Index: solenv/inc/tg_jar.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/tg_jar.mk,v
retrieving revision 1.11
diff -u -r1.11 tg_jar.mk
--- solenv/inc/tg_jar.mk	17 Dec 2003 18:06:47 -0000	1.11
+++ solenv/inc/tg_jar.mk	6 Oct 2004 17:06:08 -0000
@@ -143,6 +143,6 @@
 .IF "$(OS)$(CPU)"=="SOLARISS"
 	@+-find . -type d -user $(USER) ! -perm -5 -print | xargs test "$$1" != "" && chmod +r $$1 
 .ENDIF
-	+cd $(CLASSDIR) && zip -u -r $(@:f) $(subst,$(CLASSDIR)$/, $(JARMANIFEST)) $(subst,\,/ $(JARCLASSDIRS))
+	+cd $(CLASSDIR) && zip -u -r -o $(@:f) $(subst,$(CLASSDIR)$/, $(JARMANIFEST)) $(subst,\,/ $(JARCLASSDIRS))
 .ENDIF
 
