--- scp2/source/ooo/file_ooo.scp	21 Jan 2005 12:14:20 -0000	1.66
+++ scp2/source/ooo/file_ooo.scp	27 Jan 2005 17:36:36 -0000
@@ -371,6 +371,8 @@
     Styles = (PACKED);
 End
 
+#ifdef ENABLE_GNOMEVFS
+
 File gid_File_Bin_Gnome_Set_Default_Application
     BIN_FILE_BODY;
     Dir = gid_Dir_Program;
@@ -380,6 +382,8 @@
 
 #endif
 
+#endif
+
 File gid_File_Bin_Senddoc
     BIN_FILE_BODY;
     Dir = gid_Dir_Program;


Index: scp2/source/ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.19
diff -u -r1.19 makefile.mk
--- scp2/source/ooo/makefile.mk	21 Jan 2005 12:14:37 -0000	1.19
+++ scp2/source/ooo/makefile.mk	27 Jan 2005 17:37:34 -0000
@@ -151,6 +166,10 @@
         module_ooo.par             \
         profile_ooo.par            \
         profileitem_ooo.par
+
+.IF "$(ENABLE_GNOMEVFS)" != ""
+SCPDEFS+=-DENABLE_GNOMEVFS
+.ENDIF
 
 .IF "$(GUI)"=="WNT"
 PARFILES +=                        \
