Index: instsetoo/util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo/util/makefile.mk,v
retrieving revision 1.18.2.1
diff -u -3 -p -r1.18.2.1 makefile.mk
--- instsetoo/util/makefile.mk	12 Aug 2003 09:48:01 -0000	1.18.2.1
+++ instsetoo/util/makefile.mk	16 Aug 2003 22:42:56 -0000
@@ -97,16 +97,21 @@ SEARCH_DIR=r:\solenv\inst\ooo\$(OUTPATH)
 
 ALLTAR : pack
 
-.IF "$(alllangext)"!=""
+# Create installation sets for languages in instsetext, or in alllangext if
+# insetsetext is not defined in the format instsetext=<num>,[<num>,]...
+# e.g.: instsetext=01,49,99
+instsetext*=$(alllangext)
+  
+.IF "$(instsetext)"!=""
 
 .IF "$(BUILD_SPECIAL)"!=""  
 .IF "$(OOO_RELEASE_BUILD)"==""
 pack: pack_01
 .ELSE
-pack: $(foreach,i,$(alllangext) pack_$i)
+pack: $(foreach,i,$(instsetext) pack_$i)
 .ENDIF
 .ELSE
-pack: $(foreach,i,$(alllangext) pack_$i)
+pack: $(foreach,i,$(instsetext) pack_$i)
 .ENDIF
 
 pack_%:
@@ -117,10 +122,12 @@ pack_%:
 test:
 	+-$(LZIP) $(LZIPFLAGS) -p ${SEARCH_DIR} -l 01 -f openoffice.lst -o -n OfficeOSL 
 
-.ELSE			# "$(alllangext)"!=""
+.ELSE			# "$(instsetext)"!=""
 pack:
-	@+echo cannot pack nothing...
+	@+echo no installation sets selected to be packed.
+	@+echo instsetext: $(instsetext)
+	@+echo alllangext: $(alllangext)
 
-.ENDIF			# "$(alllangext)"!=""
+.ENDIF			# "$(instsetext)"!=""
 
 	
