Index: scptools/source/linker/scpzip.cxx
===================================================================
RCS file: /cvs/installation/scptools/source/linker/scpzip.cxx,v
retrieving revision 1.42
diff -u -p -u -r1.42 scpzip.cxx
--- scptools/source/linker/scpzip.cxx	4 Jun 2003 14:25:41 -0000	1.42
+++ scptools/source/linker/scpzip.cxx	8 Aug 2003 11:43:47 -0000
@@ -151,6 +151,7 @@ ByteString		aMergeFilename;
 ByteString		aHTMLFilename;
 
 BOOL			bTestOnly			= FALSE;
+BOOL			bDontPack			= FALSE;
 BOOL			bOldSetupZIPStyle	= FALSE;
 BOOL			bWebScript 			= FALSE;
 BOOL			bStripBinaries		= FALSE;
@@ -183,6 +184,7 @@ void PrintZipUsage()
 	fprintf( stderr, "usage:\n" );
 	fprintf( stderr, "\t-<1..9> compression (default is 5)\n" );
 	fprintf( stderr, "\t-o test only\n" );
+	fprintf( stderr, "\t-O don't pack files\n" );
 	fprintf( stderr, "\t-l <langcode,...> (first langcode is default language)\n" );
 	fprintf( stderr, "\t-h <INTEL | SPARC> (INTEL)\n" );
 	fprintf( stderr, "\t-t <WNT | OS2 | UNX> (default WNT)\n" );
@@ -211,6 +213,8 @@ void __EnumFilesCallBack( char *pFile, U
 
 void _rawZip( ByteString& aSource, SiDirEntry& aDestination )
 {
+	if(bDontPack) return;
+
 	ByteString aCompressionParam( '-' );
 	aCompressionParam += ByteString::CreateFromInt32( nCompression );
 
@@ -1578,6 +1582,9 @@ int __LOADONCALLAPI main( int argv, char
 						  break;
 
 				case 'o': bTestOnly = TRUE;
+						  break;
+
+				case 'O': bDontPack = TRUE;
 						  break;
 
 				case 'h': {
Index: scptools/source/linker/lzip.cxx
===================================================================
RCS file: /cvs/installation/scptools/source/linker/lzip.cxx,v
retrieving revision 1.19
diff -u -p -u -r1.19 lzip.cxx
--- scptools/source/linker/lzip.cxx	4 Apr 2003 15:04:25 -0000	1.19
+++ scptools/source/linker/lzip.cxx	8 Aug 2003 11:43:47 -0000
@@ -113,6 +113,7 @@ ByteString					m_aZipLst;
 
 BOOL 						m_bPrintCommandline = FALSE;
 BOOL 						m_bTestOnly = FALSE;
+BOOL 						m_bDontPack  = FALSE;
 BOOL 						m_bBuildWeb = FALSE;
 BOOL 						m_bStripBinaries = FALSE;
 
@@ -142,6 +143,7 @@ void PrintUsage()
 	fprintf( stderr, "\t-z <size>\n" );
 	fprintf( stderr, "\t-e <filename> (write a non HTML log)\n" );
 	fprintf( stderr, "\t-o test only mode\n" );
+	fprintf( stderr, "\t-O don't pack files\n" );
 	fprintf( stderr, "\t-c print scpzip commandline\n" );
 	fprintf( stderr, "\t-w generate a webinstall script\n" );
 //	fprintf( stderr, "\t-v <version> (write script version default is 100)\n" );
@@ -301,6 +303,10 @@ BOOL StartJob()
 	if( m_bTestOnly )
 		AddCmd( "-o" );
 
+	// === Don't pack ===================
+	if( m_bDontPack )
+		AddCmd( "-O" );
+
 	// === compression =================
 	if (pPrj->m_aCompression >= 0 && pPrj->m_aCompression <= 9)
 	{
@@ -604,6 +610,9 @@ int __LOADONCALLAPI main( int argc, char
 				break;
 
 			case 'o':	m_bTestOnly = TRUE;
+				break;
+
+			case 'O':	m_bDontPack = TRUE;
 				break;
 
 			case 'w':	m_bBuildWeb = TRUE;
