Index: instsetoo/util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo/util/makefile.mk,v
retrieving revision 1.18.2.1
diff -u -p -u -r1.18.2.1 makefile.mk
--- instsetoo/util/makefile.mk	12 Aug 2003 09:48:01 -0000	1.18.2.1
+++ instsetoo/util/makefile.mk	28 Aug 2003 11:17:36 -0000
@@ -98,7 +98,9 @@ SEARCH_DIR=r:\solenv\inst\ooo\$(OUTPATH)
 ALLTAR : pack
 
 .IF "$(alllangext)"!=""
-
+.IF "$(BUILD_MINIMAL)"!=""
+pack: $(foreach,i,$(alllangext) packnofiles_$i) re_clean_$(BUILD_MINIMAL) pack_$(BUILD_MINIMAL)
+.ELSE
 .IF "$(BUILD_SPECIAL)"!=""  
 .IF "$(OOO_RELEASE_BUILD)"==""
 pack: pack_01
@@ -108,11 +110,21 @@ pack: $(foreach,i,$(alllangext) pack_$i)
 .ELSE
 pack: $(foreach,i,$(alllangext) pack_$i)
 .ENDIF
+.ENDIF
+LZIP_FLAGS=-p ${SEARCH_DIR} -p ${SHARED_COM_SDK_PATH} $(LZIPFLAGS) -f openoffice.lst -n OfficeOSL 
 
-pack_%:
-	+-$(MKDIRHIER) $(INSTALLDIR)$/$(@:s/pack_//)$/normal
-	+-$(RM) $(INSTALLDIR)$/$(@:s/pack_//)$/normal$/*
-	+-$(LZIP) -p ${SEARCH_DIR} $(LZIPFLAGS) -l $(@:s/pack_//) -f openoffice.lst -d $(INSTALLDIR)$/$(@:s/pack_//) -n OfficeOSL 
+clean_%:
+	+-$(MKDIRHIER) $(INSTALLDIR)$/$(@:s/clean_//)$/normal
+	+-$(RM) $(INSTALLDIR)$/$(@:s/clean_//)$/normal$/*	
+re_clean_%:
+	+-$(MKDIRHIER) $(INSTALLDIR)$/$(@:s/re_clean_//)$/normal
+	+-$(RM) $(INSTALLDIR)$/$(@:s/re_clean_//)$/normal$/*
+
+pack_%: clean_%
+	+-$(LZIP) $(LZIP_FLAGS) -l $(@:s/pack_//) -d $(INSTALLDIR)$/$(@:s/pack_//) 
+
+packnofiles_%: clean_%
+	+-$(LZIP) $(LZIP_FLAGS) -O -l $(@:s/packnofiles_//) -d $(INSTALLDIR)$/$(@:s/packnofiles_//)
 
 test:
 	+-$(LZIP) $(LZIPFLAGS) -p ${SEARCH_DIR} -l 01 -f openoffice.lst -o -n OfficeOSL 
