Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.10.2.1
diff -u -p -u -r1.10.2.1 set_soenv.in
--- config_office/set_soenv.in	21 Jul 2003 14:41:34 -0000	1.10.2.1
+++ config_office/set_soenv.in	21 Jul 2003 16:55:33 -0000
@@ -2235,6 +2235,9 @@ sub CreateBuildEnv 
 #--------------------------------------------------------
 sub GetCorrectPath 
 {  my ( $compath, $perl_path, $tcsh_path, $compiler, $perl_compath, $compath_perl, $correct_path );
+
+   return '';
+
    $compath  = $_[ 0 ];
    $perl_path = $_[ 1 ];
    $tcsh_path = $_[ 2 ];
