Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.82
diff -u -p -u -r1.82 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	17 Jul 2003 13:07:14 -0000	1.82
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	5 Aug 2003 14:14:43 -0000
@@ -26020,9 +26020,9 @@
 					<label xml:lang="tr">Use system default user interface font</label>
 					<label xml:lang="th">Use system default user interface font</label>
 				</info>
-				<value>false</value>
-			</prop>
-			<prop oor:name="IsSelectionInReadonly" oor:type="xs:boolean">
+				<value>true</value>
+			</prop> <!-- patch1 -->
+			<prop oor:name="IsSelectionInReadonly" oor:type="xs:boolean">  <!-- patch2 -->
 				<info>
 					<author>OS</author>
 					<desc xml:lang="en-US">Indicates whether the cursor is displayed in read-only texts.</desc>
