Index: unotools/source/config/configmgr.cxx
===================================================================
RCS file: /cvs/util/unotools/source/config/configmgr.cxx,v
retrieving revision 1.36
diff -u -p -u -r1.36 configmgr.cxx
--- unotools/source/config/configmgr.cxx	10 Jun 2003 11:29:10 -0000	1.36
+++ unotools/source/config/configmgr.cxx	11 Aug 2003 15:10:23 -0000
@@ -59,6 +59,9 @@
  *
  ************************************************************************/
 
+#include <tools/isolang.hxx>
+#include <tools/intn.hxx>
+
 #ifndef _UTL_CONFIGMGR_HXX_
 #include "unotools/configmgr.hxx"
 #endif
@@ -437,6 +440,13 @@ Any ConfigManager::GetDirectConfigProper
     {
         aRet <<= aXMLFileFormatVersion;
         return aRet;
+    }
+
+    if ( !getenv ("OOOCONFIG_LOCALE") && (eProp == LOCALE || eProp == DEFAULTCURRENCY))
+    {
+	OUString aSysLang = ConvertLanguageToIsoString(GetSystemLanguage());
+        aRet <<= aSysLang;
+	return aRet;
     }
 
 	OUString sPath = C2U(cConfigBaseURL);
Index: desktop/source/app/ssoinit.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/ssoinit.cxx,v
retrieving revision 1.17
diff -u -p -u -r1.17 ssoinit.cxx
--- desktop/source/app/ssoinit.cxx	17 Apr 2003 13:33:29 -0000	1.17
+++ desktop/source/app/ssoinit.cxx	11 Aug 2003 15:11:11 -0000
@@ -220,6 +219,11 @@ void WriteUserNameToINI( const OUString 
 
 LanguageType GetLanguageType( void )
 {
+#ifdef UNX
+	if ( !getenv ("OOOCONFIG_LOCALE") )
+		return GetSystemUILanguage (INTERNATIONAL_SYSTEM_DEFAULT);
+#endif
+
 	LanguageType						theLanguageType		= LANGUAGE_DONTKNOW;
 	try
 	{
