--- sfx2/source/toolbox/tbxitem.cxx
+++ sfx2/source/toolbox/tbxitem.cxx
@@ -1485,10 +1485,20 @@
         aURL = sFallback;
 
 	BOOL bBig = ( SfxImageManager::GetCurrentSymbolSet() == SFX_SYMBOLS_LARGE );
-    GetToolBox().SetItemImage( GetId(),
-							   SvFileInformationManager::GetImage( INetURLObject( aURL ),
-							   bBig,
-								GetToolBox().GetBackground().GetColor().IsDark() ) );
+
+	Image aI = SvFileInformationManager::GetImage( INetURLObject( aURL ), bBig,
+												   GetToolBox().GetBackground().GetColor().IsDark() );
+
+	Size aBigSize( 24, 24 );
+	if ( bBig && aI.GetSizePixel() != aBigSize )
+	{
+		BitmapEx aScaleBmpEx( aI.GetBitmapEx() );
+		aScaleBmpEx.Scale( aBigSize, BMP_SCALE_INTERPOLATE );
+		GetToolBox().SetItemImage( GetId(), Image( aScaleBmpEx ) );
+	} 
+	else
+		GetToolBox().SetItemImage( GetId(), aI);
+
 	aLastURL = aURL;
 }
 
