Index: vcl/source/window/toolbox.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/toolbox.cxx,v
retrieving revision 1.77
diff -u -r1.77 toolbox.cxx
--- vcl/source/window/toolbox.cxx	9 Sep 2004 16:33:49 -0000	1.77
+++ vcl/source/window/toolbox.cxx	15 Oct 2004 15:54:04 -0000
@@ -3571,38 +3570,13 @@
             else
                 DrawSelectionBackground( pItem->maRect, bHighlight, pItem->meState == STATE_CHECK, TRUE, pItem->mbShowWindow ? TRUE : FALSE );
 
-            // draw shadows
-            if( bHighlight == 2 && pItem->meState != STATE_CHECK )
-            {
-                nImageOffX++;
-                nImageOffY++;
-
-				/* !!!
-				if( pImage->HasMaskBitmap() )
-                {
-                    Color aMaskCol = GetpApp()->GetSettings().GetStyleSettings().GetHighlightColor();
-                    USHORT h,s,b;
-                    aMaskCol.RGBtoHSB( h, s, b );
-                    if( s > 20 ) s=20;
-                    if( b > 5 ) b-=5;
-                    aMaskCol.SetColor( Color::HSBtoRGB( h, s, b ) );
-                    DrawMask( Point( nImageOffX, nImageOffY ), pImage->GetMaskBitmap(), aMaskCol );
-                }
-				*/
-
-                nImageOffX-=2;
-                nImageOffY-=2;
-            }
-
             if( bHighlight )
             {
                 if( bHighContrastWhite )
                     nImageStyle |= IMAGE_DRAW_COLORTRANSFORM;
             }
-            DrawImage( Point( nImageOffX, nImageOffY ), *pImage, nImageStyle );
         }
-        else
-            DrawImage( Point( nImageOffX, nImageOffY ), *pImage, nImageStyle );
+		DrawImage( Point( nImageOffX, nImageOffY ), *pImage, nImageStyle );
     }
 
     // draw the text
