From 4dfc33d1a1390290f126d8ef829e415a8b25f686 Mon Sep 17 00:00:00 2001
From: Jan Nieuwenhuizen <janneke@gnu.org>
Date: Tue, 12 Aug 2008 14:51:14 +0200
Subject: [PATCH] tg create: add --force option to resolve merge dependencies.


Signed-off-by: Jan Nieuwenhuizen <janneke@gnu.org>
---
 tg-create.sh |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/tg-create.sh b/tg-create.sh
index 6cce7ed..0c83460 100644
--- a/tg-create.sh
+++ b/tg-create.sh
@@ -14,6 +14,8 @@ name=
 while [ -n "$1" ]; do
 	arg="$1"; shift
 	case "$arg" in
+	--force)
+		force=1;;
 	-*)
 		echo "Usage: tg create NAME [DEPS...]" >&2
 		exit 1;;
@@ -82,6 +84,11 @@ while [ -n "$merge" ]; do
 	info "Merging $name base with $branch..."
 
 	if ! git merge "$branch"; then
+	    if test "$force" = "1"; then
+                mv .topdeps .topdeps~ && grep -Ev '(<|>|=){7}' .topdeps~ > .topdeps && rm .topdeps~
+                git add .topdeps .topmsg
+                git commit -am 'Auto-resolve topgit .top* merge failure.'
+	    else
 		info "Please commit merge resolution and call: tg create"
 		info "It is also safe to abort this operation using \`git reset --hard\`"
 		info "but please remember you are on the base branch now;"
@@ -89,6 +96,7 @@ while [ -n "$merge" ]; do
 		echo "$deps" >"$git_dir/top-deps"
 		echo "$merge" >"$git_dir/top-merge"
 		exit 2
+	    fi
 	fi
 done
 
-- 
1.6.0.rc0.44.g67270

