Index: desktop/source/deployment/misc/dp_resource.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/deployment/misc/dp_resource.cxx,v
retrieving revision 1.12.84.1
retrieving revision 1.12.84.1.6.1
diff -u -u -p -r1.12.84.1 -r1.12.84.1.6.1
--- desktop/source/deployment/misc/dp_resource.cxx	10 Aug 2006 16:37:01 -0000	1.12.84.1
+++ desktop/source/deployment/misc/dp_resource.cxx	20 Sep 2006 11:48:32 -0000	1.12.84.1.6.1
@@ -37,6 +37,7 @@
 #include "dp_resource.h"
 #include "osl/module.hxx"
 #include "osl/mutex.hxx"
+#include "rtl/ustring.h"
 #include "cppuhelper/implbase1.hxx"
 #include "unotools/configmgr.hxx"
 
@@ -55,6 +56,10 @@ struct OfficeLocale :
         if (! (::utl::ConfigManager::GetDirectConfigProperty(
                    ::utl::ConfigManager::LOCALE ) >>= slang))
             throw RuntimeException( OUSTR("Cannot determine language!"), 0 );
+        //fallback, the locale is currently only set when the user starts the
+        //office for the first time.
+        if (slang.getLength() == 0)
+            slang =  rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("en-US"));
         return toLocale(slang);
     }
 };
