--- boost/prj/d.lst	26 Jun 2007 12:40:14 -0000	1.14
+++ boost/prj/d.lst	22 Oct 2007 13:54:43 -0000	1.14.6.1
@@ -6,6 +6,7 @@ mkdir: %_DEST%\inc%_EXT%\boost
 ..\%__SRC%\misc\build\boost-1.30.2\boost\token_iterator.hpp %_DEST%\inc%_EXT%\boost\token_iterator.hpp
 ..\%__SRC%\misc\build\boost-1.30.2\boost\token_functions.hpp %_DEST%\inc%_EXT%\boost\token_functions.hpp
 ..\%__SRC%\misc\build\boost-1.30.2\boost\enable_shared_from_this.hpp %_DEST%\inc%_EXT%\boost\enable_shared_from_this.hpp
+..\%__SRC%\misc\build\boost-1.30.2\boost\array.hpp %_DEST%\inc%_EXT%\boost\array.hpp
 
 ..\%__SRC%\misc\build\boost-1.30.2\boost\cast.hpp %_DEST%\inc%_EXT%\boost\cast.hpp
 ..\%__SRC%\misc\build\boost-1.30.2\boost\enable_shared_from_this.hpp %_DEST%\inc%_EXT%\boost\enable_shared_from_this.hpp
cvs diff: Diffing boost/spirit
