--- writerfilter/inc/doctok/WW8Document.hxx	2007-12-03 10:29:49.000000000 +0100
+++ writerfilter/inc/doctok/WW8Document.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -36,7 +36,14 @@
 #ifndef INCLUDED_WW8_DOCUMENT_HXX
 #define INCLUDED_WW8_DOCUMENT_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <boost/shared_ptr.hpp>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #ifndef _SAL_TYPES_H_
 #include <sal/types.h>
 #endif // _SAL_TYPES_H_
--- writerfilter/inc/odiapi/core/Node.hxx	2006-11-01 10:14:27.000000000 +0100
+++ writerfilter/inc/odiapi/core/Node.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -46,9 +46,15 @@
 #include <odiapi/props/Properties.hxx>
 #endif
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <memory>
 #include <string>
 #include <boost/shared_ptr.hpp>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace odiapi { namespace core {
 
--- writerfilter/inc/odiapi/props/Logger.hxx	2006-11-01 10:14:27.000000000 +0100
+++ writerfilter/inc/odiapi/props/Logger.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -38,7 +38,13 @@
 #ifndef INCLUDED_LOGGER_HXX
 #define INCLUDED_LOGGER_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace util {
 
--- writerfilter/inc/odiapi/props/Properties.hxx	2006-11-01 10:14:27.000000000 +0100
+++ writerfilter/inc/odiapi/props/Properties.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -46,8 +46,14 @@
 
 #include "Iterator.hxx"
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <boost/shared_ptr.hpp>
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace util { class Logger; }
 
--- writerfilter/inc/odiapi/xxml/ContentHandler.hxx	2006-11-01 10:14:28.000000000 +0100
+++ writerfilter/inc/odiapi/xxml/ContentHandler.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -40,7 +40,14 @@
 #include <odiapi/qname/QName.hxx>
 #include <WriterFilterDllApi.hxx>
 #include <rtl/ustring.hxx>
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 
 namespace writerfilter { namespace xxml {
--- writerfilter/inc/resourcemodel/exceptions.hxx	2007-12-03 10:30:39.000000000 +0100
+++ writerfilter/inc/resourcemodel/exceptions.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -36,7 +36,14 @@
 #ifndef _DOCTOK_EXCEPTIONS_HXX
 #define _DOCTOK_EXCEPTIONS_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <WriterFilterDllApi.hxx>
 
 namespace writerfilter {
--- writerfilter/inc/resourcemodel/OutputWithDepth.hxx	2007-11-14 13:51:05.000000000 +0100
+++ writerfilter/inc/resourcemodel/OutputWithDepth.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -36,8 +36,14 @@
 #ifndef INCLUDED_OUTPUT_WITH_DEPTH
 #define INCLUDED_OUTPUT_WITH_DEPTH
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <vector>
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace writerfilter
 {
--- writerfilter/inc/resourcemodel/QNameToString.hxx	2007-12-12 08:53:42.000000000 +0100
+++ writerfilter/inc/resourcemodel/QNameToString.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -35,10 +35,17 @@
 #ifndef INCLUDED_QNAME_TO_STRING_HXX
 #define INCLUDED_QNAME_TO_STRING_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <boost/shared_ptr.hpp>
 #include <map>
 #include <string>
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <resourcemodel/WW8ResourceModel.hxx>
 #include <com/sun/star/beans/XPropertySet.hpp>
 
--- writerfilter/inc/resourcemodel/SubSequence.hxx	2007-12-03 10:30:28.000000000 +0100
+++ writerfilter/inc/resourcemodel/SubSequence.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -40,8 +40,15 @@
 #include <com/sun/star/uno/Sequence.hxx>
 #endif
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <boost/shared_ptr.hpp>
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <ctype.h>
 #include "exceptions.hxx"
 #include <WriterFilterDllApi.hxx>
--- writerfilter/inc/resourcemodel/WW8ResourceModel.hxx	2007-12-12 09:04:17.000000000 +0100
+++ writerfilter/inc/resourcemodel/WW8ResourceModel.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -36,10 +36,16 @@
 #ifndef INCLUDED_WW8_EVENT_HANDLER_HXX
 #define INCLUDED_WW8_EVENT_HANDLER_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
-#include <memory>
 #include <boost/shared_ptr.hpp>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
+#include <memory>
 #ifndef _SAL_TYPES_H_
 #include <sal/types.h>
 #endif // _SAL_TYPES_H_
--- writerfilter/qa/cppunittests/odiapi/ExternalViewLogger.cxx	2006-10-24 11:36:56.000000000 +0200
+++ writerfilter/qa/cppunittests/odiapi/ExternalViewLogger.cxx	2008-01-14 11:58:59.000000000 +0100
@@ -1,7 +1,15 @@
 /*  Copyright 2005 Sun Microsystems, Inc. */
 
 #include "ExternalViewLogger.hxx"
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <boost/assert.hpp>
 #include <stdio.h>
 
--- writerfilter/qa/cppunittests/odiapi/FileLoggerImpl.cxx	2006-10-24 11:36:57.000000000 +0200
+++ writerfilter/qa/cppunittests/odiapi/FileLoggerImpl.cxx	2008-01-14 11:58:59.000000000 +0100
@@ -1,7 +1,14 @@
 /*  Copyright 2005 Sun Microsystems, Inc. */
 
 #include "FileLoggerImpl.hxx"
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 using namespace std;
 
--- writerfilter/source/doctok/resources.xsl	2007-11-14 13:57:49.000000000 +0100
+++ writerfilter/source/doctok/resources.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -84,8 +84,14 @@
 #ifndef INCLUDED_RESOURCES_HXX
 #define INCLUDED_RESOURCES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #ifndef INCLUDED_OUTPUT_WITH_DEPTH_HXX
 #include &lt;resourcemodel/OutputWithDepth.hxx&gt;
--- writerfilter/source/doctok/util.hxx	2007-11-14 13:57:49.000000000 +0100
+++ writerfilter/source/doctok/util.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -36,8 +36,14 @@
 #ifndef INCLUDED_UTIL_HXX
 #define INCLUDED_UTIL_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace writerfilter {
 namespace doctok {
--- writerfilter/source/doctok/WW8BinTable.hxx	2007-11-14 13:57:46.000000000 +0100
+++ writerfilter/source/doctok/WW8BinTable.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -36,8 +36,14 @@
 #ifndef INCLUDED_WW8_BIN_TABLE_HXX
 #define INCLUDED_WW8_BIN_TABLE_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <boost/shared_ptr.hpp>
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #ifndef INCLUDED_WW8_FKP_HXX
 #include <WW8FKP.hxx>
--- writerfilter/source/doctok/WW8CpAndFc.cxx	2007-11-14 13:57:46.000000000 +0100
+++ writerfilter/source/doctok/WW8CpAndFc.cxx	2008-01-14 11:58:59.000000000 +0100
@@ -35,10 +35,16 @@
 
 #include <WW8CpAndFc.hxx>
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <iterator>
 #include <algorithm>
 #include <string>
 #include <map>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace writerfilter {
 namespace doctok
--- writerfilter/source/doctok/WW8OutputWithDepth.hxx	2007-11-14 13:57:46.000000000 +0100
+++ writerfilter/source/doctok/WW8OutputWithDepth.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -40,7 +40,13 @@
 #include <resourcemodel/OutputWithDepth.hxx>
 #endif
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 using namespace ::std;
 
--- writerfilter/source/odiapi/core/NodeImpl.hxx	2006-11-01 10:14:35.000000000 +0100
+++ writerfilter/source/odiapi/core/NodeImpl.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -40,8 +40,14 @@
 
 #include <odiapi/core/Node.hxx>
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
 #include <boost/shared_ptr.hpp>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace odiapi { namespace core {
 
--- writerfilter/source/odiapi/qname/QName.cxx	2006-11-01 10:14:35.000000000 +0100
+++ writerfilter/source/odiapi/qname/QName.cxx	2008-01-14 11:58:59.000000000 +0100
@@ -42,7 +42,15 @@
 #include <stack>
 #include <assert.h>
 #include <stdexcept>
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <stdio.h>
 
 namespace writerfilter
--- writerfilter/source/ooxml/fastresources_dml-baseStylesheet.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-baseStylesheet.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_BASESTYLESHEET_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_BASESTYLESHEET_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-baseTypes.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-baseTypes.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_BASETYPES_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_BASETYPES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-chartDrawing.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-chartDrawing.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_CHARTDRAWING_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_CHARTDRAWING_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-documentProperties.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-documentProperties.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_DOCUMENTPROPERTIES_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_DOCUMENTPROPERTIES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-graphicalObject.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-graphicalObject.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_GRAPHICALOBJECT_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_GRAPHICALOBJECT_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shape3DCamera.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shape3DCamera.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DCAMERA_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DCAMERA_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shape3DLighting.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shape3DLighting.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DLIGHTING_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DLIGHTING_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shape3DScene.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shape3DScene.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DSCENE_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DSCENE_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shape3DStyles.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shape3DStyles.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DSTYLES_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPE3DSTYLES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shapeEffects.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shapeEffects.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPEEFFECTS_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPEEFFECTS_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shapeGeometry.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shapeGeometry.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPEGEOMETRY_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPEGEOMETRY_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shapeLineProperties.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shapeLineProperties.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPELINEPROPERTIES_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPELINEPROPERTIES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-shapeProperties.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-shapeProperties.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_SHAPEPROPERTIES_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_SHAPEPROPERTIES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-styleDefaults.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-styleDefaults.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_STYLEDEFAULTS_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_STYLEDEFAULTS_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-stylesheet.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-stylesheet.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_STYLESHEET_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_STYLESHEET_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-textCharacter.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-textCharacter.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_TEXTCHARACTER_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_TEXTCHARACTER_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_dml-wordprocessingDrawing.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_dml-wordprocessingDrawing.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_DML_WORDPROCESSINGDRAWING_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_DML_WORDPROCESSINGDRAWING_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresourcesimpl.xsl	2007-11-14 13:59:00.000000000 +0100
+++ writerfilter/source/ooxml/fastresourcesimpl.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -75,7 +75,14 @@
     <out xml:space="preserve">
       <xsl:call-template name="licenseheader"/>
       <xsl:text>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include &lt;doctok/sprmids.hxx&gt;
 #include "OOXMLFastHelper.hxx"
 #include "OOXMLfastresources.hxx"
--- writerfilter/source/ooxml/fastresources_shared-math.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_shared-math.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_SHARED_MATH_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_SHARED_MATH_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_shared-relationshipReference.xsl	2007-11-14 13:59:00.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_shared-relationshipReference.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_SHARED_RELATIONSHIPREFERENCE_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_SHARED_RELATIONSHIPREFERENCE_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_sml-customXmlMappings.xsl	2007-11-14 13:59:00.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_sml-customXmlMappings.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_SML_CUSTOMXMLMAPPINGS_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_SML_CUSTOMXMLMAPPINGS_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
diff -u -r writerfilter/source/ooxml/fastresources_wml.xsl writerfilter/source/ooxml/fastresources_wml.xsl
--- writerfilter/source/ooxml/fastresources_wml.xsl	2007-11-14 13:59:00.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_wml.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_WML_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_WML_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources_.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources_.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_FASTRESOURCES_WML_HXX
 #define INCLUDED_OOXML_FASTRESOURCES_WML_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include "OOXMLFastContextHandler.hxx"
--- writerfilter/source/ooxml/fastresources.xsl	2007-11-14 13:58:59.000000000 +0100
+++ writerfilter/source/ooxml/fastresources.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -77,9 +77,15 @@
 #ifndef INCLUDED_OOXML_RESOURCES_HXX
 #define INCLUDED_OOXML_RESOURCES_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 #include &lt;doctok/resourceids.hxx&gt;
--- writerfilter/source/ooxml/fasttokens.xsl	2007-11-21 12:42:09.000000000 +0100
+++ writerfilter/source/ooxml/fasttokens.xsl	2008-01-14 11:58:59.000000000 +0100
@@ -78,10 +78,16 @@
 #ifndef INCLUDED_OOXML_FAST_TOKENS_HXX
 #define INCLUDED_OOXML_FAST_TOKENS_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include &lt;map&gt;
 #include &lt;vector&gt;
 #include &lt;boost/shared_ptr.hpp&gt;
 #include &lt;string&gt;
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 #include &lt;resourcemodel/WW8ResourceModel.hxx&gt;
 
--- writerfilter/source/ooxml/OOXMLFastContextHandler.hxx	2007-11-27 11:52:05.000000000 +0100
+++ writerfilter/source/ooxml/OOXMLFastContextHandler.hxx	2008-01-14 11:58:59.000000000 +0100
@@ -40,7 +40,14 @@
 #include <com/sun/star/xml/sax/XFastShapeContextHandler.hpp>
 #endif
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include "sal/config.h"
 #include "com/sun/star/uno/XComponentContext.hpp"
 #include "cppuhelper/implbase1.hxx"
--- writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx	2007-06-19 07:39:13.000000000 +0200
+++ writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx	2008-01-14 11:58:59.000000000 +0100
@@ -54,7 +54,15 @@
 #include <rtl/string.hxx>
 #include <hash_set>
 #include <assert.h>
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <cppuhelper/implbase2.hxx>
 #include <com/sun/star/embed/XTransactedObject.hpp>
 #include <com/sun/star/embed/XStorage.hpp>
@@ -66,7 +74,14 @@
 #include <comphelper/seqstream.hxx>
 
 #include <ctype.h>
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 using namespace ::com::sun::star;
 using namespace ::std;
