--- /dev/null	
+++ oox/inc/oox/xls/extrefbuffer.hxx	
@@ -0,0 +1,111 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: extrefbuffer.hxx,v $
+ *
+ *  $Revision: 1.1.2.3 $
+ *
+ *  last change: $Author: kohei $ $Date: 2007/07/09 22:30:02 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef OOX_XLS_EXTREFBUFFER_HXX
+#define OOX_XLS_EXTREFBUFFER_HXX
+
+#include "oox/xls/globaldatahelper.hxx"
+#include "oox/core/containerhelper.hxx"
+
+#include <com/sun/star/table/CellAddress.hpp>
+#include <map>
+#include <list>
+
+namespace oox { namespace core {
+    class AttributeList;
+} }
+
+namespace oox {
+namespace xls {
+
+// ============================================================================
+
+/** Stores a single external reference information. */
+struct ExtRef
+{
+    ::rtl::OUString     maPath;
+    ::rtl::OUString     maSheetName;
+
+    explicit            ExtRef();
+};
+
+struct ExtRefCell
+{
+    ::com::sun::star::table::CellAddress    maCellAddress;
+    ::rtl::OUString                         maFormulaString;
+};
+
+// ============================================================================
+
+class ExtRefBuffer : public GlobalDataHelper
+{
+public:
+    explicit            ExtRefBuffer( const GlobalDataHelper& rGlobalData );
+
+    void                setExternalLink( sal_Int32 nId, const ::rtl::OUString& aPath );
+
+    /** Get an external link data by ID.  The ID is given in the Excel formula
+        string as in '[1]Sheet1!A1'. */
+    const ExtRef*       getExternalLinkById( sal_Int32 nId ) const;
+
+    /** Check if a given Excel formula string is an external reference. */
+    bool                isExtRef( const ::rtl::OUString& aCellFormula ) const;
+
+    void                addToDestCellList( const ::com::sun::star::table::CellAddress& aAddr, const ::rtl::OUString& aFormula );
+
+    void                finalizeImport();
+
+private:
+    void                setExtSheets();
+
+    void                setExtRefCells();
+
+    void                expandFormula( const ::rtl::OUString& aFormula, ::rtl::OUString& rExpanded );
+
+private:
+    typedef ::std::map<sal_Int32, ExtRef>   ExtRefMap;
+    typedef ::std::list<ExtRefCell>         ExtRefCellList;
+
+    ExtRefMap           maExtRefs;
+    ExtRefCellList      maDestCellList;
+};
+
+// ============================================================================
+
+} // namespace xls
+} // namespace oox
+
+#endif
+
