--- oox/source/token/makefile.mk.old	2008-02-29 09:16:57.000000000 +0100
+++ oox/source/token/makefile.mk	2008-03-17 18:05:51.000000000 +0100
@@ -54,10 +54,11 @@ SLOFILES =	\
 
 .INCLUDE :  target.mk
 
-$(INCCOM)$/tokens.hxx $(INCCOM)$/tokenwords.inc $(MISC)$/tokens.gperf : tokens.txt gentoken.pl
+$(INCCOM)$/tokens.hxx $(INCCOM)$/tokenwords.inc $(MISC)$/tokens.gperf $(MISC)$/tokens.gperf.stamp : tokens.txt gentoken.pl
 		$(PERL) gentoken.pl tokens.txt $(INCCOM)$/tokens.hxx $(INCCOM)$/tokenwords.inc $(MISC)$/tokens.gperf
+		touch $(MISC)$/tokens.gperf.stamp
 
-$(INCCOM)$/tokens.inc : $(MISC)$/tokens.gperf
+$(INCCOM)$/tokens.inc : $(MISC)$/tokens.gperf.stamp
 		gperf --compare-strncmp $(MISC)$/tokens.gperf | $(SED) -e "s/(char\*)0/(char\*)0, 0/g" >$(INCCOM)$/tokens.inc
 
 $(SLO)$/tokenmap.obj : $(INCCOM)$/tokens.inc $(INCCOM)$/tokenwords.inc $(INCCOM)$/tokens.hxx
