--- solenv/inc/libs.mk	31 Jul 2007 13:06:06 -0000	1.117
+++ solenv/inc/libs.mk	4 Oct 2007 11:09:43 -0000	1.98.36.9
@@ -178,6 +178,7 @@ SO2LIB=-lso$(OFFICEUPD)$(DLLPOSTFIX)
 TKTLIB=-ltkt$(OFFICEUPD)$(DLLPOSTFIX)
 SJLIB=-lj$(OFFICEUPD)$(DLLPOSTFIX)_g
 GOODIESLIB=-lgo$(OFFICEUPD)$(DLLPOSTFIX)
+SAXLIB=-lsax$(OFFICEUPD)$(DLLPOSTFIX)
 MAILLIB=-lmail
 DOCMGRLIB=-ldmg$(OFFICEUPD)$(DLLPOSTFIX)
 BASICLIB=-lsb$(OFFICEUPD)$(DLLPOSTFIX)
@@ -396,6 +397,7 @@ AUTOMATIONLIB=ists.lib
 SVLLIB=isvl.lib
 PLUGAPPLIB=plugapp.lib
 GOODIESLIB=igo.lib
+SAXLIB=isax.lib
 MAILLIB=mail.lib
 DOCMGRLIB=docmgr.lib
 BASICLIB=basic.lib
--- solenv/inc/rules.mk	21 Sep 2007 09:17:06 -0000	1.90
+++ solenv/inc/rules.mk	4 Oct 2007 11:10:12 -0000	1.65.54.9
@@ -79,6 +79,27 @@ $(OBJ)$/%.obj : %.cpp
 .ENDIF
 .ENDIF
 
+$(OBJ)$/%.obj : %.cc
+	@echo ------------------------------
+	@echo Making: $@
+.IF "$(GUI)"=="UNX"
+	@$(RM) $@ $(@:s/.obj/.o/)
+	$(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSCXXOBJ) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $(CFLAGSINCXX)$(PWD)$/$*.cc
+.IF "$(OS)"=="SOLARIS" && "$(product)"=="full" && "$(debug)"==""    
+    $(ADJUSTVISIBILITY) -p $(@:s/.obj/.o/)
+.ENDIF          # "$(OS)"=="SOLARIS" && "$(product)"=="full" && "$(debug)"==""    
+	+$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+.ELSE
+	@+-$(RM) $@ >& $(NULLDEV)
+	@+$(TYPE) $(mktmp $(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(OBJ)$/$*.obj $(CFLAGSINCXX)$(PWD)$/$*.cc )
+	@+$(ECHONL)
+.IF "$(COM)"=="GCC"
+	$(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(OBJ)$/$*.obj $(CFLAGSINCXX)$(PWD)$/$*.cc 
+.ELSE
+	$(CXX) @$(mktmp $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(OBJ)$/$*.obj $(CFLAGSINCXX)$(PWD)$/$*.cc )
+.ENDIF
+.ENDIF
+
 .IF "$(ENABLE_PCH)"!=""
 $(SLO)$/precompiled.% .PHONY:
 	-$(MKDIRHIER) $(SLO)$/pch
@@ -194,6 +215,26 @@ $(SLO)$/%.obj : $(MISC)$/%.cxx
 .ENDIF
 .ENDIF
 
+$(SLO)$/%.obj : $(MISC)$/%.cc
+	@echo ------------------------------
+	@echo Making: $@
+.IF "$(GUI)"=="UNX"
+.IF "$(TEST)"!=""
+	$(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSCXXSLO) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) -E  $(CFLAGSINCXX)$(MISC)$/$*.cc
+.ELSE
+	@$(RM) $@ $(@:s/.obj/.o/)
+	$(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSCXXSLO) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(SLO)$/$*.o $(CFLAGSINCXX)$(MISC)$/$*.cc
+.IF "$(OS)"=="SOLARIS" && "$(product)"=="full" && "$(debug)"==""    
+    $(ADJUSTVISIBILITY) -p $(@:s/.obj/.o/)
+.ENDIF          # "$(OS)"=="SOLARIS" && "$(product)"=="full" && "$(debug)"==""    
+	+$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+.ENDIF
+.ELSE
+	@+$(TYPE) $(mktmp $(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(SLO)$/$*.obj $(CFLAGSINCXX)$(MISC)$/$*.cc )
+	@+$(ECHONL)
+	$(CXX) @$(mktmp $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(SLO)$/$*.obj $(CFLAGSINCXX)$(MISC)$/$*.cc )
+.ENDIF
+
 $(OBJ)$/%.obj : $(MISC)$/%.cxx
 	@echo ------------------------------
 	@echo Making: $@
