--- writerfilter/source/resourcemodel/util.cxx	2008-01-23 14:48:37+0000	1.1.2.3
+++ writerfilter/source/resourcemodel/util.cxx	2008-01-30 08:55:01+0000	1.1.2.4
@@ -175,7 +175,7 @@
                         sal_Int32 nValue = 0;
                         aAny >>= nValue;
                         
-                        snprintf(buffer, sizeof(buffer), "%ld", nValue);
+                        snprintf(buffer, sizeof(buffer), "%" SAL_PRIdINT32, nValue);
                         sResult += buffer;
                     } 
                     else if (sPropType == 
@@ -260,7 +260,7 @@
                         awt::Point aPoint;
                         aAny >>= aPoint;
 
-                        snprintf(buffer, sizeof(buffer), "(%ld, %ld)", aPoint.X,
+                        snprintf(buffer, sizeof(buffer), "(%" SAL_PRIdINT32 ", %" SAL_PRIdINT32 ")", aPoint.X,
                                  aPoint.Y);
 
                         sResult += buffer;
@@ -273,7 +273,7 @@
                         awt::Rectangle aRect;
                         aAny >>= aRect;
 
-                        snprintf(buffer, sizeof(buffer), "(%ld, %ld, %ld, %ld)",
+                        snprintf(buffer, sizeof(buffer), "(%" SAL_PRIdINT32 ", %" SAL_PRIdINT32 ", %" SAL_PRIdINT32 ", %" SAL_PRIdINT32 ")",
                                  aRect.X, aRect.Y, aRect.Width, aRect.Height);
                         sResult += buffer;                        
                     }
