--- writerfilter/source/dmapper/DomainMapper.cxx	12 Dec 2007 10:54:51 -0000	1.61.12.19
+++ writerfilter/source/dmapper/DomainMapper.cxx	15 Jan 2008 16:22:28 -0000
@@ -567,7 +567,8 @@
                 bool bParaStyle = (pEntry->nStyleTypeCode == STYLE_TYPE_PARA);
                 if(bParaStyle)
                     m_pImpl->SetCurrentParaStyleId(::rtl::OUString::valueOf(static_cast<sal_Int32>(nIntValue), 16));
-                m_pImpl->GetTopContext()->Insert(
+				if (m_pImpl->GetTopContext())
+	                m_pImpl->GetTopContext()->Insert(
                                                  bParaStyle ?
                                                  PROP_PARA_STYLE_NAME  : PROP_CHAR_STYLE_NAME,
                                                  true, 
@@ -1566,13 +1567,19 @@
                 switch ((nIntValue & 0x0000FF00) >> 8)
                 {
                 case 1: // vertical text
-                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION, true, uno::makeAny ( sal_Int16(900) ));
-                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION_IS_FIT_TO_LINE, true, uno::makeAny (((nIntValue & 0x00FF0000) >> 16) != 0));
+					if (m_pImpl->GetTopContext())
+					{
+	                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION, true, uno::makeAny ( sal_Int16(900) ));
+    	                m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION_IS_FIT_TO_LINE, true, uno::makeAny (((nIntValue & 0x00FF0000) >> 16) != 0));
+					}
                     break;
                 case 2: // two lines in one
-                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_IS_ON, true, uno::makeAny ( true ));
-                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_PREFIX, true, uno::makeAny ( getBracketStringFromEnum((nIntValue & 0x00FF0000) >> 16)));
-                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_SUFFIX, true, uno::makeAny ( getBracketStringFromEnum((nIntValue & 0x00FF0000) >> 16, false)));
+					if (m_pImpl->GetTopContext())
+					{
+	                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_IS_ON, true, uno::makeAny ( true ));
+	                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_PREFIX, true, uno::makeAny ( getBracketStringFromEnum((nIntValue & 0x00FF0000) >> 16)));
+	                    m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_SUFFIX, true, uno::makeAny ( getBracketStringFromEnum((nIntValue & 0x00FF0000) >> 16, false)));
+					}
                     break;
                 default:
                     break;
@@ -1586,11 +1593,12 @@
         break;
         case NS_rtf::LN_FONT: //font of footnote symbol
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->SetFootnoteFontId( nIntValue );
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->SetFootnoteFontId( nIntValue );
         break;
         case NS_ooxml::LN_CT_Sym_char:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-        if( m_pImpl->GetTopContext()->GetFootnote().is())
+        if( m_pImpl->GetTopContext() && m_pImpl->GetTopContext()->GetFootnote().is())
         {
             m_pImpl->GetTopContext()->GetFootnote()->setLabel(::rtl::OUString( sal_Unicode(nIntValue)));
             break;
@@ -1602,12 +1610,13 @@
         break;
         case NS_rtf::LN_CHAR: //footnote symbol character
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->SetFootnoteSymbol( sal_Unicode(nIntValue));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->SetFootnoteSymbol( sal_Unicode(nIntValue));
         break;
         case NS_ooxml::LN_CT_Sym_font:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
             //the footnote symbol and font are provided after the footnote is already inserted
-        if( m_pImpl->GetTopContext()->GetFootnote().is())
+        if( m_pImpl->GetTopContext() && m_pImpl->GetTopContext()->GetFootnote().is())
         {
             uno::Reference< beans::XPropertySet > xAnchorProps( m_pImpl->GetTopContext()->GetFootnote()->getAnchor(), uno::UNO_QUERY );
             xAnchorProps->setPropertyValue( 
@@ -1615,7 +1624,8 @@
                 uno::makeAny( sStringValue ));
         }
         else //a real symbol
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME, true, uno::makeAny( sStringValue ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME, true, uno::makeAny( sStringValue ));
         break;
         case NS_ooxml::LN_CT_Underline_val:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
@@ -1623,12 +1633,16 @@
             break;
         case NS_ooxml::LN_CT_Color_val:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_COLOR, true, uno::makeAny( nIntValue ) );
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_COLOR, true, uno::makeAny( nIntValue ) );
             break;
         case NS_ooxml::LN_CT_Underline_color:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_UNDERLINE_HAS_COLOR, true, uno::makeAny( true ) );
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_UNDERLINE_COLOR, true, uno::makeAny( nIntValue ) );
+			if (m_pImpl->GetTopContext())
+			{
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_UNDERLINE_HAS_COLOR, true, uno::makeAny( true ) );
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_UNDERLINE_COLOR, true, uno::makeAny( nIntValue ) );
+			}
             break;
 
         case NS_ooxml::LN_CT_TabStop_val:
@@ -1652,11 +1666,13 @@
 
         case NS_ooxml::LN_CT_Fonts_ascii:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME, true, uno::makeAny( sStringValue ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME, true, uno::makeAny( sStringValue ));
             break;
         case NS_ooxml::LN_CT_Fonts_asciiTheme:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME, true, uno::makeAny( m_pImpl->GetThemeTable()->getFontNameForTheme(nIntValue) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME, true, uno::makeAny( m_pImpl->GetThemeTable()->getFontNameForTheme(nIntValue) ));
             break;
         case NS_ooxml::LN_CT_Fonts_hAnsi: 
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
@@ -1666,30 +1682,36 @@
             break; //unsupported
         case NS_ooxml::LN_CT_Fonts_eastAsia:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_ASIAN, true, uno::makeAny( sStringValue ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_ASIAN, true, uno::makeAny( sStringValue ));
             break;
     case NS_ooxml::LN_CT_Fonts_eastAsiaTheme:
         /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-        m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_COMPLEX, true, uno::makeAny( m_pImpl->GetThemeTable()->getFontNameForTheme(nIntValue) ) );
+		if (m_pImpl->GetTopContext())
+        	m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_COMPLEX, true, uno::makeAny( m_pImpl->GetThemeTable()->getFontNameForTheme(nIntValue) ) );
         break;
         case NS_ooxml::LN_CT_Fonts_cs:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_COMPLEX, true, uno::makeAny( sStringValue ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_COMPLEX, true, uno::makeAny( sStringValue ));
             break;
         case NS_ooxml::LN_CT_Fonts_cstheme:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_COMPLEX, true, uno::makeAny( m_pImpl->GetThemeTable()->getFontNameForTheme(nIntValue) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_FONT_NAME_COMPLEX, true, uno::makeAny( m_pImpl->GetThemeTable()->getFontNameForTheme(nIntValue) ));
         break;
         case NS_ooxml::LN_CT_Spacing_before:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_PARA_TOP_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100( nIntValue ) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_PARA_TOP_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100( nIntValue ) ));
             break;
         case NS_ooxml::LN_CT_Spacing_beforeLines:
             /* WRITERFILTERSTATUS: done: 0, planned: 0.5, spent: 0 */
             break;
         case NS_ooxml::LN_CT_Spacing_after:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_PARA_BOTTOM_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100( nIntValue ) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_PARA_BOTTOM_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100( nIntValue ) ));
             break;
         case NS_ooxml::LN_CT_Spacing_afterLines:
             /* WRITERFILTERSTATUS: done: 0, planned: 0.5, spent: 0 */
@@ -1740,23 +1762,27 @@
         break;
         case NS_ooxml::LN_CT_Ind_left:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(
-                PROP_PARA_LEFT_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100(nIntValue ) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(
+    	            PROP_PARA_LEFT_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100(nIntValue ) ));
             break;
         case NS_ooxml::LN_CT_Ind_right:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(
-                PROP_PARA_RIGHT_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100(nIntValue ) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(
+    	            PROP_PARA_RIGHT_MARGIN, true, uno::makeAny( ConversionHelper::convertTwipToMM100(nIntValue ) ));
             break;
         case NS_ooxml::LN_CT_Ind_hanging:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(
-                PROP_PARA_FIRST_LINE_INDENT, true, uno::makeAny( - ConversionHelper::convertTwipToMM100(nIntValue ) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(
+    	            PROP_PARA_FIRST_LINE_INDENT, true, uno::makeAny( - ConversionHelper::convertTwipToMM100(nIntValue ) ));
             break;
         case NS_ooxml::LN_CT_Ind_firstLine:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(
-                PROP_PARA_FIRST_LINE_INDENT, true, uno::makeAny( ConversionHelper::convertTwipToMM100(nIntValue ) ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(
+    	            PROP_PARA_FIRST_LINE_INDENT, true, uno::makeAny( ConversionHelper::convertTwipToMM100(nIntValue ) ));
             break;
 
         case NS_ooxml::LN_CT_EastAsianLayout_id:
@@ -1764,10 +1790,12 @@
             break;
         case NS_ooxml::LN_CT_EastAsianLayout_combine:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_IS_ON, true, uno::makeAny ( nIntValue ? true : false ));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_COMBINE_IS_ON, true, uno::makeAny ( nIntValue ? true : false ));
             break;
         case NS_ooxml::LN_CT_EastAsianLayout_combineBrackets:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
+			if (m_pImpl->GetTopContext())
             {
                 rtl::OUString sCombinePrefix = getBracketStringFromEnum(nIntValue);
                 rtl::OUString sCombineSuffix = getBracketStringFromEnum(nIntValue, false);
@@ -1777,6 +1805,7 @@
             break;
         case NS_ooxml::LN_CT_EastAsianLayout_vert:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
+			if (m_pImpl->GetTopContext())
             {
                 sal_Int16 nRotationAngle = (nIntValue ? 900 : 0);
                 m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION, true, uno::makeAny ( nRotationAngle ));
@@ -1784,7 +1813,8 @@
             break;
         case NS_ooxml::LN_CT_EastAsianLayout_vertCompress:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
-            m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION_IS_FIT_TO_LINE, true, uno::makeAny ( nIntValue ? true : false));
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(PROP_CHAR_ROTATION_IS_FIT_TO_LINE, true, uno::makeAny ( nIntValue ? true : false));
             break;
 
         case NS_ooxml::LN_CT_PageSz_code:
@@ -1866,6 +1896,7 @@
 
         case NS_ooxml::LN_EG_RPrBase_rStyle:
             /* WRITERFILTERSTATUS: done: 100, planned: 0.5, spent: 0 */
+			if (m_pImpl->GetTopContext())
             m_pImpl->GetTopContext()->Insert( PROP_CHAR_STYLE_NAME, true, uno::makeAny( m_pImpl->GetStyleSheetTable()->ConvertStyleName( sStringValue )));
             break;
         case NS_ooxml::LN_CT_Language_val: //90314
@@ -1877,7 +1908,8 @@
         {
             LanguageType eLang = MsLangId::convertIsoStringToLanguage( sStringValue );
             lang::Locale aLocale = MsLangId::convertLanguageToLocale( eLang );
-            m_pImpl->GetTopContext()->Insert(NS_ooxml::LN_CT_Language_val== nName ? PROP_CHAR_LOCALE :
+			if (m_pImpl->GetTopContext())
+	            m_pImpl->GetTopContext()->Insert(NS_ooxml::LN_CT_Language_val== nName ? PROP_CHAR_LOCALE :
                              NS_ooxml::LN_CT_Language_eastAsia == nName ? PROP_CHAR_LOCALE_ASIAN : PROP_CHAR_LOCALE_COMPLEX,
                              true, 
                              uno::makeAny( aLocale ) );
@@ -3588,7 +3620,8 @@
         m_pImpl->SetCurrentParaStyleId( sStringValue );
         StyleSheetTablePtr pStyleTable = m_pImpl->GetStyleSheetTable();
         const ::rtl::OUString sConvertedStyleName = pStyleTable->ConvertStyleName( sStringValue, true );
-        m_pImpl->GetTopContext()->Insert( PROP_PARA_STYLE_NAME, true, uno::makeAny( sConvertedStyleName ));
+		if (m_pImpl->GetTopContext())
+	        m_pImpl->GetTopContext()->Insert( PROP_PARA_STYLE_NAME, true, uno::makeAny( sConvertedStyleName ));
         const StyleSheetEntry* pEntry = pStyleTable->FindStyleSheetByISTD(sStringValue);
         //apply numbering to paragraph if it was set at the style
         OSL_ENSURE( pEntry, "no style sheet found" );
@@ -3713,11 +3746,14 @@
     m_pImpl->getTableManager().startParagraphGroup();
     m_pImpl->PushProperties(CONTEXT_PARAGRAPH);
     static ::rtl::OUString sDefault( ::rtl::OUString::createFromAscii("Default") );
-    m_pImpl->GetTopContext()->Insert( PROP_PARA_STYLE_NAME, true, uno::makeAny( sDefault ) );
-    if (m_pImpl->isBreakDeferred(PAGE_BREAK))
-        m_pImpl->GetTopContext()->Insert( PROP_BREAK_TYPE, true, uno::makeAny( com::sun::star::style::BreakType_PAGE_BEFORE) );
-    else if (m_pImpl->isBreakDeferred(COLUMN_BREAK))
-        m_pImpl->GetTopContext()->Insert( PROP_BREAK_TYPE, true, uno::makeAny( com::sun::star::style::BreakType_COLUMN_BEFORE) );
+	if (m_pImpl->GetTopContext())
+	{
+	    m_pImpl->GetTopContext()->Insert( PROP_PARA_STYLE_NAME, true, uno::makeAny( sDefault ) );
+    	if (m_pImpl->isBreakDeferred(PAGE_BREAK))
+       		m_pImpl->GetTopContext()->Insert( PROP_BREAK_TYPE, true, uno::makeAny( com::sun::star::style::BreakType_PAGE_BEFORE) );
+    	else if (m_pImpl->isBreakDeferred(COLUMN_BREAK))
+        	m_pImpl->GetTopContext()->Insert( PROP_BREAK_TYPE, true, uno::makeAny( com::sun::star::style::BreakType_COLUMN_BEFORE) );
+	}
     m_pImpl->clearDeferredBreaks(); 
 }
 /*-- 09.06.2006 09:52:14---------------------------------------------------
@@ -3819,7 +3855,7 @@
         }
 
         PropertyMapPtr pContext = m_pImpl->GetTopContext();
-	if ( !pContext->GetFootnote().is() )
+	if ( pContext && !pContext->GetFootnote().is() )
 	{
 	    if (m_pImpl->isBreakDeferred(PAGE_BREAK))
                 m_pImpl->GetTopContext()->Insert( PROP_BREAK_TYPE, true, uno::makeAny( com::sun::star::style::BreakType_PAGE_BEFORE) );
@@ -3878,7 +3914,7 @@
         {
 
             PropertyMapPtr pContext = m_pImpl->GetTopContext();
-            if ( !pContext->GetFootnote().is() )
+            if ( pContext && !pContext->GetFootnote().is() )
             {
                 if (m_pImpl->isBreakDeferred(PAGE_BREAK))
                     m_pImpl->GetTopContext()->Insert( PROP_BREAK_TYPE, true, uno::makeAny( com::sun::star::style::BreakType_PAGE_BEFORE) );
