? filter/filter.vpj
Index: filter/source/config/fragments/fcfg_calc.mk
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/fcfg_calc.mk,v
retrieving revision 1.8
retrieving revision 1.8.192.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.8 -r1.8.192.2
--- filter/source/config/fragments/fcfg_calc.mk	27 Jan 2006 15:47:44 -0000	1.8
+++ filter/source/config/fragments/fcfg_calc.mk	26 Mar 2007 12:14:50 -0000	1.8.192.2
@@ -47,7 +47,9 @@
     calc_dBase\
 	calc8\
 	calc8_template \
-    calc_MS_Excel_2003_XML
+	calc_MS_Excel_2003_XML \
+	MS_Excel_2007_XML \
+	MS_Excel_2007_XML_Template
 
 # -----------------------------------------------
 # count = 28
@@ -75,7 +77,9 @@
 	dBase \
 	calc8 \
 	calc8_template \
-    MS_Excel_2003_XML 
+	MS_Excel_2003_XML \
+	calc_MS_Excel_2007_XML \
+	calc_MS_Excel_2007_XML_Template
 
 # -----------------------------------------------
 # count = 14
@@ -94,7 +98,9 @@
 	calc_StarOffice_XML_Calc_Template_ui \
 	calc8_ui \
 	calc8_template_ui \
-    MS_Excel_2003_XML_ui 
+	MS_Excel_2003_XML_ui \
+	calc_MS_Excel_2007_XML_ui \
+	calc_MS_Excel_2007_XML_Template_ui
     
 # -----------------------------------------------
 # count = 0
Index: filter/source/config/fragments/fcfg_impress.mk
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/fcfg_impress.mk,v
retrieving revision 1.7
retrieving revision 1.7.224.3
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.7 -r1.7.224.3
--- filter/source/config/fragments/fcfg_impress.mk	11 Nov 2005 14:07:22 -0000	1.7
+++ filter/source/config/fragments/fcfg_impress.mk	26 Mar 2007 12:14:51 -0000	1.7.224.3
@@ -28,7 +28,7 @@
 .ENDIF
 
 # -----------------------------------------------
-# count = 18
+# count = 19
 T4_IMPRESS = \
     draw_StarOffice_XML_Draw \
     impress_MS_PowerPoint_97 \
@@ -40,10 +40,12 @@
     pwp_PlaceWare\
     impress8\
     impress8_template\
-    draw8
+    draw8\
+    MS_PowerPoint_2007_XML\
+    MS_PowerPoint_2007_XML_Template
 
 # -----------------------------------------------
-# count = 19
+# count = 20
 F4_IMPRESS = \
     MS_PowerPoint_97 \
     MS_PowerPoint_97_Vorlage \
@@ -55,7 +57,9 @@
     placeware_Export\
     impress8\
     impress8_template\
-    impress8_draw
+    impress8_draw\
+    impress_MS_PowerPoint_2007_XML\
+    impress_MS_PowerPoint_2007_XML_Template
 
 # -----------------------------------------------
 # count = 12
@@ -71,7 +75,9 @@
     impress_StarOffice_XML_Impress_Template_ui \
     impress8_ui \
     impress8_template_ui \
-    impress8_draw_ui
+    impress8_draw_ui \
+    impress_MS_PowerPoint_2007_XML_ui \
+    impress_MS_PowerPoint_2007_XML_Template_ui
 
 # -----------------------------------------------
 # count = 0
Index: filter/source/config/fragments/filters/MS_Excel_4_0.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_4_0.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_4_0.xcu	20 Jul 2005 09:45:16 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_4_0.xcu	29 Mar 2007 15:13:34 -0000	1.6.300.1
@@ -1,8 +1,8 @@
-	<node oor:name="MS Excel 4.0" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT ALIEN PREFERRED</value></prop>
+<node oor:name="MS Excel 4.0" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER NOTINFILEDIALOG NOTINCHOOSER PREFERRED</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="UIName">
 			<value xml:lang="x-default">Microsoft Excel 4.0</value>
 		</prop>
@@ -10,4 +10,4 @@
 		<prop oor:name="Type"><value>calc_MS_Excel_40</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_4_0_Vorlage_Template.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_4_0_Vorlage_Template.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_4_0_Vorlage_Template.xcu	20 Jul 2005 09:45:31 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_4_0_Vorlage_Template.xcu	29 Mar 2007 15:13:34 -0000	1.6.300.1
@@ -1,10 +1,10 @@
-	<node oor:name="MS Excel 4.0 Vorlage/Template" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT TEMPLATE TEMPLATEPATH ALIEN</value></prop>
+<node oor:name="MS Excel 4.0 Vorlage/Template" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER NOTINFILEDIALOG NOTINCHOOSER TEMPLATE TEMPLATEPATH</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="FileFormatVersion"><value>0</value></prop>
 		<prop oor:name="Type"><value>calc_MS_Excel_40_VorlageTemplate</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu	20 Jul 2005 09:46:18 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_5_0_95.xcu	29 Mar 2007 15:13:34 -0000	1.6.300.1
@@ -1,8 +1,8 @@
-	<node oor:name="MS Excel 5.0/95" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN PREFERRED</value></prop>
+<node oor:name="MS Excel 5.0/95" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER NOTINFILEDIALOG NOTINCHOOSER PREFERRED</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="UIName">
 			<value xml:lang="x-default">Microsoft Excel 5.0</value>
 		</prop>
@@ -10,4 +10,4 @@
 		<prop oor:name="Type"><value>calc_MS_Excel_5095</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu	20 Jul 2005 09:46:31 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_5_0_95_Vorlage_Template.xcu	29 Mar 2007 15:13:35 -0000	1.6.300.1
@@ -1,10 +1,10 @@
-	<node oor:name="MS Excel 5.0/95 Vorlage/Template" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT TEMPLATE TEMPLATEPATH ALIEN</value></prop>
+<node oor:name="MS Excel 5.0/95 Vorlage/Template" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER NOTINFILEDIALOG NOTINCHOOSER TEMPLATE TEMPLATEPATH</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="FileFormatVersion"><value>0</value></prop>
 		<prop oor:name="Type"><value>calc_MS_Excel_5095_VorlageTemplate</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_95.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_95.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_95.xcu	20 Jul 2005 09:47:17 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_95.xcu	29 Mar 2007 15:13:35 -0000	1.6.300.1
@@ -1,8 +1,8 @@
-	<node oor:name="MS Excel 95" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN PREFERRED</value></prop>
+<node oor:name="MS Excel 95" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER PREFERRED</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="UIName">
 			<value xml:lang="x-default">Microsoft Excel 95</value>
 		</prop>
@@ -10,4 +10,4 @@
 		<prop oor:name="Type"><value>calc_MS_Excel_95</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu	20 Jul 2005 09:47:30 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_95_Vorlage_Template.xcu	29 Mar 2007 15:13:35 -0000	1.6.300.1
@@ -1,10 +1,10 @@
-	<node oor:name="MS Excel 95 Vorlage/Template" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT TEMPLATE TEMPLATEPATH ALIEN</value></prop>
+<node oor:name="MS Excel 95 Vorlage/Template" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER TEMPLATE TEMPLATEPATH</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="FileFormatVersion"><value>0</value></prop>
 		<prop oor:name="Type"><value>calc_MS_Excel_95_VorlageTemplate</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_97.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_97.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_97.xcu	20 Jul 2005 09:48:18 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_97.xcu	29 Mar 2007 15:13:35 -0000	1.6.300.1
@@ -1,8 +1,8 @@
-	<node oor:name="MS Excel 97" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN</value></prop>
+<node oor:name="MS Excel 97" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER PREFERRED</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="UIName">
 			<value xml:lang="x-default">Microsoft Excel 97/2000/XP</value>
 		</prop>
@@ -10,4 +10,4 @@
 		<prop oor:name="Type"><value>calc_MS_Excel_97</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/MS_Excel_97_Vorlage_Template.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/filters/MS_Excel_97_Vorlage_Template.xcu,v
retrieving revision 1.6
retrieving revision 1.6.300.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.300.1
--- filter/source/config/fragments/filters/MS_Excel_97_Vorlage_Template.xcu	20 Jul 2005 09:48:39 -0000	1.6
+++ filter/source/config/fragments/filters/MS_Excel_97_Vorlage_Template.xcu	29 Mar 2007 15:13:35 -0000	1.6.300.1
@@ -1,10 +1,10 @@
-	<node oor:name="MS Excel 97 Vorlage/Template" oor:op="replace">
-		<prop oor:name="Flags"><value>IMPORT EXPORT TEMPLATE TEMPLATEPATH ALIEN</value></prop>
+<node oor:name="MS Excel 97 Vorlage/Template" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER TEMPLATE TEMPLATEPATH</value></prop>
 		<prop oor:name="UIComponent"/>
-		<prop oor:name="FilterService"/>
-		<prop oor:name="UserData"><value></value></prop>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelBiffFilter</value></prop>
+	<prop oor:name="UserData"/>
 		<prop oor:name="FileFormatVersion"><value>0</value></prop>
 		<prop oor:name="Type"><value>calc_MS_Excel_97_VorlageTemplate</value></prop>
 		<prop oor:name="TemplateName"/>
 		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
-	</node>
+</node>
Index: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML.xcu
diff -N filter/source/config/fragments/filters/calc_MS_Excel_2007_XML.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/calc_MS_Excel_2007_XML.xcu	29 Mar 2007 15:13:36 -0000	1.1.2.3
@@ -0,0 +1,10 @@
+<node oor:name="Calc MS Excel 2007 XML" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER PREFERRED</value></prop>
+	<prop oor:name="UIComponent"/>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelFilter</value></prop>
+	<prop oor:name="UserData"/>
+	<prop oor:name="FileFormatVersion"/>
+	<prop oor:name="Type"><value>MS Excel 2007 XML</value></prop>
+	<prop oor:name="TemplateName"/>
+	<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+</node>
Index: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template.xcu
diff -N filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template.xcu	29 Mar 2007 15:13:36 -0000	1.1.2.2
@@ -0,0 +1,10 @@
+<node oor:name="Calc MS Excel 2007 XML Template" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER TEMPLATE TEMPLATEPATH</value></prop>
+	<prop oor:name="UIComponent"/>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.oox.ExcelFilter</value></prop>
+	<prop oor:name="UserData"/>
+	<prop oor:name="FileFormatVersion"/>
+	<prop oor:name="Type"><value>MS Excel 2007 XML Template</value></prop>
+	<prop oor:name="TemplateName"/>
+	<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+</node>
Index: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template_ui.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template_ui.xcu
diff -N filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template_ui.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_Template_ui.xcu	26 Mar 2007 12:13:35 -0000	1.1.2.1
@@ -0,0 +1,6 @@
+<node oor:name="Calc MS Excel 2007 XML Template">
+	<prop oor:name="UIName">
+		<value xml:lang="en-US">Office Open XML Spreadsheet Document Template</value>
+		<value xml:lang="de">Office Open XML Tabellendokumentvorlage</value>
+	</prop>
+</node>
Index: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_ui.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_ui.xcu
diff -N filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_ui.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/calc_MS_Excel_2007_XML_ui.xcu	26 Mar 2007 12:13:35 -0000	1.1.2.1
@@ -0,0 +1,6 @@
+<node oor:name="Calc MS Excel 2007 XML">
+	<prop oor:name="UIName">
+		<value xml:lang="en-US">Office Open XML Spreadsheet Document</value>
+		<value xml:lang="de">Office Open XML Tabellendokument</value>
+	</prop>
+</node>
Index: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML.xcu
diff -N filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML.xcu	2 Feb 2007 12:18:09 -0000	1.1.2.1
@@ -0,0 +1,10 @@
+<node oor:name="Impress MS PowerPoint 2007 XML" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER PREFERRED</value></prop>
+	<prop oor:name="UIComponent"/>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.Impress.oox.PowerPointImport</value></prop>
+	<prop oor:name="UserData"/>
+	<prop oor:name="FileFormatVersion"/>
+	<prop oor:name="Type"><value>MS PowerPoint 2007 XML</value></prop>
+	<prop oor:name="TemplateName"/>
+	<prop oor:name="DocumentService"><value>com.sun.star.presentation.PresentationDocument</value></prop>
+</node>
Index: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template.xcu
diff -N filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template.xcu	26 Mar 2007 12:13:35 -0000	1.1.2.1
@@ -0,0 +1,10 @@
+<node oor:name="Impress MS PowerPoint 2007 XML Template" oor:op="replace">
+	<prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER TEMPLATE TEMPLATEPATH PREFERRED</value></prop>
+	<prop oor:name="UIComponent"/>
+	<prop oor:name="FilterService"><value>com.sun.star.comp.Impress.oox.PowerPointImport</value></prop>
+	<prop oor:name="UserData"/>
+	<prop oor:name="FileFormatVersion"/>
+	<prop oor:name="Type"><value>MS PowerPoint 2007 XML Template</value></prop>
+	<prop oor:name="TemplateName"/>
+	<prop oor:name="DocumentService"><value>com.sun.star.presentation.PresentationDocument</value></prop>
+</node>
Index: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template_ui.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template_ui.xcu
diff -N filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template_ui.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_Template_ui.xcu	26 Mar 2007 12:13:36 -0000	1.1.2.1
@@ -0,0 +1,6 @@
+<node oor:name="Impress MS PowerPoint 2007 XML Template">
+	<prop oor:name="UIName">
+		<value xml:lang="en-US">Office Open XML Presentation Document Template</value>
+		<value xml:lang="de">Office Open XML Präsentationsdokumentvorlage</value>
+	</prop>
+</node>
Index: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_ui.xcu
===================================================================
RCS file: filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_ui.xcu
diff -N filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_ui.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/filters/impress_MS_PowerPoint_2007_XML_ui.xcu	26 Mar 2007 12:13:36 -0000	1.1.2.1
@@ -0,0 +1,6 @@
+<node oor:name="Impress MS PowerPoint 2007 XML">
+	<prop oor:name="UIName">
+		<value xml:lang="en-US">Office Open XML Presentation Document</value>
+		<value xml:lang="de">Office Open XML Präsentationsdokument</value>
+	</prop>
+</node>
Index: filter/source/config/fragments/types/MS_Excel_2007_XML.xcu
===================================================================
RCS file: filter/source/config/fragments/types/MS_Excel_2007_XML.xcu
diff -N filter/source/config/fragments/types/MS_Excel_2007_XML.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/types/MS_Excel_2007_XML.xcu	26 Mar 2007 12:14:35 -0000	1.1.2.2
@@ -0,0 +1,10 @@
+<node oor:name="MS Excel 2007 XML" oor:op="replace" >
+	<prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+	<prop oor:name="URLPattern"/>
+	<prop oor:name="Extensions"><value>xlsm xlsx</value></prop>
+	<prop oor:name="MediaType"/>
+	<prop oor:name="Preferred"><value>false</value></prop>
+	<prop oor:name="PreferredFilter"><value>Calc MS Excel 2007 XML</value></prop>
+	<prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Spreadsheet Document</value></prop>
+	<prop oor:name="ClipboardFormat"/>
+</node>
Index: filter/source/config/fragments/types/MS_Excel_2007_XML_Template.xcu
===================================================================
RCS file: filter/source/config/fragments/types/MS_Excel_2007_XML_Template.xcu
diff -N filter/source/config/fragments/types/MS_Excel_2007_XML_Template.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/types/MS_Excel_2007_XML_Template.xcu	26 Mar 2007 12:14:35 -0000	1.1.2.1
@@ -0,0 +1,10 @@
+<node oor:name="MS Excel 2007 XML Template" oor:op="replace" >
+	<prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+	<prop oor:name="URLPattern"/>
+	<prop oor:name="Extensions"><value>xltm xltx</value></prop>
+	<prop oor:name="MediaType"/>
+	<prop oor:name="Preferred"><value>false</value></prop>
+	<prop oor:name="PreferredFilter"><value>Calc MS Excel 2007 XML Template</value></prop>
+	<prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Spreadsheet Document Template</value></prop>
+	<prop oor:name="ClipboardFormat"/>
+</node>
Index: filter/source/config/fragments/types/MS_PowerPoint_2007_XML.xcu
===================================================================
RCS file: filter/source/config/fragments/types/MS_PowerPoint_2007_XML.xcu
diff -N filter/source/config/fragments/types/MS_PowerPoint_2007_XML.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/types/MS_PowerPoint_2007_XML.xcu	26 Mar 2007 12:14:35 -0000	1.1.2.2
@@ -0,0 +1,10 @@
+<node oor:name="MS PowerPoint 2007 XML" oor:op="replace" >
+	<prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+	<prop oor:name="URLPattern"/>
+	<prop oor:name="Extensions"><value>pptm pptx</value></prop>
+	<prop oor:name="MediaType"/>
+	<prop oor:name="Preferred"><value>true</value></prop>
+	<prop oor:name="PreferredFilter"><value>Impress MS PowerPoint 2007 XML</value></prop>
+	<prop oor:name="UIName"><value xml:lang="x-default">Office Open XML Presentation Document</value></prop>
+	<prop oor:name="ClipboardFormat"/>
+</node>
Index: filter/source/config/fragments/types/MS_PowerPoint_2007_XML_Template.xcu
===================================================================
RCS file: filter/source/config/fragments/types/MS_PowerPoint_2007_XML_Template.xcu
diff -N filter/source/config/fragments/types/MS_PowerPoint_2007_XML_Template.xcu
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ filter/source/config/fragments/types/MS_PowerPoint_2007_XML_Template.xcu	26 Mar 2007 12:14:35 -0000	1.1.2.1
@@ -0,0 +1,10 @@
+<node oor:name="MS PowerPoint 2007 XML Template" oor:op="replace" >
+	<prop oor:name="DetectService"><value>com.sun.star.comp.oox.FormatDetector</value></prop>
+	<prop oor:name="URLPattern"/>
+	<prop oor:name="Extensions"><value>potm potx</value></prop>
+	<prop oor:name="MediaType"/>
+	<prop oor:name="Preferred"><value>true</value></prop>
+	<prop oor:name="PreferredFilter"><value>Impress MS PowerPoint 2007 XML Template</value></prop>
+	<prop oor:name="UIName"><value>Office Open XML Presentation Document Template</value></prop>
+	<prop oor:name="ClipboardFormat"/>
+</node>
Index: filter/source/config/fragments/types/calc_MS_Excel_40.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_40.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_40.xcu	24 Jan 2006 14:37:19 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_40.xcu	29 Mar 2007 15:14:12 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_40" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xls xlw</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_40_VorlageTemplate.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_40_VorlageTemplate.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_40_VorlageTemplate.xcu	24 Jan 2006 14:37:31 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_40_VorlageTemplate.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_40_VorlageTemplate" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xlt</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_5095.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_5095.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_5095.xcu	24 Jan 2006 14:37:45 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_5095.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_5095" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xls xlw</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_5095_VorlageTemplate.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_5095_VorlageTemplate.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_5095_VorlageTemplate.xcu	24 Jan 2006 14:37:56 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_5095_VorlageTemplate.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_5095_VorlageTemplate" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xlt</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_95.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_95.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_95.xcu	24 Jan 2006 14:38:08 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_95.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_95" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xls xlw</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_95_VorlageTemplate.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_95_VorlageTemplate.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_95_VorlageTemplate.xcu	24 Jan 2006 14:38:19 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_95_VorlageTemplate.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_95_VorlageTemplate" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xlt</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_97.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_97.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_97.xcu	24 Jan 2006 14:38:34 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_97.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_97" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xls xlw</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
Index: filter/source/config/fragments/types/calc_MS_Excel_97_VorlageTemplate.xcu
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/types/calc_MS_Excel_97_VorlageTemplate.xcu,v
retrieving revision 1.6
retrieving revision 1.6.194.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.194.1
--- filter/source/config/fragments/types/calc_MS_Excel_97_VorlageTemplate.xcu	24 Jan 2006 14:38:48 -0000	1.6
+++ filter/source/config/fragments/types/calc_MS_Excel_97_VorlageTemplate.xcu	29 Mar 2007 15:14:13 -0000	1.6.194.1
@@ -1,5 +1,5 @@
 	<node oor:name="calc_MS_Excel_97_VorlageTemplate" oor:op="replace" >
-		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="DetectService"><value>com.sun.star.comp.oox.BiffFormatDetector</value></prop>
 		<prop oor:name="URLPattern"/>
 		<prop oor:name="Extensions"><value>xlt</value></prop>
 		<prop oor:name="MediaType"><value>application/vnd.ms-excel</value></prop>
