? svx/svx.vpj
Index: svx/source/unodraw/unoshap2.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/unodraw/unoshap2.cxx,v
retrieving revision 1.62
retrieving revision 1.51.276.8
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.62 -r1.51.276.8
--- svx/source/unodraw/unoshap2.cxx	10 May 2007 09:16:48 -0000	1.62
+++ svx/source/unodraw/unoshap2.cxx	29 May 2007 11:39:52 -0000	1.51.276.8
@@ -1746,6 +1746,15 @@
 				}
 			}
 		}
+		else if( aValue.getValueType() == INTERFACE_TYPE( graphic::XGraphic ))
+		{
+			Reference< graphic::XGraphic > xGraphic( aValue, UNO_QUERY );
+			if( xGraphic.is() )
+			{
+				Graphic aGraphic( xGraphic );
+				((SdrGrafObj*)mpObj.get())->SetGraphic(aGraphic);
+			}
+		}
 	}
 	else if( mpObj.is() && aPropertyName.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM(UNO_NAME_GRAPHOBJ_GRAFURL)))
 	{
@@ -2168,6 +2177,7 @@
 
 	if ( bCustomShapeGeometry )
 	{
+		((SdrObjCustomShape*)pObject)->MergeDefaultAttributes(0);
 		Rectangle aRect( pObject->GetSnapRect() );
 
 		// #i38892#
Index: svx/source/xoutdev/xattrbmp.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/xoutdev/xattrbmp.cxx,v
retrieving revision 1.15
retrieving revision 1.13.276.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.15 -r1.13.276.2
--- svx/source/xoutdev/xattrbmp.cxx	17 Sep 2006 06:23:16 -0000	1.15
+++ svx/source/xoutdev/xattrbmp.cxx	18 Jan 2007 05:02:18 -0000	1.13.276.2
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_svx.hxx"
 #include <com/sun/star/awt/XBitmap.hpp>
+#include <com/sun/star/graphic/XGraphic.hpp>
 #include <tools/stream.hxx>
 #include <vcl/window.hxx>
 #include <vcl/virdev.hxx>
@@ -802,6 +803,7 @@
     ::rtl::OUString aName;
     ::rtl::OUString aURL;
     ::com::sun::star::uno::Reference< ::com::sun::star::awt::XBitmap > xBmp;
+    ::com::sun::star::uno::Reference< ::com::sun::star::graphic::XGraphic > xGraphic;
 
     bool bSetName   = false;
     bool bSetURL    = false;
@@ -812,7 +814,11 @@
     else if( nMemberId == MID_GRAFURL )
 		bSetURL = (rVal >>= aURL);
 	else if( nMemberId == MID_BITMAP )
+	{
         bSetBitmap = (rVal >>= xBmp);
+		if ( !bSetBitmap )
+			bSetBitmap = (rVal >>= xGraphic );
+	}
     else
     {
         DBG_ASSERT( nMemberId == 0, "invalid member-id" );
@@ -843,8 +849,17 @@
     }
     if( bSetBitmap )
     {
+		Bitmap aInput;
+		if ( xBmp.is() )
+		{
         BitmapEx aInputEx( VCLUnoHelper::GetBitmap( xBmp ) );
-        Bitmap aInput( aInputEx.GetBitmap() );
+		    aInput = aInputEx.GetBitmap();
+		}
+		else if ( xGraphic.is() )
+		{
+			Graphic aGraphic( xGraphic );
+		    aInput = aGraphic.GetBitmap();
+		}
 
         // note: aXOBitmap is the member bitmap
         aXOBitmap.SetBitmap( aInput );
