--- /data4/sles/build-source-only/ooo-build/build/oog680-m5/sc/source/core/tool/address.cxx	2007-10-18 09:58:08.000000000 +0100
+++ sc/source/core/tool/address.cxx	2007-10-22 11:03:16.000000000 +0100
@@ -471,12 +472,28 @@ lcl_ScRange_Parse_XL_R1C1( ScRange& r,
         else if( NULL == (p = lcl_r1c1_get_col( p, rDetails, &r.aStart, &nFlags )))
             goto failed;
 
-        if( p[0] != ':' ||
+        // ScRangeList::Parse will hack a full reference by doing
+        // doing address = address + ':' + address
+        // but if the address contains a sheet ref then we get
+        // something like Sheet1!RC getting transformed into Sheet1!RC:Sheet1!RC
+        // the following skips the sheet ref
+        if( p[0] == ':' )
+        {
+            pTmp = lcl_XL_ParseSheetRef( p+1, &r.aEnd, pDoc,
+                aExternDocName, aEndTabName, FALSE );
+            if ( pTmp[0] == '!' )
+               p = pTmp; 
+        }
+	 
+
+
+        if( ( p[0] != ':' && p[0] != '!' ) ||
             (p[1] != 'R' && p[1] != 'r') ||
             NULL == (pTmp = lcl_r1c1_get_row( p+1, rDetails, &r.aEnd, &nFlags2 )) ||
             (*pTmp != 'C' && *pTmp != 'c') ||
             NULL == (pTmp = lcl_r1c1_get_col( pTmp, rDetails, &r.aEnd, &nFlags2 )))
         {
+
             return bOnlyAcceptSingle ? lcl_XL_LinkSheetRef( r, pDoc,
                 aExternDocName, aStartTabName, aEndTabName, nFlags ) : 0;
         }
@@ -623,7 +640,21 @@ lcl_ScRange_Parse_XL_A1( ScRange& r,
     // prepare as if it's a singleton, in case we want to fall back */
     r.aEnd.SetCol( r.aStart.Col() );
     r.aEnd.SetRow( r.aStart.Row() );    // don't overwrite sheet number as parsed in lcl_ScRange_Parse_XL_Header
-    if( *tmp2 != ':' )
+
+    // ScRangeList::Parse will hack a full reference by doing
+    // doing address = address + ':' + address
+    // but if the address contains a sheet ref then we get
+    // something like Sheet1!RC getting transformed into Sheet1!RC:Sheet1!RC
+    // the following skips the sheet ref
+    if( *tmp2 == ':' )
+    {
+        tmp1 = lcl_XL_ParseSheetRef( tmp2+1, &r.aEnd, pDoc,
+            aExternDocName, aEndTabName, FALSE );
+        if ( tmp1[0] == '!' )
+            tmp2 = tmp1; 
+    }
+
+    if( ( *tmp2 != ':' ) && ( *tmp2 != '!' ) )
     {
         if ( !bOnlyAcceptSingle )
             nFlags &= ~SCA_VALID;   // when looking for a double ref, a single-cell ref must not be accepted
