--- config_office/configure.in	10 May 2007 16:40:23 -0000	1.204
+++ config_office/configure.in	16 Jul 2007 11:07:17 -0000	1.204.2.1
@@ -2759,6 +2759,14 @@
 AC_SUBST(EPM)
 
 dnl ===================================================================
+dnl Check for gperf
+dnl ===================================================================
+AC_PATH_PROG(GPERF, gperf)
+if test -z "$GPERF"; then
+	AC_MSG_ERROR([gperf not found but needed. Install it.])
+fi
+
+dnl ===================================================================
 dnl Check for building ODK
 dnl ===================================================================
 AC_MSG_CHECKING([whether to build the ODK])
