Index: svx/inc/svxerr.hxx
===================================================================
RCS file: /cvs/graphics/svx/inc/svxerr.hxx,v
retrieving revision 1.3
retrieving revision 1.3.112.3
diff -w -u -r1.3 -r1.3.112.3
--- svx/inc/svxerr.hxx	3 Feb 2004 17:45:45 -0000	1.3
+++ svx/inc/svxerr.hxx	16 Jul 2004 17:45:58 -0000	1.3.112.3
@@ -101,6 +101,19 @@
 #define ERRCODE_SVX_VBASIC_STORAGE_EXIST       (14UL | ERRCODE_AREA_SVX     \
 													 | ERRCODE_WARNING_MASK \
 													 | ERRCODE_CLASS_WRITE )
+
+/** Error message: "Wrong password." */
+#define ERRCODE_SVX_WRONGPASS               (15UL | ERRCODE_AREA_SVX | ERRCODE_CLASS_NONE)
+
+/** Error message: "Read error. Unsupported encryption method." */
+#define ERRCODE_SVX_READ_FILTER_CRYPT       (16UL | ERRCODE_AREA_SVX | ERRCODE_CLASS_READ)
+
+/** Error message: "Read error. Passwort encrypted Powerpoint documents..." */
+#define ERRCODE_SVX_READ_FILTER_PPOINT      (17UL | ERRCODE_AREA_SVX | ERRCODE_CLASS_READ)
+
+/** Error message: "Warning. Passwort protection is not suppported when ..." */
+#define ERRCODE_SVX_EXPORT_FILTER_CRYPT      (18UL | ERRCODE_AREA_SVX | ERRCODE_CLASS_EXPORT)
+
 
 // both codes will be used twice : with ERRCODE_CLASS_READ- and ERRCODE_CLASS_WRITE-bits
 #define ERRCTX_SVX_LINGU_THESAURUS				1


Index: svx/prj/d.lst
===================================================================
RCS file: /cvs/graphics/svx/prj/d.lst,v
retrieving revision 1.89
retrieving revision 1.81.28.2
diff -w -u -r1.89 -r1.81.28.2
--- svx/prj/d.lst	28 Jun 2004 14:03:17 -0000	1.89
+++ svx/prj/d.lst	13 Jul 2004 19:12:51 -0000	1.81.28.2
@@ -543,6 +543,7 @@
 hedabu: ..\inc\unoforou.hxx %_DEST%\inc%_EXT%\svx\unoforou.hxx
 hedabu: ..\inc\e3dundo.hxx %_DEST%\inc%_EXT%\svx\e3dundo.hxx
 hedabu: ..\inc\anchorid.hxx %_DEST%\inc%_EXT%\svx\anchorid.hxx
+hedabu: ..\inc\mscodec.hxx %_DEST%\inc%_EXT%\svx\mscodec.hxx
 hedabu: ..\inc\msdffimp.hxx %_DEST%\inc%_EXT%\svx\msdffimp.hxx
 hedabu: ..\inc\msdffdef.hxx %_DEST%\inc%_EXT%\svx\msdffdef.hxx
 hedabu: ..\inc\msfiltertracer.hxx %_DEST%\inc%_EXT%\svx\msfiltertracer.hxx

Index: svx/source/items/svxerr.src
===================================================================
RCS file: /cvs/graphics/svx/source/items/svxerr.src,v
retrieving revision 1.47
retrieving revision 1.46.152.5
diff -w -u -r1.47 -r1.46.152.5
--- svx/source/items/svxerr.src	25 Jun 2004 12:13:12 -0000	1.47
+++ svx/source/items/svxerr.src	16 Jul 2004 17:46:33 -0000	1.46.152.5
@@ -216,47 +216,31 @@
 		Text [ de ] = "Der im Dokument enthaltene originale VBA Basic Code wird nicht gespeichert." ;
 		Text [ en-US ] = "The original VBA Basic Code contained in the document will not be saved.";
 	};
-};
- // ********************************************************************** EOF
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
 
+    String ERRCODE_SVX_WRONGPASS & ERRCODE_RES_MASK
+    {
+        Text [ de ] = "Das Kennwort ist ungültig. Das Dokument kann nicht geöffnet werden.";
+        Text [ en-US ] = "The password is incorrect. The document cannot be opened.";
+    };
 
+    String ERRCODE_SVX_READ_FILTER_CRYPT & ERRCODE_RES_MASK
+    {
+        Text [ de ] = "Die Verschlüsselungsmethode des aktuellen Dokumentes wird nicht unterstützt. Nur die Microsoft Office 97/2000 kompatible Kennwortverschlüsselung wird unterstützt.";
+        Text[ en-US ] = "The encryption method of this document is not supported. Only the Microsoft Office 97/2000 compatible password encryption is supported.";
+    };
 
+    String ERRCODE_SVX_READ_FILTER_PPOINT & ERRCODE_RES_MASK
+    {
+        Text [ de ] = "Das Laden von kennwortgeschützten PowerPoint Dokumenten wird nicht unterstützt.";
+        Text [ en-US ] = "The loading of password-encrypted PowerPoint documents is not supported.";
+    };
 
+    String ERRCODE_SVX_EXPORT_FILTER_CRYPT & ERRCODE_RES_MASK
+    {
+        Text [ de ] = "Der Kennwortschutz wird beim Speichern von Dokumementen im Microsoft Office Format nicht unterstützt.\nMöchten Sie Dokument ohne Kennwortschtz speichern";
+        Text[ en-US ] = "Password protection is not supported when documents are saved in Microsoft Office format.\nDo you want to save the document without password protection";
+	Text[ x-comment ] = "This error is used with an InteractionRequest Handler using a Query classification. This adds a question mark to the end of the string. So no need for one here. Note also the new line between sentences."; 
+    };
+};
+ // ********************************************************************** EOF
 

Index: svx/source/msfilter/makefile.mk
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/makefile.mk,v
retrieving revision 1.11
retrieving revision 1.11.484.1
diff -w -u -r1.11 -r1.11.484.1
--- svx/source/msfilter/makefile.mk	24 Apr 2003 13:26:39 -0000	1.11
+++ svx/source/msfilter/makefile.mk	17 Mar 2004 16:39:32 -0000	1.11.484.1
@@ -83,6 +83,7 @@
 	$(SLO)$/svxmsbas.obj	\
 	$(SLO)$/msocximex.obj	\
 	$(SLO)$/msashape3d.obj	\
+	$(SLO)$/mscodec.obj		\
 	$(SLO)$/msfiltertracer.obj
 
 EXCEPTIONSFILES= \
