diff -rup /data4/M21/ooo-build/build-src/dev300-m21/forms/source/component/ImageControl.cxx forms/source/component/ImageControl.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/forms/source/component/ImageControl.cxx	2008-04-11 09:16:50.000000000 +0100
+++ forms/source/component/ImageControl.cxx	2008-07-08 00:04:54.000000000 +0100
@@ -57,6 +57,9 @@
 #include <com/sun/star/io/NotConnectedException.hpp>
 #include <com/sun/star/beans/PropertyValue.hpp>
 #include <com/sun/star/graphic/XGraphic.hpp>
+#include <com/sun/star/graphic/XGraphicLinkProvider.hpp>
+#include <com/sun/star/embed/ElementModes.hpp>
+
 #include <tools/urlobj.hxx>
 #include <tools/stream.hxx>
 #include <tools/debug.hxx>
@@ -353,13 +356,23 @@ sal_Bool OImageControlModel::handleNewIm
     ::std::auto_ptr< SvStream > pImageStream;
     Reference< XInputStream > xImageStream;
 
-    if ( ::svt::ImageResourceAccess::isImageResourceURL( _rURL ) )
+    Reference< graphic::XGraphicLinkProvider > xProv;
+    m_xAggregateSet->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("GraphicLinkProvider" ) ) )  >>= xProv;
+    if ( xProv.is() && utl::SvPictureStreamHelper::IsPackagePictureURL( _rURL )  )
     {
-        xImageStream = ::svt::ImageResourceAccess::getImageXStream( getORB(), _rURL );
+        pImageStream.reset( utl::SvPictureStreamHelper::GetStreamFromPackagePictureURL( _rURL, xProv->getStorage() ) );
     }
-    else
+    if ( !pImageStream.get() )
     {
-        pImageStream.reset( ::utl::UcbStreamHelper::CreateStream( _rURL, STREAM_READ ) );
+        if ( ::svt::ImageResourceAccess::isImageResourceURL( _rURL ) )
+        {
+            xImageStream = ::svt::ImageResourceAccess::getImageXStream( getORB(), _rURL );
+        }
+        else
+        {
+            pImageStream.reset( ::utl::UcbStreamHelper::CreateStream( _rURL, STREAM_READ ) );
+        }
+    }
 	    sal_Bool bSetNull = ( pImageStream.get() == NULL ) || ( ERRCODE_NONE != pImageStream->GetErrorCode() );
 
         if (!bSetNull)
@@ -373,7 +386,6 @@ sal_Bool OImageControlModel::handleNewIm
 
             xImageStream = new ::utl::OInputStreamHelper( new SvLockBytes( pImageStream.get(), sal_False ), nSize );
         }
-    }
 
     if ( xImageStream.is() )
     {
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/forms/source/component/imgprod.cxx forms/source/component/imgprod.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/forms/source/component/imgprod.cxx	2008-05-05 16:21:55.000000000 +0100
+++ forms/source/component/imgprod.cxx	2008-07-07 22:49:39.000000000 +0100
@@ -49,6 +49,8 @@
 // - ImgProdLockBytes -
 // --------------------
 
+using namespace com::sun::star;
+
 class ImgProdLockBytes : public SvLockBytes
 {
 	::com::sun::star::uno::Reference< ::com::sun::star::io::XInputStream > 		xStmRef;
@@ -644,6 +646,21 @@ void ImageProducer::initialize( const ::
 			SetImage( aURL );
 		}
 	}
+	else if ( aArguments.getLength() == 2 )
+	{
+		rtl::OUString aURL;
+		uno::Reference< embed::XStorage > xStor;	
+		if ( ( aArguments[ 0 ] >>= aURL ) && ( aArguments[ 1 ] >>= xStor ) )
+		{
+			if ( utl::SvPictureStreamHelper::IsPackagePictureURL( aURL ) )
+			{
+				std::auto_ptr< SvStream > pIs( utl::SvPictureStreamHelper::GetStreamFromPackagePictureURL( aURL, xStor ) );
+				SetImage( *pIs );
+			}		
+			else
+				SetImage( aURL );
+		}
+	}
 }
 
 namespace frm
Only in forms/source/resource: localize.sdf
Only in goodies/source/filter.vcl/egif: localize.sdf
Only in goodies/source/filter.vcl/eos2met: localize.sdf
Only in goodies/source/filter.vcl/epbm: localize.sdf
Only in goodies/source/filter.vcl/epgm: localize.sdf
Only in goodies/source/filter.vcl/epict: localize.sdf
Only in goodies/source/filter.vcl/eppm: localize.sdf
Only in goodies/source/filter.vcl/eps: localize.sdf
Only in goodies/source/inv: localize.sdf
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/goodies/source/unographic/provider.cxx goodies/source/unographic/provider.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/goodies/source/unographic/provider.cxx	2008-04-11 01:08:51.000000000 +0100
+++ goodies/source/unographic/provider.cxx	2008-07-07 23:05:02.000000000 +0100
@@ -52,6 +52,8 @@
 #include <com/sun/star/io/XStream.hpp>
 #include <com/sun/star/text/GraphicCrop.hpp>
 
+#include <comphelper/storagehelper.hxx>
+
 #include "descriptor.hxx"
 #include "graphic.hxx"
 #include "provider.hxx"
@@ -319,7 +321,7 @@ uno::Reference< beans::XPropertySet > SA
 
 	::rtl::OUString aURL;
 	uno::Reference< io::XInputStream > xIStm;
-
+	uno::Reference< embed::XStorage > xStor;
 	for( sal_Int32 i = 0; ( i < rMediaProperties.getLength() ) && !xRet.is(); ++i )
 	{
 		const ::rtl::OUString	aName( rMediaProperties[ i ].Name );
@@ -333,9 +335,34 @@ uno::Reference< beans::XPropertySet > SA
 		{
 			aValue >>= xIStm;
 		}
+		else if( COMPARE_EQUAL == aName.compareToAscii( "Storage" ) )
+		{
+			aValue >>= xStor;
+		}
 	}
-
-	if( xIStm.is() )
+	
+    if ( xStor.is() && ::utl::SvPictureStreamHelper::IsPackagePictureURL( aURL ) )
+	{
+		// #TODO some asserts or something if xStream or xIStm are null
+		// #TODO - this is duplicated in queryGraphic below ( move to common func )
+		std::auto_ptr< SvStream > pIStm;
+		::GraphicFilter* pFilter = ::GraphicFilter::GetGraphicFilter();
+		
+		if( pFilter )
+		{
+			::Graphic aVCLGraphic;
+			
+			if( ( pFilter->ImportGraphic( aVCLGraphic, aURL, *pIStm ) == GRFILTER_OK ) && 
+				( aVCLGraphic.GetType() != GRAPHIC_NONE ) )
+			{
+				::unographic::Graphic* pUnoGraphic = new ::unographic::Graphic;
+				
+				pUnoGraphic->init( aVCLGraphic );
+				xRet = pUnoGraphic;
+			}
+		}
+	}
+	else if( xIStm.is() )
 	{
 		GraphicDescriptor* pDescriptor = new GraphicDescriptor;
 		pDescriptor->init( xIStm, aURL );
@@ -379,6 +406,7 @@ uno::Reference< ::graphic::XGraphic > SA
 	SvStream* 								pIStm = NULL;
 
 	uno::Reference< io::XInputStream > xIStm;
+	uno::Reference< embed::XStorage > xStor;
 
 	for( sal_Int32 i = 0; ( i < rMediaProperties.getLength() ) && !pIStm && !xRet.is(); ++i )
 	{
@@ -395,12 +423,22 @@ uno::Reference< ::graphic::XGraphic > SA
 		{
 			aValue >>= xIStm;
 		}
+		else if( COMPARE_EQUAL == aName.compareToAscii( "Storage" ) )
+		{
+			aValue >>= xStor;
+		}
 	}
 
+
 	if( xIStm.is() )
 	{
 		pIStm = ::utl::UcbStreamHelper::CreateStream( xIStm );
 	}
+
+	else if ( xStor.is() && utl::SvPictureStreamHelper::IsPackagePictureURL( aPath ) )
+	{
+		pIStm = ::utl::SvPictureStreamHelper::GetStreamFromPackagePictureURL( aPath, xStor );
+	}
 	else if( aPath.Len() )
 	{
 		xRet = implLoadMemory( aPath );
@@ -434,6 +472,8 @@ uno::Reference< ::graphic::XGraphic > SA
 				pUnoGraphic->init( aVCLGraphic );
 				xRet = pUnoGraphic;
 			}
+			if ( xStor.is() )
+				OSL_TRACE("**** xRet is %d", xRet.is() );
 		}
 	
 		delete pIStm;
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/offapi/com/sun/star/document/makefile.mk offapi/com/sun/star/document/makefile.mk
--- /data4/M21/ooo-build/build-src/dev300-m21/offapi/com/sun/star/document/makefile.mk	2008-07-09 13:18:21.000000000 +0100
+++ offapi/com/sun/star/document/makefile.mk	2008-07-09 09:32:56.000000000 +0100
@@ -92,6 +92,7 @@ IDLFILES=\
 	XFilter.idl\
 	XFilterAdapter.idl\
 	XGraphicObjectResolver.idl\
+	XGraphicResolver.idl\
 	XImporter.idl\
 	XInteractionFilterOptions.idl\
 	XInteractionFilterSelect.idl\
Only in offapi/com/sun/star/document: XGraphicResolver.idl
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/offapi/com/sun/star/graphic/makefile.mk offapi/com/sun/star/graphic/makefile.mk
--- /data4/M21/ooo-build/build-src/dev300-m21/offapi/com/sun/star/graphic/makefile.mk	2008-04-11 01:03:27.000000000 +0100
+++ offapi/com/sun/star/graphic/makefile.mk	2008-07-07 16:22:04.000000000 +0100
@@ -53,7 +53,8 @@ IDLFILES= \
 	XGraphic.idl \
 	XGraphicProvider.idl \
 	XGraphicRenderer.idl \
-	XGraphicTransformer.idl
+	XGraphicTransformer.idl \
+	XGraphicLinkProvider.idl \
 	
 # --- Targets ------------------------------------------------------
 
Only in offapi/com/sun/star/graphic: XGraphicLinkProvider.idl
Only in svx/inc/: localize.sdf
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/svx/inc/xmlgrhlp.hxx svx/inc/xmlgrhlp.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/svx/inc/xmlgrhlp.hxx	2008-04-10 20:13:29.000000000 +0100
+++ svx/inc/xmlgrhlp.hxx	2008-07-09 10:35:12.000000000 +0100
@@ -37,8 +37,9 @@
 #include <vector>
 #include <set>
 #include <utility>
-#include <com/sun/star/document/XGraphicObjectResolver.hpp>
+#include <com/sun/star/document/XGraphicResolver.hpp>
 #include <com/sun/star/document/XBinaryStreamResolver.hpp>
+#include <com/sun/star/graphic/XGraphic.hpp>
 #include <com/sun/star/embed/XStorage.hpp>
 #include "svx/svxdllapi.h"
 
@@ -61,8 +62,9 @@ struct SvxGraphicHelperStream_Impl
     ::com::sun::star::uno::Reference < ::com::sun::star::io::XStream > xStream;
 };
 
-class SVX_DLLPUBLIC SvXMLGraphicHelper : public ::cppu::WeakComponentImplHelper2<	::com::sun::star::document::XGraphicObjectResolver,
-                                                                    ::com::sun::star::document::XBinaryStreamResolver >
+typedef ::cppu::WeakComponentImplHelper2< ::com::sun::star::document::XGraphicResolver, ::com::sun::star::document::XBinaryStreamResolver > SvXMLGraphicHelper_BASE;
+
+class SVX_DLLPUBLIC SvXMLGraphicHelper : public SvXMLGraphicHelper_BASE
 {
 private:
 
@@ -118,6 +120,7 @@ public:
 
 public:
 
+    virtual ::rtl::OUString SAL_CALL resolveGraphicObject( const ::com::sun::star::uno::Reference< ::com::sun::star::graphic::XGraphic >& rxGraphic ) throw(::com::sun::star::uno::RuntimeException);
 	// XGraphicObjectResolver
     virtual ::rtl::OUString SAL_CALL resolveGraphicObjectURL( const ::rtl::OUString& aURL ) throw(::com::sun::star::uno::RuntimeException);
 
Only in svx/source/accessibility: localize.sdf
Only in svx/source/dialog: localize.sdf
Only in svx/source/editeng: localize.sdf
Only in svx/source/engine3d: localize.sdf
Only in svx/source/fmcomp: localize.sdf
Only in svx/source/form: localize.sdf
Only in svx/source/gallery2: localize.sdf
Only in svx/source/intro: localize.sdf
Only in svx/source/items: localize.sdf
Only in svx/source/mnuctrls: localize.sdf
Only in svx/source/options: localize.sdf
Only in svx/source/outliner: localize.sdf
Only in svx/source/src: localize.sdf
Only in svx/source/stbctrls: localize.sdf
Only in svx/source/svdraw: localize.sdf
Only in svx/source/svxlink: localize.sdf
Only in svx/source/table: localize.sdf
Only in svx/source/tbxctrls: localize.sdf
Only in svx/source/toolbars: localize.sdf
Only in svx/source/unodialogs/textconversiondlgs: localize.sdf
Only in svx/source/unodraw: localize.sdf
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/svx/source/xml/xmlgrhlp.cxx svx/source/xml/xmlgrhlp.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/svx/source/xml/xmlgrhlp.cxx	2008-04-11 05:11:07.000000000 +0100
+++ svx/source/xml/xmlgrhlp.cxx	2008-07-09 10:54:27.000000000 +0100
@@ -38,7 +38,9 @@
 #include <com/sun/star/lang/XMultiServiceFactory.hpp>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <com/sun/star/lang/XInitialization.hpp>
+#include <com/sun/star/graphic/XGraphicProvider.hpp>
 #include <cppuhelper/compbase4.hxx>
+#include <comphelper/processfactory.hxx>
 
 #include <unotools/ucbstreamhelper.hxx>
 #include <unotools/streamwrap.hxx>
@@ -401,16 +403,12 @@ const GraphicObject& SvXMLGraphicOutputS
 // - SvXMLGraphicHelper -
 // ----------------------
 
-SvXMLGraphicHelper::SvXMLGraphicHelper( SvXMLGraphicHelperMode eCreateMode ) :
-	::cppu::WeakComponentImplHelper2< ::com::sun::star::document::XGraphicObjectResolver,
-                                      ::com::sun::star::document::XBinaryStreamResolver >( maMutex )
+SvXMLGraphicHelper::SvXMLGraphicHelper( SvXMLGraphicHelperMode eCreateMode ) : SvXMLGraphicHelper_BASE( maMutex )
 {
 	Init( NULL, eCreateMode, sal_False );
 }
 
-SvXMLGraphicHelper::SvXMLGraphicHelper() :
-	::cppu::WeakComponentImplHelper2< ::com::sun::star::document::XGraphicObjectResolver,
-                                      ::com::sun::star::document::XBinaryStreamResolver >( maMutex )
+SvXMLGraphicHelper::SvXMLGraphicHelper() : SvXMLGraphicHelper_BASE( maMutex )
 {
 }
 
@@ -820,6 +818,22 @@ void SvXMLGraphicHelper::Destroy( SvXMLG
 
 // -----------------------------------------------------------------------------
 
+OUString SAL_CALL SvXMLGraphicHelper::resolveGraphicObject( const uno::Reference< graphic::XGraphic > & rxGraphic ) throw(uno::RuntimeException)
+{
+    rtl::OUString sRet;
+    // Create tmp GraphicObject
+    GraphicObject tmpGraphic( rxGraphic ); 
+    const OUString          aId( OUString::createFromAscii( tmpGraphic.GetUniqueID().GetBuffer() ) );
+
+    if( aId.getLength() )
+    {
+        sRet = OUString::createFromAscii( XML_GRAPHICOBJECT_URL_BASE );
+        sRet += aId;
+        sRet = resolveGraphicObjectURL( sRet ); 
+    }
+    return sRet;
+}
+
 // XGraphicObjectResolver
 OUString SAL_CALL SvXMLGraphicHelper::resolveGraphicObjectURL( const OUString& aURL )
 	throw(uno::RuntimeException)
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/inc/toolkit/controls/unocontrols.hxx toolkit/inc/toolkit/controls/unocontrols.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/inc/toolkit/controls/unocontrols.hxx	2008-04-11 10:00:24.000000000 +0100
+++ toolkit/inc/toolkit/controls/unocontrols.hxx	2008-07-07 18:02:37.000000000 +0100
@@ -54,6 +54,8 @@
 #include <com/sun/star/awt/XCurrencyField.hpp>
 #include <com/sun/star/awt/XPatternField.hpp>
 #include <com/sun/star/awt/XProgressBar.hpp>
+#include <com/sun/star/graphic/XGraphicLinkProvider.hpp>
+#include <com/sun/star/lang/XInitialization.hpp>
 #include <toolkit/controls/unocontrolmodel.hxx>
 #include <toolkit/controls/unocontrolbase.hxx>
 #include <toolkit/helper/macros.hxx>
@@ -67,6 +69,23 @@
 
 #include <list>
 
+namespace css = ::com::sun::star; 
+
+typedef ::cppu::WeakImplHelper2< css::graphic::XGraphicLinkProvider, css::lang::XInitialization >  GRAPHIC_LINKPROVIDER_BASE;
+
+class GraphLinkProviderImpl : public GRAPHIC_LINKPROVIDER_BASE
+{
+    css::uno::Reference< css::embed::XStorage > mxStorage;
+    rtl::OUString msURL;
+public:
+    GraphLinkProviderImpl();
+    // XGraphicLinkProvider
+    virtual css::uno::Reference< css::embed::XStorage > SAL_CALL getStorage() throw (css::uno::RuntimeException);
+
+    virtual ::rtl::OUString SAL_CALL getSourceURL() throw (css::uno::RuntimeException);
+    // XInitialization
+    virtual void SAL_CALL initialize( const css::uno::Sequence< css::uno::Any >& aArguments ) throw (css::uno::Exception, css::uno::RuntimeException);
+};
 
 //	----------------------------------------------------
 //	class UnoControlEditModel
@@ -219,10 +238,10 @@ private:
 	std::list< ::com::sun::star::uno::Reference< ::com::sun::star::awt::XImageConsumer > >  maListeners;
     bool                                                                                    mbAdjustingImagePosition;
     bool                                                                                    mbAdjustingGraphic;
-
+    bool                                                                                    mbAdjustingLinkProvider;
 protected:
-    ImageProducerControlModel() : mbAdjustingImagePosition( false ), mbAdjustingGraphic( false ) { }
-    ImageProducerControlModel( const ImageProducerControlModel& _rSource ) : com::sun::star::awt::XImageProducer(), UnoControlModel( _rSource ), mbAdjustingImagePosition( false ), mbAdjustingGraphic( false ) { }
+    ImageProducerControlModel();
+    ImageProducerControlModel( const ImageProducerControlModel& _rSource );
 
 	::com::sun::star::uno::Any	SAL_CALL queryInterface( const ::com::sun::star::uno::Type & rType ) throw(::com::sun::star::uno::RuntimeException);
     ::com::sun::star::uno::Any	SAL_CALL queryAggregation( const ::com::sun::star::uno::Type & rType ) throw(::com::sun::star::uno::RuntimeException);
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/inc/toolkit/helper/property.hxx toolkit/inc/toolkit/helper/property.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/inc/toolkit/helper/property.hxx	2008-07-09 13:18:03.000000000 +0100
+++ toolkit/inc/toolkit/helper/property.hxx	2008-07-07 17:01:58.000000000 +0100
@@ -190,6 +190,7 @@ namespace rtl {
 #define BASEPROPERTY_CUSTOMUNITTEXT                 136  // ::rtl::OUString
 #define BASEPROPERTY_ENABLEVISIBLE                  137  // sal_Bool
 #define BASEPROPERTY_GROUPNAME                      138  // ::rtl::OUString
+#define BASEPROPERTY_GRAPHICLINKPROVIDER            139  // css::graphic::XGraphicLinkProvider
 
 // Keine gebundenen Properties, werden immer aus der Property BASEPROPERTY_FONTDESCRIPTOR entnommen.
 #define BASEPROPERTY_FONTDESCRIPTORPART_START			1000
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/inc/toolkit/helper/servicenames.hxx toolkit/inc/toolkit/helper/servicenames.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/inc/toolkit/helper/servicenames.hxx	2008-04-11 10:05:43.000000000 +0100
+++ toolkit/inc/toolkit/helper/servicenames.hxx	2008-07-07 18:10:57.000000000 +0100
@@ -98,6 +98,7 @@ extern const sal_Char __FAR_DATA szServi
 extern const sal_Char __FAR_DATA szServiceName_UnoSimpleAnimationControl[], szServiceName_UnoSimpleAnimationControlModel[];
 extern const sal_Char __FAR_DATA szServiceName_UnoThrobberControl[], szServiceName_UnoThrobberControlModel[];
 extern const sal_Char __FAR_DATA szServiceName_UnoControlFixedHyperlink[], szServiceName_UnoControlFixedHyperlinkModel[];
+extern const sal_Char __FAR_DATA szServiceName_GraphLinkProvider[];
 
 // ExtUnoWrapper:
 extern const char __FAR_DATA szServiceName_ImageProducer[], szServiceName2_ImageProducer[];
Only in toolkit/source/awt: localize.sdf
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/controls/dialogcontrol.cxx toolkit/source/controls/dialogcontrol.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/controls/dialogcontrol.cxx	2008-07-09 13:18:03.000000000 +0100
+++ toolkit/source/controls/dialogcontrol.cxx	2008-07-07 23:08:03.000000000 +0100
@@ -2118,7 +2118,8 @@ throw (RuntimeException)
 	rbase  >>= baseLocation;
 	rUrl  >>= url;
 
-	if ( url.getLength() > 0 )
+	// Don't adjust a package url
+	if ( !utl::SvPictureStreamHelper::IsPackagePictureURL( url ) && url.getLength() > 0 )
 	{
 		INetURLObject urlObj(baseLocation);
 		urlObj.removeSegment();
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/controls/unocontrolmodel.cxx toolkit/source/controls/unocontrolmodel.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/controls/unocontrolmodel.cxx	2008-07-09 13:18:03.000000000 +0100
+++ toolkit/source/controls/unocontrolmodel.cxx	2008-07-07 17:31:51.000000000 +0100
@@ -37,6 +37,7 @@
 #include <com/sun/star/awt/FontWeight.hpp>
 #include <com/sun/star/awt/FontSlant.hpp>
 #include <com/sun/star/graphic/XGraphicProvider.hpp>
+#include <com/sun/star/graphic/XGraphicLinkProvider.hpp>
 #include <com/sun/star/io/XMarkableStream.hpp>
 #include <toolkit/controls/unocontrolmodel.hxx>
 #include <toolkit/helper/macros.hxx>
@@ -256,6 +257,8 @@ void UnoControlModel::ImplPropertyChange
         {
 			case BASEPROPERTY_GRAPHIC:				aDefault <<= ::com::sun::star::uno::makeAny( 
 																	::com::sun::star::uno::Reference< graphic::XGraphic >() ); break;
+			case BASEPROPERTY_GRAPHICLINKPROVIDER:				aDefault <<= ::com::sun::star::uno::makeAny( 
+																	::com::sun::star::uno::Reference< graphic::XGraphicLinkProvider >() ); break;
             case BASEPROPERTY_VERTICALALIGN:
             case BASEPROPERTY_BORDERCOLOR:
             case BASEPROPERTY_SYMBOL_COLOR:
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/controls/unocontrols.cxx toolkit/source/controls/unocontrols.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/controls/unocontrols.cxx	2008-04-11 10:33:44.000000000 +0100
+++ toolkit/source/controls/unocontrols.cxx	2008-07-09 12:40:45.000000000 +0100
@@ -37,6 +37,8 @@
 #include <com/sun/star/awt/VisualEffect.hpp>
 #include <com/sun/star/awt/LineEndFormat.hpp>
 #include <com/sun/star/graphic/XGraphicProvider.hpp>
+#include <com/sun/star/document/XStorageBasedDocument.hpp>
+#include <com/sun/star/frame/XModel.hpp>
 #include <com/sun/star/util/Date.hpp>
 
 
@@ -51,6 +53,7 @@
 #include <toolkit/helper/servicenames.hxx>
 #include <toolkit/helper/macros.hxx>
 #include <toolkit/helper/imagealign.hxx>
+#include <unotools/ucbstreamhelper.hxx>
 
 // for introspection
 #include <toolkit/awt/vclxwindows.hxx>
@@ -58,6 +61,7 @@
 #include <comphelper/componentcontext.hxx>
 #include <comphelper/processfactory.hxx>
 #include <comphelper/extract.hxx>
+
 #include <vcl/wrkwin.hxx>
 #include <vcl/svapp.hxx>
 #include <vcl/edit.hxx>
@@ -104,6 +108,34 @@ static void lcl_knitImageComponents( con
     }
 }
 
+
+GraphLinkProviderImpl::GraphLinkProviderImpl()
+{
+}
+
+Reference< embed::XStorage > SAL_CALL GraphLinkProviderImpl::getStorage() throw (RuntimeException)
+{
+    return mxStorage;
+}
+
+::rtl::OUString SAL_CALL GraphLinkProviderImpl::getSourceURL() throw (RuntimeException)
+{
+    return msURL;
+}
+
+void SAL_CALL GraphLinkProviderImpl::initialize( const uno::Sequence< uno::Any >& aArguments ) throw (uno::Exception, uno::RuntimeException)
+{
+    if ( aArguments.getLength() != 2 )
+        throw uno::RuntimeException();
+
+    aArguments[ 0 ] >>= msURL;
+    Reference< frame::XModel > xModel( aArguments[ 1 ], UNO_QUERY_THROW );
+
+    Reference< document::XStorageBasedDocument > xDocStorage( xModel, UNO_QUERY_THROW );
+    mxStorage.set( xDocStorage->getDocumentStorage(), UNO_QUERY_THROW );
+    // #TODO sprinkle with some asserts
+}
+
 //	----------------------------------------------------
 //	class UnoControlEditModel
 //	----------------------------------------------------
@@ -548,6 +580,17 @@ UnoFileControl::UnoFileControl()
 //	----------------------------------------------------
 //	class ImageProducerControlModel
 //	----------------------------------------------------
+
+ImageProducerControlModel::ImageProducerControlModel()  : mbAdjustingImagePosition( false ), mbAdjustingGraphic( false ), mbAdjustingLinkProvider( false ) 
+{
+	ImplRegisterProperty( BASEPROPERTY_GRAPHICLINKPROVIDER );
+}
+
+ImageProducerControlModel::ImageProducerControlModel( const ImageProducerControlModel& _rSource )  : awt::XImageProducer(), UnoControlModel( _rSource ), mbAdjustingImagePosition( false ), mbAdjustingGraphic( false ), mbAdjustingLinkProvider( false ) 
+{
+	ImplRegisterProperty( BASEPROPERTY_GRAPHICLINKPROVIDER );
+}
+
 uno::Any SAL_CALL ImageProducerControlModel::queryInterface( const uno::Type & rType ) throw(uno::RuntimeException)
 {
     return UnoControlModel::queryInterface( rType );
@@ -578,7 +621,7 @@ uno::Any ImageProducerControlModel::Impl
 }
 namespace
 {
-    uno::Reference< graphic::XGraphic > lcl_getGraphicFromURL_nothrow( const ::rtl::OUString& _rURL )
+    uno::Reference< graphic::XGraphic > lcl_getGraphicFromURL_nothrow( const ::rtl::OUString& _rURL, const uno::Reference< embed::XStorage >& rxStorage )
     {
         uno::Reference< graphic::XGraphic > xGraphic;
         if ( !_rURL.getLength() )
@@ -593,18 +636,34 @@ namespace
                 uno::Sequence< beans::PropertyValue > aMediaProperties(1);
                 aMediaProperties[0].Name = ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "URL" ) );
                 aMediaProperties[0].Value <<= _rURL;
+                if ( rxStorage.is() && ::utl::SvPictureStreamHelper::IsPackagePictureURL( _rURL ) )
+                { 
+                    sal_Int32 nOldLen = aMediaProperties.getLength();
+                    aMediaProperties.realloc( nOldLen + 1 );
+                    aMediaProperties[ nOldLen ].Name = ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "Storage" ) );
+                    aMediaProperties[ nOldLen ].Value <<= rxStorage;
+                }
                 xGraphic = xProvider->queryGraphic( aMediaProperties );
             }
         }
         catch( const Exception& )
         {
+		OSL_TRACE("unhandled details URL %s Storage is ( %d )", rtl::OUStringToOString( _rURL, RTL_TEXTENCODING_UTF8 ).getStr(), xStorage.is() );	
         	DBG_UNHANDLED_EXCEPTION();
         }
 
         return xGraphic;
     }
+
+    uno::Reference< graphic::XGraphic > lcl_getGraphicFromURL_nothrow( const ::rtl::OUString& _rURL )
+    {
+        uno::Reference< embed::XStorage > xStorage;
+        return lcl_getGraphicFromURL_nothrow( _rURL, xStorage );
+    }
+
 }
 
+
 void SAL_CALL ImageProducerControlModel::setFastPropertyValue_NoBroadcast( sal_Int32 nHandle, const ::com::sun::star::uno::Any& rValue ) throw (::com::sun::star::uno::Exception)
 {
     UnoControlModel::setFastPropertyValue_NoBroadcast( nHandle, rValue );
@@ -615,17 +674,45 @@ void SAL_CALL ImageProducerControlModel:
     {
         switch ( nHandle )
         {
+ 		// someone wants to set an image that is carried in the
+		// associated storage ( e.g the document )
+	        case BASEPROPERTY_GRAPHICLINKPROVIDER:
+			{
+				uno::Reference< graphic::XGraphicLinkProvider > xProv;
+				OSL_VERIFY( rValue >>= xProv );
+				rtl::OUString sURL;
+				if ( xProv.is() )
+				{
+					sURL = xProv->getSourceURL();
+					if ( sURL.getLength() )
+					{
+						mbAdjustingLinkProvider = true;
+						setPropertyValue( GetPropertyName( BASEPROPERTY_IMAGEURL ), uno::makeAny(  sURL ) );
+						mbAdjustingLinkProvider = false;
+					}
+				}
+				break;
+			}
+
         case BASEPROPERTY_IMAGEURL:
+        {
+            if ( !mbAdjustingLinkProvider && ImplHasProperty( BASEPROPERTY_GRAPHICLINKPROVIDER ) )
+                // reset the GraphicLinkProvider ( null ) if someone is setting
+                // the external ImageURL property directly.
+                setPropertyValue( GetPropertyName( BASEPROPERTY_GRAPHICLINKPROVIDER ), uno::makeAny(  uno::Reference< graphic::XGraphicLinkProvider >()  ) );
             if ( !mbAdjustingGraphic && ImplHasProperty( BASEPROPERTY_GRAPHIC ) )
             {
                 mbAdjustingGraphic = true;
                 ::rtl::OUString sImageURL;
                 OSL_VERIFY( rValue >>= sImageURL );
-                setPropertyValue( GetPropertyName( BASEPROPERTY_GRAPHIC ), uno::makeAny( lcl_getGraphicFromURL_nothrow( sImageURL ) ) );
+                uno::Reference< graphic::XGraphicLinkProvider > xProv;
+                if ( ImplHasProperty( BASEPROPERTY_GRAPHICLINKPROVIDER ) )
+                    getPropertyValue( GetPropertyName( BASEPROPERTY_GRAPHICLINKPROVIDER ) ) >>= xProv;
+                setPropertyValue( GetPropertyName( BASEPROPERTY_GRAPHIC ), uno::makeAny( lcl_getGraphicFromURL_nothrow( sImageURL, xProv.is() ? xProv->getStorage() : NULL ) ) );
                 mbAdjustingGraphic = false;
             }
             break;
-
+        }
         case BASEPROPERTY_GRAPHIC:
             if ( !mbAdjustingGraphic && ImplHasProperty( BASEPROPERTY_IMAGEURL ) )
             {
@@ -676,8 +763,18 @@ void ImageProducerControlModel::removeCo
 
 void ImageProducerControlModel::startProduction(  ) throw (::com::sun::star::uno::RuntimeException)
 {
-	uno::Sequence<uno::Any> aArgs(1);
+	uno::Sequence<uno::Any> aArgs(2);
 	aArgs.getArray()[0] = getPropertyValue( GetPropertyName( BASEPROPERTY_IMAGEURL ) );
+
+	uno::Reference< graphic::XGraphicLinkProvider > xProv;
+	if ( ImplHasProperty( BASEPROPERTY_GRAPHICLINKPROVIDER ) )
+			getPropertyValue( GetPropertyName( BASEPROPERTY_GRAPHICLINKPROVIDER ) ) >>= xProv;
+
+	if ( xProv.is() )
+		aArgs[1] <<= xProv->getStorage();
+	else
+		aArgs.realloc( 1 );
+
 	uno::Reference< lang::XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
 	uno::Reference< awt::XImageProducer > xImageProducer( xMSF->createInstanceWithArguments( ::rtl::OUString::createFromAscii( "com.sun.star.awt.ImageProducer" ), aArgs ), uno::UNO_QUERY );
 	if ( xImageProducer.is() )
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/helper/property.cxx toolkit/source/helper/property.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/helper/property.cxx	2008-07-09 13:18:03.000000000 +0100
+++ toolkit/source/helper/property.cxx	2008-07-07 17:34:34.000000000 +0100
@@ -53,6 +53,7 @@
 #include <com/sun/star/util/XNumberFormatsSupplier.hpp>
 #include <com/sun/star/beans/PropertyAttribute.hpp>
 #include <com/sun/star/graphic/XGraphic.hpp>
+#include <com/sun/star/graphic/XGraphicLinkProvider.hpp>
 #include <com/sun/star/resource/XStringResourceResolver.hpp>
 #include <comphelper/types.hxx>
 #include <functional>
@@ -180,6 +181,7 @@ ImplPropertyInfo* ImplGetPropertyInfos( 
 
             DECL_PROP_2     ( "Graphic",                GRAPHIC,            Reference< ::com::sun::star::graphic::XGraphic >, BOUND, TRANSIENT ),
             DECL_PROP_2     ( "GroupName",              GROUPNAME,          ::rtl::OUString,    BOUND, MAYBEDEFAULT ),
+            DECL_PROP_2     ( "GraphicLinkProvider",              GRAPHICLINKPROVIDER,          Reference< ::com::sun::star::graphic::XGraphicLinkProvider >,    BOUND, MAYBEDEFAULT ),
             DECL_PROP_2     ( "HelpText",               HELPTEXT,           ::rtl::OUString,    BOUND, MAYBEDEFAULT ),
             DECL_PROP_2     ( "HelpURL",                HELPURL,            ::rtl::OUString,    BOUND, MAYBEDEFAULT ),
             DECL_PROP_2     ( "HideInactiveSelection",  HIDEINACTIVESELECTION, bool,            BOUND, MAYBEDEFAULT ),
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/helper/registerservices.cxx toolkit/source/helper/registerservices.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/helper/registerservices.cxx	2008-07-09 13:18:24.000000000 +0100
+++ toolkit/source/helper/registerservices.cxx	2008-07-07 18:09:17.000000000 +0100
@@ -198,6 +198,7 @@ IMPL_CREATEINSTANCE( UnoSimpleAnimationC
 IMPL_CREATEINSTANCE( UnoSimpleAnimationControlModel )
 IMPL_CREATEINSTANCE( UnoThrobberControl )
 IMPL_CREATEINSTANCE( UnoThrobberControlModel )
+IMPL_CREATEINSTANCE( GraphLinkProviderImpl )
 
 extern ::com::sun::star::uno::Reference< ::com::sun::star::uno::XInterface > SAL_CALL TreeControl_CreateInstance( const ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory >& );
 extern ::com::sun::star::uno::Reference< ::com::sun::star::uno::XInterface > SAL_CALL TreeControlModel_CreateInstance( const ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory >& );
@@ -287,6 +288,7 @@ TOOLKIT_DLLPUBLIC sal_Bool SAL_CALL comp
 		registerServices( xRegistryKey, "UnoThrobberControl", szServiceName_UnoThrobberControl );
         registerServices( xRegistryKey, "UnoFixedHyperlinkControl", szServiceName_UnoControlFixedHyperlink );
         registerServices( xRegistryKey, "UnoControlFixedHyperlinkModel", szServiceName_UnoControlFixedHyperlinkModel );
+        registerServices( xRegistryKey, "GraphLinkProviderImpl", szServiceName_GraphLinkProvider );
 
         comp_AsyncCallback_component_writeInfo( _pServiceManager, _pRegistryKey );
         comp_Layout_component_writeInfo( _pServiceManager, _pRegistryKey );
@@ -369,6 +371,7 @@ TOOLKIT_DLLPUBLIC void* SAL_CALL compone
         CHECKANDCREATEFACTORY( UnoThrobberControl, szServiceName_UnoThrobberControl, NULL )
         CHECKANDCREATEFACTORY( UnoFixedHyperlinkControl, szServiceName_UnoControlFixedHyperlink, NULL )
         CHECKANDCREATEFACTORY( UnoControlFixedHyperlinkModel, szServiceName_UnoControlFixedHyperlinkModel, NULL )
+        CHECKANDCREATEFACTORY( GraphLinkProviderImpl, szServiceName_GraphLinkProvider, NULL )
 
     	if ( rtl_str_compare( sImplementationName, "com.sun.star.awt.comp.AsyncCallback" ) == 0 )
             return comp_AsyncCallback_component_getFactory( sImplementationName, _pServiceManager, _pRegistryKey );
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/helper/servicenames.cxx toolkit/source/helper/servicenames.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/toolkit/source/helper/servicenames.cxx	2008-04-11 10:37:36.000000000 +0100
+++ toolkit/source/helper/servicenames.cxx	2008-07-07 17:37:51.000000000 +0100
@@ -101,4 +101,4 @@ const sal_Char __FAR_DATA szServiceName_
 const sal_Char __FAR_DATA szServiceName_UnoThrobberControl[] = "com.sun.star.awt.UnoThrobberControl";
 const sal_Char __FAR_DATA szServiceName_UnoControlFixedHyperlink[] = "com.sun.star.awt.UnoControlFixedHyperlink";
 const sal_Char __FAR_DATA szServiceName_UnoControlFixedHyperlinkModel[] = "com.sun.star.awt.UnoControlFixedHyperlinkModel";
-
+const sal_Char __FAR_DATA szServiceName_GraphLinkProvider[] = "com.sun.star.awt.GraphicLinkProvider";
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/xmloff/inc/xmloff/xmlexp.hxx xmloff/inc/xmloff/xmlexp.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/xmloff/inc/xmloff/xmlexp.hxx	2008-04-15 15:17:24.000000000 +0100
+++ xmloff/inc/xmloff/xmlexp.hxx	2008-07-09 10:00:11.000000000 +0100
@@ -42,6 +42,7 @@
 #include <com/sun/star/xml/sax/XDocumentHandler.hpp>
 #include <com/sun/star/xml/sax/XAttributeList.hpp>
 #include <com/sun/star/xml/sax/XLocator.hpp>
+#include <com/sun/star/graphic/XGraphic.hpp>
 #include <com/sun/star/util/XNumberFormatsSupplier.hpp>
 #include <com/sun/star/lang/XUnoTunnel.hpp>
 #include <rtl/ustring.hxx>
@@ -57,6 +58,7 @@
 #include <com/sun/star/document/XFilter.hpp>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <com/sun/star/document/XExporter.hpp>
+#include <com/sun/star/document/XGraphicResolver.hpp>
 #ifndef _COM_SUN_STAR_DRAWING_XGRAPHICOBJECTRESOLVER_HPP_
 #include <com/sun/star/document/XGraphicObjectResolver.hpp>
 #endif
@@ -461,6 +463,8 @@ public:
 
 	::rtl::OUString AddEmbeddedGraphicObject(
 							const ::rtl::OUString& rGraphicObjectURL );
+	::rtl::OUString AddEmbeddedGraphicObject(
+							const com::sun::star::uno::Reference< com::sun::star::graphic::XGraphic >& rxGraphic );
     sal_Bool AddEmbeddedGraphicObjectAsBase64(
                             const ::rtl::OUString& rGraphicObjectURL );
 
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/core/xmlexp.cxx xmloff/source/core/xmlexp.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/core/xmlexp.cxx	2008-05-28 11:08:31.000000000 +0100
+++ xmloff/source/core/xmlexp.cxx	2008-07-09 10:02:45.000000000 +0100
@@ -1774,6 +1774,15 @@ sal_Int32 SvXMLExport::dataStyleForceSys
 }
 
 
+OUString SvXMLExport::AddEmbeddedGraphicObject( const Reference< graphic::XGraphic >& rxGraphic )
+{
+    rtl::OUString sRes; 
+    Reference< document::XGraphicResolver > xResolv( mxGraphicResolver, UNO_QUERY );
+    if ( xResolv.is() )
+        sRes = xResolv->resolveGraphicObject( rxGraphic );
+    return sRes;
+}
+
 OUString SvXMLExport::AddEmbeddedGraphicObject( const OUString& rGraphicObjectURL )
 {
 	OUString sRet( rGraphicObjectURL );
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/elementimport.cxx xmloff/source/forms/elementimport.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/elementimport.cxx	2008-07-09 13:18:29.000000000 +0100
+++ xmloff/source/forms/elementimport.cxx	2008-07-07 23:10:54.000000000 +0100
@@ -47,6 +47,7 @@
 #include "gridcolumnproptranslator.hxx"
 #include <comphelper/extract.hxx>
 #include <comphelper/types.hxx>
+#include <unotools/ucbstreamhelper.hxx>
 
 /** === begin UNO includes === **/
 #include <com/sun/star/text/XText.hpp>
@@ -59,7 +60,7 @@
 /** === end UNO includes === **/
 #include <tools/urlobj.hxx>
 #include <tools/time.hxx>
-
+#include <comphelper/storagehelper.hxx>
 #include <algorithm>
 #include <functional>
 
@@ -979,6 +980,13 @@ namespace xmloff
                 _rValue, OEnumMapper::getEnumMap( OEnumMapper::epImageAlign )
             ) >>= m_nImageAlign );
         }
+        else if ( _rLocalName == GetXMLToken( XML_IMAGE_DATA ) && ::utl::SvPictureStreamHelper::IsPackagePictureURL( _rValue ) )
+        {
+            uno::Sequence< uno::Any > aArgs( 2 ); 
+            aArgs[ 0 ] = uno::makeAny( _rValue );
+            aArgs[ 1 ] = uno::makeAny(  m_rContext.getGlobalContext().GetModel() );
+            m_xProv.set( m_rContext.getGlobalContext().getServiceFactory()->createInstanceWithArguments( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.awt.GraphicLinkProvider" ) ), aArgs ), uno::UNO_QUERY );
+        }
         else
             OControlImport::handleAttribute( _nNamespaceKey, _rLocalName, _rValue );
     }
@@ -1004,6 +1012,25 @@ namespace xmloff
 		    implPushBackPropertyValue( aImagePosition );
         }
 	}
+        void OImagePositionImport::EndElement()
+        {
+            OSL_TRACE("** Image::EndElement");
+            if ( m_xElement.is () )
+            {
+                OSL_TRACE("**** About to set the attribute xProv is %d", m_xProv.is() );
+                try
+                {
+                    m_xElement->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("GraphicLinkProvider" ) ), uno::makeAny( m_xProv ) );
+                }
+                catch(uno::Exception& e)
+                {
+                    OSL_TRACE("Caught exception %s", rtl::OUStringToOString( e.Message, RTL_TEXTENCODING_UTF8 ).getStr() );
+                }
+            }
+            else
+                OSL_TRACE("Phreaking null");
+            OControlImport::EndElement();
+        }
 
     //=====================================================================
 	//= OReferredControlImport
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/elementimport.hxx xmloff/source/forms/elementimport.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/elementimport.hxx	2008-04-10 23:01:44.000000000 +0100
+++ xmloff/source/forms/elementimport.hxx	2008-07-07 20:23:21.000000000 +0100
@@ -41,6 +41,7 @@
 #include <com/sun/star/lang/XMultiServiceFactory.hpp>
 #include <com/sun/star/form/XGridColumnFactory.hpp>
 #include <com/sun/star/script/XEventAttacherManager.hpp>
+#include <com/sun/star/graphic/XGraphicLinkProvider.hpp>
 /** === end UNO includes === **/
 #include <comphelper/stl_types.hxx>
 #include "eventimport.hxx"
@@ -282,7 +283,7 @@ namespace xmloff
         sal_Int16   m_nImagePosition;
         sal_Int16   m_nImageAlign;
         sal_Bool    m_bHaveImagePosition;
-
+	::com::sun::star::uno::Reference< ::com::sun::star::graphic::XGraphicLinkProvider > m_xProv;
     public:
 		OImagePositionImport(
 			IFormsImportContext& _rImport, IEventAttacherManager& _rEventManager, sal_uInt16 _nPrefix, const ::rtl::OUString& _rName,
@@ -294,7 +295,7 @@ namespace xmloff
 		// SvXMLImportContext overridables
 		virtual void StartElement(
 			const ::com::sun::star::uno::Reference< ::com::sun::star::xml::sax::XAttributeList >& _rxAttrList);
-
+		virtual void EndElement();
         // OPropertyImport overridables
 		virtual void	handleAttribute( sal_uInt16 _nNamespaceKey,
 			const ::rtl::OUString& _rLocalName,
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/propertyexport.cxx xmloff/source/forms/propertyexport.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/propertyexport.cxx	2008-04-10 23:10:41.000000000 +0100
+++ xmloff/source/forms/propertyexport.cxx	2008-07-09 12:10:11.000000000 +0100
@@ -44,6 +44,10 @@
 #include <com/sun/star/util/Date.hpp>
 #include <com/sun/star/util/Time.hpp>
 #include <com/sun/star/util/DateTime.hpp>
+#include <com/sun/star/graphic/XGraphicLinkProvider.hpp>
+#include <com/sun/star/graphic/XGraphic.hpp>
+#include <unotools/ucbstreamhelper.hxx>
+
 #include <osl/diagnose.h>
 #include <comphelper/extract.hxx>
 #include <comphelper/sequence.hxx>
@@ -62,6 +66,7 @@ namespace xmloff
 	using namespace ::com::sun::star::uno;
 	using namespace ::com::sun::star::lang;
 	using namespace ::com::sun::star::beans;
+	using namespace ::com::sun::star::graphic;
 
 	// NO using namespace ...util !!!
 	// need a tools Date/Time/DateTime below, which would conflict with the uno types then
@@ -437,6 +442,35 @@ namespace xmloff
 		exportedProperty(_sPropertyName);
 	}
 	//---------------------------------------------------------------------
+
+	void OPropertyExport::exportImageDataAttribute()
+        { 
+            rtl::OUString sURL;
+            m_xProps->getPropertyValue( PROPERTY_IMAGEURL ) >>= sURL;
+            Reference< XGraphicLinkProvider > xProv;
+            Reference< XGraphic> xGraphic;
+            m_xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "GraphicLinkProvider" ) ) ) >>= xProv;
+            if ( xProv.is() ) 
+            {
+                   m_xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "Graphic" ) ) ) >>= xGraphic;
+                   rtl::OUString newURL = m_rContext.getGlobalContext().AddEmbeddedGraphicObject( xGraphic );
+                   // stick on the Package scheme if we got a URL back
+                   if ( newURL.getLength() )
+                   {
+                       newURL = ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "vnd.sun.star.Package:" ) ) + newURL;
+                       // reset things
+                       Sequence< Any > aArgs( 2 );
+                       aArgs[ 0 ] = makeAny( newURL );
+                       aArgs[ 1 ] = makeAny(  m_rContext.getGlobalContext().GetModel() );
+                       // set up new GraphicProvider because we have a new
+                       // intra document URL
+                       xProv.set( m_rContext.getGlobalContext().getServiceFactory()->createInstanceWithArguments( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.awt.GraphicLinkProvider" ) ), aArgs ), UNO_QUERY );
+                       m_xProps->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("GraphicLinkProvider" ) ), makeAny( xProv ) );
+                   }
+            }
+            exportRelativeTargetLocation(PROPERTY_IMAGEURL,CCA_IMAGE_DATA);
+        }
+
 	void OPropertyExport::flagStyleProperties()
 	{
 		// flag all the properties which are part of the style as "handled"
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/propertyexport.hxx xmloff/source/forms/propertyexport.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/xmloff/source/forms/propertyexport.hxx	2008-04-10 23:11:00.000000000 +0100
+++ xmloff/source/forms/propertyexport.hxx	2008-07-08 13:11:03.000000000 +0100
@@ -242,7 +242,7 @@ namespace xmloff
 
 			<p>The property needs a special handling because the URL's need to be made relative</p>
 		*/
-		inline void exportImageDataAttribute() { exportRelativeTargetLocation(PROPERTY_IMAGEURL,CCA_IMAGE_DATA); }
+		void exportImageDataAttribute();
 
 		/** flag the style properties as 'already exported'
 
Only in basctl/source/basicide: localize.sdf
Only in basctl/source/dlged: localize.sdf
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/basctl/source/dlged/propbrw.cxx basctl/source/dlged/propbrw.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/basctl/source/dlged/propbrw.cxx	2008-04-11 12:05:43.000000000 +0100
+++ basctl/source/dlged/propbrw.cxx	2008-07-08 09:32:28.000000000 +0100
@@ -209,7 +209,8 @@ void PropBrw::ImplReCreateController()
         ::cppu::ContextEntry_Init aHandlerContextInfo[] =
         {
             ::cppu::ContextEntry_Init( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "DialogParentWindow" ) ), makeAny( VCLUnoHelper::GetInterface ( this ) ) ),
-            ::cppu::ContextEntry_Init( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "ContextDocument" ) ), makeAny( m_xContextDocument ) )
+            ::cppu::ContextEntry_Init( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "ContextDocument" ) ), makeAny( m_xContextDocument ) ),
+            ::cppu::ContextEntry_Init( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "SupportNonLinkedImage" ) ), makeAny( sal_False ) )
         };
         Reference< XComponentContext > xInspectorContext(
             ::cppu::createComponentContext( aHandlerContextInfo, sizeof( aHandlerContextInfo ) / sizeof( aHandlerContextInfo[0] ),
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/extensions/source/propctrlr/formcomponenthandler.cxx extensions/source/propctrlr/formcomponenthandler.cxx
--- /data4/M21/ooo-build/build-src/dev300-m21/extensions/source/propctrlr/formcomponenthandler.cxx	2008-06-16 13:42:41.000000000 +0100
+++ extensions/source/propctrlr/formcomponenthandler.cxx	2008-07-08 12:09:44.000000000 +0100
@@ -1360,10 +1360,20 @@ namespace pcr
             break;
 
         case PROPERTY_ID_IMAGE_URL:
-            if ( impl_browseForImage_nothrow( _rData, aGuard ) )
-                eResult = InteractiveSelectionResult_ObtainedValue;
+        {
+            bool bIsLink = false;
+            if ( impl_browseForImage_nothrow( _rData, bIsLink, aGuard ) )
+            {
+                if ( bIsLink )
+                    eResult = InteractiveSelectionResult_ObtainedValue;
+                else 
+                {
+                    if ( impl_handleNonImageLink_nothrow( _rData ) )
+                        eResult = InteractiveSelectionResult_Success;
+                }
+            }
             break;
-
+        }
         case PROPERTY_ID_TARGET_URL:
             if ( impl_browseForTargetURL_nothrow( _rData, aGuard ) )
                 eResult = InteractiveSelectionResult_ObtainedValue;
@@ -2656,14 +2666,31 @@ namespace pcr
     }
 
     //------------------------------------------------------------------------
-    bool FormComponentPropertyHandler::impl_browseForImage_nothrow( Any& _out_rNewValue, ::osl::ClearableMutexGuard& _rClearBeforeDialog ) const
+    bool FormComponentPropertyHandler::impl_browseForImage_nothrow( Any& _out_rNewValue, bool& bIsLink, ::osl::ClearableMutexGuard& _rClearBeforeDialog ) const
     {
+        bIsLink = false; // default
         ::rtl::OUString aStrTrans = m_pInfoService->getPropertyTranslation( PROPERTY_ID_IMAGE_URL );
 
         ::sfx2::FileDialogHelper aFileDlg(SFXWB_GRAPHIC);
 
         aFileDlg.SetTitle(aStrTrans);
-
+	sal_Bool bHandleNonLink = sal_False;
+	Reference< XModel > xDocument( impl_getContextDocument_nothrow() );
+	if ( xDocument.is() )
+	{
+		bHandleNonLink = sal_True; // in the ideal world this should be enough
+		// Bit of a hack to allow basic dialog to override support for 
+		// non-linked images ( e.g. those located in the document 
+		// stream )  However, the toolkit control CAN handle this but 
+		// the dialog code. Well, can't ( right now ) hopefully it 
+		// will soon
+		try
+		{	m_aContext.getContextValueByAsciiName( "SupportNonLinkedImage" ) >>= bHandleNonLink;
+		}
+		catch( Exception& e )  // #TODO add that debug utils thing for uncaught exceptions
+		{
+		}
+	}
         Reference< XFilePickerControlAccess > xController(aFileDlg.GetFilePicker(), UNO_QUERY);
         DBG_ASSERT(xController.is(), "FormComponentPropertyHandler::impl_browseForImage_nothrow: missing the controller interface on the file picker!");
         if (xController.is())
@@ -2672,8 +2699,8 @@ namespace pcr
             xController->setValue(ExtendedFilePickerElementIds::CHECKBOX_PREVIEW, 0, ::cppu::bool2any(sal_True));
 
             // "as link" is checked, but disabled
-            xController->setValue(ExtendedFilePickerElementIds::CHECKBOX_LINK, 0, ::cppu::bool2any(sal_True));
-            xController->enableControl(ExtendedFilePickerElementIds::CHECKBOX_LINK, sal_False);
+            xController->setValue(ExtendedFilePickerElementIds::CHECKBOX_LINK, 0, ::cppu::bool2any(!bHandleNonLink));
+            xController->enableControl(ExtendedFilePickerElementIds::CHECKBOX_LINK, bHandleNonLink );
         }
 
         ::rtl::OUString sCurValue;
@@ -2687,7 +2714,13 @@ namespace pcr
         _rClearBeforeDialog.clear();
         bool bSuccess = ( 0 == aFileDlg.Execute() );
         if ( bSuccess )
+        {
+            if ( bHandleNonLink && xController.is() )
+            {
+                xController->getValue(ExtendedFilePickerElementIds::CHECKBOX_LINK, 0) >>= bIsLink;
+            }
             _out_rNewValue <<= (::rtl::OUString)aFileDlg.GetPath();
+        }
         return bSuccess;
     }
 
@@ -3174,6 +3207,31 @@ namespace pcr
         }
         return sURL;
     }
+
+    bool FormComponentPropertyHandler::impl_handleNonImageLink_nothrow( const Any& aURL )
+    {
+        Reference< XMultiComponentFactory > xMcf = m_aContext.getUNOContext()->getServiceManager();
+
+        Sequence< Any > aArgs(2);
+        aArgs[ 0 ] = aURL;
+        aArgs[ 1 ] <<= impl_getContextDocument_nothrow();
+
+        Reference< XInterface > xLinkProvider = xMcf->createInstanceWithArgumentsAndContext( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.awt.GraphicLinkProvider" ) ), aArgs, m_aContext.getUNOContext() );
+        bool bRes = false;
+        if ( xLinkProvider.is() )
+        {
+            try
+            {
+                m_xComponent->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "GraphicLinkProvider" ) ), makeAny( xLinkProvider ) );
+                bRes = true;
+            }
+            catch( const Exception& )
+            {
+                DBG_UNHANDLED_EXCEPTION();
+            }
+        }
+        return bRes;
+    }
     // -------------------------------------------------------------------------
     ::cppu::IPropertyArrayHelper* FormComponentPropertyHandler::createArrayHelper( ) const
     {
diff -rup /data4/M21/ooo-build/build-src/dev300-m21/extensions/source/propctrlr/formcomponenthandler.hxx extensions/source/propctrlr/formcomponenthandler.hxx
--- /data4/M21/ooo-build/build-src/dev300-m21/extensions/source/propctrlr/formcomponenthandler.hxx	2008-05-20 20:11:31.000000000 +0100
+++ extensions/source/propctrlr/formcomponenthandler.hxx	2008-07-08 12:06:01.000000000 +0100
@@ -309,7 +309,7 @@ namespace pcr
                 <TRUE/> if and only if a new image URL has been chosen by the user.
                 In this case, ->_out_rNewValue is filled with the new property value
         */
-        bool impl_browseForImage_nothrow( ::com::sun::star::uno::Any& _out_rNewValue, ::osl::ClearableMutexGuard& _rClearBeforeDialog ) const;
+        bool impl_browseForImage_nothrow( ::com::sun::star::uno::Any& _out_rNewValue, bool& bIsLink, ::osl::ClearableMutexGuard& _rClearBeforeDialog ) const;
 
         /** executes a dialog which allows the user to change the TargetURL property of
             our component
@@ -438,6 +438,8 @@ namespace pcr
         */
         ::rtl::OUString impl_getDocumentURL_nothrow() const;
 
+        bool impl_handleNonImageLink_nothrow( const ::com::sun::star::uno::Any& aURL );
+
     private:
         DECL_LINK( OnDesignerClosed, void* );
 
Only in extensions/source/propctrlr/: localize.sdf
--- /dev/null	2008-04-22 00:28:44.000000000 +0100
+++ offapi/com/sun/star/document/XGraphicResolver.idl	2008-07-09 09:33:53.000000000 +0100
@@ -0,0 +1,64 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: XGraphicObjectResolver.idl,v $
+ * $Revision: 1.7 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+#ifndef __com_sun_star_document_XGraphicResolver_idl__ 
+#define __com_sun_star_document_XGraphicResolver_idl__ 
+ 
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __com_sun_star_document_XGraphicObjectResolver_idl__ 
+#include <com/sun/star/document/XGraphicObjectResolver.idl>
+#endif
+#ifndef __com_sun_star_graphic_XGraphic_idl__ 
+#include <com/sun/star/graphic/XGraphic.idl>
+#endif
+//============================================================================= 
+ 
+module com {  module sun {  module star {  module document {  
+ 
+//============================================================================= 
+ 
+/** this interface converts graphic object from one URL space to another.
+ */
+published interface XGraphicResolver
+{ 
+	interface com::sun::star::document::XGraphicObjectResolver;
+	/** converts the given XGraphic from the source URL namespace to the destination
+		URL space of this instance.
+	*/
+	string resolveGraphicObject( [in] ::com::sun::star::graphic::XGraphic gObj );
+}; 
+ 
+//============================================================================= 
+ 
+}; }; }; };  
+ 
+#endif 
--- /dev/null	2008-04-22 00:28:44.000000000 +0100
+++ offapi/com/sun/star/graphic/XGraphicLinkProvider.idl	2008-07-07 17:45:45.000000000 +0100
@@ -0,0 +1,46 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: XGraphicProvider.idl,v $
+ * $Revision: 1.5 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef com_sun_star_graphic_XGraphicLinkProvider_idl
+#define com_sun_star_graphic_XGraphicLinkProvider_idl
+
+#include <com/sun/star/uno/XInterface.idl>
+#include <com/sun/star/embed/XStorage.idl>
+
+module com { module sun { module star { module graphic
+{
+interface XGraphicLinkProvider : ::com::sun::star::uno::XInterface
+{
+    [readonly, attribute] com::sun::star::embed::XStorage Storage;
+    [readonly, attribute] string SourceURL;
+} ; } ; } ; } ; 
+};
+
+#endif
