unchanged:
--- vcl/inc/salframe.hxx	24 Apr 2003 15:16:48 -0000	1.16
+++ vcl/inc/salframe.hxx	3 Mar 2004 15:32:07 -0000
@@ -91,6 +91,14 @@
 #include <tools/gen.hxx>
 #endif
 
+#ifndef _SV_REGION_HXX
+#include <region.hxx>
+#endif
+
+#ifndef _RTL_USTRING_HXX_
+#include <rtl/ustring.hxx>
+#endif
+
 #ifdef __cplusplus
 
 class AllSettings;
@@ -159,6 +167,8 @@ struct SystemEnvData;
 #define SAL_FRAME_POSSIZE_HEIGHT            ((USHORT)0x0008)
 
 #ifdef __cplusplus
+
+using namespace rtl;
 
 // ------------
 // - SalFrame -
