unchanged:
--- vcl/inc/ctrl.hxx	16 May 2002 11:52:49 -0000	1.5
+++ vcl/inc/ctrl.hxx	3 Mar 2004 15:32:06 -0000
@@ -73,6 +73,10 @@
 #include <window.hxx>
 #endif
 
+#ifndef _SV_NATIVEWIDGETS_HXX
+#include <salnativewidgets.hxx>
+#endif
+
 // forward
 namespace vcl { struct ControlLayoutData; }
 
unchanged:
--- vcl/source/control/ctrl.cxx	4 Jun 2003 11:21:43 -0000	1.10
+++ vcl/source/control/ctrl.cxx	3 Mar 2004 15:32:07 -0000
@@ -79,6 +79,9 @@
 #ifndef _VCL_CONTROLLAYOUT_HXX
 #include <controllayout.hxx>
 #endif
+#ifndef _SV_NATIVEWIDGETS_HXX
+#include <salnativewidgets.hxx>
+#endif
 
 #pragma hdrstop
 
