Index: sal/osl/unx/signal.c
===================================================================
RCS file: /cvs/oo/porting/sal/osl/unx/signal.c,v
retrieving revision 1.7.14.1
diff -u -p -u -r1.7.14.1 signal.c
--- sal/osl/unx/signal.c	28 Oct 2002 07:33:00 -0000	1.7.14.1
+++ sal/osl/unx/signal.c	7 May 2003 15:23:28 -0000
@@ -188,6 +188,10 @@ static sal_Bool InitSignal()
 		bSetILLHandler = sal_True;
 	}
 
+	if ( getenv ("GNOME_DISABLE_CRASH_DIALOG") &&
+	     atoi (getenv ("GNOME_DISABLE_CRASH_DIALOG")) )
+		bSetSEGVHandler = bSetWINCHHandler = bSetILLHandler = bDoHardKill = sal_False;
+
 	SignalListMutex = osl_createMutex();
 
 	act.sa_handler = SignalHandlerFunction;
@@ -251,6 +255,46 @@ static sal_Bool DeInitSignal()
 	return sal_False;
 }
 
+static void
+ReportCrash( int Signal )
+{
+	int i, exec_report;
+	struct sigaction act;
+	char szShellCmd[512];
+	static sal_Bool bCrashReporterExecuted = sal_False;
+
+	fprintf (stderr, "ReportCrash %d (%d)\n",
+		 Signal, bCrashReporterExecuted);
+
+	if ( bCrashReporterExecuted )
+		return;
+
+	if ( getenv ("GNOME_DISABLE_CRASH_DIALOG") &&
+	     atoi (getenv ("GNOME_DISABLE_CRASH_DIALOG")) )
+		return;
+	     
+	fprintf (stderr, "Signal '%d'\n", Signal);
+	
+	exec_report = 0;
+
+	if (Signal < 0)
+		exec_report = 1;
+	else
+		for (i = 0; i < NoSignals; i++) {
+			if (Signals[i].Signal == Signal &&
+			    Signals[i].Action == ACT_ABORT )
+				exec_report = 1;
+		}
+
+	if (exec_report) {
+		// FIXME: we quite badly need to ungrab the mouse at this stage.
+		bCrashReporterExecuted = sal_True;
+		snprintf( szShellCmd, sizeof(szShellCmd)/sizeof(szShellCmd[0]),
+			  "bug-buddy --appname=soffice.bin --pid=%d --package=OpenOffice.org", getpid() );
+		system( szShellCmd );
+	}
+}
+
 static oslSignalAction CallSignalHandler(oslSignalInfo *pInfo)
 {
 	oslSignalHandlerImpl* pHandler = SignalList;
@@ -293,7 +337,8 @@ static void CallSystemHandler(int Signal
 					_exit(255);
 					break;
 
-				case ACT_ABORT:		/* terminate witch core dump */
+				case ACT_ABORT:		/* terminate with core dump */
+					ReportCrash( -1 );
 					act.sa_handler = SIG_DFL;
 					act.sa_flags   = 0;
 					sigemptyset(&(act.sa_mask));
@@ -324,6 +369,8 @@ static void SignalHandlerFunction(int Si
 	Info.UserSignal = Signal;
 	Info.UserData   = NULL;
 
+	fprintf (stderr, "SignalHandlerFunction %d\n", Signal);
+
 	switch (Signal)
 	{
 		case SIGBUS:
@@ -356,6 +403,8 @@ static void SignalHandlerFunction(int Si
 			break; 
 	}
 
+	ReportCrash( Signal );
+
 	/* Portal Demo HACK !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!*/
 	if (bDoHardKill && (Info.Signal == osl_Signal_AccessViolation))
 		_exit(255);
@@ -368,6 +417,7 @@ static void SignalHandlerFunction(int Si
 		break;
 
 	case osl_Signal_ActAbortApp:
+		ReportCrash( -1 );
 		act.sa_handler = SIG_DFL;
 		act.sa_flags   = 0;
 		sigemptyset(&(act.sa_mask));
