Index: solenv/inc/unxlngi4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi4.mk,v
retrieving revision 1.13
diff -u -r1.13 unxlngi4.mk
--- solenv/inc/unxlngi4.mk	4 Mar 2002 15:53:59 -0000	1.13
+++ solenv/inc/unxlngi4.mk	25 Nov 2002 12:31:47 -0000
@@ -148,7 +148,7 @@
 # name of linker
 LINK=gcc
 # default linker flags
-LINKFLAGS=
+LINKFLAGS=-O2 -z combreloc -Bsymbolic
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec
