--- scp/source/office/dir.scp	22 May 2002 22:33:22 -0000	1.50.4.4
+++ scp/source/office/dir.scp	20 Nov 2002 17:41:42 -0000
@@ -1148,7 +1148,7 @@
 // Beginn fuer Directories, die es nur unter Linux gibt - Now for all Unix ;-)
 // ************************************************************************************
 
-#if defined(UNX)
+#if 0
 
 Directory GID_DIR_KDE2
     ParentID        = PREDEFINED_HOMEDIR;
RCS file: /cvs/installation/scp/source/ookde/makefile.mk,v
retrieving revision 1.1.2.2
diff -u -r1.1.2.2 makefile.mk
--- scp/source/ookde/makefile.mk	6 Apr 2002 01:01:32 -0000	1.1.2.2
+++ scp/source/ookde/makefile.mk	20 Nov 2002 17:41:50 -0000
@@ -81,12 +81,5 @@
         module_kde.lng            \
         unxbasic_kde.lng
 
-.IF "$(GUI)"=="UNX"
-PARFILES = \
-        module_kde.par            \
-        files_kde.par             \
-        unxbasic_kde.par
-.ENDIF
-
 # --- File ---------------------------------------------------------
 .INCLUDE :  target.mk
RCS file: /cvs/installation/scp/source/player/player.scp,v
retrieving revision 1.38
diff -u -r1.38 player.scp
--- scp/source/player/player.scp	12 Nov 2001 18:47:25 -0000	1.38
+++ scp/source/player/player.scp	20 Nov 2002 17:41:51 -0000
@@ -62,7 +62,7 @@
 // directories
 ////////////////////////
 
-#ifdef LINUX
+#if 0
 
 Directory GID_DIR_KDE2
     ParentID        = PREDEFINED_HOMEDIR;
@@ -86,7 +86,7 @@
 
 #endif
 
-#ifdef UNX
+#if 0
 
 Directory GID_DIR_GNOME_PLAYER
     ParentID        = PREDEFINED_HOMEDIR;
@@ -695,13 +695,15 @@
         Name = "player_pngs.zip";
 End
 
+#if 0
 File GID_FILE_EXTRA_GNOMEPLAYER
         Dir             = GID_DIR_HOME_GNOME_APPS_STAR;
         SUBST_ARCHIVE_TXT_FILE_BODY;
         Name = "gnomeappplayer.zip";
 End
+#endif
 
-#ifdef LINUX
+#if 0
 
 File GID_FILE_EXTRA_KDEAPPPLAYER
         Dir             = GID_DIR_KDE2_SHARE_APPLNK_STAR;
Index: scp/source/oocde/module_cde.scp
===================================================================
RCS file: /cvs/installation/scp/source/oocde/module_cde.scp,v
retrieving revision 1.1.2.2
diff -u -r1.1.2.2 module_cde.scp
--- scp/source/oocde/module_cde.scp	28 May 2002 14:37:15 -0000	1.1.2.2
+++ scp/source/oocde/module_cde.scp	20 Nov 2002 17:51:04 -0000
@@ -73,7 +73,7 @@
     Styles          = (CREATE);
 End
 
-
+#if 0
  Module GID_MODULE_OPTIONAL_GNOME
         MOD_NAME_DESC ( MODULE_OPTIONAL_GNOME );
         ParentID = GID_MODULE_OPTIONAL;
@@ -95,6 +95,7 @@
         Dirs = (GID_DIR_SHARE, GID_DIR_SHARE_ICONS, GID_DIR_SHARE_GNOME, GID_DIR_HOME_GNOME,
                 GID_DIR_HOME_GNOME_APPS, GID_DIR_HOME_GNOME_APPS_STAR);
  End
+#endif
 
 #endif
 
Index: scp/util/makefile.mk
===================================================================
RCS file: /cvs/installation/scp/util/makefile.mk,v
retrieving revision 1.17.10.1
diff -u -r1.17.10.1 makefile.mk
--- scp/util/makefile.mk	1 Apr 2002 20:18:06 -0000	1.17.10.1
+++ scp/util/makefile.mk	25 Nov 2002 13:26:25 -0000
@@ -116,13 +116,10 @@
 .IF "$(GUI)"=="UNX"
 SCP1FILES += \
         dir_psprint.par   	\
-	module_kde.par 		\
 	module_cde.par 		\
         files_psprint.par 	\
-	files_kde.par 		\
 	files_cde.par		\
 	unxbasic.par		\
-	unxbasic_kde.par	\
 	unxbasic_cde.par
 .ELSE
 SCP1FILES += \
Index: scp/source/oocde/files_cde.scp
===================================================================
RCS file: /cvs/oo/installation/scp/source/oocde/files_cde.scp,v
retrieving revision 1.1.2.2
diff -u -p -u -r1.1.2.2 files_cde.scp
--- scp/source/oocde/files_cde.scp	29 Apr 2002 00:23:37 -0000	1.1.2.2
+++ scp/source/oocde/files_cde.scp	4 Apr 2003 16:30:43 -0000
@@ -329,14 +329,14 @@
 End
 
 // and now files for Gnome
-
+#if 0
 File GID_FILE_BIN_GNOMEINT
         BIN_FILE_BODY;
         Styles          = (PACKED);
         Dir             = GID_DIR_PROGRAM;
         Name            = "gnomeint";
 End
-
+#endif
 File GID_FILE_EXTRA_PNG_OOGROUP
         Dir             = GID_DIR_SHARE;
         ARCHIVE_TXT_FILE_BODY;
Index: scp/source/oocde/unxbasic_cde.scp
===================================================================
RCS file: /cvs/oo/installation/scp/source/oocde/unxbasic_cde.scp,v
retrieving revision 1.1.2.2
diff -u -p -u -r1.1.2.2 unxbasic_cde.scp
--- scp/source/oocde/unxbasic_cde.scp	29 Apr 2002 15:03:22 -0000	1.1.2.2
+++ scp/source/oocde/unxbasic_cde.scp	4 Apr 2003 16:37:49 -0000
@@ -651,6 +651,7 @@
                     kill DirEntry( officePath, "presentation.desktop" )
                 Endif
 
+#if 0
                 REM and now the executable gnomeint can be started with all selected modules for registering the
                 REM filetypes and extensions
                 REM Example: gnomeint StarOffice 6.0 /export/home/staroffice6.0 writer calc draw impress
@@ -690,7 +691,7 @@
                 shell(gnomeFileCall)
 
              rem    SyncStartApplication(gnomeFile, param2, param3, progpath, language_parameter)
-
+#endif
 End Sub
 
 Function existsFile(path as String, file as String) As Boolean
@@ -728,7 +729,7 @@
                 dim isCalc As Boolean
                 dim isDraw As Boolean
                 dim isImpress As Boolean
-
+#if 0
                 dim gnomeFile as String
                 dim gnomeFileCall as String
                 dim progpath as String
@@ -806,7 +807,7 @@
                 shell(gnomeFileCall)
 
              rem    SyncStartApplication(gnomeFile, param2, param3, progpath, language_parameter)
-
+#endif
 End Sub
 
 Function existsFile(path as String, file as String) As Boolean
