--- svtools/source/config/pathoptions.cxx	Thu May 30 06:46:39 2002
+++ svtools/source/config/pathoptions.cxx	Thu Mar 20 17:13:02 2003
@@ -123,6 +125,7 @@
 #define	SUBSTITUTE_INSTURL				ASCII_STR("$(insturl)")
 #define	SUBSTITUTE_PROGURL				ASCII_STR("$(progurl)")
 #define	SUBSTITUTE_USERURL				ASCII_STR("$(userurl)")
+#define SUBSTITUTE_USERHOME				ASCII_STR("$(userhome)")
 #define	SUBSTITUTE_PATH					ASCII_STR("$(path)")
 #define	SUBSTITUTE_LANG					ASCII_STR("$(lang)")
 #define	SUBSTITUTE_LANGID				ASCII_STR("$(langid)")
@@ -139,6 +142,7 @@
 #define	REPLACELENGTH_INSTURL			10
 #define	REPLACELENGTH_PROGURL			10
 #define	REPLACELENGTH_USERURL			10
+#define	REPLACELENGTH_USERHOME			11
 #define	REPLACELENGTH_PATH				7
 #define	REPLACELENGTH_LANG				7
 #define	REPLACELENGTH_LANGID			9
@@ -209,6 +213,7 @@
 	String			m_aProgPath;
 	String			m_aInstURL;
 	String			m_aUserURL;
+	String			m_aUserHome;
 	String			m_aProgURL;
 
 	LanguageType	m_eLanguageType;
@@ -377,6 +382,12 @@
 			aPath = aPath.replaceAt( nIdx, m_aUserURL.Len(), SUBSTITUTE_USERURL );
 			nIdx = aPath.indexOf( m_aUserURL );
 		}
+		nIdx = aPath.indexOf ( m_aUserHome );
+		while ( nIdx != -1 )
+		{
+			aPath = aPath.replaceAt( nIdx, m_aUserHome.Len(), SUBSTITUTE_USERHOME );
+			nIdx = aPath.indexOf( m_aUserHome );
+		}
 		nIdx = aPath.indexOf( m_aInstURL );
 		while ( nIdx != -1 )
 		{
@@ -406,6 +417,16 @@
 			}
         }
 
+        if( FileBase::getSystemPathFromFileURL( m_aUserHome, aTmp ) == FileBase::E_None )
+        {
+            nIdx = aPath.indexOf( aTmp );
+			while ( nIdx != -1 )
+			{
+                aPath = aPath.replaceAt( nIdx, aTmp.getLength(), SUBSTITUTE_USERHOME );
+                nIdx = aPath.indexOf( aTmp );
+			}
+        }
+
         if( FileBase::getSystemPathFromFileURL( m_aInstPath, aTmp ) == FileBase::E_None )
         {
             nIdx = aPath.indexOf( aTmp );
@@ -528,6 +549,14 @@
 		}
 		else
 		// -------------------------------------------------------------------------------------------------------------------
+		// $(userhome) - home directory of the user
+		if ( SUBSTITUTE_USERHOME == aSubString )
+		{
+			nReplaceLength = REPLACELENGTH_USERHOME;
+			aReplacement = m_aUserHome;
+		}
+		else
+		// -------------------------------------------------------------------------------------------------------------------
 		// $(progurl) - directory of the executable file as URL
 		if ( SUBSTITUTE_PROGURL == aSubString )
 		{
--- officecfg/data/org/openoffice/Office/Common.xcd	Thu Dec 19 04:39:49 2002
+++ officecfg/data/org/openoffice/Office/Common.xcd	Wed Mar 19 19:30:23 2003
@@ -1103,7 +1103,7 @@
 					<schema:documentation>
 						<schema:description>The path of the Work Folder can be modified according to the user's needs. The path specified here can be seen in the Open or Save dialog.</schema:description>
 					</schema:documentation>
-					<default:data>$(userurl)/work</default:data>
+					<default:data>$(userhome)/Documents</default:data>
 				</schema:value>
 			</schema:group>
 			<schema:group cfg:name="Default">
@@ -1174,7 +1174,7 @@
 					<default:data>$(userurl)/wordbook</default:data>
 				</schema:value>
 				<schema:value cfg:name="Work" cfg:type="string" cfg:writable="false">
-					<default:data>$(userurl)/work</default:data>
+					<default:data>$(userhome)/Documents</default:data>
 				</schema:value>
 			</schema:group>
 		</schema:group>
