--- vcl/unx/source/app/devaudiosound.cxx
+++ vcl/unx/source/app/devaudiosound.cxx
@@ -91,22 +91,34 @@
 BOOL								DevAudioSound::s_bPlaying = FALSE;
 DevAudioSoundList					DevAudioSound::s_aSounds;
 
+static const char* getAudioDevName()
+{
+    static const char* pDev = NULL;
+    if( ! pDev )
+    {
+        pDev = getenv( "AUDIODEV" );
+        if( ! pDev || ! *pDev )
+            pDev = "/dev/audio";
+    }
+    return pDev;
+}
+
 BOOL DevAudioSound::open()
 {
 	OGuard aGuard( s_aProtector );
 	if( s_nDevice == -1 )
 	{
-		s_nDevice = ::open( "/dev/audio", O_WRONLY | O_NDELAY );
+		s_nDevice = ::open( getAudioDevName(), O_WRONLY | O_NDELAY );
 		if( s_nDevice == -1 )
 		{
-			SalDbgAssert( "ERROR: could not open /dev/audio, errno=%d\n", errno );
+			SalDbgAssert( "ERROR: could not open %s, errno=%d\n", getAudioDevName(), errno );
 			return FALSE;
 		}
 #ifdef DEBUG
 		audio_device_t aDevice;
 		if( ! ioctl( s_nDevice, AUDIO_GETDEV, &aDevice ) )
-		  SalDbgAssert( "\nopened /dev/audio of type:\nName: %s\nVersion: %s\nConfig: %s\n\n",
-			   aDevice.name, aDevice.version, aDevice.config );
+		  SalDbgAssert( "\nopened %s of type:\nName: %s\nVersion: %s\nConfig: %s\n\n",
+				getAudioDevName(), aDevice.name, aDevice.version, aDevice.config );
 #endif
 	}
 	else
@@ -453,7 +465,7 @@
 
     if( s_nDevice == -1 )
     {
-        int nTryOpen = ::open( "/dev/audio", O_WRONLY | O_NDELAY );
+        int nTryOpen = ::open( getAudioDevName(), O_WRONLY | O_NDELAY );
         if( nTryOpen == -1 )
             return FALSE;
         ::close( nTryOpen );
