Index: connectivity/prj/build.lst
===================================================================
RCS file: /cvs/oo/dba/connectivity/prj/build.lst,v
retrieving revision 1.14
diff -u -p -u -r1.14 build.lst
--- connectivity/prj/build.lst	2001/11/07 11:00:01	1.14
+++ connectivity/prj/build.lst	2002/09/10 15:52:05
@@ -1,4 +1,4 @@
-cn	connectivity	:	comphelper moz svtools NULL
+cn	connectivity	:	comphelper svtools NULL
 cn	connectivity							usr1	-	all	cn_mkout NULL
 cn	connectivity\source\types				nmake	-	all	cn_types NULL
 cn	connectivity\source\commontools			nmake	-	all	cn_cmtools cn_types NULL
@@ -7,8 +7,6 @@ cn	connectivity\source\cpool				nmake	-	
 cn	connectivity\source\resource			nmake	-	all	cn_res NULL
 cn	connectivity\source\sdbcx				nmake	-	all	cn_sdbcx cn_types NULL
 cn	connectivity\source\drivers\ado			nmake	-	w	cn_ado cn_dbtools NULL
-cn	connectivity\source\drivers\mozab\mozillasrc		nmake	-	all	cn_mozab_mozillasrc cn_file NULL
-cn	connectivity\source\drivers\mozab	nmake	-	all	cn_mozab cn_mozab_mozillasrc cn_dbtools NULL
 cn	connectivity\source\drivers\calc		nmake	-	all	cn_calc cn_file NULL
 cn	connectivity\source\drivers\odbc		nmake	-	all	cn_odbc cn_dbtools NULL
 cn	connectivity\source\drivers\jdbc		nmake	-	all	cn_jdbc cn_dbtools NULL
diff -u -p -u -r1.314.2.6.2.2 files.scp
--- scp/source/office/files.scp
+++ scp/source/office/files.scp
@@ -473,17 +473,6 @@ File GID_FILE_LIB_DBODBCBASE
     Dir             = GID_DIR_PROGRAM;
 	Styles			= (PACKED);
 End
-File GID_FILE_LIB_MOZABDRV
-	BIN_FILE_BODY;
-	#ifdef UNX
-    Name            = "libmozabdrv2.so";
-	#else
-    Name            = "mozabdrv2.dll";
-	#endif
-    Dir             = GID_DIR_PROGRAM;
-	Styles			= (PACKED);
-End
-
 
 STD_UNO_LIB_FILE( GID_FILE_LIB_DBU, dbu )
 STD_UNO_LIB_FILE( GID_FILE_LIB_DBA, dba)
@@ -869,18 +858,6 @@ File GID_FILE_LIB_ODBC_2
         Name            = "libodbc2.so";
         #else
         Name            = "odbc2.dll";
-        #endif
-End
-
-File GID_FILE_LIB_MOZAB_2
-        BIN_FILE_BODY;
-        Styles          = (PACKED,UNO_COMPONENT);
-        RegistryID      = GID_STARREGISTRY_APPLICAT_RDB;
-        Dir             = GID_DIR_PROGRAM;
-        #ifdef UNX
-        Name            = "libmozab2.so";
-        #else
-        Name            = "mozab2.dll";
         #endif
 End
 
