--- officecfg/data/org/openoffice/Office/TypeDetection.xcd
+++ officecfg/data/org/openoffice/Office/TypeDetection.xcd
@@ -3904,7 +3904,7 @@
 					<default:data xml:lang="en-US">Microsoft Excel 97/2000/XP</default:data>
 				</default:value>
 				<default:value cfg:name="Data" cfg:type="string" cfg:writable="true">
-					<default:data>5,calc_MS_Excel_97,com.sun.star.sheet.SpreadsheetDocument,,3,,0,,</default:data>
+					<default:data>5,calc_MS_Excel_97,com.sun.star.sheet.SpreadsheetDocument,,259,,0,,</default:data>
 				</default:value>
 			</default:group>
                         <default:group cfg:name="MS Excel 97 Vorlage/Template">
@@ -4663,7 +4663,7 @@
 					<default:data xml:lang="en-US">Microsoft PowerPoint 97/2000/XP</default:data>
 				</default:value>
 				<default:value cfg:name="Data" cfg:type="string" cfg:writable="true">
-					<default:data>3,impress_MS_PowerPoint_97,com.sun.star.presentation.PresentationDocument,,67,emp,0,,</default:data>
+					<default:data>3,impress_MS_PowerPoint_97,com.sun.star.presentation.PresentationDocument,,323,emp,0,,</default:data>
 				</default:value>
 			</default:group>
                         <default:group cfg:name="MS PowerPoint 97 Vorlage">
@@ -5642,7 +5642,7 @@
 					<default:data xml:lang="en-US">Microsoft Word 97/2000/XP</default:data>
 				</default:value>
 				<default:value cfg:name="Data" cfg:type="string" cfg:writable="true">
-					<default:data>3,writer_MS_Word_97,com.sun.star.text.TextDocument,,67,CWW8,0,,</default:data>
+					<default:data>3,writer_MS_Word_97,com.sun.star.text.TextDocument,,323,CWW8,0,,</default:data>
 				</default:value>
 			</default:group>
                         <default:group cfg:name="MS Word 97 Vorlage">

--- officecfg/data/org/openoffice/Office/Linguistic.xcd
+++ officecfg/data/org/openoffice/Office/Linguistic.xcd
@@ -186,7 +186,7 @@
 					<schema:title>Automatic spellchecking</schema:title>
 					<schema:description>Check spelling while typing.</schema:description>
 				</schema:documentation>
-				<default:data>false</default:data>
+				<default:data>true</default:data>
 			</schema:value>
 			<schema:value cfg:name="IsSpellSpecial" cfg:type="boolean">
 				<!-- OldPath: MiscOptions / SpellOptions -->
