Index: vcl/source/window/winproc.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/winproc.cxx,v
retrieving revision 1.43.2.2
diff -u -p -u -r1.43.2.2 winproc.cxx
--- vcl/source/window/winproc.cxx	24 May 2002 14:10:51 -0000	1.43.2.2
+++ vcl/source/window/winproc.cxx	18 Feb 2003 15:44:32 -0000
@@ -318,7 +319,10 @@ static BOOL ImplHandleMouseFloatMode( Wi
 static void ImplHandleMouseHelpRequest( Window* pChild, const Point& rMousePos )
 {
     ImplSVData* pSVData = ImplGetSVData();
-    if ( pChild != pSVData->maHelpData.mpHelpWin )
+
+    if (!pSVData->maHelpData.mpHelpWin ||
+	!( pSVData->maHelpData.mpHelpWin->IsWindowOrChild( pChild ) ||
+	   pChild->IsWindowOrChild( pSVData->maHelpData.mpHelpWin ) ) )
     {
         USHORT nHelpMode = 0;
         if ( pSVData->maHelpData.mbQuickHelp )
