Index: desktop/source/app/app.cxx
===================================================================
RCS file: /cvs/oo/framework/desktop/source/app/app.cxx,v
retrieving revision 1.70.2.2
diff -u -p -u -r1.70.2.2 app.cxx
--- desktop/source/app/app.cxx	1 Jun 2002 05:36:20 -0000	1.70.2.2
+++ desktop/source/app/app.cxx	17 Mar 2003 15:47:41 -0000
@@ -2063,6 +2063,7 @@ void Desktop::HandleAppEvent( const Appl
     }
     else if ( rAppEvent.GetEvent() == "APPEAR" )
     {
+#if 0
         // find active task - the active task is always a visible task
         ::com::sun::star::uno::Reference< ::com::sun::star::frame::XTasksSupplier >
                 xDesktop( ::comphelper::getProcessServiceFactory()->createInstance( OUSTRING(RTL_CONSTASCII_USTRINGPARAM("com.sun.star.frame.Desktop")) ),
@@ -2082,8 +2083,11 @@ void Desktop::HandleAppEvent( const Appl
             xTop->toFront();
         }
         else
+#endif
+	  {
             // no visible task that could be activated found
             OpenDefault();
+	  }
     }
 	else if ( rAppEvent.GetEvent() == "QUICKSTART" )
 	{
--- framework/source/services/frame.cxx	Wed Dec 12 07:16:39 2001
+++ framework/source/services/frame.cxx	Mon Mar 17 19:43:38 2003
@@ -2751,7 +2751,9 @@
                 ( xComponentWindow.is() == sal_True )
             )
 		{
+#if 0
 			xComponentWindow->setFocus();
+#endif
             ::vos::OClearableGuard aSolarGuard( Application::GetSolarMutex() );
         	Window* pWindow = VCLUnoHelper::GetWindow( xComponentWindow );
         	if( pWindow != NULL )
