Index: vcl/source/window/winproc.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/winproc.cxx,v
retrieving revision 1.43.2.2
diff -u -p -u -r1.43.2.2 winproc.cxx
--- vcl/source/window/winproc.cxx	24 May 2002 14:10:51 -0000	1.43.2.2
+++ vcl/source/window/winproc.cxx	19 Feb 2003 10:23:06 -0000
@@ -399,6 +400,19 @@ long ImplHandleMouseEvent( Window* pWind
     long        nRet;
     USHORT      nClicks;
     USHORT      nOldCode = pWindow->mpFrameData->mnMouseCode;
+
+ //   fprintf (stderr, "Handle mouse event... %d,%d,%d\n", nX, nY, nSVEvent);
+
+    Size winSize = pWindow->GetOutputSizePixel();
+
+    if ( !bMouseLeave && !pWindow->IsMouseCaptured () &&
+        ( nX < 0 || nY < 0 || nX >= winSize.Width() ||
+          nY >= winSize.Height() ) )
+      {
+	ImplHandleMouseFloatMode( pWindow, aMousePos, nCode, nSVEvent, bMouseLeave );
+ //     fprintf (stderr, "Out of bounds ...");
+      return 0;
+    }
 
     // we need a mousemove event, befor we get a mousebuttondown or a
     // mousebuttonup event
Index: vcl/unx/source/window/salframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/window/salframe.cxx,v
retrieving revision 1.114.2.5
diff -u -p -u -r1.114.2.5 salframe.cxx
--- vcl/unx/source/window/salframe.cxx	27 Aug 2002 17:52:47 -0000	1.114.2.5
+++ vcl/unx/source/window/salframe.cxx	19 Feb 2003 10:23:16 -0000
@@ -2184,18 +2178,7 @@ long SalFrameData::HandleMouseEvent( XEv
 		}
 	}
 
-	if( nEvent == SALEVENT_MOUSELEAVE
-		|| ( aMouseEvt.mnX <  nWidth_  && aMouseEvt.mnX >  -1 &&
-			 aMouseEvt.mnY <  nHeight_ && aMouseEvt.mnY >  -1 )
-		|| pDisplay_->MouseCaptured( this )
-        )
-		return Call( nEvent, &aMouseEvt );
-
-#ifdef DBG_UTIL
-	fprintf( stderr, "SalFrameData::HandleMouseEvent %d size=%d*%d event=%d.%d\n",
-			 pEvent->type, nWidth_, nHeight_, aMouseEvt.mnX, aMouseEvt.mnY );
-#endif
-	return 0;
+	return Call( nEvent, &aMouseEvt );
 }
 
 //
