Index: vcl/source/control/fixed.cxx
===================================================================
RCS file: /cvs/oo/gsl/vcl/source/control/fixed.cxx,v
retrieving revision 1.4
diff -u -p -u -r1.4 fixed.cxx
--- vcl/source/control/fixed.cxx	27 Mar 2001 16:12:36 -0000	1.4
+++ vcl/source/control/fixed.cxx	29 Apr 2003 15:45:20 -0000
@@ -70,6 +70,9 @@
 #ifndef _SV_FIXED_HXX
 #include <fixed.hxx>
 #endif
+#ifndef _SV_BITMAPEX_HXX
+#include <bitmapex.hxx>
+#endif
 
 #include <rc.h>
 
@@ -730,13 +733,21 @@ void FixedBitmap::ImplDraw( OutputDevice
 	// Haben wir ueberhaupt eine Bitmap
 	if ( !(!maBitmap) )
 	{
+		BitmapEx aEx( maBitmap );
+
+		Color old_fill = pDev->GetFillColor();
+		const StyleSettings& rStyleSettings = GetSettings().GetStyleSettings();
+		pDev->SetFillColor( rStyleSettings.GetFaceColor() ); // FIXME: this sucks
+
 		if ( GetStyle() & WB_SCALE )
-			pDev->DrawBitmap( rPos, rSize, maBitmap );
+			pDev->DrawBitmapEx( rPos, rSize, aEx );
 		else
 		{
 			Point aPos = ImplCalcPos( GetStyle(), rPos, maBitmap.GetSizePixel(), rSize );
-			pDev->DrawBitmap( aPos, maBitmap );
+			pDev->DrawBitmapEx( aPos, aEx );
 		}
+
+		pDev->SetFillColor( old_fill );
 	}
 }
 
@@ -928,6 +939,11 @@ void FixedImage::ImplDraw( OutputDevice*
 	// Haben wir ueberhaupt ein Image
 	if ( !(!maImage) )
 	{
+		Color old_fill = pDev->GetFillColor();
+		const StyleSettings& rStyleSettings = GetSettings().GetStyleSettings();
+
+		pDev->SetFillColor( rStyleSettings.GetFaceColor() ); // FIXME: this sucks
+
 		if ( GetStyle() & WB_SCALE )
 			pDev->DrawImage( rPos, rSize, maImage, nStyle );
 		else
@@ -935,6 +951,8 @@ void FixedImage::ImplDraw( OutputDevice*
 			Point aPos = ImplCalcPos( GetStyle(), rPos, maImage.GetSizePixel(), rSize );
 			pDev->DrawImage( aPos, maImage, nStyle );
 		}
+
+		pDev->SetFillColor( old_fill );
 	}
 
 	mbInUserDraw = TRUE;
