Index: svx/util/makefile.mk
===================================================================
RCS file: /cvs/graphics/svx/util/makefile.mk,v
retrieving revision 1.16.2.6
diff -u -r1.16.2.6 makefile.mk
--- svx/util/makefile.mk	31 May 2002 12:57:57 -0000	1.16.2.6
+++ svx/util/makefile.mk	3 Dec 2002 16:53:51 -0000
@@ -83,6 +83,10 @@
 
 HELPIDFILES=    ..$/inc$/helpid.hrc
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=svx-gcc3.map
+.ENDIF
+
 SHL1TARGET= svx$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= svx
 SVXLOKAL=	$(LB)$/svx.lib
Index: svtools/util/makefile.mk
===================================================================
RCS file: /cvs/util/svtools/util/makefile.mk,v
retrieving revision 1.31.4.2
diff -u -r1.31.4.2 makefile.mk
--- svtools/util/makefile.mk	30 May 2002 11:46:50 -0000	1.31.4.2
+++ svtools/util/makefile.mk	3 Dec 2002 16:53:51 -0000
@@ -209,6 +209,11 @@
 SHL2VERSIONMAP=libsvl641ss.so.mapfile
 .ENDIF
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=svt-gcc3.map
+SHL2VERSIONMAP=svl-gcc3.map
+.ENDIF
+
 SHL1LIBS= \
 		$(SLB)$/svt.lib 	\
 		$(LB)$/svmem.lib
Index: sfx2/util/makefile.mk
===================================================================
RCS file: /cvs/framework/sfx2/util/makefile.mk,v
retrieving revision 1.21.2.3
diff -u -r1.21.2.3 makefile.mk
--- sfx2/util/makefile.mk	15 Aug 2002 15:15:49 -0000	1.21.2.3
+++ sfx2/util/makefile.mk	3 Dec 2002 16:53:51 -0000
@@ -109,6 +109,10 @@
 .ENDIF
 .ENDIF
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=$(TARGET)-$(COMNAME).map
+.ENDIF
+
 SHL1STDLIBS+=\
                 $(FWELIB) \
 		$(BASICLIB) \
Index: xmloff/util/makefile.mk
===================================================================
RCS file: /cvs/xml/xmloff/util/makefile.mk,v
retrieving revision 1.14.4.1
diff -u -r1.14.4.1 makefile.mk
--- xmloff/util/makefile.mk	15 Aug 2002 14:44:26 -0000	1.14.4.1
+++ xmloff/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -120,6 +120,10 @@
 SHL1DEF=    $(MISC)$/$(SHL1TARGET).def
 SHL1LIBS=   $(LIB1TARGET)
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=$(TARGET)-$(COMNAME).map
+.ENDIF
+
 # --- Static-Lib ---------------------------------------------------------
 
 .IF "$(SVXLIGHT)" != "" 
Index: toolkit/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/toolkit/util/makefile.mk,v
retrieving revision 1.8.4.1
diff -u -r1.8.4.1 makefile.mk
--- toolkit/util/makefile.mk	15 Aug 2002 13:57:51 -0000	1.8.4.1
+++ toolkit/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -105,6 +105,10 @@
 
 DEF1EXPORTFILE=	tk.dxp 
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=$(TARGET)-$(COMNAME).map
+.ENDIF
+
 .IF "$(OS)"=="MACOSX" 
 
 # [ed] 6/16/02 Add in X libraries if we're building X
Index: vcl/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.33.6.2
diff -u -r1.33.6.2 makefile.mk
--- vcl/util/makefile.mk	27 Aug 2002 17:58:57 -0000	1.33.6.2
+++ vcl/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -220,6 +220,9 @@
 .IF "$(OS)$(CPU)"=="SOLARISS"
 SHL1VERSIONMAP=libvcl641ss.so.mapfile
 .ENDIF
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=$(TARGET)-$(COMNAME).map
+.ENDIF
 .IF "$(remote)" != ""
 SHL1STDLIBS+=   $(UNOLIB)
 .ENDIF
Index: framework/util/makefile.mk
===================================================================
RCS file: /cvs/framework/framework/util/makefile.mk,v
retrieving revision 1.59.2.1
diff -u -r1.59.2.1 makefile.mk
--- framework/util/makefile.mk	19 Aug 2002 16:19:01 -0000	1.59.2.1
+++ framework/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -219,6 +219,10 @@
 
 SHL4TARGET=		fwk$(UPD)$(DLLPOSTFIX)
 
+.IF "$(COMNAME)"=="gcc3"
+SHL4VERSIONMAP=fwk-$(COMNAME).map
+.ENDIF
+
 SHL4IMPLIB=		ifwk
 
 SHL4OBJS=       $(SLO)$/argumentanalyzer.obj		\
Index: basic/util/makefile.mk
===================================================================
RCS file: /cvs/script/basic/util/makefile.mk,v
retrieving revision 1.9.6.1
diff -u -r1.9.6.1 makefile.mk
--- basic/util/makefile.mk	18 Feb 2002 17:45:55 -0000	1.9.6.1
+++ basic/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -106,6 +106,10 @@
 SHL1STDLIBS+=$(SJLIB)
 .ENDIF
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=$(TARGET)-$(COMNAME).map
+.ENDIF
+
 .IF "$(GUI)"=="WNT"
 #SHL1STDLIBS+=$(LIBCIMT)
 .ENDIF
Index: so3/util/makefile.mk
===================================================================
RCS file: /cvs/oi/so3/util/makefile.mk,v
retrieving revision 1.6.2.2
diff -u -r1.6.2.2 makefile.mk
--- so3/util/makefile.mk	30 May 2002 14:01:39 -0000	1.6.2.2
+++ so3/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -119,6 +119,10 @@
 		$(COMPHELPERLIB) \
 		$(UNOTOOLSLIB)
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=so-$(COMNAME).map
+.ENDIF
+
 .IF "$(OS)$(CPU)"=="SOLARISS"
 SHL1VERSIONMAP=libso641ss.so.mapfile
 .ENDIF
Index: tools/util/makefile.mk
===================================================================
RCS file: /cvs/util/tools/util/makefile.mk,v
retrieving revision 1.7.8.1
diff -u -r1.7.8.1 makefile.mk
--- tools/util/makefile.mk	30 May 2002 13:56:14 -0000	1.7.8.1
+++ tools/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -197,6 +197,9 @@
 .IF "$(OS)$(CPU)$(PRODUCT)"=="SOLARISSfull"
 SHL1VERSIONMAP=libtl641ss.so.mapfile
 .ENDIF
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=tl-$(COMNAME).map
+.ENDIF
 
 .IF "$(GUI)"=="WNT"
 SHL1STDLIBS+=   shell32.lib     \
Index: goodies/util/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/util/makefile.mk,v
retrieving revision 1.3.2.2
diff -u -r1.3.2.2 makefile.mk
--- goodies/util/makefile.mk	30 May 2002 14:28:39 -0000	1.3.2.2
+++ goodies/util/makefile.mk	3 Dec 2002 16:53:52 -0000
@@ -85,6 +85,10 @@
 SHL1TARGET= go$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= igo
 
+.IF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=$(TARGET)-$(COMNAME).map
+.ENDIF
+
 .IF "$(OS)$(CPU)"=="SOLARISS"
 SHL1VERSIONMAP=libgo641ss.so.mapfile
 .ENDIF
