--- officecfg/data/org/openoffice/Office/Common.xcd	Thu Dec 19 04:39:49 2002
+++ officecfg/data/org/openoffice/Office/Common.xcd	Mon Mar 17 21:02:38 2003
@@ -4017,12 +4017,55 @@
 				<default:data xsi:null="true"/>
 			</schema:value>
 			<schema:value cfg:name="Program" cfg:type="string">
-				<default:data/>
+				<default:data>evolution-1.4</default:data>
 			</schema:value>
 			<schema:value cfg:name="CommandProfile" cfg:type="string">
-				<default:data>Netscape 6.x</default:data>
+				<default:data>Evolution 1.4</default:data>
 			</schema:value>
 			<schema:set cfg:name="Profiles" cfg:element-type="MailCommandLineProfile">
+				<default:group cfg:name="Evolution 1.4">
+	  				<default:group cfg:name="FormatStrings">
+						<default:value cfg:name="base" cfg:type="string">
+							<default:data>mailto:?%s</default:data>
+						</default:value>
+						<default:value cfg:name="from" cfg:type="string">
+							<!-- FIXME? -->
+							<default:data/>
+						</default:value>
+						<default:value cfg:name="to" cfg:type="string">
+							<default:data>to=%s</default:data>
+						</default:value>
+						<default:value cfg:name="cc" cfg:type="string">
+							<default:data>cc=%s</default:data>
+						</default:value>
+						<default:value cfg:name="bcc" cfg:type="string">
+							<default:data>bcc=%s</default:data>
+						</default:value>
+						<default:value cfg:name="subject" cfg:type="string">
+							<default:data>subject=%s</default:data>
+						</default:value>
+						<default:value cfg:name="attachment" cfg:type="string">
+							<default:data>attach=%s</default:data>
+						</default:value>
+						<default:value cfg:name="body" cfg:type="string">
+							<default:data>body=%s</default:data>
+						</default:value>
+					</default:group>
+					<default:group cfg:name="EnumDelimiters">
+						<default:value cfg:name="base" cfg:type="string">
+							<default:data>?</default:data>
+						</default:value>
+						<default:value cfg:name="cc" cfg:type="string">
+							<default:data>,</default:data>
+						</default:value>
+						<default:value cfg:name="bcc" cfg:type="string">
+							<default:data>,</default:data>
+						</default:value>
+						<default:value cfg:name="attachment" cfg:type="string">
+							<default:data>,</default:data>
+						</default:value>
+					</default:group>
+	  			</default:group>
 				<default:group cfg:name="Netscape 6.x">
 					<default:group cfg:name="FormatStrings">
 						<default:value cfg:name="base" cfg:type="string">
