Index: tools/source/memmgr/memmgr.cxx
===================================================================
RCS file: /cvs/oo/util/tools/source/memmgr/memmgr.cxx,v
retrieving revision 1.6.2.1
diff -u -p -u -r1.6.2.1 memmgr.cxx
--- tools/source/memmgr/memmgr.cxx	28 Mar 2003 12:45:33 -0000	1.6.2.1
+++ tools/source/memmgr/memmgr.cxx	28 Apr 2003 11:00:52 -0000
@@ -96,6 +96,9 @@
 // =======================================================================
 
 _NEW_HANDLER SetNewHandler (_NEW_HANDLER pfnHandler)
+#if ! (__GNUC__ == 3 && __GNUC_MINOR__ == 0)
+	throw()
+#endif
 {
 #if defined(_MSC_VER) && (_MSC_VER <= 1200)
 	return (_NEW_HANDLER)_set_new_handler ((_PNH)pfnHandler);
