Index: source/misc/productregistration.cxx
===================================================================
RCS file: /cvs/oo/util/svtools/source/misc/productregistration.cxx,v
retrieving revision 1.1.2.2
diff -u -r1.1.2.2 productregistration.cxx
--- svtools/source/misc/productregistration.cxx	2002/05/30 11:46:45	1.1.2.2
+++ svtools/source/misc/productregistration.cxx	2002/10/14 10:08:46
@@ -263,9 +263,9 @@
 				}
 				else
 				{
+#ifdef DO_PAINFUL_REGISTRATION
 					// if we're here, the dialog should be executed during this session
 					OSL_ENSURE( RegOptions::dpThisSession == ePermission, "OProductRegistration::execute: invalid permissions!" );
-
 					// execute it
 					RegistrationDialog aDialog( Application::GetDefDialogParent() );
 					aDialog.Execute();
@@ -292,6 +292,7 @@
 						default:
 							OSL_ENSURE( sal_False, "OProductRegistration::execute: invalid response from the dialog!" );
 					}
+#endif
 				}
 			}
 
