--- sw/source/ui/inc/swmn.hrc	Sat Jun  1 00:19:32 2002
+++ sw/source/ui/inc/swmn.hrc	Sun Jan  5 17:23:17 2003
@@ -3404,14 +3404,14 @@
 			{
 				ITEM_FILE_MAIL_SENDDOC
 				Text = "Dokument als ~E-Mail..." ;
-				Text [ english ] = "Document as ~eMail..." ;
+				Text [ english ] = "Send Document as ~eMail..." ;
 				Text [ portuguese ] = "Documento como e-~mail..." ;
-				Text [ english_us ] = "Document As ~E-mail..." ;
+				Text [ english_us ] = "Send Document As ~E-mail..." ;
 				Text [ portuguese_brazilian ] = "Dokument als ~eMail..." ;
 				Text [ swedish ] = "Dokument som ~e-post..." ;
 				Text [ danish ] = "Dokument som ~e-mail..." ;
 				Text [ italian ] = "Documento come ~eMail..." ;
-				Text [ spanish ] = "~Documento como e-mail..." ;
+				Text [ spanish ] = "Enviar ~Documento como e-mail..." ;
 				Text [ french ] = "En~voyer par e-mail..." ;
 				Text [ dutch ] = "Document als ~e-mail..." ;
 				Text[ chinese_simplified ] = "æ–‡æ¡£å½“ä½œç”µå­é‚®ä»¶(~E)...";
@@ -3428,7 +3428,7 @@
 				Text[ language_user1 ] = " ";
 				Text[ catalan ] = "Document As ~E-mail...";
 				Text[ finnish ] = "Asiakirja ~sähköpostina";
-				Text[ extern ] = "Document As ~E-mail...";
+				Text[ extern ] = "Send Document As ~E-mail...";
 			};
 #endif
 #ifndef MN_SW_WEB
@@ -3436,274 +3436,226 @@
 			{
 				ITEM_FILE_VERSIONDIALOG
 			};
-			MenuItem MN_SUB_SENDTO
+			MenuItem
 			{
-				Text = "S~enden" ;
-				Text [ english ] = "S~end to" ;
-				SubMenu = Menu
-				{
-					ItemList =
-					{
-						MenuItem
-						{
-							ITEM_FILE_MAIL_SENDDOC
-							Text = "Dokument als ~E-Mail..." ;
-							Text [ english ] = "Document as ~eMail..." ;
-							Text [ english_us ] = "Document As ~E-mail..." ;
-							Text [ portuguese_brazilian ] = "Dokument als ~eMail..." ;
-							Text [ swedish ] = "Dokument som ~e-post..." ;
-							Text [ danish ] = "Dokument som ~e-mail..." ;
-							Text [ italian ] = "Documento come ~eMail..." ;
-							Text [ spanish ] = "~Documento como e-mail..." ;
-							Text [ french ] = "En~voyer par e-mail..." ;
-							Text [ dutch ] = "Document als ~e-mail..." ;
-							Text [ portuguese ] = "Documento como e-~mail..." ;
-							Text[ chinese_simplified ] = "æ–‡æ¡£å½“ä½œç”µå­é‚®ä»¶(~E)...";
-							Text[ russian ] = "Äîêóìåíò êàê ~ıë. ïèñüìî...";
-							Text[ polish ] = "Dokument jako e-mail...";
-							Text[ japanese ] = "~E-mail ã¨ã—ã¦ã®ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆ...";
-							Text[ chinese_traditional ] = "æ–‡ä»¶ç•¶ä½œé›»å­éƒµä»¶(~E)...";
-							Text[ arabic ] = "...ÇáãÓÊäÏ ßÈÑíÏ ÅáßÊÑæäí";
-							Text[ dutch ] = "Document als ~e-mail...";
-							Text[ chinese_simplified ] = "æ–‡æ¡£å½“ä½œç”µå­é‚®ä»¶(~E)...";
-							Text[ greek ] = "¸ããñáöï ùò ~E-mail...";
-							Text[ korean ] = "ë¬¸ì„œë¥¼ ì „ìë©”ì¼ë¡œ (~E)...";
-							Text[ turkish ] = "E-posta olarak...";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "Document As ~E-mail...";
-							Text[ finnish ] = "Asiakirja ~sähköpostina";
-							Text[ extern ] = "Document As ~E-mail...";
-						};
-						SEPARATOR ;
-						MenuItem
-						{
-							Identifier = FN_NEW_GLOBAL_DOC ;
-							HelpID = FN_NEW_GLOBAL_DOC ;
-							Text = "Global~dokument erzeugen" ;
-							Text [ english ] = "Create global ~document" ;
-							Text [ english_us ] = "Create Master ~Document" ;
-							Text [ portuguese_brazilian ] = "Global~dokument erzeugen" ;
-							Text [ swedish ] = "Skapa samlings~dokument" ;
-							Text [ danish ] = "Opret hoved~dokument" ;
-							Text [ italian ] = "~Crea documento master" ;
-							Text [ spanish ] = "Crear documento ~maestro" ;
-							Text [ french ] = "Créer un ~document maître" ;
-							Text [ dutch ] = "Master-~document maken" ;
-							Text [ portuguese ] = "Criar ~documento-mestre" ;
-							Text[ chinese_simplified ] = "ç”Ÿæˆä¸»æ§æ–‡æ¡£(~D)";
-							Text[ russian ] = "Ñîçäàòü ãëàâíûé äîêóìåíò";
-							Text[ polish ] = "Utwórz dokument g³ówny";
-							Text[ japanese ] = "ãƒã‚¹ã‚¿ãƒ¼ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆã®ä½œæˆ(~D)";
-							Text[ chinese_traditional ] = "è£½ä½œä¸»æ§æ–‡ä»¶(~D)";
-							Text[ arabic ] = "ÅäÔÇÁ ãÓÊäÏ ÔÇãá";
-							Text[ dutch ] = "Master-~document maken";
-							Text[ chinese_simplified ] = "ç”Ÿæˆä¸»æ§æ–‡æ¡£(~D)";
-							Text[ greek ] = "~Äçìéïõñãßá êõñßïõ åããñÜöïõ";
-							Text[ korean ] = "ë§ˆìŠ¤í„° ë¬¸ì„œ ì‘ì„±(~D)";
-							Text[ turkish ] = "Ana belge oluştur";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "Create Master ~Document";
-							Text[ finnish ] = "Luo pää~asiakirja";
-							Text[ extern ] = "Create Master ~Document";
-						};
-						MenuItem
-						{
-							Identifier = FN_NEW_HTML_DOC ;
-							HelpID = FN_NEW_HTML_DOC ;
-							Text = "~HTML-Dokument erzeugen" ;
-							Text [ english ] = "Create ~HTML document" ;
-							Text [ english_us ] = "Create ~HTML Document" ;
-							Text [ portuguese_brazilian ] = "~HTML Dokument erzeugen" ;
-							Text [ swedish ] = "Skapa ~HTML-dokument" ;
-							Text [ danish ] = "Opret ~HTML-dokument" ;
-							Text [ italian ] = "Crea documento ~HTML" ;
-							Text [ spanish ] = "~Crear documento HTML" ;
-							Text [ french ] = "Créer un document ~HTML" ;
-							Text [ dutch ] = "~HTML document maken" ;
-							Text [ portuguese ] = "Criar documento ~HTML" ;
-							Text[ chinese_simplified ] = "ç”Ÿæˆ ~HTML æ–‡æ¡£";
-							Text[ russian ] = "Ñîçäàòü äîêóìåíò HTML";
-							Text[ polish ] = "Utwórz dokument HTML";
-							Text[ japanese ] = "~HTML å½¢å¼ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆã®ä½œæˆ";
-							Text[ chinese_traditional ] = "è£½ä½œ ~HTML æ–‡ä»¶";
-							Text[ arabic ] = "HTML ÅäÔÇÁ ãÓÊäÏ";
-							Text[ dutch ] = "~HTML document maken";
-							Text[ chinese_simplified ] = "ç”Ÿæˆ ~HTML æ–‡æ¡£";
-							Text[ greek ] = "Äçìéïõñãßá åããñÜöïõ ~HTML";
-							Text[ korean ] = "~HTML ë¬¸ì„œ ì‘ì„±";
-							Text[ turkish ] = "HTML belgesi oluştur";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "Create ~HTML Document";
-							Text[ finnish ] = "Luo ~HTML-asiakirja";
-							Text[ extern ] = "Create ~HTML Document";
-						};
-						SEPARATOR ;
-						MenuItem
-						{
-							Identifier = FN_OUTLINE_TO_IMPRESS ;
-							HelpID = FN_OUTLINE_TO_IMPRESS ;
-							/* ### ACHTUNG: Neuer Text in Resource? ~Gliederung an Präsentation : ~Gliederung an Prõsentation */
-							Text = "~Gliederung an Präsentation" ;
-							Text [ english ] = "Outline to Presentation" ;
-							Text [ english_us ] = "Outline to ~Presentation" ;
-							Text [ portuguese_brazilian ] = "Gliederung an Pr?sentation" ;
-							Text [ swedish ] = "D~isposition till presentation" ;
-							Text [ danish ] = "~Disposition til præsentation" ;
-							Text [ italian ] = "~Struttura in presentazione" ;
-							Text [ spanish ] = "~Esquema en presentación" ;
-							Text [ french ] = "P~lan vers présentation" ;
-							Text [ dutch ] = "Overzicht naar ~presentatie" ;
-							Text [ portuguese ] = "Esquema em ~apresentação" ;
-							Text[ chinese_simplified ] = "å¤§çº²é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~P)";
-							Text[ russian ] = "Ñòğóêòóğó â ïğåçåíòàöèş";
-							Text[ polish ] = "Konspekt do prezentacji";
-							Text[ japanese ] = "ãƒ—ãƒ¬ã‚¼ãƒ³ãƒ†ãƒ¼ã‚·ãƒ§ãƒ³ã®ã‚¢ã‚¦ãƒˆãƒ©ã‚¤ãƒ³(~P)";
-							Text[ chinese_traditional ] = "å¤§ç¶±é€è‡³ç°¡å ±(~P)";
-							Text[ arabic ] = "ãÎØØ ÊİÕíáí ááÚÑÖ ÇáÊŞÏíãí";
-							Text[ dutch ] = "Overzicht naar ~presentatie";
-							Text[ chinese_simplified ] = "å¤§çº²é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~P)";
-							Text[ greek ] = "ÄéÜñèñùóç ğñïò ğáñïõóßáóç";
-							Text[ korean ] = "í”„ë ˆì  í…Œì´ì…˜ì—ì„œì˜ ì•„ì›ƒíŠ¸ë¼ì¸(~P)";
-							Text[ turkish ] = "~Anahatları sunuya aktar";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "Outline to ~Presentation";
-							Text[ finnish ] = "Jäsennys -> ~esitys";
-							Text[ extern ] = "Outline to ~Presentation";
-						};
-						MenuItem
-						{
-							Identifier = FN_OUTLINE_TO_CLIPBOARD ;
-							HelpID = FN_OUTLINE_TO_CLIPBOARD ;
-							Text = "Gliederung zum ~Clipboard" ;
-							Text [ English ] = "Outline to Clipboard" ;
-							Text [ portuguese ] = "Esquema em área de ~transferência" ;
-							Text [ english_us ] = "Outline to ~Clipboard" ;
-							Text [ portuguese_brazilian ] = "Gliederung zum Clipboard" ;
-							Text [ swedish ] = "Disposition till ur~klippet" ;
-							Text [ danish ] = "Disposition til ~udklipsholder" ;
-							Text [ italian ] = "Struttura in clip~board" ;
-							Text [ spanish ] = "Esquema para ~portapapeles" ;
-							Text [ french ] = "Pla~n vers le presse-papiers" ;
-							Text [ dutch ] = "Overzicht naar ~Klembord" ;
-							Text[ chinese_simplified ] = "å¤§çº²é€è‡³å‰ªè´´æ¿(~C)";
-							Text[ russian ] = "Ñòğóêòóğó â áóôåğ îáìåíà";
-							Text[ polish ] = "Konspekt do schowka";
-							Text[ japanese ] = "ã‚¯ãƒªãƒƒãƒ—ãƒœãƒ¼ãƒ‰ã®ã‚¢ã‚¦ãƒˆãƒ©ã‚¤ãƒ³(~C)";
-							Text[ chinese_traditional ] = "å¤§ç¶±é€è‡³å‰ªè²¼ç°¿(~C)";
-							Text[ arabic ] = "ãÎØØ ÊİÕíáí ááÍÇİÙÉ";
-							Text[ dutch ] = "Overzicht naar ~Klembord";
-							Text[ chinese_simplified ] = "å¤§çº²é€è‡³å‰ªè´´æ¿(~C)";
-							Text[ greek ] = "ÄéÜñèñùóç ğñïò ğñü~÷åéñï";
-							Text[ korean ] = "í´ë¦½ë³´ë“œë¥¼ ìœ„í•œ ì•„ì›ƒíŠ¸ë¼ì¸(~C)";
-							Text[ turkish ] = "Anahatları ~panoya aktar";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "Outline to ~Clipboard";
-							Text[ finnish ] = "Jäsennys -> ~leikepöytä";
-							Text[ extern ] = "Outline to ~Clipboard";
-						};
-						SEPARATOR ;
-						MenuItem
-						{
-							Identifier = FN_ABSTRACT_NEWDOC ;
-							HelpID = FN_ABSTRACT_NEWDOC ;
-							Text = "A~utoAbstract erzeugen..." ;
-							Text [ English ] = "Create summary..." ;
-							Text [ portuguese ] = "Criar Auto~Abstracto..." ;
-							Text [ english_us ] = "Create A~utoAbstract..." ;
-							Text [ portuguese_brazilian ] = "A~utoAbstract erzeugen..." ;
-							Text [ swedish ] = "Skapa A~utoUtdrag..." ;
-							Text [ danish ] = "Opret A~utoAbstract..." ;
-							Text [ italian ] = "Crea a~bstract automatico..." ;
-							Text [ spanish ] = "Crear ~AutoAbstracto..." ;
-							Text [ french ] = "Créer une A~utoSynthèse..." ;
-							Text [ dutch ] = "~AutoExtract maken..." ;
-							Text[ chinese_simplified ] = "ç”Ÿæˆè‡ªåŠ¨æ‘˜è¦(~U)...";
-							Text[ russian ] = "Ñîçäàòü àâòîğåôåğàò...";
-							Text[ polish ] = "Utwórz Autokonspekt...";
-							Text[ japanese ] = "è‡ªå‹•æŠ½å‡ºã®ä½œæˆ(~U)...";
-							Text[ chinese_traditional ] = "è£½ä½œè‡ªå‹•æ‘˜è¦(~U)...";
-							Text[ arabic ] = "...AutoAbstract ÅäÔÇÁ";
-							Text[ dutch ] = "~AutoExtract maken...";
-							Text[ chinese_simplified ] = "ç”Ÿæˆè‡ªåŠ¨æ‘˜è¦(~U)...";
-							Text[ greek ] = "Äçìéïõñãßá A~utoAbstract...";
-							Text[ korean ] = "ìë™ì¶”ìƒ ì‘ì„±(~U)...";
-							Text[ turkish ] = "Otomatik özet oluştur...";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "Create A~utoAbstract...";
-							Text[ finnish ] = "Luo automaattinen ~tiivistelmä";
-							Text[ extern ] = "Create A~utoAbstract...";
-						};
-						MenuItem
-						{
-							Identifier = FN_ABSTRACT_STARIMPRESS ;
-							HelpID = FN_ABSTRACT_STARIMPRESS ;
-							/* ### ACHTUNG: Neuer Text in Resource? AutoAbstract an ~Präsentation... : AutoAbstract an ~Prõsentation... */
-							Text = "AutoAbstract an ~Präsentation..." ;
-							Text [ English ] = "Summary to presentation..." ;
-							Text [ english_us ] = "AutoAbst~ract to Presentation..." ;
-							Text [ portuguese_brazilian ] = "AutoAbstract an ~Pr?sentation..." ;
-							Text [ swedish ] = "AutoUtdrag till ~presentation..." ;
-							Text [ danish ] = "AutoAbstract til ~præsentation..." ;
-							Text [ italian ] = "Abstract automatico alla ~presentazione..." ;
-							Text [ spanish ] = "AutoAbstracto para ~presentación..." ;
-							Text [ french ] = "AutoSynthèse vers ~présentation..." ;
-							Text [ dutch ] = "A~utoExtract aan presentatie..." ;
-							Text [ portuguese ] = "AutoAbstracto em ~apresentação..." ;
-							Text[ chinese_simplified ] = "è‡ªåŠ¨æ‘˜è¦é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~R)...";
-							Text[ russian ] = "Àâòîğåôåğàò â ïğåçåíòàöèş...";
-							Text[ polish ] = "Autokonspekt do prezentacji...";
-							Text[ japanese ] = "ãƒ—ãƒ¬ã‚¼ãƒ³ãƒ†ãƒ¼ã‚·ãƒ§ãƒ³ã®è‡ªå‹•æŠ½å‡º(~R)...";
-							Text[ chinese_traditional ] = "è‡ªå‹•æ‘˜è¦é€è‡³ç°¡å ±(~R)...";
-							Text[ arabic ] = "...Åáì ÚÑÖ ÊŞÏíãí AutoAbstract";
-							Text[ dutch ] = "A~utoExtract aan presentatie...";
-							Text[ chinese_simplified ] = "è‡ªåŠ¨æ‘˜è¦é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~R)...";
-							Text[ greek ] = "AutoAbstract ùò ğñïò ~ğáñïõóßáóç...";
-							Text[ korean ] = "í”„ë ˆì  í…Œì´ì…˜ì—ì„œì˜ ìë™ì¶”ìƒ(~R)...";
-							Text[ turkish ] = "Otomatik ~özeti sunuya aktar...";
-							Text[ language_user1 ] = " ";
-							Text[ catalan ] = "AutoAbst~ract to Presentation...";
-							Text[ finnish ] = "Automaattinen ~tiivistelmä esitykseen";
-							Text[ extern ] = "AutoAbst~ract to Presentation...";
-						};
-					};
-					Text[ english_us ] = "Send";
-				Text[ portuguese ] = "~Enviar";
-				Text[ russian ] = "Îòïğàâèòü";
-				Text[ dutch ] = "~Zenden";
-				Text[ french ] = "Envo~yer";
-				Text[ spanish ] = "En~viar";
-				Text[ italian ] = "~Invia";
-				Text[ danish ] = "Senden";
-				Text[ swedish ] = "Ski~cka";
-				Text[ polish ] = "Wyœlij";
-				Text[ portuguese_brazilian ] = "Senden";
-				Text[ japanese ] = "‘—M";
-				Text[ chinese_simplified ] = "¼Ä·¢";
+				ITEM_FILE_MAIL_SENDDOC
+				Text = "Dokument als ~E-Mail..." ;
+				Text [ english ] = "Send Document as ~eMail..." ;
+				Text [ english_us ] = "Send Document As ~E-mail..." ;
+				Text [ portuguese_brazilian ] = "Dokument als ~eMail..." ;
+				Text [ swedish ] = "Dokument som ~e-post..." ;
+				Text [ danish ] = "Dokument som ~e-mail..." ;
+				Text [ italian ] = "Documento come ~eMail..." ;
+				Text [ spanish ] = "Enviar ~Documento como e-mail..." ;
+				Text [ french ] = "En~voyer par e-mail..." ;
+				Text [ dutch ] = "Document als ~e-mail..." ;
+				Text [ portuguese ] = "Documento como e-~mail..." ;
+				Text[ chinese_simplified ] = "æ–‡æ¡£å½“ä½œç”µå­é‚®ä»¶(~E)...";
+				Text[ russian ] = "Äîêóìåíò êàê ~ıë. ïèñüìî...";
+				Text[ polish ] = "Dokument jako e-mail...";
+				Text[ japanese ] = "~E-mail ã¨ã—ã¦ã®ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆ...";
+				Text[ chinese_traditional ] = "æ–‡ä»¶ç•¶ä½œé›»å­éƒµä»¶(~E)...";
+				Text[ arabic ] = "...ÇáãÓÊäÏ ßÈÑíÏ ÅáßÊÑæäí";
+				Text[ dutch ] = "Document als ~e-mail...";
+				Text[ chinese_simplified ] = "æ–‡æ¡£å½“ä½œç”µå­é‚®ä»¶(~E)...";
+				Text[ greek ] = "¸ããñáöï ùò ~E-mail...";
+				Text[ korean ] = "ë¬¸ì„œë¥¼ ì „ìë©”ì¼ë¡œ (~E)...";
+				Text[ turkish ] = "E-posta olarak...";
+				Text[ language_user1 ] = " ";
+				Text[ catalan ] = "Document As ~E-mail...";
+				Text[ finnish ] = "Asiakirja ~sähköpostina";
+				Text[ extern ] = "Send Document As ~E-mail...";
 			};
-				Text [ portuguese ] = "~Enviar" ;
-				Text [ english_us ] = "Sen~d" ;
-				Text [ portuguese_brazilian ] = "Senden" ;
-				Text [ swedish ] = "Ski~cka" ;
-				Text [ danish ] = "S~end" ;
-				Text [ italian ] = "~Invia" ;
-				Text [ spanish ] = "En~viar" ;
-				Text [ french ] = "Envo~yer" ;
-				Text [ dutch ] = "~Zenden" ;
-				Text[ chinese_simplified ] = "è¾“å‡º(~D)";
-				Text[ russian ] = "Îòïğàâèò~ü";
-				Text[ polish ] = "Wyœlij";
-				Text[ japanese ] = "é€ã‚‹(~D)";
-				Text[ chinese_traditional ] = "å‚³é€(~D)";
-				Text[ arabic ] = "ÅÑÓÇá";
-				Text[ dutch ] = "~Zenden";
-				Text[ chinese_simplified ] = "è¾“å‡º(~D)";
-				Text[ greek ] = "Á~ğïóôïëŞ";
-				Text[ korean ] = "ë³´ë‚´ê¸°(~D)";
-				Text[ turkish ] = "Gönder";
+			SEPARATOR ;
+			MenuItem
+			{
+				Identifier = FN_NEW_GLOBAL_DOC ;
+				HelpID = FN_NEW_GLOBAL_DOC ;
+				Text = "Global~dokument erzeugen" ;
+				Text [ english ] = "Create global ~document" ;
+				Text [ english_us ] = "Create Master ~Document" ;
+				Text [ portuguese_brazilian ] = "Global~dokument erzeugen" ;
+				Text [ swedish ] = "Skapa samlings~dokument" ;
+				Text [ danish ] = "Opret hoved~dokument" ;
+				Text [ italian ] = "~Crea documento master" ;
+				Text [ spanish ] = "Crear documento ~maestro" ;
+				Text [ french ] = "Créer un ~document maître" ;
+				Text [ dutch ] = "Master-~document maken" ;
+				Text [ portuguese ] = "Criar ~documento-mestre" ;
+				Text[ chinese_simplified ] = "ç”Ÿæˆä¸»æ§æ–‡æ¡£(~D)";
+				Text[ russian ] = "Ñîçäàòü ãëàâíûé äîêóìåíò";
+				Text[ polish ] = "Utwórz dokument g³ówny";
+				Text[ japanese ] = "ãƒã‚¹ã‚¿ãƒ¼ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆã®ä½œæˆ(~D)";
+				Text[ chinese_traditional ] = "è£½ä½œä¸»æ§æ–‡ä»¶(~D)";
+				Text[ arabic ] = "ÅäÔÇÁ ãÓÊäÏ ÔÇãá";
+				Text[ dutch ] = "Master-~document maken";
+				Text[ chinese_simplified ] = "ç”Ÿæˆä¸»æ§æ–‡æ¡£(~D)";
+				Text[ greek ] = "~Äçìéïõñãßá êõñßïõ åããñÜöïõ";
+				Text[ korean ] = "ë§ˆìŠ¤í„° ë¬¸ì„œ ì‘ì„±(~D)";
+				Text[ turkish ] = "Ana belge oluştur";
+				Text[ language_user1 ] = " ";
+				Text[ catalan ] = "Create Master ~Document";
+				Text[ finnish ] = "Luo pää~asiakirja";
+				Text[ extern ] = "Create Master ~Document";
+			};
+			MenuItem
+			{
+				Identifier = FN_NEW_HTML_DOC ;
+				HelpID = FN_NEW_HTML_DOC ;
+				Text = "~HTML-Dokument erzeugen" ;
+				Text [ english ] = "Create ~HTML document" ;
+				Text [ english_us ] = "Create ~HTML Document" ;
+				Text [ portuguese_brazilian ] = "~HTML Dokument erzeugen" ;
+				Text [ swedish ] = "Skapa ~HTML-dokument" ;
+				Text [ danish ] = "Opret ~HTML-dokument" ;
+				Text [ italian ] = "Crea documento ~HTML" ;
+				Text [ spanish ] = "~Crear documento HTML" ;
+				Text [ french ] = "Créer un document ~HTML" ;
+				Text [ dutch ] = "~HTML document maken" ;
+				Text [ portuguese ] = "Criar documento ~HTML" ;
+				Text[ chinese_simplified ] = "ç”Ÿæˆ ~HTML æ–‡æ¡£";
+				Text[ russian ] = "Ñîçäàòü äîêóìåíò HTML";
+				Text[ polish ] = "Utwórz dokument HTML";
+				Text[ japanese ] = "~HTML å½¢å¼ãƒ‰ã‚­ãƒ¥ãƒ¡ãƒ³ãƒˆã®ä½œæˆ";
+				Text[ chinese_traditional ] = "è£½ä½œ ~HTML æ–‡ä»¶";
+				Text[ arabic ] = "HTML ÅäÔÇÁ ãÓÊäÏ";
+				Text[ dutch ] = "~HTML document maken";
+				Text[ chinese_simplified ] = "ç”Ÿæˆ ~HTML æ–‡æ¡£";
+				Text[ greek ] = "Äçìéïõñãßá åããñÜöïõ ~HTML";
+				Text[ korean ] = "~HTML ë¬¸ì„œ ì‘ì„±";
+				Text[ turkish ] = "HTML belgesi oluştur";
+				Text[ language_user1 ] = " ";
+				Text[ catalan ] = "Create ~HTML Document";
+				Text[ finnish ] = "Luo ~HTML-asiakirja";
+				Text[ extern ] = "Create ~HTML Document";
+			};
+			SEPARATOR ;
+			MenuItem
+			{
+				Identifier = FN_OUTLINE_TO_IMPRESS ;
+				HelpID = FN_OUTLINE_TO_IMPRESS ;
+				/* ### ACHTUNG: Neuer Text in Resource? ~Gliederung an Präsentation : ~Gliederung an Prõsentation */
+				Text = "~Gliederung an Präsentation" ;
+				Text [ english ] = "Outline to Presentation" ;
+				Text [ english_us ] = "Outline to ~Presentation" ;
+				Text [ portuguese_brazilian ] = "Gliederung an Pr?sentation" ;
+				Text [ swedish ] = "D~isposition till presentation" ;
+				Text [ danish ] = "~Disposition til præsentation" ;
+				Text [ italian ] = "~Struttura in presentazione" ;
+				Text [ spanish ] = "~Esquema en presentación" ;
+				Text [ french ] = "P~lan vers présentation" ;
+				Text [ dutch ] = "Overzicht naar ~presentatie" ;
+				Text [ portuguese ] = "Esquema em ~apresentação" ;
+				Text[ chinese_simplified ] = "å¤§çº²é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~P)";
+				Text[ russian ] = "Ñòğóêòóğó â ïğåçåíòàöèş";
+				Text[ polish ] = "Konspekt do prezentacji";
+				Text[ japanese ] = "ãƒ—ãƒ¬ã‚¼ãƒ³ãƒ†ãƒ¼ã‚·ãƒ§ãƒ³ã®ã‚¢ã‚¦ãƒˆãƒ©ã‚¤ãƒ³(~P)";
+				Text[ chinese_traditional ] = "å¤§ç¶±é€è‡³ç°¡å ±(~P)";
+				Text[ arabic ] = "ãÎØØ ÊİÕíáí ááÚÑÖ ÇáÊŞÏíãí";
+				Text[ dutch ] = "Overzicht naar ~presentatie";
+				Text[ chinese_simplified ] = "å¤§çº²é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~P)";
+				Text[ greek ] = "ÄéÜñèñùóç ğñïò ğáñïõóßáóç";
+				Text[ korean ] = "í”„ë ˆì  í…Œì´ì…˜ì—ì„œì˜ ì•„ì›ƒíŠ¸ë¼ì¸(~P)";
+				Text[ turkish ] = "~Anahatları sunuya aktar";
+				Text[ language_user1 ] = " ";
+				Text[ catalan ] = "Outline to ~Presentation";
+				Text[ finnish ] = "Jäsennys -> ~esitys";
+				Text[ extern ] = "Outline to ~Presentation";
+			};
+			MenuItem
+			{
+				Identifier = FN_OUTLINE_TO_CLIPBOARD ;
+				HelpID = FN_OUTLINE_TO_CLIPBOARD ;
+				Text = "Gliederung zum ~Clipboard" ;
+				Text [ English ] = "Outline to Clipboard" ;
+				Text [ portuguese ] = "Esquema em área de ~transferência" ;
+				Text [ english_us ] = "Outline to ~Clipboard" ;
+				Text [ portuguese_brazilian ] = "Gliederung zum Clipboard" ;
+				Text [ swedish ] = "Disposition till ur~klippet" ;
+				Text [ danish ] = "Disposition til ~udklipsholder" ;
+				Text [ italian ] = "Struttura in clip~board" ;
+				Text [ spanish ] = "Esquema para ~portapapeles" ;
+				Text [ french ] = "Pla~n vers le presse-papiers" ;
+				Text [ dutch ] = "Overzicht naar ~Klembord" ;
+				Text[ chinese_simplified ] = "å¤§çº²é€è‡³å‰ªè´´æ¿(~C)";
+				Text[ russian ] = "Ñòğóêòóğó â áóôåğ îáìåíà";
+				Text[ polish ] = "Konspekt do schowka";
+				Text[ japanese ] = "ã‚¯ãƒªãƒƒãƒ—ãƒœãƒ¼ãƒ‰ã®ã‚¢ã‚¦ãƒˆãƒ©ã‚¤ãƒ³(~C)";
+				Text[ chinese_traditional ] = "å¤§ç¶±é€è‡³å‰ªè²¼ç°¿(~C)";
+				Text[ arabic ] = "ãÎØØ ÊİÕíáí ááÍÇİÙÉ";
+				Text[ dutch ] = "Overzicht naar ~Klembord";
+				Text[ chinese_simplified ] = "å¤§çº²é€è‡³å‰ªè´´æ¿(~C)";
+				Text[ greek ] = "ÄéÜñèñùóç ğñïò ğñü~÷åéñï";
+				Text[ korean ] = "í´ë¦½ë³´ë“œë¥¼ ìœ„í•œ ì•„ì›ƒíŠ¸ë¼ì¸(~C)";
+				Text[ turkish ] = "Anahatları ~panoya aktar";
+				Text[ language_user1 ] = " ";
+				Text[ catalan ] = "Outline to ~Clipboard";
+				Text[ finnish ] = "Jäsennys -> ~leikepöytä";
+				Text[ extern ] = "Outline to ~Clipboard";
+			};
+			SEPARATOR ;
+			MenuItem
+			{
+				Identifier = FN_ABSTRACT_NEWDOC ;
+				HelpID = FN_ABSTRACT_NEWDOC ;
+				Text = "A~utoAbstract erzeugen..." ;
+				Text [ English ] = "Create summary..." ;
+				Text [ portuguese ] = "Criar Auto~Abstracto..." ;
+				Text [ english_us ] = "Create A~utoAbstract..." ;
+				Text [ portuguese_brazilian ] = "A~utoAbstract erzeugen..." ;
+				Text [ swedish ] = "Skapa A~utoUtdrag..." ;
+				Text [ danish ] = "Opret A~utoAbstract..." ;
+				Text [ italian ] = "Crea a~bstract automatico..." ;
+				Text [ spanish ] = "Crear ~AutoAbstracto..." ;
+				Text [ french ] = "Créer une A~utoSynthèse..." ;
+				Text [ dutch ] = "~AutoExtract maken..." ;
+				Text[ chinese_simplified ] = "ç”Ÿæˆè‡ªåŠ¨æ‘˜è¦(~U)...";
+				Text[ russian ] = "Ñîçäàòü àâòîğåôåğàò...";
+				Text[ polish ] = "Utwórz Autokonspekt...";
+				Text[ japanese ] = "è‡ªå‹•æŠ½å‡ºã®ä½œæˆ(~U)...";
+				Text[ chinese_traditional ] = "è£½ä½œè‡ªå‹•æ‘˜è¦(~U)...";
+				Text[ arabic ] = "...AutoAbstract ÅäÔÇÁ";
+				Text[ dutch ] = "~AutoExtract maken...";
+				Text[ chinese_simplified ] = "ç”Ÿæˆè‡ªåŠ¨æ‘˜è¦(~U)...";
+				Text[ greek ] = "Äçìéïõñãßá A~utoAbstract...";
+				Text[ korean ] = "ìë™ì¶”ìƒ ì‘ì„±(~U)...";
+				Text[ turkish ] = "Otomatik özet oluştur...";
+				Text[ language_user1 ] = " ";
+				Text[ catalan ] = "Create A~utoAbstract...";
+				Text[ finnish ] = "Luo automaattinen ~tiivistelmä";
+				Text[ extern ] = "Create A~utoAbstract...";
+			};
+			MenuItem
+			{
+				Identifier = FN_ABSTRACT_STARIMPRESS ;
+				HelpID = FN_ABSTRACT_STARIMPRESS ;
+				/* ### ACHTUNG: Neuer Text in Resource? AutoAbstract an ~Präsentation... : AutoAbstract an ~Prõsentation... */
+				Text = "AutoAbstract an ~Präsentation..." ;
+				Text [ English ] = "Summary to presentation..." ;
+				Text [ english_us ] = "AutoAbst~ract to Presentation..." ;
+				Text [ portuguese_brazilian ] = "AutoAbstract an ~Pr?sentation..." ;
+				Text [ swedish ] = "AutoUtdrag till ~presentation..." ;
+				Text [ danish ] = "AutoAbstract til ~præsentation..." ;
+				Text [ italian ] = "Abstract automatico alla ~presentazione..." ;
+				Text [ spanish ] = "AutoAbstracto para ~presentación..." ;
+				Text [ french ] = "AutoSynthèse vers ~présentation..." ;
+				Text [ dutch ] = "A~utoExtract aan presentatie..." ;
+				Text [ portuguese ] = "AutoAbstracto em ~apresentação..." ;
+				Text[ chinese_simplified ] = "è‡ªåŠ¨æ‘˜è¦é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~R)...";
+				Text[ russian ] = "Àâòîğåôåğàò â ïğåçåíòàöèş...";
+				Text[ polish ] = "Autokonspekt do prezentacji...";
+				Text[ japanese ] = "ãƒ—ãƒ¬ã‚¼ãƒ³ãƒ†ãƒ¼ã‚·ãƒ§ãƒ³ã®è‡ªå‹•æŠ½å‡º(~R)...";
+				Text[ chinese_traditional ] = "è‡ªå‹•æ‘˜è¦é€è‡³ç°¡å ±(~R)...";
+				Text[ arabic ] = "...Åáì ÚÑÖ ÊŞÏíãí AutoAbstract";
+				Text[ dutch ] = "A~utoExtract aan presentatie...";
+				Text[ chinese_simplified ] = "è‡ªåŠ¨æ‘˜è¦é€è‡³æ¼”ç¤ºæ–‡ç¨¿(~R)...";
+				Text[ greek ] = "AutoAbstract ùò ğñïò ~ğáñïõóßáóç...";
+				Text[ korean ] = "í”„ë ˆì  í…Œì´ì…˜ì—ì„œì˜ ìë™ì¶”ìƒ(~R)...";
+				Text[ turkish ] = "Otomatik ~özeti sunuya aktar...";
 				Text[ language_user1 ] = " ";
-				Text[ catalan ] = "~Envia";
-				Text[ finnish ] = "Lähe~tä";
-				Text[ extern ] = "Sen~d";
+				Text[ catalan ] = "AutoAbst~ract to Presentation...";
+				Text[ finnish ] = "Automaattinen ~tiivistelmä esitykseen";
+				Text[ extern ] = "AutoAbst~ract to Presentation...";
 			};
 #endif
 			SEPARATOR ;
Index: svx/inc/globlmn.hrc
===================================================================
RCS file: /cvs/oo/graphics/svx/inc/globlmn.hrc,v
retrieving revision 1.71.2.6
diff -u -p -u -r1.71.2.6 globlmn.hrc
--- svx/inc/globlmn.hrc	26 Mar 2003 14:52:18 -0000	1.71.2.6
+++ svx/inc/globlmn.hrc	28 Apr 2003 11:08:50 -0000
@@ -1969,13 +1969,13 @@
     Identifier = SID_MAIL_SENDDOC ; \
     HelpID = SID_MAIL_SENDDOC ; \
     Text = "Dokument als ~E-Mail..." ; \
-    Text [ english ] = "Document as ~eMail..." ; \
-    Text [ english_us ] = "Document As ~E-mail..." ; \
+    Text [ english ] = "Send Document as ~eMail..." ; \
+    Text [ english_us ] = "Send Document As ~E-mail..." ; \
     Text [ portuguese_brazilian ] = "Documento como E-mail..." ; \
     Text [ swedish ] = "Dokument som e-post..." ; \
     Text [ danish ] = "Dokument som ~e-mail..." ; \
     Text [ italian ] = "Documento come ~eMail..." ; \
-    Text [ spanish ] = "~Documento como e-mail..." ; \
+    Text [ spanish ] = "Enviar ~Documento como e-mail..." ; \
     Text [ french ] = "Envoyer par ~e-mail..." ; \
     Text [ dutch ] = "Document als e-~mail..." ; \
     Text [ portuguese ] = "~Documento como e-mail..." ;\
