diff -u -p -u -r1.4 -r1.5
--- sd/source/filter/sdpptwrp.cxx	16 Mar 2001 17:41:14 -0000	1.4
+++ sd/source/filter/sdpptwrp.cxx	19 Jun 2002 11:20:55 -0000	1.5
@@ -112,8 +112,16 @@ sal_Bool SdPPTFilter::Import()
 
 	if( ( pStorage = mrMedium.GetStorage() ) != NULL )
 	{
+		/* check if there is a dualstorage, then the
+		document is propably a PPT95 containing PPT97 */
+		SvStorageRef xDualStorage;
+		String sDualStorage( RTL_CONSTASCII_USTRINGPARAM( "PP97_DUALSTORAGE" ) );
+		if ( pStorage->IsContained( sDualStorage ) )
+		{
+			xDualStorage = pStorage->OpenStorage( sDualStorage, STREAM_STD_READ );
+			pStorage = xDualStorage;
+		}
 		SvStream* pDocStream = pStorage->OpenStream( String( RTL_CONSTASCII_USTRINGPARAM("PowerPoint Document") ), STREAM_STD_READ );
-		
 		if( pDocStream )
 		{
 			pDocStream->SetVersion( pStorage->GetVersion() );
