Index: common_gfx.cxx
===================================================================
RCS file: /cvs/oo/gsl/psprint/source/printergfx/common_gfx.cxx,v
retrieving revision 1.7
diff -u -r1.7 common_gfx.cxx
--- common_gfx.cxx	2001/12/13 17:58:09	1.7
+++ psprint/source/printergfx/common_gfx.cxx	2002/10/02 00:02:55
@@ -507,7 +507,7 @@
     {
         PSSetColor (maFillColor);
         PSSetColor ();
-        WritePS (mpPageBody, "fill\n");
+        WritePS (mpPageBody, "eofill\n");
     }
 
     // restore the current path
