--- setup2/mow/source/loader/loader.c	Thu Oct 31 14:46:52 2002
+++ setup2/mow/source/loader/loader.c	Wed Jan 29 15:33:25 2003
@@ -156,6 +156,7 @@
 
 BOOL m_bNoJunk = FALSE;
 BOOL m_bUseX = TRUE;
+BOOL m_bDebug = FALSE;
 
 typedef struct EventData
 {
@@ -1652,7 +1653,6 @@
     pid_t pid;
     int nWaitpidValue;
     int nStatus;
-    int nCheckForX = 1;
 
 #if defined LINUX
     check_system();
@@ -1685,22 +1685,17 @@
         exit(0);
     }
 
+	m_bUseX = TRUE;
     for ( i=1 ; i<argc; i++ )
     {
         if ( ( strncasecmp( argv[i], "-r:", 3 ) == 0 ) ||
              ( strncasecmp( argv[i], "-rsp1:", 6 ) == 0 ) ||
              ( strncasecmp( argv[i], "-rsp2:", 6 ) == 0 ) )
-        {
-            nCheckForX = 0;
-            break;
-        }
+            m_bUseX = FALSE;
+	else if ( ( strcasecmp( argv[i], "-debug") ) == 0 )
+            m_bDebug = TRUE;
     }
 
-    if ( nCheckForX == 1 )
-        m_bUseX = MayUseX( argc, argv );
-    else
-        m_bUseX = FALSE;
-
     // we need a LD_LIBRARY_PATH with a current directory in it so we can
     // dynamicly load a library ( needed for patching! )
     pLibPath = getenv("LD_LIBRARY_PATH");
@@ -1893,7 +1888,8 @@
             } break;
 
             case SETUP_EVENT_EXIT_LOADER :
-                KillSetupDir();
+                if ( !m_bDebug )
+                    KillSetupDir();
 
                 if ( m_bUseX )
                 {
