Index: sw/source/core/layout/paintfrm.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/layout/paintfrm.cxx,v
retrieving revision 1.69
diff -u -p -u -r1.69 paintfrm.cxx
--- sw/source/core/layout/paintfrm.cxx	4 Jul 2003 13:22:46 -0000	1.69
+++ sw/source/core/layout/paintfrm.cxx	3 Mar 2004 09:51:51 -0000
@@ -2502,7 +2502,7 @@ void SwLayoutFrm::Paint( const SwRect& r
 						 pGlobalShell->VisArea().Top() < nDeadline )
 					{
 						bOneBeepOnly = FALSE;
-						Sound::Beep();
+//						Sound::Beep();
 					}
 				}
 			}
