diff -rup cppcanvas-orig/source/mtfrenderer/emfplus.cxx cppcanvas/source/mtfrenderer/emfplus.cxx
--- cppcanvas-orig/source/mtfrenderer/emfplus.cxx	2008-02-25 12:21:51.000000000 +0100
+++ cppcanvas/source/mtfrenderer/emfplus.cxx	2008-02-25 12:42:45.000000000 +0100
@@ -783,6 +783,10 @@ namespace cppcanvas
                 EMFPBrush* brush = (EMFPBrush*) aObjects [brushIndexOrColor];
                 EMFP_DEBUG (printf ("EMF+\tbrush fill slot: %d (type: %d)\n", brushIndexOrColor, brush->GetType ()));
 
+                // give up in case something wrong happened
+                if( !brush )
+                    return;
+
                 rState.isFillColorSet = false;
                 rState.isLineColorSet = false;
 
@@ -992,7 +996,8 @@ namespace cppcanvas
                             aObjects [index] = NULL;
                         }
 
-                        switch (flags & 0xff00) {
+                        // not sure yet, what 0x8000 means
+                        switch (flags & 0x7f00) {
                         case EmfPlusObjectTypeBrush:
                             {
                                 EMFPBrush *brush;
