--- offapi/com/sun/star/rendering/makefile.mk.old	2009-04-02 10:51:32.000000000 +0000
+++ offapi/com/sun/star/rendering/makefile.mk	2009-04-06 16:42:16.000000000 +0000
@@ -60,6 +60,7 @@ IDLFILES=\
 	FontRequest.idl \
 	IntegerBitmapLayout.idl \
 	InterpolationMode.idl \
+	MtfRenderer.idl \
 	Panose.idl \
 	PanoseArmStyle.idl \
 	PanoseContrast.idl \
--- offapi/com/sun/star/rendering/MtfRenderer.idl.old	1970-01-01 00:00:00.000000000 +0000
+++ offapi/com/sun/star/rendering/MtfRenderer.idl	2009-04-06 16:42:16.000000000 +0000
@@ -0,0 +1,19 @@
+#ifndef __com_sun_star_rendering_MtfRenderer_idl__
+#define __com_sun_star_rendering_MtfRenderer_idl__
+
+#ifndef __com_sun_star_uno_XCanvas_idl__ 
+#include <com/sun/star/rendering/XCanvas.idl> 
+#endif 
+
+module com { module sun { module star { module rendering {
+
+	interface XMtfRenderer : ::com::sun::star::uno::XInterface
+	{
+		void setMetafile( [in] sequence< byte > aMtf );
+		void draw( [in] double fScaleX, [in] double fScaleY );
+	};
+
+	service MtfRenderer : XMtfRenderer;
+}; }; }; };
+
+#endif
