--- canvas/source/directx/dx_canvashelper_texturefill.cxx
+++ canvas/source/directx/dx_canvashelper_texturefill.cxx
@@ -711,8 +711,8 @@
                     const ::canvas::ParametricPolyPolygon::Values& rValues(
                         pGradient->getValues() );
 
-                    const Gdiplus::Color aColor1(tools::sequenceToArgb(rValues.maColor1));
-                    const Gdiplus::Color aColor2(tools::sequenceToArgb(rValues.maColor2));
+                    const Gdiplus::Color aColor1(tools::sequenceToArgb(rValues.maColors[0]));
+                    const Gdiplus::Color aColor2(tools::sequenceToArgb(rValues.maColors[rValues.maColors.getLength()-1]));
 
                     // TODO(E1): Return value
                     // TODO(F1): FillRule
