Index: framework/source/uielement/statusbar.cxx
===================================================================
RCS file: /cvs/framework/framework/source/uielement/statusbar.cxx,v
retrieving revision 1.4
retrieving revision 1.4.12.1
diff -u -p -u -p -r1.4 -r1.4.12.1
--- framework/source/uielement/statusbar.cxx	31 Jan 2005 13:07:51 -0000	1.4
+++ framework/source/uielement/statusbar.cxx	9 Feb 2005 18:05:58 -0000	1.4.12.1
@@ -118,6 +118,7 @@ void FrameworkStatusBar::StateChanged( S
 
 void FrameworkStatusBar::DataChanged( const DataChangedEvent& rDCEvt )
 {
+    StatusBar::DataChanged( rDCEvt );
     if ( m_pMgr )
         m_pMgr->DataChanged( rDCEvt );
 }
Index: framework/source/uielement/toolbarmanager.cxx
===================================================================
RCS file: /cvs/framework/framework/source/uielement/toolbarmanager.cxx,v
retrieving revision 1.12
retrieving revision 1.12.20.1
diff -u -p -u -p -r1.12 -r1.12.20.1
--- framework/source/uielement/toolbarmanager.cxx	21 Jan 2005 12:42:13 -0000	1.12
+++ framework/source/uielement/toolbarmanager.cxx	9 Feb 2005 17:51:48 -0000	1.12.20.1
@@ -298,7 +298,7 @@ ToolBarManager::ToolBarManager( const Re
     m_xFrame( rFrame ),
     m_pToolBar( pToolBar ),
     m_bDisposed( sal_False ),
-    m_bIsHiContrast( pToolBar->GetBackground().GetColor().IsDark() ),
+    m_bIsHiContrast( pToolBar->GetDisplayBackground().GetColor().IsDark() ),
     m_bSmallSymbols( GetCurrentSymbolSize() == SFX_SYMBOLS_SMALL ),
     m_bModuleIdentified( sal_False ),
     m_aResourceName( rResourceName ),
@@ -383,7 +383,7 @@ void ToolBarManager::CheckAndUpdateImage
     sal_Bool bRefreshImages = sal_False;
 
     // Check if high contrast/normal mode have changed
-	if ( m_pToolBar->GetBackground().GetColor().IsDark() )
+	if ( m_pToolBar->GetDisplayBackground().GetColor().IsDark() )
 	{
 		if ( !m_bIsHiContrast )
 		{
@@ -1340,7 +1340,7 @@ void ToolBarManager::RequestImages()
         ++pIter;
     }
 
-    m_bIsHiContrast = m_pToolBar->GetBackground().GetColor().IsDark();
+    m_bIsHiContrast = m_pToolBar->GetDisplayBackground().GetColor().IsDark();
     sal_Int16 j = getImageTypeFromBools( ( nSymbolSet == SFX_SYMBOLS_LARGE ), m_bIsHiContrast );
 
     if ( m_xDocImageManager.is() )
Index: officecfg/registry/data/org/openoffice/VCL.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/VCL.xcu,v
retrieving revision 1.37
retrieving revision 1.37.28.1
diff -u -p -u -p -r1.37 -r1.37.28.1
--- officecfg/registry/data/org/openoffice/VCL.xcu	31 Jan 2005 13:42:56 -0000	1.37
+++ officecfg/registry/data/org/openoffice/VCL.xcu	14 Feb 2005 17:02:12 -0000	1.37.28.1
@@ -2473,7 +2473,7 @@
       </node>
       <node oor:name="gothic" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hggothicbsun;msgothic;mspgothic;sazanamigothic;hggothic;hggothicb;hggothice;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;centurygothic;avantgarde;itcavantgarde;gothic;avantgardegothic;conga;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>hggothicbsun;msgothic;mspgothic;hggothic;hggothicb;hggothice;ipagothic;kochigothic;sazanamigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;centurygothic;avantgarde;itcavantgarde;gothic;avantgardegothic;conga;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2496,7 +2496,7 @@
       </node>
       <node oor:name="gothicb" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hggothicbsun;msgothic;mspgothic;hggothic;hggothicb;hggothice;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>hggothicbsun;msgothic;mspgothic;hggothic;hggothicb;hggothice;ipagothic;kochigothic;sazanamigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2519,7 +2519,7 @@
       </node>
       <node oor:name="gothicl" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>msgothic;mspgothic;hggothic;hggothicb;hggothice;;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;centurygothic;avantgarde;itcavantgarde;gothic;avantgardegothic;conga;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>msgothic;mspgothic;hggothic;hggothicb;hggothice;ipapgothic;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;centurygothic;avantgarde;itcavantgarde;gothic;avantgardegothic;conga;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2879,7 +2879,7 @@
       </node>
       <node oor:name="hgpmincholsun" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>mspmincho;sazanamimincho;kochimincho;andalesansui;mincho;arialunicodems;lucidaunicode</value>
+          <value>mspmincho;ipapmincho;sazanamimincho;kochimincho;andalesansui;mincho;arialunicodems;lucidaunicode</value>
         </prop>
       </node>
        <node oor:name="hggothic" oor:op="replace">
@@ -2907,7 +2907,7 @@
       </node>
       <node oor:name="hggothicb" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>msgothic;mspgothic;hggothic;hggothice;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>msgothic;mspgothic;hggothic;hggothice;ipagothic;ipapgothic;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2930,7 +2930,7 @@
       </node>
       <node oor:name="hggothice" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>msgothic;mspgothic;hggothic;hggothicb;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>msgothic;mspgothic;hggothic;hggothicb;ipagothic;ipapgothic;sazanamigothic;kochigothic;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2953,7 +2953,7 @@
       </node>
       <node oor:name="hgminchoj" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholightj;msmincho;mspmincho;sazanamimincho;kochimincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>hgmincholightj;msmincho;mspmincho;ipapmincho;sazanamimincho;kochimincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2976,7 +2976,7 @@
       </node>
       <node oor:name="hgminchol" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholsun;hgmincholightj;msmincho;mspmincho;hgminchoj;sazanamimincho;kochimincho;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgmincholsun;hgmincholightj;msmincho;mspmincho;hgminchoj;ipamincho;ipapmincho;sazanamimincho;kochimincho;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -2999,7 +2999,7 @@
       </node>
       <node oor:name="hgmincholightj" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholsun;msmincho;mspmincho;sazanamimincho;hgmincholightj;kochimincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>hgmincholsun;msmincho;mspmincho;ipapmincho;sazanamimincho;hgmincholightj;kochimincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4379,7 +4379,7 @@
       </node>
       <node oor:name="mincho" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholsun;hgmincholightj;msmincho;mspmincho;sazanamimincho;hgmincholightj;hgminchoj;kochimincho;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgmincholsun;hgmincholightj;msmincho;mspmincho;ipamincho;ipapmincho;sazanamimincho;hgmincholightj;hgminchoj;kochimincho;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4425,7 +4425,7 @@
       </node>
       <node oor:name="minchol" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholsun;hgmincholightj;msmincho;sazanamimincho;kochimincho;mspmincho;hgminchoj;hgminchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgmincholsun;hgmincholightj;msmincho;ipamincho;sazanamimincho;kochimincho;mspmincho;hgminchoj;hgminchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4448,7 +4448,7 @@
       </node>
       <node oor:name="minchou" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholsun;hgmincholightj;msmincho;sazanamimincho;kochimincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgmincholsun;hgmincholightj;msmincho;ipamicho;ipapmincho;sazanamimincho;kochimincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4701,7 +4701,7 @@
       </node>
       <node oor:name="msgothic" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hggothicbsun;mspgothic;hggothic;hggothicb;sazanamigothic;kochigothic;hggothice;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;arialunicodems;lucidaunicode</value>
+          <value>hggothicbsun;mspgothic;hggothic;hggothicb;ipagothic;sazanamigothic;kochigothic;hggothice;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4724,7 +4724,7 @@
       </node>
       <node oor:name="msmincho" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgmincholsun;hgmincholightj;sazanamimincho;kochimincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgmincholsun;hgmincholightj;ipamincho;sazanamimincho;kochimincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4770,7 +4770,7 @@
       </node>
       <node oor:name="mspmincho" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgpmincholsun;hgmincholightj;msmincho;sazanamimincho;hgminchoj;hgminchol;kochimincho;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgpmincholsun;hgmincholightj;msmincho;ipapmincho;sazanamimincho;hgminchoj;hgminchol;kochimincho;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -4793,7 +4793,7 @@
       </node>
       <node oor:name="msuigothic" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hggothicbsun;mspgothic;msgothic;hggothic;hggothicb;sazanamigothic;kochigothic;hggothice;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;arialunicodems;lucidaunicode</value>
+          <value>hggothicbsun;mspgothic;msgothic;hggothic;hggothicb;ipauigothic;sazanamigothic;kochigothic;hggothice;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -5299,7 +5299,7 @@
       </node>
       <node oor:name="pgothic" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgpgothicbsun;mspgothic;msgothic;hggothic;hggothicb;sazanamigothic;kochigothic;hggothice;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
+          <value>hgpgothicbsun;mspgothic;msgothic;hggothic;hggothicb;ipapgothic;sazanamigothic;kochigothic;hggothice;andalesansui;gothic;hgmincholightj;msmincho;mspmincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;andalesansui;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
@@ -5368,7 +5368,7 @@
       </node>
       <node oor:name="pmincho" oor:op="replace">
         <prop oor:name="SubstFonts">
-          <value>hgpminchobsun;hgmincholightj;mspmincho;msmincho;sazanamimincho;kochimincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
+          <value>hgpminchobsun;hgmincholightj;mspmincho;msmincho;ipapmincho;sazanamimincho;kochimincho;hgminchoj;hgminchol;minchol;mincho;hgheiseimin;heiseimin;minchou;msgothic;mspgothic;hggothic;hggothicb;hggothice;andalesansui;gothic;arialunicodems;lucidaunicode</value>
         </prop>
         <prop oor:name="SubstFontsMS">
           <value/>
Index: tools/source/rc/resmgr.cxx
===================================================================
RCS file: /cvs/util/tools/source/rc/resmgr.cxx,v
retrieving revision 1.36
retrieving revision 1.36.2.2
diff -u -p -u -p -r1.36 -r1.36.2.2
--- tools/source/rc/resmgr.cxx	18 Jan 2005 13:30:47 -0000	1.36
+++ tools/source/rc/resmgr.cxx	8 Feb 2005 10:35:04 -0000	1.36.2.2
@@ -603,7 +603,7 @@ ResMgr* Resource::GetResManager()
 struct ImpContent
 {
     sal_uInt64   nTypeAndId;
-    ULONG        nOffset;
+    sal_uInt32   nOffset;
 };
 
 struct ImpContentLessCompare : public ::std::binary_function< ImpContent, ImpContent, bool>
@@ -746,15 +746,15 @@ BOOL InternalResMgr::Create()
             if ( pLogFile )
             {
                 pResUseDump = new std::hash_map<sal_uInt64, int>;
-                for( ULONG i = 0; i < nEntries; ++i )
+                for( sal_uInt32 i = 0; i < nEntries; ++i )
                     (*pResUseDump)[pContent[i].nTypeAndId] = 1;
             }
 #endif
             // swap the content to the right endian
             pContent[0].nTypeAndId = ResMgr::GetUInt64( pContentBuf );
             pContent[0].nOffset = ResMgr::GetLong( pContentBuf+8 );
-			ULONG nCount = nEntries - 1;
-            for( ULONG i = 0,j=1; i < nCount; ++i,++j )
+			sal_uInt32 nCount = nEntries - 1;
+            for( sal_uInt32 i = 0,j=1; i < nCount; ++i,++j )
             {
                 // swap the content to the right endian
                 pContent[j].nTypeAndId = ResMgr::GetUInt64( pContentBuf + (12*j) );
@@ -789,9 +789,6 @@ BOOL InternalResMgr::IsGlobalAvailable( 
 											pContent + nEntries,
 											nValue,
 											ImpContentMixLessCompare());
-//    ImpContent * pFind = (ImpContent *)
-//        bsearch( (void *)((ULONG(nRT) << 16) | nId), pContent, nEntries,
-//                sizeof( ImpContent ), Search );
     return (pFind != (pContent + nEntries)) && (pFind->nTypeAndId == nValue);
 }
 
@@ -811,9 +808,6 @@ void* InternalResMgr::LoadGlobalRes( RES
 											pEnd,
 											nValue,
 											ImpContentMixLessCompare());
-//    ImpContent * pFind = (ImpContent *)
-//        bsearch( (void *)((ULONG(nRT) << 16) | nId), pContent, nEntries,
-//                sizeof( ImpContent ), Search );
     if( pFind && (pFind != pEnd) && (pFind->nTypeAndId == nValue) )
 	{
 		if( nRT == RSC_STRING && bEqual2Content )
@@ -1156,7 +1150,8 @@ void ResMgr::decStack()
     }
     else
     {
-        if( (aStack[nCurStack].Flags & RC_FALLBACK_DOWN) )
+        ImpRCStack& rTop = aStack[nCurStack];
+        if( (rTop.Flags & RC_FALLBACK_DOWN) )
         {
             #if OSL_DEBUG_LEVEL > 1
             fprintf( stderr, "returning from fallback %s\n",
@@ -1164,7 +1159,7 @@ void ResMgr::decStack()
             #endif
             delete pFallbackResMgr;
             pFallbackResMgr = NULL;
-            Resource::SetResManager( this );
+            Resource::SetResManager( rTop.pResMgr );
         }
         nCurStack--;
     }
Index: psprint/inc/cupsmgr.hxx
===================================================================
RCS file: /cvs/gsl/psprint/inc/cupsmgr.hxx,v
retrieving revision 1.5
retrieving revision 1.5.10.1
diff -u -p -u -p -r1.5 -r1.5.10.1
--- psprint/inc/cupsmgr.hxx	9 Nov 2004 16:35:33 -0000	1.5
+++ psprint/inc/cupsmgr.hxx	9 Feb 2005 16:19:05 -0000	1.5.10.1
@@ -104,6 +104,7 @@ class CUPSManager : public PrinterInfoMa
 
     virtual void initialize();
 
+    void getOptionsFromDocumentSetup( const JobData& rJob, int& rNumOptions, void** rOptions ) const;
     void runDests();
     static void runDestThread(void* pMgr);
 public:
@@ -116,7 +117,7 @@ public:
     const char* authenticateUser( const char* );
 
     virtual FILE* startSpool( const rtl::OUString& rPrinterName );
-    virtual int endSpool( const rtl::OUString& rPrinterName, const rtl::OUString& rJobTitle, FILE* pFile );
+    virtual int endSpool( const rtl::OUString& rPrinterName, const rtl::OUString& rJobTitle, FILE* pFile, const JobData& rDocumentJobData );
     virtual void setupJobContextData( JobData& rData );
 
     // changes the info about a named printer
Index: psprint/inc/psprint/printerinfomanager.hxx
===================================================================
RCS file: /cvs/gsl/psprint/inc/psprint/printerinfomanager.hxx,v
retrieving revision 1.9
retrieving revision 1.9.10.1
diff -u -p -u -p -r1.9 -r1.9.10.1
--- psprint/inc/psprint/printerinfomanager.hxx	9 Nov 2004 16:37:13 -0000	1.9
+++ psprint/inc/psprint/printerinfomanager.hxx	9 Feb 2005 16:19:05 -0000	1.9.10.1
@@ -270,7 +270,7 @@ public:
     virtual FILE* startSpool( const rtl::OUString& rPrinterName );
     // close the FILE* returned by startSpool and does the actual spooling
     // returns a numerical job id
-    virtual int endSpool( const rtl::OUString& rPrinterName, const rtl::OUString& rJobTitle, FILE* pFile );
+    virtual int endSpool( const rtl::OUString& rPrinterName, const rtl::OUString& rJobTitle, FILE* pFile, const JobData& rDocumentJobData );
     
     // for spadmin: whether adding or removing a printer is possible
     virtual bool addOrRemovePossible() const;
Index: psprint/source/fontmanager/fontmanager.cxx
===================================================================
RCS file: /cvs/gsl/psprint/source/fontmanager/fontmanager.cxx,v
retrieving revision 1.56
retrieving revision 1.56.2.1
diff -u -p -u -p -r1.56 -r1.56.2.1
--- psprint/source/fontmanager/fontmanager.cxx	31 Jan 2005 08:59:28 -0000	1.56
+++ psprint/source/fontmanager/fontmanager.cxx	11 Feb 2005 11:07:18 -0000	1.56.2.1
@@ -3323,7 +3323,7 @@ bool PrintFontManager::checkImportPossib
     // find a directory with write access
     ByteString aDir;
     for( std::list< int >::const_iterator dir_it = m_aPrivateFontDirectories.begin();
-         dir_it != m_aFontDirectories.end(); ++dir_it )
+         dir_it != m_aPrivateFontDirectories.end(); ++dir_it )
     {
         aDir = getDirectory( *dir_it );
         if( ! access( aDir.GetBuffer(), W_OK ) || createPath( aDir ) )
Index: psprint/source/printer/cupsmgr.cxx
===================================================================
RCS file: /cvs/gsl/psprint/source/printer/cupsmgr.cxx,v
retrieving revision 1.9
retrieving revision 1.9.10.1
diff -u -p -u -p -r1.9 -r1.9.10.1
--- psprint/source/printer/cupsmgr.cxx	9 Nov 2004 16:37:51 -0000	1.9
+++ psprint/source/printer/cupsmgr.cxx	9 Feb 2005 16:19:06 -0000	1.9.10.1
@@ -76,7 +76,7 @@ typedef void cups_option_t;
 #include <rtl/ustrbuf.hxx>
 #include <cupsmgr.hxx>
 
-#include <unistd.h>
+#include <algorithm>
 
 namespace psp
 {
@@ -391,6 +391,9 @@ void CUPSManager::runDests()
     int nDests = 0;
     cups_dest_t* pDests = NULL;
     nDests = m_pCUPSWrapper->cupsGetDests( &pDests );
+#if OSL_DEBUG_LEVEL > 1
+    fprintf( stderr, "came out of cupsGetDests\n" );
+#endif
 
     osl::MutexGuard aGuard( m_aCUPSMutex );
     m_nDests = nDests;
@@ -563,68 +566,75 @@ const PPDParser* CUPSManager::createCUPS
         aPrinter = rPrinter;
 
 #ifdef ENABLE_CUPS
-    if( m_aCUPSMutex.tryToAcquire() && m_nDests && m_pDests )
+    if( m_aCUPSMutex.tryToAcquire() )
     {
-        std::hash_map< OUString, int, OUStringHash >::iterator dest_it =
-            m_aCUPSDestMap.find( aPrinter );
-        if( dest_it != m_aCUPSDestMap.end() )
+        if( m_nDests && m_pDests )
         {
-            cups_dest_t* pDest = ((cups_dest_t*)m_pDests) + dest_it->second;
-            const char* pPPDFile = m_pCUPSWrapper->cupsGetPPD( pDest->name );
-#if OSL_DEBUG_LEVEL > 1
-            fprintf( stderr, "PPD for %s is %s\n", OUStringToOString( aPrinter, osl_getThreadTextEncoding() ).getStr(), pPPDFile );
-#endif
-            if( pPPDFile )
+            std::hash_map< OUString, int, OUStringHash >::iterator dest_it =
+            m_aCUPSDestMap.find( aPrinter );
+            if( dest_it != m_aCUPSDestMap.end() )
             {
-                rtl_TextEncoding aEncoding = osl_getThreadTextEncoding();
-                OUString aFileName( OStringToOUString( pPPDFile, aEncoding ) );
-                // update the printer info with context information
-                ppd_file_t* pPPD = m_pCUPSWrapper->ppdOpenFile( pPPDFile );
-                if( pPPD )
+                cups_dest_t* pDest = ((cups_dest_t*)m_pDests) + dest_it->second;
+                const char* pPPDFile = m_pCUPSWrapper->cupsGetPPD( pDest->name );
+                #if OSL_DEBUG_LEVEL > 1
+                fprintf( stderr, "PPD for %s is %s\n", OUStringToOString( aPrinter, osl_getThreadTextEncoding() ).getStr(), pPPDFile );
+                #endif
+                if( pPPDFile )
                 {
-                    // create the new parser
-                    PPDParser* pCUPSParser =  new PPDParser( aFileName );
-                    pCUPSParser->m_aFile = rPrinter;
-                    pNewParser = pCUPSParser;
-                    
-                    /*int nConflicts =*/ m_pCUPSWrapper->cupsMarkOptions( pPPD, pDest->num_options, pDest->options );
-#if OSL_DEBUG_LEVEL > 1
-                    fprintf( stderr, "processing the following options for printer %s (instance %s):\n",
-                    pDest->name, pDest->instance );
-                    for( int k = 0; k < pDest->num_options; k++ )
-                        fprintf( stderr, "   \"%s\" = \"%s\"\n",
-                    pDest->options[k].name,
-                    pDest->options[k].value );
-#endif
-                    PrinterInfo& rInfo = m_aPrinters[ aPrinter ].m_aInfo;
-                    
-                    // remember the default context for later use
-                    PPDContext& rContext = m_aDefaultContexts[ aPrinter ];
-                    rContext.setParser( pNewParser );
-                    for( int i = 0; i < pPPD->num_groups; i++ )
-                        updatePrinterContextInfo( pPPD->groups + i, rContext );
-                    
-                    rInfo.m_pParser = pNewParser;
-                    rInfo.m_aContext = rContext;
+                    rtl_TextEncoding aEncoding = osl_getThreadTextEncoding();
+                    OUString aFileName( OStringToOUString( pPPDFile, aEncoding ) );
+                    // update the printer info with context information
+                    ppd_file_t* pPPD = m_pCUPSWrapper->ppdOpenFile( pPPDFile );
+                    if( pPPD )
+                    {
+                        // create the new parser
+                        PPDParser* pCUPSParser =  new PPDParser( aFileName );
+                        pCUPSParser->m_aFile = rPrinter;
+                        pNewParser = pCUPSParser;
+                        
+                        /*int nConflicts =*/ m_pCUPSWrapper->cupsMarkOptions( pPPD, pDest->num_options, pDest->options );
+                        #if OSL_DEBUG_LEVEL > 1
+                        fprintf( stderr, "processing the following options for printer %s (instance %s):\n",
+                        pDest->name, pDest->instance );
+                        for( int k = 0; k < pDest->num_options; k++ )
+                            fprintf( stderr, "   \"%s\" = \"%s\"\n",
+                        pDest->options[k].name,
+                        pDest->options[k].value );
+                        #endif
+                        PrinterInfo& rInfo = m_aPrinters[ aPrinter ].m_aInfo;
+                        
+                        // remember the default context for later use
+                        PPDContext& rContext = m_aDefaultContexts[ aPrinter ];
+                        rContext.setParser( pNewParser );
+                        for( int i = 0; i < pPPD->num_groups; i++ )
+                            updatePrinterContextInfo( pPPD->groups + i, rContext );
+                        
+                        rInfo.m_pParser = pNewParser;
+                        rInfo.m_aContext = rContext;
+                        
+                        // clean up the mess
+                        m_pCUPSWrapper->ppdClose( pPPD );
+                    }
+                    #if OSL_DEBUG_LEVEL > 1
+                    else
+                        fprintf( stderr, "ppdOpenFile failed, fallinmg back to generic driver\n" );
+                    #endif
                     
-                    // clean up the mess
-                    m_pCUPSWrapper->ppdClose( pPPD );
+                    // remove temporary PPD file
+                    unlink( pPPDFile );
                 }
-#if OSL_DEBUG_LEVEL > 1
+                #if OSL_DEBUG_LEVEL > 1
                 else
-                    fprintf( stderr, "ppdOpenFile failed, fallinmg back to generic driver\n" );
-#endif
-
-                // remove temporary PPD file
-                unlink( pPPDFile );
+                    fprintf( stderr, "no dest found for printer %s\n", OUStringToOString( aPrinter, osl_getThreadTextEncoding() ).getStr() );
+                #endif
             }
-#if OSL_DEBUG_LEVEL > 1
-            else
-                fprintf( stderr, "no dest found for printer %s\n", OUStringToOString( aPrinter, osl_getThreadTextEncoding() ).getStr() );
-#endif
         }
         m_aCUPSMutex.release();
     }
+#if OSL_DEBUG_LEVEL >1
+    else
+        fprintf( stderr, "could not acquire CUPS mutex !!!\n" );
+#endif
 #endif // ENABLE_CUPS
 
     if( ! pNewParser )
@@ -702,7 +712,45 @@ FILE* CUPSManager::startSpool( const OUS
 #endif
 }
 
-int CUPSManager::endSpool( const OUString& rPrintername, const OUString& rJobTitle, FILE* pFile )
+struct less_ppd_key : public ::std::binary_function<double, double, bool>
+{
+    bool operator()(const PPDKey* left, const PPDKey* right) 
+    { return left->getOrderDependency() < right->getOrderDependency(); }
+};
+
+void CUPSManager::getOptionsFromDocumentSetup( const JobData& rJob, int& rNumOptions, void** rOptions ) const
+{
+    rNumOptions = 0;
+    *rOptions = NULL;
+    int i;
+
+    // emit features ordered to OrderDependency
+    // ignore features that are set to default
+
+    // sanity check
+    if( rJob.m_pParser == rJob.m_aContext.getParser() && rJob.m_pParser )
+    {
+        int nKeys = rJob.m_aContext.countValuesModified();
+        ::std::vector< const PPDKey* > aKeys( nKeys );
+        for(  i = 0; i < nKeys; i++ )
+            aKeys[i] = rJob.m_aContext.getModifiedKey( i );
+        ::std::sort( aKeys.begin(), aKeys.end(), less_ppd_key() );
+
+        for( i = 0; i < nKeys; i++ )
+        {
+            const PPDKey* pKey = aKeys[i];
+            const PPDValue* pValue = rJob.m_aContext.getValue( pKey );
+            if(pValue && pValue->m_eType == eInvocation && pValue->m_aValue.Len() )
+            {
+                OString aKey = OUStringToOString( pKey->getKey(), RTL_TEXTENCODING_ASCII_US );
+                OString aValue = OUStringToOString( pValue->m_aOption, RTL_TEXTENCODING_ASCII_US );
+                rNumOptions = m_pCUPSWrapper->cupsAddOption( aKey.getStr(), aValue.getStr(), rNumOptions, (cups_option_t**)rOptions );
+            }
+        }
+    }
+}
+
+int CUPSManager::endSpool( const OUString& rPrintername, const OUString& rJobTitle, FILE* pFile, const JobData& rDocumentJobData )
 {
     int nJobID = 0;
 
@@ -711,7 +759,7 @@ int CUPSManager::endSpool( const OUStrin
     std::hash_map< OUString, int, OUStringHash >::iterator dest_it =
         m_aCUPSDestMap.find( rPrintername );
     if( dest_it == m_aCUPSDestMap.end() )
-        return PrinterInfoManager::endSpool( rPrintername, rJobTitle, pFile );
+        return PrinterInfoManager::endSpool( rPrintername, rJobTitle, pFile, rDocumentJobData );
 
     #ifdef ENABLE_CUPS
     std::hash_map< FILE*, OString, FPtrHash >::const_iterator it = m_aSpoolFiles.find( pFile );
@@ -719,18 +767,28 @@ int CUPSManager::endSpool( const OUStrin
     {
         fclose( pFile );
         rtl_TextEncoding aEnc = osl_getThreadTextEncoding();
+        
+        // setup cups options
+        int nNumOptions = 0;
+        cups_option_t* pOptions = NULL;
+        getOptionsFromDocumentSetup( rDocumentJobData, nNumOptions, (void**)&pOptions );
 
         cups_dest_t* pDest = ((cups_dest_t*)m_pDests) + dest_it->second;
         nJobID = m_pCUPSWrapper->cupsPrintFile( pDest->name,
         it->second.getStr(),
         OUStringToOString( rJobTitle, aEnc ).getStr(),
-        0, NULL );
+        nNumOptions, pOptions );
 #if OSL_DEBUG_LEVEL > 1
-        fprintf( stderr, "cupsPrintFile( %s, %s, %s, 0, 0 ) returns %d\n",
-        pDest->name,
-        it->second.getStr(),
-        OUStringToOString( rJobTitle, aEnc ).getStr(),
-        nJobID );
+        fprintf( stderr, "cupsPrintFile( %s, %s, %s, %d, %p ) returns %d\n",
+                    pDest->name,
+                    it->second.getStr(),
+                    OUStringToOString( rJobTitle, aEnc ).getStr(),
+                    nNumOptions,
+                    pOptions,
+                    nJobID
+                    );
+        for( int n = 0; n < nNumOptions; n++ )
+            fprintf( stderr, "    option %s=%s\n", pOptions[n].name, pOptions[n].value );
         OString aCmd( "cp " );
         aCmd = aCmd + it->second;
         aCmd = aCmd + OString( " $HOME/cupsprint.ps" );
@@ -739,6 +797,8 @@ int CUPSManager::endSpool( const OUStrin
         
         unlink( it->second.getStr() );
         m_aSpoolFiles.erase( pFile );
+        if( pOptions )
+            m_pCUPSWrapper->cupsFreeOptions( nNumOptions, pOptions );
     }
 #endif // ENABLE_CUPS
 
Index: psprint/source/printer/printerinfomanager.cxx
===================================================================
RCS file: /cvs/gsl/psprint/source/printer/printerinfomanager.cxx,v
retrieving revision 1.27
retrieving revision 1.27.10.1
diff -u -p -u -p -r1.27 -r1.27.10.1
--- psprint/source/printer/printerinfomanager.cxx	9 Nov 2004 16:38:07 -0000	1.27
+++ psprint/source/printer/printerinfomanager.cxx	9 Feb 2005 16:19:06 -0000	1.27.10.1
@@ -1238,7 +1238,7 @@ FILE* PrinterInfoManager::startSpool( co
     return popen (aShellCommand.getStr(), "w");
 }
 
-int PrinterInfoManager::endSpool( const OUString& rPrintername, const OUString& rJobTitle, FILE* pFile )
+int PrinterInfoManager::endSpool( const OUString& rPrintername, const OUString& rJobTitle, FILE* pFile, const JobData& rDocumentJobData )
 {
     pclose( pFile );
     return 0; // job id ?
Index: psprint/source/printergfx/printerjob.cxx
===================================================================
RCS file: /cvs/gsl/psprint/source/printergfx/printerjob.cxx,v
retrieving revision 1.28
retrieving revision 1.28.40.1
diff -u -p -u -p -r1.28 -r1.28.40.1
--- psprint/source/printergfx/printerjob.cxx	18 May 2004 10:46:35 -0000	1.28
+++ psprint/source/printergfx/printerjob.cxx	9 Feb 2005 16:19:06 -0000	1.28.40.1
@@ -682,7 +682,7 @@ PrinterJob::EndJob ()
 #ifndef MACOSX
     {
         PrinterInfoManager& rPrinterInfoManager = PrinterInfoManager::get();
-        rPrinterInfoManager.endSpool( m_aLastJobData.m_aPrinterName, maJobTitle, pDestFILE );
+        rPrinterInfoManager.endSpool( m_aLastJobData.m_aPrinterName, maJobTitle, pDestFILE, m_aDocumentJobData );
     }
 #else
     {

Index: vcl/inc/button.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/button.hxx,v
retrieving revision 1.14
retrieving revision 1.14.10.2
diff -u -p -u -p -r1.14 -r1.14.10.2
--- vcl/inc/button.hxx	13 Jan 2005 17:27:12 -0000	1.14
+++ vcl/inc/button.hxx	14 Feb 2005 08:11:33 -0000	1.14.10.2
@@ -113,6 +113,10 @@ public:
                                               USHORT nTextStyle, Rectangle *pSymbolRect=NULL );
     SAL_DLLPRIVATE void             ImplSetFocusRect( const Rectangle &rFocusRect );
     SAL_DLLPRIVATE const Rectangle& ImplGetFocusRect() const;
+    SAL_DLLPRIVATE void             ImplSetSymbolAlign( SymbolAlign eAlign );
+    SAL_DLLPRIVATE SymbolAlign      ImplGetSymbolAlign() const;
+    SAL_DLLPRIVATE void             ImplSetSmallSymbol( BOOL bSmall = TRUE );
+
 #endif
 
 protected:
@@ -224,6 +228,8 @@ public:
 
     void            SetSymbol( SymbolType eSymbol );
     SymbolType      GetSymbol() const { return meSymbol; }
+    void            SetSymbolAlign( SymbolAlign eAlign );
+    SymbolAlign     GetSymbolAlign() const;
 
     void            SetDropDown( USHORT nStyle );
     USHORT          GetDropDown() const { return mnDDStyle; }

Index: vcl/inc/morebtn.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/morebtn.hxx,v
retrieving revision 1.2
retrieving revision 1.2.10.2
diff -u -p -u -p -r1.2 -r1.2.10.2
--- vcl/inc/morebtn.hxx	13 Jan 2005 17:41:31 -0000	1.2
+++ vcl/inc/morebtn.hxx	15 Feb 2005 14:08:03 -0000	1.2.10.2
@@ -77,7 +77,7 @@
 #include <button.hxx>
 #endif
 
-class ImplMoreWindowList;
+struct ImplMoreButtonData;
 
 // --------------
 // - MoreButton -
@@ -86,7 +86,7 @@ class ImplMoreWindowList;
 class VCL_DLLPUBLIC MoreButton : public PushButton
 {
 private:
-    ImplMoreWindowList* mpItemList;
+    ImplMoreButtonData* mpMBData;
     ULONG               mnDelta;
     MapUnit             meUnit;
     BOOL                mbState;
@@ -94,7 +94,8 @@ private:
     // Copy assignment is forbidden and not implemented.
 	SAL_DLLPRIVATE      MoreButton( const MoreButton & );
 	SAL_DLLPRIVATE      MoreButton& operator=( const MoreButton & );
-
+    SAL_DLLPRIVATE void ShowState();
+    
 protected:
     SAL_DLLPRIVATE void ImplInit( Window* pParent, WinBits nStyle );
 	SAL_DLLPRIVATE void	ImplLoadRes( const ResId& rResId );
@@ -119,7 +120,12 @@ public:
     BOOL                GetState() const { return mbState; }
 
     void                SetText( const XubString& rNewText );
-    XubString            GetText() const;
+    XubString           GetText() const;
+
+    void                SetMoreText( const XubString& rNewText );
+    void                SetLessText( const XubString& rNewText );
+    XubString           GetMoreText() const;
+    XubString           GetLessText() const;
 };
 
 inline void MoreButton::SetState( BOOL bNewState )
Index: vcl/inc/outdev.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/outdev.hxx,v
retrieving revision 1.55
retrieving revision 1.55.10.1
diff -u -p -u -p -r1.55 -r1.55.10.1
--- vcl/inc/outdev.hxx	13 Jan 2005 17:42:24 -0000	1.55
+++ vcl/inc/outdev.hxx	28 Jan 2005 15:37:39 -0000	1.55.10.1
@@ -396,7 +396,7 @@ private:
     ImplThresholdRes    maThresRes;
     OutDevType          meOutDevType;
     OutDevViewType      meOutDevViewType;
-    Region              maRegion;
+    Region              maRegion;           // contains the clip region, see SetClipRegion(...)
     Color               maLineColor;
     Color               maFillColor;
     Font                maFont;
Index: vcl/inc/svids.hrc
===================================================================
RCS file: /cvs/gsl/vcl/inc/svids.hrc,v
retrieving revision 1.10
retrieving revision 1.10.180.1
diff -u -p -u -p -r1.10 -r1.10.180.1
--- vcl/inc/svids.hrc	5 Jul 2004 15:40:41 -0000	1.10
+++ vcl/inc/svids.hrc	14 Feb 2005 08:11:34 -0000	1.10.180.1
@@ -122,6 +122,7 @@
 #define SV_BUTTONTEXT_MORE					10107
 #define SV_BUTTONTEXT_IGNORE				10108
 #define SV_BUTTONTEXT_ABORT				10109
+#define SV_BUTTONTEXT_LESS					10110
 
 #define SV_STDTEXT_FIRST					SV_STDTEXT_SERVICENOTAVAILABLE
 #define SV_STDTEXT_SERVICENOTAVAILABLE		10200
Index: vcl/inc/syswin.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/syswin.hxx,v
retrieving revision 1.16
retrieving revision 1.16.2.1
diff -u -p -u -p -r1.16 -r1.16.2.1
--- vcl/inc/syswin.hxx	21 Jan 2005 17:43:18 -0000	1.16
+++ vcl/inc/syswin.hxx	11 Feb 2005 07:20:11 -0000	1.16.2.1
@@ -230,6 +230,11 @@ private:
     USHORT          mnIcon;
 	ImplData*       mpImplData;
 
+#if _SOLAR__PRIVATE
+public:
+    SAL_DLLPRIVATE BOOL ImplIsInTaskPaneList( Window* pWin );
+#endif
+
 private:
 	// Default construction is forbidden and not implemented.
 	SystemWindow();
Index: vcl/inc/taskpanelist.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/taskpanelist.hxx,v
retrieving revision 1.5
retrieving revision 1.5.10.1
diff -u -p -u -p -r1.5 -r1.5.10.1
--- vcl/inc/taskpanelist.hxx	13 Jan 2005 17:53:18 -0000	1.5
+++ vcl/inc/taskpanelist.hxx	11 Feb 2005 07:20:11 -0000	1.5.10.1
@@ -79,6 +79,11 @@ class VCL_DLLPUBLIC TaskPaneList
 	Window *FindNextFloat( Window *pWindow, BOOL bForward = TRUE );
 	Window *FindNextSplitter( Window *pWindow, BOOL bForward = TRUE );
 
+#if _SOLAR__PRIVATE
+public:
+    BOOL IsInList( Window *pWindow );
+#endif
+
 public:
 	TaskPaneList();
 	~TaskPaneList();
Index: vcl/inc/window.h
===================================================================
RCS file: /cvs/gsl/vcl/inc/window.h,v
retrieving revision 1.18
retrieving revision 1.18.10.1
diff -u -p -u -p -r1.18 -r1.18.10.1
--- vcl/inc/window.h	13 Jan 2005 17:56:32 -0000	1.18
+++ vcl/inc/window.h	28 Jan 2005 15:37:39 -0000	1.18.10.1
@@ -313,11 +313,11 @@ public:
     ::com::sun::star::uno::Reference< ::com::sun::star::accessibility::XAccessible > mxAccessible;
     ImplAccessibleInfos* mpAccessibleInfos;
     VCLXWindow*         mpVCLXWindow;
-    Region              maWinRegion;
-    Region              maWinClipRegion;
-    Region              maInvalidateRegion;
-    Region*             mpChildClipRegion;
-    Region*             mpPaintRegion;
+    Region              maWinRegion;        // region to 'shape' the VCL window (frame coordinates)
+    Region              maWinClipRegion;    // the (clipping) region that finally corresponds to the VCL window (frame coordinates)
+    Region              maInvalidateRegion; // region that has to be redrawn (frame coordinates)
+    Region*             mpChildClipRegion;  // child clip region if CLIPCHILDREN is set (frame coordinates)
+    Region*             mpPaintRegion;      // only set during Paint() method call (window coordinates)
     WinBits             mnStyle;
     WinBits             mnPrevStyle;
     WinBits             mnExtendedStyle;
Index: vcl/inc/wintypes.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/wintypes.hxx,v
retrieving revision 1.23
retrieving revision 1.23.2.1
diff -u -p -u -p -r1.23 -r1.23.2.1
--- vcl/inc/wintypes.hxx	31 Jan 2005 09:16:44 -0000	1.23
+++ vcl/inc/wintypes.hxx	14 Feb 2005 08:11:34 -0000	1.23.2.1
@@ -313,6 +313,7 @@ enum ImageAlign { IMAGEALIGN_LEFT, IMAGE
                   IMAGEALIGN_LEFT_TOP, IMAGEALIGN_LEFT_BOTTOM, IMAGEALIGN_TOP_LEFT,
                   IMAGEALIGN_TOP_RIGHT, IMAGEALIGN_RIGHT_TOP, IMAGEALIGN_RIGHT_BOTTOM,
                   IMAGEALIGN_BOTTOM_LEFT, IMAGEALIGN_BOTTOM_RIGHT, IMAGEALIGN_CENTER };
+enum SymbolAlign { SYMBOLALIGN_LEFT, SYMBOLALIGN_RIGHT };
 
 // ------------
 // - TriState -
@@ -337,7 +338,8 @@ typedef USHORT StandardButtonType;
 #define BUTTON_MORE 		((StandardButtonType)7)
 #define BUTTON_IGNORE 		((StandardButtonType)8)
 #define BUTTON_ABORT 		((StandardButtonType)9)
-#define BUTTON_COUNT		10
+#define BUTTON_LESS 		((StandardButtonType)10)
+#define BUTTON_COUNT		11
 
 #endif // _SV_WINTYPES_HXX
 
Index: vcl/source/app/svapp.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/app/svapp.cxx,v
retrieving revision 1.55
retrieving revision 1.55.10.1
diff -u -p -u -p -r1.55 -r1.55.10.1
--- vcl/source/app/svapp.cxx	13 Jan 2005 18:38:27 -0000	1.55
+++ vcl/source/app/svapp.cxx	15 Feb 2005 08:10:35 -0000	1.55.10.1
@@ -1881,6 +1881,8 @@ BOOL InitAccessBridge( BOOL bShowCancel,
 {
     BOOL bRet = ImplInitAccessBridge( bShowCancel, rCancelled );
 
+// There is no GUI to re-enable accessibility on Unix ..
+#ifndef UNX
     if( !bRet && bShowCancel && !rCancelled )
     {
         // disable accessibility if the user chooses to continue
@@ -1890,6 +1892,7 @@ BOOL InitAccessBridge( BOOL bShowCancel,
         aSettings.SetMiscSettings( aMisc );
         Application::SetSettings( aSettings );
     }
+#endif
 
     return bRet;
 }
Index: vcl/source/app/svdata.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/app/svdata.cxx,v
retrieving revision 1.36
retrieving revision 1.36.10.1
diff -u -p -u -p -r1.36 -r1.36.10.1
--- vcl/source/app/svdata.cxx	13 Jan 2005 17:58:11 -0000	1.36
+++ vcl/source/app/svdata.cxx	15 Feb 2005 10:33:39 -0000	1.36.10.1
@@ -341,8 +341,7 @@ bool ImplInitAccessBridge(BOOL bAllowCan
 {
     rCancelled = FALSE;
 
-    // no error messages during installation (i.e., when there is no configuration)
-    bool bErrorMessage = (FALSE != vcl::SettingsConfigItem::get()->IsValidConfigMgr());
+    bool bErrorMessage = true;
 
     // Note:
     // if bAllowCancel is TRUE we were called from application startup
Index: vcl/source/control/button.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/button.cxx,v
retrieving revision 1.35
retrieving revision 1.35.2.4
diff -u -p -u -p -r1.35 -r1.35.2.4
--- vcl/source/control/button.cxx	31 Jan 2005 09:17:00 -0000	1.35
+++ vcl/source/control/button.cxx	16 Feb 2005 14:23:01 -0000	1.35.2.4
@@ -138,12 +138,14 @@ class ImplCommonButtonData
 public:
     Rectangle       maFocusRect;
     USHORT          mnButtonState;
+    BOOL            mbSmallSymbol;
 
     Image           maImage;
     Image           maImageHC;
     BitmapEx*       mpBitmapEx;
     BitmapEx*       mpBitmapExHC;
     ImageAlign      meImageAlign;
+    SymbolAlign     meSymbolAlign;
 
 public:
                     ImplCommonButtonData();
@@ -154,10 +156,12 @@ public:
 ImplCommonButtonData::ImplCommonButtonData()
 {
     mnButtonState   = 0;
+    mbSmallSymbol = FALSE;
 
     mpBitmapEx = NULL;
     mpBitmapExHC = NULL;
     meImageAlign = IMAGEALIGN_TOP;
+    meSymbolAlign = SYMBOLALIGN_LEFT;
 }
 
 // -----------------------------------------------------------------------
@@ -228,7 +232,8 @@ XubString Button::GetStandardText( Stand
         SV_BUTTONTEXT_CLOSE,
         SV_BUTTONTEXT_MORE,
         SV_BUTTONTEXT_IGNORE,
-        SV_BUTTONTEXT_ABORT
+        SV_BUTTONTEXT_ABORT,
+        SV_BUTTONTEXT_LESS
     };
 
     ResId aResId( aResIdAry[(USHORT)eButton], ImplGetResMgr() );
@@ -493,10 +498,13 @@ void Button::ImplDrawAlignedImage( Outpu
             if ( bDrawText )
             {
                 nSymbolHeight = pDev->GetTextHeight();
+                if ( mpButtonData->mbSmallSymbol )
+                    nSymbolHeight = nSymbolHeight * 3 / 4;
+
                 aSymbol = Rectangle( Point(), Size( nSymbolHeight, nSymbolHeight ) );
                 ImplCalcSymbolRect( aSymbol );
-                aRect.Left() += 2 * nSymbolHeight;
-                aTSSize.Width() = 2 * nSymbolHeight;
+                aRect.Left() += 3 * nSymbolHeight / 2;
+                aTSSize.Width() = 3 * nSymbolHeight / 2;
             }
             else
             {
@@ -642,8 +650,21 @@ void Button::ImplDrawAlignedImage( Outpu
 
     if ( bHasSymbol )
     {
-        *pSymbolRect = Rectangle( aTextPos, aSymbolSize );
-        aTextPos.X() += ( 2 * nSymbolHeight );
+        if ( mpButtonData->meSymbolAlign == SYMBOLALIGN_RIGHT )
+        {
+            Point aRightPos = Point( aTextPos.X() + aTextSize.Width() + aSymbolSize.Width()/2, aTextPos.Y() ); 
+            *pSymbolRect = Rectangle( aRightPos, aSymbolSize );
+        }
+        else
+        {
+            *pSymbolRect = Rectangle( aTextPos, aSymbolSize );
+            aTextPos.X() += ( 3 * nSymbolHeight / 2 );
+        }
+        if ( mpButtonData->mbSmallSymbol )
+        {
+            nYOffset = (aUnion.GetHeight() - aSymbolSize.Height())/2;
+            pSymbolRect->setY( aTextPos.Y() + nYOffset );
+        }
     }
 
     USHORT nStyle = 0;
@@ -718,6 +739,27 @@ USHORT Button::ImplGetButtonState() cons
 }
 
 // -----------------------------------------------------------------------
+void Button::ImplSetSymbolAlign( SymbolAlign eAlign )
+{
+    if ( mpButtonData->meSymbolAlign != eAlign )
+    {
+        mpButtonData->meSymbolAlign = eAlign;
+        StateChanged( STATE_CHANGE_DATA );
+    }
+}
+
+// -----------------------------------------------------------------------
+SymbolAlign Button::ImplGetSymbolAlign() const
+{
+    return mpButtonData->meSymbolAlign;
+}
+// -----------------------------------------------------------------------
+void Button::ImplSetSmallSymbol( BOOL bSmall )
+{
+    mpButtonData->mbSmallSymbol = bSmall;
+}
+
+// -----------------------------------------------------------------------
 void Button::EnableImageDisplay( BOOL bEnable )
 {
     if( bEnable )
@@ -1853,6 +1895,18 @@ void PushButton::SetSymbol( SymbolType e
 }
 
 // -----------------------------------------------------------------------
+void PushButton::SetSymbolAlign( SymbolAlign eAlign )
+{
+    ImplSetSymbolAlign( eAlign );
+}
+
+// -----------------------------------------------------------------------
+SymbolAlign PushButton::GetSymbolAlign() const
+{
+    return ImplGetSymbolAlign();
+}
+
+// -----------------------------------------------------------------------
 
 void PushButton::SetDropDown( USHORT nStyle )
 {

Index: vcl/source/control/morebtn.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/morebtn.cxx,v
retrieving revision 1.4
retrieving revision 1.4.32.3
diff -u -p -u -p -r1.4 -r1.4.32.3
--- vcl/source/control/morebtn.cxx	3 Jan 2005 17:40:26 -0000	1.4
+++ vcl/source/control/morebtn.cxx	16 Feb 2005 08:09:17 -0000	1.4.32.3
@@ -71,27 +71,58 @@
 
 // =======================================================================
 
-// Muss mit der Laenge der folgenden Texte uebereinstimmen
-#define EXTRA_TEXTLEN		3
-
-static sal_Char const aImplMoreOpen[]  = " >>";
-static sal_Char const aImplMoreClose[] = " <<";
-
 DECLARE_LIST( ImplMoreWindowList, Window* );
 
+struct ImplMoreButtonData
+{
+    ImplMoreWindowList *mpItemList;
+    XubString           maMoreText;
+    XubString           maLessText;
+};
+
 // =======================================================================
 
 void MoreButton::ImplInit( Window* pParent, WinBits nStyle )
 {
-	mpItemList	 = NULL;
+    mpMBData     = new ImplMoreButtonData;
 	mnDelta 	 = 0;
 	meUnit		 = MAP_PIXEL;
 	mbState 	 = FALSE;
 
+	mpMBData->mpItemList = NULL;
+
 	PushButton::ImplInit( pParent, nStyle );
 
-	SetText( Button::GetStandardText( BUTTON_MORE ) );
+	mpMBData->maMoreText = Button::GetStandardText( BUTTON_MORE );
+    mpMBData->maLessText = Button::GetStandardText( BUTTON_LESS );
+
 	SetHelpText( Button::GetStandardHelpText( BUTTON_MORE ) );
+
+    ShowState();
+
+    SetSymbolAlign( SYMBOLALIGN_RIGHT );
+    ImplSetSmallSymbol( TRUE );
+
+    if ( ! ( nStyle & ( WB_RIGHT | WB_LEFT ) ) )
+    {
+        nStyle |= WB_CENTER;
+        SetStyle( nStyle );
+    }
+}
+
+// -----------------------------------------------------------------------
+void MoreButton::ShowState()
+{
+    if ( mbState )
+    {
+        SetSymbol( SYMBOL_PAGEUP );
+        SetText( mpMBData->maLessText );
+    }
+    else
+    {
+        SetSymbol( SYMBOL_PAGEDOWN );
+        SetText( mpMBData->maMoreText );
+    }
 }
 
 // -----------------------------------------------------------------------
@@ -128,7 +159,8 @@ void MoreButton::ImplLoadRes( const ResI
 	{
 		// Nicht Methode rufen, da Dialog nicht umgeschaltet werden soll
 		mbState = (BOOL)ReadShortRes();
-		SetText( GetText() );
+		// SetText( GetText() );
+        ShowState();
 	}
 	if ( nObjMask & RSC_MOREBUTTON_MAPUNIT )
 		meUnit = (MapUnit)ReadLongRes();
@@ -141,8 +173,9 @@ void MoreButton::ImplLoadRes( const ResI
 
 MoreButton::~MoreButton()
 {
-	if ( mpItemList )
-		delete mpItemList;
+	if ( mpMBData->mpItemList )
+		delete mpMBData->mpItemList;
+    delete mpMBData;
 }
 
 // -----------------------------------------------------------------------
@@ -151,13 +184,12 @@ void MoreButton::Click()
 {
 	Window* 	pParent = GetParent();
 	Size		aSize( pParent->GetSizePixel() );
-	Window* 	pWindow = (mpItemList) ? mpItemList->First() : NULL;
+	Window* 	pWindow = (mpMBData->mpItemList) ? mpMBData->mpItemList->First() : NULL;
 	long		nDeltaPixel = LogicToPixel( Size( 0, mnDelta ), meUnit ).Height();
 
 	// Status aendern
-	XubString aText = GetText();
 	mbState = !mbState;
-	SetText( aText );
+    ShowState();
 
 	// Hier den Click-Handler rufen, damit vorher die Controls initialisiert
 	// werden koennen
@@ -170,7 +202,7 @@ void MoreButton::Click()
 		while ( pWindow )
 		{
 			pWindow->Show();
-			pWindow = mpItemList->Next();
+			pWindow = mpMBData->mpItemList->Next();
 		}
 
 		// Dialogbox anpassen
@@ -200,7 +232,7 @@ void MoreButton::Click()
 		while ( pWindow )
 		{
 			pWindow->Hide();
-			pWindow = mpItemList->Next();
+			pWindow = mpMBData->mpItemList->Next();
 		}
 	}
 }
@@ -209,10 +241,10 @@ void MoreButton::Click()
 
 void MoreButton::AddWindow( Window* pWindow )
 {
-	if ( !mpItemList )
-		mpItemList = new ImplMoreWindowList( 1024, 16, 16 );
+	if ( !mpMBData->mpItemList )
+		mpMBData->mpItemList = new ImplMoreWindowList( 1024, 16, 16 );
 
-	mpItemList->Insert( pWindow, LIST_APPEND );
+	mpMBData->mpItemList->Insert( pWindow, LIST_APPEND );
 
 	if ( mbState )
 		pWindow->Show();
@@ -224,41 +256,59 @@ void MoreButton::AddWindow( Window* pWin
 
 void MoreButton::RemoveWindow( Window* pWindow )
 {
-	if ( mpItemList )
-		mpItemList->Remove( pWindow );
+	if ( mpMBData->mpItemList )
+		mpMBData->mpItemList->Remove( pWindow );
 }
 
 // -----------------------------------------------------------------------
 
 void MoreButton::SetText( const XubString& rText )
 {
-	XubString aText = rText;
+	PushButton::SetText( rText );
+}
 
-	if ( !mbState )
-		aText.AppendAscii( aImplMoreOpen );
-	else
-		aText.AppendAscii( aImplMoreClose );
+// -----------------------------------------------------------------------
 
-	PushButton::SetText( aText );
+XubString MoreButton::GetText() const
+{
+	return PushButton::GetText();
 }
 
 // -----------------------------------------------------------------------
+void MoreButton::SetMoreText( const XubString& rText )
+{
+	if ( mpMBData )
+        mpMBData->maMoreText = rText;
 
-XubString MoreButton::GetText() const
+    if ( !mbState )
+        SetText( rText );
+}
+
+// -----------------------------------------------------------------------
+XubString MoreButton::GetMoreText() const
 {
-	XubString aText = PushButton::GetText();
+	if ( mpMBData )
+        return mpMBData->maMoreText;
+    else
+        return PushButton::GetText();
+}
 
-	XubString aSubText = aText.Copy( aText.Len()-EXTRA_TEXTLEN, EXTRA_TEXTLEN );
-	if ( !mbState )
-	{
-		if ( aSubText.EqualsAscii( aImplMoreOpen ) )
-			aText.Erase( aText.Len()-EXTRA_TEXTLEN, EXTRA_TEXTLEN );
-	}
-	else
-	{
-		if ( aSubText.EqualsAscii( aImplMoreClose ) )
-			aText.Erase( aText.Len()-EXTRA_TEXTLEN, EXTRA_TEXTLEN );
-	}
+// -----------------------------------------------------------------------
+void MoreButton::SetLessText( const XubString& rText )
+{
+	if ( mpMBData )
+        mpMBData->maLessText = rText;
+
+    if ( mbState )
+        SetText( rText );
+}
 
-	return aText;
+// -----------------------------------------------------------------------
+XubString MoreButton::GetLessText() const
+{
+	if ( mpMBData )
+        return mpMBData->maLessText;
+    else
+        return PushButton::GetText();
 }
+
Index: vcl/source/gdi/fontcfg.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/fontcfg.cxx,v
retrieving revision 1.33
retrieving revision 1.33.2.1
diff -u -p -u -p -r1.33 -r1.33.2.1
--- vcl/source/gdi/fontcfg.cxx	31 Jan 2005 09:17:41 -0000	1.33
+++ vcl/source/gdi/fontcfg.cxx	15 Feb 2005 11:42:17 -0000	1.33.2.1
@@ -581,6 +581,9 @@ static const char* const aImplKillLeadin
     "cg",
     "hg",
     "fz",
+    "ipa",
+    "sazanami",
+    "kochi",
     NULL
 };
 
Index: vcl/source/gdi/outdev3.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/outdev3.cxx,v
retrieving revision 1.192
retrieving revision 1.189.8.5
diff -u -p -u -p -r1.192 -r1.189.8.5
--- vcl/source/gdi/outdev3.cxx	31 Jan 2005 13:23:19 -0000	1.192
+++ vcl/source/gdi/outdev3.cxx	14 Feb 2005 16:49:58 -0000	1.189.8.5
@@ -451,6 +451,11 @@ static sal_Unicode const aHGPMinchoLSun[
 static sal_Unicode const aHGGothicBSun[] = { 'h', 'g', 0x30B4, 0x30B7, 0x30C3, 0x30AF, 'b', 's', 'u', 'n', 0 };
 static sal_Unicode const aHGPGothicBSun[] = { 'h', 'g', 'p', 0x30B4, 0x30B7, 0x30C3, 0x30AF, 'b', 's', 'u', 'n', 0 };
 static sal_Unicode const aHGHeiseiMin[] = { 'h', 'g', 0x5E73, 0x6210, 0x660E, 0x671D, 0x4F53, 0, 'h', 'g', 0x5E73, 0x6210, 0x660E, 0x671D, 0x4F53, 'w', '3', 'x', '1', '2', 0, 0 };
+static sal_Unicode const aIPAMincho[] =  { 'i', 'p', 'a', 0x660E, 0x671D, 0 };
+static sal_Unicode const aIPAPMincho[] = { 'i', 'p', 'a', 'p', 0x660E, 0x671D, 0 };
+static sal_Unicode const aIPAGothic[] =  { 'i', 'p', 'a',  0x30B4, 0x30B7, 0x30C3, 0x30AF, 0 };
+static sal_Unicode const aIPAPGothic[] =  { 'i', 'p', 'a', 'p', 0x30B4, 0x30B7, 0x30C3, 0x30AF, 0 };
+static sal_Unicode const aIPAUIGothic[] =  { 'i', 'p', 'a', 'u', 'i', 0x30B4, 0x30B7, 0x30C3, 0x30AF, 0 };
 static sal_Unicode const aSazanamiMincho[] = { 0x3055, 0x3056, 0x306A, 0x307F, 0x660E, 0x671D, 0, 0 };
 static sal_Unicode const aSazanamiGothic[] = { 0x3055, 0x3056, 0x306A, 0x307F, 0x30B4, 0x30B7, 0x30C3, 0x30AF, 0, 0 };
 static sal_Unicode const aKochiMincho[] = { 0x6771, 0x98A8, 0x660E, 0x671D, 0, 0 };
@@ -583,6 +588,11 @@ static ImplLocalizedFontName aImplLocali
 {   "hgpmincholsun",        aHGPMinchoLSun },
 {   "hgpgothicbsun",        aHGPGothicBSun },
 {   "hgheiseimin",          aHGHeiseiMin },
+{   "ipamincho",            aIPAMincho },
+{   "ipapmincho",           aIPAPMincho },
+{   "ipagothic",            aIPAGothic },
+{   "ipapgothic",           aIPAPGothic },
+{   "ipauigothic",          aIPAUIGothic },
 {   "sazanamimincho",       aSazanamiMincho },
 {   "sazanamigothic",       aSazanamiGothic },
 {   "kochimincho",          aKochiMincho },
@@ -3031,7 +3041,7 @@ ImplFontEntry* ImplFontCache::GetFallbac
             "arialunicodems", "cyberbit", "code2000", "",
             "andalesansui", "",
             "starsymbol", "opensymbol", "",
-            "msmincho", "fzmingti", "fzheiti", "sazanamimincho", "kochimincho", "",
+            "msmincho", "fzmingti", "fzheiti", "ipamincho", "sazanamimincho", "kochimincho", "",
             "sunbatang", "sundotum", "baekmukdotum", "gulim", "batang", "dotum", "",
             "hgmincholightj", "msunglightsc", "msunglighttc", "hymyeongjolightk", "",
             "tahoma", "timesnewroman", "lucidatypewriter", "lucidasans", "nimbussansl", "",
@@ -5197,17 +5207,16 @@ void OutputDevice::SetFont( const Font& 
 
     if ( mpMetaFile )
     {
-        const Color& rTextFillColor = aFont.GetFillColor();
-
         mpMetaFile->AddAction( new MetaFontAction( aFont ) );
+        // the color and alignment actions don't belong here
+        // TODO: get rid of them without breaking anything...
         mpMetaFile->AddAction( new MetaTextAlignAction( aFont.GetAlign() ) );
-        mpMetaFile->AddAction( new MetaTextColorAction( aFont.GetColor() ) );
         mpMetaFile->AddAction( new MetaTextFillColorAction( aFont.GetFillColor(), !aFont.IsTransparent() ) );
     }
 
 #if (OSL_DEBUG_LEVEL > 2) || defined (HDU_DEBUG)
     fprintf( stderr, "   OutputDevice::SetFont( name=\"%s\", h=%d)\n",
-	     OUStringToOString( aFont.GetName(), RTL_TEXTENCODING_UTF8 ).getStr(),
+         OUStringToOString( aFont.GetName(), RTL_TEXTENCODING_UTF8 ).getStr(),
          aFont.GetSize().Height() );
 #endif
 
@@ -5216,12 +5225,14 @@ void OutputDevice::SetFont( const Font& 
         // Optimization MT/HDU: COL_TRANSPARENT means SetFont should ignore the font color,
         // because SetTextColor() is used for this.
         // #i28759# maTextColor might have been changed behind our back, commit then, too.
-        if ( aFont.GetColor() != COL_TRANSPARENT && 
-             ( maFont.GetColor() != aFont.GetColor() || aFont.GetColor() != maTextColor ) )
-		{
-			maTextColor = aFont.GetColor();
+        if( aFont.GetColor() != COL_TRANSPARENT
+        && (aFont.GetColor() != maFont.GetColor() || aFont.GetColor() != maTextColor ) )
+        {
+            maTextColor = aFont.GetColor();
             mbInitTextColor = TRUE;
-		}
+            if( mpMetaFile )
+                mpMetaFile->AddAction( new MetaTextColorAction( aFont.GetColor() ) );
+        }
         maFont      = aFont;
         mbNewFont   = TRUE;
 
@@ -5266,9 +5277,6 @@ void OutputDevice::SetDigitLanguage( Lan
     if( mpMetaFile )
         mpMetaFile->AddAction( new MetaTextLanguageAction( eTextLanguage ) );
 
-    if( eTextLanguage == LANGUAGE_SYSTEM )
-        eTextLanguage = GetSystemLanguage();
-
     meTextLanguage = eTextLanguage;
 
     if( mpAlphaVDev )
Index: vcl/source/src/btntext.src
===================================================================
RCS file: /cvs/gsl/vcl/source/src/btntext.src,v
retrieving revision 1.35
retrieving revision 1.35.184.3
diff -u -p -u -p -r1.35 -r1.35.184.3
--- vcl/source/src/btntext.src	25 Jun 2004 15:16:33 -0000	1.35
+++ vcl/source/src/btntext.src	15 Feb 2005 14:12:46 -0000	1.35.184.3
@@ -114,6 +114,13 @@ String SV_BUTTONTEXT_MORE
 	Text [ x-comment ] = " ";
 };
 
+String SV_BUTTONTEXT_LESS
+{
+	Text [ de ] = "~Zusätze";
+	Text [ en-US ] = "~More";
+	Text [ x-comment ] = " ";
+};
+
 String SV_BUTTONTEXT_IGNORE
 {
     Text [ de ] = "~Ignorieren";
Index: vcl/source/window/menu.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/menu.cxx,v
retrieving revision 1.120
retrieving revision 1.120.2.1
diff -u -p -u -p -r1.120 -r1.120.2.1
--- vcl/source/window/menu.cxx	31 Jan 2005 13:25:29 -0000	1.120
+++ vcl/source/window/menu.cxx	15 Feb 2005 16:34:44 -0000	1.120.2.1
@@ -5082,8 +5082,7 @@ BOOL MenuBarWindow::ImplHandleKeyEvent( 
                 // Wegen Systemmenu und anderen System-HotKeys, nur
                 // eigenstaendige Character-Kombinationen auswerten
                 USHORT nKeyCode = rKEvent.GetKeyCode().GetCode();
-                if ( !nKeyCode ||
-                     ((nKeyCode >= KEY_A) && (nKeyCode <= KEY_Z)) )
+                if ( ((nKeyCode >= KEY_A) && (nKeyCode <= KEY_Z)) )
                     Sound::Beep();
             }
         }
Index: vcl/source/window/syswin.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/syswin.cxx,v
retrieving revision 1.40
retrieving revision 1.40.10.1
diff -u -p -u -p -r1.40 -r1.40.10.1
--- vcl/source/window/syswin.cxx	13 Jan 2005 18:05:34 -0000	1.40
+++ vcl/source/window/syswin.cxx	11 Feb 2005 07:23:15 -0000	1.40.10.1
@@ -150,12 +150,12 @@ SystemWindow::SystemWindow( WindowType n
     mbSysChild          = FALSE;
     mnMenuBarMode       = MENUBAR_MODE_NORMAL;
     mnIcon              = 0;
-    mpImplData->mpTaskPaneList      = NULL;
 }
 
 SystemWindow::~SystemWindow()
 {
     delete mpImplData;
+    mpImplData = NULL;
 }
 
 // -----------------------------------------------------------------------
@@ -902,3 +902,11 @@ void SystemWindow::SetMenuBarMode( USHOR
         }
     }
 }
+
+// -----------------------------------------------------------------------
+BOOL SystemWindow::ImplIsInTaskPaneList( Window* pWin )
+{
+    if( mpImplData && mpImplData->mpTaskPaneList )
+        return mpImplData->mpTaskPaneList->IsInList( pWin );
+    return FALSE;
+}
Index: vcl/source/window/taskpanelist.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/taskpanelist.cxx,v
retrieving revision 1.16
retrieving revision 1.16.92.1
diff -u -p -u -p -r1.16 -r1.16.92.1
--- vcl/source/window/taskpanelist.cxx	22 Oct 2004 12:14:28 -0000	1.16
+++ vcl/source/window/taskpanelist.cxx	11 Feb 2005 07:23:15 -0000	1.16.92.1
@@ -211,6 +211,18 @@ void TaskPaneList::RemoveWindow( Window 
 
 // --------------------------------------------------
 
+BOOL TaskPaneList::IsInList( Window *pWindow )
+{
+	::std::vector< Window* >::iterator p;
+    p = ::std::find( mTaskPanes.begin(), mTaskPanes.end(), pWindow );
+    if( p != mTaskPanes.end() )
+	    return TRUE;
+    else
+        return FALSE;
+}
+
+// --------------------------------------------------
+
 BOOL TaskPaneList::HandleKeyEvent( KeyEvent aKeyEvent )
 {
 
Index: vcl/source/window/window.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/window.cxx,v
retrieving revision 1.208
retrieving revision 1.207.2.6
diff -u -p -u -p -r1.208 -r1.207.2.6
--- vcl/source/window/window.cxx	31 Jan 2005 13:26:15 -0000	1.208
+++ vcl/source/window/window.cxx	14 Feb 2005 14:45:54 -0000	1.207.2.6
@@ -159,6 +159,7 @@
 #ifndef _SV_BUTTON_HXX
 #include <button.hxx> // Button::GetStandardText
 #endif
+
 #include <com/sun/star/awt/XWindowPeer.hpp>
 
 #ifndef _DRAFTS_COM_SUN_STAR_RENDERING_XCANVAS_HPP_
@@ -358,6 +359,7 @@ bool Window::ImplCheckUIFont( const Font
     aTestText.Append( Button::GetStandardText( BUTTON_HELP ) );
     aTestText.Append( Button::GetStandardText( BUTTON_CLOSE ) );
     aTestText.Append( Button::GetStandardText( BUTTON_MORE ) );
+    aTestText.Append( Button::GetStandardText( BUTTON_LESS ) );
     aTestText.Append( Button::GetStandardText( BUTTON_ABORT ) );
 
     return HasGlyphs( rFont, aTestText ) >= aTestText.Len();
@@ -2571,19 +2573,25 @@ void Window::ImplInvalidateFrameRegion( 
     if( ((IsPaintTransparent() && !(nFlags & INVALIDATE_NOTRANSPARENT)) || (nFlags & INVALIDATE_TRANSPARENT) ) 
             && ImplGetParent() )
     {
-        /* The following optimization shows problems when resizing (native) tabcontrols (eg the Help viewer)
-           So we pass a NULL region to the parent as well which will result in a IMPL_PAINT_PAINTALL on the parent...
+        Window *pParent = ImplGetParent();
+        while( pParent && pParent->IsPaintTransparent() )
+            pParent = pParent->ImplGetParent();
+        if( pParent )
+        {
+            Region *pChildRegion;
+            if ( mpWindowImpl->mnPaintFlags & IMPL_PAINT_PAINTALL )
+                // invalidate the whole child window region in the parent
+                pChildRegion = ImplGetWinChildClipRegion();
+            else
+                // invalidate the same region in the parent that has to be repainted in the child
+                pChildRegion = &mpWindowImpl->maInvalidateRegion;
 
-        Region aWindowRegion;
-        if( pRegion == NULL )
-            // compute real region to avoid full repaint in parent
-            aWindowRegion = Rectangle( Point( mnOutOffX, mnOutOffY ), Size( mnOutWidth, mnOutHeight ) );
-        ImplGetParent()->ImplInvalidateFrameRegion( pRegion==NULL ? &aWindowRegion : pRegion, nFlags );
-        */
-        ImplGetParent()->ImplInvalidateFrameRegion( pRegion, nFlags );
+            nFlags |= INVALIDATE_CHILDREN;  // paint should also be done on all children
+            nFlags &= ~INVALIDATE_NOERASE;  // parent should paint and erase to create proper background
+            pParent->ImplInvalidateFrameRegion( pChildRegion, nFlags );
+        }
     }
-    else
-        ImplPostPaint();
+    ImplPostPaint();
 }
 
 // -----------------------------------------------------------------------
@@ -2633,15 +2641,15 @@ void Window::ImplInvalidate( const Regio
     BOOL bInvalidateAll = !pRegion;
 
     // Transparent-Invalidate beruecksichtigen
-    Window* pWindow = this;
+    Window* pOpaqueWindow = this;
     if ( (mpWindowImpl->mbPaintTransparent && !(nFlags & INVALIDATE_NOTRANSPARENT)) || (nFlags & INVALIDATE_TRANSPARENT) )
     {
-        Window* pTempWindow = pWindow->ImplGetParent();
+        Window* pTempWindow = pOpaqueWindow->ImplGetParent();
         while ( pTempWindow )
         {
             if ( !pTempWindow->IsPaintTransparent() )
             {
-                pWindow = pTempWindow;
+                pOpaqueWindow = pTempWindow;
                 nFlags |= INVALIDATE_CHILDREN;
                 bInvalidateAll = FALSE;
                 break;
@@ -2658,15 +2666,15 @@ void Window::ImplInvalidate( const Regio
     USHORT nOrgFlags = nFlags;
     if ( !(nFlags & (INVALIDATE_CHILDREN | INVALIDATE_NOCHILDREN)) )
     {
-        if ( pWindow->GetStyle() & WB_CLIPCHILDREN )
+        if ( GetStyle() & WB_CLIPCHILDREN )
             nFlags |= INVALIDATE_NOCHILDREN;
         else
             nFlags |= INVALIDATE_CHILDREN;
     }
-    if ( (nFlags & INVALIDATE_NOCHILDREN) && pWindow->mpWindowImpl->mpFirstChild )
+    if ( (nFlags & INVALIDATE_NOCHILDREN) && mpWindowImpl->mpFirstChild )
         bInvalidateAll = FALSE;
     if ( bInvalidateAll )
-        pWindow->ImplInvalidateFrameRegion( NULL, nFlags );
+        ImplInvalidateFrameRegion( NULL, nFlags );
     else
     {
         Rectangle   aRect( Point( mnOutOffX, mnOutOffY ), Size( mnOutWidth, mnOutHeight ) );
@@ -2683,27 +2691,27 @@ void Window::ImplInvalidate( const Regio
             else
                 aRegion.Intersect( *pRegion );
         }
-        pWindow->ImplClipBoundaries( aRegion, TRUE, TRUE );
+        ImplClipBoundaries( aRegion, TRUE, TRUE );
         if ( nFlags & INVALIDATE_NOCHILDREN )
         {
             nFlags &= ~INVALIDATE_CHILDREN;
             if ( !(nFlags & INVALIDATE_NOCLIPCHILDREN) )
             {
                 if ( nOrgFlags & INVALIDATE_NOCHILDREN )
-                    pWindow->ImplClipAllChilds( aRegion );
+                    ImplClipAllChilds( aRegion );
                 else
                 {
-                    if ( pWindow->ImplClipChilds( aRegion ) )
+                    if ( ImplClipChilds( aRegion ) )
                         nFlags |= INVALIDATE_CHILDREN;
                 }
             }
         }
         if ( !aRegion.IsEmpty() )
-            pWindow->ImplInvalidateFrameRegion( &aRegion, nFlags );
+            ImplInvalidateFrameRegion( &aRegion, nFlags );  // transparency is handled here, pOpaqueWindow not required
     }
 
     if ( nFlags & INVALIDATE_UPDATE )
-        pWindow->Update();
+        pOpaqueWindow->Update();        // start painting at the opaque parent
 }
 
 // -----------------------------------------------------------------------
@@ -3278,7 +3286,8 @@ void Window::ImplPosSizeWindow( long nX,
                 nX = mpWindowImpl->mpParent->mnOutWidth - mnOutWidth - nX;
             }
         }
-        if ( mpWindowImpl->mnAbsScreenX != aPtDev.X() || nX != mpWindowImpl->mnX )
+        // check maPos as well, as it could have been changed for client windows (ImplCallMove())
+        if ( mpWindowImpl->mnAbsScreenX != aPtDev.X() || nX != mpWindowImpl->mnX || nOrgX != mpWindowImpl->maPos.X() )
         {
             if ( bCopyBits && !pOverlapRegion )
             {
@@ -3295,7 +3304,8 @@ void Window::ImplPosSizeWindow( long nX,
     }
     if ( nFlags & WINDOW_POSSIZE_Y )
     {
-        if ( nY != mpWindowImpl->mnY )
+        // check maPos as well, as it could have been changed for client windows (ImplCallMove())
+        if ( nY != mpWindowImpl->mnY || nY != mpWindowImpl->maPos.Y() )
         {
             if ( bCopyBits && !pOverlapRegion )
             {
@@ -3445,6 +3455,8 @@ void Window::ImplPosSizeWindow( long nX,
                                     ImplInvalidateFrameRegion( pOverlapRegion, INVALIDATE_CHILDREN );
                             }
                         }
+                        else
+                            bInvalidate = TRUE;
                     }
                     else
                         bInvalidate = TRUE;
@@ -4475,6 +4487,23 @@ Window::~Window()
             }
             DBG_ERROR( aTempStr.GetBuffer() );
         }
+        
+        Window* pMyParent = this;
+        SystemWindow* pMySysWin = NULL;
+
+        while ( pMyParent )
+        {
+            if ( pMyParent->IsSystemWindow() )
+                pMySysWin = (SystemWindow*)pMyParent;
+            pMyParent = pMyParent->GetParent();
+        }
+        if ( pMySysWin && pMySysWin->ImplIsInTaskPaneList( this ) )
+        {
+            ByteString aTempStr( "Window (" );
+            aTempStr += ByteString( GetText(), RTL_TEXTENCODING_UTF8 );
+            aTempStr += ") still in TaskPanelList!";
+            DBG_ERROR( aTempStr.GetBuffer() );
+        }
     }
 #endif
 
Index: vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx,v
retrieving revision 1.10
retrieving revision 1.10.2.4
diff -u -p -u -p -r1.10 -r1.10.2.4
--- vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	31 Jan 2005 09:20:23 -0000	1.10
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	9 Feb 2005 16:50:56 -0000	1.10.2.4
@@ -1397,7 +1399,7 @@ BOOL GtkSalGraphics::NWPaintGTKScrollbar
 		thumbRect.Move( (scrollbarRect.GetWidth() - slider_width) / 2, 0 );
 	}
 
-    BOOL has_slider = ! ( thumbRect.IsEmpty() );
+    BOOL has_slider = ( thumbRect.GetWidth() > 0 && thumbRect.GetHeight() > 0 );
 
 	scrollbarValues = gtk_range_get_adjustment( GTK_RANGE(scrollbarWidget) );
 	if ( scrollbarValues == NULL )
@@ -2451,7 +2453,7 @@ BOOL GtkSalGraphics::NWPaintGTKMenubar(
                                 GTK_STATE_NORMAL,
                                 GTK_SHADOW_NONE, 
                                 &clipRect,
-                                gMenubarWidget,
+                                GTK_WIDGET(m_pWindow),
                                 "base", 
                                 x, y, w, h );
             gtk_paint_box( gMenubarWidget->style,
@@ -2533,7 +2535,7 @@ BOOL GtkSalGraphics::NWPaintGTKPopupMenu
                                 GTK_STATE_NORMAL,
                                 GTK_SHADOW_NONE, 
                                 &clipRect,
-                                gMenuWidget,
+                                GTK_WIDGET(m_pWindow),
                                 "base", 
                                 x, y, w, h );
             gtk_paint_box( gMenuWidget->style,

Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.20
retrieving revision 1.19.2.8
diff -u -p -u -p -r1.20 -r1.19.2.8
--- vcl/unx/gtk/window/gtkframe.cxx	31 Jan 2005 09:20:36 -0000	1.20
+++ vcl/unx/gtk/window/gtkframe.cxx	15 Feb 2005 15:04:22 -0000	1.19.2.8
@@ -66,6 +66,7 @@
 #include <keycodes.hxx>
 #include <wmadaptor.hxx>
 #include <salbmp.h>
+#include <salprn.h>
 #include <floatwin.hxx>
 #include <salprn.h>
 #include <svapp.hxx>
@@ -93,7 +94,7 @@ static USHORT GetModCode( guint state )
     if( (state & GDK_MOD1_MASK) )
 	{
         nCode |= KEY_MOD2;
-		if( (state & GDK_MOD1_MASK) )
+		if( ! (nCode & KEY_MOD1) )
 			nCode |= KEY_CONTROLMOD;
 	}
     if( (state & GDK_BUTTON1_MASK) )
@@ -116,7 +117,7 @@ static USHORT GetKeyCode( guint keyval )
         nCode = KEY_A + (keyval-GDK_a );
     else if( keyval >= GDK_F1 && keyval <= GDK_F26 )
         nCode = KEY_F1 + (keyval-GDK_F1);
-
+    else
     {
         switch( keyval )
         {
@@ -145,7 +148,59 @@ static USHORT GetKeyCode( guint keyval )
     }
     return nCode;
 }
 
+void GtkSalFrame::doKeyCallback( guint state,
+                                 guint keyval,
+                                 guint16 hardware_keycode,
+                                 guint8 group,
+                                 guint32 time,
+                                 bool bDown,
+                                 bool bSendRelease
+                                 )
+{
+    SalKeyEvent aEvent;
+    
+    aEvent.mnTime			= time;    
+    aEvent.mnCode			= GetKeyCode( keyval ) | GetModCode( state );
+    aEvent.mnCharCode		= (USHORT)gdk_keyval_to_unicode( keyval );
+    aEvent.mnRepeat			= 0;
+    
+	vcl::DeletionListener aDel( this );
+    bool bHandled = CallCallback( bDown ? SALEVENT_KEYINPUT : SALEVENT_KEYUP, &aEvent );
+    if( bDown && ! aDel.isDeleted() )
+    {
+        /* #i42122# if application cannot handle the event (e.g. for a shortcut key)
+        *  then try the same key with the default keymap in group 0, so e.g. cyrillic
+        *  keys get translated to the english keyboard layout and can be recognized
+        *  as shortcuts (see also generic plugin)
+        */
+        if( ! bHandled && group != 0)
+        {
+            // check other mapping
+            gint eff_group, level;
+            GdkModifierType consumed;
+            guint updated_keyval = 0;
+            if( gdk_keymap_translate_keyboard_state( NULL,
+                                                     hardware_keycode,
+                                                     (GdkModifierType)0,
+                                                     0,
+                                                     &updated_keyval,
+                                                     &eff_group,
+                                                     &level,
+                                                     &consumed ) )
+            {
+                aEvent.mnCode	= GetKeyCode( updated_keyval ) | GetModCode( state );
+                aEvent.mnCharCode = (USHORT)gdk_keyval_to_unicode( updated_keyval );
+                CallCallback( SALEVENT_KEYINPUT, &aEvent );
+            }
+        }
+        if( bSendRelease && ! aDel.isDeleted() )
+        {
+            CallCallback( SALEVENT_KEYUP, &aEvent );
+        }
+    }
+}
+
 GtkSalFrame::GraphicsHolder::~GraphicsHolder()
 {
     delete pGraphics;
@@ -201,7 +263,7 @@ GtkSalFrame::~GtkSalFrame()
     
     if( m_pIMContext )
     {
-        gtk_im_context_reset( m_pIMContext );
+        hardIMReset();
         gtk_im_context_set_client_window( m_pIMContext, NULL );
         g_object_unref( m_pIMContext );
     }
@@ -215,6 +277,40 @@ GtkSalFrame::~GtkSalFrame()
         g_object_unref(G_OBJECT( m_pForeignTopLevel) );
 }
 
+void GtkSalFrame::hardIMReset()
+{
+#if OSL_DEBUG_LEVEL > 1
+	fprintf( stderr, "IMReset - this should flush the IM's pre-edit buffer\n" );
+#endif
+
+	vcl::DeletionListener aDel( this );
+
+	if( m_pIMContext )
+    {
+        gtk_im_context_reset( m_pIMContext );
+        
+        // a correctly implemented _reset method will
+        // emit a 'commit' signal if pending pre-edit
+        // and also, a predit_changed to '' which will
+        // end input.
+        
+        // Since few IM's are correctly implemented,
+        // this will end the ext text input at least
+        // from OO.o's perspective if it is still active.
+    }
+    
+	if( !aDel.isDeleted() )
+	{
+#if OSL_DEBUG_LEVEL > 1
+		if( m_bWasPreedit )
+			fprintf( stderr, "Error: ** Your IM ('%s') is broken wrt. reset **\n",
+					 g_getenv ("GTK_IM_MODULE") );
+#endif
+	    m_bWasPreedit = false;
+        CallCallback( SALEVENT_ENDEXTTEXTINPUT, (void*)NULL );
+	}
+}
+
 void GtkSalFrame::InitCommon()
 {
     // connect signals
@@ -307,6 +403,8 @@ void GtkSalFrame::InitCommon()
         maGeometry.nLeftDecoration		= 0;
         maGeometry.nRightDecoration		= 0;
     }
+
+	SetIcon(1);
 }
 
 /*  Sadly gtk_window_set_accept_focus exists only since gtk 2.4
@@ -537,6 +635,10 @@ void GtkSalFrame::SetIcon( USHORT nIcon 
 
 	GList *pPixbufs = NULL;
 
+    // default icon is icon #1
+    if( nIcon == 0 )
+        nIcon = 1;
+
 	{
 	    VCL_CUSTOM_ICON_FN *pCustomIcon = 0;
 		char *pSymbol = g_strdup_printf ("%s%d", VCL_CUSTOM_ICON_BASE, nIcon );
@@ -681,7 +783,7 @@ void GtkSalFrame::Show( BOOL bVisible, B
             if( m_pIMContext )
             {
                 gtk_im_context_focus_out( m_pIMContext );
-                gtk_im_context_reset( m_pIMContext );
+                hardIMReset();
             }
         }
         CallCallback( SALEVENT_RESIZE, NULL );
@@ -1101,6 +1203,10 @@ void GtkSalFrame::SetPointerPos( long nX
     XWarpPointer( GDK_DISPLAY_XDISPLAY (pDisplay), None,
 				  GDK_WINDOW_XID (gdk_screen_get_root_window( pScreen ) ),
                   0, 0, 0, 0, nWindowLeft, nWindowTop);
+    // #i38648# ask for the next motion hint
+    gint x, y;
+    GdkModifierType mask;
+    gdk_window_get_pointer( GTK_WIDGET(m_pWindow)->window, &x, &y, &mask );
 }
 
 void GtkSalFrame::Flush()
@@ -1149,19 +1255,8 @@ SalFrame::SalPointerState GtkSalFrame::G
     return aState;
 }
 
-void GtkSalFrame::SetInputContext( SalInputContext* pContext )
+void GtkSalFrame::createIMContext()
 {
-    if( ! pContext )
-        return;
-
-    if( ! (pContext->mnOptions & SAL_INPUTCONTEXT_TEXT) )
-    {
-        if( m_pIMContext )
-            gtk_im_context_focus_out( m_pIMContext );
-        return;
-    }
-
-    // create a new im context
     if( ! m_pIMContext )
     {
         m_pIMContext = gtk_im_multicontext_new ();
@@ -1173,17 +1268,53 @@ void GtkSalFrame::SetInputContext( SalIn
                           G_CALLBACK (signalIMRetrieveSurrounding), this );
         g_signal_connect( m_pIMContext, "delete_surrounding",
                           G_CALLBACK (signalIMDeleteSurrounding), this );
+        g_signal_connect( m_pIMContext, "preedit_start",
+                          G_CALLBACK (signalIMPreeditStart), this );
+        g_signal_connect( m_pIMContext, "preedit_end",
+                          G_CALLBACK (signalIMPreeditEnd), this );
 
         gtk_im_context_set_client_window( m_pIMContext, GTK_WIDGET(m_pWindow)->window );
 		gtk_im_context_focus_in( m_pIMContext );
+        m_bWasPreedit = false;
    }
 }
+
+void GtkSalFrame::SetInputContext( SalInputContext* pContext )
+{
+    if( ! pContext )
+        return;
+
+    if( ! (pContext->mnOptions & SAL_INPUTCONTEXT_TEXT) )
+    {
+        if( m_pIMContext )
+            gtk_im_context_focus_out( m_pIMContext );
+        return;
+    }
+
+    // create a new im context
+    createIMContext();
+}
+
 void GtkSalFrame::EndExtTextInput( USHORT nFlags )
 {
     if( m_pIMContext )
-        gtk_im_context_reset( m_pIMContext );
-    m_bWasPreedit = false;
-    CallCallback( SALEVENT_ENDEXTTEXTINPUT, NULL );
+    {
+        // since some IMs won't do a reset on gtk_im_context_reset
+        // and not empty their preedit buffer, there does not
+        // seem to be another choice than to create a completely
+        // new IC. We cannot use the same trick as the generic plugin
+        // here (which disables the preedit state) since gtk
+        // does not give us that much control
+        
+        // first give IC a chance to deinitialize
+        hardIMReset();
+        gtk_im_context_set_client_window( m_pIMContext, NULL );
+        // destroy old IC
+        g_object_unref( m_pIMContext );
+        m_pIMContext = NULL;
+        // create new IC
+        createIMContext();
+    }
 }
 
 void GtkSalFrame::updateIMSpotLocation()
@@ -1264,9 +1395,10 @@ bool GtkSalFrame::SetPluginParent( Syste
 {
     if( m_pIMContext )
     {
-        gtk_im_context_reset( m_pIMContext );
+        hardIMReset();
         gtk_im_context_set_client_window( m_pIMContext, NULL );
         g_object_unref( m_pIMContext );
+        m_pIMContext = NULL;
     }
     if( m_pFixedContainer )
         gtk_widget_destroy( GTK_WIDGET(m_pFixedContainer) );
@@ -1529,7 +1661,6 @@ gboolean GtkSalFrame::signalMotion( GtkW
 			gint x, y;
 			GdkModifierType mask;
 			gdk_window_get_pointer( GTK_WIDGET(pThis->m_pWindow)->window, &x, &y, &mask );
-
 		}
 	}
 
@@ -1596,7 +1727,7 @@ gboolean GtkSalFrame::signalFocus( GtkWi
     {
         if( pEvent->in )
         {
-            gtk_im_context_reset( pThis->m_pIMContext );
+            pThis->hardIMReset();
             gtk_im_context_set_client_window( pThis->m_pIMContext, GTK_WIDGET(pThis->m_pWindow)->window );
             gtk_im_context_focus_in( pThis->m_pIMContext );
         }
@@ -1606,6 +1737,11 @@ gboolean GtkSalFrame::signalFocus( GtkWi
         }
     }
 
+    // ask for changed printers like generic implementation
+    if( pEvent->in )
+        if( static_cast< X11SalInstance* >(GetSalData()->pInstance_)->isPrinterInit() )
+            vcl_sal::PrinterUpdate::update();
+
     // FIXME: find out who the hell steals the focus from our frame
     // if we are plugged and a float opens; why does the same not
     // happen unplugged ? Is the plugging application (SimpleViewer in SDK)
@@ -1744,6 +1880,21 @@ gboolean GtkSalFrame::signalKey( GtkWidg
             g_object_unref( pRef );
             if( bResult )
                 return TRUE;
+            else
+            {
+                DBG_ASSERT( pThis->m_nPrevKeyPresses > 0, "key press has vanished !" );
+                if( ! pThis->m_aPrevKeyPresses.empty() ) // sanity check
+                {
+                    // event was not swallowed, do not filter a following
+                    // key release event
+                    // note: this relies on gtk_im_context_filter_keypress
+                    // returning without calling a handler (in the "not swallowed"
+                    // case ) which might change the previous key press list so
+                    // we would pop the wrong event here
+                    pThis->m_aPrevKeyPresses.pop_back();
+                    pThis->m_nPrevKeyPresses--;
+                }
+            }
         }
     }
     
@@ -1875,15 +2026,7 @@ gboolean GtkSalFrame::signalKey( GtkWidg
     }
     else
     {
-        SalKeyEvent aEvent;
-
-        aEvent.mnTime			= pEvent->time;
-        aEvent.mnCode			= GetKeyCode( pEvent->keyval ) | GetModCode( pEvent->state );
-        aEvent.mnCharCode		= (USHORT)gdk_keyval_to_unicode( pEvent->keyval );
-        aEvent.mnRepeat			= 0;
-
-        pThis->CallCallback( (pEvent->type == GDK_KEY_PRESS) ? SALEVENT_KEYINPUT : SALEVENT_KEYUP, &aEvent );
-
+        pThis->doKeyCallback( pEvent->state, pEvent->keyval, pEvent->hardware_keycode, pEvent->group, pEvent->time, (pEvent->type == GDK_KEY_PRESS), false );
 		if( ! aDel.isDeleted() )
 		{
 			pThis->m_bSendModChangeOnRelease = false;
@@ -1980,25 +2123,17 @@ void GtkSalFrame::signalIMCommit( GtkIMC
         && ! pThis->m_aPrevKeyPresses.empty()
         )
     {
-        SalKeyEvent aEvent;
-
-        aEvent.mnTime			= 0;
-        aEvent.mnCode			= GetKeyCode( pThis->m_aPrevKeyPresses.back().keyval ) |
-								  GetModCode( pThis->m_aPrevKeyPresses.back().state );
-        aEvent.mnCharCode		= aTextEvent.maText.GetChar(0);
-        aEvent.mnRepeat			= 0;
-
+        const PreviousKeyPress& rKP = pThis->m_aPrevKeyPresses.back();
+        
         pThis->m_bWasPreedit = false;
-        pThis->CallCallback( SALEVENT_KEYINPUT, &aEvent );
-		if( ! aDel.isDeleted() )
-			pThis->CallCallback( SALEVENT_KEYUP, &aEvent );
+        pThis->doKeyCallback( rKP.state, rKP.keyval, rKP.hardware_keycode, rKP.group, rKP.time, true, true );
         return;
     }
 
     pThis->m_bWasPreedit = false;
     pThis->CallCallback( SALEVENT_EXTTEXTINPUT, (void*)&aTextEvent);
 	if( ! aDel.isDeleted() )
-		pThis->CallCallback( SALEVENT_ENDEXTTEXTINPUT, (void*)NULL );
+		pThis->CallCallback( SALEVENT_ENDEXTTEXTINPUT, NULL );
 }
 
 void GtkSalFrame::signalIMPreeditChanged( GtkIMContext* pContext, gpointer frame )
@@ -2013,6 +2148,11 @@ void GtkSalFrame::signalIMPreeditChanged
                                        &pText,
                                        &pAttrs,
                                        &nCursorPos );
+#if OSL_DEBUG_LEVEL > 1
+	fprintf( stderr, ":signalImPreeditChanged '%s'\n", pText );
+#endif
+	bool bEndPreedit = (!pText || !*pText) && pThis->m_bWasPreedit;
+
     SalExtTextInputEvent aTextEvent;
 
     aTextEvent.mnTime 			= 0;
@@ -2078,10 +2218,16 @@ void GtkSalFrame::signalIMPreeditChanged
     pango_attr_list_unref( pAttrs );
 
     GTK_YIELD_GRAB();
+    
+    vcl::DeletionListener aDel( pThis );
+    
     pThis->m_bWasPreedit = true;
     pThis->CallCallback( SALEVENT_EXTTEXTINPUT, (void*)&aTextEvent);
 
     delete [] pSalAttribs;
+
+	if( bEndPreedit && ! aDel.isDeleted() )
+        pThis->CallCallback( SALEVENT_ENDEXTTEXTINPUT, NULL );
 }
 
 void GtkSalFrame::signalIMPreeditStart( GtkIMContext* pContext, gpointer frame )
@@ -2091,7 +2237,12 @@ void GtkSalFrame::signalIMPreeditStart( 
 
 void GtkSalFrame::signalIMPreeditEnd( GtkIMContext* pContext, gpointer frame )
 {
-//    GtkSalFrame* pThis = (GtkSalFrame*)frame;
+    GtkSalFrame* pThis = (GtkSalFrame*)frame;
+#if OSL_DEBUG_LEVEL > 1
+	fprintf( stderr, ":signalImPreeditEnd\n" );
+#endif
+    GTK_YIELD_GRAB();
+	pThis->CallCallback( SALEVENT_ENDEXTTEXTINPUT, NULL );
 }
 
 gboolean GtkSalFrame::signalIMRetrieveSurrounding( GtkIMContext* pContext, gpointer frame )
Index: vcl/unx/inc/plugins/gtk/gtkframe.hxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/inc/plugins/gtk/gtkframe.hxx,v
retrieving revision 1.11
retrieving revision 1.11.2.2
diff -u -p -u -p -r1.11 -r1.11.2.2
--- vcl/unx/inc/plugins/gtk/gtkframe.hxx	21 Jan 2005 13:37:35 -0000	1.11
+++ vcl/unx/inc/plugins/gtk/gtkframe.hxx	15 Feb 2005 15:04:22 -0000	1.11.2.2
@@ -111,21 +111,27 @@ class GtkSalFrame : public SalFrame
         guint32 time;
         guint   state;
         guint   keyval;
+        guint16 hardware_keycode;
+        guint8  group;
 
         PreviousKeyPress (GdkEventKey *event)
         :   window (NULL),
             send_event (0),
             time (0),
             state (0),
-            keyval (0)
+            keyval (0),
+            hardware_keycode (0),
+            group (0)
         {
             if (event)
             {
-                window = event->window;
-                send_event = event->send_event;
-                time = event->time;
-                state = event->state;
-                keyval = event->keyval;
+                window              = event->window;
+                send_event          = event->send_event;
+                time                = event->time;
+                state               = event->state;
+                keyval              = event->keyval;
+                hardware_keycode    = event->hardware_keycode;
+                group               = event->group;
             }
         }
 
@@ -134,7 +140,9 @@ class GtkSalFrame : public SalFrame
             send_event( rPrev.send_event ),
             time( rPrev.time ),
             state( rPrev.state ),
-            keyval( rPrev.keyval )
+            keyval( rPrev.keyval ),
+            hardware_keycode( rPrev.hardware_keycode ),
+            group( rPrev.group )
         {}
 
         bool PreviousKeyPress::operator== (GdkEventKey *event) const
@@ -144,6 +152,8 @@ class GtkSalFrame : public SalFrame
                 && (event->send_event == send_event)
                 && (event->state == state)
                 && (event->keyval == keyval)
+                && (event->hardware_keycode == hardware_keycode)
+                && (event->group == group)
                 && (event->time - time < 3)
                 ;
         }
@@ -213,6 +223,18 @@ class GtkSalFrame : public SalFrame
     void			SetDefaultSize();
 	void			setAutoLock( bool bLock );
 	void			setScreenSaverTimeout( int nTimeout );
+	void            hardIMReset();
+    void            createIMContext();
+    
+    void            doKeyCallback( guint state,
+                                   guint keyval,
+                                   guint16 hardware_keycode,
+                                   guint8 group,
+                                   guint32 time,
+                                   bool bDown,
+                                   bool bSendRelease
+                                   );
+
 
     GdkNativeWindow findTopLevelSystemWindow( GdkNativeWindow aWindow );
 
Index: vcl/unx/source/app/saldisp.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/saldisp.cxx,v
retrieving revision 1.60
retrieving revision 1.60.126.1
diff -u -p -u -p -r1.60 -r1.60.126.1
--- vcl/unx/source/app/saldisp.cxx	9 Sep 2004 09:17:35 -0000	1.60
+++ vcl/unx/source/app/saldisp.cxx	15 Feb 2005 16:34:05 -0000	1.60.126.1
@@ -2517,10 +2517,13 @@ long SalX11Display::Dispatch( XEvent *pE
             }
             break;
         case MappingNotify:
-            if( MappingKeyboard == pEvent->xmapping.request )
+            if( MappingKeyboard == pEvent->xmapping.request ||
+                MappingModifier == pEvent->xmapping.request )
+            {
                 XRefreshKeyboardMapping( &pEvent->xmapping );
-            else if( MappingModifier == pEvent->xmapping.request )
-                ModifierMapping();
+                if( MappingModifier == pEvent->xmapping.request )
+                    ModifierMapping();
+            }
             break;
 
 		default:
Index: vcl/unx/source/plugadapt/salplug.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/plugadapt/salplug.cxx,v
retrieving revision 1.13
retrieving revision 1.13.2.1
diff -u -p -u -p -r1.13 -r1.13.2.1
--- vcl/unx/source/plugadapt/salplug.cxx	21 Jan 2005 13:38:29 -0000	1.13
+++ vcl/unx/source/plugadapt/salplug.cxx	1 Feb 2005 18:37:27 -0000	1.13.2.1
@@ -464,10 +464,6 @@ SalInstance *CreateSalInstance()
     if( pUsePlugin && *pUsePlugin )
         pInst = tryInstance( OUString::createFromAscii( pUsePlugin ) );
 
-    // fallback from kde to gtk because there is a qt->gtk theme engine
-    if( ! pInst && pUsePlugin && ! strncmp( pUsePlugin, "kde", 3 ) )
-        pInst = tryInstance( OUString( RTL_CONSTASCII_USTRINGPARAM( "gtk" ) ) );
-    
     // fallback to gen
     if( ! pInst )
         pInst = tryInstance( OUString( RTL_CONSTASCII_USTRINGPARAM( "gen" ) ) );
Index: vcl/unx/source/window/salframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/window/salframe.cxx,v
retrieving revision 1.187
retrieving revision 1.187.2.2
diff -u -p -u -p -r1.187 -r1.187.2.2
--- vcl/unx/source/window/salframe.cxx	21 Jan 2005 13:38:53 -0000	1.187
+++ vcl/unx/source/window/salframe.cxx	15 Feb 2005 16:34:06 -0000	1.187.2.2
@@ -717,7 +717,7 @@ X11SalFrame::X11SalFrame( SalFrame *pPar
     mbShaded					= false;
     mbFullScreen				= false;
 
-    mnIconID					= 0; // ICON_DEFAULT
+    mnIconID					= 1; // ICON_DEFAULT
 
 	if( mpParent )
 		mpParent->maChildren.push_back( this );
@@ -925,7 +925,11 @@ void X11SalFrame::SetIcon( USHORT nIcon 
 {
     if ( !( nStyle_ & (SAL_FRAME_STYLE_CHILD|SAL_FRAME_STYLE_FLOAT) ) )
     {
-        mnIconID = (nIcon > 0) ? nIcon : 1;
+        // 0 == default icon -> #1
+        if ( nIcon == 0 )
+            nIcon = 1;
+
+        mnIconID = nIcon;
 
         XIconSize *pIconSize = NULL;
         int nSizes = 0;
@@ -2920,6 +2924,11 @@ long X11SalFrame::HandleKeyEvent( XKeyEv
                     if ((nKeyCode != 0) && ((nKeyCode | nModCode) != aKeyEvt.mnCode))
                     {
                         aKeyEvt.mnCode = nKeyCode | nModCode;
+                        if( nKeySym )
+                        {
+                            String aKeyName = pDisplay_->GetKeyNameFromKeySym( nKeySym );
+                            aKeyEvt.mnCharCode = aKeyName.GetChar(0);
+                        }
 					    CallCallback(SALEVENT_KEYINPUT, &aKeyEvt);
                     }
                 }
Index: padmin/source/newppdlg.cxx
===================================================================
RCS file: /cvs/gsl/padmin/source/newppdlg.cxx,v
retrieving revision 1.10
retrieving revision 1.10.52.1
diff -u -p -u -p -r1.10 -r1.10.52.1
--- padmin/source/newppdlg.cxx	17 Mar 2004 10:44:51 -0000	1.10
+++ padmin/source/newppdlg.cxx	15 Feb 2005 09:44:56 -0000	1.10.52.1
@@ -202,7 +202,7 @@ void PPDImportDialog::Import()
 		{
 #if OSL_DEBUG_LEVEL > 1
 			fprintf( stderr, "Warning: File %s has empty printer name.\n",
-					 ByteString( aPath.PathToFileName(), osl_getThreadTextEncoding() ).GetBuffer() );
+					 rtl::OUStringToOString( aPath.PathToFileName(), osl_getThreadTextEncoding() ).getStr() );
 #endif
 			continue;
 		}
