Index: officecfg/registry/schema/org/openoffice/Office/Writer.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Writer.xcs,v
retrieving revision 1.35
retrieving revision 1.35.76.1
diff -u -p -u -p -r1.35 -r1.35.76.1
--- officecfg/registry/schema/org/openoffice/Office/Writer.xcs	15 Nov 2004 13:27:14 -0000	1.35
+++ officecfg/registry/schema/org/openoffice/Office/Writer.xcs	24 Jan 2005 20:14:13 -0000	1.35.76.1
@@ -1446,7 +1446,7 @@
 						<desc>Specifies if the vertical ruler is displayed.</desc>
 						<label>Vertical Ruler</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="IsVerticalRulerRight" oor:type="xs:boolean">
 					<!-- UIHints: Tools - Options - Text document - Layout - [Section] Window -->
Index: sw/source/core/frmedt/fews.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/frmedt/fews.cxx,v
retrieving revision 1.31
retrieving revision 1.30.104.2
diff -u -p -u -p -r1.31 -r1.30.104.2
--- sw/source/core/frmedt/fews.cxx	18 Jan 2005 14:27:27 -0000	1.31
+++ sw/source/core/frmedt/fews.cxx	25 Jan 2005 07:32:33 -0000	1.30.104.2
@@ -976,9 +976,9 @@ void SwFEShell::CalcBoundRect( SwRect& _
             objectpositioning::SwEnvironmentOfAnchoredObject aEnvOfObj(
                                                             _bFollowTextFlow );
             const SwLayoutFrm& rHoriEnvironLayFrm =
-                                aEnvOfObj.GetHoriEnvironmentLayoutFrm( *pFrm, false );
+                                aEnvOfObj.GetHoriEnvironmentLayoutFrm( *pFrm );
             const SwLayoutFrm& rVertEnvironLayFrm =
-                                aEnvOfObj.GetVertEnvironmentLayoutFrm( *pFrm, false );
+                                aEnvOfObj.GetVertEnvironmentLayoutFrm( *pFrm );
             SwRect aHoriEnvironRect( rHoriEnvironLayFrm.Frm() );
             SwRect aVertEnvironRect;
             if ( _bFollowTextFlow )
Index: sw/source/core/inc/environmentofanchoredobject.hxx
===================================================================
RCS file: /cvs/sw/sw/source/core/inc/environmentofanchoredobject.hxx,v
retrieving revision 1.2
retrieving revision 1.2.464.1
diff -u -p -u -p -r1.2 -r1.2.464.1
--- sw/source/core/inc/environmentofanchoredobject.hxx	8 Mar 2004 13:57:55 -0000	1.2
+++ sw/source/core/inc/environmentofanchoredobject.hxx	20 Jan 2005 15:16:34 -0000	1.2.464.1
@@ -103,8 +103,8 @@ namespace objectpositioning
 
                 this is, if object hasn't to follow the text flow:
                 - page frame.
-                - Exception: If environment layout frame is used for page alignment,
-                  it's the cell frame, if anchored inside a cell.
+                OD 2005-01-20 #118546# - no exception any more. Thus remove
+                parameter <_bForPageAlignment>
 
                 @author OD
 
@@ -113,15 +113,10 @@ namespace objectpositioning
                 oriented at (typically it's the anchor frame).
                 starting point for the search of the layout frame.
 
-                @param _bForPageAlignment
-                input parameter - indicates, if the environment layout frame
-                for the page alignments has to be determined or not.
-
                 @return reference to the layout frame, which determines the
                 the horizontal environment the object has to be positioned in.
             */
-            const SwLayoutFrm& GetHoriEnvironmentLayoutFrm( const SwFrm& _rHoriOrientFrm,
-                                                            const bool _bForPageAlignment ) const;
+            const SwLayoutFrm& GetHoriEnvironmentLayoutFrm( const SwFrm& _rHoriOrientFrm ) const;
 
             /** determine environment layout frame for possible vertical object
                 positions respectively for alignments to 'page areas'
@@ -136,8 +131,8 @@ namespace objectpositioning
 
                 this is, if object hasn't to follow the text flow:
                 - page frame.
-                - Exception: If environment layout frame is used for page alignment,
-                  it's the cell frame, if anchored inside a cell.
+                OD 2005-01-20 #118546# - no exception any more. Thus remove
+                parameter <_bForPageAlignment>
 
                 @author OD
 
@@ -146,15 +141,10 @@ namespace objectpositioning
                 oriented at (typically it's the anchor frame).
                 starting point for the search of the layout frame.
 
-                @param _bForPageAlignment
-                input parameter - indicates, if the environment layout frame
-                for the page alignments has to be determined or not.
-
                 @return reference to the layout frame, which determines the
                 the vertical environment the object has to be positioned in.
             */
-            const SwLayoutFrm& GetVertEnvironmentLayoutFrm( const SwFrm& _rVertOrientFrm,
-                                                            const bool _bForPageAlignment ) const;
+            const SwLayoutFrm& GetVertEnvironmentLayoutFrm( const SwFrm& _rVertOrientFrm ) const;
     };
 };
 
Index: sw/source/core/layout/flylay.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/layout/flylay.cxx,v
retrieving revision 1.39
retrieving revision 1.39.52.1
diff -u -p -u -p -r1.39 -r1.39.52.1
--- sw/source/core/layout/flylay.cxx	23 Dec 2004 10:07:54 -0000	1.39
+++ sw/source/core/layout/flylay.cxx	20 Jan 2005 15:20:00 -0000	1.39.52.1
@@ -1124,7 +1124,7 @@ BOOL CalcClipRect( const SdrObject *pSdr
                 objectpositioning::SwEnvironmentOfAnchoredObject
                                                 aEnvOfObj( bFollowTextFlow );
                 const SwLayoutFrm& rVertClipFrm =
-                    aEnvOfObj.GetVertEnvironmentLayoutFrm( *pVertPosOrientFrm, !bMove );
+                    aEnvOfObj.GetVertEnvironmentLayoutFrm( *pVertPosOrientFrm );
                 if ( rV.GetRelationOrient() == REL_PG_FRAME )
                 {
                     rRect = rVertClipFrm.Frm();
Index: sw/source/core/objectpositioning/anchoredobjectposition.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/objectpositioning/anchoredobjectposition.cxx,v
retrieving revision 1.9
retrieving revision 1.9.108.1
diff -u -p -u -p -r1.9 -r1.9.108.1
--- sw/source/core/objectpositioning/anchoredobjectposition.cxx	16 Nov 2004 15:50:58 -0000	1.9
+++ sw/source/core/objectpositioning/anchoredobjectposition.cxx	20 Jan 2005 15:17:49 -0000	1.9.108.1
@@ -832,7 +832,7 @@ SwTwips SwAnchoredObjectPosition::_CalcR
 {
     // determine 'page' alignment layout frame
     const SwFrm& rPageAlignLayFrm =
-            _rEnvOfObj.GetHoriEnvironmentLayoutFrm( _rHoriOrientFrm, true );
+            _rEnvOfObj.GetHoriEnvironmentLayoutFrm( _rHoriOrientFrm );
 
     const bool bEvenPage = !rPageAlignLayFrm.OnRightPage();
     const bool bToggle = _rHoriOrient.IsPosToggle() && bEvenPage;
@@ -934,7 +934,7 @@ SwTwips SwAnchoredObjectPosition::_CalcR
     // adjust calculated relative horizontal position, in order to
     // keep object inside 'page' alignment layout frame
     const SwFrm& rEnvironmentLayFrm =
-            _rEnvOfObj.GetHoriEnvironmentLayoutFrm( _rHoriOrientFrm, false );
+            _rEnvOfObj.GetHoriEnvironmentLayoutFrm( _rHoriOrientFrm );
     nRelPosX = _AdjustHoriRelPos( rEnvironmentLayFrm, nRelPosX );
 
     // if object is a Writer fly frame and it's anchored to a content and
Index: sw/source/core/objectpositioning/environmentofanchoredobject.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/objectpositioning/environmentofanchoredobject.cxx,v
retrieving revision 1.3
retrieving revision 1.3.348.1
diff -u -p -u -p -r1.3 -r1.3.348.1
--- sw/source/core/objectpositioning/environmentofanchoredobject.cxx	28 Jun 2004 13:42:54 -0000	1.3
+++ sw/source/core/objectpositioning/environmentofanchoredobject.cxx	20 Jan 2005 15:17:48 -0000	1.3.348.1
@@ -89,30 +89,16 @@ SwEnvironmentOfAnchoredObject::~SwEnviro
     @author OD
 */
 const SwLayoutFrm& SwEnvironmentOfAnchoredObject::GetHoriEnvironmentLayoutFrm(
-                                            const SwFrm& _rHoriOrientFrm,
-                                            const bool _bForPageAlignment ) const
+                                            const SwFrm& _rHoriOrientFrm ) const
 {
     const SwFrm* pHoriEnvironmentLayFrm = &_rHoriOrientFrm;
 
     if ( !mbFollowTextFlow )
     {
-        if ( _bForPageAlignment )
-        {
-            while ( !pHoriEnvironmentLayFrm->IsCellFrm() &&
-                    !pHoriEnvironmentLayFrm->IsPageFrm() )
-            {
-                pHoriEnvironmentLayFrm =
-                    pHoriEnvironmentLayFrm->IsFlyFrm()
-                    ? static_cast<const SwFlyFrm*>(pHoriEnvironmentLayFrm)->GetAnchorFrm()
-                    : pHoriEnvironmentLayFrm->GetUpper();
-                ASSERT( pHoriEnvironmentLayFrm,
-                        "SwEnvironmentOfAnchoredObject::GetHoriEnvironmentLayoutFrm(..) - no page|cell frame found" );
-            }
-        }
-        else
-        {
-            pHoriEnvironmentLayFrm = _rHoriOrientFrm.FindPageFrm();
-        }
+        // --> OD 2005-01-20 #118546# - no exception any more for page alignment.
+        // the page frame determines the horizontal layout environment.
+        pHoriEnvironmentLayFrm = _rHoriOrientFrm.FindPageFrm();
+        // <--
     }
     else
     {
@@ -139,30 +125,16 @@ const SwLayoutFrm& SwEnvironmentOfAnchor
     @author OD
 */
 const SwLayoutFrm& SwEnvironmentOfAnchoredObject::GetVertEnvironmentLayoutFrm(
-                                            const SwFrm& _rVertOrientFrm,
-                                            const bool _bForPageAlignment ) const
+                                            const SwFrm& _rVertOrientFrm ) const
 {
     const SwFrm* pVertEnvironmentLayFrm = &_rVertOrientFrm;
 
     if ( !mbFollowTextFlow )
     {
-        if ( _bForPageAlignment )
-        {
-            while ( !pVertEnvironmentLayFrm->IsCellFrm() &&
-                    !pVertEnvironmentLayFrm->IsPageFrm() )
-            {
-                pVertEnvironmentLayFrm =
-                    pVertEnvironmentLayFrm->IsFlyFrm()
-                    ? static_cast<const SwFlyFrm*>(pVertEnvironmentLayFrm)->GetAnchorFrm()
-                    : pVertEnvironmentLayFrm->GetUpper();
-                ASSERT( pVertEnvironmentLayFrm,
-                        "SwEnvironmentOfAnchoredObject::GetVertEnvironmentLayoutFrm(..) - proposed frame not found" );
-            }
-        }
-        else
-        {
-            pVertEnvironmentLayFrm = _rVertOrientFrm.FindPageFrm();
-        }
+        // --> OD 2005-01-20 #118546# - no exception any more for page alignment.
+        // the page frame determines the vertical layout environment.
+        pVertEnvironmentLayFrm = _rVertOrientFrm.FindPageFrm();
+        // <--
     }
     else
     {
Index: sw/source/core/objectpositioning/tocntntanchoredobjectposition.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/objectpositioning/tocntntanchoredobjectposition.cxx,v
retrieving revision 1.9
retrieving revision 1.9.108.1
diff -u -p -u -p -r1.9 -r1.9.108.1
--- sw/source/core/objectpositioning/tocntntanchoredobjectposition.cxx	16 Nov 2004 15:51:15 -0000	1.9
+++ sw/source/core/objectpositioning/tocntntanchoredobjectposition.cxx	20 Jan 2005 15:19:09 -0000	1.9.108.1
@@ -318,7 +318,7 @@ void SwToCntntAnchoredObjectPosition::Ca
         // OD 22.09.2003 #i18732# - determine layout frame for vertical
         // positions aligned to 'page areas'.
         const SwLayoutFrm& rPageAlignLayFrm =
-                aEnvOfObj.GetVertEnvironmentLayoutFrm( *pOrientFrm, true );
+                aEnvOfObj.GetVertEnvironmentLayoutFrm( *pOrientFrm );
 
         if ( aVert.GetVertOrient() != VERT_NONE )
         {
@@ -626,7 +626,7 @@ void SwToCntntAnchoredObjectPosition::Ca
                 // OD 08.09.2003 #110354# - allow negative position, but keep it
                 // inside environment layout frame.
                 const SwLayoutFrm& rVertEnvironLayFrm =
-                    aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm, false );
+                    aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm );
                 // --> OD 2004-07-22 #i31805# - do not check, if bottom of
                 // anchored object would fit into environment layout frame, if
                 // anchored object has to follow the text flow.
@@ -666,7 +666,7 @@ void SwToCntntAnchoredObjectPosition::Ca
                         // --> OD 2004-07-06 #i28701# - adjust calculated
                         // relative vertical position to object's environment.
                         const SwFrm& rVertEnvironLayFrm =
-                            aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm, false );
+                            aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm );
                         // --> OD 2004-08-20 - do not check, if bottom of
                         // anchored object would fit into environment layout
                         // frame, if anchored object has to follow the text flow.
@@ -759,7 +759,7 @@ void SwToCntntAnchoredObjectPosition::Ca
                             // OD 06.10.2003 #i18732# - do not follow text flow respectively
                             // align at 'page areas', but stay inside given environment
                             const SwFrm& rVertEnvironLayFrm =
-                                aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm, false );
+                                aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm );
                             nRelPosY = _AdjustVertRelPos( nTopOfAnch, bVert,
                                                           rVertEnvironLayFrm,
                                                           nRelPosY,
@@ -873,7 +873,7 @@ void SwToCntntAnchoredObjectPosition::Ca
                     else
                         nTmpRelPosY = aRelPos.Y() + nDist;
                     const SwLayoutFrm& rVertEnvironLayFrm =
-                        aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm, false );
+                        aEnvOfObj.GetVertEnvironmentLayoutFrm( *pUpperOfOrientFrm );
                     nTmpRelPosY = _AdjustVertRelPos( nTopOfAnch, bVert,
                                                      rVertEnvironLayFrm,
                                                      nTmpRelPosY,
Index: sw/source/filter/ww8/ww8graf.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/ww8graf.cxx,v
retrieving revision 1.128
retrieving revision 1.128.10.1
diff -u -p -u -p -r1.128 -r1.128.10.1
--- sw/source/filter/ww8/ww8graf.cxx	11 Jan 2005 13:26:10 -0000	1.128
+++ sw/source/filter/ww8/ww8graf.cxx	20 Jan 2005 15:13:28 -0000	1.128.10.1
@@ -2507,15 +2507,6 @@ RndStdIds SwWW8ImplReader::ProcessEscher
         }
         // <--
 
-        //#111875#
-        if ((eHoriRel == REL_PG_FRAME) && (eHoriOri == HORI_RIGHT))
-        {
-            eHoriRel = REL_FRM_RIGHT;
-            eHoriOri = HORI_NONE;
-            pFSPA->nXaRight -= pFSPA->nXaLeft;
-            pFSPA->nXaLeft = 0;
-        }
-
         //#109311# Miserable miserable hack.
         if (!bDrawingHacks)
         {
@@ -2528,6 +2519,19 @@ RndStdIds SwWW8ImplReader::ProcessEscher
             }
         }
 
+        // --> OD 2005-01-20 #118546# - if the object is anchored inside
+        // a table cell, is horizontal aligned at frame|character and
+        // has wrap through, but its attribute 'layout in table cell' isn't set,
+        // convert its horizontal alignment to page text area.
+        if ( nInTable &&
+             ( eHoriRel == FRAME || eHoriRel == REL_CHAR ) &&
+             pFSPA->nwr == 3 &&
+             pRecord->nLayoutInTableCell == 0x80000000 )
+        {
+            eHoriRel = REL_PG_PRTAREA;
+        }
+        // <--
+
         SwFmtHoriOrient aHoriOri(MakeSafePositioningValue(pFSPA->nXaLeft),
             eHoriOri, eHoriRel);
         if( 4 <= nXAlign )
Index: sw/source/ui/app/docstyle.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/app/docstyle.cxx,v
retrieving revision 1.16
retrieving revision 1.16.38.1
diff -u -p -u -p -r1.16 -r1.16.38.1
--- sw/source/ui/app/docstyle.cxx	5 Jan 2005 11:48:18 -0000	1.16
+++ sw/source/ui/app/docstyle.cxx	24 Jan 2005 19:57:00 -0000	1.16.38.1
@@ -1760,7 +1760,7 @@ void SwDocStyleSheet::Create()
 		case SFX_STYLE_FAMILY_FRAME:
 			pFrmFmt = lcl_FindFrmFmt( rDoc, aName );
 			if( !pFrmFmt )
-				pFrmFmt = rDoc.MakeFrmFmt(aName, rDoc.GetDfltFrmFmt(), FALSE);
+				pFrmFmt = rDoc.MakeFrmFmt(aName, rDoc.GetDfltFrmFmt(), FALSE, FALSE);
 
 			break;
 
Index: sw/source/ui/config/mailconfigpage.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/config/mailconfigpage.cxx,v
retrieving revision 1.3
retrieving revision 1.3.232.1
diff -u -p -u -p -r1.3 -r1.3.232.1
--- sw/source/ui/config/mailconfigpage.cxx	29 Sep 2004 09:25:11 -0000	1.3
+++ sw/source/ui/config/mailconfigpage.cxx	21 Jan 2005 10:57:02 -0000	1.3.232.1
@@ -229,8 +229,7 @@ SwMailConfigPage::SwMailConfigPage( Wind
     m_aServerED( this, ResId(         ED_SERVER)),
     m_aPortFT( this, ResId(           FT_PORT)),
     m_aPortNF( this, ResId(           NF_PORT)),
-    m_aSecureFT( this, ResId(         FT_SECURE)),
-    m_aSecureLB( this, ResId(         LB_SECURE)),
+	m_aSecureCB( this, ResId(         CB_SECURE)),
     m_aServerAuthenticationPB( this, ResId( PB_AUTHENTICATION )),
     m_aSeparatorFL( this,            ResId( FL_SEPARATOR      )),
     m_aTestPB( this, ResId(           PB_TEST)),
@@ -276,8 +275,7 @@ BOOL SwMailConfigPage::FillItemSet( SfxI
     if(m_aPortNF.IsModified())
         m_pConfigItem->SetMailPort((sal_Int16)m_aPortNF.GetValue());
     
-    if(m_aSecureLB.GetSelectEntryPos() != m_aSecureLB.GetSavedValue())
-        m_pConfigItem->SetSecureConnection(m_aSecureLB.GetSelectEntryPos() == 1);
+    m_pConfigItem->SetSecureConnection(m_aSecureCB.IsChecked());
     
     m_pConfigItem->Commit();
     return sal_True;
@@ -297,7 +295,7 @@ void SwMailConfigPage::Reset( const SfxI
     m_aServerED.SetText(m_pConfigItem->GetMailServer());
     m_aPortNF.SetValue(m_pConfigItem->GetMailPort());
     
-    m_aSecureLB.SelectEntryPos(m_pConfigItem->IsSecureConnection() ? 1 : 0);
+    m_aSecureCB.Check(m_pConfigItem->IsSecureConnection());
     
     m_aDisplayNameED.SaveValue();
     m_aAddressED    .SaveValue();
@@ -305,7 +303,7 @@ void SwMailConfigPage::Reset( const SfxI
     m_aReplyToED    .SaveValue();
     m_aServerED     .SaveValue();
     m_aPortNF       .SaveValue();
-    m_aSecureLB     .SaveValue();
+    m_aSecureCB     .SaveValue();
 }
 /*-- 06.05.2004 10:59:41---------------------------------------------------
 
@@ -495,7 +493,7 @@ void SwTestAccountSettingsDialog::Test()
                         m_pParent->m_aServerED.GetText(), 
                         m_pParent->m_aPortNF.GetValue(), 
                         ::rtl::OUString::createFromAscii( 
-                                m_pParent->m_aSecureLB.GetSelectEntryPos() == 1 ? "Ssl" : "Insecure"));
+                                m_pParent->m_aSecureCB.IsChecked() ? "Ssl" : "Insecure"));
             xMailService->connect(xConnectionContext, xAuthenticator);  
             bIsLoggedIn = xMailService->isConnected();
             if( xInMailService.is() )
Index: sw/source/ui/config/mailconfigpage.hrc
===================================================================
RCS file: /cvs/sw/sw/source/ui/config/mailconfigpage.hrc,v
retrieving revision 1.2
retrieving revision 1.2.234.1
diff -u -p -u -p -r1.2 -r1.2.234.1
--- sw/source/ui/config/mailconfigpage.hrc	20 Sep 2004 13:08:12 -0000	1.2
+++ sw/source/ui/config/mailconfigpage.hrc	21 Jan 2005 10:57:03 -0000	1.2.234.1
@@ -74,8 +74,7 @@
 #define ED_SERVER                   11
 #define FT_PORT                     12
 #define NF_PORT                     13
-#define FT_SECURE                   14
-#define LB_SECURE                   15
+#define CB_SECURE                   14
 #define PB_AUTHENTICATION           16
 #define FL_SEPARATOR                17
 #define PB_TEST                     21
Index: sw/source/ui/config/mailconfigpage.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/config/mailconfigpage.src,v
retrieving revision 1.2
retrieving revision 1.2.234.2
diff -u -p -u -p -r1.2 -r1.2.234.2
--- sw/source/ui/config/mailconfigpage.src	20 Sep 2004 13:08:23 -0000	1.2
+++ sw/source/ui/config/mailconfigpage.src	26 Jan 2005 10:52:11 -0000	1.2.234.2
@@ -81,27 +81,27 @@ TabPage TP_MAILCONFIG
     FixedText       FT_DISPLAYNAME
     {
         Pos = MAP_APPFONT ( 12 , 16 ) ;
-        Size = MAP_APPFONT ( 63 , 8 ) ;
+        Size = MAP_APPFONT ( 74 , 8 ) ;
         Text [ de ] = "~Ihr Name" ;
         Text[ en-US ] = "~Your name";
     };
     Edit  ED_DISPLAYNAME
     {
-        Pos = MAP_APPFONT ( 78 , 14 ) ;
-        Size = MAP_APPFONT ( 170 , 12 ) ;
+        Pos = MAP_APPFONT ( 88 , 14 ) ;
+        Size = MAP_APPFONT ( 160 , 12 ) ;
         Border = TRUE;
     };
     FixedText       FT_ADDRESS
     {
         Pos = MAP_APPFONT ( 12 , 31 ) ;
-        Size = MAP_APPFONT ( 63 , 8 ) ;
+        Size = MAP_APPFONT ( 74 , 8 ) ;
         Text [ de ] = "~E-Mail-Adresse" ;
         Text[ en-US ] = "~E-Mail address";
     };
     Edit   ED_ADDRESS
     {
-        Pos = MAP_APPFONT ( 78 , 29 ) ;
-        Size = MAP_APPFONT ( 170 , 12 ) ;
+        Pos = MAP_APPFONT ( 88 , 29 ) ;
+        Size = MAP_APPFONT ( 160 , 12 ) ;
         Border = TRUE;
     };
     CheckBox        CB_REPLYTO
@@ -114,14 +114,14 @@ TabPage TP_MAILCONFIG
     FixedText       FT_REPLYTO
     {
         Pos = MAP_APPFONT ( 18 , 60 ) ;
-        Size = MAP_APPFONT ( 57 , 8 ) ;
+        Size = MAP_APPFONT ( 68 , 8 ) ;
         Text [ de ] = "Ant~wortadresse" ;
         Text[ en-US ] = "~Reply address";
     };
     Edit ED_REPLYTO
     {
-        Pos = MAP_APPFONT ( 78 , 58 ) ;
-        Size = MAP_APPFONT ( 170 , 12 ) ;
+        Pos = MAP_APPFONT ( 88 , 58 ) ;
+        Size = MAP_APPFONT ( 160 , 12 ) ;
         Border = TRUE;
     };
     FixedLine       FL_SMTP
@@ -134,14 +134,14 @@ TabPage TP_MAILCONFIG
     FixedText       FT_SERVER
     {
         Pos = MAP_APPFONT ( 12 , 89 ) ;
-        Size = MAP_APPFONT ( 63 , 8 ) ;
+        Size = MAP_APPFONT ( 74 , 8 ) ;
         Text [ de ] = "~Server-Name" ;
         Text[ en-US ] = "~Server name";
     };
     Edit ED_SERVER
     {
-        Pos = MAP_APPFONT ( 78 , 87 ) ;
-        Size = MAP_APPFONT ( 115 , 12 ) ;
+        Pos = MAP_APPFONT ( 88 , 87 ) ;
+        Size = MAP_APPFONT ( 105 , 12 ) ;
         Border = TRUE;
     };
     FixedText       FT_PORT
@@ -165,34 +165,17 @@ TabPage TP_MAILCONFIG
         Spin = FALSE ;
         TabStop = TRUE ;
     };
-    FixedText       FT_SECURE
+    CheckBox        CB_SECURE
     {
         Pos = MAP_APPFONT ( 12 , 104 ) ;
-        Size = MAP_APPFONT ( 63 , 8 ) ;
-        Text [ de ] = "~Verschlüsselte Verbindung verwenden" ;
-        Text[ en-US ] = "~Use secure connection";
-    };
-    ListBox         LB_SECURE
-    {
-        Pos = MAP_APPFONT ( 78 , 102 ) ;
-        Size = MAP_APPFONT ( 40 , 50 ) ;
-        DropDown = TRUE;
-        Border = TRUE;
-        StringList =
-        {
-            < "Keine" ; > ;
-            < "SSL" ; > ;
-        };
-        stringlist [ en-US ] =
-        {
-            < "None" ; > ;
-            < "SSL" ; > ;
-        };
+        Size = MAP_APPFONT ( 242 , 10 ) ;
+        Text [ de ] = "~Verschlüsselte Verbindung verwenden (SSL)" ;
+        Text[ en-US ] = "~Use secure connection (SSL)";
     };
     PushButton PB_AUTHENTICATION
     {
-        Pos = MAP_APPFONT ( 78 , 118 ) ;
-        Size = MAP_APPFONT ( 115 , 14 ) ;
+        Pos = MAP_APPFONT ( 88 , 118 ) ;
+        Size = MAP_APPFONT ( 105 , 14 ) ;
         Text [ de ] = "Server Authen~tifizierung" ;
         Text[ en-US ] = "Server Au~thentication";
     };
@@ -203,8 +186,8 @@ TabPage TP_MAILCONFIG
     };
     PushButton      PB_TEST
     {
-        Pos = MAP_APPFONT ( 78 , 164 ) ;
-        Size = MAP_APPFONT ( 115 , 14 ) ;
+        Pos = MAP_APPFONT ( 88 , 164 ) ;
+        Size = MAP_APPFONT ( 105 , 14 ) ;
         Text [ de ] = "Einstellungen ~Testen..." ;
         Text[ en-US ] = "Test S~ettings...";
     };
Index: sw/source/ui/config/optload.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/config/optload.src,v
retrieving revision 1.68
retrieving revision 1.68.234.1
diff -u -p -u -p -r1.68 -r1.68.234.1
--- sw/source/ui/config/optload.src	20 Sep 2004 12:37:38 -0000	1.68
+++ sw/source/ui/config/optload.src	20 Jan 2005 11:16:13 -0000	1.68.234.1
@@ -202,7 +202,7 @@ TabPage TP_OPTCAPTION_PAGE
 	FixedText FT_OBJECTS
 	{
 		Pos = MAP_APPFONT ( 6 , 6 ) ;
-		Size = MAP_APPFONT ( 121 , 20 ) ;
+        Size = MAP_APPFONT ( 106 , 20 ) ;
 		WordBreak = TRUE ;
 		Text [ de ] = "Automatische Beschriftung\nbeim Einfügen von:" ;
 		Text [ en-US ] = "Add captions automatically\nwhen inserting:" ;
@@ -211,7 +211,7 @@ TabPage TP_OPTCAPTION_PAGE
 	Control CLB_OBJECTS
 	{
 		Pos = MAP_APPFONT ( 6 , 29 ) ;
-		Size = MAP_APPFONT ( 121 , 150 ) ;
+        Size = MAP_APPFONT ( 106 , 150 ) ;
 		Border = TRUE ;
 		TabStop = TRUE ;
 	};
@@ -242,16 +242,16 @@ TabPage TP_OPTCAPTION_PAGE
 
     FixedLine FL_SETTINGS_2
 	{
-		Pos = MAP_APPFONT ( 133 , 3 ) ;
-        Size = MAP_APPFONT ( 121 , 8 ) ;
+        Pos = MAP_APPFONT ( 118 , 3 ) ;
+        Size = MAP_APPFONT ( 136 , 8 ) ;
 		Text [ de ] = "Beschriftung" ;
 		Text [ en-US ] = "Caption" ;
 		Text [ x-comment ] = " ";
 	};
 	FixedText TXT_CATEGORY
 	{
-		Pos = MAP_APPFONT ( 139 , 16 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 16 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Text [ de ] = "~Kategorie" ;
 		Text [ en-US ] = "Category" ;
 		Left = TRUE ;
@@ -267,8 +267,8 @@ TabPage TP_OPTCAPTION_PAGE
 	};
 	FixedText TXT_FORMAT
 	{
-		Pos = MAP_APPFONT ( 139 , 31 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 31 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Text [ de ] = "~Nummerierung" ;
 		Text [ en-US ] = "~Numbering" ;
 		Left = TRUE ;
@@ -284,8 +284,8 @@ TabPage TP_OPTCAPTION_PAGE
 	};
 	FixedText TXT_TEXT
 	{
-		Pos = MAP_APPFONT ( 139 , 45 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 45 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Text [ de ] = "~Separator" ;
 		Text [ en-US ] = "Separator" ;
 		Left = TRUE ;
@@ -301,8 +301,8 @@ TabPage TP_OPTCAPTION_PAGE
 	};
 	FixedText TXT_POS
 	{
-		Pos = MAP_APPFONT ( 139 , 60 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 60 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Left = TRUE ;
 		Text [ de ] = "~Position" ;
 		Text [ en-US ] = "Position" ;
@@ -319,16 +319,16 @@ TabPage TP_OPTCAPTION_PAGE
 
     FixedLine FL_NUMCAPT
 	{
-		Pos = MAP_APPFONT ( 133 , 74 ) ;
-        Size = MAP_APPFONT ( 121 , 8 ) ;
+        Pos = MAP_APPFONT ( 118 , 74 ) ;
+        Size = MAP_APPFONT ( 136 , 8 ) ;
 		Text [ de ] = "Kapitelweise Nummerierung der Beschriftung" ;
 		Text [ en-US ] = "Numbering captions by chapter" ;
 		Text [ x-comment ] = " ";
 	};
 	FixedText FT_LEVEL
 	{
-		Pos = MAP_APPFONT ( 139 , 87 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 87 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Text [ de ] = "E~bene" ;
 		Text [ en-US ] = "Level" ;
 		Text [ x-comment ] = " ";
@@ -355,8 +355,8 @@ TabPage TP_OPTCAPTION_PAGE
 	};
 	FixedText FT_SEPARATOR
 	{
-		Pos = MAP_APPFONT ( 139 , 102 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 102 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Text [ de ] = "~Trenner" ;
 		Text [ en-US ] = "Separator" ;
 		Text [ x-comment ] = " ";
@@ -371,16 +371,16 @@ TabPage TP_OPTCAPTION_PAGE
 	};
     FixedLine FL_CATEGORY
 	{
-		Pos = MAP_APPFONT ( 133 , 115 ) ;
-        Size = MAP_APPFONT ( 121 , 8 ) ;
+        Pos = MAP_APPFONT ( 118 , 115 ) ;
+        Size = MAP_APPFONT ( 136 , 8 ) ;
 		Text [ de ] = "Kategorie- und Rahmenformat" ;
 		Text [ en-US ] = "Category and frame format" ;
 		Text [ x-comment ] = " ";
 	};
 	FixedText FT_CHARSTYLE
 	{
-		Pos = MAP_APPFONT ( 139 , 128 ) ;
-		Size = MAP_APPFONT ( 44 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 128 ) ;
+        Size = MAP_APPFONT ( 59 , 10 ) ;
 		Text [ de ] = "Zeichenvorlage" ;
 		Text [ en-US ] = "Character style" ;
 		Text [ x-comment ] = " ";
@@ -407,16 +407,16 @@ TabPage TP_OPTCAPTION_PAGE
 	};
 	CheckBox CB_APPLYBORDER
 	{
-		Pos = MAP_APPFONT ( 139 , 143 ) ;
-		Size = MAP_APPFONT ( 109 , 10 ) ;
+        Pos = MAP_APPFONT ( 125 , 143 ) ;
+        Size = MAP_APPFONT ( 124 , 10 ) ;
 		Text [ de ] = "~Rahmen und Schatten anwenden" ;
 		Text [ en-US ] = "Apply border and shadow" ;
 		Text [ x-comment ] = " ";
 	};
 	Window WIN_PREVIEW
 	{
-		Pos = MAP_APPFONT ( 139 , 159 ) ;
-		Size = MAP_APPFONT ( 109 , 20 ) ;
+        Pos = MAP_APPFONT ( 125 , 159 ) ;
+        Size = MAP_APPFONT ( 123 , 20 ) ;
 		Border = TRUE;
 	};
 
Index: sw/source/ui/dbui/dbmgr.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/dbui/dbmgr.cxx,v
retrieving revision 1.87
retrieving revision 1.85.12.2
diff -u -p -u -p -r1.87 -r1.85.12.2
--- sw/source/ui/dbui/dbmgr.cxx	19 Jan 2005 15:11:04 -0000	1.87
+++ sw/source/ui/dbui/dbmgr.cxx	25 Jan 2005 07:33:54 -0000	1.85.12.2
@@ -2492,10 +2492,10 @@ String SwNewDBMgr::LoadAndRegisterDataSo
     String sFilterCSV(SW_RES(STR_FILTER_CSV));
 
     xFltMgr->appendFilter( sFilterAll, C2U("*") );
-    xFltMgr->appendFilter( sFilterAllData, C2U("*.sxc;*.dbf;*.xls;*.txt;*.csv"));
+    xFltMgr->appendFilter( sFilterAllData, C2U("*.ods;*.sxc;*.dbf;*.xls;*.txt;*.csv"));
             
     xFltMgr->appendFilter( sFilterSXB, C2U("*.odb") );
-    xFltMgr->appendFilter( sFilterSXC, C2U("*.sxc") );
+    xFltMgr->appendFilter( sFilterSXC, C2U("*.ods;*.sxc") );
     xFltMgr->appendFilter( sFilterDBF, C2U("*.dbf") );
     xFltMgr->appendFilter( sFilterXLS, C2U("*.xls") );
     xFltMgr->appendFilter( sFilterTXT, C2U("*.txt") );
@@ -2518,7 +2518,8 @@ String SwNewDBMgr::LoadAndRegisterDataSo
         {
             bStore = false;
         }
-        else if(sExt.EqualsAscii("sxc") || sExt.EqualsAscii("xls"))
+        else if(sExt.EqualsAscii("ods") || 
+                sExt.EqualsAscii("sxc") || sExt.EqualsAscii("xls"))
         {
             rtl::OUString sDBURL(C2U("sdbc:calc:"));
             sDBURL += aTempURL.GetMainURL(INetURLObject::NO_DECODE);
Index: sw/source/ui/dbui/dbui.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/dbui/dbui.src,v
retrieving revision 1.26
retrieving revision 1.25.108.2
diff -u -p -u -p -r1.26 -r1.25.108.2
--- sw/source/ui/dbui/dbui.src	18 Jan 2005 15:11:43 -0000	1.26
+++ sw/source/ui/dbui/dbui.src	25 Jan 2005 07:34:03 -0000	1.25.108.2
@@ -272,8 +272,8 @@ String STR_FILTER_SXB
 };
 String STR_FILTER_SXC
 {
-    Text [ de ] = "%PRODUCTNAME Calc(*.oxc)" ;
-    Text [ en-US ] = "%PRODUCTNAME Calc(*.oxc)" ;
+    Text [ de ] = "%PRODUCTNAME Calc(*.ods;*.sxc)" ;
+    Text [ en-US ] = "%PRODUCTNAME Calc(*.ods;*.sxc)" ;
 };
 String STR_FILTER_DBF
 {
Index: sw/source/ui/dbui/mmlayoutpage.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/dbui/mmlayoutpage.cxx,v
retrieving revision 1.4
retrieving revision 1.4.108.1
diff -u -p -u -p -r1.4 -r1.4.108.1
--- sw/source/ui/dbui/mmlayoutpage.cxx	16 Nov 2004 16:59:00 -0000	1.4
+++ sw/source/ui/dbui/mmlayoutpage.cxx	20 Jan 2005 16:26:43 -0000	1.4.108.1
@@ -258,7 +258,7 @@ SwMailMergeLayoutPage::SwMailMergeLayout
     m_aUpPB.SetClickHdl(aUpDownHdl);
     m_aDownPB.SetClickHdl(aUpDownHdl);
     m_aAlignToBodyCB.SetClickHdl(LINK(this, SwMailMergeLayoutPage, AlignToTextHdl_Impl));
-
+	m_aAlignToBodyCB.Check();
 }    
 /*-- 15.04.2004 08:17:11---------------------------------------------------
 
Index: sw/source/ui/dialog/SwSpellDialogChildWindow.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/dialog/SwSpellDialogChildWindow.cxx,v
retrieving revision 1.2
retrieving revision 1.2.234.1
diff -u -p -u -p -r1.2 -r1.2.234.1
--- sw/source/ui/dialog/SwSpellDialogChildWindow.cxx	17 Sep 2004 14:01:49 -0000	1.2
+++ sw/source/ui/dialog/SwSpellDialogChildWindow.cxx	20 Jan 2005 13:30:46 -0000	1.2.234.1
@@ -166,6 +166,7 @@ struct SpellState
     bool                m_bBodySpelled;  //body already spelled
     bool                m_bOtherSpelled; //frames, footnotes, headers and footers spelled
     bool                m_bStartedInOther; //started the spelling insided of the _other_ area
+    bool                m_bStartedInSelection; // there was an initial text selection
     SwPaM*              pOtherCursor; // position where the spelling inside the _other_ area started
     bool                m_bDrawingsSpelled; //all drawings spelled 
     Reference<XTextRange> m_xStartRange; //text range that marks the start of spelling
@@ -204,6 +205,7 @@ struct SpellState
         m_bRestartDrawing(false),
         m_bTextObjectsCollected(false),
         m_bStartedInOther(false),
+        m_bStartedInSelection(false),
         pOtherCursor(0)
         {}
 
@@ -309,6 +311,8 @@ svx::SpellPortions SwSpellDialogChildWin
                 //set cursor to the start of the sentence
                 if(!pWrtShell->HasSelection())
                     pWrtShell->GoStartSentence();
+                else
+                    m_pSpellState->m_bStartedInSelection = true;
                 //determine if the selection is outside of the body text
                 bOtherText = !(pWrtShell->GetFrmType(0,sal_True) & FRMTYPE_BODY);
                 m_pSpellState->m_SpellStartPosition = bOtherText ? SPELL_START_OTHER : SPELL_START_BODY;
@@ -444,7 +448,7 @@ svx::SpellPortions SwSpellDialogChildWin
         // if the spelling started inside of the body
         // 
         bool bCloseMessage = true;
-        if(!aRet.size())
+        if(!aRet.size() && !m_pSpellState->m_bStartedInSelection)
         {
             DBG_ASSERT(m_pSpellState->m_bDrawingsSpelled && 
                         m_pSpellState->m_bOtherSpelled && m_pSpellState->m_bBodySpelled,
Index: sw/source/ui/dialog/ascfldlg.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/dialog/ascfldlg.src,v
retrieving revision 1.23
retrieving revision 1.23.348.1
diff -u -p -u -p -r1.23 -r1.23.348.1
--- sw/source/ui/dialog/ascfldlg.src	26 Jun 2004 03:23:32 -0000	1.23
+++ sw/source/ui/dialog/ascfldlg.src	21 Jan 2005 16:04:43 -0000	1.23.348.1
@@ -81,7 +81,7 @@ ModalDialog DLG_ASCII_FILTER
 	FixedText FT_CHARSET
 	{
 		Pos = MAP_APPFONT ( 12 , 15 ) ;
-		Size = MAP_APPFONT ( 53 , 10 ) ;
+        Size = MAP_APPFONT ( 66 , 10 ) ;
 		Text [ de ] = "Z~eichensatz" ;
 		Text [ en-US ] = "~Character set";
 	};
@@ -96,7 +96,7 @@ ModalDialog DLG_ASCII_FILTER
 	FixedText FT_FONT
 	{
 		Pos = MAP_APPFONT ( 12 , 31 ) ;
-		Size = MAP_APPFONT ( 53 , 10 ) ;
+        Size = MAP_APPFONT ( 66 , 10 ) ;
 		Text [ de ] = "~Grundschrift" ;
 		Text [ en-US ] = "Default fonts";
 	};
@@ -111,7 +111,7 @@ ModalDialog DLG_ASCII_FILTER
 	FixedText FT_LANGUAGE
 	{
 		Pos = MAP_APPFONT ( 12 , 47 ) ;
-		Size = MAP_APPFONT ( 53 , 10 ) ;
+        Size = MAP_APPFONT ( 66 , 10 ) ;
 		Text [ de ] = "~Sprache" ;
 		Text [ en-US ] = "Lan~guage";
 		Text [ x-comment ] = " ";
@@ -127,7 +127,7 @@ ModalDialog DLG_ASCII_FILTER
 	FixedText FT_CRLF
 	{
         Pos = MAP_APPFONT ( 12 , 64 ) ;
-        Size = MAP_APPFONT ( 53 , 8 ) ;
+        Size = MAP_APPFONT ( 66 , 8 ) ;
 		Text [ de ] = "~Absatzumbruch" ;
 		Text [ en-US ] = "~Paragraph break";
 		Text [ x-comment ] = " ";
Index: sw/source/ui/inc/mailconfigpage.hxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/inc/mailconfigpage.hxx,v
retrieving revision 1.2
retrieving revision 1.2.234.1
diff -u -p -u -p -r1.2 -r1.2.234.1
--- sw/source/ui/inc/mailconfigpage.hxx	20 Sep 2004 13:22:05 -0000	1.2
+++ sw/source/ui/inc/mailconfigpage.hxx	21 Jan 2005 10:57:07 -0000	1.2.234.1
@@ -107,8 +107,7 @@ class SwMailConfigPage : public SfxTabPa
     FixedText       m_aPortFT;
     NumericField    m_aPortNF;
 
-    FixedText       m_aSecureFT;
-    ListBox         m_aSecureLB;
+	CheckBox		m_aSecureCB;
 
     PushButton      m_aServerAuthenticationPB;
     
Index: sw/source/ui/inc/view.hxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/inc/view.hxx,v
retrieving revision 1.31
retrieving revision 1.31.212.1
diff -u -p -u -p -r1.31 -r1.31.212.1
--- sw/source/ui/inc/view.hxx	12 Oct 2004 13:23:52 -0000	1.31
+++ sw/source/ui/inc/view.hxx	24 Jan 2005 08:59:51 -0000	1.31.212.1
@@ -321,7 +321,8 @@ class SW_DLLPUBLIC SwView: public SfxVie
 					bPasteState : 1,
                     bPasteSpecialState : 1,
                     bInMailMerge : 1,
-                    bInDtor : 1 //detect destructor to prevent creating of sub shells while closing
+                    bInDtor : 1, //detect destructor to prevent creating of sub shells while closing
+                    bOldShellWasPagePreView
 					;
 
 	// Methoden fuers Suchen
Index: sw/source/ui/misc/num.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/misc/num.src,v
retrieving revision 1.33
retrieving revision 1.33.282.1
diff -u -p -u -p -r1.33 -r1.33.282.1
--- sw/source/ui/misc/num.src	2 Aug 2004 13:07:13 -0000	1.33
+++ sw/source/ui/misc/num.src	21 Jan 2005 12:16:15 -0000	1.33.282.1
@@ -108,22 +108,14 @@ TabPage TP_NUM_POSITION
 	FixedText FT_BORDERDIST
 	{
         Pos = MAP_APPFONT ( 53 , 16 ) ;
-        Size = MAP_APPFONT ( 102 , 10 ) ;
+        Size = MAP_APPFONT ( 132 , 10 ) ;
 		Text [ de ] = "Ein~rückung";
 		Text [ en-US ] = "Indent" ;
 		Text [ x-comment ] = " ";
 	};
-	CheckBox CB_RELATIVE
+    MetricField MF_BORDERDIST
 	{
-		Pos = MAP_APPFONT ( 201 , 16 ) ;
-        Size = MAP_APPFONT ( 53 , 10 ) ;
-		Text [ de ] = "Relati~v" ;
-		Text [ en-US ] = "Relati~ve" ;
-		Text [ x-comment ] = " ";
-	};
-	MetricField MF_BORDERDIST
-	{
-		Pos = MAP_APPFONT ( 158 , 14 ) ;
+        Pos = MAP_APPFONT ( 188 , 14 ) ;
 		Size = MAP_APPFONT ( 40 , 12 ) ;
 		Border = TRUE ;
 		TabStop = TRUE ;
@@ -139,17 +131,25 @@ TabPage TP_NUM_POSITION
 		Last = 1999 ;
 		SpinSize = 5 ;
 	};
-	FixedText FT_INDENT
+    CheckBox CB_RELATIVE
+    {
+        Pos = MAP_APPFONT ( 188 , 29 ) ;
+        Size = MAP_APPFONT ( 53 , 10 ) ;
+        Text [ de ] = "Relati~v" ;
+        Text [ en-US ] = "Relati~ve" ;
+        Text [ x-comment ] = " ";
+    };
+    FixedText FT_INDENT
 	{
-        Pos = MAP_APPFONT ( 53 , 32 ) ;
-        Size = MAP_APPFONT ( 102 , 10 ) ;
+        Pos = MAP_APPFONT ( 53 , 44 ) ;
+        Size = MAP_APPFONT ( 132 , 10 ) ;
 		Text [ de ] = "Abstand zum Te~xt" ;
 		Text [ en-US ] = "Space to te~xt" ;
 		Text [ x-comment ] = " ";
 	};
 	MetricField MF_INDENT
 	{
-		Pos = MAP_APPFONT ( 158 , 30 ) ;
+        Pos = MAP_APPFONT ( 188 , 42 ) ;
 		Size = MAP_APPFONT ( 40 , 12 ) ;
 		Border = TRUE ;
 		TabStop = TRUE ;
@@ -167,15 +167,15 @@ TabPage TP_NUM_POSITION
 	};
 	FixedText FT_NUMDIST
 	{
-        Pos = MAP_APPFONT ( 53 , 48 ) ;
-        Size = MAP_APPFONT ( 102 , 10 ) ;
+        Pos = MAP_APPFONT ( 53 , 60 ) ;
+        Size = MAP_APPFONT ( 132 , 10 ) ;
 		Text [ de ] = "~Mindestabstand Nummer <-> Text" ;
 		Text [ en-US ] = "~Minimum space numbering <-> text" ;
 		Text [ x-comment ] = " ";
 	};
 	MetricField MF_NUMDIST
 	{
-		Pos = MAP_APPFONT ( 158 , 46 ) ;
+        Pos = MAP_APPFONT ( 188 , 58 ) ;
 		Size = MAP_APPFONT ( 40 , 12 ) ;
 		Left = TRUE ;
 		Repeat = TRUE ;
@@ -193,16 +193,16 @@ TabPage TP_NUM_POSITION
 	};
 	FixedText FT_ALIGN
 	{
-        Pos = MAP_APPFONT ( 53 , 64 ) ;
-        Size = MAP_APPFONT ( 102 , 10 ) ;
+        Pos = MAP_APPFONT ( 53 , 76 ) ;
+        Size = MAP_APPFONT ( 132 , 10 ) ;
 		Text [ de ] = "Ausri~chtung der Nummerierung" ;
 		Text [ en-US ] = "Numbering ~alignment" ;
 		Text [ x-comment ] = " ";
 	};
 	ListBox LB_ALIGN
 	{
-		Pos = MAP_APPFONT ( 158 , 62 ) ;
-		Size = MAP_APPFONT ( 40 , 40 ) ;
+        Pos = MAP_APPFONT ( 188 , 74 ) ;
+        Size = MAP_APPFONT ( 64 , 40 ) ;
 		TabStop = TRUE ;
 		DropDown = TRUE ;
 		StringList [ de ] =
@@ -226,7 +226,7 @@ TabPage TP_NUM_POSITION
 	};
 	PushButton PB_STANDARD
 	{
-		Pos = MAP_APPFONT ( 201 , 78 ) ;
+        Pos = MAP_APPFONT ( 53 , 90 ) ;
 		Size = MAP_APPFONT ( 50 , 14 ) ;
 		Text [ de ] = "~Standard" ;
 		Text [ en-US ] = "~Default" ;
@@ -234,8 +234,8 @@ TabPage TP_NUM_POSITION
 	};
 	Window WIN_PREVIEW
 	{
-        Pos = MAP_APPFONT ( 53 , 99 ) ;
-        Size = MAP_APPFONT ( 198 , 78 ) ;
+        Pos = MAP_APPFONT ( 53 , 111 ) ;
+        Size = MAP_APPFONT ( 198 , 66 ) ;
 	};
 };
 
Index: sw/source/ui/uiview/view.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/uiview/view.cxx,v
retrieving revision 1.81
retrieving revision 1.80.106.2
diff -u -p -u -p -r1.81 -r1.80.106.2
--- sw/source/ui/uiview/view.cxx	18 Jan 2005 15:26:55 -0000	1.81
+++ sw/source/ui/uiview/view.cxx	25 Jan 2005 07:35:17 -0000	1.80.106.2
@@ -1,4 +1,4 @@
-/*************************************************************************
+ /*************************************************************************
  *
  *  $RCSfile$
  *
@@ -921,6 +921,7 @@ SwView::SwView( SfxViewFrame *pFrame, Sf
     nNewPage(USHRT_MAX),
     bInMailMerge(FALSE),
     bInDtor(FALSE),
+    bOldShellWasPagePreView(FALSE),
     pNumRuleNodeFromDoc(0) // #i23726#
 {
     // OD 18.12.2002 #103492# - According to discussion with MBA and further
@@ -966,8 +967,7 @@ SwView::SwView( SfxViewFrame *pFrame, Sf
     aUsrPref.SetOnlineSpell( aLinguOpt.bIsSpellAuto );
     aUsrPref.SetHideSpell( aLinguOpt.bIsSpellHideMarkings );
 
-	sal_Bool bOldShellWasPagePreView = FALSE,
-			 bOldShellWasSrcView = FALSE;
+	sal_Bool bOldShellWasSrcView = FALSE;
 
     // OD 18.12.2002 #103492# - determine, if there is an existing view for
     // document
@@ -1346,26 +1346,34 @@ void SwView::ReadUserData( const String 
             // OD 11.02.2003 #100556# - set flag value to avoid macro execution.
             bool bSavedFlagValue = pWrtShell->IsMacroExecAllowed();
             pWrtShell->SetMacroExecAllowed( false );
-/*!!! pb (11.08.2004): #i32536#
-            pWrtShell->SwCrsrShell::SetCrsr( aCrsrPos, !bSelectObj );
-			if( bSelectObj )
-			{
-				pWrtShell->SelectObj( aCrsrPos );
-				pWrtShell->EnterSelFrmMode( &aCrsrPos );
-			}
-*/
+//!!! pb (11.08.2004): #i32536#
+// os: changed: The user data has to be read if the view is switched back from page preview 
+            if(bOldShellWasPagePreView)
+            {
+                pWrtShell->SwCrsrShell::SetCrsr( aCrsrPos, !bSelectObj );
+            	if( bSelectObj )
+            	{
+            		pWrtShell->SelectObj( aCrsrPos );
+            		pWrtShell->EnterSelFrmMode( &aCrsrPos );
+            	}
+            }
+
             // OD 11.02.2003 #100556# - reset flag value
             pWrtShell->SetMacroExecAllowed( bSavedFlagValue );
 
             // OD 08.04.2003 #108693# - set visible area before applying
             // information from print preview. Otherwise, the applied information
             // is lost.
-/*!!! pb (11.08.2004): #i32536#
-            if ( bBrowse )
-                SetVisArea( aVis.TopLeft() );
-            else
-                SetVisArea( aVis );
-*/
+//!!! pb (11.08.2004): #i32536#
+// os: changed: The user data has to be read if the view is switched back from page preview 
+            if(bOldShellWasPagePreView)
+            {
+                if ( bBrowse )
+                    SetVisArea( aVis.TopLeft() );
+                else
+                    SetVisArea( aVis );
+            }
+
             //apply information from print preview - if available
 			if( sNewCrsrPos.Len() )
 			{
@@ -1373,14 +1381,14 @@ void SwView::ReadUserData( const String 
 				 	 nY = sNewCrsrPos.GetToken( 1, ';' ).ToInt32();
                 Point aCrsrPos( nX, nY );
                 bSelectObj = pWrtShell->IsObjSelectable( aCrsrPos );
-/*!!! pb (11.08.2004): #i32536#
+
                 pWrtShell->SwCrsrShell::SetCrsr( aCrsrPos, FALSE );
                 if( bSelectObj )
                 {
                     pWrtShell->SelectObj( aCrsrPos );
                     pWrtShell->EnterSelFrmMode( &aCrsrPos );
                 }
-*/
+                pWrtShell->MakeSelVisible();
                 sNewCrsrPos.Erase();
 			}
             else if(USHRT_MAX != nNewPage)
@@ -1507,14 +1515,18 @@ void SwView::ReadUserDataSequence ( cons
                     // OD 11.02.2003 #100556# - set flag value to avoid macro execution.
                     bool bSavedFlagValue = pWrtShell->IsMacroExecAllowed();
                     pWrtShell->SetMacroExecAllowed( false );
-/*!!! pb (11.08.2004): #i32536#
-					pWrtShell->SwCrsrShell::SetCrsr( aCrsrPos, !bSelectObj );
-					if( bSelectObj )
-					{
-						pWrtShell->SelectObj( aCrsrPos );
-						pWrtShell->EnterSelFrmMode( &aCrsrPos );
-					}
-*/
+//!!! pb (11.08.2004): #i32536#
+// os: changed: The user data has to be read if the view is switched back from page preview 
+                    if(bOldShellWasPagePreView)
+                    {
+					    pWrtShell->SwCrsrShell::SetCrsr( aCrsrPos, !bSelectObj );
+					    if( bSelectObj )
+    					{
+						    pWrtShell->SelectObj( aCrsrPos );
+						    pWrtShell->EnterSelFrmMode( &aCrsrPos );
+    					}
+    				}
+
                     // OD 11.02.2003 #100556# - reset flag value
                     pWrtShell->SetMacroExecAllowed( bSavedFlagValue );
 				}
@@ -1525,19 +1537,22 @@ void SwView::ReadUserDataSequence ( cons
 				if ( bGotZoomType && bGotZoomFactor &&
 				   ( pVOpt->GetZoom() != nZoomFactor || pVOpt->GetZoomType() != eZoom ) )
 					SetZoom( eZoom, nZoomFactor, sal_True );
-/*!!! pb (11.08.2004): #i32536#
-				if ( bBrowse && bGotVisibleLeft && bGotVisibleTop )
+//!!! pb (11.08.2004): #i32536#
+// os: changed: The user data has to be read if the view is switched back from page preview 
+                if(bOldShellWasPagePreView)
                 {
-                    Point aTopLeft(aVis.TopLeft());
-                    //check if the values are possible
-                    long nXMax = pHScrollbar->GetRangeMax() - pHScrollbar->GetVisibleSize();
-                    if( aTopLeft.X() > nXMax )
-                        aTopLeft.X() = nXMax < 0 ? 0 : nXMax;
-                    SetVisArea( aTopLeft );
-                }
-				else if (bGotVisibleLeft && bGotVisibleTop && bGotVisibleRight && bGotVisibleBottom )
-					SetVisArea( aVis );
-*/
+    				if ( bBrowse && bGotVisibleLeft && bGotVisibleTop )
+                    {
+                        Point aTopLeft(aVis.TopLeft());
+                        //check if the values are possible
+                        long nXMax = pHScrollbar->GetRangeMax() - pHScrollbar->GetVisibleSize();
+                        if( aTopLeft.X() > nXMax )
+                            aTopLeft.X() = nXMax < 0 ? 0 : nXMax;
+                        SetVisArea( aTopLeft );
+                    }
+				    else if (bGotVisibleLeft && bGotVisibleTop && bGotVisibleRight && bGotVisibleBottom )
+    					SetVisArea( aVis );
+	    	    }
 
 				pWrtShell->LockView( sal_True );
 				pWrtShell->EndAction();
Index: svx/source/dialog/SpellDialog.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/SpellDialog.cxx,v
retrieving revision 1.3
retrieving revision 1.3.308.1
diff -u -p -u -p -r1.3 -r1.3.308.1
--- svx/source/dialog/SpellDialog.cxx	27 Sep 2004 12:06:10 -0000	1.3
+++ svx/source/dialog/SpellDialog.cxx	20 Jan 2005 08:52:18 -0000	1.3.308.1
@@ -912,7 +912,12 @@ IMPL_LINK(SpellDialog, ModifyHdl, Senten
   -----------------------------------------------------------------------*/
 IMPL_LINK(SpellDialog, CancelHdl, Button *, pButton )
 {
-	Close();
+    //apply changes first - if there are any
+    if(aSentenceED.IsModified())
+    {
+        rParent.ApplyChangedSentence(aSentenceED.CreateSpellPortions());
+    }
+    Close();
 	return 0;
 }
 /*-------------------------------------------------------------------------
Index: svx/source/dialog/optjava.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/optjava.cxx,v
retrieving revision 1.6
retrieving revision 1.6.110.1
diff -u -p -u -p -r1.6 -r1.6.110.1
--- svx/source/dialog/optjava.cxx	10 Dec 2004 17:26:00 -0000	1.6
+++ svx/source/dialog/optjava.cxx	24 Jan 2005 09:16:55 -0000	1.6.110.1
@@ -261,6 +261,31 @@ SvxJavaOptionsPage::SvxJavaOptionsPage( 
 
 	EnableHdl_Impl( &m_aJavaEnableCB );
 	jfw_lock();
+	
+	//check if the text fits into the class path button
+    Size aButtonSize = m_aClassPathBtn.GetOutputSizePixel();
+    sal_Int32 nTextWidth = m_aClassPathBtn.GetTextWidth(m_aClassPathBtn.GetText());
+    //add some additional space
+    sal_Int32 nDiff = nTextWidth + 4 - aButtonSize.Width();
+    if( nDiff > 0)
+    {
+        Point aPos(m_aClassPathBtn.GetPosPixel());   
+        aPos.X() -= nDiff;
+        aButtonSize.Width() += nDiff;
+        m_aClassPathBtn.SetPosSizePixel(aPos, aButtonSize);
+        aPos = m_aAddBtn.GetPosPixel();
+        aPos.X() -= nDiff;
+        m_aAddBtn.SetPosSizePixel(aPos, aButtonSize);   
+        aPos = m_aParameterBtn.GetPosPixel();
+        aPos.X() -= nDiff;
+        m_aParameterBtn.SetPosSizePixel(aPos, aButtonSize);
+        Size aSize = m_aJavaList.GetSizePixel();
+        aSize.Width() -= nDiff;
+        m_aJavaList.SetSizePixel(aSize);   
+        
+    }            
+	
+	 
 }
 
 // -----------------------------------------------------------------------
Index: svx/source/dialog/tabstpge.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/tabstpge.cxx,v
retrieving revision 1.16
retrieving revision 1.16.494.1
diff -u -p -u -p -r1.16 -r1.16.494.1
--- svx/source/dialog/tabstpge.cxx	25 May 2004 11:28:56 -0000	1.16
+++ svx/source/dialog/tabstpge.cxx	21 Jan 2005 11:20:51 -0000	1.16.494.1
@@ -173,7 +173,6 @@ SvxTabulatorTabPage::SvxTabulatorTabPage
 	aDezChar        ( this, ResId( ED_TABTYPE_DECCHAR ) ),
 	aDezCharLabel   ( this, ResId( FT_TABTYPE_DECCHAR ) ),
     aTabTypeLabel   ( this, ResId( FL_TABTYPE ) ),
-    aTabTypeLabelVert ( this, ResId( FL_TABTYPE_VERT ) ),
 	aNoFillChar     ( this, ResId( BTN_FILLCHAR_NO ) ),
 	aFillPoints     ( this, ResId( BTN_FILLCHAR_POINTS ) ),
 	aFillDashLine   ( this, ResId( BTN_FILLCHAR_DASHLINE ) ),
Index: svx/source/dialog/tabstpge.hrc
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/tabstpge.hrc,v
retrieving revision 1.3
retrieving revision 1.3.1230.1
diff -u -p -u -p -r1.3 -r1.3.1230.1
--- svx/source/dialog/tabstpge.hrc	27 May 2002 10:44:15 -0000	1.3
+++ svx/source/dialog/tabstpge.hrc	21 Jan 2005 11:20:53 -0000	1.3.1230.1
@@ -78,7 +78,6 @@
 #define	ED_TABTYPE_DECCHAR		28
 #define FT_TABTYPE_DECCHAR		29
 #define FL_TABTYPE                      30
-#define FL_TABTYPE_VERT                 31
 
 
 #define	BTN_FILLCHAR_NO 		40
Index: svx/source/dialog/tabstpge.hxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/tabstpge.hxx,v
retrieving revision 1.2
retrieving revision 1.2.502.1
diff -u -p -u -p -r1.2 -r1.2.502.1
--- svx/source/dialog/tabstpge.hxx	10 May 2004 16:54:58 -0000	1.2
+++ svx/source/dialog/tabstpge.hxx	21 Jan 2005 11:20:55 -0000	1.2.502.1
@@ -143,7 +143,6 @@ private:
 	FixedText		aDezCharLabel;
 	Edit			aDezChar;
     FixedLine       aTabTypeLabel;
-    FixedLine       aTabTypeLabelVert;
 
 	// Fuellzeichen
 	RadioButton		aNoFillChar;
Index: svx/source/dialog/tabstpge.src
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/tabstpge.src,v
retrieving revision 1.25
retrieving revision 1.25.442.1
diff -u -p -u -p -r1.25 -r1.25.442.1
--- svx/source/dialog/tabstpge.src	25 Jun 2004 12:04:23 -0000	1.25
+++ svx/source/dialog/tabstpge.src	21 Jan 2005 11:20:57 -0000	1.25.442.1
@@ -72,33 +72,41 @@ TabPage RID_SVXPAGE_TABULATOR
 	Text [ de ] = "Tabulatoren" ;
 	Text [ en-US ] = "Tabs" ;
 	Size = MAP_APPFONT ( 260 , 185 ) ;
-	MetricBox ED_TABPOS
+    FixedLine FL_TABPOS
+    {
+        Pos = MAP_APPFONT ( 6 , 3 ) ;
+        Size = MAP_APPFONT ( 48 , 8 ) ;
+        Text [ de ] = "~Position" ;
+        Text [ en-US ] = "Position" ;
+        Text [ x-comment ] = " ";
+    };
+    MetricBox ED_TABPOS
 	{
 		Pos = MAP_APPFONT ( 12 , 14 ) ;
-        Size = MAP_APPFONT ( 39 , 95 ) ;
+        Size = MAP_APPFONT ( 39 , 165 ) ;
 		Minimum = -5600 ;
 		Maximum = 5600 ;
 		DecimalDigits = 2 ;
 		Unit = FUNIT_CM ;
 	};
-    FixedLine FL_TABPOS
-	{
-		Pos = MAP_APPFONT ( 6 , 3 ) ;
-        Size = MAP_APPFONT ( 48 , 8 ) ;
-        Text [ de ] = "~Position" ;
-        Text [ en-US ] = "Position" ;
-		Text [ x-comment ] = " ";
-	};
     FixedLine FL_TABPOS_VERT
 	{
         Pos = MAP_APPFONT ( 55 , 14 ) ;
-        Size = MAP_APPFONT ( 4 , 95 ) ;
+        Size = MAP_APPFONT ( 4 , 165 ) ;
         Vert = TRUE;
 	};
+    FixedLine FL_TABTYPE
+    {
+        Pos = MAP_APPFONT ( 60 , 3 ) ;
+        Size = MAP_APPFONT ( 138 , 8 ) ;
+        Text [ de ] = "Typ" ;
+        Text [ en-US ] = "Type" ;
+        Text [ x-comment ] = " ";
+    };
     RadioButton BTN_TABTYPE_LEFT
 	{
         Pos = MAP_APPFONT ( 66 , 14 ) ;
-        Size = MAP_APPFONT ( 49 , 10 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text [ de ] = "~Links" ;
 		Text [ en-US ] = "~Left" ;
 		Check = TRUE ;
@@ -106,52 +114,52 @@ TabPage RID_SVXPAGE_TABULATOR
 	};
 	Window WIN_TABLEFT
 	{
-        Pos = MAP_APPFONT ( 115 , 14 ) ;
+        Pos = MAP_APPFONT ( 157 , 14 ) ;
 		Size = MAP_APPFONT ( 7 , 10 ) ;
 	};
 	RadioButton BTN_TABTYPE_RIGHT
 	{
         Pos = MAP_APPFONT ( 66 , 28 ) ;
-        Size = MAP_APPFONT ( 49 , 10 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text [ de ] = "~Rechts" ;
         Text [ en-US ] = "Righ~t" ;
 		Text [ x-comment ] = " ";
 	};
 	Window WIN_TABRIGHT
 	{
-        Pos = MAP_APPFONT ( 115 , 28 ) ;
+        Pos = MAP_APPFONT ( 157 , 28 ) ;
 		Size = MAP_APPFONT ( 7 , 10 ) ;
 	};
 	RadioButton BTN_TABTYPE_CENTER
 	{
         Pos = MAP_APPFONT ( 66 , 42 ) ;
-        Size = MAP_APPFONT ( 49 , 10 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
         Text [ de ] = "~Zentriert" ;
         Text [ en-US ] = "C~entered" ;
 		Text [ x-comment ] = " ";
 	};
 	Window WIN_TABCENTER
 	{
-        Pos = MAP_APPFONT ( 115 , 42 ) ;
+        Pos = MAP_APPFONT ( 157 , 42 ) ;
 		Size = MAP_APPFONT ( 7 , 10 ) ;
 	};
 	RadioButton BTN_TABTYPE_DECIMAL
 	{
         Pos = MAP_APPFONT ( 66 , 56 ) ;
-        Size = MAP_APPFONT ( 49 , 10 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text [ de ] = "~Dezimal" ;
         Text [ en-US ] = "Deci~mal" ;
 		Text [ x-comment ] = " ";
 	};
 	Window WIN_TABDECIMAL
 	{
-        Pos = MAP_APPFONT ( 115 , 56 ) ;
+        Pos = MAP_APPFONT ( 157 , 56 ) ;
 		Size = MAP_APPFONT ( 7 , 10 ) ;
 	};
 	FixedText FT_TABTYPE_DECCHAR
 	{
         Pos = MAP_APPFONT ( 75 , 70 ) ;
-        Size = MAP_APPFONT ( 39 , 8 ) ;
+        Size = MAP_APPFONT ( 79 , 8 ) ;
         Text [ de ] = "Z~eichen" ;
 		Text [ en-US ] = "~Character" ;
 		Text [ x-comment ] = " ";
@@ -159,28 +167,22 @@ TabPage RID_SVXPAGE_TABULATOR
 	Edit ED_TABTYPE_DECCHAR
 	{
 		Border = TRUE ;
-        Pos = MAP_APPFONT ( 114 , 68 ) ;
+        Pos = MAP_APPFONT ( 156 , 68 ) ;
 		Size = MAP_APPFONT ( 9 , 12 ) ;
 		Left = TRUE ;
 	};
-    FixedLine FL_TABTYPE
-	{
-        Pos = MAP_APPFONT ( 60 , 3 ) ;
-        Size = MAP_APPFONT ( 66 , 8 ) ;
-		Text [ de ] = "Typ" ;
-		Text [ en-US ] = "Type" ;
-		Text [ x-comment ] = " ";
-	};
-    FixedLine FL_TABTYPE_VERT
-	{
-        Pos = MAP_APPFONT ( 127 , 14 ) ;
-        Size = MAP_APPFONT ( 4 , 95 ) ;
-        Vert = TRUE;
+    FixedLine FL_FILLCHAR
+    {
+        Pos = MAP_APPFONT ( 60 , 86 ) ;
+        Size = MAP_APPFONT ( 138 , 8 ) ;
+        Text [ de ] = "Füllzeichen" ;
+        Text [ en-US ] = "Fill character" ;
+        Text [ x-comment ] = " ";
     };
     RadioButton BTN_FILLCHAR_NO
 	{
-        Pos = MAP_APPFONT ( 138 , 14 ) ;
-        Size = MAP_APPFONT ( 57 , 10 ) ;
+        Pos = MAP_APPFONT ( 66 , 97 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text [ de ] = "~Keine" ;
         Text [ en-US ] = "N~one" ;
 		Check = TRUE ;
@@ -188,26 +190,26 @@ TabPage RID_SVXPAGE_TABULATOR
 	};
 	RadioButton BTN_FILLCHAR_POINTS
 	{
-        Pos = MAP_APPFONT ( 138 , 28 ) ;
-        Size = MAP_APPFONT ( 57 , 10 ) ;
+        Pos = MAP_APPFONT ( 66 , 111 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text = "~........" ;
 	};
 	RadioButton BTN_FILLCHAR_DASHLINE
 	{
-        Pos = MAP_APPFONT ( 138 , 42 ) ;
-        Size = MAP_APPFONT ( 57 , 10 ) ;
+        Pos = MAP_APPFONT ( 66 , 125 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text = "~--------" ;
 	};
 	RadioButton BTN_FILLCHAR_UNDERSCORE
 	{
-        Pos = MAP_APPFONT ( 138 , 56 ) ;
-        Size = MAP_APPFONT ( 57 , 10 ) ;
+        Pos = MAP_APPFONT ( 66 , 139 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text = "~_____" ;
 	};
 	RadioButton BTN_FILLCHAR_OTHER
 	{
-        Pos = MAP_APPFONT ( 138 , 70 ) ;
-        Size = MAP_APPFONT ( 48 , 10 ) ;
+        Pos = MAP_APPFONT ( 66 , 153 ) ;
+        Size = MAP_APPFONT ( 89 , 10 ) ;
 		Text [ de ] = "Ze~ichen" ;
         Text [ en-US ] = "Character" ;
 		Text [ x-comment ] = " ";
@@ -215,19 +217,11 @@ TabPage RID_SVXPAGE_TABULATOR
 	Edit ED_FILLCHAR_OTHER
 	{
 		Border = TRUE ;
-        Pos = MAP_APPFONT ( 186 , 68 ) ;
+        Pos = MAP_APPFONT ( 156 , 151 ) ;
 		Size = MAP_APPFONT ( 9 , 12 ) ;
 		Left = TRUE ;
 	};
-    FixedLine FL_FILLCHAR
-	{
-        Pos = MAP_APPFONT ( 132 , 3 ) ;
-        Size = MAP_APPFONT ( 66 , 8 ) ;
-		Text [ de ] = "Füllzeichen" ;
-		Text [ en-US ] = "Fill character" ;
-		Text [ x-comment ] = " ";
-	};
-	PushButton BTN_NEW
+    PushButton BTN_NEW
 	{
         Pos = MAP_APPFONT ( 204 , 6 ) ;
 		Size = MAP_APPFONT ( 50 , 14 ) ;
Index: extensions/source/bibliography/general.cxx
===================================================================
RCS file: /cvs/util/extensions/source/bibliography/general.cxx,v
retrieving revision 1.17
retrieving revision 1.17.38.1
diff -u -p -u -p -r1.17 -r1.17.38.1
--- extensions/source/bibliography/general.cxx	16 Nov 2004 11:58:26 -0000	1.17
+++ extensions/source/bibliography/general.cxx	21 Jan 2005 15:49:06 -0000	1.17.38.1
@@ -147,12 +147,13 @@ using namespace ::rtl;
 
 #define C2U(cChar) OUString::createFromAscii(cChar)
 #define C2S(cChar) String::CreateFromAscii(cChar)
+#define DISTANCE_CONTROL_TO_FIXEDTEXT 5
 
 ::Point lcl_MovePoint(const FixedText& rFixedText)
 {
 	::Point aRet(rFixedText.GetPosPixel());
 	aRet.X() += rFixedText.GetSizePixel().Width();
-	aRet.X() += 5;
+    aRet.X() += DISTANCE_CONTROL_TO_FIXEDTEXT;
 	return aRet;
 }
 
@@ -319,6 +320,9 @@ BibGeneralPage::BibGeneralPage(Window* p
 	aCustom3FT(&aControlParentWin,		ResId(FT_CUSTOM3		)),
 	aCustom4FT(&aControlParentWin,		ResId(FT_CUSTOM4		)),
 	aCustom5FT(&aControlParentWin,		ResId(FT_CUSTOM5		)),
+    aFirstFL(&aControlParentWin,        ResId(FL_1        )),
+    aSecondFL(&aControlParentWin,       ResId(FL_2        )),
+    aThirdFL(&aControlParentWin,        ResId(FL_3        )),
 	aHoriScroll(this, WB_HORZ),
 	aVertScroll(this, WB_VERT),
 	sErrorPrefix(ResId(ST_ERROR_PREFIX))
@@ -385,106 +389,109 @@ BibGeneralPage::BibGeneralPage(Window* p
 	xCtrlContnr = VCLUnoHelper::CreateControlContainer(&aControlParentWin);
 
 	xMgr = comphelper::getProcessServiceFactory();
-
-	AddControlWithError( lcl_GetColumnName( pMapping, IDENTIFIER_POS ), lcl_MovePoint( aIdentifierFT ),
-		aIdentifierFT.GetSizePixel(), sTableErrorString, aIdentifierFT.GetText(),
+    // the control should be a bit smaller than the fixed text
+    Size aControlSize(aIdentifierFT.GetSizePixel());
+    aControlSize.Width() = aControlSize.Width() * 8 / 10;
+    
+    AddControlWithError( lcl_GetColumnName( pMapping, IDENTIFIER_POS ), lcl_MovePoint( aIdentifierFT ),
+        aControlSize, sTableErrorString, aIdentifierFT.GetText(),
 		HID_BIB_IDENTIFIER_POS, 0 );
 
 	sTypeColumnName = lcl_GetColumnName(pMapping, AUTHORITYTYPE_POS);
 
-	AddControlWithError( sTypeColumnName, lcl_MovePoint(aAuthTypeFT ), aAuthTypeFT.GetSizePixel(), sTableErrorString,
+    AddControlWithError( sTypeColumnName, lcl_MovePoint(aAuthTypeFT ), aControlSize, sTableErrorString,
 		aAuthTypeFT.GetText(), HID_BIB_AUTHORITYTYPE_POS, 1 );
 
 	::Point aYearPos = lcl_MovePoint(aYearFT);
-	AddControlWithError( lcl_GetColumnName( pMapping, YEAR_POS ),
-		aYearPos, aYearFT.GetSizePixel(), sTableErrorString, aYearFT.GetText(), HID_BIB_YEAR_POS, 4 );
+    AddControlWithError( lcl_GetColumnName( pMapping, YEAR_POS ), aYearPos, 
+        aControlSize, sTableErrorString, aYearFT.GetText(), HID_BIB_YEAR_POS, 4 );
 
 	AddControlWithError( lcl_GetColumnName(pMapping, AUTHOR_POS), lcl_MovePoint(aAuthorFT),
-		aAuthorFT.GetSizePixel(), sTableErrorString, aAuthorFT.GetText(), HID_BIB_AUTHOR_POS, 2 );
+        aControlSize, sTableErrorString, aAuthorFT.GetText(), HID_BIB_AUTHOR_POS, 2 );
 
 	::Point aTitlePos( lcl_MovePoint( aTitleFT ) );
 	::Size aTitleSize = aTitleFT.GetSizePixel();
-	aTitleSize.Width() = aYearPos.X() + aYearFT.GetSizePixel().Width() - aTitlePos.X();
+    aTitleSize.Width() = aYearPos.X() + aControlSize.Width() - aTitlePos.X();
 	AddControlWithError( lcl_GetColumnName(pMapping, TITLE_POS), aTitlePos, aTitleSize, sTableErrorString,
 		aTitleFT.GetText(), HID_BIB_TITLE_POS, 22 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, PUBLISHER_POS ), lcl_MovePoint( aPublisherFT),
-		aPublisherFT.GetSizePixel(), sTableErrorString, aPublisherFT.GetText(), HID_BIB_PUBLISHER_POS, 5 );
+        aControlSize, sTableErrorString, aPublisherFT.GetText(), HID_BIB_PUBLISHER_POS, 5 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, ADDRESS_POS ), lcl_MovePoint( aAddressFT ),
-		aAddressFT.GetSizePixel(), sTableErrorString, aAddressFT.GetText(), HID_BIB_ADDRESS_POS, 7 );
+        aControlSize, sTableErrorString, aAddressFT.GetText(), HID_BIB_ADDRESS_POS, 7 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, ISBN_POS ), lcl_MovePoint( aISBNFT ),
-		aISBNFT.GetSizePixel(), sTableErrorString, aISBNFT.GetText(), HID_BIB_ISBN_POS, 6 );
+        aControlSize, sTableErrorString, aISBNFT.GetText(), HID_BIB_ISBN_POS, 6 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, CHAPTER_POS ), lcl_MovePoint(aChapterFT),
-		aChapterFT.GetSizePixel(), sTableErrorString, aChapterFT.GetText(), HID_BIB_CHAPTER_POS, 10 );
+        aControlSize, sTableErrorString, aChapterFT.GetText(), HID_BIB_CHAPTER_POS, 10 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, PAGES_POS ), lcl_MovePoint( aPagesFT ), aPagesFT.GetSizePixel(),
-		sTableErrorString, aPagesFT.GetText(), HID_BIB_PAGES_POS, 19 );
+    AddControlWithError( lcl_GetColumnName( pMapping, PAGES_POS ), lcl_MovePoint( aPagesFT ), 
+        aControlSize, sTableErrorString, aPagesFT.GetText(), HID_BIB_PAGES_POS, 19 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, EDITOR_POS ), lcl_MovePoint( aEditorFT ), aEditorFT.GetSizePixel(),
-		sTableErrorString, aEditorFT.GetText(), HID_BIB_EDITOR_POS, 12 );
+    AddControlWithError( lcl_GetColumnName( pMapping, EDITOR_POS ), lcl_MovePoint( aEditorFT ), 
+        aControlSize, sTableErrorString, aEditorFT.GetText(), HID_BIB_EDITOR_POS, 12 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, EDITION_POS ), lcl_MovePoint(aEditionFT), aEditionFT.GetSizePixel(),
-		sTableErrorString, aEditionFT.GetText(), HID_BIB_EDITION_POS, 11 );
+    AddControlWithError( lcl_GetColumnName( pMapping, EDITION_POS ), lcl_MovePoint(aEditionFT), 
+        aControlSize, sTableErrorString, aEditionFT.GetText(), HID_BIB_EDITION_POS, 11 );
 
 	AddControlWithError( lcl_GetColumnName(pMapping, BOOKTITLE_POS), lcl_MovePoint(aBooktitleFT),
-		aBooktitleFT.GetSizePixel(), sTableErrorString, aBooktitleFT.GetText(), HID_BIB_BOOKTITLE_POS, 9 );
+        aControlSize, sTableErrorString, aBooktitleFT.GetText(), HID_BIB_BOOKTITLE_POS, 9 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, VOLUME_POS ), lcl_MovePoint( aVolumeFT ), aVolumeFT.GetSizePixel(),
-		sTableErrorString, aVolumeFT.GetText(), HID_BIB_VOLUME_POS, 24 );
+    AddControlWithError( lcl_GetColumnName( pMapping, VOLUME_POS ), lcl_MovePoint( aVolumeFT ), 
+        aControlSize, sTableErrorString, aVolumeFT.GetText(), HID_BIB_VOLUME_POS, 24 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, HOWPUBLISHED_POS ), lcl_MovePoint( aHowpublishedFT ),
-		aHowpublishedFT.GetSizePixel(), sTableErrorString, aHowpublishedFT.GetText(), HID_BIB_HOWPUBLISHED_POS, 13 );
+        aControlSize, sTableErrorString, aHowpublishedFT.GetText(), HID_BIB_HOWPUBLISHED_POS, 13 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, ORGANIZATIONS_POS ), lcl_MovePoint( aOrganizationsFT ),
-		aOrganizationsFT.GetSizePixel(), sTableErrorString, aOrganizationsFT.GetText(), HID_BIB_ORGANIZATIONS_POS, 18 );
+        aControlSize, sTableErrorString, aOrganizationsFT.GetText(), HID_BIB_ORGANIZATIONS_POS, 18 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, INSTITUTION_POS ), lcl_MovePoint( aInstitutionFT ),
-		aInstitutionFT.GetSizePixel(), sTableErrorString, aInstitutionFT.GetText(), HID_BIB_INSTITUTION_POS, 14 );
+        aControlSize, sTableErrorString, aInstitutionFT.GetText(), HID_BIB_INSTITUTION_POS, 14 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, SCHOOL_POS ), lcl_MovePoint( aSchoolFT ), aSchoolFT.GetSizePixel(),
-		sTableErrorString, aSchoolFT.GetText(), HID_BIB_SCHOOL_POS, 20 );
+    AddControlWithError( lcl_GetColumnName( pMapping, SCHOOL_POS ), lcl_MovePoint( aSchoolFT ), 
+        aControlSize, sTableErrorString, aSchoolFT.GetText(), HID_BIB_SCHOOL_POS, 20 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, REPORTTYPE_POS ), lcl_MovePoint( aReportTypeFT ),
-		aReportTypeFT.GetSizePixel(), sTableErrorString, aReportTypeFT.GetText(), HID_BIB_REPORTTYPE_POS, 23 );
+        aControlSize, sTableErrorString, aReportTypeFT.GetText(), HID_BIB_REPORTTYPE_POS, 23 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, MONTH_POS ), lcl_MovePoint( aMonthFT ), aMonthFT.GetSizePixel(),
-		sTableErrorString, aMonthFT.GetText(), HID_BIB_MONTH_POS, 3 );
+    AddControlWithError( lcl_GetColumnName( pMapping, MONTH_POS ), lcl_MovePoint( aMonthFT ), 
+        aControlSize, sTableErrorString, aMonthFT.GetText(), HID_BIB_MONTH_POS, 3 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, JOURNAL_POS ), lcl_MovePoint( aJournalFT ),
-		aJournalFT.GetSizePixel(), sTableErrorString, aJournalFT.GetText(), HID_BIB_JOURNAL_POS, 15 );
+        aControlSize, sTableErrorString, aJournalFT.GetText(), HID_BIB_JOURNAL_POS, 15 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, NUMBER_POS ), lcl_MovePoint( aNumberFT ), aNumberFT.GetSizePixel(),
-		sTableErrorString, aNumberFT.GetText(), HID_BIB_NUMBER_POS, 17 );
+    AddControlWithError( lcl_GetColumnName( pMapping, NUMBER_POS ), lcl_MovePoint( aNumberFT ), 
+        aControlSize, sTableErrorString, aNumberFT.GetText(), HID_BIB_NUMBER_POS, 17 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, SERIES_POS ), lcl_MovePoint( aSeriesFT ), aSeriesFT.GetSizePixel(),
-		sTableErrorString, aSeriesFT.GetText(), HID_BIB_SERIES_POS, 21 );
+    AddControlWithError( lcl_GetColumnName( pMapping, SERIES_POS ), lcl_MovePoint( aSeriesFT ), 
+        aControlSize, sTableErrorString, aSeriesFT.GetText(), HID_BIB_SERIES_POS, 21 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, ANNOTE_POS ), lcl_MovePoint( aAnnoteFT ), aAnnoteFT.GetSizePixel(),
-		sTableErrorString, aAnnoteFT.GetText(), HID_BIB_ANNOTE_POS, 8 );
+    AddControlWithError( lcl_GetColumnName( pMapping, ANNOTE_POS ), lcl_MovePoint( aAnnoteFT ), 
+        aControlSize, sTableErrorString, aAnnoteFT.GetText(), HID_BIB_ANNOTE_POS, 8 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, NOTE_POS ), lcl_MovePoint( aNoteFT ), aNoteFT.GetSizePixel(),
-		sTableErrorString, aNoteFT.GetText(), HID_BIB_NOTE_POS, 16 );
+    AddControlWithError( lcl_GetColumnName( pMapping, NOTE_POS ), lcl_MovePoint( aNoteFT ), 
+        aControlSize, sTableErrorString, aNoteFT.GetText(), HID_BIB_NOTE_POS, 16 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, URL_POS ), lcl_MovePoint( aURLFT ), aURLFT.GetSizePixel(),
-		sTableErrorString, aURLFT.GetText(), HID_BIB_URL_POS, 25 );
+    AddControlWithError( lcl_GetColumnName( pMapping, URL_POS ), lcl_MovePoint( aURLFT ), 
+        aControlSize, sTableErrorString, aURLFT.GetText(), HID_BIB_URL_POS, 25 );
 
 	AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM1_POS ), lcl_MovePoint( aCustom1FT ),
-		aCustom1FT.GetSizePixel(), sTableErrorString, aCustom1FT.GetText(), HID_BIB_CUSTOM1_POS, 26 );
+        aControlSize, sTableErrorString, aCustom1FT.GetText(), HID_BIB_CUSTOM1_POS, 26 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM2_POS ), lcl_MovePoint( aCustom2FT ), aCustom2FT.GetSizePixel(),
-		sTableErrorString, aCustom2FT.GetText(), HID_BIB_CUSTOM2_POS, 27 );
+    AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM2_POS ), lcl_MovePoint( aCustom2FT ), 
+        aControlSize, sTableErrorString, aCustom2FT.GetText(), HID_BIB_CUSTOM2_POS, 27 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM3_POS ), lcl_MovePoint( aCustom3FT ), aCustom3FT.GetSizePixel(),
-		sTableErrorString, aCustom3FT.GetText(), HID_BIB_CUSTOM3_POS, 28 );
+    AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM3_POS ), lcl_MovePoint( aCustom3FT ), 
+        aControlSize, sTableErrorString, aCustom3FT.GetText(), HID_BIB_CUSTOM3_POS, 28 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM4_POS ), lcl_MovePoint( aCustom4FT ), aCustom4FT.GetSizePixel(),
-		sTableErrorString, aCustom4FT.GetText(), HID_BIB_CUSTOM4_POS, 29 );
+    AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM4_POS ), lcl_MovePoint( aCustom4FT ), 
+        aControlSize, sTableErrorString, aCustom4FT.GetText(), HID_BIB_CUSTOM4_POS, 29 );
 
-	AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM5_POS ), lcl_MovePoint( aCustom5FT ), aCustom5FT.GetSizePixel(),
-		sTableErrorString, aCustom5FT.GetText(), HID_BIB_CUSTOM5_POS, 30 );
+    AddControlWithError( lcl_GetColumnName( pMapping, CUSTOM5_POS ), lcl_MovePoint( aCustom5FT ), 
+        aControlSize, sTableErrorString, aCustom5FT.GetText(), HID_BIB_CUSTOM5_POS, 30 );
 
 	xPosListener = new BibPosListener(this);
 	uno::Reference< sdbc::XRowSet >  xRowSet(pDatMan->getForm(), UNO_QUERY);
Index: extensions/source/bibliography/general.hxx
===================================================================
RCS file: /cvs/util/extensions/source/bibliography/general.hxx,v
retrieving revision 1.6
retrieving revision 1.6.430.1
diff -u -p -u -p -r1.6 -r1.6.430.1
--- extensions/source/bibliography/general.hxx	17 May 2002 09:43:10 -0000	1.6
+++ extensions/source/bibliography/general.hxx	21 Jan 2005 15:49:06 -0000	1.6.430.1
@@ -123,34 +123,44 @@ class BibGeneralPage: public BibGeneralP
 	Window				aControlParentWin;
 	FixedText			aIdentifierFT;
 	FixedText			aAuthTypeFT;
+    FixedText           aYearFT;
 
 	FixedText			aAuthorFT;
-	FixedText			aMonthFT;
-	FixedText			aYearFT;
+    FixedText           aTitleFT;
 
 	FixedText			aPublisherFT;
-	FixedText			aISBNFT;
+    FixedText           aAddressFT;
+    FixedText           aISBNFT;
 
-	FixedText			aAddressFT;
-	FixedText			aAnnoteFT;
-	FixedText			aBooktitleFT;
-	FixedText			aChapterFT;
-	FixedText			aEditionFT;
-	FixedText			aEditorFT;
-	FixedText			aHowpublishedFT;
-	FixedText			aInstitutionFT;
-	FixedText			aJournalFT;
-	FixedText			aNoteFT;
-	FixedText			aNumberFT;
-	FixedText			aOrganizationsFT;
-	FixedText			aPagesFT;
-	FixedText			aSchoolFT;
-	FixedText			aSeriesFT;
-	FixedText			aTitleFT;
-	FixedText			aReportTypeFT;
-	FixedText			aVolumeFT;
-	FixedText			aURLFT;
-	FixedText			aCustom1FT;
+    FixedText           aChapterFT;
+    FixedText           aPagesFT;
+    FixedLine           aFirstFL;
+    
+    FixedText           aEditorFT;
+    FixedText           aEditionFT;
+    
+    FixedText           aBooktitleFT;
+    FixedText           aVolumeFT;
+    FixedText           aHowpublishedFT;
+    
+    FixedText           aOrganizationsFT;
+    FixedText           aInstitutionFT;
+    FixedText           aSchoolFT;
+    
+    FixedText           aReportTypeFT;
+    FixedText           aMonthFT;
+    FixedLine           aSecondFL;
+    
+    FixedText           aJournalFT;
+    FixedText           aNumberFT;
+    FixedText           aSeriesFT;
+    
+    FixedText           aAnnoteFT;
+    FixedText           aNoteFT;
+    FixedText           aURLFT;
+    FixedLine           aThirdFL;
+
+    FixedText           aCustom1FT;
 	FixedText			aCustom2FT;
 	FixedText			aCustom3FT;
 	FixedText			aCustom4FT;
Index: extensions/source/bibliography/sections.hrc
===================================================================
RCS file: /cvs/util/extensions/source/bibliography/sections.hrc,v
retrieving revision 1.1.1.1
retrieving revision 1.1.1.1.468.1
diff -u -p -u -p -r1.1.1.1 -r1.1.1.1.468.1
--- extensions/source/bibliography/sections.hrc	18 Sep 2000 16:16:45 -0000	1.1.1.1
+++ extensions/source/bibliography/sections.hrc	21 Jan 2005 15:49:06 -0000	1.1.1.1.468.1
@@ -112,6 +112,9 @@
 #define FT_CUSTOM4		    31
 #define FT_CUSTOM5		    32
 #define ST_ERROR_PREFIX		33
+#define FL_1                34
+#define FL_2                35
+#define FL_3                36
 
 //Business Address
 #define FT_OFFICE		1
Index: extensions/source/bibliography/sections.src
===================================================================
RCS file: /cvs/util/extensions/source/bibliography/sections.src,v
retrieving revision 1.34
retrieving revision 1.34.122.2
diff -u -p -u -p -r1.34 -r1.34.122.2
--- extensions/source/bibliography/sections.src	25 Jun 2004 11:22:22 -0000	1.34
+++ extensions/source/bibliography/sections.src	26 Jan 2005 11:09:26 -0000	1.34.122.2
@@ -63,19 +63,22 @@
 #include "sections.hrc"
 #include "extensio.hrc"
 
-#define FT_POS			6
-#define FTSIZE_LEFT		65
+#define FT_POS          12
+#define FTSIZE_LEFT     88
 #define ED_POS_LEFT		FT_POS+FTSIZE_LEFT+1
-#define ED_SIZE			65
-#define BTN_SIZE		12
+#define ED_SIZE         70
 #define ED_POS_LEFT2	ED_POS_LEFT + ED_SIZE +1
 #define FT_POS_MIDDLE	ED_POS_LEFT2 + 5
 #define FT_POS_RIGHT 	FT_POS_MIDDLE + FT_POS_MIDDLE - FT_POS
 
+#define FL_POS          6
+#define PAGE_WIDTH      FT_POS_RIGHT + FTSIZE_LEFT + (FTSIZE_LEFT * 8 / 10) + FL_POS + 5
+#define FL_WIDTH        PAGE_WIDTH - 2 * FL_POS
+
 TabPage RID_TP_GENERAL
 {
 	SVLook = TRUE ;
-	Size = MAP_APPFONT( 452, 195 );
+    Size = MAP_APPFONT( PAGE_WIDTH, 209 );
 	Text [ de ] = "Allgemein";
 	HelpId = HID_BIB_CONTROL_PAGE;
 
@@ -83,158 +86,204 @@ TabPage RID_TP_GENERAL
 	{
 		Pos = MAP_APPFONT( FT_POS, 6 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
+        Right = TRUE;
 	};
 	FixedText FT_AUTHTYPE
 	{
 		Pos = MAP_APPFONT( FT_POS_MIDDLE, 6 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_YEAR
 	{
 		Pos = MAP_APPFONT( FT_POS_RIGHT, 6 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_AUTHOR
 	{
 		Pos = MAP_APPFONT( FT_POS, 20 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_TITLE
 	{
 		Pos = MAP_APPFONT( FT_POS_MIDDLE, 20 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_PUBLISHER
 	{
 		Pos = MAP_APPFONT( FT_POS, 34 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_ADDRESS
 	{
 		Pos = MAP_APPFONT( FT_POS_MIDDLE, 34 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_ISBN
 	{
 		Pos = MAP_APPFONT( FT_POS_RIGHT, 34 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_CHAPTER
 	{
 		Pos = MAP_APPFONT( FT_POS, 48 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_PAGE
 	{
 		Pos = MAP_APPFONT( FT_POS_MIDDLE, 48 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
+    FixedLine FL_1 
+    {
+        Pos = MAP_APPFONT( FL_POS, 62 );
+        Size = MAP_APPFONT( FL_WIDTH, 8 );
+    };
 	FixedText FT_EDITOR
 	{
-		Pos = MAP_APPFONT( FT_POS, 69 );
+        Pos = MAP_APPFONT( FT_POS, 73 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_EDITION
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 67 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 73 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_BOOKTITLE
 	{
-		Pos = MAP_APPFONT( FT_POS, 83 );
+        Pos = MAP_APPFONT( FT_POS, 87 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_VOLUME
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 83 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 87 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
+        Right = TRUE;
     };
 	FixedText FT_HOWPUBLISHED
 	{
-		Pos = MAP_APPFONT( FT_POS_RIGHT, 83 );
+        Pos = MAP_APPFONT( FT_POS_RIGHT, 87 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_ORGANIZATION
 	{
-		Pos = MAP_APPFONT( FT_POS, 97 );
+        Pos = MAP_APPFONT( FT_POS, 101 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_INSTITUTION
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 97 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 101 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_SCHOOL
 	{
-		Pos = MAP_APPFONT( FT_POS_RIGHT, 97 );
+        Pos = MAP_APPFONT( FT_POS_RIGHT, 101 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
+        Right = TRUE;
     };
 	FixedText FT_REPORT
 	{
-		Pos = MAP_APPFONT( FT_POS, 111);
+        Pos = MAP_APPFONT( FT_POS, 115);
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
+        Right = TRUE;
     };
 	FixedText FT_MONTH
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 111 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 115 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
-	FixedText FT_JOURNAL
+        Right = TRUE;
+    };
+    FixedLine FL_2
+    {
+        Pos = MAP_APPFONT( FL_POS, 129 );
+        Size = MAP_APPFONT( FL_WIDTH, 8 );
+    };
+    FixedText FT_JOURNAL
 	{
-		Pos = MAP_APPFONT( FT_POS, 132 );
+        Pos = MAP_APPFONT( FT_POS, 140 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 
 	FixedText FT_NUMBER
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 132 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 140 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_SERIES
 	{
-		Pos = MAP_APPFONT( FT_POS_RIGHT, 132 );
+        Pos = MAP_APPFONT( FT_POS_RIGHT, 140 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_ANNOTE
 	{
-		Pos = MAP_APPFONT( FT_POS, 146 );
+        Pos = MAP_APPFONT( FT_POS, 154 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_NOTE
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 146 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 154 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_URL
 	{
-		Pos = MAP_APPFONT( FT_POS_RIGHT, 146 );
+        Pos = MAP_APPFONT( FT_POS_RIGHT, 154 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
-	FixedText FT_CUSTOM1
+        Right = TRUE;
+    };
+    FixedLine FL_3
+    {
+        Pos = MAP_APPFONT( FL_POS, 168 );
+        Size = MAP_APPFONT( FL_WIDTH, 8 );
+    };
+    FixedText FT_CUSTOM1
 	{
-		Pos = MAP_APPFONT( FT_POS, 167 );
+        Pos = MAP_APPFONT( FT_POS, 179 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_CUSTOM2
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 167 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 179 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_CUSTOM3
 	{
-		Pos = MAP_APPFONT( FT_POS_RIGHT, 167 );
+        Pos = MAP_APPFONT( FT_POS_RIGHT, 179 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_CUSTOM4
 	{
-		Pos = MAP_APPFONT( FT_POS, 181 );
+        Pos = MAP_APPFONT( FT_POS, 193 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	FixedText FT_CUSTOM5
 	{
-		Pos = MAP_APPFONT( FT_POS_MIDDLE, 181 );
+        Pos = MAP_APPFONT( FT_POS_MIDDLE, 193 );
 		Size = MAP_APPFONT( FTSIZE_LEFT, 10 );
-	};
+        Right = TRUE;
+    };
 	String ST_ERROR_PREFIX
 	{
 		Text [ de ] = "Die folgenden Spaltennamen konnten nicht zugeordnet werden:\n" ;
