--- oovbaapi/org/openoffice//excel/XName.idl	
+++ oovbaapi/org/openoffice//excel/XName.idl
@@ -58,12 +58,12 @@ interface XName
 	[attribute] string Name;
 	[attribute] string NameLocal;
 	[attribute] boolean Visible;
-        [attribute, readonly] string Value;
-        [attribute, readonly] string RefersTo;
-        [attribute, readonly] string RefersToLocal;
-        [attribute, readonly] string RefersToR1C1;
-        [attribute, readonly] string RefersToR1C1Local;
-        [attribute, readonly] XRange RefersToRange;
+    [attribute] string Value;
+    [attribute] string RefersTo;
+    [attribute] string RefersToLocal;
+    [attribute] string RefersToR1C1;
+    [attribute] string RefersToR1C1Local;
+    [attribute, readonly] XRange RefersToRange;
 
 	void Delete( );
 };
