--- oovbaapi/org/openoffice//excel/XWindow.idl	
+++ oovbaapi/org/openoffice//excel/XWindow.idl
@@ -49,6 +49,8 @@ module org {  module openoffice {  modul
 
 //=============================================================================
 interface XRange;
+interface XWorksheet;
+interface XPane;
 interface XWindow
 {
 	interface ::org::openoffice::vba::XHelperInterface;
@@ -61,6 +63,8 @@ interface XWindow
 	[attribute] boolean DisplayVerticalScrollBar; 
 	[attribute] boolean DisplayWorkbookTabs; 
 	[attribute] boolean FreezePanes; 
+	[attribute] long Height; 
+	[attribute] long Left; 
 	[attribute] boolean Split; 
 	[attribute] long SplitColumn;
 	[attribute] double SplitHorizontal;
@@ -68,6 +72,10 @@ interface XWindow
 	[attribute] double SplitVertical;
 	[attribute] any ScrollColumn;
 	[attribute] any ScrollRow;
+	[attribute] long Top; 
+	[attribute] any View;
+	[attribute] boolean Visible;
+	[attribute] long Width; 
 	[attribute] any WindowState;
 	[attribute] any Zoom;
 	any SelectedSheets( [in] any aIndex );
@@ -78,7 +86,21 @@ interface XWindow
         void Close([in] any SaveChanges, [in] any FileName, [in] any RouteWorkBook);
 	XRange ActiveCell() raises(com::sun::star::script::BasicErrorException);
 	any Selection() raises(com::sun::star::script::BasicErrorException);
+	long PointsToScreenPixelsX([in] long Points) raises(com::sun::star::script::BasicErrorException);
+	long PointsToScreenPixelsY([in] long Points) raises(com::sun::star::script::BasicErrorException);
+    void PrintOut([in] /*optional short*/ any From,
+        [in] /*optional short*/ any To,
+        [in] /*optional short*/ any Copies,
+        [in] /*optional boolean*/ any Preview,
+        [in] /*optional string*/ any ActivePrinter,
+        [in] /*optional boolean*/ any PrintToFile,
+        [in] /*optional boolean*/ any Collate,
+        [in] /*optional string*/ any PrToFileName
+        ) raises(com::sun::star::script::BasicErrorException);
+     void PrintPreview( [in]  /*Optional*/ any EnableChanges ) raises ( com::sun::star::script::BasicErrorException );
 
+	XWorksheet ActiveSheet() raises(com::sun::star::script::BasicErrorException);
+	XPane ActivePane() raises(com::sun::star::script::BasicErrorException);
 };
 
 //=============================================================================
