--- neon/makefile.mk	24 Oct 2002 11:50:58 -0000	1.5
+++ neon/makefile.mk	3 Mar 2004 01:30:41 -0000
@@ -71,6 +71,11 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(SYSTEM_NEON)" == "YES"
+@all:
+	@echo "Using system neon...."
+.ENDIF
+
 NEON_NAME=neon-0.23.5
 
 TARFILE_NAME=$(NEON_NAME)
--- ucb/source/ucp/webdav/makefile.mk	28 Jan 2004 10:15:32 -0000	1.20.4.1
+++ ucb/source/ucp/webdav/makefile.mk	3 Mar 2004 23:18:51 -0000
@@ -84,8 +84,12 @@
 .INCLUDE: settings.mk
 .INCLUDE: sv.mk
 
+.IF "$(SYSTEM_NEON)" == "YES"
+CFLAGS+= $(NEON_CFLAGS)
+.ELSE
 NEONINCDIR=external$/neon
 CFLAGS+= -I$(SOLARINCDIR)$/$(NEONINCDIR)
+.ENDIF
 
 # --- General -----------------------------------------------------
 
Index: NeonSession.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/webdav/NeonSession.cxx,v
retrieving revision 1.28.4.1
diff -u -r1.28.4.1 NeonSession.cxx
--- ucb/source/ucp/webdav/NeonSession.cxx	9 Jan 2004 19:07:09 -0000	1.28.4.1
+++ ucb/source/ucp/webdav/NeonSession.cxx	4 Mar 2004 00:07:29 -0000
@@ -63,7 +63,11 @@
 
 #include <string.h>
 
+#ifndef USE_DAV_LOCKS
+#ifndef SYSTEM_NEON
 #include <neon/config.h> /* for USE_DAV_LOCKS */
+#endif
+#endif
 
 #ifndef NE_AUTH_H
 #include <neon/ne_auth.h>
