--- /dev/null	
+++ vcl/unx/gtk/a11y/atklistener.hxx	
@@ -0,0 +1,32 @@
+#ifndef _ATK_LISTENER_HXX_
+#define _ATK_LISTENER_HXX_
+
+#include <com/sun/star/accessibility/XAccessibleEventListener.hpp>
+#include "atkwrapper.hxx"
+
+using namespace ::com::sun::star;
+
+class AtkListener : public accessibility::XAccessibleEventListener
+{
+	// UNO sucks for not providing a sane base-class for this.
+	sal_Int32 mnRefCount;
+	AtkObject *mpAccessible;
+
+public:
+	AtkListener( AtkObject *pObj );
+	virtual ~AtkListener();
+
+	// XInterface
+    virtual void acquire() throw () { ++mnRefCount; }
+    virtual void release() throw () { if (! --mnRefCount) delete this; }
+    virtual uno::Any queryInterface( const uno::Type & rType ) throw (uno::RuntimeException);
+
+	// XEventListener
+	virtual void disposing( const lang::EventObject& Source ) throw (uno::RuntimeException);
+
+	// XAccessibleEventListener
+	virtual void notifyEvent( const accessibility::AccessibleEventObject& aEvent ) throw( uno::RuntimeException );
+};
+
+#endif /* _ATK_LISTENER_HXX_ */
+
