--- /dev/null	
+++ vcl/unx/gtk/a11y/TODO	
@@ -0,0 +1,61 @@
++ AtkValue
+
+Serious problems
+    + Threading/locking:
+	+ incoming CORBA calls & the GDK lock
+		+ how are these being processed & on what thread ?
+		+ are we holding the GDK_THREADS lock ?
+		+ can we even do that ?
+		+ is it really necessary to be thread safe ?
+		+ how does this work in combination with the (unsafe) GAIL code ?
+		+ what should incoming CORBA calls be doing ?
+			+ esp. since we can't tell if they're coming from
+			  in-proc or not either [ though this is unlikely ]
+
+Missing features
+	+ notifications when the tree state changes
+		+ works for toplevels - managed by gail ...
+	+ hack in 'atklistener.cxx'
+		+ ... here ...
+
+Known bugs:
+	+ finish AtkText (!)
+	+ crashes on 'expand all' with writer open
+	+ tooltips create new toplevels with broken roles.
+
+Crasher from here:
+
+(gdb) t 1
+[Switching to thread 1 (process 24740)]#0  0x44fd892c in os::report_fatal_error ()
+   from /usr/lib/SunJava2-1.4.2/jre/lib/i386/client/libjvm.so
+(gdb) bt
+#3  0x44fdbde8 in signalHandler () from /usr/lib/SunJava2-1.4.2/jre/lib/i386/client/libjvm.so
+#4  <signal handler called>
+#5  0x40cc2725 in comphelper::AccessibleEventNotifier::addEvent () from ./libcomphelp4gcc3.so
+#6  0x54715529 in SwAccessibleContext::FireAccessibleEvent () from ./libsw680li.so
+#7  0x547156ec in SwAccessibleContext::FireStateChangedEvent () from ./libsw680li.so
+#8  0x547345dd in SwAccessibleParagraph::_InvalidateFocus () from ./libsw680li.so
+#9  0x54719cb7 in SwAccessibleContext::InvalidateFocus () from ./libsw680li.so
+#10 0x5472cc39 in SwAccessibleMap::InvalidateFocus () from ./libsw680li.so
+#11 0x547aa128 in ViewShell::InvalidateAccessibleFocus () from ./libsw680li.so
+#12 0x54d803ae in SwEditWin::GetFocus () from ./libsw680li.so
+#13 0x4033c28a in Window::ImplGrabFocus () from ./libvcl680li.so
+#14 0x403446c1 in Window::GrabFocus () from ./libvcl680li.so
+#15 0x403523bc in Window::ImplAsyncFocusHdl () from ./libvcl680li.so
+#16 0x403522d4 in Window::LinkStubImplAsyncFocusHdl () from ./libvcl680li.so
+#17 0x40352978 in ImplHandleUserEvent () from ./libvcl680li.so
+#18 0x40353499 in ImplWindowFrameProc () from ./libvcl680li.so
+#19 0x42419ff9 in SalDisplay::DispatchInternalEvent () from ./libvclplug_gen680li.so
+#20 0x41ebc95e in GtkXLib::userEventFn () at /opt/OpenOffice/src680-m90/vcl/unx/gtk/window/gtkframe.cxx:93
+#21 0x4234a221 in g_idle_dispatch () from /opt/gnome/lib/libglib-2.0.so.0
+#22 0x4234be87 in g_main_context_dispatch () from /opt/gnome/lib/libglib-2.0.so.0
+#23 0x4234e5c7 in g_main_context_iterate () from /opt/gnome/lib/libglib-2.0.so.0
+#24 0x4234e6c3 in g_main_context_iteration () from /opt/gnome/lib/libglib-2.0.so.0
+#25 0x41ebcaee in GtkXLib::Yield () at /opt/OpenOffice/src680-m90/vcl/unx/gtk/window/gtkframe.cxx:93
+#26 0x4241e4ef in X11SalInstance::Yield () from ./libvclplug_gen680li.so
+#27 0x40174c54 in Application::Yield () from ./libvcl680li.so
+#28 0x40174b6b in Application::Execute () from ./libvcl680li.so
+#29 0x0806f1b5 in desktop::Desktop::Main ()
+#30 0x4017a713 in SVMain () from ./libvcl680li.so
+#31 0x0806898d in sal_main ()
+#32 0x0806893c in main ()
