--- sfx2/source/dialog/about.cxx
+++ sfx2/source/dialog/about.cxx
@@ -222,11 +222,6 @@
 	aCopyrightText.SetControlForeground( aTextColor );
 	aBuildData.SetBackground( aWall );
 
-	Font aSmallFont = rSettings.GetInfoFont();
-	Size aSmaller = aNewFont.GetSize();
-	aSmaller.Width() = (long) (aSmaller.Width() * 0.75);
-	aSmaller.Height() = (long) (aSmaller.Height() * 0.75);
-	aNewFont.SetSize( aSmaller );
 	aBuildData.SetFont( aNewFont );
 	aBuildData.SetBackground( aWall );
 #ifdef BUILD_VER_STRING
