Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.270
diff -u -p -r1.270 configure.in
--- config_office/configure.in	6 Jun 2008 12:05:12 -0000	1.270
+++ config_office/configure.in	25 Jun 2008 11:28:30 -0000
@@ -5890,13 +6085,20 @@ cat > conftest.java << EOF
 	}
     }
 EOF
-
+mkdir confdir
 cat > conftest.xml << EOF
     <project name="conftest" default="conftest">
-	<target name="conftest">
+	<target name="conftest" depends="copytest">
             <javac srcdir="." includes="conftest.java">
 	    </javac>
 	</target>
+	<target name="copytest">
+            <copy todir="confdir">
+                <fileset dir="confdir" includes="**/*.abc" casesensitive="yes"/>
+                <filterset/>
+                <mapper type="regexp" from="^(.*[/\\])foo([/\\].*)" to="\1baa\2"/>
+            </copy>
+	</target>
     </project>
 EOF
   oldJAVA_HOME=$JAVA_HOME
@@ -5921,12 +6123,13 @@ EOF
     echo "configure: Ant test failed" >&5
     cat conftest.java >&5
     cat conftest.xml >&5
-    AC_MSG_WARN([Ant does not work - Some Java projects will not build!])
+    AC_MSG_WARN([Ant does not work - Some Java projects will not build, do you have ant-apache-regexp installed ?!])
     ANT_HOME=""
     echo "Ant does not work - Some Java projects will not build!" >>warn
   fi
   JAVA_HOME=$oldJAVA_HOME
   rm -f conftest* core core.* *.core
+  rmdir confdir
 fi
 if test -z "$ANT_HOME"; then
    ANT_HOME="NO_ANT_HOME"
