--- xmlhelp/source/com/sun/star/help/makefile.mk.ark	2006-03-01 23:44:21.000000000 +0100
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2006-03-01 23:45:50.000000000 +0100
@@ -121,21 +121,7 @@
 		-Djavax.xml.validation.SchemaFactory=org.apache.xerces.jaxp.validation.XMLSchemaFactory \
 		-Dorg.w3c.dom.DOMImplementationSourceList=org.apache.xerces.dom.DOMXSImplementationSourceImpl \
 		-Dorg.xml.sax.driver=org.apache.xerces.parsers.SAXParser \
-		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar -lxt-xmlsearch.jar
-
-.IF "$(SYSTEM_DB)"=="YES"
-$(BIN)$/HelpLinker: $(LB)$/libdb.jar.a
-
-$(LB)$/libdb.jar.a:
-	$(JAVABINCOMPILER) -c -O2 -findirect-dispatch -fjni $(DB_JAR) -o $(LB)$/libdb.jar.a
-.ENDIF
-
-.IF "$(SYSTEM_XT)"=="YES"
-$(BIN)$/HelpLinker: $(LB)$/libxt.jar.a
-
-$(LB)$/libxt.jar.a:
-	$(JAVABINCOMPILER) -c -O2 -findirect-dispatch -fjni $(XT_JAR) -o $(LB)$/libxt.jar.a
-.ENDIF
+		-ldb.jar -lxmlsearch.jar -lxt -lxercesImpl -lunoil.jar -ljurt.jar -lridl.jar -lxt-xmlsearch.jar
 
 .ELSE
 JARCLASSDIRS	= com
