diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=officecfg.vpj officecfg.clean/registry/schema/org/openoffice/Office/Common.xcs officecfg/registry/schema/org/openoffice/Office/Common.xcs
--- officecfg.clean/registry/schema/org/openoffice/Office/Common.xcs	2008-02-20 15:04:19.000000000 -0500
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2008-02-25 18:29:09.000000000 -0500
@@ -1194,6 +1194,15 @@ Dymamic border coloring means that when 
 				</info>
 				<value>false</value>
 			</prop>
+			<prop oor:name="CorrectAccidentalCapsLock" oor:type="xs:boolean">
+				<!-- UIHints: Tools  AutoCorrect/AutoFormat  Options  Correct accidental use of cAPS LOCK key -->
+				<info>
+					<author>Kohei Yoshida</author>
+					<desc>Specifies whether or not to correct accidental use of CAPS lock key.</desc>
+					<label>Correct accidental use of cAPS LOCK key</label>
+				</info>
+				<value>true</value>
+			</prop>
 			<prop oor:name="ReplaceSingleQuote" oor:type="xs:boolean">
 				<!-- OldPath: AutoCorrect/Options/All -->
 				<!-- OldLocation: Soffice.cfg -->
