diff --git offapi/com/sun/star/awt/Key.idl offapi/com/sun/star/awt/Key.idl
index c14af6d..7265d9d 100644
--- offapi/com/sun/star/awt/Key.idl
+++ offapi/com/sun/star/awt/Key.idl
@@ -449,6 +449,18 @@ published constants Key
 
 	const short QUOTELEFT = 1311;
 
+    //------------------------------------------------------------------------- 
+
+    const short CAPSLOCK = 1312;
+
+    //------------------------------------------------------------------------- 
+
+    const short NUMLOCK = 1313;
+
+    //------------------------------------------------------------------------- 
+
+    const short SCROLLLOCK = 1314;
+
 	//-------------------------------------------------------------------------
     
     const short DELETE_TO_BEGIN_OF_LINE = 1536;
