diff --git a/basic/source/classes/sbunoobj.cxx b/basic/source/classes/sbunoobj.cxx
index 539b023..b1485ab 100644
--- basic/source/classes/sbunoobj.cxx
+++ basic/source/classes/sbunoobj.cxx
@@ -2243,8 +2243,9 @@ void SbUnoObject::SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
 Reference< XInvocation > createDynamicInvocationFor( const Any& aAny );
 #endif
 
-SbUnoObject::SbUnoObject( const String& aName_, const Any& aUnoObj_ )
-	: SbxObject( aName_ )
+SbUnoObject::SbUnoObject( const String& aName_, const Any& aUnoObj_ ) :
+    SbxObject(aName_),
+    bNeedIntrospection(false)
 {
 	static Reference< XIntrospection > xIntrospection;
 
