--- basic/source/runtime/methods.cxx.ark	2008-07-11 12:16:19.216882727 +0200
+++ basic/source/runtime/methods.cxx	2008-07-11 13:22:14.260948113 +0200
@@ -51,6 +51,7 @@
 #include <unotools/ucbstreamhelper.hxx>
 #include <tools/wldcrd.hxx>
 #include <i18npool/lang.h>
+#include <rtl/string.hxx>
 
 #include "runtime.hxx"
 #include "sbunoobj.hxx"
@@ -123,6 +124,8 @@ using namespace com::sun::star::io;
 #include <io.h>
 #endif
 
+using namespace rtl;
+
 static void FilterWhiteSpace( String& rStr )
 {
 	rStr.EraseAllChars( ' ' );
