--- bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c.old	2008-04-10 18:21:42.000000000 +0200
+++ bean/native/unix/com_sun_star_beans_LocalOfficeWindow.c	2008-08-22 19:39:59.469966813 +0200
@@ -33,6 +33,12 @@
 #include <X11/Intrinsic.h>
 
 #include "jni.h"
+
+// Workaround for problematic IBM JDK 1.6.0 on ppc
+#ifndef _JNI_IMPORT_OR_EXPORT_
+#define _JNI_IMPORT_OR_EXPORT_
+#endif
+
 #include "jawt_md.h"
 #include "jawt.h"
 
--- bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c.old	2008-04-10 18:21:58.000000000 +0200
+++ bean/native/unix/com_sun_star_comp_beans_LocalOfficeWindow.c	2008-08-22 19:40:17.120161386 +0200
@@ -33,6 +33,12 @@
 #include <X11/Intrinsic.h>
 
 #include "jni.h"
+
+// Workaround for problematic IBM JDK 1.6.0 on ppc
+#ifndef _JNI_IMPORT_OR_EXPORT_
+#define _JNI_IMPORT_OR_EXPORT_
+#endif
+
 #include "jawt_md.h"
 #include "jawt.h"
 
