diff --git binfilter/bf_sw/source/filter/inc/SwAppletImpl.hxx binfilter/bf_sw/source/filter/inc/SwAppletImpl.hxx
index f309c89..959ed93 100644
--- binfilter/bf_sw/source/filter/inc/SwAppletImpl.hxx
+++ binfilter/bf_sw/source/filter/inc/SwAppletImpl.hxx
@@ -69,11 +69,6 @@
 namespace binfilter {
 
 class SfxItemSet; 
-extern sal_Char const SVTOOLS_CONSTASCII_DECL( sHTML_O_hidden, "HIDDEN" );
-extern sal_Char const SVTOOLS_CONSTASCII_DECL( sHTML_HIDDEN_false, "FALSE" );
-extern sal_Char const SVTOOLS_CONSTASCII_DECL( sHTML_O_archive, "ARCHIVE" );
-extern sal_Char const SVTOOLS_CONSTASCII_DECL( sHTML_O_archives, "ARCHIVES" );
-extern sal_Char const SVTOOLS_CONSTASCII_DECL( sHTML_O_object, "OBJECT" );

 class SwApplet_Impl
 {
