commit 26e1abf9b5716bdcfbf7df9f26997114fd83c140
Author: Jan Nieuwenhuizen <janneke@gnu.org>
Date:   Mon Jan 12 11:48:26 2009 +0100

    Compile fix for basic --enable-debug.

diff --git basic/source/app/brkpnts.cxx basic/source/app/brkpnts.cxx
index 79f702b..3649237 100644
--- basic/source/app/brkpnts.cxx
+++ basic/source/app/brkpnts.cxx
@@ -111,7 +111,7 @@ void BreakpointWindow::SetBPsInModule()
 	{
 		pModule->SetBP( (USHORT)pBrk->nLine );
 #if OSL_DEBUG_LEVEL > 1
-		DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( (USHORT)pBrk->nLine ), "Brechpunkt wurde nicht gesetzt" )
+		DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( (USHORT)pBrk->nLine ), "Brechpunkt wurde nicht gesetzt" );
 #endif
 		pBrk = Next();
 	}
commit c27bcadfd9d44921a0c1a5ec3deb10ae55a9d20e
Author: Jan Nieuwenhuizen <janneke@gnu.org>
Date:   Mon Jan 12 11:55:23 2009 +0100

    More compile fixes for basic --enable-debug.

diff --git basic/source/app/brkpnts.cxx basic/source/app/brkpnts.cxx
index 3649237..370e3da 100644
--- basic/source/app/brkpnts.cxx
+++ basic/source/app/brkpnts.cxx
@@ -152,7 +152,7 @@ void BreakpointWindow::InsertBreakpoint( USHORT nLine )
 	if ( pModule->SetBP( nLine ) )
 	{
 #if OSL_DEBUG_LEVEL > 1
-		DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( nLine ), "Brechpunkt wurde nicht gesetzt" )
+		DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( nLine ), "Brechpunkt wurde nicht gesetzt" );
 #endif
 		if ( StarBASIC::IsRunning() )
 		{
@@ -165,7 +165,7 @@ void BreakpointWindow::InsertBreakpoint( USHORT nLine )
 		}
 	}
 #if OSL_DEBUG_LEVEL > 1
-	DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( nLine ), "Brechpunkt wurde nicht gesetzt" )
+	DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( nLine ), "Brechpunkt wurde nicht gesetzt" );
 #endif
 }
 
@@ -282,7 +282,7 @@ void BreakpointWindow::Paint( const Rectangle& )
 	while ( pBrk )
 	{
 #if OSL_DEBUG_LEVEL > 1
-		DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( pBrk->nLine ), "Brechpunkt wurde nicht gesetzt" )
+		DBG_ASSERT( !pModule->IsCompiled() || pModule->IsBP( pBrk->nLine ), "Brechpunkt wurde nicht gesetzt" );
 #endif
 		ULONG nLine = pBrk->nLine-1;
 		ULONG nY = nLine*nLineHeight - nCurYOffset;
commit 3cad7f2b20092c2a06b46f664630cc5103be0fad
Author: Jan Nieuwenhuizen <janneke@gnu.org>
Date:   Mon Jan 12 12:19:05 2009 +0100

    Compile fix for basic --enable-debug.

diff --git automation/source/testtool/objtest.cxx automation/source/testtool/objtest.cxx
index 5a49c7d..ef6c6c0 100644
--- automation/source/testtool/objtest.cxx
+++ automation/source/testtool/objtest.cxx
@@ -2581,7 +2581,7 @@ void TestToolObj::SFX_NOTIFY( SfxBroadcaster&, const TypeId&,
                                 long aMS = long( aDiff.GetMSFromTime() );
                                 if ( Abs( aMS - nWait ) > 100 )
                                 {
-                                    DBG_ERROR1("Wait was off limit by %i", aDiff.GetMSFromTime() - nWait )
+                                    DBG_ERROR1("Wait was off limit by %i", aDiff.GetMSFromTime() - nWait );
                                 }
 #endif
 	                        }
commit bf3570e956d59db729762f3f6733f571336b1991
Author: Jan Nieuwenhuizen <janneke@gnu.org>
Date:   Mon Jan 12 14:17:55 2009 +0100

    Compile fix for --enable-debug.

diff --git starmath/source/dialog.cxx starmath/source/dialog.cxx
index 0fc29fd..413cf89 100644
--- starmath/source/dialog.cxx
+++ starmath/source/dialog.cxx
@@ -1874,7 +1874,7 @@ IMPL_LINK( SmSymDefineDialog, CharHighlightHdl, Control *, EMPTYARG )
     sal_UCS4 cChar = aCharsetDisplay.GetSelectCharacter();
 
 #if OSL_DEBUG_LEVEL > 1
-    DBG_ASSERT( pSubsetMap, "SubsetMap missing" )
+    DBG_ASSERT( pSubsetMap, "SubsetMap missing" );
 #endif
     if (pSubsetMap)
     {
commit c8c0f86735c8d83ba9a627266765c953972302ca
Author: Jan Nieuwenhuizen <janneke@gnu.org>
Date:   Mon Jan 12 15:27:02 2009 +0100

    Compile fix for --enable-debug.

diff --git sw/source/ui/docvw/srcedtw.cxx sw/source/ui/docvw/srcedtw.cxx
index 0eb16e0..49419d1 100644
--- sw/source/ui/docvw/srcedtw.cxx
+++ sw/source/ui/docvw/srcedtw.cxx
@@ -191,8 +191,8 @@ static void lcl_Highlight(const String& rSource, SwTextPortions& aPortionList)
 					{
 						//was war das denn?
 #if OSL_DEBUG_LEVEL > 1
-						DBG_ERROR("Token nicht erkannt!")
-						DBG_ERROR(ByteString(sToken, gsl_getSystemTextEncoding()).GetBuffer())
+						DBG_ERROR("Token nicht erkannt!");
+						DBG_ERROR(ByteString(sToken, gsl_getSystemTextEncoding()).GetBuffer());
 #endif
 					}
 
