Index: source/cairo/cairo_canvashelper.hxx
===================================================================
RCS file: /cvs/gsl/canvas/source/cairo/cairo_canvashelper.hxx,v
retrieving revision 1.5
diff -u -r1.5 cairo_canvashelper.hxx
--- canvas/source/cairo/cairo_canvashelper.hxx	9 Nov 2007 10:14:00 -0000	1.5
+++ canvas/source/cairo/cairo_canvashelper.hxx	14 Apr 2008 13:27:28 -0000
@@ -331,7 +331,8 @@
 					Operation aOperation,
 					::cairo::Cairo* pCairo,
 					const ::com::sun::star::uno::Sequence< ::com::sun::star::rendering::Texture >* pTextures=NULL,
-					SpriteCanvas* pDevice=NULL );
+					SpriteCanvas* pDevice=NULL, 
+					::com::sun::star::rendering::FillRule eFillrule=::com::sun::star::rendering::FillRule_EVEN_ODD);
     void doOperation( Operation aOperation,
 		      ::cairo::Cairo* pCairo,
 		      sal_uInt32 nPolygonIndex=0,
