--- filter/source/xsltfilter/XSLTransformer.java.old	2008-08-06 16:04:33.000000000 +0200
+++ filter/source/xsltfilter/XSLTransformer.java	2008-08-15 20:43:41.000000000 +0200
@@ -217,7 +217,10 @@ public class XSLTransformer
         // notify listeners
         t = new Thread() {
 
-            @Override
+            // @Override is optional; It just helps the compiler to catch errors, see
+            // http://java.sun.com/docs/books/tutorial/java/javaOO/annotations.html
+            // We will disable to to be able to compile with JRE 1.4
+            // @Override
             public void run() {
 
                 // Local variabes used outside try block in finally block                  
