Index: solenv/inc/unxlngi4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi4.mk,v
retrieving revision 1.17
diff -u -p -u -r1.17 unxlngi4.mk
--- solenv/inc/unxlngi4.mk	28 Apr 2003 16:46:57 -0000	1.17
+++ solenv/inc/unxlngi4.mk	13 May 2003 09:46:12 -0000
@@ -145,7 +145,7 @@ LINK*=$(CC)
 LINKFLAGSRUNPATH_BRAND=-Wl,-rpath,\''$$ORIGIN:$$ORIGIN/../basis-link/program:$$ORIGIN/../basis-link/ure-link/lib'\'
 LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_NONE=
-LINKFLAGS=-Wl,-z,combreloc $(LINKFLAGSDEFS)
+LINKFLAGS=-O2 -Bsymbolic -Wl,-z,combreloc $(LINKFLAGSDEFS)
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec
