--- reportdesign/java/com/sun/star/report/function/metadata/makefile.mk-old	2008-06-21 17:57:30.000000000 +0200
+++ reportdesign/java/com/sun/star/report/function/metadata/makefile.mk	2008-06-21 17:56:11.000000000 +0200
@@ -40,7 +40,7 @@
 
 # JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar
 .IF "$(SYSTEM_JFREEREPORT)" == "YES"
-XCLASSPATH!:==$(XCLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)$(PATH_SEPERATOR)$(LIBFORMULA_JAR)
+CLASSPATH!:=$(CLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)$(PATH_SEPERATOR)$(LIBFORMULA_JAR)
 .ELSE
 JARFILES += jcommon-1.0.10.jar libformula-0.1.14.jar
 .ENDIF
