diff --git bridges/source/cpp_uno/gcc3_linux_intel/except.cxx bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
index 163bf41..2eab1b5 100644
--- bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
+++ bridges/source/cpp_uno/gcc3_linux_intel/except.cxx
@@ -34,6 +34,7 @@
 #include <stdio.h>
 #include <dlfcn.h>
 #include <cxxabi.h>
+#include <string.h>
 #include <hash_map>
 
 #include <rtl/strbuf.hxx>
