--- config_office/configure.in.old	2008-03-20 21:10:37.000000000 +0100
+++ config_office/configure.in	2008-03-27 17:40:30.000000000 +0100
@@ -5380,7 +5380,7 @@ else
 AC_SUBST(SERVLETAPI_JAR)
 
 AC_MSG_CHECKING([whether to build the Report Builder extension])
-if test -z "$enable_reportdesign" -a "$enable_reportdesign" != "no" && test "$WITH_JAVA" != "no"; then
+if test -z "$enable_reportdesign" -o "$enable_reportdesign" != "no" && test "$WITH_JAVA" != "no"; then
   AC_MSG_RESULT([yes])
   ENABLE_REPORTDESIGN=YES
   AC_MSG_CHECKING([which jfreereport libs to use])
