--- configure.in.old	2008-08-22 12:07:03.000000000 +0200
+++ configure.in	2008-08-22 17:34:05.000000000 +0200
@@ -3387,6 +3387,19 @@ else
 fi
 
 dnl ===================================================================
+dnl Check for building stax
+dnl ===================================================================
+AC_MSG_CHECKING([whether to build the stax])
+if test -f "external/StAX/jsr173_1.0_api.jar"; then
+   BUILD_STAX=NO
+   AC_MSG_RESULT([no, will use the prebuilt external/StAX/jsr173_1.0_api.jar])
+else
+   BUILD_STAX=YES
+   AC_MSG_RESULT([yes])
+fi
+AC_SUBST(BUILD_STAX)
+
+dnl ===================================================================
 dnl Check for building ODK
 dnl ===================================================================
 AC_MSG_CHECKING([whether to build the ODK])
--- set_soenv.in.old	2008-08-22 12:07:03.000000000 +0200
+++ set_soenv.in	2008-08-22 17:35:07.000000000 +0200
@@ -2070,6 +2070,7 @@ ToFile( "ENABLE_AGG",        "@ENABLE_AG
 ToFile( "SYSTEM_AGG",	     "@SYSTEM_AGG@",       "e" );
 ToFile( "AGG_VERSION",	     "@AGG_VERSION@",      "e" );
 ToFile( "BUILD_DMAKE",       "@BUILD_DMAKE@",      "e" );
+ToFile( "BUILD_STAX",        "@BUILD_STAX@",       "e" );
 ToFile( "BUILD_UNOWINREG",   "@BUILD_UNOWINREG@",  "e" );
 ToFile( "USE_XINERAMA",      "@USE_XINERAMA@",     "e" );
 ToFile( "XINERAMA_LINK",     "@XINERAMA_LINK@",    "e" );
--- stax/makefile.mk.old	2008-07-28 13:59:31.000000000 +0200
+++ stax/makefile.mk	2008-08-22 17:40:53.000000000 +0200
@@ -51,6 +51,15 @@ all:
 	@echo "Therefore the files provided here does not need to be built in addition."
 .ENDIF
 
+EXTERNAL_DIR=$(PRJ)$/..$/external/StAX
+
+.IF "$(BUILD_STAX)" != "YES"
+
+$(OUT)$/class$/jsr173_1.0_api.jar : $(EXTERNAL_DIR)$/jsr173_1.0_api.jar
+	+$(COPY) $< $@
+
+.ELSE 			#  "$(BUILD_STAX)" != "YES"
+
 # --- Files --------------------------------------------------------
 TARFILE_NAME=stax-api-1.0-2-sources
 TARFILE_ROOTDIR=src
@@ -66,6 +75,7 @@ OUT2CLASS=jsr173_1.0_api.jar
 
 OUT2CLASS=jsr173_1.0_api.jar
 
+.ENDIF			#  "$(BUILD_STAX)" != "YES"
 .ELSE			# $(SOLAR_JAVA)!= ""
 nojava:
 	@echo "Not building $(PRJNAME) because Java is disabled"
