--- config_office/configure.in.javatarget~	2008-07-07 19:58:06.108847292 +0200
+++ config_office/configure.in	2008-07-07 20:04:44.662095262 +0200
@@ -559,6 +559,17 @@ AC_ARG_WITH(java,
                           no support for Java components, applets, accessibility
                           or XML filters. 
 ], if test "$withval" = "yes"; then WITH_JAVA=java; else WITH_JAVA=$withval; fi, WITH_JAVA=java)
+AC_ARG_WITH(java_target_version,
+[  --with-java-target-version   Generate class files that will work on JVMs with
+			  the specified version. For example, use
+			  --with-java-target-version=1.4 to make sure that the
+			  application will work with JVM 1.4 even when compiled
+			  with JDK 1.5.
+			  
+			  This option is ignored when you compile with gcj/gij.
+ 
+                          Usage: --with-java-target-version=<jvm version>
+],,)
 AC_ARG_ENABLE(gcjaot,
 [  --enable-gcjaot         Build with[[out]] using Ahead of Time java compilation
                           support to speed up buildsi by compiling the jars also
@@ -2730,6 +2741,44 @@ else
    JAVA_HOME=NO_JAVA_HOME ; export JAVA_HOME
 fi
 
+_java_target_ver="1.5"
+dnl ===================================================================
+dnl Check for target java bytecode version
+dnl ===================================================================
+if test "$SOLAR_JAVA" != ""; then
+    AC_MSG_CHECKING([for target java bytecode version])
+    if test "$JDK" = "gcj" -o "$JDK" = "kaffe"; then
+	AC_MSG_RESULT([default by $JDK])
+	if test -n "$with_java_target_version" -a "$with_java_target_version" != "no" ; then
+	    AC_MSG_WARN([Value defined by --with-java-target-version is ignored!])
+	fi
+    else
+	if test -n "$with_java_target_version" -a "$with_java_target_version" != "no" ; then
+	    _java_target_ver="$with_java_target_version"
+	    AC_MSG_RESULT([$_java_target_ver])
+	elif test $_jdk_ver -gt 10000 ; then
+	    _java_target_ver=`echo "$_jdk_ver" | $AWK '{ maj=substr($0,1,1); min=substr($0,2,2); print int(maj)"."int(min) }'`
+	    AC_MSG_RESULT([$_java_target_ver])
+	else
+    	    AC_MSG_ERROR([Unable to guess java bytecode version from java version!])
+	fi
+    fi
+
+    if ! test -z "$_java_target_ver" -o \
+	    "$_java_target_ver" = "1.1" -o \
+            "$_java_target_ver" = "1.2" -o \
+	    "$_java_target_ver" = "1.3" -o \
+	    "$_java_target_ver" = "1.4" -o \
+	    "$_java_target_ver" = "1.5" -o \
+	    "$_java_target_ver" = "1.6" -o \
+	    "$_java_target_ver" = "5" ; then
+    	AC_MSG_ERROR([$_java_target_ver is not supported java bytecode version!])
+    fi
+
+    JAVA_SOURCE_VER="$_java_target_ver"
+    JAVA_TARGET_VER="$_java_target_ver"
+fi
+
 dnl ===================================================================
 dnl Checks for javac
 dnl ===================================================================
@@ -2788,6 +2837,17 @@ if test "$SOLAR_JAVA" != ""; then
 fi
 AC_SUBST(JAVACISGCJ)
 
+JAVACISKAFFE=""
+dnl ===================================================================
+dnl Checks that javac is kaffe
+dnl ===================================================================
+if test "$SOLAR_JAVA" != ""; then
+    if test `$JAVACOMPILER -version 2>&1 | grep -c "Kaffe"` -gt 0; then
+        JAVACISKAFFE="yes"
+    fi
+fi
+AC_SUBST(JAVACISKAFFE)
+
 dnl ===================================================================
 dnl Checks for javadoc
 dnl ===================================================================
@@ -2998,6 +3058,8 @@ fi
 
 AC_SUBST(JAVA_HOME)
 AC_SUBST(JDK)
 AC_SUBST(JAVAFLAGS)
+AC_SUBST(JAVA_SOURCE_VER)
+AC_SUBST(JAVA_TARGET_VER)
 AC_SUBST(JAVAINTERPRETER)
 AC_SUBST(JAVACOMPILER)
--- config_office/set_soenv.in.javatarget~	2008-07-07 19:57:41.109846862 +0200
+++ config_office/set_soenv.in	2008-07-07 20:04:44.665095643 +0200
@@ -1842,6 +1842,7 @@ ToFile( "DMAKEROOT",         $DMAKEROOT,
 if ( $JDK ne "gcj" ) {
    ToFile( "CLASSPATH",         $CLASSPATH,         "e" );
    ToFile( "XCLASSPATH",        $XCLASSPATH,        "e" );
+   ToFile( "JAVACISKAFFE",     '@JAVACISKAFFE@',    "e" );
 }
 else {
    ToFile( "JAVACISGCJ",       '@JAVACISGCJ@',      "e" );
@@ -1852,6 +1853,8 @@ if ( $GUI eq "UNX" ) {
 if ( '@JDK@' ne '' )
 { 
    ToFile( "JDK",             "@JDK@",           "e" );
    ToFile( "JAVAFLAGS",             "@JAVAFLAGS@",           "e" );
+   ToFile( "JAVA_SOURCE_VER", "@JAVA_SOURCE_VER@","e" );
+   ToFile( "JAVA_TARGET_VER", "@JAVA_TARGET_VER@","e" );
    ToFile( "JAVAINTERPRETER", PathFormat("@JAVAINTERPRETER@"), "e" );
    ToFile( "JAVACOMPILER",    PathFormat("@JAVACOMPILER@"), "e" );
--- solenv/inc/antsettings.mk.old	2008-04-10 19:33:24.000000000 +0200
+++ solenv/inc/antsettings.mk	2008-08-25 13:52:31.000000000 +0200
@@ -44,6 +44,18 @@ PATH!:=$(ANT_HOME)$/bin$(PATH_SEPERATOR)
 ANT*:=$(ANT_HOME)$/bin$/ant
 ANT_BUILDFILE*=build.xml
 
+.IF "$(ANT_COMPILER_FLAGS)"==""
+.IF "$(JAVACISGCJ)" == "yes"
+ANT_COMPILER_FLAGS=-Dbuild.compiler=gcj
+.ENDIF
+.ENDIF
+
+.IF "$(ANT_JAVA_VER_FLAGS)"==""
+.IF "$(JDK)" != "gcj" && $(JAVACISKAFFE) != "yes"
+ANT_JAVA_VER_FLAGS=-Dant.build.javac.source=$(JAVA_SOURCE_VER) -Dant.build.javac.target=$(JAVA_TARGET_VER)
+.ENDIF
+.ENDIF
+
 .IF "$(ANT_DEBUG)"==""
 .IF "$(debug)"==""
 ANT_DEBUG=off
@@ -65,24 +77,14 @@ JAVA_HOME=
 .EXPORT : JAVA_HOME
 .ENDIF
 
-.IF "$(JAVACISGCJ)" == "yes"
-ANT_FLAGS!:=-Dbuild.compiler=gcj -Dprj=$(PRJ) -Dprjname=$(PRJNAME) -Ddebug=$(ANT_DEBUG) \
- -Doptimize=$(ANT_OPT) -Dtarget=$(TARGET) -Dsolar.update=on -Dout=$(OUT) -Dinpath=$(INPATH) \
- -Dproext="$(PROEXT)" -Dsolar.bin=$(SOLARBINDIR) -Dsolar.jar=$(SOLARBINDIR) \
- -Dsolar.doc=$(SOLARDOCDIR) -Dcommon.jar=$(SOLARCOMMONBINDIR) \
+ANT_FLAGS!:=$(ANT_COMPILER_FLAGS) -Dprj=$(PRJ) -Dprjname=$(PRJNAME) $(ANT_JAVA_VER_FLAGS) \
+ -Ddebug=$(ANT_DEBUG) -Doptimize=$(ANT_OPT) -Dtarget=$(TARGET) -Dsolar.update=on \
+ -Dout=$(OUT) -Dinpath=$(INPATH) -Dproext="$(PROEXT)" -Dsolar.bin=$(SOLARBINDIR) \
+ -Dsolar.jar=$(SOLARBINDIR) -Dsolar.doc=$(SOLARDOCDIR) -Dcommon.jar=$(SOLARCOMMONBINDIR) \
  -Dcommon.doc=$(SOLARCOMMONDOCDIR) -Dsolar.sourceversion=$(SOURCEVERSION) \
  -Dsolar.lastminor=$(LAST_MINOR) -Dsolar.build=$(BUILD) -f $(ANT_BUILDFILE) $(ANT_FLAGS) -emacs
-.ELSE
-ANT_FLAGS!:=-Dprj=$(PRJ) -Dprjname=$(PRJNAME) -Ddebug=$(ANT_DEBUG) -Doptimize=$(ANT_OPT) \
- -Dtarget=$(TARGET) -Dsolar.update=on -Dout=$(OUT) -Dinpath=$(INPATH) -Dproext="$(PROEXT)" \
- -Dsolar.bin=$(SOLARBINDIR) -Dsolar.jar=$(SOLARBINDIR) -Dsolar.doc=$(SOLARDOCDIR) \
- -Dcommon.jar=$(SOLARCOMMONBINDIR) -Dcommon.doc=$(SOLARCOMMONDOCDIR) \
- -Dsolar.sourceversion=$(SOURCEVERSION) -Dsolar.lastminor=$(LAST_MINOR) \
- -Dsolar.build=$(BUILD) -f $(ANT_BUILDFILE) $(ANT_FLAGS) -emacs
-.ENDIF
+
 .ELSE # No java
 ANT=
 ANT_FLAGS=
 .ENDIF
- 
- 
--- solenv/inc/settings.mk.javatarget~	2008-07-07 19:57:31.071845803 +0200
+++ solenv/inc/settings.mk	2008-07-07 20:04:44.669096361 +0200
@@ -157,6 +157,9 @@ JAVAI:=$(JAVAINTERPRETER)
 .IF "$(JAVACISGCJ)" == "yes"
 JAVAC+=--encoding=UTF-8 -O2 -fno-assert -Wno-deprecated -C
 .ENDIF
+.IF "$(JDK)" != "gcj" && $(JAVACISKAFFE) != "yes"
+JAVAC+=-source $(JAVA_SOURCE_VER) -target $(JAVA_TARGET_VER)
+.ENDIF
 
 #classpath and response
 .IF "$(JDK)" == "J++"
--- hsqldb/makefile.mk.old	2008-05-14 11:31:19.000000000 +0200
+++ hsqldb/makefile.mk	2008-08-25 16:11:36.000000000 +0200
@@ -67,7 +67,7 @@ JAVA_HOME=
 .EXPORT : JAVA_HOME
 BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dbuild.compiler=gcj -f $(ANT_BUILDFILE) jar
 .ELSE
-BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -f $(ANT_BUILDFILE) jar
+BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dant.build.javac.source=$(JAVA_SOURCE_VER) -Dant.build.javac.target=$(JAVA_TARGET_VER) -f $(ANT_BUILDFILE) jar
 .ENDIF
 
 .ENDIF # $(SOLAR_JAVA)!= ""
--- rhino/makefile.mk.old	2008-04-11 09:01:54.000000000 +0200
+++ rhino/makefile.mk	2008-08-25 16:15:08.000000000 +0200
@@ -55,7 +55,7 @@ JAVA_HOME=
 .EXPORT : JAVA_HOME
 BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dbuild.compiler=gcj jar
 .ELSE
-BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" jar
+BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dant.build.javac.source=$(JAVA_SOURCE_VER) -Dant.build.javac.target=$(JAVA_TARGET_VER) jar
 .ENDIF
 
 # --- Targets ------------------------------------------------------
--- jfreereport/java/sac/makefile.mk.old	2008-07-17 15:36:54.000000000 +0200
+++ jfreereport/java/sac/makefile.mk	2008-08-25 16:25:41.000000000 +0200
@@ -61,7 +61,7 @@ JAVA_HOME=
 .EXPORT : JAVA_HOME
 BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dbuild.compiler=gcj -f $(ANT_BUILDFILE) all
 .ELSE
-BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -f $(ANT_BUILDFILE) all
+BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dant.build.javac.source=$(JAVA_SOURCE_VER) -Dant.build.javac.target=$(JAVA_TARGET_VER) -f $(ANT_BUILDFILE) all
 .ENDIF
 
 .ENDIF # $(SOLAR_JAVA)!= ""
