Index: source/pdfimport/test/makefile.mk
===================================================================
RCS file: /cvs/framework/filter/source/pdfimport/test/makefile.mk,v
retrieving revision 1.5
diff -u -r1.5 makefile.mk
--- filter/source/pdfimport/test/makefile.mk	11 Apr 2008 10:51:44 -0000	1.5
+++ filter/source/pdfimport/test/makefile.mk	17 Apr 2008 07:53:23 -0000
@@ -41,6 +41,11 @@
 
 # --- test lib ------------------------------------------------------
 
+.IF "$(ENABLE_PDFIMPORT)" == "NO"
+@all:
+        @echo "PDF Import extension disabled."
+.ENDIF
+
 .IF "$(SYSTEM_ZLIB)" == "YES"
 CFLAGS+=-DSYSTEM_ZLIB
 .ENDIF
Index: source/pdfimport/test/testdocs/makefile.mk
===================================================================
RCS file: /cvs/framework/filter/source/pdfimport/test/testdocs/makefile.mk,v
retrieving revision 1.3
diff -u -r1.3 makefile.mk
--- filter/source/pdfimport/test/testdocs/makefile.mk	11 Apr 2008 10:53:50 -0000	1.3
+++ filter/source/pdfimport/test/testdocs/makefile.mk	17 Apr 2008 07:53:23 -0000
@@ -46,6 +46,7 @@
 	verticaltext.pdf
 
 # --- Fake uno bootstrap ------------------------
+.IF "$(ENABLE_PDFIMPORT)" != "NO"
 
 $(BIN)$/pdfi_unittest_test.ini : makefile.mk
 	rm -f $@
@@ -80,3 +81,5 @@
 $(foreach,i,$(TESTFILES:s/.pdf/_pdfi_unittest_writer_succeeded/:f) $(MISC)$/$i)  : $(BIN)$/pdfi_unittest_test.ini $(BIN)$/pdf2xml$(EXECPOST)
 $(foreach,i,$(TESTFILES:s/.pdf/_pdfi_unittest_impress_succeeded/:f) $(MISC)$/$i) : $(BIN)$/pdfi_unittest_test.ini $(BIN)$/pdf2xml$(EXECPOST)
 $(foreach,i,$(TESTFILES:s/.pdf/_pdfi_unittest_draw_succeeded/:f) $(MISC)$/$i)    : $(BIN)$/pdfi_unittest_test.ini $(BIN)$/pdf2xml$(EXECPOST)
+
+.ENDIF
