--- wizards/com/sun/star/wizards/ui/FilterComponent.java.old	2009-02-12 14:53:19.000000000 +0100
+++ wizards/com/sun/star/wizards/ui/FilterComponent.java	2009-02-18 17:14:54.000000000 +0100
@@ -692,10 +692,10 @@ public class FilterComponent
                 else if (AnyConverter.isBoolean(_filtercondition.Value))
                 {
                     boolean b = AnyConverter.toBoolean(_filtercondition.Value);
-                    double nValue = 0.0;
+                    Double nValue = new Double(0.0);
                     if (b)
                     {
-                        nValue = 1.0;
+                        nValue = new Double(1.0);
                     }
                     Helper.setUnoPropertyValue(UnoDialog.getModel(ControlElements[SOTXTVALUE]), "EffectiveValue", nValue);
                 }
