diff --git a/config_office/configure.in b/config_office/configure.in
index b2faf3a..68e4904 100644
--- config_office/configure.in
+++ config_office/configure.in
@@ -2369,6 +2369,7 @@ AC_SUBST(USE_SYSTEM_STL)
 dnl ===================================================================
 dnl system stl sanity tests
 dnl ===================================================================
+HAVE_GCC_VISIBILITY_BROKEN=
 if test "$USE_SYSTEM_STL" = "YES"; then
    AC_MSG_CHECKING([if hash_map will be in __gnu_cxx namespace])
    AC_LANG_PUSH([C++])
@@ -2405,9 +2406,9 @@ using namespace std;
       gccvisok=no)
       AC_MSG_RESULT([$gccvisok])
       if test "$gccvisok" = "no"; then
-         AC_MSG_WARN([Your gcc is not -fvisibility-inlines-hidden safe. Disabling visibility])
-         echo "Your gcc is not -fvisibility-inlines-hidden safe. Disabling visibility" >> warn
-         unset HAVE_GCC_VISIBILITY_FEATURE
+         AC_MSG_WARN([Your gcc is not -fvisibility-inlines-hidden safe, disabling that.])
+         echo "Your gcc is not -fvisibility-inlines-hidden safe, disabling that." >> warn
+         HAVE_GCC_VISIBILITY_BROKEN="TRUE"
       fi
 
       LDFLAGS=$sharedlink_ldflags_save
@@ -2442,6 +2442,7 @@ using namespace std;
 fi

 AC_SUBST(HAVE_GCC_VISIBILITY_FEATURE)
+AC_SUBST(HAVE_GCC_VISIBILITY_BROKEN)
 
 dnl ===================================================================
 dnl allocator
diff --git a/config_office/set_soenv.in b/config_office/set_soenv.in
index f1538b9..488597a 100644
--- config_office/set_soenv.in
+++ config_office/set_soenv.in
@@ -1917,6 +1917,8 @@ ToFile( "JAVAHOME",          $JAVAHOME,          "e" );
 ToFile( "CC",                $CC,                "e" );
 ToFile( "HAVE_GCC_VISIBILITY_FEATURE",
 		"@HAVE_GCC_VISIBILITY_FEATURE@", "e" );
+ToFile( "HAVE_GCC_VISIBILITY_BROKEN",
+		"@HAVE_GCC_VISIBILITY_BROKEN@", "e" );
 ToFile( "HAVE_LD_HASH_STYLE","@HAVE_LD_HASH_STYLE@","e" );
 ToFile( "HAVE_LD_BDIRECT",   "@HAVE_LD_BDIRECT@","e" );
 ToFile( "CXX",               $CXX,               "e" );
diff --git a/solenv/inc/unxlngx6.mk b/solenv/inc/unxlngx6.mk
index 4afa0b6..91d9bf1 100644
--- solenv/inc/unxlngx6.mk
+++ solenv/inc/unxlngx6.mk
@@ -101,7 +101,7 @@ CFLAGSCXX+= -fno-use-cxa-atexit
 CFLAGSCXX+= -Wno-ctor-dtor-privacy
 CFLAGSCXX+= -fno-use-cxa-atexit
 PICSWITCH:=-fpic
-.IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
+.IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE" && "$(HAVE_GCC_VISIBILITY_BROKEN)" != "TRUE"
 CFLAGSCXX += -fvisibility-inlines-hidden
 .ENDIF # "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
 
