--- transex3/inc/helpmerge.hxx	2008-06-16 13:11:36.000000000 +0200
+++ transex3/inc/helpmerge.hxx	2008-06-16 13:11:51.000000000 +0200
@@ -75,9 +75,9 @@ public:
     bool Merge( const ByteString &rSDFFile, const ByteString &rPathX , const ByteString &rPathY , bool bISO 
 		, const std::vector<ByteString>& aLanguages , MergeDataFile& aMergeDataFile , bool bCreateDir );
 
-private:
     static ByteString makeAbsolutePath( const ByteString& sHelpFile , const ByteString& rRoot_in );
 
+private:
     ByteString GetOutpath( const ByteString& rPathX , const ByteString& sCur , const ByteString& rPathY );
     bool MergeSingleFile( XMLFile* file , MergeDataFile& aMergeDataFile , const ByteString& sLanguage , ByteString sPath );
 
--- transex3/layout/tralay.cxx	2008-06-16 12:54:38.000000000 +0200
+++ transex3/layout/tralay.cxx	2008-06-16 13:11:09.000000000 +0200
@@ -265,13 +265,14 @@ static void insertMarker( XMLParentNode 
 void TranslateLayout::MergeLanguage( ByteString const& language )
 {
     ByteString xmlFile = mFiles.front();
+    ByteString absoluteFilePath = HelpParser::makeAbsolutePath( xmlFile, mRoot );
 
     MergeDataFile mergeData( mLocalize, xmlFile,
                              FALSE, RTL_TEXTENCODING_MS_1252 );
 
     SimpleXMLParser aParser;
     LayoutXMLFile* layoutXml = new LayoutXMLFile( mMergeMode );
-    if ( !aParser.Execute( STRING( xmlFile ), layoutXml ) )
+    if ( !aParser.Execute( STRING( absoluteFilePath ), STRING( xmlFile ), layoutXml ) )
     {
         fprintf(stderr, "error: parsing: %s\n", xmlFile.GetBuffer() );
         return;
