--- cairo/pixman/makefile.mk	2009-03-05 10:22:54.000000000 +0100
+++ cairo/pixman/makefile.mk	2009-03-05 10:23:25.000000000 +0100
@@ -89,7 +89,7 @@
 .ELSE
 # ----------- Unix ---------------------------------------------------------
 .IF "$(OS)$(COM)"=="LINUXGCC" || "$(OS)$(COM)"=="FREEBSDGCC"
-LDFLAGS:=-Wl,-rpath,'$$$$ORIGIN:$$$$ORIGIN/../ure-link/lib' -Wl,-noinhibit-exec -Wl,-z,noexecstack
+LDFLAGS:=-Wl,-rpath,'$$$$ORIGIN:$$$$ORIGIN/../ure-link/lib' -Wl,-z,noexecstack
 .ENDIF                  # "$(OS)$(COM)"=="LINUXGCC"

 .IF "$(OS)$(COM)"=="SOLARISC52"
@@ -135,7 +135,7 @@
 OUT2LIB+=pixman$/release$/*.lib
 .ENDIF
 .ELSE
-OUT2LIB+=pixman$/.libs$/libpixman-1.so
+OUT2LIB+=pixman$/.libs$/libpixman-1.so*
 .ENDIF
 
 # --- Targets ------------------------------------------------------
