diff -rup canvas/source/cairo-old/cairo_cairo.hxx canvas/source/cairo/cairo_cairo.hxx
--- canvas/source/cairo-old/cairo_cairo.hxx	2008-04-29 14:25:54.000000000 +0200
+++ canvas/source/cairo/cairo_cairo.hxx	2008-04-29 14:27:39.000000000 +0200
@@ -47,13 +47,11 @@ namespace cairo {
 #if defined QUARTZ
         // nothing needed
 #elif defined (CAIRO_HAS_XLIB_SURFACE)
-    #ifdef USE_CAIRO10_APIS
 		const void* mpSysData;
 		void* mpDisplay;
 		long mhDrawable;
 		void *mpRenderFormat;
 		bool mbFreePixmap;
-    #endif
 #elif defined (CAIRO_HAS_WIN32_SURFACE)
         // nothing needed
 #endif
@@ -93,7 +91,6 @@ namespace cairo {
 
         
 #if defined(CAIRO_HAS_XLIB_SURFACE) && !defined (QUARTZ)
-    #if defined USE_CAIRO10_APIS
 		long getPixmap()
 		{
 			return mhDrawable;
@@ -108,7 +105,6 @@ namespace cairo {
         {
             mbFreePixmap = false;
         }
-    #endif
 
         int getXFormat( Content aContent );
             
