diff --git sc/source/core/data/dpobject.cxx sc/source/core/data/dpobject.cxx
index c0ff01a..e20a6ba 100644
--- sc/source/core/data/dpobject.cxx
+++ sc/source/core/data/dpobject.cxx
@@ -56,6 +56,7 @@
 #include "scitems.hxx"
 #include "unonames.hxx"
 
+#include <com/sun/star/beans/XPropertySet.hpp>
 #include <com/sun/star/sheet/GeneralFunction.hpp>
 #include <com/sun/star/sheet/DataPilotFieldFilter.hpp>
 #include <com/sun/star/sheet/DataPilotFieldOrientation.hpp>
@@ -84,6 +85,7 @@ using ::com::sun::star::uno::Any;
 using ::com::sun::star::sheet::DataPilotTableHeaderData;
 using ::com::sun::star::sheet::DataPilotTablePositionData;
 using ::rtl::OUString;
+using ::com::sun::star::beans::XPropertySet;
 
 // -----------------------------------------------------------------------
 
@@ -887,17 +889,25 @@ void ScDPObject::FillPageList( TypedStrCollection& rStrings, long nField )
 	uno::Sequence<rtl::OUString> aNames = xMembers->getElementNames();
 	long nNameCount = aNames.getLength();
 	const rtl::OUString* pNameArr = aNames.getConstArray();
-	for (long nPos=0; nPos<nNameCount; nPos++)
+	for (long nPos = 0; nPos < nNameCount; ++nPos)
 	{
-		TypedStrData* pData = new TypedStrData( pNameArr[nPos] );
-
-//		if ( !rStrings.Insert( pData ) )
-//			delete pData;								// duplicate
+        // Make sure to insert only visible members.
+        Reference<XPropertySet> xPropSet(xMembers->getByName(pNameArr[nPos]), UNO_QUERY);
+        sal_Bool bVisible = false;
+        if (xPropSet.is())
+        {
+            Any any = xPropSet->getPropertyValue(OUString::createFromAscii(SC_UNO_ISVISIBL));
+            any >>= bVisible;
+        }
 
-		// use the order from getElementNames
-		if ( !rStrings.AtInsert( rStrings.GetCount(), pData ) )
-			delete pData;
-	}
+        if (bVisible)
+        {
+            // use the order from getElementNames
+            TypedStrData* pData = new TypedStrData( pNameArr[nPos] );
+            if ( !rStrings.AtInsert( rStrings.GetCount(), pData ) )
+                delete pData;
+        }
+    }
 
 	//	add "-all-" entry to the top (unsorted)
 	TypedStrData* pAllData = new TypedStrData( String( ScResId( SCSTR_ALL ) ) );	//! separate string? (also output)
