diff --git sc/inc/dpgroup.hxx sc/inc/dpgroup.hxx
index 0aab7c8..e98a248 100644
--- sc/inc/dpgroup.hxx
+++ sc/inc/dpgroup.hxx
@@ -33,6 +33,7 @@
 
 #include <vector>
 #include <hash_set>
+#include <boost/shared_ptr.hpp>
 
 #include "dptabdat.hxx"
 #include "scdllapi.h"
@@ -183,7 +183,7 @@ class ScDPGroupTableData : public ScDPTableData
 {
     typedef ::std::hash_set< ::rtl::OUString, ::rtl::OUStringHash, ::std::equal_to< ::rtl::OUString > > StringHashSet;
 
-    ScDPTableData*          pSourceData;
+    ::boost::shared_ptr<ScDPTableData> pSourceData;
     long                    nSourceCount;
     ScDPGroupDimensionVec   aGroups;
     ScDPNumGroupDimension*  pNumGroups;     // array[nSourceCount]
@@ -201,7 +201,7 @@ class ScDPGroupTableData : public ScDPTableData
 
 public:
                 // takes ownership of pSource
-                ScDPGroupTableData( ScDPTableData* pSource, ScDocument* pDocument );
+                ScDPGroupTableData( const ::boost::shared_ptr<ScDPTableData>& pSource, ScDocument* pDocument );
     virtual     ~ScDPGroupTableData();
 
     void        AddGroupDimension( const ScDPGroupDimension& rGroup );
diff --git sc/inc/dpobject.hxx sc/inc/dpobject.hxx
index 854e1da..503c531 100644
--- sc/inc/dpobject.hxx
+++ sc/inc/dpobject.hxx
@@ -38,7 +38,7 @@
 #include "dpoutput.hxx"
 #include <com/sun/star/sheet/XDimensionsSupplier.hpp>
 
-#include <memory>
+#include <boost/shared_ptr.hpp>
 
 //------------------------------------------------------------------
 
@@ -104,7 +104,7 @@ private:
 	ScSheetSourceDesc*		pSheetDesc;		//	for sheet data
 	ScImportSourceDesc* 	pImpDesc;		//	for database data
 	ScDPServiceDesc*		pServDesc;		//	for external service
-    ::std::auto_ptr<ScDPTableData>  mpTableData;
+    ::boost::shared_ptr<ScDPTableData>  mpTableData;
 											// cached data
 	com::sun::star::uno::Reference<com::sun::star::sheet::XDimensionsSupplier> xSource;
 	ScDPOutput*				pOutput;
diff --git sc/source/core/data/dpgroup.cxx sc/source/core/data/dpgroup.cxx
index 8b4bb57..cbb03ba 100644
--- sc/source/core/data/dpgroup.cxx
+++ sc/source/core/data/dpgroup.cxx
@@ -70,6 +70,7 @@ using ::rtl::OUStringHash;
 using ::std::vector;
 using ::std::hash_set;
 using ::std::hash_map;
+using ::boost::shared_ptr;
 
 #define D_TIMEFACTOR              86400.0
 
@@ -978,7 +978,7 @@ String lcl_GetNumGroupForValue( double fValue, const ScDPNumGroupInfo& rInfo, bo
     return lcl_GetNumGroupName( fGroupStart, rInfo, bHasNonInteger, cDecSeparator, pFormatter );
 }
 
-ScDPGroupTableData::ScDPGroupTableData( ScDPTableData* pSource, ScDocument* pDocument ) :
+ScDPGroupTableData::ScDPGroupTableData( const shared_ptr<ScDPTableData>& pSource, ScDocument* pDocument ) :
     ScDPTableData(pDocument),
     pSourceData( pSource ),
     pDoc( pDocument )
@@ -993,7 +993,6 @@ ScDPGroupTableData::ScDPGroupTableData( ScDPTableData* pSource, ScDocument* pDoc
 ScDPGroupTableData::~ScDPGroupTableData()
 {
     delete[] pNumGroups;
-    delete pSourceData;
 }
 
 void ScDPGroupTableData::AddGroupDimension( const ScDPGroupDimension& rGroup )
diff --git sc/source/core/data/dpobject.cxx sc/source/core/data/dpobject.cxx
index 7e977f0..cd4dcd1 100644
--- sc/source/core/data/dpobject.cxx
+++ sc/source/core/data/dpobject.cxx
@@ -78,7 +78,7 @@
 
 using namespace com::sun::star;
 using ::std::vector;
-using ::std::auto_ptr;
+using ::boost::shared_ptr;
 using ::com::sun::star::uno::Sequence;
 using ::com::sun::star::uno::Reference;
 using ::com::sun::star::uno::UNO_QUERY;
@@ -163,7 +163,7 @@ ScDPObject::ScDPObject( ScDocument* pD ) :
 	pSheetDesc( NULL ),
 	pImpDesc( NULL ),
 	pServDesc( NULL ),
-    mpTableData(NULL),
+    mpTableData(static_cast<ScDPTableData*>(NULL)),
 	pOutput( NULL ),
 	nAutoFormatIndex( 65535 ),
 	bSettingsChanged( FALSE ),
@@ -186,7 +186,7 @@ ScDPObject::ScDPObject(const ScDPObject& r) :
 	pSheetDesc( NULL ),
 	pImpDesc( NULL ),
 	pServDesc( NULL ),
-    mpTableData(NULL), 
+    mpTableData(static_cast<ScDPTableData*>(NULL)), 
 	pOutput( NULL ),
 	bSettingsChanged( FALSE ),
 	bAlive( FALSE ),
@@ -414,7 +414,7 @@ void ScDPObject::CreateOutput()
 
 ScDPTableData* ScDPObject::GetTableData()
 {
-    if (!mpTableData.get())
+    if (!mpTableData)
     {
         if ( pImpDesc )
         {
@@ -435,7 +435,7 @@ ScDPTableData* ScDPObject::GetTableData()
         // grouping (for cell or database data)
         if ( pSaveData && pSaveData->GetExistingDimensionData() )
         {
-            auto_ptr<ScDPGroupTableData> pGroupData(new ScDPGroupTableData(mpTableData.release(), pDoc));
+            shared_ptr<ScDPGroupTableData> pGroupData(new ScDPGroupTableData(mpTableData, pDoc));
             pSaveData->GetExistingDimensionData()->WriteToData(*pGroupData);
             mpTableData = pGroupData;
         }
