diff --git oox/inc/oox/xls/viewsettings.hxx oox/inc/oox/xls/viewsettings.hxx
index 2294475..8b9c604 100644
--- oox/inc/oox/xls/viewsettings.hxx
+++ oox/inc/oox/xls/viewsettings.hxx
@@ -83,6 +83,7 @@ struct OoxSheetViewData
     bool                mbShowZeros;                    /// True = show zero value zells.
     bool                mbShowOutline;                  /// True = show outlines.
     bool                mbZoomToFit;                    /// True = zoom chart sheet to fit window.
+    Color               maTabColor;                     /// Sheet tab color.
 
     explicit            OoxSheetViewData();
 
@@ -112,6 +113,8 @@ public:
 
     /** Imports the sheetView element containing sheet view settings. */
     void                importSheetView( const AttributeList& rAttribs );
+    /** Imports the tabcolor element containing tab color settings. */
+    void                importTabColor( const AttributeList& rAttribs );
     /** Imports the pane element containing sheet pane settings. */
     void                importPane( const AttributeList& rAttribs );
     /** Imports the selection element containing selection settings for a pane. */
@@ -121,6 +124,8 @@ public:
 
     /** Imports the SHEETVIEW record containing sheet view settings. */
     void                importSheetView( RecordInputStream& rStrm );
+    /** Imports the tabcolor element containing tab color settings. */
+    void                importTabColor( RecordInputStream& rStrm );
     /** Imports the PANE record containing sheet pane settings. */
     void                importPane( RecordInputStream& rStrm );
     /** Imports the SELECTION record containing selection settings for a pane. */
diff --git oox/inc/oox/xls/worksheetsettings.hxx oox/inc/oox/xls/worksheetsettings.hxx
index 083f2cc..0c0b91d 100644
--- oox/inc/oox/xls/worksheetsettings.hxx
+++ oox/inc/oox/xls/worksheetsettings.hxx
@@ -43,7 +43,6 @@ namespace xls {
 struct OoxSheetPrData
 {
     ::rtl::OUString     maCodeName;             /// VBA module codename.
-    Color               maTabColor;             /// Sheet tab color.
     bool                mbFilterMode;           /// True = sheet contains active filter.
     bool                mbApplyStyles;          /// True = automatic styles when creating outlines.
     bool                mbSummaryBelow;         /// True = row outline symbols below group.
diff --git oox/source/xls/chartsheetfragment.cxx oox/source/xls/chartsheetfragment.cxx
index d6fd382..b0da1c2 100644
--- oox/source/xls/chartsheetfragment.cxx
+++ oox/source/xls/chartsheetfragment.cxx
@@ -68,6 +68,8 @@ ContextWrapper OoxChartsheetFragment::onCreateContext( sal_Int32 nElement, const
                     (nElement == XLS_TOKEN( headerFooter )) ||
                     (nElement == XLS_TOKEN( picture )) ||
                     (nElement == XLS_TOKEN( drawing ));
+        case XLS_TOKEN( sheetPr ):
+            return  (nElement == XLS_TOKEN( tabColor ));
         case XLS_TOKEN( sheetViews ):
             return  (nElement == XLS_TOKEN( sheetView ));
         case XLS_TOKEN( headerFooter ):
@@ -86,6 +88,7 @@ void OoxChartsheetFragment::onStartElement( const AttributeList& rAttribs )
     switch( getCurrentElement() )
     {
         case XLS_TOKEN( sheetPr ):          getWorksheetSettings().importChartSheetPr( rAttribs );              break;
+        case XLS_TOKEN( tabColor ):         getSheetViewSettings().importTabColor( rAttribs );                  break;
         case XLS_TOKEN( sheetProtection ):  getWorksheetSettings().importChartProtection( rAttribs );           break;
         case XLS_TOKEN( sheetView ):        getSheetViewSettings().importChartSheetView( rAttribs );            break;
         case XLS_TOKEN( pageMargins ):      getPageSettings().importPageMargins( rAttribs );                    break;
diff --git oox/source/xls/viewsettings.cxx oox/source/xls/viewsettings.cxx
index 6126086..e19fa0d 100644
--- oox/source/xls/viewsettings.cxx
+++ oox/source/xls/viewsettings.cxx
@@ -183,6 +183,7 @@ const sal_Char* const sppcSheetNames[] =
     "ZoomType",
     "ZoomValue",
     "PageViewZoomValue",
+    "TabColor",
     0
 };
 
@@ -295,6 +296,12 @@ void SheetViewSettings::importSheetView( const AttributeList& rAttribs )
     rData.mbShowOutline     = rAttribs.getBool( XML_showOutlineSymbols, true );
 }
 
+void SheetViewSettings::importTabColor( const AttributeList& rAttribs )
+{
+    OoxSheetViewData& rData = maSheetDatas.empty() ? *createSheetViewData() : *maSheetDatas.back();
+    rData.maTabColor.importColor( rAttribs );
+}
+
 void SheetViewSettings::importPane( const AttributeList& rAttribs )
 {
     OSL_ENSURE( !maSheetDatas.empty(), "SheetViewSettings::importPane - missing view data" );
@@ -362,6 +369,12 @@ void SheetViewSettings::importSheetView( RecordInputStream& rStrm )
     rData.mbShowOutline  = getFlag( nFlags, OOBIN_SHEETVIEW_SHOWOUTLINE );
 }
 
+void SheetViewSettings::importTabColor( RecordInputStream& rStrm )
+{
+    OoxSheetViewData& rData = maSheetDatas.empty() ? *createSheetViewData() : *maSheetDatas.back();
+    rData.maTabColor.importColor( rStrm );
+}
+
 void SheetViewSettings::importPane( RecordInputStream& rStrm )
 {
     OSL_ENSURE( !maSheetDatas.empty(), "SheetViewSettings::importPane - missing view data" );
@@ -632,6 +645,7 @@ void SheetViewSettings::finalizeImport()
         << API_ZOOMTYPE_PERCENT
         << static_cast< sal_Int16 >( xData->getNormalZoom() )
         << static_cast< sal_Int16 >( xData->getPageBreakZoom() )
+        << xData->maTabColor.getColor( *this )
         << xData->maGridColor.getColor( *this )
         << xData->isPageBreakPreview()
         << xData->mbShowFormulas
diff --git oox/source/xls/worksheetfragment.cxx oox/source/xls/worksheetfragment.cxx
index c939804..302230f 100644
--- oox/source/xls/worksheetfragment.cxx
+++ oox/source/xls/worksheetfragment.cxx
@@ -122,7 +122,7 @@ ContextWrapper OoxWorksheetFragment::onCreateContext( sal_Int32 nElement, const
             case SHEETTYPE_WORKSHEET:   return  (nElement == XLS_TOKEN( worksheet ));
             case SHEETTYPE_CHARTSHEET:  return  false;
             case SHEETTYPE_MACROSHEET:  return  (nElement == XM_TOKEN( macrosheet ));
-            case SHEETTYPE_DIALOGSHEET: return  (nElement == XM_TOKEN( dialogsheet ));
+            case SHEETTYPE_DIALOGSHEET: return  (nElement == XLS_TOKEN( dialogsheet ));
             case SHEETTYPE_MODULESHEET: return  false;
             case SHEETTYPE_EMPTYSHEET:  return  false;
         }
@@ -130,6 +130,7 @@ ContextWrapper OoxWorksheetFragment::onCreateContext( sal_Int32 nElement, const
 
         case XLS_TOKEN( worksheet ):
         case XM_TOKEN( macrosheet ):
+        case XLS_TOKEN( dialogsheet ):
             switch( nElement )
             {
                 case XLS_TOKEN( sheetData ):
@@ -211,7 +212,7 @@ void OoxWorksheetFragment::onStartElement( const AttributeList& rAttribs )
     switch( getCurrentElement() )
     {
         case XLS_TOKEN( sheetPr ):          getWorksheetSettings().importSheetPr( rAttribs );               break;
-        case XLS_TOKEN( tabColor ):         getWorksheetSettings().importTabColor( rAttribs );              break;
+        case XLS_TOKEN( tabColor ):         getSheetViewSettings().importTabColor( rAttribs );              break;
         case XLS_TOKEN( outlinePr ):        getWorksheetSettings().importOutlinePr( rAttribs );             break;
         case XLS_TOKEN( pageSetUpPr ):      importPageSetUpPr( rAttribs );                                  break;
         case XLS_TOKEN( dimension ):        importDimension( rAttribs );                                    break;
diff --git oox/source/xls/worksheethelper.cxx oox/source/xls/worksheethelper.cxx
index e5590f8..89036e3 100644
--- oox/source/xls/worksheethelper.cxx
+++ oox/source/xls/worksheethelper.cxx
@@ -975,6 +975,7 @@ void WorksheetData::finalizeWorksheetImport()
     finalizeMergedRanges();
     maCondFormats.finalizeImport();
     maPageSett.finalizeImport();
+    maSheetSett.finalizeImport();
     maSheetViewSett.finalizeImport();
     convertColumns();
     convertRows();
diff --git oox/source/xls/worksheetsettings.cxx oox/source/xls/worksheetsettings.cxx
index 6ab8975..d30106a 100644
--- oox/source/xls/worksheetsettings.cxx
+++ oox/source/xls/worksheetsettings.cxx
@@ -34,6 +34,7 @@
 #include "oox/helper/recordinputstream.hxx"
 #include "oox/xls/biffinputstream.hxx"
 #include "oox/xls/pagesettings.hxx"
+#include "oox/xls/viewsettings.hxx"
 #include "oox/xls/workbooksettings.hxx"
 
 using ::rtl::OUString;
@@ -138,11 +139,6 @@ void WorksheetSettings::importChartSheetPr( const AttributeList& rAttribs )
     maOoxSheetData.maCodeName = rAttribs.getString( XML_codeName, OUString() );
 }
 
-void WorksheetSettings::importTabColor( const AttributeList& rAttribs )
-{
-    maOoxSheetData.maTabColor.importColor( rAttribs );
-}
-
 void WorksheetSettings::importOutlinePr( const AttributeList& rAttribs )
 {
     maOoxSheetData.mbApplyStyles  = rAttribs.getBool( XML_applyStyles, false );
@@ -187,7 +183,8 @@ void WorksheetSettings::importSheetPr( RecordInputStream& rStrm )
 {
     sal_uInt16 nFlags1;
     sal_uInt8 nFlags2;
-    rStrm >> nFlags1 >> nFlags2 >> maOoxSheetData.maTabColor;
+    rStrm >> nFlags1 >> nFlags2;
+    getSheetViewSettings().importTabColor( rStrm );
     rStrm.skip( 8 );    // sync anchor cell
     rStrm >> maOoxSheetData.maCodeName;
     // sheet settings
@@ -204,7 +201,8 @@ void WorksheetSettings::importSheetPr( RecordInputStream& rStrm )
 void WorksheetSettings::importChartSheetPr( RecordInputStream& rStrm )
 {
     rStrm.skip( 2 );    // flags, contains only the 'published' flag
-    rStrm >> maOoxSheetData.maTabColor >> maOoxSheetData.maCodeName;
+    getSheetViewSettings().importTabColor( rStrm );
+    rStrm >> maOoxSheetData.maCodeName;
 }
 
 void WorksheetSettings::importSheetProtection( RecordInputStream& rStrm )
