diff --git offapi/com/sun/star/sheet/FormulaLanguage.idl offapi/com/sun/star/sheet/FormulaLanguage.idl
index 0326520..07a9582 100644
--- offapi/com/sun/star/sheet/FormulaLanguage.idl
+++ offapi/com/sun/star/sheet/FormulaLanguage.idl
@@ -63,6 +63,10 @@ constants FormulaLanguage
      */
     const long NATIVE   = 3;
 
+    /** Function names and operators as used in the English version of 
+        Excel.
+     */
+    const long XL_ENGLISH = 4;
 };
 
 //=============================================================================
