diff --git sc/source/ui/dbgui/pvlaydlg.cxx sc/source/ui/dbgui/pvlaydlg.cxx
index 056e37e..155f3e0 100644
--- sc/source/ui/dbgui/pvlaydlg.cxx
+++ sc/source/ui/dbgui/pvlaydlg.cxx
@@ -303,20 +303,10 @@ void __EXPORT ScDPLayoutDlg::Init()
 		}
 	}
 
-	if ( thePivotData.nTab != MAXTAB+1 )
-	{
-		String aStr;
-		ScAddress( thePivotData.nCol,
-				   thePivotData.nRow,
-				   thePivotData.nTab ).Format( aStr, STD_FORMAT, pDoc, pDoc->GetAddressConvention() );
-		aEdOutPos.SetText( aStr );
-		EdModifyHdl(0);
-	}
-	else
-	{
-		aLbOutPos.SelectEntryPos( aLbOutPos.GetEntryCount()-1 );
-		SelAreaHdl(NULL);
-	}
+    // Output to a new sheet by default.
+    aLbOutPos.SelectEntryPos(1);
+    aEdOutPos.Disable();
+    aRbOutPos.Disable();
 
 	aBtnIgnEmptyRows.Check( thePivotData.bIgnoreEmptyRows );
 	aBtnDetectCat   .Check( thePivotData.bDetectCategories );
