diff --git offapi/com/sun/star/sheet/DatabaseRange.idl offapi/com/sun/star/sheet/DatabaseRange.idl
index 4bbb5f5..6e66ba4 100644
--- offapi/com/sun/star/sheet/DatabaseRange.idl
+++ offapi/com/sun/star/sheet/DatabaseRange.idl
@@ -154,7 +154,8 @@ published service DatabaseRange
     /** returns the index used to refer to this range in token arrays.
 
         <p>A token describing a database range shall contain the op-code
-        OPCODE_DB_AREA and this index as data part.</p>
+        obtained from the <const>FormulaMapGroupSpecialOffset::DB_AREA</const>
+        and this index as data part.</p>
 
         @see com::sun::star::sheet::FormulaToken
         @see com::sun::star::sheet::FormulaMapGroupSpecialOffset::DB_AREA
diff --git offapi/com/sun/star/sheet/ExternalDocLink.idl offapi/com/sun/star/sheet/ExternalDocLink.idl
new file mode 100644
index 0000000..9c7f70d
--- /dev/null
+++ offapi/com/sun/star/sheet/ExternalDocLink.idl
@@ -0,0 +1,57 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: DDELink.idl,v $
+ * $Revision: 1.9 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_ExternalDocLink_idl__
+#define __com_sun_star_sheet_ExternalDocLink_idl__
+
+#include <com/sun/star/sheet/XExternalDocLink.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** Represents a single external document link. 
+
+    <p>An external document link contains cached data used for external cell
+    and cell range references as well as external range names.</p>
+
+    @see com::sun::star::sheet::XExternalDocLink
+
+    @since OOo 3.1.0
+ */
+service ExternalDocLink : XExternalDocLink
+{
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/ExternalDocLinks.idl offapi/com/sun/star/sheet/ExternalDocLinks.idl
new file mode 100644
index 0000000..105cb8f
--- /dev/null
+++ offapi/com/sun/star/sheet/ExternalDocLinks.idl
@@ -0,0 +1,55 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: DDELink.idl,v $
+ * $Revision: 1.9 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_ExternalDocLinks_idl__
+#define __com_sun_star_sheet_ExternalDocLinks_idl__
+
+#include <com/sun/star/sheet/XExternalDocLinks.idl>
+#include <com/sun/star/sheet/XSpreadsheetDocument.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+/** Represents a collection of external document links.
+
+    <p>An external document link contains cached data used for external cell
+    and cell range references as well as external range names.</p>
+
+    @see com::sun::star::sheet::ExternalDocLink
+    @see com::sun::star::sheet::XExternalDocLinks
+
+    @since OOo 3.1.0
+ */
+service ExternalDocLinks : XExternalDocLinks
+{
+};
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/ExternalLinkInfo.idl offapi/com/sun/star/sheet/ExternalLinkInfo.idl
new file mode 100644
index 0000000..52758ec
--- /dev/null
+++ offapi/com/sun/star/sheet/ExternalLinkInfo.idl
@@ -0,0 +1,81 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: ExternalLinkInfo.idl,v $
+ * $Revision: 1.1.2.2 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+#ifndef __com_sun_star_sheet_ExternalLinkInfo_idl__
+#define __com_sun_star_sheet_ExternalLinkInfo_idl__
+
+#ifndef __com_sun_star_sheet_ExternalLinkType_idl__
+#include <com/sun/star/sheet/ExternalLinkType.idl>
+#endif
+
+//=============================================================================
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+/** describes an external link in a formula.
+
+    @since OOo3.1
+ */
+struct ExternalLinkInfo
+{
+    //-------------------------------------------------------------------------
+
+    /** Link type, one of <type>ExternalLinkType</type> constants.
+     */
+
+    long Type;
+
+    //-------------------------------------------------------------------------
+
+    /** Location of this link type.
+
+        Modes used:
+
+        1.  If <member>Type</member> is <const>ExternalLinkType::EXTERNAL</const>,
+            this member shall contain a string with the <b>URI</b> of a
+            document. The formula that would need this information for example
+            would contain =[1]Sheet1!A1 or ='[1]Sheet name'!A1 where [1] does
+            resolve to the URI contained in this Location member. Note that
+            the quotes cover both, the document name and the sheet name.
+
+        2.  If <member>Type</member> is <type>ExternalLinkType::DDE</type>,
+            this member shall contain a <type>DDELinkType</type> describing
+            service name, topic, and all known items of a DDE link.
+     */
+
+    any Data;
+
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/ExternalLinkType.idl offapi/com/sun/star/sheet/ExternalLinkType.idl
new file mode 100644
index 0000000..7c1b040
--- /dev/null
+++ offapi/com/sun/star/sheet/ExternalLinkType.idl
@@ -0,0 +1,67 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: ExternalLinkType.idl,v $
+ * $Revision: 1.1.2.1 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_ExternalLinkType_idl__
+#define __com_sun_star_sheet_ExternalLinkType_idl__
+
+//=============================================================================
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** Constants designating the link type in
+    <type>ExternalLinkInfo</type>, used with
+    <member>FormulaParser::ExternalLinks</member>.
+
+    @since OOo3.1
+ */
+constants ExternalLinkType
+{
+    /** Unknown element type
+     */
+    const long UNKNOWN  = 0;
+
+    /** URL of an external document.
+     */
+    const long DOCUMENT = 1;
+
+    /** DDE link.
+     */
+    const long DDE      = 2;
+
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
+
diff --git offapi/com/sun/star/sheet/ExternalReference.idl offapi/com/sun/star/sheet/ExternalReference.idl
new file mode 100644
index 0000000..7686310
--- /dev/null
+++ offapi/com/sun/star/sheet/ExternalReference.idl
@@ -0,0 +1,99 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: ExternalLinkInfo.idl,v $
+ * $Revision: 1.1.2.2 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_ExternalReference_idl__
+#define __com_sun_star_sheet_ExternalReference_idl__
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** Data structure to store information about an external reference.  An
+    external reference can be either a single cell reference, a cell range
+    reference, or a named range.
+
+    @see FormulaMapGroupSpecialOffset::PUSH
+
+    @since OOo 3.1
+ */
+struct ExternalReference
+{
+    //-------------------------------------------------------------------------
+
+    /** Index of an externally linked document.  Each externally-linked document
+        has a unique index value.
+
+        <p>You can get the index value of an external document from the
+        corresponding <type scope="com::sun::star::sheet">ExternalDocLink</type>
+        instance through its attribute <type scope="com::sun::star::sheet::ExternalDocLink">TokenIndex</type>.</p>
+
+        @see com::sun::star::sheet::ExternalDocLink
+        @see com::sun::star::sheet::ExternalDocLink::TokenIndex
+     */
+    long Index;
+
+#if 0
+    //-------------------------------------------------------------------------
+
+    /** Name of the sheet that the external reference points to.
+
+        <p>In case of a cell range reference that spans across multiple
+        sheets, this is the name of the first sheet in that range.</p>
+
+        <p>Note that an external range name ignores this value at the moment,
+        but <i>it may make use of this data in the future when Calc supports a
+        sheet-specific range name.</i></p>
+     */
+    string SheetName;
+#endif
+
+    //-------------------------------------------------------------------------
+
+    /** Reference data.
+
+        <p>This can store either <type>SingleReference</type> for a single
+        cell reference, <type>ComplexReference</type> for a cell range
+        reference, or simply a <type>string</type> for a defined name.</p>
+
+        <p>The <member>SingleReference::Sheet</member> member shall contain
+        the index of the external sheet cache containing the values of the
+        externally referenced cells.</p>
+
+        @see com::sun::star::sheet::SingleReference
+        @see com::sun::star::sheet::ComplexReference
+     */
+    any Reference;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/ExternalSheetCache.idl offapi/com/sun/star/sheet/ExternalSheetCache.idl
new file mode 100644
index 0000000..17af156
--- /dev/null
+++ offapi/com/sun/star/sheet/ExternalSheetCache.idl
@@ -0,0 +1,55 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: DDELink.idl,v $
+ * $Revision: 1.9 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_ExternalSheetCache_idl__
+#define __com_sun_star_sheet_ExternalSheetCache_idl__
+
+#include <com/sun/star/sheet/XExternalSheetCache.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+/** A single sheet cache for an external document.
+   
+    <p>This cached data is used for external cell and cell range references,
+    as well as external range names.  An <type scope="com::sun::star::sheet">ExternalDocLink</type>
+    instance contains a set of these sheet caches.</p>
+
+    @see com::sun::star::sheet::ExternalDocLink
+    @see com::sun::star::sheet::XExternalSheetCache
+    
+    @since OOo 3.1.0
+ */
+service ExternalSheetCache : XExternalSheetCache
+{
+};
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/FormulaMapGroup.idl offapi/com/sun/star/sheet/FormulaMapGroup.idl
index c128585..e846d72 100644
--- offapi/com/sun/star/sheet/FormulaMapGroup.idl
+++ offapi/com/sun/star/sheet/FormulaMapGroup.idl
@@ -47,7 +47,10 @@ module com {  module sun {  module star {  module sheet {
  */
 constants FormulaMapGroup
 {
-    /** Group of op-codes without a string symbol */
+    /** Group of op-codes without a string symbol.
+
+        @see FormulaMapGroupSpecialOffset
+      */
     const long SPECIAL            = 0;
 
     /** Separators and parentheses */
diff --git offapi/com/sun/star/sheet/FormulaMapGroupSpecialOffset.idl offapi/com/sun/star/sheet/FormulaMapGroupSpecialOffset.idl
index 20369b2..7ebfe52 100644
--- offapi/com/sun/star/sheet/FormulaMapGroupSpecialOffset.idl
+++ offapi/com/sun/star/sheet/FormulaMapGroupSpecialOffset.idl
@@ -31,37 +31,182 @@
 #ifndef __com_sun_star_sheet_FormulaMapGroupSpecialOffset_idl__
 #define __com_sun_star_sheet_FormulaMapGroupSpecialOffset_idl__
 
-//=============================================================================
+// ===========================================================================
 
 module com {  module sun {  module star {  module sheet {
 
-//=============================================================================
+// ===========================================================================
 
 /** Constants designating the offsets within the sequence returned by
     <member>XFormurmulaOpCodeMapper::getAvailableMappings</member> when
-    called for group <type>FormulaMapGroup</type> SPECIAL.
+    called for group <const>FormulaMapGroup::SPECIAL</const>.
 
     The number of constants may grow in future versions!
  */
 constants FormulaMapGroupSpecialOffset
 {
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe a formula operand token that will be pushed onto the formula
+        stack while the formula is interpreted.
+
+        <p>The <member>FormulaToken::Data</member> member shall contain one of
+        the following values:</p>
+
+        <ul>
+        <li>A value of type <atom>double</atom> for literal floating-point
+            constants.</li>
+        <li>A <atom>string</atom> for literal text.</li>
+        <li>A sequence of sequence of <atom>any</atom> for a literal array.
+            The contained values shall be of type <atom>double</atom> or
+            <atom>string</atom>. Floating-point values and strings may occur
+            together in an array.</li>
+        <li>A struct of type <type>SingleReference</type> for a reference to a
+            single cell in the own document.</li>
+        <li>A struct of type <type>ComplexReference</type> for a reference to
+            a range of cells in the own document.</li>
+        <li>A struct of type <type>ExternalReference</type> for a reference to
+            a cell, a range of cells, or a defined name in an external
+            document.</li>
+        </ul>
+     */
     const long PUSH               = 0;
+
+    // -----------------------------------------------------------------------
+
     const long CALL               = 1;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        instruct the formula interpreter to immediately stop interpreting the
+        formula.
+
+        <p>The <member>FormulaToken::Data</member> member is not used
+        and should be empty.</p>
+     */
     const long STOP               = 2;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe the reference to an external function (e.g. add-in function)
+        used in formulas.
+
+        <p>The <member>FormulaToken::Data</member> member shall contain a
+        <atom>string</atom> with the programmatical name of the function, e.g.
+        "com.sun.star.sheet.addin.Analysis.getEomonth" for the EOMONTH
+        function from the Analsysis add-in.</p>
+     */
     const long EXTERNAL           = 3;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe the reference to a defined name (also known as named range)
+        used in formulas.
+
+        <p>The <member>FormulaToken::Data</member> member shall contain an
+        integer value of type <atom>long</atom> specifying the index of the
+        defined name. This index can be obtained from the defined name using
+        its <type>NamedRange::TokenIndex</type> property.</p>
+
+        @see  NamedRange
+     */
     const long NAME               = 4;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe an invalid name that resolves to the #NAME? error in formulas.
+
+        <p>The <member>FormulaToken::Data</member> member is not used
+        and should be empty.</p>
+     */
     const long NO_NAME            = 5;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe an empty function parameter.
+
+        <p>Example: In the formula =SUM(1;;2) the second parameter is empty
+        and represented by a formula token containing the "empty" op-code.</p>
+
+        <p>The <member>FormulaToken::Data</member> member is not used
+        and should be empty.</p>
+     */
     const long MISSING            = 6;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe "bad" data in a formula, e.g. data the formula parser was not
+        able to parse.
+
+        <p>The <member>FormulaToken::Data</member> member shall contain a
+        <atom>string</string> with the bad data. This string will be displayed
+        in the formula.</p>
+     */
     const long BAD                = 7;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe white-space characters within the string representation of a
+        formula.
+
+        <p>White-space characters in formulas are used for readability and do
+        not affect the result of the formula.</p>
+
+        <p>The <member>FormulaToken::Data</member> member shall contain a
+        positive integer value of type <atom>long</atom> specifying the number
+        of space characters.</p>
+
+        <p>Attention: This may change in next versions to support other
+        characters than simple space characters (e.g. line feeds, horizontal
+        tabulators, non-breakable spaces).</p>
+     */
     const long SPACES             = 8;
+
+    // -----------------------------------------------------------------------
+
     const long MAT_REF            = 9;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe the reference to a database range used in formulas.
+
+        <p>The <member>FormulaToken::Data</member> member shall contain an
+        integer value of type <atom>long</atom> specifying the index of the
+        database range. This index can be obtained from the database range
+        using its <type>DatabaseRange::TokenIndex</type> property.</p>
+
+        @see  DatabaseRange
+     */
     const long DB_AREA            = 10;
+
+    // -----------------------------------------------------------------------
+
+    /** Formula tokens containing the op-code obtained from this offset
+        describe the reference to a macro function called in a formula.
+
+        <p>The <member>FormulaToken::Data</member> member shall contain a
+        <atom>string</atom> specifying the name of the macro function.</p>
+     */
     const long MACRO              = 11;
+
+    // -----------------------------------------------------------------------
+
     const long COL_ROW_NAME       = 12;
-    const long COL_ROW_NAME_AUTO  = 13;
+
+    // -----------------------------------------------------------------------
+
 };
 
-//=============================================================================
+// ===========================================================================
 
 }; }; }; };
 
diff --git offapi/com/sun/star/sheet/FormulaToken.idl offapi/com/sun/star/sheet/FormulaToken.idl
index cec2ea4..c0aff64 100644
--- offapi/com/sun/star/sheet/FormulaToken.idl
+++ offapi/com/sun/star/sheet/FormulaToken.idl
@@ -44,6 +44,8 @@ struct FormulaToken
     //-------------------------------------------------------------------------
 
     /** is the OpCode of the token.
+
+        @see com::sun::star::sheet::XFormulaOpCodeMapper
      */
     long OpCode;
 
diff --git offapi/com/sun/star/sheet/NamedRange.idl offapi/com/sun/star/sheet/NamedRange.idl
index 30af011..c4d79dc 100644
--- offapi/com/sun/star/sheet/NamedRange.idl
+++ offapi/com/sun/star/sheet/NamedRange.idl
@@ -77,8 +77,9 @@ published service NamedRange
 
     /** returns the index used to refer to this name in token arrays.
 
-        <p>A token describing a defined name shall contain the op-code
-        OPCODE_NAME and this index as data part.</p>
+        <p>A token describing a defined name shall contain the op-code obtained
+        from the <const>FormulaMapGroupSpecialOffset::NAME</const> offset and
+        this index as data part.</p>
 
         @see com::sun::star::sheet::FormulaToken
         @see com::sun::star::sheet::FormulaMapGroupSpecialOffset::NAME
diff --git offapi/com/sun/star/sheet/XExternalDocLink.idl offapi/com/sun/star/sheet/XExternalDocLink.idl
new file mode 100644
index 0000000..27a706c
--- /dev/null
+++ offapi/com/sun/star/sheet/XExternalDocLink.idl
@@ -0,0 +1,94 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: DDELink.idl,v $
+ * $Revision: 1.9 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_XExternalDocLink_idl__
+#define __com_sun_star_sheet_XExternalDocLink_idl__
+
+#include <com/sun/star/container/XEnumerationAccess.idl>
+#include <com/sun/star/container/XIndexAccess.idl>
+#include <com/sun/star/container/XNameAccess.idl>
+#include <com/sun/star/sheet/XExternalSheetCache.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** Primary interface for the <type scope="com::sun::star::sheet">ExternalDocLink</type> service.
+   
+    @see com::sun::star::sheet::ExternalDocLink
+
+    @since OOo 3.1.0
+ */
+interface XExternalDocLink
+{
+    interface com::sun::star::container::XNameAccess;
+    interface com::sun::star::container::XIndexAccess;
+    interface com::sun::star::container::XEnumerationAccess;
+
+    //-------------------------------------------------------------------------
+
+    /** <p>This method adds a new sheet cache instance to the extternal document 
+        link for a specified sheet name.  If a sheet cache instance already 
+        exists for the specified name, then the existing instance is returned.</p>
+
+        <p>Note that a sheet name lookup is performed in a case-insensitive 
+        fashion.</p>
+    
+        @param aSheetName sheet name
+
+        @return com::sun::star::sheet::XExternalSheetCache sheet cache instance
+     */
+    com::sun::star::sheet::XExternalSheetCache addSheetCache( [in] string aSheetName );
+
+    //-------------------------------------------------------------------------
+
+    /** Index corresponding to the external document link.
+    
+        <p>This index value corresponds with the external document 
+        represented by an instance of 
+        <type scope="com::sun::star::sheet">ExternalDocLink</type>.  This
+        value is stored within a formula token instance.</p>
+
+        <p>Each external document cache instance has a unique index value, and this
+        index value can be used to retrieve the corresponding external document cache
+        from the parent <type scope="com::sun::star::sheet">ExternalDocLinks</type> instance.</p>
+
+        @see com::sun::star::sheet::ExternalDocLinks
+        @see com::sun::star::sheet::FormulaToken
+        @see com::sun::star::sheet::ExternalReference
+     */
+    [attribute, readonly] long TokenIndex;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/XExternalDocLinks.idl offapi/com/sun/star/sheet/XExternalDocLinks.idl
new file mode 100644
index 0000000..384d405
--- /dev/null
+++ offapi/com/sun/star/sheet/XExternalDocLinks.idl
@@ -0,0 +1,66 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: DDELink.idl,v $
+ * $Revision: 1.9 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_XExternalDocLinks_idl__
+#define __com_sun_star_sheet_XExternalDocLinks_idl__
+
+#include <com/sun/star/sheet/XExternalDocLink.idl>
+#include <com/sun/star/container/XNameAccess.idl>
+#include <com/sun/star/container/XIndexAccess.idl>
+#include <com/sun/star/container/XEnumerationAccess.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+/** Primary interface for the <type scope="com::sun::star::sheet">ExternalDocLinks</type> service.
+   
+    @see com::sun::star::sheet::ExternalDocLinks
+
+    @since OOo 3.1.0
+ */
+interface XExternalDocLinks
+{
+    interface com::sun::star::container::XNameAccess;
+    interface com::sun::star::container::XIndexAccess;
+    interface com::sun::star::container::XEnumerationAccess;
+
+    /** This method adds a new external document link by its URL, and returns 
+        its instance.  If an document instance already exists for the specified URL,
+        then that instance gets returned instead of creating a new one.
+    
+        @param aDocName document URL (e.g. file:///path/to/document.ods)
+       
+        @return com::sun::star::sheet::XExternalDocLink external document link instance
+     */
+    com::sun::star::sheet::XExternalDocLink addDocLink( [in] string aDocName );
+};
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/XExternalSheetCache.idl offapi/com/sun/star/sheet/XExternalSheetCache.idl
new file mode 100644
index 0000000..1d91f70
--- /dev/null
+++ offapi/com/sun/star/sheet/XExternalSheetCache.idl
@@ -0,0 +1,102 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: DDELink.idl,v $
+ * $Revision: 1.9 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_XExternalSheetCache_idl__
+#define __com_sun_star_sheet_XExternalSheetCache_idl__
+
+#include <com/sun/star/lang/IllegalArgumentException.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+/** Primary interface for the <type scope="com::sun::star::sheet">ExternalSheetCache</type> service.
+
+    @see com::sun::star::sheet::ExternalSheetCache
+
+    @since OOo 3.1.0
+ */
+interface XExternalSheetCache
+{
+    /** It sets a cached value for a specified cell position.  The value is expected
+        to be either of type <type>string</type> or of type <type>double</type>.  No
+        other data types are supported.
+
+        @param nRow row position (0-based)
+        @param nColumn column position (0-based)
+        @param aValue cell value to be cached
+     */
+    void setCellValue( [in] long nColumn, [in] long nRow, [in] any aValue )
+            raises (com::sun::star::lang::IllegalArgumentException);
+
+    /** It retrieves a cached value from a specified cell position.  The cached
+        value can be either <type>string</type> or <type>double</type>.
+
+        @return any cached cell value
+     */
+    any getCellValue( [in] long nColumn, [in] long nRow )
+            raises (com::sun::star::lang::IllegalArgumentException);
+
+    /** It returns a list of all row numbers where a cached cell or cells exist.
+        The row numbers are sorted in ascending order.
+
+        @return sequence<long> list of all row numbers with cached cell(s)
+     */
+    sequence< long > getAllRows();
+
+    /** Given a row number, this method returns a list of all columns numbers
+        that store cached cell values in that row.  The column numbers are
+        sorted in ascending order.
+
+        @return sequence<long> list of all columns numbers with cached cell values
+     */
+    sequence< long > getAllColumns( [in] long nRow )
+            raises (com::sun::star::lang::IllegalArgumentException);
+
+    //-------------------------------------------------------------------------
+
+    /** Index corresponding to this instance of an external sheet cache for
+        usage in formula tokens.
+
+        <p>This index to the external sheet cache is expected in the
+        <member>SingleReference::Sheet</member> member if it is part of an
+        external reference token.
+
+        <p>Each external sheet cache has a unique index value inside the
+        <type>ExternalDocLink</type> instance.</p>
+
+        @see FormulaToken
+        @see ExternalReference
+     */
+    [attribute, readonly] long TokenIndex;
+
+};
+
+}; }; }; };
+
+#endif
diff --git offapi/com/sun/star/sheet/makefile.mk offapi/com/sun/star/sheet/makefile.mk
index c1d2844..09b33e7 100644
--- offapi/com/sun/star/sheet/makefile.mk
+++ offapi/com/sun/star/sheet/makefile.mk
@@ -123,6 +123,12 @@ IDLFILES=\
 	DatabaseRangesEnumeration.idl\
 	DDELinkMode.idl\
 	DocumentSettings.idl\
+	ExternalDocLink.idl\
+	ExternalDocLinks.idl\
+	ExternalLinkInfo.idl\
+	ExternalLinkType.idl\
+	ExternalReference.idl\
+	ExternalSheetCache.idl\
 	FillDateMode.idl\
 	FillDirection.idl\
 	FillMode.idl\
@@ -255,6 +261,9 @@ IDLFILES=\
 	XDocumentAuditing.idl\
 	XDrillDownDataSupplier.idl\
 	XEnhancedMouseClickBroadcaster.idl\
+	XExternalDocLink.idl\
+	XExternalDocLinks.idl\
+	XExternalSheetCache.idl\
 	XExternalSheetName.idl\
 	XFillAcrossSheet.idl\
 	XFormulaOpCodeMapper.idl\
