diff --git svx/inc/svx/dialogs.hrc svx/inc/svx/dialogs.hrc
index 716e7eb..a7bbc93 100644
--- svx/inc/svx/dialogs.hrc
+++ svx/inc/svx/dialogs.hrc
@@ -1562,7 +1562,7 @@
 
 #define SVX_OOO_BUILD_START             (RID_SVX_START + 1200)
 
-
+#define SID_SC_TP_FORMULA               (SVX_OOO_BUILD_START + 1)
 
 
 #endif
diff --git svx/source/dialog/treeopt.cxx svx/source/dialog/treeopt.cxx
index cacb4cf..ad44eda 100644
--- svx/source/dialog/treeopt.cxx
+++ svx/source/dialog/treeopt.cxx
@@ -453,6 +453,7 @@ static OptionsMapping_Impl __READONLY_DATA OptionsMap_Impl[] =
     { "Calc",               "View",                 SID_SC_TP_CONTENT },
     { "Calc",               "International",        RID_OFA_TP_INTERNATIONAL },
     { "Calc",               "Calculate",            SID_SC_TP_CALC },
+    { "Calc",               "Formula",              SID_SC_TP_FORMULA },
     { "Calc",               "SortLists",            SID_SC_TP_USERLISTS },
     { "Calc",               "Changes",              SID_SC_TP_CHANGES },
     { "Calc",               "Grid",                 SID_SC_TP_GRID },
diff --git svx/source/dialog/treeopt.src svx/source/dialog/treeopt.src
index bc96c6a..fcf418a 100644
--- svx/source/dialog/treeopt.src
+++ svx/source/dialog/treeopt.src
@@ -241,6 +241,7 @@ Resource RID_OFADLG_OPTIONS_TREE_PAGES
 			< "View" ;			SID_SC_TP_CONTENT			;> ;
 			< "International" ;		RID_OFA_TP_INTERNATIONAL	;> ;
 			< "Calculate" ;			SID_SC_TP_CALC				;> ;
+			< "Formula" ;       SID_SC_TP_FORMULA ;> ;
 			< "Sort Lists" ;		SID_SC_TP_USERLISTS			;> ;
 			< "Changes" ;		SID_SC_TP_CHANGES			;> ;
 			< "Grid" ;			SID_SC_TP_GRID				;> ;
