diff --git sc/source/ui/view/dbfunc3.cxx sc/source/ui/view/dbfunc3.cxx
index 514d512..cc98745 100644
--- sc/source/ui/view/dbfunc3.cxx
+++ sc/source/ui/view/dbfunc3.cxx
@@ -1735,6 +1735,13 @@ void ScDBFunc::SetDataPilotDetails( BOOL bShow, const String* pNewDimensionName
 
 void ScDBFunc::ShowDataPilotSourceData( ScDPObject& rDPObj, const Sequence<sheet::DataPilotFieldFilter>& rFilters )
 {
+    ScDocument* pDoc = GetViewData()->GetDocument();
+    if (pDoc->GetDocumentShell()->IsReadOnly())
+    {
+        ErrorMessage(STR_READONLYERR);
+        return;
+    }
+
     Reference<sheet::XDimensionsSupplier> xDimSupplier = rDPObj.GetSource();
     Reference<container::XNameAccess> xDims = xDimSupplier->getDimensions();
     Reference<sheet::XDrillDownDataSupplier> xDDSupplier(xDimSupplier, UNO_QUERY);
@@ -1749,7 +1756,6 @@ void ScDBFunc::ShowDataPilotSourceData( ScDPObject& rDPObj, const Sequence<sheet
 
     sal_Int32 nColSize = aTabData[0].getLength();
 
-    ScDocument* pDoc = GetViewData()->GetDocument();
     SCTAB nNewTab = GetViewData()->GetTabNo();
 
     auto_ptr<ScDocument> pInsDoc(new ScDocument(SCDOCMODE_CLIP));
