diff --git svtools/inc/PasswordHelper.hxx svtools/inc/PasswordHelper.hxx
index 8407114..d0788e3 100644
--- svtools/inc/PasswordHelper.hxx
+++ svtools/inc/PasswordHelper.hxx
@@ -39,11 +39,12 @@ class String;
 
 class SvPasswordHelper
 {
-	static void		GetHashPassword(com::sun::star::uno::Sequence <sal_Int8>& rPassHash, const sal_Char* pPass, sal_uInt32 nLen);
     static void     GetHashPasswordLittleEndian(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
     static void     GetHashPasswordBigEndian(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
 
 public:
+    SVL_DLLPUBLIC static void     GetHashPassword(com::sun::star::uno::Sequence <sal_Int8>& rPassHash, const sal_Char* pPass, sal_uInt32 nLen);
+
     SVL_DLLPUBLIC static void     GetHashPassword(com::sun::star::uno::Sequence<sal_Int8>& rPassHash, const String& sPass);
     /**
     Use this method to compare a given string with another given Hash value.
