diff --git sc/source/ui/docshell/externalrefmgr.cxx sc/source/ui/docshell/externalrefmgr.cxx
index ae5015d..d3f770b 100644
--- sc/source/ui/docshell/externalrefmgr.cxx
+++ sc/source/ui/docshell/externalrefmgr.cxx
@@ -161,7 +161,8 @@ ScExternalRefCache::TokenRef ScExternalRefCache::Table::getCell(SCCOL nCol, SCRO
     if (itrTable == maRows.end())
     {
         // this table doesn't have the specified row.
-        return TokenRef();
+        TokenRef p(new ScEmptyCellToken(false, false));
+        return p;
     }
 
     const RowDataType& rRowData = itrTable->second;
@@ -169,7 +170,8 @@ ScExternalRefCache::TokenRef ScExternalRefCache::Table::getCell(SCCOL nCol, SCRO
     if (itrRow == rRowData.end())
     {
         // this row doesn't have the specified column.
-        return TokenRef();
+        TokenRef p(new ScEmptyCellToken(false, false));
+        return p;
     }
 
     const Cell& rCell = itrRow->second;
