diff --git sc/source/ui/view/cellsh2.cxx sc/source/ui/view/cellsh2.cxx
index 365a2c7..832de0c 100644
--- sc/source/ui/view/cellsh2.cxx
+++ sc/source/ui/view/cellsh2.cxx
@@ -1239,7 +1239,11 @@ void __EXPORT ScCellShell::GetDBState( SfxItemSet& rSet )
 						{
 							rSet.DisableItem( nWhich );
 						}
-						else
+						else if (pDoc->GetDPAtBlock(aDummy))
+                        {    
+                            rSet.DisableItem( nWhich );
+                        }
+                        else
 							rSet.Put( SfxBoolItem( nWhich, bAutoFilter ) );
 					}
 					else
