diff --git sc/source/ui/docshell/docsh4.cxx sc/source/ui/docshell/docsh4.cxx
index 783c7f9..a9a5820 100644
--- sc/source/ui/docshell/docsh4.cxx
+++ sc/source/ui/docshell/docsh4.cxx
@@ -2473,10 +2473,14 @@ long __EXPORT ScDocShell::DdeSetData( const String& rItem,
 				pData->GetSymbol( aPos );			// continue with the name's contents
 		}
 	}
-	ScRange aRange;
-    ScAddress::Convention eConv = aDocument.GetAddressConvention();
-	BOOL bValid = ( ( aRange.Parse( aPos, &aDocument, eConv ) & SCA_VALID ) ||
-					( aRange.aStart.Parse( aPos, &aDocument, eConv ) & SCA_VALID ) );
+
+    // Address in DDE function must be always parsed as CONV_OOO so that it 
+    // would always work regardless of current address convension.  We do this
+    // because the address item in a DDE entry is *not* normalized when saved
+    // into ODF.
+    ScRange aRange;
+    bool bValid = ( (aRange.Parse(aPos, &aDocument, ScAddress::CONV_OOO ) & SCA_VALID) ||
+                    (aRange.aStart.Parse(aPos, &aDocument, ScAddress::CONV_OOO) & SCA_VALID) );
 
 	ScServerObject* pObj = NULL;			// NULL = error
 	if ( bValid )
