diff --git sc/source/core/data/dpobject.cxx sc/source/core/data/dpobject.cxx
index cd4dcd1..eddb200 100644
--- sc/source/core/data/dpobject.cxx
+++ sc/source/core/data/dpobject.cxx
@@ -448,7 +448,7 @@ void ScDPObject::CreateObjects()
 {
     // if groups are involved, create a new source with the ScDPGroupTableData
     if ( bSettingsChanged && pSaveData && pSaveData->GetExistingDimensionData() )
-        xSource = NULL;
+        InvalidateSource();
 
 	if (!xSource.is())
 	{
@@ -501,6 +501,7 @@ void ScDPObject::InvalidateSource()
 {
 	xSource = NULL;
 	bInfoValid = FALSE;
+    mpTableData.reset();
 }
 
 ScRange ScDPObject::GetNewOutputRange( BOOL& rOverflow )
