diff --git svtools/inc/tabbar.hxx svtools/inc/tabbar.hxx
index ee25537..fed6e1d 100644
--- svtools/inc/tabbar.hxx
+++ svtools/inc/tabbar.hxx
@@ -430,6 +430,11 @@ public:
                                 USHORT nPos = TAB_APPEND );
     void            RemovePage( USHORT nPageId );
     void            MovePage( USHORT nPageId, USHORT nNewPos );
+
+    Color           GetTabBgColor( USHORT nPageId ) const;
+    void            SetTabBgColor( USHORT nPageId, const Color& aTabBgColor, const BOOL& bIsDefaultTabBgColor );
+    BOOL            IsDefaultTabBgColor( USHORT nPageId );
+
     void            Clear();
 
     void            EnablePage( USHORT nPageId, BOOL bEnable = TRUE );
diff --git svtools/source/control/tabbar.cxx svtools/source/control/tabbar.cxx
index 81795e7..9a03230 100644
--- svtools/source/control/tabbar.cxx
+++ svtools/source/control/tabbar.cxx
@@ -75,6 +75,9 @@ struct ImplTabBarItem
     BOOL            mbShort;
     BOOL            mbSelect;
     BOOL            mbEnable;
+    BOOL            mbIsDefaultTabBgColor;
+    Color           maTabBgColor;
+    Color           maTabTextColor;
 
                     ImplTabBarItem( USHORT nItemId, const XubString& rText,
                                     TabBarPageBits nPageBits ) :
@@ -87,6 +90,9 @@ struct ImplTabBarItem
                         mbShort  = FALSE;
                         mbSelect = FALSE;
                         mbEnable = TRUE;
+                        mbIsDefaultTabBgColor = TRUE;
+                        maTabBgColor = Color( COL_AUTO );
+                        maTabTextColor = Color( COL_AUTO );
                     }
 };
 
@@ -1059,7 +1065,8 @@ void TabBar::Paint( const Rectangle& )
     // Font selektieren
     Font aFont = GetFont();
     Font aLightFont = aFont;
-    aLightFont.SetWeight( WEIGHT_LIGHT );
+    //aLightFont.SetWeight( WEIGHT_LIGHT );
+	aLightFont.SetWeight( WEIGHT_NORMAL );
 
     // #i36013# exclude push buttons from painting area
     Rectangle aClipRect( Point( mnOffX, 0 ), Point( mnLastOffX, GetOutputHeightPixel() - 1 ) );
@@ -1133,15 +1140,22 @@ void TabBar::Paint( const Rectangle& )
                     SetFont( aLightFont );
 
                 // Je nach Status die richtige FillInBrush setzen
+                // Set the correct FillInBrush depending upon status
                 if ( pItem->mbSelect || (pItem->mnId == mnCurPageId) )
                 {
+                    // Currently selected Tab
                     SetFillColor( aSelectColor );
                     SetTextColor( aSelectTextColor );
                 }
                 else
                 {
-                    SetFillColor( aFaceColor );
-                    SetTextColor( aFaceTextColor );
+                    if ( !pItem->mbIsDefaultTabBgColor ) {
+                        SetFillColor( pItem->maTabBgColor );
+                        SetTextColor( pItem->maTabTextColor );
+                    } else {
+                        SetFillColor( aFaceColor );
+                        SetTextColor( aFaceTextColor );
+                    }
                 }
 
                 // Muss Font Kursiv geschaltet werden
@@ -1183,11 +1197,26 @@ void TabBar::Paint( const Rectangle& )
                 aPos1.X()++;
                 aPos2.X()--;
                 aPos3.X()--;
-                SetLineColor( rStyleSettings.GetLightColor() );
+
+                // If this is the current tab, draw the left inner shadow the default color, 
+                // otherwise make it the same as the custom background color
+                if ( pItem->mbSelect || (pItem->mnId == mnCurPageId) ) {
+                    SetLineColor( rStyleSettings.GetLightColor() );
+                } else {
+                    if ( !pItem->mbIsDefaultTabBgColor )
+                    {
+                        SetLineColor( pItem->maTabBgColor );
+                    } else {
+                        SetLineColor( rStyleSettings.GetLightColor() );
+                    }
+                }
+                // Draw the left side of the tab
                 DrawLine( aPos0, aPos1 );
 
                 if ( !pItem->mbSelect && (pItem->mnId != mnCurPageId) )
                 {
+                    // Draw the top inner shadow
+                    // ToDo: Change from this static color to tab custom bg color
                     DrawLine( Point( aPos0.X(), aPos0.Y()+1 ),
                                 Point( aPos3.X(), aPos3.Y()+1 ) );
                 }
@@ -1197,8 +1226,20 @@ void TabBar::Paint( const Rectangle& )
                 aPos1.X()--;
                 aPos1.Y()--;
                 aPos2.Y()--;
+                if ( !pItem->mbIsDefaultTabBgColor && ( pItem->mbSelect || (pItem->mnId == mnCurPageId) ) )
+                {
+                    SetLineColor( pItem->maTabBgColor );
+                }
                 DrawLine( aPos1, aPos2 );
 
+                // draw a small 2px sliver of the original background color at the bottom of the selected tab
+                if ( !pItem->mbIsDefaultTabBgColor )
+                {
+                    if ( pItem->mbSelect || (pItem->mnId == mnCurPageId) ) {
+                        SetLineColor( pItem->maTabBgColor );
+                    }
+                }
+
                 // Da etwas uebermalt werden konnte, muessen wir die Polygon-
                 // umrandung nocheinmal ausgeben
                 SetLineColor( rStyleSettings.GetDarkShadowColor() );
@@ -1549,6 +1590,7 @@ void TabBar::InsertPage( USHORT nPageId, const XubString& rText,
 
     // PageItem anlegen und in die Item-Liste eintragen
     ImplTabBarItem* pItem = new ImplTabBarItem( nPageId, rText, nBits );
+    
     mpItemList->Insert( pItem, nPos );
     mbSizeFormat = TRUE;
 
@@ -1565,6 +1607,46 @@ void TabBar::InsertPage( USHORT nPageId, const XubString& rText,
 
 // -----------------------------------------------------------------------
 
+Color TabBar::GetTabBgColor( USHORT nPageId ) const
+{
+    USHORT nPos = GetPagePos( nPageId );
+
+    if ( nPos != TAB_PAGE_NOTFOUND )
+        return mpItemList->GetObject( nPos )->maTabBgColor;
+    else
+        return Color( COL_AUTO ); //TODO Find a better way to report back error FIXME
+}
+
+void TabBar::SetTabBgColor( USHORT nPageId, const Color& aTabBgColor, 
+                                            const BOOL& bIsDefaultTabBgColor )
+{
+    USHORT nPos = GetPagePos( nPageId );
+    ImplTabBarItem* pItem;
+    if ( nPos != TAB_PAGE_NOTFOUND )
+    {
+        pItem = mpItemList->GetObject( nPos );
+        // TODO: Need to take the color specification out of this code!
+        pItem->mbIsDefaultTabBgColor = bIsDefaultTabBgColor;
+        if ( aTabBgColor == Color(COL_AUTO) )
+            pItem->mbIsDefaultTabBgColor = TRUE;
+        if ( !pItem->mbIsDefaultTabBgColor  )
+        {
+            pItem->maTabBgColor = aTabBgColor;
+            if ( aTabBgColor.GetLuminance() <= 128 ) //Do not use aTabBgColor.IsDark(), because that threshold is way too low...
+                pItem->maTabTextColor = Color( COL_WHITE );
+            else
+                pItem->maTabTextColor = Color( COL_BLACK );
+        }
+        else
+        {
+            pItem->maTabBgColor = Color( COL_AUTO );
+            pItem->maTabTextColor = Color( COL_AUTO );
+        }
+    }
+}
+
+// -----------------------------------------------------------------------
+
 void TabBar::RemovePage( USHORT nPageId )
 {
     USHORT nPos = GetPagePos( nPageId );
