diff --git sc/source/ui/src/localize.sdf sc/source/ui/src/localize.sdf
index f321be9..aa5977b 100644
--- sc/source/ui/src/localize.sdf
+++ sc/source/ui/src/localize.sdf
@@ -23087,7 +23087,7 @@ sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	es	Cant
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	et	Arv				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	fa	تعداد				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	fi	LASKE				2002-02-02 02:02:02
-sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	fr	NB				2002-02-02 02:02:02
+sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	fr	Compteur				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			13691	ga	Líon				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	gl	Contaxe				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT			0	gu	ગણતરી				2002-02-02 02:02:02
@@ -23167,7 +23167,7 @@ sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	eo	Nom
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	es	Cantidad2				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	et	CountA				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	fi	LaskeA				2002-02-02 02:02:02
-sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	fr	NBVAL				2002-02-02 02:02:02
+sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	fr	Quantité				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			13691	ga	LíonA				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	gl	ContarA				2002-02-02 02:02:02
 sc	source\ui\src\globstr.src	0	string	RID_GLOBSTR	STR_FUN_TEXT_COUNT2			0	gu	CountA				2002-02-02 02:02:02
