diff --git xmloff/inc/xmloff/xmltoken.hxx xmloff/inc/xmloff/xmltoken.hxx
index 694138d..e2efd25 100644
--- xmloff/inc/xmloff/xmltoken.hxx
+++ xmloff/inc/xmloff/xmltoken.hxx
@@ -1410,6 +1410,8 @@ namespace xmloff { namespace token {
         XML_PROTECT_SIZE,
         XML_PROTECTED,
         XML_PROTECTION_KEY,
+        XML_PROTECTION_KEY_DIGEST_ALGORITHM,
+        XML_PROTECTION_KEY_DIGEST_ALGORITHM_2,
         XML_PRSUBSET,
         XML_PUBLISHER,
         XML_PUNCTUATION_WRAP,
@@ -1519,6 +1521,8 @@ namespace xmloff { namespace token {
         XML_SECTION_NAME,
         XML_SECTION_SOURCE,
         XML_SELECT_PAGE,
+        XML_SELECT_PROTECTED_CELLS,
+        XML_SELECT_UNPROTECTED_CELLS,
         XML_SELECTOR,
         XML_SEMANTICS,
         XML_SEMI_AUTOMATIC,
@@ -1731,6 +1735,7 @@ namespace xmloff { namespace token {
         XML_TABLE_OF_CONTENT_ENTRY_TEMPLATE,
         XML_TABLE_OF_CONTENT_SOURCE,
         XML_TABLE_PAGE,
+        XML_TABLE_PROTECTION,
         XML_TABLE_ROW,
         XML_TABLE_ROW_GROUP,
         XML_TABLE_ROWS,
diff --git xmloff/source/core/xmltoken.cxx xmloff/source/core/xmltoken.cxx
index e90d4a6..4c16ec6 100644
--- xmloff/source/core/xmltoken.cxx
+++ xmloff/source/core/xmltoken.cxx
@@ -1418,6 +1418,8 @@ namespace xmloff { namespace token {
         TOKEN( "size",                            XML_PROTECT_SIZE ),
         TOKEN( "protected",                       XML_PROTECTED ),
         TOKEN( "protection-key",                  XML_PROTECTION_KEY ),
+        TOKEN( "protection-key-digest-algorithm",   XML_PROTECTION_KEY_DIGEST_ALGORITHM ),
+        TOKEN( "protection-key-digest-algorithm-2", XML_PROTECTION_KEY_DIGEST_ALGORITHM_2 ),
         TOKEN( "prsubset",                        XML_PRSUBSET ),
         TOKEN( "publisher",                       XML_PUBLISHER ),
         TOKEN( "punctuation-wrap",                XML_PUNCTUATION_WRAP ),
@@ -1527,6 +1529,8 @@ namespace xmloff { namespace token {
         TOKEN( "section-name",                    XML_SECTION_NAME ),
         TOKEN( "section-source",                  XML_SECTION_SOURCE ),
         TOKEN( "select-page",                     XML_SELECT_PAGE ),
+        TOKEN( "select-protected-cells",          XML_SELECT_PROTECTED_CELLS ),
+        TOKEN( "select-unprotected-cells",        XML_SELECT_UNPROTECTED_CELLS ),
         TOKEN( "selector",                        XML_SELECTOR ),
         TOKEN( "semantics",                       XML_SEMANTICS ),
         TOKEN( "semi-automatic",                  XML_SEMI_AUTOMATIC ),
@@ -1739,6 +1743,7 @@ namespace xmloff { namespace token {
         TOKEN( "table-of-content-entry-template",     XML_TABLE_OF_CONTENT_ENTRY_TEMPLATE ),
         TOKEN( "table-of-content-source",         XML_TABLE_OF_CONTENT_SOURCE ),
         TOKEN( "table-page",                      XML_TABLE_PAGE ),
+        TOKEN( "table-protection",                XML_TABLE_PROTECTION ),
         TOKEN( "table-row",                       XML_TABLE_ROW ),
         TOKEN( "table-row-group",                 XML_TABLE_ROW_GROUP ),
         TOKEN( "table-rows",                      XML_TABLE_ROWS ),
