diff --git sc/source/filter/xml/xmlexprt.cxx sc/source/filter/xml/xmlexprt.cxx
index c086a45..3658d78 100644
--- sc/source/filter/xml/xmlexprt.cxx
+++ sc/source/filter/xml/xmlexprt.cxx
@@ -1959,7 +1959,7 @@ void ScXMLExport::_ExportAutoStyles()
 									    else
 										    pRowStyles->AddNewTable(nTable, nRows);
 									    sal_Int32 nRow = 0;
-									    while ( /*nRow <= nRows && */nRow <= MAXROW)
+									    while (nRow <= nRows && nRow <= MAXROW)
 									    {
                                             sal_Int32 nIndex = 0;
 										    uno::Reference <beans::XPropertySet> xRowProperties(xTableRows->getByIndex(nRow), uno::UNO_QUERY);
