diff --git sc/source/ui/view/cellsh2.cxx sc/source/ui/view/cellsh2.cxx
index d2e5e61..554cdf9 100644
--- sc/source/ui/view/cellsh2.cxx
+++ sc/source/ui/view/cellsh2.cxx
@@ -737,7 +737,8 @@ void ScCellShell::ExecuteDB( SfxRequest& rReq )
 						{
 							//!	use database ranges (select before type dialog?)
 							ScRange aRange;
-							if ( GetViewData()->GetSimpleArea( aRange ) == SC_MARK_SIMPLE )
+                            ScMarkType eType = GetViewData()->GetSimpleArea(aRange);
+                            if ( (eType & SC_MARK_SIMPLE) == SC_MARK_SIMPLE )
 							{
 								BOOL bOK = TRUE;
 								if ( pDoc->HasSubTotalCells( aRange ) )
diff --git sc/source/ui/view/viewdata.cxx sc/source/ui/view/viewdata.cxx
index a16efb5..0dd53be 100644
--- sc/source/ui/view/viewdata.cxx
+++ sc/source/ui/view/viewdata.cxx
@@ -774,7 +774,8 @@ BOOL ScViewData::IsMultiMarked()
     // and taking filtered in simple area marks into account.
 
     ScRange aDummy;
-    return GetSimpleArea( aDummy) != SC_MARK_SIMPLE;
+    ScMarkType eType = GetSimpleArea(aDummy);
+    return (eType & SC_MARK_SIMPLE) != SC_MARK_SIMPLE;
 }
 
 void ScViewData::SetFillMode( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow )
