diff --git offapi/com/sun/star/chart2/data/DataProvider.idl offapi/com/sun/star/chart2/data/DataProvider.idl
index c83199e..d07cf8d 100644
--- offapi/com/sun/star/chart2/data/DataProvider.idl
+++ offapi/com/sun/star/chart2/data/DataProvider.idl
@@ -57,6 +57,10 @@ service DataProvider
         internally into valid XML.
      */
     [optional] interface ::com::sun::star::chart2::data::XRangeXMLConversion;
+
+    /** If set to false <FALSE/>, values from hidden cells are not returned.
+     */
+    [optional, property] boolean                  IncludeHiddenCells;
 };
 
 } ; // data
diff --git offapi/com/sun/star/chart2/data/DataSequence.idl offapi/com/sun/star/chart2/data/DataSequence.idl
index 9702545..c97f179 100644
--- offapi/com/sun/star/chart2/data/DataSequence.idl
+++ offapi/com/sun/star/chart2/data/DataSequence.idl
@@ -140,14 +140,11 @@ service DataSequence
      */
     [property] DataSequenceRole                   Role;
 
-    /** If <TRUE/>, the values are hidden, and should therefore not be
-        used.  This may result in omitting a complete data series
-        object on rendering.
+    /** If set to false <FALSE/>, values from hidden cells are not returned.
      */
-    [optional, property] boolean                  IsHidden;
+    [optional, property] boolean                  IncludeHiddenCells;
 
-    /** a sequence of indexes that identify data points that should
-        not be rendered, because they are hidden in the underlying
+    /** a sequence of indexes that identify values that are hidden in the underlying
         data provider.
      */
     [optional, property] sequence< long >         HiddenValues;
