--- config_office/configure.in.bak	2008-07-17 07:57:50.734375000 -0700
+++ config_office/configure.in	2008-07-17 08:00:13.781250000 -0700
@@ -152,6 +152,10 @@
 [  --disable-activex       Disable the use of ActiveX for windows build.
                           This switch is mandatory when using VC++ 2005 Express.
 ],,)
+AC_ARG_ENABLE(atl,
+[  --disable-atl          Disable the use of atl.h windows build.
+                          This switch is mandatory when using VC++ 2005 Express.
+],,)
 
 AC_ARG_ENABLE(symbols,
 [  --enable-symbols        Include debugging symbols in output.
@@ -1173,6 +1177,16 @@
       AC_MSG_RESULT([no])
    fi
    AC_SUBST(DISABLE_ACTIVEX)
+
+   AC_MSG_CHECKING([whether to use atl.h])
+   if test "$enable_atl" = "yes" -o "$enable_atl" = "TRUE" -o "$enable_atl" = ""; then
+      DISABLE_ATL=""
+      AC_MSG_RESULT([yes])
+   else
+      DISABLE_ATL="TRUE"
+      AC_MSG_RESULT([no])
+   fi
+   AC_SUBST(DISABLE_ATL)
 fi
 
 dnl ===================================================================
--- config_office/set_soenv.in.bak	2008-07-17 08:01:33.000000000 -0700
+++ config_office/set_soenv.in	2008-07-17 08:02:13.890625000 -0700
@@ -1859,6 +1859,7 @@
    ToFile( "MFC_INCLUDE",       $MFC_INCLUDE,       "e" );
    ToFile( "NSIS_PATH",         $NSIS_PATH,         "e" );
    ToFile( "DISABLE_ACTIVEX",   "@DISABLE_ACTIVEX@","e" );
+   ToFile( "DISABLE_ATL",       "@DISABLE_ATL@","e" );
    ToFile( "USE_NEW_SDK",       $USE_NEW_SDK,       "e" );
 
    if ( "@NOWRAPCMD@" eq "yes" ) {
