--- configure.in.orig	2007-07-24 20:03:51.000000000 +0200
+++ configure.in	2007-07-24 21:18:11.000000000 +0200
@@ -4143,17 +4143,9 @@
 if test -z "$x_libraries"; then
    x_libraries="no_x_libraries"
 fi
-if test "$x_includes" = "default_x_includes"; then
-   XINC="/usr/include"
-else
-   XINC="$x_includes"
-fi
+XINC="$x_includes"
+XLIB="$x_libraries"
 AC_SUBST(XINC)
-if test "$x_libraries" = "default_x_libraries"; then
-   XLIB="/usr/lib"
-else
-   XLIB="$x_libraries"
-fi
 AC_SUBST(XLIB)
 AC_SUBST(DISABLE_XAW)
 AC_SUBST(XAU_LIBS)
@@ -5477,6 +5469,9 @@
    XINERAMA_LINK=dynamic
    AC_MSG_RESULT([yes])
 elif test "$_os" = "Linux" -o "$_os" = "FreeBSD"; then
+   if test "$x_libraries" = "default_x_libraries"; then
+      XLIB="/usr/lib"
+   fi
    if test -e "$XLIB/libXinerama.so" -a -e "$XLIB/libXinerama.a"; then
       # we have both versions, let the user decide but use the static one
       # per default
