--- set_soenv.in	(Revision 268589)
+++ set_soenv.in	(Arbeitskopie)
@@ -564,12 +564,7 @@
       else {
         if ('@WORDS_BIGENDIAN@' ne 'yes') {
           $EPM_FLAGS = "-a mipsel";
-          if ( $JDK eq "gcj" ) {
-	    $ARCH = "mipsel";
-          }
-          else {
-	    $ARCH = "mips32";
-          }
+	  $ARCH = "mipsel";
         }
         else {
           $ARCH = "mips";
@@ -577,7 +577,7 @@
       }
 
       $JRELIBDIR      = '$JAVA_HOME'.$ds."lib".$ds.$ARCH;
-      $JRETOOLKITDIR  = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."client";
+      $JRETOOLKITDIR  = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."server";
       $JRETHREADDIR   = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."native_threads";
       $JREEXTRALIBDIR = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."xawt";
    }
