--- configure.in	2008-06-23 14:46:06.000000000 +0200
+++ configure.in	2008-06-23 14:47:00.000000000 +0200
@@ -3534,7 +3534,7 @@
         test "$with_system_libwpd" != "no"; then
     AC_MSG_RESULT([external])
     SYSTEM_LIBWPD=YES
-    PKG_CHECK_MODULES( LIBWPD, libwpd-0.8 )
+    PKG_CHECK_MODULES( LIBWPD, libwpd-0.9 libwpd-stream-0.9 )
 else
     AC_MSG_RESULT([internal])
     SYSTEM_LIBWPD=NO
@@ -3552,7 +3552,7 @@
         test "$with_system_libwps" != "no"; then
     AC_MSG_RESULT([external])
     SYSTEM_LIBWPS=YES
-    PKG_CHECK_MODULES( LIBWPS, libwps-0.1 )
+    PKG_CHECK_MODULES( LIBWPS, libwps-0.2 )
 else
     AC_MSG_RESULT([internal])
     SYSTEM_LIBWPS=NO
@@ -3570,7 +3570,7 @@
         test "$with_system_libwpg" != "no"; then
     AC_MSG_RESULT([external])
     SYSTEM_LIBWPG=YES
-    PKG_CHECK_MODULES( LIBWPG, libwpg-0.1 )
+    PKG_CHECK_MODULES( LIBWPG, libwpg-0.2 )
 else
     AC_MSG_RESULT([internal])
     SYSTEM_LIBWPG=NO
