--- connectivity/com/sun/star/sdbcx/comp/hsqldb/NativeLibraries.java.old	2008-04-10 09:28:04.000000000 +0200
+++ connectivity/com/sun/star/sdbcx/comp/hsqldb/NativeLibraries.java	2008-10-02 16:43:19.000000000 +0200
@@ -71,8 +71,13 @@ final class NativeLibraries {
                                      invoke(url, null) })).
                             getAbsolutePath());
                     } catch (Throwable t) {
-                        throw new UnsatisfiedLinkError(
-                            e.toString()+ " - " + t.toString());
+			// final fallback for Java 1.4.2
+			try {
+                            System.load(url.getPath());
+			} catch (Throwable t2) {
+                            throw new UnsatisfiedLinkError(
+                            e.toString()+ " - " + t2.toString());
+			}
                     }
                 }
             }
