Index: source/sdstor/stgstrms.cxx
===================================================================
RCS file: /cvs/util/sot/source/sdstor/stgstrms.cxx,v
retrieving revision 1.11
diff -u -p -r1.11 stgstrms.cxx
--- sot/source/sdstor/stgstrms.cxx
+++ sot/source/sdstor/stgstrms.cxx
@@ -823,6 +823,8 @@ void* StgDataStrm::GetPtr( INT32 Pos, BO
 
 INT32 StgDataStrm::Read( void* pBuf, INT32 n )
 {
+    if (n < 0)
+        return 0;
     if( ( nPos + n ) > nSize )
         n = nSize - nPos;
     INT32 nDone = 0;
