Index: desktop/source/app/appinit.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/appinit.cxx,v
retrieving revision 1.29
diff -u -p -r1.29 appinit.cxx
--- desktop/source/app/appinit.cxx	11 Apr 2008 03:35:23 -0000	1.29
+++ desktop/source/app/appinit.cxx	31 Oct 2008 10:17:09 -0000
@@ -133,7 +133,8 @@ static bool configureUcb(bool bServer, r
 
     bool ret = 
         ::ucbhelper::ContentBroker::initialize( xServiceFactory, aArgs ) != false;
-    
+
+#ifdef GNOME_VFS_ENABLED
     // register GnomeUCP if necessary
     ::ucbhelper::ContentBroker* cb = ::ucbhelper::ContentBroker::get();
     if(cb) {
@@ -173,6 +173,7 @@ static bool configureUcb(bool bServer, r
         } catch (RuntimeException e) {
         }
     }    
+#endif // GNOME_VFS_ENABLED
     
     return ret;;
 }
Index: desktop/source/app/makefile.mk
===================================================================
RCS file: /cvs/framework/desktop/source/app/makefile.mk,v
retrieving revision 1.42
diff -u -p -r1.42 makefile.mk
--- desktop/source/app/makefile.mk	25 Jun 2008 11:10:06 -0000	1.42
+++ desktop/source/app/makefile.mk	31 Oct 2008 10:17:09 -0000
@@ -40,6 +40,10 @@ ENABLE_EXCEPTIONS=TRUE
 
 .INCLUDE :  settings.mk
 
+.IF "$(ENABLE_GNOMEVFS)"=="TRUE"
+CFLAGS+=-DGNOME_VFS_ENABLED
+.ENDIF
+
 SHL1TARGET = sofficeapp
 SHL1OBJS = \
     $(SLO)$/app.obj \
