--- desktop/win32/source/setup/setup_w.cxx.~1.5.~	2008-04-11 06:50:30.000000000 +0200
+++ desktop/win32/source/setup/setup_w.cxx	2008-11-05 15:53:43.000000000 +0100
@@ -34,5 +34,9 @@
 #define  UNICODE    1
 #define _UNICODE    1
 
+#ifdef __MINGW32__
+#define _ttempnam(a, b) (TCHAR*)tempnam ((char*)a, (char*)b)
+#endif
+
 #include "setup.cpp"
 
--- desktop/win32/source/setup/setup_a.cxx.~1.5.~	2008-04-11 06:49:31.000000000 +0200
+++ desktop/win32/source/setup/setup_a.cxx	2008-11-05 15:52:36.000000000 +0100
@@ -31,4 +31,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_desktop.hxx"
 
+#ifdef __MINGW32__
+#define _ttempnam tempnam
+#endif
+
 #include "setup.cpp"
--- desktop/win32/source/setup/rcheader.txt.~1.6.~	2008-01-15 14:33:41.000000000 +0100
+++ desktop/win32/source/setup/rcheader.txt	2008-11-05 15:16:00.000000000 +0100
@@ -7,7 +7,7 @@
 #define CB_ADDSTRING	(WM_USER+3)
 #define IDC_STATIC		(-1)
 #endif
-#include "resource.h"
+#include "Resource.h"
 
 /////////////////////////////////////////////////////////////////////////////
 //
--- desktop/win32/source/setup/setup.cpp.~1.15.12.1.~	2008-09-03 15:00:32.000000000 +0200
+++ desktop/win32/source/setup/setup.cpp	2008-11-05 15:52:16.000000000 +0100
@@ -49,7 +49,7 @@
 
 #include "setup.hxx"
 
-#include "resource.h"
+#include "Resource.h"
 
 //--------------------------------------------------------------------------
 
--- desktop/win32/source/setup/makefile.mk.~1.12.~	2008-04-11 06:48:48.000000000 +0200
+++ desktop/win32/source/setup/makefile.mk	2008-11-05 15:19:13.000000000 +0100
@@ -92,4 +92,23 @@ APP1NOSVRES=	$(RES)$/$(TARGET).res
 
 $(RCFILES) : $(ULFDIR)$/setup.ulf makefile.mk rcfooter.txt rcheader.txt rctmpl.txt
     $(WRAPCMD) lngconvex.exe -ulf $(ULFDIR)$/setup.ulf -rc $(RCFILES) -rct rctmpl.txt -rch rcheader.txt -rcf rcfooter.txt
+.IF "$(CROSS_COMPILING)" == "yes"
+	sed -i -e 's@\\x\([0-9a-f][0-9a-f]["\\]\)@\\x00\1@g' $@
+	sed -i -e 's@\\x\([0-9a-f][0-9a-f]["\\]\)@\\x00\1@g' $@
+	sed -i -e 's@\\x\([0-9a-f][0-9a-f][0-9a-f]["\\]\)@\\x0\1@g' $@
+	sed -i -e 's@\\x\([0-9a-f][0-9a-f][0-9a-f]["\\]\)@\\x0\1@g' $@
+	sed -i -e 's@\([ ,]\)"@\1L"@g' $@
+	sed -i -e 's@include L"@include "@' $@
+	pytt '\r\n' '\n' $@
+	pytt '^LANGUAGE 0x4d(.|\n)*' '' $@
+#	sed -i -e 's@^LANGUAGE 0x211@LANGUAGE 0x21@' $@
+#	sed -i -e 's@^LANGUAGE 0x226@LANGUAGE 0x26@' $@
+#	sed -i -e 's@^LANGUAGE 0x22c@LANGUAGE 0x2c@' $@
+#	sed -i -e 's@^LANGUAGE 0x22b@LANGUAGE 0x2b@' $@
+#	sed -i -e 's@^LANGUAGE 0x30@LANGUAGE 0x2d@' $@
+#	sed -i -e 's@^LANGUAGE 0x73@LANGUAGE 0x2a@' $@
+#	sed -i -e 's@^LANGUAGE 0x31@LANGUAGE 0x2e@' $@
+#	sed -i -e 's@^LANGUAGE 0x33@LANGUAGE 0x01@' $@
+#	sed -i -e 's@^LANGUAGE 0x55@LANGUAGE 0x22@' $@
+.ENDIF #"$(CROSS_COMPILING)" == "yes"
 
--- desktop/util/verinfo.rc.~1.19.12.1.~	2008-09-19 09:30:39.000000000 +0200
+++ desktop/util/verinfo.rc	2008-11-05 15:04:37.000000000 +0100
@@ -149,7 +149,11 @@ VS_VERSION_INFO versioninfo
 VS_VERSION_INFO rcdata
 {
 	0xF0, "sw", 0x0F, VER_YEAR, VER_MONTH, VER_DAY,
+#ifndef CROSS_COMPILING
 	VERSION, VERVARIANT, VER_COUNT
+#else
+	VERSION
+#endif
 };
 
 
--- desktop/util/ooverinfo.rc.~1.12.12.1.~	2008-09-19 09:30:16.000000000 +0200
+++ desktop/util/ooverinfo.rc	2008-11-05 15:07:54.000000000 +0100
@@ -144,7 +144,11 @@ VS_VERSION_INFO versioninfo
 VS_VERSION_INFO rcdata
 {
 	0xF0, "sw", 0x0F, VER_YEAR, VER_MONTH, VER_DAY,
+#ifndef CROSS_COMPILING
 	VERSION, SUBVERSION, VERVARIANT, VER_COUNT
+#else
+	VERSION
+#endif
 };
 
 
--- desktop/source/pkgchk/unopkg/makefile.mk.~1.17.~	2008-07-22 19:16:08.000000000 +0200
+++ desktop/source/pkgchk/unopkg/makefile.mk	2008-11-05 14:58:45.000000000 +0100
@@ -55,7 +55,7 @@ APP1STDLIBS = $(SALLIB) $(UNOPKGAPPLIB)
 APP1DEPN = $(SHL1TARGETN)
 APP1NOSAL = TRUE
 APP1RPATH = BRAND
-.IF "$(OS)" == "WNT"
+.IF "$(TARGET_OS)" == "WNT"
 APP1ICON = $(SOLARRESDIR)$/icons/so9_main_app.ico
 APP1LINKRES = $(MISC)$/$(TARGET)1.res
 .ENDIF
@@ -66,7 +66,7 @@ APP2STDLIBS = $(SALLIB) $(UNOPKGAPPLIB)
 APP2DEPN = $(SHL1TARGETN)
 APP2NOSAL = TRUE
 APP2RPATH = BRAND
-.IF "$(OS)" == "WNT"
+.IF "$(TARGET_OS)" == "WNT"
 APP2ICON = $(SOLARRESDIR)$/icons/ooo3_main_app.ico
 APP2LINKRES = $(MISC)$/$(TARGET)2.res
 .ENDIF
--- desktop/source/deployment/misc/dp_misc.cxx.pristine	2009-04-06 12:16:26.807994125 +0200
+++ desktop/source/deployment/misc/dp_misc.cxx	2009-04-06 12:16:33.664089092 +0200
@@ -55,7 +55,7 @@
 #define UNICODE
 #define _UNICODE
 #define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
+#include <windows.h>
 //#include "tools/postwin.h"
 #endif
 
