--- embedserv/source/inprocserv/makefile.mk.orig	2009-06-30 17:42:56.000000000 +0200
+++ embedserv/source/inprocserv/makefile.mk	2009-06-30 17:46:33.000000000 +0200
@@ -57,12 +57,19 @@
 	$(SLO)$/inprocembobj.obj
 
 SHL1TARGET=$(TARGET)
+.IF "$(COM)"=="GCC"
+SHL1STDLIBS=\
+    -luuid\
+    -lole32\
+    -lgdi32\
+    -ladvapi32
+.ELSE
 SHL1STDLIBS=\
     uuid.lib\
     ole32.lib\
     gdi32.lib\
     advapi32.lib
-
+.ENDIF
 SHL1OBJS=$(SLOFILES)
 
 SHL1DEF=$(MISC)$/$(TARGET).def
