--- solenv/bin/build.pl.orig	2008-07-22 18:46:59.000000000 +0200
+++ solenv/bin/build.pl	2008-10-21 11:18:41.000000000 +0200
@@ -419,7 +419,7 @@ sub BuildAll {
         while ($Prj = PickPrjToBuild(\%global_deps_hash)) {
             if (!defined $dead_parents{$Prj}) {              
                 if (scalar keys %broken_build) {
-                    print $echo . "Skipping project $Prj because of error(s)\n";
+                    print $echo . "Skipping project $Prj: error:" . $broken_build{$Prj} . "\n";
                     RemoveFromDependencies($Prj, \%global_deps_hash);
                     $build_is_finished{$Prj}++;
                     next;
