--- vcl/win/source/app/saltimer.cxx.pristine	2008-10-30 13:15:44.000000000 +0100
+++ vcl/win/source/app/saltimer.cxx	2008-10-30 13:13:16.000000000 +0100
@@ -32,7 +32,7 @@
 #include "precompiled_vcl.hxx"
 #include <tools/svwin.h>
 #ifdef __MINGW32__
-#include <excpt.h>
+#include <sehandler.h>
 #endif
 #include <saldata.hxx>
 #include <saltimer.h>
--- vcl/win/source/app/salinst.cxx.pristine	2009-02-18 16:18:51.000000000 +0100
+++ vcl/win/source/app/salinst.cxx	2009-04-04 16:28:58.940770892 +0200
@@ -37,7 +37,7 @@
 #include <process.h>
 #endif
 #ifdef __MINGW32__
-#include <excpt.h>
+#include <sehandler.h>
 #endif
 
 #include <vos/mutex.hxx>
@@ -67,9 +67,11 @@
 #pragma warning(push, 1)
 #endif
 
+#ifndef __MINGW32__
 #include <GdiPlus.h>
 #include <GdiPlusEnums.h>
 #include <GdiPlusColor.h>
+#endif /* __MINGW32__ */
 
 #if defined _MSC_VER
 #pragma warning(pop)
@@ -456,10 +458,12 @@ void InitSalData()
 {
 	SalData* pSalData = new SalData;
 	CoInitialize(0);
 
+#ifndef __MINGW32__
     // init GDIPlus
     static Gdiplus::GdiplusStartupInput gdiplusStartupInput;
     Gdiplus::GdiplusStartup(&pSalData->gdiplusToken, &gdiplusStartupInput, NULL);
+#endif /* __MINGW32__ */
 }
 
 
@@ -468,11 +471,13 @@ void DeInitSalData()
 	CoUninitialize();
 	SalData* pSalData = GetSalData();
 
+#ifndef __MINGW32__
     // deinit GDIPlus
     if(pSalData)
     {
         Gdiplus::GdiplusShutdown(pSalData->gdiplusToken);
     }
+#endif /* __MINGW32__ */
 
     delete pSalData;
 }
--- vcl/win/source/gdi/salnativewidgets-luna.cxx.pristine	2008-10-30 13:50:29.000000000 +0100
+++ vcl/win/source/gdi/salnativewidgets-luna.cxx	2008-10-30 13:49:24.000000000 +0100
@@ -31,6 +31,11 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_vcl.hxx"
 
+#ifdef __MINGW32__
+#undef WINVER
+#define WINVER 0x0501
+#endif /* __MINGW32__ */
+
 #define _SV_SALNATIVEWIDGETS_CXX
 
 #include "svsys.h"
--- vcl/win/source/gdi/winlayout.cxx.pristine	2008-10-30 13:43:54.000000000 +0100
+++ vcl/win/source/gdi/winlayout.cxx	2008-10-30 13:43:46.000000000 +0100
@@ -55,8 +55,8 @@
 #include <vcl/svapp.hxx>
 
 #ifdef USE_UNISCRIBE
-#include <Usp10.h>
-#include <ShLwApi.h>
+#include <usp10.h>
+#include <shlwapi.h>
 #include <winver.h>
 #endif // USE_UNISCRIBE
 
--- vcl/win/source/gdi/salprn.cxx.pristine	2008-10-30 13:15:44.000000000 +0100
+++ vcl/win/source/gdi/salprn.cxx	2008-10-30 13:42:47.000000000 +0100
@@ -35,8 +35,11 @@
 #include <tools/svwin.h>
 
 #ifdef __MINGW32__
-#include <excpt.h>
-#endif
+#undef WINVER
+#define WINVER 0x500
+#include <sehandler.h>
+#define DC_COLLATE 23
+#endif /* __MINGW32__ */
 
 #ifndef _OSL_MODULE_H
 #include <osl/module.h>
--- vcl/win/source/gdi/salgdi2.cxx.pristine	2008-10-30 13:44:20.000000000 +0100
+++ vcl/win/source/gdi/salgdi2.cxx	2008-10-30 13:36:43.000000000 +0100
@@ -31,8 +31,14 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_vcl.hxx"
 
+#ifdef __MINGW32__
+#undef WINVER
+#define WINVER 0x410
+#endif /* __MINGW32__ */
+
 #include <string.h>
 #include <stdlib.h>
+
 #include <tools/svwin.h>
 #include <tools/debug.hxx>
 #include <wincomp.hxx>
@@ -44,6 +50,7 @@
 #include <salgdi.h>
 #include <salframe.h>
 
+
 bool WinSalGraphics::supportsOperation( OutDevSupportType eType ) const
 {
     bool bRet = false;
--- vcl/win/source/gdi/salgdi3.cxx.pristine	2008-10-30 13:44:13.000000000 +0100
+++ vcl/win/source/gdi/salgdi3.cxx	2008-10-30 13:37:53.000000000 +0100
@@ -31,6 +31,11 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_vcl.hxx"
 
+#ifdef __MINGW32__
+#undef WINVER
+#define WINVER 0x500
+#endif /* __MINGW32__ */
+
 #include <string.h>
 #include <malloc.h>
 #include <tools/svwin.h>
--- vcl/win/source/window/salframe.cxx.pristine	2009-04-03 21:43:22.740500000 +0200
+++ vcl/win/source/window/salframe.cxx	2009-04-04 19:52:21.487998727 +0200
@@ -52,7 +52,12 @@
 #include <tools/svwin.h>
 #endif
 #ifdef __MINGW32__
-#include <excpt.h>
+#include <sehandler.h>
+typedef struct tagRECONVERTSTRING FAR *LPRECONVERTSTRING;
+#define SCS_CAP_SETRECONVERTSTRING 0x00000004
+#define SCS_QUERYRECONVERTSTRING 0x00020000
+#define SCS_SETRECONVERTSTRING 0x00010000
+#define IMR_CONFIRMRECONVERTSTRING 0x0005
 #endif
 #include <rtl/string.h>
 #include <rtl/ustring.h>
@@ -93,8 +98,10 @@
 #ifndef _VCL_IMPDEL_HXX
 #include <impdel.hxx>
 #endif
+#ifndef __MINGW32__
 #define COMPILE_MULTIMON_STUBS
 #include <multimon.h>
+#endif /* __MINGW32__ */
 #include <vector>
 #ifdef __MINGW32__
 #include <algorithm>
@@ -884,6 +891,7 @@ static void ImplSalCalcFullScreenSize( c
     else
         nCaptionY = 0;
 
+#ifndef __MINGW32__
 	try
 	{
         Reference< XMultiServiceFactory > xFactory( ::comphelper::getProcessServiceFactory(), UNO_QUERY_THROW );
@@ -911,7 +919,7 @@ static void ImplSalCalcFullScreenSize( c
 	catch( Exception& )
 	{
 	}
-
+#endif /* __MINGW32__ */
 	if( !nScreenDX || !nScreenDY )
 	{
 	    nScreenDX   = GetSystemMetrics( SM_CXSCREEN );
@@ -5640,6 +5648,7 @@ static BOOL ImplHandleIMEEndComposition(
 static boolean ImplHandleAppCommand( HWND hWnd, LPARAM lParam )
 {
 	sal_Int16 nCommand = 0;
+#ifndef __MINGW32__
 	switch( GET_APPCOMMAND_LPARAM(lParam) )
 	{
 	case APPCOMMAND_MEDIA_CHANNEL_DOWN:			nCommand = MEDIA_COMMAND_CHANNEL_DOWN; break;
@@ -5663,7 +5672,7 @@ static boolean ImplHandleAppCommand( HWN
 	default:
 		return false;
 	}
-
+#endif /* __MINGW32__ */
     WinSalFrame* pFrame = GetWindowPtr( hWnd );
 	Window *pWindow = pFrame ? pFrame->GetWindow() : NULL;

--- vcl/win/source/src/salsrc.rc.orig	2009-07-01 12:26:36.000000000 +0200
+++ vcl/win/source/src/salsrc.rc	2009-07-01 13:25:14.000000000 +0200
@@ -119,6 +119,6 @@
 SAL_RESID_POINTER_TAB_SELECT_SW         CURSOR          TBLSELSW.CUR
 SAL_RESID_POINTER_PAINTBRUSH            CURSOR          PNTBRSH.CUR
 
-SAL_RESID_BITMAP_50                             BITMAP          50.BMP
+SAL_RESID_BITMAP_50                             BITMAP          "50.BMP"
 
 SAL_RESID_ICON_DEFAULT				ICON		SD.ICO
 
--- vcl/util/makefile.mk.pristine	2009-04-04 19:55:53.764002766 +0200
+++ vcl/util/makefile.mk	2009-04-04 19:55:42.260224623 +0200
@@ -229,13 +229,16 @@ SHL1STDLIBS += $(PSPLIB)
 
 SHL1STDLIBS += $(UWINAPILIB)      \
                $(GDI32LIB)        \
-			   $(GDIPLUSLIB)	  \
                $(MSIMG32LIB)      \
                $(WINSPOOLLIB)     \
                $(OLE32LIB)        \
                $(SHELL32LIB)      \
                $(ADVAPI32LIB)
 
+.IF "$(CROSS_COMPILING)" != "yes"
+SHL1STDLIBS += $(GDIPLUSLIB)
+.ENDIF # "$(CROSS_COMPILING)" != "yes"
+
 .IF "$(COM)" == "GCC"
 SHL1STDLIBS += $(IMM32LIB)
 .ELSE
--- vcl/win/source/gdi/salgdi_gdiplus.cxx.pristine	2009-04-04 20:44:51.940500589 +0200
+++ vcl/win/source/gdi/salgdi_gdiplus.cxx	2009-04-04 20:46:19.256479562 +0200
@@ -39,6 +39,18 @@
 #include <salgdi.h>
 #include <tools/debug.hxx>
 
+#ifdef __MINGW32__
+bool WinSalGraphics::drawPolyPolygon( const ::basegfx::B2DPolyPolygon&, double)
+{
+  return false;
+}
+
+bool WinSalGraphics::drawPolyLine(const basegfx::B2DPolygon&, const basegfx::B2DVector&, basegfx::B2DLineJoin)
+{
+  return false;
+}
+#else /* !__MINGW32__ */
+
 #ifndef min
 #define min(a,b)	(((a) < (b)) ? (a) : (b))
 #endif
@@ -193,3 +205,4 @@ bool WinSalGraphics::drawPolyLine(const 
 }
 
 // -----------------------------------------------------------------------
+#endif /* !__MINGW32__ */

