--- embeddedobj/source/msole/olepersist.cxx.~1.39.~	2008-07-22 16:39:36.000000000 +0200
+++ embeddedobj/source/msole/olepersist.cxx	2008-11-03 15:52:02.000000000 +0100
@@ -176,7 +176,7 @@ sal_Bool KillFile_Impl( const ::rtl::OUS
 
 	return aResult;
 }
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 ::rtl::OUString GetNewFilledTempFile_Impl( const uno::Reference< embed::XOptimizedStorage >& xParentStorage, const ::rtl::OUString& aEntryName, const uno::Reference< lang::XMultiServiceFactory >& xFactory )
 	throw( io::IOException, uno::RuntimeException )
 {
@@ -229,7 +229,7 @@ void LetCommonStoragePassBeUsed_Impl( co
 	xPropSet->setPropertyValue( ::rtl::OUString::createFromAscii( "UseCommonStoragePasswordEncryption" ),
 								uno::makeAny( (sal_Bool)sal_True ) );
 }
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 //------------------------------------------------------
 void VerbExecutionController::StartControlExecution()
 {
@@ -818,7 +818,7 @@ uno::Reference< io::XStream > OleEmbedde
 										m_aTempURL = ::rtl::OUString();
 									}
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 									// retry to create the component after recovering
 									GetRidOfComponent();
 
@@ -889,7 +889,7 @@ void OleEmbeddedObject::SwitchOwnPersist
 	SwitchOwnPersistence( xNewParentStorage, xNewOwnStream, aNewName );
 }
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 //----------------------------------------------
 sal_Bool OleEmbeddedObject::SaveObject_Impl()
 {
@@ -1215,7 +1215,7 @@ void OleEmbeddedObject::StoreToLocation_
 
 	sal_Bool bStoreLoaded = sal_False;
 	if ( m_nObjectState == embed::EmbedStates::LOADED
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 		// if the object was NOT modified after storing it can be just copied
 		// as if it was in loaded state
 	  || ( m_pOleComponent && !m_pOleComponent->IsDirty() )
@@ -1255,7 +1255,7 @@ void OleEmbeddedObject::StoreToLocation_
 		bVisReplIsStored = HasVisReplInStream();
 		bStoreLoaded = sal_True;
 	}
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	else if ( m_pOleComponent )
 	{
 		xTargetStream =
@@ -1457,7 +1457,7 @@ void SAL_CALL OleEmbeddedObject::setPers
 		if ( lArguments[nInd].Name.equalsAscii( "ReadOnly" ) )
 			lArguments[nInd].Value >>= m_bReadOnly;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	sal_Int32 nStorageMode = m_bReadOnly ? embed::ElementModes::READ : embed::ElementModes::READWRITE;
 #endif
 
@@ -1467,7 +1467,7 @@ void SAL_CALL OleEmbeddedObject::setPers
 		if ( lObjArgs[nInd].Name.equalsAscii( "StoreVisualReplacement" ) )
 			lObjArgs[nInd].Value >>= m_bStoreVisRepl;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( nEntryConnectionMode == embed::EntryInitModes::DEFAULT_INIT )
 	{
 		if ( m_bFromClipboard )
@@ -1864,7 +1864,7 @@ void SAL_CALL OleEmbeddedObject::storeOw
 
 	sal_Bool bStoreLoaded = sal_True;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_nObjectState != embed::EmbedStates::LOADED && m_pOleComponent && m_pOleComponent->IsDirty() )
 	{
 		bStoreLoaded = sal_False;
@@ -2078,7 +2078,7 @@ void SAL_CALL OleEmbeddedObject::breakLi
 					uno::Reference< uno::XInterface >( static_cast< ::cppu::OWeakObject* >(this) ) );
 
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_pOleComponent )
 	{
 		// TODO: create an object based on the link
--- embeddedobj/source/msole/oleembed.cxx.~1.24.~	2008-07-22 16:39:09.000000000 +0200
+++ embeddedobj/source/msole/oleembed.cxx	2008-11-03 15:52:51.000000000 +0100
@@ -63,7 +63,7 @@
 
 using namespace ::com::sun::star;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 //----------------------------------------------
 void OleEmbeddedObject::SwitchComponentToRunningState_Impl()
 {
@@ -478,7 +478,7 @@ void SAL_CALL OleEmbeddedObject::changeS
 	if ( m_nObjectState == nNewState )
 		return;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_pOleComponent )
 	{
 		if ( m_nTargetState != -1 )
@@ -631,7 +631,7 @@ uno::Sequence< sal_Int32 > SAL_CALL OleE
 		throw embed::WrongStateException( ::rtl::OUString::createFromAscii( "The object has no persistence!\n" ),
 										uno::Reference< uno::XInterface >( static_cast< ::cppu::OWeakObject* >(this) ) );
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_pOleComponent )
 	{
 		if ( m_nObjectState == embed::EmbedStates::LOADED )
@@ -705,7 +705,7 @@ void SAL_CALL OleEmbeddedObject::doVerb(
 		throw embed::WrongStateException( ::rtl::OUString::createFromAscii( "The object has no persistence!\n" ),
 										uno::Reference< uno::XInterface >( static_cast< ::cppu::OWeakObject* >(this) ) );
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_pOleComponent )
 	{
 		sal_Int32 nOldState = m_nObjectState;
@@ -822,7 +822,7 @@ uno::Sequence< embed::VerbDescriptor > S
 	if ( m_nObjectState == -1 )
 		throw embed::WrongStateException( ::rtl::OUString::createFromAscii( "The object has no persistence!\n" ),
 										uno::Reference< uno::XInterface >( static_cast< ::cppu::OWeakObject* >(this) ) );
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_pOleComponent )
 	{
 		// registry could be used in this case
@@ -987,7 +987,7 @@ sal_Int64 SAL_CALL OleEmbeddedObject::ge
 
 	sal_Int64 nResult = 0;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_bGotStatus && m_nStatusAspect == nAspect )
 		nResult = m_nStatus;
 	else if ( m_pOleComponent )
--- embeddedobj/source/msole/makefile.mk.~1.15.~	2008-04-11 06:59:23.000000000 +0200
+++ embeddedobj/source/msole/makefile.mk	2008-11-01 14:30:00.000000000 +0100
@@ -74,7 +74,7 @@ EXCEPTIONSFILES= \
 		$(SLO)$/olemisc.obj\
 		$(SLO)$/ownview.obj
 
-.IF "$(GUI)"=="WNT"
+.IF "$(GUI)"=="WNT" && "$(CROSS_COMPILING)" == "no"
 
 SLOFILES += \
 		$(SLO)$/graphconvert.obj\
@@ -90,7 +90,7 @@ EXCEPTIONSFILES += \
 		$(SLO)$/xdialogcreator.obj\
 		$(SLO)$/advisesink.obj
 
-.ENDIF
+.ENDIF # "$(GUI)"=="WNT" && "$(CROSS_COMPILING)"
 
 SHL1OBJS= $(SLOFILES)
 
--- embeddedobj/source/msole/olemisc.cxx.~1.28.~	2008-07-22 16:39:22.000000000 +0200
+++ embeddedobj/source/msole/olemisc.cxx	2008-11-03 15:53:24.000000000 +0100
@@ -112,7 +112,7 @@ OleEmbeddedObject::OleEmbeddedObject( co
 , m_bTriedConversion( sal_False )
 {
 }
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 //------------------------------------------------------
 // this constructor let object be initialized from clipboard
 OleEmbeddedObject::OleEmbeddedObject( const uno::Reference< lang::XMultiServiceFactory >& xFactory )
@@ -188,7 +188,7 @@ void OleEmbeddedObject::MakeEventListene
 		}
 	}
 }
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 //----------------------------------------------
 void OleEmbeddedObject::StateChangeNotification_Impl( sal_Bool bBeforeChange, sal_Int32 nOldState, sal_Int32 nNewState )
 {
@@ -233,7 +233,7 @@ void OleEmbeddedObject::StateChangeNotif
 //------------------------------------------------------
 void OleEmbeddedObject::GetRidOfComponent()
 {
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	if ( m_pOleComponent )
 	{
 		if ( m_nObjectState != -1 && m_nObjectState != embed::EmbedStates::LOADED )
--- embeddedobj/source/msole/oleregister.cxx.~1.6.~	2008-04-11 07:01:19.000000000 +0200
+++ embeddedobj/source/msole/oleregister.cxx	2008-11-03 15:52:36.000000000 +0100
@@ -65,7 +65,7 @@ void * SAL_CALL component_getFactory( co
 												OleEmbeddedObjectFactory::impl_staticCreateSelfInstance,
 												OleEmbeddedObjectFactory::impl_staticGetSupportedServiceNames() );
 		}
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 		// the following service makes sence only on windows
 		else if ( aImplName.equals( MSOLEDialogObjectCreator::impl_staticGetImplementationName() ) )
 		{
@@ -102,7 +102,7 @@ sal_Bool SAL_CALL component_writeInfo( v
 			for( sal_Int32 ind = 0; ind < rServices.getLength(); ind++ )
 				xNewKey->createKey( rServices.getConstArray()[ind] );
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 		// the following service makes sence only on windows
 			xNewKey = xKey->createKey( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("/") ) + 
 										MSOLEDialogObjectCreator::impl_staticGetImplementationName() +
--- embeddedobj/source/msole/platform.h.~1.7.~	2008-04-11 07:02:49.000000000 +0200
+++ embeddedobj/source/msole/platform.h	2008-11-01 14:23:32.000000000 +0100
@@ -42,7 +42,9 @@
 #pragma warning(disable: 4548 4555 4917)
 #endif
 #include "windows.h"
-#ifdef __MINGW32__
+#if defined (__MINGW32__) && defined (__CROSS__)
+#include <oleidl.h>
+#elif defined __MINGW32__
 #include <atlbase.h>
 #else
 #if defined(_MSC_VER) && (_MSC_VER > 1310)
--- embeddedobj/source/msole/olevisual.cxx.~1.22.~	2008-07-22 16:39:49.000000000 +0200
+++ embeddedobj/source/msole/olevisual.cxx	2008-11-03 15:53:07.000000000 +0100
@@ -120,7 +120,7 @@ void SAL_CALL OleEmbeddedObject::setVisu
 		throw embed::WrongStateException( ::rtl::OUString::createFromAscii( "The object is not loaded!\n" ),
 									uno::Reference< uno::XInterface >( static_cast< ::cppu::OWeakObject* >(this) ) );
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	// RECOMPOSE_ON_RESIZE misc flag means that the object has to be switched to running state on resize.
 	// SetExtent() is called only for objects that require it,
 	// it should not be called for MSWord documents to workaround problem i49369
@@ -202,7 +202,7 @@ awt::Size SAL_CALL OleEmbeddedObject::ge
 
 	awt::Size aResult;
 
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	// TODO/LATER: Support different aspects
 	if ( m_pOleComponent && !m_bHasSizeToSet && nAspect == embed::Aspects::MSOLE_CONTENT )
 	{
@@ -365,7 +365,7 @@ embed::VisualRepresentation SAL_CALL Ole
 	{
 		return GetVisualRepresentationInNativeFormat_Impl( m_xCachedVisualRepresentation );
 	}
-#ifdef WNT
+#if defined (WNT) && !defined (__CROSS__)
 	else if ( m_pOleComponent )
 	{
 		try
