--- sccomp/source/solver/solver.cxx.pristine	2008-10-30 08:51:53.000000000 +0100
+++ sccomp/source/solver/solver.cxx	2008-10-30 08:51:44.000000000 +0100
@@ -28,6 +28,13 @@
  *
  ************************************************************************/
 
+#ifdef __MINGW32__
+#define BOOL OOO_BOOL
+#define INT32 OOO_INT32
+#define UINT32 OOO_UINT32
+#include <tools/solar.h>
+#endif /* __MINGW32__ */
+
 #undef LANGUAGE_NONE
 #define WINAPI __stdcall
 #define LoadInverseLib FALSE
--- sccomp/source/solver/solver.hxx.pristine	2008-10-30 09:23:23.000000000 +0100
+++ sccomp/source/solver/solver.hxx	2008-10-30 09:22:41.000000000 +0100
@@ -31,6 +31,7 @@
 #ifndef SOLVER_HXX
 #define SOLVER_HXX
 
+#include <sal/types.h>
 #include <com/sun/star/sheet/XSolver.hpp>
 #include <com/sun/star/sheet/XSolverDescription.hpp>
 #include <com/sun/star/lang/XServiceInfo.hpp>
