--- ./rsc/inc/vclrsc.hxx.old	2009-04-02 10:44:31.000000000 +0000
+++ ./rsc/inc/vclrsc.hxx	2009-04-06 16:41:41.000000000 +0000
@@ -176,7 +176,7 @@ typedef sal_Int64 WinBits;
 // from vcl/inc/fldunit.hxx
 enum FieldUnit { FUNIT_NONE, FUNIT_MM, FUNIT_CM, FUNIT_M, FUNIT_KM,
 				 FUNIT_TWIP, FUNIT_POINT, FUNIT_PICA,
-				 FUNIT_INCH, FUNIT_FOOT, FUNIT_MILE, FUNIT_CUSTOM,
+				 FUNIT_INCH, FUNIT_FOOT, FUNIT_MILE, FUNIT_CHAR, FUNIT_LINE, FUNIT_CUSTOM,
 				 FUNIT_PERCENT, FUNIT_100TH_MM };
 
 
--- ./rsc/source/parser/rscibas.cxx.old	2009-04-02 10:44:31.000000000 +0000
+++ ./rsc/source/parser/rscibas.cxx	2009-04-06 16:41:41.000000000 +0000
@@ -240,6 +240,9 @@ RscEnum * RscTypCont::InitFieldUnitsType
 	SETCONST( pFieldUnits, "FUNIT_FOOT", FUNIT_FOOT );
 	SETCONST( pFieldUnits, "FUNIT_MILE", FUNIT_MILE );
 	SETCONST( pFieldUnits, "FUNIT_CUSTOM", FUNIT_CUSTOM );
+	// Amelia : adds two units ,"char" and "line"
+	SETCONST( pFieldUnits, "FUNIT_CHAR", FUNIT_CHAR );
+	SETCONST( pFieldUnits, "FUNIT_LINE", FUNIT_LINE );
 	SETCONST( pFieldUnits, "FUNIT_PERCENT", FUNIT_PERCENT );
 
 	return pFieldUnits;
