--- solenv/inc/wntgcci6.mk.pristine	2008-10-29 08:42:22.000000000 +0100
+++ solenv/inc/wntgcci6.mk	2008-10-30 14:11:42.000000000 +0100
@@ -31,14 +31,19 @@
 
 # mk file for wntgcci6
 
-SOLAR_JAVA*=TRUE
+#SOLAR_JAVA*=TRUE
 FULL_DESK=TRUE
 JAVAFLAGSDEBUG=-g
 
 # SOLAR JAva Unterstuetzung nur fuer wntmsci
 
+.IF "$(CROSS_COMPILING)" == "no"
 ASM=ml
 AFLAGS=/c /Cp /coff
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+ASM=$(NASM)
+REGCOMP:=$(SOLARBINDIR)/regcomp
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 
 # architecture dependent flags for the C and C++ compiler that can be changed by
 # exporting the variable ARCH_FLAGS="..." in the shell, which is used to start build
@@ -50,7 +54,13 @@ CXX*=gcc
 #CFLAGS=-c -Wall -I$(INCLUDE) $(OLE2DEF)
 # new:
 #CYGINC=$(INCLUDE:s/-I /-I/:+"  ":s/;/ -I/:s/-I  //:s/   / /)
-CFLAGS=-fmessage-length=0 -c -nostdinc $(OLE2DEF)
+CFLAGS=-fmessage-length=0 -c $(OLE2DEF)
+.IF "$(CROSS_COMPILING)" == "no"
+CFLAGS += -nostdinc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+CFLAGS += -D__CROSS__
+SCPDEFS += -D__MINGW32__ -D__CROSS__ -DDISABLE_ATL -DUSE_SYSTEM_STL
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 ###
 CFLAGSCC=-pipe $(ARCH_FLAGS)
 CFLAGSCXX=-pipe $(ARCH_FLAGS)
@@ -109,18 +116,37 @@ DYNAMIC= -dynamic
 
 LINK*=$(CXX)
 LINKC*=$(CC)
-
-CYGLIB=$(LIB:s/;/ -L/)
-LINKFLAGS=-nostdlib -Wl,--enable-stdcall-fixup,--enable-runtime-pseudo-reloc -L$(CYGLIB)
+# FIXME: better way to strip trailing semicolon from $(LIB) to avoid
+# CYGLIB content to end with -L, eating next argument
+LIB___=$(LIB)---
+LIB_NO_TRAILING_SEMI___ = $(LIB___:s/;---//)
+LIB_NO_TRAILING_SEMI = $(LIB_NO_TRAILING_SEMI___:s/---//)
+CYGLIB=$(LIB_NO_TRAILING_SEMI:s/;/ -L/)
+LINKFLAGS=-Wl,--enable-stdcall-fixup,--enable-runtime-pseudo-reloc
+.IF "$(CYGLIB)" != ""
+LINKFLAGS += -L$(CYGLIB)
+.ENDIF #"$(CYGLIB)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
+LINKFLAGS += -nostdlib
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+LINKFLAGS += -L$(SOLARBINDIR) -L$(SOLARLIBDIR) -L$(COMPATH)/../../bin -L$(COMPATH)/../../lib
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .IF "$(USE_MINGW)"=="cygwin"
 MINGWLIBDIR=$(COMPATH)$/lib$/mingw
 .ELSE
 MINGWLIBDIR=$(COMPATH)$/lib
 .ENDIF
-LINKFLAGSAPPGUI=-mwindows $(MINGWLIBDIR)$/crt2.o
-LINKFLAGSSHLGUI=--warn-once -mwindows -shared $(MINGWLIBDIR)$/dllcrt2.o
-LINKFLAGSAPPCUI=-mconsole $(MINGWLIBDIR)$/crt2.o
-LINKFLAGSSHLCUI=--warn-once -mconsole -shared $(MINGWLIBDIR)$/dllcrt2.o
+LINKFLAGSAPPGUI=-mwindows
+LINKFLAGSSHLGUI=--warn-once -mwindows -shared
+LINKFLAGSAPPCUI=-mconsole
+LINKFLAGSSHLCUI=--warn-once -mconsole -shared
+.IF "$(CROSS_COMPILING)" == "no"
+LINKFLAGSAPPGUI+=$(MINGWLIBDIR)$/crt2.o
+LINKFLAGSSHLGUI+=$(MINGWLIBDIR)$/dllcrt2.o
+LINKFLAGSAPPCUI+=$(MINGWLIBDIR)$/crt2.o
+LINKFLAGSSHLCUI+=$(MINGWLIBDIR)$/dllcrt2.o
+.ENDIF # "$(CROSS_COMPILING)" == "no"
+
 LINKFLAGSTACK=
 LINKFLAGSPROF=
 LINKFLAGSDEBUG=-g
@@ -144,24 +170,28 @@ STDLIBCUIMT+=-lmingwthrd
 STDSHLGUIMT+=-lmingwthrd
 STDSHLCUIMT+=-lmingwthrd
 .ENDIF
-STDLIBGUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt
-STDLIBCUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt
-STDSHLGUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt
-STDSHLCUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt
+STDLIBGUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt -lversion
+STDLIBCUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt -lversion
+STDSHLGUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt -lversion
+STDSHLCUIMT+=-lmingw32 -lmoldname -lmingwex -Wl,--end-group $(UWINAPILIB) -lm -lkernel32 -luser32 -lmsvcrt -lversion
 
 LIBSTLPORT=-lstlport_gcc
 LIBSTLPORTST=-lstlport_gcc_static
 
-LIBMGR=ar
+LIBMGR=$(AR)
 LIBFLAGS=-rsu
 
-IMPLIB=ld
+IMPLIB=$(LD)
 IMPLIBFLAGS=
 
 MAPSYM=tmapsym
 MAPSYMFLAGS=
 
+.IF "$(CROSS_COMPILING)" == "no"
 RC=rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+RC=$(WINDRES)
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 RCFLAGS=-D__MINGW32__ -DWIN32 -D_WIN32_IE=0x400 -fo$@ $(RCFILES)
 RCLINK=
 RCLINKFLAGS=
@@ -170,12 +200,22 @@ RCSETVERSION=
 DLLPOSTFIX=gi
 PCHPOST=.gch
 
+.IF "$(CROSS_COMPILING)" == "no"
+LINK_PSDK_LIB=$(PSDK_HOME)/lib/
+PSDK_DOT_LIB=.lib
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+DISABLE_ACTIVEX=TRUE
+DISABLE_ATL=TRUE
+LINK_PSDK_LIB=-l
+PSDK_DOT_LIB=
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
+
 ADVAPI32LIB=-ladvapi32
 SHELL32LIB=-lshell32
 GDI32LIB=-lgdi32
 OLE32LIB=-lole32
 OLEAUT32LIB=-loleaut32
-UUIDLIB=$(PSDK_HOME)$/lib$/uuid.lib
+UUIDLIB=$(LINK_PSDK_LIB)uuid$(PSDK_DOT_LIB)
 WINSPOOLLIB=-lwinspool
 IMM32LIB=-limm32
 PSPLIB=-lpsp
@@ -189,14 +228,21 @@ USER32LIB=-luser32
 LIBCMT=-lmsvcrt
 COMDLG32LIB=-lcomdlg32
 COMCTL32LIB=-lcomctl32
-CRYPT32LIB=$(PSDK_HOME)$/lib$/crypt32.lib
-GDIPLUSLIB=$(PSDK_HOME)$/lib$/gdiplus.lib
-DBGHELPLIB=$(PSDK_HOME)$/lib$/dbghelp.lib
-MSILIB=$(PSDK_HOME)$/lib$/msi.lib
-DDRAWLIB=$(DIRECTXSDK_LIB)/ddraw.lib
-SHLWAPILIB=$(PSDK_HOME)$/lib$/shlwapi.lib
-URLMONLIB=$(PSDK_HOME)$/lib$/urlmon.lib
-UNICOWSLIB=$(PSDK_HOME)$/lib$/unicows.lib
+CRYPT32LIB=$(LINK_PSDK_LIB)crypt32$(PSDK_DOT_LIB)
+GDIPLUSLIB=$(LINK_PSDK_LIB)gdiplus$(PSDK_DOT_LIB)
+DBGHELPLIB=$(LINK_PSDK_LIB)dbghelp$(PSDK_DOT_LIB)
+MSILIB=$(LINK_PSDK_LIB)msi$(PSDK_DOT_LIB)
+DDRAWLIB=$(DIRECTXSDK_LIB)/ddraw$(PSDK_DOT_LIB)
+SHLWAPILIB=$(LINK_PSDK_LIB)shlwapi$(PSDK_DOT_LIB)
+URLMONLIB=$(LINK_PSDK_LIB)urlmon$(PSDK_DOT_LIB)
+UNICOWSLIB=$(LINK_PSDK_LIB)unicows$(PSDK_DOT_LIB)
 WININETLIB=-lwininet
 OLDNAMESLIB=-lmoldname
-MSIMG32LIB=$(PSDK_HOME)$/lib$/msimg32.lib
+MSIMG32LIB=$(LINK_PSDK_LIB)msimg32$(PSDK_DOT_LIB)
+
+.IF "$(CROSS_COMPILING)" == "yes"
+CURLLIB=-lcurl
+BERKELEYLIB=-ldb -lwsock32
+XSLTLIB=-lxslt -lxml2
+LIBXML2LIB=-lxml2
+.ENDIF
--- solenv/inc/shlinfo.rc.pristine	2008-10-29 09:33:25.000000000 +0100
+++ solenv/inc/shlinfo.rc	2008-10-29 09:33:33.000000000 +0100
@@ -29,8 +29,9 @@
  ************************************************************************/
 
 #define VERSION			3
-#define SUBVERSION		1
-#define VERVARIANT		0
+#if !defined(VERVARIANT)
+  #define VERVARIANT 		0
+#endif
 
 // .0 + VER_CONCEPT
 // .100 + VER_ALPHA
@@ -90,16 +91,16 @@
 // 				version information
 // -----------------------------------------------------------------------
 
-VS_VERSION_INFO versioninfo
+VS_VERSION_INFO VERSIONINFO
 #ifndef SUBVERSION
-	fileversion    VERSION, 0, 0, VERVARIANT
-	productversion VERSION, 0, 0, VERVARIANT
+	FILEVERSION    VERSION, 0, 0, VERVARIANT
+	PRODUCTVERSION VERSION, 0, 0, VERVARIANT
 #else
-	fileversion    VERSION, VERVARIANT, VER_COUNT
-	productversion VERSION, VERVARIANT, VER_COUNT
+	FILEVERSION    VERSION, VERVARIANT, VER_COUNT
+	PRODUCTVERSION VERSION, VERVARIANT, VER_COUNT
 #endif
-	fileflagsmask 0x3F
-	fileflags
+	FILEFLAGSMASK 0x3F
+	FILEFLAGS
 #if defined(DEBUG)
 	VS_FF_DEBUG |
 #endif
@@ -108,36 +109,36 @@
 #endif
 	0
 #ifndef WIN32
-	fileos VOS_DOS_WINDOWS16
+	FILEOS VOS_DOS_WINDOWS16
 #else
-	fileos VOS_NT_WINDOWS32
+	FILEOS VOS_NT_WINDOWS32
 #endif
-	filetype VFT_APP
+	FILETYPE VFT_APP
 	{
-		block "StringFileInfo"
+		BLOCK "StringFileInfo"
 		{
 #ifdef LG_D
-			block "040704E4"
+			BLOCK "040704E4"
 			{
 				// German StringTable
-				value "CompanyName",		"Sun Microsystems, Inc.\0"
-				value "FileVersion",		PPS(VER_LEVEL) "\0"
-				value "OriginalFilename",	PPS(ORG_NAME) "\0"
-				value "InternalName", 		PPS(INTERNAL_NAME) "\0"
-				value "LegalCopyright",		S_CRIGHT " Sun Microsystems, Inc.\0"
+				VALUE "CompanyName",		"Sun Microsystems, Inc.\0"
+				VALUE "FileVersion",		PPS(VER_LEVEL) "\0"
+				VALUE "OriginalFilename",	PPS(ORG_NAME) "\0"
+				VALUE "InternalName", 		PPS(INTERNAL_NAME) "\0"
+				VALUE "LegalCopyright",		S_CRIGHT " Sun Microsystems, Inc.\0"
 				ADDITIONAL_VERINFO1
 				ADDITIONAL_VERINFO2
 				ADDITIONAL_VERINFO3
 			}
 #else
-			block "040904E4"
+			BLOCK "040904E4"
 			{
 				// International StringTable
-				value "CompanyName", 		"Sun Microsystems, Inc.\0"
-				value "FileVersion",		PPS(VER_LEVEL) "\0"
-				value "OriginalFilename",	PPS(ORG_NAME) "\0"
-				value "InternalName", 		PPS(INTERNAL_NAME) "\0"
-				value "LegalCopyright", 	S_CRIGHT " Sun Microsystems, Inc.\0"
+				VALUE "CompanyName", 		"Sun Microsystems, Inc.\0"
+				VALUE "FileVersion",		PPS(VER_LEVEL) "\0"
+				VALUE "OriginalFilename",	PPS(ORG_NAME) "\0"
+				VALUE "InternalName", 		PPS(INTERNAL_NAME) "\0"
+				VALUE "LegalCopyright", 	S_CRIGHT " Sun Microsystems, Inc.\0"
 				ADDITIONAL_VERINFO1
 				ADDITIONAL_VERINFO2
 				ADDITIONAL_VERINFO3
@@ -145,12 +146,12 @@
 #endif
 		}
 
-		block "VarFileInfo"
+		BLOCK "VarFileInfo"
 		{
 #ifdef LG_D
-			value "Translation", 0x0407, 1252
+			VALUE "Translation", 0x0407, 1252
 #else
-			value "Translation", 0x0409, 1252
+			VALUE "Translation", 0x0409, 1252
 #endif
 		}
 	}
@@ -160,6 +161,6 @@
 {
 	0xF0, "sw", 0x0F, VER_YEAR, VER_MONTH, VER_DAY,
 	VERSION, VERVARIANT, VER_COUNT
-};
+}
 
 #endif // OS2
--- solenv/inc/_tg_shl.mk.pristine	2008-10-29 09:33:25.000000000 +0100
+++ solenv/inc/_tg_shl.mk	2008-10-29 09:33:33.000000000 +0100
@@ -275,7 +275,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL1TARGET:b) >> $(MISC)$/$(SHL1DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL1DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL1DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL1DEFAULTRES:b).rc -o $(MISC)$/$(SHL1DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL1DEFAULTRES)"!=""
 .IF "$(SHL1ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -283,18 +287,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL1ALLRES) > $(SHL1LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL1LINKRES) $(SHL1LINKRESO)
+	$(WINDRES) $(SHL1LINKRES) $(SHL1LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL1ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB1NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL1DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL1DEF) \
 		--dllname $(SHL1TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_1.cmd
 .ELSE			# "$(DEFLIB1NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL1DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL1DEF) \
 		--dllname $(SHL1TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -815,7 +819,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL2TARGET:b) >> $(MISC)$/$(SHL2DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL2DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL2DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL2DEFAULTRES:b).rc -o $(MISC)$/$(SHL2DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL2DEFAULTRES)"!=""
 .IF "$(SHL2ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -823,18 +831,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL2ALLRES) > $(SHL2LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL2LINKRES) $(SHL2LINKRESO)
+	$(WINDRES) $(SHL2LINKRES) $(SHL2LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL2ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB2NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL2DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL2DEF) \
 		--dllname $(SHL2TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_2.cmd
 .ELSE			# "$(DEFLIB2NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL2DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL2DEF) \
 		--dllname $(SHL2TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -1355,7 +1363,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL3TARGET:b) >> $(MISC)$/$(SHL3DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL3DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL3DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL3DEFAULTRES:b).rc -o $(MISC)$/$(SHL3DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL3DEFAULTRES)"!=""
 .IF "$(SHL3ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -1363,18 +1375,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL3ALLRES) > $(SHL3LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL3LINKRES) $(SHL3LINKRESO)
+	$(WINDRES) $(SHL3LINKRES) $(SHL3LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL3ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB3NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL3DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL3DEF) \
 		--dllname $(SHL3TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_3.cmd
 .ELSE			# "$(DEFLIB3NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL3DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL3DEF) \
 		--dllname $(SHL3TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -1895,7 +1907,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL4TARGET:b) >> $(MISC)$/$(SHL4DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL4DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL4DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL4DEFAULTRES:b).rc -o $(MISC)$/$(SHL4DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL4DEFAULTRES)"!=""
 .IF "$(SHL4ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -1903,18 +1919,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL4ALLRES) > $(SHL4LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL4LINKRES) $(SHL4LINKRESO)
+	$(WINDRES) $(SHL4LINKRES) $(SHL4LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL4ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB4NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL4DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL4DEF) \
 		--dllname $(SHL4TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_4.cmd
 .ELSE			# "$(DEFLIB4NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL4DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL4DEF) \
 		--dllname $(SHL4TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -2435,7 +2451,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL5TARGET:b) >> $(MISC)$/$(SHL5DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL5DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL5DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL5DEFAULTRES:b).rc -o $(MISC)$/$(SHL5DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL5DEFAULTRES)"!=""
 .IF "$(SHL5ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -2443,18 +2463,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL5ALLRES) > $(SHL5LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL5LINKRES) $(SHL5LINKRESO)
+	$(WINDRES) $(SHL5LINKRES) $(SHL5LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL5ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB5NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL5DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL5DEF) \
 		--dllname $(SHL5TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_5.cmd
 .ELSE			# "$(DEFLIB5NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL5DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL5DEF) \
 		--dllname $(SHL5TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -2975,7 +2995,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL6TARGET:b) >> $(MISC)$/$(SHL6DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL6DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL6DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL6DEFAULTRES:b).rc -o $(MISC)$/$(SHL6DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL6DEFAULTRES)"!=""
 .IF "$(SHL6ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -2983,18 +3007,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL6ALLRES) > $(SHL6LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL6LINKRES) $(SHL6LINKRESO)
+	$(WINDRES) $(SHL6LINKRES) $(SHL6LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL6ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB6NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL6DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL6DEF) \
 		--dllname $(SHL6TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_6.cmd
 .ELSE			# "$(DEFLIB6NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL6DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL6DEF) \
 		--dllname $(SHL6TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -3515,7 +3539,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL7TARGET:b) >> $(MISC)$/$(SHL7DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL7DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL7DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL7DEFAULTRES:b).rc -o $(MISC)$/$(SHL7DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL7DEFAULTRES)"!=""
 .IF "$(SHL7ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -3523,18 +3551,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL7ALLRES) > $(SHL7LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL7LINKRES) $(SHL7LINKRESO)
+	$(WINDRES) $(SHL7LINKRES) $(SHL7LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL7ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB7NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL7DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL7DEF) \
 		--dllname $(SHL7TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_7.cmd
 .ELSE			# "$(DEFLIB7NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL7DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL7DEF) \
 		--dllname $(SHL7TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -4055,7 +4083,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL8TARGET:b) >> $(MISC)$/$(SHL8DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL8DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL8DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL8DEFAULTRES:b).rc -o $(MISC)$/$(SHL8DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL8DEFAULTRES)"!=""
 .IF "$(SHL8ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -4063,18 +4095,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL8ALLRES) > $(SHL8LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL8LINKRES) $(SHL8LINKRESO)
+	$(WINDRES) $(SHL8LINKRES) $(SHL8LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL8ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB8NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL8DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL8DEF) \
 		--dllname $(SHL8TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_8.cmd
 .ELSE			# "$(DEFLIB8NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL8DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL8DEF) \
 		--dllname $(SHL8TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -4595,7 +4627,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL9TARGET:b) >> $(MISC)$/$(SHL9DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL9DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL9DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL9DEFAULTRES:b).rc -o $(MISC)$/$(SHL9DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL9DEFAULTRES)"!=""
 .IF "$(SHL9ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -4603,18 +4639,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL9ALLRES) > $(SHL9LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL9LINKRES) $(SHL9LINKRESO)
+	$(WINDRES) $(SHL9LINKRES) $(SHL9LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL9ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB9NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL9DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL9DEF) \
 		--dllname $(SHL9TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_9.cmd
 .ELSE			# "$(DEFLIB9NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL9DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL9DEF) \
 		--dllname $(SHL9TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \
@@ -5135,7 +5171,11 @@
 	@echo $(EMQ)#define INTERNAL_NAME $(SHL10TARGET:b) >> $(MISC)$/$(SHL10DEFAULTRES:b).rc
 	@echo $(EMQ)#include $(EMQ)"shlinfo.rc$(EMQ)" >> $(MISC)$/$(SHL10DEFAULTRES:b).rc
 .ENDIF			# "$(use_shl_versions)" != ""
+.IF "$(CROSS_COMPILING)" == "no"
 	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) $(MISC)$/$(SHL10DEFAULTRES:b).rc
+.ELSE # "$(CROSS_COMPILING)" == "yes"
+	$(RC) -DWIN32 $(INCLUDE) $(RCLINKFLAGS) -i $(MISC)$/$(SHL10DEFAULTRES:b).rc -o $(MISC)$/$(SHL10DEFAULTRES:b).res 
+.ENDIF # "$(CROSS_COMPILING)" == "yes"
 .ENDIF			# "$(SHL10DEFAULTRES)"!=""
 .IF "$(SHL10ALLRES)"!=""
 .IF "$(USE_SHELL)"=="4nt"
@@ -5143,18 +5183,18 @@
 .ELSE			# "$(USE_SHELL)"=="4nt"
 	$(TYPE) $(SHL10ALLRES) > $(SHL10LINKRES)
 .IF "$(COM)"=="GCC"
-	windres $(SHL10LINKRES) $(SHL10LINKRESO)
+	$(WINDRES) $(SHL10LINKRES) $(SHL10LINKRESO)
 .ENDIF			# "$(COM)"=="GCC"
 .ENDIF			# "$(USE_SHELL)"=="4nt"
 .ENDIF			# "$(SHL10ALLRES)"!=""
 .IF "$(COM)"=="GCC"	# always have to call dlltool explicitly as ld cannot handle # comment in .def
 .IF "$(DEFLIB10NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL10DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL10DEF) \
 		--dllname $(SHL10TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o > $(MISC)$/$(TARGET).$(@:b)_10.cmd
 .ELSE			# "$(DEFLIB10NAME)"!=""	# do not have to include objs
-	@echo dlltool --input-def $(SHL10DEF) \
+	@echo $(DLLTOOL) --input-def $(SHL10DEF) \
 		--dllname $(SHL10TARGET)$(DLLPOST) \
 		--kill-at \
 		--output-exp $(MISC)$/$(@:b)_exp.o \

--- solenv/inc/_tg_def.mk.orig	2009-06-27 01:53:16.000000000 +0200
+++ solenv/inc/_tg_def.mk	2009-06-27 01:55:08.000000000 +0200
@@ -29,10 +29,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL1OBJS)"!=""
-	-echo $(foreach,i,$(SHL1OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL1OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL1LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL1LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL1LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -80,7 +80,7 @@
 .IF "$(DEFLIB1NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL1TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL1TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL1TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB1NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL1TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL1TARGET).exp
@@ -313,10 +313,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL2OBJS)"!=""
-	-echo $(foreach,i,$(SHL2OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL2OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL2LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL2LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL2LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -364,7 +364,7 @@
 .IF "$(DEFLIB2NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL2TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL2TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL2TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB2NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL2TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL2TARGET).exp
@@ -597,10 +597,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL3OBJS)"!=""
-	-echo $(foreach,i,$(SHL3OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL3OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL3LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL3LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL3LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -648,7 +648,7 @@
 .IF "$(DEFLIB3NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL3TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL3TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL3TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB3NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL3TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL3TARGET).exp
@@ -881,10 +881,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL4OBJS)"!=""
-	-echo $(foreach,i,$(SHL4OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL4OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL4LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL4LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL4LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -932,7 +932,7 @@
 .IF "$(DEFLIB4NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL4TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL4TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL4TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB4NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL4TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL4TARGET).exp
@@ -1165,10 +1165,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL5OBJS)"!=""
-	-echo $(foreach,i,$(SHL5OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL5OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL5LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL5LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL5LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -1216,7 +1216,7 @@
 .IF "$(DEFLIB5NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL5TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL5TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL5TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB5NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL5TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL5TARGET).exp
@@ -1449,10 +1449,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL6OBJS)"!=""
-	-echo $(foreach,i,$(SHL6OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL6OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL6LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL6LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL6LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -1500,7 +1500,7 @@
 .IF "$(DEFLIB6NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL6TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL6TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL6TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB6NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL6TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL6TARGET).exp
@@ -1733,10 +1733,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL7OBJS)"!=""
-	-echo $(foreach,i,$(SHL7OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL7OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL7LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL7LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL7LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -1784,7 +1784,7 @@
 .IF "$(DEFLIB7NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL7TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL7TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL7TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB7NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL7TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL7TARGET).exp
@@ -2017,10 +2017,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL8OBJS)"!=""
-	-echo $(foreach,i,$(SHL8OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL8OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL8LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL8LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL8LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -2068,7 +2068,7 @@
 .IF "$(DEFLIB8NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL8TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL8TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL8TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB8NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL8TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL8TARGET).exp
@@ -2301,10 +2301,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL9OBJS)"!=""
-	-echo $(foreach,i,$(SHL9OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL9OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL9LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL9LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL9LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -2352,7 +2352,7 @@
 .IF "$(DEFLIB9NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL9TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL9TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL9TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB9NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL9TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL9TARGET).exp
@@ -2585,10 +2585,10 @@
 # Shared libraries will be build out of the *.obj files specified in SHL?OBJS and SHL?LIBS
 # Extract RTTI symbols from all the objects that will be used to build a shared library
 .IF "$(SHL10OBJS)"!=""
-	-echo $(foreach,i,$(SHL10OBJS) $i) | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-echo $(foreach,i,$(SHL10OBJS) $i) | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 .IF "$(SHL10LIBS)"!=""
-	-$(TYPE) $(foreach,j,$(SHL10LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 nm -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
+	-$(TYPE) $(foreach,j,$(SHL10LIBS) $j) | $(SED) s\#$(ROUT)\#$(PRJ)$/$(ROUT)\#g | xargs -n1 $(NM) -gP | $(SOLARENV)$/bin$/addsym-mingw.sh $@.symbols-regexp $(MISC)$/symbols-regexp.tmp >> $@.exported-symbols
 .ENDIF
 # overwrite the map file generate into the local output tree with the generated
 # exported symbols list
@@ -2636,7 +2636,7 @@
 .IF "$(DEFLIB10NAME)"!=""
 .IF "$(COM)"=="GCC"
 	@-$(RM) $(MISC)$/$(SHL10TARGET).exp
-	dlltool --output-def $(MISC)$/$(SHL10TARGET).exp --export-all-symbols \
+	$(DLLTOOL) --output-def $(MISC)$/$(SHL10TARGET).exp --export-all-symbols \
  		`$(TYPE) $(foreach,i,$(DEFLIB10NAME) $(SLB)$/$(i).lib) | sed s#$(ROUT)#$(PRJ)$/$(ROUT)#g`
 	tail --lines +3 $(MISC)$/$(SHL10TARGET).exp | sed '/^;/d' >>$@.tmpfile
 	@-$(RM) $(MISC)$/$(SHL10TARGET).exp

--- solenv/inc/_tg_lib.mk.orig	2009-06-28 00:43:24.000000000 +0200
+++ solenv/inc/_tg_lib.mk	2009-06-28 00:44:42.000000000 +0200
@@ -27,7 +27,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB1ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB1FLAGS) $(LIBFLAGS) $(LIB1ARCHIV) `cat $(LIB1TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB1ARCHIV:b).cmd
-	@+echo  ranlib $(LIB1ARCHIV) >> $(MISC)$/$(LIB1ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB1ARCHIV) >> $(MISC)$/$(LIB1ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB1ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB1ARCHIV:b).cmd
 .ELSE
@@ -120,7 +120,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB2ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB2FLAGS) $(LIBFLAGS) $(LIB2ARCHIV) `cat $(LIB2TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB2ARCHIV:b).cmd
-	@+echo  ranlib $(LIB2ARCHIV) >> $(MISC)$/$(LIB2ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB2ARCHIV) >> $(MISC)$/$(LIB2ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB2ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB2ARCHIV:b).cmd
 .ELSE
@@ -213,7 +213,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB3ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB3FLAGS) $(LIBFLAGS) $(LIB3ARCHIV) `cat $(LIB3TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB3ARCHIV:b).cmd
-	@+echo  ranlib $(LIB3ARCHIV) >> $(MISC)$/$(LIB3ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB3ARCHIV) >> $(MISC)$/$(LIB3ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB3ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB3ARCHIV:b).cmd
 .ELSE
@@ -306,7 +306,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB4ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB4FLAGS) $(LIBFLAGS) $(LIB4ARCHIV) `cat $(LIB4TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB4ARCHIV:b).cmd
-	@+echo  ranlib $(LIB4ARCHIV) >> $(MISC)$/$(LIB4ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB4ARCHIV) >> $(MISC)$/$(LIB4ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB4ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB4ARCHIV:b).cmd
 .ELSE
@@ -399,7 +399,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB5ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB5FLAGS) $(LIBFLAGS) $(LIB5ARCHIV) `cat $(LIB5TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB5ARCHIV:b).cmd
-	@+echo  ranlib $(LIB5ARCHIV) >> $(MISC)$/$(LIB5ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB5ARCHIV) >> $(MISC)$/$(LIB5ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB5ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB5ARCHIV:b).cmd
 .ELSE
@@ -492,7 +492,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB6ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB6FLAGS) $(LIBFLAGS) $(LIB6ARCHIV) `cat $(LIB6TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB6ARCHIV:b).cmd
-	@+echo  ranlib $(LIB6ARCHIV) >> $(MISC)$/$(LIB6ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB6ARCHIV) >> $(MISC)$/$(LIB6ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB6ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB6ARCHIV:b).cmd
 .ELSE
@@ -585,7 +585,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB7ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB7FLAGS) $(LIBFLAGS) $(LIB7ARCHIV) `cat $(LIB7TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB7ARCHIV:b).cmd
-	@+echo  ranlib $(LIB7ARCHIV) >> $(MISC)$/$(LIB7ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB7ARCHIV) >> $(MISC)$/$(LIB7ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB7ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB7ARCHIV:b).cmd
 .ELSE
@@ -678,7 +678,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB8ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB8FLAGS) $(LIBFLAGS) $(LIB8ARCHIV) `cat $(LIB8TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB8ARCHIV:b).cmd
-	@+echo  ranlib $(LIB8ARCHIV) >> $(MISC)$/$(LIB8ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB8ARCHIV) >> $(MISC)$/$(LIB8ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB8ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB8ARCHIV:b).cmd
 .ELSE
@@ -771,7 +771,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB9ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB9FLAGS) $(LIBFLAGS) $(LIB9ARCHIV) `cat $(LIB9TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB9ARCHIV:b).cmd
-	@+echo  ranlib $(LIB9ARCHIV) >> $(MISC)$/$(LIB9ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB9ARCHIV) >> $(MISC)$/$(LIB9ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB9ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB9ARCHIV:b).cmd
 .ELSE
@@ -864,7 +864,7 @@
 .IF "$(GUI)$(COM)"=="WNTGCC"
 	@+-$(RM) $(MISC)$/$(LIB10ARCHIV:b).cmd
 	@+echo $(LIBMGR) $(LIB10FLAGS) $(LIBFLAGS) $(LIB10ARCHIV) `cat $(LIB10TARGET) | sed s#'^'$(ROUT)#$(PRJ)$/$(ROUT)#g` > $(MISC)$/$(LIB10ARCHIV:b).cmd
-	@+echo  ranlib $(LIB10ARCHIV) >> $(MISC)$/$(LIB10ARCHIV:b).cmd
+	@+echo  $(RANLIB) $(LIB10ARCHIV) >> $(MISC)$/$(LIB10ARCHIV:b).cmd
 	@cat $(MISC)$/$(LIB10ARCHIV:b).cmd
 	@+source $(MISC)$/$(LIB10ARCHIV:b).cmd
 .ELSE

--- solenv/inc/_tg_app.mk.orig	2009-06-30 12:17:02.000000000 +0200
+++ solenv/inc/_tg_app.mk	2009-06-30 12:17:50.000000000 +0200
@@ -142,7 +142,7 @@
 	@echo mingw
 .IF "$(APP1LINKRES)" != "" || "$(APP1RES)" != ""
 	@cat $(APP1LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP1RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP1RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP1RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP1BASEX) $(APP1STACKN) -o $@ $(APP1OBJS) \
@@ -426,7 +426,7 @@
 	@echo mingw
 .IF "$(APP2LINKRES)" != "" || "$(APP2RES)" != ""
 	@cat $(APP2LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP2RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP2RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP2RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP2BASEX) $(APP2STACKN) -o $@ $(APP2OBJS) \
@@ -710,7 +710,7 @@
 	@echo mingw
 .IF "$(APP3LINKRES)" != "" || "$(APP3RES)" != ""
 	@cat $(APP3LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP3RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP3RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP3RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP3BASEX) $(APP3STACKN) -o $@ $(APP3OBJS) \
@@ -994,7 +994,7 @@
 	@echo mingw
 .IF "$(APP4LINKRES)" != "" || "$(APP4RES)" != ""
 	@cat $(APP4LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP4RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP4RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP4RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP4BASEX) $(APP4STACKN) -o $@ $(APP4OBJS) \
@@ -1278,7 +1278,7 @@
 	@echo mingw
 .IF "$(APP5LINKRES)" != "" || "$(APP5RES)" != ""
 	@cat $(APP5LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP5RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP5RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP5RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP5BASEX) $(APP5STACKN) -o $@ $(APP5OBJS) \
@@ -1562,7 +1562,7 @@
 	@echo mingw
 .IF "$(APP6LINKRES)" != "" || "$(APP6RES)" != ""
 	@cat $(APP6LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP6RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP6RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP6RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP6BASEX) $(APP6STACKN) -o $@ $(APP6OBJS) \
@@ -1846,7 +1846,7 @@
 	@echo mingw
 .IF "$(APP7LINKRES)" != "" || "$(APP7RES)" != ""
 	@cat $(APP7LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP7RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP7RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP7RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP7BASEX) $(APP7STACKN) -o $@ $(APP7OBJS) \
@@ -2130,7 +2130,7 @@
 	@echo mingw
 .IF "$(APP8LINKRES)" != "" || "$(APP8RES)" != ""
 	@cat $(APP8LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP8RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP8RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP8RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP8BASEX) $(APP8STACKN) -o $@ $(APP8OBJS) \
@@ -2414,7 +2414,7 @@
 	@echo mingw
 .IF "$(APP9LINKRES)" != "" || "$(APP9RES)" != ""
 	@cat $(APP9LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP9RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP9RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP9RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP9BASEX) $(APP9STACKN) -o $@ $(APP9OBJS) \
@@ -2698,7 +2698,7 @@
 	@echo mingw
 .IF "$(APP10LINKRES)" != "" || "$(APP10RES)" != ""
 	@cat $(APP10LINKRES) $(subst,$/res$/,$/res{$(subst,$(BIN), $(@:d))} $(APP10RES)) >  $(MISC)$/$(@:b)_all.res
-	windres $(MISC)$/$(@:b)_all.res $(APP10RESO)
+	$(WINDRES) $(MISC)$/$(@:b)_all.res $(APP10RESO)
 .ENDIF
 	@echo $(LINK) $(LINKFLAGS) $(LINKFLAGSAPP) -L$(PRJ)$/$(INPATH)$/lib $(SOLARLIB) $(STDSLO) \
 		$(APP10BASEX) $(APP10STACKN) -o $@ $(APP10OBJS) \
