--- dtrans/test/win32/dnd/transferable.hxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/test/win32/dnd/transferable.hxx	2008-10-29 20:30:12.000000000 +0100
@@ -35,7 +35,7 @@
 //_________________________________________________________________________________________________________________________
 
 
-//#include "..\ImplHelper.hxx"
+//#include "../ImplHelper.hxx"
 
 //_________________________________________________________________________________________________________________________
 //	other includes
@@ -66,7 +66,7 @@
 
 #include <process.h>
 
-#include "..\..\source\win32\ImplHelper.hxx"
+#include "../../source/win32/ImplHelper.hxx"
 
 
 //-------------------------------------------------------------
--- dtrans/source/win32/mtaole/MtaOleClipb.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/mtaole/MtaOleClipb.cxx	2008-10-29 20:30:09.000000000 +0100
@@ -52,7 +52,7 @@
 //#define UNICODE
 #include <osl/diagnose.h>
 
-#include "..\..\inc\MtaOleClipb.hxx"
+#include "../../inc/MtaOleClipb.hxx"
 #include <osl/conditn.hxx>
 
 #include <wchar.h>
--- dtrans/source/win32/clipb/WinClipbImpl.hxx.pristine	2008-10-29 17:21:58.000000000 +0100
+++ dtrans/source/win32/clipb/WinClipbImpl.hxx	2008-10-29 17:22:16.000000000 +0100
@@ -41,7 +41,7 @@
 #include <com/sun/star/datatransfer/XTransferable.hpp>
 #include <com/sun/star/datatransfer/clipboard/XClipboardListener.hpp>
 #include <com/sun/star/datatransfer/clipboard/XClipboardOwner.hpp>
-#include "..\..\inc\MtaOleClipb.hxx"
+#include "../../inc/MtaOleClipb.hxx"
 
 #if defined _MSC_VER
 #pragma warning(push,1)
--- dtrans/source/win32/clipb/WinClipbImpl.cxx.pristine	2008-10-29 17:22:36.000000000 +0100
+++ dtrans/source/win32/clipb/WinClipbImpl.cxx	2008-10-29 17:22:54.000000000 +0100
@@ -38,11 +38,11 @@
 #include "WinClipbImpl.hxx"
 
 #include <systools/win32/comtools.hxx>
-#include "..\..\inc\DtObjFactory.hxx"
-#include "..\dtobj\APNDataObject.hxx"
+#include "../../inc/DtObjFactory.hxx"
+#include "../dtobj/APNDataObject.hxx"
 #include "WinClipboard.hxx"
 #include <com/sun/star/datatransfer/clipboard/RenderingCapabilities.hpp>
-#include "..\dtobj\XNotifyingDataObject.hxx"
+#include "../dtobj/XNotifyingDataObject.hxx"
 
 #if defined _MSC_VER
 #pragma warning(push,1)
--- dtrans/source/win32/ftransl/ftransl.hxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/ftransl/ftransl.hxx	2008-10-29 20:30:05.000000000 +0100
@@ -42,7 +42,7 @@
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <com/sun/star/datatransfer/XDataFormatTranslator.hpp>
 #include <com/sun/star/datatransfer/XMimeContentTypeFactory.hpp>
-#include "..\misc\WinClip.hxx"
+#include "../misc/WinClip.hxx"
 
 #include <vector>
 
--- dtrans/source/win32/ftransl/ftransl.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/ftransl/ftransl.cxx	2008-10-29 20:30:05.000000000 +0100
@@ -40,7 +40,7 @@
 #include "ftransl.hxx"
 #endif
 #include <com/sun/star/datatransfer/XMimeContentType.hpp>
-#include "..\misc\ImplHelper.hxx"
+#include "../misc/ImplHelper.hxx"
 
 #if defined _MSC_VER
 #pragma warning(push,1)
--- dtrans/source/win32/misc/ImplHelper.cxx.pristine	2008-10-29 17:18:03.000000000 +0100
+++ dtrans/source/win32/misc/ImplHelper.cxx	2008-10-29 17:20:12.000000000 +0100
@@ -49,6 +49,129 @@
 #endif
 #ifdef __MINGW32__
 #include <excpt.h>
+#include <setjmp.h>
+#undef OPTIONAL
+#define OPTIONAL
+
+WINBASEAPI
+VOID
+WINAPI
+RtlUnwind (
+    IN PVOID TargetFrame OPTIONAL,
+    IN PVOID TargetIp OPTIONAL,
+    IN PEXCEPTION_RECORD ExceptionRecord OPTIONAL,
+    IN PVOID ReturnValue
+    );
+
+// What library is this supposed to be defined in?
+#define RtlUnwind(a,b,c,d)
+
+class __SEHandler
+{
+ public:
+  __SEHandler() {}
+  ~__SEHandler() {}
+  typedef int (*PF)(void *, LPEXCEPTION_POINTERS);
+  typedef void (*PH)(void *, LPEXCEPTION_POINTERS);
+  typedef void (*PN)(void *);
+  void Set(jmp_buf jb, void *pdata=NULL, PF pfilter=NULL, PH phandlerbody=NULL, PN pfinal=NULL)
+    {
+      __builtin_memcpy(m_jmpbuf, jb, sizeof(jmp_buf));
+      m_pData=pdata;
+      switch (reinterpret_cast<int>(pfilter))
+	{
+	default:
+	  m_filter=pfilter;
+	  break;
+	case EXCEPTION_CONTINUE_EXECUTION:
+	  m_filter=DefaultFilterContinueExecution;
+	  break;
+	case EXCEPTION_EXECUTE_HANDLER:
+	  m_filter=DefaultFilterExecuteHandler;
+	  break;
+	case EXCEPTION_CONTINUE_SEARCH:
+	  m_filter=DefaultFilterContinueSearch;
+	  break;
+	}
+      if (phandlerbody)
+	m_handlerbody=phandlerbody;
+      else
+	m_handlerbody=DefaultHandler;
+      if (pfinal)
+	m_final=pfinal;
+      else
+	m_final=DefaultFinal;
+      m_ER.pHandlerClass = this;
+      m_ER.hp = handler;
+      asm("movl %%fs:0, %%eax\n\t"
+	  "movl %%eax, %0": : "m" (m_ER.prev): "%eax" );
+      asm("movl %0, %%eax\n\t"
+	  "movl %%eax, %%fs:0": : "r" (&m_ER): "%eax" );
+    }
+  void Reset()
+    {
+      m_final(m_pData);
+      asm("movl %0, %%eax \n\t"
+	  "movl %%eax, %%fs:0"
+	  : : "m" (m_ER.prev): "%eax");
+    }
+ private:
+  __SEHandler(const __SEHandler&);
+  __SEHandler& operator=(const __SEHandler&);
+  struct _ER {
+    _ER* prev;
+    PEXCEPTION_HANDLER hp;
+    __SEHandler *pHandlerClass;
+  };
+  static EXCEPTION_DISPOSITION handler(
+		     struct _EXCEPTION_RECORD *pExceptionRecord,
+		     void * EstablisherFrame,
+		     struct _CONTEXT *ContextRecord,
+		     void * /*DispatcherContext*/)
+    {
+      __SEHandler* pThis = reinterpret_cast< _ER * >(EstablisherFrame)->pHandlerClass;
+      if ( pExceptionRecord->ExceptionFlags & EH_UNWINDING )
+	{
+	  pThis->m_final(pThis->m_pData);
+	  return ExceptionContinueSearch;
+	}
+      EXCEPTION_POINTERS ep={pExceptionRecord, ContextRecord};
+      switch ( pThis->m_filter(pThis->m_pData, &ep) )
+	{
+	case EXCEPTION_EXECUTE_HANDLER:
+	  RtlUnwind(EstablisherFrame, &&__set_label, pExceptionRecord, 0);
+__set_label:
+	  pThis->m_handlerbody(pThis->m_pData, &ep);
+	  ContextRecord->Ebp = pThis->m_jmpbuf[0];
+	  ContextRecord->Eip = pThis->m_jmpbuf[1];
+	  ContextRecord->Esp = pThis->m_jmpbuf[2];
+	  return ExceptionContinueExecution;
+	case EXCEPTION_CONTINUE_SEARCH:
+	  return ExceptionContinueSearch;
+	case EXCEPTION_CONTINUE_EXECUTION:
+	  return ExceptionContinueExecution;
+	}
+	  return ExceptionContinueExecution;
+    }
+  static int DefaultFilterContinueSearch(void *, LPEXCEPTION_POINTERS) { return EXCEPTION_CONTINUE_SEARCH; }
+  static int DefaultFilterContinueExecution(void *, LPEXCEPTION_POINTERS) { return EXCEPTION_CONTINUE_EXECUTION; }
+  static int DefaultFilterExecuteHandler(void *, LPEXCEPTION_POINTERS) { return EXCEPTION_EXECUTE_HANDLER; }
+  static void DefaultHandler(void *, LPEXCEPTION_POINTERS) {}
+  static void DefaultFinal(void *) {}
+  typedef int (*handler_p)(
+			   struct _EXCEPTION_RECORD *ExceptionRecord,
+			   void * EstablisherFrame,
+			   struct _CONTEXT *ContextRecord,
+			   void * DispatcherContext);
+  _ER m_ER;
+  void *m_pData;
+  PN m_final;
+  PH m_handlerbody;
+  PF m_filter;
+  jmp_buf m_jmpbuf;
+};
+
+#undef OPTIONAL
 #endif
 
 //------------------------------------------------------------------------

--- dtrans/source/win32/workbench/test_wincb.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/workbench/test_wincb.cxx	2008-10-29 20:30:06.000000000 +0100
@@ -37,7 +37,7 @@
 //_________________________________________________________________________________________________________________________
 
 
-#include "..\misc\ImplHelper.hxx"
+#include "../misc/ImplHelper.hxx"
 
 //_________________________________________________________________________________________________________________________
 //	other includes
--- dtrans/source/win32/workbench/XTDo.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/workbench/XTDo.cxx	2008-10-29 20:28:07.000000000 +0100
@@ -36,7 +36,7 @@
 //------------------------------------------------------------------------
 #include <osl/diagnose.h>
 
-#include "..\DTransHelper.hxx"
+#include "../DTransHelper.hxx"
 
 #ifndef _TWRAPPERDATAOBJECT_HXX_
 #include "XTDo.hxx"
--- dtrans/source/win32/dnd/source.hxx.pristine	2008-10-29 20:55:40.000000000 +0100
+++ dtrans/source/win32/dnd/source.hxx	2008-10-29 21:09:00.000000000 +0100
@@ -43,6 +43,7 @@
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include "../../inc/DtObjFactory.hxx"
 #include "globals.hxx"
+#include <windows.h>
 #include <oleidl.h>
 
 #include <systools/win32/comtools.hxx>
--- dtrans/source/win32/dnd/source.cxx.pristine	2008-10-29 20:59:32.000000000 +0100
+++ dtrans/source/win32/dnd/source.cxx	2008-10-29 21:08:39.000000000 +0100
@@ -262,7 +262,7 @@ HRESULT STDMETHODCALLTYPE DragSource::Qu
 //----------------------------------------------------
 /**
 */
-ULONG STDMETHODCALLTYPE DragSource::AddRef( void)
+ULONG STDMETHODCALLTYPE DragSource::AddRef()
 {
 	acquire();
 	return (ULONG) m_refCount;
@@ -271,7 +271,7 @@ ULONG STDMETHODCALLTYPE DragSource::AddR
 //----------------------------------------------------
 /**
 */
-ULONG STDMETHODCALLTYPE DragSource::Release( void)
+ULONG STDMETHODCALLTYPE DragSource::Release()
 {
     ULONG ref= m_refCount;
 	release();
--- dtrans/source/win32/dnd/target.hxx.pristine	2008-10-29 20:56:02.000000000 +0100
+++ dtrans/source/win32/dnd/target.hxx	2008-10-29 20:56:20.000000000 +0100
@@ -46,6 +46,7 @@
 #if defined _MSC_VER
 #pragma warning(push,1)
 #endif
+#include <windows.h>
 #include <oleidl.h>
 #if defined _MSC_VER
 #pragma warning(pop)
--- dtrans/source/win32/dtobj/TxtCnvtHlp.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/TxtCnvtHlp.cxx	2008-10-29 20:30:06.000000000 +0100
@@ -37,7 +37,7 @@
 #include <osl/diagnose.h>
 #include "TxtCnvtHlp.hxx"
 #include "DTransHelper.hxx"
-#include "..\misc\ImplHelper.hxx"
+#include "../misc/ImplHelper.hxx"
 
 using namespace ::com::sun::star::datatransfer;
 using namespace ::com::sun::star::uno;
--- dtrans/source/win32/dtobj/DOTransferable.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/DOTransferable.cxx	2008-10-29 20:30:06.000000000 +0100
@@ -40,10 +40,10 @@
 #ifndef _DOWRAPPERTRANSFERABLE_HXX_
 #include "DOTransferable.hxx"
 #endif
-#include "..\misc\ImplHelper.hxx"
-#include "..\misc\WinClip.hxx"
+#include "../misc/ImplHelper.hxx"
+#include "../misc/WinClip.hxx"
 #include "DTransHelper.hxx"
-#include "..\misc\ImplHelper.hxx"
+#include "../misc/ImplHelper.hxx"
 #include "TxtCnvtHlp.hxx"
 #include "MimeAttrib.hxx"
 #include "FmtFilter.hxx"
--- dtrans/source/win32/dtobj/FetcList.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/FetcList.cxx	2008-10-29 20:30:06.000000000 +0100
@@ -43,8 +43,8 @@
 #ifndef _DATAFORMATTRANSLATOR_HXX_
 #include "DataFmtTransl.hxx"
 #endif
-#include "..\misc\ImplHelper.hxx"
-#include "..\misc\WinClip.hxx"
+#include "../misc/ImplHelper.hxx"
+#include "../misc/WinClip.hxx"
 
 #include <algorithm>
 
--- dtrans/source/win32/dtobj/DtObjFactory.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/DtObjFactory.cxx	2008-10-29 20:30:09.000000000 +0100
@@ -34,7 +34,7 @@
 //------------------------------------------------------------------------
 // includes
 //------------------------------------------------------------------------
-#include "..\..\inc\DtObjFactory.hxx"
+#include "../../inc/DtObjFactory.hxx"
 
 #ifndef _TWRAPPERDATAOBJECT_HXX_
 #include "XTDataObject.hxx"
--- dtrans/source/win32/dtobj/Fetc.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/Fetc.cxx	2008-10-29 20:30:06.000000000 +0100
@@ -36,7 +36,7 @@
 //------------------------------------------------------------------------
 #include <osl/diagnose.h>
 #include "Fetc.hxx"
-#include "..\misc\ImplHelper.hxx"
+#include "../misc/ImplHelper.hxx"
 
 //------------------------------------------------------------------------
 // 
--- dtrans/source/win32/dtobj/DataFmtTransl.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/DataFmtTransl.cxx	2008-10-29 20:30:06.000000000 +0100
@@ -38,8 +38,8 @@
 #include <rtl/string.hxx>
 #include <osl/diagnose.h>
 #include <rtl/tencinfo.h>
-#include "..\misc\ImplHelper.hxx"
-#include "..\misc\WinClip.hxx"
+#include "../misc/ImplHelper.hxx"
+#include "../misc/WinClip.hxx"
 #include "MimeAttrib.hxx"
 #include "DTransHelper.hxx"
 #include <rtl/string.h>
--- dtrans/source/win32/dtobj/XTDataObject.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/XTDataObject.cxx	2008-10-29 20:42:42.000000000 +0100
@@ -39,8 +39,8 @@
 #ifndef _TXDATAOBJECT_HXX_
 #include "XTDataObject.hxx"
 #endif
-#include <com/sun/star/datatransfer/dataflavor.hpp>
-#include "..\misc\ImplHelper.hxx"
+#include <com/sun/star/datatransfer/DataFlavor.hpp>
+#include "../misc/ImplHelper.hxx"
 #include "DTransHelper.hxx"
 #include "TxtCnvtHlp.hxx"
 #include <com/sun/star/datatransfer/clipboard/XClipboardEx.hpp>
--- dtrans/source/win32/dtobj/XTDataObject.hxx.pristine	2008-10-29 20:41:49.000000000 +0100
+++ dtrans/source/win32/dtobj/XTDataObject.hxx	2008-10-29 20:41:28.000000000 +0100
@@ -45,7 +45,7 @@
 #endif
 
 #ifndef _FETCLIST_HXX_
-#include "FEtcList.hxx"
+#include "FetcList.hxx"
 #endif
 
 #if defined _MSC_VER
--- dtrans/source/win32/dtobj/FmtFilter.cxx.pristine	2008-10-29 20:12:03.000000000 +0100
+++ dtrans/source/win32/dtobj/FmtFilter.cxx	2008-10-29 20:51:01.000000000 +0100
@@ -38,9 +38,14 @@
 #pragma warning(push,1)
 #pragma warning(disable:4917)
 #endif
-#include <Shobjidl.h>
+#include <windows.h>
+#ifndef __MINGW32__
+#include <shobjidl.h>
+#else /* __MINGW32__ */
+#include <shlobj.h>
+#endif /* __MINGW32__ */
 #include <shlguid.h>
-#include <ObjIdl.h>
+#include <objidl.h>
 #include <shellapi.h>
 #if defined _MSC_VER
 #pragma warning(pop)
--- dtrans/source/win32/dtobj/XNotifyingDataObject.cxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/XNotifyingDataObject.cxx	2008-10-29 20:30:09.000000000 +0100
@@ -32,9 +32,9 @@
 #include "precompiled_dtrans.hxx"
 #include <osl/diagnose.h>
 #include "XNotifyingDataObject.hxx"
-#include "..\clipb\WinClipbImpl.hxx"
-#include "..\clipb\WinClipboard.hxx"
-#include "..\..\inc\DtObjFactory.hxx"
+#include "../clipb/WinClipbImpl.hxx"
+#include "../clipb/WinClipboard.hxx"
+#include "../../inc/DtObjFactory.hxx"
 
 #ifdef __MINGW32__
 #define __uuidof(I) IID_##I
--- dtrans/source/win32/dtobj/DTransHelper.hxx.pristine	2008-10-29 20:36:59.000000000 +0100
+++ dtrans/source/win32/dtobj/DTransHelper.hxx	2008-10-29 20:30:06.000000000 +0100
@@ -43,7 +43,7 @@
 #if defined _MSC_VER
 #pragma warning(pop)
 #endif
-#include "..\misc\WinClip.hxx"
+#include "../misc/WinClip.hxx"
 
 //------------------------------------------------------------------------
 // defines
--- dtrans/source/win32/dtobj/XNotifyingDataObject.hxx.pristine	2008-10-29 20:10:38.000000000 +0100
+++ dtrans/source/win32/dtobj/XNotifyingDataObject.hxx	2008-10-29 20:10:50.000000000 +0100
@@ -43,8 +43,8 @@
 #if defined _MSC_VER
 #pragma warning(push,1)
 #endif
-#include <objidl.h>
 #include <windows.h>
+#include <objidl.h>
 #if defined _MSC_VER
 #pragma warning(pop)
 #endif
--- dtrans/source/inc/MtaOleClipb.hxx.pristine	2008-10-29 20:11:24.000000000 +0100
+++ dtrans/source/inc/MtaOleClipb.hxx	2008-10-29 20:11:39.000000000 +0100
@@ -37,6 +37,7 @@
 #if defined _MSC_VER
 #pragma warning(push,1)
 #endif
+#include <windows.h>
 #include <objidl.h>
 #if defined _MSC_VER
 #pragma warning(pop)
