diff --git officecfg/registry/schema/org/openoffice/Office/Calc.xcs officecfg/registry/schema/org/openoffice/Office/Calc.xcs
index 3c3fe6b..09e5c52 100644
--- officecfg/registry/schema/org/openoffice/Office/Calc.xcs
+++ officecfg/registry/schema/org/openoffice/Office/Calc.xcs
@@ -1019,6 +1019,81 @@
 				<value>NULL</value>
 			</prop>
 		</group>
+        <group oor:name="Dialogs">
+            <info>
+                <desc>Contains the dialogs settings.</desc>
+            </info>
+            <group oor:name="CSVImport">
+                <info>
+                    <desc>Contains setting for Text CSV Import</desc>
+                </info>
+                <prop oor:name="MergeDelimiters" oor:type="xs:boolean">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>Merge Delimiter check box status</desc>
+                        <label>MergeDelimiters</label>
+                    </info>
+                    <value>false</value>
+                </prop>
+                <prop oor:name="QuotedFieldAsText" oor:type="xs:boolean">
+                    <info>
+                        <author>kyoshida</author>
+                        <desc>If true, quoted field is always imported as text 
+                            with no exception.</desc>
+                        <label>QuotedFieldAsText</label>
+                    </info>
+                    <value>true</value>
+                </prop>
+                <prop oor:name="Separators" oor:type="xs:string">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>List of Separators - as a String</desc>
+                        <label>Separators</label>
+                    </info>
+                    <value>;    </value>
+                </prop>
+                <prop oor:name="TextSeparators" oor:type="xs:string">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>Text Separators</desc>
+                        <label>TextSeparators</label>
+                    </info>
+                    <value>"</value>
+                </prop>
+                <prop oor:name="FixedWidth" oor:type="xs:boolean">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>Fixed width</desc>
+                        <label>FixedWidth</label>
+                    </info>
+                    <value>false</value>
+                </prop>
+                <prop oor:name="FromRow" oor:type="xs:int">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>From Row</desc>
+                        <label>FromRow</label>
+                    </info>
+                    <value>1</value>
+                </prop>
+                <prop oor:name="CharSet" oor:type="xs:int">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>Char Set</desc>
+                        <label>CharSet</label>
+                    </info>
+                    <value>-1</value>
+                </prop>
+                <prop oor:name="FixedWidthList" oor:type="xs:string">
+                    <info>
+                        <author>muthusuba</author>
+                        <desc>Fixed Width List of separators</desc>
+                        <label>FixedWidthList</label>
+                    </info>
+                    <value></value>
+                </prop>
+            </group>
+        </group>
 		<group oor:name="Calculate">
 			<info>
 				<desc>Contains settings that affect cell calculation.</desc>
