Index: officecfg/registry/data/org/openoffice/ucb/Configuration.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/ucb/Configuration.xcu,v
retrieving revision 1.12
retrieving revision 1.10.112.2
diff -u -p -u -p -B -r1.12 -r1.10.112.2
--- officecfg/registry/data/org/openoffice/ucb/Configuration.xcu	14 May 2008 14:54:47 -0000	1.12
+++ officecfg/registry/data/org/openoffice/ucb/Configuration.xcu	2 Jun 2008 13:30:36 -0000	1.10.112.2
@@ -106,17 +145,6 @@
                 <value/>
               </prop>
             </node>
-            <node oor:name="Provider7" oor:op="replace">
-              <prop oor:name="ServiceName">
-                <value>com.sun.star.ucb.WebDAVContentProvider</value>
-              </prop>
-              <prop oor:name="URLTemplate">
-                <value>dav</value>
-              </prop>
-              <prop oor:name="Arguments">
-                <value/>
-              </prop>
-            </node>
             <node oor:name="Provider8" oor:op="replace">
               <prop oor:name="ServiceName">
                 <value>com.sun.star.ucb.PackageContentProvider</value>
@@ -145,17 +145,50 @@
                 <value/>
               </prop>
             </node>
-            <node oor:name="Provider13" oor:op="replace">
-              <prop oor:name="ServiceName">
-                <value>com.sun.star.ucb.WebDAVContentProvider</value>
-              </prop>
-              <prop oor:name="URLTemplate">
-                <value>davs</value>
-              </prop>
-              <prop oor:name="Arguments">
-                <value/>
-              </prop>
-            </node>
+            <node oor:name="Provider13" oor:op="replace">
+                <prop oor:name="ServiceName">
+                    <value>com.sun.star.ucb.WebDAVContentProvider</value>
+                </prop>
+                <prop oor:name="URLTemplate">
+                    <value>dav</value>
+                </prop>
+                <prop oor:name="Arguments">
+                    <value/>
+                </prop>
+            </node>
+            <node oor:name="Provider14" oor:op="replace">
+                <prop oor:name="ServiceName">
+                    <value>com.sun.star.ucb.WebDAVContentProvider</value>
+                </prop>
+                <prop oor:name="URLTemplate">
+                    <value>davs</value>
+                </prop>
+                <prop oor:name="Arguments">
+                    <value/>
+                </prop>
+            </node>
+            <node oor:name="Provider15" oor:op="replace">
+                <prop oor:name="ServiceName">
+                    <value>com.sun.star.ucb.WebDAVContentProvider</value>
+                </prop>
+                <prop oor:name="URLTemplate">
+                    <value>webdav</value>
+                </prop>
+                <prop oor:name="Arguments">
+                    <value/>
+                </prop>
+            </node>
+            <node oor:name="Provider16" oor:op="replace">
+                <prop oor:name="ServiceName">
+                    <value>com.sun.star.ucb.WebDAVContentProvider</value>
+                </prop>
+                <prop oor:name="URLTemplate">
+                    <value>webdavs</value>
+                </prop>
+                <prop oor:name="Arguments">
+                    <value/>
+                </prop>
+            </node>
             <!-- We want the Provider to be the final fallback provider -->
             <node oor:name="Provider999" oor:op="replace" install:module="gio">
               <prop oor:name="ServiceName">
