Index: sw/source/filter/ww8/wrtww8.hxx
===================================================================
--- sw/source/filter/ww8/wrtww8.hxx	(revision 273252)
+++ sw/source/filter/ww8/wrtww8.hxx	(revision 273251)
@@ -296,9 +296,7 @@
     wwFont( const String &rFamilyName, FontPitch ePitch, FontFamily eFamily,
         rtl_TextEncoding eChrSet, bool bWrtWW8 );
     bool Write( SvStream *pTableStram ) const;
-#ifdef DOCX
     void WriteDocx( const DocxAttributeOutput* rAttrOutput ) const;
-#endif
     rtl::OUString GetFamilyName() const { return rtl::OUString( msFamilyNm ); }
     friend bool operator < (const wwFont &r1, const wwFont &r2);
 };
@@ -321,9 +319,7 @@
     USHORT GetId(const SvxFontItem& rFont);
     USHORT GetId(const wwFont& rFont);
     void WriteFontTable( SvStream *pTableStream, WW8Fib& pFib );
-#ifdef DOCX
     void WriteFontTable( const DocxAttributeOutput& rAttrOutput );
-#endif
 };
 
 class DrawObj
Index: sw/source/filter/ww8/wrtw8sty.cxx
===================================================================
--- sw/source/filter/ww8/wrtw8sty.cxx	(revision 273252)
+++ sw/source/filter/ww8/wrtw8sty.cxx	(revision 273251)
@@ -691,7 +691,6 @@
     return true;
 }
 
-#ifdef DOCX
 void wwFont::WriteDocx( const DocxAttributeOutput* rAttrOutput ) const
 {
     // no font embedding, panose id, subsetting, ... implemented
@@ -706,7 +705,6 @@
 
     rAttrOutput->EndFont();
 }
-#endif
 
 bool operator<(const wwFont &r1, const wwFont &r2)
 {
@@ -824,7 +822,6 @@
     }
 }
 
-#ifdef DOCX
 void wwFontHelper::WriteFontTable( const DocxAttributeOutput& rAttrOutput )
 {
     ::std::vector<const wwFont *> aFontList( AsVector() );
@@ -832,7 +829,6 @@
     ::std::for_each( aFontList.begin(), aFontList.end(),
         ::std::bind2nd( ::std::mem_fun( &wwFont::WriteDocx ), &rAttrOutput ) );
 }
-#endif
 
 /*  */
 
Index: sw/source/filter/ww8/makefile.mk
===================================================================
--- sw/source/filter/ww8/makefile.mk	(revision 273252)
+++ sw/source/filter/ww8/makefile.mk	(revision 273251)
@@ -62,6 +62,7 @@
 		$(SLO)$/wrtw8num.obj \
 		$(SLO)$/wrtw8sty.obj \
 		$(SLO)$/wrtww8.obj \
+		$(SLO)$/docxexportfilter.obj \
 		$(SLO)$/ww8atr.obj \
 		$(SLO)$/ww8par.obj \
 		$(SLO)$/ww8par6.obj \
@@ -79,6 +80,9 @@
 		$(SLO)$/wrtw8sty.obj \
 		$(SLO)$/wrtww8.obj \
 		$(SLO)$/wrtww8gr.obj \
+		$(SLO)$/docxattributeoutput.obj \
+		$(SLO)$/docxexportfilter.obj \
+		$(SLO)$/docxexport.obj \
 		$(SLO)$/ww8atr.obj \
 		$(SLO)$/ww8graf.obj \
 		$(SLO)$/ww8graf2.obj \
Index: sw/util/makefile.mk
===================================================================
--- sw/util/makefile.mk	(revision 273252)
+++ sw/util/makefile.mk	(revision 273251)
@@ -320,6 +320,8 @@
 
 SHL4STDLIBS= \
 	$(ISWLIB) \
+	$(OOXLIB) \
+	$(SAXLIB) \
 	$(SVXCORELIB) \
 	$(SVXMSFILTERLIB) \
 	$(SFXLIB) \
Index: sw/util/msword.map
===================================================================
--- sw/util/msword.map	(revision 273252)
+++ sw/util/msword.map	(revision 273251)
@@ -4,6 +4,9 @@
                 ExportRTF;
                 ImportDOC;
                 ExportDOC;
+                component_getImplementationEnvironment;
+                component_writeInfo;
+                component_getFactory;
 
         local:
                 *;
Index: filter/source/config/fragments/filters/MS_Word_2007_XML.xcu
===================================================================
--- filter/source/config/fragments/filters/MS_Word_2007_XML.xcu	(revision 273252)
+++ filter/source/config/fragments/filters/MS_Word_2007_XML.xcu	(revision 273251)
@@ -1,5 +1,5 @@
     <node oor:name="MS Word 2007 XML" oor:op="replace">
-        <prop oor:name="Flags"><value>IMPORT ALIEN 3RDPARTYFILTER</value></prop>
+        <prop oor:name="Flags"><value>IMPORT EXPORT ALIEN 3RDPARTYFILTER</value></prop>
         <prop oor:name="UIComponent"/>
         <prop oor:name="FilterService"><value>com.sun.star.comp.Writer.WriterFilter</value></prop>
         <prop oor:name="UserData"><value>OXML</value></prop>
Index: oox/prj/build.lst
===================================================================
--- oox/prj/build.lst	(revision 273252)
+++ oox/prj/build.lst	(revision 273251)
@@ -1,4 +1,4 @@
-oox	oox : vos cppu cppuhelper comphelper sal offapi sax basegfx tools vcl BOOST:boost NULL
+oox	oox : vos cppu cppuhelper comphelper sal offapi sax basegfx svx tools vcl BOOST:boost NULL
 oox	oox				usr1	-   all	oox_mkout NULL
 oox	oox\prj				get	-   all	oox_prj NULL
 oox	oox\source\token		nmake	-   all	oox_token NULL
@@ -15,4 +15,5 @@
 oox	oox\source\xls			nmake	-   all	oox_xls oox_token NULL
 oox	oox\source\dump			nmake	-   all	oox_dump oox_token NULL
 oox	oox\source\shape		nmake   -   all oox_shape oox_token NULL
-oox	oox\util			nmake   -   all oox_util oox_token oox_helper oox_core oox_ole oox_vml oox_drawingml oox_diagram oox_chart oox_table oox_ppt oox_xls oox_dump oox_shape oox_docprop NULL
+oox	oox\source\export		nmake   -   all oox_export oox_token NULL
+oox	oox\util			nmake   -   all oox_util oox_token oox_helper oox_core oox_ole oox_vml oox_drawingml oox_diagram oox_chart oox_table oox_ppt oox_xls oox_dump oox_export oox_shape oox_docprop NULL
Index: oox/prj/d.lst
===================================================================
--- oox/prj/d.lst	(revision 273252)
+++ oox/prj/d.lst	(revision 273251)
@@ -1,5 +1,6 @@
 mkdir: %_DEST%\inc%_EXT%\oox
 mkdir: %_DEST%\inc%_EXT%\oox\core
+mkdir: %_DEST%\inc%_EXT%\oox\export
 mkdir: %_DEST%\inc%_EXT%\oox\helper
 mkdir: %_DEST%\inc%_EXT%\oox\vml
 mkdir: %_DEST%\inc%_EXT%\oox\drawingml
@@ -27,6 +28,7 @@
 ..\inc\oox\drawingml\chart\chartconverter.hxx %_DEST%\inc%_EXT%\oox\drawingml\chart\chartconverter.hxx
 ..\inc\oox\drawingml\table\tablestylelist.hxx %_DEST%\inc%_EXT%\oox\drawingml\table\tablestylelist.hxx
 ..\inc\oox\vml\shape.hxx %_DEST%\inc%_EXT%\oox\vml\shape.hxx
+..\inc\oox\export\*.hxx %_DEST%\inc%_EXT%\oox\export\*.hxx
 
 ..\%__SRC%\lib\*.dylib %_DEST%\lib%_EXT%\*.dylib
 
Index: oox/util/makefile.mk
===================================================================
--- oox/util/makefile.mk	(revision 273252)
+++ oox/util/makefile.mk	(revision 273251)
@@ -57,7 +57,8 @@
     $(SLB)$/table.lib\
     $(SLB)$/shape.lib\
     $(SLB)$/dump.lib\
-    $(SLB)$/docprop.lib
+    $(SLB)$/docprop.lib\
+    $(SLB)$/export.lib
 
 # --- Shared-Library -----------------------------------------------
 
@@ -72,7 +73,13 @@
 		$(RTLLIB)		\
 		$(SALLIB)		\
 		$(BASEGFXLIB)	\
-		$(SAXLIB)
+		$(SAXLIB)		\
+		$(VCLLIB)		\
+		$(GOODIESLIB)		\
+		$(SVTOOLLIB)		\
+		$(SVXCORELIB)		\
+		$(SVXMSFILTERLIB)	\
+		$(TOOLSLIB)
 
 SHL1DEF=    $(MISC)$/$(SHL1TARGET).def
 SHL1LIBS=   $(LIB1TARGET)
