Index: reportbuilder/java/com/sun/star/report/pentaho/makefile.mk
===================================================================
--- reportbuilder/java/com/sun/star/report/pentaho/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ reportbuilder/java/com/sun/star/report/pentaho/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -42,11 +42,14 @@
 JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar java_uno.jar
 .IF "$(SYSTEM_JFREEREPORT)" == "YES"
 XCLASSPATH!:=$(XCLASSPATH)$(FLUTE_JAR)$(PATH_SEPERATOR)$(LIBBASE_JAR)$(PATH_SEPERATOR)$(LIBXML_JAR)$(PATH_SEPERATOR)$(JFREEREPORT_JAR)$(PATH_SEPERATOR)$(LIBLOADER_JAR)$(PATH_SEPERATOR)$(SAC_JAR)$(PATH_SEPERATOR)$(LIBLAYOUT_JAR)$(PATH_SEPERATOR)$(LIBSERIALIZER_JAR)$(PATH_SEPERATOR)$(LIBFONTS_JAR)$(PATH_SEPERATOR)$(LIBFORMULA_JAR)$(PATH_SEPERATOR)$(LIBREPOSITORY_JAR)$(PATH_SEPERATOR)$(COMMONS_LOGGING_JAR)
-COMP=fix_system_jfreereport
 .ELSE
 JARFILES += flute-1.3.0.jar libbase-1.0.0.jar libfonts-1.0.0.jar libformula-0.2.0.jar liblayout-0.2.9.jar libloader-1.0.0.jar librepository-1.0.0.jar libxml-1.0.0.jar flow-engine-0.9.2.jar sac.jar commons-logging-1.1.1.jar
 .ENDIF
 
+.IF "$(SYSTEM_JFREEREPORT)" == "YES" || "$(SYSTEM_APACHE_COMMONS)" == "YES"
+COMP=fix_system_libs
+.ENDIF
+
 JAVAFILES       := $(shell @$(FIND) . -name "*.java")
 # JAVACLASSFILES	= $(foreach,i,$(JAVAFILES) $(CLASSDIR)$/$(PACKAGE)$/$(i:s/.java/.class/))
 
@@ -94,13 +97,19 @@
 $(JARTARGETN) : $(COMP) $(PROPERTYFILES) $(CSSFILES) $(XSDFILES) $(TXTFILES) $(XMLFILES)
 .ENDIF          # "$(JARTARGETN)"!=""
 
-fix_system_jfreereport:
-	@echo "Fix Java Class-Path entry for JFree JFreeReport libraries from system."
+fix_system_libs:
+	@echo "Fix Java Class-Path entry for libraries from system."
+.IF ("$(SYSTEM_JFREEREPORT)" != "YES" && "$(SYSTEM_APACHE_COMMONS)" == "YES")
+	@$(SED) -r -e "s#commons-logging-1.1.1.jar#file://$(COMMONS_LOGGING_JAR)#" \
+		-i ../../../../../../$(INPATH)/class/sun-report-builder/META-INF/MANIFEST.MF
+.ENDIF
+.IF ("$(SYSTEM_JFREEREPORT)" == "YES" && "$(SYSTEM_APACHE_COMMONS)" == "YES")
 	@$(SED) '/flute/,/sac/d' -i ../../../../../../$(INPATH)/class/sun-report-builder/META-INF/MANIFEST.MF
 	@$(SED) -r -e "s#^Class-Path.*#\0\n  file://$(LIBBASE_JAR)\n  file://$(SAC_JAR)\n  file://$(LIBXML_JAR)\n\
   file://$(FLUTE_JAR)\n  file://$(JFREEREPORT_JAR)\n  file://$(LIBLAYOUT_JAR)\n  file://$(LIBLOADER_JAR)\n  file://$(LIBFORMULA_JAR)\n\
-  file://$(LIBREPOSITORY_JAR)\n  file://$(LIBFONTS_JAR)\n  file://$(LIBSERIALIZER_JAR)#" \
+  file://$(LIBREPOSITORY_JAR)\n  file://$(LIBFONTS_JAR)\n  file://$(LIBSERIALIZER_JAR)\n  file://$(COMMONS_LOGGING_JAR)#" \
 	-i ../../../../../../$(INPATH)/class/sun-report-builder/META-INF/MANIFEST.MF
+.ENDIF
 
 $(CLASSDIR)$/$(PACKAGE)$/%.properties : %.properties
 	@@-$(MKDIRHIER) $(@:d)
Index: reportbuilder/util/makefile.mk
===================================================================
--- reportbuilder/util/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ reportbuilder/util/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -110,7 +110,10 @@
 	$(EXTENSIONDIR)$/librepository-1.0.0.jar					\
 	$(EXTENSIONDIR)$/libxml-1.0.0.jar							\
 	$(EXTENSIONDIR)$/flow-engine-0.9.2.jar 	                    \
-	$(EXTENSIONDIR)$/sac.jar                                 	\
+	$(EXTENSIONDIR)$/sac.jar
+.ENDIF
+.IF "$(SYSTEM_APACHE_COMMONS)" != "YES"
+COMPONENT_EXTJARFILES += \
 	$(EXTENSIONDIR)$/commons-logging-1.1.1.jar
 .ENDIF
 
Index: apache-commons/java/lang/makefile.mk
===================================================================
--- apache-commons/java/lang/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ apache-commons/java/lang/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -43,7 +43,7 @@
 
 .INCLUDE : antsettings.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(ENABLE_MEDIAWIKI)" == "YES"
 # --- Files --------------------------------------------------------
 
 TARFILE_NAME=commons-lang-2.3-src
@@ -71,7 +71,7 @@
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(ENABLE_MEDIAWIKI)" == "YES"
 .INCLUDE : tg_ext.mk
 .ENDIF
 
Index: apache-commons/java/logging/makefile.mk
===================================================================
--- apache-commons/java/logging/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ apache-commons/java/logging/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -43,7 +43,7 @@
 
 .INCLUDE : antsettings.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && ( "$(ENABLE_MEDIAWIKI)" == "YES" || "$(ENABLE_REPORTBUILDER)" == "YES" )
 # --- Files --------------------------------------------------------
 
 TARFILE_NAME=commons-logging-1.1.1-src
@@ -75,7 +75,7 @@
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && ( "$(ENABLE_MEDIAWIKI)" == "YES" || "$(ENABLE_REPORTBUILDER)" == "YES" )
 .INCLUDE : tg_ext.mk
 .ENDIF
 
Index: apache-commons/java/codec/makefile.mk
===================================================================
--- apache-commons/java/codec/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ apache-commons/java/codec/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -43,7 +43,7 @@
 
 .INCLUDE : antsettings.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(ENABLE_MEDIAWIKI)" == "YES"
 # --- Files --------------------------------------------------------
 
 TARFILE_NAME=commons-codec-1.3-src
@@ -71,7 +71,7 @@
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(ENABLE_MEDIAWIKI)" == "YES"
 .INCLUDE : tg_ext.mk
 .ENDIF
 
Index: apache-commons/java/httpclient/makefile.mk
===================================================================
--- apache-commons/java/httpclient/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ apache-commons/java/httpclient/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -43,7 +43,7 @@
 
 .INCLUDE : antsettings.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(ENABLE_MEDIAWIKI)" == "YES"
 # --- Files --------------------------------------------------------
 
 TARFILE_NAME=commons-httpclient-3.1-src
@@ -74,7 +74,7 @@
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(ENABLE_MEDIAWIKI)" == "YES"
 .INCLUDE : tg_ext.mk
 .ENDIF
 
Index: jfreereport/java/libbase/makefile.mk
===================================================================
--- jfreereport/java/libbase/makefile.mk	(.../tags/OOO310_m5)	(Revision 269706)
+++ jfreereport/java/libbase/makefile.mk	(.../cws/rptfix06)	(Revision 269706)
@@ -66,13 +66,22 @@
 .IF "$(SOLAR_JAVA)" != ""
 .INCLUDE : tg_ext.mk
 
-$(PACKAGE_DIR)$/$(CONFIGURE_FLAG_FILE) : $(CLASSDIR)$/commons-logging-1.1.1.jar
 ALLTAR : $(CLASSDIR)$/$(TARGET)$(VERSION).jar 
 
-$(CLASSDIR)$/commons-logging-1.1.1.jar : 
-	$(COPY) $(SOLARBINDIR)$/commons-logging-1.1.1.jar $(CLASSDIR)$/commons-logging-1.1.1.jar
-	
-$(CLASSDIR)$/$(TARGET)$(VERSION).jar : $(CLASSDIR)$/commons-logging-1.1.1.jar $(PACKAGE_DIR)$/$(INSTALL_FLAG_FILE)
+# XCLASSPATH/CLASSPATH does not work and we only can give lib once. But
+# the build.xmls fortunately take *.jar out of lib so we can copy our
+# commons-logging.jar here - yes, even in the system-apache commons case.
+# Sucks.
+$(PACKAGE_DIR)$/$(CONFIGURE_FLAG_FILE) : $(CLASSDIR)$/commons-logging.jar
+
+$(CLASSDIR)$/commons-logging.jar : 
+.IF "$(SYSTEM_APACHE_COMMONS)" != "YES"
+	$(COPY) $(SOLARBINDIR)$/commons-logging-1.1.1.jar $(CLASSDIR)$/commons-logging.jar
+.ELSE
+	$(COPY) $(COMMONS_LOGGING_JAR) $(CLASSDIR)$/commons-logging.jar
+.ENDIF
+
+$(CLASSDIR)$/$(TARGET)$(VERSION).jar : $(CLASSDIR)$/commons-logging.jar $(PACKAGE_DIR)$/$(INSTALL_FLAG_FILE)
 	$(COPY) $(PACKAGE_DIR)$/$(TARFILE_ROOTDIR)$/build$/lib$/$(TARGET).jar $(CLASSDIR)$/$(TARGET)$(VERSION).jar
 .ENDIF
 
