Index: offapi/com/sun/star/awt/MaxChildrenException.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/MaxChildrenException.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/MaxChildrenException.idl	6 Mar 2008 12:41:08 -0000	1.2
+++ offapi/com/sun/star/awt/MaxChildrenException.idl	7 Apr 2008 19:41:46 -0000	1.2.26.1
@@ -15,8 +15,10 @@ module com {  module sun {  module star 
 
 	@see     XLayoutContainer
 	@see     com::sun::star::awt::XLayoutContainer::addChild
+
+    @since OOo 3.0
  */
-published exception MaxChildrenException: com::sun::star::uno::Exception
+exception MaxChildrenException: com::sun::star::uno::Exception
 { 
 }; 
  
Index: offapi/com/sun/star/awt/XDialog2.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/XDialog2.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/XDialog2.idl	6 Mar 2008 12:41:21 -0000	1.2
+++ offapi/com/sun/star/awt/XDialog2.idl	7 Apr 2008 19:41:46 -0000	1.2.26.1
@@ -13,9 +13,14 @@ module com {  module sun {  module star 
 //============================================================================= 
  
 /** Makes it possible to end a dialog.
+
+    @since OOo 3.0
  */
-published interface XDialog2: com::sun::star::awt::XDialog
+interface XDialog2: com::sun::star::awt::XDialog
 { 
+    /** hides the dialog and then causes <member>XDialog::execute</member>
+        to return with the given result value.
+     */
     void endDialog ( [in] long Result );
 }; 
  
Index: offapi/com/sun/star/awt/XLayoutContainer.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/XLayoutContainer.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/XLayoutContainer.idl	6 Mar 2008 12:41:33 -0000	1.2
+++ offapi/com/sun/star/awt/XLayoutContainer.idl	7 Apr 2008 19:41:46 -0000	1.2.26.1
@@ -26,45 +26,59 @@ module com {  module sun {  module star 
 
 //============================================================================= 
 
-published interface XLayoutUnit;
+interface XLayoutUnit;
 
 /** specifies the layout constraints for a surrounding container.
+
+    @since OOo 3.0
  */
-published interface XLayoutContainer: com::sun::star::container::XChild
+interface XLayoutContainer: com::sun::star::container::XChild
 {
-	/* Add/remove a child. Some containers provide an interface for
+	/** Adds a child. Some containers provide an interface for
 	   adding children which you should use. */
     void addChild( [in] com::sun::star::awt::XLayoutConstrains Child )
 		raises( com::sun::star::awt::MaxChildrenException ); 
+
+	/** Remove a child. Some containers provide an interface for
+	   adding children which you should use.
+    */
     void removeChild( [in] com::sun::star::awt::XLayoutConstrains Child );
+	
+    /** Returns a sequence of the children of this container. */
 	sequence< com::sun::star::awt::XLayoutConstrains > getChildren();
 
-	/* Read/modify some child layout properties. XPropertySet provides a
+	/** Read/modify some child layout properties. XPropertySet provides a
 	   rich API that allows, e.g., for introspection.
 	   Should rarely be a need to use; containers shall provide an interface
 	   for setting properties more confortably. */
     com::sun::star::beans::XPropertySet getChildProperties(
 		[in] com::sun::star::awt::XLayoutConstrains Child );
 
-	/* Set at initilization, this object should be notified when a containers state
+	/** Set at initilization, this object should be notified when a containers state
 	   changes, that may affect its size, to queue a layout re-calculation. */
 	void setLayoutUnit( [in] com::sun::star::awt::XLayoutUnit Unit );
 
 	// TEMP: no need for this...
 	com::sun::star::awt::XLayoutUnit getLayoutUnit();
 
-	/* To be used by the parent for the purpose of layout management. 
+	/** To be used by the parent for the purpose of layout management. 
 	   For widgets, use XWindow::setPosSize() */
     void allocateArea( [in] com::sun::star::awt::Rectangle NewArea );
 
-	/* Used by the layouting unit to evaluate size damage, and force a
+	/** Used by the layouting unit to evaluate size damage, and force a
 	   re-allocation. */
 	com::sun::star::awt::Size getRequestedSize();
+
+	/** Used by the layouting unit to evaluate size damage, and force a
+	   re-allocation. */
 	com::sun::star::awt::Rectangle getAllocatedArea();
 
-	/* For flow layouting (e.g. wrap-able labels). TODO: We might want to
+	/** For flow layouting (e.g. wrap-able labels). TODO: We might want to
 	   re-design this approach -- check XLayoutConstrains::calcAdjustedSize(). */
 	long getHeightForWidth( [in] long Width );
+
+	/** For flow layouting (e.g. wrap-able labels). TODO: We might want to
+	   re-design this approach -- check XLayoutConstrains::calcAdjustedSize(). */
 	boolean hasHeightForWidth();
 };
  
Index: offapi/com/sun/star/awt/XLayoutFlow.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/XLayoutFlow.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/XLayoutFlow.idl	6 Mar 2008 12:41:45 -0000	1.2
+++ offapi/com/sun/star/awt/XLayoutFlow.idl	7 Apr 2008 19:41:47 -0000	1.2.26.1
@@ -10,15 +10,18 @@ module com {  module sun {  module star 
 //============================================================================= 
  
 /** Enables height-for-width layout negociations, which allows for label wrapping
- *  and flow containers. Can be implemented by either a container or an ordinary widget;
- *  whether its parent will honor it is another story, so keep implementing
- *  getMinimumSize().
+    and flow containers. Can be implemented by either a container or an ordinary widget;
+    whether its parent will honor it is another story, so keep implementing
+    getMinimumSize().
+
+    @since OOo 3.0
  */
-published interface XLayoutFlow
+interface XLayoutFlow
 {
+    /** returns the prefered high this layout element would need for the given width. */
 	long getHeightForWidth( [in] long Width );
 
-	/* Allow the container/widget to toggle the functionality. */
+	/** Allow the container/widget to toggle the functionality. */
 	boolean hasHeightForWidth();
 };
  
Index: offapi/com/sun/star/awt/XLayoutFlowContainer.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/XLayoutFlowContainer.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/XLayoutFlowContainer.idl	6 Mar 2008 12:41:59 -0000	1.2
+++ offapi/com/sun/star/awt/XLayoutFlowContainer.idl	7 Apr 2008 19:41:47 -0000	1.2.26.1
@@ -11,9 +11,11 @@ module com {  module sun {  module star 
  
 /** Responsible to evaluate size damages and force a re-calculation. Containers
     should let it know of state changes that may affects their size.
- *  All children of a top-level window should share the same object.
+    All children of a top-level window should share the same object.
+
+    @since OOo 3.0
  */
-published interface XLayoutContainer
+interface XLayoutContainer
 {
 	void queueResize( [in] com::sun::star::awt::XLayoutContainer Container );
 };
Index: offapi/com/sun/star/awt/XLayoutRoot.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/XLayoutRoot.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/XLayoutRoot.idl	6 Mar 2008 12:42:13 -0000	1.2
+++ offapi/com/sun/star/awt/XLayoutRoot.idl	7 Apr 2008 19:41:47 -0000	1.2.26.1
@@ -18,8 +18,10 @@ module com {  module sun {  module star 
 //============================================================================= 
  
 /** specifies an interface for a top-level layoutable widget
+
+   @since OOo 3.0
  */
-published interface XLayoutRoot: com::sun::star::container::XNameAccess
+interface XLayoutRoot: com::sun::star::container::XNameAccess
 {
     com::sun::star::awt::XLayoutContainer getLayoutContainer();
 }; 
Index: offapi/com/sun/star/awt/XLayoutUnit.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/XLayoutUnit.idl,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -p -u -u -p -b -w -B -r1.2 -r1.2.26.1
--- offapi/com/sun/star/awt/XLayoutUnit.idl	6 Mar 2008 12:42:24 -0000	1.2
+++ offapi/com/sun/star/awt/XLayoutUnit.idl	7 Apr 2008 19:41:47 -0000	1.2.26.1
@@ -11,13 +11,15 @@ module com {  module sun {  module star 
  
 //============================================================================= 
 
-published interface XLayoutContainer;
+interface XLayoutContainer;
 
 /** Responsible to evaluate size damages and force a re-calculation. Containers
     should let it know of state changes that may affects their size.
- *  All children of a top-level window should share the same object.
+    All children of a top-level window should share the same object.
+    
+    @since OOo 3.0
  */
-published interface XLayoutUnit
+interface XLayoutUnit
 {
 	void queueResize( [in] com::sun::star::awt::XLayoutContainer Container );
 };
Index: offapi/com/sun/star/awt/makefile.mk
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/makefile.mk,v
retrieving revision 1.49
retrieving revision 1.47.20.2
diff -p -u -u -p -b -w -B -r1.49 -r1.47.20.2
--- offapi/com/sun/star/awt/makefile.mk	25 Jun 2008 14:50:06 -0000	1.49
+++ offapi/com/sun/star/awt/makefile.mk	30 Jul 2008 12:45:52 -0000	1.47.20.2
@@ -317,10 +317,7 @@ IDLFILES=\
 	XWindowListener.idl\
 	XWindowListener2.idl\
 	XWindowPeer.idl\
-	XMenuExtended.idl
-
-.IF "$(ENABLE_LAYOUT)" == "TRUE"
-IDLFILES+=\
+	XMenuExtended.idl\
 	MaxChildrenException.idl\
 	XDialog2.idl\
 	XLayoutContainer.idl\
@@ -328,7 +325,6 @@ IDLFILES+=\
 	XLayoutFlowContainer.idl\
 	XLayoutRoot.idl\
 	XLayoutUnit.idl
-.ENDIF # ENABLE_LAYOUT == TRUE
 
 # ------------------------------------------------------------------
 
