Index: unotools/source/ucbhelper/ucblockbytes.cxx
===================================================================
RCS file: /cvs/util/unotools/source/ucbhelper/ucblockbytes.cxx,v
retrieving revision 1.60
retrieving revision 1.58.22.2
diff -u -p -u -p -B -r1.60 -r1.58.22.2
--- unotools/source/ucbhelper/ucblockbytes.cxx	22 Apr 2008 15:10:28 -0000	1.60
+++ unotools/source/ucbhelper/ucblockbytes.cxx	2 Jun 2008 14:00:17 -0000	1.58.22.2
@@ -964,6 +964,10 @@ static sal_Bool UCBOpenContentSync(
     if( ! aScheme.equalsIgnoreAsciiCaseAscii("http")                &&
 		! aScheme.equalsIgnoreAsciiCaseAscii("https")                &&
         ! aScheme.equalsIgnoreAsciiCaseAscii("vnd.sun.star.webdav") &&
+        ! aScheme.equalsIgnoreAsciiCaseAscii("dav") &&
+        ! aScheme.equalsIgnoreAsciiCaseAscii("davs") &&
+        ! aScheme.equalsIgnoreAsciiCaseAscii("webdav") &&
+        ! aScheme.equalsIgnoreAsciiCaseAscii("webdavs") &&
         ! aScheme.equalsIgnoreAsciiCaseAscii("ftp"))
 		return _UCBOpenContentSync(
 			xLockBytes,xContent,rArg,xSink,xInteract,xProgress,xHandler);
@@ -1541,7 +1545,13 @@ ErrCode UcbLockBytes::Flush() const
     Reference <XOutputStream > xOutputStream = getOutputStream_Impl();
     if ( !xOutputStream.is() )
         return ERRCODE_IO_CANTWRITE;
-    xOutputStream->flush();
+    try {
+        xOutputStream->flush();
+    }
+    catch (...)
+    {
+        return ERRCODE_IO_CANTWRITE;
+    }
     return ERRCODE_NONE;
 }
 
