Index: sc/source/ui/docshell/docsh.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/docshell/docsh.cxx,v
retrieving revision 1.100
retrieving revision 1.98.32.2
diff -u -p -u -p -B -r1.100 -r1.98.32.2
--- sc/source/ui/docshell/docsh.cxx	15 Apr 2008 14:12:23 -0000	1.100
+++ sc/source/ui/docshell/docsh.cxx	2 Jun 2008 13:52:36 -0000	1.98.32.2
@@ -1003,7 +1003,7 @@ BOOL __EXPORT ScDocShell::ConvertFrom( S
 	//	Alle Filter brauchen die komplette Datei am Stueck (nicht asynchron),
 	//	darum vorher per CreateFileStream dafuer sorgen, dass die komplette
 	//	Datei uebertragen wird.
-	rMedium.GetPhysicalName();	//! CreateFileStream direkt rufen, wenn verfuegbar
+	rMedium.GetPhysicalName( sal_False );	//! CreateFileStream direkt rufen, wenn verfuegbar
 
     SFX_ITEMSET_ARG( rMedium.GetItemSet(), pUpdateDocItem, SfxUInt16Item, SID_UPDATEDOCMODE, sal_False);
     nCanUpdate = pUpdateDocItem ? pUpdateDocItem->GetValue() : com::sun::star::document::UpdateDocMode::NO_UPDATE;
