Index: source/plugin/base/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/base/makefile.mk,v
retrieving revision 1.14.64.1
retrieving revision 1.14.64.1.2.1
diff -u -r1.14.64.1 -r1.14.64.1.2.1
--- extensions/source/plugin/base/makefile.mk	19 Nov 2008 15:01:51 -0000	1.14.64.1
+++ extensions/source/plugin/base/makefile.mk	9 Dec 2008 16:08:38 -0000	1.14.64.1.2.1
@@ -55,11 +55,6 @@
 CFLAGSCXX+=$(OBJCXXFLAGS)
 .ENDIF  # "$(GUIBASE)"=="aqua"
 
-.IF "$(GUIBASE)" == "unx" && "$(ENABLE_GTK)" == "TRUE"
-PKGCONFIG_MODULES=gtk+-2.0
-.INCLUDE : pkg_config.mk
-.ENDIF
-
 SLOFILES=		\
 				$(SLO)$/plctrl.obj		\
 				$(SLO)$/service.obj		\
Index: source/plugin/inc/plugin/unx/plugcon.hxx
===================================================================
RCS file: /cvs/util/extensions/source/plugin/inc/plugin/unx/plugcon.hxx,v
retrieving revision 1.13.64.1
retrieving revision 1.13.64.1.2.1
diff -u -r1.13.64.1 -r1.13.64.1.2.1
--- extensions/source/plugin/inc/plugin/unx/plugcon.hxx	19 Nov 2008 15:02:20 -0000	1.13.64.1
+++ extensions/source/plugin/inc/plugin/unx/plugcon.hxx	9 Dec 2008 16:08:38 -0000	1.13.64.1.2.1
@@ -77,6 +77,27 @@
 #define MOZ_X11
 #endif
 
+//http://qa.openoffice.org/issues/show_bug.cgi?id=82545
+//https://bugzilla.mozilla.org/show_bug.cgi?id=241262
+#ifdef UNIX
+#  ifndef _UINT32
+#    if defined(__alpha) || defined(__LP64__)
+       typedef unsigned int uint32;
+#    else  /* __alpha */
+       typedef unsigned long uint32;
+#    endif
+#    define _UINT32
+#  endif
+#  ifndef _INT32
+#    if defined(__alpha) || defined(__LP64__)
+       typedef int int32;
+#    else  /* __alpha */
+       typedef long int32;
+#    endif
+#    define _INT32
+#  endif
+#endif
+
 #ifndef _NPAPI_H_
 extern "C" {
 #include <npsdk/npupp.h>
Index: source/plugin/unx/makefile.mk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/unx/makefile.mk,v
retrieving revision 1.15.64.1
retrieving revision 1.15.64.1.2.1
diff -u -r1.15.64.1 -r1.15.64.1.2.1
--- extensions/source/plugin/unx/makefile.mk	19 Nov 2008 15:02:32 -0000	1.15.64.1
+++ extensions/source/plugin/unx/makefile.mk	9 Dec 2008 16:08:38 -0000	1.15.64.1.2.1
@@ -58,12 +58,6 @@
 CDEFS+=-DDISABLE_XAW
 .ENDIF
 
-.IF "$(ENABLE_GTK)" == "TRUE"
-CDEFS+=-DENABLE_GTK
-PKGCONFIG_MODULES=gtk+-2.0 gthread-2.0
-.INCLUDE : pkg_config.mk
-.ENDIF
-
 SLOFILES=\
 	$(SLO)$/nppapi.obj		\
 	$(SLO)$/sysplug.obj		\
Index: source/plugin/unx/npnapi.cxx
===================================================================
RCS file: /cvs/util/extensions/source/plugin/unx/npnapi.cxx,v
retrieving revision 1.11.102.2
retrieving revision 1.11.102.2.2.1
diff -u -r1.11.102.2 -r1.11.102.2.2.1
--- extensions/source/plugin/unx/npnapi.cxx	3 Dec 2008 14:22:52 -0000	1.11.102.2
+++ extensions/source/plugin/unx/npnapi.cxx	9 Dec 2008 16:08:38 -0000	1.11.102.2.2.1
@@ -51,13 +51,14 @@
     if( nInstance == PluginConnector::UnknownNPPID )
         return NPERR_GENERIC_ERROR;
     
+	UINT32 nFileID = pConnector->GetStreamID( stream );
 	MediatorMessage* pMes=
 		pConnector->
 		Transact( eNPN_DestroyStream,
 				  &nInstance, sizeof( nInstance ),
-				  pConnector->GetStreamID( stream ), sizeof( int ),
+				  &nFileID, sizeof( nFileID ),
 				  POST_STRING( stream->url ),
-				  reason, sizeof( reason ),
+				  &reason, sizeof( reason ),
 				  NULL );
 
 	if( ! pMes )
Index: source/plugin/unx/nppapi.cxx
===================================================================
RCS file: /cvs/util/extensions/source/plugin/unx/nppapi.cxx,v
retrieving revision 1.7.102.1
retrieving revision 1.7.102.1.2.1
diff -u -r1.7.102.1 -r1.7.102.1.2.1
--- extensions/source/plugin/unx/nppapi.cxx	19 Nov 2008 15:03:06 -0000	1.7.102.1
+++ extensions/source/plugin/unx/nppapi.cxx	9 Dec 2008 16:08:39 -0000	1.7.102.1.2.1
@@ -397,7 +397,7 @@
 							   saved ? saved->len : 0 ) );
 
 	char *pArgnBuf, *pArgvBuf;
-	int nArgnLen = 0, nArgvLen = 0;
+	size_t nArgnLen = 0, nArgvLen = 0;
 	int i;
 	for( i = 0; i < argc; i++ )
 	{
@@ -425,7 +425,7 @@
 					  &argc, sizeof( argc ),
 					  pArgnBuf, nArgnLen,
 					  pArgvBuf, nArgvLen,
-					  saved->buf, saved->len,
+					  saved->buf, static_cast<size_t>(saved->len),
 					  NULL );
 	else
 		pMes = 
@@ -435,7 +435,7 @@
 					  &argc, sizeof( argc ),
 					  pArgnBuf, nArgnLen,
 					  pArgvBuf, nArgvLen,
-					  "0000", 4,
+					  "0000", size_t(4),
 					  NULL );
 	delete [] pArgnBuf;
 	delete [] pArgvBuf;
@@ -542,7 +542,7 @@
 				  POST_INSTANCE(),
 				  &nFileID, sizeof( nFileID ),
 				  &offset, sizeof( offset ),
-				  buffer, len,
+				  buffer, static_cast<size_t>(len),
 				  NULL );
 	if( ! pMes )
 		return 0;
Index: source/plugin/util/makefile.pmk
===================================================================
RCS file: /cvs/util/extensions/source/plugin/util/makefile.pmk,v
retrieving revision 1.8
retrieving revision 1.8.100.1
diff -u -r1.8 -r1.8.100.1
--- extensions/source/plugin/util/makefile.pmk	11 Apr 2008 10:30:16 -0000	1.8
+++ extensions/source/plugin/util/makefile.pmk	9 Dec 2008 16:08:39 -0000	1.8.100.1
@@ -34,6 +34,14 @@
 
 .IF "$(SYSTEM_MOZILLA)" == "YES"
 CFLAGS+=-DSYSTEM_MOZILLA
-PKGCONFIG_MODULES=$(MOZ_FLAVOUR)-plugin
+PKGCONFIG_MODULES+=$(MOZ_FLAVOUR)-plugin
+.ENDIF
+
+.IF "$(GUIBASE)" == "unx" && "$(ENABLE_GTK)" == "TRUE"
+CDEFS+=-DENABLE_GTK
+PKGCONFIG_MODULES+=gtk+-2.0 gthread-2.0
+.ENDIF
+
+.IF "$(PKGCONFIG_MODULES)" != ""
 .INCLUDE : pkg_config.mk
 .ENDIF
