Index: source/tools/surface.cxx
===================================================================
RCS file: /cvs/gsl/canvas/source/tools/surface.cxx,v
retrieving revision 1.6
retrieving revision 1.6.16.1
diff -u -p -r1.6 -r1.6.16.1
--- canvas/source/tools/surface.cxx	9 Jun 2008 15:03:59 -0000	1.6
+++ canvas/source/tools/surface.cxx	7 Oct 2008 15:02:20 -0000	1.6.16.1
@@ -262,7 +262,7 @@ namespace canvas
 
         ::basegfx::B2IPoint aDestOffset;
         if( mpFragment )
-            mpFragment->getPos();
+            aDestOffset = mpFragment->getPos();
 
 		// convert size to normalized device coordinates
 		const ::basegfx::B2DRectangle& rUV( 
