Index: source/table/cell.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/table/cell.cxx,v
retrieving revision 1.4
retrieving revision 1.4.80.1
diff -u -p -r1.4 -r1.4.80.1
--- svx/source/table/cell.cxx	6 Jun 2008 12:35:07 -0000	1.4
+++ svx/source/table/cell.cxx	21 Oct 2008 13:46:45 -0000	1.4.80.1
@@ -682,6 +682,7 @@ SdrTextHorzAdjust Cell::GetTextHorizonta
 void Cell::SetOutlinerParaObject( OutlinerParaObject* pTextObject )
 {
 	SdrText::SetOutlinerParaObject( pTextObject );
+    maSelection.nStartPara = 0xffff;
 }
 
 // -----------------------------------------------------------------------------
@@ -1686,6 +1687,7 @@ Reference< XTextRange > SAL_CALL Cell::g
 
 OUString SAL_CALL Cell::getString(  ) throw (RuntimeException)
 {
+    maSelection.nStartPara = 0xffff;
 	return SvxUnoTextBase::getString();
 }
 
Index: source/table/tablecolumn.hxx
===================================================================
RCS file: /cvs/graphics/svx/source/table/tablecolumn.hxx,v
retrieving revision 1.3
retrieving revision 1.3.264.1
diff -u -p -r1.3 -r1.3.264.1
--- svx/source/table/tablecolumn.hxx	11 Apr 2008 03:26:22 -0000	1.3
+++ svx/source/table/tablecolumn.hxx	21 Oct 2008 11:30:58 -0000	1.3.264.1
@@ -51,6 +51,7 @@ typedef ::cppu::ImplInheritanceHelper2< 
 class TableColumn : public TableColumnBase
 {
 	friend class TableColumnUndo;
+    friend class TableModel;
 public:
 	TableColumn( const TableModelRef& xTableModel, sal_Int32 nColumn );
 	virtual ~TableColumn();
Index: source/table/tablemodel.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/table/tablemodel.cxx,v
retrieving revision 1.4
retrieving revision 1.4.264.1
diff -u -p -r1.4 -r1.4.264.1
--- svx/source/table/tablemodel.cxx	11 Apr 2008 03:31:09 -0000	1.4
+++ svx/source/table/tablemodel.cxx	21 Oct 2008 11:30:58 -0000	1.4.264.1
@@ -248,6 +248,7 @@ void TableModel::UndoInsertRows( sal_Int
 
 	// remove the rows
 	remove_range<RowVector,RowVector::iterator>( maRows, nIndex, nCount );
+    updateRows();
 	setModified(sal_True);
 }
 
@@ -264,6 +265,7 @@ void TableModel::UndoRemoveRows( sal_Int
 	for( sal_Int32 nOffset = 0; nOffset < nCount; ++nOffset )
 		maRows[nIndex+nOffset] = aRows[nOffset];
 
+    updateRows();
 	setModified(sal_True);
 }
 
@@ -278,6 +280,8 @@ void TableModel::UndoInsertColumns( sal_
 	sal_Int32 nRows = getRowCountImpl();
 	while( nRows-- )
 		maRows[nRows]->removeColumns( nIndex, nCount );
+
+    updateColumns();
 	setModified(sal_True);
 }
 
@@ -301,6 +305,8 @@ void TableModel::UndoRemoveColumns( sal_
 	sal_Int32 nRows = getRowCountImpl();
 	for( sal_Int32 nRow = 0; nRow < nRows; ++nRow )
 		maRows[nRow]->insertColumns( nIndex, nCount, &aIter );
+
+    updateColumns();
 	setModified(sal_True);
 }
 
@@ -802,6 +808,7 @@ void TableModel::insertColumns( sal_Int3
 		{
 			DBG_ERROR("sdr::table::TableModel::insertColumns(), exception caught!");
 		}
+        updateColumns();
 		setModified(sal_True);
 	}
 }
@@ -905,6 +912,7 @@ void TableModel::removeColumns( sal_Int3
 			DBG_ERROR("sdr::table::TableModel::removeColumns(), exception caught!");
 		}
 
+        updateColumns();
 		setModified(sal_True);
 	}
 }
@@ -975,6 +983,7 @@ void TableModel::insertRows( sal_Int32 n
 		{
 			DBG_ERROR("sdr::table::TableModel::insertRows(), exception caught!");
 		}
+        updateRows();
 		setModified(sal_True);
 	}
 }
@@ -1063,6 +1072,7 @@ void TableModel::removeRows( sal_Int32 n
 			DBG_ERROR("sdr::table::TableModel::removeRows(), exception caught!");
 		}
 
+        updateRows();
 		setModified(sal_True);
 	}
 }
@@ -1178,4 +1188,28 @@ void TableModel::optimize()
 
 // -----------------------------------------------------------------------------
 
+void TableModel::updateRows()
+{
+    sal_Int32 nRow = 0;
+    RowVector::iterator iter = maRows.begin();
+    while( iter != maRows.end() )
+    {
+        (*iter++)->mnRow = nRow++;
+    }
+}
+
+// -----------------------------------------------------------------------------
+
+void TableModel::updateColumns()
+{
+    sal_Int32 nColumn = 0;
+    ColumnVector::iterator iter = maColumns.begin();
+    while( iter != maColumns.end() )
+    {
+        (*iter++)->mnColumn = nColumn++;
+    }
+}
+
+// -----------------------------------------------------------------------------
+
 } }
Index: source/table/tablemodel.hxx
===================================================================
RCS file: /cvs/graphics/svx/source/table/tablemodel.hxx,v
retrieving revision 1.3
retrieving revision 1.3.264.1
diff -u -p -r1.3 -r1.3.264.1
--- svx/source/table/tablemodel.hxx	11 Apr 2008 03:31:36 -0000	1.3
+++ svx/source/table/tablemodel.hxx	21 Oct 2008 11:30:58 -0000	1.3.264.1
@@ -176,6 +176,9 @@ private:
 	TableRowRef getRow( sal_Int32 nRow ) const throw (::com::sun::star::lang::IndexOutOfBoundsException);
 	TableColumnRef getColumn( sal_Int32 nColumn ) const throw (::com::sun::star::lang::IndexOutOfBoundsException);
 
+    void updateRows();
+    void updateColumns();
+
 	RowVector		maRows;
 	ColumnVector	maColumns;
 
Index: source/table/tablerow.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/table/tablerow.cxx,v
retrieving revision 1.3
retrieving revision 1.3.264.1
diff -u -p -r1.3 -r1.3.264.1
--- svx/source/table/tablerow.cxx	11 Apr 2008 03:32:04 -0000	1.3
+++ svx/source/table/tablerow.cxx	21 Oct 2008 11:30:58 -0000	1.3.264.1
@@ -129,36 +129,15 @@ void TableRow::insertColumns( sal_Int32 
 	throwIfDisposed();
 	if( nCount )
 	{
-		if( nIndex < static_cast< sal_Int32 >( maCells.size() ) )
-		{
-			CellVector::iterator aIter( maCells.begin() );
-			while( nIndex-- )
-				aIter++;
-			while( nCount-- )
-			{
-				CellRef xCell;
-				if( pIter )
-					xCell = (*(*pIter)++);
-				else
-					xCell = mxTableModel->createCell();
-				
-				maCells.insert( aIter, xCell );
-			}
-		}
+		if( nIndex >= static_cast< sal_Int32 >( maCells.size() ) )
+			nIndex = static_cast< sal_Int32 >( maCells.size() );
+		if ( pIter )
+			maCells.insert( maCells.begin() + nIndex, *pIter, (*pIter) + nCount );
 		else
 		{
-			nIndex = static_cast< sal_Int32 >( maCells.size() );
-			maCells.resize( nIndex + nCount );
-			while(nCount--)
-			{
-				CellRef xCell;
-				if( pIter )
-					xCell = (*(*pIter)++);
-				else
-					xCell = mxTableModel->createCell();
-			
-				maCells[nIndex++] = xCell;
-			}
+			maCells.reserve( maCells.size() + nCount );
+			for ( sal_Int32 i = 0; i < nCount; i++ )
+				maCells.insert( maCells.begin() + nIndex + i, mxTableModel->createCell() );
 		}
 	}
 }
Index: source/xml/xmlxtimp.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/xml/xmlxtimp.cxx,v
retrieving revision 1.17
retrieving revision 1.17.264.1
diff -u -p -r1.17 -r1.17.264.1
--- svx/source/xml/xmlxtimp.cxx	11 Apr 2008 04:12:08 -0000	1.17
+++ svx/source/xml/xmlxtimp.cxx	20 Oct 2008 15:02:51 -0000	1.17.264.1
@@ -260,8 +260,15 @@ void SvxXMLTableImportContext::importMar
 	(void)nPrfx;
 	(void)rLocalName;
 
-	XMLMarkerStyleImport aMarkerStyle( GetImport() );
-	aMarkerStyle.importXML( xAttrList, rAny, rName );
+    try
+    {
+	    XMLMarkerStyleImport aMarkerStyle( GetImport() );
+	    aMarkerStyle.importXML( xAttrList, rAny, rName );
+    }
+    catch( Exception& )
+    {
+        DBG_ERROR("SvxXMLTableImportContext::importMarker(), exception caught!");
+    }
 }
 
 void SvxXMLTableImportContext::importDash( USHORT nPrfx, const OUString& rLocalName, const uno::Reference< XAttributeList >& xAttrList, Any& rAny, OUString& rName )
@@ -269,8 +276,15 @@ void SvxXMLTableImportContext::importDas
 	(void)nPrfx;
 	(void)rLocalName;
 
-	XMLDashStyleImport aDashStyle( GetImport() );
-	aDashStyle.importXML( xAttrList, rAny, rName );
+    try
+    {
+    	XMLDashStyleImport aDashStyle( GetImport() );
+	    aDashStyle.importXML( xAttrList, rAny, rName );
+    }
+    catch( Exception& )
+    {
+        DBG_ERROR("SvxXMLTableImportContext::importDash(), exception caught!");
+    }
 }
 
 void SvxXMLTableImportContext::importHatch( USHORT nPrfx, const OUString& rLocalName, const uno::Reference< XAttributeList >& xAttrList, Any& rAny, OUString& rName )
@@ -278,8 +292,15 @@ void SvxXMLTableImportContext::importHat
 	(void)nPrfx;
 	(void)rLocalName;
 
-	XMLHatchStyleImport aHatchStyle( GetImport() );
-	aHatchStyle.importXML( xAttrList, rAny, rName );
+    try
+    {
+    	XMLHatchStyleImport aHatchStyle( GetImport() );
+	    aHatchStyle.importXML( xAttrList, rAny, rName );
+    }
+    catch( Exception& )
+    {
+        DBG_ERROR("SvxXMLTableImportContext::importHatch(), exception caught!");
+    }
 }
 
 void SvxXMLTableImportContext::importGradient( USHORT nPrfx, const OUString& rLocalName, const uno::Reference< XAttributeList >& xAttrList, Any& rAny, OUString& rName )
@@ -287,8 +308,15 @@ void SvxXMLTableImportContext::importGra
 	(void)nPrfx;
 	(void)rLocalName;
 
-	XMLGradientStyleImport aGradientStyle( GetImport() );
-	aGradientStyle.importXML( xAttrList, rAny, rName );
+    try
+    {
+    	XMLGradientStyleImport aGradientStyle( GetImport() );
+	    aGradientStyle.importXML( xAttrList, rAny, rName );
+    }
+    catch( Exception& )
+    {
+        DBG_ERROR("SvxXMLTableImportContext::importGradient(), exception caught!");
+    }
 }
 
 void SvxXMLTableImportContext::importBitmap( USHORT nPrfx, const OUString& rLocalName, const uno::Reference< XAttributeList >& xAttrList, Any& rAny, OUString& rName )
@@ -296,8 +324,15 @@ void SvxXMLTableImportContext::importBit
 	(void)nPrfx;
 	(void)rLocalName;
 
-	XMLImageStyle aImageStyle;
-	aImageStyle.importXML( xAttrList, rAny, rName, GetImport() );
+	try
+    {
+        XMLImageStyle aImageStyle;
+	    aImageStyle.importXML( xAttrList, rAny, rName, GetImport() );
+    }
+    catch( Exception& )
+    {
+        DBG_ERROR("SvxXMLTableImportContext::importBitmap(), exception caught!");
+    }
 }
 
 ///////////////////////////////////////////////////////////////////////
