--- sw/source/filter/docx/docx-docxattributeoutput.cxx	(revision 271190)
+++ sw/source/filter/docx/docx-docxattributeoutput.cxx	(working copy)
@@ -3198,12 +3198,12 @@
     // Get the columns attributes
     FastAttributeList *pColsAttrList = m_pSerializer->createAttrList();
 
+    pColsAttrList->add( FSNS( XML_w, XML_num ), 
+            OString::valueOf( sal_Int32( nCols ) ). getStr( ) );
+
     const char* pEquals = "false";
     if ( bEven )
     {
-        pColsAttrList->add( FSNS( XML_w, XML_num ), 
-                OString::valueOf( sal_Int32( nCols ) ). getStr( ) );
-
         USHORT nWidth = rCol.GetGutterWidth( true );
         pColsAttrList->add( FSNS( XML_w, XML_space ),
                OString::valueOf( sal_Int32( nWidth ) ).getStr( ) );
@@ -3233,7 +3233,7 @@
             if ( n + 1 != nCols )
             {
                 USHORT nSpacing = rColumns[n]->GetRight( ) + rColumns[n + 1]->GetLeft( );
-                pColAttrList->add( FSNS( XML_w, XML_w ),
+                pColAttrList->add( FSNS( XML_w, XML_space ),
                     OString::valueOf( sal_Int32( nSpacing ) ).getStr( ) );
             }
 
