Index: source/dialog/filedlghelper.cxx
===================================================================
--- sfx2/source/dialog/filedlghelper.cxx	(.../tags/DEV300_m36/sfx2)	(revision 264730)
+++ sfx2/source/dialog/filedlghelper.cxx	(.../cws/scsheetprotection02/sfx2)	(revision 264730)
@@ -556,6 +556,14 @@
 {
 	sal_Bool operator() ( const SfxFilter* _pFilter )
 	{
+        if (!_pFilter)
+            return false;
+
+        if (_pFilter->GetFilterName().EqualsAscii("MS Excel 97"))
+            // temporary hack to enable password protection for Excel 97.  Is
+            // there a better way to enable password protection of a filter?
+            return true;
+
         return  _pFilter && _pFilter->IsOwnFormat()
 			&&	_pFilter->UsesStorage()
 			&&	( SOFFICE_FILEFORMAT_60 <= _pFilter->GetVersion() );
@@ -1544,6 +1552,9 @@
 		if( !rpSet )
 			rpSet = new SfxAllItemSet( SFX_APP()->GetPool() );
 
+		// set the filter
+		getRealFilter( rFilter );
+
 		// check, wether or not we have to display a password box
 		if ( mbHasPassword && mbIsPwdEnabled && xCtrlAccess.is() )
 		{
@@ -1553,9 +1564,16 @@
 				sal_Bool bPassWord = sal_False;
 				if ( ( aValue >>= bPassWord ) && bPassWord )
 				{
-					// ask for the password
+					// ask for a password
 					SfxPasswordDialog aPasswordDlg( NULL );
 					aPasswordDlg.ShowExtras( SHOWEXTRAS_CONFIRM );
+                    if ( rFilter.EqualsAscii("MS Excel 97") )
+                    {
+                        // Excel allows 1-15 length password.
+                        aPasswordDlg.SetMinLen(1);
+                        aPasswordDlg.SetMaxLen(15);
+                    }
+
 					short nRet = aPasswordDlg.Execute();
 					if ( RET_OK == nRet )
 					{
@@ -1622,9 +1640,6 @@
 			catch( IllegalArgumentException ){}
 		}
 
-		// set the filter
-		getRealFilter( rFilter );
-
 		// fill the rpURLList
 		implGetAndCacheFiles(mxFileDlg, rpURLList, getCurentSfxFilter());
 		if ( rpURLList == NULL )
Index: source/dialog/passwd.cxx
===================================================================
--- sfx2/source/dialog/passwd.cxx	(.../tags/DEV300_m36/sfx2)	(revision 264730)
+++ sfx2/source/dialog/passwd.cxx	(.../cws/scsheetprotection02/sfx2)	(revision 264730)
@@ -112,6 +112,15 @@
 
 // -----------------------------------------------------------------------
 
+void SfxPasswordDialog::SetMaxLen( USHORT nLen )
+{
+    maPasswordED.SetMaxTextLen( nLen );
+    maConfirmED.SetMaxTextLen( nLen );
+	EditModifyHdl( NULL );
+}
+
+// -----------------------------------------------------------------------
+
 short SfxPasswordDialog::Execute()
 {
 	if ( mnExtras < SHOWEXTRAS_ALL )
Index: inc/sfx2/passwd.hxx
===================================================================
--- sfx2/inc/sfx2/passwd.hxx	(.../tags/DEV300_m36/sfx2)	(revision 264730)
+++ sfx2/inc/sfx2/passwd.hxx	(.../cws/scsheetprotection02/sfx2)	(revision 264730)
@@ -81,6 +81,7 @@
 	String			GetConfirm() const { return maConfirmED.GetText(); }
 
 	void 			SetMinLen( USHORT Len );
+    void            SetMaxLen( USHORT Len );
 	void			SetEditHelpId( ULONG nId ) { maPasswordED.SetHelpId( nId ); }
 	void			ShowExtras( USHORT nExtras ) { mnExtras = nExtras; }
 
