Index: cwscheckapi
===================================================================
RCS file: /cvs/tools/solenv/bin/cwscheckapi,v
retrieving revision 1.3.36.1
diff -u -r1.3.36.1 cwscheckapi
--- solenv/bin/cwscheckapi     14 Aug 2008 12:52:20 -0000      1.3.36.1
+++ solenv/bincwscheckapi      20 Oct 2008 11:53:11 -0000
@@ -163,7 +163,7 @@
 PARAM="$PARAM -cmd '$OFFICEBIN  -nofirststartwizard -no-oosplash -norestore -nocrashreport -nolockcheck -accept=pipe,name=$USER;urp;'"
 PARAM="$PARAM -cs pipe,name=$USER"
 PARAM="$PARAM -NoOffice true"
-PARAM="$PARAM -OOO_EnvSet setsolar"
+PARAM="$PARAM -OOO_EnvSet $SRC_ROOT/*Env.Set"
 PARAM="$PARAM -SRC_ROOT $SRC_ROOT"
 PARAM="$PARAM -COMP_ENV $INPATH"
 PARAM="$PARAM -Shell /usr/bin/tcsh"
