--- instsetoo_native/util/openoffice.lst-old	2008-08-06 00:03:46.000000000 +0200
+++ instsetoo_native/util/openoffice.lst	2008-08-06 00:04:01.000000000 +0200
@@ -23,7 +23,7 @@
 			LIBRARYVERSION 9.0.0
             REGISTRATION_HOST https://registration2.services.openoffice.org/RegistrationWeb
 			REGISTRATIONURL http://survey.services.openoffice.org/user/index.php
-			PROGRESSBARCOLOR 4,68,155
+			PROGRESSBARCOLOR 215,7,81
 			PROGRESSSIZE 412,7
 			PROGRESSPOSITION 14,230
 			PROGRESSFRAMECOLOR 58,159,232
