--- solenv/inc/unxlngi6.mk~	2006-09-21 14:19:05.398388000 +0200
+++ solenv/inc/unxlngi6.mk	2006-09-21 14:48:34.028388000 +0200
@@ -118,7 +118,7 @@
 CFLAGSDBGUTIL=
 # Compiler flags for enabling optimizations
 .IF "$(PRODUCT)"!=""
-CFLAGSOPT=-Os -fno-strict-aliasing		# optimizing for products
+CFLAGSOPT=-O2 -fno-strict-aliasing		# optimizing for products
 .ELSE 	# "$(PRODUCT)"!=""
 CFLAGSOPT=   							# no optimizing for non products
 .ENDIF	# "$(PRODUCT)"!=""
--- solenv/inc/unxlngs.mk~	2006-09-21 14:19:04.108388000 +0200
+++ solenv/inc/unxlngs.mk	2006-09-21 14:49:04.728388000 +0200
@@ -110,7 +110,7 @@
 CFLAGSDBGUTIL=
 # Compiler flags for enabling optimizations
 .IF "$(PRODUCT)"!=""
-CFLAGSOPT=-Os -fno-strict-aliasing		# optimizing for products
+CFLAGSOPT=-O2 -fno-strict-aliasing		# optimizing for products
 .ELSE 	# "$(PRODUCT)"!=""
 CFLAGSOPT=   							# no optimizing for non products
 .ENDIF	# "$(PRODUCT)"!=""
