--- tools/source/debug/debug.cxx	2006-02-28 11:31:43.000000000 +0100
+++ tools/source/debug/debug.cxx	2006-03-20 18:12:36.000000000 +0100
@@ -178,7 +178,11 @@ struct DebugData
         aDbgData.bOverwrite = TRUE;
         aDbgData.nTraceOut = DBG_OUT_NULL;
         aDbgData.nWarningOut = DBG_OUT_NULL;
+#ifdef UNX
+        aDbgData.nErrorOut = DBG_OUT_SHELL;
+#else
         aDbgData.nErrorOut = DBG_OUT_MSGBOX;
+#endif
         aDbgData.bMemInit = 0x77;
         aDbgData.bMemBound = 0x55;
         aDbgData.bMemFree = 0x33;
