diff --git a/desktop/source/migration/wizard.cxx b/desktop/source/migration/wizard.cxx
index c7c7bbd..f488c56 100644
--- desktop/source/migration/wizard.cxx
+++ desktop/source/migration/wizard.cxx
@@ -121,6 +121,8 @@ FirstStartWizard::FirstStartWizard(Window* pParent)
     ,m_bAutomaticUpdChk(sal_True)
     ,m_aLicensePath( rLicensePath )
 {
+    return;
+
     // ---
     // FreeResource();
 //	enableState(STATE_USER, sal_False);
@@ -383,7 +385,11 @@ void FirstStartWizard::overrideCheck(sal_Bool bOverride)
 
 short FirstStartWizard::Execute()
 {
-    return svt::RoadmapWizard::Execute();
+    disableWizard();
+    if ( Migration::checkMigration() )
+        Migration::doMigration();
+
+    return sal_True;
 }
 
 static OUString _makeDateTimeString (const DateTime& aDateTime, sal_Bool bUTC = sal_False)
diff --git a/scp2/source/ooo/profileitem_ooo.scp b/scp2/source/ooo/profileitem_ooo.scp
index 9773b42..ce25861 100644
--- scp2/source/ooo/common_brand.scp
+++ scp2/source/ooo/common_brand.scp
@@ -682,19 +682,14 @@ ProfileItem gid_Profileitem_Soffice_ProgressPosition
     Value = "-1,-1";
 End
 
-#ifdef WNT
 ProfileItem gid_Brand_Profileitem_Soffice_Hideeula
     ProfileID = gid_Brand_Profile_Soffice_Ini;
     ModuleID = gid_Module_Root_Brand;
     Section = "Bootstrap";
     Order = 7;
     Key = "HideEula";
-    Value = "[HIDEEULA]";
-    Inifiletablekey = "HideEula";
-    Inifiletableaction = "1";
-    Styles = (INIFILETABLE);
+    Value = "1";
 End
-#endif
 
 ProfileItem gid_Profileitem_Uno_Uno_Shared_Packages
     ProfileID = gid_Profile_Uno_Ini;
