--- officecfg/registry/data/org/openoffice/Office/Common.xcu.old	2009-04-02 11:02:04.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2009-04-06 16:41:54.000000000 +0000
@@ -546,8 +546,8 @@
     </node>
   </node>
   <node oor:name="Misc">
-    <prop oor:name="UseSystemFileDialog"  install:module="UseOOoFileDialogs">
-      <value>false</value>
+    <prop oor:name="UseSystemFileDialog"  install:module="UseOOoFileDialogs"> <!-- safe -->
+      <value>true</value>
     </prop>
     <prop oor:name="UseSystemPrintDialog"  install:module="macosx">
       <value>true</value>
