diff --git sfx2/source/appl/appreg.cxx sfx2/source/appl/appreg.cxx
index 721900b..9c833cc 100644
--- sfx2/source/appl/appreg.cxx
+++ sfx2/source/appl/appreg.cxx
@@ -70,7 +70,7 @@ void SfxApplication::Registrations_Impl()
 	SfxRecordingFloatWrapper_Impl::RegisterChildWindow();
 	SfxNavigatorWrapper::RegisterChildWindow( FALSE, NULL, SFX_CHILDWIN_NEVERHIDE );
 	SfxPartChildWnd_Impl::RegisterChildWindow();
-	SfxTemplateDialogWrapper::RegisterChildWindow(TRUE);
+	SfxTemplateDialogWrapper::RegisterChildWindow();
 	SfxDockingWrapper::RegisterChildWindow();
 
 	// Controller
