Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.84
diff -u -p -u -r1.84 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	24 Jul 2003 13:12:08 -0000	1.84
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	5 Jan 2004 14:57:29 -0000
@@ -3795,60 +3795,60 @@
 						<info>
 							<desc>Specifies if convertible MathType OLE objects are converted into Math objects.</desc>
 						</info>
-						<value>false</value>
+ 						<value>true</value>
 					</prop>
 					<prop oor:name="WinWordToWriter" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if convertible WinWord OLE objects are converted into Writer objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 					<prop oor:name="ExcelToCalc" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if convertible Excel OLE objects are converted into Calc objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 					<prop oor:name="PowerPointToImpress" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if convertible PowerPoint OLE objects are converted into Impress objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 					<prop oor:name="ImportWWFieldsAsEnhancedFields" oor:type="xs:boolean">
 						<info>
 							<desc>Flag to control use of enhanced fields.</desc>
 						</info>
 						<value>true</value>
 					</prop>
 				</group>
 				<group oor:name="Export">
 					<info>
 						<desc>Specifies default settings of Microsoft export filters.</desc>
 					</info>
 					<prop oor:name="MathToMathType" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if embedded Math objects are converted into MathType objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 					<prop oor:name="WriterToWinWord" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if embedded Writer objects are converted into WinWord objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 					<prop oor:name="CalcToExcel" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if embedded Calc objects are converted into Excel objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value>
 					</prop>
 					<prop oor:name="ImpressToPowerPoint" oor:type="xs:boolean">
 						<info>
 							<desc>Specifies if embedded Impress objects are converted into PowerPoint objects.</desc>
 						</info>
-						<value>false</value>
+						<value>true</value> 
 					</prop>
                     <prop oor:name="EnablePowerPointPreview" oor:type="xs:boolean">
                         <info>
