diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=desktop.vpj desktop.clean/source/app/app.cxx desktop/source/app/app.cxx
--- desktop.clean/source/app/app.cxx	2008-05-16 15:26:26.000000000 -0400
+++ desktop/source/app/app.cxx	2008-05-16 15:26:38.000000000 -0400
@@ -1752,8 +1752,10 @@ void Desktop::Main()
                 }
             }
         }
+#if 0
         else if ( RegistrationPage::hasReminderDateCome() )
             RegistrationPage::executeSingleMode();
+#endif
 
         RTL_LOGFILE_CONTEXT_TRACE( aLog, "} FirstStartWizard" );
 
