diff --git desktop/source/app/app.cxx desktop/source/app/app.cxx
index b1645eb..d3a3fc8 100644
--- desktop/source/app/app.cxx
+++ desktop/source/app/app.cxx
@@ -47,6 +47,7 @@
 #include "desktopcontext.hxx"
 #include "exithelper.hxx"
 #include "../migration/pages.hxx"
+#include "../migration/migration.hxx"
 
 #include <svtools/javacontext.hxx>
 #include <com/sun/star/frame/XSessionManagerListener.hpp>
@@ -1394,6 +1395,7 @@ void Desktop::Main()
         tools::InitTestToolLib();
         RTL_LOGFILE_CONTEXT_TRACE( aLog, "} tools::InitTestToolLib" );
 
+#if 0 // ooo-build doesn't use First Start Wizard at all.
         // First Start Wizard allowed ?
         if ( ! pCmdLineArgs->IsNoFirstStartWizard())
         {
@@ -1425,6 +1427,7 @@ void Desktop::Main()
             
             RTL_LOGFILE_CONTEXT_TRACE( aLog, "} FirstStartWizard" );
         }
+#endif
 
 		// keep a language options instance...
 		pLanguageOptions.reset( new SvtLanguageOptions(sal_True));
