diff --git sfx2/inc/sfx2/dinfdlg.hxx sfx2/inc/sfx2/dinfdlg.hxx
index 3e1c252..113d2db 100644
--- sfx2/inc/sfx2/dinfdlg.hxx
+++ sfx2/inc/sfx2/dinfdlg.hxx
@@ -173,6 +173,7 @@ class SfxDocumentPage : public SfxTabPage
 private:
     FixedImage                  aBmp1;
     Edit                        aNameED;
+    PushButton                  aChangePassBtn;
 
     FixedLine                   aLine1FL;
     FixedText                   aTypeFT;
@@ -212,7 +213,9 @@ private:
 
     DECL_LINK(          DeleteHdl, PushButton * );
     DECL_LINK(          SignatureHdl, PushButton * );
+    DECL_LINK( ChangePassHdl, PushButton * );
     void                ImplUpdateSignatures();
+    void                ImplCheckPasswordState();
 
 protected:
 	SfxDocumentPage( Window* pParent, const SfxItemSet& );
diff --git sfx2/source/dialog/dinfdlg.cxx sfx2/source/dialog/dinfdlg.cxx
index 075f9ee..811c068 100644
--- sfx2/source/dialog/dinfdlg.cxx
+++ sfx2/source/dialog/dinfdlg.cxx
@@ -66,6 +66,7 @@
 #include <sfx2/frame.hxx>
 #include <sfx2/viewfrm.hxx>
 #include <sfx2/request.hxx>
+#include <sfx2/passwd.hxx>
 //#include "exptypes.hxx"
 #include "helper.hxx"
 #include <sfx2/objsh.hxx>
@@ -767,6 +768,7 @@ SfxDocumentPage::SfxDocumentPage( Window* pParent, const SfxItemSet& rItemSet )
 
     aBmp1           ( this, SfxResId( BMP_FILE_1 ) ),
     aNameED         ( this, SfxResId( ED_FILE_NAME ) ),
+    aChangePassBtn  ( this, SfxResId( BTN_CHANGE_PASS ) ),
 
     aLine1FL        ( this, SfxResId( FL_FILE_1 ) ),
     aTypeFT         ( this, SfxResId( FT_FILE_TYP ) ),
@@ -808,8 +810,10 @@ SfxDocumentPage::SfxDocumentPage( Window* pParent, const SfxItemSet& rItemSet )
 	FreeResource();
 
     ImplUpdateSignatures();
+    ImplCheckPasswordState();
 	aDeleteBtn.SetClickHdl( LINK( this, SfxDocumentPage, DeleteHdl ) );
 	aSignatureBtn.SetClickHdl( LINK( this, SfxDocumentPage, SignatureHdl ) );
+    aChangePassBtn.SetClickHdl( LINK( this, SfxDocumentPage, ChangePassHdl ) );
 
     // if the button text is too wide, then broaden it
     const long nOffset = 12;
@@ -878,6 +882,32 @@ IMPL_LINK( SfxDocumentPage, SignatureHdl, PushButton*, EMPTYARG )
     return 0;
 }
 
+IMPL_LINK( SfxDocumentPage, ChangePassHdl, PushButton*, EMPTYARG )
+{
+	SfxObjectShell*	pShell = SfxObjectShell::Current();
+    do
+    {
+        if (!pShell)
+            break;
+    
+        SfxItemSet* pMedSet = pShell->GetMedium()->GetItemSet();
+        if (!pMedSet)
+            break;
+
+        ::std::auto_ptr<SfxPasswordDialog> pDlg(new SfxPasswordDialog(this));
+        pDlg->SetMinLen(1);
+        pDlg->ShowExtras(SHOWEXTRAS_CONFIRM);
+        if (pDlg->Execute() != RET_OK)
+            break;
+
+        String aNewPass = pDlg->GetPassword();
+        pMedSet->Put( SfxStringItem(SID_PASSWORD, aNewPass) );
+        pShell->SetModified(true);
+    }
+    while (false);
+    return 0;
+}
+
 void SfxDocumentPage::ImplUpdateSignatures()
 {
 	SfxObjectShell*	pDoc = SfxObjectShell::Current();
@@ -913,6 +943,34 @@ void SfxDocumentPage::ImplUpdateSignatures()
 	}
 }
 
+void SfxDocumentPage::ImplCheckPasswordState()
+{
+	SfxObjectShell*	pShell = SfxObjectShell::Current();
+    do
+    {
+        if (!pShell)
+            break;
+
+        SfxItemSet* pMedSet = pShell->GetMedium()->GetItemSet();
+        if (!pMedSet)
+            break;
+
+        const SfxPoolItem* pItem;
+        if (!pMedSet->GetItemState(SID_PASSWORD, true, &pItem))
+            break;
+
+        const SfxStringItem* pStrItem = dynamic_cast<const SfxStringItem*>(pItem);
+        if (!pStrItem)
+            break;
+
+        String aPass = pStrItem->GetValue();
+        aChangePassBtn.Enable();
+        return;
+    }
+    while (false);
+    aChangePassBtn.Disable();
+}
+
 //------------------------------------------------------------------------
 
 SfxTabPage* SfxDocumentPage::Create( Window* pParent, const SfxItemSet& rItemSet )
diff --git sfx2/source/dialog/dinfdlg.hrc sfx2/source/dialog/dinfdlg.hrc
index 1110be8..543c6ad 100644
--- sfx2/source/dialog/dinfdlg.hrc
+++ sfx2/source/dialog/dinfdlg.hrc
@@ -126,6 +126,7 @@
 #define FT_SIGNED_VAL		77
 #define STR_MULTSIGNED		78
 #define BTN_SIGNATURE		79
+#define BTN_CHANGE_PASS     80
 
 #endif
 
diff --git sfx2/source/dialog/dinfdlg.src sfx2/source/dialog/dinfdlg.src
index 5e08e3b..1502c55 100644
--- sfx2/source/dialog/dinfdlg.src
+++ sfx2/source/dialog/dinfdlg.src
@@ -123,10 +123,17 @@ TabPage TP_DOCINFODOC
 	};
 	Edit ED_FILE_NAME
 	{
-		Pos = MAP_APPFONT ( 54 , 12 ) ;
-        Size = MAP_APPFONT ( 200 , RSC_CD_TEXTBOX_HEIGHT ) ;
+		Pos = MAP_APPFONT ( 36 , 12 ) ;
+        Size = MAP_APPFONT ( 140 , RSC_CD_TEXTBOX_HEIGHT ) ;
 		Border = TRUE ;
 	};
+    PushButton BTN_CHANGE_PASS
+    {
+        Pos = MAP_APPFONT ( 189 , 11 ) ;
+        Size = MAP_APPFONT ( 65 , RSC_CD_PUSHBUTTON_HEIGHT ) ;
+        Hide = FALSE;
+        Text [ en-US ] = "Change ~Password...";
+    };
 	FixedLine FL_FILE_1
 	{
 		Pos = MAP_APPFONT ( 6 , 31 ) ;
