--- sal/osl/unx/socket.c	2006-07-19 12:39:22.000000000 +0300
+++ sal/osl/unx/socket.c	2007-07-10 14:52:47.000000000 +0300
@@ -878,6 +878,8 @@
 
 static sal_Char* _osl_getFullQualifiedDomainName (const sal_Char *pHostName)
 {
+    return strdup(pHostName);
+#if 0
 #	define DOMAINNAME_LENGTH 512
 	sal_uInt32 			nLengthOfHostName;
 	static sal_uInt32 	nLengthOfDomainName = 0;
@@ -1067,6 +1069,7 @@
 	/* always return a hostname looked up as carefully as possible
 	 * this string must be freed by the caller */
 	return pFullQualifiedName;
+ #endif
 }
 
 /*****************************************************************************/
