Index: makefile.mk
===================================================================
--- extensions/source/plugin/base/makefile.mk	(Revision 264632)
+++ extensions/source/plugin/base/makefile.mk	(Arbeitskopie)
@@ -37,10 +37,19 @@
 
 .INCLUDE :  ..$/util$/makefile.pmk
 
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+INCPRE+=-I$(MOZ_INC)$/plugin
+.ELSE
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/plugin
+.ENDIF
 .IF "$(SOLAR_JAVA)" != ""
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+INCPRE+=-I$(MOZ_INC)$/java
+INCPRE+=-I$(MOZ_INC)$/nspr
+.ELSE
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/java
 INCPRE+=-I$(SOLARINCDIR)$/mozilla$/nspr
+.ENDIF
 CDEFS+=-DOJI
 .ENDIF
 
