--- filter/source/xslt/export/xhtml/header.xsl	2007-08-06 07:26:49.453125000 +0200
+++ filter/source/xslt/export/xhtml/header.xsl	2007-08-06 07:27:22.984375000 +0200
@@ -212,9 +212,9 @@
 				<xsl:when test="$globalData/meta-file/*/office:meta/dc:title">
 					<xsl:value-of select="$globalData/meta-file/*/office:meta/dc:title" />
 				</xsl:when>
-				<!--<xsl:otherwise>
-					<xsl:text>- no title specified</xsl:text>
-				</xsl:otherwise>-->
+				<xsl:otherwise>
+					<xsl:text> </xsl:text>
+				</xsl:otherwise>
 			</xsl:choose>
 		</xsl:element>
 
