diff --git a/svx/source/dialog/textattr.cxx b/svx/source/dialog/textattr.cxx
index b4b1114..dd8f405 100644
--- svx/source/dialog/textattr.cxx
+++ svx/source/dialog/textattr.cxx
@@ -460,7 +460,7 @@ BOOL SvxTextAttrPage::FillItemSet( SfxItemSet& rAttrs)
             default: ; //prevent warning
 				DBG_ERROR( "svx::SvxTextAttrPage::FillItemSet(), unhandled state!" );
 			case STATE_NOCHECK: eFTS = SDRTEXTFIT_NONE; break;
-			case STATE_CHECK: eFTS = SDRTEXTFIT_PROPORTIONAL; break;
+			case STATE_CHECK: eFTS = SDRTEXTFIT_AUTOFIT; break;
 		}
 		rAttrs.Put( SdrTextFitToSizeTypeItem( eFTS ) );
 	}
