--- vcl/unx/gtk/a11y/atkbridge.cxx.old	2009-04-02 12:44:27.000000000 +0200
+++ vcl/unx/gtk/a11y/atkbridge.cxx	2009-04-03 20:19:54.000000000 +0200
@@ -44,7 +44,9 @@ bool InitAtkBridge(void)
     const char* pVersion = atk_get_toolkit_version();
     if( ! pVersion )
     {
-        g_warning( "unable to get gail version number" );
+        // InitAtkBridge is always called; we should not bother users with a warning
+        // if they have not enabled the accessibilty support on their system
+        // g_warning( "unable to get gail version number" );
         return false;
     }
     
