Index: formula/source/core/api/FormulaCompiler.cxx
===================================================================
--- formula/source/core/api/FormulaCompiler.cxx	(revision 266592)
+++ formula/source/core/api/FormulaCompiler.cxx	(working copy)
@@ -698,7 +698,7 @@
         if ( FormulaGrammar::GRAM_ENGLISH != _eGrammar )
             fillFromAddInCollectionUpperName( _xMap);
         else
-            fillFromAddInCollectionEnglishName( mxSymbolsEnglish);
+            fillFromAddInCollectionEnglishName( _xMap );
     }
 }
 // -----------------------------------------------------------------------------
