--- svx/source/dialog/makefile.mk.old	2009-04-02 11:03:28.000000000 +0000
+++ svx/source/dialog/makefile.mk	2009-04-06 16:41:56.000000000 +0000
@@ -43,6 +43,14 @@ ENABLE_EXCEPTIONS=TRUE
 CFLAGS+= -DENABLE_LAYOUT=1 -I../$(PRJ)/layout/inc -I../$(PRJ)/layout/$(INPATH)/inc
 .ENDIF # ENABLE_LAYOUT == TRUE
 
+.IF "$(ENABLE_GTK)" != ""
+CFLAGS+=-DENABLE_GTK
+.ENDIF
+
+.IF "$(ENABLE_KDE)" != ""
+CFLAGS+=-DENABLE_KDE
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SRS1NAME=dialogs
--- svx/source/dialog/optgdlg.cxx.old	2009-04-02 11:03:28.000000000 +0000
+++ svx/source/dialog/optgdlg.cxx	2009-04-06 16:41:56.000000000 +0000
@@ -136,7 +136,7 @@ int OfaMiscTabPage::DeactivatePage( SfxI
 	return LEAVE_PAGE;
 }
 
-#	ifdef ENABLE_GTK
+#if defined( ENABLE_GTK ) || defined( ENABLE_KDE )
 namespace
 {
 		::rtl::OUString impl_SystemFileOpenServiceName()
@@ -212,17 +212,15 @@ OfaMiscTabPage::OfaMiscTabPage(Window* p
 
     //system fileopen only available in Windows and with gtk vclplug based
     //picker and on MacOSX (aqua version)
-#if !defined( WNT ) && !defined( ENABLE_GTK ) && !(defined(MACOSX) && defined(QUARTZ))
+#if !defined( WNT ) && !defined( ENABLE_GTK ) && !(defined(MACOSX) && defined(QUARTZ)) && !defined( ENABLE_KDE )
     aFileDlgFL.Hide();
     aFileDlgCB.Hide();
-#else
-#	ifdef ENABLE_GTK
+#elif defined( ENABLE_GTK ) || defined( ENABLE_KDE )
 	if (!lcl_HasSystemFilePicker())
 	{
         aFileDlgFL.Hide();
         aFileDlgCB.Hide();
 	}
-#	endif
 #endif
 
     #if ! defined(QUARTZ)
