--- sc/inc/optutil.hxx.old	2000-11-02 20:08:15.000000000 +0100
+++ sc/inc/optutil.hxx	2005-08-02 19:32:54.000000000 +0200
@@ -70,6 +70,7 @@
 #include <unotools/configitem.hxx>
 #include <tools/link.hxx>
 
+#include "scdllapi.h"
 
 class ScOptionsUtil
 {
@@ -80,7 +84,7 @@
 
 //	ConfigItem for classes that use items from several sub trees
 
-class ScLinkConfigItem : public utl::ConfigItem
+class SC_DLLPUBLIC ScLinkConfigItem : public utl::ConfigItem
 {
 	Link	aCommitLink;
 
