Index: bin/modules/installer/servicesfile.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/servicesfile.pm,v
retrieving revision 1.23
diff -u -r1.23 servicesfile.pm
--- solenv/bin/modules/installer/servicesfile.pm	10 Jul 2006 18:51:00 -0000	1.23
+++ solenv/bin/modules/installer/servicesfile.pm	23 Feb 2007 16:12:39 -0000
@@ -258,7 +258,7 @@
 					$counter++;
 				}
 			
-				if ((( $counter > 0 ) && ( $counter%$installer::globals::javamaxservices == 0 )) || (( $counter > 0 ) && ( $i == $#{$javacomponents} )))	# limiting to $installer::globals::maxservices files
+				if ( $counter > 0 )
 				{
 					$filestring =~ s/\;\s*$//;
 					chdir($onesourcepath);
Index: bin/modules/installer/globals.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/globals.pm,v
retrieving revision 1.62.2.3
diff -u -r1.62.2.3 globals.pm
--- solenv/bin/modules/installer/globals.pm	24 Jan 2007 16:34:05 -0000	1.62.2.3
+++ solenv/bin/modules/installer/globals.pm	23 Feb 2007 11:02:01 -0000
@@ -256,7 +256,7 @@
 	@solarispatchfiles = (".diPatch", "patchinfo");
 	@environmentvariables = ( "SOLARVERSION", "GUI", "WORK_STAMP", "OUTPATH", "LOCAL_OUT", "LOCAL_COMMON_OUT" );
 	@packagelistitems = ("module", "solarispackagename", "packagename", "copyright", "vendor", "description" );
-	@regcompjars = ( "unoil.jar", "java_uno.jar", "ridl.jar", "jurt.jar", "juh.jar", "xmerge.jar", "commonwizards.jar" );
+	@regcompjars = ( "unoil.jar", "java_uno.jar", "ridl.jar", "jurt.jar", "juh.jar", "commonwizards.jar" );
 	@regcompregisterlibs = ( "javavm.uno", "javaloader.uno", "uriproc.uno" );
 	@languagepackfeature =();
 	@featurecollector =();
