--- xmlhelp/source/com/sun/star/help/makefile.mk.ark	2006-03-02 17:29:27.000000000 +0000
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2006-03-02 17:31:26.000000000 +0000
@@ -115,6 +115,12 @@
 	$(JAVABINCOMPILER) --main=com.sun.star.help.HelpLinker -o $(BIN)$/HelpLinker \
 		-O2 -Wl,-Bsymbolic -findirect-dispatch -fjni -fno-assert -Wno-deprecated \
 		$(JAVACPS) $(CLASSPATH) $(JAVAFILES) $(SOLARLIB) -L$(LB) \
+		-Djavax.xml.datatype.DatatypeFactory=org.apache.xerces.jaxp.datatype.DatatypeFactoryImpl \
+		-Djavax.xml.parsers.DocumentBuilderFactory=org.apache.xerces.jaxp.DocumentBuilderFactoryImpl \
+		-Djavax.xml.parsers.SAXParserFactory=org.apache.xerces.jaxp.SAXParserFactoryImpl \
+		-Djavax.xml.validation.SchemaFactory=org.apache.xerces.jaxp.validation.XMLSchemaFactory \
+		-Dorg.w3c.dom.DOMImplementationSourceList=org.apache.xerces.dom.DOMXSImplementationSourceImpl \
+		-Dorg.xml.sax.driver=org.apache.xerces.parsers.SAXParser \
 		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar -lxt-xmlsearch.jar
 
 .IF "$(SYSTEM_DB)"=="YES"
