--- configure.in.old	2009-04-06 16:41:44.000000000 +0000
+++ configure.in	2009-04-06 16:41:46.000000000 +0000
@@ -5249,6 +5249,7 @@ dnl Gnome VFS check
 dnl ===================================================================
 
 ENABLE_GNOMEVFS=""
+GNOMEVFS_HAS_LOCKING=""
 AC_MSG_CHECKING([whether to enable GNOME VFS support])
 if test "$_os" != "WINNT" -a "$_os" != "Darwin" -a "$enable_gnome_vfs" = "yes"; then
     ENABLE_GNOMEVFS="TRUE"
@@ -5257,10 +5258,21 @@ if test "$_os" != "WINNT" -a "$_os" != "
     if test "$ENABLE_GCONF" != "TRUE"; then
         PKG_CHECK_MODULES( GCONF, gconf-2.0 )
     fi
+
+    AC_LANG_SAVE
+    AC_LANG_CPLUSPLUS
+    save_CXXFLAGS=$CXXFLAGS
+    CXXFLAGS="$CXXFLAGS $GNOMEVFS_CFLAGS"
+    AC_CHECK_DECL(GNOME_VFS_OPEN_LOCKED,
+                  GNOMEVFS_HAS_LOCKING="YES",,
+                  [#include <libgnomevfs/gnome-vfs-handle.h>])
+    CXXFLAGS=$save_CXXFLAGS
+    AC_LANG_RESTORE
 else
     AC_MSG_RESULT([no])
 fi
 AC_SUBST(ENABLE_GNOMEVFS)
+AC_SUBST(GNOMEVFS_HAS_LOCKING)
 
 dnl ===================================================================
 dnl Check whether the gtk 2.0 libraries are available.
--- set_soenv.in.old	2009-04-06 16:41:41.000000000 +0000
+++ set_soenv.in	2009-04-06 16:41:46.000000000 +0000
@@ -1753,6 +1753,7 @@ ToFile( "SERVLETAPI_JAR",    "@SERVLETAP
 ToFile( "ENABLE_DBUS",       "@ENABLE_DBUS@",      "e" );
 ToFile( "ENABLE_GCONF",      "@ENABLE_GCONF@",     "e" );
 ToFile( "ENABLE_GNOMEVFS",   "@ENABLE_GNOMEVFS@",  "e" );
+ToFile( "GNOMEVFS_HAS_LOCKING","@GNOMEVFS_HAS_LOCKING@",  "e" );
 ToFile( "ENABLE_GIO",        "@ENABLE_GIO@",       "e" );
 ToFile( "ENABLE_KDE",        "@ENABLE_KDE@",       "e" );
 ToFile( "KDE_CFLAGS",        "@KDE_CFLAGS@",       "e" );
