--- instsetoo_native/util/openoffice.lst.orig	2009-01-01 15:59:14.000000000 +0100
+++ instsetoo_native/util/openoffice.lst	2009-01-01 16:01:25.000000000 +0100
@@ -25,8 +25,8 @@
             REGISTRATION_HOST https://registration2.services.openoffice.org/RegistrationWeb
 			REGISTRATIONURL http://survey.services.openoffice.org/user/index.php
 			PROGRESSBARCOLOR 4,68,155
-			PROGRESSSIZE 412,7
-			PROGRESSPOSITION 14,230
+			PROGRESSSIZE 376,7
+			PROGRESSPOSITION 50,30
 			PROGRESSFRAMECOLOR 58,159,232
 			REGISTRYLAYERNAME Layers
             SERVICEPACK 1
