diff -rup goodies/source/filter.vcl/eps.orig/eps.cxx goodies/source/filter.vcl/eps/eps.cxx
--- goodies/source/filter.vcl/eps.orig/eps.cxx	2007-11-26 12:08:53.000000000 +0100
+++ goodies/source/filter.vcl/eps/eps.cxx	2007-11-26 17:38:34.000000000 +0100
@@ -317,7 +317,11 @@ BOOL PSWriter::WritePS( const Graphic& r
 	// default values for the dialog options
 	mnLevel = 2;
 	mbGrayScale = FALSE;
+#ifdef UNX // don't compress by default on unix as ghostscript is unable to read LZW compressed eps
+	mbCompression = FALSE;
+#else
 	mbCompression = TRUE;
+#endif
 	mnTextMode = 0;			// default0 : export glyph outlines
 
 	// try to get the dialog selection
@@ -335,12 +339,20 @@ BOOL PSWriter::WritePS( const Graphic& r
 	        String aVersionStr( RTL_CONSTASCII_USTRINGPARAM( "Version" ) );
         	String aColorStr( RTL_CONSTASCII_USTRINGPARAM( "ColorFormat" ) );   
 	        String aComprStr( RTL_CONSTASCII_USTRINGPARAM( "CompressionMode" ) );
+#ifdef UNX // don't put binary tiff preview ahead of postscript code by default on unix as ghostscript is unable to read it
+			mnPreview = pFilterConfigItem->ReadInt32( aPreviewStr, 0 );
+#else
 			mnPreview = pFilterConfigItem->ReadInt32( aPreviewStr, 1 );
+#endif
 			mnLevel = pFilterConfigItem->ReadInt32( aVersionStr, 2 );
 			if ( mnLevel != 1 )
 				mnLevel = 2;
 			mbGrayScale = pFilterConfigItem->ReadInt32( aColorStr, 1 ) == 2;
-			mbCompression = pFilterConfigItem->ReadInt32( aComprStr, 1 ) == 1;	
+#ifdef UNX // don't compress by default on unix as ghostscript is unable to read LZW compressed eps
+			mbCompression = pFilterConfigItem->ReadInt32( aComprStr, 0 ) != 0;
+#else
+			mbCompression = pFilterConfigItem->ReadInt32( aComprStr, 1 ) == 1;
+#endif
         	String sTextMode( RTL_CONSTASCII_USTRINGPARAM( "TextMode" ) );
 			mnTextMode = pFilterConfigItem->ReadInt32( sTextMode, 0 );
 			if ( mnTextMode > 2 )
