--- desktop/source/app/cmdlineargs.hxx-old	2008-05-21 00:01:55.000000000 +0200
+++ desktop/source/app/cmdlineargs.hxx	2008-05-21 00:01:45.000000000 +0200
@@ -123,6 +123,8 @@
 		// generic methods to access parameter
 		void					SetBoolParam( BoolParam eParam, sal_Bool bNewValue );
 
+		const rtl::OUString&    GetStringParam( StringParam eParam ) const;
+
 		// Access to bool parameters
 		sal_Bool				IsMinimized() const;
         sal_Bool				IsInvisible() const;
--- desktop/source/app/cmdlineargs.cxx-old	2008-05-21 00:02:22.000000000 +0200
+++ desktop/source/app/cmdlineargs.cxx	2008-05-20 23:55:48.000000000 +0200
@@ -609,6 +609,14 @@
 	m_aBoolParams[eParam] = bNewValue;
 }
 
+const rtl::OUString& CommandLineArgs::GetStringParam( StringParam eParam ) const
+{
+       osl::MutexGuard  aMutexGuard( m_aMutex );
+
+       OSL_ASSERT( ( eParam >= 0 && eParam < CMD_STRINGPARAM_COUNT ) );
+       return m_aStrParams[eParam];
+}
+
 sal_Bool CommandLineArgs::IsMinimized() const
 {
 	osl::MutexGuard  aMutexGuard( m_aMutex );
