diff --git helpcontent2/source/text/scalc/01/04060107.xhp helpcontent2/source/text/scalc/01/04060107.xhp
index 58d4d67..995b557 100644
--- helpcontent2/source/text/scalc/01/04060107.xhp
+++ helpcontent2/source/text/scalc/01/04060107.xhp
@@ -163,7 +163,7 @@
 <paragraph role="paragraph" id="par_id3154342" xml-lang="en-US" l10n="U" oldref="5">The cells in a results array are automatically protected against changes. However, you can edit or copy the array formula by selecting the entire array cell range.</paragraph>
 <paragraph role="heading" id="hd_id8834803" xml-lang="en-US" level="3" l10n="NEW">Using Inline Array Constants in Formulas</paragraph>
 <paragraph role="paragraph" id="par_id985747" xml-lang="en-US" l10n="NEW">Calc supports inline matrix/array constants in formulas. An inline array is surrounded by curly braces '{' and '}'. Elements can be each a number (including negatives), a logical constant (TRUE, FALSE), or a literal string. Non-constant expressions are not allowed. Arrays can be entered with one or more rows, and one or more columns. All rows must consist of the same number of elements, all columns must consist of the same number of elements. </paragraph>
-<paragraph role="paragraph" id="par_id936613" xml-lang="en-US" l10n="NEW">The column separator (separating elements in one row) is the ';' semicolon. The row separator is a '|' pipe symbol. The separators are not language and locale dependent. </paragraph>
+<paragraph role="paragraph" id="par_id936613" xml-lang="en-US" l10n="NEW">The column separator (separating elements in one row) and the row separator are language and locale dependent.  But in this help content, the ';' semicolon and '|' pipe symbol are used to indicate the column and row separators, respectively.  For example, in the English locale, the ',' comma is used as the column separator, while the ';' semicolon is used as the row separator.</paragraph>
 <paragraph role="paragraph" id="par_id1877498" xml-lang="en-US" l10n="NEW">Arrays can not be nested.</paragraph>
 <paragraph role="paragraph" id="par_id4262520" xml-lang="en-US" l10n="NEW">
 <emph>Examples:</emph>
