--- configure.in-old    2007-08-07 16:43:16.000000000 +0200
+++ configure.in        2007-08-07 16:43:31.000000000 +0200
@@ -4363,7 +4363,7 @@
          ], [])
-      AC_CHECK_LIB(hunspell, main, [],
+      AC_CHECK_LIB(hunspell_pic, main, [],
            [ AC_MSG_ERROR(hunspell library not found.) ], [])
-      HUNSPELL_LIBS=-lhunspell
+      HUNSPELL_LIBS=-lhunspell_pic
      fi
 else
     AC_MSG_RESULT([internal])
