--- svx/source/editeng/svxacorr.cxx.orig     27 Jun 2007 18:00:19 -0000
+++ svx/source/editeng/svxacorr.cxx     5 Mar 2008 19:54:37 -0000
@@ -394,7 +394,7 @@
             cChar == ' '  || cChar == '\'' || cChar == '\"' ||
             cChar == '*'  || cChar == '_'  ||
             cChar == '.'  || cChar == ','  || cChar == ';' ||
-            cChar == ':'  || cChar == '?';
+            cChar == ':'  || cChar == '?'  || cChar == '!';
 }

 /* -----------------19.11.98 10:15-------------------
