--- sw/source/core/txtnode/fntcache.cxx.orig	2008-07-30 10:23:51.000000000 +0800
+++ sw/source/core/txtnode/fntcache.cxx	2008-07-30 10:25:58.000000000 +0800
@@ -1235,7 +1235,8 @@ void SwFntObj::DrawText( SwDrawTextInfo 
     // For text grid refactor
     // ASIAN LINE AND CHARACTER GRID MODE START: not snap to characters
     //
-    if ( rInf.GetFrm() && rInf.SnapToGrid() && rInf.GetFont() )
+    if ( rInf.GetFrm() && rInf.SnapToGrid() && rInf.GetFont() &&
+         SW_CJK == rInf.GetFont()->GetActual() )
     {
         GETGRID( rInf.GetFrm()->FindPageFrm() )
         
@@ -2083,7 +2084,8 @@ Size SwFntObj::GetTextSize( SwDrawTextIn
     }
     
     //for textgrid refactor
-    if ( rInf.GetFrm() && nLn && rInf.SnapToGrid() && rInf.GetFont())
+    if ( rInf.GetFrm() && nLn && rInf.SnapToGrid() && rInf.GetFont() &&
+         SW_CJK == rInf.GetFont()->GetActual() )
     {
         GETGRID( rInf.GetFrm()->FindPageFrm() )
         if ( pGrid && GRID_LINES_CHARS == pGrid->GetGridType() && !pGrid->IsSnapToChars() )
@@ -2357,7 +2359,7 @@ xub_StrLen SwFntObj::GetCrsrOfst( SwDraw
     
     //for textgrid refactor
     if ( rInf.GetFrm() && rInf.GetLen() && rInf.SnapToGrid() &&
-        rInf.GetFont() )
+         rInf.GetFont() && SW_CJK == rInf.GetFont()->GetActual() )
     {
         GETGRID( rInf.GetFrm()->FindPageFrm() )
         if ( pGrid && GRID_LINES_CHARS == pGrid->GetGridType() && !pGrid->IsSnapToChars() )
@@ -2622,7 +2624,8 @@ xub_StrLen SwFont::GetTxtBreak( SwDrawTe
     }
     
     //for text grid enhancement
-    if ( rInf.GetFrm() && nLn && rInf.SnapToGrid() && rInf.GetFont() )
+    if ( rInf.GetFrm() && nLn && rInf.SnapToGrid() &&
+         rInf.GetFont() && SW_CJK == rInf.GetFont()->GetActual() )
     {
         GETGRID( rInf.GetFrm()->FindPageFrm() )
         if ( pGrid && GRID_LINES_CHARS == pGrid->GetGridType() && !pGrid->IsSnapToChars() )
--- sw/source/filter/ww8/ww8par6.cxx.orig	2008-07-30 10:26:41.000000000 +0800
+++ sw/source/filter/ww8/ww8par6.cxx	2008-07-30 10:29:32.000000000 +0800
@@ -279,9 +279,10 @@ void SwWW8ImplReader::SetDocumentGrid(Sw
 
     aGrid.SetGridType(eType);
 
-    //Seems to force this behaviour in word ?
+    // seem to not add external leading in word, or the character would run across
+	// two line in some cases.
     if (eType != GRID_NONE)
-        rDoc.set(IDocumentSettingAccess::ADD_EXT_LEADING, true);
+        rDoc.set(IDocumentSettingAccess::ADD_EXT_LEADING, false);
 
 	//force to set document as standard page mode
 	sal_Bool bSquaredMode = sal_False;
