Index: scripting/source/pyprov/pythonscript.py
===================================================================
RCS file: /cvs/framework/scripting/source/pyprov/pythonscript.py,v
retrieving revision 1.7
diff -c -r1.7 pythonscript.py
*** scripting/source/pyprov/pythonscript.py	20 May 2008 14:32:05 -0000	1.7
--- scripting/source/pyprov/pythonscript.py	13 Nov 2008 21:42:32 -0000
***************
*** 662,667 ****
--- 662,669 ----
          uri = uri.replace( "vnd.sun.star.expand:", "",1)
          uri = uno.getComponentContext().getByName(
                      "/singletons/com.sun.star.util.theMacroExpander" ).expandMacros( uri )
+     if uri.startswith( "file:" ):
+         uri = uno.absolutize("",uri)   # necessary to get rid of .. in uri
      return uri
      
  #--------------------------------------------------------------
Index: scripting/source/provider/URIHelper.cxx
===================================================================
RCS file: /cvs/framework/scripting/source/provider/URIHelper.cxx,v
retrieving revision 1.6
diff -c -r1.6 URIHelper.cxx
*** scripting/source/provider/URIHelper.cxx	11 Apr 2008 10:37:12 -0000	1.6
--- scripting/source/provider/URIHelper.cxx	13 Nov 2008 21:42:32 -0000
***************
*** 52,63 ****
      "vnd.sun.star.expand:${$BRAND_BASE_DIR/program/" SAL_CONFIGFILE( "bootstrap") "::BaseInstallation}";
  
  static const char SHARE_UNO_PACKAGES[] = "share:uno_packages";
! static const char SHARE_UNO_PACKAGES_DIR[] = 
!     "/share/uno_packages/cache";
  
  static const char USER[] = "user";
  static const char USER_URI[] =
!     "vnd.sun.star.expand:${$BRAND_BASE_DIR/program/" SAL_CONFIGFILE( "bootstrap") ":::UserInstallation}";
  
  static const char USER_UNO_PACKAGES[] = "user:uno_packages";
  static const char USER_UNO_PACKAGES_DIR[] =
--- 52,63 ----
      "vnd.sun.star.expand:${$BRAND_BASE_DIR/program/" SAL_CONFIGFILE( "bootstrap") "::BaseInstallation}";
  
  static const char SHARE_UNO_PACKAGES[] = "share:uno_packages";
! static const char SHARE_UNO_PACKAGES_URI[] = 
!     "vnd.sun.star.expand:$UNO_SHARED_PACKAGES_CACHE";
  
  static const char USER[] = "user";
  static const char USER_URI[] =
!     "vnd.sun.star.expand:${$BRAND_BASE_DIR/program/" SAL_CONFIGFILE( "bootstrap") "::UserInstallation}";
  
  static const char USER_UNO_PACKAGES[] = "user:uno_packages";
  static const char USER_UNO_PACKAGES_DIR[] =
***************
*** 164,171 ****
      else if (m_sLocation.equalsAscii(SHARE_UNO_PACKAGES))
      {
          test = OUString::createFromAscii("uno_packages");
!         uri = OUString::createFromAscii(SHARE_URI);
!         uri = uri.concat(OUString::createFromAscii(SHARE_UNO_PACKAGES_DIR));
      }
      else if (m_sLocation.indexOf(OUString::createFromAscii(TDOC_SCHEME)) == 0)
      {
--- 164,170 ----
      else if (m_sLocation.equalsAscii(SHARE_UNO_PACKAGES))
      {
          test = OUString::createFromAscii("uno_packages");
!         uri = OUString::createFromAscii(SHARE_UNO_PACKAGES_URI);
      }
      else if (m_sLocation.indexOf(OUString::createFromAscii(TDOC_SCHEME)) == 0)
      {
