--- jvmfwk/plugins/sunmajor/pluginlib/otherjre.cxx.orig	2006-08-16 12:13:05.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/otherjre.cxx	2006-08-16 12:14:30.000000000 +0200
@@ -76,6 +76,8 @@
 #ifdef MACOSX
         "/../../../JavaVM"
 #else
+        "/bin/j9vm/libjvm.so", // for IBM Java
+        "/jre/bin/j9vm/libjvm.so", // for IBM Java
         "/lib/" JFW_PLUGIN_ARCH "/client/libjvm.so", // for Blackdown PPC
         "/lib/" JFW_PLUGIN_ARCH "/server/libjvm.so", // for Blackdown AMD64
         "/lib/" JFW_PLUGIN_ARCH "/classic/libjvm.so", // for Blackdown PPC
@@ -101,6 +103,8 @@
 #else
         "/bin",
         "/jre/bin",
+        "/bin/j9vm",
+        "/jre/bin/j9vm",
         "/bin/classic",
         "/jre/bin/classic",
         "/lib/" JFW_PLUGIN_ARCH "/client",
