--- vcl/unx/kde/kdedata.cxx.old	2008-03-22 05:08:32.000000000 +0100
+++ vcl/unx/kde/kdedata.cxx	2008-03-22 05:57:03.000000000 +0100
@@ -140,18 +140,22 @@
 	XrmInitialize();
 
 	KAboutData *kAboutData = new KAboutData( "OpenOffice.org",
-			I18N_NOOP( "OpenOffice.org" ),
-			"1.1.0",
-			I18N_NOOP( "OpenOffice.org with KDE Native Widget Support." ),
+			"OpenOffice.org",
+			ki18n( "OpenOffice.org" ),
+			"3.0.0",
+			ki18n( "OpenOffice.org with KDE Native Widget Support." ),
 			KAboutData::License_LGPL,
-			"(c) 2003, 2004 Novell, Inc",
-			I18N_NOOP( "OpenOffice.org is an office suite.\n" ),
+			ki18n( "Copyright (c) 2003, 2004, 2005, 2006, 2007, 2008 Novell, Inc"),
+			ki18n( "OpenOffice.org is an office suite.\n" ),
 			"http://kde.openoffice.org/index.html",
-			"dev@kde.openoffice.org");
-	kAboutData->addAuthor( "Jan Holesovsky",
-			I18N_NOOP( "Original author and maintainer of the KDE NWF." ),
+			"dev@kde.openoffice.org" );
+	kAboutData->addAuthor( ki18n( "Jan Holesovsky" ),
+			ki18n( "Original author and maintainer of the KDE NWF." ),
 			"kendy@artax.karlin.mff.cuni.cz",
 			"http://artax.karlin.mff.cuni.cz/~kendy" );
+	kAboutData->addAuthor( ki18n("Eric Bischoff"),
+			ki18n( "Accessibility fixes, porting to KDE 4." ),
+			"bischoff@kde.org" );
 
     m_nFakeCmdLineArgs = 1;
 	USHORT nIdx;
@@ -191,11 +195,10 @@
 
 	KCmdLineArgs::init( m_nFakeCmdLineArgs, m_pAppCmdLineArgs, kAboutData );
 
-	KApplication::disableAutoDcopRegistration();
 	m_pApplication = new VCLKDEApplication();
-    kapp->disableSessionManagement();
+	kapp->disableSessionManagement();
 	
-	Display* pDisp = QPaintDevice::x11AppDisplay();
+	Display* pDisp = QX11Info::display();
 
 	SalDisplay *pSalDisplay = new SalKDEDisplay( pDisp );
 
