diff --git scp2/source/kde/file_kde.scp scp2/source/kde/file_kde.scp
index a6d6db4..9860f29 100644
--- scp2/source/kde/file_kde.scp
+++ scp2/source/kde/file_kde.scp
@@ -38,3 +38,12 @@ File gid_File_Lib_Kdebe
     Regmergefile = "kdebe1-ucd.txt";
 End
 
+File gid_File_Lib_Kde4be
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Dir_Program;
+    Name = STRING(CONCAT2(kde4be1.uno,UNXSUFFIX));
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Regmergefile = "kde4be1-ucd.txt";
+End
+
diff --git scp2/source/kde/makefile.mk scp2/source/kde/makefile.mk
index 0cbf6a4..6ddb301 100644
--- scp2/source/kde/makefile.mk
+++ scp2/source/kde/makefile.mk
@@ -43,7 +43,16 @@ TARGETTYPE=CUI
 
 SCP_PRODUCT_TYPE=osl
 
-.IF "$(ENABLE_KDE)" == "TRUE"
+.IF "$(ENABLE_KDE)" == "TRUE" || "$(ENABLE_KDE4)" == "TRUE"
+
+.IF "$(ENABLE_KDE)" != ""
+SCPDEFS+=-DENABLE_KDE
+.ENDIF
+
+.IF "$(ENABLE_KDE4)" != ""
+SCPDEFS+=-DENABLE_KDE4
+.ENDIF
+
 PARFILES =                   \
         module_kde.par     \
         file_kde.par
diff --git scp2/source/kde/module_kde.scp scp2/source/kde/module_kde.scp
index 1dce456..28dacbe 100644
--- scp2/source/kde/module_kde.scp
+++ scp2/source/kde/module_kde.scp
@@ -30,12 +30,24 @@
 
 #include "macros.inc"
 
+#if defined( ENABLE_KDE ) || defined( ENABLE_KDE4 )
 Module gid_Module_Optional_Kde
     ParentID = gid_Module_Optional;
     Default = YES;
     PackageInfo = "packinfo_office.txt";
     MOD_NAME_DESC(MODULE_OPTIONAL_KDE);
     Styles = (DONTSHOWINUSERINSTALL);
-    Files = (gid_File_Lib_Kdebe);
+    Files = (
+#ifdef ENABLE_KDE
+    gid_File_Lib_Kdebe
+#ifdef ENABLE_KDE4
+    ,
+#endif // ENABLE_KDE4
+#endif // ENABLE_KDE
+#ifdef ENABLE_KDE4
+    gid_File_Lib_Kde4be
+#endif // ENABLE_KDE4
+    );
 End
+#endif
 
diff --git scp2/source/ooo/file_library_ooo.scp scp2/source/ooo/file_library_ooo.scp
index eeba4ff..9242bae 100644
--- scp2/source/ooo/file_library_ooo.scp
+++ scp2/source/ooo/file_library_ooo.scp
@@ -630,6 +630,18 @@ File gid_File_Bin_KdeFilePicker
 End
 #endif
 #endif
+#ifdef ENABLE_KDE4
+#ifndef MACOSX
+File gid_File_Lib_Fps_Kde4
+    TXT_FILE_BODY;
+    Styles = (PACKED,UNO_COMPONENT);
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Dir = gid_Dir_Program;
+    Name = SPECIAL_COMPONENT_LIB_NAME(fps_kde4.uno);
+    Regmergefile = "fps-kde4-ucd.txt";
+End
+#endif
+#endif
 #endif
 
 #ifdef MACOSX
@@ -1202,6 +1214,16 @@ File gid_File_Lib_Vclplug_Kde
 End
 #endif
 #endif
+#ifdef ENABLE_KDE4
+#ifndef MACOSX
+File gid_File_Lib_Vclplug_Kde4
+    Name = LIBNAME(vclplug_kde4);
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Dir_Program;
+End
+#endif
+#endif
 #endif
 
 #if defined(QUARTZ)
diff --git scp2/source/ooo/makefile.mk scp2/source/ooo/makefile.mk
index 3140056..6b289e5 100644
--- scp2/source/ooo/makefile.mk
+++ scp2/source/ooo/makefile.mk
@@ -79,6 +79,10 @@ SCPDEFS+=-DENABLE_SYSTRAY_GTK
 SCPDEFS+=-DENABLE_KDE
 .ENDIF
 
+.IF "$(ENABLE_KDE4)" != ""
+SCPDEFS+=-DENABLE_KDE4
+.ENDIF
+
 .IF "$(ENABLE_KAB)" != ""
 SCPDEFS+=-DENABLE_KAB
 .ENDIF
diff --git scp2/source/ooo/module_hidden_ooo.scp scp2/source/ooo/module_hidden_ooo.scp
index 2b93f24..51942be 100644
--- scp2/source/ooo/module_hidden_ooo.scp
+++ scp2/source/ooo/module_hidden_ooo.scp
@@ -242,6 +242,7 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Vclplug_Gen,
 	gid_File_Lib_Vclplug_Gtk,
 	gid_File_Lib_Vclplug_Kde,
+	gid_File_Lib_Vclplug_Kde4,
 	gid_File_Lib_Vclplug_Svp,
 	gid_File_Lib_Basctl,
 	gid_File_Lib_CanvasTools,
