diff --git a/workben/layout/localize.sdf b/workben/layout/localize.sdf
index 61a3e51..8c60a67 100644
--- toolkit/workben/layout/localize.sdf
+++ toolkit/workben/layout/localize.sdf
@@ -1050,3 +1050,63 @@ layout	janneke\vc\layout-cws\toolkit\workben\layout\zoom.xml	0	layout	BTN_WHOLE_
 layout	janneke\vc\layout-cws\toolkit\workben\layout\zoom.xml	0	layout	BTN_WHOLE_PAGE_label				0	zh-CN	整页(~E)				2002-02-02 02:02:02
 layout	janneke\vc\layout-cws\toolkit\workben\layout\zoom.xml	0	layout	BTN_WHOLE_PAGE_label				0	zh-TW	整頁(~E)				2002-02-02 02:02:02
 layout	janneke\vc\layout-cws\toolkit\workben\layout\zoom.xml	0	layout	BTN_WHOLE_PAGE_label				0	zu	Ikhasi ~lonke				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	STR_COL_LABEL_text				0	en-US	Range contains column la~bels				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	STR_ROW_LABEL_text				0	en-US	Range contains ~row labels				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	XSCSTR_UNDEFINED_text				0	en-US	- undefined -				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_CASESENSITIVE_label				0	en-US	Case ~sensitive				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_FORMATS_label				0	en-US	Include ~formats				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_NATURALSORT_label				0	en-US	Enable ~natural sort				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_COPYRESULT_label				0	en-US	~Copy sort results to:				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_SORT_USER_label				0	en-US	Custom sort ~order				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_LANGUAGE_label				0	en-US	~Language				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_ALGORITHM_label				0	en-US	O~ptions				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_TOP_DOWN_label				0	en-US	~Top to bottom (sort rows)				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_LEFT_RIGHT_label				0	en-US	L~eft to right (sort columns)				20080528 15:28:37
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_CASESENSITIVE_label				0	de	~Groß-/Kleinschreibung				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_CASESENSITIVE_label				0	nl	~Hoofdletters/kleine letters				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_COPYRESULT_label				0	de	Sortierergebnis ~ausgeben nach				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_COPYRESULT_label				0	nl	~Sorteerresultaten kopiëren naar:				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_FORMATS_label				0	de	~Formate einschließen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_FORMATS_label				0	nl	~Opmaak opnemen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_SORT_USER_label				0	de	~Benutzerdefinierte Sortierreihenfolge				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_SORT_USER_label				0	nl	~Door gebruiker gedefinieerde sorteervolgorde				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_SORT1_text				0	de	~Sortieren nach				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_SORT1_text				0	nl	~Sorteren op				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_SORT2_text				0	de	~Anschließend nach				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_SORT2_text				0	nl	~Daarna op				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_SORT3_text				0	de	A~nschließend nach				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_SORT3_text				0	nl	D~aarna op				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_DIRECTION_text				0	de	Richtung				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FL_DIRECTION_text				0	nl	Richting				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_ALGORITHM_label				0	de	Op~tionen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_ALGORITHM_label				0	nl	O~pties				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_AREA_LABEL_label				0	de	Datenbereich:				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_AREA_LABEL_label				0	nl	Gegevensbereik:				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_LANGUAGE_label				0	de	S~prache				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	FT_LANGUAGE_label				0	nl	~Taal				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	TP_FIELDS_label				0	de	Sortierkriterien				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	TP_FIELDS_label				0	nl	Sorteercriteria				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	TP_OPTIONS_label				0	de	Optionen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	TP_OPTIONS_label				0	nl	Opties				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_DOWN1_label				0	de	A~bsteigend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_DOWN1_label				0	nl	A~flopend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_DOWN2_label				0	de	Abstei~gend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_DOWN2_label				0	nl	Aflo~pend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_DOWN3_label				0	de	Absteigen~d				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_DOWN3_label				0	nl	Aflopen~d				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_UP1_label				0	de	A~ufsteigend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_UP1_label				0	nl	Op~lopend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_UP2_label				0	de	Au~fsteigend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_UP2_label				0	nl	Oplope~nd				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_UP3_label				0	de	Aufs~teigend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_UP3_label				0	nl	~Oplopend				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_LEFT_RIGHT_label				0	de	Von ~links nach rechts (Spalten sortieren)				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_LEFT_RIGHT_label				0	nl	Van ~links naar rechts (kolommen sorteren)				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_TOP_DOWN_label				0	de	Von ~oben nach unten (Zeilen sortieren)				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	BTN_TOP_DOWN_label				0	nl	Van ~boven naar beneden (rijen sorteren)				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	STR_COL_LABEL_label				0	de	Bereich ~enthält Spaltenbeschriftungen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	STR_COL_LABEL_label				0	nl	Bereik bevat ~kolomkoppen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	STR_ROW_LABEL_label				0	de	Bereich ~enthält Zeilenbeschriftungen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	STR_ROW_LABEL_label				0	nl	Bereik bevat ~rijkoppen				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	RID_SCDLG_SORT_title				0	de	Sortieren				2002-02-02 02:02:02
+layout	janneke\vc\layout-cws\toolkit\workben\layout\sort-options.xml	0	layout	RID_SCDLG_SORT_title				0	nl	Sorteren				2002-02-02 02:02:02
