diff --git a/scp2/source/layout/layout.scp b/scp2/source/layout/layout.scp
index 6808269..c7c2983 100644
--- scp2/source/layout/layout.scp
+++ scp2/source/layout/layout.scp
@@ -35,6 +35,13 @@ Directory gid_Dir_Share_Layout
     DosName = "layout";
 End
 
+File gid_File_Zip_Toolkit_Layout
+   Dir = gid_Dir_Share_Layout;
+   USER_FILE_BODY;
+   Styles = (ARCHIVE, PATCH);
+   Name = "toolkit-layout.zip";
+End
+
 File gid_File_Zip_Sc_Layout
    Dir = gid_Dir_Share_Layout;
    USER_FILE_BODY;
