diff --git a/scp2/source/layout/layout.scp b/scp2/source/layout/layout.scp
index 477f992..6808269 100644
--- scp2/source/layout/layout.scp
+++ scp2/source/layout/layout.scp
@@ -35,11 +35,11 @@ Directory gid_Dir_Share_Layout
     DosName = "layout";
 End
 
-File gid_File_Zip_Sw_Layout
+File gid_File_Zip_Sc_Layout
    Dir = gid_Dir_Share_Layout;
    USER_FILE_BODY;
    Styles = (ARCHIVE, PATCH);
-   Name = "sw-layout.zip";
+   Name = "sc-layout.zip";
 End
 
 File gid_File_Zip_Svx_Layout
@@ -48,3 +48,11 @@ File gid_File_Zip_Svx_Layout
    Styles = (ARCHIVE, PATCH);
    Name = "svx-layout.zip";
 End
+
+File gid_File_Zip_Sw_Layout
+   Dir = gid_Dir_Share_Layout;
+   USER_FILE_BODY;
+   Styles = (ARCHIVE, PATCH);
+   Name = "sw-layout.zip";
+End
+
