diff --git a/sd/source/ui/view/Outliner.cxx b/sd/source/ui/view/Outliner.cxx
index 0906ad7..5391c52 100644
--- sd/source/ui/view/Outliner.cxx
+++ sd/source/ui/view/Outliner.cxx
@@ -1472,7 +1472,9 @@ bool Outliner::HandleFailedSearch (void)
 }
 
 
-
+#if ENABLE_LAYOUT
+#define SvxSearchDialog Window
+#endif
 
 /** See task #95227# for discussion about correct parent for dialogs/info boxes.
 */
@@ -1495,6 +1497,7 @@ bool Outliner::HandleFailedSearch (void)
     return pParent;
 }
 
+#undef SvxSearchDialog Window
 
 
 
