diff --git a/scripting/source/protocolhandler/scripthandler.cxx b/scripting/source/protocolhandler/scripthandler.cxx
index 46c8bca..56a5315 100644
--- scripting/source/protocolhandler/scripthandler.cxx
+++ scripting/source/protocolhandler/scripthandler.cxx
@@ -539,6 +539,9 @@ const Reference< XMultiServiceFactory >& xServiceManager )
 extern "C"
 {
 
+#undef css
+#define css ::com::sun::star
+    
     void SAL_CALL component_getImplementationEnvironment( 
         const sal_Char** ppEnvironmentTypeName, uno_Environment** ppEnvironment )
     {
diff --git a/scripting/source/protocolhandler/scripthandler.hxx b/scripting/source/protocolhandler/scripthandler.hxx
index 3f00fa4..61ba5b4 100644
--- scripting/source/protocolhandler/scripthandler.hxx
+++ scripting/source/protocolhandler/scripthandler.hxx
@@ -75,8 +75,7 @@ namespace com { namespace sun { namespace star {
 namespace scripting_protocolhandler
 {
 
-// for simplification
-#define css ::com::sun::star
+namespace css = ::com::sun::star;
 
 class ScriptProtocolHandler : 
 public ::cppu::WeakImplHelper4< css::frame::XDispatchProvider,
