diff --git a/vcl/inc/vcl/button.hxx b/vcl/inc/vcl/button.hxx
index 1155c5e..db27ba4 100644
--- vcl/inc/vcl/button.hxx
+++ vcl/inc/vcl/button.hxx
@@ -103,6 +103,8 @@ public:
 
     void                SetFocusRect( const Rectangle& rFocusRect );
     const Rectangle&    GetFocusRect() const;
+    void SetSmallSymbol (bool small=true);
+    bool IsSmallSymbol () const;
 };
 
 // --------------------
diff --git a/vcl/inc/vcl/tabctrl.hxx b/vcl/inc/vcl/tabctrl.hxx
index 2b1d49a..83e83a6 100644
--- vcl/inc/vcl/tabctrl.hxx
+++ vcl/inc/vcl/tabctrl.hxx
@@ -45,8 +45,10 @@ class PushButton;
 // - TabControl-Types -
 // --------------------
 
+#ifndef TAB_APPEND
 #define TAB_APPEND          ((USHORT)0xFFFF)
 #define TAB_PAGE_NOTFOUND   ((USHORT)0xFFFF)
+#endif /* !TAB_APPEND */
 
 // --------------
 // - TabControl -
diff --git a/vcl/source/control/button.cxx b/vcl/source/control/button.cxx
index 32f3149..d7b6922 100644
--- vcl/source/control/button.cxx
+++ vcl/source/control/button.cxx
@@ -796,6 +796,15 @@ void Button::DataChanged( const DataChangedEvent& rDCEvt )
 	}
 }
 
+void Button::SetSmallSymbol (bool small)
+{
+    ImplSetSmallSymbol (small);
+}
+
+bool Button::IsSmallSymbol () const
+{
+    return mpButtonData->mbSmallSymbol;
+}
 
 // =======================================================================
 
@@ -1966,7 +1975,12 @@ Size PushButton::CalcMinimumSize( long nMaxWidth ) const
 	Size aSize;
 
 	if ( IsSymbol() )
-		aSize = Size( 12, 12 );
+    {
+        if ( IsSmallSymbol ())
+            aSize = Size( 16, 12 );
+        else
+            aSize = Size( 26, 24 );
+    }
 	else if ( IsImage() && ! (ImplGetButtonState() & BUTTON_DRAW_NOIMAGE) )
 		aSize = GetModeImage().GetSizePixel();
 	if ( PushButton::GetText().Len() && ! (ImplGetButtonState() & BUTTON_DRAW_NOTEXT) )
