diff --git a/inc/sfx2/macrconf.hxx b/inc/sfx2/macrconf.hxx
index 94bbf4b..4155b05 100644
--- sfx2/inc/sfx2/macrconf.hxx
+++ sfx2/inc/sfx2/macrconf.hxx
@@ -141,7 +141,7 @@ public:
 	sal_uInt16					GetSlotId(SfxMacroInfoPtr);
 	void					ReleaseSlotId(sal_uInt16 nId);
 	void					RegisterSlotId(sal_uInt16 nId);
-	const SfxMacroInfoPtr 	GetMacroInfo(sal_uInt16 nId) const;
+    SfxMacroInfoPtr GetMacroInfo(sal_uInt16 nId) const;
 	sal_Bool					ExecuteMacro(sal_uInt16 nId, const String& rArgs ) const;
 	sal_Bool					ExecuteMacro( SfxObjectShell*, const SvxMacro*, const String& ) const;
 	sal_Bool					CheckMacro(sal_uInt16 nId) const;
@@ -149,7 +149,7 @@ public:
 
 //#if 0 // _SOLAR__PRIVATE
 	SAL_DLLPRIVATE static void Release_Impl();
-	SAL_DLLPRIVATE const SfxMacroInfoPtr GetMacroInfo_Impl( const SvxMacro *pMacro ) const;
+    SAL_DLLPRIVATE SfxMacroInfoPtr GetMacroInfo_Impl( const SvxMacro *pMacro ) const;
 	DECL_DLLPRIVATE_LINK( CallbackHdl_Impl, SfxMacroConfig*);
 	DECL_DLLPRIVATE_LINK( EventHdl_Impl, SfxMacroInfo*);
 //#endif
diff --git a/inc/sfx2/tabdlg.hxx b/inc/sfx2/tabdlg.hxx
index d5b930e..4846f0b 100644
--- sfx2/inc/sfx2/tabdlg.hxx
+++ sfx2/inc/sfx2/tabdlg.hxx
@@ -43,6 +43,9 @@
 #include <svtools/itemset.hxx>
 #include <com/sun/star/frame/XFrame.hpp>
 
+#include <map>
+namespace layout { class TabPage; }
+
 class SfxPoolItem;
 class SfxTabDialog;
 class SfxViewFrame;
@@ -98,6 +101,7 @@ friend class SfxTabDialogController;
 	USHORT				nAppPageId;
 	BOOL				bItemsReset;
 	BOOL				bFmt;
+    std::map<int, layout::TabPage*> layoutPages;
 
 //#if 0 // _SOLAR__PRIVATE
 	DECL_DLLPRIVATE_LINK( ActivatePageHdl, TabControl * );
@@ -168,6 +172,7 @@ public:
 									BOOL bItemsOnDemand = FALSE,
 									USHORT nPos = TAB_APPEND);
 
+    void AddTabPage( int id, layout::TabPage* page );
 	void				RemoveTabPage( USHORT nId );
 
 	void				SetCurPageId( USHORT nId ) { nAppPageId = nId; }
@@ -181,7 +186,7 @@ public:
 	const SfxItemSet*	GetOutputItemSet() const { return pOutSet; }
 	const SfxItemSet*	GetOutputItemSet( USHORT nId ) const;
 	int					FillOutputItemSet();
-	const BOOL			IsFormat() const { return bFmt; }
+	BOOL IsFormat() const { return bFmt; }
 
 	const OKButton& 	GetOKButton() const { return aOKBtn; }
 	OKButton&			GetOKButton() { return aOKBtn; }
diff --git a/source/control/macrconf.cxx b/source/control/macrconf.cxx
index 0a2a595..1ab851b 100644
--- sfx2/source/control/macrconf.cxx
+++ sfx2/source/control/macrconf.cxx
@@ -662,7 +662,7 @@ void SfxMacroConfig::RegisterSlotId(sal_uInt16 nId)
 
 //==========================================================================
 
-const SfxMacroInfoPtr SfxMacroConfig::GetMacroInfo(sal_uInt16 nId) const
+SfxMacroInfoPtr SfxMacroConfig::GetMacroInfo(sal_uInt16 nId) const
 {
 	sal_uInt16 nCount = pImp->aArr.Count();
 	for (sal_uInt16 i=0; i<nCount; i++)
@@ -674,7 +674,7 @@ const SfxMacroInfoPtr SfxMacroConfig::GetMacroInfo(sal_uInt16 nId) const
 
 //==========================================================================
 
-const SfxMacroInfoPtr SfxMacroConfig::GetMacroInfo_Impl( const SvxMacro *pMacro ) const
+SfxMacroInfoPtr SfxMacroConfig::GetMacroInfo_Impl( const SvxMacro *pMacro ) const
 {
 	sal_uInt16 nCount = pImp->aArr.Count();
 	for (sal_uInt16 i=0; i<nCount; i++)
diff --git a/source/dialog/tabdlg.cxx b/source/dialog/tabdlg.cxx
index 57a8e89..7bcca09 100644
--- sfx2/source/dialog/tabdlg.cxx
+++ sfx2/source/dialog/tabdlg.cxx
@@ -56,6 +56,7 @@
 #include <sfx2/bindings.hxx>
 #include <sfx2/sfxdlg.hxx>
 #include <sfx2/itemconnect.hxx>
+#include <layout/layout.hxx>
 
 #include "dialog.hrc"
 #include "helpid.hrc"
@@ -171,6 +172,12 @@ void SfxTabDialogController::StateChanged( USHORT /*nSID*/, SfxItemState /*eStat
         for ( USHORT n=0; n<pDialog->aTabCtrl.GetPageCount(); n++ )
         {
             USHORT nPageId = pDialog->aTabCtrl.GetPageId( n );
+            if ( layout::TabPage* page = pDialog->layoutPages[nPageId] )
+            {
+                page->Reset( pSetItem->GetItemSet() );
+                bDialogStarted = TRUE;
+            }
+
             SfxTabPage* pTabPage = (SfxTabPage*) pDialog->aTabCtrl.GetTabPage( nPageId );
             if ( pTabPage )
             {
@@ -922,6 +929,11 @@ void SfxTabDialog::AddTabPage
 }
 #endif
 
+void SfxTabDialog::AddTabPage( int id, layout::TabPage* page )
+{
+    layoutPages[id] = page;
+}
+
 // -----------------------------------------------------------------------
 
 void SfxTabDialog::RemoveTabPage( USHORT nId )
@@ -1379,6 +1391,24 @@ IMPL_LINK( SfxTabDialog, ActivatePageHdl, TabControl *, pTabCtrl )
 */
 
 {
+    int newId = pTabCtrl->GetCurPageId();
+    if ( layout::TabPage* page = layoutPages[newId] )
+    {
+        TabPage *vclPage = page->GetTabPage();
+        TabControl *dialog = pTabCtrl;
+        page->Reset( page->GetItemSet() );
+        page->ActivatePage();
+        dialog->SetTabPage( newId, vclPage );
+
+        Size tabSize = vclPage->GetSizePixel();
+        Size dialogSize = dialog->GetTabPageSizePixel();
+        OSL_TRACE ("%s: tabSize: %d, %d", __FUNCTION__, tabSize.Width(), tabSize.Height() );
+        OSL_TRACE ("%s: dialogSize: %d, %d", __FUNCTION__, dialogSize.Width(), dialogSize.Height() );
+
+        dialog->SetTabPageSizePixel( tabSize );
+        return 0;
+    }
+    
 	DBG_ASSERT( pImpl->pData->Count(), "keine Pages angemeldet" );
 	const USHORT nId = pTabCtrl->GetCurPageId();
 	SFX_APP();
@@ -1455,6 +1485,9 @@ IMPL_LINK( SfxTabDialog, DeactivatePageHdl, TabControl *, pTabCtrl )
 */
 
 {
+    if ( layout::TabPage* p = layoutPages[pTabCtrl->GetCurPageId()] )
+        return SfxTabPage::LEAVE_PAGE;
+
 	USHORT nId = pTabCtrl->GetCurPageId();
 	SFX_APP();
 	SfxTabPage *pPage = (SfxTabPage*)pTabCtrl->GetTabPage( nId );
