diff --git a/sfx2/inc/sfx2/layout-post.hxx b/sfx2/inc/sfx2/layout-post.hxx
new file mode 100644
index 0000000..0eab118
--- /dev/null
+++ sfx2/inc/sfx2/layout-post.hxx
@@ -0,0 +1,44 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile$
+ *
+ * $Revision$
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef _SFX2_LAYOUT_POST_HXX
+#define _SFX2_LAYOUT_POST_HXX
+
+#include <layout/layout-post.hxx>
+
+#if ENABLE_LAYOUT
+
+/* Allow re-inclusion for cxx file. */
+#undef _SFX2_LAYOUT_PRE_HXX
+
+#endif /* ENABLE_LAYOUT */
+
+#endif /* _SFX2_LAYOUT_POST_HXX */
diff --git a/sfx2/inc/sfx2/layout-pre.hxx b/sfx2/inc/sfx2/layout-pre.hxx
new file mode 100644
index 0000000..bb8935c
--- /dev/null
+++ sfx2/inc/sfx2/layout-pre.hxx
@@ -0,0 +1,53 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile$
+ *
+ * $Revision$
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef _SFX2_LAYOUT_PRE_HXX
+#define _SFX2_LAYOUT_PRE_HXX
+
+#include <layout/layout-pre.hxx>
+
+#if ENABLE_LAYOUT
+
+///* Allow re-inclusion for next hxx file. */
+#undef _SFX2_LAYOUT_POST_HXX
+
+#undef SfxDialog
+#undef SfxModalDialog
+#undef SfxModelessDialog
+
+#define SfxDialog layout::SfxDialog
+
+#define SfxModalDialog SfxDialog
+#define SfxModelessDialog SfxDialog
+
+#endif /* ENABLE_LAYOUT */
+
+#endif /* _SFX2_LAYOUT_PRE_HXX */
diff --git a/sfx2/inc/sfx2/layout-tabdlg.hxx b/sfx2/inc/sfx2/layout-tabdlg.hxx
new file mode 100644
index 0000000..4b84083
--- /dev/null
+++ sfx2/inc/sfx2/layout-tabdlg.hxx
@@ -0,0 +1,49 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: tabdlg.hxx,v $
+ * $Revision: 1.4 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+#ifndef _LAYOUT_SFX_TABDLG_HXX
+#define _LAYOUT_SFX_TABDLG_HXX
+
+#undef ENABLE_LAYOUT_SFX_TABDIALOG
+#define ENABLE_LAYOUT_SFX_TABDIALOG 1
+
+#undef NAMESPACE_LAYOUT_SFX_TABDIALOG
+#define NAMESPACE_LAYOUT_SFX_TABDIALOG namespace layout {
+
+#undef END_NAMESPACE_LAYOUT_SFX_TABDIALOG
+#define END_NAMESPACE_LAYOUT_SFX_TABDIALOG } //end namespace layout
+
+#undef LAYOUT_NS_SFX_TABDIALOG
+#define LAYOUT_NS_SFX_TABDIALOG layout::
+
+#undef _SFXTABDLG_HXX
+#include <sfx2/tabdlg.hxx>
+
+#endif /* _LAYOUT_SFX_TABDLG_HXX */
+
diff --git a/sfx2/inc/sfx2/layout.hxx b/sfx2/inc/sfx2/layout.hxx
new file mode 100644
index 0000000..1807213
--- /dev/null
+++ sfx2/inc/sfx2/layout.hxx
@@ -0,0 +1,79 @@
+/*************************************************************************
+* DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+* 
+* Copyright 2008 by Sun Microsystems, Inc.
+*
+* OpenOffice.org - a multi-platform office productivity suite
+*
+* $RCSfile: code,v $
+*
+* $Revision: 1.3 $
+*
+* This file is part of OpenOffice.org.
+*
+* OpenOffice.org is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License version 3
+* only, as published by the Free Software Foundation.
+*
+* OpenOffice.org is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License version 3 for more details
+* (a copy is included in the LICENSE file that accompanied this code).
+*
+* You should have received a copy of the GNU Lesser General Public License
+* version 3 along with OpenOffice.org.  If not, see
+* <http://www.openoffice.org/license.html>
+* for a copy of the LGPLv3 License.
+************************************************************************/
+
+#ifndef _SFX2_LAYOUT_HXX
+#define _SFX2_LAYOUT_HXX
+
+#include <layout/layout.hxx>
+#include <sfx2/tabdlg.hxx>
+#undef ENABLE_LAYOUT_SFX_TABDIALOG
+#define ENABLE_LAYOUT_SFX_TABDIALOG 1
+#include <sfx2/layout-tabdlg.hxx>
+
+class SfxChildWindow;
+
+namespace layout
+{
+
+namespace css = ::com::sun::star;
+
+
+class SFX2_DLLPUBLIC SfxDialog
+    : public Dialog
+{
+public:
+    SfxDialog (::Window* parent, char const* xml_file, char const* id, SfxBindings* bindings=0, SfxChildWindow* child=0);
+};
+
+#if 0
+class SFX2_DLLPUBLIC SfxTabDialog
+    : public ::SfxTabDialog
+    , public InPlug
+{
+public:
+    SfxTabDialog (::Window *parent, char const* xml_file, char const* id, ResId const& res_id, SfxItemSet const* set=0);
+    ::Window* GetParent () const;
+    void FreeResource ();
+};
+#endif
+
+class SFX2_DLLPUBLIC SfxTabPage
+    : public ::SfxTabPage
+    , public InPlug
+{
+public:
+    SfxTabPage (::Window *parent, char const* xml_file, char const* id, SfxItemSet const* set=0);
+    ::Window* GetParent () const;
+    Size GetOptimalSize (WindowSizeType eType) const;
+    void FreeResource ();
+};
+
+} // end namespace layout
+
+#endif /* _SFX2_LAYOUT_HXX */
diff --git a/sfx2/inc/sfx2/sfxdlg.hxx b/sfx2/inc/sfx2/sfxdlg.hxx
index a13c5a8..9e47ca1 100644
--- sfx2/inc/sfx2/sfxdlg.hxx
+++ sfx2/inc/sfx2/sfxdlg.hxx
@@ -55,8 +55,9 @@ class SvObjectServerList;
 class TransferableDataHelper;
 struct TransferableObjectDescriptor;
 
-typedef SfxTabPage* (*CreateTabPage)(Window *pParent, const SfxItemSet &rAttrSet);
-typedef USHORT*		(*GetTabPageRanges)();
+#include <sfx2/tabdlg.hxx>
+//typedef SfxTabPage* (*CreateTabPage)(Window *pParent, const SfxItemSet &rAttrSet);
+//typedef USHORT*		(*GetTabPageRanges)();
 
 namespace sfx2
 {
diff --git a/sfx2/inc/sfx2/tabdlg.hxx b/sfx2/inc/sfx2/tabdlg.hxx
index b3eb789..b02bd15 100644
--- sfx2/inc/sfx2/tabdlg.hxx
+++ sfx2/inc/sfx2/tabdlg.hxx
@@ -34,17 +34,17 @@
 #include "sfx2/dllapi.h"
 #include "sal/types.h"
 #include <vcl/tabdlg.hxx>
-#ifndef _SV_BUTTON_HXX //autogen
 #include <vcl/button.hxx>
-#endif
 #include <vcl/tabctrl.hxx>
 #include <vcl/tabpage.hxx>
 #include <svtools/itempool.hxx>
 #include <svtools/itemset.hxx>
 #include <com/sun/star/frame/XFrame.hpp>
 
-#include <map>
-namespace layout { class TabPage; }
+#if ENABLE_LAYOUT
+#include <layout/layout.hxx>
+namespace layout { class SfxTabDialog; }
+#endif /* ENABLE_LAYOUT */
 
 class SfxPoolItem;
 class SfxTabDialog;
@@ -52,21 +52,27 @@ class SfxViewFrame;
 class SfxTabPage;
 class SfxBindings;
 
-// typedefs --------------------------------------------------------------
+#ifndef ENABLE_LAYOUT_SFX_TABDIALOG
+#define ENABLE_LAYOUT_SFX_TABDIALOG 0
+#define NAMESPACE_LAYOUT_SFX_TABDIALOG
+#define END_NAMESPACE_LAYOUT_SFX_TABDIALOG
+#define LAYOUT_NS_SFX_TABDIALOG
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG*/
 
 typedef SfxTabPage* (*CreateTabPage)(Window *pParent, const SfxItemSet &rAttrSet);
-namespace layout
-{
-class TabPage;
-typedef TabPage* (*CreateTabPage)(::Window *pParent, ::SfxItemSet const& set );
-}
 typedef USHORT*		(*GetTabPageRanges)(); // liefert internationale Which-Wert
-
-struct TabDlg_Impl;
 struct TabPageImpl;
 class SfxUs_Impl;
 
-// class SfxTabDialog ----------------------------------------------------
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#include <layout/layout-pre.hxx>
+#undef SfxTabDialog
+#undef SfxTabPage
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+
+NAMESPACE_LAYOUT_SFX_TABDIALOG
+
+class TabDlg_Impl;
 
 #define ID_TABCONTROL	1
 #define RET_USER 		100
@@ -106,8 +112,6 @@ friend class SfxTabDialogController;
 	USHORT				nAppPageId;
 	BOOL				bItemsReset;
 	BOOL				bFmt;
-    std::map<int, layout::TabPage*> layoutPages;
-    std::map<int, layout::CreateTabPage> layoutPageFuncs;
 
 //#if 0 // _SOLAR__PRIVATE
 	DECL_DLLPRIVATE_LINK( ActivatePageHdl, TabControl * );
@@ -178,8 +182,6 @@ public:
 									BOOL bItemsOnDemand = FALSE,
 									USHORT nPos = TAB_APPEND);
 
-    void AddTabPage( int id, layout::TabPage* page, bool=false );
-    void AddTabPage( int id, layout::CreateTabPage, bool=false );
 	void				RemoveTabPage( USHORT nId );
 
 	void				SetCurPageId( USHORT nId ) { nAppPageId = nId; }
@@ -215,7 +217,11 @@ public:
     void                StartExecuteModal( const Link& rEndDialogHdl );
 	void				Start( BOOL bShow = TRUE );
 
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 	const SfxItemSet*	GetExampleSet() const { return pExampleSet; }
+#else /* ENABLE_LAYOUT_SFX_TABDIALOG */
+    SfxItemSet*	GetExampleSet() const { return 0; }
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
 	SfxViewFrame*		GetViewFrame() const { return pFrame; }
 
 	void				EnableApplyButton(BOOL bEnable = TRUE);
@@ -229,13 +235,20 @@ public:
 //#endif
 };
 
-// class SfxTabPage ------------------------------------------------------
+END_NAMESPACE_LAYOUT_SFX_TABDIALOG
+
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#include <layout/layout-post.hxx>
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 
 namespace sfx { class ItemConnectionBase; }
 
 class SFX2_DLLPUBLIC SfxTabPage: public TabPage
 {
 friend class SfxTabDialog;
+friend class layout::SfxTabDialog;
 
 private:
 	const SfxItemSet*	pSet;
@@ -300,5 +313,7 @@ public:
     ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame > GetFrame();
 };
 
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
+
 #endif
 
diff --git a/sfx2/prj/build.lst b/sfx2/prj/build.lst
index d762a1d..72ba4ac 100644
--- sfx2/prj/build.lst
+++ sfx2/prj/build.lst
@@ -13,10 +13,11 @@ sf	sfx2\source\config						nmake	-	all	sf_cnfg sf_sdi sf_inc NULL
 sf	sfx2\source\control						nmake	-	all	sf_ctrl sf_sdi sf_inc NULL
 sf	sfx2\source\dialog						nmake	-	all	sf_dlg sf_sdi sf_inc NULL
 sf	sfx2\source\doc							nmake	-	all	sf_doc sf_sdi sf_inc NULL
+sf	sfx2\source\layout						nmake	-	all	sf_layout sf_sdi sf_inc NULL
 sf	sfx2\source\menu						nmake	-	all	sf_menu sf_sdi sf_inc NULL
 sf	sfx2\source\notify						nmake	-	all	sf_noti sf_sdi sf_inc NULL
 sf	sfx2\source\statbar						nmake	-	all	sf_sbar sf_sdi sf_inc NULL
 sf	sfx2\source\toolbox						nmake	-	all	sf_tbox sf_sdi sf_inc NULL
 sf	sfx2\source\inet						nmake	-	all	sf_inet sf_sdi sf_inc NULL
 sf	sfx2\source\explorer					nmake	-	all	sf_expl sf_sdi sf_inc NULL
-sf      sfx2\util                                                               nmake   -       all     sf_util sf_appl sf_bast sf_cnfg sf_ctrl sf_dlg sf_doc sf_expl sf_inet sf_menu sf_noti sf_sbar sf_tbox sf_view NULL
+sf      sfx2\util                                                               nmake   -       all     sf_util sf_appl sf_bast sf_cnfg sf_ctrl sf_dlg sf_doc sf_expl sf_inet sf_menu sf_layout sf_noti sf_sbar sf_tbox sf_view NULL
diff --git a/sfx2/prj/d.lst b/sfx2/prj/d.lst
index 03d2c01..53ed0df 100644
--- sfx2/prj/d.lst
+++ sfx2/prj/d.lst
@@ -126,3 +126,7 @@ mkdir: %_DEST%\inc%_EXT%\sfx2
 ..\inc\QuerySaveDocument.hxx %_DEST%\inc%_EXT%\sfx2\QuerySaveDocument.hxx
 ..\inc\mailmodelapi.hxx %_DEST%\inc%_EXT%\sfx2\mailmodelapi.hxx
 ..\inc\docinsert.hxx %_DEST%\inc%_EXT%\sfx2\docinsert.hxx
+..\inc\sfx2\layout.hxx %_DEST%\inc%_EXT%\sfx2\layout.hxx
+..\inc\sfx2\layout-tabdlg.hxx %_DEST%\inc%_EXT%\sfx2\layout-tabdlg.hxx
+..\inc\sfx2\layout-post.hxx %_DEST%\inc%_EXT%\sfx2\layout-post.hxx
+..\inc\sfx2\layout-pre.hxx %_DEST%\inc%_EXT%\sfx2\layout-pre.hxx
diff --git a/sfx2/source/dialog/tabdlg.cxx b/sfx2/source/dialog/tabdlg.cxx
index d3254ed..fb35c4f 100644
--- sfx2/source/dialog/tabdlg.cxx
+++ sfx2/source/dialog/tabdlg.cxx
@@ -31,19 +31,14 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_sfx2.hxx"
 
-// include ---------------------------------------------------------------
-
 #include <limits.h>
 #include <stdlib.h>
 #include <vcl/msgbox.hxx>
 #include <svtools/viewoptions.hxx>
-#ifndef GCC
-#endif
 
 #define _SVSTDARR_USHORTS
 #include <svtools/svstdarr.hxx>
 
-// wg. nAutoPageID
 #include "appdata.hxx"
 #include "sfxtypes.hxx"
 #include <sfx2/minarray.hxx>
@@ -56,20 +51,35 @@
 #include <sfx2/bindings.hxx>
 #include <sfx2/sfxdlg.hxx>
 #include <sfx2/itemconnect.hxx>
-#include <layout/layout.hxx>
 
 #include "dialog.hrc"
 #include "helpid.hrc"
 
-// for OUString and Any
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#undef TabPage
+#undef SfxTabPage
+#define SfxTabPage ::SfxTabPage
+#undef SfxTabDialog
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+
 using namespace ::com::sun::star::uno;
 using namespace ::rtl;
 
 #define USERITEM_NAME			OUString::createFromAscii( "UserItem" )
 
-/*	-----------------------------------------------------------------
-	Verwaltungsdatenstruktur f"ur jede Seite
-	----------------------------------------------------------------- */
+TYPEINIT1(LAYOUT_NS_SFX_TABDIALOG SfxTabDialogItem,SfxSetItem);
+
+struct TabPageImpl
+{
+    BOOL                        mbStandard;
+    sfx::ItemConnectionArray    maItemConn;
+    ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame > mxFrame;
+
+    TabPageImpl() : mbStandard( FALSE ) {}
+};
+
+NAMESPACE_LAYOUT_SFX_TABDIALOG
+
 struct Data_Impl
 {
 	USHORT nId;					 // Die ID
@@ -102,8 +112,6 @@ struct Data_Impl
 	}
 };
 
-TYPEINIT1(SfxTabDialogItem,SfxSetItem);
-
 SfxTabDialogItem::SfxTabDialogItem( const SfxTabDialogItem& rAttr, SfxItemPool* pItemPool )
     : SfxSetItem( rAttr, pItemPool )
 {
@@ -172,13 +180,7 @@ void SfxTabDialogController::StateChanged( USHORT /*nSID*/, SfxItemState /*eStat
         for ( USHORT n=0; n<pDialog->aTabCtrl.GetPageCount(); n++ )
         {
             USHORT nPageId = pDialog->aTabCtrl.GetPageId( n );
-            if ( layout::TabPage* page = pDialog->layoutPages[nPageId] )
-            {
-                page->Reset( pSetItem->GetItemSet() );
-                bDialogStarted = TRUE;
-            }
-
-            SfxTabPage* pTabPage = (SfxTabPage*) pDialog->aTabCtrl.GetTabPage( nPageId );
+            SfxTabPage* pTabPage = dynamic_cast<SfxTabPage*> (pDialog->aTabCtrl.GetTabPage( nPageId ));
             if ( pTabPage )
             {
                 pTabPage->Reset( pSetItem->GetItemSet() );
@@ -218,32 +220,6 @@ struct TabDlg_Impl
 	{}
 };
 
-struct TabPageImpl
-{
-    BOOL                        mbStandard;
-    sfx::ItemConnectionArray    maItemConn;
-    ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame > mxFrame;
-
-    TabPageImpl() : mbStandard( FALSE ) {}
-};
-
-void SfxTabPage::SetFrame(const ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame >& xFrame)
-{
-    if (pImpl)
-        pImpl->mxFrame = xFrame;
-}
-
-::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame > SfxTabPage::GetFrame()
-{
-    if (pImpl)
-        return pImpl->mxFrame;
-    return ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame >();
-}
-
-
-/*	-----------------------------------------------------------------
-	Hilfsfunktion: Finden einer Page anhand seiner Id
-	----------------------------------------------------------------- */
 Data_Impl* Find( SfxTabDlgData_Impl& rArr, USHORT nId, USHORT* pPos = 0 );
 
 Data_Impl* Find( SfxTabDlgData_Impl& rArr, USHORT nId, USHORT* pPos )
@@ -264,8 +240,21 @@ Data_Impl* Find( SfxTabDlgData_Impl& rArr, USHORT nId, USHORT* pPos )
 	return 0;
 }
 
-// class SfxTabPage-------------------------------------------------------
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
+
+void SfxTabPage::SetFrame(const ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame >& xFrame)
+{
+    if (pImpl)
+        pImpl->mxFrame = xFrame;
+}
 
+::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame > SfxTabPage::GetFrame()
+{
+    if (pImpl)
+        return pImpl->mxFrame;
+    return ::com::sun::star::uno::Reference< ::com::sun::star::frame::XFrame >();
+}
+          
 SfxTabPage::SfxTabPage( Window *pParent,
 						const ResId &rResId, const SfxItemSet &rAttrSet ) :
 
@@ -474,7 +463,22 @@ void SfxTabPage::AddItemConnection( sfx::ItemConnectionBase* pConnection )
     pImpl->maItemConn.AddConnection( pConnection );
 }
 
-// class SfxTabDialog ----------------------------------------------------
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
+
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#undef ResId
+#define ResId(id, foo) #id
+#undef TabDialog
+#define TabDialog(parent, res_id) Dialog (parent, "tab-dialog.xml", "tab-dialog")
+
+#define aOKBtn(this) aOKBtn (this, "BTN_OK")
+#undef PushButton
+#define PushButton(this) layout::PushButton (this, "BTN_USER")
+#define aCancelBtn(this) aCancelBtn (this, "BTN_CANCEL")
+#define aHelpBtn(this) aHelpBtn (this, "BTN_HELP")
+#define aResetBtn(this) aResetBtn (this, "BTN_RESET")
+#define aBaseFmtBtn(this) aBaseFmtBtn (this, "BTN_BASEFMT")
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
 
 #define INI_LIST(ItemSetPtr) \
 	aTabCtrl	( this, ResId(ID_TABCONTROL,*rResId.GetResMgr() ) ),\
@@ -585,11 +589,24 @@ SfxTabDialog::SfxTabDialog
 
 // -----------------------------------------------------------------------
 
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#undef ResId
+#undef TabDialog
+#undef aOKBtn
+#undef PushButton
+#undef aCancelBtn
+#undef aHelpBtn
+#undef aResetBtn
+#undef aBaseFmtBtn
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+
 SfxTabDialog::~SfxTabDialog()
 {
 	// save settings (screen position and current page)
 	SvtViewOptions aDlgOpt( E_TABDIALOG, String::CreateFromInt32( nResId ) );
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 	aDlgOpt.SetWindowState( OUString::createFromAscii( GetWindowState( WINDOWSTATE_MASK_POS ).GetBuffer() ) );
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
 	aDlgOpt.SetPageID( aTabCtrl.GetCurPageId() );
 
 	const USHORT nCount = pImpl->pData->Count();
@@ -624,10 +641,6 @@ SfxTabDialog::~SfxTabDialog()
 	delete pOutSet;
 	delete pExampleSet;
 	delete [] pRanges;
-
-    for ( size_t i = 0; i < layoutPages.size(); i++ )
-        if (layout::TabPage *page = layoutPages[i])
-            delete page;
 }
 
 // -----------------------------------------------------------------------
@@ -697,6 +710,11 @@ void SfxTabDialog::RemoveResetButton()
 
 // -----------------------------------------------------------------------
 
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#undef TabDialog
+#define TabDialog Dialog
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+
 short SfxTabDialog::Execute()
 {
 	if ( !aTabCtrl.GetPageCount() )
@@ -709,10 +727,12 @@ short SfxTabDialog::Execute()
 
 void SfxTabDialog::StartExecuteModal( const Link& rEndDialogHdl )
 {
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
     if ( !aTabCtrl.GetPageCount() )
         return;
     Start_Impl();
     TabDialog::StartExecuteModal( rEndDialogHdl );
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
 }
 
 // -----------------------------------------------------------------------
@@ -759,9 +779,10 @@ void SfxTabDialog::EnableApplyButton(BOOL bEnable)
 	if ( bEnable )
 	{
 		pImpl->pApplyButton = new PushButton( this );
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 		// in the z-order, the apply button should be behind the ok button, thus appearing at the right side of it
 		pImpl->pApplyButton->SetZOrder(&aOKBtn, WINDOW_ZORDER_BEHIND);
-
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
 		pImpl->pApplyButton->SetText( String( SfxResId( STR_APPLY ) ) );
 		pImpl->pApplyButton->Show();
 
@@ -773,9 +794,11 @@ void SfxTabDialog::EnableApplyButton(BOOL bEnable)
 		pImpl->pApplyButton = NULL;
 	}
 
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 	// adjust the layout
 	if (IsReallyShown())
 		AdjustLayout();
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
 }
 
 // -----------------------------------------------------------------------
@@ -810,7 +833,9 @@ void SfxTabDialog::Start_Impl()
 	SvtViewOptions aDlgOpt( E_TABDIALOG, String::CreateFromInt32( nResId ) );
 	if ( aDlgOpt.Exists() )
 	{
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 		SetWindowState( ByteString( aDlgOpt.GetWindowState().getStr(), RTL_TEXTENCODING_ASCII_US ) );
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
 
 		// initiale TabPage aus Programm/Hilfe/Konfig
 		nActPage = (USHORT)aDlgOpt.GetPageID();
@@ -933,16 +958,6 @@ void SfxTabDialog::AddTabPage
 }
 #endif
 
-void SfxTabDialog::AddTabPage( int id, layout::TabPage* page, bool )
-{
-    layoutPages[id] = page;
-}
-
-void SfxTabDialog::AddTabPage( int id, layout::CreateTabPage func, bool )
-{
-    layoutPageFuncs[id] = func;
-}
-
 // -----------------------------------------------------------------------
 
 void SfxTabDialog::RemoveTabPage( USHORT nId )
@@ -1188,28 +1203,7 @@ IMPL_LINK( SfxTabDialog, OkHdl, Button *, EMPTYARG )
 bool SfxTabDialog::PrepareLeaveCurrentPage()
 {
 	USHORT const nId = aTabCtrl.GetCurPageId();
-
-    if ( layout::TabPage* page = layoutPages[nId] )
-    {
-		if ( !pSet )
-			return page->DeactivatePage( 0 );
-
-        SfxItemSet set( *pSet->GetPool(), pSet->GetRanges() );
-
-        if ( page->HasExchangeSupport() )
-            return page->DeactivatePage( &set );
-
-		int result = page->DeactivatePage( 0 );
-        if ( result & layout::TabPage::LEAVE_PAGE && set.Count() )
-        {
-            pExampleSet->Put( set );
-            pOutSet->Put( set );
-		}
-		return result;
-    }
-
-	SfxTabPage* pPage = (SfxTabPage*)aTabCtrl.GetTabPage( nId );
-
+	SfxTabPage* pPage = dynamic_cast<SfxTabPage*> (aTabCtrl.GetTabPage( nId ));
 	bool bEnd = !pPage;
 
 	if ( pPage )
@@ -1281,12 +1275,6 @@ IMPL_LINK( SfxTabDialog, ResetHdl, Button *, EMPTYARG )
 
 {
     int const nId = aTabCtrl.GetCurPageId();
-    if ( layout::TabPage* page = layoutPages[nId] )
-    {
-        page->Reset( *pSet );
-        return 0;
-    }
-
 	Data_Impl* pDataObject = Find( *pImpl->pData, nId );
 	DBG_ASSERT( pDataObject, "Id nicht bekannt" );
 
@@ -1381,6 +1369,11 @@ IMPL_LINK( SfxTabDialog, BaseFmtHdl, Button *, EMPTYARG )
 
 // -----------------------------------------------------------------------
 
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+#define tabControlWindow pTabCtrl->GetWindow ()
+#else /* !ENABLE_LAYOUT_SFX_TABDIALOG */
+#define tabControlWindow pTabCtrl
+#endif /* !ENABLE_LAYOUT_SFX_TABDIALOG */
 
 IMPL_LINK( SfxTabDialog, ActivatePageHdl, TabControl *, pTabCtrl )
 
@@ -1397,42 +1390,22 @@ IMPL_LINK( SfxTabDialog, ActivatePageHdl, TabControl *, pTabCtrl )
 
 {
 	int const nId = pTabCtrl->GetCurPageId();
-    OSL_TRACE ("%s: page: %d", __FUNCTION__, nId );
-    if ( !layoutPages[nId] )
-        if ( layout::CreateTabPage func = layoutPageFuncs[nId] )
-        {
-            layout::TabPage::global_parent = pTabCtrl;
-            layout::TabPage* page = (func) (pTabCtrl, *pSet);
-            AddTabPage( nId, page );
-        }
-    if ( layout::TabPage* page = layoutPages[nId] )
-    {
-        TabPage *vclPage = page->GetTabPage();
-        TabControl *dialog = pTabCtrl;
-        page->Reset( page->GetItemSet() );
-        page->ActivatePage();
-        dialog->SetTabPage( nId, vclPage );
-
-        Size tabSize = vclPage->GetSizePixel();
-        Size dialogSize = dialog->GetTabPageSizePixel();
-        OSL_TRACE ("%s: tabSize: %d, %d", __FUNCTION__, tabSize.Width(), tabSize.Height() );
-        OSL_TRACE ("%s: dialogSize: %d, %d", __FUNCTION__, dialogSize.Width(), dialogSize.Height() );
-
-        dialog->SetTabPageSizePixel( dialogSize );
-        return 0;
-    }
     
 	DBG_ASSERT( pImpl->pData->Count(), "keine Pages angemeldet" );
 	SFX_APP();
 
 	// Tab Page schon da?
-	SfxTabPage* pTabPage = (SfxTabPage *)pTabCtrl->GetTabPage( nId );
+	SfxTabPage* pTabPage = dynamic_cast<SfxTabPage*> (pTabCtrl->GetTabPage( nId ));
 	Data_Impl* pDataObject = Find( *pImpl->pData, nId );
 	DBG_ASSERT( pDataObject, "Id nicht bekannt" );
 
 	// ggf. TabPage erzeugen:
 	if ( !pTabPage )
 	{
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+        if (layout SfxTabPage* inplug = dynamic_cast<layout SfxTabPage*> (pTabPage))
+            layout::TabPage::global_parent = pTabCtrl->GetWindow ();
+#endif
 		const SfxItemSet* pTmpSet = 0;
 
 		if ( pSet )
@@ -1444,13 +1417,16 @@ IMPL_LINK( SfxTabDialog, ActivatePageHdl, TabControl *, pTabCtrl )
 		}
 
 		if ( pTmpSet && !pDataObject->bOnDemand )
-			pTabPage = (pDataObject->fnCreatePage)( pTabCtrl, *pTmpSet );
+			pTabPage = (pDataObject->fnCreatePage)( tabControlWindow, *pTmpSet );
 		else
 			pTabPage = (pDataObject->fnCreatePage)
-							( pTabCtrl, *CreateInputItemSet( nId ) );
+							( tabControlWindow, *CreateInputItemSet( nId ) );
 		DBG_ASSERT( NULL == pDataObject->pTabPage, "create TabPage more than once" );
 		pDataObject->pTabPage = pTabPage;
+
+#if !ENABLE_LAYOUT_SFX_TABDIALOG
 		pDataObject->pTabPage->SetTabDialog( this );
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
 		SvtViewOptions aPageOpt( E_TABPAGE, String::CreateFromInt32( pDataObject->nId ) );
 		String sUserData;
 		Any aUserItem = aPageOpt.GetUserItem( USERITEM_NAME );
@@ -1459,23 +1435,48 @@ IMPL_LINK( SfxTabDialog, ActivatePageHdl, TabControl *, pTabCtrl )
 			sUserData = String( aTemp );
 		pTabPage->SetUserData( sUserData );
 		Size aSiz = pTabPage->GetSizePixel();
+
+#if ENABLE_LAYOUT
+		Size optimalSize = pTabPage->GetOptimalSize (WINDOWSIZE_MINIMUM);
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+        if (layout SfxTabPage* inplug = dynamic_cast<layout SfxTabPage*> (pTabPage))
+        {
+            if (optimalSize.Height () && optimalSize.Width ())
+            {
+                optimalSize.Width () = optimalSize.Width ();
+                optimalSize.Height () = optimalSize.Height () + 40;
+            }
+        }
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+        if (optimalSize.Height () > 0 && optimalSize.Width () > 0 )
+            aSiz = optimalSize;
+#endif /* ENABLE_LAYOUT */
+
 		Size aCtrlSiz = pTabCtrl->GetTabPageSizePixel();
 		// Gr"o/se am TabControl nur dann setzen, wenn < als TabPage
 		if ( aCtrlSiz.Width() < aSiz.Width() ||
 			 aCtrlSiz.Height() < aSiz.Height() )
+        {
 			pTabCtrl->SetTabPageSizePixel( aSiz );
+        }
+
 		PageCreated( nId, *pTabPage );
 
 		if ( pDataObject->bOnDemand )
 			pTabPage->Reset( (SfxItemSet &)pTabPage->GetItemSet() );
 		else
 			pTabPage->Reset( *pSet );
+
 		pTabCtrl->SetTabPage( nId, pTabPage );
 	}
 	else if ( pDataObject->bRefresh )
 		pTabPage->Reset( *pSet );
 	pDataObject->bRefresh = FALSE;
 
+#if ENABLE_LAYOUT_SFX_TABDIALOG
+    pTabCtrl->GetPagePos (nId);
+#endif /* ENABLE_LAYOUT_SFX_TABDIALOG */
+
 	if ( pExampleSet )
 		pTabPage->ActivatePage( *pExampleSet );
 	BOOL bReadOnly = pTabPage->IsReadOnly();
@@ -1497,12 +1498,9 @@ IMPL_LINK( SfxTabDialog, DeactivatePageHdl, TabControl *, pTabCtrl )
 */
 
 {
-    if ( /* layout::TabPage* p = */ layoutPages[pTabCtrl->GetCurPageId()] )
-        return SfxTabPage::LEAVE_PAGE;
-
 	USHORT nId = pTabCtrl->GetCurPageId();
 	SFX_APP();
-	SfxTabPage *pPage = (SfxTabPage*)pTabCtrl->GetTabPage( nId );
+	SfxTabPage *pPage = dynamic_cast<SfxTabPage*> (pTabCtrl->GetTabPage( nId ));
 	DBG_ASSERT( pPage, "keine aktive Page" );
 #ifdef DBG_UTIL
 	Data_Impl* pDataObject = Find( *pImpl->pData, pTabCtrl->GetCurPageId() );
@@ -1796,3 +1794,4 @@ long SfxTabDialog::Notify( NotifyEvent& rNEvt )
     return TabDialog::Notify( rNEvt );
 }
 
+END_NAMESPACE_LAYOUT_SFX_TABDIALOG
diff --git a/sfx2/source/layout/factory.cxx b/sfx2/source/layout/factory.cxx
new file mode 100644
index 0000000..407e626
--- /dev/null
+++ sfx2/source/layout/factory.cxx
@@ -0,0 +1,71 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: tabdlg.cxx,v $
+ * $Revision: 1.38 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#include <com/sun/star/awt/WindowAttribute.hpp>
+#include <sal/types.h>
+#include <sfx2/basedlgs.hxx>
+#include <toolkit/awt/vclxtoolkit.hxx>
+#include <toolkit/awt/vclxdialog.hxx>
+
+using rtl::OUString;
+
+class SfxModelessDialog_unprotect : public SfxModelessDialog
+{
+public:
+    SfxModelessDialog_unprotect (Window *parent, WinBits attributes);
+};
+
+SFX2_DLLPUBLIC SfxBindings* sfx2_global_bindings;
+SFX2_DLLPUBLIC SfxChildWindow* sfx2_global_child;
+
+SfxModelessDialog_unprotect::SfxModelessDialog_unprotect (Window *parent, WinBits bits)
+    : SfxModelessDialog (sfx2_global_bindings, sfx2_global_child, parent, bits)
+{
+}
+
+extern "C"
+{
+
+SAL_DLLPUBLIC_EXPORT Window* CreateWindow (VCLXWindow** vcl, OUString const& name, Window* parent, long& attributes)
+{
+	Window* window = 0;
+	if (0)
+    {
+        ;
+    }
+	else if (name.equalsAscii ("sfxmodelessdialog"))
+	{
+		window = new SfxModelessDialog_unprotect (parent, ImplGetWinBits (attributes, 0));
+		*vcl = new layoutimpl::VCLXDialog ();
+	}
+    return window;
+}
+
+}
diff --git a/sfx2/source/layout/makefile.mk b/sfx2/source/layout/makefile.mk
new file mode 100644
index 0000000..3dc2346
--- /dev/null
+++ sfx2/source/layout/makefile.mk
@@ -0,0 +1,59 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: makefile.mk,v $
+#
+# $Revision: 1.7 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJNAME=sfx2
+TARGET=layout
+ENABLE_EXCEPTIONS=true
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+# --- Files --------------------------------------------------------
+
+.IF "$(ENABLE_LAYOUT)" == "TRUE"
+
+SLOFILES = \
+	$(SLO)$/factory.obj \
+	$(SLO)$/sfxdialog.obj \
+	$(SLO)$/sfxtabpage.obj \
+	$(SLO)$/sfxtabdialog.obj \
+#
+
+.ENDIF # ENABLE_LAYOUT == TRUE
+
+# --- Tagets -------------------------------------------------------
+
+.INCLUDE :  target.mk
+
diff --git a/sfx2/source/layout/sfxdialog.cxx b/sfx2/source/layout/sfxdialog.cxx
new file mode 100644
index 0000000..32e11c9
--- /dev/null
+++ sfx2/source/layout/sfxdialog.cxx
@@ -0,0 +1,41 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: tabdlg.cxx,v $
+ * $Revision: 1.38 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#include <sfx2/layout.hxx>
+
+namespace layout
+{
+
+SfxDialog::SfxDialog(::Window* parent, char const* xml_file, char const* id, SfxBindings*, SfxChildWindow*)
+    : Dialog (parent, xml_file, id)
+{
+}
+
+} // end namespace layout
diff --git a/sfx2/source/layout/sfxtabdialog.cxx b/sfx2/source/layout/sfxtabdialog.cxx
new file mode 100644
index 0000000..bf7c521
--- /dev/null
+++ sfx2/source/layout/sfxtabdialog.cxx
@@ -0,0 +1,75 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: tabdlg.cxx,v $
+ * $Revision: 1.38 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+// MARKER(update_precomp.py): autogen include statement, do not remove
+#include "precompiled_sfx2.hxx"
+
+#include <limits.h>
+#include <stdlib.h>
+#include <vcl/msgbox.hxx>
+#include <svtools/viewoptions.hxx>
+
+#define _SVSTDARR_USHORTS
+#include <svtools/svstdarr.hxx>
+
+#include "appdata.hxx"
+#include "sfxtypes.hxx"
+#include <sfx2/minarray.hxx>
+#include <sfx2/tabdlg.hxx>
+#include <sfx2/viewfrm.hxx>
+#include <sfx2/app.hxx>
+#include "sfxresid.hxx"
+#include "sfxhelp.hxx"
+#include <sfx2/ctrlitem.hxx>
+#include <sfx2/bindings.hxx>
+#include <sfx2/sfxdlg.hxx>
+#include <sfx2/itemconnect.hxx>
+#include <layout/layout.hxx>
+
+//FIXME
+#include <toolkit/awt/vclxwindow.hxx>
+#include <com/sun/star/awt/PosSize.hpp>
+
+using namespace ::com::sun::star;
+namespace layout
+{
+using namespace ::com::sun::star;
+}
+
+#include <sfx2/layout.hxx>
+#include <layout/layout-pre.hxx>
+
+#undef TabPage
+#undef SfxTabPage
+#undef SfxTabDialog
+
+#define TabDlgCmpUS_Impl layout__TabDlgCmpUS_Impl
+
+#include "../dialog/tabdlg.cxx"
diff --git a/sfx2/source/layout/sfxtabpage.cxx b/sfx2/source/layout/sfxtabpage.cxx
new file mode 100644
index 0000000..6b7806d
--- /dev/null
+++ sfx2/source/layout/sfxtabpage.cxx
@@ -0,0 +1,81 @@
+/*************************************************************************
+* DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+* 
+* Copyright 2008 by Sun Microsystems, Inc.
+*
+* OpenOffice.org - a multi-platform office productivity suite
+*
+* $RCSfile: code,v $
+*
+* $Revision: 1.3 $
+*
+* This file is part of OpenOffice.org.
+*
+* OpenOffice.org is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License version 3
+* only, as published by the Free Software Foundation.
+*
+* OpenOffice.org is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Lesser General Public License version 3 for more details
+* (a copy is included in the LICENSE file that accompanied this code).
+*
+* You should have received a copy of the GNU Lesser General Public License
+* version 3 along with OpenOffice.org.  If not, see
+* <http://www.openoffice.org/license.html>
+* for a copy of the LGPLv3 License.
+************************************************************************/
+
+#include <sfx2/layout.hxx>
+
+#include <com/sun/star/awt/PosSize.hpp>
+#include <toolkit/awt/vclxwindow.hxx>
+
+namespace layout
+{
+
+using namespace ::com::sun::star;
+
+#if 0
+SfxTabDialog::SfxTabDialog (::Window *parent, char const* xml_file, char const* id, ResId const& res_id, SfxItemSet const* set)
+    : ::SfxTabDialog (parent, res_id, set)
+    , InPlug (parent, xml_file, id)
+{
+}
+
+::Window* SfxTabDialog::GetParent () const
+{
+    return Window::GetParent ();
+}
+
+void SfxTabDialog::FreeResource ()
+{
+    //::Window::FreeResource ();
+}
+#endif
+
+SfxTabPage::SfxTabPage (::Window *parent, char const* xml_file, char const* id, SfxItemSet const* set)
+    : ::SfxTabPage (parent, 0, *set)
+    , InPlug (parent, xml_file, id)
+{
+    dynamic_cast< ::Window* > (this)->SetComponentInterface (GetVCLXWindow ());
+}
+
+::Window* SfxTabPage::GetParent () const
+{
+    return Window::GetParent ();
+}
+
+void SfxTabPage::FreeResource ()
+{
+    //::Window::FreeResource ();
+}
+
+Size SfxTabPage::GetOptimalSize (WindowSizeType) const
+{
+    awt::Size s = GetVCLXWindow()->getMinimumSize ();
+    return Size (s.Width, s.Height);
+}
+
+} // end namespace layout
diff --git a/sfx2/util/makefile.mk b/sfx2/util/makefile.mk
index 3e08460..b987fae 100644
--- sfx2/util/makefile.mk
+++ sfx2/util/makefile.mk
@@ -61,6 +61,10 @@ LIB1FILES=  $(SLB)$/appl.lib		\
             $(SLB)$/bastyp.lib		\
             $(SLB)$/config.lib
 
+.IF "$(ENABLE_LAYOUT)" == "TRUE"
+LIB1FILES += $(SLB)$/layout.lib
+.ENDIF # ENABLE_LAYOUT == TRUE
+
 HELPIDFILES=\
 			..\inc\sfx2\sfxsids.hrc	\
 			..\source\inc\helpid.hrc
