--- sfx2/source/dialog/filedlghelper.cxx	2008-04-15 16:18:43.000000000 +0200
+++ sfx2/source/dialog/filedlghelper.cxx	2008-05-08 08:03:49.000000000 +0200
@@ -1960,11 +1960,7 @@
 
 		try
 		{
-			aValue = xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_LINK, 0 );
 			sal_Bool bValue = sal_False;
-			aValue >>= bValue;
-			aUserData.SetToken( 0, ' ', String::CreateFromInt32( (sal_Int32) bValue ) );
-
 			aValue = xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_PREVIEW, 0 );
 			bValue = sal_False;
 			aValue >>= bValue;
@@ -2089,14 +2085,6 @@
 		{
 			try
 			{
-				// respect the last "insert as link" state
-				sal_Bool bLink = (sal_Bool) aUserData.GetToken( 0, ' ' ).ToInt32();
-                if ( !xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_LINK, 0 ).hasValue() )
-                {
-				    aValue <<= bLink;
-				    xDlg->setValue( ExtendedFilePickerElementIds::CHECKBOX_LINK, 0, aValue );
-                }
-
 				// respect the last "show preview" state
 				sal_Bool bShowPreview = (sal_Bool) aUserData.GetToken( 1, ' ' ).ToInt32();
                 if  ( !xDlg->getValue( ExtendedFilePickerElementIds::CHECKBOX_PREVIEW, 0 ).hasValue() )
