Index: filter/source/bf_offwrp/makefile.mk
===================================================================
RCS file: /cvs/framework/binfilter/filter/source/bf_offwrp/makefile.mk,v
retrieving revision 1.15
diff -u -r1.15 makefile.mk
--- binfilter/filter/source/bf_offwrp/makefile.mk	10 Apr 2008 21:16:58 -0000	1.15
+++ binfilter/filter/source/bf_offwrp/makefile.mk	14 May 2008 16:19:23 -0000
@@ -57,7 +57,6 @@
 SHL1STDLIBS=    \
 				$(FWELIB)			\
 				$(BFBASICLIB) 		\
-				$(BFSO3LIB) 			\
 				$(SJLIB) 			\
 				$(TKLIB) 			\
 				$(BFSVTOOLLIB) 		\
@@ -75,6 +74,7 @@
 				$(BFSDLIB)          \
 				$(BFSCLIB)          \
 				$(BFSVXLIB)         \
+				$(BFSO3LIB) 			\
 				$(UNOTOOLSLIB)		\
 				$(COMPHELPERLIB)	\
 				$(LEGACYSMGRLIB)
