--- sw/source/ui/uiview/view2.cxx	2008-04-10 23:49:38.000000000 +0200
+++ sw/source/ui/uiview/view2.cxx	2008-05-07 18:39:44.000000000 +0200
@@ -61,6 +61,7 @@
 #include <caption.hxx>
 #include <svtools/PasswordHelper.hxx>
 #include <svtools/urihelper.hxx>
+#include <svtools/miscopt.hxx>
 #include <sfx2/passwd.hxx>
 #include <sfx2/sfxdlg.hxx>
 #include <sfx2/filedlghelper.hxx>
@@ -68,6 +69,7 @@
 #include <svx/langitem.hxx>
 #include <svx/viewlayoutitem.hxx>
 #include <svx/zoomslideritem.hxx>
+#include <svx/linkwarn.hxx>
 #include <svx/htmlmode.hxx>
 #ifndef _APP_HXX //autogen
 #include <vcl/svapp.hxx>
@@ -462,6 +464,14 @@
                     rReq.AppendItem( SfxStringItem( FN_PARAM_2, sGraphicFormat ) );
                 rReq.AppendItem( SfxBoolItem( FN_PARAM_1, bAsLink ) );
             }
+
+            // really store as link only?
+            if( bAsLink && SvtMiscOptions().ShowLinkWarningDialog() )
+            {
+                SvxLinkWarningDialog aWarnDlg(GetWindow(),pFileDlg->GetPath());
+                if( aWarnDlg.Execute() != RET_OK )
+                    bAsLink=sal_False; // don't store as link
+            }                        
         }
 
         SwWrtShell& rSh = GetWrtShell();
--- sc/source/ui/drawfunc/fuins1.cxx	2008-04-11 00:23:00.000000000 +0200
+++ sc/source/ui/drawfunc/fuins1.cxx	2008-05-07 18:40:07.000000000 +0200
@@ -42,8 +42,10 @@
 #include <svx/svdpage.hxx>
 #include <svx/svdpagv.hxx>
 #include <svx/svdview.hxx>
+#include <svx/linkwarn.hxx>
 #include <svtools/filter.hxx>
 #include <svtools/stritem.hxx>
+#include <svtools/miscopt.hxx>
 #include <vcl/msgbox.hxx>
 #include <tools/urlobj.hxx>
 #include <avmedia/mediawindow.hxx>
@@ -297,6 +299,14 @@
 				String aFilterName = aDlg.GetCurrentFilter();
 				BOOL bAsLink = aDlg.IsAsLink();
 
+                // really store as link only?
+                if( bAsLink && SvtMiscOptions().ShowLinkWarningDialog() )
+                {
+                    SvxLinkWarningDialog aWarnDlg(pWin,aFileName);
+                    if( aWarnDlg.Execute() != RET_OK )
+                        bAsLink = sal_False; // don't store as link
+                }
+
 				lcl_InsertGraphic( aGraphic, aFileName, aFilterName, bAsLink, FALSE, pViewSh, pWindow, pView );
 
 				//	append items for recording
--- sd/source/ui/func/fuinsert.cxx	2008-04-10 22:20:15.000000000 +0200
+++ sd/source/ui/func/fuinsert.cxx	2008-05-07 18:13:46.000000000 +0200
@@ -51,9 +51,11 @@
 #include <sfx2/request.hxx>
 #include <svtools/globalnameitem.hxx>
 #include <svtools/pathoptions.hxx>
+#include <svtools/miscopt.hxx>
 #include <svx/pfiledlg.hxx>
 #include <svx/impgrf.hxx>
 #include <svx/dialogs.hrc>
+#include <svx/linkwarn.hxx>
 #include <svx/linkmgr.hxx>
 #include <svx/svdetc.hxx>
 #include <avmedia/mediawindow.hxx>
@@ -178,7 +180,15 @@
 
 				if(pGrafObj && aDlg.IsAsLink())
 				{
-					// store link only?
+                    // really store as link only?
+                    if( SvtMiscOptions().ShowLinkWarningDialog() )
+                    {
+                        SvxLinkWarningDialog aWarnDlg(mpWindow,aDlg.GetPath());
+                        if( aWarnDlg.Execute() != RET_OK )
+                            return; // don't store as link
+                    }
+                        
+					// store as link
 					String aFltName(aDlg.GetCurrentFilter());
 					String aPath(aDlg.GetPath());
 					pGrafObj->SetGraphicLink(aPath, aFltName);
