Index: inc/unxlngi6.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi6.mk,v
retrieving revision 1.45
diff -u -r1.45 unxlngi6.mk
--- solenv/inc/unxlngi6.mk	10 Apr 2008 17:49:25 -0000	1.45
+++ solenv/inc/unxlngi6.mk	15 May 2008 13:29:59 -0000
@@ -146,7 +146,7 @@
 LINKC*=$(CC)
 
 # default linker flags
-LINKFLAGSDEFS*=-Wl,-z,defs
+LINKFLAGSDEFS*=-Wl,-z,defs -Wl,--as-needed
 LINKFLAGSRUNPATH_URELIB=-Wl,-rpath,\''$$ORIGIN'\'
 LINKFLAGSRUNPATH_UREBIN=-Wl,-rpath,\''$$ORIGIN/../lib:$$ORIGIN'\'
     #TODO: drop $ORIGIN once no URE executable is also shipped in OOo
Index: inc/unxlngx6.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngx6.mk,v
retrieving revision 1.15
diff -u -r1.15 unxlngx6.mk
--- solenv/inc/unxlngx6.mk	10 Apr 2008 17:52:03 -0000	1.15
+++ solenv/inc/unxlngx6.mk	15 May 2008 13:30:00 -0000
@@ -154,7 +154,7 @@
 LINKC*=$(CC)
 
 # default linker flags
-LINKFLAGSDEFS*=-Wl,-z,defs
+LINKFLAGSDEFS*=-Wl,-z,defs -Wl,--as-needed
 LINKFLAGSRUNPATH_URELIB=-Wl,-rpath,\''$$ORIGIN'\'
 LINKFLAGSRUNPATH_UREBIN=-Wl,-rpath,\''$$ORIGIN/../lib:$$ORIGIN'\'
     #TODO: drop $ORIGIN once no URE executable is also shipped in OOo
--- unodevtools/source/skeletonmaker/makefile.mk	11 Apr 2008 12:33:12 -0000	1.5
+++ unodevtools/source/skeletonmaker/makefile.mk	22 May 2008 15:46:32 -0000
@@ -51,8 +51,9 @@
 
 APP1DEPN= $(OUT)$/lib$/$(UNODEVTOOLSLIBDEPN) $(SOLARLIBDIR)$/$(CODEMAKERLIBDEPN) \
 	$(SOLARLIBDIR)$/$(COMMONCPPLIBDEPN) $(SOLARLIBDIR)$/$(COMMONJAVALIBDEPN)
-APP1STDLIBS = $(REGLIB) $(SALLIB) $(SALHELPERLIB) $(CPPULIB) $(CPPUHELPERLIB) \
-	$(UNODEVTOOLSLIBST) $(CODEMAKERLIBST) $(COMMONCPPLIBST) $(COMMONJAVALIBST)
+APP1STDLIBS =\
+	$(UNODEVTOOLSLIBST) $(CODEMAKERLIBST) $(COMMONCPPLIBST) $(COMMONJAVALIBST)\
+	$(REGLIB) $(CPPULIB) $(CPPUHELPERLIB) $(SALHELPERLIB) $(SALLIB)
 
 OBJFILES = $(APP1OBJS)
 
diff --git svx/util/makefile.mk svx/util/makefile.mk
index 4598098..d5de0b6 100644
--- svx/util/makefile.mk
+++ svx/util/makefile.mk
@@ -172,7 +172,6 @@ SHL2STDLIBS= \
             $(I18NISOLANGLIB) \
 			$(COMPHELPERLIB) \
 			$(UCBHELPERLIB)	\
-			$(CPPUHELPERLIB)	\
 			$(CPPULIB) \
             $(VOSLIB) \
             $(SALLIB) \
@@ -181,6 +181,7 @@
 .IF "$(GUI)"=="WNT"
 SHL2STDLIBS+= \
              $(SHLWAPILIB) \
+             $(CPPUHELPERLIB) \
              $(ADVAPI32LIB)
 .ENDIF # WNT
 
