Index: source/pyprov/mailmerge.py
===================================================================
RCS file: /cvs/framework/scripting/source/pyprov/mailmerge.py,v
retrieving revision 1.8
diff -u -r1.8 mailmerge.py
--- scripting/source/pyprov/mailmerge.py	6 Dec 2007 10:57:39 -0000	1.8
+++ scripting/source/pyprov/mailmerge.py	2 Dec 2008 14:09:24 -0000
@@ -83,7 +83,9 @@
 		if dbg:
 			print >> sys.stderr, connectiontype
 		if connectiontype == 'Ssl':
+			self.server.ehlo()
 			self.server.starttls()
+			self.server.ehlo()
 
 		user = xAuthenticator.getUserName()
 		password = xAuthenticator.getPassword()
