diff -p -up officecfg/registry/data/org/openoffice/Office/Common.xcu.orig2 officecfg/registry/data/org/openoffice/Office/Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu.orig2	2008-09-02 14:36:58.000000000 -0400
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2008-09-02 14:38:41.000000000 -0400
@@ -555,6 +555,10 @@
     <prop oor:name="UseSystemPrintDialog"  install:module="macosx">
       <value>true</value>
     </prop>
+    <!-- Set toolbar icon size to small as default -->
+    <prop oor:name="SymbolSet" oor:type="xs:short">
+      <value>0</value>
+    </prop>
   </node>
   <node oor:name="Save">
     <node oor:name="Document">
