diff --git basic/source/runtime/makefile.mk basic/source/runtime/makefile.mk
index 4f354ad..fdec029 100644
--- basic/source/runtime/makefile.mk
+++ basic/source/runtime/makefile.mk
@@ -61,10 +61,14 @@ SLOFILES=	\
 SLOFILES+=	$(SLO)$/win.obj
 .ENDIF
 
-.IF "$(GUI)$(CPU)" == "WNTI"
+.IF "$(GUI)$(COM)$(CPU)" == "WNTMSCI"
 SLOFILES+=	$(SLO)$/wnt.obj
 .ENDIF
 
+.IF "$(GUI)$(COM)$(CPU)" == "WNTGCCI"
+SLOFILES+=	$(SLO)$/wnt-mingw.obj
+.ENDIF
+
 .IF "$(GUI)$(CPU)" == "OS2I"
 #FIXME SLOFILES+= $(SLO)$/os2.obj
 .ENDIF
@@ -80,3 +84,10 @@ EXCEPTIONSFILES=$(SLO)$/step0.obj	\
 
 .INCLUDE :  target.mk
 
+$(SLO)$/%.obj: %.s
+#kendy: Cut'n'paste from bridges/source/cpp_uno/mingw_intel/makefile.mk
+#cmc: Ideally --noexecstack would be in operations, but with #i51385# pyuno
+#remote bridgeing breaks
+#    $(CC) -Wa,--noexecstack -c -o $(SLO)$/$(@:b).o $<
+    $(CC) -c -o $(SLO)$/$(@:b).obj $<
+    touch $@
diff --git basic/source/runtime/wnt-mingw.s basic/source/runtime/wnt-mingw.s
new file mode 100644
index 0000000..b42c527
--- /dev/null
+++ basic/source/runtime/wnt-mingw.s
@@ -0,0 +1,97 @@
+#*************************************************************************
+#
+# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+# 
+# Copyright 2008 by Sun Microsystems, Inc.
+#
+# OpenOffice.org - a multi-platform office productivity suite
+#
+# $RCSfile: wnt.asm,v $
+#
+# $Revision: 1.4 $
+#
+# This file is part of OpenOffice.org.
+#
+# OpenOffice.org is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# only, as published by the Free Software Foundation.
+#
+# OpenOffice.org is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY# without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License version 3 for more details
+# (a copy is included in the LICENSE file that accompanied this code).
+#
+# You should have received a copy of the GNU Lesser General Public License
+# version 3 along with OpenOffice.org.  If not, see
+# <http://www.openoffice.org/license.html>
+# for a copy of the LGPLv3 License.
+#
+#*************************************************************************
+
+#    Anmerkungen
+#        Direktaufruf von C- und PASCAL-Routinen, Windows und OS/2
+#
+# Inhalt:
+# type = CallXXX (far *pProc, char *pStack, short nStack)
+#
+# Kopie des Basic-Stacks (nStack Bytes) auf den C-Stack
+# und Aufruf der Prozedur.
+
+		.intel_syntax
+		#.386
+
+#_TEXT		SEGMENT	DWORD PUBLIC 'CODE' USE32
+
+		#ASSUME 	CS:_TEXT
+
+.globl		_CallINT
+.globl		_CallLNG
+.globl		_CallDBL
+.globl		_CallSTR
+.globl		_CallFIX
+
+_CallINT:
+_CallLNG:
+_CallDBL:
+_CallSTR:
+
+_CallFIX:	PUSH	EBP
+		MOV	EBP,ESP
+		PUSH	ESI
+		PUSH	EDI
+
+		PUSH	ECX
+		PUSH	EDX
+
+		MOV	DX,DS
+		MOVZX	EAX,WORD PTR [EBP+16] # EAX == nStack
+		SUB	ESP,EAX       # Stack um nStack Bytes vergroessern
+		MOV	EDI,ESP
+		MOV	AX,SS
+		MOV	ES,AX	      # ES:EDI = Startadresse des fuer
+				      # Parameter reservierten Stackbereichs
+		MOV	ESI,[EBP+12]  # DS:ESI == pStack
+
+		MOVZX	ECX,WORD PTR [EBP+16] # ECX == nStack
+		SHR	ECX,1
+		CLD
+		JCXZ	$1
+	REP	MOVSW		      # Stack uebernehmen
+$1:		MOV	DS,DX
+		CALL	DWORD PTR [EBP+8]	# Aufruf der Prozedur
+		# CLI # unter NT nicht erlaubt (privileged instruction)
+		MOV	ESP,EBP
+		SUB	ESP,16	      # wegen gepushter Register
+				      # (ESI, EDI)
+		# STI
+		POP	EDX
+		POP	ECX
+		POP	EDI
+		POP	ESI
+		POP	EBP
+		RET	12
+
+#_TEXT		ENDS
+
+		#END
diff --git config_office/configure.in config_office/configure.in
index 37be0fb..f8a7d81 100644
--- config_office/configure.in
+++ config_office/configure.in
@@ -5101,7 +5101,7 @@ AC_SUBST(CYGWIN_PATH)
 dnl ***************************************
 dnl testing ml.exe assembler path
 dnl ***************************************
-if test "$_os" = "WINNT"; then
+if test "$_os" = "WINNT" -a "$WITH_MINGWIN" != "yes"; then
   AC_MSG_CHECKING([ml.exe assembler path])
   if test -n "$with_asm_home"; then
     with_asm_home=`cygpath -u "$with_asm_home"`
