diff --git setup_native/source/win32/customactions/languagepacks/makefile.mk setup_native/source/win32/customactions/languagepacks/makefile.mk
index 6e999e2..5e82525 100644
--- setup_native/source/win32/customactions/languagepacks/makefile.mk
+++ setup_native/source/win32/customactions/languagepacks/makefile.mk
@@ -44,7 +44,9 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
 CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 #Disable precompiled header
 CDEFS+=-Dnot_used_define_to_disable_pch
@@ -60,8 +62,11 @@ SLOFILES =	$(SLO)$/lngpckinsthelper.obj \
 
 STDSHL+=	\
 	$(ADVAPI32LIB)\
-	$(MSILIB)\
-	$(LIBSTLPORTST)								
+	$(MSILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+STDSHL+=$(LIBSTLPORTST)								
+.ENDIF
 
 SHL1OBJS = $(SLOFILES)               \
     $(SLO)$/seterror.obj
diff --git setup_native/source/win32/customactions/patch/makefile.mk setup_native/source/win32/customactions/patch/makefile.mk
index 974ad3f..911da23 100755
--- setup_native/source/win32/customactions/patch/makefile.mk
+++ setup_native/source/win32/customactions/patch/makefile.mk
@@ -43,7 +43,9 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
 CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 #Disable precompiled header
 CDEFS+=-Dnot_used_define_to_disable_pch
@@ -60,9 +62,12 @@ SLOFILES = \
 STDSHL += \
 	$(ADVAPI32LIB)\
 	$(MSILIB)\
-	$(LIBSTLPORTST) \
         $(SHELL32LIB)								
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
+STDSHL += $(LIBSTLPORTST)
+.ENDIF
+
 .IF "$(COM)"=="GCC"
 STDSHL+=	\
 	$(KERNEL32LIB)\
diff --git setup_native/source/win32/customactions/quickstarter/makefile.mk setup_native/source/win32/customactions/quickstarter/makefile.mk
index 5fa7b03..2063b17 100644
--- setup_native/source/win32/customactions/quickstarter/makefile.mk
+++ setup_native/source/win32/customactions/quickstarter/makefile.mk
@@ -46,7 +46,10 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
 CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
+
 UWINAPILIB=
 
 # --- Files --------------------------------------------------------
@@ -56,8 +59,11 @@ UWINAPILIB=
 STDSHL += \
 	$(ADVAPI32LIB)\
 	$(SHELL32LIB)\
-	$(MSILIB)\
-	$(LIBSTLPORTST)								
+	$(MSILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+STDSHL += $(LIBSTLPORTST)								
+.ENDIF
 
 .IF "$(COM)"=="GCC"
 STDSHL+=	\
diff --git setup_native/source/win32/customactions/reg4allmsdoc/makefile.mk setup_native/source/win32/customactions/reg4allmsdoc/makefile.mk
index fdebb13..482cdf4 100644
--- setup_native/source/win32/customactions/reg4allmsdoc/makefile.mk
+++ setup_native/source/win32/customactions/reg4allmsdoc/makefile.mk
@@ -42,7 +42,10 @@ USE_DEFFILE=TRUE
 
 .INCLUDE : settings.mk
 
-CFLAGS+=-DUNICODE -D_UNICODE -D_STLP_USE_STATIC_LIB
+CFLAGS+=-DUNICODE -D_UNICODE
+.IF "$(USE_SYSTEM_STL)" != "YES"
+CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 # --- Files --------------------------------------------------------
 
@@ -57,8 +60,11 @@ SHL1STDLIBS=	$(KERNEL32LIB)\
 				$(ADVAPI32LIB)\
 				$(SHELL32LIB)\
 				$(MSILIB)\
-				$(SHLWAPILIB)\
-				$(LIBSTLPORTST)								
+				$(SHLWAPILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+SHL1STDLIBS+=$(LIBSTLPORTST)								
+.ENDIF
 
 SHL1LIBS = $(SLB)$/$(TARGET).lib 
 
diff --git setup_native/source/win32/customactions/reg4msdoc/makefile.mk setup_native/source/win32/customactions/reg4msdoc/makefile.mk
index 2daaca5..c3d6b42 100644
--- setup_native/source/win32/customactions/reg4msdoc/makefile.mk
+++ setup_native/source/win32/customactions/reg4msdoc/makefile.mk
@@ -43,7 +43,10 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
-CFLAGS+=-DUNICODE -D_UNICODE -D_STLP_USE_STATIC_LIB
+CFLAGS+=-DUNICODE -D_UNICODE
+.IF "$(USE_SYSTEM_STL)" != "YES"
+CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 # --- Files --------------------------------------------------------
 
@@ -70,8 +73,11 @@ SHL1STDLIBS=	$(KERNEL32LIB)\
 				$(ADVAPI32LIB)\
 				$(SHELL32LIB)\
 				$(MSILIB)\
-				$(SHLWAPILIB)\
-				$(LIBSTLPORTST)								
+				$(SHLWAPILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+SHL1STDLIBS+=$(LIBSTLPORTST)								
+.ENDIF
 
 SHL1LIBS = $(SLB)$/$(TARGET).lib 
 
diff --git setup_native/source/win32/customactions/relnotes/makefile.mk setup_native/source/win32/customactions/relnotes/makefile.mk
index 4bea6ce..fef2912 100644
--- setup_native/source/win32/customactions/relnotes/makefile.mk
+++ setup_native/source/win32/customactions/relnotes/makefile.mk
@@ -44,7 +44,11 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
-CDEFS+=-DUNICODE -D_STLP_USE_STATIC_LIB
+CDEFS+=-DUNICODE
+.IF "$(USE_SYSTEM_STL)" != "YES"
+CDEFS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
+
 UWINAPILIB=
 
 # --- Files --------------------------------------------------------
@@ -56,8 +60,11 @@ SLOFILES =	$(SLO)$/relnotes.obj
 STDSHL+=	\
 	$(ADVAPI32LIB)\
 	$(SHELL32LIB)\
-	$(MSILIB)\
-	$(LIBSTLPORTST)								
+	$(MSILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+STDSHL+=$(LIBSTLPORTST)								
+.ENDIF
 
 SHL1LIBS = $(SLB)$/$(TARGET).lib 
 
diff --git setup_native/source/win32/customactions/sellang/makefile.mk setup_native/source/win32/customactions/sellang/makefile.mk
index 2b414a6..6a874ac 100644
--- setup_native/source/win32/customactions/sellang/makefile.mk
+++ setup_native/source/win32/customactions/sellang/makefile.mk
@@ -42,7 +42,9 @@ USE_DEFFILE=TRUE
 
 .INCLUDE : settings.mk
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
 CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 # --- Files --------------------------------------------------------
 
@@ -51,12 +53,15 @@ UWINAPILIB=
 SLOFILES = \
 	$(SLO)$/sellang.obj
 
-SHL1STDLIBS=	kernel32.lib\
-				user32.lib\
-				advapi32.lib\
-				shell32.lib\
-				msi.lib\
-				$(LIBSTLPORTST)								
+SHL1STDLIBS=	$(KERNEL32LIB)\
+				$(USER32LIB)\
+				$(ADVAPI32LIB)\
+				$(SHELL32LIB)\
+				$(MSILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+SHL1STDLIBS+=$(LIBSTLPORTST)								
+.ENDIF
 
 SHL1LIBS = $(SLB)$/$(TARGET).lib 
 
diff --git setup_native/source/win32/customactions/shellextensions/makefile.mk setup_native/source/win32/customactions/shellextensions/makefile.mk
index 3c4887f..c41f8fd 100644
--- setup_native/source/win32/customactions/shellextensions/makefile.mk
+++ setup_native/source/win32/customactions/shellextensions/makefile.mk
@@ -43,7 +43,9 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
 CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 #Disable precompiled header
 CDEFS+=-Dnot_used_define_to_disable_pch
@@ -71,9 +73,12 @@ SLOFILES = \
 STDSHL += \
 	$(ADVAPI32LIB)\
 	$(MSILIB)\
-	$(LIBSTLPORTST)\
         $(SHELL32LIB)								
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
+STDSHL += $(LIBSTLPORTST)
+.ENDIF
+
 .IF "$(COM)"=="GCC"
 STDSHL+=	\
 	$(KERNEL32LIB)\
diff --git setup_native/source/win32/customactions/tools/makefile.mk setup_native/source/win32/customactions/tools/makefile.mk
index f247f0f..ee972b1 100644
--- setup_native/source/win32/customactions/tools/makefile.mk
+++ setup_native/source/win32/customactions/tools/makefile.mk
@@ -44,7 +44,9 @@ MINGW_NODLL=YES
 
 .INCLUDE : settings.mk
 
+.IF "$(USE_SYSTEM_STL)" != "YES"
 CFLAGS+=-D_STLP_USE_STATIC_LIB
+.ENDIF
 
 # --- Files --------------------------------------------------------
 
@@ -59,8 +61,11 @@ SHL1OBJS = $(SLOFILES)
 
 STDSHL+=	\
 	$(ADVAPI32LIB)\
-	$(MSILIB)\
-	$(LIBSTLPORTST)								
+	$(MSILIB)
+
+.IF "$(USE_SYSTEM_STL)" != "YES"
+STDSHL+=$(LIBSTLPORTST)								
+.ENDIF
 
 #SHL1LIBS = $(SLB)$/$(TARGET).lib 
 
