diff --git fpicker/source/win32/filepicker/VistaFilePickerImpl.cxx fpicker/source/win32/filepicker/VistaFilePickerImpl.cxx
index f3b6754..f6846fa 100644
--- fpicker/source/win32/filepicker/VistaFilePickerImpl.cxx
+++ fpicker/source/win32/filepicker/VistaFilePickerImpl.cxx
@@ -547,7 +547,11 @@ void VistaFilePickerImpl::impl_sta_setFiltersOnDialog()
     aLock.clear();
     // <- SYNCHRONIZED
 
+#ifdef __MINGW32__
+	iDialog->QueryInterface(IID_IFileDialog, (void**)(&iCustomize));
+#else
 	iDialog.query(&iCustomize);
+#endif
 
 	COMDLG_FILTERSPEC	*pFilt = &lFilters[0];
     iDialog->SetFileTypes(lFilters.size(), pFilt/*&lFilters[0]*/);
