diff --git shell/inc/internal/thumbviewer.hxx shell/inc/internal/thumbviewer.hxx
index 0028307..8cb9be3 100644
--- shell/inc/internal/thumbviewer.hxx
+++ shell/inc/internal/thumbviewer.hxx
@@ -37,6 +37,11 @@
 #endif
 #include <objidl.h>
 #include <shlobj.h>
+#ifdef __MINGW32__
+#include <algorithm>
+using std::min;
+using std::max;
+#endif
 #include <gdiplus.h>
 #if defined _MSC_VER
 #pragma warning(pop)
diff --git shell/source/win32/shlxthandler/ooofilt/ooofilt.cxx shell/source/win32/shlxthandler/ooofilt/ooofilt.cxx
index 9cba529..f9ea776 100644
--- shell/source/win32/shlxthandler/ooofilt/ooofilt.cxx
+++ shell/source/win32/shlxthandler/ooofilt/ooofilt.cxx
@@ -470,7 +470,7 @@ SCODE STDMETHODCALLTYPE COooFilter::GetText(ULONG * pcwcBuffer, WCHAR * awcBuffe
             return FILTER_E_NO_MORE_TEXT;
         }
         // Copy UNICODE characters in chunk buffer to output UNICODE buffer
-        ULONG ulToCopy = min( *pcwcBuffer, m_ulUnicodeBufferLen - m_ulUnicodeCharsRead );
+        ULONG ulToCopy = std::min( *pcwcBuffer, m_ulUnicodeBufferLen - m_ulUnicodeCharsRead );
         ZeroMemory(awcBuffer, sizeof(awcBuffer));    
         wmemcpy( awcBuffer, m_pwsBuffer.c_str() + m_ulUnicodeCharsRead, ulToCopy );
         m_ulUnicodeCharsRead += ulToCopy;
diff --git shell/source/win32/shlxthandler/util/iso8601_converter.cxx shell/source/win32/shlxthandler/util/iso8601_converter.cxx
index e5cb981..836d9e7 100644
--- shell/source/win32/shlxthandler/util/iso8601_converter.cxx
+++ shell/source/win32/shlxthandler/util/iso8601_converter.cxx
@@ -149,17 +149,35 @@ std::wstring iso8601_duration_to_local_duration(const std::wstring& iso8601durat
         hours = buff;
     }
        
-    std::wostringstream oss;
-
 #if defined(_MSC_VER) && defined(_M_X64)        
+    std::wostringstream oss;
     oss << std::setw(2) << std::setfill(wchar_t('0')) << hours   << L":" <<
            std::setw(2) << std::setfill(wchar_t('0')) << minutes << L":" << 
            std::setw(2) << std::setfill(wchar_t('0')) << seconds;
+    return oss.str();
+#elif defined( __MINGW32__ )
+#define ADD_AS_PREFILLED( st, out ) \
+    if ( st.length() == 0 ) \
+        out += L"00"; \
+    else if ( st.length() == 1 ) \
+        out += L"0"; \
+    out += st;
+
+    std::wstring result;
+    ADD_AS_PREFILLED( hours, result )
+    result += L":";
+    ADD_AS_PREFILLED( minutes, result )
+    result += L":";
+    ADD_AS_PREFILLED( seconds, result )
+
+    return result;
+#undef ADD_AS_PREFILLED
 #else
+    std::wostringstream oss;
     oss << std::setw(2) << std::setfill('0') << hours   << L":" <<
            std::setw(2) << std::setfill('0') << minutes << L":" << 
            std::setw(2) << std::setfill('0') << seconds;
-#endif    
     return oss.str();
+#endif    
 }
 
diff --git shell/source/win32/simplemail/senddoc.cxx shell/source/win32/simplemail/senddoc.cxx
index 937c139..5671d85 100644
--- shell/source/win32/simplemail/senddoc.cxx
+++ shell/source/win32/simplemail/senddoc.cxx
@@ -50,6 +50,7 @@
 #include <iostream>
 #include <vector>
 #include <sstream>
+#include <stdexcept>
 
 #if OSL_DEBUG_LEVEL > 2
     void dumpParameter();
