diff --git openssl/makefile.mk openssl/makefile.mk
index f76b998..b8120ed 100644
--- openssl/makefile.mk
+++ openssl/makefile.mk
@@ -106,14 +106,14 @@ OUT2INC += include/openssl/*
 PATCH_FILE_NAME=opensslmingw.patch
 .IF "$(USE_MINGW)" == "cygwin"
 CONFIGURE_ACTION=$(PERL) configure
-CONFIGURE_FLAGS=mingw shared 
+CONFIGURE_FLAGS=shared mingw
 INSTALL_ACTION=mv libcrypto.a libcrypto_static.a && mv libcrypto.dll.a libcrypto.a && mv libssl.a libssl_static.a && mv libssl.dll.a libssl.a
 OUT2LIB = libcrypto_static.*
 OUT2LIB += libssl_static.*
 OUT2LIB += libcrypto.*
 OUT2LIB += libssl.*
-OUT2BIN = ssleay32.dll
-OUT2BIN += libeay32.dll
+OUT2BIN = ssleay32*.dll
+OUT2BIN += cryptoeay32*.dll
 .ELSE
 CONFIGURE_ACTION=
 BUILD_ACTION=cmd /c "ms\mingw32"
diff --git openssl/opensslmingw.patch openssl/opensslmingw.patch
index 32c1105..1ae70c0 100755
--- openssl/opensslmingw.patch
+++ openssl/opensslmingw.patch
@@ -35,6 +35,23 @@
   	$(LINK_SO_O)
   link_a.cygwin:
   	@ $(CALC_VERSIONS); \
+***************
+*** 264,270 ****
+  	SHLIB_SOVER=-$(LIBVERSION); \
+  	ALLSYMSFLAGS='-Wl,--whole-archive'; \
+  	NOALLSYMSFLAGS='-Wl,--no-whole-archive'; \
+! 	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared $$base -Wl,-Bsymbolic -Wl,--out-implib,lib$(LIBNAME).dll.a"; \
+  	[ -f apps/$$SHLIB$$SHLIB_SUFFIX ] && rm apps/$$SHLIB$$SHLIB_SUFFIX; \
+  	[ -f test/$$SHLIB$$SHLIB_SUFFIX ] && rm test/$$SHLIB$$SHLIB_SUFFIX; \
+  	$(LINK_SO_A) || exit 1; \
+--- 264,270 ----
+  	SHLIB_SOVER=-$(LIBVERSION); \
+  	ALLSYMSFLAGS='-Wl,--whole-archive'; \
+  	NOALLSYMSFLAGS='-Wl,--no-whole-archive'; \
+! 	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared $$base -Wl,-Bsymbolic -Wl,--export-all-symbols -Wl,--out-implib,lib$(LIBNAME).dll.a"; \
+  	[ -f apps/$$SHLIB$$SHLIB_SUFFIX ] && rm apps/$$SHLIB$$SHLIB_SUFFIX; \
+  	[ -f test/$$SHLIB$$SHLIB_SUFFIX ] && rm test/$$SHLIB$$SHLIB_SUFFIX; \
+  	$(LINK_SO_A) || exit 1; \
 *** misc/openssl-0.9.8g/ms/mingw32.bat	Fri May  5 22:19:33 2006
 --- misc/build/openssl-0.9.8g/ms/mingw32.bat	Mon Feb  4 20:11:50 2008
 ***************
diff --git openssl/prj/d.lst openssl/prj/d.lst
index 227e211..bd7bc14 100644
--- openssl/prj/d.lst
+++ openssl/prj/d.lst
@@ -4,6 +4,8 @@ mkdir: %_DEST%\inc%_EXT%\external\openssl
 ..\%__SRC%\inc\*.h %_DEST%\inc%_EXT%\external\openssl
 ..\%__SRC%\lib\ssleay32.lib %_DEST%\lib%_EXT%\ssleay32.lib
 ..\%__SRC%\bin\ssleay32.dll %_DEST%\bin%_EXT%\ssleay32.dll
+..\%__SRC%\bin\cryptoeay32-0.9.8.dll %_DEST%\bin%_EXT%\cryptoeay32.dll
+..\%__SRC%\bin\ssleay32-0.9.8.dll %_DEST%\bin%_EXT%\ssleay32.dll
 ..\%__SRC%\lib\libeay32.lib %_DEST%\lib%_EXT%\libeay32.lib
 ..\%__SRC%\bin\libeay32.dll %_DEST%\bin%_EXT%\libeay32.dll
 ..\%__SRC%\lib\libcrypto.a %_DEST%\lib%_EXT%\libcrypto.a
