--- configure.in.mozff	2007-01-30 19:59:21.000000000 +0100
+++ configure.in	2007-02-01 12:43:06.000000000 +0100
@@ -3615,7 +3615,7 @@
     SYSTEM_MOZILLA=YES
     AC_MSG_CHECKING([which Mozilla flavour to use])
     if test -n "$with_firefox" && test "$with_firefox" != "no"; then
-      MOZ_FLAVOUR=firefox
+      MOZ_FLAVOUR=mozilla-firefox
     elif test -n "$with_seamonkey" && test "$with_seamonkey" != "no"; then
       MOZ_FLAVOUR=seamonkey
     elif test -n "$with_xulrunner" && test "$with_xulrunner" != "no"; then
