--- config_office/configure.in
+++ config_office/configure.in
@@ -1758,6 +1758,12 @@
 		  MIDL_PATH="$vstest/Common7/Tools/Bin"
       fi
 	fi
+	if test -z "$MIDL_PATH" ; then
+		psdktest=`./oowintool --psdk-home`
+		if test -x "$psdktest/Bin/midl.exe" ; then
+			MIDL_PATH="$psdktest/Bin"
+		fi
+	fi
 	if test ! -x "$MIDL_PATH/midl.exe"; then
 		AC_MSG_ERROR([midl.exe not found. Make sure it's in the path or use --with-midl-path])
 	fi
@@ -1809,7 +1809,7 @@
 	    fi
 	    if test -z "$FRAME_HOME" ; then
           frametest=`./oowintool --dotnetsdk-dir`
-          if test -x "$frametest/lib/mscoree.lib"; then
+          if test -f "$frametest/lib/mscoree.lib"; then
              FRAME_HOME="$frametest"
           fi
 	    fi
@@ -5046,10 +5046,14 @@
 fi
 
 dnl ===================================================================
-dnl Windows builds - attempt to auto-copy required dlls into external/
+dnl Windows builds - use oowintool to copy CRT dlls and manifest
 dnl ===================================================================
 if test "$_os" = "WINNT"; then
-       ./oowintool --msvc-copy-dlls ../external/msvcp
+       if ./oowintool --msvc-copy-dlls ../external/msvcp ; then
+          :
+       else
+          AC_MSG_ERROR([oowintool failed to copy CRT])
+       fi
 fi
 
 dnl ===================================================================
@@ -5067,90 +5067,6 @@
    fi
 fi
 
-dnl ===================================================================
-dnl Windows builds with .NET 2003 needs msvcp71.dll/msvcr71.dll in external/msvcp71
-dnl ===================================================================
-if test "$_os" = "WINNT" -a "$COMEX" = "10" ; then
-   AC_MSG_CHECKING([for msvcp71.dll/msvcr71.dll])
-   if test -x ../external/msvcp71/msvcp71.dll -a -x ../external/msvcp71/msvcr71.dll; then
-      AC_MSG_RESULT([found])
-   else
-      MSVCPPATH=`/bin/find "$COMPATH/../.." -iname msvcp71.dll | head -n 1`
-      MSVCRPATH=`/bin/find "$COMPATH/../.." -iname msvcr71.dll | head -n 1`
-      if test -n "$MSVCPPATH" -a -n "$MSVCRPATH"; then
-         cp "$MSVCPPATH" ../external/msvcp71/ && chmod +x ../external/msvcp71/msvcp71.dll && MSVCPCOPY="OK"
-         cp "$MSVCRPATH" ../external/msvcp71/ && chmod +x ../external/msvcp71/msvcr71.dll && MSVCRCOPY="OK"
-      fi
-      if test -z "$MSVCPCOPY" -o -z "$MSVCRCOPY"; then
-         AC_MSG_ERROR([msvcp71.dll and/or msvcr71.dll are/is missing in the default location.
-These dlls are part of the .NET installation and should be found in a directory
-similar to:
-"c:\\Program Files\\Microsoft Visual Studio .NET 2003\\Visual C++ .NET Professional - English\\"
-As the automatic detection fails please copy the files to external/msvcp71/.])
-      else
-         AC_MSG_RESULT([found and copied])
-      fi
-   fi
-fi
-
-dnl ===================================================================
-dnl Windows builds with Visual C++ 2005 needs msvc{p,r}80.dll in external/msvcp80
-dnl ===================================================================
-if test "$_os" = "WINNT" -a "$COMEX" = "11"; then
-   AC_MSG_CHECKING([for msvcp80.dll/msvcr80.dll])
-   if test -x ../external/msvcp80/msvcp80.dll -a -x ../external/msvcp80/msvcr80.dll; then
-      AC_MSG_RESULT([found])
-   else
-      MSVCPPATH=`/bin/find "$COMPATH/../.." -iname msvcp80.dll | head -n 1`
-      MSVCRPATH=`/bin/find "$COMPATH/../.." -iname msvcr80.dll | head -n 1`
-      if test -n "$MSVCPPATH" -a -n "$MSVCRPATH"; then
-         cp "$MSVCPPATH" ../external/msvcp80/ && chmod +x ../external/msvcp80/msvcp80.dll && MSVCPCOPY="OK"
-         cp "$MSVCRPATH" ../external/msvcp80/ && chmod +x ../external/msvcp80/msvcr80.dll && MSVCRCOPY="OK"
-      fi
-      if test -z "$MSVCPCOPY" -o -z "$MSVCRCOPY"; then
-         AC_MSG_ERROR([msvcp80.dll and/or msvcr80.dll are/is missing in the default location.
-These dlls are part of the VS installation and should be found in a directory
-similar to:
-"c:\\Program Files\\<no-idea>"
-As the automatic detection fails please copy the files to external/msvcp80/.])
-      else
-         AC_MSG_RESULT([found and copied])
-      fi
-   fi
-fi
-
-dnl ===================================================================
-dnl Windows builds with Visual Studio 2008 needs msvc{m,p,r}90.dll in external/msvcp90
-dnl ===================================================================
-if test "$_os" = "WINNT"; then
-   if test "$COMEX" = "12"; then
-   AC_MSG_CHECKING([for msvcm90.dll/msvcp90.dll/msvcr90.dll/Microsoft.VC90.CRT.manifest])
-   if test -x ../external/msvcp90/msvcm90.dll -a -x ../external/msvcp90/msvcp90.dll -a -x ../external/msvcp90/msvcr90.dll -a -x ../external/msvcp90/Microsoft.VC90.CRT.manifest; then
-      AC_MSG_RESULT([found])
-   else
-      MSVCMPATH=`/bin/find "$COMPATH/../.." -iname msvcm90.dll | head -n 1`
-      MSVCPPATH=`/bin/find "$COMPATH/../.." -iname msvcp90.dll | head -n 1`
-      MSVCRPATH=`/bin/find "$COMPATH/../.." -iname msvcr90.dll | head -n 1`
-      MSVCMANPATH=`/bin/find "$COMPATH/../.." -iname Microsoft.VC90.CRT.manifest | head -n 1`
-      if test -n "$MSVCMPATH" -a -n "$MSVCPPATH" -a -n "$MSVCRPATH" -a -n "$MSVCMANPATH"; then
-         cp "$MSVCMPATH" ../external/msvcp90/ && chmod +x ../external/msvcp90/msvcm90.dll && MSVCMCOPY="OK"
-         cp "$MSVCPPATH" ../external/msvcp90/ && chmod +x ../external/msvcp90/msvcp90.dll && MSVCPCOPY="OK"
-         cp "$MSVCRPATH" ../external/msvcp90/ && chmod +x ../external/msvcp90/msvcr90.dll && MSVCRCOPY="OK"
-         cp "$MSVCMANPATH" ../external/msvcp90/ && chmod +x ../external/msvcp90/Microsoft.VC90.CRT.manifest && MSVCMANCOPY="OK"
-      fi
-      if test -z "$MSVCMCOPY" -o -z "$MSVCPCOPY" -o -z "$MSVCRCOPY" -o -z "$MSVCMANCOPY"; then
-         AC_MSG_ERROR([Any or all of msvcm90.dll, msvcp90.dll, msvcr90.dll and Microsoft.VC90.CRT.manifest are missing in the default location.
-These dlls are part of the VS installation and should be found in a directory
-similar to:
-"C:\\Program Files\\Microsoft Visual Studio 9.0\\VC\\redist\\x86\\Microsoft.VC90.CRT\\"
-As the automatic detection fails please copy the files to external/msvcp90/.])
-      else
-         AC_MSG_RESULT([found and copied])
-      fi
-   fi
-fi
-fi
-
 dnl ===================================================================
 dnl Windows builds - attempt to auto-copy required instmsiX.exe into external/
 dnl ===================================================================
