--- config_office/oowintool.BAK	2008-04-25 13:12:44.546875000 +0300
+++ config_office/oowintool	2008-04-25 16:45:16.093750000 +0300
@@ -95,12 +95,16 @@
 sub print_psdk_home()
 {
     my $value;
-    $value = reg_get_value ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/MicrosoftSDK/Directories/Install Dir');
+    $value = reg_get_value ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/Microsoft SDKs/Windows/v6.1/InstallationFolder');
     if (!defined $value)
     {
-        $key = reg_find_key ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/MicrosoftSDK/InstalledSDKs/*/Install Dir');
-        $value = reg_get_value ($key);
-    }
+		$value = reg_get_value ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/MicrosoftSDK/Directories/Install Dir');
+		if (!defined $value)
+		{
+			$key = reg_find_key ('HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/MicrosoftSDK/InstalledSDKs/*/Install Dir');
+			$value = reg_get_value ($key);
+		}
+	}
     defined $value || die "psdk not found";
 
     print cygpath ($value, 'w', $output_format);
@@ -128,14 +132,14 @@
     'ver' => '7.1',
     'key' => 'Microsoft/VisualStudio/7.1/Setup/VC/ProductDir',
     'instmsi_path' => '../Common7/Tools/Deployment/MsiRedist',
-    'dll_path' => '../SDK/v1.1/Bin',
+    'dll_path' => '../Visual Studio .NET Professional 2003 - English',
     'dll_suffix' => '71'
 );
 my %msvs_net_2003 = (
     'ver' => '7.1',
     'key' => 'Microsoft/VisualStudio/7.1/Setup/VS/ProductDir',
     'instmsi_path' => 'Common7/Tools/Deployment/MsiRedist',
-    'dll_path' => 'Visual Studio .NET Professional 2003 - English', # testme ...
+    'dll_path' => 'Visual Studio .NET Professional 2003 - English',
     'dll_suffix' => '71'
 );
 my %msvs_net_2003_ea = (
@@ -159,10 +163,24 @@
     'dll_path' => '../SDK/v2.0/Bin',
     'dll_suffix' => '80'
 );
+my %msvs_2008 = (
+    'ver' => '9.0',
+    'key' => 'Microsoft/VisualStudio/9.0/Setup/VS/ProductDir',
+    'instmsi_path' => '?',
+    'dll_path' => 'VC/redist/x86/Microsoft.VC90.CRT',
+    'dll_suffix' => '90'
+);
+my %msvc_2008 = (
+    'ver' => '9.0',
+    'key' => 'Microsoft/VisualStudio/9.0/Setup/VC/ProductDir',
+    'instmsi_path' => '?',
+    'dll_path' => 'redist/x86/Microsoft.VC90.CRT',
+    'dll_suffix' => '90'
+);
 
 sub find_msvs()
 {
-    my @ms_versions = ( \%msvs_express_2005, \%msvs_net_2003_ea, \%msvs_net_2003, \%msvs_net_2002, \%msvc6 );
+    my @ms_versions = ( \%msvs_2008, \%msvs_express_2005, \%msvs_net_2003_ea, \%msvs_net_2003, \%msvs_net_2002, \%msvc6 );
 
     for $ver (@ms_versions)
     {
@@ -177,7 +195,7 @@
 
 sub find_msvc()
 {
-    my @ms_versions = ( \%msvc_express_2005, \%msvc_net_2003, \%msvc_net_2002, \%msvc6 );
+    my @ms_versions = ( \%msvc_2008, \%msvc_express_2005, \%msvc_net_2003, \%msvc_net_2002, \%msvc6 );
 
     for $ver (@ms_versions)
     {
@@ -242,6 +260,7 @@
     -f "$src/$fname" || die "can't find $src";
     -d $dest || die "no directory $dest";
 
+	print STDERR "Copying $src/$fname to $dest\n";
     copy ("$src/$fname", $dest) || die "copy failed: $!";
     chmod (0755, "$dest/$fname") || die "failed to set dll executable: $!";
 }
@@ -257,7 +276,7 @@
     $srcdir = (cygpath ($ver->{'product_dir'}, 'w', 'u') . '/' . 
 		  $ver->{$checkpath});
     -d $srcdir && return $ver;
-    return "";
+    return undef;
 }
 
 sub msvc_copy_dlls($)
