--- config_office/configure.in.OLD	2005-11-21 02:16:33.437250000 +0200
+++ config_office/configure.in	2005-11-23 05:12:58.296875000 +0200
@@ -75,6 +75,11 @@
                           distributions where AFM files or TrueType Fonts
                           are known to be available.
 ],,)
+AC_ARG_WITH(agfa-monotype-fonts,
+[  --without-agfa-monotype-fonts Remove proprietary Agfa Monotype fonts
+                                (even if present) in the openoffice.org installation
+                                set.
+],,with_agfa_monotype_fonts="yes")
 AC_ARG_ENABLE(epm,
 [  --disable-epm           OO.o includes self-packaging code, that requires
                           epm, however epm is useless for large scale
@@ -4317,6 +4322,18 @@
 fi
 AC_SUBST(WITH_FONTS)
 
+AC_MSG_CHECKING([whether to include Agfa Monotype fonts])
+if test "$with_agfa_monotype_fonts" != "no" -a -e ../agfa_monotype_fonts/download/agfa-monotype-fonts.tar.gz; then
+  AC_MSG_RESULT([yes])
+  WITH_AGFA_MONOTYPE_FONTS=YES
+  SCPDEFS="$SCPDEFS -DWITH_AGFA_MONOTYPE_FONTS"
+  BUILD_TYPE="$BUILD_TYPE AGFA_MONOTYPE_FONTS"
+else
+  AC_MSG_RESULT([no])
+  WITH_AGFA_MONOTYPE_FONTS=NO
+fi
+AC_SUBST(WITH_AGFA_MONOTYPE_FONTS)
+
 dnl ===================================================================
 dnl Test whether to include ppds
 dnl ===================================================================
--- agfa_monotype_fonts/makefile.mk.OLD	1970-01-01 02:00:00.000000000 +0200
+++ agfa_monotype_fonts/makefile.mk	2005-11-23 05:07:41.015625000 +0200
@@ -0,0 +1,78 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+
+PRJ=.
+
+PRJNAME=agfa_monotype_fonts
+TARGET=agfa_monotype_fonts
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :	settings.mk
+
+# --- Files --------------------------------------------------------
+
+TARFILE_NAME=agfa-monotype-fonts
+TARFILE_ROOTDIR=agfa-monotype-fonts
+
+PATCH_FILE_NAME=
+
+
+CONFIGURE_DIR=
+CONFIGURE_ACTION=
+
+BUILD_DIR=
+BUILD_ACTION=
+BUILD_FLAGS=
+
+# --- Targets ------------------------------------------------------
+
+all: \
+	$(MISC)$/remove_build.flag \
+	ALLTAR
+
+.INCLUDE : set_ext.mk
+.INCLUDE : target.mk
+
+.IF "$(WITH_AGFA_MONOTYPE_FONTS)"!="NO"
+.INCLUDE : tg_ext.mk
+.ENDIF
+
+# Since you never know what will be in a patch (for example, it may already
+# patch at configure level), we remove the entire package directory if a patch
+# is newer.
+$(MISC)$/remove_build.flag : $(PATCH_FILE_NAME)
+	$(REMOVE_PACKAGE_COMMAND)
+	+$(TOUCH) $(MISC)$/remove_build.flag
--- agfa_monotype_fonts/prj/d.lst.OLD	1970-01-01 02:00:00.000000000 +0200
+++ agfa_monotype_fonts/prj/d.lst	2005-11-23 04:52:41.531250000 +0200
@@ -0,0 +1,2 @@
+mkdir: %_DEST%\pck%_EXT%
+..\%__SRC%\misc\build\agfa-monotype-fonts\*.ttf %_DEST%\pck%_EXT%\*.ttf
--- agfa_monotype_fonts/prj/build.lst.OLD	1970-01-01 02:00:00.000000000 +0200
+++ agfa_monotype_fonts/prj/build.lst	2005-11-22 12:05:42.843750000 +0200
@@ -0,0 +1,3 @@
+bs	agfa_monotype_fonts : solenv NULL
+bs	agfa_monotype_fonts		usr1	-	all	amt_mkout NULL
+bs	agfa_monotype_fonts		nmake	-	all	amt_fonts NULL
--- scp2/source/ooo/file_font_ooo.scp.OLD	2005-09-09 04:37:10.000000000 +0300
+++ scp2/source/ooo/file_font_ooo.scp	2005-11-23 05:08:35.375000000 +0200
@@ -151,3 +151,31 @@
 STD_FONT_FILE( gid_File_Fnt_LiberationSerif_BoldItalic, LiberationSerif-BoldItalic.ttf, Liberation Serif Bold Italic) 
 #endif
 #endif
+
+#ifdef WNT
+#ifdef WITH_AGFA_MONOTYPE_FONTS
+STD_FONTWIN_FILE( gid_File_Fnt_albw, albw.ttf, AlbanyAMT)
+STD_FONTWIN_FILE( gid_File_Fnt_albwb, albwb.ttf, AlbanyAMT Bold)
+STD_FONTWIN_FILE( gid_File_Fnt_albwbi, albwbi.ttf, AlbanyAMT Bold Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_albwi, albwi.ttf, AlbanyAMT Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_andybol, andybol_.ttf, AndyMT Bold)
+STD_FONTWIN_FILE( gid_File_Fnt_andyreg, andyreg_.ttf, AndyMT)
+STD_FONTWIN_FILE( gid_File_Fnt_ans, ans_____.ttf, Andale Sans)
+STD_FONTWIN_FILE( gid_File_Fnt_ansb, ansb____.ttf, Andale Sans Bold)
+STD_FONTWIN_FILE( gid_File_Fnt_ansbi, ansbi___.ttf, Andale Sans Bold Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_ansi, ansi____.ttf, Andale Sans Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_bl, bl______.ttf, BellMT)
+STD_FONTWIN_FILE( gid_File_Fnt_blb, blb_____.ttf, BellMT Bold)
+STD_FONTWIN_FILE( gid_File_Fnt_blbi, blbi____.ttf, BellMT Bold Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_bli, bli_____.ttf, BellMT Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_cumbwb, cumbwb__.ttf, CumberlandAMT Bold)
+STD_FONTWIN_FILE( gid_File_Fnt_cumbwbi, cumbwbi_.ttf, CumberlandAMT Bold Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_cumbwi, cumbwi__.ttf, CumberlandAMT Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_cumbwr, cumbwr__.ttf, CumberlandAMT)
+STD_FONTWIN_FILE( gid_File_Fnt_mtsorts, mtsorts_.ttf, Monotype Sorts)
+STD_FONTWIN_FILE( gid_File_Fnt_thowb, thowb___.ttf, ThorndaleAMT Bold)
+STD_FONTWIN_FILE( gid_File_Fnt_thowbi, thowbi__.ttf, ThorndaleAMT Bold Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_thowi, thowi___.ttf, ThorndaleAMT Italic)
+STD_FONTWIN_FILE( gid_File_Fnt_thowr, thowr___.ttf, ThorndaleAMT)
+#endif
+#endif
--- postprocess/prj/build.lst	2005-11-23 05:56:34.703125000 +0200
+++ postprocess/prj/build.lst	2005-11-23 05:56:48.812500000 +0200
@@ -1,4 +1,4 @@
-po	postprocess	::	accessibility automation basctl bean fondu BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io JAVAINSTALLER2:javainstaller2 lingucomponent MATHMLDTD:MathMLDTD MSFONTEXTRACT:msfontextract ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins sccomp scp2 scripting sd ODK:sdk_oo setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top unodevtools JFREEREPORT:jfreereport REPORTDESIGN:reportdesign SDEXT:sdext SWEXT:swext writerfilter oox NULL
+po	postprocess	::	accessibility automation basctl bean fondu BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io JAVAINSTALLER2:javainstaller2 lingucomponent MATHMLDTD:MathMLDTD MSFONTEXTRACT:msfontextract ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins sccomp scp2 scripting sd ODK:sdk_oo setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts AGFA_MONOTYPE_FONTS:agfa_monotype_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top unodevtools JFREEREPORT:jfreereport REPORTDESIGN:reportdesign SDEXT:sdext SWEXT:swext writerfilter oox NULL
 po	postprocess			    	usr1	-	all	po_mkout NULL
 po	postprocess\checkxml		nmake	-	all	po_checkxml NULL
 po	postprocess\checkdeliver	nmake	-	all	po_checkdlv NULL
