--- o3tl/inc/o3tl/cow_wrapper.hxx	11 Apr 2008 11:54:52 -0000	1.6
+++ o3tl/inc/o3tl/cow_wrapper.hxx	20 Aug 2008 12:19:49 -0000	1.6.10.1
@@ -205,6 +205,7 @@ void cow_wrapper_client::queryUnmodified
     public:
         typedef T        value_type;
         typedef T*       pointer;
+        typedef const T* const_pointer;
         typedef MTPolicy mt_policy;
 
         /** Default-construct wrapped type instance
@@ -278,11 +279,11 @@ void cow_wrapper_client::queryUnmodified
         
         pointer		      operator->()       { return &make_unique(); }
         value_type&       operator*()        { return make_unique(); }
-        const pointer     operator->() const { return &m_pimpl->m_value; }
+        const_pointer     operator->() const { return &m_pimpl->m_value; }
         const value_type& operator*()  const { return m_pimpl->m_value; }
 
         pointer           get()       { return &make_unique(); }
-        const pointer	  get() const { return &m_pimpl->m_value; }
+        const_pointer	  get() const { return &m_pimpl->m_value; }
 
         /// true, if both cow_wrapper internally share the same object
         bool              same_object( const cow_wrapper& rOther ) const
--- basegfx/source/inc/hommatrixtemplate.hxx	10 Apr 2008 18:55:46 -0000	1.18
+++ basegfx/source/inc/hommatrixtemplate.hxx	15 Sep 2008 12:57:57 -0000
@@ -588,7 +588,7 @@ namespace basegfx
                 testLastLine();
             }
 
-            bool isEqual(const ImplHomMatrixTemplate& rMat)
+            bool isEqual(const ImplHomMatrixTemplate& rMat) const
             {
                 const sal_uInt16 nMaxLine(
                     sal::static_int_cast<sal_uInt16>((mpLine || rMat.mpLine) ? RowSize : (RowSize - 1)) );
