--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.old	2007-08-23 20:15:04.000000000 +0200
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2007-08-24 16:15:30.000000000 +0200
@@ -1524,7 +1524,7 @@ Dymamic border coloring means that when 
 						<author>MBA</author>
 						<desc>Specifies the templates originate from these folders and sub-folders.</desc>
 					</info>
-					<value oor:separator=":">$(insturl)/share/template/$(vlang):$(userurl)/template</value>
+					<value oor:separator=";">file:///usr/share/templates/ooo/common;file:///usr/share/templates/ooo/$(vlang);file:///usr/share/templates/ooo2/common;file:///usr/share/templates/ooo2/$(vlang);$(insturl)/share/template/common;$(insturl)/share/template/$(vlang);$(userurl)/template</value>
 				</prop>
 				<prop oor:name="UIConfig" oor:type="oor:string-list">
 					<info>
@@ -1672,7 +1672,7 @@ Dymamic border coloring means that when 
 					<info>
 						<desc>Specifies the default directory where all provided templates are located in folders and sub-folders.</desc>
 					</info>
-					<value oor:separator=":">$(insturl)/share/template/$(vlang):$(userurl)/template</value>
+					<value oor:separator=";">file:///usr/share/templates/ooo/common;file:///usr/share/templates/ooo/$(vlang);file:///usr/share/templates/ooo2/common;file:///usr/share/templates/ooo2/$(vlang);$(insturl)/share/template/common;$(insturl)/share/template/$(vlang);$(userurl)/template</value>
 				</prop>
 				<prop oor:name="UIConfig" oor:type="oor:string-list">
 					<info>
--- officecfg/registry/data/org/openoffice/Office/Common.xcu.old	2007-08-23 20:15:03.000000000 +0200
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2007-08-24 16:18:32.000000000 +0200
@@ -536,4 +536,6 @@
       </prop>
     </node>
   </node>
+  <!-- FIXME: This comment is needed to change a checksum or so.
+              Then the updated list of templates paths is read from Common.xcs-->
 </oor:component-data>
