--- sfx2/source/doc/objxtor.cxx.old	2009-04-06 16:42:00.000000000 +0000
+++ sfx2/source/doc/objxtor.cxx	2009-04-06 16:42:09.000000000 +0000
@@ -419,7 +419,16 @@ SfxObjectShell::~SfxObjectShell()
     }
 
     if ( pImp->bOwnsStorage && pImp->m_xDocStorage.is() )
-    	pImp->m_xDocStorage->dispose();
+    {
+	try
+	{
+    	    pImp->m_xDocStorage->dispose();
+	}
+	catch( Exception& )
+	{
+            /* what to do ? */
+	}
+    }
 
 	if ( pMedium )
 	{
