--- scp2/source/ooo/common_brand.scp.original	2008-05-30 10:52:54.000000000 +0200
+++ scp2/source/ooo/common_brand.scp	2008-05-30 11:32:40.000000000 +0200
@@ -965,9 +965,9 @@
     Key = "UpdateURL";
   #if defined(BUILD_SPECIAL)
   #ifdef WNT
-    Value = "${UPDATEURL}";
+    Value = "http://ooop.sourceforge.net/update/ooop.php";
   #else  // defined WNT
-    Value = "${UPDATEURL}?pkgfmt=<pkgformat>";
+    Value = "http://ooop.sourceforge.net/update/ooop.php";
   #endif //defined WNT
   #else
     Value = "";
--- scp2/util/makefile.mk	2006-09-08 12:05:19.000000000 +0200
+++ scp2/util/makefile.mk	2006-09-08 12:06:07.000000000 +0200
@@ -161,11 +161,9 @@
 SCP1FILES += cairocanvas.par
 .ENDIF
 
-.IF "$(BUILD_SPECIAL)"!=""
 SCP1FILES += \
              module_onlineupdate.par   \
              file_onlineupdate.par
-.ENDIF
 
 # ------------------------------------------------------------------------
 # OpenOffice.org with JRE
@@ -280,11 +278,9 @@
 SCP2FILES += \
              vclcanvas.par
 
-.IF "$(BUILD_SPECIAL)"!=""
 SCP2FILES += \
              module_onlineupdate.par   \
              file_onlineupdate.par
-.ENDIF
 
 # ------------------------------------------------------------------------
 # URE
