diff -rup oox-orig-1/source/drawingml/textbodypropertiescontext.cxx oox/source/drawingml/textbodypropertiescontext.cxx
--- oox-orig-1/source/drawingml/textbodypropertiescontext.cxx	2009-06-03 11:36:03.000000000 +0200
+++ oox/source/drawingml/textbodypropertiescontext.cxx	2009-06-03 11:55:15.000000000 +0200
@@ -121,7 +121,8 @@ TextBodyPropertiesContext::TextBodyPrope
 
     // ST_TextVerticalType
     mrTextBodyProp.moVert = aAttribs.getToken( XML_vert );
-    if( mrTextBodyProp.moVert.get( XML_horz ) == XML_vert ) {
+    sal_Int32 tVert = mrTextBodyProp.moVert.get( XML_horz );
+    if( tVert == XML_vert || tVert == XML_eaVert || tVert == XML_vert270 || tVert == XML_mongolianVert ) {
 	mrTextBodyProp.maPropertyMap[ CREATE_OUSTRING( "TextWritingMode" ) ]
 	    <<= WritingMode_TB_RL;
 	// workaround for TB_LR as using WritingMode2 doesn't work
