diff -rup oox-orig/source/ppt/pptshape.cxx oox/source/ppt/pptshape.cxx
--- oox-orig/source/ppt/pptshape.cxx	2009-05-12 16:05:07.000000000 +0200
+++ oox/source/ppt/pptshape.cxx	2009-05-12 16:09:59.000000000 +0200
@@ -74,8 +74,8 @@ void PPTShape::addShape(
         const awt::Rectangle* pShapeRect,
         ::oox::drawingml::ShapeIdMap* pShapeMap )
 {
-	// only placeholder from layout are being inserted
-	if ( mnSubType && ( meShapeLocation == Master ) )
+	// only placeholder in slides are being inserted
+	if ( mnSubType && ( meShapeLocation == Master || meShapeLocation == Layout ) )
 		return;
 	try
     {
