--- padmin/source/makefile.mk.old	2009-04-02 11:01:37.000000000 +0000
+++ padmin/source/makefile.mk	2009-04-06 16:41:45.000000000 +0000
@@ -47,6 +47,10 @@ LIBTARGET=NO
 
 .INCLUDE: settings.mk
 
+.IF "$(FONTCONFIG_FONTS)" != ""
+CFLAGS+=-DFONTCONFIG_FONTS
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SRS1NAME=$(TARGET)
@@ -69,11 +73,13 @@ SLOFILES=\
 	$(SLO)$/progress.obj		\
 	$(SLO)$/newppdlg.obj		\
 	$(SLO)$/prtsetup.obj		\
-	$(SLO)$/fontentry.obj		\
 	$(SLO)$/helper.obj			\
 	$(SLO)$/adddlg.obj			\
 	$(SLO)$/titlectrl.obj
 
+.IF "$(FONTCONFIG_FONTS)" == ""
+SLOFILES+= $(SLO)$/fontentry.obj
+.ENDIF
 
 OBJFILES=\
 	$(OBJ)/pamain.obj
--- padmin/source/padialog.cxx.old	2009-04-02 11:01:37.000000000 +0000
+++ padmin/source/padialog.cxx	2009-04-06 16:41:45.000000000 +0000
@@ -42,14 +42,13 @@
 #include <vcl/event.hxx>
 #include <tools/stream.hxx>
 #include <tools/color.hxx>
-#ifndef _PAD_PADIALOG_HRC_
+#include <tools/urlobj.hxx>
+#ifndef FONTCONFIG_FONTS
 #include <padialog.hrc>
 #endif
 #include <fontentry.hxx>
 #include <helper.hxx>
-#ifndef _PSP_PRINTERINFOMANAGER_HXX_
 #include <psprint/printerinfomanager.hxx>
-#endif
 #include <padialog.hxx>
 #include <adddlg.hxx>
 #include <prtsetup.hxx>
@@ -93,7 +92,9 @@ PADialog::PADialog( Window* pParent, BOO
         m_aCUPSCB( this, PaResId( RID_PA_CB_CUPSUSAGE ) ),
         m_aSepButtonFL( this, PaResId( RID_PA_FL_SEPBUTTON ) ),
 		m_aAddPB( this, PaResId( RID_PA_BTN_ADD ) ),
+	#ifndef FONTCONFIG_FONTS
 		m_aFontsPB( this, PaResId( RID_PA_BTN_FONTS ) ),
+	#endif
 		m_aCancelButton( this, PaResId( RID_PA_BTN_CANCEL ) ),
 		m_aDefPrt( PaResId( RID_PA_STR_DEFPRT ) ),
 		m_aRenameStr( PaResId( RID_PA_STR_RENAME ) ),
@@ -139,14 +140,16 @@ void PADialog::Init()
 	m_aConfPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aRenamePB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aTestPagePB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
-	m_aFontsPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aAddPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aDevicesLB.setDelPressedLink( LINK( this, PADialog, DelPressedHdl ) );
     m_aCUPSCB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 
+     #ifndef FONTCONFIG_FONTS
+        m_aFontsPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	::psp::PrintFontManager& rFontManager( ::psp::PrintFontManager::get() );
 	if( ! rFontManager.checkImportPossible() )
 		m_aFontsPB.Enable( FALSE );
+     #endif
 }
 
 PADialog::~PADialog()
@@ -212,11 +215,13 @@ IMPL_LINK( PADialog, ClickBtnHdl, PushBu
 		PrintTestPage();
 	else if( pButton == &m_aAddPB )
 		AddDevice();
+     #ifndef FONTCONFIG_FONTS
 	else if( pButton == &m_aFontsPB )
 	{
 		FontNameDlg aDialog( this );
 		aDialog.Execute();
 	}
+     #endif
     else if( static_cast<Button*>(pButton) == &m_aCUPSCB )
     {
         m_rPIManager.setCUPSDisabled( m_aCUPSCB.IsChecked() );
--- padmin/source/padialog.hxx.old	2009-04-02 11:01:37.000000000 +0000
+++ padmin/source/padialog.hxx	2009-04-06 16:41:45.000000000 +0000
@@ -77,7 +77,9 @@ namespace padmin {
 
         FixedLine							m_aSepButtonFL;
 		PushButton							m_aAddPB;
+#ifndef FONTCONFIG_FONTS
 		PushButton							m_aFontsPB;
+#endif
 		CancelButton  						m_aCancelButton;
 
 		String								m_aDefPrt;
