Index: target.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/target.mk,v
retrieving revision 1.209
diff -u -p -r1.209 target.mk
--- solenv/inc/target.mk	6 Jun 2008 12:00:28 -0000	1.209
+++ solenv/inc/target.mk	4 Jul 2008 15:25:03 -0000
@@ -48,16 +48,16 @@
 ENVINCPRE+=-I$(PRE)$/inc
 .ENDIF			# "$(PRE)"!=""
 .IF "$(BOOTSTRAP_SERVICE)"!="TRUE" && "$(NO_OFFUH)"==""
-UNOINCLUDES=$(SOLARINCDIR)$/offuh
+UNOINCLUDES*=-I$(SOLARINCDIR)$/offuh
 .ENDIF			# "$(BOOTSTRAP_SERVICE)"!="TRUE" && "$(NO_OFFUH)"==""
 SOLARINC+=$(JDKINCS)
 # make sure INCLUDE isn't set
 INCLUDE:=
 .EXPORT : INCLUDE
 .IF "$(PRJINC)"!=""
-INCLUDE!:=-I. $(ENVINCPRE) $(INCPRE:^"-I":s/-I-I/-I/) -I$(INCLOCAL) $(INCLOCPRJ:^"-I":s/-I-I/-I/) -I$(INCPCH) -I$(INC) -I$(INCGUI) -I$(INCCOM) $(SOLARINC) $(null,$(UNOINCLUDES) $(NULL) -I$(UNOINCLUDES)) -I$(INCEXT) -I$(PRJ)$/res -I$(INCPOST)
+INCLUDE!:=-I. $(ENVINCPRE) $(INCPRE:^"-I":s/-I-I/-I/) -I$(INCLOCAL) $(INCLOCPRJ:^"-I":s/-I-I/-I/) -I$(INCPCH) -I$(INC) -I$(INCGUI) -I$(INCCOM) $(SOLARINC) $(null,$(UNOINCLUDES) $(NULL) $(UNOINCLUDES)) -I$(INCEXT) -I$(PRJ)$/res -I$(INCPOST)
 .ELSE		# "$(PRJINC)"!=""
-INCLUDE!:=-I. $(ENVINCPRE) $(INCPRE:^"-I":s/-I-I/-I/) -I$(INCLOCAL) -I$(INCPCH) -I$(INC) -I$(INCGUI) -I$(INCCOM) $(SOLARINC) $(null,$(UNOINCLUDES) $(NULL) -I$(UNOINCLUDES)) -I$(INCEXT) -I$(PRJ)$/res -I$(INCPOST)
+INCLUDE!:=-I. $(ENVINCPRE) $(INCPRE:^"-I":s/-I-I/-I/) -I$(INCLOCAL) -I$(INCPCH) -I$(INC) -I$(INCGUI) -I$(INCCOM) $(SOLARINC) $(null,$(UNOINCLUDES) $(NULL) $(UNOINCLUDES)) -I$(INCEXT) -I$(PRJ)$/res -I$(INCPOST)
 .ENDIF		# "$(PRJINC)"!=""
 INCLUDE_C=$(subst,$/stl$(SPACECHAR),dont_use_stl$(SPACECHAR) $(INCLUDE))
 .EXPORT : LIB
@@ -218,6 +218,7 @@
 LOCALJARS:=$(foreach,i,$(shell @@-cd $(JARDIR) && ls -1 $(JARFILES) ) $(JARDIR)$/$i)
 NEWCLASS:=$(LOCALJARS)
 NEWCLASS+:=$(foreach,i,$(JARFILES) $(eq,$(LOCALJARS),$(subst,$i, $(LOCALJARS)) $(SOLARBINDIR)$/$i $(NULL)))
+NEWCLASS+:=$(foreach,i,$(JARFILES) $(eq,$(LOCALJARS),$(subst,$i, $(LOCALJARS)) $(SOLARPIECE)$/$(INPATH)$/bin/$i $(NULL)))
 .ENDIF			# "$(JARFILES)"!=""
 .IF "$(EXTRAJARFILES)"!=""
 NEWCLASS+=$(foreach,i,$(EXTRAJARFILES) $(COMMON_BUILD_TOOLS)$/$i)
@@ -861,6 +861,6 @@
 .IF "$(PRE)"!=""
 UNOIDLINC!:=-I$(PRE)$/idl $(UNOIDLINC)
 .ENDIF
-UNOIDLINC+=-I. -I.. -I$(PRJ) -I$(PRJ)$/inc -I$(PRJ)$/$(INPATH)$/idl -I$(OUT)$/inc -I$(SOLARIDLDIR) -I$(SOLARINCDIR)
+UNOIDLINC+=-I. -I.. -I$(PRJ) -I$(PRJ)$/inc -I$(PRJ)$/$(INPATH)$/idl -I$(OUT)$/inc $(SOLARIDLINC) -I$(SOLARINCDIR)
 
 CDEFS= -D$(OS) -D$(GUI) -D$(GVER) -D$(COM) -D$(CVER) -D$(CPUNAME)

--- solenv/inc/unxlngi6.mk
+++ solenv/inc/unxlngi6.mk
@@ -182,7 +182,7 @@
 STDLIBCPP=-lstdc++
 
 # default objectfilenames to link
-STDOBJVCL=$(L)$/salmain.o
+STDOBJVCL*=$(L)$/salmain.o
 STDOBJGUI=
 STDSLOGUI=
 STDOBJCUI=

--- solenv/inc/settings.mk
+++ solenv/inc/settings.mk
@@ -764,6 +764,7 @@
 .ELSE
 SOLARIDLDIR=$(SOLARVERSION)$/$(INPATH)$/idl$(EXT_UPDMINOR)
 .ENDIF
+SOLARIDLINC*=-I$(SOLARIDLDIR)
 
 .IF "$(UPDMINOR)" != ""
 EXT_UPDMINOR=.$(UPDMINOR)
@@ -1088,10 +1082,6 @@
 LOCAL_EXCEPTIONS_FLAGS+=$(CFLAGSEXCEPTIONS)
 LOCAL_EXCEPTIONS_FLAGS+=-DEXCEPTIONS_ON
 
-.IF "$(ENABLE_GSTREAMER)" == "TRUE"
-CDEFS+=-DGSTREAMER
-.ENDIF
-
 # compose flags and defines for GUI
 .IF "$(TARGETTYPE)"=="GUI"
 CDEFS+= $(CDEFSGUI)
