--- solenv/bin/modules/installer/setupscript.pm	2008-07-18 17:17:13.000000000 +0100
+++ solenv/bin/modules/installer/setupscript.pm	2008-07-18 17:17:13.000000000 +0100
@@ -66,10 +66,14 @@
 	
 	# and now the complete path for the setup script is needed
 	# The log file cannot be used, because this is the language independent section
-	
-	$scriptnameref = installer::scriptitems::get_sourcepath_from_filename_and_includepath(\$scriptname, $includepatharrayref, 1);
 
-	$installer::globals::setupscriptname = $$scriptnameref;
+	if (defined $ENV{OO_SOLVERDIR}) {
+	    $installer::globals::setupscriptname = "$ENV{OO_SOLVERDIR}/bin/$scriptname";
+	} else {
+	    $scriptnameref = installer::scriptitems::get_sourcepath_from_filename_and_includepath(\$scriptname, $includepatharrayref, 1);
+
+	    $installer::globals::setupscriptname = $$scriptnameref;
+	}
 
 	if ( $installer::globals::setupscriptname eq "" )
 	{
