--- extensions/source/oooimprovement/makefile.mk.old	2009-02-21 00:00:23.000000000 +0100
+++ extensions/source/oooimprovement/makefile.mk	2009-03-02 12:35:34.000000000 +0100
@@ -37,10 +37,10 @@ TARGET=oooimprovement
 ENABLE_EXCEPTIONS=TRUE
 
 PACKAGE=org.openoffice.Office
-ABSXCSROOT=$(SOLARXMLDIR)
+ABSXCSROOT=$(SPLITINSTALLDIR)$/..$/share
 XCSROOT=..
-DTDDIR=$(ABSXCSROOT)
-XSLDIR=$(ABSXCSROOT)$/processing
+DTDDIR=$(SOLARXMLDIR)
+XSLDIR=$(SOLARXMLDIR)$/processing
 PROCESSOUT=$(MISC)$/$(TARGET)
 PROCESSORDIR=$(SOLARBINDIR)
 
--- extensions/source/update/check/makefile.mk
+++ extensions/source/update/check/makefile.mk
@@ -37,10 +37,10 @@
 LIBTARGET=NO
 ENABLE_EXCEPTIONS=TRUE
 
-ABSXCSROOT=$(SOLARXMLDIR)
+ABSXCSROOT=$(SPLITINSTALLDIR)$/..$/share
 XCSROOT=..
-DTDDIR=$(ABSXCSROOT)
-XSLDIR=$(ABSXCSROOT)$/processing
+DTDDIR=$(SOLARXMLDIR)
+XSLDIR=$(SOLARXMLDIR)$/processing
 PROCESSOUT=$(MISC)$/$(TARGET)
 PROCESSORDIR=$(SOLARBINDIR)
 
