Index: automation/packimages/makefile.mk
===================================================================
RCS file: /cvs/util/automation/packimages/makefile.mk,v
retrieving revision 1.12
diff -u -r1.12 makefile.mk
--- automation/packimages/makefile.mk	11 Apr 2008 09:53:44 -0000	1.12
+++ automation/packimages/makefile.mk	13 Aug 2008 12:23:42 -0000
@@ -53,7 +53,7 @@
 #common_outdir = common
 #solarresdir = j:\so-cwsserv04\gh7\SRC680\wntmsci10\res.m50
 
-
+IMGSRC=$(SOLARVER)
 $(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/images_tt.zip .PHONY:
     @echo ------------------------------
     @echo Making: $@
@@ -63,5 +63,5 @@
     $(COPY) $(subst,$(OUTPATH),$(COMMON_OUTDIR) $(SOLARRESDIR))$/img$/vcl*.ilst $(MISC) 
     $(COPY) $(subst,$(OUTPATH),$(COMMON_OUTDIR) $(SOLARRESDIR))$/img$/sb*.ilst $(MISC) 
     -$(MKDIR) $(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))
-    $(PERL) $(SOLARENV)$/bin$/packimages.pl -g $(SOLARSRC)$/$(RSCDEFIMG) -m $(SOLARSRC)$/$(RSCDEFIMG) -c $(RSCCUSTOMIMG) -l $(MISC) -o $@
+    $(PERL) $(SOLARENV)$/bin$/packimages.pl -g $(IMGSRC)$/$(RSCDEFIMG) -m $(IMGSRC)$/$(RSCDEFIMG) -c $(RSCCUSTOMIMG) -l $(MISC) -o $@
 
