--- scripting/source/vbaevents/makefile.mk
+++ scripting/source/vbaevents/makefile.mk
@@ -23,8 +23,9 @@
         $(MISC)$/$(TARGET).don \
         $(SLOTARGET)
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+OOVBADIR=$(SOLARPIECE)$/$(INPATH)$/bin
+$(MISC)$/$(TARGET).don : $(OOVBADIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(OOVBADIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
         echo $@
 # ------------------------------------------------------------------
 
