diff -u -r io/source/acceptor/makefile.mk io/source/acceptor/makefile.mk
--- io/source/acceptor/makefile.mk	2008-04-11 12:43:42.000000000 +0100
+++ io/source/acceptor/makefile.mk	2008-07-17 11:16:04.000000000 +0100
@@ -36,15 +36,15 @@
 ENABLE_EXCEPTIONS=TRUE
 NO_BSYMBOLIC=TRUE
 COMP1TYPELIST = acceptor
-COMPRDB=$(SOLARBINDIR)$/udkapi.rdb
+COMPRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb 
 
 # --- Settings -----------------------------------------------------
 .INCLUDE :  settings.mk
 DLLPRE =
 # ------------------------------------------------------------------
 
-UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb
-UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
+UNOUCRDEP=$(COMPRDB)
+UNOUCRRDB=$(COMPRDB)
 UNOUCROUT=$(OUT)$/inc$/acceptor
 INCPRE+= $(UNOUCROUT)
 
Only in io/source/acceptor: makefile.mk~
diff -u -r io/source/connector/makefile.mk io/source/connector/makefile.mk
--- io/source/connector/makefile.mk	2008-04-11 12:45:15.000000000 +0100
+++ io/source/connector/makefile.mk	2008-07-17 11:15:19.000000000 +0100
@@ -36,15 +36,15 @@
 ENABLE_EXCEPTIONS=TRUE
 NO_BSYMBOLIC=TRUE
 COMP1TYPELIST = connectr
-COMPRDB=$(SOLARBINDIR)$/udkapi.rdb
+COMPRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb 
 
 # --- Settings -----------------------------------------------------
 .INCLUDE :  settings.mk
 DLLPRE =
 # ------------------------------------------------------------------
 
-UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb
-UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
+UNOUCRDEP=$(COMPRDB)
+UNOUCRRDB=$(COMPRDB)
 UNOUCROUT=$(OUT)$/inc$/connector
 INCPRE+= $(UNOUCROUT)
 
Only in io/source/connector: makefile.mk~
diff -u -r io/source/stm/makefile.mk io/source/stm/makefile.mk
--- io/source/stm/makefile.mk	2008-04-11 12:46:07.000000000 +0100
+++ io/source/stm/makefile.mk	2008-07-17 11:16:32.000000000 +0100
@@ -35,14 +35,14 @@
 ENABLE_EXCEPTIONS=TRUE
 NO_BSYMBOLIC=TRUE
 COMP1TYPELIST = stm
-COMPRDB=$(SOLARBINDIR)$/udkapi.rdb
+COMPRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb 
 
 # --- Settings -----------------------------------------------------
 .INCLUDE :  settings.mk
 DLLPRE =
 # --- Files --------------------------------------------------------
-UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb
-UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
+UNOUCRDEP=$(COMPRDB)
+UNOUCRRDB=$(COMPRDB)
 UNOUCROUT=$(OUT)$/inc$/stm
 INCPRE+= $(UNOUCROUT)
 
Only in io/source/stm: makefile.mk~
