--- writerfilter/source/ooxml/makefile.mk
+++ writerfilter/source/ooxml/makefile.mk
@@ -185,7 +185,7 @@
 .ENDIF
 
 $(TOKENXMLTMP): 
-    $(TYPE) $(SOLARVER)$/$(INPATH)$/inc$(UPDMINOREXT)$/oox$/token.txt | $(SED) "s#\(.*\)#<fasttoken>\1</fasttoken>#" > $@
+    $(TYPE) $(SOLARPIECE)$/$(INPATH)$/inc$(UPDMINOREXT)$/oox$/token.txt | $(SED) "s#\(.*\)#<fasttoken>\1</fasttoken>#" > $@
 
 $(TOKENXML): tokenxmlheader $(TOKENXMLTMP) tokenxmlfooter
 	$(TYPE) tokenxmlheader $(TOKENXMLTMP) tokenxmlfooter > $@
