--- solenv/inc/settings.mk.old	2008-10-08 13:41:29.000000000 +0200
+++ solenv/inc/settings.mk	2008-10-08 14:49:03.000000000 +0200
@@ -982,6 +982,7 @@ RSCDEFS=-D$(GUI) -D$(GVER) -D$(COM) -D$(
 RSCDEFS+=-DBUILD_SPECIAL=$(BUILD_SPECIAL)
 .ENDIF
 
+IMGSRC=$(SOLARVER)
 RSCDEFIMG*=default_images
 
 RSCEXTINC=.
--- solenv/inc/tg_rslb.mk.old	2008-10-08 13:41:29.000000000 +0200
+++ solenv/inc/tg_rslb.mk	2008-10-08 14:44:02.000000000 +0200
@@ -62,11 +62,11 @@ $(RSC_MULTI$(TNR)) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB$(TNR)NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB$(TNR)IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB$(TNR)IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC$(TNR)HEADER) $(RESLIB$(TNR)SRSFILES) \
@@ -77,11 +77,11 @@ $(RSC_MULTI$(TNR)) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB$(TNR)NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB$(TNR)IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB$(TNR)IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC$(TNR)HEADER) $(RESLIB$(TNR)SRSFILES) \
--- solenv/inc/_tg_rslb.mk.old	2008-10-08 13:41:29.000000000 +0200
+++ solenv/inc/_tg_rslb.mk	2008-10-08 14:42:50.000000000 +0200
@@ -30,11 +30,11 @@ $(RSC_MULTI1) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB1NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB1IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB1IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC1HEADER) $(RESLIB1SRSFILES) \
@@ -45,11 +45,11 @@ $(RSC_MULTI1) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB1NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB1IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB1IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC1HEADER) $(RESLIB1SRSFILES) \
@@ -108,11 +108,11 @@ $(RSC_MULTI2) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB2NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB2IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB2IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC2HEADER) $(RESLIB2SRSFILES) \
@@ -123,11 +123,11 @@ $(RSC_MULTI2) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB2NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB2IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB2IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC2HEADER) $(RESLIB2SRSFILES) \
@@ -186,11 +186,11 @@ $(RSC_MULTI3) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB3NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB3IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB3IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC3HEADER) $(RESLIB3SRSFILES) \
@@ -201,11 +201,11 @@ $(RSC_MULTI3) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB3NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB3IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB3IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC3HEADER) $(RESLIB3SRSFILES) \
@@ -264,11 +264,11 @@ $(RSC_MULTI4) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB4NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB4IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB4IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC4HEADER) $(RESLIB4SRSFILES) \
@@ -279,11 +279,11 @@ $(RSC_MULTI4) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB4NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB4IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB4IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC4HEADER) $(RESLIB4SRSFILES) \
@@ -342,11 +342,11 @@ $(RSC_MULTI5) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB5NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB5IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB5IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC5HEADER) $(RESLIB5SRSFILES) \
@@ -357,11 +357,11 @@ $(RSC_MULTI5) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB5NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB5IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB5IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC5HEADER) $(RESLIB5SRSFILES) \
@@ -420,11 +420,11 @@ $(RSC_MULTI6) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB6NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB6IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB6IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC6HEADER) $(RESLIB6SRSFILES) \
@@ -435,11 +435,11 @@ $(RSC_MULTI6) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB6NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB6IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB6IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC6HEADER) $(RESLIB6SRSFILES) \
@@ -498,11 +498,11 @@ $(RSC_MULTI7) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB7NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB7IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB7IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC7HEADER) $(RESLIB7SRSFILES) \
@@ -513,11 +513,11 @@ $(RSC_MULTI7) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB7NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB7IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB7IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC7HEADER) $(RESLIB7SRSFILES) \
@@ -576,11 +576,11 @@ $(RSC_MULTI8) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB8NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB8IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB8IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC8HEADER) $(RESLIB8SRSFILES) \
@@ -591,11 +591,11 @@ $(RSC_MULTI8) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB8NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB8IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB8IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC8HEADER) $(RESLIB8SRSFILES) \
@@ -654,11 +654,11 @@ $(RSC_MULTI9) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB9NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB9IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB9IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC9HEADER) $(RESLIB9SRSFILES) \
@@ -669,11 +669,11 @@ $(RSC_MULTI9) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB9NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB9IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB9IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC9HEADER) $(RESLIB9SRSFILES) \
@@ -732,11 +732,11 @@ $(RSC_MULTI10) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))$/$(RESLIB10NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB10IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB10IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil={$(subst,$(OUTPATH),$(COMMON_OUTDIR) $(BIN))} \
     -ft=$@ \
 	$(RSC10HEADER) $(RESLIB10SRSFILES) \
@@ -747,11 +747,11 @@ $(RSC_MULTI10) : \
 	$(foreach,i,$(alllangiso) -lg$i \
 	$(null,$(rescharset_{$i}) $(default$(LANG_GUI)) $(rescharset_{$i})) \
 	-fs={$(BIN)$/$(RESLIB10NAME)$i.res} \
-    $(foreach,j,$(subst,$(PRJ),$(SOLARSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB10IMAGES)) -lip={$j}$/$i \
+    $(foreach,j,$(subst,$(PRJ),$(IMGSRC)$/$(RSCDEFIMG)$/$(PRJNAME) $(RESLIB10IMAGES)) -lip={$j}$/$i \
     -lip={$j} ) \
-    -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(SOLARSRC)$/$(RSCDEFIMG)$/res ) \
-	-subMODULE=$(SOLARSRC)$/$(RSCDEFIMG) \
-	-subGLOBALRES=$(SOLARSRC)$/$(RSCDEFIMG)$/res \
+    -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res$/$i -lip=$(IMGSRC)$/$(RSCDEFIMG)$/res ) \
+	-subMODULE=$(IMGSRC)$/$(RSCDEFIMG) \
+	-subGLOBALRES=$(IMGSRC)$/$(RSCDEFIMG)$/res \
     -oil=$(BIN) \
     -ft=$@ \
 	$(RSC10HEADER) $(RESLIB10SRSFILES) \
