--- writerfilter/source/ooxml/makefile.mk
+++ writerfilter/source/ooxml/makefile.mk
@@ -184,8 +184,8 @@
 XSLTPROC=xsltproc
 .ENDIF
 
-$(TOKENXMLTMP): $(SOLARVER)$/$(INPATH)$/inc$(UPDMINOREXT)$/oox$/token.txt
-    $(TYPE) $(SOLARVER)$/$(INPATH)$/inc$(UPDMINOREXT)$/oox$/token.txt | $(SED) "s#\(.*\)#<fasttoken>\1</fasttoken>#" > $@
+ $(TOKENXMLTMP): $(SOLARPIECE)$/$(INPATH)$/inc$(UPDMINOREXT)$/oox$/token.txt
+    $(TYPE) $(SOLARPIECE)$/$(INPATH)$/inc$(UPDMINOREXT)$/oox$/token.txt | $(SED) "s#\(.*\)#<fasttoken>\1</fasttoken>#" > $@
 
 $(TOKENXML): tokenxmlheader $(TOKENXMLTMP) tokenxmlfooter
 	$(TYPE) tokenxmlheader $(TOKENXMLTMP) tokenxmlfooter > $@
