--- /opt/OpenOffice/dev300-m25/reportdesign/util/makefile.pmk	2008-06-06 09:05:19.000000000 +0100
+++ /usr/src/packages/BUILD/dev300-m25-base/reportdesign/util/makefile.pmk	2008-07-25 19:21:48.000000000 +0100
@@ -31,7 +31,7 @@
 
 XSLDIR=$(SOLARXMLDIR)$/processing
 XCSROOT=$(SOLARXMLDIR)
-ABSXCSROOT=$(SOLARXMLDIR)
+ABSXCSROOT=$(SPLITINSTALLDIR)$/..$/share
 DTDDIR=$(SOLARXMLDIR)
 PROCESSOUT=$(MISC)$/$(EXTNAME)
 PROCESSORDIR=$(SOLARBINDIR)
