--- setup_native/registry/data/org/openoffice/Office/makefile.mk	2008-07-25 02:38:34.000000000 +0100
+++ setup_native/registry/data/org/openoffice/Office/makefile.mk	2008-07-25 02:38:34.000000000 +0100
@@ -34,10 +34,10 @@
 TARGET=data_stclient
 PACKAGE=org.openoffice.Office
 
-ABSXCSROOT=$(SOLARXMLDIR)
+ABSXCSROOT=$(SPLITINSTALLDIR)$/..$/share
 XCSROOT=..
-DTDDIR=$(ABSXCSROOT)
-XSLDIR=$(ABSXCSROOT)$/processing
+DTDDIR=$(SOLARXMLDIR)
+XSLDIR=$(SOLARXMLDIR)$/processing
 PROCESSOUT=$(MISC)$/$(TARGET)
 PROCESSORDIR=$(SOLARBINDIR)
 
