--- cli_ure/unotypes/makefile.mk.old	2008-08-29 13:34:51.000000000 +0200
+++ cli_ure/unotypes/makefile.mk	2008-08-29 14:29:05.000000000 +0200
@@ -44,7 +44,7 @@ TARGET = unotypes
 .IF "$(BUILD_FOR_CLI)" != "" && ("$(GUI)" == "WNT" || "$(ENABLE_MONO_CLIMAKER)" == "YES")
 
 .IF "$(ENABLE_MONO_CLIMAKER)" == "YES"
-WRAPCMD=MONO_PATH=$(OUT)/bin:$(MONO_PATH) LD_LIBRARY_PATH=$(SOLARLIBDIR):$(OUT)/lib
+WRAPCMD=MONO_PATH=$(OUT)/bin:$(MONO_PATH) LD_LIBRARY_PATH=$(SOLARLIBDIR):$(SOLARPIECE)$/$(INPATH)$/lib:$(OUT)/lib
 CLIMAKER=climaker
 .ELSE
 CLIMAKER=climaker.exe
@@ -64,14 +64,14 @@ CLIMAKERFLAGS += --verbose
 .ENDIF
 
 #When changing the assembly version then this must also be done in scp2
-$(OUT)$/bin$/cli_uretypes.dll : $(BIN)$/$(CLIMAKER) $(SOLARBINDIR)$/types.rdb $(BIN)$/cliureversion.mk
+$(OUT)$/bin$/cli_uretypes.dll : $(BIN)$/$(CLIMAKER) $(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb $(BIN)$/cliureversion.mk
 	$(WRAPCMD) $(BIN)$/$(CLIMAKER) $(CLIMAKERFLAGS) \
 		--out $@ \
 		--keyfile $(BIN)$/cliuno.snk \
 		--assembly-version $(CLI_URETYPES_NEW_VERSION) \
 		--assembly-description "This assembly contains metadata for the StarOffice/OpenOffice.org API." \
 		--assembly-company "OpenOffice.org" \
-		$(SOLARBINDIR)$/udkapi.rdb
+		$(SOLARPIECE)$/$(INPATH)$/bin$/udkapi.rdb
 
 #do not forget to deliver cli_uretypes.config. It is NOT embedded in the policy file.
 $(POLICY_ASSEMBLY_FILE) : $(BIN)$/cli_uretypes.config $(OUT)$/bin$/cli_uretypes.dll
--- cli_ure/qa/climaker/makefile.mk.old	2008-06-25 14:05:37.000000000 +0200
+++ cli_ure/qa/climaker/makefile.mk	2008-08-29 13:50:49.000000000 +0200
@@ -106,7 +106,7 @@ $(OUTDIR)$/types.rdb: $(OUTDIR)$/types.u
 $(OUTDIR)$/cli_test_types.dll: $(OUTDIR)$/types.rdb $(BIN)$/climaker.exe $(BIN)$/cli_uretypes.dll
     $(CLIMAKER) $(CLIMAKERFLAGS) --out $@  \
 		-r $(BIN)$/cli_uretypes.dll \
-		-X $(SOLARBINDIR)$/types.rdb \
+		-X $(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb \
 		$(OUTDIR)$/types.rdb
 
 
