--- setup_native/registry/data/org/openoffice/Office/makefile.mk.old	2009-04-02 13:05:47.000000000 +0200
+++ setup_native/registry/data/org/openoffice/Office/makefile.mk	2009-04-08 12:34:02.000000000 +0200
@@ -34,10 +34,10 @@ PRJNAME=setup_native
 TARGET=data_registration
 PACKAGE=org.openoffice.Office
 
-ABSXCSROOT=$(SOLARXMLDIR)
+ABSXCSROOT=$(SPLITINSTALLDIR)$/..$/share
 XCSROOT=..
-DTDDIR=$(ABSXCSROOT)
-XSLDIR=$(ABSXCSROOT)$/processing
+DTDDIR=$(SOLARXMLDIR)
+XSLDIR=$(SOLARXMLDIR)$/processing
 PROCESSOUT=$(MISC)$/$(TARGET)
 PROCESSORDIR=$(SOLARBINDIR)
 
--- setup_native/prj/build.lst.old	2009-04-02 13:05:47.000000000 +0200
+++ setup_native/prj/build.lst	2009-04-08 12:34:02.000000000 +0200
@@ -1,7 +1,5 @@
 pk	setup_native	:	soltools xml2cmp sal officecfg unoil NULL
 pk  setup_native                                          usr1    - all  sn_mkout NULL
-pk  setup_native\scripts\source                           nmake   -  u   sn_source NULL
-pk  setup_native\scripts                                  nmake   -  u   sn_scripts sn_source.u NULL
 pk  setup_native\source\win32\customactions\tools         nmake   -  w   sn_tools NULL
 pk  setup_native\source\win32\customactions\relnotes      nmake   -  w   sn_relnotes NULL
 pk  setup_native\source\win32\customactions\regactivex    nmake   -  w   sn_regactivex NULL
