--- /tmp/makefile.mk	2008-07-17 10:16:39.000000000 +0100
+++ /usr/src/packages/BUILD/dev300-m21-ure/offuh/source/makefile.mk	2008-07-17 10:29:04.000000000 +0100
@@ -39,6 +39,7 @@
 
 # ------------------------------------------------------------------
 
+UNOUCRRDB=$(SOLARPIECE)$/$(INPATH)$/bin$/types.rdb
 $(MISC)$/$(TARGET).don : $(UNOUCRRDB)
 	@@-$(RM) $@
 	$(CPPUMAKER) -E$(UNOUCROUT)/linking_catch.hxx -Gc $(CPPUMAKERFLAGS) -B$(UNOUCRBASE) -O$(UNOUCROUT) $(UNOUCRRDB) && echo > $@
