--- solenv/bin/build.pl
+++ solenv/bin/build.pl
@@ -2158,21 +2189,28 @@
     my $old_fh = select(STDOUT);
    
     # First try to get global depencies from solver's build.lst if such exists
+    my @search_dirs;
     my $solver_inc_dir = "$ENV{SOLARVER}/common";
     $solver_inc_dir .= $ENV{PROEXT} if (defined $ENV{PROEXT});
     $solver_inc_dir .= '/inc';
     $solver_inc_dir .= $ENV{UPDMINOREXT} if (defined $ENV{UPDMINOREXT});
     $solver_inc_dir .= "/$module";
-    $solver_inc_dir = CorrectPath($solver_inc_dir);
     $dead_parents{$module}++;
+    push @search_dirs, CorrectPath($solver_inc_dir);
+    if (defined $ENV{SYSTEM_SOLVER}) {
+	push (@search_dirs, CorrectPath($ENV{SYSTEM_SOLVER}."/inc/$module"));
+    }
+
+    print "Fetching dependencies for module $module from solvers @search_dirs...\n";
-    print "Fetching dependencies for module $module from solver...";
-    foreach (@possible_build_lists) {
-        my $possible_build_lst = "$solver_inc_dir/$_";
-        if (-e $possible_build_lst) {
-            print " ok\n";
-            select($old_fh);
-            return $possible_build_lst;
-        };
+    foreach my $dir (@search_dirs) {
+	foreach (@possible_build_lists) {
+	    my $possible_build_lst = "$dir/$_";
+	    if (-e $possible_build_lst) {
+		print " ok\n";
+		select($old_fh);
+		return $possible_build_lst;
+	    };
+	}
     }
     print " failed...\n";
     print "Fetching from CVS... ";
