--- connectivity/source/drivers/hsqldb/makefile.mk.old	2008-04-10 11:21:52.000000000 +0200
+++ connectivity/source/drivers/hsqldb/makefile.mk	2008-10-08 15:52:20.000000000 +0200
@@ -59,7 +59,7 @@ RES1FILELIST=\
 	$(SRS)$/$(TARGET).srs
 
 RESLIB1NAME=$(TARGET)
-RESLIB1IMAGES=$(SOLARSRC)$/$(RSCDEFIMG)$/database
+RESLIB1IMAGES=$(IMGSRC)$/$(RSCDEFIMG)$/database
 RESLIB1SRSFILES=$(RES1FILELIST)
 
 # Note that the resource file built here is currently *not* included
