--- helpcontent2/source/auxiliary/makefile.mk
+++ helpcontent2/source/auxiliary/makefile.mk
@@ -75,7 +75,7 @@
 
 $(COMMONBIN)$/helpimg.ilst .PHONY:
 	-$(RM) $@
-	$(PERL) $(PRJ)$/helpers$/create_ilst.pl -dir=$(SOLARSRC)/default_images/res/helpimg > $@.$(INPATH)
+	$(PERL) $(PRJ)$/helpers$/create_ilst.pl -dir=$(SOLARVER)/default_images/res/helpimg > $@.$(INPATH)
 	$(RENAME) $@.$(INPATH) $@
 	
 


--- helpcontent2/settings.pmk	2008-07-25 16:23:36.000000000 +0100
+++ helpcontent2/settings.pmk	2008-07-25 16:23:36.000000000 +0100
@@ -1,10 +1,10 @@
 .INCLUDE : $(INCCOM)$/aux_langs.mk
 
-my_cp:=$(CLASSPATH)$(PATH_SEPERATOR)$(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/juh.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SPLITINSTALLDIR)$/classes$/unoil.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/ridl.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/jurt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xmlsearch.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/LuceneHelpWrapper.jar$
+my_cp:=$(CLASSPATH)$(PATH_SEPERATOR)$(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SPLITINSTALLDIR)$/classes$/juh.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SPLITINSTALLDIR)$/classes$/unoil.jar$(PATH_SEPERATOR)$(SPLITUREDIR)$/share$/java$/ridl.jar$(PATH_SEPERATOR)$(SPLITUREDIR)$/share$/java$/jurt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xmlsearch.jar$(PATH_SEPERATOR)$(SPLITINSTALLDIR)$/classes$/LuceneHelpWrapper.jar$
 .IF "$(SYSTEM_LUCENE)" == "YES"
 my_cp!:=$(my_cp)$(PATH_SEPERATOR)$(LUCENE_CORE_JAR)$(PATH_SEPERATOR)$(LUCENE_ANALYZERS_JAR)
 .ELSE
-my_cp!:=$(my_cp)$(PATH_SEPERATOR)$(SOLARBINDIR)/lucene-core-2.3.jar$(PATH_SEPERATOR)$(SOLARBINDIR)/lucene-analyzers-2.3.jar
+my_cp!:=$(my_cp)$(PATH_SEPERATOR)$(SPLITINSTALLDIR)$/classes$/lucene-core-2.3.jar$(PATH_SEPERATOR)$(SPLITINSTALLDIR)$/classes$/lucene-analyzers-2.3.jar
 .ENDIF
  
 .IF "$(SYSTEM_DB)" != "YES"
