diff -rup sd/source/filter/eppt-orig/epptooxml.cxx sd/source/filter/eppt/epptooxml.cxx
--- sd/source/filter/eppt-orig/epptooxml.cxx	2009-04-23 12:18:28.000000000 +0200
+++ sd/source/filter/eppt/epptooxml.cxx	2009-04-23 12:19:57.000000000 +0200
@@ -1124,7 +1124,8 @@ void PowerPointExport::ImplWriteSlide( s
     pFS->endElementNS( XML_p, XML_cSld );
 
     WriteTransition( pFS );
-    WriteAnimations( pFS );
+    // disable these until there's time to iron it out
+    // WriteAnimations( pFS );
 
     pFS->endElementNS( XML_p, XML_sld );
 
Only in sd/source/filter/eppt: epptooxml.cxx~
