--- wizards/util/target.pmk.orig	2006-05-17 09:41:52.000000000 +0200
+++ wizards/util/target.pmk	2006-07-02 13:54:45.125000000 +0200
@@ -45,4 +45,6 @@
 STANDARD_ALL_TARGET             = basicsrvstandard
 LAUNCHER_ALL_TARGET             = basicsrvlauncher
 TUTORIALS_ALL_TARGET            = basicsrvtutorials
+PREMIUM_ALL_TARGET              = basicsrvpremium
+OOOWIKIPEDIA_ALL_TARGET              = basicsrvooowikipedia

