--- solenv/bin/modules/installer/languagepack.pm.orig	2008-03-25 00:19:45.000000000 +0100
+++ solenv/bin/modules/installer/languagepack.pm	2008-03-23 15:36:44.000000000 +0100
@@ -98,6 +98,11 @@
 						if ( $oneitem->{'Dir'} eq "gid_Dir_Fonts_Truetype" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Fonts_$locallang"; }
 						if ( $oneitem->{'Dir'} eq "gid_Dir_Resource" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Resource_$locallang"; }
 						if ( $oneitem->{'Dir'} eq "gid_Dir_Help_Isolanguage" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Help_$locallang"; }
+						if (( $oneitem->{'Dir'} eq "gid_Dir_Samples_Language") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Drawings") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Formulas") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Presentations") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Spreadsheets") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Texts") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Texts_Kapitel") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Databases") || ($oneitem->{'Dir'} eq "gid_Dir_Samples_Databases_Dateien")) { $oneitem->{'modules'} = "gid_Module_Optional_Accessories_Samples_$locallang"; }
+						if (( $oneitem->{'Dir'} eq "gid_Dir_Template_Language") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Educate") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Finance") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Forms") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Labels") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Layout") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Misc") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Officorr") || ($oneitem->{'Dir'} eq "id_Dir_Template_Offimisc") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Personal") || ($oneitem->{'Dir'} eq "gid_Dir_Template_Presnt" )) { $oneitem->{'modules'} = "gid_Module_Optional_Accessories_Templates_$locallang"; }
+						if ( $oneitem->{'Dir'} eq "gid_Dir_Samples_Documentations" ) { $oneitem->{'modules'} = "gid_Module_Optional_Accessories_Documentations_$locallang"; }
+						if ( $oneitem->{'Dir'} eq "gid_Dir_Samples_Advertisement" ) { $oneitem->{'modules'} = "gid_Module_Optional_Accessories_Advertisement_$locallang"; }
+
 					}
 				}
 
@@ -204,6 +209,10 @@
 		if ( ${$alldirs}[$i] =~ /-fonts/ ) { next; } 
 		if ( ${$alldirs}[$i] =~ /-help/ ) { next; } 
 		if ( ${$alldirs}[$i] =~ /-res/ ) { next; } 
+        	if ( ${$alldirs}[$i] =~ /-extra-samples/ ) { next; } 
+		if ( ${$alldirs}[$i] =~ /-extra-templates/ ) { next; } 
+		if ( ${$alldirs}[$i] =~ /-extra-documentations/ ) { next; } 
+		if ( ${$alldirs}[$i] =~ /-extra-advertisement/ ) { next; } 
 	
 		$packagename = ${$alldirs}[$i];
 		last;
