Index: source/minimizer/description.xml
===================================================================
RCS file: /cvs/graphics/sdext/source/minimizer/description.xml,v
retrieving revision 1.4.4.1
diff -u -u -r1.4.4.1 description.xml
--- sdext/source/minimizer/description.xml	15 Nov 2007 16:24:05 -0000	1.4.4.1
+++ sdext/source/minimizer/description.xml	16 Nov 2007 15:00:27 -0000
@@ -6,9 +6,4 @@
        <dependencies>
                <OpenOffice.org-minimal-version value="2.3" dep:name="OpenOffice.org 2.3"/>
        </dependencies>
-       <registration>
-        <simple-license accept-by="admin" default-license-id="lic-en-US">
-            <license-text xlink:href="licensefile" lang="isocode" license-id="lic-isocode"/>
-        </simple-license>
-    </registration>
 </description>
Index: source/minimizer/makefile.mk
===================================================================
RCS file: /cvs/graphics/sdext/source/minimizer/makefile.mk,v
retrieving revision 1.6
diff -u -u -r1.6 makefile.mk
--- sdext/source/minimizer/makefile.mk	1 Nov 2007 14:10:57 -0000	1.6
+++ sdext/source/minimizer/makefile.mk	16 Nov 2007 15:00:27 -0000
@@ -48,12 +48,6 @@
 
 DESCRIPTION:=$(MISC)$/SunPresentationMinimizer$/description.xml
 
-.IF "$(GUI)" == "WIN" || "$(GUI)" == "WNT"
-PACKLICS:=$(foreach,i,$(alllangiso) $(MISC)$/SunPresentationMinimizer$/registry$/license_$i)
-.ELSE
-PACKLICS:=$(foreach,i,$(alllangiso) $(MISC)$/SunPresentationMinimizer$/registry$/LICENSE_$i)
-.ENDIF
-
 DLLPRE=
 common_build_zip=
 
@@ -117,7 +111,7 @@
 COMPONENT_LIBRARY= \
     $(MISC)$/SunPresentationMinimizer$/SunPresentationMinimizer.uno$(DLLPOST)
 
-ZIP1DEPS=		$(PACKLICS) $(DESCRIPTION) $(COMPONENT_MANIFEST) $(COMPONENT_FILES) $(COMPONENT_BITMAPS) $(COMPONENT_HELP) $(COMPONENT_LIBRARY) $(COMPONENT_MERGED_XCU)
+ZIP1DEPS=		$(DESCRIPTION) $(COMPONENT_MANIFEST) $(COMPONENT_FILES) $(COMPONENT_BITMAPS) $(COMPONENT_HELP) $(COMPONENT_LIBRARY) $(COMPONENT_MERGED_XCU)
 ZIP1TARGET=		sun-presentation-minimizer
 ZIP1DIR=		$(MISC)$/SunPresentationMinimizer
 ZIP1EXT=		.oxt
@@ -144,16 +138,6 @@
 	@@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-.IF "$(GUI)" == "WIN" || "$(GUI)" == "WNT"
-$(PACKLICS) : $(SOLARBINDIR)$/osl$/license$$(@:b:s/_/./:e:s/./_/)$$(@:e).txt
-	@@-$(MKDIRHIER) $(@:d)
-	$(GNUCOPY) $< $@
-.ELSE
-$(PACKLICS) : $(SOLARBINDIR)$/osl$/LICENSE$$(@:b:s/_/./:e:s/./_/)$$(@:e)
-	@@-$(MKDIRHIER) $(@:d)
-	$(GNUCOPY) $< $@
-.ENDIF
-
 $(MISC)$/SunPresentationMinimizer$/registry$/data$/%.xcu : $(MISC)$/$(EXTNAME)$/merge$/%.xcu
 	@@-$(MKDIRHIER) $(@:d)
 	$(GNUCOPY) $< $@
