--- solenv/inc/rules.mk.old	2009-04-02 10:53:10.000000000 +0000
+++ solenv/inc/rules.mk	2009-04-06 16:41:40.000000000 +0000
@@ -162,7 +162,9 @@ $(SLO)$/precompiled_ex.% .PHONY:
 .ENDIF			# "$(ENABLE_PCH)"!=""
 
 $(SLO)$/%.obj : %.cxx
+.IF "$(QUIET_BUILD)"==""
 	@echo ------------------------------
+.ENDIF
 	@echo Making: $@
 .IF "$(ENABLE_PCH)"!="" && ( "$(PRJNAME)"!="sw" || "$(BUILD_SPECIAL)"!="TRUE" )
 # just a helper var	
@@ -181,11 +183,11 @@ $(SLO)$/%.obj : %.cxx
 	$(CXX) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSCXXSLO) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) -E  $(CFLAGSINCXX)$(PWD)$/$*.cxx
 .ELSE
 	@$(RM) $@ $(@:s/.obj/.o/)
-	$(CAPTURE_COMMAND) $(CXX) $(ACT_PCH_SWITCHES) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSCXXSLO) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(SLO)$/$*.o $(CFLAGSINCXX)$(PWD)$/$*.cxx $(CAPTURE_OUTPUT)
+	$(QUIET_BUILD) $(CAPTURE_COMMAND) $(CXX) $(ACT_PCH_SWITCHES) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSCXXSLO) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(SLO)$/$*.o $(CFLAGSINCXX)$(PWD)$/$*.cxx $(CAPTURE_OUTPUT)
 .IF "$(OS)"=="SOLARIS" && "$(product)"=="full" && "$(debug)"=="" && "$(CAPTURE_COMMAND)"==""
     $(ADJUSTVISIBILITY) -p $(@:s/.obj/.o/)
 .ENDIF          # "$(OS)"=="SOLARIS" && "$(product)"=="full" && "$(debug)"==""    
-	$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+	$(QUIET_BUILD) $(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
 .ENDIF
 .ELSE			# "$(GUI)"=="UNX"
 	@@-$(RM) $@
--- solenv/inc/settings.mk.old	2009-04-02 10:53:10.000000000 +0000
+++ solenv/inc/settings.mk	2009-04-06 16:41:40.000000000 +0000
@@ -308,6 +308,12 @@ PROFILE=
 DBGUTIL=
 dbgutil=
 
+.IF "$(VERBOSE)"=="" && "$(verbose)"==""
+QUIET_BUILD*=@
+.ELSE
+QUIET_BUILD*=
+.ENDIF
+
 # ===========================================================================
 # unter NT werden Variablennamen an untergeordnete makefiles UPPERCASE
 # weitergereicht, aber case significant ausgewertet!
--- solenv/inc/target.mk.old	2009-04-02 10:53:10.000000000 +0000
+++ solenv/inc/target.mk	2009-04-06 16:41:40.000000000 +0000
@@ -1686,7 +1686,10 @@ ZIPDEPPHONY=.PHONY
 .ENDIF
 
 last_target:
-	@echo -------------
+.IF "$(QUIET_BUILD)"==""
+	@echo ------------- 
+.ENDIF
+	$(NULL)
 
 $(MISC)$/$(TARGET)genjava.mk: 	$(IDLFILES)
 
