--- dbaccess/source/ui/dlg/ExtensionNotPresent.cxx.old	2009-04-02 10:41:31.000000000 +0000
+++ dbaccess/source/ui/dlg/ExtensionNotPresent.cxx	2009-04-06 16:41:41.000000000 +0000
@@ -201,7 +201,7 @@ IMPL_LINK( OExtensionNotPresentDialog, D
         if (suDownloadURL.getLength() == 0)
         {
             // fallback
-            suDownloadURL = UNISTRING("http://extensions.services.openoffice.org");
+            suDownloadURL = UNISTRING("http://extensions.go-oo.org");
         }
         
         // open such URL in a browser
--- reportbuilder/registry/schema/org/openoffice/Office/ReportDesign.xcs.old	2009-04-02 10:35:53.000000000 +0000
+++ reportbuilder/registry/schema/org/openoffice/Office/ReportDesign.xcs	2009-04-06 16:41:41.000000000 +0000
@@ -55,7 +55,7 @@
                     <author>LLA</author>
                     <desc>Direct download URL to the Sun(tm) Report Builder extension.</desc>
                 </info>
-                <value>http://extensions.services.openoffice.org</value>
+                <value>http://extensions.go-oo.org</value>
             </prop>
             <prop oor:name="Name" oor:type="xs:string">
                 <info>
--- officecfg/registry/data/org/openoffice/Office/ExtensionManager.xcu.old	2009-04-02 11:02:04.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/ExtensionManager.xcu	2009-04-06 16:41:41.000000000 +0000
@@ -33,7 +33,7 @@
 <oor:component-data oor:name="ExtensionManager" oor:package="org.openoffice.Office" xmlns:oor="http://openoffice.org/2001/registry">
   <node oor:name="ExtensionRepositories">
     <prop oor:name="WebsiteLink">
-      <value>http://extensions.services.openoffice.org/getmore?cid=920794</value>
+      <value>http://extensions.go-oo.org/getmore?cid=920794</value>
     </prop>
   </node>
 </oor:component-data> 
--- sfx2/source/appl/appserv.cxx.old	2009-04-02 10:44:02.000000000 +0000
+++ sfx2/source/appl/appserv.cxx	2009-04-06 16:41:41.000000000 +0000
@@ -873,7 +873,7 @@ void SfxApplication::OfaExec_Impl( SfxRe
                     xSMGR->createInstance( DEFINE_CONST_UNICODE("com.sun.star.system.SystemShellExecute" ) ),
                     uno::UNO_QUERY_THROW );
                 if ( xSystemShell.is() )
-                    xSystemShell->execute( DEFINE_CONST_UNICODE("http://extensions.services.openoffice.org/dictionary?cid=926385"), ::rtl::OUString(), css::system::SystemShellExecuteFlags::DEFAULTS );
+                    xSystemShell->execute( DEFINE_CONST_UNICODE("http://extensions.go-oo.org/dictionary?cid=926385"), ::rtl::OUString(), css::system::SystemShellExecuteFlags::DEFAULTS );
             }
             catch( const ::com::sun::star::uno::Exception& )
             {
--- svx/source/options/optlingu.cxx.old	2009-04-02 11:03:06.000000000 +0000
+++ svx/source/options/optlingu.cxx	2009-04-06 16:41:41.000000000 +0000
@@ -1177,7 +1177,7 @@ SvxLinguTabPage::SvxLinguTabPage( Window
     aLinguOptionsCLB.SetDoubleClickHdl(LINK(this, SvxLinguTabPage, BoxDoubleClickHdl_Impl));
 
     aMoreDictsLink.SetURL( String(
-        RTL_CONSTASCII_STRINGPARAM( "http://extensions.services.openoffice.org/dictionary?cid=926386" ) ) );
+        RTL_CONSTASCII_STRINGPARAM( "http://extensions.go-oo.org/dictionary?cid=926386" ) ) );
     aMoreDictsLink.SetClickHdl( LINK( this, SvxLinguTabPage, OpenURLHdl_Impl ) );
 
 
@@ -2065,7 +2065,7 @@ SvxEditModulesDlg::SvxEditModulesDlg(Win
     aPrioDownPB.Enable( FALSE );
 
     aMoreDictsLink.SetURL( String(
-        RTL_CONSTASCII_STRINGPARAM( "http://extensions.services.openoffice.org/dictionary?cid=926386" ) ) );
+        RTL_CONSTASCII_STRINGPARAM( "http://extensions.go-oo.org/dictionary?cid=926386" ) ) );
     aMoreDictsLink.SetClickHdl( LINK( this, SvxEditModulesDlg, OpenURLHdl_Impl ) );
 
     //
--- svtools/source/contnr/templwin.cxx.old	2009-04-02 11:06:15.000000000 +0000
+++ svtools/source/contnr/templwin.cxx	2009-04-06 16:41:41.000000000 +0000
@@ -1743,7 +1743,7 @@ void SvtDocumentTemplateDialog::InitImpl
 	pImpl->aTitle = GetText();
 
     aMoreTemplatesLink.SetURL( String(
-        RTL_CONSTASCII_STRINGPARAM( "http://templates.services.openoffice.org/?cid=923508" ) ) );
+        RTL_CONSTASCII_STRINGPARAM( "http://templates.go-oo.org/?cid=923508" ) ) );
     aMoreTemplatesLink.SetClickHdl( LINK( this, SvtDocumentTemplateDialog, OpenLinkHdl_Impl ) );
 
     aManageBtn.SetClickHdl( LINK( this, SvtDocumentTemplateDialog, OrganizerHdl_Impl ) );
--- instsetoo_native/util/openoffice.lst.old	2009-04-02 10:36:19.000000000 +0000
+++ instsetoo_native/util/openoffice.lst	2009-04-06 16:41:41.000000000 +0000
@@ -88,9 +88,9 @@ OpenOffice
 			DOWNLOADBITMAP	ooobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             REGISTRATION_PRODUCT OpenOffice.org
             REGISTRATION_CID 926117
 		}
@@ -152,9 +152,9 @@ OpenOffice_wJRE
 			DOWNLOADBITMAP	ooobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             REGISTRATION_PRODUCT OpenOffice.org
             REGISTRATION_CID 926117
 		}
@@ -223,9 +223,9 @@ OpenOffice_Dev
 			DOWNLOADSETUPICO	ooosetup.ico
 			LOCALUSERDIR $ORIGIN/..
 			RELATIVE_PATHES_IN_DDF 1
-			STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-			STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+			STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+			STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             REGISTRATION_PRODUCT OpenOffice.org
             REGISTRATION_CID 926117
 		}
@@ -265,9 +265,9 @@ URE
             DONTUSESTARTMENUFOLDER 1
             SERVICESPROJEKT 1
             RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             ADD_INCLUDE_FILES cliureversion.mk,clioootypesversion.mk
 			PACKAGEMAP package_names_ext.txt
         }
@@ -309,9 +309,9 @@ OpenOffice_SDK
             DOWNLOADSETUPICO ooosetup.ico
             DONTUSESTARTMENUFOLDER 1
             RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
         }
         active 1
         compression 5
@@ -372,9 +372,9 @@ BrOffice
 			DOWNLOADBITMAP	brobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             REGISTRATION_PRODUCT OpenOffice.org
             REGISTRATION_CID 926117
 		}
@@ -440,9 +440,9 @@ BrOffice_wJRE
 			DOWNLOADBITMAP	brobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             REGISTRATION_PRODUCT OpenOffice.org
             REGISTRATION_CID 926117
 		}
@@ -514,9 +514,9 @@ BrOffice_Dev
 			DOWNLOADSETUPICO	ooosetup.ico
 			LOCALUSERDIR $ORIGIN/..
 			RELATIVE_PATHES_IN_DDF 1
-            STARTCENTER_ADDFEATURE_URL http://extensions.services.openoffice.org/getmore?cid=920794
-            STARTCENTER_INFO_URL http://www.openoffice.org
-            STARTCENTER_TEMPLREP_URL http://extensions.services.openoffice.org/templates?cid=926383
+            STARTCENTER_ADDFEATURE_URL http://extensions.go-oo.org/getmore?cid=920794
+            STARTCENTER_INFO_URL http://www.go-oo.org
+            STARTCENTER_TEMPLREP_URL http://extensions.go-oo.org/templates?cid=926383
             REGISTRATION_PRODUCT OpenOffice.org
             REGISTRATION_CID 926117
 		}
