diff --git a/automation/source/app/testbasi.cxx b/automation/source/app/testbasi.cxx
index d5d8ee2..3f77243 100644
--- automation/source/app/testbasi.cxx
+++ automation/source/app/testbasi.cxx
@@ -152,9 +152,9 @@ void TTBasic::ReportRuntimeError( AppBasEd *pEditWin )
 {
 	SbxVariableRef aDummy = new SbxVariable;
 	aDummy->SetUserData( 24 );  // ID_MaybeAddErr
-	((TestToolObj*)pTestObject)->SFX_NOTIFY( pTestObject->GetBroadcaster(), xx, SbxHint( SBX_HINT_DATAWANTED, aDummy ), xx );
+	((TestToolObj*)pTestObject)->Notify( pTestObject->GetBroadcaster(), SbxHint( SBX_HINT_DATAWANTED, aDummy ) );
 	aDummy->SetUserData( 18 );  // ID_ExceptLog
-	((TestToolObj*)pTestObject)->SFX_NOTIFY( pTestObject->GetBroadcaster(), xx, SbxHint( SBX_HINT_DATAWANTED, aDummy ), xx );
+	((TestToolObj*)pTestObject)->Notify( pTestObject->GetBroadcaster(), SbxHint( SBX_HINT_DATAWANTED, aDummy ) );
 	MyBasic::ReportRuntimeError( pEditWin );
 }
 
