Index: source/ui/dlg/ExtensionNotPresent.src
===================================================================
RCS file: /cvs/dba/dbaccess/source/ui/dlg/ExtensionNotPresent.src,v
retrieving revision 1.3
diff -u -u -r1.3 ExtensionNotPresent.src
--- dbaccess/source/ui/dlg/ExtensionNotPresent.src	2 Aug 2007 14:26:39 -0000	1.3
+++ dbaccess/source/ui/dlg/ExtensionNotPresent.src	8 Aug 2007 12:34:08 -0000
@@ -37,8 +37,8 @@
 String RID_STR_EXTENSION_NOT_PRESENT
 {
 	// #i96130# use hard coded name
-	Text [ en-US ] = "To open a report you require the extension Sun™ Report Builder.\n\nClick 'Download...' to download and install the extension.";
-	// OLD: Text [ en-US ] = "To open a report you require the extension %RPT_EXTENSION_NAME.\n\nClick 'Download...' to download and install the extension.";
+	Text [ en-US ] = "To open a report you require the extension Sun™ Report Builder.\n\nor this, install the package 'openoffice.org-report-builder'.";
+	// OLD: Text [ en-US ] = "To open a report you require the extension %RPT_EXTENSION_NAME.\n\nor this, install the package 'openoffice.org-report-builder'.";
 };
 // To open a report you require the extension Sun Report Designer weiss der Geier Hauptsache extra langer Name
 // String RID_STR_EXTENSION_NAME
@@ -73,15 +73,6 @@
         // Text will set outside from RID_STR_EXTENSION_NOT_PRESENT
 	};
 
-	PushButton PB_DOWNLOAD
-	{
-		Pos = MAP_APPFONT ( DLG_WIDTH / 2 - (CELL_PADDING/2) - BUTTON_WIDTH, ACTION_LINE_START ) ;
-		Size = MAP_APPFONT ( BUTTON_WIDTH , BUTTON_HEIGHT ) ;
-		DefButton = TRUE ;
-		TabStop = TRUE ;
-		Text [ en-US ] = "~Download..." ;
-	};
-
 	CancelButton PB_CANCEL
 	{
 		Pos = MAP_APPFONT ( DLG_WIDTH / 2 + (CELL_PADDING/2), ACTION_LINE_START) ;
--- dbaccess/source/ui/dlg/localize.sdf-old	2008-04-29 12:45:07.000000000 +0200
+++ dbaccess/source/ui/dlg/localize.sdf	2008-04-29 12:47:49.000000000 +0200
@@ -8481,62 +8481,9 @@
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	vi	Tải ~về...				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	zh-CN	下载(~D)...				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	pushbutton	RID_EXTENSION_NOT_PRESENT_DLG	PB_DOWNLOAD			0	zh-TW	下載(~D)...				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ar	لفتح تقرير، تحتاج إلى الامتداد Sun™ Report Builder.\n\nانقر فوق "تنزيل..." لتنزيل الامتداد وتثبيته.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	as-IN	এটা প্ৰতিবেদন খুলিবলৈ আপোনাক Sun™ Report Builder সম্প্ৰসাৰণৰ প্ৰয়োজন ।\n\nসম্প্ৰসাৰণ ডাউন্‌লোড কৰি তাক সংস্থাপন কৰিবলৈ 'ডাউন্‌লোড কৰক...' টিপক ।				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	be-BY	Каб адкрыць паведамленне, патрэбнае пашырэнне %RPT_EXTENSION_NAME.\n\nНацісніце 'Атрымаць...' каб атрымаць і ўстанавіць пашырэнне.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	bg	За да отворите справка, ви е необходимо разширението Sun™ Report Builder.\n\nНатиснете "Изтегляне...", за да изтеглите и инсталирате разширението.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	bn	একটি প্রতিবেদন খুলতে আপনার সমপ্রসারন Sun™ প্রতিবেদন প্রস্তুতকারকের প্রয়োজন।\n\nসমপ্রসারন ডাউনলোড এবং সংস্থাপন করতে 'ডাউনলোড...' এ ক্লিক করুন।				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	br	Evit digeriñ ur marilh ho po ezhomm eus an askouezhad %RPT_EXTENSION_NAME.\n\nKlikañ war 'Pellgargañ...' evit pellgargañ ha staliañ an askouezhad.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ca	Per a obrir un informe necessiteu l'extensió Constructor d'informes de Sun™.\n\nFeu clic a "Baixa..." per a baixar i instal·lar l'extensió.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	cs	Pro otevření sestavy je potřeba rozšíření Sun™ Report Builder.\n\nKlepněte na 'Stáhnout...', pokud chcete rozšíření stáhnout a nainstalovat.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	da	For at åbne en rapport kræves udvidelsen Sun™ Report Builder.\n\nKlik på 'Hent...' for at hente og installere udvidelsen.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	de	Zum Öffnen eines Berichts wird die Erweiterung Sun™ Report Builder benötigt.\n\nKlicken Sie auf 'Download...', um die Erweiterung herunterzuladen und zu installieren.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	el	Για να ανοίξετε μια αναφορά χρειάζεστε την επέκταση Sun™ Report Builder.\n\nΠατήστε 'Κατέβασμα...' για να κατεβάσετε και να εγκαταστήσετε την επέκταση.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	en-GB	To open a report you require the extension Sun™ Report Builder.\n\nClick 'Download...' to download and install the extension.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	eo	Por malfermi raporton vi bezonas la etendajxon %RPT_EXTENSION_NAME.\n\nAlklaku 'Elsxuti...' por elsxuti kaj instali la etendajxon.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	es	Para abrir un informe se requiere la extensión Sun™ Report Builder.\n\nHaga clic en 'Descargar...' para descargar e instalar la extensión.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	et	Aruande avamiseks on vaja laiendust Sun™-i aruandekoostaja.\n\nLaienduse allalaadimiseks ja paigaldamiseks klõpsa 'Laadi alla...'.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	eu	Txosten bat irekitzeko 'Sun™ txostenen diseinatzailea'\n(Sun™ Report Builder) hedapena behar duzu.\n\nEgin klik "Deskargatu..." hedapena deskargatu eta instalatzeko.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	fi	Avataksesi raportin tarvitset Sun™ Report Builder -lisäosan.\n\nNapsauta 'Lataa...' ladataksesi ja asentaaksesi sen.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	fr	L'ouverture d'un rapport requiert l'extension Sun™ Report Builder.\n\nCliquez sur 'Télécharger...' pour télécharger et installer l'extension.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	gl	Para abrir un informe necesita a extensión Sun™ Report Builder.\n\nPrema 'Descargar...' para descargar e instalar a extensión.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	gu	અહેવાલ ખોલવા માટે તમને એક્સટેન્સન Sun™ અહેવાલ બિલ્ડરની જરૂર છે.\n\nએક્સટેન્સન ડાઉનલોડ કરવા અને સ્થાપિત કરવા માટે 'ડાઉનલોડ...' ક્લિક કરો.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	hi-IN	एक रिपोर्ट खोलने के लिए आपको Sun™ Report Builder विस्तार की जरूरत है.\n\n'डाउनलोड करें...' को विस्तार को डाउनलोड और संस्थापित करने के लिए क्लिक करें.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	hr	Za otvaranje izvješća morate imati %RPT_EXTENSION_NAME proširenje.\n\nKliknite na  'Preuzmi...' kako bi preuzeli i instalirali proširenje.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	hu	Jelentés megnyitásához a Sun™ Report Builder kiterjesztésre van szükség.\n\nKattintson a „Letöltés...” lehetőségre a kiterjesztés letöltéséhez és telepítéséhez.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	id	Untuk membuka laporan, Anda memerlukan ekstensi Sun™ Report Builder.\n\nKlik 'Download...' untuk mendapatkan dan memasang ekstensi.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	it	Per aprire un rapporto è necessaria l'estensione Sun™ Report Builder.\n\nFate clic su 'Scarica...' per scaricare e installare l'estensione.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ja	レポートを開くには、拡張機能 Sun™ Report Builder が必要です。\n\n「ダウンロード...」をクリックすると、拡張機能のダウンロードとインストールができます。				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	km	ដើម្បី​បើក​របាយការណ៍ គឺ​អ្នក​ត្រូវការ​ផ្នែក​បន្ថែម Sun™ Report Builder ។\n\nចុច 'ទាញយក...' ដើម្បី​ទាញយក និង​ដំឡើង​ផ្នែក​បន្ថែម ។				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	kn-IN	ಒಂದು ವರದಿಯನ್ನು ತೆರೆಯಲು ನಿಮ್ಮಲ್ಲಿ Sun™ ರಿಪೋರ್ಟ್ ಬಿಲ್ಡರ್ ಎನ್ನುವ ಒಂದು ವಿಸ್ತರಣೆಯು ಇರಬೇಕು.\n\nಈ ವಿಸ್ತರಣೆಯನ್ನು ಡೌನ್‌ಲೋಡ್‌ ಮಾಡಿ ನಂತರ ಅನುಸ್ಥಾಪಿಸಲು 'ಡೌನ್‌ಲೋಡ್‌ ಮಾಡು...' ಅನ್ನು ಕ್ಲಿಕ್ಕಿಸಿ.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ko	보고서를 열려면 Sun™ Report Builder 확장이 필요합니다.\n\n'다운로드...'를 클릭하고 확장 기능을 다운로드하여 설치합니다.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	lt	Ataskaitai atverti reikalingas plėtinys „Sun™ Report Builder“.\n\nSpustelėkite „Atsisiųsti...“, kad atsisiųstumėte ir įdiegtumėte plėtinį.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	mk	За да го отворите извештајот потребна ви е екстензијата Sun™ Report Builder.\n\nКликнете на 'Симни...' за да ја симнете и инсталирате екстензијата.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	mn	Тайлан нээхийн тулд танд %RPT_EXTENSION_NAME өргөтгөл шаардлагатай.\n\nТатаж авах болон өргөтгөлийг суулгахдаа 'Татах...' товшино уу.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	mr-IN	अहवाल उघडण्यासाठी आपणास एक्सटेन्शन  %RPT_EXTENSION_NAME ची आवश्यकता आहे.\n\nएक्सटेन्शला उतरवून घेण्यासाठी  'उतरवून घ्या...' वर क्लिक करा आणि एक्सटेन्शला स्थापित करा.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	nb	Du trenger utvidelsen %RPT_EXTENSION_NAME for å åpne rapporter.\n\nTrykk på «Last ned» for å laste ned og installere utvidelsen.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ne	प्रतिवेदन खोल्न तपाईँलाई विस्तार %RPT_EXTENSION_NAME आवश्यक पर्दछ ।\n\nविस्तार डाउनलोड र स्थापना गर्न 'डाउनलोड...' क्लिक गर्नुहोस् ।				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	nl	U hebt de extensie Sun™ Report Builder nodig om een rapport te kunnen openen.\n\nKlik op 'Downloaden...' om de extensie te downloaden en te installeren.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	nn	Du treng utvidinga %RPT_EXTENSION_NAME for å opna rapportar.\n\nTrykk på «Last ned» for å lasta ned og installera utvidinga.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pa-IN	ਇੱਕ ਰਿਪੋਰਟ ਖੋਲ੍ਹਣ ਵਾਸਤੇ ਇਕਸਟੈਨਸ਼ਨ %RPT_EXTENSION_NAME ਲੋੜੀਦੀ ਹੈ।\n\nਇਕਸਟੈਨਸ਼ਨ ਡਾਊਨਲੋਡ ਅਤੇ ਇੰਸਟਾਲ ਕਰਨ ਵਾਸਤੇ 'ਡਾਊਨਲੋਡ...' ਉੱਤੇ ਕਲਿੱਕ ਕਰੋ।				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pl	Aby otworzyć raport, musisz skorzystać z rozszerzenia Sun™ Report Builder.\n\nKliknij przycisk "Pobierz...", aby pobrać i zainstalować rozszerzenie.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pt	Para abrir um relatório é necessária a extensão do Sun™ Report Builder.\n\nFaça clique em 'Transferir de...' para transferir e instalar a extensão.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	pt-BR	Para abrir um relatório, é necessário a extensão Sun™ Report Builder.\n\nClique em 'Download...' para fazer o download e instalar a extensão.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	ru	Для редактирования отчета требуется расширение Sun™ Report Builder.\n\nЩелкните  "Загрузить..." для загрузки и установки расширения.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	rw	Kugira ngo ufungure raporo usaba umugereka %RPT_EXTENSION_NAME.\n\nKanda 'Gukurura...' kugira ngo ukurure uninjizemo umugereka.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	sa-IN	रिपोर्ट् उद्घाटितुं तुभ्यं %RPT_EXTENSION_NAME विस्तारणम् आवश्यकम्।\n\nविस्तारणं संस्थापितिम् अथवा डौन्लोड् कर्तुं 'डोन्लोड्...' क्लिक् कुरु।				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	sh	Za otvaranje izveštaja potreban je dodatak Sun-ov konstruktor izveštaja.\n\nKliknite „Preuzmi...“ kako biste preuzeli i instalirali dodatak.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				1	sk	Pre otvorenie správy potrebujete rozšírenie Sun™ Report Builder.\n\nKliknite na 'Stiahnuť...' pre stahnutie a inštaláciu rozšírenia.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	sl	Če želite odpreti poročilo, potrebujete razširitev Oblikovalec poročil podjetja Sun™.\n\nKliknite 'Prenesi ...', če želite prenesti in namestiti razširitev.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	sr	За отварање извештаја потребан је додатак Sun-ов конструктор извештаја.\n\nКликните „Преузми...“ како бисте преузели и инсталирали додатак.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	sv	För att öppna en rapport behöver du tillägget Sunâ„¢ Rapportdesign.\n\nKlicka 'Hämta...' för att hämta och installera tillägget.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	ta	ஒரு அறிக்கையைத் திறக்க Sun™ Report Builder நீட்சி உங்களுக்குத் தேவை.\n\nநீட்சியை பதிவிறக்கி நிறுவ, 'பதிவிறக்கு...'ஐச் சொடுக்கவும்.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	te-IN	ఒక నివేదికను తెరువుటకు మీకు Sun™ రిపోర్టు బల్డర్ పొడిగింపు అవసరము.\n\nపొడిగింపును డౌన్‌లోడ్ చేసి సంస్థాపించుటకు 'డౌన్‌లోడ్...' నొక్కుము.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	tr	Raporu açmak için Sun™ Rapor Oluşturucu gereklidir.\n\nEklentiyi indirip yüklemek için "İndir..." düğmesine tıklayın.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	uk	Для редагування звіту потрібне розширення %RPT_EXTENSION_NAME.\n\nНатисніть  "Завантажити..." для завантаження та встановлення розширення.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	vi	Để mở một báo cáo bạn cần phần mở rộng Sun™ Report Builder.\n\nNhắp vào « Tải về... » để tải về và cài đặt phần mở rộng này.				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	zh-CN	为打开报表您需要 Sun™ 报表构建器扩展。\n\n请单击“下载...”以下载和安装该扩展。				2002-02-02 02:02:02
-dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	zh-TW	您需要有 Sun™ Report Builder 擴充軟體才能開啟報告。\n\n請按一下 [下載...] 以下載及安裝此擴充軟體。				2002-02-02 02:02:02
+dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	cs	Prootevření sestavy je potřeba rozšíření Sun™ Report Builder.\n\nProsím nainstalujte si balík 'openoffice.org-report-builder'.				2002-02-02 02:02:02
+dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				13691	de	Zum Öffnen eines Berichts wird die Erweiterung Sun™ Report Builder benötigt.\n\nInstallieren Sie dafür das Paket 'openoffice.org-report-builder'.				2002-02-02 02:02:02
+dbaccess	source\ui\dlg\ExtensionNotPresent.src	0	string	RID_STR_EXTENSION_NOT_PRESENT				0	fr	L'ouverture d'un rapport requiert l'extension %RPT_EXTENSION_NAME.\n\nPour cela, installez le package 'openoffice.org-report-builder'.				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	af	Skrapopsies				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	ar	خيارات الحذف				2002-02-02 02:02:02
 dbaccess	source\ui\dlg\RelationDlg.src	0	fixedline	DLG_REL_PROPERTIES	FL_CASC_DEL			84	as-IN	বিকল্পবোৰ ডিলিট কৰক				2002-02-02 02:02:02
