--- reportdesign/util/makefile.mk-old	2008-01-18 12:56:07.000000000 +0100
+++ reportdesign/util/makefile.mk	2008-01-18 12:55:20.000000000 +0100
@@ -225,7 +225,7 @@
 			$(ZIP1DIR)$/readme_en-US.html \
 			$(ZIP1DIR)$/readme_en-US.txt
 
-REPRORTJARFILES := \
+REPORTJARFILES := \
 	$(ZIP1DIR)$/flute-1.3-jfree-20061107.jar							\
 	$(ZIP1DIR)$/jcommon-1.0.10.jar										\
 	$(ZIP1DIR)$/jcommon-serializer-0.2.0.jar							\
