Index: util/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/util/makefile.mk,v
retrieving revision 1.3.4.1.4.4
diff -u -u -r1.3.4.1.4.4 makefile.mk
--- reportdesign/util/makefile.mk	27 Aug 2007 08:18:57 -0000	1.3.4.1.4.4
+++ reportdesign/util/makefile.mk	30 Aug 2007 14:55:14 -0000
@@ -225,14 +225,6 @@
 			$(ZIP1DIR)$/readme_en-US.html \
 			$(ZIP1DIR)$/readme_en-US.txt
 
-.IF "$(GUI)"!="WNT"
-TXTFILES:=$(foreach,i,$(alllangiso) $(ZIP1DIR)$/registration$/LICENSE_$i)
-LICLINES:=$(foreach,i,$(TXTFILES)  <license-text xlink:href="registration/$(i:f)" lang="$(subst,LICENSE_, $(i:f))" license-id="$(subst,LICENSE_, $(i:f))" />)
-.ELSE   # "$(GUI)"!="WNT"
-TXTFILES:=$(foreach,i,$(alllangiso) $(ZIP1DIR)$/registration$/license_$i.txt)
-LICLINES:=$(foreach,i,$(TXTFILES)  <license-text xlink:href="registration/$(i:f)" lang="$(subst,.txt, $(subst,license_, $(i:f)))" license-id="$(subst,.txt, $(subst,license_, $(i:f)))" />)
-.ENDIF  # "$(GUI)"!="WNT"
-
 REPORTJARFILES := \
 	$(ZIP1DIR)$/jcommon-1.0.10.jar										\
 	$(ZIP1DIR)$/sac.jar													\
@@ -251,7 +243,7 @@
 .INCLUDE : target.mk
 
 .IF "$(ZIP1TARGETN)"!=""
-$(ZIP1TARGETN) :  $(TXTFILES) $(XMLFILES) $(HTMLFILES) $(REPORTJARFILES)
+$(ZIP1TARGETN) :  $$(XMLFILES) $(HTMLFILES) $(REPORTJARFILES)
 .ENDIF          # "$(ZIP1TARGETN)"!="
 
 $(MISC)$/update_report.flag : $(XCU_FILES)
@@ -261,17 +253,8 @@
 	@@-$(MKDIRHIER) $(@:d)
 	@@-$(RM) $(ZIP1DIR)$/description.xml
 	$(TYPE) pre.xml > $@
-	$(TYPE) $(mktmp  $(LICLINES)) >> $@
 	$(TYPE) post.xml >> $@
 
-$(ZIP1DIR)$/registration$/license_%.txt : $(SOLARBINDIR)$/osl$/license_%.txt
-     @@-$(MKDIRHIER) $(@:d)
-    $(COPY) $< $@
-
-$(ZIP1DIR)$/registration$/LICENSE_% : $(SOLARBINDIR)$/osl$/LICENSE_%
-     @@-$(MKDIRHIER) $(@:d)
-    $(COPY) $< $@
-
 $(ZIP1DIR)$/%.xml : %.xml
 	@@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
Index: util/post.xml
===================================================================
RCS file: /cvs/dba/reportdesign/util/post.xml,v
retrieving revision 1.2
diff -u -u -r1.2 post.xml
--- reportdesign/util/post.xml	2 Aug 2007 14:43:01 -0000	1.2
+++ reportdesign/util/post.xml	30 Aug 2007 14:55:14 -0000
@@ -1,5 +1,3 @@
-    </simple-license>
-  </registration>
   <version value="1.0.1"/>
   <identifier value="com.sun.reportdesigner"/>
   <dependencies>
Index: util/pre.xml
===================================================================
RCS file: /cvs/dba/reportdesign/util/pre.xml,v
retrieving revision 1.2.4.1.4.1
diff -u -u -r1.2.4.1.4.1 pre.xml
--- reportdesign/util/pre.xml	27 Aug 2007 05:33:34 -0000	1.2.4.1.4.1
+++ reportdesign/util/pre.xml	30 Aug 2007 14:55:14 -0000
@@ -2,5 +2,3 @@
 <description xmlns="http://openoffice.org/extensions/description/2006"
     xmlns:d="http://openoffice.org/extensions/description/2006" 
     xmlns:xlink="http://www.w3.org/1999/xlink">
-  <registration>
-    <simple-license accept-by="admin" default-license-id="en-US" >
