diff --git a/sal/osl/unx/diagnose.c b/sal/osl/unx/diagnose.c
index c95ed40..57ab708 100644
--- sal/osl/unx/diagnose.c
+++ sal/osl/unx/diagnose.c
@@ -247,8 +247,9 @@ sal_Bool SAL_CALL osl_assertFailedLine (
 	/* output message buffer */
     OSL_DIAGNOSE_OUTPUTMESSAGE(f, szMessage);
 
-	/* output backtrace */
-	osl_diagnose_backtrace_Impl(f);
+    /* if SAL backtrace for assertions is disabled, skip here */
+    if ( !getenv("DISABLE_SAL_BACKTRACE") )
+        osl_diagnose_backtrace_Impl(f);
 
 	/* release lock and leave, w/o calling osl_breakDebug() */
 	pthread_mutex_unlock(&g_mutex);
