--- sal/osl/unx/security.c	10 Apr 2008 11:12:04 -0000	1.27
+++ sal/osl/unx/security.c	21 May 2008 15:49:31 -0000
@@ -787,7 +787,14 @@ sal_Bool SAL_CALL osl_getConfigDir(oslSe
 
 static sal_Bool SAL_CALL osl_psz_getConfigDir(oslSecurity Security, sal_Char* pszDirectory, sal_uInt32 nMax)
 {
-	return (osl_psz_getHomeDir(Security, pszDirectory, nMax));
+    sal_Char *pStr = getenv("XDG_CONFIG_HOME");
+    
+    if ((pStr == NULL) || (strlen(pStr) == 0) ||
+        (access(pStr, 0) != 0))
+        return (osl_psz_getHomeDir(Security, pszDirectory, nMax));
+
+    strncpy(pszDirectory, pStr, nMax);
+    return sal_True;
 }
 
 #else
