diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/inc/output.hxx sc/source/ui/inc/output.hxx
--- sc.clean/source/ui/inc/output.hxx	2008-03-03 11:24:45.000000000 -0500
+++ sc/source/ui/inc/output.hxx	2008-03-14 19:18:50.000000000 -0400
@@ -237,6 +237,7 @@ public:
 
 	void	DrawGrid( BOOL bGrid, BOOL bPage );
 	void	DrawStrings( BOOL bPixelToLogic = FALSE );
+    void    DrawDocumentBackground();
 	void	DrawBackground();
 	void	DrawShadow();
 	void	DrawExtraShadow(BOOL bLeft, BOOL bTop, BOOL bRight, BOOL bBottom);
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/view/gridwin4.cxx sc/source/ui/view/gridwin4.cxx
--- sc.clean/source/ui/view/gridwin4.cxx	2008-03-03 11:24:49.000000000 -0500
+++ sc/source/ui/view/gridwin4.cxx	2008-03-14 19:18:50.000000000 -0400
@@ -721,9 +721,10 @@ void ScGridWindow::Draw( SCCOL nX1, SCRO
 		aOutputData.SetSolidBackground(TRUE);
 
 	pContentDev->SetMapMode(MAP_PIXEL);
-	aOutputData.DrawBackground();
+    aOutputData.DrawDocumentBackground();
 	if ( bGridFirst && ( bGrid || bPage ) )
 		aOutputData.DrawGrid( bGrid, bPage );
+	aOutputData.DrawBackground();
 	if ( bPageMode )
 	{
 		// #87655# DrawPagePreview draws complete lines/page numbers, must always be clipped
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/view/output.cxx sc/source/ui/view/output.cxx
--- sc.clean/source/ui/view/output.cxx	2008-03-03 11:24:49.000000000 -0500
+++ sc/source/ui/view/output.cxx	2008-03-14 19:21:04.000000000 -0400
@@ -796,14 +796,23 @@ BOOL lcl_EqualBack( const RowInfo& rFirs
 	return TRUE;
 }
 
-void ScOutputData::DrawBackground()
+void ScOutputData::DrawDocumentBackground()
 {
-	FindRotated();				//! von aussen ?
+    if ( !bSolidBackground )
+        return;
 
-	ScModule* pScMod = SC_MOD();
+    Size aOnePixel = pDev->PixelToLogic(Size(1,1));
+    long nOneX = aOnePixel.Width();
+    long nOneY = aOnePixel.Height();
+    Rectangle aRect(nScrX - nOneX, nScrY - nOneY, nScrX + nScrW, nScrY + nScrH);
+    Color aBgColor( SC_MOD()->GetColorConfig().GetColorValue(svtools::DOCCOLOR).nColor );
+    pDev->SetFillColor(aBgColor);
+    pDev->DrawRect(aRect);
+}
 
-	// used only if bSolidBackground is set (only for ScGridWindow):
-    Color aBgColor( pScMod->GetColorConfig().GetColorValue(svtools::DOCCOLOR).nColor );
+void ScOutputData::DrawBackground()
+{
+	FindRotated();				//! von aussen ?
 
 	Rectangle aRect;
 	Size aOnePixel = pDev->PixelToLogic(Size(1,1));
@@ -852,7 +861,7 @@ void ScOutputData::DrawBackground()
 				long nPosX = nScrX;
 				if ( bLayoutRTL )
 					nPosX += nMirrorW - nOneX;
-				aRect = Rectangle( nPosX,nPosY, nPosX,nPosY+nRowHeight-nOneY );
+				aRect = Rectangle( nPosX, nPosY-nOneY, nPosX, nPosY+nRowHeight-nOneY );
 
 				const SvxBrushItem* pOldBackground = NULL;
 				const SvxBrushItem* pBackground;
@@ -900,15 +909,13 @@ void ScOutputData::DrawBackground()
 						if (pOldBackground)				// ==0 if hidden
 						{
 							Color aBackCol = pOldBackground->GetColor();
-							if ( bSolidBackground && aBackCol.GetTransparency() )
-								aBackCol = aBgColor;
 							if ( !aBackCol.GetTransparency() )		//! partial transparency?
 							{
 								pDev->SetFillColor( aBackCol );
 								pDev->DrawRect( aRect );
 							}
 						}
-						aRect.Left() = nPosX;
+						aRect.Left() = nPosX - nSignedOneX;
 						pOldBackground = pBackground;
 					}
 					nPosX += pRowInfo[0].pCellInfo[nX+1].nWidth * nLayoutSign;
@@ -917,8 +924,6 @@ void ScOutputData::DrawBackground()
 				if (pOldBackground)
 				{
 					Color aBackCol = pOldBackground->GetColor();
-					if ( bSolidBackground && aBackCol.GetTransparency() )
-						aBackCol = aBgColor;
 					if ( !aBackCol.GetTransparency() )		//! partial transparency?
 					{
 						pDev->SetFillColor( aBackCol );
