diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/docoptio.hxx sc/inc/docoptio.hxx
--- sc.clean/inc/docoptio.hxx	2008-04-25 18:54:41.000000000 -0400
+++ sc/inc/docoptio.hxx	2008-04-25 19:01:33.000000000 -0400
@@ -36,6 +36,8 @@
 #include "scdllapi.h"
 #include "optutil.hxx"
 
+#include "grammar.hxx"
+
 class SC_DLLPUBLIC ScDocOptions
 {
 	double fIterEps;				// Epsilon-Wert dazu
@@ -53,6 +55,7 @@ class SC_DLLPUBLIC ScDocOptions
 	BOOL   bDoAutoSpell;			// Auto-Spelling
 	BOOL   bLookUpColRowNames;		// Spalten-/Zeilenbeschriftungen automagisch suchen
     BOOL   bFormulaRegexEnabled;    // regular expressions in formulas enabled
+    ScGrammar::Grammar eFormulaGrammar;  // formula grammar used to switch different formula syntax
 
 public:
 				ScDocOptions();
@@ -101,6 +104,9 @@ public:
 
     void    SetFormulaRegexEnabled( BOOL bVal ) { bFormulaRegexEnabled = bVal; }
     BOOL    IsFormulaRegexEnabled() const       { return bFormulaRegexEnabled; }
+
+    void SetFormulaSyntax( ScGrammar::Grammar eGram ) { eFormulaGrammar = eGram; }
+    ScGrammar::Grammar GetFormulaSyntax() const { return eFormulaGrammar; }
 };
 
 
@@ -121,6 +127,7 @@ inline void ScDocOptions::CopyTo(ScDocOp
 	rOpt.bDoAutoSpell			= bDoAutoSpell;
 	rOpt.bLookUpColRowNames		= bLookUpColRowNames;
     rOpt.bFormulaRegexEnabled   = bFormulaRegexEnabled;
+    rOpt.eFormulaGrammar        = eFormulaGrammar;
 }
 
 inline const ScDocOptions& ScDocOptions::operator=( const ScDocOptions& rCpy )
@@ -140,6 +147,7 @@ inline const ScDocOptions& ScDocOptions:
 	bDoAutoSpell		= rCpy.bDoAutoSpell;
 	bLookUpColRowNames	= rCpy.bLookUpColRowNames;
     bFormulaRegexEnabled= rCpy.bFormulaRegexEnabled;
+    eFormulaGrammar     = rCpy.eFormulaGrammar;
 
 	return *this;
 }
@@ -162,6 +170,7 @@ inline int ScDocOptions::operator==( con
 			&&	rOpt.bDoAutoSpell			== bDoAutoSpell
 			&&	rOpt.bLookUpColRowNames		== bLookUpColRowNames
             &&  rOpt.bFormulaRegexEnabled   == bFormulaRegexEnabled
+            &&  rOpt.eFormulaGrammar        == eFormulaGrammar
 			);
 }
 
Only in sc/inc: docoptio.hxx.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/data/documen3.cxx sc/source/core/data/documen3.cxx
--- sc.clean/source/core/data/documen3.cxx	2008-04-25 18:54:50.000000000 -0400
+++ sc/source/core/data/documen3.cxx	2008-04-25 19:01:33.000000000 -0400
@@ -1757,6 +1757,7 @@ void ScDocument::SetDocOptions( const Sc
 	pFormatter->ChangeNullDate( d,m,y );
 	pFormatter->ChangeStandardPrec( (USHORT)rOpt.GetStdPrecision() );
 	pFormatter->SetYear2000( rOpt.GetYear2000() );
+    SetGrammar( rOpt.GetFormulaSyntax() );
 }
 
 const ScViewOptions& ScDocument::GetViewOptions() const
Only in sc/source/core/data: documen3.cxx.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/tool/docoptio.cxx sc/source/core/tool/docoptio.cxx
--- sc.clean/source/core/tool/docoptio.cxx	2008-04-25 18:54:50.000000000 -0400
+++ sc/source/core/tool/docoptio.cxx	2008-04-25 19:16:28.000000000 -0400
@@ -45,6 +45,7 @@
 #include "scresid.hxx"
 #include "sc.hrc"
 #include "miscuno.hxx"
+#include "global.hxx"
 
 using namespace utl;
 using namespace rtl;
@@ -100,7 +101,8 @@ ScDocOptions::ScDocOptions( const ScDocO
 			bMatchWholeCell( rCpy.bMatchWholeCell ),
 			bDoAutoSpell( rCpy.bDoAutoSpell ),
             bLookUpColRowNames( rCpy.bLookUpColRowNames ),
-            bFormulaRegexEnabled( rCpy.bFormulaRegexEnabled )
+            bFormulaRegexEnabled( rCpy.bFormulaRegexEnabled ),
+            eFormulaGrammar( rCpy.eFormulaGrammar )
 {
 }
 
@@ -207,6 +209,7 @@ void ScDocOptions::ResetDocOptions()
 	bDoAutoSpell		= FALSE;
 	bLookUpColRowNames	= TRUE;
     bFormulaRegexEnabled= TRUE;
+    eFormulaGrammar     = ScGrammar::GRAM_DEFAULT;
 }
 
 //========================================================================
@@ -282,7 +285,8 @@ SfxPoolItem* __EXPORT ScTpCalcItem::Clon
 #define SCCALCOPT_SEARCHCRIT		9
 #define SCCALCOPT_FINDLABEL			10
 #define SCCALCOPT_REGEX             11
-#define SCCALCOPT_COUNT             12
+#define SCCALCOPT_FORMULA_SYNTAX    12
+#define SCCALCOPT_COUNT             13
 
 #define CFGPATH_DOCLAYOUT	"Office.Calc/Layout/Other"
 
@@ -305,7 +309,8 @@ Sequence<OUString> ScDocCfg::GetCalcProp
 		"Other/Precision",					// SCCALCOPT_PRECISION
 		"Other/SearchCriteria",				// SCCALCOPT_SEARCHCRIT
         "Other/FindLabel",                  // SCCALCOPT_FINDLABEL
-        "Other/RegularExpressions"          // SCCALCOPT_REGEX
+        "Other/RegularExpressions",         // SCCALCOPT_REGEX
+        "Other/FormulaSyntax"               // SCCALCOPT_FORMULA_SYNTAX
 	};
 	Sequence<OUString> aNames(SCCALCOPT_COUNT);
 	OUString* pNames = aNames.getArray();
@@ -398,6 +403,33 @@ ScDocCfg::ScDocCfg() :
                     case SCCALCOPT_REGEX :
                         SetFormulaRegexEnabled( ScUnoHelpFunctions::GetBoolFromAny( pValues[nProp] ) );
                         break;
+                    case SCCALCOPT_FORMULA_SYNTAX:
+                    {
+                        ScGrammar::Grammar eGram = ScGrammar::GRAM_DEFAULT;
+
+                        do
+                        {
+                            if (!(pValues[nProp] >>= nIntVal))
+                                // extractino failed.
+                                break;
+
+                            switch (nIntVal)
+                            {
+                                case 0: // Calc A1
+                                    eGram = ScGrammar::GRAM_NATIVE;
+                                break;
+                                case 1: // Excel A1
+                                    eGram = ScGrammar::GRAM_NATIVE_XL_A1;
+                                break;
+                                case 2: // Excel R1C1
+                                    eGram = ScGrammar::GRAM_NATIVE_XL_R1C1;
+                                break;
+                            }
+                        }
+                        while (false);
+                        SetFormulaSyntax(eGram);
+                    }
+                    break;
 				}
 			}
 		}
@@ -481,6 +513,16 @@ IMPL_LINK( ScDocCfg, CalcCommitHdl, void
 				break;
             case SCCALCOPT_REGEX :
                 ScUnoHelpFunctions::SetBoolInAny( pValues[nProp], IsFormulaRegexEnabled() );
+            case SCCALCOPT_FORMULA_SYNTAX :
+            {
+                sal_Int32 nVal = 0;
+                switch (GetFormulaSyntax())
+                {
+                    case ScGrammar::GRAM_NATIVE_XL_A1:    nVal = 1; break;
+                    case ScGrammar::GRAM_NATIVE_XL_R1C1:  nVal = 2; break;
+                }
+                pValues[nProp] <<= nVal;
+            }
 		}
 	}
 	aCalcItem.PutProperties(aNames, aValues);
Only in sc/source/core/tool: docoptio.cxx.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/inc/optdlg.hrc sc/source/ui/inc/optdlg.hrc
--- sc.clean/source/ui/inc/optdlg.hrc	2008-04-25 18:54:43.000000000 -0400
+++ sc/source/ui/inc/optdlg.hrc	2008-04-25 19:01:33.000000000 -0400
@@ -73,6 +73,8 @@
 #define BTN_MATCH			18
 #define BTN_LOOKUP			19
 #define BTN_REGEX           20
+#define FT_FORMULA_SYNTAX   21
+#define LB_FORMULA_SYNTAX   22
 
 // TP_VIEW:
 #define BTN_VSCROLL			1
Only in sc/source/ui/inc: optdlg.hrc.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/inc/tpcalc.hxx sc/source/ui/inc/tpcalc.hxx
--- sc.clean/source/ui/inc/tpcalc.hxx	2008-04-25 18:54:43.000000000 -0400
+++ sc/source/ui/inc/tpcalc.hxx	2008-04-25 19:01:33.000000000 -0400
@@ -39,6 +39,8 @@
 #include <svtools/stdctrl.hxx>
 #include "editfield.hxx"
 
+#include "vcl/lstbox.hxx"
+
 //===================================================================
 
 class ScDocOptions;
@@ -79,6 +81,8 @@ private:
 	CheckBox		aBtnMatch;
     CheckBox        aBtnRegex;
 	CheckBox		aBtnLookUp;
+    FixedText       aFtFormulaSyntax;
+    ListBox         aLBFormulaSyntax;
 
 	FixedText		aFtPrec;
 	NumericField	aEdPrec;
Only in sc/source/ui/inc: tpcalc.hxx.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/optdlg/tpcalc.cxx sc/source/ui/optdlg/tpcalc.cxx
--- sc.clean/source/ui/optdlg/tpcalc.cxx	2008-04-25 18:54:46.000000000 -0400
+++ sc/source/ui/optdlg/tpcalc.cxx	2008-04-25 19:01:33.000000000 -0400
@@ -91,6 +91,8 @@ ScTpCalcOptions::ScTpCalcOptions( Window
 		aBtnMatch	 	( this, ScResId( BTN_MATCH ) ),
         aBtnRegex       ( this, ScResId( BTN_REGEX ) ),
 		aBtnLookUp   	( this, ScResId( BTN_LOOKUP ) ),
+        aFtFormulaSyntax( this, ScResId( FT_FORMULA_SYNTAX ) ),
+        aLBFormulaSyntax( this, ScResId( LB_FORMULA_SYNTAX ) ),
 		aFtPrec 		( this, ScResId( FT_PREC ) ),
 		aEdPrec 		( this, ScResId( ED_PREC ) ),
 		pOldOptions 	( new ScDocOptions(
@@ -156,6 +158,22 @@ void __EXPORT ScTpCalcOptions::Reset( co
 	aEdPrec    .SetValue( pLocalOptions->GetStdPrecision() );
     aEdEps     .SetValue( pLocalOptions->GetIterEps(), 6 );
 
+    ScGrammar::Grammar eGram = pLocalOptions->GetFormulaSyntax();
+    switch (eGram)
+    {
+        case ScGrammar::GRAM_NATIVE:
+            aLBFormulaSyntax.SelectEntryPos(0);
+        break;
+        case ScGrammar::GRAM_NATIVE_XL_A1:
+            aLBFormulaSyntax.SelectEntryPos(1);
+        break;
+        case ScGrammar::GRAM_NATIVE_XL_R1C1:
+            aLBFormulaSyntax.SelectEntryPos(2);
+        break;
+        default:
+            aLBFormulaSyntax.SelectEntryPos(0);
+    }
+
 	pLocalOptions->GetDate( d, m, y );
 
 	switch ( y )
@@ -187,6 +205,20 @@ BOOL __EXPORT ScTpCalcOptions::FillItemS
 	pLocalOptions->SetMatchWholeCell( aBtnMatch.IsChecked() );
     pLocalOptions->SetFormulaRegexEnabled( aBtnRegex.IsChecked() );
 	pLocalOptions->SetLookUpColRowNames( aBtnLookUp.IsChecked() );
+    ScGrammar::Grammar eGram = ScGrammar::GRAM_DEFAULT;
+    switch (aLBFormulaSyntax.GetSelectEntryPos())
+    {
+        case 0:
+            eGram = ScGrammar::GRAM_NATIVE;
+        break;
+        case 1:
+            eGram = ScGrammar::GRAM_NATIVE_XL_A1;
+        break;
+        case 2:
+            eGram = ScGrammar::GRAM_NATIVE_XL_R1C1;
+        break;
+    }
+    pLocalOptions->SetFormulaSyntax(eGram);
 
 	if ( *pLocalOptions != *pOldOptions )
 	{
Only in sc/source/ui/optdlg: tpcalc.cxx.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/src/optdlg.src sc/source/ui/src/optdlg.src
--- sc.clean/source/ui/src/optdlg.src	2008-04-25 18:54:43.000000000 -0400
+++ sc/source/ui/src/optdlg.src	2008-04-25 19:01:33.000000000 -0400
@@ -168,6 +168,27 @@ TabPage RID_SCPAGE_CALC
         Size = MAP_APPFONT ( 239 , 10 ) ;
 		Text [ en-US ] = "~Automatically find column and row labels " ;
 	};
+
+    FixedText FT_FORMULA_SYNTAX
+    {
+        Pos = MAP_APPFONT ( 21, 147 ) ;
+        Size = MAP_APPFONT ( 80, 8 ) ;
+        Text [ en-US ] = "~Formula syntax" ;
+    };
+
+    ListBox LB_FORMULA_SYNTAX
+    {
+        Border = TRUE ;
+        Pos = MAP_APPFONT ( 85, 145 ) ;
+        Size = MAP_APPFONT ( 60, 46 ) ;
+        DropDown = TRUE ;
+        StringList [ en-US ] =
+        {
+            < "Calc A1" ; Default ; > ;
+            < "Excel A1" ; Default ; > ;
+            < "Excel R1C1" ; Default ; > ;
+        };
+    };
 };
 
 /**************************************************************************/
Only in sc/source/ui/src: optdlg.src.orig
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=officecfg.vpj officecfg.clean/registry/schema/org/openoffice/Office/Calc.xcs officecfg/registry/schema/org/openoffice/Office/Calc.xcs
--- officecfg.clean/registry/schema/org/openoffice/Office/Calc.xcs	2008-04-25 19:00:58.000000000 -0400
+++ officecfg/registry/schema/org/openoffice/Office/Calc.xcs	2008-04-25 19:01:33.000000000 -0400
@@ -1129,6 +1129,30 @@
 					</info>
 					<value>true</value>
 				</prop>
+				<prop oor:name="FormulaSyntax" oor:type="xs:int">
+					<!-- UIHints: Tools - Options  Spreadsheet  Calculate -->
+					<info>
+						<author>kyoshida</author>
+						<desc>Specifies what formula syntax type to use.</desc>
+					</info>
+					<constraints>
+						<enumeration oor:value="0">
+							<info>
+								<desc>Calc A1</desc>
+							</info>
+						</enumeration>
+						<enumeration oor:value="1">
+							<info>
+								<desc>Excel A1</desc>
+							</info>
+						</enumeration>
+						<enumeration oor:value="2">
+							<info>
+								<desc>Excel R1C1</desc>
+							</info>
+						</enumeration>
+					</constraints>
+				</prop>
 				<node-ref oor:name="Date" oor:node-type="Date">
 					<info>
 						<desc>Specifies the date that is represented by the number 0.</desc>
Only in officecfg/registry/schema/org/openoffice/Office: Calc.xcs.orig
