Index: sc/source/ui/view/viewfunc.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/viewfunc.cxx,v
retrieving revision 1.38
diff -u -p -r1.38 viewfunc.cxx
--- sc/source/ui/view/viewfunc.cxx	2 Oct 2007 15:22:22 -0000	1.38
+++ sc/source/ui/view/viewfunc.cxx	26 Feb 2008 23:17:48 -0000
@@ -1334,7 +1334,8 @@ void ScViewFunc::ApplySelectionPattern( 
 		SCTAB nTab = pViewData->GetTabNo();
 		ScPatternAttr* pOldPat = new ScPatternAttr(*pDoc->GetPattern( nCol, nRow, nTab ));
 
-		pDoc->ApplyPattern( nCol, nRow, nTab, rAttr );
+		if (!bCursorOnly)
+			pDoc->ApplyPattern( nCol, nRow, nTab, rAttr );
 
 		const ScPatternAttr* pNewPat = pDoc->GetPattern( nCol, nRow, nTab );
 
