diff --git sc/inc/attarray.hxx sc/inc/attarray.hxx
index f123276..5bfa8cd 100644
--- sc/inc/attarray.hxx
+++ sc/inc/attarray.hxx
@@ -35,6 +35,7 @@
 #include "attrib.hxx"
 
 class ScDocument;
+class ScEditDataArray;
 class ScMarkArray;
 class ScPatternAttr;
 class ScStyleSheet;
@@ -104,6 +105,9 @@ friend void lcl_IterGetNumberFormat( ULONG& nFormat,
 							SCROW nStartRow, SCROW nEndRow,
 							BOOL bLeft, SCCOL nDistRight, BOOL bTop, SCROW nDistBottom );
 
+    void RemoveCellCharAttribs( SCROW nStartRow, SCROW nEndRow, 
+                              const ScPatternAttr* pPattern, ScEditDataArray* pDataArray );
+
 public:
 			ScAttrArray( SCCOL nNewCol, SCTAB nNewTab, ScDocument* pDoc );
 			~ScAttrArray();
@@ -125,9 +129,11 @@ public:
 							SCROW nStartRow, SCROW nEndRow, BOOL bLeft, SCCOL nDistRight );
 
 	void	SetPattern( SCROW nRow, const ScPatternAttr* pPattern, BOOL bPutToPool = FALSE );
-	void	SetPatternArea( SCROW nStartRow, SCROW nEndRow, const ScPatternAttr* pPattern, BOOL bPutToPool = FALSE);
+	void	SetPatternArea( SCROW nStartRow, SCROW nEndRow, const ScPatternAttr* pPattern, 
+                            BOOL bPutToPool = FALSE, ScEditDataArray* pDataArray = NULL );
 	void	ApplyStyleArea( SCROW nStartRow, SCROW nEndRow, ScStyleSheet* pStyle );
-	void	ApplyCacheArea( SCROW nStartRow, SCROW nEndRow, SfxItemPoolCache* pCache );
+	void	ApplyCacheArea( SCROW nStartRow, SCROW nEndRow, SfxItemPoolCache* pCache, 
+                            ScEditDataArray* pDataArray = NULL );
 	void	ApplyLineStyleArea( SCROW nStartRow, SCROW nEndRow,
 								const SvxBorderLine* pLine, BOOL bColorOnly );
 
diff --git sc/inc/cell.hxx sc/inc/cell.hxx
index 8484278..ac35dc6 100644
--- sc/inc/cell.hxx
+++ sc/inc/cell.hxx
@@ -34,6 +34,9 @@
 #include <stddef.h>
 
 #include <set>
+#include <vector>
+#include <boost/shared_ptr.hpp>
+
 #include <tools/mempool.hxx>
 #include <svtools/listener.hxx>
 #include "global.hxx"
@@ -58,6 +61,7 @@ class ScCodeArray;
 class ScTokenArray;
 class ScProgress;
 class ScPostIt;
+class ScPatternAttr;
 
 class ScMultipleReadHeader;
 class ScMultipleWriteHeader;
@@ -217,8 +221,57 @@ public:
 	const EditTextObject* GetData() const	{ return pData; }
 
 	void			Save( SvStream& rStream ) const;
+
+    /** Removes character attribute based on new pattern attributes. */
+    void            RemoveCharAttribs( const ScPatternAttr& rAttr );
+};
+
+// ----------------------------------------------------------------------------
+
+class ScEditDataArray
+{
+public:
+    class Item
+    {
+    public:
+        explicit Item(SCTAB nTab, SCCOL nCol, SCROW nRow, 
+                      EditTextObject* pOldData, EditTextObject* pNewData);
+        ~Item();
+    
+        const EditTextObject* GetOldData() const;
+        const EditTextObject* GetNewData() const;
+        SCTAB GetTab() const;
+        SCCOL GetCol() const;
+        SCROW GetRow() const;
+    
+    private:
+        Item(); // disabled
+    
+    private:
+        ::boost::shared_ptr<EditTextObject> mpOldData;
+        ::boost::shared_ptr<EditTextObject> mpNewData;
+        SCTAB mnTab;
+        SCCOL mnCol;
+        SCROW mnRow;
+
+    };
+
+    ScEditDataArray();
+    ~ScEditDataArray();
+
+    void AddItem(SCTAB nTab, SCCOL nCol, SCROW nRow, 
+                 EditTextObject* pOldData, EditTextObject* pNewData);
+
+    const Item* First();
+    const Item* Next();
+
+private:
+    ::std::vector<Item>::const_iterator maIter;
+    ::std::vector<Item> maArray;
 };
 
+// ----------------------------------------------------------------------------
+
 enum ScMatrixMode {
     MM_NONE      = 0,                   // No matrix formula
     MM_FORMULA   = 1,                   // Upper left matrix formula cell
diff --git sc/inc/column.hxx sc/inc/column.hxx
index 388ad6c..3621d78 100644
--- sc/inc/column.hxx
+++ sc/inc/column.hxx
@@ -56,6 +56,7 @@ class ScAttrIterator;
 class ScAttrArray;
 class ScBaseCell;
 class ScDocument;
+class ScEditDataArray;
 class ScFormulaCell;
 class ScMarkData;
 class ScMultipleReadHeader;
@@ -316,7 +317,8 @@ public:
 
 	void		ApplyAttr( SCROW nRow, const SfxPoolItem& rAttr );
 	void		ApplyPattern( SCROW nRow, const ScPatternAttr& rPatAttr );
-	void		ApplyPatternArea( SCROW nStartRow, SCROW nEndRow, const ScPatternAttr& rPatAttr );
+	void		ApplyPatternArea( SCROW nStartRow, SCROW nEndRow, const ScPatternAttr& rPatAttr, 
+                                  ScEditDataArray* pDataArray = NULL );
 	void		SetPattern( SCROW nRow, const ScPatternAttr& rPatAttr, BOOL bPutToPool = FALSE );
 	void		SetPatternArea( SCROW nStartRow, SCROW nEndRow,
 								const ScPatternAttr& rPatAttr, BOOL bPutToPool = FALSE );
@@ -348,7 +350,7 @@ public:
 
 	void		RemoveProtected( SCROW nStartRow, SCROW nEndRow );
 
-	SCsROW		ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark );
+	SCsROW		ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark, ScEditDataArray* pDataArray = NULL );
 	void		DeleteSelection( USHORT nDelFlag, const ScMarkData& rMark );
 
 	void		ClearSelectionItems( const USHORT* pWhich, const ScMarkData& rMark );
diff --git sc/inc/document.hxx sc/inc/document.hxx
index 2bf4393..6d361ae 100644
--- sc/inc/document.hxx
+++ sc/inc/document.hxx
@@ -368,6 +368,7 @@ private:
 	// kein Broadcast, keine Listener aufbauen waehrend aus einem anderen
 	// Doc (per Filter o.ae.) inserted wird, erst bei CompileAll / CalcAfterLoad
 	BOOL				bInsertingFromOtherDoc;
+    bool                bLoadingMedium;
 	BOOL				bImportingXML;		// special handling of formula text
     BOOL                bXMLFromWrapper;    // distinguish ScXMLImportWrapper from external component
 	BOOL				bCalcingAfterLoad;				// in CalcAfterLoad TRUE
@@ -1140,7 +1141,8 @@ SC_DLLPUBLIC	ScDBCollection*	GetDBCollection() const;
 									const ScPatternAttr& rAttr );
 	void			ApplyPatternArea( SCCOL nStartCol, SCROW nStartRow,
 										SCCOL nEndCol, SCROW nEndRow,
-										const ScMarkData& rMark, const ScPatternAttr& rAttr );
+										const ScMarkData& rMark, const ScPatternAttr& rAttr,
+                                      ScEditDataArray* pDataArray = NULL );
 	void			ApplyPatternAreaTab( SCCOL nStartCol, SCROW nStartRow,
 											SCCOL nEndCol, SCROW nEndRow, SCTAB nTab,
 											const ScPatternAttr& rAttr );
@@ -1207,7 +1209,8 @@ SC_DLLPUBLIC	ScDBCollection*	GetDBCollection() const;
 							SCCOL nVCol, SCROW nVRow, SCTAB nVTab,
 							const String& sValStr, double& nX);
 
-	void			ApplySelectionPattern( const ScPatternAttr& rAttr, const ScMarkData& rMark );
+	void			ApplySelectionPattern( const ScPatternAttr& rAttr, const ScMarkData& rMark, 
+                                           ScEditDataArray* pDataArray = NULL );
 	void			DeleteSelection( USHORT nDelFlag, const ScMarkData& rMark );
 	void			DeleteSelectionTab( SCTAB nTab, USHORT nDelFlag, const ScMarkData& rMark );
 
@@ -1477,6 +1480,8 @@ SC_DLLPUBLIC	SvNumberFormatter*	GetFormatTable() const;
 	BOOL			GetNoSetDirty() const { return bNoSetDirty; }
 	void			SetInsertingFromOtherDoc( BOOL bVal ) { bInsertingFromOtherDoc = bVal; }
 	BOOL			IsInsertingFromOtherDoc() const { return bInsertingFromOtherDoc; }
+    void            SetLoadingMedium( bool bVal );
+    bool            IsLoadingMedium() const;
 	void			SetImportingXML( BOOL bVal );
 	BOOL			IsImportingXML() const { return bImportingXML; }
 	void			SetXMLFromWrapper( BOOL bVal );
diff --git sc/inc/table.hxx sc/inc/table.hxx
index c9f6603..570bdd3 100644
--- sc/inc/table.hxx
+++ sc/inc/table.hxx
@@ -57,6 +57,7 @@ class ScAutoFormatData;
 class ScBaseCell;
 class ScDocument;
 class ScDrawLayer;
+class ScEditDataArray;
 class ScFormulaCell;
 class ScOutlineTable;
 class ScPostIt;
@@ -454,7 +455,8 @@ public:
 
 	void		ApplyAttr( SCCOL nCol, SCROW nRow, const SfxPoolItem& rAttr );
 	void		ApplyPattern( SCCOL nCol, SCROW nRow, const ScPatternAttr& rAttr );
-	void		ApplyPatternArea( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow, const ScPatternAttr& rAttr );
+	void		ApplyPatternArea( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow, 
+                                  const ScPatternAttr& rAttr, ScEditDataArray* pDataArray = NULL );
 	void		SetPattern( const ScAddress& rPos, const ScPatternAttr& rAttr, BOOL bPutToPool = FALSE )
 					{ aCol[rPos.Col()].SetPattern( rPos.Row(), rAttr, bPutToPool ); }
 	void		SetPattern( SCCOL nCol, SCROW nRow, const ScPatternAttr& rAttr, BOOL bPutToPool = FALSE );
@@ -482,7 +484,7 @@ public:
 	BOOL		ApplyFlags( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow, INT16 nFlags );
 	BOOL		RemoveFlags( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow, INT16 nFlags );
 
-	void		ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark );
+	void		ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark, ScEditDataArray* pDataArray = NULL );
 	void		DeleteSelection( USHORT nDelFlag, const ScMarkData& rMark );
 
 	void		ClearSelectionItems( const USHORT* pWhich, const ScMarkData& rMark );
diff --git sc/source/core/data/attarray.cxx sc/source/core/data/attarray.cxx
index 19e7684..6102796 100644
--- sc/source/core/data/attarray.cxx
+++ sc/source/core/data/attarray.cxx
@@ -41,6 +41,7 @@
 #include <svx/bolnitem.hxx>
 #include <svx/frmdiritem.hxx>
 #include <svx/shaditem.hxx>
+#include <svx/editobj.hxx>
 #include <svtools/poolcach.hxx>
 #include <svx/fontitem.hxx>
 #include <vcl/fontcvt.hxx>
@@ -55,7 +56,7 @@
 #include "markarr.hxx"
 #include "rechead.hxx"
 #include "globstr.hrc"
-
+#include "cell.hxx"
 
 #undef DBG_INVALIDATE
 #define DBGOUTPUT(s) \
@@ -299,8 +300,31 @@ void ScAttrArray::SetPattern( SCROW nRow, const ScPatternAttr* pPattern, BOOL bP
 	SetPatternArea( nRow, nRow, pPattern, bPutToPool );
 }
 
+void ScAttrArray::RemoveCellCharAttribs( SCROW nStartRow, SCROW nEndRow, 
+                                       const ScPatternAttr* pPattern, ScEditDataArray* pDataArray )
+{
+    for (SCROW nRow = nStartRow; nRow <= nEndRow; ++nRow)
+    {
+        ScBaseCell* pCell;
+        pDocument->GetCell(nCol, nRow, nTab, pCell);
+        if (pCell && pCell->GetCellType() == CELLTYPE_EDIT)
+        {
+            EditTextObject* pOldData = NULL;
+            ScEditCell* pEditCell = static_cast<ScEditCell*>(pCell);
+            if (pDataArray)
+                pOldData = pEditCell->GetData()->Clone();
+            pEditCell->RemoveCharAttribs(*pPattern);
+            if (pDataArray)
+            {
+                EditTextObject* pNewData = pEditCell->GetData()->Clone();
+                pDataArray->AddItem(nTab, nCol, nRow, pOldData, pNewData);
+            }
+        }
+    }
+}
 
-void ScAttrArray::SetPatternArea(SCROW nStartRow, SCROW nEndRow, const ScPatternAttr *pPattern, BOOL bPutToPool )
+void ScAttrArray::SetPatternArea(SCROW nStartRow, SCROW nEndRow, const ScPatternAttr *pPattern, 
+                                 BOOL bPutToPool, ScEditDataArray* pDataArray )
 {
 	if (ValidRow(nStartRow) && ValidRow(nEndRow))
 	{
@@ -470,6 +494,13 @@ void ScAttrArray::SetPatternArea(SCROW nStartRow, SCROW nEndRow, const ScPattern
                     pData[nInsert-1].nRow = nStartRow - 1;
                 pData[nInsert].nRow = nEndRow;
                 pData[nInsert].pPattern = pPattern;
+
+                // Remove character attributes from these cells if the pattern
+                // is applied during normal session.  We don't want to do this
+                // while importing document.
+                if (!pDocument->IsLoadingMedium())
+                    RemoveCellCharAttribs(nStartRow, nEndRow, pPattern ,pDataArray);
+
                 nCount++;
             }
 		}
@@ -705,7 +736,7 @@ void ScAttrArray::ApplyLineStyleArea( SCROW nStartRow, SCROW nEndRow,
 #undef SET_LINE
 
 
-void ScAttrArray::ApplyCacheArea( SCROW nStartRow, SCROW nEndRow, SfxItemPoolCache* pCache )
+void ScAttrArray::ApplyCacheArea( SCROW nStartRow, SCROW nEndRow, SfxItemPoolCache* pCache, ScEditDataArray* pDataArray )
 {
 #ifdef DBG_UTIL
 	TestData();
@@ -740,7 +771,7 @@ void ScAttrArray::ApplyCacheArea( SCROW nStartRow, SCROW nEndRow, SfxItemPoolCac
 				{
 					if (nY1 < nStartRow) nY1=nStartRow;
 					if (nY2 > nEndRow) nY2=nEndRow;
-					SetPatternArea( nY1, nY2, pNewPattern );
+					SetPatternArea( nY1, nY2, pNewPattern, false, pDataArray );
 					Search( nStart, nPos );
 				}
 				else
diff --git sc/source/core/data/cell2.cxx sc/source/core/data/cell2.cxx
index 3357180..a4c9a1c 100644
--- sc/source/core/data/cell2.cxx
+++ sc/source/core/data/cell2.cxx
@@ -49,7 +49,8 @@
 #include "editutil.hxx"
 #include "chgtrack.hxx"
 #include "indexmap.hxx"
-
+#include "scitems.hxx"
+#include "patattr.hxx"
 
 // STATIC DATA -----------------------------------------------------------
 
@@ -189,6 +190,27 @@ void ScEditCell::Save( SvStream& rStream ) const
         pData->Store( rStream );
 }
 
+void ScEditCell::RemoveCharAttribs( const ScPatternAttr& rAttr )
+{
+    const struct {
+        USHORT nAttrType; 
+        USHORT nCharType;
+    } AttrTypeMap[] = {
+        { ATTR_FONT,        EE_CHAR_FONTINFO },
+        { ATTR_FONT_HEIGHT, EE_CHAR_FONTHEIGHT },
+        { ATTR_FONT_COLOR,  EE_CHAR_COLOR }
+    };
+    USHORT nMapCount = sizeof(AttrTypeMap) / sizeof(AttrTypeMap[0]);
+
+    const SfxItemSet& rSet = rAttr.GetItemSet();
+    const SfxPoolItem* pItem;
+    for (USHORT i = 0; i < nMapCount; ++i)
+    {
+        if ( rSet.GetItemState(AttrTypeMap[i].nAttrType, false, &pItem) == SFX_ITEM_SET )
+            pData->RemoveCharAttribs(AttrTypeMap[i].nCharType);
+    }
+}
+
 void ScEditCell::SetTextObject( const EditTextObject* pObject,
             const SfxItemPool* pFromPool )
 {
@@ -224,6 +246,76 @@ void ScEditCell::SetTextObject( const EditTextObject* pObject,
         pData = NULL;
 }
 
+ScEditDataArray::ScEditDataArray()
+{
+}
+
+ScEditDataArray::~ScEditDataArray()
+{
+}
+
+void ScEditDataArray::AddItem(SCTAB nTab, SCCOL nCol, SCROW nRow, 
+                              EditTextObject* pOldData, EditTextObject* pNewData)
+{
+    maArray.push_back(Item(nTab, nCol, nRow, pOldData, pNewData));
+}
+
+const ScEditDataArray::Item* ScEditDataArray::First()
+{
+    maIter = maArray.begin();
+    if (maIter == maArray.end())
+        return NULL;
+    return &(*maIter++);
+}
+
+const ScEditDataArray::Item* ScEditDataArray::Next()
+{
+    if (maIter == maArray.end())
+        return NULL;
+    return &(*maIter++);
+}
+
+//---------------------------------------------------------------------
+
+ScEditDataArray::Item::Item(SCTAB nTab, SCCOL nCol, SCROW nRow, 
+                            EditTextObject* pOldData, EditTextObject* pNewData) :
+    mnTab(nTab),
+    mnCol(nCol),
+    mnRow(nRow)
+{
+    mpOldData.reset(pOldData);
+    mpNewData.reset(pNewData);
+}
+
+ScEditDataArray::Item::~Item()
+{
+}
+
+const EditTextObject* ScEditDataArray::Item::GetOldData() const
+{
+    return mpOldData.get();
+}
+
+const EditTextObject* ScEditDataArray::Item::GetNewData() const
+{
+    return mpNewData.get();
+}
+
+SCTAB ScEditDataArray::Item::GetTab() const
+{
+    return mnTab;
+}
+
+SCCOL ScEditDataArray::Item::GetCol() const
+{
+    return mnCol;
+}
+
+SCROW ScEditDataArray::Item::GetRow() const
+{
+    return mnRow;
+}
+
 //---------------------------------------------------------------------
 
 BOOL ScFormulaCell::IsEmpty()
diff --git sc/source/core/data/column.cxx sc/source/core/data/column.cxx
index fc848b1..67f0b70 100644
--- sc/source/core/data/column.cxx
+++ sc/source/core/data/column.cxx
@@ -369,7 +369,7 @@ ULONG ScColumn::GetNumberFormat( SCROW nRow ) const
 }
 
 
-SCsROW ScColumn::ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark )
+SCsROW ScColumn::ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark, ScEditDataArray* pDataArray )
 {
     SCROW nTop = 0;
     SCROW nBottom = 0;
@@ -380,7 +380,7 @@ SCsROW ScColumn::ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData
 		ScMarkArrayIter aMarkIter( rMark.GetArray() + nCol );
 		while (aMarkIter.Next( nTop, nBottom ))
 		{
-			pAttrArray->ApplyCacheArea( nTop, nBottom, pCache );
+			pAttrArray->ApplyCacheArea( nTop, nBottom, pCache, pDataArray );
 			bFound = TRUE;
 		}
 	}
@@ -454,11 +454,12 @@ void ScColumn::ApplyPattern( SCROW nRow, const ScPatternAttr& rPatAttr )
 }
 
 
-void ScColumn::ApplyPatternArea( SCROW nStartRow, SCROW nEndRow, const ScPatternAttr& rPatAttr )
+void ScColumn::ApplyPatternArea( SCROW nStartRow, SCROW nEndRow, const ScPatternAttr& rPatAttr, 
+                                 ScEditDataArray* pDataArray )
 {
 	const SfxItemSet* pSet = &rPatAttr.GetItemSet();
 	SfxItemPoolCache aCache( pDocument->GetPool(), pSet );
-	pAttrArray->ApplyCacheArea( nStartRow, nEndRow, &aCache );
+	pAttrArray->ApplyCacheArea( nStartRow, nEndRow, &aCache, pDataArray );
 }
 
 
diff --git sc/source/core/data/documen2.cxx sc/source/core/data/documen2.cxx
index cbf19d4..08912c2 100644
--- sc/source/core/data/documen2.cxx
+++ sc/source/core/data/documen2.cxx
@@ -187,6 +187,7 @@ ScDocument::ScDocument( ScDocumentMode	eMode,
 //		bNoSetDirty( TRUE ),
 		bNoSetDirty( FALSE ),
 		bInsertingFromOtherDoc( FALSE ),
+        bLoadingMedium(false),
 		bImportingXML( FALSE ),
         bXMLFromWrapper( FALSE ),
 		bCalcingAfterLoad( FALSE ),
diff --git sc/source/core/data/documen9.cxx sc/source/core/data/documen9.cxx
index 2b3c3f7..c29607d 100644
--- sc/source/core/data/documen9.cxx
+++ sc/source/core/data/documen9.cxx
@@ -885,6 +885,16 @@ void ScDocument::UpdateFontCharSet()
 	}
 }
 
+void ScDocument::SetLoadingMedium( bool bVal )
+{
+    bLoadingMedium = bVal;
+}
+
+bool ScDocument::IsLoadingMedium() const
+{
+    return bLoadingMedium;
+}
+
 void ScDocument::SetImportingXML( BOOL bVal )
 {
 	bImportingXML = bVal;
diff --git sc/source/core/data/document.cxx sc/source/core/data/document.cxx
index 3f6030c..c5dc8b8 100644
--- sc/source/core/data/document.cxx
+++ sc/source/core/data/document.cxx
@@ -3117,12 +3117,13 @@ void ScDocument::ApplyPattern( SCCOL nCol, SCROW nRow, SCTAB nTab, const ScPatte
 void ScDocument::ApplyPatternArea( SCCOL nStartCol, SCROW nStartRow,
 						SCCOL nEndCol, SCROW nEndRow,
 						const ScMarkData& rMark,
-						const ScPatternAttr& rAttr )
+						const ScPatternAttr& rAttr,
+                        ScEditDataArray* pDataArray )
 {
 	for (SCTAB i=0; i <= MAXTAB; i++)
 		if (pTab[i])
 			if (rMark.GetTableSelect(i))
-				pTab[i]->ApplyPatternArea( nStartCol, nStartRow, nEndCol, nEndRow, rAttr );
+				pTab[i]->ApplyPatternArea( nStartCol, nStartRow, nEndCol, nEndRow, rAttr, pDataArray );
 }
 
 
@@ -4211,7 +4212,7 @@ void ScDocument::ApplyFrameAreaTab( const ScRange& rRange,
 }
 
 
-void ScDocument::ApplySelectionPattern( const ScPatternAttr& rAttr, const ScMarkData& rMark )
+void ScDocument::ApplySelectionPattern( const ScPatternAttr& rAttr, const ScMarkData& rMark, ScEditDataArray* pDataArray )
 {
 	const SfxItemSet* pSet = &rAttr.GetItemSet();
 	BOOL bSet = FALSE;
@@ -4228,7 +4229,7 @@ void ScDocument::ApplySelectionPattern( const ScPatternAttr& rAttr, const ScMark
 			ScRange aRange;
 			rMark.GetMarkArea( aRange );
 			ApplyPatternArea( aRange.aStart.Col(), aRange.aStart.Row(),
-							  aRange.aEnd.Col(), aRange.aEnd.Row(), rMark, rAttr );
+							  aRange.aEnd.Col(), aRange.aEnd.Row(), rMark, rAttr, pDataArray );
 		}
 		else
 		{
@@ -4236,7 +4237,7 @@ void ScDocument::ApplySelectionPattern( const ScPatternAttr& rAttr, const ScMark
             for (SCTAB nTab=0; nTab<=MAXTAB; nTab++)
                 if (pTab[nTab])
                     if (rMark.GetTableSelect(nTab))
-                        pTab[nTab]->ApplySelectionCache( &aCache, rMark );
+                        pTab[nTab]->ApplySelectionCache( &aCache, rMark, pDataArray );
 		}
 	}
 }
diff --git sc/source/core/data/table2.cxx sc/source/core/data/table2.cxx
index 0ae2d0b..6a55cb5 100644
--- sc/source/core/data/table2.cxx
+++ sc/source/core/data/table2.cxx
@@ -1648,14 +1648,14 @@ void ScTable::ApplyPattern( SCCOL nCol, SCROW nRow, const ScPatternAttr& rAttr )
 
 
 void ScTable::ApplyPatternArea( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow,
-									 const ScPatternAttr& rAttr )
+									 const ScPatternAttr& rAttr, ScEditDataArray* pDataArray )
 {
 	if (ValidColRow(nStartCol, nStartRow) && ValidColRow(nEndCol, nEndRow))
 	{
 		PutInOrder(nStartCol, nEndCol);
 		PutInOrder(nStartRow, nEndRow);
 		for (SCCOL i = nStartCol; i <= nEndCol; i++)
-			aCol[i].ApplyPatternArea(nStartRow, nEndRow, rAttr);
+			aCol[i].ApplyPatternArea(nStartRow, nEndRow, rAttr, pDataArray);
 	}
 }
 
@@ -1863,10 +1863,11 @@ void ScTable::ApplyAttr( SCCOL nCol, SCROW nRow, const SfxPoolItem& rAttr )
 }
 
 
-void ScTable::ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark )
+void ScTable::ApplySelectionCache( SfxItemPoolCache* pCache, const ScMarkData& rMark,
+                                   ScEditDataArray* pDataArray )
 {
 	for (SCCOL i=0; i<=MAXCOL; i++)
-		aCol[i].ApplySelectionCache( pCache, rMark );
+		aCol[i].ApplySelectionCache( pCache, rMark, pDataArray );
 }
 
 
diff --git sc/source/ui/docshell/docsh.cxx sc/source/ui/docshell/docsh.cxx
index 9bcc078..c702954 100644
--- sc/source/ui/docshell/docsh.cxx
+++ sc/source/ui/docshell/docsh.cxx
@@ -680,10 +680,31 @@ BOOL ScDocShell::SaveXML( SfxMedium* pSaveMedium, const ::com::sun::star::uno::R
     return bRet;
 }
 
+namespace {
+
+class DocLoadChecker
+{
+public:
+    explicit DocLoadChecker(ScDocument* pDoc) :
+        mpDoc(pDoc)
+    {
+        mpDoc->SetLoadingMedium(true);
+    }
+
+    ~DocLoadChecker()
+    {
+        mpDoc->SetLoadingMedium(false);
+    }
+private:
+    ScDocument* mpDoc;
+};
+
+}
+
 BOOL __EXPORT ScDocShell::Load( SfxMedium& rMedium )
 {
 	RTL_LOGFILE_CONTEXT_AUTHOR ( aLog, "sc", "nn93723", "ScDocShell::Load" );
-
+    DocLoadChecker aChecker(&aDocument);
 	ScRefreshTimerProtector( aDocument.GetRefreshTimerControlAddress() );
 
 	//	only the latin script language is loaded
@@ -1048,7 +1069,7 @@ void __EXPORT ScDocShell::Notify( SfxBroadcaster&, const SfxHint& rHint )
 BOOL __EXPORT ScDocShell::LoadFrom( SfxMedium& rMedium )
 {
 	RTL_LOGFILE_CONTEXT_AUTHOR ( aLog, "sc", "nn93723", "ScDocShell::LoadFrom" );
-
+    DocLoadChecker aChecker(&aDocument);
 	ScRefreshTimerProtector( aDocument.GetRefreshTimerControlAddress() );
 
 	WaitObject aWait( GetActiveDialogParent() );
@@ -1075,6 +1096,7 @@ BOOL __EXPORT ScDocShell::LoadFrom( SfxMedium& rMedium )
 BOOL __EXPORT ScDocShell::ConvertFrom( SfxMedium& rMedium )
 {
 	RTL_LOGFILE_CONTEXT_AUTHOR ( aLog, "sc", "nn93723", "ScDocShell::ConvertFrom" );
+    DocLoadChecker aChecker(&aDocument);
 
 	BOOL bRet = FALSE;				// FALSE heisst Benutzerabbruch !!
 									// bei Fehler: Fehler am Stream setzen!!
diff --git sc/source/ui/inc/undoblk.hxx sc/source/ui/inc/undoblk.hxx
index cfa37e2..6f7c526 100644
--- sc/source/ui/inc/undoblk.hxx
+++ sc/source/ui/inc/undoblk.hxx
@@ -35,6 +35,10 @@
 #include "viewutil.hxx"
 #include "spellparam.hxx"
 
+#include "cell.hxx"
+
+#include <boost/shared_ptr.hpp>
+
 class ScDocShell;
 class ScDocument;
 class ScOutlineTable;
@@ -347,9 +351,11 @@ public:
 
 	virtual String	GetComment() const;
 
+    ScEditDataArray*    GetDataArray();
 private:
 	ScMarkData		aMarkData;
 	ScRange			aRange;
+    ScEditDataArray aDataArray; 
 	ScDocument*		pUndoDoc;
 	BOOL			bMulti;
 	ScPatternAttr*	pApplyPattern;
@@ -357,6 +363,7 @@ private:
 	SvxBoxInfoItem*	pLineInner;
 
 	void			DoChange( const BOOL bUndo );
+    void            ChangeEditData( const bool bUndo );
 };
 
 
diff --git sc/source/ui/inc/undocell.hxx sc/source/ui/inc/undocell.hxx
index b811744..c5aba91 100644
--- sc/source/ui/inc/undocell.hxx
+++ sc/source/ui/inc/undocell.hxx
@@ -34,6 +34,8 @@
 #include "undobase.hxx"
 #include "postit.hxx"
 
+#include <boost/shared_ptr.hpp>
+
 class ScDocShell;
 class ScBaseCell;
 class ScPatternAttr;
@@ -62,6 +64,11 @@ public:
 
 	virtual String	GetComment() const;
 
+    /** once the objects are passed to this class, their life-cycle is
+        managed by this class; the calling function must pass new'ed
+        objects to this method. */
+    void            SetEditData( EditTextObject* pOld, EditTextObject* pNew );
+
 private:
 	SCCOL			nCol;
 	SCROW			nRow;
@@ -69,9 +76,11 @@ private:
 	ScPatternAttr*	pOldPattern;
 	ScPatternAttr*	pNewPattern;
 	ScPatternAttr*	pApplyPattern;
+    ::boost::shared_ptr<EditTextObject> pOldEditData;
+    ::boost::shared_ptr<EditTextObject> pNewEditData;
 	BOOL			bIsAutomatic;
 
-	void			DoChange( const ScPatternAttr* pWhichPattern ) const;
+	void			DoChange( const ScPatternAttr* pWhichPattern, const ::boost::shared_ptr<EditTextObject>& pEditData ) const;
 };
 
 
diff --git sc/source/ui/undo/makefile.mk sc/source/ui/undo/makefile.mk
index 7c4d29a..bd35af7 100644
--- sc/source/ui/undo/makefile.mk
+++ sc/source/ui/undo/makefile.mk
@@ -78,6 +78,7 @@ SLOFILES =  \
 		$(SLO)$/undotab.obj
 
 EXCEPTIONSFILES= \
+    $(SLO)$/undocell.obj \
     $(SLO)$/undostyl.obj
 
 # --- Tagets -------------------------------------------------------
diff --git sc/source/ui/undo/undoblk3.cxx sc/source/ui/undo/undoblk3.cxx
index 9cabbd7..5cf9b92 100644
--- sc/source/ui/undo/undoblk3.cxx
+++ sc/source/ui/undo/undoblk3.cxx
@@ -474,6 +474,12 @@ String __EXPORT ScUndoSelectionAttr::GetComment() const
 	return ScGlobal::GetRscString( pLineOuter ? STR_UNDO_SELATTRLINES : STR_UNDO_SELATTR );
 }
 
+//----------------------------------------------------------------------------
+
+ScEditDataArray* ScUndoSelectionAttr::GetDataArray()
+{
+    return &aDataArray;
+}
 
 //----------------------------------------------------------------------------
 
@@ -492,6 +498,8 @@ void ScUndoSelectionAttr::DoChange( const BOOL bUndo )
 	USHORT nExtFlags = 0;
 	pDocShell->UpdatePaintExt( nExtFlags, aEffRange );
 
+    ChangeEditData(bUndo);
+
 	if (bUndo)	// nur bei Undo
 	{
 		ScRange aCopyRange = aRange;
@@ -516,6 +524,24 @@ void ScUndoSelectionAttr::DoChange( const BOOL bUndo )
 	ShowTable( aRange );
 }
 
+void ScUndoSelectionAttr::ChangeEditData( const bool bUndo )
+{
+    ScDocument* pDoc = pDocShell->GetDocument();
+    for (const ScEditDataArray::Item* pItem = aDataArray.First(); pItem; pItem = aDataArray.Next())
+    {
+        ScBaseCell* pCell;
+        pDoc->GetCell(pItem->GetCol(), pItem->GetRow(), pItem->GetTab(), pCell);
+        if (!pCell || pCell->GetCellType() != CELLTYPE_EDIT)
+            continue;
+
+        ScEditCell* pEditCell = static_cast<ScEditCell*>(pCell);
+        if (bUndo)
+            pEditCell->SetData(pItem->GetOldData(), NULL);
+        else
+            pEditCell->SetData(pItem->GetNewData(), NULL);
+    }
+}
+
 
 //----------------------------------------------------------------------------
 
diff --git sc/source/ui/undo/undocell.cxx sc/source/ui/undo/undocell.cxx
index 5a8d9fa..9ff33fe 100644
--- sc/source/ui/undo/undocell.cxx
+++ sc/source/ui/undo/undocell.cxx
@@ -61,6 +61,8 @@
 #include "chgtrack.hxx"
 #include "sc.hrc"
 
+using ::boost::shared_ptr;
+
 // STATIC DATA -----------------------------------------------------------
 
 TYPEINIT1(ScUndoCursorAttr, ScSimpleUndo);
@@ -89,6 +91,8 @@ ScUndoCursorAttr::ScUndoCursorAttr( ScDocShell* pNewDocShell,
 	nCol( nNewCol ),
 	nRow( nNewRow ),
 	nTab( nNewTab ),
+    pOldEditData( static_cast<EditTextObject*>(NULL) ),
+    pNewEditData( static_cast<EditTextObject*>(NULL) ),
 	bIsAutomatic( bAutomatic )
 {
 	ScDocumentPool* pPool = pDocShell->GetDocument()->GetPool();
@@ -113,9 +117,21 @@ String __EXPORT ScUndoCursorAttr::GetComment() const
 	return ScGlobal::GetRscString( nId );
 }
 
-void ScUndoCursorAttr::DoChange( const ScPatternAttr* pWhichPattern ) const
+void ScUndoCursorAttr::SetEditData( EditTextObject* pOld, EditTextObject* pNew )
 {
-	pDocShell->GetDocument()->SetPattern( nCol, nRow, nTab, *pWhichPattern, TRUE );
+    pOldEditData.reset(pOld);
+    pNewEditData.reset(pNew);
+}
+
+void ScUndoCursorAttr::DoChange( const ScPatternAttr* pWhichPattern, const shared_ptr<EditTextObject>& pEditData ) const
+{
+    ScDocument* pDoc = pDocShell->GetDocument();
+	pDoc->SetPattern( nCol, nRow, nTab, *pWhichPattern, TRUE );
+
+    ScBaseCell* pCell;
+    pDoc->GetCell(nCol, nRow, nTab, pCell);
+    if (pCell && pCell->GetCellType() == CELLTYPE_EDIT && pEditData.get())
+        static_cast<ScEditCell*>(pCell)->SetData(pEditData.get(), NULL);
 
 	ScTabViewShell* pViewShell = ScTabViewShell::GetActiveViewShell();
 	if (pViewShell)
@@ -141,7 +157,7 @@ void ScUndoCursorAttr::DoChange( const ScPatternAttr* pWhichPattern ) const
 void __EXPORT ScUndoCursorAttr::Undo()
 {
 	BeginUndo();
-	DoChange(pOldPattern);
+	DoChange(pOldPattern, pOldEditData);
 
 	if ( bIsAutomatic )
 	{
@@ -159,7 +175,7 @@ void __EXPORT ScUndoCursorAttr::Undo()
 void __EXPORT ScUndoCursorAttr::Redo()
 {
 	BeginRedo();
-	DoChange(pNewPattern);
+	DoChange(pNewPattern, pNewEditData);
 	EndRedo();
 }
 
diff --git sc/source/ui/view/viewfunc.cxx sc/source/ui/view/viewfunc.cxx
index 666b939..d262883 100644
--- sc/source/ui/view/viewfunc.cxx
+++ sc/source/ui/view/viewfunc.cxx
@@ -1293,6 +1293,8 @@ void ScViewFunc::ApplySelectionPattern( const ScPatternAttr& rAttr,
 		SCROW nEndRow = aMarkRange.aEnd.Row();
 		SCTAB nEndTab = aMarkRange.aEnd.Tab();
 
+        ScUndoSelectionAttr* pUndoAttr = NULL;
+        ScEditDataArray* pEditDataArray = NULL;
 		if (bRecord)
 		{
 			ScRange aCopyRange = aMarkRange;
@@ -1309,15 +1311,14 @@ void ScViewFunc::ApplySelectionPattern( const ScPatternAttr& rAttr,
 
 			aFuncMark.MarkToMulti();
 
-			pDocSh->GetUndoManager()->AddUndoAction(
-				new ScUndoSelectionAttr(
-							pDocSh, aFuncMark,
-							nStartCol, nStartRow, nStartTab,
-							nEndCol, nEndRow, nEndTab,
-							pUndoDoc, bMulti, &rAttr ) );
+            pUndoAttr = new ScUndoSelectionAttr(
+                pDocSh, aFuncMark, nStartCol, nStartRow, nStartTab,
+                nEndCol, nEndRow, nEndTab, pUndoDoc, bMulti, &rAttr );
+			pDocSh->GetUndoManager()->AddUndoAction(pUndoAttr);
+            pEditDataArray = pUndoAttr->GetDataArray();
 		}
 
-		pDoc->ApplySelectionPattern( rAttr, aFuncMark );
+		pDoc->ApplySelectionPattern( rAttr, aFuncMark, pEditDataArray );
 
 		pDocSh->PostPaint( nStartCol, nStartRow, nStartTab,
 						   nEndCol,   nEndRow,   nEndTab,
@@ -1331,6 +1332,19 @@ void ScViewFunc::ApplySelectionPattern( const ScPatternAttr& rAttr,
 		SCCOL nCol = pViewData->GetCurX();
 		SCROW nRow = pViewData->GetCurY();
 		SCTAB nTab = pViewData->GetTabNo();
+
+        ScBaseCell* pCell;
+        pDoc->GetCell(nCol, nRow, nTab, pCell);
+        EditTextObject* pOldEditData = NULL;
+        EditTextObject* pNewEditData = NULL;
+        if (pCell && pCell->GetCellType() == CELLTYPE_EDIT)
+        {
+            ScEditCell* pEditCell = static_cast<ScEditCell*>(pCell);
+            pOldEditData = pEditCell->GetData()->Clone();
+            pEditCell->RemoveCharAttribs(rAttr);
+            pNewEditData = pEditCell->GetData()->Clone();
+        }
+
 		ScPatternAttr* pOldPat = new ScPatternAttr(*pDoc->GetPattern( nCol, nRow, nTab ));
 
 		if (!bCursorOnly)
@@ -1340,11 +1354,10 @@ void ScViewFunc::ApplySelectionPattern( const ScPatternAttr& rAttr,
 
 		if (bRecord)
 		{
-			pDocSh->GetUndoManager()->AddUndoAction(
-						new ScUndoCursorAttr( pDocSh,
-											  nCol, nRow, nTab,
-											  pOldPat, pNewPat, &rAttr,
-											  FALSE ) );	// FALSE = nicht automatisch
+            ScUndoCursorAttr* pUndo = new ScUndoCursorAttr( 
+                pDocSh, nCol, nRow, nTab, pOldPat, pNewPat, &rAttr, false );
+            pUndo->SetEditData(pOldEditData, pNewEditData);
+			pDocSh->GetUndoManager()->AddUndoAction(pUndo);
 		}
 		delete pOldPat;		// wird im Undo kopiert (Pool)
 
