Index: sc/source/ui/inc/preview.hxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/inc/preview.hxx,v
retrieving revision 1.11
diff -u -p -r1.11 preview.hxx
--- sc/source/ui/inc/preview.hxx	10 Apr 2008 23:16:01 -0000	1.11
+++ sc/source/ui/inc/preview.hxx	19 May 2008 14:06:41 -0000
@@ -66,6 +66,8 @@ private:
 	ScPreviewLocationData* pLocationData;	// stores table layout for accessibility API
 	FmFormView*		pDrawView;
 
+    SCTAB           nCurTab;
+
 										// intern:
 	BOOL			bInPaint;
 	BOOL			bInGetState;
Index: sc/source/ui/view/preview.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/preview.cxx,v
retrieving revision 1.34
diff -u -p -r1.34 preview.cxx
--- sc/source/ui/view/preview.cxx	11 Apr 2008 01:35:53 -0000	1.34
+++ sc/source/ui/view/preview.cxx	19 May 2008 14:06:41 -0000
@@ -109,6 +109,7 @@ ScPreview::ScPreview( Window* pParent, S
 	bLocationValid( FALSE ),
 	pLocationData( NULL ),
 	pDrawView( NULL ),
+    nCurTab ( ScDocShell::GetCurTab() ),
 	bInPaint( FALSE ),
 	bInGetState( FALSE ),
 	pDocShell( pDocSh ),
@@ -244,8 +245,17 @@ void ScPreview::CalcPages( SCTAB nToWhic
 	//	but always all sheets are used (there is no selected sheet)
 	ScPrintOptions aOptions = SC_MOD()->GetPrintOptions();
 
+	ScMarkData aMarkData;
+	aMarkData.SelectTable( nCurTab, TRUE );
+
 	for (SCTAB i=nStart; i<nAnz; i++)
 	{
+		if (!aOptions.GetAllSheets() && !aMarkData.GetTableSelect( i )) {
+			nPages[i] = 0;
+			nFirstAttr[i] = 0;
+			continue;
+		}
+
 		long nAttrPage = i > 0 ? nFirstAttr[i-1] : 1;
 
 		long nThisStart = nTotalPages;
Index: sc/source/ui/view/tabvwsh4.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/tabvwsh4.cxx,v
retrieving revision 1.70
diff -u -p -r1.70 tabvwsh4.cxx
--- sc/source/ui/view/tabvwsh4.cxx	26 Nov 2007 18:43:29 -0000	1.70
+++ sc/source/ui/view/tabvwsh4.cxx	15 Dec 2007 03:19:14 -0000
@@ -1207,8 +1215,13 @@ PrintDialog* __EXPORT ScTabViewShell::Cr
     bool bAllTabs = SC_MOD()->GetPrintOptions().GetAllSheets();
     pDlg->CheckSheetRange( bAllTabs ? PRINTSHEETS_ALL : PRINTSHEETS_SELECTED_SHEETS );
 
+	ScMarkData aMarkData;
+	aMarkData.SelectTable( GetViewData()->GetTabNo(), TRUE );
+
 	for ( SCTAB i=0; i<nTabCount; i++ )
 	{
+		if ( !bAllTabs && !aMarkData.GetTableSelect( i ) )
+			continue;
 		ScPrintFunc aPrintFunc( pDocShell, pPrinter, i );
 		nDocPageMax += aPrintFunc.GetTotalPages();
 	}
