--- sc/source/core/data/column2.cxx.old	2009-04-06 16:41:50.000000000 +0000
+++ sc/source/core/data/column2.cxx	2009-04-06 16:41:50.000000000 +0000
@@ -1319,6 +1319,10 @@ BOOL ScColumn::IsEmpty() const
 
 BOOL ScColumn::IsEmptyBlock(SCROW nStartRow, SCROW nEndRow, bool bIgnoreNotes) const
 {
+	Rectangle aRect;
+	if (pAttrArray->HasLines(nStartRow, nEndRow, aRect, TRUE, TRUE))
+		return FALSE;
+
 	if ( nCount == 0 || !pItems )
 		return TRUE;
 
