diff --git sc/source/ui/docshell/docfunc.cxx sc/source/ui/docshell/docfunc.cxx
index 5a8f68c..34720ec 100644
--- sc/source/ui/docshell/docfunc.cxx
+++ sc/source/ui/docshell/docfunc.cxx
@@ -3436,7 +3436,7 @@ BOOL ScDocFunc::FillAuto( ScRange& rRange, const ScMarkData* pTabMark, FillDir e
 
 //------------------------------------------------------------------------
 
-BOOL ScDocFunc::MergeCells( const ScRange& rRange, BOOL bContents, BOOL bRecord, BOOL bApi )
+BOOL ScDocFunc::MergeCells( const ScRange& rRange, BOOL bContents, BOOL bRecord, BOOL bApi, BOOL bCenter )
 {
 	ScDocShellModificator aModificator( rDocShell );
 
@@ -3480,7 +3480,7 @@ BOOL ScDocFunc::MergeCells( const ScRange& rRange, BOOL bContents, BOOL bRecord,
 	if (bRecord)
 	{
 		ScDocument* pUndoDoc = NULL;
-		if (bNeedContents && bContents)
+		if ( (bNeedContents && bContents) || bCenter )
 		{
 			pUndoDoc = new ScDocument( SCDOCMODE_UNDO );
 			pUndoDoc->InitUndo( pDoc, nTab, nTab );
@@ -3490,17 +3490,23 @@ BOOL ScDocFunc::MergeCells( const ScRange& rRange, BOOL bContents, BOOL bRecord,
 		rDocShell.GetUndoManager()->AddUndoAction(
 			new ScUndoMerge( &rDocShell,
 							nStartCol, nStartRow, nTab,
-							nEndCol, nEndRow, nTab, TRUE, pUndoDoc ) );
+							nEndCol, nEndRow, nTab, TRUE, pUndoDoc, bCenter) );
 	}
 
 	if (bNeedContents && bContents)
 		pDoc->DoMergeContents( nTab, nStartCol,nStartRow, nEndCol,nEndRow );
 	pDoc->DoMerge( nTab, nStartCol,nStartRow, nEndCol,nEndRow );
+	
+	if (bCenter)
+	{
+		pDoc->ApplyAttr( nStartCol, nStartRow, nTab, SvxHorJustifyItem( SVX_HOR_JUSTIFY_CENTER, ATTR_HOR_JUSTIFY ) );
+		pDoc->ApplyAttr( nStartCol, nStartRow, nTab, SvxVerJustifyItem( SVX_VER_JUSTIFY_CENTER, ATTR_VER_JUSTIFY ) );
+	}
 
 	if ( !AdjustRowHeight( ScRange( 0,nStartRow,nTab, MAXCOL,nEndRow,nTab ) ) )
 		rDocShell.PostPaint( nStartCol, nStartRow, nTab,
 											nEndCol, nEndRow, nTab, PAINT_GRID );
-	if (bNeedContents && bContents)
+	if ( (bNeedContents && bContents) || bCenter )
 		pDoc->SetDirty( rRange );
 	aModificator.SetDocumentModified();
 
diff --git sc/source/ui/inc/docfunc.hxx sc/source/ui/inc/docfunc.hxx
index b2b7a57..be2a7e5 100644
--- sc/source/ui/inc/docfunc.hxx
+++ sc/source/ui/inc/docfunc.hxx
@@ -169,7 +169,7 @@ public:
 	BOOL			ResizeMatrix( const ScRange& rOldRange, const ScAddress& rNewEnd, BOOL bApi );
 
 	BOOL			MergeCells( const ScRange& rRange, BOOL bContents,
-								BOOL bRecord, BOOL bApi );
+								BOOL bRecord, BOOL bApi, BOOL bCenter );
 	BOOL			UnmergeCells( const ScRange& rRange, BOOL bRecord, BOOL bApi );
 
 	BOOL			SetNote( const ScAddress& rPos, const ScPostIt& rNote, BOOL bApi );
diff --git sc/source/ui/inc/undoblk.hxx sc/source/ui/inc/undoblk.hxx
index 42b4097..06c1feb 100644
--- sc/source/ui/inc/undoblk.hxx
+++ sc/source/ui/inc/undoblk.hxx
@@ -464,7 +464,8 @@ public:
 					ScUndoMerge( ScDocShell* pNewDocShell,
 								 SCCOL nStartX, SCROW nStartY, SCTAB nStartZ,
 								 SCCOL nEndX,   SCROW nEndY,   SCTAB nEndZ,
-								 BOOL bNewDoMerge, ScDocument* pNewUndoDoc = NULL );
+								 BOOL bNewDoMerge, ScDocument* pNewUndoDoc = NULL,
+								 BOOL bNewCenter = FALSE );
 	virtual 		~ScUndoMerge();
 
 	virtual void	Undo();
@@ -478,6 +479,7 @@ private:
 	ScRange			aRange;
 	BOOL			bDoMerge;				// Merge oder aufheben
 	ScDocument*		pUndoDoc;				// wenn Daten zusammengefasst
+	BOOL			bCenter;
 
 	void			DoChange( const BOOL bUndo ) const;
 };
diff --git sc/source/ui/inc/viewfunc.hxx sc/source/ui/inc/viewfunc.hxx
index 43288de..71ed0d0 100644
--- sc/source/ui/inc/viewfunc.hxx
+++ sc/source/ui/inc/viewfunc.hxx
@@ -249,7 +249,7 @@ public:
 	BOOL			TestMergeCells();
 	BOOL			TestRemoveMerge();
 
-	BOOL			MergeCells( BOOL bApi, BOOL& rDoContents, BOOL bRecord = TRUE );
+	BOOL			MergeCells( BOOL bApi, BOOL& rDoContents, BOOL bRecord = TRUE, BOOL bCenter = FALSE );
 	BOOL			RemoveMerge( BOOL bRecord = TRUE );
 
 	void			FillSimple( FillDir eDir, BOOL bRecord = TRUE );
diff --git sc/source/ui/undo/undoblk3.cxx sc/source/ui/undo/undoblk3.cxx
index d0a5c5e..8e35362 100644
--- sc/source/ui/undo/undoblk3.cxx
+++ sc/source/ui/undo/undoblk3.cxx
@@ -39,6 +39,7 @@
 // INCLUDE -------------------------------------------------------------------
 
 #include "scitems.hxx"
+#include <svx/algitem.hxx>
 #include <svx/boxitem.hxx>
 #include <svx/srchitem.hxx>
 #include <svx/linkmgr.hxx>
@@ -812,13 +813,15 @@ BOOL __EXPORT ScUndoAutoFill::CanRepeat(SfxRepeatTarget& rTarget) const
 ScUndoMerge::ScUndoMerge( ScDocShell* pNewDocShell,
 							SCCOL nStartX, SCROW nStartY, SCTAB nStartZ,
 							SCCOL nEndX, SCROW nEndY, SCTAB nEndZ,
-							BOOL bNewDoMerge, ScDocument* pNewUndoDoc )
+							BOOL bNewDoMerge, ScDocument* pNewUndoDoc,
+							BOOL bNewCenter )
 		//
 	:	ScSimpleUndo( pNewDocShell ),
 		//
 		aRange	( nStartX, nStartY, nStartZ, nEndX, nEndY, nEndZ ),
 		bDoMerge( bNewDoMerge ),
-		pUndoDoc( pNewUndoDoc )
+		pUndoDoc( pNewUndoDoc ),
+		bCenter ( bNewCenter )
 {
 }
 
@@ -865,9 +868,20 @@ void ScUndoMerge::DoChange( const BOOL bUndo ) const
 			pUndoDoc->CopyToDocument( aRange, IDF_ALL, FALSE, pDoc );
 		}
 		else
+		{
 /*!*/		pDoc->DoMergeContents( aRange.aStart.Tab(),
 								   aRange.aStart.Col(), aRange.aStart.Row(),
 								   aRange.aEnd.Col(),   aRange.aEnd.Row()   );
+			if (bCenter)
+			{
+				pDoc->ApplyAttr( aRange.aStart.Col(), aRange.aStart.Row(),
+								 aRange.aStart.Tab(),
+								 SvxHorJustifyItem( SVX_HOR_JUSTIFY_CENTER, ATTR_HOR_JUSTIFY ) );
+				pDoc->ApplyAttr( aRange.aStart.Col(), aRange.aStart.Row(),
+								 aRange.aStart.Tab(),
+								 SvxVerJustifyItem( SVX_VER_JUSTIFY_CENTER, ATTR_VER_JUSTIFY ) );
+			}
+		}
 
 	BOOL bDidPaint = FALSE;
 	ScTabViewShell* pViewShell = ScTabViewShell::GetActiveViewShell();
diff --git sc/source/ui/unoobj/cellsuno.cxx sc/source/ui/unoobj/cellsuno.cxx
index 930cf19..e825ea9 100644
--- sc/source/ui/unoobj/cellsuno.cxx
+++ sc/source/ui/unoobj/cellsuno.cxx
@@ -5360,7 +5360,7 @@ void SAL_CALL ScCellRangeObj::merge( sal_Bool bMerge ) throw(uno::RuntimeExcepti
 	{
 		ScDocFunc aFunc(*pDocSh);
 		if ( bMerge )
-			aFunc.MergeCells( aRange, FALSE, TRUE, TRUE );
+			aFunc.MergeCells( aRange, FALSE, TRUE, TRUE, FALSE );
 		else
 			aFunc.UnmergeCells( aRange, TRUE, TRUE );
 
diff --git sc/source/ui/view/cellsh3.cxx sc/source/ui/view/cellsh3.cxx
index d0431c6..7cc1eb4 100644
--- sc/source/ui/view/cellsh3.cxx
+++ sc/source/ui/view/cellsh3.cxx
@@ -758,6 +758,7 @@ void ScCellShell::Execute( SfxRequest& rReq )
             {
                 // test whether to merge or to split
                 bool bMerge = false;
+                BOOL bCenter = FALSE;
                 switch( nSlot )
                 {
                     case FID_MERGE_ON:
@@ -768,6 +769,7 @@ void ScCellShell::Execute( SfxRequest& rReq )
                     break;
                     case FID_MERGE_TOGGLE:
                     {
+                        bCenter = TRUE;
                         SfxPoolItem* pItem = 0;
                         if( rBindings.QueryState( nSlot, pItem ) >= SFX_ITEM_DEFAULT )
                             bMerge = !static_cast< SfxBoolItem* >( pItem )->GetValue();
@@ -788,7 +790,7 @@ void ScCellShell::Execute( SfxRequest& rReq )
                         bMoveContents = ((const SfxBoolItem*)pItem)->GetValue();
                     }
 
-                    if (pTabViewShell->MergeCells( bApi, bMoveContents ))
+                    if (pTabViewShell->MergeCells( bApi, bMoveContents, TRUE, bCenter ))
                     {
                         if (!bApi && bMoveContents)             // "ja" im Dialog geklickt
                             rReq.AppendItem( SfxBoolItem( nSlot, bMoveContents ) );
diff --git sc/source/ui/view/viewfun2.cxx sc/source/ui/view/viewfun2.cxx
index 3e123c4..b1c289b 100644
--- sc/source/ui/view/viewfun2.cxx
+++ sc/source/ui/view/viewfun2.cxx
@@ -1086,7 +1086,7 @@ BOOL ScViewFunc::TestMergeCells()			// Vorab-Test (fuer Menue)
 
 //----------------------------------------------------------------------------
 
-BOOL ScViewFunc::MergeCells( BOOL bApi, BOOL& rDoContents, BOOL bRecord )
+BOOL ScViewFunc::MergeCells( BOOL bApi, BOOL& rDoContents, BOOL bRecord, BOOL bCenter )
 {
 	//	Editable- und Verschachtelungs-Abfrage muss vorneweg sein (auch in DocFunc),
 	//	damit dann nicht die Inhalte-QueryBox kommt
@@ -1154,7 +1154,7 @@ BOOL ScViewFunc::MergeCells( BOOL bApi, BOOL& rDoContents, BOOL bRecord )
 	if (bOk)
 	{
 		HideCursor();
-		bOk = pDocSh->GetDocFunc().MergeCells( aMarkRange, rDoContents, bRecord, bApi );
+		bOk = pDocSh->GetDocFunc().MergeCells( aMarkRange, rDoContents, bRecord, bApi, bCenter );
 		ShowCursor();
 
 		if (bOk)
diff --git sc/uiconfig/scalc/menubar/menubar.xml sc/uiconfig/scalc/menubar/menubar.xml
index 5bced00..db8ace6 100644
--- sc/uiconfig/scalc/menubar/menubar.xml
+++ sc/uiconfig/scalc/menubar/menubar.xml
@@ -217,7 +217,13 @@
                     <menu:menuitem menu:id=".uno:SheetRightToLeft"/>
                 </menu:menupopup>
             </menu:menu>
-            <menu:menuitem menu:id=".uno:ToggleMergeCells"/>
+            <menu:menu menu:id=".uno:MergeCellsMenu">
+                <menu:menupopup>
+                    <menu:menuitem menu:id=".uno:ToggleMergeCells"/>
+                    <menu:menuitem menu:id=".uno:MergeCells"/>
+                    <menu:menuitem menu:id=".uno:SplitCell"/>
+                </menu:menupopup>
+            </menu:menu>
             <menu:menuseparator/>
             <menu:menuitem menu:id=".uno:PageFormatDialog"/>
             <menu:menu menu:id=".uno:PrintRangesMenu">
--- officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu	2006-07-03 14:43:29.000000000 +0200
+++ officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu	2006-07-03 16:44:14.000000000 +0200
@@ -1009,7 +1009,7 @@
 			</node>
 			<node oor:name=".uno:ToggleMergeCells" oor:op="replace">
 				<prop oor:name="Label" oor:type="xs:string">
-					<value xml:lang="en-US">M~erge Cells</value>
+					<value xml:lang="en-US">M~erge and Center Cells</value>
 				</prop>
 				<prop oor:name="Properties" oor:type="xs:int">
 					<value>1</value>
