--- sc/source/filter/xcl97/XclImpChangeTrack.cxx
+++ sc/source/filter/xcl97/XclImpChangeTrack.cxx
@@ -225,6 +225,8 @@
 	SvMemoryStream aMemStrm;
 	aMemStrm << (sal_uInt16) 0x0001 << nFmlSize;
     pStrm->CopyToStream( aMemStrm, nFmlSize );
+	if (nFmlSize == 0xFCFF)
+		return;
     XclImpStream aFmlaStrm( aMemStrm, GetRoot() );
 	aFmlaStrm.StartNextRecord();
     XclImpChTrFmlConverter aFmlConv( GetRoot(), *this );
