diff --git scp2/source/ooo/common_brand.scp scp2/source/ooo/common_brand.scp
index 5e29ab3..8d268c1 100644
--- scp2/source/ooo/common_brand.scp
+++ scp2/source/ooo/common_brand.scp
@@ -323,7 +323,7 @@ Unixlink gid_Brand_Unixlink_BasisLink
     // Styles = (LAYERLINK);
     Name = "basis-link";
     Dir = gid_Dir_Brand_Root;
-    Target = "../${UNIXBASISROOTNAME}/basis${OOOBASEVERSION}";
+    Target = "basis${OOOBASEVERSION}";
 End
 #endif
 
diff --git scp2/source/ooo/directory_ooo.scp scp2/source/ooo/directory_ooo.scp
index 354f8c0..95f4c17 100644
--- scp2/source/ooo/directory_ooo.scp
+++ scp2/source/ooo/directory_ooo.scp
@@ -93,7 +93,7 @@ Directory gid_Dir_Ooo_Openofficeorg
   #ifdef WNT
     HostName = "${BASISROOTNAME}";
   #else
-    HostName = "${UNIXBASISROOTNAME}";
+    HostName = ".";
   #endif
 End
 #endif
diff --git scp2/source/ooo/ooo_brand.scp scp2/source/ooo/ooo_brand.scp
index 663ce28..ecc87d2 100644
--- scp2/source/ooo/ooo_brand.scp
+++ scp2/source/ooo/ooo_brand.scp
@@ -36,7 +36,7 @@ Directory gid_Dir_Brand_Root
   #ifdef WNT
     HostName = "${PRODUCTNAME} ${BRANDPACKAGEVERSION}";
   #else
-    HostName = "${UNIXPRODUCTNAME}${BRANDPACKAGEVERSION}";
+    HostName = ".";
   #endif
     Styles = (OFFICEDIRECTORY);
 End
