--- scp2/source/ooo/directory_ooo.scp.orig	2008-04-13 23:46:59.000000000 +0200
+++ scp2/source/ooo/directory_ooo.scp	2008-05-19 17:33:02.000000000 +0200
@@ -1309,6 +1309,11 @@
     DIR_IDENT_ALL_LANG(forms);
 End
 
+Directory gid_Dir_Template_Labels
+    ParentID = gid_Dir_Template_Language;
+    DIR_IDENT_ALL_LANG(labels);
+End
+
 Directory gid_Dir_Template_Layout
     ParentID = gid_Dir_Template_Language;
     DIR_IDENT_ALL_LANG(layout);
