--- scp2/source/calc/makefile.mk-old	2006-06-20 15:27:41.000000000 +0200
+++ scp2/source/calc/makefile.mk	2006-06-20 15:27:52.000000000 +0200
@@ -49,6 +49,14 @@
 SCPDEFS+=-DWITHOUT_BINFILTER
 .ENDIF
 
+.IF "$(SYSTEM_LPSOLVE)" == "YES"
+SCPDEFS+=-DSYSTEM_LPSOLVE
+.ENDIF
+
+.IF "$(BUILD_SCSOLVER)" == "YES"
+SCPDEFS+=-DBUILD_SCSOLVER
+.ENDIF
+
 SCP_PRODUCT_TYPE=osl
 PARFILES= \
         module_calc.par              \
Index: scp2/source/calc/file_calc.scp
===================================================================
RCS file: /cvs/installation/scp2/source/calc/file_calc.scp,v
retrieving revision 1.29
diff -u -r1.29 file_calc.scp
--- scp2/source/calc/file_calc.scp	26 Apr 2006 20:46:56 -0000	1.29
+++ scp2/source/calc/file_calc.scp	1 Jul 2006 22:41:11 -0000
@@ -35,6 +35,24 @@
 
 #include "macros.inc"
 
+#ifdef BUILD_SCSOLVER
+
+File gid_File_Lib_ScSolver
+    TXT_FILE_BODY;
+    Styles = (PACKED,UNO_COMPONENT);
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Dir = gid_Dir_Program;
+  #ifdef UNX
+    Name = STRING(CONCAT2(scsolver.uno,UNXSUFFIX));
+  #else
+    Name = "scsolver.uno.dll";
+  #endif
+End
+
+STD_RES_FILE_PATCH( gid_File_Res_ScSolver, scsolver )
+
+#endif
+
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Analysis, analysis)
 
 STD_UNO_LIB_FILE( gid_File_Lib_Calc, calc)
diff --git scp2/source/ooo/makefile.mk scp2/source/ooo/makefile.mk
index 0e3094e..6cd84e3 100644
--- scp2/source/ooo/makefile.mk
+++ scp2/source/ooo/makefile.mk
@@ -55,6 +55,10 @@ SCPDEFS+=-DENABLE_CRASHDUMP
 SCPDEFS+=-DBUILD_SPECIAL
 .ENDIF
 
+.IF "$(BUILD_SCSOLVER)" == "YES"
+SCPDEFS+=-DBUILD_SCSOLVER
+.ENDIF
+
 .IF "$(JAVANUMVER)" >= "000100040000"
 SCPDEFS+=-DINCLUDE_JAVA_ACCESSBRIDGE
 .ENDIF
diff --git scp2/source/ooo/module_lang_template.scp scp2/source/ooo/module_lang_template.scp
index 860aa9e..15562a0 100755
--- scp2/source/ooo/module_lang_template.scp
+++ scp2/source/ooo/module_lang_template.scp
@@ -141,6 +141,9 @@ Module gid_Module_Langpack_Calc_Template
              gid_File_Res_Bf_Sc,
              gid_File_Res_Date,
              gid_File_Res_Sc,
+#ifdef BUILD_SCSOLVER
+             gid_File_Res_ScSolver,
+#endif
              gid_File_Res_Solver);
 End
 
