--- sd/source/ui/func/fusel.cxx	2008-04-10 22:25:23.000000000 +0200
+++ sd/source/ui/func/fusel.cxx	2008-05-10 16:26:03.000000000 +0200
@@ -414,26 +418,29 @@
                         }
                     }
 
-                    if ( !bReadOnly &&
-                        bMarked                                                   &&
-                        (!rMEvt.IsShift() || mpView->IsMarkedHit(aMDPos, nHitLog)))
-                    {
-                        /**********************************************************
-                        * Objekt verschieben
-                        **********************************************************/
-                        aDragTimer.Start();
-
-                        pHdl=mpView->PickHandle(aMDPos);
-                        if ( ! rMEvt.IsRight())
-                            mpView->BegDragObj(aMDPos, (OutputDevice*) NULL, pHdl, nDrgLog);
-                    }
-                    else
+                    if( !bDeactivateOLE )
                     {
-                        /**********************************************************
-                        * Objekt selektieren
-                        **********************************************************/
-                        if ( ! rMEvt.IsRight())
-                            mpView->BegMarkObj(aMDPos);
+                        if ( !bReadOnly &&
+                             bMarked                                                   &&
+                             (!rMEvt.IsShift() || mpView->IsMarkedHit(aMDPos, nHitLog)))
+                        {
+                            /**********************************************************
+                             * Objekt verschieben
+                             **********************************************************/
+                            aDragTimer.Start();
+
+                            pHdl=mpView->PickHandle(aMDPos);
+                            if ( ! rMEvt.IsRight())
+                                mpView->BegDragObj(aMDPos, (OutputDevice*) NULL, pHdl, nDrgLog);
+                        }
+                        else
+                        {
+                            /**********************************************************
+                             * Objekt selektieren
+                             **********************************************************/
+                            if ( ! rMEvt.IsRight())
+                                mpView->BegMarkObj(aMDPos);
+                        }
                     }
                 }
             }
