Index: sfx2/source/appl/shutdownicon.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/shutdownicon.cxx,v
retrieving revision 1.61
diff -u -p -r1.61 shutdownicon.cxx
--- sfx2/source/appl/shutdownicon.cxx	2 Nov 2007 13:13:41 -0000	1.61
+++ sfx2/source/appl/shutdownicon.cxx	17 Apr 2008 10:37:07 -0000
@@ -531,9 +531,8 @@ IMPL_STATIC_LINK( ShutdownIcon, DialogCl
         }
     }
 
-#ifdef WNT
-        LeaveModalMode();
-#endif
+    LeaveModalMode();
+
     return 0;
 }
 
