--- sfx2/source/appl/shutdownicon.cxx.old	2008-10-15 20:02:56.000000000 +0200
+++ sfx2/source/appl/shutdownicon.cxx	2008-10-15 21:00:25.000000000 +0200
@@ -52,6 +52,7 @@
 #include <com/sun/star/ui/dialogs/ControlActions.hpp>
 #include <com/sun/star/document/MacroExecMode.hpp>
 #include <com/sun/star/document/UpdateDocMode.hpp>
+#include <rtl/bootstrap.hxx>
 #include <sfx2/filedlghelper.hxx>
 #include <sfx2/fcontnr.hxx>
 #ifndef _UNOTOOLS_PROCESSFACTORY_HXX
@@ -83,6 +84,8 @@ using namespace ::vos;
 using namespace ::rtl;
 using namespace ::sfx2;
 
+extern "C" { static void SAL_CALL thisModule() {} }
+
 class SfxNotificationListener_Impl : public cppu::WeakImplHelper1< XDispatchResultListener >
 {
 public:
@@ -142,7 +145,7 @@ bool ShutdownIcon::LoadModule( osl::Modu
 
 	oslGenericFunction pTmpInit = NULL;
 	oslGenericFunction pTmpDeInit = NULL;
-	if ( pPlugin->load( OUString (RTL_CONSTASCII_USTRINGPARAM( STRING( PLUGIN_NAME ) ) ) ) )
+	if ( pPlugin->loadRelative( &thisModule, OUString (RTL_CONSTASCII_USTRINGPARAM( STRING( PLUGIN_NAME ) ) ) ) )
 	{
 		pTmpInit = pPlugin->getFunctionSymbol(
 			OUString( RTL_CONSTASCII_USTRINGPARAM( "plugin_init_sys_tray" ) ) );
@@ -825,17 +828,19 @@ void ShutdownIcon::SetAutostart( bool bA
 #else // UNX
 		getDotAutostart( true );
 
-		OUString aPath;
-		::utl::Bootstrap::locateBaseInstallation(aPath);
+		OUString aPath( RTL_CONSTASCII_USTRINGPARAM("${BRAND_BASE_DIR}/share/xdg/qstart.desktop" ) );
+		Bootstrap::expandMacros( aPath );
 
 		OUString aDesktopFile;
 		::osl::File::getSystemPathFromFileURL( aPath, aDesktopFile );
-		aDesktopFile += OUString( RTL_CONSTASCII_USTRINGPARAM( "/share/xdg/qstart.desktop" ) );
 
 		OString aDesktopFileUnx = OUStringToOString( aDesktopFile,
 													 osl_getThreadTextEncoding() );
 		OString aShortcutUnx = OUStringToOString( aShortcut,
 												  osl_getThreadTextEncoding() );
+		// call unlink just in case there is a broken link pointing to an older OOo that
+		// is not longer available on the system
+		unlink( aShortcutUnx );
 		symlink( aDesktopFileUnx, aShortcutUnx );
 
 		ShutdownIcon *pIcon = ShutdownIcon::createInstance();
