diff --git a/svx/inc/svx/colrctrl.hxx b/svx/inc/svx/colrctrl.hxx
index ab0dd63..2d8441c 100644
--- svx/inc/svx/colrctrl.hxx
+++ svx/inc/svx/colrctrl.hxx
@@ -136,8 +136,7 @@ public:
 										   const ResId& rResId );
 					~SvxColorDockingWindow();
 
-	virtual void 	SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-							const SfxHint& rHint, const TypeId& rHintType );
+	virtual void 	Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 
 	void			SetSlotIDs( USHORT nLeft, USHORT nRight )
 						{ nLeftSlot = nLeft; nRightSlot = nRight; }
diff --git a/svx/inc/svx/fntctl.hxx b/svx/inc/svx/fntctl.hxx
index cbe6e9b..5ff9f07 100644
--- svx/inc/svx/fntctl.hxx
+++ svx/inc/svx/fntctl.hxx
@@ -54,10 +54,7 @@ private:
 	DECL_LINK( MenuSelect, FontNameMenu * );
 	SVX_DLLPRIVATE virtual void	StateChanged( USHORT nSID, SfxItemState eState,
 								  const SfxPoolItem* pState );
-	SVX_DLLPRIVATE virtual void 	SFX_NOTIFY( SfxBroadcaster& rBC,
-								const TypeId& rBCType,
-								const SfxHint& rHint,
-								const TypeId& rHintType );
+	SVX_DLLPRIVATE virtual void 	Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 //#endif
 
 public:
diff --git a/svx/inc/svx/ruler.hxx b/svx/inc/svx/ruler.hxx
index e3c0d7d..caf9ec1 100644
--- svx/inc/svx/ruler.hxx
+++ svx/inc/svx/ruler.hxx
@@ -216,8 +216,7 @@ protected:
 	virtual void    ExtraDown();
 	virtual void	MouseMove( const MouseEvent& rMEvt );
 
-	virtual void SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-						 const SfxHint& rHint, const TypeId& rHintType );
+	virtual void    Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 
 	// Berechnung der Grenzwerte fuer Objectgrenzen
 	// Werte sind bezogen auf die Seite
diff --git a/svx/inc/svx/scene3d.hxx b/svx/inc/svx/scene3d.hxx
index 3951d93..6a178d0 100644
--- svx/inc/svx/scene3d.hxx
+++ svx/inc/svx/scene3d.hxx
@@ -122,10 +122,7 @@ protected:
 
 	void RebuildLists();
 
-	virtual void SFX_NOTIFY(SfxBroadcaster &rBC,
-							const TypeId   &rBCType,
-							const SfxHint  &rHint,
-							const TypeId   &rHintType);
+	virtual void Notify(SfxBroadcaster &rBC, const SfxHint  &rHint);
 
 	BOOL AreThereTransparentParts() const;
 
diff --git a/svx/inc/svx/svdedxv.hxx b/svx/inc/svx/svdedxv.hxx
index c438a6d..a187240 100644
--- svx/inc/svx/svdedxv.hxx
+++ svx/inc/svx/svdedxv.hxx
@@ -158,7 +158,7 @@ public:
 	virtual void BckAction();
 	virtual void TakeActionRect(Rectangle& rRect) const;
 
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 	virtual void ModelHasChanged();
 
 	//************************************************************************
diff --git a/svx/inc/svx/svdmrkv.hxx b/svx/inc/svx/svdmrkv.hxx
index 1dcbe6b..91edbd6 100644
--- svx/inc/svx/svdmrkv.hxx
+++ svx/inc/svx/svdmrkv.hxx
@@ -166,7 +166,7 @@ private:
     void UndirtyMrkPnt() const;
 
 protected:
-    virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+    virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
     virtual void ModelHasChanged(); // Wird von der PaintView gerufen
     virtual void SetMarkHandles();                                           // aHdl (Liste der Handles) fuellen
     virtual void SetMarkRects();                                             // Rects an den PageViews
diff --git a/svx/inc/svx/svdoattr.hxx b/svx/inc/svx/svdoattr.hxx
index 751c083..fb8bbfb 100644
--- svx/inc/svx/svdoattr.hxx
+++ svx/inc/svx/svdoattr.hxx
@@ -80,7 +80,7 @@ protected:
     FASTBOOL ImpSetShadowAttributes( const SfxItemSet& rSet, SfxItemSet& rShadowSet ) const;
 
 	// Zuhoeren, ob sich ein StyleSheet aendert
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 	SdrAttrObj();
 	virtual ~SdrAttrObj();
diff --git a/svx/inc/svx/svdocapt.hxx b/svx/inc/svx/svdocapt.hxx
index 428e9b9..c671bfe 100644
--- svx/inc/svx/svdocapt.hxx
+++ svx/inc/svx/svdocapt.hxx
@@ -113,7 +113,7 @@ public:
 
 	virtual basegfx::B2DPolyPolygon TakeXorPoly(sal_Bool bDetail) const;
 	virtual void SetModel(SdrModel* pNewModel);
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 	virtual sal_uInt32 GetHdlCount() const;
 	virtual SdrHdl* GetHdl(sal_uInt32 nHdlNum) const;
diff --git a/svx/inc/svx/svdocirc.hxx b/svx/inc/svx/svdocirc.hxx
index 634cc72..ec7b08b 100644
--- svx/inc/svx/svdocirc.hxx
+++ svx/inc/svx/svdocirc.hxx
@@ -93,7 +93,7 @@ private:
 	SVX_DLLPRIVATE virtual void RecalcXPoly();
 
 protected:
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 public:
 	TYPEINFO();
diff --git a/svx/inc/svx/svdoedge.hxx b/svx/inc/svx/svdoedge.hxx
index fda6329..087e7fd 100644
--- svx/inc/svx/svdoedge.hxx
+++ svx/inc/svx/svdoedge.hxx
@@ -203,7 +203,7 @@ public:
 	sal_Bool IsBoundRectCalculationRunning() const { return mbBoundRectCalculationRunning; }
 
 protected:
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 	XPolygon ImpCalcObjToCenter(const Point& rStPt, long nEscAngle, const Rectangle& rRect, const Point& rCenter) const;
 	void ImpRecalcEdgeTrack();  // Neuberechnung des Verbindungsverlaufs
diff --git a/svx/inc/svx/svdograf.hxx b/svx/inc/svx/svdograf.hxx
index fe5e4b1..d7bd973 100644
--- svx/inc/svx/svdograf.hxx
+++ svx/inc/svx/svdograf.hxx
@@ -222,8 +222,7 @@ public:
 
 	void 					AdjustToMaxRect( const Rectangle& rMaxRect, BOOL bShrinkOnly );
 
-	virtual void			SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-										const SfxHint& rHint, const TypeId& rHintType );
+	virtual void			Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 
 	bool					IsMirrored() { return bMirrored; }
 	void					SetMirrored( bool _bMirrored ) { bMirrored = _bMirrored; }
diff --git a/svx/inc/svx/svdorect.hxx b/svx/inc/svx/svdorect.hxx
index e4c86fa..905cc10 100644
--- svx/inc/svx/svdorect.hxx
+++ svx/inc/svx/svdorect.hxx
@@ -148,7 +148,7 @@ public:
 
 	virtual SdrObject* DoConvertToPolyObj(BOOL bBezier) const;
 
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 };
 
 #endif //_SVDORECT_HXX
diff --git a/svx/inc/svx/svdovirt.hxx b/svx/inc/svx/svdovirt.hxx
index 79ef5a2..3c19feb 100644
--- svx/inc/svx/svdovirt.hxx
+++ svx/inc/svx/svdovirt.hxx
@@ -58,7 +58,7 @@ protected:
 	Rectangle					aSnapRect;
 
 protected:
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 	virtual SdrObjGeoData* NewGeoData() const;
 	virtual void SaveGeoData(SdrObjGeoData& rGeo) const;
diff --git a/svx/inc/svx/svdpagv.hxx b/svx/inc/svx/svdpagv.hxx
index 288582e..acc0f25 100644
--- svx/inc/svx/svdpagv.hxx
+++ svx/inc/svx/svdpagv.hxx
@@ -142,7 +142,7 @@ protected:
 	sal_Bool IsLayer(const String& rName, const SetOfByte& rBS) const;
 	void SetAllLayers(SetOfByte& rB, sal_Bool bJa);
 
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 	// Nachsehen, ob AktGroup noch Inserted ist.
 	void CheckAktGroup();
diff --git a/svx/inc/svx/svdpntv.hxx b/svx/inc/svx/svdpntv.hxx
index 202b12b..e412a94 100644
--- svx/inc/svx/svdpntv.hxx
+++ svx/inc/svx/svdpntv.hxx
@@ -272,7 +272,7 @@ public:
 	BOOL ImpIsGlueVisible() { return bGlueVisible || bGlueVisible2 || bGlueVisible3 || bGlueVisible4; }
 protected:
 
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 	void GlueInvalidate() const;
 
 	void ShowEncirclement(OutputDevice* pOut);
diff --git a/svx/inc/svx/svdview.hxx b/svx/inc/svx/svdview.hxx
index 721e812..fb79b7a 100644
--- svx/inc/svx/svdview.hxx
+++ svx/inc/svx/svdview.hxx
@@ -197,7 +197,7 @@ class SVX_DLLPUBLIC SdrView: public SdrCreateView, public tools::WeakBase< SdrVi
 protected:
 	SvtAccessibilityOptions	maAccessibilityOptions;
 
-	virtual void SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType);
+	virtual void Notify(SfxBroadcaster& rBC, const SfxHint& rHint);
 
 public:
 	TYPEINFO();
diff --git a/svx/source/dialog/svxruler.cxx b/svx/source/dialog/svxruler.cxx
index df7684f..a6f5506 100644
--- svx/source/dialog/svxruler.cxx
+++ svx/source/dialog/svxruler.cxx
@@ -3621,8 +3621,7 @@ void __EXPORT SvxRuler::ExtraDown()
 }
 
 
-void __EXPORT SvxRuler::SFX_NOTIFY(SfxBroadcaster& , const TypeId& rBCType,
-							   const SfxHint& rHint, const TypeId& rHintType)
+void __EXPORT SvxRuler::Notify(SfxBroadcaster&, const SfxHint& rHint)
 /*
 
    [Beschreibung]
diff --git a/svx/source/engine3d/scene3d.cxx b/svx/source/engine3d/scene3d.cxx
index 207bf0d..2811cfd 100644
--- svx/source/engine3d/scene3d.cxx
+++ svx/source/engine3d/scene3d.cxx
@@ -1010,13 +1010,10 @@ void E3dScene::RestGeoData(const SdrObjGeoData& rGeo)
 |*
 \************************************************************************/
 
-void E3dScene::SFX_NOTIFY(SfxBroadcaster &rBC,
-						  const TypeId   &rBCType,
-						  const SfxHint  &rHint,
-						  const TypeId   &rHintType)
+void E3dScene::Notify(SfxBroadcaster &rBC, const SfxHint  &rHint)
 {
 	SetRectsDirty();
-	E3dObject::SFX_NOTIFY(rBC, rBCType, rHint, rHintType);
+	E3dObject::Notify(rBC, rHint);
 }
 
 /*************************************************************************
diff --git a/svx/source/items/boxobj.hxx b/svx/source/items/boxobj.hxx
index 77f42d6..171d4f7 100644
--- svx/source/items/boxobj.hxx
+++ svx/source/items/boxobj.hxx
@@ -66,8 +66,7 @@ private:
 
 protected:
 	// Schnittstelle zum SbxObject/SvDispatch/SfxShell..Object
-	void					SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-									const SfxHint& rHint, const TypeId& rHintType );
+	void					Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 	virtual ULONG			SetProperty( USHORT nSID, SbxVariable &rRet );
 	virtual ULONG			GetProperty( USHORT nSID, SbxVariable &rRet );
 
diff --git a/svx/source/mnuctrls/fntctl.cxx b/svx/source/mnuctrls/fntctl.cxx
index 2502874..c4e9a8a 100644
--- svx/source/mnuctrls/fntctl.cxx
+++ svx/source/mnuctrls/fntctl.cxx
@@ -132,8 +132,7 @@ void SvxFontMenuControl::StateChanged(
 	der DocumentShell gef"ullt.
 */
 
-void SvxFontMenuControl::SFX_NOTIFY( SfxBroadcaster&, const TypeId& rBCType,
-								 const SfxHint& rHint, const TypeId& rHintType )
+void SvxFontMenuControl::Notify( SfxBroadcaster&, const SfxHint& rHint )
 {
 	if ( rHint.Type() != TYPE(SfxSimpleHint) &&
 		 ( (SfxSimpleHint&)rHint ).GetId() == SFX_HINT_DOCCHANGED )
diff --git a/svx/source/sdr/properties/graphicproperties.cxx b/svx/source/sdr/properties/graphicproperties.cxx
index 78ceedc..b1a49d6 100644
--- svx/source/sdr/properties/graphicproperties.cxx
+++ svx/source/sdr/properties/graphicproperties.cxx
@@ -91,7 +91,7 @@ namespace sdr
 			rObj.SetXPolyDirty();
 
             // #i29367# Update GraphicAttr, too. This was formerly
-            // triggered by SdrGrafObj::SFX_NOTIFY, which is no longer
+            // triggered by SdrGrafObj::Notify, which is no longer
             // called nowadays. BTW: strictly speaking, the whole
             // ImpSetAttrToGrafInfo/ImpSetGrafInfoToAttr stuff could
             // be dumped, when SdrGrafObj::aGrafInfo is removed and
diff --git a/svx/source/svdraw/svdedxv.cxx b/svx/source/svdraw/svdedxv.cxx
index f419a77..11a14da 100644
--- svx/source/svdraw/svdedxv.cxx
+++ svx/source/svdraw/svdedxv.cxx
@@ -157,9 +157,9 @@ void SdrObjEditView::TakeActionRect(Rectangle& rRect) const
     }
 }
 
-void __EXPORT SdrObjEditView::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void __EXPORT SdrObjEditView::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
-    SdrGlueEditView::SFX_NOTIFY(rBC,rBCType,rHint,rHintType);
+    SdrGlueEditView::Notify(rBC,rHint);
     // Printerwechsel waerend des Editierens
     SdrHint* pSdrHint=PTR_CAST(SdrHint,&rHint);
     if (pSdrHint!=NULL && pTextEditOutliner!=NULL) {
diff --git a/svx/source/svdraw/svdmrkv.cxx b/svx/source/svdraw/svdmrkv.cxx
index 787567a..9f5063e 100644
--- svx/source/svdraw/svdmrkv.cxx
+++ svx/source/svdraw/svdmrkv.cxx
@@ -203,7 +203,7 @@ SdrMarkView::~SdrMarkView()
 	delete mpSdrViewSelection;
 }
 
-void __EXPORT SdrMarkView::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void __EXPORT SdrMarkView::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
 	SdrHint* pSdrHint=PTR_CAST(SdrHint,&rHint);
 	if (pSdrHint!=NULL) 
@@ -216,7 +216,7 @@ void __EXPORT SdrMarkView::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType
 			bMarkedPointsRectsDirty=TRUE;
 		}
 	}
-	SdrSnapView::SFX_NOTIFY(rBC,rBCType,rHint,rHintType);
+	SdrSnapView::Notify(rBC,rHint);
 }
 
 void SdrMarkView::ModelHasChanged()
diff --git a/svx/source/svdraw/svdoattr.cxx b/svx/source/svdraw/svdoattr.cxx
index 45947f8..fc67cad 100644
--- svx/source/svdraw/svdoattr.cxx
+++ svx/source/svdraw/svdoattr.cxx
@@ -141,8 +141,7 @@ void SdrAttrObj::SetModel(SdrModel* pNewModel)
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // syntactical sugar for ItemSet accesses
 
-void __EXPORT SdrAttrObj::SFX_NOTIFY(SfxBroadcaster& /*rBC*/, const TypeId& rBCType,
-	const SfxHint& rHint, const TypeId& rHintType)
+void __EXPORT SdrAttrObj::Notify(SfxBroadcaster& /*rBC*/, const SfxHint& rHint)
 {
 	SfxSimpleHint *pSimple = PTR_CAST(SfxSimpleHint, &rHint);
 	BOOL bDataChg(pSimple && SFX_HINT_DATACHANGED == pSimple->GetId());
diff --git a/svx/source/svdraw/svdocapt.cxx b/svx/source/svdraw/svdocapt.cxx
index ae8336f..ec9d030 100644
--- svx/source/svdraw/svdocapt.cxx
+++ svx/source/svdraw/svdocapt.cxx
@@ -867,9 +867,9 @@ void SdrCaptionObj::SetModel(SdrModel* pNewModel)
 	ImpRecalcTail();
 }
 
-void SdrCaptionObj::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void SdrCaptionObj::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
-	SdrRectObj::SFX_NOTIFY(rBC,rBCType,rHint,rHintType);
+	SdrRectObj::Notify(rBC,rHint);
 	ImpRecalcTail();
 }
 
diff --git a/svx/source/svdraw/svdocirc.cxx b/svx/source/svdraw/svdocirc.cxx
index 8b08c53..b0a20de 100644
--- svx/source/svdraw/svdocirc.cxx
+++ svx/source/svdraw/svdocirc.cxx
@@ -1355,10 +1355,10 @@ Point SdrCircObj::GetSnapPoint(sal_uInt32 i) const
 	}
 }
 
-void __EXPORT SdrCircObj::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void __EXPORT SdrCircObj::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
 	SetXPolyDirty();
-	SdrRectObj::SFX_NOTIFY(rBC,rBCType,rHint,rHintType);
+	SdrRectObj::Notify(rBC,rHint);
 	ImpSetAttrToCircInfo();
 }
 
diff --git a/svx/source/svdraw/svdoedge.cxx b/svx/source/svdraw/svdoedge.cxx
index 59a18d7..0c5a874 100644
--- svx/source/svdraw/svdoedge.cxx
+++ svx/source/svdraw/svdoedge.cxx
@@ -1661,7 +1661,7 @@ je Objekt variiert von 0-3:
 'C':  n  0-3  0-3   = 1+U+1
 */
 
-void __EXPORT SdrEdgeObj::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId&, const SfxHint& rHint, const TypeId&)
+void __EXPORT SdrEdgeObj::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
 	SfxSimpleHint* pSimple=PTR_CAST(SfxSimpleHint,&rHint);
 	ULONG nId=pSimple==0 ? 0 : pSimple->GetId();
@@ -1676,7 +1676,7 @@ void __EXPORT SdrEdgeObj::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId&, const S
 		if (bObj2) aCon2.pObj=NULL;
 		return; // Und mehr braucht hier nicht getan werden.
 	}
-	SdrTextObj::SFX_NOTIFY(rBC,rBCType,rHint,rHintType);
+	SdrTextObj::Notify(rBC,rHint);
 	if (nNotifyingCount==0) { // Hier nun auch ein VerriegelungsFlag
 		((SdrEdgeObj*)this)->nNotifyingCount++;
 		SdrHint* pSdrHint=PTR_CAST(SdrHint,&rHint);
@@ -1712,13 +1712,13 @@ void SdrEdgeObj::Reformat()
 	if( NULL != aCon1.pObj )
 	{
 		SfxSimpleHint aHint( SFX_HINT_DATACHANGED );
-		SFX_NOTIFY( *const_cast<SfxBroadcaster*>(aCon1.pObj->GetBroadcaster()), NULL, aHint, NULL );
+		Notify( *const_cast<SfxBroadcaster*>(aCon1.pObj->GetBroadcaster()), aHint );
 	}
 
 	if( NULL != aCon2.pObj )
 	{
 		SfxSimpleHint aHint( SFX_HINT_DATACHANGED );
-		SFX_NOTIFY( *const_cast<SfxBroadcaster*>(aCon2.pObj->GetBroadcaster()), NULL, aHint, NULL );
+		Notify( *const_cast<SfxBroadcaster*>(aCon2.pObj->GetBroadcaster()), aHint );
 	}
 }
 
diff --git a/svx/source/svdraw/svdograf.cxx b/svx/source/svdraw/svdograf.cxx
index 205fc86..481ce93 100644
--- svx/source/svdraw/svdograf.cxx
+++ svx/source/svdraw/svdograf.cxx
@@ -1259,11 +1259,10 @@ SdrObject* SdrGrafObj::DoConvertToPolyObj(BOOL bBezier) const
 
 // -----------------------------------------------------------------------------
 
-void SdrGrafObj::SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-							 const SfxHint& rHint, const TypeId& rHintType )
+void SdrGrafObj::Notify( SfxBroadcaster& rBC, const SfxHint& rHint )
 {
 	SetXPolyDirty();
-	SdrRectObj::SFX_NOTIFY( rBC, rBCType, rHint, rHintType );
+	SdrRectObj::Notify( rBC, rHint );
 	ImpSetAttrToGrafInfo();
 }
 
diff --git a/svx/source/svdraw/svdorect.cxx b/svx/source/svdraw/svdorect.cxx
index 3a7db4b..5ffee4f 100644
--- svx/source/svdraw/svdorect.cxx
+++ svx/source/svdraw/svdorect.cxx
@@ -844,9 +844,9 @@ SdrObject* SdrRectObj::DoConvertToPolyObj(BOOL bBezier) const
 	return pRet;
 }
 
-void SdrRectObj::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void SdrRectObj::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
-	SdrTextObj::SFX_NOTIFY(rBC,rBCType,rHint,rHintType);
+	SdrTextObj::Notify(rBC,rHint);
 	SetXPolyDirty(); // wg. Eckenradius
 }
 
diff --git a/svx/source/svdraw/svdovirt.cxx b/svx/source/svdraw/svdovirt.cxx
index 83799d7..dd2f777 100644
--- svx/source/svdraw/svdovirt.cxx
+++ svx/source/svdraw/svdovirt.cxx
@@ -92,7 +92,7 @@ SdrObject& SdrVirtObj::ReferencedObj()
 	return rRefObj;
 }
 
-void __EXPORT SdrVirtObj::SFX_NOTIFY(SfxBroadcaster& /*rBC*/, const TypeId& rBCType, const SfxHint& /*rHint*/, const TypeId& rHintType)
+void __EXPORT SdrVirtObj::Notify(SfxBroadcaster& /*rBC*/, const SfxHint& /*rHint*/)
 {
 	bClosedObj=rRefObj.IsClosedObj();
 	SetRectsDirty(); // hier noch Optimieren.
diff --git a/svx/source/svdraw/svdpagv.cxx b/svx/source/svdraw/svdpagv.cxx
index 529776c..9f6a388 100644
--- svx/source/svdraw/svdpagv.cxx
+++ svx/source/svdraw/svdpagv.cxx
@@ -275,7 +275,7 @@ void SdrPageView::RemovePaintWindowFromPageView(SdrPaintWindow& rPaintWindow)
 	return xReturn;
 }
 
-void __EXPORT SdrPageView::SFX_NOTIFY(SfxBroadcaster& /*rBC*/, const TypeId& /*rBCType*/, const SfxHint& /*rHint*/, const TypeId& /*rHintType*/)
+void __EXPORT SdrPageView::Notify(SfxBroadcaster& /*rBC*/, const SfxHint& /*rHint*/)
 {
     // not really interested in
 }
diff --git a/svx/source/svdraw/svdpntv.cxx b/svx/source/svdraw/svdpntv.cxx
index e88d886..3623a70 100644
--- svx/source/svdraw/svdpntv.cxx
+++ svx/source/svdraw/svdpntv.cxx
@@ -327,7 +327,7 @@ SdrPaintView::~SdrPaintView()
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void __EXPORT SdrPaintView::SFX_NOTIFY(SfxBroadcaster& /*rBC*/, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void __EXPORT SdrPaintView::Notify(SfxBroadcaster& /*rBC*/, const SfxHint& rHint)
 {
 	BOOL bObjChg=!bSomeObjChgdFlag; // TRUE= auswerten fuer ComeBack-Timer
 	if (bObjChg) {
diff --git a/svx/source/svdraw/svdview.cxx b/svx/source/svdraw/svdview.cxx
index 2e7f562..8f13ac3 100644
--- svx/source/svdraw/svdview.cxx
+++ svx/source/svdraw/svdview.cxx
@@ -1550,14 +1550,14 @@ BOOL SdrView::IsDeleteMarkedPossible() const
 	return IsDeleteMarkedObjPossible();
 }
 
-void SdrView::SFX_NOTIFY(SfxBroadcaster& rBC, const TypeId& rBCType, const SfxHint& rHint, const TypeId& rHintType)
+void SdrView::Notify(SfxBroadcaster& rBC, const SfxHint& rHint)
 {
 	if( rHint.ISA( SfxSimpleHint ) && ( (SfxSimpleHint&) rHint ).GetId() == SFX_HINT_ACCESSIBILITY_CHANGED )
 	{
 		onAccessibilityOptionsChanged();
 	}
 
- 	SdrCreateView::SFX_NOTIFY(rBC, rBCType, rHint, rHintType );
+ 	SdrCreateView::Notify(rBC, rHint);
 }
 
 SvtAccessibilityOptions& SdrView::getAccessibilityOptions()
diff --git a/svx/source/tbxctrls/colrctrl.cxx b/svx/source/tbxctrls/colrctrl.cxx
index 2fffe37..c69587b 100644
--- svx/source/tbxctrls/colrctrl.cxx
+++ svx/source/tbxctrls/colrctrl.cxx
@@ -357,10 +357,7 @@ SvxColorDockingWindow::~SvxColorDockingWindow()
 |*
 \************************************************************************/
 
-void SvxColorDockingWindow::SFX_NOTIFY( SfxBroadcaster& ,
-										 const TypeId& rBCType,
-										 const SfxHint& rHint,
-										 const TypeId& rHintType )
+void SvxColorDockingWindow::Notify( SfxBroadcaster& , const SfxHint& rHint )
 {
 	const SfxPoolItemHint *pPoolItemHint = PTR_CAST(SfxPoolItemHint, &rHint);
 	if ( pPoolItemHint
