Index: automation/source/app/testbasi.cxx
===================================================================
RCS file: /cvs/util/automation/source/app/testbasi.cxx,v
retrieving revision 1.9
retrieving revision 1.9.14.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.9 -r1.9.14.1
--- automation/source/app/testbasi.cxx	11 Apr 2008 09:54:34 -0000	1.9
+++ automation/source/app/testbasi.cxx	23 May 2008 20:00:11 -0000	1.9.14.1
@@ -152,9 +152,9 @@
 {
 	SbxVariableRef aDummy = new SbxVariable;
 	aDummy->SetUserData( 24 );  // ID_MaybeAddErr
-	((TestToolObj*)pTestObject)->SFX_NOTIFY( pTestObject->GetBroadcaster(), xx, SbxHint( SBX_HINT_DATAWANTED, aDummy ), xx );
+	((TestToolObj*)pTestObject)->Notify( pTestObject->GetBroadcaster(), SbxHint( SBX_HINT_DATAWANTED, aDummy ) );
 	aDummy->SetUserData( 18 );  // ID_ExceptLog
-	((TestToolObj*)pTestObject)->SFX_NOTIFY( pTestObject->GetBroadcaster(), xx, SbxHint( SBX_HINT_DATAWANTED, aDummy ), xx );
+	((TestToolObj*)pTestObject)->Notify( pTestObject->GetBroadcaster(), SbxHint( SBX_HINT_DATAWANTED, aDummy ) );
 	MyBasic::ReportRuntimeError( pEditWin );
 }
 
