Index: sfx2/inc/sfx2/viewfrm.hxx
===================================================================
RCS file: /cvs/framework/sfx2/inc/sfx2/viewfrm.hxx,v
retrieving revision 1.5
retrieving revision 1.5.46.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.5 -r1.5.46.1
--- sfx2/inc/sfx2/viewfrm.hxx	11 Apr 2008 11:46:36 -0000	1.5
+++ sfx2/inc/sfx2/viewfrm.hxx	23 May 2008 20:08:51 -0000	1.5.46.1
@@ -145,8 +145,7 @@
 #endif
 
 protected:
-	virtual void            SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-									const SfxHint& rHint, const TypeId& rHintType );
+	virtual void            Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 
 	SAL_DLLPRIVATE void SetWindow_Impl( Window *pWin );
 
Index: sfx2/inc/sfx2/viewsh.hxx
===================================================================
RCS file: /cvs/framework/sfx2/inc/sfx2/viewsh.hxx,v
retrieving revision 1.5
retrieving revision 1.5.46.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.5 -r1.5.46.1
--- sfx2/inc/sfx2/viewsh.hxx	11 Apr 2008 11:46:52 -0000	1.5
+++ sfx2/inc/sfx2/viewsh.hxx	23 May 2008 20:08:51 -0000	1.5.46.1
@@ -185,10 +185,8 @@
 
 	virtual void				Move();
 
-	virtual void                SFX_NOTIFY( SfxBroadcaster& rBC,
-										const TypeId& rBCType,
-										const SfxHint& rHint,
-										const TypeId& rHintType );
+	virtual void                Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
+
 public:
 	// Iteration
 	static SfxViewShell*        GetFirst( const TypeId* pType = 0, BOOL bOnlyVisible = TRUE );
Index: sfx2/source/dialog/templdlg.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/dialog/templdlg.cxx,v
retrieving revision 1.58
retrieving revision 1.58.46.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.58 -r1.58.46.1
--- sfx2/source/dialog/templdlg.cxx	11 Apr 2008 12:50:57 -0000	1.58
+++ sfx2/source/dialog/templdlg.cxx	23 May 2008 20:08:51 -0000	1.58.46.1
@@ -1562,8 +1562,7 @@
 
 
 //-------------------------------------------------------------------------
-void SfxCommonTemplateDialog_Impl::SFX_NOTIFY(SfxBroadcaster& /*rBC*/, const TypeId& rBCType,
-										  const SfxHint& rHint, const TypeId& rHintType)
+void SfxCommonTemplateDialog_Impl::Notify(SfxBroadcaster& /*rBC*/, const SfxHint& rHint)
 {
 	// Aktualisierung anstossen
 	if(rHint.Type() == TYPE(SfxSimpleHint))
Index: sfx2/source/inc/templdgi.hxx
===================================================================
RCS file: /cvs/framework/sfx2/source/inc/templdgi.hxx,v
retrieving revision 1.18
retrieving revision 1.18.46.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.18 -r1.18.46.1
--- sfx2/source/inc/templdgi.hxx	11 Apr 2008 13:28:19 -0000	1.18
+++ sfx2/source/inc/templdgi.hxx	23 May 2008 20:08:51 -0000	1.18.46.1
@@ -226,8 +226,7 @@
 	USHORT				StyleNrToInfoOffset( USHORT i );
 	USHORT				InfoOffsetToStyleNr( USHORT i );
 
-	void				SFX_NOTIFY( SfxBroadcaster& rBC, const TypeId& rBCType,
-									const SfxHint& rHint, const TypeId& rHintType );
+	void				Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 
 	void				FamilySelect( USHORT nId );
 	void				SetFamily( USHORT nId );
Index: sfx2/source/view/prnmon.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/view/prnmon.cxx,v
retrieving revision 1.25
retrieving revision 1.25.46.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.25 -r1.25.46.1
--- sfx2/source/view/prnmon.cxx	11 Apr 2008 13:39:16 -0000	1.25
+++ sfx2/source/view/prnmon.cxx	23 May 2008 20:08:52 -0000	1.25.46.1
@@ -113,17 +113,10 @@
 	SfxViewShell*			GetViewShell() const { return pViewShell; }
 	BOOL					SetPage( USHORT nPage, const String &rPage );
     void                    CreateMonitor();
-	virtual void            SFX_NOTIFY( SfxBroadcaster& rBC,
-										const TypeId& rBCType,
-										const SfxHint& rHint,
-										const TypeId& rHintType );
+	virtual void            Notify( SfxBroadcaster& rBC, const SfxHint& rHint );
 };
 
-void SfxPrintProgress_Impl::SFX_NOTIFY(
-    SfxBroadcaster& /*rBC*/,
-	const TypeId& rBCType,
-	const SfxHint& rHint,
-	const TypeId& rHintType )
+void SfxPrintProgress_Impl::Notify( SfxBroadcaster& /*rBC*/, const SfxHint& rHint )
 {
 	SfxPrintingHint* pPrintHint = PTR_CAST( SfxPrintingHint, &rHint );
 	if ( pPrintHint )
Index: sfx2/source/view/topfrm.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/view/topfrm.cxx,v
retrieving revision 1.96
retrieving revision 1.96.46.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.96 -r1.96.46.1
--- sfx2/source/view/topfrm.cxx	11 Apr 2008 13:40:01 -0000	1.96
+++ sfx2/source/view/topfrm.cxx	23 May 2008 20:08:52 -0000	1.96.46.1
@@ -1151,7 +1151,7 @@
 	}
     }
 
-    SfxViewFrame::SFX_NOTIFY( rBC, rBCType, rHint, rHintType );
+    SfxViewFrame::Notify( rBC, rHint );
 }
 
 //--------------------------------------------------------------------
Index: sfx2/source/view/viewfrm.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/view/viewfrm.cxx,v
retrieving revision 1.136
retrieving revision 1.136.8.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.136 -r1.136.8.1
--- sfx2/source/view/viewfrm.cxx	13 May 2008 13:09:22 -0000	1.136
+++ sfx2/source/view/viewfrm.cxx	23 May 2008 20:08:52 -0000	1.136.8.1
@@ -1221,13 +1221,8 @@
         pDispatcher->Flush();
         EndListening( *xObjSh );
 
-        SFX_NOTIFY( *xObjSh, xObjSh->Type(),
-                SfxSimpleHint(SFX_HINT_TITLECHANGED),
-                TYPE(SfxSimpleHint) );
-
-        SFX_NOTIFY( *xObjSh, xObjSh->Type(),
-                SfxSimpleHint(SFX_HINT_DOCCHANGED),
-                TYPE(SfxSimpleHint) );
+        Notify( *xObjSh, SfxSimpleHint(SFX_HINT_TITLECHANGED) );
+        Notify( *xObjSh, SfxSimpleHint(SFX_HINT_DOCCHANGED) );
 
         if ( 1 == xObjSh->GetOwnerLockCount() && pImp->bObjLocked && xObjSh->GetCreateMode() == SFX_CREATE_MODE_EMBEDDED )
             xObjSh->DoClose();
