Index: basic/source/app/app.cxx
===================================================================
RCS file: /cvs/script/basic/source/app/app.cxx,v
retrieving revision 1.79
retrieving revision 1.78.20.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.79 -r1.78.20.2
--- basic/source/app/app.cxx	11 Jul 2008 18:44:27 -0000	1.79
+++ basic/source/app/app.cxx	31 Jul 2008 00:12:54 -0000	1.78.20.2
@@ -826,8 +826,7 @@
 	bIsAutoRun = bAuto;
 }
 
-void BasicFrame::SFX_NOTIFY( SfxBroadcaster&, const TypeId&,
-							const SfxHint& rHint, const TypeId& )
+void BasicFrame::Notify( SfxBroadcaster&, const SfxHint& rHint )
 {
 	if ( rHint.ISA( TTExecutionStatusHint ) )
 	{
Index: basic/source/app/appbased.cxx
===================================================================
RCS file: /cvs/script/basic/source/app/appbased.cxx,v
retrieving revision 1.13
retrieving revision 1.13.20.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.13 -r1.13.20.1
--- basic/source/app/appbased.cxx	11 Apr 2008 11:29:41 -0000	1.13
+++ basic/source/app/appbased.cxx	23 May 2008 20:03:23 -0000	1.13.20.1
@@ -108,8 +108,7 @@
 	pMod->SetName( CUniString("--").Append( pMod->GetName() ) );
 }
 
-void AppBasEd::SFX_NOTIFY( SfxBroadcaster&, const TypeId&,
-							const SfxHint& rHint, const TypeId& )
+void AppBasEd::Notify( SfxBroadcaster&, const SfxHint& rHint )
 {
 	const SfxSimpleHint* p = PTR_CAST(SfxSimpleHint,&rHint);
 	if( p )
