--- slideshow/source/engine/slide/layermanager.cxx~	2008-06-24 14:12:44.000000000 +0200
+++ slideshow/source/engine/slide/layermanager.cxx	2008-11-09 14:38:17.000000000 +0100
@@ -494,8 +494,9 @@
         {
             if( !mbActive || mrViews.empty() )
                 return;
-
-            if( rShape->isVisible() )
+            
+            // hidden sprite-shape needs render() call still, to hide sprite
+            if( rShape->isVisible() || rShape->isBackgroundDetached() )
                 maUpdateShapes.insert( rShape );
             else
                 addUpdateArea( rShape );
