--- solenv/inc/rules.mk
+++ solenv/inc/rules.mk
@@ -459,15 +459,17 @@ not_existing$/o_%.dpcc : %.c;@noop $(assign all_local_obj+:=$<)
 
 # Objective-C++ files
 $(OBJ)$/%.obj : %.mm
+.IF "$(QUIET_BUILD)"==""
 	@echo ------------------------------
+.ENDIF
 	@echo Making: $@
 .IF "$(OS)"=="MACOSX"
 .IF "$(TEST)"!=""
 	$(objcpp) $(CFLAGS) $(INCLUDE) $(CFLAGSCC) $(OBJCXXFLAGS) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) -E $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $*.mm
 .ELSE
 	@$(RM) $@ $(@:s/.obj/.o/)
-	$(objcpp) $(CFLAGS) $(INCLUDE) $(CFLAGSCC) $(OBJCXXFLAGS) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $*.mm
-	$(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
+	$(QUIET_BUILD) $(objcpp) $(CFLAGS) $(INCLUDE) $(CFLAGSCC) $(OBJCXXFLAGS) $(CFLAGSOBJ) $(CDEFS) $(CDEFSOBJ) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ) $(OBJ)$/$*.o $*.mm
+	$(QUIET_BUILD) $(IFEXIST) $(@:s/.obj/.o/) $(THEN) $(TOUCH) $@ $(FI)
 .ENDIF
 .IF "$(nodep)"==""
 	$(MAKEDEPEND) @$(mktmp -f - -p$(OBJ) $(MKDEPFLAGS) $(CDEFS) $(CDEFSOBJ) $(CDEFSMT) $< ) > $(MISC)$/o_$(@:f:s/.obj/.dpcc/)
