Index: extension_post.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/extension_post.mk,v
retrieving revision 1.2
diff -u -u -r1.2 extension_post.mk
--- solenv/inc/extension_post.mk	6 Jun 2008 08:10:54 -0000	1.2
+++ solenv/inc/extension_post.mk	16 Jun 2008 15:01:36 -0000
@@ -119,21 +119,6 @@
 	$(TYPE) $(MISC)$/desc.tmp.$(EXTNAME) | sed s/UPDATED_SUPPORTED_PLATFORM/$(PLATFORMID)/ > $@
 	@@-$(RM) $(MISC)$/desc.tmp.$(EXTNAME)
 
-# default OOo license text!!!
-# may not fit...
-.IF "$(CUSTOM_LICENSE)"==""
-.IF "$(GUI)" == "WNT"
-PACKLICDEPS=$(SOLARBINDIR)$/osl$/license$$(@:b:s/_/./:e:s/./_/)$$(@:e)
-.ELSE			# "$(GUI)" == "WNT"
-PACKLICDEPS=$(SOLARBINDIR)$/osl$/LICENSE$$(@:b:s/_/./:e:s/./_/)$$(@:e)
-.ENDIF			# "$(GUI)" == "WNT"
-.ELSE			# "$(CUSTOM_LICENSE)" == ""
-PACKLICDEPS=$(CUSTOM_LICENSE)
-.ENDIF			# "$(CUSTOM_LICENSE)" == ""
-$(PACKLICS) : $(PACKLICDEPS)
-	@@-$(MKDIRHIER) $(@:d)
-	$(GNUCOPY) $< $@
-
 #$(COMPONENT_MANIFEST) : $$(@:f)
 $(COMPONENT_MANIFEST) : $(MANIFEST_SRC) $(MANIFEST_DEPS)
 	@@-$(MKDIRHIER) $(@:d)
@@ -112,7 +112,6 @@
 .ENDIF			# "$(LAST_WITH_LANG)"!="$(WITH_LANG)"
 $(DESCRIPTION) $(PHONYDESC) : $(DESCRIPTION_SRC)
 	@@-$(MKDIRHIER) $(@:d)
-	$(PERL) $(SOLARENV)$/bin$/licinserter.pl $(DESCRIPTION_SRC) $(COMPONENT_LIC_TEMPL) $@.$(EXTNAME)
 	@echo LAST_WITH_LANG=$(WITH_LANG) > $(MISC)$/$(TARGET)_lang_track.mk
 	$(TYPE) $@.$(EXTNAME) | sed s/UPDATED_IDENTIFIER/$(IMPLEMENTATION_IDENTIFIER)/ >  $(MISC)$/desc.tmp.$(EXTNAME)
 	@@-$(RM) $@.$(EXTNAME)
Index: extension_pre.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/extension_pre.mk,v
retrieving revision 1.2
diff -u -u -r1.2 extension_pre.mk
--- solenv/inc/extension_pre.mk	6 Jun 2008 08:11:07 -0000	1.2
+++ solenv/inc/extension_pre.mk	16 Jun 2008 15:01:36 -0000
@@ -37,17 +37,9 @@
 COMPONENT_MANIFEST*:=$(EXTENSIONDIR)$/META-INF$/manifest.xml
 MANIFEST_SRC*:=manifest.xml
 
-.IF "$(GUI)" == "WIN" || "$(GUI)" == "WNT"
-PACKLICS*:=$(foreach,i,$(alllangiso) $(EXTENSIONDIR)$/registration$/license_$i.txt)
-COMPONENT_LIC_TEMPL*:=registration/license_xxx.txt
-.ELSE
-PACKLICS*:=$(foreach,i,$(alllangiso) $(EXTENSIONDIR)$/registration$/LICENSE_$i)
-COMPONENT_LIC_TEMPL*:=registration/LICENSE_xxx
-.ENDIF
-
 #TODO:  check ZIP9TARGET for previous use!
 common_build_zip=
-MANIFEST_DEPS=	$(PACKLICS) $(DESCRIPTION) $(COMPONENT_FILES) $(COMPONENT_LIBRARIES) $(COMPONENT_JARFILES) $(COMPONENT_MERGED_XCU) $(EXTENSION_PACKDEPS)
+MANIFEST_DEPS=	$(DESCRIPTION) $(COMPONENT_FILES) $(COMPONENT_LIBRARIES) $(COMPONENT_JARFILES) $(COMPONENT_MERGED_XCU) $(EXTENSION_PACKDEPS)
 ZIP9DEPS=		$(PACKLICS) $(DESCRIPTION) $(COMPONENT_FILES) $(COMPONENT_LIBRARIES) $(COMPONENT_MANIFEST) $(COMPONENT_JARFILES) $(COMPONENT_MERGED_XCU) $(EXTENSION_PACKDEPS)
 ZIP9TARGET=		$(EXTENSION_ZIPNAME)
 ZIP9DIR=		$(EXTENSIONDIR)
