--- solenv/bin/modules/Cws.pm.old	2008-04-10 18:57:34.000000000 +0200
+++ solenv/bin/modules/Cws.pm	2008-06-10 13:19:53.000000000 +0200
@@ -847,7 +847,7 @@ sub get_eis_id
     my $child  = Eis::to_string($self->child());
 
     my $result;
-    eval { $result = $eis->getChildWorkspaceId($master, $child) };
+    eval { $result = int($eis->getChildWorkspaceId($master, $child)) };
     if ( $@ ) {
         carp("ERROR: get_eis_id(): EIS database transaction failed. Reason:\n$@\n");
     }
