--- solenv/inc/extension_post.mk.old	2008-09-26 16:11:11.000000000 +0200
+++ solenv/inc/extension_post.mk	2008-09-26 18:25:04.000000000 +0200
@@ -101,7 +101,7 @@ $(COMPONENT_LIBRARIES) : $(DLLDEST)$/$$(
 .ENDIF 			# "$(OS)$(CPU)"=="WNTI"
 .ENDIF			# "$(COMPONENT_LIBRARIES)"!=""
 
-IMPLEMENTATION_IDENTIFIER*="com.sun.star.$(EXTENSIONNAME)-$(PLATFORMID)"
+IMPLEMENTATION_IDENTIFIER*="com.sun.star.$(EXTENSIONNAME)"
 
 .INCLUDE .IGNORE : $(MISC)$/$(TARGET)_lang_track.mk
 .IF "$(LAST_WITH_LANG)"!="$(WITH_LANG)"
