--- configmgr/source/localbe/localfilehelper.cxx.old	2009-04-02 10:51:56.000000000 +0000
+++ configmgr/source/localbe/localfilehelper.cxx	2009-04-06 16:41:37.000000000 +0000
@@ -139,6 +139,10 @@ namespace configmgr
         if (_sURL.getLength() == 0)
             return false;
 
+		// This method has no right to be so under-performing to
+		// achieve so, so little of any usefulness.
+		return true;
+
         osl::DirectoryItem aDirItem;
         
         osl::DirectoryItem::RC rc = osl::DirectoryItem::get(_sURL, aDirItem);
--- configmgr/source/localbe/localmultistratum.cxx.old	2009-04-02 10:51:56.000000000 +0000
+++ configmgr/source/localbe/localmultistratum.cxx	2009-04-06 16:41:37.000000000 +0000
@@ -83,7 +83,7 @@ uno::Sequence< rtl::OUString > SAL_CALL
     rtl::OUString const aLayerUrl   = impl_getLayerDataDirectory(getBaseUrl());
     rtl::OUString const aComponentUrl = aLayerUrl + componentToPath(aComponent);
     
-    const sal_uInt32 k_STATUS_FIELDS =  FileStatusMask_Type | FileStatusMask_FileName; 
+    const sal_uInt32 k_STATUS_FIELDS = FileStatusMask_FileName; 
     osl::Directory aComponentDirectory(aComponentUrl);
     osl::DirectoryItem aItem;
     std::vector< rtl::OUString > aResult;
@@ -108,13 +108,12 @@ uno::Sequence< rtl::OUString > SAL_CALL
                 OSL_TRACE("Reading Component Directory - Error (%u) getting status of directory item.\n", unsigned(errcode));
                 break;
             }
-
-            OSL_ENSURE( aItemDescriptor.isValid(FileStatusMask_Type), "Could not get type of directory item");
-            if (aItemDescriptor.getFileType() != osl::FileStatus::Regular)
-                continue;
                     
             OSL_ENSURE( aItemDescriptor.isValid(FileStatusMask_FileName), "Could not get name of component found");
             rtl::OUString const aFileName = aItemDescriptor.getFileName();
+
+			// It is reasonable to assume a .xcu file is not a directory & =>
+			// not stat each directory entry at considerable cost.
             if (!aFileName.endsWithIgnoreAsciiCaseAsciiL(RTL_CONSTASCII_STRINGPARAM(kLocalDataSuffix)))
                 continue;
 
--- configmgr/source/misc/filehelper.cxx.old	2009-04-02 10:51:59.000000000 +0000
+++ configmgr/source/misc/filehelper.cxx	2009-04-06 16:41:37.000000000 +0000
@@ -102,9 +102,22 @@ namespace configmgr
 		static const TimeValue k_NullTime = {0,0};		
         sal_uInt64 aSize = 0;
         rModifyTime = k_NullTime;
+		rtl::OUString aURL;
 
-		osl::DirectoryItem aItem;		
-		if (osl::FileBase::E_None == osl::DirectoryItem::get(_sURL, aItem))
+		osl::DirectoryItem aItem;
+
+#ifndef STAT_LOTS_OF_UNUSED_FILES
+		// Statting every file takes way to long - we can use the directory
+		// time-stamp & size instead, if we need to provoke an update
+		// a simple cat > foo; rm foo will do that.
+		// The dir size is (hopefully) non-0 and a good enough proxy value.
+
+		aURL = FileHelper::getParentDir(_sURL);
+#else
+		aURL = _sURL;
+#endif
+
+		if (osl::FileBase::E_None == osl::DirectoryItem::get(aURL, aItem))
 		{
 			osl::FileStatus aStatus(osl_FileStatus_Mask_ModifyTime|osl_FileStatus_Mask_Type|osl_FileStatus_Mask_FileSize);			
 			if (osl::FileBase::E_None == aItem.getFileStatus(aStatus))
