diff -u -r ../HEAD/build/ooo310-m4/configmgr/source/treemgr/setnodeimpl.hxx configmgr/source/treemgr/setnodeimpl.hxx
--- configmgr/source/treemgr/setnodeimpl.hxx
+++ configmgr/source/treemgr/setnodeimpl.hxx
@@ -106,10 +106,21 @@
         };
 	//-----------------------------------------------------------------------------
 
+        struct ElementTreeCompare
+        {
+            bool operator()(const rtl::OUString &a, const rtl::OUString &b) const
+            { // first sort by length - order is immaterial, and it is fast.
+                if (a.getLength() == b.getLength())
+                    return a < b;
+                else
+                    return a.getLength() < b.getLength();
+            }
+        };
+
 		class ElementSet
 		{
 		public:
-			typedef std::map<rtl::OUString, ElementTreeData>	Data;
+		  typedef std::map<rtl::OUString, ElementTreeData,ElementTreeCompare>	Data;
 
 		// the following must be implemented by derived classes
 			bool isEmpty() const { return m_aData.empty(); }
