--- configure.in.old	2009-04-06 16:41:55.000000000 +0000
+++ configure.in	2009-04-06 16:41:55.000000000 +0000
@@ -1581,6 +1581,29 @@ AC_SUBST(HAVE_LD_HASH_STYLE)
 AC_SUBST(WITH_LINKER_HASH_STYLE)
 
 dnl ===================================================================
+dnl  Check for -Bsymbolic-functions linker support
+dnl ===================================================================
+
+HAVE_LD_BSYMBOLIC_FUNCTIONS=
+if test "$GCC" = "yes"; then
+   AC_MSG_CHECKING( for -Bsymbolic-functions linker support )
+   bsymbolic_functions_ldflags_save=$LDFLAGS
+   LDFLAGS="$LDFLAGS -Wl,-Bsymbolic-functions -Wl,--dynamic-list-cpp-new -Wl,--dynamic-list-cpp-typeinfo"
+   AC_LINK_IFELSE([AC_LANG_PROGRAM([
+   #include <stdio.h>
+   ],[
+    printf ("hello world\n");
+   ])], HAVE_LD_BSYMBOLIC_FUNCTIONS=TRUE, [])
+   if test "z$HAVE_LD_BSYMBOLIC_FUNCTIONS" = "zTRUE"; then
+     AC_MSG_RESULT( found )
+   else
+     AC_MSG_RESULT( not found )
+   fi
+   LDFLAGS=$bsymbolic_functions_ldflags_save
+fi
+AC_SUBST(HAVE_LD_BSYMBOLIC_FUNCTIONS)
+
+dnl ===================================================================
 dnl  Test the IRIX SGI Mips pro compiler
 dnl ===================================================================
 if test "$_os" = "IRIX" -o "$_os" = "IRIX64"; then
--- set_soenv.in.old	2009-04-06 16:41:55.000000000 +0000
+++ set_soenv.in	2009-04-06 16:41:55.000000000 +0000
@@ -1882,6 +1882,8 @@ ToFile( "HAVE_GCC_VISIBILITY_BROKEN",
 		"@HAVE_GCC_VISIBILITY_BROKEN@", "e" );
 ToFile( "HAVE_LD_HASH_STYLE","@HAVE_LD_HASH_STYLE@","e" );
 ToFile( "WITH_LINKER_HASH_STYLE","@WITH_LINKER_HASH_STYLE@","e" );
+ToFile( "HAVE_LD_BSYMBOLIC_FUNCTIONS",
+		"@HAVE_LD_BSYMBOLIC_FUNCTIONS@","e" );
 ToFile( "CXX",               $CXX,               "e" );
 ToFile( "MINGWCXX",          "@MINGWCXX@",       "e" );
 ToFile( "MINGWSTRIP",        "@MINGWSTRIP@",     "e" );
--- solenv/inc/unxlngi6.mk.old	2009-04-06 16:41:55.000000000 +0000
+++ solenv/inc/unxlngi6.mk	2009-04-06 16:41:55.000000000 +0000
@@ -158,6 +158,9 @@ LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_NONE=
 # flag -Wl,-z,noexecstack sets the NX bit on the stack
 LINKFLAGS=-Wl,-z,noexecstack -Wl,-z,combreloc $(LINKFLAGSDEFS)
+.IF "$(HAVE_LD_BSYMBOLIC_FUNCTIONS)"  == "TRUE"
+LINKFLAGS += -Wl,-Bsymbolic-functions -Wl,--dynamic-list-cpp-new -Wl,--dynamic-list-cpp-typeinfo
+.ENDIF
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec
--- solenv/inc/unxlngx6.mk.old	2009-04-06 16:41:55.000000000 +0000
+++ solenv/inc/unxlngx6.mk	2009-04-06 16:41:55.000000000 +0000
@@ -162,6 +162,9 @@ LINKFLAGSRUNPATH_BRAND=-Wl,-rpath,\''$$O
 LINKFLAGSRUNPATH_OXT=
 LINKFLAGSRUNPATH_NONE=
 LINKFLAGS=-Wl,-z,combreloc $(LINKFLAGSDEFS)
+.IF "$(HAVE_LD_BSYMBOLIC_FUNCTIONS)"  == "TRUE"
+LINKFLAGS += -Wl,-Bsymbolic-functions -Wl,--dynamic-list-cpp-new -Wl,--dynamic-list-cpp-typeinfo
+.ENDIF
 
 # linker flags for linking applications
 LINKFLAGSAPPGUI= -Wl,-export-dynamic -Wl,--noinhibit-exec
--- stlport/makefile.mk.old	2009-04-06 16:41:55.000000000 +0000
+++ stlport/makefile.mk	2009-04-06 16:41:55.000000000 +0000
@@ -151,6 +151,10 @@ BUILD_FLAGS=-f vc7.mak EXFLAGS="/EHa /Zc
 CXX+= -Wl,--hash-style=$(WITH_LINKER_HASH_STYLE)
 .ENDIF
 
+.IF "$(HAVE_LD_BSYMBOLIC_FUNCTIONS)"  == "TRUE"
+CXX+= -Wl,-Bsymbolic-functions -Wl,--dynamic-list-cpp-new -Wl,--dynamic-list-cpp-typeinfo
+.ENDIF
+
 .IF "$(COM)"=="C52"
 BUILD_ACTION=make
 .IF "$(CCNUMVER)">="00050008"
