diff -rup /home/rodo/svn/ooo-build-reference/build/oof680-m7/offapi/util/makefile.mk offapi/util/makefile.mk
--- /home/rodo/svn/ooo-build-reference/build/oof680-m7/offapi/util/makefile.mk	2007-01-02 17:57:17.000000000 +0100
+++ offapi/util/makefile.mk	2007-03-14 14:11:17.000000000 +0100
@@ -149,6 +149,8 @@ UNOTYPE_STATISTICS=$(MISC)$/unotype_stat
 
 # --- Targets ------------------------------------------------------
 
+REGMERGEFLAGS=-i
+
 ALLTAR : $(UCR)$/types.db \
        $(OUT)$/ucrdoc$/types_doc.db \
        $(REGISTRYCHECKFLAG) \
@@ -157,7 +159,7 @@ ALLTAR : $(UCR)$/types.db \
 $(UCR)$/types.db : $(UCR)$/offapi.db $(SOLARBINDIR)$/udkapi.rdb
 	-$(RM) $(REGISTRYCHECKFLAG)
 	$(GNUCOPY) -f $(UCR)$/offapi.db $@
-	$(REGMERGE) $@ / $(SOLARBINDIR)$/udkapi.rdb
+	$(REGMERGE) $(REGMERGEFLAGS) $@ / $(SOLARBINDIR)$/udkapi.rdb
 
 $(OUT)$/ucrdoc$/types_doc.db : $(OUT)$/ucrdoc$/offapi_doc.db $(SOLARBINDIR)$/udkapi_doc.rdb
 	-$(RM) $(REGISTRYCHECKFLAG)
diff -rup /home/rodo/svn/ooo-build-reference/build/oof680-m7/solenv/bin/modules/installer/servicesfile.pm solenv/bin/modules/installer/servicesfile.pm
--- /home/rodo/svn/ooo-build-reference/build/oof680-m7/solenv/bin/modules/installer/servicesfile.pm	2006-07-10 20:51:00.000000000 +0200
+++ solenv/bin/modules/installer/servicesfile.pm	2007-03-14 13:21:51.000000000 +0100
@@ -784,6 +784,11 @@ sub create_services_rdb
 		$regcompfileref = installer::scriptitems::get_sourcepath_from_filename_and_includepath(\$searchname, $includepatharrayref, 1);
 		if ( $$regcompfileref eq "" ) { installer::exiter::exit_program("ERROR: Could not find file $searchname for registering uno components!", "create_services_rdb"); }
 
+		if ($installer::globals::isunix) { $searchname = "regmerge"; }
+		else { $searchname = "regmerge.exe"; } 
+	
+		$regmergefileref = installer::scriptitems::get_sourcepath_from_filename_and_includepath(\$searchname, $includepatharrayref, 1);
+		if ( $$regmergefileref eq "" ) { installer::exiter::exit_program("ERROR: Could not find file $searchname for registering uno components!", "create_services_rdb"); }
 		# For Windows the libraries included into the mozruntime.zip have to be added to the path
 		if ($installer::globals::iswin) { add_path_to_pathvariable_directory($filesarrayref, "mozruntime_zip"); }
 		if ($installer::globals::iswin) { add_path_to_pathvariable($filesarrayref, "msvcr70.dll"); }
@@ -826,6 +831,8 @@ sub create_services_rdb
 		}
 		else
 		{
+			system ("$installer::globals::wrapcmd $$regmergefileref -i " . fix_cygwin_path ("${servicesfile}.internal") . " / " . fix_cygwin_path (${servicesfile}));
+			installer::systemactions::rename_one_file ("${servicesfile}.internal", ${servicesfile});
 			$servicesdir = installer::systemactions::rename_directory($servicesdir, $origservicesdir);
 		}
 
diff -rup /home/rodo/svn/ooo-build-reference/build/oof680-m7/solenv/inc/target.mk solenv/inc/target.mk
--- /home/rodo/svn/ooo-build-reference/build/oof680-m7/solenv/inc/target.mk	2007-01-19 13:53:46.000000000 +0100
+++ solenv/inc/target.mk	2007-03-14 11:54:31.000000000 +0100
@@ -1438,31 +1438,31 @@ makedoc:
 $(LOCALDBTARGET) : $(URDFILES) $(DEPIDLFILES)
 	$(IDLC) @$(mktmp $(IDLCFLAGS) $(UNOIDLDEFS) $(UNOIDLINCEXTRA) $(UNOIDLINC) -O$(OUT)$/ucr$/$(IDLPACKAGE) $(all_outdated_idl))
 	-$(RM) $@
-	$(REGMERGE) $@ UCR @$(mktmp $(URDFILES))
+	$(REGMERGE) $(REGMERGEFLAGS) $@ UCR @$(mktmp $(URDFILES))
 .ENDIF
 
 .IF "$(LOCALDOCDBTARGET)"!=""
 $(LOCALDOCDBTARGET) : $(URDDOCFILES) $(DEPIDLFILES)
 	$(IDLC) @$(mktmp $(IDLCFLAGS) $(UNOIDLDEFS) $(UNOIDLINCEXTRA) $(UNOIDLINC) -C -O$(OUT)$/ucrdoc$/$(IDLPACKAGE) $(all_outdated_idl))
 	-$(RM) $@
-	$(REGMERGE) $@ UCR @$(mktmp $(URDDOCFILES))
+	$(REGMERGE) $(REGMERGEFLAGS) $@ UCR @$(mktmp $(URDDOCFILES))
 .ENDIF
 
 .IF "$(UNOIDLDBTARGET)"!=""
 $(UNOIDLDBTARGET) : $(UNOIDLDBFILES) $(UNOIDLDBREGS)
 	-$(RM) $@
-	$(REGMERGE) $@ / @$(mktmp $(UNOIDLDBFILES) $(UNOIDLDBREGS))
+	$(REGMERGE) $(REGMERGEFLAGS) $@ / @$(mktmp $(UNOIDLDBFILES) $(UNOIDLDBREGS))
 .IF "$(LOCALREGDB)"!=""
-	$(REGMERGE) $(LOCALREGDB) / $@
+	$(REGMERGE) $(REGMERGEFLAGS) $(LOCALREGDB) / $@
 .ENDIF
 .ENDIF			# "$(UNOIDLDBTARGET)"!=""
 
 .IF "$(UNOIDLDBDOCTARGET)"!=""
 $(UNOIDLDBDOCTARGET) : $(UNOIDLDBDOCFILES) $(UNOIDLDBDOCREGS)
 	-$(RM) $@
-	$(REGMERGE) $@ / @$(mktmp $(UNOIDLDBDOCFILES) $(UNOIDLDBDOCREGS))
+	$(REGMERGE) $(REGMERGEFLAGS) $@ / @$(mktmp $(UNOIDLDBDOCFILES) $(UNOIDLDBDOCREGS))
 .IF "$(LOCALREGDB)"!=""
-	$(REGMERGE) $(LOCALREGDB) / $@
+	$(REGMERGE) $(REGMERGEFLAGS) $(LOCALREGDB) / $@
 .ENDIF
 .ENDIF			# "$(UNOIDLDBDOCTARGET)"!=""
 
diff -rup /home/rodo/svn/ooo-build-reference/build/oof680-m7/instsetoo_native/util/makefile.mk instsetoo_native/util/makefile.mk
--- /home/rodo/svn/ooo-build-reference/build/oof680-m7/instsetoo_native/util/makefile.mk	2007-01-16 09:12:02.000000000 +0100
+++ instsetoo_native/util/makefile.mk	2007-03-14 12:08:49.000000000 +0100
#@@ -374,6 +374,8 @@ $(MISC)$/ure$/services.rdb .ERRREMOVE:
#     + $(REGCOMP) -register -r $@ -c \
#         $(MY_Q)vnd.sun.star.expand:$$URE_INTERNAL_LIB_DIR/uuresolver.uno$(DLLPOST)$(MY_Q) \
#         -env:URE_INTERNAL_LIB_DIR=$(MY_URL)
#+	+ $(REGMERGE) -i $@-internal / $@
#+	+ $(MV) $@-internal $@
# 
# .IF "$(LOCALPYFILES)"!=""
# .IF "$(PKGFORMAT)"==""
--- oovbaapi/util/makefile.mk	2007-07-05 12:07:16.000000000 +0100
+++ oovbaapi-patched/util/makefile.mk	2007-07-05 12:43:12.000000000 +0100
@@ -57,5 +57,5 @@ UNOIDLDBFILES= \
 	$(UCR)$/constants.db
 
 # --- Targets ------------------------------------------------------
-
+REGMERGEFLAGS=-i
 .INCLUDE :  target.mk
diff -rup udkapi-orig/util/makefile.mk udkapi/util/makefile.mk
--- udkapi-orig/util/makefile.mk        2008-09-03 13:23:25.000000000 +0200
+++ udkapi/util/makefile.mk     2008-09-03 13:21:01.000000000 +0200
@@ -72,6 +72,8 @@ UNOIDLDBFILES= \
        $(UCR)$/csstask.db \
     $(UCR)$/cssuri.db
 
+REGMERGEFLAGS=-i
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
