--- svtools/source/contnr/templwin.cxx.bak	2005-08-12 12:00:28.000000000 +0530
+++ svtools/source/contnr/templwin.cxx	2005-08-12 15:40:53.766509184 +0530
@@ -1650,7 +1650,7 @@
 void SvtTemplateWindow::ReadViewSettings()
 {
 	// defaults
-	sal_Int32 nSelectedGroup	=	ICON_POS_TEMPLATES;
+	sal_Int32 nSelectedGroup	=	ICON_POS_NEWDOC;
 	sal_Int32 nSelectedView		=	TI_DOCTEMPLATE_DOCINFO;
 	double nSplitRatio			=	0.5;
 	::rtl::OUString sLastFolder;
