--- svtools/source/contnr/templwin.cxx.old	2009-04-06 16:41:53.000000000 +0000
+++ svtools/source/contnr/templwin.cxx	2009-04-06 16:41:53.000000000 +0000
@@ -108,6 +108,7 @@ using namespace ::com::sun::star::view;
 using namespace svtools;
 
 extern ::rtl::OUString CreateExactSizeText_Impl( sal_Int64 nSize ); // fileview.cxx
+#define aSeparatorStr   "----------------------------------"
 
 #define SPLITSET_ID			0
 #define COLSET_ID			1
@@ -606,7 +607,7 @@ SvtFileViewWindow_Impl::SvtFileViewWindo
 	Window( pParent, WB_DIALOGCONTROL | WB_TABSTOP | WB_BORDER | WB_3DLOOK ),
 
 	rParent				( *pParent ),
-	aFileView			( this, SvtResId( CTRL_FILEVIEW ), FILEVIEW_SHOW_TITLE ),
+	aFileView			( this, SvtResId( CTRL_FILEVIEW ), FILEVIEW_SHOW_NONE ),
 	bIsTemplateFolder	( sal_False )
 
 {
@@ -662,9 +663,14 @@ Sequence< ::rtl::OUString > SvtFileViewW
 	for ( i = 0; i < nCount; ++i )
 	{
 		GetMenuEntry_Impl( aDynamicMenuEntries[i], aTitle, aURL, aTargetFrame, aImageURL );
-
-        if ( aURL == sSeparator || aURL == sSlotURL )
+		if( aURL == sSlotURL )
 			continue;
+		if( aURL == sSeparator )
+		{
+			String aSeparator( ASCII_STR( aSeparatorStr ) );
+			::rtl::OUString* pSeparator = new ::rtl::OUString( aSeparator );
+			aNewDocs.Insert( pSeparator, LIST_APPEND );
+		}
 		else
 		{
 			// title
--- svtools/source/contnr/fileview.cxx.old	2009-04-02 11:06:15.000000000 +0000
+++ svtools/source/contnr/fileview.cxx	2009-04-06 16:41:53.000000000 +0000
@@ -103,6 +103,8 @@ using ::rtl::OUString;
 
 DECLARE_LIST( StringList_Impl, OUString* )
 
+#define aSeparatorStr  "----------------------------------"
+
 #define ROW_HEIGHT	17	// the height of a row has to be a little higher than the bitmap
 #define QUICK_SEARCH_TIMEOUT    1500    // time in mSec before the quicksearch string will be reseted
 
@@ -750,6 +752,7 @@ ViewTabListBox_Impl::ViewTabListBox_Impl
     mbEnableDelete  	( sal_True )
 
 {
+	sal_Bool bViewHeader = true;
 	Size aBoxSize = pParentWin->GetSizePixel();
 	mpHeaderBar = new HeaderBar( pParentWin, WB_BUTTONSTYLE | WB_BOTTOMBORDER );
 	mpHeaderBar->SetPosSizePixel( Point( 0, 0 ), mpHeaderBar->CalcWindowSizePixel() );
@@ -762,6 +765,8 @@ ViewTabListBox_Impl::ViewTabListBox_Impl
 		mpHeaderBar->InsertItem( COLUMN_SIZE, String( SvtResId( STR_SVT_FILEVIEW_COLUMN_SIZE ) ), 80, nBits );
 		mpHeaderBar->InsertItem( COLUMN_DATE, String( SvtResId( STR_SVT_FILEVIEW_COLUMN_DATE ) ), 500, nBits );
 	}
+	if( ( nFlags & FILEVIEW_SHOW_NONE ) == FILEVIEW_SHOW_NONE )
+		bViewHeader = false;
 	else
 		mpHeaderBar->InsertItem( COLUMN_TITLE, String( SvtResId( STR_SVT_FILEVIEW_COLUMN_TITLE ) ), 600, nBits );
 
@@ -773,7 +778,8 @@ ViewTabListBox_Impl::ViewTabListBox_Impl
 	SetEntryHeight( ROW_HEIGHT );
 
 	Show();
-	mpHeaderBar->Show();
+	if( bViewHeader )
+		mpHeaderBar->Show();
 
     maResetQuickSearch.SetTimeout( QUICK_SEARCH_TIMEOUT );
     maResetQuickSearch.SetTimeoutHdl( LINK( this, ViewTabListBox_Impl, ResetQuickSearch_Impl ) );
@@ -1133,6 +1139,7 @@ SvtFileView::SvtFileView( Window* pParen
 
 	mpImp = new SvtFileView_Impl( this, xCmdEnv, nFlags, bOnlyFolder );
     mpImp->mpView->ForbidEmptyText();
+	SetSortColumn( true );
 
     long pTabs[] = { 5, 20, 180, 320, 400, 600 };
 	mpImp->mpView->SetTabs( &pTabs[0], MAP_PIXEL );
@@ -1155,6 +1162,7 @@ SvtFileView::SvtFileView( Window* pParen
 	Reference < XCommandEnvironment > xCmdEnv = new ::ucbhelper::CommandEnvironment( xInteractionHandler, Reference< XProgressHandler >() );
 	mpImp = new SvtFileView_Impl( this, xCmdEnv, nFlags,
                                   ( nFlags & FILEVIEW_ONLYFOLDER ) == FILEVIEW_ONLYFOLDER );
+	SetSortColumn( true );
 
     if ( ( nFlags & FILEVIEW_SHOW_ALL ) == FILEVIEW_SHOW_ALL )
 	{
@@ -1168,7 +1176,8 @@ SvtFileView::SvtFileView( Window* pParen
 		long pTabs[] = { 2, 20, 600 };
 		mpImp->mpView->SetTabs( &pTabs[0], MAP_PIXEL );
 	}
-
+	if ( ( nFlags & FILEVIEW_SHOW_NONE ) == FILEVIEW_SHOW_NONE )
+        SetSortColumn( false );
 	if ( ( nFlags & FILEVIEW_MULTISELECTION ) == FILEVIEW_MULTISELECTION )
 		mpImp->mpView->SetSelectionMode( MULTIPLE_SELECTION );
 
@@ -1426,7 +1435,8 @@ sal_Bool SvtFileView::Initialize( const
 
     mpImp->Clear();
     mpImp->CreateVector_Impl( aContents );
-    mpImp->SortFolderContent_Impl();
+	if( GetSortColumn() )
+	    mpImp->SortFolderContent_Impl();
 
     mpImp->OpenFolder_Impl();
 
@@ -2330,9 +2340,11 @@ void SvtFileView_Impl::CreateVector_Impl
         pEntry->maDisplayText = aDisplayText;
 
         // detect the image
-        INetURLObject aObj( pEntry->maImageURL.getLength() ? pEntry->maImageURL : pEntry->maTargetURL );
-		pEntry->maImage = SvFileInformationManager::GetImage( aObj, FALSE, isHighContrast( mpView ) );
-
+		if( aValue != rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(aSeparatorStr) ) )
+		{
+        	INetURLObject aObj( pEntry->maImageURL.getLength() ? pEntry->maImageURL : pEntry->maTargetURL );
+			pEntry->maImage = SvFileInformationManager::GetImage( aObj, FALSE, isHighContrast( mpView ) );
+		}
         maContent.push_back( pEntry );
     }
 }
--- svtools/inc/fileview.hxx.old	2009-04-02 11:06:22.000000000 +0000
+++ svtools/inc/fileview.hxx	2009-04-06 16:41:53.000000000 +0000
@@ -51,6 +51,7 @@
 #define FILEVIEW_SHOW_SIZE			0x0020
 #define FILEVIEW_SHOW_DATE			0x0040
 #define FILEVIEW_SHOW_ALL			0x0070
+#define FILEVIEW_SHOW_NONE			0x00A0
 
 class ViewTabListBox_Impl;
 class SvtFileView_Impl;
@@ -84,6 +85,7 @@ class SVT_DLLPUBLIC SvtFileView : public
 {
 private:
     SvtFileView_Impl*       mpImp;
+	sal_Bool 				bSortColumn;
 	
 	::com::sun::star::uno::Sequence< ::rtl::OUString > mpBlackList;
 
@@ -112,6 +114,8 @@ public:
 	void					SetSizePixel( const Size& rNewSize );
     using Window::SetPosSizePixel;
 	virtual void			SetPosSizePixel( const Point& rNewPos, const Size& rNewSize );
+	void					SetSortColumn( sal_Bool bValue ) { bSortColumn = bValue; }
+	sal_Bool				GetSortColumn() { return bSortColumn; }
 
     /** initialize the view with the content of a folder given by URL, and aply an immediate filter
 
