--- svtools/prj/build.lst.old	2008-03-12 14:08:42.000000000 +0100
+++ svtools/prj/build.lst	2008-09-25 12:05:30.000000000 +0200
@@ -1,4 +1,4 @@
-st	svtools	:	offuh toolkit ucbhelper unotools JPEG:jpeg cppu cppuhelper comphelper sal jvmfwk NULL
+st	svtools	:	offuh toolkit ucbhelper unotools JPEG:jpeg cppu cppuhelper comphelper sal sot jvmfwk NULL
 st	svtools						usr1	-	all	st_mkout NULL
 st	svtools\inc					nmake	-	all	st_inc NULL
 st	svtools\inc\sane				get	-	all	st_incsa NULL
