--- svx/source/dialog/svuidlg.src.orig	2009-01-16 21:07:26.000000000 +0100
+++ svx/source/dialog/svuidlg.src	2009-01-16 23:47:44.000000000 +0100
@@ -156,37 +156,37 @@
 	};
 	CancelButton 1
 	{
-		Pos = MAP_APPFONT ( 275 , 6 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Pos = MAP_APPFONT ( 270 , 6 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
 		Text [ en-US ] = "~Close" ;
 	};
 	HelpButton 1
 	{
-		Pos = MAP_APPFONT ( 275 , 26 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Pos = MAP_APPFONT ( 270 , 26 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
 	};
 	PushButton PB_UPDATE_NOW
 	{
-		Pos = MAP_APPFONT ( 275 , 43 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Pos = MAP_APPFONT ( 270 , 43 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
 		Text [ en-US ] = "~Update" ;
 	};
 	PushButton PB_OPEN_SOURCE
 	{
-		Pos = MAP_APPFONT ( 275 , 60 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Pos = MAP_APPFONT ( 270 , 60 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
 		Text [ en-US ] = "~Open" ;
 	};
 	PushButton PB_CHANGE_SOURCE
 	{
-		Pos = MAP_APPFONT ( 275 , 77 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Pos = MAP_APPFONT ( 270 , 77 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
 		Text [ en-US ] = "~Modify..." ;
 	};
 	PushButton PB_BREAK_LINK
 	{
-		Pos = MAP_APPFONT ( 275 , 94 ) ;
-		Size = MAP_APPFONT ( 50 , 14 ) ;
+		Pos = MAP_APPFONT ( 270 , 94 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
 		Text [ en-US ] = "~Break Link" ;
 	};
 	FixedText FT_FILES2
--- svx/inc/optgenrl.hrc.orig	2009-01-16 20:33:08.000000000 +0100
+++ svx/inc/optgenrl.hrc	2009-01-16 20:33:24.000000000 +0100
@@ -74,7 +74,7 @@
 #define LINEH		12
 #define FLINEH		8
 #define	LEFT		12
-#define MID         96
+#define MID         100
 #define RIGHT		248
 #define TOP			10
 #define LINE(y)		(14+y*15)
--- svx/source/dialog/zoom.src.orig	2009-01-16 21:32:37.000000000 +0100
+++ svx/source/dialog/zoom.src	2009-01-16 23:53:36.000000000 +0100
@@ -55,7 +55,7 @@
     RadioButton BTN_WHOLE_PAGE
 	{
         Pos = MAP_APPFONT ( 12 , 27 ) ;
-        Size = MAP_APPFONT ( 91 , 10 ) ;
+        Size = MAP_APPFONT ( 104 , 10 ) ;
         Text [ en-US ] = "~Fit width and height" ;
 	};
 	RadioButton BTN_PAGE_WIDTH
--- svx/source/unodialogs/textconversiondlgs/chinese_translationdialog.src.orig	2009-01-16 22:12:48.000000000 +0100
+++ svx/source/unodialogs/textconversiondlgs/chinese_translationdialog.src	2009-01-16 22:16:07.000000000 +0100
@@ -39,7 +39,7 @@
 
 #define FULL_WIDTH (220)
 #define FULL_HEIGHT (120)
-#define EDITBUTTON_WIDTH (70)
+#define EDITBUTTON_WIDTH (75)
 
 #define COL_1 (RSC_SP_DLG_INNERBORDER_LEFT)
 #define COL_2 (COL_1+RSC_SP_FLGR_SPACE_X)
--- svx/source/dialog/hangulhanjadlg.src.orig	2009-01-16 22:41:29.000000000 +0100
+++ svx/source/dialog/hangulhanjadlg.src	2009-01-16 22:43:22.000000000 +0100
@@ -81,7 +81,7 @@
 	RadioButton RB_HANJA_HANGUL_BRACKETED
 	{
 		Pos = MAP_APPFONT( 109, 81 );
-		Size = MAP_APPFONT( 55, 8 );
+		Size = MAP_APPFONT( 56, 8 );
 
 		Text [ en-US ] = "Hanja (Han~gul)";
 	};
@@ -89,7 +89,7 @@
 	RadioButton RB_HANGUL_HANJA_BRACKETED
 	{
 		Pos = MAP_APPFONT( 167, 81 );
-		Size = MAP_APPFONT( 55, 8 );
+		Size = MAP_APPFONT( 56, 8 );
 
 		Text [ en-US ] = "Hang~ul (Hanja)";
 	};
