--- svx/source/svdraw/svdoashp.orig	2008-07-22 23:22:38.000000000 +0200
+++ svx/source/svdraw/svdoashp.cxx	2008-07-22 23:25:49.000000000 +0200
@@ -1660,8 +1660,8 @@
 	rInfo.bMirror90Allowed  =TRUE;
 	rInfo.bTransparenceAllowed = FALSE;
 	rInfo.bGradientAllowed = FALSE;
-	rInfo.bShearAllowed     =FALSE;
-	rInfo.bEdgeRadiusAllowed=FALSE;
+	rInfo.bShearAllowed     =TRUE;
+	rInfo.bEdgeRadiusAllowed=TRUE;
 	rInfo.bNoContortion     =TRUE;
 
 	// #i37011#
@@ -1698,6 +1698,11 @@
 				{
 					rInfo.bCanConvToContour = aInfo.bCanConvToContour;
 				}
+
+                if( !rInfo.bShearAllowed )
+                    rInfo.bShearAllowed=FALSE;
+                if( !aInfo.bEdgeRadiusAllowed )
+                    rInfo.bEdgeRadiusAllowed=FALSE;
 			}
 		}
 	}
