diff -rup svx/source/editeng-orig/impedit3.cxx svx/source/editeng/impedit3.cxx
--- svx/source/editeng-orig/impedit3.cxx	2009-06-05 10:19:00.000000000 +0200
+++ svx/source/editeng/impedit3.cxx	2009-06-05 10:45:28.000000000 +0200
@@ -940,7 +940,7 @@ sal_Bool ImpEditEngine::CreateLines( USH
 //						nCurPos -= rLRItem.GetTxtLeft();	// Tabs relativ zu LI
 						// Skalierung rausrechnen
 						if ( aStatus.DoStretch() && ( nStretchX != 100 ) )
-							nCurPos = nCurPos*100/nStretchX;
+						    nCurPos = nCurPos*100/std::max((sal_Int32)nStretchX, 1);
 
                         short nAllSpaceBeforeText = static_cast< short >(rLRItem.GetTxtLeft()/* + rLRItem.GetTxtLeft()*/ + nSpaceBeforeAndMinLabelWidth);
                         aCurrentTab.aTabStop = pNode->GetContentAttribs().FindTabStop( nCurPos - nAllSpaceBeforeText /*rLRItem.GetTxtLeft()*/, aEditDoc.GetDefTab() );
