Index: svx/source/msfilter/escherex.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/escherex.cxx,v
retrieving revision 1.73
diff -u -r1.73 escherex.cxx
--- svx/source/msfilter/escherex.cxx	31 Jul 2007 17:31:58 -0000	1.73
+++ svx/source/msfilter/escherex.cxx	11 Aug 2008 12:26:47 -0000
@@ -853,6 +853,29 @@
 
 	if ( nTextId )
 		AddOpt( ESCHER_Prop_lTxid, nTextId );
+
+    // n#404221: In case of rotation we need to write the txtflTextFlow 
+    // attribute too.
+	if (bIsTextFrame) {
+	    sal_uInt16 nAngle = EscherPropertyValueHelper::GetPropertyValue( 
+		aAny, 
+		rXPropSet,
+		String( RTL_CONSTASCII_USTRINGPARAM( "RotateAngle" ) ), 
+		sal_True )
+		? (sal_uInt16)( ( *((sal_Int32*)aAny.getValue() ) ) + 5 ) / 10 : 0;
+	    if (nAngle==900) {
+		AddOpt( ESCHER_Prop_txflTextFlow, 1 );
+		bSuppressRotation=true;
+	    }
+	    if (nAngle==1800) {
+		AddOpt( ESCHER_Prop_txflTextFlow, 2 );
+		bSuppressRotation=true;
+	    }
+	    if (nAngle==2700) {
+		AddOpt( ESCHER_Prop_txflTextFlow, 3 );
+		bSuppressRotation=true;
+	    }
+	}
 }
 
 static sal_Bool GetLineArrow( const sal_Bool bLineStart,
