diff -u -r /opt/OpenOffice/svn-HEAD/build/ooo310-m4/svx/source/editeng/editdoc.cxx svx/source/editeng/editdoc.cxx
--- svx/source/editeng/editdoc.cxx
+++ svx/source/editeng/editdoc.cxx
@@ -1235,6 +1235,7 @@
 	nDefTab = DEFTAB;
 	bIsVertical = FALSE;
 	bIsFixedCellHeight = FALSE;
+	nLastPos = 0;
 
 	// Don't create a empty node, Clear() will be called in EditEngine-CTOR
 
@@ -1966,6 +1967,35 @@
 	}
 }
 
+USHORT EditDoc::GetPos( ContentNode* pNode ) const
+{
+  USHORT nCount = ContentList::Count();
+
+  // Through certain filter code-paths we do a lot of appends, which in
+  // turn call GetPos - creating some N^2 nightmares. If we have a
+  // non-trivially large list, do a few checks from the end first.
+  if( nLastPos > 16 )
+    {
+      USHORT nEnd;
+      if (nLastPos > nCount - 2)
+	nEnd = nCount;
+      else
+	nEnd = nLastPos + 2;
+
+      for( USHORT nIdx = nLastPos - 2; nIdx < nEnd; nIdx++ )
+	{
+	  if( pData[ nIdx ] == pNode )
+	    {
+	      ((EditDoc *)this)->nLastPos = nIdx;
+	      return nIdx;
+	    }
+	}
+    }
+
+  ((EditDoc *)this)->nLastPos = ContentList::GetPos(pNode);
+  return nLastPos;
+}
+
 
 // -------------------------------------------------------------------------
 // class EditCharAttribList


diff -u -r /opt/OpenOffice/svn-HEAD/build/ooo310-m4/svx/source/editeng/editdoc.hxx svx/source/editeng/editdoc.hxx
--- svx/source/editeng/editdoc.hxx
+++ svx/source/editeng/editdoc.hxx
@@ -698,6 +698,7 @@
 	BOOL			bOwnerOfPool;
 	BOOL			bModified;
 
+	USHORT			nLastPos;
 protected:
 	void			ImplDestroyContents();
 
@@ -753,7 +754,7 @@
 	BOOL			RemoveAttribs( ContentNode* pNode, USHORT nStart, USHORT nEnd, EditCharAttrib*& rpStarting, EditCharAttrib*& rpEnding, USHORT nWhich = 0 );
 	void			FindAttribs( ContentNode* pNode, USHORT nStartPos, USHORT nEndPos, SfxItemSet& rCurSet );
 
-	USHORT			GetPos( ContentNode* pNode ) const { return ContentList::GetPos(pNode); }
+        USHORT			GetPos( ContentNode* pNode ) const;
 	ContentNode*	SaveGetObject( USHORT nPos ) const { return ( nPos < Count() ) ? GetObject( nPos ) : 0; }
 
 	static XubString	GetSepStr( LineEnd eEnd );
