diff -u -r svx/inc/svx/sdtfchim.hxx svx/inc/svx/sdtfchim.hxx
--- svx/inc/svx/sdtfchim.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/sdtfchim.hxx	2009-05-18 07:49:35.000000000 +0200
@@ -32,28 +32,28 @@
 
 #include <svx/svddef.hxx>
 
-class SdrTextFixedCellHeightItem : public SfxBoolItem
+class SVX_DLLPUBLIC SdrTextFixedCellHeightItem : public SfxBoolItem
 {
 public:
 
 	TYPEINFO();
-	SVX_DLLPUBLIC SdrTextFixedCellHeightItem( BOOL bUseFixedCellHeight = FALSE );
-	SdrTextFixedCellHeightItem( SvStream & rStream, sal_uInt16 nVersion );
+	SdrTextFixedCellHeightItem( BOOL bUseFixedCellHeight = FALSE );
+	SVX_DLLPRIVATE SdrTextFixedCellHeightItem( SvStream & rStream, sal_uInt16 nVersion );
 
-	virtual SfxItemPresentation GetPresentation(SfxItemPresentation ePresentation,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation(SfxItemPresentation ePresentation,
 									SfxMapUnit eCoreMetric, SfxMapUnit ePresentationMetric,
                                         String &rText, const IntlWrapper * = 0) const;
 
-	virtual SfxPoolItem*		Create( SvStream&, sal_uInt16 nItem ) const;
-	virtual SvStream&			Store( SvStream&, sal_uInt16 nVersion ) const;
-	virtual SfxPoolItem*		Clone( SfxItemPool* pPool = NULL ) const;
-	virtual	sal_uInt16			GetVersion( sal_uInt16 nFileFormatVersion ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*		Create( SvStream&, sal_uInt16 nItem ) const;
+	SVX_DLLPRIVATE virtual SvStream&			Store( SvStream&, sal_uInt16 nVersion ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*		Clone( SfxItemPool* pPool = NULL ) const;
+	SVX_DLLPRIVATE virtual	sal_uInt16			GetVersion( sal_uInt16 nFileFormatVersion ) const;
 
-	virtual	sal_Bool			QueryValue( com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 ) const;
-	virtual	sal_Bool			PutValue( const com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 );
+	SVX_DLLPRIVATE virtual	sal_Bool			QueryValue( com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 ) const;
+	SVX_DLLPRIVATE virtual	sal_Bool			PutValue( const com::sun::star::uno::Any& rVal, BYTE nMemberId = 0 );
 
 #ifdef SDR_ISPOOLABLE
-	virtual int IsPoolable() const;
+	SVX_DLLPRIVATE virtual int IsPoolable() const;
 #endif
 };
 
diff -u -r svx/inc/svx/xflbckit.hxx svx/inc/svx/xflbckit.hxx
--- svx/inc/svx/xflbckit.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xflbckit.hxx	2009-05-18 07:14:52.000000000 +0200
@@ -41,19 +41,19 @@
 |*
 \************************************************************************/
 
-class XFillBackgroundItem : public SfxBoolItem
+class SVX_DLLPUBLIC XFillBackgroundItem : public SfxBoolItem
 {
 public:
 							TYPEINFO();
 
-							SVX_DLLPUBLIC XFillBackgroundItem( BOOL bFill = FALSE );
-							XFillBackgroundItem( SvStream& rIn );
+							XFillBackgroundItem( BOOL bFill = FALSE );
+							SVX_DLLPRIVATE XFillBackgroundItem( SvStream& rIn );
 
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = NULL ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = NULL ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
diff -u -r svx/inc/svx/xflbmpit.hxx svx/inc/svx/xflbmpit.hxx
--- svx/inc/svx/xflbmpit.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xflbmpit.hxx	2009-05-18 07:22:26.000000000 +0200
@@ -39,23 +39,23 @@
 |*
 \************************************************************************/
 
-class XFillBmpPosItem : public SfxEnumItem
+class SVX_DLLPUBLIC XFillBmpPosItem : public SfxEnumItem
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpPosItem( RECT_POINT eRP = RP_MM );
-							XFillBmpPosItem( SvStream& rIn );
+							XFillBmpPosItem( RECT_POINT eRP = RP_MM );
+							SVX_DLLPRIVATE XFillBmpPosItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
 
-	virtual USHORT			GetValueCount() const;
-	RECT_POINT 				GetValue() const { return (RECT_POINT) SfxEnumItem::GetValue(); }
+	SVX_DLLPRIVATE virtual USHORT			GetValueCount() const;
+	SVX_DLLPRIVATE RECT_POINT 				GetValue() const { return (RECT_POINT) SfxEnumItem::GetValue(); }
 };
 
 
diff -u -r svx/inc/svx/xflbmsli.hxx svx/inc/svx/xflbmsli.hxx
--- svx/inc/svx/xflbmsli.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xflbmsli.hxx	2009-05-18 07:09:51.000000000 +0200
@@ -39,17 +39,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpSizeLogItem : public SfxBoolItem
+class SVX_DLLPUBLIC XFillBmpSizeLogItem : public SfxBoolItem
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpSizeLogItem( BOOL bLog = TRUE );
-							XFillBmpSizeLogItem( SvStream& rIn );
+							XFillBmpSizeLogItem( BOOL bLog = TRUE );
+							SVX_DLLPRIVATE XFillBmpSizeLogItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
diff -u -r svx/inc/svx/xflbmsxy.hxx svx/inc/svx/xflbmsxy.hxx
--- svx/inc/svx/xflbmsxy.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xflbmsxy.hxx	2009-05-18 07:30:46.000000000 +0200
@@ -39,17 +39,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpSizeXItem : public SfxMetricItem
+class SVX_DLLPUBLIC XFillBmpSizeXItem : public SfxMetricItem
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpSizeXItem( long nSizeX = 0 );
-							XFillBmpSizeXItem( SvStream& rIn );
+							XFillBmpSizeXItem( long nSizeX = 0 );
+							SVX_DLLPRIVATE XFillBmpSizeXItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -57,7 +57,7 @@
 	// Dieses Item nimmt sowohl metrische als auch
 	// Prozentwerte ( sind negativ ) auf; das muessen
 	// wir hier beruecksichtigen
-	virtual FASTBOOL		HasMetrics() const;
+	SVX_DLLPRIVATE virtual FASTBOOL		HasMetrics() const;
 };
 
 /*************************************************************************
@@ -66,17 +66,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpSizeYItem : public SfxMetricItem
+class SVX_DLLPUBLIC XFillBmpSizeYItem : public SfxMetricItem
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpSizeYItem( long nSizeY = 0 );
-							XFillBmpSizeYItem( SvStream& rIn );
+							XFillBmpSizeYItem( long nSizeY = 0 );
+							SVX_DLLPRIVATE XFillBmpSizeYItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -84,7 +84,7 @@
 	// Dieses Item nimmt sowohl metrische als auch
 	// Prozentwerte ( sind negativ ) auf; das muessen
 	// wir hier beruecksichtigen
-	virtual FASTBOOL		HasMetrics() const;
+	SVX_DLLPRIVATE virtual FASTBOOL		HasMetrics() const;
 };
 
 #endif
diff -u -r svx/inc/svx/xflboxy.hxx svx/inc/svx/xflboxy.hxx
--- svx/inc/svx/xflboxy.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xflboxy.hxx	2009-05-18 07:33:05.000000000 +0200
@@ -39,17 +39,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpPosOffsetXItem : public SfxUInt16Item
+class SVX_DLLPUBLIC XFillBmpPosOffsetXItem : public SfxUInt16Item
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpPosOffsetXItem( USHORT nOffPosX = 0 );
-							XFillBmpPosOffsetXItem( SvStream& rIn );
+							XFillBmpPosOffsetXItem( USHORT nOffPosX = 0 );
+							SVX_DLLPRIVATE XFillBmpPosOffsetXItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -62,17 +62,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpPosOffsetYItem : public SfxUInt16Item
+class SVX_DLLPUBLIC XFillBmpPosOffsetYItem : public SfxUInt16Item
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpPosOffsetYItem( USHORT nOffPosY = 0 );
-							XFillBmpPosOffsetYItem( SvStream& rIn );
+							XFillBmpPosOffsetYItem( USHORT nOffPosY = 0 );
+							SVX_DLLPRIVATE XFillBmpPosOffsetYItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
diff -u -r svx/inc/svx/xflbtoxy.hxx svx/inc/svx/xflbtoxy.hxx
--- svx/inc/svx/xflbtoxy.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xflbtoxy.hxx	2009-05-18 07:36:45.000000000 +0200
@@ -39,17 +39,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpTileOffsetXItem : public SfxUInt16Item
+class SVX_DLLPUBLIC XFillBmpTileOffsetXItem : public SfxUInt16Item
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpTileOffsetXItem( USHORT nOffX = 0 );
-							XFillBmpTileOffsetXItem( SvStream& rIn );
+							XFillBmpTileOffsetXItem( USHORT nOffX = 0 );
+							SVX_DLLPRIVATE XFillBmpTileOffsetXItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
@@ -62,17 +62,17 @@
 |*
 \************************************************************************/
 
-class XFillBmpTileOffsetYItem : public SfxUInt16Item
+class SVX_DLLPUBLIC XFillBmpTileOffsetYItem : public SfxUInt16Item
 {
 public:
 							TYPEINFO();
-							SVX_DLLPUBLIC XFillBmpTileOffsetYItem( USHORT nOffX = 0 );
-							XFillBmpTileOffsetYItem( SvStream& rIn );
+							XFillBmpTileOffsetYItem( USHORT nOffX = 0 );
+							SVX_DLLPRIVATE XFillBmpTileOffsetYItem( SvStream& rIn );
 
-	virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
-	virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone( SfxItemPool* pPool = 0 ) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create( SvStream& rIn, USHORT nVer ) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
diff -u -r svx/inc/svx/xsflclit.hxx svx/inc/svx/xsflclit.hxx
--- svx/inc/svx/xsflclit.hxx	2009-05-18 07:01:34.000000000 +0200
+++ svx/inc/svx/xsflclit.hxx	2009-05-18 07:47:38.000000000 +0200
@@ -36,24 +36,24 @@
 //------------------------------
 // class XSecondaryFillColorItem
 //------------------------------
-class XSecondaryFillColorItem : public XColorItem
+class SVX_DLLPUBLIC XSecondaryFillColorItem : public XColorItem
 {
 public:
 			TYPEINFO();
-			XSecondaryFillColorItem() {}
-			XSecondaryFillColorItem(long nIndex, const Color& rTheColor);
-			SVX_DLLPUBLIC XSecondaryFillColorItem(const String& rName, const Color& rTheColor);
-			XSecondaryFillColorItem(SvStream& rIn);
+			SVX_DLLPRIVATE XSecondaryFillColorItem() {}
+			SVX_DLLPRIVATE XSecondaryFillColorItem(long nIndex, const Color& rTheColor);
+			XSecondaryFillColorItem(const String& rName, const Color& rTheColor);
+			SVX_DLLPRIVATE XSecondaryFillColorItem(SvStream& rIn);
 
-	virtual SfxPoolItem*    Clone(SfxItemPool* pPool = 0) const;
-	virtual SfxPoolItem*    Create(SvStream& rIn, USHORT nVer) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Clone(SfxItemPool* pPool = 0) const;
+	SVX_DLLPRIVATE virtual SfxPoolItem*    Create(SvStream& rIn, USHORT nVer) const;
 
-	virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
+	SVX_DLLPRIVATE virtual SfxItemPresentation GetPresentation( SfxItemPresentation ePres,
 									SfxMapUnit eCoreMetric,
 									SfxMapUnit ePresMetric,
                                     String &rText, const IntlWrapper * = 0 ) const;
 
-	virtual USHORT GetVersion( USHORT nFileFormatVersion ) const;
+	SVX_DLLPRIVATE virtual USHORT GetVersion( USHORT nFileFormatVersion ) const;
 
 };
 
