diff -uNr sw/inc/cmdid.h sw/inc/cmdid.h
--- sw/inc/cmdid.h	2008-07-22 23:00:50.000000000 +0800
+++ sw/inc/cmdid.h	2008-09-04 23:01:48.000000000 +0800
@@ -263,6 +263,9 @@
 
 #define FN_EDIT_HYPERLINK     		 (FN_EDIT2 + 35) 	/* edit hyperlink */
 #define FN_NUM_CONTINUE              (FN_EDIT2 + 36)    /* continue previous numbering */
+#define FN_REDLINE_ACCEPT_CURRENT                (FN_EDIT2 + 37) /* accept the current selected redline */
+#define FN_REDLINE_REJECT_CURRENT                (FN_EDIT2 + 38) /* reject the current selected redline */ 
+
 /*--------------------------------------------------------------------
 	Bereich: Bearbeiten
  --------------------------------------------------------------------*/
diff -uNr sw/sdi/swriter.sdi sw/sdi/swriter.sdi
--- sw/sdi/swriter.sdi	2008-07-16 21:24:34.000000000 +0800
+++ sw/sdi/swriter.sdi	2008-09-04 23:01:47.000000000 +0800
@@ -77,6 +77,60 @@
 ]
 
 //--------------------------------------------------------------------------
+SfxBoolItem AcceptChange FN_REDLINE_ACCEPT_CURRENT
+
+[
+	/* flags: */
+	AutoUpdate = TRUE,
+	Cachable = Cachable,
+	FastCall = FALSE,
+	HasCoreId = FALSE,
+	HasDialog = FALSE,
+	ReadOnlyDoc = FALSE,
+	Toggle = TRUE,
+	Container = FALSE,
+	RecordAbsolute = FALSE,
+	RecordPerSet;
+	Synchron;
+
+	Readonly = FALSE,
+
+	/* config: */
+	AccelConfig = TRUE,
+	MenuConfig = TRUE,
+	StatusBarConfig = FALSE,
+	ToolBoxConfig = TRUE,
+	GroupId = GID_EDIT;
+]
+
+//--------------------------------------------------------------------------
+SfxBoolItem RejectChange FN_REDLINE_REJECT_CURRENT
+
+[
+	/* flags: */
+	AutoUpdate = TRUE,
+	Cachable = Cachable,
+	FastCall = FALSE,
+	HasCoreId = FALSE,
+	HasDialog = FALSE,
+	ReadOnlyDoc = FALSE,
+	Toggle = TRUE,
+	Container = FALSE,
+	RecordAbsolute = FALSE,
+	RecordPerSet;
+	Synchron;
+
+	Readonly = FALSE,
+
+	/* config: */
+	AccelConfig = TRUE,
+	MenuConfig = TRUE,
+	StatusBarConfig = FALSE,
+	ToolBoxConfig = TRUE,
+	GroupId = GID_EDIT;
+]
+
+//--------------------------------------------------------------------------
 SfxVoidItem AddAllUnknownWords FN_ADD_UNKNOWN
 ()
 [
diff -uNr sw/sdi/swslots.src sw/sdi/swslots.src
--- sw/sdi/swslots.src	2008-04-10 20:24:25.000000000 +0800
+++ sw/sdi/swslots.src	2008-09-04 23:01:48.000000000 +0800
@@ -942,6 +942,14 @@
 {
 	Slotname [ en-US ] = "Highlight Changes";
 };
+SfxSlotInfo FN_REDLINE_ACCEPT_CURRENT
+{
+	Slotname [ en-US ] = "Accept the Current Change";
+}
+SfxSlotInfo FN_REDLINE_REJECT_CURRENT
+{
+	Slotname [ en-US ] = "Reject the Current Change";
+}
 SfxSlotInfo FN_REFRESH_VIEW
 {
 	Slotname [ en-US ] = "Restore View";
diff -uNr sw/sdi/_textsh.sdi sw/sdi/_textsh.sdi
--- sw/sdi/_textsh.sdi	2008-04-11 21:59:45.000000000 +0800
+++ sw/sdi/_textsh.sdi	2008-09-07 15:22:27.000000000 +0800
@@ -936,6 +936,18 @@
 		StateMethod = StateField ;
 		DisableFlags="SW_DISABLE_ON_PROTECTED_CURSOR|SW_DISABLE_ON_MAILBOX_EDITOR";
 	]
+	FN_REDLINE_ACCEPT_CURRENT // status()
+	[
+		ExecMethod = ExecField ;
+		StateMethod = StateField ;
+		DisableFlags="SW_DISABLE_ON_PROTECTED_CURSOR|SW_DISABLE_ON_MAILBOX_EDITOR";
+	]
+	FN_REDLINE_REJECT_CURRENT // status()
+	[
+		ExecMethod = ExecField ;
+		StateMethod = StateField ;
+		DisableFlags="SW_DISABLE_ON_PROTECTED_CURSOR|SW_DISABLE_ON_MAILBOX_EDITOR";
+	]
 	FN_JAVAEDIT // status(final|play)
 	[
 		ExecMethod = ExecField ;
diff -uNr sw/source/ui/inc/popup.hrc sw/source/ui/inc/popup.hrc
--- sw/source/ui/inc/popup.hrc	2008-04-11 04:23:49.000000000 +0800
+++ sw/source/ui/inc/popup.hrc	2008-09-04 23:01:48.000000000 +0800
@@ -49,6 +49,7 @@
 #define MN_LANGUAGE_FOR_DOCUMENT	(RC_POPUPS_BEGIN + 13)
 #define MN_ANNOTATION_POPUPMENU		(RC_POPUPS_BEGIN + 14)
 #define MN_ANNOTATION_BUTTON		(RC_POPUPS_BEGIN + 15)
+#define MN_REDLINE_POPUPMENU            (RC_POPUPS_BEGIN + 16)
 
 #if MN_MEDIA_POPUPMENU > RC_POPUPS_END
 
diff -uNr sw/source/ui/inc/swmn_tmpl.hrc sw/source/ui/inc/swmn_tmpl.hrc
--- sw/source/ui/inc/swmn_tmpl.hrc	2008-04-11 04:30:37.000000000 +0800
+++ sw/source/ui/inc/swmn_tmpl.hrc	2008-09-05 00:31:13.000000000 +0800
@@ -62,7 +62,29 @@
         Identifier = FN_FORMAT_PAGE_DLG ; \
         HelpID = FN_FORMAT_PAGE_DLG ; \
         Text [ en-US ] = "Pa~ge..." ; \
-	};
+        };
+
+#define MN_REDLINE\
+        SEPARATOR ; \
+        MenuItem\
+        {\
+            Identifier = FN_REDLINE_ACCEPT_CURRENT ; \
+            HelpID = FN_REDLINE_ACCEPT_CURRENT ; \
+            Text [ en-US ] = "~Accept..."; \
+        };\
+        MenuItem \
+        { \
+            Identifier = FN_REDLINE_REJECT_CURRENT ; \
+            HelpID = FN_REDLINE_REJECT_CURRENT ; \
+            Text [ en-US ] = "Re~ject..."; \
+        }; \
+        SEPARATOR ; \
+        MenuItem \
+        { \
+            Identifier = FN_REDLINE_ON ; \
+            HelpID = FN_REDLINE_ON ; \
+            Text [ en-US ] = "~Record"; \
+        };
 
 
 #define MN_TEXT_ATTR\
@@ -315,6 +337,7 @@
         MN_TXT                                                              \
         MN_NUMBERING                                                        \
         ITEM_TRANSLITERATE_MENU                                             \
+        MN_REDLINE                                                          \
         SEPARATOR ;                                                         
 
 // not in Writer/Web
diff -uNr sw/source/ui/shells/textfld.cxx sw/source/ui/shells/textfld.cxx
--- sw/source/ui/shells/textfld.cxx	2008-04-11 22:03:59.000000000 +0800
+++ sw/source/ui/shells/textfld.cxx	2008-09-05 15:21:39.000000000 +0800
@@ -507,6 +507,22 @@
 			}
 			break;
 
+			case FN_REDLINE_ACCEPT_CURRENT :
+			{
+				const SwRedline * pRedln = rSh.GetCurrRedline();
+				if ( pRedln )
+					rSh.AcceptRedline( rSh.FindRedlineOfData( pRedln->GetRedlineData()) );
+			}
+			break;
+
+			case FN_REDLINE_REJECT_CURRENT :
+			{
+				const SwRedline * pRedln = rSh.GetCurrRedline();
+				if ( pRedln )
+					rSh.RejectRedline( rSh.FindRedlineOfData( pRedln->GetRedlineData()) );
+			}
+			break;
+
 			case FN_JAVAEDIT:
 			{
                 String aType, aText;
@@ -710,6 +726,11 @@
 				if (!rSh.GetCurrRedline())
 					rSet.DisableItem(nWhich);
 				break;
+			case FN_REDLINE_ACCEPT_CURRENT :
+			case FN_REDLINE_REJECT_CURRENT :
+				if (!(nsRedlineMode_t::REDLINE_SHOW_MASK & rSh.GetRedlineMode()) || !rSh.GetCurrRedline())
+					rSet.DisableItem(nWhich);
+			break;
 			case FN_POSTIT :
 			case FN_JAVAEDIT :
 				BOOL bCurField = FALSE;
