diff --git sw/source/core/unocore/unodraw.cxx sw/source/core/unocore/unodraw.cxx
index 09c5136..5ca1081 100644
--- sw/source/core/unocore/unodraw.cxx
+++ sw/source/core/unocore/unodraw.cxx
@@ -1966,6 +1966,7 @@ void SwXShape::attach(const uno::Reference< text::XTextRange > & xTextRange)
         OTextCursorHelper* pCursor = 0;
         SwXTextPortion* pPortion = 0;
         SwXText* pText = 0;
+        SwXParagraph* pParagraph = 0;
 
 		pRange 	= reinterpret_cast< SwXTextRange * >(
 				sal::static_int_cast< sal_IntPtr >( xRangeTunnel->getSomething( SwXTextRange::getUnoTunnelId()) ));
@@ -1975,6 +1976,8 @@ void SwXShape::attach(const uno::Reference< text::XTextRange > & xTextRange)
 				sal::static_int_cast< sal_IntPtr >( xRangeTunnel->getSomething( OTextCursorHelper::getUnoTunnelId()) ));
         pPortion = reinterpret_cast< SwXTextPortion * >(
 				sal::static_int_cast< sal_IntPtr >( xRangeTunnel->getSomething( SwXTextPortion::getUnoTunnelId()) ));
+        pParagraph = reinterpret_cast< SwXParagraph * >(
+                sal::static_int_cast< sal_IntPtr >( xRangeTunnel->getSomething( SwXParagraph::getUnoTunnelId( ) ) ) );
 
         if (pRange)
             pDoc = pRange->GetDoc();
@@ -1984,6 +1987,8 @@ void SwXShape::attach(const uno::Reference< text::XTextRange > & xTextRange)
             pDoc = pCursor->GetDoc();
 		else if ( !pDoc && pPortion && pPortion->GetCrsr() )
 			pDoc = pPortion->GetCrsr()->GetDoc();
+        else if ( !pDoc && pParagraph )
+           pDoc = pParagraph->GetCrsr( )->GetDoc( ); 
 
     }
 
