diff --git sw/source/filter/ww8/WW8TableInfo.cxx sw/source/filter/ww8/WW8TableInfo.cxx
--- sw/source/filter/ww8/WW8TableInfo.cxx
+++ sw/source/filter/ww8/WW8TableInfo.cxx
@@ -264,7 +264,7 @@ WW8TableInfo::processTableBox(const SwTable * pTable,
                 if (nDepthInsideCell == 1 && pEndOfCellInfo.get() == NULL)
                     pEndOfCellInfo = pNodeInfo;
                 
-                SwEndNode * pEndNode = dynamic_cast<SwEndNode *> (&rNode);                
+                SwEndNode * pEndNode = rNode.GetEndNode( );
                 SwStartNode * pTmpSttNd = pEndNode->StartOfSectionNode();
                 if (pTmpSttNd == pSttNd)
                     bDone = true;
