--- sw/source/filter/html/css1atr.cxx.orig	2008-05-16 10:10:24.000000000 +0200
+++ sw/source/filter/html/css1atr.cxx	2008-05-16 10:28:16.000000000 +0200
@@ -2939,7 +2939,7 @@
 	case WEIGHT_ULTRALIGHT:	pStr = sCSS1_PV_extra_light; 	break;
 	case WEIGHT_LIGHT:		pStr = sCSS1_PV_light; 			break;
 	case WEIGHT_SEMILIGHT:	pStr = sCSS1_PV_demi_light; 	break;
-	case WEIGHT_NORMAL:		pStr = sCSS1_PV_medium;			break;
+	case WEIGHT_NORMAL:		pStr = sCSS1_PV_normal;			break;
 	case WEIGHT_SEMIBOLD:	pStr = sCSS1_PV_demi_bold; 		break;
 	case WEIGHT_BOLD:
 		if( !rHTMLWrt.IsCSS1Source( CSS1_OUTMODE_PARA ) )
@@ -2953,7 +2953,7 @@
 		break;
 	case WEIGHT_ULTRABOLD:	pStr = sCSS1_PV_extra_bold;		break;
 	default:
-		;
+		pStr = sCSS1_PV_normal;
 	}
 
 	if( pStr )
