diff -u -r sw/source/filter/ww8/wrtw8sty.cxx sw/source/filter/ww8/wrtw8sty.cxx
--- sw/source/filter/ww8/wrtw8sty.cxx	2008-02-02 10:09:08.000000000 +0100
+++ sw/source/filter/ww8/wrtw8sty.cxx	2008-02-02 10:17:40.000000000 +0100
@@ -1054,6 +1054,7 @@
 {
     if( nFlag & nHFFlags )
     {
+        rWrt.bHasHdr = true;
         const SwFmtHeader& rHd = rFmt.GetHeader();
         ASSERT( rHd.GetHeaderFmt(), "KopfText nicht richtig da" );
         const SwFmtCntnt& rCntnt = rHd.GetHeaderFmt()->GetCntnt();
@@ -1065,8 +1066,12 @@
     else if( rWrt.bWrtWW8 )
     {
         pTxtPos->Append( rCpPos );
-        rWrt.WriteStringAsPara( aEmptyStr ); // CR ans Ende ( sonst mault WW )
-        rCpPos = rWrt.Fc2Cp( rWrt.Strm().Tell() );
+        if (rWrt.bHasHdr)
+        {
+            rWrt.WriteStringAsPara( aEmptyStr ); // Empty paragraph for empty header
+            rWrt.WriteStringAsPara( aEmptyStr ); // a CR that WW8 needs for end of the stream
+            rCpPos = rWrt.Fc2Cp( rWrt.Strm().Tell() );
+        }
     }
 }
 void WW8_WrPlcSepx::OutFooter( SwWW8Writer& rWrt, const SwFmt& rFmt,
@@ -1074,6 +1079,7 @@
 {
     if( nFlag & nHFFlags )
     {
+        rWrt.bHasFtr = true;
         const SwFmtFooter& rFt = rFmt.GetFooter();
         ASSERT( rFt.GetFooterFmt(), "KopfText nicht richtig da" );
         const SwFmtCntnt& rCntnt = rFt.GetFooterFmt()->GetCntnt();
@@ -1085,8 +1091,12 @@
     else if( rWrt.bWrtWW8 )
     {
         pTxtPos->Append( rCpPos );
-        rWrt.WriteStringAsPara( aEmptyStr ); // CR ans Ende ( sonst mault WW )
-        rCpPos = rWrt.Fc2Cp( rWrt.Strm().Tell() );
+        if (rWrt.bHasFtr)
+        {
+            rWrt.WriteStringAsPara( aEmptyStr ); // Empty paragraph for empty footer
+            rWrt.WriteStringAsPara( aEmptyStr ); // a CR that WW8 needs for end of the stream
+            rCpPos = rWrt.Fc2Cp( rWrt.Strm().Tell() );
+        }
     }
 }
 
diff -u -r sw/source/filter/ww8/wrtww8.cxx sw/source/filter/ww8/wrtww8.cxx
--- sw/source/filter/ww8/wrtww8.cxx	2008-02-02 10:09:08.000000000 +0100
+++ sw/source/filter/ww8/wrtww8.cxx	2008-02-02 10:10:12.000000000 +0100
@@ -2553,7 +2553,8 @@
 SwWW8Writer::SwWW8Writer(const String& rFltName, const String& rBaseURL)
     : aMainStg(sMainStream), pISet(0), pUsedNumTbl(0), mpTopNodeOfHdFtPage(0),
     pBmpPal(0), pKeyMap(0), pOLEExp(0), pOCXExp(0), pOleMap(0), nUniqueList(0),
-    mnHdFtIndex(0), pAktPageDesc(0), pPapPlc(0), pChpPlc(0), pChpIter(0), pO(0)
+    mnHdFtIndex(0), pAktPageDesc(0), pPapPlc(0), pChpPlc(0), pChpIter(0), pO(0),
+    bHasHdr(false), bHasFtr(false)
 {
     SetBaseURL( rBaseURL );
     bWrtWW8 = rFltName.EqualsAscii(FILTER_WW8);
diff -u -r sw/source/filter/ww8/wrtww8.hxx sw/source/filter/ww8/wrtww8.hxx
--- sw/source/filter/ww8/wrtww8.hxx	2008-02-02 10:09:08.000000000 +0100
+++ sw/source/filter/ww8/wrtww8.hxx	2008-02-02 10:10:12.000000000 +0100
@@ -536,6 +536,8 @@
     BYTE bInWriteTOX : 1;       // true: all content are in a TOX
     BYTE bFtnAtTxtEnd : 1;      // true: all FTN at Textend
     BYTE bEndAtTxtEnd : 1;      // true: all END at Textend
+    BYTE bHasHdr : 1;
+    BYTE bHasFtr : 1;
 
 
 
