--- sw/source/filter/ww8/ww8par2.cxx	28 Feb 2007 15:54:50 -0000	1.129
+++ sw/source/filter/ww8/ww8par2.cxx	27 May 2007 18:36:27 -0000
@@ -2551,6 +2551,8 @@ void WW8TabDesc::UseSwTable()
         GetSttNd()->FindTableNode();
     ASSERT( pTblNd, "wo ist mein TabellenNode" );
 
+	if (nRowsToRepeat == USHORT( nRows))
+	  nRowsToRepeat = 1;
     pTblNd->GetTable().SetRowsToRepeat( nRowsToRepeat );
     // ggfs. Zusatz-Zellen einfuegen u.dgl.
     AdjustNewBand();
