--- sw/source//filter/ww8/ww8par.cxx	2008-07-04 22:05:07.000000000 +0200
+++ sw/source//filter/ww8/ww8par.cxx	2008-07-04 22:01:31.000000000 +0200
@@ -3435,8 +3435,14 @@
     for (mySegIter aIter = aStart; aIter != aEnd; ++aIter)
     {
         mySegIter aNext = aIter+1;
+        mySegIter aPrev = (aIter == aStart) ? aIter : aIter-1;
 
-        bool bInsertSection = aIter != aStart ? aIter->IsContinous() : false;
+        // If two following sections are different in following properties, Word will interprete a continuous
+        // section break between them as if it was a section break next page.
+        bool bThisAndPreviousAreCompatible = ((aIter->GetPageWidth() == aPrev->GetPageWidth()) &&
+            (aIter->GetPageHeight() == aPrev->GetPageHeight()) && (aIter->IsLandScape() == aPrev->IsLandScape()));
+
+        bool bInsertSection = (aIter != aStart) ? (aIter->IsContinous() &&  bThisAndPreviousAreCompatible): false;
         bool bInsertPageDesc = !bInsertSection;
         bool bProtected = SectionIsProtected(*aIter);
 
@@ -3452,7 +3458,10 @@
             */
 
             bool bIgnoreCols = false;
-            if ((aNext != aEnd && aNext->IsContinous() || bProtected))
+            bool bThisAndNextAreCompatible = (aNext != aEnd) ? ((aIter->GetPageWidth() == aNext->GetPageWidth()) &&
+                (aIter->GetPageHeight() == aNext->GetPageHeight()) && (aIter->IsLandScape() == aNext->IsLandScape())) : true;
+
+            if ((aNext != aEnd && aNext->IsContinous() && bThisAndNextAreCompatible || bProtected))
             {
                 bIgnoreCols = true;
                 if ((aIter->NoCols() > 1) || bProtected)
