Index: sw/source/filter/ww8/ww8graf.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/ww8graf.cxx,v
retrieving revision 1.150
diff -u -r1.150 ww8graf.cxx
--- sw/source/filter/ww8/ww8graf.cxx	13 Nov 2007 15:29:52 -0000	1.150
+++ sw/source/filter/ww8/ww8graf.cxx	29 May 2008 15:01:28 -0000
@@ -2789,7 +2789,7 @@
     // Switch on 'follow text flow',
     // if object is laid out inside table cell and
     // its wrapping isn't 'SURROUND_THROUGH'
-    if (bLayoutInTableCell && eSurround != SURROUND_THROUGHT)
+    if (bLayoutInTableCell /*&& eSurround != SURROUND_THROUGHT*/)
     {
         SwFmtFollowTextFlow aFollowTextFlow( TRUE );
         aFlySet.Put( aFollowTextFlow );
Index: svx/source/msfilter/msdffimp.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/msdffimp.cxx,v
retrieving revision 1.152.182.1
diff -u -r1.152.182.1 msdffimp.cxx
--- svx/source/msfilter/msdffimp.cxx	18 Jan 2008 12:11:23 -0000	1.152.182.1
+++ svx/source/msfilter/msdffimp.cxx	29 May 2008 15:01:31 -0000
@@ -7971,7 +7971,7 @@
       nXRelTo( 2 ), //   relative to column
       nYAlign( 0 ), // position n cm below
       nYRelTo( 2 ), //   relative to paragraph
-      nLayoutInTableCell( 0 ), // element is laid out in table cell
+      nLayoutInTableCell( 0x00008000 ), // element is laid out in table cell
       nTextRotationAngle( 0 ),
       nDxTextLeft( 144 ),
       nDyTextTop( 72 ),
