Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.252
retrieving revision 1.249.12.6
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.252 -r1.249.12.6
--- config_office/configure.in	7 Mar 2008 16:56:35 -0000	1.252
+++ config_office/configure.in	25 Mar 2008 11:59:49 -0000	1.249.12.6
@@ -443,8 +449,11 @@
 AC_ARG_WITH(system-altlinuxhyph,
 [  --with-system-altlinuxhyph    Use ALTLinuxhyph already on system
 ],,)
+AC_ARG_WITH(system-lpsolve,
+[  --with-system-lpsolve         Use lpsolve already on system
+],,)
 AC_ARG_WITH(system-mozilla,
 [  --with-system-mozilla   Use mozilla already on system. Note that some
                           components cannot be built against a contemporary
                           mozilla. The flavour used can be specified by
                           --with-system-mozilla=<flavour>. Supported are:
@@ -4372,6 +4413,25 @@
 fi
 AC_SUBST(SYSTEM_MYTHES)
 
+dnl ===================================================================
+dnl Checking for lpsolve
+dnl ===================================================================
+AC_MSG_CHECKING([which lpsolve to use])
+if test -n "$with_system_lpsolve" -o -n "$with_system_libs" && \
+	test "$with_system_lpsolve" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_LPSOLVE=YES
+    AC_CHECK_HEADER(lpsolve/lp_lib.h, [],
+       [ AC_MSG_ERROR(lpsolve headers not found.)], [])
+    AC_CHECK_LIB(lpsolve55, make_lp, ,
+        [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_LPSOLVE=NO
+    BUILD_TYPE="$BUILD_TYPE LPSOLVE"
+fi
+AC_SUBST(SYSTEM_LPSOLVE)
+
 dnl ***************************************
 dnl testing libc version for Linux...
 dnl ***************************************
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.178
retrieving revision 1.174.6.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.178 -r1.174.6.2
--- config_office/set_soenv.in	7 Mar 2008 16:56:50 -0000	1.178
+++ config_office/set_soenv.in	12 Mar 2008 10:59:06 -0000	1.174.6.2
@@ -1980,6 +1980,7 @@
 ToFile( "SYSTEM_HUNSPELL",   "@SYSTEM_HUNSPELL@",  "e" );
 ToFile( "HUNSPELL_CFLAGS",   "@HUNSPELL_CFLAGS@",  "e" );
 ToFile( "HUNSPELL_LIBS",     "@HUNSPELL_LIBS@",    "e" );
+ToFile( "SYSTEM_LPSOLVE",    "@SYSTEM_LPSOLVE@",   "e" );
 ToFile( "SYSTEM_HYPH",       "@SYSTEM_HYPH@",      "e" );
 ToFile( "HYPHEN_LIB",        "@HYPHEN_LIB@",       "e" );
 ToFile( "SYSTEM_MYTHES",     "@SYSTEM_MYTHES@",    "e" );
