Index: scp2/source/calc/file_calc.scp
===================================================================
RCS file: /cvs/installation/scp2/source/calc/file_calc.scp,v
retrieving revision 1.38
retrieving revision 1.38.42.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.38 -r1.38.42.1
--- scp2/source/calc/file_calc.scp	5 Feb 2008 15:51:55 -0000	1.38
+++ scp2/source/calc/file_calc.scp	10 Mar 2008 12:19:09 -0000	1.38.42.1
@@ -37,10 +37,12 @@
 
 STD_UNO_LIB_FILE( gid_File_Lib_Solver, solver)
 
+#ifndef SYSTEM_LPSOLVE
 File gid_File_Lib_Lpsolve
     Name = SPECIAL_NAME(lpsolve55);
     PACKED_LIB_FILE_BODY;
 End
+#endif
 
 STD_UNO_LIB_FILE( gid_File_Lib_Analysis, analysis)
 
Index: scp2/source/calc/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/calc/makefile.mk,v
retrieving revision 1.6
retrieving revision 1.6.1008.1
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.1008.1
--- scp2/source/calc/makefile.mk	9 Sep 2005 01:21:34 -0000	1.6
+++ scp2/source/calc/makefile.mk	10 Mar 2008 12:19:09 -0000	1.6.1008.1
@@ -49,6 +49,10 @@
 SCPDEFS+=-DWITHOUT_BINFILTER
 .ENDIF
 
+.IF "$(SYSTEM_LPSOLVE)" == "YES"
+SCPDEFS+=-DSYSTEM_LPSOLVE
+.ENDIF
+
 SCP_PRODUCT_TYPE=osl
 PARFILES= \
         module_calc.par              \
