--- tools/inc/tools/urlobj.hxx	2006-07-20 18:09:51.000000000 +0200
+++ tools/inc/tools/urlobj.hxx	2007-06-14 11:36:58.000000000 +0200
@@ -1557,6 +1557,9 @@ private:
 
     TOOLS_DLLPRIVATE static bool scanIPv6reference(
         sal_Unicode const *& rBegin, sal_Unicode const * pEnd);
+
+private:
+    void changeScheme(INetProtocol eTargetScheme);
 };
 
 // static
--- tools/source/fsys/urlobj.cxx.old	2006-09-17 02:54:24.000000000 +0200
+++ tools/source/fsys/urlobj.cxx	2007-06-14 11:36:58.000000000 +0200
@@ -1460,10 +1460,48 @@ bool INetURLObject::setAbsURIRef(rtl::OU
 
 	m_aAbsURIRef = aSynAbsURIRef;
 
+	// At this point references of type "\\server\paths" have
+	// been converted to file:://server/path".
+#ifdef LINUX
+	if (m_eScheme==INET_PROT_FILE && !m_aHost.isEmpty()) {
+	    // Change "file:://server/path" URIs to "smb:://server/path" on
+	    // Linux
+	    // Leave "file::path" URIs unchanged.
+	    changeScheme(INET_PROT_SMB);	   
+	}
+#endif
+
+#ifdef WIN
+	if (m_eScheme==INET_PROT_SMB) {
+	    // Change "smb://server/path" URIs to "file://server/path"
+	    // URIs on Windows, since Windows doesn't understand the
+	    // SMB scheme.
+	    changeScheme(INET_PROT_FILE);
+	}
+#endif
+
 	return true;
 }
 
 //============================================================================
+void INetURLObject::changeScheme(INetProtocol eTargetScheme) {
+    ::rtl::OUString aTmpStr=m_aAbsURIRef.makeStringAndClear();
+    int oldSchemeLen=strlen(getSchemeInfo().m_pScheme);
+    m_eScheme=eTargetScheme;	   	   
+    int newSchemeLen=strlen(getSchemeInfo().m_pScheme);
+    m_aAbsURIRef.appendAscii(getSchemeInfo().m_pScheme);
+    m_aAbsURIRef.append(aTmpStr.getStr()+oldSchemeLen);
+    int delta=newSchemeLen-oldSchemeLen;
+    m_aUser+=delta;
+    m_aAuth+=delta;
+    m_aHost+=delta;
+    m_aPort+=delta;
+    m_aPath+=delta;
+    m_aQuery+=delta;
+    m_aFragment+=delta;
+}
+
+//============================================================================
 bool INetURLObject::convertRelToAbs(rtl::OUString const & rTheRelURIRef,
 									bool bOctets,
 									INetURLObject & rTheAbsURIRef,
--- sw/source/filter/ww8/wrtw8nds.cxx	26 Jun 2007 10:44:02 -0000	1.100
+++ sw/source/filter/ww8/wrtw8nds.cxx	8 Oct 2007 10:18:08 -0000
@@ -913,7 +913,7 @@
     SwWW8Writer::WriteLong( *rWrt.pDataStrm, nFlag );
 
     INetProtocol eProto = aURL.GetProtocol();
-    if (eProto == INET_PROT_FILE)
+    if (eProto == INET_PROT_FILE || eProto == INET_PROT_SMB)
     {
 // version 1 (for a document)
 
@@ -931,7 +931,7 @@
 
         // save the links to files as relative
         sURL = URIHelper::simpleNormalizedMakeRelative( rWrt.GetBaseURL(), sURL );
-        if ( sURL.EqualsAscii( "/", 0, 1 ) )
+        if ( eProto == INET_PROT_FILE && sURL.EqualsAscii( "/", 0, 1 ) )
             sURL = aURL.PathToFileName();
 
         // special case for the absolute windows names
@@ -945,6 +945,15 @@
             sURL.SearchAndReplaceAll( '/', '\\' );
         }
 
+        // n#261623 convert smb notation to '\\'
+        const char pSmb[] = "smb://";
+        if ( eProto == INET_PROT_SMB &&
+             sURL.EqualsAscii( pSmb, 0, sizeof( pSmb ) - 1 ) )
+        {
+            sURL.Erase( 0, sizeof( pSmb ) - 3 );
+            sURL.SearchAndReplaceAll( '/', '\\' );
+        }
+
         rWrt.pDataStrm->Write( MAGIC_C, sizeof(MAGIC_C) );
         SwWW8Writer::WriteLong( *rWrt.pDataStrm, sURL.Len()+1 );
         SwWW8Writer::WriteString8( *rWrt.pDataStrm, sURL, true,
