--- tools/source/rc/resmgr.cxx.old	2008-01-23 18:05:46.000000000 +0100
+++ tools/source/rc/resmgr.cxx	2008-06-17 20:00:31.000000000 +0200
@@ -323,6 +323,10 @@ void ResMgrContainer::init()
                  OUStringToOString( it->second.aFileURL, osl_getThreadTextEncoding() ).getStr() );
     }
     #endif
+
+    // set default language
+    LanguageType nLang = MsLangId::getSystemUILanguage();
+    MsLangId::convertLanguageToLocale(nLang, m_aDefLocale);
 }
 
 InternalResMgr* ResMgrContainer::getResMgr( const OUString& rPrefix,
