officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu
--- officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu	2007-12-05 15:02:57.000000000 +0100
+++ officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu	2007-12-05 16:29:43.000000000 +0100
@@ -1912,6 +1912,76 @@
					
				</prop>
			</node>
+      <node oor:name="tile-flip" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Flipping tiles</value>
+        </prop>
+      </node>
+      <node oor:name="outside-cube" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Outside turning cube</value>
+        </prop>
+      </node>
+      <node oor:name="revolving-circles" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Revolving circles</value>
+        </prop>
+      </node>
+      <node oor:name="turning-helix" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Turning helix</value>
+        </prop>
+      </node>
+      <node oor:name="inside-cube" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Inside turning cube</value>
+        </prop>
+      </node>
+      <node oor:name="fall" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Fall</value>
+        </prop>
+      </node>
+      <node oor:name="turn-around" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Turn around</value>
+        </prop>
+      </node>
+      <node oor:name="iris" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Iris</value>
+        </prop>
+      </node>
+      <node oor:name="turn-down" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Turn down</value>
+        </prop>
+      </node>
+      <node oor:name="rochade" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Rochade</value>
+        </prop>
+      </node>
+      <node oor:name="venetian3dv" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Venetian Blinds 3D Vertical</value>
+        </prop>
+      </node>
+      <node oor:name="venetian3dh" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Venetian Blinds 3D Horizontal</value>
+        </prop>
+      </node>
+      <node oor:name="static" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Static</value>
+        </prop>
+      </node>
+      <node oor:name="finedissolve" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Fine Dissolve</value>
+        </prop>
+      </node>
 		</node>
 	</node>	
 	<node oor:name="Presets">
--- officecfg/registry/schema/org/openoffice/Office/Impress.xcs	18 Jan 2008 12:59:24 -0000	1.29.126.1
+++ officecfg/registry/schema/org/openoffice/Office/Impress.xcs	5 Feb 2008 01:00:49 -0000
@@ -456,7 +456,7 @@
 					is allowed. </desc>
 					<label>List of files containing transition effects</label>
 				</info>
-                <value oor:separator=";">vnd.sun.star.expand:$OOO_BASE_DIR/share/config/soffice.cfg/simpress/transitions.xml</value>
+                <value oor:separator=";">vnd.sun.star.expand:$OOO_BASE_DIR/share/config/soffice.cfg/simpress/transitions.xml;vnd.sun.star.expand:$OOO_BASE_DIR/share/config/soffice.cfg/simpress/transitions-ogl.xml</value>
 			</prop>
 			<prop oor:name="EffectFiles" oor:type="oor:string-list">
 				<!-- OldPath: -->
