--- slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx.old	2008-09-10 05:29:42.000000000 +0200
+++ slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx	2008-09-10 16:54:02.000000000 +0200
@@ -78,6 +78,11 @@ namespace unx
 }
 #endif
 
+namespace unx {
+typedef void ( * PFNGLXBINDTEXIMAGEEXTPROC) (Display *dpy, GLXDrawable drawable, int buffer, const int *attrib_list);
+typedef void ( * PFNGLXRELEASETEXIMAGEEXTPROC) (Display *dpy, GLXDrawable drawable, int buffer);
+}
+
 #ifdef DEBUG
 #include <boost/date_time/posix_time/posix_time.hpp>
 using namespace ::boost::posix_time;
