--- slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx.old	2009-04-06 16:42:09.000000000 +0000
+++ slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx	2009-04-06 16:42:09.000000000 +0000
@@ -93,7 +93,7 @@ public:
     explicit OGLTransitionerImpl(OGLTransitionImpl* pOGLTransition);
     bool initWindowFromSlideShowView( const uno::Reference< presentation::XSlideShowView >& xView, double, double);
     void setSlides( const uno::Reference< rendering::XBitmap >& xLeavingSlide , const uno::Reference< rendering::XBitmap >& xEnteringSlide );
-    static void initialize( const uno::Reference< presentation::XSlideShowView >& xView );
+    static bool initialize( const uno::Reference< presentation::XSlideShowView >& xView );
 
     // XTransition
 	virtual void SAL_CALL update( double nTime ) throw (uno::RuntimeException);
@@ -179,14 +179,20 @@ public:
     /** Whether Mesa is the OpenGL vendor
      */
     static bool cbMesa;
+
+    /**
+       whether the display has GLX extension
+     */
+    static bool cbGLXPresent;
 };
 
 // declare the static variables as some gcc versions have problems declaring them automaticaly
 bool OGLTransitionerImpl::cbBrokenTexturesATI;
 float OGLTransitionerImpl::cnGLVersion;
 bool OGLTransitionerImpl::cbMesa;
+bool OGLTransitionerImpl::cbGLXPresent;
 
-void OGLTransitionerImpl::initialize( const uno::Reference< presentation::XSlideShowView >& xView )
+bool OGLTransitionerImpl::initialize( const uno::Reference< presentation::XSlideShowView >& xView )
 {
     // not thread safe
     static bool initialized = false;
@@ -195,28 +201,34 @@ void OGLTransitionerImpl::initialize( co
         OGLTransitionerImpl *instance;
 
         instance = new OGLTransitionerImpl( NULL );
-        instance->initWindowFromSlideShowView( xView, 0, 0 );
+        if( instance->initWindowFromSlideShowView( xView, 0, 0 ) ) {
 
-        const GLubyte* version = glGetString( GL_VERSION );
-        if( version && version[0] ) {
-            cnGLVersion = version[0] - '0';
-            if( version[1] == '.' && version[2] )
-                cnGLVersion += (version[2] - '0')/10.0;
-        } else
-            cnGLVersion = 1.0;
-        OSL_TRACE("GL version: %s parsed: %f", version, cnGLVersion );
+            const GLubyte* version = glGetString( GL_VERSION );
+            if( version && version[0] ) {
+                cnGLVersion = version[0] - '0';
+                if( version[1] == '.' && version[2] )
+                    cnGLVersion += (version[2] - '0')/10.0;
+            } else
+                cnGLVersion = 1.0;
+            OSL_TRACE("GL version: %s parsed: %f", version, cnGLVersion );
+
+            const GLubyte* vendor = glGetString( GL_VENDOR );
+            cbMesa = ( vendor && strstr( (const char *) vendor, "Mesa" ) );
+            OSL_TRACE("GL vendor: %s identified as Mesa: %d", vendor, cbMesa );
 
-        const GLubyte* vendor = glGetString( GL_VENDOR );
-        cbMesa = ( vendor && strstr( (const char *) vendor, "Mesa" ) );
-        OSL_TRACE("GL vendor: %s identified as Mesa: %d", vendor, cbMesa );
+            /* TODO: check for version once the bug in fglrx driver is fixed */
+            cbBrokenTexturesATI = (vendor && strcmp( (const char *) vendor, "ATI Technologies Inc." ) == 0 );
 
-        /* TODO: check for version once the bug in fglrx driver is fixed */
-        cbBrokenTexturesATI = (strcmp( (const char *) glGetString( GL_VENDOR ), "ATI Technologies Inc." ) == 0 );
+            instance->disposing();
+            cbGLXPresent = true;
+        } else
+            cbGLXPresent = false;
 
-        instance->disposing();
         delete instance;
         initialized = true;
     }
+
+    return cbGLXPresent;
 }
 
 bool OGLTransitionerImpl::initWindowFromSlideShowView( const uno::Reference< presentation::XSlideShowView >& xView, double, double)
@@ -243,6 +255,10 @@ bool OGLTransitionerImpl::initWindowFrom
 	GLWin.hWnd = sysData->hWnd;
 #elif defined( UNX )
     GLWin.dpy = reinterpret_cast<unx::Display*>(sysData->pDisplay);
+
+    if( unx::glXQueryExtension( GLWin.dpy, NULL, NULL ) == false )
+        return false;
+
     GLWin.win = sysData->aWindow;
     GLWin.screen = unx::XDefaultScreen(GLWin.dpy);
     unx::XVisualInfo* vi( NULL );
@@ -770,7 +786,7 @@ void SAL_CALL OGLTransitionerImpl::updat
 {
     osl::MutexGuard const guard( m_aMutex );
 
-    if (isDisposed() || pTransition->mnRequiredGLVersion > cnGLVersion)
+    if (isDisposed() || !cbGLXPresent || pTransition->mnRequiredGLVersion > cnGLVersion)
         return;
 
     glEnable(GL_DEPTH_TEST);
@@ -904,7 +920,7 @@ public:
         if( !hasTransition( transitionType, transitionSubType ) )
             return uno::Reference< presentation::XTransition >();
 
-        OGLTransitionerImpl::initialize( view );
+        bool bGLXPresent = OGLTransitionerImpl::initialize( view );
 
         if( OGLTransitionerImpl::cbMesa && (
             ( transitionType == animations::TransitionType::FADE && transitionSubType == animations::TransitionSubType::CROSSFADE ) ||
@@ -974,9 +990,11 @@ public:
 
         rtl::Reference<OGLTransitionerImpl> xRes(
             new OGLTransitionerImpl(pTransition) );
-        if(!xRes->initWindowFromSlideShowView(view,slideOffset.X,slideOffset.Y))
-            return uno::Reference< presentation::XTransition >();
-        xRes->setSlides(leavingBitmap,enteringBitmap);
+        if( bGLXPresent ) {
+            if( !xRes->initWindowFromSlideShowView(view,slideOffset.X,slideOffset.Y))
+                return uno::Reference< presentation::XTransition >();
+            xRes->setSlides(leavingBitmap,enteringBitmap);
+        }
 
         return uno::Reference<presentation::XTransition>(xRes.get());
     }
