Index: typesconfig/typesconfig.c
===================================================================
--- sal/typesconfig/typesconfig.c	(revision 269860)
+++ sal/typesconfig/typesconfig.c	(working copy)
@@ -163,20 +163,26 @@
 |*	Letzte Aenderung
 |*
 *************************************************************************/
-static int dummy(void* unused);
+#if defined(IA64) || defined(ARM32)
 
 int GetAtAddress( Type eT, void* p )
 {
-#if defined(IA64) || defined(ARM32)
   switch ( eT )
   {
   case t_char:		return *((char*)p);
-  case t_short:		if ((long)p % sizeof(short)) abort(); else return *((short*)p);
-  case t_int:		if ((long)p % sizeof(int)) abort(); else return *((int*)p);
-  case t_long:		if ((long)p % sizeof(long)) abort(); else return *((long*)p);
-  case t_double:	if ((long)p % sizeof(double)) abort(); else return *((double*)p);
+  case t_short:		if ((long)p % sizeof(short)) raise(SIGBUS); else return *((short*)p);
+  case t_int:		if ((long)p % sizeof(int)) raise(SIGBUS); else return *((int*)p);
+  case t_long:		if ((long)p % sizeof(long)) raise(SIGBUS); else return *((long*)p);
+  case t_double:	if ((long)p % sizeof(double)) raise(SIGBUS); else return *((double*)p);
   }
+  abort();
+}
+
 #else
+static int dummy(void* unused);
+
+int GetAtAddress( Type eT, void* p )
+{
   switch ( eT )
   {
   case t_char: { char x = *(char*)p; return dummy(&x); }
@@ -185,7 +191,6 @@
   case t_long: { long x = *(long*)p; return dummy(&x); }
   case t_double: { double x = *(double*)p; return dummy(&x); }
   }
-#endif
   abort();
 }
 
@@ -195,6 +200,7 @@
     return 0;
 }
 
+#endif
 /*************************************************************************
 |*
 |*	SetAtAddress()
