--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx.orig	2009-03-19 23:28:01.000000000 -0500
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2009-03-19 23:32:27.000000000 -0500
@@ -184,7 +184,7 @@
 
 	gtk_dialog_set_default_response( GTK_DIALOG (m_pDialog), GTK_RESPONSE_ACCEPT );
 
-	gtk_file_chooser_set_local_only( GTK_FILE_CHOOSER( m_pDialog ), FALSE );
+	gtk_file_chooser_set_local_only( GTK_FILE_CHOOSER( m_pDialog ), TRUE );
 	gtk_file_chooser_set_select_multiple( GTK_FILE_CHOOSER( m_pDialog ), FALSE );
 
 	m_pVBox = gtk_vbox_new( FALSE, 0 );
--- fpicker/source/unx/gnome/SalGtkFolderPicker.cxx.orig	2009-03-19 23:28:01.000000000 -0500
+++ fpicker/source/unx/gnome/SalGtkFolderPicker.cxx	2009-03-19 23:32:14.000000000 -0500
@@ -96,7 +96,7 @@
 		GTK_STOCK_OK, GTK_RESPONSE_ACCEPT, (char *)NULL );
 
 	gtk_dialog_set_default_response( GTK_DIALOG (m_pDialog), GTK_RESPONSE_ACCEPT );
-	gtk_file_chooser_set_local_only( GTK_FILE_CHOOSER( m_pDialog ), FALSE );
+	gtk_file_chooser_set_local_only( GTK_FILE_CHOOSER( m_pDialog ), TRUE );
 	gtk_file_chooser_set_select_multiple( GTK_FILE_CHOOSER( m_pDialog ), FALSE );
 }
 
