--- jvmfwk/plugins/sunmajor/javaenvsetup/javaldx.cxx.orig	2008-04-11 05:19:31.000000000 -0500
+++ jvmfwk/plugins/sunmajor/javaenvsetup/javaldx.cxx	2009-03-05 19:54:38.000000000 -0600
@@ -111,6 +111,8 @@
         if (errcode == JFW_E_NO_JAVA_FOUND)
         {
             fprintf(stderr,"javaldx: Could not find a Java Runtime Environment! \n");
+            fprintf(stderr,"Please ensure that the package openoffice.org-java-common is installed.\n");
+            fprintf(stderr,"If it is already installed then try removing ~/.openoffice.org/3/user/config/javasettings_Linux_*.xml\n");
             return 0;
         }
         else if (errcode != JFW_E_NONE && errcode != JFW_E_DIRECT_MODE)
