--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx.orig	2009-03-12 21:32:50.000000000 -0500
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2009-03-12 21:38:04.000000000 -0500
@@ -845,9 +845,9 @@
 	OSL_ASSERT( m_pDialog != NULL );
 	::vos::OGuard aGuard( Application::GetSolarMutex() );
 
-	GSList* pPathList = gtk_file_chooser_get_uris( GTK_FILE_CHOOSER(m_pDialog) );
+	GSList* pFileList = gtk_file_chooser_get_files( GTK_FILE_CHOOSER(m_pDialog) );
 	
-	int nCount = g_slist_length( pPathList );
+	int nCount = g_slist_length( pFileList );
 	int nIndex = 0;
 	OSL_TRACE( "GETFILES called %d files\n", nCount );
 
@@ -858,9 +858,12 @@
 	uno::Sequence< rtl::OUString > aSelectedFiles(nCount);
 
 	// Convert to OOo
-    for( GSList *pElem = pPathList; pElem; pElem = pElem->next)
+    for( GSList *pElem = pFileList; pElem; pElem = pElem->next)
 	{
-		gchar *pURI = reinterpret_cast<gchar*>(pElem->data);
+		GFile *pGFILE = reinterpret_cast<GFile*>(pElem->data);
+		gchar *pPATH = g_file_get_path(pGFILE);
+		gchar *pURI = g_filename_to_uri (pPATH, NULL, NULL);
+		g_free(pPATH);
 		aSelectedFiles[ nIndex ] = uritounicode(pURI);
 
 		if( GTK_FILE_CHOOSER_ACTION_SAVE == eAction )
@@ -962,7 +965,7 @@
 		g_free( pURI );
 	}
 
-	g_slist_free( pPathList );
+	g_slist_free( pFileList );
 
 	return aSelectedFiles;
 }
--- fpicker/source/unx/gnome/SalGtkFolderPicker.cxx.orig	2009-03-12 21:32:50.000000000 -0500
+++ fpicker/source/unx/gnome/SalGtkFolderPicker.cxx	2009-03-12 21:37:01.000000000 -0500
@@ -131,8 +131,10 @@
 	OSL_ASSERT( m_pDialog != NULL );
 	::vos::OGuard aGuard( Application::GetSolarMutex() );
 
-	gchar* pCurrentFolder =
-		gtk_file_chooser_get_current_folder_uri( GTK_FILE_CHOOSER( m_pDialog ) );
+	GFile* pGFILE = gtk_file_chooser_get_current_folder_file(GTK_FILE_CHOOSER(m_pDialog));
+	gchar* pPATH = g_file_get_path(pGFILE);
+	gchar* pCurrentFolder = g_filename_to_uri (pPATH, NULL, NULL);
+	g_free(pPATH);
 	::rtl::OUString aCurrentFolderName = uritounicode(pCurrentFolder);
 	g_free( pCurrentFolder );
 
--- fpicker/source/unx/gnome/SalGtkPicker.cxx.orig	2009-03-12 21:32:50.000000000 -0500
+++ fpicker/source/unx/gnome/SalGtkPicker.cxx	2009-03-12 21:37:13.000000000 -0500
@@ -222,8 +222,10 @@
 	OSL_ASSERT( m_pDialog != NULL );
 	::vos::OGuard aGuard( Application::GetSolarMutex() );
 	
-	gchar* pCurrentFolder = 
-		gtk_file_chooser_get_current_folder_uri( GTK_FILE_CHOOSER( m_pDialog ) );
+	GFile* pGFILE = gtk_file_chooser_get_current_folder_file(GTK_FILE_CHOOSER(m_pDialog));
+	gchar* pPATH = g_file_get_path(pGFILE);
+	gchar* pCurrentFolder = g_filename_to_uri (pPATH, NULL, NULL);
+	g_free(pPATH);
 	::rtl::OUString aCurrentFolderName = uritounicode(pCurrentFolder);
 	g_free( pCurrentFolder );
 
--- fpicker/source/unx/gnome/SalGtkPicker.hxx.orig	2009-03-12 21:32:50.000000000 -0500
+++ fpicker/source/unx/gnome/SalGtkPicker.hxx	2009-03-12 21:34:38.000000000 -0500
@@ -46,6 +46,8 @@
 #include <com/sun/star/awt/XTopWindowListener.hpp>
 #include <com/sun/star/awt/XExtendedToolkit.hpp>
 
+#include <glib.h>
+#include <gio/gio.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
 
