--- ucb/source/ucp/webdav/NeonPropFindRequest.cxx.old	2008-01-02 16:49:35.000000000 +0100
+++ ucb/source/ucp/webdav/NeonPropFindRequest.cxx	2008-01-15 13:59:06.000000000 +0100
@@ -176,11 +176,19 @@ extern "C" int NPFR_propfind_iter( void*
 
 // -------------------------------------------------------------------
 extern "C" void NPFR_propfind_results( void* userdata,
+#if NEON_VERSION >= 0260
                                        const ne_uri* uri,
+#else
+                                       const char* href,
+#endif
                                        const NeonPropFindResultSet* set )
 {
 	DAVResource theResource(
+#if NEON_VERSION >= 0260
 						OStringToOUString( uri->path, RTL_TEXTENCODING_UTF8 ) );
+#else
+						OStringToOUString( href, RTL_TEXTENCODING_UTF8 ) );
+#endif
 
 	ne_propset_iterate( set, NPFR_propfind_iter, &theResource );
 
@@ -208,14 +216,22 @@ extern "C" int NPFR_propnames_iter( void
 
 // -------------------------------------------------------------------
 extern "C" void NPFR_propnames_results( void* userdata,
+#if NEON_VERSION >= 0260
                                         const ne_uri* uri,
+#else
+                                        const char* href,
+#endif
                                         const NeonPropFindResultSet* results )
 {
 	// @@@ href is not the uri! DAVResourceInfo ctor wants uri!
 
 	// Create entry for the resource.
 	DAVResourceInfo theResource(
+#if NEON_VERSION >= 0260
 						OStringToOUString( uri->path, RTL_TEXTENCODING_UTF8 ) );
+#else
+						OStringToOUString( href, RTL_TEXTENCODING_UTF8 ) );
+#endif
 	// Fill entry.
     ne_propset_iterate( results, NPFR_propnames_iter, &theResource );
 
