--- ucb/source/ucp/webdav/webdavcontent.cxx	2007-07-27 15:43:02.000000000 +0200
+++ ucb/source/ucp/webdav/webdavcontent.cxx	2007-07-27 15:56:21.000000000 +0200
@@ -1414,6 +1414,18 @@ uno::Reference< sdbc::XRow > Content::ge
     uno::Reference< ucb::XContentIdentifier >    xIdentifier;
     rtl::Reference< ::ucbhelper::ContentProviderImplHelper > xProvider;
 
+    // webdav does not support active streaming, we can immediately return
+    if ( ( rProperties.getLength() == 1 ) &&
+            rProperties[ 0 ].Name.equalsAsciiL(
+                RTL_CONSTASCII_STRINGPARAM( "SupportsActiveStreaming" ) ) )
+    {
+        rtl::Reference< ::ucbhelper::PropertyValueSet > xRow
+            = new ::ucbhelper::PropertyValueSet( m_xSMgr );
+        xRow->appendBoolean( rProperties[0], sal_False );
+
+        return uno::Reference< sdbc::XRow >( xRow.get() );
+    }
+
     {
         osl::Guard< osl::Mutex > aGuard( m_aMutex );
 
