--- solenv/inc/unxlngi6.mk~	2006-09-04 16:51:53.010718000 +0200
+++ solenv/inc/unxlngi6.mk	2006-09-04 17:05:38.280718000 +0200
@@ -64,7 +64,7 @@
 
 # architecture dependent flags for the C and C++ compiler that can be changed by
 # exporting the variable ARCH_FLAGS="..." in the shell, which is used to start build
-ARCH_FLAGS*=-mtune=pentiumpro
+#ARCH_FLAGS*=-mtune=pentiumpro
 
 # name of C++ Compiler
 CXX*=g++
