--- vcl/source/glyphs/gcach_ftyp.cxx	2008-05-26 15:17:03.000000000 +0100
+++ vcl/source/glyphs/gcach_ftyp.cxx	2008-06-03 13:01:17.000000000 +0100
@@ -522,7 +522,10 @@
 void* FreetypeServerFont::GetFtFace() const
 {
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     return maFaceFT;
 }
@@ -742,6 +745,7 @@
     {
         pFTNewSize( maFaceFT, &maSizeFT );
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
     }
     FT_Error rc = FT_Set_Pixel_Sizes( maFaceFT, mnWidth, rFSD.mnHeight );
     if( rc != FT_Err_Ok )
@@ -975,7 +979,10 @@
     }
 
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     rFactor = 0x100;
 
@@ -1294,7 +1304,10 @@
 void FreetypeServerFont::InitGlyphData( int nGlyphIndex, GlyphData& rGD ) const
 {
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     int nGlyphFlags;
     SplitGlyphFlags( *this, nGlyphIndex, nGlyphFlags );
@@ -1382,7 +1395,10 @@
 bool FreetypeServerFont::GetGlyphBitmap1( int nGlyphIndex, RawBitmap& rRawBitmap ) const
 {
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     int nGlyphFlags;
     SplitGlyphFlags( *this, nGlyphIndex, nGlyphFlags );
@@ -1542,7 +1558,10 @@
 bool FreetypeServerFont::GetGlyphBitmap8( int nGlyphIndex, RawBitmap& rRawBitmap ) const
 {
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     int nGlyphFlags;
     SplitGlyphFlags( *this, nGlyphIndex, nGlyphFlags );
@@ -1796,7 +1815,10 @@
     // when font faces of different sizes share the same maFaceFT
     // then we have to make sure that it uses the correct maSizeFT
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     // use Freetype's kerning info
     FT_Vector aKernVal;
@@ -1831,7 +1853,10 @@
     // when font faces of different sizes share the same maFaceFT
     // then we have to make sure that it uses the correct maSizeFT
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+        FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     // first figure out which glyph pairs are involved in kerning
     ULONG nKernLength = 0;
@@ -2231,7 +2257,10 @@
     ::basegfx::B2DPolyPolygon& rB2DPolyPoly ) const
 {
     if( maSizeFT )
+    {
         pFTActivateSize( maSizeFT );
+	FT_Set_Transform( maFaceFT, NULL, NULL );
+    }
 
     rB2DPolyPoly.clear();
 
