diff -rup vcl/unx/source/app-orig/saldisp.cxx vcl/unx/source/app/saldisp.cxx
--- vcl/unx/source/app-orig/saldisp.cxx	2008-04-15 11:49:38.000000000 +0200
+++ vcl/unx/source/app/saldisp.cxx	2008-04-15 11:50:44.000000000 +0200
@@ -70,7 +70,7 @@
 
 #ifdef USE_XINERAMA
 #ifdef USE_XINERAMA_XORG
-#if defined(X86) || defined(MACOSX)
+#if defined(X86) || defined(X86_64) || defined(MACOSX)
 #include <X11/extensions/Xinerama.h>
 #endif
 #elif defined USE_XINERAMA_XSUN
@@ -2723,7 +2723,7 @@ void SalDisplay::InitXinerama()
         }
     }
 #elif defined(USE_XINERAMA_XORG)
-#if defined( X86 ) || defined( MACOSX )
+#if defined( X86 ) || defined( X86_64 ) || defined( MACOSX )
 if( XineramaIsActive( pDisp_ ) )
 {
     int nFramebuffers = 1;
