--- vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx.old	2008-06-16 16:30:38.000000000 +0200
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	2008-11-03 16:39:51.000000000 +0100
@@ -3180,22 +3180,6 @@ void GtkSalGraphics::updateSettings( All
     aStyleSet.SetHighlightColor( aHighlightColor );
     aStyleSet.SetHighlightTextColor( aHighlightTextColor );
 
-    // hyperlink colors
-    GdkColor *link_color = NULL;
-    gtk_widget_style_get (m_pWindow, "link-color", &link_color, NULL);
-    if (link_color)
-    {
-        aStyleSet.SetLinkColor(getColor(*link_color));
-        gdk_color_free (link_color);
-        link_color = NULL;
-    }
-    gtk_widget_style_get (m_pWindow, "visited-link-color", &link_color, NULL);
-    if (link_color)
-    {
-        aStyleSet.SetVisitedLinkColor(getColor(*link_color));
-        gdk_color_free (link_color);
-    }
-
     // Tab colors
     aStyleSet.SetActiveTabColor( aBackFieldColor ); // same as the window color.
     Color aSelectedBackColor = getColor( pStyle->bg[GTK_STATE_ACTIVE] );
