--- vcl/unx/gtk/window/gtkframe.cxx	25 Jun 2008 14:35:09 -0000	1.83
+++ vcl/unx/gtk/window/gtkframe.cxx	11 Jul 2008 17:07:48 -0000	1.83.2.1
@@ -582,6 +582,7 @@ void GtkSalFrame::InitCommon()
 
     //system data
     SalDisplay* pDisp = GetX11SalData()->GetDisplay();
+    m_aSystemData.nSize 		= sizeof( SystemChildData );
     m_aSystemData.pDisplay		= pDisp->GetDisplay();
     m_aSystemData.aWindow		= GDK_WINDOW_XWINDOW(m_pWindow->window);
     m_aSystemData.pSalFrame		= this;
--- vcl/unx/gtk/window/gtkobject.cxx	24 Jun 2008 11:42:57 -0000	1.14
+++ vcl/unx/gtk/window/gtkobject.cxx	11 Jul 2008 17:07:48 -0000	1.14.2.1
@@ -59,6 +59,7 @@ GtkSalObject::GtkSalObject( GtkSalFrame*
 
         //system data
         SalDisplay* pDisp = GetX11SalData()->GetDisplay();
+        m_aSystemData.nSize 		= sizeof( SystemChildData );
         m_aSystemData.pDisplay		= pDisp->GetDisplay();
         m_aSystemData.aWindow		= GDK_WINDOW_XWINDOW(m_pSocket->window);
         m_aSystemData.pSalFrame		= NULL;
