diff --git a/basic/source/app/process.cxx b/basic/source/app/process.cxx
index 6cb5b25..516816d 100644
--- basic/source/app/process.cxx
+++ basic/source/app/process.cxx
@@ -46,9 +46,6 @@
 #ifndef _ERRCODE_HXX //autogen
 #include <tools/errcode.hxx>
 #endif
-#ifndef _VOS_PROCESS_HXX_
-#include <vos/process.hxx>
-#endif
 #ifndef _SBXCORE_HXX
 #include <basic/sbxcore.hxx>
 #endif
diff --git a/extensions/test/sax/testsax.cxx b/extensions/test/sax/testsax.cxx
index 1b2d845..b3b70e3 100644
--- extensions/test/sax/testsax.cxx
+++ extensions/test/sax/testsax.cxx
@@ -42,7 +42,6 @@
 
 #include <rtl/wstring.hxx>
 #include <osl/time.h>
-//#include <vos/dynload.hxx>
 //#include <osl/diagnose.h>
 #include <usr/weak.hxx>
 #include <tools/string.hxx>
diff --git a/extensions/workben/pythonautotest.cxx b/extensions/workben/pythonautotest.cxx
index 950d28f..3f8c20d 100644
--- extensions/workben/pythonautotest.cxx
+++ extensions/workben/pythonautotest.cxx
@@ -41,7 +41,6 @@
 #include <stardiv/uno/beans/exactname.hxx>
 
 #include <rtl/ustring.hxx>
-#include <vos/dynload.hxx>
 #include <osl/diagnose.h>
 #include <usr/services.hxx>
 #include <vcl/svapp.hxx>
diff --git a/extensions/workben/pythontest.cxx b/extensions/workben/pythontest.cxx
index 2c8d894..26905e9 100644
--- extensions/workben/pythontest.cxx
+++ extensions/workben/pythontest.cxx
@@ -42,7 +42,6 @@
 #include <stardiv/uno/beans/exactname.hxx>
 
 #include <rtl/ustring.hxx>
-#include <vos/dynload.hxx>
 #include <osl/diagnose.h>
 #include <usr/services.hxx>
 #include <vcl/svapp.hxx>
diff --git a/extensions/workben/testcomponent.cxx b/extensions/workben/testcomponent.cxx
index cd49e03..9a586a3 100644
--- extensions/workben/testcomponent.cxx
+++ extensions/workben/testcomponent.cxx
@@ -49,7 +49,6 @@
 #include <smart/com/sun/star/lang/XComponent.hxx>
 //#include <com/sun/star/registry/ stardiv/uno/repos/simplreg.hxx>
 
-#include <vos/dynload.hxx>
 #include <osl/diagnose.h>
 #include <usr/services.hxx>
 #include <vcl/svapp.hxx>
diff --git a/extensions/workben/testframecontrol.cxx b/extensions/workben/testframecontrol.cxx
index dfbb793..9c79532 100644
--- extensions/workben/testframecontrol.cxx
+++ extensions/workben/testframecontrol.cxx
@@ -37,7 +37,6 @@
 #include "precompiled_extensions.hxx"
 
 //#include <osl/mutex.hxx>
-#include <vos/dynload.hxx>
 
 #include <vcl/wrkwin.hxx>
 #include <vcl/svapp.hxx>
diff --git a/extensions/workben/testresource.cxx b/extensions/workben/testresource.cxx
index 0aed67a..8725e9b 100644
--- extensions/workben/testresource.cxx
+++ extensions/workben/testresource.cxx
@@ -40,7 +40,6 @@
 #include <smart/com/sun/star/script/XInvocation.hxx>
 
 #include <rtl/ustring.hxx>
-#include <vos/dynload.hxx>
 #include <osl/diagnose.h>
 #include <usr/services.hxx>
 #include <vcl/svapp.hxx>
diff --git a/sax/test/saxdemo.cxx b/sax/test/saxdemo.cxx
index 9d73d6e..acca666 100644
--- sax/test/saxdemo.cxx
+++ sax/test/saxdemo.cxx
@@ -58,7 +58,6 @@
 #include <cppuhelper/implbase1.hxx>
 #include <cppuhelper/implbase3.hxx>
 
-#include <vos/dynload.hxx>
 #include <osl/diagnose.h>
 
 using namespace ::rtl;
diff --git a/sax/test/testcomponent.cxx b/sax/test/testcomponent.cxx
index 1fa63f3..49da513 100644
--- sax/test/testcomponent.cxx
+++ sax/test/testcomponent.cxx
@@ -49,7 +49,6 @@
 
 #include <cppuhelper/servicefactory.hxx>
 
-#include <vos/dynload.hxx>
 #include <osl/diagnose.h>
 
 using namespace ::rtl;
diff --git a/sc/inc/pch/precompiled_sc.hxx b/sc/inc/pch/precompiled_sc.hxx
index bbef9da..607220b 100644
--- sc/inc/pch/precompiled_sc.hxx
+++ sc/inc/pch/precompiled_sc.hxx
@@ -252,8 +252,6 @@
 #include <vcl/window.hxx>
 #include <vcl/wintypes.hxx>
 #include <sal/macros.h>
-#include <vos/object.hxx>
-#include <vos/types.hxx>
 #include <wchar.h>
 
 #endif
diff --git a/sc/source/filter/inc/filt_pch.hxx b/sc/source/filter/inc/filt_pch.hxx
index d31ae05..df03b56 100644
--- sc/source/filter/inc/filt_pch.hxx
+++ sc/source/filter/inc/filt_pch.hxx
@@ -171,8 +171,6 @@
 #include <so3/so2ref.hxx>
 #include <so3/so2dll.hxx>
 #include <svx/fhgtitem.hxx>
-#include <vos/types.hxx>
-#include <vos/object.hxx>
 #include <sal/macros.h>
 #include <tools/unqid.hxx>
 #include <com/sun/star/uno/Sequence.h>
@@ -238,7 +236,6 @@
 #include <com/sun/star/container/NoSuchElementException.hpp>
 #include <vcl/svapp.hxx>
 #include <osl/thread.hxx>
-#include <vos/runnable.hxx>
 #include <salhelper/simplereferenceobject.hxx>
 #include <vcl/apptypes.hxx>
 #include <svx/editdata.hxx>
diff --git a/starmath/source/math_pch.cxx b/starmath/source/math_pch.cxx
index dd32c2e..f4e69a9 100644
--- starmath/source/math_pch.cxx
+++ starmath/source/math_pch.cxx
@@ -126,10 +126,8 @@
 #include <usr/xiface.hxx>
 #include <usr/ustring.hxx>
 #include <salhelper/simplereferenceobject.hxx>
-#include <vos/types.hxx>
 #include <osl/types.h>
 #include <osl/interlck.h>
-#include <vos/object.hxx>
 #include <sal/macros.h>
 #include <smdll.hxx>
 #include <sfx2/sfxsids.hrc>
@@ -152,7 +150,6 @@
 #include <smmod.hxx>
 #include <osl/thread.hxx>
 #include <osl/thread.h>
-#include <vos/runnable.hxx>
 #include <vcl/apptypes.hxx>
 #include <tools/dynary.hxx>
 #include <vcl/metric.hxx>
diff --git a/stoc/test/testintrosp.cxx b/stoc/test/testintrosp.cxx
index 396c944..5be43aa 100644
--- stoc/test/testintrosp.cxx
+++ stoc/test/testintrosp.cxx
@@ -45,8 +45,6 @@
 #include <osl/diagnose.h>
 #endif
 
-//#include <vos/dynload.hxx>
-
 #include <ModuleA/XIntroTest.hpp>
 #include <com/sun/star/beans/XPropertySet.hpp>
 #include <com/sun/star/beans/XIntrospection.hpp>
diff --git a/vos/inc/vos/execabl.hxx b/vos/inc/vos/execabl.hxx
index f229f09..bfb7604 100644
--- vos/inc/vos/execabl.hxx
+++ vos/inc/vos/execabl.hxx
@@ -37,9 +37,6 @@
 #ifndef _VOS_EXECABL_HXX_
 #define _VOS_EXECABL_HXX_
 
-#ifndef _VOS_TYPES_HXX_
-#	include <vos/types.hxx>
-#endif
 #ifndef _VOS_REFERNCE_HXX_
 #	include <vos/refernce.hxx>
 #endif
diff --git a/vos/inc/vos/istream.hxx b/vos/inc/vos/istream.hxx
index ab0b719..1fd78ec 100644
--- vos/inc/vos/istream.hxx
+++ vos/inc/vos/istream.hxx
@@ -40,10 +40,6 @@
 #include "sal/types.h"
 #endif
 
-#ifndef _VOS_TYPES_HXX_
-#	include	<vos/types.hxx>
-#endif
-
 namespace vos
 {     
 
diff --git a/vos/inc/vos/object.hxx b/vos/inc/vos/object.hxx
index 5efb555..340900a 100644
--- vos/inc/vos/object.hxx
+++ vos/inc/vos/object.hxx
@@ -37,9 +37,6 @@
 #ifndef _VOS_OBJECT_HXX_
 #define _VOS_OBJECT_HXX_
 
-#ifndef _VOS_TYPES_HXX_
-#	include <vos/types.hxx>
-#endif
 #ifndef _VOS_MACROS_HXX_
 #	include <vos/macros.hxx>
 #endif
diff --git a/vos/inc/vos/pipe.hxx b/vos/inc/vos/pipe.hxx
index 3067a42..2982ae2 100644
--- vos/inc/vos/pipe.hxx
+++ vos/inc/vos/pipe.hxx
@@ -40,9 +40,6 @@
 #ifndef _OSL_PIPE_H_
 #   include <osl/pipe.h>
 #endif
-#ifndef _VOS_TYPES_HXX_
-#   include <vos/types.hxx>
-#endif
 #ifndef _VOS_OBJECT_HXX_
 #   include <vos/object.hxx>
 #endif
diff --git a/vos/inc/vos/process.hxx b/vos/inc/vos/process.hxx
deleted file mode 100644
index eb59c0c..0000000
--- vos/inc/vos/process.hxx
+++ /dev/null
@@ -1,333 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: process.hxx,v $
- *
- *  $Revision: 1.10 $
- *
- *  last change: $Author: hr $ $Date: 2006/06/20 04:05:29 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-#ifndef _VOS_PROCESS_HXX_
-#define _VOS_PROCESS_HXX_
-
-#ifndef _RTL_USTRING_HXX_
-#   include <rtl/ustring.hxx>
-#endif
-#ifndef _OSL_SECURITY_HXX_
-#   include <osl/security.hxx>
-#endif
-#ifndef _VOS_OBJECT_HXX_
-#   include <vos/object.hxx>
-#endif
-#ifndef _OSL_PROCESS_H_
-#   include <osl/process.h>
-#endif
-
-namespace vos
-{
-
-class OProcess;
-
-/** helper class to fill a vector of command line arguments
- */
-class OArgumentList
-{
-    sal_uInt32 n_Args;
-    rtl_uString** m_aVec;
-
-public:
-
-    OArgumentList();
-    OArgumentList( sal_uInt32 nArgs, const ::rtl::OUString* aArgument1, ... );
-    // switched argument list to avoid ambiguity with previous constructor.
-    OArgumentList( const ::rtl::OUString aArgumentList[], sal_uInt32 nArgs );
-
-    OArgumentList( const OArgumentList& rOther);
-
-    OArgumentList& operator=( const OArgumentList& rOther);
-
-    virtual ~OArgumentList();
-
-    friend class OProcess;
-};
-
-/** helper class to fill a vector of environment settings
- */
-class OEnvironment
-{
-    sal_Int32 n_Vars;
-    rtl_uString** m_aVec;
-
-public:
-
-    OEnvironment();
-    OEnvironment( sal_Int32 nVars, const ::rtl::OUString* aVariable1, ... );
-    // switched argument list to avoid ambiguity with previous constructor.
-    OEnvironment( const ::rtl::OUString aVariableList[], sal_Int32 nVars );
-
-    OEnvironment( const OEnvironment& rOther );
-
-    OEnvironment& operator=( const OEnvironment& rOther );
-
-    virtual ~OEnvironment();
-
-    friend class OProcess;
-};
-
-
-/** startup child processes.
-    @see OStartupInfo
-    Used for starting an monitoring child processes with special features:
-    <ul><li>setting enviroments,
-    <li>setting working directories,
-    <li>setting user rights and security,
-    <li>providing ioresources like file descriptors and sockets.</ul>
-*/
-class OProcess : public OObject
-{
-    VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OProcess, vos));
-
-public:
-
-    /** Options for execution mode:
-    */
-    enum TProcessOption
-    {
-        TOption_Wait       = osl_Process_WAIT,       // wait for completion
-        TOption_SearchPath = osl_Process_SEARCHPATH, // search path for executable
-        TOption_Detached   = osl_Process_DETACHED,   // run detached
-        TOption_Normal     = osl_Process_NORMAL,     // run in normal window
-        TOption_Hidden     = osl_Process_HIDDEN,     // run hidden
-        TOption_Minimized  = osl_Process_MINIMIZED,  // run in minimized window
-        TOption_Maximized  = osl_Process_MAXIMIZED,  // run in maximized window
-        TOption_FullScreen = osl_Process_FULLSCREEN  // run in fullscreen window
-    };
-
-    /** Errorcodes:
-    */
-    enum TProcessError {
-        E_None         = osl_Process_E_None,            /* no error */
-        E_NotFound     = osl_Process_E_NotFound,        /* image not found */
-        E_TimedOut     = osl_Process_E_TimedOut,        /* timout occured */
-        E_NoPermission = osl_Process_E_NoPermission,    /* permission denied */
-        E_Unknown      = osl_Process_E_Unknown,         /* unknown error */
-        E_InvalidError = osl_Process_E_InvalidError     /* unmapped error */
-    };
-
-    enum TDescriptorFlags
-    {
-        TFlags_None = osl_Process_DFNONE,
-        TFlags_Wait = osl_Process_DFWAIT
-    };
-
-    enum TProcessData
-    {
-        TData_Identifier = osl_Process_IDENTIFIER,
-        TData_ExitCode   = osl_Process_EXITCODE,
-        TData_CpuTimes   = osl_Process_CPUTIMES,
-        TData_HeapUsage  = osl_Process_HEAPUSAGE
-    };
-
-    struct TProcessInfo : public oslProcessInfo
-    {
-        TProcessInfo() { Size = sizeof(*this); }
-    };
-
-    typedef oslProcessIdentifier TProcessIdentifier;
-
-    /** Creating a process object by naming the executable.
-        Does not yet start the process.
-        @see execute
-    */
-
-    OProcess( );
-
-    OProcess(const ::rtl::OUString& strImageName);
-
-    OProcess(const ::rtl::OUString& strImageName,
-             const ::rtl::OUString& strWorkingDirectory);
-
-    /// destroying a process object
-    virtual ~OProcess();
-
-    SAL_CALL operator oslProcess()
-        { return m_Process; }
-
-    SAL_CALL operator oslProcess() const
-        { return m_Process; }
-
-    static OProcess* SAL_CALL getProcess(TProcessIdentifier Identifier);
-
-    /** execute the given process.
-        This process becomes a child of the caller.
-        If there are any ioresources provided from the calling process, this
-        function returns only, if the child process calls OStartupInfo::acceptIOResource().
-        @param Options [in] describes the execution mode.
-        @return only not eNONE, if too much enviroments are added.
-        @see OStartupInfo::acceptIOResource
-    */
-    TProcessError SAL_CALL execute(TProcessOption Options,
-                          const OArgumentList& aArgumentList = OArgumentList(),
-                          const OEnvironment&  aEnvironment  = OEnvironment()
-                          );
-
-    /** execute the given process with the specified security.
-        This process becomes a child of the caller.
-        The process is executed with the rights of the user, for whom the security object is created.
-        If there are any ioresources provided from the calling process, this
-        function returns only, if the child process calls OStartupInfo::acceptIOResource().
-        @param Options [in] describes the execution mode.
-        @param Security [in] is a given security object for one logged in user.
-        @return eNONE, if the proccess could be executed, otherwise an errorcode.
-        @see OStartupInfo::acceptIOResource
-    */
-    TProcessError SAL_CALL execute(TProcessOption Options,
-                          const osl::Security &Security,
-                          const OArgumentList& aArgumentList = OArgumentList(),
-                          const OEnvironment&  aEnvironment  = OEnvironment()
-                         );
-
-    TProcessError SAL_CALL terminate();
-
-    TProcessError SAL_CALL getInfo(TProcessData Data, TProcessInfo* pInfo) const;
-
-    static TProcessError SAL_CALL getCurrentInfo(TProcessData Data, TProcessInfo* pInfo);
-
-    /** wait for the completation of this child process
-        @return eNONE if child process exits, otherwise nothing.
-    */
-    TProcessError SAL_CALL join();
-
-protected:
-    const ::rtl::OUString m_strImageName;
-    const ::rtl::OUString m_strDirectory;
-
-    oslProcess         m_Process;
-};
-
-/** informations for client processes provided by the parent.
-    @see OProcess
-*/
-
-
-class OStartupInfo : public OObject
-{
-    VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OStartupInfo, vos));
-
-public:
-    /** Errorcodes:
-    */
-    enum TStartupError {
-        E_None         = osl_Process_E_None,            /* no error */
-        E_NotFound     = osl_Process_E_NotFound,        /* image not found */
-        E_TimedOut     = osl_Process_E_TimedOut,        /* timout occured */
-        E_NoPermission = osl_Process_E_NoPermission,    /* permission denied */
-        E_Unknown      = osl_Process_E_Unknown,         /* unknown error */
-        E_InvalidError = osl_Process_E_InvalidError     /* unmapped error */
-    };
-
-    /** Constructor.
-    */
-    OStartupInfo();
-
-    /** Destructor
-    */
-    ~OStartupInfo();
-
-    /** @return the number of command line arguments.
-     */
-    sal_uInt32 SAL_CALL getCommandArgCount();
-
-    /** get the nArg-th command argument passed to the main-function of this process.
-        @param nArg [in] the number of arguments to return.
-        @param strCommandArg [out] the string that receives the argument.
-        @return eNONE
-    */
-    TStartupError SAL_CALL getCommandArg(sal_uInt32 nArg, ::rtl::OUString& strCommandArg);
-
-    TStartupError SAL_CALL getExecutableFile(::rtl::OUString& strImageName)
-        const;
-
-    /** Get the value of one enviroment variable.
-        @param Name [in] denotes the name of the variable to get.
-        @param Buffer [out] is the buffer where the value of this variable is returned.
-        @param Max [in] is the size of this buffer.
-        @return eNONE, if the variable exist in the enviroment, otherwise False.
-    */
-    TStartupError SAL_CALL getEnvironment(const ::rtl::OUString& strVar, ::rtl::OUString& strValue);
-};
-
-
-
-/** get extended arguments from command line and an argument file
-    the file name is given on the command line as "@filename"
-    (filename must be in our UNC notation!)
-    enumeration starts with 0 (i.e. argv[1])
-    each line in the file will be treated as one argument
-    @see also OProcess and OStartupInfo
-*/
-
-class OExtCommandLineImpl;
-
-class OExtCommandLine : public OObject
-{
-    VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OExtCommandLine, vos));
-    static vos::OExtCommandLineImpl* pExtImpl;
-
-public:
-
-    /** Constructor.
-    */
-    OExtCommandLine();
-
-    /** Destructor
-    */
-    virtual ~OExtCommandLine();
-
-    /** @return the number of extended command line arguments.
-     */
-    sal_uInt32 SAL_CALL getCommandArgCount();
-
-
-    /** get the nArg-th extended command argument
-        @param nArg [in] the number of extended argument to return.
-        @param strCommandArg [out] the string that receives the argument.
-        @return sal_True   if the nArg-th argument has been retriveded successfully
-        @return sal_False  on all other cases
-    */
-    sal_Bool SAL_CALL getCommandArg(sal_uInt32 nArg, ::rtl::OUString& strCommandArg);
-
-
-};
-
-}
-
-#endif  // _VOS_PROCESS_HXX_
-
-
diff --git a/vos/inc/vos/refernce.hxx b/vos/inc/vos/refernce.hxx
index 6256f6e..46e695f 100644
--- vos/inc/vos/refernce.hxx
+++ vos/inc/vos/refernce.hxx
@@ -36,9 +36,6 @@
 #ifndef _VOS_REFERNCE_HXX_
 #define _VOS_REFERNCE_HXX_
 
-#ifndef _VOS_TYPES_HXX_
-#	include <vos/types.hxx>
-#endif
 #ifndef _OSL_INTERLCK_H_
 #	include <osl/interlck.h>
 #endif
diff --git a/vos/inc/vos/runnable.hxx b/vos/inc/vos/runnable.hxx
deleted file mode 100644
index 7b1533b..0000000
--- vos/inc/vos/runnable.hxx
+++ /dev/null
@@ -1,79 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: runnable.hxx,v $
- *
- *  $Revision: 1.3 $
- *
- *  last change: $Author: vg $ $Date: 2006/06/02 12:39:03 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-#ifndef _VOS_RUNNABLE_HXX_
-#define _VOS_RUNNABLE_HXX_
-
-#ifndef _VOS_TYPES_HXX_
-#	include <vos/types.hxx>
-#endif
-#ifndef _VOS_REFERNCE_HXX_
-#	include <vos/refernce.hxx>
-#endif
-
-namespace vos
-{     
-
-
-/** IRunnable
-
-	The IRunnable-interface is to be understood as wrapper around
-	your application-code, which allows it to be executed asynchronously.
-
-
-	@author  Bernd Hofner
-	@version 0.1
-*/
-
-class IRunnable
-{
-public:
-
-    IRunnable() { }
-    virtual ~IRunnable() { }
-    
-    
-	/** overload to implement your functionality.
-	*/
-	virtual void SAL_CALL run()= 0;
-
-};
-
-
-}
-
-
-#endif  // _VOS_RUNNABLE_HXX_
-
diff --git a/vos/inc/vos/stream.hxx b/vos/inc/vos/stream.hxx
index 096f075..c617cf2 100644
--- vos/inc/vos/stream.hxx
+++ vos/inc/vos/stream.hxx
@@ -36,9 +36,6 @@
 #ifndef _VOS_STREAM_HXX_
 #define _VOS_STREAM_HXX_
 
-#ifndef _VOS_TYPES_HXX_
-#	include <vos/types.hxx>
-#endif
 #ifndef _VOS_OBJECT_HXX_
 #	include <vos/object.hxx>
 #endif
diff --git a/vos/inc/vos/thread.hxx b/vos/inc/vos/thread.hxx
deleted file mode 100644
index 57fccb9..0000000
--- vos/inc/vos/thread.hxx
+++ /dev/null
@@ -1,255 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: thread.hxx,v $
- *
- *  $Revision: 1.7 $
- *
- *  last change: $Author: hr $ $Date: 2006/06/20 11:18:52 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-#ifndef _VOS_THREAD_HXX_
-#define _VOS_THREAD_HXX_
-
-#ifndef _VOS_TYPES_HXX_
-#	include <vos/types.hxx>
-#endif
-#ifndef _VOS_OBJECT_HXX_
-#	include <vos/object.hxx>
-#endif
-#ifndef _OSL_THREAD_H_
-#	include <osl/thread.h>
-#endif
-#ifndef _OSL_CONDITION_H_
-#	include <osl/conditn.h>
-#endif
-#ifndef _VOS_RUNNABLE_HXX_
-#	include <vos/runnable.hxx>
-#endif
-
-#ifndef _OSL_TIME_H_
-#include <osl/time.h>
-#endif
-
-namespace vos
-{
-
-extern "C" typedef void ThreadWorkerFunction_impl(void *);
-ThreadWorkerFunction_impl threadWorkerFunction_impl;
-
-/** OThread is an objectoriented interface for threads.
-	This class should be the base class for all objects using threads. The
-	main working function is the run() method and should be overriden in the
-	derived class. To support soft termination of a thread, yield() should
-	be called in regular intervalls and the return value should be checked.
-	If yield returned False the run method should return.
-
-	@author  Bernd Hofner
-	@version 1.0
-*/
-
-class OThread : public vos::IRunnable,
-				public vos::OObject
-{
-
-	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OThread, vos));
-    
-    oslCondition m_aCondition;
-
-public:
-	/** priority of thread.
-	*/
-	enum TThreadPriority
-	{
-		TPriority_Highest     = osl_Thread_PriorityHighest,
-		TPriority_AboveNormal = osl_Thread_PriorityAboveNormal,
-		TPriority_Normal      = osl_Thread_PriorityNormal,
-		TPriority_BelowNormal = osl_Thread_PriorityBelowNormal,
-		TPriority_Lowest      = osl_Thread_PriorityLowest,
-		TPriority_Unknown     = osl_Thread_PriorityUnknown
-	};
-
-	/** 
-	*/
-	enum TThreadSleep 
-	{   
-		TSleep_Normal,
-		TSleep_Cancel,
-		TSleep_Pending,
-		TSleep_Active,
-		TSleep_Error,
-		TSleep_Unknown	
-	};
-
-	typedef oslThreadIdentifier TThreadIdentifier;
-
-	/// Constructor 
-	OThread();              
-
-	/// Destructor kills thread if neccessary
-	virtual ~OThread();
-
-	/** Create running instance of a thread.
-		@returns True if thread could be created.
-	*/
-	sal_Bool SAL_CALL create();
-
-	/** Create suspended instance of a thread.
-		@returns True if thread could be created.
-	*/
-	sal_Bool SAL_CALL createSuspended();
-
-	/// Suspend a runnng thread
-	void SAL_CALL suspend();
-
-	/// Resume a suspended thread
-	void SAL_CALL resume();
-
-	/** Tries to kill the thread.
-		will not block and might not succeed when run() won't heed isTerminationRequested().
-	*/
-	virtual void SAL_CALL terminate();
-
-	/// Kill thread hard and block until it is actually gone
-	virtual void SAL_CALL kill();
-
-	/// Block till thread is terminated
-	void SAL_CALL join();
-
-	/// Check if thread is running.
-	sal_Bool SAL_CALL isRunning();
-
-	/** Change thread priority. 
-		The valid priority levels are:
-	<ul>
-		<li>ThreadPriorityHighest,
-		<li>ThreadPriorityAboveNormal,
-		<li>ThreadPriorityNormal,
-		<li>ThreadPriorityBelowNormal,
-		<li>ThreadPriorityLowest,
-	</ul>
-	*/
-	void SAL_CALL setPriority(TThreadPriority Priority);
-
-	/** Query thread priority. 
-		Valid return values are:
-	<ul>
-		<li>ThreadPriorityHighest,
-		<li>ThreadPriorityAboveNormal,
-		<li>ThreadPriorityNormal,
-		<li>ThreadPriorityBelowNormal,
-		<li>ThreadPriorityLowest,
-		<li>ThreadPriorityUnknown (returned if thread is killed)
-	</ul>
-	*/
-	TThreadPriority SAL_CALL getPriority();
-
-	TThreadIdentifier SAL_CALL getIdentifier() const;
-
-	static TThreadIdentifier SAL_CALL getCurrentIdentifier();
-
-	/** Let thread sleep a specified amout of time.
-		@param Delay specifies the number of time to sleep.
-	*/
-	TThreadSleep SAL_CALL sleep(const TimeValue& Delay);
-
-	/** Awake the sleeping thread. 
-		@returns False if at least one of the handles is invalid 
-		or the thread is not sleeping. 
-	*/
-	sal_Bool SAL_CALL awake();
-
-	/** Let current thread wait a specified amout of time.
-		@param Delay specifies the number of time
-		to wait. Note, if you need to interrupt the waiting operation
-		use sleep instead.
-	*/
-	static void SAL_CALL wait(const TimeValue& Delay);
-
-	/** Reschedules threads. 
-		Call within your loop if you
-		want other threads offer some processing time.
-		This method is static, so it might be used by the
-		main-thread.
-	*/
-	static void SAL_CALL yield();
-
-protected:
-
-	/// Working method which should be overridden.
-	virtual void SAL_CALL run() = 0;
-
-	/** Checks if thread should terminate.
-		isTerminationRequested() will return True if someone called
-		terminate().
-		@return True if thread should terminate, False if he can continue.
-	*/
-	virtual sal_Bool SAL_CALL schedule();
-
-	/** Called when run() is done. 
-		You might want to override it to do some cleanup.
-	*/
-	virtual void SAL_CALL onTerminated();
-
-protected:
-	oslThread m_hThread;
-	sal_Bool   m_bTerminating;
-
-	friend void threadWorkerFunction_impl(void *);
-};
-
-class OThreadData : public vos::OObject
-{
-	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OThreadData, vos));
-
-public:
-	/// Create a thread specific local data key 
-	OThreadData( oslThreadKeyCallbackFunction = 0 );              
-
-	/// Destroy a thread specific local data key 
-	virtual ~OThreadData();
-
-	/** Set the data associated with the data key.
-		@returns True if operation was successfull
-	*/
-	sal_Bool SAL_CALL setData(void *pData);
-
-	/** Get the data associated with the data key.
-		@returns The data asscoitaed with the data key or 
-		NULL if no data was set
-	*/
-	void* SAL_CALL getData();
-
-protected:
-	oslThreadKey m_hKey;
-};
-
-}
-
-#endif // _VOS_THREAD_HXX_
-
diff --git a/vos/inc/vos/types.hxx b/vos/inc/vos/types.hxx
deleted file mode 100644
index c66a226..0000000
--- vos/inc/vos/types.hxx
+++ /dev/null
@@ -1,48 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: types.hxx,v $
- *
- *  $Revision: 1.3 $
- *
- *  last change: $Author: rt $ $Date: 2005/09/09 09:05:37 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-
-#ifndef _VOS_TYPES_HXX_
-#define _VOS_TYPES_HXX_
-
-
-/* disable some warnings for MS-C */
-#ifdef _MSC_VER
-#	pragma warning (disable : 4786) // 4786: truncated names (longer 255 chars)
-#	pragma warning (disable : 4355)	// 4355: this used in initializer-list
-#endif
-
-#endif //_VOS_TYPES_HXX_
-
diff --git a/vos/source/makefile.mk b/vos/source/makefile.mk
index b258071..2c02290 100644
--- vos/source/makefile.mk
+++ vos/source/makefile.mk
@@ -48,10 +48,8 @@ ENABLE_EXCEPTIONS=TRUE
 SLOFILES= \
 				$(SLO)$/object.obj      \
 				$(SLO)$/refernce.obj    \
-				$(SLO)$/thread.obj      \
 				$(SLO)$/stream.obj      \
 				$(SLO)$/timer.obj       \
-				$(SLO)$/process.obj     \
 				$(SLO)$/pipe.obj
 
 
@@ -59,10 +57,8 @@ SLOFILES= \
 OBJFILES= \
 				$(OBJ)$/object.obj      \
 				$(OBJ)$/refernce.obj    \
-				$(OBJ)$/thread.obj      \
 				$(OBJ)$/stream.obj      \
 				$(OBJ)$/timer.obj       \
-				$(OBJ)$/process.obj     \
 				$(OBJ)$/pipe.obj
 
 .ENDIF
diff --git a/vos/source/process.cxx b/vos/source/process.cxx
deleted file mode 100644
index 3872bbc..0000000
--- vos/source/process.cxx
+++ /dev/null
@@ -1,536 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: process.cxx,v $
- *
- *  $Revision: 1.13 $
- *
- *  last change: $Author: hr $ $Date: 2006/06/20 04:07:14 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-
-#include <cstdarg>
-#include <vector>
-#include <rtl/ustring.hxx>
-#ifndef INCLUDED_RTL_INSTANCE_HXX
-#include <rtl/instance.hxx>
-#endif
-
-#include "vos/process.hxx"
-#include "osl/diagnose.h"
-#include <osl/file.hxx>
-#include <osl/mutex.hxx>
-
-#define MAX_RESOURCES	100
-#define MAX_ARGS		100
-#define MAX_ENVIROMENTS 100
-
-using namespace vos;
-
-/////////////////////////////////////////////////////////////////////////////
-/// Argument
-
-OArgumentList::OArgumentList() :
-	n_Args(0),
-	m_aVec(0)
-{
-	// empty 
-}
-
-OArgumentList::OArgumentList( sal_uInt32 nArgs, const ::rtl::OUString* aArgument1, ... ) :
-	n_Args( nArgs )
-{
-	m_aVec = new rtl_uString* [nArgs];
-	std::va_list pArgs;
-	sal_uInt32 i = 0;
-	const rtl::OUString* aArgument;
-	
-	va_start ( pArgs, aArgument1 );
-	aArgument = aArgument1;
-
-    while( true ) {
-		m_aVec[i] = aArgument->pData;
-		rtl_uString_acquire( m_aVec[i++] );
-		if( i < n_Args )
-			aArgument = va_arg( pArgs,rtl::OUString* );
-		else
-			break;
-	}
-	va_end( pArgs );
-}
-
-
-OArgumentList::OArgumentList( const rtl::OUString aArgumentList[], sal_uInt32 nArgs ) :
-	n_Args( nArgs )
-{
-	m_aVec = new rtl_uString* [n_Args];
-	for( sal_uInt32 i = 0; i < n_Args; ++ i )  {
-		m_aVec[i] = aArgumentList[i].pData;
-		rtl_uString_acquire( m_aVec[i] );
-	}
-}
-
-OArgumentList::OArgumentList( const OArgumentList& rOther ) : n_Args( rOther.n_Args )
-{
-	m_aVec = new rtl_uString* [n_Args];
-	
-	sal_uInt32 i;
-	for ( i = 0; i < n_Args; ++i )
-	{		
-		m_aVec[i] = rOther.m_aVec[i];
-		rtl_uString_acquire( m_aVec[i] );
-	}
-}
-
-OArgumentList& OArgumentList::operator=( const OArgumentList& rOther )
-{
-	if ( this != &rOther )
-	{
-
-		// delete the old one
-		sal_uInt32 i;
-		for ( i = 0; i < n_Args; ++i )
-			rtl_uString_release( m_aVec[i] );
-
-		delete [] m_aVec;
-
-
-		n_Args = rOther.n_Args;
-		m_aVec = new rtl_uString* [n_Args];
-		for( i = 0; i < n_Args; ++i )
-		{
-			m_aVec[i] = rOther.m_aVec[i];
-			rtl_uString_acquire( m_aVec[i] );
-		}
-	}
-
-	return *this;
-}
-
-OArgumentList::~OArgumentList( )
-{
-	for( sal_uInt32 i = 0; i < n_Args; ++i ) rtl_uString_release( m_aVec[i] );
-	delete[] m_aVec;
-}
-
-
-////////////////////////////////////////////////////////////////////////////////
-/// Environment
-
-OEnvironment::OEnvironment() :
-	n_Vars( 0 ),
-	m_aVec( 0 )
-{
-}
-
-OEnvironment::OEnvironment( sal_Int32 nVars, const ::rtl::OUString* aArgument1, ... ) :
-	n_Vars( nVars )
-{
-	m_aVec = new rtl_uString* [nVars];
-	std::va_list pArgs;
-	sal_Int32 i = 0;
-	const rtl::OUString* aArgument;
-	
-	va_start ( pArgs, aArgument1 );
-	aArgument = aArgument1;
-
-    while( true ) {
-		m_aVec[i] = aArgument->pData;
-		rtl_uString_acquire( m_aVec[i++] );
-		if( i < n_Vars )
-			aArgument = va_arg( pArgs,rtl::OUString* );
-		else
-			break;
-	}
-	va_end( pArgs );
-}
-
-
-OEnvironment::OEnvironment( const ::rtl::OUString aVariableList[], sal_Int32 nVars ) :
-	n_Vars( nVars )
-{
-	m_aVec = new rtl_uString* [n_Vars];
-	for( sal_Int32 i = 0; i < n_Vars; ++ i )  {
-		m_aVec[i] = aVariableList[i].pData;
-		rtl_uString_acquire( m_aVec[i] );
-	}	
-}
-
-OEnvironment::OEnvironment( const OEnvironment& rOther ) : n_Vars( rOther.n_Vars )
-{
-	m_aVec = new rtl_uString* [n_Vars];
-	
-	sal_Int32 i;
-	for ( i = 0; i < n_Vars; ++i )
-	{
-		m_aVec[i] = rOther.m_aVec[i];
-		rtl_uString_acquire( m_aVec[i] );
-	}
-}
-
-OEnvironment& OEnvironment::operator=( const OEnvironment& rOther )
-{
-	if ( this != &rOther )
-	{
-		sal_Int32 i;
-		for ( i = 0; i < n_Vars; ++i )
-			rtl_uString_release( m_aVec[i] );
-
-		delete [] m_aVec;
-
-		n_Vars = rOther.n_Vars;
-		m_aVec = new rtl_uString* [n_Vars];
-		for ( i = 0; i < n_Vars; ++i ) 
-		{
-			m_aVec[i] = rOther.m_aVec[i];
-			rtl_uString_acquire( m_aVec[i] );
-		}
-	}
-
-	return *this;
-}
-
-OEnvironment::~OEnvironment()
-{
-	for( sal_Int32 i = 0; i < n_Vars; ++i ) rtl_uString_release( m_aVec[i] );
-	delete[] m_aVec;
-}
-
-/////////////////////////////////////////////////////////////////////////////
-// Process
-
-
-VOS_IMPLEMENT_CLASSINFO(
-	VOS_CLASSNAME(OProcess, vos),
-	VOS_NAMESPACE(OProcess, vos),
-	VOS_NAMESPACE(OObject, vos), 0);
-
-
-OProcess::OProcess( ) :
-	m_strImageName( ),
-	m_strDirectory(),
-	m_Process(0)
-{
-}
-
-
-OProcess::OProcess( const rtl::OUString& strImageName ) :
-	m_strImageName( strImageName ),
-	m_strDirectory(),
-	m_Process(0)
-{
-	// empty
-}
-
-
-OProcess::OProcess(const rtl::OUString& strImageName, const rtl::OUString& strWorkingDirectory) :
-	m_strImageName( strImageName ),
-	m_strDirectory( strWorkingDirectory ),
-	m_Process(0)
-{
-	// empty
-}
-
-
-OProcess::~OProcess()
-{
-	osl_freeProcessHandle(m_Process);
-}
-
-OProcess* OProcess::getProcess(TProcessIdentifier Identifier)
-{
-	oslProcess hProcess = osl_getProcess(Identifier);
-
-	if (hProcess)
-	{
-		OProcess* pProcess = new OProcess( );
-
-		pProcess->m_Process = hProcess;
-
-		return pProcess;
-	}
-
-	return 0;
-}
-
-
-OProcess::TProcessError OProcess::execute(TProcessOption Options,
-										  const OArgumentList& aArgumentList,
-										  const OEnvironment&  aEnvironment )
-{
-	return ((TProcessError)osl_executeProcess(m_strImageName.pData, 
-								              aArgumentList.m_aVec,
-											  aArgumentList.n_Args,
-											  Options, 
-								              0,
-								              m_strDirectory.pData,
-								              aEnvironment.m_aVec,
-											  aEnvironment.n_Vars,
-											  &m_Process));
-}
-
-
-OProcess::TProcessError OProcess::execute( TProcessOption Options,
-										   const osl::Security &Security,
-										   const OArgumentList& aArgumentList,
-										   const OEnvironment&  aEnvironment )
-{
-	return ((TProcessError)osl_executeProcess(m_strImageName.pData, 
-								              aArgumentList.m_aVec,
-											  aArgumentList.n_Args,
-											  Options, 
-								              Security.getHandle(),
-								              m_strDirectory.pData,
-								              aEnvironment.m_aVec,
-											  aEnvironment.n_Vars,
-											  &m_Process));
-}
-
-
-OProcess::TProcessError OProcess::terminate()
-{
-   return (TProcessError)osl_terminateProcess(m_Process);
-}
-
-OProcess::TProcessError OProcess::getInfo(TProcessData Data, TProcessInfo* pInfo) const
-{
-   return (TProcessError)osl_getProcessInfo(m_Process, Data, pInfo);
-}
-
-OProcess::TProcessError OProcess::getCurrentInfo(TProcessData Data, TProcessInfo* pInfo)
-{
-   return (TProcessError)osl_getProcessInfo(0, Data, pInfo);
-}
-
-OProcess::TProcessError OProcess::join()
-{
-	return (TProcessError)osl_joinProcess(m_Process);
-}
-
-
-/*
-OProcess::TProcessError OProcess::searchPath(const sal_Char* Name, sal_Char *Buffer, sal_uInt32 Max, 
-                                             const sal_Char* Path, sal_Char Separator)
-{
-	return (TProcessError)osl_searchPath(Name, Path, Separator, Buffer, Max);
-}
-*/
-
-/////////////////////////////////////////////////////////////////////////////
-// StartupInfo
-
-VOS_IMPLEMENT_CLASSINFO(
-	VOS_CLASSNAME(OStartupInfo, vos),
-	VOS_NAMESPACE(OStartupInfo, vos), 
-	VOS_NAMESPACE(OObject, vos), 0);
-
-OStartupInfo::OStartupInfo()
-{
-}
-
-OStartupInfo::~OStartupInfo()
-{
-}
-
-OStartupInfo::TStartupError OStartupInfo::getExecutableFile(
-    rtl::OUString& strImageName ) const
-{
-	return (TStartupError) osl_getExecutableFile( &strImageName.pData );
-}
-
-
-OStartupInfo::TStartupError OStartupInfo::getCommandArg(sal_uInt32 nArg, rtl::OUString& strCommandArg)
-{
-	return ( TStartupError ) osl_getCommandArg( nArg,&strCommandArg.pData );
-}
-
-sal_uInt32 OStartupInfo::getCommandArgCount()
-{
-	return osl_getCommandArgCount();
-}
-
-OStartupInfo::TStartupError OStartupInfo::getEnvironment(const rtl::OUString& strVar,
-														 rtl::OUString& strValue)
-{
-	return ( TStartupError ) osl_getEnvironment( strVar.pData, &strValue.pData );
-}
-
-
-
-/////////////////////////////////////////////////////////////////////////////
-//
-// OExtCommandLineImpl
-//
-
-namespace vos
-{  
-
-class OExtCommandLineImpl
-{
-    void init();
-    
-    ::std::vector< ::rtl::OUString > aExtArgVector;
-    sal_uInt32 m_nArgCount;
-    
-public:
-    
-    OExtCommandLineImpl();
-    ~OExtCommandLineImpl();
-
-    sal_uInt32 SAL_CALL getCommandArgCount();
-    
-    sal_Bool SAL_CALL getCommandArg(sal_uInt32 nArg, ::rtl::OUString& strCommandArg);
-};
- 
-}
-
-OExtCommandLineImpl::OExtCommandLineImpl() 
-    : m_nArgCount(0)
-{
-    init();
-}
-
-OExtCommandLineImpl::~OExtCommandLineImpl()
-{
-    
-}
-
-
-sal_uInt32 SAL_CALL OExtCommandLineImpl::getCommandArgCount()
-{
-    return m_nArgCount;
-}
-
-
-sal_Bool SAL_CALL OExtCommandLineImpl::getCommandArg(sal_uInt32 nArg, ::rtl::OUString& strCommandArg) 
-{
-    if ( nArg >= m_nArgCount )
-    {
-        return sal_False;
-    }
-
-    strCommandArg = aExtArgVector[nArg];
-
-    return sal_True;
-}
-
-
-void OExtCommandLineImpl::init()
-{
-    OStartupInfo aStartInfo;
-    sal_uInt32 nIndex=0;
-    sal_uInt32 nArgs = aStartInfo.getCommandArgCount();
-
-    for ( nIndex = 0 ; nIndex < nArgs ; ++nIndex )
-    {
-        ::rtl::OUString aString;
-        aStartInfo.getCommandArg(nIndex,aString);
-        
-        if ( aString[0] == (sal_Unicode) '@' )
-        {
-            ::rtl::OUString aFileName = aString.copy(1);
-            ::osl::File aFile(aFileName);
-            ::rtl::ByteSequence aSeq;
-
-            ::osl::FileBase::RC aErr = aFile.open(OpenFlag_Read);
-
-            if ( aErr != ::osl::FileBase::E_None )
-            {
-                break;
-            }
-
-            do
-            {
-                aErr = aFile.readLine(aSeq);
-                if ( aSeq.getLength() != 0 )
-                {
-                    ::rtl::OUString newString((sal_Char*)aSeq.getArray(), aSeq.getLength(), RTL_TEXTENCODING_ASCII_US);
-                    aExtArgVector.push_back( newString );
-                    m_nArgCount++;
-                }
-            }
-            while ( aErr == ::osl::FileBase::E_None && aSeq.getLength() > 0 );
-
-			aFile.close();
-			aFile.remove(aFileName);
-        }
-        else
-        {
-            aExtArgVector.push_back( aString );
-            m_nArgCount++;
-        }
-    }
-}
-
-
-
-/////////////////////////////////////////////////////////////////////////////
-//
-// OExtCommandLine
-//
-
-namespace 
-{ 
-    struct lclMutex : public rtl::Static< osl::Mutex, lclMutex > {}; 
-}
-
-OExtCommandLineImpl* OExtCommandLine::pExtImpl=0;
-
-
-VOS_IMPLEMENT_CLASSINFO(
-	VOS_CLASSNAME(OExtCommandLine, vos),
-	VOS_NAMESPACE(OExtCommandLine, vos),
-	VOS_NAMESPACE(OObject, vos), 0);
-
-OExtCommandLine::OExtCommandLine()
-{
-    osl::MutexGuard Guard(lclMutex::get());
-
-    if ( pExtImpl == NULL )
-    {    
-        pExtImpl = new OExtCommandLineImpl;
-    }
-}
-
-OExtCommandLine::~OExtCommandLine()
-{
-
-    
-}
-
-sal_uInt32 SAL_CALL OExtCommandLine::getCommandArgCount()
-{
-    return pExtImpl->getCommandArgCount();
-}
-
-
-sal_Bool SAL_CALL OExtCommandLine::getCommandArg(sal_uInt32 nArg, ::rtl::OUString& strCommandArg) 
-{
-    return pExtImpl->getCommandArg(nArg,strCommandArg);
-}
-
diff --git a/vos/source/thread.cxx b/vos/source/thread.cxx
deleted file mode 100644
index 324da86..0000000
--- vos/source/thread.cxx
+++ /dev/null
@@ -1,250 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: thread.cxx,v $
- *
- *  $Revision: 1.8 $
- *
- *  last change: $Author: hr $ $Date: 2006/06/20 11:23:08 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-#ifndef _OSL_TIME_H_
-#include <osl/time.h>
-#endif
-#include <osl/diagnose.h>
-#include <vos/object.hxx>
-#include <vos/thread.hxx>
-
-using namespace vos;
-
-void vos::threadWorkerFunction_impl(void * pthis)
-{
-	vos::OThread* pThis= (vos::OThread*)pthis;
-
-	// call Handler-Function of OThread-derived class
-	pThis->run();
-
-	// if not already terminating, by a kill do normal shutdown 
-	if (! pThis->m_bTerminating)
-	{
-		pThis->m_bTerminating = sal_True;
-
-		pThis->onTerminated();		// could e.g. delete this
-	}
-}
-
-/////////////////////////////////////////////////////////////////////////////
-//
-// Thread class
-//
-
-VOS_IMPLEMENT_CLASSINFO(VOS_CLASSNAME(OThread, vos), 
-						VOS_NAMESPACE(OThread, vos), 
-						VOS_NAMESPACE(OObject, vos), 0);
-
-OThread::OThread() 
-{
-	m_hThread      = 0;
-	m_bTerminating = sal_False;
-    m_aCondition   = osl_createCondition();
-}
-
-OThread::~OThread() 
-{
-	if (m_hThread != 0)
-	{
-		osl_destroyThread(m_hThread);
-	}
-    
-    osl_destroyCondition( m_aCondition );
-}
-
-sal_Bool OThread::create() 
-{
-	OSL_ASSERT(m_hThread == 0);	// only one running thread per instance
-
-    m_hThread = osl_createSuspendedThread(
-        threadWorkerFunction_impl, (void*)this);
-	if (m_hThread)
-		osl_resumeThread(m_hThread);							             
-
-	return m_hThread != 0;
-}
-
-sal_Bool OThread::createSuspended() 
-{
-	OSL_ASSERT(m_hThread == 0);	// only one running thread per instance
-
-	m_hThread= osl_createSuspendedThread(threadWorkerFunction_impl, (void*)this);
-	return m_hThread != 0;
-}
-
-void OThread::suspend() 
-{
-	OSL_ASSERT(m_hThread != 0);	// use only on running thread
-
-	osl_suspendThread(m_hThread);
-}
-
-void OThread::resume() 
-{
-	OSL_ASSERT(m_hThread != 0);	// use only on running thread
-
-	osl_resumeThread(m_hThread);
-}
-
-sal_Bool OThread::isRunning() 
-{
-	return m_hThread != 0 && osl_isThreadRunning(m_hThread);
-}
-
-OThread::TThreadIdentifier OThread::getIdentifier() const
-{
-	return (TThreadIdentifier)osl_getThreadIdentifier(m_hThread);
-}
-
-OThread::TThreadIdentifier OThread::getCurrentIdentifier()
-{
-	return (TThreadIdentifier)osl_getThreadIdentifier(0);
-}
-
-void OThread::join() 
-{
-	if (m_hThread) {
-		OSL_ASSERT(getCurrentIdentifier() != getIdentifier());
-		osl_joinWithThread(m_hThread);
-	}
-}
-
-OThread::TThreadSleep OThread::sleep(const TimeValue& Delay) 
-{
-    TThreadSleep eRet;
-    
-    switch( osl_waitCondition( m_aCondition, &Delay ) )
-    {
-    case osl_cond_result_ok:
-        eRet = TSleep_Normal;
-        break;
-        
-    case osl_cond_result_timeout:
-        eRet = TSleep_Cancel;
-        break;   
-     
-    default:
-        eRet = TSleep_Error;
-        break;
-    }
-    
-    return eRet;
-}
-
-void OThread::wait(const TimeValue& Delay) {
-	osl_waitThread(&Delay);
-}
-
-sal_Bool OThread::awake() 
-{
-    osl_setCondition( m_aCondition );
-    return osl_resetCondition( m_aCondition );
-}
-
-void OThread::terminate() 
-{
-	osl_terminateThread(m_hThread);
-}
-
-sal_Bool OThread::schedule() {
-	return osl_scheduleThread(m_hThread);
-}
-
-void OThread::kill() 
-{
-	if (osl_isThreadRunning(m_hThread)) 
-	{
-		// flag we are shutting down
-		m_bTerminating = sal_True;
-
-		terminate();
-		join();
-	}
-}
-
-void OThread::setPriority(OThread::TThreadPriority Priority)
-{
-	osl_setThreadPriority(m_hThread, (oslThreadPriority)Priority);
-}
-
-OThread::TThreadPriority OThread::getPriority()
-{
-	return 	(TThreadPriority)osl_getThreadPriority(m_hThread);
-}
-
-
-void OThread::yield() 
-{ 
-	osl_yieldThread();
-}
-
-void OThread::onTerminated() 
-{
-}
-
-/////////////////////////////////////////////////////////////////////////////
-//
-// ThreadData class
-//
-
-VOS_IMPLEMENT_CLASSINFO(VOS_CLASSNAME(OThreadData, vos), 
-						VOS_NAMESPACE(OThreadData, vos), 
-						VOS_NAMESPACE(OObject, vos), 0);
-
-OThreadData::OThreadData( oslThreadKeyCallbackFunction pCallback )
-{
-    m_hKey = osl_createThreadKey( pCallback );
-	OSL_VERIFY(m_hKey);
-}
-
-OThreadData::~OThreadData()
-{
-	osl_destroyThreadKey(m_hKey);
-}
-
-sal_Bool OThreadData::setData(void *pData)
-{
-	OSL_ASSERT(m_hKey != 0);
-
-	return (osl_setThreadKeyData(m_hKey, pData));
-}
-
-void *OThreadData::getData()
-{
-	OSL_ASSERT(m_hKey != 0);
-
-	return (osl_getThreadKeyData(m_hKey));
-}
-
diff --git a/vos/source/timer.cxx b/vos/source/timer.cxx
index 00a3089..ed86c9a 100644
--- vos/source/timer.cxx
+++ vos/source/timer.cxx
@@ -40,7 +40,7 @@
 #include <vos/timer.hxx>
 #include <osl/diagnose.h>
 #include <vos/ref.hxx>
-#include <vos/thread.hxx>
+#include <osl/thread.hxx>
 #include <osl/conditn.hxx>
 
 
@@ -51,7 +51,7 @@
 
 class OTimerManagerCleanup;
 
-class vos::OTimerManager : public vos::OThread
+class vos::OTimerManager : public osl::Thread
 {
     
 public:
@@ -463,7 +463,7 @@ void OTimerManager::checkForTimeout()
 
 void OTimerManager::run() 
 {
-	setPriority(TPriority_BelowNormal);
+	setPriority(osl_Thread_PriorityBelowNormal);
 
 	while (schedule())
 	{
