diff --git a/configmgr/inc/pch/precompiled_configmgr.hxx b/configmgr/inc/pch/precompiled_configmgr.hxx
index 13f06d4..3502b7c 100644
--- configmgr/inc/pch/precompiled_configmgr.hxx
+++ configmgr/inc/pch/precompiled_configmgr.hxx
@@ -241,7 +241,7 @@
 
 
 
-#include "vos/conditn.hxx"
+#include "osl/conditn.hxx"
 #include "vos/pipe.hxx"
 #include "rtl/ref.hxx"
 #include "salhelper/simplereferenceobject.hxx"
diff --git a/configmgr/workben/apitest/cfgregistry.cxx b/configmgr/workben/apitest/cfgregistry.cxx
index 945782a..e593ef8 100644
--- configmgr/workben/apitest/cfgregistry.cxx
+++ configmgr/workben/apitest/cfgregistry.cxx
@@ -72,8 +72,8 @@
 #ifndef _CPPUHELPER_EXTRACT_HXX_
 #include <cppuhelper/extract.hxx>
 #endif
-#ifndef _VOS_CONDITN_HXX_
-#include <vos/conditn.hxx>
+#ifndef _OSL_CONDITN_HXX_
+#include <osl/conditn.hxx>
 #endif
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
diff --git a/configmgr/workben/local_io/cfglocal.cxx b/configmgr/workben/local_io/cfglocal.cxx
index 3cf1777..b77089d 100644
--- configmgr/workben/local_io/cfglocal.cxx
+++ configmgr/workben/local_io/cfglocal.cxx
@@ -62,8 +62,8 @@
 #ifndef _CPPUHELPER_IMPLBASE1_HXX_
 #include <cppuhelper/implbase1.hxx>
 #endif
-#ifndef _VOS_CONDITN_HXX_
-#include <vos/conditn.hxx>
+#ifndef _OSL_CONDITN_HXX_
+#include <osl/conditn.hxx>
 #endif
 
 #include <osl/time.h>
@@ -126,7 +126,7 @@ class OOpenNodeCallback : public IOpenObjectCallback, public OOpenNodeCallback_B
 {
 protected:
 	OUString				m_sNodeId;
-	::vos::OCondition&		m_rFinishCondition;
+	osl::Condition&		m_rFinishCondition;
 
 	enum ACTION	{ STARTELEMENT, CHARACTERS, ENDELEMENT };
 	ACTION					m_eLastAction;
@@ -139,7 +139,7 @@ protected:
 	}
 
 public:
-	OOpenNodeCallback(::vos::OCondition& _rFinishCond) : m_rFinishCondition(_rFinishCond),  m_nLevel(0), m_eLastAction(ENDELEMENT), m_bCloseStartTag(sal_False) { }
+	OOpenNodeCallback(osl::Condition& _rFinishCond) : m_rFinishCondition(_rFinishCond),  m_nLevel(0), m_eLastAction(ENDELEMENT), m_bCloseStartTag(sal_False) { }
 
 	OUString getNodeId() const { return m_sNodeId; }
 
@@ -530,10 +530,10 @@ void testSession(const Reference< XMultiServiceFactory > &xORB, bool bPrint)
 	aTimeout.Seconds = 30;
 	switch (aWaitForSomething.wait(&aTimeout))
 	{
-	case ICondition::result_error:
+	case osl_cond_result_error:
 		printf("error while waiting for the callback ... exiting\n\n");
 		return;
-	case ICondition::result_timeout:
+	case osl_cond_result_timeout:
 		if (bPrint) printf("timed out ... exiting\n\n");
 		return;
 	}
@@ -554,10 +554,10 @@ void testSession(const Reference< XMultiServiceFactory > &xORB, bool bPrint)
 // 
 // 		   switch (aWaitForSomething.wait(&aTimeout))
 // 		   {
-// 			   case ICondition::result_error:
+// 			   case osl_cond_result_error:
 // 				   printf("error while waiting for the callback ... exiting\n\n");
 // 				   return;
-// 			   case ICondition::result_timeout:
+// 			   case osl_cond_result_timeout:
 // 				   printf("timed out ... exiting\n\n");
 // 				   return;
 // 		   }
@@ -569,10 +569,10 @@ void testSession(const Reference< XMultiServiceFactory > &xORB, bool bPrint)
 	aSession.addNode(sOpenedNode, UNI_STRING(pLayoutNode), &aAddTabstops, new OOpenNodeCallback(aWaitForSomething));
 	switch (aWaitForSomething.wait(&aTimeout))
 	{
-	case ICondition::result_error:
+	case osl_cond_result_error:
 		printf("error while waiting for the callback ... exiting\n\n");
 		return;
-	case ICondition::result_timeout:
+	case osl_cond_result_timeout:
 		if (bPrint) printf("timed out ... exiting\n\n");
 		return;
 	}
@@ -586,10 +586,10 @@ void testSession(const Reference< XMultiServiceFactory > &xORB, bool bPrint)
 
 	switch (aWaitForSomething.wait(&aTimeout))
 	{
-	case ICondition::result_error:
+	case osl_cond_result_error:
 		printf("error while waiting for the callback ... exiting\n\n");
 		return;
-	case ICondition::result_timeout:
+	case osl_cond_result_timeout:
 		if (bPrint) printf("timed out ... exiting\n\n");
 		return;
 	}
@@ -602,10 +602,10 @@ void testSession(const Reference< XMultiServiceFactory > &xORB, bool bPrint)
 	aSession.deleteNode(sOpenedNode, UNI_STRING(pLayoutNode2), new OOpenNodeCallback(aWaitForSomething));
 	switch (aWaitForSomething.wait(&aTimeout))
 	{
-	case ICondition::result_error:
+	case osl_cond_result_error:
 		printf("error while waiting for the callback ... exiting\n\n");
 		return;
-	case ICondition::result_timeout:
+	case osl_cond_result_timeout:
 		if (bPrint) printf("timed out ... exiting\n\n");
 		return;
 	}
@@ -628,10 +628,10 @@ void testSession(const Reference< XMultiServiceFactory > &xORB, bool bPrint)
 // 		   aSession.getNode(UNI_STRING(pLayoutNode), new OOpenNodeCallback(aWaitForSomething));
 // 		   switch (aWaitForSomething.wait(&aTimeout))
 // 		   {
-// 			   case ICondition::result_error:
+// 			   case osl_cond_result_error:
 // 				   printf("error while waiting for the callback ... exiting\n\n");
 // 				   return;
-// 			   case ICondition::result_timeout:
+// 			   case osl_cond_result_timeout:
 // 				   printf("timed out ... exiting\n\n");
 // 				   return;
 // 		   }
diff --git a/extensions/inc/pch/precompiled_extensions.hxx b/extensions/inc/pch/precompiled_extensions.hxx
index 24167fd..54fdd33 100644
--- extensions/inc/pch/precompiled_extensions.hxx
+++ extensions/inc/pch/precompiled_extensions.hxx
@@ -543,7 +543,7 @@
 
 #include "vcl/fldunit.hxx"
 
-#include "vos/conditn.hxx"
+#include "osl/conditn.hxx"
 #include "osl/diagnose.h"
 #include "vos/macros.hxx"
 #include "vos/module.hxx"
diff --git a/extensions/source/plugin/inc/plugin/unx/mediator.hxx b/extensions/source/plugin/inc/plugin/unx/mediator.hxx
index a7d9fbb..bbc9b98 100644
--- extensions/source/plugin/inc/plugin/unx/mediator.hxx
+++ extensions/source/plugin/inc/plugin/unx/mediator.hxx
@@ -53,8 +53,8 @@
 #ifndef _OSL_MUTEX_HXX_
 #include <osl/mutex.hxx>
 #endif
-#ifndef _VOS_CONDITN_HXX_
-#include <vos/conditn.hxx>
+#ifndef _OSL_CONDITN_HXX_
+#include <osl/conditn.hxx>
 #endif
 #ifndef _THREAD_HXX_
 #include <osl/thread.hxx>
@@ -117,7 +117,7 @@ protected:
     osl::Mutex                          m_aQueueMutex;
     osl::Mutex                          m_aSendMutex;
 	// only one thread can send a message at any given time
-	NAMESPACE_VOS(OCondition)			m_aNewMessageCdtn;
+	osl::Condition						m_aNewMessageCdtn;
 	MediatorListener*					m_pListener;
 	// thread to fill the queue
 
diff --git a/extensions/test/sax/testsax.cxx b/extensions/test/sax/testsax.cxx
index e778dce..1b2d845 100644
--- extensions/test/sax/testsax.cxx
+++ extensions/test/sax/testsax.cxx
@@ -46,7 +46,7 @@
 //#include <osl/diagnose.h>
 #include <usr/weak.hxx>
 #include <tools/string.hxx>
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 
 #if OSL_DEBUG_LEVEL == 0
 #define NDEBUG
diff --git a/extensions/test/stm/datatest.cxx b/extensions/test/stm/datatest.cxx
index fa4cd8e..fce1697 100644
--- extensions/test/stm/datatest.cxx
+++ extensions/test/stm/datatest.cxx
@@ -53,7 +53,7 @@
 #include <usr/reflserv.hxx>  // for EXTERN_SERVICE_CALLTYPE
 #include <usr/weak.hxx>      // OWeakObject
 
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 #include <osl/mutex.hxx>
 #include <osl/thread.hxx>
 
diff --git a/extensions/test/stm/marktest.cxx b/extensions/test/stm/marktest.cxx
index f99e57d..5bb24cd 100644
--- extensions/test/stm/marktest.cxx
+++ extensions/test/stm/marktest.cxx
@@ -49,7 +49,7 @@
 #include <usr/reflserv.hxx>  // for EXTERN_SERVICE_CALLTYPE
 #include <usr/weak.hxx>      // OWeakObject
 
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 #include <osl/mutex.hxx>
 #include <osl/thread.hxx>
 
diff --git a/extensions/test/stm/pipetest.cxx b/extensions/test/stm/pipetest.cxx
index 0aa6d41..cee0050 100644
--- extensions/test/stm/pipetest.cxx
+++ extensions/test/stm/pipetest.cxx
@@ -46,7 +46,7 @@
 #include <usr/reflserv.hxx>  // for EXTERN_SERVICE_CALLTYPE
 #include <usr/weak.hxx>      // OWeakObject
 
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 #include <osl/mutex.hxx>
 #include <osl/thread.hxx>
 
diff --git a/extensions/workben/pythonautotest.cxx b/extensions/workben/pythonautotest.cxx
index d41a255..950d28f 100644
--- extensions/workben/pythonautotest.cxx
+++ extensions/workben/pythonautotest.cxx
@@ -48,7 +48,7 @@
 #include <usr/ustring.hxx>
 #include <usr/weak.hxx>
 #include <tools/string.hxx>
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 #if OSL_DEBUG_LEVEL == 0
 #define NDEBUG
 #endif
diff --git a/extensions/workben/pythontest.cxx b/extensions/workben/pythontest.cxx
index 73d466a..2c8d894 100644
--- extensions/workben/pythontest.cxx
+++ extensions/workben/pythontest.cxx
@@ -49,7 +49,7 @@
 #include <usr/ustring.hxx>
 #include <usr/weak.hxx>
 #include <tools/string.hxx>
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 #if OSL_DEBUG_LEVEL == 0
 #define NDEBUG
 #endif
diff --git a/extensions/workben/testcomponent.cxx b/extensions/workben/testcomponent.cxx
index 607e4bd..cd49e03 100644
--- extensions/workben/testcomponent.cxx
+++ extensions/workben/testcomponent.cxx
@@ -55,7 +55,7 @@
 #include <vcl/svapp.hxx>
 #include <usr/ustring.hxx>
 #include <tools/string.hxx>
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 
 #if OSL_DEBUG_LEVEL == 0
 #define NDEBUG
diff --git a/sal/typesconfig/core b/sal/typesconfig/core
index b5b6cf5..97fe99e 100644
Binary files a/sal/typesconfig/core and b/sal/typesconfig/core differ
diff --git a/unotools/inc/unotools/ucblockbytes.hxx b/unotools/inc/unotools/ucblockbytes.hxx
index a87730f..8e8942c 100644
--- unotools/inc/unotools/ucblockbytes.hxx
+++ unotools/inc/unotools/ucblockbytes.hxx
@@ -18,7 +18,7 @@
 #endif
 
 #include <osl/thread.hxx>
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 #include <osl/mutex.hxx>
 #include <tools/stream.hxx>
 #include <tools/link.hxx>
@@ -88,8 +88,8 @@ SV_DECL_IMPL_REF( UcbLockBytesHandler )
 
 class UNOTOOLS_DLLPUBLIC UcbLockBytes : public virtual SvLockBytes
 {
-	vos::OCondition			m_aInitialized;
-	vos::OCondition			m_aTerminated;
+	osl::Condition			m_aInitialized;
+	osl::Condition			m_aTerminated;
 	osl::Mutex				m_aMutex;
 
     String                  m_aContentType;
diff --git a/vos/inc/vos/conditn.hxx b/vos/inc/vos/conditn.hxx
deleted file mode 100644
index cd1b76d..0000000
--- vos/inc/vos/conditn.hxx
+++ /dev/null
@@ -1,150 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: conditn.hxx,v $
- *
- *  $Revision: 1.4 $
- *
- *  last change: $Author: vg $ $Date: 2006/06/02 12:35:55 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-
-#ifndef _VOS_CONDITN_HXX_
-#define _VOS_CONDITN_HXX_
-
-#ifndef _OSL_CONDITN_H_
-#	include <osl/conditn.h>
-#endif
-#ifndef _OSL_TIME_H_
-#include <osl/time.h>
-#endif
-#ifndef _VOS_OBJECT_HXX_
-#	include <vos/object.hxx>
-#endif
-
-namespace vos
-{
-
-/** ICondition
-	
-	Interface for a thread-spanning condition. If a condition-object
-	is created, its initial condition is False. You can check the
-	condition nonblocking with "check()" or wait for it to become set
-	with "wait()". The methods "set()" and "reset()" are used to change
-	the conditions state.
-
-    @author  Bernd Hofner
-	@version 1.0
-*/
-class ICondition
-{
-public:
-
-    ICondition() { }
-    virtual ~ICondition() { }
-    
-    
-    
-	enum TResult
-    {
-	    result_ok          = osl_cond_result_ok,
-	    result_error       = osl_cond_result_error,
-		result_timeout     = osl_cond_result_timeout
-	};
-
-	/** set condition to True =>
-		wait() will not block, check() returns True
-	*/
-	virtual void SAL_CALL set()= 0;
-
-	/** set condition to False =>
-		wait() will block, check() returns False
-	*/
-	virtual void SAL_CALL reset()= 0;
-
-	/** Blocks if condition is not set<BR>
-		If condition has been destroyed prematurely, wait() will
-		return with False.
-	*/
-	virtual TResult SAL_CALL wait(const TimeValue* pTimeout = 0)= 0;
-
-	/** True: condition is set <BR>
-		False: condition is not set <BR>
-		does not block
-	*/
-	virtual sal_Bool SAL_CALL check()= 0;
-};
-
-
-/** OCondition
-	
-	Implements the ICondition interface.
-
-    @author  Bernd Hofner
-	@version 1.0
-
-*/
-class OCondition : public OObject, public ICondition
-{
-
-	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OCondition, vos));
-
-public:
-
-	/// initial state of condition is not set
-	OCondition();
-	virtual ~OCondition();
-
-	/// set condition to True => wait() will not block, check() returns True
-	virtual void SAL_CALL set();
-
-	/// set condition to False => wait() will block, check() returns False
-	virtual void SAL_CALL reset();
-
-	/** Blocks if condition is not set<BR>
-		If condition has been destroyed prematurely, wait() will
-		return with False.
-	*/
-	TResult SAL_CALL wait(const TimeValue* pTimeout = 0);
-
-	/** True: condition is set <BR>
-		False: condition is not set <BR>
-		does not block
-	*/
-	virtual sal_Bool SAL_CALL check();
-
-protected:
-
-	oslCondition	m_Condition;
-
-};
-
-}
-
-#endif	// _VOS_CONDITN_HXX_
-
diff --git a/vos/source/conditn.cxx b/vos/source/conditn.cxx
deleted file mode 100644
index 828f3e9..0000000
--- vos/source/conditn.cxx
+++ /dev/null
@@ -1,86 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: conditn.cxx,v $
- *
- *  $Revision: 1.4 $
- *
- *  last change: $Author: vg $ $Date: 2006/06/02 12:41:52 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-#ifndef _OSL_TIME_H_
-#include <osl/time.h>
-#endif
-#include <vos/conditn.hxx>
-#include <osl/diagnose.h>
-
-using namespace vos;
-
-VOS_IMPLEMENT_CLASSINFO(VOS_CLASSNAME(OCondition, vos), VOS_NAMESPACE(OCondition, vos), VOS_NAMESPACE(OObject, vos), 0);
-
-/// initial state of condition is not set
-OCondition::OCondition()
-{
-	m_Condition= osl_createCondition();
-}
-
-OCondition::~OCondition()
-{
-	osl_destroyCondition(m_Condition);
-}
-
-/// set condition to sal_True => wait() will not block, check() returns sal_True
-void OCondition::set()
-{
-	osl_setCondition(m_Condition);
-}
-
-/// set condition to sal_False => wait() will block, check() returns sal_False
-void OCondition::reset()
-{
-	osl_resetCondition(m_Condition);
-}
-
-/** Blocks if condition is not set<BR>
-	If condition has been destroyed prematurely, wait() will
-	return with sal_False.
-*/
-OCondition::TResult OCondition::wait(const TimeValue* pTimeout)
-{
-	return (TResult)osl_waitCondition(m_Condition, pTimeout);
-}
-
-/** sal_True: condition is set <BR>
-	sal_False: condition is not set <BR>
-	does not block
-*/
-sal_Bool OCondition::check()
-{
-	return osl_checkCondition(m_Condition);
-}
-
diff --git a/vos/source/makefile.mk b/vos/source/makefile.mk
index 2cf9daf..5efea26 100644
--- vos/source/makefile.mk
+++ vos/source/makefile.mk
@@ -45,7 +45,7 @@ ENABLE_EXCEPTIONS=TRUE
 
 # --- Files --------------------------------------------------------
 
-SLOFILES=       $(SLO)$/conditn.obj     \
+SLOFILES= \
 				$(SLO)$/object.obj      \
 				$(SLO)$/refernce.obj    \
 				$(SLO)$/socket.obj      \
@@ -61,7 +61,7 @@ SLOFILES=       $(SLO)$/conditn.obj     \
 
 
 .IF "$(UPDATER)"=="YES"
-OBJFILES=       $(OBJ)$/conditn.obj     \
+OBJFILES= \
 				$(OBJ)$/object.obj      \
 				$(OBJ)$/refernce.obj    \
 				$(OBJ)$/socket.obj      \
diff --git a/vos/source/timer.cxx b/vos/source/timer.cxx
index 3ba8830..4750ac6 100644
--- vos/source/timer.cxx
+++ vos/source/timer.cxx
@@ -41,7 +41,7 @@
 #include <osl/diagnose.h>
 #include <vos/ref.hxx>
 #include <vos/thread.hxx>
-#include <vos/conditn.hxx>
+#include <osl/conditn.hxx>
 
 
 /////////////////////////////////////////////////////////////////////////////
@@ -91,7 +91,7 @@ protected:
     // List Protection
     osl::Mutex					m_Lock;
     // Signal the insertion of a timer
-    NAMESPACE_VOS(OCondition)	m_notEmpty;
+    osl::Condition				m_notEmpty;
 
     // Synchronize access to OTimerManager
 	static osl::Mutex 			m_Access;
