Index: unx/gtk/gdi/salnativewidgets-gtk.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx,v
retrieving revision 1.47
diff -u -p -r1.47 salnativewidgets-gtk.cxx
--- vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	16 Jun 2008 14:30:38 -0000	1.47
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	28 Oct 2008 16:05:04 -0000
@@ -3635,7 +3640,7 @@ static void NWEnsureGTKSpinButton( int n
 {
 	if ( !gWidgetData[nScreen].gSpinButtonWidget )
 	{
-		GtkAdjustment *adj = GTK_ADJUSTMENT( gtk_adjustment_new(0, 0, 2, 1, 1, 1) );
+		GtkAdjustment *adj = GTK_ADJUSTMENT( gtk_adjustment_new(0, 0, 2, 1, 1, 0) );
 		gWidgetData[nScreen].gSpinButtonWidget = gtk_spin_button_new( adj, 1, 2 );
 
 		//Setting non-editable means it doesn't blink, so there's no timeouts
