--- scp2/source/ooo/file_ooo.scp.org	2008-07-17 08:37:09.000000000 +0200
+++ scp2/source/ooo/file_ooo.scp	2008-07-18 07:23:31.000000000 +0200
@@ -2920,6 +2920,19 @@
 
 // Dictionary extensions
 
+#ifdef WNT
+
+#ifndef WITHOUT_MYSPELL_DICTS
+File gid_File_Extension_Dictionary_Bundle
+	Dir = gid_Brand_Dir_Share_Extension_Install;
+	Name = "dict-bundle.oxt";
+	Styles = (PACKED);
+	UnixRights = 444;
+End
+#endif
+
+#else
+
 #ifndef WITHOUT_MYSPELL_DICTS
 File gid_File_Extension_Dictionary_Af
 	Dir = gid_Brand_Dir_Share_Extension_Install;
@@ -3136,6 +3149,8 @@
 End
 #endif
 
+#endif
+
 // #ifdef BUILD_SPECIAL
 // File gid_File_Extension_Oxt_Oooimprovement
 //    Name = "OOoImprovement.oxt";
