--- scp2/source/base/folderitem_base.scp
+++ scp2/source/base/folderitem_base.scp
@@ -40,5 +40,4 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_BASE);
 End
--- scp2/source/calc/folderitem_calc.scp
+++ scp2/source/calc/folderitem_calc.scp
@@ -40,5 +40,4 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_CALC);
 End
--- scp2/source/impress/folderitem_impress.scp
+++ scp2/source/impress/folderitem_impress.scp
@@ -40,5 +40,4 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_IMPRESS);
 End
--- scp2/source/math/folderitem_math.scp
+++ scp2/source/math/folderitem_math.scp
@@ -40,5 +40,4 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_MATH);
 End
--- scp2/source/draw/folderitem_draw.scp
+++ scp2/source/draw/folderitem_draw.scp
@@ -40,5 +40,4 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_DRAW);
 End
--- scp2/source/ooo/folderitem_ooo.scp
+++ scp2/source/ooo/folderitem_ooo.scp
@@ -40,7 +40,6 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_SOFFICE);
 End
 
 FolderItem gid_Folderitem_Soffice_Desktop
@@ -54,6 +54,5 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
 	Styles = (NOWEB,NON_ADVERTISED,USE_HELPER_FILENAME);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_SOFFICE);
     ComponentCondition = "CREATEDESKTOPLINK=1";
 End
--- scp2/source/writer/folderitem_writer.scp
+++ scp2/source/writer/folderitem_writer.scp
@@ -40,5 +40,4 @@
     WkDir = gid_Dir_Program;
     Parameter = "";
     Styles = (NOWEB, NON_ADVERTISED);
-    ALL_LANG(Tooltip, STR_FI_TOOLTIP_WRITER);
 End
