--- configure.in	Wed May  3 15:15:00 2006
+++ configure.in	Wed May  3 14:38:03 2006
@@ -74,6 +74,9 @@
                           epm, however epm is useless for large scale
                           package building.
 ],,enable_epm="yes")
+AC_ARG_ENABLE(activex_component,
+[  --disable-activex-component Remove support for ActiveX embedding of OOo
+],,enable_activex_component=yes)
 AC_ARG_WITH(epm,
 [  --with-epm		  Decides which epm to use. Default is to use
 			  the one from the system if one is built. When
@@ -4268,6 +4272,20 @@
 AC_SUBST(WITH_WRITER2LATEX)
 
 AC_SUBST(SCPDEFS)
+
+if test "$_os" = "WINNT"; then
+  AC_MSG_CHECKING([whether to enable ActiveX embedding of OOo components])
+  if test "$enable_activex_component" = "yes" -o "$enable_activex_component" = "TRUE" -o "$enable_activex_component" = ""; then
+    ENABLE_ACTIVEX_COMPONENT="TRUE"
+    AC_MSG_RESULT([yes])
+    SCPDEFS="$SCPDEFS -DWITH_ACTIVEX_COMPONENT"
+  else
+    ENABLE_ACTIVEX_COMPONENT=""
+    AC_MSG_RESULT([no])
+  fi
+  AC_SUBST(ENABLE_ACTIVEX_COMPONENT)
+  AC_SUBST(SCPDEFS)
+fi
 
 AC_MSG_CHECKING([whether and how to use Xinerama])
 if test "$_os" = "Darwin"; then
--- scp2/source/activex/file_activex.scp	Thu Oct 27 17:08:13 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/activex/file_activex.scp	Wed May  3 14:24:12 2006
@@ -36,4 +36,6 @@
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 File gid_File_Lib_Soactivex
     TXT_FILE_BODY;
@@ -57,2 +59,3 @@
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
--- scp2/source/activex/module_activex.scp	Fri Sep  9 04:17:32 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/activex/module_activex.scp	Wed May  3 14:24:29 2006
@@ -36,6 +36,8 @@
 
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 Module gid_Module_Optional_Activexcontrol
     PackageInfo = "packinfo_office.txt";
     MOD_NAME_DESC ( MODULE_OPTIONAL_ACTIVEXCONTROL );
@@ -43,3 +45,5 @@
     ParentID = gid_Module_Optional;
     Files = (gid_File_Lib_Soactivex, gid_File_Lib_Regactivex_Msi, gid_File_Lib_Regpatchactivex_Msi);
 End
+
+#endif // WITH_ACTIVEX_COMPONENT
--- scp2/source/calc/registryitem_calc.scp	Thu Nov 10 18:18:08 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/calc/registryitem_calc.scp	Wed May  3 14:25:20 2006
@@ -36,6 +36,8 @@
 
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 // MSOLE embedded object in SO7 file format
 RegistryItem gid_Regitem_Clsid__7b342dc4_139a_4a46_8a93_Db0827ccee9c_
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -272,6 +274,8 @@
     Value = "opendocument.CalcDocument.1";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starcalcdocument_6
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Calc_Bin;
@@ -359,6 +363,8 @@
     REG_VALUE_LNG(SO60_SPREADSHEET)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starcalcdocument_6_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Calc_Bin;
@@ -366,6 +372,8 @@
     Value = "{7B342DC4-139A-4a46-8A93-DB0827CCEE9C}";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starcalcdocument_6_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Calc_Bin;
@@ -688,6 +696,8 @@
     REG_VALUE_LNG(OO_CALC)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Openoffice_Calcdocument_1_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Calc_Bin;
@@ -694,6 +704,8 @@
     Subkey = "opendocument.CalcDocument.1\CLSID";
     Value = "{7FA8AE11-B3E3-4D88-AABF-255526CD1CE8}";
 End
+
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Openoffice_Calcdocument_1_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
--- scp2/source/draw/registryitem_draw.scp	Thu Nov 10 18:18:21 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/draw/registryitem_draw.scp	Wed May  3 14:25:46 2006
@@ -36,6 +36,8 @@
 
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 // MSOLE embedded object in SO7 file format
 RegistryItem gid_Regitem_Clsid__41662fc2_0d57_4aff_Ab27_Ad2e12e7c273_
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -272,7 +274,7 @@
     Value = "opendocument.DrawDocument.1";
 End
 
-
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Soffice_Stardrawdocument_6
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -361,6 +363,8 @@
     REG_VALUE_LNG(SO60_DRAWING)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Stardrawdocument_6_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Draw_Bin;
@@ -368,6 +372,8 @@
     Value = "{41662FC2-0D57-4aff-AB27-AD2E12E7C273}";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Stardrawdocument_6_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Draw_Bin;
@@ -690,6 +696,8 @@
     REG_VALUE_LNG(OO_DRAW)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Openoffice_Drawdocument_1_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Draw_Bin;
@@ -696,6 +704,8 @@
     Subkey = "opendocument.DrawDocument.1\CLSID";
     Value = "{448BB771-CFE2-47C4-BCDF-1FBF378E202C}";
 End
+
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Openoffice_Drawdocument_1_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
--- scp2/source/impress/registryitem_impress.scp	Thu Nov 10 18:18:33 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/impress/registryitem_impress.scp	Wed May  3 14:26:13 2006
@@ -36,6 +36,8 @@
 
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 // MSOLE embedded object in SO7 file format
 RegistryItem gid_Regitem_Clsid__E5a0b632_Dfba_4549_9346_E414da06e6f8_
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -272,7 +274,7 @@
     Value = "opendocument.ImpressDocument.1";
 End
 
-
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Soffice_Starimpressdocument_6
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -377,6 +379,8 @@
     REG_VALUE_LNG(SO60_PRESENT)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starimpressdocument_6_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Impress_Bin;
@@ -384,6 +388,8 @@
     Value = "{E5A0B632-DFBA-4549-9346-E414DA06E6F8}";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starimpressdocument_6_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Impress_Bin;
@@ -727,6 +733,8 @@
     REG_VALUE_LNG(OO_PRESENT)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Openoffice_Impressdocument_1_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Impress_Bin;
@@ -733,6 +741,8 @@
     Subkey = "opendocument.ImpressDocument.1\CLSID";
     Value = "{EE5D1EA4-D445-4289-B2FC-55FC93693917}";
 End
+
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Openoffice_Impressdocument_1_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
--- scp2/source/math/registryitem_math.scp	Thu Nov 10 18:18:46 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/math/registryitem_math.scp	Wed May  3 14:26:36 2006
@@ -35,6 +35,8 @@
  ************************************************************************/
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 // MSOLE embedded object in SO7 file format
 RegistryItem gid_Regitem_Clsid__D0484de6_Aaee_468a_991f_8d4b0737b57a_
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -271,7 +273,7 @@
     Value = "opendocument.MathDocument.1";
 End
 
-
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Soffice_Starmathdocument_6
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -360,6 +362,8 @@
     REG_VALUE_LNG(SO60_FORMULA)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starmathdocument_6_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Math_Bin;
@@ -367,6 +371,8 @@
     Value = "{D0484DE6-AAEE-468a-991F-8D4B0737B57A}";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starmathdocument_6_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Math_Bin;
@@ -582,6 +588,8 @@
     REG_VALUE_LNG(OO_FORMULA)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Openoffice_Mathdocument_1_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Math_Bin;
@@ -588,6 +596,8 @@
     Subkey = "opendocument.MathDocument.1\CLSID";
     Value = "{D2D59CD1-0A6A-4D36-AE20-47817077D57C}";
 End
+
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Openoffice_Mathdocument_1_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
--- scp2/source/writer/registryitem_writer.scp	Thu Nov 10 18:19:00 2005
+++ /mnt/pippi/src/2.0.2/build/OOO_2_0_2/scp2/source/writer/registryitem_writer.scp	Wed May  3 14:27:50 2006
@@ -36,6 +36,8 @@
 
 #include "macros.inc"
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 // MSOLE embedded object in SO7 file format
 RegistryItem gid_Regitem_Clsid__30a2652a_Ddf7_45e7_Aca6_3eab26fc8a4e_
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -272,6 +274,7 @@
     Value = "opendocument.WriterDocument.1";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Soffice_Starwriterdocument_6
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
@@ -376,6 +379,8 @@
     REG_VALUE_LNG(SO60_TEXT)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starwriterdocument_6_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Wrt_Bin;
@@ -383,6 +388,8 @@
     Value = "{30A2652A-DDF7-45e7-ACA6-3EAB26FC8A4E}";
 End
 
+#endif // WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Soffice_Starwriterdocument_6_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Wrt_Bin;
@@ -882,6 +889,8 @@
     REG_VALUE_LNG(OO_WRITER)
 End
 
+#ifdef WITH_ACTIVEX_COMPONENT
+
 RegistryItem gid_Regitem_Openoffice_Writerdocument_1_Clsid
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
     ModuleID = gid_Module_Prg_Wrt_Bin;
@@ -888,6 +897,8 @@
     Subkey = "opendocument.WriterDocument.1\CLSID";
     Value = "{F616B81F-7BB8-4F22-B8A5-47428D59F8AD}";
 End
+
+#endif // WITH_ACTIVEX_COMPONENT
 
 RegistryItem gid_Regitem_Openoffice_Writerdocument_1_Defaulticon
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
