--- wizards/com/sun/star/wizards/form/Finalizer.java.old	2006-04-07 14:41:55.000000000 +0200
+++ wizards/com/sun/star/wizards/form/Finalizer.java	2008-05-14 18:52:51.000000000 +0200
@@ -75,7 +75,7 @@ public class Finalizer {
         );
         CurUnoDialog.insertLabel("lblProceed", 
           new String[] {"Height", "Label", "PositionX", "PositionY", "Step", "TabIndex", "Width"},
-          new Object[] { UIConsts.INTEGERS[8], slblProceed, new Integer(97), new Integer(62),UIConsts.INTEGERS[8], new Short(curtabindex++), new Integer(154)}
+          new Object[] { UIConsts.INTEGERS[8], slblProceed, new Integer(97), new Integer(62),UIConsts.INTEGERS[8], new Short(curtabindex++), new Integer(185)}
         );
         XRadioButton optWorkWithForm = CurUnoDialog.insertRadioButton("optWorkWithForm", null,
           new String[] {"Height", "HelpURL", "Label", "PositionX", "PositionY", "State", "Step", "TabIndex", "Width"},
