Index: source/writer/registryitem_writer.scp
===================================================================
--- scp2/source/writer/registryitem_writer.scp	(Revision 266578)
+++ scp2/source/writer/registryitem_writer.scp	(Arbeitskopie)
@@ -1278,6 +1278,15 @@
     Value = " ";
 End
 
+// .WPS
+RegistryItem gid_Regitem_OpenOffice_WPS_OpenWith_Writer
+    ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
+    Subkey = ".wps\OpenWithProgIDs";
+    ModuleID = gid_Module_Prg_Wrt_Bin;
+    Name = "opendocument.WriterDocument.1";
+    Value = " ";
+End
+
 // .HWP
 RegistryItem gid_Regitem_OpenOffice_HWP_OpenWith_Writer
     ParentID = PREDEFINED_HKEY_CLASSES_ROOT;
