Index: sfx2/source/appl/app.src
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/app.src,v
retrieving revision 1.88
diff -u -r1.88 app.src
--- sfx2/source/appl/app.src	17 Jul 2003 13:14:26 -0000	1.88
+++ sfx2/source/appl/app.src	20 Aug 2003 21:36:38 -0000
@@ -3657,6 +3657,14 @@
 		};
 		ToolBoxItem
 		{
+			Identifier = SID_ATTR_ZOOM ;
+		};
+		ToolBoxItem
+		{
+			Type = TOOLBOXITEM_SEPARATOR ;
+		};
+		ToolBoxItem
+		{
 			Identifier = SID_CLOSEDOC;
 			Hide = TRUE;
 		};
Index: svx/inc/helpid.hrc
===================================================================
RCS file: /cvs/graphics/svx/inc/helpid.hrc,v
retrieving revision 1.38
diff -u -r1.38 helpid.hrc
--- svx/inc/helpid.hrc	30 Apr 2003 08:23:57 -0000	1.38
+++ svx/inc/helpid.hrc	17 Aug 2003 16:31:26 -0000
@@ -768,6 +768,7 @@
 #define HID_SVX_OPTIONS_ONLINEUPDATE                (HID_SVX_EXT0_START + 186)
 #define HID_OPTIONS_CHART_DEFCOLORS                 (HID_SVX_EXT0_START + 187)
 #define HID_OPTIONS_MULTIPATH_LIST                  (HID_SVX_EXT0_START + 188)
+#define HID_ZOOM_COMBOBOX                           (HID_SVX_EXT0_START + 189)
 
 // please adjust ACT_SVX_HID_END2 below if you add entries here!
 
@@ -806,7 +806,7 @@
 #error Resource-Ueberlauf in #line, #file
 #endif
 
-#define ACT_SVX_HID_END2        HID_OPTIONS_MULTIPATH_LIST
+#define ACT_SVX_HID_END2        HID_ZOOM_COMBOBOX
 #if ACT_SVX_HID_END2 > HID_SVX_EXT0_END
 #error resource overrun in #line, #file
 #endif
Index: svx/source/tbxctrls/tbcontrl.src
===================================================================
RCS file: /cvs/graphics/svx/source/tbxctrls/tbcontrl.src,v
retrieving revision 1.35
diff -u -r1.35 tbcontrl.src
--- svx/source/tbxctrls/tbcontrl.src	17 Jul 2003 11:33:35 -0000	1.35
+++ svx/source/tbxctrls/tbcontrl.src	17 Aug 2003 16:31:38 -0000
@@ -263,4 +263,14 @@
 };
+ComboBox RID_SVXTBX_ZOOM
+{
+	HelpId = HID_ZOOM_COMBOBOX ;
+	Size = MAP_APPFONT ( 30 , 86 ) ;
+	DropDown = TRUE ;
+	AutoHScroll = TRUE ;
+	DDExtraWidth = TRUE ;
+	Border = TRUE ;
+	Hide = TRUE ;
+};
 String RID_SVXSTR_EXTRAS_CHARBACKGROUND
 {
 	Text [ en-US ] = "Highlighting" ;
dummy line to avoid confusing diff-mode
--- sw/source/ui/app/swmodule.cxx	2003-07-07 18:10:23.000000000 +0200
+++ sw/source/ui/app/swmodule.cxx	2003-08-13 22:52:29.000000000 +0200
@@ -121,6 +121,7 @@
 #include <svx/fillctrl.hxx>
 #include <svx/tbcontrl.hxx>
 #include <svx/verttexttbxctrl.hxx>
+#include <svx/zoomtbxctrl.hxx>
 #include <svx/formatpaintbrushctrl.hxx>
 #include <svx/contdlg.hxx>
 #include <svx/layctrl.hxx>
@@ -573,6 +576,7 @@
 	SvxFontSizeMenuControl::RegisterControl(SID_ATTR_CHAR_FONTHEIGHT, pMod );
 
 	SwZoomControl::RegisterControl(SID_ATTR_ZOOM, pMod );
+	SvxZoomToolBoxControl::RegisterControl(SID_ATTR_ZOOM, pMod);
     SwPreviewZoomControl::RegisterControl(FN_PREVIEW_ZOOM, pMod);
 	SwHyperlinkControl::RegisterControl(FN_STAT_HYPERLINKS, pMod );
 	SvxPosSizeStatusBarControl::RegisterControl(0, pMod );

Index: sw/source/ui/uiview/pview.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/uiview/pview.src,v
retrieving revision 1.28
diff -u -r1.28 pview.src
--- sw/source/ui/uiview/pview.src	17 Jul 2003 12:04:56 -0000	1.28
+++ sw/source/ui/uiview/pview.src	20 Aug 2003 21:47:00 -0000
@@ -1043,11 +1043,6 @@
             Identifier = SID_ZOOM_IN;
             HelpID = SID_ZOOM_IN ;
         };
-        ToolBoxItem
-		{
-			Identifier = FN_PREVIEW_ZOOM;
-			HelpID = HID_PREVIEW_ZOOM ;
-		};
 		ToolBoxItem { Type = TOOLBOXITEM_SEPARATOR ; };
 		ToolBoxItem
 		{
Index: sc/source/ui/app/scdll.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/app/scdll.cxx,v
retrieving revision 1.14
diff -u -r1.14 scdll.cxx
--- sc/source/ui/app/scdll.cxx	26 Mar 2003 18:05:47 -0000	1.14
+++ sc/source/ui/app/scdll.cxx	18 Aug 2003 19:08:47 -0000
@@ -105,6 +105,7 @@
 #include <svx/selctrl.hxx>
 #include <svx/insctrl.hxx>
 #include <svx/zoomctrl.hxx>
+#include <svx/zoomtbxctrl.hxx>
 #include <svx/flditem.hxx>
 #include <svx/modctrl.hxx>
 #include <svx/pszctrl.hxx>
@@ -243,6 +244,7 @@
 	SvxInsertStatusBarControl		::RegisterControl(SID_ATTR_INSERT,		pMod);
 	SvxSelectionModeControl			::RegisterControl(SID_STATUS_SELMODE,	pMod);
 	SvxZoomStatusBarControl			::RegisterControl(SID_ATTR_ZOOM,		pMod);
+	SvxZoomToolBoxControl			::RegisterControl(SID_ATTR_ZOOM,		pMod);
 	SvxModifyControl 				::RegisterControl(SID_DOC_MODIFIED,		pMod);
 	XmlSecStatusBarControl          ::RegisterControl( SID_SIGNATURE,       pMod );
 
Index: sd/source/ui/app/sddll2.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/sddll2.cxx,v
retrieving revision 1.10
diff -u -r1.10 sddll2.cxx
--- sd/source/ui/app/sddll2.cxx	13 Sep 2002 13:42:54 -0000	1.10
+++ sd/source/ui/app/sddll2.cxx	18 Aug 2003 19:03:03 -0000
@@ -114,6 +114,7 @@
 #include <svx/linectrl.hxx>
 #include <svx/tbcontrl.hxx>
 #include <svx/zoomctrl.hxx>
+#include <svx/zoomtbxctrl.hxx>
 #include <svx/pszctrl.hxx>
 #include <svx/modctrl.hxx>
 #include <svx/fntctl.hxx>
@@ -242,6 +245,7 @@
 
 	// StatusBarControls registrieren
 	SvxZoomStatusBarControl::RegisterControl( SID_ATTR_ZOOM, pMod );
+	SvxZoomToolBoxControl::RegisterControl( SID_ATTR_ZOOM, pMod );
 	SvxPosSizeStatusBarControl::RegisterControl( SID_ATTR_SIZE, pMod );
 	SvxModifyControl::RegisterControl( SID_DOC_MODIFIED, pMod );
 	//SvxInsertStatusBarControl::RegisterControl(0, pModd);
Index: starmath/source/smdll.cxx
===================================================================
RCS file: /cvs/sw/starmath/source/smdll.cxx,v
retrieving revision 1.5
diff -u -r1.5 smdll.cxx
--- starmath/source/smdll.cxx	12 Oct 2001 15:54:37 -0000	1.5
+++ starmath/source/smdll.cxx	18 Aug 2003 19:10:56 -0000
@@ -73,6 +73,7 @@
 #endif
 #include <svx/modctrl.hxx>
 #include <svx/zoomctrl.hxx>
+#include <svx/zoomtbxctrl.hxx>
 #include <sfx2/docfac.hxx>
 #include <svx/lboxctrl.hxx>
 #include <sfx2/docfile.hxx>
@@ -147,6 +150,7 @@
 	SmViewShell::RegisterFactory(1);
 
 	SvxZoomStatusBarControl::RegisterControl( SID_ATTR_ZOOM, pp );
+	SvxZoomToolBoxControl::RegisterControl( SID_ATTR_ZOOM, pp );
 	SvxModifyControl::RegisterControl( SID_TEXTSTATUS, pp );
 	SvxUndoRedoControl::RegisterControl( SID_UNDO, pp );
 	SvxUndoRedoControl::RegisterControl( SID_REDO, pp );
--- sfx2/inc/sfx2/sfxsids.hrc	2005-02-07 22:41:53.005932054 -0800
+++ sfx2/inc/sfx2/sfxsids.hrc	2005-02-07 22:42:55.732927520 -0800
@@ -1114,6 +1116,7 @@
 
 #define SID_SVX_START                       (SID_LIB_START +   0)
 
+#define SID_ATTR_ZOOM                       (SID_SVX_START +   0)
 #define SID_ATTR_CHAR                       (SID_SVX_START +   6)
 #define SID_ATTR_CHAR_FONT                  (SID_SVX_START +   7)
 #define SID_ATTR_CHAR_POSTURE               (SID_SVX_START +   8)

--- svx/inc/svx/svxids.hrc	8 Sep 2005 19:01:37 -0000	1.102
+++ svx/inc/svx/svxids.hrc	24 Oct 2005 14:48:34 -0000
@@ -434,6 +434,6 @@
 // --------------------------------------------------------------------------
 
-#define SID_ATTR_ZOOM                                   ( SID_SVX_START + 0 )
+//#define SID_ATTR_ZOOM                                   ( SID_SVX_START + 0 )     -> sfxsids.hrc
 #define SID_ATTR_BRUSH                                  ( SID_SVX_START + 1 )
 #define SID_ATTR_TABSTOP                                ( SID_SVX_START + 2 )
 #define SID_ATTR_TABSTOP_DEFAULTS                       ( SID_SVX_START + 3 )

--- svx/source/tbxctrls/makefile.mk	2005-02-10 21:18:28.391273964 -0800
+++ svx/source/tbxctrls/makefile.mk	2005-02-10 21:18:58.339248608 -0800
@@ -97,7 +97,8 @@
         $(SLO)$/toolbarmenu.obj \
         $(SLO)$/tbxcolor.obj \
         $(SLO)$/tbxcolorupdate.obj \
-        $(SLO)$/tbunocontroller.obj
+        $(SLO)$/tbunocontroller.obj \
+        $(SLO)$/zoomtbxctrl.obj	
 
 SRS1NAME=$(TARGET)
 SRC1FILES = \

--- svx/inc/svx/dialogs.hrc	2005-02-02 19:27:13.000000000 +0530
+++ svx/inc/svx/dialogs.hrc	2005-03-29 14:16:08.000000000 +0530
@@ -243,6 +243,8 @@
 
 // fuer ToolboxControl-Style
 #define RID_SVXTBX_STYLE                    (RID_SVX_START + 120)
+// for Zoom Toolbox Control
+#define RID_SVXTBX_ZOOM						(RID_SVX_START + 121)
 
 #define RID_SVXDLG_HYPERLINK                (RID_SVX_START + 140)
 #define RID_SVXDLG_PASSWORD					(RID_SVX_START + 141)
@@ -1133,6 +1135,11 @@
 #define RID_SVXSTR_COLOR_CONFIG_SAVE2       (RID_SVX_START + 854)
 #define RID_SVXSTR_COLOR_CONFIG_DELETE      (RID_SVX_START + 855)
 
+// Zoom ComboBox Strings
+#define RID_SVX_ZOOM_PAGEWIDTH					(RID_SVX_START + 856)
+#define RID_SVX_ZOOM_WHOLEPAGE					(RID_SVX_START + 857)
+#define RID_SVX_ZOOM_OPTIMAL					(RID_SVX_START + 858)
+
 // Accessibility strings.  The actual string ids are defined in
 // svx/inc/accessibility.hrc, the strings are defined in
 // svx/source/accessibility/accessibility.src

--- svx/source/tbxctrls/tbcontrl.src	2006-07-13 18:04:12.000000000 +0200
+++ svx/source/tbxctrls/tbcontrl.src	2007-03-15 16:38:38.000000000 +0100
@@ -129,5 +139,240 @@
     Text [ en-US ] = "Pages";
 };
+String RID_SVX_ZOOM_WHOLEPAGE
+{
+	Text [ de ] = "~Ganze Seite" ;
+	Text [ en-US ] = "~Entire Page" ;
+	Text [ af ] = "~Hele bladsy" ;
+	Text [ ar ] = "~صفحة كاملة" ;
+	Text [ as-IN ] = "সম্পূর্ণ পৃষ্ঠা" ;
+	Text [ be-BY ] = "Старонка цалкам" ;
+	Text [ bg ] = "~Цяла страница" ;
+	Text [ br ] = "Pajenn a-bezh" ;
+	Text [ bs ] = "~Cijela stranica" ;
+	Text [ ca ] = "~Tota la pàgina" ;
+	Text [ cs ] = "Celá stránka" ;
+	Text [ cy ] = "~Tudalen Gyfan" ;
+	Text [ da ] = "H~ele siden" ;
+	Text [ el ] = "Ολόκληρη σελίδα" ;
+	Text [ en-GB ] = "~Entire Page" ;
+	Text [ en-ZA ] = "~Entire Page" ;
+	Text [ eo ] = "Integra Paĝo" ;
+	Text [ es ] = "~Toda la página" ;
+	Text [ et ] = "Terve lehekülg" ;
+	Text [ fa ] = "~سراسر صفحه" ;
+	Text [ fi ] = "~Koko sivu" ;
+	Text [ fr ] = "Page entière" ;
+	Text [ ga ] = "L~eathanach ar fad" ;
+	Text [ gu ] = "આખુ પાનું(~E)" ;
+	Text [ gu-IN ] = "આખુ પાનું(~E)" ;
+	Text [ he ] = "עמוד שלם" ;
+	Text [ hi-IN ] = "सम्पूर्ण पृष्ठ (~E)" ;
+	Text [ hr ] = "~Cijela stranica" ;
+	Text [ hu ] = "~Teljes oldal" ;
+	Text [ it ] = "~Pagina intera" ;
+	Text [ ja ] = "ページ全体(~E)" ;
+	Text [ ka ] = "~მთლიანი გვერდი" ;
+	Text [ km ] = "ទំព័រទាំងមូល " ;
+	Text [ ko ] = "전체 페이지(~E)" ;
+	Text [ lt ] = "~Visas puslapis" ;
+	Text [ mk ] = "~Цела страница" ;
+	Text [ ml-IN ] = "മുഴുവന് പേജ്" ;
+	Text [ mr-IN ] = "~संपूर्ण पृष्ठ" ;
+	Text [ nb ] = "~Hele siden" ;
+	Text [ ne ] = "सम्पूर्ण पृष्ठ" ;
+	Text [ nl ] = "H~ele pagina" ;
+	Text [ nn ] = "~Heile sida" ;
+	Text [ nr ] = "~Ikhasi Loke" ;
+	Text [ ns ] = "~Letlakala Ka moka" ;
+	Text [ or-IN ] = "ସମଗ୍ର ପୃଷ୍ଠା" ;
+	Text [ pa-IN ] = "ਮੁਕੰਮਲ ਸਫ਼ਾ(~E)" ;
+	Text [ pl ] = "~Cała strona" ;
+	Text [ pt ] = "~Página inteira" ;
+	Text [ pt-BR ] = "Página int~eira" ;
+	Text [ ru ] = "Вся страница" ;
+	Text [ rw ] = "Ipaji yose" ;
+	Text [ sh-YU ] = "~Cela strana" ;
+	Text [ sk ] = "~Celá strana" ;
+	Text [ sl ] = "~Cela stran" ;
+	Text [ sr-CS ] = "~Цела страна" ;
+	Text [ ss ] = "~Lonkhe lipheji" ;
+	Text [ st ] = "~Leqephe kaofela" ;
+	Text [ sv ] = "Hela sidan" ;
+	Text [ sw-TZ ] = "Ukurasa Mzima" ;
+	Text [ ta ] = "முழுப்பக்கம்" ;
+	Text [ ta-IN ] = "முழுப்பக்கம்" ;
+	Text [ te-IN ] = "పుటము మొత్తము" ;
+	Text [ tg ] = "Ба тамомми саҳифа" ;
+	Text [ th ] = "~ทั้งหน้า" ;
+	Text [ tn ] = "~Yotlhe Tsebe" ;
+	Text [ tr ] = "Tüm Sayfa" ;
+	Text [ ts ] = "~Pheji hinkwaro" ;
+	Text [ uk ] = "~Entire Page" ;
+	Text [ ur-IN ] = "پوراصفحہ" ;
+	Text [ ve ] = "~Siatari Lothe" ;
+	Text [ vi ] = "~Toàn trang" ;
+	Text [ xh ] = "~Ikhasi Liphelele" ;
+	Text [ zh-CN ] = "整页(~E)" ;
+	Text [ zh-TW ] = "整頁(~E)" ;
+	Text [ zu ] = "Ikhasi ~lonke" ;
+	Text [ x-comment ] = " ";
+};
+String RID_SVX_ZOOM_PAGEWIDTH
+{
+	Text [ de ] = "Seiten~breite" ;
+	Text [ en-US ] = "~Page Width" ;
+	Text [ af ] = "~Bladsybreedte" ;
+	Text [ ar ] = "~عرض الصفحة" ;
+	Text [ as-IN ] = "পৃষ্ঠাৰ প্রস্থ" ;
+	Text [ be-BY ] = "Шырыня старонкі" ;
+	Text [ bg ] = "~Ширина на страницата" ;
+	Text [ br ] = "~Led pajenn" ;
+	Text [ bs ] = "Širina ~stranice" ;
+	Text [ ca ] = "~Ample de pàgina" ;
+	Text [ cs ] = "Šířka stránky" ;
+	Text [ cy ] = "~Lled tudalen" ;
+	Text [ da ] = "Side~bredde" ;
+	Text [ el ] = "Πλάτος σελίδας" ;
+	Text [ en-GB ] = "~Page Width" ;
+	Text [ en-ZA ] = "~Page Width" ;
+	Text [ eo ] = "Paĝa Larĝo" ;
+	Text [ es ] = "~Ancho de página" ;
+	Text [ et ] = "Lehekülje laius" ;
+	Text [ fa ] = "~عرض صفحه" ;
+	Text [ fi ] = "~Sivun leveys" ;
+	Text [ fr ] = "~Largeur de page" ;
+	Text [ ga ] = "Leithead ~Leathanaigh" ;
+	Text [ gu ] = "પાનું પહોળાઈ(~P)" ;
+	Text [ gu-IN ] = "પાનું પહોળાઈ(~P)" ;
+	Text [ he ] = "רוחב עמוד" ;
+	Text [ hi-IN ] = "पृष्ठ चौड़ाई (~P)" ;
+	Text [ hr ] = "~Širina stranice" ;
+	Text [ hu ] = "~Oldalszélesség" ;
+	Text [ it ] = "Larg~h. pagina" ;
+	Text [ ja ] = "ページ幅(~P)" ;
+	Text [ ka ] = "~გვერდის სიგანე" ;
+	Text [ km ] = "ទទឹង​ទំព័រ " ;
+	Text [ ko ] = "페이지 너비(~P)" ;
+	Text [ lt ] = "~Puslapio plotis" ;
+	Text [ mk ] = "Ширина на ~страница" ;
+	Text [ ml-IN ] = "പേജിന്റെ വീതി" ;
+	Text [ mr-IN ] = "~पृष्ठाची रुंदी" ;
+	Text [ nb ] = "~Sidebredde" ;
+	Text [ ne ] = "पृष्ठ चौडाइ" ;
+	Text [ nl ] = "~Paginabreedte" ;
+	Text [ nn ] = "~Sidebreidd" ;
+	Text [ nr ] = "~Ububanzi bekhasi" ;
+	Text [ ns ] = "~Bophara bja Letlakala" ;
+	Text [ or-IN ] = "ପୃଷ୍ଠାର ଓସାର" ;
+	Text [ pa-IN ] = "ਸਫ਼ਾ ਚੌੜਾਈ(~P)" ;
+	Text [ pl ] = "Szerokość ~strony" ;
+	Text [ pt ] = "~Largura da página" ;
+	Text [ pt-BR ] = "Largura da ~página" ;
+	Text [ ru ] = "По ширине страницы" ;
+	Text [ rw ] = "Ubugari bw'Urupapuro" ;
+	Text [ sh-YU ] = "~Širina strane" ;
+	Text [ sk ] = "Šírka ~strany" ;
+	Text [ sl ] = "Širina s~trani" ;
+	Text [ sr-CS ] = "~Ширина стране" ;
+	Text [ ss ] = "~Bubanti belipheji" ;
+	Text [ st ] = "~Bophara ba Leqephe" ;
+	Text [ sv ] = "Sidbredd" ;
+	Text [ sw-TZ ] = "Upana Ukurasa" ;
+	Text [ ta ] = "பக்க அகலம்" ;
+	Text [ ta-IN ] = "பக்க அகலம்" ;
+	Text [ te-IN ] = "పుటము యొక్క వెడెల్పు" ;
+	Text [ tg ] = "Ба бари саҳифа" ;
+	Text [ th ] = "ความกว้างของ~หน้า" ;
+	Text [ tn ] = "~Tsebe Bophara" ;
+	Text [ tr ] = "Sayfa Yüksekliği" ;
+	Text [ ts ] = "~Ku anama ka pheji" ;
+	Text [ uk ] = "~Page Width" ;
+	Text [ ur-IN ] = "صفحہ کی چوڑائی" ;
+	Text [ ve ] = "~Vhuphara ha Siatari" ;
+	Text [ vi ] = "Độ ~rộng trang" ;
+	Text [ xh ] = "~Ububanzi Bekhasi" ;
+	Text [ zh-CN ] = "页面宽度(~P)" ;
+	Text [ zh-TW ] = "頁面寬度(~P)" ;
+	Text [ zu ] = "Ububanzi be~khasi" ;
+	Text [ x-comment ] = " ";
+};
+String RID_SVX_ZOOM_OPTIMAL
+{
+	Text [ de ] = "~Optimal" ;
+	Text [ en-US ] = "~Optimal" ;
+	Text [ af ] = "~Optimale" ;
+	Text [ ar ] = "الأم~ثل" ;
+	Text [ be-BY ] = "Аптымальна" ;
+	Text [ bg ] = "~Оптимален" ;
+	Text [ br ] = "~Gwellek" ;
+	Text [ bs ] = "~Optimalno" ;
+	Text [ ca ] = "Ò~ptim" ;
+	Text [ cs ] = "Optimální" ;
+	Text [ cy ] = "~Gorau" ;
+	Text [ da ] = "~Optimal" ;
+	Text [ el ] = "Βέ~λτιστα" ;
+	Text [ en-GB ] = "~Optimal" ;
+	Text [ en-ZA ] = "~Optimal" ;
+	Text [ eo ] = "~Optimal" ;
+	Text [ es ] = "Ó~ptimo" ;
+	Text [ et ] = "Optimaalne" ;
+	Text [ fa ] = "~بهینه" ;
+	Text [ fi ] = "O~ptimaalinen" ;
+	Text [ fr ] = "~Optimal" ;
+	Text [ ga ] = "~Optamach" ;
+	Text [ gu ] = "શ્રેષ્ઠ (~O)" ;
+	Text [ gu-IN ] = "શ્રેષ્ઠ (~O)" ;
+	Text [ he ] = "מיטבי" ;
+	Text [ hi-IN ] = "~Optimal" ;
+	Text [ hr ] = "~Optimalno" ;
+	Text [ hu ] = "~Optimális" ;
+	Text [ it ] = "~Ottimale" ;
+	Text [ ja ] = "最適(~O)" ;
+	Text [ ka ] = "~ოფტიმალური" ;
+	Text [ km ] = "ប្រសើរ​បំផុត " ;
+	Text [ ko ] = "최적(~O)" ;
+	Text [ ku ] = "D~înamik" ;
+	Text [ lt ] = "~Optimalus" ;
+	Text [ mk ] = "~Оптимално" ;
+	Text [ nb ] = "~Best mulig" ;
+	Text [ ne ] = "इष्टतम" ;
+	Text [ nl ] = "~Optimaal" ;
+	Text [ nn ] = "~Best mogleg" ;
+	Text [ nr ] = "~Ngaletha ipumelelo" ;
+	Text [ ns ] = "~Kgolo" ;
+	Text [ or-IN ] = "ଅପଟିମାଲ" ;
+	Text [ pa-IN ] = "ਅਨੁਕੂਲ(~O)" ;
+	Text [ pl ] = "~Optymalny" ;
+	Text [ pt ] = "~Optimizado" ;
+	Text [ pt-BR ] = "~Ideal" ;
+	Text [ ru ] = "Оптимально" ;
+	Text [ rw ] = "Gihamye" ;
+	Text [ sh-YU ] = "~Optimalan" ;
+	Text [ sk ] = "~Optimálny" ;
+	Text [ sl ] = "~Optimalno" ;
+	Text [ sr-CS ] = "~Оптималан" ;
+	Text [ ss ] = "~Kwekugcina" ;
+	Text [ st ] = "~E fetisisang" ;
+	Text [ sv ] = "Optimal" ;
+	Text [ sw-TZ ] = "Sadifu" ;
+	Text [ ta ] = "~சிறப்பு" ;
+	Text [ ta-IN ] = "~சிறப்பு" ;
+	Text [ te-IN ] = "అనుకూలమైన" ;
+	Text [ tg ] = "Оптималӣ" ;
+	Text [ th ] = "เ~หมาะที่สุด" ;
+	Text [ tn ] = "~Nepagetse" ;
+	Text [ tr ] = "En Uygun" ;
+	Text [ ts ] = "~Mpimo wa le makumu" ;
+	Text [ uk ] = "Оптимальний" ;
+	Text [ ur-IN ] = "مناسب" ;
+	Text [ ve ] = "~Oputimala" ;
+	Text [ vi ] = "Tối ư~u" ;
+	Text [ xh ] = "~Okufaneleke ncam" ;
+	Text [ zh-CN ] = "最佳(~O)" ;
+	Text [ zh-TW ] = "最合適(~O)" ;
+	Text [ zu ] = "~Okuthandwayo" ;
+	Text [ x-comment ] = " ";
+};
 Image RID_SVXIMG_GRAF_RED
 {
 	ImageBitmap = Bitmap { File = "sc10865.bmp" ; };

--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ svx/source/tbxctrls/zoomtbxctrl.cxx	2005-04-01 15:02:35.346989875 +0530
@@ -0,0 +1,281 @@
+/*************************************************************************
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#include <string> // HACK: prevent conflict between STLPORT and Workshop headers
+
+#include <vcl/toolbox.hxx>
+#include <sfx2/app.hxx>
+#include <sfx2/dispatch.hxx>
+#include <sfx2/viewsh.hxx>
+#include <vcl/combobox.hxx>
+                       
+#include <svx/dialogs.hrc>
+
+#include <svx/zoomtbxctrl.hxx>
+#include <svx/zoomitem.hxx>
+#include <svx/dialmgr.hxx>
+
+#define MINZOOM 20
+#define MAXZOOM 600
+#define DEFAULT_ZOOM 100
+
+using namespace ::rtl;
+using namespace ::com::sun::star::uno;
+using namespace ::com::sun::star::beans;
+using namespace ::com::sun::star::frame;
+
+using namespace ::com::sun::star::frame;
+
+SFX_IMPL_TOOLBOX_CONTROL( SvxZoomToolBoxControl, SvxZoomItem);
+
+class SvxZoomBox_Impl : public ComboBox
+{
+    USHORT          nSlotId;
+    BOOL            bRelease;
+    OUString          maWholePage;
+    OUString          maOptimal;
+    OUString          maPageWidth;
+
+    ::com::sun::star::uno::Reference< ::com::sun::star::frame::XDispatchProvider > m_xDispatchProvider;
+
+public:
+    SvxZoomBox_Impl( Window* pParent, USHORT nSlot, const Reference< XDispatchProvider >& rDispatchProvider );
+    ~SvxZoomBox_Impl();
+
+protected:
+    virtual void    Select();
+    virtual long    Notify( NotifyEvent& rNEvt );
+
+    void ReleaseFocus();
+
+};
+
+static String RemoveTilda( USHORT nId )
+{
+    return String( SVX_RESSTR ( nId ) ).EraseAllChars('~');
+}
+
+SvxZoomBox_Impl::SvxZoomBox_Impl( 
+    Window* pParent, 
+    USHORT nSlot, 
+    const Reference< XDispatchProvider >& rDispatchProvider ):
+    ComboBox( pParent, SVX_RES(RID_SVXTBX_ZOOM)),
+    nSlotId(nSlot),
+    bRelease(TRUE),
+    m_xDispatchProvider( rDispatchProvider ),
+    maPageWidth( RemoveTilda ( RID_SVX_ZOOM_PAGEWIDTH ) ),
+    maOptimal( RemoveTilda ( RID_SVX_ZOOM_OPTIMAL  ) ),
+    maWholePage( RemoveTilda ( RID_SVX_ZOOM_WHOLEPAGE ) )
+{
+	EnableAutocomplete( FALSE );
+	USHORT aZoomValues[] =
+	    {25, 50, 75, 100, 150, 200 };
+	for(USHORT i = 0; i < sizeof(aZoomValues)/sizeof(USHORT); i++)
+	{
+	String sEntry = String::CreateFromInt32( aZoomValues[i] );
+	sEntry += '%';
+	InsertEntry( sEntry );
+	}
+	InsertEntry( maPageWidth );
+	InsertEntry( maOptimal );
+	InsertEntry( maWholePage );
+}
+
+SvxZoomBox_Impl::~SvxZoomBox_Impl()
+{}
+
+void    SvxZoomBox_Impl::Select()
+{
+    if ( !IsTravelSelect() )
+    {
+        String sEntry( GetText() );
+        sEntry.EraseAllChars( '%' );
+	SvxZoomType ZoomType; 
+        USHORT nZoom = DEFAULT_ZOOM;
+	if( maWholePage == OUString( sEntry ) )
+	{
+	    nSlotId = SID_SIZE_PAGE;
+	    ZoomType = SVX_ZOOM_WHOLEPAGE;
+	}
+	else if( maPageWidth == OUString( sEntry ) )
+	{
+	    nSlotId = SID_SIZE_PAGE_WIDTH;
+	    ZoomType = SVX_ZOOM_PAGEWIDTH;
+	}
+	else if( maOptimal == OUString( sEntry ) )
+	{
+	    nSlotId = SID_SIZE_ALL;
+	    ZoomType = SVX_ZOOM_OPTIMAL;
+	}
+	else
+	{
+	    nZoom = (USHORT)sEntry.ToInt32();
+	    if(nZoom < MINZOOM)
+		nZoom = MINZOOM;
+	    if(nZoom > MAXZOOM)
+		nZoom = MAXZOOM;
+
+	    nSlotId =  SID_ATTR_ZOOM;
+	    ZoomType = SVX_ZOOM_PERCENT;
+	}
+
+	SvxZoomItem aItem ( ZoomType , nZoom, nSlotId );
+	Any a;
+	Sequence< PropertyValue > aArgs( 1 );
+	aArgs[0].Name = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "Zoom" ));
+	aItem.QueryValue( a );
+	aArgs[0].Value = a;
+	SfxToolBoxControl::Dispatch( 
+	    m_xDispatchProvider,
+	    rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ".uno:Zoom" )),
+	    aArgs );
+    }
+    ReleaseFocus();
+}
+
+
+long SvxZoomBox_Impl::Notify( NotifyEvent& rNEvt )
+{
+    long nHandled = 0;
+    if ( rNEvt.GetType() == EVENT_KEYINPUT )
+    {
+        USHORT nCode = rNEvt.GetKeyEvent()->GetKeyCode().GetCode();
+        switch ( nCode )
+        {
+            case KEY_RETURN:
+            case KEY_TAB:
+            {
+                if ( KEY_TAB == nCode )
+                    bRelease = FALSE;
+                else
+                    nHandled = 1;
+                Select();
+                break;
+            }
+
+            case KEY_ESCAPE:
+                SetText( GetSavedValue() );
+                ReleaseFocus();
+                break;
+        }
+    }
+    else if ( EVENT_LOSEFOCUS == rNEvt.GetType() )
+    {
+        Window* pFocusWin = Application::GetFocusWindow();
+        if ( !HasFocus() && GetSubEdit() != pFocusWin )
+            SetText( GetSavedValue() );
+    }
+
+    return nHandled ? nHandled : ComboBox::Notify( rNEvt );
+}
+
+void SvxZoomBox_Impl::ReleaseFocus()
+{
+    if ( !bRelease )
+    {
+        bRelease = TRUE;
+        return;
+    }
+    SfxViewShell* pCurSh = SfxViewShell::Current();
+
+    if ( pCurSh )
+    {
+        Window* pShellWnd = pCurSh->GetWindow();
+
+        if ( pShellWnd )
+            pShellWnd->GrabFocus();
+    }
+}
+
+SvxZoomToolBoxControl::SvxZoomToolBoxControl(  USHORT          nSlotID,
+    USHORT          nID,
+    ToolBox&        rBox  )
+:  SfxToolBoxControl( nSlotID, nID, rBox )
+{
+}
+
+SvxZoomToolBoxControl::~SvxZoomToolBoxControl()
+{
+}
+
+void SvxZoomToolBoxControl::StateChanged( USHORT nSID,
+                                         SfxItemState eState,
+                                         const SfxPoolItem* pState )
+{
+    USHORT nId = GetId();
+    GetToolBox().EnableItem( nId, (GetItemState(pState) != SFX_ITEM_DISABLED) );
+    SvxZoomBox_Impl* pBox = (SvxZoomBox_Impl*)GetToolBox().GetItemWindow( GetId() );
+    if(SFX_ITEM_AVAILABLE <= eState)
+    {  
+        String sZoom(String::CreateFromInt32(((const SvxZoomItem*)pState)->GetValue()));
+        sZoom += '%';
+        pBox->SetText(sZoom);
+        pBox->SaveValue();
+    }
+}    
+
+Window* SvxZoomToolBoxControl::CreateItemWindow( Window *pParent )
+{
+
+    SvxZoomBox_Impl* pRet = new SvxZoomBox_Impl( pParent, GetSlotId(), Reference< XDispatchProvider >( m_xFrame->getController(), UNO_QUERY ));
+    
+    return pRet;
+}

--- svx/prj/d.lst	2007-05-02 21:06:36.000000000 +0200
+++ svx/prj/d.lst	2007-05-02 21:12:15.000000000 +0200
@@ -425,6 +425,7 @@ mkdir: %_DEST%\inc%_EXT%\svx
 ..\inc\xexch.hxx %_DEST%\inc%_EXT%\svx\xexch.hxx
 ..\inc\zoomctrl.hxx %_DEST%\inc%_EXT%\svx\zoomctrl.hxx
 ..\inc\svx\zoomitem.hxx %_DEST%\inc%_EXT%\svx\zoomitem.hxx
+..\inc\svx\zoomtbxctrl.hxx %_DEST%\inc%_EXT%\svx\zoomtbxctrl.hxx
 ..\inc\svx\viewlayoutitem.hxx %_DEST%\inc%_EXT%\svx\viewlayoutitem.hxx
 ..\inc\svx\zoomslideritem.hxx %_DEST%\inc%_EXT%\svx\zoomslideritem.hxx
 ..\inc\svx\zoomsliderctrl.hxx %_DEST%\inc%_EXT%\svx\zoomsliderctrl.hxx
--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ svx/inc/svx/zoomtbxctrl.hxx	2005-04-07 16:50:49.483537921 +0530
@@ -0,0 +1,90 @@
+/*************************************************************************
+ *
+ *
+ *
+ *
+ *
+ *
+ *
+ *	The Contents of this file are made available subject to the terms of
+ *	either of the following licenses
+ *
+ *		   - GNU Lesser General Public License Version 2.1
+ *		   - Sun Industry Standards Source License Version 1.1
+ *
+ *	Sun Microsystems Inc., October, 2000
+ *
+ *	GNU Lesser General Public License Version 2.1
+ *	=============================================
+ *	Copyright 2000 by Sun Microsystems, Inc.
+ *	901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *	This library is free software; you can redistribute it and/or
+ *	modify it under the terms of the GNU Lesser General Public
+ *	License version 2.1, as published by the Free Software Foundation.
+ *
+ *	This library is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *	Lesser General Public License for more details.
+ *
+ *	You should have received a copy of the GNU Lesser General Public
+ *	License along with this library; if not, write to the Free Software
+ *	Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *	MA	02111-1307	USA
+ *
+ *
+ *	Sun Industry Standards Source License Version 1.1
+ *	=================================================
+ *	The contents of this file are subject to the Sun Industry Standards
+ *	Source License Version 1.1 (the "License"); You may not use this file
+ *	except in compliance with the License. You may obtain a copy of the
+ *	License at http://www.openoffice.org/license.html.
+ *
+ *	Software provided under this License is provided on an "AS IS" basis,
+ *	WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *	WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *	MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *	See the License for the specific provisions governing your rights and
+ *	obligations concerning the Software.
+ *
+ *	The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *	Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *	All Rights Reserved.
+ *
+ *	Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+#ifndef _SVX_ZOOMTBXCTRL_HXX
+#define _SVX_ZOOMTBXCTRL_HXX
+
+#ifndef _SFXTBXCTRL_HXX
+#include <sfx2/tbxctrl.hxx>
+#endif
+
+#ifndef INCLUDED_SVXDLLAPI_H
+#include <svx/svxdllapi.h>
+#endif
+
+class SVX_DLLPUBLIC SvxZoomToolBoxControl : public SfxToolBoxControl
+{
+public:
+	SFX_DECL_TOOLBOX_CONTROL();
+
+	SvxZoomToolBoxControl(  USHORT          nSlotID,
+    USHORT          nID,
+    ToolBox&        rBox
+     );
+	~SvxZoomToolBoxControl();
+
+	virtual void	StateChanged( USHORT nSID,
+								  SfxItemState eState,
+								  const SfxPoolItem* pState );
+
+	virtual Window*	CreateItemWindow( Window *pParent );
+};
+
+#endif // _SVX_ZOOMTBXCTRL_HXX
