Index: source/api2/apinodeaccess.hxx
===================================================================
RCS file: /cvs/util/configmgr/source/api2/apinodeaccess.hxx,v
retrieving revision 1.7
diff -u -p -r1.7 apinodeaccess.hxx
--- configmgr/source/api2/apinodeaccess.hxx	30 Jul 2004 15:07:20 -0000	1.7
+++ configmgr/source/api2/apinodeaccess.hxx	15 Feb 2005 12:58:26 -0000
@@ -66,11 +66,9 @@
 #include "utility.hxx"
 #endif
 
-#if defined(_MSC_VER) && (_MSC_VER >=1310)
 #ifndef CONFIGMGR_CONFIGNODE_HXX_
 #include "noderef.hxx"
 #endif
-#endif
 
 #ifndef CONFIGMGR_ACCESSOR_HXX
 #include <accessor.hxx>
--- config_office/configure.in.gcc4	2005-03-01 09:29:14.000000000 +0000
+++ config_office/configure.in	2005-03-01 09:31:12.000000000 +0000
@@ -895,14 +895,14 @@
    _gcc_longver=`echo $_gcc_version | $AWK -F. '{ print \$1*10000+\$2*100+\$3 }'`
    GCCVER=`echo $_gcc_version | $AWK -F. '{ print \$1*10000+\$2*100+\$3 }'`
 
-   if test "$_gcc_major" = "3"; then
+   if test "$_gcc_major" -lt "3"; then
+      AC_MSG_ERROR([found version "$_gcc_version", use version 3+ of the gcc compiler])
+   else
       if test "$GCCVER" -eq "030203"; then
 	    if test "$ENABLE_SYMBOLS" = "SMALL"; then
            AC_MSG_ERROR([version "$_gcc_version" gives internal error with small.])
 	    fi
       fi
-   else
-      AC_MSG_ERROR([found version "$_gcc_version", use version 3+ of the gcc compiler])
    fi
    AC_MSG_RESULT([checked (gcc $_gcc_version)])
    if test "$_os" = "SunOS"; then
Index: source/drivers/file/FNumericFunctions.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/file/FNumericFunctions.cxx,v
retrieving revision 1.3
diff -u -p -r1.3 FNumericFunctions.cxx
--- connectivity/source/drivers/file/FNumericFunctions.cxx	15 Mar 2004 12:46:43 -0000	1.3
+++ connectivity/source/drivers/file/FNumericFunctions.cxx	1 Mar 2005 14:28:56 -0000
@@ -60,7 +60,6 @@
  ************************************************************************/
 
 #include "file/FNumericFunctions.hxx"
-#include <cmath>
 #include <rtl/math.hxx>
 
 using namespace connectivity;
Index: inc/uno/lbnames.h
===================================================================
RCS file: /cvs/udk/cppu/inc/uno/lbnames.h,v
retrieving revision 1.12
diff -u -p -r1.12 lbnames.h
--- cppu/inc/uno/lbnames.h	30 Jul 2004 15:03:44 -0000	1.12
+++ cppu/inc/uno/lbnames.h	15 Feb 2005 10:36:03 -0000
@@ -96,6 +96,8 @@ provoking error here, because PP ignores
 #error "Tested gcc 2 versions are 2.91 and 2.95.  Patch uno/lbnames.h to try your gcc 2 version."
 #elif (__GNUC__ == 3 && __GNUC_MINOR__ <= 4)
 #define __CPPU_ENV gcc3
+#elif (__GNUC__ == 4 && __GNUC_MINOR__ == 0)
+#define __CPPU_ENV gcc3
 #elif (__GNUC__ == 3)
 #error "Tested gcc 3 version is <= 3.4.  Patch uno/lbnames.h to try your gcc 3 version."
 #else
Index: source/deployment/registry/component/dp_component.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/deployment/registry/component/dp_component.cxx,v
retrieving revision 1.7
diff -u -p -r1.7 dp_component.cxx
--- desktop/source/deployment/registry/component/dp_component.cxx	21 Jan 2005 17:13:33 -0000	1.7
+++ desktop/source/deployment/registry/component/dp_component.cxx	15 Feb 2005 19:17:38 -0000
@@ -114,7 +114,7 @@ class BackendImpl : public ::dp_registry
         const OUString m_loader;
         Reference<XComponentContext> m_xRemoteContext;
         
-        enum {
+        enum reg {
             REG_UNINIT, REG_VOID, REG_REGISTERED, REG_NOT_REGISTERED
         } m_registered;
         
Index: source/deployment/registry/inc/dp_backend.h
===================================================================
RCS file: /cvs/framework/desktop/source/deployment/registry/inc/dp_backend.h,v
retrieving revision 1.7
diff -u -p -r1.7 dp_backend.h
--- desktop/source/deployment/registry/inc/dp_backend.h	21 Jan 2005 17:14:01 -0000	1.7
+++ desktop/source/deployment/registry/inc/dp_backend.h	15 Feb 2005 19:17:38 -0000
@@ -262,7 +262,7 @@ class PackageRegistryBackend
 protected:
     ::rtl::OUString m_context;
     // currently only for library containers:
-    enum {
+    enum context {
         CONTEXT_UNKNOWN,
         CONTEXT_USER, CONTEXT_SHARED,
         CONTEXT_DOCUMENT
Index: source/component/refvaluecomponent.hxx
===================================================================
RCS file: /cvs/gsl/forms/source/component/refvaluecomponent.hxx,v
retrieving revision 1.2
diff -u -p -r1.2 refvaluecomponent.hxx
--- forms/source/component/refvaluecomponent.hxx	16 Nov 2004 10:42:28 -0000	1.2
+++ forms/source/component/refvaluecomponent.hxx	15 Feb 2005 19:13:29 -0000
@@ -74,7 +74,7 @@ namespace frm
 {
 //........................................................................
 
-    enum { STATE_NOCHECK, STATE_CHECK, STATE_DONTKNOW };
+    enum state { STATE_NOCHECK, STATE_CHECK, STATE_DONTKNOW };
 
 	//====================================================================
 	//= OReferenceValueComponent
Index: source/drawdef.h
===================================================================
RCS file: /cvs/sw/hwpfilter/source/drawdef.h,v
retrieving revision 1.2
diff -u -p -r1.2 drawdef.h
--- hwpfilter/source/drawdef.h	16 Feb 2005 18:13:50 -0000	1.2
+++ hwpfilter/source/drawdef.h	1 Mar 2005 11:13:27 -0000
@@ -77,7 +77,7 @@ class HWPPara;
 /**
  * Constant for drawing objects
  */
-enum
+enum hwpdo
 {
     HWPDO_CONTAINER,
     HWPDO_LINE,
Index: source/hbox.h
===================================================================
RCS file: /cvs/sw/hwpfilter/source/hbox.h,v
retrieving revision 1.2
diff -u -p -r1.2 hbox.h
--- hwpfilter/source/hbox.h	16 Feb 2005 18:14:57 -0000	1.2
+++ hwpfilter/source/hbox.h	1 Mar 2005 11:13:29 -0000
@@ -233,7 +233,7 @@ struct Tab: public HBox
 
 // tbox(10) TABLE BOX MATH BUTTON HYPERTEXT
 
-enum { TBL_TYPE, TXT_TYPE, EQU_TYPE, BUTTON_TYPE, HYPERTEXT_TYPE };
+enum hwptype { TBL_TYPE, TXT_TYPE, EQU_TYPE, BUTTON_TYPE, HYPERTEXT_TYPE };
 enum                                              /* TxtBox->margin[n][?] */
 {
     OUT_M, IN_M, CELL_M
@@ -242,7 +242,7 @@ enum                                    
 {
     L_M, R_M, T_M, B_M
 };
-enum { CHAR_ANCHOR, PARA_ANCHOR, PAGE_ANCHOR, PAPER_ANCHOR };
+enum hwpanchor { CHAR_ANCHOR, PARA_ANCHOR, PAGE_ANCHOR, PAPER_ANCHOR };
 enum { TF_OCCUPY_SPACE, TF_TRANSPARENT,TF_ARROUND_TEXT };
 enum
 {
@@ -588,7 +588,7 @@ struct Table
 };
 
 /* picture (11) 그림, OLE그림, 삽입그림, 그리기 */
-enum
+enum hwppictype
 {
     PICTYPE_FILE, PICTYPE_OLE, PICTYPE_EMBED,
     PICTYPE_DRAW, PICTYPE_UNKNOWN
Index: xmlsec1-1.2.4.patch
===================================================================
RCS file: /cvs/external/libxmlsec/xmlsec1-1.2.4.patch,v
retrieving revision 1.11
diff -u -p -r1.11 xmlsec1-1.2.4.patch
--- libxmlsec/xmlsec1-1.2.4.patch	16 Nov 2004 10:29:29 -0000	1.11
+++ libxmlsec/xmlsec1-1.2.4.patch	15 Feb 2005 11:22:23 -0000
@@ -7839,111 +7839,6 @@
 ! 	return(0);
 ! }
 ! 
-! /**
-!  * Block cipher transform final
-!  */
-! static int 
-! xmlSecNssKeyWrapCtxFinal(
-! 	xmlSecNssKeyWrapCtxPtr		ctx ,
-! 	xmlSecBufferPtr 			in ,
-! 	xmlSecBufferPtr 			out ,
-! 	int 						encrypt ,
-! 	xmlSecTransformCtxPtr 		transformCtx
-! ) {
-! 	PK11SymKey*			targetKey ;
-! 	xmlSecSize			blockSize ;
-! 	xmlSecBufferPtr		result ;
-! 
-! 	xmlSecAssert2( ctx != NULL , -1 ) ;
-! 	xmlSecAssert2( ctx->cipher != CKM_INVALID_MECHANISM , -1 ) ;
-! 	xmlSecAssert2( ctx->symkey != NULL , -1 ) ;
-! 	xmlSecAssert2( ctx->keyId != NULL , -1 ) ;
-! 	xmlSecAssert2( ctx->material != NULL , -1 ) ;
-! 	xmlSecAssert2( in != NULL , -1 ) ;
-! 	xmlSecAssert2( out != NULL , -1 ) ;
-! 	xmlSecAssert2( transformCtx != NULL , -1 ) ;
-! 
-! 	/* read raw key material and append into context */
-! 	if( xmlSecBufferAppend( ctx->material, xmlSecBufferGetData(in), xmlSecBufferGetSize(in) ) < 0 ) {
-! 		xmlSecError( XMLSEC_ERRORS_HERE ,
-! 			NULL ,
-! 			"xmlSecBufferAppend" ,
-! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 		return(-1);    
-! 	}
-! 
-! 	if( xmlSecBufferRemoveHead( in , xmlSecBufferGetSize(in) ) < 0 ) {
-! 		xmlSecError( XMLSEC_ERRORS_HERE ,
-! 			NULL ,
-! 			"xmlSecBufferRemoveHead" ,
-! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 		return(-1);    
-! 	}
-! 
-! 	/* Now we get all of the key materail */
-! 	/* from now on we will wrap or unwrap the key */
-! 	if( ( blockSize = PK11_GetBlockSize( ctx->cipher , NULL ) ) < 0 ) {
-! 		xmlSecError( XMLSEC_ERRORS_HERE ,
-! 			NULL ,
-! 			"PK11_GetBlockSize" ,
-! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 		return(-1);    
-! 	}
-! 
-! 	result = xmlSecBufferCreate( blockSize ) ;
-! 	if( result == NULL ) {
-! 		xmlSecError( XMLSEC_ERRORS_HERE ,
-! 			NULL ,
-! 			"xmlSecBufferCreate" ,
-! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 		return(-1);    
-! 	}
-! 
-! 	switch( ctx->cipher ) {
-! 		case CKM_DES3_CBC :
-! 			if( xmlSecNssKeyWrapDesOp(ctx, encrypt, result) < 0 ) {
-! 				xmlSecError( XMLSEC_ERRORS_HERE ,
-! 					NULL ,
-! 					"xmlSecNssKeyWrapDesOp" ,
-! 					XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 					XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 				xmlSecBufferDestroy(result);
-! 				return(-1);    
-! 			}
-! 			break ;
-! 	//	case CKM_NETSCAPE_AES_KEY_WRAP :
-! 		case CKM_AES_CBC :
-! 			if( xmlSecNssKeyWrapAesOp(ctx, encrypt, result) < 0 ) {
-! 				xmlSecError( XMLSEC_ERRORS_HERE ,
-! 					NULL ,
-! 					"xmlSecNssKeyWrapAesOp" ,
-! 					XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 					XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 				xmlSecBufferDestroy(result);
-! 				return(-1);    
-! 			}
-! 			break ;
-! 	}
-! 
-! 	/* Write output */
-! 	if( xmlSecBufferAppend( out, xmlSecBufferGetData(result), xmlSecBufferGetSize(result) ) < 0 ) {
-! 		xmlSecError( XMLSEC_ERRORS_HERE ,
-! 			NULL ,
-! 			"xmlSecBufferAppend" ,
-! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
-! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
-! 		xmlSecBufferDestroy(result);
-! 		return(-1);    
-! 	}
-! 	xmlSecBufferDestroy(result);
-! 
-! 	return(0);
-! }
-! 
 ! static int 
 ! xmlSecNssKWDes3BufferReverse(xmlSecByte *buf, xmlSecSize size) {
 !     xmlSecSize s;
@@ -8398,6 +8293,111 @@
 ! 	return 0 ;
 ! }
 ! 
+! /**
+!  * Block cipher transform final
+!  */
+! static int 
+! xmlSecNssKeyWrapCtxFinal(
+! 	xmlSecNssKeyWrapCtxPtr		ctx ,
+! 	xmlSecBufferPtr 			in ,
+! 	xmlSecBufferPtr 			out ,
+! 	int 						encrypt ,
+! 	xmlSecTransformCtxPtr 		transformCtx
+! ) {
+! 	PK11SymKey*			targetKey ;
+! 	xmlSecSize			blockSize ;
+! 	xmlSecBufferPtr		result ;
+! 
+! 	xmlSecAssert2( ctx != NULL , -1 ) ;
+! 	xmlSecAssert2( ctx->cipher != CKM_INVALID_MECHANISM , -1 ) ;
+! 	xmlSecAssert2( ctx->symkey != NULL , -1 ) ;
+! 	xmlSecAssert2( ctx->keyId != NULL , -1 ) ;
+! 	xmlSecAssert2( ctx->material != NULL , -1 ) ;
+! 	xmlSecAssert2( in != NULL , -1 ) ;
+! 	xmlSecAssert2( out != NULL , -1 ) ;
+! 	xmlSecAssert2( transformCtx != NULL , -1 ) ;
+! 
+! 	/* read raw key material and append into context */
+! 	if( xmlSecBufferAppend( ctx->material, xmlSecBufferGetData(in), xmlSecBufferGetSize(in) ) < 0 ) {
+! 		xmlSecError( XMLSEC_ERRORS_HERE ,
+! 			NULL ,
+! 			"xmlSecBufferAppend" ,
+! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 		return(-1);    
+! 	}
+! 
+! 	if( xmlSecBufferRemoveHead( in , xmlSecBufferGetSize(in) ) < 0 ) {
+! 		xmlSecError( XMLSEC_ERRORS_HERE ,
+! 			NULL ,
+! 			"xmlSecBufferRemoveHead" ,
+! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 		return(-1);    
+! 	}
+! 
+! 	/* Now we get all of the key materail */
+! 	/* from now on we will wrap or unwrap the key */
+! 	if( ( blockSize = PK11_GetBlockSize( ctx->cipher , NULL ) ) < 0 ) {
+! 		xmlSecError( XMLSEC_ERRORS_HERE ,
+! 			NULL ,
+! 			"PK11_GetBlockSize" ,
+! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 		return(-1);    
+! 	}
+! 
+! 	result = xmlSecBufferCreate( blockSize ) ;
+! 	if( result == NULL ) {
+! 		xmlSecError( XMLSEC_ERRORS_HERE ,
+! 			NULL ,
+! 			"xmlSecBufferCreate" ,
+! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 		return(-1);    
+! 	}
+! 
+! 	switch( ctx->cipher ) {
+! 		case CKM_DES3_CBC :
+! 			if( xmlSecNssKeyWrapDesOp(ctx, encrypt, result) < 0 ) {
+! 				xmlSecError( XMLSEC_ERRORS_HERE ,
+! 					NULL ,
+! 					"xmlSecNssKeyWrapDesOp" ,
+! 					XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 					XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 				xmlSecBufferDestroy(result);
+! 				return(-1);    
+! 			}
+! 			break ;
+! 	//	case CKM_NETSCAPE_AES_KEY_WRAP :
+! 		case CKM_AES_CBC :
+! 			if( xmlSecNssKeyWrapAesOp(ctx, encrypt, result) < 0 ) {
+! 				xmlSecError( XMLSEC_ERRORS_HERE ,
+! 					NULL ,
+! 					"xmlSecNssKeyWrapAesOp" ,
+! 					XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 					XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 				xmlSecBufferDestroy(result);
+! 				return(-1);    
+! 			}
+! 			break ;
+! 	}
+! 
+! 	/* Write output */
+! 	if( xmlSecBufferAppend( out, xmlSecBufferGetData(result), xmlSecBufferGetSize(result) ) < 0 ) {
+! 		xmlSecError( XMLSEC_ERRORS_HERE ,
+! 			NULL ,
+! 			"xmlSecBufferAppend" ,
+! 			XMLSEC_ERRORS_R_CRYPTO_FAILED ,
+! 			XMLSEC_ERRORS_NO_MESSAGE ) ;
+! 		xmlSecBufferDestroy(result);
+! 		return(-1);    
+! 	}
+! 	xmlSecBufferDestroy(result);
+! 
+! 	return(0);
+! }
+! 
 ! static int 
 ! xmlSecNssKeyWrapExecute(xmlSecTransformPtr transform, int last, xmlSecTransformCtxPtr transformCtx) {
 ! 	xmlSecNssKeyWrapCtxPtr 	context = NULL ;
Index: inc/Outliner.hxx
===================================================================
RCS file: /cvs/graphics/sd/inc/Outliner.hxx,v
retrieving revision 1.9
diff -u -p -r1.9 Outliner.hxx
--- sd/inc/Outliner.hxx	16 Feb 2005 16:55:34 -0000	1.9
+++ sd/inc/Outliner.hxx	1 Mar 2005 18:58:02 -0000
@@ -223,7 +223,7 @@ public:
 private:
     /// Specifies whether to search and replace, to spell check or to do a
     /// text conversion.
-    enum {SEARCH, SPELL, TEXT_CONVERSION} meMode;
+    enum mode {SEARCH, SPELL, TEXT_CONVERSION} meMode;
 
     /// The view which displays the searched objects.
 	::sd::View* mpView;
Index: inc/lnkbase.hxx
===================================================================
RCS file: /cvs/framework/sfx2/inc/lnkbase.hxx,v
retrieving revision 1.5
diff -u -p -r1.5 lnkbase.hxx
--- sfx2/inc/lnkbase.hxx	18 Jan 2005 15:18:14 -0000	1.5
+++ sfx2/inc/lnkbase.hxx	15 Feb 2005 19:18:17 -0000
@@ -106,7 +106,7 @@ class SvLinkSource;
 #define	OBJECT_CLIENT_GRF			0x91
 #define	OBJECT_CLIENT_OLE			0x92 // embedded link
 
-enum {
+enum sfxlink {
 	// Ole2 compatibel und persistent
 	LINKUPDATE_ALWAYS = 1,
 	LINKUPDATE_ONCALL = 3,
Index: inc/tabdlg.hxx
===================================================================
RCS file: /cvs/framework/sfx2/inc/tabdlg.hxx,v
retrieving revision 1.11
diff -u -p -r1.11 tabdlg.hxx
--- sfx2/inc/tabdlg.hxx	11 Jan 2005 14:53:00 -0000	1.11
+++ sfx2/inc/tabdlg.hxx	15 Feb 2005 19:18:24 -0000
@@ -311,7 +311,7 @@ public:
 	void				SetExchangeSupport( BOOL bNew = TRUE )
 							{ bHasExchangeSupport = bNew; }
 
-	enum {
+	enum sfxpg {
 		KEEP_PAGE = 0x0000,		// Fehlerbehandlung; Seite nicht wechseln
 			// 2. F"ullen eines ItemSets f"ur die Aktualilsierung
 			// "ubergeordneter Beispiele; dieser Pointer kann immer
Index: inc/lnkbase.hxx
===================================================================
RCS file: /cvs/oi/so3/inc/lnkbase.hxx,v
retrieving revision 1.6
diff -u -p -r1.6 lnkbase.hxx
--- so3/inc/lnkbase.hxx	31 Jul 2001 08:13:19 -0000	1.6
+++ so3/inc/lnkbase.hxx	15 Feb 2005 19:05:39 -0000
@@ -97,7 +97,7 @@ class SvLinkSource;
 #define	OBJECT_CLIENT_FILE			0x90
 #define	OBJECT_CLIENT_GRF			0x91
 
-enum {
+enum so3link {
 	// Ole2 compatibel und persistent
 	LINKUPDATE_ALWAYS = 1,
 	LINKUPDATE_ONCALL = 3,
Index: STLport-4.5.patch
===================================================================
RCS file: /cvs/external/stlport/STLport-4.5.patch,v
retrieving revision 1.19
diff -u -p -r1.19 STLport-4.5.patch
--- stlport/STLport-4.5.patch	16 Feb 2005 16:45:56 -0000	1.19
+++ stlport/STLport-4.5.patch	1 Mar 2005 09:42:50 -0000
@@ -277,7 +277,7 @@
   # define _STLP_USE_GLIBC
   #endif
   
-+ # if (__GNUC_MINOR__ < 4)  && (__GNUC__ >= 3)
++ # if (__GNUC_MINOR__ < 4)  && (__GNUC__ == 3)
   #   define _STLP_NO_MEMBER_TEMPLATE_KEYWORD
 + # endif
   
Index: source/invocation/invocation.cxx
===================================================================
RCS file: /cvs/udk/stoc/source/invocation/invocation.cxx,v
retrieving revision 1.13
diff -u -p -r1.13 invocation.cxx
--- stoc/source/invocation/invocation.cxx	27 Mar 2003 12:00:44 -0000	1.13
+++ stoc/source/invocation/invocation.cxx	15 Feb 2005 14:08:19 -0000
@@ -805,7 +805,7 @@ struct MemberItem
 	OUString aName;
 
 	// Defines where the member comes from
-	enum { NAMEACCESS, PROPERTYSET, METHOD } eMode;
+	enum Mode { NAMEACCESS, PROPERTYSET, METHOD } eMode;
 
 	// Index to respective sequence 
 	// (Index to NameAccess sequence for eMode==NAMEACCESS etc.)
Index: source/security/access_controller.cxx
===================================================================
RCS file: /cvs/udk/stoc/source/security/access_controller.cxx,v
retrieving revision 1.5
diff -u -p -r1.5 access_controller.cxx
--- stoc/source/security/access_controller.cxx	19 Apr 2002 15:16:45 -0000	1.5
+++ stoc/source/security/access_controller.cxx	15 Feb 2005 14:08:25 -0000
@@ -445,7 +445,7 @@ class AccessController
         SAL_THROW( (RuntimeException) );
     
     // mode
-    enum { OFF, ON, DYNAMIC_ONLY, SINGLE_USER, SINGLE_DEFAULT_USER } m_mode;
+    enum mode { OFF, ON, DYNAMIC_ONLY, SINGLE_USER, SINGLE_DEFAULT_USER } m_mode;
     
     PermissionCollection m_defaultPermissions;
     // for single-user mode
Index: source/filter/basflt/fltini.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/basflt/fltini.cxx,v
retrieving revision 1.39
diff -u -p -r1.39 fltini.cxx
--- sw/source/filter/basflt/fltini.cxx	27 Jan 2005 11:29:57 -0000	1.39
+++ sw/source/filter/basflt/fltini.cxx	2 Mar 2005 08:45:51 -0000
@@ -806,8 +806,9 @@ void CalculateFlySize(SfxItemSet& rFlySe
 	if( SFX_ITEM_SET != rFlySet.GetItemState( RES_FRM_SIZE, TRUE, &pItem ) ||
 		MINFLY > ((SwFmtFrmSize*)pItem)->GetWidth() )
 	{
-		SwFmtFrmSize aSz( (SwFmtFrmSize&) (pItem ? *pItem
-									: rFlySet.Get( RES_FRM_SIZE, TRUE )) );
+		SwFmtFrmSize aSz((SwFmtFrmSize&)rFlySet.Get(RES_FRM_SIZE, TRUE));
+		if (pItem)
+			aSz = (SwFmtFrmSize&)(*pItem);
 
 		SwTwips nWidth;
 		// dann die Breite des Flys selbst bestimmen. Ist eine Tabelle
Index: source/ui/dbui/dbinsdlg.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/dbui/dbinsdlg.cxx,v
retrieving revision 1.50
diff -u -p -r1.50 dbinsdlg.cxx
--- sw/source/ui/dbui/dbinsdlg.cxx	17 Feb 2005 10:37:38 -0000	1.50
+++ sw/source/ui/dbui/dbinsdlg.cxx	2 Mar 2005 08:49:03 -0000
@@ -319,7 +319,7 @@ const char cDBFldEnd 	= '>';
 // Hilfsstruktur fuers einfuegen von Datenbankspalten als Felder oder Text
 struct _DB_Column
 {
-	enum { DB_FILLTEXT, DB_COL_FIELD, DB_COL_TEXT, DB_SPLITPARA } eColType;
+	enum ColType { DB_FILLTEXT, DB_COL_FIELD, DB_COL_TEXT, DB_SPLITPARA } eColType;
 
 	union {
 		String* pText;
Index: source/ui/inc/wrtsh.hxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/inc/wrtsh.hxx,v
retrieving revision 1.30
diff -u -p -r1.30 wrtsh.hxx
--- sw/source/ui/inc/wrtsh.hxx	21 Jan 2005 16:42:39 -0000	1.30
+++ sw/source/ui/inc/wrtsh.hxx	2 Mar 2005 08:50:46 -0000
@@ -303,7 +303,7 @@ typedef FASTBOOL (SwWrtShell:: *FNSimple
 	// Gemaess den Regeln fuer intelligentes Cut / Paste
 	// werden umgebende Spaces rausgeschnitten.
 	// Liefert Art der Wortselektion zurueck (siehe enum)
-	enum {
+	enum word {
 			NO_WORD = 0,
 			WORD_SPACE_BEFORE = 1,
 			WORD_SPACE_AFTER = 2,
Index: source/parser_i/idl/tk_keyw.cxx
===================================================================
RCS file: /cvs/tools/autodoc/source/parser_i/idl/tk_keyw.cxx,v
retrieving revision 1.5
diff -u -p -r1.5 tk_keyw.cxx
--- autodoc/source/parser_i/idl/tk_keyw.cxx	12 Jul 2004 15:42:51 -0000	1.5
+++ autodoc/source/parser_i/idl/tk_keyw.cxx	7 Mar 2005 18:27:27 -0000
@@ -139,15 +139,15 @@ TokParameterHandling::EV_TokenId   	ev_p
 namespace lux
 {
 
-EnumValueMap &
+template<> EnumValueMap &
 TokBuiltInType::EV_TokenId::Values_()		{ return G_aTokBuiltInType_EV_TokenId_Values; }
-EnumValueMap &
+template<> EnumValueMap &
 TokTypeModifier::EV_TokenId::Values_()		{ return G_aTokTypeModifier_EV_TokenId_Values; }
-EnumValueMap &
+template<> EnumValueMap &
 TokMetaType::EV_TokenId::Values_()			{ return G_aTokMetaType_EV_TokenId_Values; }
-EnumValueMap &
+template<> EnumValueMap &
 TokStereotype::EV_TokenId::Values_()		{ return G_aTokStereotype_EV_TokenId_Values; }
-EnumValueMap &
+template<> EnumValueMap &
 TokParameterHandling::EV_TokenId::Values_()	{ return G_aTokParameterHandling_EV_TokenId_Values; }
 
 }   // namespace lux
Index: source/parser_i/idl/tk_punct.cxx
===================================================================
RCS file: /cvs/tools/autodoc/source/parser_i/idl/tk_punct.cxx,v
retrieving revision 1.4
diff -u -p -r1.4 tk_punct.cxx
--- autodoc/source/parser_i/idl/tk_punct.cxx	27 Jan 2005 11:28:39 -0000	1.4
+++ autodoc/source/parser_i/idl/tk_punct.cxx	7 Mar 2005 18:27:27 -0000
@@ -93,7 +93,7 @@ TokPunctuation::EV_TokenId  Greater(TokP
 
 namespace lux
 {
-EnumValueMap &
+template<> EnumValueMap &
 TokPunctuation::EV_TokenId::Values_()		{ return G_aTokPunctuation_EV_TokenId_Values; }
 }
 
Index: source/parser_i/idoc/tk_atag2.cxx
===================================================================
RCS file: /cvs/tools/autodoc/source/parser_i/idoc/tk_atag2.cxx,v
retrieving revision 1.3
diff -u -p -r1.3 tk_atag2.cxx
--- autodoc/source/parser_i/idoc/tk_atag2.cxx	10 Jun 2003 11:35:20 -0000	1.3
+++ autodoc/source/parser_i/idoc/tk_atag2.cxx	7 Mar 2005 18:27:28 -0000
@@ -90,7 +90,7 @@ Tok_AtTag::EV_TokenId ev_since(Tok_AtTag
 
 namespace lux
 {
-EnumValueMap &
+template<> EnumValueMap &
 Tok_AtTag::EV_TokenId::Values_()	{ return G_aTokAtTag_EV_TokenId_Values; }
 }
 
Index: source/parser_i/idoc/tk_xml.cxx
===================================================================
RCS file: /cvs/tools/autodoc/source/parser_i/idoc/tk_xml.cxx,v
retrieving revision 1.2
diff -u -p -r1.2 tk_xml.cxx
--- autodoc/source/parser_i/idoc/tk_xml.cxx	14 May 2002 09:02:21 -0000	1.2
+++ autodoc/source/parser_i/idoc/tk_xml.cxx	7 Mar 2005 18:27:28 -0000
@@ -94,11 +94,11 @@ Tok_XmlFormat_Tag::EV_TokenId   ev_atom(
 namespace lux
 {
 
-EnumValueMap &
+template<> EnumValueMap &
 Tok_XmlConst::EV_TokenId::Values_()	        { return G_aTok_XmlConst_EV_TokenId_Values; }
-EnumValueMap &
+template<> EnumValueMap &
 Tok_XmlLink_Tag::EV_TokenId::Values_()	    { return G_aTok_XmlLink_Tag_EV_TokenId_Values; }
-EnumValueMap &
+template<> EnumValueMap &
 Tok_XmlFormat_Tag::EV_TokenId::Values_()	{ return G_aTok_XmlFormat_Tag_EV_TokenId_Values; }
 
 }   // namespace lux
Index: runner/makefile.mk
===================================================================
RCS file: /cvs/qa/qadevOOo/runner/makefile.mk,v
retrieving revision 1.8
diff -u -p -r1.8 makefile.mk
--- qadevOOo/runner/makefile.mk	25 Jan 2005 15:39:17 -0000	1.8
+++ qadevOOo/runner/makefile.mk	3 Mar 2005 12:05:15 -0000
@@ -86,10 +86,12 @@ OWNJAR: ALLTAR
 .INCLUDE :  target.mk
 
 OWNJAR: LIGHT
-    +jar cvfm $(CLASSDIR)$/$(TARGET).jar manifest -C $(PRJ) objdsc $(JARCOMMANDS_TESTS) $(JARCOMMANDS_RUNNER)
+#    +jar cvfm $(CLASSDIR)$/$(TARGET).jar manifest -C $(PRJ) objdsc $(JARCOMMANDS_TESTS) $(JARCOMMANDS_RUNNER)
+    +jar cvfm $(CLASSDIR)$/$(TARGET).jar manifest -C $(PRJ) objdsc
 
 LIGHT:
-    +jar cvfm $(CLASSDIR)$/$(TARGET)Light.jar manifest $(JARCOMMANDS_RUNNER)
+#    +jar cvfm $(CLASSDIR)$/$(TARGET)Light.jar manifest $(JARCOMMANDS_RUNNER)
+    +jar cvfm $(CLASSDIR)$/$(TARGET)Light.jar manifest
 
 .ELSE
 all:
