Index: cppu/inc/uno/environment.h
===================================================================
RCS file: /cvs/udk/cppu/inc/uno/environment.h,v
retrieving revision 1.8
retrieving revision 1.8.18.1
diff -u -p -r1.8 -r1.8.18.1
--- cppu/inc/uno/environment.h	21 Apr 2004 12:56:31 -0000	1.8
+++ cppu/inc/uno/environment.h	26 Aug 2004 17:05:59 -0000	1.8.18.1
@@ -172,7 +172,7 @@ typedef void (SAL_CALL * uno_freeProxyFu
 	@param nBytes amount of memory in bytes
 	@return pointer to allocated memory
 */
-typedef void * (SAL_CALL * uno_memAlloc)( sal_uInt32 nBytes );
+typedef void * (SAL_CALL * uno_memAlloc)( sal_Size nBytes );
 
 /** The binary specification of an UNO environment supporting interface registration.
 */
