Index: extensions/source/bibliography/datman.hxx
===================================================================
RCS file: /cvs/util/extensions/source/bibliography/datman.hxx,v
retrieving revision 1.11
retrieving revision 1.11.6.1
diff -u -p -u -r1.11 -r1.11.6.1
--- extensions/source/bibliography/datman.hxx	17 Jun 2004 16:15:42 -0000	1.11
+++ extensions/source/bibliography/datman.hxx	9 Sep 2004 09:55:20 -0000	1.11.6.1
@@ -232,7 +232,7 @@ public:
 		void						saveCtrModel(const ::rtl::OUString& rName,
 													const ::com::sun::star::uno::Reference< ::com::sun::star::awt::XControlModel > & rCtrModel);
 
-		sal_Bool 						moveRelative(long nMove);
+		sal_Bool 						moveRelative(sal_Int32 nMove);
 
 		void						CreateMappingDialog(Window* pParent);
         ::rtl::OUString             CreateDBChangeDialog(Window* pParent);
Index: extensions/source/propctrlr/browserlistbox.hxx
===================================================================
RCS file: /cvs/util/extensions/source/propctrlr/browserlistbox.hxx,v
retrieving revision 1.5
retrieving revision 1.5.72.1
diff -u -p -u -r1.5 -r1.5.72.1
--- extensions/source/propctrlr/browserlistbox.hxx	19 Mar 2004 12:00:33 -0000	1.5
+++ extensions/source/propctrlr/browserlistbox.hxx	9 Sep 2004 09:55:21 -0000	1.5.72.1
@@ -118,7 +118,7 @@ namespace pcr
 				void						UpdatePlayGround();
 				void						UpdateVScroll();
 				void						ShowEntry(sal_uInt16 nPos);
-				void						MoveThumbTo(long nNewThumbPos);
+				void						MoveThumbTo(sal_Int32 nNewThumbPos);
 				void						Resize();
 
 
Index: extensions/source/svg/svgaction.cxx
===================================================================
RCS file: /cvs/util/extensions/source/svg/svgaction.cxx,v
retrieving revision 1.7
retrieving revision 1.7.100.1
diff -u -p -u -r1.7 -r1.7.100.1
--- extensions/source/svg/svgaction.cxx	6 Jan 2004 12:41:25 -0000	1.7
+++ extensions/source/svg/svgaction.cxx	9 Sep 2004 09:55:21 -0000	1.7.100.1
@@ -802,7 +802,7 @@ void SVGActionWriter::ImplWriteGradientE
 // -----------------------------------------------------------------------------
 
 void SVGActionWriter::ImplWriteText( const Point& rPos, const String& rText, 
-									 const long* pDXArray, long nWidth,
+									 const sal_Int32* pDXArray, long nWidth,
 									 const NMSP_RTL::OUString* pStyle )
 {
 	String	aText( rText ); aText.EraseLeadingChars( ' ' );
@@ -811,19 +811,19 @@ void SVGActionWriter::ImplWriteText( con
 	if( nLen )
 	{
 		Size	aNormSize;
-		long*	pOwnArray;
-		long*	pDX;
+		sal_Int32*	pOwnArray;
+		sal_Int32*	pDX;
 
 		// get text sizes
 		if( pDXArray )
 		{
 			pOwnArray = NULL;
 			aNormSize = Size( mpVDev->GetTextWidth( aText ), 0 );
-			pDX = (long*) pDXArray;
+			pDX = pDXArray;
 		}
 		else
 		{	
-			pOwnArray = new long[ nLen ];
+			pOwnArray = new sal_Int32[ nLen ];
 			aNormSize = Size( mpVDev->GetTextArray( aText, pOwnArray ), 0 );
 			pDX = pOwnArray;
 		}
Index: extensions/source/svg/svgaction.hxx
===================================================================
RCS file: /cvs/util/extensions/source/svg/svgaction.hxx,v
retrieving revision 1.2
retrieving revision 1.2.346.1
diff -u -p -u -r1.2 -r1.2.346.1
--- extensions/source/svg/svgaction.hxx	22 Mar 2001 17:49:17 -0000	1.2
+++ extensions/source/svg/svgaction.hxx	9 Sep 2004 09:55:21 -0000	1.2.346.1
@@ -168,7 +168,7 @@ private:
 	void					ImplWritePolygon( const Polygon& rPoly, sal_Bool bLineOnly, const NMSP_RTL::OUString* pStyle = NULL );
 	void					ImplWritePolyPolygon( const PolyPolygon& rPolyPoly, sal_Bool bLineOnly, const NMSP_RTL::OUString* pStyle = NULL );
 	void					ImplWriteGradientEx( const PolyPolygon& rPolyPoly, const Gradient& rGradient, const NMSP_RTL::OUString* pStyle = NULL );
-	void					ImplWriteText( const Point& rPos, const String& rText, const long* pDXArray, long nWidth, const NMSP_RTL::OUString* pStyle = NULL );
+	void					ImplWriteText( const Point& rPos, const String& rText, const sal_Int32* pDXArray, long nWidth, const NMSP_RTL::OUString* pStyle = NULL );
 	void					ImplWriteBmp( const BitmapEx& rBmpEx, const Point& rPt, const Size& rSz, const Point& rSrcPt, const Size& rSrcSz, const NMSP_RTL::OUString* pStyle = NULL );
 
 	void					ImplCheckFontAttributes();
