Index: so3/inc/factory.hxx
===================================================================
RCS file: /cvs/oi/so3/inc/factory.hxx,v
retrieving revision 1.9
retrieving revision 1.9.50.1
diff -u -p -r1.9 -r1.9.50.1
--- so3/inc/factory.hxx	19 Sep 2003 08:38:07 -0000	1.9
+++ so3/inc/factory.hxx	21 Jul 2004 18:50:23 -0000	1.9.50.1
@@ -160,8 +160,8 @@ public:
 	static SvGlobalName GetSvClass( INT32 nFileFormat, const SvGlobalName & rClass );
 	static BOOL			IsIntern31( const SvGlobalName & rClass );
 	static BOOL			IsIntern( const SvGlobalName & rClass,
-							   	  long *pFileFormat	);
-	static SvGlobalName GetServerName( long nStorageFormat );
+							   	  sal_Int32 *pFileFormat	);
+	static SvGlobalName GetServerName( sal_Int32 nStorageFormat );
 	static String	    GetServiceName( const SvGlobalName& );
 
 private:
Index: so3/source/misc/factory.cxx
===================================================================
RCS file: /cvs/oi/so3/source/misc/factory.cxx,v
retrieving revision 1.18
retrieving revision 1.18.16.1
diff -u -p -r1.18 -r1.18.16.1
--- so3/source/misc/factory.cxx	3 May 2004 14:00:16 -0000	1.18
+++ so3/source/misc/factory.cxx	21 Jul 2004 18:50:23 -0000	1.18.16.1
@@ -936,7 +936,7 @@ SvGlobalName SvFactory::GetAutoConvertTo
 //=========================================================================
 SvGlobalName SvFactory::GetSvClass
 (
-	long nFileFormat,
+	sal_Int32 nFileFormat,
 	const SvGlobalName & rClass
 )
 /*  [Beschreibung]
@@ -1046,7 +1046,7 @@ BOOL SvFactory::IsIntern31
 BOOL SvFactory::IsIntern
 (
 	const SvGlobalName & rClass,
- 	long *pFileFormat
+ 	sal_Int32 *pFileFormat
 )
 /*  [Description]
 
@@ -1091,7 +1091,7 @@ BOOL SvFactory::IsIntern
 	return FALSE;
 }
 
-SvGlobalName SvFactory::GetServerName( long nStorageFormat )
+SvGlobalName SvFactory::GetServerName( sal_Int32 nStorageFormat )
 {
 	SvGlobalName aRet;
 
Index: so3/source/persist/persist.cxx
===================================================================
RCS file: /cvs/oi/so3/source/persist/persist.cxx,v
retrieving revision 1.57
retrieving revision 1.56.50.2
diff -u -p -r1.57 -r1.56.50.2
--- so3/source/persist/persist.cxx	13 Jul 2004 07:52:37 -0000	1.57
+++ so3/source/persist/persist.cxx	13 Oct 2004 14:58:35 -0000	1.56.50.2
@@ -840,7 +840,7 @@ BOOL SvPersist::ImplCopy( SvPersist* pSr
     SvStorageRef aSrcEleStor = pSrc->GetStorage();
     SvStorage* pStor = GetStorage();
 
-    long nObjVersion = aSrcEleStor->GetVersion();
+    sal_Int32 nObjVersion = aSrcEleStor->GetVersion();
     BOOL bIntern = SvFactory::IsIntern( aSrcEleStor->GetClassName(), &nObjVersion );
 	if( nObjVersion >= SOFFICE_FILEFORMAT_60 )
 	{
@@ -1862,7 +1862,7 @@ BOOL SvPersist::SaveElement( SvStorage* 
 		pEle->SetClassName( aSrcEleStor->GetClassName() );
 
 	DBG_ASSERT( aSrcEleStor.Is(), "no object storage" );
-    long nObjVersion = aSrcEleStor->GetVersion();
+    sal_Int32 nObjVersion = aSrcEleStor->GetVersion();
 
     // element must be saved as OLEStorage if it currently has an OLE storage or
     // the target storage is an OLEStorage
Index: so3/source/persist/transprt.cxx
===================================================================
RCS file: /cvs/oi/so3/source/persist/transprt.cxx,v
retrieving revision 1.5
retrieving revision 1.5.220.1
diff -u -p -r1.5 -r1.5.220.1
--- so3/source/persist/transprt.cxx	20 Nov 2001 17:16:56 -0000	1.5
+++ so3/source/persist/transprt.cxx	21 Jul 2004 18:50:24 -0000	1.5.220.1
@@ -1130,7 +1130,7 @@ sal_Int32 SAL_CALL UcbTransportInputStre
 	sal_Int32 nSize = 0;
 	while (nSize < nBytesToRead)
 	{
-		ULONG nCount = 0;
+		sal_Size nCount = 0;
 		ErrCode nError = m_xLockBytes->ReadAt (
 			m_nPosition,
 			rData.getArray() + nSize,
@@ -1138,8 +1138,8 @@ sal_Int32 SAL_CALL UcbTransportInputStre
 			&nCount);
 		if (nError != ERRCODE_NONE && nError != ERRCODE_IO_PENDING)
 			throw IOException();
-		m_nPosition += nCount;
-		nSize += nCount;
+		m_nPosition += svstream_Size_to_u32( nCount );
+		nSize += svstream_Size_to_u32( nCount );
 		if (nError == ERRCODE_NONE && nCount == 0)
 			break;
 	}
@@ -1159,7 +1159,7 @@ sal_Int32 SAL_CALL UcbTransportInputStre
 		throw NotConnectedException();
 
 	rData.realloc(nMaxBytesToRead);
-	ULONG nCount;
+	sal_Size nCount;
 	ErrCode nError;
 	do
 	{
@@ -1170,12 +1170,12 @@ sal_Int32 SAL_CALL UcbTransportInputStre
 			&nCount);
 		if (nError != ERRCODE_NONE && nError != ERRCODE_IO_PENDING)
 			throw IOException();
-		m_nPosition += nCount;
+		m_nPosition += svstream_Size_to_u32( nCount );
 	}
 	while (nCount == 0 && nError == ERRCODE_IO_PENDING);
 
-	rData.realloc(sal_Int32(nCount));
-	return sal_Int32(nCount);
+	rData.realloc(sal_Int32(svstream_Size_to_u32(nCount)));
+	return sal_Int32(svstream_Size_to_u32(nCount));
 }
 
 /*
