Index: desktop/source/deployment/gui/dp_gui_dialog.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/deployment/gui/dp_gui_dialog.cxx,v
retrieving revision 1.3
retrieving revision 1.2.12.2
diff -u -p -r1.3 -r1.2.12.2
--- desktop/source/deployment/gui/dp_gui_dialog.cxx	12 Aug 2004 12:04:24 -0000	1.3
+++ desktop/source/deployment/gui/dp_gui_dialog.cxx	13 Oct 2004 21:22:15 -0000	1.2.12.2
@@ -242,14 +242,14 @@ DialogImpl::DialogImpl(
     sal_Int32 statusWidth = that->m_treelb->GetTextWidth(
         that->m_treelb->m_strEnabled );
     statusWidth = ::std::max(
-        statusWidth, that->m_treelb->GetTextWidth(
+        statusWidth, (sal_Int32) that->m_treelb->GetTextWidth(
             that->m_treelb->m_strDisabled ) );
     statusWidth = ::std::max(
-        statusWidth, that->m_treelb->GetTextWidth(
+        statusWidth, (sal_Int32) that->m_treelb->GetTextWidth(
             that->m_treelb->m_strUnknown ) );
     String strStatus( getResourceString(RID_STR_PACKAGE_STATUS) );
     statusWidth = ::std::max(
-        statusWidth, that->m_treelb->GetTextWidth( strStatus ) );
+        statusWidth, (sal_Int32) that->m_treelb->GetTextWidth( strStatus ) );
     statusWidth += 1;
     
     that->m_headerBar->InsertItem(
@@ -257,7 +257,7 @@ DialogImpl::DialogImpl(
         selWidth - statusWidth - vscrollWidth );
     that->m_headerBar->InsertItem( ITEM_ID_STATUS, strStatus, statusWidth );
     
-    sal_Int32 tabs[ 3 ];
+    long tabs[ 3 ];
     tabs[ 0 ] = 2; // two tabs
     tabs[ 1 ] = statusWidth;
     tabs[ 2 ] = selWidth - statusWidth - vscrollWidth;
