Index: tools/bootstrp/command.cxx
===================================================================
RCS file: /cvs/util/tools/bootstrp/command.cxx,v
retrieving revision 1.8
retrieving revision 1.8.178.1
diff -u -p -r1.8 -r1.8.178.1
--- tools/bootstrp/command.cxx	28 Apr 2003 17:20:08 -0000	1.8
+++ tools/bootstrp/command.cxx	6 Oct 2004 11:42:55 -0000	1.8.178.1
@@ -59,10 +59,6 @@
  *
  ************************************************************************/
 
-#ifdef SCO
-#define _IOSTREAM_H
-#endif
-
 #ifdef PRECOMPILED
 #include "first.hxx"
 #endif
@@ -84,7 +80,7 @@
 
 //#define MH_TEST2	1			// fuers direkte Testen
 
-#if defined (DOS) || defined (WNT) || defined (WIN) || defined(OS2)
+#if defined (WNT)
 #include <process.h>    // for _SPAWN
 #endif
 #ifdef UNX
@@ -98,30 +94,20 @@
 #endif
 #define P_WAIT 1 		// erstmal einen dummz
 #endif
-#ifdef MAC
-#define P_WAIT 1 		// erstmal einen dummz
-#endif
 #ifdef WTC
 #define _spawnv spawnv
 #endif
 
-#ifdef OS2
-#include <svpm.h>
-#define _spawnv spawnv
-#endif
-#if defined (WIN) || defined (WNT)
+#if defined (WNT)
 #include <svwin.h>
 #endif
 
-#if defined( WNT ) || defined ( OS2 ) || defined ( WIN ) || defined (DOS )
+#if defined( WNT )
 #define  	cPathSeperator ';'
 #endif
 #ifdef UNX
 #define	 	cPathSeperator	':'
 #endif
-#ifdef MAC
-#define	 	cPathSeperator	','
-#endif
 
 /*****************************************************************************/
 CommandLine::CommandLine(BOOL bWrite)
@@ -444,7 +430,7 @@ void CCommand::ImplInit()
 		(void) strcpy( pTmpStr, aCommandLine.GetToken(i, ' ' ).GetBuffer() );
 		USHORT nStrLen = strlen( pTmpStr ) + 1;
 		strcpy( pChar, pTmpStr );
-		*pPtr = (unsigned int) pChar;
+		*pPtr = (sal_uIntPtr) pChar;
 		pChar += nStrLen;
 		pPtr += 1;
 #ifdef UNX
@@ -457,7 +443,7 @@ void CCommand::ImplInit()
 				aCommandLine.Erase(0, 16);
 			i = nArgc;
 			strcpy( pChar, aCommandLine.GetBuffer() );
-			*pPtr = ( unsigned int ) pChar;
+			*pPtr = (sal_uIntPtr) pChar;
 			pPtr += 1;
 		}
 #endif
@@ -471,15 +457,8 @@ CCommand::operator const int()
 {
 	int nRet;
 #ifndef UNX
-#if defined( WNT ) || defined( DOS ) || defined( WIN )
+#if defined( WNT )
 	nRet = _spawnv( P_WAIT, ppArgv[0], (const char **) ppArgv );
-#else
-#if defined( UNX ) || defined( OS2 )
-	nRet = _spawnv( P_WAIT, ppArgv[0], ppArgv );
-#endif
-#ifdef MAC
-//Mac-Implem.
-#endif
 #endif
 #else
 #ifdef UNX
@@ -505,14 +484,8 @@ CCommand::operator const int()
 	}
 #endif
 
-#ifdef MAC
-//Mac_impl
-=======
 #endif
 
-#endif
-
-#ifndef MAC
 	switch ( errno )
 	{
 		case	E2BIG :
@@ -533,7 +506,6 @@ CCommand::operator const int()
 		default:
 				nError = COMMAND_UNKNOWN;
 	}
-#endif
 
 	if ( nRet )
 		fprintf( stderr, "Program returned with errros\n");
Index: tools/inc/fsys.hxx
===================================================================
RCS file: /cvs/util/tools/inc/fsys.hxx,v
retrieving revision 1.6
retrieving revision 1.6.98.1
diff -u -p -r1.6 -r1.6.98.1
--- tools/inc/fsys.hxx	20 Oct 2003 15:39:20 -0000	1.6
+++ tools/inc/fsys.hxx	10 Sep 2004 13:46:10 -0000	1.6.98.1
@@ -81,6 +81,8 @@
 #include <errcode.hxx>
 #endif
 
+#include <stdarg.h>
+
 #define FEAT_FSYS_DOUBLESPEED
 
 // --------------
Index: tools/inc/pstm.hxx
===================================================================
RCS file: /cvs/util/tools/inc/pstm.hxx,v
retrieving revision 1.2
retrieving revision 1.2.288.1
diff -u -p -r1.2 -r1.2.288.1
--- tools/inc/pstm.hxx	7 Dec 2000 12:43:16 -0000	1.2
+++ tools/inc/pstm.hxx	9 Sep 2004 09:25:33 -0000	1.2.288.1
@@ -100,16 +100,6 @@ public:
 };
 
 /************************** S v R t t i B a s e **************************/
-/*
-#if defined (DOS) && defined (STC)
-#ifdef WIN
-#error Fuer Win muss der Kram virtual sein (MM/MH)
-#endif
-class SvRttiBase : public SvRefBase
-#else
-class SvRttiBase : virtual public SvRefBase
-#endif
-*/
 class SvRttiBase : public SvRefBase
 {
 public:
@@ -257,9 +247,9 @@ class SvPersistStream : public SvStream
     const SvPersistStream * pRefStm;
     UINT32                  nFlags;
 
-    virtual ULONG       GetData( void* pData, ULONG nSize );
-    virtual ULONG       PutData( const void* pData, ULONG nSize );
-    virtual ULONG       SeekPos( ULONG nPos );
+    virtual sal_Size    GetData( void* pData, sal_Size nSize );
+    virtual sal_Size    PutData( const void* pData, sal_Size nSize );
+    virtual sal_Size    SeekPos( sal_Size nPos );
     virtual void        FlushData();
 protected:
     ULONG               GetCurMaxIndex( const SvPersistUIdx & ) const;
Index: tools/inc/solar.h
===================================================================
RCS file: /cvs/util/tools/inc/solar.h,v
retrieving revision 1.38
retrieving revision 1.36.4.2
diff -u -p -r1.38 -r1.36.4.2
--- tools/inc/solar.h	4 Oct 2004 19:59:56 -0000	1.38
+++ tools/inc/solar.h	13 Oct 2004 15:00:40 -0000	1.36.4.2
@@ -101,7 +101,7 @@
 typedef sal_Bool		BOOL;	/* typedef unsigned char   BOOL; */
 typedef sal_uInt8		BYTE;	/* typedef unsigned char   BYTE; */
 typedef sal_uInt16		USHORT;	/* typedef unsigned short  USHORT; */
-typedef sal_uInt32		ULONG;	/* typedef unsigned long   ULONG; */
+typedef sal_uIntPtr		ULONG;	/* typedef unsigned long   ULONG; */
 typedef int 			FASTBOOL;
 
 #ifndef VCL_NEED_BASETSD
Index: tools/inc/stream.hxx
===================================================================
RCS file: /cvs/util/tools/inc/stream.hxx,v
retrieving revision 1.6
retrieving revision 1.6.4.3
diff -u -p -r1.6 -r1.6.4.3
--- tools/inc/stream.hxx	17 Jun 2004 13:10:29 -0000	1.6
+++ tools/inc/stream.hxx	9 Sep 2004 14:15:37 -0000	1.6.4.3
@@ -78,6 +78,8 @@
 #include <rtti.hxx>
 #endif
 
+#include <limits>
+
 class StreamData;
 
 // ------------------------
@@ -92,6 +94,30 @@ inline rtl_TextEncoding GetStoreCharSet(
 		return eEncoding;
 }
 
+// ------------------
+// - 64-bit helpers -
+// ------------------
+
+// The only reason these inlines exist is to make conversion
+// to full 64-bit stream APIs easier later on.  If all places
+// that make direct use of the stream APIs use these inlines
+// to convert back to 32-bit numbers, we should
+// just be able to grep the code for places that depend on
+// 32-bit interaction with SvStream.
+
+inline sal_uInt32 svstream_Size_to_u32( sal_Size n )
+{
+	OSL_ENSURE((sal_uInt32)n <= ::std::numeric_limits< sal_uInt32 >::max(),
+		"svstream_Size_to_u32(): WARNING! truncation will during cast from 64-bit to 32-bit.");
+	return (sal_uInt32) n;
+}
+
+inline sal_Size svstream_u32_to_Size( sal_uInt32 n )
+{
+	return (sal_Size) n;
+}
+
+
 // ---------------
 // - StreamTypes -
 // ---------------
@@ -398,6 +424,9 @@ public:
 	SvStream&		operator>>( sal_uInt16& rUInt16 );
 	SvStream&		operator>>( sal_uInt32& rUInt32 );
 	SvStream&		operator>>( long& rLong );
+#if SAL_TYPES_SIZEOFLONG == 8
+	SvStream&		operator>>( sal_Size& rSize );
+#endif
 	SvStream&		operator>>( short& rShort );
 	SvStream&		operator>>( int& rInt );
 	SvStream&		operator>>( signed char& rChar );
@@ -416,6 +445,9 @@ public:
 	SvStream&		operator<<( sal_uInt16 nUInt16 );
 	SvStream&		operator<<( sal_uInt32 nUInt32 );
 	SvStream&		operator<<( long nLong );
+#if SAL_TYPES_SIZEOFLONG == 8
+	SvStream&		operator<<( sal_Size nSize );
+#endif
 	SvStream&		operator<<( short nShort );
 	SvStream&		operator<<( int nInt );
 	SvStream&		operator<<( signed char nChar );
Index: tools/source/debug/debug.cxx
===================================================================
RCS file: /cvs/util/tools/source/debug/debug.cxx,v
retrieving revision 1.7
retrieving revision 1.7.120.3
diff -u -p -r1.7 -r1.7.120.3
--- tools/source/debug/debug.cxx	7 Aug 2003 11:57:20 -0000	1.7
+++ tools/source/debug/debug.cxx	10 Sep 2004 13:42:45 -0000	1.7.120.3
@@ -71,6 +71,7 @@
 
 #include <time.h>
 #include <cstdarg>  // combinations
+#include <stdarg.h> // some platforms need cstdarg, other stdarg.h
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
Index: tools/source/generic/line.cxx
===================================================================
RCS file: /cvs/util/tools/source/generic/line.cxx,v
retrieving revision 1.2
retrieving revision 1.2.194.1
diff -u -p -r1.2 -r1.2.194.1
--- tools/source/generic/line.cxx	27 Mar 2003 17:04:14 -0000	1.2
+++ tools/source/generic/line.cxx	10 Sep 2004 14:02:48 -0000	1.2.194.1
@@ -74,7 +74,7 @@
 #endif
 
 #include <cstdlib>
-#include <cmath>
+#include <math.h>
 
 inline long FRound( double fVal )
 {
Index: tools/source/memtools/mempool.cxx
===================================================================
RCS file: /cvs/util/tools/source/memtools/mempool.cxx,v
retrieving revision 1.3
retrieving revision 1.3.4.2
diff -u -p -r1.3 -r1.3.4.2
--- tools/source/memtools/mempool.cxx	17 Jun 2004 13:12:06 -0000	1.3
+++ tools/source/memtools/mempool.cxx	10 Sep 2004 13:41:45 -0000	1.3.4.2
Index: tools/source/ref/pstm.cxx
===================================================================
RCS file: /cvs/util/tools/source/ref/pstm.cxx,v
retrieving revision 1.3
retrieving revision 1.3.64.1
diff -u -p -r1.3 -r1.3.64.1
--- tools/source/ref/pstm.cxx	26 Feb 2004 13:21:01 -0000	1.3
+++ tools/source/ref/pstm.cxx	9 Sep 2004 09:26:10 -0000	1.3.64.1
@@ -341,7 +341,7 @@ void SvPersistStream::ResetError()
 /*************************************************************************
 |*	  SvPersistStream::GetData()
 *************************************************************************/
-ULONG SvPersistStream::GetData( void* pData, ULONG nSize )
+sal_Size SvPersistStream::GetData( void* pData, sal_Size nSize )
 {
 	DBG_ASSERT( pStm, "stream not set" )
 	ULONG nRet = pStm->Read( pData, nSize );
@@ -352,7 +352,7 @@ ULONG SvPersistStream::GetData( void* pD
 /*************************************************************************
 |*	  SvPersistStream::PutData()
 *************************************************************************/
-ULONG SvPersistStream::PutData( const void* pData, ULONG nSize )
+sal_Size SvPersistStream::PutData( const void* pData, sal_Size nSize )
 {
 	DBG_ASSERT( pStm, "stream not set" )
 	ULONG nRet = pStm->Write( pData, nSize );
@@ -363,7 +363,7 @@ ULONG SvPersistStream::PutData( const vo
 /*************************************************************************
 |*	  SvPersistStream::Seek()
 *************************************************************************/
-ULONG SvPersistStream::SeekPos( ULONG nPos )
+sal_Size SvPersistStream::SeekPos( sal_Size nPos )
 {
 	DBG_ASSERT( pStm, "stream not set" )
 	ULONG nRet = pStm->Seek( nPos );
Index: tools/source/stream/stream.cxx
===================================================================
RCS file: /cvs/util/tools/source/stream/stream.cxx,v
retrieving revision 1.17
retrieving revision 1.16.4.4
diff -u -p -r1.17 -r1.16.4.4
--- tools/source/stream/stream.cxx	8 Sep 2004 16:01:59 -0000	1.17
+++ tools/source/stream/stream.cxx	13 Oct 2004 15:05:09 -0000	1.16.4.4
@@ -120,6 +120,10 @@ inline static void SwapLong( long& r )
     {   r = SWAPLONG(r);   }
 inline static void SwapULong( sal_uInt32& r )
     {   r = SWAPLONG(r);   }
+#if SAL_TYPES_SIZEOFLONG == 8
+inline static void SwapSize( sal_Size& r )
+    {   r = SWAPLONG(r);   }
+#endif
 inline static void SwapLongInt( int& r )
     {   r = SWAPLONG(r);   }
 inline static void SwapLongUInt( unsigned int& r )
@@ -178,7 +182,7 @@ int tmp = eIOMode; \
 if( (tmp == STREAM_IO_READ) && sizeof(datatype)<=nBufFree) \
 {\
     for (int i = 0; i < sizeof(datatype); i++)\
-        ((char *)&r)[i] = pBufPos[i];\
+        ((char *)&value)[i] = pBufPos[i];\
     nBufActualPos += sizeof(datatype);\
     pBufPos += sizeof(datatype);\
     nBufFree -= sizeof(datatype);\
@@ -1158,6 +1162,7 @@ SvStream& SvStream::operator>> ( sal_uIn
     return *this;
 }
 
+
 SvStream& SvStream::operator >> ( long& r )
 {
 #if(SAL_TYPES_SIZEOFLONG != 4)
@@ -1174,6 +1179,29 @@ SvStream& SvStream::operator >> ( long& 
     return *this;
 }
 
+#if SAL_TYPES_SIZEOFLONG == 8
+SvStream& SvStream::operator >> ( sal_Size& r )
+{
+
+    DBG_ASSERT( FALSE, "SvStream operator >> sal_Size not yet implemented!\n" );
+/*
+#if(SAL_TYPES_SIZEOFLONG != 4)
+    int tmp = r;
+    READNUMBER_WITHOUT_SWAP(sal_Size,tmp)
+    r = tmp;
+    if( bSwap )
+        SwapLong(r);
+#else
+    READNUMBER_WITHOUT_SWAP(long,r)
+    if( bSwap )
+        SwapLong(r);
+#endif
+*/
+    return *this;
+}
+
+#endif // SAL_TYPES_SIZEOFLONG == 8
+
 SvStream& SvStream::operator >> ( short& r )
 {
     READNUMBER_WITHOUT_SWAP(short,r)
@@ -1315,6 +1343,24 @@ SvStream& SvStream::operator<< ( long v 
     return *this;
 }
 
+#if(SAL_TYPES_SIZEOFLONG == 8)
+SvStream& SvStream::operator<< ( sal_Size v )
+{
+#if(SAL_TYPES_SIZEOFLONG != 4)
+    int tmp = v;
+    if( bSwap )
+        SwapInt(tmp);
+    WRITENUMBER_WITHOUT_SWAP(int,tmp)
+#else
+    if( bSwap )
+        SwapSize(v);
+    WRITENUMBER_WITHOUT_SWAP(long,v)
+#endif
+    return *this;
+}
+
+#endif
+
 SvStream& SvStream::operator<<  ( short v )
 {
     if( bSwap )
@@ -2023,7 +2069,7 @@ SvStream& SvStream::ReadNumber( sal_uInt
     }
     char *pEndPtr;
     rUInt32 = strtoul( buf, &pEndPtr, (int)nRadix );
-    nFPtr += ( (sal_uInt32)pEndPtr - (sal_uInt32)buf );
+    nFPtr += ( (sal_uIntPtr)pEndPtr - (sal_uIntPtr)buf );
     Seek( nFPtr );
     bIsEof = FALSE;
     return *this;
