--- jvmfwk/plugins/sunmajor/pluginlib/sunjre.cxx.orig	2006-05-21 21:05:55.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/sunjre.cxx	2006-05-21 21:45:05.000000000 +0200
@@ -90,6 +91,7 @@
     static char const * ar[] = {
 
         "/lib/" JFW_PLUGIN_ARCH "/client",
+        "/lib/" JFW_PLUGIN_ARCH "/server",
         "/lib/" JFW_PLUGIN_ARCH "/native_threads", 
         "/lib/" JFW_PLUGIN_ARCH
 
