Index: basegfx/prj/d.lst
===================================================================
RCS file: /cvs/graphics/basegfx/prj/d.lst,v
retrieving revision 1.19
retrieving revision 1.19.30.1
diff -u -p -r1.19 -r1.19.30.1
--- basegfx/prj/d.lst	16 Feb 2004 17:03:10 -0000	1.19
+++ basegfx/prj/d.lst	10 Sep 2004 15:26:57 -0000	1.19.30.1
@@ -2,7 +2,7 @@
 
 ..\%__SRC%\bin\basegfx?????.dll %_DEST%\bin%_EXT%\basegfx?????.dll
 
-..\%__SRC%\lib\libbasegfx?????.* %_DEST%\lib%_EXT%\libbasegfx?????.*
+..\%__SRC%\lib\libbasegfx*.so %_DEST%\lib%_EXT%\libbasegfx*.so
 ..\%__SRC%\lib\*.a %_DEST%\lib%_EXT%\*.a
 ..\%__SRC%\lib\*.dylib %_DEST%\lib%_EXT%\*.dylib
 
