Index: framework/inc/recording/dispatchrecorder.hxx
===================================================================
RCS file: /cvs/framework/framework/inc/recording/dispatchrecorder.hxx,v
retrieving revision 1.6
retrieving revision 1.6.222.1
diff -u -p -u -r1.6 -r1.6.222.1
--- framework/inc/recording/dispatchrecorder.hxx	25 Mar 2003 18:19:49 -0000	1.6
+++ framework/inc/recording/dispatchrecorder.hxx	21 Jul 2004 18:34:36 -0000	1.6.222.1
@@ -126,9 +126,9 @@ class DispatchRecorder
 
 	virtual sal_Int32 SAL_CALL getCount() throw (::com::sun::star::uno::RuntimeException);
 
-	virtual com::sun::star::uno::Any SAL_CALL getByIndex(long int)  throw (com::sun::star::uno::RuntimeException, com::sun::star::lang::WrappedTargetException, com::sun::star::lang::IndexOutOfBoundsException);
+	virtual com::sun::star::uno::Any SAL_CALL getByIndex(sal_Int32)  throw (com::sun::star::uno::RuntimeException, com::sun::star::lang::WrappedTargetException, com::sun::star::lang::IndexOutOfBoundsException);
 
-	virtual void SAL_CALL replaceByIndex(long int, const com::sun::star::uno::Any&)  throw (::com::sun::star::lang::IllegalArgumentException, ::com::sun::star::lang::IndexOutOfBoundsException, ::com::sun::star::lang::WrappedTargetException, ::com::sun::star::uno::RuntimeException);
+	virtual void SAL_CALL replaceByIndex(sal_Int32, const com::sun::star::uno::Any&)  throw (::com::sun::star::lang::IllegalArgumentException, ::com::sun::star::lang::IndexOutOfBoundsException, ::com::sun::star::lang::WrappedTargetException, ::com::sun::star::uno::RuntimeException);
 
     // private functions
     private:
Index: framework/source/uiconfiguration/moduleuiconfigurationmanager.cxx
===================================================================
RCS file: /cvs/framework/framework/source/uiconfiguration/moduleuiconfigurationmanager.cxx,v
retrieving revision 1.9
retrieving revision 1.4.14.2
diff -u -p -u -r1.9 -r1.4.14.2
--- framework/source/uiconfiguration/moduleuiconfigurationmanager.cxx	4 Oct 2004 14:41:27 -0000	1.9
+++ framework/source/uiconfiguration/moduleuiconfigurationmanager.cxx	12 Oct 2004 13:26:05 -0000	1.4.14.2
@@ -918,7 +918,7 @@ void SAL_CALL ModuleUIConfigurationManag
             Reference< XPropertySet > xPropSet( m_xUserConfigStorage, UNO_QUERY );
             if ( xPropSet.is() )
             {
-                long nOpenMode;
+                sal_Int32 nOpenMode;
                 Any a = xPropSet->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "OpenMode" )));
                 if ( a >>= nOpenMode )
                     m_bReadOnly = !( nOpenMode & ElementModes::WRITE );
Index: framework/source/uiconfiguration/uiconfigurationmanager.cxx
===================================================================
RCS file: /cvs/framework/framework/source/uiconfiguration/uiconfigurationmanager.cxx,v
retrieving revision 1.9
retrieving revision 1.4.14.2
diff -u -p -u -r1.9 -r1.4.14.2
--- framework/source/uiconfiguration/uiconfigurationmanager.cxx	20 Sep 2004 10:09:23 -0000	1.9
+++ framework/source/uiconfiguration/uiconfigurationmanager.cxx	12 Oct 2004 13:25:54 -0000	1.4.14.2
@@ -1224,7 +1224,7 @@ void SAL_CALL UIConfigurationManager::se
         {
             try
             {
-                long nOpenMode;
+                sal_Int32 nOpenMode;
                 Any a = xPropSet->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "OpenMode" )));
                 if ( a >>= nOpenMode )
                     m_bReadOnly = !( nOpenMode & ElementModes::WRITE );
