--- sfx2/source/explorer/exptypes.cxx	2006-02-16 15:23:01.000000000 +0000
+++ sfx2/source/explorer/exptypes.cxx	2006-02-16 15:22:11.000000000 +0000
@@ -162,8 +162,8 @@
 	BigInt nSize2 = nSize1;
 	String aSecondUnitStr( ' ' );
 	BOOL bGB = FALSE;
-	ULONG nMega = 1024 * 1024;
-	ULONG nGiga = nMega * 1024;
+	sal_uInt32 nMega = 1024 * 1024;
+	sal_uInt32 nGiga = nMega * 1024;
 
 	static const BigInt nTenKB( 10000 );
 	static const BigInt nOneMB( nMega );
@@ -391,8 +391,8 @@
 	BigInt nSize = rSize;
 	double fSize = nSize;
 
-	ULONG nMega = 1024 * 1024;
-	ULONG nGiga = nMega * 1024;
+	sal_uInt32 nMega = 1024 * 1024;
+	sal_uInt32 nGiga = nMega * 1024;
 	static const BigInt nTenKB( 10000 );
 	static const BigInt nOneMB( nMega );
 	static const BigInt nOneGB( nGiga );
