--- xmerge/java/org/openoffice/xmerge/converter/xml/xslt/DocumentDeserializerImpl.java.orig	2004-12-22 17:22:27.982258021 +0100
+++ xmerge/java/org/openoffice/xmerge/converter/xml/xslt/DocumentDeserializerImpl.java	2004-12-22 17:22:39.704173178 +0100
@@ -155,14 +155,14 @@ public final class DocumentDeserializerI
      */
     public Document deserialize() throws ConvertException, IOException {
 	log("\nFound the XSLT deserializer");
-	Enumeration enum = cd.getDocumentEnumeration();
+	Enumeration enumer = cd.getDocumentEnumeration();
 	org.w3c.dom.Document domDoc=null;
 	DOMDocument docOut=null;
 	GenericOfficeDocument doc = null;
 	ByteArrayOutputStream baos =null;
 	GenericOfficeDocument sxwDoc = new GenericOfficeDocument("output");
-	while (enum.hasMoreElements()) {       
-		 docOut = (DOMDocument) enum.nextElement();
+	while (enumer.hasMoreElements()) {       
+		 docOut = (DOMDocument) enumer.nextElement();
 	}
 	domDoc = docOut.getContentDOM();	
 	try{
--- xmerge/java/org/openoffice/xmerge/converter/xml/xslt/PluginFactoryImpl.java.orig	2004-12-22 17:23:19.555086519 +0100
+++ xmerge/java/org/openoffice/xmerge/converter/xml/xslt/PluginFactoryImpl.java	2004-12-22 17:23:31.172021027 +0100
@@ -183,9 +183,9 @@ public final class PluginFactoryImpl ext
 	String ext= ".txt";
 	String mimeType = null;
 	ConverterInfo ci = this.getConverterInfo();
-	Enumeration enum = ci.getDeviceMime();
-	while (enum.hasMoreElements()) {       
-	    mimeType= (String) enum.nextElement();
+	Enumeration enumer = ci.getDeviceMime();
+	while (enumer.hasMoreElements()) {       
+	    mimeType= (String) enumer.nextElement();
 	}
 	try {
 	    props.load(is);
--- xmerge/java/org/openoffice/xmerge/ConvertData.java.orig	2004-12-22 17:24:06.098811949 +0100
+++ xmerge/java/org/openoffice/xmerge/ConvertData.java	2004-12-22 17:24:15.936063386 +0100
@@ -127,8 +127,8 @@ public class ConvertData {
      *           <code>Vector</code> of <code>Document</code> objects.
      */
     public Enumeration getDocumentEnumeration() {
-        Enumeration enum = v.elements();
-		return (enum);
+        Enumeration enumer = v.elements();
+		return (enumer);
 	}
 
 
