--- dbaccess/source/ui/app/AppDetailPageHelper.cxx	2004-11-16 10:49:09.000000000 +0100
+++ dbaccess/source/ui/app/AppDetailPageHelper.cxx	2005-01-05 11:03:56.988049918 +0100
@@ -534,7 +534,7 @@ sal_Bool OAppDetailPageHelper::isLeaf(Sv
 	if ( nPos < CONTROL_COUNT && _pEntry )
 	{
 		DBTreeListBox& rTree = *m_pLists[nPos];
-		bLeafSelected = reinterpret_cast<sal_Int32>(_pEntry->GetUserData()) != FOLDER_TYPE;
+		bLeafSelected = reinterpret_cast<sal_IntPtr>(_pEntry->GetUserData()) != FOLDER_TYPE;
 	}
 	return bLeafSelected;
 }
@@ -549,7 +549,7 @@ sal_Bool OAppDetailPageHelper::isALeafSe
 		SvLBoxEntry* pEntry = rTree.FirstSelected( );
 		while( !bLeafSelected && pEntry )
 		{
-			bLeafSelected = reinterpret_cast<sal_Int32>(pEntry->GetUserData()) != FOLDER_TYPE;
+			bLeafSelected = reinterpret_cast<sal_IntPtr>(pEntry->GetUserData()) != FOLDER_TYPE;
 			pEntry = rTree.NextSelected(pEntry);
 		}
 	}
--- dbaccess/source/ui/control/tabletree.cxx	2004-11-16 10:49:11.000000000 +0100
+++ dbaccess/source/ui/control/tabletree.cxx	2005-01-05 11:03:56.989049743 +0100
@@ -153,7 +153,7 @@ void OTableTreeListBox::notifyHiContrast
 	SvLBoxEntry* pEntryLoop = First();
 	while (pEntryLoop)
 	{
-		sal_Int32 nType = reinterpret_cast<sal_Int32>(pEntryLoop->GetUserData());
+		sal_Int32 nType = reinterpret_cast<sal_IntPtr>(pEntryLoop->GetUserData());
 		if ( nType )
 		{
 			USHORT nCount = pEntryLoop->ItemCount();
--- dbaccess/source/ui/dlg/generalpage.cxx	2004-11-16 10:49:11.000000000 +0100
+++ dbaccess/source/ui/dlg/generalpage.cxx	2005-01-05 11:12:03.072098005 +0100
@@ -482,7 +482,7 @@ namespace dbaui
         else
         {
 		    USHORT nEntry = m_aDatasourceType.GetSelectEntryPos();
-			DATASOURCE_TYPE eSelectedType = static_cast<DATASOURCE_TYPE>(reinterpret_cast<sal_Int32>(m_aDatasourceType.GetEntryData(nEntry)));
+			DATASOURCE_TYPE eSelectedType = static_cast<DATASOURCE_TYPE>(reinterpret_cast<sal_IntPtr>(m_aDatasourceType.GetEntryData(nEntry)));
 			if (m_DBWizardMode)
 			{
 				sal_Bool bCurCreationMode = IsDatabaseToBeCreated();
@@ -511,7 +511,7 @@ namespace dbaui
 	{
 		// get the type from the entry data
 		sal_Int16 nSelected = _pBox->GetSelectEntryPos();
-		DATASOURCE_TYPE eSelectedType = static_cast<DATASOURCE_TYPE>(reinterpret_cast<sal_Int32>(_pBox->GetEntryData(nSelected)));
+		DATASOURCE_TYPE eSelectedType = static_cast<DATASOURCE_TYPE>(reinterpret_cast<sal_IntPtr>(_pBox->GetEntryData(nSelected)));
 
 		setParentTitle(eSelectedType);
 		// let the impl method do all the stuff
--- dbaccess/source/ui/inc/QueryDesignView.hxx	2004-11-16 10:49:13.000000000 +0100
+++ dbaccess/source/ui/inc/QueryDesignView.hxx	2005-01-05 11:03:56.991049393 +0100
@@ -180,7 +180,7 @@ namespace dbaui
 		// called when a table from tabeview was deleted
 		void TableDeleted(const ::rtl::OUString& rAliasName);
 
-		BOOL getColWidth( const ::rtl::OUString& rAliasName, const ::rtl::OUString& rFieldName, ULONG& nWidth );
+		BOOL getColWidth( const ::rtl::OUString& rAliasName, const ::rtl::OUString& rFieldName, sal_uInt32& nWidth );
 		void fillValidFields(const ::rtl::OUString& strTableName, ComboBox* pFieldList);
 		void zoomTableView(const Fraction& _rFraction);
 
--- dbaccess/source/ui/tabledesign/TEditControl.hxx	2003-03-19 18:53:02.000000000 +0100
+++ dbaccess/source/ui/tabledesign/TEditControl.hxx	2005-01-05 11:03:56.991049393 +0100
@@ -132,7 +132,7 @@ namespace dbaui
 			OTableEditorCtrl* m_pOwner;
 
 		public:
-			ClipboardInvalidator(ULONG nTimeout,OTableEditorCtrl*);
+			ClipboardInvalidator(sal_uInt32 nTimeout,OTableEditorCtrl*);
 			~ClipboardInvalidator();
 
 		protected:
@@ -160,7 +160,7 @@ namespace dbaui
 		virtual void Undo();
 		virtual void Redo();
 		virtual String GetCellText(long nRow, USHORT nColId) const;
-		virtual ULONG GetTotalCellWidth(long nRow, USHORT nColId);
+		virtual sal_uInt32 GetTotalCellWidth(long nRow, USHORT nColId);
 
 		virtual void CopyRows();
 		virtual void InsertRows( long nRow );
