--- xmlhelp/source/com/sun/star/help/HelpLinker.java.orig	2004-12-22 17:08:22.901974311 +0100
+++ xmlhelp/source/com/sun/star/help/HelpLinker.java	2004-12-22 17:08:53.581544726 +0100
@@ -505,10 +505,10 @@ class HelpLinker {
                 bais = new ByteArrayInputStream(baos.toByteArray());
                 ois = new ObjectInputStream(bais);
                 Hashtable anchorToLL = (Hashtable) ois.readObject();
-                Enumeration enum = anchorToLL.keys();
+                Enumeration enumer = anchorToLL.keys();
                 String fakedHid = URLEncoder.encode(documentPath);
-                while (enum.hasMoreElements()) {
-                    String anchor = (String) enum.nextElement();
+                while (enumer.hasMoreElements()) {
+                    String anchor = (String) enumer.nextElement();
                     addBookmark(
                         dbBase,
                         documentPath,
@@ -600,9 +600,9 @@ class HelpLinker {
 		// last, all files which should be copied into the jarFile
 		/////////////////////////////////////////////////////////////////////////
 
-		Enumeration enum = additionalFiles.keys();
-		while (enum.hasMoreElements()) {
-			String additionalFileKey = (String) enum.nextElement();
+		Enumeration enumer = additionalFiles.keys();
+		while (enumer.hasMoreElements()) {
+			String additionalFileKey = (String) enumer.nextElement();
 			String additionalFileName =
 				(String) additionalFiles.get(additionalFileKey);
 			copyFileToJarfile(additionalFileName, null, additionalFileKey);
--- xmlhelp/source/com/sun/star/help/HelpKeyword.java.orig	2004-12-22 17:10:13.982319803 +0100
+++ xmlhelp/source/com/sun/star/help/HelpKeyword.java	2004-12-22 17:10:28.790700481 +0100
@@ -59,11 +59,11 @@ public class HelpKeyword {
 	}
 
 	void dump(Db table) {
-		Enumeration enum = _hash.keys();
+		Enumeration enumer = _hash.keys();
 		int j = 0;
 		String[] list = new String[_hash.size()];
-		while (enum.hasMoreElements()) {
-			list[j++] = (String) enum.nextElement();
+		while (enumer.hasMoreElements()) {
+			list[j++] = (String) enumer.nextElement();
 		}
 
 		for (int i = 0; i < list.length; ++i) {
--- xmlhelp/source/com/sun/star/help/HelpTransformer.java.orig	2004-12-22 17:11:10.251367944 +0100
+++ xmlhelp/source/com/sun/star/help/HelpTransformer.java	2004-12-22 17:11:43.160548897 +0100
@@ -14,11 +14,11 @@ class DefinitionsSorted
 	{
 		String styleName;
 		Hashtable styles;
-		Enumeration enum = defs.getTable().keys();
+		Enumeration enumer = defs.getTable().keys();
 		
-		while( enum.hasMoreElements() )
+		while( enumer.hasMoreElements() )
 		{
-			styleName = ( String ) enum.nextElement();
+			styleName = ( String ) enumer.nextElement();
 			styles = defs.getEffectiveStyle( styleName );
 			
 			Enumeration styleOptions = styles.keys();
@@ -112,11 +112,11 @@ class Definitions
 		while( it.hasNext() )
 		{
 			Hashtable curr;
-			Enumeration enum = (curr = ( Hashtable )it.next()).keys();
-			while( enum.hasMoreElements() )
+			Enumeration enumer = (curr = ( Hashtable )it.next()).keys();
+			while( enumer.hasMoreElements() )
 			{
 				String str;
-				if( result.get( str = ( String ) enum.nextElement() ) == null )
+				if( result.get( str = ( String ) enumer.nextElement() ) == null )
 					result.put( str,curr.get( str ) );
 			}
 		}
@@ -129,11 +129,11 @@ class Definitions
 	{
 		String str;
 		Hashtable a = getEffectiveStyle( "p.def1" );
-		Enumeration enum = a.keys();
+		Enumeration enumer = a.keys();
 		
-		while( enum.hasMoreElements() )
+		while( enumer.hasMoreElements() )
 		{
-			System.out.println( (str = ( String ) enum.nextElement()) + " : " + ((String)a.get(str)) );
+			System.out.println( (str = ( String ) enumer.nextElement()) + " : " + ((String)a.get(str)) );
 		}
 	}
 	
--- xmlhelp/source/com/sun/star/help/HelpIndexer.java.orig	2004-12-22 17:12:21.878703949 +0100
+++ xmlhelp/source/com/sun/star/help/HelpIndexer.java	2004-12-22 17:12:56.931508151 +0100
@@ -239,11 +239,11 @@ public class HelpIndexer {
 			table.close(0);
 
 			System.out.println("Indexing...");
-			Enumeration enum = _hashDocInfo.elements();
+			Enumeration enumer = _hashDocInfo.elements();
 			int cut = 0;
-			while (enum.hasMoreElements() && cut < 100000000) {
+			while (enumer.hasMoreElements() && cut < 100000000) {
 				try {
-					DocInfo info = (DocInfo) enum.nextElement();
+					DocInfo info = (DocInfo) enumer.nextElement();
 					String url = info.getURL();
 					if (url == null) {
 						System.out.println(
@@ -423,11 +423,11 @@ public class HelpIndexer {
 		}
 
 		void dump() {
-			Enumeration enum = _hash.keys();
+			Enumeration enumer = _hash.keys();
 			int j = 0;
 			String[] list = new String[_hash.size()];
-			while (enum.hasMoreElements()) {
-				list[j++] = (String) enum.nextElement();
+			while (enumer.hasMoreElements()) {
+				list[j++] = (String) enumer.nextElement();
 			}
 
 			Db table;
@@ -458,11 +458,11 @@ public class HelpIndexer {
 	}
 
 	void dumpHelptext() {
-		Enumeration enum = _hashHelptext.keys();
+		Enumeration enumer = _hashHelptext.keys();
 		int j = 0;
 		String[] list = new String[_hashHelptext.size()];
-		while (enum.hasMoreElements()) {
-			list[j++] = (String) enum.nextElement();
+		while (enumer.hasMoreElements()) {
+			list[j++] = (String) enumer.nextElement();
 		}
 
 		Db table;
