--- extensions/source/scanner/grid.cxx	2003-10-27 14:21:34.000000000 +0100
+++ extensions/source/scanner/grid.cxx	2005-01-11 13:04:39.927291339 +0100
@@ -63,7 +63,7 @@
 #include <cmath>
 #include <grid.hxx>
 
-ResId SaneResId( ULONG );
+ResId SaneResId( sal_uInt32 );
 
 /***********************************************************************
  *
--- extensions/source/scanner/sanedlg.cxx	2004-02-26 14:12:12.000000000 +0100
+++ extensions/source/scanner/sanedlg.cxx	2005-01-11 13:03:12.427481480 +0100
@@ -74,7 +74,7 @@
 #define USE_SAVE_STATE
 #undef  SAVE_ALL_STATES
 
-ResId SaneResId( ULONG nID )
+ResId SaneResId( sal_uInt32 nID )
 {
 	static ResMgr* pResMgr = ResMgr::CreateResMgr( "san" MAKE_NUMSTR(SUPD) );
 	return ResId( nID, pResMgr );
