diff -pur xmlhelp.orig/source/cxxhelp/provider/db.cxx xmlhelp/source/cxxhelp/provider/db.cxx
--- xmlhelp.orig/source/cxxhelp/provider/db.cxx	2005-03-30 10:37:57.000000000 +0200
+++ xmlhelp/source/cxxhelp/provider/db.cxx	2005-04-21 10:23:14.662386760 +0200
@@ -234,7 +234,7 @@ Dbt::Dbt()
 }
 
 
-Dbt::Dbt(void *data_arg, u_int32_t size_arg)
+Dbt::Dbt(void *data_arg, size_t size_arg)
 {
     using namespace std;
     DBT * thispod = this;
Only in xmlhelp/source/cxxhelp/provider: .db.cxx.swp
Only in xmlhelp/source/cxxhelp/provider: .db.hxx.swp
