--- tools/inc/bigint.hxx	2004-06-17 15:08:56.000000000 +0200
+++ tools/inc/bigint.hxx	2004-12-27 15:51:46.677420277 +0100
@@ -111,8 +111,8 @@ public:
     operator        long()  const;
     operator        int()   const;
     operator        double() const;
-    operator        USHORT() const;
-    operator        ULONG() const;
+    operator        sal_uInt16() const;
+    operator        sal_uInt32() const;
 
     void            Set( sal_Bool bSet ) { bIsSet = bSet; }
     ByteString      GetByteString() const;
@@ -230,7 +230,7 @@
         return 0;
 }
 
-inline BigInt::operator USHORT() const
+inline BigInt::operator sal_uInt16() const
 {
     if ( !bIsBig && (nVal == (sal_uInt32)(sal_uInt16)nVal) )
         return (sal_uInt16)nVal;
--- tools/inc/date.hxx	2003-09-04 09:44:38.000000000 +0200
+++ tools/inc/date.hxx	2004-12-27 09:48:14.465562808 +0100
@@ -86,21 +86,21 @@ enum DayOfWeek { MONDAY, TUESDAY, WEDNES
 class Date
 {
 private:
-	ULONG			nDate;
+	sal_uInt32		nDate;
 
 public:
 					Date();
 					Date( const ResId & rResId );
-					Date( ULONG _nDate ) { Date::nDate = _nDate; }
+					Date( sal_uInt32 _nDate ) { Date::nDate = _nDate; }
 					Date( const Date& rDate )
 						{ nDate = rDate.nDate; }
 					Date( USHORT nDay, USHORT nMonth, USHORT nYear )
-						{ nDate = ((ULONG)(nDay%100)) +
-								  (((ULONG)(nMonth%100))*100) +
-								  (((ULONG)nYear%10000)*10000); }
+						{ nDate = ((sal_uInt32)(nDay%100)) +
+								  (((sal_uInt32)(nMonth%100))*100) +
+								  (((sal_uInt32)nYear%10000)*10000); }
 
-	void			SetDate( ULONG nNewDate ) { nDate = nNewDate; }
-	ULONG			GetDate() const { return nDate; }
+	void			SetDate( sal_uInt32 nNewDate ) { nDate = nNewDate; }
+	sal_uInt32		GetDate() const { return nDate; }
 
 	void			SetDay( USHORT nNewDay );
 	void			SetMonth( USHORT nNewMonth );
--- tools/source/generic/bigint.cxx	2004-06-17 15:11:15.000000000 +0200
+++ tools/source/generic/bigint.cxx	2004-12-27 15:52:43.680572912 +0100
@@ -687,7 +687,7 @@ BigInt::BigInt( sal_uInt32 nValue )
 
 // -----------------------------------------------------------------------
 
-BigInt::operator ULONG() const
+BigInt::operator sal_uInt32() const
 {
     if ( !bIsBig )
         return (sal_uInt32)nVal;
