--- rsc/source/res/rscrange.cxx	2004-06-17 13:53:09.000000000 +0200
+++ rsc/source/res/rscrange.cxx	2005-01-04 10:35:06.533204731 +0100
@@ -239,7 +239,7 @@ RSCINST RscRange::Create( RSCINST * pIns
 void RscRange::WriteSrc( const RSCINST & rInst, FILE * fOutput,
 						 RscTypCont *, USHORT, const char * )
 {
-	fprintf( fOutput, "%ld", ((RscRangeInst *)rInst.pData)->nValue + nMin );
+	fprintf( fOutput, "%ld", long( ((RscRangeInst *)rInst.pData)->nValue + nMin ) );
 }
 
 /*************************************************************************
