--- scp2/util/makefile.mk	2005-06-08 13:35:31.840198136 +0200
+++ scp2/util/makefile.mk	2005-06-08 13:37:38.297993146 +0200
@@ -105,10 +105,13 @@ SCP1FILES  = installation_ooo.par       
              module_lingu.par              \
              file_lingu.par                \
              module_xsltfilter.par         \
-             file_xsltfilter.par           \
-             module_python.par             \
+             file_xsltfilter.par
+
+.IF "$(CPUNAME)" != "X86_64"
+SCP1FILES += module_python.par             \
              module_python_mailmerge.par   \
              file_python.par               \
              profileitem_python.par
+.ENDIF
 
 .IF "$(SOLAR_JAVA)"!=""
 SCP1FILES +=                               \
