--- cppcanvas/source/mtfrenderer/implrenderer.cxx	2005-03-30 10:27:39.000000000 +0200
+++ cppcanvas/source/mtfrenderer/implrenderer.cxx	2005-04-21 10:04:04.454713029 +0200
@@ -2596,7 +2596,7 @@ namespace cppcanvas
             {
                 // only a single action. Setup subset, and render
                 Action::Subset aSubset;
-                aSubset.mnSubsetBegin = ::std::max( 0L,
+                aSubset.mnSubsetBegin = ::std::max( 0,
                                                     nStartIndex - aRangeBegin->mnOrigIndex );
                 aSubset.mnSubsetEnd   = ::std::min( aRangeBegin->mpAction->getActionCount(),
                                                     nEndIndex - aRangeBegin->mnOrigIndex );
@@ -2615,7 +2615,7 @@ namespace cppcanvas
                 // render partial first, full intermediate, and
                 // partial last action
                 Action::Subset aSubset;
-                aSubset.mnSubsetBegin = ::std::max( 0L,
+                aSubset.mnSubsetBegin = ::std::max( 0,
                                                     nStartIndex - aRangeBegin->mnOrigIndex );
                 aSubset.mnSubsetEnd   = aRangeBegin->mpAction->getActionCount();
 
--- cppcanvas/source/mtfrenderer/textaction.cxx	2005-03-30 10:31:59.000000000 +0200
+++ cppcanvas/source/mtfrenderer/textaction.cxx	2005-04-21 10:04:55.103991006 +0200
@@ -450,7 +450,7 @@ namespace cppcanvas
                                                 ::std::min( 
                                                     rSubset.mnSubsetEnd - rSubset.mnSubsetBegin,
                                                     rOrigContext.Length ),
-                                                0L ) );
+                                                0 ) );
 
                 const rendering::StringContext aContext( rOrigContext.Text,
                                                          nNewStartPos,
