--- sw/source/core/edit/editsh.cxx	2005-11-14 19:22:54.000000000 +0100
+++ sw/source/core/edit/editsh.cxx	2005-11-14 19:37:09.000000000 +0100
@@ -960,8 +960,8 @@ void SwEditShell::SetNumberingRestart()
         else
             MakeFindRange(DOCPOS_OTHERSTART, DOCPOS_OTHEREND, pCrsr); //extra content
         SwPosition* pSttPos = pCrsr->Start(), *pEndPos = pCrsr->End();
-        sal_uInt32 nCurrNd = pSttPos->nNode.GetIndex();
-        sal_uInt32 nEndNd = pEndPos->nNode.GetIndex();
+        ULONG nCurrNd = pSttPos->nNode.GetIndex();
+        ULONG nEndNd = pEndPos->nNode.GetIndex();
         if( nCurrNd <= nEndNd )
         {
             SwCntntFrm* pCntFrm;
