cvs diff: Diffing source
Index: padmin/source/padialog.cxx
===================================================================
RCS file: /cvs/gsl/padmin/source/padialog.cxx,v
retrieving revision 1.14
diff -u -r1.14 padialog.cxx
--- padmin/source/padialog.cxx	31 Jan 2005 09:28:06 -0000	1.14
+++ padmin/source/padialog.cxx	4 Jun 2005 19:17:22 -0000
@@ -219,7 +228,7 @@
 String PADialog::getSelectedDevice()
 {
 	int nPos = m_aDevicesLB.GetSelectEntryPos();
-	int nLen = (int)m_aDevicesLB.GetEntryData( nPos );
+	int nLen = (int)(sal_IntPtr)m_aDevicesLB.GetEntryData( nPos );
 	return m_aDevicesLB.GetEntry( nPos ).Copy( 0, nLen );
 }
 
