Index: stoc/source/javavm/javavm.cxx
===================================================================
--- stoc/source/javavm/javavm.cxx.orig	2005-06-17 12:06:59.000000000 +0200
+++ stoc/source/javavm/javavm.cxx	2005-07-05 13:33:22.070266296 +0200
@@ -1037,13 +1037,7 @@
                         " that the requested JavaVM pointer is not available")),
                 static_cast< cppu::OWeakObject * >(this));
         }
-        if (sizeof (m_pJavaVm) <= sizeof (sal_Int32)) {
-            return css::uno::makeAny(reinterpret_cast< sal_Int32 >(m_pJavaVm));
-        } else {
-            OSL_ASSERT(
-                sizeof (sal_Int64) >= sizeof (jvmaccess::VirtualMachine *));
-            return css::uno::makeAny(reinterpret_cast< sal_Int64 >(m_pJavaVm));
-        }
+        return css::uno::makeAny(reinterpret_cast< sal_IntPtr >(m_pJavaVm));
     case RETURN_VIRTUALMACHINE:
         OSL_ASSERT(sizeof (sal_Int64) >= sizeof (jvmaccess::VirtualMachine *));
         return css::uno::makeAny(
