--- svx/inc/galmisc.hxx	2005-03-23 13:41:31.000000000 +0100
+++ svx/inc/galmisc.hxx	2005-04-21 12:26:58.068654915 +0200
@@ -127,7 +127,7 @@ BOOL			FileExists( const INetURLObject& 
 BOOL			CreateDir(  const INetURLObject& rURL );
 BOOL			CopyFile(  const INetURLObject& rSrcURL, const INetURLObject& rDstURL );
 BOOL			KillFile( const INetURLObject& rURL );
-BitmapEx		GalleryResGetBitmapEx( ULONG nId );
+BitmapEx		GalleryResGetBitmapEx( sal_uInt32 nId );
 
 
 // ---------------
--- svx/inc/outliner.hxx	2005-04-21 11:32:30.504424756 +0200
+++ svx/inc/outliner.hxx	2005-04-21 11:32:09.801442410 +0200
@@ -875,7 +875,7 @@ public:
 	ULONG           GetLineHeight( ULONG nParagraph, ULONG nLine = 0 );
 
 	// nFormat muss ein Wert aus dem enum EETextFormat sein (wg.CLOOKS)
-	ULONG			Read( SvStream& rInput, const String& rBaseURL, USHORT, SvKeyValueIterator* pHTTPHeaderAttrs = NULL );
+    ULONG           Read( SvStream& rInput, const String& rBaseURL, USHORT, SvKeyValueIterator* pHTTPHeaderAttrs = NULL );
 
 	SfxUndoManager& GetUndoManager();
 
--- svx/source/form/fmvwimp.cxx	2005-04-21 11:32:30.716363153 +0200
+++ svx/source/form/fmvwimp.cxx	2005-04-21 11:40:54.963795566 +0200
@@ -1665,8 +1665,8 @@ SdrObject* FmXFormView::implCreateXForms
 			const sal_uInt16 nObjID = OBJ_FM_BUTTON;
 			::Size controlSize(4000, 500);
 			FmFormObj *pControl = static_cast<FmFormObj*>(SdrObjFactory::MakeNewObject( FmFormInventor, nObjID, NULL, NULL ));
-			controlSize.Width() = sal_Int32(Fraction(controlSize.Width(), 1) * eTargetMode.GetScaleX());
-			controlSize.Height() = sal_Int32(Fraction(controlSize.Height(), 1) * eTargetMode.GetScaleY());
+			controlSize.Width() = Fraction(controlSize.Width(), 1) * eTargetMode.GetScaleX();
+			controlSize.Height() = Fraction(controlSize.Height(), 1) * eTargetMode.GetScaleY();
 			::Point controlPos(_pOutDev->LogicToLogic(Point(controlSize.Width(),0),eSourceMode,eTargetMode));
 			::Rectangle controlRect(controlPos,_pOutDev->LogicToLogic(controlSize, eSourceMode, eTargetMode));
 			pControl->SetLogicRect(controlRect);
--- svx/source/gallery2/galmisc.cxx	2005-03-23 13:42:21.000000000 +0100
+++ svx/source/gallery2/galmisc.cxx	2005-04-21 11:44:13.365127044 +0200
@@ -716,7 +716,7 @@ void GalleryTransferable::StartDrag( Win
     }
 }
 
-BitmapEx GalleryResGetBitmapEx( ULONG nId )
+BitmapEx GalleryResGetBitmapEx( sal_uInt32 nId )
 {
 	BitmapEx aBmpEx( GAL_RESID( nId ) );
 	
