--- padmin/source/helper.cxx	2004-06-26 05:11:23.000000000 +0200
+++ padmin/source/helper.cxx	2005-01-11 12:03:56.739054845 +0100
@@ -131,7 +131,7 @@ using namespace com::sun::star::ui::dial
  *	PaResId
  */
 
-ResId padmin::PaResId( ULONG nId )
+ResId padmin::PaResId( sal_uInt32 nId )
 {
 	static ResMgr* pPaResMgr = NULL;
 	if( ! pPaResMgr )
--- padmin/source/helper.hxx	2004-03-30 15:46:19.000000000 +0200
+++ padmin/source/helper.hxx	2005-01-11 12:05:59.538922220 +0100
@@ -156,7 +156,7 @@ public:
 
 BOOL AreYouSure( Window*, int nRid = -1 );
 
-ResId PaResId( ULONG nId );
+ResId PaResId( sal_uInt32 nId );
 
 void FindFiles( const String& rDirectory, ::std::list< String >& rResult, const String& rSuffixes, bool bRecursive = false );
 
