--- tools/source/communi/simplecm.cxx	2005-03-15 09:56:54.000000000 +0100
+++ tools/source/communi/simplecm.cxx	2005-04-20 15:47:31.427509413 +0200
@@ -309,7 +309,7 @@ BOOL SimpleCommunicationLinkViaSocket::D
 {
 	BOOL bWasError = FALSE;
 	char* pBuffer = NULL;
-    ULONG nLen;
+	comm_UINT32 nLen;
 	bWasError = pPacketHandler->ReceiveData( (void* &)pBuffer, nLen ) != C_ERROR_NONE;
 	if ( !bWasError )
 	{
