--- automation/source/testtool/cmdstrm.cxx	2004-12-16 11:35:33.000000000 +0100
+++ automation/source/testtool/cmdstrm.cxx	2004-12-27 14:55:23.593863549 +0100
@@ -379,7 +379,7 @@ void CmdStream::GenCmdSlot( USHORT nNr, 
 				case SbxDOUBLE:
                     if ( !bWriteUnoSlot )
                         Write( (USHORT)BinULONG );
-					Write(rPar->Get( 2*n )->GetULong());
+					Write( (comm_ULONG) rPar->Get( 2*n )->GetULong());
 					break;
 				case SbxSTRING:
 				case SbxCHAR:
