--- sc/inc/address.hxx	2006-03-20 18:30:06.000000000 +0100
+++ sc/inc/address.hxx	2006-03-27 19:01:34.000000000 +0200
@@ -81,11 +81,6 @@ typedef sal_Int32 SCsCOLROW;
 // size_t typedef to be able to find places where code was changed from USHORT
 // to size_t and is used to read/write from/to streams.
 typedef size_t SCSIZE;
-#ifndef MACOSX
-// resolve ambiguity, TODO: move to tools/solar.h, or even better use ::std::min/max
-inline size_t Min( size_t a, size_t b) { return a<b ? a : b; }
-inline size_t Max( size_t a, size_t b) { return a>b ? a : b; }
-#endif
 
 // Maximum possible value of data type, NOT maximum row value.
 // MSC confuses numeric_limit max() with macro max() if vcl/wintypes.hxx is
