--- dbaccess/source/ui/app/AppController.cxx	2005-04-06 13:40:09.000000000 +0200
+++ dbaccess/source/ui/app/AppController.cxx	2005-04-21 13:28:51.871178772 +0200
@@ -1751,7 +1751,7 @@ Reference< XComponent > OApplicationCont
 // -----------------------------------------------------------------------------
 IMPL_LINK( OApplicationController, OnCreateWithPilot, void*, _pType )
 {
-    ElementType eType = (ElementType)reinterpret_cast< int >( _pType );
+    ElementType eType = (ElementType)reinterpret_cast< sal_IntPtr >( _pType );
     newElementWithPilot( eType );
     return 0L;
 }
--- dbaccess/source/ui/tabledesign/TEditControl.cxx	2005-03-10 17:54:31.000000000 +0100
+++ dbaccess/source/ui/tabledesign/TEditControl.cxx	2005-04-21 13:39:46.194989084 +0200
@@ -190,7 +190,7 @@ DBG_NAME(OTableEditorCtrl);
 
 //==================================================================
 //------------------------------------------------------------------
-OTableEditorCtrl::ClipboardInvalidator::ClipboardInvalidator(ULONG nTimeout,OTableEditorCtrl* _pOwner)
+OTableEditorCtrl::ClipboardInvalidator::ClipboardInvalidator(sal_uInt32 nTimeout,OTableEditorCtrl* _pOwner)
 : m_pOwner(_pOwner)
 {
 	m_aInvalidateTimer.SetTimeout(nTimeout);
@@ -1313,7 +1313,7 @@ String OTableEditorCtrl::GetCellText( lo
 }
 
 //------------------------------------------------------------------------------
-ULONG OTableEditorCtrl::GetTotalCellWidth(long nRow, sal_uInt16 nColId)
+sal_uInt32 OTableEditorCtrl::GetTotalCellWidth(long nRow, sal_uInt16 nColId)
 {
 	DBG_CHKTHIS(OTableEditorCtrl,NULL);
 	return GetTextWidth(GetCellText(nRow, nColId)) + 2 * GetTextWidth('0');
