--- sc/source/ui/dbgui/csvtablebox.cxx	23 Aug 2004 09:29:32 -0000	1.8
+++ sc/source/ui/dbgui/csvtablebox.cxx	19 Oct 2004 13:19:12 -0000	1.7.378.2
@@ -409,33 +409,33 @@ IMPL_LINK( ScCsvTableBox, CsvCmdHdl, ScC
     switch( eType )
     {
         case CSVCMD_SETPOSCOUNT:
-            maData.mnPosCount = Max( nParam1, 1L );
+            maData.mnPosCount = Max( nParam1, sal_Int32( 1 ) );
             ImplSetPosOffset( GetFirstVisPos() );
         break;
         case CSVCMD_SETPOSOFFSET:
             ImplSetPosOffset( nParam1 );
         break;
         case CSVCMD_SETHDRWIDTH:
-            maData.mnHdrWidth = Max( nParam1, 0L );
+            maData.mnHdrWidth = Max( nParam1, sal_Int32( 0 ) );
             ImplSetPosOffset( GetFirstVisPos() );
         break;
         case CSVCMD_SETCHARWIDTH:
-            maData.mnCharWidth = Max( nParam1, 1L );
+            maData.mnCharWidth = Max( nParam1, sal_Int32( 1 ) );
             ImplSetPosOffset( GetFirstVisPos() );
         break;
         case CSVCMD_SETLINECOUNT:
-            maData.mnLineCount = Max( nParam1, 1L );
+            maData.mnLineCount = Max( nParam1, sal_Int32( 1 ) );
             ImplSetLineOffset( GetFirstVisLine() );
         break;
         case CSVCMD_SETLINEOFFSET:
             ImplSetLineOffset( nParam1 );
         break;
         case CSVCMD_SETHDRHEIGHT:
-            maData.mnHdrHeight = Max( nParam1, 0L );
+            maData.mnHdrHeight = Max( nParam1, sal_Int32( 0 ) );
             ImplSetLineOffset( GetFirstVisLine() );
         break;
         case CSVCMD_SETLINEHEIGHT:
-            maData.mnLineHeight = Max( nParam1, 1L );
+            maData.mnLineHeight = Max( nParam1, sal_Int32( 1 ) );
             ImplSetLineOffset( GetFirstVisLine() );
         break;
         case CSVCMD_MOVERULERCURSOR:
--- sc/source/ui/inc/csvtablebox.hxx	23 Aug 2004 09:33:53 -0000	1.5
+++ sc/source/ui/inc/csvtablebox.hxx	19 Oct 2004 12:49:53 -0000	1.4.484.2
@@ -143,10 +143,10 @@ private:
 
     /** Calculates and sets valid position offset nearest to nPos. */
     SC_DLLPRIVATE inline void                 ImplSetPosOffset( sal_Int32 nPos )
-                                    { maData.mnPosOffset = Max( Min( nPos, GetMaxPosOffset() ), 0L ); }
+                                    { maData.mnPosOffset = Max( Min( nPos, GetMaxPosOffset() ), sal_Int32( 0 ) ); }
     /** Calculates and sets valid line offset nearest to nLine. */
     SC_DLLPRIVATE inline void                 ImplSetLineOffset( sal_Int32 nLine )
-                                    { maData.mnLineOffset = Max( Min( nLine, GetMaxLineOffset() ), 0L ); }
+                                    { maData.mnLineOffset = Max( Min( nLine, GetMaxLineOffset() ), sal_Int32( 0 ) ); }
     /** Moves controls (not cursors!) so that nPos becomes visible. */
     SC_DLLPRIVATE void                        MakePosVisible( sal_Int32 nPos );
 
--- sc/source/ui/dbgui/csvruler.cxx	26 Mar 2003 18:05:52 -0000	1.9
+++ sc/source/ui/dbgui/csvruler.cxx	9 Sep 2004 11:02:30 -0000	1.9.378.1
@@ -231,7 +231,7 @@ sal_Int32 ScCsvRuler::GetNoScrollPos( sa
         else if( nNewPos > GetLastVisPos() - CSV_SCROLL_DIST - 1L )
         {
             sal_Int32 nScroll = (GetFirstVisPos() < GetMaxPosOffset()) ? CSV_SCROLL_DIST : 0;
-            nNewPos = Min( nNewPos, GetLastVisPos() - nScroll - 1L );
+            nNewPos = Min( nNewPos, GetLastVisPos() - nScroll - sal_Int32( 1 ) );
         }
     }
     return nNewPos;
@@ -374,7 +374,7 @@ void ScCsvRuler::MouseMove( const MouseE
         if( IsTracking() )
         {
             // on mouse tracking: keep position valid
-            nPos = Max( Min( nPos, GetPosCount() - 1L ), 1L );
+            nPos = Max( Min( nPos, GetPosCount() - sal_Int32( 1 ) ), sal_Int32( 1 ) );
             MoveMouseTracking( nPos );
         }
         else
@@ -513,7 +513,7 @@ void ScCsvRuler::ImplDrawArea( sal_Int32
 
     aRect = maActiveRect;
     aRect.Left() = Max( GetFirstX(), nPosX );
-    aRect.Right() = Min( Min( GetX( GetPosCount() ), GetLastX() ), nPosX + nWidth - 1L );
+    aRect.Right() = Min( Min( GetX( GetPosCount() ), GetLastX() ), nPosX + nWidth - sal_Int32( 1 ) );
     if( aRect.Left() <= aRect.Right() )
     {
         maBackgrDev.SetFillColor( maActiveColor );
--- sc/source/ui/dbgui/csvgrid.cxx	11 Oct 2004 12:28:48 -0000	1.21
+++ sc/source/ui/dbgui/csvgrid.cxx	18 Jan 2005 04:32:56 -0000	1.19.42.3
@@ -179,7 +179,7 @@ void ScCsvGrid::UpdateOffsetX()
     sal_Int32 nLastLine = GetLastVisLine() + 1;
     sal_Int32 nDigits = 2;
     while( nLastLine /= 10 ) ++nDigits;
-    nDigits = Max( nDigits, 3L );
+    nDigits = Max( nDigits, sal_Int32( 3 ) );
     Execute( CSVCMD_SETHDRWIDTH, GetTextWidth( String( '0' ) ) * nDigits );
 }
 
@@ -704,8 +704,8 @@ void ScCsvGrid::MoveCursor( sal_uInt32 n
     {
         sal_Int32 nPosBeg = GetColumnPos( nColIndex );
         sal_Int32 nPosEnd = GetColumnPos( nColIndex + 1 );
-        sal_Int32 nMinPos = Max( nPosBeg - CSV_SCROLL_DIST, 0L );
-        sal_Int32 nMaxPos = Min( nPosEnd - GetVisPosCount() + CSV_SCROLL_DIST + 1L, nMinPos );
+        sal_Int32 nMinPos = Max( nPosBeg - CSV_SCROLL_DIST, sal_Int32( 0 ) );
+        sal_Int32 nMaxPos = Min( nPosEnd - GetVisPosCount() + CSV_SCROLL_DIST + sal_Int32( 1 ), nMinPos );
         if( nPosBeg - CSV_SCROLL_DIST + 1 <= GetFirstVisPos() )
             Execute( CSVCMD_SETPOSOFFSET, nMinPos );
         else if( nPosEnd + CSV_SCROLL_DIST >= GetLastVisPos() )
@@ -787,7 +787,7 @@ void ScCsvGrid::ImplSetTextLineSep(
         pChar = ScImportExport::ScanNextFieldFromString( pChar, aCellText, cTextSep, pSepChars, bMergeSep );
 
         // update column width
-        sal_Int32 nWidth = Max( CSV_MINCOLWIDTH, aCellText.Len() + 1L );
+        sal_Int32 nWidth = Max( CSV_MINCOLWIDTH, aCellText.Len() + sal_Int32( 1 ) );
         if( IsValidColumn( nColIx ) )
         {
             // expand existing column
@@ -918,7 +918,7 @@ void ScCsvGrid::Tracking( const Tracking
 
         sal_Int32 nPos = (rMEvt.GetPosPixel().X() - GetFirstX()) / GetCharWidth() + GetFirstVisPos();
         // on mouse tracking: keep position valid
-        nPos = Max( Min( nPos, GetPosCount() - 1L ), 0L );
+        nPos = Max( Min( nPos, GetPosCount() - sal_Int32( 1 ) ), sal_Int32( 0 ) );
         Execute( CSVCMD_MAKEPOSVISIBLE, nPos );
 
         sal_uInt32 nColIx = GetColumnFromPos( nPos );
@@ -1366,7 +1366,7 @@ void ScCsvGrid::ImplDrawTrackingRect( sa
     if( HasFocus() && IsVisibleColumn( nColIndex ) )
     {
         sal_Int32 nX1 = Max( GetColumnX( nColIndex ), GetFirstX() ) + 1;
-        sal_Int32 nX2 = Min( GetColumnX( nColIndex + 1 ) - 1L, GetLastX() );
+        sal_Int32 nX2 = Min( GetColumnX( nColIndex + 1 ) - sal_Int32( 1 ), GetLastX() );
         sal_Int32 nY2 = Min( GetY( GetLastVisLine() + 1 ), GetHeight() ) - 1;
         InvertTracking( Rectangle( nX1, 0, nX2, nY2 ), SHOWTRACK_SMALL | SHOWTRACK_WINDOW );
     }
