--- vcl/inc/dockwin.hxx	2004-12-23 13:53:23.371814191 +0100
+++ vcl/inc/dockwin.hxx	2004-12-23 13:53:59.319605085 +0100
@@ -135,10 +135,10 @@ private:
 	long			mnTrackY;
 	long			mnTrackWidth;
 	long			mnTrackHeight;
-	long			mnDockLeft;
-	long			mnDockTop;
-	long			mnDockRight;
-	long			mnDockBottom;
+	sal_Int32		mnDockLeft;
+	sal_Int32		mnDockTop;
+	sal_Int32		mnDockRight;
+	sal_Int32		mnDockBottom;
 	WinBits 		mnFloatBits;
 	BOOL			mbDockCanceled:1,
 					mbFloatPrevented:1,
--- vcl/source/gdi/pdfwriter_impl.cxx	2004-12-23 14:10:40.341697954 +0100
+++ vcl/source/gdi/pdfwriter_impl.cxx	2004-12-23 14:13:18.581364393 +0100
@@ -3497,7 +3497,7 @@ void PDFWriterImpl::createDefaultPushBut
     OStringBuffer aDA( 256 );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetButtonTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[m_nCurrentPage].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[m_nCurrentPage].appendMappedLength( (sal_Int32) aFont.GetHeight(), aDA );
     aDA.append( " Tf" );
     rButton.m_aDAString = aDA.makeStringAndClear();
 
@@ -3554,7 +3554,7 @@ void PDFWriterImpl::createDefaultEditApp
     OStringBuffer aDA( 32 );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetFieldTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( (sal_Int32) aFont.GetHeight(), aDA );
     aDA.append( " Tf" );
 
     /*  create an empty appearance stream, let the viewer create
@@ -3655,7 +3655,7 @@ void PDFWriterImpl::createDefaultListBox
 #endif
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetFieldTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( (sal_Int32) aFont.GetHeight(), aDA );
     aDA.append( " Tf" );
     rBox.m_aDAString = aDA.makeStringAndClear();
 }
@@ -3743,7 +3743,7 @@ void PDFWriterImpl::createDefaultCheckBo
     aDA.append( "/Tx BMC\r\nq BT\r\n" );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ), aDA );
     aDA.append( " /ZaDb " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aCheckRect.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( (sal_Int32) aCheckRect.GetHeight(), aDA );
     aDA.append( " Tf\r\n" );
     m_aPages[ m_nCurrentPage ].appendMappedLength( nCharXOffset, aDA );
     aDA.append( " " );
@@ -3838,7 +3838,7 @@ void PDFWriterImpl::createDefaultRadioBu
     aDA.append( "/Tx BMC\r\nq BT\r\n" );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ), aDA );
     aDA.append( " /ZaDb " );
-    m_aPages[m_nCurrentPage].appendMappedLength( aCheckRect.GetHeight(), aDA );
+    m_aPages[m_nCurrentPage].appendMappedLength( (sal_Int32) aCheckRect.GetHeight(), aDA );
     aDA.append( " Tf\r\n0 0 Td\r\nET\r\nQ\r\n" );
     writeBuffer( aDA.getStr(), aDA.getLength() );
     setFillColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ) );
