--- qadevOOo/runner/util/DBTools.java	2004-12-22 16:50:25.004648608 +0100
+++ qadevOOo/runner/util/DBTools.java	2004-12-22 16:50:55.538299660 +0100
@@ -680,12 +680,12 @@ public class DBTools {
         XEnumerationAccess dbContEA = (XEnumerationAccess)
             UnoRuntime.queryInterface(XEnumerationAccess.class, dbContext) ;
 
-        XEnumeration enum = dbContEA.createEnumeration() ;
+        XEnumeration enumer = dbContEA.createEnumeration() ;
 
         out.println("DatabaseContext registered DataSource's :") ;
-        while (enum.hasMoreElements()) {
+        while (enumer.hasMoreElements()) {
             try {
-                DataSourceInfo inf = new DataSourceInfo(enum.nextElement()) ;
+                DataSourceInfo inf = new DataSourceInfo(enumer.nextElement()) ;
                 inf.printInfo(out) ;
             } catch (com.sun.star.container.NoSuchElementException e) {}
             catch (com.sun.star.lang.WrappedTargetException e) {}
--- qadevOOo/tests/java/ifc/style/_ParagraphProperties.java	2004-12-22 16:57:11.171534053 +0100
+++ qadevOOo/tests/java/ifc/style/_ParagraphProperties.java	2004-12-22 16:57:44.064778408 +0100
@@ -459,18 +459,18 @@ public class _ParagraphProperties extend
         }
         
         public String[] getElementNames() {
-            Enumeration enum = members.keys();
+            Enumeration enumer = members.keys();
             int count = members.size();
             String[] res = new String[count];
             int i=0;
-            while(enum.hasMoreElements())
-                res[i] = (String)enum.nextElement();
+            while(enumer.hasMoreElements())
+                res[i] = (String)enumer.nextElement();
             return res;
         }
         
         public com.sun.star.uno.Type getElementType() {
-            Enumeration enum = members.keys();
-            String key = (String)enum.nextElement();
+            Enumeration enumer = members.keys();
+            String key = (String)enumer.nextElement();
             Object o = members.get(key);
             return new Type(o.getClass());
         }
--- qadevOOo/tests/java/ifc/reflection/_XTypeDescriptionEnumerationAccess.java	2004-12-22 17:14:27.113572901 +0100
+++ qadevOOo/tests/java/ifc/reflection/_XTypeDescriptionEnumerationAccess.java	2004-12-22 17:14:39.756339480 +0100
@@ -95,13 +95,13 @@ public class _XTypeDescriptionEnumeratio
             return;
         }
         try {
-            XTypeDescriptionEnumeration enum = 
+            XTypeDescriptionEnumeration enumer = 
                         oObj.createTypeDescriptionEnumeration(sString, tClass, 
                         TypeDescriptionSearchDepth.INFINITE);
             try {
                 log.println("Got an enumeration.");
                 while (true) {
-                    XTypeDescription desc = enum.nextTypeDescription();
+                    XTypeDescription desc = enumer.nextTypeDescription();
                     i++;
                     log.println("\tdesc name: " + desc.getName()); 
                 }
--- qadevOOo/tests/java/mod/_smgr/OServiceManager.java	2004-12-22 17:15:38.850901895 +0100
+++ qadevOOo/tests/java/mod/_smgr/OServiceManager.java	2004-12-22 17:15:55.421975563 +0100
@@ -164,12 +164,12 @@ public class OServiceManager extends Tes
                 (XSet.class, oInterface) ;
             XSet set1 = (XSet) UnoRuntime.queryInterface
                 (XSet.class, srvMan) ;
-            XEnumeration enum = set1.createEnumeration();
-            Object srv = enum.nextElement();
+            XEnumeration enumer = set1.createEnumeration();
+            Object srv = enumer.nextElement();
 
             set.insert(srv) ;
 
-            newElement = enum.nextElement();
+            newElement = enumer.nextElement();
 
             XPropertySet xProp = (XPropertySet)UnoRuntime.queryInterface(
                                             XPropertySet.class, oInterface);
--- qadevOOo/tests/java/mod/_servicemgr.uno/OServiceManager.java	2004-12-22 17:16:46.101027450 +0100
+++ qadevOOo/tests/java/mod/_servicemgr.uno/OServiceManager.java	2004-12-22 17:17:02.116200189 +0100
@@ -162,12 +162,12 @@ public class OServiceManager extends Tes
                 (XSet.class, oInterface) ;
             XSet set1 = (XSet) UnoRuntime.queryInterface
                 (XSet.class, srvMan) ;
-            XEnumeration enum = set1.createEnumeration();
-            Object srv = enum.nextElement();
+            XEnumeration enumer = set1.createEnumeration();
+            Object srv = enumer.nextElement();
 
             set.insert(srv) ;
 
-            newElement = enum.nextElement();
+            newElement = enumer.nextElement();
 
             XPropertySet xProp = (XPropertySet)UnoRuntime.queryInterface(
                                             XPropertySet.class, oInterface);
