--- chart2/source/controller/dialogs/dlg_ChartType.cxx	2005-01-04 23:46:33.743527771 +0100
+++ chart2/source/controller/dialogs/dlg_ChartType.cxx	2005-01-04 23:41:37.795308293 +0100
@@ -1258,7 +1258,7 @@
     }
 }
 
-void SchDiagramTypeDlg::SetMaximumNumberOfLines( long nMaxLines )
+void SchDiagramTypeDlg::SetMaximumNumberOfLines( sal_Int32 nMaxLines )
 {
     aMtrFldNumLines.SetLast( nMaxLines );
     aMtrFldNumLines.SetMax( nMaxLines );
--- chart2/source/controller/dialogs/tp_Scale.cxx	2005-01-04 23:46:33.747526814 +0100
+++ chart2/source/controller/dialogs/tp_Scale.cxx	2005-01-04 23:41:37.798307575 +0100
@@ -380,7 +380,7 @@
         return LEAVE_PAGE;
     }
 
-	ULONG nIndex = pNumFormatter->GetStandardIndex(LANGUAGE_SYSTEM);
+	sal_uInt32 nIndex = pNumFormatter->GetStandardIndex(LANGUAGE_SYSTEM);
 	Edit* pEdit = NULL;
 	USHORT nErrStrId = 0;
 	double fDummy;
