Index: svtools/inc/instrm.hxx
===================================================================
RCS file: /cvs/util/svtools/inc/instrm.hxx,v
retrieving revision 1.1.1.1
retrieving revision 1.1.1.1.650.1
diff -u -p -u -p -r1.1.1.1 -r1.1.1.1.650.1
--- svtools/inc/instrm.hxx	18 Sep 2000 16:58:51 -0000	1.1.1.1
+++ svtools/inc/instrm.hxx	21 Jan 2005 15:09:32 -0000	1.1.1.1.650.1
@@ -84,19 +84,19 @@ class SvInputStream: public SvStream
 	com::sun::star::uno::Reference< com::sun::star::io::XSeekable >
 	    m_xSeekable;
 	SvDataPipe_Impl * m_pPipe;
-	ULONG m_nSeekedFrom;
+	sal_Size m_nSeekedFrom;
 
 	SVL_DLLPRIVATE bool open();
 
-	SVL_DLLPRIVATE virtual ULONG GetData(void * pData, ULONG nSize);
+	SVL_DLLPRIVATE virtual sal_Size GetData(void * pData, sal_Size nSize);
 
-	SVL_DLLPRIVATE virtual ULONG PutData(void const *, ULONG);
+	SVL_DLLPRIVATE virtual sal_Size PutData(void const *, sal_Size);
 
-	SVL_DLLPRIVATE virtual ULONG SeekPos(ULONG nPos);
+	SVL_DLLPRIVATE virtual sal_Size SeekPos(sal_Size nPos);
 
 	SVL_DLLPRIVATE virtual void FlushData();
 
-	SVL_DLLPRIVATE virtual void SetSize(ULONG);
+	SVL_DLLPRIVATE virtual void SetSize(sal_Size);
 
 public:
 	SvInputStream(
@@ -108,9 +108,9 @@ public:
 
 	virtual USHORT IsA() const;
 
-	virtual void AddMark(ULONG nPos);
+	virtual void AddMark(sal_Size nPos);
 
-	virtual void RemoveMark(ULONG nPos);
+	virtual void RemoveMark(sal_Size nPos);
 };
 
 #endif // SVTOOLS_INSTRM_HXX
Index: svtools/inc/outstrm.hxx
===================================================================
RCS file: /cvs/util/svtools/inc/outstrm.hxx,v
retrieving revision 1.1.1.1
retrieving revision 1.1.1.1.650.1
diff -u -p -u -p -r1.1.1.1 -r1.1.1.1.650.1
--- svtools/inc/outstrm.hxx	18 Sep 2000 16:58:52 -0000	1.1.1.1
+++ svtools/inc/outstrm.hxx	21 Jan 2005 15:09:33 -0000	1.1.1.1.650.1
@@ -79,15 +79,15 @@ class SvOutputStream: public SvStream
 	com::sun::star::uno::Reference< com::sun::star::io::XOutputStream >
 		m_xStream;
 
-	SVL_DLLPRIVATE virtual ULONG GetData(void *, ULONG);
+	SVL_DLLPRIVATE virtual sal_Size GetData(void *, sal_Size);
 
-	SVL_DLLPRIVATE virtual ULONG PutData(void const * pData, ULONG nSize);
+	SVL_DLLPRIVATE virtual sal_Size PutData(void const * pData, sal_Size nSize);
 
-	SVL_DLLPRIVATE virtual ULONG SeekPos(ULONG);
+	SVL_DLLPRIVATE virtual sal_Size SeekPos(sal_Size);
 
 	SVL_DLLPRIVATE virtual void FlushData();
 
-	SVL_DLLPRIVATE virtual void SetSize(ULONG);
+	SVL_DLLPRIVATE virtual void SetSize(sal_Size);
 
 public:
 	SvOutputStream(com::sun::star::uno::Reference<

Index: svtools/inc/poolitem.hxx
===================================================================
RCS file: /cvs/util/svtools/inc/poolitem.hxx,v
retrieving revision 1.8
retrieving revision 1.7.10.2
diff -u -p -u -p -r1.8 -r1.7.10.2
--- svtools/inc/poolitem.hxx	18 Jan 2005 14:22:53 -0000	1.8
+++ svtools/inc/poolitem.hxx	9 Mar 2005 23:07:05 -0000	1.7.10.2
@@ -230,17 +230,17 @@ friend class SfxItemPoolCache;
 friend class SfxItemSet;
 friend class SfxVoidItem;
 
-	ULONG                    nRefCount;                    // Referenzzaehler
+	sal_uInt32                   nRefCount;                    // Referenzzaehler
 	USHORT                   nWhich;
 	USHORT					 nKind;
 
 private:
-	inline void              SetRefCount( ULONG n );
+	inline void              SetRefCount( sal_uInt32 n );
 	inline void				 SetKind( USHORT n );
 public:
-	inline ULONG             AddRef( ULONG n = 1 ) const;
+	inline sal_uInt32             AddRef( sal_uInt32 n = 1 ) const;
 private:
-	inline ULONG             ReleaseRef( ULONG n = 1 ) const;
+	inline sal_uInt32             ReleaseRef( sal_uInt32 n = 1 ) const;
 	SVL_DLLPRIVATE long      Delete_Impl(void*);
 
 #if 0
@@ -284,7 +284,7 @@ public:
 	virtual SvStream&        Store( SvStream &, USHORT nItemVersion ) const;
 	virtual SfxPoolItem*     Clone( SfxItemPool *pPool = 0 ) const = 0;
 
-	ULONG                    GetRefCount() const { return nRefCount; }
+	sal_uInt32                    GetRefCount() const { return nRefCount; }
 	inline USHORT			 GetKind() const { return nKind; }
 
 	/** Read in a Unicode string from a streamed byte string representation.
@@ -340,7 +340,7 @@ private:
 
 // -----------------------------------------------------------------------
 
-inline void SfxPoolItem::SetRefCount( ULONG n )
+inline void SfxPoolItem::SetRefCount( sal_uInt32 n )
 {
 	DBG_CHKTHIS( SfxPoolItem, 0 );
 	nRefCount = n;
@@ -354,15 +354,15 @@ inline void SfxPoolItem::SetKind( USHORT
 	nKind = n;
 }
 
-inline ULONG SfxPoolItem::AddRef( ULONG n ) const
+inline sal_uInt32 SfxPoolItem::AddRef( sal_uInt32 n ) const
 {
 	DBG_CHKTHIS( SfxPoolItem, 0 );
 	DBG_ASSERT( nRefCount <= SFX_ITEMS_MAXREF, "AddRef mit nicht-Pool-Item" );
-	DBG_ASSERT( ULONG_MAX - nRefCount > n, "AddRef: Referenzzaehler ueberschlaegt sich" );
+	DBG_ASSERT( UINT_MAX - nRefCount > n, "AddRef: Referenzzaehler ueberschlaegt sich" );
 	return ( ((SfxPoolItem *)this)->nRefCount += n );
 }
 
-inline ULONG SfxPoolItem::ReleaseRef( ULONG n ) const
+inline sal_uInt32 SfxPoolItem::ReleaseRef( sal_uInt32 n ) const
 {
 	DBG_CHKTHIS( SfxPoolItem, 0 );
 	DBG_ASSERT( nRefCount <= SFX_ITEMS_MAXREF, "AddRef mit nicht-Pool-Item" );
Index: svtools/inc/zforlist.hxx
===================================================================
RCS file: /cvs/util/svtools/inc/zforlist.hxx,v
retrieving revision 1.32
retrieving revision 1.32.146.1
diff -u -p -u -p -r1.32 -r1.32.146.1
--- svtools/inc/zforlist.hxx	25 Jun 2004 17:22:53 -0000	1.32
+++ svtools/inc/zforlist.hxx	21 Jan 2005 17:16:07 -0000	1.32.146.1
@@ -453,14 +453,14 @@ public:
 			nType contains the type of the format.
 			nKey contains the index key of the format.
 	 */
-	BOOL PutEntry( String& rString, xub_StrLen& nCheckPos, short& nType, ULONG& nKey,
+	BOOL PutEntry( String& rString, xub_StrLen& nCheckPos, short& nType, sal_uInt32& nKey,
 				  LanguageType eLnge = LANGUAGE_DONTKNOW );
 
 	/** Same as <method>PutEntry</method> but the format code string is
 	 	considered to be of language/country eLnge and is converted to
 		language/country eNewLnge */
 	BOOL PutandConvertEntry( String& rString, xub_StrLen& nCheckPos,
-							 short& nType, ULONG& nKey,
+							 short& nType, sal_uInt32& nKey,
 							 LanguageType eLnge, LanguageType eNewLnge );
 
 	/** Same as <method>PutandConvertEntry</method> but the format code string
@@ -468,12 +468,12 @@ public:
 		converted to another System language/country eNewLnge. In this case
 	 	the automatic currency is converted too. */
 	BOOL PutandConvertEntrySystem( String& rString, xub_StrLen& nCheckPos,
-							 short& nType, ULONG& nKey,
+							 short& nType, sal_uInt32& nKey,
 							 LanguageType eLnge, LanguageType eNewLnge );
 
 	/** Create a format code string using format nIndex as a template and
 		applying other settings (passed from the dialog) */
-	void GenerateFormat( String& sString, ULONG nIndex,
+	void GenerateFormat( String& sString, sal_uInt32 nIndex,
 						LanguageType eLnge = LANGUAGE_DONTKNOW,
 						BOOL bThousand = FALSE, BOOL IsRed = FALSE,
 						USHORT nPrecision = 0, USHORT nAnzLeading = 1 );
@@ -537,14 +537,14 @@ public:
 	void PrepareSave();
 
 	/// Flag format index as used
-	void SetFormatUsed(ULONG nFIndex);
+	void SetFormatUsed(sal_uInt32 nFIndex);
 
 	/// Get additional info of a format index, e.g. for dialog box
-	void GetFormatSpecialInfo(ULONG nFormat, BOOL& bThousand, BOOL& IsRed,
+	void GetFormatSpecialInfo(sal_uInt32 nFormat, BOOL& bThousand, BOOL& IsRed,
 							  USHORT& nPrecision, USHORT& nAnzLeading);
 
 	/// Count of decimals
-	USHORT GetFormatPrecision( ULONG nFormat ) const;
+	USHORT GetFormatPrecision( sal_uInt32 nFormat ) const;
 
 	/** Get additional info of a format code string, e.g. for dialog box.
 		Uses a temporary parse, if possible use only if format code is not
@@ -553,7 +553,7 @@ public:
 			0 if format code string parsed without errors, otherwise error
 			position (like nCheckPos on <method>PutEntry</method>)
 	 */
-	ULONG GetFormatSpecialInfo( const String&, BOOL& bThousand, BOOL& IsRed,
+	sal_uInt32 GetFormatSpecialInfo( const String&, BOOL& bThousand, BOOL& IsRed,
 							  USHORT& nPrecision, USHORT& nAnzLeading,
 							  LanguageType eLnge = LANGUAGE_DONTKNOW );
 
@@ -655,7 +655,7 @@ public:
     String GetDecimalSep() const { return GetNumDecimalSep(); }
 
 	/// Return the decimal separator matching the locale of the given format
-	String GetFormatDecimalSep( ULONG nFormat ) const;
+	String GetFormatDecimalSep( sal_uInt32 nFormat ) const;
 
 	/// Return a <type>SvPtrArr</type> with pointers to <type>NfCurrencyEntry</type> entries
     static const NfCurrencyTable& GetTheCurrencyTable();
@@ -708,7 +708,7 @@ public:
 		If ppEntry is not NULL and exactly one entry is found, a [$xxx-nnn] is
 		returned, even if the format code only contains [$xxx] !
 	 */
-	BOOL	GetNewCurrencySymbolString( ULONG nFormat, String& rSymbol,
+	BOOL	GetNewCurrencySymbolString( sal_uInt32 nFormat, String& rSymbol,
 				const NfCurrencyEntry** ppEntry = NULL, BOOL* pBank = NULL ) const;
 
 	/** Look up the corresponding <type>NfCurrencyEntry</type> matching
@@ -868,15 +868,15 @@ private:
 
 	// Return the format index of the currency format of the system locale.
 	// Format is created if not already present.
-	SVT_DLLPRIVATE ULONG	ImpGetDefaultSystemCurrencyFormat();
+	SVT_DLLPRIVATE sal_uInt32	ImpGetDefaultSystemCurrencyFormat();
 
 	// Return the format index of the currency format of the current locale.
 	// Format is created if not already present.
-	SVT_DLLPRIVATE ULONG	ImpGetDefaultCurrencyFormat();
+	SVT_DLLPRIVATE sal_uInt32	ImpGetDefaultCurrencyFormat();
 
 	// Return the default format for a given type and current locale.
 	// May ONLY be called from within GetStandardFormat().
-	SVT_DLLPRIVATE ULONG	ImpGetDefaultFormat( short nType );
+	SVT_DLLPRIVATE sal_uInt32	ImpGetDefaultFormat( short nType );
 
 	// Return the index in a sequence of format codes matching an enum of
 	// NfIndexTableOffset. If not found 0 is returned. If the sequence doesn't
