--- fpicker/source/office/iodlg.cxx	16 Feb 2005 18:00:57 -0000	1.84
+++ fpicker/source/office/iodlg.cxx	9 Mar 2005 23:13:58 -0000	1.79.44.4
@@ -2673,7 +2673,7 @@ void SvtFileDialog::Resize()
 	Size aDlgSize = GetResizeOutputSizePixel();
 	Size aOldSize = _pImp->_aDlgSize;
 	_pImp->_aDlgSize = aDlgSize;
-	long nWinDeltaW = 0;
+	sal_Int32 nWinDeltaW = 0;
 
 	if ( _pPrevWin &&
          _pPrevWin->GetPosPixel().X() > _pFileView->GetPosPixel().X() )
@@ -2683,8 +2683,8 @@ void SvtFileDialog::Resize()
 
 	Size aNewSize = _pFileView->GetSizePixel();
 	Point aBoxPos( _pFileView->GetPosPixel() );
-	long nDeltaY = aNewSize.Height();
-	long nDeltaX = aNewSize.Width();
+	sal_Int32 nDeltaY = aNewSize.Height();
+	sal_Int32 nDeltaX = aNewSize.Width();
 	aNewSize.Height() = aDlgSize.Height() - _pImp->_nFixDeltaHeight;
 	aNewSize.Width() = aDlgSize.Width() - aBoxPos.X() - _pImp->_a6Size.Width() - nWinDeltaW;
 	if ( aOldSize.Height() )
@@ -2704,7 +2704,7 @@ void SvtFileDialog::Resize()
 		// Dieses Resize wurde nur zum Ein - oder Ausblenden des Indicators aufgerufen
 		return;
 
-	long nMaxYPos = 0;
+	sal_Int32 nMaxYPos = 0;
 
 	// -------------
 	// move controls
