--- sw/source/core/doc/dbgoutsw.cxx	2005-05-09 10:39:39.087821518 +0200
+++ sw/source/core/doc/dbgoutsw.cxx	2005-05-09 10:39:35.542937888 +0200
@@ -381,7 +381,7 @@ static String lcl_dbg_out(const SwPositi
     aStr += String::CreateFromInt32(rPos.nContent.GetIndex());
     aStr += String(": ", RTL_TEXTENCODING_ASCII_US);
     aStr += String::CreateFromInt32
-        (reinterpret_cast<sal_Int32>(rPos.nContent.GetIdxReg()), 16);
+        (reinterpret_cast<sal_IntPtr>(rPos.nContent.GetIdxReg()), 16);
 
     aStr += String(" )", RTL_TEXTENCODING_ASCII_US);
 
