Index: vcl/inc/bitmap.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/bitmap.hxx,v
retrieving revision 1.18
retrieving revision 1.17.76.2
diff -u -p -u -p -r1.18 -r1.17.76.2
--- vcl/inc/bitmap.hxx	2 Nov 2005 13:26:51 -0000	1.18
+++ vcl/inc/bitmap.hxx	16 Jan 2006 13:20:20 -0000	1.17.76.2
@@ -290,7 +290,7 @@ public:
     SAL_DLLPRIVATE static BOOL          ImplWriteDIBFileHeader( SvStream& rOStm, BitmapReadAccess& rAcc );
     SAL_DLLPRIVATE static BOOL          ImplWriteDIBPalette( SvStream& rOStm, BitmapReadAccess& rAcc );
     SAL_DLLPRIVATE static BOOL          ImplWriteDIBBits( SvStream& rOStm, BitmapReadAccess& rAcc,
-                                                             ULONG nCompression, ULONG& rImageSize );
+                                                             ULONG nCompression, sal_uInt32& rImageSize );
     SAL_DLLPRIVATE static void          ImplDecodeRLE( BYTE* pBuffer, DIBInfoHeader& rHeader,
                                            BitmapWriteAccess& rAcc, BOOL bRLE4 );
     SAL_DLLPRIVATE static BOOL          ImplWriteRLE( SvStream& rOStm, BitmapReadAccess& rAcc, BOOL bRLE4 );
Index: vcl/inc/dockwin.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/dockwin.hxx,v
retrieving revision 1.10
retrieving revision 1.10.78.1
diff -u -p -u -p -r1.10 -r1.10.78.1
--- vcl/inc/dockwin.hxx	9 Sep 2005 10:54:15 -0000	1.10
+++ vcl/inc/dockwin.hxx	26 Oct 2005 18:07:07 -0000	1.10.78.1
@@ -114,10 +114,10 @@ private:
 	long			mnTrackY;
 	long			mnTrackWidth;
 	long			mnTrackHeight;
-	long			mnDockLeft;
-	long			mnDockTop;
-	long			mnDockRight;
-	long			mnDockBottom;
+	sal_Int32		mnDockLeft;
+	sal_Int32		mnDockTop;
+	sal_Int32		mnDockRight;
+	sal_Int32		mnDockBottom;
 	WinBits 		mnFloatBits;
 	BOOL			mbDockCanceled:1,
 					mbFloatPrevented:1,
Index: vcl/inc/imgcons.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/imgcons.hxx,v
retrieving revision 1.5
retrieving revision 1.4.78.2
diff -u -p -u -p -r1.5 -r1.4.78.2
--- vcl/inc/imgcons.hxx	14 Dec 2005 14:58:07 -0000	1.5
+++ vcl/inc/imgcons.hxx	16 Jan 2006 13:21:20 -0000	1.4.78.2
@@ -84,7 +84,7 @@ public:					
 	BOOL				GetData( BitmapEx& rBmpEx ) const;
 	BOOL				GetData( Image& rImage ) const;
 	const Rectangle&	GetChangedRect() const { return maChangedRect; }
-	ULONG				GetStatus() const;
+	sal_uInt32			GetStatus() const;
 						
 	void				SetDataChangedLink( const Link& rLink ) { maChgLink = rLink; }
 	const Link&			GetDataChangedLink() const { return maChgLink; }
@@ -94,22 +94,22 @@ public:					
 
 public:
 
-	virtual void		Init( ULONG nWidth, ULONG nHeight );
+	virtual void		Init( sal_uInt32 nWidth, sal_uInt32 nHeight );
 
 	virtual void		SetColorModel( USHORT nBitCount,
-									   ULONG nPalEntries, const ULONG* pRGBAPal, 
-									   ULONG nRMask, ULONG nGMask, ULONG nBMask, ULONG nAMask  );
+									   sal_uInt32 nPalEntries, const sal_uInt32* pRGBAPal, 
+									   sal_uInt32 nRMask, sal_uInt32 nGMask, sal_uInt32 nBMask, sal_uInt32 nAMask  );
 						
-	virtual	void		SetPixelsByBytes( ULONG nConsX, ULONG nConsY, 
-										  ULONG nConsWidth, ULONG nConsHeight,
-										  const BYTE* pProducerData, ULONG nOffset, ULONG nScanSize );
+	virtual	void		SetPixelsByBytes( sal_uInt32 nConsX, sal_uInt32 nConsY, 
+										  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+										  const BYTE* pProducerData, sal_uInt32 nOffset, sal_uInt32 nScanSize );
 						
-	virtual	void		SetPixelsByLongs( ULONG nConsX, ULONG nConsY, 
-										  ULONG nConsWidth, ULONG nConsHeight,
-										  const ULONG* pProducerData, ULONG nOffset, ULONG nScanSize );
+	virtual	void		SetPixelsByLongs( sal_uInt32 nConsX, sal_uInt32 nConsY, 
+										  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+										  const sal_uInt32* pProducerData, sal_uInt32 nOffset, sal_uInt32 nScanSize );
 						
-	virtual void		Completed( ULONG nStatus );
-//	virtual void		Completed( ULONG nStatus, ImageProducer& rProducer );
+	virtual void		Completed( sal_uInt32 nStatus );
+//	virtual void		Completed( sal_uInt32 nStatus, ImageProducer& rProducer );
 };
 
 #endif // _IMGCONS_HXX
Index: vcl/source/control/ilstbox.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/ilstbox.cxx,v
retrieving revision 1.54
retrieving revision 1.53.44.2
diff -u -p -u -p -r1.54 -r1.53.44.2
--- vcl/source/control/ilstbox.cxx	26 Jan 2006 18:08:11 -0000	1.54
+++ vcl/source/control/ilstbox.cxx	23 Feb 2006 11:10:25 -0000	1.53.44.2
@@ -2627,7 +2627,7 @@ void ImplWin::ImplDraw( bool bLayout )
 	        // The listbox is painted over the entire control including the
 	        // border, but ImplWin does not contain the border => correction
 	        // needed.
-	        long nLeft, nTop, nRight, nBottom;
+	        sal_Int32 nLeft, nTop, nRight, nBottom;
 	        pWin->GetBorder( nLeft, nTop, nRight, nBottom );
 	        Point aPoint( -nLeft, -nTop );
 	        Region aCtrlRegion( Rectangle( aPoint - GetPosPixel(), pWin->GetSizePixel() ) );
Index: vcl/source/gdi/bitmap2.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/bitmap2.cxx,v
retrieving revision 1.10
retrieving revision 1.9.76.2
diff -u -p -u -p -r1.10 -r1.9.76.2
--- vcl/source/gdi/bitmap2.cxx	9 Feb 2006 14:50:28 -0000	1.10
+++ vcl/source/gdi/bitmap2.cxx	23 Feb 2006 11:10:50 -0000	1.9.76.2
@@ -867,7 +867,7 @@ BOOL Bitmap::ImplWriteDIBPalette( SvStre
 // ------------------------------------------------------------------
 
 BOOL Bitmap::ImplWriteDIBBits( SvStream& rOStm, BitmapReadAccess& rAcc,
-							   ULONG nCompression, ULONG& rImageSize )
+							   ULONG nCompression, sal_uInt32& rImageSize )
 {
 	if( BITFIELDS == nCompression )
 	{
Index: vcl/source/gdi/imgcons.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/imgcons.cxx,v
retrieving revision 1.4
retrieving revision 1.4.78.1
diff -u -p -u -p -r1.4 -r1.4.78.1
--- vcl/source/gdi/imgcons.cxx	9 Sep 2005 12:00:11 -0000	1.4
+++ vcl/source/gdi/imgcons.cxx	26 Oct 2005 18:07:08 -0000	1.4.78.1
@@ -134,7 +134,7 @@ ImageConsumer::~ImageConsumer()
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::Init( ULONG nWidth, ULONG nHeight )
+void ImageConsumer::Init( sal_uInt32 nWidth, sal_uInt32 nHeight )
 {
 	maSize = Size( nWidth, nHeight );
 	maBitmap = maMask = Bitmap();
@@ -145,8 +145,8 @@ void ImageConsumer::Init( ULONG nWidth, 
 // -----------------------------------------------------------------------------
 
 void ImageConsumer::SetColorModel( USHORT nBitCount,
-								   ULONG nPalEntries, const ULONG* pRGBAPal, 
-								   ULONG nRMask, ULONG nGMask, ULONG nBMask, ULONG nAMask )
+								   sal_uInt32 nPalEntries, const sal_uInt32* pRGBAPal, 
+								   sal_uInt32 nRMask, sal_uInt32 nGMask, sal_uInt32 nBMask, sal_uInt32 nAMask )
 {
 	DBG_ASSERT( maSize.Width() && maSize.Height(), "Missing call to ImageConsumer::Init(...)!" );
 
@@ -221,9 +221,9 @@ void ImageConsumer::SetColorModel( USHOR
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::SetPixelsByBytes( ULONG nConsX, ULONG nConsY,
-									  ULONG nConsWidth, ULONG nConsHeight,
-									  const BYTE* pData, ULONG nOffset, ULONG nScanSize )
+void ImageConsumer::SetPixelsByBytes( sal_uInt32 nConsX, sal_uInt32 nConsY,
+									  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+									  const BYTE* pData, sal_uInt32 nOffset, sal_uInt32 nScanSize )
 {
 	DBG_ASSERT( !!maBitmap && !!maMask, "Missing call to ImageConsumer::SetColorModel(...)!" );
 
@@ -358,9 +358,9 @@ void ImageConsumer::SetPixelsByBytes( UL
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::SetPixelsByLongs( ULONG nConsX, ULONG nConsY, 
-									  ULONG nConsWidth, ULONG nConsHeight,
-									  const ULONG* pData, ULONG nOffset, ULONG nScanSize )
+void ImageConsumer::SetPixelsByLongs( sal_uInt32 nConsX, sal_uInt32 nConsY, 
+									  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+									  const sal_uInt32* pData, sal_uInt32 nOffset, sal_uInt32 nScanSize )
 {
 	DBG_ASSERT( !!maBitmap && !!maMask, "Missing call to ImageConsumer::SetColorModel(...)!" );
 
@@ -495,7 +495,7 @@ void ImageConsumer::SetPixelsByLongs( UL
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::Completed( ULONG nStatus /*, ImageProducer& rProducer */ )
+void ImageConsumer::Completed( sal_uInt32 nStatus /*, ImageProducer& rProducer */ )
 {
 	delete mpMapper;
 	mpMapper = NULL;
@@ -539,7 +539,7 @@ void ImageConsumer::DataChanged()
 
 // -----------------------------------------------------------------------------
 
-ULONG ImageConsumer::GetStatus() const
+sal_uInt32 ImageConsumer::GetStatus() const
 {
 	return mnStatus;
 }
Index: vcl/source/gdi/pdfwriter_impl.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/pdfwriter_impl.cxx,v
retrieving revision 1.89
retrieving revision 1.84.10.3
diff -u -p -u -p -r1.89 -r1.84.10.3
--- vcl/source/gdi/pdfwriter_impl.cxx	9 Feb 2006 13:39:47 -0000	1.89
+++ vcl/source/gdi/pdfwriter_impl.cxx	23 Feb 2006 11:11:40 -0000	1.84.10.3
@@ -3618,7 +3618,7 @@ void PDFWriterImpl::createDefaultPushBut
     OStringBuffer aDA( 256 );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetButtonTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[m_nCurrentPage].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[m_nCurrentPage].appendMappedLength( sal_Int32( aFont.GetHeight() ), aDA );
     aDA.append( " Tf" );
     rButton.m_aDAString = aDA.makeStringAndClear();
 
@@ -3706,7 +3706,7 @@ void PDFWriterImpl::createDefaultEditApp
     OStringBuffer aDA( 32 );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetFieldTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( sal_Int32( aFont.GetHeight() ), aDA );
     aDA.append( " Tf" );
 
     /*  create an empty appearance stream, let the viewer create
@@ -3787,7 +3787,7 @@ void PDFWriterImpl::createDefaultListBox
 #endif
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetFieldTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( sal_Int32( aFont.GetHeight() ), aDA );
     aDA.append( " Tf" );
     rBox.m_aDAString = aDA.makeStringAndClear();
 }
@@ -3875,7 +3875,7 @@ void PDFWriterImpl::createDefaultCheckBo
     aDA.append( "/Tx BMC\r\nq BT\r\n" );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ), aDA );
     aDA.append( " /ZaDb " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aCheckRect.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( sal_Int32( aCheckRect.GetHeight() ), aDA );
     aDA.append( " Tf\r\n" );
     m_aPages[ m_nCurrentPage ].appendMappedLength( nCharXOffset, aDA );
     aDA.append( " " );
@@ -3970,7 +3970,7 @@ void PDFWriterImpl::createDefaultRadioBu
     aDA.append( "/Tx BMC\r\nq BT\r\n" );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ), aDA );
     aDA.append( " /ZaDb " );
-    m_aPages[m_nCurrentPage].appendMappedLength( aCheckRect.GetHeight(), aDA );
+    m_aPages[m_nCurrentPage].appendMappedLength( sal_Int32( aCheckRect.GetHeight() ), aDA );
     aDA.append( " Tf\r\n0 0 Td\r\nET\r\nQ\r\n" );
     writeBuffer( aDA.getStr(), aDA.getLength() );
     setFillColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ) );
Index: vcl/source/gdi/pngwrite.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/pngwrite.cxx,v
retrieving revision 1.6
retrieving revision 1.6.78.1
diff -u -p -u -p -r1.6 -r1.6.78.1
--- vcl/source/gdi/pngwrite.cxx	9 Sep 2005 12:09:34 -0000	1.6
+++ vcl/source/gdi/pngwrite.cxx	26 Oct 2005 18:07:09 -0000	1.6.78.1
@@ -123,7 +123,7 @@ private:
 	void				ImplWritePalette();
 	void				ImplOpenChunk( ULONG nChunkType );
 	void				ImplWriteChunk( BYTE nNumb );
-	void				ImplWriteChunk( ULONG nNumb );
+	void				ImplWriteChunk( sal_uInt32 nNumb );
 	void				ImplWriteChunk( unsigned char* pSource, sal_uInt32 nDatSize );
 	void				ImplCloseChunk( void );
 };
@@ -306,8 +306,8 @@ std::vector< vcl::PNGWriter::ChunkData >
 BOOL PNGWriterImpl::ImplWriteHeader()
 {
 	ImplOpenChunk(PNGCHUNK_IHDR);
-	ImplWriteChunk( ( mnWidth = (ULONG)mpAccess->Width() ) );
-	ImplWriteChunk( ( mnHeight = (ULONG)mpAccess->Height() ) );
+	ImplWriteChunk( sal_uInt32( mnWidth =  mpAccess->Width() ) );
+	ImplWriteChunk( sal_uInt32( mnHeight = mpAccess->Height() ) );
 
 	if ( mnWidth && mnHeight && mnBitsPerPixel && mbStatus )
 	{
@@ -664,7 +664,7 @@ void PNGWriterImpl::ImplWriteChunk ( BYT
 	maChunkSeq.back().aData.push_back( nSource );
 }
 
-void PNGWriterImpl::ImplWriteChunk ( ULONG nSource )
+void PNGWriterImpl::ImplWriteChunk ( sal_uInt32 nSource )
 {
 	vcl::PNGWriter::ChunkData& rChunkData = maChunkSeq.back();
 	rChunkData.aData.push_back( (sal_uInt8)( nSource >> 24 ) );
Index: vcl/source/window/dockmgr.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/dockmgr.cxx,v
retrieving revision 1.16
retrieving revision 1.12.78.3
diff -u -p -u -p -r1.16 -r1.12.78.3
--- vcl/source/window/dockmgr.cxx	9 Feb 2006 17:12:43 -0000	1.16
+++ vcl/source/window/dockmgr.cxx	23 Feb 2006 11:12:54 -0000	1.12.78.3
@@ -224,7 +224,7 @@ IMPL_LINK( ImplDockFloatWin2, DockingHdl
         {
             Point aPt;
             Rectangle aBorderRect( aPt, pBorder->GetSizePixel() );
-            long nLeft, nTop, nRight, nBottom;
+            sal_Int32 nLeft, nTop, nRight, nBottom;
             GetBorder( nLeft, nTop, nRight, nBottom );
             // limit borderrect to the caption part only and without the resizing borders
             aBorderRect.nBottom = aBorderRect.nTop + nTop;
Index: vcl/unx/source/app/i18n_cb.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/i18n_cb.cxx,v
retrieving revision 1.27
retrieving revision 1.25.76.3
diff -u -p -u -p -r1.27 -r1.25.76.3
--- vcl/unx/source/app/i18n_cb.cxx	16 Jan 2006 13:07:13 -0000	1.27
+++ vcl/unx/source/app/i18n_cb.cxx	23 Feb 2006 11:15:08 -0000	1.25.76.3
@@ -507,8 +507,8 @@ PreeditCaretCallback ( XIC ic, XPointer 
 		case XIMDontChange:   direction = "Dont change";   break;
 	}
 
-	fprintf (stderr, "PreeditCaretCallback( ic=%i, client=%i,\n",
-		(unsigned int)ic, (unsigned int)client_data );
+	fprintf (stderr, "PreeditCaretCallback( ic=%p, client=%p,\n",
+		ic, client_data );
 	fprintf (stderr, "\t position=%i, direction=\"%s\", style=\"%s\" )\n",
 		call_data->position, direction, style );
     #endif
Index: vcl/unx/source/gdi/salgdi3.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/salgdi3.cxx,v
retrieving revision 1.131
retrieving revision 1.127.40.3
diff -u -p -u -p -r1.131 -r1.127.40.3
--- vcl/unx/source/gdi/salgdi3.cxx	27 Jan 2006 13:50:19 -0000	1.131
+++ vcl/unx/source/gdi/salgdi3.cxx	23 Feb 2006 11:16:42 -0000	1.127.40.3
@@ -447,7 +447,7 @@ SalDisplay::GetFont( const ExtendedXlfd 
 				if( pFontCache_->GetCurPos() )
 				{
 					pFontCache_->Remove( pItem );
-					pFontCache_->Insert( pItem, (sal_uInt32)0 );
+					pFontCache_->Insert( pItem, 0UL );
 				}
 				return pItem;
 			}
@@ -475,7 +475,7 @@ SalDisplay::GetFont( const ExtendedXlfd 
 	ExtendedFontStruct *pItem = new ExtendedFontStruct( GetDisplay(),
 										rPixelSize, bVertical,
 										const_cast<ExtendedXlfd*>(pRequestedFont) );
-	pFontCache_->Insert( pItem, (sal_uInt32)0 );
+	pFontCache_->Insert( pItem, 0UL );
 	pItem->AddRef();
 
 	return pItem;
