--- solenv/inc/unxlngx6.mk	2005-11-07 13:13:22.000000000 +0100
+++ solenv/inc/unxlngx6.mk	2005-11-07 13:08:02.000000000 +0100
@@ -120,7 +120,7 @@ CFLAGSDEBUG=-g
 CFLAGSDBGUTIL=
 # Compiler flags for enabling optimizations
 .IF "$(PRODUCT)"!=""
-CFLAGSOPT=-Os -fno-strict-aliasing		# optimizing for products
+CFLAGSOPT=-O2 -fno-strict-aliasing		# optimizing for products
 CFLAGSOPT+=-Wuninitialized				# not supported without optimization
 .ELSE 	# "$(PRODUCT)"!=""
 CFLAGSOPT=   							# no optimizing for non products
