--- vcl/inc/window.hxx	2005-06-15 08:24:57.981585729 +0200
+++ vcl/inc/window.hxx	2005-06-15 08:29:26.562535439 +0200
@@ -876,8 +876,8 @@
     BOOL                IsCompoundControl() const;
     BOOL                HasCompoundControlFocus() const;
 
-    static ULONG        SaveFocus();
-    static BOOL         EndSaveFocus( ULONG nSaveId, BOOL bRestore = TRUE );
+    static sal_IntPtr   SaveFocus();
+    static BOOL         EndSaveFocus( sal_IntPtr nSaveId, BOOL bRestore = TRUE );
 
     void                CaptureMouse();
     void                ReleaseMouse();
--- vcl/source/window/window2.cxx	2005-06-15 08:24:58.185536166 +0200
+++ vcl/source/window/window2.cxx	2005-06-15 08:28:18.613956292 +0200
@@ -903,7 +903,7 @@
 
 // -----------------------------------------------------------------------
 
-ULONG Window::SaveFocus()
+sal_IntPtr Window::SaveFocus()
 {
     ImplSVData* pSVData = ImplGetSVData();
     if ( pSVData->maWinData.mpFocusWin )
@@ -911,7 +911,7 @@
         ImplFocusDelData* pDelData = new ImplFocusDelData;
         pSVData->maWinData.mpFocusWin->ImplAddDel( pDelData );
         pDelData->mpFocusWin = pSVData->maWinData.mpFocusWin;
-        return (ULONG)(void*)pDelData;
+        return (sal_IntPtr)(void*)pDelData;
     }
     else
         return 0;
@@ -919,7 +919,7 @@
 
 // -----------------------------------------------------------------------
 
-BOOL Window::EndSaveFocus( ULONG nSaveId, BOOL bRestore )
+BOOL Window::EndSaveFocus( sal_IntPtr nSaveId, BOOL bRestore )
 {
     if ( !nSaveId )
         return FALSE;
--- svtools/source/dialogs/wizardmachine.cxx	2005-06-15 08:24:16.181692999 +0200
+++ svtools/source/dialogs/wizardmachine.cxx	2005-06-15 08:26:16.251585812 +0200
@@ -502,7 +502,7 @@
 				pChildLoop->SetStyle(pChildLoop->GetStyle() & ~WB_DEFBUTTON);
 
 				// give the button the focus, this is the best method to enforce it to be repainted
-				sal_uInt32 nSaveFocusId = Window::SaveFocus();
+				sal_IntPtr nSaveFocusId = Window::SaveFocus();
 				pChildLoop->GrabFocus();
 				Window::EndSaveFocus(nSaveFocusId);
 			}
@@ -524,7 +524,7 @@
 		{
 			_pNewDefButton->SetStyle(_pNewDefButton->GetStyle() | WB_DEFBUTTON);
 			// give the button the focus, this is the best method to enforce it to be repainted
-			sal_uInt32 nSaveFocusId = Window::SaveFocus();
+			sal_IntPtr nSaveFocusId = Window::SaveFocus();
 			_pNewDefButton->GrabFocus();
 			Window::EndSaveFocus(nSaveFocusId);
 		}
