cvs diff: Diffing source/polygon
Index: basegfx/source/polygon/b2dpolypolygonrasterconverter.cxx
===================================================================
RCS file: /cvs/graphics/basegfx/source/polygon/b2dpolypolygonrasterconverter.cxx,v
retrieving revision 1.3
diff -u -r1.3 b2dpolypolygonrasterconverter.cxx
--- basegfx/source/polygon/b2dpolypolygonrasterconverter.cxx	10 Mar 2005 13:38:47 -0000	1.3
+++ basegfx/source/polygon/b2dpolypolygonrasterconverter.cxx	4 Jun 2005 16:16:29 -0000
@@ -147,7 +147,7 @@
 			m_indices2 = new sal_uInt32[nNumElements];
 
 			// check for out of memory situation
-			if(!((sal_uInt32)m_indices1|(sal_uInt32)m_indices2)) {
+			if(!m_indices1 || !m_indices2) {
 				delete [] m_indices1;
 				delete [] m_indices2;
 				m_indices1 = NULL;
