--- fpicker/source/office/asyncfilepicker.cxx	22 Oct 2004 12:34:51 -0000	1.2
+++ fpicker/source/office/asyncfilepicker.cxx	21 Jan 2005 17:16:25 -0000	1.2.122.1
@@ -205,7 +205,7 @@ namespace svt
         DBG_TESTSOLARMUTEX();
             // if this asserts, we'd need to have an own mutex per instance
 
-        FileViewResult eResult = static_cast< FileViewResult >( reinterpret_cast< int >( pEmptyArg ) );
+        FileViewResult eResult = static_cast< FileViewResult >( reinterpret_cast< sal_IntPtr >( pEmptyArg ) );
         OSL_ENSURE( eStillRunning != eResult, "AsyncPickerAction::OnActionDone: invalid result!" );
 
         // release once (since we acquired in |execute|), but keep alive until the
