Index: svx/source/dialog/frmdirlbox.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/frmdirlbox.cxx,v
retrieving revision 1.2
retrieving revision 1.2.150.1
diff -u -p -u -p -r1.2 -r1.2.150.1
--- svx/source/dialog/frmdirlbox.cxx	2 Aug 2004 17:44:27 -0000	1.2
+++ svx/source/dialog/frmdirlbox.cxx	21 Jan 2005 16:07:02 -0000	1.2.150.1
@@ -78,7 +78,7 @@ inline void* lclEnumToVoid( SvxFrameDire
 
 inline SvxFrameDirection lclVoidToEnum( void* pDirection )
 {
-    return static_cast< SvxFrameDirection >( reinterpret_cast< sal_uInt32 >( pDirection ) );
+    return static_cast< SvxFrameDirection >( reinterpret_cast< sal_IntPtr >( pDirection ) );
 }
 
 } // namespace
