Index: extensions/source/dbpilots/commonpagesdbp.cxx
===================================================================
RCS file: /cvs/util/extensions/source/dbpilots/commonpagesdbp.cxx,v
retrieving revision 1.10
retrieving revision 1.10.16.1
diff -u -p -u -p -r1.10 -r1.10.16.1
--- extensions/source/dbpilots/commonpagesdbp.cxx	5 Jan 2005 12:42:15 -0000	1.10
+++ extensions/source/dbpilots/commonpagesdbp.cxx	21 Jan 2005 16:21:00 -0000	1.10.16.1
@@ -220,7 +220,7 @@ namespace dbp
             for ( USHORT nLookup = 0; nLookup < m_aTable.GetEntryCount(); ++nLookup )
             {
                 if ( m_aTable.GetEntry( nLookup ) == sLookup )
-                    if ( reinterpret_cast< sal_Int32 >( m_aTable.GetEntryData( nLookup ) ) == nCommandType )
+                    if ( reinterpret_cast< sal_IntPtr >( m_aTable.GetEntryData( nLookup ) ) == nCommandType )
                     {
                         m_aTable.SelectEntryPos( nLookup );
                         break;
@@ -251,7 +251,7 @@ namespace dbp
 				rContext.xForm->setPropertyValue( ::rtl::OUString::createFromAscii("DataSourceName"), makeAny( sDataSource ) );
 			}
             ::rtl::OUString sCommand = m_aTable.GetSelectEntry();
-            sal_Int32 nCommandType = reinterpret_cast< sal_Int32 >( m_aTable.GetEntryData( m_aTable.GetSelectEntryPos() ) );
+            sal_Int32 nCommandType = reinterpret_cast< sal_IntPtr >( m_aTable.GetEntryData( m_aTable.GetSelectEntryPos() ) );
 			
 			rContext.xForm->setPropertyValue( ::rtl::OUString::createFromAscii("Command"), makeAny( sCommand ) );
 			rContext.xForm->setPropertyValue( ::rtl::OUString::createFromAscii("CommandType"), makeAny( nCommandType ) );
Index: extensions/source/dbpilots/gridwizard.cxx
===================================================================
RCS file: /cvs/util/extensions/source/dbpilots/gridwizard.cxx,v
retrieving revision 1.12
retrieving revision 1.11.36.2
diff -u -p -u -p -r1.12 -r1.11.36.2
--- extensions/source/dbpilots/gridwizard.cxx	17 Feb 2005 11:11:53 -0000	1.12
+++ extensions/source/dbpilots/gridwizard.cxx	9 Mar 2005 16:00:57 -0000	1.11.36.2
@@ -470,7 +470,7 @@ namespace dbp
 		// the index of the selected entry
 		WizardState nSelected = bMoveRight ? m_aExistFields.GetSelectEntryPos() : m_aSelFields.GetSelectEntryPos();
 		// the (original) relative position of the entry
-		sal_Int32 nRelativeIndex = reinterpret_cast<sal_Int32>(bMoveRight ? m_aExistFields.GetEntryData(nSelected) : m_aSelFields.GetEntryData(nSelected));
+		sal_IntPtr nRelativeIndex = reinterpret_cast<sal_IntPtr>(bMoveRight ? m_aExistFields.GetEntryData(nSelected) : m_aSelFields.GetEntryData(nSelected));
 
 		WizardState nInsertPos = LISTBOX_APPEND;
 		if (!bMoveRight)
@@ -478,7 +478,7 @@ namespace dbp
 			nInsertPos = 0;
 			while (nInsertPos < rMoveTo.GetEntryCount())
 			{
-				if (reinterpret_cast<sal_Int32>(rMoveTo.GetEntryData(nInsertPos)) > nRelativeIndex)
+				if (reinterpret_cast<sal_IntPtr>(rMoveTo.GetEntryData(nInsertPos)) > nRelativeIndex)
 					break;
 				++nInsertPos;
 			}
Index: extensions/source/scanner/grid.cxx
===================================================================
RCS file: /cvs/util/extensions/source/scanner/grid.cxx,v
retrieving revision 1.3
diff -u -r1.3 grid.cxx
--- extensions/source/scanner/grid.cxx	27 Oct 2003 13:21:34 -0000	1.3
+++ extensions/source/scanner/grid.cxx	5 Jun 2005 10:21:12 -0000
@@ -582,7 +582,7 @@
 {
 	if( pButton == &m_aResetButton )
 	{
-		int nType = (int)m_aResetTypeBox.GetEntryData( m_aResetTypeBox.GetSelectEntryPos() );
+		int nType = (int)(sal_IntPtr)m_aResetTypeBox.GetEntryData( m_aResetTypeBox.GetSelectEntryPos() );
 		switch( nType )
 		{
 			case RESET_TYPE_LINEAR_ASCENDING:
