Index: toolkit/source/awt/vclxaccessiblelist.cxx
===================================================================
RCS file: /cvs/gsl/toolkit/source/awt/vclxaccessiblelist.cxx,v
retrieving revision 1.18
retrieving revision 1.18.158.1
diff -u -p -u -p -r1.18 -r1.18.158.1
--- toolkit/source/awt/vclxaccessiblelist.cxx	22 May 2003 13:00:20 -0000	1.18
+++ toolkit/source/awt/vclxaccessiblelist.cxx	29 Nov 2004 19:28:09 -0000	1.18.158.1
@@ -300,13 +300,13 @@ void VCLXAccessibleList::ProcessWindowEv
 
         case VCLEVENT_LISTBOX_ITEMREMOVED:
         case VCLEVENT_COMBOBOX_ITEMREMOVED:
-            HandleChangedItemList (false, reinterpret_cast<sal_Int32>(
+            HandleChangedItemList (false, reinterpret_cast<sal_IntPtr>(
                 rVclWindowEvent.GetData()));
             break;
 
         case VCLEVENT_LISTBOX_ITEMADDED:
         case VCLEVENT_COMBOBOX_ITEMADDED:
-            HandleChangedItemList (true, reinterpret_cast<sal_Int32>(
+            HandleChangedItemList (true, reinterpret_cast<sal_IntPtr>(
                 rVclWindowEvent.GetData()));
             break;
 		case VCLEVENT_WINDOW_GETFOCUS:
