--- sc/source/core/tool/zforauto.cxx	17 Sep 2004 19:40:46 -0000	1.3
+++ sc/source/core/tool/zforauto.cxx	21 Jan 2005 16:39:13 -0000	1.3.52.1
@@ -154,7 +154,7 @@ ULONG ScNumFormatAbbrev::GetFormatIndex(
 
 	xub_StrLen nCheckPos;
 	short nType;
-	ULONG nKey = rFormatter.GetEntryKey(sFormatstring, eLnge);
+	sal_uInt32 nKey = rFormatter.GetEntryKey(sFormatstring, eLnge);
 	if (nKey == NUMBERFORMAT_ENTRY_NOT_FOUND)
 	{
 		BOOL res = rFormatter.PutEntry(sFormatstring,
--- sc/source/filter/excel/xestyle.cxx	21 Feb 2005 13:30:50 -0000	1.18
+++ sc/source/filter/excel/xestyle.cxx	9 Mar 2005 21:14:21 -0000	1.11.130.5
@@ -1330,7 +1330,7 @@ void XclExpNumFmtBuffer::WriteFormatReco
             {
                 xub_StrLen nCheckPos;
                 short nType = NUMBERFORMAT_DEFINED;
-                ULONG nKey;
+                sal_uInt32 nKey;
                 String aTemp( pEntry->GetFormatstring() );
                 mxFormatter->PutandConvertEntry( aTemp, nCheckPos, nType, nKey, eLang, LANGUAGE_ENGLISH_US );
                 DBG_ASSERT( nCheckPos == 0, "XclExpNumFmtBuffer::WriteFormatRecord - format code not convertible" );
--- sc/source/filter/excel/xistyle.cxx	21 Feb 2005 13:34:00 -0000	1.21
+++ sc/source/filter/excel/xistyle.cxx	9 Mar 2005 21:17:47 -0000	1.20.32.2
@@ -570,7 +570,7 @@ void XclImpNumFmtBuffer::CreateScFormats
         // insert/convert the Excel number format
         xub_StrLen nCheckPos;
         short nType = NUMBERFORMAT_DEFINED;
-        ULONG nKey;
+        sal_uInt32 nKey;
         if( rNumFmt.maFormat.Len() )
         {
             String aFormat( rNumFmt.maFormat );
--- sc/source/core/tool/interpr4.cxx	16 Feb 2005 18:06:55 -0000	1.34
+++ sc/source/core/tool/interpr4.cxx	9 Mar 2005 20:57:03 -0000	1.32.68.2
@@ -289,7 +289,7 @@
 					((ScStringCell*)pCell)->GetString( aStr );
 				else
 					((ScEditCell*)pCell)->GetString( aStr );
-				ULONG nFIndex = 0;					// damit default Land/Spr.
+				sal_uInt32 nFIndex = 0;					// damit default Land/Spr.
 				if ( !pFormatter->IsNumberFormat( aStr, nFIndex, fValue ) )
 				{
 					SetError(errNoValue);
--- sc/inc/document.hxx	15 Nov 2004 16:32:04 -0000	1.85
+++ sc/inc/document.hxx	21 Jan 2005 16:33:22 -0000	1.85.68.1
@@ -195,10 +195,10 @@ namespace com { namespace sun { namespac
 
 
 #ifdef _ZFORLIST_DECLARE_TABLE
-class SvULONGTable;
+class Svsal_uInt32Table;
 #else
 class Table;
-typedef Table SvULONGTable;
+typedef Table Svsal_uInt32Table;
 #endif
 
 
@@ -287,7 +287,7 @@ private:
 	XColorTable*		pColorTable;
 	ScConditionalFormatList* pCondFormList;				// bedingte Formate
 	ScValidationDataList* pValidationList;				// Gueltigkeit
-	SvULONGTable*		pFormatExchangeList;			// zum Umsetzen von Zahlenformaten
+	Svsal_uInt32Table*	pFormatExchangeList;			// zum Umsetzen von Zahlenformaten
 	ScTable*			pTab[MAXTABCOUNT];
 	ScRangeName*		pRangeName;
 	ScDBCollection*		pDBCollection;
@@ -523,7 +523,7 @@ SC_DLLPUBLIC	ScDBCollection*	GetDBCollec
 	SC_DLLPUBLIC BOOL			GetName( SCTAB nTab, String& rName ) const;
 	BOOL			GetTable( const String& rName, SCTAB& rTab ) const;
 	inline SCTAB	GetTableCount() const { return nMaxTableNumber; }
-	SvULONGTable*	GetFormatExchangeList() const { return pFormatExchangeList; }
+	Svsal_uInt32Table* GetFormatExchangeList() const { return pFormatExchangeList; }
 
 	void			SetDocProtection( BOOL bProtect, const com::sun::star::uno::Sequence <sal_Int8>& aPass );
 	void			SetTabProtection( SCTAB nTab, BOOL bProtect, const com::sun::star::uno::Sequence <sal_Int8>& aPass );
@@ -734,8 +734,8 @@ SC_DLLPUBLIC	ScDBCollection*	GetDBCollec
 	void			GetValue( SCCOL nCol, SCROW nRow, SCTAB nTab, double& rValue );
 	double			RoundValueAsShown( double fVal, ULONG nFormat );
 	void			GetNumberFormat( SCCOL nCol, SCROW nRow, SCTAB nTab,
-									 ULONG& rFormat );
-	ULONG			GetNumberFormat( const ScAddress& ) const;
+									 sal_uInt32& rFormat );
+	sal_uInt32		GetNumberFormat( const ScAddress& ) const;
 					/// if no number format attribute is set the calculated
 					/// number format of the formula cell is returned
 	void			GetNumberFormatInfo( short& nType, ULONG& nIndex,
@@ -1125,7 +1125,7 @@ SC_DLLPUBLIC	ScDBCollection*	GetDBCollec
 									BOOL bPutToPool = FALSE );
 	void			SetPattern( SCCOL nCol, SCROW nRow, SCTAB nTab, const ScPatternAttr& rAttr,
 									BOOL bPutToPool = FALSE );
-	void            DeleteNumberFormat( const ULONG* pDelKeys, ULONG nCount );
+	void            DeleteNumberFormat( const sal_uInt32* pDelKeys, sal_uInt32 nCount );
 
 	void			AutoFormat( SCCOL nStartCol, SCROW nStartRow, SCCOL nEndCol, SCROW nEndRow,
 									USHORT nFormatNo, const ScMarkData& rMark );
--- sc/source/core/data/column3.cxx	11 Jan 2005 12:38:02 -0000	1.16
+++ sc/source/core/data/column3.cxx	21 Jan 2005 16:33:44 -0000	1.16.6.1
@@ -1257,7 +1257,7 @@ BOOL ScColumn::SetString( SCROW nRow, SC
 		if (rString.Len() > 0)
 		{
 			double nVal;
-			ULONG nIndex, nOldIndex;
+			sal_uInt32 nIndex, nOldIndex;
 			sal_Unicode cFirstChar;
 			SvNumberFormatter* pFormatter = pDocument->GetFormatTable();
 			SfxObjectShell* pDocSh = pDocument->GetDocumentShell();
--- sc/source/core/data/dociter.cxx	15 Nov 2004 16:33:34 -0000	1.12
+++ sc/source/core/data/dociter.cxx	21 Jan 2005 16:34:00 -0000	1.12.68.1
@@ -524,7 +524,7 @@ ScQueryValueIterator::ScQueryValueIterat
 	for (i=0; (i<nCount) && (aParam.GetEntry(i).bDoQuery); i++)
 	{
 		ScQueryEntry& rEntry = aParam.GetEntry(i);
-		ULONG nIndex = 0;
+		sal_uInt32 nIndex = 0;
 		rEntry.bQueryByString =
 				!(pDoc->GetFormatTable()->IsNumberFormat(*rEntry.pStr, nIndex, rEntry.nVal));
 	}
@@ -806,7 +806,7 @@ ScQueryCellIterator::ScQueryCellIterator
 		for (i=0; (i<MAXQUERY) && (aParam.GetEntry(i).bDoQuery); i++)
 		{
 			ScQueryEntry& rEntry = aParam.GetEntry(i);
-			ULONG nIndex = 0;
+			sal_uInt32 nIndex = 0;
 			rEntry.bQueryByString =
 					 !(pDoc->GetFormatTable()->IsNumberFormat(*rEntry.pStr,
 															  nIndex, rEntry.nVal));
--- sc/source/core/data/documen2.cxx	16 Feb 2005 18:05:58 -0000	1.48
+++ sc/source/core/data/documen2.cxx	9 Mar 2005 20:54:26 -0000	1.47.68.2
@@ -1367,7 +1367,7 @@ void ScDocument::SetLostData()
 	bLostData = TRUE;
 }
 
-void ScDocument::DeleteNumberFormat( const ULONG* pDelKeys, ULONG nCount )
+void ScDocument::DeleteNumberFormat( const sal_uInt32* pDelKeys, sal_uInt32 nCount )
 {
 /*
 	for (ULONG i = 0; i < nCount; i++)
@@ -1737,7 +1737,7 @@ ULONG ScDocument::TransferTab( ScDocumen
 		SvNumberFormatter* pOtherFormatter = pSrcDoc->xPoolHelper->GetFormTable();
 		if (pOtherFormatter && pOtherFormatter != pThisFormatter)
 		{
-			SvULONGTable* pExchangeList =
+			Svsal_uInt32Table* pExchangeList =
 					 pThisFormatter->MergeFormatter(*(pOtherFormatter));
 			if (pExchangeList->Count() > 0)
 				pFormatExchangeList = pExchangeList;
--- sc/source/core/data/documen8.cxx	23 Dec 2004 10:43:45 -0000	1.38
+++ sc/source/core/data/documen8.cxx	21 Jan 2005 16:35:03 -0000	1.38.32.1
@@ -357,7 +357,7 @@ void ScDocument::CopyStdStylesFrom( ScDo
     SvNumberFormatter* pOtherFormatter = pSrcDoc->xPoolHelper->GetFormTable();
     if (pOtherFormatter && pOtherFormatter != pThisFormatter)
     {
-        SvULONGTable* pExchangeList =
+        Svsal_uInt32Table* pExchangeList =
                 pThisFormatter->MergeFormatter(*(pOtherFormatter));
         if (pExchangeList->Count() > 0)
             pFormatExchangeList = pExchangeList;
--- sc/source/core/data/document.cxx	16 Feb 2005 18:06:13 -0000	1.64
+++ sc/source/core/data/document.cxx	9 Mar 2005 20:55:27 -0000	1.63.68.2
@@ -1641,7 +1641,7 @@ void ScDocument::CopyFromClip( const ScR
 			SvNumberFormatter* pOtherFormatter = pClipDoc->xPoolHelper->GetFormTable();
 			if (pOtherFormatter && pOtherFormatter != pThisFormatter)
 			{
-				SvULONGTable* pExchangeList =
+				Svsal_uInt32Table* pExchangeList =
 						 pThisFormatter->MergeFormatter(*(pOtherFormatter));
 				if (pExchangeList->Count() > 0)
 					pFormatExchangeList = pExchangeList;
@@ -2131,7 +2131,7 @@ double ScDocument::GetValue( const ScAdd
 
 
 void ScDocument::GetNumberFormat( SCCOL nCol, SCROW nRow, SCTAB nTab,
-								  ULONG& rFormat )
+								  sal_uInt32& rFormat )
 {
 	if (VALIDTAB(nTab))
 		if (pTab[nTab])
@@ -2143,7 +2143,7 @@ void ScDocument::GetNumberFormat( SCCOL 
 }
 
 
-ULONG ScDocument::GetNumberFormat( const ScAddress& rPos ) const
+sal_uInt32 ScDocument::GetNumberFormat( const ScAddress& rPos ) const
 {
 	SCTAB nTab = rPos.Tab();
 	if ( pTab[nTab] )
--- sc/source/core/data/dpdimsave.cxx	3 Aug 2004 11:30:39 -0000	1.2
+++ sc/source/core/data/dpdimsave.cxx	21 Jan 2005 16:35:19 -0000	1.2.100.1
@@ -156,7 +156,7 @@ void ScDPSaveGroupItem::AddToData( ScDPG
 
     for ( std::vector<String>::const_iterator aIter(aElements.begin()); aIter != aElements.end(); aIter++ )
     {
-        ULONG nFormat = 0;      //! ...
+        sal_uInt32 nFormat = 0;      //! ...
         double fValue;
         if ( pFormatter->IsNumberFormat( *aIter, nFormat, fValue ) )
             aData = ScDPItemData( *aIter, fValue, TRUE );
--- sc/source/core/data/dpshttab.cxx	4 Jun 2004 10:24:58 -0000	1.5
+++ sc/source/core/data/dpshttab.cxx	21 Jan 2005 16:35:39 -0000	1.5.42.1
@@ -130,7 +130,7 @@ ScSheetDPData::ScSheetDPData( ScDocument
             }
             else
             {
-		        ULONG nIndex = 0;
+		        sal_uInt32 nIndex = 0;
 		        rEntry.bQueryByString =
 					        !(pD->GetFormatTable()->
 						        IsNumberFormat(*rEntry.pStr, nIndex, rEntry.nVal));
--- sc/source/core/data/table3.cxx	15 Nov 2004 16:34:51 -0000	1.20
+++ sc/source/core/data/table3.cxx	21 Jan 2005 16:36:18 -0000	1.16.42.4
@@ -1332,7 +1332,7 @@ SCSIZE ScTable::Query(ScQueryParam& rPar
 
 		if ( rEntry.bQueryByString )
 		{
-			ULONG nIndex = 0;
+			sal_uInt32 nIndex = 0;
 			rEntry.bQueryByString = !(pDocument->GetFormatTable()->
 				IsNumberFormat( *rEntry.pStr, nIndex, rEntry.nVal ));
 		}
--- sc/source/core/data/validat.cxx	4 Jun 2004 10:29:49 -0000	1.12
+++ sc/source/core/data/validat.cxx	21 Jan 2005 16:36:37 -0000	1.12.42.1
@@ -560,7 +560,7 @@ BOOL ScValidationData::IsDataValid( cons
 
 	//	Test, was es denn ist - wie in ScColumn::SetString
 
-	ULONG nFormat = rPattern.GetNumberFormat( pFormatter );
+	sal_uInt32 nFormat = rPattern.GetNumberFormat( pFormatter );
 
 	double nVal;
 	BOOL bIsVal = pFormatter->IsNumberFormat( rTest, nFormat, nVal );
--- sc/source/core/tool/chartarr.cxx	20 Aug 2004 09:11:50 -0000	1.13
+++ sc/source/core/tool/chartarr.cxx	21 Jan 2005 16:37:52 -0000	1.13.76.1
@@ -758,7 +758,7 @@ SchMemChart* ScChartArray::CreateMemChar
 							nVal = ((ScValueCell*)pCell)->GetValue();
 							if ( bCalcAsShown && nVal != 0.0 )
 							{
-								ULONG nFormat;
+								sal_uInt32 nFormat;
 								pDocument->GetNumberFormat( pCols[nCol],
 									pRows[nRow], nTab1, nFormat );
 								nVal = pDocument->RoundValueAsShown( nVal, nFormat );
--- sc/source/core/tool/ddelink.cxx	4 Oct 2004 20:05:51 -0000	1.11
+++ sc/source/core/tool/ddelink.cxx	21 Jan 2005 16:37:29 -0000	1.11.48.1
@@ -239,7 +239,7 @@ void __EXPORT ScDdeLink::DataChanged( co
 			for (SCSIZE nC=0; nC<nCols; nC++)
 			{
 				aEntry = aLine.GetToken( (xub_StrLen) nC, '\t' );
-				ULONG nIndex = nStdFormat;
+				sal_uInt32 nIndex = nStdFormat;
 				double fVal;
 				if ( nMode != SC_DDE_TEXT && pFormatter->IsNumberFormat( aEntry, nIndex, fVal ) )
 					pResult->PutDouble( fVal, nC, nR );
--- sc/source/core/tool/interpr1.cxx	28 Jan 2005 17:20:19 -0000	1.33
+++ sc/source/core/tool/interpr1.cxx	9 Mar 2005 20:56:34 -0000	1.32.68.2
@@ -2355,7 +2355,7 @@ void ScInterpreter::ScT()
 void ScInterpreter::ScValue()
 {
 	String aInputString = GetString();
-	ULONG nFIndex = 0;					// damit default Land/Spr.
+	sal_uInt32 nFIndex = 0;					// damit default Land/Spr.
 	double fVal;
 	if (pFormatter->IsNumberFormat(aInputString, nFIndex, fVal))
 		PushDouble(fVal);
@@ -2665,7 +2665,7 @@ double ScInterpreter::IterateParameters(
 				if( eFunc == ifCOUNT )
 				{
 					String aStr( PopString() );
-					ULONG nFIndex = 0;					// damit default Land/Spr.
+					sal_uInt32 nFIndex = 0;					// damit default Land/Spr.
 					if ( bTextAsZero || pFormatter->IsNumberFormat(aStr, nFIndex, fVal))
 						nCount++;
 				}
@@ -3710,7 +3710,7 @@ void ScInterpreter::ScCountIf()
 			else if( rString.Len() )
 			{
 				rParam.FillInExcelSyntax(rString, 0);
-				ULONG nIndex = 0;
+				sal_uInt32 nIndex = 0;
 				rEntry.bQueryByString =
 					!(pFormatter->IsNumberFormat(
 						*rEntry.pStr, nIndex, rEntry.nVal));
@@ -3892,7 +3892,7 @@ void ScInterpreter::ScSumIf()
 			else
 			{
 				rParam.FillInExcelSyntax(rString, 0);
-				ULONG nIndex = 0;
+				sal_uInt32 nIndex = 0;
 				rEntry.bQueryByString =
 					!(pFormatter->IsNumberFormat(
 						*rEntry.pStr, nIndex, rEntry.nVal));
@@ -5027,7 +5027,7 @@ BOOL ScInterpreter::GetDBParams(SCTAB& r
 						ScQueryEntry& rEntry = rParam.GetEntry(i);
 						if ( rEntry.bDoQuery )
 						{
-							ULONG nIndex = 0;
+							sal_uInt32 nIndex = 0;
 							rEntry.bQueryByString = !pFormatter->IsNumberFormat(
 								*rEntry.pStr, nIndex, rEntry.nVal );
 							if ( rEntry.bQueryByString && !rParam.bRegExp )
--- sc/source/core/tool/interpr2.cxx	15 Nov 2004 16:35:29 -0000	1.23
+++ sc/source/core/tool/interpr2.cxx	21 Jan 2005 16:38:39 -0000	1.23.68.1
@@ -197,7 +197,7 @@ void ScInterpreter::ScGetHour()
 void ScInterpreter::ScGetDateValue()
 {
 	String aInputString = GetString();
-	ULONG nFIndex = 0;					// damit default Land/Spr.
+	sal_uInt32 nFIndex = 0;					// damit default Land/Spr.
 	double fVal;
 	if (pFormatter->IsNumberFormat(aInputString, nFIndex, fVal))
 	{
@@ -414,7 +414,7 @@ void ScInterpreter::ScGetDiffDate360()
 void ScInterpreter::ScGetTimeValue()
 {
 	String aInputString = GetString();
-	ULONG nFIndex = 0;					// damit default Land/Spr.
+	sal_uInt32 nFIndex = 0;					// damit default Land/Spr.
 	double fVal;
 	if (pFormatter->IsNumberFormat(aInputString, nFIndex, fVal))
 	{
--- sc/source/core/tool/interpr4.cxx	16 Feb 2005 18:06:55 -0000	1.34
+++ sc/source/core/tool/interpr4.cxx	9 Mar 2005 20:57:03 -0000	1.32.68.2
@@ -1459,7 +1459,7 @@ double ScInterpreter::GetDouble()
 		case svString:
 		{
 			String aStr(PopString());
-			ULONG nFIndex = 0;					// damit default Land/Spr.
+			sal_uInt32 nFIndex = 0;					// damit default Land/Spr.
 			if(!pFormatter->IsNumberFormat( aStr, nFIndex, nVal ) )
 			{
 				SetError(errIllegalArgument);
--- sc/source/filter/dif/difimp.cxx	21 Feb 2005 13:20:55 -0000	1.8
+++ sc/source/filter/dif/difimp.cxx	9 Mar 2005 20:57:12 -0000	1.7.42.2
@@ -520,7 +520,7 @@ DATASET	DifParser::GetNextDataset( void 
 				{	// ...und zur Strafe mit'm Numberformatter...
 					DBG_ASSERT( pNumFormatter, "-DifParser::GetNextDataset(): No Formatter, more fun!" );
 					String			aTestVal( pAkt );
-					ULONG			nFormat = 0;
+					sal_uInt32		nFormat = 0;
 					double			fTmpVal;
 					if( pNumFormatter->IsNumberFormat( aTestVal, nFormat, fTmpVal ) )
 					{
--- sc/source/filter/excel/excrecds.cxx	21 Feb 2005 13:25:15 -0000	1.78
+++ sc/source/filter/excel/excrecds.cxx	9 Mar 2005 21:17:30 -0000	1.76.82.2
@@ -678,7 +678,7 @@ BOOL XclExpAutofilter::AddEntry( const S
 	else
 	{
 		double	fVal	= 0.0;
-		ULONG	nIndex	= 0;
+		sal_uInt32	nIndex	= 0;
         BOOL    bIsNum  = bLen ? GetFormatter().IsNumberFormat( sText, nIndex, fVal ) : TRUE;
 		String*	pText	= bIsNum ? NULL : &sText;
 
--- sc/source/ui/app/inputhdl.cxx	16 Feb 2005 18:09:54 -0000	1.59
+++ sc/source/ui/app/inputhdl.cxx	9 Mar 2005 20:36:36 -0000	1.58.138.2
@@ -2096,7 +2096,7 @@ void ScInputHandler::EnterHandler( BYTE 
 			ScDocument* pDoc = pActiveViewSh->GetViewData()->GetDocument();
 			SvNumberFormatter* pFormatter = pDoc->GetFormatTable();
 			// without conditional format, as in ScColumn::SetString
-			ULONG nFormat = pLastPattern->GetNumberFormat( pFormatter );
+			sal_uInt32 nFormat = pLastPattern->GetNumberFormat( pFormatter );
 			double nVal;
 			if ( pFormatter->IsNumberFormat( aString, nFormat, nVal ) )
 			{
--- sc/source/ui/docshell/docfunc.cxx	8 Sep 2004 13:52:34 -0000	1.52
+++ sc/source/ui/docshell/docfunc.cxx	21 Jan 2005 16:43:14 -0000	1.52.56.1
@@ -1008,7 +1008,7 @@ ScBaseCell* ScDocFunc::InterpretEnglishS
 	else		// (nur) auf englisches Zahlformat testen
 	{
 		SvNumberFormatter* pFormatter = pDoc->GetFormatTable();
-		ULONG nEnglish = pFormatter->GetStandardIndex(LANGUAGE_ENGLISH_US);
+		sal_uInt32 nEnglish = pFormatter->GetStandardIndex(LANGUAGE_ENGLISH_US);
 		double fVal;
 		if ( pFormatter->IsNumberFormat( rText, nEnglish, fVal ) )
 			pNewCell = new ScValueCell( fVal );
--- sc/source/ui/docshell/docsh.cxx	21 Feb 2005 13:50:23 -0000	1.78
+++ sc/source/ui/docshell/docsh.cxx	9 Mar 2005 20:52:07 -0000	1.77.6.2
@@ -1530,7 +1530,7 @@ void ScDocShell::AsciiSave( SvStream& rS
                     }
                     else if ( ((ScFormulaCell*)pCell)->IsValue() )
                     {
-                        ULONG nFormat;
+                        sal_uInt32 nFormat;
                         aDocument.GetNumberFormat( nCol, nRow, nTab, nFormat );
                         if ( bFixedWidth )
                         {
@@ -1564,7 +1564,7 @@ void ScDocShell::AsciiSave( SvStream& rS
                 break;
             case CELLTYPE_VALUE :
                 {
-                    ULONG nFormat;
+                    sal_uInt32 nFormat;
                     aDocument.GetNumberFormat( nCol, nRow, nTab, nFormat );
                     if ( bFixedWidth )
                     {
--- sc/source/ui/docshell/docsh8.cxx	4 Jun 2004 11:24:52 -0000	1.17
+++ sc/source/ui/docshell/docsh8.cxx	21 Jan 2005 16:43:54 -0000	1.17.42.1
@@ -607,7 +607,7 @@ void lcl_GetColumnTypes( ScDocShell& rDo
 				nDbType = sdbc::DataType::VARCHAR;
 			else
 			{
-				ULONG nFormat;
+				sal_uInt32 nFormat;
 				pDoc->GetNumberFormat( nCol, nFirstDataRow, nTab, nFormat );
 				if ( pCell && pCell->GetCellType() == CELLTYPE_FORMULA
 				  && ((nFormat % SV_COUNTRY_LANGUAGE_OFFSET) == 0) )
@@ -994,7 +994,7 @@ ULONG ScDocShell::DBaseExport( const Str
 								}
 								else
 								{
-									ULONG nFormat;
+									sal_uInt32 nFormat;
 									Color* pColor;
 									aDocument.GetNumberFormat( nDocCol, nDocRow, nTab, nFormat );
 									ScCellFormat::GetString( pCell, nFormat, aString, &pColor, *pNumFmt );
--- sc/source/ui/docshell/impex.cxx	11 Jan 2005 13:20:36 -0000	1.31
+++ sc/source/ui/docshell/impex.cxx	21 Jan 2005 16:44:08 -0000	1.31.6.1
@@ -835,7 +835,7 @@ bool lcl_PutString( ScDocument* pDoc, SC
 		//!	SetString mit Extra-Flag ???
 
 		SvNumberFormatter* pFormatter = pDoc->GetFormatTable();
-		ULONG nEnglish = pFormatter->GetStandardIndex(LANGUAGE_ENGLISH_US);
+		sal_uInt32 nEnglish = pFormatter->GetStandardIndex(LANGUAGE_ENGLISH_US);
 		double fVal;
 		if ( pFormatter->IsNumberFormat( rStr, nEnglish, fVal ) )
 		{
--- sc/source/ui/miscdlgs/filldlg.cxx	23 Aug 2004 09:36:22 -0000	1.4
+++ sc/source/ui/miscdlgs/filldlg.cxx	21 Jan 2005 16:44:38 -0000	1.4.78.1
@@ -287,7 +287,7 @@ BOOL __EXPORT ScFillSeriesDlg::CheckStar
 
 BOOL __EXPORT ScFillSeriesDlg::CheckIncrementVal()
 {
-	ULONG  nKey = 0;
+	sal_uInt32 nKey = 0;
 	String aStr( aEdIncrement.GetText() );
 
 	return rDoc.GetFormatTable()->IsNumberFormat( aStr, nKey, fIncrement );
--- sc/source/ui/miscdlgs/solvrdlg.cxx	4 Feb 2004 14:28:04 -0000	1.7
+++ sc/source/ui/miscdlgs/solvrdlg.cxx	21 Jan 2005 16:44:46 -0000	1.7.146.1
@@ -248,7 +248,7 @@ BOOL ScSolverDlg::IsRefInputMode() const
 
 BOOL __EXPORT ScSolverDlg::CheckTargetValue( String& rStrVal )
 {
-	ULONG  n1 = 0;
+	sal_uInt32 n1 = 0;
 	double n2;
 
 	return pDoc->GetFormatTable()->IsNumberFormat( rStrVal, n1, n2 );
--- sc/source/ui/unoobj/cellsuno.cxx	23 Dec 2004 10:46:30 -0000	1.89
+++ sc/source/ui/unoobj/cellsuno.cxx	21 Jan 2005 16:45:11 -0000	1.89.30.1
@@ -1348,7 +1348,7 @@ String lcl_GetInputString( ScDocShell* p
                 // Since the English formatter was constructed with
                 // LANGUAGE_ENGLISH_US the "General" format has index key 0,
                 // we don't have to query.
-				ULONG nNumFmt = bEnglish ?
+				sal_uInt32 nNumFmt = bEnglish ?
 //						pFormatter->GetStandardIndex(LANGUAGE_ENGLISH_US) :
                         0 :
 						pDoc->GetNumberFormat( rPosition );
--- sc/source/ui/view/cellsh1.cxx	21 Feb 2005 13:53:16 -0000	1.37
+++ sc/source/ui/view/cellsh1.cxx	9 Mar 2005 20:43:15 -0000	1.36.48.2
@@ -683,7 +683,7 @@ void ScCellShell::ExecuteEdit( SfxReques
 				else // (pReqArgs == NULL) => Dialog hochziehen
 				{
 					//
-					ULONG nPrivFormat;
+					sal_uInt32 nPrivFormat;
 					CellType eCellType;
 					pDoc->GetNumberFormat( nStartCol, nStartRow, nStartTab, nPrivFormat );
 					pDoc->GetCellType( nStartCol, nStartRow, nStartTab,eCellType );
--- sc/source/ui/view/tabvwsh5.cxx	4 Jun 2004 12:07:53 -0000	1.7
+++ sc/source/ui/view/tabvwsh5.cxx	21 Jan 2005 16:46:41 -0000	1.7.42.1
@@ -443,11 +443,11 @@ void ScTabViewShell::UpdateNumberFormatt
 						ScDocument* 			 pDoc,
 						const SvxNumberInfoItem& rInfoItem )
 {
-	const ULONG nDelCount = rInfoItem.GetDelCount();
+	const sal_uInt32 nDelCount = rInfoItem.GetDelCount();
 
 	if ( nDelCount > 0 )
 	{
-		const ULONG* pDelArr = rInfoItem.GetDelArray();
+		const sal_uInt32* pDelArr = rInfoItem.GetDelArray();
 
 		for ( USHORT i=0; i<nDelCount; i++ )
 			rInfoItem.GetNumberFormatter()->DeleteEntry( pDelArr[i] );
--- sc/source/ui/view/tabvwsha.cxx	28 Jan 2005 17:23:33 -0000	1.15
+++ sc/source/ui/view/tabvwsha.cxx	9 Mar 2005 20:43:58 -0000	1.14.48.2
@@ -131,7 +131,7 @@ BOOL ScTabViewShell::GetFunction( String
 		aStr += '=';
 
 		//	Anzahl im Standardformat, die anderen nach Cursorposition
-		ULONG nNumFmt = 0;
+		sal_uInt32 nNumFmt = 0;
 		SvNumberFormatter* pFormatter = pDoc->GetFormatTable();
 		if ( eFunc != SUBTOTAL_FUNC_CNT && eFunc != SUBTOTAL_FUNC_CNT2 )
 		{
@@ -560,7 +560,7 @@ void ScTabViewShell::UpdateInputHandler(
 					//!	Auch bei Zahlformat "Text"? -> dann beim Editieren wegnehmen
 
 					SvNumberFormatter* pFormatter = pDoc->GetFormatTable();
-					ULONG nNumFmt;
+					sal_uInt32 nNumFmt;
 					pDoc->GetNumberFormat( nPosX, nPosY, nTab, nNumFmt );
 					double fDummy;
 					if ( pFormatter->IsNumberFormat(aString, nNumFmt, fDummy) )
--- sc/source/ui/view/viewfunc.cxx	9 Sep 2004 09:31:12 -0000	1.27
+++ sc/source/ui/view/viewfunc.cxx	21 Jan 2005 16:46:48 -0000	1.27.56.1
@@ -425,7 +425,7 @@ void ScViewFunc::EnterData( SCCOL nCol, 
 
 		BOOL bNumFmtChanged = FALSE;
 		double fVal;
-		ULONG nIndex = 0;
+		sal_uInt32 nIndex = 0;
 
 		// einzelnes '=' ist String (wird fuer Spezialfilter so gebraucht)
 		if ( rString.Len() > 1 && 
@@ -999,9 +999,9 @@ void ScViewFunc::ApplyAttributes( const 
 
 	if ( pDialogSet->GetItemState( ATTR_VALUE_FORMAT ) == SFX_ITEM_SET )
 	{	// #82521# don't reset to default SYSTEM GENERAL if not intended
-		ULONG nOldFormat =
+		sal_uInt32 nOldFormat =
 			((const SfxUInt32Item&)pOldSet->Get( ATTR_VALUE_FORMAT )).GetValue();
-		ULONG nNewFormat =
+		sal_uInt32 nNewFormat =
 			((const SfxUInt32Item&)pDialogSet->Get( ATTR_VALUE_FORMAT )).GetValue();
 		if ( nNewFormat != nOldFormat )
 		{
@@ -1019,7 +1019,7 @@ void ScViewFunc::ApplyAttributes( const 
 					SvxLanguageItem( eNewLang, ATTR_LANGUAGE_FORMAT ) );
 
 				//	#40606# nur die Sprache geaendert -> Zahlformat-Attribut nicht anfassen
-				ULONG nNewMod = nNewFormat % SV_COUNTRY_LANGUAGE_OFFSET;
+				sal_uInt32 nNewMod = nNewFormat % SV_COUNTRY_LANGUAGE_OFFSET;
 				if ( nNewMod == ( nOldFormat % SV_COUNTRY_LANGUAGE_OFFSET ) &&
 				 	nNewMod <= SV_MAX_ANZ_STANDARD_FORMATE )
 					aNewAttrs.GetItemSet().ClearItem( ATTR_VALUE_FORMAT );
@@ -2393,7 +2393,7 @@ void ScViewFunc::SetNumberFormat( short 
 		return;
 	}
 
-	ULONG				nNumberFormat = 0;
+	sal_uInt32			nNumberFormat = 0;
 	ScViewData*			pViewData = GetViewData();
 	ScDocument*			pDoc = pViewData->GetDocument();
 	SvNumberFormatter*	pNumberFormatter = pDoc->GetFormatTable();
@@ -2402,7 +2402,7 @@ void ScViewFunc::SetNumberFormat( short 
 
 	//	#67936# always take language from cursor position, even if there is a selection
 
-	ULONG nCurrentNumberFormat;
+	sal_uInt32 nCurrentNumberFormat;
 	pDoc->GetNumberFormat( pViewData->GetCurX(),
 						   pViewData->GetCurY(),
 						   pViewData->GetTabNo(),
@@ -2435,7 +2435,7 @@ void ScViewFunc::SetNumFmtByStr( const S
 
 	//	Sprache immer von Cursorposition
 
-	ULONG nCurrentNumberFormat;
+	sal_uInt32 nCurrentNumberFormat;
 	pDoc->GetNumberFormat( pViewData->GetCurX(), pViewData->GetCurY(),
 						   pViewData->GetTabNo(), nCurrentNumberFormat );
 	const SvNumberformat* pEntry = pFormatter->GetEntry( nCurrentNumberFormat );
@@ -2444,7 +2444,7 @@ void ScViewFunc::SetNumFmtByStr( const S
 	//	Index fuer String bestimmen
 
 	BOOL bOk = TRUE;
-	ULONG nNumberFormat = pFormatter->GetEntryKey( rCode, eLanguage );
+	sal_uInt32 nNumberFormat = pFormatter->GetEntryKey( rCode, eLanguage );
 	if ( nNumberFormat == NUMBERFORMAT_ENTRY_NOT_FOUND )
 	{
 		//	neu eintragen
@@ -2484,7 +2484,7 @@ void ScViewFunc::ChangeNumFmtDecimals( B
 	SCROW nRow = GetViewData()->GetCurY();
 	SCTAB nTab = GetViewData()->GetTabNo();
 
-	ULONG nOldFormat;
+	sal_uInt32 nOldFormat;
 	pDoc->GetNumberFormat( nCol, nRow, nTab, nOldFormat );
 	const SvNumberformat* pOldEntry = pFormatter->GetEntry( nOldFormat );
 	if (!pOldEntry)
@@ -2495,7 +2495,7 @@ void ScViewFunc::ChangeNumFmtDecimals( B
 
 	//	was haben wir denn da?
 
-	ULONG nNewFormat = nOldFormat;
+	sal_uInt32 nNewFormat = nOldFormat;
 	BOOL bError = FALSE;
 
 	LanguageType eLanguage = pOldEntry->GetLanguage();
