--- svtools/inc/cacheoptions.hxx	8 May 2001 09:05:17 -0000	1.2
+++ svtools/inc/cacheoptions.hxx	21 Jan 2005 17:16:00 -0000	1.2.638.2
@@ -159,14 +159,14 @@ class SvtCacheOptions
 
 		sal_Int32	GetWriterOLE_Objects() const;
 		sal_Int32	GetDrawingEngineOLE_Objects() const;
-		sal_Int32	GetGraphicManagerTotalCacheSize() const; 
-		sal_Int32	GetGraphicManagerObjectCacheSize() const;
+		sal_Size	GetGraphicManagerTotalCacheSize() const; 
+		sal_Size	GetGraphicManagerObjectCacheSize() const;
 		sal_Int32	GetGraphicManagerObjectReleaseTime() const;
 
 		void		SetWriterOLE_Objects( sal_Int32 nObjects );
 		void		SetDrawingEngineOLE_Objects( sal_Int32 nObjects );
-		void		SetGraphicManagerTotalCacheSize( sal_Int32 nTotalCacheSize );
-		void		SetGraphicManagerObjectCacheSize( sal_Int32 nObjectCacheSize );
+		void		SetGraphicManagerTotalCacheSize( sal_Size nTotalCacheSize );
+		void		SetGraphicManagerObjectCacheSize( sal_Size nObjectCacheSize );
 		void        SetGraphicManagerObjectReleaseTime( sal_Int32 nReleaseTimeSeconds );
 
 	//-------------------------------------------------------------------------------------------------------------
--- svtools/source/config/cacheoptions.cxx	16 Jun 2004 10:06:14 -0000	1.5
+++ svtools/source/config/cacheoptions.cxx	21 Jan 2005 17:16:13 -0000	1.5.16.2
@@ -141,14 +141,14 @@ public:
 
 	sal_Int32		GetWriterOLE_Objects() const;
 	sal_Int32		GetDrawingEngineOLE_Objects() const;
-	sal_Int32		GetGraphicManagerTotalCacheSize() const; 
-	sal_Int32		GetGraphicManagerObjectCacheSize() const;
+	sal_Size		GetGraphicManagerTotalCacheSize() const; 
+	sal_Size		GetGraphicManagerObjectCacheSize() const;
 	sal_Int32		GetGraphicManagerObjectReleaseTime() const;
 					
 	void			SetWriterOLE_Objects( sal_Int32 nObjects );
 	void			SetDrawingEngineOLE_Objects( sal_Int32 nObjects );
-	void			SetGraphicManagerTotalCacheSize( sal_Int32 nTotalCacheSize );
-	void			SetGraphicManagerObjectCacheSize( sal_Int32 nObjectCacheSize );
+	void			SetGraphicManagerTotalCacheSize( sal_Size nTotalCacheSize );
+	void			SetGraphicManagerObjectCacheSize( sal_Size nObjectCacheSize );
 	void            SetGraphicManagerObjectReleaseTime( sal_Int32 nReleaseTimeSeconds );
 
 //-------------------------------------------------------------------------------------------------------------
@@ -167,8 +167,8 @@ private:
 
 		sal_Int32	mnWriterOLE;
 		sal_Int32	mnDrawingOLE;
-		sal_Int32	mnGrfMgrTotalSize;
-		sal_Int32	mnGrfMgrObjectSize;
+		sal_Size	mnGrfMgrTotalSize;
+		sal_Size	mnGrfMgrObjectSize;
         sal_Int32   mnGrfMgrObjectRelease;
 };
 
@@ -306,7 +306,7 @@ sal_Int32 SvtCacheOptions_Impl::GetDrawi
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-sal_Int32 SvtCacheOptions_Impl::GetGraphicManagerTotalCacheSize() const
+sal_Size SvtCacheOptions_Impl::GetGraphicManagerTotalCacheSize() const
 {
 	return mnGrfMgrTotalSize;
 }
@@ -314,7 +314,7 @@ sal_Int32 SvtCacheOptions_Impl::GetGraph
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-sal_Int32 SvtCacheOptions_Impl::GetGraphicManagerObjectCacheSize() const
+sal_Size SvtCacheOptions_Impl::GetGraphicManagerObjectCacheSize() const
 {
 	return mnGrfMgrObjectSize;
 }
@@ -348,7 +348,7 @@ void SvtCacheOptions_Impl::SetDrawingEng
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-void SvtCacheOptions_Impl::SetGraphicManagerTotalCacheSize( sal_Int32 nGrfMgrTotalSize )
+void SvtCacheOptions_Impl::SetGraphicManagerTotalCacheSize( sal_Size nGrfMgrTotalSize )
 {
 	mnGrfMgrTotalSize = nGrfMgrTotalSize;
 	SetModified();
@@ -357,7 +357,7 @@ void SvtCacheOptions_Impl::SetGraphicMan
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-void SvtCacheOptions_Impl::SetGraphicManagerObjectCacheSize( sal_Int32 nGrfMgrObjectSize )
+void SvtCacheOptions_Impl::SetGraphicManagerObjectCacheSize( sal_Size nGrfMgrObjectSize )
 {
 	mnGrfMgrObjectSize = nGrfMgrObjectSize;
 	SetModified();
@@ -455,7 +455,7 @@ sal_Int32 SvtCacheOptions::GetDrawingEng
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-sal_Int32 SvtCacheOptions::GetGraphicManagerTotalCacheSize() const
+sal_Size SvtCacheOptions::GetGraphicManagerTotalCacheSize() const
 {
     MutexGuard aGuard( GetOwnStaticMutex() );
 	return m_pDataContainer->GetGraphicManagerTotalCacheSize();
@@ -464,7 +464,7 @@ sal_Int32 SvtCacheOptions::GetGraphicMan
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-sal_Int32 SvtCacheOptions::GetGraphicManagerObjectCacheSize() const
+sal_Size SvtCacheOptions::GetGraphicManagerObjectCacheSize() const
 {
     MutexGuard aGuard( GetOwnStaticMutex() );
 	return m_pDataContainer->GetGraphicManagerObjectCacheSize();
@@ -500,7 +500,7 @@ void SvtCacheOptions::SetDrawingEngineOL
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-void SvtCacheOptions::SetGraphicManagerTotalCacheSize( sal_Int32 nGrfMgrTotalSize )
+void SvtCacheOptions::SetGraphicManagerTotalCacheSize( sal_Size nGrfMgrTotalSize )
 {
     MutexGuard aGuard( GetOwnStaticMutex() );
 	m_pDataContainer->SetGraphicManagerTotalCacheSize( nGrfMgrTotalSize );
@@ -509,7 +509,7 @@ void SvtCacheOptions::SetGraphicManagerT
 //*****************************************************************************************************************
 //	public method
 //*****************************************************************************************************************
-void SvtCacheOptions::SetGraphicManagerObjectCacheSize( sal_Int32 nGrfMgrObjectSize )
+void SvtCacheOptions::SetGraphicManagerObjectCacheSize( sal_Size nGrfMgrObjectSize )
 {
     MutexGuard aGuard( GetOwnStaticMutex() );
 	m_pDataContainer->SetGraphicManagerObjectCacheSize( nGrfMgrObjectSize );
--- svtools/source/control/taskmisc.cxx	16 Jun 2004 10:13:08 -0000	1.2
+++ svtools/source/control/taskmisc.cxx	21 Jul 2004 17:34:09 -0000	1.2.16.1
@@ -307,10 +307,10 @@ void WindowArrange::ImplCascade( const R
 	long		nRectHeight = nHeight;
 	long		nOff;
 	long		nCascadeWins;
-	long		nLeftBorder;
-	long		nTopBorder;
-	long		nRightBorder;
-	long		nBottomBorder;
+	sal_Int32	nLeftBorder;
+	sal_Int32	nTopBorder;
+	sal_Int32	nRightBorder;
+	sal_Int32	nBottomBorder;
 	long		nStartOverWidth;
 	long		nStartOverHeight;
 	long		nOverWidth = 0;
--- svtools/source/edit/svmedit.cxx	20 Sep 2004 14:10:21 -0000	1.28
+++ svtools/source/edit/svmedit.cxx	19 Oct 2004 16:45:46 -0000	1.26.64.2
@@ -1238,7 +1238,7 @@ Size MultiLineEdit::CalcMinimumSize() co
 {
 	Size aSz = pImpSvMEdit->CalcMinimumSize();
 
-	long nLeft, nTop, nRight, nBottom;
+	sal_Int32 nLeft, nTop, nRight, nBottom;
 	((Window*)this)->GetBorder( nLeft, nTop, nRight, nBottom );
 	aSz.Width() += nLeft+nRight;
 	aSz.Height() += nTop+nBottom;
@@ -1249,14 +1249,14 @@ Size MultiLineEdit::CalcMinimumSize() co
 Size MultiLineEdit::CalcAdjustedSize( const Size& rPrefSize ) const
 {
 	Size aSz = rPrefSize;
-	long nLeft, nTop, nRight, nBottom;
+	sal_Int32 nLeft, nTop, nRight, nBottom;
 	((Window*)this)->GetBorder( nLeft, nTop, nRight, nBottom );
 
 	// In der Hoehe auf ganze Zeilen justieren
 
-	long nHeight = aSz.Height() - nTop - nBottom;
-	long nLineHeight = pImpSvMEdit->CalcSize( 1, 1 ).Height();
-	long nLines = nHeight / nLineHeight;
+	sal_Int32 nHeight = aSz.Height() - nTop - nBottom;
+	sal_Int32 nLineHeight = pImpSvMEdit->CalcSize( 1, 1 ).Height();
+	sal_Int32 nLines = nHeight / nLineHeight;
 	if ( nLines < 1 )
 		nLines = 1;
 
@@ -1270,7 +1270,7 @@ Size MultiLineEdit::CalcSize( USHORT nCo
 {
 	Size aSz = pImpSvMEdit->CalcSize( nColumns, nLines );
 
-	long nLeft, nTop, nRight, nBottom;
+	sal_Int32 nLeft, nTop, nRight, nBottom;
 	((Window*)this)->GetBorder( nLeft, nTop, nRight, nBottom );
 	aSz.Width() += nLeft+nRight;
 	aSz.Height() += nTop+nBottom;
@@ -1389,13 +1389,13 @@ void MultiLineEdit::Draw( OutputDevice* 
 
 	XubString aText = GetText();
 	Size aTextSz( pDev->GetTextWidth( aText ), pDev->GetTextHeight() );
-	ULONG nLines = (ULONG) (aSize.Height() / aTextSz.Height());
+	sal_uInt32 nLines = sal_uInt32(aSize.Height() / aTextSz.Height());
 	if ( !nLines )
 		nLines = 1;
 	aTextSz.Height() = nLines*aTextSz.Height();
-	long nOnePixel = GetDrawPixel( pDev, 1 );
-	long nOffX = 3*nOnePixel;
-	long nOffY = 2*nOnePixel;
+	sal_Int32 nOnePixel = GetDrawPixel( pDev, 1 );
+	sal_Int32 nOffX = 3*nOnePixel;
+	sal_Int32 nOffY = 2*nOnePixel;
 
 	// Clipping?
 	if ( ( nOffY < 0  ) || ( (nOffY+aTextSz.Height()) > aSize.Height() ) || ( (nOffX+aTextSz.Width()) > aSize.Width() ) )
