--- vcl/inc/bitmap.hxx	13 Jan 2005 17:26:04 -0000	1.14
+++ vcl/inc/bitmap.hxx	10 Mar 2005 02:24:40 -0000	1.13.14.2
@@ -305,7 +305,7 @@ public:
     SAL_DLLPRIVATE static BOOL          ImplWriteDIBFileHeader( SvStream& rOStm, BitmapReadAccess& rAcc );
     SAL_DLLPRIVATE static BOOL          ImplWriteDIBPalette( SvStream& rOStm, BitmapReadAccess& rAcc );
     SAL_DLLPRIVATE static BOOL          ImplWriteDIBBits( SvStream& rOStm, BitmapReadAccess& rAcc,
-                                                             ULONG nCompression, ULONG& rImageSize );
+                                                             ULONG nCompression, sal_uInt32& rImageSize );
     SAL_DLLPRIVATE static void          ImplDecodeRLE( BYTE* pBuffer, DIBInfoHeader& rHeader,
                                            BitmapWriteAccess& rAcc, BOOL bRLE4 );
     SAL_DLLPRIVATE static BOOL          ImplWriteRLE( SvStream& rOStm, BitmapReadAccess& rAcc, BOOL bRLE4 );
--- vcl/inc/dockwin.hxx	13 Jan 2005 17:30:41 -0000	1.9
+++ vcl/inc/dockwin.hxx	10 Mar 2005 02:17:23 -0000	1.8.40.2
@@ -140,10 +140,10 @@ private:
 	long			mnTrackY;
 	long			mnTrackWidth;
 	long			mnTrackHeight;
-	long			mnDockLeft;
-	long			mnDockTop;
-	long			mnDockRight;
-	long			mnDockBottom;
+	sal_Int32		mnDockLeft;
+	sal_Int32		mnDockTop;
+	sal_Int32		mnDockRight;
+	sal_Int32		mnDockBottom;
 	WinBits 		mnFloatBits;
 	BOOL			mbDockCanceled:1,
 					mbFloatPrevented:1,
--- vcl/inc/imgcons.hxx	13 Jan 2005 17:36:46 -0000	1.3
+++ vcl/inc/imgcons.hxx	10 Mar 2005 02:09:00 -0000	1.2.144.2
@@ -110,7 +110,7 @@ public:					
 	BOOL				GetData( BitmapEx& rBmpEx ) const;
 	BOOL				GetData( Image& rImage ) const;
 	const Rectangle&	GetChangedRect() const { return maChangedRect; }
-	ULONG				GetStatus() const;
+	sal_uInt32			GetStatus() const;
 						
 	void				SetDataChangedLink( const Link& rLink ) { maChgLink = rLink; }
 	const Link&			GetDataChangedLink() const { return maChgLink; }
@@ -120,22 +120,22 @@ public:					
 
 public:
 
-	virtual void		Init( ULONG nWidth, ULONG nHeight );
+	virtual void		Init( sal_uInt32 nWidth, sal_uInt32 nHeight );
 
 	virtual void		SetColorModel( USHORT nBitCount,
-									   ULONG nPalEntries, const ULONG* pRGBAPal, 
-									   ULONG nRMask, ULONG nGMask, ULONG nBMask, ULONG nAMask  );
+									   sal_uInt32 nPalEntries, const sal_uInt32* pRGBAPal, 
+									   sal_uInt32 nRMask, sal_uInt32 nGMask, sal_uInt32 nBMask, sal_uInt32 nAMask  );
 						
-	virtual	void		SetPixelsByBytes( ULONG nConsX, ULONG nConsY, 
-										  ULONG nConsWidth, ULONG nConsHeight,
-										  const BYTE* pProducerData, ULONG nOffset, ULONG nScanSize );
-						
-	virtual	void		SetPixelsByLongs( ULONG nConsX, ULONG nConsY, 
-										  ULONG nConsWidth, ULONG nConsHeight,
-										  const ULONG* pProducerData, ULONG nOffset, ULONG nScanSize );
+	virtual	void		SetPixelsByBytes( sal_uInt32 nConsX, sal_uInt32 nConsY, 
+										  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+										  const BYTE* pProducerData, sal_uInt32 nOffset, sal_uInt32 nScanSize );
+						
+	virtual	void		SetPixelsByLongs( sal_uInt32 nConsX, sal_uInt32 nConsY, 
+										  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+										  const sal_uInt32* pProducerData, sal_uInt32 nOffset, sal_uInt32 nScanSize );
 						
-	virtual void		Completed( ULONG nStatus );
-//	virtual void		Completed( ULONG nStatus, ImageProducer& rProducer );
+	virtual void		Completed( sal_uInt32 nStatus );
+//	virtual void		Completed( sal_uInt32 nStatus, ImageProducer& rProducer );
 };
 
 #endif // _IMGCONS_HXX
--- vcl/source/control/ilstbox.cxx	26 Nov 2004 20:41:57 -0000	1.50
+++ vcl/source/control/ilstbox.cxx	18 Jan 2005 17:36:18 -0000	1.47.6.3
@@ -2662,7 +2662,7 @@ void ImplWin::ImplDraw( bool bLayout )
 	        // The listbox is painted over the entire control including the
 	        // border, but ImplWin does not contain the border => correction
 	        // needed.
-	        long nLeft, nTop, nRight, nBottom;
+	        sal_Int32 nLeft, nTop, nRight, nBottom;
 	        pWin->GetBorder( nLeft, nTop, nRight, nBottom );
 	        Point aPoint( -nLeft, -nTop );
 	        Region aCtrlRegion( Rectangle( aPoint - GetPosPixel(), pWin->GetSizePixel() ) );
--- vcl/source/gdi/bitmap2.cxx	17 Jun 2004 12:14:50 -0000	1.6
+++ vcl/source/gdi/bitmap2.cxx	10 Sep 2004 15:33:33 -0000	1.6.6.1
@@ -870,7 +870,7 @@ BOOL Bitmap::ImplWriteDIBPalette( SvStre
 // ------------------------------------------------------------------
 
 BOOL Bitmap::ImplWriteDIBBits( SvStream& rOStm, BitmapReadAccess& rAcc,
-							   ULONG nCompression, ULONG& rImageSize )
+							   ULONG nCompression, sal_uInt32& rImageSize )
 {
 	if( BITFIELDS == nCompression )
 	{
--- vcl/source/gdi/imgcons.cxx	6 Jan 2004 13:41:30 -0000	1.3
+++ vcl/source/gdi/imgcons.cxx	14 Sep 2004 18:06:33 -0000	1.3.144.1
@@ -160,7 +160,7 @@ ImageConsumer::~ImageConsumer()
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::Init( ULONG nWidth, ULONG nHeight )
+void ImageConsumer::Init( sal_uInt32 nWidth, sal_uInt32 nHeight )
 {
 	maSize = Size( nWidth, nHeight );
 	maBitmap = maMask = Bitmap();
@@ -171,8 +171,8 @@ void ImageConsumer::Init( ULONG nWidth, 
 // -----------------------------------------------------------------------------
 
 void ImageConsumer::SetColorModel( USHORT nBitCount,
-								   ULONG nPalEntries, const ULONG* pRGBAPal, 
-								   ULONG nRMask, ULONG nGMask, ULONG nBMask, ULONG nAMask )
+								   sal_uInt32 nPalEntries, const sal_uInt32* pRGBAPal, 
+								   sal_uInt32 nRMask, sal_uInt32 nGMask, sal_uInt32 nBMask, sal_uInt32 nAMask )
 {
 	DBG_ASSERT( maSize.Width() && maSize.Height(), "Missing call to ImageConsumer::Init(...)!" );
 
@@ -247,9 +247,9 @@ void ImageConsumer::SetColorModel( USHOR
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::SetPixelsByBytes( ULONG nConsX, ULONG nConsY,
-									  ULONG nConsWidth, ULONG nConsHeight,
-									  const BYTE* pData, ULONG nOffset, ULONG nScanSize )
+void ImageConsumer::SetPixelsByBytes( sal_uInt32 nConsX, sal_uInt32 nConsY,
+									  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+									  const BYTE* pData, sal_uInt32 nOffset, sal_uInt32 nScanSize )
 {
 	DBG_ASSERT( !!maBitmap && !!maMask, "Missing call to ImageConsumer::SetColorModel(...)!" );
 
@@ -384,9 +384,9 @@ void ImageConsumer::SetPixelsByBytes( UL
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::SetPixelsByLongs( ULONG nConsX, ULONG nConsY, 
-									  ULONG nConsWidth, ULONG nConsHeight,
-									  const ULONG* pData, ULONG nOffset, ULONG nScanSize )
+void ImageConsumer::SetPixelsByLongs( sal_uInt32 nConsX, sal_uInt32 nConsY, 
+									  sal_uInt32 nConsWidth, sal_uInt32 nConsHeight,
+									  const sal_uInt32* pData, sal_uInt32 nOffset, sal_uInt32 nScanSize )
 {
 	DBG_ASSERT( !!maBitmap && !!maMask, "Missing call to ImageConsumer::SetColorModel(...)!" );
 
@@ -521,7 +521,7 @@ void ImageConsumer::SetPixelsByLongs( UL
 
 // -----------------------------------------------------------------------------
 
-void ImageConsumer::Completed( ULONG nStatus /*, ImageProducer& rProducer */ )
+void ImageConsumer::Completed( sal_uInt32 nStatus /*, ImageProducer& rProducer */ )
 {
 	delete mpMapper;
 	mpMapper = NULL;
@@ -565,7 +565,7 @@ void ImageConsumer::DataChanged()
 
 // -----------------------------------------------------------------------------
 
-ULONG ImageConsumer::GetStatus() const
+sal_uInt32 ImageConsumer::GetStatus() const
 {
 	return mnStatus;
 }
--- vcl/source/gdi/pdfwriter_impl.cxx	31 Jan 2005 13:23:40 -0000	1.78
+++ vcl/source/gdi/pdfwriter_impl.cxx	10 Mar 2005 02:39:04 -0000	1.75.50.2
@@ -3497,7 +3497,7 @@ void PDFWriterImpl::createDefaultPushBut
     OStringBuffer aDA( 256 );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetButtonTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[m_nCurrentPage].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[m_nCurrentPage].appendMappedLength( sal_Int32( aFont.GetHeight() ), aDA );
     aDA.append( " Tf" );
     rButton.m_aDAString = aDA.makeStringAndClear();
 
@@ -3554,7 +3554,7 @@ void PDFWriterImpl::createDefaultEditApp
     OStringBuffer aDA( 32 );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetFieldTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( sal_Int32( aFont.GetHeight() ), aDA );
     aDA.append( " Tf" );
 
     /*  create an empty appearance stream, let the viewer create
@@ -3655,7 +3655,7 @@ void PDFWriterImpl::createDefaultListBox
 #endif
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetFieldTextColor() ), aDA );
     aDA.append( " /HelvReg " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aFont.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( sal_Int32( aFont.GetHeight() ), aDA );
     aDA.append( " Tf" );
     rBox.m_aDAString = aDA.makeStringAndClear();
 }
@@ -3743,7 +3743,7 @@ void PDFWriterImpl::createDefaultCheckBo
     aDA.append( "/Tx BMC\r\nq BT\r\n" );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ), aDA );
     aDA.append( " /ZaDb " );
-    m_aPages[ m_nCurrentPage ].appendMappedLength( aCheckRect.GetHeight(), aDA );
+    m_aPages[ m_nCurrentPage ].appendMappedLength( sal_Int32( aCheckRect.GetHeight() ), aDA );
     aDA.append( " Tf\r\n" );
     m_aPages[ m_nCurrentPage ].appendMappedLength( nCharXOffset, aDA );
     aDA.append( " " );
@@ -3838,7 +3838,7 @@ void PDFWriterImpl::createDefaultRadioBu
     aDA.append( "/Tx BMC\r\nq BT\r\n" );
     appendNonStrokingColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ), aDA );
     aDA.append( " /ZaDb " );
-    m_aPages[m_nCurrentPage].appendMappedLength( aCheckRect.GetHeight(), aDA );
+    m_aPages[m_nCurrentPage].appendMappedLength( sal_Int32( aCheckRect.GetHeight() ), aDA );
     aDA.append( " Tf\r\n0 0 Td\r\nET\r\nQ\r\n" );
     writeBuffer( aDA.getStr(), aDA.getLength() );
     setFillColor( replaceColor( rWidget.TextColor, rSettings.GetRadioCheckTextColor() ) );
--- vcl/source/gdi/pngwrite.cxx	28 Jun 2004 16:23:56 -0000	1.4
+++ vcl/source/gdi/pngwrite.cxx	20 Oct 2004 09:48:49 -0000	1.3.6.2
@@ -149,7 +149,7 @@ private:
 	void				ImplWritePalette();
 	void				ImplOpenChunk( ULONG nChunkType );
 	void				ImplWriteChunk( BYTE nNumb );
-	void				ImplWriteChunk( ULONG nNumb );
+	void				ImplWriteChunk( sal_uInt32 nNumb );
 	void				ImplWriteChunk( unsigned char* pSource, sal_uInt32 nDatSize );
 	void				ImplCloseChunk( void );
 };
@@ -331,8 +331,8 @@ std::vector< vcl::PNGWriter::ChunkData >
 BOOL PNGWriterImpl::ImplWriteHeader()
 {
 	ImplOpenChunk(PNGCHUNK_IHDR);
-	ImplWriteChunk( ( mnWidth = (ULONG)mpAccess->Width() ) );
-	ImplWriteChunk( ( mnHeight = (ULONG)mpAccess->Height() ) );
+	ImplWriteChunk( sal_uInt32( mnWidth =  mpAccess->Width() ) );
+	ImplWriteChunk( sal_uInt32( mnHeight = mpAccess->Height() ) );
 
 	if ( mnWidth && mnHeight && mnBitsPerPixel && mbStatus )
 	{
@@ -689,7 +689,7 @@ void PNGWriterImpl::ImplWriteChunk ( BYT
 	maChunkSeq.back().aData.push_back( nSource );
 }
 
-void PNGWriterImpl::ImplWriteChunk ( ULONG nSource )
+void PNGWriterImpl::ImplWriteChunk ( sal_uInt32 nSource )
 {
 	vcl::PNGWriter::ChunkData& rChunkData = maChunkSeq.back();
 	rChunkData.aData.push_back( (sal_uInt8)( nSource >> 24 ) );
--- vcl/source/window/dockmgr.cxx	31 Jan 2005 13:25:12 -0000	1.10
+++ vcl/source/window/dockmgr.cxx	10 Mar 2005 02:45:01 -0000	1.4.40.4
@@ -241,7 +241,7 @@ IMPL_LINK( ImplDockFloatWin2, DockingHdl
         {
             Point aPt;
             Rectangle aBorderRect( aPt, pBorder->GetSizePixel() );
-            long nLeft, nTop, nRight, nBottom;
+            sal_Int32 nLeft, nTop, nRight, nBottom;
             GetBorder( nLeft, nTop, nRight, nBottom );
             // limit borderrect to the caption part only and without the resizing borders
             aBorderRect.nBottom = aBorderRect.nTop + nTop;
--- vcl/unx/source/app/i18n_cb.cxx	8 Sep 2004 15:37:43 -0000	1.22
+++ vcl/unx/source/app/i18n_cb.cxx	20 Oct 2004 09:38:01 -0000	1.21.144.2
@@ -573,8 +573,8 @@ PreeditCaretCallback ( XIC ic, XPointer 
 		case XIMDontChange:   direction = "Dont change";   break;
 	}
 
-	fprintf (stderr, "PreeditCaretCallback( ic=%i, client=%i,\n",
-		(unsigned int)ic, (unsigned int)client_data );
+	fprintf (stderr, "PreeditCaretCallback( ic=%p, client=%p,\n",
+		ic, client_data );
 	fprintf (stderr, "\t position=%i, direction=\"%s\", style=\"%s\" )\n",
 		call_data->position, direction, style );
 
--- vcl/unx/source/gdi/salgdi3.cxx	7 Jan 2005 09:26:25 -0000	1.123
+++ vcl/unx/source/gdi/salgdi3.cxx	18 Jan 2005 17:23:11 -0000	1.119.4.4
@@ -475,7 +475,7 @@ SalDisplay::GetFont( const ExtendedXlfd 
 				if( pFontCache_->GetCurPos() )
 				{
 					pFontCache_->Remove( pItem );
-					pFontCache_->Insert( pItem, (sal_uInt32)0 );
+					pFontCache_->Insert( pItem, 0UL );
 				}
 				return pItem;
 			}
@@ -503,7 +503,7 @@ SalDisplay::GetFont( const ExtendedXlfd 
 	ExtendedFontStruct *pItem = new ExtendedFontStruct( GetDisplay(),
 										rPixelSize, bVertical,
 										const_cast<ExtendedXlfd*>(pRequestedFont) );
-	pFontCache_->Insert( pItem, (sal_uInt32)0 );
+	pFontCache_->Insert( pItem, 0UL );
 	pItem->AddRef();
 
 	return pItem;
