--- starmath/source/accessibility.cxx	7 Nov 2003 15:21:58 -0000	1.26
+++ starmath/source/accessibility.cxx	10 Sep 2004 13:53:27 -0000	1.26.80.1
@@ -638,7 +638,7 @@ awt::Rectangle SAL_CALL SmGraphicAccessi
                 aTLPos.X() -= 0;
                 Size  aSize (pNode->GetSize());
 
-                long *pXAry = new long[ aNodeText.Len() ];
+                sal_Int32 *pXAry = new sal_Int32[ aNodeText.Len() ];
                 pWin->SetFont( pNode->GetFont() );
                 pWin->GetTextArray( aNodeText, pXAry, 0, aNodeText.Len() );
                 aTLPos.X()    += nNodeIndex > 0 ? pXAry[nNodeIndex - 1] : 0;
@@ -721,7 +721,7 @@ sal_Int32 SAL_CALL SmGraphicAccessible::
 
                 long nNodeX = pNode->GetLeft();
 
-                long *pXAry = new long[ aTxt.Len() ];
+                sal_Int32 *pXAry = new sal_Int32[ aTxt.Len() ];
                 pWin->SetFont( pNode->GetFont() );
                 pWin->GetTextArray( aTxt, pXAry, 0, aTxt.Len() );
                 for (sal_Int32 i = 0;  i < aTxt.Len()  &&  nRes == -1;  ++i)
--- starmath/source/parse.cxx	11 Aug 2004 15:08:01 -0000	1.27
+++ starmath/source/parse.cxx	14 Oct 2004 02:21:56 -0000	1.26.58.2
@@ -620,7 +620,7 @@ void SmParser::NextToken()
 	}
 	else if (aRes.TokenType & KParseType::BOOLEAN)
 	{
-		long   &rnEndPos = aRes.EndPos;
+		sal_Int32   &rnEndPos = aRes.EndPos;
 		String	aName( BufferString.Copy( nRealStart, rnEndPos - nRealStart ) );
 		if (2 >= aName.Len())
 		{
@@ -724,7 +724,7 @@ void SmParser::NextToken()
 	}
 	else if (aRes.TokenType & KParseType::ONE_SINGLE_CHAR)
 	{
-		long   &rnEndPos = aRes.EndPos;
+		sal_Int32   &rnEndPos = aRes.EndPos;
 		String	aName( BufferString.Copy( nRealStart, rnEndPos - nRealStart ) );
 
 		if (1 == aName.Len())
