--- desktop/source/deployment/gui/dp_gui_dialog.cxx	2005-12-16 16:01:42.000000000 +0100
+++ desktop/source/deployment/gui/dp_gui_dialog.cxx	2005-12-16 16:02:32.000000000 +0100
@@ -205,11 +205,11 @@ DialogImpl::~DialogImpl()
     that->Resize();
     
     // calculate tabs:
-    sal_Int32 selWidth = that->m_treelb->GetSizePixel().getWidth();
-    sal_Int32 vscrollWidth =
+    long selWidth = that->m_treelb->GetSizePixel().getWidth();
+    long vscrollWidth =
         Application::GetSettings().GetStyleSettings().GetScrollBarSize();
     
-    sal_Int32 statusWidth = that->m_treelb->GetTextWidth(
+    long statusWidth = that->m_treelb->GetTextWidth(
         that->m_treelb->m_strEnabled );
     statusWidth = ::std::max(
         statusWidth, that->m_treelb->GetTextWidth(
@@ -227,7 +227,7 @@ DialogImpl::~DialogImpl()
         selWidth - statusWidth - vscrollWidth );
     that->m_headerBar->InsertItem( ITEM_ID_STATUS, strStatus, statusWidth );
     
-    sal_Int32 tabs[ 3 ];
+    long tabs[ 3 ];
     tabs[ 0 ] = 2; // two tabs
     tabs[ 1 ] = statusWidth;
     tabs[ 2 ] = selWidth - statusWidth - vscrollWidth;
