--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ libwpd/prj/d.lst	2004-07-30 15:02:53.926973008 +0100
@@ -0,0 +1,4 @@
+mkdir: %_DEST%\inc%_EXT%\libwpd
+..\%__SRC%\misc\build\libwpd-0.8.0\src\lib\*.h %_DEST%\inc%_EXT%\libwpd\*.h
+..\%__SRC%\lib\*.so %_DEST%\lib%_EXT%\*.so
+..\%__SRC%\bin\*.dll %_DEST%\lib%_EXT%\*.dll
--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ libwpd/prj/build.lst	2004-07-30 15:04:08.946568296 +0100
@@ -0,0 +1,2 @@
+wp	libwpd	: sfx2 NULL
+wp	libwpd	nmake	-	all	libpwd_mkout NULL
