Index: sw/prj/build.lst
===================================================================
RCS file: /cvs/sw/sw/prj/build.lst,v
retrieving revision 1.6
diff -u -r1.6 build.lst
--- sw/prj/build.lst	27 May 2003 15:49:00 -0000	1.6
+++ sw/prj/build.lst	24 Aug 2004 10:17:56 -0000
@@ -1,4 +1,4 @@
-sw	sw	:	connectivity offmgr NULL
+sw	sw	:	connectivity offmgr writerperfect NULL
 sw	sw										usr1	-	all	sw_mkout NULL
 sw	sw\inc									get		-	all	sw_inc NULL
 sw	sw\prj									get		-	all	sw_prj NULL
