Index: sw/inc/iodetect.hxx
===================================================================
RCS file: /cvs/sw/sw/inc/Attic/iodetect.hxx,v
retrieving revision 1.19
diff -u -r1.19 iodetect.hxx
--- sw/inc/iodetect.hxx	15 Apr 2003 16:47:52 -0000	1.19
+++ sw/inc/iodetect.hxx	24 Aug 2004 09:14:14 -0000
@@ -154,7 +154,7 @@
 #ifdef DEBUG_SH
 		1 +
 #endif
-		21;
+		22;
 
 #define FORAMTNAME_SW4 		"StarWriter 4.0"
 #define FORAMTNAME_SW3   	"StarWriter 3.0"
@@ -195,6 +195,7 @@
 sal_Char __FAR_DATA FILTER_XMLV[] 	= "CXMLV";	                         \
 sal_Char __FAR_DATA FILTER_XMLVW[] 	= "CXMLVWEB";	                     \
 sal_Char __FAR_DATA sSwDos[] 		= "SW6";	                            \
+sal_Char __FAR_DATA sWPD[]			= "WPD";	                            \
 																	        \
 																	        \
 SwIoDetect aReaderWriter[ MAXFILTER ] = { 							        \
@@ -217,7 +218,8 @@
 /* 16*/ SwIoEntry(sWW1,  			STRING_LEN,	0,				  	TRUE),  \
 /* 17*/ SwIoEntry(sWW5,				STRING_LEN,	0, 					FALSE), \
 /* 18*/ SwIoEntry(sSwg1,            4,          0,  				FALSE), \
-/* 19*/ SwIoEntry(FILTER_XML,		4,			&::GetXMLWriter,	TRUE)	\
+/* 19*/ SwIoEntry(sWPD,				4,			0,					TRUE),	\
+/* 20*/ SwIoEntry(FILTER_XML,		4,			&::GetXMLWriter,	TRUE)	\
                                                                             \
 /* opt*/ DEB_SH_SwIoEntry(sW4W_Int, STRING_LEN, 0,				  	TRUE)   \
 /* opt*/ DEB_DBWRT_SwIoEntry(sDebug,STRING_LEN, &::GetDebugWriter,	FALSE)  \
@@ -307,6 +309,11 @@
 		bRet = 0 == strncmp( sSw6_FormatStt, pHeader, 12 ) &&               \
 		   	   0 == strncmp( sSw6_FormatEnd, pHeader + 12 + 1, 4 );			\
 	}                                                                       \
+	else if( sWPD == pName )												\
+	{																		\
+		sal_Char __READONLY_DATA aWPD_Magic[] = "WPC";						\
+		bRet = 0 == strncmp( aWPD_Magic, pHeader + 1, 3);					\
+	}																		\
     else if (FILTER_TEXT == pName) \
         bRet = SwIoSystem::IsDetectableText(pHeader, nLen); \
     else if (FILTER_W4W == pName) \
@@ -334,9 +342,9 @@
 	if( rUserData.EqualsAscii(sWW6) || rUserData.EqualsAscii(FILTER_WW8) )            \
 		return String::CreateFromAscii(                                               \
 				RTL_CONSTASCII_STRINGPARAM( "WordDocument" ));                        \
-	if( rUserData.EqualsAscii(sExcel) || rUserData.EqualsAscii(sCExcel) )             \
+	if( rUserData.EqualsAscii(sWPD) )										          \
 		return String::CreateFromAscii(                                               \
-				RTL_CONSTASCII_STRINGPARAM( "Book" ));                                \
+				RTL_CONSTASCII_STRINGPARAM( "PerfectOffice_MAIN" ));				  \
 	return String::CreateFromAscii( RTL_CONSTASCII_STRINGPARAM( "" ));                \
 }                                                                                     \
 const SfxFilter* SwIoSystem::GetFilterOfFormat( const String& rFmtNm,                 \
