--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ writerperfect/util/makefile.mk	2004-07-30 22:26:16.383863680 +0100
@@ -0,0 +1,34 @@
+PRJ=..
+PRJNAME=wpft
+TARGET=wpft
+VERSION=$(UPD)
+
+.INCLUDE :  settings.mk
+
+LIB1TARGET= $(SLB)$/$(TARGET).lib
+LIB1FILES= \
+	$(SLB)$/ooo.lib  \
+	$(SLB)$/stream.lib  \
+	$(SLB)$/filter.lib
+SHL1LIBS=$(LIB1TARGET) 
+SHL1STDLIBS+= \
+	$(SVLLIB)	\
+	$(SOTLIB) \
+	$(SVXLIB) \
+	$(SO2LIB) \
+	$(SVTOOLLIB) \
+	$(UNOTOOLSLIB) \
+	$(TOOLSLIB) \
+	$(COMPHELPERLIB) \
+	$(UCBHELPERLIB) \
+	$(CPPUHELPERLIB) \
+	$(CPPULIB) \
+	$(SALLIB) \
+	-lwpd
+SHL1TARGET = $(TARGET)$(UPD)$(DLLPOSTFIX)
+SHL1IMPLIB = i$(SHL1TARGET)
+SHL1LIBS = $(LIB1TARGET)
+SHL1VERSIONMAP=$(TARGET).map
+DEF1NAME=$(SHL1TARGET)
+
+.INCLUDE :  target.mk
--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ writerperfect/util/wpft.map	2004-07-30 22:26:28.461027672 +0100
@@ -0,0 +1,8 @@
+WPFT_1_0 {
+	global:
+		component_getImplementationEnvironment;
+		component_writeInfo;
+		component_getFactory;
+	local:
+		*;
+};
