--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ writerperfect/source/stream/makefile.mk	2004-07-30 18:13:05.301180440 +0100
@@ -0,0 +1,14 @@
+PRJ=..$/..
+
+PRJNAME=stream
+TARGET=stream
+ENABLE_EXCEPTIONS=true
+
+.INCLUDE :  settings.mk
+
+# broken but ... necessary, internal include shafted ...
+INCPRE+=$(SOLARVER)$/$(UPD)$/$(INPATH)$/inc$/libwpd
+
+SLOFILES= $(SLO)$/WPXSvStream.obj
+
+.INCLUDE :  target.mk
