--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ /opt/OpenOffice/OOO_1_1_2/writerperfect/source/ooo/makefile.mk	2004-07-30 18:20:03.127661160 +0100
@@ -0,0 +1,17 @@
+PRJ=..$/..
+
+PRJNAME=ooo
+TARGET=ooo
+ENABLE_EXCEPTIONS=true
+
+.INCLUDE :  settings.mk
+
+# broken but ... necessary, internal include shafted ...
+INCPRE+=$(SOLARVER)$/$(UPD)$/$(INPATH)$/inc$/libwpd
+
+SLOFILES= \
+	$(SLO)$/attrlist.obj	\
+	$(SLO)$/xmlkywd.obj	\
+	$(SLO)$/xmltoken.obj
+
+.INCLUDE :  target.mk
