--- /dev/null	2004-06-30 20:04:37.000000000 +0100
+++ writerperfect/source/filter/FontMap.cxx	2004-07-30 22:22:29.488357040 +0100
@@ -0,0 +1,20 @@
+#include "FontMap.hxx"
+#include <string.h>
+
+#ifdef FONTMAPPING
+const char * mapFont(const char *font)
+{
+        for (int i = 0; fontConversionTable[i].pWordPerfectFont; i++) 
+	{
+                if (!strcmp(font, fontConversionTable[i].pWordPerfectFont)) {
+                        return fontConversionTable[i].pConvertedFont;
+                }
+        }
+	return font;
+}
+#else
+const char * mapFont(const char *font)
+{
+	return font;
+}
+#endif
