diff -rup scp2-orig/source/ooo/file_library_ooo.scp scp2/source/ooo/file_library_ooo.scp
--- scp2-orig/source/ooo/file_library_ooo.scp	2006-08-01 12:47:31.000000000 +0200
+++ scp2/source/ooo/file_library_ooo.scp	2006-08-01 12:51:18.000000000 +0200
@@ -2452,9 +2452,10 @@ STD_UNO_LIB_FILE( gid_File_Lib_Mdb2_Impl
 STD_LIB_FILE_PATCH( gid_File_Lib_AVMedia, avmedia )
 
 #if defined UNX
-#ifdef SOLAR_JAVA
-UNO_JAR_FILE( gid_File_Jar_AVmedia, avmedia )
-#endif
+SPECIAL_UNO_LIB_FILE_PATCH( gid_File_Lib_AVMediaGStreamer, avmediagst )
+//#ifdef SOLAR_JAVA
+//UNO_JAR_FILE( gid_File_Jar_AVmedia, avmedia )
+//#endif
 #elif defined WNT
 #ifdef ENABLE_DIRECTX
 SPECIAL_UNO_LIB_FILE_PATCH( gid_File_Lib_AVMediaWin, avmediawin )
diff -rup scp2-orig/source/ooo/module_hidden_ooo.scp scp2/source/ooo/module_hidden_ooo.scp
--- scp2-orig/source/ooo/module_hidden_ooo.scp	2006-08-01 12:47:31.000000000 +0200
+++ scp2/source/ooo/module_hidden_ooo.scp	2006-08-01 12:51:33.000000000 +0200
@@ -238,6 +238,7 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Svx,
 	gid_File_Lib_AVMedia,
 	gid_File_Lib_AVMediaWin,
+	gid_File_Lib_AVMediaGStreamer,
 	gid_File_Lib_BaseGfx,
 	gid_File_Lib_Sysdtrans,
 	gid_File_Lib_Sw,
