--- scp2/source/ooo/file_library_ooo.scp.old	2009-04-06 16:42:01.000000000 +0000
+++ scp2/source/ooo/file_library_ooo.scp	2009-04-06 16:42:09.000000000 +0000
@@ -1642,6 +1642,9 @@ STD_UNO_LIB_FILE( gid_File_Lib_Unoxml, u
 // AVMedia
 STD_UNO_LIB_FILE( gid_File_Lib_AVMedia, avmedia )
 
+#ifdef GSTREAMER
+SPECIAL_UNO_LIB_FILE_PATCH( gid_File_Lib_AVMediaGStreamer, avmediagst )
+#else
 #if defined UNX
 #ifdef SOLAR_JAVA
 UNO_JAR_FILE( gid_File_Jar_AVmedia, avmedia )
@@ -1651,6 +1654,7 @@ UNO_JAR_FILE( gid_File_Jar_AVmedia, avme
 SPECIAL_UNO_LIB_FILE( gid_File_Lib_AVMediaWin, avmediawin )
 #endif
 #endif
+#endif
 
 #ifndef WITHOUT_MOZILLA
 File gid_File_Lib_XSec_Framework
--- scp2/source/ooo/module_hidden_ooo.scp.old	2009-04-02 10:51:54.000000000 +0000
+++ scp2/source/ooo/module_hidden_ooo.scp	2009-04-06 16:42:09.000000000 +0000
@@ -217,6 +217,7 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Svx,
 	gid_File_Lib_AVMedia,
 	gid_File_Lib_AVMediaWin,
+	gid_File_Lib_AVMediaGStreamer,
 	gid_File_Lib_BaseGfx,
 	gid_File_Lib_Sysdtrans,
 	gid_File_Lib_Sw,
