diff -rup sd-orig/source/ui/slideshow/slideshowimpl.cxx sd/source/ui/slideshow/slideshowimpl.cxx
--- sd-orig/source/ui/slideshow/slideshowimpl.cxx	2006-08-17 15:48:52.000000000 +0200
+++ sd/source/ui/slideshow/slideshowimpl.cxx	2006-08-17 15:54:35.000000000 +0200
@@ -135,13 +135,17 @@
 // TODO(Q3): This breaks encapsulation. Either export
 // these strings from avmedia, or provide an XManager
 // factory there
+#ifdef GSTREAMER
+#	define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_GStreamer"
+#else
 #ifdef WNT
 #	define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_DirectX"
 #elif defined QUARTZ
 #   define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_QuickTime"
 #else
 #	define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_Java"
 #endif 
+#endif
 
 using ::com::sun::star::uno::UNO_QUERY;
 using ::com::sun::star::uno::UNO_QUERY_THROW;
