diff -rup avmedia-orig/source/viewer/mediawindow.cxx avmedia/source/viewer/mediawindow.cxx
--- avmedia-orig/source/viewer/mediawindow.cxx	2006-09-21 12:10:28.000000000 +0200
+++ avmedia/source/viewer/mediawindow.cxx	2006-09-21 12:15:19.000000000 +0200
@@ -382,6 +382,7 @@ void MediaWindow::getMediaFilters( Filte
                                         "AU Audio", "au",
                                         "AVI", "avi",
                                         "CD Audio", "cda",
+                                        "FLAC Audio", "flac",
                                         "MIDI Audio", "mid;midi",
                                         "MPEG Audio", "mp2;mp3;mpa",
                                         "MPEG Video", "mpg;mpeg;mpv;mp4",
