--- cli_ure/source/basetypes/makefile.mk	2006-03-21 17:43:24.000000000 +0100
+++ cli_ure/source/basetypes/makefile.mk	2006-03-21 17:45:05.000000000 +0100
@@ -95,7 +95,7 @@ $(POLICY_ASSEMBLY_FILE) : $(BIN)$/cli_ba
 	+$(WRAPCMD) $(AL) -out:$@ \
 			-version:$(CLI_BASETYPES_POLICY_VERSION) \
 			-keyfile:$(BIN)$/cliuno.snk \
-			-link:$(BIN)$/cli_basetypes.config
+			-link:cli_basetypes.config,$(BIN)$/cli_basetypes.config
 
 #Create the config file that is used with the policy assembly
 $(BIN)$/cli_basetypes.config: cli_basetypes_config $(BIN)$/cliureversion.mk 
--- cli_ure/source/native/makefile.mk	2006-03-21 17:24:10.000000000 +0100
+++ cli_ure/source/native/makefile.mk	2006-03-21 17:45:16.000000000 +0100
@@ -128,7 +128,7 @@ $(POLICY_ASSEMBLY_FILE) : $(BIN)$/cli_cp
 	+$(WRAPCMD) $(AL) -out:$@ \
 			-version:$(CLI_CPPUHELPER_POLICY_VERSION) \
 			-keyfile:$(BIN)$/cliuno.snk \
-			-link:$(BIN)$/cli_cppuhelper.config
+			-link:cli_cppuhelper.config,$(BIN)$/cli_cppuhelper.config
 
 #Create the config file that is used with the policy assembly
 $(BIN)$/cli_cppuhelper.config: cli_cppuhelper_config $(BIN)$/cliureversion.mk 
--- cli_ure/source/ure/makefile.mk	2006-03-21 17:24:10.000000000 +0100
+++ cli_ure/source/ure/makefile.mk	2006-03-21 17:45:27.000000000 +0100
@@ -93,7 +93,7 @@ $(POLICY_ASSEMBLY_FILE) : $(BIN)$/cli_ur
 	+$(WRAPCMD) $(AL) -out:$@ \
 			-version:$(CLI_URE_POLICY_VERSION) \
 			-keyfile:$(BIN)$/cliuno.snk \
-			-link:$(BIN)$/cli_ure.config
+			-link:cli_ure.config,$(BIN)$/cli_ure.config
 
 #Create the config file that is used with the policy assembly
 $(BIN)$/cli_ure.config: cli_ure_config $(BIN)$/cliureversion.mk 
--- cli_ure/unotypes/makefile.mk	2006-03-09 15:04:42.000000000 +0100
+++ cli_ure/unotypes/makefile.mk	2006-03-21 17:45:36.000000000 +0100
@@ -81,7 +81,7 @@ $(POLICY_ASSEMBLY_FILE) : $(BIN)$/cli_ty
 	+$(WRAPCMD) AL.exe -out:$@ \
 			-version:$(CLI_TYPES_POLICY_VERSION) \
 			-keyfile:$(BIN)$/cliuno.snk \
-			-link:$(BIN)$/cli_types.config
+			-link:cli_types.config,$(BIN)$/cli_types.config
 
 #Create the config file that is used with the policy assembly
 $(BIN)$/cli_types.config: cli_types_config $(BIN)$/cliureversion.mk 
