--- cli_ure/source/climaker/makefile.mk	2007-03-30 14:01:14.000000000 +0200
+++ cli_ure/source/climaker/makefile.mk	2007-03-30 14:05:06.000000000 +0200
@@ -50,7 +50,7 @@ ENABLE_EXCEPTIONS = TRUE
 
 ALLTAR :   # do nothing
 
-.ELSE
+.ELSE # "$(ENABLE_MONO_CLIMAKER)" != "YES"
 
 SLOFILES = $(SLO)$/climaker_mono.obj
 SHL1OBJS=$(SLOFILES)
@@ -64,11 +64,15 @@ $(BIN)$/climaker : climaker_csharp.cs
 	mkbundle2 --static -o $(BIN)$/climaker $(BIN)$/climaker.exe -L $(OUT)/lib
  
 .INCLUDE :  target.mk
-.ENDIF
+.ENDIF # "$(ENABLE_MONO_CLIMAKER)" != "YES"
 
-.ELSE
+.ELSE # "$(ENABLE_MONO)" == "YES"
 
-.IF "$(BUILD_FOR_CLI)" != ""
+.IF "$(BUILD_FOR_CLI)" == ""
+
+ALLTAR :   # do nothing
+
+.ELSE # "$(BUILD_FOR_CLI)" == ""
 
 NO_OFFUH=TRUE
 CPPUMAKERFLAGS =
@@ -128,11 +132,11 @@ $(BIN)$/climaker.exe.config : climaker.e
 	$(GNUCOPY) -f $? $@
 .IF "$(USE_SHELL)"!="4nt"
 	chmod +x $@
-.ENDIF
+.ENDIF # "$(USE_SHELL)"!="4nt"
 
 .INCLUDE :  target.mk
 .INCLUDE : $(PRJ)$/util$/target.pmk
 
-.ENDIF
+.ENDIF # "$(BUILD_FOR_CLI)" == ""
 
-.ENDIF
+.ENDIF # "$(ENABLE_MONO)" == "YES"
