--- cli_ure-orig/prj/build.lst	2007-07-27 14:59:51.000000000 +0200
+++ cli_ure/prj/build.lst	2007-07-27 14:53:12.000000000 +0200
@@ -7,7 +7,7 @@ ure    cli_ure\source\basetypes        n
 ure    cli_ure\source\bootstrap        nmake   -       u       ure_source_bootstrap ure_source_source ure_source_basetypes ure_inc NULL
 ure    cli_ure\source\climaker         nmake   -       all     ure_source_climaker ure_source_basetypes ure_inc NULL
 ure    cli_ure\unotypes                        nmake   -       all     ure_unotypes ure_source_version ure_source_source ure_source_climaker ure_source_basetypes ure_inc NULL
-ure    cli_ure\source\ure                      nmake   -       all     ure_source_ure ure_source_version ure_source_source ure_unotypes ure_inc NULL
+ure    cli_ure\source\ure                      nmake   -       all     ure_source_ure ure_source_bootstrap.u ure_source_version ure_source_source ure_unotypes ure_inc NULL
 ure    cli_ure\source\uno_bridge       nmake   -       w,vc7   ure_source_uno_bridge ure_unotypes ure_source_ure ure_inc NULL
 ure	cli_ure\source\mono_bridge      nmake   -       u       ure_source_mono_bridge ure_unotypes ure_source_ure ure_inc NULL
 ure    cli_ure\source\native           nmake   -       w,vc7   ure_source_native ure_source_version ure_source_source ure_source_ure ure_unotypes ure_source_uno_bridge ure_inc NULL
