--- connectivity/prj/build.lst.old	2009-04-02 10:52:08.000000000 +0000
+++ connectivity/prj/build.lst	2009-04-06 16:42:08.000000000 +0000
@@ -19,6 +19,7 @@ cn  connectivity\source\drivers\calc
 cn  connectivity\source\drivers\odbcbase            nmake   -   all cn_odbcbase cn_dbtools cn_inc NULL
 cn  connectivity\source\drivers\odbc                nmake   -   all cn_odbc cn_odbcbase cn_inc NULL
 cn  connectivity\source\drivers\mysql               nmake   -   all cn_mysql cn_dbtools cn_inc NULL
+cn  connectivity\source\drivers\postgresql          nmake   -   all cn_postgresql cn_dbtools cn_inc NULL
 cn  connectivity\source\drivers\jdbc                nmake   -   all cn_jdbc cn_dbtools cn_inc NULL
 cn  connectivity\source\drivers\adabas              nmake   -   all cn_adabas cn_dbtools cn_odbcbase cn_inc NULL
 cn  connectivity\source\drivers\file                nmake   -   all cn_file cn_dbtools cn_inc NULL
