Index: makefile.mk
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/postgresql/Attic/makefile.mk,v
retrieving revision 1.1.2.11
diff -u -u -r1.1.2.11 makefile.mk
--- connectivity/source/drivers/postgresql/makefile.mk	15 Feb 2007 20:04:47 -0000	1.1.2.11
+++ connectivity/source/drivers/postgresql/makefile.mk	10 May 2007 14:05:08 -0000
@@ -71,7 +71,10 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
+.IF "$(SYSTEM_POSTGRESQL)" != "YES"
 .INCLUDE :  $(SOLARINCDIR)$/postgresql/postgresql-version.mk
+.ENDIF
+
 #-------------------------------------------------------------------
 
 # uno component naming scheme
@@ -80,8 +83,19 @@
 PQ_SDBC_MAJOR=0
 PQ_SDBC_MINOR=7
 PQ_SDBC_MICRO=4
+.IF "$(SYSTEM_POSTGRESQL)" == "YES"
+POSTGRESQL_MAJOR=`pg_config --version | awk '{ print $2 }' | cut -d. -f1`
+POSTGRESQL_MINOR=`pg_config --version | awk '{ print $2 }' | cut -d. -f2`
+POSTGRESQL_MICRO=`pg_config --version | awk '{ print $2 }' | cut -d. -f3`
+.ENDIF
+
+.IF "$(SYSTEM_POSTGRESQL)" != "YES"
+POSTGRESQL_INCLUDES=-I$(SOLARINCDIR)$/postgresql
+.ELSE
+POSTGRESQL_INCLUDES=-I`pg_config --includedir`
+.ENDIF
 
-CFLAGS+=-I$(SOLARINCDIR)$/postgresql \
+CFLAGS+=$(POSTGRESQL_INCLUDES) \
 	-DPOSTGRESQL_MAJOR=$(POSTGRESQL_MAJOR) \
 	-DPOSTGRESQL_MINOR=$(POSTGRESQL_MINOR) \
 	-DPOSTGRESQL_MICRO=$(POSTGRESQL_MICRO) \
