Index: set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.10.6.8
diff -u -u -r1.10.6.8 set_soenv.in
--- config_office/set_soenv.in	3 May 2004 15:26:57 -0000	1.10.6.8
+++ condig_office/set_soenv.in	20 May 2004 13:44:12 -0000
@@ -407,7 +407,7 @@
          $JRETHREADDIR   = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."ppc".$ds."native_threads";
       }
    }
-   elsif ($machine_type eq "s390")
+   elsif (($machine_type eq "s390") or ($machine_type eq "s390x"))
    {  print "Setting Linux S/390 specific values... ";
       $outfile        = "LinuxS390Env.Set";
       $CPU            = "3";
