Index: sal/osl/unx/signal.c
===================================================================
RCS file: /cvs/porting/sal/osl/unx/signal.c,v
retrieving revision 1.12
diff -u -p -u -r1.12 signal.c
--- sal/osl/unx/signal.c	28 May 2003 15:56:05 -0000	1.12
+++ sal/osl/unx/signal.c	4 Jun 2003 10:00:01 -0000
@@ -202,6 +202,10 @@ static sal_Bool InitSignal()
 		bSetILLHandler = sal_True;
 	}
 
+	if ( getenv ("GNOME_DISABLE_CRASH_DIALOG") &&
+	     atoi (getenv ("GNOME_DISABLE_CRASH_DIALOG")) )
+		bSetSEGVHandler = bSetWINCHHandler = bSetILLHandler = bDoHardKill = sal_False;
+
 	SignalListMutex = osl_createMutex();
 
 	act.sa_handler = SignalHandlerFunction;
@@ -309,7 +313,7 @@ int ReportCrash( int Signal )
 				char *pXMLTempName = NULL;
 				char *pStackTempName = NULL;
 
-#ifdef INCLUDE_BACKTRACE
+#if 0
 				char szXMLTempNameBuffer[L_tmpnam];
 				char szStackTempNameBuffer[L_tmpnam];
 				void *stackframes[1024];
@@ -402,8 +406,10 @@ int ReportCrash( int Signal )
 #endif
 
 #else /* defined INCLUDE BACKTRACE */
-				snprintf( szShellCmd, sizeof(szShellCmd)/sizeof(szShellCmd[0]),
-				"crash_report -p %d -s %d", getpid(), Signal );
+				// FIXME: we quite badly need to ungrab the mouse at this stage.
+ 				snprintf( szShellCmd, sizeof(szShellCmd)/sizeof(szShellCmd[0]),
+					  "bug-buddy --appname=soffice.bin --pid=%d --package=OpenOffice.org",
+					  getpid() );
 #endif /* defined INCLUDE BACKTRACE */
 
 
