--- dbaccess/source/ui/querydesign/QueryDesignView.cxx
+++ dbaccess/source/ui/querydesign/QueryDesignView.cxx
@@ -458,8 +458,12 @@
 
 				::rtl::OUString aQuote = xMetaData->getIdentifierQuoteString();
 				::rtl::OUString aTableListStr(sComposedName);
-				aTableListStr += ::rtl::OUString(' ');
-				aTableListStr += ::dbtools::quoteName(aQuote, pEntryTab->GetAliasName());
+				rtl::OUString aAliasName = ::dbtools::quoteName(aQuote, pEntryTab->GetAliasName());
+				if( !aAliasName.equals( sComposedName ) )
+				{
+					aTableListStr += ::rtl::OUString(' ');
+					aTableListStr += aAliasName;
+				}
 				aDBName = aTableListStr;
 			}
 			catch(SQLException&)
@@ -1173,14 +1181,12 @@
 			}
 		}
 		// all tables that haven't a connection to anyone
 		OQueryTableView::OTableWindowMap::const_iterator aTabIter = pTabList->begin();
+		map< ::rtl::OUString,sal_Bool,::comphelper::UStringMixLess> aSingleNames;
 		for(;aTabIter != pTabList->end();++aTabIter)
 		{
 			const OQueryTableWindow* pEntryTab = static_cast<const OQueryTableWindow*>(aTabIter->second);
 			if(!pEntryTab->ExistsAConn())
- 			{
-				aTableListStr += BuildTable(_xConnection,pEntryTab);
-				aTableListStr += ::rtl::OUString(',');
+				searchAndAppendName(_xConnection, pEntryTab, aSingleNames, aTableListStr);
- 			}
 		}
 
