--- sw/source/ui/envelp/mailmrge.cxx
+++ sw/source/ui/envelp/mailmrge.cxx
@@ -249,6 +249,52 @@
 {
     DBG_ERROR("disposing")
 }
+
+void SwMailMergeDlg::ShiftLowerControls(sal_Int16 nPixels)
+{
+	Window* aCntrlArr[] = {
+		&aAllRB       ,
+		&aMarkedRB    ,
+		&aFromRB      ,
+		&aFromNF      ,
+		&aBisFT       ,
+		&aToNF        ,
+		&aRecordFL    ,
+		&aPrinterRB   ,
+		&aMailingRB   ,
+		&aFileRB      ,
+		&aSingleJobsCB,
+		&aPathFT      ,
+		&aPathED      ,
+		&aPathPB      ,
+		&aFilenameFT  ,
+		&aColumnRB    ,
+		&aFilenameRB  ,
+		&aColumnLB    ,
+		&aFilenameED  ,
+		&aAddressFT   ,
+		&aAddressFldLB,
+		&aSubjectFT   ,
+		&aSubjectED   ,
+		&aFormatFT    ,
+		&aAttachFT    ,
+		&aAttachED    ,
+		&aAttachPB    ,
+		&aFormatHtmlCB,
+		&aFormatRtfCB ,
+		&aFormatSwCB  ,
+		&aDestFL      ,
+		&aSeparatorFL ,
+		0};
+
+	for( Window** ppW = aCntrlArr; *ppW; ++ppW )
+	{
+		::Point aPnt( (*ppW)->GetPosPixel() );
+		aPnt.Y() += nPixels;
+		(*ppW)->SetPosPixel( aPnt );
+	}
+}
+
 /*------------------------------------------------------------------------
  Beschreibung:
 ------------------------------------------------------------------------*/
@@ -335,51 +382,21 @@
         ::Size aSize = GetSizePixel();
         aSize.Height() -= nDiff;
         SetSizePixel(aSize);
-        Window* aCntrlArr[] = {
-            &aAllRB       ,
-            &aMarkedRB    ,
-            &aFromRB      ,
-            &aFromNF      ,
-            &aBisFT       ,
-            &aToNF        ,
-            &aRecordFL    ,
-            &aPrinterRB   ,
-            &aMailingRB   ,
-            &aFileRB      ,
-            &aSingleJobsCB,
-            &aPathFT      ,
-            &aPathED      ,
-            &aPathPB      ,
-            &aFilenameFT  ,
-            &aColumnRB    ,
-            &aFilenameRB  ,
-            &aColumnLB    ,
-            &aFilenameED  ,
-            &aAddressFT   ,
-            &aAddressFldLB,
-            &aSubjectFT   ,
-            &aSubjectED   ,
-            &aFormatFT    ,
-            &aAttachFT    ,
-            &aAttachED    ,
-            &aAttachPB    ,
-            &aFormatHtmlCB,
-            &aFormatRtfCB ,
-            &aFormatSwCB  ,
-            &aDestFL      ,
-            &aSeparatorFL ,
-            0};
-
-        for( Window** ppW = aCntrlArr; *ppW; ++ppW )
-		{
-            ::Point aPnt( (*ppW)->GetPosPixel() );
-			aPnt.Y() -= nDiff;
-			(*ppW)->SetPosPixel( aPnt );
-		}
+		ShiftLowerControls( -nDiff );
     }
     else
     {
+		// Records view designed to be too small, double it's size
+        long nDiff = aRecordFL.GetPosPixel().Y() - pBeamerWin->GetPosPixel().Y();
+        ::Size aSize = GetSizePixel();
+        aSize.Height() += nDiff;
+        SetSizePixel(aSize);
+		aSize = pBeamerWin->GetSizePixel();
+		aSize.Height() += nDiff;
+		pBeamerWin->SetSizePixel(aSize);
+		ShiftLowerControls( +nDiff );
+
         try
         {
             // create a frame wrapper for myself



--- sw/source/ui/inc/mailmrge.hxx
+++ sw/source/ui/inc/mailmrge.hxx
@@ -177,6 +177,7 @@
 
 	virtual void 	Apply();
 	void 			ExecQryShell(BOOL bVisible);
+	void            ShiftLowerControls(sal_Int16 nPixels);
 
 public:
 	 SwMailMergeDlg(Window* pParent, SwWrtShell& rSh,
