Index: desktop/source/app/app.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/app.cxx,v
retrieving revision 1.121.10.8
diff -u -r1.121.10.8 app.cxx
--- desktop/source/app/app.cxx	2 Mar 2004 10:06:34 -0000	1.121.10.8
+++ desktop/source/app/app.cxx	2 Oct 2004 16:36:23 -0000
@@ -2128,14 +2128,14 @@
         OUString aMsg = OUString::createFromAscii(
             "Desktop::OpenDefault() IllegalArgumentException while calling loadComponentFromURL: ") 
             + iae.Message;
-        OSL_ENSURE( sal_False, aMsg);
+        OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
     }
     catch (com::sun::star::io::IOException& ioe)
     {
         OUString aMsg = OUString::createFromAscii(
             "Desktop::OpenDefault() IOException while calling loadComponentFromURL: ") 
             + ioe.Message;
-        OSL_ENSURE( sal_False, aMsg);
+        OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
     }
 
     // shut down again if no component could be loaded
@@ -2286,14 +2286,14 @@
                     OUString aMsg = OUString::createFromAscii(
                         "handle app event IllegalArgumentException while calling loadComponentFromURL: ") 
                         + iae.Message;
-                    OSL_ENSURE( sal_False, aMsg);
+                    OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
                 }
                 catch (com::sun::star::io::IOException& ioe)
                 {
                     OUString aMsg = OUString::createFromAscii(
                         "handle app event IOException while calling loadComponentFromURL: ") 
                         + ioe.Message;
-                    OSL_ENSURE( sal_False, aMsg);
+                    OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
                 }
 
                 if ( !xDoc.is() )
Index: desktop/source/app/dispatchwatcher.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/dispatchwatcher.cxx,v
retrieving revision 1.15.82.1
diff -u -r1.15.82.1 dispatchwatcher.cxx
--- desktop/source/app/dispatchwatcher.cxx	9 Jan 2004 16:43:26 -0000	1.15.82.1
+++ desktop/source/app/dispatchwatcher.cxx	2 Oct 2004 16:36:24 -0000
@@ -353,14 +353,14 @@
                 OUString aMsg = OUString::createFromAscii(
                     "Dispatchwatcher IllegalArgumentException while calling loadComponentFromURL: ") 
                     + iae.Message;
-                OSL_ENSURE( sal_False, aMsg);
+                OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
             }
             catch (com::sun::star::io::IOException& ioe)
             {
                 OUString aMsg = OUString::createFromAscii(
                     "Dispatchwatcher IOException while calling loadComponentFromURL: ") 
                     + ioe.Message;
-                OSL_ENSURE( sal_False, aMsg);
+                OSL_ENSURE( sal_False, OUStringToOString(aMsg, RTL_TEXTENCODING_ASCII_US).getStr());
             }
 			if ( aDispatchRequest.aRequestType == REQUEST_OPEN ||
 				 aDispatchRequest.aRequestType == REQUEST_VIEW ||
