diff -urN desktop.orig/prj/d.lst desktop/prj/d.lst
--- desktop.orig/prj/d.lst	2004-01-28 14:06:17.000000000 +0100
+++ desktop/prj/d.lst	2004-08-02 12:33:55.844955508 +0200
@@ -60,6 +60,7 @@
 ..\%__SRC%\misc\sweb.sh %_DEST%\bin%_EXT%\sweb
 ..\%__SRC%\misc\swriter.sh %_DEST%\bin%_EXT%\swriter
 ..\%__SRC%\misc\fromTemplate.sh %_DEST%\bin%_EXT%\fromtemplate
+..\%__SRC%\misc\mozwrapper.sh %_DEST%\bin%_EXT%\mozwrapper
 ..\%__SRC%\misc\nswrapper.sh %_DEST%\bin%_EXT%\nswrapper
 
 dos: sh -c "if test %OS% = MACOSX; then create-bundle %_DEST%\lib%_EXT%\*.dylib; fi"
diff -urN desktop.orig/scripts/makefile.mk desktop/scripts/makefile.mk
--- desktop.orig/scripts/makefile.mk	2003-03-25 14:50:59.000000000 +0100
+++ desktop/scripts/makefile.mk	2004-08-02 12:33:56.130905875 +0200
@@ -68,6 +68,7 @@
 # --- Targets -------------------------------------------------------------
 
 UNIXTEXT= \
+	$(MISC)$/mozwrapper.sh \
 	$(MISC)$/nswrapper.sh \
 	$(MISC)$/fromTemplate.sh \
 	$(MISC)$/sagenda.sh \
diff -urN desktop.orig/scripts/mozwrapper.sh desktop/scripts/mozwrapper.sh
--- desktop.orig/scripts/mozwrapper.sh	1970-01-01 01:00:00.000000000 +0100
+++ desktop/scripts/mozwrapper.sh	2004-08-02 12:33:56.046920452 +0200
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+# if mozilla is not found, specify full path here
+MOZILLA=mozilla
+
+if ${MOZILLA} -remote "openURL($1,new-window)" 2>&1 | egrep -si "not running on display"; then
+	${MOZILLA} $1
+fi
diff -urN scp.orig/inc/gid.inc scp/inc/gid.inc
--- scp.orig/inc/gid.inc	2004-05-03 18:58:52.000000000 +0200
+++ scp/inc/gid.inc	2004-08-02 12:37:27.888151345 +0200
@@ -635,6 +635,7 @@
 #define GID_FILE_EXE_SWEB                           gid_File_Exe_Sweb
 #define GID_FILE_EXE_SWRITER                        gid_File_Exe_Swriter
 #define GID_FILE_EXE_QUICKSTART                     gid_File_Exe_Quickstart
+#define GID_FILE_EXE_MOZWRAPPER                     gid_File_Exe_Mozwrapper
 #define GID_FILE_EXE_NSWRAPPER                      gid_File_Exe_Nswrapper
 
 #define GID_FILE_BIN_GNOMEINT                       gid_File_Bin_Gnomeint
diff -urN scp.orig/source/office/files.scp scp/source/office/files.scp
--- scp.orig/source/office/files.scp	2004-05-12 17:46:28.000000000 +0200
+++ scp/source/office/files.scp	2004-08-02 12:37:27.866155162 +0200
@@ -2724,6 +2724,13 @@
         Name            = "nswrapper";
 End
 
+File GID_FILE_EXE_MOZWRAPPER
+        BIN_FILE_BODY;
+        Dir             = GID_DIR_PROGRAM;
+        Styles          = (PACKED);
+        Name            = "mozwrapper";
+End
+
 #endif
 
 #endif // ACHUNG: Hier Ende für reine Unix-Dateien (Solaris)
