--- config_office/configure.in
+++ config_office/configure.in
@@ -279,6 +279,12 @@
 
                           Usage: --with-dict=ENGB,ENUS,ITIT
 ],,)
+AC_ARG_WITH(myspell-dicts,
+[  --without-myspell-dicts   Removes myspell dictionaries from openoffice.org
+			  installation set, for people building for specific
+			  distributions where the myspell dictionaries are
+			  installed from other sources
+],,)
 AC_ARG_WITH(asm-home,
 [  --with-asm-home         For Windows users, please supply the path for the
                           ml.exe assembler.
@@ -2356,6 +2362,13 @@
 fi
 AC_SUBST(WITH_DICT)
 
+if test -z "$with_myspell_dicts"; then
+   WITH_MYSPELL_DICTIONARIES="TRUE"
+else
+   WITH_MYSPELL_DICTIONARIES="FALSE"
+fi
+AC_SUBST(WITH_MYSPELL_DICTIONARIES)
+
 if test -n "$enable_static_gtk"; then
    ENABLE_STATIC_GTK="TRUE"
 else
--- config_office/set_soenv.in
+++ config_office/set_soenv.in
@@ -1448,6 +1448,7 @@
        }
    }
 }
+ToFile( "WITH_MYSPELL_DICTIONARIES",  "@WITH_MYSPELL_DICTIONARIES@",       "e" );
 # Languages
 if ('@WITH_LANG@' =~ m/ALL,/s)
 {
--- dictionaries/en_GB/makefile.mk
+++ dictionaries/en_GB/makefile.mk
@@ -77,15 +77,21 @@
 all_target: $(MISC)$/$(TARGET).don 
 
 DIC2BIN= \
-	en_GB.aff \
-	en_GB.dic \
 	hyph_en_GB.dic \
 	README_en_GB.txt
 
+.IF "$(WITH_MYSPELL_DICTIONARIES)" != "FALSE"
+DIC2BIN+= \
+	en_GB.aff \
+	en_GB.dic
+.ENDIF
+
 
 $(MISC)$/$(TARGET).don : 
 	+$(COPY) $(foreach,i,$(DIC2BIN) $i) $(BIN)
+.IF "$(WITH_MYSPELL_DICTIONARIES)" != "FALSE"
 	@echo DICT en GB en_GB >>$(BIN)$/dictionary.lst
+.ENDIF
 	@echo HYPH en GB hyph_en_GB >>$(BIN)$/dictionary.lst
 	@touch $@
 
--- dictionaries/en_US/makefile.mk
+++ dictionaries/en_US/makefile.mk
@@ -83,14 +83,21 @@
 .ENDIF
 
 DIC2BIN= \
-	en_US.aff \
-	en_US.dic \
 	hyph_en_US.dic
 
+.IF "$(WITH_MYSPELL_DICTIONARIES)" != "FALSE"
+
+DIC2BIN+= \
+	en_US.aff \
+	en_US.dic
+
+.ENDIF
 
 $(MISC)$/$(TARGET).don : $(MISC)$/th_en_US.don
 	+$(COPY) $(foreach,i,$(DIC2BIN) $i) $(BIN)
+.IF "$(WITH_MYSPELL_DICTIONARIES)" != "FALSE"
 	@echo DICT en US en_US >>$(BIN)$/dictionary.lst
+.ENDIF
 	@echo HYPH en US hyph_en_US >>$(BIN)$/dictionary.lst
 	@echo THES en US th_en_US >>$(BIN)$/dictionary.lst
 	@touch $@
--- dictionaries/it_IT/makefile.mk
+++ dictionaries/it_IT/makefile.mk
@@ -77,15 +77,20 @@
 all_target: $(MISC)$/$(TARGET).don 
 
 DIC2BIN= \
-	it_IT.aff \
-	it_IT.dic \
 	hyph_it_IT.dic \
 	README_it_IT.txt
 
+.IF "$(WITH_MYSPELL_DICTIONARIES)" != "FALSE"
+DIC2BIN+= \
+	it_IT.aff \
+	it_IT.dic
+.ENDIF
 
 $(MISC)$/$(TARGET).don : 
 	+$(COPY) $(foreach,i,$(DIC2BIN) $i) $(BIN)
+.IF "$(WITH_MYSPELL_DICTIONARIES)" != "FALSE"
 	@echo DICT it IT it_IT >>$(BIN)$/dictionary.lst
+.ENDIF
 	@echo HYPH it IT hyph_it_IT >>$(BIN)$/dictionary.lst
 	@touch $@
 
