Index: solenv/inc/unxlngi4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi4.mk,v
retrieving revision 1.19.18.1
diff -u -p -u -r1.19.18.1 unxlngi4.mk
--- solenv/inc/unxlngi4.mk	15 Aug 2003 12:06:59 -0000	1.19.18.1
+++ solenv/inc/unxlngi4.mk	21 Oct 2003 05:00:48 -0000
@@ -89,7 +89,7 @@ CXX*=g++
 # name of C Compiler
 CC*=gcc
 # flags for C and C++ Compiler
-CFLAGS+=-fmessage-length=0 -c $(INCLUDE)
+CFLAGS+=-fmessage-length=0 -c $(INCLUDE)  -DFONTCONFIG_FONTS
 
 # flags to enable build with symbols; required for crashdump feature
 .IF "$(ENABLE_SYMBOLS)"=="SMALL"
@@ -106,7 +106,7 @@ CFLAGSEXCEPTIONS=-fexceptions -fno-enfor
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # -fpermissive should be removed as soon as possible
-CFLAGSCXX= -pipe -mcpu=pentiumpro -fno-for-scope -fpermissive -fno-rtti
+CFLAGSCXX= -pipe -mcpu=pentiumpro -fno-for-scope -fpermissive -fno-rtti -DFONTCONFIG_FONTS
 
 # HACK: enable Hamburg developers to build on glibc-2.2 machines but compile vs. glibc-2.1 headers
 .IF "$(BUILD_SPECIAL)"==""
