--- offmgr/source/offapp/dialog/makefile.mk.orig	2004-06-15 10:42:41.992786992 +0200
+++ offmgr/source/offapp/dialog/makefile.mk	2004-06-15 10:46:01.819408728 +0200
@@ -80,6 +80,10 @@ PROJECTPCHSOURCE=$(PRJ)$/util\ofapch
 CDEFS=$(CDEFS) -DPRECOMPILED
 .ENDIF
 
+.IF "$(ENABLE_GTK)" = "TRUE" || "$(ENABLE_KDE)" = "TRUE"
+CDEFS+= -DUNX_FPICKER
+.ENDIF
+
 IMGLST_SRS=$(SRS)$/dialog.srs
 BMP_IN=$(PRJ)$/res
 
--- offmgr/source/offapp/dialog/optgdlg.cxx.orig	2004-06-15 09:42:26.972353632 +0200
+++ offmgr/source/offapp/dialog/optgdlg.cxx	2004-06-15 10:47:59.950450080 +0200
@@ -250,8 +250,7 @@ OfaMiscTabPage::OfaMiscTabPage(Window* p
 {
 	FreeResource();
 
-    //quick launch only available in Win
-#ifndef WNT
+#if !defined( WNT ) && !defined( UNX_FPICKER )
     aFileDlgFL.Hide();
     aFileDlgCB.Hide();
 #endif
