Index: gcach_ftyp.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/glyphs/gcach_ftyp.cxx,v
retrieving revision 1.98
diff -u -r1.98 gcach_ftyp.cxx
--- vcl/source/glyphs/gcach_ftyp.cxx	16 Jul 2003 17:46:21 -0000	1.98
+++ vcl/source/glyphs/gcach_ftyp.cxx	18 Nov 2003 01:35:57 -0000
@@ -77,7 +77,8 @@
 #include <osl/file.hxx>
 #include <osl/thread.hxx>
 
-#include "freetype/freetype.h"
+#include "ft2build.h"
+#include FT_FREETYPE_H
 #include "freetype/ftglyph.h"
 #include "freetype/ftoutln.h"
 #include "freetype/tttables.h"
Index: gcach_layout.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/glyphs/gcach_layout.cxx,v
retrieving revision 1.23
diff -u -r1.23 gcach_layout.cxx
--- vcl/source/glyphs/gcach_layout.cxx	16 Jul 2003 17:46:32 -0000	1.23
+++ vcl/source/glyphs/gcach_layout.cxx	18 Nov 2003 01:49:35 -0000
@@ -68,7 +68,8 @@
 #endif
 
 #ifndef _SV_GCACHFTYP_HXX
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <gcach_ftyp.hxx>
 #endif
 
