Index: sal/osl/unx/semaphor.c
===================================================================
RCS file: /cvs/porting/sal/osl/unx/semaphor.c,v
retrieving revision 1.2
diff -u -p -u -r1.2 semaphor.c
--- sal/osl/unx/semaphor.c	1 Mar 2001 13:26:57 -0000	1.2
+++ sal/osl/unx/semaphor.c	5 May 2004 12:27:28 -0000
@@ -85,7 +85,7 @@
 typedef struct {
 	struct { long status; int spinlock; } sem_lock;
 	int sem_value;
-	_pthread_descr sem_waiting;
+	void *sem_waiting;
 } glibc_21_sem_t;
 #endif
 
