Index: vcl/unx/source/gdi/gcach_xpeer.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/gcach_xpeer.cxx,v
retrieving revision 1.30
diff -u -p -u -r1.30 gcach_xpeer.cxx
--- vcl/unx/source/gdi/gcach_xpeer.cxx	28 May 2003 12:33:58 -0000	1.30
+++ vcl/unx/source/gdi/gcach_xpeer.cxx	4 Jun 2003 09:07:47 -0000
@@ -297,16 +297,22 @@ void X11GlyphPeer::RemovingGlyph( Server
         case XRENDER_KIND:
             {
                 GlyphSet aGlyphSet = GetGlyphSet( rServerFont );
-                Glyph nGlyphId = GetGlyphId( rServerFont, nGlyphIndex );
-                // XRenderFreeGlyphs not implemented yet for version<=0.2
-                // #108209# disabled because of crash potential,
-                // the glyph leak is not too bad because they will
-                // be cleaned up when the glyphset is released
+		GlyphData& rGlyphData = rServerFont.GetGlyphData( nGlyphIndex );
+
+		if( rGlyphData.GetExtInfo() != EMPTY_KIND )
+                {
+		    Glyph nGlyphId = GetGlyphId( rServerFont, nGlyphIndex );
+
+		    // XRenderFreeGlyphs not implemented yet for version<=0.2
+		    // #108209# disabled because of crash potential,
+		    // the glyph leak is not too bad because they will
+		    // be cleaned up when the glyphset is released
 #if 0   // TODO: reenable when it works without problems
-                if( nRenderVersion >= 0x05 )
-                    (*pXRenderFreeGlyphs)( mpDisplay, aGlyphSet, &nGlyphId, 1 );
+		    if( nRenderVersion >= 0x05 )
+                        (*pXRenderFreeGlyphs)( mpDisplay, aGlyphSet, &nGlyphId, 1 );
 #endif
-                mnBytesUsed -= nHeight * ((nWidth + 3) & ~3);
+		    mnBytesUsed -= nHeight * ((nWidth + 3) & ~3);
+		}
             }
             break;
     }
