Index: svx/source/dialog/dlgname.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/dlgname.cxx,v
retrieving revision 1.2
diff -u -p -u -r1.2 dlgname.cxx
--- svx/source/dialog/dlgname.cxx	28 Feb 2002 16:47:13 -0000	1.2
+++ svx/source/dialog/dlgname.cxx	24 Oct 2003 11:54:38 -0000
@@ -97,6 +97,7 @@ SvxNameDialog::SvxNameDialog( Window* pW
 
 	aFtDescription.SetText( rDesc );
 	aEdtName.SetText( rName );
+	aEdtName.SetSelection( Selection( SELECTION_MIN, SELECTION_MAX ) );
     ModifyHdl(&aEdtName);
     aEdtName.SetModifyHdl(LINK(this, SvxNameDialog, ModifyHdl));
 }
Index: sc/source/ui/miscdlgs/strindlg.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/miscdlgs/strindlg.cxx,v
retrieving revision 1.2
diff -u -p -u -r1.2 strindlg.cxx
--- sc/source/ui/miscdlgs/strindlg.cxx	5 Dec 2001 21:57:17 -0000	1.2
+++ sc/source/ui/miscdlgs/strindlg.cxx	24 Oct 2003 11:55:20 -0000
@@ -93,6 +93,7 @@ ScStringInputDlg::ScStringInputDlg( Wind
 	SetText( rTitle );
 	aFtEditTitle.SetText( rEditTitle );
 	aEdInput.SetText( rDefault );
+	aEdInput.SetSelection( Selection( SELECTION_MIN, SELECTION_MAX ) );
 
 	//	HelpId for Edit different for different uses
 
Index: sw/source/ui/utlui/unotools.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/utlui/unotools.cxx,v
retrieving revision 1.15
diff -u -p -u -r1.15 unotools.cxx
--- sw/source/ui/utlui/unotools.cxx	17 Apr 2003 15:57:34 -0000	1.15
+++ sw/source/ui/utlui/unotools.cxx	1 Dec 2003 15:32:16 -0000
@@ -182,6 +182,7 @@ SwRenameXNamedDlg::SwRenameXNamedDlg( Wi
 
 	String sTmp(GetText());
 	aNewNameED.SetText(xNamed->getName());
+	aNewNameED.SetSelection( Selection( SELECTION_MIN, SELECTION_MAX ) );
 	sTmp += String(xNamed->getName());
 	SetText(sTmp);
 
