Index: sfx2/source/toolbox/tbxcust.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/toolbox/tbxcust.cxx,v
retrieving revision 1.26
diff -u -p -u -r1.26 tbxcust.cxx
--- sfx2/source/toolbox/tbxcust.cxx	4 Apr 2003 17:39:20 -0000	1.26
+++ sfx2/source/toolbox/tbxcust.cxx	14 May 2003 14:32:23 -0000
@@ -1080,7 +1080,7 @@ SfxUserBitmapDialog_Impl::SfxUserBitmapD
 				if ( !!aBitmap )
 				{
 					if ( aBitmap.GetSizePixel() != aSize )
-						aBitmap.Scale( aSize, BMP_SCALE_FAST );
+						aBitmap.Scale( aSize, BMP_SCALE_INTERPOLATE );
 
 					// Add image to the image list which will be used to initialize toolbox
 					aBitmapUrls.push_back( rUrl );
