Index: officecfg/registry/data/org/openoffice/Office/Common.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Common.xcu,v
retrieving revision 1.13.34.1.18.1
diff -u -r1.13.34.1.18.1 Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu	9 May 2004 15:22:21 -0000	1.13.34.1.18.1
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	17 Aug 2004 11:27:03 -0000
@@ -108,22 +108,31 @@
   </node>
   <node oor:name="ExternalApps">
     <prop oor:name="http" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>desktop-launch</value>
     </prop>
     <prop oor:name="https" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>desktop-launch</value>
     </prop>
     <prop oor:name="ftp" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>desktop-launch</value>
     </prop>
     <prop oor:name="file" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>desktop-launch</value>
     </prop>
     <prop oor:name="mailto" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>desktop-launch</value>
     </prop>
   </node>
   <node oor:name="ExternalMailer">
+    <prop oor:name="CommandProfile" oor:type="xs:string">
+       <value>Desktop Mailer</value>
+    </prop>
+    <prop oor:name="UseDefaultMailer" oor:type="xs:boolean">
+      <value>false</value>
+    </prop>
+    <prop oor:name="Program" oor:type="xs:string">
+      <value>desktop-launch</value>
+    </prop>
     <node oor:name="Profiles">
       <node oor:name="Netscape 6.x - 7.0 (Option 1)" oor:op="replace">
         <node oor:name="FormatStrings">
@@ -419,6 +428,48 @@
       </prop>
      </node>
     </node>   
+     <node oor:name="Desktop Mailer" oor:op="replace">
+      <node oor:name="FormatStrings">
+       <prop oor:name="base">
+        <value>mailto:?%s</value>
+       </prop>
+       <prop oor:name="from">
+        <value/>
+       </prop>
+       <prop oor:name="to">
+        <value>to=%s</value>
+       </prop>
+       <prop oor:name="cc">
+        <value>cc=%s</value>
+       </prop>
+       <prop oor:name="bcc">
+        <value>bcc=%s</value>
+       </prop>
+       <prop oor:name="subject">
+        <value>subject=%s</value>
+       </prop>
+       <prop oor:name="attachment">
+        <value>attachment=%s</value>
+       </prop>
+       <prop oor:name="body">
+        <value>body=%s</value>
+       </prop>
+      </node>
+      <node oor:name="EnumDelimiters">
+       <prop oor:name="base">
+        <value>&amp;</value>
+       </prop>
+       <prop oor:name="cc">
+        <value>&amp;cc=</value>
+       </prop>
+       <prop oor:name="bcc">
+        <value>&amp;bcc=</value>
+       </prop>
+       <prop oor:name="attachment">
+        <value>&amp;attachment=</value>
+       </prop>
+      </node>
+     </node>   
    </node>
   </node>
   <node oor:name="Startup">
