diff -urN tools.orig/inc/resmgr.hxx tools/inc/resmgr.hxx
--- tools.orig/inc/resmgr.hxx	2000-12-05 20:19:20.000000000 +0100
+++ tools/inc/resmgr.hxx	2004-07-28 11:09:44.703247375 +0200
@@ -222,6 +222,11 @@
 									  const UniString* pResPath );
 	static ResMgr*		CreateResMgr( const sal_Char* pPrefixName,
 									  LanguageType eLanguage = LANGUAGE_DONTKNOW ); // nur in VCL
+
+	static ResMgr*		PriorSearchCreateResMgr( const sal_Char* pPrefixName,
+									  LanguageType& eLanguage,
+									  const UniString* pAppName,
+									  const UniString* pResPath );
 	static ResMgr*		SearchCreateResMgr( const sal_Char* pPrefixName,
 											LanguageType& rLanguageReturn );
 
diff -urN tools.orig/source/rc/resmgr.cxx tools/source/rc/resmgr.cxx
--- tools.orig/source/rc/resmgr.cxx	2004-05-03 19:05:50.000000000 +0200
+++ tools/source/rc/resmgr.cxx	2004-07-28 11:09:44.701247721 +0200
@@ -1309,9 +1309,11 @@
         switch ( nType )
         {
             case LANGUAGE_FRENCH_CANADIAN:
+		nType = LANGUAGE_FRENCH;
                 return "33";
 
             case LANGUAGE_PORTUGUESE_BRAZILIAN:
+		nType = LANGUAGE_PORTUGUESE;
                 return "03";
 
             default:
@@ -1319,11 +1321,18 @@
         }
     }
     else if ( nPrio == 2 )
+    {
+	nType = LANGUAGE_ENGLISH_US;
         return "01";
-    else if ( nPrio == 3 )
+    }
+    else if ( nPrio == 3 ) {
+	nType = LANGUAGE_ENGLISH;
         return "44";
-    else if ( nPrio == 4 )
+    }
+    else if ( nPrio == 4 ) {
+	nType = LANGUAGE_GERMAN;
         return "49";
+    }
     else
         return "99";
 }
@@ -1335,6 +1344,16 @@
                               const UniString* pAppName,
                               const UniString* pResPath )
 {
+    return PriorSearchCreateResMgr(pPrefixName, nType, pAppName, pResPath);
+}
+
+// -----------------------------------------------------------------------
+
+ResMgr* ResMgr::PriorSearchCreateResMgr( const sal_Char* pPrefixName,
+                              LanguageType& nType,
+                              const UniString* pAppName,
+                              const UniString* pResPath )
+{
     // Suchreihenfolge festlegen
     const sal_Char* pLang[6];
 
diff -urN xmlhelp.orig/inc/provider/urlparameter.hxx xmlhelp/inc/provider/urlparameter.hxx
--- xmlhelp.orig/inc/provider/urlparameter.hxx	2002-06-27 10:27:02.000000000 +0200
+++ xmlhelp/inc/provider/urlparameter.hxx	2004-07-28 11:32:30.252798753 +0200
@@ -295,6 +295,8 @@
 		bool name( bool modulePresent );
 	
 		bool query();
+		
+		bool check_help( const rtl::OUString& language );
 
 	};   // end class URLParameter
 
diff -urN xmlhelp.orig/prj/build.lst xmlhelp/prj/build.lst
--- xmlhelp.orig/prj/build.lst	2002-02-14 14:22:16.000000000 +0100
+++ xmlhelp/prj/build.lst	2004-07-28 11:32:30.258797709 +0200
@@ -1,4 +1,4 @@
-xh	xmlhelp	:	ucbhelper XmlSearch sablot jut unoil berkeleydb NULL
+xh	xmlhelp	:	ucbhelper XmlSearch sablot jut unoil berkeleydb tools NULL
 xh	xmlhelp									usr1	-	all	xh_mkout NULL
 xh	xmlhelp\source\helpprovider				nmake	-	all	xh_helpprovider NULL
 xh	xmlhelp\source\treeview					nmake	-	all	xh_treeview NULL
diff -urN xmlhelp.orig/source/cxxhelp/provider/urlparameter.cxx xmlhelp/source/cxxhelp/provider/urlparameter.cxx
--- xmlhelp.orig/source/cxxhelp/provider/urlparameter.cxx	2003-08-05 11:42:33.000000000 +0200
+++ xmlhelp/source/cxxhelp/provider/urlparameter.cxx	2004-07-28 11:32:30.253798579 +0200
@@ -105,6 +105,15 @@
 #ifndef _RTL_URI_HXX_
 #include <rtl/uri.hxx>
 #endif
+#ifndef _TOOLS_INTN_HXX
+#include <tools/intn.hxx>
+#endif
+#ifndef _ISOLANG_HXX
+#include <tools/isolang.hxx>
+#endif
+#ifndef _TOOLS_RESMGR_HXX
+#include <tools/resmgr.hxx> 
+#endif
 #ifndef _COM_SUN_STAR_IO_XACTIVEDATASINK_HPP_
 #include <com/sun/star/io/XActiveDataSink.hpp>
 #endif
@@ -272,13 +281,59 @@
 	
 	return m_aTitle;
 }
-		
+
+
+// Check that we have the necessary files. The rationale is if
+// help/<lang>/err.html exists then the localized help data are installed.
+
+bool URLParameter::check_help( const rtl::OUString& language )
+{
+	rtl::OUString fileURL = 
+		m_pDatabases->getInstallPathAsURL()
+		+ m_pDatabases->lang( language )
+		+ rtl::OUString::createFromAscii( "/err.html" );
+	
+	osl::DirectoryItem aDirItem;
+	osl::File aFile( fileURL );
+	osl::FileStatus aStatus( FileStatusMask_FileSize );
+	
+	if( osl::FileBase::E_None != osl::DirectoryItem::get( fileURL,aDirItem ) ||
+		osl::FileBase::E_None != aFile.open( OpenFlag_Read )       ||
+		osl::FileBase::E_None != aDirItem.getFileStatus( aStatus ) )
+		return FALSE;
+
+	return TRUE;
+}
 
 rtl::OUString URLParameter::get_language()
 {
 	if( m_aLanguage.getLength() == 0 )
 		return m_aDefaultLanguage;
-	
+
+	// test the default localization
+	if (check_help(m_aLanguage))
+		return m_aLanguage;
+
+
+	// test localization which is used for menu
+	LanguageType rLangType = GetSystemUILanguage();
+	ResMgr *testRM=ResMgr::PriorSearchCreateResMgr(CREATEVERSIONRESMGR_NAME(vcl), rLangType, NULL, NULL);
+	if (testRM)
+	{
+		delete testRM;
+		UniString rLang, rCountry;
+		ConvertLanguageToIsoNames(rLangType, rLang, rCountry);
+		m_aLanguage =
+			rtl::OUString(rLang.GetBuffer())
+			+ rtl::OUString::createFromAscii( "-" )
+			+ rtl::OUString(rCountry.GetBuffer());
+			
+		if (check_help(m_aLanguage))
+			return m_aLanguage;
+	}
+			
+	// ultimate fallback to the English help
+	m_aLanguage = rtl::OUString::createFromAscii( "en-US" );
 	return m_aLanguage;
 }
 
diff -urN xmlhelp.orig/source/treeview/makefile.mk xmlhelp/source/treeview/makefile.mk
--- xmlhelp.orig/source/treeview/makefile.mk	2001-07-18 12:30:44.000000000 +0200
+++ xmlhelp/source/treeview/makefile.mk	2004-07-28 11:32:30.255798231 +0200
@@ -89,6 +89,7 @@
 	$(EXPATASCII3RDLIB)      \
 	$(SALLIB)  \
 	$(VOSLIB) \
+	$(TOOLSLIB) \
 	$(UCBHELPERLIB)
 SHL1VERSIONMAP=	$(TARGET).map
 
diff -urN xmlhelp.orig/source/treeview/tvread.cxx xmlhelp/source/treeview/tvread.cxx
--- xmlhelp.orig/source/treeview/tvread.cxx	2001-11-08 17:42:46.000000000 +0100
+++ xmlhelp/source/treeview/tvread.cxx	2004-07-28 11:32:30.256798057 +0200
@@ -14,6 +14,15 @@
 #ifndef _OSL_FILE_HXX_
 #include <osl/file.hxx>
 #endif
+#ifndef _TOOLS_INTN_HXX
+#include <tools/intn.hxx>
+#endif
+#ifndef _ISOLANG_HXX
+#include <tools/isolang.hxx>
+#endif
+#ifndef _TOOLS_RESMGR_HXX
+#include <tools/resmgr.hxx> 
+#endif
 #ifndef _COM_SUN_STAR_FRAME_XCONFIGMANAGER_HPP_
 #include <com/sun/star/frame/XConfigManager.hpp>
 #endif
@@ -633,6 +642,24 @@
 
 
 
+// Check that we have the necessary files. The rationale is if
+// help/<lang>/err.html exists then the localized help data are installed.
+
+bool TVChildTarget::checkHelp(const rtl::OUString& url)
+{
+	rtl::OUString errfileURL = url + rtl::OUString::createFromAscii( "/err.html" );
+		
+	osl::DirectoryItem aDirItem;
+	osl::File aFile( errfileURL );
+	osl::FileStatus aStatus( FileStatusMask_FileSize );
+	
+	if( osl::FileBase::E_None != osl::DirectoryItem::get( errfileURL,aDirItem ) ||
+		osl::FileBase::E_None != aFile.open( OpenFlag_Read )       ||
+		osl::FileBase::E_None != aDirItem.getFileStatus( aStatus ) )
+		return FALSE;
+
+	return TRUE;
+}
 
 
 
@@ -703,9 +730,48 @@
 			 osl::FileBase::E_None == osl::DirectoryItem::get( url + locale.copy( 0,idx ),
 															   aDirItem ) )
 		ret = locale.copy( 0,idx );
+
+	bool urlFound = FALSE;
+
+	// test the localization found by locales
+	if (checkHelp( url + ret ))
+	{
+		url = url + ret;
+		urlFound = TRUE;
+	}
 	
-	url = url + ret;
-	
+	// test localization which is used for menu
+	if (!urlFound)
+	{
+		LanguageType rLangType = GetSystemUILanguage();
+		ResMgr *testRM=ResMgr::PriorSearchCreateResMgr(CREATEVERSIONRESMGR_NAME(vcl), rLangType, NULL, NULL);
+		if (testRM)
+		{
+			delete testRM;
+			UniString rLangUni, rCountryUni;
+			ConvertLanguageToIsoNames(rLangType, rLangUni, rCountryUni);
+			rtl::OUString rLang=rtl::OUString(rLangUni.GetBuffer());
+			rtl::OUString rCountry=rtl::OUString(rCountryUni.GetBuffer());
+			
+			if (checkHelp(url + rLang + rtl::OUString::createFromAscii( "-" ) + rCountry))
+			{
+				url = url + rLang + rtl::OUString::createFromAscii( "-" ) + rCountry;
+				urlFound = TRUE;
+			}
+			else if (checkHelp(url + rLang))
+			{
+				url = url + rLang;
+				urlFound = TRUE;
+			}
+		}
+	}
+
+	// ultimate fallback to the English help
+	if (!urlFound)
+	{
+		url = url + rtl::OUString::createFromAscii( "en" );
+	}
+
 	// first of all, try do determine whether there are any *.tree files present
 	
 	osl::Directory aDirectory( url );
diff -urN xmlhelp.orig/source/treeview/tvread.hxx xmlhelp/source/treeview/tvread.hxx
--- xmlhelp.orig/source/treeview/tvread.hxx	2001-10-31 14:51:55.000000000 +0100
+++ xmlhelp/source/treeview/tvread.hxx	2004-07-28 11:32:30.256798057 +0200
@@ -295,6 +295,8 @@
 	private:
 		std::vector< rtl::Reference< TVRead > >   Elements;
 
+		bool checkHelp( const rtl::OUString& url );
+
 		ConfigData init(
 			const com::sun::star::uno::Reference< com::sun::star::lang::XMultiServiceFactory >& xMSF );
 
diff -urN xmlhelp.orig/util/makefile.mk xmlhelp/util/makefile.mk
--- xmlhelp.orig/util/makefile.mk	2003-04-28 18:18:57.000000000 +0200
+++ xmlhelp/util/makefile.mk	2004-07-28 11:32:30.257797883 +0200
@@ -98,6 +98,7 @@
 
 # Add additional libs here.
 SHL1STDLIBS=                     \
+	$(TOOLSLIB)              \
 	$(CPPUHELPERLIB)         \
 	$(CPPULIB)               \
 	$(SALLIB)                \
