Index: xmlhelp/source/treeview/tvread.cxx
===================================================================
RCS file: /cvs/util/xmlhelp/source/treeview/tvread.cxx,v
retrieving revision 1.13
diff -u -p -u -r1.13 tvread.cxx
--- xmlhelp/source/treeview/tvread.cxx	8 Nov 2001 16:42:46 -0000	1.13
+++ xmlhelp/source/treeview/tvread.cxx	12 Aug 2003 10:19:42 -0000
@@ -24,6 +24,9 @@
 #include <com/sun/star/beans/PropertyState.hpp>
 #endif
 
+#include <osl/process.h>
+#include <rtl/locale.hxx>
+
 namespace treeview {
 
 	
@@ -684,7 +687,16 @@ ConfigData TVChildTarget::init( const Re
 	rtl::OUString productVersion( setupversion + 
 								  rtl::OUString::createFromAscii( " " ) +
 								  setupextension );
+#ifdef UNX
+	rtl_Locale *pRtlLocale = NULL;
+	osl_getProcessLocale( &pRtlLocale );
+	rtl::OLocale aLocale( pRtlLocale );
+	rtl::OUString locale = ( aLocale.getLanguage() + 
+				 rtl::OUString::createFromAscii( "-" ) +
+				 aLocale.getCountry() );
+#else
 	rtl::OUString locale( getKey( xHierAccess,"L10N/ooLocale" ) );
+#endif
 
 	
 	// Determine fileurl from url and locale
Index: desktop/source/app/ssoinit.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/ssoinit.cxx,v
retrieving revision 1.17
diff -u -p -u -r1.17 ssoinit.cxx
--- desktop/source/app/ssoinit.cxx	17 Apr 2003 13:33:29 -0000	1.17
+++ desktop/source/app/ssoinit.cxx	12 Aug 2003 10:25:29 -0000
@@ -220,6 +219,10 @@ void WriteUserNameToINI( const OUString 
 
 LanguageType GetLanguageType( void )
 {
+#ifdef UNX
+	return GetSystemUILanguage (INTERNATIONAL_SYSTEM_DEFAULT);
+#endif
+
 	LanguageType						theLanguageType		= LANGUAGE_DONTKNOW;
 	try
 	{
Index: unotools/source/config/configmgr.cxx
===================================================================
RCS file: /cvs/util/unotools/source/config/configmgr.cxx,v
retrieving revision 1.36
diff -u -p -u -r1.36 configmgr.cxx
--- unotools/source/config/configmgr.cxx	10 Jun 2003 11:29:10 -0000	1.36
+++ unotools/source/config/configmgr.cxx	12 Aug 2003 10:27:35 -0000
@@ -89,6 +92,12 @@
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
 #endif
+#ifndef _OSL_PROCESS_H_
+#include <osl/process.h>
+#endif
+#ifndef _RTL_LOCALE_HXX_
+#include <rtl/locale.hxx>
+#endif
 
 #include <list>
 
@@ -438,6 +447,20 @@ Any ConfigManager::GetDirectConfigProper
         aRet <<= aXMLFileFormatVersion;
         return aRet;
     }
+
+#ifdef UNX
+    if ( (eProp == LOCALE || eProp == DEFAULTCURRENCY) )
+    {
+	rtl_Locale *pRtlLocale = NULL;
+	osl_getProcessLocale( &pRtlLocale );
+	OLocale aLocale( pRtlLocale );
+	OUString aSysLang = ( aLocale.getLanguage() + 
+			      OUString::createFromAscii( "-" ) +
+			      aLocale.getCountry() );
+        aRet <<= aSysLang;
+	return aRet;
+    }
+#endif
 
 	OUString sPath = C2U(cConfigBaseURL);
 	switch(eProp)
Index: stoc/source/javavm/javavm.cxx
===================================================================
RCS file: /cvs/udk/stoc/source/javavm/javavm.cxx,v
retrieving revision 1.59
diff -u -p -u -r1.59 javavm.cxx
--- stoc/source/javavm/javavm.cxx	12 Jun 2003 11:08:36 -0000	1.59
+++ stoc/source/javavm/javavm.cxx	12 Aug 2003 10:29:54 -0000
@@ -137,6 +137,8 @@ int main( int argc, char * argv[])
 #include "jvmaccess/virtualmachine.hxx"
 #include "osl/file.hxx"
 #include "osl/thread.h"
+#include "osl/process.h"
+#include "rtl/locale.hxx"
 #include "rtl/bootstrap.hxx"
 #include "rtl/process.h"
 #include "rtl/string.h"
@@ -437,6 +439,14 @@ void getDefaultLocaleFromConfig(stoc_jav
                                 const css::uno::Reference<css::lang::XMultiComponentFactory> & xSMgr,
                                 const css::uno::Reference<css::uno::XComponentContext> &xCtx ) throw(css::uno::Exception)
 {
+#ifdef UNX
+	rtl_Locale *pRtlLocale = NULL;
+	osl_getProcessLocale( &pRtlLocale );
+	rtl::OLocale aOLocale( pRtlLocale );
+	rtl::OUString aLocale = ( aOLocale.getLanguage() + 
+				  rtl::OUString::createFromAscii( "-" ) +
+				  aOLocale.getCountry() );
+#else
 	css::uno::Reference<css::uno::XInterface> xConfRegistry = xSMgr->createInstanceWithContext(
 		rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("com.sun.star.configuration.ConfigurationRegistry")),
 		xCtx );
@@ -450,15 +460,22 @@ void getDefaultLocaleFromConfig(stoc_jav
 
 	// read locale
 	css::uno::Reference<css::registry::XRegistryKey> locale = xRegistryRootKey->openKey(rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("L10N/ooLocale")));
-	if(locale.is() && locale->getStringValue().getLength()) {
+	rtl::OUString aLocale;
+
+	if(locale.is() && locale->getStringValue().getLength())
+		 aLocale = locale->getStringValue();
+
+	xConfRegistry_simple->close();
+#endif
+	if (aLocale.getLength()) {
 		rtl::OUString language;
 		rtl::OUString country;
 
-		sal_Int32 index = locale->getStringValue().indexOf((sal_Unicode) '-');
+		sal_Int32 index = aLocale.indexOf((sal_Unicode) '-');
 
 		if(index >= 0) {
-			language = locale->getStringValue().copy(0, index);
-			country = locale->getStringValue().copy(index + 1);
+			language = aLocale.copy(0, index);
+			country = aLocale.copy(index + 1);
 
 			if(language.getLength()) {
 				rtl::OUString prop(RTL_CONSTASCII_USTRINGPARAM("user.language="));
@@ -475,8 +492,6 @@ void getDefaultLocaleFromConfig(stoc_jav
 			}
 		}
 	}
-
-	xConfRegistry_simple->close();
 }
 
 void getJavaPropsFromConfig(stoc_javavm::JVM * pjvm,
