Index: unotools/source/config/configmgr.cxx
===================================================================
RCS file: /cvs/util/unotools/source/config/configmgr.cxx,v
retrieving revision 1.36
diff -u -p -u -r1.36 configmgr.cxx
--- unotools/source/config/configmgr.cxx	10 Jun 2003 11:29:10 -0000	1.36
+++ unotools/source/config/configmgr.cxx	11 Aug 2003 15:10:23 -0000
@@ -59,6 +59,9 @@
  *
  ************************************************************************/
 
+#include <tools/isolang.hxx>
+#include <tools/intn.hxx>
+
 #ifndef _UTL_CONFIGMGR_HXX_
 #include "unotools/configmgr.hxx"
 #endif
@@ -437,6 +440,13 @@ Any ConfigManager::GetDirectConfigProper
     {
         aRet <<= aXMLFileFormatVersion;
         return aRet;
+    }
+
+    if ( !getenv ("OOOCONFIG_LOCALE") && (eProp == LOCALE || eProp == DEFAULTCURRENCY))
+    {
+	OUString aSysLang = ConvertLanguageToIsoString(GetSystemLanguage());
+        aRet <<= aSysLang;
+	return aRet;
     }
 
 	OUString sPath = C2U(cConfigBaseURL);
Index: desktop/source/app/ssoinit.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/ssoinit.cxx,v
retrieving revision 1.17
diff -u -p -u -r1.17 ssoinit.cxx
--- desktop/source/app/ssoinit.cxx	17 Apr 2003 13:33:29 -0000	1.17
+++ desktop/source/app/ssoinit.cxx	11 Aug 2003 15:11:11 -0000
@@ -220,6 +219,11 @@ void WriteUserNameToINI( const OUString 
 
 LanguageType GetLanguageType( void )
 {
+#ifdef UNX
+	if ( !getenv ("OOOCONFIG_LOCALE") )
+		return GetSystemUILanguage (INTERNATIONAL_SYSTEM_DEFAULT);
+#endif
+
 	LanguageType						theLanguageType		= LANGUAGE_DONTKNOW;
 	try
 	{
Index: stoc/source/javavm/javavm.cxx
===================================================================
RCS file: /cvs/udk/stoc/source/javavm/javavm.cxx,v
retrieving revision 1.59
diff -u -p -u -r1.59 javavm.cxx
--- stoc/source/javavm/javavm.cxx	12 Jun 2003 11:08:36 -0000	1.59
+++ stoc/source/javavm/javavm.cxx	11 Aug 2003 15:31:49 -0000
@@ -148,6 +148,12 @@ int main( int argc, char * argv[])
 #include "uno/environment.h"
 #include "uno/lbnames.h"
 
+#ifdef UNX_AND_WE_CAN_USE_THE_CRUCIAL_TOOLS
+#  include <tools/isolang.hxx>
+#  include <tools/intn.hxx>
+#  include <tools/string.hxx>
+#endif
+
 #include "jni.h"
 
 #include <setjmp.h>
@@ -437,6 +443,9 @@ void getDefaultLocaleFromConfig(stoc_jav
                                 const css::uno::Reference<css::lang::XMultiComponentFactory> & xSMgr,
                                 const css::uno::Reference<css::uno::XComponentContext> &xCtx ) throw(css::uno::Exception)
 {
+#ifdef UNX_AND_WE_CAN_USE_THE_CRUCIAL_TOOLS
+	rtl::OUString aLocale = ConvertLanguageToIsoString(GetSystemLanguage());
+#else
 	css::uno::Reference<css::uno::XInterface> xConfRegistry = xSMgr->createInstanceWithContext(
 		rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("com.sun.star.configuration.ConfigurationRegistry")),
 		xCtx );
@@ -450,15 +459,22 @@ void getDefaultLocaleFromConfig(stoc_jav
 
 	// read locale
 	css::uno::Reference<css::registry::XRegistryKey> locale = xRegistryRootKey->openKey(rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("L10N/ooLocale")));
-	if(locale.is() && locale->getStringValue().getLength()) {
+	rtl::OUString aLocale;
+
+	if(locale.is() && locale->getStringValue().getLength())
+		 aLocale = locale->getStringValue();
+
+	xConfRegistry_simple->close();
+#endif
+	if (aLocale.getLength()) {
 		rtl::OUString language;
 		rtl::OUString country;
 
-		sal_Int32 index = locale->getStringValue().indexOf((sal_Unicode) '-');
+		sal_Int32 index = aLocale.indexOf((sal_Unicode) '-');
 
 		if(index >= 0) {
-			language = locale->getStringValue().copy(0, index);
-			country = locale->getStringValue().copy(index + 1);
+			language = aLocale.copy(0, index);
+			country = aLocale.copy(index + 1);
 
 			if(language.getLength()) {
 				rtl::OUString prop(RTL_CONSTASCII_USTRINGPARAM("user.language="));
@@ -475,8 +491,6 @@ void getDefaultLocaleFromConfig(stoc_jav
 			}
 		}
 	}
-
-	xConfRegistry_simple->close();
 }
 
 void getJavaPropsFromConfig(stoc_javavm::JVM * pjvm,
Index: xmlhelp/source/treeview/tvread.cxx
===================================================================
RCS file: /cvs/util/xmlhelp/source/treeview/tvread.cxx,v
retrieving revision 1.13
diff -u -p -u -r1.13 tvread.cxx
--- xmlhelp/source/treeview/tvread.cxx	8 Nov 2001 16:42:46 -0000	1.13
+++ xmlhelp/source/treeview/tvread.cxx	11 Aug 2003 15:45:40 -0000
@@ -24,6 +24,12 @@
 #include <com/sun/star/beans/PropertyState.hpp>
 #endif
 
+#ifdef UNX_AND_WE_CAN_USE_THE_CRUCIAL_TOOLS
+#  include <tools/isolang.hxx>
+#  include <tools/intn.hxx>
+#  include <tools/string.hxx>
+#endif
+
 namespace treeview {
 
 	
@@ -684,7 +690,11 @@ ConfigData TVChildTarget::init( const Re
 	rtl::OUString productVersion( setupversion + 
 								  rtl::OUString::createFromAscii( " " ) +
 								  setupextension );
+#ifdef UNX_AND_WE_CAN_USE_THE_CRUCIAL_TOOLS
+	rtl::OUString locale = ConvertLanguageToIsoString(GetSystemLanguage());
+#else
 	rtl::OUString locale( getKey( xHierAccess,"L10N/ooLocale" ) );
+#endif
 
 	
 	// Determine fileurl from url and locale
Index: stoc/source/javavm/javavm.cxx
===================================================================
RCS file: /cvs/udk/stoc/source/javavm/javavm.cxx,v
retrieving revision 1.59
diff -u -p -u -r1.59 javavm.cxx
--- stoc/source/javavm/javavm.cxx	12 Jun 2003 11:08:36 -0000	1.59
+++ stoc/source/javavm/javavm.cxx	11 Aug 2003 15:46:16 -0000
@@ -148,6 +148,12 @@ int main( int argc, char * argv[])
 #include "uno/environment.h"
 #include "uno/lbnames.h"
 
+#ifdef UNX_AND_WE_CAN_USE_THE_CRUCIAL_TOOLS
+#  include <tools/isolang.hxx>
+#  include <tools/intn.hxx>
+#  include <tools/string.hxx>
+#endif
+
 #include "jni.h"
 
 #include <setjmp.h>
@@ -437,6 +443,9 @@ void getDefaultLocaleFromConfig(stoc_jav
                                 const css::uno::Reference<css::lang::XMultiComponentFactory> & xSMgr,
                                 const css::uno::Reference<css::uno::XComponentContext> &xCtx ) throw(css::uno::Exception)
 {
+#ifdef UNX_AND_WE_CAN_USE_THE_CRUCIAL_TOOLS
+	rtl::OUString aLocale = ConvertLanguageToIsoString(GetSystemLanguage());
+#else
 	css::uno::Reference<css::uno::XInterface> xConfRegistry = xSMgr->createInstanceWithContext(
 		rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("com.sun.star.configuration.ConfigurationRegistry")),
 		xCtx );
@@ -450,15 +459,22 @@ void getDefaultLocaleFromConfig(stoc_jav
 
 	// read locale
 	css::uno::Reference<css::registry::XRegistryKey> locale = xRegistryRootKey->openKey(rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("L10N/ooLocale")));
-	if(locale.is() && locale->getStringValue().getLength()) {
+	rtl::OUString aLocale;
+
+	if(locale.is() && locale->getStringValue().getLength())
+		 aLocale = locale->getStringValue();
+
+	xConfRegistry_simple->close();
+#endif
+	if (aLocale.getLength()) {
 		rtl::OUString language;
 		rtl::OUString country;
 
-		sal_Int32 index = locale->getStringValue().indexOf((sal_Unicode) '-');
+		sal_Int32 index = aLocale.indexOf((sal_Unicode) '-');
 
 		if(index >= 0) {
-			language = locale->getStringValue().copy(0, index);
-			country = locale->getStringValue().copy(index + 1);
+			language = aLocale.copy(0, index);
+			country = aLocale.copy(index + 1);
 
 			if(language.getLength()) {
 				rtl::OUString prop(RTL_CONSTASCII_USTRINGPARAM("user.language="));
@@ -475,8 +491,6 @@ void getDefaultLocaleFromConfig(stoc_jav
 			}
 		}
 	}
-
-	xConfRegistry_simple->close();
 }
 
 void getJavaPropsFromConfig(stoc_javavm::JVM * pjvm,
