Index: vcl/source/gdi/bitmapex.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/bitmapex.cxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 bitmapex.cxx
--- vcl/source/gdi/bitmapex.cxx	11 Apr 2003 17:28:27 -0000	1.6
+++ vcl/source/gdi/bitmapex.cxx	15 Sep 2003 10:19:39 -0000
@@ -113,6 +147,12 @@ BitmapEx::BitmapEx( const Bitmap& rBmp )
 		eTransparent( TRANSPARENT_NONE ),
 		bAlpha		( FALSE )
 {
+    if (aBitmap.GetBitCount () == 32)
+	{
+        eTransparent = TRANSPARENT_BITMAP;
+		bAlpha = TRUE;
+		aMask = aBitmap.GetAlphaMask();
+	}
 }
 
 // ------------------------------------------------------------------
@@ -124,12 +164,8 @@ BitmapEx::BitmapEx( const Bitmap& rBmp, 
 		eTransparent	( !rMask ? TRANSPARENT_NONE : TRANSPARENT_BITMAP ),
 		bAlpha			( FALSE )
 {
-    // #105489# Ensure a mask is exactly one bit deep
     if( !!aMask && aMask.GetBitCount() != 1 )
-    {
-        OSL_TRACE("BitmapEx: forced mask to monochrome");
-        aMask.ImplMakeMono( 255 );
-    }
+		bAlpha = TRUE;
 }
 
 // ------------------------------------------------------------------
@@ -152,7 +188,13 @@ BitmapEx::BitmapEx( const Bitmap& rBmp, 
 		eTransparent		( TRANSPARENT_BITMAP ),
 		bAlpha				( FALSE )
 {
-	aMask = aBitmap.CreateMask( aTransparentColor );
+    if( rBmp.GetBitCount() > 24 )
+	{
+        aMask = rBmp.GetAlphaMask();
+		bAlpha = TRUE;
+	}
+	else
+	    aMask = aBitmap.CreateMask( aTransparentColor );
 }
 
 // ------------------------------------------------------------------
@@ -287,12 +329,7 @@ BitmapEx BitmapEx::GetColorTransformedBi
 
 Bitmap BitmapEx::GetMask() const
 {
-	Bitmap aRet( aMask );
-
-	if( IsAlpha() )
-		aRet.ImplMakeMono( 255 );
-
-	return aRet;
+	return Bitmap( aMask );
 }
 
 // ------------------------------------------------------------------
@@ -395,7 +433,7 @@ BOOL BitmapEx::Scale( const double& rSca
                bRet = aBitmap.Scale( rScaleX, rScaleY, nScaleFlag );
 
                if( bRet && ( eTransparent == TRANSPARENT_BITMAP ) && !!aMask )
-                       aMask.Scale( rScaleX, rScaleY, BMP_SCALE_FAST );
+                       aMask.Scale( rScaleX, rScaleY, aMask.GetBitCount() == 1 ? BMP_SCALE_FAST : nScaleFlag);
 
                aBitmapSize = aBitmap.GetSizePixel();
        }
