--- vcl/unx/inc/XIM.h	18 Sep 2000 17:05:40 -0000	1.1.1.1
+++ vcl/unx/inc/XIM.h	24 Feb 2004 02:38:04 -0000
@@ -155,6 +155,8 @@
 /* XIM attributes for multilingual IM extension */
 #define XNMultiLingualInput "multiLingualInput"
 #define XNQueryUnicodeCharacterSubset "unicodeCharacterSubset"
+#define XNQueryExtensionVersion "queryExtensionersion"
+#define XIIIMP_MULTILINGUAL_EXTENSION_VERSION 2
 
 /* XIC attributes for multilingual IM extension */
 
--- vcl/unx/source/app/i18n_ic.cxx	2 Jul 2003 13:40:36 -0000	1.26
+++ vcl/unx/source/app/i18n_ic.cxx	24 Feb 2004 02:38:04 -0000
@@ -200,11 +200,9 @@
                                 ),
         mpFocusFrame( NULL )
 {
-#ifdef SOLARIS
     static const char* pIIIMPEnable = getenv( "SAL_DISABLE_OWN_IM_STATUS" );
     if( pIIIMPEnable && *pIIIMPEnable )
         mnSupportedStatusStyle &= ~XIMStatusCallbacks;
-#endif
 
     maClientData.aText.pUnicodeBuffer = NULL;
     maClientData.aText.pCharStyle     = NULL;
--- vcl/unx/source/app/i18n_im.cxx	28 Jan 2004 15:23:17 -0000	1.23.110.2
+++ vcl/unx/source/app/i18n_im.cxx	24 Feb 2004 02:38:04 -0000
@@ -438,11 +438,16 @@
 			// get ml-input flag from input-method
 			if ( maMethod == (XIM)NULL )
 				mbMultiLingual = False;
-#if !defined(LINUX)
 			else
-			if ( XGetIMValues(maMethod,
-					XNMultiLingualInput, &mbMultiLingual, NULL ) != NULL )
+			{
+			    int xiiimp_version = 1;
+			    if ( XGetIMValues(maMethod,
+					      XNQueryExtensionVersion, &xiiimp_version, NULL) != NULL || xiiimp_version < XIIIMP_MULTILINGUAL_EXTENSION_VERSION)
 				mbMultiLingual = False;
+			    if ( XGetIMValues(maMethod,
+					      XNMultiLingualInput, &mbMultiLingual, NULL ) != NULL )
+				mbMultiLingual = False;
+			}
             if( mbMultiLingual )
             {
                 XIMUnicodeCharacterSubsets* subsets;
@@ -470,9 +475,6 @@
                     fprintf( stderr, "query subsets failed\n" );
 #endif
             }
-#else
-            mbMultiLingual = False;
-#endif
 		}
 		else
 	    {
