Index: scp/source/javafilter/module_javafilter.scp
===================================================================
RCS file: /cvs/installation/scp/source/javafilter/Attic/module_javafilter.scp,v
retrieving revision 1.16.10.1
diff -u -p -u -r1.16.10.1 module_javafilter.scp
--- scp/source/javafilter/module_javafilter.scp	15 Jan 2004 12:31:12 -0000	1.16.10.1
+++ scp/source/javafilter/module_javafilter.scp	28 Apr 2004 15:39:30 -0000
@@ -13,7 +13,9 @@
 Module GID_MODULE_OPTIONAL_XSLTFILTERSAMPLES
         MOD_NAME_DESC(MODULE_OPTIONAL_XSLTFILTERSAMPLES);
         ParentID = GID_MODULE_OPTIONAL;
-        Files = ( GID_FILE_XSL_SOFFTODOCBOOKHEADINGS,
+        Files = (
+#ifdef SOLAR_JAVA
+			GID_FILE_XSL_SOFFTODOCBOOKHEADINGS,
 			GID_FILE_XSL_DOCBOOKTOSOFFHEADINGS,
 			GID_FILE_STW_DOCBOOKTEMPLATE,
 			GID_FILE_XSL_COMMON,
@@ -31,10 +33,10 @@ Module GID_MODULE_OPTIONAL_XSLTFILTERSAM
  			GID_FILE_XSL_TABLE_WML,
 			GID_FILE_XSL_MSWORDXMLIMPORT,
 			GID_FILE_XSL_MSEXCELXMLIMPORT
-			
+#endif
  );
         Minimal = NO;
-        Default = NO;
+        Default = YES;
         Styles = ( );
 End
 
