diff -Nurd jvmaccess/source.orig/javainfoimpl.cxx jvmaccess/source/javainfoimpl.cxx
--- jvmaccess/source.orig/javainfoimpl.cxx	2004-01-28 09:19:00.000000000 -0200
+++ jvmaccess/source/javainfoimpl.cxx	2003-05-31 23:18:00.000000000 -0300
@@ -211,7 +211,10 @@
                         "j2sdk", \
                         "jdk", \
                         "jre", \
-                        "java"
+                        "java", \
+			"1.4", \
+			"1.3"
+
 /* These are directories in which a java installation is
    looked for.
 */
@@ -227,7 +230,8 @@
                          "jdk/", \
                          "jre/", \
                          "j2sdk/", \
-                         "j2re/"
+                         "j2re/", \
+                         "j2se/"
 
 
 #endif // WNT, UNX
