--- svtools/source/contnr/svimpbox.cxx	Fri Dec  7 09:40:39 2001
+++ svtools/source/contnr/svimpbox.cxx	Wed Jan  8 16:40:55 2003
@@ -676,6 +676,7 @@
 		return TRUE;
 
 	Rectangle aRect( pView->GetFocusRect( pEntry, nLine ));
+	aRect.Right () = GetOutputSize().Width() - pView->GetMapMode().GetOrigin().X();
 	if( pView->IsA() == SV_LISTBOX_ID_TREEBOX )
 	{
 		SvLBoxContextBmp* pBmp = (SvLBoxContextBmp*)(pEntry->GetFirstItem(SV_ITEM_ID_LBOXCONTEXTBMP));
