--- desktop/scripts/nswrapper.sh
+++ desktop/scripts/nswrapper.sh
@@ -1,10 +1,10 @@
 #!/bin/sh
 
-# if netscape is not found, specify full path here
-NETSCAPE=netscape
+# if mozilla is not found, specify full path here
+MOZILLA=mozilla
 
-if ${NETSCAPE} -remote "openURL($1,new-window)" 2>&1 | egrep -si "not running on display"; then
-	${NETSCAPE} $1
+if ${MOZILLA} -remote "openURL($1,new-window)" 2>&1 | egrep -si "not running on display"; then
+	${MOZILLA} $1
 fi
 
 
