--- odk/source/OOSupport/org/netbeans/modules/openoffice/wizard/OOIDLWizardIterator.java.javafix	2003-11-18 11:03:25.000000000 -0500
+++ odk/source/OOSupport/org/netbeans/modules/openoffice/wizard/OOIDLWizardIterator.java	2003-11-18 11:03:59.000000000 -0500
@@ -207,7 +207,7 @@
    index--;
   }
 
-  public org.openide.WizardDescriptor$Panel current()
+  public org.openide.WizardDescriptor.Panel current()
    {
      return panels[index];
    }
--- odk/source/OOSupport/org/netbeans/modules/openoffice/wizard/OOServiceCreateIterator.java.javafix	2003-11-18 11:04:51.000000000 -0500
+++ odk/source/OOSupport/org/netbeans/modules/openoffice/wizard/OOServiceCreateIterator.java	2003-11-18 11:05:28.000000000 -0500
@@ -214,7 +214,7 @@
         index--;
     }
     
-    public org.openide.WizardDescriptor$Panel current() {
+    public org.openide.WizardDescriptor.Panel current() {
         return panels[index];
     }
     
--- odk/source/OOSupport/org/netbeans/modules/openoffice/wizard/OOAddInCreateIterator.java.javafix	2003-11-18 11:06:09.000000000 -0500
+++ odk/source/OOSupport/org/netbeans/modules/openoffice/wizard/OOAddInCreateIterator.java	2003-11-18 11:06:26.000000000 -0500
@@ -200,7 +200,7 @@
         index--;
     }
     
-    public org.openide.WizardDescriptor$Panel current() {
+    public org.openide.WizardDescriptor.Panel current() {
         return panels[index];
     }
     
