Index: svtools/source/dialogs/printdlg.cxx
===================================================================
--- svtools/source/dialogs/printdlg.cxx.orig	2003-11-25 15:07:08.000000000 +0100
+++ svtools/source/dialogs/printdlg.cxx	2004-01-08 00:13:50.000000000 +0100
@@ -429,8 +429,13 @@ bool PrintDialog::ImplGetFilename()
         Reference< XFilterManager > xFilterMgr( xFilePicker, UNO_QUERY );
         if( xFilePicker.is() && xFilterMgr.is() )
         {
+	    	INetURLObject aLocation( maFiPrintFile.GetText(), INET_PROT_FILE );
+
+	    	xFilePicker->setDefaultName( aLocation.GetLastName() );
+	    	xFilePicker->setDisplayDirectory ( aLocation.GetPartBeforeLastName() );
+
             try
             {
 #ifdef UNX
                 // add PostScript and PDF
                 if( ! Application::IsRemoteServer() )
@@ -449,6 +454,10 @@ bool PrintDialog::ImplGetFilename()
                         xFilterMgr->appendFilter( OUString( RTL_CONSTASCII_USTRINGPARAM( "PostScript" ) ), OUString( RTL_CONSTASCII_USTRINGPARAM( "*.ps" ) ) );
                     if( bPDF )
                         xFilterMgr->appendFilter( OUString( RTL_CONSTASCII_USTRINGPARAM( "Portable Document Format" ) ), OUString( RTL_CONSTASCII_USTRINGPARAM( "*.pdf" ) ) );
+
+		    String aFilter = pPrinter->GetJobValue( String::CreateFromAscii( "Filter#" ) );
+		    if (aFilter.Len())
+			    xFilterMgr->setCurrentFilter( aFilter );
                 }
 #elif defined WNT
                 xFilterMgr->appendFilter( OUString( RTL_CONSTASCII_USTRINGPARAM( "*.PRN" ) ), OUString( RTL_CONSTASCII_USTRINGPARAM( "*.prn" ) ) );
@@ -465,6 +474,13 @@ bool PrintDialog::ImplGetFilename()
                 Sequence< OUString > aPathSeq( xFilePicker->getFiles() );
                 INetURLObject aObj( aPathSeq[0] );
                 maFiPrintFile.SetText( aObj.PathToFileName() );
+
+#ifdef UNX
+				Printer* pPrinter = TEMPPRINTER() ? TEMPPRINTER() : mpPrinter;
+				if ( pPrinter )
+					pPrinter->SetJobValue( String::CreateFromAscii( "Filter#" ),
+										   xFilterMgr->getCurrentFilter() );
+#endif
                 return true;
             }
             return false;
