Index: sc/sdi/cellsh.sdi
===================================================================
RCS file: /cvs/sc/sc/sdi/cellsh.sdi,v
retrieving revision 1.11
diff -u -r1.11 cellsh.sdi
--- sc/sdi/cellsh.sdi	28 Nov 2002 14:56:06 -0000	1.11
+++ sc/sdi/cellsh.sdi	17 Aug 2004 15:15:42 -0000
@@ -138,6 +138,7 @@
 	SID_GET_CLPBRD_FORMAT_NAME		[ ExecMethod = ExecuteEdit; ]
 	SID_EXTERNAL_SOURCE	[ ExecMethod = ExecuteEdit; StateMethod = GetBlockState; ]
 	FID_MERGE_ON	[ ExecMethod = Execute; StateMethod = GetState; ]
+	FID_MERGE_CENTER	[ ExecMethod = Execute; StateMethod = GetState; ]
 	FID_MERGE_OFF	[ ExecMethod = Execute; StateMethod = GetState; ]
 	SID_OUTLINE_HIDE	[ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
 	SID_OUTLINE_SHOW	[ ExecMethod = ExecuteEdit; StateMethod = GetState; ]
Index: sc/sdi/scalc.sdi
===================================================================
RCS file: /cvs/sc/sc/sdi/scalc.sdi,v
retrieving revision 1.24
diff -u -r1.24 scalc.sdi
--- sc/sdi/scalc.sdi	28 Nov 2002 14:56:05 -0000	1.24
+++ sc/sdi/scalc.sdi	17 Aug 2004 15:15:42 -0000
@@ -4084,6 +4084,31 @@
 ]
 
 //--------------------------------------------------------------------------
+SfxVoidItem MergeAndCenter FID_MERGE_CENTER
+(SfxBoolItem MoveContents FID_MERGE_CENTER)
+[
+	/* flags: */
+	AutoUpdate = FALSE,
+	Cachable = Cachable,
+	FastCall = FALSE,
+	HasCoreId = FALSE,
+	HasDialog = FALSE,
+	ReadOnlyDoc = TRUE,
+	Toggle = FALSE,
+	Container = FALSE,
+	RecordAbsolute = FALSE,
+	RecordPerSet;
+	Synchron;
+
+	/* config: */
+	AccelConfig = TRUE,
+	MenuConfig = TRUE,
+	StatusBarConfig = FALSE,
+	ToolBoxConfig = TRUE,
+	GroupId = GID_FORMAT;
+]
+
+//--------------------------------------------------------------------------
 SfxBoolItem Mirror SID_OBJECT_MIRROR
 ()
 [
Index: sc/sdi/scslots.src
===================================================================
RCS file: /cvs/sc/sc/sdi/scslots.src,v
retrieving revision 1.59.28.5
diff -u -r1.59.28.5 scslots.src
--- sc/sdi/scslots.src	27 Apr 2004 16:54:55 -0000	1.59.28.5
+++ sc/sdi/scslots.src	17 Aug 2004 15:15:48 -0000
@@ -1866,6 +1866,12 @@
 	SlotName[ zulu ] = "Hlanganisa Amagumbi";
 	SlotName[ norwegian_nynorsk ] = "Slå saman celler";
 };
+SfxSlotInfo FID_MERGE_CENTER
+{
+	SlotName = "Merge and Center";
+	SlotName [ english ] =  "Merge and Center";
+	SlotName [ english_us ] =  "Merge and Center";
+};
 SfxSlotInfo FID_NOTE_VISIBLE
 {
 	SlotName = "Notiz anzeigen";
