Index: sc/source/core/tool/interpr4.cxx
===================================================================
RCS file: /cvs/sc/sc/source/core/tool/interpr4.cxx,v
retrieving revision 1.26
diff -u -p -u -r1.26 interpr4.cxx
--- sc/source/core/tool/interpr4.cxx	28 Apr 2003 15:32:01 -0000	1.26
+++ sc/source/core/tool/interpr4.cxx	26 Feb 2004 19:15:09 -0000
@@ -1248,6 +1248,13 @@ StackVar ScInterpreter::GetStackType( BY
 
 BOOL ScInterpreter::DoubleRefToPosSingleRef( const ScRange& rRange, ScAddress& rAdr )
 {
+	// Check for a singleton first - no implicit intersection for them.
+	if( rRange.aStart == rRange.aEnd )
+	{
+		rAdr = rRange.aStart;
+		return TRUE;
+	}
+
 	USHORT nMyCol = aPos.Col();
 	USHORT nMyRow = aPos.Row();
 	USHORT nMyTab = aPos.Tab();
