--- sc/source/ui/inc/viewfunc.hxx
+++ sc/source/ui/inc/viewfunc.hxx
@@ -243,7 +243,8 @@
 										ScMarkData* pMarkData = NULL );
 	void			SetMarkedWidthOrHeight( BOOL bWidth, ScSizeMode eMode, USHORT nSizeTwips,
 										BOOL bRecord = TRUE, BOOL bPaint = TRUE );
+	BOOL                    GetRowColHidden(BOOL col);
 	void			ShowMarkedColumns( BOOL bShow, BOOL bRecord = TRUE );
 	void			ShowMarkedRows( BOOL bShow, BOOL bRecord = TRUE );
 

--- sc/source/core/data/document.cxx	2003-11-22 15:35:08.000000000 +0530
+++ sc/source/core/data/document.cxx	2004-01-06 18:30:55.000000000 +0530
@@ -475,6 +475,28 @@
 	return FALSE;
 }
 
+BOOL ScDocument::IsRowHidden(USHORT nStartRow, USHORT nEndRow, USHORT nTab) const
+{
+        BOOL	rHidden = FALSE;
+        for(USHORT i=nStartRow; i<=nEndRow; i++)
+	  {
+		  if( RowHidden(i,nTab) )
+			  rHidden = TRUE;
+	  }
+	return rHidden;
+}
+
+BOOL ScDocument::IsColHidden(USHORT nStartCol, USHORT nEndCol, USHORT nTab) const
+{
+        BOOL	cHidden = FALSE;
+        for(USHORT i=nStartCol; i<=nEndCol; i++)
+	  {
+		  if(  ColHidden(i,nTab) )
+			  cHidden = TRUE;
+	  }
+	return cHidden;
+}
+
 /* ----------------------------------------------------------------------------
 	benutzten Bereich suchen:
 
Index: sc/source/ui/view/cellsh.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/cellsh.cxx,v
retrieving revision 1.20
diff -u -p -u -r1.20 cellsh.cxx
--- sc/source/ui/view/cellsh.cxx	8 Apr 2003 16:32:11 -0000	1.20
+++ sc/source/ui/view/cellsh.cxx	6 Jan 2004 14:56:52 -0000
@@ -775,13 +775,20 @@ void ScCellShell::GetState(SfxItemSet &r
 				break;
 
 			case FID_ROW_HIDE:
-			case FID_ROW_SHOW:
 			case FID_COL_HIDE:
-			case FID_COL_SHOW:
 			case FID_COL_OPT_WIDTH:
 			case FID_ROW_OPT_HEIGHT:
 			case FID_DELETE_CELL:
 				if ( pDoc->IsTabProtected(nTab) || pDocSh->IsReadOnly())
+					rSet.DisableItem( nWhich );
+				break;
+
+			case FID_ROW_SHOW:
+				if( pDoc->IsTabProtected(nTab) || pDocSh->IsReadOnly() || !pTabViewShell->GetRowColHidden(FALSE) ) 
+					rSet.DisableItem( nWhich );
+				break;
+			case FID_COL_SHOW:
+				if( pDoc->IsTabProtected(nTab) || pDocSh->IsReadOnly() || !pTabViewShell->GetRowColHidden(TRUE) ) 
 					rSet.DisableItem( nWhich );
 				break;
 
Index: sc/inc/document.hxx
===================================================================
RCS file: /cvs/sc/sc/inc/document.hxx,v
retrieving revision 1.67
diff -u -p -u -r1.67 document.hxx
--- sc/inc/document.hxx	28 Apr 2003 15:30:35 -0000	1.67
+++ sc/inc/document.hxx	6 Jan 2004 22:11:41 -0000
@@ -494,6 +494,7 @@ private:
 
 
 	inline BOOL 		RowHidden( USHORT nRow, USHORT nTab );		// FillInfo
+    inline BOOL         ColHidden( USHORT nCol, USHORT nTab );          // FillInfo
 
 public:
 	long			GetCellCount() const;		// alle Zellen
@@ -655,6 +656,8 @@ public:
 	void			ClearDrawPage(USHORT nTab);
 	void			SetVisible( USHORT nTab, BOOL bVisible );
 	BOOL			IsVisible( USHORT nTab ) const;
+        BOOL                    IsRowHidden(USHORT nStartRow, USHORT nEndRow, USHORT nTab) const;
+        BOOL                    IsColHidden(USHORT nStartCol, USHORT nEndCol, USHORT nTab) const;
 	void			SetScenario( USHORT nTab, BOOL bFlag );
 	BOOL			IsScenario( USHORT nTab ) const;
 	void			GetScenarioData( USHORT nTab, String& rComment,
@@ -1673,6 +1676,16 @@ private: // CLOOK-Impl-Methoden
 inline USHORT ScDocument::FastGetRowHeight( USHORT nRow, USHORT nTab ) const
 {
 	return ( pTab[nTab]->pRowFlags[nRow] & CR_HIDDEN ) ? 0 : pTab[nTab]->pRowHeight[nRow];
+}
+
+inline BOOL ScDocument::ColHidden( USHORT nCol, USHORT nTab )
+{
+	return ( pTab[nTab]->pColFlags[nCol] & CR_HIDDEN ) != 0;
+}
+
+inline BOOL ScDocument::RowHidden( USHORT nRow, USHORT nTab )
+{
+	return ( pTab[nTab]->pRowFlags[nRow] & CR_HIDDEN ) != 0;
 }
 
 
Index: sc/source/core/data/fillinfo.cxx
===================================================================
RCS file: /cvs/sc/sc/source/core/data/fillinfo.cxx,v
retrieving revision 1.3
diff -u -p -u -r1.3 fillinfo.cxx
--- sc/source/core/data/fillinfo.cxx	5 Feb 2001 11:32:30 -0000	1.3
+++ sc/source/core/data/fillinfo.cxx	6 Jan 2004 22:14:46 -0000
@@ -184,12 +184,6 @@ void lcl_GetMergeRange( short nX, short 
 	rEndY = rStartY + pMerge->GetRowMerge() - 1;
 }
 
-inline BOOL ScDocument::RowHidden( USHORT nRow, USHORT nTab )
-{
-	return ( pTab[nTab]->pRowFlags[nRow] & CR_HIDDEN ) != 0;
-}
-
-
 #define CELLINFO(x,y) pRowInfo[nArrY+y].pCellInfo[nArrX+x]
 
 USHORT ScDocument::FillInfo( RowInfo* pRowInfo, USHORT nX1, USHORT nY1, USHORT nX2, USHORT nY2,
Index: sc/source/ui/view/viewfunc.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/viewfunc.cxx,v
retrieving revision 1.19
diff -u -p -u -p -r1.19 viewfunc.cxx
--- sc/source/ui/view/viewfunc.cxx	8 Apr 2003 16:34:56 -0000	1.19
+++ sc/source/ui/view/viewfunc.cxx	6 Feb 2004 05:37:49 -0000
@@ -2114,6 +2114,38 @@ void ScViewFunc::SetMarkedWidthOrHeight(
 	rMark.MarkToSimple();
 }
 
+BOOL ScViewFunc::GetRowColHidden(BOOL col)
+{
+	ScDocument* pDoc = GetViewData()->GetDocument();
+	ScMarkData& rMark = GetViewData()->GetMarkData();
+	USHORT nCol = 0,nRow = 0,nTab = 0, nRangeCnt = 0;
+	USHORT* pRanges = new USHORT[MAXROW+1];
+	BOOL hidden = FALSE;
+
+	if ( col )
+		nRangeCnt = rMark.GetMarkColumnRanges( pRanges ); 
+	else
+		nRangeCnt = rMark.GetMarkRowRanges( pRanges );		
+
+	const USHORT* pTabRanges = pRanges;
+
+	for ( USHORT nRangeNo=0; nRangeNo<nRangeCnt; nRangeNo++ )
+	{
+		USHORT nStartNo = *(pTabRanges++);
+		USHORT nEndNo = *(pTabRanges++);
+		if ( col )
+			hidden = pDoc->IsColHidden(nStartNo,nEndNo,nTab);
+		else
+			hidden = pDoc->IsRowHidden(nStartNo,nEndNo,nTab);
+		/* Even if a single col/row is hidden set the var and break */
+		if ( hidden == TRUE ) 
+			break;
+	}
+
+	delete[] pRanges;
+	return hidden;
+}
+
 void ScViewFunc::ModifyCellSize( ScDirection eDir, BOOL bOptimal )
 {
 	//!	Schrittweiten einstellbar
