Index: inc/compiler.hxx
===================================================================
RCS file: /cvs/sc/sc/inc/compiler.hxx,v
retrieving revision 1.15
diff -u -p -u -r1.15 compiler.hxx
--- inc/compiler.hxx	26 Mar 2003 18:03:34 -0000	1.15
+++ inc/compiler.hxx	26 Feb 2004 11:25:03 -0000
@@ -99,7 +99,7 @@
 
 #define errIllegalChar			501
 #define errIllegalArgument		502
-#define	errIllegalFPOperation	503
+#define	errIllegalFPOperation	503		// #DIV/0!
 #define errIllegalParameter		504
 #define errIllegalJump			505
 #define errSeparator			506
@@ -115,13 +115,13 @@
 #define	errUnknownVariable		516
 #define errUnknownOpCode		517
 #define errUnknownStackVariable	518
-#define errNoValue				519
+#define errNoValue				519		// #VALUE!
 #define errUnknownToken			520
 #define errNoCode				521
 #define	errCircularReference	522
 #define errNoConvergence        523
-#define errNoRef				524
-#define errNoName				525
+#define errNoRef				524     // #REF!
+#define errNoName				525     // #NAME?
 #define errDoubleRef			526
 #define errInterpOverflow		527
 // wird nicht angezeigt, temporaer fuer TrackFormulas
Index: source/filter/excel/excform.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/excform.cxx,v
retrieving revision 1.21
diff -u -p -u -r1.21 excform.cxx
--- source/filter/excel/excform.cxx	21 May 2003 07:56:13 -0000	1.21
+++ source/filter/excel/excform.cxx	26 Feb 2004 11:25:04 -0000
@@ -2406,7 +2408,7 @@ const ScTokenArray* ExcelToSc::GetBoolEr
 	{
         case xlErrNull:     eOc = ocStop;       nError = errNoCode;             break;
         case xlErrDiv0:     eOc = ocStop;       nError = errIllegalFPOperation; break;
-        case xlErrValue:    eOc = ocNoValue;    nError = NOVALUE;               break;
+        case xlErrValue:    eOc = ocNoValue;    nError = errNoValue;            break;
         case xlErrRef:      eOc = ocStop;       nError = errNoRef;              break;
         case xlErrName:     eOc = ocStop;       nError = errNoName;             break;
         case xlErrNum:      eOc = ocStop;       nError = errIllegalFPOperation; break;
@@ -2522,7 +2526,7 @@ BOOL ExcelToSc::SetCurVal( ScFormulaCell
 				{// nInd temp. missbrauchen
 					case 0x00:	nInd = errNoCode; break;			// #NULL!
 					case 0x07:	nInd = errIllegalFPOperation; break;// #DIV/0!
-					case 0x0F:	nInd = NOVALUE; break;				// #VALUE!
+					case 0x0F:	nInd = errNoValue; break;			// #VALUE!
 					case 0x17:	nInd = errNoRef; break;				// #REF!
 					case 0x1D:	nInd = errNoName; break;			// #NAME?
 					case 0x24:	nInd = errIllegalFPOperation; break;// #NUM!
