--- /dev/null	2002-08-31 00:31:37.000000000 +0100
+++ sd/source/filter/html/htmltheme.hxx	2003-08-16 16:56:34.000000000 +0100
@@ -0,0 +1,26 @@
+#ifndef _SD_HTML_THEME_HXX
+#define _SD_HTML_THEME_HXX
+
+#include <list>
+
+class SdHtmlTheme {
+public:
+	static const SdHtmlTheme &getTheme();
+public:
+	SdHtmlTheme();
+	~SdHtmlTheme();
+	static const int nButtonsPerTheme = 13;
+	int      getThemeCount() const;
+	Bitmap   getThumbnail( int nThemeId ) const;
+	long     getBitmap( int nThemeId, int nImage, BitmapEx &rBitmap ) const;
+private:
+	/* list */
+	void loadDirectory( String aPath );
+	struct Theme {
+		ByteString aName;
+		BitmapEx   aImages[nButtonsPerTheme];
+	};
+	std::list<Theme> m_aThemes;
+};
+
+#endif // _SD_SIMPLE_THEME_HXX
