--- officecfg/registry/data/org/openoffice/Office/Common.xcu	2 Jul 2003 14:19:36 -0000	1.13
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	4 Dec 2003 02:56:49 -0000
@@ -107,13 +107,13 @@
   </node>
   <node oor:name="ExternalApps">
     <prop oor:name="http" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>sensible-browser</value>
     </prop>
     <prop oor:name="https" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>sensible-browser</value>
     </prop>
     <prop oor:name="ftp" oor:op="replace" oor:type="xs:string">
-      <value/>
+      <value>sensible-browser</value>
     </prop>
     <prop oor:name="file" oor:op="replace" oor:type="xs:string">
       <value/>
