Index: setup2/mow/source/loader/loader.c
===================================================================
RCS file: /cvs/installation/setup2/mow/source/loader/loader.c,v
retrieving revision 1.49.14.1
diff -u -3 -p -r1.49.14.1 loader.c
--- setup2/mow/source/loader/loader.c	17 Sep 2003 08:32:24 -0000	1.49.14.1
+++ setup2/mow/source/loader/loader.c	30 Sep 2003 13:16:15 -0000
@@ -2195,6 +2195,10 @@ BOOL __getFullPath(const char* pszFilena
             return (FALSE);
         }
 
+#if 0
+/* This check is only enough for an NFS mount.  A bind mount
+has the same device numbers even though the inodes do not
+match */
         if (status.st_dev == parent.st_dev)
         {
             do
@@ -2211,6 +2215,7 @@ BOOL __getFullPath(const char* pszFilena
             while (entry->d_ino != status.st_ino);
         }
         else
+#endif
         {
             /*
              * if this point is a mount point we have to check
