Index: vcl/unx/source/gdi/salgdi2.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/salgdi2.cxx,v
retrieving revision 1.18
diff -u -p -u -r1.18 salgdi2.cxx
--- vcl/unx/source/gdi/salgdi2.cxx	10 Jun 2003 14:31:08 -0000	1.18
+++ vcl/unx/source/gdi/salgdi2.cxx	9 Oct 2003 16:40:49 -0000
@@ -775,10 +779,13 @@ void SalGraphics::DrawBitmap( const SalT
         XGCValues			aOldVal, aNewVal;
         int					nValues = GCForeground | GCBackground;
 
-        // set foreground/background values for 1Bit bitmaps
-        XGetGCValues( pXDisp, aGC, nValues, &aOldVal );
-        aNewVal.foreground = rColMap.GetWhitePixel(), aNewVal.background = rColMap.GetBlackPixel();
-        XChangeGC( pXDisp, aGC, nValues, &aNewVal );
+		if( rSalBitmap.GetBitCount() == 1 )
+		{
+				// set foreground/background values for 1Bit bitmaps
+				XGetGCValues( pXDisp, aGC, nValues, &aOldVal );
+				aNewVal.foreground = rColMap.GetWhitePixel(), aNewVal.background = rColMap.GetBlackPixel();
+				XChangeGC( pXDisp, aGC, nValues, &aNewVal );
+		}
 
 #if defined(_USE_PRINT_EXTENSION_)
 		rSalBitmap.ImplDraw( pSalDisp, aDrawable, nDepth, *pPosAry,  aGC );
@@ -786,7 +793,8 @@ void SalGraphics::DrawBitmap( const SalT
 		rSalBitmap.ImplDraw( aDrawable, nDepth, *pPosAry, aGC );
 #endif
 
-        XChangeGC( pXDisp, aGC, nValues, &aOldVal );
+		if( rSalBitmap.GetBitCount() == 1 )
+				XChangeGC( pXDisp, aGC, nValues, &aOldVal );
         XFlush( pXDisp );
 
 #ifndef _USE_PRINT_EXTENSION_
