Index: oo_cvs_641/solenv/inc/unxlngi4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi4.mk,v
retrieving revision 1.13
diff -u -3 -p -r1.13 unxlngi4.mk
--- solenv/inc/unxlngi4.mk	4 Mar 2002 15:53:59 -0000	1.13
+++ solenv/inc/unxlngi4.mk	5 Feb 2003 15:54:46 -0000
@@ -196,7 +196,11 @@ STDSHLGUIST=-lX11 -lXext -ldl -lm
 STDSHLCUIST=-ldl -lm
 
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
+.IF "$(STLPORT4)"=="NO_STLPORT4"
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ELSE
+LIBSTLPORTST=$(LIBSTLPORT)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
Index: oo_cvs_641/solenv/inc/unxlngppc4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc4.mk,v
retrieving revision 1.1.2.3
diff -u -3 -p -r1.1.2.3 unxlngppc4.mk
--- solenv/inc/unxlngppc4.mk	7 Jun 2002 14:11:58 -0000	1.1.2.3
+++ solenv/inc/unxlngppc4.mk	5 Feb 2003 15:54:46 -0000
@@ -199,7 +199,11 @@ STDSHLGUIMT=-lX11 -lXext -ldl -lpthread 
 STDSHLCUIMT=-ldl -lpthread -lm
 
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
+.IF "$(STLPORT4)"=="NO_STLPORT4"
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ELSE
+LIBSTLPORTST=$(LIBSTLPORT)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
