Index: sw/source/core/unocore/unocoll.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/unocore/unocoll.cxx,v
retrieving revision 1.23
diff -u -r1.23 unocoll.cxx
--- sw/source/core/unocore/unocoll.cxx	30 Jun 2003 15:52:15 -0000	1.23
+++ sw/source/core/unocore/unocoll.cxx	30 Aug 2004 10:58:08 -0000
@@ -345,10 +345,13 @@
   -----------------------------------------------------------------------*/
 sal_uInt16	SwXServiceProvider::GetProviderType(const OUString& rServiceName)
 {
-	for(sal_uInt16 i = 0; i <= SW_SERVICE_LAST; i++ )
+        if ( rServiceName.getLength() )
 	{
-		if( COMPARE_EQUAL == rServiceName.compareToAscii(aProvNames[i]))
-			return i;
+	       for(sal_uInt16 i = 0; i <= SW_SERVICE_LAST; i++ )
+	       {
+		      if( COMPARE_EQUAL == rServiceName.compareToAscii(aProvNames[i]))
+			     return i;		  
+	       }
 	}
 	return SW_SERVICE_INVALID;
 }
Index: sw/source/ui/uno/unotxdoc.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/uno/unotxdoc.cxx,v
retrieving revision 1.78.32.2
diff -u -r1.78.32.2 unotxdoc.cxx
--- sw/source/ui/uno/unotxdoc.cxx	28 Jan 2004 17:02:08 -0000	1.78.32.2
+++ sw/source/ui/uno/unotxdoc.cxx	30 Aug 2004 10:58:09 -0000
@@ -1763,13 +1763,14 @@
 	::vos::OGuard aGuard(Application::GetSolarMutex());
 	if(!IsValid())
 		throw RuntimeException();
-	Reference< XInterface >  xRet;
+	Reference< XInterface >  xRet = 0;
 	sal_uInt16 nType = SwXServiceProvider::GetProviderType(rServiceName);
 	if(nType != SW_SERVICE_INVALID)
 	{
 		xRet = SwXServiceProvider::MakeInstance(nType, pDocShell->GetDoc());
 	}
-	else
+	else 
+	if ( rServiceName.getLength() )
 	{
         if( rServiceName.compareToAscii( "com.sun.star.", 13 ) == 0 )
         {
