Index: sw/inc/iodetect.hxx
===================================================================
RCS file: /cvs/sw/sw/inc/Attic/iodetect.hxx,v
retrieving revision 1.19
diff -u -r1.19 iodetect.hxx
--- sw/inc/iodetect.hxx	15 Apr 2003 16:47:52 -0000	1.19
+++ sw/inc/iodetect.hxx	13 Sep 2004 13:22:15 -0000
@@ -154,7 +154,7 @@
 #ifdef DEBUG_SH
 		1 +
 #endif
-		21;
+		22;
 
 #define FORAMTNAME_SW4 		"StarWriter 4.0"
 #define FORAMTNAME_SW3   	"StarWriter 3.0"
@@ -195,8 +195,8 @@
 sal_Char __FAR_DATA FILTER_XMLV[] 	= "CXMLV";	                         \
 sal_Char __FAR_DATA FILTER_XMLVW[] 	= "CXMLVWEB";	                     \
 sal_Char __FAR_DATA sSwDos[] 		= "SW6";	                            \
-																	        \
-																	        \
+sal_Char __FAR_DATA sLWP[]		= "LWP";			     \	
+																	       \ 
 SwIoDetect aReaderWriter[ MAXFILTER ] = { 							        \
 /*	0*/ SwIoEntry(FILTER_SW5, 		4, 			&::GetSw3Writer, 	TRUE),  \
 /*	1*/ SwIoEntry(FILTER_SW4, 		4, 			&::GetSw3Writer, 	FALSE), \
@@ -217,8 +217,9 @@
 /* 16*/ SwIoEntry(sWW1,  			STRING_LEN,	0,				  	TRUE),  \
 /* 17*/ SwIoEntry(sWW5,				STRING_LEN,	0, 					FALSE), \
 /* 18*/ SwIoEntry(sSwg1,            4,          0,  				FALSE), \
-/* 19*/ SwIoEntry(FILTER_XML,		4,			&::GetXMLWriter,	TRUE)	\
-                                                                            \
+/* 19*/ SwIoEntry(sLWP,				4,			0,					TRUE),	\
+/* 20*/ SwIoEntry(FILTER_XML,		4,			&::GetXMLWriter,	FALSE)	\
+					\
 /* opt*/ DEB_SH_SwIoEntry(sW4W_Int, STRING_LEN, 0,				  	TRUE)   \
 /* opt*/ DEB_DBWRT_SwIoEntry(sDebug,STRING_LEN, &::GetDebugWriter,	FALSE)  \
 /* opt*/ DEB_DBWRT_SwIoEntry(sUndo,	STRING_LEN, &::GetUndoWriter,	FALSE)  \
@@ -306,7 +307,12 @@
 																			\
 		bRet = 0 == strncmp( sSw6_FormatStt, pHeader, 12 ) &&               \
 		   	   0 == strncmp( sSw6_FormatEnd, pHeader + 12 + 1, 4 );			\
-	}                                                                       \
+	}                 									\ 
+	else if( sLWP == pName )												\
+	{																		\
+		sal_Char __READONLY_DATA aLWP[] = "WordPro";						\
+		bRet = 0 == strncmp( aLWP, pHeader, 7);					\
+	}		                                                     \
     else if (FILTER_TEXT == pName) \
         bRet = SwIoSystem::IsDetectableText(pHeader, nLen); \
     else if (FILTER_W4W == pName) \
@@ -323,7 +329,7 @@
 		rUserData.EqualsAscii(FILTER_SWW5V) ||                                        \
     	rUserData.EqualsAscii(FILTER_SW4 )|| rUserData.EqualsAscii(FILTER_SW4V) ||    \
 		rUserData.EqualsAscii(FILTER_SWW4V) ||                                        \
-		rUserData.EqualsAscii(FILTER_SW3 )|| rUserData.EqualsAscii(FILTER_SW3V) )     \
+		rUserData.EqualsAscii(FILTER_SW3 )|| rUserData.EqualsAscii(FILTER_SW3V) || rUserData.EqualsAscii(sLWP))     \
 		return String::CreateFromAscii(                                               \
 				RTL_CONSTASCII_STRINGPARAM( "StarWriterDocument" ));                  \
 	if( rUserData.EqualsAscii(FILTER_XML) || 										  \
Index: sw/source/filter/basflt/fltini.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/basflt/fltini.cxx,v
retrieving revision 1.23
diff -u -r1.23 fltini.cxx
--- sw/source/filter/basflt/fltini.cxx	17 Apr 2003 14:49:22 -0000	1.23
+++ sw/source/filter/basflt/fltini.cxx	13 Sep 2004 13:22:15 -0000
@@ -234,6 +234,7 @@
 	_SetFltPtr( nCnt, new WW1Reader, sWW1 );
 	_SetFltPtr( nCnt, pWW8Rd, sWW5 );
 	_SetFltPtr( nCnt, ReadSwg, sSwg1 );
+	_SetFltPtr( nCnt, (ReadXML = new LotusReader), sLWP );
 	_SetFltPtr( nCnt, (ReadXML = new XMLReader), FILTER_XML );
 
 #ifdef NEW_WW97_EXPORT
