--- solenv/inc/libs.mk	31 Oct 2003 16:23:21 -0000	1.66.14.1
+++ solenv/inc/libs.mk	9 Dec 2003 11:50:20 -0000
@@ -135,9 +135,12 @@
 BSTRPLIB=-lbtstrp
 VCLLIBST=-lvcl
 VCLLIB=$(SVLIB)
-FREETYPELIB=-lfreetype
-FREETYPELIBST=$(STATIC) -lfreetype $(DYNAMIC)
+.IF "$(SYSTEM_FREETYPE)"!="YES"
+  FREETYPE_LIBS=-lfreetype
+.ENDIF
+FREETYPELIB=$(FREETYPE_LIBS)
+FREETYPELIBST=$(STATIC) $(FREETYPE_LIBS) $(DYNAMIC)
 XPLIB=-l_xp
 TKLIB=-ltk$(OFFICEUPD)$(DLLPOSTFIX)
 SVTOOLLIB=-lsvt$(OFFICEUPD)$(DLLPOSTFIX)

--- vcl/source/glyphs/makefile.mk	15 Feb 2002 15:52:27 -0000	1.5
+++ vcl/source/glyphs/makefile.mk	9 Dec 2003 11:50:47 -0000
@@ -73,6 +73,7 @@
 # --- Files --------------------------------------------------------
 
 .IF "$(USE_BUILTIN_RASTERIZER)" != ""
+CFLAGS += $(FREETYPE_CFLAGS)
 SLOFILES=\
 		$(SLO)$/glyphcache.obj		\
 		$(SLO)$/gcach_rbmp.obj		\
Index: vcl/source/glyphs/gcach_ftyp.cxx
===================================================================
--- vcl/source/glyphs/gcach_ftyp.cxx.orig	2004-01-07 23:21:33.000000000 +0100
+++ vcl/source/glyphs/gcach_ftyp.cxx	2004-01-08 00:48:40.000000000 +0100
@@ -77,12 +77,13 @@
 #include <osl/file.hxx>
 #include <osl/thread.hxx>
 
-#include "freetype/freetype.h"
-#include "freetype/ftglyph.h"
-#include "freetype/ftoutln.h"
-#include "freetype/tttables.h"
-#include "freetype/tttags.h"
-#include "freetype/ttnameid.h"
+#include "ft2build.h"
+#include  FT_FREETYPE_H
+#include  FT_GLYPH_H
+#include  FT_OUTLINE_H
+#include  FT_TRUETYPE_TABLES_H
+#include  FT_TRUETYPE_TAGS_H
+#include  FT_TRUETYPE_IDS_H
 
 #ifndef FREETYPE_PATCH
     // VERSION_MINOR in freetype.h is too coarse
Index: gcach_layout.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/glyphs/gcach_layout.cxx,v
retrieving revision 1.23
diff -u -r1.23 gcach_layout.cxx
--- vcl/source/glyphs/gcach_layout.cxx	16 Jul 2003 17:46:32 -0000	1.23
+++ vcl/source/glyphs/gcach_layout.cxx	18 Nov 2003 01:49:35 -0000
@@ -68,7 +68,8 @@
 #endif
 
 #ifndef _SV_GCACHFTYP_HXX
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <gcach_ftyp.hxx>
 #endif
 
Index: freetype/makefile.mk
===================================================================
--- freetype/makefile.mk.orig	2003-05-28 14:41:19.000000000 +0200
+++ freetype/makefile.mk	2004-01-08 00:48:40.000000000 +0100
@@ -75,12 +75,12 @@ TARGET=so_freetype
 # the X11 port because there are some header conflicts
 # and linktime conflicts as well.  It is simply less of
 # a headache to use the one provided.
-.IF "$(OS)" == "MACOSX"
+.IF "$(OS)" == "MACOSX" || "$(SYSTEM_FREETYPE)" == "YES"
 
 dummy:
-	@echo "Nothing to build for MACOSX"
+	@echo "Nothing to build for OS $(OS)"
 
-.ELSE	# "$(OS)" == "MACOSX"
+.ELSE	# "$(OS)" == "MACOSX" || "$(SYSTEM_FREETYPE)" == "YES"
 
 TARFILE_NAME=freetype-2.1.4
 
Index: scp/source/global/setupzip.scp
===================================================================
--- scp/source/global/setupzip.scp.orig	2003-12-12 21:11:03.000000000 +0100
+++ scp/source/global/setupzip.scp	2004-01-08 00:49:26.000000000 +0100
@@ -556,6 +556,8 @@ End
 
 #ifdef UNX
 
+#ifndef SYSTEM_FREETYPE
+
 #ifndef MACOSX
 File GID_FILE_LIB_FREETYPE
     #if defined (LINUX)  &&  ( defined (CLASSIC_FAT_PRODUCT) || defined (OSL_PRODUCT) || defined (FAM_PRODUCT) )
@@ -573,6 +575,8 @@ File GID_FILE_LIB_FREETYPE
 End
 #endif
 
+#endif
+
 #if defined (OSL_PRODUCT) && !defined(WITHOUT_FONTS)
 STD_FONTUNX_FILE( GID_FILE_FNT_VERA, Vera.ttf, Vera Sans)
 #endif
