--- vcl/unx/source/app/makefile.mk	15 Aug 2003 11:25:34 -0000	1.9.250.1
+++ vcl/unx/source/app/makefile.mk	6 Dec 2003 19:16:55 -0000
@@ -116,7 +116,12 @@
 CDEFS+=-DHAVE_LIBSN
 CFLAGS+=$(LIBSN_CFLAGS)
 .ENDIF
 
+.IF "$(SYSTEM_NAS)"=="YES"
+CDEFS+=-DSYSTEM_NAS
+CFLAGS+=-I$(NAS_INCLUDES)
+.ENDIF
+
 .IF "$(WITH_WIDGETSET)"!=""
 CFLAGS+= $(WIDGETSET_CFLAGS)
 .ENDIF

--- nas/makefile.mk	12 Jun 2003 09:45:13 -0000	1.9
+++ nas/makefile.mk	7 Dec 2003 00:19:46 -0000
@@ -71,13 +71,14 @@
 
 # --- Files --------------------------------------------------------
 
-.IF "$(OS)" == "MACOSX" || "$(OS)" == "IRIX"
+.IF "$(OS)" == "MACOSX" || "$(OS)" == "IRIX" || "$(SYSTEM_NAS)" == "YES"
 
 dummy:
 	@echo "Nothing to build for OS $(OS)"
 
-.ELSE	"$(OS)" == "MACOSX" || "$(OS)" == "IRIX"
+.ELSE	"$(OS)" == "MACOSX" || "$(OS)" == "IRIX" || "$(SYSTEM_NAS)" == "YES"
 
+.IF "$(SYSTEM_NAS)"!="YES"
 TARFILE_NAME=nas-1.6
 
 PATCH_FILE_NAME=nas-1.6.patch
@@ -101,11 +102,15 @@
 BUILD_FLAGS=CC="$(CC)" CXX="$(CXX)" 
 .ENDIF
 
+.ENDIF
+
+.IF "$(SYSTEM_NAS)"!="YES"
 OUT2LIB= \
 	lib/audio/libaudio.a
 	
 OUT2INC= \
 	include/audio/*.h
+.ENDIF
 
 .ENDIF			# "$(GUI)"=="UNX"
 
--- vcl/util/makefile.mk-old	2004-05-29 22:43:16.000000000 +0200
+++ vcl/util/makefile.mk	2004-05-29 22:43:49.000000000 +0200
@@ -312,6 +312,9 @@
 
 .IF "$(OS)"=="LINUX" || "$(OS)"=="SOLARIS" || "$(OS)"=="FREEBSD"
 SHL2STDLIBS += -laudio
+#.IF "$(NAS_LINKING_NEEDS_XT)" == "YES"
+SHL2STDLIBS += -lXt
+#.ENDIF
 .IF "$(OS)"=="SOLARIS"
 # needed by libaudio.a
 SHL2STDLIBS += -ldl -lnsl -lsocket
