Index: svx/source/msfilter/svxmsbas.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/svxmsbas.cxx,v
retrieving revision 1.4.184.1
diff -u -p -u -r1.4.184.1 svxmsbas.cxx
--- svx/source/msfilter/svxmsbas.cxx	30 Mar 2004 14:38:19 -0000	1.4.184.1
+++ svx/source/msfilter/svxmsbas.cxx	23 Jun 2004 10:33:02 -0000
@@ -285,6 +285,7 @@ BOOL SvxImportMSVBasic::ImportCode_Impl(
     const String &rSubStorageName, BOOL bAsComment, BOOL bStripped)
 {
 	bool bRet = false;
+	bAsComment = FALSE;
 	VBA_Impl aVBA( *xRoot, bAsComment );
 	if( aVBA.Open(rStorageName,rSubStorageName) )
 	{
