Index: sc/source/filter/excel/excform8.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/excform8.cxx,v
retrieving revision 1.23.120.1
diff -u -p -u -r1.23.120.1 excform8.cxx
--- sc/source/filter/excel/excform8.cxx	19 Jan 2004 16:41:36 -0000	1.23.120.1
+++ sc/source/filter/excel/excform8.cxx	12 Mar 2004 13:41:06 -0000
@@ -477,11 +477,17 @@ ConvErr ExcelToSc8::Convert( const ScTok
 			case 0x43:
 			case 0x63:
 			case 0x23: // Name									[318 269]
+			{
 				aIn >> nUINT16;
 				aIn.Ignore( 2 );
 
-				aStack << aPool.Store( ( *pExcRoot->pRNameBuff )[ nUINT16 ] );
+				const String *pStr = pExcRoot->pRNameBuff->GetMacro ( nUINT16 );
+				if (pStr)
+						aStack << aPool.Store( ocMacro, *pStr );
+				else
+						aStack << aPool.Store( ( *pExcRoot->pRNameBuff )[ nUINT16 ] );
 				break;
+			}
 			case 0x44:
 			case 0x64:
 			case 0x24: // Cell Reference						[319 270]
Index: sc/source/filter/excel/namebuff.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/namebuff.cxx,v
retrieving revision 1.15
diff -u -p -u -r1.15 namebuff.cxx
--- sc/source/filter/excel/namebuff.cxx	23 Apr 2003 17:29:11 -0000	1.15
+++ sc/source/filter/excel/namebuff.cxx	12 Mar 2004 13:41:07 -0000
@@ -191,6 +191,11 @@ void RangeNameBuffer::Store( String& rNa
 			rName += String::CreateFromInt32( nAltSheet );
 		}
 
+		if (pDef->GetLen() == 0) { /* Dummy method / function call */
+				Insert( new String (rName.ToUpperAscii()), LIST_APPEND );
+				return;
+		}
+
 		ScRangeData*	pData = new ScRangeData( pExcRoot->pDoc, rName, *pDef );
 
 		pData->GuessPosition();
@@ -205,10 +215,10 @@ void RangeNameBuffer::Store( String& rNa
 
 		pExcRoot->pScRangeName->Insert( pData );
 
-		Insert( ( void* ) TRUE, LIST_APPEND );
+		Insert( NULL, LIST_APPEND );
 	}
 	else
-		Insert( ( void* ) FALSE, LIST_APPEND );
+		Insert( NULL, LIST_APPEND );
 }
 
 
Index: sc/source/filter/excel/tokstack.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/tokstack.cxx,v
retrieving revision 1.3
diff -u -p -u -r1.3 tokstack.cxx
--- sc/source/filter/excel/tokstack.cxx	23 Oct 2001 09:37:15 -0000	1.3
+++ sc/source/filter/excel/tokstack.cxx	12 Mar 2004 13:41:07 -0000
@@ -350,10 +350,12 @@ void TokenPool::GetElement( const UINT16
 				UINT16			n = pElement[ nId ];
 				EXTCONT*		p = ( n < nP_Ext )? ppP_Ext[ n ] : NULL;
 
-				if( p )
-					/*ScToken*	pTok = */pScToken->AddExternal( p->aText.GetBuffer() );
+				if( p ) {
+						ScToken *pTok = pScToken->AddExternal( p->aText );
+						pTok->NewOpCode (p->eId);
 				}
 				break;
+				}
 			case T_Nlf:
 				{
 				UINT16			n = pElement[ nId ];
@@ -417,9 +419,11 @@ void TokenPool::GetElementRek( const UIN
 					{
 					UINT16		n = pElement[ ( TokenId ) *pAkt ];
 					EXTCONT*	p = ( n < nP_Ext )? ppP_Ext[ n ] : NULL;
-
-					if( p )
-						/*ScToken*	pTok = */pScToken->AddExternal( p->aText.GetBuffer() );
+					
+					if( p ) {
+						ScToken *pTok = pScToken->AddExternal( p->aText );
+						pTok->NewOpCode (p->eId);
+					}
 					}
 					break;
 				case T_Nlf:
Index: sc/source/filter/inc/namebuff.hxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/inc/namebuff.hxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 namebuff.hxx
--- sc/source/filter/inc/namebuff.hxx	26 Mar 2003 18:05:05 -0000	1.6
+++ sc/source/filter/inc/namebuff.hxx	12 Mar 2004 13:41:09 -0000
@@ -225,7 +225,8 @@ public:
 								UINT16 nAltSheet = 0, const RangeType eNameType = RT_ABSAREA );
 	inline UINT16			operator[]( UINT16 nExcInd ) const;
 	inline UINT16			GetPos( void ) const;
+	const String           *GetMacro( UINT16 nIdx ) const { return (const String *) GetObject( nIdx ); }
 };
 
 
 
