Index: berkeleydb/makefile.mk
===================================================================
RCS file: /cvs/external/berkeleydb/makefile.mk,v
retrieving revision 1.13.12.1
diff -u -p -u -r1.13.12.1 makefile.mk
--- berkeleydb/makefile.mk	9 Jan 2004 13:44:31 -0000	1.13.12.1
+++ berkeleydb/makefile.mk	1 Apr 2004 23:56:37 -0000
@@ -106,7 +106,14 @@ OUT2BIN=java$/classes$/db.jar
 EXT_USE_STLPORT=TRUE
 
 BUILD_DIR=build_win32
-.IF "$(COMEX)"=="8"
+.IF "$(SOLAR_JAVA)" != ""
+BUILD_JAVA_DEVENV= /build Release /project db_java /useenv
+BUILD_JAVA_MSDEV= /MAKE "db_java - RELEASE"
+BUILD_JAVA_BIN=java$/classes$/db.jar \
+		$(BUILD_DIR)$/Release$/libdb_java32.dll
+BUILD_JAVA_LIB=$(BUILD_DIR)$/Release$/libdb_java32.lib
+.ENDIF
+.IF "$(COMEX)"=="8" || "$(COMEX)"=="10"
 CONFIGURE_DIR=build_win32
 CONFIGURE_ACTION=wdevenv Berkeley_DB Release
 .IF "$(USE_SHELL)"!="4nt"
@@ -114,17 +121,16 @@ BUILD_ACTION_SEP=;
 .ELSE # "$(USE_SHELL)"!="4nt"
 BUILD_ACTION_SEP=^
 .ENDIF # "$(USE_SHELL)"!="4nt"
-BUILD_ACTION=devenv Berkeley_DB.sln /build Release /project db_buildall /useenv $(BUILD_ACTION_SEP) devenv Berkeley_DB.sln /build Release /project db_java /useenv
+BUILD_ACTION=devenv Berkeley_DB.sln /build Release /project db_buildall /useenv $(BUILD_ACTION_SEP) devenv Berkeley_DB.sln  $(BUILD_JAVA_DEVENV)
 .ELSE
-BUILD_ACTION=msdev Berkeley_DB.dsw /useenv /MAKE "db_buildall - RELEASE" /MAKE "db_java - RELEASE"
+BUILD_ACTION=msdev Berkeley_DB.dsw /useenv /MAKE "db_buildall - RELEASE" $(BUILD_JAVA_MSDEV)
 .ENDIF
 
-OUT2BIN=java$/classes$/db.jar \
-	$(BUILD_DIR)$/Release$/libdb_java32.dll \
-	$(BUILD_DIR)$/Release$/libdb32.dll
+OUT2BIN= $(BUILD_JAVA_BIN) \
+	 $(BUILD_DIR)$/Release$/libdb32.dll
 
 OUT2LIB= \
-	$(BUILD_DIR)$/Release$/libdb_java32.lib \
+	$(BUILD_JAVA_LIB) \
 	$(BUILD_DIR)$/Release$/libdb32.lib
 
 .ENDIF			# "$(GUI)"=="WNT"
