--- sd/source/filter/html/htmltheme.hxx	2004-04-15 13:40:11.000000000 +0530
+++ sd/source/filter/html/htmltheme.hxx	2004-04-15 13:45:11.000000000 +0530
@@ -3,13 +3,15 @@
 
 #include <list>
 
+	static const int nButtonsPerTheme = 13;
 class SdHtmlTheme {
 public:
 	static const SdHtmlTheme &getTheme();
 public:
 	SdHtmlTheme();
 	~SdHtmlTheme();
-	static const int nButtonsPerTheme = 13;
+	//Win32 Build fails 
+	//static const int nButtonsPerTheme = 13;
 	int      getThemeCount() const;
 	Bitmap   getThumbnail( int nThemeId ) const;
 	long     getBitmap( int nThemeId, int nImage, BitmapEx &rBitmap ) const;
--- sd/source/ui/view/drviews2.cxx	2004-04-15 13:27:38.000000000 +0530
+++ sd/source/ui/view/drviews2.cxx	2004-04-15 13:34:03.000000000 +0530
@@ -1401,7 +1368,8 @@ void lcl_GetPreferedExtension( String &r
 	// dann ggfs. ueber die native-Info der Grafik den "besten"
 	// Filter vorschlagen
 	const sal_Char* pExt = "png";
-	switch( rGrf.GetLink().GetType() )
+
+	switch( ( const_cast<Graphic &>(rGrf).GetLink()).GetType() )
 	{
 		case GFX_LINK_TYPE_NATIVE_GIF:      pExt = "gif"; break;
 		case GFX_LINK_TYPE_NATIVE_TIF:      pExt = "tif"; break;
