--- shell/source/win32/simplemail/smplmailclient.cxx
+++ shell/source/win32/simplemail/smplmailclient.cxx
@@ -452,14 +452,19 @@
                 static_cast< XSimpleMailClient* >( this ), 
                 1 );
 
-        m_AttchmtsSysPathList.push_back( 
-            OString( 
-                aSysPath.getStr( ), 
-                aSysPath.getLength( ), 
-                osl_getThreadTextEncoding( ) ) );
-       
+	OString aFullPath = OUStringToOString( aSysPath,
+					       osl_getThreadTextEncoding() );
+	sal_Int32 nLastSlash = aFullPath.lastIndexOf('\\');
+	OString aFileName = aFullPath.copy( nLastSlash + 1 );
+
+        m_AttchmtsSysPathList.push_back( aFullPath );
         m_AttachementList[i].lpszPathName = const_cast< LPSTR >(
             m_AttchmtsSysPathList.back( ).getStr( ) );  
+
+        m_AttchmtsSysPathList.push_back( aFileName );
+        m_AttachementList[i].lpszFileName = const_cast< LPSTR >(
+            m_AttchmtsSysPathList.back( ).getStr( ) );  
+
         m_AttachementList[i].nPosition = -1;               
     }
 }
