Index: scp/source/office/profile.scp
===================================================================
RCS file: /cvs/installation/scp/source/office/Attic/profile.scp,v
retrieving revision 1.182.10.6
Index: scp/source/office/profile.scp
===================================================================
--- scp/source/office/profile.scp.orig	2004-07-30 22:56:27.620315216 +0200
+++ scp/source/office/profile.scp	2004-07-30 22:59:24.030496792 +0200
@@ -3540,6 +3540,7 @@ End
 
 // HACK: write accessibility bridge service info
 
+#ifdef SOLAR_JAVA
 #if ! ( defined (WEB_PRODUCT) || defined (SRV_PRODUCT) )
 
     StarRegistryItem GID_STARREGISTRYITEM_SERVICES_ACCESSBRIDGE
@@ -3575,6 +3576,7 @@ End
     End
 
 #endif
+#endif
 
 ////////////////////////////////////////////////////////////////////////////////////////
 // profile objects              WNT
Index: scp/source/global/setupzip.scp
===================================================================
--- scp/source/global/setupzip.scp.orig	2004-07-30 22:56:37.046882160 +0200
+++ scp/source/global/setupzip.scp	2004-07-30 22:59:24.032496488 +0200
@@ -524,7 +524,6 @@ File GID_FILE_JAR_UNOIL
 	Dir             = GCFG_CLASSES_DIR;
 	Styles          = (PACKED,DONT_OVERWRITE,SETUPZIP);
 End
-#endif
 
 #ifdef INCLUDE_JAVA_ACCESSBRIDGE
 
@@ -537,6 +536,8 @@ End
 
 #endif
 
+#endif
+
 #ifdef UNX
 #ifdef SOLAR_JAVA
 File GID_FILE_BIN_JAVALDX
@@ -574,13 +575,14 @@ STD_FONTUNX_FILE( GID_FILE_FNT_VERA, Ver
 #endif
 
 #ifdef WNT
-
+#ifdef SOLAR_JAVA
 File GID_FILE_LIB_ACCESSBRIDGE
         BIN_FILE_BODY;
         Styles          = (PACKED,SETUPZIP);
         Dir             = GCFG_BINARY_DIR;
         Name            = "java_uno_accessbridge.dll";
 End
+#endif
 
 #ifdef OSL_PRODUCT
 
