Index: stlport/makefile.mk
===================================================================
RCS file: /cvs/external/stlport/makefile.mk,v
retrieving revision 1.22.6.4
diff -u -p -u -r1.22.6.4 makefile.mk
--- stlport/makefile.mk	28 Jan 2004 10:52:04 -0000	1.22.6.4
+++ stlport/makefile.mk	30 Apr 2004 12:37:15 -0000
@@ -113,9 +113,9 @@ BUILD_DIR=src
 .IF "$(COM)"=="MSC"
 BUILD_ACTION=nmake
 .IF "$(CCNUMVER)"<="001300000000"
-BUILD_FLAGS=-f vc6.mak
+BUILD_FLAGS=/p /f vc6.mak
 .ELSE			# "$(CCNUMVER)"<="001300000000"
-BUILD_FLAGS=-f vc7.mak
+BUILD_FLAGS=/p /f vc7.mak
 .ENDIF			# "$(CCNUMVER)"<="001300000000"
 .ENDIF
 
