Index: ucb/prj/build.lst
===================================================================
RCS file: /cvs/ucb/ucb/prj/build.lst,v
retrieving revision 1.11
diff -u -p -u -r1.11 build.lst
--- ucb/prj/build.lst	27 Mar 2003 17:26:37 -0000	1.11
+++ ucb/prj/build.lst	13 May 2003 20:58:30 -0000
@@ -8,6 +8,7 @@ uc	ucb\source\cacher		    nmake	-	all	uc
 uc	ucb\source\sorter		    nmake	-	all	uc_sorter uc_unotypes NULL
 uc	ucb\source\ucp\ftp		    nmake	-	all	uc_ftp uc_unotypes NULL
 uc	ucb\source\ucp\file		    nmake	-	all	uc_file uc_unotypes NULL
+uc	ucb\source\ucp\odma		nmake	-	all	uc_odma uc_unotypes NULL
 uc	ucb\source\ucp\hierarchy	nmake	-	all	uc_hierarchy uc_unotypes NULL
 uc	ucb\source\ucp\remote		nmake	-	all	uc_remote uc_regexp NULL
 uc	ucb\source\ucp\webdav		nmake	-	all	uc_webdav uc_unotypes NULL

Index: officecfg/registry/data/org/openoffice/ucb/Configuration.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/ucb/Configuration.xcu,v
retrieving revision 1.4
diff -u -p -u -r1.4 Configuration.xcu
--- officecfg/registry/data/org/openoffice/ucb/Configuration.xcu	8 Nov 2002 16:50:28 -0000	1.4
+++ officecfg/registry/data/org/openoffice/ucb/Configuration.xcu	13 May 2003 20:52:35 -0000
@@ -154,6 +154,17 @@
                 <value/>
               </prop>
             </node>
+            <node oor:name="Provider9" oor:op="replace">
+              <prop oor:name="ServiceName">
+                <value>com.sun.star.comp.odma.ContentProvider</value>
+              </prop>
+              <prop oor:name="URLTemplate">
+                <value>odma</value>
+              </prop>
+              <prop oor:name="Arguments">
+                <value/>
+              </prop>
+            </node>
           </node>
         </node>
       </node>

Index: scp/source/office/files.scp
===================================================================
RCS file: /cvs/installation/scp/source/office/files.scp,v
retrieving revision 1.367
diff -u -p -u -r1.367 files.scp
--- scp/source/office/files.scp	17 Jul 2003 08:54:46 -0000	1.367
+++ scp/source/office/files.scp	5 Aug 2003 14:48:48 -0000
@@ -879,6 +883,8 @@ SPECIAL_UNO_LIB_FILE(GID_FILE_LIB_PACKAG
 
 SPECIAL_UNO_LIB_FILE(GID_FILE_LIB_UCPHIER1,ucphier1)
 SPECIAL_UNO_LIB_FILE(GID_FILE_LIB_UCPDAV1,ucpdav1)
+SPECIAL_UNO_LIB_FILE(GID_FILE_LIB_UCPODMA1,ucpodma1)
+SPECIAL_UNO_LIB_FILE(GID_FILE_LIB_UCPODMALIB,odma32)
 SPECIAL_UNO_LIB_FILE(GID_FILE_LIB_FILEACC,fileacc)
 SPECIAL_UNO_COMPONENT_LIB_FILE(GID_FILE_LIB_TINSTRM,textinstream.uno)
 SPECIAL_UNO_COMPONENT_LIB_FILE(GID_FILE_LIB_TOUTSTRM,textoutstream.uno)

Index: ucb/prj/d.lst
===================================================================
RCS file: /cvs/ucb/ucb/prj/d.lst,v
retrieving revision 1.11.38.1
diff -u -p -u -r1.11.38.1 d.lst
--- ucb/prj/d.lst	28 Jan 2004 10:12:30 -0000	1.11.38.1
+++ ucb/prj/d.lst	4 May 2004 13:29:07 -0000
@@ -16,6 +16,7 @@
 ..\%__SRC%\misc\ucb*.sym %_DEST%\bin%_EXT%\ucb*.sym
 ..\source\core\ucb.xml %_DEST%\xml%_EXT%\ucb.xml
 
+..\source\ucp\odma\*.dll %_DEST%\bin%_EXT%\*.dll
 ..\%__SRC%\bin\ucp*.dll %_DEST%\bin%_EXT%\ucp*.dll
 ..\%__SRC%\lib\libucp*.so %_DEST%\lib%_EXT%\libucp*.so
 ..\%__SRC%\lib\libucp*.dylib %_DEST%\lib%_EXT%\libucp*.dylib
Index: tools/source/fsys/urlobj.cxx
===================================================================
RCS file: /cvs/util/tools/source/fsys/urlobj.cxx,v
retrieving revision 1.40
diff -u -p -u -r1.40 urlobj.cxx
--- tools/source/fsys/urlobj.cxx	22 May 2003 08:45:17 -0000	1.40
+++ tools/source/fsys/urlobj.cxx	4 May 2004 13:58:35 -0000
@@ -452,6 +491,8 @@ static INetURLObject::SchemeInfo const a
           false, false, false, false, false, false },
 		{ "vnd.sun.star.odma", "vnd.sun.star.odma:", 0, false, false, false,
 		  false, false, false, true, false },
+		{ "odma", "odma://", 0, false, false, false,
+		  false, false, false, true, false },
 		{ "telnet", "telnet://", 23, true, true, false, true, true, true, true,
 		  false } };
 
@@ -2015,6 +2103,7 @@ INetURLObject::getPrefix(sal_Unicode con
 			  PrefixInfo::INTERNAL },
 			{ "mailto:", 0, INET_PROT_MAILTO, PrefixInfo::OFFICIAL },
 			{ "news:", 0, INET_PROT_NEWS, PrefixInfo::OFFICIAL },
+			{ "odma:", 0, INET_PROT_ODMA, PrefixInfo::OFFICIAL },
 			{ "out:", "staroffice.out:", INET_PROT_OUT,
 			  PrefixInfo::INTERNAL },
 			{ "pop3:", "staroffice.pop3:", INET_PROT_POP3,
Index: tools/inc/urlobj.hxx
===================================================================
RCS file: /cvs/util/tools/inc/urlobj.hxx,v
retrieving revision 1.23
diff -u -p -u -r1.23 urlobj.hxx
--- tools/inc/urlobj.hxx	27 Mar 2003 17:03:29 -0000	1.23
+++ tools/inc/urlobj.hxx	4 May 2004 13:58:35 -0000
@@ -181,7 +181,8 @@ enum INetProtocol
     INET_PROT_VND_SUN_STAR_SCRIPT = 27,
 	INET_PROT_VND_SUN_STAR_ODMA = 28,
     INET_PROT_TELNET = 29,
-	INET_PROT_END = 30
+    INET_PROT_ODMA = 30,
+	INET_PROT_END = 31
 };
 
 //============================================================================
