Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.55.6.7
diff -u -p -u -r1.55.6.7 configure.in
--- config_office/configure.in	19 Feb 2004 10:59:25 -0000	1.55.6.7
+++ config_office/configure.in	22 Mar 2004 10:33:58 -0000
@@ -1653,7 +2006,11 @@ WITH_LIBART=NO
 if test -n "$enable_libart"; then
   PKG_CHECK_MODULES( LIBART, libart-2.0 >= 2.3.13 )
   WITH_LIBART=YES
-	
+
+  if test $_os = "WINNT"; then
+    LIBART_LIBS=${LIBART_LIBS/-l/}
+    LIBART_LIBS=${LIBART_LIBS/_2/_2.lib }
+  fi
 elif test "$with_gpc" != "no" ; then
   WITH_GPC=YES
 
