Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.55.6.7
diff -u -p -u -r1.55.6.7 configure.in
--- config_office/configure.in	19 Feb 2004 10:59:25 -0000	1.55.6.7
+++ config_office/configure.in	30 Apr 2004 16:03:45 -0000
@@ -1529,11 +1921,13 @@ dnl ************************************
 dnl testing bash tools path on Windows
 dnl ***************************************
 if test "$_os" = "WINNT"; then
-	AC_PATH_PROG(CYGWIN, bash)
-	CYGWIN=`dirname "$CYGWIN"`
+	AC_PATH_PROG(CYGWIN_PATH, bash)
+	CYGWIN_PATH=`dirname "$CYGWIN_PATH"`
 fi
-if test -z "$CYGWIN"; then
+if test -z "$CYGWIN_PATH"; then
    CYGWIN="NO_CYGWIN"
+else
+   CYGWIN="$CYGWIN_PATH"
 fi
 AC_SUBST(CYGWIN)
 
