===================================================================
RCS file: /var/cvsup/gsl/vcl/source/gdi/bitmap.cxx,v
retrieving revision 1.7.178.1
retrieving revision 1.7.178.2
diff -u -r1.7.178.1 -r1.7.178.2
--- vcl/source/gdi/bitmap.cxx	2003/11/25 18:08:00	1.7.178.1
+++ vcl/source/gdi/bitmap.cxx	2003/12/10 18:25:44	1.7.178.2
@@ -1792,7 +1792,7 @@
     if (eColorMode == BMP_COLOR_HIGHCONTRAST && GetBitCount() >= 24) {
 
 	    // Yet another special case for icons
-	    BitmapReadAccess* pAcc = AcquireReadAccess();
+	    BitmapReadAccess* pAcc = ((Bitmap*)this)->AcquireReadAccess();
 	    BitmapWriteAccess* pDis = aRet.AcquireWriteAccess();
 
 	    Size aSize = GetSizePixel();
@@ -1811,7 +1811,7 @@
 	    }
 
 	    aRet.ReleaseAccess( pDis );
-	    ReleaseAccess( pAcc );
+	    ((Bitmap*)this)->ReleaseAccess( pAcc );
     } else {
 	
 		Color*  pSrcColors = NULL;
===================================================================
RCS file: /var/cvsup/gsl/vcl/source/gdi/bitmap2.cxx,v
retrieving revision 1.4.178.1
retrieving revision 1.4.178.2
diff -u -r1.4.178.1 -r1.4.178.2
--- bitmap2.cxx	2003/11/25 18:08:00	1.4.178.1
+++ vcl/source/gdi/bitmap2.cxx	2003/12/10 18:25:45	1.4.178.2
@@ -1079,7 +1079,7 @@
 
 	aAlpha.Erase( COL_WHITE );
 
-	BitmapReadAccess *pSrc = AcquireReadAccess();
+	BitmapReadAccess *pSrc = ((Bitmap*)this)->AcquireReadAccess();
 	BitmapWriteAccess *pDst = aAlpha.AcquireWriteAccess();
 
 	for( long nY = 0; nY < aSize.Height(); nY++ )
@@ -1092,8 +1092,8 @@
 		}
 	}
 	
-	ReleaseAccess( pSrc );
-	ReleaseAccess( pDst );
+	((Bitmap*)this)->ReleaseAccess( pSrc );
+	((Bitmap*)this)->ReleaseAccess( pDst );
 		
 	return aAlpha;
 }
===================================================================
RCS file: /var/cvsup/gsl/vcl/source/gdi/image.cxx,v
retrieving revision 1.8.202.1
retrieving revision 1.8.202.2
diff -u -r1.8.202.1 -r1.8.202.2
--- image.cxx	2003/11/25 18:08:02	1.8.202.1
+++ vcl/source/gdi/image.cxx	2003/12/10 18:25:47	1.8.202.2
@@ -804,9 +804,11 @@
 		return;
 	}
 
+    BOOL bUseAlphaMask = FALSE;
 	if( !rMaskBmp && rBitmap.GetBitCount() == 32 )
 	{
-		rMaskBmp = rBitmap.GetAlphaMask();
+        bUseAlphaMask = TRUE;
+		//rMaskBmp = rBitmap.GetAlphaMask();
 		bColor = FALSE;
 	}
 
@@ -859,7 +861,7 @@
 	}
 
 	pThis->mpImplData->mpImageBitmap = new ImplImageBmp;
-	pThis->mpImplData->mpImageBitmap->Create( rBitmap, rMaskBmp,
+    pThis->mpImplData->mpImageBitmap->Create( rBitmap, bUseAlphaMask ? rBitmap.GetAlphaMask() : rMaskBmp,
 											  rColor, bColor,
 											  pThis->mpImplData->maImageSize.Width(),
 											  pThis->mpImplData->maImageSize.Height(),
===================================================================
RCS file: /var/cvsup/gsl/vcl/win/source/gdi/salbmp.cxx,v
retrieving revision 1.2
retrieving revision 1.2.346.1
diff -u -r1.2 -r1.2.346.1
--- salbmp.cxx	2001/06/28 13:13:55	1.2
+++ vcl/win/source/gdi/salbmp.cxx	2003/12/17 09:56:08	1.2.346.1
@@ -360,11 +360,11 @@
 
 HGLOBAL SalBitmap::ImplCreateDIB( const Size& rSize, USHORT nBits, const BitmapPalette& rPal )
 {
-	DBG_ASSERT( nBits == 1 || nBits == 4 || nBits == 8 || nBits == 24, "Unsupported BitCount!" );
+	DBG_ASSERT( nBits == 1 || nBits == 4 || nBits == 8 || nBits == 24 || nBitCount == 32, "Unsupported BitCount!" );
 
 	HGLOBAL hDIB = 0;
 
-	if ( rSize.Width() && rSize.Height() && ( nBits == 1 || nBits == 4 || nBits == 8 || nBits == 24 ) )
+	if ( rSize.Width() && rSize.Height() && ( nBits == 1 || nBits == 4 || nBits == 8 || nBits == 24 || nBits == 32 ) )
 	{
 		const ULONG 	nImageSize = AlignedWidth4Bytes( nBits * rSize.Width() ) * rSize.Height();
 		const USHORT	nColors = ( nBits <= 8 ) ? ( 1 << nBits ) : 0;
@@ -496,7 +496,9 @@
 								  pBIH->biBitCount == 8 ? BMP_FORMAT_8BIT_PAL :
 								  pBIH->biBitCount == 16 ? BMP_FORMAT_16BIT_TC_LSB_MASK :
 								  pBIH->biBitCount == 24 ? BMP_FORMAT_24BIT_TC_BGR :
-								  pBIH->biBitCount == 32 ? BMP_FORMAT_32BIT_TC_MASK : 0UL );
+								  //pBIH->biBitCount == 32 ? BMP_FORMAT_32BIT_TC_MASK : 0UL );
+                                  // #i22149# always assume BGRA for 32bit
+								  pBIH->biBitCount == 32 ? BMP_FORMAT_32BIT_TC_BGRA : 0UL );
 
 			if( BMP_SCANLINE_FORMAT( pBuffer->mnFormat ) )
 			{
