--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/btree/makefile.mk	2004-04-02 17:27:34.000000000 +0100
@@ -0,0 +1,29 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=btree
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/bt_compare.obj	\
+			$(SLO)$/bt_conv.obj	\
+			$(SLO)$/bt_curadj.obj	\
+			$(SLO)$/bt_cursor.obj	\
+			$(SLO)$/bt_delete.obj	\
+			$(SLO)$/bt_method.obj	\
+			$(SLO)$/bt_open.obj	\
+			$(SLO)$/bt_put.obj	\
+			$(SLO)$/bt_rec.obj	\
+			$(SLO)$/bt_reclaim.obj	\
+			$(SLO)$/bt_recno.obj	\
+			$(SLO)$/bt_rsearch.obj	\
+			$(SLO)$/bt_search.obj	\
+			$(SLO)$/bt_split.obj	\
+			$(SLO)$/bt_stat.obj	\
+			$(SLO)$/bt_upgrade.obj	\
+			$(SLO)$/bt_verify.obj	\
+			$(SLO)$/btree_auto.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/db/makefile.mk	2004-04-02 17:33:37.000000000 +0100
@@ -0,0 +1,33 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=db_src
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/crdel_auto.obj	\
+			$(SLO)$/crdel_rec.obj	\
+			$(SLO)$/db.obj		\
+			$(SLO)$/db_am.obj	\
+			$(SLO)$/db_auto.obj	\
+			$(SLO)$/db_cam.obj	\
+			$(SLO)$/db_conv.obj	\
+			$(SLO)$/db_dispatch.obj \
+			$(SLO)$/db_dup.obj	\
+			$(SLO)$/db_iface.obj	\
+			$(SLO)$/db_join.obj	\
+			$(SLO)$/db_meta.obj	\
+			$(SLO)$/db_method.obj	\
+			$(SLO)$/db_overflow.obj \
+			$(SLO)$/db_pr.obj	\
+			$(SLO)$/db_rec.obj	\
+			$(SLO)$/db_reclaim.obj	\
+			$(SLO)$/db_ret.obj	\
+			$(SLO)$/db_upg.obj	\
+			$(SLO)$/db_upg_opd.obj	\
+			$(SLO)$/db_vrfy.obj	\
+			$(SLO)$/db_vrfyutil.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/cxx/makefile.mk	2004-04-02 17:35:57.000000000 +0100
@@ -0,0 +1,18 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=cxx
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/cxx_app.obj	    \
+			$(SLO)$/cxx_except.obj	\
+			$(SLO)$/cxx_lock.obj	\
+			$(SLO)$/cxx_log.obj		\
+			$(SLO)$/cxx_mpool.obj	\
+			$(SLO)$/cxx_table.obj	\
+			$(SLO)$/cxx_txn.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/hash/makefile.mk	2004-04-02 17:37:20.000000000 +0100
@@ -0,0 +1,24 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=hash
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/hash.obj	\
+			$(SLO)$/hash_auto.obj	\
+			$(SLO)$/hash_conv.obj	\
+			$(SLO)$/hash_dup.obj	\
+			$(SLO)$/hash_func.obj	\
+			$(SLO)$/hash_meta.obj	\
+			$(SLO)$/hash_method.obj	\
+			$(SLO)$/hash_page.obj	\
+			$(SLO)$/hash_rec.obj	\
+			$(SLO)$/hash_reclaim.obj\
+			$(SLO)$/hash_stat.obj	\
+			$(SLO)$/hash_upgrade.obj\
+			$(SLO)$/hash_verify.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/common/makefile.mk	2004-04-02 17:38:21.000000000 +0100
@@ -0,0 +1,17 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=common
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/db_byteorder.obj	\
+			$(SLO)$/db_err.obj	\
+			$(SLO)$/db_getlong.obj	\
+			$(SLO)$/db_log2.obj	\
+			$(SLO)$/util_log.obj	\
+			$(SLO)$/util_sig.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/env/makefile.mk	2004-04-02 17:39:27.000000000 +0100
@@ -0,0 +1,17 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=env
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/db_salloc.obj	\
+			$(SLO)$/env_method.obj	\
+			$(SLO)$/env_open.obj	\
+			$(SLO)$/env_recover.obj	\
+			$(SLO)$/env_region.obj	\
+			$(SLO)$/db_shash.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/lock/makefile.mk	2004-04-02 17:40:20.000000000 +0100
@@ -0,0 +1,18 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=lock
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/lock.obj	\
+			$(SLO)$/lock_conflict.obj	\
+			$(SLO)$/lock_deadlock.obj	\
+			$(SLO)$/lock_method.obj	\
+			$(SLO)$/lock_region.obj	\
+			$(SLO)$/lock_stat.obj	\
+			$(SLO)$/lock_util.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/log/makefile.mk	2004-04-02 17:41:18.000000000 +0100
@@ -0,0 +1,21 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=log
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/log.obj	\
+			$(SLO)$/log_archive.obj	\
+			$(SLO)$/log_auto.obj	\
+			$(SLO)$/log_compare.obj	\
+			$(SLO)$/log_findckp.obj	\
+			$(SLO)$/log_get.obj	\
+			$(SLO)$/log_method.obj	\
+			$(SLO)$/log_put.obj	\
+			$(SLO)$/log_rec.obj	\
+			$(SLO)$/log_register.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/mp/makefile.mk	2004-04-02 17:42:24.000000000 +0100
@@ -0,0 +1,23 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=mp
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/mp_alloc.obj	\
+			$(SLO)$/mp_bh.obj	\
+			$(SLO)$/mp_fget.obj	\
+			$(SLO)$/mp_fopen.obj	\
+			$(SLO)$/mp_fput.obj	\
+			$(SLO)$/mp_fset.obj	\
+			$(SLO)$/mp_method.obj	\
+			$(SLO)$/mp_region.obj	\
+			$(SLO)$/mp_register.obj	\
+			$(SLO)$/mp_stat.obj	\
+			$(SLO)$/mp_sync.obj	\
+			$(SLO)$/mp_trickle.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/qam/makefile.mk	2004-04-02 17:43:30.000000000 +0100
@@ -0,0 +1,21 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=qam
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/qam.obj	\
+			$(SLO)$/qam_auto.obj	\
+			$(SLO)$/qam_conv.obj	\
+			$(SLO)$/qam_files.obj	\
+			$(SLO)$/qam_method.obj	\
+			$(SLO)$/qam_open.obj	\
+			$(SLO)$/qam_rec.obj	\
+			$(SLO)$/qam_stat.obj	\
+			$(SLO)$/qam_upgrade.obj	\
+			$(SLO)$/qam_verify.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/txn/makefile.mk	2004-04-02 17:44:17.000000000 +0100
@@ -0,0 +1,15 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=txn
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/txn.obj	\
+			$(SLO)$/txn_auto.obj	\
+			$(SLO)$/txn_rec.obj	\
+			$(SLO)$/txn_region.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/xa/makefile.mk	2004-04-02 17:45:22.000000000 +0100
@@ -0,0 +1,14 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=xa
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/xa.obj	\
+			$(SLO)$/xa_db.obj	\
+			$(SLO)$/xa_map.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/os_win32/makefile.mk	2004-04-02 17:48:54.000000000 +0100
@@ -0,0 +1,23 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=os_win32
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/os_abs.obj	\
+			$(SLO)$/os_dir.obj	\
+			$(SLO)$/os_errno.obj	\
+			$(SLO)$/os_fid.obj	\
+			$(SLO)$/os_finit.obj	\
+			$(SLO)$/os_map.obj	\
+			$(SLO)$/os_open.obj	\
+			$(SLO)$/os_rename.obj	\
+			$(SLO)$/os_seek.obj	\
+			$(SLO)$/os_sleep.obj	\
+			$(SLO)$/os_spin.obj	\
+			$(SLO)$/os_type.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/os/makefile.mk	2004-04-02 17:50:18.000000000 +0100
@@ -0,0 +1,23 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=os
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/os_alloc.obj	\
+			$(SLO)$/os_fsync.obj	\
+			$(SLO)$/os_handle.obj	\
+			$(SLO)$/os_method.obj	\
+			$(SLO)$/os_oflags.obj	\
+			$(SLO)$/os_region.obj	\
+			$(SLO)$/os_root.obj	\
+			$(SLO)$/os_rpath.obj	\
+			$(SLO)$/os_rw.obj	\
+			$(SLO)$/os_stat.obj	\
+			$(SLO)$/os_tmpdir.obj	\
+			$(SLO)$/os_unlink.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/mutex/makefile.mk	2004-04-02 17:50:55.000000000 +0100
@@ -0,0 +1,13 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=mutex
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= \
+			$(SLO)$/mut_tas.obj	\
+			$(SLO)$/mutex.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/dbm/makefile.mk	2004-04-02 17:51:34.000000000 +0100
@@ -0,0 +1,11 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=dbm
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= $(SLO)$/dbm.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/clib/makefile.mk	2004-04-02 17:52:09.000000000 +0100
@@ -0,0 +1,11 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=clib
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= $(SLO)$/strcasecmp.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/hsearch/makefile.mk	2004-04-02 17:53:01.000000000 +0100
@@ -0,0 +1,11 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db
+TARGET=hsearch
+
+.INCLUDE :  settings.mk
+
+INCPRE+=..$/build_win32 ../include
+SLOFILES= $(SLO)$/hsearch.obj
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/libdb_java/makefile.mk	2004-04-03 02:20:59.000000000 +0100
@@ -0,0 +1,28 @@
+PRJ=..$/..$/..$/..$/..
+
+PRJNAME=db_java
+TARGET=db_java
+
+.INCLUDE :  settings.mk
+
+INCPRE=..$/build_win32 ../include
+
+.IF "$(SOLAR_JAVA)"!=""
+SLOFILES= \
+			$(SLO)$/java_Db.obj	\
+			$(SLO)$/java_Dbc.obj	\
+			$(SLO)$/java_DbE.obj	\
+			$(SLO)$/java_DbLock.obj	\
+			$(SLO)$/java_DbLsn.obj	\
+			$(SLO)$/java_Dbt.obj	\
+			$(SLO)$/java_DbTxn.obj	\
+			$(SLO)$/java_info.obj	\
+			$(SLO)$/java_locked.obj	\
+			$(SLO)$/java_util.obj
+.ELSE
+dummy:
+	@echo "Nothing to build for OS $(OS)"
+
+.ENDIF
+
+.INCLUDE :  target.mk
--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ misc/build/db-3.2.9/makefile.mk	2004-04-03 02:23:51.000000000 +0100
@@ -0,0 +1,52 @@
+PRJ=..$/..$/..$/..
+PRJNAME=db
+TARGET=db
+
+.INCLUDE :  settings.mk
+
+CHILDREN=btree clib common cxx db dbm env hash hsearch \
+	 lock log mp mutex os os_win32 qam txn xa
+
+LIB1TARGET= $(SLB)$/$(TARGET).lib
+LIB1FILES=  $(SLB)$/btree.lib	    \
+            $(SLB)$/clib.lib	    \
+            $(SLB)$/common.lib	    \
+            $(SLB)$/cxx.lib	    \
+            $(SLB)$/db_src.lib	    \
+            $(SLB)$/dbm.lib	    \
+            $(SLB)$/env.lib	    \
+            $(SLB)$/hash.lib	    \
+            $(SLB)$/hsearch.lib	    \
+            $(SLB)$/lock.lib	    \
+            $(SLB)$/log.lib	    \
+            $(SLB)$/mp.lib	    \
+            $(SLB)$/mutex.lib	    \
+            $(SLB)$/os.lib	    \
+            $(SLB)$/os_win32.lib    \
+            $(SLB)$/qam.lib	    \
+            $(SLB)$/txn.lib	    \
+            $(SLB)$/xa.lib
+
+SHL1LIBS=$(LIB1TARGET)
+SHL1TARGET	=	libdb32
+SHL1IMPLIB	=	$(TARGET)
+SHL1DEF 	=	build_win32$/libdb.def
+SHL1LIBS	=	$(SLB)$/$(TARGET).lib
+SHL1DEPS	=	$(CHILDREN)
+DEF1NAME=$(SHL1TARGET)
+
+.IF "$(SOLAR_JAVA)"!=""
+# untested by me ...
+CHILDREN+=db_java
+SHL2LIBS=$(SLB)$/db_java.lib
+SHL2TARGET=libdb_java32
+SHL2IMPLIB=db_java
+SHL2LIBS=$(SLB)$/db_java.lib
+SHL2DEPS	=	$(CHILDREN)
+DEF2NAME=$(SHL2TARGET)
+.ENDIF
+
+$(CHILDREN) .PHONY :
+	+cd $@ ; $(MAKECMD) subdmake=true $(MFLAGS) $(CALLMACROS)
+
+.INCLUDE :  target.mk
