--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ /foo/common/makefile.mk	2004-04-05 17:40:56.000000000 +0100
@@ -0,0 +1,49 @@
+PRJ=..$/..$/..$/..$/..$/..
+
+PRJNAME=icuuc
+TARGET=icuuc
+
+.INCLUDE :  settings.mk
+
+# INCPRE=..$/build_win32 ../include
+
+CDEFS+=-DU_COMMON_IMPLEMENTATION
+SLOFILES= \
+	$(SLO)$/putil.obj $(SLO)$/uobject.obj $(SLO)$/locmap.obj \
+	$(SLO)$/mutex.obj $(SLO)$/umutex.obj $(SLO)$/udata.obj \
+	$(SLO)$/ucmndata.obj $(SLO)$/udatamem.obj $(SLO)$/umapfile.obj \
+	$(SLO)$/filestrm.obj $(SLO)$/uresbund.obj $(SLO)$/uresdata.obj \
+	$(SLO)$/resbund.obj $(SLO)$/cwchar.obj $(SLO)$/uloc.obj \
+	$(SLO)$/locid.obj $(SLO)$/uhash.obj $(SLO)$/uhash_us.obj \
+	$(SLO)$/ucnv.obj $(SLO)$/ucnv_bld.obj $(SLO)$/ucnv_cb.obj \
+	$(SLO)$/ucnv_cnv.obj $(SLO)$/ucnv_err.obj $(SLO)$/ucnv_io.obj \
+	$(SLO)$/ucnvlat1.obj $(SLO)$/ucnv_u7.obj $(SLO)$/ucnv_u8.obj \
+	$(SLO)$/ucnv_u16.obj $(SLO)$/ucnv_u32.obj $(SLO)$/ucnvmbcs.obj \
+	$(SLO)$/ucnv2022.obj $(SLO)$/ucnvhz.obj $(SLO)$/ucnv_lmb.obj \
+	$(SLO)$/ucnvscsu.obj $(SLO)$/ucnvbocu.obj $(SLO)$/ucnvisci.obj \
+	$(SLO)$/unistr.obj $(SLO)$/utf_impl.obj $(SLO)$/ustring.obj \
+	$(SLO)$/ustrcase.obj $(SLO)$/cstring.obj $(SLO)$/ustrfmt.obj \
+	$(SLO)$/ustrtrns.obj $(SLO)$/normlzr.obj $(SLO)$/unorm.obj \
+	$(SLO)$/chariter.obj $(SLO)$/schriter.obj $(SLO)$/uchriter.obj \
+	$(SLO)$/uiter.obj $(SLO)$/uchar.obj $(SLO)$/uprops.obj \
+	$(SLO)$/bidi.obj $(SLO)$/ubidi.obj $(SLO)$/ubidiwrt.obj \
+	$(SLO)$/ubidiln.obj $(SLO)$/ushape.obj $(SLO)$/unames.obj \
+	$(SLO)$/ucln_cmn.obj $(SLO)$/uscript.obj $(SLO)$/usc_impl.obj \
+	$(SLO)$/umemstrm.obj $(SLO)$/ucmp8.obj $(SLO)$/uvector.obj \
+	$(SLO)$/digitlst.obj $(SLO)$/brkiter.obj $(SLO)$/brkdict.obj \
+	$(SLO)$/ubrk.obj $(SLO)$/dbbi.obj $(SLO)$/dbbi_tbl.obj \
+	$(SLO)$/rbbi.obj $(SLO)$/rbbidata.obj $(SLO)$/rbbinode.obj \
+	$(SLO)$/rbbirb.obj $(SLO)$/rbbiscan.obj $(SLO)$/rbbisetb.obj \
+	$(SLO)$/rbbistbl.obj $(SLO)$/rbbitblb.obj $(SLO)$/unicode.obj \
+	$(SLO)$/convert.obj $(SLO)$/utrie.obj $(SLO)$/uset.obj \
+	$(SLO)$/cmemory.obj $(SLO)$/unifilt.obj $(SLO)$/unifunct.obj \
+	$(SLO)$/uniset.obj $(SLO)$/upropset.obj $(SLO)$/usetiter.obj \
+	$(SLO)$/util.obj $(SLO)$/uenum.obj
+
+SHL1LIBS=$(TARGET).lib
+SHL1TARGET = lib$(TARGET)
+SHL1IMPLIB = $(TARGET)
+SHL1LIBS   = $(SLB)$/$(TARGET).lib
+DEF1NAME = $(SHL1TARGET)
+
+.INCLUDE :  target.mk

--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ /foo/i18n/makefile.mk	2004-04-05 18:56:21.000000000 +0100
@@ -0,0 +1,41 @@
+PRJ=..$/..$/..$/..$/..$/..
+
+PRJNAME=icui18n
+TARGET=icui18n
+
+.INCLUDE :  settings.mk
+
+INCPRE+=../common
+CDEFS+=-DU_I18N_IMPLEMENTATION
+SLOFILES= \
+	$(SLO)$/ucln_in.obj $(SLO)$/fmtable.obj $(SLO)$/format.obj \
+	$(SLO)$/msgfmt.obj $(SLO)$/umsg.obj $(SLO)$/numfmt.obj \
+	$(SLO)$/unum.obj $(SLO)$/decimfmt.obj $(SLO)$/dcfmtsym.obj \
+	$(SLO)$/choicfmt.obj $(SLO)$/datefmt.obj $(SLO)$/smpdtfmt.obj \
+	$(SLO)$/dtfmtsym.obj $(SLO)$/udat.obj $(SLO)$/ucal.obj \
+	$(SLO)$/calendar.obj $(SLO)$/gregocal.obj $(SLO)$/timezone.obj \
+	$(SLO)$/simpletz.obj $(SLO)$/sortkey.obj $(SLO)$/bocsu.obj \
+	$(SLO)$/coleitr.obj $(SLO)$/coll.obj $(SLO)$/ucoleitr.obj \
+	$(SLO)$/ucol.obj $(SLO)$/ucol_bld.obj $(SLO)$/ucol_cnt.obj \
+	$(SLO)$/ucol_elm.obj $(SLO)$/ucol_tok.obj $(SLO)$/ucol_wgt.obj \
+	$(SLO)$/tblcoll.obj $(SLO)$/strmatch.obj $(SLO)$/usearch.obj \
+	$(SLO)$/search.obj $(SLO)$/stsearch.obj $(SLO)$/unifltlg.obj \
+	$(SLO)$/translit.obj $(SLO)$/utrans.obj $(SLO)$/cpdtrans.obj \
+	$(SLO)$/hextouni.obj $(SLO)$/rbt.obj $(SLO)$/rbt_data.obj \
+	$(SLO)$/rbt_pars.obj $(SLO)$/rbt_rule.obj $(SLO)$/rbt_set.obj \
+	$(SLO)$/nultrans.obj $(SLO)$/remtrans.obj $(SLO)$/titletrn.obj \
+	$(SLO)$/tolowtrn.obj $(SLO)$/toupptrn.obj $(SLO)$/anytrans.obj \
+	$(SLO)$/name2uni.obj $(SLO)$/uni2name.obj $(SLO)$/unitohex.obj \
+	$(SLO)$/nortrans.obj $(SLO)$/quant.obj $(SLO)$/transreg.obj \
+	$(SLO)$/nfrs.obj $(SLO)$/nfrule.obj $(SLO)$/nfsubs.obj \
+	$(SLO)$/rbnf.obj $(SLO)$/esctrn.obj $(SLO)$/unesctrn.obj \
+	$(SLO)$/funcrepl.obj $(SLO)$/strrepl.obj $(SLO)$/tridpars.obj \
+	$(SLO)$/caniter.obj $(SLO)$/ucurr.obj
+
+SHL1LIBS=$(TARGET).lib
+SHL1TARGET = lib$(TARGET)
+SHL1IMPLIB = $(TARGET)
+SHL1LIBS   = $(SLB)$/$(TARGET).lib
+DEF1NAME = $(SHL1TARGET)
+
+.INCLUDE :  target.mk

--- /dev/null	2003-09-23 18:59:22.000000000 +0100
+++ /foo/layout/makefile.mk	2004-04-05 19:06:18.000000000 +0100
@@ -0,0 +1,77 @@
+PRJ=..$/..$/..$/..$/..$/..
+
+PRJNAME=icule
+TARGET=icule
+
+.INCLUDE :  settings.mk
+
+INCPRE+=../common
+CDEFS+=-DU_LAYOUT_IMPLEMENTATION
+SLOFILES= \
+	    $(SLO)$/loengine.obj \
+	    $(SLO)$/OpenTypeLayoutEngine.obj \
+	    $(SLO)$/ThaiLayoutEngine.obj \
+	    $(SLO)$/ThaiShaping.obj \
+	    $(SLO)$/ThaiStateTables.obj \
+	    $(SLO)$/ArabicLayoutEngine.obj \
+	    $(SLO)$/GXLayoutEngine.obj \
+	    $(SLO)$/IndicLayoutEngine.obj \
+	    $(SLO)$/LayoutEngine.obj \
+	    $(SLO)$/ContextualGlyphSubstProc.obj \
+	    $(SLO)$/IndicRearrangementProcessor.obj \
+	    $(SLO)$/LigatureSubstProc.obj \
+	    $(SLO)$/LookupTables.obj \
+	    $(SLO)$/MorphTables.obj \
+	    $(SLO)$/NonContextualGlyphSubstProc.obj \
+	    $(SLO)$/SegmentArrayProcessor.obj \
+	    $(SLO)$/SegmentSingleProcessor.obj \
+	    $(SLO)$/SimpleArrayProcessor.obj \
+	    $(SLO)$/SingleTableProcessor.obj \
+	    $(SLO)$/StateTableProcessor.obj \
+	    $(SLO)$/SubtableProcessor.obj \
+	    $(SLO)$/TrimmedArrayProcessor.obj \
+	    $(SLO)$/AlternateSubstSubtables.obj \
+	    $(SLO)$/AnchorTables.obj \
+	    $(SLO)$/ArabicLigatureData.obj \
+	    $(SLO)$/ArabicShaping.obj \
+	    $(SLO)$/ClassDefinitionTables.obj \
+	    $(SLO)$/ContextualSubstSubtables.obj \
+	    $(SLO)$/CoverageTables.obj \
+	    $(SLO)$/CursiveAttachmentSubtables.obj \
+	    $(SLO)$/DeviceTables.obj \
+	    $(SLO)$/Features.obj \
+	    $(SLO)$/GDEFMarkFilter.obj \
+	    $(SLO)$/GlyphDefinitionTables.obj \
+	    $(SLO)$/GlyphIterator.obj \
+	    $(SLO)$/GlyphPosnLookupProc.obj \
+	    $(SLO)$/GlyphPositioningTables.obj \
+	    $(SLO)$/GlyphSubstLookupProc.obj \
+	    $(SLO)$/GlyphSubstitutionTables.obj \
+	    $(SLO)$/HebrewLigatureData.obj \
+	    $(SLO)$/HebrewShaping.obj \
+	    $(SLO)$/IndicClassTables.obj \
+	    $(SLO)$/IndicReordering.obj \
+	    $(SLO)$/LigatureSubstSubtables.obj \
+	    $(SLO)$/LookupProcessor.obj \
+	    $(SLO)$/Lookups.obj \
+	    $(SLO)$/MarkArrays.obj \
+	    $(SLO)$/MarkToBasePosnSubtables.obj \
+	    $(SLO)$/MarkToLigaturePosnSubtables.obj \
+	    $(SLO)$/MarkToMarkPosnSubtables.obj \
+	    $(SLO)$/MultipleSubstSubtables.obj \
+	    $(SLO)$/OpenTypeUtilities.obj \
+	    $(SLO)$/PairPositioningSubtables.obj \
+	    $(SLO)$/ScriptAndLanguage.obj \
+	    $(SLO)$/ScriptAndLanguageTags.obj \
+	    $(SLO)$/SinglePositioningSubtables.obj \
+	    $(SLO)$/SingleSubstitutionSubtables.obj \
+	    $(SLO)$/SubstitutionLookups.obj \
+	    $(SLO)$/ValueRecords.obj
+
+SHL1LIBS=$(TARGET).lib
+SHL1TARGET = lib$(TARGET)
+SHL1IMPLIB = $(TARGET)
+SHL1LIBS   = $(SLB)$/$(TARGET).lib
+DEF1NAME = $(SHL1TARGET)
+
+.INCLUDE :  target.mk
