Index: offmgr/source/offapp/intro/ooo.src
===================================================================
RCS file: /cvs/framework/offmgr/source/offapp/intro/ooo.src,v
retrieving revision 1.44.16.3
diff -u -p -u -r1.44.16.3 ooo.src
--- offmgr/source/offapp/intro/ooo.src	22 Sep 2003 13:36:28 -0000	1.44.16.3
+++ offmgr/source/offapp/intro/ooo.src	23 Sep 2003 09:57:00 -0000
@@ -77,9 +77,9 @@
 #include "intro.hrc"
 
 #ifdef BUILD_SPECIAL
-#define OOO_VENDOR "Sun Microsystems Inc."
+#define OOO_VENDOR "Novell Ximian Services, Inc."
 #define OOO_LICENSE "LGPL"
-#define OOO_INTRO "openintro_sun.bmp"
+#define OOO_INTRO "openintro_ximian.bmp"
 #else
 //#define OOO_VENDOR "my company"
 #define OOO_LICENSE "LGPL"
@@ -88,7 +88,7 @@
 
 Bitmap RID_DEFAULT_ABOUT_BMP_LOGO
 {
-	File = "openabout.bmp" ;
+	File = "openabout_ximian.bmp" ;
 };
 
 Bitmap RID_DEFAULTINTRO
