Index: vcl/util/makefile.mk
===================================================================
--- vcl/util/makefile.mk.orig	2004-08-13 16:11:36.573807768 +0200
+++ vcl/util/makefile.mk	2004-08-13 16:35:20.939271304 +0200
@@ -320,7 +320,7 @@ SHL2STDLIBS += -lXext -lSM -lICE -lX11
 .IF "$(XINERAMA_LINK)" == "dynamic"
 SHL2STDLIBS += -lXinerama
 .ELSE
-SHL2STDLIBS += -Wl,-Bstatic -lXinerama -Wl,-Bdynamic 
+SHL2STDLIBS += -Wl,-Bstatic -lXinerama_pic -Wl,-Bdynamic 
 .ENDIF          # $(XINERAMA_LINK)
 .ENDIF          # $(USE_XINERAMA)
 SHL2STDLIBS += -lXext -lSM -lICE -lX11
Index: vcl/unx/source/app/saldisp.cxx
===================================================================
--- vcl/unx/source/app/saldisp.cxx.orig	2004-08-13 15:51:32.208899064 +0200
+++ vcl/unx/source/app/saldisp.cxx	2004-08-13 16:34:50.245937400 +0200
@@ -2779,7 +2779,7 @@ void SalDisplay::InitXinerama()
     }
     osl_unloadModule( hModule );
 #else
-#if defined( X86 ) || defined( MACOSX )
+#if defined( LINUX ) || defined( MACOSX )
     if( XineramaIsActive( pDisp_ ) )
     {
         int nFramebuffers = 1;
