Index: vcl/util/makefile.mk
===================================================================
--- vcl/util/makefile.mk.orig	2004-07-30 22:30:00.284626920 +0200
+++ vcl/util/makefile.mk	2004-07-30 22:35:29.460584568 +0200
@@ -317,9 +317,7 @@ SHL2STDLIBS+=$(LIBSN_LIBS)
 SHL2STDLIBS += -lXext -lSM -lICE -lX11
 # Others
 .ELSE           # "$(OS)"=="SOLARIS"
-.IF "$(CPU)" == "I"
-SHL2STDLIBS += -Wl,-Bstatic -lXinerama -Wl,-Bdynamic 
-.ENDIF 			# "$(CPU)=="I"
+SHL2STDLIBS += -Wl,-Bstatic -lXinerama_pic -Wl,-Bdynamic 
 SHL2STDLIBS += -lXext -lSM -lICE -lX11
 .ENDIF          # "$(OS)"=="SOLARIS"
 
Index: vcl/unx/source/app/saldisp.cxx
===================================================================
--- vcl/unx/source/app/saldisp.cxx.orig	2004-07-30 22:07:14.879200096 +0200
+++ vcl/unx/source/app/saldisp.cxx	2004-07-30 22:33:50.905567208 +0200
@@ -91,7 +91,7 @@
 
 #include <X11/Xatom.h>
 #ifndef SOLARIS
-#if defined(X86) || defined(MACOSX)
+#if defined(LINUX) || defined(MACOSX)
 #include <X11/extensions/Xinerama.h>
 #endif
 #endif
@@ -2778,7 +2778,7 @@ void SalDisplay::InitXinerama()
     }
     osl_unloadModule( hModule );
 #else
-#if defined( X86 ) || defined( MACOSX )
+#if defined( LINUX ) || defined( MACOSX )
     if( XineramaIsActive( pDisp_ ) )
     {
         int nFramebuffers = 1;
