diff -purN src680-m60/scp2/prj/build.lst cws_src680_pj10/scp2/prj/build.lst
--- scp2/prj/build.lst	2004-10-25 10:13:06.000000000 +0200
+++ scp2/prj/build.lst	2004-11-03 10:02:07.000000000 +0100
@@ -20,4 +20,4 @@ cp    scp2\source\testtool        nmake 
 cp    scp2\source\writer          nmake    -    all    cp_writer NULL
 cp    scp2\source\xsltfilter      nmake    -    all    cp_xsltfilter NULL
 cp    scp2\source\winexplorerext  nmake    -    all    cp_winexplorerext NULL
-cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_calc cp_cde cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_lingu cp_mailcap cp_math cp_ooo cp_python cp_testtool cp_writer cp_xsltfilter cp_winexplorerext NULL
+cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_calc cp_canvas cp_cde cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_lingu cp_mailcap cp_math cp_ooo cp_python cp_testtool cp_writer cp_xsltfilter cp_winexplorerext NULL
