Index: rsc/source/rscpp/cpp1.c
===================================================================
RCS file: /cvs/gsl/rsc/source/rscpp/cpp1.c,v
retrieving revision 1.4
diff -u -p -u -r1.4 cpp1.c
--- rsc/source/rscpp/cpp1.c	2 Jul 2003 14:52:27 -0000	1.4
+++ rsc/source/rscpp/cpp1.c	10 Sep 2003 16:12:10 -0000
@@ -419,10 +419,11 @@ nRunde++;
 /* alt:        fclose(stdout); */
 
 #if OSL_DEBUG_LEVEL > 1
-        fclose( pDefOut );
+		if( pDefOut != stdout )
+            fclose( pDefOut );
 #endif
-        fclose( pCppOut );
-        fclose( pCppIn  );
+		if( pCppOut != stdout )
+            fclose( pCppOut );
 
 
         if (errors > 0) {
