--- /dev/null	
+++ accessibility/unx/makefile.mk	
@@ -0,0 +1,32 @@
+PRJ=..
+PRJNAME=accessibility
+# Version
+A11Y_MAJOR=1
+TARGET=a11y
+ENABLE_EXCEPTIONS=TRUE
+NO_BSYMBOLIC=TRUE
+
+
+.INCLUDE: settings.mk
+
+COMPILER_WARN_ALL=TRUE
+# no "lib" prefix
+DLLPRE =
+
+SLOFILES=\
+	$(SLO)$/atkbridge.obj
+
+SHL1NOCHECK=TRUE
+SHL1TARGET=$(TARGET)$(A11Y_MAJOR).uno
+SHL1DEF=$(MISC)$/$(SHL1TARGET).def
+SHL1LIBS=$(SLB)$/$(TARGET).lib
+SHL1IMPLIB=i$(TARGET)
+SHL1STDLIBS=\
+	$(CPPUHELPERLIB) \
+	$(CPPULIB) \
+	$(SALLIB)  \
+	$(UNOTOOLSLIB)
+#	$(TOOLSLIB)
+
+.INCLUDE: target.mk
+
