--- /dev/null	
+++ accessibility/unx/atkwrapper.hxx	
@@ -0,0 +1,28 @@
+#ifndef __ATK_WRAPPER_H__
+#define __ATK_WRAPPER_H__
+
+#include <atkbridge.hxx>
+
+namespace a11y { extern "C" {
+
+typedef struct _AtkObjectWrapper       AtkObjectWrapper;
+typedef struct _AtkObject_WrapperClass AtkObjectWrapperClass;
+
+struct _AtkObjectWrapper
+{
+	AtkObject parent;
+	accessibilty::XAccessible *accessible;
+};
+
+struct _AtkObjectWrapperClass
+{
+	AtkObjectClass parent_class;x
+};
+
+GType atk_object_wrapper_get_type (void) G_GNUC_CONST;
+
+AtkObjectWrapper *atk_object_wrapper_construct( const Reference< accessibilty::XAccessible > &rAccessible );
+
+}; };
+
+#endif /* __ATK_WRAPPER_H__ */
