Index: xmloff/inc/xmloff/xmltoken.hxx
===================================================================
RCS file: /cvs/xml/xmloff/inc/xmloff/xmltoken.hxx,v
retrieving revision 1.9
retrieving revision 1.6.14.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.9 -r1.6.14.2
--- xmloff/inc/xmloff/xmltoken.hxx	28 Aug 2007 13:49:45 -0000	1.9
+++ xmloff/inc/xmloff/xmltoken.hxx	1 Oct 2007 15:28:50 -0000	1.6.14.2
@@ -732,6 +732,7 @@
         XML_FILTER_NAME,
         XML_FILTER_OPTIONS,
         XML_FILTER_OR,
+        XML_FILTER_SET_ITEM,
         XML_FIRST_DATE_TIME,
         XML_FIRST_PAGE,
         XML_FIRST_PAGE_NUMBER,
Index: xmloff/source/core/xmltoken.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/core/xmltoken.cxx,v
retrieving revision 1.112
retrieving revision 1.109.14.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.112 -r1.109.14.2
--- xmloff/source/core/xmltoken.cxx	28 Aug 2007 13:34:08 -0000	1.112
+++ xmloff/source/core/xmltoken.cxx	1 Oct 2007 15:28:56 -0000	1.109.14.2
@@ -740,6 +740,7 @@
         TOKEN( "filter-name",                     XML_FILTER_NAME ),
         TOKEN( "filter-options",                  XML_FILTER_OPTIONS ),
         TOKEN( "filter-or",                       XML_FILTER_OR ),
+        TOKEN( "filter-set-item",                 XML_FILTER_SET_ITEM ),
         TOKEN( "first-date-time",                 XML_FIRST_DATE_TIME ),
         TOKEN( "first-page",                      XML_FIRST_PAGE ),
         TOKEN( "first-page-number",               XML_FIRST_PAGE_NUMBER ),
