? sfx2/sfx2.vpj
Index: sfx2/source/dialog/filedlghelper.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/dialog/filedlghelper.cxx,v
retrieving revision 1.138
retrieving revision 1.136.42.2
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.138 -r1.136.42.2
--- sfx2/source/dialog/filedlghelper.cxx	4 Feb 2008 14:19:33 -0000	1.138
+++ sfx2/source/dialog/filedlghelper.cxx	15 Feb 2008 00:45:45 -0000	1.136.42.2
@@ -691,6 +691,11 @@
 {
 	sal_Bool operator() ( const SfxFilter* _pFilter )
 	{
+        if (_pFilter->GetFilterName().EqualsAscii("MS Excel 97"))
+            // temporary hack to enable password protection for Excel 97.  Is
+            // there a better way to enable password protection of a filter?
+            return true;
+
         return  _pFilter && _pFilter->IsOwnFormat()
 			&&	_pFilter->UsesStorage()
 			&&	( SOFFICE_FILEFORMAT_60 <= _pFilter->GetVersion() );
