diff --git sc/inc/compiler.hrc sc/inc/compiler.hrc
index 92b7e82..22d77cb 100644
--- sc/inc/compiler.hrc
+++ sc/inc/compiler.hrc
@@ -39,25 +39,26 @@
 #define SC_OPCODE_STOP                2
 #define SC_OPCODE_EXTERNAL            3
 #define SC_OPCODE_NAME                4
-#define SC_OPCODE_IF                  5     /* jump commands */
-#define SC_OPCODE_CHOSE               6
-#define SC_OPCODE_OPEN                7     /* parentheses and separators */
-#define SC_OPCODE_CLOSE               8
-#define SC_OPCODE_SEP                 9
-#define SC_OPCODE_MISSING            10     /* special OpCodes */
-#define SC_OPCODE_BAD                11
-#define SC_OPCODE_SPACES             12
-#define SC_OPCODE_MAT_REF            13
-#define SC_OPCODE_DB_AREA            14     /* additional access operators */
-#define SC_OPCODE_MACRO              15
-#define SC_OPCODE_COL_ROW_NAME       16
-#define SC_OPCODE_COL_ROW_NAME_AUTO  17
-#define SC_OPCODE_PERCENT_SIGN       18     /* operator _follows_ value */
-#define SC_OPCODE_ARRAY_OPEN         19
-#define SC_OPCODE_ARRAY_CLOSE        20
-#define SC_OPCODE_ARRAY_ROW_SEP      21
-#define SC_OPCODE_ARRAY_COL_SEP      22     /* some convs use sep != col_sep */
-#define SC_OPCODE_STOP_DIV           23
+#define SC_OPCODE_EXTERNAL_NAME       5
+#define SC_OPCODE_IF                  6     /* jump commands */
+#define SC_OPCODE_CHOSE               7
+#define SC_OPCODE_OPEN                8     /* parentheses and separators */
+#define SC_OPCODE_CLOSE               9
+#define SC_OPCODE_SEP                10
+#define SC_OPCODE_MISSING            11     /* special OpCodes */
+#define SC_OPCODE_BAD                12
+#define SC_OPCODE_SPACES             13
+#define SC_OPCODE_MAT_REF            14
+#define SC_OPCODE_DB_AREA            15     /* additional access operators */
+#define SC_OPCODE_MACRO              16
+#define SC_OPCODE_COL_ROW_NAME       17
+#define SC_OPCODE_COL_ROW_NAME_AUTO  18
+#define SC_OPCODE_PERCENT_SIGN       19     /* operator _follows_ value */
+#define SC_OPCODE_ARRAY_OPEN         20
+#define SC_OPCODE_ARRAY_CLOSE        21
+#define SC_OPCODE_ARRAY_ROW_SEP      22
+#define SC_OPCODE_ARRAY_COL_SEP      23     /* some convs use sep != col_sep */
+#define SC_OPCODE_STOP_DIV           24
 
 /*** error constants #... ***/
 #define SC_OPCODE_START_ERRORS       30
diff --git sc/inc/compiler.hxx sc/inc/compiler.hxx
index 8036115..047ff2c 100644
--- sc/inc/compiler.hxx
+++ sc/inc/compiler.hxx
@@ -45,6 +45,7 @@
 #include <unotools/charclass.hxx>
 #include <rtl/ustrbuf.hxx>
 #include <com/sun/star/uno/Sequence.hxx>
+#include <vector>
 
 namespace com { namespace sun { namespace star {
     namespace sheet {
@@ -150,6 +151,10 @@ public:
             bool        bHasForceArray;
         } sbyte;
         ComplRefData aRef;
+        struct {
+            sal_uInt16  nFileId;
+            sal_Unicode cName[MAXSTRLEN+1];
+        } extname;
         ScMatrix*    pMat;
         USHORT       nIndex;                // index into name collection
         sal_Unicode  cStr[ MAXSTRLEN+1 ];   // string (up to 255 characters + 0)
@@ -179,6 +184,7 @@ public:
     void SetDouble( double fVal );
     void SetInt( int nVal );
     void SetName( USHORT n );
+    void SetExternalName( sal_uInt16 nFileId, const String& rName );
     void SetMatrix( ScMatrix* p );
     void SetExternal(const sal_Unicode* pStr);
     // These methods are ok to use, reference count not cleared.
@@ -227,6 +233,16 @@ public:
                                    xub_StrLen nSrcPos,
                                    const CharClass* pCharClass) const = 0;
 
+        /** 
+         * Parse the symbol string and pick up the file name and the external 
+         * range name. 
+         *
+         * @return true on successful parse, or false otherwise.
+         */
+        virtual bool parseExternalName( const String& rSymbol, String& rFile, String& rName ) const = 0;
+
+        virtual String makeExternalNameStr( const String& rFile, const String& rName ) const = 0;
+
         enum SpecialSymbolType
         {
             /** 
@@ -443,6 +459,7 @@ private:
     BOOL IsDoubleReference( const String& );
     BOOL IsMacro( const String& );
     BOOL IsNamedRange( const String& );
+    bool IsExternalNamedRange( const String& rSymbol );
     BOOL IsDBRange( const String& );
     BOOL IsColRowName( const String& );
     BOOL IsBoolean( const String& );
diff --git sc/inc/document.hxx sc/inc/document.hxx
index 8a648d3..9b373cc 100644
--- sc/inc/document.hxx
+++ sc/inc/document.hxx
@@ -92,6 +92,7 @@ class ScDocOptions;
 class ScDocumentPool;
 class ScDrawLayer;
 class ScExtDocOptions;
+class ScExternalRefManager;
 class ScFormulaCell;
 class ScMarkData;
 class ScOutlineTable;
@@ -281,6 +282,7 @@ private:
 	ScFieldEditEngine*	pCacheFieldEditEngine;
 
     ::std::auto_ptr<ScDocProtection> pDocProtection;
+    ::std::auto_ptr<ScExternalRefManager> pExternalRefMgr;
 	String              aDocName;                       // opt: Dokumentname
 	String              aDocCodeName;                       // opt: Dokumentname
 	ScRangePairListRef	xColNameRanges;
@@ -624,6 +626,8 @@ SC_DLLPUBLIC	ScDBCollection*	GetDBCollection() const;
 									const String& aFileName,
 									const String& aTabName );
 
+    ScExternalRefManager* GetExternalRefManager();
+
     /** Creates a new sheet, and makes it linked to the specified sheet in an external document.
         @param rnTab  (out-param) Returns the sheet index, if sheet could be inserted).
         @return  TRUE = Sheet created, rnTab contains valid sheet index. */
diff --git sc/inc/externalrefmgr.hxx sc/inc/externalrefmgr.hxx
new file mode 100644
index 0000000..b5427ec
--- /dev/null
+++ sc/inc/externalrefmgr.hxx
@@ -0,0 +1,164 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: document.hxx,v $
+ * $Revision: 1.112 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+#ifndef SC_EXTERNALREFMGR_HXX
+#define SC_EXTERNALREFMGR_HXX
+
+#include "sfx2/objsh.hxx"
+#include "sfx2/lnkbase.hxx"
+#include "global.hxx"
+#include "address.hxx"
+
+#include <hash_map>
+#include <boost/shared_ptr.hpp>
+#include <vector>
+
+class ScDocument;
+class ScToken;
+class ScTokenArray;
+class String;
+class SfxObjectShellRef;
+class Window;
+
+class ScExternalRefLink : public ::sfx2::SvBaseLink
+{
+public:
+    ScExternalRefLink(ScDocument* pDoc, const String& rFile, const String& rFilter);
+    virtual ~ScExternalRefLink();
+
+    virtual void Closed();
+    virtual void DataChanged(const String& rMimeType, const ::com::sun::star::uno::Any & rValue);
+    virtual void Edit(Window* pParent, const Link& rEndEditHdl);
+
+private:
+    ScExternalRefLink(); // disabled
+    ScExternalRefLink(const ScExternalRefLink&); // disabled
+
+    DECL_LINK(EndEditHdl, void*);
+
+    String      maFileName;
+    String      maFilterName;
+    ScDocument* mpDoc;
+};
+
+// ============================================================================
+
+class ScExternalRefManager
+{
+    struct AddressHash
+    {
+        size_t operator()(const ScAddress& rAddr) const
+        {
+            // dumb hashing - how do we properly hash a cell address?
+            return rAddr.Tab() + rAddr.Col() + rAddr.Row();
+        }
+    };
+
+    struct RangeHash
+    {
+        size_t operator()(const ScRange& rRange) const
+        {
+            const ScAddress& s = rRange.aStart;
+            const ScAddress& e = rRange.aEnd;
+            return s.Tab() + s.Col() + s.Row() + e.Tab() + e.Col() + e.Row();
+        }
+    };
+
+    typedef ::boost::shared_ptr<ScToken>        TokenRef;
+    typedef ::boost::shared_ptr<ScTokenArray>   TokenArrayRef;
+
+    typedef ::std::hash_map<String, SfxObjectShellRef, ScStringHashCode, ::std::equal_to<String> >  DocShellMap;
+    typedef ::std::hash_map<ScAddress, TokenRef, AddressHash, ::std::equal_to<ScAddress> >          SingleTokenMap;
+    typedef ::std::hash_map<ScRange, TokenArrayRef, RangeHash, ::std::equal_to<ScRange> >           DoubleTokenMap;
+    typedef ::std::hash_map<String, TokenArrayRef, ScStringHashCode, ::std::equal_to<String> >      RangeNameMap;
+
+    /** 
+     * Cached content of a single external document
+     */
+    struct DocCache
+    {
+        SingleTokenMap  maSingleTokens;
+        DoubleTokenMap  maDoubleTokens;
+        RangeNameMap    maRangeNames;
+    };
+    typedef ::boost::shared_ptr<DocCache> DocCacheRef;
+
+    typedef ::std::hash_map<String, DocCacheRef, ScStringHashCode, ::std::equal_to<String> >   DocCacheMap;
+
+public:
+    explicit ScExternalRefManager(ScDocument* pDoc);
+    ~ScExternalRefManager();
+
+    ScToken* getSingleRefToken(const String& rFile, const ScAddress& rCell);
+    ScTokenArray* getDoubleRefTokens(const String& rFile, const ScRange& rRange);
+
+    /** 
+     * Get an array of tokens corresponding with a specified name in a 
+     * specified file. 
+     * 
+     * @return a new instance of ScTokenArray.  The caller must manage its 
+     *         life cycle.
+     */
+    ScTokenArray* getRangeNameTokens(const String& rFile, const String& rName, const ScAddress& rCurPos);
+
+    sal_uInt16 getExternalFileId(const String& rFile);
+    const String* getExternalFileName(sal_uInt16 nIndex) const;
+    
+    void refreshNames(const String& rFile);
+    void switchSrcFile(const String& rOldFile, const String& rNewFile);
+    void clear();
+
+private:
+    ScExternalRefManager();
+    ScExternalRefManager(const ScExternalRefManager&); 
+
+    DocCache* getDocumentCache(const String& rFile);
+    ScDocument* getSrcDocument(const String& rFile);
+    void insertExternalFileLink(const String& rFile, const String& rFilterName);
+
+private:
+    ScDocument* mpDoc;
+
+    /** 
+     * Source document cache.  This stores the original source document 
+     * instances.  The source documents get purged after certain period of 
+     * time. 
+     */
+    DocShellMap maDocShells;
+
+    /** cache only of referenced ranges and names from source documents. */
+    DocCacheMap maCachedDocContents;
+
+    /** original source file index. */
+    ::std::vector<String> maFileNames;
+};
+
+
+#endif
diff --git sc/inc/opcode.hxx sc/inc/opcode.hxx
index 285104d..b72ebad 100644
--- sc/inc/opcode.hxx
+++ sc/inc/opcode.hxx
@@ -44,6 +44,7 @@ enum OpCodeEnum
 		ocStop				= SC_OPCODE_STOP,
 		ocExternal			= SC_OPCODE_EXTERNAL,
 		ocName				= SC_OPCODE_NAME,
+        ocExternalName      = SC_OPCODE_EXTERNAL_NAME,
 	// Jump commands
 		ocIf				= SC_OPCODE_IF,
 		ocChose				= SC_OPCODE_CHOSE,
diff --git sc/inc/token.hxx sc/inc/token.hxx
index 23decd6..01e67ca 100644
--- sc/inc/token.hxx
+++ sc/inc/token.hxx
@@ -64,7 +64,7 @@ enum StackVarEnum
                                         // cell during import, having a double
                                         // and/or string result and a formula
                                         // string to be compiled.
-
+    svExternalName,
     svError,                            // error token
     svMissing = 0x70,                   // 0 or ""
     svUnknown                           // unknown StackType
@@ -447,6 +447,21 @@ public:
 };
 
 
+class ScExternalNameToken : public ScOpToken
+{
+private:
+    sal_uInt16                  mnFileId;
+    String                      maName;
+public:
+                                ScExternalNameToken( sal_uInt16 nFileId, const String& rName );
+                                ScExternalNameToken( const ScExternalNameToken& r );
+    virtual                     ~ScExternalNameToken();
+    virtual USHORT              GetIndex() const;
+    virtual const String&       GetString() const;
+    virtual BOOL                operator==( const ScToken& rToken ) const;
+};
+
+
 class ScJumpToken : public ScOpToken
 {
 private:
diff --git sc/inc/tokenarray.hxx sc/inc/tokenarray.hxx
index 1d903ce..f7c4d6b 100644
--- sc/inc/tokenarray.hxx
+++ sc/inc/tokenarray.hxx
@@ -193,6 +193,7 @@ public:
     ScToken* AddDoubleReference( const ComplRefData& rRef );
     ScToken* AddName( USHORT n );
     ScToken* AddMatrix( ScMatrix* p );
+    ScToken* AddExternalName( sal_uInt16 nFileId, const String& rName );
     ScToken* AddExternal( const sal_Unicode* pStr );
     /** Xcl import may play dirty tricks with OpCode!=ocExternal.
         Others don't use! */
diff --git sc/source/core/data/documen2.cxx sc/source/core/data/documen2.cxx
index 6dd6697..69e45fb 100644
--- sc/source/core/data/documen2.cxx
+++ sc/source/core/data/documen2.cxx
@@ -140,6 +140,7 @@ ScDocument::ScDocument( ScDocumentMode	eMode,
 		pScriptTypeData( NULL ),
         pCacheFieldEditEngine( NULL ),
         pDocProtection( NULL ),
+        pExternalRefMgr( NULL ),
 		pViewOptions( NULL ),
 		pDocOptions( NULL ),
 		pExtDocOptions( NULL ),
diff --git sc/source/core/data/documen3.cxx sc/source/core/data/documen3.cxx
index 541e2e4..20dae1e 100644
--- sc/source/core/data/documen3.cxx
+++ sc/source/core/data/documen3.cxx
@@ -61,6 +61,7 @@
 #include "brdcst.hxx"
 #include "bcaslot.hxx"
 #include "tablink.hxx"
+#include "externalrefmgr.hxx"
 #include "markdata.hxx"
 #include "validat.hxx"
 #include "dociter.hxx"
@@ -81,7 +82,10 @@
 #include "svtools/PasswordHelper.hxx"
 #include "tabprotection.hxx"
 
+#include <memory>
+
 using namespace com::sun::star;
+using ::std::auto_ptr;
 
 //------------------------------------------------------------------------
 
@@ -478,6 +482,14 @@ BOOL ScDocument::LinkExternalTab( SCTAB& rTab, const String& aDocTab,
 	return TRUE;
 }
 
+ScExternalRefManager* ScDocument::GetExternalRefManager()
+{
+    if (!pExternalRefMgr.get())
+        pExternalRefMgr.reset(new ScExternalRefManager(this));
+
+    return pExternalRefMgr.get();
+}
+
 BOOL ScDocument::InsertLinkedEmptyTab( SCTAB& rnTab, const String& rFileName,
         const String& rFilterName, const String& rFilterOpt, const String& rTabName )
 {
diff --git sc/source/core/tool/compiler.cxx sc/source/core/tool/compiler.cxx
index 7caea76..106514c 100644
--- sc/source/core/tool/compiler.cxx
+++ sc/source/core/tool/compiler.cxx
@@ -73,9 +73,13 @@
 #include "errorcodes.hxx"
 #include "parclass.hxx"
 #include "autonamecache.hxx"
+#include "externalrefmgr.hxx"
+
+#include <stdio.h>
 
 using namespace ::com::sun::star;
 using rtl::OUString;
+using ::std::vector;
 
 #if OSL_DEBUG_LEVEL > 1
 // For some unknown reason the identical dbg_dump utilities in
@@ -1361,6 +1365,64 @@ struct ConventionOOO_A1 : public Convention_A1
 
         return sal_Unicode(0);
     }
+
+    virtual bool parseExternalName( const String& rSymbol, String& rFile, String& rName ) const
+    {
+        // in OOO A1, an external name syntax is as follows:
+        // 
+        //   'file:///path/to/source'#ExternalName
+
+        String aTmpFile, aTmpName;
+        xub_StrLen nLen = rSymbol.Len();
+        const sal_Unicode* p = rSymbol.GetBuffer();
+        bool bInFilePath = false;
+        bool bInName = false;
+        sal_Unicode cPrev = 0;
+        for (xub_StrLen i = 0; i < nLen; ++i, ++p)
+        {
+            const sal_Unicode c = *p;
+            if (c == '\'')
+            {
+                if (!bInFilePath && i != 0)
+                    // The first character must be a single quote.
+                    return false;
+
+                bInFilePath = !bInFilePath;
+            }
+            else if (bInFilePath)
+                aTmpFile.Append(c);
+            else if (bInName)
+                aTmpName.Append(c);
+            else if (c == '#')
+            {
+                if (cPrev != '\'')
+                    // '#' must immediately follow the closing single quote.
+                    return false;
+
+                bInName = true;
+            }
+
+            cPrev = c;
+        }
+
+        if (aTmpFile.Len() == 0 || aTmpName.Len() == 0)
+            return false;
+
+        rFile = aTmpFile;
+        rName = aTmpName;
+
+        return true;
+    }
+
+    virtual String makeExternalNameStr( const String& rFile, const String& rName ) const
+    {
+        String aStr;
+        aStr.Append(sal_Unicode('\''));
+        aStr.Append(rFile);
+        aStr.AppendAscii("\'#");
+        aStr.Append(rName);
+        return aStr;
+    }
 };
 
 
@@ -1480,6 +1542,16 @@ struct ConventionXL
         }
         return sal_Unicode(0);
     }
+
+    static bool parseExternalName( const String& rSymbol, String& rFile, String& rName )
+    {
+        return false;
+    }
+
+    static String makeExternalNameStr( const String& rFile, const String& rName )
+    {
+        return String();
+    }
 };
 
 struct ConventionXL_A1 : public Convention_A1, public ConventionXL
@@ -1561,6 +1633,16 @@ struct ConventionXL_A1 : public Convention_A1, public ConventionXL
     {
         return ConventionXL::getSpecialSymbol(eSymType);
     }
+
+    virtual bool parseExternalName( const String& rSymbol, String& rFile, String& rName ) const
+    {
+        return ConventionXL::parseExternalName(rSymbol, rFile, rName);
+    }
+
+    virtual String makeExternalNameStr( const String& rFile, const String& rName ) const
+    {
+        return ConventionXL::makeExternalNameStr(rFile, rName);
+    }
 };
 
 static const ConventionXL_A1 ConvXL_A1;
@@ -1688,6 +1770,16 @@ struct ConventionXL_R1C1 : public ScCompiler::Convention, public ConventionXL
     {
         return ConventionXL::getSpecialSymbol(eSymType);
     }
+
+    virtual bool parseExternalName( const String& rSymbol, String& rFile, String& rName ) const
+    {
+        return ConventionXL::parseExternalName(rSymbol, rFile, rName);
+    }
+
+    virtual String makeExternalNameStr( const String& rFile, const String& rName ) const
+    {
+        return ConventionXL::makeExternalNameStr(rFile, rName);
+    }
 };
 
 static const ConventionXL_R1C1 ConvXL_R1C1;
@@ -2597,6 +2689,28 @@ BOOL ScCompiler::IsNamedRange( const String& rUpperName )
         return FALSE;
 }
 
+bool ScCompiler::IsExternalNamedRange( const String& rSymbol )
+{
+    if (!pConv)
+        return false;
+
+    String aFile, aName;
+    if (!pConv->parseExternalName(rSymbol, aFile, aName))
+        return false;
+
+    ScRawToken aToken;
+    if (aFile.Len() > MAXSTRLEN || aName.Len() > MAXSTRLEN)
+        return false;
+
+    sal_uInt16 nFileId = pDoc->GetExternalRefManager()->getExternalFileId(aFile);
+    fprintf(stdout, "ScCompiler::IsExternalNamedRange:   file name = '%s'; file id = %d\n",
+            rtl::OUStringToOString(aFile, RTL_TEXTENCODING_UTF8).getStr(), nFileId);
+
+    aToken.SetExternalName(nFileId, aName);
+    pRawToken = aToken.Clone();
+    return true;
+}
+
 BOOL ScCompiler::IsDBRange( const String& rName )
 {
     USHORT n;
@@ -3190,10 +3304,11 @@ BOOL ScCompiler::NextNewToken( bool bAllowBooleans )
             // IsBoolean before isValue to catch inline bools without the kludge
             //    for inline arrays.
             if ( !(bMayBeFuncName && IsOpCode( aUpper ))
-              && !IsReference( aOrg )
+              && !IsReference(aOrg)
               && !(bAllowBooleans && IsBoolean( aUpper ))
               && !IsValue( aUpper )
-              && !IsNamedRange( aUpper )
+              && !IsNamedRange(aUpper)
+              && !IsExternalNamedRange(aOrg)
               && !IsDBRange( aUpper )
               && !IsColRowName( aUpper )
               && !(bMayBeFuncName && IsMacro( aUpper ))
@@ -3452,6 +3567,22 @@ BOOL ScCompiler::GetToken()
     }
     if( pToken->GetOpCode() == ocSubTotal )
         glSubTotal = TRUE;
+    else if ( pToken->GetOpCode() == ocExternalName )
+    {
+        ScExternalRefManager* pRefMgr = pDoc->GetExternalRefManager();
+        const String* pFile = pRefMgr->getExternalFileName(pToken->GetIndex());
+        if (!pFile)
+            SetError(errNoName);
+
+        const String& rName = pToken->GetString();
+        ScTokenArray* pNew = pRefMgr->getRangeNameTokens(*pFile, rName, aPos);
+        if (pNew)
+        {
+            PushTokenArray(pNew, true);
+            return GetToken();
+        }
+        SetError(errNoName);
+    }
     else if( pToken->GetOpCode() == ocName )
     {
         ScRangeData* pRangeData = pDoc->GetRangeName()->FindIndex( pToken->GetIndex() );
@@ -5402,6 +5533,15 @@ ScToken* ScCompiler::CreateStringFromToken( rtl::OUStringBuffer& rBuffer, ScToke
     BOOL bSpaces = FALSE;
     ScToken* t = pTokenP;
     OpCode eOp = t->GetOpCode();
+    if (eOp == ocExternalName)
+    {
+        const String *pStr = pDoc->GetExternalRefManager()->getExternalFileName(t->GetIndex());
+        if (pStr)
+        {
+            rBuffer.append(pConv->makeExternalNameStr(*pStr, t->GetString()));
+        }
+    }
+
     if( eOp >= ocAnd && eOp <= ocOr )
     {
         // AND, OR infix?
diff --git sc/source/core/tool/token.cxx sc/source/core/tool/token.cxx
index 7312257..8ccba56 100644
--- sc/source/core/tool/token.cxx
+++ sc/source/core/tool/token.cxx
@@ -52,6 +52,8 @@
 #include "parclass.hxx"
 #include "jumpmatrix.hxx"
 
+using ::std::vector;
+
 // ImpTokenIterator wird je Interpreter angelegt, mehrfache auch durch
 // SubCode via ScTokenIterator Push/Pop moeglich
 IMPL_FIXEDMEMPOOL_NEWDEL( ImpTokenIterator, 32, 16 )
@@ -196,6 +198,19 @@ void ScRawToken::SetName( USHORT n )
     nRefCnt = 0;
 }
 
+void ScRawToken::SetExternalName( sal_uInt16 nFileId, const String& rName )
+{
+    eOp = ocExternalName;
+    eType = svExternalName;
+    nRefCnt = 0;
+
+    extname.nFileId = nFileId;
+
+    xub_StrLen n = rName.Len();
+    memcpy(extname.cName, rName.GetBuffer(), n*sizeof(sal_Unicode));
+    extname.cName[n] = 0;
+}
+
 ComplRefData& ScRawToken::GetReference()
 {
     DBG_ASSERT( lcl_IsReference( eOp, GetType() ), "GetReference: no Ref" );
@@ -255,6 +270,7 @@ ScRawToken* ScRawToken::Clone() const
             case svSingleRef:
             case svDoubleRef:   n += sizeof(aRef); break;
             case svMatrix:      n += sizeof(ScMatrix*); break;
+            case svExternalName: n += sizeof(extname); break;
             case svIndex:       n += sizeof(USHORT); break;
             case svJump:        n += nJump[ 0 ] * 2 + 2; break;
             case svExternal:    n = sal::static_int_cast<USHORT>( n + GetStrLenBytes( cStr+1 ) + GetStrLenBytes( 2 ) ); break;
@@ -311,8 +327,14 @@ ScToken* ScRawToken::CreateToken() const
             return new ScMatrixToken( pMat );
         //break;
         case svIndex :
+            fprintf(stdout, "ScRawToken::CreateToken:   creating index token ... (op = %d)\n", eOp);
             return new ScIndexToken( eOp, nIndex );
         //break;
+        case svExternalName:
+        {
+            String aName(extname.cName);
+            return new ScExternalNameToken( extname.nFileId, aName );
+        }
         case svJump :
             return new ScJumpToken( eOp, (short*) nJump );
         //break;
@@ -505,6 +527,9 @@ ScToken* ScToken::Clone() const
         case svExternal :
             return new ScExternalToken( *static_cast<const ScExternalToken*>(this) );
         //break;
+        case svExternalName :
+            return new ScExternalNameToken( *static_cast<const ScExternalNameToken*>(this) );
+        //break;
         case svFAP :
             return new ScFAPToken( *static_cast<const ScFAPToken*>(this) );
         //break;
@@ -1016,6 +1041,56 @@ BOOL ScIndexToken::operator==( const ScToken& r ) const
 }
 
 
+ScExternalNameToken::ScExternalNameToken( sal_uInt16 nFileId, const String& rName ) :
+    ScOpToken( ocExternalName, svExternalName ),
+    mnFileId(nFileId),
+    maName(rName)
+{
+}
+
+ScExternalNameToken::ScExternalNameToken( const ScExternalNameToken& r ) :
+    ScOpToken( r ),
+    mnFileId(r.mnFileId),
+    maName(r.maName)
+{
+}
+
+ScExternalNameToken::~ScExternalNameToken() {}
+
+USHORT ScExternalNameToken::GetIndex() const
+{
+    return mnFileId;
+}
+
+const String& ScExternalNameToken::GetString() const
+{
+    return maName;
+}
+
+BOOL ScExternalNameToken::operator==( const ScToken& r ) const
+{
+    if ( !ScToken::operator==(r) )
+        return false;
+
+    if (mnFileId != r.GetIndex())
+        return false;
+
+    xub_StrLen nLen = maName.Len();
+    const String& rName = r.GetString();
+    if (nLen != rName.Len())
+        return false;
+
+    const sal_Unicode* p1 = maName.GetBuffer();
+    const sal_Unicode* p2 = rName.GetBuffer();
+    for (xub_StrLen j = 0; j < nLen; ++j)
+    {
+        if (p1[j] != p2[j])
+            return false;
+    }
+    return true;
+}
+
+
 short* ScJumpToken::GetJump() const                     { return pJump; }
 BOOL ScJumpToken::operator==( const ScToken& r ) const
 {
@@ -1873,6 +1948,11 @@ ScToken* ScTokenArray::AddMatrix( ScMatrix* p )
     return Add( new ScMatrixToken( p ) );
 }
 
+ScToken* ScTokenArray::AddExternalName( sal_uInt16 nFileId, const String& rName )
+{
+    return Add( new ScExternalNameToken(nFileId, rName) );
+}
+
 ScToken* ScTokenArray::AddColRowName( const SingleRefData& rRef )
 {
     return Add( new ScSingleRefOpToken( ocColRowName, rRef ) );
diff --git sc/source/filter/excel/excform8.cxx sc/source/filter/excel/excform8.cxx
index ebf8543..da9ded9 100644
--- sc/source/filter/excel/excform8.cxx
+++ sc/source/filter/excel/excform8.cxx
@@ -41,6 +41,7 @@
 #include "xilink.hxx"
 #include "xiname.hxx"
 
+#include "externalrefmgr.hxx"
 
 ExcelToSc8::ExcelToSc8( const XclImpRoot& rRoot ) :
     ExcelToSc( rRoot ),
@@ -608,8 +609,22 @@ ConvErr ExcelToSc8::Convert( const ScTokenArray*& rpTokArray, XclImpStream& aIn,
                     {
                         case xlExtName:
                         {
-                            aStack << aPool.Store( ocNoName, pExtName->GetName() );
-                            GetTracer().TraceFormulaExtName();
+                            const String* pFileUrl = rLinkMan.GetSupbookUrl(nXtiIndex);
+                            if (!pFileUrl)
+                            {    
+                                aStack << aPool.Store(ocNoName, pExtName->GetName());
+                                break;
+                            }
+
+                            String aFileUrl = ScGlobal::GetAbsDocName(*pFileUrl, GetDocShell());
+                            ScExternalRefManager* pRefMgr = GetDoc().GetExternalRefManager();
+                            sal_uInt16 nFileId = pRefMgr->getExternalFileId(aFileUrl);
+                            fprintf(stdout, "ExcelToSc8::Convert:   external name (file: %s; name: %s; file id: %d)\n",
+                                    rtl::OUStringToOString(aFileUrl, RTL_TEXTENCODING_UTF8).getStr(),
+                                    rtl::OUStringToOString(pExtName->GetName(), RTL_TEXTENCODING_UTF8).getStr(),
+                                    nFileId);
+
+                            aStack << aPool.StoreExtName(nFileId, pExtName->GetName());
                         }
                         break;
 
diff --git sc/source/filter/excel/tokstack.cxx sc/source/filter/excel/tokstack.cxx
index 20ada63..091de84 100644
--- sc/source/filter/excel/tokstack.cxx
+++ sc/source/filter/excel/tokstack.cxx
@@ -395,6 +395,16 @@ void TokenPool::GetElement( const UINT16 nId )
                         pScToken->AddMatrix( p );
                 }
                 break;
+            case T_ExtName:
+            {
+                UINT16 n = pElement[nId];
+                if (n < maExtNames.size())
+                {
+                    const ExtName& r = maExtNames[n];
+                    pScToken->AddExternalName(r.mnFileId, r.maName);
+                }
+            }
+            break;
 			default:
 				DBG_ERROR("-TokenPool::GetElement(): Zustand undefiniert!?");
 		}
@@ -477,6 +487,16 @@ void TokenPool::GetElementRek( const UINT16 nId )
                             pScToken->AddMatrix( p );
                     }
                     break;
+                case T_ExtName:
+                {
+                    UINT16 n = pElement[*pAkt];
+                    if (n < maExtNames.size())
+                    {
+                        const ExtName& r = maExtNames[n];
+                        pScToken->AddExternalName(r.mnFileId, r.maName);
+                    }
+                }
+                break;
 				default:
 					DBG_ERROR("-TokenPool::GetElementRek(): Zustand undefiniert!?");
 			}
@@ -724,6 +744,24 @@ const TokenId TokenPool::StoreMatrix( SCSIZE nC, SCSIZE nR )
     return ( const TokenId ) nElementAkt;
 }
 
+const TokenId TokenPool::StoreExtName( sal_uInt16 nFileId, const String& rName )
+{
+    if ( nElementAkt >= nElement )
+        GrowElement();
+
+    pElement[nElementAkt] = static_cast<UINT16>(maExtNames.size());
+    pType[nElementAkt] = T_ExtName;
+    
+    maExtNames.push_back(ExtName());
+    ExtName& r = maExtNames.back();
+    r.mnFileId = nFileId;
+    r.maName = rName;
+
+    ++nElementAkt;
+
+    return static_cast<const TokenId>(nElementAkt);
+}
+
 void TokenPool::Reset( void )
 {
     nP_IdAkt = nP_IdLast = nElementAkt = nP_StrAkt = nP_DblAkt = nP_ErrAkt = nP_RefTrAkt = nP_ExtAkt = nP_NlfAkt = nP_MatrixAkt = 0;
diff --git sc/source/filter/excel/xilink.cxx sc/source/filter/excel/xilink.cxx
index e351863..b58f6ec 100644
--- sc/source/filter/excel/xilink.cxx
+++ sc/source/filter/excel/xilink.cxx
@@ -191,6 +191,11 @@ public:
                             sal_uInt16 nXtiIndex ) const;
     /** Returns the specified external name or 0 on error. */
     const XclImpExtName* GetExternName( sal_uInt16 nXtiIndex, sal_uInt16 nExtName ) const;
+
+    /** Returns the absolute file URL of a supporting workbook specified by
+        the index. */
+    const String*       GetSupbookUrl( sal_uInt16 nXtiIndex ) const;
+
     /** Tries to decode the URL of the specified XTI entry to OLE or DDE link components.
         @descr  For DDE links: Decodes to application name and topic.
         For OLE object links: Decodes to class name and document URL.
@@ -584,6 +589,14 @@ const XclImpExtName* XclImpLinkManagerImpl::GetExternName( sal_uInt16 nXtiIndex,
     return pSupbook ? pSupbook->GetExternName( nExtName ) : 0;
 }
 
+const String* XclImpLinkManagerImpl::GetSupbookUrl( sal_uInt16 nXtiIndex ) const
+{
+    const XclImpSupbook* p = GetSupbook( nXtiIndex );
+    if (!p)
+        return NULL;
+    return &p->GetXclUrl();
+}
+
 bool XclImpLinkManagerImpl::GetLinkData( String& rApplic, String& rTopic, sal_uInt16 nXtiIndex ) const
 {
     const XclImpSupbook* pSupbook = GetSupbook( nXtiIndex );
@@ -707,6 +720,11 @@ const XclImpExtName* XclImpLinkManager::GetExternName( sal_uInt16 nXtiIndex, sal
     return mxImpl->GetExternName( nXtiIndex, nExtName );
 }
 
+const String* XclImpLinkManager::GetSupbookUrl( sal_uInt16 nXtiIndex ) const
+{
+    return mxImpl->GetSupbookUrl(nXtiIndex);
+}
+
 bool XclImpLinkManager::GetLinkData( String& rApplic, String& rTopic, sal_uInt16 nXtiIndex ) const
 {
     return mxImpl->GetLinkData( rApplic, rTopic, nXtiIndex );
diff --git sc/source/filter/inc/tokstack.hxx sc/source/filter/inc/tokstack.hxx
index fef6b03..c319704 100644
--- sc/source/filter/inc/tokstack.hxx
+++ sc/source/filter/inc/tokstack.hxx
@@ -37,6 +37,8 @@
 #include <tools/debug.hxx>
 #include "compiler.hxx"
 
+#include <vector>
+
 
 typedef OpCode DefTokenId;
 // in PRODUCT version: ambiguity between OpCode (being USHORT) and UINT16
@@ -80,6 +82,7 @@ enum E_TYPE
 	T_Ext,		// irgendwas Unbekanntes mit Funktionsnamen
 	T_Nlf,		// token for natural language formula
 	T_Matrix,	// token for inline arrays
+    T_ExtName,  // token for external names
 	T_Error		// fuer Abfrage im Fehlerfall
 };
 
@@ -136,6 +139,14 @@ class TokenPool
 		UINT16						nP_Matrix;
 		UINT16						nP_MatrixAkt;
 
+        /** for storage of external names */
+        struct ExtName
+        {
+            sal_uInt16  mnFileId;
+            String      maName;
+        };
+        ::std::vector<ExtName>      maExtNames;
+
 		UINT16*						pElement;	// Array mit Indizes fuer Elemente
 		E_TYPE*						pType;		// ...mit Typ-Info
 		UINT16*						pSize;		// ...mit Laengenangabe (Anz. UINT16)
@@ -182,6 +193,7 @@ class TokenPool
 										// 4 externals (e.g. AddIns, Makros...)
 		const TokenId				StoreNlf( const SingleRefData& rTr );
 		const TokenId				StoreMatrix( SCSIZE nC, SCSIZE nR );
+        const TokenId               StoreExtName( sal_uInt16 nFileId, const String& rName );
 
 		inline const TokenId		LastId( void ) const;
 		inline const ScTokenArray*	operator []( const TokenId nId );
diff --git sc/source/filter/inc/xilink.hxx sc/source/filter/inc/xilink.hxx
index 0d547fe..3c2d0d9 100644
--- sc/source/filter/inc/xilink.hxx
+++ sc/source/filter/inc/xilink.hxx
@@ -179,6 +179,9 @@ public:
                             sal_uInt16 nXtiIndex ) const;
     /** Returns the specified external name or 0 on error. */
     const XclImpExtName* GetExternName( sal_uInt16 nXtiIndex, sal_uInt16 nExtName ) const;
+
+    const String* GetSupbookUrl( sal_uInt16 nXtiIndex ) const;
+
     /** Tries to decode the URL of the specified XTI entry to OLE or DDE link components.
         @descr  For DDE links: Decodes to application name and topic.
         For OLE object links: Decodes to class name and document URL.
diff --git sc/source/ui/docshell/externalrefmgr.cxx sc/source/ui/docshell/externalrefmgr.cxx
new file mode 100644
index 0000000..52074f9
--- /dev/null
+++ sc/source/ui/docshell/externalrefmgr.cxx
@@ -0,0 +1,564 @@
+/*************************************************************************
+ *
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ * 
+ * Copyright 2008 by Sun Microsystems, Inc.
+ *
+ * OpenOffice.org - a multi-platform office productivity suite
+ *
+ * $RCSfile: document.hxx,v $
+ * $Revision: 1.112 $
+ *
+ * This file is part of OpenOffice.org.
+ *
+ * OpenOffice.org is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License version 3
+ * only, as published by the Free Software Foundation.
+ *
+ * OpenOffice.org is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License version 3 for more details
+ * (a copy is included in the LICENSE file that accompanied this code).
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * version 3 along with OpenOffice.org.  If not, see
+ * <http://www.openoffice.org/license.html>
+ * for a copy of the LGPLv3 License.
+ *
+ ************************************************************************/
+
+// MARKER(update_precomp.py): autogen include statement, do not remove
+#include "precompiled_sc.hxx"
+
+
+
+// INCLUDE ---------------------------------------------------------------
+
+#include "externalrefmgr.hxx"
+#include "document.hxx"
+#include "token.hxx"
+#include "tokenarray.hxx"
+#include "address.hxx"
+#include "tablink.hxx"
+#include "docsh.hxx"
+#include "scextopt.hxx"
+#include "rangenam.hxx"
+#include "cell.hxx"
+
+#include "sfx2/app.hxx"
+#include "sfx2/docfilt.hxx"
+#include "sfx2/docfile.hxx"
+#include "sfx2/fcontnr.hxx"
+#include "sfx2/sfxsids.hrc"
+#include "sfx2/objsh.hxx"
+#include "svtools/itemset.hxx"
+#include "svtools/stritem.hxx"
+#include "svx/linkmgr.hxx"
+
+#include <memory>
+#include <stdio.h>
+
+using ::std::auto_ptr;
+using ::com::sun::star::uno::Any;
+using ::std::vector;
+using ::std::find;
+using ::std::distance;
+
+ScExternalRefLink::ScExternalRefLink(ScDocument* pDoc, const String& rFile, const String& rFilter) :
+    ::sfx2::SvBaseLink(::sfx2::LINKUPDATE_ONCALL, FORMAT_FILE),
+    maFileName(rFile),
+    maFilterName(rFilter),
+    mpDoc(pDoc)
+{
+}
+
+ScExternalRefLink::~ScExternalRefLink()
+{
+}
+
+void ScExternalRefLink::Closed()
+{
+    fprintf(stdout, "ScExternalRefLink::Closed: --begin\n");
+    fprintf(stdout, "ScExternalRefLink::Closed: --end\n");
+}
+
+void ScExternalRefLink::DataChanged(const String& /*rMimeType*/, const Any& /*rValue*/)
+{
+    String aFile, aFilter;
+    mpDoc->GetLinkManager()->GetDisplayNames(this, NULL, &aFile, NULL, &aFilter);
+    ScExternalRefManager* pMgr = mpDoc->GetExternalRefManager();
+
+    if (maFileName.Equals(aFile))
+    {
+        pMgr->refreshNames(aFile);
+    }
+    else
+    {
+        pMgr->switchSrcFile(maFileName, aFile);
+        maFileName   = aFile;
+        maFilterName = aFilter;
+    }
+}
+
+void ScExternalRefLink::Edit(Window* pParent, const Link& /*rEndEditHdl*/)
+{
+    SvBaseLink::Edit(pParent, LINK(this, ScExternalRefLink, EndEditHdl));
+}
+
+IMPL_LINK(ScExternalRefLink, EndEditHdl, void*, EMPTYARG)
+{
+    return 0;
+}
+
+// ============================================================================
+
+static ScToken* lcl_convertToToken(ScBaseCell* pCell)
+{
+    if (!pCell)
+        return NULL;
+
+    switch (pCell->GetCellType())
+    {
+        case CELLTYPE_STRING:
+        {
+            String aStr;
+            static_cast<ScStringCell*>(pCell)->GetString(aStr);
+            ScStringToken aToken(aStr);
+            return aToken.Clone();
+        }
+        break;
+        case CELLTYPE_VALUE:
+        {
+            double fVal = static_cast<ScValueCell*>(pCell)->GetValue();
+            ScDoubleToken aToken(fVal);
+            return aToken.Clone();
+        }
+        break;
+        case CELLTYPE_FORMULA:
+        {
+            ScFormulaCell* pFCell = static_cast<ScFormulaCell*>(pCell);
+            if (pFCell->IsValue())
+            {
+                double fVal = pFCell->GetValue();
+                ScDoubleToken aToken(fVal);
+                return aToken.Clone();
+            }
+            else
+            {
+                String aStr;
+                pFCell->GetString(aStr);
+                ScStringToken aToken(aStr);
+                return aToken.Clone();
+            }
+        }
+        break;
+        default:
+            DBG_ERROR("attempted to convert an unknown cell type.");
+    }
+
+    return NULL;
+}
+
+static ScTokenArray* lcl_convertToTokenArray(ScDocument* pSrcDoc, const ScRange& rRange)
+{
+    const ScAddress& s = rRange.aStart;
+    const ScAddress& e = rRange.aEnd;
+
+    SCTAB nTab1 = s.Tab(), nTab2 = e.Tab();
+    SCCOL nCol1 = s.Col(), nCol2 = e.Col();
+    SCROW nRow1 = s.Row(), nRow2 = e.Row();
+
+    auto_ptr<ScTokenArray> pArray(new ScTokenArray);
+    bool bFirstTab = true;
+    for (SCTAB nTab = nTab1; nTab <= nTab2; ++nTab)
+    {
+        ScMatrixRef xMat = new ScMatrix(
+            static_cast<SCSIZE>(nCol2-nCol1+1),
+            static_cast<SCSIZE>(nRow2-nRow1+1));
+
+        for (SCCOL nCol = nCol1; nCol <= nCol2; ++nCol)
+        {
+            for (SCROW nRow = nRow1; nRow <= nRow2; ++nRow)
+            {
+                SCSIZE nC = nCol - nCol1, nR = nRow - nRow1;
+                ScBaseCell* pCell;
+                pSrcDoc->GetCell(nCol, nRow, nTab, pCell);
+                if (pCell)
+                {
+                    switch (pCell->GetCellType())
+                    {
+                        case CELLTYPE_STRING:
+                        {
+                            String aStr;
+                            static_cast<ScStringCell*>(pCell)->GetString(aStr);
+                            xMat->PutString(aStr, nC, nR);
+                        }
+                        break;
+                        case CELLTYPE_VALUE:
+                        {
+                            double fVal = static_cast<ScValueCell*>(pCell)->GetValue();
+                            xMat->PutDouble(fVal, nC, nR);
+                        }
+                        break;
+                        case CELLTYPE_FORMULA:
+                        {
+                            ScFormulaCell* pFCell = static_cast<ScFormulaCell*>(pCell);
+                            if (pFCell->IsValue())
+                            {
+                                double fVal = pFCell->GetValue();
+                                xMat->PutDouble(fVal, nC, nR);
+                            }
+                            else
+                            {
+                                String aStr;
+                                pFCell->GetString(aStr);
+                                xMat->PutString(aStr, nC, nR);
+                            }
+                        }
+                        break;
+                        default:
+                            DBG_ERROR("attempted to convert an unknown cell type.");
+                    }
+                }
+                else
+                {
+                    xMat->PutEmpty(nC, nR);
+                }
+            }
+        }
+        if (!bFirstTab)
+            pArray->AddOpCode(ocSep);
+
+        ScMatrix* pMat2 = xMat;
+        ScMatrixToken aToken(pMat2);
+        pArray->AddToken(aToken);
+
+        bFirstTab = false;
+    }
+    return pArray.release();
+}
+
+ScExternalRefManager::ScExternalRefManager(ScDocument* pDoc) :
+    mpDoc(pDoc)
+{
+}
+
+ScExternalRefManager::~ScExternalRefManager()
+{
+    clear();
+}
+
+ScToken* ScExternalRefManager::getSingleRefToken(const String& rFile, const ScAddress& rCell)
+{
+    {
+        String aStr;
+        rCell.Format(aStr, SCA_ABS_3D);
+        fprintf(stdout, "ScExternalRefManager::getSingleRefToken: --begin (file = '%s'; address = '%s')\n",
+                rtl::OUStringToOString(rFile, RTL_TEXTENCODING_UTF8).getStr(),
+                rtl::OUStringToOString(aStr, RTL_TEXTENCODING_UTF8).getStr());
+    }
+
+    SingleTokenMap& rMap = getDocumentCache(rFile)->maSingleTokens;
+    SingleTokenMap::iterator itr = rMap.find(rCell);
+    if (itr != rMap.end())
+    {
+        fprintf(stdout, "ScExternalRefManager::getSingleRefToken:   reference cached\n");
+        // this single reference is cached.
+        return itr->second.get();
+    }
+
+    // reference not cached.  read from the source document.
+    ScDocument* pSrcDoc = getSrcDocument(rFile);
+    if (!pSrcDoc)
+        return NULL;
+
+    ScBaseCell* pCell = NULL;
+    pSrcDoc->GetCell(rCell.Col(), rCell.Row(), rCell.Tab(), pCell);
+    TokenRef pTok(lcl_convertToToken(pCell));
+
+    if (!pTok.get())
+    {
+        // Cell in the source document is probably empty.
+        pTok.reset(new ScEmptyCellToken(false, false));
+    }
+
+    rMap.insert(SingleTokenMap::value_type(rCell, pTok));
+    fprintf(stdout, "ScExternalRefManager::getSingleRefToken: --end\n");
+    return pTok.get();
+}
+
+ScTokenArray* ScExternalRefManager::getDoubleRefTokens(const String& rFile, const ScRange& rRange)
+{
+    {
+        String aStr;
+        rRange.Format(aStr, SCR_ABS_3D, mpDoc);
+        fprintf(stdout, "ScExternalRefManager::getDoubleRefToken: --begin (file = '%s'; range = '%s)\n",
+                rtl::OUStringToOString(rFile, RTL_TEXTENCODING_UTF8).getStr(), 
+                rtl::OUStringToOString(aStr, RTL_TEXTENCODING_UTF8).getStr());
+    }
+
+    DoubleTokenMap& rMap = getDocumentCache(rFile)->maDoubleTokens;
+    DoubleTokenMap::iterator itr = rMap.find(rRange);
+    if (itr != rMap.end())
+    {
+        fprintf(stdout, "ScExternalRefManager::getDoubleRefToken:   range reference cached\n");
+        // this range reference is cached.
+        return itr->second.get();
+    }
+
+    // reference not cached.  read from the source document.
+    ScDocument* pSrcDoc = getSrcDocument(rFile);
+    if (!pSrcDoc)
+        return NULL;
+
+    TokenArrayRef pArray(lcl_convertToTokenArray(pSrcDoc, rRange));
+    if (!pArray.get())
+    {
+        // highly unlikely since lcl_convertToTokenArray never returns NULL.
+        fprintf(stdout, "ScExternalRefManager::getDoubleRefToken: --end (NULL)\n");
+        return NULL;
+    }
+
+    rMap.insert(DoubleTokenMap::value_type(rRange, pArray));
+    fprintf(stdout, "ScExternalRefManager::getDoubleRefToken: --end\n");
+    return pArray.get();
+}
+
+ScTokenArray* ScExternalRefManager::getRangeNameTokens(const String& rFile, const String& rName, const ScAddress& rCurPos)
+{
+    String aCellStr;
+    rCurPos.Format(aCellStr, SCA_ABS_3D);
+    fprintf(stdout, "ScExternalRefManager::getRangeNameTokens: --begin (file = '%s'; name = '%s'; pos = '%s')\n",
+            rtl::OUStringToOString(rFile, RTL_TEXTENCODING_UTF8).getStr(),
+            rtl::OUStringToOString(rName, RTL_TEXTENCODING_UTF8).getStr(),
+            rtl::OUStringToOString(aCellStr, RTL_TEXTENCODING_UTF8).getStr());
+
+    // First, check if this name has already been cached.
+    RangeNameMap& rMap = getDocumentCache(rFile)->maRangeNames;
+    RangeNameMap::iterator itr = rMap.find(rName);
+    if (itr != rMap.end())
+    {
+        fprintf(stdout, "ScExternalRefManager::getRangeNameTokens:   name is cached\n");
+        // this name is cached.
+        return itr->second->Clone();
+    }
+
+    ScDocument* pSrcDoc = getSrcDocument(rFile);
+    if (!pSrcDoc)
+        return NULL;
+
+    ScRangeName* pExtNames = pSrcDoc->GetRangeName();
+    String aUpperName = ScGlobal::pCharClass->upper(rName);
+    USHORT n;
+    bool bRes = pExtNames->SearchNameUpper(aUpperName, n);
+    if (!bRes)
+        return NULL;
+
+    ScRangeData* pRangeData = (*pExtNames)[n];
+    if (!pRangeData)
+        return NULL;
+
+    // Parse all tokens in this external range data, and replace each absolute 
+    // reference token with an external reference token, and cache them.  Also
+    // register the source document with the link manager if it's a new
+    // source.
+
+    TokenArrayRef pNew(new ScTokenArray);
+
+    ScTokenArray* pCode = pRangeData->GetCode();
+    for (ScToken* pToken = pCode->First(); pToken; pToken = pCode->Next())
+    {
+        bool bTokenAdded = false;
+        switch (pToken->GetType())
+        {
+            case svSingleRef:
+            {
+                fprintf(stdout, "ScDocument::FindExternalRangeName:   single ref\n");
+                SingleRefData aRef(pToken->GetSingleRef());
+                if (aRef.IsTabRel())
+                    break;
+
+                // The sheet reference is absolute.  Get the cell from the 
+                // source document, and convert it to a static value.
+    
+                aRef.CalcAbsIfRel(rCurPos);
+                ScToken* pTok = getSingleRefToken(rFile, ScAddress(aRef.nCol, aRef.nRow, aRef.nTab));
+                if (pTok)
+                {
+                    pNew->AddToken(*pTok);
+                    bTokenAdded = true;
+                }
+            }
+            break;
+            case svDoubleRef:
+            {
+                fprintf(stdout, "ScDocument::FindExternalRangeName:   double ref\n");
+                ComplRefData aComRef(pToken->GetDoubleRef());
+                if (aComRef.Ref1.IsTabRel() || aComRef.Ref2.IsTabRel())
+                    break;
+
+                aComRef.CalcAbsIfRel(rCurPos);
+                SCsTAB nTab1 = aComRef.Ref1.nTab;
+                SCsCOL nCol1 = aComRef.Ref1.nCol;
+                SCsROW nRow1 = aComRef.Ref1.nRow;
+                SCsTAB nTab2 = aComRef.Ref2.nTab;
+                SCsCOL nCol2 = aComRef.Ref2.nCol;
+                SCsROW nRow2 = aComRef.Ref2.nRow;
+
+                ScTokenArray* pArray = getDoubleRefTokens(rFile, ScRange(nCol1, nRow1, nTab1, nCol2, nRow2, nTab2));
+                if (pArray)
+                {
+                    for (ScToken* p = pArray->First(); p; p = pArray->Next())
+                    {
+                        pNew->AddToken(*p);
+                        bTokenAdded = true;
+                    }
+                }
+            }
+            break;
+        }
+        
+        if (!bTokenAdded)
+            pNew->AddToken(*pToken);
+    }
+
+    rMap.insert(RangeNameMap::value_type(rName, pNew));
+    fprintf(stdout, "ScExternalRefManager::getRangeNameTokens: --end\n");
+    return pNew->Clone();
+}
+
+ScExternalRefManager::DocCache* ScExternalRefManager::getDocumentCache(const String& rFile)
+{
+    DocCacheMap::iterator itr = maCachedDocContents.find(rFile);
+    if (itr == maCachedDocContents.end())
+    {
+        DocCacheRef pCache(new DocCache);
+        maCachedDocContents.insert(DocCacheMap::value_type(rFile, pCache));
+        return pCache.get();
+    }
+
+    return itr->second.get();
+}
+
+ScDocument* ScExternalRefManager::getSrcDocument(const String& rFile)
+{
+    DocShellMap::iterator itrEnd = maDocShells.end();
+    DocShellMap::iterator itr = maDocShells.find(rFile);
+    if (itr == itrEnd)
+    {
+        fprintf(stdout, "ScExternalRefManager::getSourceDocument:   file not found: '%s'\n",
+                rtl::OUStringToOString(rFile, RTL_TEXTENCODING_UTF8).getStr());
+
+        String aFilter, aOptions;
+        ScDocumentLoader::GetFilterName(rFile, aFilter, aOptions, true, false);
+        const SfxFilter* pFilter = ScDocShell::Factory().GetFilterContainer()->GetFilter4FilterName(aFilter);
+
+        SfxItemSet* pSet = new SfxAllItemSet(SFX_APP()->GetPool());
+        if (aOptions.Len())
+            pSet->Put(SfxStringItem(SID_FILE_FILTEROPTIONS, aOptions));
+
+        auto_ptr<SfxMedium> pMedium(new SfxMedium(rFile, STREAM_STD_READ, FALSE, pFilter, pSet));
+        if (pMedium->GetError() != ERRCODE_NONE)
+            return NULL;
+
+        pMedium->UseInteractionHandler(false);
+
+        ScDocShell* pNewShell = new ScDocShell(SFX_CREATE_MODE_INTERNAL);
+        SfxObjectShellRef aRef = pNewShell;
+
+        // increment the recursive link count of the source document.
+        ScExtDocOptions* pExtOpt = mpDoc->GetExtDocOptions();
+        sal_uInt32 nLinkCount = pExtOpt ? pExtOpt->GetDocSettings().mnLinkCnt : 0;
+        ScDocument* pSrcDoc = pNewShell->GetDocument();
+        ScExtDocOptions* pExtOptNew = pSrcDoc->GetExtDocOptions();
+        if (!pExtOptNew)
+        {
+            pExtOptNew = new ScExtDocOptions;
+            pSrcDoc->SetExtDocOptions(pExtOptNew);
+        }
+        pExtOptNew->GetDocSettings().mnLinkCnt = nLinkCount + 1;
+
+        pNewShell->DoLoad(pMedium.release());
+
+        maDocShells.insert(DocShellMap::value_type(rFile, aRef));
+        insertExternalFileLink(rFile, aFilter);
+        return pSrcDoc;
+    }
+
+    SfxObjectShell* p = itr->second;
+    return static_cast<ScDocShell*>(p)->GetDocument();
+}
+
+void ScExternalRefManager::insertExternalFileLink(const String& rFile, const String& rFilterName)
+{
+    SvxLinkManager* pLinkMgr = mpDoc->GetLinkManager();
+    ScExternalRefLink* pLink = new ScExternalRefLink(mpDoc, rFile, rFilterName);
+    pLinkMgr->InsertFileLink(*pLink, OBJECT_CLIENT_FILE, rFile, &rFilterName);
+    pLink->Update();
+}
+
+sal_uInt16 ScExternalRefManager::getExternalFileId(const String& rFile)
+{
+    using namespace std;
+
+    vector<String>::const_iterator itrBeg = maFileNames.begin(), itrEnd = maFileNames.end();
+    vector<String>::const_iterator itr = find(itrBeg, itrEnd, rFile);
+    if (itr != itrEnd)
+    {
+        size_t nId = distance(itrBeg, itr);
+        return static_cast<sal_uInt16>(nId);
+    }
+
+    maFileNames.push_back(rFile);
+    return static_cast<sal_uInt16>(maFileNames.size() - 1);
+}
+
+const String* ScExternalRefManager::getExternalFileName(sal_uInt16 nIndex) const
+{
+    if (nIndex >= maFileNames.size())
+        return NULL;
+
+    return &maFileNames[nIndex];
+}
+
+void ScExternalRefManager::refreshNames(const String& rFile)
+{
+    fprintf(stdout, "ScExternalRefManager::refreshNames: --begin\n");
+    // 1. Find the cached content (DocCache) for a given file name.
+    // 2. Clear the single and double reference caches.
+    // 3. Re-parse the cached names from the source documents.
+    // 4. Update those cells that contain the cached names.
+
+
+}
+
+void ScExternalRefManager::switchSrcFile(const String& rOldFile, const String& rNewFile)
+{
+    fprintf(stdout, "ScExternalRefManager::switchSrcFile: --begin (old file = '%s'; new file = '%s')\n",
+            rtl::OUStringToOString(rOldFile, RTL_TEXTENCODING_UTF8).getStr(), 
+            rtl::OUStringToOString(rNewFile, RTL_TEXTENCODING_UTF8).getStr());
+
+    /**
+     * This is a tricky one.  First, remove all cached data associated with 
+     * the old file (and purge the cached source document if it's still 
+     * alive), swap the file location with the new one, then recalculate those 
+     * cells that contain reference to the old file. 
+     *  
+     * To implement this, I need to switch from storing the file path string 
+     * with the external name token, to storing the index of it and storing 
+     * the string itself with ScExternalRefManager.  This way I can just 
+     * switch the file name here centrally instead of converting the token 
+     * arrays of all affected cells.
+     */
+}
+
+void ScExternalRefManager::clear()
+{
+    DocShellMap::iterator itrEnd = maDocShells.end();
+    for (DocShellMap::iterator itr = maDocShells.begin(); itr != itrEnd; ++itr)
+        itr->second->DoClose();
+
+    maDocShells.clear();
+}
+
diff --git sc/source/ui/docshell/makefile.mk sc/source/ui/docshell/makefile.mk
index db83fec..6286dfb 100644
--- sc/source/ui/docshell/makefile.mk
+++ sc/source/ui/docshell/makefile.mk
@@ -53,6 +53,7 @@ CXXFILES = \
         docsh6.cxx 	\
         docsh7.cxx 	\
         docsh8.cxx 	\
+        externalrefmgr.cxx \
         tablink.cxx 	\
         arealink.cxx 	\
         dbdocfun.cxx 	\
@@ -79,6 +80,7 @@ SLOFILES =  \
         $(SLO)$/docsh6.obj   	\
         $(SLO)$/docsh7.obj   	\
         $(SLO)$/docsh8.obj   	\
+        $(SLO)$/externalrefmgr.obj \
         $(SLO)$/tablink.obj   	\
         $(SLO)$/arealink.obj   	\
         $(SLO)$/dbdocfun.obj 	\
@@ -101,6 +103,7 @@ EXCEPTIONSFILES= \
         $(SLO)$/docsh3.obj	\
         $(SLO)$/docsh4.obj \
         $(SLO)$/docsh8.obj \
+        $(SLO)$/externalrefmgr.obj \
         $(SLO)$/dbdocimp.obj
 
 SRS1NAME=$(TARGET)
@@ -117,6 +120,7 @@ LIB1OBJFILES =  \
         $(SLO)$/docsh6.obj   	\
         $(SLO)$/docsh7.obj   	\
         $(SLO)$/docsh8.obj   	\
+        $(SLO)$/externalrefmgr.obj \
         $(SLO)$/tablink.obj   	\
         $(SLO)$/arealink.obj   	\
         $(SLO)$/dbdocfun.obj 	\
