--- svtools/source/fsstor/fsstorage.cxx.orig	2005-08-24 11:18:17.000000000 -0700
+++ svtools/source/fsstor/fsstorage.cxx	2005-09-06 13:23:58.753415152 -0700
@@ -697,10 +697,11 @@
 	INetURLObject aFolderURL( m_pImpl->m_aURL );
 	aFolderURL.Append( aStorName );
 
-	if ( ::utl::UCBContentHelper::IsDocument( aFolderURL.GetMainURL( INetURLObject::NO_DECODE ) ) )
-		throw io::IOException(); // TODO:
-
 	sal_Bool bFolderExists = ::utl::UCBContentHelper::IsFolder( aFolderURL.GetMainURL( INetURLObject::NO_DECODE ) );
+	if (!bFolderExists)
+		if ( ::utl::UCBContentHelper::IsDocument( aFolderURL.GetMainURL( INetURLObject::NO_DECODE ) ) )
+			throw io::IOException(); // TODO:
+
 	if ( ( nStorageMode & embed::ElementModes::NOCREATE ) && !bFolderExists )
 		throw io::IOException(); // TODO:
 
@@ -721,9 +722,6 @@
 		else if ( ( nStorageMode & embed::ElementModes::TRUNCATE ) )
 			throw io::IOException(); // TODO: access denied
 
-		if ( !::utl::UCBContentHelper::IsFolder( aFolderURL.GetMainURL( INetURLObject::NO_DECODE ) ) )
-			throw io::IOException(); // there is no such folder
-
 		::ucb::Content aResultContent( aFolderURL.GetMainURL( INetURLObject::NO_DECODE ), xDummyEnv );
 		xResult = uno::Reference< embed::XStorage >( 
 							static_cast< OWeakObject* >( new FSStorage(	aResultContent,
