--- /dev/null	2004-07-26 12:51:14.000000000 +0100
+++ svx/workben/gal.cxx	2005-02-21 14:39:40.505711696 +0000
@@ -0,0 +1,126 @@
+#include <stdio.h>
+#include <unistd.h>
+
+#include <comphelper/processfactory.hxx>
+#include <comphelper/regpathhelper.hxx>
+#include <cppuhelper/servicefactory.hxx>
+#include <cppuhelper/bootstrap.hxx>
+#include <com/sun/star/lang/XMultiServiceFactory.hpp>
+#include <com/sun/star/lang/XInitialization.hpp>
+#include <com/sun/star/registry/XSimpleRegistry.hpp>
+
+#include <ucbhelper/contentbroker.hxx>
+
+#include <tools/urlobj.hxx>
+#include <tools/fsys.hxx>
+#include <tools/intn.hxx>
+#include <svtools/filedlg.hxx>
+
+#include <vcl/window.hxx>
+#include <vcl/svapp.hxx>
+#include <vcl/font.hxx>
+#include <vcl/sound.hxx>
+#include <vcl/print.hxx>
+#include <vcl/toolbox.hxx>
+#include <vcl/help.hxx>
+#include <vcl/scrbar.hxx>
+#include <vcl/wrkwin.hxx>
+#include <vcl/msgbox.hxx>
+
+#include <osl/file.hxx>
+#include <osl/process.h>
+#include <rtl/bootstrap.hxx>
+
+#include <galtheme.hxx>
+#include <gallery1.hxx>
+
+#define GALLERY_PATH "file:///tmp/gallery"
+
+using namespace ::vos;
+using namespace ::rtl;
+using namespace ::com::sun::star::uno;
+using namespace ::com::sun::star::registry;
+using namespace ::com::sun::star::lang;
+
+class GalApp : public Application
+{
+public:
+	virtual void Main();
+};
+
+Reference< XMultiServiceFactory > createApplicationServiceManager()
+{
+    Reference< XMultiServiceFactory > xMS;
+	try
+	{
+		Reference< XComponentContext > xComponentContext = ::cppu::defaultBootstrap_InitialComponentContext();
+        if ( xComponentContext.is() )
+		    xMS = xMS.query( xComponentContext->getServiceManager() );
+	}
+	catch( ::com::sun::star::uno::Exception& )
+	{
+	}
+
+	return xMS;
+}
+
+static void createTheme()
+{
+	Gallery* pGallery;
+
+	String aGalleryName = String::CreateFromAscii( GALLERY_PATH );
+	String aThemeName = String::CreateFromAscii( "FooTheme" );
+
+	pGallery = Gallery::AcquireGallery( aGalleryName );
+	if (!pGallery ) {
+			fprintf( stderr, "Could't acquire '%s'\n",
+					 GALLERY_PATH );
+			exit( 1 );
+	}
+
+	fprintf( stderr, "Existing themes: %d\n",
+			 pGallery->GetThemeCount() );
+
+	GalleryTheme *pGalTheme;
+	if (!pGallery->CreateTheme( aThemeName ) ) {
+			fprintf( stderr, "Failed to create theme\n" );
+			exit( 1 );
+	}
+
+	fprintf( stderr, "Existing themes: %d\n",
+			 pGallery->GetThemeCount() );
+
+	SfxListener aListener;
+
+	if ( !( pGalTheme = pGallery->AcquireTheme( aThemeName, aListener ) ) ) {
+			fprintf( stderr, "Failed to acquire theme\n" );
+			exit( 1 );
+	}
+
+	pGallery->ReleaseTheme( pGalTheme, aListener );
+
+	Gallery::ReleaseGallery( pGallery );
+}
+
+void GalApp::Main()
+{
+    ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory >
+		xMSF = cppu::createRegistryServiceFactory(
+				rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "gal.rdb" ) ), sal_True );
+	::comphelper::setProcessServiceFactory( xMSF );
+
+	// Without this no file access works ...
+	ucb::ContentProviderDataList aData;
+	ucb::ContentProviderData aFileProvider(
+			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.ucb.FileContentProvider" ) ),
+			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "file" ) ),
+			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "" ) ) );
+	aData.push_back( aFileProvider );
+	ucb::ContentBroker::initialize( xMSF, aData );
+
+	fprintf( stderr, "Hello World\n" );
+
+	createTheme();
+}
+
+GalApp aGalApp;
