? svtools/workben/test.src.good
? svtools/workben/workben..dpr
Index: svtools/workben/makefile.mk
===================================================================
RCS file: /cvs/util/svtools/workben/makefile.mk,v
retrieving revision 1.4
diff -u -p -u -r1.4 makefile.mk
--- svtools/workben/makefile.mk	1 Oct 2002 12:28:04 -0000	1.4
+++ svtools/workben/makefile.mk	27 Aug 2003 12:16:28 -0000
@@ -91,6 +91,11 @@ APP1STDLIBS=	$(SVTOOLLIB)	\
 APP1DEPN=		$(L)$/itools.lib $(SVLIBDEPEND) 
 APP1OBJS=		$(OBJ)$/svdem.obj
 
+SRC1FILES=test.src
+SRS1NAME=test
+RES1TARGET=test
+SRS1FILES=$(SRS)$/test.srs
+
 
 # --- Targets -------------------------------------------------------
 
Index: svtools/workben/svdem.cxx
===================================================================
RCS file: /cvs/util/svtools/workben/svdem.cxx,v
retrieving revision 1.3
diff -u -p -u -r1.3 svdem.cxx
--- svtools/workben/svdem.cxx	15 Jan 2002 17:25:59 -0000	1.3
+++ svtools/workben/svdem.cxx	27 Aug 2003 12:16:28 -0000
@@ -257,6 +257,22 @@ public:
 
 // -----------------------------------------------------------------------
 
+class LayoutDlg : public ModalDialog
+{
+	PushButton aOk;
+	PushButton aCancel;
+	PushButton aSomething;
+  public:
+	LayoutDlg( Window *pParent) :
+		ModalDialog( pParent, ResId( 1001 ) ),
+		aOk        ( this,    ResId( 1002 ) ),
+		aCancel    ( this,    ResId( 1003 ) ),
+		aSomething ( this,    ResId( 1004 ) )
+	{
+		FreeResource();
+	}
+};
+
 void MyApp::Main()
 {
     try
@@ -273,7 +289,13 @@ void MyApp::Main()
         Help::EnableExtHelp();
         Help::EnableBalloonHelp();
         Help::EnableQuickHelp();
-        
+ 
+	ResMgr *pAppResMgr = new ResMgr( String( RTL_CONSTASCII_USTRINGPARAM( "test.res" ) ), NULL, NULL );
+	Resource::SetResManager( pAppResMgr );
+
+	LayoutDlg aResWin( NULL );
+	aResWin.Execute();
+       
         MyWin aMainWin( NULL, WinBits( WB_APP | WB_STDWORK | WB_CLIPCHILDREN ) );
         aMainWin.SetText( XubString( RTL_CONSTASCII_USTRINGPARAM( "SVTOOLS - Workbench" ) ) );
         aMainWin.GrabFocus();
Index: svtools/workben/test.src
===================================================================
RCS file: svtools/workben/test.src
diff -N svtools/workben/test.src
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ svtools/workben/test.src	27 Aug 2003 12:16:28 -0000
@@ -0,0 +1,55 @@
+#define DLG_TEST        1001
+#define BTN_OK          1002
+#define BTN_CANCEL      1003
+#define BTN_SOMETHING   1004
+#define DLG_TEST_LAYOUT 1100
+
+ModalDialog DLG_TEST
+{
+	OutputSize = TRUE ;
+	SVLook = TRUE;
+	Moveable = TRUE ;
+	Closeable = TRUE;
+	Text = "Layout test dialog";
+	Sizeable = TRUE;
+	LayoutId = DLG_TEST_LAYOUT;
+	Size = MAP_APPFONT(320, 50);
+
+	PushButton BTN_OK
+	{
+		Text = "Expand/Fill";
+	};
+	PushButton BTN_CANCEL
+	{
+		Text = "NoExpand";
+	};
+	PushButton BTN_SOMETHING
+	{
+		Text = "Expand/NoFill";
+		Size = MAP_APPFONT(50, 50);
+		Pos  = MAP_APPFONT(100, 100);
+	};
+
+	LayoutHBox DLG_TEST_LAYOUT {
+		Homogeneous = False;
+		Border = 4;
+		Layout {
+			PeerId = BTN_OK;
+			Expand = True;
+			Fill = True;
+			Padding = 3;
+		};
+		Layout {
+			PeerId = BTN_CANCEL;
+			Expand = False;
+			Fill = False;
+			Padding = 1;
+		};
+		Layout {
+			PeerId = BTN_SOMETHING;
+			Expand = True;
+			Fill = False;
+			Padding = 0;
+		};
+	};
+};
Index: svtools/prj/d.lst
===================================================================
RCS file: /cvs/util/svtools/prj/d.lst,v
retrieving revision 1.104
diff -u -p -u -r1.104 d.lst
--- svtools/prj/d.lst	16 Jul 2003 18:11:00 -0000	1.104
+++ svtools/prj/d.lst	27 Aug 2003 12:16:28 -0000
@@ -27,6 +27,7 @@ mkdir: %_DEST%\inc%_EXT%\svtools
 ..\%__SRC%\bin\svt?????.sym %_DEST%\bin%_EXT%\svt?????.sym
 ..\%__SRC%\bin\svl?????.dll %_DEST%\bin%_EXT%\svl?????.dll
 ..\%__SRC%\bin\bmp.* %_DEST%\bin%_EXT%\bmp.*
+..\%__SRC%\bin\test.* %_DEST%\bin%_EXT%\test.*
 ..\%__SRC%\bin\bmpsum.* %_DEST%\bin%_EXT%\bmpsum.*
 ..\%__SRC%\bin\bmpgui.* %_DEST%\bin%_EXT%\bmpgui.*
 ..\%__SRC%\bin\g2g.* %_DEST%\bin%_EXT%\g2g.*
