--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ offapi/org/openoffice/vba/XRange.idl	2005-05-13 13:26:08.071763202 +0530
@@ -0,0 +1,76 @@
+#ifndef __org_openoffice_vba_XRange_idl__
+#define __org_openoffice_vba_Range_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+//=============================================================================
+
+/** 
+ */
+published interface XRange: com::sun::star::uno::XInterface
+{
+
+	[attribute, readonly] string Address;
+
+	[attribute] long Column;
+
+	[attribute, readonly] long Count;
+
+	[attribute, readonly] org::openoffice::vba::XRange CurrentArray; 
+
+	[attribute, readonly] org::openoffice::vba::XRange CurrentRegion; 
+	
+	[attribute] string Formula; 
+
+	[attribute] string FormulaArray; //type?
+
+	[attribute] string NumberFormat; 
+
+	[attribute, readonly] long Row; 
+
+	[attribute] string Value; 
+	
+	//Methods  Note: Need to filter/prioritize the relevant methods
+
+	void cut([in] org::openoffice::vba::XRange rangeDestination);
+
+	void copy([in] org::openoffice::vba::XRange rangeDestination);
+
+	void clear();
+
+	void clearComments();
+
+	void clearContents();
+
+	void clearFormats();
+
+	void dataseries([in] string aRowCol, [in] string aType, [in] string aDate, [in] long nStep, [in] long nStop, [in] boolean bTrend);
+
+	void filldown();
+
+	void fillleft();
+
+	void fillright();
+
+	void fillup();
+
+	void insert([in] string aShift);
+
+	void resize([in] long nRowSize, [in] long nColumnSize);
+
+	/*more methods to go here...*/
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
