Index: qatesttool/global/win/dial_a_c.win
===================================================================
RCS file: /cvs/qa/qatesttool/global/win/dial_a_c.win,v
retrieving revision 1.12
diff -u -p -r1.12 dial_a_c.win
--- qatesttool/global/win/dial_a_c.win	13 Jan 2005 09:23:06 -0000	1.12
+++ qatesttool/global/win/dial_a_c.win	6 Feb 2006 10:33:03 -0000
@@ -79,7 +79,7 @@ CrLf sw:RadioButton:DLG_ASCII_FILTER:RB_
 Cr sw:RadioButton:DLG_ASCII_FILTER:RB_CR
 Lf sw:RadioButton:DLG_ASCII_FILTER:RB_LF
 
-*Attribute HID_SEARCHATTR
+*SearchAttribute HID_SEARCHATTR
 Auswahl HID_SEARCHATTR_CTL_ATTR
 
 ' *Aufzaehlungszeichen SID_OUTLINE_BULLET
Index: qatesttool/writer/level1/inc/regexp/search.inc
===================================================================
RCS file: /cvs/qa/qatesttool/writer/level1/inc/regexp/search.inc,v
retrieving revision 1.12
diff -u -p -r1.12 search.inc
--- qatesttool/writer/level1/inc/regexp/search.inc	7 Dec 2005 14:35:48 -0000	1.12
+++ qatesttool/writer/level1/inc/regexp/search.inc	6 Feb 2006 10:33:29 -0000
@@ -992,8 +992,8 @@ endcase
 
 function wCountNumberOfAttributes() as Integer
    Dim i as integer, AttributeCount as integer
-   Attribute.Click
-   Kontext "Attribute"
+   SearchAttribute.Click
+   Kontext "SearchAttribute"
    wait 500
    ' Count how much Attributes are in the list
    For i = 1 to 200
@@ -1006,7 +1006,7 @@ function wCountNumberOfAttributes() as I
       i = 202
     endcatch
    next i
-   Attribute.Close
+   SearchAttribute.Close
 end function
 
 ' ----------------------------------------------------------------------------------
@@ -1019,7 +1019,7 @@ function wSelectAttributeName(aNumber as
  Dim i as integer, aSelectIt as String
  Kontext "FindAndReplace"
  Attributes.Click
- Kontext "Attribute"
+ Kontext "SearchAttribute"
 
  Select Case aNumber
   Case 1
@@ -1341,7 +1341,7 @@ function wSelectAttributeName(aNumber as
    if AsianSupport = True then Warnlog "Unable to locate Attribute in list!"
    wSelectAttributeName = ""
   endif
-  Attribute.Ok
+  SearchAttribute.Ok
 end function
 
 
@@ -1596,13 +1596,13 @@ sub wSetFindAndReplaceToDefault(WhichDoc
     if CurrentSelectionOnly.IsEnabled then CurrentSelectionOnly.Uncheck
     'Column 3
     Attributes.Click
-    Kontext "Attribute"
+    Kontext "SearchAttribute"
     ' List has 32 Attributes
     For i = 1 to 32
         Auswahl.Select i
         Auswahl.Uncheck
     Next i
-    Attribute.OK
+    SearchAttribute.OK
     kontext "FindAndReplace"
     if NoFormat.IsEnabled then NoFormat.Click
     Kontext "FindAndReplace"
@@ -1689,7 +1689,7 @@ end sub
 function ToggleAttribute(sNumber as integer, sCheck as boolean)
     Kontext "FindAndReplace"
     Attributes.Click
-    Kontext "Attribute"
+    Kontext "SearchAttribute"
     Auswahl.Select sNumber             'Check Attributes underlined
     wait 500
     if sCheck = "True" then
@@ -1698,7 +1698,7 @@ function ToggleAttribute(sNumber as inte
         Auswahl.Uncheck
     endif
     wait 500
-    Attribute.OK
+    SearchAttribute.OK
     Kontext "FindAndReplace"
 end function
 
Index: qatesttool/writer/update/inc/w_002_.inc
===================================================================
RCS file: /cvs/qa/qatesttool/writer/update/inc/w_002_.inc,v
retrieving revision 1.35
diff -u -p -r1.35 w_002_.inc
--- qatesttool/writer/update/inc/w_002_.inc	30 Jan 2006 09:23:33 -0000	1.35
+++ qatesttool/writer/update/inc/w_002_.inc	6 Feb 2006 10:33:56 -0000
@@ -404,10 +404,10 @@ testcase tEditFindAndReplace
     
     Attributes.Click
     '/// in 'Find&Replace' dialog click 'Attributess' ///
-    Kontext "Attribute"
-    Call DialogTest ( Attribute )
+    Kontext "SearchAttribute"
+    Call DialogTest ( SearchAttribute )
     '/// Cancel dialog 'Attributess' ///
-    Attribute.Cancel
+    SearchAttribute.Cancel
 
     Kontext "FindAndReplace"
     Call DialogTest ( FindAndReplace )
