--- psprintBoldItalicPatch/inc/psprint/fontmanager.hxx	2005-08-24 15:07:47.287701000 +0100
+++ psprint/inc/psprint/fontmanager.hxx	2005-08-24 16:30:39.737278761 +0100
@@ -395,6 +395,7 @@
     std::hash_map< fontID, PrintFont* >       m_aFonts;
     std::hash_map< int, family::type >        m_aFamilyTypes;
     std::hash_map< int,  int >                m_aFamilyStyles;
+    std::list< fontID >                     m_aArtificialFonts;
     std::list< rtl::OUString >              m_aPrinterDrivers;
     std::list< rtl::OString >               m_aFontDirectories;
     std::list< int >							m_aPrivateFontDirectories;
@@ -497,6 +498,21 @@
     // get the font list and fast font info. see getFontList for pParser
     void getFontListWithFastInfo( std::list< FastPrintFontInfo >& rFonts, const PPDParser* pParser = NULL ) const;
 
+    // true if the font is an artificial one
+    bool isArtificial( fontID nFontID )
+    {
+        bool result = false;
+        ::std::list<int>::const_iterator it = m_aArtificialFonts.begin();
+        for ( ; it != m_aArtificialFonts.end(); ++it )
+        {
+            if ( *it == nFontID )
+            {
+                result = true;;
+                break;
+            }
+        }
+        return result;
+    }
     // get font info for a specific font
     bool getFontInfo( fontID nFontID, PrintFontInfo& rInfo ) const;
     // get fast font info for a specific font
--- psprintBoldItalicPatch/source/printergfx/text_gfx.cxx	2005-08-25 09:43:37.959840156 +0100
+++ psprint/source/printergfx/text_gfx.cxx	2005-08-24 16:41:48.354072375 +0100
@@ -193,6 +193,9 @@
     mnTextAngle                       = nAngle;
     mbTextVertical                    = bVertical;
 
+    bool fontIsArtificial = mrFontMgr.isArtificial(mnFontID );
+    maVirtualStatus.mbArtItalic = ( fontIsArtificial && ( mrFontMgr.getFontItalic(mnFontID)  == italic::Italic ) );
+    maVirtualStatus.mbArtBold = ( fontIsArtificial && (  mrFontMgr.getFontWeight(mnFontID) >= weight::Medium ) );
     return 0;
 }
 
--- psprintBoldItalicPatch/source/fontmanager/fontmanager.cxx	2005-08-24 15:38:14.320780000 +0100
+++ psprint/source/fontmanager/fontmanager.cxx	2005-08-24 16:29:40.876864139 +0100
@@ -246,6 +246,7 @@
             else if( (result)->m_eType == fonttype::Builtin )
                 m_aFontFileToFontID[ static_cast<BuiltinFont*>(result)->m_aMetricFile ].insert( aFont );
             m_aFonts[ aFont ] = result;
+            m_aArtificialFonts.push_back(aFont);
             // track what family's have what styles {italic, bold, 
             // italic-bold }
             updateFamilyStyles( result); 
@@ -2222,6 +2223,7 @@
         m_aFontDirectories.clear();
         m_aPrivateFontDirectories.clear();
         m_aFamilyStyles.clear();
+        m_aArtificialFonts.clear();
     }
 
 #if OSL_DEBUG_LEVEL > 1
