Index: rdbmaker/inc/codemaker/global.hxx
===================================================================
RCS file: /cvs/udk/rdbmaker/inc/codemaker/global.hxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 global.hxx
--- rdbmaker/inc/codemaker/global.hxx	19 Jul 2006 16:24:51 -0000	1.6
+++ rdbmaker/inc/codemaker/global.hxx	26 Jan 2007 15:20:30 -0000
@@ -94,10 +94,6 @@ sal_Bool fileExists(const ::rtl::OString
 sal_Bool checkFileContent(const ::rtl::OString& targetFileName, const ::rtl::OString& tmpFileName);
 
 const ::rtl::OString inGlobalSet(const ::rtl::OUString & r);
-inline const ::rtl::OString inGlobalSet(sal_Char* p)
-{
-	return inGlobalSet( ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(p) ) );
-}
 
 ::rtl::OUString convertToFileUrl(const ::rtl::OString& fileName);
 
Index: configmgr/source/backend/multistratumbackend.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/backend/multistratumbackend.cxx,v
retrieving revision 1.11
diff -u -p -u -r1.11 multistratumbackend.cxx
--- configmgr/source/backend/multistratumbackend.cxx	16 Sep 2006 15:06:31 -0000	1.11
+++ configmgr/source/backend/multistratumbackend.cxx	26 Jan 2007 16:43:38 -0000
@@ -359,8 +359,7 @@ void MultiStratumBackend::initializeBack
 		{
 			if(!bOptional)
 			{
-				static const sal_Char sErrContext[] = "MultiStratumBackend: Could not create Backend Stratum Service: ";
-				OUString const sContext(RTL_CONSTASCII_USTRINGPARAM(sErrContext));
+				OUString const sContext(RTL_CONSTASCII_USTRINGPARAM("MultiStratumBackend: Could not create Backend Stratum Service: "));
 				OUString const sMessage = sContext.concat(exception.Message);
 				
                 backenduno::StratumCreationException error(sMessage,*this,
@@ -817,10 +816,8 @@ void SAL_CALL MultiStratumBackend::dispo
  }
 //------------------------------------------------------------------------------
 
-static const sal_Char * const kBackendService = "com.sun.star.configuration.backend.Backend" ;
-
-static const sal_Char * const kImplementation =
-                "com.sun.star.comp.configuration.backend.MultiStratumBackend" ;
+#define kBackendService "com.sun.star.configuration.backend.Backend"
+#define kImplementation "com.sun.star.comp.configuration.backend.MultiStratumBackend"
 
 static const AsciiServiceName kServiceNames [] = 
 { 
Index: configmgr/source/backend/singlebackendadapter.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/backend/singlebackendadapter.cxx,v
retrieving revision 1.12
diff -u -p -u -r1.12 singlebackendadapter.cxx
--- configmgr/source/backend/singlebackendadapter.cxx	16 Sep 2006 15:06:59 -0000	1.12
+++ configmgr/source/backend/singlebackendadapter.cxx	26 Jan 2007 16:43:38 -0000
@@ -295,12 +295,10 @@ uno::Reference<backenduno::XUpdateHandle
 }
 //------------------------------------------------------------------------------
 
-static const sal_Char * const kBackendService = "com.sun.star.configuration.backend.Backend" ;
-static const sal_Char * const kAdapterService = "com.sun.star.configuration.backend.BackendAdapter" ;
-static const sal_Char * const kOnlineService  = "com.sun.star.configuration.backend.OnlineBackend" ;
-
-static const sal_Char * const kImplementation =
-                "com.sun.star.comp.configuration.backend.SingleBackendAdapter" ;
+#define kBackendService "com.sun.star.configuration.backend.Backend"
+#define kAdapterService "com.sun.star.configuration.backend.BackendAdapter"
+#define kOnlineService  "com.sun.star.configuration.backend.OnlineBackend"
+#define kImplementation "com.sun.star.comp.configuration.backend.SingleBackendAdapter"
 
 static const AsciiServiceName kServiceNames [] = 
 { 
Index: configmgr/source/inc/strings.hxx
===================================================================
RCS file: /cvs/util/configmgr/source/inc/strings.hxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 strings.hxx
--- configmgr/source/inc/strings.hxx	8 Sep 2005 03:57:34 -0000	1.5
+++ configmgr/source/inc/strings.hxx	26 Jan 2007 16:43:38 -0000
@@ -63,7 +63,6 @@ struct UStringDescription
 		, m_aString (_pName, _nLen, _encoding)
 		{
 		}
-
 	sal_Int32 getLength() const { return m_nLen; }
 	operator ::rtl::OUString const&() const { return m_aString; }
 	operator const sal_Char*() const { return m_pZeroTerminatedName; }
@@ -77,7 +76,8 @@ private:
 	extern ::configmgr::UStringDescription name
 
 #define IMPLEMENT_CONSTASCII_USTRING(name, asciivalue)	\
-	::configmgr::UStringDescription name(RTL_CONSTASCII_USTRINGPARAM(asciivalue))
+	::configmgr::UStringDescription name( \
+        asciivalue, ((sal_Int32)(sizeof(asciivalue)-1)), RTL_TEXTENCODING_ASCII_US )
 
 //.........................................................................
 }	// namespace frm
Index: configmgr/source/localbe/localmultistratum.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/localbe/localmultistratum.cxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 localmultistratum.cxx
--- configmgr/source/localbe/localmultistratum.cxx	16 Sep 2006 15:12:17 -0000	1.8
+++ configmgr/source/localbe/localmultistratum.cxx	26 Jan 2007 16:43:38 -0000
@@ -247,12 +247,9 @@ void LocalMultiStratum::getLayerDirector
 //------------------------------------------------------------------------------
 //------------------------------------------------------------------------------
 
-static const sal_Char * const kMultiStratumImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalMultiStratum" ;
-static const sal_Char * const kBackendService = 
-                "com.sun.star.configuration.backend.MultiStratum" ;
-static const sal_Char * const kLocalService = 
-                "com.sun.star.configuration.backend.LocalMultiStratum" ;
+#define kLocalService   "com.sun.star.configuration.backend.LocalMultiStratum"
+#define kBackendService "com.sun.star.configuration.backend.MultiStratum"
+#define kMultiStratumImplementation "com.sun.star.comp.configuration.backend.LocalMultiStratum"
 
 static AsciiServiceName kServiceNames [] = { kLocalService, 0, kBackendService, 0 } ;
 static const ServiceImplementationInfo kMultiStratumServiceInfo   = { kMultiStratumImplementation  , kServiceNames, kServiceNames + 2 } ;
Index: configmgr/source/localbe/localschemasupplier.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/localbe/localschemasupplier.cxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 localschemasupplier.cxx
--- configmgr/source/localbe/localschemasupplier.cxx	16 Sep 2006 15:12:46 -0000	1.5
+++ configmgr/source/localbe/localschemasupplier.cxx	26 Jan 2007 16:43:38 -0000
@@ -277,12 +277,9 @@ uno::Reference<backend::XSchema> SAL_CAL
 
 //------------------------------------------------------------------------------
 
-static const sal_Char * const kImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalSchemaSupplier" ;
-static const sal_Char * const kSchemaService = 
-                "com.sun.star.configuration.backend.SchemaSupplier" ;
-static const sal_Char * const kLocalService = 
-                "com.sun.star.configuration.backend.LocalSchemaSupplier" ;
+#define kImplementation "com.sun.star.comp.configuration.backend.LocalSchemaSupplier"
+#define kSchemaService  "com.sun.star.configuration.backend.SchemaSupplier"
+#define kLocalService   "com.sun.star.configuration.backend.LocalSchemaSupplier"
 
 static AsciiServiceName kServiceNames [] = {kLocalService, 0, kSchemaService, 0 } ;
 static const ServiceImplementationInfo kServiceInfo = { kImplementation, kServiceNames,kServiceNames+2 } ;
Index: configmgr/source/localbe/localsinglebackend.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/localbe/localsinglebackend.cxx,v
retrieving revision 1.21
diff -u -p -u -r1.21 localsinglebackend.cxx
--- configmgr/source/localbe/localsinglebackend.cxx	16 Sep 2006 15:13:02 -0000	1.21
+++ configmgr/source/localbe/localsinglebackend.cxx	26 Jan 2007 16:43:38 -0000
@@ -789,12 +789,9 @@ uno::Reference<backend::XUpdatableLayer>
 }
 //------------------------------------------------------------------------------
 
-static const sal_Char * const kImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalSingleBackend" ;
-static const sal_Char * const kBackendService = 
-                "com.sun.star.configuration.backend.SingleBackend" ;
-static const sal_Char * const kLocalService = 
-                "com.sun.star.configuration.backend.LocalSingleBackend" ;
+#define kImplementation "com.sun.star.comp.configuration.backend.LocalSingleBackend"
+#define kBackendService "com.sun.star.configuration.backend.SingleBackend"
+#define kLocalService   "com.sun.star.configuration.backend.LocalSingleBackend"
 
 static AsciiServiceName kServiceNames [] = { kLocalService, 0, kBackendService, 0 } ;
 static const ServiceImplementationInfo kServiceInfo = { kImplementation, kServiceNames, kServiceNames + 2 } ;
Index: configmgr/source/localbe/localsinglestratum.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/localbe/localsinglestratum.cxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 localsinglestratum.cxx
--- configmgr/source/localbe/localsinglestratum.cxx	16 Sep 2006 15:13:17 -0000	1.8
+++ configmgr/source/localbe/localsinglestratum.cxx	26 Jan 2007 16:43:38 -0000
@@ -178,18 +178,12 @@ void LocalResourceStratum::getLayerDirec
 //------------------------------------------------------------------------------
 //------------------------------------------------------------------------------
 
-static const sal_Char * const kLegacyStratumImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalSingleStratum" ;
-static const sal_Char * const kDataStratumImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalStratum" ;
-static const sal_Char * const kReadonlyStratumImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalReadonlyStratum" ;
-static const sal_Char * const kResourceStratumImplementation = 
-                "com.sun.star.comp.configuration.backend.LocalResourceStratum" ;
-static const sal_Char * const kBackendService = 
-                "com.sun.star.configuration.backend.SingleStratum" ;
-static const sal_Char * const kLocalService = 
-                "com.sun.star.configuration.backend.LocalSingleStratum" ;
+#define kLegacyStratumImplementation "com.sun.star.comp.configuration.backend.LocalSingleStratum"
+#define kDataStratumImplementation   "com.sun.star.comp.configuration.backend.LocalStratum"
+#define kReadonlyStratumImplementation "com.sun.star.comp.configuration.backend.LocalReadonlyStratum"
+#define kResourceStratumImplementation "com.sun.star.comp.configuration.backend.LocalResourceStratum"
+#define kBackendService "com.sun.star.configuration.backend.SingleStratum"
+#define kLocalService "com.sun.star.configuration.backend.LocalSingleStratum"
 
 static AsciiServiceName kServiceNames [] = { kLocalService, 0, kBackendService, 0 } ;
 static const ServiceImplementationInfo kLegacyStratumServiceInfo   = { kLegacyStratumImplementation  , kServiceNames, kServiceNames + 2 } ;
Index: configmgr/source/localbe/localstratumbase.hxx
===================================================================
RCS file: /cvs/util/configmgr/source/localbe/localstratumbase.hxx,v
retrieving revision 1.3
diff -u -p -u -r1.3 localstratumbase.hxx
--- configmgr/source/localbe/localstratumbase.hxx	8 Sep 2005 04:07:34 -0000	1.3
+++ configmgr/source/localbe/localstratumbase.hxx	26 Jan 2007 16:43:38 -0000
@@ -89,7 +89,7 @@ typedef cppu::WeakComponentImplHelper3<l
 
 struct LocalStratumMutexHolder { osl::Mutex mMutex; };
 //------------------------------------------------------------------------------
-static const char kLocalDataSuffix[] = ".xcu";
+#define kLocalDataSuffix ".xcu"
 //------------------------------------------------------------------------------
 
 /**
Index: configmgr/source/misc/bootstrapcontext.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/misc/bootstrapcontext.cxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 bootstrapcontext.cxx
--- configmgr/source/misc/bootstrapcontext.cxx	16 Sep 2006 15:14:28 -0000	1.6
+++ configmgr/source/misc/bootstrapcontext.cxx	26 Jan 2007 16:43:38 -0000
@@ -275,7 +275,7 @@ bool ComponentContext::lookupInBootstrap
 // ---------------------------------------------------------------------------
 // ---------------------------------------------------------------------------
 
-static const char k_TunneledContext[] = "/services/com.sun.star.configuration.bootstrap.Context";
+#define k_TunneledContext "/services/com.sun.star.configuration.bootstrap.Context"
 
 class UnoContextTunnel::Tunnel
 : public ::cppu::WeakImplHelper2< uno::XCurrentContext, lang::XUnoTunnel >
Index: configmgr/source/misc/configinteractionhandler.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/misc/configinteractionhandler.cxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 configinteractionhandler.cxx
--- configmgr/source/misc/configinteractionhandler.cxx	16 Sep 2006 15:14:56 -0000	1.5
+++ configmgr/source/misc/configinteractionhandler.cxx	26 Jan 2007 16:43:38 -0000
@@ -55,7 +55,7 @@ namespace {
 
 namespace css = com::sun::star;
 
-static char const INTERACTION_HANDLER[] = "configuration.interaction-handler";
+#define INTERACTION_HANDLER "configuration.interaction-handler"
 
 }
 
Index: configmgr/source/misc/providerfactory.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/misc/providerfactory.cxx,v
retrieving revision 1.23
diff -u -p -u -r1.23 providerfactory.cxx
--- configmgr/source/misc/providerfactory.cxx	16 Sep 2006 15:16:55 -0000	1.23
+++ configmgr/source/misc/providerfactory.cxx	26 Jan 2007 16:43:39 -0000
@@ -102,9 +102,7 @@ namespace configmgr
 		{
             using ::com::sun::star::configuration::CannotLoadConfigurationException;
 
-			static sal_Char const sCannotCreate[] = "Cannot create ConfigurationProvider. Unknown backend or factory error.";
-
-            throw CannotLoadConfigurationException( OUString(RTL_CONSTASCII_USTRINGPARAM(sCannotCreate)), *this );
+            throw CannotLoadConfigurationException( OUString(RTL_CONSTASCII_USTRINGPARAM("Cannot create ConfigurationProvider. Unknown backend or factory error.")), *this );
 		}
 
 		return xResult;
@@ -132,8 +130,7 @@ namespace configmgr
 
 			OSL_ASSERT(aContext.isBootstrapValid());
 
-			static const sal_Char sErrContext[] = "Cannot open Configuration: ";
-            OUString const sContext(RTL_CONSTASCII_USTRINGPARAM(sErrContext));
+            OUString const sContext(RTL_CONSTASCII_USTRINGPARAM("Cannot open Configuration: "));
 			e.Message = sContext.concat(e.Message);
 			throw;
 		}
Index: configmgr/source/treemgr/configexcept.cxx
===================================================================
RCS file: /cvs/util/configmgr/source/treemgr/configexcept.cxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 configexcept.cxx
--- configmgr/source/treemgr/configexcept.cxx	16 Sep 2006 15:27:40 -0000	1.5
+++ configmgr/source/treemgr/configexcept.cxx	26 Jan 2007 16:43:39 -0000
@@ -73,8 +73,8 @@ namespace configmgr
 		}
 		//---------------------------------------------------------------------
 
-		static const char c_sInvalidNamePre[] = "CONFIGURATION: Invalid Path or Name: ";
-		static const char c_sInvalidName[] = "CONFIGURATION: <Invalid Path or Name>";
+#define c_sInvalidNamePre "CONFIGURATION: Invalid Path or Name: "
+#define c_sInvalidName    "CONFIGURATION: <Invalid Path or Name>"
 //-----------------------------------------------------------------------------
 
 		InvalidName::InvalidName(OUString const& sName)
@@ -112,7 +112,8 @@ namespace configmgr
 			
 //-----------------------------------------------------------------------------
 
-		static const char c_sTypeMismatch[] = "CONFIGURATION: Data Types do not match: ";
+#define c_sTypeMismatch "CONFIGURATION: Data Types do not match: "
+
 	//---------------------------------------------------------------------
 		OUString TypeMismatch::describe(OUString const& sFoundType, OUString const& sExpectedType)
 		{
? eventattacher/unxlngx6.pro
Index: eventattacher/source/eventattacher.cxx
===================================================================
RCS file: /cvs/util/eventattacher/source/eventattacher.cxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 eventattacher.cxx
--- eventattacher/source/eventattacher.cxx	20 Jun 2006 00:15:13 -0000	1.6
+++ eventattacher/source/eventattacher.cxx	26 Jan 2007 16:59:16 -0000
@@ -925,7 +925,7 @@ void * SAL_CALL component_getFactory(
 	{
 		Reference< XSingleServiceFactory > xFactory( createOneInstanceFactory(
 			reinterpret_cast< XMultiServiceFactory * >( pServiceManager ),
-			OUString( RTL_CONSTASCII_USTRINGPARAM( pImplName ) ),
+			OUString( RTL_CONSTASCII_USTRINGPARAM( IMPLNAME ) ),
 			::comp_EventAttacher::EventAttacherImpl_CreateInstance, 
 			::comp_EventAttacher::EventAttacherImpl::getSupportedServiceNames_Static() ) );
 		
Index: tools/source/fsys/urlobj.cxx
===================================================================
RCS file: /cvs/util/tools/source/fsys/urlobj.cxx,v
retrieving revision 1.59
diff -u -p -u -r1.59 urlobj.cxx
--- tools/source/fsys/urlobj.cxx	17 Sep 2006 00:54:24 -0000	1.59
+++ tools/source/fsys/urlobj.cxx	26 Jan 2007 17:07:36 -0000
@@ -4777,8 +4777,7 @@ bool INetURLObject::setFSysPath(rtl::OUS
 	}
 
 	rtl::OUStringBuffer aSynAbsURIRef(
-		rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("file://"), 
-		RTL_TEXTENCODING_ASCII_US));
+		rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("file://")));
 	switch (eStyle)
 	{
 		case FSYS_VOS:
Index: tools/inc/string.hxx
===================================================================
RCS file: /cvs/util/tools/inc/string.hxx,v
retrieving revision 1.11
diff -u -p -u -r1.11 string.hxx
--- tools/inc/string.hxx	19 Jun 2006 13:30:16 -0000	1.11
+++ tools/inc/string.hxx	26 Jan 2007 17:43:39 -0000
@@ -489,6 +489,7 @@ public:
 						UniString( const ResId& rResId );
 						UniString( const UniString& rStr );
 						UniString( const UniString& rStr, xub_StrLen nPos, xub_StrLen nLen );
+                        UniString( rtl_uString * str, __sal_NoAcquire );
 						UniString( const rtl::OUString& rStr );
 						UniString( const sal_Unicode* pCharStr );
 						UniString( const sal_Unicode* pCharStr, xub_StrLen nLen );
Index: tools/source/string/strucvt.cxx
===================================================================
RCS file: /cvs/util/tools/source/string/strucvt.cxx,v
retrieving revision 1.10
diff -u -p -u -r1.10 strucvt.cxx
--- tools/source/string/strucvt.cxx	17 Sep 2006 01:03:58 -0000	1.10
+++ tools/source/string/strucvt.cxx	26 Jan 2007 17:43:58 -0000
@@ -143,6 +143,25 @@ UniString::UniString( const rtl::OUStrin
 	}
 }
 
+UniString::UniString( rtl_uString * pData, __sal_NoAcquire )
+    : mpData(NULL)
+{
+	DBG_CTOR( UniString, DbgCheckUniString );
+
+    OSL_ENSURE(rStr.pData->length < STRING_MAXLEN,
+               "Overflowing rtl::OUString -> UniString cut to zero length");
+
+	if (pData->length < STRING_MAXLEN)
+	{
+	    mpData = reinterpret_cast< UniStringData * >(pData);
+	    STRING_ACQUIRE((STRING_TYPE *)mpData);
+	}
+	else
+	{
+		STRING_NEW((STRING_TYPE **)&mpData);
+	}
+}
+
 // -----------------------------------------------------------------------
 
 UniString& UniString::Assign( const rtl::OUString& rStr )
Index: svtools/source/config/colorcfg.cxx
===================================================================
RCS file: /cvs/util/svtools/source/config/colorcfg.cxx,v
retrieving revision 1.14
diff -u -p -u -r1.14 colorcfg.cxx
--- svtools/source/config/colorcfg.cxx	12 Oct 2006 15:07:43 -0000	1.14
+++ svtools/source/config/colorcfg.cxx	26 Jan 2007 18:50:24 -0000
@@ -178,9 +178,14 @@ uno::Sequence< OUString> ColorConfig_Imp
 	OUString* pNames = aNames.getArray();
     struct ColorConfigEntryData_Impl
     {
+#ifdef RTL_INLINE_STRINGS
+        rtl_uString   * pString;
+        __sal_NoAcquire eDummy;
+#else
         const sal_Char* cName;
         sal_Int32       nLength;
         rtl_TextEncoding eEncoding;
+#endif
         sal_Bool bCanBeVisible;
     };
     static const ColorConfigEntryData_Impl cNames[] =
@@ -235,7 +240,11 @@ uno::Sequence< OUString> ColorConfig_Imp
     {
 		rtl::OUString sBaseName(sBase);
         sal_Int32 nPos = i / 4;
+#ifdef RTL_INLINE_STRINGS
+        sBaseName += OUString(cNames[nPos].pString);
+#else
         sBaseName += OUString(cNames[nPos].cName, cNames[nPos].nLength, cNames[nPos].eEncoding);
+#endif
         pNames[nIndex] += sBaseName;
         pNames[nIndex++] += sColor;
         if(cNames[nPos].bCanBeVisible)
Index: svtools/source/uno/unoevent.cxx
===================================================================
RCS file: /cvs/util/svtools/source/uno/unoevent.cxx,v
retrieving revision 1.9
diff -u -p -u -r1.9 unoevent.cxx
--- svtools/source/uno/unoevent.cxx	17 Sep 2006 15:30:27 -0000	1.9
+++ svtools/source/uno/unoevent.cxx	26 Jan 2007 18:59:33 -0000
@@ -77,8 +77,8 @@ using ::rtl::OUString;
 using ::rtl::OUStringBuffer;
 
 
-const sal_Char sAPI_ServiceName[] = "com.sun.star.container.XNameReplace";
-const sal_Char sAPI_SvDetachedEventDescriptor[] = "SvDetachedEventDescriptor";
+#define sAPI_ServiceName "com.sun.star.container.XNameReplace"
+#define sAPI_SvDetachedEventDescriptor "SvDetachedEventDescriptor"
 
 //
 // SvBaseEventDescriptor
Index: xmloff/source/core/DomBuilderContext.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/core/DomBuilderContext.cxx,v
retrieving revision 1.4
diff -u -p -u -r1.4 DomBuilderContext.cxx
--- xmloff/source/core/DomBuilderContext.cxx	17 Sep 2006 10:18:13 -0000	1.4
+++ xmloff/source/core/DomBuilderContext.cxx	26 Jan 2007 19:17:13 -0000
@@ -206,8 +206,6 @@ void DomBuilderContext::Characters( cons
 // helper function implementations
 //
 
-const sal_Char sDocumentBuilder[] = "com.sun.star.xml.dom.DocumentBuilder";
-
 Reference<XNode> lcl_createDomInstance()
 {
     Reference<XMultiServiceFactory> xFactory = utl::getProcessServiceFactory();
@@ -215,7 +213,7 @@ Reference<XNode> lcl_createDomInstance()
 
     Reference<XDocumentBuilder> xBuilder(
         xFactory->createInstance( 
-            OUString( RTL_CONSTASCII_USTRINGPARAM( sDocumentBuilder ) ) ), 
+            OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.xml.dom.DocumentBuilder" ) ) ), 
         UNO_QUERY_THROW );
 
     return Reference<XNode>( xBuilder->newDocument(), UNO_QUERY_THROW );
