--- /dev/null	2006-04-07 20:25:18.000000000 +0100
+++ filter/source/config/fragments/filters/OfficeOpen.xcu	2006-04-19 13:57:17.000000000 +0100
@@ -0,1 +1,14 @@
+	<node oor:name="Office Open XML" oor:op="replace">
+		<prop oor:name="Flags"><value>IMPORT EXPORT TEMPLATE</value></prop>
+		<prop oor:name="UIComponent"/>
+		<prop oor:name="FilterService"/>
+		<prop oor:name="UserData"><value></value></prop>
+		<prop oor:name="FileFormatVersion"><value>6800</value></prop>
+		<prop oor:name="Type"><value>calc_OfficeOpen</value></prop>
+		<prop oor:name="TemplateName"/>
+		<prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+		<prop oor:name="UIName">
+			<value xml:lang="x-default">ECMA Office Open</value>
+		</prop>
+	</node>
--- /dev/null	2006-04-07 20:25:18.000000000 +0100
+++ filter/source/config/fragments/types/calc_OfficeOpen.xcu	2006-04-19 13:55:29.000000000 +0100
@@ -0,0 +1,12 @@
+	<node oor:name="calc_OfficeOpen" oor:op="replace">
+		<prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+		<prop oor:name="URLPattern"/>
+		<prop oor:name="Extensions"><value>xlsx</value></prop>
+		<prop oor:name="MediaType"><value>application/vnd.ms-excel-FIXME</value></prop>
+		<prop oor:name="Preferred"><value>false</value></prop>
+		<prop oor:name="PreferredFilter"><value>Office Open XML</value></prop>
+		<prop oor:name="UIName">
+			<value>ECMA Office Open</value>
+		</prop>
+		<prop oor:name="ClipboardFormat"><value>OfficeOpen</value></prop>
+	</node>
Index: sc/source/ui/docshell/docsh.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/docshell/docsh.cxx,v
retrieving revision 1.83
diff -u -p -u -r1.83 docsh.cxx
--- sc/source/ui/docshell/docsh.cxx	27 Jan 2006 15:51:57 -0000	1.83
+++ sc/source/ui/docshell/docsh.cxx	20 Apr 2006 10:51:09 -0000
@@ -185,6 +185,7 @@ static const sal_Char __FAR_DATA pFilter
 static const sal_Char __FAR_DATA pFilterHtml[]		= "HTML (StarCalc)";
 static const sal_Char __FAR_DATA pFilterHtmlWebQ[]	= "calc_HTML_WebQuery";
 static const sal_Char __FAR_DATA pFilterRtf[]		= "Rich Text Format (StarCalc)";
+static const sal_Char __FAR_DATA pFilterOOXml[]		= "ooxml";
 
 //----------------------------------------------------------------------
 
@@ -886,6 +888,14 @@ BOOL __EXPORT ScDocShell::ConvertFrom( S
 			bSetColWidths = TRUE;
 			bSetRowHeights = TRUE;
 		}
+		else if (aFltName.EqualsAscii(pFilterOOXml))
+		{
+			fprintf (stderr, "Load OO.o XML\n");
+            ScColumn::bDoubleAlloc = TRUE;
+            FltError eError = ScImportECMA( rMedium, &aDocument);
+            ScColumn::bDoubleAlloc = FALSE;
+			bRet = TRUE;
+		}
 		else if ( aFltName.EqualsAscii(pFilterExcel4) || aFltName.EqualsAscii(pFilterExcel5) ||
 				   aFltName.EqualsAscii(pFilterExcel95) || aFltName.EqualsAscii(pFilterExcel97) ||
 				   aFltName.EqualsAscii(pFilterEx4Temp) || aFltName.EqualsAscii(pFilterEx5Temp) ||

Index: sc/util/makefile.mk
===================================================================
RCS file: /cvs/sc/sc/util/makefile.mk,v
retrieving revision 1.33
diff -u -p -u -r1.33 makefile.mk
--- sc/util/makefile.mk	27 Jan 2006 15:52:35 -0000	1.33
+++ sc/util/makefile.mk	19 Apr 2006 13:08:14 -0000
@@ -182,6 +182,7 @@ LIB4FILES=	\
 	$(SLB)$/html.lib \
 	$(SLB)$/lotus.lib \
 	$(SLB)$/qpro.lib\
+	$(SLB)$/ecma.lib\
 	$(SLB)$/rtf.lib \
 	$(SLB)$/xml.lib \
 	$(SLB)$/accessibility.lib \


Index: sc/prj/build.lst
===================================================================
RCS file: /cvs/sc/sc/prj/build.lst,v
retrieving revision 1.11
diff -u -p -u -r1.11 build.lst
--- sc/prj/build.lst	27 Jan 2006 15:49:40 -0000	1.11
+++ sc/prj/build.lst	19 Apr 2006 13:08:47 -0000
@@ -39,6 +40,7 @@ sc	sc\source\filter\pch					get		-	all	s
 sc	sc\source\filter\excel					nmake	-	all	sc_excel sc_sdi NULL
 sc	sc\source\filter\lotus					nmake	-	all	sc_lotus sc_sdi NULL
 sc      sc\source\filter\qpro                                  nmake   -       all     sc_qpro sc_sdi NULL
+sc      sc\source\filter\ecma                                  nmake   -       all     sc_ecma sc_sdi NULL
 sc	sc\source\filter\starcalc				nmake	-	all	sc_scalc sc_sdi NULL
 sc	sc\source\filter\dif					nmake	-	all	sc_dif sc_sdi NULL
 sc	sc\source\filter\rtf					nmake	-	all	sc_rtf sc_sdi NULL
@@ -48,4 +50,4 @@ sc	sc\source\filter\xcl97					nmake	-	al
 sc	sc\addin\datefunc						nmake	-	all	sc_addfu sc_sdi NULL
 sc	sc\addin\rot13							nmake	-	all	sc_adrot sc_sdi NULL
 sc	sc\addin\util							nmake	-	all	sc_adutil sc_addfu sc_adrot sc_sdi NULL
-sc      sc\util                                                                 nmake   -       all     sc_util sc_addfu sc_adrot sc_adutil sc_app sc_attr sc_cctrl sc_cosrc sc_data sc_dbgui sc_dif sc_docsh sc_drfnc sc_excel sc_form sc_html sc_lotus sc_qpro sc_misc sc_name sc_nvipi sc_opt sc_page sc_rtf sc_scalc sc_style sc_tool sc_uisrc sc_undo sc_unobj sc_view sc_xcl97 sc_xml sc_acc sc_ftools sc_vba NULL
+sc      sc\util                                                                 nmake   -       all     sc_util sc_addfu sc_adrot sc_adutil sc_app sc_attr sc_cctrl sc_cosrc sc_data sc_dbgui sc_dif sc_docsh sc_drfnc sc_excel sc_form sc_html sc_lotus sc_ecma sc_qpro sc_misc sc_name sc_nvipi sc_opt sc_page sc_rtf sc_scalc sc_style sc_tool sc_uisrc sc_undo sc_unobj sc_view sc_xcl97 sc_xml sc_acc sc_ftools sc_vba NULL
