--- /dev/null	
+++ sc/source/ui/unoobj/vbaworkbook.hxx	
@@ -0,0 +1,65 @@
+#ifndef _VBA_WORKBOOK_HXX_
+#define _VBA_WORKBOOK_HXX_
+
+#include <org/openoffice/vba/XWorkBook.hpp>
+
+class ScModelObj;
+
+class ScVbaWorkBook : public org::openoffice::vba::XWorkBook
+{
+	// UNO sucks for not providing a sane, thread-safe
+	// base-class for this.
+	sal_Int32   mnRefCount;
+	ScModelObj *mpModel;
+
+public:
+	ScVbaWorkBook( ScModelObj *pModel );
+	virtual ~ScVbaWorkBook();
+
+	// XInterface
+    virtual void acquire() throw () { ++mnRefCount; }
+    virtual void release() throw () { if (! --mnRefCount) delete this; }
+    virtual com::sun::star::uno::Any queryInterface(
+			const com::sun::star::uno::Type & rType )
+				throw (com::sun::star::uno::RuntimeException);
+
+    // Attributes
+    virtual ::rtl::OUString SAL_CALL getName() throw (::com::sun::star::uno::RuntimeException)
+			{ return rtl::OUString::createFromAscii( "VBA workbook name" ); }
+    virtual ::rtl::OUString SAL_CALL getPath() throw (::com::sun::star::uno::RuntimeException)
+			{ return rtl::OUString::createFromAscii( "VBA dummy path" ); }
+    virtual ::rtl::OUString SAL_CALL getFullName() throw (::com::sun::star::uno::RuntimeException)
+			{ return rtl::OUString::createFromAscii( "VBA full name" ); }
+    virtual void SAL_CALL setFullName( const ::rtl::OUString& _fullname ) throw (::com::sun::star::uno::RuntimeException)
+			{ }
+    virtual ::sal_Int32 SAL_CALL getCalculationVersion() throw (::com::sun::star::uno::RuntimeException)
+			{ return 1; }
+    virtual ::sal_Bool SAL_CALL getProtectStructure() throw (::com::sun::star::uno::RuntimeException)
+			{ return TRUE; }
+    virtual ::sal_Bool SAL_CALL getProtectWindows() throw (::com::sun::star::uno::RuntimeException)
+			{ return TRUE; }
+    virtual ::sal_Bool SAL_CALL getSaved() throw (::com::sun::star::uno::RuntimeException) 
+			{ return TRUE; }
+    virtual void SAL_CALL setSaved( ::sal_Bool _saved ) throw (::com::sun::star::uno::RuntimeException) 
+			{ }
+    virtual ::sal_Bool SAL_CALL getVBASigned() throw (::com::sun::star::uno::RuntimeException)
+			{ return FALSE; }
+
+    // Methods
+    virtual void SAL_CALL activate() throw (::com::sun::star::uno::RuntimeException)
+			{}
+    virtual void SAL_CALL close( ::sal_Bool bSaveChanges, const ::rtl::OUString& aFileName, ::sal_Bool bRouteWorkBook ) throw (::com::sun::star::uno::RuntimeException)
+			{}
+    virtual void SAL_CALL save() throw (::com::sun::star::uno::RuntimeException)
+			{}
+    virtual void SAL_CALL saveas( const ::rtl::OUString& aFileName ) throw (::com::sun::star::uno::RuntimeException)
+			{}
+    virtual void SAL_CALL printout() throw (::com::sun::star::uno::RuntimeException)
+			{}
+    virtual void SAL_CALL printpreview() throw (::com::sun::star::uno::RuntimeException)
+			{}
+    virtual void SAL_CALL protect() throw (::com::sun::star::uno::RuntimeException) 
+			{}
+};
+
+#endif /* _VBA_WORKBOOK_HXX_ */
