--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ sc/source/ui/unoobj/vbaworksheet.cxx	2005-05-13 11:58:02.250476165 +0530
@@ -0,0 +1,28 @@
+#include <cppuhelper/queryinterface.hxx>
+
+#include "docuno.hxx"
+#include "vbaworksheet.hxx"
+
+using namespace ::org::openoffice;
+using namespace ::com::sun::star;
+
+ScVbaWorkSheet::ScVbaWorkSheet( ScModelObj *pModel )
+{
+	mnRefCount = 1;
+    mpModel = pModel;
+}
+
+ScVbaWorkSheet::~ScVbaWorkSheet()
+{
+}
+
+// XInterface implementation
+uno::Any ScVbaWorkSheet::queryInterface( const uno::Type & rType ) throw (uno::RuntimeException)
+{
+    return cppu::queryInterface
+			(rType, 
+			 static_cast< uno::XInterface * >( static_cast< vba::XWorkSheet * >( this ) ),
+			 static_cast< vba::XWorkSheet * >( this ) );
+}
+
+
