--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ sc/source/ui/unoobj/vbarange.cxx	2005-05-17 11:42:27.000000000 +0530
@@ -0,0 +1,26 @@
+#include <cppuhelper/queryinterface.hxx>
+
+#include "docuno.hxx"
+#include "vbarange.hxx"
+
+using namespace ::org::openoffice;
+using namespace ::com::sun::star;
+
+ScVbaRange::ScVbaRange( ScTableSheetsObj *pTableSheets )
+{
+	mnRefCount = 1;
+    mpTableSheets = pTableSheets;
+}
+
+ScVbaRange::~ScVbaRange()
+{
+}
+
+// XInterface implementation
+uno::Any ScVbaRange::queryInterface( const uno::Type & rType ) throw (uno::RuntimeException)
+{
+    return cppu::queryInterface
+			(rType, 
+			 static_cast< uno::XInterface * >( static_cast< vba::XRange * >( this ) ),
+			 static_cast< vba::XRange * >( this ) );
+}
