--- /dev/null	
+++ sc/source/ui/unoobj/vbaworkbook.cxx	
@@ -0,0 +1,26 @@
+#include <cppuhelper/queryinterface.hxx>
+
+#include "docuno.hxx"
+#include "vbaworkbook.hxx"
+
+using namespace ::org::openoffice;
+using namespace ::com::sun::star;
+
+ScVbaWorkBook::ScVbaWorkBook( ScModelObj *pModel )
+{
+	mnRefCount = 1;
+    mpModel = pModel;
+}
+
+ScVbaWorkBook::~ScVbaWorkBook()
+{
+}
+
+// XInterface implementation
+uno::Any ScVbaWorkBook::queryInterface( const uno::Type & rType ) throw (uno::RuntimeException)
+{
+    return cppu::queryInterface
+			(rType, 
+			 static_cast< uno::XInterface * >( static_cast< vba::XWorkBook * >( this ) ),
+			 static_cast< vba::XWorkBook * >( this ) );
+}
