diff --git a/sc/source/core/data/document.cxx b/sc/source/core/data/document.cxx
index b3d0607..372fb72 100644
--- sc/source/core/data/document.cxx
+++ sc/source/core/data/document.cxx
@@ -193,7 +193,34 @@ BOOL ScDocument::GetTable( const String& rName, SCTAB& rTab ) const
 
 BOOL ScDocument::ValidTabName( const String& rName ) const
 {
-    return rName.Len() > 0;
+    xub_StrLen nLen = rName.Len();
+    if (!nLen)
+        return false;
+
+    for (xub_StrLen i = 0; i < nLen; ++i)
+    {
+        const sal_Unicode c = rName.GetChar(i);
+        switch (c)
+        {
+            case ':':
+            case '\\':
+            case '/':
+            case '?':
+            case '*':
+            case '[':
+            case ']':
+                // these characters are not allowed to match XL's convention.
+                return false;
+            case '\'':
+                if (i == 0 || i == nLen - 1)
+                    // single quote is not allowed at the first or last 
+                    // character position.
+                    return false;
+            break;
+        }
+    }
+
+    return true;
 }
 
 
