diff -r -u framework/source/xml/menudocumenthandler.cxx /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/menudocumenthandler.cxx
--- framework/source/xml/menudocumenthandler.cxx	2007-01-29 18:19:20.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/menudocumenthandler.cxx	2006-12-01 17:54:14.000000000 +0000
@@ -163,7 +163,6 @@
         if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_COMMANDURL ))
         {
             rProp[i].Value >>= rCommandURL;
-            rCommandURL = rCommandURL.intern();
         }
         else if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_HELPURL ))
         {
Only in framework/source/xml: menudocumenthandler.cxx~
diff -r -u framework/source/xml/toolboxdocumenthandler.cxx /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/toolboxdocumenthandler.cxx
--- framework/source/xml/toolboxdocumenthandler.cxx	2007-01-29 17:42:34.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/toolboxdocumenthandler.cxx	2006-12-01 17:54:14.000000000 +0000
@@ -122,10 +122,7 @@
     for ( sal_Int32 i = 0; i < rProp.getLength(); i++ )
     {
         if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_COMMANDURL ))
-        {
             rProp[i].Value >>= rCommandURL;
-            rCommandURL = rCommandURL.intern();
-        }
         else if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_HELPURL ))
             rProp[i].Value >>= rHelpURL;
         else if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_LABEL ))
Only in framework/source/xml: toolboxdocumenthandler.cxx~
--- framework/source/services/urltransformer.cxx	2007-01-29 18:25:38.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/services/urltransformer.cxx	2006-12-01 17:54:13.000000000 +0000
@@ -229,7 +229,6 @@
 				// INetURLObject supports only an intelligent method of parsing URL's. So write
 				// back Complete to have a valid encoded URL in all cases!
 				aURL.Complete	= aParser.GetMainURL( INetURLObject::NO_DECODE           );
-                aURL.Complete   = aURL.Complete.intern();
 
 				aParser.SetMark	( OUString() );
 				aParser.SetParam( OUString() );
diff -r -u framework/source/xml/menudocumenthandler.cxx /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/menudocumenthandler.cxx
--- framework/source/xml/menudocumenthandler.cxx	2007-01-29 18:19:20.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/menudocumenthandler.cxx	2006-12-01 17:54:14.000000000 +0000
@@ -163,7 +163,6 @@
         if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_COMMANDURL ))
         {
             rProp[i].Value >>= rCommandURL;
-            rCommandURL = rCommandURL.intern();
         }
         else if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_HELPURL ))
         {
Only in framework/source/xml: menudocumenthandler.cxx~
diff -r -u framework/source/xml/toolboxdocumenthandler.cxx /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/toolboxdocumenthandler.cxx
--- framework/source/xml/toolboxdocumenthandler.cxx	2007-01-29 17:42:34.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/framework/source/xml/toolboxdocumenthandler.cxx	2006-12-01 17:54:14.000000000 +0000
@@ -122,10 +122,7 @@
     for ( sal_Int32 i = 0; i < rProp.getLength(); i++ )
     {
         if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_COMMANDURL ))
-        {
             rProp[i].Value >>= rCommandURL;
-            rCommandURL = rCommandURL.intern();
-        }
         else if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_HELPURL ))
             rProp[i].Value >>= rHelpURL;
         else if ( rProp[i].Name.equalsAscii( ITEM_DESCRIPTOR_LABEL ))
Only in framework/source/xml: toolboxdocumenthandler.cxx~
--- configmgr/source/backend/binaryreader.cxx	2007-01-29 18:22:49.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/configmgr/source/backend/binaryreader.cxx	2006-12-01 17:52:27.000000000 +0000
@@ -542,8 +542,8 @@
 
                 rtl_TextEncoding const enc = bIsAscii ? RTL_TEXTENCODING_ASCII_US : RTL_TEXTENCODING_UTF8;
 
-                rtl_uString_internConvert( &result.pData, pUTF, nStrLength, 
-                                           enc, OSTRING_TO_OUSTRING_CVTFLAGS);
+                rtl_string2UString( &result.pData, pUTF, nStrLength, 
+					                enc, OSTRING_TO_OUSTRING_CVTFLAGS);
 			}
 
 			return result;
--- vcl/source/gdi/impimagetree.cxx	2007-01-29 18:28:35.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/vcl/source/gdi/impimagetree.cxx	2006-12-01 17:58:01.000000000 +0000
@@ -594,7 +594,7 @@
 	}
 
 	if( !rReturn.IsEmpty() )
-		aBmpExHashMap[ rName.intern() ] = rReturn;
+		aBmpExHashMap[ rName ] = rReturn;
 	
 	return( !rReturn.IsEmpty() );
 }
--- package/source/zipapi/ZipFile.cxx	2007-01-29 18:31:27.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/package/source/zipapi/ZipFile.cxx	2006-12-01 17:55:38.000000000 +0000
@@ -798,9 +798,9 @@
 			if ( aEntry.nExtraLen > ZIP_MAXEXTRA )
 				throw ZipException( OUString( RTL_CONSTASCII_USTRINGPARAM ( "extra header info exceeds ZIP_MAXEXTRA bytes") ), Reference < XInterface > () );
 
-			aEntry.sName = rtl::OUString::intern ( (sal_Char *) aMemGrabber.getCurrentPos(), 
-                                                   aEntry.nNameLen, 
-                                                   RTL_TEXTENCODING_ASCII_US);
+			aEntry.sName = OUString ( (sal_Char *) aMemGrabber.getCurrentPos(), 
+									  aEntry.nNameLen, 
+									  RTL_TEXTENCODING_ASCII_US);
 
 			aMemGrabber.skipBytes( aEntry.nNameLen + aEntry.nExtraLen + nCommentLen );
 			aEntries[aEntry.sName] = aEntry;	
--- vcl/source/gdi/impimage.cxx	2007-01-29 19:19:06.000000000 +0000
+++ /opt/OpenOffice/HEAD/build/OOE680_m6/vcl/source/gdi/impimage.cxx	2007-01-26 17:25:30.000000000 +0000
@@ -142,11 +142,10 @@
 void ImplImageList::AddImage( const ::rtl::OUString &aName,
 							  USHORT nId, const BitmapEx &aBitmapEx )
 {
-    rtl::OUString aIName = aName.intern();
-	ImageAryData *pImg = new ImageAryData( aIName, nId, aBitmapEx );
+	ImageAryData *pImg = new ImageAryData( aName, nId, aBitmapEx );
 	maImages.push_back( pImg );
-	if( aIName.getLength() )
-		maNameHash [ aIName ] = pImg;
+	if( aName.getLength() )
+		maNameHash [ aName ] = pImg;
 }
 
 void ImplImageList::RemoveImage( USHORT nPos )
