Index: svx/source/gallery2/galctrl.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/gallery2/galctrl.cxx,v
retrieving revision 1.21
diff -u -p -u -r1.21 galctrl.cxx
--- svx/source/gallery2/galctrl.cxx	8 Sep 2005 23:06:32 -0000	1.21
+++ svx/source/gallery2/galctrl.cxx	1 Nov 2005 11:57:49 -0000
@@ -64,7 +64,6 @@ GalleryPreview::GalleryPreview( GalleryB
 {
 	SetHelpId( HID_GALLERY_WINDOW );
     InitSettings();
-	aSound.SetNotifyHdl( LINK( this, GalleryPreview, SoundEndHdl ) );
 }
 
 // ------------------------------------------------------------------------
@@ -77,7 +76,6 @@ GalleryPreview::GalleryPreview( Window* 
 {
 	SetHelpId( HID_GALLERY_PREVIEW );
     InitSettings();
-    aSound.SetNotifyHdl( LINK( this, GalleryPreview, SoundEndHdl ) );
 }
 
 // ------------------------------------------------------------------------
@@ -276,14 +274,6 @@ void GalleryPreview::PreviewMedia( const
 	}
 }
 
-// ------------------------------------------------------------------------
-
-IMPL_LINK( GalleryPreview, SoundEndHdl, Sound*, pSound )
-{
-	aSound.SetSoundName( String() );
-	return 0L;
-}
-
 // -------------------
 // - GalleryIconView -
 // -------------------
Index: svx/inc/galctrl.hxx
===================================================================
RCS file: /cvs/graphics/svx/inc/galctrl.hxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 galctrl.hxx
--- svx/inc/galctrl.hxx	8 Sep 2005 17:48:15 -0000	1.5
+++ svx/inc/galctrl.hxx	1 Nov 2005 11:57:49 -0000
@@ -45,7 +45,6 @@
 #include <vcl/menu.hxx>
 #include <vcl/edit.hxx>
 #include <vcl/combobox.hxx>
-#include <vcl/sound.hxx>
 #include <svtools/slstitm.hxx>
 #include <svtools/transfer.hxx> 
 #include <svtools/valueset.hxx>
@@ -74,7 +73,6 @@ private:
 	
     GraphicObject		aGraphicObj;
 	Rectangle			aPreviewRect;
-	Sound				aSound;
     GalleryTheme*       mpTheme;
 						
 	SVX_DLLPRIVATE BOOL				ImplGetGraphicCenterRect( const Graphic& rGraphic, Rectangle& rResultRect ) const;
@@ -95,8 +93,7 @@ private:
     // DragSourceHelper
 	SVX_DLLPRIVATE virtual void		StartDrag( sal_Int8 nAction, const Point& rPosPixel );
 
-						DECL_LINK( MenuSelectHdl, Menu* );
-						DECL_LINK( SoundEndHdl, Sound* );
+    DECL_LINK( MenuSelectHdl, Menu* );
 						
 public:					
 						
