Index: svtools/source/filter.vcl/filter/filter.cxx
===================================================================
RCS file: /cvs/util/svtools/source/filter.vcl/filter/filter.cxx,v
retrieving revision 1.58.14.1
diff -u -p -r1.58.14.1 filter.cxx
--- svtools/source/filter.vcl/filter/filter.cxx	14 Oct 2005 12:54:49 -0000	1.58.14.1
+++ svtools/source/filter.vcl/filter/filter.cxx	1 Dec 2005 14:54:18 -0000
@@ -1451,8 +1451,6 @@ USHORT GraphicFilter::ImportGraphic( Gra
             }
             else
             {
-                vcl::PNGReader::DisablePreviewMode();
-
                 // check if this PNG contains a GIF chunk!
                 const std::vector< vcl::PNGReader::ChunkData >&    rChunkData = aPNGReader.GetChunks();
                 std::vector< vcl::PNGReader::ChunkData >::const_iterator aIter( rChunkData.begin() );
@@ -1487,6 +1485,8 @@ USHORT GraphicFilter::ImportGraphic( Gra
 					eLinkType = GFX_LINK_TYPE_NATIVE_PNG;
 				}
 			}
+            if( nImportFlags & GRFILTER_I_FLAGS_FOR_PREVIEW )
+				vcl::PNGReader::DisablePreviewMode();
 		}
 		else if( aFilterName.EqualsIgnoreCaseAscii( IMP_JPEG ) )
 		{
