Index: solenv/inc/libs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/libs.mk,v
retrieving revision 1.96
diff -u -p -u -r1.96 libs.mk
--- solenv/inc/libs.mk	20 Jan 2006 14:22:05 -0000	1.96
+++ solenv/inc/libs.mk	20 Mar 2006 13:41:17 -0000
@@ -64,12 +64,13 @@ DUNOUTILLIBST=-ldunoutil
 DVCLUTILLIBST=-ldvclutil
 BPICONVLIB=-lbpiconv
 TOOLSLIB=-ltl$(OFFICEUPD)$(DLLPOSTFIX)
+UNOTYPESLIB=-lexcept$(OFFICEUPD)$(DLLPOSTFIX)
 # make sure some ancient libs are empty
 OSLLIB=
 RTLLIB=
 TECLIB=
 CPPULIB=-luno_cppu
-CPPUHELPERLIB=-luno_cppuhelper$(COMID)
+CPPUHELPERLIB=-luno_cppuhelper$(COMID) $(UNOTYPESLIB)
 .INCLUDE .IGNORE : ucbhelper$/version.mk
 UCBHELPERLIB=-lucbhelper$(UCBHELPER_MAJOR)$(COMID)
 REGLIB=-lreg

Index: cppuhelper/source/makefile.mk
===================================================================
RCS file: /cvs/udk/cppuhelper/source/makefile.mk,v
retrieving revision 1.41
diff -u -p -u -r1.41 makefile.mk
--- cppuhelper/source/makefile.mk	27 Oct 2005 17:19:33 -0000	1.41
+++ cppuhelper/source/makefile.mk	20 Mar 2006 13:11:41 -0000
@@ -151,8 +151,9 @@ SHL1TARGET=uno_$(TARGET)$(COMID)
 SHL1STDLIBS= \
 		$(SALLIB)		\
 		$(SALHELPERLIB)	\
-		$(CPPULIB)
+		$(CPPULIB) \
+		$(UNOTYPESLIB)
 
 SHL1DEPN=
 SHL1IMPLIB=i$(TARGET)
--- /dev/null	2006-03-11 13:25:00.000000000 +0000
+++ offuh/except/except.cxx	2006-03-17 20:34:49.000000000 +0000
@@ -0,1 +1,25 @@
+#include <sal/config.h>
+#include <cppu/macros.hxx>
+
+// We don't want to hide all this shared goodness:
+#undef CPPU_GCC_DLLPUBLIC_EXPORT
+#define CPPU_GCC_DLLPUBLIC_EXPORT 
+#undef CPPU_GCC_DLLPRIVATE
+#define CPPU_GCC_DLLPRIVATE 
+
+#define CPPU_INTERNAL_IMPL 1 
+
+// Evilness: sal/config.h includes stdlib.h
+// that has some workarounds for 'major' / 'minor'
+// issues, at least on Linux. These then cause problems
+// with chained constructors: FooClass : major(0) tec.
+
+// obscure CORBA compat interfaces
+#undef major
+#undef minor
+
+// com/sun/star/drawing/CaptionEscapeDirection.idl: keyword genius
+#define auto not_auto
+
+#include <catch.hxx>
--- /dev/null	2006-03-11 13:25:00.000000000 +0000
+++ offuh/except/makefile.mk	2006-03-17 11:44:41.000000000 +0000
@@ -0,0 +1,23 @@
+# --- Settings -----------------------------------------------------
+
+PRJ=..
+
+PRJNAME=offuh
+TARGET=unotypes
+ENABLE_EXCEPTIONS=TRUE
+
+.INCLUDE :  settings.mk
+
+LIB1TARGET= $(SLB)$/$(TARGET).lib
+LIB1OBJFILES= $(SLO)$/except.obj
+
+SHL1TARGET=except$(UPD)$(DLLPOSTFIX)
+SHL1LIBS=$(LIB1TARGET)
+SHL1STDLIBS= $(SALLIB) $(SALHELPERLIB) $(REGLIB) $(CPPULIB)
+
+# --- Targets -------------------------------------------------------
+
+.INCLUDE :  target.mk
+
+$(LIB1OBJFILES) : $(MISC)$/offuh.don
+


Index: bridges/source/cpp_uno/gcc3_linux_intel/makefile.mk
===================================================================
RCS file: /cvs/udk/bridges/source/cpp_uno/gcc3_linux_intel/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- bridges/source/cpp_uno/gcc3_linux_intel/makefile.mk	7 Sep 2005 22:21:49 -0000	1.7
+++ bridges/source/cpp_uno/gcc3_linux_intel/makefile.mk	27 Mar 2006 14:05:35 -0000
@@ -74,7 +74,8 @@ SHL1LIBS = $(SLB)$/cpp_uno_shared.lib
 
 SHL1STDLIBS= \
 	$(CPPULIB)			\
-	$(SALLIB)
+	$(SALLIB) \
+	$(UNOTYPESLIB)
 
 .ENDIF
 
Index: bridges/source/jni_uno/makefile.mk
===================================================================
RCS file: /cvs/udk/bridges/source/jni_uno/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- bridges/source/jni_uno/makefile.mk	9 Jan 2006 09:47:17 -0000	1.7
+++ bridges/source/jni_uno/makefile.mk	27 Mar 2006 14:05:36 -0000
@@ -65,7 +65,8 @@ SHL1STDLIBS= \
 	$(JVMACCESSLIB)			\
 	$(CPPULIB)			\
     $(SALLIB) \
-    $(SALHELPERLIB)
+    $(SALHELPERLIB) \
+	$(UNOTYPESLIB)
 
 SHL1VERSIONMAP=$(TARGET).map
 
Index: padmin/source/makefile.mk
===================================================================
RCS file: /cvs/gsl/padmin/source/makefile.mk,v
retrieving revision 1.18
diff -u -p -u -r1.18 makefile.mk
--- padmin/source/makefile.mk	8 Sep 2005 16:25:28 -0000	1.18
+++ padmin/source/makefile.mk	27 Mar 2006 19:56:52 -0000
@@ -44,6 +44,10 @@ LIBTARGET=NO
 
 .INCLUDE: settings.mk
 
+.IF "$(FONTCONFIG_FONTS)" != ""
+CFLAGS+=-DFONTCONFIG_FONTS
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SRS1NAME=$(TARGET)
@@ -67,11 +71,13 @@ SLOFILES=\
 	$(SLO)$/newppdlg.obj		\
 	$(SLO)$/prtsetup.obj		\
 	$(SLO)$/prgsbar.obj			\
-	$(SLO)$/fontentry.obj		\
 	$(SLO)$/helper.obj			\
 	$(SLO)$/adddlg.obj			\
 	$(SLO)$/titlectrl.obj
 
+.IF "$(FONTCONFIG_FONTS)" == ""
+SLOFILES+= $(SLO)$/fontentry.obj
+.ENDIF
 
 OBJFILES=\
 	$(OBJ)/pamain.obj
@@ -85,7 +91,8 @@ SHL1STDLIBS=\
 	$(TOOLSLIB)						\
 	$(COMPHELPERLIB)				\
 	$(CPPULIB)						\
-	$(SALLIB)
+	$(SALLIB) \
+	$(UNOTYPESLIB)
 
 APP1TARGET=spadmin.bin
 APP1DEPN+=$(SHL1TARGETN)
@@ -103,6 +110,7 @@ APP1STDLIBS=	\
 	$(CPPUHELPERLIB)	\
 	$(CPPULIB)			\
 	$(SALLIB)			\
+	$(UNOTYPESLIB)			\
 	-lXext -lX11
 
 .ENDIF

Index: goodies/source/filter.vcl/egif/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/egif/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/egif/makefile.mk	9 Sep 2005 02:31:59 -0000	1.8
+++ goodies/source/filter.vcl/egif/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -66,7 +66,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET=     egi$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     egif
-SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB)
+SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=       $(LB)$/egif.lib
 SHL1LIBS=       $(SLB)$/egif.lib
Index: goodies/source/filter.vcl/eos2met/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/eos2met/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/eos2met/makefile.mk	9 Sep 2005 02:34:15 -0000	1.8
+++ goodies/source/filter.vcl/eos2met/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -64,7 +64,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET=     eme$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     eos2met
-SHL1STDLIBS=    $(TOOLSLIB) $(SVLIB) $(SVTOOLLIB) $(CPPULIB) $(SALLIB)
+SHL1STDLIBS=    $(TOOLSLIB) $(SVLIB) $(SVTOOLLIB) $(CPPULIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=       $(LB)$/eos2met.lib
 SHL1LIBS=       $(SLB)$/eos2met.lib

Index: goodies/source/filter.vcl/epbm/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/epbm/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/epbm/makefile.mk	9 Sep 2005 02:36:32 -0000	1.8
+++ goodies/source/filter.vcl/epbm/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -64,7 +64,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET= 	epb$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	epbm
-SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB)	$(SVTOOLLIB) $(SALLIB)
+SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB)	$(SVTOOLLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=		$(LB)$/epbm.lib
 SHL1LIBS=		$(SLB)$/epbm.lib

Index: goodies/source/filter.vcl/epgm/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/epgm/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/epgm/makefile.mk	9 Sep 2005 02:38:50 -0000	1.8
+++ goodies/source/filter.vcl/epgm/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -65,7 +65,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET= 	epg$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	epgm
-SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB)
+SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=		$(LB)$/epgm.lib
 SHL1LIBS=		$(SLB)$/epgm.lib
Index: goodies/source/filter.vcl/epict/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/epict/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/epict/makefile.mk	9 Sep 2005 02:41:02 -0000	1.8
+++ goodies/source/filter.vcl/epict/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -65,7 +65,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET=     ept$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     epict
-SHL1STDLIBS=    $(TOOLSLIB) $(SVLIB) $(SVTOOLLIB) $(CPPULIB) $(SALLIB)
+SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=       $(LB)$/epict.lib
 SHL1LIBS=       $(SLB)$/epict.lib
Index: goodies/source/filter.vcl/eppm/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/eppm/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/eppm/makefile.mk	9 Sep 2005 02:43:20 -0000	1.8
+++ goodies/source/filter.vcl/eppm/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -64,7 +64,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET= 	epp$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	eppm
-SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB)
+SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(CPPULIB) $(SVTOOLLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=		$(LB)$/eppm.lib
 SHL1LIBS=		$(SLB)$/eppm.lib
Index: goodies/source/filter.vcl/eps/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/eps/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/eps/makefile.mk	9 Sep 2005 02:45:33 -0000	1.9
+++ goodies/source/filter.vcl/eps/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -65,7 +65,7 @@ RESLIB1SRSFILES=$(SRS)$/$(TARGET).srs
 
 SHL1TARGET= 	eps$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	eps
-SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(SVLIB) $(SVTOOLLIB) $(CPPULIB) $(SALLIB)
+SHL1STDLIBS=	$(TOOLSLIB) $(SVLIB) $(SVLIB) $(SVTOOLLIB) $(CPPULIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=		$(LB)$/eps.lib
 SHL1LIBS=		$(SLB)$/eps.lib
Index: goodies/source/filter.vcl/eras/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/eras/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/eras/makefile.mk	9 Sep 2005 02:46:14 -0000	1.9
+++ goodies/source/filter.vcl/eras/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =	$(SLO)$/eras.obj
 
 SHL1TARGET= 	era$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	eras
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/eras.lib
 SHL1LIBS=		$(SLB)$/eras.lib
 
Index: goodies/source/filter.vcl/etiff/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/etiff/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/etiff/makefile.mk	9 Sep 2005 02:46:42 -0000	1.9
+++ goodies/source/filter.vcl/etiff/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =	$(SLO)$/etiff.obj
 
 SHL1TARGET= 	eti$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	etiff
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/etiff.lib
 SHL1LIBS=		$(SLB)$/etiff.lib
 
Index: goodies/source/filter.vcl/expm/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/expm/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/expm/makefile.mk	9 Sep 2005 02:47:11 -0000	1.9
+++ goodies/source/filter.vcl/expm/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -56,7 +56,7 @@ SLOFILES =	$(SLO)$/expm.obj
 
 SHL1TARGET= 	exp$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	expm
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/expm.lib
 SHL1LIBS=		$(SLB)$/expm.lib
 
Index: goodies/source/filter.vcl/icgm/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/icgm/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- goodies/source/filter.vcl/icgm/makefile.mk	9 Sep 2005 02:53:03 -0000	1.7
+++ goodies/source/filter.vcl/icgm/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -123,7 +123,8 @@ SHL1STDLIBS = \
 			$(UNOTOOLSLIB)	\
 			$(TOOLSLIB)		\
 			$(CPPULIB)		\
-			$(SALLIB)
+			$(SALLIB)		\
+			$(UNOTYPESLIB)
 
 # --- Targets --------------------------------------------------------------
 
Index: goodies/source/filter.vcl/idxf/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/idxf/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/idxf/makefile.mk	9 Sep 2005 02:57:42 -0000	1.9
+++ goodies/source/filter.vcl/idxf/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -62,7 +62,7 @@ SLOFILES =  $(SLO)$/dxfgrprd.obj \
 
 SHL1TARGET=     idx$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     idxf
-SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=       $(LB)$/idxf.lib
 SHL1LIBS=       $(SLB)$/idxf.lib # $(LB)$/rtftoken.lib
Index: goodies/source/filter.vcl/ieps/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ieps/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/ieps/makefile.mk	9 Sep 2005 02:58:11 -0000	1.9
+++ goodies/source/filter.vcl/ieps/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =  $(SLO)$/ieps.obj
 
 SHL1TARGET=     ips$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     ieps
-SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=       $(LB)$/ieps.lib
 SHL1LIBS=       $(SLB)$/ieps.lib
 
Index: goodies/source/filter.vcl/ios2met/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ios2met/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/ios2met/makefile.mk	9 Sep 2005 02:58:41 -0000	1.9
+++ goodies/source/filter.vcl/ios2met/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =  $(SLO)$/ios2met.obj
 
 SHL1TARGET=     ime$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     ios2met
-SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 
 SHL1DEPN=       $(LB)$/ios2met.lib
 SHL1LIBS=       $(SLB)$/ios2met.lib # $(LB)$/rtftoken.lib
Index: goodies/source/filter.vcl/ipbm/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ipbm/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/ipbm/makefile.mk	9 Sep 2005 02:59:11 -0000	1.9
+++ goodies/source/filter.vcl/ipbm/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =	$(SLO)$/ipbm.obj
 
 SHL1TARGET= 	ipb$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	ipbm
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/ipbm.lib
 SHL1LIBS=		$(SLB)$/ipbm.lib
 
Index: goodies/source/filter.vcl/ipcd/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ipcd/makefile.mk,v
retrieving revision 1.10
diff -u -p -u -r1.10 makefile.mk
--- goodies/source/filter.vcl/ipcd/makefile.mk	9 Sep 2005 02:59:40 -0000	1.10
+++ goodies/source/filter.vcl/ipcd/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =  $(SLO)$/ipcd.obj
 
 SHL1TARGET=     icd$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     ipcd
-SHL1STDLIBS=    $(SVTOOLLIB) $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(SVTOOLLIB) $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=       $(LB)$/ipcd.lib
 SHL1LIBS=       $(SLB)$/ipcd.lib # $(LB)$/rtftoken.lib
 
Index: goodies/source/filter.vcl/ipcx/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ipcx/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/ipcx/makefile.mk	9 Sep 2005 03:00:10 -0000	1.9
+++ goodies/source/filter.vcl/ipcx/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =  $(SLO)$/ipcx.obj
 
 SHL1TARGET=     ipx$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     ipcx
-SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=       $(LB)$/ipcx.lib
 SHL1LIBS=       $(SLB)$/ipcx.lib
 
Index: goodies/source/filter.vcl/ipict/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ipict/makefile.mk,v
retrieving revision 1.9
diff -u -p -u -r1.9 makefile.mk
--- goodies/source/filter.vcl/ipict/makefile.mk	9 Sep 2005 03:00:39 -0000	1.9
+++ goodies/source/filter.vcl/ipict/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =  $(SLO)$/ipict.obj
 
 SHL1TARGET=     ipt$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     ipict
-SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=       $(LB)$/ipict.lib
 SHL1LIBS=       $(SLB)$/ipict.lib
 
Index: goodies/source/filter.vcl/ipsd/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/ipsd/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/ipsd/makefile.mk	9 Sep 2005 03:01:08 -0000	1.8
+++ goodies/source/filter.vcl/ipsd/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =	$(SLO)$/ipsd.obj
 
 SHL1TARGET= 	ipd$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	ipsd
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/ipsd.lib
 SHL1LIBS=		$(SLB)$/ipsd.lib
 
Index: goodies/source/filter.vcl/iras/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/iras/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/iras/makefile.mk	9 Sep 2005 03:01:37 -0000	1.8
+++ goodies/source/filter.vcl/iras/makefile.mk	27 Mar 2006 21:04:11 -0000
@@ -55,7 +55,7 @@ SLOFILES =	$(SLO)$/iras.obj
 
 SHL1TARGET= 	ira$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	iras
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/iras.lib
 SHL1LIBS=		$(SLB)$/iras.lib
 
Index: goodies/source/filter.vcl/itga/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/itga/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/itga/makefile.mk	9 Sep 2005 03:02:08 -0000	1.8
+++ goodies/source/filter.vcl/itga/makefile.mk	27 Mar 2006 21:04:13 -0000
@@ -55,7 +55,7 @@ SLOFILES =	$(SLO)$/itga.obj
 
 SHL1TARGET= 	itg$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB= 	itga
-SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=	$(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=		$(LB)$/itga.lib
 SHL1LIBS=		$(SLB)$/itga.lib
 
Index: goodies/source/filter.vcl/itiff/makefile.mk
===================================================================
RCS file: /cvs/graphics/goodies/source/filter.vcl/itiff/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- goodies/source/filter.vcl/itiff/makefile.mk	9 Sep 2005 03:03:37 -0000	1.8
+++ goodies/source/filter.vcl/itiff/makefile.mk	27 Mar 2006 21:04:13 -0000
@@ -57,7 +57,7 @@ SLOFILES =  $(SLO)$/itiff.obj    \
 
 SHL1TARGET=     iti$(UPD)$(DLLPOSTFIX)
 SHL1IMPLIB=     itiff
-SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB)
+SHL1STDLIBS=    $(VCLLIB) $(TOOLSLIB) $(SALLIB) $(UNOTYPESLIB)
 SHL1DEPN=       $(LB)$/itiff.lib
 SHL1LIBS=       $(SLB)$/itiff.lib # $(LB)$/rtftoken.lib
 
Index: sd/source/filter/eppt/makefile.mk
===================================================================
RCS file: /cvs/graphics/sd/source/filter/eppt/makefile.mk,v
retrieving revision 1.8
diff -u -p -u -r1.8 makefile.mk
--- sd/source/filter/eppt/makefile.mk	9 Sep 2005 03:20:06 -0000	1.8
+++ sd/source/filter/eppt/makefile.mk	28 Mar 2006 14:39:31 -0000
@@ -73,7 +73,8 @@ SHL1STDLIBS = 	\
 				$(TOOLSLIB) 		\
 				$(UCBHELPERLIB)		\
 				$(CPPULIB)			\
-				$(SALLIB)
+				$(SALLIB)		\
+				$(UNOTYPESLIB)
 
 # --- Targets --------------------------------------------------------------
 
