Index: solenv/inc/libs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/libs.mk,v
retrieving revision 1.96
diff -u -p -u -r1.96 libs.mk
--- solenv/inc/libs.mk	20 Jan 2006 14:22:05 -0000	1.96
+++ solenv/inc/libs.mk	20 Mar 2006 13:41:17 -0000
@@ -64,12 +64,13 @@ DUNOUTILLIBST=-ldunoutil
 DVCLUTILLIBST=-ldvclutil
 BPICONVLIB=-lbpiconv
 TOOLSLIB=-ltl$(OFFICEUPD)$(DLLPOSTFIX)
+UNOTYPESLIB=-lexcept$(OFFICEUPD)$(DLLPOSTFIX)
 # make sure some ancient libs are empty
 OSLLIB=
 RTLLIB=
 TECLIB=
 CPPULIB=-luno_cppu
-CPPUHELPERLIB=-luno_cppuhelper$(COMID)
+CPPUHELPERLIB=-luno_cppuhelper$(COMID) $(UNOTYPESLIB)
 .INCLUDE .IGNORE : ucbhelper$/version.mk
 UCBHELPERLIB=-lucbhelper$(UCBHELPER_MAJOR)$(COMID)
 REGLIB=-lreg

Index: cppuhelper/source/makefile.mk
===================================================================
RCS file: /cvs/udk/cppuhelper/source/makefile.mk,v
retrieving revision 1.41
diff -u -p -u -r1.41 makefile.mk
--- cppuhelper/source/makefile.mk	27 Oct 2005 17:19:33 -0000	1.41
+++ cppuhelper/source/makefile.mk	20 Mar 2006 13:11:41 -0000
@@ -151,8 +151,9 @@ SHL1TARGET=uno_$(TARGET)$(COMID)
 SHL1STDLIBS= \
 		$(SALLIB)		\
 		$(SALHELPERLIB)	\
-		$(CPPULIB)
+		$(CPPULIB) \
+		$(UNOTYPESLIB)
 
 SHL1DEPN=
 SHL1IMPLIB=i$(TARGET)
--- /dev/null	2006-03-11 13:25:00.000000000 +0000
+++ offuh/except/except.cxx	2006-03-17 20:34:49.000000000 +0000
@@ -0,1 +1,25 @@
+#include <sal/config.h>
+#include <cppu/macros.hxx>
+
+// We don't want to hide all this shared goodness:
+#undef CPPU_GCC_DLLPUBLIC_EXPORT
+#define CPPU_GCC_DLLPUBLIC_EXPORT 
+#undef CPPU_GCC_DLLPRIVATE
+#define CPPU_GCC_DLLPRIVATE 
+
+#define CPPU_INTERNAL_IMPL 1 
+
+// Evilness: sal/config.h includes stdlib.h
+// that has some workarounds for 'major' / 'minor'
+// issues, at least on Linux. These then cause problems
+// with chained constructors: FooClass : major(0) tec.
+
+// obscure CORBA compat interfaces
+#undef major
+#undef minor
+
+// com/sun/star/drawing/CaptionEscapeDirection.idl: keyword genius
+#define auto not_auto
+
+#include <catch.hxx>
--- /dev/null	2006-03-11 13:25:00.000000000 +0000
+++ offuh/except/makefile.mk	2006-03-17 11:44:41.000000000 +0000
@@ -0,0 +1,23 @@
+# --- Settings -----------------------------------------------------
+
+PRJ=..
+
+PRJNAME=offuh
+TARGET=unotypes
+ENABLE_EXCEPTIONS=TRUE
+
+.INCLUDE :  settings.mk
+
+LIB1TARGET= $(SLB)$/$(TARGET).lib
+LIB1OBJFILES= $(SLO)$/except.obj
+
+SHL1TARGET=except$(UPD)$(DLLPOSTFIX)
+SHL1LIBS=$(LIB1TARGET)
+SHL1STDLIBS= $(SALLIB) $(SALHELPERLIB) $(REGLIB) $(CPPULIB)
+
+# --- Targets -------------------------------------------------------
+
+.INCLUDE :  target.mk
+
+$(LIB1OBJFILES) : $(MISC)$/offuh.don
+


