Index: registry/source/regimpl.hxx
===================================================================
RCS file: /cvs/udk/registry/source/regimpl.hxx,v
retrieving revision 1.3
diff -u -p -u -r1.3 regimpl.hxx
--- registry/source/regimpl.hxx	9 Sep 2005 05:16:28 -0000	1.3
+++ registry/source/regimpl.hxx	20 Oct 2005 14:24:03 -0000
@@ -50,7 +50,8 @@
 #include	<store/store.hxx>
 #endif
 
-#define REG_PAGESIZE 512
+/* #define REG_PAGESIZE 512 */
+#define REG_PAGESIZE 128
 
 #define	REG_MODE_CREATE		store_AccessCreate
 #define	REG_MODE_OPEN		store_AccessReadWrite 
