--- sfx2/source/doc/docvor.hrc.OLD	Wed Sep  7 21:41:58 2005
+++ sfx2/source/doc/docvor.hrc	Tue Feb  7 15:34:11 2006
@@ -66,6 +66,7 @@
 #define BTN_EDIT 105
 #define BTN_FILES 3
 #define BTN_ADDRESSTEMPLATE	4
+#define BTN_IMPORT 5
 #define LB_LEFT 1
 #define FT_DEFAULT_TEMPLATE_LABEL 20
 #define FT_DEFAULT_TEMPLATE 21
--- sfx2/source/doc/docvor.src.OLD	Fri Sep 23 18:28:14 2005
+++ sfx2/source/doc/docvor.src	Tue Feb  7 15:34:08 2006
@@ -70,6 +70,15 @@
 		DefButton = TRUE ;
 		Text [ x-comment ] = " ";
 	};
+	PushButton BTN_IMPORT
+	{
+		Pos = MAP_APPFONT ( 205 , 63 ) ;
+		Size = MAP_APPFONT ( 60 , 14 ) ;
+		Text [ de ] = "Vorlage ~importieren..." ;
+		Text [ en-US ] = "Import Template..." ;
+		TabStop = TRUE ;
+		Text [ x-comment ] = " ";
+	};
 	PushButton BTN_FILES
 	{
 		Pos = MAP_APPFONT ( 205 , 143 ) ;
--- sfx2/source/doc/docvor.cxx.OLD	Fri Nov 11 16:24:31 2005
+++ sfx2/source/doc/docvor.cxx	Tue Feb  7 15:43:57 2006
@@ -193,6 +193,7 @@
 	MenuButton              aEditBtn;
 	HelpButton              aHelpBtn;
 	PushButton              aAddressTemplateBtn;
+    PushButton              aImportBtn;
 	PushButton              aFilesBtn;
 
 	Accelerator             aEditAcc;
@@ -202,6 +203,7 @@
 
 	SvStringsDtor* 			GetAllFactoryURLs_Impl() const;
 	sal_Bool				GetServiceName_Impl( String& rFactoryURL, String& rFileURL ) const;
+    void				    do_import( SvLBoxEntry *pEntry );
 	long                    Dispatch_Impl( USHORT nId, Menu* _pMenu );
 	String                  GetPath_Impl( BOOL bOpen, const String& rFileName );
     ::com::sun::star::uno::Sequence< ::rtl::OUString > GetPaths_Impl( const String& rFileName );
@@ -213,6 +215,7 @@
 	DECL_LINK( AccelSelect_Impl, Accelerator * );
 	DECL_LINK( MenuSelect_Impl, Menu * );
 	DECL_LINK( MenuActivate_Impl, Menu * );
+    DECL_LINK( Import_Impl, Button * );
 	DECL_LINK( AddFiles_Impl, Button * );
 	DECL_LINK( OnAddressTemplateClicked, Button * );
 	BOOL                    DontDelete_Impl( SvLBoxEntry *pEntry);
@@ -248,6 +251,7 @@
 	aEditBtn			( pParent, ResId( BTN_EDIT ) ),
 	aHelpBtn			( pParent, ResId( BTN_HELP ) ),
 	aAddressTemplateBtn	( pParent, ResId( BTN_ADDRESSTEMPLATE ) ),
+    aImportBtn			( pParent, ResId( BTN_IMPORT ) ),
 	aFilesBtn			( pParent, ResId( BTN_FILES ) ),
 
 	aEditAcc	( ResId( ACC_EDIT ) ),
@@ -298,6 +302,8 @@
 	aEditAcc.SetSelectHdl( LINK( this, SfxOrganizeDlg_Impl, AccelSelect_Impl ) );
 	GetpApp()->InsertAccel( &aEditAcc );
 
+	aImportBtn.SetClickHdl(
+		LINK(this,SfxOrganizeDlg_Impl, Import_Impl));
 	aFilesBtn.SetClickHdl(
 		LINK(this,SfxOrganizeDlg_Impl, AddFiles_Impl));
 	aAddressTemplateBtn.SetClickHdl(
@@ -1828,6 +1834,52 @@
 	return bRet;
 }
 
+void SfxOrganizeDlg_Impl::do_import( SvLBoxEntry *pEntry )
+{
+	USHORT nRegion = 0, nIndex = 0;
+	GetIndices_Impl( pFocusBox, pEntry, nRegion, nIndex );
+
+	::com::sun::star::uno::Sequence < ::rtl::OUString > Paths = GetPaths_Impl( String() );
+	sal_Int32 nCount = Paths.getLength();
+	if( nCount == 1 )
+	{
+		String aPath = String( Paths.getArray()[0] );
+		if ( aPath.Len() && !aMgr.CopyFrom( pFocusBox, nRegion, nIndex, aPath ) )
+		{
+			String aText( SfxResId( STR_ERROR_COPY_TEMPLATE ) );
+			aText.SearchAndReplaceAscii( "$1", aPath );
+			ErrorBox( pDialog, WB_OK, aText ).Execute();
+		}
+	}
+	else if( nCount > 1)
+	{
+		INetURLObject aPathObj( Paths[0] );
+		aPathObj.setFinalSlash();
+		for ( USHORT i = 1; i<nCount; ++i )
+		{
+			if ( i == 1 )
+				aPathObj.Append( Paths[i] );
+			else
+				aPathObj.setName( Paths[i] );
+			String aPath = aPathObj.GetMainURL( INetURLObject::NO_DECODE );
+			if ( aPath.Len() && !aMgr.CopyFrom( pFocusBox, nRegion, nIndex, aPath ) )
+			{
+				String aText( SfxResId( STR_ERROR_COPY_TEMPLATE ) );
+				aText.SearchAndReplaceAscii( "$1", aPath );
+				ErrorBox( pDialog, WB_OK, aText ).Execute();
+			}
+		}
+	}
+	
+	/*String aPath = GetPath_Impl( TRUE, String() );
+	if ( aPath.Len() && !aMgr.CopyFrom( pFocusBox, nRegion, nIndex, aPath ) )
+	{
+	    String aText( SfxResId( STR_ERROR_COPY_TEMPLATE ) );
+	    aText.SearchAndReplaceAscii( "$1", aPath );
+	    ErrorBox( pDialog, WB_OK, aText ).Execute();
+	}*/
+}
+
 long SfxOrganizeDlg_Impl::Dispatch_Impl( USHORT nId, Menu* _pMenu )
 
 /*  [Beschreibung]
@@ -1960,48 +2012,7 @@
 		{
 			if ( !pEntry )
 				return 1;
-			USHORT nRegion = 0, nIndex = 0;
-			GetIndices_Impl( pFocusBox, pEntry, nRegion, nIndex );
-
-            ::com::sun::star::uno::Sequence < ::rtl::OUString > Paths = GetPaths_Impl( String() );
-            sal_Int32 nCount = Paths.getLength();
-            if( nCount == 1 )
- 			{
-               String aPath = String( Paths.getArray()[0] );
-               if ( aPath.Len() && !aMgr.CopyFrom( pFocusBox, nRegion, nIndex, aPath ) )
-               {
-                   String aText( SfxResId( STR_ERROR_COPY_TEMPLATE ) );
-                   aText.SearchAndReplaceAscii( "$1", aPath );
-                   ErrorBox( pDialog, WB_OK, aText ).Execute();
-               }
-            }
-            else if( nCount > 1)
-            {
-                INetURLObject aPathObj( Paths[0] );
-                aPathObj.setFinalSlash();
-                for ( USHORT i = 1; i<nCount; ++i )
-				{
-					if ( i == 1 )
-                        aPathObj.Append( Paths[i] );
-					else
-                        aPathObj.setName( Paths[i] );
-                    String aPath = aPathObj.GetMainURL( INetURLObject::NO_DECODE );
-                    if ( aPath.Len() && !aMgr.CopyFrom( pFocusBox, nRegion, nIndex, aPath ) )
-                    {
-                        String aText( SfxResId( STR_ERROR_COPY_TEMPLATE ) );
-                        aText.SearchAndReplaceAscii( "$1", aPath );
-                        ErrorBox( pDialog, WB_OK, aText ).Execute();
-                    }
-                }
-            }
-
-            /*String aPath = GetPath_Impl( TRUE, String() );
-			if ( aPath.Len() && !aMgr.CopyFrom( pFocusBox, nRegion, nIndex, aPath ) )
-			{
-				String aText( SfxResId( STR_ERROR_COPY_TEMPLATE ) );
-				aText.SearchAndReplaceAscii( "$1", aPath );
-				ErrorBox( pDialog, WB_OK, aText ).Execute();
-            }*/
+			do_import( pEntry );
 			break;
 		}
 
@@ -2346,6 +2357,16 @@
 	svt::AddressBookSourceDialog aDialog(pDialog, ::comphelper::getProcessServiceFactory());
 	aDialog.Execute();
 	return 0L;
+}
+
+//-------------------------------------------------------------------------
+
+IMPL_LINK( SfxOrganizeDlg_Impl, Import_Impl, Button *, pButton )
+{
+	BOOL bEnable = ( pFocusBox && pFocusBox->GetSelectionCount() );
+	SvLBoxEntry* pEntry = bEnable ? pFocusBox->FirstSelected() : NULL;
+	do_import( pEntry );
+    return 0;
 }
 
 //-------------------------------------------------------------------------
