Index: sc/source/ui/unoobj/makefile.mk
===================================================================
RCS file: /cvs/sc/sc/source/ui/unoobj/makefile.mk,v
retrieving revision 1.16
diff -u -p -u -r1.16 makefile.mk
--- sc/source/ui/unoobj/makefile.mk	11 Aug 2004 09:08:22 -0000	1.16
+++ sc/source/ui/unoobj/makefile.mk	11 May 2005 11:00:42 -0000
@@ -122,7 +122,8 @@ SLO1FILES =  \
         $(SLO)$/listenercalls.obj \
         $(SLO)$/cellvaluebinding.obj \
         $(SLO)$/celllistsource.obj \
-        $(SLO)$/warnpassword.obj
+        $(SLO)$/warnpassword.obj \
+	$(SLO)$/vbaworkbook.obj
 
 SLO2FILES = \
 	$(SLO)$/scdetect.obj								\
