diff --git a/solenv/inc/_tg_srs.mk b/solenv/inc/_tg_srs.mk
index 5ed60b2..eaea344 100755
--- a/solenv/inc/_tg_srs.mk
+++ b/solenv/inc/_tg_srs.mk
@@ -29,6 +29,8 @@ $(foreach,i,$(SRC1FILES) $(COMMONMISC)$/$(TARGET)$/$i) : $$(@:f) localize.sdf
     $(RENAME) $@.$(INPATH) $@
     -$(RM) $@.$(INPATH)
 
+COLON_HACK := :
+
 $(SRS)$/$(SRS1NAME).srs: $(foreach,i,$(SRC1FILES) $(COMMONMISC)$/$(TARGET)$/$i)
 .ELSE			# "$(WITH_LANG)"!=""
 $(SRS)$/$(SRS1NAME).srs: $(SRC1FILES)
@@ -37,7 +39,7 @@ $(SRS)$/$(SRS1NAME).srs: $(SRC1FILES)
 	@echo Making: $@
 	$(RSC) -presponse @$(mktmp \
 		$(SRSDEFAULT) $(RSC_SRS_CHARSET) $(RSCFLAGS) -I$(RSCEXTINC) \
-		$(INCLUDE) $(RSCDEFS) $(RSCUPDVERDEF) \
+		$(subst,-Iz$(COLON_HACK),-I $(INCLUDE)) $(RSCDEFS) $(RSCUPDVERDEF) \
 		-fp=$@.$(INPATH) \
 		$< \
 	)
diff --git a/solenv/inc/rules.mk b/solenv/inc/rules.mk
index 626ef87..917c799 100755
--- a/solenv/inc/rules.mk
+++ b/solenv/inc/rules.mk
@@ -194,7 +194,7 @@ $(SLO)$/%.obj : %.cxx
 .IF "$(COM)"=="GCC"
 	$(CAPTURE_COMMAND) $(CXX) $(ACT_PCH_SWITCHES) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(SLO)$/$*.obj $(CFLAGSINCXX)$(PWD)$/$*.cxx $(CAPTURE_OUTPUT)
 .ELSE
-	$(CAPTURE_COMMAND) $(CXX) $(ACT_PCH_SWITCHES) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(SLO)$/$*.obj $(CFLAGSINCXX)$(PWD)$/$*.cxx $(CAPTURE_OUTPUT)
+	$(CAPTURE_COMMAND) $(CXX) $(ACT_PCH_SWITCHES) $(CFLAGS) $(INCLUDE) $(CFLAGSCXX) $(CFLAGSSLO) $(CDEFS) $(CDEFSSLO) $(CDEFSMT) $(!eq,$(EXCEPTIONSFILES),$(subst,$@, $(EXCEPTIONSFILES)) $(LOCAL_EXCEPTIONS_FLAGS) $(GLOBAL_EXCEPTIONS_FLAGS)) $(CFLAGSAPPEND) $(CFLAGSOUTOBJ)$(SLO)$/$*.obj $(CFLAGSINCXX)z:$(PWD)$/$*.cxx $(CAPTURE_OUTPUT)
 .ENDIF		    # "$(COM)"=="GCC"	
 .ENDIF			# "$(GUI)"=="UNX"
 .IF "$(nodep)"==""
diff --git a/solenv/inc/settings.mk b/solenv/inc/settings.mk
index f44bc59..8b0c1ce 100755
--- a/solenv/inc/settings.mk
+++ b/solenv/inc/settings.mk
@@ -967,7 +967,7 @@ SCPLINKFLAGS+=-v $(SCPLINKVERSION)
 .IF "$(make_srs_deps)"!=""
 RSC=rscdep
 .ELSE # "$(make_srs_deps)"!=""
-RSC=$(FLIPCMD) rsc
+RSC=rsc
 .ENDIF # "$(make_srs_deps)"!=""
 
 #new
diff --git a/solenv/inc/startup/wnt/CVS/Entries b/solenv/inc/startup/wnt/CVS/Entries
deleted file mode 100755
index 207dd3d..0000000
--- a/solenv/inc/startup/wnt/CVS/Entries
+++ /dev/null
@@ -1,2 +0,0 @@
-/macros.mk/1.25/Thu Mar  6 15:16:17 2008//TOOO300_m3
-D
diff --git a/solenv/inc/startup/wnt/CVS/Repository b/solenv/inc/startup/wnt/CVS/Repository
deleted file mode 100755
index c635f5b..0000000
--- a/solenv/inc/startup/wnt/CVS/Repository
+++ /dev/null
@@ -1 +0,0 @@
-tools/solenv/inc/startup/wnt
diff --git a/solenv/inc/startup/wnt/CVS/Root b/solenv/inc/startup/wnt/CVS/Root
deleted file mode 100755
index ab83a92..0000000
--- a/solenv/inc/startup/wnt/CVS/Root
+++ /dev/null
@@ -1 +0,0 @@
-:pserver:anoncvs@anoncvs.services.openoffice.org:/cvs
diff --git a/solenv/inc/startup/wnt/CVS/Tag b/solenv/inc/startup/wnt/CVS/Tag
deleted file mode 100755
index 0156927..0000000
--- a/solenv/inc/startup/wnt/CVS/Tag
+++ /dev/null
@@ -1 +0,0 @@
-NOOO300_m3
diff --git a/solenv/inc/startup/wnt/CVS/Template b/solenv/inc/startup/wnt/CVS/Template
deleted file mode 100755
index eec9bab..0000000
--- a/solenv/inc/startup/wnt/CVS/Template
+++ /dev/null
@@ -1,42 +0,0 @@
-Issue number:  
-Submitted by:  
-Reviewed by:   
-CVS: ----------------------------------------------------------------------
-CVS: Issue number:
-CVS:   If this change addresses one or more issues,
-CVS:   then enter the issue number(s) here.
-CVS: Submitted by:
-CVS:   If this code has been contributed to the project by someone else; i.e.,
-CVS:   they sent us a patch or a set of diffs, then include their name/email
-CVS:   address here. If this is your work then delete this line.
-CVS: Reviewed by:
-CVS:   If we are doing pre-commit code reviews and someone else has
-CVS:   reviewed your changes, include their name(s) here.
-CVS:   If you have not had it reviewed then delete this line.
-CVS: ----------------------------------------------------------------------
-CVS: Committers, 
-CVS: 
-CVS: Please follow these protocols: 
-CVS: 
-CVS: * Please include in the log message 
-CVS: reference(s) by ID / number and/or URL 
-CVS: to any and all relevant OpenOffice.org issue(s). 
-CVS: 
-CVS: * If the code is contributed from outside Sun 
-CVS: then please verify using the list at the following URL 
-CVS: http://www.openoffice.org/copyright/copyrightapproved.html 
-CVS: that Sun has received a signed Copyright Assignment Form 
-CVS: from the submitter.
-CVS: 
-CVS: Otherwise, 
-CVS: please send an email TO: the submitter; and CC: OOCRequest@eng.sun.com
-CVS: the letter (CopyRightRequest.txt) to request assignment of copyright to Sun 
-CVS: (http://www.openoffice.org/copyright/assign_copyright.html). 
-CVS: 
-CVS: Please do NOT commit code until you have verified (as detailed above) that 
-CVS: Sun has received a signed Copyright Assignment Form from the submitter.
-CVS: 
-CVS: * Please send an email TO: the submitter 
-CVS: (particularly, if from outside Sun) 
-CVS: advising that the code has been committed, 
-CVS: and gratefully recognizing the contribution. 
diff --git a/solenv/inc/startup/wnt/macros.mk b/solenv/inc/startup/wnt/macros.mk
deleted file mode 100755
index 58b0097..0000000
--- a/solenv/inc/startup/wnt/macros.mk
+++ /dev/null
@@ -1,91 +0,0 @@
-# Define additional MSDOS specific settings.
-#
-
-# --- Set Wrapper command ---
-# Provide a macro that can be used to access the wrapper and to avoid
-# hardcoding the program name everywhere
-GUWCMD*=guw.exe
-
-# This is a no-op for native W32 dmake
-.WINPATH !:= yes
-
-# Directory cache configuration.
-.DIRCACHE  *:= no
-
-# Applicable suffix definitions
-E *:= .exe	# Executables
-
-# Now set the remaining arguments depending on which SHELL we
-# are going to use.
-.IF $(USE_SHELL) == 4nt
-
-.IMPORT : COMSPEC
-
-SHELL *:= $(COMSPEC)
-
-   SHELLFLAGS       *:= $(SWITCHAR)c
-   GROUPFLAGS       *:= $(SHELLFLAGS)
-   SHELLMETAS       *:= "<>|&%^
-# Fix syntax highlighting: "
-   GROUPSUFFIX      *:= .bat
-   DIVFILE          *=  $(TMPFILE:s,/,\,)
-   RM               *=  +del
-   RMFLAGS          *= /y /E
-   MV	            *=  +rename
-   __.DIVSEP-sh-yes *:= \\
-   __.DIVSEP-sh-no  *:= \\
-   DIRSEPSTR        := \\
-
-.ELSE	# Non 4nt case
-
-.IF $(USE_SHELL) == bash
-   SHELLFLAGS       *:= -c
-.ELSE
-   # See iz50689 why -f is needed.
-   SHELLFLAGS       *:= -fc
-.ENDIF # $(USE_SHELL) == bash
-
-   GROUPFLAGS       *:=
-   SHELLMETAS       *:= *";?<>|()&][$$\#`'
-# Fix syntax highlighting: "
-   GROUPSUFFIX      *:= .csh
-# This is hopefully not used. Only in: dmake/msdos/spawn.c  
-#  .MKSARGS         *:= yes
-   DIVFILE          *=  $(TMPFILE:s,/,${__.DIVSEP-sh-${USESHELL}})
-   RM               *=  $/bin$/rm
-   RMFLAGS          *=  -f
-   MV	            *=  $/bin$/mv
-   __.DIVSEP-sh-yes *:= \\\
-   __.DIVSEP-sh-no  *:= \\
-   DIRSEPSTR :=/
-
-.ENDIF
-
-.EXPORT : GUWCMD
-
-.IF $(USE_SHELL) == 4nt
-
-my4ver:=$(shell @echo %_4ver)
-
-.IF "$(my4ver:s/.//:s/,//)" >= "400"
-.ELSE			# "$(my4ver:s/.//:s/,//)" >= "400"
-.ERROR : ; @echo Forced error: 4NT version too old! Version 4.0 or newer is required.
-4NT_version_too_old
-.ENDIF			# "$(my4ver:s/.//:s/,//)" >= "400"
-
-.ELSE # $(USE_SHELL) == 4nt
-
-.IF $(SHELL) == $(NULL)
-# SHELL has to be set in (non-4NT) OOobuild environment
-.ERROR : ; @echo Forced error: Environment variable SHELL has to be set for OOo build!
-SHELL_variable_needed
-.END
-
-.END # $(USE_SHELL) == 4nt
-
-GROUPSHELL *:= $(SHELL)
-
-# Does not respect case of filenames.
-.DIRCACHERESPCASE := no
-NAMEMAX	=	256
-
diff --git a/solenv/inc/unitools.mk b/solenv/inc/unitools.mk
index 9ffeeac..46dd60b 100755
--- a/solenv/inc/unitools.mk
+++ b/solenv/inc/unitools.mk
@@ -39,7 +39,7 @@ XSLTPROC*=xsltproc
 
 ULFCONV*=ulfconv
 
-MAKEDEPEND*=$(SOLARBINDIR)$/makedepend
+MAKEDEPEND*=/local/ooo-build/ooo-build/build/ooo300-m3/solver/300/unxlngx6.pro/bin/makedepend
 
 SCP_CHECK_TOOL:=checkscp$E
 
diff --git a/solenv/inc/wntmsci11.mk b/solenv/inc/wntmsci11.mk
index 5d3bb44..dd8cd6b 100755
--- a/solenv/inc/wntmsci11.mk
+++ b/solenv/inc/wntmsci11.mk
@@ -305,7 +305,7 @@ MFC_INCLUDE*=$(COMPATH)$/atlmfc$/include
 MFC_LIB*=$(COMPATH)$/atlmfc$/lib
 .ENDIF
 
-LIBMGR=lib $(NOLOGO)
+LIBMGR=wine "c:\\Program Files\\Microsoft Visual Studio 9.0\\VC\\bin\\lib.exe" $(NOLOGO)
 IMPLIB=lib
 LIBFLAGS=
 
diff --git a/winenv.set.sh b/winenv.set.sh
index 6ad392c..55163c0 100755
--- a/winenv.set.sh
+++ b/winenv.set.sh
@@ -5,16 +5,16 @@
 # 1. exec sh/bash/ksh 
 # 2. source winenv.set 
 # #################################################################
-SRC_ROOT="c:/ooo-build/ooo-build/build/ooo300-m3"
-SHELL="/usr/bin/bash"
+SRC_ROOT="z:/local/ooo-build/ooo-build-win/build/ooo300-m3"
+SHELL="/bin/bash"
 JAVA_HOME="c:/PROGRA~1/Java/JDK16~1.0_0"
-ANT_HOME="c:/ooo-build/ooo-build/build/apache-ant-1.7.0"
-ANT_LIB="c:/ooo-build/ooo-build/build/apache-ant-1.7.0/lib"
-ANT="c:/ooo-build/ooo-build/build/apache-ant-1.7.0/bin/ant"
+ANT_HOME="z:/local/ooo-build/ooo-build-win/build/apache-ant-1.7.0"
+ANT_LIB="z:/local/ooo-build/ooo-build-win/build/apache-ant-1.7.0/lib"
+ANT="z:/local/ooo-build/ooo-build-win/build/apache-ant-1.7.0/bin/ant"
 JDKLIB="c:/PROGRA~1/Java/JDK16~1.0_0/lib"
 STLPORT4="NO_STLPORT4"
 STLPORT_VER="400"
-ASM_PATH="c:/PROGRA~1/MICROS~1.0/VC/bin"
+ASM_PATH="\"c:\\Program Files\\Microsoft Visual Studio 9.0\\VC\\bin\""
 BUILD_SOSL="true"
 
 
@@ -75,7 +75,7 @@ MKDEPENDSOLVER="TRUE"
 #
 SOLAR_JAVA="TRUE"
 COM="MSC"
-COMPATH="c:/PROGRA~1/MICROS~1.0/VC"
+COMPATH="\"c:\\Program Files\\Microsoft Visual Studio 9.0\\VC\""
 CPU="I"
 CPUNAME="INTEL"
 CVER="M1500"
@@ -97,7 +97,7 @@ FLIPCMD="slfl.pl"
 # Variable values.
 #
 CYGWIN_PATH="C:/cygwin/bin"
-PSDK_HOME="c:/PROGRA~1/MICROS~4/Windows/v6.1"
+PSDK_HOME="\"c:\\Program Files\\Microsoft SDKs\\Windows\\v6.1\""
 WINDOWS_VISTA_PSDK="TRUE"
 DIRECTXSDK_LIB="c:/PROGRA~1/MI22C6~1/lib/x86"
 ATL_LIB="c:/PROGRA~1/MICROS~4/Windows/v6.1/lib"
@@ -108,14 +108,14 @@ NSIS_PATH="c:/PROGRA~1/NSIS"
 DISABLE_ACTIVEX="TRUE"
 DISABLE_ATL="TRUE"
 USE_NEW_SDK="TRUE"
-TMPDIR="C:/cygwin/tmp"
-PERL="C:/cygwin/bin/perl"
+TMPDIR="/tmp"
+PERL="/usr/bin/perl"
 GNUMAKE="make"
 UPD="300"
 WORK_STAMP="OOO300"
-SOLARVER="c:/ooo-build/ooo-build/build/ooo300-m3/solver/300"
+SOLARVER="z:/local/ooo-build/ooo-build-win/build/ooo300-m3/solver/300"
 SOLARVERSION="$SOLARVER"
-SOLARENV="c:/ooo-build/ooo-build/build/ooo300-m3/solenv"
+SOLARENV="/local/ooo-build/ooo-build-win/build/ooo300-m3/solenv"
 SOLARENVINC="$SOLARENV/inc"
 LOCALINI="$SOLARENV/config"
 STAR_INIROOT="$SOLARENV/config"
@@ -133,13 +133,15 @@ JAVAINTERPRETER="c:/PROGRA~1/Java/JDK16~1.0_0/bin/java.exe"
 JAVACOMPILER="c:/PROGRA~1/Java/JDK16~1.0_0/bin/javac.exe"
 JAVADOC="c:/PROGRA~1/Java/JDK16~1.0_0/bin/javadoc.exe"
 SCPDEFS=" -DWITH_ACTIVEX_COMPONENT"
-PATH=".:/cygdrive/c/ooo-build/ooo-build/build/ooo300-m3/solver/300/wntmsci12.pro/bin:/cygdrive/c/ooo-build/ooo-build/build/ooo300-m3/solenv/bin:/cygdrive/c/ooo-build/ooo-build/build/ooo300-m3/solenv/wntmsci12/bin:/cygdrive/c/PROGRA~1/Java/JDK16~1.0_0/bin:/cygdrive/c/PROGRA~1/Java/JDK16~1.0_0/jre/bin/client:/cygdrive/c/PROGRA~1/MICROS~1.0/VC/bin:/cygdrive/c/PROGRA~1/MICROS~1.0/Common7/IDE:/cygdrive/c/PROGRA~1/MICROS~4/Windows/v6.1/Bin:/cygdrive/c/Windows/MICROS~1.NET/FRAMEW~1/v3.5:/cygdrive/c/PROGRA~1/MICROS~1.0/VC/bin:/usr/bin:/cygdrive/c/PROGRA~1/NSIS:/cygdrive/c/PROGRA~1/MICROS~4/Windows/v6.1/bin:/cygdrive/c/ooo-build/ooo-build/build/bin:/usr/local/bin:/bin:/usr/X11R6/bin:/cygdrive/c/PROGRA~1/MI22C6~1/UTILIT~1/Bin/x86:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Program Files/ATI Technologies/ATI.ACE/Core-Static:/cygdrive/c/ooo-build/ooo-build/bin"
+#PATH=".:/cygdrive/c/ooo-build/ooo-build/build/ooo300-m3/solver/300/wntmsci12.pro/bin:/cygdrive/c/ooo-build/ooo-build/build/ooo300-m3/solenv/bin:/cygdrive/c/ooo-build/ooo-build/build/ooo300-m3/solenv/wntmsci12/bin:/cygdrive/c/PROGRA~1/Java/JDK16~1.0_0/bin:/cygdrive/c/PROGRA~1/Java/JDK16~1.0_0/jre/bin/client:/cygdrive/c/PROGRA~1/MICROS~1.0/VC/bin:/cygdrive/c/PROGRA~1/MICROS~1.0/Common7/IDE:/cygdrive/c/PROGRA~1/MICROS~4/Windows/v6.1/Bin:/cygdrive/c/Windows/MICROS~1.NET/FRAMEW~1/v3.5:/cygdrive/c/PROGRA~1/MICROS~1.0/VC/bin:/usr/bin:/cygdrive/c/PROGRA~1/NSIS:/cygdrive/c/PROGRA~1/MICROS~4/Windows/v6.1/bin:/cygdrive/c/ooo-build/ooo-build/build/bin:/usr/local/bin:/bin:/usr/X11R6/bin:/cygdrive/c/PROGRA~1/MI22C6~1/UTILIT~1/Bin/x86:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Program Files/ATI Technologies/ATI.ACE/Core-Static:/cygdrive/c/ooo-build/ooo-build/bin"
 SOLARLIB=" -L$SOLARVER/$INPATH/lib -L../lib -L$SOLARENV/$OUTPATH/lib"
-SOLARINC=" -I. -I$SOLARVER/$INPATH/inc/stl -I$SOLARVER/$INPATH/inc/external -I$SOLARVER/$INPATH/inc -I$SOLARENV/$OUTPATH/inc -I$SOLARENV/inc -I$SRC_ROOT/res -Ic:/ooo-build/ooo-build/build/ooo300-m3/solver/300/wntmsci12.pro/inc/stl -I$JAVA_HOME/include/win32 -I$JAVA_HOME/include -I$PSDK_HOME/include -I$COMPATH/include -Ic:/PROGRA~1/MI22C6~1/include -Ic:/PROGRA~1/MI22C6~1/include"
+SOLARINC=" -I. -I$SOLARVER/$INPATH/inc/stl -I$SOLARVER/$INPATH/inc/external -I$SOLARVER/$INPATH/inc -I$SOLARENV/$OUTPATH/inc -I$SOLARENV/inc -I$SRC_ROOT/res -Iz:/local/ooo-build/ooo-build-win/build/ooo300-m3/solver/300/wntmsci12.pro/inc/stl -I$JAVA_HOME/include/win32 -I$JAVA_HOME/include -I$PSDK_HOME/include -I$COMPATH/include -Ic:/PROGRA~1/MI22C6~1/include -Ic:/PROGRA~1/MI22C6~1/include"
 COMP_ENV="wntmsci12"
 JAVAHOME="$JAVA_HOME"
-CC="c:/PROGRA~1/MICROS~1.0/VC/bin/cl.exe"
-CXX="c:/PROGRA~1/MICROS~1.0/VC/bin/cl.exe"
+CC="wine \"c:\\Program Files\\Microsoft Visual Studio 9.0\\VC\\bin\\cl.exe\""
+CXX="wine \"c:\\Program Files\\Microsoft Visual Studio 9.0\\VC\\bin\\cl.exe\""
+#CC="wine c:/PROGRA~1/MICROS~1.0/VC/bin/cl.exe"
+#CXX="wine c:/PROGRA~1/MICROS~1.0/VC/bin/cl.exe"
 LFS_CFLAGS="-D_FILE_OFFSET_BITS=unknown -D_LARGE_FILES"
 ENABLE_DIRECTX="TRUE"
 ENABLE_LAYOUT="TRUE"
@@ -160,7 +162,7 @@ SYSTEM_STDLIBS="NO"
 SYSTEM_ZLIB="NO"
 SYSTEM_OPENSSL="NO"
 SYSTEM_PYTHON="NO"
-HOME="/home/kendy"
+HOME="/local/home/kendy"
 SYSTEM_ICU="NO"
 SYSTEM_JPEG="NO"
 SYSTEM_POPPLER="NO"
@@ -195,18 +197,18 @@ SYSTEM_MYTHES="NO"
 GXX_INCLUDE_PATH="NO_GXX_INCLUDE"
 COMMON_BUILD_TOOLS="$SRC_ROOT/external/common"
 SOLARSRC="$SRC_ROOT"
-ILIB=".;c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib;c:\\PROGRA~1\\Java\\JDK16~1.0_0\\lib;c:\\PROGRA~1\\MICROS~4\\Windows\\v6.1\\lib;c:\\PROGRA~1\\MICROS~4\\Windows\\v6.1\\lib;c:\\PROGRA~1\\MICROS~1.0\\VC\\lib;c:\\PROGRA~1\\MI22C6~1\\lib\\x86"
-GREP="$CYGWIN_PATH/grep.exe"
-FIND="$CYGWIN_PATH/find.exe"
-LS="$CYGWIN_PATH/ls.exe"
-GNUCOPY="$CYGWIN_PATH/cp.exe"
-TOUCH="$CYGWIN_PATH/touch.exe"
+ILIB=".;z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib;c:\\PROGRA~1\\Java\\JDK16~1.0_0\\lib;c:\\PROGRA~1\\MICROS~4\\Windows\\v6.1\\lib;c:\\PROGRA~1\\MICROS~4\\Windows\\v6.1\\lib;c:\\PROGRA~1\\MICROS~1.0\\VC\\lib;c:\\PROGRA~1\\MI22C6~1\\lib\\x86"
+GREP="/usr/bin/grep"
+FIND="/usr/bin/find"
+LS="/bin/ls"
+GNUCOPY="/bin/cp"
+TOUCH="/usr/bin/touch"
 DELIVER="deliver.pl"
 MKOUT="mkout.pl"
 ZIPDEP="zipdep.pl"
 USE_SHELL="bash"
-PYTHONPATH=".;c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib;c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib\\python;c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib\\python\\lib-dynload;c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib\\pyuno;c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\bin\\pyuno"
-PYTHONHOME="c:\\ooo-build\\ooo-build\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro"
+PYTHONPATH=".;z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib;z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib\\python;z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib\\python\\lib-dynload;z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\lib\\pyuno;z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro\\bin\\pyuno"
+PYTHONHOME="z:\\local\\ooo-build\\ooo-build-win\\build\\ooo300-m3\\solver\\300\\wntmsci12.pro"
 
 #
 # Aliases.
