Index: sfx2/source/toolbox/tbxitem.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/toolbox/tbxitem.cxx,v
retrieving revision 1.30
diff -u -p -u -r1.30 tbxitem.cxx
--- sfx2/source/toolbox/tbxitem.cxx	4 Apr 2003 17:39:32 -0000	1.30
+++ sfx2/source/toolbox/tbxitem.cxx	2 Jun 2003 14:16:35 -0000
@@ -746,19 +746,29 @@ BOOL Impl_ExistURLInMenu( const PopupMen
 
 void SfxAppToolBoxControl_Impl::SetImage( const String &rURL )
 {
-    /* We accept URL's here only, which exist as items of our internal popup menu.
-       All other ones will be ignored and a fallback is used ... */
-    String aURL = rURL;
-    String sFallback;
-    BOOL bValid = Impl_ExistURLInMenu(pMenu,aURL,&sFallback);
-    if (!bValid)
-        aURL = sFallback;
+	/* We accept URL's here only, which exist as items of our internal popup menu.
+	   All other ones will be ignored and a fallback is used ... */
+	String aURL = rURL;
+	String sFallback;
+	BOOL bValid = Impl_ExistURLInMenu(pMenu,aURL,&sFallback);
+	if (!bValid)
+		aURL = sFallback;
 
 	BOOL bBig = ( SfxImageManager::GetCurrentSymbolSet() == SFX_SYMBOLS_LARGE );
-    GetToolBox().SetItemImage( GetId(),
-							   SvFileInformationManager::GetImage( INetURLObject( aURL ),
-							   bBig,
-								GetToolBox().GetBackground().GetColor().IsDark() ) );
+	Image aI = SvFileInformationManager::GetImage( INetURLObject( aURL ), bBig,
+						       GetToolBox().GetBackground().GetColor().IsDark() );
+	Size aBigSize( 24, 24 );
+	if ( bBig && aI.GetSizePixel() != aBigSize )
+	{
+		BitmapEx aScaleBmp( aI.GetBitmap(), aI.GetMaskBitmap() );
+		aScaleBmp.Scale( aBigSize, BMP_SCALE_INTERPOLATE );
+		Image aSizedImage( aScaleBmp );
+
+		GetToolBox().SetItemImage( GetId(), aSizedImage);
+	} 
+	else
+		GetToolBox().SetItemImage( GetId(), aI);
+
 	aLastURL = aURL;
 }
 
