Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.79
diff -u -p -u -r1.79 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	29 Apr 2003 18:26:57 -0000	1.79
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	13 May 2003 12:48:36 -0000
@@ -7133,7 +7133,7 @@
 						<desc xml:lang="en-US">Specifies an additional limit if Font Antialiasing is enabled. Fonts that are smaller than the specified limit are not anti-aliased.</desc>
 						<label xml:lang="en-US">Font Antialiasing Minimum Pixel Height</label>
 					</info>
-					<value>8</value>
+					<value>1</value>
 				</prop>
 			</group>
 		</group>
