Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.79
diff -u -p -u -r1.79 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	29 Apr 2003 18:26:57 -0000	1.79
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	5 Jun 2003 13:14:25 -0000
@@ -9262,11 +9262,16 @@
 					</enumeration>
 					<enumeration oor:value="1">
 						<info>
-							<desc xml:lang="en-US">32x32 pixel icons</desc>
+							<desc xml:lang="en-US">24x24 pixel icons</desc>
+						</info>
+					</enumeration>
+					<enumeration oor:value="2">
+						<info>
+							<desc xml:lang="en-US">automatic icon sizing</desc>
 						</info>
 					</enumeration>
 				</constraints>
-				<value>0</value>
+				<value>1</value>
 			</prop>
 			<prop oor:name="ToolboxStyle" oor:type="xs:short">
 				<!-- UIHints: Tools  Options General View -->
