Index: moz/zipped/makefile.mk
===================================================================
RCS file: /cvs/external/moz/zipped/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- moz/zipped/makefile.mk	14 Aug 2002 16:22:52 -0000	1.7
+++ moz/zipped/makefile.mk	18 Jul 2003 09:01:40 -0000
@@ -71,10 +71,10 @@ TARGET=moz_unzip
 
 # --- Files --------------------------------------------------------
 
-.IF "$(OS)" == "MACOSX"
+.IF "$(OS)" == "MACOSX" || "$(WITH_MOZILLA)" == "YES"
 
 dummy:
-	@echo "Nothing to build for OS $(OS)"
+	@echo "Nothing to build for OS $(OS), With system mozilla: $(WITH_MOZILLA)"
 
 .ELSE	"$(OS)" == "MACOSX"
 
