--- dbaccess/source/ui/dlg/dbwizsetup.cxx.bakj	2005-04-06 18:15:46.000000000 +0530
+++ dbaccess/source/ui/dlg/dbwizsetup.cxx	2005-04-06 18:21:11.000000000 +0530
@@ -268,6 +268,7 @@
 #define THUNDERBIRD_PATH       17
 #define CREATENEW_PATH         18
 #define USERDEFINED_PATH       19
+#define EVOLUTION_PATH_LOCAL   20
 
 OFinalDBPageSetup*          pFinalPage;
 
@@ -411,6 +412,14 @@
     else
         declarePath( THUNDERBIRD_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
 
+if ( m_pCollection->hasAuthentication(DST_EVOLUTION_LOCAL))
+        declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+    else
+        declarePath( EVOLUTION_PATH_LOCAL, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+ if ( m_pCollection->hasAuthentication(DST_EVOLUTION_LDAP))
+        declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+    else
+        declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
     if ( m_pCollection->hasAuthentication(DST_EVOLUTION))
         declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
     else
@@ -595,6 +604,14 @@
                 activatePath( THUNDERBIRD_PATH, sal_True);
 			    break;
 
+            case DST_EVOLUTION_LOCAL:
+                activatePath( EVOLUTION_PATH_LOCAL, sal_True);
+                break;
+
+            case DST_EVOLUTION_LDAP:
+                activatePath( EVOLUTION_PATH, sal_True);
+                break;
+
             case DST_EVOLUTION:
                 activatePath( EVOLUTION_PATH, sal_True);
                 break;
@@ -643,6 +660,8 @@
 	switch ( m_eType )
     {
         case DST_EVOLUTION:
+        case DST_EVOLUTION_LOCAL:
+        case DST_EVOLUTION_LDAP:
         case DST_OUTLOOK:
         case DST_OUTLOOKEXP:
 		case DST_MOZILLA:
--- dbaccess/source/ui/inc/dsntypes.hxx.bakj	2005-04-06 18:05:46.000000000 +0530
+++ dbaccess/source/ui/inc/dsntypes.hxx	2005-04-06 18:07:10.000000000 +0530
@@ -101,6 +101,8 @@
 	DST_EVOLUTION	=	16,
 	DST_EMBEDDED	=	17,
  	DST_THUNDERBIRD	=	18,	
+	DST_EVOLUTION_LOCAL =   19,
+	DST_EVOLUTION_LDAP  =   20,
 
 	DST_USERDEFINE1,	/// first user defined driver
 	DST_USERDEFINE2,
--- dbaccess/source/ui/misc/dsntypes.cxx.bakj	2005-04-06 17:57:59.000000000 +0530
+++ dbaccess/source/ui/misc/dsntypes.cxx	2005-04-06 18:05:24.000000000 +0530
@@ -412,7 +412,9 @@
 			case DST_OUTLOOK:
 			case DST_OUTLOOKEXP:
 			case DST_FLAT:
-            case DST_EVOLUTION:
+			case DST_EVOLUTION_LDAP:
+            		case DST_EVOLUTION:
+			case DST_EVOLUTION_LOCAL:
 	 		case DST_THUNDERBIRD:
 			case DST_CALC:
                 bSupportsTableCreation = FALSE;
@@ -457,7 +459,9 @@
 		case DST_OUTLOOK:
 		case DST_OUTLOOKEXP:
 		case DST_JDBC:
-        case DST_EVOLUTION:
+                case DST_EVOLUTION_LDAP:
+                case DST_EVOLUTION:
+                case DST_EVOLUTION_LOCAL:
 			bEnableBrowseButton = FALSE;
 			break;
 		default:
@@ -491,12 +495,14 @@
 		case DST_ODBC:
 		case DST_ADO:
 		case DST_LDAP:
+           	case DST_EVOLUTION_LDAP:
+                case DST_EVOLUTION:
 			return sal_True;
 			break;
 		case DST_MSACCESS:
 		case DST_MOZILLA:
 		case DST_THUNDERBIRD:
-		case DST_EVOLUTION:
+		case DST_EVOLUTION_LOCAL:
 		case DST_OUTLOOK:
         case DST_OUTLOOKEXP: //????
 		case DST_DBASE:
@@ -574,8 +580,13 @@
 			return DST_OUTLOOK;
 		if (_rDsn.EqualsIgnoreCaseAscii("outlookexp", nSeparator,_rDsn.Len() - nSeparator))
 			return DST_OUTLOOKEXP;
-		if (_rDsn.EqualsIgnoreCaseAscii("evolution", nSeparator,_rDsn.Len() - nSeparator))
-			return DST_EVOLUTION;
+                if (_rDsn.EqualsIgnoreCaseAscii("evolution:ldap", nSeparator,_rDsn.Len() - nSeparator))
+                       	return DST_EVOLUTION_LDAP;
+               if (_rDsn.EqualsIgnoreCaseAscii("evolution:groupwise", nSeparator,_rDsn.Len() - nSeparator))
+                        return DST_EVOLUTION;
+               if (_rDsn.EqualsIgnoreCaseAscii("evolution:local", nSeparator,_rDsn.Len() - nSeparator))
+                       	return DST_EVOLUTION_LOCAL;
+
 	}
 
 	// find third :
--- dbaccess/source/ui/misc/dbumiscres.src.bakj	2005-04-06 18:11:26.000000000 +0530
+++ dbaccess/source/ui/misc/dbumiscres.src	2005-04-06 18:15:20.000000000 +0530
@@ -133,9 +133,17 @@
 	{
 		Text  ="sdbc:address:outlookexp";
 	};
-	String STR_EVOLUTION
-	{
-		Text  ="sdbc:address:evolution";
+        String STR_EVOLUTION_LOCAL
+        {
+               Text  ="sdbc:address:evolution:local";
+        };
+        String STR_EVOLUTION_LDAP
+        {
+                Text  ="sdbc:address:evolution:ldap";
+        };
+        String STR_EVOLUTION_GROUPWISE
+        {
+                Text  ="sdbc:address:evolution:groupwise";
 	};	
 	String STR_EMBEDDED
 	{
@@ -221,11 +229,21 @@
 		Text [ de ] = "Microsoft Windows Adressbuch";
 		Text[ en-US ] = "Microsoft Windows Address Book";
 	};
-	String STR_EVOLUTION
-	{
-		Text [ de ] = "Evolution Adressbuch";
-		Text[ en-US ] = "Evolution Address Book";
-	};
+        String STR_EVOLUTION_LOCAL
+        {
+                Text [ de ] = "Evolution Local Adressbuch";
+                Text[ en-US ] = "Evolution Local";
+        };
+        String STR_EVOLUTION_LDAP
+        {
+               Text [ de ] = "Evolution LDAP Adressbuch";
+               Text[ en-US ] = "Evolution LDAP";
+        };
+        String STR_EVOLUTION_GROUPWISE
+        {
+                Text [ de ] = "Groupwise Adressbuch";
+                Text[ en-US ] = "Groupwise";
+        };
 	String STR_EMBEDDED
 	{
 		Text [ de ] = "Interne Datenbank";
--- dbaccess/source/ui/misc/dbumiscres.hrc.bakj	2005-04-06 18:07:39.000000000 +0530
+++ dbaccess/source/ui/misc/dbumiscres.hrc	2005-04-06 18:11:00.000000000 +0530
@@ -80,11 +80,12 @@
 #define STR_LDAP				13
 #define STR_OUTLOOK				14
 #define STR_OUTLOOKEXP			15
-#define STR_EVOLUTION			16 
+#define STR_EVOLUTION_GROUPWISE			16 
 #define STR_EMBEDDED			17
 #define STR_THUNDERBIRD			18
-
-#define STR_END					STR_THUNDERBIRD + 1
+#define STR_EVOLUTION_LOCAL             19
+#define STR_EVOLUTION_LDAP              20
+#define STR_END					STR_EVOLUTION_LDAP + 1
 
 #endif // _DBU_MISCRES_HRC_
 
