--- extensions/source/abpilot/abspilot.cxx	19 Dec 2005 17:26:24 -0000	1.13
+++ extensions/source/abpilot/abspilot.cxx	16 Feb 2006 12:45:38 -0000
@@ -151,12 +151,9 @@ namespace abp
 		m_pCancel->SetClickHdl( LINK( this, OAddessBookSourcePilot, OnCancelClicked) );
 
 		// some initial settings
-#ifdef WITH_MOZILLA
 #ifdef UNX
-		m_aSettings.eType = AST_MORK;
-#else
-		m_aSettings.eType = AST_OE;
-#endif
+// FIXME: if KDE use KAB instead
+		m_aSettings.eType = AST_EVOLUTION;
 #else
 		m_aSettings.eType = AST_OTHER;
 #endif
