--- connectivity/source/drivers/evoab2/EApi.cxx.orig	2006-02-28 11:32:00.000000000 +0100
+++ connectivity/source/drivers/evoab2/EApi.cxx	2006-09-05 12:03:56.000000000 +0200
@@ -41,9 +41,10 @@
 #include "EApi.h"
 #endif
 static char *eBookLibNames[] = {
-	"libebook.so.8",	 // evolution-2.0
+	"libebook-1.2.so.9", // evolution-2.8
+	"libebook-1.2.so.5", // evolution-2.4 and 2.6+
 	"libebook-1.2.so.3", // evolution-2.2
-	"libebook-1.2.so.5"	 // evolution-2.4 and 2.6+
+	"libebook.so.8"      // evolution-2.0
 };
 
 typedef void (*SymbolFunc) (void);
