--- connectivity/source/drivers/evoab2/NStatement.cxx	Fri Feb  3 17:19:14 2006
+++ connectivity/source/drivers/evoab2/NStatement.cxx.new	Wed Feb  8 14:54:03 2006
@@ -379,19 +379,28 @@
 			else if( (aMatchString.indexOf ( WILDCARD ) == aMatchString.lastIndexOf ( WILDCARD ) ) )
 			{   // One occurance of '%'  matches...
                 if ( aMatchString.indexOf ( WILDCARD ) == 0 )
-					pResult = createTest( aColumnName, E_BOOK_QUERY_BEGINS_WITH, aMatchString.copy( 1 ) );
+					pResult = createTest( aColumnName, E_BOOK_QUERY_ENDS_WITH, aMatchString.copy( 1 ) );
 
                 else if ( aMatchString.indexOf ( WILDCARD ) == aMatchString.getLength() - 1 )
-					pResult = createTest( aColumnName, E_BOOK_QUERY_ENDS_WITH, aMatchString.copy( 0, aMatchString.getLength() - 1 ) );
+					pResult = createTest( aColumnName, E_BOOK_QUERY_BEGINS_WITH, aMatchString.copy( 0, aMatchString.getLength() - 1 ) );
 
 				else
-				{
 					::dbtools::throwGenericSQLException(
-						::rtl::OUString::createFromAscii( "like statement contains too many wildcards" ), NULL );
-				}
+						::rtl::OUString::createFromAscii( "like statement contains wildcard in the middle" ), NULL );
+					
+
 				if( pResult && bNotLike )
 					pResult = e_book_query_not( pResult, TRUE );
             }
+			else if( aMatchString.getLength() >= 3 &&
+					 aMatchString.indexOf ( WILDCARD ) == 0 &&
+					 aMatchString.indexOf ( WILDCARD, 1) == aMatchString.getLength() - 1 ) {
+				// one '%' at the start and another at the end
+				pResult = createTest( aColumnName, E_BOOK_QUERY_CONTAINS, aMatchString.copy (1, aMatchString.getLength() - 2) );
+			}
+			else
+				::dbtools::throwGenericSQLException(
+					::rtl::OUString::createFromAscii( "like statement contains too many wildcards" ), NULL );
 		}
 		else
 				OSL_ASSERT( "Serious internal error" );
