Index: scp/source/office/files.scp
===================================================================
RCS file: /cvs/installation/scp/source/office/Attic/files.scp,v
retrieving revision 1.369.16.14.2.2
diff -u -p -u -r1.369.16.14.2.2 files.scp
--- scp/source/office/files.scp	12 May 2004 15:46:28 -0000	1.369.16.14.2.2
+++ scp/source/office/files.scp	2 Jul 2004 14:33:01 -0000
@@ -804,12 +804,12 @@ End
 
 #ifdef UNX
 
-File GID_FILE_LIB_EVOAB_1
+File GID_FILE_LIB_EVOAB_2
         TXT_FILE_BODY;
         Styles          = (PACKED,UNO_COMPONENT);
         RegistryID      = GID_STARREGISTRY_SERVICES_RDB;
         Dir             = GID_DIR_PROGRAM;
-        Name            = STRING(CONCAT2(libevoab1,UNXSUFFIX));
+        Name            = STRING(CONCAT2(libevoab2,UNXSUFFIX));
 End
 
 #endif
