Index: crashrep/source/unx/makefile.mk
===================================================================
RCS file: /cvs/porting/crashrep/source/unx/makefile.mk,v
retrieving revision 1.7.2.2
diff -u -p -u -r1.7.2.2 makefile.mk
--- crashrep/source/unx/makefile.mk	22 Jul 2003 10:37:38 -0000	1.7.2.2
+++ crashrep/source/unx/makefile.mk	20 Aug 2003 16:53:01 -0000
@@ -86,7 +86,7 @@ dummy:
 # Only build crash reporter if either a product build with debug info 
 # or a non-pro build is done.
 
-.IF "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" != "FULL"
+.IF "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" != "full"
 
 CFLAGS+=`pkg-config --cflags gtk+-2.0`
 
@@ -113,7 +113,7 @@ APP1STDLIBS=`pkg-config --libs gtk+-2.0`
 
 # Build statically linked version for special builds and non-pros
 
-.IF "$(ENABLE_CRASHDUMP)" == "STATIC" || "$(PRODUCT)" != "FULL"
+.IF "$(ENABLE_CRASHDUMP)" == "STATIC" || "$(PRODUCT)" != "full"
 APP2NOSAL=TRUE
 APP2TARGET=$(TARGET2)
 APP2OBJS=$(OBJFILES)
@@ -131,7 +131,7 @@ APP2STDLIBS=$(STATIC) `pkg-config --libs
 
 ALL: ALLTAR $(BIN)$/crash_dump.res.01
 
-.ENDIF #  "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" != "FULL"
+.ENDIF #  "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" != "full"
 
 .ENDIF #  "$(OS)"=="MACOSX"
 
Index: scp/util/makefile.mk
===================================================================
RCS file: /cvs/installation/scp/util/makefile.mk,v
retrieving revision 1.24
diff -u -p -u -r1.24 makefile.mk
--- scp/util/makefile.mk	15 Jul 2003 14:36:52 -0000	1.24
+++ scp/util/makefile.mk	2 Sep 2003 14:45:55 -0000
@@ -152,13 +152,13 @@ SCP1FILES += \
 
 .ENDIF
 
-.IF "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" != "FULL" 
+.IF "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" != "full"
 SCP1FILES += \
         crashrep_dynamic.par
 .ENDIF
 
 .IF "$(GUI)"=="UNX"
-.IF "$(ENABLE_CRASHDUMP)" == "STATIC" || "$(PRODUCT)" != "FULL"
+.IF "$(ENABLE_CRASHDUMP)" == "STATIC" || "$(PRODUCT)" != "full"
 SCP1FILES += \
         crashrep_static.par
 .ENDIF
