--- xmlhelp/source/cxxhelp/provider/provider.cxx	2004-11-01 14:23:44.848784824 +0100
+++ xmlhelp/source/cxxhelp/provider/provider.cxx	2004-11-02 12:53:57.137045016 +0100
@@ -108,6 +108,10 @@
 #ifndef _COM_SUN_STAR_CONTAINER_XNAMEREPLACE_HPP_
 #include <com/sun/star/container/XNameReplace.hpp>
 #endif
+#ifndef _UNO_CURRENT_CONTEXT_HXX_
+#include <uno/current_context.hxx>
+#endif
+
 
 using namespace com::sun::star::frame;
 using namespace com::sun::star::beans;
@@ -415,10 +419,42 @@ void ContentProvider::init()
     if(!found) {
         imageZip = getKey(xHierAccess,"Path/Current/Config");
         subst(imageZip);
-        if(1+imageZip.lastIndexOf('/') == imageZip.getLength())
-            imageZip += rtl::OUString::createFromAscii("images.zip");
-        else
-            imageZip += rtl::OUString::createFromAscii("/images.zip");        
+        if(1+imageZip.lastIndexOf('/') != imageZip.getLength())
+            imageZip += rtl::OUString::createFromAscii("/");
+
+        rtl::OUString aZipName;
+        try
+        {
+            Reference< XCurrentContext > xCurrentContext( getCurrentContext() );
+
+            if ( xCurrentContext.is() )
+            {
+                Any aValue = xCurrentContext->getValueByName(
+                        rtl::OUString::createFromAscii( "system.desktop-environment" ) );
+
+                rtl::OUString aDesktopEnvironment;
+                if ( aValue >>= aDesktopEnvironment )
+                {
+                    if ( aDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) )
+                        aZipName = rtl::OUString::createFromAscii( "images_gnome.zip" );
+                    else if ( aDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
+                        aZipName = rtl::OUString::createFromAscii( "images_kde.zip" );
+
+                    osl::DirectoryItem aDirItem;
+                    if ( aZipName.getLength() > 0 &&
+                         osl::DirectoryItem::get( imageZip + aZipName, aDirItem ) == osl::FileBase::E_None )
+                    {
+                        found = true;
+                    }
+                }
+            }
+        }
+        catch( RuntimeException ) {}
+
+        if ( !found || aZipName.getLength() == 0 )
+            aZipName = rtl::OUString::createFromAscii( "images.zip" );
+
+        imageZip += aZipName;
     }
 	
     m_pDatabases = new Databases( instPath,
