Index: scp/source/office/files.scp
===================================================================
RCS file: /cvs/installation/scp/source/office/Attic/files.scp,v
retrieving revision 1.369.16.14.2.2
diff -u -p -u -r1.369.16.14.2.2 files.scp
--- scp/source/office/files.scp	12 May 2004 15:46:28 -0000	1.369.16.14.2.2
+++ scp/source/office/files.scp	9 Jul 2004 13:10:11 -0000
@@ -2731,6 +2711,15 @@ End
 // ************************************************************************************
 // ************************************************************************************
 
+#ifdef UNX
+File GID_FILE_LIB_FPS_GNOME
+        TXT_FILE_BODY;
+        Styles          = (PACKED,UNO_COMPONENT);
+        RegistryID      = GID_STARREGISTRY_SERVICES_RDB;
+        Dir             = GID_DIR_PROGRAM;
+        Name            = "fps_gnome.dll";
+End
+#endif
 
 // ************************************************************************************
 // ************************************************************************************
