Index: framework/source/services/logindialog.cxx
===================================================================
RCS file: /cvs/framework/framework/source/services/logindialog.cxx,v
retrieving revision 1.11
retrieving revision 1.12
diff -u -p -u -r1.11 -r1.12
--- framework/source/services/logindialog.cxx	12 Oct 2001 05:59:01 -0000	1.11
+++ framework/source/services/logindialog.cxx	20 Nov 2003 17:08:00 -0000	1.12
@@ -114,6 +114,7 @@
 #ifndef _SV_SYSDATA_HXX
     #if defined( WIN ) || defined( WNT ) || defined( OS2 )
         typedef sal_Int32 HWND;
+        typedef sal_Int32 HMENU;
     #endif
     #include <vcl/sysdata.hxx>
 #endif
