Index: sj2/source/jscpp/sjapplet_impl.cxx
===================================================================
RCS file: /cvs/oi/sj2/source/jscpp/sjapplet_impl.cxx,v
retrieving revision 1.15.18.2
retrieving revision 1.19
diff -u -p -u -r1.15.18.2 -r1.19
--- sj2/source/jscpp/sjapplet_impl.cxx	3 May 2004 11:20:01 -0000	1.15.18.2
+++ sj2/source/jscpp/sjapplet_impl.cxx	1 Jun 2004 09:13:40 -0000	1.19
@@ -62,6 +62,7 @@
 #include "sjapplet_impl.hxx"
 
 #define HWND int
+#define HMENU int
 
 #include <rtl/ustring>
 #include <rtl/process.h>
