Index: bmp.cxx
===================================================================
RCS file: /cvs/util/svtools/bmpmaker/bmp.cxx,v
retrieving revision 1.11
retrieving revision 1.14
diff -u -p -u -r1.11 -r1.14
--- bmp.cxx	24 Apr 2003 13:01:36 -0000	1.11
+++ bmp.cxx	19 Mar 2004 16:33:01 -0000	1.14
@@ -258,11 +255,20 @@ int BmpApp::Start( const ::std::vector< 
 
 int main( int nArgCount, char* ppArgs[] )
 {
+#ifdef UNX
+    static char aDisplayVar[ 1024 ];
+
+    strcpy( aDisplayVar, "DISPLAY=" );
+    putenv( aDisplayVar );
+#endif
+    
     ::std::vector< String > aArgs;
     BmpApp                  aBmpApp;
 
+    InitVCL( com::sun::star::uno::Reference< com::sun::star::lang::XMultiServiceFactory >() );
+
     for( int i = 1; i < nArgCount; i++ )
         aArgs.push_back( String( ppArgs[ i ], RTL_TEXTENCODING_ASCII_US ) );
     
     return aBmpApp.Start( aArgs );
 }

