Index: vcl/win/inc/salbmp.h
===================================================================
RCS file: vcl/win/inc/salbmp.h
diff -N vcl/win/inc/salbmp.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/win/inc/salbmp.h	18 Nov 2003 14:48:14 -0000	1.2
@@ -0,0 +1,132 @@
+/*************************************************************************
+ *
+ *  
+ *
+ *  
+ *
+ *  
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_SALBMP_H
+#define _SV_SALBMP_H
+
+#ifndef _SV_WINCOMP_HXX
+#include <wincomp.hxx>
+#endif
+
+#ifndef _GEN_HXX
+#include <tools/gen.hxx>
+#endif
+
+#ifndef _SV_SV_H
+#include <sv.h>
+#endif
+
+#ifndef _SV_SALBMP_HXX
+#include <salbmp.hxx>
+#endif
+
+// --------------
+// - SalBitmap	-
+// --------------
+
+struct	BitmapBuffer;
+class	BitmapColor;
+class	BitmapPalette;
+class	SalGraphics;
+
+class WinSalBitmap : public SalBitmap
+{
+private:
+
+	Size				maSize;
+	HGLOBAL 			mhDIB;
+	HBITMAP 			mhDDB;
+	USHORT				mnBitCount;
+
+public:
+
+	HGLOBAL 			ImplGethDIB() const { return mhDIB; }
+	HBITMAP 			ImplGethDDB() const { return mhDDB; }
+
+	static HGLOBAL		ImplCreateDIB( const Size& rSize, USHORT nBitCount, const BitmapPalette& rPal );
+	static HANDLE		ImplCopyDIBOrDDB( HANDLE hHdl, bool bDIB );
+	static USHORT		ImplGetDIBColorCount( HGLOBAL hDIB );
+	static void 		ImplDecodeRLEBuffer( const BYTE* pSrcBuf, BYTE* pDstBuf,
+											 const Size& rSizePixel, bool bRLE4 );
+
+public:
+
+						WinSalBitmap();
+	virtual				~WinSalBitmap();
+
+public:
+
+	bool                        Create( HANDLE hBitmap, bool bDIB, bool bCopyHandle );
+	virtual bool                Create( const Size& rSize, USHORT nBitCount, const BitmapPalette& rPal );
+	virtual bool                Create( const SalBitmap& rSalBmpImpl );
+	virtual bool                Create( const SalBitmap& rSalBmpImpl, SalGraphics* pGraphics );
+	virtual bool                Create( const SalBitmap& rSalBmpImpl, USHORT nNewBitCount );
+
+	virtual void                Destroy();
+
+	virtual Size                GetSize() const { return maSize; }
+	virtual USHORT              GetBitCount() const { return mnBitCount; }
+
+	virtual BitmapBuffer*		AcquireBuffer( bool bReadOnly );
+	virtual void                ReleaseBuffer( BitmapBuffer* pBuffer, bool bReadOnly );
+};
+
+#endif // _SV_SALBMP_HXX
Index: vcl/win/inc/saldata.hxx
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/saldata.hxx,v
retrieving revision 1.11
retrieving revision 1.14
diff -u -p -u -r1.11 -r1.14
--- vcl/win/inc/saldata.hxx	28 May 2003 12:34:54 -0000	1.11
+++ vcl/win/inc/saldata.hxx	10 May 2004 16:01:23 -0000	1.14
@@ -75,25 +75,12 @@
 #include <wincomp.hxx>
 #endif
 
-// as long as we're using the old skd header files, define some of the new XP constants here
-
-#ifndef SPI_GETFLATMENU
-#define SPI_GETFLATMENU     0x1022
-#endif
-#ifndef COLOR_MENUBAR
-#define COLOR_MENUBAR       30
-#endif
-#ifndef COLOR_MENUHILIGHT
-#define COLOR_MENUHILIGHT   29   
-#endif
-
-
 class AutoTimer;
-class SalInstance;
-class SalObject;
-class SalFrame;
-class SalVirtualDevice;
-class SalPrinter;
+class WinSalInstance;
+class WinSalObject;
+class WinSalFrame;
+class WinSalVirtualDevice;
+class WinSalPrinter;
 class Font;
 struct HDCCache;
 struct TempFontItem;
@@ -122,13 +109,20 @@ struct SalIcon
 // - SalData -
 // -----------
 
-struct SalData
+class SalData
 {
+public:
+    SalData();
+    ~SalData();
+
+    // native widget framework
+    void    initNWF();
+    void    deInitNWF();
+
+public:
     HINSTANCE               mhInst;                 // default instance handle
     HINSTANCE               mhPrevInst;             // previous instance handle
     int                     mnCmdShow;              // default frame show style
-    // Erst hier koennen Daten kompatible eingefuegt werden, da die
-    // oberen Daten in salmain.cxx modifiziert werden
     HPALETTE                mhDitherPal;            // dither palette
     HGLOBAL                 mhDitherDIB;            // dither memory handle
     BYTE*                   mpDitherDIB;            // dither memory
@@ -139,16 +133,15 @@ struct SalData
     ULONG                   mnTimerMS;              // Current Time (in MS) of the Timer
     ULONG                   mnTimerOrgMS;           // Current Original Time (in MS)
     UINT                    mnTimerId;              // windows timer id
-    SALTIMERPROC            mpTimerProc;            // timer callback proc
     BOOL                    mbInTimerProc;          // timer event is currently being dispatched
     HHOOK                   mhSalObjMsgHook;        // hook to get interesting msg for SalObject
     HWND                    mhWantLeaveMsg;         // window handle, that want a MOUSELEAVE message
     AutoTimer*              mpMouseLeaveTimer;      // Timer for MouseLeave Test
-    SalInstance*            mpFirstInstance;        // pointer of first instance
-    SalFrame*               mpFirstFrame;           // pointer of first frame
-    SalObject*              mpFirstObject;          // pointer of first object window
-    SalVirtualDevice*       mpFirstVD;              // first VirDev
-    SalPrinter*             mpFirstPrinter;         // first printing printer
+    WinSalInstance*         mpFirstInstance;        // pointer of first instance
+    WinSalFrame*            mpFirstFrame;           // pointer of first frame
+    WinSalObject*           mpFirstObject;          // pointer of first object window
+    WinSalVirtualDevice*    mpFirstVD;              // first VirDev
+    WinSalPrinter*          mpFirstPrinter;         // first printing printer
     HDCCache*               mpHDCCache;             // Cache for three DC's
     HBITMAP                 mh50Bmp;                // 50% Bitmap
     HBRUSH                  mh50Brush;              // 50% Brush
@@ -169,6 +162,8 @@ struct SalData
     SysAgt_Enable_PROC      mpSageEnableProc;       // funktion to deactivate the system agent
     SalIcon*                mpFirstIcon;            // icon cache, points to first icon, NULL if none
     TempFontItem*           mpTempFontItem;
+    BOOL                    mbThemeChanged;         // true if visual theme was changed: throw away theme handles
+
 };
 
 inline void SetSalData( SalData* pData ) { ImplGetSVData()->mpSalData = (void*)pData; }
@@ -266,7 +261,7 @@ long ImplHandleSalObjSysCharMsg( HWND hW
 BOOL ImplHandleGlobalMsg( HWND hWnd, UINT nMsg, WPARAM wParam, LPARAM lParam, LRESULT& rlResult );
 
 // \WIN\SOURCE\WINDOW\SALOBJ.CXX
-SalObject* ImplFindSalObject( HWND hWndChild );
+WinSalObject* ImplFindSalObject( HWND hWndChild );
 BOOL ImplSalPreDispatchMsg( MSG* pMsg );
 void ImplSalPostDispatchMsg( MSG* pMsg, LRESULT nDispatchResult );
 
@@ -383,24 +378,24 @@ WIN_BOOL    ImplGetMessage( LPMSG lpMsg,
 WIN_BOOL    ImplPeekMessage( LPMSG lpMsg, HWND hWnd, UINT wMsgFilterMin, UINT wMsgFilterMax, UINT wRemoveMsg );
 LONG        ImplDispatchMessage( CONST MSG *lpMsg );
 
-inline void SetWindowPtr( HWND hWnd, SalFrame* pThis )
+inline void SetWindowPtr( HWND hWnd, WinSalFrame* pThis )
 {
     ImplSetWindowLong( hWnd, SAL_FRAME_THIS, (LONG)pThis );
 }
 
-inline SalFrame* GetWindowPtr( HWND hWnd )
+inline WinSalFrame* GetWindowPtr( HWND hWnd )
 {
-    return (SalFrame*)ImplGetWindowLong( hWnd, SAL_FRAME_THIS );
+    return (WinSalFrame*)ImplGetWindowLong( hWnd, SAL_FRAME_THIS );
 }
 
-inline void SetSalObjWindowPtr( HWND hWnd, SalObject* pThis )
+inline void SetSalObjWindowPtr( HWND hWnd, WinSalObject* pThis )
 {
     ImplSetWindowLong( hWnd, SAL_OBJECT_THIS, (LONG)pThis );
 }
 
-inline SalObject* GetSalObjWindowPtr( HWND hWnd )
+inline WinSalObject* GetSalObjWindowPtr( HWND hWnd )
 {
-    return (SalObject*)ImplGetWindowLong( hWnd, SAL_OBJECT_THIS );
+    return (WinSalObject*)ImplGetWindowLong( hWnd, SAL_OBJECT_THIS );
 }
 
 #endif  // _SV_SALDATA_HXX
Index: vcl/win/inc/salframe.h
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/salframe.h,v
retrieving revision 1.7
retrieving revision 1.9
diff -u -p -u -r1.7 -r1.9
--- vcl/win/inc/salframe.h	19 Mar 2002 17:09:35 -0000	1.7
+++ vcl/win/inc/salframe.h	18 Nov 2003 16:03:05 -0000	1.9
@@ -70,21 +70,27 @@
 #include <sysdata.hxx>
 #endif
 
+#ifndef _SV_SALFRAME_HXX
+#include <salframe.hxx>
+#endif
+
+class WinSalGraphics;
+
 // ----------------
-// - SalFrameData -
+// - WinSalFrame -
 // ----------------
 
-class SalFrameData
+class WinSalFrame : public SalFrame
 {
 public:
     HWND                    mhWnd;                  // Window handle
     HCURSOR                 mhCursor;               // cursor handle
     HIMC                    mhDefIMEContext;        // default IME-Context
-    SalGraphics*            mpGraphics;             // current frame graphics
-    SalGraphics*            mpGraphics2;            // current frame graphics for other threads
-    SalFrame*               mpNextFrame;            // pointer to next frame
-    void*                   mpInst;                 // instance handle for callback
-    SALFRAMEPROC            mpProc;                 // callback proc
+    WinSalGraphics*         mpGraphics;             // current frame graphics
+    WinSalGraphics*         mpGraphics2;            // current frame graphics for other threads
+    WinSalFrame*            mpNextFrame;            // pointer to next frame
+    HMENU                   mSelectedhMenu;         // the menu where highlighting is currently going on
+    HMENU                   mLastActivatedhMenu;    // the menu that was most recently opened
     SystemEnvData           maSysData;              // system data
     SalFrameState           maState;                // frame state
     int                     mnShowState;            // show state
@@ -118,7 +124,51 @@ public:
     BOOL                    mbSpezIME;              // TRUE: Spez IME
     BOOL                    mbAtCursorIME;          // TRUE: Wir behandeln nur einige IME-Messages
     BOOL                    mbCandidateMode;        // TRUE: Wir befinden uns im Candidate-Modus
-    BOOL                    mbInReparent;           // TRUE: ignore focus lost and gain due to reparenting
+    static BOOL             mbInReparent;           // TRUE: ignore focus lost and gain due to reparenting
+
+public:
+    WinSalFrame();
+    virtual ~WinSalFrame();
+
+    virtual SalGraphics*		GetGraphics();
+    virtual void				ReleaseGraphics( SalGraphics* pGraphics );
+    virtual BOOL				PostEvent( void* pData );
+    virtual void				SetTitle( const XubString& rTitle );
+    virtual void				SetIcon( USHORT nIcon );
+    virtual void                                SetMenu( SalMenu* pSalMenu );
+    virtual void                                DrawMenuBar();
+    virtual void				Show( BOOL bVisible, BOOL bNoActivate = FALSE );
+    virtual void				Enable( BOOL bEnable );
+    virtual void              SetMinClientSize( long nWidth, long nHeight );
+    virtual void				SetPosSize( long nX, long nY, long nWidth, long nHeight, USHORT nFlags );
+    virtual void				GetClientSize( long& rWidth, long& rHeight );
+    virtual void				GetWorkArea( Rectangle& rRect );
+    virtual SalFrame*			GetParent() const;
+    virtual void				SetWindowState( const SalFrameState* pState );
+    virtual BOOL				GetWindowState( SalFrameState* pState );
+    virtual void				ShowFullScreen( BOOL bFullScreen );
+    virtual void				StartPresentation( BOOL bStart );
+    virtual void				SetAlwaysOnTop( BOOL bOnTop );
+    virtual void				ToTop( USHORT nFlags );
+    virtual void				SetPointer( PointerStyle ePointerStyle );
+    virtual void				CaptureMouse( BOOL bMouse );
+    virtual void				SetPointerPos( long nX, long nY );
+    virtual void				Flush();
+    virtual void				Sync();
+    virtual void				SetInputContext( SalInputContext* pContext );
+    virtual void				EndExtTextInput( USHORT nFlags );
+    virtual String				GetKeyName( USHORT nKeyCode );
+    virtual String				GetSymbolKeyName( const XubString& rFontName, USHORT nKeyCode );
+    virtual LanguageType		GetInputLanguage();
+    virtual SalBitmap*			SnapShot();
+    virtual void				UpdateSettings( AllSettings& rSettings );
+    virtual void				Beep( SoundType eSoundType );
+    virtual const SystemEnvData*	GetSystemData() const;
+    virtual ULONG				GetCurrentModButtons();
+    virtual void				SetParent( SalFrame* pNewParent );
+    virtual bool				SetPluginParent( SystemParentData* pNewParent );
 };
+
+void ImplSalGetWorkArea( HWND hWnd, RECT *pRect, const RECT *pParentRect );
 
 #endif // _SV_SALFRAME_H
Index: vcl/win/inc/salgdi.h
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/salgdi.h,v
retrieving revision 1.8
retrieving revision 1.12
diff -u -p -u -r1.8 -r1.12
--- vcl/win/inc/salgdi.h	28 May 2003 12:35:06 -0000	1.8
+++ vcl/win/inc/salgdi.h	10 May 2004 16:01:37 -0000	1.12
@@ -68,6 +68,9 @@
 #ifndef _SV_SALLAYOUT_HXX
 #include <sallayout.hxx>
 #endif
+#ifndef _SV_SALGDI_HXX
+#include <salgdi.hxx>
+#endif
 
 #include "boost/scoped_ptr.hpp"
 
@@ -100,7 +103,7 @@ public:
 // - SalGraphicsData -
 // -------------------
 
-class SalGraphicsData
+class WinSalGraphics : public SalGraphics
 {
 public:
 	HDC 					mhDC;				// HDC
@@ -141,11 +144,177 @@ public:
     // SalGraphics::GetTextLayout, to cache data derived from mhDC's font;
     // flushed whenever a new font is selected into mhDC:
     boost::scoped_ptr< ImplTextLayoutCache > mxTextLayoutCache;
+public:
+    WinSalGraphics();
+    virtual ~WinSalGraphics();
+
+protected:
+    virtual BOOL		unionClipRegion( long nX, long nY, long nWidth, long nHeight );
+    // draw --> LineColor and FillColor and RasterOp and ClipRegion
+    virtual void		drawPixel( long nX, long nY );
+    virtual void		drawPixel( long nX, long nY, SalColor nSalColor );
+    virtual void		drawLine( long nX1, long nY1, long nX2, long nY2 );
+    virtual void		drawRect( long nX, long nY, long nWidth, long nHeight );
+    virtual void		drawPolyLine( ULONG nPoints, const SalPoint* pPtAry );
+    virtual void		drawPolygon( ULONG nPoints, const SalPoint* pPtAry );
+    virtual void		drawPolyPolygon( ULONG nPoly, const ULONG* pPoints, PCONSTSALPOINT* pPtAry );
+    virtual sal_Bool	drawPolyLineBezier( ULONG nPoints, const SalPoint* pPtAry, const BYTE* pFlgAry );
+    virtual sal_Bool	drawPolygonBezier( ULONG nPoints, const SalPoint* pPtAry, const BYTE* pFlgAry );
+    virtual sal_Bool	drawPolyPolygonBezier( ULONG nPoly, const ULONG* pPoints, const SalPoint* const* pPtAry, const BYTE* const* pFlgAry );
+
+    // CopyArea --> No RasterOp, but ClipRegion
+    virtual void		copyArea( long nDestX, long nDestY, long nSrcX, long nSrcY, long nSrcWidth,
+                                  long nSrcHeight, USHORT nFlags );
+    
+    // CopyBits and DrawBitmap --> RasterOp and ClipRegion
+    // CopyBits() --> pSrcGraphics == NULL, then CopyBits on same Graphics
+    virtual void		copyBits( const SalTwoRect* pPosAry, SalGraphics* pSrcGraphics );
+    virtual void		drawBitmap( const SalTwoRect* pPosAry, const SalBitmap& rSalBitmap );
+    virtual void		drawBitmap( const SalTwoRect* pPosAry,
+                                    const SalBitmap& rSalBitmap,
+                                    SalColor nTransparentColor );
+    virtual void		drawBitmap( const SalTwoRect* pPosAry,
+                                    const SalBitmap& rSalBitmap,
+                                    const SalBitmap& rTransparentBitmap );
+    virtual void		drawMask( const SalTwoRect* pPosAry,
+                                  const SalBitmap& rSalBitmap,
+                                  SalColor nMaskColor );
+    
+    virtual SalBitmap*	getBitmap( long nX, long nY, long nWidth, long nHeight );
+    virtual SalColor	getPixel( long nX, long nY );
+
+    // invert --> ClipRegion (only Windows or VirDevs)
+    virtual void		invert( long nX, long nY, long nWidth, long nHeight, SalInvert nFlags);
+    virtual void		invert( ULONG nPoints, const SalPoint* pPtAry, SalInvert nFlags );
+
+    virtual BOOL		drawEPS( long nX, long nY, long nWidth, long nHeight, void* pPtr, ULONG nSize );
+
+    // native widget rendering methods that require mirroring
+    virtual BOOL        hitTestNativeControl( ControlType nType, ControlPart nPart, const Region& rControlRegion,
+                                              const Point& aPos, SalControlHandle& rControlHandle, BOOL& rIsInside );
+    virtual BOOL        drawNativeControl( ControlType nType, ControlPart nPart, const Region& rControlRegion,
+                                           ControlState nState, const ImplControlValue& aValue, SalControlHandle& rControlHandle,
+                                           rtl::OUString aCaption );
+    virtual BOOL        drawNativeControlText( ControlType nType, ControlPart nPart, const Region& rControlRegion,
+                                               ControlState nState, const ImplControlValue& aValue,
+                                               SalControlHandle& rControlHandle, rtl::OUString aCaption );
+    virtual BOOL        getNativeControlRegion( ControlType nType, ControlPart nPart, const Region& rControlRegion, ControlState nState,
+                                                const ImplControlValue& aValue, SalControlHandle& rControlHandle, rtl::OUString aCaption,
+                                                Region &rNativeBoundingRegion, Region &rNativeContentRegion );
+
+public:
+    // public SalGraphics methods, the interface to teh independent vcl part
+
+    // get device resolution
+    virtual void			GetResolution( long& rDPIX, long& rDPIY );
+    // get resolution for fonts (an implementations specific adjustment,
+    // ideally would be the same as the Resolution)
+    virtual void			GetScreenFontResolution( long& rDPIX, long& rDPIY );
+    // get the depth of the device
+    virtual USHORT			GetBitCount();
+    // get the width of the device
+    virtual long			GetGraphicsWidth();
+
+    // set the clip region to empty
+    virtual void			ResetClipRegion();
+    // begin setting the clip region, add rectangles to the
+    // region with the UnionClipRegion call
+    virtual void			BeginSetClipRegion( ULONG nCount );
+    // all rectangles were added and the clip region should be set now
+    virtual void			EndSetClipRegion();
+
+    // set the line color to transparent (= don't draw lines)
+    virtual void			SetLineColor();
+    // set the line color to a specific color
+    virtual void			SetLineColor( SalColor nSalColor );
+    // set the fill color to transparent (= don't fill)
+    virtual void			SetFillColor();
+    // set the fill color to a specific color, shapes will be
+    // filled accordingly
+    virtual void          	SetFillColor( SalColor nSalColor );
+    // enable/disable XOR drawing
+    virtual void			SetXORMode( BOOL bSet );
+    // set line color for raster operations
+    virtual void			SetROPLineColor( SalROPColor nROPColor );
+    // set fill color for raster operations
+    virtual void			SetROPFillColor( SalROPColor nROPColor );
+    // set the text color to a specific color
+    virtual void			SetTextColor( SalColor nSalColor );
+    // set the font
+    virtual USHORT         SetFont( ImplFontSelectData*, int nFallbackLevel );
+    // get the current font's etrics
+    virtual void			GetFontMetric( ImplFontMetricData* );
+    // get kernign pairs of the current font
+    // return only PairCount if (pKernPairs == NULL)
+    virtual ULONG			GetKernPairs( ULONG nPairs, ImplKernPairData* pKernPairs );
+    // get the repertoire of the current font; the code pairs returned
+    // contain unicode ranges. if pCodePairs is NULL return only the
+    // number of pairs which would be filled
+    virtual ULONG			GetFontCodeRanges( sal_uInt32* pCodePairs ) const;
+    // graphics must fill supplied font list
+    virtual void			GetDevFontList( ImplDevFontList* );
+    // graphics should call ImplAddDevFontSubstitute on supplied
+    // OutputDevice for all its device specific preferred font substitutions
+    virtual void			GetDevFontSubstList( OutputDevice* );
+    virtual ImplFontData*	AddTempDevFont( const String& rFileURL, const String& rFontName );
+    // CreateFontSubset: a method to get a subset of glyhps of a font
+    // inside a new valid font file
+    // returns TRUE if creation of subset was successfull
+    // parameters: rToFile: contains a osl file URL to write the subset to
+    //             pFont: describes from which font to create a subset
+    //             pGlyphIDs: the glyph ids to be extracted
+    //             pEncoding: the character code corresponding to each glyph
+    //             pWidths: the advance widths of the correspoding glyphs (in PS font units)
+    //             nGlyphs: the number of glyphs
+    //             rInfo: additional outgoing information
+    // implementation note: encoding 0 with glyph id 0 should be added implicitly
+    // as "undefined character"
+    virtual BOOL			CreateFontSubset( const rtl::OUString& rToFile,
+                                              ImplFontData* pFont,
+                                              long* pGlyphIDs,
+                                              sal_uInt8* pEncoding,
+                                              sal_Int32* pWidths,
+                                              int nGlyphs,
+                                              FontSubsetInfo& rInfo // out parameter
+                                              );
+
+    // GetFontEncodingVector: a method to get the encoding map Unicode
+	// to font encoded character; this is only used for type1 fonts and
+    // may return NULL in case of unknown encoding vector
+    // if ppNonEncoded is set and non encoded characters (that is type1
+    // glyphs with only a name) exist it is set to the corresponding
+    // map for non encoded glyphs; the encoding vector contains -1
+    // as encoding for these cases
+    virtual const std::map< sal_Unicode, sal_Int32 >* GetFontEncodingVector( ImplFontData* pFont, const std::map< sal_Unicode, rtl::OString >** ppNonEncoded );
+
+    // GetEmbedFontData: gets the font data for a font marked
+    // embeddable by GetDevFontList or NULL in case of error
+    // parameters: pFont: describes the font in question
+    //             pWidths: the widths of all glyphs from char code 0 to 255
+    //                      pWidths MUST support at least 256 members;
+    //             rInfo: additional outgoing information
+    //             pDataLen: out parameter, contains the byte length of the returned buffer
+    virtual const void*	GetEmbedFontData( ImplFontData* pFont,
+                                          const sal_Unicode* pUnicodes,
+                                          sal_Int32* pWidths,
+                                          FontSubsetInfo& rInfo,
+                                          long* pDataLen );
+    // frees the font data again
+    virtual void			FreeEmbedFontData( const void* pData, long nDataLen );
+
+    virtual BOOL                    GetGlyphBoundRect( long nIndex, Rectangle& );
+    virtual BOOL                    GetGlyphOutline( long nIndex, PolyPolygon& );
+
+    virtual SalLayout*              GetTextLayout( ImplLayoutArgs&, int nFallbackLevel );
+    virtual void					 DrawServerFontLayout( const ServerFontLayout& );
+
+    // Query the platform layer for control support
+    virtual BOOL IsNativeControlSupported( ControlType nType, ControlPart nPart );
 };
 
 // Init/Deinit Graphics
-void	ImplSalInitGraphics( SalGraphicsData* mpData );
-void	ImplSalDeInitGraphics( SalGraphicsData* mpData );
+void	ImplSalInitGraphics( WinSalGraphics* mpData );
+void	ImplSalDeInitGraphics( WinSalGraphics* mpData );
 void	ImplUpdateSysColorEntries();
 int 	ImplIsSysColorEntry( SalColor nSalColor );
 void	ImplGetLogFontFromFontSelect( HDC hDC,
@@ -157,11 +326,7 @@ void	ImplGetLogFontFromFontSelect( HDC h
 // - Defines -
 // -----------
 
-#ifdef WIN
-#define MAX_64KSALPOINTS	((((USHORT)0xFFFF)-4)/sizeof(POINT))
-#else
 #define MAX_64KSALPOINTS	((((USHORT)0xFFFF)-8)/sizeof(POINTS))
-#endif
 
 // -----------
 // - Inlines -
Index: vcl/win/inc/salinst.h
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/salinst.h,v
retrieving revision 1.4
retrieving revision 1.7
diff -u -p -u -r1.4 -r1.7
--- vcl/win/inc/salinst.h	11 Apr 2003 17:35:32 -0000	1.4
+++ vcl/win/inc/salinst.h	18 May 2004 10:59:00 -0000	1.7
@@ -66,34 +66,74 @@
 #include <sv.h>
 #endif
 
+#ifndef _SV_SALINST_HXX
+#include <salinst.hxx>
+#endif
+
 namespace vos { class OMutex; }
-class SalYieldMutex;
-class SalInstance;
-class SalFrame;
-class SalObject;
 
 // -------------------
 // - SalInstanceData -
 // -------------------
 
-class SalInstanceData
+class SalYieldMutex;
+
+class WinSalInstance : public SalInstance
 {
 public:
 	HINSTANCE			mhInst; 				// Instance Handle
 	HWND				mhComWnd;				// window, for communication (between threads and the main thread)
-	void*				mpFilterInst;
-	void*				mpFilterCallback;
 	SalYieldMutex*		mpSalYieldMutex;		// Sal-Yield-Mutex
 	vos::OMutex*		mpSalWaitMutex; 		// Sal-Wait-Mutex
 	USHORT				mnYieldWaitCount;		// Wait-Count
+public:
+    WinSalInstance();
+    virtual ~WinSalInstance();
+
+    virtual SalFrame*      	CreateChildFrame( SystemParentData* pParent, ULONG nStyle );
+    virtual SalFrame*      	CreateFrame( SalFrame* pParent, ULONG nStyle );
+    virtual void			DestroyFrame( SalFrame* pFrame );
+    virtual SalObject*		CreateObject( SalFrame* pParent );
+    virtual void			DestroyObject( SalObject* pObject );
+    virtual SalVirtualDevice*	CreateVirtualDevice( SalGraphics* pGraphics,
+                                                     long nDX, long nDY,
+                                                     USHORT nBitCount );
+    virtual void			DestroyVirtualDevice( SalVirtualDevice* pDevice );
+
+    virtual SalInfoPrinter*	CreateInfoPrinter( SalPrinterQueueInfo* pQueueInfo,
+                                               ImplJobSetup* pSetupData );
+    virtual void			DestroyInfoPrinter( SalInfoPrinter* pPrinter );
+    virtual SalPrinter*		CreatePrinter( SalInfoPrinter* pInfoPrinter );
+    virtual void			DestroyPrinter( SalPrinter* pPrinter );
+    virtual void			GetPrinterQueueInfo( ImplPrnQueueList* pList );
+    virtual void			GetPrinterQueueState( SalPrinterQueueInfo* pInfo );
+    virtual void			DeletePrinterQueueInfo( SalPrinterQueueInfo* pInfo );
+    virtual String             GetDefaultPrinter();
+    virtual SalSound*			CreateSalSound();
+    virtual SalTimer*			CreateSalTimer();
+    virtual SalOpenGL*			CreateSalOpenGL( SalGraphics* pGraphics );
+    virtual SalI18NImeStatus*	CreateI18NImeStatus();
+    virtual SalSystem*			CreateSalSystem();
+    virtual SalBitmap*			CreateSalBitmap();
+    virtual vos::IMutex*		GetYieldMutex();
+    virtual ULONG				ReleaseYieldMutex();
+    virtual void				AcquireYieldMutex( ULONG nCount );
+    virtual void				Yield( BOOL bWait );
+    virtual bool				AnyInput( USHORT nType );
+    virtual SalMenu*				CreateMenu( BOOL bMenuBar );
+    virtual void				DestroyMenu( SalMenu* );
+    virtual SalMenuItem*			CreateMenuItem( const SalItemParams* pItemData );
+    virtual void				DestroyMenuItem( SalMenuItem* );
+    virtual SalSession*                         CreateSalSession();
+    virtual void*				GetConnectionIdentifier( ConnectionIdentifierType& rReturnedType, int& rReturnedBytes );
 };
 
 // --------------
 // - Prototypen -
 // --------------
 
-SalFrame* ImplSalCreateFrame( SalInstance* pInst, HWND hWndParent, ULONG nSalFrameStyle );
-SalObject* ImplSalCreateObject( SalInstance* pInst, SalFrame* pParent );
+SalFrame* ImplSalCreateFrame( WinSalInstance* pInst, HWND hWndParent, ULONG nSalFrameStyle );
+SalObject* ImplSalCreateObject( WinSalInstance* pInst, WinSalFrame* pParent );
 HWND ImplSalReCreateHWND( HWND hWndParent, HWND oldhWnd, BOOL bAsChild );
 void ImplSalStartTimer( ULONG nMS, BOOL bMutex = FALSE );
 void ImplSalPrinterAbortJobAsync( HDC hPrnDC );
Index: vcl/win/inc/salmenu.h
===================================================================
RCS file: vcl/win/inc/salmenu.h
diff -N vcl/win/inc/salmenu.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/win/inc/salmenu.h	20 Nov 2003 13:03:11 -0000	1.3
@@ -0,0 +1,118 @@
+/*************************************************************************
+ *
+ *  
+ *
+ *  
+ *
+ *  
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_SALMENU_H
+#define _SV_SALMENU_H
+
+#ifndef _SV_SV_H
+#include <sv.h>
+#endif
+#ifndef _SV_BITMAP_HXX
+#include <bitmap.hxx>
+#endif
+#ifndef _SV_SALMENU_HXX
+#include <salmenu.hxx>
+#endif
+
+
+class WinSalMenu : public SalMenu
+{
+public:
+    WinSalMenu();
+    virtual ~WinSalMenu();
+    virtual BOOL VisibleMenuBar();  // must return TRUE to actually DISPLAY native menu bars
+                            // otherwise only menu messages are processed (eg, OLE on Windows)
+
+    virtual void InsertItem( SalMenuItem* pSalMenuItem, unsigned nPos );
+    virtual void RemoveItem( unsigned nPos );
+    virtual void SetSubMenu( SalMenuItem* pSalMenuItem, SalMenu* pSubMenu, unsigned nPos );
+    virtual void SetFrame( const SalFrame* pFrame );
+    virtual void CheckItem( unsigned nPos, BOOL bCheck );
+    virtual void EnableItem( unsigned nPos, BOOL bEnable );
+    virtual void SetItemText( unsigned nPos, SalMenuItem* pSalMenuItem, const XubString& rText );
+    virtual void SetItemImage( unsigned nPos, SalMenuItem* pSalMenuItem, const Image& rImage );
+    virtual void SetAccelerator( unsigned nPos, SalMenuItem* pSalMenuItem, const KeyCode& rKeyCode, const XubString& rKeyName );
+    virtual void GetSystemMenuData( SystemMenuData* pData );
+
+    HMENU mhMenu;           // the menu handle
+    BOOL  mbMenuBar;        // true for menu bars
+    HWND  mhWnd;            // the window handle where the menubar is attached, may be NULL
+    WinSalMenu *mpParentMenu;  // the parent menu
+};
+
+class WinSalMenuItem : public SalMenuItem
+{
+public:
+    WinSalMenuItem();
+    virtual ~WinSalMenuItem();
+
+
+    MENUITEMINFOW mInfo;
+    void*     mpMenu;       // pointer to corresponding VCL menu
+    XubString mText;        // the item text
+    XubString mAccelText;   // the accelerator string
+    Bitmap    maBitmap;     // item image
+    int       mnId;         // item id
+    WinSalMenu*  mpSalMenu;    // the menu where this item is inserted
+};
+
+#endif // _SV_SALMENU_H
+
Index: vcl/win/inc/salnativewidgets.h
===================================================================
RCS file: vcl/win/inc/salnativewidgets.h
diff -N vcl/win/inc/salnativewidgets.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/win/inc/salnativewidgets.h	10 May 2004 16:01:47 -0000	1.2
@@ -0,0 +1,90 @@
+/*************************************************************************
+ *
+ *  
+ *
+ *  
+ *
+ *  
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_NATIVEWIDGETS_H
+#define _SV_NATIVEWIDGETS_H
+
+
+#ifdef __cplusplus
+
+#ifndef _SV_SV_H
+#include <sv.h>
+#endif
+
+/* SalControlHandleData:
+ * 
+ *   Holds platform specific theming data.
+ */
+
+class SalControlHandleData
+{
+	public:
+		SalControlHandleData( void );
+		~SalControlHandleData( void );
+
+	public:
+        // nothing needed on Win32
+};
+
+
+#endif	/* __cplusplus */
+
+#endif
Index: vcl/win/inc/salobj.h
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/salobj.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -p -u -r1.1.1.1 -r1.2
--- vcl/win/inc/salobj.h	18 Sep 2000 17:05:49 -0000	1.1.1.1
+++ vcl/win/inc/salobj.h	18 Nov 2003 14:49:11 -0000	1.2
@@ -68,12 +68,15 @@
 #ifndef _SV_SYSDATA_HXX
 #include <sysdata.hxx>
 #endif
+#ifndef _SV_SALOBJ_HXX
+#include <salobj.hxx>
+#endif
 
 // -----------------
 // - SalObjectData -
 // -----------------
 
-class SalObjectData
+class WinSalObject : public SalObject
 {
 public:
 	HWND					mhWnd;					// Window handle
@@ -84,9 +87,24 @@ public:
 	RGNDATA*				mpStdClipRgnData;		// Cache Standard-ClipRegion-Data
 	RECT*					mpNextClipRect; 		// Naechstes ClipRegion-Rect
 	BOOL					mbFirstClipRect;		// Flag for first cliprect to insert
-	SalObject*				mpNextObject;			// pointer to next object
-	void*					mpInst; 				// instance handle for callback
-	SALOBJECTPROC			mpProc; 				// callback proc
+	WinSalObject*				mpNextObject;			// pointer to next object
+
+
+    WinSalObject();
+    virtual ~WinSalObject();
+
+	virtual void					ResetClipRegion();
+	virtual USHORT					GetClipRegionType();
+	virtual void					BeginSetClipRegion( ULONG nRects );
+	virtual void					UnionClipRegion( long nX, long nY, long nWidth, long nHeight );
+	virtual void					EndSetClipRegion();
+	virtual void					SetPosSize( long nX, long nY, long nWidth, long nHeight );
+	virtual void					Show( BOOL bVisible );
+	virtual void					Enable( BOOL nEnable );
+	virtual void					GrabFocus();
+	virtual void					SetBackground();
+	virtual void					SetBackground( SalColor nSalColor );
+	virtual const SystemEnvData*	GetSystemData() const;
 };
 
 #endif // _SV_SALOBJ_H
Index: vcl/win/inc/salogl.h
===================================================================
RCS file: vcl/win/inc/salogl.h
diff -N vcl/win/inc/salogl.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/win/inc/salogl.h	18 Nov 2003 14:49:19 -0000	1.2
@@ -0,0 +1,121 @@
+/*************************************************************************
+ *
+ *  
+ *
+ *  
+ *
+ *  
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_SALOGL_H
+#define _SV_SALOGL_H
+
+#define _OPENGL_EXT
+
+#ifndef _SV_WINCOMP_HXX
+#include <wincomp.hxx>
+#endif
+#ifndef _GEN_HXX
+#include <tools/gen.hxx>
+#endif
+#ifndef _SV_SV_H
+#include <sv.h>
+#endif
+#ifndef _SV_SALOGL_HXX
+#include <salogl.hxx>
+#endif
+
+// -----------------
+// - State defines -
+// -----------------
+
+#define OGL_STATE_UNLOADED		(0x00000000)
+#define OGL_STATE_INVALID		(0x00000001)
+#define OGL_STATE_VALID 		(0x00000002)
+
+// -------------
+// - SalOpenGL -
+// -------------
+
+class SalGraphics;
+class String;
+
+class WinSalOpenGL : public SalOpenGL
+{
+private:
+	static HGLRC		mhOGLContext;
+	static HDC			mhOGLLastDC;
+	static ULONG		mnOGLState;
+
+private:
+	static BOOL 		ImplInitLib();
+	static BOOL 		ImplInit();
+	static void 		ImplFreeLib();
+
+public:
+    WinSalOpenGL( SalGraphics* );
+    virtual ~WinSalOpenGL();
+						
+	virtual bool		IsValid();
+	virtual void*		GetOGLFnc( const char * );
+	virtual void		OGLEntry( SalGraphics* pGraphics );
+	virtual void		OGLExit( SalGraphics* pGraphics );
+    virtual void		StartScene( SalGraphics* pGraphics );
+	virtual void		StopScene();
+
+    static void         Release();
+};
+
+#endif // _SV_SALOGL_H
Index: vcl/win/inc/salprn.h
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/salprn.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -p -u -r1.1.1.1 -r1.2
--- vcl/win/inc/salprn.h	18 Sep 2000 17:05:49 -0000	1.1.1.1
+++ vcl/win/inc/salprn.h	18 Nov 2003 14:49:31 -0000	1.2
@@ -66,8 +66,9 @@
 #include <sv.h>
 #endif
 
-class SalGraphics;
-class SalInfoPrinter;
+#ifndef _SV_SALPRN_HXX
+#include <salprn.hxx>
+#endif
 
 // -----------------
 // - SalDriverData -
@@ -105,14 +106,16 @@ struct SalSysQueueData
 	BOOL					mbAnsi; 				// TRUE - use A functions
 };
 
-// ----------------------
-// - SalInfoPrinterData -
-// ----------------------
+// ---------------------
+// - WinSalInfoPrinter -
+// ---------------------
+
+class WinSalGraphics;
 
-class SalInfoPrinterData
+class WinSalInfoPrinter : public SalInfoPrinter
 {
 public:
-	SalGraphics*			mpGraphics; 			// current Printer graphics
+	WinSalGraphics*			mpGraphics; 			// current Printer graphics
 	XubString				maDriverName;			// printer driver name
 	XubString				maDeviceName;			// printer device name
 	XubString				maPortName; 			// printer port name
@@ -122,23 +125,57 @@ public:
 	HDC 					mhDC;					// printer hdc
 	BOOL					mbGraphics; 			// is Graphics used
 	BOOL					mbAnsi;
+
+public:
+    WinSalInfoPrinter();
+    virtual ~WinSalInfoPrinter();
+
+	virtual SalGraphics*			GetGraphics();
+	virtual void					ReleaseGraphics( SalGraphics* pGraphics );
+	virtual BOOL					Setup( SalFrame* pFrame, ImplJobSetup* pSetupData );
+	virtual BOOL					SetPrinterData( ImplJobSetup* pSetupData );
+	virtual BOOL					SetData( ULONG nFlags, ImplJobSetup* pSetupData );
+	virtual void					GetPageInfo( const ImplJobSetup* pSetupData,
+                                                 long& rOutWidth, long& rOutHeight,
+                                                 long& rPageOffX, long& rPageOffY,
+                                                 long& rPageWidth, long& rPageHeight );
+	virtual ULONG					GetCapabilities( const ImplJobSetup* pSetupData, USHORT nType );
+	virtual ULONG					GetPaperBinCount( const ImplJobSetup* pSetupData );
+	virtual String					GetPaperBinName( const ImplJobSetup* pSetupData, ULONG nPaperBin );
+    virtual void					InitPaperFormats( const ImplJobSetup* pSetupData );
+    virtual int					GetLandscapeAngle( const ImplJobSetup* pSetupData );
 };
 
-// ------------------
-// - SalPrinterData -
-// ------------------
+// -----------------
+// - WinSalPrinter -
+// -----------------
 
-class SalPrinterData
+class WinSalPrinter : public SalPrinter
 {
 public:
-	SalGraphics*			mpGraphics; 			// current Printer graphics
-	SalInfoPrinter* 		mpInfoPrinter;			// pointer to the compatible InfoPrinter
-	SalPrinter* 			mpNextPrinter;			// next printing printer
+	WinSalGraphics*			mpGraphics; 			// current Printer graphics
+	WinSalInfoPrinter* 		mpInfoPrinter;			// pointer to the compatible InfoPrinter
+	WinSalPrinter* 			mpNextPrinter;			// next printing printer
 	HDC 					mhDC;					// printer hdc
 	ULONG					mnError;				// Error Code
 	ULONG					mnCopies;				// Kopien
 	BOOL					mbCollate;				// Sortierte Kopien
 	BOOL					mbAbort;				// Job Aborted
+
+public:
+    WinSalPrinter();
+    virtual ~WinSalPrinter();
+
+	virtual BOOL					StartJob( const XubString* pFileName,
+                                              const XubString& rJobName,
+                                              const XubString& rAppName,
+                                              ULONG nCopies, BOOL bCollate,
+                                              ImplJobSetup* pSetupData );
+	virtual BOOL					EndJob();
+	virtual BOOL					AbortJob();
+	virtual SalGraphics*			StartPage( ImplJobSetup* pSetupData, BOOL bNewJobData );
+	virtual BOOL					EndPage();
+	virtual ULONG					GetErrorCode();
 };
 
 #endif // _SV_SALPRN_H
Index: vcl/win/inc/salsound.h
===================================================================
RCS file: vcl/win/inc/salsound.h
diff -N vcl/win/inc/salsound.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/win/inc/salsound.h	18 Nov 2003 14:49:40 -0000	1.2
@@ -0,0 +1,132 @@
+/*************************************************************************
+ *
+ *  
+ *
+ *  
+ *
+ *  
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_SALSOUND_H
+#define _SV_SALSOUND_H
+
+#ifndef _SV_WINCOMP_HXX
+#include <wincomp.hxx>
+#endif
+
+#ifndef _GEN_HXX
+#include <tools/gen.hxx>
+#endif
+#ifndef _STRING_HXX
+#include <tools/string.hxx>
+#endif
+
+#ifndef _SV_SV_H
+#include <sv.h>
+#endif
+
+#ifndef _SV_SALFRAME_HXX
+#include <salframe.hxx>
+#endif
+#ifndef _SV_SALSTYPE_HXX
+#include <salstype.hxx>
+#endif
+#ifndef _SV_SALSOUND_HXX
+#include <salsound.hxx>
+#endif
+
+// ------------
+// - SalSound -
+// ------------
+
+class WinSalSound : public SalSound
+{
+private:
+	static HINSTANCE	mhMCILib;
+	static ULONG		mnSoundState;
+	static void*		mpMCIFnc;
+	ULONG				mnStartTime;
+	ULONG				mnPlayLen;
+	HWND				mhSoundWnd;
+	UINT				mnDeviceId;
+	bool				mbLoop;
+	bool				mbPaused;
+
+public:
+	void				ImplSetError( DWORD nMciErr );
+	void				ImplNotify( SoundNotification eNotification, ULONG nError );
+
+public:
+    WinSalSound();
+    virtual ~WinSalSound();
+    
+    virtual bool		IsValid();
+    virtual bool		Init( const String&	rSoundName,
+                              ULONG&		rSoundLen );
+   	virtual void		Play( ULONG nStartTime, ULONG nPlayTime, bool bLoop );
+	virtual void		Stop();
+	virtual void		Pause();
+	virtual void		Continue();
+    virtual bool		IsLoopMode() const;
+	virtual bool		IsPlaying() const;
+    virtual bool		IsPaused() const;
+
+
+	bool				ImplCreate();
+	void				ImplDestroy();
+	static void 		Release();
+};
+
+#endif // _SV_SALSOUND_H
Index: vcl/win/inc/saltimer.h
===================================================================
RCS file: vcl/win/inc/saltimer.h
diff -N vcl/win/inc/saltimer.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/win/inc/saltimer.h	18 Nov 2003 14:49:47 -0000	1.2
@@ -0,0 +1,80 @@
+/*************************************************************************
+ *
+ *  
+ *
+ *  
+ *
+ *  
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_SALTIMER_H
+#define _SV_SALTIMER_H
+
+#ifndef _SV_SALTIMER_HXX
+#include <saltimer.hxx>
+#endif
+
+class WinSalTimer : public SalTimer
+{
+public:
+    WinSalTimer() {}
+    virtual ~WinSalTimer();
+
+    // overload all pure virtual methods
+	void 			Start( ULONG nMS );
+	void 			Stop();
+};
+
+#endif
Index: vcl/win/inc/salvd.h
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/salvd.h,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -p -u -r1.1.1.1 -r1.2
--- vcl/win/inc/salvd.h	18 Sep 2000 17:05:49 -0000	1.1.1.1
+++ vcl/win/inc/salvd.h	18 Nov 2003 14:50:00 -0000	1.2
@@ -66,23 +66,33 @@
 #include <sv.h>
 #endif
 
-class SalGraphics;
-class SalVirtualDevice;
+#ifndef _SV_SALVD_HXX
+#include <salvd.hxx>
+#endif
+
+class WinSalGraphics;
 
 // -----------------
 // - SalVirDevData -
 // -----------------
 
-class SalVirDevData
+class WinSalVirtualDevice : public SalVirtualDevice
 {
 public:
 	HDC 					mhDC;					// HDC or 0 for Cache Device
 	HBITMAP 				mhBmp;					// Memory Bitmap
 	HBITMAP 				mhDefBmp;				// Default Bitmap
-	SalGraphics*			mpGraphics; 			// current VirDev graphics
-	SalVirtualDevice*		mpNext; 				// next VirDev
+	WinSalGraphics*			mpGraphics; 			// current VirDev graphics
+	WinSalVirtualDevice*	mpNext; 				// next VirDev
 	USHORT					mnBitCount; 			// BitCount (0 or 1)
 	BOOL					mbGraphics; 			// is Graphics used
+
+    WinSalVirtualDevice();
+    virtual ~WinSalVirtualDevice();
+
+    virtual SalGraphics*			GetGraphics();
+    virtual void					ReleaseGraphics( SalGraphics* pGraphics );
+    virtual BOOL                    SetSize( long nNewDX, long nNewDY );
 };
 
 #endif // _SV_SALVD_H
Index: vcl/win/inc/wincomp.hxx
===================================================================
RCS file: /cvs/gsl/vcl/vcl/win/inc/wincomp.hxx,v
retrieving revision 1.1.1.1
retrieving revision 1.2
diff -u -p -u -r1.1.1.1 -r1.2
--- vcl/win/inc/wincomp.hxx	18 Sep 2000 17:05:49 -0000	1.1.1.1
+++ vcl/win/inc/wincomp.hxx	6 Jan 2004 14:50:04 -0000	1.2
@@ -186,241 +186,69 @@ inline HFONT GetWindowFont( HWND hWnd )
 	return (HFONT)(UINT)SendMessage( hWnd, WM_GETFONT, 0, 0 );
 }
 
-// ---------------------
-// - Windows/Window NT -
-// ---------------------
-
-// Anpassung fuer Unterschiede zwischen 3.x und NT
-
 inline void SetClassCursor( HWND hWnd, HCURSOR hCursor )
 {
-#ifndef WNT
-	SetClassWord( hWnd, GCW_HCURSOR, (WORD)hCursor );
-#else
 	SetClassLong( hWnd, GCL_HCURSOR, (DWORD)hCursor );
-#endif
 }
 
 inline HCURSOR GetClassCursor( HWND hWnd )
 {
-#ifndef WNT
-	return (HCURSOR)GetClassWord( hWnd, GCW_HCURSOR );
-#else
 	return (HCURSOR)GetClassLong( hWnd, GCL_HCURSOR );
-#endif
 }
 
 inline void SetClassIcon( HWND hWnd, HICON hIcon )
 {
-#ifndef WNT
-	SetClassWord( hWnd, GCW_HICON, (WORD)hIcon );
-#else
 	SetClassLong( hWnd, GCL_HICON, (DWORD)hIcon );
-#endif
 }
 
 inline HICON GetClassIcon( HWND hWnd )
 {
-#ifndef WNT
-	return (HICON)GetClassWord( hWnd, GCW_HICON );
-#else
 	return (HICON)GetClassLong( hWnd, GCL_HICON );
-#endif
 }
 
 inline HBRUSH SetClassBrush( HWND hWnd, HBRUSH hBrush )
 {
-#ifndef WNT
-	return (HBRUSH)SetClassWord( hWnd, GCW_HBRBACKGROUND, (WORD)hBrush );
-#else
 	return (HBRUSH)SetClassLong( hWnd, GCL_HBRBACKGROUND, (DWORD)hBrush );
-#endif
 }
 
 inline HBRUSH GetClassBrush( HWND hWnd )
 {
-#ifndef WNT
-	return (HBRUSH)GetClassWord( hWnd, GCW_HBRBACKGROUND );
-#else
 	return (HBRUSH)GetClassLong( hWnd, GCL_HBRBACKGROUND );
-#endif
 }
 
 inline HINSTANCE GetWindowInstance( HWND hWnd )
 {
-#ifndef WNT
-	return (HINSTANCE)GetWindowWord( hWnd, GWW_HINSTANCE );
-#else
 	return (HINSTANCE)GetWindowLong( hWnd, GWL_HINSTANCE );
-#endif
-}
-
-#ifndef WNT
-inline UINT CharLowerBuff( LPSTR lpStr, UINT nLen )
-{
-	return AnsiLowerBuff( lpStr, nLen );
-}
-#endif
-
-#ifndef WNT
-inline UINT CharUpperBuff( LPSTR lpStr, UINT nLen )
-{
-	return AnsiUpperBuff( lpStr, nLen );
-}
-#endif
-
-#ifndef WNT
-inline void OemToChar( LPCSTR lpStr1, LPSTR lpStr2 )
-{
-	OemToAnsi( lpStr1, lpStr2 );
 }
-#endif
-
-
-// -----------------------------------
-// - Unterschiede zwischen 16/32-Bit -
-// -----------------------------------
-
-#ifdef WIN
-#define SVWINAPI	WINAPI
-#else
-#define SVWINAPI	APIENTRY
-#endif
-
-#ifdef WIN
-#define NEARDATA	_near
-#else
-#define NEARDATA
-#endif
-
-// Zum kopieren von mehr als 64 KB
-#ifdef WIN
-inline void lmemcpy( void* pDst, const void* pSrc, ULONG nSize )
-{
-	hmemcpy( pDst, pSrc, nSize );
-}
-#else
-inline void lmemcpy( void* pDst, const void* pSrc, ULONG nSize )
-{
-	memcpy( pDst, pSrc, nSize );
-}
-#endif
-
-#ifdef WNT
-typedef LONG	WinWeight;
-#else
-typedef int 	WinWeight;
-#endif
-
-
-// ----------------------------------------------------
-// - Steuerungen fuer Versionen und Laufzeit-Abfragen -
-// ----------------------------------------------------
-
-#if defined( WNT )
-#define W95_VERSION 		400
-#else
-#define W95_VERSION 		395
-#endif
-
-// Wenn eine 32-Bit SV Version die nur unter W95 laeuft gebildet werden soll,
-// muss nur dieses Define W40ONLY definiert werden
-#if ( WINVER >= 0x0400 )
-#define W40ONLY
-#endif
-
-// Wenn wir sowieso erst ab W95 laufen, brauchen wir auch keine
-// Laufzeit-Abfragen
-#ifdef W40ONLY
-#define W40IF
-#define W40NIF
-#define W40ELSE
-
-#else
-
-// Nur ein 32-Bit-SDK definiert WINVER >= 0x0400 und somit brauchen wir
-// diese W40-Abfragen auch nur hier. Die Abfragen, die sowohl fuer 3.1
-// als auch fuer NT gelten sind als normale if-Abfragen kodiert
-#ifdef WIN
-#define W40NIF
-#else
-#define W40IF				if ( aSalShlData.mbW40 )
-#define W40NIF				if ( !aSalShlData.mbW40 )
-#define W40ELSE 			else
-#endif
-
-#endif
-
-/****************************
-
-Beispiel fuer Klammerung:
-
-#if ( WINVER >= 0x0400 )
-	W40IF
-	{
-	... W40-Code
-	}
-	W40ELSE
-#endif
-#ifndef W40ONLY
-	{
-	... Normaler 3.1 und NT 3.5(1)-Code
-	}
-#endif
-
-*****************************/
-
 
 // ------------------------
 // - ZMouse Erweiterungen -
 // ------------------------
 
-#if defined( WNT )
-
-#ifdef UNICODE
-#define MSH_MOUSEWHEEL L"MSWHEEL_ROLLMSG"
-#else
 #define MSH_MOUSEWHEEL "MSWHEEL_ROLLMSG"
-#endif
 
-// Default value for rolling one notch
-#ifndef WHEEL_DELTA
-#define WHEEL_DELTA 	 120
-#endif
-
-#ifndef WM_MOUSEWHEEL
-#define WM_MOUSEWHEEL					0x020A
-#endif
-
-#ifndef  WHEEL_PAGESCROLL
-// signifies to scroll a page, also defined in winuser.h in the NT4.0 SDK
-#define WHEEL_PAGESCROLL  (UINT_MAX)
-#endif
-
-#ifdef UNICODE
-#define MOUSEZ_CLASSNAME  L"MouseZ"           // wheel window class
-#define MOUSEZ_TITLE	  L"Magellan MSWHEEL" // wheel window title
-#else
 #define MOUSEZ_CLASSNAME  "MouseZ"            // wheel window class
 #define MOUSEZ_TITLE	  "Magellan MSWHEEL"  // wheel window title
-#endif
 
 #define MSH_WHEELMODULE_CLASS (MOUSEZ_CLASSNAME)
 #define MSH_WHEELMODULE_TITLE (MOUSEZ_TITLE)
 
-#ifdef UNICODE
-#define MSH_SCROLL_LINES L"MSH_SCROLL_LINES_MSG"
-#else
 #define MSH_SCROLL_LINES "MSH_SCROLL_LINES_MSG"
-#endif
 
+#ifndef WHEEL_DELTA
+#define WHEEL_DELTA 				120
+#endif
+#ifndef WM_MOUSEWHEEL
+#define WM_MOUSEWHEEL				0x020A
+#endif
 #ifndef SPI_GETWHEELSCROLLLINES
-#define SPI_GETWHEELSCROLLLINES   104
+#define SPI_GETWHEELSCROLLLINES		104
 #endif
 #ifndef SPI_SETWHEELSCROLLLINES
-#define SPI_SETWHEELSCROLLLINES   105
+#define SPI_SETWHEELSCROLLLINES		105
 #endif
-
+#ifndef WHEEL_PAGESCROLL
+#define WHEEL_PAGESCROLL			(UINT_MAX)
 #endif
 
 
@@ -428,26 +256,46 @@ Beispiel fuer Klammerung:
 // - SystemAgent Erweiterungen -
 // -----------------------------
 
-#if ( WINVER >= 0x0400 )
 #define ENABLE_AGENT			1
 #define DISABLE_AGENT			2
 #define GET_AGENT_STATUS		3
-
-typedef int (SVWINAPI* SysAgt_Enable_PROC)( int );
-#endif
+typedef int (APIENTRY* SysAgt_Enable_PROC)( int );
 
 // ---------------------
 // - 5.0-Erweiterungen -
 // ---------------------
 
-#ifndef COLOR_HOTLIGHT
-#define COLOR_HOTLIGHT					26
-#endif
 #ifndef COLOR_GRADIENTACTIVECAPTION
 #define COLOR_GRADIENTACTIVECAPTION 	27
 #endif
 #ifndef COLOR_GRADIENTINACTIVECAPTION
 #define COLOR_GRADIENTINACTIVECAPTION	28
+#endif
+
+#ifndef SPI_GETFLATMENU
+#define SPI_GETFLATMENU     0x1022
+#endif
+#ifndef COLOR_MENUBAR
+#define COLOR_MENUBAR       30
+#endif
+#ifndef COLOR_MENUHILIGHT
+#define COLOR_MENUHILIGHT   29   
+#endif
+
+#ifndef CS_DROPSHADOW
+#define CS_DROPSHADOW       0x00020000
+#endif
+
+// -------------------------------------------------------
+// MT 12/03: From winuser.h, only needed in salframe.cxx
+// Better change salframe.cxx to include winuser.h
+// -------------------------------------------------------
+
+#define WS_EX_LAYERED           0x00080000
+
+#ifndef WM_UNICHAR
+#define WM_UNICHAR              0x0109
+#define UNICODE_NOCHAR          0xFFFF
 #endif
 
 #endif // _SV_WINCOMP_HXX
