--- vcl/source/window/menu.cxx.before	2004-07-20 17:53:04.546875000 +0100
+++ vcl/source/window/menu.cxx	2004-07-20 17:53:43.812500000 +0100
@@ -2395,7 +2395,7 @@
                     aSz.Width() += pData->aSz.Width();
                 }
 				else
-					pData->aSz.Height() = std::max( std::max( nTextHeight, pData->aSz.Height() ), nMinMenuItemHeight );
+					pData->aSz.Height() = Max( Max( nTextHeight, pData->aSz.Height() ), nMinMenuItemHeight );
 		nWidth += nTextWidth;
             }
 
@@ -2414,7 +2414,7 @@
                 if ( nFontHeight > nWidth )
                     nWidth += nFontHeight;
 
-				pData->aSz.Height() = std::max( std::max( nFontHeight, pData->aSz.Height() ), nMinMenuItemHeight );
+				pData->aSz.Height() = Max( Max( nFontHeight, pData->aSz.Height() ), nMinMenuItemHeight );
 
 //                if ( nFontHeight > pData->aSz.Height() )
 //                    pData->aSz.Height() = nFontHeight;
@@ -2435,7 +2435,7 @@
         int nCheckWidth = 0;
         if ((nMenuFlags & MENU_FLAG_SHOWCHECKIMAGES) || aMaxImgSz.Width() <= 0)
 			nCheckWidth = nFontHeight/2 + nExtra;
-        USHORT gfxExtra = (USHORT) std::max( nExtra, 7L ); // #107710# increase space between checkmarks/images/text
+        USHORT gfxExtra = (USHORT) Max( nExtra, 7L ); // #107710# increase space between checkmarks/images/text
         nCheckPos = nExtra;
         nImagePos = nCheckPos + nCheckWidth;
         nTextPos = nImagePos + aMaxImgSz.Width();
