--- sc/source/ui/vba/vbaapplication.cxx.orig	2008-01-16 14:14:06.000000000 +0800
+++ sc/source/ui/vba/vbaapplication.cxx	2008-01-16 16:47:25.000000000 +0800
@@ -62,6 +62,7 @@
 #include "gridwin.hxx"
 #include "vbanames.hxx"
 #include "vbashape.hxx"
+#include "vbaassistant.hxx"
 #include "sc.hrc"
 
 #include <osl/file.hxx>
@@ -129,6 +130,12 @@ ScVbaApplication::getThisWorkbook() thro
 	return getActiveWorkbook();
 }
 
+uno::Reference< vba::XAssistant > SAL_CALL
+ScVbaApplication::getAssistant() throw (css::uno::RuntimeException)
+{
+    return uno::Reference< vba::XAssistant >( new ScVbaAssistant( this, mxContext ) );
+}
+
 uno::Any SAL_CALL
 ScVbaApplication::getSelection() throw (uno::RuntimeException)
 {
@@ -433,12 +440,17 @@ ScVbaApplication::Range( const uno::Any&
 }
 
 uno::Any SAL_CALL
-ScVbaApplication::Names( ) throw ( uno::RuntimeException )
+ScVbaApplication::Names( const css::uno::Any& aIndex ) throw ( uno::RuntimeException )
 {
     uno::Reference< frame::XModel > xModel( getCurrentDocument(), uno::UNO_QUERY_THROW );
-    uno::Reference< sheet::XNamedRanges > xNamedRanges( getActiveSheet() , uno::UNO_QUERY_THROW );
+    uno::Reference< beans::XPropertySet > xPropertySet( xModel, uno::UNO_QUERY_THROW );
+    uno::Reference< sheet::XNamedRanges > xNamedRanges( xPropertySet->getPropertyValue( rtl::OUString::createFromAscii("NamedRanges")) , uno::UNO_QUERY_THROW );
     css::uno::Reference< excel::XNames > xNames ( new ScVbaNames( this , mxContext , xNamedRanges , xModel ) );
-        return uno::makeAny( xNames );
+    if (  aIndex.getValueTypeClass() == uno::TypeClass_VOID )
+    {
+        return uno::Any( xNames );
+    }
+    return uno::Any( xNames->Item( aIndex, uno::Any() ) );
 }
 
 
