--- canvas/source/cairo/cairo_canvashelper.cxx	2007-07-25 17:28:13.000000000 +0200
+++ canvas/source/cairo/cairo_canvashelper.cxx	2007-07-25 19:27:41.000000000 +0200
@@ -845,9 +845,10 @@ namespace cairocanvas
 	cairo_set_matrix( pCairo, &aIdentityMatrix );
 
 	for( sal_uInt32 nPolygonIndex = 0; nPolygonIndex < aPolyPolygon.count(); nPolygonIndex++ ) {
-	    ::basegfx::B2DPolygon aPolygon = aPolyPolygon.getB2DPolygon( nPolygonIndex );
+	    ::basegfx::B2DPolygon aPolygon( aPolyPolygon.getB2DPolygon( nPolygonIndex ) );
+	    const sal_uInt32 nPointCount( aPolygon.count() );
 
-	    if( aPolygon.count() > 1) {
+	    if( nPointCount > 1) {
 		bool bIsBezier = aPolygon.areControlPointsUsed();
 		::basegfx::B2DPoint aA, aB, aP;
 
@@ -871,11 +872,11 @@ namespace cairocanvas
 		OSL_TRACE( "move to %f,%f\n", nX, nY );
 
 		if( bIsBezier ) {
-		    aA = aPolygon.getControlPointA( 0 );
-		    aB = aPolygon.getControlPointB( 0 );
+		    aA = aPolygon.getNextControlPoint( 0 );
+		    aB = aPolygon.getPrevControlPoint( 1 );
 		}
 
-		for( sal_uInt32 j = 1; j < aPolygon.count(); j++ ) {
+		for( sal_uInt32 j = 1; j < nPointCount; j++ ) {
 		    aP = aPolygon.getB2DPoint( j );
 
 		    nX = aP.getX();
@@ -908,8 +909,8 @@ namespace cairocanvas
 			cairo_matrix_transform_point( &aOrigMatrix, &nBX, &nBY );
 			cairo_curve_to( pCairo, nAX, nAY, nBX, nBY, nX, nY );
 			
-			aA = aPolygon.getControlPointA( j );
-			aB = aPolygon.getControlPointB( j );
+			aA = aPolygon.getNextControlPoint( j );
+			aB = aPolygon.getPrevControlPoint( ( j + 1 ) % nPointCount );
 		    } else {
 			cairo_line_to( pCairo, nX, nY );
 			OSL_TRACE( "line to %f,%f\n", nX, nY );
