--- canvas/source/tools/spriteredrawmanager.cxx	2005-11-02 13:55:15.000000000 +0100
+++ canvas/source/tools/spriteredrawmanager.cxx	2006-02-08 23:07:11.000000000 +0100
@@ -399,7 +399,7 @@ namespace canvas
         // now, calc the _true_ update area, by merging all sprite's
         // true update areas into one rectangle
         ::basegfx::B2DRange aTrueArea( aBegin->second.getUpdateArea() );
-        ::std::for_each( ++rUpdateArea.maComponentList.begin(),
+        ::std::for_each( aBegin,
                          aEnd,
                          ::boost::bind( ::basegfx::B2DRangeExpander(aTrueArea),
                                         ::boost::bind( &SpriteInfo::getUpdateArea,
--- canvas/inc/canvas/spriteredrawmanager.hxx	2005-11-02 13:41:12.000000000 +0100
+++ canvas/inc/canvas/spriteredrawmanager.hxx	2006-02-08 23:07:48.000000000 +0100
@@ -97,6 +97,8 @@ namespace canvas
         class SpriteInfo
         {
         public:
+			~SpriteInfo() {}
+
             /** Create sprite info
 
             	@param rRef
@@ -153,7 +155,7 @@ namespace canvas
             }
 
             const Sprite::Reference&	getSprite() const { return mpSprite; }
-            const ::basegfx::B2DRange&	getUpdateArea() const { return maTrueUpdateArea; }
+            ::basegfx::B2DRange	        getUpdateArea() const { return maTrueUpdateArea; }
             bool						needsUpdate() const { return mbNeedsUpdate; }
             bool						isPureMove() const { return mbIsPureMove; }
 
