--- canvas/prj/build.lst	2006-11-02 13:13:45.000000000 +0100
+++ canvas/prj/build.lst	2006-11-02 13:14:17.000000000 +0100
@@ -1,4 +1,4 @@
-cv	canvas	:	javaunohelper comphelper cppuhelper offuh unoil tools goodies vcl AGG:agg basegfx NULL
+cv	canvas	:	javaunohelper comphelper cppuhelper offuh unoil tools goodies vcl AGG:agg basegfx CAIRO:cairo NULL
 cv	canvas											 usr1	-	all	cv_mkout NULL
 cv	canvas\inc										 nmake	-	all	cv_inc NULL
 cv	canvas\source\tools								 nmake	-	all cv_tools cv_inc NULL
--- canvas/source/cairo/makefile.mk	2008-04-11 10:59:46.000000000 +0200
+++ canvas/source/cairo/makefile.mk	2008-05-19 14:12:29.000000000 +0200
@@ -82,7 +82,7 @@
 .IF "$(SYSTEM_CAIRO)" == "YES"
 SHL1STDLIBS+= $(CAIRO_LIBS)
 .ELSE
-SHL1STDLIBS+= -lcairo -lpixman-1
+SHL1STDLIBS+= -lcairo -lfontconfig $(FREETYPELIB)
 .ENDIF
 
 .IF "$(GUIBASE)"=="aqua"
