--- ../src680-m152-orig/vcl/unx/source/gdi/salgdi3.cxx	2006-02-17 14:31:27.000000000 +0100
+++ vcl/unx/source/gdi/salgdi3.cxx	2006-02-22 16:34:14.000000000 +0100
@@ -742,12 +742,17 @@ void X11SalGraphics::DrawServerAAFontStr
 	}
 
     // create xrender Picture for font foreground
+	bool bDeleteSrc = false;
+	Pixmap aSrcPixmap;
+	Pixmap aSrcPicture;
+
     static Pixmap aPixmap;
     static Picture aSrc = 0;
+	static unsigned nDepth;
     if( !aSrc )
     {
         int iDummy;
-        unsigned uDummy, nDepth;
+        unsigned uDummy;
         XLIB_Window wDummy;
         XGetGeometry( pDisplay, hDrawable_, &wDummy, &iDummy, &iDummy,
             &uDummy, &uDummy, &uDummy, &nDepth );
@@ -760,15 +765,31 @@ void X11SalGraphics::DrawServerAAFontStr
 #else
         aSrc = (*aX11GlyphPeer.pXRenderCreatePicture)( pDisplay, aPixmap, pVisualFormat, CPRepeat, &aAttr );
 #endif
-    }
+		aSrcPicture = aSrc;
+		aSrcPixmap = aPixmap;
+    } else if( pVisualFormat && pVisualFormat->depth != nDepth ) {
+        aSrcPixmap = XCreatePixmap( pDisplay, hDrawable_, 1, 1, pVisualFormat->depth );
+
+        XRenderPictureAttributes aAttr;
+        aAttr.repeat = true;
+#ifdef XRENDER_LINK
+        aSrcPicture = XRenderCreatePicture ( pDisplay, aSrcPixmap, pVisualFormat, CPRepeat, &aAttr );
+#else
+        aSrcPicture = (*aX11GlyphPeer.pXRenderCreatePicture)( pDisplay, aSrcPixmap, pVisualFormat, CPRepeat, &aAttr );
+#endif
+		bDeleteSrc = true;
+	} else {
+		aSrcPicture = aSrc;
+		aSrcPixmap = aPixmap;
+	}
 
     // set font foreground
     GC nGC = SelectFont();
     XGCValues aGCVal;
     XGetGCValues( pDisplay, nGC, GCForeground, &aGCVal );
     aGCVal.clip_mask = None;
-    GC tmpGC = XCreateGC( pDisplay, aPixmap, GCForeground | GCClipMask, &aGCVal );
-    XDrawPoint( pDisplay, aPixmap, tmpGC, 0, 0 );
+    GC tmpGC = XCreateGC( pDisplay, aSrcPixmap, GCForeground | GCClipMask, &aGCVal );
+    XDrawPoint( pDisplay, aSrcPixmap, tmpGC, 0, 0 );
     XFreeGC( pDisplay, tmpGC );
 
     // notify xrender of target drawable
@@ -810,10 +831,10 @@ void X11SalGraphics::DrawServerAAFontStr
              aRenderAry[ i ] = aX11GlyphPeer.GetGlyphId( rFont, aGlyphAry[i] );
 #ifdef XRENDER_LINK
         XRenderCompositeString32 ( pDisplay, PictOpOver,
-            aSrc, aDst, 0, aGlyphSet, 0, 0, aPos.X(), aPos.Y(), aRenderAry, nGlyphs );
+            aSrcPicture, aDst, 0, aGlyphSet, 0, 0, aPos.X(), aPos.Y(), aRenderAry, nGlyphs );
 #else
         (*aX11GlyphPeer.pXRenderCompositeString32)( pDisplay, PictOpOver,
-            aSrc, aDst, 0, aGlyphSet, 0, 0, aPos.X(), aPos.Y(), aRenderAry, nGlyphs );
+            aSrcPicture, aDst, 0, aGlyphSet, 0, 0, aPos.X(), aPos.Y(), aRenderAry, nGlyphs );
 #endif
     }
 
@@ -823,6 +844,16 @@ void X11SalGraphics::DrawServerAAFontStr
 #else
     (*aX11GlyphPeer.pXRenderFreePicture)( pDisplay, aDst );
 #endif
+
+	if( bDeleteSrc ) {
+		XFreePixmap( pDisplay, aSrcPixmap );
+
+#ifdef XRENDER_LINK
+		XRenderFreePicture ( pDisplay, aSrcPicture );
+#else
+		(*aX11GlyphPeer.pXRenderFreePicture)( pDisplay, aSrcPicture );
+#endif
+	}
 }
 
 //--------------------------------------------------------------------------
