--- config_office/configure.in.old	2005-08-03 15:23:40.000000000 +0200
+++ config_office/configure.in	2005-08-03 15:28:42.000000000 +0200
@@ -137,6 +137,10 @@
 [  --disable-gtk           Determines whether to use Gtk+ vclplug on platforms
                           where Gtk+ is available.
 ],,enable_gtk=yes)
+AC_ARG_ENABLE(cairo,
+[  --enable-cairo         Determines whether to use Cairo canvas backend on platforms
+                          where Cairo is available.
+],,)
 AC_ARG_ENABLE(gnome-vfs,
 [  --disable-gnome-vfs     Determines whether to use the Gnome Virtual Filing
                           System on platforms where that VFS is available
@@ -205,6 +209,9 @@
 AC_ARG_WITH(system-freetype,
 [  --with-system-freetype  Use freetype already on system
 ],,)
+AC_ARG_WITH(system-cairo,
+[  --with-system-cairo  Use cairo already on system
+],,)
 AC_ARG_WITH(system-libwpd,
 [  --with-system-libwpd    Use libwpd already on system
 ],,)
@@ -587,6 +594,7 @@
 	solaris*)
 		test_x=yes
 		test_gtk=yes
+		build_cairo=yes
 		test_kde=yes
 		test_cups=yes
 		_os=SunOS
@@ -613,6 +621,7 @@
 	linux-gnu)
 		test_x=yes
 		test_gtk=yes
+		build_cairo=yes
 		test_kde=yes
 		test_cups=yes
 		if getconf GNU_LIBPTHREAD_VERSION |grep -q NPTL; then
@@ -634,11 +643,13 @@
 		test_x=no
 		test_cups=no
 		test_gtk=no
+		build_cairo=no
 		_os=Darwin
 		;;
 	freebsd*)
 		test_x=yes
 		test_gtk=yes
+		build_cairo=yes
 		test_kde=yes
 		test_cups=yes
 		AC_MSG_CHECKING([the FreeBSD operating system release])
@@ -670,6 +681,7 @@
 	netbsd)
 		test_x=yes
 		test_gtk=yes
+		build_cairo=yes
 		test_kde=yes
 		test_cups=no
 		PTHREAD_CFLAGS="-pthread"
@@ -2473,6 +2485,25 @@
 AC_SUBST(FREETYPE_LIBS)
 
 dnl ===================================================================
+dnl Check for system cairo
+dnl ===================================================================
+AC_MSG_CHECKING([which cairo to use])
+if test -n "$with_system_cairo" -o -n "$with_system_libs" && \
+	test "$with_system_cairo" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_CAIRO=YES
+    PKG_CHECK_MODULES( CAIRO, cairo >= 0.5.2 )
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_CAIRO=NO
+    BUILD_TYPE="$BUILD_TYPE CAIRO"
+fi
+AC_SUBST(SYSTEM_CAIRO)
+AC_SUBST(CAIRO_CFLAGS)
+AC_SUBST(CAIRO_LIBS)
+
+
+dnl ===================================================================
 dnl Check for system libwpd
 dnl ===================================================================
 AC_MSG_CHECKING([which libwpd to use])
@@ -3770,6 +3801,25 @@
 AC_SUBST(GTK_LIBS)
 
 dnl ===================================================================
+dnl Check whether the Cairo libraries are available.
+dnl ===================================================================
+
+ENABLE_CAIRO=""
+
+if test  "$build_cairo" = "yes"; then
+
+    AC_MSG_CHECKING([whether to enable the Cairo canvas backend])
+    if test "x$enable_cairo" != "xno" ; then
+        ENABLE_CAIRO="TRUE"
+       AC_MSG_RESULT([yes])
+    else
+       AC_MSG_RESULT([no])
+    fi
+fi
+
+AC_SUBST(ENABLE_CAIRO)
+
+dnl ===================================================================
 dnl Check whether the Qt and KDE libraries are available.
 dnl ===================================================================
 
