diff -up ../src680-m152-orig/canvas/source/cairo/cairo_cairo.cxx canvas/source/cairo/cairo_cairo.cxx
--- ../src680-m152-orig/canvas/source/cairo/cairo_cairo.cxx	2006-02-17 14:31:48.000000000 +0100
+++ canvas/source/cairo/cairo_cairo.cxx	2006-02-22 15:26:16.000000000 +0100
@@ -80,4 +80,13 @@ namespace cairo
 	{
 		cairo_xlib_surface_set_size( mpSurface, width, height );
 	}
+
+	int
+	Surface::getDepth()
+	{
+		if( mpRenderFormat )
+			return ( ( XRenderPictFormat * ) mpRenderFormat )->depth;
+
+		return -1;
+	}
 }
diff -up ../src680-m152-orig/canvas/source/cairo/cairo_cairo.hxx canvas/source/cairo/cairo_cairo.hxx
--- ../src680-m152-orig/canvas/source/cairo/cairo_cairo.hxx	2006-02-17 14:31:48.000000000 +0100
+++ canvas/source/cairo/cairo_cairo.hxx	2006-02-22 15:25:27.000000000 +0100
@@ -37,6 +37,7 @@ namespace cairo {
 			  mpDisplay( NULL ),
 			  mhDrawable( 0 ),
 			  mpSysData( NULL ),
+			  mpRenderFormat( NULL ),
 			  mbFreePixmap( false ),
 			  mnRefCount( 1 )
 		{
@@ -79,6 +80,8 @@ namespace cairo {
 
 		// use only for surfaces made on X Drawables
 		void Resize( int width, int height );
+
+		int getDepth();
 	};
 }
 
diff -up ../src680-m152-orig/canvas/source/cairo/cairo_canvashelper_text.cxx canvas/source/cairo/cairo_canvashelper_text.cxx
--- ../src680-m152-orig/canvas/source/cairo/cairo_canvashelper_text.cxx	2006-02-17 14:31:33.000000000 +0100
+++ canvas/source/cairo/cairo_canvashelper_text.cxx	2006-02-22 16:35:34.000000000 +0100
@@ -83,7 +83,7 @@ namespace cairocanvas
 		aSystemGraphicsData.hDrawable = pSurface->getPixmap();
 		aSystemGraphicsData.pRenderFormat = pSurface->getRenderFormat();
 
-		return new VirtualDevice( &aSystemGraphicsData, 24 );
+		return new VirtualDevice( &aSystemGraphicsData, pSurface->getDepth() );
 	}
 
 	static bool
diff -up ../src680-m152-orig/canvas/source/cairo/cairo_helper.cxx canvas/source/cairo/cairo_helper.cxx
--- ../src680-m152-orig/canvas/source/cairo/cairo_helper.cxx	2006-02-17 14:31:34.000000000 +0100
+++ canvas/source/cairo/cairo_helper.cxx	2006-02-07 14:51:30.000000000 +0100
@@ -129,7 +129,7 @@ cairoHelperGetSurface( const void* pSysD
 	cairo_surface_set_device_offset( pSurface, x, y );
     }
 
-    //XSynchronize( (Display*) pSysData->pDisplay, TRUE );
+    XSynchronize( (Display*) pSysData->pDisplay, TRUE );
 
     return pSurface;
 }
