Index: vcl/unx/inc/salgdi.h
===================================================================
RCS file: /cvs/gsl/vcl/unx/inc/salgdi.h,v
retrieving revision 1.33.54.1
diff -u -p -r1.33.54.1 salgdi.h
--- vcl/unx/inc/salgdi.h	14 Dec 2005 15:36:39 -0000	1.33.54.1
+++ vcl/unx/inc/salgdi.h	10 Jan 2006 10:09:00 -0000
@@ -199,7 +199,7 @@ public:
     virtual				~X11SalGraphics();
 
             void            Init( SalFrame *pFrame, Drawable aDrawable );
-            void            Init( X11SalVirtualDevice *pVirtualDevice );
+            void            Init( X11SalVirtualDevice *pVirtualDevice, SalColormap* pColormap = NULL, bool bDeleteColormap = false );
             void            Init( class ImplSalPrinterData *pPrinter );
             void            DeInit();
 
Index: vcl/unx/source/gdi/salvd.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/salvd.cxx,v
retrieving revision 1.10.126.1
diff -u -p -r1.10.126.1 salvd.cxx
--- vcl/unx/source/gdi/salvd.cxx	14 Dec 2005 15:36:40 -0000	1.10.126.1
+++ vcl/unx/source/gdi/salvd.cxx	10 Jan 2006 10:09:00 -0000
@@ -33,6 +33,10 @@
  *
  ************************************************************************/
 
+#include <prex.h>
+#include <X11/extensions/Xrender.h>
+#include <postx.h>
+
 #include <salunx.h>
 
 #ifndef _SV_SALDATA_HXX
@@ -89,13 +93,20 @@ void X11SalInstance::DestroyVirtualDevic
 
 // -=-= SalGraphicsData =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 // -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-void X11SalGraphics::Init( X11SalVirtualDevice *pDevice )
+void X11SalGraphics::Init( X11SalVirtualDevice *pDevice, SalColormap* pColormap, bool bDeleteColormap )
 {
 	SalDisplay *pDisplay  = pDevice->GetDisplay();
 	
 	int nVisualDepth = pDisplay->GetColormap().GetVisual()->GetDepth();
 	int nDeviceDepth = pDevice->GetDepth();
 
+	if( pColormap )
+    {
+		m_pColormap = pColormap;
+		if( bDeleteColormap )
+			m_pDeleteColormap = m_pColormap;
+	}
+	else
 	if( nDeviceDepth == nVisualDepth )
 		m_pColormap = &pDisplay->GetColormap();
 	else 
@@ -120,12 +131,22 @@ BOOL X11SalVirtualDevice::Init( SalDispl
                                 long nDX, long nDY,
 								USHORT nBitCount,
 								Pixmap hDrawable,
-								void* pRenderFormat )
+								void* pRenderFormatVoid )
 {
+	SalColormap* pColormap = NULL;
+	bool bDeleteColormap = false;
+
     pDisplay_               = pDisplay;
     pGraphics_				= new X11SalGraphics();
-	if( pRenderFormat )
+	if( pRenderFormatVoid ) {
+		XRenderPictFormat *pRenderFormat = ( XRenderPictFormat* )pRenderFormatVoid;
 		pGraphics_->SetXRenderFormat( pRenderFormat );
+		if( pRenderFormat->colormap )
+			pColormap = new SalColormap( pDisplay, pRenderFormat->colormap );
+		else
+			pColormap = new SalColormap( nBitCount );
+ 		bDeleteColormap = true;
+	}
     pGraphics_->SetLayout( 0 ); // by default no! mirroring for VirtualDevices, can be enabled with EnableRTL()
 	nDX_                    = nDX;
 	nDY_                    = nDY;
@@ -142,7 +163,7 @@ BOOL X11SalVirtualDevice::Init( SalDispl
 		bExternPixmap_ = TRUE;
 	}
 
-    pGraphics_->Init( this );
+    pGraphics_->Init( this, pColormap, bDeleteColormap );
 
 	return hDrawable_ != None ? TRUE : FALSE;
 }
