--- canvas/source/factory/cf_service.cxx	2005-04-18 11:08:12 +02:00
+++ canvas/source/factory/cf_service.cxx	2005-06-10 16:13:06 +02:00
@@ -198,12 +198,13 @@
     // append the usual preferred ones:
     sal_Int32 pos = m_services.getLength();
 #if defined WNT
-    m_services.realloc( pos + 3 );
+    m_services.realloc( pos + 4 );
     m_services[ pos++ ] = OUSTR("com.sun.star.rendering.DXCanvas");
 #else
-    m_services.realloc( pos + 2 );
+    m_services.realloc( pos + 3 );
 #endif
     m_services[ pos++ ] = OUSTR("com.sun.star.rendering.JavaCanvas");
+    m_services[ pos++ ] = OUSTR("com.sun.star.rendering.CairoCanvas");
     m_services[ pos   ] = OUSTR("com.sun.star.rendering.VCLCanvas");
 }
 
