--- canvas/prj/build.lst	2005-07-27 16:55:25.000000000 +0200
+++ canvas/prj/build.lst	2005-07-27 12:54:50.000000000 +0200
@@ -1,6 +1,7 @@
-uc	canvas	:	javaunohelper comphelper cppuhelper offuh unoil tools goodies vcl basegfx NULL
+uc	canvas	:	javaunohelper comphelper cppuhelper offuh unoil tools goodies vcl basegfx cairo NULL
 uc	canvas											 usr1	-	all	uc_mkout NULL
 uc	canvas\source\tools								 nmake	-	all uc_tools NULL
 uc	canvas\source\vcl								 nmake	-	all	uc_vcl uc_tools NULL
+uc	canvas\source\cairo								 nmake	-	all	uc_cairo uc_tools NULL
 uc	canvas\source\java								 nmake	-	all uc_java NULL
 uc	canvas\source\factory							 nmake	-	all uc_factory NULL
