--- /dev/null	
+++ canvas/source/cairo/cairo_helper.hxx	
@@ -0,0 +1,10 @@
+int   cairoHelperGetDefaultScreen( void* display );
+void* cairoHelperGetGlitzDrawable( const void* pSysData, int width, int height );
+void* cairoHelperGetGlitzSurface( const void* pSysData, void *drawable, int x, int y, int width, int height );
+void* cairoHelperGetSurface( const void* pSysData, int x, int y, int width, int height );
+void  cairoHelperFlush( const void* pSysData );
+void* cairoHelperGetSurface( const void* pSysData, const void *pBmpData, int width, int height );
+void* cairoHelperGetDisplay( const void* pSysData );
+long  cairoHelperGetWindow( const void* pSysData );
+void* cairoHelperGetVisual( const void* pSysData );
+int   cairoHelperGetDepth( const void* pSysData );
