Index: officecfg/registry/data/org/openoffice/VCL.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/VCL.xcu,v
retrieving revision 1.42
diff -u -p -u -r1.42 VCL.xcu
--- officecfg/registry/data/org/openoffice/VCL.xcu	21 Jul 2005 14:22:30 -0000	1.42
+++ officecfg/registry/data/org/openoffice/VCL.xcu	1 Sep 2005 10:29:42 -0000
@@ -62,6 +62,11 @@
 <!DOCTYPE oor:component-data SYSTEM "../../../component-update.dtd">
 <oor:component-data oor:name="VCL" oor:package="org.openoffice" xmlns:oor="http://openoffice.org/2001/registry" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <node oor:name="Settings">
+    <node oor:name="Canvas" oor:op="replace">
+      <prop oor:name="PreferredServices" oor:type="xs:string" oor:op="replace">
+        <value>com.sun.star.rendering.VCLCanvas</value>
+      </prop>
+    </node>
     <node oor:name="Accessibility" oor:op="replace">
       <prop oor:name="EnableATToolSupport" oor:type="xs:string" oor:op="replace">
         <value>false</value>

