--- scp/inc/gid.inc
+++ scp/inc/gid.inc
@@ -2833,7 +2833,8 @@
 #define GID_RVP_FILE_LIB_REMOTE_VCL                 gid_Rvp_File_Lib_Remote_VCL
 #define GID_RVP_FILE_LIB_REMOTE_TL                  gid_Rvp_File_Lib_Remote_TL
 #define GID_RVP_FILE_LIB_VCL                        gid_Rvp_File_Lib_Vcl
+#define GID_FILE_LIB_VCLPLUG_GEN		    gid_File_Lib_Vclplug_Gen
 #define GID_RVP_FILE_LIB_IMR                        gid_Rvp_File_Lib_Imr
 #define GID_RVP_FILE_LIB_TL                         gid_Rvp_File_Lib_Tl
 #define GID_RVP_FILE_LIB_PORTAL                     gid_Rvp_File_Lib_Portal
--- scp/source/global/setupzip.scp
+++ scp/source/global/setupzip.scp
@@ -66,6 +66,9 @@
 
 STD_SETUPZIP_LIB_FILE( GID_FILE_LIB_VCL, vcl )
 STD_SETUPZIP_LIB_FILE( GID_FILE_LIB_XCR, xcr)
+#ifdef UNX
+STD_SETUPZIP_LIB_FILE( GID_FILE_LIB_VCLPLUG_GEN, vclplug_gen )
+#endif
 
 File GID_FILE_LIB_LOCALEDATA_EN
 	TXT_FILE_BODY;
@@ -453,14 +456,7 @@
 
 #if defined(UNX)
 
-File Gid_FILE_BIN_GETSTYLEGNOME
-		BIN_FILE_BODY;
-		Name			= "getstyle-gnome";
-		Dir				= GCFG_BINARY_DIR;
-		Styles			= (PACKED, SETUPZIP);
-End
-
 File GID_FILE_BINARY_SETUP
 	BIN_FILE_BODY;
     Name            = "setup.bin";
--- scp/source/office/files.scp
+++ scp/source/office/files.scp
@@ -1013,17 +1013,6 @@
 #endif
 #endif
 
-#ifdef UNX
-
-File GID_FILE_MSGBOX_GNOME
-        BIN_FILE_BODY;
-        Styles          = (PACKED);
-        Dir             = GID_DIR_PROGRAM;
-        Name            = "msgbox-gnome";
-End
-
-#endif
-
 // Helpfiles for version src640
 
 File GID_FILE_HELP_HELP_ZIP
@@ -2602,6 +2591,8 @@
         Styles          = (PACKED);
 End
 
+STD_LIB_FILE( GID_FILE_LIB_VCLPLUG_GTK, vclplug_gtk )
+
 #ifdef OSL_PRODUCT
 
 STD_FONTUNX_FILE( GID_FILE_FNT_OPENSYMB, opens___.ttf, OpenSymbol)
