Index: vcl/unx/gtk/app/makefile.mk
===================================================================
RCS file: vcl/unx/gtk/app/makefile.mk
diff -N vcl/unx/gtk/app/makefile.mk
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/gtk/app/makefile.mk	12 Dec 2003 16:52:35 -0000	1.1.2.3
@@ -0,0 +1,102 @@
+#*************************************************************************
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to the terms of
+#   either of the following licenses
+#
+#          - GNU Lesser General Public License Version 2.1
+#          - Sun Industry Standards Source License Version 1.1
+#
+#   Sun Microsystems Inc., October, 2000
+#
+#   GNU Lesser General Public License Version 2.1
+#   =============================================
+#   Copyright 2000 by Sun Microsystems, Inc.
+#   901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License version 2.1, as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#   MA  02111-1307  USA
+#
+#
+#   Sun Industry Standards Source License Version 1.1
+#   =================================================
+#   The contents of this file are subject to the Sun Industry Standards
+#   Source License Version 1.1 (the "License"); You may not use this file
+#   except in compliance with the License. You may obtain a copy of the
+#   License at http://www.openoffice.org/license.html.
+#
+#   Software provided under this License is provided on an "AS IS" basis,
+#   WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+#   WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+#   MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+#   See the License for the specific provisions governing your rights and
+#   obligations concerning the Software.
+#
+#   The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+#
+#   Copyright: 2000 by Sun Microsystems, Inc.
+#
+#   All Rights Reserved.
+#
+#   Contributor(s): _______________________________________
+#
+#
+#
+#*************************************************************************
+
+PRJ=..$/..$/..
+
+PRJNAME=vcl
+TARGET=gtkapp
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  svpre.mk
+.INCLUDE :  settings.mk
+.INCLUDE :  sv.mk
+
+# --- Files --------------------------------------------------------
+
+.IF "$(GUIBASE)"!="unx"
+
+dummy:
+	@echo "Nothing to build for GUIBASE $(GUIBASE)"
+
+.ELSE		# "$(GUIBASE)"!="unx"
+
+CFLAGS+=`pkg-config --cflags gtk+-2.0`
+
+SLOFILES=\
+			$(SLO)$/gtkdata.obj	\
+			$(SLO)$/gtkinst.obj	\
+			$(SLO)$/gtksys.obj
+
+.IF "$(WITH_LIBSN)"=="YES"
+CDEFS+=-DHAVE_LIBSN
+CFLAGS+=$(LIBSN_CFLAGS)
+.ENDIF
+
+.ENDIF		# "$(GUIBASE)"!="unx"
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE :  target.mk
+
+.INCLUDE :  $(PRJ)$/util$/target.pmk
Index: vcl/unx/gtk/window/makefile.mk
===================================================================
RCS file: vcl/unx/gtk/window/makefile.mk
diff -N vcl/unx/gtk/window/makefile.mk
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/gtk/window/makefile.mk	19 Jan 2004 18:55:29 -0000	1.1.2.2
@@ -0,0 +1,101 @@
+#*************************************************************************
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to the terms of
+#   either of the following licenses
+#
+#          - GNU Lesser General Public License Version 2.1
+#          - Sun Industry Standards Source License Version 1.1
+#
+#   Sun Microsystems Inc., October, 2000
+#
+#   GNU Lesser General Public License Version 2.1
+#   =============================================
+#   Copyright 2000 by Sun Microsystems, Inc.
+#   901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License version 2.1, as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#   MA  02111-1307  USA
+#
+#
+#   Sun Industry Standards Source License Version 1.1
+#   =================================================
+#   The contents of this file are subject to the Sun Industry Standards
+#   Source License Version 1.1 (the "License"); You may not use this file
+#   except in compliance with the License. You may obtain a copy of the
+#   License at http://www.openoffice.org/license.html.
+#
+#   Software provided under this License is provided on an "AS IS" basis,
+#   WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+#   WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+#   MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+#   See the License for the specific provisions governing your rights and
+#   obligations concerning the Software.
+#
+#   The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+#
+#   Copyright: 2000 by Sun Microsystems, Inc.
+#
+#   All Rights Reserved.
+#
+#   Contributor(s): _______________________________________
+#
+#
+#
+#*************************************************************************
+
+PRJ=..$/..$/..
+
+PRJNAME=vcl
+TARGET=gtkwin
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  svpre.mk
+.INCLUDE :  settings.mk
+.INCLUDE :  sv.mk
+
+# --- Files --------------------------------------------------------
+
+.IF "$(GUIBASE)"!="unx"
+
+dummy:
+	@echo "Nothing to build for GUIBASE $(GUIBASE)"
+
+.ELSE		# "$(GUIBASE)"!="unx"
+
+CFLAGS+=`pkg-config --cflags gtk+-2.0`
+
+SLOFILES=\
+			$(SLO)$/gtkframe.obj				\
+			$(SLO)$/gtkobject.obj
+
+.IF "$(WITH_LIBSN)"=="YES"
+CDEFS+=-DHAVE_LIBSN
+CFLAGS+=$(LIBSN_CFLAGS)
+.ENDIF
+
+.ENDIF		# "$(GUIBASE)"!="unx"
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE :  target.mk
+
+.INCLUDE :  $(PRJ)$/util$/target.pmk
Index: vcl/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.50
retrieving revision 1.49.2.8
diff -u -p -u -r1.50 -r1.49.2.8
--- vcl/util/makefile.mk	6 Jan 2004 14:47:00 -0000	1.50
+++ vcl/util/makefile.mk	22 Jan 2004 14:00:44 -0000	1.49.2.8
@@ -257,9 +257,7 @@ SHL1STDLIBS += -framework Cocoa
 .ENDIF
 
 
-.IF "$(OS)"=="MACOSX"
-SHL1STDLIBS += -lXinerama
-.ENDIF
+SHL1STDLIBS += -lX11
 
 .ENDIF          # "$(GUI)"=="UNX"
 
@@ -272,11 +270,11 @@ LIB2FILES= \
             $(SLB)$/salwin.lib  \
             $(SLB)$/salgdi.lib  \
             $(SLB)$/salapp.lib
-SHL2TARGET=pure_x_$(UPD)$(DLLPOSTFIX)
+SHL2TARGET=vclplug_gen$(UPD)$(DLLPOSTFIX)
 SHL2IMPLIB=ipure_x
 SHL2LIBS=  $(LIB2TARGET)
 
-# libs for pure_x plugin
+# libs for generic plugin
 SHL2STDLIBS=\
 			$(VCLLIB)\
 			-lpsp$(VERSION)$(DLLPOSTFIX)\
@@ -310,11 +308,45 @@ SHL2STDLIBS += -lXext -lSM -lICE -lX11
 .ELSE           # "$(OS)"=="SOLARIS"
 .IF "$(CPU)" == "I"
 SHL2STDLIBS += -Wl,-Bstatic -lXinerama -Wl,-Bdynamic 
-SHL2STDLIBS += -lXext -lSM -lICE -lX11
 .ENDIF 			# "$(CPU)=="I"
+SHL2STDLIBS += -lXext -lSM -lICE -lX11
 .ENDIF          # "$(OS)"=="SOLARIS"
 
 .ENDIF          # "$(GUIBASE)"=="unx"
+
+# dummy plugin
+LIB3TARGET=$(SLB)$/idummy_plug_
+LIB3FILES= \
+            $(SLB)$/dapp.lib
+SHL3TARGET=vclplug_dummy$(UPD)$(DLLPOSTFIX)
+SHL3IMPLIB=idummy_plug_
+SHL3LIBS=  $(LIB3TARGET)
+
+# libs for dummy plugin
+SHL3STDLIBS=\
+			$(VCLLIB)\
+            $(SOTLIB)           \
+            $(UNOTOOLSLIB)      \
+            $(TOOLSLIB)         \
+            $(COMPHELPERLIB)	\
+            $(UCBHELPERLIB)     \
+            $(CPPUHELPERLIB)    \
+            $(CPPULIB)          \
+            $(VOSLIB)           \
+            $(SALLIB)
+
+# gtk plugin
+LIB4TARGET=$(SLB)$/igtk_plug_
+LIB4FILES=\
+			$(SLB)$/gtkapp.lib\
+			$(SLB)$/gtkwin.lib
+SHL4TARGET=vclplug_gtk$(UPD)$(DLLPOSTFIX)
+SHL4IMPLIB=igtk_plug_
+SHL4LIBS=$(LIB4TARGET)
+# libs for gtk plugin
+SHL4STDLIBS=`pkg-config --libs gtk+-2.0 gthread-2.0`
+SHL4STDLIBS+=-l$(SHL2TARGET)
+SHL4STDLIBS+=$(SHL3STDLIBS) -lX11
 
 .ENDIF # UNX
 
--- vcl/unx/source/gdi/makefile.mk	2004-01-24 15:41:05.000000000 +0000
+++ vcl/unx/source/gdi/makefile.mk.new	2004-01-29 03:36:19.000000000 +0000
@@ -90,7 +90,6 @@
 		$(SLO)$/charnames.obj	\
 		$(SLO)$/dtint.obj		\
 		$(SLO)$/kdeint.obj		\
-		$(SLO)$/gnomeint.obj	\
 		$(SLO)$/salcvt.obj		\
 		$(SLO)$/ansi1252.obj	\
 		$(SLO)$/xfont.obj		\
