Index: solenv/inc/tg_ext.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/tg_ext.mk,v
retrieving revision 1.48
retrieving revision 1.47.34.1
diff -u -p -u -r1.48 -r1.47.34.1
--- solenv/inc/tg_ext.mk	16 Dec 2003 11:38:45 -0000	1.48
+++ solenv/inc/tg_ext.mk	9 Dec 2003 13:59:07 -0000	1.47.34.1
@@ -313,9 +313,9 @@ create_patch : $(MISC)$/$(TARFILE_ROOTDI
 	@+-$(RM) $(NEW_PATCH_FILE_NAME).bak >& $(NULLDEV)
 #ignore returncode of 1 (indicates differences...)	
 # hard coded again to get the same directory level as before. quite ugly...
-	+-cd $(ROUT) && diff -rc misc$/$(TARFILE_ROOTDIR) misc$/build$/$(TARFILE_ROOTDIR) | $(PERL) $(SOLARENV)$/bin$/cleandiff.pl | tr -d "\015" > misc$/$(NEW_PATCH_FILE_NAME).tmp
+	+-cd $(PRJ)$/$(ROUT) && diff -rc misc$/$(TARFILE_ROOTDIR) misc$/build$/$(TARFILE_ROOTDIR) | $(PERL) $(SOLARENV)$/bin$/cleandiff.pl | tr -d "\015" > misc$/$(NEW_PATCH_FILE_NAME).tmp
 	@+-mv $(NEW_PATCH_FILE_NAME) $(NEW_PATCH_FILE_NAME).bak >& $(NULLDEV)
-	@+-mv $(MISC)$/$(NEW_PATCH_FILE_NAME).tmp $(NEW_PATCH_FILE_NAME) >& $(NULLDEV)
+	@+-mv $(MISC)$/$(NEW_PATCH_FILE_NAME).tmp $(PRJ)$/$(NEW_PATCH_FILE_NAME) >& $(NULLDEV)
 	@+echo still some problems with win32 generated patches...
 
 create_clean : $(PACKAGE_DIR)$/$(UNTAR_FLAG_FILE)
