Index: vcl/unx/dummy/dbmp.cxx
===================================================================
RCS file: vcl/unx/dummy/dbmp.cxx
diff -N vcl/unx/dummy/dbmp.cxx
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/dummy/dbmp.cxx	28 Nov 2003 13:43:12 -0000	1.1.2.1
@@ -0,0 +1,226 @@
+/*************************************************************************
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#include <dbmp.hxx>
+#include <bmpacc.hxx>
+
+// -------------
+// - SalBitmap -
+// -------------
+
+DummyBitmap::DummyBitmap() :
+	mpDIB( NULL )
+{
+}
+
+// -----------------------------------------------------------------------------
+
+DummyBitmap::~DummyBitmap()
+{
+	Destroy();
+}
+
+// -----------------------------------------------------------------------------
+
+// -----------------------------------------------------------------------------
+
+BitmapBuffer* DummyBitmap::ImplCreateDIB( const Size& rSize, USHORT nBitCount, const BitmapPalette& rPal )
+{
+	DBG_ASSERT( nBitCount == 1 || nBitCount == 4 || nBitCount == 8 || nBitCount == 24, "Unsupported BitCount!" );
+
+	BitmapBuffer* pDIB;
+
+	if( rSize.Width() && rSize.Height() )
+	{
+		pDIB = new BitmapBuffer;
+
+		if( pDIB )
+		{
+			const USHORT nColors = ( nBitCount <= 8 ) ? ( 1 << nBitCount ) : 0;
+			
+			pDIB->mnFormat = BMP_FORMAT_BOTTOM_UP;
+
+			switch( nBitCount )
+			{
+				case( 1 ): pDIB->mnFormat |= BMP_FORMAT_1BIT_MSB_PAL; break;
+				case( 4 ): pDIB->mnFormat |= BMP_FORMAT_4BIT_MSN_PAL; break;
+				case( 8 ): pDIB->mnFormat |= BMP_FORMAT_8BIT_PAL; break;
+			
+				default:
+					pDIB->mnFormat |= BMP_FORMAT_24BIT_TC_BGR;
+				break;
+			}
+
+			pDIB->mnWidth = rSize.Width();
+			pDIB->mnHeight = rSize.Height();
+			pDIB->mnScanlineSize = AlignedWidth4Bytes( pDIB->mnWidth * nBitCount );
+			pDIB->mnBitCount = nBitCount;
+			
+			if( nColors )
+			{
+				pDIB->maPalette = rPal;
+				pDIB->maPalette.SetEntryCount( nColors );
+			}
+
+			pDIB->mpBits = new BYTE[ pDIB->mnScanlineSize * pDIB->mnHeight ];
+			rtl_zeroMemory( pDIB->mpBits, pDIB->mnScanlineSize * pDIB->mnHeight );
+		}
+	}
+	else
+		pDIB = NULL;
+
+	return pDIB;
+}
+
+// -----------------------------------------------------------------------------
+
+bool DummyBitmap::Create( const Size& rSize, USHORT nBitCount, const BitmapPalette& rPal )
+{
+	Destroy();
+	mpDIB = ImplCreateDIB( rSize, nBitCount, rPal );
+
+	return( mpDIB != NULL );
+}
+
+// -----------------------------------------------------------------------------
+
+bool DummyBitmap::Create( const SalBitmap& rSSalBmp )
+{
+	Destroy();
+
+    const DummyBitmap& rSalBmp = static_cast<const DummyBitmap&>( rSSalBmp );
+
+	if( rSalBmp.mpDIB )
+	{
+		mpDIB = ImplCreateDIB( rSalBmp.GetSize(), rSalBmp.GetBitCount(), rSalBmp.mpDIB->maPalette );
+		
+		if( mpDIB )
+			memcpy( mpDIB->mpBits, rSalBmp.mpDIB->mpBits, mpDIB->mnScanlineSize * mpDIB->mnHeight );
+	}
+	return( ! rSalBmp.mpDIB  || 
+			( rSalBmp.mpDIB && ( mpDIB != NULL ) )
+            );
+}
+
+// -----------------------------------------------------------------------------
+
+bool DummyBitmap::Create( const SalBitmap& rSalBmp, SalGraphics* pGraphics )
+{
+	return FALSE;
+}
+
+// -----------------------------------------------------------------------------
+
+bool DummyBitmap::Create( const SalBitmap& rSalBmp, USHORT nNewBitCount )
+{
+	return FALSE;
+}
+
+// -----------------------------------------------------------------------------
+
+void DummyBitmap::Destroy()
+{
+	if( mpDIB )
+	{
+		delete[] mpDIB->mpBits;
+		delete mpDIB, mpDIB = NULL;
+	}
+}
+
+// -----------------------------------------------------------------------------
+
+Size DummyBitmap::GetSize() const
+{
+	Size aSize;
+
+	if( mpDIB )
+		aSize.Width() = mpDIB->mnWidth, aSize.Height() = mpDIB->mnHeight;
+	
+	return aSize;
+}
+
+// -----------------------------------------------------------------------------
+
+USHORT DummyBitmap::GetBitCount() const
+{
+	USHORT nBitCount;
+
+	if( mpDIB )
+		nBitCount = mpDIB->mnBitCount;
+	else
+		nBitCount = 0;
+
+	return nBitCount;
+}
+
+// -----------------------------------------------------------------------------
+
+BitmapBuffer* DummyBitmap::AcquireBuffer( bool bReadOnly )
+{
+	return mpDIB;
+}
+
+// -----------------------------------------------------------------------------
+
+void DummyBitmap::ReleaseBuffer( BitmapBuffer* pBuffer, bool bReadOnly )
+{
+}
+
Index: vcl/unx/dummy/dbmp.hxx
===================================================================
RCS file: vcl/unx/dummy/dbmp.hxx
diff -N vcl/unx/dummy/dbmp.hxx
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/dummy/dbmp.hxx	28 Nov 2003 13:43:12 -0000	1.1.2.1
@@ -0,0 +1,110 @@
+/*************************************************************************
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#ifndef _SV_DBMP_HXX
+#define _SV_DBMP_HXX
+
+#ifndef _SV_SALBMP_HXX
+#include <salbmp.hxx>
+#endif
+
+struct	BitmapBuffer;
+class	BitmapPalette;
+class	SalGraphics;
+
+// -------------
+// - SalBitmap -
+// -------------
+
+class DummyBitmap : public SalBitmap
+{
+private:
+
+	static BitmapBuffer*		ImplCreateDIB( const Size& rSize,
+											   USHORT nBitCount, 
+											   const BitmapPalette& rPal );
+	BitmapBuffer*	mpDIB;
+public:
+
+    DummyBitmap();
+    virtual ~DummyBitmap();
+
+    // overload pure virtual methods
+	virtual bool			Create( const Size& rSize, 
+                            USHORT nBitCount, 
+							const BitmapPalette& rPal );
+	virtual bool			Create( const SalBitmap& rSalBmp );
+	virtual bool			Create( const SalBitmap& rSalBmp, 
+                                    SalGraphics* pGraphics );
+	virtual bool			Create( const SalBitmap& rSalBmp,
+                                    USHORT nNewBitCount );
+						
+	virtual void			Destroy();
+						
+	virtual Size			GetSize() const;
+	virtual USHORT			GetBitCount() const;
+						
+	virtual BitmapBuffer*	AcquireBuffer( bool bReadOnly );
+	virtual void			ReleaseBuffer( BitmapBuffer* pBuffer, bool bReadOnly );
+};
+
+#endif // _SV_DBMP_HXX
+
Index: vcl/unx/dummy/dinst.cxx
===================================================================
RCS file: vcl/unx/dummy/dinst.cxx
diff -N vcl/unx/dummy/dinst.cxx
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/dummy/dinst.cxx	28 Nov 2003 13:43:13 -0000	1.1.2.1
@@ -0,0 +1,269 @@
+/*************************************************************************
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to the terms of
+ *  either of the following licenses
+ *
+ *         - GNU Lesser General Public License Version 2.1
+ *         - Sun Industry Standards Source License Version 1.1
+ *
+ *  Sun Microsystems Inc., October, 2000
+ *
+ *  GNU Lesser General Public License Version 2.1
+ *  =============================================
+ *  Copyright 2000 by Sun Microsystems, Inc.
+ *  901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License version 2.1, as published by the Free Software Foundation.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *  MA  02111-1307  USA
+ *
+ *
+ *  Sun Industry Standards Source License Version 1.1
+ *  =================================================
+ *  The contents of this file are subject to the Sun Industry Standards
+ *  Source License Version 1.1 (the "License"); You may not use this file
+ *  except in compliance with the License. You may obtain a copy of the
+ *  License at http://www.openoffice.org/license.html.
+ *
+ *  Software provided under this License is provided on an "AS IS" basis,
+ *  WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+ *  WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+ *  MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+ *  See the License for the specific provisions governing your rights and
+ *  obligations concerning the Software.
+ *
+ *  The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+ *
+ *  Copyright: 2000 by Sun Microsystems, Inc.
+ *
+ *  All Rights Reserved.
+ *
+ *  Contributor(s): _______________________________________
+ *
+ *
+ ************************************************************************/
+
+#include <salinst.hxx>
+#include <dbmp.hxx>
+
+class DummyInstance : public SalInstance
+{
+public:
+    DummyInstance() {}
+    virtual ~DummyInstance();
+
+    virtual SalFrame*      	CreateChildFrame( SystemParentData* pParent, ULONG nStyle );
+    virtual SalFrame*      	CreateFrame( SalFrame* pParent, ULONG nStyle );
+    virtual void				DestroyFrame( SalFrame* pFrame );
+    virtual SalObject*			CreateObject( SalFrame* pParent );
+    virtual void				DestroyObject( SalObject* pObject );
+    virtual SalVirtualDevice*	CreateVirtualDevice( SalGraphics* pGraphics,
+                                                     long nDX, long nDY,
+                                                     USHORT nBitCount );
+    virtual void				DestroyVirtualDevice( SalVirtualDevice* pDevice );
+
+    virtual SalInfoPrinter*	CreateInfoPrinter( SalPrinterQueueInfo* pQueueInfo,
+                                               ImplJobSetup* pSetupData );
+    virtual void				DestroyInfoPrinter( SalInfoPrinter* pPrinter );
+    virtual SalPrinter*		CreatePrinter( SalInfoPrinter* pInfoPrinter );
+    virtual void				DestroyPrinter( SalPrinter* pPrinter );
+
+    virtual void				GetPrinterQueueInfo( ImplPrnQueueList* pList );
+    virtual void				GetPrinterQueueState( SalPrinterQueueInfo* pInfo );
+    virtual void				DeletePrinterQueueInfo( SalPrinterQueueInfo* pInfo );
+    virtual String             GetDefaultPrinter();
+
+    virtual SalSound*			CreateSalSound();
+    virtual SalTimer*			CreateSalTimer();
+    virtual SalOpenGL*			CreateSalOpenGL( SalGraphics* pGraphics );
+    virtual SalI18NImeStatus*	CreateI18NImeStatus();
+    virtual SalSystem*			CreateSalSystem();
+    virtual SalBitmap*			CreateSalBitmap();
+    virtual vos::IMutex*		GetYieldMutex();
+    virtual ULONG				ReleaseYieldMutex();
+    virtual void				AcquireYieldMutex( ULONG nCount );
+    virtual void				Yield( BOOL bWait );
+    virtual bool				AnyInput( USHORT nType );
+
+                            // Menues
+    virtual SalMenu*        CreateMenu( BOOL bMenuBar );
+    virtual void            DestroyMenu( SalMenu* pMenu);
+    virtual SalMenuItem*    CreateMenuItem( const SalItemParams* pItemData );
+    virtual void            DestroyMenuItem( SalMenuItem* pItem );
+    virtual void*				GetConnectionIdentifier( ConnectionIdentifierType& rReturnedType, int& rReturnedBytes );
+};
+
+// plugin factory function
+extern "C" 
+{
+    SalInstance* create_SalInstance()
+    {
+        return new DummyInstance();
+    }
+}
+
+
+DummyInstance::~DummyInstance()
+{
+}
+
+SalFrame*			DummyInstance::CreateChildFrame( SystemParentData*, ULONG )
+{
+    return NULL;
+}
+
+SalFrame*			DummyInstance::CreateFrame( SalFrame*, ULONG )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyFrame( SalFrame* )
+{
+}
+
+SalObject*			DummyInstance::CreateObject( SalFrame* )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyObject( SalObject* )
+{
+}
+
+SalVirtualDevice*	DummyInstance::CreateVirtualDevice( SalGraphics*, long, long, USHORT )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyVirtualDevice( SalVirtualDevice* )
+{
+}
+
+SalInfoPrinter*		DummyInstance::CreateInfoPrinter( SalPrinterQueueInfo*, ImplJobSetup* )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyInfoPrinter( SalInfoPrinter* )
+{
+}
+
+SalPrinter*			DummyInstance::CreatePrinter( SalInfoPrinter* )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyPrinter( SalPrinter* )
+{
+}
+
+void				DummyInstance::GetPrinterQueueInfo( ImplPrnQueueList* )
+{
+}
+
+void				DummyInstance::GetPrinterQueueState( SalPrinterQueueInfo* )
+{
+}
+
+void				DummyInstance::DeletePrinterQueueInfo( SalPrinterQueueInfo* )
+{
+}
+
+String				DummyInstance::GetDefaultPrinter()
+{
+    return String();
+}
+
+SalSound*			DummyInstance::CreateSalSound()
+{
+    return NULL;
+}
+
+SalTimer*			DummyInstance::CreateSalTimer()
+{
+    return NULL;
+}
+
+SalOpenGL*			DummyInstance::CreateSalOpenGL( SalGraphics* )
+{
+    return NULL;
+}
+
+SalI18NImeStatus*	DummyInstance::CreateI18NImeStatus()
+{
+    return NULL;
+}
+
+SalSystem*			DummyInstance::CreateSalSystem()
+{
+    return NULL;
+}
+
+SalBitmap*			DummyInstance::CreateSalBitmap()
+{
+    return new DummyBitmap();
+}
+
+vos::IMutex*		DummyInstance::GetYieldMutex()
+{
+    return NULL;
+}
+
+ULONG				DummyInstance::ReleaseYieldMutex()
+{
+    return 0;
+}
+
+void				DummyInstance::AcquireYieldMutex( ULONG nCount )
+{
+}
+
+void				DummyInstance::Yield( BOOL )
+{
+}
+
+bool				DummyInstance::AnyInput( USHORT )
+{
+    return false;
+}
+
+SalMenu*			DummyInstance::CreateMenu( BOOL )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyMenu( SalMenu* )
+{
+}
+
+SalMenuItem*		DummyInstance::CreateMenuItem( const SalItemParams* )
+{
+    return NULL;
+}
+
+void				DummyInstance::DestroyMenuItem( SalMenuItem* )
+{
+}
+
+void*				DummyInstance::GetConnectionIdentifier( ConnectionIdentifierType& rReturnedType, int& rReturnedBytes )
+{
+    rReturnedType		= Blob;
+    rReturnedBytes		= 0;
+    return NULL;
+}
Index: vcl/unx/dummy/makefile.mk
===================================================================
RCS file: vcl/unx/dummy/makefile.mk
diff -N vcl/unx/dummy/makefile.mk
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/dummy/makefile.mk	28 Nov 2003 13:43:14 -0000	1.1.2.1
@@ -0,0 +1,94 @@
+#*************************************************************************
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to the terms of
+#   either of the following licenses
+#
+#          - GNU Lesser General Public License Version 2.1
+#          - Sun Industry Standards Source License Version 1.1
+#
+#   Sun Microsystems Inc., October, 2000
+#
+#   GNU Lesser General Public License Version 2.1
+#   =============================================
+#   Copyright 2000 by Sun Microsystems, Inc.
+#   901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License version 2.1, as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#   MA  02111-1307  USA
+#
+#
+#   Sun Industry Standards Source License Version 1.1
+#   =================================================
+#   The contents of this file are subject to the Sun Industry Standards
+#   Source License Version 1.1 (the "License"); You may not use this file
+#   except in compliance with the License. You may obtain a copy of the
+#   License at http://www.openoffice.org/license.html.
+#
+#   Software provided under this License is provided on an "AS IS" basis,
+#   WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+#   WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+#   MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+#   See the License for the specific provisions governing your rights and
+#   obligations concerning the Software.
+#
+#   The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+#
+#   Copyright: 2000 by Sun Microsystems, Inc.
+#
+#   All Rights Reserved.
+#
+#   Contributor(s): _______________________________________
+#
+#
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJNAME=vcl
+TARGET=dapp
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  svpre.mk
+.INCLUDE :  settings.mk
+.INCLUDE :  sv.mk
+
+# --- Files --------------------------------------------------------
+
+.IF "$(GUIBASE)"!="unx"
+
+dummy:
+	@echo "Nothing to build for GUIBASE $(GUIBASE)"
+
+.ELSE		# "$(GUIBASE)"!="unx"
+
+SLOFILES=\
+			$(SLO)$/dinst.obj\
+			$(SLO)$/dbmp.obj
+
+.ENDIF		# "$(GUIBASE)"!="unx"
+
+# --- Targets ------------------------------------------------------
+
+.INCLUDE :  target.mk
+
+.INCLUDE :  $(PRJ)$/util$/target.pmk
