Index: sbxmod.cxx
===================================================================
RCS file: /cvs/script/basic/source/classes/sbxmod.cxx,v
retrieving revision 1.38
diff -c -r1.38 sbxmod.cxx
*** basic/source/classes/sbxmod.cxx	3 Aug 2007 09:55:16 -0000	1.38
--- basic/source/classes/sbxmod.cxx	17 Aug 2007 08:22:19 -0000
***************
*** 704,710 ****
--- 704,714 ----
  		pINST = new SbiInstance( (StarBASIC*) GetParent() );
  
  		// Launcher problem
+         // i80726 The Find below will genarate an error in Testtool so we reset it unless there was one before already
+         BOOL bWasError = SbxBase::GetError() != 0;
  		SbxVariable* pMSOMacroRuntimeLibVar = Find( aMSOMacroRuntimeLibName, SbxCLASS_OBJECT );
+         if ( !bWasError && (SbxBase::GetError() == SbxERR_PROC_UNDEFINED) )
+             SbxBase::ResetError();
  		if( pMSOMacroRuntimeLibVar )
  		{
  			StarBASIC* pMSOMacroRuntimeLib = PTR_CAST(StarBASIC,pMSOMacroRuntimeLibVar);
