--- basic.orig/source/runtime/methods.cxx	2005-10-17 14:55:39.000000000 +0100
+++ basic/source/runtime/methods.cxx	2005-11-10 15:44:54.000000000 +0000
@@ -1977,6 +1977,18 @@ RTLFUNC(Time)
 	}
 }
 
+RTLFUNC(MilliTimer)
+{
+	TimeValue pTimeVal;
+	long milliSeconds =  -1;
+	if ( osl_getSystemTime( &pTimeVal ) )	
+	{
+		milliSeconds = pTimeVal.Seconds * 1000;
+		milliSeconds +=  ( pTimeVal.Nanosec / 1000000 );
+	}		
+	
+	rPar.Get(0)->PutLong( (long)milliSeconds );
+}
 RTLFUNC(Timer)
 {
 	Time aTime;
--- basic.orig/source/runtime/rtlproto.hxx	2005-09-07 22:40:58.000000000 +0100
+++ basic/source/runtime/rtlproto.hxx	2005-11-10 15:32:14.000000000 +0000
@@ -218,6 +218,7 @@ extern RTLFUNC(MonthName);
 extern RTLFUNC(Now);
 extern RTLFUNC(Second);
 extern RTLFUNC(Time);
+extern RTLFUNC(MilliTimer);
 extern RTLFUNC(Timer);
 extern RTLFUNC(Weekday);
 extern RTLFUNC(WeekdayName);
--- basic.orig/source/runtime/stdobj.cxx	2005-09-07 22:41:29.000000000 +0100
+++ basic/source/runtime/stdobj.cxx	2005-11-10 15:32:46.000000000 +0000
@@ -516,6 +516,7 @@ static Methods aMethods[] = {
   { "number",       SbxDOUBLE },
 { "Time",           SbxVARIANT,       _LFUNCTION,RTLNAME(Time)              },
 { "Timer",          SbxDATE,          _FUNCTION, RTLNAME(Timer)             },
+{ "MilliTimer",          SbxLONG,          _FUNCTION, RTLNAME(MilliTimer)             },
 { "TimeSerial",     SbxDATE,      3 | _FUNCTION, RTLNAME(TimeSerial)        },
   { "Hour",         SbxLONG },
   { "Minute",       SbxLONG },
