--- basic/source/sbx/sbxstr.cxx.orig	2008-07-22 13:53:07.000000000 +0800
+++ basic/source/sbx/sbxstr.cxx	2008-07-22 14:25:53.000000000 +0800
@@ -98,7 +98,7 @@ XubString ImpGetString( const SbxValues*
 			if( pVal )
 				aRes = pVal->GetString();
 			else if( p->pObj && p->pObj->IsFixed()
-					&& (p->pObj->GetType() & (SbxARRAY | SbxBYTE )) )
+					&& (p->pObj->GetType() == (SbxARRAY | SbxBYTE )) )
 			{
 				// convert byte array to string
 				SbxArray* pArr = PTR_CAST(SbxArray, p->pObj);
--- basic/source/sbx/sbxvalue.cxx.orig	2008-07-22 13:54:30.000000000 +0800
+++ basic/source/sbx/sbxvalue.cxx	2008-07-22 14:25:47.000000000 +0800
@@ -309,7 +309,7 @@ SbxValue& SbxValue::operator=( const Sbx
 		{
 			// string -> byte array
 			if( IsFixed() && (aData.eType == SbxOBJECT)
-				&& aData.pObj && ( aData.pObj->GetType() & (SbxARRAY | SbxBYTE) )		   	
+				&& aData.pObj && ( aData.pObj->GetType() == (SbxARRAY | SbxBYTE) )		   	
 				&& (r.aData.eType == SbxSTRING) )
 			{
 				String aStr = r.GetString();
@@ -319,7 +319,7 @@ SbxValue& SbxValue::operator=( const Sbx
 			}
 			// byte array -> string
 			if( r.IsFixed() && (r.aData.eType == SbxOBJECT)
-				&& r.aData.pObj && ( r.aData.pObj->GetType() & (SbxARRAY | SbxBYTE) )
+				&& r.aData.pObj && ( r.aData.pObj->GetType() == (SbxARRAY | SbxBYTE) )
 				&& (aData.eType == SbxSTRING) )
 			{
 				SbxBase* pObj = r.GetObject();
