--- /data4/sles/ooo-buildm165/ooo-build/build/src680-m165/sc/source/core/tool/address.cxx	2006-05-10 12:09:35.000000000 +0100
+++ sc/source/core/tool/address.cxx	2006-05-18 18:13:58.000000000 +0100
@@ -427,6 +427,13 @@ lcl_ConvertSingleRef( BOOL& bExternal, c
         if( !nBits )
             p = q;
     }
+    else // assume its "E:F" for full Column
+    {
+        nBits = SCA_VALID_ROW ;
+        nBits |= SCA_VALID_COLADDR;
+        nRow = MAXROW;
+        nRes |= nBits;
+    }
     if ( bNeedExtTab )
     {
         if ( (nRes & SCA_VALID_ROW) && (nRes & SCA_VALID_COL)
@@ -600,6 +607,9 @@ USHORT ScRange::Parse( const String& r, 
             aEnd = aStart;  // die Tab _muss_ gleich sein, so ist`s weniger Code
             if ( nRes2 = lcl_ConvertSingleRef( bExternal, p + nPos+ 1, pDoc, aEnd, rDetails ) )
             {
+                bool bColFormat1 = ( ( nRes1 & SCA_VALID_COLADDR ) == SCA_VALID_COLADDR );
+                bool bColFormat2 = ( ( nRes2 & SCA_VALID_COLADDR ) == SCA_VALID_COLADDR );
+
                 if ( bExternal && aStart.Tab() != aEnd.Tab() )
                     nRes2 &= ~SCA_VALID_TAB;    // #REF!
                 else
@@ -640,6 +650,18 @@ USHORT ScRange::Parse( const String& r, 
                             == ( SCA_TAB_ABSOLUTE | SCA_TAB_3D ))
                             && !(nRes2 & SCA_TAB_3D) )
                         nRes2 |= SCA_TAB_ABSOLUTE;
+
+                    // "E:F" is legal but "E1:F" is not
+                    if  ( bColFormat2 || bColFormat1 ) 
+                    {
+                    	if ( ( bColFormat2 && bColFormat1 ) ) 
+                                // make sure the whole column range
+                                // is in the range
+                        	aStart.SetRow(0);
+		
+                        else
+                            nRes2 &= ~SCA_VALID_COL; // raise error
+                    }
                 }
             }
             else
@@ -661,10 +673,15 @@ USHORT ScRange::ParseAny( const String& 
         SCA_VALID_TAB2;
     if ( (nRet & nValid) != nValid )
     {
+	// single address
         ScAddress aAdr;
         nRet = aAdr.Parse( r, pDoc, rDetails );
         if ( nRet & SCA_VALID )
+        {
             aStart = aEnd = aAdr;
+            if  ( ( nRet & SCA_VALID_COLADDR ) == SCA_VALID_COLADDR )
+                aStart.SetRow(0);
+        }
     }
     return nRet;
 }
--- /data4/sles/ooo-buildm165/ooo-build/build/src680-m165/sc/inc/address.hxx	2006-05-10 12:09:35.000000000 +0100
+++ sc/inc/address.hxx	2006-05-18 18:25:08.000000000 +0100
@@ -221,6 +221,7 @@ inline bool ValidColRowTab( SCCOL nCol, 
 #define SCA_VALID_ROW       0x0100
 #define SCA_VALID_COL       0x0200
 #define SCA_VALID_TAB       0x0400
+#define SCA_VALID_COLADDR   0x0800
 #define SCA_VALID_ROW2      0x1000
 #define SCA_VALID_COL2      0x2000
 #define SCA_VALID_TAB2      0x4000
