Index: extensions/source/propctrlr/formcontroller.cxx
===================================================================
RCS file: /cvs/util/extensions/source/propctrlr/formcontroller.cxx,v
retrieving revision 1.85
diff -u -p -u -r1.85 formcontroller.cxx
--- extensions/source/propctrlr/formcontroller.cxx	23 Mar 2005 11:56:05 -0000	1.85
+++ extensions/source/propctrlr/formcontroller.cxx	8 Sep 2005 15:25:06 -0000
@@ -1354,6 +1354,9 @@ class EventsNameReplace_Impl:
 
 									if	(	(pEvent->ScriptCode.getLength() > 0)
 										&&	(pEvent->ScriptType.getLength() > 0)
+#ifdef ENABLE_VBA
+										&& (pEvent->ScriptType.compareToAscii("VBAInterop") != 0 )
+#endif
 										)
 									{
 										::rtl::OUString sScriptType = pEvent->ScriptType;
@@ -1725,6 +1728,12 @@ class EventsNameReplace_Impl:
 						for (sal_uInt32 nI=0; nI<nScrEvts;nI++)
 						{
 							const ScriptEventDescriptor& rEvDe = pEvDes[nI];
+#ifdef ENABLE_VBA
+							if ( ( pEvDes[ nI ].ScriptType.compareToAscii("VBAInterop") == 0  ) )
+								// don't process or show "VBAInterop" bindings
+								continue; 
+#endif
+
 							if ( (aListenerClassName.Equals((const sal_Unicode*)rEvDe.ListenerType)
 								 || aListener.Equals((const sal_Unicode*)rEvDe.ListenerType) )
 								&& pMethods->equals(rEvDe.EventMethod))
