diff --git basic/source/comp/scanner.cxx basic/source/comp/scanner.cxx
index c7fb06c..49e512e 100644
--- basic/source/comp/scanner.cxx
+++ basic/source/comp/scanner.cxx
@@ -479,7 +479,17 @@ PrevLineCommentLbl:
 eoln:
 	if( nCol && *--pLine == '_' )
 	{
-		pLine = NULL; return NextSym();
+		pLine = NULL; 
+                bool bRes = NextSym();
+                if ( bVBASupportOn && aSym.GetBuffer()[0] == '.' )
+                {
+                // object _
+                //    .Method
+                // ^^^  <- spaces is legal in MSO VBA
+                    OSL_TRACE("*** resetting bSpaces***"); 
+                    bSpaces = FALSE; 
+                }
+                return bRes;
 	}
 	else
 	{
