--- /dev/null	
+++ offapi/org/openoffice/vba/XBorders.idl	
@@ -0,0 +1,27 @@
+#ifndef __org_openoffice_vba_XBorders_idl__
+#define __org_openoffice_vba_XBorders_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+#ifndef  __org_openoffice_vba_XBorder_idl__
+#include <org/openoffice/vba/XBorder.idl>
+#endif
+
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+//=============================================================================
+interface XBorders: ::org::openoffice::vba::XBorder
+{
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
