--- /dev/null	
+++ offapi/org/openoffice/vba/XWorksheet.idl	
@@ -0,0 +1,61 @@
+#ifndef __org_openoffice_vba_XWorksheet_idl__
+#define __org_openoffice_vba_XWorksheet_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+//=============================================================================
+
+interface XComments;
+interface XRange;
+interface XOutline;
+interface XWorksheet: com::sun::star::uno::XInterface
+{
+	[attribute] string Name;
+	[attribute] boolean Visible; 
+	[attribute, readonly] long StandardHeight;
+	[attribute, readonly] long StandardWidth;
+	[attribute, readonly] boolean ProtectionMode;
+	[attribute, readonly] boolean ProtectContents;
+	[attribute, readonly] boolean ProtectDrawingObjects;
+	[attribute, readonly] XRange UsedRange; 
+	[attribute, readonly] XWorksheet Next;
+	[attribute, readonly] XWorksheet Previous;
+	
+	void Activate();
+	void Calculate( );
+	void Select();
+	void Move([in] any Before,[in] any After );
+	void Copy([in] any Before,[in] any After );
+	void Paste([in] any Destination,[in] any Link);
+	void Delete( );
+	void Protect([in] any Password,[in] any DrawingObjects ,[in] any Contents,[in] any Scenarios,[in] any UserInterfaceOnly);
+	void Unprotect([in] any Password );
+	void CheckSpelling([in] any CustomDictionary,[in] any IgnoreUppercase, [in] any AlwaysSuggest,[in] any SpellingLang );
+	XRange Range([in] any Cell1, [in] any Cell2 );
+	any ChartObjects([in] any Index);
+	any PivotTables([in] any Index);
+	any Comments([in] any Index);
+	XOutline Outline();
+
+
+// FIXME: should prolly inherit from Range somehow...
+	XRange Cells([in] any RowIndex, [in] any ColumnIndex);
+	XRange Rows([in] any aIndex);
+	XRange Columns([in] any aIndex);
+
+	any Evaluate( [in] string Name);
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
