--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSheetType.idl	
@@ -0,0 +1,14 @@
+#ifndef __org_openoffice_vba_Excel_XlSheetType_idl__
+#define __org_openoffice_vba_Excel_XlSheetType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSheetType
+{
+        const long xlChart = -4109; //(&HFFFFEFF3)
+        const long xlDialogSheet = -4116; //(&HFFFFEFEC)
+        const long xlExcel4IntMacroSheet = 4;
+        const long xlExcel4MacroSheet = 3;
+        const long xlWorkSheet = -4167; //(&HFFFFEFB9)
+};
+}; }; }; };
+#endif
