--- /dev/null	2005-10-10 09:55:51.712408000 +0100
+++ offapi/org/openoffice/vba/XCollection.idl	2005-08-09 11:46:10.000000000 +0100
@@ -0,0 +1,28 @@
+#ifndef __org_openoffice_vba_XCollection_idl__
+#define __org_openoffice_vba_XCollection_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+//=============================================================================
+
+published interface XApplication;
+
+published interface XCollection : ::com::sun::star::uno::XInterface
+{
+	[attribute, readonly] long Count;
+	[attribute, readonly] any   Parent;
+	[attribute, readonly] long Creator;
+	[attribute, readonly] XApplication Application;
+
+	any  Item( [in] any Index );
+};
+
+}; }; };
+
+#endif
