--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlRoutingSlipStatus.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlRoutingSlipStatus_idl__
+#define __org_openoffice_vba_Excel_XlRoutingSlipStatus_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlRoutingSlipStatus
+{
+        const short xlNotYetRouted = 0;
+        const short xlRoutingComplete = 2;
+        const short xlRoutingInProgress = 1;
+};
+}; }; }; };
+#endif
