--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPivotFieldCalculation.idl	
@@ -0,0 +1,18 @@
+#ifndef __org_openoffice_vba_Excel_XlPivotFieldCalculation_idl__
+#define __org_openoffice_vba_Excel_XlPivotFieldCalculation_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPivotFieldCalculation
+{
+        const long xlDifferenceFrom = 2;
+        const long xlIndex = 9;
+        const long xlNoAdditionalCalculation = -4143; //(&HFFFFEFD1)
+        const long xlPercentDifferenceFrom = 4;
+        const long xlPercentOf = 3;
+        const long xlPercentOfColumn = 7;
+        const long xlPercentOfRow = 6;
+        const long xlPercentOfTotal = 8;
+        const long xlRunningTotal = 5;
+};
+}; }; }; };
+#endif
