--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSmartTagControlType.idl	
@@ -0,0 +1,23 @@
+#ifndef __org_openoffice_vba_Excel_XlSmartTagControlType_idl__
+#define __org_openoffice_vba_Excel_XlSmartTagControlType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSmartTagControlType
+{
+        const short xlSmartTagControlActiveX = 13;
+        const short xlSmartTagControlButton = 6;
+        const short xlSmartTagControlCheckbox = 9;
+        const short xlSmartTagControlCombo = 12;
+        const short xlSmartTagControlHelp = 3;
+        const short xlSmartTagControlHelpURL = 4;
+        const short xlSmartTagControlImage = 8;
+        const short xlSmartTagControlLabel = 7;
+        const short xlSmartTagControlLink = 2;
+        const short xlSmartTagControlListbox = 11;
+        const short xlSmartTagControlRadioGroup = 14;
+        const short xlSmartTagControlSeparator = 5;
+        const short xlSmartTagControlSmartTag = 1;
+        const short xlSmartTagControlTextbox = 10;
+};
+}; }; }; };
+#endif
