--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPhoneticAlignment.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlPhoneticAlignment_idl__
+#define __org_openoffice_vba_Excel_XlPhoneticAlignment_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPhoneticAlignment
+{
+        const short XlPhoneticAlignCenter = 2;
+        const short XlPhoneticAlignDistributed = 3;
+        const short XlPhoneticAlignLeft = 1;
+        const short XlPhoneticAlignNoControl = 0;
+};
+}; }; }; };
+#endif
