--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlBorderWeight.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlBorderWeight_idl__
+#define __org_openoffice_vba_Excel_XlBorderWeight_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlBorderWeight
+{
+        const long xlHairline = 1;
+        const long xlMedium = -4138; //(&HFFFFEFD6)
+        const long xlThick = 4;
+        const long xlThin = 2;
+};
+}; }; }; };
+#endif
