--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlFindLookIn.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlFindLookIn_idl__
+#define __org_openoffice_vba_Excel_XlFindLookIn_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlFindLookIn
+{
+        const long xlComments = -4144; //(&HFFFFEFD0)
+        const long xlFormulas = -4123; //(&HFFFFEFE5)
+        const long xlValues = -4163; //(&HFFFFEFBD)
+};
+}; }; }; };
+#endif
