--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlAutoFillType.idl	
@@ -0,0 +1,20 @@
+#ifndef __org_openoffice_vba_Excel_XlAutoFillType_idl__
+#define __org_openoffice_vba_Excel_XlAutoFillType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlAutoFillType
+{
+        const short xlFillCopy = 1;
+        const short xlFillDays = 5;
+        const short xlFillDefault = 0;
+        const short xlFillFormats = 3;
+        const short xlFillMonths = 7;
+        const short xlFillSeries = 2;
+        const short xlFillValues = 4;
+        const short xlFillWeekdays = 6;
+        const short xlFillYears = 8;
+        const short xlGrowthTrend = 10;
+        const short xlLinearTrend = 9;
+};
+}; }; }; };
+#endif
