--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlErrorChecks.idl	
@@ -0,0 +1,17 @@
+#ifndef __org_openoffice_vba_Excel_XlErrorChecks_idl__
+#define __org_openoffice_vba_Excel_XlErrorChecks_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlErrorChecks
+{
+        const short xlEmptyCellReferences = 7;
+        const short xlEvaluateToError = 1;
+        const short xlInconsistentFormula = 4;
+        const short xlListDataValidation = 8;
+        const short xlNumberAsText = 3;
+        const short xlOmittedCells = 5;
+        const short xlTextDate = 2;
+        const short xlUnlockedFormulaCells = 6;
+};
+}; }; }; };
+#endif
