--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlCalculationState.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlCalculationState_idl__
+#define __org_openoffice_vba_Excel_XlCalculationState_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlCalculationState
+{
+        const short xlCalculating = 1; 
+        const short xlDone = 0;
+        const short xlPending = 2;
+};
+}; }; }; };
+#endif
