--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlCmdType.idl	
@@ -0,0 +1,14 @@
+#ifndef __org_openoffice_vba_Excel_XlCmdType_idl__
+#define __org_openoffice_vba_Excel_XlCmdType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlCmdType
+{
+        const short xlCmdCube = 1;
+        const short xlCmdDefault = 4;
+        const short xlCmdList = 5;
+        const short xlCmdSql = 2;
+        const short xlCmdTable = 3;
+};
+}; }; }; };
+#endif
