--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlEditionOptionsOption.idl	
@@ -0,0 +1,17 @@
+#ifndef __org_openoffice_vba_Excel_XlEditionOptionsOption_idl__
+#define __org_openoffice_vba_Excel_XlEditionOptionsOption_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlEditionOptionsOption
+{
+        const short xlAutomaticUpdate = 4;
+        const short xlCancel = 1;
+        const short xlChangeAttributes = 6;
+        const short xlManualUpdate = 5;
+        const short xlOpenSource = 3;
+        const short xlSelect = 3;
+        const short xlSendPublisher = 2;
+        const short xlUpdateSubscriber = 2;
+};
+}; }; }; };
+#endif
