--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlEnableCancelKey.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlEnableCancelKey_idl__
+#define __org_openoffice_vba_Excel_XlEnableCancelKey_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlEnableCancelKey
+{
+        const short xlDisabled = 0;
+        const short xlErrorHandler = 2;
+        const short xlInterrupt = 1;
+};
+}; }; }; };
+#endif
