--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLink.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlLink_idl__
+#define __org_openoffice_vba_Excel_XlLink_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLink
+{
+        const short xlExcelLink = 1;
+        const short xlOLELink = 2;
+        const short xlPublishers = 5;
+        const short xlSubscribers = 6;
+};
+}; }; }; };
+#endif
