--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlFormControl.idl	
@@ -0,0 +1,19 @@
+#ifndef __org_openoffice_vba_Excel_XlFormControl_idl__
+#define __org_openoffice_vba_Excel_XlFormControl_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlFormControl
+{
+        const short xlButtonControl = 0;
+        const short xlCheckBox = 1;
+        const short xlDropDown = 2;
+        const short xlEditBox = 3;
+        const short xlGroupBox = 4;
+        const short xlLabel = 5;
+        const short xlListBox = 6;
+        const short xlOptionButton = 7;
+        const short xlSchollBar = 8;
+        const short xlSpinner = 9;
+};
+}; }; }; };
+#endif
