--- /dev/null	
+++ offapi/org/openoffice/vba/XWorksheets.idl	
@@ -0,0 +1,29 @@
+#ifndef __org_openoffice_vba_XWorksheets_idl__
+#define __org_openoffice_vba_XWorksheets_idl__
+                                                                                                                             
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+                                                                                                                             
+#ifndef __org_openoffice_vba_XCollection_idl__
+#include <org/openoffice/vba/XCollection.idl>
+#endif
+                                                                                                                             
+//=============================================================================
+                                                                                                                             
+module org {  module openoffice {  module vba {
+                                                                                                                             
+//=============================================================================
+                                                                                                                             
+interface XWorksheets : ::com::sun::star::uno::XInterface
+{
+	[attribute] any Visible; 
+        any  Add( [in] any Before, [in] any After, [in] any Count, [in] any Type );                                 
+        void Delete();
+	void PrintOut( [in] any From, [in] any To, [in] any Copies, [in] any Preview, [in] any ActivePrinter, [in] any PrintToFile, [in] any Collate, [in] any PrToFileName );
+};
+                                                                                                                             
+}; }; };
+                                                                                                                             
+#endif
+
