--- /dev/null	
+++ offapi/org/openoffice/vba/XComment.idl	
@@ -0,0 +1,38 @@
+#ifndef __org_openoffice_vba_XComment_idl__
+#define __org_openoffice_vba_XComment_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+#ifndef __com_sun_star_lang_IllegalArgumentException_idl__
+#include <com/sun/star/lang/IllegalArgumentException.idl>
+#endif
+//=============================================================================
+
+module org {  module openoffice {  module vba {
+
+interface XApplication;
+interface XRange;
+
+interface XComment: com::sun::star::uno::XInterface
+{
+	[attribute, readonly] XApplication Application;
+	[attribute] string Author;
+	[attribute, readonly] long Creator;
+	[attribute, readonly] XRange Parent;
+//	[attribute, readonly] Shape Shape;
+	[attribute] boolean Visible;
+
+	void Delete();
+	XComment Next();
+	XComment Previous();
+	string Text([in] any Text, [in] any Start, [in] any Overwrite);
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
