--- /dev/null	
+++ offapi/org/openoffice/vba/XGlobals.idl	
@@ -0,0 +1,34 @@
+#ifndef __org_openoffice_vba_XGlobals_idl__
+#define __org_openoffice_vba_XGlobals_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+module org {  module openoffice {  module vba { 
+
+interface XApplication;
+interface XWorkbook;
+interface XWorksheet;
+interface XWorksheets;
+interface XWorkbooks;
+
+interface XGlobals: com::sun::star::uno::XInterface
+{
+        // FIXME, need better way to expose globals
+	sequence< any > getGlobals();
+
+	[attribute, readonly] org::openoffice::vba::XApplication Application;
+	[attribute, readonly] org::openoffice::vba::XWorkbook ActiveWorkbook;
+	[attribute, readonly] org::openoffice::vba::XWorksheet ActiveSheet;
+	any WorkSheets( [in] any aIndex );
+	any WorkBooks( [in] any aIndex );
+	any Sheets( [in] any aIndex );
+	any Range( [in] any Cell1, [in] any Cell2 );
+};
+
+}; }; };
+
+#endif
+
+
