--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlAutoFilterOperator.idl	
@@ -0,0 +1,15 @@
+#ifndef __org_openoffice_vba_Excel_XlAutoFilterOperator_idl__
+#define __org_openoffice_vba_Excel_XlAutoFilterOperator_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlAutoFilterOperator
+{
+        const short xlAnd = 1;
+        const short xlBottom10Items = 4;
+        const short xlBottom10Percent = 6;
+        const short xlOr = 2;
+        const short xlTop10Items = 3;
+        const short xlTop10Percent = 5;
+};
+}; }; }; };
+#endif
