--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLayoutFormType.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlLayoutFormType_idl__
+#define __org_openoffice_vba_Excel_XlLayoutFormType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLayoutFormType
+{
+        const short xlOutline = 1;
+        const short xlTabular = 0;
+};
+}; }; }; };
+#endif
