--- /dev/null	2006-01-03 09:47:36.312320500 +0000
+++ offapi/org/openoffice/vba/Excel/makefile.mk	2006-01-16 09:43:57.000000000 +0000
@@ -0,0 +1,30 @@
+PRJ=..$/..$/..$/..
+
+PRJNAME=api
+
+TARGET=excel
+PACKAGE=org$/openoffice$/vba$/Excel
+
+# --- Settings -----------------------------------------------------
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+.IF "$(ENABLE_VBA)"!="YES"
+dummy:
+        @echo "not building vba..."
+.ENDIF
+
+# ------------------------------------------------------------------------
+
+IDLFILES=\
+	XlCreator.idl \
+	XlCutCopyMode.idl \
+	XlFindLookIn.idl\
+	Constants.idl \
+	XlCalculation.idl \
+	XlCellType.idl \
+	XlCommentDisplayMode.idl
+
+# ------------------------------------------------------------------
+
+.INCLUDE :  target.mk
+
