--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlInsertFormatOrigin.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlInsertFormatOrigin_idl__
+#define __org_openoffice_vba_Excel_XlInsertFormatOrigin_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlInsertFormatOrigin
+{
+        const short xlFormatFromLeftOrAbove = 0;
+        const short xlFormatFromRightOrAbove = 1;
+};
+}; }; }; };
+#endif
