--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlChartLocation.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlChartLocation_idl__
+#define __org_openoffice_vba_Excel_XlChartLocation_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlChartLocation
+{
+        const short xlLocationAsNewSheet = 1;
+        const short xlLocationAsObject = 2;
+        const short xlLocationAutomatic = 3;
+};
+}; }; }; };
+#endif
