--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlTrendlineType.idl	
@@ -0,0 +1,15 @@
+#ifndef __org_openoffice_vba_Excel_XlTrendlineType_idl__
+#define __org_openoffice_vba_Excel_XlTrendlineType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlTrendlineType
+{
+        const long xlExponential = 5;
+        const long xlLinear = -4132; // (&HFFFFEFDC)
+        const long xlLogarithmic = -4133; // (&HFFFFEFDB)
+        const long xlMovingAvg = 6;
+        const long xlPolynomial = 3;
+        const long xlPower = 4;
+};
+}; }; }; };
+#endif
