--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlParameterType.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlParameterType_idl__
+#define __org_openoffice_vba_Excel_XlParameterType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlParameterType
+{
+        const short xlConstant = 1;
+        const short xlPrompt = 0;
+        const short xlRange = 2;
+};
+}; }; }; };
+#endif
