--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlUnderlineStyle.idl	
@@ -0,0 +1,14 @@
+#ifndef __org_openoffice_vba_Excel_XlUnderlineStyle_idl__
+#define __org_openoffice_vba_Excel_XlUnderlineStyle_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlUnderlineStyle
+{
+        const long XlUnderlineStyleDouble = -4119; // (&HFFFFEFE9)
+        const long XlUnderlineStyleDoubleAccounting = 5;
+        const long XlUnderlineStyleNone = -4142; // (&HFFFFEFD2)
+        const long XlUnderlineStyleSingle = 2;
+        const long XlUnderlineStyleSingleAccounting = 4;
+};
+}; }; }; };
+#endif
