--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPivotCellType.idl	
@@ -0,0 +1,19 @@
+#ifndef __org_openoffice_vba_Excel_XlPivotCellType_idl__
+#define __org_openoffice_vba_Excel_XlPivotCellType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPivotCellType
+{
+        const short xlPivotCellBlankCell = 0;
+        const short xlPivotCellCustomSubtotal = 7;
+        const short xlPivotCellDataField = 4;
+        const short xlPivotCellDataPivotField = 8;
+        const short xlPivotCellGrandTotal = 3;
+        const short xlPivotCellPageFieldItem = 6;
+        const short xlPivotCellPivotField = 5;
+        const short xlPivotCellPivotItem = 1;
+        const short xlPivotCellSubtotal = 2;
+        const short xlPivotCellValue = 0;
+};
+}; }; }; };
+#endif
