--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlTextVisualLayoutType.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlTextVisualLayoutType_idl__
+#define __org_openoffice_vba_Excel_XlTextVisualLayoutType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlTextVisualLayoutType
+{
+        const short xlTextVisualLTR = 1;
+        const short xlTextVisualRTL = 2;
+};
+}; }; }; };
+#endif
