--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLinkStatus.idl	
@@ -0,0 +1,20 @@
+#ifndef __org_openoffice_vba_Excel_XlLinkStatus_idl__
+#define __org_openoffice_vba_Excel_XlLinkStatus_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLinkStatus
+{
+        const short xlLinkStatusCopiedValues = 10;
+        const short xlLinkStatusIndeterminate = 5;
+        const short xlLinkStatusInvalidName = 7;
+        const short xlLinkStatusMissingFile = 1;
+        const short xlLinkStatusMissingSheet = 2;
+        const short xlLinkStatusNotStarted = 6;
+        const short xlLinkStatusOK = 0;
+        const short xlLinkStatusOld = 3;
+        const short xlLinkStatusSourceNotCalculated = 4; 
+        const short xlLinkStatusSourceNotOpen = 8;
+        const short xlLinkStatusSourceOpen = 9;
+};
+}; }; }; };
+#endif
