--- /dev/null	2005-06-05 14:44:09.000000000 +0100
+++ offapi/org/openoffice/vba/MSFormReturnTypes.idl	2005-09-13 12:05:27.560595825 +0100
@@ -0,0 +1,28 @@
+#ifndef __MSForms_idl__
+#define __MSForms_idl__
+module MSForms
+{
+    struct ReturnBoolean
+    {
+        boolean Value;
+    };
+                                                                                
+    struct ReturnEffect
+    {
+        //fmDropEffect Value;
+        short Value;
+    };
+                                                                                
+    struct ReturnInteger
+    {
+        long Value;
+    };
+                                                                                
+    struct ReturnSingle
+    {
+        float Value;
+    };
+                                                                                
+
+};
+#endif
