--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlToolbarProtection.idl	
@@ -0,0 +1,15 @@
+#ifndef __org_openoffice_vba_Excel_XlToolbarProtection_idl__
+#define __org_openoffice_vba_Excel_XlToolbarProtection_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlToolbarProtection
+{
+        const long xlNoButtonChanges = 1;
+        const long xlNoChanges = 4;
+        const long xlNoDockingChanges = 3;
+        const long xlNoShapeChanges = 2;
+        const long xlToolbarProtectionNone = -4143; //(&HFFFFEFD1)
+
+};
+}; }; }; };
+#endif
