--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLegendPosition.idl	
@@ -0,0 +1,14 @@
+#ifndef __org_openoffice_vba_Excel_XlLegendPosition_idl__
+#define __org_openoffice_vba_Excel_XlLegendPosition_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLegendPosition
+{
+        const long xlLegendPositionBottom = -4107; //(&HFFFFEFF5)
+        const long xlLegendPositionCorner = 2;
+        const long xlLegendPositionLeft = -4131; //(&HFFFFEFDD)
+        const long xlLegendPositionRight = -4152; //(&HFFFFEFC8)
+        const long xlLegendPositionTop = -4160; //(&HFFFFEFC0)
+};
+}; }; }; };
+#endif
