--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSortOrientation.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlSortOrientation_idl__
+#define __org_openoffice_vba_Excel_XlSortOrientation_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSortOrientation
+{
+        const short xlSortColumns = 1;
+        const short xlSortRows = 2;
+};
+}; }; }; };
+#endif
