--- /dev/null	2004-07-26 17:21:14.000000000 +0530
+++ offapi/org/openoffice/vba/XWorksheets.idl	2005-07-25 13:50:31.000000000 +0530
@@ -0,0 +1,27 @@
+#ifndef __org_openoffice_vba_XWorksheets_idl__
+#define __org_openoffice_vba_XWorksheets_idl__
+                                                                                                                             
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+                                                                                                                             
+#ifndef __org_openoffice_vba_XCollection_idl__
+#include <org/openoffice/vba/XCollection.idl>
+#endif
+                                                                                                                             
+//=============================================================================
+                                                                                                                             
+module org {  module openoffice {  module vba {
+                                                                                                                             
+//=============================================================================
+                                                                                                                             
+published interface XWorksheets : ::org::openoffice::vba::XCollection
+{
+        any  Add( [in] any Before, [in] any After, [in] any Count, [in] any Type );                                 
+        void Delete();
+};
+                                                                                                                             
+}; }; };
+                                                                                                                             
+#endif
+
