--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSaveAsAccessMode.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlSaveAsAccessMode_idl__
+#define __org_openoffice_vba_Excel_XlSaveAsAccessMode_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSaveAsAccessMode
+{
+        const short xlExclusive = 3;
+        const short xlNoChange = 1;
+        const short xlShared = 2;
+};
+}; }; }; };
+#endif
