--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSmartTagDisplayMode.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlSmartTagDisplayMode_idl__
+#define __org_openoffice_vba_Excel_XlSmartTagDisplayMode_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSmartTagDisplayMode
+{
+        const short xlButtonOnly = 2;
+        const short xlDisplayNone = 1;
+        const short xlIndicatorAndButton = 0;
+};
+}; }; }; };
+#endif
