--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlObjectSize.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlObjectSize_idl__
+#define __org_openoffice_vba_Excel_XlObjectSize_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlObjectSize
+{
+        const short xlFitToPage = 2;
+        const short xlFullPage = 3;
+        const short xlScreenSize = 1;
+};
+}; }; }; };
+#endif
