--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlHtmlType.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlHtmlType_idl__
+#define __org_openoffice_vba_Excel_XlHtmlType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlHtmlType
+{
+        const short xlHtmlCalc = 1;
+        const short xlHtmlChart = 3;
+        const short xlHtmlList = 2;
+        const short xlHtmlStatic = 0;
+};
+}; }; }; };
+#endif
