--- /dev/null	2005-06-05 14:44:09.000000000 +0100
+++ offapi/org/openoffice/vba/makefile.mk	2005-08-31 17:03:20.000000000 +0100
@@ -0,0 +1,35 @@
+PRJ=..$/..$/..
+
+PRJNAME=api
+
+TARGET=vba
+PACKAGE=org$/openoffice$/vba
+
+# --- Settings -----------------------------------------------------
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+.IF "$(ENABLE_VBA)"!="YES"
+dummy:
+        @echo "not building vba..."
+.ENDIF
+
+# ------------------------------------------------------------------------
+
+IDLFILES=\
+	XApplication.idl\
+	XCollection.idl\
+	XGlobals.idl\
+	XRange.idl\
+	XWorkbook.idl\
+	XWorkbooks.idl\
+	XWorksheet.idl\
+	XWorksheets.idl\
+	XFont.idl\
+	XRows.idl\
+	XOutline.idl\
+	XColumns.idl \
+	MSFormReturnTypes.idl 
+
+# ------------------------------------------------------------------
+
+.INCLUDE :  target.mk
