--- /dev/null	2006-01-28 08:50:32.192375500 +0000
+++ offapi/org/openoffice/vba/Excel/XlSortOrder.idl	2006-01-30 15:02:53.000000000 +0000
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlSortOrder_idl__
+#define __org_openoffice_vba_Excel_XlSortOrder_idl__
+
+module org  { module openoffice { module vba { module Excel { 
+constants XlSortOrder
+{
+	const short xlAscending = 1;
+	const short xlDescending = 2;
+};
+}; }; }; };
+#endif
