--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPlatform.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlPlatform_idl__
+#define __org_openoffice_vba_Excel_XlPlatform_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPlatform
+{
+        const short xlMacintosh = 1;
+        const short xlMSDOS = 3;
+        const short xlWindows = 2;};
+}; }; }; };
+#endif
