--- /dev/null	
+++ offapi/org/openoffice/vba/XWorkbook.idl	
@@ -0,0 +1,36 @@
+#ifndef __org_openoffice_vba_XWorkbook_idl__
+#define __org_openoffice_vba_XWorkbook_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+//=============================================================================
+
+interface XWorksheet;
+interface XWorksheets;
+
+interface XWorkbook: com::sun::star::uno::XInterface
+{
+	[attribute, readonly] string  Name;
+	[attribute, readonly] string  Path;
+	[attribute, readonly] string  FullName;
+	[attribute, readonly] boolean ProtectStructure; 
+	[attribute, readonly] XWorksheet ActiveSheet;
+	[attribute] boolean Saved;
+
+	any  Worksheets([in] any sheet);
+	void Close([in] any SaveChanges, [in] any FileName, [in] any RouteWorkBook);
+	void Protect( [in] any Password );
+	void Unprotect( [in] any Password );
+	void Save();
+	void Activate();
+};
+
+}; }; };
+
+#endif
