--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSubtototalLocationType.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlSubtototalLocationType_idl__
+#define __org_openoffice_vba_Excel_XlSubtototalLocationType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSubtototalLocationType
+{
+        const short xlAtBottom = 2;
+        const short xlAtTop = 1;
+};
+}; }; }; };
+#endif
