--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLinkInfo.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlLinkInfo_idl__
+#define __org_openoffice_vba_Excel_XlLinkInfo_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLinkInfo
+{
+        const short xlEditionDate = 2;
+        const short xlLinkInfoStatus = 3;
+        const short xlUpdateState = 1;
+};
+}; }; }; };
+#endif
