--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlDisplayDrawingObjects.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlDisplayDrawingObjects_idl__
+#define __org_openoffice_vba_Excel_XlDisplayDrawingObjects_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlDisplayDrawingObjects
+{
+        const long xlDisplayShapes = -4104; //(&HFFFFEFF8)
+        const long xlHide = 3;
+        const long xlPlaceholders = 2;
+};
+}; }; }; };
+#endif
