--- /dev/null	
+++ offapi/org/openoffice/vba/XChartObjects.idl	
@@ -0,0 +1,30 @@
+#ifndef __org_openoffice_vba_XChartObjects_idl__
+#define __org_openoffice_vba_XChartObjects_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __org_openoffice_vba_XCollection_idl__
+#include <org/openoffice/vba/XCollection.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+//=============================================================================
+interface XChartObjects: ::com::sun::star::uno::XInterface
+{
+	// #TODO #FIXME the very basic functionality e.g.
+	//any Add( [in] double Left, [in] double Top, [in] double Width, [in] double Height );
+	//void Delete();
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
