--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlQueryType.idl	
@@ -0,0 +1,15 @@
+#ifndef __org_openoffice_vba_Excel_XlQueryType_idl__
+#define __org_openoffice_vba_Excel_XlQueryType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlQueryType
+{
+        const short xlADORecordset = 7;
+        const short xlDAORecordset = 2;
+        const short xlODBCQuery = 1;
+        const short xlOLEDBQuery = 5;
+        const short xlTextImport = 6;
+        const short xlWebQuery = 4;
+};
+}; }; }; };
+#endif
