--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLinkInfoType.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlLinkInfoType_idl__
+#define __org_openoffice_vba_Excel_XlLinkInfoType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLinkInfoType
+{
+        const short xlLinkInfoOLELinks = 2;
+        const short xlLinkInfoPublishers = 5;
+        const short xlLinkInfoSubscribers = 6;
+};
+}; }; }; };
+#endif
