--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlDataSeriesType.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlDataSeriesType_idl__
+#define __org_openoffice_vba_Excel_XlDataSeriesType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlDataSeriesType
+{
+        const long xlAutoFill = 4;
+        const long xlChronological = 3;
+        const long xlDataSeriesLinear = -4132; //(&HFFFFEFDC)
+        const long xlGrowth = 2;
+};
+}; }; }; };
+#endif
