--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlLineStyle.idl	
@@ -0,0 +1,17 @@
+#ifndef __org_openoffice_vba_Excel_XlLineStyle_idl__
+#define __org_openoffice_vba_Excel_XlLineStyle_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlLineStyle
+{
+        const long xlContinuous = 1;
+        const long xlDash = -4115; //(&HFFFFEFED)
+        const long xlDashDot = 4;
+        const long xlDashDotDot = 5;
+        const long xlDot = -4118; //(&HFFFFEFEA)
+        const long xlDouble = -4119; //(&HFFFFEFE9)
+        const long xlLineStyleNone = -4142; //(&HFFFFEFD2)
+        const long xlSlantDashDot = 13;
+};
+}; }; }; };
+#endif
