--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlBarShape.idl	
@@ -0,0 +1,15 @@
+#ifndef __org_openoffice_vba_Excel_XlBarShape_idl__
+#define __org_openoffice_vba_Excel_XlBarShape_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlBarShape
+{
+        const short xlBox = 0;
+        const short xlConeToMax = 5;
+        const short xlConeToPoint = 4;
+        const short xlCylinder = 3;
+        const short xlPyramidToMax = 2;
+        const short xlPyramidToPoint = 1;
+};
+}; }; }; };
+#endif
