--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlColumnDataType.idl	
@@ -0,0 +1,20 @@
+#ifndef __org_openoffice_vba_Excel_XlColumnDataType_idl__
+#define __org_openoffice_vba_Excel_XlColumnDataType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlColumnDataType
+{
+        const short xlDMYFormat = 4;
+        const short xlDYMFormat = 7;
+        const short xlEMDFormat = 10;
+        const short xlGeneralFormat = 1;
+        const short xlMDYFormat = 3;
+        const short xlMYDFormat = 6;
+        const short xlSkipColumn = 9;
+        const short xlTextFormat = 2;
+        const short xlYDMFormat = 8;
+        const short xlYMDFormat = 5;
+
+};
+}; }; }; };
+#endif
