--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlWindowState.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlWindowState_idl__
+#define __org_openoffice_vba_Excel_XlWindowState_idl__
+module org  { module openoffice { module vba { module Excel {
+constants XlWindowState
+{
+        const long xlMaximized = -4137; // (&HFFFFEFD7)
+        const long xlMinimized = -4140; // (&HFFFFEFD4)
+        const long xlNormal = -4143; // (&HFFFFEFD1)
+};
+}; }; }; };
+#endif
