--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlOLEType.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlOLEType_idl__
+#define __org_openoffice_vba_Excel_XlOLEType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlOLEType
+{
+        const short XlOLEControl = 2;
+        const short XlOLEEmbed = 1;
+        const short XlOLELink = 0;
+};
+}; }; }; };
+#endif
