--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlArrangeStyle.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlArrangeStyle_idl__
+#define __org_openoffice_vba_Excel_XlArrangeStyle_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlArrangeStyle
+{
+        const long xlArrangeStyleCascade = 7;
+        const long xlArrangeStyleHorizontal = -4128; //(&HFFFFEFE0)
+        const long xlArrangeStyleTiled = 1;
+        const long xlArrangeStyleVertical = -4166; //(&HFFFFEFBA)
+};
+}; }; }; };
+#endif
