--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPriority.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlPriority_idl__
+#define __org_openoffice_vba_Excel_XlPriority_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPriority
+{
+        const long xlPriorityHigh = -4127; //(&HFFFFEFE1)
+        const long xlPriorityLow = -4134; //(&HFFFFEFDA)
+        const long xlPriorityNormal = -4143; //(&HFFFFEFD1)
+};
+}; }; }; };
+#endif
