--- /dev/null	2006-01-28 08:50:32.192375500 +0000
+++ offapi/org/openoffice/vba/XComments.idl	2006-02-01 16:27:02.000000000 +0000
@@ -0,0 +1,24 @@
+#ifndef __org_openoffice_vba_XComments_idl__
+#define __org_openoffice_vba_XComments_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+#ifndef __com_sun_star_lang_IllegalArgumentException_idl__
+#include <com/sun/star/lang/IllegalArgumentException.idl>
+#endif
+//=============================================================================
+
+module org {  module openoffice {  module vba {
+
+published interface XComments: com::sun::star::uno::XInterface
+{
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
