--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPasteType.idl	
@@ -0,0 +1,19 @@
+#ifndef __org_openoffice_vba_Excel_XlPasteType_idl__
+#define __org_openoffice_vba_Excel_XlPasteType_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPasteType
+{
+        const long xlPasteAll = -4104; //(&HFFFFEFF8)
+        const long xlPasteAllExceptBorders = 7; 
+        const long xlPasteAllColumnWidths = 8;
+        const long xlPasteComments = -4144; //(&HFFFFEFDO)
+        const long xlPasteFormats = -4122; //(&HFFFFEFE6)
+        const long xlPasteFormulas = -4123; //(&HFFFFEFE5)
+        const long xlPasteFormulasAndNumberFormats = 11;
+        const long xlPasteValidation = 6;
+        const long xlPasteValues = -4163; //(&HFFFFEFBD)
+        const long xlPasteValuesAndNumberFormats = 12;
+};
+}; }; }; };
+#endif
