--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlListConflict.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlListConflict_idl__
+#define __org_openoffice_vba_Excel_XlListConflict_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlListConflict
+{
+        const short xlListConflictDialog = 0;
+        const short xlListConflictDiscardAllConflicts = 2;
+        const short xlListConflictError = 3;
+        const short xlListConflictRetryAllConflicts = 1;
+};
+}; }; }; };
+#endif
