--- /dev/null	
+++ offapi/org/openoffice/vba/XlPasteType.idl	
@@ -0,0 +1,34 @@
+#ifndef __org_openoffice_vba_XlPasteType_idl__
+#define __org_openoffice_vba_XlPasteType_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba {
+
+//=============================================================================
+
+constants xlPasteType
+{
+/*noelp #FIXME check these numbers, they don't look right*/
+	const unsigned short xlPasteAll = 0 ;
+	const unsigned short xlPasteAllExceptBorders = 2;
+	const unsigned short xlPasteColumnWidths = 3;
+	const unsigned short xlPasteComments = 4;
+	const unsigned short xlPasteFormats = 5;
+	const unsigned short xlPasteFormulas = 6;
+	const unsigned short xlPasteFormulasAndNumberFormats = 7;
+	const unsigned short xlPasteValidation = 8;
+	const unsigned short xlPasteValues = 9 ;
+	const unsigned short xlPasteValuesAndNumberFormats =10;
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
