--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlMouseButton.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlMouseButton_idl__
+#define __org_openoffice_vba_Excel_XlMouseButton_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlMouseButton
+{
+        const short xlNoButton = 0;
+        const short xlPrimaryButton = 1;
+        const short xlSecondaryButton = 2;
+};
+}; }; }; };
+#endif
