--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlArrowHeadLength.idl	
@@ -0,0 +1,12 @@
+#ifndef __org_openoffice_vba_Excel_XlArrowHeadLength_idl__
+#define __org_openoffice_vba_Excel_XlArrowHeadLength_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlArrowHeadLength
+{
+        const long xlArrowHeadLengthLong = 3;
+        const long xlArrowHeadLengthMedium = -4138; //(&HFFFFEFD6)
+        const long xlArrowHeadLengthShort = 1;
+};
+}; }; }; };
+#endif
