--- /dev/null	
+++ offapi/org/openoffice/vba/XRange.idl	
@@ -0,0 +1,27 @@
+#ifndef __org_openoffice_vba_XRange_idl__
+#define __org_openoffice_vba_Range_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+published interface XRange: com::sun::star::uno::XInterface
+{
+	[attribute] double Value;
+	void Clear();
+	void ClearComments();
+	void ClearContents();
+	void ClearFormats();
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
