--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlRunAutoMacro.idl	
@@ -0,0 +1,13 @@
+#ifndef __org_openoffice_vba_Excel_XlRunAutoMacro_idl__
+#define __org_openoffice_vba_Excel_XlRunAutoMacro_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlRunAutoMacro
+{
+        const short xlAutoActivate = 3;
+        const short xlAutoClose = 2;
+        const short xlAutoDeactivate = 4;
+        const short xlAutoOpen = 1;
+};
+}; }; }; };
+#endif
