--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlSizeRepresents.idl	
@@ -0,0 +1,11 @@
+#ifndef __org_openoffice_vba_Excel_XlSizeRepresents_idl__
+#define __org_openoffice_vba_Excel_XlSizeRepresents_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlSizeRepresents
+{
+        const short xlSizeIsArea = 1;
+        const short xlSizeIsWidth = 2;
+};
+}; }; }; };
+#endif
