--- /dev/null	
+++ offapi/org/openoffice/vba/Excel/XlPaperSize.idl	
@@ -0,0 +1,50 @@
+#ifndef __org_openoffice_vba_Excel_XlPaperSize_idl__
+#define __org_openoffice_vba_Excel_XlPaperSize_idl__
+module org  { module openoffice { module vba { module Excel {
+
+constants XlPaperSize
+{
+        const short xlPaper10x14 = 16; //(&H10)
+        const short xlPaper11x17 = 17; //(&H11)
+        const short xlPaperA3 = 8;
+        const short xlPaperA4Small = 9;
+        const short xlPaperA5 = 10;
+        const short xlPaperB4 = 12;
+        const short xlPaperB5 = 13;
+        const short xlPaperCsheet = 24; //(&H18)
+        const short xlPaperDsheet = 25; //(&H19)
+        const short xlPaperEnvelope10 = 20; //(&H14)
+        const short xlPaperEnvelope11 = 21; //(&H15)
+        const short xlPaperEnvelope12 = 22; //(&H16)
+        const short xlPaperEnvelope14 = 23; //(&H17)
+        const short xlPaperEnvelope9 = 19; //(&H13)
+        const short xlPaperEnvelopeB4 = 33; //(&H21)
+        const short xlPaperEnvelopeB5 = 34; //(&H22)
+        const short xlPaperEnvelopeB6 = 35; //(&H23)
+        const short xlPaperEnvelopeC3 = 29; //(&H1D)
+        const short xlPaperEnvelopeC4 = 30; //(&H1E)
+        const short xlPaperEnvelopeC5 = 28; //(&H1C)
+        const short xlPaperEnvelopeC6 = 31; //(&H1F)
+        const short xlPaperEnvelopeC65 = 32; //(&H20)
+        const short xlPaperEnvelopeDL = 27; //(&H1B)
+        const short xlPaperEnvelopeItaly = 36; //(&H24)
+        const short xlPaperEnvelopeMonarch = 37; //(&H25)
+        const short xlPaperEnvelopePersonal = 38; //(&H26)
+        const short xlPaperEsheet = 26; //(&H1A)
+        const short xlPaperExective = 7;
+        const short xlPaperFanfoldLegalGerman = 41; //(&H29)
+        const short xlPaperFanfoldStdGerman = 40; //(&H28)
+        const short xlPaperFanfoldUS = 39; //(&H27)
+        const short xlPaperFolio = 14;
+        const short xlPaperLedger = 4;
+        const short xlPaperLegal = 5;
+        const short xlPaperLetter = 1;
+        const short xlPaperLetterSmall = 2;
+        const short xlPaperNote = 18; //(&H12)
+        const short xlPaperQuarto = 15;
+        const short xlPaperStatement = 6;
+        const short xlPaperTabloid = 3;
+        const short xlPaperUser = 256; //(&H100)
+};
+}; }; }; };
+#endif
