--- oovbaapi/org/openoffice/excel/XPivotTable.idl	
+++ oovbaapi/org/openoffice/excel/XPivotTable.idl
@@ -8,14 +8,21 @@
 #include <org/openoffice/excel/XPivotCache.idl>
 #endif
 
+#ifndef __org_openoffice_vba__XHelperInterface_idl__
+#define __org_openoffice_vba__XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
+
 //=============================================================================
 
 module org {  module openoffice {  module excel { 
 
 //=============================================================================
 
-interface XPivotTable: com::sun::star::uno::XInterface
+//interface XPivotTable: com::sun::star::uno::XInterface
+interface XPivotTable
 {
+	interface org::openoffice::vba::XHelperInterface;
 	XPivotCache PivotCache();
 };
 
