--- oovbaapi/org/openoffice/excel/XDialogs.idl	
+++ oovbaapi/org/openoffice/excel/XDialogs.idl
@@ -4,20 +4,28 @@
 #ifndef __com_sun_star_uno_XInterface_idl__
 #include <com/sun/star/uno/XInterface.idl>
 #endif
-                                                                                                                             
 #ifndef __org_openoffice_vba_XCollection_idl__
 #include <org/openoffice/vba/XCollection.idl>
 #endif
-                                                                                                                             
+
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
+
 //=============================================================================
                                                                                                                              
 module org {  module openoffice {  module excel {
-                                                                                                                             
+interface XApplication;                                                                                                                             
 //=============================================================================
                                                                                                                              
-interface XDialogs : ::org::openoffice::vba::XCollection
+//interface XDialogs : ::org::openoffice::vba::XCollection
+interface XDialogs 
 {
-	void Dummy();
+	interface ::org::openoffice::vba::XHelperInterface;
+
+        [attribute, readonly] long Count;
+
+	any Item( [in] any Index );
 };
                                                                                                                              
 }; }; };
