--- oovbaapi/org/openoffice/msforms/XShapeRange.idl	
+++ oovbaapi/org/openoffice/msforms/XShapeRange.idl
@@ -43,17 +43,20 @@
 #include <org/openoffice/msforms/XShape.idl>
 #endif
 
-#ifndef __org_openoffice_msforms_XShapes_idl__
-#include <org/openoffice/msforms/XShapes.idl>
+#ifndef  __org_openoffice_vba_XCollection_idl__
+#include <org/openoffice/vba/XCollection.idl>
 #endif
 
 module org { module openoffice { module msforms {
 
-interface XShapeRange : org::openoffice::vba::XHelperInterface
+interface XShapeRange 
 {
+    interface org::openoffice::vba::XCollection;
     void Select();
     XShape Group();
-    long Count();
+    void IncrementRotation( [in] double Increment );
+    void IncrementLeft( [in] double Increment );
+    void IncrementTop( [in] double Increment );
 };
 }; }; };
 
