--- oovbaapi/org/openoffice/excel/XWindows.idl	
+++ oovbaapi/org/openoffice/excel/XWindows.idl
@@ -5,13 +5,19 @@
 #include <com/sun/star/uno/XInterface.idl>
 #endif
 
+#ifndef __org_openoffice_vba_XCollection_idl__
+#include <org/openoffice/vba/XCollection.idl>
+#endif
+
 //=============================================================================
 
 module org {  module openoffice {  module excel { 
 
 //=============================================================================
-interface XWindows: ::com::sun::star::uno::XInterface
+interface XWindows
 {
+	interface ::org::openoffice::vba::XCollection;
+
 	void Arrange( [in] long ArrangeStyle, [in] any ActiveWorkbook, [in] any SyncHorizontal, [in] any SyncVertical );	
 };
 
