--- /dev/null	
+++ oovbaapi//org/openoffice/office/MsoLineDashStyle.idl
@@ -0,0 +1,21 @@
+#ifndef __org_openoffice_office_MsoLineDashStyle_idl__
+#define __org_openoffice_office_MsoLineDashStyle_idl__
+
+module org{ module openoffice { module office {
+
+constants MsoLineDashStyle
+{
+    const long msoLineDash = 4;
+    const long msoLineDashDot = 5;
+    const long msoLineDashDotDot = 6;
+    const long msoLineDashStyleMixed = -2;
+    const long msoLineLongDash = 7;
+    const long msoLineLongDashDot = 8;
+    const long msoLineRoundDot = 3;
+    const long msoLineSolid = 1;
+    const long msoLineSquareDot = 2;
+};
+
+};};};
+
+#endif
