--- /dev/null	
+++ oovbaapi/org/openoffice/excel/XOLEObject.idl
@@ -0,0 +1,37 @@
+#ifndef __org_openoffice_excel_XOLEObject_idl__
+#define __org_openoffice_excel_XOLEObject_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
+
+//=============================================================================
+
+module org {  module openoffice {  module excel { 
+
+
+//=============================================================================
+interface XOLEObject
+{
+	interface ::org::openoffice::vba::XHelperInterface;
+
+	[attribute, readonly] com::sun::star::uno::XInterface Object;
+    [attribute] boolean Enabled;
+    [attribute] boolean Visible;
+    [attribute] double  Left;
+    [attribute] double  Top;
+    [attribute] double  Height;
+    [attribute] double  Width;
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
