--- oovbaapi/org/openoffice/excel/XSeries.idl	
+++ oovbaapi/org/openoffice/excel/XSeries.idl
@@ -5,14 +5,20 @@
 #include <com/sun/star/uno/XInterface.idl>
 #endif
 
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
+
 //=============================================================================
 
 module org {  module openoffice {  module excel { 
 
 //=============================================================================
 
-interface XSeries: com::sun::star::uno::XInterface
+interface XSeries
 {
+	interface ::org::openoffice::vba::XHelperInterface;
+
 	[attribute, readonly] string Name;
 	any Series( [in] any Item );
 	any Values ( [in] any Values );
