--- oovbaapi/org/openoffice/vba/XCollection.idl	
+++ oovbaapi/org/openoffice/vba/XCollection.idl
@@ -4,22 +4,28 @@
 #ifndef __com_sun_star_uno_XInterface_idl__
 #include <com/sun/star/uno/XInterface.idl>
 #endif
-#ifndef __org_openoffice_excel_XApplication_idl__
-#include <org/openoffice/excel/XApplication.idl>
+#ifndef __com_sun_star_script_XDefaultMethod_idl__
+#include <com/sun/star/script/XDefaultMethod.idl>
+#endif
+#ifndef __com_sun_star_container_XEnumerationAccess_idl__
+#include <com/sun/star/container/XEnumerationAccess.idl>
+#endif
+
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
 #endif
 //=============================================================================
 
 module org {  module openoffice {  module vba { 
-
 //=============================================================================
 
-interface XCollection : ::com::sun::star::uno::XInterface
+interface XCollection
 {
-	[attribute, readonly] long Count;
-	[attribute, readonly] any   Parent;
-	[attribute, readonly] long Creator;
-	[attribute, readonly] org::openoffice::excel::XApplication Application;
+	interface ::org::openoffice::vba::XHelperInterface;
+	interface ::com::sun::star::script::XDefaultMethod;
+	interface ::com::sun::star::container::XEnumerationAccess;
 
+	[attribute, readonly] long Count;
 	any  Item( [in] any Index );
 };
 
