--- oovbaapi/org/openoffice/excel/XWorksheets.idl	
+++ oovbaapi/org/openoffice/excel/XWorksheets.idl
@@ -56,8 +56,8 @@ interface XWorksheets
         any  Add( [in] any Before, [in] any After, [in] any Count, [in] any Type );                                 
         void Delete();
 	void PrintOut( [in] any From, [in] any To, [in] any Copies, [in] any Preview, [in] any ActivePrinter, [in] any PrintToFile, [in] any Collate, [in] any PrToFileName );
+	void Select( [in] any Replace );
 };
-                                                                                                                             
 }; }; };
                                                                                                                              
 #endif
