--- oovbaapi/genconstidl/makefile.mk	2007-08-30 19:26:17.000000000 +0200
+++ oovbaapi/genconstidl/makefile.mk	2007-08-30 19:32:11.000000000 +0200
@@ -40,11 +40,6 @@ PRJNAME=oovbaapi
 
 .INCLUDE : settings.mk
 
-.IF "$(ENABLE_VBA)"!="YES"
-dummy:
-        @echo "not building vba..."
-.ENDIF
-
 # symbol files of Microsoft API
 MYSYMFILES = access.api vba.api adodb.api msforms.api word.api dao.api powerpoint.api excel.api stdole.api
 
--- oovbaapi/ooo/vba/constants/makefile.mk	2007-08-30 19:30:00.000000000 +0200
+++ oovbaapi/ooo/vba/constants/makefile.mk	2007-08-30 19:31:48.000000000 +0200
@@ -42,11 +42,6 @@ PACKAGE=ooo$/vba$/constants
 # --- Settings -----------------------------------------------------
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
-.IF "$(ENABLE_VBA)"!="YES"
-dummy:
-        @echo "not building vba..."
-.ELSE
-
 # ------------------------------------------------------------------------
 
 # I tried to use the IDLFILES target but it failed dismally
@@ -64,5 +64,3 @@ $(MY_GEN_UCR_PATH)$/%.urd: $(MY_GEN_IDL_
 $(MYDBTARGET) : $(MYURDFILES)  $(MYIDLFILES)
     $(REGMERGE) $(OUT)$/ucr/constants.db /UCR @$(mktmp $(MYURDFILES))

-.ENDIF
-
--- oovbaapi/ooo/vba/excel/makefile.mk	2007-09-01 11:42:58.000000000 +0200
+++ oovbaapi/ooo/vba/excel/makefile.mk	2007-09-01 11:55:50.000000000 +0200
@@ -42,11 +42,6 @@ PACKAGE=ooo$/vba$/Excel
 # --- Settings -----------------------------------------------------
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
-.IF "$(ENABLE_VBA)"!="YES"
-dummy:
-        @echo "not building vba..."
-.ENDIF
-
 # ------------------------------------------------------------------------
 
 
--- oovbaapi/ooo/vba/msforms/makefile.mk	2007-08-30 19:30:20.000000000 +0200
+++ oovbaapi/ooo/vba/msforms/makefile.mk	2007-08-30 19:31:31.000000000 +0200
@@ -42,11 +42,6 @@ PACKAGE=ooo$/vba$/msforms
 # --- Settings -----------------------------------------------------
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
-.IF "$(ENABLE_VBA)"!="YES"
-dummy:
-        @echo "not building vba..."
-.ENDIF
-
 # ------------------------------------------------------------------------
 
 IDLFILES=\
--- oovbaapi/ooo/vba/makefile.mk	2007-08-30 19:30:25.000000000 +0200
+++ oovbaapi/ooo/vba/makefile.mk	2007-08-30 19:31:38.000000000 +0200
@@ -42,11 +42,6 @@ PACKAGE=ooo$/vba
 # --- Settings -----------------------------------------------------
 .INCLUDE :  $(PRJ)$/util$/makefile.pmk
 
-.IF "$(ENABLE_VBA)"!="YES"
-dummy:
-        @echo "not building vba..."
-.ENDIF
-
 # ------------------------------------------------------------------------
 
 IDLFILES=\
--- oovbaapi/util/makefile.mk	2007-08-30 17:32:39.000000000 +0200
+++ oovbaapi/util/makefile.mk	2007-08-30 19:32:03.000000000 +0200
@@ -44,11 +44,6 @@ TARGET=oovbaapi_db
 
 .INCLUDE :  makefile.pmk
 
-.IF "$(ENABLE_VBA)"!="YES"
-dummy:
-        @echo "not building vba..."
-.ENDIF
-
 # ------------------------------------------------------------------
 UNOIDLDBFILES= \
 	$(UCR)$/vba.db \
