--- /dev/null	
+++ oovbaapi/org/openoffice/vba/XPropValue.idl
@@ -0,0 +1,26 @@
+#ifndef __org_openoffice_vba_XPropValue_idl__
+#define __org_openoffice_vba_XPropValue_idl__
+
+#ifndef __com_sun_star_script_XDefaultProperty_idl__
+#include <com/sun/star/script/XDefaultProperty.idl>
+#endif
+
+
+//=============================================================================
+
+module org {  module openoffice {  module vba { 
+
+
+//=============================================================================
+interface XPropValue: com::sun::star::script::XDefaultProperty
+{
+	[attribute] any Value;
+};
+
+//=============================================================================
+
+}; }; };
+
+#endif
+
+
