--- /dev/null	
+++ oovbaapi//org/openoffice/office/MsoScaleFrom.idl
@@ -0,0 +1,15 @@
+#ifndef __org_openoffice_office_MsoScaleFrom_idl__
+#define __org_openoffice_office_MsoScaleFrom_idl__
+
+module org{ module openoffice { module office {
+
+constants MsoScaleFrom
+{
+    const long msoScaleFromBottomRight = 2;
+    const long msoScaleFromMiddle = 1;
+    const long msoScaleFromTopLeft = 0;
+};
+
+};};};
+
+#endif
