--- oovbaapi/org/openoffice/excel/XRange.idl	
+++ oovbaapi/org/openoffice/excel/XRange.idl
@@ -7,6 +7,15 @@
 #ifndef __com_sun_star_lang_IllegalArgumentException_idl__
 #include <com/sun/star/lang/IllegalArgumentException.idl>
 #endif
+#ifndef __com_sun_star_container_XEnumerationAccess_idl__
+#include <com/sun/star/container/XEnumerationAccess.idl>
+#endif
+#ifndef __com_sun_star_script_XDefaultProperty_idl__
+#include <com/sun/star/script/XDefaultProperty.idl>
+#endif
+#ifndef __com_sun_star_script_XDefaultMethod_idl__
+#include <com/sun/star/script/XDefaultMethod.idl>
+#endif
 #ifndef __org_openoffice_excel_XlPasteType_idl__
 #include <org/openoffice/excel/XlPasteType.idl>
 #endif
@@ -16,6 +25,9 @@
 #ifndef __org_openoffice_vba_XCollection_idl__
 #include <org/openoffice/vba/XCollection.idl>
 #endif
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
 //=============================================================================
 
 module org {  module openoffice {  module excel { 
@@ -30,8 +42,13 @@ interface XBorders;
 interface XValidation;
 published interface XWorksheet;
 
-interface XRange: com::sun::star::uno::XInterface
+interface XRange
 {
+ 	interface com::sun::star::container::XEnumerationAccess;
+	interface com::sun::star::script::XDefaultMethod;
+	interface com::sun::star::script::XDefaultProperty;
+	interface ::org::openoffice::vba::XHelperInterface;
+
 	[attribute] any Value; 
 	[attribute] any Formula;
 	[attribute] any FormulaArray;
