--- oovbaapi/org/openoffice/excel/XChartObject.idl	
+++ oovbaapi/org/openoffice/excel/XChartObject.idl
@@ -4,15 +4,19 @@
 #ifndef __com_sun_star_uno_XInterface_idl__
 #include <com/sun/star/uno/XInterface.idl>
 #endif
-
+#ifndef __org_openoffice_vba_XHelperInterface_idl__
+#include <org/openoffice/vba/XHelperInterface.idl>
+#endif
 //=============================================================================
 
 module org {  module openoffice {  module excel { 
 
 //=============================================================================
 interface XChart;
-interface XChartObject: com::sun::star::uno::XInterface
+interface XChartObject
 {
+	interface ::org::openoffice::vba::XHelperInterface;
+
 	[attribute, readonly] XChart Chart;
 	[attribute, readonly] string Name;
 	any test();
