diff --git basic/source/runtime/step0.cxx basic/source/runtime/step0.cxx
index 25adf15..cbd2fad 100644
--- basic/source/runtime/step0.cxx
+++ basic/source/runtime/step0.cxx
@@ -109,19 +109,24 @@ void SbiRuntime::StepCompare( SbxOperato
 	// values ( and type ) set as appropriate
 	SbxDataType p1Type = p1->GetType(); 
 	SbxDataType p2Type = p2->GetType(); 
+	if ( p1Type == SbxEMPTY )
+	{
+		p1->Broadcast( SBX_HINT_DATAWANTED );
+		p1Type = p1->GetType(); 
+	}
+	if ( p2Type == SbxEMPTY )
+	{
+		p2->Broadcast( SBX_HINT_DATAWANTED );
+		p2Type = p2->GetType(); 
+	}
 	if ( p1Type == p2Type )
 	{
-		if ( p1Type == SbxEMPTY )
-		{
-			p1->Broadcast( SBX_HINT_DATAWANTED );
-			p2->Broadcast( SBX_HINT_DATAWANTED );
-		}
 		// if both sides are an object and have default props
 		// then we need to use the default props
 		// we don't need to worry if only one side ( lhs, rhs ) is an 
 		// object ( object side will get coerced to correct type in 
 		// Compare )
-		else if ( p1Type ==  SbxOBJECT )
+		if ( p1Type ==  SbxOBJECT )
 		{
 			SbxVariable* pDflt = getDefaultProp( p1 );
 			if ( pDflt )
