diff --git sc/source/ui/vba/vbarange.cxx sc/source/ui/vba/vbarange.cxx
index 95f312c..cd25749 100644
--- sc/source/ui/vba/vbarange.cxx
+++ sc/source/ui/vba/vbarange.cxx
@@ -3388,7 +3391,7 @@ ScVbaRange::createEnumeration() throw (uno::RuntimeException)
 ::rtl::OUString SAL_CALL 
 ScVbaRange::getDefaultMethodName(  ) throw (uno::RuntimeException)
 {
-	const static rtl::OUString sName( RTL_CONSTASCII_USTRINGPARAM("Cells") );
+	const static rtl::OUString sName( RTL_CONSTASCII_USTRINGPARAM("Item") );
 	return sName;
 }
 
