diff --git svx/source/msfilter/msvbasic.cxx svx/source/msfilter/msvbasic.cxx
index a8aac8a..9cec8e5 100644
--- svx/source/msfilter/msvbasic.cxx
+++ svx/source/msfilter/msvbasic.cxx
@@ -681,8 +681,7 @@ ReferenceRegisteredRecord::read( SvStream* pStream )
     *pStream >> Id >> Size >> SizeOfLibid;
     if ( SizeOfLibid )
     {
-        pLibid = new sal_uInt8[ SizeOfLibid ];
-        OSL_TRACE("ReferenceRegisteredRecord about to read Libid [0x%x]", pStream->Tell() ); 
+        pLibid = new sal_uInt8[ SizeOfLibid];
         pStream->Read( pLibid, SizeOfLibid );
     }
     *pStream >> Reserved1 >> Reserved2;
@@ -758,7 +757,14 @@ ReferenceControlRecord() : Id( 0x2F ), SizeTwiddled( 0 ), SizeOfLibidTwiddled( 0
 bool read( SvStream* pStream )
 {
     OSL_TRACE("ReferenceControlRecord [0x%x]", pStream->Tell() ); 
-    OriginalRecord.read( pStream );
+    long nPos = pStream->Tell();
+
+    *pStream >> Id;
+    pStream->Seek( nPos ); // point before the peeked Id
+    if ( Id == 0x33 ) // we have an OriginalRecord
+    {
+        OriginalRecord.read( pStream );
+    }
     *pStream >> Id >> SizeTwiddled >> SizeOfLibidTwiddled;
 
     if ( SizeOfLibidTwiddled )
@@ -769,7 +775,7 @@ bool read( SvStream* pStream )
 
     *pStream >> Reserved1 >> Reserved2;
 
-    long nPos = pStream->Tell();
+    nPos = pStream->Tell();
     // peek at the id for optional NameRecord
     sal_Int16 nTmpId;
     *pStream >> nTmpId;
