--- sc/source/ui/inc/dbfunc.hxx	2007-07-16 20:34:07.000000000 +0100
+++ sc/source/ui/inc/dbfunc.hxx	2007-07-17 13:07:35.000000000 +0100
@@ -66,7 +66,7 @@ public:
 
 	void			Sort( const ScSortParam& rSortParam,
 						  BOOL bRecord = TRUE, BOOL bPaint = TRUE );
-	void			Query( const ScQueryParam& rQueryParam,
+	SC_DLLPUBLIC void			Query( const ScQueryParam& rQueryParam,
 						   const ScRange* pAdvSource, BOOL bRecord );
 	void			DoSubTotals( const ScSubTotalParam& rParam, BOOL bRecord = TRUE,
 							const ScSortParam* pForceNewSort = NULL );
