Index: sc/inc/docuno.hxx
===================================================================
RCS file: /cvs/sc/sc/inc/docuno.hxx,v
retrieving revision 1.21
diff -u -p -r1.21 docuno.hxx
--- sc/inc/docuno.hxx	8 Sep 2005 17:33:48 -0000	1.21
+++ sc/inc/docuno.hxx	7 Nov 2005 20:22:05 -0000
@@ -181,7 +181,7 @@ public:
 	static void				CreateAndSet(ScDocShell* pDocSh);
 
 	ScDocument*				GetDocument() const;
-    SfxObjectShell*         GetEmbeddedObject() const;
+SC_DLLPUBLIC    SfxObjectShell*         GetEmbeddedObject() const;
 	BOOL					IsEmbedded() const;
 	double					GetOutputFactor() const;
 
