--- /dev/null	2005-07-25 08:49:40.924536240 +0100
+++ sc/source/ui/vba/vbaoutline.cxx	2005-07-26 12:39:11.000000000 +0100
@@ -0,0 +1,18 @@
+#include "vbaoutline.hxx"
+
+void
+ScVbaOutline::ShowLevels( const uno::Any& RowLevels, const uno::Any& ColumnLevels ) throw (::com::sun::star::uno::RuntimeException) 
+{
+	sal_Int16 nLevel;
+	if (mxOutline.is()) 
+	{
+		if (RowLevels >>= nLevel)
+		{
+			mxOutline->showLevel(nLevel, table::TableOrientation_ROWS);
+		} 
+		if (ColumnLevels >>= nLevel)
+		{
+			mxOutline->showLevel(nLevel,table::TableOrientation_COLUMNS);
+		}
+	}
+}
