--- /dev/null	2005-07-25 08:49:40.924536240 +0100
+++ sc/source/ui/vba/vbahelper.hxx	2005-07-05 15:34:17.000000000 +0100
@@ -0,0 +1,15 @@
+#ifndef SC_VBA_HELPER_HXX
+#define SC_VBA_HELPER_HXX
+
+using namespace ::com::sun::star;
+using namespace ::org::openoffice;
+
+#ifdef DEBUG
+#  define SC_VBA_FIXME(a) OSL_TRACE( a )
+#  define SC_VBA_STUB() SC_VBA_FIXME(( "%s - stubbed\n", __FUNCTION__ ))
+#else
+#  define SC_VBA_FIXME(a)
+#  define SC_VBA_STUB()
+#endif
+
+#endif
