--- sc/source/ui/vba/vbashaperange.hxx	
+++ sc/source/ui/vba/vbashaperange.hxx
@@ -38,27 +38,34 @@
 #include <com/sun/star/drawing/XShapes.hpp>
 #include <org/openoffice/msforms/XShapeRange.hpp>
 
-#include "vbahelperinterface.hxx"
-#include "vbahelper.hxx"
+#include "vbacollectionimpl.hxx"
 
-typedef InheritedHelperInterfaceImpl1< oo::msforms::XShapeRange > ScVbaShapeRange_BASE;
+typedef CollTestImplHelper< oo::msforms::XShapeRange > ScVbaShapeRange_BASE;
 
 class ScVbaShapeRange : public ScVbaShapeRange_BASE
 {
 private:
-    css::uno::Reference< css::drawing::XShapes > m_xShapes;
     css::uno::Reference< css::drawing::XDrawPage > m_xDrawPage;
+    css::uno::Reference< css::drawing::XShapes > m_xShapes;
     sal_Int32 m_nShapeGroupCount;
 protected:
 	virtual rtl::OUString& getServiceImplName();
 	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
+	css::uno::Reference< css::drawing::XShapes > getShapes() throw (css::uno::RuntimeException) ;
 public:
-    ScVbaShapeRange( const css::uno::Reference< oo::vba::XHelperInterface >& xParent, const css::uno::Reference< css::uno::XComponentContext >& xContext, css::uno::Reference< css::drawing::XShapes > xShapes, css::uno::Reference< css::drawing::XDrawPage> xDrawShape );
+    ScVbaShapeRange( const css::uno::Reference< oo::vba::XHelperInterface >& xParent, const css::uno::Reference< css::uno::XComponentContext >& xContext, const css::uno::Reference< css::container::XIndexAccess >& xShapes, const css::uno::Reference< css::drawing::XDrawPage>& xDrawShape );
 
     // Methods
     virtual void SAL_CALL Select(  ) throw (css::uno::RuntimeException);
     virtual css::uno::Reference< ::org::openoffice::msforms::XShape > SAL_CALL Group() throw (css::uno::RuntimeException);
-    virtual sal_Int32 SAL_CALL Count(  ) throw (css::uno::RuntimeException);
+    virtual void SAL_CALL IncrementRotation( double Increment ) throw (css::uno::RuntimeException);
+    virtual void SAL_CALL IncrementLeft( double Increment ) throw (css::uno::RuntimeException) ;
+    virtual void SAL_CALL IncrementTop( double Increment ) throw (css::uno::RuntimeException);
+    //XEnumerationAccess
+    virtual css::uno::Type SAL_CALL getElementType() throw (css::uno::RuntimeException);
+    virtual css::uno::Reference< css::container::XEnumeration > SAL_CALL createEnumeration() throw (css::uno::RuntimeException); 
+    // ScVbaCollectionBaseImpl
+    virtual css::uno::Any createCollectionObject( const css::uno::Any& aSource );
 };
 
 #endif//SC_VBA_SHAPERANGE_HXX
