--- sc/source/ui/vba/vbaworksheet.hxx	2007-03-02 14:56:07.000000000 +0000
+++ sc/source/ui/vba/vbaworksheet.hxx	2007-02-13 18:05:42.000000000 +0000
@@ -60,6 +26,7 @@ class ScVbaWorksheet : public ::cppu::We
 	css::uno::Reference< oo::excel::XRange > getSheetRange() throw (css::uno::RuntimeException);
 
 	css::uno::Any getControl( const rtl::OUString& sName );
+	css::uno::Any getControlShape( const rtl::OUString& sName );
 protected:
 
 	virtual css::uno::Reference< css::frame::XModel > getModel()
@@ -114,6 +81,8 @@ public:
 	virtual css::uno::Any SAL_CALL PivotTables( const css::uno::Any& Index ) throw (css::uno::RuntimeException);
 	virtual css::uno::Any SAL_CALL Comments( const css::uno::Any& Index ) throw (css::uno::RuntimeException);
 
+	virtual css::uno::Any SAL_CALL OLEObjects( const css::uno::Any& Index ) throw (css::uno::RuntimeException);
+
 	// XInvocation
 	virtual css::uno::Reference< css::beans::XIntrospectionAccess > SAL_CALL getIntrospection(  ) throw (css::uno::RuntimeException);
 	virtual css::uno::Any SAL_CALL invoke( const ::rtl::OUString& aFunctionName, const css::uno::Sequence< css::uno::Any >& aParams, css::uno::Sequence< ::sal_Int16 >& aOutParamIndex, css::uno::Sequence< css::uno::Any >& aOutParam ) throw (css::lang::IllegalArgumentException, css::script::CannotConvertException, css::reflection::InvocationTargetException, css::uno::RuntimeException);
