--- sc/source/ui/vba/vbaworksheet.hxx	
+++ sc/source/ui/vba/vbaworksheet.hxx
@@ -36,6 +36,8 @@
 #define SC_VBA_WORKSHEET_HXX
 
 #include <cppuhelper/implbase2.hxx>
+#include <comphelper/unwrapargs.hxx>
+
 #include <com/sun/star/sheet/XSpreadsheet.hpp>
 #include <com/sun/star/script/XInvocation.hpp>
 #include <org/openoffice/excel/XWorksheet.hpp>
@@ -75,6 +77,7 @@ public:
 		const css::uno::Reference< css::uno::XComponentContext >& xContext,
 		const css::uno::Reference< css::sheet::XSpreadsheet >& xSheet, 
 		const css::uno::Reference< css::frame::XModel >& xModel )throw (css::uno::RuntimeException)  ;
+	ScVbaWorksheet( css::uno::Sequence< css::uno::Any > const& aArgs, css::uno::Reference< css::uno::XComponentContext >const& xContext ) throw ( css::lang::IllegalArgumentException );
 
 	virtual ~ScVbaWorksheet() {}
 
