--- /dev/null	
+++ sc/source/ui/vba/vbaapplication.hxx	
@@ -0,0 +1,28 @@
+#ifndef SC_VBA_APPLICATION_HXX
+#define SC_VBA_APPLICATION_HXX
+
+#include <org/openoffice/vba/XApplication.hpp>
+#include "vbahelper.hxx"
+
+class ScModelObj;
+
+class ScVbaApplication : public org::openoffice::vba::XApplication
+{
+	SC_VBA_UNO_HELPER_MEMBERS;
+	ScModelObj *mpModel;
+
+public:
+	ScVbaApplication( ScModelObj *pModel );
+	virtual ~ScVbaApplication();
+
+	// XInterface
+	SC_VBA_UNO_HELPER_XINTERFACE;
+
+    // Attributes
+    virtual ::com::sun::star::uno::Reference< ::org::openoffice::vba::XWorkSheet >
+				getActiveSheet() throw (::com::sun::star::uno::RuntimeException);
+    virtual ::com::sun::star::uno::Reference< ::org::openoffice::vba::XWorkBook >
+				getActiveWorkBook() throw (::com::sun::star::uno::RuntimeException);
+};
+
+#endif /* SC_VBA_APPLICATION_HXX */
