--- sc/source/ui/vba//vbawindow.hxx	
+++ sc/source/ui/vba//vbawindow.hxx
@@ -37,20 +37,24 @@
 #include <cppuhelper/implbase1.hxx>
 #include <org/openoffice/excel/XWindow.hpp>
 #include <com/sun/star/uno/XComponentContext.hpp>
+#include <com/sun/star/sheet/XViewPane.hpp>
 #include <com/sun/star/frame/XModel.hpp>
+#include <org/openoffice/excel/XPane.hpp>
 
-#include "vbahelper.hxx"
+#include "vbahelperinterface.hxx"
 #include "vbaworkbook.hxx"
 
-typedef ::cppu::WeakImplHelper1<oo::excel::XWindow > WindowImpl_BASE;
+typedef InheritedHelperInterfaceImpl1<oo::excel::XWindow > WindowImpl_BASE;
 
 class ScVbaWindow : public WindowImpl_BASE
 {		
-	css::uno::Reference< css::uno::XComponentContext > m_xContext;
 	css::uno::Reference< css::frame::XModel > m_xModel;
+	css::uno::Reference< css::sheet::XViewPane > m_xViewPane;
+	css::uno::Reference< oo::excel::XPane > m_xPane;
 	void  Scroll( const css::uno::Any& Down, const css::uno::Any& Up, const css::uno::Any& ToRight, const css::uno::Any& ToLeft, bool bLargeScroll = false ) throw (css::uno::RuntimeException);	
 public:
-	ScVbaWindow( const css::uno::Reference< css::uno::XComponentContext >& xContext, const css::uno::Reference< css::frame::XModel >& xModel ) : m_xContext(xContext), m_xModel( xModel ) {}
+	ScVbaWindow( const css::uno::Reference< oo::vba::XHelperInterface >& xParent, const css::uno::Reference< css::uno::XComponentContext >& xContext, const css::uno::Reference< css::frame::XModel >& xModel );
+ 	css::uno::Reference< oo::excel::XPane > ActivePane();
 	// XWindow
 	virtual void SAL_CALL setCaption( const css::uno::Any& _caption ) throw (css::uno::RuntimeException);
 	virtual css::uno::Any SAL_CALL getCaption() throw (css::uno::RuntimeException);
@@ -68,6 +72,9 @@ public:
 	virtual void SAL_CALL ScrollWorkbookTabs( const css::uno::Any& Sheets, const css::uno::Any& Position ) throw (css::uno::RuntimeException);
 	virtual void SAL_CALL Activate(  ) throw (css::uno::RuntimeException);
 	virtual void SAL_CALL Close( const css::uno::Any& SaveChanges, const css::uno::Any& FileName, const css::uno::Any& RouteWorkBook ) throw (css::uno::RuntimeException);
+	// XHelperInterface
+	virtual rtl::OUString& getServiceImplName();
+	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
 };
 
 #endif //SC_VBA_WINDOW_HXX
