--- sc/source/ui/vba//vbapivottables.hxx	
+++ sc/source/ui/vba//vbapivottables.hxx
@@ -42,16 +42,18 @@
 #include <com/sun/star/uno/XComponentContext.hpp>
  
 #include "vbahelper.hxx"
+#include "vbahelperinterface.hxx"
 #include "vbacollectionimpl.hxx"
 
 
-typedef ::cppu::ImplInheritanceHelper1< ScVbaCollectionBaseImpl, oo::excel::XPivotTables > ScVbaPivotTables_BASE;
+typedef CollTestImplHelper< oo::excel::XPivotTables > ScVbaPivotTables_BASE;
 
 class ScVbaPivotTables : public ScVbaPivotTables_BASE
 {
-protected:
+	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
+
 public:
-	ScVbaPivotTables( const css::uno::Reference< css::uno::XComponentContext > & xContext, const css::uno::Reference< css::container::XIndexAccess >& xIndexAccess );
+	ScVbaPivotTables( const css::uno::Reference< oo::vba::XHelperInterface >& xParent, const css::uno::Reference< css::uno::XComponentContext > & xContext, const css::uno::Reference< css::container::XIndexAccess >& xIndexAccess );
 	virtual ~ScVbaPivotTables() {}
 
 	// XEnumerationAccess
@@ -61,8 +63,10 @@ public:
 
 	// XPivotTables
 
-	// ScVbaCollectionBaseImpl	
+	// ScVbaPivotTables_BASE
 	virtual css::uno::Any createCollectionObject( const css::uno::Any& aSource ); 
+	
+	virtual rtl::OUString& getServiceImplName();
 
 };
 
