--- /dev/null	
+++ sc/source/ui/vba/vbaglobals.cxx	
@@ -0,0 +1,28 @@
+#include <cppuhelper/queryinterface.hxx>
+
+#include "vbaglobals.hxx"
+
+using namespace ::org::openoffice;
+using namespace ::com::sun::star;
+
+// XInterface implementation
+uno::Any ScVbaGlobals::queryInterface( const uno::Type & rType ) throw (uno::RuntimeException)
+{
+    return cppu::queryInterface
+			(rType, 
+			 static_cast< uno::XInterface * >( static_cast< vba::XGlobals * >( this ) ),
+			 static_cast< vba::XGlobals * >( this ) );
+}
+
+uno::Sequence< uno::Reference< uno::XInterface > > ScVbaGlobals::getGlobals()
+{
+		uno::Sequence< uno::Reference< uno::XInterface > > aRet(1);
+		aRet[0] = uno::Reference< uno::XInterface >(new ScVbaGlobals());
+		return aRet;
+}
+
+uno::Reference<vba::XApplication >
+ScVbaGlobals::getApplication() throw (uno::RuntimeException)
+{
+	return uno::Reference< vba::XApplication >(NULL);
+}
