--- sc/source/ui/vba/vbacombobox.cxx	2007-03-02 14:56:07.000000000 +0000
+++ sc/source/ui/vba/vbacombobox.cxx	2007-02-13 18:05:42.000000000 +0000
@@ -47,13 +13,19 @@ const sal_Int32 RANGE_PROPERTY_ID_DFLT=1
 // another property/method of the same name
 const ::rtl::OUString RANGE_PROPERTY_DFLT( RTL_CONSTASCII_USTRINGPARAM( "_$DefaultProp" ) );
 
-ScVbaComboBox::ScVbaComboBox( const uno::Reference< uno::XComponentContext >& xContext, const uno::Reference< beans::XPropertySet >& xProps ) : OPropertyContainer(GetBroadcastHelper()), m_xContext(xContext), m_xProps( xProps ) 
+ScVbaComboBox::ScVbaComboBox( const uno::Reference< uno::XComponentContext >& xContext, const uno::Reference< css::drawing::XControlShape >& xControlShape ) : ComboBoxImpl_BASE( xContext, xControlShape ), OPropertyContainer(GetBroadcastHelper())
 {
 	// grab the default value property name
 	m_xProps->getPropertyValue( CONTROLSOURCEPROP ) >>= sSourceName;
 	setDfltPropHandler();
 }
 
+ScVbaComboBox::ScVbaComboBox( const uno::Reference< uno::XComponentContext >& xContext, const uno::Reference< beans::XPropertySet >& xProps, const css::uno::Reference< css::drawing::XControlShape > xControlShape ) : ComboBoxImpl_BASE( xContext, xProps, xControlShape ), OPropertyContainer(GetBroadcastHelper())
+{
+	m_xProps->getPropertyValue( CONTROLSOURCEPROP ) >>= sSourceName;
+	setDfltPropHandler();
+}
+
 
 // Attributes
 uno::Any SAL_CALL 
