--- sc/source/ui/vba/vbaworkbook.hxx	
+++ sc/source/ui/vba/vbaworkbook.hxx
@@ -87,6 +87,10 @@ public:
 	virtual css::uno::Any SAL_CALL Colors( const css::uno::Any& Index ) throw (css::script::BasicErrorException, css::uno::RuntimeException);
 	virtual ::sal_Int32 SAL_CALL FileFormat(  ) throw (css::script::BasicErrorException, css::uno::RuntimeException);
 	virtual void SAL_CALL SaveCopyAs( const rtl::OUString& Filename ) throw ( css::uno::RuntimeException);
+    // code name
+    virtual ::rtl::OUString SAL_CALL getCodeName() throw ( css::uno::RuntimeException);
+    virtual void SAL_CALL setCodeName( const ::rtl::OUString& sGlobCodeName ) throw (css::uno::RuntimeException);
+    
 	// XHelperInterface
 	virtual rtl::OUString& getServiceImplName();
 	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
