--- /dev/null	
+++ sc/source/ui/vba/vbaapplication.cxx	
@@ -0,0 +1,38 @@
+#include <cppuhelper/queryinterface.hxx>
+
+#include "docuno.hxx"
+#include "vbaapplication.hxx"
+
+using namespace ::org::openoffice;
+using namespace ::com::sun::star;
+
+ScVbaApplication::ScVbaApplication( ScModelObj *pModel )
+{
+	mnRefCount = 1;
+    mpModel = pModel;
+}
+
+ScVbaApplication::~ScVbaApplication()
+{
+}
+
+// XInterface implementation
+uno::Any ScVbaApplication::queryInterface( const uno::Type & rType ) throw (uno::RuntimeException)
+{
+    return cppu::queryInterface
+			(rType, 
+			 static_cast< uno::XInterface * >( static_cast< vba::XApplication * >( this ) ),
+			 static_cast< vba::XApplication * >( this ) );
+}
+
+uno::Reference< vba::XWorkSheet >
+ScVbaApplication::getActiveSheet() throw (uno::RuntimeException)
+{
+	return uno::Reference< vba::XWorkSheet >(NULL);
+}
+
+uno::Reference< vba::XWorkBook >
+ScVbaApplication::getActiveWorkBook() throw (uno::RuntimeException)
+{
+	return uno::Reference< vba::XWorkBook >(NULL);
+}
