--- sc/source/ui/vba/vbaapplication.cxx	
+++ sc/source/ui/vba/vbaapplication.cxx
@@ -504,18 +504,8 @@ ScVbaApplication::GoTo( const uno::Any& 
         ScGridWindow* gridWindow = (ScGridWindow*)pShell->GetWindow(); 
         try
         {
-            uno::Reference< beans::XPropertySet > xProps( getCurrentDocument(), uno::UNO_QUERY_THROW );
-            uno::Reference< container::XNameAccess > xNameAccess( xProps->getPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("NamedRanges") ) ), uno::UNO_QUERY_THROW );
-            sal_Bool isNamedReference = xNameAccess->hasByName( sRangeName ); 
-            uno::Reference< table::XCellRange > xRange;
-
-            if ( isNamedReference )
-                xRange = ScVbaRange::getCellRangeForName( sRangeName, xDoc );
-            else
-                xRange = ScVbaRange::getCellRangeForName( sRangeName, xDoc, ScAddress::CONV_XL_R1C1 );
-
-            ScVbaRange* pRange = new ScVbaRange( this, mxContext, xRange );
-            uno::Reference< excel::XRange > xVbaSheetRange( pRange );
+            uno::Reference< excel::XRange > xVbaSheetRange = ScVbaRange::getRangeObjectForName( mxContext, sRangeName, getDocShell( xModel ), ScAddress::CONV_XL_R1C1 );
+;
             if( bScroll )
             {
                 xVbaSheetRange->Select();
