--- sc/source/ui/vba/vbawindow.cxx	2007-02-28 11:33:18.000000000 +0000
+++ sc/source/ui/vba/vbawindow.cxx	2007-03-02 14:58:02.000000000 +0000
@@ -405,8 +405,6 @@ ScVbaWindow::getWindowState() throw (uno
 	SfxTopViewFrame *pTop= PTR_CAST( SfxTopViewFrame, pViewFrame -> GetTopViewFrame() );
 	if ( pTop )
     {
-	// FOR_UPSTREAM_BUILD
-	/*
         WorkWindow* pWork = (WorkWindow*) pTop->GetTopFrame_Impl()->GetTopWindow_Impl();
         if ( pWork )
         {
@@ -415,7 +413,6 @@ ScVbaWindow::getWindowState() throw (uno
             else if (pWork -> IsMinimized())
                 nwindowState = xlMinimized;
         }
-	*/
     }
     return uno::makeAny( nwindowState );
 }
@@ -430,8 +427,6 @@ ScVbaWindow::setWindowState( const uno::
 	SfxTopViewFrame *pTop= PTR_CAST( SfxTopViewFrame, pViewFrame -> GetTopViewFrame() );
 	if ( pTop )
     {
-	// FOR_UPSTREAM_BUILD
-	/*
         WorkWindow* pWork = (WorkWindow*) pTop->GetTopFrame_Impl()->GetTopWindow_Impl();
         if ( pWork )
         {
@@ -444,7 +439,6 @@ ScVbaWindow::setWindowState( const uno::
             else
                 throw uno::RuntimeException( rtl::OUString(RTL_CONSTASCII_USTRINGPARAM( "Invalid Parameter" ) ), uno::Reference< uno::XInterface >() );
         }
-	*/
     }
 }
 
