--- sc/source/ui/vba/vbaworkbook.cxx	
+++ sc/source/ui/vba/vbaworkbook.cxx
@@ -45,6 +45,7 @@
 #include <com/sun/star/beans/XPropertySet.hpp>
 #include <org/openoffice/excel/XlFileFormat.hpp>
 
+#include "scextopt.hxx"
 #include "vbaworksheet.hxx"
 #include "vbaworksheets.hxx"
 #include "vbaworkbook.hxx"
@@ -63,15 +64,6 @@
 using namespace ::org::openoffice;
 using namespace ::com::sun::star;
 
-namespace workbook
-{
-uno::Reference< vba::XHelperInterface >
-lcl_getParentFromArgs( uno::Sequence< uno::Any > const& args ) throw ( lang::IllegalArgumentException );
-uno::Reference< frame::XModel > 
-lcl_getModelFromArgs( uno::Sequence< uno::Any > const& args ) throw ( lang::IllegalArgumentException );
-};
-
-
 class ActiveSheet : public ScVbaWorksheet
 {
 protected:
@@ -222,7 +214,7 @@ ScVbaWorkbook::ScVbaWorkbook( 	const css
 }
 
 ScVbaWorkbook::ScVbaWorkbook( uno::Sequence< uno::Any> const & args,
-    uno::Reference< uno::XComponentContext> const & xContext ) : ScVbaWorkbook_BASE( workbook::lcl_getParentFromArgs( args ), xContext ),  mxModel( workbook::lcl_getModelFromArgs( args ) ) 
+    uno::Reference< uno::XComponentContext> const & xContext ) : ScVbaWorkbook_BASE( getXSomethingFromArgs< vba::XHelperInterface >( args, 0 ), xContext ),  mxModel( getXSomethingFromArgs< frame::XModel >( args, 1 ) ) 
 
 {
 	init();
@@ -274,17 +266,21 @@ ScVbaWorkbook::Sheets( const uno::Any& a
 {
 	return Worksheets( aIndex );
 }
+
+
+
 uno::Any SAL_CALL
 ScVbaWorkbook::Worksheets( const uno::Any& aIndex ) throw (uno::RuntimeException)
 {
 	uno::Reference< frame::XModel > xModel( getModel() );	
 	uno::Reference <sheet::XSpreadsheetDocument> xSpreadDoc( xModel, uno::UNO_QUERY_THROW );
-	uno::Reference<sheet::XSpreadsheets > xSheets( xSpreadDoc->getSheets(), uno::UNO_QUERY_THROW );
+	uno::Reference<container::XIndexAccess > xSheets( xSpreadDoc->getSheets(), uno::UNO_QUERY_THROW );
 	uno::Reference< vba::XCollection > xWorkSheets(  new ScVbaWorksheets( this, mxContext, xSheets, xModel ) );
 	if (  aIndex.getValueTypeClass() == uno::TypeClass_VOID )
 	{
 		return uno::Any( xWorkSheets );	
 	}
+	// pass on to collection
 	return uno::Any( xWorkSheets->Item( aIndex, uno::Any() ) );
 }
 uno::Any SAL_CALL
@@ -469,25 +465,9 @@ ScVbaWorkbook::getServiceNames()
 	}
 	return aServiceNames;
 }
+
 namespace workbook
 {
-uno::Reference< vba::XHelperInterface >
-lcl_getParentFromArgs( uno::Sequence< uno::Any > const& args ) throw ( lang::IllegalArgumentException )
-{
-        boost::optional< uno::Reference< vba::XHelperInterface >  >xParent;
-        uno::Reference< frame::XModel > xModel;
-        comphelper::unwrapArgs( args, xParent, xModel );
-        return *xParent;
-}
-
-uno::Reference< frame::XModel > 
-lcl_getModelFromArgs( uno::Sequence< uno::Any > const& args ) throw ( lang::IllegalArgumentException )
-{
-        boost::optional< uno::Reference< vba::XHelperInterface >  >xParent;
-        uno::Reference< frame::XModel > xModel;
-        comphelper::unwrapArgs( args, xParent, xModel );
-        return xModel;
-}
 namespace sdecl = comphelper::service_decl;
 sdecl::vba_service_class_<ScVbaWorkbook, sdecl::with_args<true> > serviceImpl;
 extern sdecl::ServiceDecl const serviceDecl(
