--- sc/source/ui/vba//vbapalette.hxx	
+++ sc/source/ui/vba//vbapalette.hxx
@@ -44,7 +44,7 @@ class ScVbaPalette
 private:
 	SfxObjectShell* m_pShell;
 public:
-	ScVbaPalette(  SfxObjectShell* pShell ) : m_pShell( pShell ){}
+	ScVbaPalette(  SfxObjectShell* pShell = NULL ) : m_pShell( pShell ){}
 	// if no palette available e.g. because the document doesn't have a 
 	// palette defined then a default palette will be returned.
 	css::uno::Reference< css::container::XIndexAccess > getPalette();
