--- sc/source/ui/vba//vbavalidation.cxx	
+++ sc/source/ui/vba//vbavalidation.cxx
@@ -317,3 +317,21 @@ ScVbaValidation::getFormula2() throw (un
 	return xCond->getFormula2();
 }
 
+rtl::OUString& 
+ScVbaValidation::getServiceImplName()
+{
+	static rtl::OUString sImplName( RTL_CONSTASCII_USTRINGPARAM("ScVbaValidation") );
+	return sImplName;
+}
+
+uno::Sequence< rtl::OUString > 
+ScVbaValidation::getServiceNames()
+{
+	static uno::Sequence< rtl::OUString > aServiceNames;
+	if ( aServiceNames.getLength() == 0 )
+	{
+		aServiceNames.realloc( 1 );
+		aServiceNames[ 0 ] = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("org.openoffice.excel.Validation" ) );
+	}
+	return aServiceNames;
+}
