--- /dev/null	2005-11-06 11:49:58.372324250 +0000
+++ sc/source/ui/vba/vbawsfunction.hxx	2005-10-27 18:06:14.000000000 +0100
@@ -0,0 +1,36 @@
+#ifndef SC_VBA_INTERIOR_HXX
+#define SC_VBA_INTERIOR_HXX
+
+#include <cppuhelper/implbase1.hxx>
+#include <cppuhelper/implbase2.hxx>
+#include <org/openoffice/vba/XWorksheetFunction.hpp>
+#include <com/sun/star/uno/XComponentContext.hpp>
+#include "vbarange.hxx"
+
+#include <com/sun/star/script/XInvocation.hpp>
+#include <com/sun/star/sheet/XFunctionAccess.hpp>
+
+typedef ::cppu::WeakImplHelper2< oo::vba::XWorksheetFunction,css::script::XInvocation > ScVbaWSFunction_BASE;
+
+class ScVbaWSFunction :  public ScVbaWSFunction_BASE
+{
+	css::uno::Reference< css::uno::XComponentContext > m_xContext;
+public:
+	ScVbaWSFunction( css::uno::Reference< css::uno::XComponentContext >& xContext
+				):m_xContext(xContext)
+	{
+	}
+	virtual ~ScVbaWSFunction(){}
+
+	virtual void SAL_CALL dummy(  ) throw (css::uno::RuntimeException);
+
+	virtual css::uno::Reference< css::beans::XIntrospectionAccess >  SAL_CALL getIntrospection(void)  throw(css::uno::RuntimeException);
+	virtual css::uno::Any  SAL_CALL invoke(const rtl::OUString& FunctionName, const css::uno::Sequence< css::uno::Any >& Params, css::uno::Sequence< sal_Int16 >& OutParamIndex, css::uno::Sequence< css::uno::Any >& OutParam) throw(css::lang::IllegalArgumentException, css::script::CannotConvertException, css::reflection::InvocationTargetException, css::uno::RuntimeException);
+	virtual void  SAL_CALL setValue(const rtl::OUString& PropertyName, const css::uno::Any& Value) throw(css::beans::UnknownPropertyException, css::script::CannotConvertException, css::reflection::InvocationTargetException, css::uno::RuntimeException);
+	virtual css::uno::Any  SAL_CALL getValue(const rtl::OUString& PropertyName) throw(css::beans::UnknownPropertyException, css::uno::RuntimeException);
+	virtual sal_Bool  SAL_CALL hasMethod(const rtl::OUString& Name)  throw(css::uno::RuntimeException);
+	virtual sal_Bool  SAL_CALL hasProperty(const rtl::OUString& Name)  throw(css::uno::RuntimeException);
+
+};
+#endif
+
