diff -rup sc/source/ui/vba/vbawindow.cxx sc/source/ui/vba/vbawindow.cxx
--- sc/source/ui/vba/vbawindow.cxx	2007-02-07 17:08:52.000000000 +0000
+++ sc/source/ui/vba/vbawindow.cxx	2007-02-10 10:10:37.000000000 +0000
@@ -371,8 +371,6 @@ ScVbaWindow::getWindowState() throw (uno
 	SfxTopViewFrame *pTop= PTR_CAST( SfxTopViewFrame, pViewFrame -> GetTopViewFrame() );
 	if ( pTop )
     {
-	// FOR_UPSTREAM_BUILD
-	/*
         WorkWindow* pWork = (WorkWindow*) pTop->GetTopFrame_Impl()->GetTopWindow_Impl();
         if ( pWork )
         {
@@ -381,7 +379,6 @@ ScVbaWindow::getWindowState() throw (uno
             else if (pWork -> IsMinimized())
                 nwindowState = xlMinimized;
         }
-	*/
     }
     return uno::makeAny( nwindowState );
 }
