--- build.lst	2005-08-02 17:14:58.000000000 +0530
+++ sc/prj/build.lst	2005-07-31 09:53:33.000000000 +0530
@@ -11,8 +11,8 @@
 sc	sc\source\filter\inc					get		-	all	sc_fiinc NULL
 sc	sc\addin\inc							get		-	all	sc_adinc NULL
 sc	sc\sdi									nmake	-	all	sc_sdi NULL
-sc      sc\source\ui\vba                                                nmake   -             all     sc_vba sc_sdi NULL
 sc	sc\source\ui\Accessibility						nmake	-	all	sc_acc sc_sdi NULL
+sc	sc\source\ui\vba						nmake	-	all	sc_vba sc_sdi sc_view NULL
 sc	sc\source\ui\pch						get		-	all	sc_upch sc_sdi NULL
 sc	sc\source\ui\view						nmake	-	all	sc_view sc_sdi NULL
 sc	sc\source\ui\app						nmake	-	all	sc_app sc_sdi NULL
