--- sc/inc/scextopt.hxx	2007-09-03 11:16:00.000000000 +0800
+++ sc/inc/scextopt.hxx	2007-08-30 18:15:05.000000000 +0800
@@ -111,7 +111,7 @@
     (struct ScExtDocSettings), settings for all sheets in the document
     (struct ScExtTabSettings), and a list of codenames used for VBA import/export.
  */
-class ScExtDocOptions
+class SC_DLLPUBLIC ScExtDocOptions
 {
 public:
     explicit            ScExtDocOptions();
