Index: svx/source/form/fmctrler.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/form/fmctrler.cxx,v
retrieving revision 1.51
diff -u -r1.51 fmctrler.cxx
--- svx/source/form/fmctrler.cxx	13 Apr 2005 08:28:58 -0000	1.51
+++ svx/source/form/fmctrler.cxx	30 Aug 2005 15:50:53 -0000
@@ -232,6 +232,7 @@
 #ifndef _TOOLKIT_UNOHLP_HXX
 #include <toolkit/helper/vclunohelper.hxx>
 #endif
+
 #ifndef _COMPHELPER_SEQUENCE_HXX_
 #include <comphelper/sequence.hxx>
 #endif
@@ -245,6 +246,8 @@
 #include <algorithm>
 #include <functional>
 
+#include "formControlVBAHandler.hxx"
+
 using namespace ::com::sun::star;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::awt;
@@ -265,6 +268,7 @@
 using namespace ::svxform;
 using namespace ::connectivity::simple;
 
+
 //==================================================================
 // FmXAutoControl
 //==================================================================
@@ -1279,7 +1283,7 @@
 sal_Bool FmXFormController::determineLockState() const
 {
     OSL_ENSURE(!FmXFormController_BASE1::rBHelper.bDisposed,"FmXFormController: Object already disposed!");
-    // a.) in filter mode we are always locked
+    // a.) in filter mode we arealways locked
     // b.) if we have no valid model or our model (a result set) is not alive -> we're locked
     // c.) if we are inserting everything is OK and we are not locked
     // d.) if are not updatable or on invalid position
@@ -1619,6 +1623,8 @@
             {
                 Reference< XInterface >  xIfc(xControl, UNO_QUERY);
                 m_xModelAsManager->attach( nPos, xIfc, makeAny(xControl) );
+                FormControlVBAHandler handler( m_xModelAsManager, xControl );
+                handler.createEventBindings(nPos);
                 break;
             }
         }
@@ -2054,6 +2060,7 @@
     return Reference< XControl > ();
 }
 
+
 //------------------------------------------------------------------------------
 void FmXFormController::implControlInserted( const Reference< XControl>& _rxControl, bool _bAddToEventAttacher )
 {
