--- svx/source/msfilter/msocximex.cxx	2008-09-17 18:22:32.000000000 +0100
+++ svx/source/msfilter/msocximex.cxx	2008-09-17 18:24:52.000000000 +0100
@@ -4279,6 +4279,18 @@ sal_Bool OCX_UserForm::Read(SvStorageStr
 		pS->Read(pIcon,nIconLen);
 	}
 
+
+
+    ReadAlign( pS, pS->Tell() - nStart, 4);
+
+    if (pBlockFlags[2] & 0x10)
+    {
+        //Font Stuff..
+        pS->SeekRel(0x1a);
+        sal_uInt8 nFontLen;
+        *pS >> nFontLen;
+        pS->SeekRel(nFontLen);
+    }
 	if (nPicture)
 	{
 		pS->Read(pPictureHeader,20);
@@ -4293,18 +4305,6 @@ sal_Bool OCX_UserForm::Read(SvStorageStr
 		// make sure the stream position should be pointing after the image
 		pS->Seek( imagePos + nPictureLen );
 	}
-
-    ReadAlign( pS, pS->Tell() - nStart, 4);
-
-    if (pBlockFlags[2] & 0x10)
-    {
-        //Font Stuff..
-        pS->SeekRel(0x1a);
-        sal_uInt8 nFontLen;
-        *pS >> nFontLen;
-        pS->SeekRel(nFontLen);
-    }
-
     sal_Int16 numTrailingRecs = 0;
     *pS >> numTrailingRecs;
     // seems to be no. of trailing records,
