diff --git sc/source/core/tool/interpr1.cxx sc/source/core/tool/interpr1.cxx
index 2ae49af..8699b1b 100644
--- sc/source/core/tool/interpr1.cxx
+++ sc/source/core/tool/interpr1.cxx
@@ -66,6 +66,7 @@
 #include "lookupcache.hxx"
 #include "rangenam.hxx"
 #include "compiler.hxx"
+#include <basic/sbstar.hxx>
 
 #define SC_DOUBLE_MAXVALUE  1.7e307
 
@@ -4092,7 +4093,20 @@ void ScInterpreter::ScMatch()
                 }
             }
             if ( rEntry.bQueryByString )
-                rParam.bRegExp = MayBeRegExp( *rEntry.pStr, pDok );
+            {
+		BOOL bIsVBAMode = FALSE;
+                if ( pDok ) 
+                {
+                   SfxObjectShell* pDocSh = pDok->GetDocumentShell();
+                   if ( pDocSh )
+                        bIsVBAMode = pDocSh->GetBasic()->isVBAEnabled();
+                }
+                // #TODO handle MSO wildcards
+                if ( bIsVBAMode )
+                    rParam.bRegExp = FALSE;
+                else
+                    rParam.bRegExp = MayBeRegExp( *rEntry.pStr, pDok );
+            }
 
             if (pMatSrc) // The source data is matrix array.
             {
