--- sc/source/ui/vba/vbahelper.cxx.orig	2008-08-15 16:40:22.000000000 +0800
+++ sc/source/ui/vba/vbahelper.cxx	2008-08-15 16:40:32.000000000 +0800
@@ -249,7 +249,7 @@ void
 implnPaste()
 {
 	PasteCellsWarningReseter resetWarningBox;
-	ScTabViewShell* pViewShell = ScTabViewShell::GetActiveViewShell();
+	ScTabViewShell* pViewShell = getCurrentBestViewShell();
 	if ( pViewShell )
 	{
 		pViewShell->PasteFromSystem();
