Index: sc/util/makefile.mk
===================================================================
RCS file: /cvs/sc/sc/util/makefile.mk,v
retrieving revision 1.39
diff -u -p -r1.39 makefile.mk
--- sc/util/makefile.mk	25 Apr 2007 16:15:20 -0000	1.39
+++ sc/util/makefile.mk	1 May 2007 08:32:20 -0000
@@ -278,6 +278,7 @@ DEF9NAME=$(SHL9TARGET)
 
 SHL9STDLIBS= \
 		$(CPPUHELPERLIB) \
+		$(VCLLIB)		\
 		$(CPPULIB) \
 		$(COMPHELPERLIB) \
 		$(SVLIB) \
Index: sc/source/ui/vba/vbahelper.hxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/vba/vbahelper.hxx,v
retrieving revision 1.2
retrieving revision 1.2.2.1
diff -u -p -r1.2 -r1.2.2.1
--- sc/source/ui/vba/vbahelper.hxx	25 Apr 2007 16:07:33 -0000	1.2
+++ sc/source/ui/vba/vbahelper.hxx	3 May 2007 11:16:34 -0000	1.2.2.1
@@ -38,8 +38,8 @@
 #include <com/sun/star/frame/XModel.hpp>
 #include <docsh.hxx>
 
-#define css com::sun::star
-#define oo org::openoffice
+namespace css = ::com::sun::star;
+
 namespace org 
 {
 	namespace openoffice 
@@ -69,6 +69,8 @@ namespace org 
 	}
 }
 
+namespace oo = org::openoffice;
+
 #ifdef DEBUG
 #  define SC_VBA_FIXME(a) OSL_TRACE( a )
 #  define SC_VBA_STUB() SC_VBA_FIXME(( "%s - stubbed\n", __FUNCTION__ ))
