diff --git oovbaapi/ooo/vba/XCommandBar.idl oovbaapi/ooo/vba/XCommandBar.idl
index 5e27c6b..13d54da 100644
--- oovbaapi/ooo/vba/XCommandBar.idl
+++ oovbaapi/ooo/vba/XCommandBar.idl
@@ -54,6 +54,7 @@ interface XCommandBar
 
     [attribute] string Name;
     [attribute] boolean Visible;
+    [attribute] boolean Enabled;
     
     void Delete() raises ( com::sun::star::script::BasicErrorException );
     any Controls( [in] any Index ) raises ( com::sun::star::script::BasicErrorException );
diff --git vbahelper/source/vbahelper/vbacommandbar.cxx vbahelper/source/vbahelper/vbacommandbar.cxx
index 55ba7b6..ba9f49b 100644
--- vbahelper/source/vbahelper/vbacommandbar.cxx
+++ vbahelper/source/vbahelper/vbacommandbar.cxx
@@ -267,6 +267,22 @@ ScVbaCommandBar::setVisible( ::sal_Bool _visible ) throw (uno::RuntimeException)
         OSL_TRACE( "SetVisible get an exception\n" );
     }
 }
+
+::sal_Bool SAL_CALL 
+ScVbaCommandBar::getEnabled() throw (uno::RuntimeException)
+{
+    // #FIXME: seems tha main menu is alwayse enabled.
+    // for the toolbar, need to investigate
+    return sal_True;
+}
+
+void SAL_CALL
+ScVbaCommandBar::setEnabled( sal_Bool/* _enabled */ ) throw (uno::RuntimeException)
+{
+    // #FIXME: seems tha main menu is alwayse enabled.
+    // for the toolbar, need to investigate
+}
+
 void SAL_CALL 
 ScVbaCommandBar::Delete(  ) throw (css::script::BasicErrorException, css::uno::RuntimeException)
 {
diff --git vbahelper/source/vbahelper/vbacommandbar.hxx vbahelper/source/vbahelper/vbacommandbar.hxx
index 0b830b2..c53a5cd 100644
--- vbahelper/source/vbahelper/vbacommandbar.hxx
+++ vbahelper/source/vbahelper/vbacommandbar.hxx
@@ -93,6 +93,8 @@ public:
     virtual void SAL_CALL setName( const ::rtl::OUString& _name ) throw (css::uno::RuntimeException);
     virtual ::sal_Bool SAL_CALL getVisible() throw (css::uno::RuntimeException);
     virtual void SAL_CALL setVisible( ::sal_Bool _visible ) throw (css::uno::RuntimeException);
+    virtual ::sal_Bool SAL_CALL getEnabled() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setEnabled( ::sal_Bool _enabled ) throw (css::uno::RuntimeException);
 
     // Methods
     virtual void SAL_CALL Delete(  ) throw (css::script::BasicErrorException, css::uno::RuntimeException);
