--- basic/source/classes/makefile.mk	2007-03-01 16:50:05.000000000 +0100
+++ basic/source/classes/makefile.mk	2007-03-01 16:50:39.000000000 +0100
@@ -53,10 +53,6 @@ ALLTAR : \
         $(MISC)$/$(TARGET).don \
         $(SLOTARGET)
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
-        echo $@
-
 # --- Allgemein -----------------------------------------------------------
 
 COMMON_SLOFILES=	\
@@ -85,3 +81,9 @@ LIB1OBJFILES = $(SLOFILES)
 
 .INCLUDE :  target.mk
 
+$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+        echo $@
+
+$(OBJFILES) : $(MISC)$/$(TARGET).don
+
--- sc/source/ui/vba/makefile.mk	2007-03-01 14:51:01.000000000 +0100
+++ sc/source/ui/vba/makefile.mk	2007-03-01 15:24:47.000000000 +0100
@@ -1,6 +1,6 @@
-PRJ=../..$/..
+PRJ=..$/..$/..
 
-PRJNAME=
+PRJNAME=sc
 TARGET=vbaobj
 .IF "$(ENABLE_VBA)"!="YES"
 dummy:
@@ -21,10 +21,6 @@ ALLTAR : \
         $(MISC)$/$(TARGET).don \
         $(SLOTARGET)
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
-        echo $@
-
 # ------------------------------------------------------------------
 
 SLOFILES= \
@@ -65,3 +61,10 @@ SLOFILES= \
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
+
+$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+        echo $@
+
+$(OBJFILES) : $(MISC)$/$(TARGET).don
+
--- scripting/source/vbaevents/makefile.mk	2007-03-01 16:54:36.000000000 +0100
+++ scripting/source/vbaevents/makefile.mk	2007-03-01 16:56:15.000000000 +0100
@@ -24,9 +24,6 @@ ALLTAR : \
         $(MISC)$/$(TARGET).don \
         $(SLOTARGET)
 
-$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
-        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
-        echo $@
 # ------------------------------------------------------------------
 
 #.INCLUDE :  ..$/cppumaker.mk
@@ -57,3 +54,10 @@ SHL1LIBS=$(SLB)$/$(TARGET).lib
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
+
+$(MISC)$/$(TARGET).don : $(SOLARBINDIR)$/oovbaapi.rdb
+        +$(CPPUMAKER) -O$(OUT)$/inc -BUCR $(SOLARBINDIR)$/oovbaapi.rdb -X$(SOLARBINDIR)$/types.rdb && echo > $@
+        echo $@
+
+$(OBJFILES) : $(MISC)$/$(TARGET).don
+
