diff -u -p -r1.18 exprtree.cxx
--- basic/source/comp/exprtree.cxx	12 Oct 2006 14:27:12 -0000	1.18
+++ basic/source/comp/exprtree.cxx	24 Jul 2007 21:21:14 -0000
@@ -933,11 +968,13 @@ SbiDimList::SbiDimList( SbiParser* p ) :
 		SbiToken eTok;
 		for( ;; )
 		{
-			pExpr1 = new SbiExpression( pParser );
+                        // we have already processed a LPAREN so we are already 
+			// in a nested mode, so... propagate that fact
+			pExpr1 = new SbiExpression( pParser, SbSTDEXPR, TRUE );
 			eTok = pParser->Next();
 			if( eTok == TO )
 			{
-				pExpr2 = new SbiExpression( pParser );
+				pExpr2 = new SbiExpression( pParser, SbSTDEXPR, TRUE );
 				eTok = pParser->Next();
 				bConst &= pExpr1->IsIntConstant() & pExpr2->IsIntConstant();
 				bError |= !pExpr1->IsValid();x
