--- /data4/sles/ooo-buildCopy/ooo-build/build/src680-m125/basic/source/runtime/step2.cxx	2005-09-20 15:48:29.000000000 +0100
+++ basic/source/runtime/step2.cxx	2005-09-21 14:21:00.000000000 +0100
@@ -106,6 +106,27 @@
 		BOOL bFatalError = FALSE;
 		SbxDataType t = (SbxDataType) nOp2;
 		String aName( pImg->GetString( nOp1 & 0x7FFF ) );
+#ifdef ENABLE_VBA
+		SvtFilterOptions* pFiltOpt = SvtFilterOptions::Get();	
+		bool bIsVBAInterOp = pFiltOpt->IsLoadExcelBasicCode(); 
+		// Hacky capture of Evaluate [] syntax
+		// this should be tackled I feel at the pcode level
+		if ( bIsVBAInterOp && aName.Search('[') == 0 ) 
+		{
+			// emulate pcode here
+			StepARGC();
+			// psuedo StepLOADSC
+			String sArg = aName.Copy( 1, aName.Len() - 2 );
+			SbxVariable* p = new SbxVariable;
+			p->PutString( sArg );
+			PushVar( p );
+			//
+			StepARGV();
+			nOp1 = nOp1 | 0x8000; // indicate params are present
+			aName = String::CreateFromAscii("Evaluate");
+		}
+#endif //ENABLE_VBA
+
 		if( bLocal )
 			pElem = refLocals->Find( aName, SbxCLASS_DONTCARE );
 		if( !pElem )
@@ -142,9 +163,6 @@
 					aRes.pObj = pUnoClass;
 					pElem->SbxVariable::Put( aRes );
 				}
-#ifdef ENABLE_VBA
-				SvtFilterOptions* pFiltOpt = SvtFilterOptions::Get();	
-#endif //ENABLE_VBA
 				// #62939 Wenn eine Uno-Klasse gefunden wurde, muss
 				// das Wrapper-Objekt gehalten werden, da sonst auch
 				// die Uno-Klasse, z.B. "stardiv" immer wieder neu
@@ -161,7 +179,7 @@
 					refLocals->Put( pElem, refLocals->Count() );
 				}
 #ifdef ENABLE_VBA
-				else if ( pFiltOpt->IsLoadExcelBasicCode() )
+				else if ( bIsVBAInterOp )
 				{
 					// last gasp see if name matches a 
 					// vba idl constant (with no namespace)
