--- basic/source/classes/sbunoobj.cxx	2008-04-30 14:45:37.000000000 +0100
+++ basic/source/classes/sbunoobj.cxx	2008-05-08 16:10:05.000000000 +0100
@@ -786,7 +786,11 @@ void unoToSbxValue( SbxVariable* pVar, c
 			// SbUnoObject instanzieren
 			String aName;
 			SbUnoObject* pSbUnoObject = new SbUnoObject( aName, aValue );
-			if ( SbiRuntime::isVBAEnabled() )
+//			If this is called externally e.g. from the scripting 
+//			framework then there is no 'active' runtime the default property will not be set up
+//			only a vba object will have XDefaultProp set anyway so... this
+//			test seems a bit of overkill
+//			if ( SbiRuntime::isVBAEnabled() )
 			{
 				String sDfltPropName;
 		
