diff --git sw/source/filter/ww8/ww8toolbar.cxx sw/source/filter/ww8/ww8toolbar.cxx
index dfcaa17..b5648bf 100644
--- sw/source/filter/ww8/ww8toolbar.cxx
+++ sw/source/filter/ww8/ww8toolbar.cxx
@@ -216,7 +216,8 @@ bool Customization::Read( SvStream *pS)
             if (!aTBDelta.Read( pS ) )
                 return false;
             customizationDataTBDelta.push_back( aTBDelta );
-            if ( aTBDelta.ControlDropsToolBar() )
+            // Only set the drop down for menu's associated with standard toolbar
+            if ( aTBDelta.ControlDropsToolBar() && tbidForTBD == 0x25 )
                 pWrapper->InsertDropIndex( aTBDelta.CustomizationIndex() );
         }
     }
