diff --git sc/source/ui/vba/vbaworksheets.cxx sc/source/ui/vba/vbaworksheets.cxx
index f2c0312..a927236 100644
--- sc/source/ui/vba/vbaworksheets.cxx
+++ sc/source/ui/vba/vbaworksheets.cxx
@@ -209,10 +209,23 @@ ScVbaWorksheets::Add( const uno::Any& Before, const uno::Any& After,
 	Count >>= nNewSheets;
 	Type >>= nType;
 	SCTAB nCount = 0;
-	Before >>= aStringSheet;
-	if (!aStringSheet.getLength())
+
+	uno::Reference< excel::XWorksheet > xBeforeAfterSheet;
+
+	if ( Before.hasValue() )
 	{
-		After >>= aStringSheet;
+        	if ( Before >>= xBeforeAfterSheet )
+			aStringSheet = xBeforeAfterSheet->getName();
+		else
+			Before >>= aStringSheet;
+	}
+
+	if (!aStringSheet.getLength() && After.hasValue() )
+	{
+        	if ( After >>= xBeforeAfterSheet )
+			aStringSheet = xBeforeAfterSheet->getName();
+		else
+			After >>= aStringSheet;
 		bBefore = sal_False;
 	}
 	if (!aStringSheet.getLength())
