diff --git a/xmlscript/source/xmldlg_imexp/xmldlg_import.cxx b/xmlscript/source/xmldlg_imexp/xmldlg_import.cxx
index 81d7cd0..45f53ac 100644
--- xmlscript/source/xmldlg_imexp/xmldlg_import.cxx
+++ xmlscript/source/xmldlg_imexp/xmldlg_import.cxx
@@ -1452,7 +1452,18 @@ void ImportContext::importEvents(
                             descr.ScriptCode = buf.makeStringAndClear();
                         }
                     }
-
+                    else if ( descr.ScriptType.equals( OUString( RTL_CONSTASCII_USTRINGPARAM( "Script" ) ) ) )
+                    { 
+                        // Check if there is a protocol, if not assume
+                        // this is an early scripting framework url ( without
+                        // the protocol ) and fix it up!! 
+                        if ( descr.ScriptCode.indexOf( ':' ) == -1 )
+                        {
+                            ::rtl::OUStringBuffer buf;
+                            buf.append( OUString( RTL_CONSTASCII_USTRINGPARAM( "vnd.sun.star.script:" ) ) );
+                            buf.append( descr.ScriptCode );
+                        }
+                    }
 
                     // script:event element
                     if (aLocalName.equalsAsciiL( RTL_CONSTASCII_STRINGPARAM("event") ))
