diff --git forms/source/component/FormComponent.cxx forms/source/component/FormComponent.cxx
index 2919abd..2b73ec6 100644
--- forms/source/component/FormComponent.cxx
+++ forms/source/component/FormComponent.cxx
@@ -2556,8 +2556,8 @@ void SAL_CALL OBoundControlModel::setVal
     OSL_PRECOND( m_bSupportsExternalBinding, "OBoundControlModel::setValueBinding: How did you reach this method?" );
         // the interface for this method should not have been exposed if we do not
         // support binding to external data
-
-    if ( !impl_approveValueBinding_nolock( _rxBinding ) )
+    // allow reset
+    if ( _rxBinding.is() && !impl_approveValueBinding_nolock( _rxBinding ) )
     {
         throw IncompatibleTypesException(
             FRM_RES_STRING( RID_STR_INCOMPATIBLE_TYPES ),
