diff --git vbahelper/source/msforms/vbacombobox.cxx vbahelper/source/msforms/vbacombobox.cxx
index f1ea440..7c27734 100644
--- vbahelper/source/msforms/vbacombobox.cxx
+++ vbahelper/source/msforms/vbacombobox.cxx
@@ -63,9 +63,17 @@ ScVbaComboBox::getValue() throw (uno::RuntimeException)
 void SAL_CALL 
 ScVbaComboBox::setListIndex( const uno::Any& _value ) throw (uno::RuntimeException)
 {
-    uno::Sequence< sal_Int16 > sSelection(1);
-    _value >>= sSelection[ 0 ];
-    m_xProps->setPropertyValue( SELECTEDITEMS, uno::makeAny( sSelection ) );
+    sal_Int16 nIndex = 0;
+    if( _value >>= nIndex )
+    {
+	    uno::Sequence< rtl::OUString > sItems;
+    	m_xProps->getPropertyValue( ITEMS ) >>= sItems;
+        if( ( nIndex >= 0 ) && ( sItems.getLength() > nIndex ) )
+        {
+            rtl::OUString sText = sItems[ nIndex ];
+            m_xProps->setPropertyValue( TEXT, uno::makeAny( sText ) );
+        }
+    }
 }
 
 uno::Any SAL_CALL 
