Index: sc/source/ui/unoobj/docuno.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/unoobj/docuno.cxx,v
retrieving revision 1.50
diff -u -p -u -r1.50 docuno.cxx
--- sc/source/ui/unoobj/docuno.cxx	23 Mar 2005 13:08:32 -0000	1.50
+++ sc/source/ui/unoobj/docuno.cxx	11 May 2005 10:25:42 -0000
@@ -93,6 +93,8 @@
 #include <comphelper/processfactory.hxx>
 #endif
 
+#include "vbaglobals.hxx"
+
 #include "docuno.hxx"
 #include "cellsuno.hxx"
 #include "nameuno.hxx"
@@ -164,6 +170,7 @@ const SfxItemPropertyMap* lcl_GetDocOptP
 		{MAP_CHAR_LEN(SC_UNO_STANDARDDEC),	0,	&getCppuType((sal_Int16*)0),							  0},
         {MAP_CHAR_LEN(SC_UNO_REGEXENABLED), 0,  &getBooleanCppuType(),                                    0},
         {MAP_CHAR_LEN(SC_UNO_RUNTIMEUID),   0,  &getCppuType(static_cast< const rtl::OUString * >(0)),    beans::PropertyAttribute::READONLY},
+        {MAP_CHAR_LEN(SC_VBA_GLOBALS_PROP), 0,  &getCppuType((uno::Sequence<Reference< uno::XInterface > >*)0), beans::PropertyAttribute::READONLY},
 
 		{0,0,0,0}
 	};
@@ -1476,6 +1483,10 @@ uno::Any SAL_CALL ScModelObj::getPropert
         {
             aRet <<= getRuntimeUID();
         }
+		else if ( aString.EqualsAscii( SC_VBA_GLOBALS_PROP ) )
+		{
+			aRet <<= ScVbaGlobals::getGlobals();
+		}
 	}
 
 	return aRet;
