diff --git sc/source/ui/view/tabview3.cxx sc/source/ui/view/tabview3.cxx
index c71049a..1a87bfb 100644
--- sc/source/ui/view/tabview3.cxx
+++ sc/source/ui/view/tabview3.cxx
@@ -1439,10 +1439,15 @@ void ScTabView::MarkRange( const ScRange& rRange, BOOL bSetCursor, BOOL bContinu
 	{
 		SCCOL nAlignX = rRange.aStart.Col();
 		SCROW nAlignY = rRange.aStart.Row();
-		if ( rRange.aStart.Col() == 0 && rRange.aEnd.Col() == MAXCOL )
-			nAlignX = aViewData.GetPosX(WhichH(aViewData.GetActivePart()));
-		if ( rRange.aStart.Row() == 0 && rRange.aEnd.Row() == MAXROW )
-			nAlignY = aViewData.GetPosY(WhichV(aViewData.GetActivePart()));
+		bool bCol = ( rRange.aStart.Col() == 0 && rRange.aEnd.Col() == MAXCOL );
+		bool bRow = ( rRange.aStart.Row() == 0 && rRange.aEnd.Row() == MAXROW );
+		if ( !bCol && !bRow )
+		{
+			if ( bCol ) 
+				nAlignX = aViewData.GetPosX(WhichH(aViewData.GetActivePart()));
+			if ( bRow )
+				nAlignY = aViewData.GetPosY(WhichV(aViewData.GetActivePart()));
+		}
 		AlignToCursor( nAlignX, nAlignY, SC_FOLLOW_JUMP );
 	}
 	InitBlockMode( rRange.aStart.Col(), rRange.aStart.Row(), nTab );
