--- oovbaapi/org/openoffice/excel/XInterior.idl.orig	2008-02-29 17:01:26.000000000 +0800
+++ oovbaapi/org/openoffice/excel/XInterior.idl	2008-02-29 16:29:29.000000000 +0800
@@ -56,6 +56,9 @@ interface XInterior
 
 	[attribute]  any  Color;
 	[attribute]  any  ColorIndex;
+	[attribute]  any  Pattern;
+	[attribute]  any  PatternColor;
+	[attribute]  any  PatternColorIndex;
 };
 
 //=============================================================================
--- sc/source/ui/vba/vbainterior.hxx.orig	2008-02-29 16:48:30.000000000 +0800
+++ sc/source/ui/vba/vbainterior.hxx	2008-02-29 17:19:05.000000000 +0800
@@ -64,6 +64,12 @@ public:
 
 	virtual css::uno::Any SAL_CALL getColorIndex() throw ( css::uno::RuntimeException);
 	virtual void SAL_CALL setColorIndex( const css::uno::Any& _colorindex ) throw ( css::uno::RuntimeException );
+    virtual css::uno::Any SAL_CALL getPattern() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setPattern( const css::uno::Any& _pattern ) throw (css::uno::RuntimeException);
+    virtual css::uno::Any SAL_CALL getPatternColor() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setPatternColor( const css::uno::Any& _patterncolor ) throw (css::uno::RuntimeException);
+    virtual css::uno::Any SAL_CALL getPatternColorIndex() throw (css::uno::RuntimeException);
+    virtual void SAL_CALL setPatternColorIndex( const css::uno::Any& _patterncolorindex ) throw (css::uno::RuntimeException); 
 	//XHelperInterface
 	virtual rtl::OUString& getServiceImplName();
 	virtual css::uno::Sequence<rtl::OUString> getServiceNames();
--- sc/source/ui/vba/vbainterior.cxx.orig	2008-02-29 16:48:30.000000000 +0800
+++ sc/source/ui/vba/vbainterior.cxx	2008-03-03 16:53:25.000000000 +0800
@@ -41,6 +41,7 @@
 #include <com/sun/star/beans/MethodConcept.hpp>
 
 #include <org/openoffice/excel/XlColorIndex.hpp>
+#include <org/openoffice/excel/XlPattern.hpp>
 
 #include <comphelper/processfactory.hxx>
 #include <cppuhelper/queryinterface.hxx>
@@ -125,6 +126,42 @@ ScVbaInterior::getColorIndex() throw ( c
 	}     
 	return uno::makeAny( nIndex );
 }
+// OOo do not support below API
+uno::Any SAL_CALL 
+ScVbaInterior::getPattern() throw (uno::RuntimeException)
+{
+    // XlPattern
+    return uno::makeAny( excel::XlPattern::xlPatternNone );
+}
+void SAL_CALL 
+ScVbaInterior::setPattern( const uno::Any& _pattern ) throw (uno::RuntimeException)
+{
+    (void) _pattern;
+}
+uno::Any SAL_CALL 
+ScVbaInterior::getPatternColor() throw (uno::RuntimeException)
+{
+    // 0 is the default color. no filled.
+    // RGB
+    return getColor();
+}
+void SAL_CALL 
+ScVbaInterior::setPatternColor( const uno::Any& _patterncolor ) throw (uno::RuntimeException)
+{
+    setColor( _patterncolor );
+}
+uno::Any SAL_CALL 
+ScVbaInterior::getPatternColorIndex() throw (uno::RuntimeException)
+{
+    // 0 means color is not set. 
+    // XlColorIndex
+    return getColorIndex();
+}
+void SAL_CALL 
+ScVbaInterior::setPatternColorIndex( const uno::Any& _patterncolorindex ) throw (uno::RuntimeException)
+{
+    setColorIndex( _patterncolorindex );
+}
 
 rtl::OUString& 
 ScVbaInterior::getServiceImplName()
