--- sc/source/ui/vba/vbawsfunction.cxx	2009-02-02 15:33:32.000000000 +0000
+++ sc/source/ui/vba/vbawsfunction.cxx.patched	2009-02-02 16:18:58.000000000 +0000
@@ -88,12 +88,13 @@ ScVbaWSFunction::invoke(const rtl::OUStr
 			count, rtl::OUStringToOString( comphelper::anyToString( aParamTemp[count] ), RTL_TEXTENCODING_UTF8 ).getStr()  );
 	
 	uno::Any aRet = xFunctionAccess->callFunction(FunctionName,aParamTemp);
+
 	// MATCH function should alwayse return a double value, but currently if the first argument is XCellRange, MATCH function returns an array instead of a double value. Don't know why?
 	// To fix this issue in safe, current solution is to convert this array to a double value just for MATCH function.
 	String aUpper( FunctionName );
 	ScCompiler aCompiler( NULL, ScAddress() ); 
 	OpCode eOp = aCompiler.GetEnglishOpCode( aUpper.ToUpperAscii() );
-	if( eOp == ocMatch )
+	if( eOp == ocMatch || eOp == ocIsError )
 	{
 		double fVal = 0.0;
 		if( aRet >>= fVal )
