--- oovbaapi/org/openoffice/msforms/XScrollBar.idl.bak	2008-07-03 15:26:45.000000000 +0800
+++ oovbaapi/org/openoffice/msforms/XScrollBar.idl	2008-07-03 15:27:10.000000000 +0800
@@ -32,8 +32,8 @@
  *    MA  02111-1307  USA
  *
  ************************************************************************/
-#ifndef __org_openoffice_msforms_XRadioButton_idl__
-#define __org_openoffice_msforms_XRadioButton_idl__
+#ifndef __org_openoffice_msforms_XScrollBar_idl__
+#define __org_openoffice_msforms_XScrollBar_idl__
 
 #ifndef __com_sun_star_uno_XInterface_idl__
 #include <com/sun/star/uno/XInterface.idl>
