diff --git svx/source/msfilter/svxmsbas.cxx svx/source/msfilter/svxmsbas.cxx
index 3b2feb0..2826ed4 100644
--- svx/source/msfilter/svxmsbas.cxx
+++ svx/source/msfilter/svxmsbas.cxx
@@ -457,8 +457,15 @@ BOOL SvxImportMSVBasic::ImportCode_Impl( const String& rStorageName,
 				// get the api object for the codename
 				if ( xVBACodeNamedObjectAccess.is() )
 				{
-					sModuleInfo.ModuleObject.set( xVBACodeNamedObjectAccess->getByName( sModuleInfo.ModuleName ), uno::UNO_QUERY );    
-					OSL_TRACE("** Straight up creation of Module");
+					try
+					{
+						sModuleInfo.ModuleObject.set( xVBACodeNamedObjectAccess->getByName( sModuleInfo.ModuleName ), uno::UNO_QUERY );    
+						OSL_TRACE("** Straight up creation of Module");
+					}
+					catch(uno::Exception& e)
+					{
+						OSL_TRACE("Failed to get documument object for %s", rtl::OUStringToOString( sModuleInfo.ModuleName, RTL_TEXTENCODING_UTF8 ).getStr() );
+					}
 				}
 			}
                         aSourceAny <<= sModuleInfo;
