diff --git vbahelper/source/vbahelper/vbadocumentsbase.cxx vbahelper/source/vbahelper/vbadocumentsbase.cxx
index 813e084..6d5ddb3 100644
--- vbahelper/source/vbahelper/vbadocumentsbase.cxx
+++ vbahelper/source/vbahelper/vbadocumentsbase.cxx
@@ -45,6 +45,7 @@
 #include <com/sun/star/util/XCloseable.hpp>
 #include <com/sun/star/lang/IndexOutOfBoundsException.hpp>
 #include <com/sun/star/document/XTypeDetection.hpp>
+#include <com/sun/star/document/MacroExecMode.hpp>
 #include <com/sun/star/uri/XUriReference.hpp>
 #include <com/sun/star/uri/XUriReferenceFactory.hpp>
 #include <com/sun/star/lang/XServiceInfo.hpp>
@@ -271,8 +272,13 @@ VbaDocumentsBase::Open( const rtl::OUString& rFileName, const uno::Any& ReadOnly
 		::rtl::OUString::createFromAscii("com.sun.star.frame.Desktop"),
 		mxContext),
 		uno::UNO_QUERY_THROW );
+
 	uno::Sequence< beans::PropertyValue > sProps( rProps );
-	sal_Int32 nIndex = 0;
+	sProps.realloc( sProps.getLength() + 1 );
+	sProps[ sProps.getLength() - 1 ].Name = rtl::OUString( RTL_CONSTASCII_USTRINGPARAM("MacroExecutionMode") );
+	sProps[ sProps.getLength() - 1 ].Value <<= uno::makeAny( document::MacroExecMode::ALWAYS_EXECUTE_NO_WARN );
+
+	sal_Int32 nIndex = sProps.getLength() - 1;
 	
 	if ( ReadOnly.hasValue()  )
 	{
