Index: basic/source/inc/runtime.hxx
===================================================================
RCS file: /cvs/script/basic/source/inc/runtime.hxx,v
retrieving revision 1.25
diff -u -p -r1.25 runtime.hxx
--- basic/source/inc/runtime.hxx	29 Sep 2005 18:41:20 -0000	1.25
+++ basic/source/inc/runtime.hxx	1 Dec 2005 13:24:13 -0000
@@ -491,6 +491,8 @@ public:
 	SbxArray* GetParams();
 
 	SbxBase* FindElementExtern( const String& rName );
+	static bool isVBAEnabled();
+
 };
 
 inline void checkArithmeticOverflow( double d )
Index: basic/source/runtime/runtime.cxx
===================================================================
RCS file: /cvs/script/basic/source/runtime/runtime.cxx,v
retrieving revision 1.24
diff -u -p -r1.24 runtime.cxx
--- basic/source/runtime/runtime.cxx	29 Sep 2005 18:42:01 -0000	1.24
+++ basic/source/runtime/runtime.cxx	1 Dec 2005 13:24:22 -0000
@@ -66,6 +66,15 @@
 // Makro MEMBER()
 #include <macfix.hxx>
 
+#include <svtools/fltrcfg.hxx>
+bool SbiRuntime::isVBAEnabled()
+{
+	bool result = false;
+	SvtFilterOptions* pFiltOpt = SvtFilterOptions::Get();
+	if ( pFiltOpt )
+		result = pFiltOpt->IsLoadExcelBasicCode(); 
+	return result; 
+}
 
 // #91147 Global reschedule flag
 static BOOL bStaticGlobalEnableReschedule = TRUE;
