diff --git sc/source/ui/vba/vbarange.cxx sc/source/ui/vba/vbarange.cxx
index 3b62d7b..c149923 100644
--- sc/source/ui/vba/vbarange.cxx
+++ sc/source/ui/vba/vbarange.cxx
@@ -3636,12 +3642,14 @@ ScVbaRange::getRowHeight() throw (uno::R
 	double nHeight = getCalcRowHeight( thisAddress );
 	// #TODO probably possible to use the SfxItemSet ( and see if
 	//  SFX_ITEM_DONTCARE is set ) to improve performance
-	if ( mbIsRows ) 
+        // seems not just for Rows but ordinary RowHeights too
+//	if ( mbIsRows ) 
 	{
 		for ( sal_Int32 nRow = nStartRow ; nRow <= nEndRow; ++nRow )
 		{
 			thisAddress.StartRow = nRow;
 			double nCurHeight = getCalcRowHeight( thisAddress );
+			OSL_TRACE("row %d nHeight %f nCurHeight %f", nRow, nHeight, nCurHeight );
 			if ( nHeight != nCurHeight )
 				return aNULL();
 		}
