Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.128
diff -u -r1.128 configure.in
--- config_office/configure.in	14 Jul 2005 13:45:35 -0000	1.128
+++ config_office/configure.in	28 Jul 2005 17:15:04 -0000
@@ -26,6 +26,9 @@
 AC_ARG_ENABLE(lockdown,
 [  --enable-lockdown       Enables the gconf integration work in OOo
 ],,)
+AC_ARG_ENABLE(vba,
+[  --disable-vba           Disables the compilation of the vba feature
+],,)
 AC_ARG_ENABLE(libsn,
 [  --enable-libsn          Enables the use of libstartup-notification
 ],,)
@@ -1306,6 +1309,16 @@
    AC_HEADER_STDC
 fi
 
+AC_MSG_CHECKING([whether to disable vba feature])
+if test -n "$enable_vba" && test "$enable_vba" = "no"; then
+   AC_MSG_RESULT([yes])
+   ENABLE_VBA=NO
+else
+   ENABLE_VBA=YES
+   AC_MSG_RESULT([no])
+fi
+AC_SUBST(ENABLE_VBA)
+
 dnl ===================================================================
 dnl Check if cups/cups.h is available 
 dnl ===================================================================

Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.70
diff -u -r1.70 set_soenv.in
--- config_office/set_soenv.in	14 Jul 2005 13:45:49 -0000	1.70
+++ config_office/set_soenv.in	28 Jul 2005 17:15:09 -0000
@@ -1482,6 +1482,7 @@
 ToFile( "MOZILLA_VERSION",   $MOZILLA_VERSION,   "e" );
 ToFile( "DEFAULT_MOZILLA_TOOLKIT", $MOZILLA_TOOLKIT, "e" );
 ToFile( "BUILD_VER_STRING",  "@BUILD_VER_STRING@", "e" );
+ToFile( "ENABLE_VBA",  "@ENABLE_VBA@", "e" );
 if ($platform =~ m/linux-gnu/ && $platform =~ m/powerpc/) {
 ToFile( "JITC_PROCESSOR_TYPE","6",                 "e" );
 }
