--- vcl/unx/gtk/window/gtkframe.cxx.bak	2004-11-24 11:28:39.000000000 +0530
+++ vcl/unx/gtk/window/gtkframe.cxx	2004-11-30 10:51:59.785225136 +0530
@@ -76,6 +76,8 @@
 #include <dlfcn.h>
 #include <soicon.hxx>
 
+#include <svapp.hxx>
+
 #if OSL_DEBUG_LEVEL > 1
 #include <cstdio>
 #endif
@@ -795,6 +797,8 @@
 	{
 		if( m_pParent )
 		{
+		        if( Application::GetSettings().GetLayoutRTL() )
+			    nX = m_pParent->maGeometry.nWidth-maGeometry.nWidth-1-nX;
 			nX += m_pParent->maGeometry.nX;
 			nY += m_pParent->maGeometry.nY;
 		}
@@ -1568,6 +1570,10 @@
             gdk_display_pointer_ungrab( pThis->getGdkDisplay(), GDK_CURRENT_TIME );
     }
 
+    // --- RTL --- (mirror mouse pos)
+    if( Application::GetSettings().GetLayoutRTL() )
+         aEvent.mnX = pThis->maGeometry.nWidth-1-aEvent.mnX;
+
     GTK_YIELD_GRAB();
 	
 	vcl::DeletionListener aDel( pThis );
@@ -1628,6 +1630,10 @@
     aEvent.mnCode			= GetModCode( pSEvent->state );
     aEvent.mbHorz			= (pSEvent->direction == GDK_SCROLL_LEFT || pSEvent->direction == GDK_SCROLL_RIGHT);
     
+    // --- RTL --- (mirror mouse pos)
+    if( Application::GetSettings().GetLayoutRTL() )
+         aEvent.mnX = pThis->maGeometry.nWidth-1-aEvent.mnX;
+
     GTK_YIELD_GRAB();
     pThis->CallCallback( SALEVENT_WHEELMOUSE, &aEvent );
     
@@ -1645,7 +1645,10 @@
     aEvent.mnCode	= GetModCode( pEvent->state );
     aEvent.mnButton	= 0;
     
     
+    // --- RTL --- (mirror mouse pos)
+    if( Application::GetSettings().GetLayoutRTL() )
+         aEvent.mnX = pThis->maGeometry.nWidth-1-aEvent.mnX;
     GTK_YIELD_GRAB();
 
  	vcl::DeletionListener aDel( pThis );
@@ -1703,6 +1702,10 @@
     aEvent.mnBoundWidth		= pEvent->area.width;
     aEvent.mnBoundHeight	= pEvent->area.height;
 
+    // --- RTL --- (mirror mouse pos)
+    if( Application::GetSettings().GetLayoutRTL() )
+         aEvent.mnBoundX = pThis->maGeometry.nWidth-aEvent.mnBoundWidth-aEvent.mnBoundX;
+
     GTK_YIELD_GRAB();
     pThis->CallCallback( SALEVENT_PAINT, &aEvent );
     
