--- glib-1.2.10/makefile.msc	Thu Mar 15 17:48:49 2001
+++ glib-1.2.10/makefile.msc	Fri Oct 14 12:30:05 2005
@@ -1,23 +1,20 @@
-## Makefile for building the GLib, gmodule and gthread dlls with Microsoft C
+## Makefile for building the GLib and gmodule dlls with Microsoft C
 ## Use: nmake -f makefile.msc install
+# THREAD SUPPORT HAS BEEN HACKED OUT
 
 # Change this to wherever you want to install the DLLs. This directory
 # should be in your PATH.
-BIN = C:\bin
-
-# This is the location of pthreads for Win32, 
-# see http://sourceware.cygnus.com/pthreads-win32/
-# We want the 1999-04-07 snapshot.
-PTHREADS = ..\pthreads-snap-1999-04-07
-PTHREAD_LIB = $(PTHREADS)\pthread.lib
-PTHREAD_INC = -I$(PTHREADS)
+PREFIX=.
+BIN = $(PREFIX)\bin
+LIBDIR = $(PREFIX)\lib
+INCDIR = $(PREFIX)\include
 
 ################################################################
 
 # Nothing much configurable below
 
 # cl -? describes the options
-CC = cl -G5 -GF -Ox -W3 -MD -nologo
+CC = cl -GF -Ox -W3 -MD -D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -nologo
 
 # No general LDFLAGS needed
 LDFLAGS = /link
@@ -33,17 +30,18 @@
 	glib-$(GLIB_VER).dll	\
 	gmodule\gmoduleconf.h \
 	gmodule-$(GLIB_VER).dll	\
-	gthread-$(GLIB_VER).dll \
 	testglib.exe	\
 	testgmodule.exe	\
 	testgdate.exe	\
 	testgdateparser.exe \
-	testgthread.exe
 
 install : all
 	$(INSTALL) glib-$(GLIB_VER).dll $(BIN)
 	$(INSTALL) gmodule-$(GLIB_VER).dll $(BIN)
-	$(INSTALL) gthread-$(GLIB_VER).dll $(BIN)
+	$(INSTALL) glib-$(GLIB_VER).lib $(LIBDIR)
+	$(INSTALL) gmodule-$(GLIB_VER).lib $(LIBDIR)
+	$(INSTALL) glib.h $(INCDIR)
+	$(INSTALL) glibconfig.h $(INCDIR)
 
 glib_OBJECTS = \
 	garray.obj	\
@@ -80,7 +80,7 @@
 	copy config.h.win32 config.h
 
 .c.obj :
-	$(CC) $(CFLAGS) -GD -c -DGLIB_COMPILATION -DG_LOG_DOMAIN=g_log_domain_glib $<
+	$(CC) $(CFLAGS) -c -DGLIB_COMPILATION -DG_LOG_DOMAIN=g_log_domain_glib $<
 
 gmodule_OBJECTS = \
 	gmodule.obj
@@ -96,15 +94,6 @@
 gmodule\gmoduleconf.h: gmodule\gmoduleconf.h.win32
 	copy gmodule\gmoduleconf.h.win32 gmodule\gmoduleconf.h
 
-gthread_OBJECTS = \
-	gthread.obj
-
-gthread-$(GLIB_VER).dll : $(gthread_OBJECTS)
-	$(CC) $(CFLAGS) -LD -Fegthread-$(GLIB_VER).dll $(gthread_OBJECTS) glib-$(GLIB_VER).lib $(PTHREAD_LIB) glib-$(GLIB_VER).lib $(LDFLAGS) /def:gthread\gthread.def
-
-gthread.obj : gthread\gthread.c gthread\gthread-posix.c
-	$(CC) $(CFLAGS) -GD $(PTHREAD_INC) -DG_LOG_DOMAIN=\"GThread\" -c gthread\gthread.c
-
 testglib.exe : glib-$(GLIB_VER).dll testglib.obj
 	$(CC) $(CFLAGS) -Fetestglib.exe testglib.obj glib-$(GLIB_VER).lib $(LDFLAGS) /subsystem:console
 
@@ -141,25 +130,19 @@
 libgplugin_b.obj : gmodule\libgplugin_b.c
 	$(CC) $(CFLAGS) -Igmodule -c gmodule\libgplugin_b.c
 
-testgthread.exe : glib-$(GLIB_VER).dll gthread-$(GLIB_VER).dll testgthread.obj
-	$(CC) $(CFLAGS) testgthread.obj glib-$(GLIB_VER).lib gthread-$(GLIB_VER).lib $(PTHREAD_LIB) $(LDFLAGS) /subsystem:console 
-
-testgthread.obj : gthread\testgthread.c
-	$(CC) $(CFLAGS) $(PTHREAD_INC) -c gthread\testgthread.c
-
 clean:
-	del config.h
-	del glibconfig.h
-	del gmodule\gmoduleconf.h
-	del *.exe
-	del *.obj
-	del *.dll
-	del *.lib
-	del *.err
-	del *.map
-	del *.sym
-	del *.exp
-	del *.lk1
-	del *.mk1
-	del *.pdb
-	del *.ilk
+	-del config.h
+	-del glibconfig.h
+	-del gmodule\gmoduleconf.h
+	-del *.exe
+	-del *.obj
+	-del *.dll
+	-del *.lib
+	-del *.err
+	-del *.map
+	-del *.sym
+	-del *.exp
+	-del *.lk1
+	-del *.mk1
+	-del *.pdb
+	-del *.ilk
--- glib-1.2.10/glib.h
+++ glib-1.2.10/glib.h
@@ -2635,18 +2635,7 @@
  * For some functions, we provide emulators in glib, which are prefixed
  * with gwin_.
  */
-#    define getcwd		_getcwd
-#    define getpid		_getpid
-#    define access		_access
-#    define open		_open
-#    define read		_read
-#    define write		_write
-#    define lseek		_lseek
-#    define close		_close
 #    define pipe(phandles)	_pipe (phandles, 4096, _O_BINARY)
-#    define popen		_popen
-#    define pclose		_pclose
-#    define fdopen		_fdopen
 #    define ftruncate(fd, size)	gwin_ftruncate (fd, size)
 #    define opendir		gwin_opendir
 #    define readdir		gwin_readdir
