--- bridges/source/cpp_uno/gcc3_linux_x86-64/uno2cpp.cxx	16 Sep 2006 15:50:53 -0000	1.6
+++ bridges/source/cpp_uno/gcc3_linux_x86-64/uno2cpp.cxx	21 Nov 2006 16:15:19 -0000
@@ -122,7 +122,7 @@ invoke_copy_to_stack(sal_uInt64 * pDS,  
                 // value verbatim, as a double without conversion.
 				pFPR[nr_fpr++] = *reinterpret_cast<double *>( pSV++ );
 			else
-				*pDS++ = *reinterpret_cast<double *>( pSV++ );
+				*pDS++ = *pSV++;
 			break;
 
 		case 'H': // 64-bit Word
