Index: unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.53.4.1
diff -u -r1.53.4.1 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	10 Aug 2006 14:53:49 -0000	1.53.4.1
+++ vcl/unx/gtk/window/gtkframe.cxx	12 Oct 2006 14:30:50 -0000
@@ -647,6 +650,9 @@
 AtkRole
 GtkSalFrame::GetAtkRole( GtkWindow* window )
 {
+    if (GTK_IS_FILE_CHOOSER_DIALOG(window))
+	return ATK_ROLE_FILE_CHOOSER;
+
     static AtkRole aDefaultRole = ATK_ROLE_INVALID;
     
     // Special role for sub-menu and combo-box popups that are exposed directly
