--- extras/util/target.orig	2005-09-09 00:22:16.000000000 +0200
+++ extras/util/target.pmk	2005-12-30 10:29:41.754670700 +0100
@@ -76,6 +76,53 @@
 GALLERY_WWWBACK_TARGET          = gallwwwback
 GALLERY_WWWGRAF_TARGET          = gallwwwgraf

+GALLERY_ACCESSORIES_ROOT_TARGET = kamigallery
+GALLERY_ACCESSORIES_DRAWS_PEOPLE_TARGET = kamigallerydrawspeople
+GALLERY_ACCESSORIES_PHOTOS_BUILDINGS_TARGET = kamigalleryphotosbuildings
+GALLERY_ACCESSORIES_PHOTOS_CELEBRATION_TARGET = kamigalleryphotoscelebration
+GALLERY_ACCESSORIES_PHOTOS_CITIES_TARGET = kamigalleryphotoscities
+GALLERY_ACCESSORIES_PHOTOS_FAUNA_TARGET = kamigalleryphotosfauna
+GALLERY_ACCESSORIES_PHOTOS_FLOWERS_TARGET = kamigalleryphotosflowers
+GALLERY_ACCESSORIES_PHOTOS_FOODSANDDRINKS_TARGET = kamigalleryphotosfoodsanddrinks
+GALLERY_ACCESSORIES_PHOTOS_HUMANS_TARGET = kamigalleryphotoshumans
+GALLERY_ACCESSORIES_PHOTOS_LANDSCAPES_TARGET = kamigalleryphotoslandscapes
+GALLERY_ACCESSORIES_PHOTOS_OBJECTS_TARGET = kamigalleryphotosobjects
+GALLERY_ACCESSORIES_PHOTOS_PLANTS_TARGET = kamigalleryphotosplants
+GALLERY_ACCESSORIES_PHOTOS_SPACE_TARGET = kamigalleryphotosspace
+GALLERY_ACCESSORIES_PHOTOS_STATUES_TARGET = kamigalleryphotosstatues
+GALLERY_ACCESSORIES_PHOTOS_TRAVEL_TARGET = kamigalleryphotostravel
+GALLERY_ACCESSORIES_ELEMENTS_BULLETS2_TARGET = kamigalleryelementsbullets2
+
+GALLERY_ACCESSORIES_NONFREE_ROOT_TARGET = kamigallerynonfree
+
+PREMIUM_TEMPLATES_LAYOUT_IMPRESS_TARGET     = kamitemplateslayout
+PREMIUM_TEMPLATES_PRESNT_IMPRESS_TARGET     = kamitemplatespresent
+PREMIUM_TEMPLATES_EDUCATE_TARGET            = kamitemplateseducate
+PREMIUM_TEMPLATES_FINANCE_TARGET            = kamitemplatesfinance
+PREMIUM_TEMPLATES_FORMS_TARGET              = kamitemplatesforms
+PREMIUM_TEMPLATES_MISC_TARGET               = kamitemplatesmisc
+PREMIUM_TEMPLATES_OFFICORR_TARGET           = kamitemplatesofficorr
+PREMIUM_TEMPLATES_OFFIMISC_TARGET           = kamitemplatesoffimisc
+PREMIUM_TEMPLATES_PERSONAL_TARGET           = kamitemplatespersonal
+PREMIUM_TEMPLATES_ROOT_TARGET               = kamitemplates
+PREMIUM_SAMPLES_ROOT_TARGET                 = kamisamples
+PREMIUM_SAMPLES_DOCUMENTATION_TARGET        = kamisamplesdocumentation
+PREMIUM_SAMPLES_ADVERTISEMENT_TARGET        = kamisamplesadvertisement
+
+PREMIUM_TEMPLATES_NONFREE_LAYOUT_IMPRESS_TARGET     = kamitemplatesnonfreelayout
+PREMIUM_TEMPLATES_NONFREE_PRESNT_IMPRESS_TARGET     = kamitemplatesnonfreepresent
+PREMIUM_TEMPLATES_NONFREE_EDUCATE_TARGET            = kamitemplatesnonfreeeducate
+PREMIUM_TEMPLATES_NONFREE_FINANCE_TARGET            = kamitemplatesnonfreefinance
+PREMIUM_TEMPLATES_NONFREE_FORMS_TARGET              = kamitemplatesnonfreeforms
+PREMIUM_TEMPLATES_NONFREE_MISC_TARGET               = kamitemplatesnonfreemisc
+PREMIUM_TEMPLATES_NONFREE_OFFICORR_TARGET           = kamitemplatesnonfreeofficorr
+PREMIUM_TEMPLATES_NONFREE_OFFIMISC_TARGET           = kamitemplatesnonfreeoffimisc
+PREMIUM_TEMPLATES_NONFREE_PERSONAL_TARGET           = kamitemplatesnonfreepersonal
+PREMIUM_TEMPLATES_NONFREE_ROOT_TARGET               = kamitemplatesnonfree
+PREMIUM_SAMPLES_NONFREE_ROOT_TARGET                 = kamisamplesnonfree
+PREMIUM_SAMPLES_NONFREE_DOCUMENTATION_TARGET        = kamisamplesnonfreedocumentation
+PREMIUM_SAMPLES_NONFREE_ADVERTISEMENT_TARGET        = kamisamplesnonfreeadvertisement
+
 INSO_DUTCH_TARGET               = inso31
 INSO_ENGLISH_UK_TARGET          = inso44
 INSO_ENGLISH_US_TARGET          = inso01
