Index: configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.110
diff -u -u -r1.110 configure.in
--- config_office/configure.in	17 Mar 2005 12:15:54 -0000	1.110
+++ config_office/configure.in	26 Mar 2005 11:28:13 -0000
@@ -3523,106 +3523,6 @@
 AC_SUBST(USE_XINERAMA)
 AC_SUBST(XINERAMA_LINK)
 
-dnl ===================================================================
-dnl Test for the presence of Ant and that it works
-dnl ===================================================================
-
-if test "$SOLAR_JAVA" != ""; then
-ANT_HOME=; export ANT_HOME
-WITH_ANT_HOME=; export WITH_ANT_HOME
-if test -z "$with_ant_home"; then
-   AC_PATH_PROGS(ANT, [jakarta-ant ant ant.sh ant.bat])
-else
-   AC_PATH_PROGS(ANT, [jakarta-ant ant ant.sh ant.bat],,$with_ant_home/bin:$PATH)
-   WITH_ANT_HOME=$with_ant_home
-fi
-
-if test -z "$ANT"; then
-  AC_MSG_ERROR([Ant not found - Make sure it's in the path or use --with-ant-home])
-else
-  # resolve relative or absolute symlink
-  while test -h "$ANT"; do
-     a_cwd=`pwd`
-     a_basename=`basename "$ANT"`
-     a_script=`ls -l "$ANT" | sed "s/.*${a_basename} -> //g"`
-     cd "`dirname "$ANT"`"
-     cd "`dirname "$a_script"`"
-     ANT="`pwd`"/"`basename "$a_script"`"
-     cd "$a_cwd"
-  done
-  AC_MSG_CHECKING([if $ANT works])
-cat > conftest.java << EOF
-    public class conftest {
-	int testmethod(int a, int b) {
-            return a + b;
-	}
-    }
-EOF
-
-cat > conftest.xml << EOF
-    <project name="conftest" default="conftest">
-	<target name="conftest">
-            <javac srcdir="." includes="conftest.java">
-	    </javac>
-	</target>
-    </project>
-EOF
-  oldJAVA_HOME=$JAVA_HOME
-  if test "$JDK" = "gcj"; then
-    JAVA_HOME=; export JAVA_HOME
-    ant_cmd="$ANT -Dbuild.compiler=gcj -buildfile conftest.xml 1>&2"
-  else
-    ant_cmd="$ANT -buildfile conftest.xml 1>&2"
-  fi
-  AC_TRY_EVAL(ant_cmd)
-  if test $? = 0 && test -f ./conftest.class ; then
-    AC_MSG_RESULT([Ant works]) 
-    if test -z "$WITH_ANT_HOME"; then
-        ANT_HOME=`echo $ANT | $SED -n "s/\/bin\/ant.*\$//p"`
-    else
-        ANT_HOME="$WITH_ANT_HOME"
-    fi
-  else
-    echo "configure: Ant test failed" >&5
-    cat conftest.java >&5
-    cat conftest.xml >&5
-    AC_MSG_WARN([Ant does not work - Some Java projects will not build!])
-    ANT_HOME=""
-    echo "Ant does not work - Some Java projects will not build!" >>warn
-  fi
-  JAVA_HOME=$oldJAVA_HOME
-  rm -f conftest* core core.* *.core
-fi
-if test -z "$ANT_HOME"; then
-   ANT_HOME="NO_ANT_HOME"
-fi
-AC_SUBST(ANT_HOME)
-
-dnl Checking for ant.jar
-if test "$ANT_HOME" != "NO_ANT_HOME"; then
-   AC_MSG_CHECKING([Ant lib directory])
-   if test -f $ANT_HOME/lib/ant.jar; then
-	  ANT_LIB="$ANT_HOME/lib"
-   else
-      if test -f $ANT_HOME/ant.jar; then
-	     ANT_LIB="$ANT_HOME"
-      else
-          if test -f /usr/share/java/ant.jar; then
-              ANT_LIB=/usr/share/java
-          else
-             if test -f /usr/share/ant-core/lib/ant.jar; then
-                 ANT_LIB=/usr/share/ant-core/lib
-             else
-                 AC_MSG_ERROR([Ant libraries not found!])
-             fi
-          fi
-      fi
-   fi
-   AC_MSG_RESULT([Ant lib directory found.])
-fi
-AC_SUBST(ANT_LIB)
-fi
-
 AC_MSG_CHECKING([which languages to be built])
 WITH_LANG="$with_lang"
 if test -z "$WITH_LANG"; then
Index: inc/ant.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/ant.mk,v
retrieving revision 1.27
diff -u -u -r1.27 ant.mk
--- solenv/inc/ant.mk	15 Mar 2005 09:10:44 -0000	1.27
+++ solenv/inc/ant.mk	26 Mar 2005 11:30:59 -0000
@@ -142,6 +142,13 @@
 .EXPORT : CLASSPATH
 .EXPORT : PATH
 
+# Debian: local hack to only use libant1.6-java. see /usr/share/doc/libant1.6-java/README.Debian
+# FIXME: conditionalize this and don't hardcode it. configure? (configure needs to be fixed anyhow
+# because we are not allowed to use $ANT_HOME/bin/ant in any way)
+ANT_HOME:=/usr/share/ant1.6
+ANT_LIB:=$(ANT_HOME)$/lib
+ANT:=$(JAVAI) $(JAVACPS) $(CLASSPATH)$(PATH_SEPERATOR)$(ANT_LIB)/ant-launcher.jar$(PATH_SEPERATOR)$(ANT_LIB)/ant.jar -Dant.home=$(ANT_HOME) org.apache.tools.ant.Main
+
 # --- TARGETS -----------------------------------------------------
 
 #$(CLASSDIR)$/solar.properties .PHONY:
Index: set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.54
diff -u -u -r1.54 set_soenv.in
--- config_office/set_soenv.in	15 Mar 2005 09:46:44 -0000	1.54
+++ config_office/set_soenv.in	26 Mar 2005 11:31:36 -0000
@@ -1361,9 +1361,6 @@
    ToFile( "JAVADOC",         "@JAVADOC@",       "e" );
 }
 ToFile( "XSLTPROC",          "@XSLTPROC@",       "e" );
-ToFile( "ANT_HOME",          "@ANT_HOME@",       "e" );
-ToFile( "ANT_LIB",           "@ANT_LIB@",        "e" );
-ToFile( "ANT",               "@ANT@",            "e" );
 ToFile( "JDKLIB",            $JAVA_LIB,          "e" );
 ToFile( "STLPORT4",          $STLPORT4,          "e" );
 ToFile( "ASM_PATH",          $ASM_PATH,          "e" );
? f
Index: makefile.mk
===================================================================
RCS file: /cvs/external/hsqldb/makefile.mk,v
retrieving revision 1.5
diff -u -u -r1.5 makefile.mk
--- hsqldb/makefile.mk	15 Mar 2005 09:08:29 -0000	1.5
+++ hsqldb/makefile.mk	26 Mar 2005 21:32:36 -0000
@@ -90,6 +90,13 @@
 
 ANT_BUILDFILE=build$/build.xml
 
+# Debian: local hack to only use libant1.6-java. see /usr/share/doc/libant1.6-java/README.Debian
+# # FIXME: conditionalize this and don't hardcode it. configure? (configure needs to be fixed anyhow
+# # because we are not allowed to use $ANT_HOME/bin/ant in any way)
+ANT_HOME:=/usr/share/ant1.6
+ANT_LIB:=$(ANT_HOME)$/lib
+ANT:=$(JAVAI) $(JAVACPS) $(CLASSPATH)$(PATH_SEPERATOR)$(ANT_LIB)/ant-launcher.jar$(PATH_SEPERATOR)$(ANT_LIB)/ant.jar -Dant.home=$(ANT_HOME) org.apache.tools.ant.Main
+
 .IF "$(JDK)"=="gcj"
 JAVA_HOME=
 .EXPORT : JAVA_HOME
