diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sw.vpj sw.clean/source/core/edit/edws.cxx sw/source/core/edit/edws.cxx
--- sw.clean/source/core/edit/edws.cxx	2008-02-28 19:19:41.000000000 -0500
+++ sw/source/core/edit/edws.cxx	2008-02-28 22:12:10.000000000 -0500
@@ -364,7 +364,7 @@ void SwEditShell::AutoCorrect( SvxAutoCo
 	SwAutoCorrDoc aSwAutoCorrDoc( *this, *pCrsr, cChar );
 	rACorr.AutoCorrect( aSwAutoCorrDoc,
 					pTNd->GetTxt(), pCrsr->GetPoint()->nContent.GetIndex(),
-					cChar, bInsert );
+					cChar, bInsert, GetWin() );
 	if( cChar )
 		SaveTblBoxCntnt( pCrsr->GetPoint() );
 	EndAllAction();
