diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=solenv.vpj solenv.clean/inc/unxlngi6.mk solenv/inc/unxlngi6.mk
--- solenv.clean/inc/unxlngi6.mk	2008-02-27 10:53:41.000000000 -0500
+++ solenv/inc/unxlngi6.mk	2008-02-28 13:04:07.000000000 -0500
@@ -208,6 +208,10 @@ STDLIBCUIMT+=-ldl -lpthread -lm
 STDSHLGUIMT+=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT+=-ldl -lpthread -lm
 
+# XTest - X11 extension
+STDLIBGUIMT+=-lXtst
+STDSHLGUIMT+=-lXtst
+
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
 .IF "$(USE_STLP_DEBUG)" != ""
