Index: native/unix/makefile.mk
===================================================================
RCS file: /cvs/api/bean/native/unix/makefile.mk,v
retrieving revision 1.6
diff -u -u -r1.6 makefile.mk
--- bean/native/unix/makefile.mk	6 Jun 2005 17:42:10 -0000	1.6
+++ bean/native/unix/makefile.mk	9 Jun 2005 12:19:26 -0000
@@ -27,10 +27,10 @@
 
 SHL1TARGET=$(TARGET)
 SHL1LIBS=	$(SLB)$/$(TARGET).lib
+SHL1STDLIBS=-ljawt
 .IF "$(JDK)" == "gcj"
-SHL1STDLIBS=-lgcjawt -lgcj
+SHL1STDLIBS+=-lgcj
 .ELSE
-SHL1STDLIBS=-ljawt
 .ENDIF
 
 .IF "$(OS)" == "LINUX"
