--- solenv/bin/modules/installer/servicesfile.pm.orig	2004-11-04 17:09:30.354448680 +0100
+++ solenv/bin/modules/installer/servicesfile.pm	2004-11-04 17:27:59.466837968 +0100
@@ -783,7 +783,10 @@ sub create_services_rdb
 
 	# Adding the created file "legacy_binfilters.rdb" to the filearray
 	
-	add_legacy_binfilters_rdb_file_into_filearray($filesarrayref, $includepatharrayref);
+	if ( $ENV{'WITH_BINFILTER'} != "NO" )
+	{
+		add_legacy_binfilters_rdb_file_into_filearray($filesarrayref, $includepatharrayref);
+	}
 
 }
 
