Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.69.4.1
diff -u -r1.69.4.1 configure.in
--- config_office/configure.in	28 Sep 2004 09:11:31 -0000	1.69.4.1
+++ config_office/configure.in	8 Oct 2004 09:47:33 -0000
@@ -96,6 +96,9 @@
                           the static libraries instead of the dynamic ones. 
                           (enables -Bstatic linker flag for gtk libraries)
 ],,)
+AC_ARG_ENABLE(binfilter,
+[  --disable-binfilter:    Disable legacy binary file formats filters
+],,)
 AC_ARG_ENABLE(rpath,
 [  --disable-rpath:        Disable the use of relative paths in shared libraries
 ],,)
@@ -573,6 +576,16 @@
    ENABLE_DIRECTX=""
 fi
 AC_SUBST(ENABLE_DIRECTX)
+
+dnl ===================================================================
+dnl Disable legacy binary file formats filters
+dnl ===================================================================
+if test "$enable_binfilter" = "no"; then
+   WITH_BINFILTER="NO"
+else
+   WITH_BINFILTER="YES"
+fi
+AC_SUBST(WITH_BINFILTER)
 
 dnl ===================================================================
 dnl Disable rpath in shared libraries?
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.22
diff -u -r1.22 set_soenv.in
--- config_office/set_soenv.in	9 Sep 2004 11:22:23 -0000	1.22
+++ config_office/set_soenv.in	8 Oct 2004 09:47:54 -0000
@@ -1691,6 +1691,7 @@
 ToFile( "LIBSN_LIBS",        "@LIBSN_LIBS@",       "e" );
 ToFile( "WITH_MOZILLA",      "@WITH_MOZILLA@",     "e" );
 ToFile( "WITH_FONTS",        "@WITH_FONTS@",       "e" );
+ToFile( "WITH_BINFILTER",    "@WITH_BINFILTER@",   "e" );
 if ( '@ENABLE_RPATH@' eq "no" ) {
   ToFile( "LINKFLAGSRUNPATH",'',                   "e" );
 }
Index: instsetoo/prj/build.lst
===================================================================
RCS file: /cvs/installation/instsetoo/prj/build.lst,v
retrieving revision 1.37
diff -u -r1.37 build.lst
--- instsetoo/prj/build.lst	9 Sep 2004 11:13:31 -0000	1.37
+++ instsetoo/prj/build.lst	8 Oct 2004 09:42:22 -0000
@@ -1,4 +1,4 @@
-oo	instsetoo	:	helpcontent2 UnoControls basctl configmgr cpputools dbaccess desktop eventattacher extensions extras fileaccess forms io package padmin readlicense_oo remotebridges scaddins scp2 shell sc sch chart2 sd starmath sw ucb wizards officecfg xmlhelp MathMLDTD dictionaries bitstream_vera_fonts filter psprint_config fpicker odk embedserv pyuno crashrep scripting embeddedobj hwpfilter unoxml slideshow binfilter setup_native msfontextract xmlsecurity NULL
+oo	instsetoo	:	helpcontent2 UnoControls basctl configmgr cpputools dbaccess desktop eventattacher extensions extras fileaccess forms io package padmin readlicense_oo remotebridges scaddins scp2 shell sc sch chart2 sd starmath sw ucb wizards officecfg xmlhelp MathMLDTD dictionaries bitstream_vera_fonts filter psprint_config fpicker odk embedserv pyuno crashrep scripting embeddedobj hwpfilter unoxml slideshow $(WITH_BINFILTER):binfilter setup_native msfontextract xmlsecurity NULL
 oo	instsetoo								usr1	-	all	oo_mkout NULL
 oo	instsetoo\packimages					nmake	-	all	oo_pack NULL
 oo  instsetoo\packconfig                    nmake   -   all oo_packconfig NULL
Index: instsetoo_native/prj/build.lst
===================================================================
RCS file: /cvs/installation/instsetoo_native/prj/build.lst,v
retrieving revision 1.9
diff -u -r1.9 build.lst
--- instsetoo_native/prj/build.lst	9 Sep 2004 11:16:04 -0000	1.9
+++ instsetoo_native/prj/build.lst	8 Oct 2004 09:42:28 -0000
@@ -1,4 +1,4 @@
-oon	instsetoo_native	:	helpcontent2 UnoControls basctl configmgr cpputools dbaccess desktop eventattacher extensions extras fileaccess forms io package padmin readlicense_oo remotebridges scaddins scp2 setup_native SO:epm shell sc sch chart2 sd starmath sw ucb wizards officecfg xmlhelp MathMLDTD dictionaries bitstream_vera_fonts filter psprint_config fpicker odk embedserv pyuno crashrep scripting embeddedobj hwpfilter unoxml slideshow binfilter msfontextract xmlsecurity NULL
+oon	instsetoo_native	:	helpcontent2 UnoControls basctl configmgr cpputools dbaccess desktop eventattacher extensions extras fileaccess forms io package padmin readlicense_oo remotebridges scaddins scp2 setup_native SO:epm shell sc sch chart2 sd starmath sw ucb wizards officecfg xmlhelp MathMLDTD dictionaries bitstream_vera_fonts filter psprint_config fpicker odk embedserv pyuno crashrep scripting embeddedobj hwpfilter unoxml slideshow $(WITH_BINFILTER):binfilter msfontextract xmlsecurity NULL
 oon	instsetoo_native						usr1	-	all	oon_mkout NULL
 oon	instsetoo_native\packimages				nmake	-	all	oon_pack NULL
 oon	instsetoo_native\packconfig				nmake	-	all	oon_config NULL
