Index: ./source/customshapes/EnhancedCustomShapeFunctionParser.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/customshapes/EnhancedCustomShapeFunctionParser.cxx,v
retrieving revision 1.10
diff -u -r1.10 EnhancedCustomShapeFunctionParser.cxx
--- svx/source/customshapes/EnhancedCustomShapeFunctionParser.cxx	24 Oct 2006 13:42:15 -0000	1.10
+++ svx/source/customshapes/EnhancedCustomShapeFunctionParser.cxx	6 Sep 2007 08:10:27 -0000
@@ -1201,7 +1201,7 @@
     const ::boost::spirit::parse_info<StringIteratorT> aParseInfo( 
             ::boost::spirit::parse( aStart,
                                     aEnd,
-                                    aExpressionGrammer,
+                                    aExpressionGrammer >> ::boost::spirit::end_p,
                                     ::boost::spirit::space_p ) );
     OSL_DEBUG_ONLY(::std::cout.flush()); // needed to keep stdout and cout in sync
 
