--- configmgr/source/inc/noderef.hxx.old	2005-09-08 05:52:19.000000000 +0200
+++ configmgr/source/inc/noderef.hxx	2005-11-07 16:16:37.000000000 +0100
@@ -196,11 +196,7 @@
 			TreeRef getContextTree() const;
 			/// gets the parent node of this tree ('s root node), if available
 			NodeRef getContextNode() const;
-			#if defined(_MSC_VER) && (_MSC_VER > 1300 )
 			friend bool equalTreeRef(TreeRef const& lhs, TreeRef const& rhs);
-			#else
-			friend bool equalTreeRef(TreeRef const& lhs, TreeRef const& rhs) { return lhs.m_pImpl == rhs.m_pImpl; }
-			#endif
 	private:
 			friend class Tree;
 			friend class TreeImplHelper;
@@ -210,9 +206,7 @@
 
 			TreeImpl* m_pImpl;
         };
-		#if defined(_MSC_VER) && (_MSC_VER > 1300 )
 		inline bool equalTreeRef(TreeRef const& lhs, TreeRef const& rhs) { return lhs.m_pImpl == rhs.m_pImpl; }
-		#endif
 
 		/**	represents a hierarchy of config entries (identified by <type>NodeRef</type>s and <type>ValueRef</type>s)
 
