Index: solenv/bin/build.pl
===================================================================
RCS file: /cvs/tools/solenv/bin/build.pl,v
retrieving revision 1.115
diff -u -r1.115 build.pl
--- solenv/bin/build.pl	7 Sep 2004 11:32:13 -0000	1.115
+++ solenv/bin/build.pl	8 Oct 2004 09:42:12 -0000
@@ -1935,6 +1935,29 @@
 };
 
 #
+# Extract a boolean value from an environment variable
+#
+sub get_env_boolean($$) {
+    my $var_key = shift;
+    my $default = shift;
+    my $var = $ENV{$var_key};
+    
+    defined $var || return $default;
+    $var eq '' && return $default;
+
+    if ($var =~ m/yes/i || $var =~ m/true/i ||
+	$var =~ m/enabled/i || $var =~ m/^[1-9]+$/) {
+	return '1';
+    }
+    elsif ($var =~ m/no/i || $var =~ m/false/i ||
+	   $var =~ m/disabled/i || $var =~ m/^0+$/) {
+	return '';
+    }
+
+    return $default;
+}
+
+#
 # pick only the modules, that should be built for
 # build types from %build_modes
 #
@@ -1943,7 +1966,10 @@
     my @mod_array = split(/\s+/, $modules);
     my $new_modules = '';
     foreach (@mod_array) {
-        if (/(\w+):(\S+)/o) {
+	if (/\$\(\w+\):(\S+)/o) {
+	    $new_modules .= $2 . ' ' if (get_env_boolean($1,1));
+	}
+	elsif (/(\w+):(\S+)/o) {
             $new_modules .= $2 . ' ' if (defined $build_modes{$1});
             next;
         };
