--- solenv/bin/modules/installer/scriptitems.pm
+++ solenv/bin/modules/installer/scriptitems.pm
@@ -466,6 +469,8 @@
 		$value =~ s/\<productbuildid\>/$installer::globals::buildid/;
 		$value =~ s/\<sourceid\>/$installer::globals::build/;
 		$value =~ s/\<productupdate\>/$productupdatestring/;
+		$value =~ s/\<vendor\>/$ENV{'OOO_VENDOR'}/;
+		$value =~ s/\<buildversion\>/$ENV{'BUILD_VER_STRING'}/;
 
 		$oneitem->{'Value'} = $value;
 	}
--- scp2/source/ooo/profileitem_ooo.scp
+++ scp2/source/ooo/profileitem_ooo.scp
@@ -217,6 +217,24 @@
   #endif
 End
 
+ProfileItem gid_Profileitem_Version_Vendor
+    ProfileID = gid_Profile_Version_Ini;
+    ModuleID = gid_Module_Root;
+    Section = "Version";
+    Order = 16;
+    Key = "Vendor";
+    Value = "<vendor>";
+End
+
+ProfileItem gid_Profileitem_Version_BuildVersion
+    ProfileID = gid_Profile_Version_Ini;
+    ModuleID = gid_Module_Root;
+    Section = "Version";
+    Order = 17;
+    Key = "BuildVersion";
+    Value = "<buildversion>";
+End
+
 // for the configmgr
 
 ProfileItem gid_Profileitem_Configmgr_Cfg_Servertype
