Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.82
diff -u -r1.82 configure.in
--- config_office/configure.in	5 Jan 2005 12:09:48 -0000	1.82
+++ config_office/configure.in	26 Jan 2005 16:19:58 -0000
@@ -354,6 +379,13 @@
 
                           Usage: --with-use-shell=tcsh|4nt
 ],with_use_shell=$withval,with_use_shell="tcsh")
+AC_ARG_WITH(build-version,
+[  --with-build-version   Allows the builder to add a custom version tag
+                          that will appear in the Help/About box for QA
+                          purposes.
+
+                          Usage: --with-build-version="Built by Jim"
+],with_build_version=$withval)
 AC_ARG_ENABLE(sgistl,
 [  --enable-sgistl         for IRIX users, use this option option to build
                           OpenOffice.org using SGI's STL.
@@ -1436,6 +1495,17 @@
 fi
 
 dnl ===================================================================
+dnl Custom build version
+dnl ===================================================================
+
+if test "z$with_build_version" != "z"; then
+   BUILD_VER_STRING=$with_build_version
+else
+   BUILD_VER_STRING=
+fi
+AC_SUBST(BUILD_VER_STRING)
+
+dnl ===================================================================
 dnl Java support enable
 dnl ===================================================================
 AC_MSG_CHECKING([whether to build with Java support])
