Index: sd/source/ui/slidesorter/view/SlsPageNotificationObjectContact.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/slidesorter/view/SlsPageNotificationObjectContact.cxx,v
retrieving revision 1.4
diff -u -r1.4 SlsPageNotificationObjectContact.cxx
--- sd/source/ui/slidesorter/view/SlsPageNotificationObjectContact.cxx	19 Oct 2005 12:26:21 -0000	1.4
+++ sd/source/ui/slidesorter/view/SlsPageNotificationObjectContact.cxx	31 Oct 2005 15:10:49 -0000
@@ -51,7 +52,7 @@
       mrBroadcaster(rBroadcaster),
       mrListener(rListener)
 {
-    ObjectContact::mbIsPreviewRenderer = sal_True;
+	mbIsPreviewRenderer = sal_True;
 }
 
 
