--- xmerge/prj/build.lst       27 Oct 2005 08:30:27 -0000      1.5
+++ xmerge/prj/build.lst       24 Oct 2005 16:00:30 -0000      1.4
@@ -1,4 +1,4 @@
-xmrg xmerge :  solenv NULL
+xmrg xmerge : solenv unoil javaunohelper NULL
 xmrg xmerge                        usr1  - all xmrg_mkout NULL
 xmrg xmerge\source\xmerge          nmake - all xmrg_xmerge NULL
 xmrg xmerge\source\bridge          nmake - all xmrg_bridge xmrg_xmerge NULL
