Index: ucb/source/ucp/gvfs/makefile.mk
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/gvfs/makefile.mk,v
retrieving revision 1.4
diff -u -r1.4 makefile.mk
--- ucb/source/ucp/gvfs/makefile.mk	21 Jan 2005 11:38:43 -0000	1.4
+++ ucb/source/ucp/gvfs/makefile.mk	19 Apr 2005 20:39:54 -0000
@@ -75,7 +75,7 @@
 
 .IF "$(ENABLE_GNOMEVFS)"!=""
 COMPILER_WARN_ALL=TRUE
-PKGCONFIG_MODULES=gnome-vfs-2.0
+PKGCONFIG_MODULES=gnome-vfs-2.0 gthread-2.0 glib-2.0
 .INCLUDE: pkg_config.mk
 
 .IF "$(OS)" == "SOLARIS"
Index: vcl/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.65
diff -u -r1.65 makefile.mk
--- vcl/util/makefile.mk	13 Jan 2005 18:16:05 -0000	1.65
+++ vcl/util/makefile.mk	20 Apr 2005 06:43:17 -0000
@@ -362,7 +362,7 @@
 
 # gtk plugin
 .IF "$(ENABLE_GTK)" != ""
-PKGCONFIG_MODULES=gtk+-2.0 gthread-2.0
+PKGCONFIG_MODULES=gtk+-2.0 gthread-2.0 gdk-x11-2.0 glib-2.0 gdk-pixbuf-2.0 gobject-2.0 pango
 .INCLUDE: pkg_config.mk
 
 LIB4TARGET=$(SLB)$/igtk_plug_
Index: fpicker/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/fpicker/util/makefile.mk,v
retrieving revision 1.10
diff -u -r1.10 makefile.mk
--- fpicker/util/makefile.mk	18 Jan 2005 13:26:32 -0000	1.10
+++ fpicker/util/makefile.mk	20 Apr 2005 12:23:59 -0000
@@ -74,7 +74,7 @@
 
 .IF "$(GUI)"!="WNT"
 .IF "$(ENABLE_GTK)" == "TRUE"
-PKGCONFIG_MODULES=gtk+-2.0
+PKGCONFIG_MODULES=gtk+-2.0 glib-2.0 gdk-pixbuf-2.0 gobject-2.0
 .INCLUDE: pkg_config.mk
 #check gtk version
 GTK_TWO_FOUR:=$(shell +-$(PKGCONFIG) --exists 'gtk+-2.0 >= 2.4.0' && echo ok)
