diff --git a/writerfilter/inc/odiapi/xxml/ContentHandler.hxx b/writerfilter/inc/odiapi/xxml/ContentHandler.hxx
index b38cc2f..da847f3 100644
--- writerfilter/inc/odiapi/xxml/ContentHandler.hxx
+++ writerfilter/inc/odiapi/xxml/ContentHandler.hxx
@@ -40,7 +40,13 @@
 #include <odiapi/qname/QName.hxx>
 #include <WriterFilterDllApi.hxx>
 #include <rtl/ustring.hxx>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 
 namespace writerfilter { namespace xxml {
diff --git a/writerfilter/inc/resourcemodel/OutputWithDepth.hxx b/writerfilter/inc/resourcemodel/OutputWithDepth.hxx
index 20b564c..b06db5d 100644
--- writerfilter/inc/resourcemodel/OutputWithDepth.hxx
+++ writerfilter/inc/resourcemodel/OutputWithDepth.hxx
@@ -37,7 +37,14 @@
 #define INCLUDED_OUTPUT_WITH_DEPTH
 
 #include <vector>
+
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <iostream>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 
 namespace writerfilter
 {
diff --git a/writerfilter/inc/resourcemodel/WW8ResourceModel.hxx b/writerfilter/inc/resourcemodel/WW8ResourceModel.hxx
index 0beaa93..dceabb7 100644
--- writerfilter/inc/resourcemodel/WW8ResourceModel.hxx
+++ writerfilter/inc/resourcemodel/WW8ResourceModel.hxx
@@ -36,7 +36,13 @@
 #ifndef INCLUDED_WW8_EVENT_HANDLER_HXX
 #define INCLUDED_WW8_EVENT_HANDLER_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
 #include <memory>
 #include <boost/shared_ptr.hpp>
 
diff --git a/writerfilter/inc/resourcemodel/exceptions.hxx b/writerfilter/inc/resourcemodel/exceptions.hxx
index 0bde3a2..5f8c442 100644
--- writerfilter/inc/resourcemodel/exceptions.hxx
+++ writerfilter/inc/resourcemodel/exceptions.hxx
@@ -36,7 +36,14 @@
 #ifndef _DOCTOK_EXCEPTIONS_HXX
 #define _DOCTOK_EXCEPTIONS_HXX
 
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility push(default)
+#endif
 #include <string>
+#ifdef GCC_VISIBILITY_BROKEN
+#pragma GCC visibility pop
+#endif
+
 #include <WriterFilterDllApi.hxx>
 
 namespace writerfilter {
