--- svx/source/dialog/cuifmsearch.hxx	2007-08-02 16:24:27.000000000 +0200
+++ svx/source/dialog/cuifmsearch.hxx	2007-08-02 16:24:14.000000000 +0200
@@ -91,6 +91,7 @@ namespace svxform {
 // = class FmSearchDialog - Dialog fuer Suchen in Formularen/Tabellen
 // ===================================================================================================
 
+class FmSearchEngine;
 struct FmSearchProgress;
 
 class FmSearchDialog : public ModalDialog
