Index: icu/makefile.mk
===================================================================
RCS file: /cvs/external/icu/makefile.mk,v
retrieving revision 1.28
diff -u -p -u -r1.28 makefile.mk
--- icu/makefile.mk	28 Dec 2005 16:54:29 -0000	1.28
+++ icu/makefile.mk	16 Feb 2006 15:33:41 -0000
@@ -201,5 +206,6 @@ $(PACKAGE_DIR)$/$(CONFIGURE_FLAG_FILE) :
 # configure is affected.
 $(MISC)$/remove_build.flag : $(BINARY_PATCH_FILE_NAME) $(PATCH_FILE_NAME) makefile.mk
 	$(REMOVE_PACKAGE_COMMAND)
+	@+-$(RM) $(PACKAGE_DIR)$/$(UNTAR_FLAG_FILE)
 	+$(TOUCH) $(MISC)$/remove_build.flag
 
