--- ./UnoControls/source/controls/progressbar.cxx.gcc43~	2006-06-20 01:09:11.000000000 +0200
+++ ./UnoControls/source/controls/progressbar.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -69,6 +69,7 @@
 #endif
 
 #include <math.h>
+#include <limits.h>
 
 //____________________________________________________________________________________________________________
 //	includes of my project
--- ./bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx.gcc43~	2007-11-13 15:17:59.000000000 +0100
+++ ./bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -39,6 +39,7 @@
 #include <stdio.h>
 #include <dlfcn.h>
 #include <cxxabi.h>
+#include <string.h>
 #include <hash_map>
 
 #include <rtl/strbuf.hxx>
--- ./bridges/source/remote/context/context.cxx.gcc43~	2006-09-16 17:58:33.000000000 +0200
+++ ./bridges/source/remote/context/context.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_bridges.hxx"
 #include <stdio.h>
+#include <string.h>
 #include <list>
 #include <hash_map>
 #include <utility>
--- ./bridges/source/remote/urp/urp_marshal.hxx.gcc43~	2005-09-08 00:47:08.000000000 +0200
+++ ./bridges/source/remote/urp/urp_marshal.hxx	2008-01-25 18:52:13.000000000 +0100
@@ -54,6 +54,8 @@
 #include "urp_marshal_decl.hxx"
 #endif
 
+#include <string.h>
+
 struct remote_Interface;
 
 namespace bridges_urp
--- ./connectivity/source/commontools/sqlerror.cxx.gcc43~	2007-11-21 15:59:17.000000000 +0100
+++ ./connectivity/source/commontools/sqlerror.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -46,6 +46,8 @@
 #include <cppuhelper/exc_hlp.hxx>
 #include <rtl/ustrbuf.hxx>
 
+#include <string.h>
+
 //........................................................................
 namespace connectivity
 {
--- ./connectivity/source/drivers/adabas/BConnection.cxx.gcc43~	2008-01-18 15:52:32.000000000 +0100
+++ ./connectivity/source/drivers/adabas/BConnection.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -76,6 +76,8 @@
 #include <connectivity/sqlparse.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity::adabas;
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
--- ./connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx.gcc43~	2008-01-25 18:08:21.000000000 +0100
+++ ./connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -63,6 +63,7 @@
 #include <com/sun/star/sdbc/ColumnSearch.hpp>
 
 #include <vector>
+#include <cstring>
 
 #ifndef _CONNECTIVITY_EVOAB_EVOLUTION_API_HXX_
 #include "EApi.h"
--- ./connectivity/source/drivers/odbc/OConnection.cxx.gcc43~	2008-01-18 16:02:30.000000000 +0100
+++ ./connectivity/source/drivers/odbc/OConnection.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -81,6 +81,8 @@
 #include <connectivity/dbexception.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity::odbc;
 using namespace connectivity;
 using namespace dbtools;
--- ./connectivity/source/drivers/odbc/OPreparedStatement.cxx.gcc43~	2008-01-18 16:02:57.000000000 +0100
+++ ./connectivity/source/drivers/odbc/OPreparedStatement.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -82,6 +82,8 @@
 #include "connectivity/sqlparse.hxx"
 #endif
 
+#include <string.h>
+
 using namespace ::comphelper;
 using namespace connectivity;
 using namespace connectivity::odbc;
--- ./connectivity/source/parse/sqlbison.y.gcc43~	2008-01-18 16:09:25.000000000 +0100
+++ ./connectivity/source/parse/sqlbison.y	2008-01-25 18:52:13.000000000 +0100
@@ -76,6 +76,8 @@
 #include "connectivity/dbconversion.hxx"
 #endif
 
+#include <string.h>
+
 #if defined __GNUC__
     #pragma GCC system_header
 #elif defined __SUNPRO_CC
--- ./connectivity/source/parse/sqlnode.cxx.gcc43~	2008-01-18 16:09:42.000000000 +0100
+++ ./connectivity/source/parse/sqlnode.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -147,6 +147,7 @@
 #include <tools/diagnose_ex.h>
 #endif
 
+#include <string.h>
 
 using namespace ::com::sun::star::sdbc;
 using namespace ::com::sun::star::util;
--- ./dbaccess/source/core/api/KeySet.cxx.gcc43~	2008-01-18 16:34:22.000000000 +0100
+++ ./dbaccess/source/core/api/KeySet.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -116,6 +116,7 @@
 #ifndef _TOOLS_DEBUG_HXX
 #include <tools/debug.hxx>
 #endif
+#include <string.h>
 
 using namespace dbaccess;
 using namespace ::connectivity;
--- ./desktop/source/so_comp/services.cxx.gcc43~	2006-10-12 16:17:20.000000000 +0200
+++ ./desktop/source/so_comp/services.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -61,6 +61,7 @@
 #include "oemjob.hxx"
 #include "evaluation.hxx"
 
+#include <string.h>
 
 using namespace rtl;
 using namespace ::com::sun::star::uno;
--- ./desktop/unx/splash/services_unxsplash.cxx.gcc43~	2008-01-25 18:08:15.000000000 +0100
+++ ./desktop/unx/splash/services_unxsplash.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -54,6 +54,7 @@
 #endif
 
 #include "unxsplash.hxx"
+#include <string.h>
 
 using namespace rtl;
 using namespace ::com::sun::star::uno;
--- ./filter/source/flash/swffilter.cxx.gcc43~	2007-01-18 15:09:23.000000000 +0100
+++ ./filter/source/flash/swffilter.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -86,6 +86,7 @@
 
 #include "swfexporter.hxx"
 
+#include <string.h>
 //#include <stdlib.h>
 //#include <windows.h>
 
--- ./filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx.gcc43~	2006-12-01 15:31:48.000000000 +0100
+++ ./filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_filter.hxx"
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
--- ./filter/source/xmlfilterdetect/filterdetect.cxx.gcc43~	2007-10-23 13:40:19.000000000 +0200
+++ ./filter/source/xmlfilterdetect/filterdetect.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_filter.hxx"
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
--- ./forms/source/component/ComboBox.cxx.gcc43~	2007-11-21 18:15:24.000000000 +0100
+++ ./forms/source/component/ComboBox.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -67,6 +67,7 @@
 #include <tools/debug.hxx>
 #include <tools/diagnose_ex.h>
 #include <unotools/sharedunocomponent.hxx>
+#include <limits.h>
 
 using namespace dbtools;
 
--- ./fpicker/source/unx/kde/kdefpmain.cxx.gcc43~	2008-01-25 18:08:25.000000000 +0100
+++ ./fpicker/source/unx/kde/kdefpmain.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -68,6 +68,7 @@
 #include <kcmdlineargs.h>
 
 #include <iostream>
+#include <cstdlib>
 
 //////////////////////////////////////////////////////////////////////////
 // Main
--- ./fpicker/source/unx/kde_unx/UnxCommandThread.cxx.gcc43~	2008-01-25 18:08:25.000000000 +0100
+++ ./fpicker/source/unx/kde_unx/UnxCommandThread.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -76,6 +76,7 @@
 #endif
 
 #include <unistd.h>
+#include <string.h>
 
 #include <iostream>
 
--- ./framework/source/services/substitutepathvars.cxx.gcc43~	2006-09-16 16:11:52.000000000 +0200
+++ ./framework/source/services/substitutepathvars.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -56,6 +56,8 @@
 #include "services.h"
 #endif
 
+#include <string.h>
+
 //_________________________________________________________________________________________________________________
 //	interface includes
 //_________________________________________________________________________________________________________________
--- ./hwpfilter/source/formula.cpp.gcc43~	2006-06-20 02:52:23.000000000 +0200
+++ ./hwpfilter/source/formula.cpp	2008-01-25 18:52:22.000000000 +0100
@@ -39,7 +39,7 @@
 #include "nodes.h"
 #include "mapping.h"
 #include "hwpeq.h"
-#include <iostream.h>
+#include <iostream>
 
 extern LinkedList<Node> nodelist;
 
--- ./i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx.gcc43~	2007-11-13 15:32:24.000000000 +0100
+++ ./i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -73,6 +73,7 @@
 #include <com/sun/star/i18n/XLocaleData.hpp>
 
 #include <bullet.h>
+#include <string.h>
 
 using namespace com::sun::star;
 using namespace com::sun::star::uno;
--- ./i18npool/source/localedata/saxparser.cxx.gcc43~	2006-09-17 11:21:51.000000000 +0200
+++ ./i18npool/source/localedata/saxparser.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -37,6 +37,7 @@
 #include "precompiled_i18npool.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #include <stack>
 
 #include "sal/main.h"
--- ./i18npool/source/search/textsearch.cxx.gcc43~	2007-12-05 17:44:55.000000000 +0100
+++ ./i18npool/source/search/textsearch.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -78,6 +78,8 @@
 #include <cppuhelper/weak.hxx>
 #endif
 
+#include <string.h>
+
 #ifdef _MSC_VER
 // get rid of that dumb compiler warning
 // identifier was truncated to '255' characters in the debug information
--- ./idlc/source/idlccompile.cxx.gcc43~	2007-11-02 13:25:51.000000000 +0100
+++ ./idlc/source/idlccompile.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -71,6 +71,8 @@
 #endif
 #endif
 
+#include <string.h>
+
 using namespace ::rtl;
 using namespace ::osl;
 
--- ./idlc/source/idlcproduce.cxx.gcc43~	2007-10-15 14:44:05.000000000 +0200
+++ ./idlc/source/idlcproduce.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -63,6 +63,8 @@
 #include <errno.h>
 #endif
 
+#include <string.h>
+
 using namespace ::rtl;
 using namespace ::osl;
 
--- ./jvmfwk/source/elements.cxx.gcc43~	2007-06-20 19:13:36.000000000 +0200
+++ ./jvmfwk/source/elements.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -50,6 +50,7 @@
 #include "libxml/xpathInternals.h"
 #include "rtl/bootstrap.hxx"
 #include "boost/optional.hpp"
+#include <string.h>
 // #define NS_JAVA_FRAMEWORK "http://openoffice.org/2004/java/framework/1.0"
 // #define NS_SCHEMA_INSTANCE "http://www.w3.org/2001/XMLSchema-instance"
 
--- ./jvmfwk/source/fwkutil.cxx.gcc43~	2007-10-15 14:07:17.000000000 +0200
+++ ./jvmfwk/source/fwkutil.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -47,6 +47,7 @@
 #endif
 
 #include <string>
+#include <string.h>
 #include "osl/mutex.hxx"
 #include "osl/module.hxx"
 #include "osl/thread.hxx"
--- ./lingucomponent/source/languageguessing/guess.cxx.gcc43~	2007-06-22 10:32:19.000000000 +0200
+++ ./lingucomponent/source/languageguessing/guess.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -37,6 +37,7 @@
 #include "precompiled_lingucomponent.hxx"
 
 #include <iostream>
+#include <string.h>
 
 #include <libtextcat/textcat.h>
 #include <altstrfunc.hxx>
--- ./oox/inc/oox/helper/helper.hxx.gcc43~	2007-12-10 15:47:49.000000000 +0100
+++ ./oox/inc/oox/helper/helper.hxx	2008-01-25 18:52:22.000000000 +0100
@@ -40,6 +40,7 @@
 #include <osl/endian.h>
 #include <rtl/string.hxx>
 #include <rtl/ustring.hxx>
+#include <string.h>
 
 namespace oox {
 
--- ./oox/source/core/binarycodec.cxx.gcc43~	2007-12-10 15:47:49.000000000 +0100
+++ ./oox/source/core/binarycodec.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -36,6 +36,7 @@
 #include "oox/core/binarycodec.hxx"
 #include <algorithm>
 #include <osl/diagnose.h>
+#include <string.h>
 
 namespace oox {
 namespace core {
--- ./oox/source/helper/binaryinputstream.cxx.gcc43~	2007-12-10 15:47:50.000000000 +0100
+++ ./oox/source/helper/binaryinputstream.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -36,6 +36,7 @@
 #include "oox/helper/binaryinputstream.hxx"
 #include <com/sun/star/io/XInputStream.hpp>
 #include <osl/diagnose.h>
+#include <string.h>
 
 using ::com::sun::star::uno::Reference;
 using ::com::sun::star::uno::Sequence;
--- ./oox/source/helper/binaryoutputstream.cxx.gcc43~	2007-12-10 15:47:50.000000000 +0100
+++ ./oox/source/helper/binaryoutputstream.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -37,6 +37,7 @@
 #include <osl/diagnose.h>
 #include <com/sun/star/io/XOutputStream.hpp>
 #include "oox/helper/binaryinputstream.hxx"
+#include <string.h>
 
 using ::com::sun::star::uno::Reference;
 using ::com::sun::star::uno::Sequence;
--- ./oox/source/helper/recordinputstream.cxx.gcc43~	2007-12-10 15:47:50.000000000 +0100
+++ ./oox/source/helper/recordinputstream.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -35,6 +35,7 @@
 
 #include "oox/helper/recordinputstream.hxx"
 #include <vector>
+#include <string.h>
 
 using ::rtl::OUString;
 
--- ./oox/source/ppt/commonbehaviorcontext.cxx.gcc43~	2007-12-10 15:47:50.000000000 +0100
+++ ./oox/source/ppt/commonbehaviorcontext.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -38,6 +38,7 @@
 #include <comphelper/processfactory.hxx>
 #include <osl/diagnose.h>
 #include <rtl/ustrbuf.hxx>
+#include <string.h>
 
 #include <com/sun/star/animations/XTimeContainer.hpp>
 #include <com/sun/star/animations/XAnimationNode.hpp>
--- ./oox/source/xls/biffcodec.cxx.gcc43~	2007-12-10 15:47:50.000000000 +0100
+++ ./oox/source/xls/biffcodec.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -35,6 +35,7 @@
 
 #include "oox/xls/biffcodec.hxx"
 #include <osl/thread.h>
+#include <string.h>
 
 using ::rtl::OString;
 using ::rtl::OUString;
--- ./package/source/zipapi/MemoryByteGrabber.hxx.gcc43~	2006-06-20 08:12:56.000000000 +0200
+++ ./package/source/zipapi/MemoryByteGrabber.hxx	2008-01-25 18:52:22.000000000 +0100
@@ -42,6 +42,8 @@
 #include <com/sun/star/io/XSeekable.hpp>
 #endif
 
+#include <string.h>
+
 class MemoryByteGrabber 
 {
 protected:
--- ./package/source/zipapi/XUnbufferedStream.cxx.gcc43~	2006-09-17 19:27:25.000000000 +0200
+++ ./package/source/zipapi/XUnbufferedStream.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -61,6 +61,8 @@
 #endif
 #include <algorithm>
 
+#include <string.h>
+
 #if 0
 // for debugging purposes here
 #include <com/sun/star/ucb/XSimpleFileAccess.hpp>
--- ./psprint/source/helper/helper.cxx.gcc43~	2006-09-16 14:34:24.000000000 +0200
+++ ./psprint/source/helper/helper.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -39,6 +39,7 @@
 #include <cstring>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <psprint/helper.hxx>
 #include <tools/string.hxx>
--- ./rdbmaker/source/rdbmaker/rdboptions.cxx.gcc43~	2006-06-20 06:10:04.000000000 +0200
+++ ./rdbmaker/source/rdbmaker/rdboptions.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -33,6 +33,7 @@
  *
  ************************************************************************/
 #include 	<stdio.h>
+#include	<string.h>
 
 #include	"rdboptions.hxx"
 
--- ./registry/tools/checksingleton.cxx.gcc43~	2007-10-15 14:28:07.000000000 +0200
+++ ./registry/tools/checksingleton.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_registry.hxx"
 #include <stdio.h>
+#include <string.h>
 
 #include <set>
 
--- ./sax/source/expatwrap/sax_expat.cxx.gcc43~	2006-06-20 01:07:03.000000000 +0200
+++ ./sax/source/expatwrap/sax_expat.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -33,6 +33,7 @@
  *
  ************************************************************************/
 #include <stdlib.h>
+#include <string.h>
 #include <sal/alloca.h>
 #include <vector>
 
--- ./sax/source/fastparser/fastparser.cxx.gcc43~	2008-01-25 18:08:33.000000000 +0100
+++ ./sax/source/fastparser/fastparser.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -66,6 +66,8 @@
 #include "expat/xmlparse.h"
 #endif
 
+#include <string.h>
+
 using ::rtl::OUString;
 using ::rtl::OString;
 using namespace ::std;
--- ./shell/source/unix/sysshell/recently_used_file_handler.cxx.gcc43~	2007-06-19 18:12:06.000000000 +0200
+++ ./shell/source/unix/sysshell/recently_used_file_handler.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -69,6 +69,7 @@
 #include <vector>
 #include <algorithm>
 #include <functional>
+#include <string.h>
 
 namespace /* private */ {
     
--- ./svx/source/msfilter/mscodec.cxx.gcc43~	2006-10-12 14:57:57.000000000 +0200
+++ ./svx/source/msfilter/mscodec.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -43,6 +43,7 @@
 #include <osl/diagnose.h>
 #include <algorithm>
 #include <tools/solar.h>
+#include <string.h>
 
 namespace svx {
 
--- ./tools/inc/bootstrp/command.hxx.gcc43~	2007-11-02 13:59:18.000000000 +0100
+++ ./tools/inc/bootstrp/command.hxx	2008-01-25 18:52:22.000000000 +0100
@@ -36,7 +36,7 @@
 #ifndef COMMAND_HXX
 #define COMMAND_HXX
 
-#include <iostream.h>
+#include <iostream>
 
 #include <tools/stream.hxx>
 #define STRLEN 100
--- ./tools/source/fsys/urlobj.cxx.gcc43~	2008-01-25 18:08:14.000000000 +0100
+++ ./tools/source/fsys/urlobj.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -81,6 +81,8 @@
 #define INCLUDED_LIMITS
 #endif
 
+#include <string.h>
+
 namespace unnamed_tools_urlobj {} using namespace unnamed_tools_urlobj;
 	// unnamed namespaces don't work well yet...
 
--- ./unoxml/source/dom/attr.cxx.gcc43~	2007-12-06 11:57:36.000000000 +0100
+++ ./unoxml/source/dom/attr.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -36,6 +36,7 @@
 #include "attr.hxx"
 #include "element.hxx"
 #include <com/sun/star/xml/dom/DOMException.hdl>
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/attributesmap.cxx.gcc43~	2006-06-20 02:43:00.000000000 +0200
+++ ./unoxml/source/dom/attributesmap.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "attributesmap.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/characterdata.cxx.gcc43~	2007-12-06 11:58:17.000000000 +0100
+++ ./unoxml/source/dom/characterdata.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -36,6 +36,7 @@
 #include <com/sun/star/xml/dom/events/XDocumentEvent.hpp>
 #include "characterdata.hxx"
 #include "../events/mutationevent.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/document.cxx.gcc43~	2007-12-06 12:17:14.000000000 +0100
+++ ./unoxml/source/dom/document.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -54,6 +54,8 @@
 #include "../events/uievent.hxx"
 #include "../events/mouseevent.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
--- ./unoxml/source/dom/documenttype.cxx.gcc43~	2006-06-20 02:44:45.000000000 +0200
+++ ./unoxml/source/dom/documenttype.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -36,7 +36,7 @@
 #include "documenttype.hxx"
 #include "entitiesmap.hxx"
 #include "notationsmap.hxx"
-
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/element.cxx.gcc43~	2007-12-06 11:59:29.000000000 +0100
+++ ./unoxml/source/dom/element.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -39,6 +39,7 @@
 #include "elementlist.hxx"
 #include "attributesmap.hxx"
 #include "../events/mutationevent.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/elementlist.cxx.gcc43~	2006-06-20 02:45:55.000000000 +0200
+++ ./unoxml/source/dom/elementlist.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "elementlist.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/entity.cxx.gcc43~	2006-06-20 02:46:26.000000000 +0200
+++ ./unoxml/source/dom/entity.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "entity.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/dom/entityreference.cxx.gcc43~	2006-06-20 02:46:48.000000000 +0200
+++ ./unoxml/source/dom/entityreference.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "entityreference.hxx"
+#include <string.h>
 namespace DOM
 {
   CEntityReference::CEntityReference(const xmlNodePtr /*aNodePtr*/)
--- ./unoxml/source/dom/node.cxx.gcc43~	2007-12-06 12:17:57.000000000 +0100
+++ ./unoxml/source/dom/node.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include <stdio.h>
+#include <string.h>
 #include "node.hxx"
 #include "element.hxx"
 #include "text.hxx"
--- ./unoxml/source/dom/notation.cxx.gcc43~	2005-09-09 12:03:31.000000000 +0200
+++ ./unoxml/source/dom/notation.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "notation.hxx"
+#include <string.h>
 namespace DOM
 {
     CNotation::CNotation(const xmlNotationPtr aNotationPtr)
--- ./unoxml/source/dom/processinginstruction.cxx.gcc43~	2007-12-06 12:01:02.000000000 +0100
+++ ./unoxml/source/dom/processinginstruction.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "processinginstruction.hxx"
+#include <string.h>
 
 namespace DOM
 {
--- ./unoxml/source/fastsax/fastparser.cxx.gcc43~	2007-06-05 15:52:07.000000000 +0200
+++ ./unoxml/source/fastsax/fastparser.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -39,6 +39,7 @@
 #include <stack>
 #include <vector>
 #include <hash_map>
+#include <string.h>
 #include <boost/scoped_ptr.hpp>
 #include <boost/shared_ptr.hpp>
 
--- ./vcl/unx/source/app/i18n_im.cxx.gcc43~	2006-09-17 14:31:47.000000000 +0200
+++ ./vcl/unx/source/app/i18n_im.cxx	2008-01-25 18:52:22.000000000 +0100
@@ -37,6 +37,7 @@
 #include "precompiled_vcl.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #ifdef LINUX
 #  ifndef __USE_XOPEN
 #    define __USE_XOPEN
--- ./writerfilter/source/dmapper/DomainMapper_Impl.hxx.gcc43~	2008-01-25 18:08:35.000000000 +0100
+++ ./writerfilter/source/dmapper/DomainMapper_Impl.hxx	2008-01-25 18:52:47.000000000 +0100
@@ -50,6 +50,7 @@
 #endif
 #include <vector>
 #include <stack>
+#include <cstring>
 #ifdef GCC_VISIBILITY_BROKEN
 #pragma GCC visibility pop
 #endif
--- ./writerfilter/source/dmapper/GraphicImport.cxx.gcc43~	2007-12-12 09:00:48.000000000 +0100
+++ ./writerfilter/source/dmapper/GraphicImport.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -104,7 +104,7 @@
 #endif
 #include <rtl/ustrbuf.hxx>
 
-
+#include <cstring>
 #include <iostream>
 #include <resourcemodel/QNameToString.hxx>
 
--- ./writerfilter/source/odiapi/qname/QName.cxx.gcc43~	2008-01-25 18:08:35.000000000 +0100
+++ ./writerfilter/source/odiapi/qname/QName.cxx	2008-01-25 18:53:12.000000000 +0100
@@ -55,6 +55,7 @@
 #pragma GCC visibility push(default)
 #endif
 #include <string>
+#include <cstring>
 #ifdef GCC_VISIBILITY_BROKEN
 #pragma GCC visibility pop
 #endif
--- ./writerfilter/unocomponent/debugservices/rtfsl/RTFSLParser.cxx.gcc43~	2008-01-25 18:08:35.000000000 +0100
+++ ./writerfilter/unocomponent/debugservices/rtfsl/RTFSLParser.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -74,6 +74,7 @@
 #include <odiapi/qname/QName.hxx>
 #include <odiapi/props/Properties.hxx>
 #include <stack>
+#include <cstring>
 
 #include <ctype.h>
 
--- ./writerfilter/unocomponent/debugservices/rtftok/ScannerTestService.cxx.gcc43~	2008-01-25 18:08:35.000000000 +0100
+++ ./writerfilter/unocomponent/debugservices/rtftok/ScannerTestService.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -72,6 +72,8 @@
 #include <com/sun/star/beans/XPropertySet.hpp>
 #include <comphelper/seqstream.hxx>
 
+#include <cstring>
+
 #include <ctype.h>
 
 using namespace ::com::sun::star;
--- ./writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx.gcc43~	2008-01-25 18:08:35.000000000 +0100
+++ ./writerfilter/unocomponent/debugservices/rtftok/XMLScanner.cxx	2008-01-25 18:53:37.000000000 +0100
@@ -67,6 +67,7 @@
 #pragma GCC visibility push(default)
 #endif
 #include <string>
+#include <cstring>
 #ifdef GCC_VISIBILITY_BROKEN
 #pragma GCC visibility pop
 #endif
--- ./writerperfect/source/filter/DocumentCollector.hxx.gcc43~	2008-01-25 18:08:08.000000000 +0100
+++ ./writerperfect/source/filter/DocumentCollector.hxx	2008-01-25 18:52:13.000000000 +0100
@@ -43,6 +43,7 @@
 #include <vector>
 #include <map>
 #include <stack>
+#include <cstring>
 
 class DocumentElement;
 class DocumentHandler;
--- ./writerperfect/source/filter/TableStyle.cxx.gcc43~	2007-01-23 13:15:30.000000000 +0100
+++ ./writerperfect/source/filter/TableStyle.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -30,6 +30,7 @@
 #include "FilterInternal.hxx"
 #include "TableStyle.hxx"
 #include "DocumentElement.hxx"
+#include <cstring>
 
 #ifdef _MSC_VER
 #include <minmax.h>
--- ./writerperfect/source/filter/TextRunStyle.cxx.gcc43~	2008-01-25 18:08:11.000000000 +0100
+++ ./writerperfect/source/filter/TextRunStyle.cxx	2008-01-25 18:52:13.000000000 +0100
@@ -32,6 +32,8 @@
 #include "WriterProperties.hxx"
 #include "DocumentElement.hxx"
 
+#include <cstring>
+
 #ifdef _MSC_VER
 #include <minmax.h>
 #endif
--- ./xmlhelp/source/com/sun/star/help/HelpCompiler.cxx.gcc43~	2007-11-19 13:58:51.000000000 +0100
+++ ./xmlhelp/source/com/sun/star/help/HelpCompiler.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -37,6 +37,7 @@
 #include "HelpCompiler.hxx"
 #include <limits.h>
 #include <stdlib.h>
+#include <string.h>
 #include <libxslt/xslt.h>
 #include <libxslt/xsltInternals.h>
 #include <libxslt/transform.h>
--- ./xmlhelp/source/com/sun/star/help/HelpLinker.cxx.gcc43~	2007-11-26 19:03:59.000000000 +0100
+++ ./xmlhelp/source/com/sun/star/help/HelpLinker.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -53,6 +53,9 @@
 #include <sal/types.h>
 #include <osl/time.h>
 
+#include <string.h>
+#include <limits.h>
+
 class JarOutputStream
 {
 private:
--- ./xmlhelp/source/cxxhelp/provider/databases.cxx.gcc43~	2008-01-25 18:08:14.000000000 +0100
+++ ./xmlhelp/source/cxxhelp/provider/databases.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -74,6 +74,7 @@
 #endif
 #include "inputstream.hxx"
 #include <algorithm>
+#include <cstring>
 
 // Extensible help
 #include "com/sun/star/deployment/thePackageManagerFactory.hpp"
--- ./xmloff/inc/xmloff/PropertySetInfoHash.hxx.gcc43~	2007-04-11 15:20:28.000000000 +0200
+++ ./xmloff/inc/xmloff/PropertySetInfoHash.hxx	2008-01-25 18:52:21.000000000 +0100
@@ -40,6 +40,7 @@
 #endif
 
 #include <memory>
+#include <string.h>
 
 struct PropertySetInfoHash
 {
--- ./xmloff/inc/xmloff/nmspmap.hxx.gcc43~	2007-04-11 15:27:35.000000000 +0200
+++ ./xmloff/inc/xmloff/nmspmap.hxx	2008-01-25 18:52:21.000000000 +0100
@@ -61,6 +61,8 @@
 #include <cppuhelper/weak.hxx>
 #endif
 
+#include <limits.h>
+
 const sal_uInt16 XML_NAMESPACE_XMLNS  = (USHRT_MAX-2);
 const sal_uInt16 XML_NAMESPACE_NONE  = (USHRT_MAX-1);
 const sal_uInt16 XML_NAMESPACE_UNKNOWN  = (USHRT_MAX);
--- ./xmloff/source/draw/EnhancedCustomShapeToken.cxx.gcc43~	2006-09-17 12:24:05.000000000 +0200
+++ ./xmloff/source/draw/EnhancedCustomShapeToken.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -43,6 +43,7 @@
 #include <osl/mutex.hxx>
 #endif
 #include <hash_map>
+#include <cstring>
 
 namespace xmloff { namespace EnhancedCustomShapeToken {
 
--- ./xmlsecurity/source/helper/xsecparser.cxx.gcc43~	2006-09-16 16:40:42.000000000 +0200
+++ ./xmlsecurity/source/helper/xsecparser.cxx	2008-01-25 18:52:21.000000000 +0100
@@ -43,6 +43,8 @@
 #endif
 #include "cppuhelper/exc_hlp.hxx"
 
+#include <cstring>
+
 namespace cssu = com::sun::star::uno;
 namespace cssxs = com::sun::star::xml::sax;
 
