--- toolkit/inc/toolkit/awt/vclxaccessiblelist.hxx.old	2005-09-09 14:36:25.000000000 +0200
+++ toolkit/inc/toolkit/awt/vclxaccessiblelist.hxx	2005-11-22 10:24:49.000000000 +0100
@@ -190,7 +190,7 @@
     /** This function is called from the implementation helper during a
         XComponent::dispose call.  Free the list of items and the items themselves.
     */
-    virtual void SAL_CALL VCLXAccessibleList::disposing (void);
+    virtual void SAL_CALL disposing (void);
 
     /** This method adds the states
         <const>AccessibleStateType::FOCUSABLE</const> and possibly
