--- connectivity/source/drivers/evoab2/NStatement.cxx	2006-07-13 21:10:46.000000000 +0200
+++ connectivity/source/drivers/evoab2/NStatement.cxx	2006-07-13 21:10:43.000000000 +0200
@@ -94,8 +94,8 @@ OStatement_Base::OStatement_Base(OEvoabC
 	  m_xResultSet(NULL),
 	  m_pResultSet(NULL),
 	  m_pParseTree(NULL),
-	  m_aSQLIterator(_pConnection->createCatalog()->getTables(), _pConnection->getMetaData(), NULL),
-	  m_aParser(_pConnection->getDriver()->getMSFactory())
+	  m_aParser(_pConnection->getDriver()->getMSFactory()),
+	  m_aSQLIterator(_pConnection, _pConnection->createCatalog()->getTables(), m_aParser, NULL)
 {
 	m_pConnection->acquire();
 }
