--- desktop/source/deployment/manager/dp_manager.cxx
+++ desktop/source/deployment/manager/dp_manager.cxx
@@ -619,7 +619,7 @@
             if (m_activePackagesDB->has( title ))
             {
                 // package already deployed, interact --force:
-                Any nameClashResolveRequest(
+                Any nameClashResolveRequest = Any(
                     NameClashResolveRequest(
                         getResourceString(
                             RID_STR_PACKAGE_ALREADY_ADDED ) + title,
