--- xmloff/source/draw/animationimport.cxx.old	2005-10-19 14:21:49.000000000 +0200
+++ xmloff/source/draw/animationimport.cxx	2005-11-22 17:34:20.000000000 +0100
@@ -1359,7 +1359,7 @@
 	AnimationsImport( const Reference< XMultiServiceFactory > & rSMgr );
 	~AnimationsImport() throw ();
 
-	SvXMLImportContext *AnimationsImport::CreateContext(USHORT nPrefix, const OUString& rLocalName,	const Reference<XAttributeList>& xAttrList);
+	SvXMLImportContext* CreateContext(USHORT nPrefix, const OUString& rLocalName,	const Reference<XAttributeList>& xAttrList);
 
 	// XInterface
     virtual Any SAL_CALL queryInterface( const Type& aType ) throw (RuntimeException);
@@ -1367,7 +1367,7 @@
     virtual void SAL_CALL release() throw ();
 
 	// XAnimationNodeSupplier
-	Reference< XAnimationNode > SAL_CALL AnimationsImport::getAnimationNode() throw (RuntimeException);
+	Reference< XAnimationNode > SAL_CALL getAnimationNode() throw (RuntimeException);
 
 	// XServiceInfo
     virtual OUString SAL_CALL getImplementationName() throw(RuntimeException);
--- xmloff/source/forms/elementexport.hxx.old	2005-09-09 16:05:46.000000000 +0200
+++ xmloff/source/forms/elementexport.hxx	2005-11-22 17:39:58.000000000 +0100
@@ -267,7 +267,7 @@
 
         /** retrieves the string specifying the ListSource of a list or combo box
         */
-        ::rtl::OUString OControlExport::getScalarListSourceValue() const;
+        ::rtl::OUString getScalarListSourceValue() const;
 
         /** determines whether the list entries (of a combo or list box) are supplied by the user
 
--- xmloff/inc/shapeimport.hxx.old	2005-09-09 15:08:16.000000000 +0200
+++ xmloff/inc/shapeimport.hxx	2005-11-22 17:22:41.000000000 +0100
@@ -373,7 +373,7 @@
 		const com::sun::star::uno::Reference< com::sun::star::xml::sax::XAttributeList >& xAttrList,
 		com::sun::star::uno::Reference< com::sun::star::drawing::XShapes >& rShapes,
 		const com::sun::star::uno::Reference< com::sun::star::xml::sax::XAttributeList >& xFrameAttrList);
-	SvXMLImportContext *XMLShapeImportHelper::CreateFrameChildContext( 
+	SvXMLImportContext* CreateFrameChildContext( 
 		SvXMLImportContext *pThisContext, USHORT nPrefix, const rtl::OUString& rLocalName,
 		const com::sun::star::uno::Reference< com::sun::star::xml::sax::XAttributeList >& xAttrList );
 
--- xmloff/inc/SchXMLExportHelper.hxx.old	2005-09-09 14:40:58.000000000 +0200
+++ xmloff/inc/SchXMLExportHelper.hxx	2005-11-22 17:24:45.000000000 +0100
@@ -194,7 +194,7 @@
 					  sal_Bool bIncludeTable );
 
 	/// returns the string corresponding to the current FileFormat CLSID for Chart
-	const rtl::OUString& SchXMLExportHelper::getChartCLSID();
+	const rtl::OUString& getChartCLSID();
 
 	UniReference< XMLPropertySetMapper > GetPropertySetMapper() const { return mxPropertySetMapper; }
 
--- xmloff/source/chart/SchXMLTableContext.hxx.old	2005-09-09 15:26:59.000000000 +0200
+++ xmloff/source/chart/SchXMLTableContext.hxx	2005-11-22 17:26:34.000000000 +0100
@@ -98,7 +98,7 @@
 
 public:
 	/// The data for the ChartDocument is applied linearly
-	static void SchXMLTableHelper::applyTableSimple(
+	static void applyTableSimple(
 		const SchXMLTable& rTable,
 		com::sun::star::uno::Reference< com::sun::star::chart::XChartDocument > xChartDoc );
 
--- xmloff/inc/txtflde.hxx.old	2005-09-09 15:08:51.000000000 +0200
+++ xmloff/inc/txtflde.hxx	2005-11-22 17:48:37.000000000 +0100
@@ -248,7 +248,7 @@
 	SvXMLExport& GetExport() { return rExport; }
 
 	/// export a field after <text:span> is already written
-	void XMLTextFieldExport::ExportFieldHelper(
+	void ExportFieldHelper(
 		const ::com::sun::star::uno::Reference<
 				::com::sun::star::text::XTextField> & rTextField,
 		const ::com::sun::star::uno::Reference<
--- xmloff/inc/XMLFootnoteConfigurationImportContext.hxx.old	2005-09-09 14:50:45.000000000 +0200
+++ xmloff/inc/XMLFootnoteConfigurationImportContext.hxx	2005-11-22 17:45:55.000000000 +0100
@@ -113,8 +113,7 @@
     /// set configuration at document; calls ProcessSettings
     // --> OD 2005-01-31 #i40579# - move code from <CreateAndInsertLate(..)>
     // to <Finish(..)>, because at this time all styles it references have been set.
-    virtual void XMLFootnoteConfigurationImportContext::Finish(
-		sal_Bool bOverwrite );
+    virtual void Finish( sal_Bool bOverwrite);
     // <--
 
 	/// set configuration at document
--- xmloff/inc/xmlnumi.hxx.old	2005-09-09 15:17:51.000000000 +0200
+++ xmloff/inc/xmlnumi.hxx	2005-11-22 17:31:28.000000000 +0100
@@ -106,11 +106,11 @@
 
 	static ::com::sun::star::uno::Reference <
 		::com::sun::star::container::XIndexReplace >
-	SvxXMLListStyleContext::CreateNumRule(
+	CreateNumRule(
 		const ::com::sun::star::uno::Reference <
 			::com::sun::star::frame::XModel > & rModel );
 
-	static void SvxXMLListStyleContext::SetDefaultStyle(
+	static void SetDefaultStyle(
 		const ::com::sun::star::uno::Reference <
 			::com::sun::star::container::XIndexReplace > & rNumRule,
 		sal_Int16 nLevel,
