--- dbaccess/source/ui/misc/UITools.cxx.ark	2006-11-08 14:08:43.000000000 +0100
+++ dbaccess/source/ui/misc/UITools.cxx	2006-11-08 14:10:46.000000000 +0100
@@ -739,19 +739,19 @@
 			pInfo->bNullable		= (sal_Int32)aValue == ColumnValue::NULLABLE;
 			++nPos;
             aValue.fill(nPos,aTypes[nPos],xRow);
-			pInfo->bCaseSensitive	= aValue;
+			pInfo->bCaseSensitive	= (sal_Bool)aValue;
 			++nPos;
             aValue.fill(nPos,aTypes[nPos],xRow);
 			pInfo->nSearchType		= aValue;
 			++nPos;
             aValue.fill(nPos,aTypes[nPos],xRow);
-			pInfo->bUnsigned		= aValue;
+			pInfo->bUnsigned		= (sal_Bool)aValue;
 			++nPos;
             aValue.fill(nPos,aTypes[nPos],xRow);
-			pInfo->bCurrency		= aValue;
+			pInfo->bCurrency		= (sal_Bool)aValue;
 			++nPos;
             aValue.fill(nPos,aTypes[nPos],xRow);
-			pInfo->bAutoIncrement	= aValue;
+			pInfo->bAutoIncrement	= (sal_Bool)aValue;
 			++nPos;
             aValue.fill(nPos,aTypes[nPos],xRow);
 			pInfo->aLocalTypeName	= aValue;
--- dbaccess/source/ui/misc/DExport.cxx.ark	2006-11-08 14:26:16.000000000 +0100
+++ dbaccess/source/ui/misc/DExport.cxx	2006-11-08 14:27:26.000000000 +0100
@@ -321,19 +321,19 @@
 				m_pTypeInfo->bNullable		= (sal_Int32)aValue == ColumnValue::NULLABLE;
 				++nPos;
 				aValue.fill(nPos,aTypes[nPos],xRow);
-				m_pTypeInfo->bCaseSensitive	= aValue;
+				m_pTypeInfo->bCaseSensitive	= (sal_Bool)aValue;
 				++nPos;
 				aValue.fill(nPos,aTypes[nPos],xRow);
 				m_pTypeInfo->nSearchType		= aValue;
 				++nPos;
 				aValue.fill(nPos,aTypes[nPos],xRow);
-				m_pTypeInfo->bUnsigned		= aValue;
+				m_pTypeInfo->bUnsigned		= (sal_Bool)aValue;
 				++nPos;
 				aValue.fill(nPos,aTypes[nPos],xRow);
-				m_pTypeInfo->bCurrency		= aValue;
+				m_pTypeInfo->bCurrency		= (sal_Bool)aValue;
 				++nPos;
 				aValue.fill(nPos,aTypes[nPos],xRow);
-				m_pTypeInfo->bAutoIncrement	= aValue;
+				m_pTypeInfo->bAutoIncrement	= (sal_Bool)aValue;
 				++nPos;
 				aValue.fill(nPos,aTypes[nPos],xRow);
 				m_pTypeInfo->aLocalTypeName	= aValue;
