--- configmgr/source/backend/binaryreadhandler.hxx.old	2005-09-08 05:26:54.000000000 +0200
+++ configmgr/source/backend/binaryreadhandler.hxx	2005-11-07 16:42:41.000000000 +0100
@@ -132,26 +132,26 @@
 			void readChildNodes(ISubtree & rSubTree)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
-            binary::NodeType::Type BinaryReadHandler::readNodeType()
+            binary::NodeType::Type readNodeType()
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
-			binary::ValueFlags::Type BinaryReadHandler::readValueFlags(bool& bSeq, bool& hasValue, bool& hasDefault )
+			binary::ValueFlags::Type readValueFlags(bool& bSeq, bool& hasValue, bool& hasDefault )
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 
-			void BinaryReadHandler::readString(rtl::OUString &_aString)
+			void readString(rtl::OUString &_aString)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 
-			void BinaryReadHandler::readAttributes(node::Attributes  &_aAttributes)
+			void readAttributes(node::Attributes  &_aAttributes)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 
-			void BinaryReadHandler::readGroup(rtl::OUString &_aName, node::Attributes &_aAttributes)
+			void readGroup(rtl::OUString &_aName, node::Attributes &_aAttributes)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 
-			void BinaryReadHandler::readSet(rtl::OUString &_aName, node::Attributes &_aAttributes,
+			void readSet(rtl::OUString &_aName, node::Attributes &_aAttributes,
 											rtl::OUString &_sInstanceName, rtl::OUString &_sInstanceModule)
 			    SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
-			void BinaryReadHandler::readValue(rtl::OUString &_aName, node::Attributes &_aAttributes,
+			void readValue(rtl::OUString &_aName, node::Attributes &_aAttributes,
 											  uno::Any& _aValue, uno::Any& _aDefaultValue,uno::Type& _aType)
 			    SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
--- configmgr/source/backend/binarywritehandler.hxx.old	2005-09-08 05:27:22.000000000 +0200
+++ configmgr/source/backend/binarywritehandler.hxx	2005-11-07 16:45:15.000000000 +0100
@@ -128,16 +128,16 @@
 			void writeAttributes(node::Attributes const& _aAttributes)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
-            void BinaryWriteHandler::writeNodeType(binary::NodeType::Type _eType)
+            void writeNodeType(binary::NodeType::Type _eType)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
-			void BinaryWriteHandler::writeStop()
+			void writeStop()
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
 			void writeType(uno::Type const& _aType)
     			SAL_THROW( (io::IOException, uno::RuntimeException) );
 			
-			void BinaryWriteHandler::writeValue( uno::Any const& _aValue)
+			void writeValue( uno::Any const& _aValue)
 				SAL_THROW( (io::IOException, uno::RuntimeException) );
 		
 		};
