--- sc/source/ui/Accessibility/AccessibleDocument.cxx.old	2005-09-08 22:17:14.000000000 +0200
+++ sc/source/ui/Accessibility/AccessibleDocument.cxx	2005-11-25 10:35:17.000000000 +0100
@@ -320,7 +320,7 @@
     void SetDrawBroadcaster();
 
     sal_Int32 GetCount() const;
-    uno::Reference< XAccessible > ScChildrenShapes::Get(const ScAccessibleShapeData* pData) const;
+    uno::Reference< XAccessible > Get(const ScAccessibleShapeData* pData) const;
     uno::Reference< XAccessible > Get(sal_Int32 nIndex) const;
 	uno::Reference< XAccessible > GetAt(const awt::Point& rPoint) const;
 
@@ -363,7 +363,7 @@
     ScAddress* GetAnchor(const uno::Reference<drawing::XShape>& xShape) const;
     uno::Reference<XAccessibleRelationSet> GetRelationSet(const ScAccessibleShapeData* pData) const;
     void CheckWhetherAnchorChanged(const uno::Reference<drawing::XShape>& xShape) const;
-    void ScChildrenShapes::SetAnchor(const uno::Reference<drawing::XShape>& xShape, ScAccessibleShapeData* pData) const;
+    void SetAnchor(const uno::Reference<drawing::XShape>& xShape, ScAccessibleShapeData* pData) const;
     void AddShape(const uno::Reference<drawing::XShape>& xShape, sal_Bool bCommitChange) const;
     void RemoveShape(const uno::Reference<drawing::XShape>& xShape) const;
 
