--- filter/source/xmlfilterdetect/filterdetect.hxx.old	2005-09-08 23:59:46.000000000 +0200
+++ filter/source/xmlfilterdetect/filterdetect.hxx	2005-11-24 18:43:57.000000000 +0100
@@ -136,7 +136,7 @@
 
 
  	//XExtendedFilterDetection
-     virtual ::rtl::OUString SAL_CALL FilterDetect::detect( com::sun::star::uno::Sequence< com::sun::star::beans::PropertyValue >& lDescriptor ) 
+     virtual ::rtl::OUString SAL_CALL detect( com::sun::star::uno::Sequence< com::sun::star::beans::PropertyValue >& lDescriptor ) 
 			throw( com::sun::star::uno::RuntimeException );
 
 
--- filter/source/filtertracer/filtertracer.hxx.old	2005-09-08 23:40:34.000000000 +0200
+++ filter/source/filtertracer/filtertracer.hxx	2005-11-24 18:40:33.000000000 +0100
@@ -182,7 +182,7 @@
 	REF( NMSP_UTIL::XTextSearch )		mxTextSearch;
 	NMSP_UTIL::SearchOptions			maSearchOptions;
 
-	sal_Bool				FilterTracer::ImplFilter( const rtl::OUString& rFilter, const rtl::OUString& rString );
+	sal_Bool				ImplFilter( const rtl::OUString& rFilter, const rtl::OUString& rString );
 
 public:
 							FilterTracer( const REF( NMSP_LANG::XMultiServiceFactory )& rxMgr );
