--- sd/source/ui/unoidl/unomodel.cxx	2005-04-04 15:08:14.582780314 +0200
+++ sd/source/ui/unoidl/unomodel.cxx	2005-04-04 15:08:26.384748695 +0200
@@ -1927,7 +1927,7 @@ void SAL_CALL SdXImpressDocument::render
 				vcl::PDFExtOutDevData* pPDFExtOutDevData = PTR_CAST( vcl::PDFExtOutDevData, pOut->GetExtOutDevData() );
 
 	            ::sd::ClientView* pView = new ::sd::ClientView( pDocShell, pOut, NULL );
-				Rectangle				aVisArea( Point(), pDoc->GetSdPage( (USHORT)nPageNumber - 1, ePageKind )->GetSize() );	
+				Rectangle				aVisArea = Rectangle( Point(), pDoc->GetSdPage( (USHORT)nPageNumber - 1, ePageKind )->GetSize() );	
         		Region					aRegion( aVisArea );
                 Point					aOrigin;
 
