--- transex3/inc/export.hxx.old	2005-09-23 16:27:41.000000000 +0200
+++ transex3/inc/export.hxx	2005-11-04 12:16:45.000000000 +0100
@@ -347,7 +347,7 @@
 	BOOL WriteExportList( ResData *pResData, ExportList *pExportList,
 						const ByteString &rTyp, BOOL bCreateNew = FALSE );
 	
-	ByteString Export::MergePairedList( ByteString& sLine , ByteString& sText );
+	ByteString MergePairedList( ByteString& sLine , ByteString& sText );
 
 	ByteString FullId();					// creates cur. GID
     
@@ -355,7 +355,7 @@
     
 	ByteString GetPairedListID		( const ByteString& sText );
     ByteString GetPairedListString	( const ByteString& sText );
-	ByteString Export::StripList	( const ByteString& sText );
+	ByteString StripList	( const ByteString& sText );
 
 	void UnmergeUTF8( ByteString& sOrig );
 	void InsertListEntry( const ByteString &rText, const ByteString &rLine );
--- transex3/inc/helpmerge.hxx.old	2005-09-09 16:55:29.000000000 +0200
+++ transex3/inc/helpmerge.hxx	2005-11-04 12:19:16.000000000 +0100
@@ -77,8 +77,8 @@
     bool Merge( const ByteString &rSDFFile, const ByteString &rPathX , const ByteString &rPathY , bool bISO );
 
 private:
-    ByteString HelpParser::GetOutpath( const ByteString& rPathX , const ByteString& sCur , const ByteString& rPathY );
+    ByteString GetOutpath( const ByteString& rPathX , const ByteString& sCur , const ByteString& rPathY );
     void Process( LangHashMap* aLangHM , ByteString& sCur , ResData *pResData , MergeDataFile& aMergeDataFile );
     void ProcessHelp( LangHashMap* aLangHM , ByteString& sCur , ResData *pResData , MergeDataFile& aMergeDataFile );
-    void HelpParser::MakeDir( const ByteString& sPath );
+    void MakeDir( const ByteString& sPath );
 };
--- transex3/inc/xmlparse.hxx.old	2005-09-09 16:58:09.000000000 +0200
+++ transex3/inc/xmlparse.hxx	2005-11-04 12:18:37.000000000 +0100
@@ -322,7 +322,7 @@
     /// Return the alpha strings representation
     ByteString	        GetIsoLangByIndex( USHORT nIndex );
     
-    static XMLUtil&     XMLUtil::Instance(); 
+    static XMLUtil&     Instance(); 
     ~XMLUtil();
 
     void         dump();
