--- sfx2/source/appl/newhelp.cxx.ark	2005-03-19 01:46:16.000000000 +0100
+++ sfx2/source/appl/newhelp.cxx	2005-03-19 01:46:34.000000000 +0100
@@ -249,9 +249,9 @@
 #include <vcl/waitobj.hxx>
 #include <unotools/ucbhelper.hxx>
 
-#include <sfx2/viewfrm.hxx>
-#include <sfx2/objsh.hxx>
-#include <sfx2/docfac.hxx>
+#include <../../inc/viewfrm.hxx>
+#include <../../inc/objsh.hxx>
+#include <../../inc/docfac.hxx>
 
 using namespace ::ucb;
 using namespace com::sun::star::ucb;
