--- comphelper/inc/comphelper/scopeguard.hxx.ark	2005-03-14 20:21:49.000000000 +0100
+++ comphelper/inc/comphelper/scopeguard.hxx	2005-03-14 21:43:24.000000000 +0100
@@ -69,7 +69,7 @@
 
 /** ScopeGuard to ease writing exception-safe code.
  */
-class ScopeGuard : private boost::noncopyable // noncopyable until we have
+class COMPHELPER_DLLPUBLIC ScopeGuard : private boost::noncopyable // noncopyable until we have
                                               // good reasons...
 {
 public:
