--- basic/inc/sbprop.hxx.old	2005-11-21 15:55:23.000000000 +0100
+++ basic/inc/sbprop.hxx	2005-11-22 18:08:34.000000000 +0100
@@ -76,7 +76,7 @@
 	SB_DLLPRIVATE virtual ~SbProcedureProperty();
 
 public:
-	SbProcedureProperty::SbProcedureProperty( const String& r, SbxDataType t )
+	SbProcedureProperty( const String& r, SbxDataType t )
 			: SbxProperty( r, t ) // , pMod( p )
 			, mbSet( false )
 	{}
