Index: svx/source/fmcomp/fmgridcl.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/fmcomp/fmgridcl.cxx,v
retrieving revision 1.48
diff -u -r1.48 fmgridcl.cxx
--- svx/source/fmcomp/fmgridcl.cxx	13 Apr 2005 08:27:50 -0000	1.48
+++ svx/source/fmcomp/fmgridcl.cxx	20 Apr 2005 11:14:22 -0000
@@ -1135,10 +1135,10 @@
 			else
 			{
 				// Standardlabel setzen
-				::rtl::OUString sLabelBase(String(SVX_RES(RID_STR_COLUMN)));
+				String sLabelBase(SVX_RES(RID_STR_COLUMN));
 				// disambiguate the name
 				Reference< XNameAccess > xColNames(xCols, UNO_QUERY);
-				::rtl::OUString sLabel;
+				rtl::OUString sLabel;
 				for (sal_Int32 i=1; i<65535; ++i)
 				{
 					sLabel = sLabelBase;
