--- solenv/bin/deliver.pl	2005-12-20 16:36:12.000000000 +0100
+++ solenv/bin/deliver.pl	2005-12-20 16:39:29.000000000 +0100
@@ -116,11 +116,12 @@ if ( $ENV{GUI} eq 'WNT' ) {
         warn("Warning: do we need stripping for windows gcc? Nothing defined yet.");
     }
 } else {
-    $strip = 'strip' if (((defined $ENV{ENABLE_SYMBOLS}) && ($ENV{ENABLE_SYMBOLS} ne "TRUE") && ($ENV{ENABLE_SYMBOLS} ne "SMALL")) || (!defined $ENV{ENABLE_SYMBOLS}));
+    if (((defined $ENV{ENABLE_SYMBOLS}) && ($ENV{ENABLE_SYMBOLS} ne "TRUE") && ($ENV{ENABLE_SYMBOLS} ne "SMALL")) || (!defined $ENV{ENABLE_SYMBOLS})) {
+        $strip = 'strip';
+        $strip .= " -R '.comment' -s" if ($ENV{OS} eq 'LINUX');
+    }
 }
 
-$strip .= " -R '.comment' -s" if ($ENV{OS} eq 'LINUX');
-
 $upd           = $ENV{'UPD'};
 ($gui		= lc($ENV{GUI})) 		|| die "can't determine GUI";
 $tempcounter        = 0;
