--- scp2/source/base/file_base.scp	2005-07-07 15:57:08.526055527 +0200
+++ scp2/source/base/file_base.scp	2005-07-07 16:04:56.511797555 +0200
@@ -75,12 +75,16 @@ End
 
 #endif
 
+#ifdef SOLAR_JAVA
+
 File gid_File_Help_Sdatabase_Zip
     Dir = gid_Dir_Help_Isolanguage;
     ARCHIVE_TXT_FILE_BODY_PATCH;
     EXTRA_ALL_LANG(sdatabase);
 End
 
+#endif
+
 File gid_File_Registry_Spool_Oo_Setup_Base_Xcu
     TXT_FILE_BODY;
     Styles = (PACKED);
