--- ridljar/source/unoloader/com/sun/star/lib/unoloader/makefile.mk	2005-06-17 11:45:10.000000000 +0200
+++ ridljar/source/unoloader/com/sun/star/lib/unoloader/makefile.mk	2005-06-24 17:26:09.836032814 +0200
@@ -81,6 +81,8 @@ CUSTOMMANIFESTFILE = manifest
 
 .INCLUDE: target.mk
 
+.IF "$(SOLAR_JAVA)" != ""
+
 .IF "$(depend)" == ""
 $(JAVACLASSFILES): $(MISC)$/$(TARGET).mkdir.done
 .ENDIF
@@ -88,3 +90,5 @@ $(JAVACLASSFILES): $(MISC)$/$(TARGET).mk
 $(MISC)$/$(TARGET).mkdir.done .ERRREMOVE:
     $(MKDIR) $(CLASSDIR)
     $(TOUCH) $@
+
+.ENDIF
