--- qadevOOo/runner/helper/PropertyHandlerImpl.java	14 Mar 2006 11:49:09 -0000	1.2
+++ qadevOOo/runner/helper/PropertyHandlerImpl.java	31 Jul 2006 16:06:58 -0000
@@ -122,12 +122,12 @@ public class PropertyHandlerImpl impleme
      *
      * @throws com.sun.star.lang.NullPointerException ::com::sun::star::lang::NullPointerException if ControlFactory is NULL .
      */
-    public void describePropertyLine(
+    public com.sun.star.inspection.LineDescriptor describePropertyLine(
             String PropertyName,
-            com.sun.star.inspection.LineDescriptor[] out_Descriptor,
             com.sun.star.inspection.XPropertyControlFactory ControlFactory)
             throws com.sun.star.beans.UnknownPropertyException,
             com.sun.star.lang.NullPointerException {
+        return null;
     }
     
     /**
