--- solenv/bin/modules/installer/languages.pm.all_lang	2005-09-08 05:04:40.000000000 -0400
+++ solenv/bin/modules/installer/languages.pm	2005-10-23 09:33:18.000000000 -0400
@@ -64,6 +64,12 @@
 		$installer::globals::unixmultipath = $installer::globals::languagelist;
 		$installer::globals::unixmultipath =~ s/\,/\_/g;	# hashes not allowed, comma to underline
 		$installer::globals::alllanguagesinproductarrayref = installer::converter::convert_stringlist_into_array(\$installer::globals::unixmultipath, "_");
+
+		$installer::globals::unixmultipath_orig = $installer::globals::unixmultipath;
+		if (length($installer::globals::unixmultipath) > 120) {
+			chomp(my $shorter = `echo $installer::globals::unixmultipath | md5sum | sed -e "s/  -//g"` );
+			$installer::globals::unixmultipath = $shorter;
+		}
 	}	
 	
 	while ($first =~ /^(\S+)\#(\S+?)$/)	# Minimal matching, to keep the order of languages
--- solenv/bin/modules/installer/globals.pm.all_lang	2005-10-20 14:49:34.000000000 -0400
+++ solenv/bin/modules/installer/globals.pm	2005-10-23 09:15:28.000000000 -0400
@@ -124,6 +124,7 @@
 	$addpackagelist = "";
 	$is_unix_multi = 0;
 	$unixmultipath = "";
+	$unixmultipath_orig = "";
 	$alllanguagesinproductarrayref = "";
 	$shiptestdirectory = "";
 	$makelinuxlinkrpm = 0;
--- solenv/bin/make_installer.pl.all_lang	2005-10-20 14:49:34.000000000 -0400
+++ solenv/bin/make_installer.pl	2005-10-23 09:31:02.000000000 -0400
@@ -515,13 +515,29 @@
 	else { $logminor = $installer::globals::minor; }
 
 	my $loglanguagestring = $$languagestringref;
-	if ( $installer::globals::is_unix_multi ) { $loglanguagestring = $installer::globals::unixmultipath; }
+	if ( $installer::globals::is_unix_multi ) { $loglanguagestring = $installer::globals::unixmultipath_orig; }
+
+	my $loglanguagestring_orig = $loglanguagestring;
+	if (length($loglanguagestring) > 120) {
+	    chomp(my $shorter = `echo $loglanguagestring | md5sum | sed -e "s/  -//g"`);
+	    $loglanguagestring = $shorter;
+	}
 
 	$installer::globals::logfilename = "log_" . $installer::globals::build . "_" . $logminor . "_" . $loglanguagestring . ".log";
 
 	if (( ! $installer::globals::is_unix_multi ) || ( $isfirstrun )) { $loggingdir = $loggingdir . $loglanguagestring . $installer::globals::separator; }
 
 	installer::systemactions::create_directory($loggingdir);
+
+	if ($loglanguagestring ne $loglanguagestring_orig) {
+	    (my $dir = $loggingdir) =~ s!/$!!;
+	    open(my $F1, "> $dir.dir");
+	    open(my $F2, "> " . $loggingdir . $installer::globals::logfilename . '.file');
+	    my @s = map { "$_\n" } split('_', $loglanguagestring_orig);
+	    print $F1 @s;
+	    print $F2 @s;
+	}
+
 	$installer::globals::exitlog = $loggingdir;
 
 	##############################################################
