--- sw/inc/unochart.hxx	2007-07-17 19:01:26.000000000 +0200
+++ sw/inc/unochart.hxx	2007-07-17 19:03:31.000000000 +0200
@@ -220,6 +220,13 @@ public:
     virtual ::com::sun::star::uno::Reference< ::com::sun::star::chart2::data::XDataSequence > SAL_CALL createDataSequenceByRangeRepresentation( const ::rtl::OUString& aRangeRepresentation ) throw (::com::sun::star::lang::IllegalArgumentException, ::com::sun::star::uno::RuntimeException);
     virtual ::com::sun::star::uno::Reference< ::com::sun::star::sheet::XRangeSelection > SAL_CALL getRangeSelection(  ) throw (::com::sun::star::uno::RuntimeException);
 
+    virtual ::sal_Bool SAL_CALL createDataSourcePossible(
+        const ::com::sun::star::uno::Sequence< ::com::sun::star::beans::PropertyValue >& aArguments )
+        throw (::com::sun::star::uno::RuntimeException) { return true; }
+    virtual ::sal_Bool SAL_CALL createDataSequenceByRangeRepresentationPossible(
+        const ::rtl::OUString& aRangeRepresentation )
+        throw (::com::sun::star::uno::RuntimeException) { return true; }
+
     // XRangeXMLConversion
     virtual ::rtl::OUString SAL_CALL convertRangeToXML( const ::rtl::OUString& aRangeRepresentation ) throw (::com::sun::star::lang::IllegalArgumentException, ::com::sun::star::uno::RuntimeException);
     virtual ::rtl::OUString SAL_CALL convertRangeFromXML( const ::rtl::OUString& aXMLRange ) throw (::com::sun::star::lang::IllegalArgumentException, ::com::sun::star::uno::RuntimeException);
