Index: svx/source/dialog/cfg.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/cfg.cxx,v
retrieving revision 1.19
diff -u -r1.19 cfg.cxx
--- svx/source/dialog/cfg.cxx	13 Apr 2005 08:25:54 -0000	1.19
+++ svx/source/dialog/cfg.cxx	19 Apr 2005 17:25:59 -0000
@@ -5347,19 +5347,19 @@
 
 	if ( rejectedCount != 0 )
 	{
-		OUString message( String( ResId( RID_SVXSTR_IMPORT_ICON_ERROR, DIALOG_MGR() ) ) );
+		String message( ResId( RID_SVXSTR_IMPORT_ICON_ERROR, DIALOG_MGR() ) );
 		if ( m_nExpectedSize != 16 )
 		{
 			message = replaceSixteen( message, m_nExpectedSize );
 		}
 
-		OUString newLine = OUString::createFromAscii("\n");
+		String newLine = OUString::createFromAscii("\n");
 		message += newLine;
 		message += newLine;
 
 		for ( sal_Int32 i = 0; i < rejectedCount; i++ )
 		{
-			message += rejected[i];
+			message += String( rejected[i] );
 			message += newLine;
 		}
 
