--- sysui/desktop/mandrakelinux/mandrakelinux-menus.spec.ark	2005-03-14 13:17:19.000000000 +0100
+++ sysui/desktop/mandrakelinux/mandrakelinux-menus.spec	2005-03-14 13:17:26.000000000 +0100
@@ -3,7 +3,7 @@
 Version: 2.0
 Release: 1
 Group: Office
-Copyright: LGPL / SISSL
+License: LGPL / SISSL
 AutoReqProv: no
 Requires: openofficeorg-core01, mandrake-release
 %define _unpackaged_files_terminate_build 0
