--- binfilter/bf_sw/source/core/inc/flowfrm.hxx.old	2005-09-08 11:28:10.000000000 +0200
+++ binfilter/bf_sw/source/core/inc/flowfrm.hxx	2005-12-05 13:53:51.000000000 +0100
@@ -34,6 +34,11 @@
  ************************************************************************/
 #ifndef _FLOWFRM_HXX
 #define _FLOWFRM_HXX
+
+#ifndef _FRMTOOL_HXX
+#include <frmtool.hxx>
+#endif
+
 namespace binfilter {
 
 //Der FlowFrm gibt die Funktionalitaet fuer alle Frms vor, die fliessen und
--- binfilter/bf_sw/source/core/inc/frmtool.hxx.old	2005-09-08 11:30:39.000000000 +0200
+++ binfilter/bf_sw/source/core/inc/frmtool.hxx	2005-12-05 14:43:59.000000000 +0100
@@ -93,6 +93,9 @@
 //Implementierung in feshview.cxx
 SwFlyFrm *GetFlyFromMarked( const SdrMarkList *pLst, ViewShell *pSh );
 
+SwFrm *SaveCntnt( SwLayoutFrm *pLay, SwFrm *pStart );
+void RestoreCntnt( SwFrm *pSav, SwLayoutFrm *pParent, SwFrm *pSibling );
+
 //Nicht gleich die math.lib anziehen.
 ULONG MA_FASTCALL SqRt( BigInt nX );
 
--- binfilter/bf_sw/source/core/inc/pagefrm.hxx.old	2005-09-08 11:35:07.000000000 +0200
+++ binfilter/bf_sw/source/core/inc/pagefrm.hxx	2005-12-05 18:19:30.000000000 +0100
@@ -70,6 +70,8 @@
 	CHG_CHGPAGE
 };
 
+void SetLastPage( SwPageFrm* );
+
 class SwPageFrm: public SwFtnBossFrm
 {
 	friend class SwFrm;
--- binfilter/bf_sw/source/core/inc/rootfrm.hxx.old	2005-09-08 11:36:05.000000000 +0200
+++ binfilter/bf_sw/source/core/inc/rootfrm.hxx	2005-12-05 14:03:36.000000000 +0100
@@ -69,6 +69,8 @@
 #define INV_LINENUM 32
 #define INV_DIRECTION 64
 
+void SetShell( ViewShell *pSh );
+
 class SwRootFrm: public SwLayoutFrm
 {
 	//Muss das Superfluous temporaer abschalten.
--- binfilter/inc/bf_svx/svdetc.hxx.old	2005-09-09 16:34:08.000000000 +0200
+++ binfilter/inc/bf_svx/svdetc.hxx	2005-12-05 11:12:42.000000000 +0100
@@ -139,6 +139,13 @@
 	friend SdrOutliner* SdrMakeOutliner( USHORT nOutlinerMode, SdrModel* pMod );
 };
 
+// Einen Outliner mit den engineglobalen
+// Defaulteinstellungen auf dem Heap erzeugen.
+// Ist pMod<>NULL, dann wird der MapMode des uebergebenen
+// Models verwendet. Die resultierende Default-Fonthoehe bleibt
+// jedoch dieselbe (die logische Fonthoehe wird umgerechnet).
+SdrOutliner* SdrMakeOutliner( USHORT nOutlinerMode, SdrModel* pMod );
+
 // Liefert eine Ersatzdarstellung fuer einen XFillStyle
 // Bei XFILL_NONE gibt's FALSE und rCol bleibt unveraendert.
 FASTBOOL GetDraftFillColor(const SfxItemSet& rSet, Color& rCol);
