Index: xmloff/source/draw/xexptran.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/draw/xexptran.cxx,v
retrieving revision 1.17
diff -u -r1.17 xexptran.cxx
--- xmloff/source/draw/xexptran.cxx	13 Apr 2005 09:29:54 -0000	1.17
+++ xmloff/source/draw/xexptran.cxx	19 Apr 2005 16:51:17 -0000
@@ -448,8 +448,8 @@
 const OUString& SdXMLImExTransform2D::GetExportString(const SvXMLUnitConverter& rConv)
 {
 	OUString aNewString;
-	OUString aClosingBrace(String(sal_Unicode(')')));
-	OUString aEmptySpace(String(sal_Unicode(' ')));
+	OUString aClosingBrace(sal_Unicode(')'));
+	OUString aEmptySpace(sal_Unicode(' '));
 	
 	for(sal_uInt32 a(0L); a < maList.Count(); a++)
 	{
@@ -861,8 +861,8 @@
 const OUString& SdXMLImExTransform3D::GetExportString(const SvXMLUnitConverter& rConv)
 {
 	OUString aNewString;
-	OUString aClosingBrace(String(sal_Unicode(')')));
-	OUString aEmptySpace(String(sal_Unicode(' ')));
+	OUString aClosingBrace(sal_Unicode(')'));
+	OUString aEmptySpace(sal_Unicode(' '));
 	
 	for(sal_uInt32 a(0L); a < maList.Count(); a++)
 	{
@@ -1267,7 +1267,7 @@
 const OUString& SdXMLImExViewBox::GetExportString(const SvXMLUnitConverter& rConv)
 {
 	OUString aNewString;
-	OUString aEmptySpace(String(sal_Unicode(' ')));
+	OUString aEmptySpace(sal_Unicode(' '));
 
 	Imp_PutNumberChar(aNewString, rConv, mnX);
 	aNewString += aEmptySpace;
