--- dbaccess/source/ui/inc/genericcontroller.hxx	2006-07-14 13:36:10.000000000 +0200
+++ dbaccess/source/ui/inc/genericcontroller.hxx	2006-07-14 13:37:04.000000000 +0200
@@ -143,7 +143,7 @@ namespace dbaui
     public:
                  optional ( ) : base_type( ) { }
         explicit optional ( T const& val ) : base_type( val ) { }
-                 optional ( optional const& rhs ) : base_type( rhs ) { }
+                 optional ( optional const& rhs ) : base_type( (base_type const&)rhs ) { }
 
     public:
         optional& operator= ( T const& rhs )
