--- vcl/source/window/dockingarea.cxx	16 Nov 2004 15:10:38 -0000	1.2
+++ vcl/source/window/dockingarea.cxx	6 Dec 2004 16:09:51 -0000
@@ -209,6 +209,7 @@
         if( !ImplGetSVData()->maNWFData.mbDockingAreaSeparateTB )
         {
             // draw a single toolbar background covering the whole docking area
-            Region aCtrlRegion( Rectangle( Point(), GetOutputSizePixel() ) );
+            Point tmp;
+            Region aCtrlRegion( Rectangle( tmp, GetOutputSizePixel() ) );
             
             DrawNativeControl( CTRL_TOOLBAR, IsHorizontal() ? PART_DRAW_BACKGROUND_HORZ : PART_DRAW_BACKGROUND_VERT, 


--- vcl/source/window/menu.cxx	17 Nov 2004 13:19:35 -0000	1.114
+++ vcl/source/window/menu.cxx	6 Dec 2004 16:10:01 -0000
@@ -2201,7 +2201,8 @@
             ImplControlValue aVal;
             Region aNativeBounds;
             Region aNativeContent;
-            Region aCtrlRegion( Rectangle( Point( 0, 0 ), Size( 100, 15 ) ) );
+            Point tmp( 0, 0 );
+            Region aCtrlRegion( Rectangle( tmp, Size( 100, 15 ) ) );
             if( pWindow->GetNativeControlRegion( ControlType(CTRL_MENUBAR),
                                                  ControlPart(PART_ENTIRE_CONTROL),
                                                  aCtrlRegion,
@@ -4837,7 +4838,8 @@
                         aMenubarValue.maTopDockingAreaHeight = ImplGetTopDockingAreaHeight( this );
                         aControlValue.setOptionalVal( (void *)(&aMenubarValue) );
 
-                        Region aBgRegion( Rectangle( Point(0,0), GetOutputSizePixel() ) );
+                        Point tmp(0,0);
+                        Region aBgRegion( Rectangle( tmp, GetOutputSizePixel() ) );
                         DrawNativeControl( CTRL_MENUBAR, PART_ENTIRE_CONTROL,
                                            aBgRegion,
                                            CTRL_STATE_ENABLED,
