--- animations/inc/animations/animationnodehelper.hxx	2005-07-19 00:33:54.000000000 +0200
+++ animations/inc/animations/animationnodehelper.hxx	2005-07-19 00:34:30.000000000 +0200
@@ -100,7 +100,7 @@
         @return true, if the functor was successfully applied to
         all children, false otherwise.
     */
-    template< typename Functor > bool for_each_childNode( const ::com::sun::star::uno::Reference< ::com::sun::star::animations::XAnimationNode >&	xNode,
+    template< typename Functor > inline bool for_each_childNode( const ::com::sun::star::uno::Reference< ::com::sun::star::animations::XAnimationNode >&	xNode,
                                                           Functor&																					rFunctor )
     {
         try
@@ -133,7 +133,7 @@
 
 	/** pushes the given node to the given vector and recursivly calls itself for each child node.
 	*/
-	void create_deep_vector( const ::com::sun::star::uno::Reference< ::com::sun::star::animations::XAnimationNode >& xNode,
+	inline void create_deep_vector( const ::com::sun::star::uno::Reference< ::com::sun::star::animations::XAnimationNode >& xNode,
 								std::vector< ::com::sun::star::uno::Reference< ::com::sun::star::animations::XAnimationNode > >& rVector )
 	{
 		rVector.push_back( xNode );
