--- vcl/source/control/button.cxx.~1.37.~	2005-03-18 18:51:14.000000000 +0100
+++ vcl/source/control/button.cxx	2005-03-31 21:59:19.578104528 +0200
@@ -3089,7 +3089,8 @@
     if( pThis->IsNativeControlSupported( CTRL_RADIOBUTTON, PART_ENTIRE_CONTROL ) )
     {
         ImplControlValue aControlValue;
-        Region			 aCtrlRegion( Rectangle( Point( 0, 0 ), GetSizePixel() ) );
+        Point            aPoint( 0, 0 );
+        Region			 aCtrlRegion( Rectangle( aPoint, GetSizePixel() ) );
         ControlState	 nState = CTRL_STATE_DEFAULT|CTRL_STATE_ENABLED;
         Region aBoundingRgn, aContentRgn;
         
@@ -3907,7 +3908,8 @@
     if( pThis->IsNativeControlSupported( CTRL_CHECKBOX, PART_ENTIRE_CONTROL ) )
     {
         ImplControlValue aControlValue;
-        Region			 aCtrlRegion( Rectangle( Point( 0, 0 ), GetSizePixel() ) );
+        Point            aPoint( 0, 0 );
+        Region			 aCtrlRegion( Rectangle( aPoint, GetSizePixel() ) );
         ControlState	 nState = CTRL_STATE_DEFAULT|CTRL_STATE_ENABLED;
         Region aBoundingRgn, aContentRgn;
         
