--- ucbhelper/source/provider/backgrounddownload.cxx.old	2005-09-09 18:35:46.000000000 +0200
+++ ucbhelper/source/provider/backgrounddownload.cxx	2005-11-04 10:48:12.000000000 +0100
@@ -340,7 +340,7 @@
         
         InputStream* GetSink(size_t nMaxMem);
 
-        Reference<XCommandEnvironment> DownloadThread_Impl::GetEnv();
+        Reference<XCommandEnvironment> GetEnv();
         
         
         enum THREADSTATE {
