--- sw/source/core/swg/SwXMLTextBlocks1.cxx.old	2005-09-09 06:43:40.000000000 +0200
+++ sw/source/core/swg/SwXMLTextBlocks1.cxx	2005-11-25 15:31:53.000000000 +0100
@@ -136,11 +136,8 @@
 	}
 	else
 	{
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-        String aStreamName = aFolderName + (OUString) String::CreateFromAscii(".xml");
-#else
-        String aStreamName = aFolderName + String::CreateFromAscii(".xml");
-#endif
+        String aStreamName = aFolderName;
+        aStreamName += String::CreateFromAscii(".xml");
 		try
 		{
             xRoot = xBlkRoot->openStorageElement( aFolderName, embed::ElementModes::READ );
@@ -357,11 +354,8 @@
 	sal_Bool bTextOnly = sal_True;
 	String aFolderName;
 	GeneratePackageName ( rShort, aFolderName );
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-	String aStreamName = aFolderName + (OUString) String::CreateFromAscii(".xml");
-#else
-	String aStreamName = aFolderName + String::CreateFromAscii(".xml");
-#endif
+	String aStreamName = aFolderName;
+	aStreamName += String::CreateFromAscii(".xml");
 	rText.Erase();
 
     try
@@ -446,11 +440,8 @@
 	}
 	*/
 	String aFolderName( rPackageName );
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-    String aStreamName = aFolderName + (OUString) String::CreateFromAscii(".xml");
-#else
-    String aStreamName = aFolderName + String::CreateFromAscii(".xml");
-#endif
+	String aStreamName = aFolderName;
+	aStreamName += String::CreateFromAscii(".xml");
 
 	Reference< lang::XMultiServiceFactory > xServiceFactory =
 		comphelper::getProcessServiceFactory();
