--- i18npool/inc/localedata.hxx.old	2005-11-24 11:47:23.000000000 +0100
+++ i18npool/inc/localedata.hxx	2005-11-28 15:01:22.000000000 +0100
@@ -90,6 +90,10 @@
 
 namespace com { namespace sun { namespace star { namespace i18n {
 
+inline sal_Bool operator ==(const com::sun::star::lang::Locale& l1, const com::sun::star::lang::Locale& l2) {
+    return l1.Language == l2.Language && l1.Country == l2.Country && l1.Variant == l2.Variant;
+};
+
 class I18NPOOL_DLLPUBLIC LocaleData : public cppu::WeakImplHelper2
 <
 	XLocaleData,
@@ -131,13 +135,7 @@
 	virtual com::sun::star::uno::Sequence< rtl::OUString > SAL_CALL getSupportedServiceNames() throw( com::sun::star::uno::RuntimeException );
 
 private :
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-	inline friend sal_Bool operator ==(const com::sun::star::lang::Locale& l1, const com::sun::star::lang::Locale& l2);
-#else
-	inline friend sal_Bool operator ==(const com::sun::star::lang::Locale& l1, const com::sun::star::lang::Locale& l2) {
-	    return l1.Language == l2.Language && l1.Country == l2.Country && l1.Variant == l2.Variant;
-	};
-#endif
+	friend sal_Bool operator ==(const com::sun::star::lang::Locale& l1, const com::sun::star::lang::Locale& l2);
 
 	struct lookupTableItem {
 	    lookupTableItem(const sal_Char *name, osl::Module* m) : dllName(name), module(m) {}
@@ -163,11 +161,6 @@
 		const com::sun::star::uno::Sequence< com::sun::star::i18n::Calendar >& calendarsSeq,
 		sal_Int16 len, sal_Int16 item) throw( com::sun::star::uno::RuntimeException );
 };
-#if defined(_MSC_VER) && (_MSC_VER >= 1310 )
-	inline sal_Bool operator ==(const com::sun::star::lang::Locale& l1, const com::sun::star::lang::Locale& l2) {
-	    return l1.Language == l2.Language && l1.Country == l2.Country && l1.Variant == l2.Variant;
-	};
-#endif
 
 } } } }
 
