--- solenv/inc/unxlngi6.mk.old	2005-04-14 19:21:35.000000000 +0200
+++ solenv/inc/unxlngi6.mk	2005-04-19 14:27:02.000000000 +0200
@@ -89,6 +89,11 @@
 .ENDIF
 .ENDIF
 
+# define default arch flags when needed
+.IF "$(ARCH_FLAGS)"==""
+ARCH_FLAGS=-mtune=pentiumpro
+.ENDIF
+
 # name of C++ Compiler
 CXX*=g++
 # name of C Compiler
@@ -112,14 +117,14 @@
 .ENDIF
 
 # flags for the C++ Compiler
-CFLAGSCC= -pipe -mtune=pentiumpro
+CFLAGSCC= -pipe $(ARCH_FLAGS)
 # Flags for enabling exception handling
 CFLAGSEXCEPTIONS=-fexceptions -fno-enforce-eh-specs
 # Flags for disabling exception handling
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # -fpermissive should be removed as soon as possible
-CFLAGSCXX= -pipe -mtune=pentiumpro
+CFLAGSCXX= -pipe $(ARCH_FLAGS)
 CFLAGSCXX+= -Wno-ctor-dtor-privacy
 PICSWITCH:=-fpic
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
