--- framework/source/helper/uiconfigelementwrapperbase.cxx.ark	2006-11-06 21:25:32.000000000 +0100
+++ framework/source/helper/uiconfigelementwrapperbase.cxx	2006-11-06 21:41:50.000000000 +0100
@@ -256,7 +256,7 @@
 	{
         case UIELEMENT_PROPHANDLE_CONFIGLISTENER:
             bReturn = PropHelper::willPropertyBeChanged(
-                        com::sun::star::uno::makeAny(m_bConfigListener),
+                        com::sun::star::uno::makeAny((sal_Bool)m_bConfigListener),
                         aValue,
                         aOldValue,
                         aConvertedValue);
@@ -283,7 +283,7 @@
 
         case UIELEMENT_PROPHANDLE_PERSISTENT:
             bReturn = PropHelper::willPropertyBeChanged(
-                        com::sun::star::uno::makeAny(m_bPersistent),
+                        com::sun::star::uno::makeAny((sal_Bool)m_bPersistent),
                         aValue,
                         aOldValue,
                         aConvertedValue);
@@ -315,7 +315,7 @@
 
         case UIELEMENT_PROPHANDLE_NOCLOSE:
             bReturn = PropHelper::willPropertyBeChanged(
-                        com::sun::star::uno::makeAny(m_bNoClose),
+                        com::sun::star::uno::makeAny((sal_Bool)m_bNoClose),
                         aValue,
                         aOldValue,
                         aConvertedValue);
@@ -421,7 +421,7 @@
     switch( nHandle )
 	{
         case UIELEMENT_PROPHANDLE_CONFIGLISTENER: 
-            aValue <<= m_bConfigListener;
+            aValue <<= (sal_Bool)m_bConfigListener;
             break;
         case UIELEMENT_PROPHANDLE_CONFIGSOURCE: 
             aValue <<= m_xConfigSource;
@@ -433,7 +433,7 @@
             break;
         }
         case UIELEMENT_PROPHANDLE_PERSISTENT: 
-            aValue <<= m_bPersistent;
+            aValue <<= (sal_Bool)m_bPersistent;
             break;
         case UIELEMENT_PROPHANDLE_RESOURCEURL: 
             aValue <<= m_aResourceURL;
@@ -445,7 +445,7 @@
             aValue <<= m_xMenuBar;
             break;
         case UIELEMENT_PROPHANDLE_NOCLOSE:
-            aValue <<= m_bNoClose;
+            aValue <<= (sal_Bool)m_bNoClose;
             break;
 	}
 }
--- framework/source/layoutmanager/layoutmanager.cxx.ark	2006-11-06 21:55:27.000000000 +0100
+++ framework/source/layoutmanager/layoutmanager.cxx	2006-11-06 22:01:19.000000000 +0100
@@ -7102,14 +7102,14 @@
 	{
         case LAYOUTMANAGER_PROPHANDLE_MENUBARCLOSER :
             bReturn = PropHelper::willPropertyBeChanged(
-                        com::sun::star::uno::makeAny(m_bMenuBarCloser),
+                        com::sun::star::uno::makeAny((sal_Bool)m_bMenuBarCloser),
                         aValue,
                         aOldValue,
                         aConvertedValue);
                 break;
         case LAYOUTMANAGER_PROPHANDLE_AUTOMATICTOOLBARS:
             bReturn = PropHelper::willPropertyBeChanged(
-                        com::sun::star::uno::makeAny(m_bAutomaticToolbars),
+                        com::sun::star::uno::makeAny((sal_Bool)m_bAutomaticToolbars),
                         aValue,
                         aOldValue,
                         aConvertedValue);
@@ -7162,10 +7162,10 @@
     switch( nHandle )
 	{
         case LAYOUTMANAGER_PROPHANDLE_MENUBARCLOSER:
-            aValue <<= m_bMenuBarCloser;
+            aValue <<= (sal_Bool)m_bMenuBarCloser;
             break;
         case LAYOUTMANAGER_PROPHANDLE_AUTOMATICTOOLBARS:
-            aValue <<= m_bAutomaticToolbars;
+            aValue <<= (sal_Bool)m_bAutomaticToolbars;
             break;
         case LAYOUTMANAGER_PROPHANDLE_REFRESHVISIBILITY:
             aValue <<= sal_False;
--- framework/source/uiconfiguration/windowstateconfiguration.cxx.ark	2006-11-06 22:05:23.000000000 +0100
+++ framework/source/uiconfiguration/windowstateconfiguration.cxx	2006-11-06 22:07:44.000000000 +0100
@@ -675,21 +675,21 @@
             switch ( i )
             {
                 case PROPERTY_LOCKED:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bLocked ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bLocked ); break;
                 case PROPERTY_DOCKED:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bDocked ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bDocked ); break;
                 case PROPERTY_VISIBLE:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bVisible ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bVisible ); break;
                 case PROPERTY_CONTEXT:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bContext ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bContext ); break;
                 case PROPERTY_HIDEFROMMENU:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bHideFromMenu ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bHideFromMenu ); break;
                 case PROPERTY_NOCLOSE:   
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bNoClose ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bNoClose ); break;
                 case PROPERTY_SOFTCLOSE:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bSoftClose ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bSoftClose ); break;
                 case PROPERTY_CONTEXTACTIVE:
-                    aPropSeq[nIndex].Value = makeAny( rWinStateInfo.bContextActive ); break;
+                    aPropSeq[nIndex].Value = makeAny( (sal_Bool)rWinStateInfo.bContextActive ); break;
                 case PROPERTY_DOCKINGAREA:
                     aPropSeq[nIndex].Value = makeAny( rWinStateInfo.aDockingArea ); break;
                 case PROPERTY_POS:
--- framework/source/uielement/uicommanddescription.cxx.ark	2006-11-06 22:10:27.000000000 +0100
+++ framework/source/uielement/uicommanddescription.cxx	2006-11-06 22:10:34.000000000 +0100
@@ -432,7 +432,7 @@
         aPropSeq[1].Name  = m_aPropName;
         aPropSeq[1].Value = makeAny( pIter->second.aCommandName );
         aPropSeq[2].Name  = m_aPropPopup;
-        aPropSeq[2].Value = makeAny( pIter->second.bPopup );
+        aPropSeq[2].Value = makeAny( (sal_Bool)pIter->second.bPopup );
         return makeAny( aPropSeq );
     }
 
