--- basctl/source/basicide/moduldl2.cxx	2005-01-19 17:33:07.757914991 +0100
+++ basctl/source/basicide/moduldl2.cxx	2005-01-19 17:33:04.473487969 +0100
@@ -821,7 +821,7 @@ void LibPage::InsertLib()
 	xFP->setTitle( String( IDEResId( RID_STR_APPENDLIBS ) ) );
 
     // filter
-    ::rtl::OUString aTitle( ::rtl::OUString( IDEResId( RID_STR_BASIC ) ) );
+    ::rtl::OUString aTitle = ::rtl::OUString( IDEResId( RID_STR_BASIC ) );
     ::rtl::OUString aFilter;
     aFilter =  ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "*.sdw;*.sxw;*.odt" ) );       // text
     aFilter += ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( ";*.vor;*.stw;*.ott" ) );      // text template
