Index: sfx2/source/doc/guisaveas.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/doc/guisaveas.cxx,v
retrieving revision 1.15
diff -u -r1.15 guisaveas.cxx
--- sfx2/source/doc/guisaveas.cxx	13 Apr 2005 12:40:30 -0000	1.15
+++ sfx2/source/doc/guisaveas.cxx	19 Apr 2005 17:03:52 -0000
@@ -793,10 +793,10 @@
 
 		if ( xControlAccess.is() )
 		{
-			::rtl::OUString aCtrlText( String( SfxResId( STR_EXPORTBUTTON ) ) );
+			String aCtrlText( SfxResId( STR_EXPORTBUTTON ) );
 			xControlAccess->setLabel( ui::dialogs::CommonFilePickerElementIds::PUSHBUTTON_OK, aCtrlText );
 
-			aCtrlText = ::rtl::OUString( String( SfxResId( STR_LABEL_FILEFORMAT ) ) );
+			aCtrlText = String( SfxResId( STR_LABEL_FILEFORMAT ) );
 			xControlAccess->setLabel( ui::dialogs::CommonFilePickerElementIds::LISTBOX_FILTER_LABEL, aCtrlText );
 		}
 
