Index: sysui/desktop/freedesktop/makefile.mk
===================================================================
RCS file: /cvs/gsl/sysui/desktop/freedesktop/makefile.mk,v
retrieving revision 1.7
retrieving revision 1.7.4.1
diff -u -p -u -p -r1.7 -r1.7.4.1
--- sysui/desktop/freedesktop/makefile.mk	13 Jan 2006 14:57:10 -0000	1.7
+++ sysui/desktop/freedesktop/makefile.mk	24 Jan 2006 15:34:20 -0000	1.7.4.1
@@ -68,6 +68,7 @@ ALLTAR : $(PACKAGE_RPM)
 $(PACKAGE_RPM) : $(MISC)$/redhat.rpmflag
 
 $(PACKAGE_RPM) : $(MISC)/$(TARGET)-menus.spec
+	@-$(MKDIRHIER) $(@:d)
 	-@$(RM) $(PKGDIR)$/openoffice.org-$(TARGET)-*.noarch.rpm $(BIN)/noarch/openoffice.org-$(TARGET)-*.noarch.rpm
 	@$(MKDIRHIER) $(MISC)$/$(TARGET)
 	@$(MKDIRHIER) $(MISC)$/$(TARGET)$/BUILD
Index: sysui/desktop/mandriva/makefile.mk
===================================================================
RCS file: /cvs/gsl/sysui/desktop/mandriva/makefile.mk,v
retrieving revision 1.7
retrieving revision 1.7.4.1
diff -u -p -u -p -r1.7 -r1.7.4.1
--- sysui/desktop/mandriva/makefile.mk	13 Jan 2006 14:57:24 -0000	1.7
+++ sysui/desktop/mandriva/makefile.mk	24 Jan 2006 15:34:20 -0000	1.7.4.1
@@ -221,6 +221,7 @@ $(MISC)/$(TARGET)/etc/$(UNIXFILENAME) :
 # --- packaging ---------------------------------------------------
 	
 $(RPMFILE) : $(RPMDEPN) $(MISC)/$(TARGET)-menus.spec
+	@-$(MKDIRHIER) $(@:d)
 	-$(RM) $(@:d)$/$(PKGNAME)-*.noarch.rpm $(BIN)$/noarch$/$(PKGNAME)-*.noarch.rpm
 	@$(RPM) $(RPMMACROS) -bb $(MISC)/$(TARGET)-menus.spec \
 		--buildroot $(LAUNCHERDIR) \
Index: sysui/desktop/redhat/makefile.mk
===================================================================
RCS file: /cvs/gsl/sysui/desktop/redhat/makefile.mk,v
retrieving revision 1.18
retrieving revision 1.18.4.1
diff -u -p -u -p -r1.18 -r1.18.4.1
--- sysui/desktop/redhat/makefile.mk	13 Jan 2006 14:57:38 -0000	1.18
+++ sysui/desktop/redhat/makefile.mk	24 Jan 2006 15:34:21 -0000	1.18.4.1
@@ -227,6 +227,7 @@ $(MISC)/$(TARGET)/etc/$(UNIXFILENAME) :
 # --- packaging ---------------------------------------------------
 	
 $(RPMFILE) : $(RPMDEPN) $(MISC)/$(TARGET)-menus.spec
+	@-$(MKDIRHIER) $(@:d)
 	-$(RM) $(@:d)$/$(PKGNAME)-* $(BIN)$/noarch$/$(PKGNAME)-*
 	@$(RPM) -bb $(MISC)/$(TARGET)-menus.spec $(RPMMACROS) \
 		--buildroot $(LAUNCHERDIR) \
Index: sysui/desktop/suse/makefile.mk
===================================================================
RCS file: /cvs/gsl/sysui/desktop/suse/makefile.mk,v
retrieving revision 1.19
retrieving revision 1.19.4.1
diff -u -p -u -p -r1.19 -r1.19.4.1
--- sysui/desktop/suse/makefile.mk	13 Jan 2006 14:58:05 -0000	1.19
+++ sysui/desktop/suse/makefile.mk	24 Jan 2006 15:34:22 -0000	1.19.4.1
@@ -226,6 +226,7 @@ $(MISC)/$(TARGET)/etc/$(UNIXFILENAME) :
 # --- packaging ---------------------------------------------------
 	
 $(RPMFILE) : $(RPMDEPN) $(MISC)/$(TARGET)-menus.spec
+	@-$(MKDIRHIER) $(@:d)
 	-$(RM) $(@:d)$/$(PKGNAME)-*.noarch.rpm $(BIN)$/noarch$/$(PKGNAME)-*.noarch.rpm 
 	@$(RPM) -bb $(MISC)/$(TARGET)-menus.spec $(RPMMACROS) \
 		--buildroot $(LAUNCHERDIR) \
Index: sysui/desktop/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/sysui/desktop/util/makefile.mk,v
retrieving revision 1.11
retrieving revision 1.11.4.1
diff -u -p -u -p -r1.11 -r1.11.4.1
--- sysui/desktop/util/makefile.mk	13 Jan 2006 14:58:26 -0000	1.11
+++ sysui/desktop/util/makefile.mk	24 Jan 2006 15:34:23 -0000	1.11.4.1
@@ -81,12 +81,13 @@ PKGCONFIG_LIBS!:=-Wl,--export-dynamic $(
 
 .ENDIF          # "$(ENABLE_GNOMEVFS)"!=""
 
-.IF "$(OS)"!="SOLARIS"
+.IF "$(RPM)" != ""
 TARFILE=$(BIN)$/rpm$/openoffice.org-desktop-integration.tar.gz
+.ENDIF  # "$(RPM)" != ""
+
 .IF "$(DPKG)" != ""
 TARFILE2=$(BIN)$/deb$/openoffice.org-desktop-integration.tar.gz
 .ENDIF  # "$(DPKG)" != ""
-.ENDIF  # "$(OS)"!="SOLARIS"
 
 # --- Files --------------------------------------------------------
 
@@ -113,12 +114,13 @@ $(SCRIPTS) : $$(@:f)
     +@cat $(@:f) | tr -d "\015" > $@
     +@chmod 555 $@
 
-.IF "$(OS)"!="SOLARIS"
+.IF "$(TARFILE)" != ""
 
 $(TARFILE) : $(PKGDIR)$/{$(shell ls $(PKGDIR))}
 	$(MKDIRHIER) $(@:d)
 	tar -C $(PKGDIR:d:d) -cf - $(PKGDIR:f)/ | gzip > $@
-.ENDIF
+
+.ENDIF # "$(TARFILE)" != ""
 
 .IF "$(DPKG)" != ""
 
Index: solenv/inc/packtools.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/packtools.mk,v
retrieving revision 1.6
retrieving revision 1.6.8.1
diff -u -p -u -p -r1.6 -r1.6.8.1
--- solenv/inc/packtools.mk	13 Jan 2006 15:02:07 -0000	1.6
+++ solenv/inc/packtools.mk	25 Jan 2006 10:57:27 -0000	1.6.8.1
@@ -34,6 +34,13 @@
 #*************************************************************************
 
 .IF "$(GUI)"!="WNT"
+
+.IF "$(PKGFORMAT)"!="$(PKGFORMAT:s/rpm//)"
 RPM*:=$(shell +-ls -l `which rpm` >& /dev/null && set i=`which rpm` && echo rpm`test -x $${{i}}build && echo build`)
+.ENDIF			# "$(PKGFORMAT)"!="$(PKGFORMAT:s/rpm//)"
+
+.IF "$(PKGFORMAT)"!="$(PKGFORMAT:s/deb//)"
 DPKG*:=$(shell +-ls -l `which dpkg` >& /dev/null && echo dpkg)
+.ENDIF			# "$(PKGFORMAT)"!="$(PKGFORMAT:s/deb//)"
+
 .ENDIF
