Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.99
diff -u -r1.99 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	8 Sep 2004 16:01:25 -0000	1.99
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	11 Oct 2004 20:05:22 -0000
@@ -971,7 +971,7 @@
 					<desc>Specifies if the letter at the beginning of a sentence should be capitalized automatically.</desc>
 					<label>Capitalize first letter of every sentence</label>
 				</info>
-				<value>true</value>
+				<value>false</value>
 			</prop>
 			<prop oor:name="ChangeUnderlineWeight" oor:type="xs:boolean">
 				<!-- OldPath: AutoCorrect/Options/All -->
