--- salnativewidgets-gtk.cxx.bak	2005-03-15 14:52:50.550577928 +0530
+++ vcl/unx/gtk/gdi/salnativewidgets-gtk.cxx	2005-03-15 14:46:59.050014136 +0530
@@ -2848,6 +2848,7 @@
     
     // background colors
     Color aBackColor = getColor( pStyle->bg[GTK_STATE_NORMAL] );
+    Color aBackActiveColor = getColor( pStyle->bg[GTK_STATE_ACTIVE] );
     Color aBackFieldColor = getColor( pStyle->base[ GTK_STATE_NORMAL ] );
     aStyleSet.Set3DColors( aBackColor );
     aStyleSet.SetFaceColor( aBackColor );
@@ -2855,7 +2856,7 @@
     aStyleSet.SetWorkspaceColor( aBackColor );
     aStyleSet.SetFieldColor( aBackFieldColor );
     aStyleSet.SetWindowColor( aBackFieldColor );
-    aStyleSet.SetHelpColor( aBackColor );
+    aStyleSet.SetHelpColor( aBackActiveColor );
    // ancient wisdom tells us a mystic algorithm how to set checked color
     if( aBackColor == COL_LIGHTGRAY )
         aStyleSet.SetCheckedColor( Color( 0xCC, 0xCC, 0xCC ) );
