Index: tools/inc/ref.hxx
===================================================================
RCS file: /cvs/util/tools/inc/Attic/ref.hxx,v
retrieving revision 1.5
diff -u -p -r1.5 ref.hxx
--- tools/inc/ref.hxx	19 Jun 2006 13:29:01 -0000	1.5
+++ tools/inc/ref.hxx	19 Apr 2007 10:30:39 -0000
@@ -126,33 +126,6 @@ PRV_SV_IMPL_REF_COUNTERS( ClassName, Loc
 						  OwnerLock( TRUE ), OwnerLock( FALSE ), 	\
 						  EMPTYARG, pObj )
 
-#define SV_DECL_COMPAT_REF( ClassName )			\
-class ClassName;								\
-class ClassName##Ref							\
-{												\
-protected:										\
-	SvRefBase* pRefbase;						\
-	PRV_SV_DECL_REF( ClassName ) 			    \
-public:											\
-	static void Destroy( void* );		        \
-};
-
-#define SV_COMPAT_REF_INIT												\
-pObj ? ( pRefbase =  													\
- (SfxPointerServer::GetServer()->HasPointer( pObj ) ? NULL :			\
- ((*(SvRefBase **)SfxPointerServer::GetServer()->CreatePointer(pObj))=	\
-   new SvCompatRefBase( &Destroy, pObj ) ),								\
-   (SvRefBase *)SfxPointerServer::GetServer()->GetPointer(pObj))) : 0;
-
-#define SV_IMPL_COMPAT_REF( ClassName )										\
-PRV_SV_IMPL_REF_COUNTERS( ClassName, Ref, AddRef(), AddNextRef(), 			\
-						  ReleaseReference(), SV_COMPAT_REF_INIT, pRefbase )\
-inline void ClassName##Ref::Destroy( void* pObj)							\
-{																			\
-	SfxPointerServer::GetServer()->ReleasePointer( pObj );					\
-	delete ( ClassName* ) pObj;												\
-}
-
 #define SV_DECL_IMPL_REF(ClassName)				\
     SV_DECL_REF(ClassName)						\
     SV_IMPL_REF(ClassName)
@@ -161,10 +134,6 @@ inline void ClassName##Ref::Destroy( voi
     SV_DECL_LOCK(ClassName)						\
     SV_IMPL_LOCK(ClassName)
 
-#define SV_DECL_IMPL_COMPAT_REF( ClassName )	\
-    SV_DECL_COMPAT_REF(ClassName)				\
-    SV_IMPL_COMPAT_REF(ClassName)
-
 
 /************************** S v R e f L i s t ****************************/
 #define PRV_SV_DECL_REF_LIST(CN,EN,vis) \
@@ -414,17 +383,6 @@ public:
     UINT32          GetRefCount() const { return nRefCount; }
 };
 
-class SvCompatRefBase : public SvRefBase
-{
-	void (*pFunc)( void* );
-	void *pObj;
-protected:
-	virtual         ~SvCompatRefBase();
-public:
-	SvCompatRefBase( void (*pFuncP)( void*), void *pObjP ) :
-		pFunc( pFuncP ), pObj( pObjP ) {}
-};
-
 #if _SOLAR__PRIVATE
 #ifndef EMPTYARG
 #define EMPTYARG
Index: tools/source/ref/ref.cxx
===================================================================
RCS file: /cvs/util/tools/source/ref/ref.cxx,v
retrieving revision 1.4
diff -u -p -r1.4 ref.cxx
--- tools/source/ref/ref.cxx	17 Sep 2006 01:01:04 -0000	1.4
+++ tools/source/ref/ref.cxx	19 Apr 2007 10:30:39 -0000
@@ -57,8 +57,4 @@ void SvRefBase::QueryDelete()
 	delete this;
 }
 
-SvCompatRefBase::~SvCompatRefBase()
-{
-	(*pFunc)( pObj );
-}
 
