Index: instsetoo_native/prj/build.lst
===================================================================
RCS file: /cvs/installation/instsetoo_native/prj/build.lst,v
retrieving revision 1.12
diff -u -p -u -r1.12 build.lst
--- instsetoo_native/prj/build.lst	2 Nov 2004 09:27:38 -0000	1.12
+++ instsetoo_native/prj/build.lst	15 Nov 2004 14:54:53 -0000
@@ -1,4 +1,22 @@
-oon	instsetoo_native	:	helpcontent2 UnoControls basctl configmgr cpputools dbaccess desktop eventattacher extensions extras fileaccess forms io package padmin readlicense_oo remotebridges scaddins scp2 setup_native SO:epm shell sc sch chart2 sd starmath sw ucb wizards officecfg xmlhelp MathMLDTD dictionaries bitstream_vera_fonts filter psprint_config fpicker odk embedserv pyuno crashrep scripting embeddedobj hwpfilter unoxml slideshow binfilter msfontextract xmlsecurity bean NULL
+oon	instsetoo_native  :  \	
+	helpcontent2 UnoControls basctl \
+	configmgr cpputools dbaccess \
+	desktop eventattacher extensions \
+	extras fileaccess forms io \
+	package padmin readlicense_oo \
+	remotebridges scaddins scp2 \
+	setup_native SO:epm shell sc \
+	sch chart2 sd starmath sw ucb \
+	wizards officecfg xmlhelp MathMLDTD \
+	filter \
+	psprint_config fpicker odk embedserv \
+	pyuno crashrep scripting embeddedobj \
+	hwpfilter unoxml slideshow \
+	msfontextract xmlsecurity bean \
+	$(WITH_FONTS):bitstream_vera_fonts \
+	$(WITH_BINFILTER):binfilter \
+       $(WITH_MYSPELL_DICTS):dictionaries lingucomponent \
+	NULL
 oon	instsetoo_native						usr1	-	all	oon_mkout NULL
 oon	instsetoo_native\packimages				nmake	-	all	oon_pack NULL
 oon	instsetoo_native\packconfig				nmake	-	all	oon_config NULL
