Index: ucb/prj/build.lst
===================================================================
RCS file: /cvs/ucb/ucb/prj/build.lst,v
retrieving revision 1.17
diff -u -p -u -r1.17 build.lst
--- ucb/prj/build.lst	13 Oct 2004 15:04:57 -0000	1.17
+++ ucb/prj/build.lst	15 Nov 2004 14:56:38 -0000
@@ -1,4 +1,9 @@
-uc ucb : unotools tools cppuhelper curl neon offapi sal salhelper ucbhelper udkapi vos SO:gnomelibs NULL
+uc ucb : unotools tools cppuhelper \
+	 !$(SYSTEM_CURL):curl \
+	 !$(SYSTEM_NEON):neon \
+	 offapi sal salhelper \
+	 ucbhelper udkapi vos \
+	 SO:gnomelibs NULL
 uc	ucb							usr1	-	all	uc_mkout NULL
 uc	ucb\unotypes			    nmake	-	all	uc_unotypes NULL
 uc	ucb\source\regexp		    nmake	-	all	uc_regexp uc_unotypes NULL
