Index: vcl/prj/build.lst
===================================================================
RCS file: /cvs/gsl/vcl/prj/build.lst,v
retrieving revision 1.34
diff -u -p -u -r1.34 build.lst
--- vcl/prj/build.lst	8 Oct 2004 15:19:49 -0000	1.34
+++ vcl/prj/build.lst	15 Nov 2004 14:57:36 -0000
@@ -1,4 +1,7 @@
-vc	vcl	:	boost nas freetype psprint rsc sot ucbhelper unotools rvpapi icu i18npool unoil ridljar x11_extensions SO:gtk offuh basegfx NULL
+vc	vcl	:	boost nas psprint rsc sot ucbhelper \
+			unotools rvpapi icu i18npool unoil \
+			ridljar x11_extensions SO:gtk offuh basegfx \
+			!$(SYSTEM_FREETYPE):freetype NULL
 vc	vcl										usr1	-	all	vc_mkout NULL
 vc	vcl\source\unotypes						nmake	-	all	vc_unot NULL
 vc	vcl\source\glyphs						nmake	-	all	vc_glyphs vc_unot NULL
