--- config_office/configure.in.old	2007-10-29 14:01:52.000000000 +0100
+++ config_office/configure.in	2007-10-29 14:04:46.000000000 +0100
@@ -1830,7 +1830,7 @@
     AC_CHECK_HEADER(security/pam_appl.h, [],
                     [AC_MSG_ERROR([pam_appl.h could not be found. libpam-dev or pam-devel missing?])], [])
     AC_MSG_CHECKING([whether to link to libpam])
-    if test -n "$enable_pam_link"; then
+    if test -n "$enble_pam_link" -a "$enable_pam_link" != "no"; then
       AC_MSG_RESULT([yes])
       PAM_LINK=YES
       AC_CHECK_LIB(pam, pam_start, [],
@@ -1853,7 +1853,7 @@
 
 if test "$_os" = "Linux"; then
     AC_MSG_CHECKING([whether to link to libcrypt])
-    if test -n "$enable_crypt_link"; then
+    if test -n "$enable_crypt_link" -a "$enable_crypt_link" != "no"; then
         AC_MSG_RESULT([yes])
         CRYPT_LINK=YES
         AC_CHECK_LIB(crypt, crypt, [],
@@ -4208,7 +4208,7 @@
 dnl Check for system Xrender
 dnl ===================================================================
 AC_MSG_CHECKING([whether to link to Xrender])
-if test -n "$enable_xrender_link"; then
+if test -n "$enable_xrender_link" -a "$enable_xrender_link" != "no"; then
    AC_MSG_RESULT([yes])
    XRENDER_LINK=YES
    with_system_xrender_headers=yes
