--- config_office/configure.in	2007-03-22 16:54:38.000000000 +0100
+++ config_office/configure.in	2007-03-29 15:24:14.000000000 +0200
@@ -1548,6 +1616,10 @@ if test "$_os" = "WINNT"; then
 		if test -e "$with_mspdb_path/mspdb71.dll"; then
 			MSPDB_PATH="$with_mspdb_path"
 		fi
+		dnl  .NET 2005 Compiler
+		if test -e "$with_mspdb_path/mspdb80.dll"; then
+			MSPDB_PATH="$with_mspdb_path"
+		fi
 		dnl .NET case
 		if test -z "$MSPDB_PATH" -a -e "$with_cl_home/../Common7/IDE/mspdb71.dll"; then
 			MSPDB_PATH="$with_cl_home/../Common7/IDE"
