--- config_office/configure.in	23 Mar 2005 08:46:00 -0000	1.111
+++ config_office/configure.in	30 Mar 2005 18:40:39 -0000	1.111.2.2
@@ -3146,6 +3155,7 @@
 
 KDE_CFLAGS=""
 KDE_LIBS=""
+MOC="moc"
 if test "$test_kde" = "yes" -a "$ENABLE_KDE" = "TRUE" ; then
     dnl Search paths for Qt and KDE
     qt_incdirs="$QTINC /usr/local/qt/include /usr/include/qt /usr/include /usr/X11R6/include/X11/qt /usr/X11R6/include/qt /usr/lib/qt3/include /usr/lib/qt/include /usr/share/qt3/include $x_includes"
@@ -3197,6 +3207,12 @@
 your Qt installation by exporting QTDIR before running "configure".])
     fi
 
+    AC_PATH_PROG( MOC, moc, no, [$QTDIR/bin:$PATH] )
+    if test "$MOC" = "no" ; then
+        AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify
+the root of your Qt installation by exporting QTDIR before running "configure".])
+     fi
+     
     dnl Check for KDE headers
     AC_MSG_CHECKING([for KDE headers])
     kde_incdir="no"
@@ -3233,6 +3249,7 @@
 fi
 AC_SUBST(KDE_CFLAGS)
 AC_SUBST(KDE_LIBS)
+AC_SUBST(MOC)
 
 dnl ===================================================================
 dnl Test for the presence of libstartup-notification
--- config_office/set_soenv.in	15 Mar 2005 09:46:44 -0000	1.54
+++ config_office/set_soenv.in	30 Mar 2005 18:40:39 -0000	1.54.6.2
@@ -1443,6 +1443,7 @@
 ToFile( "ENABLE_KDE",        "@ENABLE_KDE@",       "e" );
 ToFile( "KDE_CFLAGS",        "@KDE_CFLAGS@",       "e" );
 ToFile( "KDE_LIBS",          "@KDE_LIBS@",         "e" );
+ToFile( "MOC",               "@MOC@",              "e" );
 ToFile( "CUSTOM_IMAGE_SETS", "@CUSTOM_IMAGE_SETS@","e" );
 ToFile( "PSPRINT",           "TRUE",             "e" );
 # [ed] 5/14/02 If we're building Aqua graphics, insert a C macro to indicate this.
