--- config_office/configure.in.orig	2006-03-30 12:23:12.000000000 +0200
+++ config_office/configure.in	2006-03-30 12:40:42.000000000 +0200
@@ -30,8 +30,12 @@
 [  --enable-libart         Enables the use of libart, instead of GPC for
                           polygon clipping.
 ],,)
-AC_ARG_ENABLE(openldap,
-[  --enable-openldap       Enables the use of the OpenLDAP LDAP SDK instead
+AC_ARG_ENABLE(ldap,
+[  --disable-ldap       Disables the use of LDAP backend via Netscape/Mozilla or
+                          OpenLDAP LDAP SDK
+],,)
+AC_ARG_WITH(openldap,
+[  --with-openldap       Enables the use of the OpenLDAP LDAP SDK instead
                           of the Netscape/Mozilla one
 ],,)
 AC_ARG_ENABLE(lockdown,
@@ -2842,15 +2846,17 @@
 fi
 
 AC_MSG_CHECKING([whether to build LDAP configuration backend])
-if test "$enable_mozilla" = "no" && test "$enable_openldap" != "yes"; then
-   AC_MSG_RESULT([no. Either Mozilla or OpenLDAP needed.])
-   WITH_LDAP=NO
-elif test "$with_system_mozilla" = "yes" || test "$enable_openldap" = "yes"; then
-   AC_MSG_RESULT([yes])
-   WITH_LDAP=YES
+if test -z "$enable_ldap" || test "$enable_ldap" = "yes"; then
+   if test "$enable_mozilla" = "yes" || test "$with_openldap" = "yes"; then
+      AC_MSG_RESULT([yes])
+      WITH_LDAP=YES
+   else
+      AC_MSG_RESULT([no. Either Mozilla or OpenLDAP needed])
+      WITH_LDAP=NO
+   fi
 else
-   AC_MSG_RESULT([yes])
-   WITH_LDAP=YES
+   AC_MSG_RESULT([no. Either Mozilla or OpenLDAP needed.])
+   WITH_LDAP=NO
 fi
 
 if test "$WITH_LDAP" = "YES"; then
@@ -2858,7 +2864,7 @@
    dnl Test whether we want to use the Mozilla or the OpenLDAP LDAP SDK
    dnl ===================================================================
    AC_MSG_CHECKING([which LDAP SDK to use])
-   if test -n "$enable_openldap" && test "$enable_openldap" != "no"; then
+   if test -n "$with_openldap" && test "$with_openldap" != "no"; then
   	   AC_MSG_RESULT([OpenLDAP])
 	   WITH_OPENLDAP=YES
 	   AC_CHECK_HEADERS(ldap.h, [],
@@ -2876,6 +2882,7 @@
 	   WITH_OPENLDAP=NO
    fi
 fi
+AC_SUBST(WITH_LDAP)
 AC_SUBST(WITH_OPENLDAP)
 
 dnl ===================================================================
@@ -2923,7 +2930,7 @@
     MOZ_INC=`$PKG_CONFIG --variable=includedir $moz_flavour-xpcom`
     MOZ_LIB=`$PKG_CONFIG --variable=libdir $moz_flavour-xpcom`
     MOZ_LIB_XPCOM=$MOZILLAXPCOM_LIBS
-    if test "$WITH_OPENLDAP" != "YES"; then
+    if test "$WITH_LDAP" != "NO" && test "$WITH_OPENLDAP" != "YES"; then
       AC_MSG_CHECKING([whether $tmp was compiled with --enable-ldap])
       if test -d "$MOZ_INC/ldap"; then
          AC_MSG_RESULT([yes])
--- config_office/set_soenv.in.orig	2006-03-30 13:00:11.000000000 +0200
+++ config_office/set_soenv.in	2006-03-30 13:00:44.000000000 +0200
@@ -1721,6 +1721,7 @@
 ToFile( "WITH_LIBSN",        "@WITH_LIBSN@",       "e" );
 ToFile( "LIBSN_CFLAGS",      "@LIBSN_CFLAGS@",     "e" );
 ToFile( "LIBSN_LIBS",        "@LIBSN_LIBS@",       "e" );
+ToFile( "WITH_LDAP",         "@WITH_LDAP@",        "e" );
 ToFile( "WITH_OPENLDAP",     "@WITH_OPENLDAP@",    "e" );
 ToFile( "WITH_MOZILLA",      "@WITH_MOZILLA@",     "e" );
 ToFile( "SYSTEM_MOZILLA",    "@SYSTEM_MOZILLA@",   "e" );
--- extensions/source/config/ldap/makefile.mk.orig	2006-03-30 12:23:20.000000000 +0200
+++ extensions/source/config/ldap/makefile.mk	2006-03-30 12:42:42.000000000 +0200
@@ -48,9 +48,9 @@
 .INCLUDE : settings.mk
 DLLPRE =
 
-.IF "$(WITH_MOZILLA)" == "NO" && "$(WITH_OPENLDAP)" != "YES"
+.IF "$(WITH_LDAP)" != "YES"
 @all:
-	@echo "Mozilla disabled and no OpenLDAP wanted -> LDAP disabled."
+	@echo "LDAP disabled."
 .ENDIF
 
 .IF "$(LDAPSDKINCLUDES)" == ""
--- scp2/source/ooo/makefile.mk.orig	2006-03-30 12:44:01.000000000 +0200
+++ scp2/source/ooo/makefile.mk	2006-03-30 12:45:05.000000000 +0200
@@ -119,8 +119,8 @@
 SCPDEFS+=-DSYSTEM_MOZILLA
 .ENDIF
 
-.IF "$(WITH_OPENLDAP)" == "YES"
-SCPDEFS+=-DWITH_OPENLDAP
+.IF "$(WITH_LDAP)" == "YES"
+SCPDEFS+=-DWITH_LDAP
 .ENDIF
 
 .IF "$(SYSTEM_CURL)" == "YES"
--- scp2/source/ooo/file_library_ooo.scp.orig	2006-03-30 12:23:30.000000000 +0200
+++ scp2/source/ooo/file_library_ooo.scp	2006-03-30 12:45:33.000000000 +0200
@@ -157,8 +157,7 @@
   #endif
 End
 
-#ifndef WITHOUT_LDAP
-#if !defined WITHOUT_MOZILLA || defined WITH_OPENLDAP
+#ifdef WITH_LDAP
 File gid_File_Lib_Ldapbe2
     TXT_FILE_BODY;
     Styles = (PACKED,UNO_COMPONENT,PATCH);
@@ -171,7 +170,6 @@
   #endif
 End
 #endif
-#endif
 
 // BM: #117495# remove chart2 from setup.  I just comment this
 // out, because it will later be re-added
