diff -rup ../src680-m165-orig/config_office/configure.in config_office/configure.in
--- ../src680-m165-orig/config_office/configure.in	2006-05-25 10:56:39.000000000 +0200
+++ config_office/configure.in	2006-05-25 11:08:26.000000000 +0200
@@ -4264,6 +4320,7 @@ if test  "$test_gtk" = "yes"; then
 
     if test "$ENABLE_GTK" = "TRUE" ; then
         PKG_CHECK_MODULES(GTK, gtk+-2.0 >= 2.4)
+        PKG_CHECK_MODULES(DBUSGLIB, dbus-glib-1 >= 0.60)
         PKG_CHECK_MODULES(GDKPIXBUF, gdk-pixbuf-xlib-2.0 >= 2.2)
         GTK_CFLAGS="$GTK_CFLAGS $GDKPIXBUF_CFLAGS"
         GTK_LIBS="$GTK_LIBS $GDKPIXBUF_LIBS"
