--- officecfg/registry/schema/org/openoffice/Office/Linguistic.xcs	2005-04-04 18:09:25.000000000 +0530
+++ officecfg/registry/schema/org/openoffice/Office/Linguistic.xcs	2005-08-04 14:17:21.066786557 +0530
@@ -144,7 +144,7 @@
 						<desc>Specifies the custom dictionaries to be used. The Ignore list (IgnoreAllList (All)) is administered in the RAM only for the current spellcheck.</desc>
 						<label>Custom dictionaries</label>
 					</info>
-					<value oor:separator=";">IgnoreAllList;soffice.dic;standard.dic;sun.dic</value>
+					<value oor:separator=";">IgnoreAllList;soffice.dic;standard.dic;Novell.dic;sun.dic;corporate.dic</value>
 				</prop>
 				<prop oor:name="IsUseDictionaryList" oor:type="xs:boolean">
 					<!-- OldLocation: NEW -->

Index: extras/source/wordbook/makefile.mk
===================================================================
RCS file: /cvs/installation/extras/source/wordbook/makefile.mk,v
retrieving revision 1.2
diff -u -p -u -r1.2 makefile.mk
--- extras/source/wordbook/makefile.mk	8 Sep 2005 21:20:19 -0000	1.2
+++ extras/source/wordbook/makefile.mk	16 Feb 2006 11:14:03 -0000
@@ -47,7 +47,7 @@ TARGET=wordbookall
 
 ZIP1TARGET      = $(WORDBOOK_ALL_TARGET)
 
-ZIP1LIST        = sun.dic
+ZIP1LIST        = sun.dic Novell.dic
 
 
 ZIP2TARGET      = $(WORDBOOK_USER_TARGET)
