Index: sw/source/ui/app/docsh.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/app/docsh.cxx,v
retrieving revision 1.52
retrieving revision 1.52.40.1
diff -u -p -u -r1.52 -r1.52.40.1
--- sw/source/ui/app/docsh.cxx	14 Jul 2005 11:41:20 -0000	1.52
+++ sw/source/ui/app/docsh.cxx	15 Aug 2005 11:18:13 -0000	1.52.40.1
@@ -435,7 +435,8 @@ BOOL SwDocShell::ConvertFrom( SfxMedium&
 	RTL_LOGFILE_CONTEXT_AUTHOR( aLog, "SW", "JP93722",  "SwDocShell::ConvertFrom" );
 
 	SwReader* pRdr;
-	SwRead pRead = StartConvertFrom(rMedium, &pRdr);
+	SwRead pRead = StartConvertFrom(rMedium, &pRdr); 
+	SotStorageRef pStg=pRead->getSotStorageRef(); // #i45333# save sot storage ref in case of recursive calls
 	if (!pRead)
 		return FALSE;
 
@@ -492,6 +493,8 @@ BOOL SwDocShell::ConvertFrom( SfxMedium&
 	if( bOk && !pDoc->IsInLoadAsynchron() )
 		StartLoadFinishedTimer();
 
+	pRead->setSotStorageRef(pStg); // #i45333# save sot storage ref in case of recursive calls
+
 	return bOk;
 }
 
Index: sw/inc/shellio.hxx
===================================================================
RCS file: /cvs/sw/sw/inc/shellio.hxx,v
retrieving revision 1.28
retrieving revision 1.28.334.1
diff -u -p -u -r1.28 -r1.28.334.1
--- sw/inc/shellio.hxx	25 Jan 2005 13:58:38 -0000	1.28
+++ sw/inc/shellio.hxx	15 Aug 2005 11:19:05 -0000	1.28.334.1
@@ -338,6 +338,9 @@ public:
 	virtual USHORT GetSectionList( SfxMedium& rMedium,
 									SvStrings& rStrings ) const;
 
+	SotStorageRef getSotStorageRef() { return pStg; };
+	void setSotStorageRef(SotStorageRef pStg) { this->pStg=pStg; };
+
 private:
     virtual ULONG Read(SwDoc &, const String& rBaseURL, SwPaM &,const String &)=0;
 
