Index: ucb/source/ucp/gvfs/stream.cxx
===================================================================
RCS file: /cvs/ucb/ucb/source/ucp/gvfs/stream.cxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 stream.cxx
--- ucb/source/ucp/gvfs/stream.cxx	17 Sep 2006 13:54:55 -0000	1.5
+++ ucb/source/ucp/gvfs/stream.cxx	8 Feb 2007 21:22:17 -0000
@@ -165,7 +165,7 @@ sal_Int32 SAL_CALL Stream::readSomeBytes
 {
 	// Again - having 2 methods here just sucks; cf. filinpstr.cxx
 	// This can never be an effective non-blocking API - so why bother ?
-	return readSomeBytes( aData, nMaxBytesToRead );
+	return readBytes( aData, nMaxBytesToRead );
 }
 
 void SAL_CALL Stream::skipBytes( sal_Int32 nBytesToSkip )
