Index: fpicker/source/unx/gnome/SalGtkFilePicker.cxx
===================================================================
RCS file: /cvs/gsl/fpicker/source/unx/gnome/SalGtkFilePicker.cxx,v
retrieving revision 1.7
diff -u -p -u -r1.7 SalGtkFilePicker.cxx
--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx	12 Jul 2005 11:58:49 -0000	1.7
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	1 Sep 2005 11:30:51 -0000
@@ -955,7 +963,9 @@ uno::Sequence<rtl::OUString> SAL_CALL Sa
 			FilterList::iterator aListIter = ::std::find_if( 
                 m_pFilterList->begin(), m_pFilterList->end(), FilterTitleMatch(sFilterName) );
 
-			OUString aFilter = aListIter->getFilter();
+			OUString aFilter;
+			if (aListIter != m_pFilterList->end())
+				aFilter = aListIter->getFilter();
 
 			OSL_TRACE( "turned into %s\n", 
 				OUStringToOString( aFilter, RTL_TEXTENCODING_UTF8 ).getStr() );
