Index: fpicker/source/unx/gnome/SalGtkFilePicker.cxx
===================================================================
RCS file: /cvs/gsl/fpicker/source/unx/gnome/SalGtkFilePicker.cxx,v
retrieving revision 1.10
diff -u -p -u -r1.10 SalGtkFilePicker.cxx
--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx	28 Sep 2005 14:09:38 -0000	1.10
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	17 Feb 2006 14:58:22 -0000
@@ -905,7 +905,14 @@ uno::Sequence<rtl::OUString> SAL_CALL Sa
     {
 		gchar *path = gtk_file_chooser_get_current_folder_uri( 
             GTK_FILE_CHOOSER( m_pDialog ));
-        nURLOffset = strlen(path) + 1;
+		if (!path)
+		{
+		    g_warning ("unusual (NULL) file chooser URI");
+			path = g_strdup ("");
+			nURLOffset = 0;
+		}
+		else
+			nURLOffset = strlen (path) + 1;
         aSelectedFiles[0] = uritounicode(path);
         g_free(path);
     }

Index: fpicker/source/unx/gnome/SalGtkPicker.cxx
===================================================================
RCS file: /cvs/gsl/fpicker/source/unx/gnome/SalGtkPicker.cxx,v
retrieving revision 1.9
diff -u -p -u -r1.9 SalGtkPicker.cxx
--- fpicker/source/unx/gnome/SalGtkPicker.cxx	8 Sep 2005 23:35:50 -0000	1.9
+++ fpicker/source/unx/gnome/SalGtkPicker.cxx	17 Feb 2006 14:58:22 -0000
@@ -80,6 +80,8 @@ using namespace ::com::sun::star::uno;
 
 rtl::OUString SalGtkPicker::uritounicode(const gchar* pIn)
 {
+	g_return_val_if_fail (pIn != NULL, rtl::OUString());
+
     rtl::OUString sURL( const_cast<const sal_Char *>(pIn), strlen(pIn),
         RTL_TEXTENCODING_UTF8 );
     INetURLObject aURL(sURL);
