Index: vcl/source/app/svmain.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/app/svmain.cxx,v
retrieving revision 1.53
diff -u -r1.53 svmain.cxx
--- vcl/source/app/svmain.cxx	24 Feb 2005 15:23:19 -0000	1.53
+++ vcl/source/app/svmain.cxx	22 Mar 2005 13:41:35 -0000
@@ -248,6 +248,15 @@
 {
     RTL_LOGFILE_CONTEXT( aLog, "vcl (ss112471) ::SVMain" );
 
+#ifdef UNX
+	if( !getenv( "STAR_RESOURCEPATH" ) )
+	{
+		fprintf( stderr, "** Error: Your environment looks broken, if you're using linkoo "
+				 "you need to source ./env before running soffice.bin\n" );
+		return FALSE;
+	}
+#endif
+
     ImplSVData* pSVData = ImplGetSVData();
 
     DBG_ASSERT( pSVData->mpApp, "no instance of class Application" );
