--- vcl/source/gdi/image.cxx
+++ vcl/source/gdi/image.cxx
@@ -970,7 +970,11 @@
 	if( mpImplData )
 	{
 		for( USHORT i = 0; i < mpImplData->maImages.size(); i++ )
-			rNames.push_back( mpImplData->maImages[ i ]->maName );
+        {
+            const rtl::OUString &rName = mpImplData->maImages[ i ]->maName;
+            if( rName.getLength() > 0 )
+                rNames.push_back( rName );
+        }
 	}
 }
 


--- svx/source/dialog/cfg.cxx
+++ svx/source/dialog/cfg.cxx
@@ -5457,13 +5462,16 @@
 			// added to the list
 		}
 
+        uno::Reference< graphic::XGraphic > xGraph;
 		if ( graphics.getLength() > 0 )
+            xGraph = graphics[0];
+        if ( xGraph.is() )
 		{
-			Image img = Image( graphics[ 0 ] );
+			Image img = Image( xGraph );
 			aTbSymbol.InsertItem( nId, img, pConstIter->first );
 
-			graphics[ 0 ]->acquire();
+			xGraph->acquire();
 
 			aTbSymbol.SetItemData(
 				nId, static_cast< void * > ( graphics[ 0 ].get() ) );
 
