Index: slideshow/prj/build.lst
===================================================================
RCS file: /cvs/graphics/slideshow/prj/build.lst,v
retrieving revision 1.6
retrieving revision 1.6.12.1
diff -u -p -u -p -r1.6 -r1.6.12.1
--- slideshow/prj/build.lst	17 Jul 2007 14:32:58 -0000	1.6
+++ slideshow/prj/build.lst	29 Aug 2007 14:52:39 -0000	1.6.12.1
@@ -1,4 +1,4 @@
-pe	slideshow	:	tools sal cppu vcl comphelper cppuhelper offuh cppcanvas basegfx goodies unotools toolkit animations NULL
+pe	slideshow	:	tools sal cppu vcl comphelper cppuhelper offuh cppcanvas basegfx goodies unotools toolkit animations avmedia NULL
 pe	slideshow								usr1	-	all	pe_mkout NULL
 pe	slideshow\inc							nmake	-	all	pe_inc NULL
 pe	slideshow\source\api					nmake	-	all	pe_api pe_inc NULL
Index: slideshow/source/engine/soundplayer.cxx
===================================================================
RCS file: /cvs/graphics/slideshow/source/engine/soundplayer.cxx,v
retrieving revision 1.9
retrieving revision 1.8.14.2
diff -u -p -u -p -r1.9 -r1.8.14.2
--- slideshow/source/engine/soundplayer.cxx	17 Aug 2007 12:43:15 -0000	1.9
+++ slideshow/source/engine/soundplayer.cxx	4 Oct 2007 11:24:15 -0000	1.8.14.2
@@ -48,6 +48,8 @@
 
 #include <tools/urlobj.hxx>
 
+#include <avmedia/mediawindow.hxx>
+
 #include "soundplayer.hxx"
 
 #include <algorithm>
@@ -55,16 +57,6 @@
 using namespace ::com::sun::star;
 
 
-// TODO(Q3): This breaks encapsulation. Either export
-// these strings from avmedia, or provide an XManager
-// factory there
-#ifdef WNT
-#	define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_DirectX"
-#else
-#	define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_Java"
-#endif 
-
-
 namespace slideshow
 {
     namespace internal
@@ -122,22 +114,10 @@ namespace slideshow
 
             try
             {
-                uno::Reference<lang::XMultiComponentFactory> xFac( 
-                    rComponentContext->getServiceManager() );
-                        
-                uno::Reference< ::com::sun::star::media::XManager > xManager(
-                    xFac->createInstanceWithContext( 
-                        ::rtl::OUString::createFromAscii( AVMEDIA_MANAGER_SERVICE_NAME ),
-                        rComponentContext ),
-                    uno::UNO_QUERY );
-                    
-                if( xManager.is() )
-                {
-                    const INetURLObject aURL( rSoundURL );
-                    mxPlayer.set( xManager->createPlayer( 
-                                      aURL.GetMainURL( INetURLObject::DECODE_UNAMBIGUOUS ) ), 
-                                  uno::UNO_QUERY );
-                }
+        		const INetURLObject aURL( rSoundURL );		
+                mxPlayer.set( avmedia::MediaWindow::createPlayer(
+                                aURL.GetMainURL( INetURLObject::DECODE_UNAMBIGUOUS ) ),
+                                uno::UNO_QUERY);
             }
             catch( uno::RuntimeException& )
             {
Index: slideshow/source/engine/shapes/viewmediashape.cxx
===================================================================
RCS file: /cvs/graphics/slideshow/source/engine/shapes/viewmediashape.cxx,v
retrieving revision 1.2
retrieving revision 1.2.14.1
diff -u -p -u -p -r1.2 -r1.2.14.1
--- slideshow/source/engine/shapes/viewmediashape.cxx	17 Jul 2007 14:55:30 -0000	1.2
+++ slideshow/source/engine/shapes/viewmediashape.cxx	29 Aug 2007 14:52:39 -0000	1.2.14.1
@@ -60,6 +60,7 @@
 #include <cppcanvas/vclfactory.hxx>
 #include <cppcanvas/basegfxfactory.hxx>
 #include <cppcanvas/basegfxfactory.hxx>
+#include <avmedia/mediawindow.hxx>
 
 #include <com/sun/star/media/XManager.hpp>
 #include <com/sun/star/media/XPlayer.hpp>
@@ -75,12 +76,6 @@
 #include "mediashape.hxx"
 #include "tools.hxx"
 
-#ifdef WNT
-#define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_DirectX"
-#else
-#define AVMEDIA_MANAGER_SERVICE_NAME "com.sun.star.media.Manager_Java"
-#endif
-
 using namespace ::com::sun::star;
 
 namespace slideshow
@@ -375,24 +370,8 @@ namespace slideshow
 				{
 					if( rMediaURL.getLength() )
 					{
-                        uno::Reference<lang::XMultiComponentFactory> xFactory( 
-                            mxComponentContext->getServiceManager() );
-
-						if( xFactory.is() )
-						{
-							uno::Reference< media::XManager > xManager;
-								
-							xManager.set( 
-                                xFactory->createInstanceWithContext( 
-                                    ::rtl::OUString::createFromAscii( 
-                                        AVMEDIA_MANAGER_SERVICE_NAME ),
-                                    mxComponentContext ), 
-                                uno::UNO_QUERY );
-                            
-							if( xManager.is() )
-								mxPlayer.set( xManager->createPlayer( rMediaURL ), 
-                                              uno::UNO_QUERY );
-						}
+						mxPlayer.set( avmedia::MediaWindow::createPlayer( rMediaURL ), 
+                            uno::UNO_QUERY );
 					}
 				}
                 catch( uno::RuntimeException& )
Index: slideshow/util/makefile.mk
===================================================================
RCS file: /cvs/graphics/slideshow/util/makefile.mk,v
retrieving revision 1.8
retrieving revision 1.8.12.1
diff -u -p -u -p -r1.8 -r1.8.12.1
--- slideshow/util/makefile.mk	17 Jul 2007 15:22:33 -0000	1.8
+++ slideshow/util/makefile.mk	29 Aug 2007 14:52:39 -0000	1.8.12.1
@@ -76,7 +76,8 @@ SHL1STDLIBS= $(TOOLSLIB) \
 			 $(CPPCANVASLIB) \
 			 $(UNOTOOLSLIB) \
 			 $(GOODIESLIB) \
-			 $(TKLIB)
+			 $(TKLIB) \
+			 $(AVMEDIALIB)
 
 SHL1IMPLIB=i$(TARGET)
 SHL1LIBS=$(SLB)$/$(TARGET).lib
