--- solenv/bin/cws-commit-patch	2006-12-01 17:23:55.000000000 +0100
+++ solenv/bin/cws-commit-patch	2008-03-25 18:49:02.000000000 +0100
@@ -38,7 +38,7 @@ eval 'exec perl -wS $0 ${1+"$@"}'
 
 use Cwd 'abs_path';
 
-$default_mws = 'SRC680';
+$default_mws = 'DEV300';
 
 sub usage()
 {
@@ -109,13 +109,20 @@ sub cws_add_modules($$$@)
             print STDERR "Unable to open patch '$patch'.";
             exit 1;
         }
+        my $minus_name;
         while (<$PATCH>) {
-            if ( /^\+\+\+ ([^\/]*)/ ) {
+            if ( /^--- ([^\/]*)/ ) {
+                $minus_name = $1;
+            }
+            elsif ( /^\+\+\+ ([^\/]*)/ ) {
                 if ( $1 ne "" ) {
                     $modules{$1} = "add";
                 }
+                elsif ( $minus_name ne "" ) {
+                    $modules{$minus_name} = "add";
+                }
                 else {
-                    print STDERR "Unable to guess module name ($1) from $patch.\n";
+                    print STDERR "Unable to guess module name from $patch.\n";
                     close( $PATCH );
                     exit 1;
                 }
@@ -183,10 +190,10 @@ sub apply_patches(@)
     }
 }
 
-sub cvs_add_files(@)
+sub cvs_add_rm_files(@)
 {
     my ( @patches ) = @_;
-    system( "echo '===== cvs_add_files =====' >> .log" );
+    system( "echo '===== cvs_add_rm_files =====' >> .log" );
 
     my %modules = ();
 
@@ -195,30 +202,48 @@ sub cvs_add_files(@)
             print STDERR "Unable to open patch '$patch'.";
             exit 1;
         }
-	my $file_add = "";
+        my $file_add = "";
+        my $file_rm = "";
         while (<$PATCH>) {
-            if ( /^\+\+\+ ([^\s]*)/ ) {
+            if ( /^--- ([^\s]*)/ ) {
+                if ( $1 ne "" ) {
+                    $file_rm = $1;
+                }
+            }
+            elsif ( /^\+\+\+ ([^\s]*)/ ) {
                 if ( $1 ne "" ) {
-		    $file_add = $1;
+                    $file_add = $1;
+                }
+            }
+            elsif ( /^\@\@ .* \+0,0 \@\@$/ && $file_rm ne "" ) {
+                if ( $file_rm =~ /^([^\/]+)\/(.*)/ ) {
+                    my $dir = $1;
+                    my $file = $2;
+
+                    if ( system( "cd $dir && cvs rm $file" ) != 0 ) {
+                        print STDERR "Unable to exec 'cd $dir && cvs rm $file'.\n";
+                        close( $PATCH );
+                        exit 1;
+                    }
                 }
             }
             elsif ( /^\@\@ -0,0 / && $file_add ne "" ) {
-		my @to_add = split( /\//, $file_add );
-		my $current = "";
-		for my $add ( @to_add ) {
-		    $current .= "/" unless ( $current eq "" );
-		    my $where_add_dir = $current;
-		    $current .= $add;
-
-		    if ( ( -d "$current" && ! -d "$current/CVS" ) || ( -f "$current" ) ) {
-			if ( system( "cd $where_add_dir && cvs add $add" ) != 0 ) {
-			    print STDERR "Unable to exec 'cd $where_add_dir && cvs add $add'.\n";
-			    close( $PATCH );
-			    exit 1;
-			}
-		    }
-		}
-	    }
+                my @to_add = split( /\//, $file_add );
+                my $current = "";
+                for my $add ( @to_add ) {
+                    $current .= "/" unless ( $current eq "" );
+                    my $where_add_dir = $current;
+                    $current .= $add;
+
+                    if ( ( -d "$current" && ! -d "$current/CVS" ) || ( -f "$current" ) ) {
+                        if ( system( "cd $where_add_dir && cvs add $add" ) != 0 ) {
+                            print STDERR "Unable to exec 'cd $where_add_dir && cvs add $add'.\n";
+                            close( $PATCH );
+                            exit 1;
+                        }
+                    }
+                }
+            }
         }
         close( $PATCH );
     }
@@ -234,7 +259,7 @@ sub cws_commit($$$$$@)
 
     apply_patches( @patches );
 
-    cvs_add_files( @patches );
+    cvs_add_rm_files( @patches );
 
     my @cws_modules = cws_modules( $mws, $cws );
     my $commit_message = "#i$iz#\n" . $summary;
@@ -261,10 +286,10 @@ if ( !defined( $ENV{'SOLARENV'} ) || $EN
     $build_dir =~ s#/solenv/bin/[^/]*$##;
     my $env_script = "";
     if ( -f "$build_dir/LinuxIntelEnv.Set.sh" ) {
-	$env_script = "$build_dir/LinuxIntelEnv.Set.sh";
+        $env_script = "$build_dir/LinuxIntelEnv.Set.sh";
     }
     elsif ( -f "$build_dir/LinuxX86-64Env.Set.sh" ) {
-	$env_script = "$build_dir/LinuxX86-64Env.Set.sh";
+        $env_script = "$build_dir/LinuxX86-64Env.Set.sh";
     }
     else {
         print STDERR "Unable to find '$build_dir/Linux(Intel|X86-64)Env.Set.sh'.\n";
@@ -312,10 +337,10 @@ while (@ARGV) {
     }
     elsif ( $opt eq "-m" || $opt eq "--milestone" ) {
         $milestone = shift @ARGV;
-	if (! ( $milestone =~ m/^m/ ) ) {
-	    $milestone = "m$milestone";
-	    print "Re-writing milestone to $milestone\n";
-	}
+        if (! ( $milestone =~ m/^m/ ) ) {
+            $milestone = "m$milestone";
+            print "Re-writing milestone to $milestone\n";
+        }
     }
     elsif ( $opt eq "-w" || $opt eq "--mws" ) {
         $mws = shift @ARGV;
