? cws-chart16.diff
Index: chart2/source/controller/chartapiwrapper/AxisWrapper.cxx
===================================================================
RCS file: /cvs/graphics/chart2/source/controller/chartapiwrapper/AxisWrapper.cxx,v
retrieving revision 1.6
retrieving revision 1.6.24.1
diff -u -u -r1.6 -r1.6.24.1
--- chart2/source/controller/chartapiwrapper/AxisWrapper.cxx	25 Jul 2007 08:26:11 -0000	1.6
+++ chart2/source/controller/chartapiwrapper/AxisWrapper.cxx	12 Sep 2007 14:59:52 -0000	1.6.24.1
@@ -4,9 +4,9 @@
  *
  *  $RCSfile: AxisWrapper.cxx,v $
  *
- *  $Revision: 1.6 $
+ *  $Revision: 1.6.24.1 $
  *
- *  last change: $Author: rt $ $Date: 2007/07/25 08:26:11 $
+ *  last change: $Author: iha $ $Date: 2007/09/12 14:59:52 $
  *
  *  The Contents of this file are made available subject to
  *  the terms of GNU Lesser General Public License Version 2.1.
@@ -452,11 +452,9 @@
 Any AxisWrapper::getReferenceSize()
 {
     Any aRet;
-    /*
     Reference< beans::XPropertySet > xProp( this->getAxis(), uno::UNO_QUERY );
     if( xProp.is() )
         aRet = xProp->getPropertyValue( C2U("ReferenceDiagramSize") );
-    */
     return aRet;
 }
 awt::Size AxisWrapper::getCurrentSizeForReference()
Index: chart2/source/controller/chartapiwrapper/DataSeriesPointWrapper.cxx
===================================================================
RCS file: /cvs/graphics/chart2/source/controller/chartapiwrapper/DataSeriesPointWrapper.cxx,v
retrieving revision 1.9.2.1
retrieving revision 1.9.2.1.6.1
diff -u -u -r1.9.2.1 -r1.9.2.1.6.1
--- chart2/source/controller/chartapiwrapper/DataSeriesPointWrapper.cxx	17 Aug 2007 09:19:04 -0000	1.9.2.1
+++ chart2/source/controller/chartapiwrapper/DataSeriesPointWrapper.cxx	12 Sep 2007 14:58:41 -0000	1.9.2.1.6.1
@@ -4,9 +4,9 @@
  *
  *  $RCSfile: DataSeriesPointWrapper.cxx,v $
  *
- *  $Revision: 1.9.2.1 $
+ *  $Revision: 1.9.2.1.6.1 $
  *
- *  last change: $Author: obo $ $Date: 2007/08/17 09:19:04 $
+ *  last change: $Author: iha $ $Date: 2007/09/12 14:58:41 $
  *
  *  The Contents of this file are made available subject to
  *  the terms of GNU Lesser General Public License Version 2.1.
@@ -592,11 +592,9 @@
 Any DataSeriesPointWrapper::getReferenceSize()
 {
     Any aRet;
-    /*
     Reference< beans::XPropertySet > xProp( this->getInnerPropertySet(), uno::UNO_QUERY );
     if( xProp.is() )
         aRet = xProp->getPropertyValue( C2U("ReferenceDiagramSize") );
-    */
     return aRet;
 }
 awt::Size DataSeriesPointWrapper::getCurrentSizeForReference()
Index: chart2/source/controller/dialogs/tp_ChartType.cxx
===================================================================
RCS file: /cvs/graphics/chart2/source/controller/dialogs/tp_ChartType.cxx,v
retrieving revision 1.5
retrieving revision 1.5.18.1
diff -u -u -r1.5 -r1.5.18.1
--- chart2/source/controller/dialogs/tp_ChartType.cxx	3 Aug 2007 13:36:45 -0000	1.5
+++ chart2/source/controller/dialogs/tp_ChartType.cxx	12 Sep 2007 13:19:48 -0000	1.5.18.1
@@ -4,9 +4,9 @@
  *
  *  $RCSfile: tp_ChartType.cxx,v $
  *
- *  $Revision: 1.5 $
+ *  $Revision: 1.5.18.1 $
  *
- *  last change: $Author: hr $ $Date: 2007/08/03 13:36:45 $
+ *  last change: $Author: iha $ $Date: 2007/09/12 13:19:48 $
  *
  *  The Contents of this file are made available subject to
  *  the terms of GNU Lesser General Public License Version 2.1.
@@ -472,6 +472,8 @@
     SplinePropertiesDialog( Window* pParent );
     virtual ~SplinePropertiesDialog();
 
+    void adjustSize();
+
     void fillControls( const ChartTypeParameter& rParameter );
     void fillParameter( ChartTypeParameter& rParameter, bool bSmoothLines );
 
@@ -546,7 +548,21 @@
 SplinePropertiesDialog::~SplinePropertiesDialog()
 {
 }
-
+void SplinePropertiesDialog::adjustSize()
+{
+    Size aDlgSize( this->GetSizePixel() );
+    long nBorder = m_aRB_Splines_Cubic.GetPosPixel().X();
+    long nX = m_aMF_SplineOrder.GetPosPixel().X() + m_aMF_SplineOrder.GetSizePixel().Width();
+    if(aDlgSize.Width()< (nX+nBorder) )
+    {
+        aDlgSize.Width() = (nX+nBorder);
+        this->SetSizePixel(aDlgSize);
+
+        Size aLineSize( m_aFL_DialogButtons.GetSizePixel() );
+        aLineSize.Width() = aDlgSize.Width();
+        m_aFL_DialogButtons.SetSizePixel(aLineSize);
+    }
+}
 void SplinePropertiesDialog::fillControls( const ChartTypeParameter& rParameter )
 {
     switch(rParameter.eCurveStyle)
@@ -642,6 +658,7 @@
     if( !m_pSplinePropertiesDialog.get() )
     {
         m_pSplinePropertiesDialog = ::std::auto_ptr< SplinePropertiesDialog >( new SplinePropertiesDialog( m_aPB_DetailsDialog.GetParent() ) );
+        m_pSplinePropertiesDialog->adjustSize();
     }
     return *m_pSplinePropertiesDialog;
 }
Index: chart2/source/controller/dialogs/tp_ChartType.src
===================================================================
RCS file: /cvs/graphics/chart2/source/controller/dialogs/tp_ChartType.src,v
retrieving revision 1.2
retrieving revision 1.2.58.1
diff -u -u -r1.2 -r1.2.58.1
--- chart2/source/controller/dialogs/tp_ChartType.src	22 May 2007 17:44:32 -0000	1.2
+++ chart2/source/controller/dialogs/tp_ChartType.src	12 Sep 2007 13:20:13 -0000	1.2.58.1
@@ -4,9 +4,9 @@
  *
  *  $RCSfile: tp_ChartType.src,v $
  *
- *  $Revision: 1.2 $
+ *  $Revision: 1.2.58.1 $
  *
- *  last change: $Author: vg $ $Date: 2007/05/22 17:44:32 $
+ *  last change: $Author: iha $ $Date: 2007/09/12 13:20:13 $
  *
  *  The Contents of this file are made available subject to
  *  the terms of GNU Lesser General Public License Version 2.1.
@@ -61,8 +61,8 @@
 
 #define POS_X_SPLINES_1 (8)
 #define POS_Y_SPLINES_2 (8)
-#define POS_Y_SPLINES_3 (POS_Y_SPLINES_2+12)
-#define POS_Y_SPLINES_4 (POS_Y_SPLINES_3+12)
+#define POS_Y_SPLINES_3 (POS_Y_SPLINES_2+13)
+#define POS_Y_SPLINES_4 (POS_Y_SPLINES_3+13)
 #define POS_Y_SPLINES_5 (POS_Y_SPLINES_4+RSC_CD_FIXEDTEXT_HEIGHT)
 
 #define SPLINES_WIDTH_RADIO 45
@@ -72,7 +72,7 @@
 #define WIDTH_XVALUE_SORTING (WIDTH_IMAGELIST)
 
 #define SPLINE_DIALOG_WIDTH 172
-#define SPLINE_DIALOG_HEIGHT 57
+#define SPLINE_DIALOG_HEIGHT 59
 
 TabPage TP_CHARTTYPE
 {
