Index: bin/modules/CwsConfig.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/CwsConfig.pm,v
retrieving revision 1.7
diff -u -w -p -r1.7 CwsConfig.pm
--- bin/modules/CwsConfig.pm	8 Sep 2005 08:56:07 -0000	1.7
+++ bin/modules/CwsConfig.pm	26 May 2006 18:46:05 -0000
@@ -378,9 +378,13 @@ sub read_config
     	    }
 	    }
 	    defined $config{$section} || croak("ERROR: unknown / no section '$section'\n");
-    	if ( m/(\w[\w\d]*)=(.*)\s*/ ) {
-	        $config{$section}->{$1} = $2;
-    #	    print "Set '$1' to '$2'\n";
+	    if ( m/(\w(\w\d)+)=(.*)$/ ) {
+		my $var = $1;
+		my $val = $2;
+		$val =~ s/(\s*['"])?(.*)\1\s*$/\2/;
+		$val =~ s/\\(["'])/\1/g;
+	        $config{$section}->{$var} = $val;
+#		print "Set '$var' to '$val'\n";
 	    }
     }
     close ($fhandle) || croak("ERROR: Failed to close: $!");
