Index: connectivity/source/drivers/hsqldb/HDriver.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/hsqldb/HDriver.cxx,v
retrieving revision 1.23
retrieving revision 1.23.50.2
diff -u -u -p -r1.23 -r1.23.50.2
--- connectivity/source/drivers/hsqldb/HDriver.cxx	12 Mar 2007 10:40:59 -0000	1.23
+++ connectivity/source/drivers/hsqldb/HDriver.cxx	8 Aug 2007 09:16:44 -0000	1.23.50.2
@@ -241,7 +241,13 @@ namespace connectivity
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("JavaDriverClass"));
 				aConvertedProperties[nPos++].Value <<= ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("org.hsqldb.jdbcDriver"));
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("JavaDriverClassPath"));
-				aConvertedProperties[nPos++].Value <<= ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("vnd.sun.star.expand:$ORIGIN/classes/hsqldb.jar vnd.sun.star.expand:$ORIGIN/classes/sdbc_hsqldb.jar"));
+				aConvertedProperties[nPos++].Value <<= ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM(
+#ifdef SYSTEM_HSQLDB
+					HSQLDB_JAR
+#else
+					"vnd.sun.star.expand:$ORIGIN/classes/hsqldb.jar"
+#endif
+					" vnd.sun.star.expand:$ORIGIN/classes/sdbc_hsqldb.jar"));
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("IsAutoRetrievingEnabled"));
 				aConvertedProperties[nPos++].Value <<= sal_True;
 				aConvertedProperties[nPos].Name = ::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM("AutoRetrievingStatement"));
Index: connectivity/source/drivers/hsqldb/makefile.mk
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/hsqldb/makefile.mk,v
retrieving revision 1.11
retrieving revision 1.11.50.1
diff -u -u -p -r1.11 -r1.11.50.1
--- connectivity/source/drivers/hsqldb/makefile.mk	9 Mar 2007 08:58:53 -0000	1.11
+++ connectivity/source/drivers/hsqldb/makefile.mk	8 Aug 2007 08:40:13 -0000	1.11.50.1
@@ -50,6 +50,9 @@ nojava:
 	@echo "Not building jurt because Java is disabled"
 .ENDIF
 
+.IF "$(SYSTEM_HSQLDB)" == "YES"
+CDEFS+=-DSYSTEM_HSQLDB -DHSQLDB_JAR=\""file://$(HSQLDB_JAR)"\"
+.ENDIF
 # --- Resources ---------------------------------
 
 SRS1NAME=$(TARGET)
Index: scp2/source/ooo/profileitem_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/profileitem_ooo.scp,v
retrieving revision 1.49
retrieving revision 1.49.14.1
diff -u -u -p -r1.49 -r1.49.14.1
--- scp2/source/ooo/profileitem_ooo.scp	26 Jul 2007 08:57:23 -0000	1.49
+++ scp2/source/ooo/profileitem_ooo.scp	8 Aug 2007 09:22:22 -0000	1.49.14.1
@@ -859,7 +859,7 @@ ProfileItem gid_Profileitem_Jvmfwk_UnoSe
   #endif
 End
 
-#if defined SYSTEM_BSH || defined SYSTEM_HSQLDB || defined SYSTEM_XML_APIS || \
+#if defined SYSTEM_BSH || defined SYSTEM_XML_APIS || \
     defined SYSTEM_XERCES || defined SYSTEM_XALAN || defined SYSTEM_SERIALIZER
 ProfileItem gid_Profileitem_Jvmfwk_Uno_Java_Jfw_Classpath_Urls
     ProfileID = gid_Profile_Jvmfwk_Ini;
@@ -871,20 +871,17 @@ ProfileItem gid_Profileitem_Jvmfwk_Uno_J
 #ifdef SYSTEM_BSH
     ValueList1 = BSH_JAR;
 #endif
-#ifdef SYSTEM_HSQLDB
-    ValueList2 = HSQLDB_JAR;
-#endif
 #ifdef SYSTEM_XML_APIS
-    ValueList3 = XML_APIS_JAR;
+    ValueList2 = XML_APIS_JAR;
 #endif
 #ifdef SYSTEM_XERCES
-    ValueList4 = XERCES_JAR;
+    ValueList3 = XERCES_JAR;
 #endif
 #ifdef SYSTEM_XALAN
-    ValueList5 = XALAN_JAR;
+    ValueList4 = XALAN_JAR;
 #endif
 #ifdef SYSTEM_SERIALIZER
-    ValueList6 = SERIALIZER_JAR;
+    ValueList5 = SERIALIZER_JAR;
 #endif
 End
 #endif
