Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.116
retrieving revision 1.112.4.2
diff -u -u -r1.116 -r1.112.4.2
--- config_office/configure.in	11 Apr 2005 09:31:15 -0000	1.116
+++ config_office/configure.in	15 May 2005 14:12:56 -0000	1.112.4.2
@@ -152,6 +152,15 @@
 AC_ARG_ENABLE(pasf,
 [  --disable-pasf          disable support for sound using portaudio/sndfile
 ],,enable_pasf=yes)
+AC_ARG_ENABLE(pam-link,
+[   --enable-pam-link       link with libpam instead of dynamically open it
+],,)
+AC_ARG_ENABLE(crypt-link,
+[   --enable-crypt-link     link with libcrypt instead of dynamically open it
+],,)
+AC_ARG_ENABLE(xrender-link,
+[   --enable-xrender-link   link with libXrender instead of dynamically open it
+],,)
 AC_ARG_WITH(myspell-dicts,
 [  --without-myspell-dicts Removes myspell dictionaries from openoffice.org
                           installation set, for people building for specific
@@ -224,6 +233,15 @@
 AC_ARG_WITH(system-neon,
 [  --with-system-neon      Use neon already on system
 ],,)
+AC_ARG_WITH(system-myspell,
+[  --with-system-myspell  Use myspell already on system
+],,)
+AC_ARG_WITH(system-mythes,
+[  --with-system-mythes   Use mythes already on system
+],,)
+AC_ARG_WITH(system-altlinuxhyph,
+[  --with-system-altlinuxhyph    Use ALTLinuxhyph already on system
+],,)
 AC_ARG_WITH(system-mozilla,
 [  --with-system-mozilla   Use mozilla already on system. Note that some
                           components cannot be built against a contemporary
@@ -1286,12 +1304,35 @@
 fi
 
 dnl ===================================================================
-dnl Check if pam_appl.h is available on Linux or FreeBSD
+dnl Check if PAM/pam_appl.h is available on Linux or FreeBSD
 dnl ===================================================================
 if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "GNU"; then
     AC_CHECK_HEADER(security/pam_appl.h, [],
                     [AC_MSG_ERROR([pam_appl.h could not be found. libpam-dev or pam-devel missing?])], [])
+    AC_MSG_CHECKING([whether to link to libpam])
+    if test -n "$enable_pam_link"; then
+      AC_MSG_RESULT([yes])
+      PAM_LINK=YES
+      AC_CHECK_LIB(pam, pam_start, [],
+          [AC_MSG_ERROR(libpam not found or functional)], [])
+    else
+      AC_MSG_RESULT([no, dynamically open it])
+      PAM_LINK=NO
+    fi
+fi
+AC_SUBST(PAM_LINK)
+
+AC_MSG_CHECKING([whether to link to libcrypt])
+if test -n "$enable_crypt_link"; then
+   AC_MSG_RESULT([yes])
+   CRYPT_LINK=YES
+   AC_CHECK_LIB(crypt, crypt, [],
+       [AC_MSG_ERROR(libcrypt not found or functional)], [])
+else
+   AC_MSG_RESULT([no, dynamically open it])
+   CRYPT_LINK=NO
 fi
+AC_SUBST(CRYPT_LINK)
 
 dnl ===================================================================
 dnl Testing for c++ compiler and version...
@@ -2876,8 +2917,17 @@
 dnl ===================================================================
 dnl Check for system Xrender
 dnl ===================================================================
-AC_MSG_CHECKING([which XRender headers to use])
-if test -n "$with_system_xrender_headers" && test -n "$with_system_xrender_headers" != "no"; then
+AC_MSG_CHECKING([whether to link to Xrender])
+if test -n "$enable_xrender_link"; then
+   AC_MSG_RESULT([yes])
+   XRENDER_LINK=YES
+   with_system_xrender_headers=yes
+else
+   AC_MSG_RESULT([no, dynamically open it])
+   XRENDER_LINK=NO
+fi
+AC_MSG_CHECKING([which Xrender headers to use])
+if test -n "$with_system_xrender_headers" && test "$with_system_xrender_headers" != "no"; then
    AC_MSG_RESULT([external])
    SYSTEM_XRENDER_HEADERS=YES
    AC_CHECK_HEADER(X11/extensions/Xrender.h, [],
@@ -2887,7 +2937,12 @@
    SYSTEM_XRENDER_HEADERS=NO
    BUILD_TYPE="$BUILD_TYPE X11_EXTENSIONS"
 fi
+if test "$XRENDER_LINK" = "YES"; then
+   AC_CHECK_LIB(Xrender, XRenderQueryVersion, [],
+      [AC_MSG_ERROR(libXrender not found or functional)], [])
+fi
 AC_SUBST(SYSTEM_XRENDER_HEADERS)
+AC_SUBST(XRENDER_LINK)
 
 dnl ===================================================================
 dnl Check for system nas
@@ -2984,6 +3039,71 @@
 AC_SUBST(SYSTEM_NEON)
 AC_SUBST(NEON_LIBS)
 AC_SUBST(NEON_CFLAGS)
+
+dnl ===================================================================
+dnl Check for system myspell
+dnl ===================================================================
+AC_MSG_CHECKING([which myspell to use])
+if test -n "$with_system_myspell" && test "$with_system_myspell" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_MYSPELL=YES
+    PKG_CHECK_MODULES( MYSPELL, myspell,
+      [ MYSPELL_PKGCONFIG=YES ],
+      [
+         AC_CHECK_HEADER(myspell/myspell.hxx, [],
+            [AC_MSG_ERROR(myspell headers not found.)], [])
+         MYSPELL_CFLAGS=-I/usr/include/myspell
+         AC_LANG(C++)
+         AC_CHECK_LIB(myspell, main, [],
+           [ AC_MSG_ERROR(myspell library not found.) ], [])
+         MYSPELL_LIBS=-lmyspell
+      ]
+    )
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_MYSPELL=NO
+    MYSPELL_LIBS=-lmyspell
+fi
+AC_SUBST(SYSTEM_MYSPELL)
+AC_SUBST(MYSPELL_CFLAGS)
+AC_SUBST(MYSPELL_LIBS)
+
+dnl ===================================================================
+dnl Checking for altlinuxhyph
+dnl ===================================================================
+AC_MSG_CHECKING([which altlinuxhyph to use])
+if test -n "$with_system_altlinuxhyph" && test "$with_system_altlinuxyph" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_HYPH=YES
+    AC_CHECK_HEADER(hyphen.h, [],
+       [ AC_MSG_ERROR(altlinuxhyph headers not found.)], [])
+    AC_CHECK_MEMBER(struct _HyphenDict.cset, [],
+       [ AC_MSG_ERROR(no. You are sure you have altlinuyhyph headers?)],
+       [#include <hyphen.h>])
+    AC_CHECK_LIB(hyph, hnj_hyphen_hyphenate, [],
+        [ AC_MSG_ERROR(altlinuxhyph library not found.)], [])
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_HYPH=NO
+fi
+AC_SUBST(SYSTEM_HYPH)
+
+dnl ===================================================================
+dnl Checking for mythes
+dnl ===================================================================
+AC_MSG_CHECKING([which mythes to use])
+if test -n "$with_system_mythes" && test "$with_system_mythes" != "no"; then
+    AC_MSG_RESULT([external])
+    SYSTEM_MYTHES=YES
+    AC_CHECK_HEADER(mythes.hxx, [],
+       [ AC_MSG_ERROR(mythes.hxx headers not found.)], [])
+    AC_CHECK_LIB(mythes, main, [],
+        [ AC_MSG_ERROR(mythes library not found.)], [])
+else
+    AC_MSG_RESULT([internal])
+    SYSTEM_MYTHES=NO
+fi
+AC_SUBST(SYSTEM_MYTHES)
 
 dnl ***************************************
 dnl testing libc version for Linux...
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.60
retrieving revision 1.55.4.2
diff -u -u -r1.60 -r1.55.4.2
--- config_office/set_soenv.in	11 May 2005 11:30:25 -0000	1.60
+++ config_office/set_soenv.in	15 May 2005 14:13:12 -0000	1.55.4.2
@@ -1738,6 +1738,13 @@
 ToFile( "BUILD_DMAKE",       "@BUILD_DMAKE@",      "e" );
 ToFile( "USE_XINERAMA",      "@USE_XINERAMA@",     "e" );
 ToFile( "XINERAMA_LINK",     "@XINERAMA_LINK@",    "e" );
+ToFile( "SYSTEM_MYSPELL",    "@SYSTEM_MYSPELL@",   "e" );
+ToFile( "MYSPELL_CFLAGS",    "@MYSPELL_CFLAGS@",   "e" );
+ToFile( "MYSPELL_LIBS",      "@MYSPELL_LIBS@",     "e" );
+ToFile( "SYSTEM_HYPH",       "@SYSTEM_HYPH@",      "e" );
+ToFile( "SYSTEM_MYTHES",     "@SYSTEM_MYTHES@",    "e" );
+ToFile( "PAM_LINK",          "@PAM_LINK@",         "e" );
+ToFile( "CRYPT_LINK",        "@CRYPT_LINK@",       "e" );
 ToFile( "GXX_INCLUDE_PATH",  "@GXX_INCLUDE_PATH@", "e" );
 ToFile( "COMMON_BUILD_TOOLS",$COMMON_BUILD_TOOLS,  "e" );
 if ($platform !~ m/cygwin/)
Index: external/gcc3_specific/makefile.mk
===================================================================
RCS file: /cvs/util/external/gcc3_specific/makefile.mk,v
retrieving revision 1.3
retrieving revision 1.3.68.1
diff -u -u -r1.3 -r1.3.68.1
--- external/gcc3_specific/makefile.mk	18 Jun 2003 12:44:10 -0000	1.3
+++ external/gcc3_specific/makefile.mk	1 Apr 2005 10:21:33 -0000	1.3.68.1
@@ -12,7 +12,7 @@
 .IF "$(OS)"!="MACOSX"
 .IF "$(OS)"!="IRIX"
 .IF "$(OS)"!="FREEBSD"
-
+.IF "$(SYSTEM_STDLIBS)"!="YES"
 
 all : $(SOLARLIBDIR)$/libstdc++.so.$(SHORTSTDCPP3) $(SOLARLIBDIR)$/libgcc_s.so.1 $(SOLARLIBDIR)$/libgcc_s.so 
 
@@ -27,6 +27,7 @@
         -rm -f $@
         +ln -s libgcc_s.so.1 $@
 
+.ENDIF
 .ENDIF
 .ENDIF 
 .ENDIF 
Index: lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk,v
retrieving revision 1.7
retrieving revision 1.7.14.1
diff -u -u -r1.7 -r1.7.14.1
--- lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	9 Mar 2004 12:39:27 -0000	1.7
+++ lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	1 Apr 2005 10:21:34 -0000	1.7.14.1
@@ -152,8 +152,14 @@
 		$(UCBHELPERLIB)	\
 		$(UNOTOOLSLIB)	\
 		$(LNGLIB) \
-		$(HNJLIB) \
-                $(ULINGULIB) 
+		$(HNJLIB)
+
+.IF "$(SYSTEM_MYSPELL)" != "YES"
+SHL1STDLIBS+=   $(ULINGULIB)
+.ELSE
+SHL1STDLIBS+=   $(MYSPELL_LIBS)
+.ENDIF
+
 
 
 # build DLL
Index: lingucomponent/source/hyphenator/altlinuxhyph/libhnj/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/libhnj/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.74.1
diff -u -u -r1.4 -r1.4.74.1
--- lingucomponent/source/hyphenator/altlinuxhyph/libhnj/makefile.mk	26 Mar 2003 13:02:19 -0000	1.4
+++ lingucomponent/source/hyphenator/altlinuxhyph/libhnj/makefile.mk	18 May 2005 00:09:49 -0000	1.4.74.1
@@ -49,6 +49,11 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(SYSTEM_HYPH)" == "YES"
+@all:
+	@echo "Using system altlinuxhyph..."
+.ENDIF
+
 all_target: ALLTAR
 
 
Index: lingucomponent/source/lingutil/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/lingutil/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.14.1
diff -u -u -r1.4 -r1.4.14.1
--- lingucomponent/source/lingutil/makefile.mk	9 Mar 2004 12:40:32 -0000	1.4
+++ lingucomponent/source/lingutil/makefile.mk	1 Apr 2005 10:21:35 -0000	1.4.14.1
@@ -70,6 +70,10 @@
 
 .INCLUDE : settings.mk
 
+.IF "$(SYSTEM_MYSPELL)" == "YES"
+@all:
+	@echo "Nothing to do here; using system myspell..."
+.ENDIF
 
 SLOFILES=	\
 		$(SLO)$/dictmgr.obj
Index: lingucomponent/source/spellcheck/myspell/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/spellcheck/myspell/makefile.mk,v
retrieving revision 1.8
retrieving revision 1.8.14.1
diff -u -u -r1.8 -r1.8.14.1
--- lingucomponent/source/spellcheck/myspell/makefile.mk	9 Mar 2004 12:42:12 -0000	1.8
+++ lingucomponent/source/spellcheck/myspell/makefile.mk	1 Apr 2005 10:21:36 -0000	1.8.14.1
@@ -71,6 +71,11 @@
 .INCLUDE : settings.mk
 
 # --- Files --------------------------------------------------------
+
+.IF "$(SYSTEM_MYSPELL)" == "YES"
+@all:
+	@echo "Using system myspell..."
+.ENDIF
 
 # all_target: ALLTAR DICTIONARY
 all_target: ALLTAR
Index: lingucomponent/source/spellcheck/spell/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/spellcheck/spell/makefile.mk,v
retrieving revision 1.7
retrieving revision 1.7.14.1
diff -u -u -r1.7 -r1.7.14.1
--- lingucomponent/source/spellcheck/spell/makefile.mk	9 Mar 2004 12:42:40 -0000	1.7
+++ lingucomponent/source/spellcheck/spell/makefile.mk	1 Apr 2005 10:21:37 -0000	1.7.14.1
@@ -69,7 +69,7 @@
 
 .IF "$(MYSPELLLIB)"==""
 .IF "$(GUI)"=="UNX"
-MYSPELLLIB=-lmyspell
+MYSPELLLIB=$(MYSPELL_LIBS)
 .ENDIF # unx
 .IF "$(GUI)"=="WNT"
 MYSPELLLIB=myspell.lib
@@ -125,9 +125,15 @@
 	com.sun.star.linguistic2.XThesaurus
 
 
-CXXFLAGS +=  -I..$/myspell -I..$/..$/lingutil
+.IF "$(SYSTEM_MYSPELL)" != "YES"
+CXXFLAGS += -I..$/myspell -I..$/..$/lingutil
 CFLAGSCXX += -I..$/myspell -I..$/..$/lingutil
-CFLAGSCC +=  -I..$/myspell -I..$/..$/lingutil
+CFLAGSCC += -I..$/myspell  -I..$/..$/lingutil
+.ELSE
+CXXGLAGS += $(MYSPELL_CFLAGS)
+CFLAGSCXX += $(MYSPELL_CFLAGS)
+CFLAGSCC += $(MYSPELL_CFLAGS)
+.ENDIF
 
 .IF "$(header)" == ""
 
@@ -157,9 +163,11 @@
 		$(UCBHELPERLIB)	\
 		$(UNOTOOLSLIB)	\
 		$(LNGLIB) \
-                $(MYSPELLLIB) \
-                $(ULINGULIB) 
+                $(MYSPELLLIB)
 
+.IF "$(SYSTEM_MYSPELL)" != "YES"
+SHL1STDLIBS+=   $(ULINGULIB)
+.ENDIF
 
 # build DLL
 SHL1LIBS=		$(SLB)$/$(TARGET).lib
Index: lingucomponent/source/thesaurus/libnth/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/thesaurus/libnth/makefile.mk,v
retrieving revision 1.3
retrieving revision 1.3.14.1
diff -u -u -r1.3 -r1.3.14.1
--- lingucomponent/source/thesaurus/libnth/makefile.mk	9 Mar 2004 12:42:53 -0000	1.3
+++ lingucomponent/source/thesaurus/libnth/makefile.mk	1 Apr 2005 10:21:37 -0000	1.3.14.1
@@ -124,10 +124,26 @@
 	com.sun.star.linguistic2.XThesaurus
 
 
-
+.IF "$(SYSTEM_MYSPELL)" == "YES" && "$(SYSTEM_MYTHES)" == "YES"
+CXXFLAGS += $(MYSPELL_CFLAGS)
+CFLAGSCXX += $(MYSPELL_CFLAGS)
+CFLAGSCC += $(MYSPELL_CFLAGS)
+.ENDIF
+.IF "$(SYSTEM_MYSPELL)" == "YES" && "$(SYSTEM_MYTHES)" != "YES"
+CXXFLAGS += -I..$/mythes $(MYSPELL_CFLAGS)
+CFLAGSCXX += -I..$/mythes $(MYSPELL_CFLAGS)
+CFLAGSCC += -I..$/mythes $(MYSPELL_CFLAGS)
+.ENDIF
+.IF "$(SYSTEM_MYPSPELL)" != "YES" && "$(SYSTEM_MYTHES)" == "YES"
+CXXFLAGS += -I..$/..$/lingutil
+CFLAGSCXX += -I..$/..$/lingutil
+CFLAGSCC += -I..$/..$/lingutil
+.ENDIF
+.IF "$(SYSTEM_MYSPELL)" != "YES" && "$(SYSTEM_MYTHES)" != "YES"
 CXXFLAGS += -I..$/mythes -I..$/..$/lingutil
 CFLAGSCXX += -I..$/mythes -I..$/..$/lingutil
 CFLAGSCC += -I..$/mythes -I..$/..$/lingutil
+.ENDIF
 
 .IF "$(header)" == ""
 
@@ -158,9 +174,13 @@
 		$(UCBHELPERLIB)	\
 		$(UNOTOOLSLIB)	\
 		$(LNGLIB) \
-                $(MYTHESLIB) \
-		$(ULINGULIB)
+                $(MYTHESLIB)
 
+.IF "$(SYSTEM_MYSPELL)" != "YES"
+SHL1STDLIBS+=   $(ULINGULIB)
+.ELSE
+SHL1STDLIBS+=   $(MYSPELL_LIBS)
+.ENDIF
 # build DLL
 SHL1LIBS=		$(SLB)$/$(TARGET).lib
 SHL1IMPLIB=		i$(TARGET)
Index: lingucomponent/source/thesaurus/mythes/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/thesaurus/mythes/makefile.mk,v
retrieving revision 1.2
retrieving revision 1.2.18.1
diff -u -u -r1.2 -r1.2.18.1
--- lingucomponent/source/thesaurus/mythes/makefile.mk	4 Feb 2004 13:08:28 -0000	1.2
+++ lingucomponent/source/thesaurus/mythes/makefile.mk	1 Apr 2005 10:21:38 -0000	1.2.18.1
@@ -71,6 +71,11 @@
 .INCLUDE : settings.mk
 
 # --- Files --------------------------------------------------------
+
+.IF "$(SYSTEM_MYTHES)" == "YES"
+@all:
+	@echo "Using system mythes..."
+.ENDIF
 
 all_target: ALLTAR
 
Index: sal/osl/unx/makefile.mk
===================================================================
RCS file: /cvs/porting/sal/osl/unx/makefile.mk,v
retrieving revision 1.25
retrieving revision 1.25.146.1
diff -u -u -r1.25 -r1.25.146.1
--- sal/osl/unx/makefile.mk	4 Feb 2004 15:30:47 -0000	1.25
+++ sal/osl/unx/makefile.mk	1 Apr 2005 10:21:39 -0000	1.25.146.1
@@ -146,6 +146,15 @@
 
 .IF "$(COM)"=="C50"
 APP1STDLIBS+=-lC
+.ENDIF
+
+.IF "$(LINUX)" == "YES"
+.IF "$(PAM_LINK)" == "YES"
+CFLAGS+=-DPAM_LINK
+.ENDIF
+.IF "$(CRYPT_LINK)" == "YES"
+CFLAGS+=-DCRYPT_LINK
+.ENDIF
 .ENDIF
 
 .INCLUDE :  target.mk
Index: sal/osl/unx/security.c
===================================================================
RCS file: /cvs/porting/sal/osl/unx/security.c,v
retrieving revision 1.13
retrieving revision 1.13.14.1
diff -u -u -r1.13 -r1.13.14.1
--- sal/osl/unx/security.c	21 Feb 2005 17:11:15 -0000	1.13
+++ sal/osl/unx/security.c	1 Apr 2005 10:21:39 -0000	1.13.14.1
@@ -233,6 +233,7 @@
 	return PAM_SUCCESS;
 }
 
+#ifndef PAM_LINK
 /*
  * avoid linking against libpam.so, since it is not available on all systems,
  * instead load-on-call, returns structure which holds pointer to 
@@ -284,6 +285,7 @@
 
 	return pam_module;	
 }
+#endif
 
 /*
  * User Identification using PAM
@@ -294,11 +296,13 @@
 {
 	sal_Bool success = sal_False;
 
+#ifndef PAM_LINK
 	sal_PamModule* pam_module;
 
 	pam_module = osl_getPAM();
 	if ( pam_module != NULL )
 	{
+#endif
 		pam_handle_t   *pam_handle = NULL;
 		struct pam_conv pam_conversation;
 		sal_PamData     pam_data;
@@ -311,21 +315,38 @@
 		pam_conversation.conv 		 = osl_PamConversation;
 		pam_conversation.appdata_ptr = (void*)(&pam_data); 
 
+#ifndef PAM_LINK
   		return_value = pam_module->pam_start( "su", name, 
 			&pam_conversation, &pam_handle);
+#else
+  		return_value = pam_start( "su", name, 
+			&pam_conversation, &pam_handle);
+#endif
 		if (return_value == PAM_SUCCESS ) 
+#ifndef PAM_LINK
 			return_value = pam_module->pam_authenticate(pam_handle, 0);
+#else
+			return_value = pam_authenticate(pam_handle, 0);
+#endif
   		if (return_value == PAM_SUCCESS ) 
+#ifndef PAM_LINK
 			return_value = pam_module->pam_acct_mgmt(pam_handle, 0);
 		pam_module->pam_end( pam_handle, return_value );
+#else
+			return_value = pam_acct_mgmt(pam_handle, 0);
+		pam_end( pam_handle, return_value );
+#endif
 	
 		success = (sal_Bool)(return_value == PAM_SUCCESS);
+#ifndef PAM_LINK
 	}
+#endif
 
   	return success;
 }
 
 
+#ifndef CRYPT_LINK
 /* dummy crypt, matches the interface of 
    crypt() but does not encrypt at all */
 static const sal_Char* SAL_CALL
@@ -370,6 +391,7 @@
 
 	return dynamic_crypt( (sal_Char*)key, (sal_Char*)salt );
 }
+#endif
 
 /*
  * compare an encrypted and an unencrypted password for equality
@@ -393,7 +415,11 @@
 	
 	pthread_mutex_lock(&crypt_mutex);
 
+#ifndef CRYPT_LINK
 	encrypted_plain = (sal_Char *)osl_dynamicCrypt( pPlainPassword, salt );
+#else
+	encrypted_plain = (sal_Char *)crypt( pPlainPassword, salt );
+#endif
 	success = (sal_Bool) (strcmp(pEncryptedPassword, encrypted_plain) == 0);
 	
 	pthread_mutex_unlock(&crypt_mutex);
Index: sal/util/makefile.mk
===================================================================
RCS file: /cvs/porting/sal/util/makefile.mk,v
retrieving revision 1.35
retrieving revision 1.35.54.1
diff -u -u -r1.35 -r1.35.54.1
--- sal/util/makefile.mk	3 Nov 2004 09:16:28 -0000	1.35
+++ sal/util/makefile.mk	1 Apr 2005 10:21:40 -0000	1.35.54.1
@@ -176,6 +176,15 @@
 
 .IF "$(OS)"=="MACOSX"
 SHL1STDLIBS+=-framework CoreFoundation
+.ENDIF
+
+.IF "$(OS)" == "LINUX"
+.IF "$(PAM_LINK)" == "YES"
+SHL1STDLIBS+=-lpam
+.ENDIF
+.IF "$(CRYPT_LINK)" == "YES"
+SHL1STDLIBS+=-lcrypt
+.ENDIF
 .ENDIF
 
 SHL1LIBS+=$(SLB)$/$(TARGET).lib
Index: vcl/unx/source/gdi/gcach_xpeer.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/gcach_xpeer.cxx,v
retrieving revision 1.37
retrieving revision 1.37.354.1
diff -u -u -r1.37 -r1.37.354.1
--- vcl/unx/source/gdi/gcach_xpeer.cxx	4 Feb 2004 11:05:23 -0000	1.37
+++ vcl/unx/source/gdi/gcach_xpeer.cxx	1 Apr 2005 10:21:41 -0000	1.37.354.1
@@ -127,6 +127,7 @@
     if( !XQueryExtension( mpDisplay, "RENDER", &nDummy, &nDummy, &nDummy ) )
         return;
 
+#ifndef XRENDER_LINK
     // we don't know if we are running on a system with xrender library
     // we don't want to install system libraries ourselves
     // => load them dynamically when they are there
@@ -204,24 +205,42 @@
     pFunc=osl_getSymbol(pRenderLib, freePicFuncName.pData);
     if( !pFunc ) return;
     pXRenderFreePicture             = (void(*)(Display*,Picture))pFunc;
+#endif
 
     // needed to initialize libXrender internals, we already know its there
+#ifdef XRENDER_LINK
+    XRenderQueryExtension( mpDisplay, &nDummy, &nDummy );
+#else
     (*pXRenderQueryExtension)( mpDisplay, &nDummy, &nDummy );
+#endif
 
     int nMajor, nMinor;
+#ifdef XRENDER_LINK
+    XRenderQueryVersion( mpDisplay, &nMajor, &nMinor );
+#else
     (*pXRenderQueryVersion)( mpDisplay, &nMajor, &nMinor );
+#endif
     nRenderVersion = 16*nMajor + nMinor;
     // TODO: enable/disable things depending on version
 
     // the 8bit alpha mask format must be there
     XRenderPictFormat aPictFormat={0,0,8,{0,0,0,0,0,0,0,0xFF},0};
+#ifdef XRENDER_LINK
+    mpGlyphFormat = XRenderFindFormat ( mpDisplay,
+        PictFormatAlphaMask|PictFormatDepth, &aPictFormat, 0 );
+#else
     mpGlyphFormat = (*pXRenderFindFormat)( mpDisplay,
         PictFormatAlphaMask|PictFormatDepth, &aPictFormat, 0 );
+#endif
 
     if( mpGlyphFormat != NULL )
     {
         // and the visual must be supported too
+#ifdef XRENDER_LINK
+      XRenderPictFormat* pVisualFormat = XRenderFindVisualFormat ( mpDisplay, _pVisual);
+#else
         XRenderPictFormat* pVisualFormat = (*pXRenderFindVisualFormat)( mpDisplay, _pVisual );
+#endif
         if( pVisualFormat != NULL )
             mbUsingXRender = true;
     }
@@ -250,7 +269,11 @@
             break;
 
         case XRENDER_KIND:
+#ifdef XRENDER_LINK
+	    XRenderFreeGlyphSet( mpDisplay,(GlyphSet)rServerFont.GetExtPointer() );
+#else
             (*pXRenderFreeGlyphSet)( mpDisplay,(GlyphSet)rServerFont.GetExtPointer() );
+#endif
             break;
     }
 
@@ -352,7 +375,11 @@
                 int nHeight = rServerFont.GetFontSelData().mnHeight;
                 if( nHeight<250 && rServerFont.GetAntialiasAdvice() )
                 {
+#ifdef XRENDER_LINK
+		    aGlyphSet = XRenderCreateGlyphSet ( mpDisplay, mpGlyphFormat );
+#else
                     aGlyphSet = (*pXRenderCreateGlyphSet)( mpDisplay, mpGlyphFormat );
+#endif
                     rServerFont.SetExtended( XRENDER_KIND, (void*)aGlyphSet );
                 }
                 else
@@ -516,8 +543,13 @@
 
             aGlyphId = nGlyphIndex & 0x00FFFFFF;
             const ULONG nBytes = maRawBitmap.mnScanlineSize * maRawBitmap.mnHeight;
+#ifdef XRENDER_LINK
+	    XRenderAddGlyphs ( mpDisplay, aGlyphSet, &aGlyphId, &aGlyphInfo, 1,
+                (char*)maRawBitmap.mpBits, nBytes );
+#else
             (*pXRenderAddGlyphs)( mpDisplay, aGlyphSet, &aGlyphId, &aGlyphInfo, 1,
                 (char*)maRawBitmap.mpBits, nBytes );
+#endif
             mnBytesUsed += nBytes;
         }
         else
Index: vcl/unx/source/gdi/gcach_xpeer.hxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/gcach_xpeer.hxx,v
retrieving revision 1.7
retrieving revision 1.7.628.1
diff -u -u -r1.7 -r1.7.628.1
--- vcl/unx/source/gdi/gcach_xpeer.hxx	28 Apr 2003 17:10:05 -0000	1.7
+++ vcl/unx/source/gdi/gcach_xpeer.hxx	1 Apr 2005 10:21:41 -0000	1.7.628.1
@@ -97,6 +97,7 @@
     bool                mbUsingXRender;
     XRenderPictFormat*  mpGlyphFormat;
 
+#ifndef XRENDER_LINK
 public:
     XRenderPictFormat*  (*pXRenderFindFormat)(Display*,unsigned long,XRenderPictFormat*,int);
     XRenderPictFormat*  (*pXRenderFindVisualFormat)(Display*,Visual*);
@@ -110,6 +111,7 @@
     Picture     (*pXRenderCreatePicture)(Display*,Drawable,XRenderPictFormat*,unsigned long,XRenderPictureAttributes*);
     void        (*pXRenderSetPictureClipRegion)(Display*,Picture,XLIB_Region);
     void        (*pXRenderFreePicture)(Display*,Picture);
+#endif
 };
 
 #endif // _SV_GCACH_XPEER_HXX
Index: vcl/unx/source/gdi/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/makefile.mk,v
retrieving revision 1.15
retrieving revision 1.15.88.1
diff -u -u -r1.15 -r1.15.88.1
--- vcl/unx/source/gdi/makefile.mk	13 Jan 2005 18:14:58 -0000	1.15
+++ vcl/unx/source/gdi/makefile.mk	1 Apr 2005 10:21:41 -0000	1.15.88.1
@@ -107,6 +107,10 @@
 .IF "$(OS)"=="SOLARIS"
 SLOFILES+=$(SLO)$/cdeint.obj
 ENVCFLAGS+=-DUSE_CDE
+.ENDIF
+
+.IF "$(XRENDER_LINK)" == "YES"
+CFLAGS+=-DXRENDER_LINK
 .ENDIF
 
 .IF "$(USE_BUILTIN_RASTERIZER)" != ""
Index: vcl/unx/source/gdi/salgdi3.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/salgdi3.cxx,v
retrieving revision 1.123
retrieving revision 1.123.94.1
diff -u -u -r1.123 -r1.123.94.1
--- vcl/unx/source/gdi/salgdi3.cxx	7 Jan 2005 09:26:25 -0000	1.123
+++ vcl/unx/source/gdi/salgdi3.cxx	1 Apr 2005 10:21:42 -0000	1.123.94.1
@@ -756,7 +756,11 @@
 {
     Display* pDisplay = GetXDisplay();
     Visual* pVisual = GetDisplay()->GetVisual()->GetVisual();
+#ifdef XRENDER_LINK
+    XRenderPictFormat* pVisualFormat = XRenderFindVisualFormat ( pDisplay, pVisual );
+#else
     XRenderPictFormat* pVisualFormat = (*aX11GlyphPeer.pXRenderFindVisualFormat)( pDisplay, pVisual );
+#endif
 
     // create xrender Picture for font foreground
     static Pixmap aPixmap;
@@ -772,7 +776,11 @@
 
         XRenderPictureAttributes aAttr;
         aAttr.repeat = true;
+#ifdef XRENDER_LINK
+        aSrc = XRenderCreatePicture ( pDisplay, aPixmap, pVisualFormat, CPRepeat, &aAttr );
+#else
         aSrc = (*aX11GlyphPeer.pXRenderCreatePicture)( pDisplay, aPixmap, pVisualFormat, CPRepeat, &aAttr );
+#endif
     }
 
     // set font foreground
@@ -786,10 +794,19 @@
 
     // notify xrender of target drawable
     XRenderPictureAttributes aAttr;
+#ifdef XRENDER_LINK
+    Picture aDst = XRenderCreatePicture ( pDisplay, hDrawable_, pVisualFormat, 0, &aAttr );
+#else
     Picture aDst = (*aX11GlyphPeer.pXRenderCreatePicture)( pDisplay, hDrawable_, pVisualFormat, 0, &aAttr );
+#endif
+
     // set clipping
     if( pClipRegion_ && !XEmptyRegion( pClipRegion_ ) )
+#ifdef XRENDER_LINK
+        XRenderSetPictureClipRegion( pDisplay, aDst, pClipRegion_ );
+#else
         (*aX11GlyphPeer.pXRenderSetPictureClipRegion)( pDisplay, aDst, pClipRegion_ );
+#endif
 
     ServerFont& rFont = rLayout.GetServerFont();
     GlyphSet aGlyphSet = aX11GlyphPeer.GetGlyphSet( rFont );
@@ -807,12 +824,21 @@
         unsigned int aRenderAry[ MAXGLYPHS ];
         for( int i = 0; i < nGlyphs; ++i )
              aRenderAry[ i ] = aX11GlyphPeer.GetGlyphId( rFont, aGlyphAry[i] );
+#ifdef XRENDER_LINK
+        XRenderCompositeString32 ( pDisplay, PictOpOver,
+            aSrc, aDst, 0, aGlyphSet, 0, 0, aPos.X(), aPos.Y(), aRenderAry, nGlyphs );
+#else
         (*aX11GlyphPeer.pXRenderCompositeString32)( pDisplay, PictOpOver,
             aSrc, aDst, 0, aGlyphSet, 0, 0, aPos.X(), aPos.Y(), aRenderAry, nGlyphs );
+#endif
     }
 
     // cleanup
+#ifdef XRENDER_LINK
+    XRenderFreePicture ( pDisplay, aDst );
+#else
     (*aX11GlyphPeer.pXRenderFreePicture)( pDisplay, aDst );
+#endif
 }
 
 //--------------------------------------------------------------------------
Index: vcl/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.66
retrieving revision 1.65.88.2
diff -u -u -r1.66 -r1.65.88.2
--- vcl/util/makefile.mk	22 Apr 2005 11:41:26 -0000	1.66
+++ vcl/util/makefile.mk	15 May 2005 14:19:19 -0000	1.65.88.2
@@ -311,6 +311,10 @@
 .ENDIF # OS == MACOSX
 
 SHL2STDLIBS += $(XINERAMALIBS)
+
+.IF "$(XRENDER_LINK)" == "YES"
+SHL2STDLIBS+=`pkg-config --libs xrender`
+.ENDIF
 
 .ENDIF # USE_XINERAMA != no
 
