Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.232
retrieving revision 1.221.4.23
diff -u -u -r1.232 -r1.221.4.23
--- config_office/configure.in	26 Nov 2007 18:03:39 -0000	1.232
+++ config_office/configure.in	4 Dec 2007 23:46:26 -0000	1.221.4.23
@@ -197,6 +197,11 @@
 [  --enable-cairo          Determines whether to use Cairo canvas backend on
                           platforms where Cairo is available.
 ],,enable_cairo=no)
+AC_ARG_ENABLE(opengl,
+[  --enable-opengl         Determines whether to use OpenGL (for 3D transitions)
+                          This is built as an UNO extension and not deployed
+			  in the install per default.
+],,enable_opengl=no)
 AC_ARG_ENABLE(dbus,
 [  --enable-dbus           Determines whether to enable presentation mode 
                           screensaver control under GNOME via DBUS 
@@ -319,6 +324,58 @@
 [  --with-beanshell-jar=JARFILE   Specify path to jarfile manually ],
 [ BSH_JAR="$withval"
 ])
+AC_ARG_ENABLE(sdext,
+[  --enable-sdext          enables the build of the Presentation Minimizer extension
+],,)
+AC_ARG_ENABLE(ogltrans,
+[  --enable-ogltrans       enables the build of the OGLTrans extension
+],,)
+AC_ARG_ENABLE(reportdesign,
+[  --disable-reportdesign  disables the build of the Report Builder extension
+],,)
+AC_ARG_WITH(system-jfreereport,
+[  --with-system-jfreereport      Use JFreeReport already on system
+],,)
+AC_ARG_WITH(jcommon-jar,
+[  --with-jcommon-jar=JARFILE   Specify path to jarfile manually ],
+[ JCOMMON_JAR="$withval"
+])
+AC_ARG_WITH(sac-jar,
+[  --with-sac-jar=JARFILE   Specify path to jarfile manually ],
+[ SAC_JAR="$withval"
+])
+AC_ARG_WITH(libxml-jar,
+[  --with-libxml-jar=JARFILE   Specify path to jarfile manually ],
+[ LIBXML_JAR="$withval"
+])
+AC_ARG_WITH(flute-jar,
+[  --with-flute-jar=JARFILE   Specify path to jarfile manually ],
+[ FLUTE_JAR="$withval"
+])
+AC_ARG_WITH(jfreereport-jar,
+[  --with-jfreereport-jar=JARFILE   Specify path to jarfile manually ],
+[ JFREEREPORT_JAR="$withval"
+])
+AC_ARG_WITH(liblayout-jar,
+[  --with-liblayout-jar=JARFILE   Specify path to jarfile manually ],
+[ LIBLAYOUT_JAR="$withval"
+])
+AC_ARG_WITH(libformula-jar,
+[  --with-libformula-jar=JARFILE   Specify path to jarfile manually ],
+[ LIBFORMULA_JAR="$withval"
+])
+AC_ARG_WITH(librepository-jar,
+[  --with-librepository-jar=JARFILE   Specify path to jarfile manually ],
+[ LIBREPOSITORY_JAR="$withval"
+])
+AC_ARG_WITH(libfonts-java-jar,
+[  --with-libfonts-jar=JARFILE   Specify path to jarfile manually ],
+[ LIBFONTS_JAR="$withval"
+])
+AC_ARG_WITH(jcommon-serializer-jar,
+[  --with-jcommon-serializer-jar=JARFILE   Specify path to jarfile manually ],
+[ JCOMMON_SERIALIZER_JAR="$withval"
+])
 AC_ARG_WITH(system-xml-apis,
 [  --with-system-xml-apis  Use xml-apis already on system
 ],,)
@@ -403,15 +460,15 @@
                           --with-system-mozilla=<flavour>. Supported are:
                           xulrunner (default), firefox, seamonkey, mozilla
 ], WITH_SYSTEM_MOZILLA=$withval, WITH_SYSTEM_MOZILLA=no)
-AC_ARG_WITH(stlport4,
-[  --with-stlport4         The location that STLport4 is installed in. The STL
+AC_ARG_WITH(stlport,
+[  --with-stlport         The location that STLport is installed in. The STL
                           header files are assumed to be in
-                          stlport4-home/stlport and the STLPort4 library in
-                          stlport4-home/lib.
+                          stlport-home/stlport and the STLPort library in
+                          stlport-home/lib.
 
-                          Usage: --with-stlport4=<absolute path to stlport4 home>
+                          Usage: --with-stlport=<absolute path to stlport home>
 
-                          Warning!!, --without-stlport4 is possible with
+                          Warning!!, --without-stlport is possible with
                           gcc >= 3.3.3, but will break ABI compatability
 ], WITH_STLPORT=$withval , WITH_STLPORT=yes)
 AC_ARG_WITH(jdk-home,
@@ -1152,15 +1209,6 @@
 AC_SUBST(USE_SHELL)
 AC_SUBST(WITH_MINGWIN)
 
-AC_SYS_LARGEFILE
-if test -n "$ac_cv_sys_file_offset_bits"; then
-   LFS_CFLAGS="-D_FILE_OFFSET_BITS=$ac_cv_sys_file_offset_bits"
-fi
-if test -n "$ac_cv_sys_large_files" && test "$ac_cv_sys_large_files" != "no"; then
-   LFS_CFLAGS="$LFS_CFLAGS -D_LARGE_FILES"
-fi
-AC_SUBST(LFS_CFLAGS)
-
 dnl ===================================================================
 dnl Extra check for Windows. Non 4nt-builds need gcc to build dmake
 dnl and g++ to build guw.exe although MS cl (or MinGW) is used to
@@ -1646,6 +1694,42 @@
 fi
 
 dnl ===================================================================
+dnl Find pre-processors.
+dnl ===================================================================
+if test "$_os" != "WINNT" -o "$WITH_MINGWIN" = "yes"; then
+   AC_PROG_CXXCPP
+
+dnl Check whether there's a C pre-processor.
+dnl ===================================================================
+dnl When using SunStudio compiler, there is a bug with the cc
+dnl preprocessor, so use CC preprocessor as the cc preprocessor
+dnl See Issuezilla #445.
+dnl ===================================================================
+   if test "$_os" = "SunOS"; then
+      CPP=$CXXCPP
+   else
+      AC_PROG_CPP
+   fi
+fi
+
+AC_CHECK_SIZEOF(long)
+SIZEOF_LONG=$ac_cv_sizeof_long
+AC_SUBST(SIZEOF_LONG)
+AC_C_BIGENDIAN
+WORDS_BIGENDIAN=$ac_cv_c_bigendian
+AC_SUBST(WORDS_BIGENDIAN)
+
+dnl Check for large file support
+AC_SYS_LARGEFILE
+if test -n "$ac_cv_sys_file_offset_bits"; then
+   LFS_CFLAGS="-D_FILE_OFFSET_BITS=$ac_cv_sys_file_offset_bits"
+fi
+if test -n "$ac_cv_sys_large_files" && test "$ac_cv_sys_large_files" != "no"; then
+   LFS_CFLAGS="$LFS_CFLAGS -D_LARGE_FILES"
+fi
+AC_SUBST(LFS_CFLAGS)
+
+dnl ===================================================================
 dnl Check if we are to enable vba macro interoperability feature
 dnl ===================================================================
 AC_MSG_CHECKING([whether to disable vba feature])
@@ -1706,7 +1790,7 @@
     AC_CHECK_HEADER(security/pam_appl.h, [],
                     [AC_MSG_ERROR([pam_appl.h could not be found. libpam-dev or pam-devel missing?])], [])
     AC_MSG_CHECKING([whether to link to libpam])
-    if test -n "$enable_pam_link"; then
+    if test -n "$enable_pam_link" -a "$enable_pam_link" != "no"; then
       AC_MSG_RESULT([yes])
       PAM_LINK=YES
       AC_CHECK_LIB(pam, pam_start, [],
@@ -1729,7 +1813,7 @@
 
 if test "$_os" = "Linux"; then
     AC_MSG_CHECKING([whether to link to libcrypt])
-    if test -n "$enable_crypt_link"; then
+    if test -n "$enable_crypt_link" -a "$enable_crypt_link" != "no"; then
         AC_MSG_RESULT([yes])
         CRYPT_LINK=YES
         AC_CHECK_LIB(crypt, crypt, [],
@@ -1925,8 +2009,7 @@
 dnl Testing for exception handling - dwarf2 or sjlj exceptions...
 dnl *************************************************************
 AC_MSG_CHECKING([exception type])
-AC_LANG_SAVE
-AC_LANG_CPLUSPLUS
+AC_LANG_PUSH([C++])
 AC_CACHE_VAL(exceptions_type, [AC_TRY_LINK(
 	[#include <iostream>
 
@@ -1939,7 +2022,7 @@
 ])
 
 AC_MSG_RESULT($exceptions_type)
-AC_LANG_RESTORE
+AC_LANG_POP([C++])
 EXCEPTIONS="$exceptions_type"
 AC_SUBST(EXCEPTIONS)
 
@@ -2045,16 +2128,6 @@
 fi
 
 dnl ===================================================================
-dnl When using SunStudio compiler, there is a bug with the cc
-dnl preprocessor, so use CC preprocessor for the following tests.
-dnl See Issuezilla #445.
-dnl ===================================================================
-if test "$_os" = "SunOS"; then
-AC_LANG_CPLUSPLUS
-AC_TRY_CPP("", , )
-fi
-
-dnl ===================================================================
 dnl Checks for SGI STL
 dnl ===================================================================
 if test -n "$enable_sgistl" && "$enable_sgistl" != "no"; then
@@ -2072,9 +2145,9 @@
 else
 
 dnl ===================================================================
-dnl Checks for STLPORT4
+dnl Checks for STLPORT
 dnl ===================================================================
-   AC_MSG_CHECKING([for STLport4 headers])
+   AC_MSG_CHECKING([for STLport headers])
    STLPORT4=""
    USE_SYSTEM_STL=""
    if test "$WITH_STLPORT" = "yes"; then
@@ -2085,28 +2158,34 @@
    else
       STLPORT4=$WITH_STLPORT
       if test "$_os" != "WINNT" -o "$WITH_MINGWIN" = "yes"; then
-         AC_TRY_CPP($STLPORT4/stlport/hash_map, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([STLport4 headers not found.]))
+         AC_TRY_CPP($STLPORT4/stlport/hash_map, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([STLport headers not found.]))
       else
          dnl AC_TRY_CPP doesn't work for MSVC because C++ preprocessor is not found by autoconf.
          if test -f "$STLPORT4/stlport/hash_map"; then
             AC_MSG_RESULT([checked.])
          else
-            AC_MSG_ERROR([STLport4 headers not found.])
+            AC_MSG_ERROR([STLport headers not found.])
          fi
       fi
       if test "$_os" != "WINNT" -o "$WITH_MINGWIN" = "yes"; then
-         AC_MSG_CHECKING([for STLport4 libraries])
+         AC_MSG_CHECKING([for STLport libraries])
          if test "$_os" = "SunOS"; then
 		      if test -f "$STLPORT4/lib/libstlport_sunpro.so"; then
 			      AC_MSG_RESULT([checked])
+		      elif test -f "$STLPORT4/lib/libstlport.so"; then
+			      AC_MSG_RESULT([checked])
+			      STLPORT_VER=500
 		      else
-			      AC_MSG_ERROR([STLport4 libraries not found])
+			      AC_MSG_ERROR([STLport libraries not found])
 		      fi
 	      elif test "$_os" = "Darwin"; then
 		      if test -f "$STLPORT4/lib/libstlport_gcc.dylib"; then
 			      AC_MSG_RESULT([checked])
+		      elif test -f "$STLPORT4/lib/libstlport.dylib"; then
+			      AC_MSG_RESULT([checked])
+			      STLPORT_VER=500
 		      else
-			      AC_MSG_ERROR([STLport4 libraries not found])
+			      AC_MSG_ERROR([STLport libraries not found])
 		      fi
 	      elif test "$_os" = "IRIX" -o "$_os" = "IRIX64"; then
 		      if test -f "$STLPORT4/lib/libstlport_mipspro_41.so"; then
@@ -2114,15 +2193,21 @@
 			   else
 				   if test -f "$STLPORT4/lib/libstlport_gcc.so"; then
 					   AC_MSG_RESULT([checked])
+				   elif test -f "$STLPORT4/lib/libstlport.so"; then
+					   AC_MSG_RESULT([checked])
+					   STLPORT_VER=500
 				   else
-					   AC_MSG_ERROR([STLport4 libraries not found])
+					   AC_MSG_ERROR([STLport libraries not found])
 				   fi
 			   fi
 		   else
 			   if test -f "$STLPORT4/lib/libstlport_gcc.so"; then
 				   AC_MSG_RESULT([checked])
+			   elif test -f "$STLPORT4/lib/libstlport.so"; then
+				   AC_MSG_RESULT([checked])
+				   STLPORT_VER=500
 			   else
-				   AC_MSG_ERROR([STLport4 libraries not found])
+				   AC_MSG_ERROR([STLport libraries not found])
 			   fi
 		   fi
 	   fi
@@ -2132,7 +2217,11 @@
 if test -z "$STLPORT4"; then
   STLPORT4="NO_STLPORT4"
 fi
+if test -z "$STLPORT_VER"; then
+  STLPORT_VER=400
+fi
 AC_SUBST(STLPORT4)
+AC_SUBST(STLPORT_VER)
 AC_SUBST(USE_SYSTEM_STL)
 
 dnl ===================================================================
@@ -2163,13 +2252,12 @@
 dnl ===================================================================
 if test "$USE_SYSTEM_STL" = "YES"; then
    AC_MSG_CHECKING([if hash_map will be in __gnu_cxx namespace])
-   AC_LANG_SAVE
-   AC_LANG_CPLUSPLUS
+   AC_LANG_PUSH([C++])
    AC_TRY_COMPILE([#include <ext/hash_map>
 using namespace __gnu_cxx;
 ],[hash_map<int, int> t; return 0;],
   ac_cv_cxx_have_ext_hash_map=yes, ac_cv_cxx_have_ext_hash_map=no)
-   AC_LANG_RESTORE
+   AC_LANG_POP([C++])
    if test "$ac_cv_cxx_have_ext_hash_map" = "no"; then
       AC_MSG_ERROR([Can't find hash_map. Try with stlport enabled])
    else
@@ -2623,17 +2711,10 @@
 dnl ===================================================================
 dnl Checks for specific files.
 dnl ===================================================================
+
 dnl ===================================================================
 dnl Checks for programs.
 dnl ===================================================================
-dnl Check whether there's a C pre-processor.
-if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" ; then
-   AC_PROG_CPP
-fi
-dnl Check whether there's a C++ pre-processor.
-if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" ; then
-   AC_PROG_CXXCPP
-fi
 
 dnl ===================================================================
 dnl Check whether we already have dmake
@@ -2935,8 +3016,7 @@
       if test "$MINGWSTRIP" = "false"; then
          AC_MSG_ERROR(MinGW32 binutils needed. Install them.)
       fi
-      AC_LANG_SAVE
-      AC_LANG_CPLUSPLUS
+      AC_LANG_PUSH([C++])
       save_CXX=$CXX
       save_CXXCPP=$CXXCPP
       CXX=$MINGWCXX
@@ -2957,7 +3037,7 @@
       CXX=$save_CXX
       CXXCPP=$save_CXXCPP
       LIBS=$save_LIBS
-      AC_LANG_RESTORE
+      AC_LANG_POP([C++])
       fi
    fi
    BUILD_TYPE="$BUILD_TYPE ODK"
@@ -3240,6 +3302,13 @@
    if test "z$HOME" = "z"; then
       export HOME="";
    fi
+   # bz2 tarball and bzip2 is not standard
+   if test -z "$BZIP2"; then
+	AC_PATH_PROG( BZIP2, bzip2)
+	if test -z "$BZIP2"; then
+   		AC_MSG_ERROR([the internal Python module has a .tar.bz2. You need bzip2])
+   	fi
+   fi
 fi
 AC_SUBST(SYSTEM_PYTHON)
 AC_SUBST(PYTHON_CFLAGS)
@@ -3263,7 +3332,7 @@
         ], []
     )
     AC_MSG_CHECKING([whether db is at least 4.1])
-    for v in 1 2 3 4 5; do
+    for v in 1 2 3 4 5 6; do
        AC_TRY_RUN([
 #include <db.h>
 
@@ -3515,13 +3584,12 @@
 if test -n "$with_system_boost" && test "$with_system_boost" != "no"; then
    AC_MSG_RESULT([external])
    SYSTEM_BOOST=YES
-   AC_LANG_SAVE
-   AC_LANG_CPLUSPLUS
+   AC_LANG_PUSH([C++])
    AC_CHECK_HEADER(boost/shared_ptr.hpp, [],
        [AC_MSG_ERROR(boost/shared_ptr.hpp not found. install boost)], [])
    AC_CHECK_HEADER(boost/spirit/core.hpp, [],
        [AC_MSG_ERROR(boost/spirit/core.hpp not found. install boost)], [])
-   AC_LANG_RESTORE
+   AC_LANG_POP([C++])
 else
    AC_MSG_RESULT([internal])
    BUILD_TYPE="$BUILD_TYPE BOOST"
@@ -3536,11 +3604,10 @@
 if test -n "$with_system_vigra" && test "$with_system_vigra" != "no"; then
    AC_MSG_RESULT([external])
    SYSTEM_VIGRA=YES
-   AC_LANG_SAVE
-   AC_LANG_CPLUSPLUS
+   AC_LANG_PUSH([C++])
    AC_CHECK_HEADER(vigra/copyimage.hxx, [],
        [AC_MSG_ERROR(vigra/copyimage.hxx not found. install vigra)], [])
-   AC_LANG_RESTORE
+   AC_LANG_POP([C++])
 else
    AC_MSG_RESULT([internal])
    BUILD_TYPE="$BUILD_TYPE VIGRA"
@@ -3578,7 +3645,7 @@
 AC_MSG_CHECKING([whether to build Mozilla addressbook connectivity])
 if test "$enable_mozilla" = "no"; then
    AC_MSG_RESULT([no])
-elif test -n "$with_system_mozilla"; then
+elif test "$with_system_mozilla" = "yes"; then
    AC_MSG_RESULT([no, not possible with system-mozilla])
 else
    AC_MSG_RESULT([yes])
@@ -3587,8 +3654,6 @@
 AC_MSG_CHECKING([whether to build XML Security support])
 if test "$enable_mozilla" = "no"; then
    AC_MSG_RESULT([no, since Mozilla (NSS) disabled but needed])
-elif test "$with_system_mozilla" = "yes"; then
-   AC_MSG_RESULT([yes])
 else
    AC_MSG_RESULT([yes])
 fi
@@ -3858,15 +3923,15 @@
    SYSTEM_ICU=YES
    AC_MSG_CHECKING([for unicode/rbbi.h])
    AC_TRY_CPP(unicode/rbbi.h, AC_MSG_RESULT([checked.]), AC_MSG_ERROR([icu headers not found.]))
-   AC_PATH_PROG(SYSTEM_GENBRK, genbrk)
+   AC_PATH_PROG(SYSTEM_GENBRK, genbrk, [], [$PATH:/usr/sbin:/sbin])
    if test -z "$SYSTEM_GENBRK"; then
       AC_MSG_ERROR([\"genbrk\" not found in \$PATH, install the icu development tool \"genbrk"\])
    fi
-   AC_PATH_PROG(SYSTEM_GENCCODE, genccode)
+   AC_PATH_PROG(SYSTEM_GENCCODE, genccode, [], [$PATH:/usr/sbin:/sbin])
    if test -z "$SYSTEM_GENCCODE"; then
       AC_MSG_ERROR([\"genccode\" not found in \$PATH, install the icu development tool \"genccode"\])
    fi
-   AC_PATH_PROG(SYSTEM_GENCMN, gencmn)
+   AC_PATH_PROG(SYSTEM_GENCMN, gencmn, [], [$PATH:/usr/sbin:/sbin])
    if test -z "$SYSTEM_GENCMN"; then
       AC_MSG_ERROR([\"gencmn\" not found in \$PATH, install the icu development tool \"gencmn"\])
    fi
@@ -3875,13 +3940,14 @@
 #include <unicode/uversion.h>
 
 int main(int argc, char **argv) {
-      if(U_ICU_VERSION_MAJOR_NUM == 3 && U_ICU_VERSION_MINOR_NUM == 6) return 0;
+      if(U_ICU_VERSION_MAJOR_NUM == 3 && (U_ICU_VERSION_MINOR_NUM == 6 ||
+      	U_ICU_VERSION_MINOR_NUM == 8)) return 0;
       else return 1;
 }
       ], [AC_MSG_RESULT(OK)], [AC_MSG_RESULT([not suitable, only 3.6 supported currently])])
      AC_MSG_CHECKING(whether setBreakType is public)
-     save_LDFLAGS="$LDFLAGS"
-     LDFLAGS="$LDFLAGS -licuuc"
+     save_LIBS="$LIBS"
+     LIBS="$LIBS -licuuc"
 	   AC_TRY_LINK([#include <unicode/rbbi.h>], [
 	   RuleBasedBreakIterator *rbi = NULL;
 	   rbi = new RuleBasedBreakIterator();
@@ -3892,7 +3958,7 @@
       else
         AC_MSG_RESULT(yes)
       fi
-      LDFLAGS="$save_LDFLAGS"
+      LIBS="$save_LIBS"
 else
     AC_MSG_RESULT([internal])
     SYSTEM_ICU=NO
@@ -4021,7 +4087,7 @@
 dnl Check for system Xrender
 dnl ===================================================================
 AC_MSG_CHECKING([whether to link to Xrender])
-if test -n "$enable_xrender_link"; then
+if test -n "$enable_xrender_link" -a "$enable_xrender_link" != "no"; then
    AC_MSG_RESULT([yes])
    XRENDER_LINK=YES
    with_system_xrender_headers=yes
@@ -4151,7 +4217,7 @@
 dnl ===================================================================
 dnl Check for building neon
 dnl ===================================================================
-AC_MSG_CHECKING([whether to build neon])
+AC_MSG_CHECKING([whether to use neon])
 if test "$enable_neon" = "no"; then
     AC_MSG_RESULT([no])
     DISABLE_NEON=TRUE
@@ -4240,7 +4334,7 @@
 if test -n "$with_system_hunspell" && test "$with_system_hunspell" != "no"; then
     AC_MSG_RESULT([external])
     SYSTEM_HUNSPELL=YES
-    AC_LANG(C++)
+    AC_LANG_PUSH([C++])
     PKG_CHECK_MODULES(HUNSPELL, hunspell, HUNSPELL_PC="TRUE", HUNSPELL_PC="" )
     if test "$HUNSPELL_PC" != "TRUE"; then
       AC_CHECK_HEADER(hunspell.hxx, [],
@@ -4252,6 +4346,7 @@
            [ AC_MSG_ERROR(hunspell library not found.) ], [])
       HUNSPELL_LIBS=-lhunspell
      fi
+    AC_LANG_POP([C++])
 else
     AC_MSG_RESULT([internal])
     SYSTEM_HUNSPELL=NO
@@ -4272,13 +4367,22 @@
     AC_CHECK_MEMBER(struct _HyphenDict.cset, [],
        [ AC_MSG_ERROR(no. You are sure you have altlinuyhyph headers?)],
        [#include <hyphen.h>])
-    AC_CHECK_LIB(hyph, hnj_hyphen_hyphenate2, [],
+    AC_CHECK_LIB(hyphen, hnj_hyphen_hyphenate2, [HYPHEN_LIB=-lhyphen],
         [ AC_MSG_ERROR(altlinuxhyph library not found or too old.)], [])
+    if test -z "$HYPHEN_LIB"; then
+       AC_CHECK_LIB(hyph, hnj_hyphen_hyphenate2, [HYPHEN_LIB=-lhyph],
+           [ AC_MSG_ERROR(altlinuxhyph library not found or too old.)], [])
+    fi
+    if test -z "$HYPHEN_LIB"; then
+       AC_CHECK_LIB(hnj, hnj_hyphen_hyphenate2, [HYPHEN_LIB=-lhnj],
+           [ AC_MSG_ERROR(altlinuxhyph library not found or too old.)], [])
+    fi
 else
     AC_MSG_RESULT([internal])
     SYSTEM_HYPH=NO
 fi
 AC_SUBST(SYSTEM_HYPH)
+AC_SUBST(HYPHEN_LIB)
 
 dnl ===================================================================
 dnl Checking for mythes
@@ -4906,6 +5010,124 @@
 AC_SUBST(CAIRO_LIBS)
 
 dnl ===================================================================
+dnl Check whether the OpenGL libraries are available
+dnl ===================================================================
+
+AC_MSG_CHECKING([whether to build the OGLTrans extension])
+ENABLE_OPENGL=
+
+if test "x$enable_opengl" != "xno" ; then
+   AC_MSG_RESULT([yes])
+   AC_CHECK_HEADER(GL/gl.h, [],
+                   [AC_MSG_ERROR([OpenGL headers not found])], [])
+   AC_CHECK_LIB(GL, main, [],
+     [AC_MSG_ERROR(libGL not installed or functional)], [])
+   AC_CHECK_LIB(GLU, main, [],
+     [AC_MSG_ERROR(libGLU not installed or functional)], [])
+   ENABLE_OPENGL=TRUE
+else
+   AC_MSG_RESULT([no])
+fi
+
+AC_SUBST(ENABLE_OPENGL)
+
+AC_MSG_CHECKING([whether to build the Presentation Minimizer extension])
+if test -n "$enable_sdext" -a "$enable_sdext" != "no"; then
+   AC_MSG_RESULT([yes])
+   BUILD_TYPE="$BUILD_TYPE SDEXT"
+else
+   AC_MSG_RESULT([no])
+fi
+
+AC_MSG_CHECKING([whether to build the Report Builder extension])
+if test -z "$enable_reportdesign" -a "$enable_reportdesign" != "no" && test "$WITH_JAVA" != "no"; then
+  AC_MSG_RESULT([yes])
+  ENABLE_REPORTDESIGN=YES
+  AC_MSG_CHECKING([which jfreereport libs to use])
+  if test "$with_system_jfreereport" == "yes"; then
+   	SYSTEM_JFREEREPORT=YES
+  	AC_MSG_RESULT([external])
+        if test -z $JCOMMON_JAR; then
+               JCOMMON_JAR=/usr/share/java/jcommon-1.0.10.jar
+        fi
+        AC_CHECK_FILE($JCOMMON_JAR, [],
+               [AC_MSG_ERROR(jcommon.jar not found.)], [])
+        if test -z $SAC_JAR; then
+               SAC_JAR=/usr/share/java/sac.jar
+        fi
+        AC_CHECK_FILE($SAC_JAR, [],
+               [AC_MSG_ERROR(sac.jar not found.)], [])
+        if test -z $LIBXML_JAR; then
+               LIBXML_JAR=/usr/share/java/libxml-0.9.5.jar
+        fi
+        AC_CHECK_FILE($LIBXML_JAR, [],
+               [AC_MSG_ERROR(libxml.jar not found.)], [])
+        if test -z $FLUTE_JAR; then
+               FLUTE_JAR=/usr/share/java/flute-1.3-jfree-20061107.jar
+        fi
+        AC_CHECK_FILE($FLUTE_JAR, [],
+               [AC_MSG_ERROR(flute-1.3-jfree.jar not found.)], [])
+        if test -z $JFREEREPORT_JAR; then
+               JFREEREPORT_JAR=/usr/share/java/jfreereport-0.9.0-05.jar
+        fi
+        AC_CHECK_FILE($JFREEREPORT_JAR, [],
+               [AC_MSG_ERROR(jfreereport.jar not found.)], [])
+        if test -z $LIBLAYOUT_JAR; then
+               LIBLAYOUT_JAR=/usr/share/java/liblayout-0.2.6.jar
+        fi
+        AC_CHECK_FILE($LIBLAYOUT_JAR, [],
+               [AC_MSG_ERROR(liblayout.jar not found.)], [])
+        if test -z $LIBLOADER_JAR; then
+               LIBLOADER_JAR=/usr/share/java/libloader-0.3.4.jar
+        fi
+        AC_CHECK_FILE($LIBLOADER_JAR, [],
+               [AC_MSG_ERROR(libloader.jar not found.)], [])
+        if test -z $LIBFORMULA_JAR; then
+               LIBFORMULA_JAR=/usr/share/java/libformula-0.1.8.jar
+        fi
+        AC_CHECK_FILE($LIBFORMULA_JAR, [],
+               [AC_MSG_ERROR(libformula.jar not found.)], [])
+        if test -z $LIBREPOSITORY_JAR; then
+               LIBREPOSITORY_JAR=/usr/share/java/librepository-0.1.1.jar
+        fi
+        AC_CHECK_FILE($LIBREPOSITORY_JAR, [],
+               [AC_MSG_ERROR(librepository.jar not found.)], [])
+        if test -z $LIBFONTS_JAR; then
+               LIBFONTS_JAR=/usr/share/java/libfonts-0.2.6.jar
+        fi
+        AC_CHECK_FILE($LIBFONTS_JAR, [],
+               [AC_MSG_ERROR(libfonts.jar not found.)], [])
+        if test -z $JCOMMON_SERIALIZER_JAR; then
+               JCOMMON_SERIALIZER_JAR=/usr/share/java/jcommon-serializer-0.1.0.jar
+        fi
+        AC_CHECK_FILE($JCOMMON_SERIALIZER_JAR, [],
+               [AC_MSG_ERROR(jcommon-serializer.jar not found.)], [])
+  else
+     AC_MSG_RESULT([internal])
+     SYSTEM_JFREEREPORT=NO
+     BUILD_TYPE="$BUILD_TYPE JFREEREPORT"
+  fi
+  BUILD_TYPE="$BUILD_TYPE REPORTDESIGN"
+else
+  AC_MSG_RESULT([no])
+  ENABLE_REPORTDESIGN=NO
+  SYSTEM_JFREEREPORT=NO
+fi
+AC_SUBST(ENABLE_REPORTDESIGN)
+AC_SUBST(SYSTEM_JFREEREPORT)
+AC_SUBST(JCOMMON_JAR)
+AC_SUBST(SAC_JAR)
+AC_SUBST(LIBXML_JAR)
+AC_SUBST(FLUTE_JAR)
+AC_SUBST(JFREEREPORT_JAR)
+AC_SUBST(LIBLAYOUT_JAR)
+AC_SUBST(LIBLOADER_JAR)
+AC_SUBST(LIBFORMULA_JAR)
+AC_SUBST(LIBREPOSITORY_JAR)
+AC_SUBST(LIBFONTS_JAR)
+AC_SUBST(JCOMMON_SERIALIZER_JAR)
+
+dnl ===================================================================
 dnl Check whether the Qt and KDE libraries are available.
 dnl ===================================================================
 
@@ -5076,8 +5298,7 @@
 AC_MSG_CHECKING([whether to enable KDE address book support])
 if test "$enable_kdeab" = "yes" && test "$enable_kde" = "yes"; then
    AC_MSG_RESULT([yes])
-   AC_LANG_SAVE
-   AC_LANG_CPLUSPLUS
+   AC_LANG_PUSH([C++])
    save_CXXFLAGS=$CXXFLAGS
    CXXFLAGS="$CXXFLAGS $KDE_CFLAGS"
    AC_MSG_CHECKING([whether KDE is between 3.2 and 3.6])
@@ -5090,7 +5311,7 @@
 }
        ], [AC_MSG_RESULT([yes])], [AC_MSG_ERROR([KDE version too old or too recent, please use another version of KDE or disable KDE address book support])])
    CXXFLAGS=$save_CXXFLAGS
-   AC_LANG_RESTORE
+   AC_LANG_POP([C++])
    ENABLE_KAB=TRUE
 else
    AC_MSG_RESULT([no])
@@ -5206,10 +5427,10 @@
    AC_MSG_RESULT([yes])
 elif test "$_os" = "Linux" -o "$_os" = "FreeBSD"; then
    if test -e "$XLIB/libXinerama.so" -a -e "$XLIB/libXinerama.a"; then
-      # we have both versions, let the user decide but use the static one
+      # we have both versions, let the user decide but use the dynamic one
       # per default
       USE_XINERAMA=YES
-      if test -n "$with_dynamic_xinerama" -o -n "$with_system_libs"; then
+      if test -z "$with_static_xinerama" -o -n "$with_system_libs"; then
          XINERAMA_LINK=dynamic
       else
          XINERAMA_LINK=static
@@ -5254,13 +5475,6 @@
 AC_SUBST(USE_XINERAMA)
 AC_SUBST(XINERAMA_LINK)
 
-AC_CHECK_SIZEOF(long)
-SIZEOF_LONG=$ac_cv_sizeof_long
-AC_SUBST(SIZEOF_LONG)
-AC_C_BIGENDIAN
-WORDS_BIGENDIAN=$ac_cv_c_bigendian
-AC_SUBST(WORDS_BIGENDIAN)
-
 dnl ===================================================================
 dnl Test for the presence of Ant and that it works
 dnl ===================================================================
Index: config_office/oowintool
===================================================================
RCS file: /cvs/tools/config_office/oowintool,v
retrieving revision 1.3
retrieving revision 1.3.60.1
diff -u -u -r1.3 -r1.3.60.1
--- config_office/oowintool	27 Jun 2007 17:51:45 -0000	1.3
+++ config_office/oowintool	22 Nov 2007 15:46:15 -0000	1.3.60.1
@@ -209,7 +209,9 @@
 
 sub print_jdk_dir()
 {
-    my $dir = reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.4/JavaHome");
+    my $dir =
+	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.5/JavaHome") ||
+	  reg_get_value ("HKEY_LOCAL_MACHINE/SOFTWARE/JavaSoft/Java\ Development\ Kit/1.4/JavaHome");
     print cygpath($dir, 'w', $output_format); 
 }
 
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.164
retrieving revision 1.157.6.8
diff -u -u -r1.164 -r1.157.6.8
--- config_office/set_soenv.in	26 Nov 2007 18:04:00 -0000	1.164
+++ config_office/set_soenv.in	2 Dec 2007 12:34:14 -0000	1.157.6.8
@@ -1563,6 +1563,7 @@
 ToFile( "ANT",               "@ANT@",            "e" );
 ToFile( "JDKLIB",            $JAVA_LIB,          "e" );
 ToFile( "STLPORT4",          $STLPORT4,          "e" );
+ToFile( "STLPORT_VER",       "@STLPORT_VER@",    "e" );
 ToFile( "ASM_PATH",          $ASM_PATH,          "e" );
 ToFile( "BUILD_SOSL",        $BUILD_SOSL,        "e" );
 ToFile( "FRAMEWORKSHOME",    $FRAMEWORKSHOME,    "e" );
@@ -1651,6 +1652,20 @@
 ToFile( "ENABLE_SYSTRAY_GTK", "@ENABLE_SYSTRAY_GTK@", "e" );
 ToFile( "ENABLE_STATIC_GTK", "@ENABLE_STATIC_GTK@", "e" );
 ToFile( "ENABLE_CAIRO",      "@ENABLE_CAIRO@",     "e" );
+ToFile( "ENABLE_OPENGL",     "@ENABLE_OPENGL@",    "e" );
+ToFile( "ENABLE_REPORTDESIGN","@ENABLE_REPORTDESIGN@","e" );
+ToFile( "SYSTEM_JFREEREPORT","@SYSTEM_JFREEREPORT@","e" );
+ToFile( "JCOMMON_JAR",       "@JCOMMON_JAR@",      "e" );
+ToFile( "SAC_JAR",           "@SAC_JAR@",          "e" );
+ToFile( "LIBXML_JAR",        "@LIBXML_JAR@",       "e" );
+ToFile( "FLUTE_JAR",         "@FLUTE_JAR@",        "e" );
+ToFile( "JFREEREPORT_JAR",   "@JFREEREPORT_JAR@",  "e" );
+ToFile( "LIBLAYOUT_JAR",     "@LIBLAYOUT_JAR@",    "e" );
+ToFile( "LIBLOADER_JAR",     "@LIBLOADER_JAR@",    "e" );
+ToFile( "LIBFORMULA_JAR",    "@LIBFORMULA_JAR@",   "e" );
+ToFile( "LIBREPOSITORY_JAR", "@LIBREPOSITORY_JAR@","e" );
+ToFile( "LIBFONTS_JAR",      "@LIBFONTS_JAR@",     "e" );
+ToFile( "JCOMMON_SERIALIZER_JAR","@JCOMMON_SERIALIZER_JAR@","e" );
 ToFile( "ENABLE_DBUS",       "@ENABLE_DBUS@",      "e" );
 ToFile( "ENABLE_GNOMEVFS",   "@ENABLE_GNOMEVFS@",  "e" );
 ToFile( "ENABLE_KDE",        "@ENABLE_KDE@",       "e" );
@@ -1975,6 +1990,7 @@
 ToFile( "HUNSPELL_CFLAGS",   "@HUNSPELL_CFLAGS@",  "e" );
 ToFile( "HUNSPELL_LIBS",     "@HUNSPELL_LIBS@",    "e" );
 ToFile( "SYSTEM_HYPH",       "@SYSTEM_HYPH@",      "e" );
+ToFile( "HYPHEN_LIB",        "@HYPHEN_LIB@",       "e" );
 ToFile( "SYSTEM_MYTHES",     "@SYSTEM_MYTHES@",    "e" );
 ToFile( "PAM",               "@PAM@",              "e" );
 ToFile( "NEW_SHADOW_API",    "@NEW_SHADOW_API@",   "e" );
Index: lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk,v
retrieving revision 1.15
retrieving revision 1.15.16.2
diff -u -u -r1.15 -r1.15.16.2
--- lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	20 Sep 2007 14:31:18 -0000	1.15
+++ lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	4 Dec 2007 20:59:44 -0000	1.15.16.2
@@ -41,7 +41,11 @@
 USE_DEFFILE=TRUE
 
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_HYPH)" == "YES"
+HNJLIB=$(HYPHEN_LIB)
+.ELSE
 HNJLIB=-lhnj
+.ENDIF
 .ELIF  "$(GUI)"=="OS2"
 HNJLIB=$(SLB)\libhnj.lib
 .ELSE
Index: odk/settings/settings.mk
===================================================================
RCS file: /cvs/api/odk/settings/settings.mk,v
retrieving revision 1.23
retrieving revision 1.23.14.1
diff -u -u -r1.23 -r1.23.14.1
--- odk/settings/settings.mk	31 Jul 2007 13:58:07 -0000	1.23
+++ odk/settings/settings.mk	15 Oct 2007 22:46:22 -0000	1.23.14.1
@@ -289,7 +289,11 @@
 SALHELPERLIB=-luno_salhelper$(COMID)
 REGLIB=-lreg
 STORELIB=-lstore
+ifeq "$(STLPORT_VER)" "500"
+STLPORTLIB=-lstlport
+else
 STLPORTLIB=-lstlport_gcc
+endif
 
 EMPTYSTRING=
 PATH_SEPARATOR=:
@@ -387,7 +391,11 @@
 SALHELPERLIB=-luno_salhelper$(COMID)
 REGLIB=-lreg
 STORELIB=-lstore
+ifeq "$(STLPORT_VER)" "500"
+STLPORTLIB=-lstlport -lstdc++
+else
 STLPORTLIB=-lstlport_gcc -lstdc++
+endif
 
 EMPTYSTRING=
 PATH_SEPARATOR=:
@@ -483,7 +491,11 @@
 SALHELPERLIB=-luno_salhelper$(COMID)
 REGLIB=-lreg
 STORELIB=-lstore
+ifeq "$(STLPORT_VER)" "500"
+STLPORTLIB=-lstlport
+else
 STLPORTLIB=-lstlport_gcc
+endif
 
 EMPTYSTRING=
 PATH_SEPARATOR=:
Index: postprocess/prj/build.lst
===================================================================
RCS file: /cvs/installation/postprocess/prj/build.lst,v
retrieving revision 1.33
retrieving revision 1.33.6.1
diff -u -u -r1.33 -r1.33.6.1
--- postprocess/prj/build.lst	20 Aug 2007 15:43:32 -0000	1.33
+++ postprocess/prj/build.lst	2 Dec 2007 12:34:15 -0000	1.33.6.1
@@ -1,4 +1,4 @@
-po	postprocess	::	accessibility automation basctl bean fondu BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io JAVAINSTALLER2:javainstaller2 lingucomponent MATHMLDTD:MathMLDTD MSFONTEXTRACT:msfontextract ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins scp2 scripting sd ODK:sdk_oo setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top unodevtools jfreereport reportdesign NULL
+po	postprocess	::	accessibility automation basctl bean fondu BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io JAVAINSTALLER2:javainstaller2 lingucomponent MATHMLDTD:MathMLDTD MSFONTEXTRACT:msfontextract ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins scp2 scripting sd ODK:sdk_oo setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top unodevtools JFREEREPORT:jfreereport REPORTDESIGN:reportdesign SDEXT:sdext NULL
 po	postprocess			    	usr1	-	all	po_mkout NULL
 po	postprocess\checkxml		nmake	-	all	po_checkxml NULL
 po	postprocess\checkdeliver	nmake	-	all	po_checkdlv NULL
Index: reportdesign/java/com/sun/star/report/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/java/com/sun/star/report/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.56.1
diff -u -u -r1.4 -r1.4.56.1
--- reportdesign/java/com/sun/star/report/makefile.mk	3 Aug 2007 09:48:29 -0000	1.4
+++ reportdesign/java/com/sun/star/report/makefile.mk	2 Dec 2007 12:34:15 -0000	1.4.56.1
@@ -42,7 +42,13 @@
 .INCLUDE: settings.mk
 #----- compile .java files -----------------------------------------
 
-JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar jcommon-1.0.10.jar
+JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+XCLASSPATH!:==$(XCLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)
+.ELSE
+JARFILES += jcommon-1.0.10.jar
+.ENDIF
+
 JAVAFILES       :=	DataRow.java\
 					DataSource.java\
 					DataSourceException.java\
Index: reportdesign/java/com/sun/star/report/pentaho/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/java/com/sun/star/report/pentaho/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.4.1
diff -u -u -r1.4 -r1.4.4.1
--- reportdesign/java/com/sun/star/report/pentaho/makefile.mk	6 Nov 2007 15:42:30 -0000	1.4
+++ reportdesign/java/com/sun/star/report/pentaho/makefile.mk	2 Dec 2007 12:34:15 -0000	1.4.4.1
@@ -44,7 +44,13 @@
 
 #----- compile .java files -----------------------------------------
 
-JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar jcommon-1.0.10.jar libxml-0.9.5.jar jfreereport-0.9.0-05.jar libloader-0.3.4.jar sac.jar liblayout-0.2.6.jar jcommon-serializer-0.1.0.jar libfonts-0.2.6.jar libformula-0.1.8.jar
+JARFILES        = ridl.jar unoil.jar jurt.jar juh.jar jut.jar java_uno.jar
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)$(PATH_SEPERATOR)$(LIBXML_JAR)$(PATH_SEPERATOR)$(JFREEREPORT_JAR)$(PATH_SEPERATOR)$(LIBLOADER_JAR)$(PATH_SEPERATOR)$(SAC_JAR)$(PATH_SEPERATOR)$(LIBLAYOUT_JAR)$(PATH_SEPERATOR)$(JCOMMON_SERIALIZER_JAR)$(PATH_SEPERATOR)$(LIBFONTS_JAR)$(PATH_SEPERATOR)$(LIBFORMULA_JAR)
+.ELSE
+JARFILES += jcommon-1.0.10.jar libxml-0.9.5.jar jfreereport-0.9.0-05.jar libloader-0.3.4.jar sac.jar liblayout-0.2.6.jar jcommon-serializer-0.1.0.jar libfonts-0.2.6.jar libformula-0.1.8.jar
+.ENDIF
+
 JAVAFILES       := $(shell @$(FIND) . -name "*.java")
 # JAVACLASSFILES	= $(foreach,i,$(JAVAFILES) $(CLASSDIR)$/$(PACKAGE)$/$(i:s/.java/.class/))
 
Index: reportdesign/java/com/sun/star/report/util/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/java/com/sun/star/report/util/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.56.1
diff -u -u -r1.4 -r1.4.56.1
--- reportdesign/java/com/sun/star/report/util/makefile.mk	3 Aug 2007 09:52:41 -0000	1.4
+++ reportdesign/java/com/sun/star/report/util/makefile.mk	2 Dec 2007 12:34:15 -0000	1.4.56.1
@@ -42,7 +42,12 @@
 .INCLUDE: settings.mk
 #----- compile .java files -----------------------------------------
 
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+CLASSPATH!:=$(CLASSPATH)$(PATH_SEPERATOR)$(JCOMMON_JAR)$(PATH_SEPERATOR)$(LIBXML_JAR)$(PATH_SEPERATOR)$(JFREEREPORT_JAR)
+.ELSE
 JARFILES        = jcommon-1.0.10.jar libxml-0.9.5.jar jfreereport-0.9.0-05.jar
+.ENDIF
+
 JAVAFILES       = DefaultJobProperties.java DefaultParameterMap.java DefaultReportJobDefinition.java FileRepository.java ManifestWriter.java
 # --- Targets ------------------------------------------------------
 .INCLUDE :  target.mk
Index: reportdesign/prj/build.lst
===================================================================
RCS file: /cvs/dba/reportdesign/prj/build.lst,v
retrieving revision 1.5
retrieving revision 1.5.4.2
diff -u -u -r1.5 -r1.5.4.2
--- reportdesign/prj/build.lst	27 Nov 2007 11:57:56 -0000	1.5
+++ reportdesign/prj/build.lst	3 Dec 2007 15:39:29 -0000	1.5.4.2
@@ -1,4 +1,4 @@
-rd  reportdesign    :   boost jfreereport comphelper dbaccess NULL
+rd  reportdesign    :   boost JFREEREPORT:jfreereport comphelper dbaccess readlicense_oo NULL
 rd  reportdesign                                                        usr1    -   all rd_mkout NULL
 rd  reportdesign\inc                                                    nmake   -   all rd_inc NULL
 rd  reportdesign\java							nmake   -   all rd_java NULL
Index: reportdesign/util/makefile.mk
===================================================================
RCS file: /cvs/dba/reportdesign/util/makefile.mk,v
retrieving revision 1.9
retrieving revision 1.9.4.2
diff -u -u -r1.9 -r1.9.4.2
--- reportdesign/util/makefile.mk	27 Nov 2007 11:59:51 -0000	1.9
+++ reportdesign/util/makefile.mk	3 Dec 2007 15:39:29 -0000	1.9.4.2
@@ -233,7 +233,7 @@
 LICLINES:=$(foreach,i,$(TXTFILES)  <license-text xlink:href="registration/$(i:f)" lang="$(subst,.txt, $(subst,license_, $(i:f)))" license-id="$(subst,.txt, $(subst,license_, $(i:f)))" />)
 .ENDIF  # "$(GUI)"!="WNT"
 
-REPRORTJARFILES := \
+REPORTJARFILES := \
 	$(ZIP1DIR)$/jcommon-1.0.10.jar										\
 	$(ZIP1DIR)$/sac.jar													\
 	$(ZIP1DIR)$/libxml-0.9.5.jar										\
@@ -245,14 +244,14 @@
 	$(ZIP1DIR)$/libformula-0.1.8.jar									\
 	$(ZIP1DIR)$/librepository-0.1.1.jar									\
 	$(ZIP1DIR)$/libfonts-0.2.6.jar										\
-	$(ZIP1DIR)$/jcommon-serializer-0.1.0.jar							\
+	$(ZIP1DIR)$/jcommon-serializer-0.1.0.jar								\
 	$(ZIP1DIR)$/sun-report-builder.jar
 
 # --- Targets ----------------------------------
 .INCLUDE : target.mk
 
 .IF "$(ZIP1TARGETN)"!=""
-$(ZIP1TARGETN) :  $(TXTFILES) $(XMLFILES) $(HTMLFILES) $(REPRORTJARFILES)
+$(ZIP1TARGETN) :  $(TXTFILES) $(XMLFILES) $(HTMLFILES) $(REPORTJARFILES)
 .ENDIF          # "$(ZIP1TARGETN)"!="
 
 $(MISC)$/update_report.flag : $(XCU_FILES)
@@ -286,9 +285,15 @@
 	echo $(XCU_TMP)
     $(COPY) $< $@
 
+.IF "$(SYSTEM_JFREEREPORT)" == "YES"
+$(ZIP1DIR)$/%.jar : $(JFREEREPORT_JAR:d:d)$/%.jar
+	@@-$(MKDIRHIER) $(@:d)
+    $(COPY) $< $@
+.ELSE
 $(ZIP1DIR)$/%.jar : $(SOLARBINDIR)$/%.jar
 	@@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
+.ENDIF
 
 $(ZIP1DIR)$/%.jar : $(CLASSDIR)$/%.jar
 	@@-$(MKDIRHIER) $(@:d)
Index: scp2/source/base/file_base.scp
===================================================================
RCS file: /cvs/installation/scp2/source/base/file_base.scp,v
retrieving revision 1.22
retrieving revision 1.22.4.1
diff -u -u -r1.22 -r1.22.4.1
--- scp2/source/base/file_base.scp	21 Nov 2007 16:25:37 -0000	1.22
+++ scp2/source/base/file_base.scp	2 Dec 2007 12:34:16 -0000	1.22.4.1
@@ -64,6 +64,8 @@
 STD_RES_FILE_PATCH( gid_File_Res_Sdbcl, sdbcl)
 STD_RES_FILE_PATCH( gid_File_Res_Sdberr, sdberr)
 
+#ifdef ENABLE_REPORTDESIGN
+
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Rpt, rpt )
 STD_RES_FILE_PATCH( gid_File_Res_Rpt, rpt)
 
@@ -72,6 +74,8 @@
 
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Rptxml, rptxml )
 
+#endif
+
 #if defined UNX && ! defined MACOSX
 
 File gid_File_Share_Xdg_Base
Index: scp2/source/base/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/base/makefile.mk,v
retrieving revision 1.5
retrieving revision 1.5.820.2
diff -u -u -r1.5 -r1.5.820.2
--- scp2/source/base/makefile.mk	15 Nov 2005 12:54:19 -0000	1.5
+++ scp2/source/base/makefile.mk	4 Dec 2007 20:59:44 -0000	1.5.820.2
@@ -45,6 +45,10 @@
 
 .INCLUDE :	settings.mk
 
+.IF "$(ENABLE_REPORTDESIGN)" == "YES"
+SCPDEFS += -DENABLE_REPORTDESIGN
+.ENDIF
+
 SCP_PRODUCT_TYPE=osl
 
 PARFILES= \
Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.265
retrieving revision 1.265.4.1
diff -u -u -r1.265 -r1.265.4.1
--- scp2/source/ooo/file_library_ooo.scp	23 Nov 2007 13:37:59 -0000	1.265
+++ scp2/source/ooo/file_library_ooo.scp	2 Dec 2007 23:21:56 -0000	1.265.4.1
@@ -2692,6 +2692,7 @@
 End
 #endif
 
+#ifdef OPENSSL
 File gid_File_Lib_Openssl
     TXT_FILE_BODY;
     Styles = (PACKED,PATCH);
@@ -2700,6 +2701,7 @@
     Name = "ssleay32.dll";
   #endif
 End
+#endif
 
 File gid_File_Lib_Crypto
     TXT_FILE_BODY;
Index: scp2/source/ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.58
retrieving revision 1.58.14.1
diff -u -u -r1.58 -r1.58.14.1
--- scp2/source/ooo/makefile.mk	6 Nov 2007 15:52:23 -0000	1.58
+++ scp2/source/ooo/makefile.mk	2 Dec 2007 23:21:56 -0000	1.58.14.1
@@ -203,6 +203,11 @@
 SCPDEFS+=-DDISABLE_NEON
 .ENDIF
 
+# if yes or unset (neon not used) -> do not install openssl library!
+.IF $(SYSTEM_OPENSSL) == "NO"
+SCPDEFS+=-DOPENSSL
+.ENDIF
+
 .IF "$(ENABLE_VBA)" == "YES"
     .IF "$(VBA_EXTENSION)" == "YES"
         SCPDEFS+=-DVBA_EXTENSION
Index: slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx
===================================================================
RCS file: /cvs/graphics/slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx,v
retrieving revision 1.2
retrieving revision 1.2.2.2
diff -u -u -r1.2 -r1.2.2.2
--- slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx	9 Nov 2007 10:17:20 -0000	1.2
+++ slideshow/source/engine/OGLTrans/OGLTrans_TransitionerImpl.cxx	4 Dec 2007 21:10:09 -0000	1.2.2.2
@@ -502,12 +502,21 @@
     // XTransitionFactory
     virtual ::sal_Bool SAL_CALL hasTransition( ::sal_Int16 transitionType, ::sal_Int16 transitionSubType ) throw (uno::RuntimeException)
     {
-        return createTransition(transitionType, 
-                                transitionSubType, 
-                                uno::Reference< presentation::XSlideShowView >(),
-                                uno::Reference< rendering::XBitmap >(),
-                                uno::Reference< rendering::XBitmap >(),
-                                geometry::RealPoint2D()).is();
+        if( transitionType != animations::TransitionType::MISCSHAPEWIPE )
+            return sal_False;
+
+        switch( transitionSubType )
+        {
+            case animations::TransitionSubType::ACROSS:
+            case animations::TransitionSubType::CORNERSOUT:
+            case animations::TransitionSubType::CIRCLE:
+            case animations::TransitionSubType::FANOUTHORIZONTAL:
+            case animations::TransitionSubType::CORNERSIN:  
+                return sal_True;
+
+            default:
+                return sal_False;
+        }
     }
 
     virtual uno::Reference< presentation::XTransition > SAL_CALL createTransition( 
@@ -525,13 +534,13 @@
         switch( transitionSubType )
         {
             case animations::TransitionSubType::ACROSS:
-                pTransition->makeNByMTileFlip(5,5);
+                pTransition->makeNByMTileFlip(8,6);
                 break;   
             case animations::TransitionSubType::CORNERSOUT:
                 pTransition->makeOutsideCubeFaceToLeft();
                 break;            
             case animations::TransitionSubType::CIRCLE:
-                pTransition->makeRevolvingCircles(5,20);
+                pTransition->makeRevolvingCircles(8,128);
                 break;
             case animations::TransitionSubType::FANOUTHORIZONTAL:
                 pTransition->makeHelix(20);
Index: slideshow/source/engine/OGLTrans/makefile.mk
===================================================================
RCS file: /cvs/graphics/slideshow/source/engine/OGLTrans/makefile.mk,v
retrieving revision 1.2
retrieving revision 1.2.2.1
diff -u -u -r1.2 -r1.2.2.1
--- slideshow/source/engine/OGLTrans/makefile.mk	9 Nov 2007 10:17:54 -0000	1.2
+++ slideshow/source/engine/OGLTrans/makefile.mk	22 Nov 2007 14:24:53 -0000	1.2.2.1
@@ -53,12 +53,6 @@
 # --- Common ----------------------------------------------------------
 #.IF "$(GUI)"=="UNX"
 
-#.IF "$(SYSTEM_OPENGL)" == "YES"
-CFLAGS+=$(OGLTRANS_CFLAGS)
-#.ELSE
-#CFLAGS+=-I$(SOLARINCDIR)/OGLTrans
-#.ENDIF
-
 SLOFILES = \
         $(SLO)$/OGLTrans_TransitionImpl.obj \
         $(SLO)$/OGLTrans_TransitionerImpl.obj
Index: solenv/bin/cwsclone.pl
===================================================================
RCS file: /cvs/tools/solenv/bin/cwsclone.pl,v
retrieving revision 1.5
retrieving revision 1.5.770.1
diff -u -u -r1.5 -r1.5.770.1
--- solenv/bin/cwsclone.pl	3 Nov 2005 10:32:05 -0000	1.5
+++ solenv/bin/cwsclone.pl	4 Dec 2007 14:01:20 -0000	1.5.770.1
@@ -133,7 +133,7 @@
 $cvs_module->cvs_repository($config->get_cvs_server_repository());
 my %cvs_aliases = $cvs_module->get_aliases_hash();
 
-foreach my $module (split( /\s+/, $cvs_aliases{'OpenOffice'})) {
+foreach my $module (split( /\s+/, $cvs_aliases{'OpenOffice2'})) {
     my $use_tag;
     my $result1;
     my $result2;
Index: solenv/inc/unxbsdi.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxbsdi.mk,v
retrieving revision 1.15
retrieving revision 1.15.236.1
diff -u -u -r1.15 -r1.15.236.1
--- solenv/inc/unxbsdi.mk	9 Mar 2007 09:07:07 -0000	1.15
+++ solenv/inc/unxbsdi.mk	15 Oct 2007 22:46:21 -0000	1.15.236.1
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=-DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+=-DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # this is a platform with JAVA support
 .IF "$(SOLAR_JAVA)"!=""
@@ -154,8 +154,13 @@
 STDSHLGUIMT=-lX11 -lXext -lpthread -lm
 STDSHLCUIMT=-lpthread -lm
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 
 # name of library manager
Index: solenv/inc/unxbsdi2.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxbsdi2.mk,v
retrieving revision 1.13
retrieving revision 1.12.236.2
diff -u -u -r1.13 -r1.12.236.2
--- solenv/inc/unxbsdi2.mk	15 Oct 2007 12:40:03 -0000	1.13
+++ solenv/inc/unxbsdi2.mk	14 Nov 2007 21:10:09 -0000	1.12.236.2
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+= -DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+= -DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # this is a platform with JAVA support
 .IF "$(SOLAR_JAVA)"!=""
@@ -168,8 +168,13 @@
 
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport -lstdc++
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
Index: solenv/inc/unxbsds.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxbsds.mk,v
retrieving revision 1.16
retrieving revision 1.16.236.1
diff -u -u -r1.16 -r1.16.236.1
--- solenv/inc/unxbsds.mk	9 Mar 2007 09:07:44 -0000	1.16
+++ solenv/inc/unxbsds.mk	15 Oct 2007 22:46:22 -0000	1.16.236.1
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=-D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400 -DOSL_USE_SYS_V_SEMAPHORE
+CDEFS+=-D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER) -DOSL_USE_SYS_V_SEMAPHORE
 
 # this is a platform with JAVA support
 .IF "$(SOLAR_JAVA)"!=""
@@ -154,8 +154,13 @@
 
 # STLport always needs pthread. This may yield some redundant -lpthread
 # but that doesn't matter. 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport -lpthread
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC) -lpthread
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lpthread
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC) -lpthread
+.ENDIF
 
 
 # name of library manager
Index: solenv/inc/unxfbsd.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxfbsd.mk,v
retrieving revision 1.5
retrieving revision 1.4.128.2
diff -u -u -r1.5 -r1.4.128.2
--- solenv/inc/unxfbsd.mk	15 Oct 2007 12:40:17 -0000	1.5
+++ solenv/inc/unxfbsd.mk	14 Nov 2007 21:10:03 -0000	1.4.128.2
@@ -192,11 +192,21 @@
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
 .IF "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlportstlg
+LIBSTLPORTST=$(STATIC) -lstlportstlg $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_stldebug
 LIBSTLPORTST=$(STATIC) -lstlport_gcc_stldebug $(DYNAMIC)
+.ENDIF
 .ELSE # "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 .ENDIF # "$(USE_STLP_DEBUG)" != ""
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
Index: solenv/inc/unxirgm.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxirgm.mk,v
retrieving revision 1.13
retrieving revision 1.13.236.1
diff -u -u -r1.13 -r1.13.236.1
--- solenv/inc/unxirgm.mk	9 Mar 2007 09:08:29 -0000	1.13
+++ solenv/inc/unxirgm.mk	15 Oct 2007 22:46:22 -0000	1.13.236.1
@@ -136,8 +136,13 @@
 STDSHLCUIMT=	-L/usr/lib32 $(THREADLIB) $(DYNAMIC) -lm -lgcc -lc
 THREADLIB=		-lpthread
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST= -lstlport
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST= -lstlport_gcc
+.ENDIF
 
 LIBMGR=			ar
 LIBFLAGS=		-r
Index: solenv/inc/unxlngi4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi4.mk,v
retrieving revision 1.44
retrieving revision 1.43.236.2
diff -u -u -r1.44 -r1.43.236.2
--- solenv/inc/unxlngi4.mk	15 Oct 2007 12:40:36 -0000	1.44
+++ solenv/inc/unxlngi4.mk	14 Nov 2007 21:10:06 -0000	1.43.236.2
@@ -44,7 +44,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # enable visibility define in "sal/types.h"
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
@@ -189,8 +189,13 @@
 
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport -lstdc++
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
Index: solenv/inc/unxlngi6.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngi6.mk,v
retrieving revision 1.41
retrieving revision 1.40.34.2
diff -u -u -r1.41 -r1.40.34.2
--- solenv/inc/unxlngi6.mk	15 Oct 2007 12:41:01 -0000	1.41
+++ solenv/inc/unxlngi6.mk	14 Nov 2007 21:10:27 -0000	1.40.34.2
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -DX86 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # enable visibility define in "sal/types.h"
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
@@ -212,11 +212,21 @@
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
 .IF "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlportstlg
+LIBSTLPORTST=$(STATIC) -lstlportstlg $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_stldebug
 LIBSTLPORTST=$(STATIC) -lstlport_gcc_stldebug $(DYNAMIC)
+.ENDIF
 .ELSE # "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 .ENDIF # "$(USE_STLP_DEBUG)" != ""
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
Index: solenv/inc/unxlngppc.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc.mk,v
retrieving revision 1.28
retrieving revision 1.28.104.1
diff -u -u -r1.28 -r1.28.104.1
--- solenv/inc/unxlngppc.mk	31 Jul 2007 13:06:33 -0000	1.28
+++ solenv/inc/unxlngppc.mk	15 Oct 2007 22:46:22 -0000	1.28.104.1
@@ -46,7 +46,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl 
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400 -DPOWERPC -DPPC 
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER) -DPOWERPC -DPPC 
 
 # enable visibility define in "sal/types.h"
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
@@ -182,8 +182,13 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 # name of library manager
 LIBMGR=ar 
Index: solenv/inc/unxlngppc4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc4.mk,v
retrieving revision 1.29
retrieving revision 1.28.106.2
diff -u -u -r1.29 -r1.28.106.2
--- solenv/inc/unxlngppc4.mk	15 Oct 2007 12:41:14 -0000	1.29
+++ solenv/inc/unxlngppc4.mk	14 Nov 2007 21:10:24 -0000	1.28.106.2
@@ -44,7 +44,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400 -DPOWERPC -DPPC
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER) -DPOWERPC -DPPC
 
 # enable visibility define in "sal/types.h"
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
@@ -181,8 +181,13 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport -lstdc++
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
Index: solenv/inc/unxlngs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngs.mk,v
retrieving revision 1.22
retrieving revision 1.21.102.2
diff -u -u -r1.22 -r1.21.102.2
--- solenv/inc/unxlngs.mk	15 Oct 2007 12:41:27 -0000	1.22
+++ solenv/inc/unxlngs.mk	14 Nov 2007 21:10:15 -0000	1.21.102.2
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DSPARC -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DSPARC -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # enable visibility define in "sal/types.h"
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
@@ -185,8 +185,13 @@
 
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport $(STDLIBCPP)
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc $(STDLIBCPP)
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
Index: solenv/inc/unxlngs3904.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngs3904.mk,v
retrieving revision 1.17
retrieving revision 1.17.104.1
diff -u -u -r1.17 -r1.17.104.1
--- solenv/inc/unxlngs3904.mk	31 Jul 2007 13:07:15 -0000	1.17
+++ solenv/inc/unxlngs3904.mk	15 Oct 2007 22:46:22 -0000	1.17.104.1
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # this is a platform with JAVA support
 .IF "$(SOLAR_JAVA)"!=""
@@ -165,8 +165,13 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport -lstdc++
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
Index: solenv/inc/unxlngx6.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngx6.mk,v
retrieving revision 1.12
retrieving revision 1.10.102.2
diff -u -u -r1.12 -r1.10.102.2
--- solenv/inc/unxlngx6.mk	13 Nov 2007 14:17:07 -0000	1.12
+++ solenv/inc/unxlngx6.mk	14 Nov 2007 21:10:12 -0000	1.10.102.2
@@ -45,7 +45,7 @@
 #LINKOUTPUT_FILTER=" |& $(SOLARENV)$/bin$/msg_filter"
 
 # _PTHREADS is needed for the stl
-CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -DX86_64 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=400
+CDEFS+=$(PTHREAD_CFLAGS) -DGLIBC=2 -DX86_64 -D_PTHREADS -D_REENTRANT -DNEW_SOLAR -D_USE_NAMESPACE=1 -DSTLPORT_VERSION=$(STLPORT_VER)
 
 # enable visibility define in "sal/types.h"
 .IF "$(HAVE_GCC_VISIBILITY_FEATURE)" == "TRUE"
@@ -206,11 +206,21 @@
 LIBSALCPPRT*=-Wl,--whole-archive -lsalcpprt -Wl,--no-whole-archive
 
 .IF "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlportstlg
+LIBSTLPORTST=$(STATIC) -lstlportstlg $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_stldebug
 LIBSTLPORTST=$(STATIC) -lstlport_gcc_stldebug $(DYNAMIC)
+.ENDIF
 .ELSE # "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 .ENDIF # "$(USE_STLP_DEBUG)" != ""
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
Index: solenv/inc/unxmacx.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxmacx.mk,v
retrieving revision 1.22
retrieving revision 1.21.8.2
diff -u -u -r1.22 -r1.21.8.2
--- solenv/inc/unxmacx.mk	12 Nov 2007 16:20:07 -0000	1.22
+++ solenv/inc/unxmacx.mk	14 Nov 2007 21:10:36 -0000	1.21.8.2
@@ -51,7 +51,7 @@
 # Definitions that we may need on the compile line.
 # -D_PTHREADS and -D_REENTRANT are needed for STLport, and must be specified when
 #  compiling STLport sources too, either internally or externally.
-CDEFS+=-DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNO_PTHREAD_PRIORITY $(PROCESSOR_DEFINES) -DSTLPORT_VERSION=400 -D_USE_NAMESPACE=1
+CDEFS+=-DGLIBC=2 -D_PTHREADS -D_REENTRANT -DNO_PTHREAD_PRIORITY $(PROCESSOR_DEFINES) -DSTLPORT_VERSION=$(STLPORT_VER) -D_USE_NAMESPACE=1
 .IF "$(GUIBASE)"=="unx"
 CDEFS+= -DX_LOCALE
 .ENDIF
@@ -162,11 +162,21 @@
 #  STLport library names
 # ---------------------------------
 .IF "$(USE_STLP_DEBUG)" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=-lstlportstlg
+LIBSTLPORTST=$(STATIC) -lstlportstlg
+.ELSE
 LIBSTLPORT=-lstlport_gcc_stldebug
 LIBSTLPORTST=$(SOLARVERSION)$/$(INPATH)$/lib$/libstlport_gcc_stldebug.a
+.ENDIF
 .ELSE # "$(USE_STLP_DEBUG" != ""
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=-lstlport
+LIBSTLPORTST=$(STATIC) -lstlport
+.ELSE
 LIBSTLPORT=-lstlport_gcc
 LIBSTLPORTST=$(SOLARVERSION)$/$(INPATH)$/lib$/libstlport_gcc.a
+.ENDIF
 .ENDIF # "$(USE_STLP_DEBUG" != ""
 
 # ---------------------------------
Index: solenv/inc/unxsogi.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxsogi.mk,v
retrieving revision 1.9
retrieving revision 1.9.236.1
diff -u -u -r1.9 -r1.9.236.1
--- solenv/inc/unxsogi.mk	9 Mar 2007 09:11:34 -0000	1.9
+++ solenv/inc/unxsogi.mk	15 Oct 2007 22:46:22 -0000	1.9.236.1
@@ -41,7 +41,7 @@
 JAVAFLAGSDEBUG=-g
 
 # _PTHREADS is needed for the stl
-CDEFS+=-D_PTHREADS -D_REENTRANT -DSYSV -DSUN -DSUN4 -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=400 -D_USE_NAMESPACE=1 
+CDEFS+=-D_PTHREADS -D_REENTRANT -DSYSV -DSUN -DSUN4 -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=$(STLPORT_VER) -D_USE_NAMESPACE=1 
 
 
 .IF "$(SOLAR_JAVA)"!=""
@@ -120,8 +120,13 @@
 
 STDLIBGUIMT+=-lX11
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 LIBMGR=ar
 LIBFLAGS=-r
Index: solenv/inc/unxsogs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxsogs.mk,v
retrieving revision 1.14
retrieving revision 1.14.236.1
diff -u -u -r1.14 -r1.14.236.1
--- solenv/inc/unxsogs.mk	9 Mar 2007 09:11:46 -0000	1.14
+++ solenv/inc/unxsogs.mk	15 Oct 2007 22:46:22 -0000	1.14.236.1
@@ -41,7 +41,7 @@
 JAVAFLAGSDEBUG=-g
 
 # _PTHREADS is needed for the stl
-CDEFS+=-D_PTHREADS -D_REENTRANT -DSYSV -DSUN -DSUN4 -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=400 -D_USE_NAMESPACE=1 
+CDEFS+=-D_PTHREADS -D_REENTRANT -DSYSV -DSUN -DSUN4 -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=$(STLPORT_VER) -D_USE_NAMESPACE=1 
 
 
 .IF "$(SOLAR_JAVA)"!=""
@@ -119,8 +119,13 @@
 
 STDLIBGUIMT+=-lX11
 
+.IF "$(STLPORT_VER)" >= "500"
+LIBSTLPORT=$(DYNAMIC) -lstlport
+LIBSTLPORTST=$(STATIC) -lstlport $(DYNAMIC)
+.ELSE
 LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
 LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+.ENDIF
 
 LIBMGR=ar
 LIBFLAGS=-r
Index: solenv/inc/unxsoli4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxsoli4.mk,v
retrieving revision 1.39
retrieving revision 1.38.34.2
diff -u -u -r1.39 -r1.38.34.2
--- solenv/inc/unxsoli4.mk	15 Oct 2007 12:41:49 -0000	1.39
+++ solenv/inc/unxsoli4.mk	14 Nov 2007 21:10:33 -0000	1.38.34.2
@@ -38,7 +38,7 @@
 AFLAGS=-P
 
 
-CDEFS+=-mt -D_PTHREADS -DSYSV -DSUN -DSUN4 -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=400
+CDEFS+=-mt -D_PTHREADS -DSYSV -DSUN -DSUN4 -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=$(STLPORT_VER)
 
 SOLAR_JAVA*=TRUE
 .IF "$(SOLAR_JAVA)"!=""
Index: solenv/inc/unxsols4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxsols4.mk,v
retrieving revision 1.45
retrieving revision 1.44.34.2
diff -u -u -r1.45 -r1.44.34.2
--- solenv/inc/unxsols4.mk	15 Oct 2007 12:42:02 -0000	1.45
+++ solenv/inc/unxsols4.mk	14 Nov 2007 21:09:57 -0000	1.44.34.2
@@ -39,7 +39,7 @@
 AFLAGS=-P -xarch=v8plus -D__sparcv8plus
 
 
-CDEFS+=-D_PTHREADS -DSYSV -DSUN -DSUN4 -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=400
+CDEFS+=-D_PTHREADS -DSYSV -DSUN -DSUN4 -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DSTLPORT_VERSION=$(STLPORT_VER)
 
 SOLAR_JAVA*=TRUE
 .IF "$(SOLAR_JAVA)"!=""
Index: solenv/inc/wntmsci10.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/wntmsci10.mk,v
retrieving revision 1.34
retrieving revision 1.33.34.2
diff -u -u -r1.34 -r1.33.34.2
--- solenv/inc/wntmsci10.mk	15 Oct 2007 12:42:27 -0000	1.34
+++ solenv/inc/wntmsci10.mk	14 Nov 2007 21:10:00 -0000	1.33.34.2
@@ -222,7 +222,7 @@
     lingucomponent \
     soldep
 
-CDEFS+=-DSTLPORT_VERSION=400 -DWINVER=0x400 -D_WIN32_IE=0x400
+CDEFS+=-DSTLPORT_VERSION=$(STLPORT_VER) -DWINVER=0x400 -D_WIN32_IE=0x400
 CDEFS+=-D_MT
 
 COMMENTFLAG=/COMMENT:"$(PRJNAME)_$(UPD)_$(VCSID)_"
Index: store/util/makefile.mk
===================================================================
RCS file: /cvs/ucb/store/util/makefile.mk,v
retrieving revision 1.22
retrieving revision 1.20.8.2
diff -u -u -r1.22 -r1.20.8.2
--- store/util/makefile.mk	2 Nov 2007 13:04:39 -0000	1.22
+++ store/util/makefile.mk	14 Nov 2007 21:10:50 -0000	1.20.8.2
@@ -70,7 +70,8 @@
 
 SHL1STDLIBS=	$(SALLIB)
 
-.IF "$(USE_STLP_DEBUG)" != ""
+# system STLport5 needs it
+.IF "$(USE_STLP_DEBUG)" != "" || "$(STLPORT_VER)" >= "500"
 SHL1STDLIBS+=$(LIBSTLPORT)
 .ENDIF
 
