Index: connectivity/source/drivers/dbase/DTable.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/dbase/DTable.cxx,v
retrieving revision 1.94
retrieving revision 1.94.4.1
diff -u -p -u -p -r1.94 -r1.94.4.1
--- connectivity/source/drivers/dbase/DTable.cxx	20 Jun 2006 01:21:12 -0000	1.94
+++ connectivity/source/drivers/dbase/DTable.cxx	27 Jun 2006 20:00:34 -0000	1.94.4.1
@@ -2408,7 +2408,7 @@ BOOL ODbaseTable::ReadMemo(ULONG nBlockN
 				return sal_False;
 			}
 
-			ULONG nLength;
+			sal_uInt32 nLength(0);
 			(*m_pMemoStream) >> nLength;
 
 			if (m_aMemoHeader.db_typ == MemodBaseIV)
Index: xmloff/source/style/impastp4.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/style/impastp4.cxx,v
retrieving revision 1.17
retrieving revision 1.17.8.1
diff -u -p -u -p -r1.17 -r1.17.8.1
--- xmloff/source/style/impastp4.cxx	19 Jun 2006 18:33:22 -0000	1.17
+++ xmloff/source/style/impastp4.cxx	27 Jun 2006 13:15:34 -0000	1.17.8.1
@@ -276,7 +276,7 @@ OUString SvXMLAutoStylePoolP_Impl::Find(
 {
 	OUString sName;
 
-	sal_uInt32 nPos;
+	ULONG nPos;
 	XMLFamilyData_Impl aTemporary( nFamily );
 	XMLFamilyData_Impl *pFamily = 0;
 	if( maFamilyList.Seek_Entry( &aTemporary, &nPos ) )
Index: soltools/cpp/_tokens.c
===================================================================
RCS file: /cvs/tools/soltools/cpp/_tokens.c,v
retrieving revision 1.4
retrieving revision 1.4.2.1
diff -u -p -u -p -r1.4 -r1.4.2.1
--- soltools/cpp/_tokens.c	20 Jun 2006 05:07:05 -0000	1.4
+++ soltools/cpp/_tokens.c	27 Jun 2006 14:27:36 -0000	1.4.2.1
@@ -364,7 +364,7 @@ void
     if (str)
         fprintf(stderr, "%s ", str);
     if (tp < trp->bp || tp > trp->lp)
-        fprintf(stderr, "(tp offset %d) ", tp - trp->bp);
+        fprintf(stderr, "(tp offset %ld) ", (long int) (tp - trp->bp));
     for (tp = trp->bp; tp < trp->lp && tp < trp->bp + 32; tp++)
     {
         if (tp->type != NL)
Index: soltools/testhxx/create.pl
===================================================================
RCS file: /cvs/tools/soltools/testhxx/create.pl,v
retrieving revision 1.2
retrieving revision 1.2.2.1
diff -u -p -u -p -r1.2 -r1.2.2.1
--- soltools/testhxx/create.pl	20 Jun 2006 05:09:44 -0000	1.2
+++ soltools/testhxx/create.pl	27 Jun 2006 12:34:08 -0000	1.2.2.1
@@ -74,7 +74,7 @@ if ($ENV{OS} eq 'LINUX') {
     print STDOUT 'unlink ${my_tmp} || exit 1', "\n";
     print STDOUT 'unlink ${my_tmp}.out || exit 1', "\n";
     print STDOUT 'exit ${my_ret}', "\n";
-} elsif ($ENV{OS} eq 'WNT') {
+} elsif ($ENV{OS} eq 'WNT' and $ENV{USE_SHELL} eq '4nt') {
     1 while $in =~ s!\s+-I\s*\.\S*\s*! !g; # discard relative includes
     $in =~ s!(\s+-I\s*)(?i:$solarversion)(\S*)!$1%SOLARVERSION%$2!og;
         # macrofy includes to solver
@@ -94,6 +94,10 @@ if ($ENV{OS} eq 'LINUX') {
     print STDOUT 'DEL /EQ %my_tmp% %my_tmp%.obj %my_tmp%.pdb %my_tmp%.out',
         "\n";
     print STDOUT 'QUIT %my_ret%', "\n";
+} elsif ($ENV{OS} eq 'WNT' and $ENV{USE_SHELL} ne '4nt') {
+    print STDOUT '#!/bin/bash', "\n";
+    print STDOUT 'echo testhxx not yet available on this platform 1>&2', "\n";
+    print STDOUT 'exit 1', "\n";
 } else {
     print STDOUT 'echo \'no testhxx on this platform\'', "\n";
 }
Index: soltools/testhxx/makefile.mk
===================================================================
RCS file: /cvs/tools/soltools/testhxx/makefile.mk,v
retrieving revision 1.2
retrieving revision 1.2.2.2
diff -u -p -u -p -r1.2 -r1.2.2.2
--- soltools/testhxx/makefile.mk	20 Jun 2006 05:09:53 -0000	1.2
+++ soltools/testhxx/makefile.mk	27 Jun 2006 12:34:08 -0000	1.2.2.2
@@ -38,15 +38,15 @@ TARGET := testhxx
 LIBTARGET := NO
 ENABLE_EXCEPTIONS := TRUE
 
-.INCLUDE: $(PRJ)$/util$/makefile.pmk
-.INCLUDE: settings.mk
-
 CAPTURE_COMMAND = echo
 CAPTURE_OUTPUT = > $(MISC)$/testhxx.output
 
+.INCLUDE: $(PRJ)$/util$/makefile.pmk
+.INCLUDE: settings.mk
+
 SLOFILES = $(SLO)$/testhxx.obj
 
-.IF "$(OS)" == "WNT"
+.IF "$(USE_SHELL)" == "4nt"
 MY_SCRIPT_EXT := .btm
 MY_EXECUTABLE := @ echo
 .ELSE
Index: berkeleydb/db-4.2.52.patch
===================================================================
RCS file: /cvs/external/berkeleydb/db-4.2.52.patch,v
retrieving revision 1.10
retrieving revision 1.10.6.1
diff -u -p -u -p -r1.10 -r1.10.6.1
--- berkeleydb/db-4.2.52.patch	19 Jun 2006 10:39:16 -0000	1.10
+++ berkeleydb/db-4.2.52.patch	27 Jun 2006 11:51:29 -0000	1.10.6.1
@@ -816,7 +816,7 @@
 ! 
 ! EXTERNAL_WARNINGS_NOT_ERRORS := TRUE
 ! 
-! SOLARINC=-I..$/build_win32 -I..$/ $(PSDKINC) -I$(COMPATH)$/include
+! SOLARINC=-I..$/build_win32 -I..$/ -I$(PSDK_HOME)$/include $(PSDKINC) -I$(COMPATH)$/include
 ! .INCLUDE :  settings.mk
 ! 
 ! 
Index: vcl/unx/kde/kdedata.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/kde/kdedata.cxx,v
retrieving revision 1.12
retrieving revision 1.12.12.1
diff -u -p -u -p -r1.12 -r1.12.12.1
--- vcl/unx/kde/kdedata.cxx	19 Jun 2006 19:47:55 -0000	1.12
+++ vcl/unx/kde/kdedata.cxx	27 Jun 2006 14:12:29 -0000	1.12.12.1
@@ -35,13 +35,7 @@
 
 #define _SV_SALDATA_CXX
 
-#define Region QtXRegion
-#include <kaboutdata.h>
-#include <kapplication.h>
-#include <kcmdlineargs.h>
-#include <kstartupinfo.h>
-#include <qpaintdevice.h>
-#undef Region
+#include "kdewrapper.h"
 
 #include <unistd.h>
 #include <fcntl.h>
Index: vcl/unx/kde/kdewrapper.h
===================================================================
RCS file: vcl/unx/kde/kdewrapper.h
diff -N vcl/unx/kde/kdewrapper.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ vcl/unx/kde/kdewrapper.h	27 Jun 2006 14:13:13 -0000	1.1.2.1
@@ -0,0 +1,84 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2006 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef INCLUDED_VCL_UNX_KDE_KDEWRAPPER_H
+#define INCLUDED_VCL_UNX_KDE_KDEWRAPPER_H
+
+#ifndef _SAL_CONFIG_H_
+#include "sal/config.h"
+#endif
+
+#if defined __GNUC__
+#pragma GCC system_header
+#endif
+
+// Hack, but needed because of conflicting types...
+#define Region QtXRegion
+
+#include <qcheckbox.h>
+#include <qcombobox.h>
+#include <qframe.h>
+#include <qlineedit.h>
+#include <qlistview.h>
+#include <qmainwindow.h>
+#include <qpaintdevice.h>
+#include <qpainter.h>
+#include <qpushbutton.h>
+#include <qradiobutton.h>
+#include <qrangecontrol.h>
+#include <qtabbar.h>
+#include <qtabwidget.h>
+#include <qtoolbar.h>
+#include <qtoolbutton.h>
+#include <qwidget.h>
+
+// Ugly hack to be able to access QMenuItem::is_enabled
+#define private public
+#include <qmenudata.h>
+#undef private
+
+#include <kaboutdata.h>
+#include <kapplication.h>
+#include <kcmdlineargs.h>
+#include <kconfig.h>
+#include <kglobal.h>
+#include <kmainwindow.h>
+#include <kmenubar.h>
+#include <kstartupinfo.h>
+#include <kstyle.h>
+
+#undef Region
+
+#endif
Index: vcl/unx/kde/salnativewidgets-kde.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/kde/salnativewidgets-kde.cxx,v
retrieving revision 1.13
retrieving revision 1.13.12.1
diff -u -p -u -p -r1.13 -r1.13.12.1
--- vcl/unx/kde/salnativewidgets-kde.cxx	19 Jun 2006 19:48:09 -0000	1.13
+++ vcl/unx/kde/salnativewidgets-kde.cxx	27 Jun 2006 14:12:29 -0000	1.13.12.1
@@ -35,41 +35,7 @@
 
 #define _SV_SALNATIVEWIDGETS_KDE_CXX
 
-// Hack, but needed because of conflicting types...
-#define Region QtXRegion
-
-#include <qcheckbox.h>
-#include <qcombobox.h>
-#include <qframe.h>
-#include <qlineedit.h>
-#include <qlistview.h>
-#include <qmainwindow.h>
-#include <qpainter.h>
-#include <qpushbutton.h>
-#include <qradiobutton.h>
-#include <qrangecontrol.h>
-#include <qtabbar.h>
-#include <qtabwidget.h>
-#include <qtoolbar.h>
-#include <qtoolbutton.h>
-#include <qwidget.h>
-
-// Ugly hack to be able to access QMenuItem::is_enabled
-#define private public
-#include <qmenudata.h>
-#undef private
-
-#include <kaboutdata.h>
-#include <kapplication.h>
-#include <kcmdlineargs.h>
-#include <kconfig.h>
-#include <kglobal.h>
-#include <kmainwindow.h>
-#include <kmenubar.h>
-#include <kstartupinfo.h>
-#include <kstyle.h>
-
-#undef Region
+#include "kdewrapper.h"
 
 #include <salunx.h>
 
Index: automation/source/server/sta_list.cxx
===================================================================
RCS file: /cvs/util/automation/source/server/sta_list.cxx,v
retrieving revision 1.18
retrieving revision 1.18.4.1
diff -u -p -u -p -r1.18 -r1.18.4.1
--- automation/source/server/sta_list.cxx	20 Jun 2006 00:24:33 -0000	1.18
+++ automation/source/server/sta_list.cxx	28 Jun 2006 07:43:33 -0000	1.18.4.1
@@ -992,7 +992,11 @@ String StatementList::ClientTree(Window 
 	WRITEc("UId : ");
 	WRITE(UIdString(pBase->GetSmartUniqueOrHelpId()));
 	WRITEc(":0x");
-    WRITE(String::CreateFromInt32( sal_Int32(pBase), 16 ));
+    WRITE(
+        String::CreateFromInt64(
+            sal::static_int_cast< sal_Int64 >(
+                reinterpret_cast< sal_IntPtr >(pBase)),
+            16 ));
 	WRITEc(":");
 	WRITE(pBase->GetQuickHelpText());
 	WRITEc(":");
Index: automation/source/testtool/objtest.cxx
===================================================================
RCS file: /cvs/util/automation/source/testtool/objtest.cxx,v
retrieving revision 1.25
retrieving revision 1.25.4.2
diff -u -p -u -p -r1.25 -r1.25.4.2
--- automation/source/testtool/objtest.cxx	20 Jun 2006 00:27:12 -0000	1.25
+++ automation/source/testtool/objtest.cxx	30 Jun 2006 08:18:58 -0000	1.25.4.2
@@ -1731,6 +1731,11 @@ void TestToolObj::SFX_NOTIFY( SfxBroadca
     			case ID_StartUse:
     				if ( !rPar )  // rPar = NULL  <=>  Kein Parameter
     				{
+                        {
+    					    BasicRuntime aRun = BasicRuntimeAccess::GetRuntime();
+    					    aLogFileName = DirEntry(aRun.GetModuleName(SbxNAME_NONE)).GetBase().AppendAscii(".res");
+                        }
+
     					ADD_RUN_LOG();
     					ADD_CASE_LOG(GEN_RES_STR0(S_READING_FILE));
 
@@ -2066,11 +2071,11 @@ void TestToolObj::SFX_NOTIFY( SfxBroadca
     				}
     				break;
     			case ID_MaybeAddErr:
-    				if ( ((StarBASIC*)GetParent())->GetErr() && ( !IS_ERROR() ||
-    					 pFehlerListe->GetObject(pFehlerListe->Count()-1)->nError != ((StarBASIC*)GetParent())->GetErr() ) )
+    				if ( ((StarBASIC*)GetParent())->GetErrBasic() && ( !IS_ERROR() ||
+    					 pFehlerListe->GetObject(pFehlerListe->Count()-1)->nError != ((StarBASIC*)GetParent())->GetErrBasic() ) )
     				{
-    					((StarBASIC*)GetParent())->MakeErrorText(((StarBASIC*)GetParent())->GetErr(),String());
-    					ADD_ERROR_QUIET(((StarBASIC*)GetParent())->GetErr() , ((StarBASIC*)GetParent())->GetErrorText())
+    					((StarBASIC*)GetParent())->MakeErrorText(((StarBASIC*)GetParent())->GetErrBasic(),String());
+    					ADD_ERROR_QUIET(((StarBASIC*)GetParent())->GetErrBasic() , ((StarBASIC*)GetParent())->GetErrorText())
     				}
     				break;
     			case ID_GetNextCloseWindow:
@@ -4193,4 +4198,3 @@ String TTFormat::ms2s( ULONG nMilliSecon
 }
 
 
-
Index: tools/source/inet/inetmsg.cxx
===================================================================
RCS file: /cvs/util/tools/source/inet/inetmsg.cxx,v
retrieving revision 1.8
retrieving revision 1.8.6.1
diff -u -p -u -p -r1.8 -r1.8.6.1
--- tools/source/inet/inetmsg.cxx	19 Jun 2006 13:46:21 -0000	1.8
+++ tools/source/inet/inetmsg.cxx	27 Jun 2006 11:30:46 -0000	1.8.6.1
@@ -1390,9 +1390,11 @@ BOOL INetMIMEMessage::EnableAttachChild 
         // Generate a unique boundary from current time.
         sal_Char sTail[16 + 1];
         Time aCurTime;
+        sal_uInt64 nThis = reinterpret_cast< sal_uIntPtr >( this ); // we can be on a 64bit architecture
+        nThis = ( ( nThis >> 32 ) ^ nThis ) & SAL_MAX_UINT32;
         sprintf (sTail, "%08X%08X",
                  static_cast< unsigned int >(aCurTime.GetTime()),
-                 reinterpret_cast< unsigned int >(this));
+                 static_cast< unsigned int >(nThis));
         m_aBoundary = "------------_4D48";
         m_aBoundary += sTail;
 
Index: bridges/source/cpp_uno/gcc3_linux_x86-64/uno2cpp.cxx
===================================================================
RCS file: /cvs/udk/bridges/source/cpp_uno/gcc3_linux_x86-64/uno2cpp.cxx,v
retrieving revision 1.4
retrieving revision 1.4.20.1
diff -u -p -u -p -r1.4 -r1.4.20.1
--- bridges/source/cpp_uno/gcc3_linux_x86-64/uno2cpp.cxx	8 Mar 2006 08:51:19 -0000	1.4
+++ bridges/source/cpp_uno/gcc3_linux_x86-64/uno2cpp.cxx	27 Jun 2006 10:54:34 -0000	1.4.20.1
@@ -514,9 +514,11 @@ static void cpp_call(
 
 
 //==================================================================================================
-void bridges::cpp_uno::shared::UnoInterfaceProxy::dispatch(
+namespace bridges { namespace cpp_uno { namespace shared {
+
+void unoInterfaceProxyDispatch(
 	uno_Interface * pUnoI, const typelib_TypeDescription * pMemberDescr,
-	void * pReturn, void * pArgs[], uno_Any ** ppException ) SAL_THROW(())
+	void * pReturn, void * pArgs[], uno_Any ** ppException )
 {
 	// is my surrogate
 	bridges::cpp_uno::shared::UnoInterfaceProxy * pThis
@@ -643,3 +645,5 @@ void bridges::cpp_uno::shared::UnoInterf
 	}
 	}
 }
+
+} } }
Index: bridges/source/cpp_uno/gcc3_macosx_intel/uno2cpp.cxx
===================================================================
RCS file: /cvs/udk/bridges/source/cpp_uno/gcc3_macosx_intel/uno2cpp.cxx,v
retrieving revision 1.2
retrieving revision 1.2.26.1
diff -u -p -u -p -r1.2 -r1.2.26.1
--- bridges/source/cpp_uno/gcc3_macosx_intel/uno2cpp.cxx	3 Feb 2006 17:35:54 -0000	1.2
+++ bridges/source/cpp_uno/gcc3_macosx_intel/uno2cpp.cxx	28 Jun 2006 07:25:09 -0000	1.2.26.1
@@ -342,9 +342,11 @@ static void cpp_call(
 }
 
 //==================================================================================================
-void bridges::cpp_uno::shared::UnoInterfaceProxy::dispatch(
+
+namespace bridges { namespace cpp_uno { namespace shared {
+void unoInterfaceProxyDispatch(
 	uno_Interface * pUnoI, const typelib_TypeDescription * pMemberDescr,
-    void * pReturn, void * pArgs[], uno_Any ** ppException ) SAL_THROW(())
+	void * pReturn, void * pArgs[], uno_Any ** ppException )
 {
 	// is my surrogate
 	bridges::cpp_uno::shared::UnoInterfaceProxy * pThis
@@ -461,3 +463,5 @@ void bridges::cpp_uno::shared::UnoInterf
 	}
 	}
 }
+
+} } }
Index: transex3/source/export.cxx
===================================================================
RCS file: /cvs/l10n/transex3/source/export.cxx,v
retrieving revision 1.50
retrieving revision 1.50.2.1
diff -u -p -u -p -r1.50 -r1.50.2.1
--- transex3/source/export.cxx	19 Jun 2006 17:21:29 -0000	1.50
+++ transex3/source/export.cxx	27 Jun 2006 18:08:45 -0000	1.50.2.1
@@ -561,7 +561,7 @@ int Export::Execute( int nToken, const c
 /*	printf("+---------------\n");
 	printf("sToken = %s\n",sToken.GetBuffer());
 	printf("nToken = %d\n",nToken);
-	printf("+---------------\n");*/
+	printf("+---------------\n"); */
     BOOL bWriteToMerged = bMergeMode;
 
 	if ( nToken == CONDITION ) {
@@ -2350,7 +2350,9 @@ void Export::MergeRest( ResData *pResDat
 							nMaxIndex = pList->GetGermanEntryCount();
 						//if( pMergeDataFile ) pMergeDataFile->Dump();
 						pEntrys = pMergeDataFile->GetPFormEntrys( pResData );
-                        while( pEntrys && ( nLIndex < nMaxIndex )) {
+                        while( pEntrys  && ( nLIndex < nMaxIndex )) {
+						 
+							//while( pEntrys && ( nLIndex < nMaxIndex )) {
 							//if( pEntrys ) pEntrys->Dump();
 							ByteString sText;
                             BOOL bText;
@@ -2451,6 +2453,7 @@ void Export::MergeRest( ResData *pResDat
 									sText1 += " ;\n";
 								sText1 += sSpace;
 								sText1 += "\t";
+								//printf("Writing '%s'\n",sText1.GetBuffer());
 								WriteToMerged( sText1 ,true );
 							   
 								// Set matching pairedlist identifier 
@@ -2468,6 +2471,7 @@ void Export::MergeRest( ResData *pResDat
 							else
 								break;
 							nLIndex ++;
+							pEntrys = pMergeDataFile->GetPFormEntrys( pResData );
 						}
 						if ( nIdx > 1 ) {
 							ByteString sFooter( sSpace.Copy( 1 ));
@@ -2520,7 +2524,8 @@ void Export::MergeRest( ResData *pResDat
 					sLine.SearchAndReplace( "\"", "< \"" );
 				}
 			}
-			while( PrepareTextToMerge( sLine, nList, nListLang, pResData ) && ( nListIndex <= nMaxIndex )) {
+
+			while( PrepareTextToMerge( sLine, nList, nListLang, pResData ) && ( nListIndex <= nMaxIndex )) { 
 				ByteString sText( "\t" );
 				sText += sLine;
 				sText += " ;";
Index: shell/source/backends/kdebe/kdeinetlayer.hxx
===================================================================
RCS file: /cvs/gsl/shell/source/backends/kdebe/kdeinetlayer.hxx,v
retrieving revision 1.2
retrieving revision 1.2.10.1
diff -u -p -u -p -r1.2 -r1.2.10.1
--- shell/source/backends/kdebe/kdeinetlayer.hxx	2 Jun 2006 12:26:08 -0000	1.2
+++ shell/source/backends/kdebe/kdeinetlayer.hxx	27 Jun 2006 13:57:30 -0000	1.2.10.1
@@ -27,9 +27,7 @@
 #include <cppuhelper/implbase2.hxx>
 #endif
 
-#ifndef QSTRING_H
-#include <qstring.h>
-#endif
+#include "qstring_wrapper.h"
 
 namespace css = com::sun::star ;
 namespace uno = css::uno ;
Index: shell/source/backends/kdebe/qstring_wrapper.h
===================================================================
RCS file: shell/source/backends/kdebe/qstring_wrapper.h
diff -N shell/source/backends/kdebe/qstring_wrapper.h
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ shell/source/backends/kdebe/qstring_wrapper.h	27 Jun 2006 13:59:34 -0000	1.1.2.1
@@ -0,0 +1,49 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2006 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef INCLUDED_SHELL_SOURCE_BACKENDS_KDEBE_QSTRING_WRAPPER_H
+#define INCLUDED_SHELL_SOURCE_BACKENDS_KDEBE_QSTRING_WRAPPER_H
+
+#ifndef _SAL_CONFIG_H_
+#include "sal/config.h"
+#endif
+
+#if defined __GNUC__
+#pragma GCC system_header
+#endif
+
+#include <qstring.h>
+
+#endif
Index: sot/source/sdstor/ucbstorage.cxx
===================================================================
RCS file: /cvs/util/sot/source/sdstor/ucbstorage.cxx,v
retrieving revision 1.90
retrieving revision 1.90.2.1
diff -u -p -u -p -r1.90 -r1.90.2.1
--- sot/source/sdstor/ucbstorage.cxx	20 Jun 2006 05:56:00 -0000	1.90
+++ sot/source/sdstor/ucbstorage.cxx	27 Jun 2006 10:01:10 -0000	1.90.2.1
@@ -554,7 +554,7 @@ public:
 																		// this class is close to be unusable
 																		// since it can not read and write
     using SvStream::SetError;
-    void                        SetError( long nError );
+    void                        SetError( sal_uInt32 nError );
 	void						PrepareCachedForReopen( StreamMode nMode );
 };
 
@@ -1216,7 +1216,7 @@ void  UCBStorageStream_Impl::FlushData()
     m_bCommited = TRUE;
 }
 
-void UCBStorageStream_Impl::SetError( long nErr )
+void UCBStorageStream_Impl::SetError( sal_uInt32 nErr )
 {
     if ( !m_nError )
     {
Index: sfx2/source/dialog/acccfg.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/dialog/acccfg.cxx,v
retrieving revision 1.30
retrieving revision 1.30.4.1
diff -u -p -u -p -r1.30 -r1.30.4.1
--- sfx2/source/dialog/acccfg.cxx	19 Jun 2006 22:19:38 -0000	1.30
+++ sfx2/source/dialog/acccfg.cxx	28 Jun 2006 10:48:55 -0000	1.30.4.1
@@ -670,7 +670,7 @@ void SfxAcceleratorConfigPage::Init(cons
         if (!sKey.Len())
             continue;
         TAccInfo*    pEntry   = new TAccInfo(i1, nListPos, aKey);
-        SvLBoxEntry* pLBEntry = aEntriesBox.InsertEntry(sKey, 0L, LIST_APPEND, 0xFFFF);
+        SvLBoxEntry* pLBEntry = aEntriesBox.InsertEntryToColumn(sKey, 0L, LIST_APPEND, 0xFFFF);
         pLBEntry->SetUserData(pEntry);
     }
 
@@ -768,7 +768,7 @@ void SfxAcceleratorConfigPage::ResetConf
 String FileDialog_Impl( Window* /*pParent*/, WinBits nBits, const String& rTitle )
 {
 	BOOL bSave = ( ( nBits & WB_SAVEAS ) == WB_SAVEAS );
-    short nDialogType = bSave? ::sfx2::FILESAVE_SIMPLE : ::sfx2::FILEOPEN_SIMPLE;
+    short nDialogType = bSave? short(::sfx2::FILESAVE_SIMPLE) : short(::sfx2::FILEOPEN_SIMPLE);
 
     sfx2::FileDialogHelper aFileDlg( nDialogType, 0 );
 	aFileDlg.SetTitle( rTitle );
@@ -1073,7 +1073,7 @@ IMPL_LINK( SfxAcceleratorConfigPage, Sel
                 )
                 {
                     TAccInfo*    pU1 = new TAccInfo(-1, -1, pUserData->m_aKey);
-				    SvLBoxEntry* pE1 = aKeyBox.InsertEntry( pUserData->m_aKey.GetName(), 0, LIST_APPEND, 0xFFFF);
+				    SvLBoxEntry* pE1 = aKeyBox.InsertEntry( pUserData->m_aKey.GetName(), 0L, TRUE, LIST_APPEND );
                     pE1->SetUserData(pU1);
                     pE1->EnableChildsOnDemand( FALSE );
                 }
Index: sfx2/source/dialog/mgetempl.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/dialog/mgetempl.cxx,v
retrieving revision 1.9
retrieving revision 1.9.4.1
diff -u -p -u -p -r1.9 -r1.9.4.1
--- sfx2/source/dialog/mgetempl.cxx	19 Jun 2006 22:23:08 -0000	1.9
+++ sfx2/source/dialog/mgetempl.cxx	27 Jun 2006 08:52:59 -0000	1.9.4.1
@@ -296,6 +296,8 @@ SfxManageStyleSheetPage::~SfxManageStyle
 */
 
 {
+    aNameEd.SetGetFocusHdl( Link() );
+    aNameEd.SetLoseFocusHdl( Link() );
 	delete pFamilies;
 	pItem = 0;
 	pStyle = 0;
Index: sal/osl/unx/thread.c
===================================================================
RCS file: /cvs/porting/sal/osl/unx/thread.c,v
retrieving revision 1.27
retrieving revision 1.27.4.1
diff -u -p -u -p -r1.27 -r1.27.4.1
--- sal/osl/unx/thread.c	20 Jun 2006 04:19:58 -0000	1.27
+++ sal/osl/unx/thread.c	27 Jun 2006 17:56:42 -0000	1.27.4.1
@@ -597,14 +597,11 @@ static sal_uInt16 LastIdent = 0;
 
 static sal_uInt16 lookupThreadId (pthread_t hThread)
 {
-	int		  i, n;
 	HashEntry *pEntry;
 
 	pthread_mutex_lock(&HashLock);
 
-	for (n = 0, i = HASHID(hThread); n < HashSize; n++, i = ++i % HashSize)
-	{
-		pEntry = HashTable[i];
+		pEntry = HashTable[HASHID(hThread)];
 		while (pEntry != NULL)
 		{
 			if (pthread_equal(pEntry->Handle, hThread))
@@ -614,7 +611,6 @@ static sal_uInt16 lookupThreadId (pthrea
 			}
 			pEntry = pEntry->Next;
 		}
-	}
 
 	pthread_mutex_unlock(&HashLock);
 
