Index: basebmp/prj/build.lst
===================================================================
RCS file: /cvs/gsl/basebmp/prj/build.lst,v
retrieving revision 1.3
retrieving revision 1.3.4.1
diff -u -u -p -r1.3 -r1.3.4.1
--- basebmp/prj/build.lst	27 Jun 2007 12:41:44 -0000	1.3
+++ basebmp/prj/build.lst	26 Jul 2007 08:41:30 -0000	1.3.4.1
@@ -1,4 +1,4 @@
-bx	basebmp	:	sal vigra basegfx BOOST:boost NULL
+bx	basebmp	:	sal VIGRA:vigra basegfx BOOST:boost NULL
 bx	basebmp									usr1	-	all	bx_mkout NULL
 bx	basebmp\source							nmake	-	all	bx_source NULL
 bx	basebmp\util							nmake	-	all	bx_util bx_source NULL
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.213
retrieving revision 1.213.6.6
diff -u -u -p -r1.213 -r1.213.6.6
--- config_office/configure.in	3 Jul 2007 12:15:43 -0000	1.213
+++ config_office/configure.in	26 Jul 2007 11:54:49 -0000	1.213.6.6
@@ -370,6 +370,9 @@ AC_ARG_WITH(system-curl,
 AC_ARG_WITH(system-boost,
 [  --with-system-boost     Use boost already on system
 ],,)
+AC_ARG_WITH(system-vigra,
+[  --with-system-vigra     Use vigra already on system
+],,)
 AC_ARG_WITH(system-nas,
 [  --with-system-nas       Use nas already on system
 ],,)
@@ -3414,6 +3407,25 @@ fi
 AC_SUBST(SYSTEM_BOOST)
 
 dnl ===================================================================
+dnl Check for system vigra
+dnl ===================================================================
+AC_MSG_CHECKING([which vigra to use])
+if test -n "$with_system_vigra" && test "$with_system_vigra" != "no"; then
+   AC_MSG_RESULT([external])
+   SYSTEM_VIGRA=YES
+   AC_LANG_SAVE
+   AC_LANG_CPLUSPLUS
+   AC_CHECK_HEADER(vigra/copyimage.hxx, [],
+       [AC_MSG_ERROR(vigra/copyimage.hxx not found. install vigra)], [])
+   AC_LANG_RESTORE
+else
+   AC_MSG_RESULT([internal])
+   BUILD_TYPE="$BUILD_TYPE VIGRA"
+   SYSTEM_VIGRA=NO
+fi
+AC_SUBST(SYSTEM_VIGRA)
+
+dnl ===================================================================
 dnl Check for system odbc
 dnl ===================================================================
 AC_MSG_CHECKING([which odbc headers to use])
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.152
retrieving revision 1.152.8.2
diff -u -u -p -r1.152 -r1.152.8.2
--- config_office/set_soenv.in	27 Jun 2007 17:52:06 -0000	1.152
+++ config_office/set_soenv.in	26 Jul 2007 08:40:02 -0000	1.152.8.2
@@ -1834,6 +1834,7 @@ ToFile( "SYSTEM_CURL",       "@SYSTEM_CU
 ToFile( "CURL_CFLAGS",       "@CURL_CFLAGS@",      "e" );
 ToFile( "CURL_LIBS",         "@CURL_LIBS@",        "e" );
 ToFile( "SYSTEM_BOOST",      "@SYSTEM_BOOST@",     "e" );
+ToFile( "SYSTEM_VIGRA",      "@SYSTEM_VIGRA@",     "e" );
 ToFile( "SYSTEM_NAS",        "@SYSTEM_NAS@",       "e" );
 ToFile( "NAS_INCLUDES",      "@NAS_INCLUDES@",     "e" );
 ToFile( "ENABLE_NAS",        "@ENABLE_NAS@",       "e" );
