cvs diff: Diffing helpcontent2
Index: helpcontent2/settings.pmk
===================================================================
RCS file: /cvs/documentation/helpcontent2/settings.pmk,v
retrieving revision 1.4
retrieving revision 1.4.2.1
diff -u -p -r1.4 -r1.4.2.1
--- helpcontent2/settings.pmk	11 Apr 2005 09:29:48 -0000	1.4
+++ helpcontent2/settings.pmk	20 Apr 2005 16:59:01 -0000	1.4.2.1
@@ -1,10 +1,10 @@
 
-my_cp=$(CLASSPATH)$(PATH_SEPERATOR)$(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/unoil.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/ridl.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/jurt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/jut.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xmlsearch.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xmlhelp.jar
+my_cp:=$(CLASSPATH)$(PATH_SEPERATOR)$(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/unoil.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/ridl.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/jurt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/jut.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xmlsearch.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/xmlhelp.jar
 
 .IF "$(SYSTEM_DB)" == "YES"
-my_cp:=$(my_cp)$(PATH_SEPERATOR)$(DB_JAR)
+my_cp!:=$(my_cp)$(PATH_SEPERATOR)$(DB_JAR)
 .ELSE
-my_cp:=$(my_cp)$(PATH_SEPERATOR)$(SOLARBINDIR)$/db.jar
+my_cp!:=$(my_cp)$(PATH_SEPERATOR)$(SOLARBINDIR)$/db.jar
 .ENDIF
 
 .INCLUDE : $(INCCOM)$/aux_langs.mk
cvs diff: Diffing helpcontent2/helpers
cvs diff: Diffing helpcontent2/helpers/helpauthoring
cvs diff: Diffing helpcontent2/helpers/helpauthoring/filter
cvs diff: Diffing helpcontent2/helpers/helpauthoring/guide
cvs diff: Diffing helpcontent2/helpers/helpauthoring/guide/img
cvs diff: Diffing helpcontent2/helpers/helpauthoring/macros
cvs diff: Diffing helpcontent2/helpers/helpauthoring/macros/HelpAuthoring
cvs diff: Diffing helpcontent2/helpers/helpdebug
cvs diff: Diffing helpcontent2/prj
cvs diff: Diffing helpcontent2/source
cvs diff: Diffing helpcontent2/source/auxiliary
cvs diff: Diffing helpcontent2/source/auxiliary/de
cvs diff: Diffing helpcontent2/source/auxiliary/en-US
cvs diff: Diffing helpcontent2/source/auxiliary/es
cvs diff: Diffing helpcontent2/source/auxiliary/fr
cvs diff: Diffing helpcontent2/source/auxiliary/it
cvs diff: Diffing helpcontent2/source/auxiliary/ja
cvs diff: Diffing helpcontent2/source/auxiliary/ko
cvs diff: Diffing helpcontent2/source/auxiliary/nl
cvs diff: Diffing helpcontent2/source/auxiliary/pt-BR
cvs diff: Diffing helpcontent2/source/auxiliary/sv
cvs diff: Diffing helpcontent2/source/auxiliary/zh-CN
cvs diff: Diffing helpcontent2/source/auxiliary/zh-TW
cvs diff: Diffing helpcontent2/source/text
cvs diff: Diffing helpcontent2/source/text/sbasic
cvs diff: Diffing helpcontent2/source/text/sbasic/guide
cvs diff: Diffing helpcontent2/source/text/sbasic/shared
cvs diff: Diffing helpcontent2/source/text/sbasic/shared/01
cvs diff: Diffing helpcontent2/source/text/sbasic/shared/02
cvs diff: Diffing helpcontent2/source/text/scalc
cvs diff: Diffing helpcontent2/source/text/scalc/00
cvs diff: Diffing helpcontent2/source/text/scalc/01
cvs diff: Diffing helpcontent2/source/text/scalc/02
cvs diff: Diffing helpcontent2/source/text/scalc/04
cvs diff: Diffing helpcontent2/source/text/scalc/05
cvs diff: Diffing helpcontent2/source/text/scalc/guide
cvs diff: Diffing helpcontent2/source/text/schart
cvs diff: Diffing helpcontent2/source/text/schart/00
cvs diff: Diffing helpcontent2/source/text/schart/01
cvs diff: Diffing helpcontent2/source/text/schart/02
cvs diff: Diffing helpcontent2/source/text/schart/04
cvs diff: Diffing helpcontent2/source/text/sdraw
cvs diff: Diffing helpcontent2/source/text/sdraw/00
cvs diff: Diffing helpcontent2/source/text/sdraw/01
cvs diff: Diffing helpcontent2/source/text/sdraw/04
cvs diff: Diffing helpcontent2/source/text/sdraw/guide
cvs diff: Diffing helpcontent2/source/text/shared
cvs diff: Diffing helpcontent2/source/text/shared/00
cvs diff: Diffing helpcontent2/source/text/shared/01
cvs diff: Diffing helpcontent2/source/text/shared/02
cvs diff: Diffing helpcontent2/source/text/shared/04
cvs diff: Diffing helpcontent2/source/text/shared/05
cvs diff: Diffing helpcontent2/source/text/shared/07
cvs diff: Diffing helpcontent2/source/text/shared/autokorr
cvs diff: Diffing helpcontent2/source/text/shared/autopi
cvs diff: Diffing helpcontent2/source/text/shared/explorer
cvs diff: Diffing helpcontent2/source/text/shared/explorer/database
cvs diff: Diffing helpcontent2/source/text/shared/guide
cvs diff: Diffing helpcontent2/source/text/shared/optionen
cvs diff: Diffing helpcontent2/source/text/simpress
cvs diff: Diffing helpcontent2/source/text/simpress/00
cvs diff: Diffing helpcontent2/source/text/simpress/01
cvs diff: Diffing helpcontent2/source/text/simpress/02
cvs diff: Diffing helpcontent2/source/text/simpress/04
cvs diff: Diffing helpcontent2/source/text/simpress/guide
cvs diff: Diffing helpcontent2/source/text/smath
cvs diff: Diffing helpcontent2/source/text/smath/00
cvs diff: Diffing helpcontent2/source/text/smath/01
cvs diff: Diffing helpcontent2/source/text/smath/02
cvs diff: Diffing helpcontent2/source/text/smath/04
cvs diff: Diffing helpcontent2/source/text/smath/guide
cvs diff: Diffing helpcontent2/source/text/swriter
cvs diff: Diffing helpcontent2/source/text/swriter/00
cvs diff: Diffing helpcontent2/source/text/swriter/01
cvs diff: Diffing helpcontent2/source/text/swriter/02
cvs diff: Diffing helpcontent2/source/text/swriter/04
cvs diff: Diffing helpcontent2/source/text/swriter/guide
cvs diff: Diffing helpcontent2/util
cvs diff: Diffing helpcontent2/util/sbasic
cvs diff: Diffing helpcontent2/util/scalc
cvs diff: Diffing helpcontent2/util/schart
cvs diff: Diffing helpcontent2/util/sdatabase
cvs diff: Diffing helpcontent2/util/sdraw
cvs diff: Diffing helpcontent2/util/shared
cvs diff: Diffing helpcontent2/util/simpress
cvs diff: Diffing helpcontent2/util/smath
cvs diff: Diffing helpcontent2/util/swriter
cvs diff: Diffing officecfg
cvs diff: Diffing officecfg/org
cvs diff: Diffing officecfg/org/openoffice
cvs diff: Diffing officecfg/org/openoffice/configuration
cvs diff: Diffing officecfg/org/openoffice/helper
cvs diff: Diffing officecfg/prj
cvs diff: Diffing officecfg/registry
cvs diff: Diffing officecfg/registry/data
cvs diff: Diffing officecfg/registry/data/org
cvs diff: Diffing officecfg/registry/data/org/openoffice
cvs diff: Diffing officecfg/registry/data/org/openoffice/Office
cvs diff: Diffing officecfg/registry/data/org/openoffice/Office/UI
Index: officecfg/registry/data/org/openoffice/Office/UI/DbuCommands.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/UI/DbuCommands.xcu,v
retrieving revision 1.9
retrieving revision 1.9.2.1
diff -u -p -r1.9 -r1.9.2.1
--- officecfg/registry/data/org/openoffice/Office/UI/DbuCommands.xcu	18 Apr 2005 08:38:09 -0000	1.9
+++ officecfg/registry/data/org/openoffice/Office/UI/DbuCommands.xcu	20 Apr 2005 17:10:48 -0000	1.9.2.1
@@ -122,7 +122,7 @@
 			</node>
 			<node oor:name=".uno:DBTableRename" oor:op="replace">
 				<prop oor:name="Label" oor:type="xs:string">
-					<value xml:lang="de">Umbennen...</value>
+					<value xml:lang="de">Umbenennen...</value>
 					<value xml:lang="en-US">Rename...</value>
 				</prop>
 				<prop oor:name="Properties" oor:type="xs:int">
@@ -158,7 +158,7 @@
 			</node>
 			<node oor:name=".uno:DBQueryRename" oor:op="replace">
 				<prop oor:name="Label" oor:type="xs:string">
-					<value xml:lang="de">Umbennen...</value>
+					<value xml:lang="de">Umbenennen...</value>
 					<value xml:lang="en-US">Rename...</value>
 				</prop>
 				<prop oor:name="Properties" oor:type="xs:int">
@@ -194,7 +194,7 @@
 			</node>
 			<node oor:name=".uno:DBFormRename" oor:op="replace">
 				<prop oor:name="Label" oor:type="xs:string">
-					<value xml:lang="de">Umbennen...</value>
+					<value xml:lang="de">Umbenennen...</value>
 					<value xml:lang="en-US">Rename...</value>
 				</prop>
 				<prop oor:name="Properties" oor:type="xs:int">
@@ -230,7 +230,7 @@
 			</node>
 			<node oor:name=".uno:DBReportRename" oor:op="replace">
 				<prop oor:name="Label" oor:type="xs:string">
-					<value xml:lang="de">Umbennen...</value>
+					<value xml:lang="de">Umbenennen...</value>
 					<value xml:lang="en-US">Rename...</value>
 				</prop>
 				<prop oor:name="Properties" oor:type="xs:int">
cvs diff: Diffing officecfg/registry/data/org/openoffice/TypeDetection
cvs diff: Diffing officecfg/registry/data/org/openoffice/ucb
cvs diff: Diffing officecfg/registry/schema
cvs diff: Diffing officecfg/registry/schema/org
cvs diff: Diffing officecfg/registry/schema/org/openoffice
cvs diff: Diffing officecfg/registry/schema/org/openoffice/Office
cvs diff: Diffing officecfg/registry/schema/org/openoffice/Office/UI
cvs diff: Diffing officecfg/registry/schema/org/openoffice/TypeDetection
cvs diff: Diffing officecfg/registry/schema/org/openoffice/ucb
cvs diff: Diffing officecfg/util
cvs diff: Diffing xmloff
cvs diff: Diffing xmloff/dtd
cvs diff: Diffing xmloff/inc
cvs diff: Diffing xmloff/inc/xmloff
cvs diff: Diffing xmloff/prj
cvs diff: Diffing xmloff/qa
cvs diff: Diffing xmloff/qa/unoapi
cvs diff: Diffing xmloff/qa/unoapi/testdocuments
cvs diff: Diffing xmloff/source
cvs diff: Diffing xmloff/source/chart
cvs diff: Diffing xmloff/source/core
cvs diff: Diffing xmloff/source/draw
Index: xmloff/source/draw/xexptran.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/draw/xexptran.cxx,v
retrieving revision 1.17
retrieving revision 1.17.4.1
diff -u -p -r1.17 -r1.17.4.1
--- xmloff/source/draw/xexptran.cxx	13 Apr 2005 09:29:54 -0000	1.17
+++ xmloff/source/draw/xexptran.cxx	20 Apr 2005 19:10:35 -0000	1.17.4.1
@@ -448,8 +448,8 @@ void SdXMLImExTransform2D::AddMatrix(con
 const OUString& SdXMLImExTransform2D::GetExportString(const SvXMLUnitConverter& rConv)
 {
 	OUString aNewString;
-	OUString aClosingBrace(String(sal_Unicode(')')));
-	OUString aEmptySpace(String(sal_Unicode(' ')));
+	OUString aClosingBrace(sal_Unicode(')'));
+	OUString aEmptySpace(sal_Unicode(' '));
 	
 	for(sal_uInt32 a(0L); a < maList.Count(); a++)
 	{
@@ -861,8 +861,8 @@ void SdXMLImExTransform3D::AddHomogenMat
 const OUString& SdXMLImExTransform3D::GetExportString(const SvXMLUnitConverter& rConv)
 {
 	OUString aNewString;
-	OUString aClosingBrace(String(sal_Unicode(')')));
-	OUString aEmptySpace(String(sal_Unicode(' ')));
+	OUString aClosingBrace(sal_Unicode(')'));
+	OUString aEmptySpace(sal_Unicode(' '));
 	
 	for(sal_uInt32 a(0L); a < maList.Count(); a++)
 	{
@@ -1267,7 +1267,7 @@ SdXMLImExViewBox::SdXMLImExViewBox(const
 const OUString& SdXMLImExViewBox::GetExportString(const SvXMLUnitConverter& rConv)
 {
 	OUString aNewString;
-	OUString aEmptySpace(String(sal_Unicode(' ')));
+	OUString aEmptySpace(sal_Unicode(' '));
 
 	Imp_PutNumberChar(aNewString, rConv, mnX);
 	aNewString += aEmptySpace;
cvs diff: Diffing xmloff/source/forms
cvs diff: Diffing xmloff/source/meta
cvs diff: Diffing xmloff/source/script
cvs diff: Diffing xmloff/source/style
cvs diff: Diffing xmloff/source/text
cvs diff: Diffing xmloff/source/transform
cvs diff: Diffing xmloff/source/xforms
cvs diff: Diffing xmloff/util
cvs diff: Diffing xmloff/util/defs
cvs diff: Diffing xmloff/xml
cvs diff: Diffing sd
cvs diff: Diffing sd/inc
cvs diff: Diffing sd/prj
cvs diff: Diffing sd/qa
cvs diff: Diffing sd/qa/unoapi
cvs diff: Diffing sd/res
cvs diff: Diffing sd/res/imagelst
cvs diff: Diffing sd/res/webview
cvs diff: Diffing sd/sdi
cvs diff: Diffing sd/source
cvs diff: Diffing sd/source/core
cvs diff: Diffing sd/source/filter
cvs diff: Diffing sd/source/filter/bin
cvs diff: Diffing sd/source/filter/cgm
cvs diff: Diffing sd/source/filter/eppt
cvs diff: Diffing sd/source/filter/grf
cvs diff: Diffing sd/source/filter/html
cvs diff: Diffing sd/source/filter/ppt
cvs diff: Diffing sd/source/filter/xml
cvs diff: Diffing sd/source/ui
cvs diff: Diffing sd/source/ui/accessibility
cvs diff: Diffing sd/source/ui/animations
Index: sd/source/ui/animations/CustomAnimationList.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/animations/CustomAnimationList.cxx,v
retrieving revision 1.7
retrieving revision 1.7.6.2
diff -u -p -r1.7 -r1.7.6.2
--- sd/source/ui/animations/CustomAnimationList.cxx	13 Apr 2005 09:35:53 -0000	1.7
+++ sd/source/ui/animations/CustomAnimationList.cxx	20 Apr 2005 19:18:02 -0000	1.7.6.2
@@ -724,7 +724,7 @@ void CustomAnimationList::update()
 			{
 				SvLBoxEntry* pLBoxEntry = new CustomAnimationListEntry;
 				pLBoxEntry->AddItem( new SvLBoxContextBmp( pLBoxEntry, 0, Image(), Image(), 0));
-				OUString aDescription( String( SdResId( STR_CUSTOMANIMATION_TRIGGER ) ) );
+				OUString aDescription = String( SdResId( STR_CUSTOMANIMATION_TRIGGER ) );
 				aDescription += OUString( RTL_CONSTASCII_USTRINGPARAM(": ") );
 				aDescription += getShapeDescription( xShape, false );
 				pLBoxEntry->AddItem( new CustomAnimationTriggerEntryItem( pLBoxEntry, 0, aDescription, this ) );
cvs diff: Diffing sd/source/ui/app
cvs diff: Diffing sd/source/ui/dlg
cvs diff: Diffing sd/source/ui/docshell
cvs diff: Diffing sd/source/ui/func
cvs diff: Diffing sd/source/ui/inc
cvs diff: Diffing sd/source/ui/inc/taskpane
cvs diff: Diffing sd/source/ui/inc/tools
cvs diff: Diffing sd/source/ui/notes
cvs diff: Diffing sd/source/ui/slideshow
cvs diff: Diffing sd/source/ui/slidesorter
cvs diff: Diffing sd/source/ui/slidesorter/cache
cvs diff: Diffing sd/source/ui/slidesorter/controller
cvs diff: Diffing sd/source/ui/slidesorter/inc
cvs diff: Diffing sd/source/ui/slidesorter/inc/cache
cvs diff: Diffing sd/source/ui/slidesorter/inc/controller
cvs diff: Diffing sd/source/ui/slidesorter/inc/model
cvs diff: Diffing sd/source/ui/slidesorter/inc/view
cvs diff: Diffing sd/source/ui/slidesorter/model
cvs diff: Diffing sd/source/ui/slidesorter/shell
cvs diff: Diffing sd/source/ui/slidesorter/view
cvs diff: Diffing sd/source/ui/toolpanel
cvs diff: Diffing sd/source/ui/toolpanel/controls
cvs diff: Diffing sd/source/ui/tools
cvs diff: Diffing sd/source/ui/unoidl
cvs diff: Diffing sd/source/ui/view
cvs diff: Diffing sd/source/unotypes
cvs diff: Diffing sd/uiconfig
cvs diff: Diffing sd/uiconfig/sdraw
cvs diff: Diffing sd/uiconfig/sdraw/accelerator
cvs diff: Diffing sd/uiconfig/sdraw/accelerator/de
cvs diff: Diffing sd/uiconfig/sdraw/accelerator/en-US
cvs diff: Diffing sd/uiconfig/sdraw/accelerator/es
cvs diff: Diffing sd/uiconfig/sdraw/accelerator/fr
cvs diff: Diffing sd/uiconfig/sdraw/menubar
cvs diff: Diffing sd/uiconfig/sdraw/statusbar
cvs diff: Diffing sd/uiconfig/sdraw/toolbar
cvs diff: Diffing sd/uiconfig/simpress
cvs diff: Diffing sd/uiconfig/simpress/accelerator
cvs diff: Diffing sd/uiconfig/simpress/accelerator/de
cvs diff: Diffing sd/uiconfig/simpress/accelerator/en-US
cvs diff: Diffing sd/uiconfig/simpress/accelerator/es
cvs diff: Diffing sd/uiconfig/simpress/accelerator/fr
cvs diff: Diffing sd/uiconfig/simpress/menubar
cvs diff: Diffing sd/uiconfig/simpress/statusbar
cvs diff: Diffing sd/uiconfig/simpress/toolbar
cvs diff: Diffing sd/util
cvs diff: Diffing sd/util/defs
cvs diff: Diffing sd/workben
cvs diff: Diffing sd/workben/testdll
cvs diff: Diffing sd/xml
cvs diff: Diffing sd/xsl
cvs diff: Diffing svx
cvs diff: Diffing svx/doc
cvs diff: Diffing svx/doc/UML
cvs diff: Diffing svx/inc
cvs diff: Diffing svx/inc/svx
cvs diff: Diffing svx/inc/svx/sdr
cvs diff: Diffing svx/inc/svx/sdr/animation
cvs diff: Diffing svx/inc/svx/sdr/contact
cvs diff: Diffing svx/inc/svx/sdr/event
cvs diff: Diffing svx/inc/svx/sdr/mixer
cvs diff: Diffing svx/inc/svx/sdr/properties
cvs diff: Diffing svx/prj
cvs diff: Diffing svx/qa
cvs diff: Diffing svx/qa/unoapi
cvs diff: Diffing svx/qa/unoapi/testdocuments
cvs diff: Diffing svx/res
cvs diff: Diffing svx/sdi
cvs diff: Diffing svx/source
cvs diff: Diffing svx/source/accessibility
cvs diff: Diffing svx/source/customshapes
cvs diff: Diffing svx/source/dialog
Index: svx/source/dialog/cfg.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/cfg.cxx,v
retrieving revision 1.20
retrieving revision 1.20.4.1
diff -u -p -r1.20 -r1.20.4.1
--- svx/source/dialog/cfg.cxx	18 Apr 2005 11:52:21 -0000	1.20
+++ svx/source/dialog/cfg.cxx	20 Apr 2005 19:22:15 -0000	1.20.4.1
@@ -5362,7 +5362,7 @@ void SvxIconSelectorDialog::ImportGraphi
 
 	if ( rejectedCount != 0 )
 	{
-		OUString message( String( ResId( RID_SVXSTR_IMPORT_ICON_ERROR, DIALOG_MGR() ) ) );
+		OUString message = String( ResId( RID_SVXSTR_IMPORT_ICON_ERROR, DIALOG_MGR() ) );
 		if ( m_nExpectedSize != 16 )
 		{
 			message = replaceSixteen( message, m_nExpectedSize );
cvs diff: Diffing svx/source/editeng
cvs diff: Diffing svx/source/engine3d
cvs diff: Diffing svx/source/fmcomp
Index: svx/source/fmcomp/fmgridcl.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/fmcomp/fmgridcl.cxx,v
retrieving revision 1.48
retrieving revision 1.48.8.1
diff -u -p -r1.48 -r1.48.8.1
--- svx/source/fmcomp/fmgridcl.cxx	13 Apr 2005 08:27:50 -0000	1.48
+++ svx/source/fmcomp/fmgridcl.cxx	20 Apr 2005 19:23:09 -0000	1.48.8.1
@@ -1135,7 +1135,7 @@ void FmGridHeader::PostExecuteColumnCont
 			else
 			{
 				// Standardlabel setzen
-				::rtl::OUString sLabelBase(String(SVX_RES(RID_STR_COLUMN)));
+				::rtl::OUString sLabelBase = String(SVX_RES(RID_STR_COLUMN));
 				// disambiguate the name
 				Reference< XNameAccess > xColNames(xCols, UNO_QUERY);
 				::rtl::OUString sLabel;
cvs diff: Diffing svx/source/form
Index: svx/source/form/fmpgeimp.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/form/fmpgeimp.cxx,v
retrieving revision 1.27
retrieving revision 1.27.8.1
diff -u -p -r1.27 -r1.27.8.1
--- svx/source/form/fmpgeimp.cxx	13 Apr 2005 08:29:22 -0000	1.27
+++ svx/source/form/fmpgeimp.cxx	20 Apr 2005 19:24:00 -0000	1.27.8.1
@@ -364,7 +364,7 @@ Reference< ::com::sun::star::form::XForm
 			{
 			}
 
-			::rtl::OUString aName (String(SVX_RES(RID_STR_STDFORMNAME)));
+			::rtl::OUString aName = String(SVX_RES(RID_STR_STDFORMNAME));
 			xSet->setPropertyValue(FM_PROP_NAME, makeAny(aName));
 
 
cvs diff: Diffing svx/source/gallery2
cvs diff: Diffing svx/source/inc
cvs diff: Diffing svx/source/init
cvs diff: Diffing svx/source/intro
cvs diff: Diffing svx/source/items
cvs diff: Diffing svx/source/mnuctrls
cvs diff: Diffing svx/source/msfilter
cvs diff: Diffing svx/source/options
cvs diff: Diffing svx/source/outliner
cvs diff: Diffing svx/source/sdr
cvs diff: Diffing svx/source/sdr/animation
cvs diff: Diffing svx/source/sdr/contact
cvs diff: Diffing svx/source/sdr/event
cvs diff: Diffing svx/source/sdr/mixer
cvs diff: Diffing svx/source/sdr/properties
cvs diff: Diffing svx/source/src
cvs diff: Diffing svx/source/stbctrls
cvs diff: Diffing svx/source/svdraw
cvs diff: Diffing svx/source/svrtf
cvs diff: Diffing svx/source/svxlink
cvs diff: Diffing svx/source/tbxctrls
cvs diff: Diffing svx/source/toolbars
cvs diff: Diffing svx/source/unodialogs
cvs diff: Diffing svx/source/unodialogs/inc
cvs diff: Diffing svx/source/unodialogs/textconversiondlgs
cvs diff: Diffing svx/source/unodraw
cvs diff: Diffing svx/source/unoedit
cvs diff: Diffing svx/source/unogallery
cvs diff: Diffing svx/source/xml
cvs diff: Diffing svx/source/xoutdev
cvs diff: Diffing svx/uiconfig
cvs diff: Diffing svx/uiconfig/accelerator
cvs diff: Diffing svx/uiconfig/accelerator/en-US
cvs diff: Diffing svx/uiconfig/accelerator/es
cvs diff: Diffing svx/util
cvs diff: Diffing svx/util/defs
cvs diff: Diffing svx/win
cvs diff: Diffing svx/win/res
cvs diff: Diffing svx/workben
cvs diff: Diffing svx/xml
cvs diff: Diffing sfx2
cvs diff: Diffing sfx2/inc
cvs diff: Diffing sfx2/inc/sfx2
cvs diff: Diffing sfx2/prj
cvs diff: Diffing sfx2/qa
cvs diff: Diffing sfx2/qa/complex
cvs diff: Diffing sfx2/qa/complex/DocHelper
cvs diff: Diffing sfx2/qa/complex/docinfo
cvs diff: Diffing sfx2/qa/unoapi
cvs diff: Diffing sfx2/qa/unoapi/testdocuments
cvs diff: Diffing sfx2/res
cvs diff: Diffing sfx2/sdi
cvs diff: Diffing sfx2/source
cvs diff: Diffing sfx2/source/accel
cvs diff: Diffing sfx2/source/appl
cvs diff: Diffing sfx2/source/bastyp
cvs diff: Diffing sfx2/source/config
cvs diff: Diffing sfx2/source/control
cvs diff: Diffing sfx2/source/dialog
cvs diff: Diffing sfx2/source/doc
Index: sfx2/source/doc/guisaveas.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/doc/guisaveas.cxx,v
retrieving revision 1.15
retrieving revision 1.15.6.1
diff -u -p -r1.15 -r1.15.6.1
--- sfx2/source/doc/guisaveas.cxx	13 Apr 2005 12:40:30 -0000	1.15
+++ sfx2/source/doc/guisaveas.cxx	20 Apr 2005 19:19:46 -0000	1.15.6.1
@@ -793,7 +793,7 @@ sal_Bool ModelData_Impl::OutputFileDialo
 
 		if ( xControlAccess.is() )
 		{
-			::rtl::OUString aCtrlText( String( SfxResId( STR_EXPORTBUTTON ) ) );
+			::rtl::OUString aCtrlText = String( SfxResId( STR_EXPORTBUTTON ) );
 			xControlAccess->setLabel( ui::dialogs::CommonFilePickerElementIds::PUSHBUTTON_OK, aCtrlText );
 
 			aCtrlText = ::rtl::OUString( String( SfxResId( STR_LABEL_FILEFORMAT ) ) );
cvs diff: Diffing sfx2/source/explorer
cvs diff: Diffing sfx2/source/inc
cvs diff: Diffing sfx2/source/inet
cvs diff: Diffing sfx2/source/menu
cvs diff: Diffing sfx2/source/notify
cvs diff: Diffing sfx2/source/statbar
cvs diff: Diffing sfx2/source/toolbox
cvs diff: Diffing sfx2/source/view
cvs diff: Diffing sfx2/util
cvs diff: Diffing sfx2/util/defs
cvs diff: Diffing sfx2/win
cvs diff: Diffing sfx2/win/res
