Index: bean/native/unix/makefile.mk
===================================================================
RCS file: /cvs/api/bean/native/unix/makefile.mk,v
retrieving revision 1.6
retrieving revision 1.6.2.2
diff -u -u -r1.6 -r1.6.2.2
--- bean/native/unix/makefile.mk	6 Jun 2005 17:42:10 -0000	1.6
+++ bean/native/unix/makefile.mk	14 Jun 2005 11:36:12 -0000	1.6.2.2
@@ -27,10 +27,9 @@
 
 SHL1TARGET=$(TARGET)
 SHL1LIBS=	$(SLB)$/$(TARGET).lib
+SHL1STDLIBS=$(AWTLIB)
 .IF "$(JDK)" == "gcj"
-SHL1STDLIBS=-lgcjawt -lgcj
-.ELSE
-SHL1STDLIBS=-ljawt
+SHL1STDLIBS+=-lgcj
 .ENDIF
 
 .IF "$(OS)" == "LINUX"
Index: bean/native/win32/makefile.mk
===================================================================
RCS file: /cvs/api/bean/native/win32/makefile.mk,v
retrieving revision 1.2
retrieving revision 1.2.24.1
diff -u -u -r1.2 -r1.2.24.1
--- bean/native/win32/makefile.mk	18 Oct 2004 07:17:33 -0000	1.2
+++ bean/native/win32/makefile.mk	14 Jun 2005 11:36:13 -0000	1.2.24.1
@@ -20,7 +20,7 @@
 SHL1TARGET=$(TARGET)
 SHL1LIBS=	$(SLB)$/$(TARGET).lib
 #SHL1IMPLIB=	i$(TARGET)
-SHL1STDLIBS=jawt.lib $(SALLIB)
+SHL1STDLIBS=$(AWTLIB) $(SALLIB)
 
 SHL1DEF=	$(MISC)$/$(SHL1TARGET).def
 
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.119
retrieving revision 1.119.6.3
diff -u -u -r1.119 -r1.119.6.3
--- config_office/configure.in	31 May 2005 17:03:12 -0000	1.119
+++ config_office/configure.in	14 Jun 2005 11:33:17 -0000	1.119.6.3
@@ -1882,10 +1882,23 @@
 
 if test "$JDK" = "gcj"; then 
    save_CFLAGS=$CFLAGS
+   save_LDFLAGS=$LDFLAGS
    CFLAGS="$CFLAGS -I$JAVA_HOME/include"
+   LDFLAGS="$LDFLAGS -L$JAVA_HOME/lib -lgcj"
    AC_CHECK_HEADER(jni.h, [],
                     [AC_MSG_ERROR([jni.h could not be found. Mismatch between gcc and libgcj or libgcj-devel missing?])], [])
+
+   AC_CHECK_LIB(gcjawt, JAWT_GetAWT, [ AWTLIB=-lgcjawt ] )
+   if test -z "$AWTLIB"; then
+      AC_CHECK_LIB(jawt, JAWT_GetAWT, [ AWTLIB=-ljawt ] )
+   fi
+   if test -z "$AWTLIB"; then
+      AC_MSG_ERROR([either -lgcjawt or -ljawt with JAWT_GetAWT is required to build wih gcj])
+   fi
+   AC_SUBST(AWTLIB)
+
    CFLAGS=$save_CFLAGS
+   LDFLAGS=$save_LDFLAGS
 fi
 
 AC_SUBST(JAVA_HOME)
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.64
retrieving revision 1.64.4.2
diff -u -u -r1.64 -r1.64.4.2
--- config_office/set_soenv.in	31 May 2005 17:03:33 -0000	1.64
+++ config_office/set_soenv.in	14 Jun 2005 11:33:18 -0000	1.64.4.2
@@ -1642,6 +1642,7 @@
 else {
    ToFile( "GCJ_DATABASE",     $GCJ_DATABASE,       "e" );
    ToFile( "JAVACACHE",        '@JAVACACHE@',       "e" );
+   ToFile( "AWTLIB",           '@AWTLIB@',          "e" );
 }
 if ( $platform =~ m/darwin/ )
 {  ToFile( "DYLD_LIBRARY_PATH", $LD_LIBRARY_PATH,   "e" );
Index: filter/source/config/fragments/makefile.mk
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/makefile.mk,v
retrieving revision 1.10
retrieving revision 1.10.8.2
diff -u -u -r1.10 -r1.10.8.2
--- filter/source/config/fragments/makefile.mk	2 Jun 2005 10:00:48 -0000	1.10
+++ filter/source/config/fragments/makefile.mk	14 Jun 2005 11:55:23 -0000	1.10.8.2
@@ -176,7 +176,11 @@
 MERGE:=python ../tools/merge/pyAltFCFGMerge
 .ELSE
 MERGE    := $(JAVA) -jar $(CLASSDIR)$/FCFGMerge.jar
+.IF "$(JDK)" == "gcj"
+PACKLANG := $(JAVA) $(JAVACPS) $(SOLARBINDIR)/xalan.jar org.apache.xalan.xslt.Process -XSL langfilter.xsl
+.ELSE
 PACKLANG := $(JAVA) org.apache.xalan.xslt.Process -XSL langfilter.xsl
+.ENDIF
 .ENDIF
 
 # -----------------------------------------------------------------------------
Index: solenv/inc/libs.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/libs.mk,v
retrieving revision 1.89
retrieving revision 1.89.26.1
diff -u -u -r1.89 -r1.89.26.1
--- solenv/inc/libs.mk	23 May 2005 15:34:17 -0000	1.89
+++ solenv/inc/libs.mk	14 Jun 2005 11:34:40 -0000	1.89.26.1
@@ -70,6 +70,7 @@
 #
 #externe libs in plattform.mk
 #
+AWTLIB*=-ljawt
 AVMEDIALIB=-lavmedia$(OFFICEUPD)$(DLLPOSTFIX)
 ICUINLIB=-licui18n
 ICULELIB=-licule
@@ -298,6 +299,7 @@
 					$(USED_UNO_LIBS)
 
 .ELSE				# "$(GUI)"=="UNX" || "$(GUI)"=="MAC"
+AWTLIB*=jawt.lib
 AVMEDIALIB=$(LIBPRE) iavmedia.lib
 ICUINLIB=icuin.lib
 ICULELIB=icule.lib
