Index: framework/source/loadenv/loadenv.cxx
===================================================================
RCS file: /cvs/framework/framework/source/loadenv/loadenv.cxx,v
retrieving revision 1.28.20.1
retrieving revision 1.28.20.1.2.1
diff -u -u -p -b -w -B -r1.28.20.1 -r1.28.20.1.2.1
--- framework/source/loadenv/loadenv.cxx	8 Feb 2008 10:33:37 -0000	1.28.20.1
+++ framework/source/loadenv/loadenv.cxx	12 Feb 2008 16:07:06 -0000	1.28.20.1.2.1
@@ -398,7 +398,9 @@ css::uno::Reference< css::lang::XCompone
     if (xModel.is ())
     {
         ::comphelper::MediaDescriptor lModelDescriptor(xModel->getArgs());
-        lDescriptor.update (lModelDescriptor);
+        ::comphelper::MediaDescriptor::iterator pIt = lModelDescriptor.find( ::comphelper::MediaDescriptor::PROP_MACROEXECUTIONMODE() );
+		if ( pIt != lModelDescriptor.end() )
+			lDescriptor[::comphelper::MediaDescriptor::PROP_MACROEXECUTIONMODE()] = pIt->second;
     }
 
     return lDescriptor;
