Index: svtools/source/contnr/svimpicn.cxx
===================================================================
RCS file: /cvs/util/svtools/source/contnr/svimpicn.cxx,v
retrieving revision 1.10
retrieving revision 1.8.210.3
diff -u -p -u -p -r1.10 -r1.8.210.3
--- svtools/source/contnr/svimpicn.cxx	18 Jul 2007 08:52:40 -0000	1.10
+++ svtools/source/contnr/svimpicn.cxx	23 Jul 2007 09:28:53 -0000	1.8.210.3
@@ -218,7 +218,7 @@ MultiTextLineInfo::~MultiTextLineInfo()
 {
 	for ( USHORT i = 0; i < mnLines; i++ )
 		delete mpLines[i];
-	delete mpLines;
+	delete [] mpLines;
 }
 
 void MultiTextLineInfo::AddLine( TextLineInfo* pLine )
Index: svtools/source/filter.vcl/wmf/winwmf.cxx
===================================================================
RCS file: /cvs/util/svtools/source/filter.vcl/wmf/winwmf.cxx,v
retrieving revision 1.33
retrieving revision 1.33.188.1
diff -u -p -u -p -r1.33 -r1.33.188.1
--- svtools/source/filter.vcl/wmf/winwmf.cxx	14 Nov 2006 15:42:51 -0000	1.33
+++ svtools/source/filter.vcl/wmf/winwmf.cxx	29 Jun 2007 10:57:47 -0000	1.33.188.1
@@ -886,7 +886,7 @@ void WMFReader::ReadRecordParams( USHORT
 								break;
 							}
 						}
-						delete pData;
+						delete[] pData;
 					}
 				}
 			}
Index: svtools/source/filter.vcl/wmf/wmfwr.cxx
===================================================================
RCS file: /cvs/util/svtools/source/filter.vcl/wmf/wmfwr.cxx,v
retrieving revision 1.28
retrieving revision 1.28.114.1
diff -u -p -u -p -r1.28 -r1.28.114.1
--- svtools/source/filter.vcl/wmf/wmfwr.cxx	29 Jan 2007 14:47:57 -0000	1.28
+++ svtools/source/filter.vcl/wmf/wmfwr.cxx	29 Jun 2007 10:57:47 -0000	1.28.114.1
@@ -1278,7 +1278,7 @@ void WMFWriter::WriteRecords( const GDIM
 					for ( i = 0; i < ( nLen - 1 ); i++ )
 						pDXAry[ i ] = pDXAry[ i ] * (sal_Int32)pA->GetWidth() / nNormSize;
 					if ( ( nLen <= 1 ) || ( (sal_Int32)pA->GetWidth() == nNormSize ) )
-						delete pDXAry, pDXAry = NULL;
+						delete[] pDXAry, pDXAry = NULL;
 					aSrcLineInfo = LineInfo();
 					SetAllAttr();
 					if ( !WMFRecord_Escape_Unicode( pA->GetPoint(), aTemp, pDXAry ) )
Index: libxslt/makefile.mk
===================================================================
RCS file: /cvs/external/libxslt/makefile.mk,v
retrieving revision 1.7
retrieving revision 1.4.4.4
diff -u -p -u -p -r1.7 -r1.4.4.4
--- libxslt/makefile.mk	19 Jul 2007 11:21:19 -0000	1.7
+++ libxslt/makefile.mk	23 Jul 2007 18:00:54 -0000	1.4.4.4
@@ -78,19 +78,29 @@ BUILD_ACTION=nmake
 BUILD_DIR=$(CONFIGURE_DIR)
 .ENDIF
 .ELSE
+
+.IF "$(OS)$(COM)"=="LINUXGCC" || "$(OS)$(COM)"=="FREEBSDGCC"
+LDFLAGS:=-Wl,-rpath,'$$$$ORIGIN' -Wl,-noinhibit-exec -Wl,-z,noexecstack
+.ENDIF                  # "$(OS)$(COM)"=="LINUXGCC"
+.IF "$(OS)$(COM)"=="SOLARISC52"
+LDFLAGS:=-Wl,-R'$$$$ORIGIN'
+.ENDIF                  # "$(OS)$(COM)"=="SOLARISC52"
+
 .IF "$(SYSBASE)"!=""
 xslt_CFLAGS+=-I$(SYSBASE)$/usr$/include -I$(SOLARINCDIR)$/external
 .IF "$(OS)"=="SOLARIS" || "$(OS)"=="LINUX"
-xslt_LDFLAGS+=-L$(SYSBASE)$/lib -L$(SYSBASE)$/usr$/lib -L$(SOLARLIBDIR) -lpthread -ldl
+LDFLAGS+=-L$(SYSBASE)$/lib -L$(SYSBASE)$/usr$/lib -L$(SOLARLIBDIR) -lpthread -ldl
 .ENDIF
 .ENDIF			# "$(SYSBASE)"!=""
+
+.EXPORT: LDFLAGS
+
 .IF "$(COMNAME)"=="sunpro5"
 xslt_CFLAGS+=-xc99=none
 .ENDIF                  # "$(COMNAME)"=="sunpro5"
 CONFIGURE_DIR=
-xslt_LDFLAGS+=-lxml2 -lz
 CONFIGURE_ACTION=chmod 777 libxml2-config && .$/configure
-CONFIGURE_FLAGS=--enable-ipv6=no --without-crypto --without-python --enable-static=no --with-sax1=yes CFLAGS="$(xslt_CFLAGS)" LDFLAGS="$(xslt_LDFLAGS)" 
+CONFIGURE_FLAGS=--enable-ipv6=no --without-crypto --without-python --enable-static=no --with-sax1=yes CFLAGS="$(xslt_CFLAGS)" LIBXML2LIB=$(LIBXML2LIB) ZLIB3RDLIB=$(ZLIB3RDLIB)
 BUILD_ACTION=$(GNUMAKE)
 BUILD_FLAGS+= -j$(EXTMAXPROCESS)
 BUILD_DIR=$(CONFIGURE_DIR)
Index: tools/bootstrp/command.cxx
===================================================================
RCS file: /cvs/util/tools/bootstrp/command.cxx,v
retrieving revision 1.14
retrieving revision 1.13.100.2
diff -u -p -u -p -r1.14 -r1.13.100.2
--- tools/bootstrp/command.cxx	27 Jun 2007 22:05:51 -0000	1.14
+++ tools/bootstrp/command.cxx	5 Jul 2007 12:59:20 -0000	1.13.100.2
@@ -195,6 +195,8 @@ CommandLine& CommandLine::operator=(cons
 		delete [] ppArgv[i];
 	}
 	delete [] ppArgv;
+        ppArgv = new char * [1];
+        ppArgv[0] = NULL;
 	BuildCommand(CommandBuffer);
 	return *this;
 }
@@ -208,6 +210,8 @@ CommandLine& CommandLine::operator=(cons
 		delete [] ppArgv[i];
 	}
 	delete [] ppArgv;
+        ppArgv = new char * [1];
+        ppArgv[0] = NULL;
 	BuildCommand(CommandBuffer);
 
 	return *this;
Index: tools/bootstrp/cppdep.cxx
===================================================================
RCS file: /cvs/util/tools/bootstrp/cppdep.cxx,v
retrieving revision 1.14
retrieving revision 1.13.100.2
diff -u -p -u -p -r1.14 -r1.13.100.2
--- tools/bootstrp/cppdep.cxx	27 Jun 2007 22:06:03 -0000	1.14
+++ tools/bootstrp/cppdep.cxx	5 Jul 2007 12:57:22 -0000	1.13.100.2
@@ -64,6 +64,7 @@ CppDep::CppDep()
 
 CppDep::~CppDep()
 {
+	delete pSources;
 	delete pSearchPath;
 	delete pFileList;
 }
Index: lingucomponent/source/hyphenator/altlinuxhyph/libhnj/hyphen.c
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/libhnj/hyphen.c,v
retrieving revision 1.7
retrieving revision 1.7.30.2
diff -u -p -u -p -r1.7 -r1.7.30.2
--- lingucomponent/source/hyphenator/altlinuxhyph/libhnj/hyphen.c	4 Oct 2006 16:11:00 -0000	1.7
+++ lingucomponent/source/hyphenator/altlinuxhyph/libhnj/hyphen.c	9 Jul 2007 07:47:49 -0000	1.7.30.2
@@ -292,9 +292,11 @@ hnj_hyphen_load (const char *fn)
             if (index) {
                 char * index2 = strchr(index + 1, ',');
                 *index = '\0';
-                *index2 = '\0';
-                replindex = (signed char) atoi(index + 1) - 1;
-                replcut = (signed char) atoi(index2 + 1);                
+				if (index2) {
+                   *index2 = '\0';
+                   replindex = (signed char) atoi(index + 1) - 1;
+                   replcut = (signed char) atoi(index2 + 1);                
+                }
             } else {
                 hnj_strchomp(repl + 1);
                 replindex = 0;
@@ -416,7 +418,8 @@ hnj_hyphen_load (const char *fn)
 #ifndef VERBOSE
   hnj_hash_free (hashtab);
 #endif
-
+  
+  fclose(f);
   return dict;
 }
 
Index: lingucomponent/source/thesaurus/mythes/mythes.cxx
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/thesaurus/mythes/mythes.cxx,v
retrieving revision 1.7
retrieving revision 1.7.36.2
diff -u -p -u -p -r1.7 -r1.7.36.2
--- lingucomponent/source/thesaurus/mythes/mythes.cxx	16 Sep 2006 16:13:38 -0000	1.7
+++ lingucomponent/source/thesaurus/mythes/mythes.cxx	9 Jul 2007 08:44:47 -0000	1.7.36.2
@@ -74,7 +74,6 @@ int MyThes::thInitialize(const char* idx
     // open the index file
     FILE * pifile = fopen(idxpath,"r");
     if (!pifile) {
-        pifile = NULL;
         return 0;
     } 
 
@@ -84,6 +83,7 @@ int MyThes::thInitialize(const char* idx
     if (!wrd) {
        fprintf(stderr,"Error - bad memory allocation\n");
        fflush(stderr);
+       fclose(pifile);
        return 0;
     }
     int len = readLine(pifile,wrd,MAX_WD_LEN);
@@ -99,6 +99,7 @@ int MyThes::thInitialize(const char* idx
     if ( (!(list)) || (!(offst)) ) {
        fprintf(stderr,"Error - bad memory allocation\n");
        fflush(stderr);
+       fclose(pifile);
        return 0;
     }
 
@@ -114,6 +115,7 @@ int MyThes::thInitialize(const char* idx
                 if (!list[nw]) {
                     fprintf(stderr,"Error - bad memory allocation\n");
                     fflush(stderr);
+                    fclose(pifile);
                     return 0;
                 }
                 memcpy((list[nw]),wrd,np);
@@ -126,12 +128,10 @@ int MyThes::thInitialize(const char* idx
 
     free((void *)wrd);
     fclose(pifile);
-    pifile=NULL;
 
     /* next open the data file */
     pdfile = fopen(datpath,"r");
     if (!pdfile) {
-        pdfile = NULL;
         return 0;
     } 
         
@@ -147,17 +147,20 @@ void MyThes::thCleanup()
         pdfile=NULL;
     }
 
-    /* now free up all the allocated strings on the list */
-    for (int i=0; i < nw; i++) 
+    if (list)
     {
-        if (list[i]) {
-            free(list[i]);
-            list[i] = 0;
+        /* now free up all the allocated strings on the list */
+        for (int i=0; i < nw; i++) 
+        {
+            if (list[i]) {
+                free(list[i]);
+                list[i] = 0;
+            }
         }
+        free((void*)list);
     }
 
     if (encoding) free((void*)encoding);
-    if (list)  free((void*)list);
     if (offst) free((void*)offst);
 
     encoding = NULL;
Index: soltools/giparser/gi_list.cxx
===================================================================
RCS file: /cvs/tools/soltools/giparser/gi_list.cxx,v
retrieving revision 1.4
retrieving revision 1.4.36.1
diff -u -p -u -p -r1.4 -r1.4.36.1
--- soltools/giparser/gi_list.cxx	17 Sep 2006 11:33:21 -0000	1.4
+++ soltools/giparser/gi_list.cxx	29 Jun 2007 08:51:57 -0000	1.4.36.1
@@ -155,13 +155,12 @@ List_GenericInfo::InsertInfoByPath( Gene
                         - i_sKeyPath );
 
         GenericInfo * pNew = new GenericInfo(aKey);
-        InsertInfo(pNew,false);
-
-        return pNew->SubList().InsertInfoByPath(
+        bool bSuccess = InsertInfo(pNew,false);
+        return bSuccess ? pNew->SubList().InsertInfoByPath(
                                     let_dpInfo,
                                     sNextPathSegment,
                                     i_bCreatePath,
-                                    i_bOverwrite );
+                                    i_bOverwrite ) : 0;
     }
     else
     {
Index: sax/source/expatwrap/saxwriter.cxx
===================================================================
RCS file: /cvs/xml/sax/source/expatwrap/saxwriter.cxx,v
retrieving revision 1.18
retrieving revision 1.18.20.1
diff -u -p -u -p -r1.18 -r1.18.20.1
--- sax/source/expatwrap/saxwriter.cxx	19 Jun 2006 23:07:40 -0000	1.18
+++ sax/source/expatwrap/saxwriter.cxx	6 Jul 2007 11:40:39 -0000	1.18.20.1
@@ -71,7 +71,6 @@ using namespace ::com::sun::star::io;
 #include "xml2utf.hxx"
 
 #define LINEFEED 10
-#define SAXWRITER_CHECK_FOR_INVALID_CHARS
 #define SEQUENCESIZE 1024
 #define MAXCOLUMNCOUNT 72
 
@@ -221,25 +220,23 @@ public:
 	inline sal_Bool comment(const rtl::OUString& rComment) throw( SAXException );
 };
 
-const sal_Bool g_bValidCharsBelow32[31] =
+const sal_Bool g_bValidCharsBelow32[32] =
 {
 //  0 1 2 3 4 5 6 7
 	0,0,0,0,0,0,0,0,  //0
 	0,1,1,0,0,1,0,0,  //8
 	0,0,0,0,0,0,0,0,  //16
-	0,0,0,0,0,0,0
+	0,0,0,0,0,0,0,0
 };
 
 inline sal_Bool IsInvalidChar(const sal_Unicode aChar)
 {
 	sal_Bool bRet(sal_False);
-#ifdef SAXWRITER_CHECK_FOR_INVALID_CHARS
-		// check first for the most common characters
+	// check first for the most common characters
 	if( aChar < 32 || aChar >= 0xd800 )
 		bRet = ( (aChar < 32 && ! g_bValidCharsBelow32[aChar]) ||
 			aChar == 0xffff ||
 			aChar == 0xfffe );
-#endif
 	return bRet;
 }
 
Index: sax/source/expatwrap/xml2utf.cxx
===================================================================
RCS file: /cvs/xml/sax/source/expatwrap/xml2utf.cxx,v
retrieving revision 1.9
retrieving revision 1.9.20.1
diff -u -p -u -p -r1.9 -r1.9.20.1
--- sax/source/expatwrap/xml2utf.cxx	26 Jun 2006 09:42:10 -0000	1.9
+++ sax/source/expatwrap/xml2utf.cxx	6 Jul 2007 12:18:35 -0000	1.9.20.1
@@ -459,7 +459,7 @@ Sequence<sal_Unicode> Text2UnicodeConver
 
 
 	if( pbTempMem ) {
-		delete pbTempMem;
+		delete [] pbTempMem;
 	}
 
 	// set to correct unicode size
@@ -574,7 +574,7 @@ Sequence<sal_Int8> Unicode2TextConverter
 	}
 
 	if( puTempMem ) {
-		delete puTempMem;
+		delete [] puTempMem;
 	}
 
 	// reduce the size of the buffer (fast, no copy necessary)
Index: salhelper/source/gcc3.map
===================================================================
RCS file: salhelper/source/gcc3.map
diff -N salhelper/source/gcc3.map
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ salhelper/source/gcc3.map	24 Jul 2007 09:36:02 -0000	1.1.2.2
@@ -0,0 +1,73 @@
+UDK_3_0_0 { # should have been UDK_3.0
+    global:
+	GetVersionInfo;
+	_ZN9salhelper18ORealDynamicLoader11newInstanceEPPS0_RKN3rtl8OUStringES6_;
+	_ZN9salhelper18ORealDynamicLoader7acquireEv;
+	_ZN9salhelper18ORealDynamicLoader7releaseEv;
+	_ZN9salhelper18ORealDynamicLoaderC1EPPS0_RKN3rtl8OUStringES6_PvS7_;
+	_ZN9salhelper18ORealDynamicLoaderC2EPPS0_RKN3rtl8OUStringES6_PvS7_;
+	_ZN9salhelper18ORealDynamicLoaderD0Ev;
+	_ZN9salhelper18ORealDynamicLoaderD1Ev;
+	_ZN9salhelper18ORealDynamicLoaderD2Ev;
+	_ZN9salhelper21SimpleReferenceObjectD0Ev;
+	_ZN9salhelper21SimpleReferenceObjectD1Ev;
+	_ZN9salhelper21SimpleReferenceObjectD2Ev;
+	_ZN9salhelper21SimpleReferenceObjectdlEPv;
+
+	# Introducing a question mark at the end because of
+	# marginal type discrepancy there is a difference in the
+	# mangled name between Linux and Mac OS X, see #i69351#
+	_ZN9salhelper21SimpleReferenceObjectnwE?; # salhelper::SimpleReferenceObject::operator new (std::size_t)
+
+	_ZNK9salhelper18ORealDynamicLoader6getApiEv;
+    # _ZTIN9salhelper18ORealDynamicLoaderE;
+    # _ZTSN9salhelper18ORealDynamicLoaderE;
+	_ZTVN9salhelper18ORealDynamicLoaderE;
+    # _ZTIN9salhelper21SimpleReferenceObjectE;
+    # _ZTSN9salhelper21SimpleReferenceObjectE;
+	_ZTVN9salhelper21SimpleReferenceObjectE;
+
+    local:
+        *;
+};
+
+UDK_3.1 {
+    global:
+        _ZN9salhelper21SimpleReferenceObjectdlEPvRKSt9nothrow_t;
+
+	# Introducing a wildcard right in the middle because due to 
+	# marginal type discrepancy there is a difference in the 
+	# mangled name between Linux and Mac OS X see #i69351#
+	_ZN9salhelper21SimpleReferenceObjectnwE?RKSt9nothrow_t; # salhelper::SimpleReferenceObject::operator new (std::size_t, std::nothrow_t const&)
+ 
+        _ZN9salhelper9ConditionC1ERN3osl5MutexE;
+        _ZN9salhelper9ConditionC2ERN3osl5MutexE;
+        _ZN9salhelper9ConditionD0Ev;
+        _ZN9salhelper9ConditionD1Ev;
+        _ZN9salhelper9ConditionD2Ev;
+        # _ZTIN9salhelper9ConditionE;
+        # _ZTIS9salhelper9ConditionE;
+
+        _ZN9salhelper17ConditionModifierC1ERNS_9ConditionE;
+        _ZN9salhelper17ConditionModifierC2ERNS_9ConditionE;
+        _ZN9salhelper17ConditionModifierD1Ev;
+        _ZN9salhelper17ConditionModifierD2Ev;
+
+        _ZN9salhelper15ConditionWaiterC1ERNS_9ConditionE;
+        _ZN9salhelper15ConditionWaiterC1ERNS_9ConditionE?;
+        _ZN9salhelper15ConditionWaiterC2ERNS_9ConditionE;
+        _ZN9salhelper15ConditionWaiterC2ERNS_9ConditionE?;
+        _ZN9salhelper15ConditionWaiterD1Ev;
+        _ZN9salhelper15ConditionWaiterD2Ev;
+
+        _ZN9salhelper15ConditionWaiter8timedoutaSERKS1_;
+        _ZN9salhelper15ConditionWaiter8timedoutC1ERKS1_;
+        _ZN9salhelper15ConditionWaiter8timedoutC1Ev;
+        _ZN9salhelper15ConditionWaiter8timedoutC2ERKS1_;
+        _ZN9salhelper15ConditionWaiter8timedoutC2Ev;
+        _ZN9salhelper15ConditionWaiter8timedoutD0Ev;
+        _ZN9salhelper15ConditionWaiter8timedoutD1Ev;
+        _ZN9salhelper15ConditionWaiter8timedoutD2Ev;
+        # _ZTIN9salhelper15ConditionWaiter8timedoutE;
+        # _ZTSN9salhelper15ConditionWaiter8timedoutE;
+} UDK_3_0_0;
Index: salhelper/source/makefile.mk
===================================================================
RCS file: /cvs/udk/salhelper/source/makefile.mk,v
retrieving revision 1.27
retrieving revision 1.27.4.5
diff -u -p -u -p -r1.27 -r1.27.4.5
--- salhelper/source/makefile.mk	25 May 2007 10:55:32 -0000	1.27
+++ salhelper/source/makefile.mk	24 Jul 2007 12:09:13 -0000	1.27.4.5
@@ -80,16 +80,8 @@ DEF1NAME=	$(SHL1TARGET)
 SHL1VERSIONMAP=msci.map
 .ELIF "$(COMNAME)"=="sunpro5"
 SHL1VERSIONMAP=sols.map
-.ELIF "$(OS)$(CPU)$(COMNAME)"=="LINUXIgcc3"
-SHL1VERSIONMAP=gcc3_linux_intel.map
-.ELIF "$(OS)$(CPU)$(COMNAME)"=="FREEBSDIgcc3"
-SHL1VERSIONMAP=gcc3_linux_intel.map
-.ELIF "$(OS)$(CPU)$(COMNAME)"=="LINUXSgcc3"
-SHL1VERSIONMAP=gcc3_linux_intel.map
-.ELIF "$(OS)$(CPU)$(COMNAME)"=="MACOSXIgcc3"
-SHL1VERSIONMAP=gcc3_linux_intel.map
-.ELIF "$(GUI)$(COM)"=="WNTGCC"
-SHL1VERSIONMAP=mingw_intel.map
+.ELIF "$(COMNAME)"=="gcc3"
+SHL1VERSIONMAP=gcc3.map
 .ENDIF
 
 # --- Targets ------------------------------------------------------
Index: salhelper/source/mingw_intel.map
===================================================================
RCS file: salhelper/source/mingw_intel.map
diff -N salhelper/source/mingw_intel.map
--- salhelper/source/mingw_intel.map	26 Jun 2007 17:39:24 -0000	1.3
+++ /dev/null	1 Jan 1970 00:00:00 -0000
@@ -1,57 +0,0 @@
-UDK_3_0_0 { # should have been UDK_3.0
-    global:
-	GetVersionInfo;
-	_ZN9salhelper18ORealDynamicLoader11newInstanceEPPS0_RKN3rtl8OUStringES6_;
-	_ZN9salhelper18ORealDynamicLoader7acquireEv;
-	_ZN9salhelper18ORealDynamicLoader7releaseEv;
-	_ZN9salhelper18ORealDynamicLoaderC1EPPS0_RKN3rtl8OUStringES6_PvS7_;
-	_ZN9salhelper18ORealDynamicLoaderC2EPPS0_RKN3rtl8OUStringES6_PvS7_;
-	_ZN9salhelper18ORealDynamicLoaderD0Ev;
-	_ZN9salhelper18ORealDynamicLoaderD1Ev;
-	_ZN9salhelper18ORealDynamicLoaderD2Ev;
-	_ZN9salhelper21SimpleReferenceObjectD0Ev;
-	_ZN9salhelper21SimpleReferenceObjectD1Ev;
-	_ZN9salhelper21SimpleReferenceObjectD2Ev;
-	_ZN9salhelper21SimpleReferenceObjectdlEPv;
-	_ZN9salhelper21SimpleReferenceObjectnwEj;
-	_ZNK9salhelper18ORealDynamicLoader6getApiEv;
-    # _ZTIN9salhelper18ORealDynamicLoaderE;
-    # _ZTSN9salhelper18ORealDynamicLoaderE;
-	_ZTVN9salhelper18ORealDynamicLoaderE;
-    # _ZTIN9salhelper21SimpleReferenceObjectE;
-    # _ZTSN9salhelper21SimpleReferenceObjectE;
-	_ZTVN9salhelper21SimpleReferenceObjectE;
-
-        _ZN9salhelper21SimpleReferenceObjectdlEPvRKSt9nothrow_t;
-        _ZN9salhelper21SimpleReferenceObjectnwEjRKSt9nothrow_t;
-        _ZN9salhelper9ConditionC1ERN3osl5MutexE;
-        _ZN9salhelper9ConditionC2ERN3osl5MutexE;
-        _ZN9salhelper9ConditionD0Ev;
-        _ZN9salhelper9ConditionD1Ev;
-        _ZN9salhelper9ConditionD2Ev;
-        # _ZTIN9salhelper9ConditionE;
-        # _ZTIS9salhelper9ConditionE;
-
-        _ZN9salhelper17ConditionModifierC1ERNS_9ConditionE;
-        _ZN9salhelper17ConditionModifierC2ERNS_9ConditionE;
-        _ZN9salhelper17ConditionModifierD1Ev;
-        _ZN9salhelper17ConditionModifierD2Ev;
-
-        _ZN9salhelper15ConditionWaiterC1ERNS_9ConditionE;
-        _ZN9salhelper15ConditionWaiterC1ERNS_9ConditionEm;
-        _ZN9salhelper15ConditionWaiterC2ERNS_9ConditionE;
-        _ZN9salhelper15ConditionWaiterC2ERNS_9ConditionEm;
-        _ZN9salhelper15ConditionWaiterD1Ev;
-        _ZN9salhelper15ConditionWaiterD2Ev;
-
-        _ZN9salhelper15ConditionWaiter8timedoutaSERKS1_;
-        _ZN9salhelper15ConditionWaiter8timedoutC1ERKS1_;
-        _ZN9salhelper15ConditionWaiter8timedoutC1Ev;
-        _ZN9salhelper15ConditionWaiter8timedoutC2ERKS1_;
-        _ZN9salhelper15ConditionWaiter8timedoutC2Ev;
-        _ZN9salhelper15ConditionWaiter8timedoutD0Ev;
-        _ZN9salhelper15ConditionWaiter8timedoutD1Ev;
-        _ZN9salhelper15ConditionWaiter8timedoutD2Ev;
-        # _ZTIN9salhelper15ConditionWaiter8timedoutE;
-        # _ZTSN9salhelper15ConditionWaiter8timedoutE;
-} UDK_3_0_0;
Index: i18npool/source/breakiterator/gendict.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/breakiterator/gendict.cxx,v
retrieving revision 1.8
retrieving revision 1.8.68.1
diff -u -p -u -p -r1.8 -r1.8.68.1
--- i18npool/source/breakiterator/gendict.cxx	17 Sep 2006 09:14:30 -0000	1.8
+++ i18npool/source/breakiterator/gendict.cxx	3 Jul 2007 10:42:40 -0000	1.8.68.1
@@ -56,13 +56,17 @@ SAL_IMPLEMENT_MAIN_WITH_ARGS(argc, argv)
 
 	sfp = fopen(argv[1], "rb");	// open the source file for read;
 	if (sfp == NULL) 
+	{
 	    printf("Open the dictionary source file failed.");
+	    return -1;
+	}
 
 	// create the C source file to write
 	cfp = fopen(argv[2], "wb");
 	if (cfp == NULL) {
 	    fclose(sfp);
 	    printf("Can't create the C source file.");
+	    return -1;
 	}
 
 	fprintf(cfp, "/*\n");
Index: i18npool/source/collator/gencoll_rule.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/collator/gencoll_rule.cxx,v
retrieving revision 1.10
retrieving revision 1.10.38.1
diff -u -p -u -p -r1.10 -r1.10.38.1
--- i18npool/source/collator/gencoll_rule.cxx	12 Dec 2006 16:15:22 -0000	1.10
+++ i18npool/source/collator/gencoll_rule.cxx	24 Jul 2007 09:58:58 -0000	1.10.38.1
@@ -45,6 +45,8 @@
 
 #include "warnings_guard_unicode_tblcoll.h"
 
+U_CAPI void U_EXPORT2 uprv_free(void *mem);
+
 using namespace ::rtl;
 
 /* Main Procedure */
@@ -133,6 +135,8 @@ SAL_IMPLEMENT_MAIN_WITH_ARGS(argc, argv)
         else {
             printf("Could not get rule data from collator\n");
         }
+
+	if (data) uprv_free(data);
     } else {
         printf("\nRule parsering error\n");
     }
Index: i18npool/source/textconversion/genconv_dict.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/textconversion/genconv_dict.cxx,v
retrieving revision 1.10
retrieving revision 1.10.68.1
diff -u -p -u -p -r1.10 -r1.10.68.1
--- i18npool/source/textconversion/genconv_dict.cxx	17 Sep 2006 09:23:37 -0000	1.10
+++ i18npool/source/textconversion/genconv_dict.cxx	3 Jul 2007 10:42:40 -0000	1.10.68.1
@@ -61,13 +61,17 @@ SAL_IMPLEMENT_MAIN_WITH_ARGS(argc, argv)
 
 	sfp = fopen(argv[2], "rb");	// open the source file for read;
 	if (sfp == NULL) 
+    {
 	    printf("Open the dictionary source file failed.");
+        return -1;
+    }
 
 	// create the C source file to write
 	cfp = fopen(argv[3], "wb");
 	if (cfp == NULL) {
 	    fclose(sfp);
 	    printf("Can't create the C source file.");
+        return -1;
 	}
 
 	fprintf(cfp, "/*\n");
