Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.77
retrieving revision 1.77.8.3
diff -u -r1.77 -r1.77.8.3
--- config_office/configure.in	15 Nov 2004 14:11:17 -0000	1.77
+++ config_office/configure.in	9 Dec 2004 16:47:27 -0000	1.77.8.3
@@ -116,9 +116,18 @@
 [  --enable-kde            Determines whether to use Qt/KDE vclplug on platforms
                           where Qt and KDE are available.
 ],,)
+AC_ARG_ENABLE(binfilter,
+[  --disable-binfilter:    Disable legacy binary file formats filters
+],,)
 AC_ARG_ENABLE(rpath,
 [  --disable-rpath:        Disable the use of relative paths in shared libraries
 ],,)
+AC_ARG_WITH(myspell-dicts,
+[  --without-myspell-dicts   Removes myspell dictionaries from openoffice.org
+                         installation set, for people building for specific
+                         distributions where the myspell dictionaries are
+                         installed from other sources
+],,)
 AC_ARG_WITH(system-libs,
 [  --with-system-libs      Use libs already on system -- enables all
 			   --with-system-* flags except mozilla, python and
@@ -372,6 +381,9 @@
 
                           Usage: --enable-check-only=yes
 ],,)
+
+BUILD_TYPE="OOo EXT"
+
 dnl ===================================================================
 dnl Message.
 dnl ===================================================================
@@ -639,6 +651,17 @@
 fi
 AC_SUBST(ENABLE_FONTCONFIG)
 
+dnl ===================================================================
+dnl Disable legacy binary file formats filters
+dnl ===================================================================
+if test "$enable_binfilter" = "no"; then
+   WITH_BINFILTER="NO"
+else
+   WITH_BINFILTER="YES"
+   BUILD_TYPE="$BUILD_TYPE BINFILTER"
+fi
+AC_SUBST(WITH_BINFILTER)
+
 if test "$_os" = "WINNT"; then
    AC_MSG_CHECKING([whether to use DirectX])
    if test "$enable_directx" = "yes" -o "$enable_directx" = "TRUE" -o "$enable_directx" = ""; then
@@ -657,6 +680,19 @@
 fi
 AC_SUBST(ENABLE_RPATH)
 
+dnl Test whether to include MySpell dictionaries
+dnl ===================================================================
+AC_MSG_CHECKING([whether to include MySpell dictionaries])
+if test -z "$with_myspell_dicts"; then
+  AC_MSG_RESULT([yes])
+  WITH_MYSPELL_DICTS=YES
+  BUILD_TYPE="$BUILD_TYPE DICTIONARIES"
+else
+  AC_MSG_RESULT([no])
+  WITH_MYSPELL_DICTS=NO
+fi
+AC_SUBST(WITH_MYSPELL_DICTS)
+
 dnl ===================================================================
 dnl WINNT uses either 4nt or tcsh, all other O/S use tcsh.
 dnl The following sets the with_use_shell variable.
@@ -1769,6 +1805,7 @@
 else
     AC_MSG_RESULT([internal])
     SYSTEM_ZLIB=NO
+    BUILD_TYPE="$BUILD_TYPE ZLIB"
 fi
 AC_SUBST(SYSTEM_ZLIB)
 
@@ -1787,6 +1824,7 @@
 else
     AC_MSG_RESULT([internal])
     SYSTEM_JPEG=NO
+    BUILD_TYPE="$BUILD_TYPE JPEG"
 fi
 AC_SUBST(SYSTEM_JPEG)
 
@@ -1805,6 +1843,7 @@
 else
     AC_MSG_RESULT([internal])
     SYSTEM_EXPAT=NO
+    BUILD_TYPE="$BUILD_TYPE EXPAT"
 fi
 AC_SUBST(SYSTEM_EXPAT)
 
@@ -1820,6 +1859,7 @@
 else
     AC_MSG_RESULT([internal])
     SYSTEM_FREETYPE=NO
+    BUILD_TYPE="$BUILD_TYPE FREETYPE"
 fi
 AC_SUBST(SYSTEM_FREETYPE)
 AC_SUBST(FREETYPE_CFLAGS)
@@ -1834,9 +1874,11 @@
     AC_MSG_RESULT([external])
     SYSTEM_LIBXML=YES
     PKG_CHECK_MODULES( LIBXML, libxml-2.0 >= 2.0 )
+    BUILD_TYPE="$BUILD_TYPE LIBXMLSEC"
 else
     AC_MSG_RESULT([internal])
     SYSTEM_LIBXML=NO
+    BUILD_TYPE="$BUILD_TYPE LIBXML2 LIBXMLSEC"
 fi
 AC_SUBST(SYSTEM_LIBXML)
 AC_SUBST(LIBXML_CFLAGS)
@@ -1863,6 +1905,7 @@
    CPPFLAGS="$save_CPPFLAGS"
 else
    SYSTEM_PYTHON=NO
+   BUILD_TYPE="$BUILD_TYPE PYTHON"
    AC_MSG_RESULT([internal])
 fi
 AC_SUBST(SYSTEM_PYTHON)
@@ -1928,6 +1971,7 @@
 else
    AC_MSG_RESULT([internal])
    SYSTEM_CURL=NO
+   BUILD_TYPE="$BUILD_TYPE CURL"
 fi
 AC_SUBST(SYSTEM_CURL)
 AC_SUBST(CURL_CFLAGS)
@@ -2087,6 +2132,7 @@
 else
     AC_MSG_RESULT([internal])
     SYSTEM_NAS=NO
+    BUILD_TYPE="$BUILD_TYPE NAS"
 fi
 AC_SUBST(SYSTEM_NAS)
 AC_SUBST(NAS_INCLUDES)
@@ -2120,6 +2166,7 @@
     SYSTEM_NEON=NO
     NEON_LIBS=-lneon
     NEON_CFLAGS=
+    BUILD_TYPE="$BUILD_TYPE NEON"
 fi
 AC_SUBST(SYSTEM_NEON)
 AC_SUBST(NEON_LIBS)
@@ -1968,6 +2012,7 @@
 else
     AC_MSG_RESULT([internal])
     SYSTEM_MOZILLA=NO
+    BUILD_TYPE="$BUILD_TYPE MOZ"
 fi
 AC_SUBST(SYSTEM_MOZILLA)
 AC_SUBST(MOZILLA_CFLAGS)
@@ -2683,6 +2730,7 @@
 if test "$with_fonts" != "no" ; then
   AC_MSG_RESULT([yes])
   WITH_FONTS=YES
+  BUILD_TYPE="$BUILD_TYPE BITSTREAM_VERA_FONTS"
 else
   AC_MSG_RESULT([no])
   WITH_FONTS=NO
@@ -2883,6 +2931,8 @@
 fi
 AC_SUBST(CUSTOM_IMAGE_SETS)
 
+AC_SUBST(BUILD_TYPE)
+
 dnl ===================================================================
 dnl Setting up the environment.
 dnl ===================================================================
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.26
retrieving revision 1.26.8.2
diff -u -r1.26 -r1.26.8.2
--- config_office/set_soenv.in	15 Nov 2004 14:11:32 -0000	1.26
+++ config_office/set_soenv.in	9 Dec 2004 14:44:29 -0000	1.26.8.2
@@ -1701,6 +1701,7 @@
 ToFile( "WITH_GPC",          "@WITH_GPC@",         "e" );
 ToFile( "WITH_LIBART",       "@WITH_LIBART@",      "e" );
 ToFile( "LIBART_CFLAGS",     "@LIBART_CFLAGS@",    "e" );
+ToFile( "WITH_MYSPELL_DICTS","@WITH_MYSPELL_DICTS@","e");
 ToFile( "LIBART_LIBS",       "@LIBART_LIBS@",      "e" );
 ToFile( "WITH_LIBSN",        "@WITH_LIBSN@",       "e" );
 ToFile( "LIBSN_CFLAGS",      "@LIBSN_CFLAGS@",     "e" );
@@ -1708,6 +1709,8 @@
 ToFile( "WITH_MOZILLA",      "@WITH_MOZILLA@",     "e" );
 ToFile( "WITH_OPENLDAP",     "@WITH_OPENLDAP@",    "e" );
 ToFile( "WITH_FONTS",        "@WITH_FONTS@",       "e" );
+ToFile( "WITH_BINFILTER",    "@WITH_BINFILTER@",   "e" );
+ToFile( "BUILD_TYPE",        "@BUILD_TYPE@",       "e" );
 ToFile( "WITH_FONTOOO",      "@WITH_FONTOOO@",     "e" );
 ToFile( "SYSTEM_MSPACK",     "@SYSTEM_MSPACK@",    "e" );
 if ( '@ENABLE_RPATH@' eq "no" ) {

--- connectivity/prj/build.lst	19 Jan 2004 10:57:17 -0000
+++ connectivity/prj/build.lst	9 Dec 2004 16:43:34 -0000
@@ -1,4 +1,4 @@
-cn	connectivity	:	comphelper moz svtools unixODBC unoil jurt javaunohelper hsqldb NULL
+cn	connectivity	:	comphelper MOZ:moz svtools unixODBC unoil jurt javaunohelper hsqldb NULL
 cn	connectivity							usr1	-	all	cn_mkout NULL
 cn	connectivity\com\sun\star\sdbcx\comp\hsqldb	nmake	-	all	cn_jhsqldbdb cn_hsqldb NULL
 cn	connectivity\source\types			nmake	-	all	cn_types NULL
Index: hwpfilter/prj/build.lst
===================================================================
RCS file: /cvs/sw/hwpfilter/prj/build.lst,v
retrieving revision 1.3
retrieving revision 1.3.24.1
diff -u -r1.3 -r1.3.24.1
--- hwpfilter/prj/build.lst	19 Jan 2004 10:57:17 -0000	1.3
+++ hwpfilter/prj/build.lst	9 Dec 2004 16:43:34 -0000	1.3.24.1
@@ -1,4 +1,4 @@
-hw	hwpfilter	:	offuh cppuhelper zlib NULL
+hw	hwpfilter	:	offuh cppuhelper ZLIB:zlib NULL
 hw	hwpfilter							usr1	-	all	hw_mkout NULL
 hw	hwpfilter\prj						get		-	all	hw_prj NULL
 hw	hwpfilter\source					nmake	-	all	hw_source NULL
Index: jvmfwk/prj/build.lst
===================================================================
RCS file: /cvs/udk/jvmfwk/prj/build.lst,v
retrieving revision 1.9
retrieving revision 1.9.30.1
diff -u -r1.9 -r1.9.30.1
--- jvmfwk/prj/build.lst	1 Jun 2004 12:11:15 -0000	1.9
+++ jvmfwk/prj/build.lst	9 Dec 2004 16:19:08 -0000	1.9.30.1
@@ -1,4 +1,4 @@
-jv jvmfwk : cppu cppuhelper sal libxml2 NULL
+jv jvmfwk : cppu cppuhelper sal LIBXML2:libxml2 NULL
 jv jvmfwk\source nmake - all jf_framework NULL
 jv jvmfwk\plugins\sunmajor\pluginlib nmake - all jf_sunmajorlib NULL
 jv jvmfwk\plugins\sunmajor\javaenvsetup nmake - u jf_sunjavaldx jf_framework NULL
Index: libxmlsec/prj/build.lst
===================================================================
RCS file: /cvs/external/libxmlsec/prj/build.lst,v
retrieving revision 1.5
retrieving revision 1.5.6.1
diff -u -r1.5 -r1.5.6.1
--- libxmlsec/prj/build.lst	28 Oct 2004 16:07:00 -0000	1.5
+++ libxmlsec/prj/build.lst	9 Dec 2004 16:34:34 -0000	1.5.6.1
@@ -1,3 +1,3 @@
-ls  libxmlsec   : libxml2 moz NULL
+ls  libxmlsec   : stlport soltools LIBXML2:libxml2 MOZ:moz NULL
 ls	libxmlsec			    usr1	-	all	ls_mkout NULL
 ls	libxmlsec			    nmake	-	all	ls_xmlsec1 NULL
Index: package/prj/build.lst
===================================================================
RCS file: /cvs/xml/package/prj/build.lst,v
retrieving revision 1.5
retrieving revision 1.5.68.1
diff -u -r1.5 -r1.5.68.1
--- package/prj/build.lst	25 Feb 2004 15:27:32 -0000	1.5
+++ package/prj/build.lst	9 Dec 2004 16:43:09 -0000	1.5.68.1
@@ -1,4 +1,4 @@
-pk	package	:	cppu cppuhelper comphelper ucbhelper sal zlib NULL
+pk	package	:	cppu cppuhelper comphelper ucbhelper sal ZLIB:zlib NULL
 pk	package									usr1	-	all	pk_mkout NULL
 pk	package\source\zipapi					nmake	-	all	pk_zipapi NULL
 pk	package\source\zippackage				nmake	-	all	pk_zippackage NULL
Index: postprocess/prj/build.lst
===================================================================
RCS file: /cvs/installation/postprocess/prj/build.lst,v
retrieving revision 1.8
retrieving revision 1.8.2.1
diff -u -r1.8 -r1.8.2.1
--- postprocess/prj/build.lst	29 Nov 2004 08:42:37 -0000	1.8
+++ postprocess/prj/build.lst	9 Dec 2004 14:46:49 -0000	1.8.2.1
@@ -1,4 +1,4 @@
-po	postprocess	::	accessibility automation basctl bean binfilter chart2 configmgr crashrep dbaccess desktop dtrans embeddedobj embedserv eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io lingucomponent MathMLDTD msfontextract odk officecfg package padmin psprint_config remotebridges sc scaddins scp2 scripting sd setup_native slideshow starmath sw testshl testshl2 testtools ucb UnoControls unoxml unzip wizards xmlsecurity OOo:bitstream_vera_fonts OOo:dictionaries OOo:pyuno OOo:readlicense_oo SO:top NULL
+po	postprocess	::	accessibility automation basctl bean BINFILTER:binfilter chart2 configmgr crashrep dbaccess desktop dtrans embeddedobj embedserv eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io lingucomponent MathMLDTD msfontextract odk officecfg package padmin psprint_config remotebridges sc scaddins scp2 scripting sd setup_native slideshow starmath sw testshl testshl2 testtools ucb UnoControls unoxml unzip wizards xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top NULL
 po	postprocess			    	usr1	-	all	po_mkout NULL
 po	postprocess\packconfig		nmake	-	all	po_packconfig NULL
 po	postprocess\rebase	    	nmake	-	w	po_rebase NULL
Index: pyuno/prj/build.lst
===================================================================
RCS file: /cvs/udk/pyuno/prj/build.lst,v
retrieving revision 1.4
retrieving revision 1.4.76.1
diff -u -r1.4 -r1.4.76.1
--- pyuno/prj/build.lst	13 Jun 2003 07:51:26 -0000	1.4
+++ pyuno/prj/build.lst	9 Dec 2004 14:52:16 -0000	1.4.76.1
@@ -1,4 +1,4 @@
-pu	pyuno	:	stoc cpputools cppuhelper python NULL
+pu	pyuno	:	stoc cpputools cppuhelper PYTHON:python NULL
 pu	pyuno				  usr1	-	all	br_mkout NULL
 pu	pyuno\zipcore	nmake	-	all	pu_zipcore NULL
 pu	pyuno\unotypes		nmake	-	all	pu_unotypes NULL
Index: scp2/source/calc/file_calc.scp
===================================================================
RCS file: /cvs/installation/scp2/source/calc/file_calc.scp,v
retrieving revision 1.11
retrieving revision 1.11.24.1
diff -u -r1.11 -r1.11.24.1
--- scp2/source/calc/file_calc.scp	18 Nov 2004 08:53:53 -0000	1.11
+++ scp2/source/calc/file_calc.scp	9 Dec 2004 14:35:22 -0000	1.11.24.1
@@ -77,8 +77,10 @@
 // start binfilter #110743#
 //
 
+#ifndef WITHOUT_BINFILTER
 STD_LIB_FILE( gid_File_Lib_Bf_Sc, bf_sc)
 STD_RES_FILE( gid_File_Res_Bf_Sc, bf_sc)
+#endif
 
 //
 // end binfilter #110743#
Index: scp2/source/calc/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/calc/makefile.mk,v
retrieving revision 1.4
retrieving revision 1.4.192.1
diff -u -r1.4 -r1.4.192.1
--- scp2/source/calc/makefile.mk	5 Jul 2004 13:11:28 -0000	1.4
+++ scp2/source/calc/makefile.mk	9 Dec 2004 14:35:23 -0000	1.4.192.1
@@ -71,6 +71,10 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
+
+.IF "$(WITH_BINFILTER)" == "NO"
+SCPDEFS+=-DWITHOUT_BINFILTER
+.ENDIF
 
 SCP_PRODUCT_TYPE=osl
 PARFILES= \
Index: scp2/source/lingu/file_lingu.scp
===================================================================
RCS file: /cvs/installation/scp2/source/lingu/file_lingu.scp,v
retrieving revision 1.2
retrieving revision 1.2.302.1
diff -u -r1.2 -r1.2.302.1
--- scp2/source/lingu/file_lingu.scp	11 Feb 2004 08:48:24 -0000	1.2
+++ scp2/source/lingu/file_lingu.scp	9 Dec 2004 14:35:23 -0000	1.2.302.1
@@ -77,9 +77,11 @@
     Dir = gid_Dir_Program;
 End
 
+#ifndef WITHOUT_MYSPELL_DICTS
 File gid_File_Zip_Writingaids
     TXT_FILE_BODY;
     Styles = (ARCHIVE);
     Dir = gid_Dir_Share_Dict_Ooo;
     Name = "writingaids.zip";
 End
+#endif
Index: scp2/source/lingu/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/lingu/makefile.mk,v
retrieving revision 1.3
retrieving revision 1.3.200.1
diff -u -r1.3 -r1.3.200.1
--- scp2/source/lingu/makefile.mk	25 Jun 2004 16:22:25 -0000	1.3
+++ scp2/source/lingu/makefile.mk	9 Dec 2004 14:35:23 -0000	1.3.200.1
@@ -71,6 +71,10 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :  settings.mk
+
+.IF "$(WITH_MYSPELL_DICTS)" != "YES"
+SCPDEFS+=-DWITHOUT_MYSPELL_DICTS
+.ENDIF
 
 SCP_PRODUCT_TYPE=osl
 
Index: scp2/source/math/file_math.scp
===================================================================
RCS file: /cvs/installation/scp2/source/math/file_math.scp,v
retrieving revision 1.10
retrieving revision 1.10.24.1
diff -u -r1.10 -r1.10.24.1
--- scp2/source/math/file_math.scp	18 Nov 2004 08:58:10 -0000	1.10
+++ scp2/source/math/file_math.scp	9 Dec 2004 14:35:24 -0000	1.10.24.1
@@ -85,8 +85,10 @@
 // start binfilter #110743#
 //
 
+#ifndef WITHOUT_BINFILTER
 STD_LIB_FILE( gid_File_Lib_Bf_Sm, bf_sm)
 STD_RES_FILE( gid_File_Res_Bf_Sm, bf_sm)
+#endif
 
 //
 // end binfilter #110743#
Index: scp2/source/math/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/math/makefile.mk,v
retrieving revision 1.6
retrieving revision 1.6.24.1
diff -u -r1.6 -r1.6.24.1
--- scp2/source/math/makefile.mk	18 Nov 2004 08:58:51 -0000	1.6
+++ scp2/source/math/makefile.mk	9 Dec 2004 14:35:24 -0000	1.6.24.1
@@ -71,6 +71,10 @@
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
+
+.IF "$(WITH_BINFILTER)" == "NO"
+SCPDEFS+=-DWITHOUT_BINFILTER
+.ENDIF
 
 SCP_PRODUCT_TYPE=osl
 PARFILES= \
Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.71
retrieving revision 1.71.10.1
diff -u -r1.71 -r1.71.10.1
--- scp2/source/ooo/file_library_ooo.scp	26 Nov 2004 21:52:02 -0000	1.71
+++ scp2/source/ooo/file_library_ooo.scp	9 Dec 2004 14:35:25 -0000	1.71.10.1
@@ -1505,6 +1505,7 @@
 // start binfilter #110743#
 //
 
+#ifndef WITHOUT_BINFILTER
 STD_LIB_FILE( gid_File_Lib_Bf_Sch, bf_sch)
 STD_LIB_FILE( gid_File_Lib_Bf_Lng, bf_lng)
 STD_LIB_FILE( gid_File_Lib_Bf_Xo, bf_xo)
@@ -1517,6 +1518,7 @@
 STD_LIB_FILE( gid_File_Lib_Legacy_Binfilters, legacy_binfilters)
 STD_UNO_LIB_FILE( gid_File_Lib_Bf_Migratefilter, bf_migratefilter)
 STD_UNO_LIB_FILE( gid_File_Lib_Bindet, bindet)
+#endif
 
 //
 // end binfilter #110743#
Index: scp2/source/ooo/file_resource_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_resource_ooo.scp,v
retrieving revision 1.14
retrieving revision 1.14.10.1
diff -u -r1.14 -r1.14.10.1
--- scp2/source/ooo/file_resource_ooo.scp	30 Nov 2004 12:47:40 -0000	1.14
+++ scp2/source/ooo/file_resource_ooo.scp	9 Dec 2004 14:35:25 -0000	1.14.10.1
@@ -142,6 +142,7 @@
 // start binfilter #110743#
 //
 
+#ifndef WITHOUT_BINFILTER
 STD_RES_FILE( gid_File_Res_Bf_Sch, bf_sch )
 STD_RES_FILE( gid_File_Res_Bf_Svx, bf_svx )
 STD_RES_FILE( gid_File_Res_Bf_Frm, bf_frm )
@@ -149,6 +150,7 @@
 STD_RES_FILE( gid_File_Res_Bf_Sd, bf_sd )
 STD_RES_FILE( gid_File_Res_Bf_Sw, bf_sw )
 STD_RES_FILE( gid_File_Res_Bf_Sfx, bf_sfx )
+#endif
 
 //
 // end binfilter #110743#
Index: scp2/source/ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.13
retrieving revision 1.13.10.1
diff -u -r1.13 -r1.13.10.1
--- scp2/source/ooo/makefile.mk	26 Nov 2004 22:25:05 -0000	1.13
+++ scp2/source/ooo/makefile.mk	9 Dec 2004 14:35:26 -0000	1.13.10.1
@@ -124,6 +124,10 @@
 
 .IF "$(SYSTEM_CURL)" == "YES"
 SCPDEFS+=-DSYSTEM_CURL
+.ENDIF
+
+.IF "$(WITH_BINFILTER)" == "NO"
+SCPDEFS+=-DWITHOUT_BINFILTER
 .ENDIF
 
 SCP_PRODUCT_TYPE=osl
Index: svtools/prj/build.lst
===================================================================
RCS file: /cvs/util/svtools/prj/build.lst,v
retrieving revision 1.28
retrieving revision 1.28.10.1
diff -u -r1.28 -r1.28.10.1
--- svtools/prj/build.lst	26 Nov 2004 20:38:27 -0000	1.28
+++ svtools/prj/build.lst	9 Dec 2004 15:58:07 -0000	1.28.10.1
@@ -1,4 +1,4 @@
-st	svtools	:	offuh toolkit ucbhelper unotools jpeg cppu cppuhelper sal NULL
+st	svtools	:	offuh toolkit ucbhelper unotools JPEG:jpeg cppu cppuhelper sal NULL
 st	svtools						usr1	-	all	st_mkout NULL
 st	svtools\inc					get	-	all	st_inc NULL
 st	svtools\inc\sane				get	-	all	st_incsa NULL
Index: tools/prj/build.lst
===================================================================
RCS file: /cvs/util/tools/prj/build.lst,v
retrieving revision 1.19
retrieving revision 1.19.42.1
diff -u -r1.19 -r1.19.42.1
--- tools/prj/build.lst	4 Oct 2004 20:00:08 -0000	1.19
+++ tools/prj/build.lst	9 Dec 2004 16:41:54 -0000	1.19.42.1
@@ -1,4 +1,4 @@
-tl	tools	:	cppu external offuh vos zlib expat basegfx comphelper NULL
+tl	tools	:	cppu external offuh vos ZLIB:zlib expat basegfx comphelper NULL
 tl	tools								usr1	-	all	tl_mkout NULL
 tl	tools\bootstrp\isdll				get		-	all	tl_bsisdll NULL
 tl	tools\bootstrp\addexes				get		-	all	tl_bsexes NULL
Index: ucb/prj/build.lst
===================================================================
RCS file: /cvs/ucb/ucb/prj/build.lst,v
retrieving revision 1.17
retrieving revision 1.17.26.1
diff -u -r1.17 -r1.17.26.1
--- ucb/prj/build.lst	13 Oct 2004 15:04:57 -0000	1.17
+++ ucb/prj/build.lst	9 Dec 2004 14:50:54 -0000	1.17.26.1
@@ -1,4 +1,4 @@
-uc ucb : unotools tools cppuhelper curl neon offapi sal salhelper ucbhelper udkapi vos NULL
+uc ucb : unotools tools cppuhelper CURL:curl NEON:neon offapi sal salhelper ucbhelper udkapi vos NULL
 uc	ucb							usr1	-	all	uc_mkout NULL
 uc	ucb\unotypes			    nmake	-	all	uc_unotypes NULL
 uc	ucb\source\regexp		    nmake	-	all	uc_regexp uc_unotypes NULL
Index: unoxml/prj/build.lst
===================================================================
RCS file: /cvs/framework/unoxml/prj/build.lst,v
retrieving revision 1.5
retrieving revision 1.5.20.1
diff -u -r1.5 -r1.5.20.1
--- unoxml/prj/build.lst	30 Mar 2004 09:58:22 -0000	1.5
+++ unoxml/prj/build.lst	9 Dec 2004 16:33:43 -0000	1.5.20.1
@@ -1,4 +1,4 @@
-ux	unoxml	:	offuh cppuhelper libxml2 sfx2 NULL
+ux	unoxml	:	offuh cppuhelper LIBXML2:libxml2 sfx2 NULL
 ux	unoxml\source\dom			nmake	-	all	ux_dom      NULL
 ux	unoxml\source\xpath			nmake	-	all	ux_xpath    ux_dom NULL
 ux	unoxml\source\events		nmake	-	all	ux_events   ux_dom NULL
Index: vcl/prj/build.lst
===================================================================
RCS file: /cvs/gsl/vcl/prj/build.lst,v
retrieving revision 1.35
retrieving revision 1.35.34.1
diff -u -r1.35 -r1.35.34.1
--- vcl/prj/build.lst	9 Nov 2004 16:43:11 -0000	1.35
+++ vcl/prj/build.lst	9 Dec 2004 16:16:54 -0000	1.35.34.1
@@ -1,4 +1,4 @@
-vc	vcl	:	boost nas freetype psprint rsc sot ucbhelper unotools rvpapi icu i18npool unoil ridljar x11_extensions offuh basegfx SO:sndfile SO:portaudio transex3 NULL
+vc	vcl	:	boost NAS:nas FREETYPE:freetype psprint rsc sot ucbhelper unotools rvpapi icu i18npool unoil ridljar x11_extensions offuh basegfx SO:sndfile SO:portaudio transex3 NULL
 vc	vcl										usr1	-	all	vc_mkout NULL
 vc	vcl\source\unotypes						nmake	-	all	vc_unot NULL
 vc	vcl\source\glyphs						nmake	-	all	vc_glyphs vc_unot NULL
Index: xmlsecurity/prj/build.lst
===================================================================
RCS file: /cvs/xml/xmlsecurity/prj/build.lst,v
retrieving revision 1.3
retrieving revision 1.3.18.1
diff -u -r1.3 -r1.3.18.1
--- xmlsecurity/prj/build.lst	30 Sep 2004 16:06:29 -0000	1.3
+++ xmlsecurity/prj/build.lst	9 Dec 2004 16:33:42 -0000	1.3.18.1
@@ -1,4 +1,4 @@
-xs	xmlsecurity	:	xmloff unotools offapi unoil svx moz libxmlsec NULL
+xs	xmlsecurity	:	xmloff unotools offapi unoil svx MOZ:moz LIBXMLSEC:libxmlsec NULL
 xs	xmlsecurity                                                        usr1	 -	all	xs_mkout	NULL
 xs	xmlsecurity\source\framework                                       nmake -	all	xs_fw NULL
 xs	xmlsecurity\source\xmlsec\nss                                      nmake -	all	xs_nss NULL
Index: scp2/prj/build.lst
===================================================================
RCS file: /cvs/installation/scp2/prj/build.lst,v
retrieving revision 1.8
diff -u -p -u -r1.8 build.lst
--- scp2/prj/build.lst	15 Nov 2004 13:15:48 -0000	1.8
+++ scp2/prj/build.lst	9 Dec 2004 19:50:18 -0000
@@ -1,4 +1,4 @@
-cp    scp2    :    setup_native python NULL
+cp    scp2    :    setup_native PYTHON:python NULL
 cp    scp2                        usr1     -    all    cp_mkout NULL
 cp    scp2\source\activex         nmake    -    all    cp_activex NULL
 cp    scp2\source\calc            nmake    -    all    cp_calc NULL

