Index: filter/source/xsltfilter/makefile.mk
===================================================================
RCS file: /cvs/framework/filter/source/xsltfilter/makefile.mk,v
retrieving revision 1.17
retrieving revision 1.17.18.1
diff -u -p -u -p -r1.17 -r1.17.18.1
--- filter/source/xsltfilter/makefile.mk	14 Aug 2006 15:31:13 -0000	1.17
+++ filter/source/xsltfilter/makefile.mk	27 Sep 2006 11:19:41 -0000	1.17.18.1
@@ -74,6 +74,9 @@ JARFILES 		= ridl.jar unoil.jar jurt.jar
 
 .IF "$(SYSTEM_XALAN)" == "YES"
 XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(XALAN_JAR)
+.IF "$(SERIALIZER_JAR)" != ""
+XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(SERIALIZER_JAR)
+.ENDIF
 .ELSE
 JARFILES += xalan.jar
 .ENDIF
Index: svtools/inc/brwbox.hxx
===================================================================
RCS file: /cvs/util/svtools/inc/brwbox.hxx,v
retrieving revision 1.29
retrieving revision 1.29.2.1
diff -u -p -u -p -r1.29 -r1.29.2.1
--- svtools/inc/brwbox.hxx	29 Sep 2006 15:52:33 -0000	1.29
+++ svtools/inc/brwbox.hxx	2 Oct 2006 14:23:37 -0000	1.29.2.1
@@ -45,6 +45,9 @@
 #ifndef _SV_CTRL_HXX //autogen
 #include <vcl/ctrl.hxx>
 #endif
+#ifndef _SV_MULTISEL_HXX
+#include <tools/multisel.hxx>
+#endif
 
 #ifndef _HEADBAR_HXX
 #include <headbar.hxx>
@@ -94,7 +97,7 @@ namespace utl {
 // -------------------
 
 #define BROWSER_INVALIDID           USHRT_MAX
-#define BROWSER_ENDOFSELECTION      -1
+#define BROWSER_ENDOFSELECTION      (long)(SFX_ENDOFSELECTION)
 
 typedef ULONG BrowserMode;
 
Index: svtools/source/filter.vcl/wmf/enhwmf.cxx
===================================================================
RCS file: /cvs/util/svtools/source/filter.vcl/wmf/enhwmf.cxx,v
retrieving revision 1.31
retrieving revision 1.31.18.1
diff -u -p -u -p -r1.31 -r1.31.18.1
--- svtools/source/filter.vcl/wmf/enhwmf.cxx	17 Sep 2006 14:55:15 -0000	1.31
+++ svtools/source/filter.vcl/wmf/enhwmf.cxx	2 Oct 2006 14:38:05 -0000	1.31.18.1
@@ -373,33 +373,37 @@ BOOL EnhWMFReader::ReadEnhWMF() // SvStr
 				UINT16*	pnPoints;
 				Point*	pPtAry;
 
-				INT32	i, nPoly, nGesPoints;
+				UINT32	i, nPoly, nGesPoints;
 				pWMF->SeekRel( 0x10 );
 
 				// Anzahl der Polygone:
 				*pWMF >> nPoly >> nGesPoints;
 
-				// Anzahl der Punkte eines jeden Polygons holen, Gesammtzahl der Punkte ermitteln:
-				pnPoints = new UINT16[ nPoly ];
-
-				for ( i = 0; i < nPoly; i++ )
+				if (nGesPoints < SAL_MAX_UINT32 / sizeof(Point))
 				{
-					*pWMF >> nPoints;
-					pnPoints[ i ] = (UINT16)nPoints;
-				}
-				// Polygonpunkte holen:
-				pPtAry  = (Point*) new char[ nGesPoints * sizeof(Point) ];
 
-				for ( i = 0; i < nGesPoints; i++ )
-				{
-					*pWMF >> nX32 >> nY32;
-					pPtAry[ i ] = Point( nX32, nY32 );
+					// Anzahl der Punkte eines jeden Polygons holen, Gesammtzahl der Punkte ermitteln:
+					pnPoints = new UINT16[ nPoly ];
+	
+					for ( i = 0; i < nPoly; i++ )
+					{
+						*pWMF >> nPoints;
+						pnPoints[ i ] = (UINT16)nPoints;
+					}
+					// Polygonpunkte holen:
+					pPtAry  = (Point*) new char[ nGesPoints * sizeof(Point) ];
+	
+					for ( i = 0; i < nGesPoints; i++ )
+					{
+						*pWMF >> nX32 >> nY32;
+						pPtAry[ i ] = Point( nX32, nY32 );
+					}
+					// PolyPolygon Actions erzeugen
+					PolyPolygon aPolyPoly( (UINT16)nPoly, pnPoints, pPtAry );
+					pOut->DrawPolyPolygon( aPolyPoly, bRecordPath );
+					delete[] (char*) pPtAry;
+					delete[] pnPoints;
 				}
-				// PolyPolygon Actions erzeugen
-				PolyPolygon aPolyPoly( (UINT16)nPoly, pnPoints, pPtAry );
-				pOut->DrawPolyPolygon( aPolyPoly, bRecordPath );
-				delete[] (char*) pPtAry;
-				delete[] pnPoints;
 			}
 			break;
 
@@ -1101,31 +1105,34 @@ BOOL EnhWMFReader::ReadEnhWMF() // SvStr
 				UINT16*	pnPoints;
 				Point*	pPtAry;
 
-				INT32	i, nPoly, nGesPoints;
+				UINT32	i, nPoly, nGesPoints;
 				pWMF->SeekRel( 0x10 );
 				// Anzahl der Polygone:
 				*pWMF >> nPoly >> nGesPoints;
-				// Anzahl der Punkte eines jeden Polygons holen, Gesammtzahl der Punkte ermitteln:
-				pnPoints = new UINT16[ nPoly ];
-				for ( i = 0; i < nPoly; i++ )
-				{
-					*pWMF >> nPoints;
-					pnPoints[ i ] = (UINT16)nPoints;
-				}
-				// Polygonpunkte holen:
-				pPtAry  = (Point*) new char[ nGesPoints * sizeof(Point) ];
-				for ( i = 0; i < nGesPoints; i++ )
+				if (nGesPoints < SAL_MAX_UINT32 / sizeof(Point))
 				{
-					*pWMF >> nX16 >> nY16;
-					pPtAry[ i ] = Point( nX16, nY16 );
+					// Anzahl der Punkte eines jeden Polygons holen, Gesammtzahl der Punkte ermitteln:
+					pnPoints = new UINT16[ nPoly ];
+					for ( i = 0; i < nPoly; i++ )
+					{
+						*pWMF >> nPoints;
+						pnPoints[ i ] = (UINT16)nPoints;
+					}
+					// Polygonpunkte holen:
+					pPtAry  = (Point*) new char[ nGesPoints * sizeof(Point) ];
+					for ( i = 0; i < nGesPoints; i++ )
+					{
+						*pWMF >> nX16 >> nY16;
+						pPtAry[ i ] = Point( nX16, nY16 );
+					}
+	
+					// PolyPolygon Actions erzeugen
+					PolyPolygon aPolyPoly( (UINT16)nPoly, pnPoints, pPtAry );
+					pOut->DrawPolyPolygon( aPolyPoly, bRecordPath );
+					delete[] (char*) pPtAry;
+					delete[] pnPoints;
 				}
-
-				// PolyPolygon Actions erzeugen
-				PolyPolygon aPolyPoly( (UINT16)nPoly, pnPoints, pPtAry );
-				pOut->DrawPolyPolygon( aPolyPoly, bRecordPath );
-				delete[] (char*) pPtAry;
-				delete[] pnPoints;
-			};
+			}
 			break;
 
 			case EMR_FILLRGN :
Index: svtools/source/filter.vcl/wmf/winwmf.cxx
===================================================================
RCS file: /cvs/util/svtools/source/filter.vcl/wmf/winwmf.cxx,v
retrieving revision 1.30
retrieving revision 1.30.18.1
diff -u -p -u -p -r1.30 -r1.30.18.1
--- svtools/source/filter.vcl/wmf/winwmf.cxx	17 Sep 2006 14:55:43 -0000	1.30
+++ svtools/source/filter.vcl/wmf/winwmf.cxx	2 Oct 2006 14:38:05 -0000	1.30.18.1
@@ -866,17 +866,21 @@ void WMFReader::ReadRecordParams( USHORT
 										aMemoryStream >> aPt.X()
 													  >> aPt.Y()
 													  >> nStringLen;
-										sal_Unicode* pBuf = aString.AllocBuffer( (sal_uInt16)nStringLen );
-										for ( i = 0; i < nStringLen; i++ )
-											aMemoryStream >> pBuf[ i ];
-										aMemoryStream >> nDXCount;
-										if ( nDXCount )
-											pDXAry = new sal_Int32[ nDXCount ];
-										for  ( i = 0; i < nDXCount; i++ )
-											aMemoryStream >> pDXAry[ i ];
-										aMemoryStream >> nSkipActions;
-										pOut->DrawText( aPt, aString, pDXAry );
-										delete[] pDXAry;
+
+										if (nStringLen < STRING_MAXLEN)
+										{
+											sal_Unicode* pBuf = aString.AllocBuffer( (xub_StrLen)nStringLen );
+											for ( i = 0; i < nStringLen; i++ )
+												aMemoryStream >> pBuf[ i ];
+											aMemoryStream >> nDXCount;
+											if ( nDXCount )
+												pDXAry = new sal_Int32[ nDXCount ];
+											for  ( i = 0; i < nDXCount; i++ )
+												aMemoryStream >> pDXAry[ i ];
+											aMemoryStream >> nSkipActions;
+											pOut->DrawText( aPt, aString, pDXAry );
+											delete[] pDXAry;
+										}
 									}
 								}
 								break;
Index: lingucomponent/source/hyphenator/altlinuxhyph/hyphen/hyphenimp.cxx
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/hyphen/hyphenimp.cxx,v
retrieving revision 1.18
retrieving revision 1.18.8.1
diff -u -p -u -p -r1.18 -r1.18.8.1
--- lingucomponent/source/hyphenator/altlinuxhyph/hyphen/hyphenimp.cxx	16 Sep 2006 16:07:26 -0000	1.18
+++ lingucomponent/source/hyphenator/altlinuxhyph/hyphen/hyphenimp.cxx	27 Sep 2006 09:27:28 -0000	1.18.8.1
@@ -93,7 +93,6 @@
 #include "dictmgr.hxx"
 
 #include <stdio.h>
-#include <iostream>
 using namespace utl;
 using namespace osl;
 using namespace rtl;
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.179
retrieving revision 1.177.8.2
diff -u -p -u -p -r1.179 -r1.177.8.2
--- config_office/configure.in	27 Sep 2006 09:43:11 -0000	1.179
+++ config_office/configure.in	2 Oct 2006 09:54:05 -0000	1.177.8.2
@@ -310,6 +310,10 @@ AC_ARG_WITH(xalan-jar,
 [  --with-xalan-jar=JARFILE   Specify path to jarfile manually ],
 [ XALAN_JAR="$withval"
 ])
+AC_ARG_WITH(serializer-jar,
+[  --with-serializer-jar=JARFILE   Specify path to jarfile manually ],
+[ SERIALIZER_JAR="$withval"
+])
 AC_ARG_WITH(system-sablot,
 [  --with-system-sablot    Use sablot already on system
 ],,)
@@ -3076,6 +3080,11 @@ if test -n "$with_system_xalan" -o -n "$
           AC_CHECK_FILE($XALAN_JAR, [],
                [AC_MSG_ERROR(xalan.jar replacement not found.)], [])
         fi
+        if test -n "$SERIALIZER_JAR"; then
+          AC_CHECK_FILE($SERIALIZER_JAR, [],
+               [AC_MSG_ERROR(serializer.jar not found.)], [])
+          AC_SUBST(SERIALIZER_JAR)
+        fi
 else
     AC_MSG_RESULT([internal])
     SYSTEM_XALAN=NO
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.118
retrieving revision 1.116.8.2
diff -u -p -u -p -r1.118 -r1.116.8.2
--- config_office/set_soenv.in	27 Sep 2006 09:43:23 -0000	1.118
+++ config_office/set_soenv.in	2 Oct 2006 09:53:54 -0000	1.116.8.2
@@ -1822,6 +1822,7 @@ ToFile( "SYSTEM_XERCES",     "@SYSTEM_XE
 ToFile( "XERCES_JAR",        "@XERCES_JAR@",       "e" );
 ToFile( "SYSTEM_XALAN",      "@SYSTEM_XALAN@",     "e" );
 ToFile( "XALAN_JAR",         "@XALAN_JAR@",        "e" );
+ToFile( "SERIALIZER_JAR",    "@SERIALIZER_JAR@",   "e" );
 ToFile( "SYSTEM_SABLOT",     "@SYSTEM_SABLOT@",    "e" );
 ToFile( "SABLOT_LIBS",       "@SABLOT_LIBS@",      "e" );
 ToFile( "SYSTEM_ODBC_HEADERS","@SYSTEM_ODBC_HEADERS@","e" );
Index: sd/source/ui/app/optsitem.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/optsitem.cxx,v
retrieving revision 1.35
retrieving revision 1.35.14.1
diff -u -p -u -p -r1.35 -r1.35.14.1
--- sd/source/ui/app/optsitem.cxx	16 Sep 2006 18:30:28 -0000	1.35
+++ sd/source/ui/app/optsitem.cxx	27 Sep 2006 09:58:27 -0000	1.35.14.1
@@ -129,7 +129,7 @@ void SdOptionsGeneric::Init() const
 {
 	if( !mbInit )
 	{
-		SdOptionsGeneric* pThis	= (SdOptionsGeneric*) this;
+		SdOptionsGeneric* pThis	= const_cast<SdOptionsGeneric*>(this);
 
 		if( !mpCfgItem )
 			pThis->mpCfgItem = new SdOptionsItem( *this, maSubTree );
@@ -167,7 +167,7 @@ void SdOptionsGeneric::Commit( SdOptions
 
 	if( aNames.getLength() && ( aValues.getLength() == aNames.getLength() ) )
 	{
-		if( ( (SdOptionsGeneric*) this )->WriteData( aValues.getArray() ) )
+		if( (const_cast<SdOptionsGeneric*>(this))->WriteData( aValues.getArray() ) )
 			rCfgItem.PutProperties( aNames, aValues );
 		else
 		{
Index: sw/source/core/fields/cellfml.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/fields/cellfml.cxx,v
retrieving revision 1.10
retrieving revision 1.10.20.1
diff -u -p -u -p -r1.10 -r1.10.20.1
--- sw/source/core/fields/cellfml.cxx	16 Sep 2006 21:10:00 -0000	1.10
+++ sw/source/core/fields/cellfml.cxx	3 Oct 2006 09:59:27 -0000	1.10.20.1
@@ -376,18 +376,14 @@ void SwTableFormula::_MakeFormel( const 
 	// ein Bereich in dieser Klammer ?
 	if( pLastBox )
 	{
-	//TODOUNICODE: does it work?
-//		pEndBox = (SwTableBox*)(long)(*pLastBox);
-		pEndBox = (SwTableBox*)pLastBox->ToInt32();
+		pEndBox = (SwTableBox*)pLastBox->ToInt64();
 
 		// ist das ueberhaupt ein gueltiger Pointer ??
 		if( !rTbl.GetTabSortBoxes().Seek_Entry( pEndBox ))
 			pEndBox = 0;
 		rFirstBox.Erase( 0, pLastBox->Len()+1 );
 	}
-	//TODOUNICODE: does it work?
-//	pSttBox = (SwTableBox*)(long)rFirstBox;
-	pSttBox = (SwTableBox*)rFirstBox.ToInt32();
+	pSttBox = (SwTableBox*)rFirstBox.ToInt64();
 	// ist das ueberhaupt ein gueltiger Pointer ??
 	if( !rTbl.GetTabSortBoxes().Seek_Entry( pSttBox ))
 		pSttBox = 0;
@@ -465,7 +461,7 @@ void SwTableFormula::RelBoxNmsToPtr( con
 	if( pLastBox )
 	{
 		if( 0 != ( pRelBox = lcl_RelToBox( rTbl, pBox, *pLastBox )) )
-			rNewStr += String::CreateFromInt32( (long)pRelBox );
+			rNewStr += String::CreateFromInt64( (sal_PtrDiff)pRelBox );
 		else
 			rNewStr += '0';
 		rNewStr += ':';
@@ -473,7 +469,7 @@ void SwTableFormula::RelBoxNmsToPtr( con
 	}
 
 	if( 0 != ( pRelBox = lcl_RelToBox( rTbl, pBox, rFirstBox )) )
-		rNewStr += String::CreateFromInt32( (long)pRelBox );
+		rNewStr += String::CreateFromInt64( (sal_PtrDiff)pRelBox );
 	else
 		rNewStr += '0';
 
@@ -527,8 +523,7 @@ void SwTableFormula::PtrToBoxNms( const 
 	rFirstBox.Erase(0,1);
 	if( pLastBox )
 	{
-//		pBox = (SwTableBox*)(long)(*pLastBox);
-		pBox = (SwTableBox*)pLastBox->ToInt32();
+		pBox = (SwTableBox*)pLastBox->ToInt64();
 
 		// ist das ueberhaupt ein gueltiger Pointer ??
 		if( rTbl.GetTabSortBoxes().Seek_Entry( pBox ))
@@ -539,8 +534,7 @@ void SwTableFormula::PtrToBoxNms( const 
 		rFirstBox.Erase( 0, pLastBox->Len()+1 );
 	}
 
-//	pBox = (SwTableBox*)(long)rFirstBox;
-	pBox = (SwTableBox*)rFirstBox.ToInt32();
+	pBox = (SwTableBox*)rFirstBox.ToInt64();
 	// ist das ueberhaupt ein gueltiger Pointer ??
 	if( rTbl.GetTabSortBoxes().Seek_Entry( pBox ))
 		rNewStr += pBox->GetName();
@@ -562,13 +556,13 @@ void SwTableFormula::BoxNmsToPtr( const 
 	if( pLastBox )
 	{
 		pBox = rTbl.GetTblBox( *pLastBox );
-		rNewStr += String::CreateFromInt32( (long)pBox );
+		rNewStr += String::CreateFromInt64( (sal_PtrDiff)pBox );
 		rNewStr += ':';
 		rFirstBox.Erase( 0, pLastBox->Len()+1 );
 	}
 
 	pBox = rTbl.GetTblBox( rFirstBox );
-	rNewStr += String::CreateFromInt32( (long)pBox );
+	rNewStr += String::CreateFromInt64( (sal_PtrDiff)pBox );
 
 	// Kennung fuer Box erhalten
 	rNewStr += rFirstBox.GetChar( rFirstBox.Len() - 1 );
@@ -897,8 +891,7 @@ String lcl_BoxNmToRel( const SwTable& rT
 	if( !bExtrnlNm )
 	{
 		// in die Externe Darstellung umwandeln.
-//		SwTableBox* pBox = (SwTableBox*)(long)sTmp;
-		SwTableBox* pBox = (SwTableBox*)sTmp.ToInt32();
+		SwTableBox* pBox = (SwTableBox*)sTmp.ToInt64();
 		if( !rTbl.GetTabSortBoxes().Seek_Entry( pBox ))
 			return '?';
 		sTmp = pBox->GetName();
@@ -954,8 +947,7 @@ void SwTableFormula::_GetFmlBoxes( const
 	// ein Bereich in dieser Klammer ?
 	if( pLastBox )
 	{
-//		pEndBox = (SwTableBox*)(long)(*pLastBox);
-		pEndBox = (SwTableBox*)pLastBox->ToInt32();
+		pEndBox = (SwTableBox*)pLastBox->ToInt64();
 
 		// ist das ueberhaupt ein gueltiger Pointer ??
 		if( !rTbl.GetTabSortBoxes().Seek_Entry( pEndBox ))
@@ -963,8 +955,7 @@ void SwTableFormula::_GetFmlBoxes( const
 		rFirstBox.Erase( 0, pLastBox->Len()+1 );
 	}
 
-//	pSttBox = (SwTableBox*)(long)rFirstBox;
-	pSttBox = (SwTableBox*)rFirstBox.ToInt32();
+	pSttBox = (SwTableBox*)rFirstBox.ToInt64();
 	// ist das ueberhaupt ein gueltiger Pointer ??
 	if( !rTbl.GetTabSortBoxes().Seek_Entry( pSttBox ))
 		pSttBox = 0;
@@ -1054,12 +1045,8 @@ void SwTableFormula::_HasValidBoxes( con
 		{
 		case INTRNL_NAME:
 			if( pLastBox )
-			{
-//				pEndBox = (SwTableBox*)(long)(*pLastBox);
-				pEndBox = (SwTableBox*)pLastBox->ToInt32();
-			}
-//			pSttBox = (SwTableBox*)(long)rFirstBox;
-			pSttBox = (SwTableBox*)rFirstBox.ToInt32();
+				pEndBox = (SwTableBox*)pLastBox->ToInt64();
+			pSttBox = (SwTableBox*)rFirstBox.ToInt64();
 			break;
 
 		case REL_NAME:
@@ -1169,12 +1156,8 @@ void SwTableFormula::_SplitMergeBoxNm( c
 	{
 	case INTRNL_NAME:
 		if( pLastBox )
-		{
-//			pEndBox = (SwTableBox*)(long)(*pLastBox);
-			pEndBox = (SwTableBox*)pLastBox->ToInt32();
-		}
-//		pSttBox = (SwTableBox*)(long)rFirstBox;
-		pSttBox = (SwTableBox*)rFirstBox.ToInt32();
+			pEndBox = (SwTableBox*)pLastBox->ToInt64();
+		pSttBox = (SwTableBox*)rFirstBox.ToInt64();
 		break;
 
 	case REL_NAME:
@@ -1260,8 +1243,8 @@ void SwTableFormula::_SplitMergeBoxNm( c
 	}
 
 	if( pLastBox )
-		( rNewStr += String::CreateFromInt32((long)pEndBox )) += ':';
-	( rNewStr += String::CreateFromInt32((long)pSttBox ))
+		( rNewStr += String::CreateFromInt64((sal_PtrDiff)pEndBox)) += ':';
+	( rNewStr += String::CreateFromInt64((sal_PtrDiff)pSttBox))
 			  += rFirstBox.GetChar( rFirstBox.Len() - 1 );
 }
 
Index: svx/source/editeng/impedit3.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/editeng/impedit3.cxx,v
retrieving revision 1.110
retrieving revision 1.110.14.1
diff -u -p -u -p -r1.110 -r1.110.14.1
--- svx/source/editeng/impedit3.cxx	17 Sep 2006 04:52:31 -0000	1.110
+++ svx/source/editeng/impedit3.cxx	27 Sep 2006 10:23:52 -0000	1.110.14.1
@@ -1314,10 +1314,11 @@ sal_Bool ImpEditEngine::CreateLines( USH
 				if ( nPara || IsFixedCellHeight() || pLine->GetStartPortion() ) // Nicht die aller erste Zeile
 				{
                     // #100508# There are documents with PropLineSpace 0, why?
+                    // (cmc: re above question :-) such documents can be seen by importing a .ppt
                     if ( rLSItem.GetPropLineSpace() && ( rLSItem.GetPropLineSpace() != 100 ) )
                     {
 					    sal_uInt16 nTxtHeight = pLine->GetHeight();
-					    sal_uInt32 nH = nTxtHeight;
+					    sal_Int32 nH = nTxtHeight;
 					    nH *= rLSItem.GetPropLineSpace();
 					    nH /= 100;
 					    // Der Ascent muss um die Differenz angepasst werden:
@@ -1656,10 +1657,11 @@ void ImpEditEngine::CreateAndInsertEmpty
 			if ( nPara || IsFixedCellHeight() || pTmpLine->GetStartPortion() ) // Nicht die aller erste Zeile
 			{
                 // #100508# There are documents with PropLineSpace 0, why?
+                // (cmc: re above question :-) such documents can be seen by importing a .ppt
                 if ( rLSItem.GetPropLineSpace() && ( rLSItem.GetPropLineSpace() != 100 ) )
                 {
 				    sal_uInt16 nTxtHeight = pTmpLine->GetHeight();
-				    sal_uInt32 nH = nTxtHeight;
+				    sal_Int32 nH = nTxtHeight;
 				    nH *= rLSItem.GetPropLineSpace();
 				    nH /= 100;
 				    // Der Ascent muss um die Differenz angepasst werden:
Index: scp2/source/ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.43
retrieving revision 1.43.172.1
diff -u -p -u -p -r1.43 -r1.43.172.1
--- scp2/source/ooo/makefile.mk	20 Apr 2006 13:35:04 -0000	1.43
+++ scp2/source/ooo/makefile.mk	27 Sep 2006 11:20:28 -0000	1.43.172.1
@@ -166,6 +166,9 @@ SCPDEFS+=-DSYSTEM_XERCES -DXERCES_JAR=\"
 
 .IF "$(SYSTEM_XALAN)" == "YES"
 SCPDEFS+=-DSYSTEM_XALAN -DXALAN_JAR=\""file://$(XALAN_JAR)"\"
+.IF "$(SERIALIZER_JAR)" != ""
+SCPDEFS+=-DSYSTEM_SERIALIZER -DSERIALIZER_JAR=\""file://$(SERIALIZER_JAR)"\"
+.ENDIF
 .ENDIF
 
 .IF "$(JDK)" == "gcj"
Index: scp2/source/ooo/profileitem_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/profileitem_ooo.scp,v
retrieving revision 1.39
retrieving revision 1.39.158.1
diff -u -p -u -p -r1.39 -r1.39.158.1
--- scp2/source/ooo/profileitem_ooo.scp	9 May 2006 10:45:06 -0000	1.39
+++ scp2/source/ooo/profileitem_ooo.scp	27 Sep 2006 11:20:28 -0000	1.39.158.1
@@ -841,5 +841,8 @@ ProfileItem gid_Profileitem_Jvmfwk_Uno_J
 #endif
 #ifdef SYSTEM_XALAN
     ValueList29 = XALAN_JAR;
+#ifdef SYSTEM_SERIALIZER
+    ValueList30 = SERIALIZER_JAR;
+#endif
 #endif
 End
