Index: instsetoo_native/packimages/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo_native/packimages/makefile.mk,v
retrieving revision 1.7
retrieving revision 1.7.12.1
diff -u -r1.7 -r1.7.12.1
--- instsetoo_native/packimages/makefile.mk	17 Nov 2004 18:10:31 -0000	1.7
+++ instsetoo_native/packimages/makefile.mk	20 Dec 2004 17:09:58 -0000	1.7.12.1
@@ -69,7 +69,12 @@
 RSCCUSTOMIMG*=$(PRJNAME)$/util
 .INCLUDE: target.mk
 
-ALLTAR : $(COMMONBIN)$/images.zip
+IMAGES := $(COMMONBIN)$/images.zip
+# Custom sets, at 24x24 & 16x16 fall-back to industrial preferentially
+CUSTOM_PREFERRED_FALLBACK=$(SOLARSRC)$/ooo_custom_images/industrial
+CUSTOM_IMAGES=$(foreach,i,$(CUSTOM_IMAGE_SETS) images_$i)
+
+ALLTAR : $(IMAGES) $(CUSTOM_IMAGES)
 
 $(RES)$/img$/commandimagelist.ilst .PHONY :
     +-$(MKDIR) $(RES)$/img
@@ -84,3 +89,5 @@
 $(COMMONBIN)$/images.zip .PHONY: $(RES)$/img$/commandimagelist.ilst
     +$(PERL) $(SOLARENV)$/bin$/packimages.pl -g $(SOLARSRC)$/$(RSCDEFIMG) -m $(SOLARSRC)$/$(RSCDEFIMG) -c $(SOLARSRC)$/$(RSCCUSTOMIMG) -l $(SOLARCOMMONRESDIR)$/img -l $(RES)$/img -o $@
 
+images_% : 
+    +$(PERL) $(SOLARENV)$/bin$/packimages.pl -g $(SOLARSRC)$/$(RSCDEFIMG) -m $(SOLARSRC)$/$(RSCDEFIMG) -c $(SOLARSRC)$/ooo_custom_images/$(@:s/images_//) -c $(CUSTOM_PREFERRED_FALLBACK) -l $(SOLARCOMMONRESDIR)$/img -l $(RES)$/img -o $(COMMONBIN)$/$@.zip
Index: solenv/bin/packimages.pl
===================================================================
RCS file: /cvs/tools/solenv/bin/packimages.pl,v
retrieving revision 1.8
retrieving revision 1.8.46.1
diff -u -r1.8 -r1.8.46.1
--- solenv/bin/packimages.pl	19 Nov 2004 11:41:35 -0000	1.8
+++ solenv/bin/packimages.pl	20 Dec 2004 17:31:27 -0000	1.8.46.1
@@ -82,7 +82,7 @@
 my $tmp_out_file;            # path to temporary output file
 my $global_path;             # path to global images directory
 my $module_path;             # path to module images directory
-my $custom_path;             # path to custom images directory
+my @custom_path;             # path to custom images directory
 my @imagelist_path;          # pathes to directories containing the image lists 
 my $verbose;                 # be verbose
 my $extra_verbose;           # be extra verbose
@@ -135,14 +135,14 @@
                              '-o=s' => \$out_file,
                              '-g=s' => \$global_path,
                              '-m=s' => \$module_path,
-                             '-c=s' => \$custom_path,
+                             '-c=s' => \@custom_path,
                              '-l=s' => \@imagelist_path,
                              '-v'   => \$verbose,
                              '-vv'  => \$extra_verbose
                             );
 
     if ( $opt_help || !$success || !$out_file || !$global_path 
-        || !$module_path || !$custom_path || !@imagelist_path ) 
+        || !$module_path || !@custom_path || !@imagelist_path ) 
     {
         usage();
         exit(1);
@@ -159,7 +159,7 @@
     print_error("directory is not writable: '$out_dir'", 2) if ! -w $out_dir;
 
     # Check paths.
-    foreach ($global_path, $module_path, $custom_path, @imagelist_path) {
+    foreach ($global_path, $module_path, @custom_path, @imagelist_path) {
         print_error("no such directory: '$_'", 2) if ! -d $_;
         print_error("can't search directory: '$_'", 2) if ! -x $_;
     }
@@ -235,12 +235,16 @@
 {
     my $custom_hash_ref = shift;
     my $keep_back;
-    find({ wanted => \&wanted, no_chdir => 0 }, "$custom_path");
-    foreach ( @custom_list ) {
-        if ( /^\Q$custom_path\E\/(.*)$/o ) {
-            $keep_back=$1;
-            $custom_hash_ref->{$keep_back}++;
-        }
+    for my $path (@custom_path) {
+	find({ wanted => \&wanted, no_chdir => 0 }, $path);
+	foreach ( @custom_list ) {
+	    if ( /^\Q$path\E\/(.*)$/o ) {
+		$keep_back=$1;
+		if (!defined $custom_hash_ref->{$keep_back}) {
+		    $custom_hash_ref->{$keep_back} = $path;
+		}
+	    }
+	}
     }
 }
 
@@ -270,7 +274,7 @@
             next;
         }
         if ( exists $custom_hash_ref->{$_} ) {
-            $zip_hash{$_} = $custom_path;
+            $zip_hash{$_} = $custom_hash_ref->{$_};
             next;
         }
         # it's neither in 'module' nor 'custom', record it in zip hash
@@ -278,15 +282,12 @@
     }
     foreach ( keys %{$module_hash_ref} ) {
         if ( exists $custom_hash_ref->{$_} ) {
-            $zip_hash{$_} = $custom_path;
+            $zip_hash{$_} = $custom_hash_ref->{$_};
             next;
         }
         # it's not in 'custom', record it in zip hash
         $zip_hash{$_} = $module_path;
     }
-#    foreach ( keys %{$custom_hash_ref} ) {
-#        $zip_hash{$_} = $custom_path;
-#    }
 
     if ( @warn_list ) {
         foreach ( @warn_list ) {
? config_office/autom4te.cache
? config_office/config.log
? config_office/config.parms
? config_office/config.status
? config_office/oowintool
? config_office/set_soenv
? config_office/warn
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.80
retrieving revision 1.80.2.1
diff -u -r1.80 -r1.80.2.1
--- config_office/configure.in	13 Dec 2004 12:16:35 -0000	1.80
+++ config_office/configure.in	20 Dec 2004 17:26:43 -0000	1.80.2.1
@@ -310,6 +310,12 @@
                           
                           Usage: --enable-macos9
 ],,)
+AC_ARG_WITH(images,
+[  --with-images          Select different image sets to build & install in
+			  addition to the default set.
+
+                          Usage: --with-images="industrial crystal"
+],with_images=$withval,with_images=)
 AC_ARG_WITH(lang,
 [  --with-lang             Use this option to build OpenOffice.org with
                           additional language support. English (US) is always
@@ -2845,6 +2851,9 @@
    LOCAL_SOLVER=$with_local_solver
 fi
 AC_SUBST(LOCAL_SOLVER)
+
+CUSTOM_IMAGE_SETS="$with_images"
+AC_SUBST(CUSTOM_IMAGE_SETS)
 
 WITH_LANG="$with_lang"
 AC_SUBST(WITH_LANG)
Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.29
retrieving revision 1.29.2.1
diff -u -r1.29 -r1.29.2.1
--- config_office/set_soenv.in	13 Dec 2004 12:24:03 -0000	1.29
+++ config_office/set_soenv.in	20 Dec 2004 17:26:43 -0000	1.29.2.1
@@ -1421,6 +1421,7 @@
 ToFile( "ENABLE_KDE",        "@ENABLE_KDE@",       "e" );
 ToFile( "KDE_CFLAGS",        "@KDE_CFLAGS@",       "e" );
 ToFile( "KDE_LIBS",          "@KDE_LIBS@",         "e" );
+ToFile( "CUSTOM_IMAGE_SETS", "@CUSTOM_IMAGE_SETS@","e" );
 ToFile( "PSPRINT",           "TRUE",             "e" );
 # [ed] 5/14/02 If we're building Aqua graphics, insert a C macro to indicate this.
 # There may be a better way to do this, like splitting unxmacxp into two, but


Index: file_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_ooo.scp,v
retrieving revision 1.60
diff -u -r1.60 file_ooo.scp
--- scp2/source/ooo/file_ooo.scp
+++ scp2/source/ooo/file_ooo.scp
@@ -606,7 +606,7 @@
 File gid_File_Images_Zip
     TXT_FILE_BODY;
     Dir = gid_Dir_Share_Config;
-    Name = "images.zip";
+    Name = "images*.zip";
     Styles = (PACKED,SETUPZIP);
 End
 
