--- dbaccess/source/ui/misc/dsntypes.cxx	2005-02-01 20:32:42.457538835 +0530
+++ dbaccess/source/ui/misc/dsntypes.cxx	2005-02-01 19:10:37.279948003 +0530
@@ -413,7 +413,9 @@
 			case DST_OUTLOOK:
 			case DST_OUTLOOKEXP:
 			case DST_FLAT:
+            case DST_EVOLUTION_LDAP:
             case DST_EVOLUTION:
+	    case DST_EVOLUTION_LOCAL:
 	 		case DST_THUNDERBIRD:
 			case DST_CALC:
                 bSupportsTableCreation = FALSE;
@@ -456,7 +458,9 @@
 		case DST_OUTLOOK:
 		case DST_OUTLOOKEXP:
 		case DST_JDBC:
+            case DST_EVOLUTION_LDAP:
         case DST_EVOLUTION:
+	    case DST_EVOLUTION_LOCAL:
 			bEnableBrowseButton = FALSE;
 			break;
 		default:
@@ -495,7 +499,9 @@
 		case DST_MSACCESS:
 		case DST_MOZILLA:
 		case DST_THUNDERBIRD:
-		case DST_EVOLUTION:
+            case DST_EVOLUTION_LDAP:
+		case DST_EVOLUTION: sal_True; break;
+	    case DST_EVOLUTION_LOCAL:
 		case DST_OUTLOOK:
         case DST_OUTLOOKEXP: //????
 		case DST_DBASE:
@@ -570,8 +576,12 @@
 			return DST_OUTLOOK;
 		if (_rDsn.EqualsIgnoreCaseAscii("outlookexp", nSeparator,_rDsn.Len() - nSeparator))
 			return DST_OUTLOOKEXP;
-		if (_rDsn.EqualsIgnoreCaseAscii("evolution", nSeparator,_rDsn.Len() - nSeparator))
+		if (_rDsn.EqualsIgnoreCaseAscii("evolution:ldap", nSeparator,_rDsn.Len() - nSeparator))
+			return DST_EVOLUTION_LDAP;
+		if (_rDsn.EqualsIgnoreCaseAscii("evolution:groupwise", nSeparator,_rDsn.Len() - nSeparator))
 			return DST_EVOLUTION;
+		if (_rDsn.EqualsIgnoreCaseAscii("evolution:local", nSeparator,_rDsn.Len() - nSeparator))
+			return DST_EVOLUTION_LOCAL;
 	}
 
 	// find third :
--- dbaccess/source/ui/inc/dsntypes.hxx	2005-02-01 20:24:30.215332500 +0530
+++ dbaccess/source/ui/inc/dsntypes.hxx	2005-02-01 20:23:25.089946518 +0530
@@ -100,6 +100,8 @@
 	DST_OUTLOOKEXP	=	15,	
 	DST_EVOLUTION	=	16,
 	DST_EMBEDDED	=	17,
+	DST_EVOLUTION_LOCAL =   18,
+	DST_EVOLUTION_LDAP	=	19,
 
 	DST_USERDEFINE1,	/// first user defined driver
 	DST_USERDEFINE2,
--- dbaccess/source/ui/misc/dbumiscres.hrc	2005-02-01 20:35:36.578688808 +0530
+++ dbaccess/source/ui/misc/dbumiscres.hrc	2005-02-01 20:25:03.141473238 +0530
@@ -80,10 +80,12 @@
 #define STR_LDAP				13
 #define STR_OUTLOOK				14
 #define STR_OUTLOOKEXP			15
-#define STR_EVOLUTION			16 
+#define STR_EVOLUTION_LDAP		16 
 #define STR_EMBEDDED			17
+#define STR_EVOLUTION_GROUPWISE		18 
+#define STR_EVOLUTION_LOCAL		19 
 
-#define STR_END					STR_EMBEDDED + 1
+#define STR_END					STR_EVOLUTION_LOCAL + 1
 
 #endif // _DBU_MISCRES_HRC_
 
--- dbaccess/source/ui/misc/dbumiscres.src	2005-02-01 20:35:36.578688808 +0530
+++ dbaccess/source/ui/misc/dbumiscres.src	2005-02-01 18:57:28.302374764 +0530
@@ -129,9 +129,17 @@
 	{
 		Text  ="sdbc:address:outlookexp";
 	};
-	String STR_EVOLUTION
+	String STR_EVOLUTION_LOCAL
 	{
-		Text  ="sdbc:address:evolution";
+		Text  ="sdbc:address:evolution:local";
+	};	
+	String STR_EVOLUTION_LDAP
+	{
+		Text  ="sdbc:address:evolution:ldap";
+	};	
+	String STR_EVOLUTION_GROUPWISE
+	{
+		Text  ="sdbc:address:evolution:groupwise";
 	};	
 	String STR_EMBEDDED
 	{
@@ -217,10 +225,20 @@
 		Text [ de ] = "Microsoft Windows Adressbuch";
 		Text[ en-US ] = "Microsoft Windows Address Book";
 	};
-	String STR_EVOLUTION
+	String STR_EVOLUTION_LOCAL
+	{
+		Text [ de ] = "Evolution Adressbuch";
+		Text[ en-US ] = "Evolution Local";
+	};
+	String STR_EVOLUTION_LDAP
+	{
+		Text [ de ] = "Evolution Adressbuch";
+		Text[ en-US ] = "Evolution LDAP";
+	};
+	String STR_EVOLUTION_GROUPWISE
 	{
 		Text [ de ] = "Evolution Adressbuch";
-		Text[ en-US ] = "Evolution Address Book";
+		Text[ en-US ] = "Groupwise";
 	};
 	String STR_EMBEDDED
 	{
--- dbaccess/source/ui/dlg/dbwizsetup.cxx	2005-02-01 20:36:42.616893069 +0530
+++ dbaccess/source/ui/dlg/dbwizsetup.cxx	2005-02-01 19:17:20.450143488 +0530
@@ -262,6 +262,7 @@
 #define EVOLUTION_PATH         16
 #define CREATENEW_PATH         17
 #define USERDEFINED_PATH       18
+#define EVOLUTION_PATH_LOCAL   19
 
 OFinalDBPageSetup*          pFinalPage;
 
@@ -403,6 +404,14 @@
     else
         declarePath( MOZILLA_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
     
+    if ( m_pCollection->hasAuthentication(DST_EVOLUTION_LOCAL))
+        declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+    else
+        declarePath( EVOLUTION_PATH_LOCAL, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+ if ( m_pCollection->hasAuthentication(DST_EVOLUTION_LDAP))
+        declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
+    else
+        declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
     if ( m_pCollection->hasAuthentication(DST_EVOLUTION))
         declarePath( EVOLUTION_PATH, PAGE_DBSETUPWIZARD_INTRO, PAGE_DBSETUPWIZARD_AUTHENTIFICATION, PAGE_DBSETUPWIZARD_FINAL, WZS_INVALID_STATE);
     else
@@ -583,6 +592,14 @@
                 activatePath( MOZILLA_PATH, sal_True);
 			    break;
 
+            case DST_EVOLUTION_LOCAL:
+                activatePath( EVOLUTION_PATH_LOCAL, sal_True);
+                break;
+
+            case DST_EVOLUTION_LDAP:
+                activatePath( EVOLUTION_PATH, sal_True);
+                break;
+
             case DST_EVOLUTION:
                 activatePath( EVOLUTION_PATH, sal_True);
                 break;
@@ -630,6 +647,8 @@
 	DATASOURCE_TYPE eType = getDatasourceType(*m_pOutSet);
 	switch ( m_eType )
     {
+        case DST_EVOLUTION_LOCAL:
+        case DST_EVOLUTION_LDAP:
         case DST_EVOLUTION:
         case DST_OUTLOOK:
         case DST_OUTLOOKEXP:
