Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.99
diff -u -r1.99 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	8 Sep 2004 16:01:25 -0000	1.99
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	14 Oct 2004 16:23:24 -0000
@@ -2646,7 +2646,7 @@
 						<desc>Specifies an additional limit if Font Antialiasing is enabled. Fonts that are smaller than the specified limit are not anti-aliased.</desc>
 						<label>Font Antialiasing Minimum Pixel Height</label>
 					</info>
-					<value>8</value>
+					<value>1</value>
 				</prop>
 			</group>
 		</group>
