Index: desktop/source/migration/wizard.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/migration/wizard.cxx,v
retrieving revision 1.2
diff -u -p -u -r1.2 wizard.cxx
--- desktop/source/migration/wizard.cxx	15 Nov 2004 15:49:35 -0000	1.2
+++ desktop/source/migration/wizard.cxx	10 Dec 2004 11:18:12 -0000
@@ -118,5 +118,6 @@ FirstStartWizard::FirstStartWizard(Windo
     ,m_bDone(sal_False)
     ,m_bAccepted(sal_False)
 {
+	return;
     // ---
     // FreeResource();
@@ -318,10 +295,9 @@ sal_Bool FirstStartWizard::onFinish(sal_
 
 short FirstStartWizard::Execute()
 {
-    if (isFirstStart() || !isLicenseAccepted())    
-        return svt::RoadmapWizard::Execute();
-    else
-        return sal_True;
+    if (isFirstStart() && Migration::checkMigration())
+		Migration::doMigration();
+	return sal_True;
 }

