Index: vcl/unx/source/gdi/salgdi3.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/salgdi3.cxx,v
retrieving revision 1.122
diff -u -r1.122 salgdi3.cxx
--- vcl/unx/source/gdi/salgdi3.cxx	8 Sep 2004 15:58:06 -0000	1.122
+++ vcl/unx/source/gdi/salgdi3.cxx	15 Oct 2004 16:11:02 -0000
@@ -1204,9 +1204,11 @@
 
 void X11SalGraphics::GetDevFontList( ImplDevFontList *pList )
 {
+#if defined( NASTY_SLOW_UGLY_X_FONTS )
     // announce X11 fonts
     XlfdStorage* pX11FontList = GetDisplay()->GetXlfdList();
     pX11FontList->AnnounceFonts( pList );
+#endif
 
     // prepare the GlyphCache using psprint's font infos
     aX11GlyphPeer.SetDisplay( GetXDisplay(),
