Index: officecfg/registry/schema/org/openoffice/Office/Impress.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Impress.xcs,v
retrieving revision 1.24
diff -u -p -u -r1.24 Impress.xcs
--- officecfg/registry/schema/org/openoffice/Office/Impress.xcs	21 Jan 2005 18:15:46 -0000	1.24
+++ officecfg/registry/schema/org/openoffice/Office/Impress.xcs	19 Sep 2005 10:14:48 -0000
@@ -99,7 +99,7 @@
 						<desc>Indicates whether to show the rulers at the left and at the top of the working area.</desc>
 						<label>Rulers visible</label>
 					</info>
-					<value>false</value>
+					<value>true</value>
 				</prop>
 				<prop oor:name="Helpline" oor:type="xs:boolean">
 					<!-- OldPath: Impress/Layout/Display -->
