--- desktop/source/app/userinstall.cxx.old	2006-03-06 19:19:26.000000000 +0100
+++ desktop/source/app/userinstall.cxx	2006-03-06 20:18:54.000000000 +0100
@@ -280,6 +280,11 @@
         FileBase::RC rc = Directory::create(aUserPath);
         if ((rc != FileBase::E_None) && (rc != FileBase::E_EXIST)) return UserInstall::E_Creation;
 
+#ifdef UNIX
+	// set safer permissions for the user directory by default
+	File::setAttributes(aUserPath, Attribute_OwnWrite| Attribute_OwnRead| Attribute_OwnExe);
+#endif
+
             // copy data from shared data directory of base installation
         for (sal_Int32 i=0; pszSrcList[i]!=NULL && pszDstList[i]!=NULL; i++)
         {
