Index: prj/build.lst
===================================================================
RCS file: /cvs/util/xmlhelp/prj/build.lst,v
retrieving revision 1.8
diff -u -u -r1.8 build.lst
--- xmlhelp/prj/build.lst	14 Feb 2002 13:22:16 -0000	1.8
+++ xmlhelp/prj/build.lst	3 Jun 2005 11:34:12 -0000
@@ -1,8 +1,7 @@
-xh	xmlhelp	:	ucbhelper XmlSearch sablot jut unoil berkeleydb svtools NULL
+xh	xmlhelp	:	ucbhelper sablot jut unoil berkeleydb svtools NULL
 xh	xmlhelp									usr1	-	all	xh_mkout NULL
 xh	xmlhelp\source\helpprovider				nmake	-	all	xh_helpprovider NULL
 xh	xmlhelp\source\treeview					nmake	-	all	xh_treeview NULL
-xh	xmlhelp\source\com\sun\star\help		nmake	-	all	xh_help NULL
 xh	xmlhelp\source\cxxhelp\util				nmake	-	all	xh_cutil NULL
 xh	xmlhelp\source\cxxhelp\qe				nmake	-	all	xh_qe NULL
 xh	xmlhelp\source\cxxhelp\db				nmake	-	all	xh_db NULL
