Index: file_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_ooo.scp,v
retrieving revision 1.74
diff -u -u -r1.74 file_ooo.scp
--- scp2/source/ooo/file_ooo.scp	15 Mar 2005 09:23:29 -0000	1.74
+++ scp2/source/ooo/file_ooo.scp	23 Mar 2005 12:37:12 -0000
@@ -736,7 +736,9 @@
 #endif
 
 #ifdef SOLAR_JAVA
+#ifndef GCJ
 STD_JAR_FILE( gid_File_Jar_Hsqldb, hsqldb )
+#endif
 STD_JAR_FILE( gid_File_Jar_Hsqldb_Sdbc, sdbc_hsqldb )
 #endif
 
Index: makefile.mk
===================================================================
RCS file: /cvs/external/hsqldb/makefile.mk,v
retrieving revision 1.5
diff -u -u -r1.5 makefile.mk
--- hsqldb/makefile.mk	15 Mar 2005 09:08:29 -0000	1.5
+++ hsqldb/makefile.mk	23 Mar 2005 02:26:05 -0000
@@ -68,7 +68,7 @@
 .INCLUDE :	settings.mk
 .INCLUDE :  version.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(JDK)" != "gcj"
 # --- Files --------------------------------------------------------
 
 TARFILE_NAME=hsqldb_$(HSQLDB_VERSION)
@@ -105,7 +105,7 @@
 .INCLUDE : set_ext.mk
 .INCLUDE : target.mk
 
-.IF "$(SOLAR_JAVA)" != ""
+.IF "$(SOLAR_JAVA)" != "" && "$(JDK)" != "gcj"
 .INCLUDE : tg_ext.mk
 .ENDIF
 
