--- config_office/configure.in	2005-07-29 14:22:24.000000000 +0530
+++ config_office/configure.in	2005-08-04 16:36:39.908621384 +0530
@@ -88,6 +91,9 @@ AC_ARG_ENABLE(directx,
 AC_ARG_ENABLE(atlmfc,
 [  --disable-atlmfc        Disable the use of ATL/MFC for windows build.
 ],,)
+AC_ARG_ENABLE(access,
+[  --disable-access        Disable the Access import pieces.
+],,)
 
 AC_ARG_ENABLE(symbols,
 [  --enable-symbols        Include debugging symbols in output.
@@ -866,6 +872,18 @@ if test "$_os" = "WINNT"; then
 fi
 
 dnl ===================================================================
+dnl Disable the Access import pieces?
+dnl ===================================================================
+AC_MSG_CHECKING([whether to enable the Access import pieces])
+if test "$enable_access" = "no"; then
+   ENABLE_ACCESS="NO"
+else
+   ENABLE_ACCESS="YES"
+fi
+AC_MSG_RESULT([$ENABLE_ACCESS])
+AC_SUBST(ENABLE_ACCESS)
+
+dnl ===================================================================
 dnl Disable rpath in shared libraries?
 dnl ===================================================================
 AC_MSG_CHECKING([whether to use RPATH in shared libraries])
--- config_office/set_soenv.in	2005-07-29 14:22:24.000000000 +0530
+++ config_office/set_soenv.in	2005-08-04 16:39:57.758227869 +0530
@@ -1682,6 +1682,7 @@ ToFile( "ENABLE_CRASHDUMP",  "@ENABLE_CR
 ToFile( "ENABLE_CUPS",       "@ENABLE_CUPS@",       "e" );
 ToFile( "ENABLE_FONTCONFIG", "@ENABLE_FONTCONFIG@", "e" );
 ToFile( "ENABLE_DIRECTX",    "@ENABLE_DIRECTX@",    "e" );
+ToFile( "ENABLE_ACCESS",     "@ENABLE_ACCESS@",    "e" );
 ToFile( "VC_STANDARD",       "@VC_STANDARD@",      "e" );
 ToFile( "WITH_GPC",          "@WITH_GPC@",         "e" );
 ToFile( "WITH_LIBART",       "@WITH_LIBART@",      "e" );
