--- bitstream_vera_fonts/makefile.mk-old	2004-06-04 19:50:37.000000000 +0200
+++ bitstream_vera_fonts/makefile.mk	2004-06-04 19:50:14.000000000 +0200
@@ -71,6 +71,11 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(WITH_FONTS)" != "YES"
+@all:
+	@echo "Not installing Bitstream Vera Fonts, nothing to do here..."
+.ENDIF
+
 TARFILE_NAME=ttf-bitstream-vera-1.10
 TARFILE_ROOTDIR=bitstram-vera-fonts
 
