--- scp2/source/base/file_base.scp~	2006-03-29 14:45:27.000000000 +0200
+++ scp2/source/base/file_base.scp	2006-04-28 12:11:26.636190600 +0200
@@ -64,29 +64,6 @@
 
 #endif
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Sdatabase_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(sdatabase);
-    Patchfiles = ("sdatabase.db",
-                  "sdatabase.ht",
-                  "sdatabase.jar",
-                  "sdatabase.key",
-                  "sdatabase.idx/CONTEXTS",
-                  "sdatabase.idx/DICTIONARY",
-                  "sdatabase.idx/DOCS",
-                  "sdatabase.idx/DOCS.TAB",
-                  "sdatabase.idx/EDGE",
-                  "sdatabase.idx/LINKNAMES",
-                  "sdatabase.idx/OFFSETS",
-                  "sdatabase.idx/POSITIONS",
-                  "sdatabase.idx/SCHEMA"); 
-End
-
-#endif
-
 File gid_File_Registry_Spool_Oo_Setup_Base_Xcu
     TXT_FILE_BODY;
     Styles = (PACKED,PATCH);
--- scp2/source/calc/file_calc.scp~	2006-03-27 11:38:53.000000000 +0200
+++ scp2/source/calc/file_calc.scp	2006-04-28 12:11:26.644189384 +0200
@@ -77,30 +77,6 @@
     Name = "/registry/spool/org/openoffice/Office/Embedding-calc.xcu";
 End
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Scalc_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(scalc);
-    Patchfiles = ("scalc.db",
-                  "scalc.ht",
-                  "scalc.jar",
-                  "scalc.key",
-                  "scalc.tree",
-                  "scalc.idx/CONTEXTS",
-                  "scalc.idx/DICTIONARY",
-                  "scalc.idx/DOCS",
-                  "scalc.idx/DOCS.TAB",
-                  "scalc.idx/EDGE",
-                  "scalc.idx/LINKNAMES",
-                  "scalc.idx/OFFSETS",
-                  "scalc.idx/POSITIONS",
-                  "scalc.idx/SCHEMA"); 
-End
-
-#endif
-
 // new user interface configuration files
 /*
 File gid_File_Share_Config_Sofficecfg_Calc_Menubar_Xml
--- scp2/source/calc/module_calc.scp~	2005-09-09 03:21:49.000000000 +0200
+++ scp2/source/calc/module_calc.scp	2006-04-28 12:11:26.645189232 +0200
@@ -52,13 +52,6 @@
     gid_File_Lib_Bf_Sc,gid_File_Res_Bf_Sc,gid_File_Oo_Office_Ui_Calcwindowstate_Xcu,gid_File_Registry_Spool_Oo_TypeDetection_Calc_Types_Xcu,gid_File_Registry_Spool_Oo_TypeDetection_Calc_Filters_Xcu, gid_File_Tmp_Userinstall_Calc_Inf);
 End
 
-Module gid_Module_Prg_Calc_Help
-    MOD_NAME_DESC ( MODULE_PRG_CALC_HELP );
-    ParentID = gid_Module_Prg_Calc;
-    Default = YES;
-    Files = (gid_File_Help_Scalc_Zip);
-End
-
 Module gid_Module_Prg_Calc_Addins
     MOD_NAME_DESC ( MODULE_OPTIONAL_ADDINS );
     ParentID = gid_Module_Prg_Calc;
--- scp2/source/draw/file_draw.scp~	2006-02-09 15:18:30.000000000 +0100
+++ scp2/source/draw/file_draw.scp	2006-04-28 12:11:26.646189080 +0200
@@ -42,29 +42,6 @@
     Name = "/registry/spool/org/openoffice/Office/Embedding-draw.xcu";
 End
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Sdraw_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(sdraw);
-    Patchfiles = ("sdraw.db",
-                  "sdraw.ht",
-                  "sdraw.jar",
-                  "sdraw.key",
-                  "sdraw.idx/CONTEXTS",
-                  "sdraw.idx/DICTIONARY",
-                  "sdraw.idx/DOCS",
-                  "sdraw.idx/DOCS.TAB",
-                  "sdraw.idx/EDGE",
-                  "sdraw.idx/LINKNAMES",
-                  "sdraw.idx/OFFSETS",
-                  "sdraw.idx/POSITIONS",
-                  "sdraw.idx/SCHEMA"); 
-End
-
-#endif
-
 File gid_File_Exe_Sdraw
     BIN_FILE_BODY;
     Dir = gid_Dir_Program;
--- scp2/source/draw/module_draw.scp~	2005-09-09 03:25:47.000000000 +0200
+++ scp2/source/draw/module_draw.scp	2006-04-28 12:11:26.647188928 +0200
@@ -51,9 +51,3 @@
     gid_File_Registry_Spool_Oo_TypeDetection_Draw_Types_Xcu,gid_File_Registry_Spool_Oo_TypeDetection_Draw_Filters_Xcu, gid_File_Tmp_Userinstall_Draw_Inf);
 End
 
-Module gid_Module_Prg_Draw_Help
-    MOD_NAME_DESC ( MODULE_PRG_DRAW_HELP );
-    ParentID = gid_Module_Prg_Draw;
-    Default = YES;
-    Files = (gid_File_Help_Sdraw_Zip);
-End
--- scp2/source/impress/file_impress.scp~	2006-02-09 15:18:46.000000000 +0100
+++ scp2/source/impress/file_impress.scp	2006-04-28 12:11:26.647188928 +0200
@@ -52,31 +52,6 @@
     Name = "/registry/spool/org/openoffice/Office/Embedding-impress.xcu";
 End
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Simpress_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    TXT_FILE_BODY;
-    Styles = (ARCHIVE,DONT_OVERWRITE,PATCH);
-    EXTRA_ALL_LANG(simpress);
-    Patchfiles = ("simpress.db",
-                  "simpress.ht",
-                  "simpress.jar",
-                  "simpress.key",
-                  "simpress.tree",
-                  "simpress.idx/CONTEXTS",
-                  "simpress.idx/DICTIONARY",
-                  "simpress.idx/DOCS",
-                  "simpress.idx/DOCS.TAB",
-                  "simpress.idx/EDGE",
-                  "simpress.idx/LINKNAMES",
-                  "simpress.idx/OFFSETS",
-                  "simpress.idx/POSITIONS",
-                  "simpress.idx/SCHEMA"); 
-End
-
-#endif
-
 File gid_File_Exe_Simpress
     BIN_FILE_BODY;
     Dir = gid_Dir_Program;
--- scp2/source/impress/module_impress.scp~	2005-09-09 03:30:13.000000000 +0200
+++ scp2/source/impress/module_impress.scp	2006-04-28 12:11:26.648188776 +0200
@@ -53,9 +53,3 @@
     gid_File_Registry_Spool_Oo_TypeDetection_Impress_Filters_Xcu,gid_File_Registry_Spool_Oo_TypeDetection_Impress_Types_Xcu, gid_File_Tmp_Userinstall_Impress_Inf);
 End
 
-Module gid_Module_Prg_Impress_Help
-    MOD_NAME_DESC ( MODULE_PRG_IMPRESS_HELP );
-    ParentID = gid_Module_Prg_Impress;
-    Default = YES;
-    Files = (gid_File_Help_Simpress_Zip);
-End
--- scp2/source/math/module_math.scp~	2005-09-09 03:35:19.000000000 +0200
+++ scp2/source/math/module_math.scp	2006-04-28 12:11:28.011981448 +0200
@@ -53,9 +53,3 @@
     gid_File_Registry_Spool_Oo_TypeDetection_Math_Types_Xcu,gid_File_Registry_Spool_Oo_TypeDetection_Math_Filters_Xcu, gid_File_Tmp_Userinstall_Math_Inf);
 End
 
-Module gid_Module_Prg_Math_Help
-    MOD_NAME_DESC ( MODULE_PRG_MATH_HELP );
-    ParentID = gid_Module_Prg_Math;
-    Default = YES;
-    Files = (gid_File_Help_Smath_Zip);
-End
--- scp2/source/math/file_math.scp~	2006-03-13 15:19:50.000000000 +0100
+++ scp2/source/math/file_math.scp	2006-04-28 12:13:49.022544584 +0200
@@ -77,31 +77,6 @@
     Name = "/registry/spool/org/openoffice/Office/Embedding-math.xcu";
 End
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Smath_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(smath);
-    Patchfiles = ("smath.cfg",
-                  "smath.db",                  
-                  "smath.ht",
-                  "smath.jar",
-                  "smath.key",
-                  "smath.tree",
-                  "smath.idx/CONTEXTS",
-                  "smath.idx/DICTIONARY",
-                  "smath.idx/DOCS",
-                  "smath.idx/DOCS.TAB",
-                  "smath.idx/EDGE",
-                  "smath.idx/LINKNAMES",
-                  "smath.idx/OFFSETS",
-                  "smath.idx/POSITIONS",
-                  "smath.idx/SCHEMA"); 
-End
-
-#endif
-
 File gid_File_Exe_Smath
     BIN_FILE_BODY;
     Dir = gid_Dir_Program;
--- scp2/source/ooo/file_ooo.scp~	2006-04-28 12:08:33.177560320 +0200
+++ scp2/source/ooo/file_ooo.scp	2006-04-28 12:15:19.531785088 +0200
@@ -587,54 +587,6 @@
 
 STD_RES_FILE( gid_File_Res_Eme, eme )
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Common_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(shared);
-    Patchfiles = ("shared.db",
-                  "shared.ht",
-                  "shared.jar",
-                  "shared.key",
-                  "shared.tree",
-                  "err.html");
-End
-
-File gid_File_Help_Help_Zip
-    Dir = gid_Dir_Help;
-    ARCHIVE_TXT_FILE_BODY;
-    Name = "helpxsl.zip";
-End
-
-File gid_File_Help_Sbasic_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(sbasic);
-    Patchfiles = ("sbasic.db",
-                  "sbasic.ht",
-                  "sbasic.jar",
-                  "sbasic.key",
-                  "sbasic.tree",
-                  "sbasic.idx/CONTEXTS",
-                  "sbasic.idx/DICTIONARY",
-                  "sbasic.idx/DOCS",
-                  "sbasic.idx/DOCS.TAB",
-                  "sbasic.idx/EDGE",
-                  "sbasic.idx/LINKNAMES",
-                  "sbasic.idx/OFFSETS",
-                  "sbasic.idx/POSITIONS",
-                  "sbasic.idx/SCHEMA");
-End
-
-File gid_File_Help_Schart_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(schart);
-End
-
-#endif
-
 File gid_File_Html_Thirdpartylicensereadme
     Name = "THIRDPARTYLICENSEREADME.html";
     Dir = PREDEFINED_PROGDIR;
--- scp2/source/testtool/file_testtool.scp~	2005-09-09 03:45:36.000000000 +0200
+++ scp2/source/testtool/file_testtool.scp	2006-04-28 12:11:28.976834768 +0200
@@ -49,9 +49,3 @@
   #endif
 End
 
-File gid_File_Hid_Testtool
-    BIN_FILE_BODY;
-    Dir = gid_Dir_Program;
-    Styles = (PACKED);
-    Name = "hid.lst";
-End
--- scp2/source/writer/file_writer.scp~	2006-02-09 15:19:28.000000000 +0100
+++ scp2/source/writer/file_writer.scp	2006-04-28 12:11:28.977834616 +0200
@@ -74,30 +74,6 @@
     Name = "/registry/spool/org/openoffice/Office/Embedding-writer.xcu";
 End
 
-#ifdef SOLAR_JAVA
-
-File gid_File_Help_Swriter_Zip
-    Dir = gid_Dir_Help_Isolanguage;
-    ARCHIVE_TXT_FILE_BODY_PATCH;
-    EXTRA_ALL_LANG(swriter);
-    Patchfiles = ("swriter.db",
-                  "swriter.ht",
-                  "swriter.jar",
-                  "swriter.key",
-                  "swriter.tree",
-                  "swriter.idx/CONTEXTS",
-                  "swriter.idx/DICTIONARY",
-                  "swriter.idx/DOCS",
-                  "swriter.idx/DOCS.TAB",
-                  "swriter.idx/EDGE",
-                  "swriter.idx/LINKNAMES",
-                  "swriter.idx/OFFSETS",
-                  "swriter.idx/POSITIONS",
-                  "swriter.idx/SCHEMA"); 
-End
-
-#endif
-
 /*
 File gid_File_Share_Config_Sofficecfg_Writer_Menubar_Xml
     TXT_FILE_BODY;
--- scp2/source/writer/module_writer.scp~	2006-04-28 12:09:06.299525016 +0200
+++ scp2/source/writer/module_writer.scp	2006-04-28 12:11:28.978834464 +0200
@@ -76,10 +76,3 @@
     gid_File_Tmp_Userinstall_Wrt_Inf);
 End
 
-Module gid_Module_Prg_Wrt_Help
-    MOD_NAME_DESC ( MODULE_PRG_WRT_HELP );
-    ParentID = gid_Module_Prg_Wrt;
-    Default = YES;
-    Minimal = YES;
-    Files = (gid_File_Help_Swriter_Zip);
-End
--- postprocess/prj/build.lst~	2006-02-06 15:25:22.000000000 +0100
+++ postprocess/prj/build.lst	2006-04-28 12:11:28.978834464 +0200
@@ -1,4 +1,4 @@
-po	postprocess	::	accessibility automation basctl bean BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker helpcontent2 hwpfilter io lingucomponent MATHMLDTD:MathMLDTD MSFONTEXTRACT:msfontextract ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins scp2 scripting sd ODK:sdk_oo setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top unodevtools NULL
+po	postprocess	::	accessibility automation basctl bean BINFILTER:binfilter chart2 configmgr CRASHREP:crashrep dbaccess desktop dtrans embeddedobj embedserv EPM:epm eventattacher extensions extras fileaccess filter forms fpicker xmlhelp hwpfilter io lingucomponent MATHMLDTD:MathMLDTD MSFONTEXTRACT:msfontextract ODK:odk officecfg package padmin psprint_config remotebridges sc scaddins scp2 scripting sd ODK:sdk_oo setup_native slideshow starmath sw sysui testshl2 testtools ucb UnoControls unoxml ure wizards xmerge xmlsecurity BITSTREAM_VERA_FONTS:bitstream_vera_fonts DICTIONARIES:dictionaries OOo:pyuno OOo:readlicense_oo SO:top unodevtools NULL
 po	postprocess			    	usr1	-	all	po_mkout NULL
 po	postprocess\checkxml		nmake	-	all	po_checkxml NULL
 po	postprocess\checkdeliver	nmake	-	all	po_checkdlv NULL
