--- unotools/inc/unotools/tempfile.hxx	2005-03-03 15:24:23.000000000 +0530
+++ unotools/inc/unotools/tempfile.hxx	2005-03-03 15:25:02.000000000 +0530
@@ -107,7 +107,7 @@
                     The extension string may be f.e. ".txt" or "", if no extension string is given, ".tmp" is used
                     */
                     TempFile( const String& rLeadingChars, const String* pExtension=NULL, const String* pParent=NULL,
-                                sal_Bool bDirectory=sal_False );
+                                sal_Bool bDirectory=sal_False, sal_Bool bEmail=sal_False  );
 
                     /**
                     TempFile will be removed from disk in dtor if EnableKillingTempFile was called before.
--- unotools/source/ucbhelper/tempfile.cxx	2004-10-05 02:00:08.000000000 +0530
+++ unotools/source/ucbhelper/tempfile.cxx	2005-03-03 18:12:25.511642721 +0530
@@ -300,7 +300,7 @@
     CreateTempName_Impl( pImp->aName, sal_True, bDirectory );
 }
 
-TempFile::TempFile( const String& rLeadingChars, const String* pExtension, const String* pParent, sal_Bool bDirectory )
+TempFile::TempFile( const String& rLeadingChars, const String* pExtension, const String* pParent, sal_Bool bDirectory ,sal_Bool bEmail)
     : pImp( new TempFile_Impl )
     , bKillingFileEnabled( sal_False )
 {
@@ -314,6 +314,7 @@
     for ( sal_Int32 i=0;; i++ )
     {
         String aTmp( aName );
+	if (!bEmail)
         aTmp += String::CreateFromInt32( i );
         if ( pExtension )
             aTmp += *pExtension;
@@ -334,16 +335,18 @@
         else
         {
             File aFile( aTmp );
-            FileBase::RC err = aFile.open(osl_File_OpenFlag_Create);
-            if ( err == FileBase::E_None )
+            FileBase::RC err = aFile.open(osl_File_OpenFlag_Write | osl_File_OpenFlag_Create);
+            if ( (err == FileBase::E_None)|| (err == FileBase::E_EXIST) )
             {
                 pImp->aName = aTmp;
                 aFile.close();
                 break;
             }
-            else if ( err != FileBase::E_EXIST )
+            else  {
                 // if f.e. name contains invalid chars stop trying to create dirs
                 break;
+		}
+
         }
     }
 }
--- sfx2/source/dialog/mailmodel.cxx	2005-01-21 23:03:16.000000000 +0530
+++ sfx2/source/dialog/mailmodel.cxx	2005-03-03 18:19:42.987278772 +0530
@@ -290,12 +290,10 @@
 				pExt = new String( String::CreateFromAscii( "." ) + (OUString) aFileObj.getExtension() );
 				aFileObj.removeExtension();
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				aLeadingStr += String::CreateFromAscii( "_" );
 			}
 			else
 			{
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				aLeadingStr += String::CreateFromAscii( "_" );
 			}
 		}
 
@@ -306,7 +304,7 @@
 			pExt->Erase( 0, 1 );
 		}
 
-        ::utl::TempFile aTempFile( aLeadingStr, pExt );
+        ::utl::TempFile aTempFile( aLeadingStr, pExt,NULL,sal_False,sal_True );
 
         delete pExt;
 
@@ -392,12 +390,10 @@
 				pExt = new String( aPDFExtension );
 				aFileObj.removeExtension();
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				aLeadingStr += String::CreateFromAscii( "_" );
 			}
 			else
 			{
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
-				aLeadingStr += String::CreateFromAscii( "_" );
 			}
 		}
 
