--- config_office/configure.in
+++ config_office/configure.in
@@ -50,6 +50,13 @@
  
                           Usage: --disable-mozilla
 ],,enable_mozilla="yes")
+AC_ARG_ENABLE(epm,
+[  --disable-epm          OO.o includes self-packaging code, that requires
+			  epm, however epm is useless for large scale
+			  package building.
+
+                          Usage: --disable-epm
+],,enable_epm="yes")
 
 AC_ARG_ENABLE(cups,
 [  --enable-cups           enable cups support in the psprint project
@@ -1746,7 +1753,9 @@
 dnl Check for epm
 dnl ===================================================================
 AC_PATH_PROG(EPM, epm, no)
-if test "$EPM" = "no"; then
-   AC_MSG_ERROR([epm is needed for native installers. install epm])
+if test "z$enable_epm" = "z" -o "$enable_epm" != "no"; then
+    if test "$EPM" = "no"; then
+        AC_MSG_ERROR([epm is needed for native installers. install epm])
+    fi
 fi
 
