Index: officecfg/registry/data/org/openoffice/Office/DataAccess.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/DataAccess.xcu,v
retrieving revision 1.14
diff -u -p -u -r1.14 DataAccess.xcu
--- officecfg/registry/data/org/openoffice/Office/DataAccess.xcu	25 Jul 2003 08:26:19 -0000	1.14
+++ officecfg/registry/data/org/openoffice/Office/DataAccess.xcu	10 Mar 2004 15:48:18 -0000
@@ -1539,7 +1539,7 @@
     <node oor:name="com.sun.star.comp.sdbc.evoab.OEvoabDriver">
       <node oor:name="EvolutionPreferences">
         <prop oor:name="FullPathExportingCommand">
-          <value>file:///usr/share/evolution/*/evolution-addressbook-export</value>
+          <value>evolution-addressbook-export</value>
         </prop>
       </node>
       <node oor:name="ColumnAliases">
