Index: fpicker/source/unx/gnome/SalGtkFilePicker.cxx
===================================================================
RCS file: /cvs/gsl/fpicker/source/unx/gnome/SalGtkFilePicker.cxx,v
retrieving revision 1.11
diff -u -p -r1.11 SalGtkFilePicker.cxx
--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx	29 Mar 2006 07:41:24 -0000	1.11
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	12 May 2006 11:28:22 -0000
@@ -1864,10 +1864,10 @@ case_insensitive_filter (const GtkFileFi
     g_return_val_if_fail( data != NULL, FALSE );
     g_return_val_if_fail( filter_info != NULL, FALSE );
 
-    if( !filter_info->filename )
+    if( !filter_info->uri )
         return FALSE;
 
-    const char *pExtn = strrchr( filter_info->filename, '.' );
+    const char *pExtn = strrchr( filter_info->uri, '.' );
     if( !pExtn )
         return FALSE;
     pExtn++;
@@ -1911,7 +1914,7 @@ int SalGtkFilePicker::implAddFilter( con
                 if (aTokens.getLength())
                     aTokens += OUString::createFromAscii(",");
                 aTokens = aTokens += aToken;
-                gtk_file_filter_add_custom (filter, GTK_FILE_FILTER_FILENAME,
+                gtk_file_filter_add_custom (filter, GTK_FILE_FILTER_URI,
                     case_insensitive_filter, 
                     g_strdup( rtl::OUStringToOString( aToken, RTL_TEXTENCODING_UTF8 ) ),
                     (GDestroyNotify) g_free );
