Index: filter/source/config/fragments/makefile.mk
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/makefile.mk,v
retrieving revision 1.16
diff -u -p -u -p -r1.16 makefile.mk
--- filter/source/config/fragments/makefile.mk	14 Dec 2005 14:48:16 -0000	1.16
+++ filter/source/config/fragments/makefile.mk	28 Dec 2005 20:59:56 -0000
@@ -156,7 +156,18 @@ MERGE    := $(JAVAI) -jar $(CLASSDIR)$/F
 MERGE:=python ../tools/merge/pyAltFCFGMerge
 .ELSE
 MERGE    := $(JAVAI) -jar $(CLASSDIR)$/FCFGMerge.jar
+.ENDIF
+
+.IF "$(XSLTPROC)"=="NO_XSLTPROC"
 PACKLANG := $(JAVAI) $(JAVACPS) $(SOLARBINDIR)/xalan.jar org.apache.xalan.xslt.Process -XSL langfilter.xsl
+PACKLANG_IN := -IN
+PACKLANG_PARAM := -PARAM
+PACKLANG_XSL := -XSL
+.ELSE
+PACKLANG := $(XSLTPROC)
+PACKLANG_IN :=
+PACKLANG_PARAM := --stringparam
+PACKLANG_XSL :=
 .ENDIF
 
 # -----------------------------------------------------------------------------
@@ -189,7 +200,7 @@ $(REALFILTERPACKAGES_FILTERS_UI_LANGPACK
      +@echo Building language packages $(@:b:s/Filter_//) for filters
      +@echo ===================================================================
      +-$(MKDIRHIER) $(@:d)
-     +$(PACKLANG) -IN $< -PARAM lang $(@:d:d:d:d:d:d:d:d:b) > $@
+     +$(PACKLANG) $(PACKLANG_PARAM) lang $(@:d:d:d:d:d:d:d:d:b) $(PACKLANG_XSL) langfilter.xsl $(PACKLANG_IN) $< > $@
 
 $(REALFILTERPACKAGES_FILTERS_UI_LANGPACKS_PACKED) : $(REALFILTERPACKAGES_FILTERS_UI_LANGPACKS)
 
