diff -u -r filter.bak/source/xslt/export/excelml/ooo2spreadsheetml.xsl filter/source/xslt/export/excelml/ooo2spreadsheetml.xsl
--- filter.bak/source/xslt/export/excelml/ooo2spreadsheetml.xsl	2005-03-08 13:52:38.105858338 +0530
+++ filter/source/xslt/export/excelml/ooo2spreadsheetml.xsl	2005-03-08 13:54:15.389987063 +0530
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <!--
 
    $RCSfile$
diff -u -r filter.bak/source/xslt/export/excelml/style_mapping.xsl filter/source/xslt/export/excelml/style_mapping.xsl
--- filter.bak/source/xslt/export/excelml/style_mapping.xsl	2005-03-08 13:52:38.107858093 +0530
+++ filter/source/xslt/export/excelml/style_mapping.xsl	2005-03-08 13:54:25.580743563 +0530
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <!--
 
    $RCSfile$
diff -u -r filter.bak/source/xslt/export/excelml/styles.xsl filter/source/xslt/export/excelml/styles.xsl
--- filter.bak/source/xslt/export/excelml/styles.xsl	2005-03-08 13:52:38.108857971 +0530
+++ filter/source/xslt/export/excelml/styles.xsl	2005-03-08 13:54:25.600741123 +0530
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <!--
 
    $RCSfile$
diff -u -r filter.bak/source/xslt/export/excelml/table.xsl filter/source/xslt/export/excelml/table.xsl
--- filter.bak/source/xslt/export/excelml/table.xsl	2005-03-08 13:52:38.110857727 +0530
+++ filter/source/xslt/export/excelml/table.xsl	2005-03-08 13:54:30.240175011 +0530
@@ -1,4 +1,4 @@
-﻿<!--
+<!--
 
    $RCSfile$
 
diff -u -r filter.bak/source/xslt/import/excelml/spreadsheetml2ooo.xsl filter/source/xslt/import/excelml/spreadsheetml2ooo.xsl
--- filter.bak/source/xslt/import/excelml/spreadsheetml2ooo.xsl	2005-03-08 13:52:38.288836009 +0530
+++ filter/source/xslt/import/excelml/spreadsheetml2ooo.xsl	2005-03-08 13:54:43.953501698 +0530
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <!--
 
    $RCSfile$
