--- desktop/source/deployment/gui/license_dialog.cxx.orig	2006-07-20 02:13:17.000000000 +0200
+++ desktop/source/deployment/gui/license_dialog.cxx	2006-07-20 02:18:42.000000000 +0200
@@ -33,6 +33,10 @@
  *
  ************************************************************************/
 
+// include before comphelper/unwrapargs.hxx
+// to make function getCppuType(com::sun::star::uno::Reference<com::sun::star::awt::XWindow>*) valid
+#include "license_dialog.hxx"
+
 #include "cppuhelper/implbase2.hxx"
 #include "cppuhelper/implementationentry.hxx"
 #include "unotools/configmgr.hxx"
@@ -59,7 +63,6 @@
 
 //#include "boost/bind.hpp"
 #include "dp_gui_shared.hxx"
-#include "license_dialog.hxx"
 #include "dp_gui.hrc"
 
 using namespace ::dp_misc;
