Index: svx/source/editeng/impedit2.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/editeng/impedit2.cxx,v
retrieving revision 1.108
diff -u -p -r1.108 impedit2.cxx
--- svx/source/editeng/impedit2.cxx	14 Mar 2006 09:40:32 -0000	1.108
+++ svx/source/editeng/impedit2.cxx	18 Apr 2006 10:36:42 -0000
@@ -1595,7 +1595,7 @@ sal_Bool ImpEditEngine::IsInputSequenceC
     if (!pCTLOptions)
         pCTLOptions = new SvtCTLOptions;
 
-    rtl::OUString aTxt( (const sal_Unicode *) &nChar );
+    rtl::OUString aTxt( nChar );
     
     // get the index that really is first
     USHORT nFirstPos = rCurSel.Min().GetIndex();
 

