--- svtools/source/contnr/fileview.cxx.m163	2006-04-21 00:53:00.000000000 +0800
+++ svtools/source/contnr/fileview.cxx	2006-04-21 15:08:15.000000000 +0800
@@ -1105,10 +1105,13 @@
     if ( bFound )
     {
         SvLBoxEntry* pEntry = GetEntry( mnSearchIndex );
-		SelectAll( FALSE );
-        Select( pEntry );
-        SetCurEntry( pEntry );
-        MakeVisible( pEntry );
+        if( pEntry != NULL )
+        {
+            SelectAll( FALSE );
+            Select( pEntry ); 
+            SetCurEntry( pEntry );
+            MakeVisible( pEntry );
+        }
     }
     else
         Sound::Beep();
