--- psprint/inc/psprint/fontmanager.hxx.old	2005-01-31 09:58:54.000000000 +0100
+++ psprint/inc/psprint/fontmanager.hxx	2005-05-23 16:56:15.000000000 +0200
@@ -243,6 +243,8 @@
     KernPair() : first( 0 ), second( 0 ), kern_x( 0 ), kern_y( 0 ) {}
 };
 
+class FontCache;
+
 // a class to manage printable fonts
 // aims are type1 and truetype fonts
 
--- psprint/inc/psprint/printergfx.hxx.old	2004-09-08 16:00:04.000000000 +0200
+++ psprint/inc/psprint/printergfx.hxx	2005-05-23 17:00:30.000000000 +0200
@@ -164,6 +164,7 @@
  * forward declarations
  */
 
+class Font3;
 class GlyphSet;
 class PrinterJob;
 class PrintFontManager;
