--- configmgr/source/api2/confproviderimpl2.hxx.old	2003-03-19 17:18:31.000000000 +0100
+++ configmgr/source/api2/confproviderimpl2.hxx	2005-05-23 16:05:23.000000000 +0200
@@ -79,6 +79,8 @@
 	namespace uno		= css::uno;
 	namespace script	= css::script;
 
+	class OConfigurationProvider;
+
 	class OConfigurationProviderImpl : public OProviderImpl			
 	{		
 		friend class OConfigurationProvider;			
--- configmgr/source/api2/providerimpl.hxx.old	2004-11-15 14:35:31.000000000 +0100
+++ configmgr/source/api2/providerimpl.hxx	2005-05-23 16:01:45.000000000 +0200
@@ -130,6 +130,7 @@
     class IConfigDefaultProvider;
     class TreeManager;
     class ContextReader;
+    class OProvider;
     
     struct IConfigBroadcaster;
 
