--- dbaccess/source/core/dataaccess/ModelImpl.hxx.old	2005-03-30 13:55:09.000000000 +0200
+++ dbaccess/source/core/dataaccess/ModelImpl.hxx	2005-05-25 12:03:03.000000000 +0200
@@ -191,6 +191,9 @@
 typedef ::com::sun::star::uno::WeakReference< ::com::sun::star::sdbc::XConnection > OWeakConnection;
 typedef std::vector< OWeakConnection > OWeakConnectionArray;
 
+class ODatabaseContext;
+class OSharedConnectionManager;
+
 //============================================================
 //= ODatabaseModelImpl
 //============================================================
