--- starmath/inc/symbol.hxx.old	2005-04-04 10:03:27.000000000 +0200
+++ starmath/inc/symbol.hxx	2005-05-26 12:04:16.000000000 +0200
@@ -95,6 +95,8 @@
 #define SYMBOLSET_NONE	0xFFFF
 #define SYMBOL_NONE		0xFFFF
 
+class SmSymSetManager;
+
 ////////////////////////////////////////////////////////////////////////////////
 
 inline const String GetExportSymbolName( const String &rUiSymbolName )
--- starmath/inc/view.hxx.old	2003-04-24 19:29:12.000000000 +0200
+++ starmath/inc/view.hxx	2005-05-26 12:01:10.000000000 +0200
@@ -98,6 +98,7 @@
 
 class Menu;
 class DataChangedEvent;
+class SmClipboardChangeListener;
 class SmDocShell;
 class SmViewShell;
 
