--- svtools/inc/ctrltool.hxx.old	2005-04-13 12:04:30.000000000 +0200
+++ svtools/inc/ctrltool.hxx	2005-05-24 13:58:03.000000000 +0200
@@ -77,6 +77,7 @@
 #endif
 
 class ImplFontListNameInfo;
+class OutputDevice;
 
 /*************************************************************************
 
--- svtools/inc/inettbc.hxx.old	2005-04-13 12:16:04.000000000 +0200
+++ svtools/inc/inettbc.hxx	2005-05-23 18:50:07.000000000 +0200
@@ -80,6 +80,8 @@
 #endif
 
 class IUrlFilter;
+class SvtMatchContext_Impl;
+class SvtURLBox_Impl;
 class SVT_DLLPUBLIC SvtURLBox : public ComboBox
 {
 friend class SvtMatchContext_Impl;
--- svtools/inc/poolitem.hxx.old	2005-04-14 20:42:40.000000000 +0200
+++ svtools/inc/poolitem.hxx	2005-05-23 18:45:44.000000000 +0200
@@ -213,6 +213,8 @@
 DBG_NAMEEX(SfxItemHandle);
 
 class SvXMLUnitConverter;
+class SfxItemPool;
+class SfxItemSet;
 
 class String;
 namespace rtl
