--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2005-02-25 11:45:15.000000000 +0530
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2005-02-28 13:43:49.515957013 +0530
@@ -169,6 +169,10 @@ SalGtkFilePicker::SalGtkFilePicker( cons
 		lang::XServiceInfo>( m_rbHelperMtx ),
 	m_xServiceMgr( xServiceMgr ),
 	m_aAsyncEventNotifier( rBHelper ),
+	m_pFilterComboBox( NULL ),
+	m_pFilterComboHBox( NULL ),
+	m_pFilterComboAlign( NULL ),
+	m_pFilterComboLabel( NULL ),
 	m_pVBox ( NULL ),
 	m_pFilterList( NULL ),
     bVersionWidthUnset( false ),
@@ -302,6 +302,27 @@ SalGtkFilePicker::SalGtkFilePicker
         gtk_box_pack_end( GTK_BOX( m_pVBox ), m_pHBoxs[i], FALSE, FALSE, 0 );
 	}
 
+	m_pFilterComboBox = gtk_combo_box_new_text();
+	m_pFilterComboAlign = gtk_alignment_new(0, 0, 0, 1);
+	m_pFilterComboHBox = gtk_hbox_new( FALSE, 0 );
+	m_pFilterComboLabel = gtk_label_new( "" );
+
+	aLabel = aResProvider.getResString( FILE_PICKER_FILE_TYPE );
+	aLabel += OUString::createFromAscii(": ");
+	g_object_set( m_pFilterComboLabel, "label",
+		OUStringToOString( aLabel, RTL_TEXTENCODING_UTF8 ).getStr(),
+		"use_underline", TRUE, NULL );
+
+	gtk_container_add( GTK_CONTAINER( m_pFilterComboAlign ), m_pFilterComboBox );
+	gtk_box_pack_end( GTK_BOX( m_pFilterComboHBox ), m_pFilterComboAlign, FALSE, FALSE, 0 );
+	gtk_box_pack_end( GTK_BOX( m_pFilterComboHBox ), m_pFilterComboLabel, FALSE, FALSE, 0 );
+	gtk_box_pack_end( GTK_BOX( m_pVBox ), m_pFilterComboHBox, FALSE, FALSE, 0 );
+
+	gtk_widget_show( m_pFilterComboLabel );
+	gtk_widget_show( m_pFilterComboHBox );
+	gtk_widget_show( m_pFilterComboAlign );
+	gtk_widget_show( m_pFilterComboBox );
+
 	gtk_file_chooser_set_extra_widget( GTK_FILE_CHOOSER( m_pDialog ), m_pVBox );
 
     gtk_widget_show( m_pVBox );
@@ -739,6 +739,8 @@ rtl::OUString SAL_CALL SalGtkFilePicker:
 
 	OSL_TRACE( "GetCURRENTfilter\n" );
 
+	GtkFileChooserAction eAction = gtk_file_chooser_get_action( GTK_FILE_CHOOSER( m_pDialog ) );
+
 	// Update the filtername from the users selection if they have had a chance to do so.
 	if( GtkFileFilter *filter = gtk_file_chooser_get_filter( GTK_FILE_CHOOSER( m_pDialog ) ) )
 	{
@@ -755,6 +755,26 @@ rtl::OUString SAL_CALL SalGtkFilePicker:
         }
 	}
 
+	if( GTK_FILE_CHOOSER_ACTION_SAVE == eAction ||
+		( GTK_FILE_CHOOSER_ACTION_OPEN == eAction && 
+		  m_aCurrentFilter.equals( shrinkFilterName( m_pFilterList->begin()->getTitle() ) ) ) )
+	{
+		gint nFilterChoice = gtk_combo_box_get_active( GTK_COMBO_BOX( m_pFilterComboBox ) );
+
+		gint nFilterPos = 0;
+		for     (       FilterList::iterator aListIter = m_pFilterList->begin();
+				aListIter != m_pFilterList->end();
+				++aListIter, ++nFilterPos
+			)
+		{
+			if ( nFilterPos == nFilterChoice )
+			{
+				m_aCurrentFilter = OUString( aListIter->getTitle() );
+				break;
+			}
+		}
+	}
+
 	OSL_TRACE( "Returning current filter of %s\n", 
 		OUStringToOString( m_aCurrentFilter, RTL_TEXTENCODING_UTF8 ).getStr() );
 	
@@ -878,18 +878,22 @@ uno::Sequence<rtl::OUString> SAL_CALL Sa
 
 		if( GTK_FILE_CHOOSER_ACTION_SAVE == eAction )
 		{
-			const gchar* filtername = 
-				gtk_file_filter_get_name( gtk_file_chooser_get_filter( GTK_FILE_CHOOSER( m_pDialog ) ) );
-
-			OSL_TRACE( "2: current filter is %s\n", filtername );
-
+			OUString aFilter;
 
-			FilterList::iterator aListIter = ::std::find_if( 
-			       			m_pFilterList->begin(), m_pFilterList->end(), 
-						FilterTitleMatch( OUString( filtername,
-							strlen( filtername ), RTL_TEXTENCODING_UTF8 ) ) );
+			gint nFilterChoice = gtk_combo_box_get_active( GTK_COMBO_BOX( m_pFilterComboBox ) );
 
-			OUString aFilter = aListIter->getFilter();
+               		gint nFilterPos = 0;
+		        for     (       FilterList::iterator aListIter = m_pFilterList->begin();
+					aListIter != m_pFilterList->end();
+					++aListIter, ++nFilterPos
+				)
+			{
+				if ( nFilterPos == nFilterChoice )
+				{
+					aFilter = OUString( aListIter->getFilter() );
+					break;
+				}
+			}
 
 			OSL_TRACE( "turned into %s\n", 
 				OUStringToOString( aFilter, RTL_TEXTENCODING_UTF8 ).getStr() );
@@ -1733,6 +1733,21 @@ uno::Sequence<rtl::OUString> SAL_CALL Sa
 //-------------------------------------------------
 void SalGtkFilePicker::SetCurFilter( const OUString& rFilter )
 {
+	OUString aShrunkName = shrinkFilterName( rFilter );
+
+	gint nFilterPos = 0;
+	for     (       FilterList::iterator aListIter = m_pFilterList->begin();
+			aListIter != m_pFilterList->end();
+			++aListIter, ++nFilterPos
+		)
+	{
+		if ( aShrunkName.equals( aListIter->getTitle() ) )
+		{
+			gtk_combo_box_set_active( GTK_COMBO_BOX( m_pFilterComboBox ), nFilterPos );
+			break;
+		}
+	}
+
 	// Get all the filters already added
 	GSList *filters = gtk_file_chooser_list_filters ( GTK_FILE_CHOOSER( m_pDialog ) );
 	bool bFound = false;
@@ -1740,7 +1740,6 @@ uno::Sequence<rtl::OUString> SAL_CALL Sa
 		G_CONST_RETURN gchar * filtername = gtk_file_filter_get_name( pFilter );
 		OUString sFilterName( filtername, strlen( filtername ), RTL_TEXTENCODING_UTF8 );
 
-		OUString aShrunkName = shrinkFilterName( rFilter );
 		if( aShrunkName.equals( sFilterName) )
 		{
 			OSL_TRACE( "actually setting %s\n", filtername );
@@ -1741,6 +1811,8 @@ void SalGtkFilePicker::implAddFilter( co
 		while( nIndex >= 0 );
 	}
 
+	gtk_combo_box_append_text( GTK_COMBO_BOX( m_pFilterComboBox ), aFilterName );
+
 	gtk_file_chooser_add_filter( GTK_FILE_CHOOSER( m_pDialog ), filter );
 }
 
@@ -1782,6 +1854,7 @@ void SalGtkFilePicker::SetFilters()
 		}
 	}
 
+	gtk_combo_box_set_active( GTK_COMBO_BOX( m_pFilterComboBox ), 0 );
 
 	// set the default filter
 	if( m_aCurrentFilter && (m_aCurrentFilter.getLength() > 0) )
@@ -1814,6 +1887,11 @@ SalGtkFilePicker::~SalGtkFilePicker()
 
 	delete m_pFilterList;
 
+	gtk_widget_destroy( m_pFilterComboBox );
+	gtk_widget_destroy( m_pFilterComboAlign );
+	gtk_widget_destroy( m_pFilterComboLabel );
+	gtk_widget_destroy( m_pFilterComboHBox );
+
 	gtk_widget_destroy( m_pVBox );
 }




 
--- fpicker/source/unx/gnome/SalGtkFilePicker.hxx	2005-02-25 11:45:15.000000000 +0530
+++ fpicker/source/unx/gnome/SalGtkFilePicker.hxx	2005-02-25 12:17:01.000000000 +0530
@@ -324,6 +324,10 @@ class SalGtkFilePicker : 
 		SalGtkAsyncEventNotifier m_aAsyncEventNotifier;
 		FilterList *m_pFilterList;
 		GtkWidget  *m_pVBox;
+		GtkWidget  *m_pFilterComboBox;
+		GtkWidget  *m_pFilterComboLabel;
+		GtkWidget  *m_pFilterComboHBox;
+		GtkWidget  *m_pFilterComboAlign;
 
 		enum { 
 			AUTOEXTENSION,
