--- fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2005-03-03 15:26:08.000000000 +0530
+++ fpicker/source/unx/gnome/SalGtkFilePicker.cxx	2005-03-08 13:32:15.264113002 +0530
@@ -202,6 +202,8 @@ SalGtkFilePicker::SalGtkFilePicker( cons
 		m_pListLabels[i] = NULL;
 		mbListVisibility[i] = false;
     }
+
+	initialize(0);
 }
 
 //------------------------------------------------------------------------------------
@@ -1422,23 +1424,15 @@ sal_Bool SAL_CALL SalGtkFilePicker::getS
 void SAL_CALL SalGtkFilePicker::initialize( const uno::Sequence<uno::Any>& aArguments ) 
 	throw( uno::Exception, uno::RuntimeException )
 {
-	// parameter checking	    
-	uno::Any aAny;
-	if( 0 == aArguments.getLength() )
-		throw lang::IllegalArgumentException(
-			rtl::OUString::createFromAscii( "no arguments" ),
-			static_cast<XFilePicker*>( this ), 1 );
-   
-	aAny = aArguments[0];
-
-	if( ( aAny.getValueType() != ::getCppuType( ( sal_Int16* )0 ) ) &&
-	     (aAny.getValueType() != ::getCppuType( ( sal_Int8* )0 ) ) )
-		 throw lang::IllegalArgumentException(
-			rtl::OUString::createFromAscii( "invalid argument type" ),
-			static_cast<XFilePicker*>( this ), 1 );
+	if( m_pDialog )
+	{
+		CleanUp();
+		gtk_widget_destroy( m_pDialog );
+	}
 
-	sal_Int16 templateId = -1;
-	aAny >>= templateId;
+	sal_Int16 templateId = TemplateDescription::FILEOPEN_SIMPLE;
+	if ( aArguments.getLength() == 1 )
+		aArguments[0] >>= templateId;
 
 	GtkFileChooserAction eAction = GTK_FILE_CHOOSER_ACTION_OPEN;
 	gchar *first_button_text = GTK_STOCK_OPEN;
@@ -1874,6 +1868,11 @@ void SalGtkFilePicker::SetFilters()
 
 SalGtkFilePicker::~SalGtkFilePicker()
 {
+	CleanUp();
+}
+
+void SalGtkFilePicker::CleanUp()
+{
 	int i;
 
 	for( i = 0; i < TOGGLE_LAST; i++ )
--- fpicker/source/unx/gnome/SalGtkFilePicker.hxx	2005-03-03 15:26:08.000000000 +0530
+++ fpicker/source/unx/gnome/SalGtkFilePicker.hxx	2005-03-08 13:32:38.980239030 +0530
@@ -372,6 +372,7 @@ class SalGtkFilePicker : 
 
 		void SetCurFilter( const OUString& rFilter );
 		void SetFilters();
+		void CleanUp();
 
 		void implAddFilter( const OUString& rFilter, const OUString& rType);
 		void implAddFilterGroup( const OUString& rFilter, 
