--- svx/prj/build.lst.old	2004-11-17 11:20:45.000000000 +0100
+++ svx/prj/build.lst	2005-04-08 11:50:32.000000000 +0200
@@ -44,3 +44,4 @@
 sx	svx\source\customshapes					nmake	-	all sx_customshapes NULL
 sx	svx\source\toolbars						nmake	-	all sx_toolbars NULL
 sx	svx\util								nmake	-	all	sx_util sx_3deng sx_dlg sx_draw sx_properties sx_contact sx_mixer sx_event sx_animation sx_eeng sx_fmcmp sx_form sx_gall sx_items sx_link sx_mnuc sx_msfilt sx_opt sx_outl sx_rtf sx_sdi sx_stbc sx_tbxc sx_undrw sx_unedt sx_ungal sx_xml sx_xout sx_accessibility sx_intro sx_init sx_customshapes sx_toolbars NULL
+sx	svx\source\gengal						nmake	-	all	sx_gengal sx_util NULL
--- svx/prj/d.lst.old	2005-04-06 21:07:37.000000000 +0200
+++ svx/prj/d.lst	2005-04-08 15:18:30.000000000 +0200
@@ -25,6 +25,7 @@
 ..\%__SRC%\lib\lib*.* %_DEST%\lib%_EXT%\lib*.*
 ..\%__SRC%\lib\isvx.lib %_DEST%\lib%_EXT%\isvx.lib
 ..\%__SRC%\slb\xdraw.lib %_DEST%\lib%_EXT%\svxsmrtd.lib
+..\%__SRC%\bin\gengal* %_DEST%\bin%_EXT%\gengal*
 ..\%__SRC%\bin\svxslots.tlb %_DEST%\bin%_EXT%\svxslots.tlb
 ..\%__SRC%\bin\cui*.dll %_DEST%\bin%_EXT%\cui*.dll
 ..\%__SRC%\bin\svx*.dll %_DEST%\bin%_EXT%\svx*.dll
--- scp2/source/ooo/file_ooo.scp.old	2005-04-06 21:07:35.000000000 +0200
+++ scp2/source/ooo/file_ooo.scp	2005-04-08 12:37:27.000000000 +0200
@@ -266,6 +266,35 @@
 
 #endif
 
+File gid_File_Bin_Gengal
+    BIN_FILE_BODY;
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+  #ifdef UNX
+    Name = "gengal.bin";
+  #else
+    Name = "gengal.exe";
+  #endif
+End
+
+#ifdef UNX
+
+File gid_File_Script_Gengal
+    BIN_FILE_BODY;
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+    Name = "gengal";
+End
+
+#endif
+
+File gid_File_Rdb_Gengal
+    TXT_FILE_BODY;
+    Dir = gid_Dir_Program;
+    Styles = (PACKED, OVERWRITE);
+    Name = "gengal.rdb";
+End
+
 #ifdef UNX
 
 File gid_File_Bin_Pluginapp
--- /dev/null	2005-03-19 20:36:14.000000000 +0100
+++ svx/source/gengal/gengal.cxx	2005-04-12 12:17:34.000000000 +0200
@@ -0,0 +1,272 @@
+#include <stdio.h>
+#include <unistd.h>
+
+#include <list>
+
+#include <unotools/streamwrap.hxx>
+#include <unotools/ucbstreamhelper.hxx>
+
+#include <comphelper/processfactory.hxx>
+#include <comphelper/regpathhelper.hxx>
+#include <cppuhelper/servicefactory.hxx>
+#include <cppuhelper/bootstrap.hxx>
+#include <com/sun/star/lang/XMultiServiceFactory.hpp>
+#include <com/sun/star/lang/XInitialization.hpp>
+#include <com/sun/star/registry/XSimpleRegistry.hpp>
+
+#include <ucbhelper/contentbroker.hxx>
+#include <ucbhelper/configurationkeys.hxx>
+
+#include <tools/urlobj.hxx>
+#include <tools/fsys.hxx>
+#include <tools/intn.hxx>
+#include <svtools/filedlg.hxx>
+
+#include <vcl/window.hxx>
+#include <vcl/svapp.hxx>
+#include <vcl/font.hxx>
+#include <vcl/sound.hxx>
+#include <vcl/print.hxx>
+#include <vcl/toolbox.hxx>
+#include <vcl/help.hxx>
+#include <vcl/scrbar.hxx>
+#include <vcl/wrkwin.hxx>
+#include <vcl/msgbox.hxx>
+
+#include <osl/file.hxx>
+#include <osl/process.h>
+#include <rtl/bootstrap.hxx>
+
+#include <galtheme.hxx>
+#include <gallery1.hxx>
+
+using namespace ::vos;
+using namespace ::rtl;
+using namespace ::com::sun::star::uno;
+using namespace ::com::sun::star::registry;
+using namespace ::com::sun::star::lang;
+
+typedef ::std::list<rtl::OUString> FileNameList;
+
+class GalApp : public Application
+{
+public:
+	virtual void Main();
+};
+
+Reference< XMultiServiceFactory > createApplicationServiceManager()
+{
+    Reference< XMultiServiceFactory > xMS;
+	try
+	{
+		Reference< XComponentContext > xComponentContext = ::cppu::defaultBootstrap_InitialComponentContext();
+        if ( xComponentContext.is() )
+		    xMS = xMS.query( xComponentContext->getServiceManager() );
+	}
+	catch( ::com::sun::star::uno::Exception& )
+	{
+	}
+
+	return xMS;
+}
+
+static void createTheme( rtl::OUString aThemeName,
+						 rtl::OUString aGalleryURL,
+						 FileNameList &rFiles )
+{
+	Gallery* pGallery;
+	
+	pGallery = Gallery::AcquireGallery( aGalleryURL );
+	if (!pGallery ) {
+			fprintf( stderr, "Could't acquire '%s'\n",
+					 (const sal_Char *) rtl::OUStringToOString( aGalleryURL,
+																RTL_TEXTENCODING_UTF8 ) );
+			exit( 1 );
+	}
+	fprintf( stderr, "Work on gallery '%s'\n",
+					 (const sal_Char *) rtl::OUStringToOString( aGalleryURL, RTL_TEXTENCODING_UTF8 ) );
+
+	fprintf( stderr, "Existing themes: %d\n",
+			 pGallery->GetThemeCount() );
+
+	GalleryTheme *pGalTheme;
+	if( !pGallery->HasTheme( aThemeName) ) {
+			if( !pGallery->CreateTheme( aThemeName ) ) {
+					fprintf( stderr, "Failed to create theme\n" );
+					exit( 1 );
+			}
+	}
+
+	fprintf( stderr, "Existing themes: %d\n",
+			 pGallery->GetThemeCount() );
+
+	SfxListener aListener;
+
+	if ( !( pGalTheme = pGallery->AcquireTheme( aThemeName, aListener ) ) ) {
+			fprintf( stderr, "Failed to acquire theme\n" );
+			exit( 1 );
+	}
+
+	FileNameList::const_iterator aIter;
+	
+	for( aIter = rFiles.begin(); aIter != rFiles.end(); aIter++ )
+	{
+//  Should/could use:
+//	if ( ! pGalTheme->InsertFileOrDirURL( aURL ) ) {
+//	Requires a load more components ...
+
+		Graphic aGraphic;
+		String aFormat;
+
+#if 1
+		if ( ! pGalTheme->InsertURL( *aIter ) )
+			fprintf( stderr, "Failed to import '%s'\n",
+					 (const sal_Char *) rtl::OUStringToOString( *aIter, RTL_TEXTENCODING_UTF8 ) );
+		else
+			fprintf( stderr, "Imported file '%s' (%d)\n",
+					 (const sal_Char *) rtl::OUStringToOString( *aIter, RTL_TEXTENCODING_UTF8 ),
+					 pGalTheme->GetObjectCount() );
+
+#else // only loads BMPs
+		SvStream *pStream = ::utl::UcbStreamHelper::CreateStream( *aIter, STREAM_READ );
+		if (!pStream) {
+			fprintf( stderr, "Can't find image to import\n" );
+			exit (1);
+		}
+		*pStream >> aGraphic;
+		delete pStream;
+		if( aGraphic.GetType() == GRAPHIC_NONE )
+		{
+			fprintf( stderr, "Failed to load '%s'\n",
+					 (const sal_Char *) rtl::OUStringToOString( *aIter, RTL_TEXTENCODING_UTF8 ) );
+			continue;
+		}
+
+		SgaObjectBmp aObject( aGraphic, *aIter, aFormat );
+		if ( ! aObject.IsValid() ) {
+			fprintf( stderr, "Failed to create thumbnail for image\n" );
+			continue;
+		}
+		
+		if ( ! pGalTheme->InsertObject( aObject ) ) {
+			fprintf( stderr, "Failed to insert file or URL\n" );
+			continue;
+		} 
+#endif
+	}
+
+	pGallery->ReleaseTheme( pGalTheme, aListener );
+
+	Gallery::ReleaseGallery( pGallery );
+}
+
+static void PrintHelp()
+{
+	fprintf( stdout, "Utility to generate OO.o gallery files\n\n" );
+	
+	fprintf( stdout, "using: gengal.bin --name <name> --path <dir> [ files ... ]\n\n");
+	
+	fprintf( stdout, "options:\n");
+	fprintf( stdout, " --name <theme>\t\tdefines a name of the created or updated theme.\n");
+	fprintf( stdout, " --path <dir>\t\tdefines directory where the gallery files are created\n");
+	fprintf( stdout, "\t\t\tor updated.\n");
+	fprintf( stdout, " files\t\t\tlists files to be added to the gallery. Absolute paths\n");
+	fprintf( stdout, "\t\t\tare required.\n");
+}
+
+static OUString Smartify( const OUString &rPath )
+{
+	INetURLObject aURL;
+	aURL.SetSmartURL( rPath );
+	return aURL.GetMainURL( INetURLObject::NO_DECODE );
+}
+
+// Cut/paste [ evilness ] from Desktop::CreateApplicationServiceManager()
+Reference< XMultiServiceFactory > CreateApplicationServiceManager()
+{
+	try
+	{
+		Reference<XComponentContext> xComponentContext = ::cppu::defaultBootstrap_InitialComponentContext();
+		Reference<XMultiServiceFactory> xMS(xComponentContext->getServiceManager(), UNO_QUERY);
+
+		return xMS;
+	}
+	catch( ::com::sun::star::uno::Exception& )
+	{
+	}
+
+	return Reference< XMultiServiceFactory >();
+}
+
+void GalApp::Main()
+{
+#ifdef TOO_SIMPLE_BY_HALF
+    ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory >
+		xMSF = cppu::createRegistryServiceFactory(
+				rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "gengal.rdb" ) ), sal_True );
+	::comphelper::setProcessServiceFactory( xMSF );
+
+	// Without this no file access works ...
+	ucb::ContentProviderDataList aData;
+	ucb::ContentProviderData aFileProvider(
+			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.ucb.FileContentProvider" ) ),
+			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "file" ) ),
+			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "" ) ) );
+	aData.push_back( aFileProvider );
+	ucb::ContentBroker::initialize( xMSF, aData );
+#else
+	Reference<XComponentContext> xComponentContext = ::cppu::defaultBootstrap_InitialComponentContext();
+	Reference<XMultiServiceFactory> xMSF(xComponentContext->getServiceManager(), UNO_QUERY);
+	if( !xMSF.is() )
+		fprintf( stderr, "Failed to bootstrap\n" );
+	::comphelper::setProcessServiceFactory( xMSF );
+	
+	OUString aEmpty;
+	Sequence< Any > aArgs(6);
+	aArgs[0]
+		<<= rtl::OUString::createFromAscii(UCB_CONFIGURATION_KEY1_LOCAL);
+	aArgs[1]
+		<<= rtl::OUString::createFromAscii(UCB_CONFIGURATION_KEY2_OFFICE);
+	aArgs[2] <<= rtl::OUString::createFromAscii("PIPE");
+	aArgs[3] <<= aEmpty;
+	aArgs[4] <<= rtl::OUString::createFromAscii("PORTAL");
+	aArgs[5] <<= aEmpty;
+
+    if (! ::ucb::ContentBroker::initialize( xMSF, aArgs ) )
+			fprintf( stderr, "Failed to init content broker\n" );
+#endif
+
+
+	bool bHelp = false;
+	rtl::OUString aPath;
+	rtl::OUString aName = rtl::OUString::createFromAscii( "Default name" );
+	FileNameList aFiles;
+
+	for( USHORT i = 0; i < GetCommandLineParamCount(); i++ )
+	{
+		OUString aParam = GetCommandLineParam( i );
+
+		if( aParam.equalsAscii( "--help" ) ||
+			aParam.equalsAscii( "-h" ) )
+				bHelp = true;
+
+		else if ( aParam.equalsAscii( "--name" ) )
+			aName = GetCommandLineParam( ++i );
+
+		else if ( aParam.equalsAscii( "--path" ) )
+			aPath = Smartify( GetCommandLineParam( ++i ) );
+
+		else
+			aFiles.push_back( Smartify( aParam ) );
+	}
+
+	if( bHelp )
+	{
+		PrintHelp();
+		return;
+	}
+
+	createTheme( aName, aPath, aFiles );
+}
+
+GalApp aGalApp;
--- /dev/null	2005-03-19 20:36:14.000000000 +0100
+++ svx/source/gengal/gengal.sh	2005-04-08 15:02:24.000000000 +0200
@@ -0,0 +1,155 @@
+#!/bin/sh
+
+#*****************************************************************************
+# 
+#  gengal - wrapper for gengal.bin
+#  gengal.bin - utility to generare or update galleries usable with OpenOffice.org
+# 
+#  Initial version by: Petr Mladek <pmladek@suse.cz>
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License version 2, as
+#  published by the Free Software Foundation.
+# 
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+# 
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+# 
+#*****************************************************************************
+
+# default setting
+GAL_NAME=
+GAL_PATH=
+GAL_NUM=1
+GAL_FORCE=false
+
+# suffixes of gallery-related files
+GAL_SUFFIXES=".sdg .sdv .thm"
+
+usage()
+{
+    echo -e "Utility to generare or update galleries usable with OpenOffice.org"
+    echo -e
+    echo -e "Usage: ${0##*/} --name name --path path [--num number] [--force] file..."
+    echo -e 
+    echo -e "Options:"
+    echo -e "\t--name name:  defines a name of the gallery that will be shown in OOo."
+    echo -e "\t--path path:  defines a path where the actual version of the gallery is"
+    echo -e "\t\t      located and where to put the new version."
+    echo -e "\t--num number: defines number of the gallery. It is 1 by default."
+    echo -e "\t--force:      only a gallery with the same name can be updated. This"
+    echo -e "\t\t      option forces to replace the old uncompatible gallery with"
+    echo -e "\t\t      the new one."
+    echo -e "\tfile...:      list of file that have to be added to the gallery."
+    echo -e "\t\t      Absolute paths are required."
+}
+
+clean_and_exit()
+{
+    test "z$GAL_TMP" != "z" -a -d $GAL_TMP && rm -rf $GAL_TMP
+    exit 1;
+}
+
+
+while test "${1#--}" != "$1" ; do
+    case "$1" in
+	"--help") usage
+		exit 0;
+		;;
+	"--name")
+		shift
+		GAL_NAME="$1"
+		;;
+	"--path")
+		shift;
+		GAL_PATH="$1"
+		if test "z${GAL_PATH##/*}" != "z" ; then
+		    echo "Error: --path option requires an absolute path as the parameter"
+		    exit 1;
+		fi
+		;;
+	"--num")
+		shift;
+		GAL_NUM="$1"
+		if test "z${GAL_NUM#*[!0-9]}" != "z$GAL_NUM" ; then
+		    echo "Error: --num option requires a number as the parameter"
+		    exit 1;
+		fi
+		;;
+	"--force")
+		GAL_FORCE=true
+		;;
+	*)
+		echo "Error: unknown option \"$1\", try --help"
+		exit 1;
+		;;
+    esac
+    shift;
+done
+
+
+# checks for required options
+
+if test "z$GAL_NAME" = "z" ; then
+    echo "Error: --name is required option"
+    exit 1;
+fi
+
+if test "z$GAL_PATH" = "z" ; then
+    echo "Error: --path is required option"
+    exit 1;
+fi
+
+# checks for absolute paths
+for file in "$@" ; do
+    if test "z${file##/*}" != "z" ; then
+	echo "Error: Abosolute file names are required, such that \"$file\" is not a valid parameter."
+	exit 1;
+    fi
+done
+
+# temp dir
+GAL_TMP=`mktemp -d /tmp/gal-XXXXXXXX`
+
+# move already existing version to temp dir
+gal_orig_exist=true
+for suffix in $GAL_SUFFIXES ; do
+    test -f $GAL_PATH/sg$GAL_NUM$suffix || gal_orig_exist=false
+done
+
+if $gal_orig_exist ; then
+    for suffix in $GAL_SUFFIXES ; do
+	cp $GAL_PATH/sg$GAL_NUM$suffix $GAL_TMP/sg1$suffix || clean_and_exit;
+    done
+fi
+
+# create/update gallery
+"$0.bin" --name "$GAL_NAME" --path $GAL_TMP "$@" || clean_and_exit;
+
+# if $GAL_TMP/sd2.sdg exists, it means that the older gallery cannot be updated
+# (used another name), so a new one was created instead
+GAL_CURRENT=sg1
+if test -f $GAL_TMP/sg2.sdg ; then
+    if "$GAL_FORCE" ; then
+	# remove the older gallery, if will replaced
+	for suffix in $GAL_SUFFIXES ; do
+	    rm $GAL_TMP/$GAL_CURRENT$suffix || clean_and_exit;
+	done
+	GAL_CURRENT=sg2
+    else
+	echo "Error: The gallery \"sg$GAL_NUM\" already exists with another name."
+	echo "       Use --foce to replace it."
+	clean_and_exit
+    fi
+fi
+
+# move the greated gallery from 
+for suffix in $GAL_SUFFIXES ; do
+    mv -f $GAL_TMP/$GAL_CURRENT$suffix $GAL_PATH/sg$GAL_NUM$suffix || clean_and_exit
+done
+rmdir $GAL_TMP
--- /dev/null	2005-03-19 20:36:14.000000000 +0100
+++ svx/source/gengal/makefile.mk	2005-04-08 12:13:01.000000000 +0200
@@ -0,0 +1,141 @@
+#*************************************************************************
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to the terms of
+#   either of the following licenses
+#
+#          - GNU Lesser General Public License Version 2.1
+#          - Sun Industry Standards Source License Version 1.1
+#
+#   Sun Microsystems Inc., October, 2000
+#
+#   GNU Lesser General Public License Version 2.1
+#   =============================================
+#   Copyright 2000 by Sun Microsystems, Inc.
+#   901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#   This library is free software; you can redistribute it and/or
+#   modify it under the terms of the GNU Lesser General Public
+#   License version 2.1, as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#   Lesser General Public License for more details.
+#
+#   You should have received a copy of the GNU Lesser General Public
+#   License along with this library; if not, write to the Free Software
+#   Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#   MA  02111-1307  USA
+#
+#
+#   Sun Industry Standards Source License Version 1.1
+#   =================================================
+#   The contents of this file are subject to the Sun Industry Standards
+#   Source License Version 1.1 (the "License"); You may not use this file
+#   except in compliance with the License. You may obtain a copy of the
+#   License at http://www.openoffice.org/license.html.
+#
+#   Software provided under this License is provided on an "AS IS" basis,
+#   WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING,
+#   WITHOUT LIMITATION, WARRANTIES THAT THE SOFTWARE IS FREE OF DEFECTS,
+#   MERCHANTABLE, FIT FOR A PARTICULAR PURPOSE, OR NON-INFRINGING.
+#   See the License for the specific provisions governing your rights and
+#   obligations concerning the Software.
+#
+#   The Initial Developer of the Original Code is: Sun Microsystems, Inc.
+#
+#   Copyright: 2000 by Sun Microsystems, Inc.
+#
+#   All Rights Reserved.
+#
+#   Contributor(s): _______________________________________
+#
+#
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJNAME=svx
+
+TARGET=gengal.bin
+TARGETTYPE=GUI
+LIBTARGET=NO
+ENABLE_EXCEPTIONS=TRUE
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+.INCLUDE :  $(PRJ)$/util$/makefile.pmk
+
+# --- Files --------------------------------------------------------
+
+SCRIPTFILES =	$(BIN)$/gengal
+
+CXXFILES=	gengal.cxx
+
+OBJFILES=	$(OBJ)$/gengal.obj
+
+APP1TARGET= $(TARGET)
+APP1OBJS=	$(OBJFILES)
+APP1STDLIBS=$(TOOLSLIB) 		\
+			$(OSLLIB)			\
+			$(ONELIB)			\
+			$(VOSLIB)			\
+			$(SO2LIB)			\
+			$(SVTOOLLIB)		\
+			$(SVLLIB)			\
+			$(COMPHELPERLIB)		\
+			$(CPPULIB)		\
+			$(CPPUHELPERLIB)		\
+			$(SALLIB)		\
+			$(GOODIESLIB)		\
+			$(SFXLIB)			\
+			$(SOTLIB)			\
+			$(DLLIB)			\
+			$(SVLIB)			\
+			$(SVXLIB)
+
+.IF "$(GUI)"!="UNX"
+APP1DEF=	$(MISC)$/$(TARGET).def
+.ENDIF
+
+
+.IF "$(COM)"=="GCC"
+ADDOPTFILES=$(OBJ)$/gengal.obj
+add_cflagscxx="-frtti -fexceptions"
+.ENDIF
+
+
+# --- Targets ------------------------------------------------------
+
+ALL : \
+	ALLTAR \
+	$(SCRIPTFILES) \
+	$(BIN)$/gengal.rdb 
+
+.INCLUDE :	target.mk
+
+$(SCRIPTFILES) : $$(@:f:+".sh")
+	+@tr -d "\015" < $(@:f:+".sh") > $@
+
+$(BIN)$/gengal.rdb : makefile.mk $(UNOUCRRDB)
+	rm -f $@
+	$(GNUCOPY) $(UNOUCRRDB) $@
+	+cd $(BIN) && \
+		regcomp -register -r gengal.rdb \
+			-c i18nsearch.uno$(DLLPOST) \
+			-c i18npool.uno$(DLLPOST) \
+			-c configmgr2.uno$(DLLPOST) \
+			-c servicemgr.uno$(DLLPOST) \
+			-c libfwl$(UPD)$(DLLSUFFIX)$(DLLPOST) \
+			-c libucpfile1$(DLLPOST) \
+			-c libfileacc$(DLLPOST) \
+			-c libucb1$(DLLPOST)
+
--- scp2/source/ooo/module_hidden_ooo.scp.old	2005-02-24 17:29:25.000000000 +0100
+++ scp2/source/ooo/module_hidden_ooo.scp	2005-04-08 12:36:05.000000000 +0200
@@ -433,6 +433,9 @@
 	gid_File_Script_Unopkg,
 	gid_File_Bin_Configimport,
 	gid_File_Script_Configimport,
+	gid_File_Bin_Gengal,
+	gid_File_Script_Gengal,
+	gid_File_Rdb_Gengal,
 	gid_File_Bin_Pluginapp,
 	gid_File_Bin_Soffice,
 	gid_File_Bin_Soffice_Langpack,
