--- bridges/source/cpp_uno/gcc3_linux_powerpc/cpp2uno.cxx.old   2006-09-16 17:48:08.000000000 +0200
+++ bridges/source/cpp_uno/gcc3_linux_powerpc/cpp2uno.cxx       2007-11-22 19:56:53.000000000 +0100
@@ -36,6 +36,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_bridges.hxx"

+#include <string.h>
+
 #include <com/sun/star/uno/genfunc.hxx>
 #include <uno/data.h>
 #include <typelib/typedescription.hxx>
--- registry/tools/checksingleton.cxx	15 Oct 2007 12:28:30 -0000	1.11
+++ registry/tools/checksingleton.cxx	25 Jan 2008 09:45:26 -0000
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_registry.hxx"
 #include <stdio.h>
+#include <string.h>
 
 #include <set>

--- idlc/source/idlccompile.cxx	2 Nov 2007 12:26:14 -0000	1.22
+++ idlc/source/idlccompile.cxx	25 Jan 2008 10:05:13 -0000
@@ -71,6 +71,8 @@
 #endif
 #endif
 
+#include <string.h>
+
 using namespace ::rtl;
 using namespace ::osl;

--- idlc/source/idlcproduce.cxx	15 Oct 2007 12:44:28 -0000	1.15
+++ idlc/source/idlcproduce.cxx	25 Jan 2008 10:06:19 -0000
@@ -63,6 +63,8 @@
 #include <errno.h>
 #endif
 
+#include <string.h>
+
 using namespace ::rtl;
 using namespace ::osl;
 
Index: source/rdbmaker/rdboptions.cxx
===================================================================
RCS file: /cvs/udk/rdbmaker/source/rdbmaker/rdboptions.cxx,v
retrieving revision 1.6
diff -u -u -r1.6 rdboptions.cxx
--- rdbmaker/source/rdbmaker/rdboptions.cxx	20 Jun 2006 04:10:27 -0000	1.6
+++ rdbmaker/source/rdbmaker/rdboptions.cxx	25 Jan 2008 10:15:03 -0000
@@ -33,6 +33,7 @@
  *
  ************************************************************************/
 #include 	<stdio.h>
+#include	<string.h>
 
 #include	"rdboptions.hxx"
 
Index: gcc3_linux_x86-64/except.cxx
===================================================================
RCS file: /cvs/udk/bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx,v
retrieving revision 1.6
diff -u -u -r1.6 except.cxx
--- bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx	13 Nov 2007 14:18:22 -0000	1.6
+++ bridges/source/cpp_uno/gcc3_linux_x86-64/except.cxx	25 Jan 2008 10:18:40 -0000
@@ -37,6 +37,7 @@
 #include "precompiled_bridges.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #include <dlfcn.h>
 #include <cxxabi.h>
 #include <hash_map>
Index: context.cxx
===================================================================
RCS file: /cvs/udk/bridges/source/remote/context/context.cxx,v
retrieving revision 1.10
diff -u -u -r1.10 context.cxx
--- bridges/source/remote/context/context.cxx	16 Sep 2006 15:58:56 -0000	1.10
+++ bridges/source/remote/context/context.cxx	25 Jan 2008 10:20:28 -0000
@@ -36,6 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_bridges.hxx"
 #include <stdio.h>
+#include <string.h>
 #include <list>
 #include <hash_map>
 #include <utility>
Index: urp_marshal.hxx
===================================================================
RCS file: /cvs/udk/bridges/source/remote/urp/urp_marshal.hxx,v
retrieving revision 1.7
diff -u -u -r1.7 urp_marshal.hxx
--- bridges/source/remote/urp/urp_marshal.hxx	7 Sep 2005 22:47:30 -0000	1.7
+++ bridges/source/remote/urp/urp_marshal.hxx	25 Jan 2008 10:22:13 -0000
@@ -54,6 +54,8 @@
 #include "urp_marshal_decl.hxx"
 #endif
 
+#include <string.h>
+
 struct remote_Interface;
 
 namespace bridges_urp
Index: source/elements.cxx
===================================================================
RCS file: /cvs/udk/jvmfwk/source/elements.cxx,v
retrieving revision 1.19
diff -u -u -r1.19 elements.cxx
--- jvmfwk/source/elements.cxx	20 Jun 2007 17:13:59 -0000	1.19
+++ jvmfwk/source/elements.cxx	25 Jan 2008 10:25:05 -0000
@@ -50,6 +50,7 @@
 #include "libxml/xpathInternals.h"
 #include "rtl/bootstrap.hxx"
 #include "boost/optional.hpp"
+#include <string.h>
 // #define NS_JAVA_FRAMEWORK "http://openoffice.org/2004/java/framework/1.0"
 // #define NS_SCHEMA_INSTANCE "http://www.w3.org/2001/XMLSchema-instance"
 
Index: source/fwkutil.cxx
===================================================================
RCS file: /cvs/udk/jvmfwk/source/fwkutil.cxx,v
retrieving revision 1.27
diff -u -u -r1.27 fwkutil.cxx
--- jvmfwk/source/fwkutil.cxx	15 Oct 2007 12:07:40 -0000	1.27
+++ jvmfwk/source/fwkutil.cxx	25 Jan 2008 10:26:40 -0000
@@ -47,6 +47,7 @@
 #endif
 
 #include <string>
+#include <string.h>
 #include "osl/mutex.hxx"
 #include "osl/module.hxx"
 #include "osl/thread.hxx"
Index: plugins/sunmajor/pluginlib/sunjavaplugin.cxx
===================================================================
RCS file: /cvs/udk/jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx,v
retrieving revision 1.23
diff -u -u -r1.23 sunjavaplugin.cxx
--- jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx	13 Nov 2007 14:20:50 -0000	1.23
+++ jvmfwk/plugins/sunmajor/pluginlib/sunjavaplugin.cxx	25 Jan 2008 10:27:48 -0000
@@ -38,6 +38,7 @@
 #if OSL_DEBUG_LEVEL > 0
 #include <stdio.h>
 #endif
+#include <string.h>
 
 #include "boost/scoped_array.hpp"
 #include "osl/diagnose.h"
Index: plugins/sunmajor/pluginlib/util.cxx
===================================================================
RCS file: /cvs/udk/jvmfwk/plugins/sunmajor/pluginlib/util.cxx,v
retrieving revision 1.14
diff -u -u -r1.14 util.cxx
--- jvmfwk/plugins/sunmajor/pluginlib/util.cxx	6 Dec 2007 17:44:50 -0000	1.14
+++ jvmfwk/plugins/sunmajor/pluginlib/util.cxx	25 Jan 2008 10:28:59 -0000
@@ -61,6 +61,7 @@
 #pragma warning(pop)
 #endif
 #endif
+#include <string.h>
 
 #include "sunjre.hxx"
 #include "vendorlist.hxx"
Index: sax_expat.cxx
===================================================================
RCS file: /cvs/xml/sax/source/expatwrap/sax_expat.cxx,v
retrieving revision 1.15
diff -u -u -r1.15 sax_expat.cxx
--- sax/source/expatwrap/sax_expat.cxx	19 Jun 2006 23:07:26 -0000	1.15
+++ sax/source/expatwrap/sax_expat.cxx	25 Jan 2008 10:33:34 -0000
@@ -33,6 +33,7 @@
  *
  ************************************************************************/
 #include <stdlib.h>
+#include <string.h>
 #include <sal/alloca.h>
 #include <vector>

--- sax/source/fastparser/fastparser.cxx-old	2008-01-25 11:34:58.000000000 +0100
+++ sax/source/fastparser/fastparser.cxx	2008-01-25 11:34:33.000000000 +0100
@@ -65,6 +65,7 @@
 #else
 #include "expat/xmlparse.h"
 #endif
+#include <string.h>
 
 using ::rtl::OUString;
 using ::rtl::OString;

--- hwpfilter/source/formula.cpp	20 Jun 2006 00:52:46 -0000	1.3
+++ hwpfilter/source/formula.cpp	25 Jan 2008 10:40:54 -0000
@@ -39,7 +39,7 @@
 #include "nodes.h"
 #include "mapping.h"
 #include "hwpeq.h"
-#include <iostream.h>
+#include <iostream>
 
 extern LinkedList<Node> nodelist;
 
Index: source/zipapi/MemoryByteGrabber.hxx
===================================================================
RCS file: /cvs/xml/package/source/zipapi/MemoryByteGrabber.hxx,v
retrieving revision 1.5
diff -u -u -r1.5 MemoryByteGrabber.hxx
--- package/source/zipapi/MemoryByteGrabber.hxx	20 Jun 2006 06:13:19 -0000	1.5
+++ package/source/zipapi/MemoryByteGrabber.hxx	25 Jan 2008 10:47:26 -0000
@@ -41,6 +41,7 @@
 #ifndef _COM_SUN_STAR_IO_XSEEKABLE_HPP_
 #include <com/sun/star/io/XSeekable.hpp>
 #endif
+#include <string.h>
 
 class MemoryByteGrabber 
 {
Index: source/zipapi/XUnbufferedStream.cxx
===================================================================
RCS file: /cvs/xml/package/source/zipapi/XUnbufferedStream.cxx,v
retrieving revision 1.12
diff -u -u -r1.12 XUnbufferedStream.cxx
--- package/source/zipapi/XUnbufferedStream.cxx	17 Sep 2006 17:27:48 -0000	1.12
+++ package/source/zipapi/XUnbufferedStream.cxx	25 Jan 2008 10:47:26 -0000
@@ -60,6 +60,7 @@
 #include <EncryptedDataHeader.hxx>
 #endif
 #include <algorithm>
+#include <string.h>
 
 #if 0
 // for debugging purposes here
? unxlngx6.pro
Index: source/defaultnumberingprovider/defaultnumberingprovider.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx,v
retrieving revision 1.28
diff -u -u -r1.28 defaultnumberingprovider.cxx
--- i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx	13 Nov 2007 14:32:47 -0000	1.28
+++ i18npool/source/defaultnumberingprovider/defaultnumberingprovider.cxx	25 Jan 2008 10:52:32 -0000
@@ -51,6 +51,7 @@
 #include <localedata.hxx>
 #include <nativenumbersupplier.hxx>
 #include <stdio.h>
+#include <string.h>
 
 #if OSL_DEBUG_LEVEL == 0 && !defined(NDEBUG)
 #define NDEBUG
Index: source/localedata/saxparser.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/localedata/saxparser.cxx,v
retrieving revision 1.13
diff -u -u -r1.13 saxparser.cxx
--- i18npool/source/localedata/saxparser.cxx	17 Sep 2006 09:22:14 -0000	1.13
+++ i18npool/source/localedata/saxparser.cxx	25 Jan 2008 10:52:32 -0000
@@ -37,6 +37,7 @@
 #include "precompiled_i18npool.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #include <stack>
 
 #include "sal/main.h"
Index: source/search/textsearch.cxx
===================================================================
RCS file: /cvs/l10n/i18npool/source/search/textsearch.cxx,v
retrieving revision 1.10
diff -u -u -r1.10 textsearch.cxx
--- i18npool/source/search/textsearch.cxx	5 Dec 2007 16:45:18 -0000	1.10
+++ i18npool/source/search/textsearch.cxx	25 Jan 2008 10:52:32 -0000
@@ -85,6 +85,8 @@
 #pragma warning( disable: 4786 )
 #endif
 
+#include <string.h>
+
 using namespace ::com::sun::star::util;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::lang;
? qa
? unxlngx6.pro
? prj/tests.lst
Index: inc/bootstrp/command.hxx
===================================================================
RCS file: /cvs/util/tools/inc/bootstrp/command.hxx,v
retrieving revision 1.4
diff -u -u -r1.4 command.hxx
--- tools/inc/bootstrp/command.hxx	2 Nov 2007 12:59:41 -0000	1.4
+++ tools/inc/bootstrp/command.hxx	25 Jan 2008 10:56:47 -0000
@@ -36,7 +36,7 @@
 #ifndef COMMAND_HXX
 #define COMMAND_HXX
 
-#include <iostream.h>
+#include <iostream>
 
 #include <tools/stream.hxx>
 #define STRLEN 100
Index: source/fsys/urlobj.cxx
===================================================================
RCS file: /cvs/util/tools/source/fsys/urlobj.cxx,v
retrieving revision 1.60
diff -u -u -r1.60 urlobj.cxx
--- tools/source/fsys/urlobj.cxx	27 Jun 2007 22:12:02 -0000	1.60
+++ tools/source/fsys/urlobj.cxx	25 Jan 2008 10:56:47 -0000
@@ -81,6 +81,8 @@
 #define INCLUDED_LIMITS
 #endif
 
+#include <string.h>
+
 namespace unnamed_tools_urlobj {} using namespace unnamed_tools_urlobj;
 	// unnamed namespaces don't work well yet...

--- psprint/source/helper/helper.cxx	16 Sep 2006 12:34:47 -0000	1.29
+++ psprint/source/helper/helper.cxx	25 Jan 2008 11:48:05 -0000
@@ -39,6 +39,7 @@
 #include <cstring>
 #include <sys/stat.h>
 #include <unistd.h>
+#include <limits.h>
 
 #include <psprint/helper.hxx>
 #include <tools/string.hxx>
Index: unx/source/app/i18n_im.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/i18n_im.cxx,v
retrieving revision 1.37
diff -u -u -r1.37 i18n_im.cxx
--- vcl/unx/source/app/i18n_im.cxx	17 Sep 2006 12:32:10 -0000	1.37
+++ vcl/unx/source/app/i18n_im.cxx	25 Jan 2008 11:55:21 -0000
@@ -37,6 +37,7 @@
 #include "precompiled_vcl.hxx"
 
 #include <stdio.h>
+#include <string.h>
 #ifdef LINUX
 #  ifndef __USE_XOPEN
 #    define __USE_XOPEN
Index: inc/xmloff/PropertySetInfoHash.hxx
===================================================================
RCS file: /cvs/xml/xmloff/inc/xmloff/PropertySetInfoHash.hxx,v
retrieving revision 1.2
diff -u -u -r1.2 PropertySetInfoHash.hxx
--- xmloff/inc/xmloff/PropertySetInfoHash.hxx	11 Apr 2007 13:20:51 -0000	1.2
+++ xmloff/inc/xmloff/PropertySetInfoHash.hxx	25 Jan 2008 12:11:09 -0000
@@ -39,6 +39,7 @@
 #include <xmloff/PropertySetInfoKey.hxx>
 #endif
 
+#include <string.h>
 #include <memory>
 
 struct PropertySetInfoHash
--- xmloff/inc/xmloff/nmspmap.hxx.gcc43~        2007-04-11 15:27:58.000000000 +0200
+++ xmloff/inc/xmloff/nmspmap.hxx       2007-11-06 00:27:51.000000000 +0100
@@ -61,6 +61,8 @@
 #include <cppuhelper/weak.hxx>
 #endif

+#include <limits.h>
+
 const sal_uInt16 XML_NAMESPACE_XMLNS  = (USHRT_MAX-2);
 const sal_uInt16 XML_NAMESPACE_NONE  = (USHRT_MAX-1);
 const sal_uInt16 XML_NAMESPACE_UNKNOWN  = (USHRT_MAX);
Index: source/draw/EnhancedCustomShapeToken.cxx
===================================================================
RCS file: /cvs/xml/xmloff/source/draw/EnhancedCustomShapeToken.cxx,v
retrieving revision 1.9
diff -u -u -r1.9 EnhancedCustomShapeToken.cxx
--- xmloff/source/draw/EnhancedCustomShapeToken.cxx	17 Sep 2006 10:24:28 -0000	1.9
+++ xmloff/source/draw/EnhancedCustomShapeToken.cxx	25 Jan 2008 12:15:15 -0000
@@ -43,6 +43,7 @@
 #include <osl/mutex.hxx>
 #endif
 #include <hash_map>
+#include <string.h>
 
 namespace xmloff { namespace EnhancedCustomShapeToken {
 
Index: source/parse/sqlnode.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/parse/sqlnode.cxx,v
retrieving revision 1.50.16.2
diff -u -u -r1.50.16.2 sqlnode.cxx
--- connectivity/source/parse/sqlnode.cxx	18 Jan 2008 15:10:05 -0000	1.50.16.2
+++ connectivity/source/parse/sqlnode.cxx	25 Jan 2008 12:24:10 -0000
@@ -146,6 +146,7 @@
 #ifndef TOOLS_DIAGNOSE_EX_H
 #include <tools/diagnose_ex.h>
 #endif
+#include <string.h>
 
 
 using namespace ::com::sun::star::sdbc;
Index: source/parse/sqlbison.y
===================================================================
RCS file: /cvs/dba/connectivity/source/parse/sqlbison.y,v
retrieving revision 1.60.16.1
diff -u -u -r1.60.16.1 sqlbison.y
--- connectivity/source/parse/sqlbison.y	18 Jan 2008 15:09:48 -0000	1.60.16.1
+++ connectivity/source/parse/sqlbison.y	25 Jan 2008 12:25:54 -0000
@@ -17,6 +17,7 @@
 //--------------------------------------------------------------------------
 
 #include <vector>
+#include <string.h>
 
 #ifndef _CONNECTIVITY_SQLNODE_HXX
 #include <connectivity/sqlnode.hxx>
Index: source/commontools/sqlerror.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/commontools/sqlerror.cxx,v
retrieving revision 1.2
diff -u -u -r1.2 sqlerror.cxx
--- connectivity/source/commontools/sqlerror.cxx	21 Nov 2007 14:59:40 -0000	1.2
+++ connectivity/source/commontools/sqlerror.cxx	25 Jan 2008 12:28:06 -0000
@@ -46,6 +46,8 @@
 #include <cppuhelper/exc_hlp.hxx>
 #include <rtl/ustrbuf.hxx>
 
+#include <string.h>
+
 //........................................................................
 namespace connectivity
 {
Index: source/drivers/hsqldb/HStorageAccess.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/hsqldb/HStorageAccess.cxx,v
retrieving revision 1.11
diff -u -u -r1.11 HStorageAccess.cxx
--- connectivity/source/drivers/hsqldb/HStorageAccess.cxx	17 Sep 2006 02:40:38 -0000	1.11
+++ connectivity/source/drivers/hsqldb/HStorageAccess.cxx	25 Jan 2008 12:29:58 -0000
@@ -60,6 +60,7 @@
 #include "diagnose_ex.h"
 #endif
 
+#include <string.h>
 
 using namespace ::com::sun::star::container;
 using namespace ::com::sun::star::uno;
Index: source/drivers/odbc/OConnection.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/odbc/OConnection.cxx,v
retrieving revision 1.41.70.1
diff -u -u -r1.41.70.1 OConnection.cxx
--- connectivity/source/drivers/odbc/OConnection.cxx	18 Jan 2008 15:02:53 -0000	1.41.70.1
+++ connectivity/source/drivers/odbc/OConnection.cxx	25 Jan 2008 12:30:56 -0000
@@ -81,6 +81,8 @@
 #include <connectivity/dbexception.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity::odbc;
 using namespace connectivity;
 using namespace dbtools;
Index: source/drivers/odbc/OPreparedStatement.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/odbc/OPreparedStatement.cxx,v
retrieving revision 1.45.26.1
diff -u -u -r1.45.26.1 OPreparedStatement.cxx
--- connectivity/source/drivers/odbc/OPreparedStatement.cxx	18 Jan 2008 15:03:20 -0000	1.45.26.1
+++ connectivity/source/drivers/odbc/OPreparedStatement.cxx	25 Jan 2008 12:31:59 -0000
@@ -37,6 +37,7 @@
 #include "precompiled_connectivity.hxx"
 
 #include <stdio.h>
+#include <string.h>
 
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
Index: source/drivers/adabas/BConnection.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/adabas/BConnection.cxx,v
retrieving revision 1.23.70.1
diff -u -u -r1.23.70.1 BConnection.cxx
--- connectivity/source/drivers/adabas/BConnection.cxx	18 Jan 2008 14:52:55 -0000	1.23.70.1
+++ connectivity/source/drivers/adabas/BConnection.cxx	25 Jan 2008 12:33:39 -0000
@@ -76,6 +76,8 @@
 #include <connectivity/sqlparse.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity::adabas;
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
Index: source/drivers/evoab2/NDatabaseMetaData.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx,v
retrieving revision 1.6.178.1
diff -u -u -r1.6.178.1 NDatabaseMetaData.cxx
--- connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx	18 Jan 2008 14:56:06 -0000	1.6.178.1
+++ connectivity/source/drivers/evoab2/NDatabaseMetaData.cxx	25 Jan 2008 12:35:48 -0000
@@ -63,6 +63,7 @@
 #include <com/sun/star/sdbc/ColumnSearch.hpp>
 
 #include <vector>
+#include <string.h>
 
 #ifndef _CONNECTIVITY_EVOAB_EVOLUTION_API_HXX_
 #include "EApi.h"
Index: source/drivers/jdbc/Blob.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/Blob.cxx,v
retrieving revision 1.10
diff -u -u -r1.10 Blob.cxx
--- connectivity/source/drivers/jdbc/Blob.cxx	17 Sep 2006 02:44:18 -0000	1.10
+++ connectivity/source/drivers/jdbc/Blob.cxx	25 Jan 2008 12:36:51 -0000
@@ -52,6 +52,8 @@
 #include <connectivity/dbexception.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity;
 //**************************************************************
 //************ Class: java.sql.Blob
Index: source/drivers/jdbc/CallableStatement.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/CallableStatement.cxx,v
retrieving revision 1.18
diff -u -u -r1.18 CallableStatement.cxx
--- connectivity/source/drivers/jdbc/CallableStatement.cxx	27 Jun 2007 14:35:19 -0000	1.18
+++ connectivity/source/drivers/jdbc/CallableStatement.cxx	25 Jan 2008 12:37:44 -0000
@@ -64,6 +64,8 @@
 #include <comphelper/sequence.hxx>
 #endif
 
+#include <string.h>
+
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::beans;
Index: source/drivers/jdbc/InputStream.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/InputStream.cxx,v
retrieving revision 1.11
diff -u -u -r1.11 InputStream.cxx
--- connectivity/source/drivers/jdbc/InputStream.cxx	17 Sep 2006 02:47:00 -0000	1.11
+++ connectivity/source/drivers/jdbc/InputStream.cxx	25 Jan 2008 12:38:56 -0000
@@ -42,6 +42,8 @@
 #include "java/tools.hxx"
 #endif
 
+#include <string.h>
+
 using namespace connectivity;
 //**************************************************************
 //************ Class: java.io.InputStream
Index: source/drivers/jdbc/JStatement.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/JStatement.cxx,v
retrieving revision 1.5
diff -u -u -r1.5 JStatement.cxx
--- connectivity/source/drivers/jdbc/JStatement.cxx	27 Nov 2007 12:02:27 -0000	1.5
+++ connectivity/source/drivers/jdbc/JStatement.cxx	25 Jan 2008 12:39:45 -0000
@@ -74,6 +74,7 @@
 #include "resource/jdbc_log.hrc"
 
 #include <algorithm>
+#include <string.h>
 
 using namespace ::comphelper;
 using namespace connectivity;
Index: source/drivers/jdbc/PreparedStatement.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/PreparedStatement.cxx,v
retrieving revision 1.20.16.2
diff -u -u -r1.20.16.2 PreparedStatement.cxx
--- connectivity/source/drivers/jdbc/PreparedStatement.cxx	21 Jan 2008 15:50:39 -0000	1.20.16.2
+++ connectivity/source/drivers/jdbc/PreparedStatement.cxx	25 Jan 2008 12:41:13 -0000
@@ -68,6 +68,8 @@
 #endif
 #include "resource/jdbc_log.hrc"
 
+#include <string.h>
+
 using namespace connectivity;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::beans;
Index: source/drivers/jdbc/Reader.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/Reader.cxx,v
retrieving revision 1.10
diff -u -u -r1.10 Reader.cxx
--- connectivity/source/drivers/jdbc/Reader.cxx	17 Sep 2006 02:48:02 -0000	1.10
+++ connectivity/source/drivers/jdbc/Reader.cxx	25 Jan 2008 12:41:47 -0000
@@ -42,6 +42,7 @@
 #ifndef _INC_MEMORY
 //#include <memory.h>
 #endif
+#include <string.h>
 using namespace connectivity;
 //**************************************************************
 //************ Class: java.io.Reader
Index: source/drivers/jdbc/ResultSet.cxx
===================================================================
RCS file: /cvs/dba/connectivity/source/drivers/jdbc/ResultSet.cxx,v
retrieving revision 1.32.16.1
diff -u -u -r1.32.16.1 ResultSet.cxx
--- connectivity/source/drivers/jdbc/ResultSet.cxx	18 Jan 2008 14:59:49 -0000	1.32.16.1
+++ connectivity/source/drivers/jdbc/ResultSet.cxx	25 Jan 2008 12:42:27 -0000
@@ -97,6 +97,8 @@
 #include "connectivity/dbexception.hxx"
 #endif
 
+#include <string.h>
+
 using namespace ::comphelper;
 
 using namespace connectivity;
Index: source/services/substitutepathvars.cxx
===================================================================
RCS file: /cvs/framework/framework/source/services/substitutepathvars.cxx,v
retrieving revision 1.17
diff -u -u -r1.17 substitutepathvars.cxx
--- framework/source/services/substitutepathvars.cxx	16 Sep 2006 14:12:15 -0000	1.17
+++ framework/source/services/substitutepathvars.cxx	25 Jan 2008 12:57:57 -0000
@@ -130,6 +130,8 @@
 
 #include <comphelper/configurationhelper.hxx>
 
+#include <string.h>
+
 //_________________________________________________________________________________________________________________
 //	Defines
 //_________________________________________________________________________________________________________________
Index: source/backends/gconfbe/gconflayer.cxx
===================================================================
RCS file: /cvs/gsl/shell/source/backends/gconfbe/gconflayer.cxx,v
retrieving revision 1.13
diff -u -u -r1.13 gconflayer.cxx
--- shell/source/backends/gconfbe/gconflayer.cxx	19 Jun 2007 16:11:58 -0000	1.13
+++ shell/source/backends/gconfbe/gconflayer.cxx	25 Jan 2008 13:52:14 -0000
@@ -72,6 +72,8 @@
 #include <com/sun/star/uno/Sequence.hxx>
 #endif
 
+#include <string.h>
+
 using namespace rtl;
 
 //==============================================================================
--- shell/source/unix/sysshell/recently_used_file_handler.cxx-old	2008-01-25 14:53:34.000000000 +0100
+++ shell/source/unix/sysshell/recently_used_file_handler.cxx	2008-01-25 14:53:53.000000000 +0100
@@ -69,6 +69,7 @@
 #include <vector>
 #include <algorithm>
 #include <functional>
+#include <string.h>
 
 namespace /* private */ {
     
Index: databases.cxx
===================================================================
RCS file: /cvs/util/xmlhelp/source/cxxhelp/provider/databases.cxx,v
retrieving revision 1.49
diff -u -u -r1.49 databases.cxx
--- xmlhelp/source/cxxhelp/provider/databases.cxx	19 Nov 2007 13:00:39 -0000	1.49
+++ xmlhelp/source/cxxhelp/provider/databases.cxx	25 Jan 2008 14:01:28 -0000
@@ -71,6 +74,7 @@
 #endif
 #include "inputstream.hxx"
 #include <algorithm>
+#include <string.h>
 
 // Extensible help
 #include "com/sun/star/deployment/thePackageManagerFactory.hpp"
Index: HelpCompiler.cxx
===================================================================
RCS file: /cvs/util/xmlhelp/source/com/sun/star/help/HelpCompiler.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 HelpCompiler.cxx
--- xmlhelp/source/com/sun/star/help/HelpCompiler.cxx	19 Nov 2007 12:59:14 -0000	1.4
+++ xmlhelp/source/com/sun/star/help/HelpCompiler.cxx	25 Jan 2008 14:03:28 -0000
@@ -37,6 +37,7 @@
 #include "HelpCompiler.hxx"
 #include <limits.h>
 #include <stdlib.h>
+#include <string.h>
 #include <libxslt/xslt.h>
 #include <libxslt/xsltInternals.h>
 #include <libxslt/transform.h>
Index: HelpLinker.cxx
===================================================================
RCS file: /cvs/util/xmlhelp/source/com/sun/star/help/HelpLinker.cxx,v
retrieving revision 1.10
diff -u -u -r1.10 HelpLinker.cxx
--- xmlhelp/source/com/sun/star/help/HelpLinker.cxx	26 Nov 2007 18:04:22 -0000	1.10
+++ xmlhelp/source/com/sun/star/help/HelpLinker.cxx	25 Jan 2008 14:05:27 -0000
@@ -37,6 +37,9 @@
 
 #include <map>
 
+#include <string.h>
+#include <limits.h>
+
 #include <boost/shared_ptr.hpp>
 #include <boost/tokenizer.hpp>
 
Index: source/controls/progressbar.cxx
===================================================================
RCS file: /cvs/gsl/UnoControls/source/controls/progressbar.cxx,v
retrieving revision 1.7
diff -u -u -r1.7 progressbar.cxx
--- UnoControls/source/controls/progressbar.cxx	19 Jun 2006 23:09:34 -0000	1.7
+++ UnoControls/source/controls/progressbar.cxx	25 Jan 2008 14:13:26 -0000
@@ -69,6 +69,7 @@
 #endif
 
 #include <math.h>
+#include <limits.h>
 
 //____________________________________________________________________________________________________________
 //	includes of my project
Index: pppoptimizertoken.cxx
===================================================================
RCS file: /cvs/graphics/sdext/source/minimizer/pppoptimizertoken.cxx,v
retrieving revision 1.7
diff -u -u -r1.7 pppoptimizertoken.cxx
--- sdext/source/minimizer/pppoptimizertoken.cxx	17 Aug 2007 09:38:07 -0000	1.7
+++ sdext/source/minimizer/pppoptimizertoken.cxx	25 Jan 2008 15:00:25 -0000
@@ -40,6 +40,7 @@
 #include <osl/mutex.hxx>
 #endif
 #include <hash_map>
+#include <string.h>
 
 struct TCheck
 {
Index: mscodec.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/mscodec.cxx,v
retrieving revision 1.5
diff -u -u -r1.5 mscodec.cxx
--- svx/source/msfilter/mscodec.cxx	12 Oct 2006 12:58:20 -0000	1.5
+++ svx/source/msfilter/mscodec.cxx	25 Jan 2008 15:30:50 -0000
@@ -42,6 +42,7 @@
 
 #include <osl/diagnose.h>
 #include <algorithm>
+#include <string.h>
 #include <tools/solar.h>
 
 namespace svx {
Index: KeySet.cxx
===================================================================
RCS file: /cvs/dba/dbaccess/source/core/api/KeySet.cxx,v
retrieving revision 1.68.18.1
diff -u -u -r1.68.18.1 KeySet.cxx
--- dbaccess/source/core/api/KeySet.cxx	18 Jan 2008 15:34:45 -0000	1.68.18.1
+++ dbaccess/source/core/api/KeySet.cxx	25 Jan 2008 15:48:19 -0000
@@ -116,6 +116,7 @@
 #ifndef _TOOLS_DEBUG_HXX
 #include <tools/debug.hxx>
 #endif
+#include <string.h>
 
 using namespace dbaccess;
 using namespace ::connectivity;
Index: services.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/so_comp/services.cxx,v
retrieving revision 1.7
diff -u -u -r1.7 services.cxx
--- desktop/source/so_comp/services.cxx	12 Oct 2006 14:17:43 -0000	1.7
+++ desktop/source/so_comp/services.cxx	25 Jan 2008 17:16:35 -0000
@@ -61,6 +61,7 @@
 #include "oemjob.hxx"
 #include "evaluation.hxx"
 
+#include <string.h>
 
 using namespace rtl;
 using namespace ::com::sun::star::uno;
Index: swffilter.cxx
===================================================================
RCS file: /cvs/framework/filter/source/flash/swffilter.cxx,v
retrieving revision 1.6
diff -u -u -r1.6 swffilter.cxx
--- filter/source/flash/swffilter.cxx	18 Jan 2007 14:09:46 -0000	1.6
+++ filter/source/flash/swffilter.cxx	25 Jan 2008 17:42:51 -0000
@@ -88,6 +88,7 @@
 
 //#include <stdlib.h>
 //#include <windows.h>
+#include <string.h>
 
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::frame;
Index: XmlFilterAdaptor.cxx
===================================================================
RCS file: /cvs/framework/filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx,v
retrieving revision 1.18
diff -u -u -r1.18 XmlFilterAdaptor.cxx
--- filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx	1 Dec 2006 14:32:11 -0000	1.18
+++ filter/source/xmlfilteradaptor/XmlFilterAdaptor.cxx	25 Jan 2008 17:45:45 -0000
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_filter.hxx"
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
Index: filterdetect.cxx
===================================================================
RCS file: /cvs/framework/filter/source/xmlfilterdetect/filterdetect.cxx,v
retrieving revision 1.9
diff -u -u -r1.9 filterdetect.cxx
--- filter/source/xmlfilterdetect/filterdetect.cxx	23 Oct 2007 11:40:42 -0000	1.9
+++ filter/source/xmlfilterdetect/filterdetect.cxx	25 Jan 2008 17:47:40 -0000
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_filter.hxx"
-#include <iostream.h>
+#include <iostream>
 #include <stdlib.h>
 #include <ctype.h>
 #include <stdio.h>
Index: ComboBox.cxx
===================================================================
RCS file: /cvs/gsl/forms/source/component/ComboBox.cxx,v
retrieving revision 1.38
diff -u -u -r1.38 ComboBox.cxx
--- forms/source/component/ComboBox.cxx	21 Nov 2007 17:15:47 -0000	1.38
+++ forms/source/component/ComboBox.cxx	25 Jan 2008 19:47:49 -0000
@@ -68,6 +68,8 @@
 #include <tools/diagnose_ex.h>
 #include <unotools/sharedunocomponent.hxx>
 
+#include <limits.h>
+
 using namespace dbtools;
 
 //.........................................................................
Index: guess.cxx
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/languageguessing/guess.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 guess.cxx
--- lingucomponent/source/languageguessing/guess.cxx	22 Jun 2007 08:32:42 -0000	1.4
+++ lingucomponent/source/languageguessing/guess.cxx	25 Jan 2008 20:15:03 -0000
@@ -37,6 +37,7 @@
 #include "precompiled_lingucomponent.hxx"
 
 #include <iostream>
+#include <string.h>
 
 #include <libtextcat/textcat.h>
 #include <altstrfunc.hxx>
Index: source/dom/attr.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/attr.cxx,v
retrieving revision 1.7
diff -u -u -r1.7 attr.cxx
--- unoxml/source/dom/attr.cxx	6 Dec 2007 10:57:59 -0000	1.7
+++ unoxml/source/dom/attr.cxx	25 Jan 2008 22:02:25 -0000
@@ -36,6 +36,7 @@
 #include "attr.hxx"
 #include "element.hxx"
 #include <com/sun/star/xml/dom/DOMException.hdl>
+#include <string.h>
 
 namespace DOM
 {
Index: source/dom/attributesmap.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/attributesmap.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 attributesmap.cxx
--- unoxml/source/dom/attributesmap.cxx	20 Jun 2006 00:43:23 -0000	1.4
+++ unoxml/source/dom/attributesmap.cxx	25 Jan 2008 22:02:25 -0000
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "attributesmap.hxx"
+#include <string.h>
 
 namespace DOM
 {
Index: source/dom/characterdata.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/characterdata.cxx,v
retrieving revision 1.9
diff -u -u -r1.9 characterdata.cxx
--- unoxml/source/dom/characterdata.cxx	6 Dec 2007 10:58:40 -0000	1.9
+++ unoxml/source/dom/characterdata.cxx	25 Jan 2008 22:02:26 -0000
@@ -36,6 +36,7 @@
 #include <com/sun/star/xml/dom/events/XDocumentEvent.hpp>
 #include "characterdata.hxx"
 #include "../events/mutationevent.hxx"
+#include <string.h>
 
 namespace DOM
 {
Index: source/dom/document.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/document.cxx,v
retrieving revision 1.12
diff -u -u -r1.12 document.cxx
--- unoxml/source/dom/document.cxx	6 Dec 2007 11:17:37 -0000	1.12
+++ unoxml/source/dom/document.cxx	25 Jan 2008 22:02:26 -0000
@@ -54,6 +54,8 @@
 #include "../events/uievent.hxx"
 #include "../events/mouseevent.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
Index: source/dom/documenttype.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/documenttype.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 documenttype.cxx
--- unoxml/source/dom/documenttype.cxx	20 Jun 2006 00:45:08 -0000	1.4
+++ unoxml/source/dom/documenttype.cxx	25 Jan 2008 22:02:26 -0000
@@ -37,6 +37,7 @@
 #include "entitiesmap.hxx"
 #include "notationsmap.hxx"
 
+#include <string.h>
 
 namespace DOM
 {
Index: source/dom/element.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/element.cxx,v
retrieving revision 1.11
diff -u -u -r1.11 element.cxx
--- unoxml/source/dom/element.cxx	6 Dec 2007 10:59:52 -0000	1.11
+++ unoxml/source/dom/element.cxx	25 Jan 2008 22:02:26 -0000
@@ -40,6 +40,8 @@
 #include "attributesmap.hxx"
 #include "../events/mutationevent.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
Index: source/dom/elementlist.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/elementlist.cxx,v
retrieving revision 1.5
diff -u -u -r1.5 elementlist.cxx
--- unoxml/source/dom/elementlist.cxx	20 Jun 2006 00:46:18 -0000	1.5
+++ unoxml/source/dom/elementlist.cxx	25 Jan 2008 22:02:26 -0000
@@ -35,6 +35,8 @@
 
 #include "elementlist.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
Index: source/dom/entity.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/entity.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 entity.cxx
--- unoxml/source/dom/entity.cxx	20 Jun 2006 00:46:49 -0000	1.4
+++ unoxml/source/dom/entity.cxx	25 Jan 2008 22:02:26 -0000
@@ -35,6 +35,8 @@
 
 #include "entity.hxx"
 
+#include <string.h>
+
 namespace DOM
 {
 
Index: source/dom/entityreference.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/entityreference.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 entityreference.cxx
--- unoxml/source/dom/entityreference.cxx	20 Jun 2006 00:47:11 -0000	1.4
+++ unoxml/source/dom/entityreference.cxx	25 Jan 2008 22:02:26 -0000
@@ -34,6 +34,8 @@
  ************************************************************************/
 
 #include "entityreference.hxx"
+#include <string.h>
+
 namespace DOM
 {
   CEntityReference::CEntityReference(const xmlNodePtr /*aNodePtr*/)
Index: source/dom/node.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/node.cxx,v
retrieving revision 1.13
diff -u -u -r1.13 node.cxx
--- unoxml/source/dom/node.cxx	6 Dec 2007 11:18:20 -0000	1.13
+++ unoxml/source/dom/node.cxx	25 Jan 2008 22:02:27 -0000
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include <stdio.h>
+#include <string.h>
 #include "node.hxx"
 #include "element.hxx"
 #include "text.hxx"
Index: source/dom/notation.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/notation.cxx,v
retrieving revision 1.3
diff -u -u -r1.3 notation.cxx
--- unoxml/source/dom/notation.cxx	9 Sep 2005 10:03:53 -0000	1.3
+++ unoxml/source/dom/notation.cxx	25 Jan 2008 22:02:27 -0000
@@ -34,6 +34,8 @@
  ************************************************************************/
 
 #include "notation.hxx"
+#include <string.h>
+
 namespace DOM
 {
     CNotation::CNotation(const xmlNotationPtr aNotationPtr)
Index: source/dom/processinginstruction.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/dom/processinginstruction.cxx,v
retrieving revision 1.7
diff -u -u -r1.7 processinginstruction.cxx
--- unoxml/source/dom/processinginstruction.cxx	6 Dec 2007 11:01:25 -0000	1.7
+++ unoxml/source/dom/processinginstruction.cxx	25 Jan 2008 22:02:27 -0000
@@ -34,6 +34,7 @@
  ************************************************************************/
 
 #include "processinginstruction.hxx"
+#include <string.h>
 
 namespace DOM
 {
Index: source/fastsax/fastparser.cxx
===================================================================
RCS file: /cvs/framework/unoxml/source/fastsax/fastparser.cxx,v
retrieving revision 1.2
diff -u -u -r1.2 fastparser.cxx
--- unoxml/source/fastsax/fastparser.cxx	5 Jun 2007 13:52:30 -0000	1.2
+++ unoxml/source/fastsax/fastparser.cxx	25 Jan 2008 22:02:27 -0000
@@ -66,6 +66,8 @@
 #include "expat/xmlparse.h"
 #endif
 
+#include <string.h>
+
 using ::rtl::OUString;
 using ::rtl::OString;
 using namespace ::std;
Index: DocumentCollector.hxx
===================================================================
RCS file: /cvs/sw/writerperfect/source/filter/DocumentCollector.hxx,v
retrieving revision 1.3
diff -u -u -r1.3 DocumentCollector.hxx
--- writerperfect/source/filter/DocumentCollector.hxx	27 Sep 2007 12:54:46 -0000	1.3
+++ writerperfect/source/filter/DocumentCollector.hxx	25 Jan 2008 23:19:42 -0000
@@ -42,6 +43,7 @@
 #include <vector>
 #include <map>
 #include <stack>
+#include <string.h>
 
 class DocumentElement;
 class DocumentHandler;
Index: TableStyle.cxx
===================================================================
RCS file: /cvs/sw/writerperfect/source/filter/TableStyle.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 TableStyle.cxx
--- writerperfect/source/filter/TableStyle.cxx	23 Jan 2007 12:15:53 -0000	1.4
+++ writerperfect/source/filter/TableStyle.cxx	25 Jan 2008 23:19:42 -0000
@@ -27,6 +27,7 @@
  * Corel Corporation or Corel Corporation Limited."
  */
 #include <math.h>
+#include <string.h>
 #include "FilterInternal.hxx"
 #include "TableStyle.hxx"
 #include "DocumentElement.hxx"
Index: TextRunStyle.cxx
===================================================================
RCS file: /cvs/sw/writerperfect/source/filter/TextRunStyle.cxx,v
retrieving revision 1.4
diff -u -u -r1.4 TextRunStyle.cxx
--- writerperfect/source/filter/TextRunStyle.cxx	23 Jan 2007 12:16:30 -0000	1.4
+++ writerperfect/source/filter/TextRunStyle.cxx	25 Jan 2008 23:19:42 -0000
@@ -36,6 +36,8 @@
 #include <minmax.h>
 #endif
 
+#include <string.h>
+
 ParagraphStyle::ParagraphStyle(WPXPropertyList *pPropList, const WPXPropertyListVector &xTabStops, const WPXString &sName) :
 	mpPropList(pPropList),
 	mxTabStops(xTabStops),
Index: xsecparser.cxx
===================================================================
RCS file: /cvs/xml/xmlsecurity/source/helper/xsecparser.cxx,v
retrieving revision 1.6
diff -u -u -r1.6 xsecparser.cxx
--- xmlsecurity/source/helper/xsecparser.cxx	16 Sep 2006 14:41:05 -0000	1.6
+++ xmlsecurity/source/helper/xsecparser.cxx	26 Jan 2008 08:57:19 -0000
@@ -43,6 +43,8 @@
 #endif
 #include "cppuhelper/exc_hlp.hxx"
 
+#include <string.h>
+
 namespace cssu = com::sun::star::uno;
 namespace cssxs = com::sun::star::xml::sax;
 
