--- sc/inc/miscuno.hxx.old	2005-03-23 13:44:54.000000000 +0100
+++ sc/inc/miscuno.hxx	2005-08-02 19:38:06.000000000 +0200
@@ -337,11 +337,11 @@
 												com::sun::star::beans::XPropertySet>& xProp,
 											const ::rtl::OUString& rName, long nDefault );
 
-	static sal_Bool			GetBoolFromAny( const com::sun::star::uno::Any& aAny );
+	SC_DLLPUBLIC static sal_Bool			GetBoolFromAny( const com::sun::star::uno::Any& aAny );
 	static sal_Int16		GetInt16FromAny( const com::sun::star::uno::Any& aAny );
 	static sal_Int32		GetInt32FromAny( const com::sun::star::uno::Any& aAny );
 	static sal_Int32		GetEnumFromAny( const com::sun::star::uno::Any& aAny );
-	static void				SetBoolInAny( com::sun::star::uno::Any& rAny, sal_Bool bValue );
+	SC_DLLPUBLIC static void				SetBoolInAny( com::sun::star::uno::Any& rAny, sal_Bool bValue );
 };
 
 
--- sc/inc/optutil.hxx.old	2000-11-02 20:08:15.000000000 +0100
+++ sc/inc/optutil.hxx	2005-08-02 19:32:54.000000000 +0200
@@ -70,6 +70,10 @@
 #include <tools/link.hxx>
 #endif
 
+#ifndef INCLUDED_SCDLLAPI_H
+#include "scdllapi.h"
+#endif
+
 
 class ScOptionsUtil
 {
@@ -80,7 +84,7 @@
 
 //	ConfigItem for classes that use items from several sub trees
 
-class ScLinkConfigItem : public utl::ConfigItem
+class SC_DLLPUBLIC ScLinkConfigItem : public utl::ConfigItem
 {
 	Link	aCommitLink;
 
