--- config_office/configure.in.ark	2005-06-02 13:58:53.000000000 +0200
+++ config_office/configure.in	2005-06-02 14:02:02.000000000 +0200
@@ -261,6 +261,8 @@
                           or XML filters. Use --with-java=gij to enable gij/gcj
                           as java/javac replacements
 ], if test "$withval" = "yes"; then WITH_JAVA=java; else WITH_JAVA=$withval; fi, WITH_JAVA=java)
+AC_ARG_ENABLE(gcj-dbtool,
+[  --enable-gcj-dbtool    Use gcj-dbtool to cache te used jars to speed up using them (gcj-only)],,)
 AC_ARG_WITH(ant-home,
 [  --with-ant-home         If you have installed Jakarta Ant on your system,
                           please supply the path here.
@@ -1898,18 +1900,24 @@
 dnl Check for optional gcj-dbtool
 dnl ===================================================================
 if test "$JDK" = "gcj"; then 
-    javacache=`echo $WITH_JAVA | $SED -e "s/gij/gcj-dbtool/g"`
-    if test -z "$with_jdk_home"; then
-        AC_PATH_PROG(JAVACACHE, $javacache)
-    else
-        _javac_path="$with_jdk_home/bin/$javacache"
-        dnl Check if there is a gcj-dbtool at all.
-        if test -x "$_javac_path"; then
-            JAVACACHE=$_javac_path
+    AC_MSG_CHECKING([whether to cache jars])
+    if test -n "$enable_gcj_dbtool" && test "$enable_gcj_dbtool" != "no"; then
+        AC_MSG_RESULT([yes])
+        javacache=`echo $WITH_JAVA | $SED -e "s/gij/gcj-dbtool/g"`
+        if test -z "$with_jdk_home"; then
+            AC_PATH_PROG(JAVACACHE, $javacache)
+        else
+            _javac_path="$with_jdk_home/bin/$javacache"
+            dnl Check if there is a gcj-dbtool at all.
+            if test -x "$_javac_path"; then
+                JAVACACHE=$_javac_path
+            fi
         fi
-    fi
-    if test -z "$JAVACACHE"; then
-      AC_MSG_WARN([$javacache not found set with_jdk_home])
+        if test -z "$JAVACACHE"; then
+            AC_MSG_WARN([$javacache not found set with_jdk_home])
+        fi
+    else
+        AC_MSG_RESULT([no])
     fi
 fi
 AC_SUBST(JAVACACHE)
--- solenv/bin/deliver.pl.ark	2005-06-02 14:02:08.000000000 +0200
+++ solenv/bin/deliver.pl	2005-06-02 14:03:08.000000000 +0200
@@ -782,7 +782,8 @@
         # hard link if possible
         if( link($from, $to) ){
             print "LINK: $from -> $to\n";
-            if ($ENV{JDK} eq 'gcj' && is_jar($from)) {
+            if (defined $ENV{JDK} && $ENV{JDK} eq 'gcj' &&
+                defined $ENV{JAVACACHE} && $ENV{JAVACACHE} ne '' && is_jar($from)) {
                 cachejar($to);
             }
             return 1;
@@ -818,7 +818,7 @@
         fix_file_permissions($$from_stat_ref[2], $temp_file);
         $rc = rename($temp_file, $to);
         if ( $rc ) {
-            if (defined($ENV{JDK}) && $ENV{JDK} eq 'gcj' && is_jar($from)) {
+            if (defined($ENV{JDK}) && $ENV{JDK} eq 'gcj' && defined($ENV{JAVACACHE}) && $ENV{CACHEJAR} ne '' && is_jar($from)) {
                 cachejar($to);
             }
             # handle special packaging of *.dylib files for Mac OS X
