--- solenv/inc/settings.mk.orig	2006-01-27 17:34:07.000000000 +0100
+++ solenv/inc/settings.mk	2006-01-27 17:35:28.000000000 +0100
@@ -186,12 +186,11 @@
 .ENDIF
 
 #required arguments
-.IF "$(JAVACACHE)" != ""
-JAVAC=$(JAVACOMPILER) --encoding=ISO-8859-15 -O2 -fno-assert -Wno-deprecated -C
-JAVAI=$(JAVAINTERPRETER) -Dgnu.gcj.precompiled.db.path=$(GCJ_DATABASE)
-.ELSE
 JAVAC=$(JAVACOMPILER)
 JAVAI=$(JAVAINTERPRETER)
+.IF "$(JDK)" == "gcj"
+JAVAC+=--encoding=UTF-8 -O2 -fno-assert -Wno-deprecated -C
+JAVAI+=-Dgnu.gcj.precompiled.db.path=$(GCJ_DATABASE)
 .ENDIF
 
 #classpath and response
