--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx.old	2006-03-13 12:11:30.359949000 +0000
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	2006-03-13 12:13:00.779949000 +0000
@@ -37,6 +37,11 @@
 #include "osl/thread.h"
 #include "gnujre.hxx"
 #include "util.hxx"
+extern "C"
+{
+#include <sys/utsname.h>
+#include <string.h>
+}
 
 using namespace rtl;
 using namespace std;
@@ -53,10 +58,12 @@
 char const* const* GnuInfo::getJavaExePaths(int * size)
 {
     static char const * ar[] = {
-        "gij",
-        "bin/gij",
+	"gij-4.1",
+	"bin/gij-4.1"
 	"gij-4.0",
 	"bin/gij-4.0"
+        "gij",
+        "bin/gij",
     };
     *size = sizeof (ar) / sizeof (char*);
     return ar;
@@ -72,8 +79,27 @@
         , "/lib/libgcj.so.4"
 #endif
     };
+#if defined(__linux__) && defined (__i386__)
+    struct utsname uts;
+    static char const* ar64[]= {
+          "/lib32/libgcj.so.7"
+          "/lib32/libgcj.so.6"
+    };
+    uname(&uts);
+    if (!strcmp("x86_64", uts.machine))
+    {
+        *size = sizeof(ar64) / sizeof (char*);
+	return ar64;
+    }
+    else
+    {
+        *size = sizeof(ar) / sizeof (char*);
+	return ar;
+    }
+#else
     *size = sizeof(ar) / sizeof (char*);
     return ar;
+#endif
 }
 
 bool GnuInfo::initialize(vector<pair<OUString, OUString> > props)
