--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx~	2005-08-10 20:29:10.119466648 +0000
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	2005-08-10 20:43:55.646846024 +0000
@@ -63,6 +63,11 @@
 #include "osl/thread.h"
 #include "gnujre.hxx"
 #include "util.hxx"
+extern "C"
+{
+#include <sys/utsname.h>
+#include <string.h>
+}
 
 using namespace rtl;
 using namespace std;
@@ -97,8 +102,26 @@
         , "/lib/libgcj.so.4"
 #endif
     };
+#if defined(__linux__) && defined (__i386__)
+    struct utsname uts;
+    static char const* ar64[]= {
+          "/lib32/libgcj.so.7"
+    };
+    uname(&uts);
+    if (!strcmp("x86_64", uts.machine))
+    {
+        *size = sizeof(ar64) / sizeof (char*);
+	return ar64;
+    }
+    else
+    {
+        *size = sizeof(ar) / sizeof (char*);
+	return ar;
+    }
+#else
     *size = sizeof(ar) / sizeof (char*);
     return ar;
+#endif
 }
 
 bool GnuInfo::initialize(vector<pair<OUString, OUString> > props)
