Index: config_office/set_soenv.in
===================================================================
--- config_office/set_soenv.in.orig	2006-01-18 21:23:14.000000000 +0100
+++ config_office/set_soenv.in	2006-01-18 21:34:37.000000000 +0100
@@ -78,7 +78,7 @@ my ( $oldPATH, $SRC_ROOT, $SO_HOME, $JAV
      $UPD, $SOLARUPD, $WORK_STAMP, $TF_ONE51, $TF_UCB, 
      $URD_ONLY, $SOLARROOT, $SOLARSRC, $DEVROOT, $SOLARVER, $SOLARVERSION, $SOLARENV, 
      $STAR_INIROOT, $STAR_INIROOTOLD, $STAR_STANDLST, $STAR_SSCOMMON, $STAR_SSOLARINI, 
-     $STAR_REGISTRY, $STAR_RESOURCEPATH, $DMAKEROOT, $CLASSPATH, $XCLASSPATH, $COMPATH, $GCJ_DATABASE,
+     $STAR_REGISTRY, $STAR_RESOURCEPATH, $DMAKEROOT, $CLASSPATH, $XCLASSPATH, $COMPATH, $GCJ_DATABASE, $GCJ_CACHEDIR,
      $MSPDB_PATH, $MIDL_PATH, $CSC_PATH, $NMAKE_PATH, 
      $LD_LIBRARY_PATH, $PATH, $SOLARDEF, $SOLAREXTRAINC, $SOLAREXTRALIB, $SOLARLIB, 
      $SOLARINC, $LOCALINI, $PATHEXTRA, $FRAMEWORKSHOME, $COMEX, $MULTITHREAD_OBJ, $PERL, 
@@ -906,7 +906,9 @@ $CLASSPATH            = '$JAVA_HOME'.$ds
 # see above for why the change
 $XCLASSPATH           = '$JAVA_HOME'.$ds.'jre'.$LIB.$ds."rt.jar".$ps.'.';
 # Location of gcj cache
-$GCJ_DATABASE = '$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH'.$LIB.$ds."openoffice.org.gcjdb";
+$GCJ_CACHEDIR = '$SOLARVER'.$ds.'$UPD'.$ds.'$INPATH'.$LIB.$ds;
+# Location of gcj database
+$GCJ_DATABASE = $GCJ_CACHEDIR."openoffice.org.gcjdb";
 
 # Paths to run time shared libraries.
 if ($platform =~ m/solaris/) 
@@ -1669,6 +1671,7 @@ if ( $JDK ne "gcj" ) {
    ToFile( "XCLASSPATH",        $XCLASSPATH,        "e" );
 }
 else {
+   ToFile( "GCJ_CACHEDIR",     $GCJ_CACHEDIR,       "e" );
    ToFile( "GCJ_DATABASE",     $GCJ_DATABASE,       "e" );
    ToFile( "JAVACACHE",        '@JAVACACHE@',       "e" );
    ToFile( "AWTLIB",           '@AWTLIB@',          "e" );
Index: helpcontent2/util/target.pmk
===================================================================
--- helpcontent2/util/target.pmk.orig	2006-01-05 15:44:39.000000000 +0100
+++ helpcontent2/util/target.pmk	2006-01-18 21:34:37.000000000 +0100
@@ -50,5 +50,8 @@ JAVA_VM_HEAP_SIZE = -Xms256m -Xmx256m
 
 $(LINKALLTARGETS) : $(foreach,i,$(LINKLINKFILES) $(COMMONMISC)$/$$(@:b:s/_/./:e:s/.//)/$i) $(subst,LANGUAGE,$$(@:b:s/_/./:e:s/.//) $(LINKADDEDDEPS)) $(COMMONMISC)$/xhp_changed.flag
 	@+-$(MKDIR) -p $(@:d)
+.IF "$(JDK)" != "gcj"
 	$(JAVAI) $(JAVA_VM_HEAP_SIZE) $(JAVA_LIBRARY_PATH) -Djavax.xml.parsers.SAXParserFactory=org.apache.xerces.jaxp.SAXParserFactoryImpl -Djavax.xml.parsers.DocumentBuilderFactory=org.apache.xerces.jaxp.DocumentBuilderFactoryImpl -cp $(my_cp) com.sun.star.help.HelpLinker @$(mktmp -mod $(LINKNAME) -hid $(PRJ)$/helpers/hid.lst -src $(COMMONMISC) -sty $(PRJ)$/source$/auxiliary$/embed.xsl -idx $(PRJ)$/source$/auxiliary$/index.xsl -lang {$(subst,$(LINKNAME)_, $(@:b))} $(subst,LANGUAGE,{$(subst,$(LINKNAME)_, $(@:b))} $(LINKADDEDFILES)) $(foreach,i,$(LINKLINKFILES) $(COMMONMISC)$/{$(subst,$(LINKNAME)_, $(@:b))}/$i) -o $@)
-
+.ELSE
+	HelpLinker @$(mktmp -mod $(LINKNAME) -hid $(PRJ)$/helpers/hid.lst -src $(COMMONMISC) -sty $(PRJ)$/source$/auxiliary$/embed.xsl -idx $(PRJ)$/source$/auxiliary$/index.xsl -lang {$(subst,$(LINKNAME)_, $(@:b))} $(subst,LANGUAGE,{$(subst,$(LINKNAME)_, $(@:b))} $(LINKADDEDFILES)) $(foreach,i,$(LINKLINKFILES) $(COMMONMISC)$/{$(subst,$(LINKNAME)_, $(@:b))}/$i) -o $@)
+.ENDIF
Index: solenv/bin/deliver.pl
===================================================================
--- solenv/bin/deliver.pl.orig	2006-01-13 17:36:04.000000000 +0100
+++ solenv/bin/deliver.pl	2006-01-18 21:35:43.000000000 +0100
@@ -746,8 +746,10 @@ sub strip_target {
 
 sub cachejar {
     if (defined($ENV{JAVACACHE}) && $ENV{JAVACACHE} ne '') {
+        my $todir = $ENV{'GCJ_CACHEDIR'};
         my $file = shift;
-        my $to = $file.".so";
+        $lib = basename($file);
+        my $to = $todir."lib".$lib.".so";
         my $JAVALINKER = "$ENV{JAVACOMPILER} -shared -fPIC -Wl,-Bsymbolic -O2 -findirect-dispatch -fjni -o";
         print "CACHEJAR: $file -> $to with $ENV{GCJ_DATABASE}\n";
         print "Caching 1/2: $JAVALINKER $to $file\n";
Index: xmlhelp/prj/d.lst
===================================================================
--- xmlhelp/prj/d.lst.orig	2006-01-16 13:34:13.000000000 +0100
+++ xmlhelp/prj/d.lst	2006-01-18 21:34:37.000000000 +0100
@@ -1,4 +1,5 @@
 ..\%__SRC%\bin\*.dll %_DEST%\bin%_EXT%\*.*
+..\%__SRC%\bin\HelpLinker %_DEST%\bin%_EXT%\HelpLinker
 ..\%__SRC%\lib\lib*.so %_DEST%\lib%_EXT%
 ..\%__SRC%\lib\*.dylib %_DEST%\lib%_EXT%\*.*
 ..\%__SRC%\class\*.jar %_DEST%\bin%_EXT%\*.*
Index: xmlhelp/source/com/sun/star/help/makefile.mk
===================================================================
--- xmlhelp/source/com/sun/star/help/makefile.mk.orig	2006-01-18 21:26:53.000000000 +0100
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2006-01-18 21:34:37.000000000 +0100
@@ -91,10 +91,27 @@ JAVACLASSFILES += \
 	$(CLASSDIR)$/$(PACKAGE)$/FileURLStreamHandler.class
 .ENDIF
 
+.IF "$(JDK)"=="gcj"
+
+.IF "$(SYSTEM_DB)"=="YES"
+$(BIN)$/HelpLinker: $(LB)$/libdb.jar.a
+.ELSE
+$(BIN)$/HelpLinker:
+.ENDIF
+	$(JAVACOMPILER) --main=com.sun.star.help.HelpLinker -o $(BIN)$/HelpLinker \
+		-O2 -Wl,-Bsymbolic -findirect-dispatch -fjni -fno-assert -Wno-deprecated \
+		$(JAVACPS) $(CLASSPATH) $(JAVAFILES) $(SOLARLIB) -L$(LB) \
+		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar
+
+$(LB)$/libdb.jar.a:
+	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(DB_JAR) -o $(LB)$/libdb.jar.a
+
+.ELSE
 JARCLASSDIRS	= com
 JARTARGET		= $(PRJNAME).jar
 JARCOMPRESS 	= TRUE
 CUSTOMMANIFESTFILE = manifest
+.ENDIF
 
 
 # --- Targets ------------------------------------------------------
