--- scp2/source/ooo/file_ooo.scp.old	2006-05-22 15:23:18.000000000 +0200
+++ scp2/source/ooo/file_ooo.scp	2006-05-22 15:44:45.000000000 +0200
@@ -286,13 +286,6 @@
 
 #endif
 
-File gid_File_Rdb_Gengal
-    TXT_FILE_BODY;
-    Dir = gid_Dir_Program;
-    Styles = (PACKED, OVERWRITE);
-    Name = "gengal.rdb";
-End
-
 #if !defined(WITHOUT_MOZILLA) && defined(UNX)
 
 File gid_File_Bin_Pluginapp
--- svx/source/gengal/gengal.cxx.old	2006-05-22 15:23:18.000000000 +0200
+++ svx/source/gengal/gengal.cxx	2006-05-22 15:32:31.000000000 +0200
@@ -2,6 +2,11 @@
  *
  *  OpenOffice.org - a multi-platform office productivity suite
  *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to
  *  the terms of GNU Lesser General Public License Version 2.1.
@@ -241,21 +246,6 @@
 
 void GalApp::Main()
 {
-#ifdef TOO_SIMPLE_BY_HALF
-    ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory >
-		xMSF = cppu::createRegistryServiceFactory(
-				rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "gengal.rdb" ) ), sal_True );
-	::comphelper::setProcessServiceFactory( xMSF );
-
-	// Without this no file access works ...
-	ucb::ContentProviderDataList aData;
-	ucb::ContentProviderData aFileProvider(
-			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "com.sun.star.ucb.FileContentProvider" ) ),
-			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "file" ) ),
-			rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "" ) ) );
-	aData.push_back( aFileProvider );
-	ucb::ContentBroker::initialize( xMSF, aData );
-#else
 	Reference<XComponentContext> xComponentContext = ::cppu::defaultBootstrap_InitialComponentContext();
 	Reference<XMultiServiceFactory> xMSF(xComponentContext->getServiceManager(), UNO_QUERY);
 	if( !xMSF.is() )
@@ -273,10 +263,8 @@
 	aArgs[4] <<= rtl::OUString::createFromAscii("PORTAL");
 	aArgs[5] <<= aEmpty;
 
-    if (! ::ucb::ContentBroker::initialize( xMSF, aArgs ) )
-			fprintf( stderr, "Failed to init content broker\n" );
-#endif
-
+	if (! ::ucb::ContentBroker::initialize( xMSF, aArgs ) )
+		fprintf( stderr, "Failed to init content broker\n" );
 
 	bool bHelp = false;
 	rtl::OUString aPath, aDestDir;
--- svx/source/gengal/gengal.sh.old	2006-05-22 15:23:18.000000000 +0200
+++ svx/source/gengal/gengal.sh	2006-05-22 15:35:43.000000000 +0200
@@ -4,6 +4,14 @@
 #   The Contents of this file are made available subject to
 #   the terms of GNU Lesser General Public License Version 2.1.
 #
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
 #
 #     GNU Lesser General Public License Version 2.1
 #     =============================================
@@ -62,64 +70,34 @@
 sd_platform=`uname -s`
 case $sd_platform in
   AIX)
-    LIBPATH="$sd_prog":$LIBPATH
+    LIBPATH=${sd_prog}${LIBPATH+:${LIBPATH}}
     export LIBPATH
     ;;
 
   Darwin)
-    DYLD_LIBRARY_PATH="$sd_prog":$DYLD_LIBRARY_PATH
+    DYLD_LIBRARY_PATH=${sd_prog}${DYLD_LIBRARY_PATH+:${DYLD_LIBRARY_PATH}}
     export DYLD_LIBRARY_PATH
     ;;
 
   HP-UX)
-    SHLIB_PATH="$sd_prog":/usr/openwin/lib:$SHLIB_PATH
+    SHLIB_PATH=${sd_prog}:/usr/openwin/lib${SHLIB_PATH+:${SHLIB_PATH}}
     export SHLIB_PATH
     ;;
 
   IRIX*)
-    LD_LIBRARYN32_PATH=:"$sd_prog":$LD_LIBRARYN32_PATH
+    LD_LIBRARYN32_PATH=${sd_prog}${LD_LIBRARYN32_PATH+:${LD_LIBRARYN32_PATH}}
     export LD_LIBRARYN32_PATH
     ;;
 
   *)
-    LD_LIBRARY_PATH="$sd_prog":$LD_LIBRARY_PATH
+    LD_LIBRARY_PATH=${sd_prog}${LD_LIBRARY_PATH+:${LD_LIBRARY_PATH}}
     export LD_LIBRARY_PATH
     ;;
 esac
 
-# extend the ld_library_path for java: javaldx checks the sofficerc for us
-if [ -x "$sd_prog/javaldx" ] ; then
-    java_ld_library_path=`"$sd_prog/javaldx"`
-    if [ "$java_ld_library_path" != "" ] ; then
-        case $sd_platform in
-            AIX)
-                LIBPATH=${java_ld_library_path}:${LIBPATH}
-                ;;
-            Darwin)
-                DYLD_LIBRARY_PATH=${java_ld_library_path}:${DYLD_LIBRARY_PATH}
-                ;;
-            HP-UX)
-                SHLIB_PATH=${java_ld_library_path}:${SHLIB_PATH}
-                ;;
-            IRIX*)
-                LD_LIBRARYN32_PATH=${java_ld_library_path}:${LD_LIBRARYN32_PATH}
-                ;;
-            *)
-                LD_LIBRARY_PATH=${java_ld_library_path}:${LD_LIBRARY_PATH}
-                ;;
-        esac
-    fi
-fi
-
 # misc. environment variables
-OPENOFFICE_MOZILLA_FIVE_HOME="$sd_inst/program"
-export OPENOFFICE_MOZILLA_FIVE_HOME
-
 unset XENVIRONMENT
 
-# uncomment line below to disable anti aliasing of fonts
-# SAL_ANTIALIAS_DISABLE=true; export SAL_ANTIALIAS_DISABLE
-
 # set path so that other apps can be started just by name
 PATH="$sd_prog":$PATH
 export PATH
--- svx/source/gengal/makefile.mk.old	2006-05-22 15:23:18.000000000 +0200
+++ svx/source/gengal/makefile.mk	2006-05-22 15:42:43.000000000 +0200
@@ -58,7 +58,9 @@
 .ELSE			# .IF "$(GUI)"=="WNT"
 APP1TARGET= $(TARGET).bin
 .ENDIF			# .IF "$(GUI)"=="WNT"
-APP1OBJS=	$(OBJFILES)
+
+APP1OBJS=   $(OBJFILES)
+
 APP1STDLIBS=$(TOOLSLIB) 		\
 			$(OSLLIB)			\
 			$(ONELIB)			\
@@ -82,35 +84,18 @@
 APP1DEF=	$(MISC)$/$(TARGET).def
 .ENDIF
 
-
 .IF "$(COM)"=="GCC"
 ADDOPTFILES=$(OBJ)$/gengal.obj
 add_cflagscxx="-frtti -fexceptions"
 .ENDIF
 
-
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
 
 ALLTAR : \
-	$(SCRIPTFILES) \
-	$(BIN)$/gengal.rdb 
+	$(SCRIPTFILES)
 
 $(SCRIPTFILES) : $$(@:f:+".sh")
 	+@tr -d "\015" < $(@:f:+".sh") > $@
 
-$(BIN)$/gengal.rdb : makefile.mk $(UNOUCRRDB)
-	rm -f $@
-	$(GNUCOPY) $(UNOUCRRDB) $@
-	+cd $(BIN) && \
-		regcomp -register -r gengal.rdb \
-			-c i18nsearch.uno$(DLLPOST) \
-			-c i18npool.uno$(DLLPOST) \
-			-c configmgr2.uno$(DLLPOST) \
-			-c servicemgr.uno$(DLLPOST) \
-			-c $(DLLPRE)fwl$(UPD)$(DLLPOSTFIX)$(DLLPOST) \
-			-c $(DLLPRE)ucpfile1$(DLLPOST) \
-			-c $(DLLPRE)fileacc$(DLLPOST) \
-			-c $(DLLPRE)ucb1$(DLLPOST)
-
