--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx.orig	2006-08-27 22:35:55.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	2006-08-27 22:36:25.000000000 +0200
@@ -53,12 +53,9 @@
 char const* const* GnuInfo::getJavaExePaths(int * size)
 {
     static char const * ar[] = {
-        "gij",
-        "bin/gij",
-	"gij-4.1",
-	"bin/gij-4.1",
-	"gij-4.2",
-	"bin/gij-4.2",
+        "java",
+        "bin/java",
+        "jre/bin/java",
     };
     *size = sizeof (ar) / sizeof (char*);
     return ar;
@@ -67,9 +64,7 @@
 char const* const* GnuInfo::getRuntimePaths(int * size)
 {
     static char const* ar[]= {
-          "/lib/" JFW_PLUGIN_ARCH "/client/libjvm.so",
-          "/gcj-4.1.1/libjvm.so",
-	  "/gcj-4.1/libjvm.so",
+	  "/../jre/lib/" JFW_PLUGIN_ARCH "/client/libjvm.so",
 	  "/libgcj.so.8",
 	  "/libgcj.so.70",
           "/libgcj.so.7",
