--- config_office/configure.in.orig	2006-04-04 11:05:16.000000000 +0200
+++ config_office/configure.in	2006-04-04 12:47:06.000000000 +0200
@@ -2186,10 +2186,20 @@
 dnl Checks for javac
 dnl ===================================================================
 if test "$SOLAR_JAVA" != ""; then
+    javacompiler="javac"
     if test "$JDK" = "gcj"; then 
-        javacompiler=`echo $WITH_JAVA | $SED -e "s/gij/gcj/g" | $SED -e "s/java/javac/g"`
-    else
-        javacompiler="javac"
+        javabincompiler="gcj"
+        if test -z "$with_jdk_home"; then
+            AC_PATH_PROG(JAVABINCOMPILER, $javabincompiler)
+        else
+            _javabinc_path="$with_jdk_home/bin/$javabincompiler"
+            if test -x "$_javabinc_path"; then
+                JAVABINCOMPILER=$_javabinc_path
+            fi
+        fi
+        if test -z "$JAVABINCOMPILER"; then
+            AC_MSG_ERROR([$javabincompiler not found set with_jdk_home])
+        fi
     fi
     if test -z "$with_jdk_home"; then
         AC_PATH_PROG(JAVACOMPILER, $javacompiler)
@@ -2294,12 +2304,8 @@
 dnl ===================================================================
 dnl Check for optional gcj-dbtool, use if compiler is gcj
 dnl ===================================================================
-if test `$JAVACOMPILER --version 2>&1 | grep -c "gcj"` -gt 0; then
-   if test `echo $WITH_JAVA | grep -c "gij"` -eq 0; then
-      javacache="gcj-dbtool"
-   else
-      javacache=`echo $WITH_JAVA | $SED -e "s/gij/gcj-dbtool/g"`
-   fi
+if test "$JDK" = "gcj"; then
+   javacache="gcj-dbtool"
    if test -z "$with_jdk_home"; then
       AC_PATH_PROG(JAVACACHE, $javacache)
    else
@@ -2318,6 +2324,7 @@
 AC_SUBST(JDK)
 AC_SUBST(JAVAINTERPRETER)
 AC_SUBST(JAVACOMPILER)
+AC_SUBST(JAVABINCOMPILER)
 AC_SUBST(JAVACACHE)
 AC_SUBST(JAVADOC)
 
@@ -4776,7 +4783,8 @@
   oldJAVA_HOME=$JAVA_HOME
   if test "$JDK" = "gcj"; then
     JAVA_HOME=; export JAVA_HOME
-    ant_cmd="$ANT -Dbuild.compiler=gcj -buildfile conftest.xml 1>&2"
+    dnl Execute external 'javac' binary/wrapper
+    ant_cmd="$ANT -Dbuild.compiler=extJavac -buildfile conftest.xml 1>&2"
   else
     ant_cmd="$ANT -buildfile conftest.xml 1>&2"
   fi
--- config_office/set_soenv.in.orig	2006-04-04 12:49:15.000000000 +0200
+++ config_office/set_soenv.in	2006-04-04 12:49:52.000000000 +0200
@@ -1681,6 +1681,7 @@
    ToFile( "JDK",             "@JDK@",           "e" );
    ToFile( "JAVAINTERPRETER", "@JAVAINTERPRETER@","e" );
    ToFile( "JAVACOMPILER",    "@JAVACOMPILER@",  "e" );
+   ToFile( "JAVABINCOMPILER", "@JAVABINCOMPILER@", "e" );
    ToFile( "JAVADOC",         "@JAVADOC@",       "e" );
 }
 if ( $platform =~ m/darwin/ )
--- hsqldb/makefile.mk.orig	2006-04-04 12:04:46.000000000 +0200
+++ hsqldb/makefile.mk	2006-04-04 12:05:23.000000000 +0200
@@ -76,7 +76,8 @@
 .IF "$(JDK)"=="gcj"
 JAVA_HOME=
 .EXPORT : JAVA_HOME
-BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dbuild.compiler=gcj -f $(ANT_BUILDFILE) jar
+# Execute external 'javac' binary/wrapper
+BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -Dbuild.compiler=extJavac -f $(ANT_BUILDFILE) jar
 .ELSE
 BUILD_ACTION=$(ANT) -Dbuild.label="build-$(RSCREVISION)" -f $(ANT_BUILDFILE) jar
 .ENDIF
--- solenv/bin/deliver.pl.orig	2006-04-04 12:26:28.000000000 +0200
+++ solenv/bin/deliver.pl	2006-04-04 12:29:01.000000000 +0200
@@ -750,7 +750,7 @@
         my $file = shift;
         $lib = basename($file);
         my $to = $todir."lib".$lib.".so";
-        my $JAVALINKER = "$ENV{JAVACOMPILER} -shared -fPIC -Wl,-Bsymbolic -O2 -findirect-dispatch -fjni -o";
+        my $JAVALINKER = "$ENV{JAVABINCOMPILER} -shared -fPIC -Wl,-Bsymbolic -O2 -findirect-dispatch -fjni -o";
         print "CACHEJAR: $file -> $to with $ENV{GCJ_DATABASE}\n";
         print "Caching 1/2: $JAVALINKER $to $file\n";
         system("$JAVALINKER $to $file");
--- solenv/inc/antsettings.mk.orig	2006-04-04 12:00:32.000000000 +0200
+++ solenv/inc/antsettings.mk	2006-04-04 12:00:41.000000000 +0200
@@ -72,7 +72,8 @@
 .IF "$(JDK)"=="gcj"
 JAVA_HOME=
 .EXPORT : JAVA_HOME
-ANT_FLAGS!:=-Dbuild.compiler=gcj -Dprj=$(PRJ) -Dprjname=$(PRJNAME) -Ddebug=$(ANT_DEBUG) \
+# Execute external 'javac' binary/wrapper
+ANT_FLAGS!:=-Dbuild.compiler=extJavac -Dprj=$(PRJ) -Dprjname=$(PRJNAME) -Ddebug=$(ANT_DEBUG) \
  -Doptimize=$(ANT_OPT) -Dtarget=$(TARGET) -Dsolar.update=on -Dout=$(OUT) -Dinpath=$(INPATH) \
  -Dproext="$(PROEXT)" -Dsolar.bin=$(SOLARBINDIR) -Dsolar.jar=$(SOLARBINDIR) \
  -Dsolar.doc=$(SOLARDOCDIR) -Dcommon.jar=$(SOLARCOMMONBINDIR) \
--- solenv/inc/settings.mk.orig	2006-04-04 11:28:09.000000000 +0200
+++ solenv/inc/settings.mk	2006-04-04 11:33:57.000000000 +0200
@@ -171,14 +171,10 @@
 JAVACOMPILER=$(DEVROOT)$/cafepro$/bin$/sj.exe
 .ENDIF
 .ELSE
-.IF "$(JDK)" == "gcj"
-JAVACOMPILER=gcj
-.ELSE
 JAVACOMPILER=javac
 .ENDIF
 .ENDIF
 .ENDIF
-.ENDIF
 
 #if javadoc is not already set
 .IF "$(JAVADOC)"==""
@@ -189,7 +185,6 @@
 JAVAC=$(JAVACOMPILER)
 JAVAI=$(JAVAINTERPRETER)
 .IF "$(JAVACACHE)" != ""
-JAVAC+=--encoding=UTF-8 -O2 -fno-assert -Wno-deprecated -C
 JAVAI+=-Dgnu.gcj.precompiled.db.path=$(GCJ_DATABASE)
 .ENDIF
 
--- xmlhelp/source/com/sun/star/help/makefile.mk.orig	2006-04-04 12:55:02.000000000 +0200
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2006-04-04 12:55:48.000000000 +0200
@@ -112,7 +112,7 @@
 .IF "$(JDK)"=="gcj"
 
 $(BIN)$/HelpLinker:
-	$(JAVACOMPILER) --main=com.sun.star.help.HelpLinker -o $(BIN)$/HelpLinker \
+	$(JAVABINCOMPILER) --main=com.sun.star.help.HelpLinker -o $(BIN)$/HelpLinker \
 		-O2 -Wl,-Bsymbolic -findirect-dispatch -fjni -fno-assert -Wno-deprecated \
 		$(JAVACPS) $(CLASSPATH) $(JAVAFILES) $(SOLARLIB) -L$(LB) \
 		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar -lxt-xmlsearch.jar
@@ -121,14 +121,14 @@
 $(BIN)$/HelpLinker: $(LB)$/libdb.jar.a
 
 $(LB)$/libdb.jar.a:
-	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(DB_JAR) -o $(LB)$/libdb.jar.a
+	$(JAVABINCOMPILER) -c -O2 -findirect-dispatch -fjni $(DB_JAR) -o $(LB)$/libdb.jar.a
 .ENDIF
 
 .IF "$(SYSTEM_XT)"=="YES"
 $(BIN)$/HelpLinker: $(LB)$/libxt.jar.a
 
 $(LB)$/libxt.jar.a:
-	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(XT_JAR) -o $(LB)$/libxt.jar.a
+	$(JAVABINCOMPILER) -c -O2 -findirect-dispatch -fjni $(XT_JAR) -o $(LB)$/libxt.jar.a
 .ENDIF
 
 .ELSE
