--- goodies/source/filter.vcl/eos2met/eos2met.cxx.ark	2006-01-25 14:19:56.000000000 +0100
+++ goodies/source/filter.vcl/eos2met/eos2met.cxx	2006-01-25 15:25:13.000000000 +0100
@@ -435,7 +435,7 @@
 
 	for (pCS=pChrSetList; pCS!=NULL; pCS=pCS->pSucc)
 	{
-		if (pCS->aName==ByteString(rFont.GetName(),gsl_getSystemTextEncoding()) && pCS->eWeight==rFont.GetWeight() )
+		if (pCS->aName==String(ByteString(rFont.GetName(),gsl_getSystemTextEncoding()), osl_getThreadTextEncoding()) && pCS->eWeight==rFont.GetWeight() )
 			return pCS->nSet;
 	}
 
