--- officecfg/registry/data/org/openoffice/VCL.xcu.bak	2005-04-21 18:20:08.000000000 +0200
+++ officecfg/registry/data/org/openoffice/VCL.xcu	2005-04-22 19:52:21.000000000 +0200
@@ -412,6 +412,40 @@
         <value>Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
      </prop>
     </node>
+    <node oor:name="gu-IN" oor:op="replace">
+     <prop oor:name="CTL_DISPLAY" oor:type="xs:string" oor:op="replace">
+	<value>Gujrati_padmaa;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:type="xs:string" oor:op="replace">
+	<value>Gujrati_padmaa;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:type="xs:string" oor:op="replace">
+	<value>Gujrati_padmaa;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:type="xs:string" oor:op="replace">
+	<value>Gujrati_padmaa;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:type="xs:string" oor:op="replace">
+	<value>Gujrati_padmaa;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
+    <node oor:name="hi-IN" oor:op="replace">
+     <prop oor:name="CTL_DISPLAY" oor:type="xs:string" oor:op="replace">
+	<value>Hindi_Gargi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_HEADING" oor:type="xs:string" oor:op="replace">
+	<value>Hindi_Gargi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_PRESENTATION" oor:type="xs:string" oor:op="replace">
+	<value>Hindi_Gargi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_SPREADSHEET" oor:type="xs:string" oor:op="replace">
+	<value>Hindi_Gargi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+     <prop oor:name="CTL_TEXT" oor:type="xs:string" oor:op="replace">
+	<value>Hindi_Gargi;Tahoma;Lucidasans;Lucida Sans;Arial Unicode MS</value>
+     </prop>
+    </node>
     <node oor:name="he" oor:op="replace">
      <prop oor:name="UI_SANS" oor:op="replace" oor:type="xs:string">
         <value>lucidagrande;nachlieli;tahoma;lucidasans;supplement;Andale Sans UI;arialunicodems;lucidasansunicode;interfaceuser;WarpSans;Geneva;MS Sans Serif;Helv;Dialog;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Helmet;Interface System;Sans Serif</value>
