Index: sd/source/ui/slidesorter/controller/SlsSelectionFunction.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/slidesorter/controller/SlsSelectionFunction.cxx,v
retrieving revision 1.22
diff -u -p -u -r1.22 SlsSelectionFunction.cxx
--- sd/source/ui/slidesorter/controller/SlsSelectionFunction.cxx	14 Jul 2005 10:16:28 -0000	1.22
+++ sd/source/ui/slidesorter/controller/SlsSelectionFunction.cxx	16 Sep 2005 20:49:12 -0000
@@ -481,6 +484,7 @@ BOOL SelectionFunction::KeyInput (const 
             break;
 
         case KEY_DELETE:
+        case KEY_BACKSPACE:
         {
             int	nSelectedPagesCount = 0;
             int nPageCount = mrController.GetModel().GetPageCount();
