Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.47
diff -u -p -u -r1.47 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	5 May 2006 11:01:02 -0000	1.47
+++ vcl/unx/gtk/window/gtkframe.cxx	19 May 2006 09:03:56 -0000
@@ -190,7 +190,7 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_KP_Equal:
             case GDK_equal:			nCode = KEY_EQUAL;		break;
             case GDK_Find:			nCode = KEY_FIND;		break;
-            case GDK_Menu:			nCode = KEY_MENU;		break;
+            case GDK_Menu:			nCode = KEY_CONTEXTMENU;break;
             case GDK_Help:			nCode = KEY_HELP;		break;
             case GDK_Undo:			nCode = KEY_UNDO;		break;
             case GDK_Redo:          nCode = KEY_REPEAT;     break;
