Index: sw/source/ui/app/app.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/app/app.src,v
retrieving revision 1.108
diff -u -r1.108 app.src
--- sw/source/ui/app/app.src	20 Sep 2004 15:15:43 -0000	1.108
+++ sw/source/ui/app/app.src	30 Nov 2004 10:02:12 -0000
@@ -268,7 +268,10 @@
 		FN_SHADOWCURSOR;\
 		FN_INSERT_CAPTION;\
 		FN_INSERT_REF_FIELD;\
-        FN_INSERT_REGION;\
+       FN_INSERT_REGION;\
+                SID_HELPINDEX;\
+		SID_HELP_SUPPORTPAGE;\
+		SID_ABOUT;\
 		SID_EXTRUSION_TOOGLE;\
 		SID_EXTRUSION_TILT_DOWN;\
 		SID_EXTRUSION_TILT_UP;\
@@ -283,7 +286,7 @@
 										\
 		};								\
 	IdCount = { 						\
-        173;                            \
+        176;                            \
 	};
 
 
Index: sfx2/source/appl/app.src
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/app.src,v
retrieving revision 1.98
diff -u -r1.98 app.src
--- sfx2/source/appl/app.src	4 Oct 2004 20:42:11 -0000	1.98
+++ sfx2/source/appl/app.src	30 Nov 2004 10:05:28 -0000
@@ -1126,9 +1126,12 @@
 		SID_EXPLORER_SEARCHORIGINAL;\
 		SID_EXPLORER_DETAILS;\
 		SID_EXPLORER_ICONS;\
-    };\
+	        SID_HELPINDEX;\
+		SID_HELP_SUPPORTPAGE;\
+		SID_ABOUT;\
+};\
 	IdCount = {\
-        40;\
+        43;\
 	};
 
 #ifndef IMAGE_STDBTN_COLOR
Index: sc/source/ui/src/toolbox.src
===================================================================
RCS file: /cvs/sc/sc/source/ui/src/toolbox.src,v
retrieving revision 1.71
diff -u -r1.71 toolbox.src
--- sc/source/ui/src/toolbox.src	17 Sep 2004 13:53:15 -0000	1.71
+++ sc/source/ui/src/toolbox.src	30 Nov 2004 10:09:52 -0000
@@ -1565,6 +1565,13 @@
 		SID_ATTR_FILL_SHADOW; /*#i25616# 10299 */ \
 		SID_ENABLE_HYPHENATION;			/* 26087	*/	\
         SID_ATTR_ZOOM; \
+        	SID_PRINTPREVIEW;\
+		SID_HELPINDEX;\
+		SID_HELP_SUPPORTPAGE;\
+		SID_ABOUT;\
+                SID_DELETE_CONTENTS;\
+                SID_SELECTALL;\
+                FID_MERGE_CENTER;\
 		SID_EXTRUSION_TOOGLE;\
 		SID_EXTRUSION_TILT_DOWN;\
 		SID_EXTRUSION_TILT_UP;\
@@ -1582,7 +1589,7 @@
         SID_ADD_PRINTAREA;\
     };                                                  \
     IdCount = {                                         \
-        179;                                            \
+        186;                                            \
     };
 
 
Index: sd/source/ui/app/app.src
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/app.src,v
retrieving revision 1.26
diff -u -r1.26 app.src
--- sd/source/ui/app/app.src	12 Oct 2004 13:08:52 -0000	1.26
+++ sd/source/ui/app/app.src	30 Nov 2004 10:11:58 -0000
@@ -391,9 +391,16 @@
 			SID_HYPHENATION;\
 			SID_THESAURUS;\
 			SID_AUTOSPELL_CHECK;\
+                SID_INSERT_FLD_DATE_FIX;\
+                SID_INSERT_FLD_TIME_FIX;\
+                SID_INSERT_FLD_AUTHOR;\
+                SID_INSERT_FLD_PAGE;\
+		SID_HELPINDEX;\
+		SID_HELP_SUPPORTPAGE;\
+		SID_ABOUT;\
 	};\
     IdCount = {\
-        283;\
+        290;\
     };
 
 
Index: svx/source/src/app.src
===================================================================
RCS file: /cvs/graphics/svx/source/src/app.src,v
retrieving revision 1.12
diff -u -r1.12 app.src
--- svx/source/src/app.src	12 Oct 2004 14:16:03 -0000	1.12
+++ svx/source/src/app.src	30 Nov 2004 10:14:55 -0000
@@ -277,6 +277,9 @@
 		SID_DIA;                    \
 		SID_HIDE_SLIDE;             \
 		SID_CHOOSE_POLYGON;         \
+		SID_HELPINDEX;		    \
+		SID_HELP_SUPPORTPAGE;	    \
+		SID_ABOUT;                  \
         SID_FM_REMOVE_FILTER_SORT;  \
         SID_FM_SORTUP;              \
         SID_FM_SORTDOWN;            \
@@ -405,7 +408,7 @@
 		SID_AVMEDIA_PLAYER; \
     };\
 	IdCount = {\
-        317;\
+        320;\
 	};
 
 #define OFFICE_ID_LIST\
