Index: sd/source/ui/view/viewshel.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/view/viewshel.cxx,v
retrieving revision 1.48
diff -u -p -u -r1.48 viewshel.cxx
--- sd/source/ui/view/viewshel.cxx	31 Jan 2005 14:56:31 -0000	1.48
+++ sd/source/ui/view/viewshel.cxx	11 Mar 2005 13:27:48 -0000
@@ -1143,9 +1143,9 @@ SfxUndoManager* ViewShell::ImpGetUndoMan
     SfxShell* pObjectBar = GetObjectBarManager().GetTopObjectBar();
     // The object bar may be temporarily missing, i.e. when one is
     // replaced by another.  Return a NULL pointer in this case.
-    if (pObjectBar != NULL)
-        return pObjectBar->GetUndoManager();
-    else
+//    if (pObjectBar != NULL)
+//        return pObjectBar->GetUndoManager();
+//    else
 	{
 		//#i39635# this state occurs e.g. initially with the left slide sorter pane
 		SfxViewFrame* pViewFrame = GetViewFrame();
