--- sfx2/source/appl/sfxhelp.cxx~	2005-09-07 19:47:17.000000000 +0200
+++ sfx2/source/appl/sfxhelp.cxx	2005-10-10 19:26:00.111394472 +0200
@@ -151,7 +151,7 @@
 
 NoHelpErrorBox::NoHelpErrorBox( Window* _pParent ) :
 
-    ErrorBox( _pParent, WB_OK, String( SfxResId( RID_STR_HLPAPPNOTSTARTED ) ) )
+    ErrorBox( _pParent, WB_OK, String( SfxResId( RID_STR_HLPAPPNOTSTARTED ) ).Append(String(DEFINE_CONST_UNICODE(".\n"))).Append(String( SfxResId( RID_STR_HLPFILENOTEXIST ))).Append(String(DEFINE_CONST_UNICODE("."))) )
 {
     // Error message: "No help available"
 }


