Index: makefile.mk
===================================================================
RCS file: /cvs/external/hsqldb/makefile.mk,v
retrieving revision 1.6
diff -u -p -r1.6 makefile.mk
--- hsqldb/makefile.mk	23 Mar 2005 14:55:56 -0000	1.6
+++ hsqldb/makefile.mk	7 Apr 2005 13:43:40 -0000
@@ -75,9 +75,9 @@ TARFILE_NAME=hsqldb_$(HSQLDB_VERSION)
       
 TARFILE_ROOTDIR=hsqldb
 
-PATCH_FILE_NAME=hsqldb_1_8_0
+#PATCH_FILE_NAME=hsqldb_1_8_0
 
-ADDITIONAL_FILES=   src$/org$/hsqldb$/Collation.java \
+#ADDITIONAL_FILES=   src$/org$/hsqldb$/Collation.java \
                     src$/org$/hsqldb$/TxManager.java \
                     src$/org$/hsqldb$/lib$/LongKeyIntValueHashMap.java \
                     src$/org$/hsqldb$/persist$/ScaledRAFileInJar.java \
