--- officecfg/registry/schema/org/openoffice/Office/Common.xcs~	2007-03-23 09:21:03.860193118 +0100
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2007-03-27 12:32:45.892690837 +0200
@@ -5302,7 +5302,12 @@
 					</enumeration>
 					<enumeration oor:value="tango">
 						<info>
-							<desc>Tango - the Gnome default theme</desc>
+							<desc>Tango</desc>
+						</info>
+					</enumeration>
+					<enumeration oor:value="tango">
+						<info>
+							<desc>Human - the Gnome default theme for Ubuntu</desc>
 						</info>
 					</enumeration>
 				</constraints>
--- solenv/inc/settings.mk~	2007-03-23 09:21:06.110227059 +0100
+++ solenv/inc/settings.mk	2007-03-27 12:38:14.107641885 +0200
@@ -998,7 +998,7 @@
 .ENDIF
 
 # additional image sets
-CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango
+CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango human
 
 # settings for mozilla idl compiler
 XPIDL=xpidl
--- svtools/inc/imgdef.hxx~	2007-03-23 09:21:00.380140622 +0100
+++ svtools/inc/imgdef.hxx	2007-03-27 12:33:14.063115782 +0200
@@ -50,7 +50,8 @@
     SFX_SYMBOLS_STYLE_HICONTRAST,
     SFX_SYMBOLS_STYLE_INDUSTRIAL,
     SFX_SYMBOLS_STYLE_CRYSTAL,
-    SFX_SYMBOLS_STYLE_TANGO
+    SFX_SYMBOLS_STYLE_TANGO,
+    SFX_SYMBOLS_STYLE_HUMAN
 };
 
 #define SFX_TOOLBOX_CHANGESYMBOLSET		0x0001
--- svtools/source/config/miscopt.cxx~	2007-03-23 09:21:03.860193118 +0100
+++ svtools/source/config/miscopt.cxx	2007-03-27 12:34:17.434071718 +0200
@@ -292,6 +292,7 @@
 		case STYLE_SYMBOLS_INDUSTRIAL: return SFX_SYMBOLS_STYLE_INDUSTRIAL;
 		case STYLE_SYMBOLS_CRYSTAL:    return SFX_SYMBOLS_STYLE_CRYSTAL;
 		case STYLE_SYMBOLS_TANGO:      return SFX_SYMBOLS_STYLE_TANGO;
+		case STYLE_SYMBOLS_HUMAN:      return SFX_SYMBOLS_STYLE_HUMAN;
 	}
 
 	return SFX_SYMBOLS_STYLE_AUTO;
@@ -307,6 +308,7 @@
 		case SFX_SYMBOLS_STYLE_INDUSTRIAL: return STYLE_SYMBOLS_INDUSTRIAL;
 		case SFX_SYMBOLS_STYLE_CRYSTAL:    return STYLE_SYMBOLS_CRYSTAL;
 		case SFX_SYMBOLS_STYLE_TANGO:      return STYLE_SYMBOLS_TANGO;
+		case SFX_SYMBOLS_STYLE_HUMAN:      return STYLE_SYMBOLS_HUMAN;
 	}
 
 	return STYLE_SYMBOLS_AUTO;
--- svx/source/dialog/optgdlg.cxx~	2007-03-23 09:21:00.710145600 +0100
+++ svx/source/dialog/optgdlg.cxx	2007-03-27 12:35:38.615296319 +0200
@@ -916,6 +916,7 @@
             case 3: eSet = SFX_SYMBOLS_STYLE_INDUSTRIAL; break;
             case 4: eSet = SFX_SYMBOLS_STYLE_CRYSTAL;    break;
             case 5: eSet = SFX_SYMBOLS_STYLE_TANGO;      break;
+            case 6: eSet = SFX_SYMBOLS_STYLE_HUMAN;      break;
             default:
                 DBG_ERROR( "OfaViewTabPage::FillItemSet(): This state of aIconStyleLB should not be possible!" );
         }
@@ -1087,6 +1088,7 @@
             case SFX_SYMBOLS_STYLE_INDUSTRIAL: nStyleLB_InitialSelection = 3; break;
             case SFX_SYMBOLS_STYLE_CRYSTAL:    nStyleLB_InitialSelection = 4; break;
             case SFX_SYMBOLS_STYLE_TANGO:      nStyleLB_InitialSelection = 5; break;
+            case SFX_SYMBOLS_STYLE_HUMAN:      nStyleLB_InitialSelection = 6; break;
             default:                           nStyleLB_InitialSelection = 0; break;
         }
     }
--- svx/source/dialog/optgdlg.src~	2007-03-23 09:21:00.380140622 +0100
+++ svx/source/dialog/optgdlg.src	2007-03-27 12:36:11.615794124 +0200
@@ -274,6 +274,7 @@
             < "Industrial" ; > ;
             < "Crystal" ; > ;
             < "Tango" ; > ;
+            < "Human" ; > ;
         };
         StringList [ en-US ] =
         {
@@ -283,6 +284,7 @@
             < "Industrial" ; > ;
             < "Crystal" ; > ;
             < "Tango" ; > ;
+            < "Human" ; > ;
         };
     };
 	CheckBox CB_SYSTEM_FONT
--- vcl/inc/settings.hxx~	2007-03-23 09:21:00.380140622 +0100
+++ vcl/inc/settings.hxx	2007-03-27 12:37:39.127114212 +0200
@@ -529,6 +529,7 @@
 #define STYLE_SYMBOLS_INDUSTRIAL	((ULONG)3)
 #define STYLE_SYMBOLS_CRYSTAL		((ULONG)4)
 #define STYLE_SYMBOLS_TANGO		((ULONG)5)
+#define STYLE_SYMBOLS_HUMAN		((ULONG)6)
 
 #define STYLE_CURSOR_NOBLINKTIME    ((ULONG)0xFFFFFFFF)
 
--- vcl/source/app/settings.cxx~	2007-03-23 09:21:00.420141225 +0100
+++ vcl/source/app/settings.cxx	2007-03-27 12:47:53.956388772 +0200
@@ -735,6 +735,7 @@
 		case STYLE_SYMBOLS_INDUSTRIAL: return ::rtl::OUString::createFromAscii( "industrial" );
 		case STYLE_SYMBOLS_CRYSTAL:    return ::rtl::OUString::createFromAscii( "crystal" );
 		case STYLE_SYMBOLS_TANGO:      return ::rtl::OUString::createFromAscii( "tango" );
+		case STYLE_SYMBOLS_HUMAN:      return ::rtl::OUString::createFromAscii( "human" );
 	}
 
 	return ::rtl::OUString::createFromAscii( "auto" );
@@ -754,6 +755,8 @@
 		return STYLE_SYMBOLS_CRYSTAL;
 	else if ( rName == ::rtl::OUString::createFromAscii( "tango" ) )
 		return STYLE_SYMBOLS_TANGO;
+	else if ( rName == ::rtl::OUString::createFromAscii( "human" ) )
+		return STYLE_SYMBOLS_HUMAN;
 
 	return STYLE_SYMBOLS_AUTO;
 }
@@ -775,7 +778,7 @@
 
 			if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
 			    rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
-				snDesktopStyle = STYLE_SYMBOLS_TANGO;
+				snDesktopStyle = STYLE_SYMBOLS_HUMAN;
 			else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
 				snDesktopStyle = STYLE_SYMBOLS_CRYSTAL;
 
