--- officecfg/registry/schema/org/openoffice/Office/Common.xcs.old	2007-07-18 05:44:03.000000000 -0500
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	2007-08-04 02:48:14.000000000 -0500
@@ -5383,7 +5383,12 @@
 					</enumeration>
 					<enumeration oor:value="tango">
 						<info>
-							<desc>Tango - the Gnome default theme</desc>
+							<desc>Tango</desc>
+						</info>
+					</enumeration>
+					<enumeration oor:value="human">
+						<info>
+							<desc>Human - the Gnome default theme for Ubuntu</desc>
 						</info>
 					</enumeration>
 				</constraints>
--- solenv/inc/settings.mk.old	2007-08-04 02:51:40.000000000 -0500
+++ solenv/inc/settings.mk	2007-08-04 02:51:55.000000000 -0500
@@ -946,7 +946,7 @@
 .ENDIF
 
 # additional image sets
-CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango
+CUSTOM_IMAGE_SETS=hicontrast industrial crystal tango human
 
 # settings for mozilla idl compiler
 XPIDL=xpidl
--- svtools/inc/imgdef.hxx.old	2007-06-04 09:47:50.000000000 -0500
+++ svtools/inc/imgdef.hxx	2007-08-04 02:53:12.000000000 -0500
@@ -50,7 +50,8 @@
     SFX_SYMBOLS_STYLE_HICONTRAST,
     SFX_SYMBOLS_STYLE_INDUSTRIAL,
     SFX_SYMBOLS_STYLE_CRYSTAL,
-    SFX_SYMBOLS_STYLE_TANGO
+    SFX_SYMBOLS_STYLE_TANGO,
+    SFX_SYMBOLS_STYLE_HUMAN
 };
 
 #define SFX_TOOLBOX_CHANGESYMBOLSET		0x0001
--- svtools/source/config/miscopt.cxx.old	2007-06-27 16:14:41.000000000 -0500
+++ svtools/source/config/miscopt.cxx	2007-08-04 02:56:14.000000000 -0500
@@ -286,6 +286,7 @@
 		case STYLE_SYMBOLS_INDUSTRIAL: return SFX_SYMBOLS_STYLE_INDUSTRIAL;
 		case STYLE_SYMBOLS_CRYSTAL:    return SFX_SYMBOLS_STYLE_CRYSTAL;
 		case STYLE_SYMBOLS_TANGO:      return SFX_SYMBOLS_STYLE_TANGO;
+		case STYLE_SYMBOLS_HUMAN:      return SFX_SYMBOLS_STYLE_HUMAN;
 	}
 
 	return SFX_SYMBOLS_STYLE_AUTO;
@@ -301,6 +302,7 @@
 		case SFX_SYMBOLS_STYLE_INDUSTRIAL: return STYLE_SYMBOLS_INDUSTRIAL;
 		case SFX_SYMBOLS_STYLE_CRYSTAL:    return STYLE_SYMBOLS_CRYSTAL;
 		case SFX_SYMBOLS_STYLE_TANGO:      return STYLE_SYMBOLS_TANGO;
+		case SFX_SYMBOLS_STYLE_HUMAN:      return STYLE_SYMBOLS_HUMAN;
 	}
 
 	return STYLE_SYMBOLS_AUTO;
--- svx/source/dialog/optgdlg.cxx.old	2007-07-11 06:03:03.000000000 -0500
+++ svx/source/dialog/optgdlg.cxx	2007-08-04 02:59:17.000000000 -0500
@@ -918,6 +918,7 @@
             case 3: eSet = SFX_SYMBOLS_STYLE_INDUSTRIAL; break;
             case 4: eSet = SFX_SYMBOLS_STYLE_CRYSTAL;    break;
             case 5: eSet = SFX_SYMBOLS_STYLE_TANGO;      break;
+            case 6: eSet = SFX_SYMBOLS_STYLE_HUMAN;      break;
             default:
                 DBG_ERROR( "OfaViewTabPage::FillItemSet(): This state of aIconStyleLB should not be possible!" );
         }
@@ -1089,6 +1090,7 @@
             case SFX_SYMBOLS_STYLE_INDUSTRIAL: nStyleLB_InitialSelection = 3; break;
             case SFX_SYMBOLS_STYLE_CRYSTAL:    nStyleLB_InitialSelection = 4; break;
             case SFX_SYMBOLS_STYLE_TANGO:      nStyleLB_InitialSelection = 5; break;
+            case SFX_SYMBOLS_STYLE_HUMAN:      nStyleLB_InitialSelection = 6; break;
             default:                           nStyleLB_InitialSelection = 0; break;
         }
     }
--- svx/source/dialog/optgdlg.src.old	2007-06-27 12:27:39.000000000 -0500
+++ svx/source/dialog/optgdlg.src	2007-08-04 03:18:01.000000000 -0500
@@ -243,6 +243,7 @@
             < "Industrial" ; > ;
             < "Crystal" ; > ;
             < "Tango" ; > ;
+            < "Human" ; > ;
         };
     };
 	CheckBox CB_SYSTEM_FONT
--- vcl/inc/vcl/settings.hxx.old	2007-06-04 09:57:34.000000000 -0500
+++ vcl/inc/vcl/settings.hxx	2007-08-04 03:06:54.000000000 -0500
@@ -529,7 +529,8 @@
 #define STYLE_SYMBOLS_INDUSTRIAL	((ULONG)3)
 #define STYLE_SYMBOLS_CRYSTAL		((ULONG)4)
 #define STYLE_SYMBOLS_TANGO		((ULONG)5)
-#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)5)
+#define STYLE_SYMBOLS_HUMAN		((ULONG)6)
+#define STYLE_SYMBOLS_THEMES_MAX	((ULONG)6)
 
 #define STYLE_CURSOR_NOBLINKTIME    ((ULONG)0xFFFFFFFF)
 
--- vcl/source/app/settings.cxx.old	2007-08-04 03:38:19.000000000 -0500
+++ vcl/source/app/settings.cxx	2007-08-04 03:43:37.000000000 -0500
@@ -84,6 +84,11 @@
 #include <tools/postwin.h>
 #endif
 
+extern "C" {
+#include <unistd.h>
+#include <string.h>
+}
+
 using namespace rtl;
 
 // =======================================================================
@@ -728,6 +733,7 @@
 		case STYLE_SYMBOLS_INDUSTRIAL: return ::rtl::OUString::createFromAscii( "industrial" );
 		case STYLE_SYMBOLS_CRYSTAL:    return ::rtl::OUString::createFromAscii( "crystal" );
 		case STYLE_SYMBOLS_TANGO:      return ::rtl::OUString::createFromAscii( "tango" );
+		case STYLE_SYMBOLS_HUMAN:      return ::rtl::OUString::createFromAscii( "human" );
 	}
 
 	return ::rtl::OUString::createFromAscii( "auto" );
@@ -747,6 +753,8 @@
 		return STYLE_SYMBOLS_CRYSTAL;
 	else if ( rName == ::rtl::OUString::createFromAscii( "tango" ) )
 		return STYLE_SYMBOLS_TANGO;
+	else if ( rName == ::rtl::OUString::createFromAscii( "human" ) )
+		return STYLE_SYMBOLS_HUMAN;
 
 	return STYLE_SYMBOLS_AUTO;
 }
@@ -773,11 +781,34 @@
 
 // -----------------------------------------------------------------------
 
+static ULONG _checked_style = 99;
+
 ULONG StyleSettings::GetCurrentSymbolsStyle() const
 {
 	// style selected in Tools -> Options... -> OpenOffice.org -> View
 	ULONG nStyle = GetSymbolsStyle();
 
+	// if the images*.zip file doesn't exist, fall back to auto
+	if ( nStyle != STYLE_SYMBOLS_AUTO && _checked_style != nStyle )
+	{
+		char zipfile[100] = "/usr/lib/openoffice/share/config/images";
+		switch ( nStyle )
+		{
+			case STYLE_SYMBOLS_DEFAULT:    strcat (zipfile, ""); break;
+			case STYLE_SYMBOLS_HICONTRAST: strcat (zipfile, "_hicontrast"); break;
+			case STYLE_SYMBOLS_INDUSTRIAL: strcat (zipfile, "_industrial"); break;
+			case STYLE_SYMBOLS_CRYSTAL:    strcat (zipfile, "_crystal"); break;
+			case STYLE_SYMBOLS_TANGO:      strcat (zipfile, "_tango"); break;
+			case STYLE_SYMBOLS_HUMAN:      strcat (zipfile, "_human"); break;
+		}
+		strcat (zipfile, ".zip");
+		if ( access( zipfile, R_OK ) )
+		{
+			_checked_style = nStyle;
+			nStyle = STYLE_SYMBOLS_AUTO;
+		}
+	}
+
 	if ( nStyle == STYLE_SYMBOLS_AUTO )
 	{
 		// the preferred style can be read from the desktop setting by the desktop native widgets modules
@@ -788,7 +819,7 @@
 		
 			// use a hardcoded desktop-specific fallback if no preferred style has been detected
 			static bool sbFallbackDesktopChecked = false;
-			static ULONG snFallbackDesktopStyle = STYLE_SYMBOLS_DEFAULT;
+			static ULONG snFallbackDesktopStyle = STYLE_SYMBOLS_HUMAN;
 
 			if ( !sbFallbackDesktopChecked )
 			{
@@ -796,7 +827,7 @@
 
 				if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "gnome" ) ||
 				    rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "windows" ) )
-					snFallbackDesktopStyle = STYLE_SYMBOLS_TANGO;
+					snFallbackDesktopStyle = STYLE_SYMBOLS_HUMAN;
 				else if( rDesktopEnvironment.equalsIgnoreAsciiCaseAscii( "kde" ) )
 					snFallbackDesktopStyle = STYLE_SYMBOLS_CRYSTAL;
 
