--- hunspell/prj/build.lst.orig	2005-07-21 11:27:17.000000000 +0200
+++ hunspell/prj/build.lst	2005-08-09 16:22:37.000000000 +0200
@@ -1,3 +1,3 @@
-mb	hunspell	:	linguistic NULL
+mb	hunspell	:	linguistic libhunspell NULL
 mb	hunspell								usr1	-	all	hs_mkout NULL
 mb	hunspell\source		nmake	-	all	hs_libhunspell NULL
--- hunspell/prj/d.lst.old	2005-02-18 02:27:28.000000000 +0100
+++ hunspell/prj/d.lst	2005-08-09 17:45:44.000000000 +0200
@@ -1,5 +1 @@
-..\%__SRC%\bin\morphbase* %_DEST%\bin%_EXT%\morphbase*
-..\%__SRC%\lib\ilmorphbase* %_DEST%\lib%_EXT%\ilmorphbase*
-..\%__SRC%\lib\libmorphbase.a %_DEST%\lib%_EXT%\libmorphbase.a
-..\%__SRC%\lib\libmorphbase*.dylib %_DEST%\lib%_EXT%\libmorphbase*.dylib
-..\%__SRC%\lib\libmorphbase*.so %_DEST%\lib%_EXT%\libmorphbase*.so
+..\%__SRC%\lib\*.so %_DEST%\lib%_EXT%\*.so
--- hunspell/source/makefile.mk.orig	2005-07-21 11:53:51.000000000 +0200
+++ hunspell/source/makefile.mk	2005-08-09 17:00:24.000000000 +0200
@@ -71,12 +71,7 @@
 
 .INCLUDE : settings.mk
 
-.IF "$(OS)"!="WNT"
-HUNSPELLLIB=/usr/local/lib/libhunspell.a
-.ELSE
-HUNSPELLLIB=./libhunspell.a
-.ENDIF
-
+HUNSPELLLIB=-lhunspell
 
 # --- Files --------------------------------------------------------
 
@@ -111,6 +106,10 @@
 	com.sun.star.linguistic2.XThesaurus
 
 
+.IF "$(SYSTEM_HUNSPELL)" != "YES"
+SOLARINC += -I$(SOLARVER)/$(UPD)/$(INPATH)/inc/libhunspell
+.ENDIF
+
 CXXFLAGS += -I. -I./include  -DHAVE_CONFIG_H
 CFLAGSCXX += -I. -I./include  -DHAVE_CONFIG_H
 CFLAGSCC += -I. -I./include -DHAVE_CONFIG_H
