--- lingucomponent/source/spellcheck/spell/makefile.mk-old	2006-03-16 19:03:49.946824328 +0100
+++ lingucomponent/source/spellcheck/spell/makefile.mk	2006-03-16 19:04:44.188578328 +0100
@@ -47,7 +47,11 @@
 
 .IF "$(HUNSPELLLIB)"==""
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_HUNSPELL)" == "YES"
+HUNSPELLLIB=-lhunspell_pic
+.ELSE
 HUNSPELLLIB=-lhunspell
+.ENDIF
 .ENDIF # unx
 .IF "$(GUI)"=="WNT"
 HUNSPELLLIB=hunspell.lib
