diff --git a/i18npool/source/localedata/data/makefile.mk b/i18npool/source/localedata/data/makefile.mk
index e075310..de58df0 100644
--- i18npool/source/localedata/data/makefile.mk
+++ i18npool/source/localedata/data/makefile.mk
@@ -301,7 +301,8 @@ SHL2OBJS= \
 	$(SLO)$/localedata_es_PY.obj \
 	$(SLO)$/localedata_es_SV.obj \
 	$(SLO)$/localedata_es_UY.obj \
-	$(SLO)$/localedata_es_VE.obj
+	$(SLO)$/localedata_es_VE.obj \
+	$(SLO)$/localedata_gl_ES.obj
 
 LIB2TARGET=	$(SLB)$/$(SHL2TARGET).lib
 LIB2OBJFILES=$(SHL2OBJS)
@@ -317,7 +318,7 @@ SHL3VERSIONMAP=localedata_euro.map
 SHL3DEF=$(MISC)$/$(SHL3TARGET).def
 DEF3NAME=$(SHL3TARGET)
 
-SHL3STDLIBS=	$(LINK_LOCALEDATA_EN_LIB) $(LINK_LOCALEDATA_ES_LIB)
+SHL3STDLIBS=	$(LINK_LOCALEDATA_EN_LIB)
 SHL3DEPN=$(SHL1TARGETN) $(SHL2TARGETN)
 SHL3OBJS=	\
 	$(SLO)$/localedata_be_BY.obj	\
@@ -348,7 +349,6 @@ SHL3OBJS=	\
 	$(SLO)$/localedata_fur_IT.obj	\
 	$(SLO)$/localedata_fy_NL.obj	\
 	$(SLO)$/localedata_ga_IE.obj	\
-	$(SLO)$/localedata_gl_ES.obj	\
 	$(SLO)$/localedata_gsc_FR.obj	\
 	$(SLO)$/localedata_hr_HR.obj	\
 	$(SLO)$/localedata_is_IS.obj	\
diff --git a/i18npool/source/localedata/localedata.cxx b/i18npool/source/localedata/localedata.cxx
index 106c90e..885f6bd 100644
--- i18npool/source/localedata/localedata.cxx
+++ i18npool/source/localedata/localedata.cxx
@@ -115,6 +115,7 @@ static const struct {
     { "es_SV",  lcl_DATA_ES },
     { "es_UY",  lcl_DATA_ES },
     { "es_VE",  lcl_DATA_ES },
+    { "gl_ES",  lcl_DATA_ES },
 
     { "de_DE",  lcl_DATA_EURO },
     { "de_AT",  lcl_DATA_EURO },
@@ -133,7 +134,6 @@ static const struct {
     { "sv_SE",  lcl_DATA_EURO },
     { "sv_FI",  lcl_DATA_EURO },
     { "ca_ES",  lcl_DATA_EURO },
-    { "gl_ES",  lcl_DATA_EURO },
     { "cs_CZ",  lcl_DATA_EURO },
     { "sk_SK",  lcl_DATA_EURO },
     { "da_DK",  lcl_DATA_EURO },
