--- jvmfwk/plugins/sunmajor/pluginlib/otherjre.cxx.orig	2006-08-16 12:13:05.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/otherjre.cxx	2006-08-16 12:14:30.000000000 +0200
@@ -80,6 +80,7 @@
         "/jre/bin/classic/libjvm.so", // for IBM Java
         "/lib/" JFW_PLUGIN_ARCH "/client/libjvm.so", // for Blackdown PPC
         "/lib/" JFW_PLUGIN_ARCH "/server/libjvm.so", // for Blackdown AMD64
+        "/lib/" JFW_PLUGIN_ARCH "/j9vm/libjvm.so", // for IBM Java
         "/lib/" JFW_PLUGIN_ARCH "/classic/libjvm.so", // for Blackdown PPC
 	"/lib/" JFW_PLUGIN_ARCH "/jrockit/libjvm.so" // for Java of BEA Systems
 #endif
@@ -105,6 +106,7 @@
         "/jre/bin/classic",
         "/lib/" JFW_PLUGIN_ARCH "/client",
         "/lib/" JFW_PLUGIN_ARCH "/server",
+        "/lib/" JFW_PLUGIN_ARCH "/j9vm",
         "/lib/" JFW_PLUGIN_ARCH "/classic",
         "/lib/" JFW_PLUGIN_ARCH "/jrockit",
         "/lib/" JFW_PLUGIN_ARCH "/native_threads",
