--- svx/source/dialog/optgdlg.cxx	2005-06-01 15:59:12.546822000 +0200
+++ svx/source/dialog/optgdlg.cxx	2005-06-02 15:22:45.537194972 +0200
@@ -677,8 +677,9 @@ OfaViewTabPage::OfaViewTabPage(Window* p
     aUserInterfaceFL    ( this, ResId( FL_USERINTERFACE ) ),
     aWindowSizeFT       ( this, ResId( FT_WINDOWSIZE ) ),
     aWindowSizeMF       ( this, ResId( MF_WINDOWSIZE ) ),
-    aIconSizeFT              ( this, ResId( FT_ICONSIZE ) ),
+    aIconSizeStyleFT    ( this, ResId( FT_ICONSIZESTYLE ) ),
     aIconSizeLB              ( this, ResId( LB_ICONSIZE ) ),
+    aIconStyleLB        ( this, ResId( LB_ICONSTYLE ) ),
     m_aSystemFont				(this, ResId( CB_SYSTEM_FONT ) ),
 #if defined( UNX )
 	aFontAntiAliasing	( this, ResId( CB_FONTANTIALIASING )),
@@ -824,22 +825,38 @@ BOOL OfaViewTabPage::FillItemSet( SfxIte
 	BOOL bMenuOptModified = FALSE;
 
     SvtMiscOptions aMiscOptions;
-    UINT16 nBigLB_NewSelection = aIconSizeLB.GetSelectEntryPos();
-	if( nBigLB_InitialSelection != nBigLB_NewSelection )
+    UINT16 nSizeLB_NewSelection = aIconSizeLB.GetSelectEntryPos();
+    if( nSizeLB_InitialSelection != nSizeLB_NewSelection )
 	{
 		// from now on it's modified, even if via auto setting the same size was set as now selected in the LB
         sal_Int16 eSet = SFX_SYMBOLS_SIZE_AUTO;
-        switch( nBigLB_NewSelection )
+        switch( nSizeLB_NewSelection )
 		{
             case 0: eSet = SFX_SYMBOLS_SIZE_AUTO;  break;
             case 1: eSet = SFX_SYMBOLS_SIZE_SMALL; break;
             case 2: eSet = SFX_SYMBOLS_SIZE_LARGE; break;
             default:
-				DBG_ERROR( "OfaViewTabPage::FillItemSet(): This state should not be possible!" );
+                DBG_ERROR( "OfaViewTabPage::FillItemSet(): This state of aIconSizeLB should not be possible!" );
 		}
         aMiscOptions.SetSymbolsSize( eSet );
     }
 
+    UINT16 nStyleLB_NewSelection = aIconStyleLB.GetSelectEntryPos();
+    if( nStyleLB_InitialSelection != nStyleLB_NewSelection )
+    {
+        sal_Int16 eSet = SFX_SYMBOLS_STYLE_AUTO;
+        switch( nStyleLB_NewSelection )
+        {
+            case 0: eSet = SFX_SYMBOLS_STYLE_AUTO;       break;
+            case 1: eSet = SFX_SYMBOLS_STYLE_DEFAULT;    break;
+            case 2: eSet = SFX_SYMBOLS_STYLE_INDUSTRIAL; break;
+            case 3: eSet = SFX_SYMBOLS_STYLE_CRYSTAL;    break;
+            default:
+                DBG_ERROR( "OfaViewTabPage::FillItemSet(): This state of aIconStyleLB should not be possible!" );
+        }
+        aMiscOptions.SetSymbolsStyle( eSet );
+    }
+
     BOOL bAppearanceChanged = FALSE;
 
 
@@ -1009,10 +1026,23 @@ void OfaViewTabPage::Reset( const SfxIte
     SvtMiscOptions aMiscOptions;
 
     if( aMiscOptions.GetSymbolsSize() != SFX_SYMBOLS_SIZE_AUTO )
-        nBigLB_InitialSelection = ( aMiscOptions.AreCurrentSymbolsLarge() )? 2 : 1;
-    aIconSizeLB.SelectEntryPos( nBigLB_InitialSelection );
+        nSizeLB_InitialSelection = ( aMiscOptions.AreCurrentSymbolsLarge() )? 2 : 1;
+    aIconSizeLB.SelectEntryPos( nSizeLB_InitialSelection );
     aIconSizeLB.SaveValue();
 
+    if( aMiscOptions.GetSymbolsStyle() != SFX_SYMBOLS_STYLE_AUTO )
+    {
+        switch ( aMiscOptions.GetCurrentSymbolsStyle() )
+        {
+            case SFX_SYMBOLS_STYLE_DEFAULT:    nStyleLB_InitialSelection = 1; break;
+            case SFX_SYMBOLS_STYLE_INDUSTRIAL: nStyleLB_InitialSelection = 2; break;
+            case SFX_SYMBOLS_STYLE_CRYSTAL:    nStyleLB_InitialSelection = 3; break;
+            default:                           nStyleLB_InitialSelection = 0; break;
+        }
+    }
+    aIconStyleLB.SelectEntryPos( nStyleLB_InitialSelection );
+    aIconStyleLB.SaveValue();
+
     if( m_aSystemFont.IsEnabled() )
     {
         SvtAccessibilityOptions aAccessibilityOptions;
--- svx/source/dialog/optgdlg.hrc	2005-06-01 15:57:42.345380000 +0200
+++ svx/source/dialog/optgdlg.hrc	2005-06-01 15:59:55.744371325 +0200
@@ -96,9 +96,9 @@
 //#define OFFS_PUSHBUTTON_FIXEDTEXT(base)	(base+(RSC_CD_PUSHBUTTON_HEIGHT-RSC_CD_FIXEDTEXT_HEIGHT)/2)
 
 #define ROW_MF_SCALING          (13)
-#define ROW_FT_ICONSIZE         (ROW_MF_SCALING+RSC_CD_DROPDOWN_HEIGHT+ROWSPACE)
-#define ROW_LB_ICONSIZE         (ROW_FT_ICONSIZE+RSC_CD_FIXEDTEXT_HEIGHT+ROWSPACE)
-#define ROW_CB_SYSTEM_FONT      (ROW_LB_ICONSIZE+RSC_CD_DROPDOWN_HEIGHT+ROWSPACE)
+#define ROW_FT_ICONSIZESTYLE    (ROW_MF_SCALING+RSC_CD_DROPDOWN_HEIGHT+ROWSPACE)
+#define ROW_LB_ICONSIZESTYLE    (ROW_FT_ICONSIZESTYLE+RSC_CD_FIXEDTEXT_HEIGHT+ROWSPACE)
+#define ROW_CB_SYSTEM_FONT      (ROW_LB_ICONSIZESTYLE+RSC_CD_DROPDOWN_HEIGHT+ROWSPACE)
 #define ROW_CB_FONTANTIALIASING (ROW_CB_SYSTEM_FONT+ (2 * RSC_CD_DROPDOWN_HEIGHT) +ROWSPACE)
 #define ROW_NF_AA_POINTLIMIT	(ROW_CB_FONTANTIALIASING+RSC_CD_FIXEDLINE_HEIGHT+ROWSPACE)
 
@@ -158,8 +158,9 @@
 #define BTN_DOCVIEW					51
 #define BTN_OPENWIN					52
 #define LB_ICONSIZE					53
-#define FT_ICONSIZE					54
-#define CB_SYSTEM_FONT				55
+#define FT_ICONSIZESTYLE			54
+#define LB_ICONSTYLE				55
+#define CB_SYSTEM_FONT				56
 
 #define CB_FONT_SHOW				60
 #define CB_SHOW_INACTIVE			61
--- svx/source/dialog/optgdlg.hxx	2005-06-01 15:57:42.345380000 +0200
+++ svx/source/dialog/optgdlg.hxx	2005-06-01 16:27:29.788070657 +0200
@@ -135,8 +135,9 @@ private:
     FixedLine       aUserInterfaceFL;
     FixedText       aWindowSizeFT;
     MetricField     aWindowSizeMF;
-    FixedText       aIconSizeFT;
+    FixedText       aIconSizeStyleFT;
     ListBox         aIconSizeLB;
+    ListBox         aIconStyleLB;
 	CheckBox   		m_aSystemFont;
 
 #if defined( UNX )
@@ -172,7 +173,8 @@ private:
 	FixedText		aMouseMiddleFT;
 	ListBox			aMouseMiddleLB;
 
-	UINT16			nBigLB_InitialSelection;
+	UINT16			nSizeLB_InitialSelection;
+	UINT16			nStyleLB_InitialSelection;
 	BOOL			bSfxSymbolsAuto;
 
     SvtTabAppearanceCfg* pAppearanceCfg;
--- svx/source/dialog/optgdlg.src	2005-06-01 15:57:42.346380000 +0200
+++ svx/source/dialog/optgdlg.src	2005-06-01 15:59:55.745371153 +0200
@@ -249,17 +249,17 @@ TabPage OFA_TP_VIEW
 		Unit = FUNIT_CUSTOM ;
 		CustomUnitText = "%" ;
 	};
-    FixedText FT_ICONSIZE
+    FixedText FT_ICONSIZESTYLE
 	{
-        Pos = MAP_APPFONT ( 12 , ROW_FT_ICONSIZE  ) ;
+        Pos = MAP_APPFONT ( 12 , ROW_FT_ICONSIZESTYLE  ) ;
         Size = MAP_APPFONT ( 114 , 8 ) ;
-        Text [ de ] = "Symbolgröße" ;
-        Text [ en-US ] = "Icon size" ;
+        Text [ de ] = "Symbolgröße und Symbolstil" ;
+        Text [ en-US ] = "Icon size and style" ;
     };
     ListBox LB_ICONSIZE
 	{
-        Pos = MAP_APPFONT ( 12 , ROW_LB_ICONSIZE ) ;
-        Size = MAP_APPFONT ( 114 , 60 ) ;
+        Pos = MAP_APPFONT ( 12 , ROW_LB_ICONSIZESTYLE ) ;
+        Size = MAP_APPFONT ( 55 , 60 ) ;
 		DropDown = TRUE ;
 		Border = TRUE ;
 		TabStop = TRUE ;
@@ -276,6 +276,28 @@ TabPage OFA_TP_VIEW
 			< "Large" ; > ;
 		};
 	};
+    ListBox LB_ICONSTYLE
+    {
+        Pos = MAP_APPFONT ( 71 , ROW_LB_ICONSIZESTYLE ) ;
+        Size = MAP_APPFONT ( 55 , 60 ) ;
+        DropDown = TRUE ;
+        Border = TRUE ;
+        TabStop = TRUE ;
+        StringList [ de ] =
+        {
+            < "Automatisch" ; > ;
+            < "Standard" ; > ;
+            < "Industrial" ; > ;
+            < "Crystal" ; > ;
+        };
+        StringList [ en-US ] =
+        {
+            < "Automatic" ; > ;
+            < "Default" ; > ;
+            < "Industrial" ; > ;
+            < "Crystal" ; > ;
+        };
+    };
 	CheckBox CB_SYSTEM_FONT
 	{
 		TabStop = TRUE ;
