Index: instsetoo_native/packimages/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo_native/packimages/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- instsetoo_native/packimages/makefile.mk	17 Nov 2004 18:10:31 -0000	1.7
+++ instsetoo_native/packimages/makefile.mk	1 Dec 2004 08:51:07 -0000
@@ -69,7 +69,17 @@ TARGET=packimages
 RSCCUSTOMIMG*=$(PRJNAME)$/util
 .INCLUDE: target.mk
 
-ALLTAR : $(COMMONBIN)$/images.zip
+IMAGES := $(COMMONBIN)$/images.zip
+
+.IF "$(ICONS_GNOME)" != ""
+IMAGES += $(COMMONBIN)$/images_gnome.zip
+.ENDIF
+
+.IF "$(ICONS_KDE)" != ""
+IMAGES += $(COMMONBIN)$/images_kde.zip
+.ENDIF
+
+ALLTAR : $(IMAGES)
 
 $(RES)$/img$/commandimagelist.ilst .PHONY :
     +-$(MKDIR) $(RES)$/img
@@ -84,3 +94,8 @@ $(RES)$/img$/commandimagelist.ilst .PHON
 $(COMMONBIN)$/images.zip .PHONY: $(RES)$/img$/commandimagelist.ilst
     +$(PERL) $(SOLARENV)$/bin$/packimages.pl -g $(SOLARSRC)$/$(RSCDEFIMG) -m $(SOLARSRC)$/$(RSCDEFIMG) -c $(SOLARSRC)$/$(RSCCUSTOMIMG) -l $(SOLARCOMMONRESDIR)$/img -l $(RES)$/img -o $@
 
+$(COMMONBIN)$/images_gnome.zip: $(ICONS_GNOME)
+    +$(COPY) $< $@
+
+$(COMMONBIN)$/images_kde.zip: $(ICONS_KDE)
+    +$(COPY) $< $@
