--- config_office/configure.in	2004-10-26 15:17:20.072058712 +0200
+++ config_office/configure.in	2004-10-26 15:44:33.812692384 +0200
@@ -159,6 +159,12 @@ AC_ARG_ENABLE(kde,
 
                           Usage: --enable-kde
 ],,)
+AC_ARG_WITH(gnome-icons,
+[  --with-gnome-icons      Zip archive with icons for Gnome session.
+],,)
+AC_ARG_WITH(kde-icons,
+[  --with-kde-icons        Zip archive with icons for KDE session.
+],,)
 AC_ARG_WITH(stlport4,
 [  --with-stlport4         The location that STLport4 is installed in. The STL
                           header files are assumed to be in stlport4-home/stlport
@@ -2554,6 +2569,18 @@ AC_SUBST(WIDGETSET_KDE_LIBS)
 AC_SUBST(WIDGETSET_GTK_CFLAGS)
 AC_SUBST(WIDGETSET_GTK_LIBS)
 
+ICONS_GNOME=
+ICONS_KDE=
+if test -f "$with_gnome_icons"; then
+   ICONS_GNOME="$with_gnome_icons"
+fi
+if test -f "$with_kde_icons"; then
+   ICONS_KDE="$with_kde_icons"
+fi
+
+AC_SUBST(ICONS_GNOME)
+AC_SUBST(ICONS_KDE)
+
 dnl ===================================================================
 dnl Disable mozilla if we can
 dnl ===================================================================
--- config_office/set_soenv.in	2004-10-26 15:19:05.986957200 +0200
+++ config_office/set_soenv.in	2004-10-26 15:21:36.285108408 +0200
@@ -1769,6 +1769,8 @@ ToFile( "WIDGETSET_GTK_CFLAGS",  "@WIDGE
 ToFile( "WIDGETSET_GTK_LIBS",    "@WIDGETSET_GTK_LIBS@",   "e" );
 ToFile( "WIDGETSET_KDE_CFLAGS",  "@WIDGETSET_KDE_CFLAGS@", "e" );
 ToFile( "WIDGETSET_KDE_LIBS",    "@WIDGETSET_KDE_LIBS@",   "e" );
+ToFile( "ICONS_GNOME",       "@ICONS_GNOME@",      "e" );
+ToFile( "ICONS_KDE",         "@ICONS_KDE@",        "e" );
 if ($platform ne "$Winnt")
 {  ToFile( "TEMP",              $TEMP,              "e" );
    ToFile( "TMP",               $TEMP,              "e" );
