Index: solenv/bin/make_installer.pl
===================================================================
RCS file: /cvs/tools/solenv/bin/make_installer.pl,v
retrieving revision 1.20
diff -u -p -u -r1.20 make_installer.pl
--- solenv/bin/make_installer.pl	18 Oct 2004 13:51:13 -0000	1.20
+++ solenv/bin/make_installer.pl	11 Nov 2004 15:47:35 -0000
@@ -117,6 +117,65 @@ use installer::windows::shortcut;
 use installer::windows::upgrade;
 use installer::ziplist;
 
+sub install_simple ($$$$)
+{
+    my ($packagename, $directoriesarray, $filesarray, $linksarray) = @_;
+    my $destdir = $installer::globals::destdir;
+    my @lines = (); 
+
+    `mkdir -p $destdir` if $destdir ne "";
+    `mkdir -p $destdir$installer::globals::rootpath`;
+
+    # Create Directories
+    for ( my $i = 0; $i <= $#{$directoriesarray}; $i++ )
+    {
+	my $onedir = ${$directoriesarray}[$i];
+	my $dir = "";
+	
+	if ( $onedir->{'Dir'} ) { $dir = $onedir->{'Dir'}; }
+	
+	if ((!($dir =~ /\bPREDEFINED_/ )) || ( $dir =~ /\bPREDEFINED_PROGDIR\b/ ))
+	{
+#	    printf "mkdir $destdir$onedir->{'HostName'}\n";
+	    mkdir $destdir . $onedir->{'HostName'};
+	    push @lines, "%dir " . $onedir->{'HostName'} . "\n";
+	} 
+    }
+
+    for ( my $i = 0; $i <= $#{$filesarray}; $i++ )
+    {
+	my $onefile = ${$filesarray}[$i];
+	my $unixrights = $onefile->{'UnixRights'};
+	my $destination = $onefile->{'destination'};
+	my $sourcepath = $onefile->{'sourcepath'};
+	
+#        printf "mv $sourcepath $destdir$destination\n";
+	`cp -af '$sourcepath' '$destdir$destination'`;
+	`chmod $unixrights '$destdir$destination'`;
+	push @lines, "$destination\n";
+    }
+    
+    for ( my $i = 0; $i <= $#{$linksarray}; $i++ )
+    {
+	my $onelink = ${$linksarray}[$i];
+	my $destination = $onelink->{'destination'};
+	my $destinationfile = $onelink->{'destinationfile'};
+	
+#	print "link $destinationfile -> $destdir$destination\n";
+	`ln -sf '$destinationfile' '$destdir$destination'`;
+	push @lines, "$destination\n";
+    }
+
+    if ( $destdir ne "" )
+    {
+	my $filelist;
+	my $fname = $installer::globals::destdir . "/$packagename";
+	open ($filelist, ">$fname") || die "Can't open $fname: $!";
+	print $filelist @lines;
+	close ($filelist);
+    }
+}
+
 #################################################
 # Main program
 #################################################
@@ -1059,37 +1118,45 @@ for ( my $n = 0; $n <= $#installer::glob
 			# Example for a link: l 000 root sys /usr/bin/linkname filename
 			# The source field specifies the file to link to
 
-			my $epmfilename = "epm_" . $installer::globals::product . "_" . $onepackagename . "_" . $installer::globals::compiler . "_" . $installer::globals::build . "_" . $installer::globals::minor . "_" . $$languagestringref . ".lst";
+			if ( $installer::globals::simple )
+			{
+				print "... installing module $onepackagename ... \n";
+			        install_simple ($onepackagename, $dirsinpackage, $filesinpackage, $linksinpackage);
+			}
+			else # use EPM to package
+			{
+				my $epmfilename = "epm_" . $installer::globals::product . "_" . $onepackagename . "_" . $installer::globals::compiler . "_" . $installer::globals::build . "_" . $installer::globals::minor . "_" . $$languagestringref . ".lst";
 
-			print "... creating epm list file $epmfilename ... \n";
+				print "... creating epm list file $epmfilename ... \n";
 
-			my $completeepmfilename = $listfiledir . $installer::globals::separator . $epmfilename;
+				my $completeepmfilename = $listfiledir . $installer::globals::separator . $epmfilename;
 
-			my @epmfile = ();
+				my @epmfile = ();
 
-			my $epmheaderref = installer::epmfile::create_epm_header($allvariableshashref, $filesinproductlanguageresolvedarrayref, $languagesarrayref, $onepackage);
-			installer::epmfile::adding_header_to_epm_file(\@epmfile, $epmheaderref);
+				my $epmheaderref = installer::epmfile::create_epm_header($allvariableshashref, $filesinproductlanguageresolvedarrayref, $languagesarrayref, $onepackage);
+				installer::epmfile::adding_header_to_epm_file(\@epmfile, $epmheaderref);
 
-			# adding directories, files and links into epm file
+				# adding directories, files and links into epm file
 			
-			installer::epmfile::put_directories_into_epmfile($dirsinpackage, \@epmfile );
-			installer::epmfile::put_files_into_epmfile($filesinpackage, \@epmfile );
-			installer::epmfile::put_links_into_epmfile($linksinpackage, \@epmfile );
-
-			if ((!( $shellscriptsfilename eq "" )) && (!($installer::globals::iswindowsbuild))) { installer::epmfile::adding_shellscripts_to_epm_file(\@epmfile, $shellscriptsfilename, $packagerootpath, $allvariableshashref); }
+				installer::epmfile::put_directories_into_epmfile($dirsinpackage, \@epmfile );
+				installer::epmfile::put_files_into_epmfile($filesinpackage, \@epmfile );
+				installer::epmfile::put_links_into_epmfile($linksinpackage, \@epmfile );
+				
+				if ((!( $shellscriptsfilename eq "" )) && (!($installer::globals::iswindowsbuild))) { installer::epmfile::adding_shellscripts_to_epm_file(\@epmfile, $shellscriptsfilename, $packagerootpath, $allvariableshashref); }
 
-			installer::files::save_file($completeepmfilename ,\@epmfile);
+				installer::files::save_file($completeepmfilename ,\@epmfile);
 
-			# ... splitting the rootpath into a relocatable part and a static part, if possible
+				# ... splitting the rootpath into a relocatable part and a static part, if possible
 
-			my $staticpath = "";
-			my $relocatablepath = "";
-			installer::epmfile::analyze_rootpath($packagerootpath, \$staticpath, \$relocatablepath);
+				my $staticpath = "";
+				my $relocatablepath = "";
+				installer::epmfile::analyze_rootpath($packagerootpath, \$staticpath, \$relocatablepath);
 
-			# ... replacing the variable PRODUCTDIRECTORYNAME in the shellscriptfile by $staticpath
+				# ... replacing the variable PRODUCTDIRECTORYNAME in the shellscriptfile by $staticpath
 
-			installer::epmfile::resolve_path_in_epm_list_before_packaging(\@epmfile, $completeepmfilename, "PRODUCTDIRECTORYNAME", $staticpath);
-			installer::files::save_file($completeepmfilename ,\@epmfile);
+				installer::epmfile::resolve_path_in_epm_list_before_packaging(\@epmfile, $completeepmfilename, "PRODUCTDIRECTORYNAME", $staticpath);
+				installer::files::save_file($completeepmfilename ,\@epmfile);
+			}
 	
 			# changing into the "install" directory to create installation sets
 	
@@ -1168,8 +1235,11 @@ for ( my $n = 0; $n <= $#installer::glob
 			}
 			else	# this is the standard epm (not relocatable) or ( nonlinux and nonsolaris )
 			{			
-				installer::epmfile::resolve_path_in_epm_list_before_packaging(\@epmfile, $completeepmfilename, "\$\$PRODUCTINSTALLLOCATION", $relocatablepath);
-				installer::files::save_file($completeepmfilename ,\@epmfile);
+			        if ( !$installer::globals::simple )
+				{
+					installer::epmfile::resolve_path_in_epm_list_before_packaging(\@epmfile, $completeepmfilename, "\$\$PRODUCTINSTALLLOCATION", $relocatablepath);
+					installer::files::save_file($completeepmfilename ,\@epmfile);
+				}
 
 				if ( $installer::globals::call_epm )
 				{
Index: solenv/bin/modules/installer/globals.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/globals.pm,v
retrieving revision 1.18
diff -u -p -u -r1.18 globals.pm
--- solenv/bin/modules/installer/globals.pm	18 Oct 2004 13:52:14 -0000	1.18
+++ solenv/bin/modules/installer/globals.pm	11 Nov 2004 15:47:35 -0000
@@ -77,6 +77,7 @@ BEGIN
 	$product = "";
 	$languagelist = "";
 
+	$destdir = "";
 	$rootpath = "";
 	
 	$productextension = "";
@@ -129,6 +130,7 @@ BEGIN
 	$is_special_epm = 0;
 	$epm_in_path = 0;
 	$epm_path = "";
+	$simple = 0;
 	$call_epm = 1;
 	$packageformat = "";
 	$packagename = "";
Index: solenv/bin/modules/installer/parameter.pm
===================================================================
RCS file: /cvs/tools/solenv/bin/modules/installer/parameter.pm,v
retrieving revision 1.10
diff -u -p -u -r1.10 parameter.pm
--- solenv/bin/modules/installer/parameter.pm	18 Oct 2004 13:53:07 -0000	1.10
+++ solenv/bin/modules/installer/parameter.pm	11 Nov 2004 15:47:35 -0000
@@ -88,6 +88,7 @@ The following parameter are needed:
 -l: Language of the product (comma and hash) (optional, defined in productlist)
 -b: Build, e.g. srx645 (optional)
 -m: Minor, e.g. m10 (optional)
+-simple: Path to do a simple install to
 -c: Compiler, e.g. wntmsci8, unxlngi5, unxsols4, ... (optional) 
 -u: Path, in which zipfiles are unpacked (optional)
 -msitemplate: Source of the msi file templates (Windows compiler only)
@@ -159,6 +160,7 @@ sub getparameter
 {
 	if ( $installer::globals::debug ) { installer::logger::debuginfo("installer::parameter::getparameter"); }
 
+	my $destdir;
 	while ( $#ARGV >= 0 )
 	{
 		my $param = shift(@ARGV);
@@ -190,6 +192,19 @@ sub getparameter
 		elsif ($param eq "-copyproject") { $installer::globals::is_copy_only_project = 1; }
 		elsif ($param eq "-languagepack") { $installer::globals::languagepack = 1; }
 		elsif ($param eq "-addchildprojects") { $installer::globals::addchildprojects = 1; }
+		elsif ($param eq "-destdir")
+		{
+		    $installer::globals::rootpath ne "" && die "must set destdir before -i or -simple";
+		    $installer::globals::destdir = shift @ARGV;
+		}
+		elsif ($param eq "-simple")
+		{
+		    $installer::globals::simple = 1;
+		    $installer::globals::call_epm = 0;
+		    my $path = shift(@ARGV);
+		    $path =~ s/^$installer::globals::destdir//;
+		    $installer::globals::rootpath = $path;
+		}
 		else
 		{
 			print("\n*************************************\n");
