--- solenv/bin/modules/installer/globals.pm.old	2005-07-18 16:03:28.000000000 +0200
+++ solenv/bin/modules/installer/globals.pm	2005-08-17 18:56:19.000000000 +0200
@@ -158,7 +158,7 @@
 	@linuxlinks = ();
 	@linkrpms = ();
 	
-	$strip = 0;
+	$strip = 1;
 	$solarjava = 0;
 	$jdklib = "";
 	$jrepath = "";
--- solenv/bin/make_installer.pl.old	2005-07-20 12:07:41.000000000 +0200
+++ solenv/bin/make_installer.pl	2005-08-17 19:50:52.000000000 +0200
@@ -1200,7 +1200,7 @@
 			# 1. copy all files that need to be stripped locally 
 			# 2. strip all these files
 
-			installer::strip::strip_libraries($filesinpackage, $languagestringref);
+			if ( $installer::globals::strip ) { installer::strip::strip_libraries($filesinpackage, $languagestringref); }
 			if ( $installer::globals::globallogging ) { installer::files::save_array_of_hashes($loggingdir . $packagename ."_files.log", $filesinpackage); }
 
 			###############################################################
--- solenv/bin/modules/installer/parameter.pm.old	2005-07-18 16:03:40.000000000 +0200
+++ solenv/bin/modules/installer/parameter.pm	2005-08-17 19:11:55.000000000 +0200
@@ -98,6 +98,7 @@
 -format: Package format
 -debian: Create Debian packages for Linux
 -dontunzip: do not unzip all files with flag ARCHIVE
+-dontstrip : do not strip binaries, keep the debugging symbols
 -dontcallepm : do not call epm to create install sets (opt., non-Windows only)
 -ispatchedepm : Usage of a patched (non-standard) epm (opt., non-Windows only)
 -packagelist : file, containing a list of module gids (opt., non-Windows only)
@@ -181,6 +182,7 @@
 		elsif ($param eq "-u") { $installer::globals::unpackpath = shift(@ARGV); }
 		elsif ($param eq "-i") { $installer::globals::rootpath = shift(@ARGV); }
 		elsif ($param eq "-dontcallepm") { $installer::globals::call_epm = 0; }
+		elsif ($param eq "-dontstrip") { $installer::globals::strip = 0; }
 		elsif ($param eq "-msitemplate") { $installer::globals::idttemplatepath = shift(@ARGV); }
 		elsif ($param eq "-msilanguage") { $installer::globals::idtlanguagepath = shift(@ARGV); }
 		elsif ($param eq "-patchinc") { $installer::globals::patchincludepath = shift(@ARGV); }
