Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.112
diff -u -r1.112 configure.in
--- config_office/configure.in	30 Mar 2005 11:38:08 -0000	1.112
+++ config_office/configure.in	4 Apr 2005 11:49:17 -0000
@@ -2516,9 +2556,7 @@
     CPPFLAGS="$CPPFLAGS $MOZ_NSS_CFLAGS"
     LDFLAGS="$LDFLAGS $MOZ_NSS_LIBS"
     AC_CHECK_LIB(nss3, PK11_GetCertFromPrivateKey, [], 
-      [AC_MSG_ERROR(PK11_GetCertFromPrivateKey missing but needed.
-See https://bugzilla.mozilla.org/show_bug.cgi?id=262274.
-Fixed since nss 3.9.3 (contained by e.g. mozilla >= 1.7.5))], [])
+      [MOZ_NSS_HACK=YES],[])
     LDFLAGS="$save_LDFLAGS"
     CPPFLAGS="$save_CPPFLAGS"
     LIBS="$save_LIBS"
@@ -2637,6 +2675,7 @@
 AC_SUBST(MOZ_LIB_XPCOM)
 AC_SUBST(MOZ_NSS_CFLAGS)
 AC_SUBST(MOZ_LDAP_CFLAGS)
+AC_SUBST(MOZ_NSS_HACK)
 
 dnl ===================================================================
 dnl Check for system sane


Index: config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.55
diff -u -r1.55 set_soenv.in
--- config_office/set_soenv.in	30 Mar 2005 11:38:22 -0000	1.55
+++ config_office/set_soenv.in	4 Apr 2005 11:51:09 -0000
@@ -1670,6 +1678,7 @@
 ToFile( "MOZ_LIB_XPCOM",     "@MOZ_LIB_XPCOM@",    "e" );
 ToFile( "MOZ_NSS_CFLAGS",    "@MOZ_NSS_CFLAGS@",   "e" );
 ToFile( "MOZ_LDAP_CFLAGS",   "@MOZ_LDAP_CFLAGS@",  "e" );
+ToFile( "MOZ_NSS_HACK",      "@MOZ_NSS_HACK@",     "e" );
 ToFile( "WITH_FONTS",        "@WITH_FONTS@",       "e" );
 ToFile( "WITH_BINFILTER",    "@WITH_BINFILTER@",   "e" );
 ToFile( "BUILD_TYPE",        "@BUILD_TYPE@",       "e" );
