--- xmerge/source/xmerge/java/org/openoffice/xmerge/ConvertData.java.old	2005-12-14 16:21:42.000000000 +0200
+++ xmerge/source/xmerge/java/org/openoffice/xmerge/ConvertData.java	2005-12-14 16:22:05.000000000 +0200
@@ -127,8 +127,8 @@
      *           <code>Vector</code> of <code>Document</code> objects.
      */
     public Enumeration getDocumentEnumeration() {
-        Enumeration enum = v.elements();
-		return (enum);
+        Enumeration enumerate = v.elements();
+		return (enumerate);
 	}
 
 
--- xmerge/source/xmerge/java/org/openoffice/xmerge/converter/xml/xslt/DocumentDeserializerImpl.java.old	2005-12-14 16:22:34.000000000 +0200
+++ xmerge/source/xmerge/java/org/openoffice/xmerge/converter/xml/xslt/DocumentDeserializerImpl.java	2005-12-14 16:22:59.000000000 +0200
@@ -155,14 +155,14 @@
      */
     public Document deserialize() throws ConvertException, IOException {
 	log("\nFound the XSLT deserializer");
-	Enumeration enum = cd.getDocumentEnumeration();
+	Enumeration enumerate = cd.getDocumentEnumeration();
 	org.w3c.dom.Document domDoc=null;
 	DOMDocument docOut=null;
 	GenericOfficeDocument doc = null;
 	ByteArrayOutputStream baos =null;
 	GenericOfficeDocument sxwDoc = new GenericOfficeDocument("output");
-	while (enum.hasMoreElements()) {       
-		 docOut = (DOMDocument) enum.nextElement();
+	while (enumerate.hasMoreElements()) {       
+		 docOut = (DOMDocument) enumerate.nextElement();
 	}
 	domDoc = docOut.getContentDOM();	
 	try{
--- xmerge/source/xmerge/java/org/openoffice/xmerge/converter/xml/xslt/PluginFactoryImpl.java.old	2005-12-14 16:23:11.000000000 +0200
+++ xmerge/source/xmerge/java/org/openoffice/xmerge/converter/xml/xslt/PluginFactoryImpl.java	2005-12-14 16:23:27.000000000 +0200
@@ -183,9 +183,9 @@
 	String ext= ".txt";
 	String mimeType = null;
 	ConverterInfo ci = this.getConverterInfo();
-	Enumeration enum = ci.getDeviceMime();
-	while (enum.hasMoreElements()) {       
-	    mimeType= (String) enum.nextElement();
+	Enumeration enumerate = ci.getDeviceMime();
+	while (enumerate.hasMoreElements()) {       
+	    mimeType= (String) enumerate.nextElement();
 	}
 	try {
 	    props.load(is);
