Index: gnujre.cxx
===================================================================
RCS file: /cvs/udk/jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx,v
retrieving revision 1.9.14.1
diff -u -u -r1.9.14.1 gnujre.cxx
--- jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	11 Aug 2006 10:44:52 -0000	1.9.14.1
+++ jvmfwk/plugins/sunmajor/pluginlib/gnujre.cxx	27 Aug 2006 00:59:25 -0000
@@ -55,6 +55,12 @@
     static char const * ar[] = {
         "gij",
         "bin/gij",
+	"gij-4.3",
+	"bin/gij-4.3",
+	"gij-4.2",
+	"bin/gij-4.2",
+	"gij-4.1",
+	"bin/gij-4.1",
     };
     *size = sizeof (ar) / sizeof (char*);
     return ar;
@@ -65,6 +69,19 @@
     static char const* ar[]= {
           "/lib/" JFW_PLUGIN_ARCH "/client/libjvm.so",
           "/gcj-4.1.1/libjvm.so",
+	  "/gcj-4.3-90/libjvm.so",
+	  "/gcj-4.2-81/libjvm.so",
+	  "/gcj-4.2/libjvm.so",
+	  "/gcj-4.2.1/libjvm.so",
+	  "/gcj-4.2.2/libjvm.so",
+	  "/gcj-4.1-71/libjvm.so",
+	  "/gcj-4_1/libjvm.so",
+	  "/gcj-4.1/libjvm.so",
+	  "/libgcj.so.81",
+	  "/libgcj.so.80",
+	  "/libgcj.so.8",
+	  "/libgcj.so.71",
+	  "/libgcj.so.70",
           "/libgcj.so.7",
           "/libgcj.so.6"
     };
