--- jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx.old	2006-06-20 02:10:31.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx	2006-11-10 21:31:27.000000000 +0100
@@ -51,12 +51,17 @@
    by "\xXX\xXX"
  */
 BEGIN_VENDOR_MAP()
+#if defined(LINUX) && defined(X86_64)
+    VENDOR_MAP_ENTRY("Free Software Foundation, Inc.", GnuInfo)
+#endif
     VENDOR_MAP_ENTRY("Sun Microsystems Inc.", SunInfo)
     VENDOR_MAP_ENTRY("IBM Corporation", OtherInfo)
     VENDOR_MAP_ENTRY("Blackdown Java-Linux Team", OtherInfo)
     VENDOR_MAP_ENTRY("Apple Computer, Inc.", OtherInfo)
     VENDOR_MAP_ENTRY("BEA Systems, Inc.", OtherInfo)
+#if !defined(LINUX) || !defined(X86_64)
     VENDOR_MAP_ENTRY("Free Software Foundation, Inc.", GnuInfo)
+#endif
 END_VENDOR_MAP()    
 
 
--- /dev/null	2006-06-16 15:54:00.000000000 +0200
+++ jvmfwk/distributions/OpenOfficeorg/javavendors_linux_x86_64.xml	2006-11-10 21:36:33.000000000 +0100
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<javaSelection xmlns="http://openoffice.org/2004/java/framework/1.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+ 
+ <updated>2004-01-30</updated>
+ 
+ <vendorInfos> 
+  <vendor name="Sun Microsystems Inc.">
+    <minVersion>1.3.1</minVersion>
+  </vendor>
+  <vendor name="IBM Corporation">
+    <minVersion>1.3.1</minVersion>
+    <maxVersion>1.4.2</maxVersion>
+  </vendor>
+  <vendor name="Blackdown Java-Linux Team">
+    <minVersion>1.3.1</minVersion>
+  </vendor>
+  <vendor name="Free Software Foundation, Inc.">
+    <minVersion>1.4.2</minVersion>
+  </vendor>
+  <vendor name="BEA Systems, Inc.">
+    <minVersion>1.4.1</minVersion>
+  </vendor>
+ </vendorInfos>
+
+ <plugins>
+  <library vendor="Free Software Foundation, Inc.">sunjavaplugin.so</library>
+  <library vendor="Sun Microsystems Inc.">sunjavaplugin.so</library>
+  <library vendor="IBM Corporation">sunjavaplugin.so</library>
+  <library vendor="Blackdown Java-Linux Team">sunjavaplugin.so</library>
+  <library vendor="BEA Systems, Inc.">sunjavaplugin.so</library>
+ </plugins>
+</javaSelection>
--- jvmfwk/distributions/OpenOfficeorg/makefile.mk.old	2006-01-31 19:28:42.000000000 +0100
+++ jvmfwk/distributions/OpenOfficeorg/makefile.mk	2006-11-10 22:05:48.000000000 +0100
@@ -49,6 +49,8 @@
 .IF "$(GUI)"=="UNX"
 .IF "$(OS)"=="MACOSX"
 	+-$(COPY) javavendors_macosx.xml $(BIN)$/javavendors_ooo.xml
+.ELIF "$(OS)"=="LINUX" && "$(CPUNAME)"=="X86_64"
+	+-$(COPY) javavendors_linux_x86_64.xml $(BIN)$/javavendors_ooo.xml
 .ELIF "$(OS)"=="LINUX" || "$(OS)"=="FREEBSD"
 	+-$(COPY) javavendors_linux.xml $(BIN)$/javavendors_ooo.xml
 .ELSE
