--- /dev/null	2006-01-31 01:27:56.000000000 +0100
+++ scp2/source/kde/file_kde.scp	2006-04-04 09:54:22.000000000 +0200
@@ -0,0 +1,45 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+#include "macros.inc"
+
+File gid_File_Lib_Kdebe
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Dir_Program;
+    Name = STRING(CONCAT2(kdebe1.uno,UNXSUFFIX));
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Regmergefile = "kdebe1-ucd.txt";
+End
+
--- /dev/null	2006-01-31 01:27:56.000000000 +0100
+++ scp2/source/kde/localize.sdf	2006-04-03 17:39:07.000000000 +0200
@@ -0,0 +1,103 @@
+#
+#    ####    ###     #   #   ###   #####    #####  ####   #####  #####  
+#    #   #  #   #    ##  #  #   #    #      #      #   #    #      #    
+#    #   #  #   #    # # #  #   #    #      ###    #   #    #      #    
+#    #   #  #   #    #  ##  #   #    #      #      #   #    #      #    
+#    ####    ###     #   #   ###     #      #####  ####   #####    #    
+#
+#    DO NOT EDIT! This file will be overwritten by localisation process
+#
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	cs	Tento modul umožňuje integraci %PRODUCTNAME %PRODUCTVERSION do pracovního prostředí KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	es	Este módulo permite la integración de %PRODUCTNAME %PRODUCTVERSION en el entorno de escritorio de KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	fr	Ce module se charge de l'intégration de %PRODUCTNAME %PRODUCTVERSION dans l'environnement de bureau KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	hu	Ez a modul lehetővé teszi a %PRODUCTNAME %PRODUCTVERSION együttműködését a KDE grafikus munkakörnyezettel.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	it	Questo modulo permette di integrare %PRODUCTNAME %PRODUCTVERSION nel Desktop Environment di KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	ja	このモジュールは %PRODUCTNAME %PRODUCTVERSION を KDE Desktop Environment に統合します。				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	ko	이 모듈은 %PRODUCTNAME %PRODUCTVERSION을(를) KDE Desktop Environment로 통합시킵니다.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	nl	Deze module zorgt voor de integratie van %PRODUCTNAME %PRODUCTVERSION in het KDE DesktopEnvironment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	pt-BR	Este módulo se encarrega da integração do %PRODUCTNAME %PRODUCTVERSION no desktop KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	sk	Tento modul umožňuje integráciu %PRODUCTNAME %PRODUCTVERSION do KDE Desktop Environment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	sv	Den här modulen ser till att %PRODUCTNAME %PRODUCTVERSION integreras i skrivbordsmiljön KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	tr	Bu modül KDE masaüstü dağıtımı %PRODUCTNAME %PRODUCTVERSION uyumu sağlar.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	zh-CN	这个模块将 %PRODUCTNAME %PRODUCTVERSION 整合到 SOLARIS 的 KDE 操作界面内。				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	zh-TW	這個模組將 %PRODUCTNAME %PRODUCTVERSION 整合到 KDE 操作界面。				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	cs	Integrace do KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	da	KDE integration				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	es	Integración KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	fr	Intégration KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	hu	KDE-környezet				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	it	Integrazione KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	ja	KDE 統合				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	ko	KDE 통합				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	nl	KDE integratie				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	pt-BR	Integração KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	sk	Integrácia do KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	sv	KDE-integration				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	tr	KDE Entegrasyonu				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	zh-CN	KDE 整合				2002-02-02 02:02:02
+scp2	source\kde\module_kde.lng	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	zh-TW	KDE 整合				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	cs	Tento modul umožňuje integraci %PRODUCTNAME %PRODUCTVERSION do pracovního prostředí KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	es	Integración del sistema de %PRODUCTNAME %PRODUCTVERSION con KDE Desktop Environment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	fr	Intégration système de %PRODUCTNAME %PRODUCTVERSION dans l'environnement de bureau KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	hu	Ez a modul lehetővé teszi a %PRODUCTNAME %PRODUCTVERSION együttműködését a KDE grafikus munkakörnyezettel.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	it	Integrazione di %PRODUCTNAME %PRODUCTVERSION nel KDE Desktop Environment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	ja	KDE Desktop Environment への %PRODUCTNAME %PRODUCTVERSION のシステム統合。				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	km	ការ​រួមបញ្ចូល​ប្រព័ន្ធ​នៃ %PRODUCTNAME %PRODUCTVERSION ទៅ​ក្នុង KDE Desktop Environment ។				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	ko	KDE 데스크톱 환경에 %PRODUCTNAME %PRODUCTVERSION 시스템 통합				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	nl	Systeemintegratie van %PRODUCTNAME %PRODUCTVERSION in het KDE DesktopEnvironment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	pt-BR	Integração do sistema do %PRODUCTNAME %PRODUCTVERSION ao KDE Desktop Environment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	sk	Tento modul umožňuje integráciu %PRODUCTNAME %PRODUCTVERSION do KDE Desktop Environment.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	sv	Systemintegration av %PRODUCTNAME %PRODUCTVERSION i skrivbordsmiljön KDE.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	tr	Bu modül KDE masaüstü dağıtımı %PRODUCTNAME %PRODUCTVERSION uyumu sağlar.				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	zh-CN	将 %PRODUCTNAME %PRODUCTVERSION 系统集成到 KDE 桌面环境。				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_DESC_MODULE_OPTIONAL_KDE				0	zh-TW	%PRODUCTNAME %PRODUCTVERSION 與 KDE 桌面環境的系統整合。				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	cs	Integrace do KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	da	KDE integration				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	es	Integración KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	fr	Intégration KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	hu	KDE-környezet				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	it	Integrazione KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	ja	KDE 統合				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	km	ការ​រួមបញ្ចូល KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	ko	KDE 통합				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	nl	KDE-integratie				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	pt-BR	Integração KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	sk	Integrácia do KDE				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	sv	KDE-integration				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	tr	KDE Entegrasyonu				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	zh-CN	KDE 整合				2002-02-02 02:02:02
+scp2	source\kde\module_kde.ulf	0	LngText	STR_NAME_MODULE_OPTIONAL_KDE				0	zh-TW	KDE 整合				2002-02-02 02:02:02
--- /dev/null	2006-01-31 01:27:56.000000000 +0100
+++ scp2/source/kde/makefile.mk	2006-04-03 17:39:07.000000000 +0200
@@ -0,0 +1,58 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile$
+#
+#   $Revision$
+#
+#   last change: $Author$ $Date$
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJPCH=
+
+PRJNAME=scp2
+TARGET=kde
+TARGETTYPE=CUI
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+SCP_PRODUCT_TYPE=osl
+
+.IF "$(ENABLE_KAB)" == "TRUE"
+PARFILES =                   \
+        module_kde.par     \
+        file_kde.par
+.ENDIF
+
+# --- File ---------------------------------------------------------
+
+.INCLUDE :  target.mk
--- /dev/null	2006-01-31 01:27:56.000000000 +0100
+++ scp2/source/kde/module_kde.scp	2006-04-04 13:53:33.000000000 +0200
@@ -0,0 +1,46 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile$
+ *
+ *  $Revision$
+ *
+ *  last change: $Author$ $Date$
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+Module gid_Module_Optional_Kde
+    ParentID = gid_Module_Optional;
+    Default = YES;
+	Name = "gid_Module_Optional_Kde";
+	Description = "gid_Module_Optional_Kde";
+    Files = (gid_File_Lib_Kdebe);
+    Dirs = ();
+End
+
