diff -urN vcl/unx/inc/i18n_im.hxx ooo_SRC680_m76_src/vcl/unx/inc/i18n_im.hxx
--- vcl/unx/inc/i18n_im.hxx	Wed Feb  2 09:16:01 2005
+++ vcl/unx/inc/i18n_im.hxx	Wed Feb  2 09:16:31 2005
@@ -62,11 +62,15 @@
 #ifndef _SAL_I18N_INPUTMETHOD_HXX
 #define _SAL_I18N_INPUTMETHOD_HXX
 
+#ifndef _VCL_DLLAPI_H
+#include "dllapi.h"
+#endif
+
 extern "C" char* GetMethodName( XIMStyle nStyle, char *pBuf, int nBufSize);
 
 #define bUseInputMethodDefault True
 
-class SalI18N_InputMethod
+class VCL_DLLPUBLIC SalI18N_InputMethod
 {
 	Bool    	mbUseable; 	// system supports locale as well as status 
 							// and preedit style ?  
--- vcl/unx/inc/saldisp.hxx	Wed Feb  2 09:16:01 2005
+++ vcl/unx/inc/saldisp.hxx	Wed Feb  2 09:16:31 2005
@@ -524,7 +524,7 @@
 inline int SalColormap::GetClass() const
 { return pVisual_ ? pVisual_->GetClass() : PseudoColor; }
 
-class SalX11Display : public SalDisplay
+class VCL_DLLPUBLIC SalX11Display : public SalDisplay
 {
 public:
 		     SalX11Display( Display* pDisp,
diff -urN ooo_SRC680_m76_src.orig/vcl/unx/inc/salframe.h ooo_SRC680_m76_src/vcl/unx/inc/salframe.h
--- vcl/unx/inc/salframe.h	Wed Feb  2 09:16:01 2005
+++ vcl/unx/inc/salframe.h	Wed Feb  2 09:16:31 2005
@@ -87,6 +87,10 @@
 #include <timer.hxx>
 #endif
 
+#ifndef _VCL_DLLAPI_H
+#include "dllapi.h"
+#endif
+
 #include <salunx.h>
 #include <list>
 
@@ -104,7 +108,7 @@
 #define SHOWSTATE_NORMAL		1
 #define SHOWSTATE_HIDDEN		2
 
-class X11SalFrame : public SalFrame
+class VCL_DLLPUBLIC X11SalFrame : public SalFrame
 {
     friend class vcl_sal::WMAdaptor;
     friend class vcl_sal::NetWMAdaptor;
