--- desktop/source/app/desktopcontext.cxx	2005-09-08 17:07:10.000000000 +0000
+++ desktop/source/app/desktopcontext.cxx	2006-06-05 17:20:10.000000000 +0000
@@ -56,11 +56,7 @@ Any SAL_CALL DesktopContext::getValueByN
 {
     Any retVal;
 
-    if ( 0 == Name.compareToAscii( DESKTOP_ENVIRONMENT_NAME ) )
-    {
-        retVal = makeAny( Application::GetDesktopEnvironment() );
-    }
-    else if ( 0 == Name.compareToAscii( JAVA_INTERACTION_HANDLER_NAME ))
+    if ( 0 == Name.compareToAscii( JAVA_INTERACTION_HANDLER_NAME ))
     {
         retVal = makeAny( Reference< XInteractionHandler >( new svt::JavaInteractionHandler()) );
     }
--- desktop/source/app/desktopcontext.hxx	2006-06-05 16:56:33.000000000 +0000
+++ desktop/source/app/desktopcontext.hxx	2006-06-05 16:56:48.000000000 +0000
@@ -44,8 +44,6 @@
 #include <uno/current_context.hxx>
 #endif
 
-#define DESKTOP_ENVIRONMENT_NAME "system.desktop-environment"
-
 namespace desktop
 {
     class DesktopContext: public cppu::WeakImplHelper1< com::sun::star::uno::XCurrentContext >
--- vcl/source/app/svmain.cxx	2006-05-18 06:41:04.000000000 +0000
+++ vcl/source/app/svmain.cxx	2006-06-05 17:11:21.000000000 +0000
@@ -155,6 +155,12 @@ using namespace ::com::sun::star::lang;
 #include <fontcfg.hxx>
 #include <configsettings.hxx>
 
+#ifndef _CPPUHELPER_IMPLBASE1_HXX_
+#include <cppuhelper/implbase1.hxx>
+#endif
+#ifndef _UNO_CURRENT_CONTEXT_HXX_
+#include <uno/current_context.hxx>
+#endif
 
 
 // =======================================================================
@@ -284,6 +291,36 @@ public:
     void                Main(){};
 };
 
+class DesktopEnvironmentContext: public cppu::WeakImplHelper1< com::sun::star::uno::XCurrentContext >
+{
+public:
+    DesktopEnvironmentContext( const com::sun::star::uno::Reference< com::sun::star::uno::XCurrentContext > & ctx)
+        : m_xNextContext( ctx ) {}
+
+    // XCurrentContext
+    virtual com::sun::star::uno::Any SAL_CALL getValueByName( const rtl::OUString& Name )
+            throw (com::sun::star::uno::RuntimeException);
+
+private:
+    com::sun::star::uno::Reference< com::sun::star::uno::XCurrentContext > m_xNextContext;
+};
+
+Any SAL_CALL DesktopEnvironmentContext::getValueByName( const rtl::OUString& Name) throw (RuntimeException)
+{
+    Any retVal;
+
+    if ( 0 == Name.compareToAscii( "system.desktop-environment" ) )
+    {
+        retVal = makeAny( Application::GetDesktopEnvironment() );
+    }
+    else if( m_xNextContext.is() )
+    {
+        // Call next context in chain if found
+        retVal = m_xNextContext->getValueByName( Name );
+    }
+    return retVal;
+}
+
 BOOL InitVCL( const ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory > & rSMgr )
 {
     RTL_LOGFILE_CONTEXT( aLog, "vcl (ss112471) ::InitVCL" );
@@ -325,6 +362,10 @@ BOOL InitVCL( const ::com::sun::star::un
         return FALSE;
     RTL_LOGFILE_CONTEXT_TRACE( aLog, "} ::CreateSalInstance" );
 
+    // Desktop Environment context (to be able to get value of "system.desktop-environment" as soon as possible)
+    com::sun::star::uno::setCurrentContext(
+        new DesktopEnvironmentContext( com::sun::star::uno::getCurrentContext() ) );
+
 	// Initialize application instance (should be done after initialization of VCL SAL part)
     if( pSVData->mpApp )
         // call init to initialize application class
