Index: vcl/inc/keycodes.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/keycodes.hxx,v
retrieving revision 1.6
diff -u -p -u -r1.6 keycodes.hxx
--- vcl/inc/keycodes.hxx	9 Sep 2005 11:07:18 -0000	1.6
+++ vcl/inc/keycodes.hxx	18 Oct 2005 10:43:55 -0000
@@ -161,6 +161,8 @@
 #define KEY_HELP		((USHORT)(KEYGROUP_MISC+27))
 #define KEY_HANGUL_HANJA ((USHORT)(KEYGROUP_MISC+28))
 #define KEY_DECIMAL     ((USHORT)(KEYGROUP_MISC+29))
+#define KEY_TILDE       ((USHORT)(KEYGROUP_MISC+30))
+#define KEY_QUOTELEFT   ((USHORT)(KEYGROUP_MISC+31))
 
 #define KEY_CODE		((USHORT)0x0FFF)
 

Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.34.8.1
diff -u -p -u -r1.34.8.1 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	5 Oct 2005 10:28:28 -0000	1.34.8.1
+++ vcl/unx/gtk/window/gtkframe.cxx	18 Oct 2005 10:44:11 -0000
@@ -162,6 +162,9 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_Redo:          nCode = KEY_REPEAT;     break;
             case GDK_KP_Decimal:
             case GDK_KP_Separator:	nCode = KEY_DECIMAL;	break;
+            case GDK_asciitilde:	nCode = KEY_TILDE;		break;
+            case GDK_leftsinglequotemark:
+            case GDK_quoteleft:	nCode = KEY_QUOTELEFT;		break;
         }
     }
     return nCode;

Index: vcl/unx/source/app/saldisp.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/saldisp.cxx,v
retrieving revision 1.67
diff -u -p -u -r1.67 saldisp.cxx
--- vcl/unx/source/app/saldisp.cxx	9 Sep 2005 13:00:41 -0000	1.67
+++ vcl/unx/source/app/saldisp.cxx	18 Oct 2005 10:45:06 -0000
@@ -1508,6 +1508,14 @@ USHORT SalDisplay::GetKeyCode( KeySym ke
                 nKey = KEY_DIVIDE;
                 *pcPrintable = '/';
                 break;
+            case XK_asciitilde:
+                nKey = KEY_TILDE;
+                *pcPrintable = '~';
+                break;
+            case XK_quoteleft:
+                nKey = KEY_QUOTELEFT;
+                *pcPrintable = '`';
+                break;
         }
     }
     else if( IsFunctionKey( keysym ) )
@@ -1691,6 +1695,12 @@ USHORT SalDisplay::GetKeyCode( KeySym ke
         case XK_slash:
             nKey = KEY_DIVIDE;
             break;
+        case XK_asciitilde:
+            nKey = KEY_TILDE;
+            break;
+        case XK_quoteleft:
+            nKey = KEY_QUOTELEFT;
+            break;
         case XK_period:
             nKey = KEY_POINT;
             break;

Index: framework/collector/cmduicollector.cxx
===================================================================
RCS file: /cvs/framework/framework/collector/cmduicollector.cxx,v
retrieving revision 1.12
diff -u -p -u -r1.12 cmduicollector.cxx
--- framework/collector/cmduicollector.cxx	8 Sep 2005 23:54:40 -0000	1.12
+++ framework/collector/cmduicollector.cxx	18 Oct 2005 10:45:49 -0000
@@ -337,7 +337,12 @@ KeyIdentifierInfo AWTKeyIdentifierMap[] 
     {css::awt::Key::PROPERTIES    , "KEY_PROPERTIES" },
     {css::awt::Key::FRONT         , "KEY_FRONT"      },
     {css::awt::Key::CONTEXTMENU   , "KEY_CONTEXTMENU"},
+    {css::awt::Key::MENU          , "KEY_MENU"       },
     {css::awt::Key::HELP          , "KEY_HELP"       },
+    {css::awt::Key::HANGUL_HANJA  , "KEY_HANGUL_HANJA"},
+    {css::awt::Key::DECIMAL       , "KEY_DECIMAL"    },
+    {css::awt::Key::TILDE         , "KEY_TILDE"      },
+    {css::awt::Key::QUOTELEFT     , "KEY_QUOTELEFT"  },
 	{0                            , ""               } // mark the end of this array!
 };                             
 

Index: offapi/com/sun/star/awt/Key.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/awt/Key.idl,v
retrieving revision 1.10
diff -u -p -u -r1.10 Key.idl
--- offapi/com/sun/star/awt/Key.idl	7 Sep 2005 21:07:36 -0000	1.10
+++ offapi/com/sun/star/awt/Key.idl	18 Oct 2005 10:46:14 -0000
@@ -431,8 +431,28 @@ published constants Key
 	const short CONTEXTMENU = 1305; 
  
 	//------------------------------------------------------------------------- 
+
+	const short MENU = 1306;
+
+	//------------------------------------------------------------------------- 
 	
-	const short HELP = 1306; 
+	const short HELP = 1307;
+
+	//------------------------------------------------------------------------- 
+
+	const short HANGUL_HANJA = 1308;
+
+	//------------------------------------------------------------------------- 
+
+	const short DECIMAL = 1309;
+
+	//------------------------------------------------------------------------- 
+
+	const short TILDE = 1310;
+
+	//------------------------------------------------------------------------- 
+
+	const short QUOTELEFT = 1311;
  
 }; 
  
