Index: source/control/button.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/button.cxx,v
retrieving revision 1.52.18.2
diff -p -u -r1.52.18.2 button.cxx
--- vcl/source/control/button.cxx	17 Aug 2007 14:46:07 -0000	1.52.18.2
+++ vcl/source/control/button.cxx	22 Nov 2007 14:42:55 -0000
@@ -1975,33 +1975,34 @@ void PushButton::EndSelection()
 
 Size PushButton::CalcMinimumSize( long nMaxWidth ) const
 {
-    Size aSize;
+	Size aSize;
 
-    if ( IsSymbol() )
-        aSize = Size( 12, 12 );
-    else if ( IsImage() && ! (ImplGetButtonState() & BUTTON_DRAW_NOIMAGE) )
-        aSize = GetModeImage().GetSizePixel();
-    else if ( PushButton::GetText().Len() && ! (ImplGetButtonState() & BUTTON_DRAW_NOTEXT) )
-    {
-        ULONG nDrawFlags = 0;
-        aSize = GetTextRect( Rectangle( Point(), Size( nMaxWidth ? nMaxWidth : 0x7fffffff, 0x7fffffff ) ),
-                             PushButton::GetText(), ImplGetTextStyle( nDrawFlags ) ).GetSize();
-    }
+	if ( IsSymbol() )
+		aSize = Size( 12, 12 );
+	else if ( IsImage() && ! (ImplGetButtonState() & BUTTON_DRAW_NOIMAGE) )
+		aSize = GetModeImage().GetSizePixel();
+	if ( PushButton::GetText().Len() && ! (ImplGetButtonState() & BUTTON_DRAW_NOTEXT) )
+	{
+		ULONG nDrawFlags = 0;
+		Size textSize = GetTextRect( Rectangle( Point(), Size( nMaxWidth ? nMaxWidth : 0x7fffffff, 0x7fffffff ) ),
+									 PushButton::GetText(), ImplGetTextStyle( nDrawFlags ) ).GetSize();
+	   aSize.Width() += int( textSize.Width () * 1.15 );
+	   aSize.Height() = std::max( aSize.Height(), long( textSize.Height() * 1.15 ) );
+	}
 
-    // cf. ImplDrawPushButton ...
-    aSize.Width() += 8;
-    aSize.Height() += 8;
-    
-    return CalcWindowSize( aSize );
-}
+	// cf. ImplDrawPushButton ...
+	aSize.Width() += 8;
+	aSize.Height() += 8;
 
-// -----------------------------------------------------------------------
+    return CalcWindowSize( aSize );
+}    
 
 Size PushButton::GetOptimalSize(WindowSizeType eType) const
 {
     switch (eType) {
-    case WINDOWSIZE_MINIMUM:
+    case WINDOWSIZE_MINIMUM: {
         return CalcMinimumSize();
+    }
     default:
         return Button::GetOptimalSize( eType );
     }
Index: source/control/edit.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/edit.cxx,v
retrieving revision 1.86.10.2
diff -p -u -I .*$\(Author\|Date\|RCSfile\|Revision\): .*$ -r1.86.10.2 edit.cxx
--- vcl/source/control/edit.cxx	17 Aug 2007 14:45:57 -0000	1.86.10.2
+++ vcl/source/control/edit.cxx	23 Nov 2007 11:05:32 -0000
@@ -2707,9 +2707,8 @@ void Edit::SetSubEdit( Edit* pEdit )
 
 Size Edit::CalcMinimumSize() const
 {
-	Size aSz( GetTextWidth( GetText() ), GetTextHeight() );
-	aSz = CalcWindowSize( aSz );
-	return aSz;
+	Size aSize ( GetTextWidth( GetText() ), GetTextHeight() );
+	return CalcWindowSize( aSize );
 }
 
 // -----------------------------------------------------------------------
Index: source/control/fixed.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/fixed.cxx,v
retrieving revision 1.21.66.1
diff -p -u -I .*$\(Author\|Date\|RCSfile\|Revision\): .*$ -r1.21.66.1 fixed.cxx
--- vcl/source/control/fixed.cxx	17 Aug 2007 13:52:57 -0000	1.21.66.1
+++ vcl/source/control/fixed.cxx	23 Nov 2007 18:32:38 -0000
@@ -400,25 +400,30 @@ void FixedText::DataChanged( const DataC
 
 // -----------------------------------------------------------------------
 
-Size FixedText::CalcMinimumSize( long nMaxWidth ) const
+Size FixedText::CalcMinimumTextSize( Control const *pControl, long nMaxWidth )
 {
-	USHORT nStyle = ImplGetTextStyle( GetStyle() );
-	if ( !( GetStyle() & WB_NOLABEL ) )
+	USHORT nStyle = ImplGetTextStyle( pControl->GetStyle() );
+	if ( !( pControl->GetStyle() & WB_NOLABEL ) )
 		nStyle |= TEXT_DRAW_MNEMONIC;
 
-	Size aSize = GetTextRect( Rectangle( Point(), Size( (nMaxWidth ? nMaxWidth : 0x7fffffff), 0x7fffffff ) ),
-							  GetText(), nStyle ).GetSize();
+	Size aSize = pControl->GetTextRect( Rectangle( Point(), Size( (nMaxWidth ? nMaxWidth : 0x7fffffff), 0x7fffffff ) ),
+									   pControl->GetText(), nStyle ).GetSize();
 
-    if ( GetStyle() & WB_EXTRAOFFSET )
+    if ( pControl->GetStyle() & WB_EXTRAOFFSET )
         aSize.Width() += 2;
     
 	// GetTextRect verkraftet keinen leeren String:
 	if ( aSize.Width() < 0 )
 		aSize.Width() = 0;
 	if ( aSize.Height() <= 0 )
-		aSize.Height() = GetTextHeight();
+		aSize.Height() = pControl->GetTextHeight();
+
+	return aSize;
+}
 
-	return CalcWindowSize( aSize );
+Size FixedText::CalcMinimumSize( long nMaxWidth ) const
+{
+	return CalcWindowSize( CalcMinimumTextSize ( this, nMaxWidth ) );
 }
 // -----------------------------------------------------------------------
 
@@ -689,6 +694,18 @@ void FixedLine::DataChanged( const DataC
 	}
 }
 
+// -----------------------------------------------------------------------
+
+Size FixedLine::GetOptimalSize(WindowSizeType eType) const
+{
+    switch (eType) {
+    case WINDOWSIZE_MINIMUM:
+        return CalcWindowSize( FixedText::CalcMinimumTextSize ( this ) );
+    default:
+        return Control::GetOptimalSize( eType );
+    }
+}
+
 // =======================================================================
 
 void FixedBitmap::ImplInit( Window* pParent, WinBits nStyle )
Index: inc/vcl/fixed.hxx
===================================================================
RCS file: /cvs/gsl/vcl/inc/vcl/fixed.hxx,v
retrieving revision 1.2.142.1
diff -p -u -I .*$\(Author\|Date\|RCSfile\|Revision\): .*$ -r1.2.142.1 fixed.hxx
--- vcl/inc/vcl/fixed.hxx	17 Aug 2007 13:52:57 -0000	1.2.142.1
+++ vcl/inc/vcl/fixed.hxx	23 Nov 2007 18:29:56 -0000
@@ -85,6 +85,7 @@ public:
     virtual void    StateChanged( StateChangedType nType );
     virtual void    DataChanged( const DataChangedEvent& rDCEvt );
 
+    static Size     CalcMinimumTextSize( Control const* pControl, long nMaxWidth = 0 );
     Size            CalcMinimumSize( long nMaxWidth = 0 ) const;
     virtual Size    GetOptimalSize(WindowSizeType eType) const;
 };
@@ -114,6 +115,8 @@ public:
     virtual void    Resize();
     virtual void    StateChanged( StateChangedType nType );
     virtual void    DataChanged( const DataChangedEvent& rDCEvt );
+
+    virtual Size    GetOptimalSize(WindowSizeType eType) const;
 };
 
 // ---------------
Index: source/control/spinfld.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/spinfld.cxx,v
retrieving revision 1.24.46.2
diff -p -u -r1.24.46.2 spinfld.cxx
--- vcl/source/control/spinfld.cxx	17 Aug 2007 14:46:42 -0000	1.24.46.2
+++ vcl/source/control/spinfld.cxx	22 Nov 2007 14:57:29 -0000
@@ -974,7 +974,7 @@ Size SpinField::CalcMinimumSize() const
 	if ( GetStyle() & WB_DROPDOWN )
 		aSz.Width() += GetSettings().GetStyleSettings().GetScrollBarSize();
 	if ( GetStyle() & WB_SPIN )
-		aSz.Width() += GetSettings().GetStyleSettings().GetSpinSize();
+		aSz.Width() += maUpperRect.GetWidth();
 
 	return aSz;
 }
