Index: svtools/source/control/fmtfield.cxx
===================================================================
RCS file: /cvs/util/svtools/source/control/fmtfield.cxx,v
retrieving revision 1.38
diff -u -p -u -r1.38 fmtfield.cxx
--- svtools/source/control/fmtfield.cxx	27 Jun 2007 21:24:43 -0000	1.38
+++ svtools/source/control/fmtfield.cxx	17 Aug 2007 16:12:15 -0000
@@ -1198,7 +1198,10 @@ BOOL DoubleNumericField::CheckText(const
 
 	return FALSE;
 #else
-	return m_pNumberValidator->isValidNumericFragment( sText );
+    if (m_pNumberValidator != NULL)
+        return m_pNumberValidator->isValidNumericFragment( sText );
+    else
+        return TRUE;
 #endif
 }
 
