Index: vcl/source/control/button.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/button.cxx,v
retrieving revision 1.45.132.1
diff -u -p -u -r1.45.132.1 button.cxx
--- vcl/source/control/button.cxx	24 Jan 2007 13:33:19 -0000	1.45.132.1
+++ vcl/source/control/button.cxx	8 Jun 2007 17:30:58 -0000
@@ -1989,6 +1991,10 @@ Size PushButton::CalcMinimumSize( long n
                              PushButton::GetText(), ImplGetTextStyle( nDrawFlags ) ).GetSize();
     }
 
+    // cf. ImplDrawPushButton ...
+    aSize.Width() += 8;
+    aSize.Height() += 8;
+
     return CalcWindowSize( aSize );
 }
 
