Index: sal/osl/unx/nlsupport.c
===================================================================
RCS file: /cvs/porting/sal/osl/unx/nlsupport.c,v
retrieving revision 1.24
diff -u -r1.24 nlsupport.c
--- sal/osl/unx/nlsupport.c	30 Mar 2004 16:29:07 -0000	1.24
+++ sal/osl/unx/nlsupport.c	15 Jun 2005 14:37:23 -0000
@@ -216,6 +216,7 @@
     rtl_uString * pLanguage = NULL;
     rtl_uString * pCountry  = NULL;
     rtl_uString * pVariant  = NULL;
+	rtl_Locale * pResult = NULL;
     
     /* check if locale contains a valid string */
     if( locale )
@@ -244,13 +245,20 @@
                 OSL_ASSERT(pVariant != NULL);
             }
 
-            return rtl_locale_register( pLanguage->buffer, pCountry ? pCountry->buffer : c_locale + 1, pVariant ? pVariant->buffer : c_locale + 1 );
+            pResult = rtl_locale_register( pLanguage->buffer, pCountry ? pCountry->buffer : c_locale + 1, pVariant ? pVariant->buffer : c_locale + 1 );
         }
         else
-            return rtl_locale_register( c_locale, c_locale + 1, c_locale + 1 ); 
+            pResult = rtl_locale_register( c_locale, c_locale + 1, c_locale + 1 ); 
+
+		if( pLanguage )
+			rtl_uString_release( pLanguage );
+		if( pCountry )
+			rtl_uString_release( pCountry );
+		if( pVariant )
+			rtl_uString_release( pVariant );
     }
     
-    return NULL;
+    return pResult;
 }
         
 #if defined(LINUX) || defined(SOLARIS) || defined(IRIX) || defined(NETBSD) || defined(FREEBSD)
