Index: sch/source/core/chtmod3d.cxx
===================================================================
RCS file: /cvs/graphics/sch/source/core/chtmod3d.cxx,v
retrieving revision 1.22
diff -u -r1.22 chtmod3d.cxx
--- sch/source/core/chtmod3d.cxx	24 Feb 2005 14:07:46 -0000	1.22
+++ sch/source/core/chtmod3d.cxx	15 Jun 2005 16:42:33 -0000
@@ -2139,6 +2139,8 @@
 				aPolyPolygon.Insert( aPoly );
 
 				GetPage( 0 )->RemoveObject( 0 );
+
+				delete pTmp;
 			}
 
 			SchE3dExtrudeObj* pObj=new SchE3dExtrudeObj(aDefltAttr3D,
