Index: svx/source/sdr/properties/e3dsceneproperties.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/sdr/properties/e3dsceneproperties.cxx,v
retrieving revision 1.5
diff -u -r1.5 e3dsceneproperties.cxx
--- svx/source/sdr/properties/e3dsceneproperties.cxx	7 Mar 2005 17:33:11 -0000	1.5
+++ svx/source/sdr/properties/e3dsceneproperties.cxx	16 Jun 2005 08:15:05 -0000
@@ -179,15 +179,15 @@
 			if(nCount)
 			{
 				// Generate filtered ItemSet which contains all but the SDRATTR_3DSCENE items.
-				SfxItemSet& rNewSet = *(rSet.Clone(TRUE));
-				DBG_ASSERT(&rNewSet, "E3dSceneProperties::SetMergedItemSet(): Could not clone ItemSet (!)");
+				SfxItemSet *rNewSet = rSet.Clone(TRUE);
+				DBG_ASSERT(rNewSet != NULL, "E3dSceneProperties::SetMergedItemSet(): Could not clone ItemSet (!)");
 				
 				for(sal_uInt16 b(SDRATTR_3DSCENE_FIRST); b <= SDRATTR_3DSCENE_LAST; b++)
 				{
-					rNewSet.ClearItem(b);
+					rNewSet->ClearItem(b);
 				}
 
-				if(rNewSet.Count())
+				if(rNewSet->Count())
 				{
 					for(sal_uInt32 a(0L); a < nCount; a++)
 					{
@@ -196,10 +196,11 @@
 						if(pObj && pObj->ISA(E3dCompoundObject))
 						{
 							// set merged ItemSet at contained 3d object.
-							pObj->SetMergedItemSet(rNewSet, bClearAllItems);
+							pObj->SetMergedItemSet(*rNewSet, bClearAllItems);
 						}
 					}
 				}
+				delete rNewSet;
 			}
 
 			// call parent. This will set items on local object, too.
