Index: configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.173.2.2
diff -u -u -r1.173.2.2 configure.in
--- config_office/configure.in	17 Aug 2006 07:27:21 -0000	1.173.2.2
+++ config_office/configure.in	28 Aug 2006 00:43:46 -0000
@@ -928,6 +928,15 @@
 fi
 AC_SUBST(ENABLE_SYMBOLS)
 
+AC_SYS_LARGEFILE
+if test -n "$ac_cv_sys_file_offset_bits"; then
+   LFS_CFLAGS="-D_FILE_OFFSET_BITS=$ac_cv_sys_file_offset_bits"
+fi
+if test -n "$ac_cv_sys_large_files" && test "$ac_cv_sys_large_files" != "no"; then
+   LFS_CFLAGS="$LFS_CFLAGS -D_LARGE_FILES"
+fi
+AC_SUBST(LFS_CFLAGS)
+
 dnl ===================================================================
 dnl Determine if the solver is to be stripped or not.
 dnl ===================================================================
Index: set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.112.2.1
diff -u -u -r1.112.2.1 set_soenv.in
--- config_office/set_soenv.in	11 Aug 2006 10:44:20 -0000	1.112.2.1
+++ config_office/set_soenv.in	28 Aug 2006 00:43:47 -0000
@@ -1754,6 +1754,7 @@
 ToFile( "ALLOC",             "@ALLOC@",          "e" );
 ToFile( "ENABLE_SYMBOLS",    "@ENABLE_SYMBOLS@",    "e" );
 ToFile( "DISABLE_STRIP",     "@DISABLE_STRIP@",     "e" );
+ToFile( "LFS_CFLAGS",        "@LFS_CFLAGS@",     "e" );
 ToFile( "ENABLE_CRASHDUMP",  "@ENABLE_CRASHDUMP@",  "e" );
 ToFile( "ENABLE_CUPS",       "@ENABLE_CUPS@",       "e" );
 ToFile( "ENABLE_FONTCONFIG", "@ENABLE_FONTCONFIG@", "e" );
Index: inc/settings.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/settings.mk,v
retrieving revision 1.199
diff -u -u -r1.199 settings.mk
--- solenv/inc/settings.mk	21 Jul 2006 10:38:35 -0000	1.199
+++ solenv/inc/settings.mk	28 Aug 2006 00:44:03 -0000
@@ -1291,6 +1291,11 @@
 CDEFS+= -DTF_NEWEX
 .ENDIF
 
+.IF "$(LFS_CFLAGS)" != ""
+CFLAGS+= $(LFS_CFLAGS)
+CXXFLAGS+= $(LFS_CFLAGS)
+.ENDIF
+
 #defaults for UCR HEADER
 UNOUCRBASE*=UCR
 UNOUCROUT*=$(OUT)$/inc$/$(TARGET)
