--- libxmlsec/makefile.mk.ark	2005-03-13 18:41:36.000000000 +0100
+++ libxmlsec/makefile.mk	2005-03-13 18:42:48.000000000 +0100
@@ -106,7 +106,11 @@
 .ELSE
 CONFIGURE_DIR=
 CONFIGURE_ACTION=chmod 777 libxml2-config && .$/configure
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+CONFIGURE_FLAGS=--with-libxslt=no --with-openssl=no --with-gnutls=no --enable-pkgconfig=yes
+.ELSE
 CONFIGURE_FLAGS=--with-libxslt=no --with-openssl=no --with-gnutls=no --enable-pkgconfig=no
+.ENDIF
 BUILD_ACTION=$(GNUMAKE)
 BUILD_DIR=$(CONFIGURE_DIR)
 .ENDIF
