Index: solenv/bin/linkoo
===================================================================
RCS file: /cvs/tools/solenv/bin/linkoo,v
retrieving revision 1.7
diff -u -r1.7 linkoo
--- solenv/bin/linkoo	6 May 2005 09:24:30 -0000	1.7
+++ solenv/bin/linkoo	26 May 2005 16:55:14 -0000
@@ -174,8 +207,10 @@
     my $dirh_toplevel;
     opendir ($dirh_toplevel, $build_path) || die "Can't open '$build_path': $!";
     while (my $subdir = readdir ($dirh_toplevel)) {
+	$subdir =~ m/\./ && next; # eg. vcl.old,
 	my $test = "$build_path/$subdir/$target";
-	-d $test && push @modules, $test;
+	-d $test || next;
+	push @modules, $test;
     }
     closedir ($dirh_toplevel);
 
