Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.26
diff -u -r1.26 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	30 Mar 2005 09:11:10 -0000	1.26
+++ vcl/unx/gtk/window/gtkframe.cxx	6 Apr 2005 15:56:19 -0000
@@ -729,8 +729,12 @@
     
     for( nIndex = 0; nIndex < sizeof(nOffsets)/ sizeof(USHORT); nIndex++ )
     {
+		ResMgr *pResMgr = ImplGetResMgr();
+		if( !pResMgr )
+			g_error( "Can't locate resource manager, if you're using linkoo - did "
+					 "you remember to source ./ooenv before running ./soffice.bin ?" );
         // #i44723# workaround gcc temporary problem
-        ResId aResId( nOffsets[nIndex] + nIcon, ImplGetResMgr() );
+        ResId aResId( nOffsets[nIndex] + nIcon, pResMgr );
         BitmapEx aIcon( aResId );
 
         ImpBitmap *pIconImpBitmap = aIcon.ImplGetBitmapImpBitmap();
