Index: solenv/bin/linkoo
===================================================================
RCS file: /cvs/tools/solenv/bin/linkoo,v
retrieving revision 1.3
diff -u -r1.3 linkoo
--- solenv/bin/linkoo	23 Dec 2004 10:52:43 -0000	1.3
+++ solenv/bin/linkoo	31 Jan 2005 16:31:15 -0000
@@ -69,8 +69,8 @@
 export GNOME_DISABLE_CRASH_DIALOG=1
 ';
 
-my @exceptions = ( 'cppuhelper', 'configmgr', 'cfgmgr' );
+my @exceptions = ( 'cppuhelper', 'configmgr', 'cfgmgr', 'sunjavaplugin' );
 
 %replaceable = (
     'program' => '\.so$',
     'program/resource' => '\.res$',
