--- configmgr/source/treemgr/nodeimplobj.cxx.orig	2007-08-02 21:27:38.000000000 +0200
+++ configmgr/source/treemgr/nodeimplobj.cxx	2007-08-02 21:33:12.000000000 +0200
@@ -66,6 +66,8 @@
 #include "viewfactory.hxx"
 #endif
 
+#include <exception>
+
 namespace configmgr
 {
 	namespace configuration
--- configmgr/source/treemgr/setnodeimpl.cxx.orig	2007-08-02 21:31:12.000000000 +0200
+++ configmgr/source/treemgr/setnodeimpl.cxx	2007-08-02 21:32:47.000000000 +0200
@@ -104,6 +104,8 @@
 #define INCLUDED_VECTOR
 #endif
 
+#include <exception>
+
 namespace configmgr
 {
 	namespace configuration
--- psprint/source/printergfx/printerjob.cxx.orig	2007-08-02 22:07:05.000000000 +0200
+++ psprint/source/printergfx/printerjob.cxx	2007-08-02 22:09:48.000000000 +0200
@@ -75,6 +75,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <time.h>
 #include <pwd.h>
 #include <sal/alloca.h>
 
--- dtrans/source/X11/X11_selection.cxx.orig	2007-08-02 22:11:07.000000000 +0200
+++ dtrans/source/X11/X11_selection.cxx	2007-08-02 22:11:23.000000000 +0200
@@ -37,6 +37,7 @@
 #include "precompiled_dtrans.hxx"
 
 #include <unistd.h>
+#include <time.h>
 #include <stdio.h>
 #include <string.h>
 #include <sys/time.h>
