diff -u -r ../../../sfx2.bak/inc/sfxsids.hrc sfx2/inc/sfxsids.hrc
--- ../../../sfx2.bak/inc/sfxsids.hrc	2005-01-04 11:32:19.000000000 +0530
+++ sfx2/inc/sfxsids.hrc	2005-01-04 13:39:10.000000000 +0530
@@ -148,7 +148,9 @@
 #define SID_MAIL_NOTIFY                     (SID_SFX_START + 330)
 #define SID_MAIL_SENDDOC                    (SID_SFX_START + 331)
 #define	SID_MAIL_SENDDOCASPDF				(SID_SFX_START + 1672)
-
+#define       SID_MAIL_SENDDOCASMSDOC                 (SID_SFX_START + 1705)
+#define       SID_MAIL_SENDDOCASMSEXCEL               (SID_SFX_START + 1706)
+#define       SID_MAIL_SENDDOCASMSPOWERPOINT          (SID_SFX_START + 1707)
 
 #define SID_MAIL_RECIPIENT					(SID_SFX_START + 334)
 #define SID_MAIL_TEXT                       (SID_SFX_START + 335)
@@ -565,7 +567,7 @@
 #define SID_APPLY_STYLE                     (SID_SFX_START + 1703)
 #define SID_MACRO_SIGNATURE                 (SID_SFX_START + 1704)
 
-#define SID_SFX_free_START                  (SID_SFX_START + 1705)
+#define SID_SFX_free_START                  (SID_SFX_START + 1708)
 #define SID_SFX_free_END  					(SID_SFX_START + 3999)
 
 #define SID_OPEN_NEW_VIEW                   (SID_SFX_START + 520)
Only in sfx2/inc/: sfxsids.hrc.orig
Only in sfx2/inc/: sfxsids.hrc.rej
diff -u -r ../../../sfx2.bak/sdi/sfx.sdi sfx2/sdi/sfx.sdi
--- ../../../sfx2.bak/sdi/sfx.sdi	2005-01-04 11:32:19.000000000 +0530
+++ sfx2/sdi/sfx.sdi	2005-01-04 12:16:27.000000000 +0530
@@ -6340,7 +6340,6 @@
 	ToolBoxConfig = TRUE,
 	GroupId = GID_DOCUMENT;
 ]
-
 //--------------------------------------------------------------------------
 SfxBoolItem SendMail SID_MAIL_SENDDOC
 (SfxStringItem Recipient SID_MAIL_RECIPIENT,SfxStringItem Subject SID_MAIL_SUBJECT,SfxStringItem MailText SID_MAIL_TEXT,SfxUInt16Item Priority SID_MAIL_PRIORITY,SfxBoolItem AttachedDoc SID_MAIL_ATTACH,SfxStringItem AttachFiles SID_MAIL_ATTACH_FILE)
@@ -6368,7 +6367,88 @@
 	ToolBoxConfig = TRUE,
 	GroupId = GID_DOCUMENT;
 ]
+SfxBoolItem SendMailDocAsMSDoc SID_MAIL_SENDDOCASMSDOC
+(SfxStringItem Recipient SID_MAIL_RECIPIENT,SfxStringItem Subject SID_MAIL_SUBJECT,SfxStringItem MailText SID_MAIL_TEXT,SfxUInt16Item Priority SID_MAIL_PRIORITY,SfxBoolItem AttachedDoc SID_MAIL_ATTACH,SfxStringItem AttachFiles SID_MAIL_ATTACH_FILE)
+[
+	/* flags: */
+	AutoUpdate = FALSE,
+	Cachable = Cachable,
+	FastCall = FALSE,
+	HasCoreId = FALSE,
+	HasDialog = TRUE,
+	ReadOnlyDoc = TRUE,
+	Toggle = FALSE,
+	Container = FALSE,
+	RecordAbsolute = FALSE,
+	RecordPerSet;
+	Asynchron;
+
+	/* status: */
+	SlotType = SfxStringItem
 
+	/* config: */
+	AccelConfig = TRUE,
+	MenuConfig = TRUE,
+	StatusBarConfig = FALSE,
+	ToolBoxConfig = TRUE,
+	GroupId = GID_DOCUMENT;
+]
+//--------------------------------------------------------------------------
+SfxBoolItem SendMailDocAsMSExcel SID_MAIL_SENDDOCASMSEXCEL
+(SfxStringItem Recipient SID_MAIL_RECIPIENT,SfxStringItem Subject SID_MAIL_SUBJECT,SfxStringItem MailText SID_MAIL_TEXT,SfxUInt16Item Priority SID_MAIL_PRIORITY,SfxBoolItem AttachedDoc SID_MAIL_ATTACH,SfxStringItem AttachFiles SID_MAIL_ATTACH_FILE)
+[
+        /* flags: */
+        AutoUpdate = FALSE,
+        Cachable = Cachable,
+        FastCall = FALSE,
+        HasCoreId = FALSE,
+        HasDialog = TRUE,
+        ReadOnlyDoc = TRUE,
+        Toggle = FALSE,
+        Container = FALSE,
+        RecordAbsolute = FALSE,
+        RecordPerSet;
+        Asynchron;
+                                                                                                    
+        /* status: */
+        SlotType = SfxStringItem
+                                                                                                    
+        /* config: */
+        AccelConfig = TRUE,
+        MenuConfig = TRUE,
+        StatusBarConfig = FALSE,
+        ToolBoxConfig = TRUE,
+        GroupId = GID_DOCUMENT;
+]
+
+//--------------------------------------------------------------------------
+SfxBoolItem SendMailDocAsMSPowerPoint SID_MAIL_SENDDOCASMSPOWERPOINT
+(SfxStringItem Recipient SID_MAIL_RECIPIENT,SfxStringItem Subject SID_MAIL_SUBJECT,SfxStringItem MailText SID_MAIL_TEXT,SfxUInt16Item Priority SID_MAIL_PRIORITY,SfxBoolItem AttachedDoc SID_MAIL_ATTACH,SfxStringItem AttachFiles SID_MAIL_ATTACH_FILE)
+[
+        /* flags: */
+        AutoUpdate = FALSE,
+        Cachable = Cachable,
+        FastCall = FALSE,
+        HasCoreId = FALSE,
+        HasDialog = TRUE,
+        ReadOnlyDoc = TRUE,
+        Toggle = FALSE,
+        Container = FALSE,
+        RecordAbsolute = FALSE,
+        RecordPerSet;
+        Asynchron;
+                                                                                                    
+        /* status: */
+        SlotType = SfxStringItem
+                                                                                                    
+        /* config: */
+        AccelConfig = TRUE,
+        MenuConfig = TRUE,
+        StatusBarConfig = FALSE,
+        ToolBoxConfig = TRUE,
+        GroupId = GID_DOCUMENT;
+]
+//--------------------------------------------------------------------------
 SfxBoolItem SendMailDocAsPDF SID_MAIL_SENDDOCASPDF
 (SfxStringItem Recipient SID_MAIL_RECIPIENT,SfxStringItem Subject SID_MAIL_SUBJECT,SfxStringItem MailText SID_MAIL_TEXT,SfxUInt16Item Priority SID_MAIL_PRIORITY,SfxBoolItem AttachedDoc SID_MAIL_ATTACH,SfxStringItem AttachFiles SID_MAIL_ATTACH_FILE)
 [
Only in sfx2/sdi/: sfx.sdi.orig
diff -u -r ../../../sfx2.bak/sdi/sfxslots.src sfx2/sdi/sfxslots.src
--- ../../../sfx2.bak/sdi/sfxslots.src	2005-01-04 11:32:19.000000000 +0530
+++ sfx2/sdi/sfxslots.src	2005-01-04 13:42:26.000000000 +0530
@@ -575,6 +575,21 @@
 	Slotname [ de ] = "Dokument versenden";
 	Slotname [ en-US ] = "Send Document";
 };
+SfxSlotInfo SID_MAIL_SENDDOCASMSDOC
+{
+        SlotName  =  "Send document as MS-DOC Attachment";
+        SlotName [ en-US ] =  "Send document as MS-DOC Attachment";
+};
+SfxSlotInfo SID_MAIL_SENDDOCASMSEXCEL
+{
+        SlotName  =  "Send document as MS-EXCEL Attachment";
+        SlotName [ en-US ] =  "Send document as MS-EXCEL Attachment";
+};
+SfxSlotInfo SID_MAIL_SENDDOCASMSPOWERPOINT
+{
+        SlotName  =  "Send document as MS-POWERPOINT Attachment";
+        SlotName [ en-US ] =  "Send document as MS-POWERPOINT Attachment";
+};
 SfxSlotInfo SID_MAIL_SENDDOCASPDF
 {
 	Slotname [ de ] = "Dokument als PDF-Anhang versenden";
Only in sfx2/sdi/: sfxslots.src.orig
diff -u -r ../../../sfx2.bak/sdi/viwslots.sdi sfx2/sdi/viwslots.sdi
--- ../../../sfx2.bak/sdi/viwslots.sdi	2005-01-04 11:32:19.000000000 +0530
+++ sfx2/sdi/viwslots.sdi	2005-01-04 12:16:27.000000000 +0530
@@ -22,6 +22,21 @@
 		ExecMethod = ExecMisc_Impl ;
 		StateMethod = GetState_Impl ;
 	]
+	SID_MAIL_SENDDOCASMSDOC // ole(no) api(todo)
+	[
+		ExecMethod = ExecMisc_Impl ;
+		StateMethod = GetState_Impl ;
+	]
+	SID_MAIL_SENDDOCASMSEXCEL // ole(no) api(todo)
+	[
+		ExecMethod = ExecMisc_Impl ;
+		StateMethod = GetState_Impl ;
+	]
+	SID_MAIL_SENDDOCASMSPOWERPOINT // ole(no) api(todo)
+	[
+		ExecMethod = ExecMisc_Impl ;
+		StateMethod = GetState_Impl ;
+	]
 	 //---------------------------------------------------------------------
 	SID_PRINTDOC // ole(no) api(play/rec)
 	[
diff -u -r ../../../sfx2.bak/source/dialog/mailmodel.cxx sfx2/source/dialog/mailmodel.cxx
--- ../../../sfx2.bak/source/dialog/mailmodel.cxx	2005-01-04 11:32:21.000000000 +0530
+++ sfx2/source/dialog/mailmodel.cxx	2005-01-04 13:36:26.000000000 +0530
@@ -349,7 +349,7 @@
 	return eRet;
 }
 
-SfxMailModel_Impl::SaveResult SfxMailModel_Impl::SaveDocAsPDF( String& rFileName, String& rType )
+SfxMailModel_Impl::SaveResult SfxMailModel_Impl::SaveDocAs( String& rFileName, String& rType,MailDocType mailDocType )
 {
 	SaveResult eRet = SAVE_CANCELLED;
 	SfxViewFrame* pTopViewFrm = mpBindings->GetDispatcher_Impl()->GetFrame()->GetTopViewFrame();
@@ -366,9 +366,16 @@
 
 		// Get PDF Filter from document
 		SfxFilterMatcher aMatcher( String::CreateFromAscii(xDocShell->GetFactory().GetShortName()) );
-		String aPDFExtension = String::CreateFromAscii( ".pdf" );
+		String aExtension ;
+                switch (mailDocType){
+	                case TYPE_ASPDF   : aExtension = String::CreateFromAscii( ".pdf" );break;
+        	        case TYPE_ASMSDOC : aExtension = String::CreateFromAscii( ".doc" );break;
+                	case TYPE_ASMSEXCEL : aExtension = String::CreateFromAscii( ".xls" );break;
+	                case TYPE_ASMSPOWERPOINT : aExtension = String::CreateFromAscii( ".ppt" );break;
+                }
+
 
-		const SfxFilter*	pFilter		= aMatcher.GetFilter4Extension( aPDFExtension, SFX_FILTER_EXPORT );
+		const SfxFilter*	pFilter		= aMatcher.GetFilter4Extension( aExtension, SFX_FILTER_EXPORT );
 		sal_Bool			bHasFilter	= pFilter ? sal_True : sal_False;
 
 		// create temp file name with leading chars and extension
@@ -384,7 +391,7 @@
 			String aName;
 			if ( aFileObj.hasExtension() )
 			{
-				pExt = new String( aPDFExtension );
+				pExt = new String( aExtension );
 				aFileObj.removeExtension();
 				aLeadingStr = aFileObj.getName( INetURLObject::LAST_SEGMENT, true, INetURLObject::DECODE_WITH_CHARSET );
 				aLeadingStr += String::CreateFromAscii( "_" );
@@ -407,11 +414,25 @@
 		delete pExt;
 
 		rFileName = aTempFile.GetURL();
+                SfxBoolItem aPicklist( SID_PICKLIST, FALSE );
+                SfxBoolItem aSaveTo( SID_SAVETO, TRUE );
+                SfxStringItem* pFilterName = NULL;
+                if ( pFilter && bHasFilter )
+              pFilterName = new SfxStringItem( SID_FILTER_NAME, pFilter->GetFilterName() );
+ 
+                SfxStringItem* pPassItem = NULL;
+                String aPasswd;
+                if ( GetPasswd_Impl( xDocShell->GetMedium()->GetItemSet(), aPasswd ) )
+                        pPassItem = new SfxStringItem( SID_PASSWORD, aPasswd );
 
 		// save document to temp file
 		SfxStringItem aFileName( SID_FILE_NAME, rFileName );
-		const SfxBoolItem *pRet = (const SfxBoolItem*)pDisp->Execute( SID_EXPORTDOCASPDF, SFX_CALLMODE_SYNCHRON, &aFileName, 0L );
-		BOOL bRet = pRet ? pRet->GetValue() : FALSE;
+		const SfxBoolItem *pRet = (const SfxBoolItem*)pDisp->Execute( SID_SAVEASDOC, SFX_CALLMODE_SYNCHRON, &aFileName, &aPicklist, &aSaveTo,
+                 pFilterName ? pFilterName : pPassItem,
+                 pFilterName ? pPassItem : 0L, 0L );
+ 
+  /*            const SfxBoolItem *pRet = (const SfxBoolItem*)pDisp->Execute( SID_EXPORTDOCASPDF, SFX_CALLMODE_SYNCHRON, &aFileName, 0L );
+  */            BOOL bRet = pRet ? pRet->GetValue() : FALSE;
 		eRet = bRet ? SAVE_SUCCESSFULL : SAVE_CANCELLED;
 
 		if ( pFilter )
@@ -509,11 +530,16 @@
 	String aFileName, aContentType;
 
 	sal_Bool bSuccessfull = sal_False;
-	if ( eMailDocType == TYPE_SELF )
-		eSaveResult = SaveDocument( aFileName, aContentType );
-	else
-		eSaveResult = SaveDocAsPDF( aFileName, aContentType );
-
+	switch (eMailDocType){
+	    case TYPE_ASMSDOC:
+	    case TYPE_ASMSPOWERPOINT:
+	    case TYPE_ASPDF:
+	    case TYPE_ASMSEXCEL:
+			 	eSaveResult = SaveDocAs( aFileName, aContentType,eMailDocType );
+				break;
+	    default:
+				eSaveResult = SaveDocument( aFileName, aContentType);
+	} 
     if ( eSaveResult == SAVE_SUCCESSFULL )
 	{
     	SfxFrame* pViewFrm = mpBindings->GetDispatcher_Impl()->GetFrame()->GetFrame();
Only in sfx2/source/dialog: mailmodel.cxx.orig
Only in sfx2/source/dialog: mailmodel.cxx.rej
diff -u -r ../../../sfx2.bak/source/inc/mailmodel.hxx sfx2/source/inc/mailmodel.hxx
--- ../../../sfx2.bak/source/inc/mailmodel.hxx	2005-01-04 11:32:20.000000000 +0530
+++ sfx2/source/inc/mailmodel.hxx	2005-01-04 12:16:27.000000000 +0530
@@ -90,6 +90,9 @@
 	enum MailDocType
 	{
 		TYPE_SELF,
+		TYPE_ASMSDOC,
+		TYPE_ASMSEXCEL,
+		TYPE_ASMSPOWERPOINT,
 		TYPE_ASPDF
 	};
 
@@ -114,7 +117,7 @@
 	void				ClearList( AddressList_Impl* pList );
 	void				MakeValueList( AddressList_Impl* pList, String& rValueList );
 	SaveResult			SaveDocument( String& rFileName, String& rType );
-	SaveResult			SaveDocAsPDF( String& rFileName, String& rType );
+	SaveResult			SaveDocAs( String& rFileName, String& rType,MailDocType mailDocType );
 
 	DECL_LINK( DoneHdl, void* );
 
diff -u -r ../../../sfx2.bak/source/view/viewsh.cxx sfx2/source/view/viewsh.cxx
--- ../../../sfx2.bak/source/view/viewsh.cxx	2005-01-04 11:32:20.000000000 +0530
+++ sfx2/source/view/viewsh.cxx	2005-01-04 14:13:17.000000000 +0530
@@ -191,6 +191,9 @@
 
 		case SID_MAIL_SENDDOCASPDF:
 		case SID_MAIL_SENDDOC:
+                case SID_MAIL_SENDDOCASMSDOC:
+                case SID_MAIL_SENDDOCASMSEXCEL:
+                case SID_MAIL_SENDDOCASMSPOWERPOINT:
 		{
 			SfxObjectShell* pDoc = GetObjectShell();
 			if ( pDoc && pDoc->QueryHiddenInformation(
@@ -221,10 +224,18 @@
 				}
 
 				SfxMailModel_Impl::SendMailResult eResult = SfxMailModel_Impl::SEND_MAIL_ERROR;
-				if ( nId == SID_MAIL_SENDDOCASPDF )
-					eResult = aModel.Send( SfxMailModel_Impl::TYPE_ASPDF );
-				else
-					eResult = aModel.Send( SfxMailModel_Impl::TYPE_SELF );
+				switch (nId) {
+					case SID_MAIL_SENDDOCASPDF: 
+						 eResult = aModel.Send( SfxMailModel_Impl::TYPE_ASPDF );break;
+					case SID_MAIL_SENDDOCASMSDOC:
+						 eResult = aModel.Send( SfxMailModel_Impl::TYPE_ASMSDOC );break;
+					case SID_MAIL_SENDDOCASMSEXCEL:  
+						eResult = aModel.Send( SfxMailModel_Impl::TYPE_ASMSEXCEL );break;
+					case SID_MAIL_SENDDOCASMSPOWERPOINT: 
+						 eResult = aModel.Send( SfxMailModel_Impl::TYPE_ASMSPOWERPOINT );break;
+					default:
+						eResult = aModel.Send( SfxMailModel_Impl::TYPE_SELF );
+				}
 
 				if ( eResult == SfxMailModel_Impl::SEND_MAIL_ERROR )
 				{
@@ -333,6 +344,9 @@
 			// Mail-Funktionen
 			case SID_MAIL_SENDDOCASPDF:
 			case SID_MAIL_SENDDOC:
+			case SID_MAIL_SENDDOCASMSDOC:
+			case SID_MAIL_SENDDOCASMSEXCEL:
+			case SID_MAIL_SENDDOCASMSPOWERPOINT:
 			{
                 BOOL bEnable = !GetViewFrame()->HasChildWindow( SID_MAIL_CHILDWIN );
 				if ( !bEnable )
@@ -353,6 +367,60 @@
 					rSet.DisableItem( nSID );
 				}
 
+				else if ( nSID == SID_MAIL_SENDDOCASMSDOC )
+				{
+					SfxObjectShellRef xDocShell = GetViewFrame()->GetObjectShell();
+					if ( xDocShell.Is() )
+					{
+						// Get PDF Filter from container
+				                SfxFilterMatcher aMatcher( String::CreateFromAscii(xDocShell->GetFactory().GetShortName()) );
+						//if ( pFilterContainer )
+						{
+							String aPDFExtension = String::CreateFromAscii( ".doc" );
+							const SfxFilter*        pFilter         = aMatcher.GetFilter4Extension( aPDFExtension, SFX_FILTER_EXPORT );
+							if ( pFilter != NULL )
+								break;
+						}
+					}
+
+					rSet.DisableItem( nSID );
+				}
+				else if ( nSID == SID_MAIL_SENDDOCASMSEXCEL )
+				{
+					SfxObjectShellRef xDocShell = GetViewFrame()->GetObjectShell();
+					if ( xDocShell.Is() )
+					{
+						// Get PDF Filter from container
+						 SfxFilterMatcher aMatcher( String::CreateFromAscii(xDocShell->GetFactory().GetShortName()) );
+						// if ( pFilterContainer )
+						{
+							String aPDFExtension = String::CreateFromAscii( ".xls" );
+							const SfxFilter*        pFilter         = aMatcher.GetFilter4Extension( aPDFExtension, SFX_FILTER_EXPORT );
+							if ( pFilter != NULL )
+								break;
+						}
+					}
+
+					rSet.DisableItem( nSID );
+				}
+				else if ( nSID == SID_MAIL_SENDDOCASMSPOWERPOINT )
+				{
+					SfxObjectShellRef xDocShell = GetViewFrame()->GetObjectShell();
+					if ( xDocShell.Is() )
+					{
+						// Get PDF Filter from container
+						SfxFilterMatcher aMatcher( String::CreateFromAscii(xDocShell->GetFactory().GetShortName()) );
+						//if ( pFilterContainer )
+						{
+							String aPDFExtension = String::CreateFromAscii( ".ppt" );
+							const SfxFilter*        pFilter         = aMatcher.GetFilter4Extension( aPDFExtension, SFX_FILTER_EXPORT );
+							if ( pFilter != NULL )
+								break;
+						}
+					}
+
+					rSet.DisableItem( nSID );
+				}
 				break;
 			}
 
Only in sfx2/source/view: viewsh.cxx.orig
Only in sfx2/source/view: viewsh.cxx.rej
diff -u -r ../../../svx.bak/inc/globlmn_tmpl.hrc svx/inc/globlmn_tmpl.hrc
--- ../../../svx.bak/inc/globlmn_tmpl.hrc	2005-01-04 11:32:48.000000000 +0530
+++ svx/inc/globlmn_tmpl.hrc	2005-01-05 10:59:10.666110552 +0530
@@ -580,6 +580,24 @@
     Text [ en-US ] = "~AutoMail" ; \
 	Text [ x-comment ] = " ";	\
 
+#define ITEM_FILE_MAIL_SENDDOCASMSDOC \
+    Identifier = SID_MAIL_SENDDOCASMSDOC ; \
+       Command = ".uno:SendMailDocAsMSDoc" ; \
+    HelpID = SID_MAIL_SENDDOCASMSDOC ; \
+    Text [ en-US ] = "Document as MS-~Doc Attachment..." ; \
+
+#define ITEM_FILE_MAIL_SENDDOCASMSEXCEL \
+    Identifier = SID_MAIL_SENDDOCASMSEXCEL ; \
+       Command = ".uno:SendMailDocAsMSExcel" ; \
+    HelpID = SID_MAIL_SENDDOCASMSEXCEL ; \
+    Text [ en-US ] = "Document as MS-~Excel Attachment..." ; \
+
+#define ITEM_FILE_MAIL_SENDDOCASMSPOWERPOINT \
+    Identifier = SID_MAIL_SENDDOCASMSPOWERPOINT ; \
+       Command = ".uno:SendMailDocAsMSPowerPoint" ; \
+    HelpID = SID_MAIL_SENDDOCASMSPOWERPOINT ; \
+    Text [ en-US ] = "Document as MS-~PPT Attachment..." ; \
+
 #define ITEM_FILE_MAIL_SENDDOC \
     Identifier = SID_MAIL_SENDDOC ; \
 	Command = ".uno:SendMail" ; \
diff -u -r ../../../sw.bak/uiconfig/swriter/menubar/menubar.xml sw/uiconfig/swriter/menubar/menubar.xml
--- ../../../sw.bak/uiconfig/swriter/menubar/menubar.xml	2005-01-04 10:28:19.000000000 +0530
+++ sw/uiconfig/swriter/menubar/menubar.xml	2005-01-05 11:16:17.643986184 +0530
@@ -22,6 +22,7 @@
 				<menu:menupopup>
 					<menu:menuitem menu:id=".uno:SendMail" menu:helpid="5331" menu:label="" />
 					<menu:menuitem menu:id=".uno:SendMailDocAsPDF" menu:helpid="6672" menu:label="" />
+					<menu:menuitem menu:id=".uno:SendMailDocAsMSDoc" menu:helpid="6705" menu:label="" /
 					<menu:menuseparator/>
 					<menu:menuitem menu:id=".uno:NewGlobalDoc" menu:helpid="20004" menu:label="" />
 					<menu:menuitem menu:id=".uno:NewHtmlDoc" menu:helpid="20040" menu:label="" />
Only in sc/uiconfig/scalc/accelerator/en-US: default.xml.orig
diff -u -r ../../../sc.bak/uiconfig/scalc/menubar/menubar.xml sc/uiconfig/scalc/menubar/menubar.xml
--- ../../../sc.bak/uiconfig/scalc/menubar/menubar.xml	2004-12-22 14:05:07.000000000 +0530
+++ sc/uiconfig/scalc/menubar/menubar.xml	2005-01-05 11:16:41.045428624 +0530
@@ -22,6 +22,7 @@
 				<menu:menupopup>
 					<menu:menuitem menu:id=".uno:SendMail" menu:helpid="5331" menu:label="" />
 					<menu:menuitem menu:id=".uno:SendMailDocAsPDF" menu:helpid="6672" menu:label="" />
+					<menu:menuitem menu:id=".uno:SendMailDocAsMSExcel" menu:helpid="6706" menu:label="" />
 				</menu:menupopup>
 			</menu:menu>
 			<menu:menuseparator/>
diff -u -r ../../../sd.bak/uiconfig/simpress/menubar/menubar.xml sd/uiconfig/simpress/menubar/menubar.xml
--- ../../../sd.bak/uiconfig/simpress/menubar/menubar.xml	2005-01-03 10:48:26.000000000 +0530
+++ sd/uiconfig/simpress/menubar/menubar.xml	2005-01-05 11:27:07.242232240 +0530
@@ -22,6 +22,7 @@
 				<menu:menupopup>
 					<menu:menuitem menu:id=".uno:SendMail" menu:helpid="5331" menu:label="" />
 					<menu:menuitem menu:id=".uno:SendMailDocAsPDF" menu:helpid="6672" menu:label="" />
+					<menu:menuitem menu:id=".uno:SendMailDocAsMSPowerPoint" menu:helpid="6707" menu:label="" />
 				</menu:menupopup>
 			</menu:menu>
 			<menu:menuseparator/>
