Index: sal/osl/unx/file.cxx
===================================================================
RCS file: /cvs/porting/sal/osl/unx/file.cxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 file.cxx
--- sal/osl/unx/file.cxx	31 May 2005 17:07:41 -0000	1.8
+++ sal/osl/unx/file.cxx	23 Aug 2005 16:42:43 -0000
@@ -650,6 +652,8 @@ oslFileError osl_openFile( rtl_uString* 
             fd = open( buffer, flags, mode );
             if ( fd >= 0 )
             {
+				sal_Bool got_flock;
+
                 /* check if file lock is enabled and clear l_type member of flock otherwise */
                 if( (char *) -1 == pFileLockEnvVar )
                 {
@@ -663,8 +667,21 @@ oslFileError osl_openFile( rtl_uString* 
                 if( NULL == pFileLockEnvVar )
                     aflock.l_type = 0;
 
+				got_flock = sal_False;
                 /* lock the file if flock.l_type is set */
-                if( F_WRLCK != aflock.l_type || -1 != fcntl( fd, F_SETLK, &aflock ) )
+                if( F_WRLCK != aflock.l_type)
+					got_flock = sal_True;
+				if( -1 != fcntl( fd, F_SETLK, &aflock ) )
+					got_flock = sal_True;
+				else if( errno == EAGAIN )
+					errno = EACCES; /* misleading synonym */
+				else if( errno == ENOLCK )
+				{ /* locking not supported */
+					errno = 0;
+					got_flock = sal_True;
+				}
+
+				if (got_flock)
                 {
                     /* allocate memory for impl structure */
                     pHandleImpl = (oslFileHandleImpl*) rtl_allocateMemory( sizeof(oslFileHandleImpl) );
