diff -u -p -u -r1.44.16.7 ooo.src
--- svx/source/intro/ooo.src
+++ svx/source/intro/ooo.src
@@ -75,10 +75,10 @@
 #include "svxids.hrc"
 #include "intro.hrc"
 
-#ifdef BUILD_SPECIAL
-#define OOO_VENDOR "Sun Microsystems Inc."
+#if 1
+#define OOO_VENDOR "Novell, Inc."
 #define OOO_LICENSE "LGPL"
-#define OOO_INTRO "openintro_sun.bmp"
+#define OOO_INTRO "openintro_nld.bmp"
 #else
 //#define OOO_VENDOR "my company"
 #define OOO_LICENSE "LGPL"
@@ -87,7 +87,7 @@
 
 Bitmap RID_DEFAULT_ABOUT_BMP_LOGO
 {
-	File = "openabout.bmp" ;
+	File = "openabout_nld.bmp" ;
 };
 
 Bitmap RID_DEFAULTINTRO
