Index: svx/prj/d.lst
===================================================================
RCS file: /cvs/graphics/svx/prj/d.lst,v
retrieving revision 1.108
diff -u -r1.108 d.lst
--- svx/prj/d.lst	9 Dec 2004 16:36:45 -0000	1.108
+++ svx/prj/d.lst	16 Dec 2004 12:43:47 -0000
@@ -55,7 +55,9 @@
 ..\res\cjkintro.bmp %COMMON_DEST%\pck%_EXT%\starsuite_bitmap\intro.bmp
 ..\res\ofaabout.bmp %COMMON_DEST%\pck%_EXT%\staroffice_bitmap\about.bmp
 ..\res\soffice.bmp %COMMON_DEST%\pck%_EXT%\staroffice_bitmap\intro.bmp
-..\res\ooointro.bmp %COMMON_DEST%\pck%_EXT%\openoffice\intro.bmp
+..\res\openintro_suse.bmp %COMMON_DEST%\pck%_EXT%\openoffice\intro.bmp
+..\res\openintro_nld.bmp %COMMON_DEST%\pck%_EXT%\openoffice\intro-nld.bmp
+..\res\openabout_nld.bmp %COMMON_DEST%\pck%_EXT%\openoffice\about.bmp
 ..\res\ooonologointro.bmp %COMMON_DEST%\pck%_EXT%\openoffice\nologointro.bmp
 
 mkdir: %_DEST%\inc%_EXT%\svx

--- svx/source/intro/makefile.mk	2005-12-20 15:28:19.000000000 +0100
+++ svx/source/intro/makefile.mk	2005-12-20 15:28:35.000000000 +0100
@@ -46,7 +46,7 @@ TARGET=intro
 RSCLOCINC!:=$(RSCLOCINC);$(PRJ)$/RES
 
 # setting the compiled by $user$ string in the About box
-OOO_VENDOR*="$(USER)$(USERNAME)"
+OOO_VENDOR*="Novell, Inc."
 RSCDEFS+=-DOOO_VENDOR=$(OOO_VENDOR)
 
 # --- Allgemein -----------------------------------------------------------
Index: svx/source/intro/ooo.src
===================================================================
RCS file: /cvs/graphics/svx/source/intro/ooo.src,v
retrieving revision 1.11
diff -u -r1.11 ooo.src
--- svx/source/intro/ooo.src	7 Jan 2005 09:49:30 -0000	1.11
+++ svx/source/intro/ooo.src	1 Feb 2005 16:48:49 -0000
@@ -50,7 +50,7 @@
 
 Bitmap RID_DEFAULT_ABOUT_BMP_LOGO
 {
-	File = "openabout.bmp" ;
+	File = "openabout_suse.bmp" ;
 };
 
 String RID_APPTITLE


This piece of the patch is not necessary for people doing stock
re-branding, but it allows us to package our enterprise branding as
2 separate images in an RPM.


Index: desktop/source/splash/splash.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/splash/splash.cxx,v
retrieving revision 1.15
diff -u -p -u -r1.15 splash.cxx
--- desktop/source/splash/splash.cxx
+++ desktop/source/splash/splash.cxx
@@ -334,6 +335,7 @@ void SplashScreen::initBitmap()
 	{
 		xub_StrLen nIndex = 0;
 		String aBmpFileName( DEFINE_CONST_UNICODE("intro.bmp") );
+		String aNLDBmpFileName( DEFINE_CONST_UNICODE("intro-nld.bmp") );
 
         bool haveBitmap = false;
 
@@ -361,6 +363,22 @@ void SplashScreen::initBitmap()
             }
         }
 
+        // Then, look for NLD branded bitmap
+        if ( !haveBitmap )
+        {
+            INetURLObject aObj( _sExecutePath, INET_PROT_FILE );
+            aObj.insertName( aNLDBmpFileName );
+
+            SvFileStream aStrm( aObj.PathToFileName(), STREAM_STD_READ );
+            if ( !aStrm.GetError() )
+            {
+                // Default case, we load the intro bitmap from a seperate file
+                // (e.g. staroffice_intro.bmp or starsuite_intro.bmp)
+                aStrm >> _aIntroBmp;
+                haveBitmap = true;
+            }
+        }
+
         // Then, try to use bitmap located in the same directory as the executable.
         if ( !haveBitmap )
         {


--- scp2/source/ooo/file_ooo.scp
+++ scp2/source/ooo/file_ooo.scp
@@ -437,6 +437,20 @@
     Styles = (PACKED);
 End
 
+File gid_File_Bmp_Intro_Nld
+    TXT_FILE_BODY;
+    Dir = gid_Dir_Program;
+    Name = "intro-nld.bmp";
+    Styles = (PACKED);
+End
+
+File gid_File_Bmp_About_Nld
+    TXT_FILE_BODY;
+    Dir = gid_Dir_Program;
+    Name = "about.bmp";
+    Styles = (PACKED);
+End
+
 File gid_File_Dtd_Accelerator
     TXT_FILE_BODY;
     Styles = (PACKED);
