--- odk/setsdkenv_unix.sh.in.old	2005-11-09 10:29:22.000000000 +0100
+++ odk/setsdkenv_unix.sh.in	2006-11-15 16:05:58.000000000 +0100
@@ -3,11 +3,16 @@
 # This script starts a new shell and sets all enviroment variables, which
 # are necessary for building the examples of the Office Development Kit.
 # The Script was developed for the operating systems Solaris and Linux.
-sdkscript=`echo $0 | grep setsdkenv_unix`
+sdkscript=`echo $0 | grep setsdkenv_unix | true`
 if [ "$sdkscript" = "" ]
 then
 sdkscript=`history 1 | sed "s/.*\. //"`
 fi
+if [ "$sdkscript" = "" ]
+then
+# total falback; needed when history does not work
+sdkscript=@OO_SDK_HOME@/setsdkenv_unix
+fi
 sdkdir=`dirname $sdkscript`
 
 # The SDK name
