Index: odk/pack/copying/makefile.mk
===================================================================
RCS file: /cvs/api/odk/pack/copying/makefile.mk,v
retrieving revision 1.72
diff -u -u -r1.72 makefile.mk
--- odk/pack/copying/makefile.mk	29 Mar 2005 12:18:55 -0000	1.72
+++ odk/pack/copying/makefile.mk	17 May 2005 17:24:11 -0000
@@ -293,7 +293,6 @@
 	$(DESTIDLLIST)  \
 	$(DESTINCLUDELIST) \
 	$(DESTCLASSESLIST) \
-	$(DESTDIRJAR)$/win$/unowinreg.dll \
-	$(DESTDIRDOCU)$/common$/spec$/xml_format$/xml_specification.pdf
+	$(DESTDIRJAR)$/win$/unowinreg.dll
 
 DIR_DIRECTORY_LIST=$(uniq $(DIR_FILE_LIST:d))
Index: odk/util/check.pl
===================================================================
RCS file: /cvs/api/odk/util/check.pl,v
retrieving revision 1.16
diff -u -u -r1.16 check.pl
--- odk/util/check.pl	29 Mar 2005 12:19:55 -0000	1.16
+++ odk/util/check.pl	17 May 2005 17:25:46 -0000
@@ -153,14 +153,6 @@
     }
     print "\n";
 
-    # check file format specification
-    print "check xml format specifiaction: ";
-    if (! -e "$StartDir/docs/common/spec/xml_format/xml_specification.pdf") {
-	print "\nERROR: \"$StartDir/docs/common/spec/xml_format/xml_specification.pdf\" is missing\n";
-	$return++;
-    }
-    print "\n";
-
     #check configure files
     print "check config files: ";
     if ($OperatingSystem eq "windows") {
Index: odk/util/odk_rules.pmk
===================================================================
RCS file: /cvs/api/odk/util/odk_rules.pmk,v
retrieving revision 1.13
diff -u -u -r1.13 odk_rules.pmk
--- odk/util/odk_rules.pmk	7 Mar 2005 17:31:28 -0000	1.13
+++ odk/util/odk_rules.pmk	17 May 2005 17:27:05 -0000
@@ -315,10 +315,6 @@
 	$(MY_TEXTCOPY) $(MY_TEXTCOPY_SOURCEPRE) $< $(MY_TEXTCOPY_TARGETPRE) $@
 .ENDIF
 
-$(DESTDIRCOMMONDOCU)$/spec$/xml_format$/xml_specification.pdf : xml_specification.pdf
-	+-rm -f $@ >& $(NULLDEV)
-	$(GNUCOPY) xml_specification.pdf $@
-        
 $(DESTDIRXML)$/module-description.dtd : $(XMLOUT)$/module-description.dtd
 	+-rm -f $@ >& $(NULLDEV)
 	$(MY_TEXTCOPY) $(MY_TEXTCOPY_SOURCEPRE) $(XMLOUT)$/module-description.dtd $(MY_TEXTCOPY_TARGETPRE) $@  
