? offapi/offapi.vpj
Index: offapi/com/sun/star/sheet/DataPilotFieldFilter.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/sheet/DataPilotFieldFilter.idl,v
retrieving revision 1.2
retrieving revision 1.1.2.3
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.2 -r1.1.2.3
--- offapi/com/sun/star/sheet/DataPilotFieldFilter.idl	26 Nov 2007 15:22:52 -0000	1.2
+++ offapi/com/sun/star/sheet/DataPilotFieldFilter.idl	7 Dec 2007 05:39:58 -0000	1.1.2.3
@@ -42,12 +42,34 @@
 
 //=============================================================================
 
+/** A single filtering condition used when constructing a drill-down sheet.
+
+   <p>Each instance of <type>DataPilotFieldFilter</type> represents a filtering 
+   condition.  It is used when evaluating a row in a data source whether or not 
+   to include it in a filtered subset of the original data.  For example, for 
+   a given row to be included in the output of a drill-down data table, the row 
+   must match every single instance of <type>DataPilotFieldFilter</type> to be 
+   qualified.  If the field specified by the value of <member>FieldName</member>
+   is not found, then that filter condition will not be evaluated.</p>
+
+   <p>Note that this struct is typically used in a sequence and is not normally 
+   used as a single instance.</p>
+
+   @see <method>XDrillDownDataSupplier::getDrillDownData</method>
+   @see <type>DataPilotTableResultData</type>
+
+   @since OOo 3.0.0
+ */
 struct DataPilotFieldFilter
 {
-    /** Field name. */
+    /** Field name.
+
+        @see <type>DataPilotField</type>
+     */
     string FieldName;
 
-    /** String value that needs to match against. */
+    /** String value to match against.
+     */
     string MatchValue;
 };
 
Index: offapi/com/sun/star/sheet/DataPilotOutputRangeType.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/DataPilotOutputRangeType.idl
diff -N offapi/com/sun/star/sheet/DataPilotOutputRangeType.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/DataPilotOutputRangeType.idl	7 Dec 2007 05:29:36 -0000	1.1.2.3
@@ -0,0 +1,80 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: DataPilotOutputRangeType.idl,v $
+ *
+ *  $Revision: 1.1.2.3 $
+ *
+ *  last change: $Author: kohei $ $Date: 2007/12/07 05:29:36 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_DataPilotOutputRangeType_idl__
+#define __com_sun_star_sheet_DataPilotOutputRangeType_idl__
+
+module com { module sun { module star { module sheet {
+
+//============================================================================
+
+/** specifies region type of DataPilot table range
+
+    <p>This constant set is used to indicate the type of output range desired when
+    <method>XDataPilotTable2::getOutputRangeByType</method> is called, which 
+    returns a different cell range depending upon the value passed to it as the argument.</p>
+
+    @see com::sun::star::sheet::XDataPilotTable2
+
+    @since OOo 3.0.0
+ */
+constants DataPilotOutputRangeType
+{
+    //------------------------------------------------------------------------
+
+    /** whole DataPilot output range including the header area above the table
+        where the filter and page field buttons are located. */
+    const long WHOLE = 0;
+
+    //------------------------------------------------------------------------
+
+    /** whole table but without the header area where the filter and page field 
+        buttons are located. */
+    const long TABLE = 1;
+
+    //------------------------------------------------------------------------
+
+    /** result area where the result values are displayed.  This also includes
+        the column and row subtotal areas when they are displayed. */
+    const long RESULT = 2;
+};
+
+//============================================================================
+
+}; }; }; };
+
+
+
+#endif
Index: offapi/com/sun/star/sheet/DataPilotSource.idl
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/sheet/DataPilotSource.idl,v
retrieving revision 1.6
retrieving revision 1.6.404.3
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.6 -r1.6.404.3
--- offapi/com/sun/star/sheet/DataPilotSource.idl	8 Sep 2005 04:34:04 -0000	1.6
+++ offapi/com/sun/star/sheet/DataPilotSource.idl	6 Dec 2007 22:19:05 -0000	1.6.404.3
@@ -104,6 +104,29 @@
 	 */
 	[property] boolean RowGrand;
 
+    //-------------------------------------------------------------------------
+
+    /** specifies the number of row fields.
+
+        @since OOo 3.0.0
+     */
+    [readonly, property, optional] long RowFieldCount;
+
+    //-------------------------------------------------------------------------
+
+    /** specifies the number of column fields. 
+
+        @since OOo 3.0.0
+     */
+    [readonly, property, optional] long ColumnFieldCount;
+
+    //-------------------------------------------------------------------------
+
+    /** specifies the number of data fields. 
+
+        @since OOo 3.0.0
+     */
+    [readonly, property, optional] long DataFieldCount;
 };
 
 //=============================================================================
Index: offapi/com/sun/star/sheet/DataPilotTableHeaderData.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/DataPilotTableHeaderData.idl
diff -N offapi/com/sun/star/sheet/DataPilotTableHeaderData.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/DataPilotTableHeaderData.idl	7 Dec 2007 05:29:36 -0000	1.1.2.3
@@ -0,0 +1,87 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: DataPilotTableHeaderData.idl,v $
+ *
+ *  $Revision: 1.1.2.3 $
+ *
+ *  last change: $Author: kohei $ $Date: 2007/12/07 05:29:36 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_DataPilotTableHeaderData_idl__
+#define __com_sun_star_sheet_DataPilotTableHeaderData_idl__
+
+#include <com/sun/star/sheet/DataPilotFieldFilter.idl>
+#include <com/sun/star/sheet/DataResult.idl>
+
+//=============================================================================
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** information about a cell within the column or row header area of a DataPilot table.
+
+    This struct contains information about a particular cell located within the 
+    column or row header area of a DataPilot table.  This is the type that is
+    contained in <member>DataPilotTablePositionData::PositionData</member> when 
+    the value of <member>DataPilotTablePositionData::PositionType</member> is
+    either <const>DataPilotTablePositionType::ROW_HEADER</const> or 
+    <const>DataPilotTablePositionType::COLUMN_HEADER</const>.
+
+    @see com::sun::star::sheet::DataPilotTablePositionData
+    @see com::sun::star::sheet::DataPilotTablePositionType
+    @see com::sun::star::sheet::DataPilotFieldFilter
+    @see com::sun::star::sheet::DataResult
+
+    @since OOo 3.0.0
+ */
+struct DataPilotTableHeaderData
+{
+    /** number of dimensions */
+    long    Dimension;
+
+    /** hierarchy */
+    long    Hierarchy;
+
+    /** level */
+    long    Level;
+
+    /** flag */
+    long    Flags;
+
+    /** member name */
+    string  MemberName;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
+
Index: offapi/com/sun/star/sheet/DataPilotTablePositionData.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/DataPilotTablePositionData.idl
diff -N offapi/com/sun/star/sheet/DataPilotTablePositionData.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/DataPilotTablePositionData.idl	7 Dec 2007 05:29:36 -0000	1.1.2.5
@@ -0,0 +1,98 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: DataPilotTablePositionData.idl,v $
+ *
+ *  $Revision: 1.1.2.5 $
+ *
+ *  last change: $Author: kohei $ $Date: 2007/12/07 05:29:36 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_DataPilotTablePositionData_idl__
+#define __com_sun_star_sheet_DataPilotTablePositionData_idl__
+
+#include <com/sun/star/sheet/DataPilotFieldFilter.idl>
+
+//=============================================================================
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** This structure contains information on a cell within a DataPilot table.
+
+    <p>This structure contains information on a particular cell within a DataPilot
+    table, and is used to retrieve its metadata.  The <member>PositionType</member>
+    member specifies in which sub-area of the table the cell is positioned, which
+    in turn determines the type of metadata contained in the <member>PositionData</member>
+    member.</p>
+
+    @see com::sun::star::sheet::DataPilotTablePositionType
+    @see com::sun::star::sheet::DataPiotTableResultData
+    @see com::sun::star::sheet::DataPiotTableHeaderData
+
+    @since OOo 3.0.0
+ */
+struct DataPilotTablePositionData
+{
+	//-------------------------------------------------------------------------
+
+    /** This parameter specifies which sub-area of a DataPilot table a given 
+        cell is positioned.  See <type>DataPilotTablePositionType</type>
+        for how to interpret the value of this parameter.
+
+        @see com::sun::star::sheet::DataPilotTablePositionType
+     */
+    long    PositionType;
+
+	//-------------------------------------------------------------------------
+
+    /** <p>This member contains a structure of different types depending on the 
+        position type specified in <member>PositionType</member> member.</p>
+
+        <p>When the value of <member>PositionType</member> is 
+        <const>DataPilotTablePositionType::RESULT</const>, <member>DataPilotTablePositionData::PositionData</member> 
+        contains an instance of type <type>DataPilotTableResultData</type>, 
+        whereas when the value of <member>DataPilotTablePositionData::PositionType</member>
+        is either <const>DataPilotTablePositionType::ROW_HEADER</const> 
+        or <const>DataPilotTablePositionType::COLUMN_HEADER</const>, then the 
+        <member>PositionData</member> member contains an instance of type <type>
+        DataPilotTableHeaderData</type>.</p>
+
+        @see com::sun::star::sheet::DataPiotTableResultData
+        @see com::sun::star::sheet::DataPiotTableHeaderData
+     */
+    any     PositionData;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
+
Index: offapi/com/sun/star/sheet/DataPilotTablePositionType.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/DataPilotTablePositionType.idl
diff -N offapi/com/sun/star/sheet/DataPilotTablePositionType.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/DataPilotTablePositionType.idl	18 Apr 2008 19:19:41 -0000	1.1.2.7
@@ -0,0 +1,86 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: DataPilotTablePositionType.idl,v $
+ *
+ *  $Revision: 1.1.2.7 $
+ *
+ *  last change: $Author: kohei $ $Date: 2008/04/18 19:19:41 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_DataPilotTablePositionType_idl__
+#define __com_sun_star_sheet_DataPilotTablePositionType_idl__
+
+module com { module sun { module star { module sheet {
+
+//============================================================================
+
+/** specifies in which sub-area a cell is positioned within a DataPilot table.
+
+    @see com::sun::star::sheet::DataPilotTablePositionData
+    @see com::sun::star::sheet::DataPilotTableResultData
+    @see com::sun::star::sheet::DataPilotTableHeaderData
+
+    @since OOo 3.0.0
+ */
+constants DataPilotTablePositionType
+{
+    //------------------------------------------------------------------------
+
+    /** indicates that the specified cell is not in the DataPilot table. */
+    const long NOT_IN_TABLE = 0;
+
+    //------------------------------------------------------------------------
+
+    /** indicates that the specified cell is within the result area. */
+    const long RESULT = 1;
+
+    //------------------------------------------------------------------------
+
+    /** indicates that the specified cell is within the row header area. */
+    const long ROW_HEADER = 2;
+
+    //------------------------------------------------------------------------
+
+    /** indicates that the specified cell is within the column header area. */
+    const long COLUMN_HEADER = 3;
+
+    //------------------------------------------------------------------------
+
+    /** indicates that the specified cell is within the table but in areas 
+        other than the result or header areas. */
+    const long OTHER = 4;
+};
+
+//============================================================================
+
+}; }; }; };
+
+
+
+#endif
Index: offapi/com/sun/star/sheet/DataPilotTableResultData.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/DataPilotTableResultData.idl
diff -N offapi/com/sun/star/sheet/DataPilotTableResultData.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/DataPilotTableResultData.idl	7 Dec 2007 05:29:36 -0000	1.1.2.5
@@ -0,0 +1,94 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: DataPilotTableResultData.idl,v $
+ *
+ *  $Revision: 1.1.2.5 $
+ *
+ *  last change: $Author: kohei $ $Date: 2007/12/07 05:29:36 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_DataPilotTableResultData_idl__
+#define __com_sun_star_sheet_DataPilotTableResultData_idl__
+
+#include <com/sun/star/sheet/DataPilotFieldFilter.idl>
+#include <com/sun/star/sheet/DataResult.idl>
+
+//=============================================================================
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** information about a cell positioned within the result area of a DataPilot table.
+
+    <p><type>DataPilotTableResultData</type> contains information about a 
+    particular cell positioned within the result area of a DataPilot table.</p>
+
+    @see com::sun::star::sheet::DataPilotTablePositionData
+    @see com::sun::star::sheet::DataPilotTablePositionType
+    @see com::sun::star::sheet::DataPilotFieldFilter
+    @see com::sun::star::sheet::DataResult
+
+    @since OOo 3.0.0
+ */
+struct DataPilotTableResultData
+{
+    //------------------------------------------------------------------------
+
+    /** This is a set of filter criteria that can be used to re-create those
+        data rows that contribute to the value shown in the cell.
+
+        @see com::sun::star::sheet::DataPilotFieldFilter
+     */
+    sequence< DataPilotFieldFilter >    FieldFilters;
+
+    //------------------------------------------------------------------------
+
+    /** This is a 0-based index that specifies which data field the data displayed
+        in the cell is for; the value of 0 means the cell is for the first data
+        field, 1 for the second, and so on.
+     */
+    long                                DataFieldIndex;
+
+    //------------------------------------------------------------------------
+
+    /** more information about the result contained in the 
+        <type>DataResult</type> type.
+
+        @see com::sun::star::sheet::DataResult
+     */
+    DataResult                          Result;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
+
Index: offapi/com/sun/star/sheet/XDataPilotTable2.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/XDataPilotTable2.idl
diff -N offapi/com/sun/star/sheet/XDataPilotTable2.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/XDataPilotTable2.idl	18 Apr 2008 15:06:44 -0000	1.1.2.7
@@ -0,0 +1,139 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: XDataPilotTable2.idl,v $
+ *
+ *  $Revision: 1.1.2.7 $
+ *
+ *  last change: $Author: kohei $ $Date: 2008/04/18 15:06:44 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_XDataPilotTable2_idl__
+#define __com_sun_star_sheet_XDataPilotTable2_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __com_sun_star_table_CellAddress_idl__
+#include <com/sun/star/table/CellAddress.idl>
+#endif
+
+#ifndef __com_sun_star_table_CellRangeAddress_idl__
+#include <com/sun/star/table/CellRangeAddress.idl>
+#endif
+
+#ifndef __com_sun_star_sheet_XDataPilotTable_idl__
+#include <com/sun/star/sheet/XDataPilotTable.idl>
+#endif
+
+#ifndef __com_sun_star_sheet_DataPilotOutputRangeType_idl__
+#include <com/sun/star/sheet/DataPilotOutputRangeType.idl>
+#endif
+
+#ifndef __com_sun_star_sheet_DataPilotFieldFilter_idl__
+#include <com/sun/star/sheet/DataPilotFieldFilter.idl>
+#endif
+
+#include <com/sun/star/sheet/DataPilotTablePositionData.idl>
+#include <com/sun/star/lang/IllegalArgumentException.idl>
+
+module com {  module sun {  module star {  module sheet {
+
+/** additional methods to extend <type scope="com::sun::star::sheet">XDataPilotTable</type>. 
+
+   <type scope="com::sun::star::sheet">XDataPilotTable2</type> extends the old 
+   <type scope="com::sun::star::sheet">XDataPilotTable</type> interface with 
+   additional methods.
+
+   @see com::sun::star::sheet::XDataPilotTable
+
+   @since OOo 3.0.0
+ */
+interface XDataPilotTable2: com::sun::star::sheet::XDataPilotTable
+{
+    /** When the address of a cell within the result area is given, 
+        <method>XDataPilotTable2::getDrillDownData</method> returns its drill-down 
+        output table that includes only those rows that contribute to the value of
+        that cell.
+
+        @param aAddr cell address within the result area of a DataPilot table.
+
+        @returns drill-down output as 2-dimensional sequence, including the header row.
+
+        @see <method>XDataPilotTable2::insertDrillDownSheet</method>
+     */
+    sequence< sequence< any > > getDrillDownData( [in] com::sun::star::table::CellAddress aAddr );
+
+    /** Given a cell address, it returns the information about that cell.  The 
+        type of information returned depends upon whether the cell is within 
+        the result area or column/row header area.
+
+        @param aAddr address of the cell whose information is to be returned.
+
+        @returns <type>DataPilotTablePositionData</type> which contains the 
+        position type and the information for that cell position.
+
+        @see com::sun::star::sheet::DataPilotTablePositionData
+        @see com::sun::star::sheet::DataPilotTableHeaderData
+        @see com::sun::star::sheet::DataPilotTableResultData
+     */
+    DataPilotTablePositionData getPositionData( [in] com::sun::star::table::CellAddress aAddr );
+
+    /** <p>This method inserts a new sheet to display the drill-down data for a 
+        specified result cell.  A drill-down data for a result cell consists 
+        of a subset of rows from the original data source that contribute to 
+        the value displayed in that cell.</p>
+
+        <p>The new sheet is always inserted to the immediate left of the current
+        sheet where the DataPilot table is.  Note that when the drill-down data
+        is empty, no new sheet is inserted.</p>
+
+        @param aAddr address of a result cell
+
+        @returns <void/>
+     */
+    void insertDrillDownSheet( [in] com::sun::star::table::CellAddress aAddr );
+
+    /** <p>This method returns a different output range of a DataPilot table 
+        per specified output range type.</p>
+
+        @returns <type scope="com::sun::star::table">CellRangeAddress</type> depicting
+        the range specified.  See <type>DataPilotOutputRangeType</type> for a set
+        of possible output range types.
+
+        @see com::sun::star::sheet::DataPilotOutputRangeType
+     */
+    com::sun::star::table::CellRangeAddress getOutputRangeByType( [in] long nType ) 
+        raises (com::sun::star::lang::IllegalArgumentException);
+};
+
+}; }; }; };
+
+#endif
+
Index: offapi/com/sun/star/sheet/XDrillDownDataSupplier.idl
===================================================================
RCS file: offapi/com/sun/star/sheet/XDrillDownDataSupplier.idl
diff -N offapi/com/sun/star/sheet/XDrillDownDataSupplier.idl
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ offapi/com/sun/star/sheet/XDrillDownDataSupplier.idl	7 Dec 2007 18:37:34 -0000	1.1.2.2
@@ -0,0 +1,95 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: XDrillDownDataSupplier.idl,v $
+ *
+ *  $Revision: 1.1.2.2 $
+ *
+ *  last change: $Author: kohei $ $Date: 2007/12/07 18:37:34 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef __com_sun_star_sheet_XDrillDownDataSupplier_idl__
+#define __com_sun_star_sheet_XDrillDownDataSupplier_idl__
+
+#ifndef __com_sun_star_uno_XInterface_idl__
+#include <com/sun/star/uno/XInterface.idl>
+#endif
+
+#ifndef __com_sun_star_sheet_DataPilotFieldFilter_idl__
+#include <com/sun/star/sheet/DataPilotFieldFilter.idl>
+#endif
+
+#ifndef __com_sun_star_table_CellAddress_idl__
+#include <com/sun/star/table/CellAddress.idl>
+#endif
+
+//=============================================================================
+
+module com {  module sun {  module star {  module sheet {
+
+//=============================================================================
+
+/** supplies a filtered subset of the original data source based on filtering criteria.
+
+    A service that acts as a DataPilot data source can optionally implement this
+    interface to allow drill-down of result data.  The method this interface provides
+    is used internally when calling <method>XDataPilotTable2::getDrillDownData</method>
+    or <method>XDataPilotTable2::insertDrillDownSheet</method>.  If the data source
+    service does not implement this interface, then the aformentioned two methods
+    will have no effect.
+
+    @see com::sun::star::sheet::DataPilotSource
+
+    @since OOo 3.0.0
+ */
+interface XDrillDownDataSupplier: com::sun::star::uno::XInterface
+{
+    /** This method returns filtered subset of the original source data based on 
+        a given set of filtering criteria.
+
+        @param aFilters filtering criteria
+
+        @returns a filtered subset of the origintal source data as 2-dimensional
+        sequences of <type scope="com::sun::star::uno">Any</any>.  The first row
+        must be the header row.  Each <type scope="com::sun::star::uno">Any</type>
+        instance must contain either <type>double</type> value for a numeric cell, or 
+        a <type>string</type> value for a string cell.
+
+        @see com::sun::star::sheet::DataPilotFieldFilter
+        @see com::sun::star::sheet::XDataPilotTable2
+     */
+    sequence< sequence< any > > getDrillDownData( 
+        [in] sequence< com::sun::star::sheet::DataPilotFieldFilter > aFilters );
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
+
Index: offapi/com/sun/star/sheet/makefile.mk
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/sheet/makefile.mk,v
retrieving revision 1.27
retrieving revision 1.24.138.5
diff -u -b -I $Revision.*$ -I $Author.*$ -r1.27 -r1.24.138.5
--- offapi/com/sun/star/sheet/makefile.mk	5 Feb 2008 15:53:10 -0000	1.27
+++ offapi/com/sun/star/sheet/makefile.mk	25 Feb 2008 14:45:39 -0000	1.24.138.5
@@ -101,6 +101,7 @@
 	DataPilotItem.idl\
 	DataPilotItems.idl\
 	DataPilotItemsEnumeration.idl\
+	DataPilotOutputRangeType.idl\
 	DataPilotSource.idl\
 	DataPilotSourceDimension.idl\
 	DataPilotSourceDimensions.idl\
@@ -111,6 +112,10 @@
 	DataPilotSourceMember.idl\
 	DataPilotSourceMembers.idl\
 	DataPilotTable.idl\
+	DataPilotTableHeaderData.idl\
+	DataPilotTablePositionData.idl\
+	DataPilotTablePositionType.idl\
+	DataPilotTableResultData.idl\
 	DataPilotTables.idl\
 	DataPilotTablesEnumeration.idl\
 	DataResult.idl\
@@ -244,12 +249,14 @@
 	XDataPilotMemberResults.idl\
 	XDataPilotResults.idl\
 	XDataPilotTable.idl\
+	XDataPilotTable2.idl\
 	XDataPilotTables.idl\
 	XDataPilotTablesSupplier.idl\
 	XDatabaseRange.idl\
 	XDatabaseRanges.idl\
 	XDimensionsSupplier.idl\
 	XDocumentAuditing.idl\
+	XDrillDownDataSupplier.idl\
 	XEnhancedMouseClickBroadcaster.idl\
 	XExternalSheetName.idl\
 	XFillAcrossSheet.idl\
