--- officecfg/registry/data/org/openoffice/VCL.xcu.old	2006-02-03 18:13:21.000000000 +0100
+++ officecfg/registry/data/org/openoffice/VCL.xcu	2006-04-20 16:04:09.000000000 +0200
@@ -160,6 +160,41 @@
         <value>Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma;Luxi Sans;Interface User;WarpSans;Geneva;Tahoma;MS Sans Serif;Helv;Dialog;Albany;Lucida;Helvetica;Charcoal;Chicago;Arial;Helmet;Interface System;Sans Serif</value>
       </prop>
     </node>
+    <node oor:name="el" oor:op="replace">
+      <prop oor:name="LATIN_DISPLAY" oor:type="xs:string" oor:op="replace">
+        <value>Albany AMT;Albany;Arial;DejaVu Sans;CMU Sans Serif;FreeSans;Nimbus Sans L;Helvetica;Lucida;Geneva;Helmet;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
+      </prop>
+      <prop oor:name="LATIN_HEADING" oor:type="xs:string" oor:op="replace">
+        <value>Albany AMT;Albany;Arial;DejaVu Sans;CMU Sans Serif;FreeSans;Nimbus Sans L;Helvetica;Lucida;Geneva;Helmet;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
+      </prop>
+      <prop oor:name="LATIN_FIXED" oor:type="xs:string" oor:op="replace">
+        <value>Cumberland AMT;Cumberland;Courier New;DejaVu Sans Mono;CMU Typewriter Text;FreeMono;Nimbus Mono L;Courier;Lucida Sans Typewriter;Lucida Typewriter;Monaco;Monospaced</value>
+      </prop>
+      <prop oor:name="LATIN_PRESENTATION" oor:type="xs:string" oor:op="replace">
+        <value>Albany AMT;Albany;Arial;DejaVu Sans;CMU Sans Serif;FreeSans;Nimbus Sans L;Helvetica;Lucida;Geneva;Helmet;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
+      </prop>
+      <prop oor:name="LATIN_SPREADSHEET" oor:type="xs:string" oor:op="replace">
+        <value>Albany AMT;Albany;Arial;DejaVu Sans;CMU Sans Serif;FreeSans;Nimbus Sans L;Helvetica;Lucida;Geneva;Helmet;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
+      </prop>
+      <prop oor:name="LATIN_TEXT" oor:type="xs:string" oor:op="replace">
+        <value>Thorndale AMT;Thorndale;Times New Roman;DejaVu Serif;CMU Serif;FreeSerif;Nimbus Roman No9 L;Luxi Serif;Times;Lucida Serif;Lucida Bright;Timmons;New York;Serif</value>
+      </prop>
+      <prop oor:name="SANS" oor:type="xs:string" oor:op="replace">
+        <value>Albany AMT;Albany;Arial;DejaVu Sans;CMU Sans Serif;FreeSans;Nimbus Sans L;Helvetica;Lucida;Geneva;Helmet;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
+      </prop>
+      <prop oor:name="SERIF" oor:type="xs:string" oor:op="replace">
+        <value>Thorndale AMT;Thorndale;Times New Roman;DejaVu Serif;CMU Serif;FreeSerif;Nimbus Roman No9 L;Luxi Serif;Times;Lucida Serif;Lucida Bright;Timmons;New York;Serif</value>
+      </prop>
+      <prop oor:name="UI_FIXED" oor:type="xs:string" oor:op="replace">
+        <value>Cumberland AMT;Cumberland;Courier New;DejaVu Sans Mono;CMU Typewriter Text;FreeMono;Nimbus Mono L;Courier;Lucida Sans Typewriter;Lucida Typewriter;Monaco;Monospaced</value>
+      </prop>
+      <prop oor:name="FIXED" oor:type="xs:string" oor:op="replace">
+        <value>Cumberland AMT;Cumberland;Courier New;DejaVu Sans Mono;CMU Typewriter Text;FreeMono;Nimbus Mono L;Courier;Lucida Sans Typewriter;Lucida Typewriter;Monaco;Monospaced</value>
+      </prop>
+      <prop oor:name="UI_SANS" oor:type="xs:string" oor:op="replace">
+        <value>Albany AMT;Albany;Arial;DejaVu Sans;CMU Sans Serif;FreeSans;Nimbus Sans L;Helvetica;Lucida;Geneva;Helmet;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
+      </prop>
+    </node>
    <node oor:name="hr-HR" oor:op="replace">
       <prop oor:name="LATIN_DISPLAY" oor:type="xs:string" oor:op="replace">
         <value>Albany;Arial;Luxi Serif;Helvetica;Lucida;Geneva;Helmet;SansSerif;Andale Sans UI;Arial Unicode MS;Lucida Sans Unicode;Tahoma</value>
@@ -1749,6 +1784,75 @@
           <value>Special,Italic,Script</value>
         </prop>
       </node>
+      <node oor:name="cmusansserif" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>albanyamt;albany;arial;dejavusans;cmusansserif;freesans;nimbussansl;helvetica;lucida;geneva;helmet;andalesansui;arialunicodems;lucidasansunicode;tahoma</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,SansSerif</value>
+        </prop>
+      </node>
+      <node oor:name="cmuserif" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>thorndaleamt;thorndale;timesnewroman;dejavuserif;cmuserif;freeserif;nimbusromanno9l;luxiserif;times;lucidaserif;lucidabright;timmons;newyork;serif</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,Serif</value>
+        </prop>
+      </node>
+      <node oor:name="cmutypewritertext" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>cumberlandamt;cumberland;couriernew;dejavusansmono;cmutypewritertext;freemono;nimbusmonol;courier;lucidasanstypewriter;lucidatypewriter;monaco;monospaced</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,Fixed,Typewriter</value>
+        </prop>
+      </node>
       <node oor:name="colonna" oor:op="replace">
         <prop oor:name="SubstFonts">
           <value>monotypeoldstyleboldoutline;monotypeoldstyleoutline;chevaraoutline;imprintmtshadow;imprintshadow;imprint;colonnamt;castellar;imprintmtshadow;imprintshadow;imprint;chevaraoutline;chevara;gallia;colonnamt;algerian;castellar;imprintmtshadow;imprintshadow;imprint;chevara;gallia;algerian</value>
@@ -1978,6 +2082,75 @@
           <value>Italic,Script</value>
         </prop>
       </node>
+      <node oor:name="dejavusans" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>albanyamt;albany;arial;dejavusans;cmusansserif;freesans;nimbussansl;helvetica;lucida;geneva;helmet;andalesansui;arialunicodems;lucidasansunicode;tahoma</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,SansSerif</value>
+        </prop>
+      </node>
+      <node oor:name="dejavusansmono" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>cumberlandamt;cumberland;couriernew;dejavusansmono;cmutypewritertext;freemono;nimbusmonol;courier;lucidasanstypewriter;lucidatypewriter;monaco;monospaced</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,Fixed,Typewriter</value>
+        </prop>
+      </node>
+      <node oor:name="dejavuserif" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>thorndaleamt;thorndale;timesnewroman;dejavuserif;cmuserif;freeserif;nimbusromanno9l;luxiserif;times;lucidaserif;lucidabright;timmons;newyork;serif</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,Serif</value>
+        </prop>
+      </node>
       <node oor:name="din" oor:op="replace">
         <prop oor:name="SubstFonts">
           <value>albany;arial;helvetica;lucidasans;lucida;geneva;helmet;sansserif;nimbussansl;nimbussans;andalesansui;arialunicodems;lucidaunicode</value>
@@ -2392,6 +2565,75 @@
           <value>Decorative,Italic,Script,Comic</value>
         </prop>
       </node>
+      <node oor:name="freemono" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>cumberlandamt;cumberland;couriernew;dejavusansmono;cmutypewritertext;freemono;nimbusmonol;courier;lucidasanstypewriter;lucidatypewriter;monaco;monospaced</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,Fixed,Typewriter</value>
+        </prop>
+      </node>
+      <node oor:name="freesans" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>albanyamt;albany;arial;dejavusans;cmusansserif;freesans;nimbussansl;helvetica;lucida;geneva;helmet;andalesansui;arialunicodems;lucidasansunicode;tahoma</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,SansSerif</value>
+        </prop>
+      </node>
+      <node oor:name="freeserif" oor:op="replace">
+        <prop oor:name="SubstFonts">
+          <value>thorndaleamt;thorndale;timesnewroman;dejavuserif;cmuserif;freeserif;nimbusromanno9l;luxiserif;times;lucidaserif;lucidabright;timmons;newyork;serif</value>
+        </prop>
+        <prop oor:name="SubstFontsMS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsPS">
+          <value></value>
+        </prop>
+        <prop oor:name="SubstFontsHTML">
+          <value></value>
+        </prop>
+        <prop oor:name="FontWeight">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontWidth">
+          <value>Normal</value>
+        </prop>
+        <prop oor:name="FontType">
+          <value>Standard,Normal,Serif</value>
+        </prop>
+      </node>
       <node oor:name="frenchscript" oor:op="replace">
         <prop oor:name="SubstFonts">
           <value>palacescript;palacescriptmt;arioso;shelley;zapfchancery;itczapfchancery;monotypecorsiva;corsiva;chancery;chanceryl;lucidacalligraphy;lucidahandwriting;comicsansms;kidprint;</value>
