--- sfx2/source/doc/docinf.cxx	2006-09-17 12:39:49.000000000 -0400
+++ sfx2/source/doc/docinf.cxx	2006-11-08 08:58:37.000000000 -0500
@@ -947,7 +947,7 @@ private:
 
     bool                SeekToPropertyPos( SvStream& rStrm, sal_uInt32 nPropPos ) const;
     void                LoadProperty( SvStream& rStrm, sal_Int32 nPropId );
-    void                SaveProperty( SvStream& rStrm, SfxOlePropertyBase& rProp, sal_Size& rnPropPosPos );
+    sal_Size            SaveProperty( SvStream& rStrm, SfxOlePropertyBase& rProp, sal_Size& rnPropPosPos );
 
 private:
     SfxOlePropMap       maPropMap;              /// All properties in this section, by identifier.
@@ -1195,6 +1195,8 @@ void SfxOleSection::ImplLoad( SvStream& 
 
 void SfxOleSection::ImplSave( SvStream& rStrm )
 {
+    sal_Size nEndPos;
+
     /*  Always export with UTF-8 encoding. All dependent properties (bytestring
         and dictionary) will be updated automatically. */
     maCodePageProp.SetTextEncoding( RTL_TEXTENCODING_UTF8 );
@@ -1212,17 +1214,16 @@ void SfxOleSection::ImplSave( SvStream& 
 
     // write dictionary property
     if( maDictProp.HasPropertyNames() )
-        SaveProperty( rStrm, maDictProp, nPropPosPos );
+        nEndPos = SaveProperty( rStrm, maDictProp, nPropPosPos );
     // write codepage property
-    SaveProperty( rStrm, maCodePageProp, nPropPosPos );
+    nEndPos = SaveProperty( rStrm, maCodePageProp, nPropPosPos );
     // write other properties
     for( SfxOlePropMap::const_iterator aIt = maPropMap.begin(), aEnd = maPropMap.end(); aIt != aEnd; ++aIt )
-        SaveProperty( rStrm, *aIt->second, nPropPosPos );
+        nEndPos = SaveProperty( rStrm, *aIt->second, nPropPosPos );
 
     // write section size (first field in section header)
-    sal_uInt32 nSectSize = static_cast< sal_uInt32 >( rStrm.Tell() - mnStartPos );
     rStrm.Seek( mnStartPos );
-    rStrm << nSectSize;
+    rStrm << static_cast< sal_uInt32 >( nEndPos - mnStartPos );
 }
 
 bool SfxOleSection::SeekToPropertyPos( SvStream& rStrm, sal_uInt32 nPropPos ) const
@@ -1264,7 +1265,7 @@ void SfxOleSection::LoadProperty( SvStre
     }
 }
 
-void SfxOleSection::SaveProperty( SvStream& rStrm, SfxOlePropertyBase& rProp, sal_Size& rnPropPosPos )
+sal_Size SfxOleSection::SaveProperty( SvStream& rStrm, SfxOlePropertyBase& rProp, sal_Size& rnPropPosPos )
 {
     rStrm.Seek( STREAM_SEEK_TO_END );
     sal_uInt32 nPropPos = static_cast< sal_uInt32 >( rStrm.Tell() - mnStartPos );
@@ -1275,10 +1276,15 @@ void SfxOleSection::SaveProperty( SvStre
     // align to 32-bit
     while( (rStrm.Tell() & 3) != 0 )
         rStrm << sal_uInt8( 0 );
+
+    sal_Size nEndPos = rStrm.Tell();
+
     // write property ID/position pair
     rStrm.Seek( rnPropPosPos );
     rStrm << rProp.GetPropId() << nPropPos;
     rnPropPosPos = rStrm.Tell();
+
+    return nEndPos;
 }
 
 // ============================================================================
