Index: ooo.src
===================================================================
RCS file: /cvs/graphics/svx/source/intro/ooo.src,v
retrieving revision 1.11
diff -u -u -r1.11 ooo.src
--- svx/source/intro/ooo.src	7 Jan 2005 09:49:30 -0000	1.11
+++ svx/source/intro/ooo.src	29 May 2005 15:41:24 -0000
@@ -78,7 +78,7 @@
 #define OOO_VENDOR "Sun Microsystems Inc."
 #define OOO_LICENSE "LGPL"
 #else
-//#define OOO_VENDOR "my company"
+#define OOO_VENDOR "Debian"
 #define OOO_LICENSE "LGPL"
 #endif
 
