--- slideshow/source/engine/userpaintoverlay.cxx.old	2005-10-11 10:37:00.000000000 +0200
+++ slideshow/source/engine/userpaintoverlay.cxx	2006-09-27 16:10:35.678388000 +0200
@@ -125,9 +125,14 @@
 
                 ::osl::MutexGuard aGuard( maMutex );
 
-                maViews.erase( ::std::remove( maViews.begin(),
-                                              maViews.end(),
-                                              rView ) );
+#if defined(_MSC_VER) && (_MSC_VER > 1310)
+                maViews.erase( remove( maViews.begin(),
+else
+                maViews.erase( std::remove( maViews.begin(),
+#endif
+                                            maViews.end(),
+                                            rView ),
+                               maViews.end() );
 
                 return true;
             }
