--- sc/source/filter/xml/xmltabi.cxx.old	2006-09-27 15:53:44.038388000 +0200
+++ sc/source/filter/xml/xmltabi.cxx	2006-09-27 15:55:19.238388000 +0200
@@ -248,15 +248,11 @@
             for (i = 0; i < nDepth; ++i)
 			{
 				sal_Int32 nCount(pColArray->GetCount(static_cast<USHORT>(i)));
-				sal_Bool bChanged(sal_False);
-				for (sal_Int32 j = 0; j < nCount && !bChanged; ++j)
+				for (sal_Int32 j = 0; j < nCount; ++j)
 				{
 					ScOutlineEntry* pEntry(pColArray->GetEntry(static_cast<USHORT>(i), static_cast<USHORT>(j)));
 					if (pEntry->IsHidden())
-					{
 						pColArray->SetVisibleBelow(static_cast<USHORT>(i), static_cast<USHORT>(j), sal_False);
-						bChanged = sal_True;
-					}
 				}
 			}
 			ScOutlineArray* pRowArray(pOutlineTable->GetRowArray());
@@ -264,15 +260,11 @@
 			for (i = 0; i < nDepth; ++i)
 			{
 				sal_Int32 nCount(pRowArray->GetCount(static_cast<USHORT>(i)));
-				sal_Bool bChanged(sal_False);
-				for (sal_Int32 j = 0; j < nCount && !bChanged; ++j)
+				for (sal_Int32 j = 0; j < nCount; ++j)
 				{
 					ScOutlineEntry* pEntry(pRowArray->GetEntry(static_cast<USHORT>(i), static_cast<USHORT>(j)));
 					if (pEntry->IsHidden())
-					{
 						pRowArray->SetVisibleBelow(static_cast<USHORT>(i), static_cast<USHORT>(j), sal_False);
-						bChanged = sal_True;
-					}
 				}
 			}
 		}
