--- svx/source/dialog/cuihyperdlg.cxx.old	2006-09-27 15:53:42.978388000 +0200
+++ svx/source/dialog/cuihyperdlg.cxx	2006-09-27 15:55:18.178388000 +0200
@@ -368,7 +368,7 @@
 
 	SvxHyperlinkTabPageBase* pCurrentPage = (SvxHyperlinkTabPageBase*)GetTabPage( nPageId );
 
-	mbIsHTMLDoc = BOOL( pItem->GetInsertMode() & HLINK_HTMLMODE );
+	mbIsHTMLDoc = (pItem->GetInsertMode() & HLINK_HTMLMODE) ? true : false;
 
 	SfxItemSet& aPageSet =  (SfxItemSet&)GetTabPage (nPageId)->GetItemSet ();
 	aPageSet.Put ( *pItem );
