Index: source/ui/src/localize.sdf
===================================================================
RCS file: /cvs/sc/sc/source/ui/src/localize.sdf,v
retrieving revision 1.29.38.1
diff -u -r1.29.38.1 localize.sdf
--- sc/source/ui/src/localize.sdf	2 Feb 2007 12:04:51 -0000	1.29.38.1
+++ sc/source/ui/src/localize.sdf	27 Mar 2007 14:45:42 -0000
@@ -48423,7 +48423,7 @@
 sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		0	nr	~Cima amaKholomu				2002-02-02 02:02:02
 sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		0	ns	~Phumula Dikholomo				2002-02-02 02:02:02
 sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		0	or-IN	ସ୍ତମ୍ଭଗୁଡିକୁ ବିଲୋପକର				2002-02-02 02:02:02
-sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		13691	pa-IN	ਕਾਲਮ ਸ਼ਾਮਿਲ(~d)				2002-02-02 02:02:02
+sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		13691	pa-IN	ਕਾਲਮ ਹਟਾਓ(~d)				2002-02-02 02:02:02
 sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		0	pl	~Usuń kolumny				2002-02-02 02:02:02
 sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		0	pt	Eliminar ~colunas				2002-02-02 02:02:02
 sc	source\ui\src\hdrcont.src	0	menuitem	RID_POPUP_COLHEADER	SID_DEL_COLS	SID_DEL_COLS		0	pt-BR	~Excluir Colunas				2002-02-02 02:02:02
