--- svx/source/dialog/multipat.cxx	27 Jun 2007 17:22:03 -0000	1.11
+++ svx/source/dialog/multipat.cxx	7 Feb 2008 08:54:15 -0000
@@ -138,7 +138,7 @@
         if ( pImpl->bIsRadioButtonMode )
         {
             ULONG nPos = aRadioLB.GetEntryPos( sInsPath, 1 );
-            if ( (ULONG)-1 == nPos )
+            if ( 0xffffffff == nPos ) //See svtools/source/contnr/svtabbx.cxx SvTabListBox::GetEntryPos
             {
                 String sNewEntry( '\t' );
                 sNewEntry += sInsPath;
