--- solenv/bin/packimages.pl	2005-02-14 16:55:04.129927103 +0100
+++ solenv/bin/packimages.pl	2005-02-14 16:55:09.167058489 +0100
@@ -243,7 +243,7 @@ sub find_custom
     for my $path (@custom_path) {
 	find({ wanted => \&wanted, no_chdir => 0 }, $path);
 	foreach ( @custom_list ) {
-	    if ( /^\Q$path\E\/(.*)$/o ) {
+	    if ( /^\Q$path\E\/(.*)$/ ) {
 		$keep_back=$1;
 		if (!defined $custom_hash_ref->{$keep_back}) {
 		    $custom_hash_ref->{$keep_back} = $path;
