Index: padmin/source/makefile.mk
===================================================================
RCS file: /cvs/gsl/padmin/source/makefile.mk,v
retrieving revision 1.14
diff -u -p -r1.14 makefile.mk
--- padmin/source/makefile.mk	29 Oct 2004 12:45:30 -0000	1.14
+++ padmin/source/makefile.mk	25 Nov 2004 12:37:44 -0000
@@ -71,6 +71,10 @@ LIBTARGET=NO
 
 .INCLUDE: settings.mk
 
+.IF "$(FONTCONFIG_FONTS)" != ""
+CFLAGS+=-DFONTCONFIG_FONTS
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SRS1NAME=$(TARGET)
@@ -94,11 +98,13 @@ SLOFILES=\
 	$(SLO)$/newppdlg.obj		\
 	$(SLO)$/prtsetup.obj		\
 	$(SLO)$/prgsbar.obj			\
-	$(SLO)$/fontentry.obj		\
 	$(SLO)$/helper.obj			\
 	$(SLO)$/adddlg.obj			\
 	$(SLO)$/titlectrl.obj
 
+.IF "$(FONTCONFIG_FONTS)" == ""
+SLOFILES+= $(SLO)$/fontentry.obj
+.ENDIF
 
 OBJFILES=\
 	$(OBJ)/pamain.obj
Index: padmin/source/padialog.cxx
===================================================================
RCS file: /cvs/gsl/padmin/source/padialog.cxx,v
retrieving revision 1.13
diff -u -p -r1.13 padialog.cxx
--- padmin/source/padialog.cxx	9 Nov 2004 16:50:55 -0000	1.13
+++ padmin/source/padialog.cxx	25 Nov 2004 12:39:13 -0000
@@ -89,9 +89,14 @@
 #ifndef _TOOLS_COLOR_HXX
 #include <tools/color.hxx>
 #endif
+#ifndef _URLOBJ_HXX
+#include <tools/urlobj.hxx>
+#endif
+#ifndef FONTCONFIG_FONTS
 #ifndef _PAD_PADIALOG_HRC_
 #include <padialog.hrc>
 #endif
+#endif
 #ifndef _PAD_FONTENTRY_HXX_
 #include <fontentry.hxx>
 #endif
@@ -156,7 +161,9 @@ PADialog::PADialog( Window* pParent, BOO
 		m_aComment( this, PaResId( RID_PA_TXT_COMMENT_STRING ) ),
         m_aSepButtonFL( this, PaResId( RID_PA_FL_SEPBUTTON ) ),
 		m_aAddPB( this, PaResId( RID_PA_BTN_ADD ) ),
+	#ifndef FONTCONFIG_FONTS
 		m_aFontsPB( this, PaResId( RID_PA_BTN_FONTS ) ),
+	#endif
 		m_aCancelButton( this, PaResId( RID_PA_BTN_CANCEL ) ),
 		m_aDefPrt( PaResId( RID_PA_STR_DEFPRT ) ),
 		m_aRenameStr( PaResId( RID_PA_STR_RENAME ) ),
@@ -184,13 +191,15 @@ void PADialog::Init()
 	m_aConfPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aRenamePB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aTestPagePB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
-	m_aFontsPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aAddPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aDevicesLB.setDelPressedLink( LINK( this, PADialog, DelPressedHdl ) );
 
+     #ifndef FONTCONFIG_FONTS
+        m_aFontsPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	::psp::PrintFontManager& rFontManager( ::psp::PrintFontManager::get() );
 	if( ! rFontManager.checkImportPossible() )
 		m_aFontsPB.Enable( FALSE );
+     #endif
 }
 
 PADialog::~PADialog()
@@ -227,12 +236,13 @@ IMPL_LINK( PADialog, ClickBtnHdl, PushBu
 		PrintTestPage();
 	else if( pButton == &m_aAddPB )
 		AddDevice();
+     #ifndef FONTCONFIG_FONTS
 	else if( pButton == &m_aFontsPB )
 	{
 		FontNameDlg aDialog( this );
 		aDialog.Execute();
 	}
-
+     #endif
 	return 0;
 }
 
Index: padmin/source/padialog.hxx
===================================================================
RCS file: /cvs/gsl/padmin/source/padialog.hxx,v
retrieving revision 1.5
diff -u -p -r1.5 padialog.hxx
--- padmin/source/padialog.hxx	6 Jan 2004 16:50:50 -0000	1.5
+++ padmin/source/padialog.hxx	25 Nov 2004 12:39:50 -0000
@@ -118,7 +118,9 @@ namespace padmin {
 
         FixedLine							m_aSepButtonFL;
 		PushButton							m_aAddPB;
+#ifndef FONTCONFIG_FONTS
 		PushButton							m_aFontsPB;
+#endif
 		CancelButton  						m_aCancelButton;
 
 		String								m_aDefPrt;
