--- idlc/source/idlcproduce.cxx	2006-07-11 12:06:59.000000000 +0200
+++ idlc/source/idlcproduce.cxx	2006-07-11 12:27:04.000000000 +0200
@@ -150,7 +150,15 @@ sal_Int32 SAL_CALL produceFile(const OSt
 {
 	Options* pOptions = idlc()->getOptions();
 
-    OString regTmpName = regFileName.replaceAt(regFileName.getLength() -3, 3, "_idlc_");
+	OString regTmpName = regFileName.replaceAt(regFileName.getLength() -3, 3, "_idlc_XXXXXX");
+	mktemp( const_cast< char * >( regTmpName.getStr() ) );
+
+	if ( !regTmpName.getLength() )	
+	{
+		fprintf(stderr, "%s: Failed to create a tempname for '%s'.\n", 
+				pOptions->getProgramName().getStr(), regFileName.getStr());
+		return 1;
+	}	
 
 	if ( !checkOutputPath(regFileName) )	
 	{
