--- python/makefile.mk.orig	2004-10-19 18:05:34.913872472 +0200
+++ python/makefile.mk	2004-10-19 18:18:32.484663744 +0200
@@ -106,9 +106,9 @@ CONFIGURE_ACTION= ./configure --prefix=.
 CONFIGURE_ACTION += --disable-ipv6
 .ENDIF
 .IF "$(OS)" == "IRIX"
-BUILD_ACTION=$(ENV_BUILD) gmake ; gmake install
+BUILD_ACTION=$(ENV_BUILD) gmake -j$(MAXPROCESS) ; gmake install
 .ELSE
-BUILD_ACTION=$(ENV_BUILD) make ; make install
+BUILD_ACTION=$(ENV_BUILD) make -j$(MAXPROCESS) ; make install
 .ENDIF
 .IF "$(OS)"=="MACOSX"
 PYTHONCORESHL=$(OUT)$/lib$/$(DLLPRE)python$(PYMAJOR)$(PYMINOR)$(DLLPOST)
