--- i18npool/source/breakiterator/makefile.mk	2005-03-21 16:51:04.366837129 +0100
+++ i18npool/source/breakiterator/makefile.mk	2005-03-21 16:51:12.537425643 +0100
@@ -74,13 +74,13 @@ ENABLE_EXCEPTIONS=TRUE
 # --- Files --------------------------------------------------------
 
 # grab all .txt files under data directory, which are breakiterator rule files.
-MY_BRK_TXTFILES=$(shell ls data/*.txt)
+MY_BRK_TXTFILES:=$(shell ls data/*.txt)
 
 # insert "OpenOffice" as icu package name in front of the  name of each rule file for searching on application provided data
-MY_BRK_BRKFILES=$(subst,data/,$(MISC)$/OpenOffice_ $(MY_BRK_TXTFILES:s/.txt/.brk/))
+MY_BRK_BRKFILES:=$(subst,data/,$(MISC)$/OpenOffice_ $(MY_BRK_TXTFILES:s/.txt/.brk/))
 
 # OpenOffice_icu_dat.c is a generated file from the rule file list by gencmn
-MY_MISC_CXXFILES = \
+MY_MISC_CXXFILES := \
         $(MISC)$/OpenOffice_icu_dat.c \
         $(MY_BRK_BRKFILES:s/.brk/_brk.c/)
 
