--- solenv/inc/target.mk	12 Jul 2006 10:11:48 -0000	1.175.4.2
+++ solenv/inc/target.mk	12 Jul 2006 11:20:47 -0000
@@ -2037,19 +2036,25 @@
 .IF "$(IDLFILES)"!=""
 
 $(URDTARGET) : $(DEPIDLFILES)
-#.IF "$(MAXPROCESS)"<="1"
+		@+-mkdir $(OUT)$/ucr >& $(NULLDEV)
+# use this target only to speedup single proceess builds
+# see according rules in rules.mk
+.IF "$(MAXPROCESS)"<="1"
 		+$(IDLC) @$(mktmp $(UNOIDLDEFS) $(UNOIDLINCEXTRA) $(UNOIDLINC) -O$(OUT)$/ucr$/$(IDLPACKAGE) $(DEPIDLFILES:+"\n"))
-#.ENDIF			# "$(MAXPROCESS)"<="1"
-		@+echo > $@
+.ENDIF			# "$(MAXPROCESS)"<="1"
+	@+echo > $@
 
 .IF "$(URDDOC)"!=""
 
 $(URDDOCTARGET) : $(DEPIDLFILES)
 		@+-mkdir $(OUT)$/ucrdoc >& $(NULLDEV)
-#.IF "$(MAXPROCESS)"<="1"
-		+$(IDLC) @$(mktmp $(UNOIDLDEFS) $(UNOIDLINCEXTRA) $(UNOIDLINC) -C -O$(OUT)$/ucrdoc$/$(IDLPACKAGE) $(DEPIDLFILES:+"\n"))		
-#.ENDIF			# "$(MAXPROCESS)"<="1"
-		@+echo > $@
+# use this target only to speedup single proceess builds
+# see according rules in rules.mk
+.IF "$(MAXPROCESS)"<="1"
+		+$(IDLC) @$(mktmp $(UNOIDLDEFS) $(UNOIDLINCEXTRA) $(UNOIDLINC) -C -O$(OUT)$/ucrdoc$/$(IDLPACKAGE) $(DEPIDLFILES:+"\n"))
+.ENDIF			# "$(MAXPROCESS)"<="1"
+	@+echo > $@
+	
 .ENDIF			# "$(URDDOC)"!=""
 .ENDIF			# "$(IDLFILES)"!=""
 
