--- officecfg/registry/data/org/openoffice/Office/Common.xcu.ark	2006-03-12 03:53:26.000000000 +0000
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	2006-03-12 03:59:16.000000000 +0000
@@ -536,6 +536,11 @@
       </prop>
     </node>
   </node>
+  <node oor:name="ExternalMailer">
+    <prop oor:name="Program" oor:type="xs:string">
+      <value>/usr/kde/3.5/bin/kmail</value>
+    </prop>
+  </node>
   <!-- FIXME: This comment is needed to change a checksum or so.
               Then the updated list of templates paths is read from Common.xcs-->
 </oor:component-data>
