--- scp2/source/ooo/file_extra_ooo.scp.orig	2005-10-09 01:39:27.123969504 +0400
+++ scp2/source/ooo/file_extra_ooo.scp	2005-10-09 01:49:33.404404593 +0400
@@ -181,6 +181,13 @@
     Name = "gallroot.zip";
 End
 
+File gid_File_Extra_Gallopenclipart_Lang
+    Dir = gid_Dir_Gallery;
+    ARCHIVE_TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(openclipart);
+End
+
 File gid_File_Extra_Gallrulers
     Dir = gid_Dir_Gallery_Rulers;
     ARCHIVE_TXT_FILE_BODY;
