Index: unxlngppc4.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc4.mk,v
retrieving revision 1.24
diff -u -u -r1.24 unxlngppc4.mk
--- solenv/inc/unxlngppc4.mk	5 Jul 2006 22:03:12 -0000	1.24
+++ solenv/inc/unxlngppc4.mk	3 Aug 2006 11:49:54 -0000
@@ -103,6 +103,11 @@
 
 # Compiler flags for debugging
 CFLAGSDEBUG=-g
+.IF "$(ENABLE_SYMBOLS)" == "SMALL"
+CFLAGSENABLESYMBOLS=-g1
+.ELSE
+CFLAGSENABLESYMBOLS=-g
+.ENDIF
 CFLAGSDBGUTIL=
 
 # Compiler flags for enabling optimizations
Index: unxlngppc.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/unxlngppc.mk,v
retrieving revision 1.25
diff -u -u -r1.25 unxlngppc.mk
--- solenv/inc/unxlngppc.mk	5 Jul 2006 22:03:00 -0000	1.25
+++ solenv/inc/unxlngppc.mk	3 Aug 2006 11:49:54 -0000
@@ -72,6 +72,12 @@
 # name of C Compiler
 CC*=gcc
 
+.IF "$(ENABLE_SYMBOLS)" == "SMALL"
+CFLAGSENABLESYMBOLS=-g1
+.ELSE
+CFLAGSENABLESYMBOLS=-g 
+.ENDIF
+
 # source code is still not signed versus unsigned char clean 
 CFLAGS=-fsigned-char -nostdinc -c
 CFLAGSCC=-fsigned-char $(ARCH_FLAGS)
--- config_office/configure.in-old	2006-08-03 14:09:42.000000000 +0200
+++ config_office/configure.in	2006-08-03 14:10:23.000000000 +0200
@@ -950,6 +950,7 @@
 	else
 		if test "$enable_symbols" = "SMALL" -o "$enable_symbols" = "small"; then
 			ENABLE_SYMBOLS="SMALL"
+			AC_MSG_RESULT([yes, small ones])
 		else if test "$enable_symbols" != "no" ; then
 			     echo enable symbols is: $enable_symbols
 			     AC_MSG_ERROR([--enable-symbols only accepts yes, TRUE or SMALL as parameter.])
