--- solenv/bin/modules/installer/languagepack.pm.orig	2007-09-06 11:52:02.000000000 +0200
+++ solenv/bin/modules/installer/languagepack.pm	2008-01-15 18:25:40.000000000 +0100
@@ -98,6 +98,8 @@
 						if ( $oneitem->{'Dir'} eq "gid_Dir_Fonts_Truetype" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Fonts_$locallang"; }
 						if ( $oneitem->{'Dir'} eq "gid_Dir_Resource" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Resource_$locallang"; }
 						if ( $oneitem->{'Dir'} eq "gid_Dir_Help_Isolanguage" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Help_$locallang"; }
+						if ( $oneitem->{'Dir'} eq "gid_Dir_Samples_Language" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Accessories_Samples_$locallang"; }
+						if ( $oneitem->{'Dir'} eq "gid_Dir_Template_Language" ) { $oneitem->{'modules'} = "gid_Module_Langpack_Accessories_Templates_$locallang"; }
 					}
 				}
 
@@ -204,6 +206,8 @@
 		if ( ${$alldirs}[$i] =~ /-fonts/ ) { next; } 
 		if ( ${$alldirs}[$i] =~ /-help/ ) { next; } 
 		if ( ${$alldirs}[$i] =~ /-res/ ) { next; } 
+        	if ( ${$alldirs}[$i] =~ /-samples/ ) { next; } 
+		if ( ${$alldirs}[$i] =~ /-templates/ ) { next; } 
 	
 		$packagename = ${$alldirs}[$i];
 		last;
