diff -ur psprint_config.clean/configuration/psprint.conf psprint_config/configuration/psprint.conf
--- psprint_config.clean/configuration/psprint.conf	2007-12-04 17:18:21.000000000 -0500
+++ psprint_config/configuration/psprint.conf	2007-12-04 17:18:29.000000000 -0500
@@ -48,7 +48,8 @@
 ;         defaults for newly added printers.
 ; PerformFontSubstitution: wether to replace downloadable fonts
 ;         with resident fonts or not
-PerformFontSubstitution=true
+; PJ: changed to false
+PerformFontSubstitution=false
 
 ; SubstFont_<downloadable font>: the resident font to replace
 ;         <downloadable font> if PerformFontSubstitution is true
@@ -84,7 +85,8 @@
 
 ; PerformFontSubstitution: wether to replace downloadable fonts
 ;         with resident fonts or not
-PerformFontSubstitution=true
+; PJ: changed to false        
+PerformFontSubstitution=false
 
 ; SubstFont_<downloadable font>: the resident font to replace
 ;         <downloadable font> if PerformFontSubstitution is true
