Index: configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.221
diff -u -u -r1.221 configure.in
--- config_office/configure.in	4 Oct 2007 07:48:20 -0000	1.221
+++ config_office/configure.in	15 Oct 2007 09:02:22 -0000
@@ -3133,7 +3133,8 @@
    dnl check if the headers really work:
    save_CPPFLAGS="$CPPFLAGS"
    CPPFLAGS="$CPPFLAGS $PYTHON_CFLAGS"
-   AC_TRY_CPP(Python.h, , AC_MSG_ERROR([Python headers not found.]))
+   AC_CHECK_HEADER(Python.h, [],
+       [AC_MSG_ERROR(Python headers not found)], [])
    CPPFLAGS="$save_CPPFLAGS"
 else
    SYSTEM_PYTHON=NO
