--- pyuno/source/module/makefile.mk-old	2007-05-03 22:07:11.000000000 +0200
+++ pyuno/source/module/makefile.mk	2007-05-03 22:08:35.000000000 +0200
@@ -131,7 +131,7 @@
 .ELIF "$(OS)" == "MACOSX"
 	@echo $(CC) -bundle -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELSE
-	@echo $(LINK) $(LINKFLAGSSHLCUI) -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
+	@echo $(LINK) -Wl,-z,combreloc -Wl,-rpath,\''$$ORIGIN'\':/usr/lib/openoffice/program $(LINKFLAGSSHLCUI) -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ENDIF
 	cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
