--- pyuno/source/module/makefile.mk	2005-08-07 17:35:23.198884312 +0200
+++ pyuno/source/module/makefile.mk	2005-08-07 17:41:05.457853040 +0200
@@ -145,7 +145,7 @@
 .IF "$(GUI)" == "UNX"
 $(PYUNO_MODULE) : $(SLO)$/pyuno_dlopenwrapper.obj
 .IF "$(OS)" == "LINUX"
-	@+ echo $(LINK) $(LINKFLAGS) $(LINKFLAGSSHLCUI) -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
+	@+ echo $(LINK) -Wl,-z,combreloc $(LINKFLAGSDEFS) -Wl,-rpath,\''$$ORIGIN'\':/usr/lib/openoffice2/program $(LINKFLAGSSHLCUI) -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELIF "$(OS)" == "SOLARIS"
 	@+ echo ld -G -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELIF "$(OS)" == "FREEBSD"
