--- pyuno/source/module/makefile.mk-old	2007-05-03 22:07:11.000000000 +0200
+++ pyuno/source/module/makefile.mk	2007-05-03 22:08:35.000000000 +0200
@@ -121,7 +121,7 @@
 .IF "$(GUI)" == "UNX"
 $(PYUNO_MODULE) : $(SLO)$/pyuno_dlopenwrapper.obj
 .IF "$(OS)" == "LINUX"
-       @echo $(LINK) $(LINKFLAGS) $(LINKFLAGSSHLCUI) -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
+       @echo $(LINK) -Wl,-z,combreloc $(LINKFLAGS) -Wl,-rpath,/usr/lib/openoffice/program $(LINKFLAGSSHLCUI) -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELIF "$(OS)" == "SOLARIS"
        @echo ld -G -ldl -o $@ $(SLO)$/pyuno_dlopenwrapper.o > $(MISC)$/$(@:b).cmd
 .ELIF "$(OS)" == "FREEBSD"
