--- /dev/null	2004-07-26 17:21:14.000000000 +0530
+++ qpro.hxx	2005-02-25 15:17:05.000000000 +0530
@@ -0,0 +1,43 @@
+#ifndef _QPRO_HXX
+#define _QPRO_HXX
+
+#include "flttools.hxx"
+#include "filter.hxx"
+#include "scerrors.hxx"
+#include "document.hxx"
+#include "cell.hxx"
+#include "math.h"
+
+#include <tools/color.hxx>
+                                                                                                                  
+// Stream wrapper class
+class BiffReader
+{
+    protected:
+        sal_uInt16 mnId;
+       	sal_uInt16 mnLength;
+       	sal_uInt32 mnOffset;
+       	rtl_TextEncoding meCharSet;
+                                                                                                                  
+    public:
+	sal_uInt8 Align[ 256 ];
+	sal_uInt8 Font[ 256 ];
+	sal_uInt16 FontRecord[ 256 ];
+	String FontType[ 256 ];
+	bool EndOfFile;
+	SvStream *mpStream;
+        BiffReader( SfxMedium& rMedium );
+        ~BiffReader();
+       	bool recordsLeft() { return mpStream && !mpStream->IsEof(); }
+       	bool nextRecord();
+       	void SetFormat( ScDocument* pDoc, sal_uInt8 nCol, sal_uInt16 nRow, sal_uInt8 nTab, sal_uInt16 nStyle );
+       	sal_uInt16 getId() { return mnId; }
+       	sal_uInt16 getLength() { return mnLength; }
+       	void readString( String &rString, sal_uInt16 nLength );
+       	SvStream& operator>>( sal_uInt8& rByte ); 
+       	SvStream& operator>>( sal_uInt16& rWord );
+       	SvStream& operator>>( sal_Int8& rByte );
+	SvStream& operator>>( double& rDouble );
+};
+#endif
+
