--- scdetect.cxx.bak	2005-01-05 18:33:21.677208880 +0530
+++ sc/source/ui/unoobj/scdetect.cxx	2005-01-03 16:54:26.000000000 +0530
@@ -176,6 +176,7 @@
 static const sal_Char __FAR_DATA pFilterXML[]		= "StarOffice XML (Calc)";
 static const sal_Char __FAR_DATA pFilterAscii[]		= "Text - txt - csv (StarCalc)";
 static const sal_Char __FAR_DATA pFilterLotus[]		= "Lotus";
+static const sal_Char __FAR_DATA pFilterQPro[]		= "Quattro Pro";
 static const sal_Char __FAR_DATA pFilterExcel4[]	= "MS Excel 4.0";
 static const sal_Char __FAR_DATA pFilterEx4Temp[]	= "MS Excel 4.0 Vorlage/Template";
 static const sal_Char __FAR_DATA pFilterExcel5[]	= "MS Excel 5.0/95";
@@ -613,6 +614,13 @@
                             0x0004, 0x0000,                   // File Revision Subcode
                             M_ENDE };
 
+			const UINT16 pQPro[] =
+                               { 0x0000, 0x0000, 0x0002, 0x0000,
+                                 M_ALT(4), 0x0001, 0x0002, // WB1, WB2
+                                 0x0006, 0x0007,           // QPro 6/7 (?)
+                                 0x0010,
+                                 M_ENDE };
+
                         const UINT16 pDIF1[] =          // DIF mit CR-LF
                             {
                             'T', 'A', 'B', 'L', 'E',
@@ -647,7 +655,8 @@
                             pDIF2,
                             pSylk,
                             pLotusNew,
-                            pLotus2
+                            pLotus2,
+                            pQPro
                             };
                         const UINT16 nFilterCount = sizeof(ppFilterPatterns) / sizeof(ppFilterPatterns[0]);
 
@@ -662,7 +671,8 @@
                             pFilterDif,
                             pFilterSylk,
                             pFilterLotus,
-                            pFilterLotus
+                            pFilterLotus,
+			    pFilterQPro
                             };
 
                         const UINT16 nByteMask = 0xFF;
--- docsh.cxx.bak	2005-01-05 18:43:35.558884680 +0530
+++ sc/source/ui/docshell/docsh.cxx	2005-01-03 11:47:12.000000000 +0530
@@ -197,6 +197,7 @@
 static const sal_Char __FAR_DATA pFilterXML[]		= "StarOffice XML (Calc)";
 static const sal_Char __FAR_DATA pFilterAscii[]		= "Text - txt - csv (StarCalc)";
 static const sal_Char __FAR_DATA pFilterLotus[]		= "Lotus";
+static const sal_Char __FAR_DATA pFilterQPro[]		= "Quattro Pro";
 static const sal_Char __FAR_DATA pFilterExcel4[]	= "MS Excel 4.0";
 static const sal_Char __FAR_DATA pFilterEx4Temp[]	= "MS Excel 4.0 Vorlage/Template";
 static const sal_Char __FAR_DATA pFilterExcel5[]	= "MS Excel 5.0/95";
@@ -1115,6 +1116,35 @@
 			bSetColWidths = TRUE;
 			bSetSimpleTextColWidths = TRUE;
 		}
+		else if (aFltName.EqualsAscii(pFilterQPro))
+                {
+                         String sItStr;
+                                SfxItemSet*      pSet = rMedium.GetItemSet();
+                                const SfxPoolItem* pItem;
+                                if ( pSet && SFX_ITEM_SET ==
+                                         pSet->GetItemState( SID_FILE_FILTEROPTIONS, TRUE, &pItem ) )
+                                {
+                                        sItStr = ((const SfxStringItem*)pItem)->GetValue();
+                                }
+                                                                                                                             
+                                if (sItStr.Len() == 0)
+                                {
+                                                                                                                                                                       sItStr = ScGlobal::GetCharsetString( RTL_TEXTENCODING_UTF8 );
+                                }
+                        ScColumn::bDoubleAlloc = TRUE;
+                        FltError eError = ScImportQuattroPro( rMedium, &aDocument);
+                        ScColumn::bDoubleAlloc = FALSE;
+                        if (eError != eERR_OK)
+                        {
+                            if (!GetError())
+                                    SetError( eError );
+                                                                                                                             
+                                if( ( eError & ERRCODE_WARNING_MASK ) == ERRCODE_WARNING_MASK )
+                                        bRet = TRUE;
+                        }
+                        else
+                                bRet = TRUE;
+                }
 		else if (aFltName.EqualsAscii(pFilterRtf))
 		{
 			FltError eError = SCERR_IMPORT_UNKNOWN;

--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ filter/source/config/fragments/filters/QPro.xcu	2005-01-05 18:47:47.122641176 +0530
@@ -0,0 +1,14 @@
+<node oor:name="QPro" oor:op="replace" oor:finalized="true" oor:mandatory="true">
+                <prop oor:name="FileFormatVersion"><value>0</value></prop>
+                <prop oor:name="Type"><value>calc_QPro</value></prop>
+                <prop oor:name="DocumentService"><value>com.sun.star.sheet.SpreadsheetDocument</value></prop>
+                <prop oor:name="UIComponent"><value>com.sun.star.comp.Calc.FilterOptionsDialog</value></prop>
+                <prop oor:name="UserData"/>
+                <prop oor:name="FilterService"/>
+                <prop oor:name="TemplateName"/>
+                <prop oor:name="UIName">
+                        <value>Quattro Pro</value>
+                </prop>
+                <prop oor:name="Flags"><value>IMPORT ALIEN USESOPTIONS PREFERRED</value></prop>
+        </node>
+

--- /dev/null	2004-08-25 23:04:59.000000000 +0530
+++ filter/source/config/fragments/types/calc_QPro.xcu	2005-01-05 18:47:03.452280080 +0530
@@ -0,0 +1,13 @@
+<node oor:name="calc_QPro" oor:op="replace" oor:finalized="true" oor:mandatory="true">
+                <prop oor:name="DetectService"><value>com.sun.star.comp.calc.FormatDetector</value></prop>
+                <prop oor:name="URLPattern"/>
+                <prop oor:name="Extensions"><value>wb2</value></prop>
+                <prop oor:name="MediaType"/>
+                <prop oor:name="Preferred"><value>true</value></prop>
+                <prop oor:name="PreferredFilter"><value>Quattro Pro</value></prop>
+                <prop oor:name="UIName">
+                        <value>Quattro Pro</value>
+                </prop>
+                <prop oor:name="ClipboardFormat"/>
+        </node>
+
--- filter.hxx.bak	2005-01-05 18:35:40.237144568 +0530
+++ sc/inc/filter.hxx	2005-01-05 18:37:43.515403424 +0530
@@ -139,6 +139,7 @@
 
 FltError ScImportHTML( SvStream&, ScDocument*, ScRange& rRange, double nOutputFactor = 1.0, BOOL bCalcWidthHeight = TRUE );
 
+FltError ScImportQuattroPro( SfxMedium &rMedium, ScDocument *pDoc );
 // ***********************************************************************
 // Diverse Exportfilter
 // ***********************************************************************
Index: filter/source/config/fragments/fcfg_calc.mk
===================================================================
RCS file: /cvs/framework/filter/source/config/fragments/fcfg_calc.mk,v
retrieving revision 1.3
diff -u -r1.3 fcfg_calc.mk
--- filter/source/config/fragments/fcfg_calc.mk	13 Jul 2004 09:17:13 -0000	1.3
+++ filter/source/config/fragments/fcfg_calc.mk	10 Jan 2005 15:15:31 -0000
@@ -12,6 +12,7 @@
     calc_MS_Excel_95_VorlageTemplate \
     calc_MS_Excel_97 \
     calc_MS_Excel_97_VorlageTemplate \
+    calc_QPro \
     writer_Rich_Text_Format \
     calc_SYLK \
     calc_StarCalc_10 \
@@ -43,6 +44,7 @@
 	MS_Excel_95_Vorlage_Template \
 	MS_Excel_97 \
 	MS_Excel_97_Vorlage_Template \
+	QPro \
 	Rich_Text_Format__StarCalc_ \
 	SYLK \
 	StarCalc_1_0 \
