--- /dev/null	2004-07-26 17:21:14.000000000 +0530
+++ sc/source/filter/inc/qproform.hxx	2005-02-25 14:55:03.000000000 +0530
@@ -0,0 +1,67 @@
+#ifndef _QPROFORM_HXX
+#define _QPROFORM_HXX
+
+#ifndef SC_XLTOOLS_HXX
+#include "xltools.hxx"
+#endif                                                                                                                  
+#ifndef _FORMEL_HXX
+#include "formel.hxx"
+#endif
+#ifndef _STRING_HXX
+#include <tools/string.hxx>
+#endif
+#ifndef _QPRO_HXX
+#include "qpro.hxx"
+#endif
+
+enum FUNC_TYPE
+{
+    FT_Return,  
+    FT_FuncFix0,
+    FT_FuncFix1,   
+    FT_FuncFix2,  
+    FT_FuncFix3,    
+    FT_FuncFix4,    
+    FT_FuncVar,            
+    FT_Neg,               
+    FT_Op,              
+    FT_NotImpl,        
+    FT_ConstFloat,  
+    FT_Variable,   
+    FT_Range,   
+    FT_Braces,
+    FT_ConstInt,    
+    FT_ConstString,
+    FT_NOP,                 
+    FT_Cref,               
+    FT_Rref,               
+    FT_Nrref,              
+    FT_Absnref,            
+    FT_Erref,              
+    FT_Ecref,              
+    FT_Econstant,   
+};
+                                                                                                                  
+class QProToSc : public ConverterBase
+{
+    private:
+        TokenId nAddToken;  
+        TokenId nSubToken;  
+        TokenId n0Token;  
+        BiffReader& aIn; 
+        void ReadSRD( SingleRefData& rR, sal_Int8 nPage, sal_Int8 nCol, sal_uInt16 rRel ); 
+    public:
+	QProToSc( BiffReader& aStr );
+        ~QProToSc(){ };
+	ConvErr Convert( const ScTokenArray*& rpErg, sal_uInt16 nLen,
+                                                                        const FORMULA_TYPE eFT = FT_CellFormula );
+	void DoFunc( DefTokenId eOc, sal_uInt8 nAnz, const sal_Char* pExtString );
+ 
+        void IncToken( TokenId &rParam );
+        void DecToken( TokenId& rParam );
+        void NegToken( TokenId& rParam );
+        DefTokenId IndexToToken( sal_uInt16 nToken );
+	FUNC_TYPE IndexToType( sal_uInt8 nToken );
+	const sal_Char*	GetAddInName( const sal_uInt8 nId ); 
+};
+#endif
