--- svx/source/msfilter/msocximex.cxx	2007-09-27 14:02:19.000000000 +0100
+++ svx/source/msfilter/msocximex.cxx	2008-02-04 23:56:17.000000000 +0000
@@ -1758,11 +1758,15 @@ sal_Bool OCX_OptionButton::Import(com::s
         aTmp <<= lclCreateOUString( pCaption, nCaptionLen );
 		rPropSet->setPropertyValue( WW8_ASCII2STR("Label"), aTmp);
 	}
-
+    
     // #i40279# always centered vertically
     aTmp <<= ::com::sun::star::style::VerticalAlignment_MIDDLE;
     rPropSet->setPropertyValue( WW8_ASCII2STR("VerticalAlign"), aTmp );
-
+    if ( pGroupName )
+    {
+        aTmp <<= lclCreateOUString( pGroupName, nGroupNameLen );
+        rPropSet->setPropertyValue( WW8_ASCII2STR("GroupName"), aTmp);
+    }
 	aFontData.Import(rPropSet);
 	return sal_True;
 }
