--- svx/source/dialog/docrecovery.cxx	2006-05-08 14:51:13.000000000 +0000
+++ svx/source/dialog/docrecovery.cxx	2006-05-12 10:53:43.000000000 +0000
@@ -1071,6 +1071,7 @@ RecoveryDialog::RecoveryDialog(Window*  
     , m_aCancelBtn          ( this           , ResId  ( BTN_RECOV_CANCEL               ) )
     , m_aNextStr            (                  ResId  ( STR_RECOVERY_NEXT              ) )
     , m_aTitleRecoveryInProgress(              ResId  ( STR_RECOVERY_INPROGRESS        ) )
+    , m_aTitleRecoveryReport(                  ResId  ( STR_RECOVERY_REPORT            ) )
     , m_aRecoveryOnlyDescr  (                  ResId  ( STR_RECOVERYONLY_DESCR         ) )
     , m_aRecoveryOnlyFinish (                  ResId  ( STR_RECOVERYONLY_FINISH        ) )
     , m_pDefButton          ( NULL                                                       )
@@ -1209,6 +1210,7 @@ short RecoveryDialog::execute()
                  }
                  else
                  {
+                    m_aDescrFT.SetText(m_aTitleRecoveryReport);
                     m_aNextBtn.SetText(m_aNextStr);
                     m_aNextBtn.Enable(TRUE);
                     m_aCancelBtn.Enable(TRUE);
dummy line to avoid confusing diff-mode
--- svx/source/dialog/docrecovery.hrc	2005-11-11 12:50:14.000000000 +0000
+++ svx/source/dialog/docrecovery.hrc	2006-05-12 10:39:43.000000000 +0000
@@ -123,8 +123,9 @@
 #define STR_NOTRECOVYET             24
 #define STR_RECOVERY_NEXT           25
 #define STR_RECOVERY_INPROGRESS     26
-#define STR_RECOVERYONLY_DESCR      27
-#define STR_RECOVERYONLY_FINISH     28
+#define STR_RECOVERY_REPORT         27
+#define STR_RECOVERYONLY_DESCR      28
+#define STR_RECOVERYONLY_FINISH     29
 
 #define RECOV_WIDTH					SAVE_WIDTH
 #define RECOV_HEIGHT				SAVE_HEIGHT
--- svx/source/dialog/docrecovery.src	2006-05-08 14:51:28.000000000 +0000
+++ svx/source/dialog/docrecovery.src	2006-05-12 10:38:33.000000000 +0000
@@ -278,8 +278,13 @@ TabPage RID_SVXPAGE_DOCRECOVERY_RECOVER
 	};
 	String STR_RECOVERY_INPROGRESS
 	{
-	    Text[ de ] = "%PRODUCTNAME %PRODUCTVERSION beginnt jetzt mit der Wiederherstellung Ihrer Dokumente. Dieses kann abhängig von der Größe der Dokumente einige Zeit in Anspruch nehmen.\n\nEs wurde ein Fehlerbericht erstellt, der uns bei der Identifizierung des Problems in %PRODUCTNAME hilft. Betätigen Sie die 'Weiter' Schaltfläche, um zum Fehlerbericht-Programm zu gelangen oder 'Abbrechen' um diesen Schritt zu überspringen";
-	    Text[ en-US ] = "%PRODUCTNAME %PRODUCTVERSION starts now to recover your documents. Depending on the size of the documents this process can take some time.\n\nA report of the crash was created helping us to identify the cause why %PRODUCTNAME crashed. Click 'Next' to get to the Error Report Tool or press 'Cancel' to skip this step." ;
+	    Text[ de ] = "%PRODUCTNAME %PRODUCTVERSION beginnt jetzt mit der Wiederherstellung Ihrer Dokumente. Dieses kann abhängig von der Größe der Dokumente einige Zeit in Anspruch nehmen.";
+	    Text[ en-US ] = "%PRODUCTNAME %PRODUCTVERSION starts now to recover your documents. Depending on the size of the documents this process can take some time." ;
+	};
+	String STR_RECOVERY_REPORT
+	{
+	    Text[ de ] = "Es wurde ein Fehlerbericht erstellt, der uns bei der Identifizierung des Problems in %PRODUCTNAME hilft. Betätigen Sie die 'Weiter' Schaltfläche, um zum Fehlerbericht-Programm zu gelangen oder 'Abbrechen' um diesen Schritt zu überspringen.";
+	    Text[ en-US ] = "A report of the crash was created helping us to identify the cause why %PRODUCTNAME crashed. Click 'Next' to get to the Error Report Tool or press 'Cancel' to skip this step." ;
 	};
 	String STR_RECOVERYONLY_DESCR
 	{
--- svx/source/inc/docrecovery.hxx	2006-05-08 14:51:55.000000000 +0000
+++ svx/source/inc/docrecovery.hxx	2006-05-12 10:52:42.000000000 +0000
@@ -693,6 +693,7 @@ class RecoveryDialog : public IExtendedT
         CancelButton    m_aCancelBtn;
         String          m_aNextStr;
         String          m_aTitleRecoveryInProgress;
+        String          m_aTitleRecoveryReport;
         String          m_aRecoveryOnlyDescr;
         String          m_aRecoveryOnlyFinish;
 
