--- reportdesign/util/makefile.mk.orig	2007-07-15 13:26:16.000000000 +0200
+++ reportdesign/util/makefile.mk	2007-07-15 13:31:19.000000000 +0200
@@ -47,6 +47,8 @@
 .INCLUDE :	settings.mk
 # ------------------------------------------------------------------
 
+.IF "$(SOLAR_JAVA)"!=""
+
 # --- reportdesign core (rpt) -----------------------------------
 
 LIB1TARGET=$(SLB)$/$(TARGET).lib
@@ -228,7 +230,6 @@
 	$(ZIP1DIR)$/jcommon-serializer-0.1.0.jar							\
 	$(ZIP1DIR)$/reportdesign.jar
 
-.INCLUDE : target.mk
 # --- Targets ----------------------------------
 .IF "$(ZIP1TARGETN)"!=""
 $(ZIP1TARGETN) :  $(TXTFILES) $(XMLFILES) $(HTMLFILES) $(REPRORTJARFILES)
@@ -257,3 +258,7 @@
 $(ZIP1DIR)$/THIRDPARTYREADMELICENSE.html : THIRDPARTYREADMELICENSE.html
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
+
+.ENDIF
+
+.INCLUDE : target.mk
