diff -ru rsc.old/source/rscpp/cpp1.c rsc/source/rscpp/cpp1.c
--- rsc.old/source/rscpp/cpp1.c	2004-09-22 09:18:00.765479724 -0400
+++ rsc/source/rscpp/cpp1.c	2004-09-22 09:30:49.503520149 -0400
@@ -422,9 +422,14 @@
 #endif
         if( pCppOut != stdout && pCppOut != stderr )
             fclose( pCppOut );
+/* The file actually gets closed earlier when the infile stack
+ * is wound down.  Closing it again here causes a double-free
+ * in fclose().
+ */
+#if 0
         if( pCppIn != stdin )
             fclose( pCppIn  );
-
+#endif
 
         if (errors > 0) {
             fprintf(stderr, (errors == 1)
