Index: sc/source/filter/excel/biffdump.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/biffdump.cxx,v
retrieving revision 1.78
diff -u -r1.78 biffdump.cxx
--- sc/source/filter/excel/biffdump.cxx	29 Mar 2005 13:35:36 -0000	1.78
+++ sc/source/filter/excel/biffdump.cxx	6 Apr 2005 13:37:33 -0000
@@ -274,6 +273,10 @@
 	r += p;
 }
 
+inline static void __AddDec( ByteString& r, sal_Int8 n )
+{
+	__AddDec( r, ( INT32 ) n );
+}
 
 inline static void __AddDec( ByteString& r, INT16 n )
 {
