diff -rup /data4/scratch/oog680-m5/sc/source/core/tool/compiler.cxx sc/source/core/tool/compiler.cxx
--- /data4/scratch/oog680-m5/sc/source/core/tool/compiler.cxx	2007-09-24 15:00:30.000000000 +0100
+++ sc/source/core/tool/compiler.cxx	2007-09-19 01:05:47.000000000 +0100
@@ -1543,6 +1546,8 @@ ScCompiler::ScCompiler( ScDocument* pDoc
     nMaxTab = pDoc ? pDoc->GetTableCount() - 1 : 0;
     pStack = NULL;
     nNumFmt = NUMBERFORMAT_UNDEFINED;
+    SetCompileEnglish( TRUE ); 
+
 }
 
 ScCompiler::ScCompiler(ScDocument* pDocument, const ScAddress& rPos )
