Index: sc/source/ui/view/gridwin.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/gridwin.cxx,v
retrieving revision 1.88
diff -u -p -r1.88 gridwin.cxx
--- sc/source/ui/view/gridwin.cxx	12 Dec 2007 13:21:19 -0000	1.88
+++ sc/source/ui/view/gridwin.cxx	20 Mar 2008 02:45:50 -0000
@@ -5028,6 +5131,59 @@ void ScGridWindow::DeleteCursorOverlay()
     DELETEZ( mpOOCursors );
 }
 
+static void AddCopySelection( std::vector<Rectangle>& rPixelRects, ScDocument* pClipDoc, 
+    ScViewData* pViewData, ScSplitPos eWhich )
+{
+    if( !pClipDoc || !pViewData->IsPasteMode())
+        return;
+
+    SCCOL nX = pViewData->GetCurX();
+    SCROW nY = pViewData->GetCurY();
+
+    SCCOL nClipStartX;
+    SCROW nClipStartY;
+    pClipDoc->GetClipStart( nClipStartX, nClipStartY );
+
+    // we don't want to "draw-over" the actual cursor, as this causes the
+    // cursor to disappear.
+    if (nX == nClipStartX && nY == nClipStartY)
+        return;
+
+    SCCOL nClipAreaX;
+    SCROW nClipAreaY;
+    pClipDoc->GetClipArea( nClipAreaX, nClipAreaY, TRUE);
+
+    Point aClipStartScrPos = pViewData->GetScrPos( nClipStartX, nClipStartY, eWhich );
+    Point aClipEndScrPos   = pViewData->GetScrPos( nClipStartX + nClipAreaX + 1, 
+        nClipStartY + nClipAreaY + 1, eWhich);
+
+    long nSizeXPix = aClipEndScrPos.X() - aClipStartScrPos.X();
+    long nSizeYPix = aClipEndScrPos.Y() - aClipStartScrPos.Y();
+
+    aClipStartScrPos.X() -= 2;
+    aClipStartScrPos.Y() -= 2;
+
+    Rectangle aRect( aClipStartScrPos, Size( nSizeXPix + 3, nSizeYPix + 3 ) );
+
+    // cursor borders XOR wth each other -- if a border is drawn over the same
+    // area twice, no border is drawn.  Consequently, we can't draw a border
+    // around the copy selection of the cursor is next to the copied cell(s),
+    // otherwise we'll get a "missing" border.
+
+    // left edge
+    if (!(nY == nClipStartY && nX == nClipStartX-1))
+        rPixelRects.push_back(Rectangle( aRect.Left(), aRect.Top(), aRect.Left()+2, aRect.Bottom() ));
+    // right edge
+    if (!(nY == nClipStartY && nX == nClipStartX+1))
+        rPixelRects.push_back(Rectangle( aRect.Right()-2, aRect.Top(), aRect.Right(), aRect.Bottom() ));
+    // top edge
+    if (!(nX == nClipStartX && nY == nClipStartY-1))
+        rPixelRects.push_back(Rectangle( aRect.Left()+3, aRect.Top(), aRect.Right()-3, aRect.Top()+2 ));
+    // bottom edge
+    if (!(nX == nClipStartX && nY == nClipStartY+1))
+        rPixelRects.push_back(Rectangle( aRect.Left()+3, aRect.Bottom()-2, aRect.Right()-3, aRect.Bottom() ));
+}
+
 void ScGridWindow::UpdateCursorOverlay()
 {
     MapMode aDrawMode = GetDrawMapMode();
@@ -5042,6 +5198,12 @@ void ScGridWindow::UpdateCursorOverlay()
 
     std::vector<Rectangle> aPixelRects;
 
+    ScTransferObj* pTransObj = ScTransferObj::GetOwnClipboard( pViewData->GetActiveWin() );
+    if (pTransObj)
+    {
+        AddCopySelection( aPixelRects, pTransObj->GetDocument(), pViewData, eWhich );
+    }
+
     //
     //  determine the cursor rectangles in pixels (moved from ScGridWindow::DrawCursor)
     //
