--- sc/source/ui/docshell/docsh.cxx	2005-08-09 22:09:17.000000000 +0530
+++ sc/source/ui/docshell/docsh.cxx	2005-08-09 22:05:01.000000000 +0530
@@ -1389,6 +1389,7 @@ void ScDocShell::AsciiSave( SvStream& rS
     sal_Unicode cStrDelim = rAsciiOpt.nTextSepCode;
     CharSet eCharSet      = rAsciiOpt.eCharSet;
     BOOL bFixedWidth      = rAsciiOpt.bFixedWidth;
+	BOOL bSaveAsRendered  = rAsciiOpt.bSaveAsRendered;
 
 	CharSet eOldCharSet = rStream.GetStreamCharSet();
 	rStream.SetStreamCharSet( eCharSet );
@@ -1575,7 +1576,7 @@ void ScDocShell::AsciiSave( SvStream& rS
                 {
                     ULONG nFormat;
                     aDocument.GetNumberFormat( nCol, nRow, nTab, nFormat );
-                    if ( bFixedWidth )
+                    if ( bFixedWidth || bSaveAsRendered )
                     {
                         Color* pDummy;
                         ScCellFormat::GetString( pCell, nFormat, aString, &pDummy, rFormatter );
--- sc/source/ui/dbgui/imoptdlg.cxx	2005-08-09 22:08:50.000000000 +0530
+++ sc/source/ui/dbgui/imoptdlg.cxx	2005-08-09 22:08:28.064027648 +0530
@@ -368,6 +368,7 @@ ScImportOptions::ScImportOptions( const 
 		nTextSepCode  = (sal_Unicode) rStr.GetToken(1,',').ToInt32();
 		aStrFont	  = rStr.GetToken(2,',');
 		eCharSet	  = ScGlobal::GetCharsetValue(aStrFont);
+		bSaveAsRendered   = (BOOL) (rStr.GetToken( 3, ',' ).ToInt32());
 	}
 }
 
@@ -385,6 +386,8 @@ String ScImportOptions::BuildString() co
 	aResult += String::CreateFromInt32(nTextSepCode);
 	aResult += ',';
 	aResult += aStrFont;
+	aResult += ',';
+	aResult += String::CreateFromInt32( (sal_Int32)bSaveAsRendered );
 
 	return aResult;
 }
--- sc/source/ui/dbgui/imoptdlg.hrc	2005-08-09 20:21:43.000000000 +0530
+++ sc/source/ui/dbgui/imoptdlg.hrc	2005-08-09 22:10:14.064913064 +0530
@@ -74,4 +74,5 @@
 #define LB_FONT			10
 #define FL_FIELDOPT     11
 #define CB_FIXEDWIDTH   12
+#define CB_SAVERENDER	13
 
--- sc/source/ui/inc/imoptdlg.hxx	2005-08-09 20:22:05.000000000 +0530
+++ sc/source/ui/inc/imoptdlg.hxx	2005-08-09 22:07:49.000000000 +0530
@@ -143,44 +143,47 @@ class SC_DLLPUBLIC ScImportOptions
 {
 public:
 		ScImportOptions()
-            : nFieldSepCode(0),nTextSepCode(0),eCharSet(RTL_TEXTENCODING_DONTKNOW),bFixedWidth(FALSE)
+            : nFieldSepCode(0),nTextSepCode(0),eCharSet(RTL_TEXTENCODING_DONTKNOW),bFixedWidth(FALSE),bSaveAsRendered(FALSE)
 		{}
 		ScImportOptions( const String& rStr );
 
 		ScImportOptions( sal_Unicode nFieldSep, sal_Unicode nTextSep, const String& rStr )
-            : nFieldSepCode(nFieldSep),nTextSepCode(nTextSep),aStrFont(rStr),bFixedWidth(FALSE)
+            : nFieldSepCode(nFieldSep),nTextSepCode(nTextSep),aStrFont(rStr),bFixedWidth(FALSE),bSaveAsRendered(FALSE)
 		{ eCharSet = ScGlobal::GetCharsetValue(aStrFont); }
 
 		ScImportOptions( sal_Unicode nFieldSep, sal_Unicode nTextSep, rtl_TextEncoding nEnc )
-            : nFieldSepCode(nFieldSep),nTextSepCode(nTextSep),bFixedWidth(FALSE)
+            : nFieldSepCode(nFieldSep),nTextSepCode(nTextSep),bFixedWidth(FALSE),bSaveAsRendered(FALSE)
 		{ SetTextEncoding( nEnc ); }
 
 		ScImportOptions( const ScImportOptions& rCpy )
-			: nFieldSepCode	(rCpy.nFieldSepCode),
-			  nTextSepCode	(rCpy.nTextSepCode),
-			  aStrFont		(rCpy.aStrFont),
-              eCharSet      (rCpy.eCharSet),
-              bFixedWidth   (rCpy.bFixedWidth)
+			: nFieldSepCode		(rCpy.nFieldSepCode),
+			  nTextSepCode		(rCpy.nTextSepCode),
+			  aStrFont			(rCpy.aStrFont),
+              eCharSet      	(rCpy.eCharSet),
+              bFixedWidth   	(rCpy.bFixedWidth),
+			  bSaveAsRendered	(rCpy.bSaveAsRendered)
 		{}
 
 	ScImportOptions& operator=( const ScImportOptions& rCpy )
 						{
-							nFieldSepCode = rCpy.nFieldSepCode;
-							nTextSepCode  = rCpy.nTextSepCode;
-							aStrFont	  = rCpy.aStrFont;
-							eCharSet	  = rCpy.eCharSet;
-                            bFixedWidth   = rCpy.bFixedWidth;
+							nFieldSepCode 	= rCpy.nFieldSepCode;
+							nTextSepCode  	= rCpy.nTextSepCode;
+							aStrFont	  	= rCpy.aStrFont;
+							eCharSet	  	= rCpy.eCharSet;
+                            bFixedWidth   	= rCpy.bFixedWidth;
+							bSaveAsRendered	= rCpy.bSaveAsRendered;
 							return *this;
 						}
 
 	BOOL			 operator==( const ScImportOptions& rCmp )
 						{
 							return
-								   nFieldSepCode == rCmp.nFieldSepCode
-								&& nTextSepCode  == rCmp.nTextSepCode
-								&& eCharSet		 == rCmp.eCharSet
-                                && aStrFont      == rCmp.aStrFont
-                                && bFixedWidth   == rCmp.bFixedWidth;
+								   nFieldSepCode 	== rCmp.nFieldSepCode
+								&& nTextSepCode  	== rCmp.nTextSepCode
+								&& eCharSet		 	== rCmp.eCharSet
+                                && aStrFont      	== rCmp.aStrFont
+                                && bFixedWidth   	== rCmp.bFixedWidth
+								&& bSaveAsRendered	== rCmp.bSaveAsRendered;
 						}
 	String	BuildString() const;
 
@@ -191,6 +194,7 @@ public:
     String      aStrFont;
     CharSet     eCharSet;
     BOOL        bFixedWidth;
+	BOOL		bSaveAsRendered;
 };
 
 
--- sc/source/ui/dbgui/scuiimoptdlg.cxx	2005-08-09 20:22:31.000000000 +0530
+++ sc/source/ui/dbgui/scuiimoptdlg.cxx	2005-08-09 22:10:52.432080368 +0530
@@ -173,7 +173,8 @@ ScImportOptionsDlg::ScImportOptionsDlg(
 		aFtFont		( this, ScResId( FT_FONT ) ),
         aLbFont     ( this, ScResId( bAscii ? DDLB_FONT : LB_FONT ) ),
         aFlFieldOpt ( this, ScResId( FL_FIELDOPT ) ),
-        aCbFixed    ( this, ScResId( CB_FIXEDWIDTH ) )
+        aCbFixed    ( this, ScResId( CB_FIXEDWIDTH ) ),
+		aCbRender	( this, ScResId( CB_SAVERENDER ) )
 {
 	// im Ctor-Initializer nicht moeglich (MSC kann das nicht):
 	pFieldSepTab = new ScDelimiterTable( String(ScResId(SCSTR_FIELDSEP)) );
@@ -250,6 +251,8 @@ ScImportOptionsDlg::ScImportOptionsDlg(
         aCbFixed.Show();
         aCbFixed.SetClickHdl( LINK( this, ScImportOptionsDlg, FixedWidthHdl ) );
         aCbFixed.Check( FALSE );
+		aCbRender.Show();
+		aCbRender.Check( FALSE );
     }
     else
     {
@@ -260,6 +263,7 @@ ScImportOptionsDlg::ScImportOptionsDlg(
 		aEdFieldSep.Hide();
         aEdTextSep.Hide();
         aCbFixed.Hide();
+		aCbRender.Hide();
 		aLbFont.GrabFocus();
         aLbFont.SetDoubleClickHdl( LINK( this, ScImportOptionsDlg, DoubleClickHdl ) );
     }
@@ -293,6 +297,7 @@ void ScImportOptionsDlg::GetImportOption
 		rOptions.nFieldSepCode = GetCodeFromCombo( aEdFieldSep );
 		rOptions.nTextSepCode  = GetCodeFromCombo( aEdTextSep );
         rOptions.bFixedWidth = aCbFixed.IsChecked();
+		rOptions.bSaveAsRendered = aCbRender.IsChecked();
 	}
 }
 
@@ -335,6 +340,7 @@ IMPL_LINK( ScImportOptionsDlg, FixedWidt
         aEdFieldSep.Enable( bEnable );
         aFtTextSep.Enable( bEnable );
         aEdTextSep.Enable( bEnable );
+		aCbRender.Enable( bEnable );
     }
     return 0;
 }
--- sc/source/ui/inc/scuiimoptdlg.hxx	2005-08-09 22:19:09.248552784 +0530
+++ sc/source/ui/inc/scuiimoptdlg.hxx	2005-08-09 22:19:27.905716464 +0530
@@ -96,6 +96,7 @@ private:
 	OKButton			aBtnOk;
 	CancelButton		aBtnCancel;
 	HelpButton			aBtnHelp;
+	CheckBox			aCbRender;
 
 	ScDelimiterTable*	pFieldSepTab;
 	ScDelimiterTable*	pTextSepTab;
--- sc/source/ui/dbgui/imoptdlg.src	2005-08-09 20:21:58.000000000 +0530
+++ sc/source/ui/dbgui/imoptdlg.src	2005-08-12 21:46:10.131170096 +0530
@@ -135,11 +135,21 @@ ModalDialog RID_SCDLG_IMPORTOPT
     CheckBox CB_FIXEDWIDTH
     {
         Pos = MAP_APPFONT( 12, 66 );
-        Size = MAP_APPFONT( 172, 10 );
+        Size = MAP_APPFONT( 100, 10 );
         Hide = TRUE;
         Text [ de ] = "Feste Spalten~breite";
         Text [ en-US ] = "Fixed column ~width";
 	};
+	CheckBox CB_SAVERENDER
+	{
+		Pos = MAP_APPFONT( 120, 66 );
+		Size = MAP_APPFONT( 100, 10 ); 
+		TabStop = TRUE;
+		Hide = TRUE;
+		Text [ de ] = " ";
+		Text [ en-US ] = "Save as ~Rendered";
+		Text [ x-comment ] = " ";
+	};
 	OKButton BTN_OK
 	{
 		Pos = MAP_APPFONT ( 202 , 6 ) ;
