--- sc/source/ui/view/preview.cxx	2005-02-08 19:07:30.537603611 +0530
+++ sc/source/ui/view/preview.cxx	2005-02-08 19:08:35.536532612 +0530
@@ -521,6 +521,7 @@
         switch(nKey)
         {
             case KEY_ADD:      nSlot = SID_PREVIEW_ZOOMIN;  break;
+	    case KEY_ESCAPE:    nSlot = SID_PREVIEW_CLOSE; break;
             case KEY_SUBTRACT: nSlot = SID_PREVIEW_ZOOMOUT; break;
         }
         if(nSlot)
