diff -u -r ../../../sc.bak/source/core/data/table4.cxx sc/source/core/data/table4.cxx
--- ../../../sc.bak/source/core/data/table4.cxx	2004-12-22 14:05:08.000000000 +0530
+++ sc/source/core/data/table4.cxx	2005-01-07 12:21:07.726010848 +0530
@@ -1296,6 +1296,8 @@
 				{
 					for (rInner = nIMin; rInner <= nIMax; rInner++)
 					{
+                                                if (pDocument->RowFiltered( rInner, nTab))
+                                                        continue;
 						ULONG nInd = nActFormCnt;
 						FillFormula(nInd, bFirst, (ScFormulaCell*)pSrcCell,
 							static_cast<SCCOL>(nCol), nRow, (rInner == nIEnd) );
@@ -1307,6 +1309,8 @@
 				{
 					for (rInner = nIMin; rInner <= nIMax; rInner++)
 					{
+                                                if (pDocument->RowFiltered( rInner, nTab))
+                                                        continue;
 						aCol[nCol].Insert(static_cast<SCROW>(nRow), pSrcCell->Clone(pDocument));
 					}
 					nProgress += nIMax - nIMin + 1;
Index: sc/inc/document.hxx
===================================================================
--- sc/inc/document.hxx.orig	2004-11-26 13:35:38.064526880 +0000
+++ sc/inc/document.hxx	2004-11-26 17:10:55.795733240 +0000
@@ -508,6 +508,8 @@ public:
 	ULONG			GetWeightedCount() const;	// Formeln und Edit staerker gewichtet
 	ULONG			GetCodeCount() const;		// RPN-Code in Formeln
 	DECL_LINK( GetUserDefinedColor, USHORT * );
+	BOOL 		RowFiltered( USHORT nRow, USHORT nTab ) const;	// FillInfo
+	BOOL		ColFiltered( USHORT nCol, USHORT nTab ) const;	// FillInfo
 																// Numberformatter
 
 public:
@@ -1698,6 +1700,16 @@ inline BOOL ScDocument::RowHidden( USHOR
     return pTab[nTab]->pRowHeight->GetValue(nRow);
 }
 
+inline BOOL ScDocument::ColFiltered( USHORT nCol, USHORT nTab ) const
+{
+	return ( pTab[nTab]->pColFlags[nCol] & CR_FILTERED) != 0;
+}
+
+inline BOOL ScDocument::RowFiltered( USHORT nRow, USHORT nTab ) const
+{
+	return pTab[nTab]->IsFiltered(nRow);
+}
+ 
 #endif
 
 
