diff -u -r ../../../sc.bak/source/core/data/table4.cxx sc/source/core/data/table4.cxx
--- ../../../sc.bak/source/core/data/table4.cxx	2004-12-22 14:05:08.000000000 +0530
+++ sc/source/core/data/table4.cxx	2005-01-07 12:21:07.726010848 +0530
@@ -1296,6 +1296,8 @@
 				{
 					for (rInner = nIMin; rInner <= nIMax; rInner++)
 					{
+                                                if (pDocument->RowFiltered( rInner, nTab))
+                                                        continue;
 						ULONG nInd = nActFormCnt;
 						FillFormula(nInd, bFirst, (ScFormulaCell*)pSrcCell,
 							static_cast<SCCOL>(nCol), nRow, (rInner == nIEnd) );
@@ -1307,6 +1309,8 @@
 				{
 					for (rInner = nIMin; rInner <= nIMax; rInner++)
 					{
+                                                if (pDocument->RowFiltered( rInner, nTab))
+                                                        continue;
 						aCol[nCol].Insert(static_cast<SCROW>(nRow), pSrcCell->Clone(pDocument));
 					}
 					nProgress += nIMax - nIMin + 1;
