--- sc/source/ui/src/popup.src
+++ sc/source/ui/src/popup.src
@@ -142,6 +142,22 @@
 		 //------------------------------
 		MenuItem
 		{
+			Identifier = SID_INSERT_POSTIT ;
+			HelpId = SID_INSERT_POSTIT ;
+			Text [ en-US ] = "Insert Not~e" ;
+		Text[ de ] = "Hinw~eis einfügen";
+		Text[ cs ] = "Poznámka";
+		Text[ es ] = "Insertar Not~a";
+		Text[ fr ] = "Insérer une not~e";
+		Text[ it ] = "Inserisci nota";
+		Text[ ja ] = "Not~eを挿入";
+		Text[ pt-BR ] = "Inserir Observaçã~o";
+		Text[ zh-CN ] = "插入注释(~E)";
+		Text[ zh-TW ] = "插入記事(~E)";
+			Text [ x-comment ] = " ";
+		};
+		MenuItem
+		{
 			Identifier = FID_NOTE_VISIBLE ;
 			HelpId = FID_NOTE_VISIBLE ;
 			Text [ de ] = "~Notiz anzeigen" ;


--- sc/source/ui/view/cellsh.cxx
+++ sc/source/ui/view/cellsh.cxx
@@ -308,6 +308,14 @@
 					}
 				}
 				break;
+			case SID_INSERT_POSTIT:
+				{
+					ScPostIt aNote(pDoc);
+					ScViewData* pView	= GetViewData();
+					if ( pDoc->GetNote( pView->GetCurX(), pView->GetCurY(), pView->GetTabNo(), aNote ) )
+					       	rSet.DisableItem( nWhich );		
+				}
+				break;
 			case SID_OPENDLG_FUNCTION:
 				{
 					ScMarkData aMarkData=GetViewData()->GetMarkData();
