--- sc/source/ui/src/popup.src.back	2005-02-15 10:55:00.000000000 +0530
+++ sc/source/ui/src/popup.src	2005-03-26 12:06:20.000000000 +0530
@@ -165,6 +165,13 @@
 		 //------------------------------
 		MenuItem
 		{
+			Identifier = SID_INSERT_POSTIT ;
+			HelpId = SID_INSERT_POSTIT ;
+			Text [ en-US ] = "Insert Not~e" ;
+			Text [ x-comment ] = " ";
+		};
+		MenuItem
+		{
 			Identifier = FID_NOTE_VISIBLE ;
 			HelpId = FID_NOTE_VISIBLE ;
 			Text [ de ] = "~Notiz anzeigen" ;
--- sc/source/ui/view/cellsh.cxx.back1	2005-11-17 18:58:00.000000000 +0530
+++ sc/source/ui/view/cellsh.cxx	2005-03-30 10:40:14.000000000 +0530
@@ -308,6 +308,14 @@
 					}
 				}
 				break;
+			case SID_INSERT_POSTIT:
+				{
+					ScPostIt aNote(pDoc);
+					ScViewData* pView	= GetViewData();
+					if ( pDoc->GetNote( pView->GetCurX(), pView->GetCurY(), pView->GetTabNo(), aNote ) )
+					       	rSet.DisableItem( nWhich );		
+				}
+				break;
 			case SID_OPENDLG_FUNCTION:
 				{
 					ScMarkData aMarkData=GetViewData()->GetMarkData();
