diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/compiler.hxx sc/inc/compiler.hxx
--- sc.clean/inc/compiler.hxx	2008-03-03 11:24:43.000000000 -0500
+++ sc/inc/compiler.hxx	2008-03-03 16:16:53.000000000 -0500
@@ -236,26 +236,6 @@ typedef ::std::hash_map< String, String,
 class SC_DLLPUBLIC ScCompiler
 {
 public:
-	struct Convention
-	{
-		const ScAddress::Convention meConv;
-		const ULONG*   				mpCharTable;
-
-
-		Convention( ScAddress::Convention eConvP );
-        virtual ~Convention();
-
-		virtual void MakeRefStr( rtl::OUStringBuffer&	rBuffer,
-								 const ScCompiler&		rCompiler,
-								 const ComplRefData&	rRef,
-								 BOOL bSingleRef ) const = 0;
-		virtual ::com::sun::star::i18n::ParseResult
-					parseAnyToken( const String& rFormula,
-								   xub_StrLen nSrcPos,
-								   const CharClass* pCharClass) const = 0;
-	};
-    friend struct Convention;
-
     /** Mappings from strings to OpCodes and vice versa. */
     class OpCodeMap
     {
@@ -362,6 +342,30 @@ private:
     typedef ::boost::shared_ptr< const OpCodeMap > OpCodeMapPtr;
     typedef ::boost::shared_ptr< OpCodeMap > NonConstOpCodeMapPtr;
 
+    struct Convention
+    {
+        const ScAddress::Convention meConv;
+        const ULONG*                mpCharTable;
+
+
+        Convention( ScAddress::Convention eConvP );
+        virtual ~Convention();
+
+        virtual void MakeRefStr( rtl::OUStringBuffer&   rBuffer,
+                                 const ScCompiler&      rCompiler,
+                                 const ComplRefData&    rRef,
+                                 BOOL bSingleRef ) const = 0;
+        virtual ::com::sun::star::i18n::ParseResult
+                    parseAnyToken( const String& rFormula,
+                                   xub_StrLen nSrcPos,
+                                   const CharClass* pCharClass) const = 0;
+
+        virtual const String& GetSymbol( const OpCodeMapPtr& rpMap, OpCode e ) const = 0;
+
+        virtual OpCode GetOpCode( const OpCodeMapPtr& rpMap, const String& rName, bool bInArray ) const = 0;
+    };
+    friend struct Convention;
+
 private:
 
     static NonConstOpCodeMapPtr  mxSymbolsODFF;                          // ODFF symbols
@@ -415,7 +419,7 @@ private:
     bool        mbCloseBrackets;            // whether to close open brackets automatically, default TRUE
 
 	BOOL   GetToken();
-	BOOL   NextNewToken(bool bAllowBooleans = false);
+	BOOL   NextNewToken(bool bInArray = false);
 	OpCode NextToken();
 	void PutCode( ScTokenRef& );
 	void Factor();
@@ -433,7 +437,7 @@ private:
 	void SetError(USHORT nError);
 	xub_StrLen NextSymbol();
 	BOOL IsValue( const String& );
-	BOOL IsOpCode( const String& );
+	BOOL IsOpCode( const String& rName, bool bInArray );
 	BOOL IsOpCode2( const String& );
 	BOOL IsString();
 	BOOL IsReference( const String& );
@@ -539,7 +543,8 @@ public:
     const ScDocument* GetDoc() const { return pDoc; }
     const ScAddress& GetPos() const { return aPos; }
 
-    static const String& GetStringFromOpCode( OpCode eOpCode, bool bEnglish = false );
+    static const String& GetStringFromOpCode( OpCode eOpCode, bool bEnglish = false, 
+                                              ScAddress::Convention eConv = ScAddress::CONV_OOO );
 
 	ScToken* CreateStringFromToken( String& rFormula, ScToken* pToken,
 									BOOL bAllowArrAdvance = FALSE );
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/docoptio.hxx sc/inc/docoptio.hxx
--- sc.clean/inc/docoptio.hxx	2008-03-03 11:24:44.000000000 -0500
+++ sc/inc/docoptio.hxx	2008-03-19 18:27:17.000000000 -0400
@@ -52,6 +52,8 @@
 #include "optutil.hxx"
 #endif
 
+#include "address.hxx"
+
 class SC_DLLPUBLIC ScDocOptions
 {
 	double fIterEps;				// Epsilon-Wert dazu
@@ -69,6 +71,7 @@ class SC_DLLPUBLIC ScDocOptions
 	BOOL   bDoAutoSpell;			// Auto-Spelling
 	BOOL   bLookUpColRowNames;		// Spalten-/Zeilenbeschriftungen automagisch suchen
     BOOL   bFormulaRegexEnabled;    // regular expressions in formulas enabled
+    ScAddress::Convention eFormulaConv; // formula address convention used to switch different formula syntax
 
 public:
 				ScDocOptions();
@@ -117,6 +120,9 @@ public:
 
     void    SetFormulaRegexEnabled( BOOL bVal ) { bFormulaRegexEnabled = bVal; }
     BOOL    IsFormulaRegexEnabled() const       { return bFormulaRegexEnabled; }
+
+    void SetFormulaSyntax( ScAddress::Convention eConv ) { eFormulaConv = eConv; }
+    ScAddress::Convention GetFormulaSyntax() const { return eFormulaConv; }
 };
 
 
@@ -137,6 +143,7 @@ inline void ScDocOptions::CopyTo(ScDocOp
 	rOpt.bDoAutoSpell			= bDoAutoSpell;
 	rOpt.bLookUpColRowNames		= bLookUpColRowNames;
     rOpt.bFormulaRegexEnabled   = bFormulaRegexEnabled;
+    rOpt.eFormulaConv           = eFormulaConv;
 }
 
 inline const ScDocOptions& ScDocOptions::operator=( const ScDocOptions& rCpy )
@@ -156,6 +163,7 @@ inline const ScDocOptions& ScDocOptions:
 	bDoAutoSpell		= rCpy.bDoAutoSpell;
 	bLookUpColRowNames	= rCpy.bLookUpColRowNames;
     bFormulaRegexEnabled= rCpy.bFormulaRegexEnabled;
+    eFormulaConv        = rCpy.eFormulaConv;
 
 	return *this;
 }
@@ -178,6 +186,7 @@ inline int ScDocOptions::operator==( con
 			&&	rOpt.bDoAutoSpell			== bDoAutoSpell
 			&&	rOpt.bLookUpColRowNames		== bLookUpColRowNames
             &&  rOpt.bFormulaRegexEnabled   == bFormulaRegexEnabled
+            &&  rOpt.eFormulaConv           == eFormulaConv
 			);
 }
 
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/global.hxx sc/inc/global.hxx
--- sc.clean/inc/global.hxx	2008-03-03 11:24:43.000000000 -0500
+++ sc/inc/global.hxx	2008-03-03 11:33:34.000000000 -0500
@@ -712,11 +712,11 @@ public:
     void        InitArgumentInfo() const;
 
     /** Returns a semicolon separated list of all parameter names. */
-    String  GetParamList        () const;
+    String  GetParamList        ( ScAddress::Convention eConv = ScAddress::CONV_OOO ) const;
     /** Returns the full function siganture: "FUNCTIONNAME( parameter list )". */
-	String	GetSignature		() const;
+	String	GetSignature		( ScAddress::Convention eConv = ScAddress::CONV_OOO) const;
     /** Returns the function siganture with parameters from the passed string array. */
-	String	GetFormulaString	( String** aArgArr ) const;
+	String	GetFormulaString	( String** aArgArr, ScAddress::Convention eConv = ScAddress::CONV_OOO ) const;
 
 	USHORT		nFIndex;		// eindeutiger Funktionsindex
 	USHORT		nCategory;		// Kategorie
@@ -786,6 +786,7 @@ class ScFormulaUtil
 public:
 	static BOOL					GetNextFunc( const String&	rFormula,
 											 BOOL			bBack,
+                                             ScAddress::Convention eConv,
 											 xub_StrLen&	rFStart, // Ein- und Ausgabe
 											 xub_StrLen*	pFEnd = NULL,
 											 const ScFuncDesc**	ppFDesc = NULL,
@@ -794,21 +795,27 @@ public:
 	static const ScFuncDesc*	GetDefaultFuncDesc();
 
 	static xub_StrLen			GetFunctionStart( const String& rFormula, xub_StrLen nStart,
-													BOOL bBack, String* pFuncName = NULL );
+												  BOOL bBack, 
+                                                  ScAddress::Convention eConv,
+                                                  String* pFuncName = NULL );
 
-	static xub_StrLen			GetFunctionEnd	( const String& rFormula, xub_StrLen nStart );
+	static xub_StrLen			GetFunctionEnd	( const String& rFormula, xub_StrLen nStart,
+                                                  ScAddress::Convention eConv );
 
 	static xub_StrLen			GetArgStart		( const String& rFormula, xub_StrLen nStart,
-												  USHORT nArg );
+												  USHORT nArg,
+                                                  ScAddress::Convention eConv );
 
 	static String**				GetArgStrings	( const String&	rFormula,
 												  xub_StrLen	nFuncPos,
-												  USHORT		nArgs );
+												  USHORT		nArgs,
+                                                  ScAddress::Convention eConv );
 
 	static void					FillArgStrings	( const String&	rFormula,
 												  xub_StrLen	nFuncPos,
 												  USHORT		nArgs,
-												  String**		pArgs );
+												  String**		pArgs,
+                                                  ScAddress::Convention eConv );
 };
 
 //==================================================================
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/data/documen3.cxx sc/source/core/data/documen3.cxx
--- sc.clean/source/core/data/documen3.cxx	2008-03-03 11:24:50.000000000 -0500
+++ sc/source/core/data/documen3.cxx	2008-03-19 18:30:50.000000000 -0400
@@ -1757,6 +1757,7 @@ void ScDocument::SetDocOptions( const Sc
 	pFormatter->ChangeNullDate( d,m,y );
 	pFormatter->ChangeStandardPrec( (USHORT)rOpt.GetStdPrecision() );
 	pFormatter->SetYear2000( rOpt.GetYear2000() );
+    SetAddressConvention( rOpt.GetFormulaSyntax() );
 }
 
 const ScViewOptions& ScDocument::GetViewOptions() const
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/data/global.cxx sc/source/core/data/global.cxx
--- sc.clean/source/core/data/global.cxx	2008-03-03 11:24:50.000000000 -0500
+++ sc/source/core/data/global.cxx	2008-03-04 00:34:14.000000000 -0500
@@ -1061,6 +1061,7 @@ const ScFuncDesc* ScFormulaUtil::GetDefa
 
 BOOL ScFormulaUtil::GetNextFunc( const String&	rFormula,
 								 BOOL			bBack,
+                                 ScAddress::Convention eConv,
 								 xub_StrLen&	rFStart,   // Ein- und Ausgabe
 								 xub_StrLen*	pFEnd, 	   // = NULL
 								 const ScFuncDesc**	ppFDesc,   // = NULL
@@ -1070,13 +1071,13 @@ BOOL ScFormulaUtil::GetNextFunc( const S
 	xub_StrLen	nOldStart = rFStart;
 	String		aFname;
 
-	rFStart = GetFunctionStart( rFormula, rFStart, bBack, ppFDesc ? &aFname : NULL );
+	rFStart = GetFunctionStart( rFormula, rFStart, bBack, eConv, ppFDesc ? &aFname : NULL );
 	bFound  = ( rFStart != FUNC_NOTFOUND );
 
 	if ( bFound )
 	{
 		if ( pFEnd )
-			*pFEnd = GetFunctionEnd( rFormula, rFStart );
+			*pFEnd = GetFunctionEnd( rFormula, rFStart, eConv );
 
 		if ( ppFDesc )
 		{
@@ -1085,7 +1086,7 @@ BOOL ScFormulaUtil::GetNextFunc( const S
 			if ( *ppFDesc )
 			{
 				if (pppArgs)
-					*pppArgs = GetArgStrings( rFormula, rFStart, (*ppFDesc)->nArgCount );
+					*pppArgs = GetArgStrings( rFormula, rFStart, (*ppFDesc)->nArgCount, eConv );
 			}
 			else
 			{
@@ -1104,7 +1105,8 @@ BOOL ScFormulaUtil::GetNextFunc( const S
 void ScFormulaUtil::FillArgStrings( const String&	rFormula,
 									xub_StrLen		nFuncPos,
 									USHORT			nArgs,
-									String**		aArgArr )
+									String**		aArgArr,
+                                    ScAddress::Convention eConv )
 {
 	if ( !aArgArr ) return;
 
@@ -1115,11 +1117,11 @@ void ScFormulaUtil::FillArgStrings( cons
 
 	for ( i=0; i<nArgs && !bLast; i++ )
 	{
-		nStart = GetArgStart( rFormula, nFuncPos, i );
+		nStart = GetArgStart( rFormula, nFuncPos, i, eConv );
 
 		if ( i+1<nArgs ) // letztes Argument?
 		{
-			nEnd = GetArgStart( rFormula, nFuncPos, i+1 );
+			nEnd = GetArgStart( rFormula, nFuncPos, i+1, eConv );
 
 			if ( nEnd != nStart )
 				aArgArr[i] = new String( rFormula.Copy( nStart, nEnd-1-nStart ) );
@@ -1128,7 +1130,7 @@ void ScFormulaUtil::FillArgStrings( cons
 		}
 		else
 		{
-			nEnd = GetFunctionEnd( rFormula, nFuncPos )-1;
+			nEnd = GetFunctionEnd( rFormula, nFuncPos, eConv )-1;
 			if ( nStart < nEnd )
 				aArgArr[i] = new String( rFormula.Copy( nStart, nEnd-nStart ) );
 			else
@@ -1145,13 +1147,14 @@ void ScFormulaUtil::FillArgStrings( cons
 
 String** ScFormulaUtil::GetArgStrings( const String& rFormula,
 									   xub_StrLen nFuncPos,
-									   USHORT nArgs )
+									   USHORT nArgs,
+                                       ScAddress::Convention eConv )
 {
 	String** aArgArr = NULL;
 	if (nArgs)
 	{
 		aArgArr = new String*[nArgs];
-		FillArgStrings( rFormula, nFuncPos, nArgs, aArgArr );
+		FillArgStrings( rFormula, nFuncPos, nArgs, aArgArr, eConv );
 	}
 	return aArgArr;
 }
@@ -1174,6 +1177,7 @@ inline BOOL IsFormulaText( const String&
 xub_StrLen ScFormulaUtil::GetFunctionStart( const String&	rFormula,
 										xub_StrLen		nStart,
 										BOOL			bBack,
+                                        ScAddress::Convention /*eConv*/,
 										String*			pFuncName )
 {
 	xub_StrLen nStrLen = rFormula.Len();
@@ -1265,14 +1269,23 @@ xub_StrLen ScFormulaUtil::GetFunctionSta
 
 //------------------------------------------------------------------------
 
-xub_StrLen	ScFormulaUtil::GetFunctionEnd( const String& rStr, xub_StrLen nStart )
+xub_StrLen	ScFormulaUtil::GetFunctionEnd( const String& rStr, xub_StrLen nStart,
+                                           ScAddress::Convention eConv )
 {
 	xub_StrLen nStrLen = rStr.Len();
 
 	if ( nStrLen < nStart )
 		return nStart;
 
+    // We assume the following tokens are single-character tokens.
+    const sal_Unicode open       = ScCompiler::GetStringFromOpCode(ocOpen,  false, eConv).GetChar(0);
+    const sal_Unicode close      = ScCompiler::GetStringFromOpCode(ocClose, false, eConv).GetChar(0);
+    const sal_Unicode sep        = ScCompiler::GetStringFromOpCode(ocSep, false, eConv).GetChar(0);
+    const sal_Unicode arrayOpen  = ScCompiler::GetStringFromOpCode(ocArrayOpen,  false, eConv).GetChar(0);
+    const sal_Unicode arrayClose = ScCompiler::GetStringFromOpCode(ocArrayClose, false, eConv).GetChar(0);
+
 	short	nParCount = 0;
+    bool    bInArray = false;
 	BOOL	bFound = FALSE;
 
 	while ( !bFound && (nStart < nStrLen) )
@@ -1285,9 +1298,9 @@ xub_StrLen	ScFormulaUtil::GetFunctionEnd
 			while ( (nStart < nStrLen) && rStr.GetChar(nStart) != '"' )
 				nStart++;
 		}
-		else if ( c == '(' )
+		else if ( c == open )
 			nParCount++;
-		else if ( c == ')' )
+		else if ( c == close )
 		{
 			nParCount--;
 			if ( nParCount == 0 )
@@ -1298,9 +1311,17 @@ xub_StrLen	ScFormulaUtil::GetFunctionEnd
 				nStart--;	// einen zu weit gelesen
 			}
 		}
-		else if ( c == ';' )
+        else if ( c == arrayOpen )
+        {
+            bInArray = true;
+        }
+        else if ( c == arrayClose )
+        {
+            bInArray = false;
+        }
+		else if ( c == sep )
 		{
-			if ( nParCount == 0 )
+			if ( !bInArray && nParCount == 0 )
 			{
 				bFound = TRUE;
 				nStart--;	// einen zu weit gelesen
@@ -1314,14 +1335,23 @@ xub_StrLen	ScFormulaUtil::GetFunctionEnd
 
 //------------------------------------------------------------------
 
-xub_StrLen ScFormulaUtil::GetArgStart( const String& rStr, xub_StrLen nStart, USHORT nArg )
+xub_StrLen ScFormulaUtil::GetArgStart( const String& rStr, xub_StrLen nStart, USHORT nArg,
+                                       ScAddress::Convention eConv )
 {
 	xub_StrLen nStrLen = rStr.Len();
 
 	if ( nStrLen < nStart )
 		return nStart;
 
+    // We assume the following tokens are single-character tokens.
+    const sal_Unicode open       = ScCompiler::GetStringFromOpCode(ocOpen,  false, eConv).GetChar(0);
+    const sal_Unicode close      = ScCompiler::GetStringFromOpCode(ocClose, false, eConv).GetChar(0);
+    const sal_Unicode sep        = ScCompiler::GetStringFromOpCode(ocSep, false, eConv).GetChar(0);
+    const sal_Unicode arrayOpen  = ScCompiler::GetStringFromOpCode(ocArrayOpen,  false, eConv).GetChar(0);
+    const sal_Unicode arrayClose = ScCompiler::GetStringFromOpCode(ocArrayClose, false, eConv).GetChar(0);
+
 	short	nParCount	= 0;
+    bool    bInArray    = false;
 	BOOL	bFound		= FALSE;
 
 	while ( !bFound && (nStart < nStrLen) )
@@ -1334,19 +1364,27 @@ xub_StrLen ScFormulaUtil::GetArgStart( c
 			while ( (nStart < nStrLen) && rStr.GetChar(nStart) != '"' )
 				nStart++;
 		}
-		else if ( c == '(' )
+		else if ( c == open )
 		{
 			bFound = ( nArg == 0 );
 			nParCount++;
 		}
-		else if ( c == ')' )
+		else if ( c == close )
 		{
 			nParCount--;
 			bFound = ( nParCount == 0 );
 		}
-		else if ( c == ';' )
+        else if ( c == arrayOpen )
+        {
+            bInArray = true;
+        }
+        else if ( c == arrayClose )
+        {
+            bInArray = false;
+        }
+		else if ( c == sep )
 		{
-			if ( nParCount == 1 )
+			if ( !bInArray && nParCount == 1 )
 			{
 				nArg--;
 				bFound = ( nArg == 0  );
@@ -1702,8 +1740,10 @@ void ScFuncDesc::InitArgumentInfo() cons
 
 //------------------------------------------------------------------------
 
-String ScFuncDesc::GetParamList() const
+String ScFuncDesc::GetParamList( ScAddress::Convention eConv ) const
 {
+    const String& sep = ScCompiler::GetStringFromOpCode(ocSep, false, eConv);
+
     String aSig;
 
     if ( nArgCount > 0 )
@@ -1714,7 +1754,10 @@ String ScFuncDesc::GetParamList() const
             {
                 aSig += *(aDefArgNames[i]);
                 if ( i != nArgCount-1 )
-                    aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( "; " ));
+                {
+                    aSig.Append(sep);
+                    aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( " " ));
+                }
             }
         }
         else
@@ -1723,14 +1766,17 @@ String ScFuncDesc::GetParamList() const
             for ( USHORT nArg = 0; nArg < nFix; nArg++ )
             {
                 aSig += *(aDefArgNames[nArg]);
-                aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( "; " ));
+                aSig.Append(sep);
+                aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( " " ));
             }
             aSig += *(aDefArgNames[nFix]);
             aSig += '1';
-            aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( "; " ));
+            aSig.Append(sep);
+            aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( " " ));
             aSig += *(aDefArgNames[nFix]);
             aSig += '2';
-            aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( "; ... " ));
+            aSig.Append(sep);
+            aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( " ... " ));
         }
     }
 
@@ -1739,7 +1785,7 @@ String ScFuncDesc::GetParamList() const
 
 //------------------------------------------------------------------------
 
-String ScFuncDesc::GetSignature() const
+String ScFuncDesc::GetSignature( ScAddress::Convention eConv ) const
 {
 	String aSig;
 
@@ -1747,7 +1793,7 @@ String ScFuncDesc::GetSignature() const
 	{
 		aSig = *pFuncName;
 
-        String aParamList( GetParamList() );
+        String aParamList( GetParamList(eConv) );
         if( aParamList.Len() )
         {
 			aSig.AppendAscii(RTL_CONSTASCII_STRINGPARAM( "( " ));
@@ -1763,8 +1809,10 @@ String ScFuncDesc::GetSignature() const
 
 //------------------------------------------------------------------------
 
-String ScFuncDesc::GetFormulaString( String** aArgArr ) const
+String ScFuncDesc::GetFormulaString( String** aArgArr, ScAddress::Convention eConv ) const
 {
+    const String& sep = ScCompiler::GetStringFromOpCode(ocSep, false, eConv);
+
 	String aFormula;
 
 	if(pFuncName)
@@ -1787,7 +1835,7 @@ String ScFuncDesc::GetFormulaString( Str
 					{
 						bLastArg = !( aArgArr[i+1]->Len() > 0 );
 						if ( !bLastArg )
-							aFormula += ';';
+							aFormula += sep;
 					}
 				}
 			}
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/tool/compiler.cxx sc/source/core/tool/compiler.cxx
--- sc.clean/source/core/tool/compiler.cxx	2008-03-03 11:24:50.000000000 -0500
+++ sc/source/core/tool/compiler.cxx	2008-03-19 20:54:49.000000000 -0400
@@ -439,6 +439,7 @@ void ScCompiler::SetCompileEnglish( BOOL
         if (!pCharClassEnglish)
             InitCharClassEnglish();
         pCharClass = pCharClassEnglish;
+        SetRefConvention(ScAddress::CONV_OOO); // hack!  when compiling English, it's most likely handling ODF formula syntax.
     }
     else
     {
@@ -928,7 +929,7 @@ for (i = 97; i < 123; i++)
 /* ) */     t[41] |=   SC_COMPILER_C_WORD;
 /* * */     t[42] |=   SC_COMPILER_C_WORD;
 /* + */     t[43] |=   SC_COMPILER_C_WORD;
-#if 0 /* this really needs to be locale specific. */
+#if 1 /* this really needs to be locale specific. */
 /* , */     t[44]  =   SC_COMPILER_C_CHAR | SC_COMPILER_C_WORD_SEP | SC_COMPILER_C_VALUE_SEP;
 #else
 /* , */     t[44] |=   SC_COMPILER_C_WORD;
@@ -980,6 +981,20 @@ struct Convention_A1 : public ScCompiler
         return pCharClass->parseAnyToken( rFormula,
                 nSrcPos, nStartFlags, aAddAllowed, nContFlags, aAddAllowed );
     }
+
+    virtual const String& GetSymbol( const ScCompiler::OpCodeMapPtr& rpMap, OpCode e ) const
+    {
+        return rpMap->getSymbol(e);
+    }
+
+    virtual OpCode GetOpCode( const ScCompiler::OpCodeMapPtr& rpMap, const String& rName, bool bInArray ) const
+    {
+        const ScOpCodeHashMap* pHashMap = rpMap->getHashMap();
+        ScOpCodeHashMap::const_iterator itr = pHashMap->find(rName);
+        if (itr == pHashMap->end())
+            return ocNone;
+        return itr->second;
+    }
 };
 
 String Convention_A1::MakeColStr( SCCOL nCol )
@@ -1232,6 +1247,10 @@ static const ScCompiler::Convention *pCo
 
 struct ConventionXL
 {
+    static const String maSep;
+    static const String maArrayRowSep;
+    static const String maArrayColSep;
+
     static bool GetDocAndTab( const ScCompiler& rComp,
                               const SingleRefData& rRef,
                               String& rDocName,
@@ -1314,8 +1333,50 @@ struct ConventionXL
             rBuf.append( sal_Unicode( '!' ) );
         }
     }
+
+    static OpCode GetOpCode( const ScCompiler::OpCodeMapPtr& rpMap, const String& rName, bool bInArray )
+    {
+        // TODO: switch to using a hash table for better string lookup performance.
+        if (bInArray)
+        {
+            if (rName.Equals(ConventionXL::maArrayRowSep))
+            {
+                return ocArrayRowSep;
+            }
+            else if (rName.Equals(ConventionXL::maArrayColSep))
+            {
+                return ocArrayColSep;
+            }       
+        }
+
+        if (rName.Equals(ConventionXL::maSep))
+        {
+            return ocSep;
+        }
+
+        const ScOpCodeHashMap* pHashMap = rpMap->getHashMap();
+        ScOpCodeHashMap::const_iterator itr = pHashMap->find(rName);
+        if (itr == pHashMap->end())
+        {
+            return ocNone;
+        }
+
+        switch (itr->second)
+        {
+            // opcode found, but the following opcodes must not be valid for XL_A1.
+            case ocSep:
+            case ocArrayColSep:
+            case ocArrayRowSep:
+                return ocNone;
+        }
+        return itr->second;
+    }
 };
 
+const String ConventionXL::maSep = String::CreateFromAscii(",");
+const String ConventionXL::maArrayRowSep = String::CreateFromAscii(";");
+const String ConventionXL::maArrayColSep = String::CreateFromAscii(",");
+
 struct ConventionXL_A1 : public Convention_A1, public ConventionXL
 {
     ConventionXL_A1() : Convention_A1( ScAddress::CONV_XL_A1 ) { }
@@ -1325,6 +1386,7 @@ struct ConventionXL_A1 : public Conventi
                      BOOL bSingleRef ) const
     {
         ComplRefData aRef( rRef );
+        aRef.Ref1.CalcAbsIfRel( rComp.GetPos() );
 
         MakeDocStr( rBuf, rComp, aRef, bSingleRef );
 
@@ -1389,6 +1451,25 @@ struct ConventionXL_A1 : public Conventi
             MakeRowStr( rBuf, aRef.Ref2.nRow );
         }
     }
+
+    virtual const String& GetSymbol( const ScCompiler::OpCodeMapPtr& rpMap, OpCode e ) const
+    {
+        switch (e)
+        {
+            case ocSep:
+                return ConventionXL::maSep;
+            case ocArrayRowSep:
+                return ConventionXL::maArrayRowSep;
+            case ocArrayColSep:
+                return ConventionXL::maArrayColSep;
+        }
+        return rpMap->getSymbol(e);
+    }
+
+    virtual OpCode GetOpCode( const ScCompiler::OpCodeMapPtr& rpMap, const String& rName, bool bInArray ) const
+    {
+        return ConventionXL::GetOpCode(rpMap, rName, bInArray);
+    }
 };
 
 static const ConventionXL_A1 ConvXL_A1;
@@ -1508,6 +1589,25 @@ struct ConventionXL_R1C1 : public ScComp
         return pCharClass->parseAnyToken( rFormula,
                 nSrcPos, nStartFlags, aAddAllowed, nContFlags, aAddAllowed );
     }
+
+    virtual const String& GetSymbol( const ScCompiler::OpCodeMapPtr& rpMap, OpCode e ) const
+    {
+        switch (e)
+        {
+            case ocSep:
+                return ConventionXL::maSep;
+            case ocArrayRowSep:
+                return ConventionXL::maArrayRowSep;
+            case ocArrayColSep:
+                return ConventionXL::maArrayColSep;
+        }
+        return rpMap->getSymbol(e);
+    }
+
+    virtual OpCode GetOpCode( const ScCompiler::OpCodeMapPtr& rpMap, const String& rName, bool bInArray ) const
+    {
+        return ConventionXL::GetOpCode(rpMap, rName, bInArray);
+    }
 };
 
 static const ConventionXL_R1C1 ConvXL_R1C1;
@@ -1626,7 +1726,8 @@ void ScCompiler::CheckTabQuotes( String&
 
 void ScCompiler::SetCompileXML( BOOL bVal )
 {
-    SetRefConvention( bVal ? pConvOOO_A1_XML : pConvOOO_A1 );
+    if (bVal)
+        SetRefConvention( pConvOOO_A1_XML );
 }
 void ScCompiler::SetRefConvention( ScAddress::Convention eConv )
 {
@@ -1968,15 +2069,16 @@ xub_StrLen ScCompiler::NextSymbol()
 // Convert symbol to token
 //---------------------------------------------------------------------------
 
-BOOL ScCompiler::IsOpCode( const String& rName )
+BOOL ScCompiler::IsOpCode( const String& rName, bool bInArray )
 {
-    ScOpCodeHashMap::const_iterator iLook( mxSymbols->getHashMap()->find( rName));
-    BOOL bFound = (iLook != mxSymbols->getHashMap()->end());
-    if (bFound)
+    bool bFound = false;
+    OpCode oc = pConv->GetOpCode(mxSymbols, rName, bInArray);
+    if (oc != ocNone)
     {
         ScRawToken aToken;
-        aToken.SetOpCode( (*iLook).second );
+        aToken.SetOpCode(oc);
         pRawToken = aToken.Clone();
+        bFound = true;
     }
     else
     {
@@ -2776,8 +2878,9 @@ void ScCompiler::AutoCorrectParsedSymbol
     }
 }
 
-BOOL ScCompiler::NextNewToken( bool bAllowBooleans )
+BOOL ScCompiler::NextNewToken( bool bInArray )
 {
+    bool bAllowBooleans = bInArray;
     xub_StrLen nSpaces = NextSymbol();
 
 #if 0
@@ -2836,7 +2939,7 @@ BOOL ScCompiler::NextNewToken( bool bAll
             // IsReference().
             // IsBoolean before isValue to catch inline bools without the kludge
             //    for inline arrays.
-            if ( !(bMayBeFuncName && IsOpCode( aUpper ))
+            if ( !(bMayBeFuncName && IsOpCode( aUpper, bInArray ))
               && !IsReference( aOrg )
               && !(bAllowBooleans && IsBoolean( aUpper ))
               && !IsValue( aUpper )
@@ -2988,7 +3091,7 @@ ScTokenArray* ScCompiler::CompileString(
                 SetError(errCodeOverflow);
             }
             else if ( bAutoCorrect )
-                aCorrectedFormula += mxSymbols->getSymbol(ocArrayClose);
+                aCorrectedFormula += pConv->GetSymbol(mxSymbols, ocArrayClose);
         }
 
         ScByteToken aToken( ocClose );
@@ -2999,7 +3102,7 @@ ScTokenArray* ScCompiler::CompileString(
                 SetError(errCodeOverflow); break;
             }
             if ( bAutoCorrect )
-                aCorrectedFormula += mxSymbols->getSymbol(ocClose);
+                aCorrectedFormula += pConv->GetSymbol(mxSymbols, ocClose);
         }
     }
     if ( nForced >= 2 )
@@ -5001,9 +5104,17 @@ ScRangeData* ScCompiler::UpdateMoveTab( 
 }
 
 
-const String& ScCompiler::GetStringFromOpCode( OpCode eOpCode, bool bEnglish )
+const String& ScCompiler::GetStringFromOpCode( OpCode eOpCode, bool bEnglish, ScAddress::Convention eConv )
 {
-    return bEnglish ? mxSymbolsEnglish->getSymbol( eOpCode ) : mxSymbolsNative->getSymbol( eOpCode );
+    const ScCompiler::Convention* pConv = pConvOOO_A1; // default
+    switch (eConv)
+    {
+        case ScAddress::CONV_XL_A1:     pConv = pConvXL_A1;     break;
+        case ScAddress::CONV_XL_R1C1:   pConv = pConvXL_R1C1;   break;
+        // more case likely to follow in the future....
+    }
+    OpCodeMapPtr mxSymbols = bEnglish ? mxSymbolsEnglish : mxSymbolsNative;
+    return pConv->GetSymbol( mxSymbols, eOpCode );
 }
 
 ScToken* ScCompiler::CreateStringFromToken( String& rFormula, ScToken* pTokenP,
@@ -5046,7 +5157,7 @@ ScToken* ScCompiler::CreateStringFromTok
     else if( eOp >= ocInternalBegin && eOp <= ocInternalEnd )
         rBuffer.appendAscii( pInternal[ eOp - ocInternalBegin ] );
     else if( (USHORT) eOp < mxSymbolsNative->getSymbolCount())        // Keyword:
-        rBuffer.append(mxSymbols->getSymbol(eOp));
+        rBuffer.append( pConv->GetSymbol(mxSymbols, eOp) );
     else
     {
         DBG_ERROR("Unbekannter OpCode");
@@ -5175,20 +5286,19 @@ void ScCompiler::CreateStringFromScMatri
     SCSIZE nC, nMaxC, nR, nMaxR;
 
     pMatrix->GetDimensions( nMaxC, nMaxR);
-
-    rBuffer.append( mxSymbols->getSymbol(ocArrayOpen) );
+    rBuffer.append( pConv->GetSymbol(mxSymbols, ocArrayOpen) );
     for( nR = 0 ; nR < nMaxR ; nR++)
     {
         if( nR > 0)
         {
-            rBuffer.append( mxSymbols->getSymbol(ocArrayRowSep) );
+            rBuffer.append( pConv->GetSymbol(mxSymbols, ocArrayRowSep) );
         }
 
         for( nC = 0 ; nC < nMaxC ; nC++)
         {
             if( nC > 0)
             {
-                rBuffer.append( mxSymbols->getSymbol(ocArrayColSep) );
+                rBuffer.append( pConv->GetSymbol(mxSymbols, ocArrayColSep) );
             }
 
             if( pMatrix->IsValue( nC, nR ) )
@@ -5213,7 +5323,7 @@ void ScCompiler::CreateStringFromScMatri
                 AppendString( rBuffer, pMatrix->GetString( nC, nR ) );
         }
     }
-    rBuffer.append( mxSymbols->getSymbol(ocArrayClose) );
+    rBuffer.append( pConv->GetSymbol(mxSymbols, ocArrayClose) );
 }
 
 void ScCompiler::AppendBoolean( rtl::OUStringBuffer& rBuffer, bool bVal )
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/tool/docoptio.cxx sc/source/core/tool/docoptio.cxx
--- sc.clean/source/core/tool/docoptio.cxx	2008-03-03 11:24:50.000000000 -0500
+++ sc/source/core/tool/docoptio.cxx	2008-03-19 20:37:44.000000000 -0400
@@ -50,6 +50,7 @@
 #include "scresid.hxx"
 #include "sc.hrc"
 #include "miscuno.hxx"
+#include "global.hxx"
 
 using namespace utl;
 using namespace rtl;
@@ -105,7 +106,8 @@ ScDocOptions::ScDocOptions( const ScDocO
 			bMatchWholeCell( rCpy.bMatchWholeCell ),
 			bDoAutoSpell( rCpy.bDoAutoSpell ),
             bLookUpColRowNames( rCpy.bLookUpColRowNames ),
-            bFormulaRegexEnabled( rCpy.bFormulaRegexEnabled )
+            bFormulaRegexEnabled( rCpy.bFormulaRegexEnabled ),
+            eFormulaConv( rCpy.eFormulaConv )
 {
 }
 
@@ -212,6 +214,7 @@ void ScDocOptions::ResetDocOptions()
 	bDoAutoSpell		= FALSE;
 	bLookUpColRowNames	= TRUE;
     bFormulaRegexEnabled= TRUE;
+    eFormulaConv        = ScAddress::CONV_OOO;
 }
 
 //========================================================================
@@ -287,7 +290,8 @@ SfxPoolItem* __EXPORT ScTpCalcItem::Clon
 #define SCCALCOPT_SEARCHCRIT		9
 #define SCCALCOPT_FINDLABEL			10
 #define SCCALCOPT_REGEX             11
-#define SCCALCOPT_COUNT             12
+#define SCCALCOPT_FORMULA_SYNTAX    12
+#define SCCALCOPT_COUNT             13
 
 #define CFGPATH_DOCLAYOUT	"Office.Calc/Layout/Other"
 
@@ -310,7 +314,8 @@ Sequence<OUString> ScDocCfg::GetCalcProp
 		"Other/Precision",					// SCCALCOPT_PRECISION
 		"Other/SearchCriteria",				// SCCALCOPT_SEARCHCRIT
         "Other/FindLabel",                  // SCCALCOPT_FINDLABEL
-        "Other/RegularExpressions"          // SCCALCOPT_REGEX
+        "Other/RegularExpressions",         // SCCALCOPT_REGEX
+        "Other/FormulaSyntax"               // SCCALCOPT_FORMULA_SYNTAX
 	};
 	Sequence<OUString> aNames(SCCALCOPT_COUNT);
 	OUString* pNames = aNames.getArray();
@@ -403,6 +408,14 @@ ScDocCfg::ScDocCfg() :
                     case SCCALCOPT_REGEX :
                         SetFormulaRegexEnabled( ScUnoHelpFunctions::GetBoolFromAny( pValues[nProp] ) );
                         break;
+                    case SCCALCOPT_FORMULA_SYNTAX:
+                    {
+                        if ((pValues[nProp] >>= nIntVal) && nIntVal > ScAddress::CONV_UNSPECIFIED && nIntVal < ScAddress::CONV_LAST)
+                            SetFormulaSyntax( static_cast<ScAddress::Convention>(nIntVal) );
+                        else
+                            SetFormulaSyntax(ScAddress::CONV_OOO);
+                    }
+                    break;
 				}
 			}
 		}
@@ -486,6 +499,10 @@ IMPL_LINK( ScDocCfg, CalcCommitHdl, void
 				break;
             case SCCALCOPT_REGEX :
                 ScUnoHelpFunctions::SetBoolInAny( pValues[nProp], IsFormulaRegexEnabled() );
+                break;
+            case SCCALCOPT_FORMULA_SYNTAX:
+                pValues[nProp] <<= static_cast<sal_Int32>(GetFormulaSyntax());
+                break;
 		}
 	}
 	aCalcItem.PutProperties(aNames, aValues);
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/app/scmod.cxx sc/source/ui/app/scmod.cxx
--- sc.clean/source/ui/app/scmod.cxx	2008-03-03 11:24:45.000000000 -0500
+++ sc/source/ui/app/scmod.cxx	2008-03-19 17:38:48.000000000 -0400
@@ -1253,6 +1253,7 @@ void ScModule::ModifyOptions( const SfxI
 						 || rOldOpt.IsMatchWholeCell() != rNewOpt.IsMatchWholeCell()
 						 || rOldOpt.GetYear2000()	!= rNewOpt.GetYear2000()
                          || rOldOpt.IsFormulaRegexEnabled() != rNewOpt.IsFormulaRegexEnabled()
+                         || rOldOpt.GetFormulaSyntax() != rNewOpt.GetFormulaSyntax()
 						 );
 			pDoc->SetDocOptions( rNewOpt );
 			pDocSh->SetDocumentModified();
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/formdlg/formula.cxx sc/source/ui/formdlg/formula.cxx
--- sc.clean/source/ui/formdlg/formula.cxx	2008-03-03 11:24:49.000000000 -0500
+++ sc/source/ui/formdlg/formula.cxx	2008-03-04 00:32:04.000000000 -0500
@@ -306,7 +306,7 @@ ScFormulaDlg::ScFormulaDlg( SfxBindings*
 
 			xub_StrLen nFStart = 0;
 			xub_StrLen nFEnd   = 0;
-			if ( ScFormulaUtil::GetNextFunc( aFormula, FALSE, nFStart, &nFEnd) )
+			if ( ScFormulaUtil::GetNextFunc( aFormula, FALSE, pDoc->GetAddressConvention(), nFStart, &nFEnd, NULL, NULL ) )
 			{
 				pScMod->InputReplaceSelection( aFormula );
 				pScMod->InputSetSelection( nFStart, nFEnd );
@@ -523,7 +523,7 @@ void ScFormulaDlg::FillControls()
 	const ScFuncDesc*		pOldFuncDesc=pFuncDesc;
 	BOOL			bTestFlag=FALSE;
 
-	if ( ScFormulaUtil::GetNextFunc( aFormula, FALSE,
+	if ( ScFormulaUtil::GetNextFunc( aFormula, FALSE, pDoc->GetAddressConvention(),
 									 nNextFStart, &nNextFEnd, &pFuncDesc, &pArgArr ) )
 	{
 		bTestFlag=(pOldFuncDesc!=pFuncDesc);
@@ -564,7 +564,7 @@ void ScFormulaDlg::FillControls()
 
 		if(bTestFlag) aScParaWin.SetArgCount(nArgs,nOffset);
 		USHORT nActiv=0;
-		xub_StrLen nArgPos=ScFormulaUtil::GetArgStart( aFormula, nFStart, 0 );
+		xub_StrLen nArgPos=ScFormulaUtil::GetArgStart( aFormula, nFStart, 0, pDoc->GetAddressConvention() );
 		xub_StrLen nEditPos=(xub_StrLen) pMEdit->GetSelection().Min();
 		BOOL	bFlag=FALSE;
 
@@ -595,11 +595,11 @@ void ScFormulaDlg::FillControls()
 	}
 		//	Test, ob vorne/hinten noch mehr Funktionen sind
 
-	xub_StrLen nTempStart = ScFormulaUtil::GetArgStart( aFormula, nFStart, 0 );
-	BOOL bNext = ScFormulaUtil::GetNextFunc( aFormula, FALSE, nTempStart );
+	xub_StrLen nTempStart = ScFormulaUtil::GetArgStart( aFormula, nFStart, 0, pDoc->GetAddressConvention() );
+	BOOL bNext = ScFormulaUtil::GetNextFunc( aFormula, FALSE, pDoc->GetAddressConvention(), nTempStart );
 	nTempStart=(xub_StrLen)pMEdit->GetSelection().Min();
 	pData->SetFStart(nTempStart);
-	BOOL bPrev = ScFormulaUtil::GetNextFunc( aFormula, TRUE, nTempStart );
+	BOOL bPrev = ScFormulaUtil::GetNextFunc( aFormula, TRUE, pDoc->GetAddressConvention(), nTempStart );
 	aBtnBackward.Enable(bPrev);
 	aBtnForward.Enable(bNext);
 }
@@ -809,7 +809,7 @@ void ScFormulaDlg::UpdateFunctionDesc()
 		{
             pDesc->InitArgumentInfo();      // full argument info is needed
 
-			String aSig = pDesc->GetSignature();
+			String aSig = pDesc->GetSignature(pDoc->GetAddressConvention());
 
 			aFtFuncName.SetText( aSig );
 			if(pDesc->pFuncDesc)
@@ -819,7 +819,7 @@ void ScFormulaDlg::UpdateFunctionDesc()
 			ResizeArgArr( pDesc );
 
 			if ( pArgArr && pArgArr[0] && pArgArr[0]->Len() )		// noch Argumente da?
-				aSig = pDesc->GetFormulaString( pArgArr );			// fuer Eingabezeile
+				aSig = pDesc->GetFormulaString( pArgArr, pDoc->GetAddressConvention() );			// fuer Eingabezeile
 			//@ pScMod->InputReplaceSelection( aSig );
 		}
 	}
@@ -899,7 +899,7 @@ void ScFormulaDlg::EditThisFunc(xub_StrL
 
 	//@bFound = ScFormulaUtil::GetNextFunc( aFormula, FALSE, nNextFStart, &nNextFEnd, &pFuncDesc );
 
-	bFound = ScFormulaUtil::GetNextFunc( aFormula, FALSE, nNextFStart, &nNextFEnd);
+	bFound = ScFormulaUtil::GetNextFunc( aFormula, FALSE, pDoc->GetAddressConvention(), nNextFStart, &nNextFEnd);
 	if ( bFound )
 	{
 		xub_StrLen nFEnd;
@@ -959,15 +959,15 @@ void ScFormulaDlg::EditNextFunc( BOOL bF
 	BOOL bFound;
 	if ( bForward )
 	{
-		nNextFStart	= ScFormulaUtil::GetArgStart( aFormula, nFStart, 0 );
+		nNextFStart	= ScFormulaUtil::GetArgStart( aFormula, nFStart, 0, pDoc->GetAddressConvention() );
 		//@bFound = ScFormulaUtil::GetNextFunc( aFormula, FALSE, nNextFStart, &nNextFEnd, &pFuncDesc );
-		bFound = ScFormulaUtil::GetNextFunc( aFormula, FALSE, nNextFStart, &nNextFEnd);
+		bFound = ScFormulaUtil::GetNextFunc( aFormula, FALSE, pDoc->GetAddressConvention(), nNextFStart, &nNextFEnd);
 	}
 	else
 	{
 		nNextFStart	= nFStart;
 		//@bFound = ScFormulaUtil::GetNextFunc( aFormula, TRUE, nNextFStart, &nNextFEnd, &pFuncDesc );
-		bFound = ScFormulaUtil::GetNextFunc( aFormula, TRUE, nNextFStart, &nNextFEnd);
+		bFound = ScFormulaUtil::GetNextFunc( aFormula, TRUE, pDoc->GetAddressConvention(), nNextFStart, &nNextFEnd);
 	}
 
 	if ( bFound )
@@ -1012,8 +1012,8 @@ void ScFormulaDlg::EditFuncParas(xub_Str
 
 		nArgs = pFuncDesc->nArgCount;
 
-		xub_StrLen nArgPos=ScFormulaUtil::GetArgStart( aFormula, nFStart, 0 );
-		pArgArr = ScFormulaUtil::GetArgStrings( aFormula, nFStart, pFuncDesc->nArgCount );
+		xub_StrLen nArgPos=ScFormulaUtil::GetArgStart( aFormula, nFStart, 0, pDoc->GetAddressConvention() );
+		pArgArr = ScFormulaUtil::GetArgStrings( aFormula, nFStart, pFuncDesc->nArgCount, pDoc->GetAddressConvention() );
 
 		USHORT nActiv=aScParaWin.GetSliderPos();
 		BOOL	bFlag=FALSE;
@@ -1068,8 +1068,6 @@ BOOL ScFormulaDlg::CalcValue( const Stri
 		{
 			ScFormulaCell* pFCell = new ScFormulaCell( pDoc, aCursorPos, rStrExp );
 
-			ScCompiler*	pCompi=new ScCompiler( pDoc, aCursorPos, *(pFCell->GetCode()));
-
 			// #35521# HACK! um bei ColRowNames kein #REF! zu bekommen,
 			// wenn ein Name eigentlich als Bereich in die Gesamt-Formel
 			// eingefuegt wird, bei der Einzeldarstellung aber als
@@ -1129,7 +1127,6 @@ BOOL ScFormulaDlg::CalcValue( const Stri
 			}
 
 			delete pFCell;
-			delete pCompi;
 		}
 		else
 			bResult = FALSE;
@@ -1143,7 +1140,7 @@ void ScFormulaDlg::UpdateValues()
 	ScModule* pScMod = SC_MOD();
 	String aStrResult;
 
-	if ( CalcValue( pFuncDesc->GetFormulaString( pArgArr ), aStrResult ) )
+	if ( CalcValue( pFuncDesc->GetFormulaString( pArgArr, pDoc->GetAddressConvention() ), aStrResult ) )
 		aWndResult.SetValue( aStrResult );
 
 	aStrResult.Erase();
@@ -1206,7 +1203,7 @@ IMPL_LINK( ScFormulaDlg, FxHdl, ScParaWi
 		xub_StrLen nFormulaStrPos = pData->GetFStart();
 
 		String aFormula = pScMod->InputGetFormulaStr();
-		xub_StrLen n1 = ScFormulaUtil::GetArgStart( aFormula, nFormulaStrPos, nEdFocus+pData->GetOffset() );
+		xub_StrLen n1 = ScFormulaUtil::GetArgStart( aFormula, nFormulaStrPos, nEdFocus+pData->GetOffset(), pDoc->GetAddressConvention() );
 
 		pData->SetEdFocus( nEdFocus );
 		pData->SaveValues();
@@ -1335,7 +1332,7 @@ IMPL_LINK( ScFormulaDlg, FormulaCursorHd
 
 		if(nFStart<nPos)
 		{
-			xub_StrLen nPos1=ScFormulaUtil::GetFunctionEnd(aString,nFStart);
+			xub_StrLen nPos1=ScFormulaUtil::GetFunctionEnd(aString, nFStart, pDoc->GetAddressConvention());
 
 			if(nPos1>nPos || nPos1==STRING_NOTFOUND)
 			{
@@ -1356,7 +1353,7 @@ IMPL_LINK( ScFormulaDlg, FormulaCursorHd
 				}
 				if(nCount==0)
 				{
-					nFStart=ScFormulaUtil::GetFunctionStart(aString,n,TRUE);
+					nFStart = ScFormulaUtil::GetFunctionStart( aString, n, TRUE, pDoc->GetAddressConvention() );
 					EditThisFunc(nFStart);
 				}
 				else
@@ -1382,7 +1379,7 @@ void ScFormulaDlg::UpdateSelection()
 
 	pScMod->InputSetSelection((xub_StrLen)aFuncSel.Min(),(xub_StrLen)aFuncSel.Max());
 
-	pScMod->InputReplaceSelection( pFuncDesc->GetFormulaString( pArgArr ) );
+	pScMod->InputReplaceSelection( pFuncDesc->GetFormulaString( pArgArr, pDoc->GetAddressConvention() ) );
 	pMEdit->SetText(pScMod->InputGetFormulaStr());
 	xub_StrLen PrivStart, PrivEnd;
 	pScMod->InputGetSelection( PrivStart, PrivEnd);
@@ -1392,7 +1389,7 @@ void ScFormulaDlg::UpdateSelection()
 	nArgs = pFuncDesc->nArgCount;
 
 	String aFormula=pMEdit->GetText();
-	xub_StrLen nArgPos=ScFormulaUtil::GetArgStart( aFormula,PrivStart,0);
+	xub_StrLen nArgPos=ScFormulaUtil::GetArgStart( aFormula, PrivStart, 0, pDoc->GetAddressConvention() );
 
 	USHORT nPos=aScParaWin.GetActiveLine();
 
@@ -1648,6 +1645,8 @@ void ScFormulaDlg::UpdateTokenArray( con
 
 xub_StrLen ScFormulaDlg::GetFunctionPos(xub_StrLen nPos)
 {
+    const sal_Unicode sep = ScCompiler::GetStringFromOpCode(ocSep, false, pDoc->GetAddressConvention()).GetChar(0);
+
 	xub_StrLen nTokPos=1;
 	xub_StrLen nOldTokPos=1;
 	xub_StrLen nFuncPos=STRING_NOTFOUND;    //@ Testweise
@@ -1673,7 +1672,7 @@ xub_StrLen ScFormulaDlg::GetFunctionPos(
 
 			if(eOp==ocPush || eOp==ocSpaces)
 			{
-				xub_StrLen n1=aFormString.Search(';',nTokPos);
+				xub_StrLen n1=aFormString.Search(sep, nTokPos);
 				xub_StrLen n2=aFormString.Search(')',nTokPos);
 				xub_StrLen nXXX=nTokPos;
 				if(n1<n2)
@@ -1903,7 +1902,7 @@ IMPL_LINK( ScFormulaDlg, FuncSelHdl, ScF
 		{
             pDesc->InitArgumentInfo();      // full argument info is needed
 
-			String aSig = pDesc->GetSignature();
+			String aSig = pDesc->GetSignature(pDoc->GetAddressConvention());
 			if(pDesc->pFuncName)
 				aFtHeadLine.SetText( *(pDesc->pFuncName) );
 			else
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/inc/optdlg.hrc sc/source/ui/inc/optdlg.hrc
--- sc.clean/source/ui/inc/optdlg.hrc	2008-03-03 11:24:45.000000000 -0500
+++ sc/source/ui/inc/optdlg.hrc	2008-03-19 16:21:34.000000000 -0400
@@ -78,6 +78,8 @@
 #define BTN_MATCH			18
 #define BTN_LOOKUP			19
 #define BTN_REGEX           20
+#define FT_FORMULA_SYNTAX   21
+#define LB_FORMULA_SYNTAX   22
 
 // TP_VIEW:
 #define BTN_VSCROLL			1
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/inc/tpcalc.hxx sc/source/ui/inc/tpcalc.hxx
--- sc.clean/source/ui/inc/tpcalc.hxx	2008-03-03 11:24:45.000000000 -0500
+++ sc/source/ui/inc/tpcalc.hxx	2008-03-19 18:34:51.000000000 -0400
@@ -61,6 +61,8 @@
 #include "editfield.hxx"
 #endif
 
+#include "vcl/lstbox.hxx"
+
 //===================================================================
 
 class ScDocOptions;
@@ -101,6 +103,8 @@ private:
 	CheckBox		aBtnMatch;
     CheckBox        aBtnRegex;
 	CheckBox		aBtnLookUp;
+    FixedText       aFtFormulaSyntax;
+    ListBox         aLBFormulaSyntax;
 
 	FixedText		aFtPrec;
 	NumericField	aEdPrec;
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/optdlg/tpcalc.cxx sc/source/ui/optdlg/tpcalc.cxx
--- sc.clean/source/ui/optdlg/tpcalc.cxx	2008-03-03 11:24:49.000000000 -0500
+++ sc/source/ui/optdlg/tpcalc.cxx	2008-03-19 18:18:38.000000000 -0400
@@ -96,6 +96,8 @@ ScTpCalcOptions::ScTpCalcOptions( Window
 		aBtnMatch	 	( this, ScResId( BTN_MATCH ) ),
         aBtnRegex       ( this, ScResId( BTN_REGEX ) ),
 		aBtnLookUp   	( this, ScResId( BTN_LOOKUP ) ),
+        aFtFormulaSyntax( this, ScResId( FT_FORMULA_SYNTAX ) ),
+        aLBFormulaSyntax( this, ScResId( LB_FORMULA_SYNTAX ) ),
 		aFtPrec 		( this, ScResId( FT_PREC ) ),
 		aEdPrec 		( this, ScResId( ED_PREC ) ),
 		pOldOptions 	( new ScDocOptions(
@@ -161,6 +163,18 @@ void __EXPORT ScTpCalcOptions::Reset( co
 	aEdPrec    .SetValue( pLocalOptions->GetStdPrecision() );
     aEdEps     .SetValue( pLocalOptions->GetIterEps(), 6 );
 
+    ScAddress::Convention eConv = pLocalOptions->GetFormulaSyntax();
+    switch (eConv)
+    {
+        case ScAddress::CONV_OOO:
+        case ScAddress::CONV_XL_A1:
+        case ScAddress::CONV_XL_R1C1:
+            aLBFormulaSyntax.SelectEntryPos(static_cast<USHORT>(eConv));
+        break;
+        default:
+            aLBFormulaSyntax.SelectEntryPos(0);
+    }
+
 	pLocalOptions->GetDate( d, m, y );
 
 	switch ( y )
@@ -192,6 +206,8 @@ BOOL __EXPORT ScTpCalcOptions::FillItemS
 	pLocalOptions->SetMatchWholeCell( aBtnMatch.IsChecked() );
     pLocalOptions->SetFormulaRegexEnabled( aBtnRegex.IsChecked() );
 	pLocalOptions->SetLookUpColRowNames( aBtnLookUp.IsChecked() );
+    pLocalOptions->SetFormulaSyntax(
+        static_cast<ScAddress::Convention>(aLBFormulaSyntax.GetSelectEntryPos()) );
 
 	if ( *pLocalOptions != *pOldOptions )
 	{
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/src/optdlg.src sc/source/ui/src/optdlg.src
--- sc.clean/source/ui/src/optdlg.src	2008-03-03 11:24:45.000000000 -0500
+++ sc/source/ui/src/optdlg.src	2008-03-19 16:40:53.000000000 -0400
@@ -173,6 +173,27 @@ TabPage RID_SCPAGE_CALC
         Size = MAP_APPFONT ( 239 , 10 ) ;
 		Text [ en-US ] = "~Automatically find column and row labels " ;
 	};
+
+    FixedText FT_FORMULA_SYNTAX
+    {
+        Pos = MAP_APPFONT ( 21, 147 ) ;
+        Size = MAP_APPFONT ( 80, 8 ) ;
+        Text [ en-US ] = "~Formula syntax" ;
+    };
+
+    ListBox LB_FORMULA_SYNTAX
+    {
+        Border = TRUE ;
+        Pos = MAP_APPFONT ( 85, 145 ) ;
+        Size = MAP_APPFONT ( 60, 46 ) ;
+        DropDown = TRUE ;
+        StringList [ en-US ] =
+        {
+            < "Calc A1" ; Default ; > ;
+            < "Excel A1" ; Default ; > ;
+            < "Excel R1C1" ; Default ; > ;
+        };
+    };
 };
 
 /**************************************************************************/
Only in sc/source/ui/view: tabvwshf.cxx.~1.13.~
Only in sc/util: makefile.mk.~1.45.~
Only in officecfg: officecfg.argsep
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=officecfg.vpj officecfg.clean/registry/schema/org/openoffice/Office/Calc.xcs officecfg/registry/schema/org/openoffice/Office/Calc.xcs
--- officecfg.clean/registry/schema/org/openoffice/Office/Calc.xcs	2008-03-03 11:28:58.000000000 -0500
+++ officecfg/registry/schema/org/openoffice/Office/Calc.xcs	2008-03-19 18:32:54.000000000 -0400
@@ -1134,6 +1134,30 @@
 					</info>
 					<value>true</value>
 				</prop>
+				<prop oor:name="FormulaSyntax" oor:type="xs:int">
+					<!-- UIHints: Tools - Options  Spreadsheet  Calculate -->
+					<info>
+						<author>kyoshida</author>
+						<desc>Specifies what formula syntax type to use.</desc>
+					</info>
+					<constraints>
+						<enumeration oor:value="0">
+							<info>
+								<desc>Calc A1</desc>
+							</info>
+						</enumeration>
+						<enumeration oor:value="1">
+							<info>
+								<desc>Excel A1</desc>
+							</info>
+						</enumeration>
+						<enumeration oor:value="2">
+							<info>
+								<desc>Excel R1C1</desc>
+							</info>
+						</enumeration>
+					</constraints>
+				</prop>
 				<node-ref oor:name="Date" oor:node-type="Date">
 					<info>
 						<desc>Specifies the date that is represented by the number 0.</desc>
