Index: sc/source/core/data/column.cxx
===================================================================
--- sc/source/core/data/column.cxx.orig	2004-11-26 13:35:37.996537216 +0000
+++ sc/source/core/data/column.cxx	2004-11-26 17:09:57.853541792 +0000
@@ -1322,6 +1322,9 @@ void ScColumn::CopyToClip(USHORT nRow1, 
 	for (i = 0; i < nCount; i++)
 		if ((pItems[i].nRow >= nRow1) && (pItems[i].nRow <= nRow2))
 		{
+
+			if ( pDocument->RowFiltered( pItems[i].nRow , nTab))
+				continue;
 			if (!nBlockCount)
 				nStartIndex = i;
 			nEndIndex = i;
@@ -1342,7 +1345,11 @@ void ScColumn::CopyToClip(USHORT nRow1, 
 	{
 		rColumn.Resize( rColumn.GetCellCount() + nBlockCount );
 		for (i = nStartIndex; i <= nEndIndex; i++)
+			{
+			if ( pDocument->RowFiltered( pItems[i].nRow , nTab))
+				continue;			
 			rColumn.Append(pItems[i].nRow, pItems[i].pCell->Clone(rColumn.pDocument));
+			}
 	}
 }
 
Index: sc/source/core/data/column3.cxx
===================================================================
--- sc/source/core/data/column3.cxx.orig	2004-11-26 13:35:37.999536760 +0000
+++ sc/source/core/data/column3.cxx	2004-11-26 17:09:57.867539664 +0000
@@ -398,6 +398,17 @@ void ScColumn::DeleteRange( USHORT nStar
 	BOOL bSimple = ((nDelFlag & IDF_CONTENTS) == IDF_CONTENTS);
 	USHORT i;
 
+	/*
+	  *If we are deleting a range that has filtered column, let it not be simple 
+	  */
+	  
+	if (bSimple)
+	{
+		for (i = nStartIndex; i <= nEndIndex && bSimple; i++)
+			if ( pDocument->RowFiltered(pItems[i].nRow, nTab))
+				bSimple = FALSE;
+	}
+
 		//	Notiz-Zeichenobjekte
 	if (nDelFlag & IDF_NOTE)
 	{
@@ -452,6 +463,16 @@ void ScColumn::DeleteRange( USHORT nStar
 			BOOL bDelete = FALSE;
 			ScBaseCell* pOldCell = pItems[j].pCell;
 			CellType eCellType = pOldCell->GetCellType();
+			
+			/*
+			  * If the rows are filtered continue instead deleting.
+			  */
+			if ( pDocument->RowFiltered( pItems[j].nRow, nTab))
+				{
+				++j;
+				continue;
+				}
+			
 			switch ( eCellType )
 			{
 				case CELLTYPE_VALUE:
Index: sc/inc/document.hxx
===================================================================
--- sc/inc/document.hxx.orig	2004-11-26 13:35:38.064526880 +0000
+++ sc/inc/document.hxx	2004-11-26 17:10:55.795733240 +0000
@@ -508,6 +508,8 @@ public:
 	long			GetWeightedCount() const;	// Formeln und Edit staerker gewichtet
 	ULONG			GetCodeCount() const;		// RPN-Code in Formeln
 	DECL_LINK( GetUserDefinedColor, USHORT * );
+	BOOL 		RowFiltered( USHORT nRow, USHORT nTab ) const;	// FillInfo
+	BOOL		ColFiltered( USHORT nCol, USHORT nTab ) const;	// FillInfo
 																// Numberformatter
 
 public:
@@ -1698,6 +1700,15 @@ inline BOOL ScDocument::RowHidden( USHOR
 	return ( pTab[nTab]->pRowFlags[nRow] & CR_HIDDEN ) != 0;
 }
 
+inline BOOL ScDocument::ColFiltered( USHORT nCol, USHORT nTab ) const
+{
+	return ( pTab[nTab]->pColFlags[nCol] & CR_FILTERED) != 0;
+}
+
+inline BOOL ScDocument::RowFiltered( USHORT nRow, USHORT nTab ) const
+{
+	return pTab[nTab]->IsFiltered(nRow);
+}
 
 
 #endif
