Index: sc/source/core/tool/interpr1.cxx
===================================================================
RCS file: /cvs/sc/sc/source/core/tool/interpr1.cxx,v
retrieving revision 1.53
diff -u -r1.53 interpr1.cxx
--- sc/source/core/tool/interpr1.cxx	1 Nov 2007 16:23:17 -0000	1.53
+++ sc/source/core/tool/interpr1.cxx	8 Jan 2008 17:17:34 -0000
@@ -2768,7 +2768,9 @@
 void ScInterpreter::ScMin( BOOL bTextAsZero )
 {
 	BYTE nParamCount = GetByte();
-	double nMin = SC_DOUBLE_MAXVALUE;
+    if (!MustHaveParamCountMin( nParamCount, 1))
+        return;
+    double nMin = ::std::numeric_limits<double>::max();
 	double nVal = 0.0;
 	ScAddress aAdr;
 	ScRange aRange;
@@ -2805,18 +2807,18 @@
 				USHORT nErr = 0;
 				PopDoubleRef( aRange );
 				ScValueIterator aValIter( pDok, aRange, glSubTotal, bTextAsZero );
-				if (aValIter.GetFirst(nVal, nErr))
-				{
-					if (nMin > nVal)
-						nMin = nVal;
-					aValIter.GetCurNumFmtInfo( nFuncFmtType, nFuncFmtIndex );
-					while ((nErr == 0) && aValIter.GetNext(nVal, nErr))
-					{
-						if (nMin > nVal)
-							nMin = nVal;
-					}
-					SetError(nErr);
-				}
+                if (aValIter.GetFirst(nVal, nErr))
+                {
+                    if (nMin > nVal)
+                        nMin = nVal;
+                    aValIter.GetCurNumFmtInfo( nFuncFmtType, nFuncFmtIndex );
+                    while ((nErr == 0) && aValIter.GetNext(nVal, nErr))
+                    {
+                        if (nMin > nVal)
+                            nMin = nVal;
+                    }
+                    SetError(nErr);
+                }
 			}
 			break;
 			case svMatrix :
@@ -2875,9 +2877,9 @@
 				SetError(errIllegalParameter);
 		}
 	}
-	if (nMin == SC_DOUBLE_MAXVALUE)
-		SetIllegalArgument();
-	else
+	if ( nVal < nMin  )
+		PushDouble(0.0);
+	else		
 		PushDouble(nMin);
 }
 
@@ -2888,7 +2890,9 @@
 void ScInterpreter::ScMax( BOOL bTextAsZero )
 {
 	BYTE nParamCount = GetByte();
-	double nMax = -SC_DOUBLE_MAXVALUE;
+    if (!MustHaveParamCountMin( nParamCount, 1))
+        return;
+    double nMax = -(::std::numeric_limits<double>::max());
 	double nVal = 0.0;
 	ScAddress aAdr;
 	ScRange aRange;
@@ -2995,8 +2999,8 @@
 				SetError(errIllegalParameter);
 		}
 	}
-	if (nMax == -SC_DOUBLE_MAXVALUE)
-		SetIllegalArgument();
+    if ( nVal > nMax  )
+        PushDouble(0.0);
 	else
 		PushDouble(nMax);
 }
