Index: sc/source/ui/view/preview.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/preview.cxx,v
retrieving revision 1.32
diff -u -p -r1.32 preview.cxx
--- sc/source/ui/view/preview.cxx	10 May 2007 17:01:54 -0000	1.32
+++ sc/source/ui/view/preview.cxx	15 Dec 2007 03:19:12 -0000
@@ -231,8 +231,17 @@ void ScPreview::CalcPages( SCTAB nToWhic
 	//	but always all sheets are used (there is no selected sheet)
 	ScPrintOptions aOptions = SC_MOD()->GetPrintOptions();
 
+	ScMarkData aMarkData;
+	aMarkData.SelectTable( ScDocShell::GetCurTab(), TRUE );
+
 	for (SCTAB i=nStart; i<nAnz; i++)
 	{
+		if (!aOptions.GetAllSheets() && !aMarkData.GetTableSelect( i )) {
+			nPages[i] = 0;
+			nFirstAttr[i] = 0;
+			continue;
+		}
+
 		long nAttrPage = i > 0 ? nFirstAttr[i-1] : 1;
 
 		long nThisStart = nTotalPages;
Index: sc/source/ui/view/tabvwsh4.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/tabvwsh4.cxx,v
retrieving revision 1.70
diff -u -p -r1.70 tabvwsh4.cxx
--- sc/source/ui/view/tabvwsh4.cxx	26 Nov 2007 18:43:29 -0000	1.70
+++ sc/source/ui/view/tabvwsh4.cxx	15 Dec 2007 03:19:14 -0000
@@ -1207,8 +1215,13 @@ PrintDialog* __EXPORT ScTabViewShell::Cr
     bool bAllTabs = SC_MOD()->GetPrintOptions().GetAllSheets();
     pDlg->CheckSheetRange( bAllTabs ? PRINTSHEETS_ALL : PRINTSHEETS_SELECTED_SHEETS );
 
+	ScMarkData aMarkData;
+	aMarkData.SelectTable( GetViewData()->GetTabNo(), TRUE );
+
 	for ( SCTAB i=0; i<nTabCount; i++ )
 	{
+		if ( !bAllTabs && !aMarkData.GetTableSelect( i ) )
+			continue;
 		ScPrintFunc aPrintFunc( pDocShell, pPrinter, i );
 		nDocPageMax += aPrintFunc.GetTotalPages();
 	}
