--- sc/source/ui/view/tabcont.cxx	2005-03-17 10:55:46.000000000 -0800
+++ sc/source/ui/view/tabcont.cxx	2005-03-17 10:55:10.000000000 -0800
@@ -199,6 +199,11 @@
     // mouse button down and up on same page?
     if( nMouseClickPageId != GetPageId( aPos ) )
         nMouseClickPageId = TAB_PAGE_NOTFOUND;
+    else if(rMEvt.GetClicks()==2 && rMEvt.IsLeft())
+                {
+                        SfxDispatcher* pDispatcher = pViewData->GetViewShell()->GetViewFrame()->GetDispatcher();
+                        pDispatcher->Execute(FID_TAB_MENU_RENAME, SFX_CALLMODE_SYNCHRON | SFX_CALLMODE_RECORD);
+                }
 
 
     if( nMouseClickPageId == 0 )

