diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=officecfg.vpj officecfg.clean/registry/data/org/openoffice/Office/UI/CalcCommands.xcu officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu
--- officecfg.clean/registry/data/org/openoffice/Office/UI/CalcCommands.xcu	2008-02-20 15:04:18.000000000 -0500
+++ officecfg/registry/data/org/openoffice/Office/UI/CalcCommands.xcu	2008-02-21 11:25:19.000000000 -0500
@@ -1439,6 +1439,14 @@
                     <value xml:lang="en-US">E-mail as ~OpenDocument Spreadsheet...</value>
                 </prop>
             </node>
+            <node oor:name=".uno:ToggleSheetGrid" oor:op="replace">
+                <prop oor:name="Label" oor:type="xs:string">
+                    <value xml:lang="en-US">Toggle Grid Lines for Current Sheet</value>
+                </prop>
+                <prop oor:name="Properties" oor:type="xs:int">
+                    <value>1</value>
+                </prop>
+            </node>
         </node>
 		<node oor:name="Popups">
 			<node oor:name=".uno:AuditMenu" oor:op="replace">
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/document.hxx sc/inc/document.hxx
--- sc.clean/inc/document.hxx	2008-02-05 11:16:54.000000000 -0500
+++ sc/inc/document.hxx	2008-02-20 00:41:43.000000000 -0500
@@ -162,6 +162,7 @@ class ScAutoNameCache;
 class ScTemporaryChartLock;
 class ScLookupCache;
 struct ScLookupCacheMapImpl;
+struct ScTableViewOptions;
 
 namespace com { namespace sun { namespace star {
     namespace lang {
@@ -1703,6 +1704,8 @@ public:
                     /// an ID unique to each document instance
     sal_uInt32      GetDocumentID() const;
 
+    ScTableViewOptions* GetTableViewOptions( SCTAB nTab ) const;
+
     void            InvalidateStyleSheetUsage()
                         { bStyleSheetUsageInvalid = TRUE; }
 	void GetSortParam( ScSortParam& rParam, SCTAB nTab );
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/sc.hrc sc/inc/sc.hrc
--- sc.clean/inc/sc.hrc	2008-02-05 11:16:54.000000000 -0500
+++ sc/inc/sc.hrc	2008-02-20 18:35:44.000000000 -0500
@@ -821,6 +821,7 @@
 #define FID_TAB_APPEND			(TAB_POPUP_START+5)
 #define FID_TAB_INDEX			(TAB_POPUP_START+6)
 #define FID_TAB_RTL				(TAB_POPUP_START+7)
 #define FID_TAB_USE_R1C1		(TAB_POPUP_START+8)
+#define FID_TAB_TOGGLE_GRID     (TAB_POPUP_START+9)
 
 #define TAB_POPUP_END			(DATA_MENU_END + 20)
 
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/table.hxx sc/inc/table.hxx
--- sc.clean/inc/table.hxx	2008-02-05 11:16:54.000000000 -0500
+++ sc/inc/table.hxx	2008-02-20 00:30:49.000000000 -0500
@@ -37,6 +37,7 @@
 #define SC_TABLE_HXX
 
 #include <vector>
+#include <memory>
 
 #ifndef _GEN_HXX //autogen
 #include <tools/gen.hxx>
@@ -89,6 +90,7 @@ class ScIndexMap;
 struct RowInfo;
 struct ScFunctionData;
 struct ScLineFlags;
+struct ScTableViewOptions;
 class CollatorWrapper;
 
 
@@ -166,6 +168,8 @@ private:
 	USHORT			nScenarioFlags;
 	BOOL			bActiveScenario;
 
+    ::std::auto_ptr<ScTableViewOptions> pViewOptions;
+
 friend class ScDocument;					// fuer FillInfo
 friend class ScDocumentIterator;
 friend class ScValueIterator;
@@ -183,6 +187,7 @@ public:
 							BOOL bColInfo = TRUE, BOOL bRowInfo = TRUE );
 				~ScTable();
 
+    ScTableViewOptions* GetViewOptions();
 	ScOutlineTable*	GetOutlineTable()				{ return pOutlineTable; }
 
 	ULONG		GetCellCount() const;
Only in sc/inc: tabviewoptions.hxx
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/inc/ViewSettingsSequenceDefines.hxx sc/inc/ViewSettingsSequenceDefines.hxx
--- sc.clean/inc/ViewSettingsSequenceDefines.hxx	2008-02-05 11:16:54.000000000 -0500
+++ sc/inc/ViewSettingsSequenceDefines.hxx	2008-02-21 00:06:54.000000000 -0500
@@ -69,7 +69,7 @@
 // this are the defines for the position of the settings in the
 // TableViewSettingsSequence
 
-#define SC_TABLE_VIEWSETTINGS_COUNT         14
+#define SC_TABLE_VIEWSETTINGS_COUNT         15
 
 #define SC_CURSOR_X							0
 #define SC_CURSOR_Y							1
@@ -85,6 +85,7 @@
 #define SC_TABLE_ZOOM_TYPE                  11
 #define SC_TABLE_ZOOM_VALUE                 12
 #define SC_TABLE_PAGE_VIEW_ZOOM_VALUE       13
+#define SC_TABLE_SHOWGRID                   14
 
 #define SC_CURSORPOSITIONX					"CursorPositionX"
 #define SC_CURSORPOSITIONY					"CursorPositionY"
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/sdi/docsh.sdi sc/sdi/docsh.sdi
--- sc.clean/sdi/docsh.sdi	2008-02-05 11:16:54.000000000 -0500
+++ sc/sdi/docsh.sdi	2008-02-20 18:33:41.000000000 -0500
@@ -68,6 +68,7 @@ interface TableSelection
 	FID_DELETE_TABLE	[ ExecMethod = ExecuteTable; StateMethod = GetStateTable; ]
 	FID_TAB_RENAME		[ ExecMethod = ExecuteTable; StateMethod = GetStateTable; ]
 	FID_TAB_RTL			[ ExecMethod = ExecuteTable; StateMethod = GetStateTable; ]
 	FID_TAB_USE_R1C1	[ ExecMethod = ExecuteTable; StateMethod = GetStateTable; ]
+    FID_TAB_TOGGLE_GRID [ ExecMethod = ExecuteTable; StateMethod = GetStateTable; ]
 
 	SID_TABLE_ACTIVATE	[ ExecMethod = Execute; ]
 }
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/sdi/scalc.sdi sc/sdi/scalc.sdi
--- sc.clean/sdi/scalc.sdi	2008-02-05 11:16:54.000000000 -0500
+++ sc/sdi/scalc.sdi	2008-02-20 18:05:05.000000000 -0500
@@ -7706,3 +7706,28 @@ SfxBoolItem SheetRightToLeft FID_TAB_RTL
 	GroupId = GID_FORMAT;
 ]
 
+//--------------------------------------------------------------------------
+SfxBoolItem ToggleSheetGrid FID_TAB_TOGGLE_GRID
+
+[
+	/* flags: */
+	AutoUpdate = FALSE,
+	Cachable = Cachable,
+	FastCall = FALSE,
+	HasCoreId = FALSE,
+	HasDialog = FALSE,
+	ReadOnlyDoc = TRUE,
+	Toggle = FALSE,
+	Container = FALSE,
+	RecordAbsolute = FALSE,
+	RecordPerSet;
+	Synchron;
+
+	/* config: */
+	AccelConfig = TRUE,
+	MenuConfig = TRUE,
+	StatusBarConfig = FALSE,
+	ToolBoxConfig = TRUE,
+	GroupId = GID_FORMAT;
+]
+
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/data/documen2.cxx sc/source/core/data/documen2.cxx
--- sc.clean/source/core/data/documen2.cxx	2008-02-05 11:16:59.000000000 -0500
+++ sc/source/core/data/documen2.cxx	2008-02-20 00:46:26.000000000 -0500
@@ -277,6 +277,13 @@ sal_uInt32 ScDocument::GetDocumentID() c
     return nCrc;
 }
 
+ScTableViewOptions* ScDocument::GetTableViewOptions( SCTAB nTab ) const
+{
+    if (ValidTab(nTab) && pTab[nTab])
+        return pTab[nTab]->GetViewOptions();
+
+    return NULL;
+}
 
 void ScDocument::StartChangeTracking()
 {
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/data/makefile.mk sc/source/core/data/makefile.mk
--- sc.clean/source/core/data/makefile.mk	2008-02-05 11:16:59.000000000 -0500
+++ sc/source/core/data/makefile.mk	2008-02-20 01:16:25.000000000 -0500
@@ -113,7 +113,8 @@ SLOFILES =  \
 	$(SLO)$/table6.obj \
 	$(SLO)$/userdat.obj \
 	$(SLO)$/validat.obj \
-	$(SLO)$/postit.obj
+	$(SLO)$/postit.obj \
+	$(SLO)$/tabviewoptions.obj
 
 EXCEPTIONSFILES= \
     $(SLO)$/autonamecache.obj \
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/core/data/table1.cxx sc/source/core/data/table1.cxx
--- sc.clean/source/core/data/table1.cxx	2008-02-05 11:16:59.000000000 -0500
+++ sc/source/core/data/table1.cxx	2008-02-20 00:33:00.000000000 -0500
@@ -119,6 +119,7 @@
 #include "progress.hxx"
 #include "hints.hxx"		// fuer Paint-Broadcast
 #include "prnsave.hxx"
+#include "tabviewoptions.hxx"
 
 // STATIC DATA -----------------------------------------------------------
 
@@ -158,7 +159,8 @@ ScTable::ScTable( ScDocument* pDoc, SCTA
 	pScenarioRanges( NULL ),
 	aScenarioColor( COL_LIGHTGRAY ),
 	nScenarioFlags( 0 ),
-	bActiveScenario( FALSE )
+	bActiveScenario( FALSE ),
+    pViewOptions( NULL )
 {
 
 	if (bColInfo)
@@ -228,6 +230,13 @@ ScTable::~ScTable()
 	DestroySortCollator();
 }
 
+ScTableViewOptions* ScTable::GetViewOptions()
+{
+    if (!pViewOptions.get())
+        pViewOptions.reset(new ScTableViewOptions);
+    return pViewOptions.get();
+}
+
 void ScTable::GetName( String& rName ) const
 {
 	rName = aName;
Only in sc/source/core/data: tabviewoptions.cxx
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/filter/excel/xeview.cxx sc/source/filter/excel/xeview.cxx
--- sc.clean/source/filter/excel/xeview.cxx	2008-02-05 11:16:59.000000000 -0500
+++ sc/source/filter/excel/xeview.cxx	2008-02-20 12:40:45.000000000 -0500
@@ -57,6 +57,8 @@
 #include "xestyle.hxx"
 #endif
 
+#include "tabviewoptions.hxx"
+
 // Workbook view settings records =============================================
 
 XclExpWindow1::XclExpWindow1( const XclExpRoot& rRoot ) :
@@ -249,11 +251,14 @@ XclExpTabViewSettings::XclExpTabViewSett
 
     const ScViewOptions& rViewOpt = GetDoc().GetViewOptions();
     maData.mbShowFormulas   = rViewOpt.GetOption( VOPT_FORMULAS );
-    maData.mbShowGrid       = rViewOpt.GetOption( VOPT_GRID );
     maData.mbShowHeadings   = rViewOpt.GetOption( VOPT_HEADER );
     maData.mbShowZeros      = rViewOpt.GetOption( VOPT_NULLVALS );
     maData.mbShowOutline    = rViewOpt.GetOption( VOPT_OUTLINER );
 
+    // Save per-sheet view options.
+    const ScTableViewOptions* pTabViewOpt = GetDoc().GetTableViewOptions(nScTab);
+    maData.mbShowGrid = pTabViewOpt->mbDisplayGrid;
+
     // *** sheet options: cursor, selection, splits, grid color, zoom ***
 
     if( const ScExtTabSettings* pTabSett = GetExtDocOptions().GetTabSettings( nScTab ) )
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/filter/excel/xiview.cxx sc/source/filter/excel/xiview.cxx
--- sc.clean/source/filter/excel/xiview.cxx	2008-02-05 11:16:59.000000000 -0500
+++ sc/source/filter/excel/xiview.cxx	2008-02-20 12:28:11.000000000 -0500
@@ -60,6 +60,8 @@
 #include "xistyle.hxx"
 #endif
 
+#include "tabviewoptions.hxx"
+
 // Document view settings =====================================================
 
 XclImpDocViewSettings::XclImpDocViewSettings( const XclImpRoot& rRoot ) :
@@ -295,12 +297,16 @@ void XclImpTabViewSettings::Finalize()
         // set Excel sheet settings globally at Calc document, take settings from displayed sheet
         ScViewOptions aViewOpt( rDoc.GetViewOptions() );
         aViewOpt.SetOption( VOPT_FORMULAS, maData.mbShowFormulas );
-        aViewOpt.SetOption( VOPT_GRID,     maData.mbShowGrid );
         aViewOpt.SetOption( VOPT_HEADER,   maData.mbShowHeadings );
         aViewOpt.SetOption( VOPT_NULLVALS, maData.mbShowZeros );
         aViewOpt.SetOption( VOPT_OUTLINER, maData.mbShowOutline );
         rDoc.SetViewOptions( aViewOpt );
     }
+
+    // Store sheet view options.
+
+    ScTableViewOptions* pTabViewOptions = rDoc.GetTableViewOptions(nScTab);
+    pTabViewOptions->mbDisplayGrid = maData.mbShowGrid;
 }
 
 // ============================================================================
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/view/gridwin4.cxx sc/source/ui/view/gridwin4.cxx
--- sc.clean/source/ui/view/gridwin4.cxx	2008-02-05 11:16:56.000000000 -0500
+++ sc/source/ui/view/gridwin4.cxx	2008-02-21 10:13:46.000000000 -0500
@@ -79,6 +79,7 @@
 #include "inputopt.hxx"
 #include "fillinfo.hxx"
 #include "sc.hrc"
+#include "tabviewoptions.hxx"
 
 #ifndef _SV_VIRDEV_HXX
 #include <vcl/virdev.hxx>
@@ -564,7 +565,13 @@ void ScGridWindow::Draw( SCCOL nX1, SCRO
 	aOutputData.SetEditObject( GetEditObject() );
 	aOutputData.SetViewShell( pViewData->GetViewShell() );
 
-	BOOL bGrid = rOpts.GetOption( VOPT_GRID );
+    BOOL bGrid = rOpts.GetOption( VOPT_GRID );
+    if (bGrid)
+    {
+        const ScTableViewOptions* pTabOptions = pDoc->GetTableViewOptions(nTab);
+        bGrid = pTabOptions->mbDisplayGrid;
+    }
+
 	BOOL bPage = rOpts.GetOption( VOPT_PAGEBREAKS );
 
 	if ( eMode == SC_UPDATE_CHANGED )
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/view/tabvwshf.cxx sc/source/ui/view/tabvwshf.cxx
--- sc.clean/source/ui/view/tabvwshf.cxx	2008-02-05 11:16:56.000000000 -0500
+++ sc/source/ui/view/tabvwshf.cxx	2008-02-20 23:48:40.000000000 -0500
@@ -42,6 +42,8 @@
 
 #include "scitems.hxx"
 #include <sfx2/request.hxx>
+#include <sfx2/bindings.hxx>
+#include <sfx2/viewfrm.hxx>
 #include <basic/sbstar.hxx>
 #include <svtools/languageoptions.hxx>
 #include <svtools/stritem.hxx>
@@ -59,7 +61,7 @@
 //CHINA001 #include "strindlg.hxx"
 //CHINA001 #include "mvtabdlg.hxx"
 #include "docfunc.hxx"
-
+#include "tabviewoptions.hxx"
 
 #include "scabstdlg.hxx" //CHINA001
 
@@ -685,6 +687,17 @@ void ScTabViewShell::ExecuteTable( SfxRe
 			}
 			break;
 
+        case FID_TAB_TOGGLE_GRID:
+            {
+                ScTableViewOptions* pViewOpt = pDoc->GetTableViewOptions(nCurrentTab);
+                pViewOpt->mbDisplayGrid = !pViewOpt->mbDisplayGrid;
+                SfxBindings& rBindings = GetViewFrame()->GetBindings();
+                rBindings.Invalidate( FID_TAB_TOGGLE_GRID );
+                PaintGrid();
+                rReq.Done();
+            }
+            break;
+
 		default:
 			DBG_ERROR("Unbekannte Message bei ViewShell");
 			break;
@@ -800,6 +813,13 @@ void ScTabViewShell::GetStateTable( SfxI
 						rSet.Put( SfxBoolItem( nWhich, pDoc->IsLayoutRTL( nTab ) ) );
 				}
 				break;
+
+            case FID_TAB_TOGGLE_GRID:
+                {
+                    const ScTableViewOptions* pTabViewOptions = pDoc->GetTableViewOptions(nTab);
+                    rSet.Put( SfxBoolItem(nWhich, pTabViewOptions->mbDisplayGrid) );
+                }
+                break;
 		}
 		nWhich = aIter.NextWhich();
 	}
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/source/ui/view/viewdata.cxx sc/source/ui/view/viewdata.cxx
--- sc.clean/source/ui/view/viewdata.cxx	2008-02-05 11:16:56.000000000 -0500
+++ sc/source/ui/view/viewdata.cxx	2008-02-21 00:17:16.000000000 -0500
@@ -73,6 +73,7 @@
 #include "miscuno.hxx"
 #include "unonames.hxx"
 #include "inputopt.hxx"
+#include "tabviewoptions.hxx"
 
 #ifndef _XMLOFF_XMLUCONV_HXX
 #include <xmloff/xmluconv.hxx>
@@ -137,7 +138,7 @@ ScViewDataTable::~ScViewDataTable()
 {
 }
 
-void ScViewDataTable::WriteUserDataSequence(uno::Sequence <beans::PropertyValue>& rSettings, const ScViewData& /*rViewData*/, SCTAB /*nTab*/)
+void ScViewDataTable::WriteUserDataSequence(uno::Sequence <beans::PropertyValue>& rSettings, const ScViewData& rViewData, SCTAB nTab)
 {
 	rSettings.realloc(SC_TABLE_VIEWSETTINGS_COUNT);
 	beans::PropertyValue* pSettings = rSettings.getArray();
@@ -180,6 +180,10 @@ void ScViewDataTable::WriteUserDataSeque
         pSettings[SC_TABLE_ZOOM_VALUE].Value <<= nZoomValue;
         pSettings[SC_TABLE_PAGE_VIEW_ZOOM_VALUE].Name = rtl::OUString(RTL_CONSTASCII_USTRINGPARAM(SC_PAGEVIEWZOOMVALUE));
         pSettings[SC_TABLE_PAGE_VIEW_ZOOM_VALUE].Value <<= nPageZoomValue;
+
+        const ScTableViewOptions* pTabViewOpt = rViewData.GetDocument()->GetTableViewOptions(nTab);
+        pSettings[SC_TABLE_SHOWGRID].Name = rtl::OUString(RTL_CONSTASCII_USTRINGPARAM(SC_UNO_SHOWGRID));
+        pSettings[SC_TABLE_SHOWGRID].Value <<= static_cast<sal_Bool>(pTabViewOpt->mbDisplayGrid);
 	}
 }

@@ -287,6 +292,12 @@ void ScViewDataTable::ReadUserDataSequen
             aPageZoomX = aPageZoomY = aZoom;
             rHasZoom = true;
         }
+        else if (sName.compareToAscii(SC_UNO_SHOWGRID) == 0)
+        {
+            bool bShowGrid = true;
+            aSettings[i].Value >>= bShowGrid;
+            rViewData.GetDocument()->GetTableViewOptions(nTab)->mbDisplayGrid = bShowGrid;
+        }
         else if (sName.compareToAscii(SC_TABLESELECTED) == 0)
         {
             bool bSelected = false;
diff -urp --exclude=CVS --exclude=unxlngi6.pro --exclude=sc.vpj sc.clean/uiconfig/scalc/toolbar/formatobjectbar.xml sc/uiconfig/scalc/toolbar/formatobjectbar.xml
--- sc.clean/uiconfig/scalc/toolbar/formatobjectbar.xml	2008-02-05 11:16:54.000000000 -0500
+++ sc/uiconfig/scalc/toolbar/formatobjectbar.xml	2008-02-21 11:32:04.000000000 -0500
@@ -50,4 +50,6 @@
  <toolbar:toolbaritem xlink:href=".uno:InsertColumns" toolbar:helpid="helpid:26268"  toolbar:visible="false" />
  <toolbar:toolbaritem xlink:href=".uno:DeleteRows" toolbar:helpid="helpid:26236"  toolbar:visible="false" />
  <toolbar:toolbaritem xlink:href=".uno:DeleteColumns" toolbar:helpid="helpid:26237"  toolbar:visible="false" />
+ <toolbar:toolbarseparator/>
+ <toolbar:toolbaritem xlink:href=".uno:ToggleSheetGrid" toolbar:helpid="helpid:26237"  toolbar:visible="true" />
 </toolbar:toolbar>
--- /dev/null	2007-03-30 11:25:40.000000000 -0400
+++ sc/inc/tabviewoptions.hxx	2008-02-20 01:15:22.000000000 -0500
@@ -0,0 +1,50 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: table.hxx,v $
+ *
+ *  $Revision: 1.32 $
+ *
+ *  last change: $Author: vg $ $Date: 2007/09/21 09:22:40 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2008 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#ifndef SC_TAB_VIEW_OPTIONS_HXX
+#define SC_TAB_VIEW_OPTIONS_HXX
+
+/** This class stores a set of persistent sheet view options that are saved
+    with the document.  This class is still work in progress, thus will
+    likely grow with more options in future. */
+struct ScTableViewOptions
+{
+    bool mbDisplayGrid;
+
+    ScTableViewOptions();
+};
+
+
+#endif
--- /dev/null	2007-03-30 11:25:40.000000000 -0400
+++ sc/source/core/data/tabviewoptions.cxx	2008-02-20 16:04:37.000000000 -0500
@@ -0,0 +1,41 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: table.hxx,v $
+ *
+ *  $Revision: 1.32 $
+ *
+ *  last change: $Author: vg $ $Date: 2007/09/21 09:22:40 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2008 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "tabviewoptions.hxx"
+
+ScTableViewOptions::ScTableViewOptions() :
+    mbDisplayGrid(true)
+{
+}
