--- sc/source/ui/src/localize.sdf.orig	2006-02-06 20:07:43.000000000 +0100
+++ sc/source/ui/src/localize.sdf	2006-06-14 18:09:53.000000000 +0200
@@ -9149,6 +9149,42 @@
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	10			0	zh-CN	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	10			0	zh-TW	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	10			0	zu	I-% elincane kakhulu				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	en-GB	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	de	Enthält				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	es	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	fr	Contient				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	it	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ja	を含む				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	pt-BR	Contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	zh-CN	包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	zh-TW	包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	de	Enthält nicht				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	es	No contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	fr	Ne contient pas				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	it	Non contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ja	を含まない				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	pt-BR	Não contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	zh-CN	不包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	zh-TW	不包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	en-GB	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	de	Beginnt mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	es	Empieza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	fr	Commence par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	it	Inizia con				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ja	で始まる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	pt-BR	Começa com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	zh-CN	开头是				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	zh-TW	開頭以				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	en-GB	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	de	Endet mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	es	Finaliza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	fr	Finit par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	it	Termina con				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ja	で終わる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	pt-BR	Termina com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	zh-CN	结尾是				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	zh-TW	結束以				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	2			0	af	<				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	2			0	ar	>				20040507 14:23:12
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	2			0	be-BY	<				2002-02-02 02:02:02
@@ -9683,6 +9719,42 @@
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	10			0	zh-CN	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	10			0	zh-TW	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	10			0	zu	I-% elincane kakhulu				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	en-GB	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	de	Enthält				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	es	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	fr	Contient				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	it	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ja	を含む				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	pt-BR	Contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	zh-CN	包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	zh-TW	包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	de	Enthält nicht				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	es	No contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	fr	Ne contient pas				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	it	Non contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ja	を含まない				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	pt-BR	Não contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	zh-CN	不包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	zh-TW	不包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	en-GB	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	de	Beginnt mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	es	Empieza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	fr	Commence par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	it	Inizia con				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ja	で始まる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	pt-BR	Começa com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	zh-CN	开头是				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	zh-TW	開頭以				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	en-GB	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	de	Endet mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	es	Finaliza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	fr	Finit par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	it	Termina con				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ja	で終わる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	pt-BR	Termina com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	zh-CN	结尾是				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	zh-TW	結束以				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	2			0	af	<				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	2			0	ar	>				20040507 14:23:12
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	2			0	be-BY	<				2002-02-02 02:02:02
@@ -10217,6 +10289,42 @@
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	10			0	zh-CN	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	10			0	zh-TW	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	10			0	zu	I-% elincane kakhulu				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	en-GB	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	de	Enthält				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	es	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	fr	Contient				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	it	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ja	を含む				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	pt-BR	Contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	zh-CN	包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	zh-TW	包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	de	Enthält nicht				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	es	No contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	fr	Ne contient pas				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	it	Non contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ja	を含まない				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	pt-BR	Não contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	zh-CN	不包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	zh-TW	不包含				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	en-GB	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	de	Beginnt mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	es	Empieza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	fr	Commence par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	it	Inizia con				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ja	で始まる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	pt-BR	Começa com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	zh-CN	开头是				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	zh-TW	開頭以				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	en-GB	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	de	Endet mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	es	Finaliza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	fr	Finit par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	it	Termina con				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ja	で終わる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	pt-BR	Termina com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	zh-CN	结尾是				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	zh-TW	結束以				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	2			0	af	<				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	2			0	ar	>				20040507 14:23:12
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	2			0	be-BY	<				2002-02-02 02:02:02
