--- sc/source/ui/src/localize.sdf.old	2006-06-14 18:09:53.000000000 +0200
+++ sc/source/ui/src/localize.sdf	2006-06-14 18:34:10.000000000 +0200
@@ -9149,42 +9149,190 @@
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	10			0	zh-CN	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	10			0	zh-TW	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	10			0	zu	I-% elincane kakhulu				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	en-GB	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	af	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ar	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	be-BY	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	bg	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ca	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	cs	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	cy	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	da	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	de	Enthält				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	el	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	en-GB	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	es	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	et	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	fi	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	fr	Contient				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	gu-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	hi-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	hr	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	hu	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	it	Contiene				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ja	を含む				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	km	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ko	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	lt	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	mk	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	nb	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	nl	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	nn	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	pa-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	pl	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	pt-BR	Contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	pt	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ru	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	rw	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	sk	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	sl	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	sr-CS	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	st	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	sv	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	tr	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	ts	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	vi	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	xh	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	zh-CN	包含				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	zh-TW	包含				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	11			0	zu	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	af	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ar	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	be-BY	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	bg	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ca	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	cs	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	cy	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	da	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	de	Enthält nicht				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	el	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	es	No contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	et	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	fi	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	fr	Ne contient pas				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	gu-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	hi-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	hr	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	hu	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	it	Non contiene				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ja	を含まない				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	km	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ko	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	lt	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	mk	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	nb	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	nl	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	nn	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	pa-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	pl	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	pt-BR	Não contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	pt	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ru	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	rw	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	sk	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	sl	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	sr-CS	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	st	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	sv	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	tr	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	ts	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	vi	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	xh	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	zh-CN	不包含				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	zh-TW	不包含				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	en-GB	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	12			0	zu	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	af	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ar	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	be-BY	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	bg	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ca	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	cs	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	cy	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	da	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	de	Beginnt mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	el	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	en-GB	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	es	Empieza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	et	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	fi	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	fr	Commence par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	gu-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	hi-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	hr	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	hu	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	it	Inizia con				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ja	で始まる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	km	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ko	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	lt	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	mk	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	nb	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	nl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	nn	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	pa-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	pl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	pt	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	pt-BR	Começa com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ru	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	rw	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	sk	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	sl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	sr-CS	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	st	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	sv	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	tr	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	ts	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	vi	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	xh	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	zh-CN	开头是				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	zh-TW	開頭以				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	en-GB	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	13			0	zu	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	af	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ar	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	be-BY	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	bg	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ca	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	cs	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	cy	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	da	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	de	Endet mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	el	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	en-GB	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	es	Finaliza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	et	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	fi	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	fr	Finit par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	gu-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	hi-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	hr	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	hu	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	it	Termina con				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ja	で終わる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	km	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ko	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	lt	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	mk	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	nb	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	nl	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	nn	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	pa-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	pl	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	pt-BR	Termina com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	pt	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ru	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	rw	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	sk	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	sl	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	sr-CS	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	st	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	sv	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	tr	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	ts	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	vi	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	xh	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	zh-CN	结尾是				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	zh-TW	結束以				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	14			0	zu	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	2			0	af	<				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	2			0	ar	>				20040507 14:23:12
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND1	2			0	be-BY	<				2002-02-02 02:02:02
@@ -9719,42 +9867,190 @@
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	10			0	zh-CN	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	10			0	zh-TW	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	10			0	zu	I-% elincane kakhulu				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	en-GB	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	af	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ar	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	be-BY	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	bg	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ca	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	cs	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	cy	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	da	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	de	Enthält				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	el	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	en-GB	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	es	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	et	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	fi	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	fr	Contient				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	gu-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	hi-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	hr	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	hu	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	it	Contiene				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ja	を含む				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	km	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ko	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	lt	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	mk	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	nb	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	nl	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	nn	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	pa-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	pl	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	pt-BR	Contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	pt	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ru	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	rw	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	sk	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	sl	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	sr-CS	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	st	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	sv	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	tr	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	ts	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	vi	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	xh	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	zh-CN	包含				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	zh-TW	包含				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	11			0	zu	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	af	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ar	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	be-BY	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	bg	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ca	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	cs	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	cy	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	da	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	de	Enthält nicht				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	el	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	es	No contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	et	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	fi	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	fr	Ne contient pas				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	gu-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	hi-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	hr	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	hu	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	it	Non contiene				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ja	を含まない				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	km	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ko	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	lt	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	mk	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	nb	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	nl	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	nn	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	pa-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	pl	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	pt-BR	Não contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	pt	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ru	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	rw	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	sk	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	sl	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	sr-CS	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	st	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	sv	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	tr	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	ts	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	vi	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	xh	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	zh-CN	不包含				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	zh-TW	不包含				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	en-GB	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	12			0	zu	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	af	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ar	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	be-BY	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	bg	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ca	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	cs	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	cy	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	da	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	de	Beginnt mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	el	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	en-GB	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	es	Empieza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	et	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	fi	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	fr	Commence par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	gu-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	hi-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	hr	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	hu	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	it	Inizia con				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ja	で始まる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	km	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ko	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	lt	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	mk	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	nb	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	nl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	nn	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	pa-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	pl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	pt	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	pt-BR	Começa com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ru	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	rw	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	sk	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	sl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	sr-CS	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	st	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	sv	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	tr	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	ts	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	vi	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	xh	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	zh-CN	开头是				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	zh-TW	開頭以				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	en-GB	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	13			0	zu	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	af	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ar	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	be-BY	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	bg	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ca	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	cs	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	cy	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	da	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	de	Endet mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	el	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	en-GB	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	es	Finaliza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	et	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	fi	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	fr	Finit par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	gu-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	hi-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	hr	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	hu	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	it	Termina con				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ja	で終わる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	km	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ko	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	lt	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	mk	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	nb	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	nl	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	nn	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	pa-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	pl	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	pt-BR	Termina com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	pt	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ru	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	rw	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	sk	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	sl	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	sr-CS	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	st	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	sv	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	tr	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	ts	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	vi	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	xh	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	zh-CN	结尾是				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	zh-TW	結束以				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	14			0	zu	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	2			0	af	<				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	2			0	ar	>				20040507 14:23:12
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND2	2			0	be-BY	<				2002-02-02 02:02:02
@@ -10289,42 +10585,190 @@
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	10			0	zh-CN	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	10			0	zh-TW	最小 %				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	10			0	zu	I-% elincane kakhulu				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	en-GB	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	af	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ar	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	be-BY	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	bg	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ca	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	cs	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	cy	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	da	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	de	Enthält				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	el	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	en-GB	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	es	Contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	et	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	fi	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	fr	Contient				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	gu-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	hi-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	hr	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	hu	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	it	Contiene				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ja	を含む				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	km	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ko	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	lt	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	mk	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	nb	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	nl	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	nn	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	pa-IN	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	pl	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	pt-BR	Contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	pt	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ru	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	rw	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	sk	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	sl	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	sr-CS	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	st	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	sv	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	tr	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	ts	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	vi	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	xh	Contains				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	zh-CN	包含				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	zh-TW	包含				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	11			0	zu	Contains				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	af	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ar	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	be-BY	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	bg	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ca	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	cs	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	cy	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	da	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	de	Enthält nicht				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	el	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	en-GB	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	es	No contiene				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	et	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	fi	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	fr	Ne contient pas				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	gu-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	hi-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	hr	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	hu	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	it	Non contiene				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ja	を含まない				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	km	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ko	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	lt	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	mk	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	nb	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	nl	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	nn	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	pa-IN	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	pl	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	pt-BR	Não contém				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	pt	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ru	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	rw	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	sk	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	sl	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	sr-CS	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	st	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	sv	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	tr	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	ts	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	vi	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	xh	Does not Contain				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	zh-CN	不包含				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	zh-TW	不包含				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	en-GB	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	12			0	zu	Does not Contain				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	af	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ar	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	be-BY	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	bg	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ca	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	cs	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	cy	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	da	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	de	Beginnt mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	el	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	en-GB	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	es	Empieza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	et	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	fi	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	fr	Commence par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	gu-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	hi-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	hr	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	hu	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	it	Inizia con				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ja	で始まる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	km	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ko	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	lt	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	mk	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	nb	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	nl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	nn	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	pa-IN	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	pl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	pt	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	pt-BR	Começa com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ru	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	rw	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	sk	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	sl	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	sr-CS	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	st	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	sv	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	tr	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	ts	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	vi	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	xh	Begins with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	zh-CN	开头是				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	zh-TW	開頭以				2002-02-02 02:02:02
-sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	en-GB	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	13			0	zu	Begins with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	af	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ar	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	be-BY	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	bg	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ca	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	cs	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	cy	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	da	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	de	Endet mit				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	el	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	en-GB	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	es	Finaliza por				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	et	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	fi	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	fr	Finit par				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	gu-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	hi-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	hr	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	hu	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	it	Termina con				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ja	で終わる				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	km	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ko	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	lt	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	mk	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	nb	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	nl	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	nn	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	pa-IN	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	pl	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	pt-BR	Termina com				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	pt	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ru	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	rw	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	sk	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	sl	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	sr-CS	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	st	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	sv	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	tr	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	ts	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	vi	Ends with				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	xh	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	zh-CN	结尾是				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	zh-TW	結束以				2002-02-02 02:02:02
+sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	14			0	zu	Ends with				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	2			0	af	<				2002-02-02 02:02:02
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	2			0	ar	>				20040507 14:23:12
 sc	source\ui\src\filter.src	0	stringlist	RID_SCDLG_FILTER.LB_COND3	2			0	be-BY	<				2002-02-02 02:02:02
