diff -u -r ../src680-m78.bak/sfx2/source/dialog/passwd.cxx sfx2/source/dialog/passwd.cxx
--- ../src680-m78.bak/sfx2/source/dialog/passwd.cxx	2005-02-23 17:00:37.000000000 +0530
+++ sfx2/source/dialog/passwd.cxx	2005-02-24 09:47:01.491560482 +0530
@@ -126,6 +126,7 @@
 	maPasswordED	( this, ResId( ED_PASSWD_PASSWORD ) ),
 	maConfirmFT		( this, ResId( FT_PASSWD_CONFIRM ) ),
 	maConfirmED		( this, ResId( ED_PASSWD_CONFIRM ) ),
+	maMinLengthFT	( this, ResId( FT_PASSWD_MINLEN ) ),
 	maPasswordBox	( this, ResId( GB_PASSWD_PASSWORD ) ),
 	maOKBtn			( this, ResId( BTN_PASSWD_OK ) ),
 	maCancelBtn		( this, ResId( BTN_PASSWD_CANCEL ) ),
@@ -205,8 +206,10 @@
 			aPos = maUserED.GetPosPixel();
 			maPasswordED.SetPosPixel( aPos );
 
+			aPos = maConfirmFT.GetPosPixel();
 			maConfirmFT.SetPosPixel( aPwdPos1 );
 			maConfirmED.SetPosPixel( aPwdPos2 );
+			maMinLengthFT.SetPosPixel(aPos);
 		}
 
 		Size aBoxSize = maPasswordBox.GetSizePixel();
diff -u -r ../src680-m78.bak/sfx2/source/dialog/passwd.hrc sfx2/source/dialog/passwd.hrc
--- ../src680-m78.bak/sfx2/source/dialog/passwd.hrc	2005-02-23 17:00:37.000000000 +0530
+++ sfx2/source/dialog/passwd.hrc	2005-02-24 09:39:59.089229986 +0530
@@ -70,6 +70,7 @@
 #define ED_PASSWD_PASSWORD 14
 #define FT_PASSWD_CONFIRM 15
 #define ED_PASSWD_CONFIRM 16
+#define FT_PASSWD_MINLEN 17
 
 #define BTN_PASSWD_OK 20
 #define BTN_PASSWD_CANCEL 21
diff -u -r ../src680-m78.bak/sfx2/source/dialog/passwd.src sfx2/source/dialog/passwd.src
--- ../src680-m78.bak/sfx2/source/dialog/passwd.src	2005-02-23 17:00:37.000000000 +0530
+++ sfx2/source/dialog/passwd.src	2005-02-24 09:56:29.023722648 +0530
@@ -117,6 +117,13 @@
 		Pos = MAP_APPFONT( 57, 44 );
 		Size = MAP_APPFONT( 75, 12 );
 	};
+	FixedText FT_PASSWD_MINLEN
+	{
+		Pos = MAP_APPFONT(12, 65 );
+		Size = MAP_APPFONT( 82, 10 );
+		Text [ de ] = "";
+		Text [ en-US ] = "(Minimum 5 characters)"; 
+	};
     FixedLine GB_PASSWD_PASSWORD
 	{
 		Pos = MAP_APPFONT( 6, 3 );
diff -u -r ../src680-m78.bak/sfx2/inc/passwd.hxx sfx2/inc/passwd.hxx
--- ../src680-m78.bak/sfx2/inc/passwd.hxx	2005-02-23 17:00:31.000000000 +0530
+++ sfx2/inc/passwd.hxx	2005-02-24 09:34:28.079468128 +0530
@@ -100,6 +100,7 @@
 	Edit			maPasswordED;
 	FixedText		maConfirmFT;
 	Edit			maConfirmED;
+	FixedText		maMinLengthFT;
     FixedLine       maPasswordBox;
 
 	OKButton		maOKBtn;
