--- sc/source/ui/view/viewfun2.cxx	2006-05-02 15:53:53.000000000 +0000
+++ sc/source/ui/view/viewfun2.cxx	2006-06-29 13:03:59.000000000 +0000
@@ -871,10 +871,15 @@ BOOL ScViewFunc::RemoveMerge( BOOL bReco
 	ScRange aRange;
 	if (GetViewData()->GetSimpleArea( aRange ))
 	{
+		ScRange aExtended( aRange );
+		GetViewData()->GetDocument()->ExtendMerge( aExtended );
 		ScDocShell* pDocSh = GetViewData()->GetDocShell();
+
 		HideCursor();
 		BOOL bOk = pDocSh->GetDocFunc().UnmergeCells( aRange, bRecord, FALSE );
+		MarkRange( aExtended );
 		ShowCursor();
+
 		if (bOk)
 			pDocSh->UpdateOle(GetViewData());
 	}
