Index: sc/inc/errorcodes.hxx
===================================================================
RCS file: /cvs/sc/sc/inc/errorcodes.hxx,v
retrieving revision 1.2
diff -u -p -u -r1.2 errorcodes.hxx
--- sc/inc/errorcodes.hxx	8 Mar 2004 11:40:52 -0000	1.2
+++ sc/inc/errorcodes.hxx	18 Apr 2005 11:26:47 -0000
@@ -74,7 +74,7 @@ namespace ScErrorCodes
 
 const USHORT errIllegalChar          = 501;
 const USHORT errIllegalArgument      = 502;
-const USHORT errIllegalFPOperation   = 503;
+const USHORT errIllegalFPOperation   = 503; // #NUM!
 const USHORT errIllegalParameter     = 504;
 const USHORT errIllegalJump          = 505;
 const USHORT errSeparator            = 506;
@@ -90,13 +90,13 @@ const USHORT errUnknownState         = 5
 const USHORT errUnknownVariable      = 516;
 const USHORT errUnknownOpCode        = 517;
 const USHORT errUnknownStackVariable = 518;
-const USHORT errNoValue              = 519;
+const USHORT errNoValue              = 519; // #VALUE!
 const USHORT errUnknownToken         = 520;
-const USHORT errNoCode               = 521;
+const USHORT errNoCode               = 521; // #NULL!
 const USHORT errCircularReference    = 522;
 const USHORT errNoConvergence        = 523;
-const USHORT errNoRef                = 524;
-const USHORT errNoName               = 525;
+const USHORT errNoRef                = 524; // #REF!
+const USHORT errNoName               = 525; // #NAME?
 const USHORT errDoubleRef            = 526;
 const USHORT errInterpOverflow       = 527;
 // Not displayed, temporary for TrackFormulas,
@@ -108,6 +108,7 @@ const USHORT errCellNoValue          = 5
 const USHORT errNoAddin              = 530;
 // Interpreter: needed Macro not found
 const USHORT errNoMacro              = 531;
+const USHORT errDivisionByZero       = 532; // #DIV/0!
 
 // Interpreter: NA() not available condition, not a real error
 const USHORT NOVALUE                 = 0x7fff;
Index: sc/inc/globstr.hrc
===================================================================
RCS file: /cvs/sc/sc/inc/globstr.hrc,v
retrieving revision 1.12
diff -u -p -u -r1.12 globstr.hrc
--- sc/inc/globstr.hrc	17 Sep 2004 13:52:04 -0000	1.12
+++ sc/inc/globstr.hrc	18 Apr 2005 11:26:47 -0000
@@ -576,7 +576,12 @@
 
 #define STR_ERR_DATAPILOT_INPUT     411
 
-#define STR_COUNT                   412
+#define STR_DIV_ZERO                412
+#define STR_LONG_ERR_DIV_ZERO       413
+#define STR_NUM_ERROR               414
+#define STR_NULL_ERROR              415
+
+#define STR_COUNT                   416
 
 #endif
 
Index: sc/source/filter/excel/excform.cxx
===================================================================
RCS file: /cvs/sc/sc/source/filter/excel/excform.cxx,v
retrieving revision 1.34
diff -u -p -u -r1.34 excform.cxx
--- sc/source/filter/excel/excform.cxx	21 Feb 2005 13:23:38 -0000	1.34
+++ sc/source/filter/excel/excform.cxx	18 Apr 2005 11:26:47 -0000
@@ -1546,8 +1546,8 @@ const ScTokenArray* ExcelToSc::GetBoolEr
 	switch( eType )
 	{
         case xlErrNull:     eOc = ocStop;       nError = errNoCode;             break;
-        case xlErrDiv0:     eOc = ocStop;       nError = errIllegalFPOperation; break;
-        case xlErrValue:    eOc = ocNoValue;    nError = NOVALUE;               break;
+        case xlErrDiv0:     eOc = ocStop;       nError = errDivisionByZero;     break;
+        case xlErrValue:    eOc = ocNoValue;    nError = errNoValue;            break;
         case xlErrRef:      eOc = ocStop;       nError = errNoRef;              break;
         case xlErrName:     eOc = ocStop;       nError = errNoName;             break;
         case xlErrNum:      eOc = ocStop;       nError = errIllegalFPOperation; break;
@@ -1662,8 +1662,8 @@ BOOL ExcelToSc::SetCurVal( ScFormulaCell
 				switch( nVal )
 				{// nInd temp. missbrauchen
 					case 0x00:	nInd = errNoCode; break;			// #NULL!
-					case 0x07:	nInd = errIllegalFPOperation; break;// #DIV/0!
-					case 0x0F:	nInd = NOVALUE; break;				// #VALUE!
+					case 0x07:	nInd = errDivisionByZero; break;    // #DIV/0!
+					case 0x0F:	nInd = errNoValue; break;			// #VALUE!
 					case 0x17:	nInd = errNoRef; break;				// #REF!
 					case 0x1D:	nInd = errNoName; break;			// #NAME?
 					case 0x24:	nInd = errIllegalFPOperation; break;// #NUM!
Index: sc/source/ui/src/globstr.src
===================================================================
RCS file: /cvs/sc/sc/source/ui/src/globstr.src,v
retrieving revision 1.61
diff -u -p -u -r1.61 globstr.src
--- sc/source/ui/src/globstr.src	14 Jan 2005 12:15:01 -0000	1.61
+++ sc/source/ui/src/globstr.src	18 Apr 2005 11:26:48 -0000
@@ -2587,6 +2588,29 @@ Resource RID_GLOBSTR
         Text [ de ] = "Sie können diesen Teil der Datenpilot-Tabelle nicht ändern.";
         Text[ en-US ] = "You cannot change this part of the DataPilot table.";
     };
-
+	String STR_DIV_ZERO
+	{
+		Text [ de ] = "#DIV/0!" ;
+		Text [ en-US ] = "#DIV/0!" ;
+		Text [ x-comment ] = " ";
+	};
+	String STR_LONG_ERR_DIV_ZERO
+	{
+		Text [ de ] = "Division by zero" ;
+		Text [ en-US ] = "Division by zero" ;
+		Text [ x-comment ] = " ";
+	};
+	String STR_NUM_ERROR
+	{
+		Text [ de ] = "#NUM!" ;
+		Text [ en-US ] = "#NUM!" ;
+		Text [ x-comment ] = " ";
+	};
+	String STR_NULL_ERROR
+	{
+		Text [ de ] = "#NUL!" ;
+		Text [ en-US ] = "#NULL!" ;
+		Text [ x-comment ] = " ";
+	};
 };
 
Index: sc/source/core/data/global.cxx
===================================================================
RCS file: /cvs/sc/sc/source/core/data/global.cxx,v
retrieving revision 1.42
diff -u -p -u -r1.42 global.cxx
--- sc/source/core/data/global.cxx	16 Feb 2005 18:06:28 -0000	1.42
+++ sc/source/core/data/global.cxx	18 Apr 2005 11:26:48 -0000
@@ -422,6 +422,9 @@ String ScGlobal::GetErrorString(USHORT n
         case errNoMacro  : nErrNumber = STR_NO_MACRO; break;
 		case errDoubleRef:
 		case errNoValue  : nErrNumber = STR_NO_VALUE; break;
+		case errNoCode : nErrNumber = STR_NULL_ERROR; break;
+		case errDivisionByZero : nErrNumber = STR_DIV_ZERO; break;
+		case errIllegalFPOperation : nErrNumber = STR_NUM_ERROR; break;
 
 		default		     : sResStr = GetRscString(STR_ERROR_STR);
 						   sResStr += String::CreateFromInt32( nErrNumber );
@@ -508,6 +511,9 @@ String ScGlobal::GetLongErrorString(USHO
         case errNoMacro:
             nErrNumber = STR_LONG_ERR_NO_MACRO;
 		break;
+        case errDivisionByZero:
+            nErrNumber = STR_LONG_ERR_DIV_ZERO;
+		break;
 		case errNoValue:
 			nErrNumber = STR_LONG_ERR_NO_VALUE;
 		break;
