Index: sc/inc/unonames.hxx
===================================================================
RCS file: /cvs/sc/sc/inc/unonames.hxx,v
retrieving revision 1.69
diff -u -p -r1.69 unonames.hxx
--- sc/inc/unonames.hxx	8 Sep 2005 18:02:11 -0000	1.69
+++ sc/inc/unonames.hxx	8 Feb 2006 09:52:31 -0000
@@ -344,6 +344,8 @@
 #define SC_UNO_SRCHSTYLES			"SearchStyles"
 #define SC_UNO_SRCHTYPE				"SearchType"
 #define SC_UNO_SRCHWORDS			"SearchWords"
+#define SC_UNO_SRCHSEARCH			"SearchString"
+#define SC_UNO_SRCHREPLACE			"ReplaceString"
 
 //	old (5.2) property names for page styles - for compatibility only!
 #define OLD_UNO_PAGE_BACKCOLOR		"BackgroundColor"
Index: sc/source/ui/unoobj/srchuno.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/unoobj/srchuno.cxx,v
retrieving revision 1.4
diff -u -p -r1.4 srchuno.cxx
--- sc/source/ui/unoobj/srchuno.cxx	8 Sep 2005 22:50:20 -0000	1.4
+++ sc/source/ui/unoobj/srchuno.cxx	8 Feb 2006 09:53:45 -0000
@@ -43,6 +43,7 @@
 #include <svx/srchitem.hxx>
 #include <osl/mutex.hxx>
 #include <rtl/uuid.h>
+#include <comphelper/types.hxx>
 
 #include "srchuno.hxx"
 #include "docsh.hxx"
@@ -181,6 +182,8 @@ void SAL_CALL ScCellSearchObj::setProper
 	else if (aString.EqualsAscii( SC_UNO_SRCHSIMEX ))  pSearchItem->SetLEVOther( ScUnoHelpFunctions::GetInt16FromAny( aValue ) );
 	else if (aString.EqualsAscii( SC_UNO_SRCHSIMREM )) pSearchItem->SetLEVShorter( ScUnoHelpFunctions::GetInt16FromAny( aValue ) );
 	else if (aString.EqualsAscii( SC_UNO_SRCHTYPE ))   pSearchItem->SetCellType( ScUnoHelpFunctions::GetInt16FromAny( aValue ) );
+	else if (aString.EqualsAscii( SC_UNO_SRCHSEARCH ))   setSearchString( ::comphelper::getString( aValue ) ) ;
+	else if (aString.EqualsAscii( SC_UNO_SRCHREPLACE ))  setReplaceString( ::comphelper::getString( aValue ) ) ;
 }
 
 uno::Any SAL_CALL ScCellSearchObj::getPropertyValue( const rtl::OUString& aPropertyName )
@@ -203,6 +206,8 @@ uno::Any SAL_CALL ScCellSearchObj::getPr
 	else if (aString.EqualsAscii( SC_UNO_SRCHSIMEX ))  aRet <<= (sal_Int16) pSearchItem->GetLEVOther();
 	else if (aString.EqualsAscii( SC_UNO_SRCHSIMREM )) aRet <<= (sal_Int16) pSearchItem->GetLEVShorter();
 	else if (aString.EqualsAscii( SC_UNO_SRCHTYPE ))   aRet <<= (sal_Int16) pSearchItem->GetCellType();
+	else if (aString.EqualsAscii( SC_UNO_SRCHSEARCH )) aRet <<= getSearchString();
+	else if (aString.EqualsAscii( SC_UNO_SRCHREPLACE )) aRet <<= getReplaceString();
 
 	return aRet;
 }
