diff -Naur -x .svn null/file_accessories.scp scp2/source/accessories/file_accessories.scp
--- null/file_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/file_accessories.scp	2007-12-24 09:30:07.000000000 +0100
@@ -0,0 +1,54 @@
+/*************************************************************************
+*
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: file_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/07/02 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+************************************************************************/
+
+#include "macros.inc"
+
+/* *** KAMI Extras *** */
+
+/* ** Macro - Premium ** */
+
+File gid_File_Basic_Premium
+    Dir = gid_Dir_Basic_Premium;
+    ARCHIVE_TXT_FILE_BODY_PATCH;
+    Name = "basicsrvpremium.zip";
+End
+
+/* ** Macro - OOoWikipedia ** */
+
+File gid_File_Basic_OOoWikipedia
+    Dir = gid_Dir_Basic_OOoWikipedia;
+    ARCHIVE_TXT_FILE_BODY_PATCH;
+    Name = "basicsrvooowikipedia.zip";
+End
diff -Naur -x .svn null/file_font_accessories.scp scp2/source/accessories/file_font_accessories.scp
--- null/file_font_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/file_font_accessories.scp	2007-12-24 08:47:44.000000000 +0100
@@ -0,0 +1,1790 @@
+/*************************************************************************
+*
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: file_font_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+************************************************************************/
+
+#include "macros.inc"
+
+/* Airmole font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_airmole, airmole.ttf, Airmole ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_airmole, airmole.ttf, Airmole ) 
+#endif 
+#endif 
+
+/* Airmole Antique font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_airmolea, airmolea.ttf, Airmole Antique ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_airmolea, airmolea.ttf, Airmole Antique ) 
+#endif 
+#endif 
+
+/* Airmole Shaded font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_airmoles, airmoles.ttf, Airmole Shaded ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_airmoles, airmoles.ttf, Airmole Shaded ) 
+#endif 
+#endif 
+
+/* Airmole Stripe font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_airmoleq, airmoleq.ttf, Airmole Stripe ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_airmoleq, airmoleq.ttf, Airmole Stripe ) 
+#endif 
+#endif 
+
+/* Astron Boy font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_astronbo, astronbo.ttf, Astron Boy ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_astronbo, astronbo.ttf, Astron Boy ) 
+#endif 
+#endif 
+
+/* Astron Boy Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_astronbi, astronbi.ttf, Astron Boy Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_astronbi, astronbi.ttf, Astron Boy Italic ) 
+#endif 
+#endif 
+
+/* Astron Boy Video font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_astronbv, astronbv.ttf, Astron Boy Video ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_astronbv, astronbv.ttf, Astron Boy Video ) 
+#endif 
+#endif 
+
+/* Astron Boy Wonder font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_astronbw, astronbw.ttf, Astron Boy Wonder ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_astronbw, astronbw.ttf, Astron Boy Wonder ) 
+#endif 
+#endif 
+
+/* Baveuse font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_baveuse, baveuse.ttf, Baveuse ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_baveuse, baveuse.ttf, Baveuse ) 
+#endif 
+#endif 
+
+/* Berylium font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Berylium, Berylium.ttf, Berylium ) 
+#endif 
+#endif 
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Berylium, Berylium.ttf, Berylium ) 
+#endif 
+#endif 
+
+/* Berylium Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Beryliub, Beryliub.ttf, Berylium Bold )
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Beryliub, Beryliub.ttf, Berylium Bold ) 
+#endif 
+#endif 
+
+/* Berylium Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Berylibi, Berylibi.ttf, Berylium Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Berylibi, Berylibi.ttf, Berylium Bold Italic ) 
+#endif 
+#endif 
+
+/* Berylium Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Beryliui, Beryliui.ttf, Berylium Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Beryliui, Beryliui.ttf, Berylium Italic ) 
+#endif 
+#endif 
+
+/* Betsy Flanagan 2 font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_BETSY2, BETSY2.ttf, Betsy Flanagan 2 ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_BETSY2, BETSY2.ttf, Betsy Flanagan 2 ) 
+#endif 
+#endif 
+
+/* Blue Highway font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bluehigh, bluehigh.ttf, Blue Highway ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bluehigh, bluehigh.ttf, Blue Highway ) 
+#endif 
+#endif 
+
+/* Blue Highway Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bluebold, bluebold.ttf, Blue Highway Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bluebold, bluebold.ttf, Blue Highway Bold ) 
+#endif 
+#endif 
+
+/* Blue Highway Condensed font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bluecond, bluecond.ttf, Blue Highway Condensed ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bluecond, bluecond.ttf, Blue Highway Condensed ) 
+#endif 
+#endif 
+
+/* Blue Highway Linocut font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bluehigl, bluehigl.ttf, Blue Highway Linocut ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bluehigl, bluehigl.ttf, Blue Highway Linocut ) 
+#endif 
+#endif 
+
+/* Bullpen font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bullpen_, bullpen_.ttf, Bullpen ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bullpen_, bullpen_.ttf, Bullpen ) 
+#endif 
+#endif 
+
+/* Bullpen 3D font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bullpen3, bullpen3.ttf, Bullpen 3D ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bullpen3, bullpen3.ttf, Bullpen 3D ) 
+#endif 
+#endif 
+
+/* Bullpen Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_bullpeni, bullpeni.ttf, Bullpen Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_bullpeni, bullpeni.ttf, Bullpen Italic ) 
+#endif 
+#endif 
+
+/* Colourbars font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_colourba, colourba.ttf, Colourbars ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_colourba, colourba.ttf, Colourbars ) 
+#endif 
+#endif 
+
+/* Colourbars Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_colourbb, colourbb.ttf, Colourbars Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_colourbb, colourbb.ttf, Colourbars Bold ) 
+#endif 
+#endif 
+
+/* Counterscraps font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_counters, counters.ttf, Counterscraps ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_counters, counters.ttf, Counterscraps ) 
+#endif 
+#endif 
+
+/* Credit Valley font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_creditva, creditva.ttf, Credit Valley ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_creditva, creditva.ttf, Credit Valley ) 
+#endif 
+#endif 
+
+/* Credit Valley Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_creditvb, creditvb.ttf, Credit Valley Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_creditvb, creditvb.ttf, Credit Valley Bold ) 
+#endif 
+#endif 
+
+/* Credit Valley Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_creditvz, creditvz.ttf, Credit Valley Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_creditvz, creditvz.ttf, Credit Valley Bold Italic ) 
+#endif 
+#endif 
+
+/* Credit Valley Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_creditvi, creditvi.ttf, Credit Valley Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_creditvi, creditvi.ttf, Credit Valley Italic ) 
+#endif
+#endif 
+
+/* Edmunds font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_edmunds, edmunds.ttf, Edmunds ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_edmunds, edmunds.ttf, Edmunds ) 
+#endif 
+#endif 
+
+/* Edmunds Distressed font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_edmundis, edmundis.ttf, Edmunds Distressed ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_edmundis, edmundis.ttf, Edmunds Distressed ) 
+#endif 
+#endif 
+
+/* Engebrechtre font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeregu, engeregu.ttf, Engebrechtre ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeregu, engeregu.ttf, Engebrechtre ) 
+#endif 
+#endif 
+
+/* Engebrechtre Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engebold, engebold.ttf, Engebrechtre Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engebold, engebold.ttf, Engebrechtre Bold ) 
+#endif 
+#endif 
+
+/* Engebrechtre Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeboit, engeboit.ttf, Engebrechtre Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeboit, engeboit.ttf, Engebrechtre Bold Italic ) 
+#endif 
+#endif 
+
+/* Engebrechtre Expanded font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeexpa, engeexpa.ttf, Engebrechtre Expanded ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeexpa, engeexpa.ttf, Engebrechtre Expanded ) 
+#endif 
+#endif 
+
+/* Engebrechtre Expanded Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeexbo, engeexbo.ttf, Engebrechtre Expanded Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeexbo, engeexbo.ttf, Engebrechtre Expanded Bold ) 
+#endif 
+#endif 
+
+/* Engebrechtre Expanded Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeexbi, engeexbi.ttf, Engebrechtre Expanded Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeexbi, engeexbi.ttf, Engebrechtre Expanded Bold Italic ) 
+#endif 
+#endif 
+
+/* Engebrechtre Expanded Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeexit, engeexit.ttf, Engebrechtre Expanded Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeexit, engeexit.ttf, Engebrechtre Expanded Italic ) 
+#endif 
+#endif 
+
+/* Engebrechtre Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_engeital, engeital.ttf, Engebrechtre Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_engeital, engeital.ttf, Engebrechtre Italic ) 
+#endif 
+#endif 
+
+/* Goodfish font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_goodfish, goodfish.ttf, Goodfish ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_goodfish, goodfish.ttf, Goodfish ) 
+#endif 
+#endif 
+
+/* Goodfish Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_goodfisb, goodfisb.ttf, Goodfish Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_goodfisb, goodfisb.ttf, Goodfish Bold ) 
+#endif 
+#endif 
+
+/* Goodfish Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_goodfisc, goodfisc.ttf, Goodfish Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_goodfisc, goodfisc.ttf, Goodfish Bold Italic ) 
+#endif 
+#endif 
+
+/* Goodfish Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_goodfisi, goodfisi.ttf, Goodfish Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_goodfisi, goodfisi.ttf, Goodfish Italic ) 
+#endif 
+#endif 
+
+/* Guanine font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_guanine_, guanine_.ttf, Guanine ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_guanine_, guanine_.ttf, Guanine ) 
+#endif 
+#endif 
+
+/* Gunplay font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_gunplay, gunplay.ttf, Gunplay ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_gunplay, gunplay.ttf, Gunplay ) 
+#endif 
+#endif 
+
+/* Kirsty font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_kirsty__, kirsty__.ttf, Kirsty ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_kirsty__, kirsty__.ttf, Kirsty ) 
+#endif 
+#endif 
+
+/* Kirsty Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_kirsty_b, kirsty_b.ttf, Kirsty Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_kirsty_b, kirsty_b.ttf, Kirsty Bold ) 
+#endif 
+#endif 
+
+/* Kirsty Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_kirstybi, kirstybi.ttf, Kirsty Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_kirstybi, kirstybi.ttf, Kirsty Bold Italic ) 
+#endif 
+#endif 
+
+/* Kirsty Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_kirsty_i, kirsty_i.ttf, Kirsty Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_kirsty_i, kirsty_i.ttf, Kirsty Italic ) 
+#endif 
+#endif 
+
+/* Kirsty Ink font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_kirstyin, kirstyin.ttf, Kirsty Ink ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_kirstyin, kirstyin.ttf, Kirsty Ink ) 
+#endif 
+#endif 
+
+/* Mufferaw font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_mufferaw, mufferaw.ttf, Mufferaw ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_mufferaw, mufferaw.ttf, Mufferaw ) 
+#endif 
+#endif 
+
+/* Nasalization font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_nasaliza, nasaliza.ttf, Nasalization ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_nasaliza, nasaliza.ttf, Nasalization ) 
+#endif 
+#endif 
+
+/* Neuropol font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_NEUROPOL, NEUROPOL.ttf, Neuropol ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_NEUROPOL, NEUROPOL.ttf, Neuropol ) 
+#endif 
+#endif 
+
+/* Numberpile font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_numberpi, numberpi.ttf, Numberpile ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_numberpi, numberpi.ttf, Numberpile ) 
+#endif 
+#endif 
+
+/* Pricedown font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_pricedow, pricedow.ttf, Pricedown ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_pricedow, pricedow.ttf, Pricedown ) 
+#endif 
+#endif 
+
+/* Pupcat font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_pupcat__, pupcat__.ttf, Pupcat ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_pupcat__, pupcat__.ttf, Pupcat ) 
+#endif 
+#endif 
+
+/* Rina font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_rina, rina.ttf, Rina ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_rina, rina.ttf, Rina ) 
+#endif 
+#endif 
+
+/* Sandoval font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_sandoval, sandoval.ttf, Sandoval ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_sandoval, sandoval.ttf, Sandoval )
+#endif 
+#endif 
+
+/* Sappy Mugs font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_SAPPM___, SAPPM___.ttf, Sappy Mugs ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_SAPPM___, SAPPM___.ttf, Sappy Mugs ) 
+#endif 
+#endif 
+
+/* Screengem font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_screenge, screenge.ttf, Screengem ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_screenge, screenge.ttf, Screengem ) 
+#endif 
+#endif 
+
+/* Steelfish font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_steelfis, steelfis.ttf, Steelfish ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_steelfis, steelfis.ttf, Steelfish ) 
+#endif 
+#endif 
+
+/* Steelfish Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_steelfib, steelfib.ttf, Steelfish Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_steelfib, steelfib.ttf, Steelfish Bold ) 
+#endif 
+#endif 
+
+/* Steelfish Outline font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_steelout, steelout.ttf, Steelfish Outline ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_steelout, steelout.ttf, Steelfish Outline ) 
+#endif 
+#endif 
+
+/* Strenuous font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_strenuou, strenuou.ttf, Strenuous ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_strenuou, strenuou.ttf, Strenuous ) 
+#endif 
+#endif 
+
+/* Strenuous 3D font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_strenu3d, strenu3d.ttf, Strenuous 3D ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_strenu3d, strenu3d.ttf, Strenuous 3D ) 
+#endif 
+#endif 
+
+/* Subpear font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_subpear_, subpear_.ttf, Subpear ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_subpear_, subpear_.ttf, Subpear ) 
+#endif 
+#endif 
+
+/* Teen font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_teen____, teen____.ttf, Teen ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_teen____, teen____.ttf, Teen ) 
+#endif 
+#endif 
+
+/* Teen Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_teenbold, teenbold.ttf, Teen Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_teenbold, teenbold.ttf, Teen Bold ) 
+#endif 
+#endif 
+
+/* Teen Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_teenbdit, teenbdit.ttf, Teen Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_teenbdit, teenbdit.ttf, Teen Bold Italic ) 
+#endif 
+#endif 
+
+/* Teen Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_teenital, teenital.ttf, Teen Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_teenital, teenital.ttf, Teen Italic ) 
+#endif 
+#endif 
+
+/* Teen Light font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_teenlite, teenlite.ttf, Teen Light ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_teenlite, teenlite.ttf, Teen Light ) 
+#endif 
+#endif 
+
+/* Teen Light Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_teenlita, teenlita.ttf, Teen Light Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_teenlita, teenlita.ttf, Teen Light Italic ) 
+#endif 
+#endif 
+
+/* Typodermic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_typoderm, typoderm.ttf, Typodermic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_typoderm, typoderm.ttf, Typodermic ) 
+#endif 
+#endif 
+
+/* Vahika font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vahika_, vahika_.ttf, Vahika ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vahika_, vahika_.ttf, Vahika ) 
+#endif 
+#endif 
+
+/* Vahika Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vahikab, vahikab.ttf, Vahika Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vahikab, vahikab.ttf, Vahika Bold ) 
+#endif 
+#endif 
+
+/* Vahika Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vahikac, vahikac.ttf, Vahika Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vahikac, vahikac.ttf, Vahika Bold Italic ) 
+#endif 
+#endif 
+
+/* Vahika Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vahikai, vahikai.ttf, Vahika Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vahikai, vahikai.ttf, Vahika Italic ) 
+#endif 
+#endif 
+
+/* Vectroid font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vectroid, vectroid.ttf, Vectroid ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vectroid, vectroid.ttf, Vectroid ) 
+#endif 
+#endif 
+
+/* Velvenda Cooler font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_velvendc, velvendc.ttf, Velvenda Cooler ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_velvendc, velvendc.ttf, Velvenda Cooler ) 
+#endif 
+#endif 
+
+/* Velvenda Megablack font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_velvenda, velvenda.ttf, Velvenda Megablack ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_velvenda, velvenda.ttf, Velvenda Megablack ) 
+#endif 
+#endif 
+
+/* Vibrocentric font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vibrocen, vibrocen.ttf, Vibrocentric ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vibrocen, vibrocen.ttf, Vibrocentric ) 
+#endif 
+#endif 
+
+/* Vibrocentric Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vibroceb, vibroceb.ttf, Vibrocentric Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vibroceb, vibroceb.ttf, Vibrocentric Bold ) 
+#endif 
+#endif 
+
+/* Vibrocentric Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vibrocex, vibrocex.ttf, Vibrocentric Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vibrocex, vibrocex.ttf, Vibrocentric Bold Italic ) 
+#endif 
+#endif 
+
+/* Vibrocentric Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_vibrocei, vibrocei.ttf, Vibrocentric Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_vibrocei, vibrocei.ttf, Vibrocentric Italic ) 
+#endif 
+#endif 
+
+/* Wintermute font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_wintermu, wintermu.ttf, Wintermute ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_wintermu, wintermu.ttf, Wintermute ) 
+#endif 
+#endif 
+
+/* You're Gone font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_youregon, youregon.ttf, You are Gone )
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_youregon, youregon.ttf, You are Gone )
+#endif 
+#endif 
+
+/* You're Gone Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_youregoi, youregoi.ttf, You are Gone Italic )
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_youregoi, youregoi.ttf, You are Gone Italic ) 
+#endif 
+#endif 
+
+/* Zekton font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_zekton__, zekton__.ttf, Zekton ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_zekton__, zekton__.ttf, Zekton ) 
+#endif 
+#endif 
+
+/* Zekton Bold font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_zektonbo, zektonbo.ttf, Zekton Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_zektonbo, zektonbo.ttf, Zekton Bold ) 
+#endif 
+#endif 
+
+/* Zekton Bold Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_zektonbi, zektonbi.ttf, Zekton Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_zektonbi, zektonbi.ttf, Zekton Bold Italic )
+#endif 
+#endif 
+
+/* Zekton Italic font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_zektonit, zektonit.ttf, Zekton Italic )
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_zektonit, zektonit.ttf, Zekton Italic ) 
+#endif 
+#endif 
+
+/* Zorque font */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_zorque, zorque.ttf, Zorque ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_zorque, zorque.ttf, Zorque )
+#endif 
+#endif
+
+/* Additional fonts for 2.0.3 */
+
+/* -URW-Gothic L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Gothic_L_Avantgarde, a010013l.ttf, -URW-Gothic L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Gothic_L_Avantgarde, a010013l.ttf, -URW-Gothic L ) 
+#endif 
+#endif 
+ 
+/* -URW-Gothic L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Gothic_L_Bold_Avantgarde, a010015l.ttf, -URW-Gothic L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Gothic_L_Bold_Avantgarde, a010015l.ttf, -URW-Gothic L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Gothic L Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Gothic_L_Italic_Avantgarde, a010033l.ttf, -URW-Gothic L Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Gothic_L_Italic_Avantgarde, a010033l.ttf, -URW-Gothic L Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Gothic L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Gothic_L_Bold_Italic_Avantgarde, a010035l.ttf, -URW-Gothic L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Gothic_L_Bold_Italic_Avantgarde, a010035l.ttf, -URW-Gothic L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Bookman L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Bookman_L_Bookman, b018012l.ttf, -URW-Bookman L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Bookman_L_Bookman, b018012l.ttf, -URW-Bookman L ) 
+#endif 
+#endif 
+ 
+/* -URW-Bookman L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Bookman_L_Bold_Bookman, b018015l.ttf, -URW-Bookman L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Bookman_L_Bold_Bookman, b018015l.ttf, -URW-Bookman L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Bookman L Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Bookman_L_Italic_Bookman, b018032l.ttf, -URW-Bookman L Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Bookman_L_Italic_Bookman, b018032l.ttf, -URW-Bookman L Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Bookman L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Bookman_L_Bold_Italic_Bookman, b018035l.ttf, -URW-Bookman L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Bookman_L_Bold_Italic_Bookman, b018035l.ttf, -URW-Bookman L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Century Schoolbook L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_New_Century_SchoolBook, c059013l.ttf, -URW-Century Schoolbook L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_New_Century_SchoolBook, c059013l.ttf, -URW-Century Schoolbook L ) 
+#endif 
+#endif 
+ 
+/* -URW-Century Schoolbook L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_Bold_New_Century_SchoolBook, c059016l.ttf, -URW-Century Schoolbook L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_Bold_New_Century_SchoolBook, c059016l.ttf, -URW-Century Schoolbook L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Century Schoolbook L Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_Italic_New_Century_SchoolBook, c059033l.ttf, -URW-Century Schoolbook L Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_Italic_New_Century_SchoolBook, c059033l.ttf, -URW-Century Schoolbook L Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Century Schoolbook L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_Bold_Italic_New_Century_SchoolBook, c059036l.ttf, -URW-Century Schoolbook L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Century_Schoolbook_L_Bold_Italic_New_Century_SchoolBook, c059036l.ttf, -URW-Century Schoolbook L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Helvetica, n019003l.ttf, -URW-Nimbus Sans L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Helvetica, n019003l.ttf, -URW-Nimbus Sans L ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Helvetica, n019004l.ttf, -URW-Nimbus Sans L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Helvetica, n019004l.ttf, -URW-Nimbus Sans L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Italic_Helvetica, n019023l.ttf, -URW-Nimbus Sans L Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Italic_Helvetica, n019023l.ttf, -URW-Nimbus Sans L Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Italic_Helvetica, n019024l.ttf, -URW-Nimbus Sans L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Italic_Helvetica, n019024l.ttf, -URW-Nimbus Sans L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Condensed */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Helvetica_Condensed, n019043l.ttf, -URW-Nimbus Sans L Condensed ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Helvetica_Condensed, n019043l.ttf, -URW-Nimbus Sans L Condensed ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Bold Condensed */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Helvetica_Condensed, n019044l.ttf, -URW-Nimbus Sans L Bold Condensed ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Helvetica_Condensed, n019044l.ttf, -URW-Nimbus Sans L Bold Condensed ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Italic Condensed */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Italic_Helvetica_Condensed, n019063l.ttf, -URW-Nimbus Sans L Italic Condensed ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Italic_Helvetica_Condensed, n019063l.ttf, -URW-Nimbus Sans L Italic Condensed ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Sans L Bold Italic Condensed */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Italic_Helvetica_Condensed, n019064l.ttf, -URW-Nimbus Sans L Bold Italic Condensed ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Italic_Helvetica_Condensed, n019064l.ttf, -URW-Nimbus Sans L Bold Italic Condensed ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Roman No9 L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Times, n021003l.ttf, -URW-Nimbus Roman No9 L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Times, n021003l.ttf, -URW-Nimbus Roman No9 L ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Roman No9 L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Bold_Times, n021004l.ttf, -URW-Nimbus Roman No9 L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Bold_Times, n021004l.ttf, -URW-Nimbus Roman No9 L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Roman No9 L Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Italic_Times, n021023l.ttf, -URW-Nimbus Roman No9 L Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Italic_Times, n021023l.ttf, -URW-Nimbus Roman No9 L Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Roman No9 L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Bold_Italic_Times, n021024l.ttf, -URW-Nimbus Roman No9 L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Bold_Italic_Times, n021024l.ttf, -URW-Nimbus Roman No9 L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Mono L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Mono_L_Courier, n022003l.ttf, -URW-Nimbus Mono L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Mono_L_Courier, n022003l.ttf, -URW-Nimbus Mono L ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Mono L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Mono_L_Bold_Courier, n022004l.ttf, -URW-Nimbus Mono L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Mono_L_Bold_Courier, n022004l.ttf, -URW-Nimbus Mono L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Mono LK Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Mono_LK_Italic_Courier, n022023l.ttf, -URW-Nimbus Mono LK Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Mono_LK_Italic_Courier, n022023l.ttf, -URW-Nimbus Mono LK Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Nimbus Mono L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Nimbus_Mono_L_Bold_Italic_Courier, n022024l.ttf, -URW-Nimbus Mono L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Nimbus_Mono_L_Bold_Italic_Courier, n022024l.ttf, -URW-Nimbus Mono L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Palladio L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Palladio_L_Palitino, p052003l.ttf, -URW-Palladio L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Palladio_L_Palitino, p052003l.ttf, -URW-Palladio L ) 
+#endif
+#endif 
+ 
+/* -URW-Palladio L Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Palladio_L_Bold_Palitino, p052004l.ttf, -URW-Palladio L Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Palladio_L_Bold_Palitino, p052004l.ttf, -URW-Palladio L Bold ) 
+#endif 
+#endif 
+ 
+/* -URW-Palladio L Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Palladio_L_Italic_Palitino, p052023l.ttf, -URW-Palladio L Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Palladio_L_Italic_Palitino, p052023l.ttf, -URW-Palladio L Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Palladio L Bold Italic */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Palladio_L_Bold_Italic_Palitino, p052024l.ttf, -URW-Palladio L Bold Italic ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Palladio_L_Bold_Italic_Palitino, p052024l.ttf, -URW-Palladio L Bold Italic ) 
+#endif 
+#endif 
+ 
+/* -URW-Standard Symbols L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Standard_Symbols_L_Symbol, s050000l.ttf, -URW-Standard Symbols L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Standard_Symbols_L_Symbol, s050000l.ttf, -URW-Standard Symbols L ) 
+#endif 
+#endif 
+ 
+/* -URW-Chancery L */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_-URW-Chancery_L_Zapf_Chancery, z003034l.ttf, -URW-Chancery L ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_-URW-Chancery_L_Zapf_Chancery, z003034l.ttf, -URW-Chancery L ) 
+#endif 
+#endif 
+ 
+/* Dingbats */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Dingbats, d050000l.ttf, Dingbats ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Dingbats, d050000l.ttf, Dingbats ) 
+#endif 
+#endif 
+ 
+/* Luxi Mono Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Mono_Bold, luximb.ttf, Luxi Mono Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Mono_Bold, luximb.ttf, Luxi Mono Bold ) 
+#endif 
+#endif 
+ 
+/* Luxi Mono Bold Oblique */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Mono_Bold_Oblique, luximbi.ttf, Luxi Mono Bold Oblique ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Mono_Bold_Oblique, luximbi.ttf, Luxi Mono Bold Oblique ) 
+#endif 
+#endif 
+ 
+/* Luxi Mono Regular */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Mono_Regular, luximr.ttf, Luxi Mono Regular ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Mono_Regular, luximr.ttf, Luxi Mono Regular ) 
+#endif 
+#endif 
+ 
+/* Luxi Mono Oblique */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Mono_Oblique, luximri.ttf, Luxi Mono Oblique ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Mono_Oblique, luximri.ttf, Luxi Mono Oblique ) 
+#endif 
+#endif 
+ 
+/* Luxi Serif Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Serif_Bold, luxirb.ttf, Luxi Serif Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Serif_Bold, luxirb.ttf, Luxi Serif Bold ) 
+#endif 
+#endif 
+ 
+/* Luxi Serif Bold Oblique */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Serif_Bold_Oblique, luxirbi.ttf, Luxi Serif Bold Oblique ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Serif_Bold_Oblique, luxirbi.ttf, Luxi Serif Bold Oblique ) 
+#endif 
+#endif 
+ 
+/* Luxi Serif Regular */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Serif_Regular, luxirr.ttf, Luxi Serif Regular ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Serif_Regular, luxirr.ttf, Luxi Serif Regular ) 
+#endif 
+#endif 
+ 
+/* Luxi Serif Oblique */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Serif_Oblique, luxirri.ttf, Luxi Serif Oblique ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Serif_Oblique, luxirri.ttf, Luxi Serif Oblique ) 
+#endif 
+#endif 
+ 
+/* Luxi Sans Bold */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Sans_Bold, luxisb.ttf, Luxi Sans Bold ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Sans_Bold, luxisb.ttf, Luxi Sans Bold ) 
+#endif 
+#endif 
+ 
+/* Luxi Sans Bold Oblique */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Sans_Bold_Oblique, luxisbi.ttf, Luxi Sans Bold Oblique ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Sans_Bold_Oblique, luxisbi.ttf, Luxi Sans Bold Oblique ) 
+#endif 
+#endif 
+ 
+/* Luxi Sans Regular */ 
+#ifdef UNX 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Sans_Regular, luxisr.ttf, Luxi Sans Regular ) 
+#endif 
+#endif 
+#ifdef WNT 
+#ifndef WITHOUT_EXTRA_FONT 
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Sans_Regular, luxisr.ttf, Luxi Sans Regular ) 
+#endif 
+#endif 
+ 
+/* Luxi Sans Oblique */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_Luxi_Sans_Oblique, luxisri.ttf, Luxi Sans Oblique )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_Luxi_Sans_Oblique, luxisri.ttf, Luxi Sans Oblique )
+#endif
+#endif
+
+/* Gentium */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_Gentium, GenR102.TTF, Gentium )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_Gentium, GenR102.TTF, Gentium )
+#endif
+#endif
+
+/* Gentium Italic */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_Gentium_Italic, GenI102.TTF, Gentium Italic )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_Gentium_Italic, GenI102.TTF, Gentium Italic )
+#endif
+#endif
+
+/* GentiumAlt */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_GentiumAlt, GenAR102.TTF, GentiumAlt )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_GentiumAlt, GenAR102.TTF, GentiumAlt )
+#endif
+#endif
+
+/* GentiumAlt Italic */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_GentiumAlt_Italic, GenAI102.TTF, GentiumAlt Italic )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_GentiumAlt_Italic, GenAI102.TTF, GentiumAlt Italic )
+#endif
+#endif
+
+/* MgOpen Canonica Bold */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_MgOpen_Canonica_Bold, MgOpenCanonicaBold.ttf, MgOpen Canonica Bold )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_MgOpen_Canonica_Bold, MgOpenCanonicaBold.ttf, MgOpen Canonica Bold )
+#endif
+#endif
+
+/* MgOpen Canonica Bold Italic */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_MgOpen_Canonica_Bold_Italic, MgOpenCanonicaBoldItalic.ttf, MgOpen Canonica Bold Italic)
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_MgOpen_Canonica_Bold_Italic, MgOpenCanonicaBoldItalic.ttf, MgOpen Canonica Bold Italic)
+#endif
+#endif
+
+/* MgOpen Canonica Italic */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_MgOpen_Canonica_Italic, MgOpenCanonicaItalic.ttf, MgOpen Canonica Italic )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_MgOpen_Canonica_Italic, MgOpenCanonicaItalic.ttf, MgOpen Canonica Italic )
+#endif
+#endif
+
+/* MgOpen Canonica Regular */
+#ifdef UNX
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTUNX_FILE( gid_File_Fnt_MgOpen_Canonica_Regular, MgOpenCanonicaRegular.ttf, MgOpen Canonica )
+#endif
+#endif
+#ifdef WNT
+#ifndef WITHOUT_EXTRA_FONT
+STD_FONTWIN_FILE( gid_File_Fnt_MgOpen_Canonica_Regular, MgOpenCanonicaRegular.ttf, MgOpen Canonica )
+#endif
+#endif
diff -Naur -x .svn null/file_gallery_accessories.scp scp2/source/accessories/file_gallery_accessories.scp
--- null/file_gallery_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/file_gallery_accessories.scp	2007-12-24 09:30:28.000000000 +0100
@@ -0,0 +1,146 @@
+/*************************************************************************
+*
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: file_gallery_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+************************************************************************/
+
+#include "macros.inc"
+
+/* *** KAMI Extras *** */
+
+/* ** Gallery ** */
+
+#ifndef WITHOUT_EXTRA_GALLERY
+
+File gid_File_Extra_Gallery_KAMI
+    Dir = gid_Dir_Gallery;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigallery.zip";
+End
+
+File gid_File_Extra_Gallery_Draws_People_KAMI
+    Dir = gid_Dir_Gallery_Draws_People;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigallerydrawspeople.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Plants_KAMI
+    Dir = gid_Dir_Gallery_Photos_Plants;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosplants.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Celebration_KAMI
+    Dir = gid_Dir_Gallery_Photos_Celebration;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotoscelebration.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Fauna_KAMI
+    Dir = gid_Dir_Gallery_Photos_Fauna;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosfauna.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Flowers_KAMI
+    Dir = gid_Dir_Gallery_Photos_Flowers;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosflowers.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_FoodsandDrinks_KAMI
+    Dir = gid_Dir_Gallery_Photos_FoodsandDrinks;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosfoodsanddrinks.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Humans_KAMI
+    Dir = gid_Dir_Gallery_Photos_Humans;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotoshumans.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Landscapes_KAMI
+    Dir = gid_Dir_Gallery_Photos_Landscapes;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotoslandscapes.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Buildings_KAMI
+    Dir = gid_Dir_Gallery_Photos_Buildings;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosbuildings.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Objects_KAMI
+    Dir = gid_Dir_Gallery_Photos_Objects;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosobjects.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Cities_KAMI
+    Dir = gid_Dir_Gallery_Photos_Cities;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotoscities.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Space_KAMI
+    Dir = gid_Dir_Gallery_Photos_Space;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosspace.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Statues_KAMI
+    Dir = gid_Dir_Gallery_Photos_Statues;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotosstatues.zip";
+End
+
+File gid_File_Extra_Gallery_Photos_Travel_KAMI
+    Dir = gid_Dir_Gallery_Photos_Travel;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryphotostravel.zip";
+End
+
+File gid_File_Extra_Gallery_Elements_Bullets2_KAMI
+    Dir = gid_Dir_Gallery_Elements_Bullets2;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigalleryelementsbullets2.zip";
+End
+
+File gid_File_Extra_Gallery_NONFREE_KAMI
+    Dir = gid_Dir_Gallery;
+    ARCHIVE_TXT_FILE_BODY;
+    Name = "kamigallerynonfree.zip";
+End
+
+#endif
diff -Naur -x .svn null/file_samples_accessories.scp scp2/source/accessories/file_samples_accessories.scp
--- null/file_samples_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/file_samples_accessories.scp	2007-12-24 09:30:43.000000000 +0100
@@ -0,0 +1,88 @@
+/*************************************************************************
+*
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: file_samples_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+************************************************************************/
+
+#include "macros.inc"
+
+/* *** KAMI Extras *** */
+
+/* ** Samples ** */
+
+#ifndef WITHOUT_EXTRA_SAMPLE
+
+File gid_File_Extra_Samples_KAMI
+    Dir = gid_Dir_Samples_Language;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamisamples);
+End
+
+File gid_File_Extra_Samples_Documentations_KAMI
+    Dir = gid_Dir_Samples_Documentations;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamisamplesdocumentation);
+End
+
+File gid_File_Extra_Samples_Advertisement_KAMI
+    Dir = gid_Dir_Samples_Advertisement;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamisamplesadvertisement);
+End
+
+/* ** Samples - NON FREE** */
+
+File gid_File_Extra_Samples_NONFREE_KAMI
+    Dir = gid_Dir_Samples_Language;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamisamplesnonfree);
+End
+
+File gid_File_Extra_Samples_NONFREE_Documentations_KAMI
+    Dir = gid_Dir_Samples_Documentations;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamisamplesnonfreedocumentation);
+End
+
+File gid_File_Extra_Samples_NONFREE_Advertisement_KAMI
+    Dir = gid_Dir_Samples_Advertisement;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamisamplesnonfreeadvertisement);
+End
+
+#endif
diff -Naur -x .svn null/file_templates_accessories.scp scp2/source/accessories/file_templates_accessories.scp
--- null/file_templates_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/file_templates_accessories.scp	2007-12-24 09:31:01.000000000 +0100
@@ -0,0 +1,186 @@
+/*************************************************************************
+*
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: file_templates_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+************************************************************************/
+
+#include "macros.inc"
+
+/* *** KAMI Extras *** */
+
+/* ** Templates ** */
+
+#ifndef WITHOUT_EXTRA_TEMPLATE
+
+File gid_File_Extra_Templates_KAMI
+    Dir = gid_Dir_Template_Language;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplates);
+End
+
+File gid_File_Extra_Templates_Layout_KAMI
+    Dir = gid_Dir_Template_Layout;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplateslayout);
+End
+
+File gid_File_Extra_Templates_Present_KAMI
+    Dir = gid_Dir_Template_Presnt;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatespresent);
+End
+
+File gid_File_Extra_Templates_Educate_KAMI
+    Dir = gid_Dir_Template_Educate;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplateseducate);
+End
+
+File gid_File_Extra_Templates_Finance_KAMI
+    Dir = gid_Dir_Template_Finance;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesfinance);
+End
+
+File gid_File_Extra_Templates_Forms_KAMI
+    Dir = gid_Dir_Template_Forms;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesforms);
+End
+
+File gid_File_Extra_Templates_Misc_KAMI
+    Dir = gid_Dir_Template_Misc;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesmisc);
+End
+
+File gid_File_Extra_Templates_Officorr_KAMI
+    Dir = gid_Dir_Template_Officorr;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesofficorr);
+End
+
+File gid_File_Extra_Templates_Offimisc_KAMI
+    Dir = gid_Dir_Template_Offimisc;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesoffimisc);
+End
+
+File gid_File_Extra_Templates_Personal_KAMI
+    Dir = gid_Dir_Template_Personal;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatespersonal);
+End
+
+/* ** Templates - NON FREE ** */
+
+File gid_File_Extra_Templates_NONFREE_KAMI
+    Dir = gid_Dir_Template_Language;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfree);
+End
+
+File gid_File_Extra_Templates_NONFREE_Layout_KAMI
+    Dir = gid_Dir_Template_Layout;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreelayout);
+End
+
+File gid_File_Extra_Templates_NONFREE_Present_KAMI
+    Dir = gid_Dir_Template_Presnt;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreepresent);
+End
+
+File gid_File_Extra_Templates_NONFREE_Educate_KAMI
+    Dir = gid_Dir_Template_Educate;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreeeducate);
+End
+
+File gid_File_Extra_Templates_NONFREE_Finance_KAMI
+    Dir = gid_Dir_Template_Finance;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreefinance);
+End
+
+File gid_File_Extra_Templates_NONFREE_Forms_KAMI
+    Dir = gid_Dir_Template_Forms;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreeforms);
+End
+
+File gid_File_Extra_Templates_NONFREE_Misc_KAMI
+    Dir = gid_Dir_Template_Misc;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreemisc);
+End
+
+File gid_File_Extra_Templates_NONFREE_Officorr_KAMI
+    Dir = gid_Dir_Template_Officorr;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreeofficorr);
+End
+
+File gid_File_Extra_Templates_NONFREE_Offimisc_KAMI
+    Dir = gid_Dir_Template_Offimisc;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreeoffimisc);
+End
+
+File gid_File_Extra_Templates_NONFREE_Personal_KAMI
+    Dir = gid_Dir_Template_Personal;
+    TXT_FILE_BODY;
+    Styles = (ARCHIVE, DONT_OVERWRITE);
+    EXTRA_ALL_LANG(kamitemplatesnonfreepersonal);
+End
+
+#endif
diff -Naur -x .svn null/makefile.mk scp2/source/accessories/makefile.mk
--- null/makefile.mk	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/makefile.mk	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,70 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile: makefile.mk,v $
+#
+#   $Revision: 1.0 $
+#
+#   last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJPCH=
+
+PRJNAME=scp2
+TARGET=accessories
+TARGETTYPE=CUI
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :	settings.mk
+
+SCP_PRODUCT_TYPE=osl
+PARFILES= \
+        module_accessories.par             \
+        module_gallery_accessories.par     \
+        module_templates_accessories.par   \
+        module_samples_accessories.par     \
+        module_font_accessories.par        \
+        file_accessories.par               \
+        file_gallery_accessories.par       \
+        file_templates_accessories.par     \
+        file_samples_accessories.par       \
+        file_font_accessories.par
+
+ULFFILES= \
+        module_accessories.ulf             \
+        module_gallery_accessories.ulf     \
+        module_templates_accessories.ulf   \
+        module_samples_accessories.ulf     \
+        module_font_accessories.ulf
+
+
+# --- File ---------------------------------------------------------
+.INCLUDE :  target.mk
diff -Naur -x .svn null/module_accessories.scp scp2/source/accessories/module_accessories.scp
--- null/module_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_accessories.scp	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,46 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+Module gid_Module_Optional_Accessories
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES);
+    ParentID = gid_Module_Optional;
+    Files = (
+        gid_File_Basic_Premium  ); 
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
diff -Naur -x .svn null/module_accessories.ulf scp2/source/accessories/module_accessories.ulf
--- null/module_accessories.ulf	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_accessories.ulf	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,42 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.ulf,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES]
+de = "Zubehör"
+en-US = "Accessories"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES]
+de = "Wertvolles %PRODUCTNAME Zubehör inklusive verschiedener Gallerythemen, Vorlagen, Beispieldokumenten und Fonts."
+en-US = "Useful %PRODUCTNAME accessories including various Galleries, Templates, Sample documents and Fonts."
diff -Naur -x .svn null/module_font_accessories.scp scp2/source/accessories/module_font_accessories.scp
--- null/module_font_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_font_accessories.scp	2007-12-24 09:31:39.000000000 +0100
@@ -0,0 +1,195 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+#ifndef WITHOUT_EXTRA_FONT
+
+Module gid_Module_Optional_Accessories_Fonts
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES_FONTS);
+    ParentID = gid_Module_Optional_Accessories;
+    Files = ( 
+        gid_File_Fnt_airmole,
+        gid_File_Fnt_airmolea,
+        gid_File_Fnt_airmoles,
+        gid_File_Fnt_airmoleq,
+        gid_File_Fnt_astronbo,
+        gid_File_Fnt_astronbi,
+        gid_File_Fnt_astronbv,
+        gid_File_Fnt_astronbw,
+        gid_File_Fnt_baveuse,
+        gid_File_Fnt_Berylium,
+        gid_File_Fnt_Beryliub,
+        gid_File_Fnt_Berylibi,
+        gid_File_Fnt_Beryliui,
+        gid_File_Fnt_BETSY2,
+        gid_File_Fnt_bluehigh,
+        gid_File_Fnt_bluebold,
+        gid_File_Fnt_bluecond,
+        gid_File_Fnt_bluehigl,
+        gid_File_Fnt_bullpen_,
+        gid_File_Fnt_bullpen3,
+        gid_File_Fnt_bullpeni,
+        gid_File_Fnt_colourba,
+        gid_File_Fnt_colourbb,
+        gid_File_Fnt_counters,
+        gid_File_Fnt_creditva,
+        gid_File_Fnt_creditvb,
+        gid_File_Fnt_creditvz,
+        gid_File_Fnt_creditvi,
+        gid_File_Fnt_edmunds,
+        gid_File_Fnt_edmundis,
+        gid_File_Fnt_engeregu,
+        gid_File_Fnt_engebold,
+        gid_File_Fnt_engeboit,
+        gid_File_Fnt_engeexpa,
+        gid_File_Fnt_engeexbo,
+        gid_File_Fnt_engeexbi,
+        gid_File_Fnt_engeexit,
+        gid_File_Fnt_engeital,
+        gid_File_Fnt_goodfish,
+        gid_File_Fnt_goodfisb,
+        gid_File_Fnt_goodfisc,
+        gid_File_Fnt_goodfisi,
+        gid_File_Fnt_guanine_,
+        gid_File_Fnt_gunplay,
+        gid_File_Fnt_kirsty__,
+        gid_File_Fnt_kirsty_b,
+        gid_File_Fnt_kirstybi,
+        gid_File_Fnt_kirsty_i,
+        gid_File_Fnt_kirstyin,
+        gid_File_Fnt_mufferaw,
+        gid_File_Fnt_nasaliza,
+        gid_File_Fnt_NEUROPOL,
+        gid_File_Fnt_numberpi,
+        gid_File_Fnt_pricedow,
+        gid_File_Fnt_pupcat__,
+        gid_File_Fnt_rina,
+        gid_File_Fnt_sandoval,
+        gid_File_Fnt_SAPPM___,
+        gid_File_Fnt_screenge,
+        gid_File_Fnt_steelfis,
+        gid_File_Fnt_steelfib,
+        gid_File_Fnt_steelout,
+        gid_File_Fnt_strenuou,
+        gid_File_Fnt_strenu3d,
+        gid_File_Fnt_subpear_,
+        gid_File_Fnt_teen____,
+        gid_File_Fnt_teenbold,
+        gid_File_Fnt_teenbdit,
+        gid_File_Fnt_teenital,
+        gid_File_Fnt_teenlite,
+        gid_File_Fnt_teenlita,
+        gid_File_Fnt_typoderm,
+        gid_File_Fnt_vahika_,
+        gid_File_Fnt_vahikab,
+        gid_File_Fnt_vahikac,
+        gid_File_Fnt_vahikai,
+        gid_File_Fnt_vectroid,
+        gid_File_Fnt_velvendc,
+        gid_File_Fnt_velvenda,
+        gid_File_Fnt_vibrocen,
+        gid_File_Fnt_vibroceb,
+        gid_File_Fnt_vibrocex,
+        gid_File_Fnt_vibrocei,
+        gid_File_Fnt_wintermu,
+        gid_File_Fnt_youregon,
+        gid_File_Fnt_youregoi,
+        gid_File_Fnt_zekton__,
+        gid_File_Fnt_zektonbo,
+        gid_File_Fnt_zektonbi,
+        gid_File_Fnt_zektonit,
+        gid_File_Fnt_zorque,
+        gid_File_Fnt_-URW-Gothic_L_Avantgarde,
+        gid_File_Fnt_-URW-Gothic_L_Bold_Avantgarde,
+        gid_File_Fnt_-URW-Gothic_L_Italic_Avantgarde,
+        gid_File_Fnt_-URW-Gothic_L_Bold_Italic_Avantgarde,
+        gid_File_Fnt_-URW-Bookman_L_Bookman,
+        gid_File_Fnt_-URW-Bookman_L_Bold_Bookman,
+        gid_File_Fnt_-URW-Bookman_L_Italic_Bookman,
+        gid_File_Fnt_-URW-Bookman_L_Bold_Italic_Bookman,
+        gid_File_Fnt_-URW-Century_Schoolbook_L_New_Century_SchoolBook,
+        gid_File_Fnt_-URW-Century_Schoolbook_L_Bold_New_Century_SchoolBook,
+        gid_File_Fnt_-URW-Century_Schoolbook_L_Italic_New_Century_SchoolBook,
+        gid_File_Fnt_-URW-Century_Schoolbook_L_Bold_Italic_New_Century_SchoolBook,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Helvetica,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Helvetica,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Italic_Helvetica,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Italic_Helvetica,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Helvetica_Condensed,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Helvetica_Condensed,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Italic_Helvetica_Condensed,
+        gid_File_Fnt_-URW-Nimbus_Sans_L_Bold_Italic_Helvetica_Condensed,
+        gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Times,
+        gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Bold_Times,
+        gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Italic_Times,
+        gid_File_Fnt_-URW-Nimbus_Roman_No9_L_Bold_Italic_Times,
+        gid_File_Fnt_-URW-Nimbus_Mono_L_Courier,
+        gid_File_Fnt_-URW-Nimbus_Mono_L_Bold_Courier,
+        gid_File_Fnt_-URW-Nimbus_Mono_LK_Italic_Courier,
+        gid_File_Fnt_-URW-Nimbus_Mono_L_Bold_Italic_Courier,
+        gid_File_Fnt_-URW-Palladio_L_Palitino,
+        gid_File_Fnt_-URW-Palladio_L_Bold_Palitino,
+        gid_File_Fnt_-URW-Palladio_L_Italic_Palitino,
+        gid_File_Fnt_-URW-Palladio_L_Bold_Italic_Palitino,
+        gid_File_Fnt_-URW-Standard_Symbols_L_Symbol,
+        gid_File_Fnt_-URW-Chancery_L_Zapf_Chancery,
+        gid_File_Fnt_Dingbats,
+        gid_File_Fnt_Luxi_Mono_Bold,
+        gid_File_Fnt_Luxi_Mono_Bold_Oblique,
+        gid_File_Fnt_Luxi_Mono_Regular,
+        gid_File_Fnt_Luxi_Mono_Oblique,
+        gid_File_Fnt_Luxi_Serif_Bold,
+        gid_File_Fnt_Luxi_Serif_Bold_Oblique,
+        gid_File_Fnt_Luxi_Serif_Regular,
+        gid_File_Fnt_Luxi_Serif_Oblique,
+        gid_File_Fnt_Luxi_Sans_Bold,
+        gid_File_Fnt_Luxi_Sans_Bold_Oblique,
+        gid_File_Fnt_Luxi_Sans_Regular,
+        gid_File_Fnt_Luxi_Sans_Oblique,
+        gid_File_Fnt_Gentium,
+        gid_File_Fnt_Gentium_Italic,
+        gid_File_Fnt_GentiumAlt,
+        gid_File_Fnt_GentiumAlt_Italic,
+        gid_File_Fnt_MgOpen_Canonica_Bold,
+        gid_File_Fnt_MgOpen_Canonica_Bold_Italic,
+        gid_File_Fnt_MgOpen_Canonica_Italic,
+        gid_File_Fnt_MgOpen_Canonica_Regular );
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
+
+#endif
diff -Naur -x .svn null/module_font_accessories.ulf scp2/source/accessories/module_font_accessories.ulf
--- null/module_font_accessories.ulf	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_font_accessories.ulf	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,42 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.ulf,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES_FONTS]
+de = "Fonts"
+en-US = "Fonts"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES_FONTS]
+de = "Zusätzliche Fonts für %PRODUCTNAME und andere Applikationen."
+en-US = "Additional fonts for %PRODUCTNAME and other applications."
diff -Naur -x .svn null/module_gallery_accessories.scp scp2/source/accessories/module_gallery_accessories.scp
--- null/module_gallery_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_gallery_accessories.scp	2007-12-24 09:31:48.000000000 +0100
@@ -0,0 +1,66 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+#ifndef WITHOUT_EXTRA_GALLERY
+
+Module gid_Module_Optional_Accessories_Gallery
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES_GALLERY);
+    ParentID = gid_Module_Optional_Accessories;
+    Files = (
+        gid_File_Extra_Gallery_KAMI,
+        gid_File_Extra_Gallery_Draws_People_KAMI,
+        gid_File_Extra_Gallery_Photos_Plants_KAMI,
+        gid_File_Extra_Gallery_Photos_Celebration_KAMI,
+        gid_File_Extra_Gallery_Photos_Fauna_KAMI,
+        gid_File_Extra_Gallery_Photos_Flowers_KAMI,
+        gid_File_Extra_Gallery_Photos_FoodsandDrinks_KAMI,
+        gid_File_Extra_Gallery_Photos_Humans_KAMI,
+        gid_File_Extra_Gallery_Photos_Landscapes_KAMI,
+        gid_File_Extra_Gallery_Photos_Objects_KAMI,
+        gid_File_Extra_Gallery_Photos_Buildings_KAMI,
+        gid_File_Extra_Gallery_Photos_Cities_KAMI,
+        gid_File_Extra_Gallery_Photos_Space_KAMI,
+        gid_File_Extra_Gallery_Photos_Statues_KAMI
+        gid_File_Extra_Gallery_Photos_Travel_KAMI,
+        gid_File_Extra_Gallery_Elements_Bullets2_KAMI,
+        gid_File_Extra_Gallery_NONFREE_KAMI );
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
+
+#endif
diff -Naur -x .svn null/module_gallery_accessories.ulf scp2/source/accessories/module_gallery_accessories.ulf
--- null/module_gallery_accessories.ulf	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_gallery_accessories.ulf	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,42 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.ulf,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES_GALLERY]
+de = "Zusätzliche Gallerythemen"
+en-US = "Additional Gallery Themes"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES_GALLERY]
+de = "Die %PRODUCTNAME Gallerythemen beinhalten mehr als 2900 Komponenten aus verschiedenen Themen."
+en-US = "The %PRODUCTNAME Gallery contains more than 2900 elements in various themes."
diff -Naur -x .svn null/module_samples_accessories.scp scp2/source/accessories/module_samples_accessories.scp
--- null/module_samples_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_samples_accessories.scp	2007-12-24 09:31:57.000000000 +0100
@@ -0,0 +1,77 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+#ifndef WITHOUT_EXTRA_SAMPLE
+
+Module gid_Module_Optional_Accessories_Samples
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES_SAMPLES);
+    ParentID = gid_Module_Optional_Accessories;
+    Files = (
+        gid_File_Extra_Samples_KAMI
+        gid_File_Extra_Samples_NONFREE_KAMI  );
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
+
+Module gid_Module_Optional_Accessories_Documentations
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES_DOCUMENTATIONS);
+    ParentID = gid_Module_Optional_Accessories;
+    Files = (
+        gid_File_Extra_Samples_KAMI,
+        gid_File_Extra_Samples_Documentations_KAMI
+        gid_File_Extra_Samples_NONFREE_KAMI,
+        gid_File_Extra_Samples_NONFREE_Documentations_KAMI );
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
+
+Module gid_Module_Optional_Accessories_Advertisement
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES_ADVERTISEMENT);
+    ParentID = gid_Module_Optional_Accessories;
+    Files = (
+        gid_File_Extra_Samples_KAMI,
+        gid_File_Extra_Samples_Advertisement_KAMI
+        gid_File_Extra_Samples_NONFREE_KAMI,
+        gid_File_Extra_Samples_NONFREE_Advertisement_KAMI );
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
+
+#endif
diff -Naur -x .svn null/module_samples_accessories.ulf scp2/source/accessories/module_samples_accessories.ulf
--- null/module_samples_accessories.ulf	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_samples_accessories.ulf	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,59 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.ulf,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES_SAMPLES]
+de = "Beispieldokumente"
+en-US = "Sample documents"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES_SAMPLES]
+de = "Beispieldokumente sind nützlich für das Verständnis."
+en-US = "Sample documents are good for learning."
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES_DOCUMENTATIONS]
+de = "Dokumentation"
+en-US = "Documentations"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES_DOCUMENTATIONS]
+de = "%PRODUCTNAME Dokumentation."
+en-US = "%PRODUCTNAME documentations."
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES_ADVERTISEMENT]
+de = "Werbung"
+en-US = "Advertisement"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES_ADVERTISEMENT]
+de = "%PRODUCTNAME Werbematerial."
+en-US = "%PRODUCTNAME advertisement materials."
+
diff -Naur -x .svn null/module_templates_accessories.scp scp2/source/accessories/module_templates_accessories.scp
--- null/module_templates_accessories.scp	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_templates_accessories.scp	2007-12-24 09:32:06.000000000 +0100
@@ -0,0 +1,69 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.scp,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+#ifndef WITHOUT_EXTRA_TEMPLATE
+
+Module gid_Module_Optional_Accessories_Templates
+    MOD_NAME_DESC(MODULE_OPTIONAL_ACCESSORIES_TEMPLATES);
+    ParentID = gid_Module_Optional_Accessories;
+    Files = ( 
+        gid_File_Extra_Templates_KAMI,
+        gid_File_Extra_Templates_Layout_KAMI,
+        gid_File_Extra_Templates_Present_KAMI,
+        gid_File_Extra_Templates_Educate_KAMI,
+        gid_File_Extra_Templates_Finance_KAMI,
+        gid_File_Extra_Templates_Forms_KAMI,
+        gid_File_Extra_Templates_Misc_KAMI,
+        gid_File_Extra_Templates_Officorr_KAMI,
+        gid_File_Extra_Templates_Offimisc_KAMI,
+        gid_File_Extra_Templates_Personal_KAMI
+        gid_File_Extra_Templates_NONFREE_KAMI,
+        gid_File_Extra_Templates_NONFREE_Layout_KAMI,
+        gid_File_Extra_Templates_NONFREE_Present_KAMI,
+        gid_File_Extra_Templates_NONFREE_Educate_KAMI,
+        gid_File_Extra_Templates_NONFREE_Finance_KAMI,
+        gid_File_Extra_Templates_NONFREE_Forms_KAMI,
+        gid_File_Extra_Templates_NONFREE_Misc_KAMI,
+        gid_File_Extra_Templates_NONFREE_Officorr_KAMI,
+        gid_File_Extra_Templates_NONFREE_Offimisc_KAMI,
+        gid_File_Extra_Templates_NONFREE_Personal_KAMI );
+    Minimal = NO;
+    Default = YES;
+    Styles = ( );
+End
+
+#endif
diff -Naur -x .svn null/module_templates_accessories.ulf scp2/source/accessories/module_templates_accessories.ulf
--- null/module_templates_accessories.ulf	1970-01-01 01:00:00.000000000 +0100
+++ scp2/source/accessories/module_templates_accessories.ulf	2007-01-09 19:03:16.000000000 +0100
@@ -0,0 +1,44 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_accessories.ulf,v $
+ *
+ *  $Revision: 1.0 $
+ *
+ *  last change: $Author: KAMI $ $Date: 2006/02/18 14:50:39 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+
+[STR_NAME_MODULE_OPTIONAL_ACCESSORIES_TEMPLATES]
+de = "Vorlagen"
+en-US = "Templates"
+
+[STR_DESC_MODULE_OPTIONAL_ACCESSORIES_TEMPLATES]
+de = "Erstelle perfekte Dokumente innerhalb von Sekunden mit Dokumentvorlagen."
+en-US = "Create perfect documents within seconds with document templates."
+
