--- scp2/source/ooo/file_ooo.scp.old	2005-02-22 19:50:37.000000000 +0100
+++ scp2/source/ooo/file_ooo.scp	2005-02-23 19:24:05.000000000 +0100
@@ -2385,12 +2385,14 @@
 End
 #endif
 
+#ifndef WITHOUT_BINFILTER
 File gid_Starregistry_Legacy_Binfilters_Rdb
     TXT_FILE_BODY;
     Name = "legacy_binfilters.rdb";
     Dir = gid_Dir_Program;
     Styles = (PACKED, STARREGISTRY);
 End
+#endif
 
 File gid_Starregistry_Services_Rdb
     TXT_FILE_BODY;
