--- scp2/prj/build.lst	2007-11-07 10:33:28.000000000 +0100
+++ scp2/prj/build.lst	2007-11-07 11:13:56.000000000 +0100
@@ -25,4 +25,5 @@
 cp    scp2\source\ure             nmake    -    all    cp_ure cp_langmacros NULL
 cp    scp2\source\onlineupdate    nmake    -    all    cp_update cp_langmacros NULL
 cp    scp2\source\ooo_wo_ure      nmake    -    all    cp_ooo_wo_ure cp_langmacros NULL
-cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_calc cp_canvas cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_layout cp_lingu cp_math cp_ooo cp_python cp_quickstart cp_testtool cp_writer cp_base cp_xsltfilter cp_winexplorerext cp_ure cp_sdkoo cp_update cp_ooo_wo_ure NULL
+cp    scp2\source\glib2           nmake    -    all    cp_glib2 cp_langmacros NULL
+cp    scp2\util                   nmake    -    all    cp_util cp_activex cp_calc cp_canvas cp_crashrep cp_draw cp_gnome cp_graphicfilter cp_impress cp_javafilter cp_kde cp_layout cp_lingu cp_math cp_ooo cp_python cp_quickstart cp_testtool cp_writer cp_base cp_xsltfilter cp_winexplorerext cp_ure cp_sdkoo cp_update cp_ooo_wo_ure cp_glib2 NULL
--- /dev/null	2007-05-04 12:54:36.000000000 +0200
+++ scp2/source/glib2/file_glib2.scp	2007-11-07 11:21:40.000000000 +0100
@@ -0,0 +1,70 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: file_glib2.scp,v $
+ *
+ *  $Revision: 1.5 $
+ *
+ *  last change: $Author: hr $ $Date: 2006/06/20 04:52:09 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+#include "macros.inc"
+
+#if ! (defined (FREEBSD) || defined (NETBSD) || defined(MACOSX) || defined(SYSTEM_GLIB2) || defined(WNT))
+#if defined( _gcc3 )
+
+File gid_File_Lib_Glib20
+    TXT_FILE_BODY;
+    Name = STRING(libglib-2.0.so.0);
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Gmodule20
+    TXT_FILE_BODY;
+    Name = STRING(libgmodule-2.0.so.0);
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Gobject20
+    TXT_FILE_BODY;
+    Name = STRING(libgobject-2.0.so.0);
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Gthread20
+    TXT_FILE_BODY;
+    Name = STRING(libgthread-2.0.so.0);
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+#endif
+
+#endif
--- /dev/null	2007-05-04 12:54:36.000000000 +0200
+++ scp2/source/glib2/makefile.mk	2007-11-07 11:14:17.000000000 +0100
@@ -0,0 +1,62 @@
+#*************************************************************************
+#
+#   OpenOffice.org - a multi-platform office productivity suite
+#
+#   $RCSfile: makefile.mk,v $
+#
+#   $Revision: 1.5 $
+#
+#   last change: $Author: hr $ $Date: 2007/01/02 15:28:27 $
+#
+#   The Contents of this file are made available subject to
+#   the terms of GNU Lesser General Public License Version 2.1.
+#
+#
+#     GNU Lesser General Public License Version 2.1
+#     =============================================
+#     Copyright 2005 by Sun Microsystems, Inc.
+#     901 San Antonio Road, Palo Alto, CA 94303, USA
+#
+#     This library is free software; you can redistribute it and/or
+#     modify it under the terms of the GNU Lesser General Public
+#     License version 2.1, as published by the Free Software Foundation.
+#
+#     This library is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#     Lesser General Public License for more details.
+#
+#     You should have received a copy of the GNU Lesser General Public
+#     License along with this library; if not, write to the Free Software
+#     Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+#     MA  02111-1307  USA
+#
+#*************************************************************************
+
+PRJ=..$/..
+
+PRJPCH=
+
+PRJNAME=scp2
+TARGET=glib2
+TARGETTYPE=CUI
+
+# --- Settings -----------------------------------------------------
+
+.INCLUDE :  settings.mk
+
+SCP_PRODUCT_TYPE=osl
+
+.IF "$(SYSTEM_GLIB2)" != "YES"
+PARFILES =                   \
+        module_glib2.par     \
+        file_glib2.par
+
+ULFFILES= \
+        module_glib2.ulf
+
+.ENDIF
+
+# --- File ---------------------------------------------------------
+
+.INCLUDE :  target.mk
--- /dev/null	2007-05-04 12:54:36.000000000 +0200
+++ scp2/source/glib2/module_glib2.ulf	2007-11-07 11:16:06.000000000 +0100
@@ -0,0 +1,41 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_gnome.ulf,v $
+ *
+ *  $Revision: 1.6 $
+ *
+ *  last change: $Author: ihi $ $Date: 2007/04/19 15:24:53 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+[STR_NAME_MODULE_OPTIONAL_GLIB2]
+en-US = "Glib2 libraries"
+
+[STR_DESC_MODULE_OPTIONAL_GLIB2]
+en-US = "Glib2 libraries for %PRODUCTNAME %PRODUCTVERSION."
+
--- scp2/util/makefile.mk	2007-11-07 12:32:58.000000000 +0100
+++ scp2/util/makefile.mk	2007-11-07 12:33:05.000000000 +0100
@@ -183,6 +183,12 @@
 SCP1FILES += aqua_ooo.par
 .ENDIF
 
+.IF "$(SYSTEM_GLIB2)"!="YES"
+SCP1FILES += \
+             module_glib2.par	\
+	     file_glib2.par
+.ENDIF
+
 # ------------------------------------------------------------------------
 # OpenOffice.org with JRE
 
--- /dev/null	2007-05-04 12:54:36.000000000 +0200
+++ scp2/source/glib2/module_glib2.scp	2007-11-07 12:39:27.000000000 +0100
@@ -0,0 +1,52 @@
+/*************************************************************************
+ *
+ *  OpenOffice.org - a multi-platform office productivity suite
+ *
+ *  $RCSfile: module_glib2.scp,v $
+ *
+ *  $Revision: 1.6 $
+ *
+ *  last change: $Author: rt $ $Date: 2007/07/03 12:06:27 $
+ *
+ *  The Contents of this file are made available subject to
+ *  the terms of GNU Lesser General Public License Version 2.1.
+ *
+ *
+ *    GNU Lesser General Public License Version 2.1
+ *    =============================================
+ *    Copyright 2005 by Sun Microsystems, Inc.
+ *    901 San Antonio Road, Palo Alto, CA 94303, USA
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License version 2.1, as published by the Free Software Foundation.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Lesser General Public
+ *    License along with this library; if not, write to the Free Software
+ *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ *    MA  02111-1307  USA
+ *
+ ************************************************************************/
+
+#include "macros.inc"
+
+#if ! (defined (FREEBSD) || defined (NETBSD) || defined(MACOSX) || defined(SYSTEM_GLIB2) || defined(WNT))
+#if defined( _gcc3 )
+
+Module gid_Module_Optional_Glib2
+    ParentID = gid_Module_Optional;
+    Default = YES;
+    MOD_NAME_DESC(MODULE_OPTIONAL_GLIB2);
+    Styles = (DONTSHOWINUSERINSTALL);
+    Files = (gid_File_Lib_Glib20, gid_File_Lib_Gmodule20, gid_File_Lib_Gobject20, gid_File_Lib_Gthread20);
+    Dirs = ();
+End
+
+#endif
+
+#endif
