--- scp2/source/ooo/file_library_ooo.scp	2007-11-01 21:36:36.000000000 +0100
+++ scp2/source/ooo/file_library_ooo.scp	2007-11-01 23:04:20.000000000 +0100
@@ -1082,6 +1082,41 @@
 
 #endif
 
+#if ! (defined (FREEBSD) || defined (NETBSD) || defined(MACOSX) || defined(SYSTEM_GLIB2) || defined(WNT))
+#if defined( _gcc3 )
+
+File gid_File_Lib_Glib20
+    TXT_FILE_BODY;
+    Name = STRING(libglib-2.0.so.0));
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Gmodule20
+    TXT_FILE_BODY;
+    Name = STRING(libgmodule-2.0.so.0));
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Gobject20
+    TXT_FILE_BODY;
+    Name = STRING(libgobject-2.0.so.0));
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+File gid_File_Lib_Gthread20
+    TXT_FILE_BODY;
+    Name = STRING(libgthread-2.0.so.0));
+    Dir = gid_Dir_Program;
+    Styles = (PACKED);
+End
+
+#endif
+
+#endif
+
 #if defined(LINUX) || defined(FREEBSD) || defined(NETBSD) || defined(IRIX) || defined(MACOSX)
 
 File gid_File_Lib_Gcc2_Uno
