--- scp2/util/makefile.mk.orig	2007-08-10 19:15:34.000000000 +0200
+++ scp2/util/makefile.mk	2007-09-12 15:16:23.177216800 +0200
@@ -90,7 +90,18 @@
              module_python_mailmerge.par   \
              file_python.par               \
              profileitem_python.par        \
-             shortcut_python.par
+             shortcut_python.par           \
+             module_accessories.par        \
+             module_font_accessories.par      \
+             module_gallery_accessories.par   \
+             module_samples_accessories.par   \
+             module_templates_accessories.par \
+             file_accessories.par     \
+             file_font_accessories.par     \
+             file_gallery_accessories.par  \
+             file_samples_accessories.par  \
+             file_templates_accessories.par
+
 
 .IF "$(SOLAR_JAVA)"!=""
 SCP1FILES +=                               \
@@ -223,7 +234,18 @@
              module_python.par             \
              module_python_mailmerge.par   \
              file_python.par               \
-             profileitem_python.par
+             profileitem_python.par        \
+             module_accessories.par        \
+             module_font_accessories.par      \
+             module_gallery_accessories.par   \
+             module_samples_accessories.par   \
+             module_templates_accessories.par \
+             file_accessories.par     \
+             file_font_accessories.par     \
+             file_gallery_accessories.par  \
+             file_samples_accessories.par  \
+             file_templates_accessories.par
+
 
 .IF "$(SOLAR_JAVA)"!=""
 SCP2FILES +=                               \
