Index: sd/source/ui/app/optsitem.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/optsitem.cxx,v
retrieving revision 1.31
diff -u -p -r1.31 optsitem.cxx
--- sd/source/ui/app/optsitem.cxx	27 Jan 2005 14:12:09 -0000	1.31
+++ sd/source/ui/app/optsitem.cxx	21 May 2005 16:11:16 -0000
@@ -1711,8 +1711,9 @@ SfxPoolItem* SdOptionsPrintItem::Clone( 
 
 int SdOptionsPrintItem::operator==( const SfxPoolItem& rAttr ) const
 {
-	DBG_ASSERT( SfxPoolItem::operator==(rAttr), "unterschiedliche Typen" );
-	return( (SdOptionsPrint&) *this == (const SdOptionsPrint&)(const SdOptionsPrintItem&) rAttr );
+	const SdOptionsPrint& rRHS = 
+		dynamic_cast<const SdOptionsPrint&>(rAttr);
+	return SdOptionsPrint::operator==(rRHS);
 }
 
 // -----------------------------------------------------------------------
