Index: sd/inc/sdfilter.hxx
===================================================================
--- sd/inc/sdfilter.hxx.orig	2003-11-26 21:36:47.000000000 +0100
+++ sd/inc/sdfilter.hxx	2003-11-26 21:48:35.000000000 +0100
@@ -104,6 +104,7 @@ protected:
 	SfxProgress*				mpProgress;
 	sal_Bool					mbIsDraw : 1;
 	sal_Bool					mbShowProgress : 1;
+	sal_Bool					mbOldHideSpell : 1;
 
 	::vos::OModule*				OpenLibrary( const ::rtl::OUString& rLibraryName ) const;
 	void						CreateStatusIndicator();
Index: sd/source/filter/sdfilter.cxx
===================================================================
--- sd/source/filter/sdfilter.cxx.orig	2003-11-26 21:39:53.000000000 +0100
+++ sd/source/filter/sdfilter.cxx	2003-11-26 21:48:41.000000000 +0100
@@ -107,8 +107,10 @@ SdFilter::SdFilter( SfxMedium& rMedium, 
 	mxModel( rDocShell.GetModel() ),
 	mpProgress( NULL ),
 	mbIsDraw( rDocShell.GetDocumentType() == DOCUMENT_TYPE_DRAW ),
-	mbShowProgress( bShowProgress )
+	mbShowProgress( bShowProgress ),
+	mbOldHideSpell( mrDocument.GetHideSpell() )
 {
+	mrDocument.SetHideSpell( sal_True );
 }
 
 // -----------------------------------------------------------------------------
@@ -117,6 +119,8 @@ SdFilter::~SdFilter()
 {
 	if( !mrDocShell.HasSpecialProgress() )
 		delete mpProgress;
+
+	mrDocument.SetHideSpell( mbOldHideSpell );
 }
 
 // -----------------------------------------------------------------------------
