--- sd/source/ui/view/viewshe3.cxx.orig	2005-09-01 15:08:37 +02:00
+++ sd/source/ui/view/viewshe3.cxx	2005-09-01 15:09:47 +02:00
@@ -1214,7 +1214,7 @@ SdPage* ViewShell::CreateOrDuplicatePage
         SfxBoolItem aMakeToolPaneVisible (ID_VAL_ISVISIBLE, TRUE);
         SfxUInt32Item aPanelId (ID_VAL_PANEL_INDEX,
             toolpanel::TaskPaneViewShell::PID_LAYOUT);
-        GetDispatcher()->Execute (
+        GetViewFrame()->GetDispatcher()->Execute (
             SID_TASK_PANE,
             SFX_CALLMODE_ASYNCHRON | SFX_CALLMODE_RECORD,
             &aMakeToolPaneVisible,
