diff -rup sd/source/ui/slideshow-orig/slideshowviewimpl.cxx sd/source/ui/slideshow/slideshowviewimpl.cxx
--- sd/source/ui/slideshow-orig/slideshowviewimpl.cxx	2008-02-22 15:31:52.000000000 +0100
+++ sd/source/ui/slideshow/slideshowviewimpl.cxx	2008-02-22 15:31:26.000000000 +0100
@@ -368,13 +368,7 @@ geometry::AffineMatrix2D SAL_CALL SlideS
     ::osl::MutexGuard aGuard( m_aMutex );
 	::vos::OGuard aSolarGuard( Application::GetSolarMutex() );
 
-	const Size& rTmpSize( mrOutputWindow.GetSizePixel() );
-    
-    // Reduce available width by one, as the slides might actually
-    // render one pixel wider and higher as aPageSize below specifies
-    // (when shapes of page size have visible border lines)
-	const Size  aWindowSize( rTmpSize.Width()-1, 
-                             rTmpSize.Height()-1 );
+	const Size aWindowSize( mrOutputWindow.GetSizePixel() );
     Size aOutputSize( aWindowSize );
 
 	if( meAnimationMode != ANIMATIONMODE_SHOW )
@@ -401,6 +395,12 @@ geometry::AffineMatrix2D SAL_CALL SlideS
 	Point aOutputOffset( ( aWindowSize.Width() - aOutputSize.Width() ) >> 1,
 							( aWindowSize.Height() - aOutputSize.Height() ) >> 1 );
 
+    // Reduce available width by one, as the slides might actually
+    // render one pixel wider and higher as aPageSize below specifies
+    // (when shapes of page size have visible border lines)
+    aOutputSize.Width() --;
+    aOutputSize.Height() --;
+    
     ::basegfx::B2DHomMatrix aMatrix;
 
 	maPresentationArea = Rectangle( aOutputOffset, aOutputSize );
