Index: Common.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Common.xcu,v
retrieving revision 1.31
diff -u -u -r1.31 Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu	23 Mar 2005 10:53:46 -0000	1.31
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	25 May 2005 23:23:30 -0000
@@ -597,4 +597,9 @@
       <value>false</value>
     </prop>
   </node>
+  <node oor:name="ExternalMailer">
+    <prop oor:name="Program" oor:type="xs:string">
+      <value>sensible-ooomua</value>
+    </prop>
+  </node>
 </oor:component-data>
