Index: Common.xcu
===================================================================
RCS file: /cvs/util/officecfg/registry/data/org/openoffice/Office/Common.xcu,v
retrieving revision 1.31
diff -u -u -r1.31 Common.xcu
--- officecfg/registry/data/org/openoffice/Office/Common.xcu	23 Mar 2005 10:53:46 -0000	1.31
+++ officecfg/registry/data/org/openoffice/Office/Common.xcu	25 May 2005 23:23:30 -0000
@@ -597,4 +597,9 @@
       <value>false</value>
     </prop>
   </node>
+  <node oor:name="ExternalMailer">
+    <prop oor:name="Program" oor:type="xs:string">
+      <value>sensible-ooomua</value>
+    </prop>
+  </node>
 </oor:component-data>
Index: shell/source/unix/misc/senddoc.sh
===================================================================
RCS file: /cvs/gsl/shell/source/unix/misc/senddoc.sh,v
retrieving revision 1.8
diff -u -u -r1.8 senddoc.sh
--- shell/source/unix/misc/senddoc.sh	27 Sep 2005 12:55:00 -0000	1.8
+++ shell/source/unix/misc/senddoc.sh	16 Feb 2006 00:38:29 -0000
@@ -91,6 +91,19 @@
 	shift
 fi
 
+case `basename "$MAILER"` in
+	sensible-ooomua)
+		if [ -n "$GNOME_DESKTOP_SESSION_ID" -a -x /usr/bin/evolution ]; then
+			MAILER=/usr/bin/evolution
+		elif [ -n "$KDE_FULL_SESSION" -a -x /usr/bin/kmail ]; then
+			MAILER=/usr/bin/kmail
+		elif [ -x /usr/bin/evolution ]; then
+			# default
+			MAILER=/usr/bin/evolution
+		fi
+		;;
+esac
+
 # autodetect mail client from executable name
 case `basename "$MAILER" | sed 's/-.*$//'` in
 
