--- sfx2/source/doc/doctempl.src.old	2007-06-28 01:21:07.000000000 +0200
+++ sfx2/source/doc/doctempl.src	2007-08-24 16:25:12.000000000 +0200
@@ -51,6 +51,7 @@ StringArray TEMPLATE_SHORT_NAMES_ARY
 		< "layout" ; > ;
 		< "presnt" ; > ;
 		< "misc" ; > ;
+		< "labels" ; > ;
 	};
 };
 
@@ -68,6 +69,7 @@ StringArray TEMPLATE_LONG_NAMES_ARY
         < "Presentation Backgrounds" ; > ;
         < "Presentations" ; > ;
         < "Miscellaneous" ; > ;
+        < "Labels" ; > ;
     };
 
 	};
