--- vcl/inc/toolbox.hxx	2005-09-09 13:33:28.000000000 +0200
+++ vcl/inc/toolbox.hxx	2005-11-02 17:03:13.000000000 +0100
@@ -288,7 +288,6 @@ private:
     SAL_DLLPRIVATE void            ImplHideFocus();
 	SAL_DLLPRIVATE void			   ImplUpdateInputEnable();
     SAL_DLLPRIVATE void			   ImplFillLayoutData() const;
-    SAL_DLLPRIVATE const Size&     ImplGetDefaultImageSize() const;
     SAL_DLLPRIVATE void            ImplUpdateCustomMenu();
     SAL_DLLPRIVATE BOOL            ImplHasClippedItems();
     SAL_DLLPRIVATE Point           ImplGetPopupPosition( const Rectangle& rRect, const Size& rSize ) const;
@@ -598,6 +597,8 @@ public:
     Pair GetTextStartEnd( long nText ) const;
     // returns the item id for text portion nText or 0 if nText is invalid
     USHORT GetDisplayItemId( long nText ) const;
+
+    const Size&         ImplGetDefaultImageSize() const;
 };
 
 inline void ToolBox::CheckItem( USHORT nItemId, BOOL bCheck )
--- sfx2/source/toolbox/tbxitem.cxx	27 Jan 2005 10:19:01 -0000	1.46
+++ sfx2/source/toolbox/tbxitem.cxx	3 Feb 2005 13:00:20 -0000
@@ -1704,7 +1704,15 @@
     if ( !aImage )
         aImage = !!aMenuImage ? aMenuImage :
             SvFileInformationManager::GetImage( INetURLObject( aURL ), bBig, bHC );
-    GetToolBox().SetItemImage( GetId(), aImage );
+    Size aBigSize( GetToolBox().ImplGetDefaultImageSize() );
+    if ( bBig && aImage.GetSizePixel() != aBigSize )
+    {
+        BitmapEx aScaleBmpEx( aImage.GetBitmapEx() );
+        aScaleBmpEx.Scale( aBigSize, BMP_SCALE_INTERPOLATE );
+        GetToolBox().SetItemImage( GetId(), Image( aScaleBmpEx ) );
+    } 
+    else
+        GetToolBox().SetItemImage( GetId(), aImage );
 	aLastURL = aURL;
 }
 
