? sfx2/sfx2.vpj
Index: sfx2/source/doc/guisaveas.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/doc/guisaveas.cxx,v
retrieving revision 1.31
diff -u -r1.31 guisaveas.cxx
--- sfx2/source/doc/guisaveas.cxx	26 Nov 2007 16:47:57 -0000	1.31
+++ sfx2/source/doc/guisaveas.cxx	2 Feb 2008 03:46:35 -0000
@@ -679,8 +679,9 @@
 		GetMediaDescr() = aAcceptedArgs;
 
     // the document must be modified
-    if ( !GetModifiable()->isModified() && !bVersInfoNeedsStore )
-		return STATUS_NO_ACTION;
+// always allow save (kohei)
+//  if ( !GetModifiable()->isModified() && !bVersInfoNeedsStore )
+//      return STATUS_NO_ACTION;
 
     // if the document is readonly or a new one a SaveAs operation must be used
     if ( !GetStorable()->hasLocation() || GetStorable()->isReadonly() )
Index: sfx2/source/doc/objserv.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/doc/objserv.cxx,v
retrieving revision 1.101
diff -u -r1.101 objserv.cxx
--- sfx2/source/doc/objserv.cxx	21 Nov 2007 16:47:51 -0000	1.101
+++ sfx2/source/doc/objserv.cxx	2 Feb 2008 03:46:35 -0000
@@ -1058,8 +1058,9 @@
 					if ( !bMediumRO && GetMedium() && IsModified() )
 						rSet.Put(SfxStringItem(
 							nWhich, String(SfxResId(STR_SAVEDOC))));
-					else
-                    	rSet.DisableItem(nWhich);
+// always allow save (kohei)
+//                  else
+//                      rSet.DisableItem(nWhich);
 				}
 				break;
 
