--- sfx2/source/dialog/filedlghelper.cxx	2006-02-17 14:27:13.000000000 +0000
+++ sfx2/source/dialog/filedlghelper.cxx	2006-02-17 14:32:27.000000000 +0000
@@ -2394,6 +2394,27 @@ ErrCode FileDialogHelper::GetGraphic( Gr
 }
 
 // ------------------------------------------------------------------------
+int impl_isFolder( const OUString& rPath )
+{
+	try
+	{
+		::ucb::Content aContent( 
+			rPath, star::uno::Reference< star::ucb::XCommandEnvironment >() );
+		if ( aContent.isFolder() )
+			return 1;
+
+		return 0;
+	}
+	catch ( star::ucb::ContentCreationException const & )
+	{
+	}
+	catch ( star::ucb::InteractiveAugmentedIOException const & )
+	{
+	}
+
+	return -1;
+}
+
 void FileDialogHelper::SetDisplayDirectory( const String& _rPath )
 {
 	if ( !_rPath.Len() )
@@ -2403,29 +2425,24 @@ void FileDialogHelper::SetDisplayDirecto
 	// and take it as filename and the rest of the path should be
 	// the folder
 
-	::rtl::OUString sPath;
-	::rtl::OUString sFileName;
-
 	INetURLObject aObj( _rPath );
-	try
-	{
-		::ucb::Content aContent( 
-			_rPath, star::uno::Reference< star::ucb::XCommandEnvironment >() );
-		if ( !aContent.isFolder() )
-		{
-			sFileName = aObj.GetName( INetURLObject::DECODE_WITH_CHARSET );
-			aObj.removeSegment();
-			mpImp->setFileName( sFileName );
-		}
-	}
-	catch ( star::ucb::ContentCreationException const & )
+
+	::rtl::OUString sFileName = aObj.GetName( INetURLObject::DECODE_WITH_CHARSET );
+	aObj.removeSegment();
+	::rtl::OUString sPath = aObj.GetMainURL( INetURLObject::NO_DECODE );
+
+	int nIsFolder = impl_isFolder( _rPath );
+	if ( nIsFolder == 0 ||
+		 ( nIsFolder == -1 && impl_isFolder( sPath ) == 1 ) )
 	{
+		mpImp->setFileName( sFileName );
+		mpImp->displayFolder( sPath );
 	}
-	catch ( star::ucb::InteractiveAugmentedIOException const & )
+	else
 	{
+		INetURLObject aObjPathName( _rPath );
+		mpImp->displayFolder( aObjPathName.GetMainURL( INetURLObject::NO_DECODE ) );
 	}
-
-    mpImp->displayFolder( aObj.GetMainURL( INetURLObject::NO_DECODE ) );
 }
 
 // ------------------------------------------------------------------------
