--- shell/source/unix/misc/senddoc.sh.old	2008-03-19 15:24:48.000000000 +0100
+++ shell/source/unix/misc/senddoc.sh	2008-03-19 16:46:00.000000000 +0100
@@ -27,7 +27,7 @@ run_mozilla() {
 		moz=$1
 	fi
 
-	if file "$moz" | grep "script" > /dev/null && grep "[NM]PL" "$moz" > /dev/null; then
+	if file -L "$moz" | grep "script" > /dev/null && grep "[NM]PL" "$moz" > /dev/null; then
 		"$moz" -remote 'ping()' 2>/dev/null >/dev/null
 		if [ $? -eq 2 ]; then
 			"$1" -compose "$2" &
