--- shell/source/unix/sysshell/recently_used_file.cxx.old	2005-09-07 21:55:52.000000000 +0200
+++ shell/source/unix/sysshell/recently_used_file.cxx	2006-03-07 18:08:02.000000000 +0100
@@ -58,6 +58,8 @@
 #endif
 
 #include <sys/file.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 #include <unistd.h>
 
@@ -97,8 +99,11 @@
         file_ = fopen(tmp.getStr(), "r+");
         
         /* create if not exist */
-        if (NULL == file_)
+        if (NULL == file_) {
+            mode_t umask_ = umask(S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH);
             file_ = fopen(tmp.getStr(), "w+");
+            umask(umask_);
+        }
         
         if (NULL == file_)
             throw "I/O error opening ~/.recently-used";
