--- solenv/bin/build.pl.orig	2007-11-30 18:30:42.000000000 +0100
+++ solenv/bin/build.pl	2007-11-30 18:31:47.000000000 +0100
@@ -222,7 +222,7 @@
     $StandDir = get_stand_dir();   # This also sets $CurrentPrj
     provide_consistency() if (defined $ENV{CWS_WORK_STAMP} && defined($ENV{COMMON_ENV_TOOLS}));
 
-    $deliver_command = $ENV{DELIVER};
+    $deliver_command = $ENV{DELIVER} ." -quiet";
     if ($dlv_switch) {
         $deliver_command .= ' '. $dlv_switch;
     } else {
