--- binfilter/bf_sc/source/core/tool/sc_rechead.cxx	2006-10-27 16:37:48.000000000 +0200
+++ binfilter/bf_sc/source/core/tool/sc_rechead.cxx	2007-02-22 19:07:47.000000000 +0100
@@ -98,7 +98,7 @@ namespace binfilter {
 /*N*/ 	if ( nPos - nDataPos != nDataSize )				// Default getroffen?
 /*N*/ 	{
 /*N*/ 		nDataSize = nPos - nDataPos;
-/*N*/ 		rStream.Seek(nDataPos - sizeof(ULONG));
+/*N*/ 		rStream.Seek(nDataPos - sizeof(sal_uInt32));
 /*N*/ 		rStream << nDataSize;						// Groesse am Anfang eintragen
 /*N*/ 		rStream.Seek(nPos);
 /*N*/ 	}
@@ -214,7 +214,7 @@ namespace binfilter {
 /*N*/ 	{
 /*N*/ 		nDataSize = nDataEnd - nDataPos;
 /*N*/ 		ULONG nPos = rStream.Tell();
-/*N*/ 		rStream.Seek(nDataPos-sizeof(ULONG));
+/*N*/ 		rStream.Seek(nDataPos-sizeof(sal_uInt32));
 /*N*/ 		rStream << nDataSize;								// Groesse am Anfang eintragen
 /*N*/ 		rStream.Seek(nPos);
 /*N*/ 	}
--- binfilter/bf_svx/source/items/svx_flditem.cxx	2006-10-27 23:15:22.000000000 +0200
+++ binfilter/bf_svx/source/items/svx_flditem.cxx	2007-02-22 19:06:25.000000000 +0100
@@ -403,7 +403,7 @@ namespace binfilter {
 /*N*/ {
 /*N*/ 	USHORT nFormat;
 /*N*/ 	sal_uInt32 nFrameMarker, nCharSetMarker;
-/*N*/ 	long nUlongSize = (long)sizeof(ULONG);
+/*N*/ 	long nUlongSize = (long)sizeof(sal_uInt32);
 /*N*/ 	String aTmpURL;
 /*N*/ 
 /*N*/ 	rStm >> nFormat;
