--- extensions/source/packer/packer2.hxx	2006-03-16 14:00:21.000000000 +0100
+++ extensions/source/packer/packer2.hxx	2007-02-22 19:04:50.000000000 +0100
@@ -93,7 +93,7 @@ class INetURLObject;
 \* ******************************************************************/
 
 const sal_uInt32 PACK_ID = ((('S' << 8 | 'Z' ) << 8 | 'I' ) << 8 | 'P' );
-const USHORT HEADER_SIZE = 2*sizeof( ULONG ) + 3*sizeof( USHORT );
+const USHORT HEADER_SIZE = 2*sizeof( sal_uInt32 ) + 3*sizeof( USHORT );
 
 class LocalFileHeader
 {
--- svx/source/items/flditem.cxx	2006-10-12 14:53:54.000000000 +0200
+++ svx/source/items/flditem.cxx	2007-02-22 19:06:03.000000000 +0100
@@ -402,7 +402,7 @@ void SvxURLField::Load( SvPersistStream 
 {
 	USHORT nFormat;
 	sal_uInt32 nFrameMarker, nCharSetMarker;
-	long nUlongSize = (long)sizeof(ULONG);
+	long nUlongSize = (long)sizeof(sal_uInt32);
 	String aTmpURL;
 
 	rStm >> nFormat;
--- svtools/source/items1/poolio.cxx	2006-10-12 17:22:39.000000000 +0200
+++ svtools/source/items1/poolio.cxx	2007-02-22 19:09:40.000000000 +0100
@@ -788,7 +788,7 @@ SvStream &SfxItemPool::Load1_Impl(SvStre
 	if ( pImp->nMajorVer > 1 || pImp->nMinorVer >= 3 )
 	{
 		// Version-Map finden (letztes ULONG der Size-Table gibt Pos an)
-		rStream.Seek( nEndOfSizes - sizeof(ULONG) );
+		rStream.Seek( nEndOfSizes - sizeof(sal_uInt32) );
 		sal_uInt32 nVersionMapPos;
 		rStream >> nVersionMapPos;
 		rStream.Seek( nVersionMapPos );
--- svtools/source/numbers/numhead.cxx	2007-02-22 13:59:11.000000000 +0100
+++ svtools/source/numbers/numhead.cxx	2007-02-22 19:09:23.000000000 +0100
@@ -106,7 +106,7 @@ SvNumWriteHeader::~SvNumWriteHeader()
 	if ( nPos - nDataPos != nDataSize )				// Default getroffen?
 	{
 		nDataSize = nPos - nDataPos;
-		rStream.Seek(nDataPos - sizeof(ULONG));
+		rStream.Seek(nDataPos - sizeof(sal_uInt32));
 		rStream << nDataSize;						// Groesse am Anfang eintragen
 		rStream.Seek(nPos);
 	}
