Index: sd/source/ui/dlg/filedlg.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/dlg/filedlg.cxx,v
retrieving revision 1.13
diff -u -p -u -r1.13 filedlg.cxx
--- sd/source/ui/dlg/filedlg.cxx	9 Sep 2005 04:00:47 -0000	1.13
+++ sd/source/ui/dlg/filedlg.cxx	1 Nov 2005 12:42:01 -0000
@@ -75,9 +75,6 @@
 #include <com/sun/star/ui/dialogs/XFilePicker.hpp>
 #endif
 
-#ifndef _SOUND_HXX //autogen
-#include <vcl/sound.hxx>
-#endif
 #ifndef _SV_MSGBOX_HXX
 #include <vcl/msgbox.hxx>
 #endif
@@ -126,7 +123,6 @@ private:
 
 	css::uno::Reference< css::ui::dialogs::XFilePickerControlAccess > 	mxControlAccess;
 
-	Sound						maSound;
     ULONG                       mnPlaySoundEvent;
 	BOOL						mbUsableSelection;
 	BOOL						mbLabelPlaying;
@@ -180,11 +176,11 @@ IMPL_LINK( SdFileDialog_Imp, PlayMusicHd
 	if( mbLabelPlaying )
 	{
         // switch from playing to not playing
-
+#if 0
 		// reset, so that sound file gets unlocked
         maSound.Stop();
 		maSound.SetSoundName( String() );
-		
+#endif		
         try
         {
             mxControlAccess->setLabel( css::ui::dialogs::ExtendedFilePickerElementIds::PUSHBUTTON_PLAY, 
@@ -201,6 +197,7 @@ IMPL_LINK( SdFileDialog_Imp, PlayMusicHd
     }
 	else
 	{
+#if 0
         // switch from not playing to playing of current file
 		if( maSound.IsPlaying() )
 		{
@@ -208,12 +205,13 @@ IMPL_LINK( SdFileDialog_Imp, PlayMusicHd
 			maSound.Stop();
             maSound.SetSoundName( String() );
 		}
-
+#endif
         INetURLObject	aUrl( GetPath() );
         String			aSoundFile( aUrl.GetMainURL( INetURLObject::NO_DECODE ) );
 		
         if( aSoundFile.Len() > 0 && Sound::IsSoundFile(aSoundFile) )
         {
+#if 0
             maSound.SetNotifyHdl( LINK( this, SdFileDialog_Imp, StopMusicHdl ) );
 			mbDuringPreparePlaying=TRUE;
             maSound.SetSoundName( aSoundFile );
@@ -243,6 +241,7 @@ IMPL_LINK( SdFileDialog_Imp, PlayMusicHd
 				//reset error state of sound
 				maSound.SetSoundName( String() );
 			}
+#endif
 		}
 	}
 
@@ -257,8 +256,10 @@ IMPL_LINK( SdFileDialog_Imp, StopMusicHd
 
 	 ::vos::OGuard aGuard( Application::GetSolarMutex() );
 
+#if 0
 	// reset, so that sound file gets unlocked
 	maSound.SetSoundName( String() );
+#endif
 	
 	if( mxControlAccess.is() )
 	{
Index: sd/source/ui/view/drviewse.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/view/drviewse.cxx,v
retrieving revision 1.55
diff -u -p -u -r1.55 drviewse.cxx
--- sd/source/ui/view/drviewse.cxx	9 Sep 2005 07:12:08 -0000	1.55
+++ sd/source/ui/view/drviewse.cxx	1 Nov 2005 12:45:23 -0000
@@ -122,9 +122,6 @@
 #ifndef _URLOBJ_HXX //autogen
 #include <tools/urlobj.hxx>
 #endif
-#ifndef _SV_SOUND_HXX
-#include <vcl/sound.hxx>
-#endif
 
 // #UndoRedo#
 #ifndef _SFXSLSTITM_HXX
@@ -1717,12 +1714,14 @@ void DrawViewShell::InsertURLButton(cons
 			form::FormButtonType eButtonType = form::FormButtonType_URL;
 			aTmp <<= eButtonType;
 			xPropSet->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "ButtonType" )), aTmp );
+#if 0
 			if ( Sound::IsSoundFile( rURL ) )
 			{
 				// #105638# OJ
 				aTmp <<= sal_True;
 				xPropSet->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "DispatchURLInternal" )), aTmp );
 			}
+#endif
 	    }
     }
 
@@ -1757,12 +1756,14 @@ void DrawViewShell::InsertURLButton(cons
 		form::FormButtonType eButtonType = form::FormButtonType_URL;
 		aTmp <<= eButtonType;
 		xPropSet->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "ButtonType" )), aTmp );
+#if 0
 		// #105638# OJ
 		if ( Sound::IsSoundFile( rURL ) )
 		{
 			aTmp <<= sal_True;
 			xPropSet->setPropertyValue( rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "DispatchURLInternal" )), aTmp );
 		}
+#endif
 
         Point aPos;
 
