? smoketestoo_native/OpenOffice
? smoketestoo_native/bootstraprc
Index: smoketestoo_native/smoketest.pl
===================================================================
RCS file: /cvs/installation/smoketestoo_native/smoketest.pl,v
retrieving revision 1.18
diff -u -p -u -r1.18 smoketest.pl
--- smoketestoo_native/smoketest.pl	14 Aug 2006 09:32:50 -0000	1.18
+++ smoketestoo_native/smoketest.pl	20 Dec 2006 13:55:59 -0000
@@ -51,8 +51,8 @@ use Getopt::Long;
 #   Globale Variablen   #
 #                       #
 #########################
-$is_debug = 0;
-$is_command_infos = 0;
+$is_debug = 0;           # run without executing commands
+$is_command_infos = 0;   # print command details before exec
 $is_protocol_test = 0;
 $is_remove_on_error = 0;
 $is_remove_at_end = 1;
@@ -167,17 +167,17 @@ else {
 	);
 }
 
-@error_messages = (	'',
+@error_messages = (	'', #0
 			'',
 			'Error during installation!',
 			'Error: patching configuration failed!',
 			'Error: starting office failed!',
-			'Error during testing',
+			'Error during testing', #5
 			'',
 			'Error in setup log',
 			'installationsset is not complete',
 			'can not copy all basic scripts',
-			'can not patch bottstrapini',
+			'can not patch bootstrapini', #10
 			'msiexec failed. Maybe you have got an installed version',
 			'deinstallation is incomplete'
 );
@@ -546,7 +546,31 @@ sub doInstall {
 		chomp $system;
 		$mach = `uname -m`;
 		chomp $mach;
-		if ( (defined($system)) && ($system eq "Linux") ) {
+		if ($ENV{EPM} eq 'NO') { # do the install ourselves ...
+# FIXME - this tool should work nicely without such hacks
+# cut/paste from ooo-build/bin/ooinstall
+		    my $instoo_dir = "$ENV{SOLARROOT}/instsetoo_native";
+
+		    if ( $ENV{'SYSTEM_MOZILLA'} eq 'YES' ) {
+			$ENV{'LD_LIBRARY_PATH'} = "$ENV{'MOZ_LIB'}:$ENV{'LD_LIBRARY_PATH'}";
+		    }
+		    $ENV{'PYTHONPATH'} = "$ENV{SOLARROOT}/instsetoo_native/$ENV{INPATH}/bin:$ENV{SOLARVER}/$ENV{UPD}/$ENV{INPATH}/lib";
+		    $ENV{OUT} = "../$ENV{INPATH}";
+		    $ENV{LOCAL_OUT} = "../$ENV{INPATH}";
+		    $ENV{LOCAL_COMMON_OUT} = "../$ENV{INPATH}";
+		    my $sane_destdir = $dest_installdir;
+		    $sane_destdir .= "oootest";
+		    createPath ($sane_destdir, $error_setup);  
+		    $Command = "cd $instoo_dir/util ; perl -w $ENV{SOLARENV}/bin/make_installer.pl " .
+			"-f openoffice.lst -l en-US -p OpenOffice " .
+			"-packagelist ../inc_openoffice/unix/packagelist.txt " .
+			"-addpackagelist ../inc_openoffice/unix/packagelist_language.txt " .
+			"-buildid \"smoketestoo\" -simple $sane_destdir";
+# FIXME - this tool should work nicely without such evil
+		    execute_Command ($Command, $error_setup, $show_Message, $command_normal);
+		    return "$sane_destdir/"
+
+		} elsif ( (defined($system)) && ($system eq "Linux") ) {
 			$installsetpath .= "RPMS$PathSeparator";
 			$optdir = "$dest_installdir" . "opt" . $PathSeparator; 
 			$rpmdir = "$dest_installdir" . "rpm" . $PathSeparator; 
@@ -658,6 +682,11 @@ sub getInstset {
 	my ($INSTSET, $NEWINSTSET);
 	my (@DirArray, $InstDir, $RootDir, $TestDir1, $TestDir2);
 	print "get Instset\n" if $is_debug;
+
+	if ($ENV{EPM} eq 'NO') { # we do the install ourselves ...
+	    return ();
+	}
+
 	$NEWINSTSET = "";
 	if (defined($smoketest_install)) {
 		my $mask = "\\" . $PathSeparator . "\$";
