--- solenv/inc/unxlngppc.mk.old	2007-08-23 20:15:01.000000000 +0200
+++ solenv/inc/unxlngppc.mk	2007-08-27 12:12:19.000000000 +0200
@@ -83,9 +83,8 @@ LINKFLAGS += -Wl,--hash-style=gnu
 LINKFLAGS += -Wl,-zdynsort
 .ENDIF
 
-# source code is still not signed versus unsigned char clean 
-CFLAGS=-fsigned-char -nostdinc -c
-CFLAGSCC=-fsigned-char $(ARCH_FLAGS)
+CFLAGS=-nostdinc -c
+CFLAGSCC=$(ARCH_FLAGS)
 
 # Flags for enabling exception handling
 CFLAGSEXCEPTIONS=-fexceptions 
--- solenv/inc/unxmacx.mk.old	2007-08-03 15:55:43.000000000 +0200
+++ solenv/inc/unxmacx.mk	2007-08-27 14:36:43.000000000 +0200
@@ -88,13 +88,13 @@ CXX*=g++
 CC*=gcc
 objc=cc
 
-CFLAGS=-fsigned-char -fmessage-length=0 -malign-natural -c $(EXTRA_CFLAGS)
+CFLAGS=-fmessage-length=0 -malign-natural -c $(EXTRA_CFLAGS)
 
 # ---------------------------------
 #  Compilation flags
 # ---------------------------------
 # Normal C compilation flags
-CFLAGSCC=-pipe -fsigned-char -malign-natural $(ARCH_FLAGS)
+CFLAGSCC=-pipe -malign-natural $(ARCH_FLAGS)
 
 # Normal Objective C compilation flags
 #OBJCFLAGS=-no-precomp
@@ -108,7 +108,7 @@ CFLAGSEXCEPTIONS=-fexceptions -fno-enfor
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # Normal C++ compilation flags
-CFLAGSCXX=-pipe -malign-natural -fsigned-char -Wno-long-double $(ARCH_FLAGS)
+CFLAGSCXX=-pipe -malign-natural -Wno-long-double $(ARCH_FLAGS)
 CFLAGSCXX+= -Wno-ctor-dtor-privacy
 PICSWITCH:=-fPIC
 # Other flags
--- solenv/inc/unxlngppc4.mk.old	2007-07-24 11:14:04.000000000 +0200
+++ solenv/inc/unxlngppc4.mk	2007-08-27 15:56:25.000000000 +0200
@@ -70,10 +70,10 @@ CXX*=g++
 # name of C Compiler
 CC*=gcc
 
-CFLAGS+=-fsigned-char -fmessage-length=0 -c
+CFLAGS+=-fmessage-length=0 -c
 
 # flags for the C++ Compiler
-CFLAGSCC= -fsigned-char -pipe $(ARCH_FLAGS)
+CFLAGSCC= -pipe $(ARCH_FLAGS)
 
 # Flags for enabling exception handling
 CFLAGSEXCEPTIONS=-fexceptions -fno-enforce-eh-specs
@@ -81,7 +81,7 @@ CFLAGSEXCEPTIONS=-fexceptions -fno-enfor
 # Flags for disabling exception handling
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
-CFLAGSCXX= -fsigned-char -pipe -frtti $(ARCH_FLAGS)
+CFLAGSCXX= -pipe -frtti $(ARCH_FLAGS)
 PICSWITCH:=-fPIC
 
 # Compiler flags for compiling static object in multi threaded environment with graphical user interface
