--- sot/source/unoolestorage/xolesimplestorage.hxx.ark	2006-04-30 22:14:56.000000000 +0200
+++ sot/source/unoolestorage/xolesimplestorage.hxx	2006-04-30 22:18:13.000000000 +0200
@@ -103,7 +103,7 @@ class OLESimpleStorage	: public ::cppu::
 	static void InsertInputStreamToStorage_Impl( BaseStorage* pStorage, ::rtl::OUString aName, const ::com::sun::star::uno::Reference< ::com::sun::star::io::XInputStream >& xInputStream )
 	throw ( ::com::sun::star::uno::Exception );
 
-	static void OLESimpleStorage::InsertNameAccessToStorage_Impl( BaseStorage* pStorage, ::rtl::OUString aName, const ::com::sun::star::uno::Reference< ::com::sun::star::container::XNameAccess >& xNameAccess )
+	static void InsertNameAccessToStorage_Impl( BaseStorage* pStorage, ::rtl::OUString aName, const ::com::sun::star::uno::Reference< ::com::sun::star::container::XNameAccess >& xNameAccess )
 	throw ( ::com::sun::star::uno::Exception );
 
 public:
