Index: svx/source/dialog/optmemory.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/optmemory.cxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 optmemory.cxx
--- svx/source/dialog/optmemory.cxx	8 Sep 2005 21:47:15 -0000	1.8
+++ svx/source/dialog/optmemory.cxx	11 Nov 2005 16:04:28 -0000
@@ -237,8 +237,7 @@ OfaMemoryOptionsPage::OfaMemoryOptionsPa
 {
 	FreeResource();
 
-    //quick launch only available in Win
-#ifndef WNT
+#if !defined(WNT) && !defined(ENABLE_GTK)
     aQuickLaunchFL.Hide();
     aQuickLaunchCB.Hide();
 #endif

--- /dev/null	2005-11-11 09:08:10.800476000 +0000
+++ sysui/desktop/menus/qstart.desktop	2005-11-11 16:37:17.000000000 +0000
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=1.0
+Encoding=UTF-8
+Terminal=false
+Type=Application
+Categories=Application;Office
+Exec=ooffice -quickstart -nologo -nodefault
+NoDisplay=true
+Name=%PRODUCTNAME Quickstarter
+Comment=Hook for quickstarter startup
Index: sfx2/util/makefile.mk
===================================================================
RCS file: /cvs/framework/sfx2/util/makefile.mk,v
retrieving revision 1.39
diff -u -p -u -r1.39 makefile.mk
--- sfx2/util/makefile.mk	7 Sep 2005 19:33:33 -0000	1.39
+++ sfx2/util/makefile.mk	11 Nov 2005 17:03:08 -0000
@@ -148,6 +148,25 @@ RESLIB1NAME=$(TARGET)
 RESLIB1IMAGES=$(PRJ)$/res
 RESLIB1SRSFILES=$(SFXSRSLIST)
 
+# gtk quick-starter
+.IF "$(ENABLE_GTK)" != ""
+PKGCONFIG_MODULES=gtk+-2.0
+.INCLUDE: pkg_config.mk
+CFLAGS+=$(PKGCONFIG_CFLAGS)
+
+SHL3TARGET=qstart_gtk$(UPD)$(DLLPOSTFIX)
+SHL3LIBS=$(SLB)$/quickstart.lib
+SHL3DEPN=$(SHL1IMPLIBN) $(SHL1TARGETN)
+# libs for gtk plugin
+SHL3STDLIBS=$(SHL1STDLIBS) $(SFX2LIB) $(EGGTRAYLIB)
+SHL3STDLIBS+=$(PKGCONFIG_LIBS:s/ -lpangoxft-1.0//)
+# hack for faked SO environment
+.IF "$(PKGCONFIG_ROOT)"!=""
+SHL3SONAME+=-z nodefs
+SHL3NOCHECK=TRUE
+.ENDIF          # "$(PKGCONFIG_ROOT)"!=""
+.ENDIF # "$(ENABLE_GTK)" != ""
+
 # --- Targets ------------------------------------------------------
 
 
Index: sfx2/source/appl/makefile.mk
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/makefile.mk,v
retrieving revision 1.39
diff -u -p -u -r1.39 makefile.mk
--- sfx2/source/appl/makefile.mk	7 Sep 2005 17:43:35 -0000	1.39
+++ sfx2/source/appl/makefile.mk	11 Nov 2005 17:03:49 -0000
@@ -38,6 +38,7 @@ PRJ=..$/..
 PRJNAME=sfx2
 TARGET=appl
 ENABLE_EXCEPTIONS=TRUE
+LIBTARGET=NO
 
 # --- Settings -----------------------------------------------------
 
@@ -47,10 +48,18 @@ ENABLE_EXCEPTIONS=TRUE
 # w.g. compilerbugs
 .IF "$(GUI)"=="WNT"
 CFLAGS+=-Od
+CFLAGS+=-DENABLE_QUICKSTART_APPLET
 .ENDIF
 
 .IF "$(GUI)"=="UNX"
         CDEFS+=-DDLL_NAME=libsfx$(UPD)$(DLLPOSTFIX)$(DLLPOST)
+.IF "$(ENABLE_GTK)" != ""
+        PKGCONFIG_MODULES=gtk+-2.0
+        .INCLUDE: pkg_config.mk
+        CFLAGS+=$(PKGCONFIG_CFLAGS)
+        CFLAGS+=-DENABLE_QUICKSTART_APPLET
+        CDEFS+=-DPLUGIN_NAME=libqstart_gtk$(UPD)$(DLLPOSTFIX)$(DLLPOST)
+.ENDIF # "$(ENABLE_QUICKSTART_APPLET)"=="TRUE"
 .ELSE
         CDEFS+=-DDLL_NAME=sfx$(UPD)$(DLLPOSTFIX)$(DLLPOST)
 .ENDIF
@@ -65,7 +74,7 @@ SRS2NAME=sfx
 SRC2FILES =  \
 		sfx.src
 
-SLOFILES =  \
+SFX_OBJECTS = \
 	$(SLO)$/imagemgr.obj\
 	$(SLO)$/appuno.obj \
 	$(SLO)$/appmail.obj \
@@ -105,6 +114,17 @@ SLOFILES =  \
     $(SLO)$/lnkbase2.obj\
     $(SLO)$/impldde.obj
 
+QUICKSTART_OBJECTS = \
+	$(SLO)$/shutdowniconunx.obj
+
+SLOFILES = $(SFX_OBJECTS) $(QUICKSTART_OBJECTS)
+LIB1TARGET= $(SLB)$/$(TARGET).lib
+LIB1OBJFILES= $(SFX_OBJECTS)
+
+LIB2TARGET= $(SLB)$/quickstart.lib
+LIB2OBJFILES= $(QUICKSTART_OBJECTS)
+
+
 EXCEPTIONSFILES=\
 	$(SLO)$/imagemgr.obj		\
 	$(SLO)$/appopen.obj \
Index: sfx2/source/appl/shutdownicon.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/shutdownicon.cxx,v
retrieving revision 1.42
diff -u -p -u -r1.42 shutdownicon.cxx
--- sfx2/source/appl/shutdownicon.cxx	7 Sep 2005 17:48:37 -0000	1.42
+++ sfx2/source/appl/shutdownicon.cxx	11 Nov 2005 17:03:50 -0000
@@ -102,7 +102,19 @@
 #ifndef _URLOBJ_HXX
 #include <tools/urlobj.hxx>
 #endif
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
+#endif
+#ifndef _OSL_FILE_HXX_
+#include <osl/file.hxx>
+#endif
+#ifndef _UTL_BOOTSTRAP_HXX
+#include <unotools/bootstrap.hxx>
+#endif
 
+#ifdef UNX // need symlink
+#include <unistd.h>
+#endif
 #include "sfxresid.hxx"
 
 using namespace ::com::sun::star::uno;
@@ -126,9 +138,7 @@ public:
 
 void SAL_CALL SfxNotificationListener_Impl::dispatchFinished( const DispatchResultEvent& aEvent ) throw( RuntimeException )
 {
-#ifdef WNT
 	ShutdownIcon::LeaveModalMode();
-#endif
 }
 
 void SAL_CALL SfxNotificationListener_Impl::disposing( const EventObject& aEvent ) throw( RuntimeException )
@@ -138,62 +148,84 @@ void SAL_CALL SfxNotificationListener_Im
 SFX_IMPL_XSERVICEINFO( ShutdownIcon, "com.sun.star.office.Quickstart", "com.sun.star.comp.desktop.QuickstartWrapper" )	\
 SFX_IMPL_ONEINSTANCEFACTORY( ShutdownIcon );
 
+bool ShutdownIcon::bModalMode = false;
 ShutdownIcon* ShutdownIcon::pShutdownIcon = 0;
 
-ShutdownIcon::ShutdownIcon( Reference< XMultiServiceFactory > aSMgr ) :
-	ShutdownIconServiceBase( m_aMutex ),
-	m_bVeto ( false ),
-	m_pResMgr( 0 ),
-	m_xServiceManager( aSMgr )
-{
+// To remove conditionals
+extern "C" {
+	static void disabled_initSystray()
+	{
+		// disable shutdown
+		ShutdownIcon::getInstance()->SetVeto( true );
+		ShutdownIcon::getInstance()->addTerminateListener();
+	}
+	static void disabled_deInitSystray() { }
 }
+#define DOSTRING( x )			   			#x
+#define STRING( x )				   			DOSTRING( x )
 
-ShutdownIcon::~ShutdownIcon()
+void ShutdownIcon::initSystray()
 {
-#ifdef WNT
-    deInitSystray();
-#endif
+	if (!m_pInitSystray)
+	{
+#ifdef ENABLE_QUICKSTART_APPLET
+#  ifdef WIN32
+		m_pInitSystray = win32_init_sys_tray;
+		m_pDeInitSystray = win32_shutdown_sys_tray;
+#  else // UNX
+		m_pPlugin = new osl::Module();
+		if ( m_pPlugin->load( OUString (RTL_CONSTASCII_USTRINGPARAM( STRING( PLUGIN_NAME ) ) ) ) )
+		{
+			m_pInitSystray = (void (*)()) m_pPlugin->getSymbol(
+					OUString( RTL_CONSTASCII_USTRINGPARAM( "plugin_init_sys_tray" ) ) );
+			m_pDeInitSystray = (void (*)()) m_pPlugin->getSymbol(
+					OUString( RTL_CONSTASCII_USTRINGPARAM( "plugin_shutdown_sys_tray" ) ) );
+			OSL_ASSERT (m_pInitSystray && m_pDeInitSystray);
+		}
+		else
+		{
+			delete m_pPlugin;
+			m_pPlugin = 0;
+		}
+	}
+#  endif // UNX
+#endif // ENABLE_QUICKSTART_APPLET
+	if (!m_pInitSystray ||
+		!m_pDeInitSystray)
+	{
+		m_pInitSystray = disabled_initSystray;
+		m_pDeInitSystray = disabled_deInitSystray;
+	}
+	m_pInitSystray();
 }
 
-// ---------------------------------------------------------------------------
-
-void ShutdownIcon::SetAutostart( bool bActivate )
+void ShutdownIcon::deInitSystray()
 {
-#ifdef WNT
-    OUString aShortcutName( RTL_CONSTASCII_USTRINGPARAM( "StarOffice 6.0" ) );
-	ResMgr* pMgr = SfxResId::GetResMgr();
-    if( pMgr )
-    {
-        ::vos::OGuard aGuard( Application::GetSolarMutex() );
-        UniString aRes( SfxResId( STR_QUICKSTART_LNKNAME ) );
-        aShortcutName = OUString( aRes );
-    }
+	m_pDeInitSystray();
 
-    aShortcutName += OUString( RTL_CONSTASCII_USTRINGPARAM( ".lnk" ) );
-
-    SetAutostartW32( aShortcutName, bActivate );
-#endif
+	m_pInitSystray = 0;
+	m_pDeInitSystray = 0;
+	if (m_pPlugin)
+		delete m_pPlugin;
+	m_pPlugin = 0;
 }
 
-bool ShutdownIcon::GetAutostart( )
-{
-#ifdef WNT
-    OUString aShortcutName( RTL_CONSTASCII_USTRINGPARAM( "StarOffice 6.0" ) );
-	ResMgr* pMgr = SfxResId::GetResMgr();
-    if( pMgr )
-    {
-        ::vos::OGuard aGuard( Application::GetSolarMutex() );
-        UniString aRes( SfxResId( STR_QUICKSTART_LNKNAME ) );
-        aShortcutName = OUString( aRes );
-    }
-    aShortcutName += OUString( RTL_CONSTASCII_USTRINGPARAM( ".lnk" ) );
 
-    return GetAutostartW32( aShortcutName );
-#else
-    return false;
-#endif
+ShutdownIcon::ShutdownIcon( Reference< XMultiServiceFactory > aSMgr ) :
+	ShutdownIconServiceBase( m_aMutex ),
+	m_bVeto ( false ),
+	m_pResMgr( 0 ),
+	m_xServiceManager( aSMgr ),
+	m_pPlugin( 0 ),
+	m_pInitSystray( 0 ),
+	m_pDeInitSystray( 0 )
+{
 }
 
+ShutdownIcon::~ShutdownIcon()
+{
+	deInitSystray();
+}
 
 // ---------------------------------------------------------------------------
 
@@ -231,9 +263,7 @@ void ShutdownIcon::FileOpen()
     if ( getInstance() && getInstance()->m_xDesktop.is() )
     {
         ::vos::OGuard aGuard( Application::GetSolarMutex() );
-#ifdef WNT
 		EnterModalMode();
-#endif
 		// use ctor for filling up filters automatically! #89169#
 		FileDialogHelper dlg( WB_OPEN | SFXWB_MULTISELECTION, String() );
         if ( ERRCODE_NONE == dlg.Execute() )
@@ -357,9 +387,7 @@ void ShutdownIcon::FileOpen()
 			{
 			}
 		}
-#ifdef WNT
 		LeaveModalMode();
-#endif
     }
 }
 
@@ -395,9 +423,7 @@ void ShutdownIcon::FromTemplate()
             Reference< ::com::sun::star::frame::XNotifyingDispatch > xNotifyer( xDisp, UNO_QUERY );
             if ( xNotifyer.is() )
 			{
-#ifdef WNT
 				EnterModalMode();
-#endif
                 xNotifyer->dispatchWithNotification( aTargetURL, aArgs, new SfxNotificationListener_Impl() );
 			}
             else
@@ -544,10 +570,9 @@ void SAL_CALL ShutdownIcon::initialize( 
 				if ( !m_xDesktop.is() )
 					return;
 
+				/* Create a sub-classed instance - foo */
 				ShutdownIcon::pShutdownIcon = this;
-#ifdef WNT
 				initSystray();
-#endif
 			}
 			catch(const ::com::sun::star::lang::IllegalArgumentException&)
 			{
@@ -565,3 +590,178 @@ void SAL_CALL ShutdownIcon::initialize( 
     }
 
 }
+
+// -------------------------------
+
+void ShutdownIcon::EnterModalMode()
+{
+	bModalMode = TRUE;
+}
+
+// -------------------------------
+
+void ShutdownIcon::LeaveModalMode()
+{
+	bModalMode = FALSE;
+}
+
+// ------------------
+// install/uninstall
+
+bool ShutdownIcon::IsQuickstarterInstalled()
+{
+#ifndef ENABLE_QUICKSTART_APPLET
+	return false;
+#endif
+#ifdef UNX
+	return true;
+#endif
+#ifdef WNT
+	bool	bQuickstarterInstalled = false;
+	HKEY	hKey;
+
+	if ( ERROR_SUCCESS == RegOpenKeyEx( HKEY_CURRENT_USER, TEXT("Software\\OpenOffice.org"), 0, KEY_READ, &hKey ) )
+	{
+		if ( ERROR_SUCCESS == RegQueryValueEx( hKey, TEXT("QuickStarterInstalled"),  NULL, NULL, NULL, NULL ) )
+			bQuickstarterInstalled = true;
+
+		RegCloseKey( hKey );
+	}
+
+	if ( !bQuickstarterInstalled )
+	{
+		if ( ERROR_SUCCESS == RegOpenKeyEx( HKEY_LOCAL_MACHINE, TEXT("Software\\OpenOffice.org"), 0, KEY_READ, &hKey ) )
+		{
+			if ( ERROR_SUCCESS == RegQueryValueEx( hKey, TEXT("QuickStarterInstalled"),  NULL, NULL, NULL, NULL ) )
+				bQuickstarterInstalled = true;
+
+			RegCloseKey( hKey );
+		}
+	}
+
+	return bQuickstarterInstalled;
+#endif
+}
+
+// ---------------------------------------------------------------------------
+
+rtl::OUString ShutdownIcon::getShortcutName()
+{
+#ifndef ENABLE_QUICKSTART_APPLET
+	return OUString();
+#else
+
+    OUString aShortcutName( RTL_CONSTASCII_USTRINGPARAM( "StarOffice 6.0" ) );
+	ResMgr* pMgr = SfxResId::GetResMgr();
+    if( pMgr )
+    {
+        ::vos::OGuard aGuard( Application::GetSolarMutex() );
+        UniString aRes( SfxResId( STR_QUICKSTART_LNKNAME ) );
+        aShortcutName = OUString( aRes );
+    }
+#ifdef WNT
+    aShortcutName += OUString( RTL_CONSTASCII_USTRINGPARAM( ".lnk" ) );
+
+	OUString aShortcut(SHGetAutostartFolderName());
+	aShortcut += OUString( RTL_CONSTASCII_USTRINGPARAM( "\\" ) );
+	aShortcut += aShortcutName;
+#else // UNX
+	OUString aHomeURL;
+	osl::Security().getHomeDir( aHomeURL );
+	OUString aShortcut;
+	::osl::File::getSystemPathFromFileURL( aHomeURL, aShortcut );
+	aShortcut += OUString( RTL_CONSTASCII_USTRINGPARAM( "/.autostart/ooqstart.desktop" ) );
+#endif // UNX
+	return aShortcut;
+#endif // ENABLE_QUICKSTART_APPLET
+}
+
+bool ShutdownIcon::GetAutostart( )
+{
+	bool bRet = false;
+#ifdef ENABLE_QUICKSTART_APPLET
+	OUString aShortcut( getShortcutName() );
+	OUString aShortcutUrl;
+	osl::File::getFileURLFromSystemPath( aShortcut, aShortcutUrl );
+	osl::File f( aShortcutUrl );
+	osl::File::RC error = f.open( OpenFlag_Read );
+	if( error == osl::File::E_None )
+	{
+		f.close();
+		bRet = true;
+	}
+#endif // ENABLE_QUICKSTART_APPLET
+    return bRet;
+}
+
+#if defined (ENABLE_QUICKSTART_APPLET) && defined (UNX)
+static void ensureDotAutostart ()
+{
+	OUString aDir;
+	osl::Security().getHomeDir( aDir );
+	aDir += OUString( RTL_CONSTASCII_USTRINGPARAM( "/.autostart" ) );
+	osl::Directory::create( aDir );
+}
+#endif
+
+void ShutdownIcon::SetAutostart( bool bActivate )
+{
+#ifdef ENABLE_QUICKSTART_APPLET
+	OUString aShortcut( getShortcutName() );
+
+    if( bActivate && IsQuickstarterInstalled() )
+    {
+#ifdef WNT
+        wchar_t aPath[_MAX_PATH];
+        if( isNT() )
+        {
+            GetModuleFileNameW( NULL, aPath, _MAX_PATH-1);
+        }
+        else
+        {
+            char szPathA[_MAX_PATH];
+            int len = GetModuleFileNameA( NULL, szPathA, _MAX_PATH-1);
+
+		    // calc the string wcstr len
+		    int nNeededWStrBuffSize = MultiByteToWideChar( CP_ACP, 0, szPathA, -1, NULL, 0 );
+
+		    // copy the string if necessary
+		    if ( nNeededWStrBuffSize > 0 )
+			    MultiByteToWideChar( CP_ACP, 0, szPathA, -1, aPath, nNeededWStrBuffSize );
+	    }
+
+        OUString aOfficepath( aPath );
+        int i = aOfficepath.lastIndexOf((sal_Char) '\\');
+        if( i != -1 )
+            aOfficepath = aOfficepath.copy(0, i);
+
+        OUString quickstartExe(aOfficepath);
+        quickstartExe += OUString( RTL_CONSTASCII_USTRINGPARAM( "\\quickstart.exe" ) );
+
+        CreateShortcut( quickstartExe, aOfficepath, aShortcut, OUString(), OUString() );
+#else // UNX
+		ensureDotAutostart();
+
+		OUString aPath;
+		::utl::Bootstrap::locateBaseInstallation(aPath);
+
+		OUString aDesktopFile;
+		::osl::File::getSystemPathFromFileURL( aPath, aDesktopFile );
+		aDesktopFile += OUString( RTL_CONSTASCII_USTRINGPARAM( "/share/xdg/qstart.desktop" ) );
+
+		OString aDesktopFileUnx = OUStringToOString( aDesktopFile,
+													 osl_getThreadTextEncoding() );
+		OString aShortcutUnx = OUStringToOString( aShortcut,
+												  osl_getThreadTextEncoding() );
+		symlink( aDesktopFileUnx, aShortcutUnx );
+#endif // UNX
+    }
+    else
+    {
+        OUString aShortcutUrl;
+        ::osl::File::getFileURLFromSystemPath( aShortcut, aShortcutUrl );
+        ::osl::File::remove( aShortcutUrl );
+    }
+#endif // ENABLE_QUICKSTART_APPLET
+}
+
Index: sfx2/source/appl/shutdownicon.hxx
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/shutdownicon.hxx,v
retrieving revision 1.18
diff -u -p -u -r1.18 shutdownicon.hxx
--- sfx2/source/appl/shutdownicon.hxx	30 Mar 2005 08:41:34 -0000	1.18
+++ sfx2/source/appl/shutdownicon.hxx	11 Nov 2005 17:03:50 -0000
@@ -35,12 +35,18 @@
 #ifndef _OSL_MUTEX_HXX_
 #include <osl/mutex.hxx>
 #endif
+#ifndef _OSL_MODULE_HXX_
+#include <osl/module.hxx>
+#endif
 #ifndef _SFX_SFXUNO_HXX
 #include <sfxuno.hxx>
 #endif
 #ifndef _CPPUHELPER_COMPBASE3_HXX_
 #include <cppuhelper/compbase3.hxx>
 #endif
+#ifndef INCLUDED_SFX2_DLLAPI_H
+#include <sfx2/dllapi.h>
+#endif
 
 class ResMgr;
 
@@ -49,25 +55,38 @@ typedef ::cppu::WeakComponentImplHelper3
 	::com::sun::star::frame::XTerminateListener,
 	::com::sun::star::lang::XServiceInfo > ShutdownIconServiceBase;
 
-class ShutdownIcon :	public ShutdownIconServiceBase
+#if defined(USE_APP_SHORTCUTS)
+#define WRITER_URL      "private:factory/swriter"
+#define CALC_URL        "private:factory/scalc"
+#define IMPRESS_URL     "private:factory/simpress"
+#define IMPRESS_WIZARD_URL     "private:factory/simpress?slot=10425"
+#define DRAW_URL        "private:factory/sdraw"
+#define MATH_URL        "private:factory/smath"
+#define BASE_URL        "private:factory/sdatabase?Interactive"
+#endif
+
+class SFX2_DLLPUBLIC ShutdownIcon :	public ShutdownIconServiceBase
 {
 		::osl::Mutex	m_aMutex;
         bool            m_bVeto;
         ResMgr          *m_pResMgr;
 		::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory > m_xServiceManager;
 
+		static bool bModalMode;
         static ShutdownIcon *pShutdownIcon; // one instance
 
-#ifdef WNT
-        void initSystray();
-        void deInitSystray();
-        static void SetAutostartW32( const ::rtl::OUString& aShortcutName, bool bActivate );
-        static bool GetAutostartW32( const ::rtl::OUString& aShortcutName );
+		void (*m_pInitSystray)   ();
+		void (*m_pDeInitSystray) ();
+		::osl::Module  *m_pPlugin;
+
+		void initSystray();
+		void deInitSystray();
+
 		static void EnterModalMode();
 		static void LeaveModalMode();
+		static rtl::OUString getShortcutName();
 
 		friend class SfxNotificationListener_Impl;
-#endif
 
 	public:
 		ShutdownIcon( ::com::sun::star::uno::Reference< ::com::sun::star::lang::XMultiServiceFactory > aSMgr );
@@ -99,9 +118,7 @@ class ShutdownIcon :	public ShutdownIcon
         void SetVeto( bool bVeto )  { m_bVeto = bVeto;}
         bool GetVeto()              { return m_bVeto; }
 
-#ifdef WNT
         static bool IsQuickstarterInstalled();
-#endif
 
 		// Component Helper - force override
 		virtual void SAL_CALL disposing();
@@ -123,4 +140,15 @@ class ShutdownIcon :	public ShutdownIcon
 		::com::sun::star::uno::Reference< ::com::sun::star::frame::XDesktop > m_xDesktop;
 };
 
+extern "C" {
+#  ifdef WNT
+	// builtin win32 systray
+	void win32_init_sys_tray();
+	void win32_shutdown_sys_tray();
+#  endif
+	// external plugin systray impl.
+	void plugin_init_sys_tray();
+	void plugin_shutdown_sys_tray();
+}
+
 #endif
Index: sfx2/source/appl/shutdowniconw32.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/appl/shutdowniconw32.cxx,v
retrieving revision 1.33
diff -u -p -u -r1.33 shutdowniconw32.cxx
--- sfx2/source/appl/shutdowniconw32.cxx	7 Sep 2005 17:48:58 -0000	1.33
+++ sfx2/source/appl/shutdowniconw32.cxx	11 Nov 2005 17:03:50 -0000
@@ -55,7 +55,6 @@
 #include <stdio.h>
 #include <io.h>
 #include <osl/thread.h>
-#include <osl/file.hxx>
 #include <setup_native/qswin32.h>
 
 #ifndef _COMPHELPER_PROCESSFACTORY_HXX_
@@ -102,16 +101,6 @@ using namespace ::osl;
 #define IDM_UNINSTALL               11
 
 
-#if defined(USE_APP_SHORTCUTS)
-#define WRITER_URL      "private:factory/swriter"
-#define CALC_URL        "private:factory/scalc"
-#define IMPRESS_URL     "private:factory/simpress"
-#define IMPRESS_WIZARD_URL     "private:factory/simpress?slot=10425"
-#define DRAW_URL        "private:factory/sdraw"
-#define MATH_URL        "private:factory/smath"
-#define BASE_URL        "private:factory/sdatabase?Interactive"
-#endif
-
 #define ICON_SO_DEFAULT 				1
 #define ICON_TEXT_DOCUMENT				2
 #define ICON_TEXT_TEMPLATE				3
@@ -419,10 +408,10 @@ LRESULT CALLBACK listenerWndProc( HWND h
                     // update status before showing menu, could have been changed from option page
                     CheckMenuItem( popupMenu, IDM_INSTALL, MF_BYCOMMAND| (ShutdownIcon::GetAutostart() ? MF_CHECKED : MF_UNCHECKED) );
 
-					EnableMenuItem( popupMenu, IDM_EXIT, MF_BYCOMMAND | (bModalMode ? MF_GRAYED : MF_ENABLED) );
+					EnableMenuItem( popupMenu, IDM_EXIT, MF_BYCOMMAND | (ShutdownIcon::bModalMode ? MF_GRAYED : MF_ENABLED) );
 #if defined(USE_APP_SHORTCUTS)
-					EnableMenuItem( popupMenu, IDM_OPEN, MF_BYCOMMAND | (bModalMode ? MF_GRAYED : MF_ENABLED) );
-					EnableMenuItem( popupMenu, IDM_TEMPLATE, MF_BYCOMMAND | (bModalMode ? MF_GRAYED : MF_ENABLED) );
+					EnableMenuItem( popupMenu, IDM_OPEN, MF_BYCOMMAND | (ShutdownIcon::bModalMode ? MF_GRAYED : MF_ENABLED) );
+					EnableMenuItem( popupMenu, IDM_TEMPLATE, MF_BYCOMMAND | (ShutdownIcon::bModalMode ? MF_GRAYED : MF_ENABLED) );
 #endif
                     int m = TrackPopupMenuEx( popupMenu, TPM_RETURNCMD|TPM_LEFTALIGN|TPM_RIGHTBUTTON,
                                               pt.x, pt.y, hWnd, NULL );
@@ -517,7 +506,7 @@ LRESULT CALLBACK executerWndProc( HWND h
             {
 #if defined(USE_APP_SHORTCUTS)
                 case IDM_OPEN:
-					if ( !bModalMode && checkOEM() )
+					if ( !ShutdownIcon::bModalMode && checkOEM() )
 						ShutdownIcon::FileOpen();
                 break;
                 case IDM_WRITER:
@@ -541,7 +530,7 @@ LRESULT CALLBACK executerWndProc( HWND h
                     ShutdownIcon::OpenURL( OUString( RTL_CONSTASCII_USTRINGPARAM( BASE_URL ) ), OUString( RTL_CONSTASCII_USTRINGPARAM( "_default" ) ) );
                 break;
                 case IDM_TEMPLATE:
-					if ( !bModalMode && checkOEM())
+					if ( !ShutdownIcon::bModalMode && checkOEM())
 						ShutdownIcon::FromTemplate();
                 break;
 #endif
@@ -551,7 +540,7 @@ LRESULT CALLBACK executerWndProc( HWND h
                 case IDM_EXIT:
                     // remove listener and
                     //  terminate office if running in background
-					if ( !bModalMode )
+					if ( !ShutdownIcon::bModalMode )
 						ShutdownIcon::terminateDesktop();
                     break;
             }
@@ -595,9 +584,9 @@ DWORD WINAPI SystrayThread( LPVOID lpPar
 
 // -------------------------------
 
-void ShutdownIcon::initSystray()
+void win32_init_sys_tray()
 {
-	if ( IsQuickstarterInstalled() )
+	if ( ShutdownIcon::IsQuickstarterInstalled() )
 	{
 		WNDCLASSEXA listenerClass;
 		listenerClass.cbSize		= sizeof(WNDCLASSEX);
@@ -652,9 +641,9 @@ void ShutdownIcon::initSystray()
 
 // -------------------------------
 
-void ShutdownIcon::deInitSystray()
+void win32_shutdown_sys_tray()
 {
-	if ( IsQuickstarterInstalled() )
+	if ( ShutdownIcon::IsQuickstarterInstalled() )
 	{
 		if( IsWindow( aListenerWindow ) )
 		{
@@ -669,19 +658,6 @@ void ShutdownIcon::deInitSystray()
 }
 
 
-// -------------------------------
-
-void ShutdownIcon::EnterModalMode()
-{
-	bModalMode = TRUE;
-}
-
-// -------------------------------
-
-void ShutdownIcon::LeaveModalMode()
-{
-	bModalMode = FALSE;
-}
 
 // -------------------------------
 
@@ -886,102 +862,6 @@ BOOL CreateShortcut( const OUString& rAb
 	return TRUE;
 }
 
-// ------------------
-// install/uninstall
-
-bool ShutdownIcon::IsQuickstarterInstalled()
-{
-	bool	bQuickstarterInstalled = false;
-	HKEY	hKey;
-
-	if ( ERROR_SUCCESS == RegOpenKeyEx( HKEY_CURRENT_USER, TEXT("Software\\OpenOffice.org"), 0, KEY_READ, &hKey ) )
-	{
-		if ( ERROR_SUCCESS == RegQueryValueEx( hKey, TEXT("QuickStarterInstalled"),  NULL, NULL, NULL, NULL ) )
-		{
-			bQuickstarterInstalled = true;
-		}
-
-		RegCloseKey( hKey );
-	}
-
-	if ( !bQuickstarterInstalled )
-	{
-		if ( ERROR_SUCCESS == RegOpenKeyEx( HKEY_LOCAL_MACHINE, TEXT("Software\\OpenOffice.org"), 0, KEY_READ, &hKey ) )
-		{
-			if ( ERROR_SUCCESS == RegQueryValueEx( hKey, TEXT("QuickStarterInstalled"),  NULL, NULL, NULL, NULL ) )
-			{
-				bQuickstarterInstalled = true;
-			}
-
-			RegCloseKey( hKey );
-		}
-	}
-
-	return bQuickstarterInstalled;
-}
-
-void ShutdownIcon::SetAutostartW32( const OUString& aShortcutName, bool bActivate )
-{
-    OUString aShortcut(SHGetAutostartFolderName());
-    aShortcut += OUString( RTL_CONSTASCII_USTRINGPARAM( "\\" ) );
-    aShortcut += aShortcutName;
-
-    if( bActivate && IsQuickstarterInstalled() )
-    {
-        wchar_t aPath[_MAX_PATH];
-        if( isNT() )
-        {
-            GetModuleFileNameW( NULL, aPath, _MAX_PATH-1);
-        }
-        else
-        {
-            char szPathA[_MAX_PATH];
-            int len = GetModuleFileNameA( NULL, szPathA, _MAX_PATH-1);
-
-		    // calc the string wcstr len
-		    int nNeededWStrBuffSize = MultiByteToWideChar( CP_ACP, 0, szPathA, -1, NULL, 0 );
-
-		    // copy the string if necessary
-		    if ( nNeededWStrBuffSize > 0 )
-			    MultiByteToWideChar( CP_ACP, 0, szPathA, -1, aPath, nNeededWStrBuffSize );
-	    }
-
-        OUString aOfficepath( aPath );
-        int i = aOfficepath.lastIndexOf((sal_Char) '\\');
-        if( i != -1 )
-            aOfficepath = aOfficepath.copy(0, i);
-
-        OUString quickstartExe(aOfficepath);
-        quickstartExe += OUString( RTL_CONSTASCII_USTRINGPARAM( "\\quickstart.exe" ) );
-
-        CreateShortcut( quickstartExe, aOfficepath, aShortcut, OUString(), OUString() );
-    }
-    else
-    {
-        OUString aShortcutUrl;
-        ::osl::File::getFileURLFromSystemPath( aShortcut, aShortcutUrl );
-        ::osl::File::remove( aShortcutUrl );
-    }
-}
-
-bool ShutdownIcon::GetAutostartW32( const OUString& aShortcutName )
-{
-	OUString aShortcut(SHGetAutostartFolderName());
-	aShortcut += OUString( RTL_CONSTASCII_USTRINGPARAM( "\\" ) );
-	aShortcut += aShortcutName;
-
-	OUString aShortcutUrl;
-	File::getFileURLFromSystemPath( aShortcut, aShortcutUrl );
-	File f( aShortcutUrl );
-	File::RC error = f.open( OpenFlag_Read );
-	if( error == File::E_None )
-	{
-		f.close();
-		return true;
-	}
-	else
-		return false;
-}
 #endif // WNT
 
 
--- /dev/null	2005-11-11 09:08:10.800476000 +0000
+++ sfx2/source/appl/shutdowniconunx.cxx	2005-11-11 16:42:22.000000000 +0000
@@ -0,0 +1,334 @@
+
+#ifdef ENABLE_QUICKSTART_APPLET
+
+#define OU2A(rtlOUString)  (::rtl::OUStringToOString((rtlOUString), RTL_TEXTENCODING_ASCII_US).getStr())
+
+#include <gtk/gtk.h>
+#include <glib.h>
+#include <eggtray/eggtrayicon.h>
+
+#ifndef _SV_BITMAPEX_HXX
+#include <vcl/bitmapex.hxx>
+#endif
+#ifndef _SV_BMPACC_HXX
+#include <vcl/bmpacc.hxx>
+#endif
+#ifndef _SV_SVIDS_HRC
+#include <vcl/svids.hrc>
+#endif
+#ifndef _SFXAPP_HXX
+#include "app.hxx"
+#endif
+#ifndef _SFX_APP_HRC
+#include "app.hrc"
+#endif
+#ifndef __SHUTDOWNICON_HXX__
+#define USE_APP_SHORTCUTS
+#include "shutdownicon.hxx"
+#endif
+
+using namespace ::rtl;
+using namespace ::osl;
+
+static bool bModalMode = FALSE;
+
+EggTrayIcon *tray_icon;
+GtkWidget *button, *icon_image, *menu;
+ResMgr *pVCLResMgr;
+USHORT nResMgrOffset;
+
+GdkPixbuf *writer_pixbuf, *calc_pixbuf, *impress_pixbuf, *draw_pixbuf, *template_pixbuf;
+OUString writer_label, calc_label, impress_label, draw_label;
+OUString template_label, fileopen_label, prelaunch_label, exit_label;
+OUString aWriterURL, aCalcURL, aImpressURL, aDrawURL;
+
+static void open_url_cb( GtkWidget *button, gpointer *data )
+{
+	ShutdownIcon::OpenURL( *(OUString *)data, OUString( RTL_CONSTASCII_USTRINGPARAM( "_default" ) ) );
+}
+
+static void open_file_cb( GtkWidget *button )
+{
+	if ( !bModalMode )
+		ShutdownIcon::FileOpen();
+}
+
+static void open_template_cb( GtkWidget *button )
+{
+	if ( !bModalMode )
+		ShutdownIcon::FromTemplate();
+}
+
+static void prelaunch_toggled_cb( GtkWidget *button )
+{
+	ShutdownIcon::SetAutostart( !ShutdownIcon::GetAutostart() );
+}
+
+static void exit_quickstarter_cb( GtkWidget *button, EggTrayIcon *icon )
+{
+	egg_tray_icon_cancel_message (icon, 1 );
+
+	ShutdownIcon::getInstance()->terminateDesktop();
+}
+
+static void reset_toggle_cb( GtkWidget *menu )
+{
+	gtk_toggle_button_set_active( GTK_TOGGLE_BUTTON( button ), FALSE );
+
+	gtk_menu_popdown( GTK_MENU( menu ) );
+}
+
+static void remove_menu_item( gpointer child, gpointer menu )
+{
+	gtk_container_remove( GTK_CONTAINER( menu ), GTK_WIDGET( child ) );
+}
+
+static void populate_menu( void )
+{
+	GtkWidget *menuitem, *image;
+
+	g_list_foreach( GTK_MENU_SHELL( menu)->children,
+			remove_menu_item, menu );
+
+	image = gtk_image_new_from_pixbuf( writer_pixbuf );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( writer_label ) );
+	gtk_image_menu_item_set_image( ( GtkImageMenuItem * ) menuitem, image );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+	g_signal_connect( menuitem, "activate", G_CALLBACK( open_url_cb ), &aWriterURL );
+
+	image = gtk_image_new_from_pixbuf( calc_pixbuf );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( calc_label ) );
+	gtk_image_menu_item_set_image( ( GtkImageMenuItem * ) menuitem, image );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+	g_signal_connect( menuitem, "activate", G_CALLBACK( open_url_cb ), &aCalcURL );
+
+	image = gtk_image_new_from_pixbuf( impress_pixbuf );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( impress_label ) );
+	gtk_image_menu_item_set_image( ( GtkImageMenuItem * ) menuitem, image );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+	g_signal_connect( menuitem, "activate", G_CALLBACK( open_url_cb ), &aImpressURL );
+
+	image = gtk_image_new_from_pixbuf( draw_pixbuf );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( draw_label ) );
+	gtk_image_menu_item_set_image( ( GtkImageMenuItem * ) menuitem, image );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem);
+	g_signal_connect( menuitem, "activate", G_CALLBACK( open_url_cb ), &aDrawURL );
+
+	image = gtk_image_new_from_pixbuf( template_pixbuf );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( template_label ) );
+	gtk_image_menu_item_set_image( ( GtkImageMenuItem * ) menuitem, image );
+	gtk_menu_shell_append(GTK_MENU_SHELL( menu ), menuitem );
+	gtk_widget_set_sensitive( menuitem, bModalMode ? FALSE : TRUE );
+	g_signal_connect( menuitem, "activate", G_CALLBACK( open_template_cb ), NULL );
+
+	menuitem = gtk_separator_menu_item_new();
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+
+	image = gtk_image_new_from_stock( GTK_STOCK_OPEN, GTK_ICON_SIZE_MENU );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( fileopen_label ) );
+	gtk_image_menu_item_set_image( ( GtkImageMenuItem * ) menuitem, image );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+	gtk_widget_set_sensitive( menuitem, bModalMode ? FALSE : TRUE );
+	g_signal_connect( menuitem, "activate", G_CALLBACK( open_file_cb ), NULL );
+
+	menuitem = gtk_separator_menu_item_new();
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+
+	menuitem = gtk_check_menu_item_new_with_label( OU2A( prelaunch_label ) );
+	gtk_check_menu_item_set_active( ( GtkCheckMenuItem * ) menuitem,
+		ShutdownIcon::GetAutostart() );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem);
+	g_signal_connect( menuitem, "activate", G_CALLBACK( prelaunch_toggled_cb ), NULL );
+
+	menuitem = gtk_separator_menu_item_new();
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+
+	image = gtk_image_new_from_stock( GTK_STOCK_QUIT,  GTK_ICON_SIZE_MENU );
+	menuitem = gtk_image_menu_item_new_with_label( OU2A( exit_label ) );
+	gtk_image_menu_item_set_image( GTK_IMAGE_MENU_ITEM( menuitem ), image );
+	gtk_menu_shell_append( GTK_MENU_SHELL( menu ), menuitem );
+	gtk_widget_set_sensitive( menuitem, bModalMode ? FALSE : TRUE );
+	g_signal_connect( menuitem, "activate", G_CALLBACK( exit_quickstarter_cb ), tray_icon );
+
+	gtk_widget_show_all( menu );
+}
+
+static void display_menu_cb( GtkWidget *button )
+{
+	populate_menu();
+
+	gtk_menu_popup( GTK_MENU( menu ), NULL, NULL,
+		NULL, NULL, 0, gtk_get_current_event_time() );
+}
+
+static inline BYTE * getRow( BitmapReadAccess *pBuffer, ULONG nRow )
+{
+    if( BMP_SCANLINE_ADJUSTMENT( pBuffer->GetScanlineFormat() ) == BMP_FORMAT_TOP_DOWN )
+        return pBuffer->GetBuffer() + nRow * pBuffer->GetScanlineSize();
+    else
+        return pBuffer->GetBuffer() + ( pBuffer->Height() - nRow - 1 ) * pBuffer->GetScanlineSize();
+}
+
+static GdkPixbuf * ResIdToPixbuf( USHORT nResId )
+{
+	ResId aResId( nResMgrOffset + nResId, pVCLResMgr );
+	BitmapEx aIcon( aResId );
+	Bitmap pInSalBitmap = aIcon.GetBitmap();
+	AlphaMask pInSalAlpha = aIcon.GetAlpha();
+
+	BitmapReadAccess* pSalBitmap = pInSalBitmap.AcquireReadAccess();
+	BitmapReadAccess* pSalAlpha = pInSalAlpha.AcquireReadAccess();
+
+	g_return_val_if_fail( pSalBitmap != NULL, NULL );
+	g_return_val_if_fail( pSalAlpha != NULL, NULL );
+
+	g_return_val_if_fail( pSalBitmap->GetBitCount() == 24, NULL );
+	g_return_val_if_fail( pSalAlpha->GetBitCount() == 8, NULL );
+
+	Size aSize( pSalBitmap->Width(), pSalBitmap->Height() );
+	g_return_val_if_fail( Size( pSalAlpha->Width(), pSalAlpha->Height() ) == aSize, NULL );
+
+#warning FIXME - make this safer & more generic ...
+
+	int nX, nY;
+	guchar *pPixbufData = ( guchar * )g_malloc( 4 * aSize.Width() * aSize.Height() );
+	guchar *pDestData = pPixbufData;
+
+	for( nY = 0; nY < pSalBitmap->Height(); nY++ )
+	{
+		BYTE *pData = getRow( pSalBitmap, nY );
+		BYTE *pAlphaData = getRow( pSalAlpha, nY );
+
+		for( nX = 0; nX < pSalBitmap->Width(); nX++ )
+		{
+			if( pSalBitmap->GetScanlineFormat() == BMP_FORMAT_24BIT_TC_BGR )
+			{
+				pDestData[2] = *pData++;
+				pDestData[1] = *pData++;
+				pDestData[0] = *pData++;
+			}
+			else // BMP_FORMAT_24BIT_TC_RGB
+			{
+				pDestData[0] = *pData++;
+				pDestData[1] = *pData++;
+				pDestData[2] = *pData++;
+			}
+			pDestData += 3;
+			*pDestData++ = 255 - *pAlphaData++;
+		}
+	}
+
+	if( pSalBitmap )
+		pInSalBitmap.ReleaseAccess( pSalBitmap );
+	if( pSalAlpha )
+		pInSalAlpha.ReleaseAccess( pSalAlpha );
+
+	return gdk_pixbuf_new_from_data( pPixbufData,
+		GDK_COLORSPACE_RGB, TRUE, 8,
+		aSize.Width(), aSize.Height(),
+		aSize.Width() * 4,
+		(GdkPixbufDestroyNotify) g_free,
+		NULL );
+}
+
+extern "C" {
+	static gboolean
+	show_at_idle (gpointer tray_icon)
+	{
+		gtk_widget_show_all( GTK_WIDGET( tray_icon ) );
+		return FALSE;
+	}
+}
+
+void SAL_DLLPUBLIC_EXPORT plugin_init_sys_tray()
+{
+	if( !g_type_from_name( "GdkDisplay" ) )
+		return;
+
+	ShutdownIcon *pShutdownIcon = ShutdownIcon::getInstance();
+
+	tray_icon = egg_tray_icon_new(
+		OU2A( pShutdownIcon->GetResString( STR_QUICKSTART_TIP ) ) );
+
+	button = gtk_toggle_button_new();
+	GtkTooltips *tooltips = gtk_tooltips_new();
+	gtk_tooltips_set_tip( GTK_TOOLTIPS( tooltips ), button, 
+		OU2A( pShutdownIcon->GetResString( STR_QUICKSTART_TIP ) ), NULL );
+
+	icon_image = gtk_image_new();
+	gtk_container_add( GTK_CONTAINER( button ), icon_image );
+
+	// Use high contrast icons where appropriate
+	if( Application::GetSettings().GetStyleSettings().GetFaceColor().IsDark() )
+		nResMgrOffset = SV_ICON_SMALL_HC_START;
+	else
+		nResMgrOffset = SV_ICON_SMALL_START;
+
+	pVCLResMgr = CREATEVERSIONRESMGR( vcl );
+
+	GdkPixbuf *buffer = ResIdToPixbuf( SV_ICON_ID_OFFICE ); 
+	gtk_image_set_from_pixbuf( GTK_IMAGE( icon_image ), buffer );
+	g_object_unref( buffer );
+
+	writer_pixbuf	= ResIdToPixbuf( SV_ICON_ID_TEXT );
+	calc_pixbuf		= ResIdToPixbuf( SV_ICON_ID_SPREADSHEET );
+	impress_pixbuf	= ResIdToPixbuf( SV_ICON_ID_PRESENTATION );
+	draw_pixbuf		= ResIdToPixbuf( SV_ICON_ID_DRAWING );
+	template_pixbuf	= ResIdToPixbuf( SV_ICON_ID_TEMPLATE );
+
+	writer_label	= pShutdownIcon->GetUrlDescription(
+		OUString( RTL_CONSTASCII_USTRINGPARAM( WRITER_URL ) ) );
+	calc_label		= pShutdownIcon->GetUrlDescription(
+		OUString( RTL_CONSTASCII_USTRINGPARAM( CALC_URL ) ) );
+	impress_label	= pShutdownIcon->GetUrlDescription(
+		OUString( RTL_CONSTASCII_USTRINGPARAM( IMPRESS_URL ) ) );
+	draw_label		= pShutdownIcon->GetUrlDescription(
+		OUString( RTL_CONSTASCII_USTRINGPARAM( DRAW_URL ) ) );
+
+	template_label	= pShutdownIcon->GetResString( STR_QUICKSTART_FROMTEMPLATE );
+	fileopen_label	= pShutdownIcon->GetResString( STR_QUICKSTART_FILEOPEN );
+	prelaunch_label	= pShutdownIcon->GetResString( STR_QUICKSTART_PRELAUNCH );
+	exit_label		= pShutdownIcon->GetResString( STR_QUICKSTART_EXIT );
+
+	aWriterURL	= OUString( RTL_CONSTASCII_USTRINGPARAM( WRITER_URL ) ) ;
+	aCalcURL	= OUString( RTL_CONSTASCII_USTRINGPARAM( CALC_URL ) ) ;
+	aImpressURL	= OUString( RTL_CONSTASCII_USTRINGPARAM( IMPRESS_URL ) ) ;
+	aDrawURL	= OUString( RTL_CONSTASCII_USTRINGPARAM( DRAW_URL ) ) ;
+
+	g_signal_connect( button, "clicked",
+		G_CALLBACK( display_menu_cb ), tray_icon );
+
+	gtk_container_add( GTK_CONTAINER( tray_icon ), button );
+	g_idle_add (show_at_idle, (gpointer) tray_icon);
+
+	menu = gtk_menu_new();
+	g_signal_connect (menu, "deactivate",
+		G_CALLBACK (reset_toggle_cb), NULL);
+
+	populate_menu();
+
+	// disable shutdown
+	pShutdownIcon->SetVeto( true );
+	pShutdownIcon->addTerminateListener();
+}
+
+void SAL_DLLPUBLIC_EXPORT plugin_shutdown_sys_tray()
+{
+	if( !tray_icon )
+		return;
+
+	g_object_unref( writer_pixbuf );
+	g_object_unref( calc_pixbuf );
+	g_object_unref( impress_pixbuf );
+	g_object_unref( draw_pixbuf );
+	g_object_unref( template_pixbuf );
+
+	g_list_foreach( GTK_MENU_SHELL( menu)->children,
+			remove_menu_item, menu );
+	gtk_widget_destroy( menu );
+	gtk_widget_destroy( icon_image );
+	gtk_widget_destroy( button );
+	gtk_widget_destroy( GTK_WIDGET( tray_icon ) );
+}
+
+#endif // ENABLE_QUICKSTART_APPLET
Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.156.16.1
diff -u -p -u -r1.156.16.1 file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	9 Nov 2005 15:29:24 -0000	1.156.16.1
+++ scp2/source/ooo/file_library_ooo.scp	11 Nov 2005 16:49:42 -0000
@@ -417,6 +417,24 @@ End
 
 #endif
 
+#ifdef UNX
+#ifdef ENABLE_GTK
+File gid_File_Lib_Egg_Tray
+ TXT_FILE_BODY;
+ Styles = (PACKED);
+ Dir = gid_Dir_Program;
+ Name = STRING(CONCAT4(libeggtray,OFFICEUPD,DLLSUFFIX,UNXSUFFIX));
+End
+
+File gid_File_Lib_QStart_Gtk
+    Name = LIBNAME(qstart_gtk);
+    TXT_FILE_BODY;
+    Styles = (PACKED,PATCH);
+    Dir = gid_Dir_Program;
+End
+#endif
+#endif
+
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Dba, dba)
 
 STD_UNO_LIB_FILE( gid_File_Lib_Dbacfg, dbacfg)
Index: scp2/source/ooo/file_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_ooo.scp,v
retrieving revision 1.119
diff -u -p -u -r1.119 file_ooo.scp
--- scp2/source/ooo/file_ooo.scp	27 Oct 2005 14:08:26 -0000	1.119
+++ scp2/source/ooo/file_ooo.scp	11 Nov 2005 16:50:31 -0000
@@ -412,6 +412,15 @@ File gid_File_Share_Xdg_Printeradmin
     Name = "printeradmin.desktop";
 End
 
+#ifdef ENABLE_GTK
+File gid_File_Share_Xdg_QStart
+    TXT_FILE_BODY;
+    Dir = gid_Dir_Share_Xdg;
+    Styles = (PACKED,SCPZIP_REPLACE);
+    Name = "qstart.desktop";
+End
+#endif
+
 #endif
 
 File gid_File_Bmp_Installation
Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.139
diff -u -p -u -r1.139 configure.in
--- config_office/configure.in	25 Oct 2005 11:23:57 -0000	1.139
+++ config_office/configure.in	11 Nov 2005 17:12:25 -0000
@@ -3794,6 +3959,7 @@ if test  "$test_gtk" = "yes"; then
         PKG_CHECK_MODULES(GDKPIXBUF, gdk-pixbuf-xlib-2.0 >= 2.2)
         GTK_CFLAGS="$GTK_CFLAGS $GDKPIXBUF_CFLAGS"
         GTK_LIBS="$GTK_LIBS $GDKPIXBUF_LIBS"
+	BUILD_TYPE="$BUILD_TYPE GTK"
     fi
 
 fi


Index: sfx2/prj/build.lst
===================================================================
RCS file: /cvs/framework/sfx2/prj/build.lst,v
retrieving revision 1.9
diff -u -p -u -r1.9 build.lst
--- sfx2/prj/build.lst	21 Jun 2005 09:45:59 -0000	1.9
+++ sfx2/prj/build.lst	11 Nov 2005 17:15:04 -0000
@@ -1,4 +1,4 @@
-sf      sfx2    :       idl basic xmlscript framework shell setup_native NULL
+sf      sfx2    :       idl basic xmlscript framework shell setup_native GTK:eggtray NULL
 sf	sfx2									usr1	-	all	sf_mkout NULL
 sf	sfx2\inc								get		-	all	sf_inc NULL
 sf	sfx2\prj								get		-	all	sf_prj NULL


