Index: solenv/inc/settings.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/settings.mk,v
retrieving revision 1.185.36.1
diff -u -p -u -r1.185.36.1 settings.mk
--- solenv/inc/settings.mk	10 Feb 2006 12:08:47 -0000	1.185.36.1
+++ solenv/inc/settings.mk	23 Feb 2006 12:30:59 -0000
@@ -932,15 +932,6 @@ LIB:=$(LB);$(SLB);$(ILIB)
 CPUNAME=CPUNAME_HAS_TO_BE_SET_IN_ENVIRONMENT
 .ENDIF
 
-.IF "$(GUI)"=="UNX" || "$(USE_SHELL)"!="4nt"
-DATESTRING:=$(shell date +%d%m%Y)
-.ELSE			# "$(GUI)"=="UNX" || "$(USE_SHELL)"!="4nt"
-#can't override 4nt internal date to use date.exe :-(
-DATESTRING:=$(shell +echo %@IF[%@LEN[%_DAY%]==1,0%_DAY%,%_DAY%]%@IF[%@LEN[%_MONTH%]==1,0%_MONTH%,%_MONTH%]%_YEAR%)
-.ENDIF			# "$(GUI)"=="UNX" || "$(USE_SHELL)"!="4nt"
-SCPDEFS+=-DSCP_DATESTRING=$(DATESTRING)
-
-
 .IF "$(USE_STLP_DEBUG)" != ""
 SCPCDEFS+=-D_STLP_DEBUG
 .ENDIF
