? unotools/unotools.vpj
Index: unotools/inc/unotools/textsearch.hxx
===================================================================
RCS file: /cvs/util/unotools/inc/unotools/textsearch.hxx,v
retrieving revision 1.8
diff -u -r1.8 textsearch.hxx
--- unotools/inc/unotools/textsearch.hxx	19 Jun 2006 14:02:47 -0000	1.8
+++ unotools/inc/unotools/textsearch.hxx	20 Aug 2007 19:43:39 -0000
@@ -148,8 +148,8 @@
 
 class UNOTOOLS_DLLPUBLIC TextSearch
 {
-	com::sun::star::uno::Reference < com::sun::star::util::XTextSearch >
-			xTextSearch;
+    static ::com::sun::star::uno::Reference< ::com::sun::star::util::XTextSearch > 
+            sxTextSearch;
 
 	void Init( const SearchParam & rParam,
 			   const ::com::sun::star::lang::Locale& rLocale );
Index: unotools/source/i18n/textsearch.cxx
===================================================================
RCS file: /cvs/util/unotools/source/i18n/textsearch.cxx,v
retrieving revision 1.13
diff -u -r1.13 textsearch.cxx
--- unotools/source/i18n/textsearch.cxx	26 Mar 2007 12:46:42 -0000	1.13
+++ unotools/source/i18n/textsearch.cxx	20 Aug 2007 19:43:39 -0000
@@ -119,6 +119,8 @@
 //  ( Die Unterscheidung der Gross/Klein-Schreibung kann mit einen Flag
 //  unterdrueckt werden )
 
+Reference<XTextSearch> TextSearch::sxTextSearch = 0;
+
 TextSearch::TextSearch(const SearchParam & rParam, LanguageType eLang )
 {
 	if( LANGUAGE_NONE == eLang )
@@ -138,11 +140,14 @@
 {
 	try
 	{
-		Reference< XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
-		xTextSearch = Reference< XTextSearch > ( xMSF->createInstance(
-			::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(
-                        "com.sun.star.util.TextSearch" ) ) ), UNO_QUERY_THROW );
-		xTextSearch->setOptions( rPara );
+        if ( !sxTextSearch.is() )
+        {
+            Reference< XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
+            sxTextSearch.set( xMSF->createInstance(
+                ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(
+                            "com.sun.star.util.TextSearch" ) ) ), UNO_QUERY_THROW );
+        }
+        sxTextSearch->setOptions( rPara );
 	}
 	catch ( Exception& )
 	{
@@ -193,11 +198,14 @@
 
 	try
 	{
-		Reference< XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
-		xTextSearch = Reference< XTextSearch > ( xMSF->createInstance(
-			::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(
-                        "com.sun.star.util.TextSearch" ) ) ), UNO_QUERY_THROW );
-		xTextSearch->setOptions( aSOpt );
+        if ( !sxTextSearch.is() )
+        {
+            Reference< XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
+            sxTextSearch.set ( xMSF->createInstance(
+                ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(
+                            "com.sun.star.util.TextSearch" ) ) ), UNO_QUERY_THROW );
+        }
+        sxTextSearch->setOptions( aSOpt );
 	}
 	catch ( Exception& )
 	{
@@ -214,11 +222,14 @@
 
     try
     {
-        Reference< XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
-        xTextSearch = Reference< XTextSearch > ( xMSF->createInstance(
-            ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(
-                        "com.sun.star.util.TextSearch" ) ) ), UNO_QUERY_THROW );
-        xTextSearch->setOptions( aSOpt );
+        if ( !sxTextSearch.is() )
+        {
+            Reference< XMultiServiceFactory > xMSF = ::comphelper::getProcessServiceFactory();
+            sxTextSearch.set ( xMSF->createInstance(
+                ::rtl::OUString( RTL_CONSTASCII_USTRINGPARAM(
+                            "com.sun.star.util.TextSearch" ) ) ), UNO_QUERY_THROW );
+        }
+        sxTextSearch->setOptions( aSOpt );
     }
     catch ( Exception& )
     {
@@ -245,10 +256,9 @@
 	int nRet = 0;
 	try
 	{
-		if( xTextSearch.is() )
-		{
-			SearchResult aRet( xTextSearch->searchForward(
-													rStr, *pStart, *pEnde ));
+        if ( sxTextSearch.is() )
+        {
+            SearchResult aRet( sxTextSearch->searchForward(rStr, *pStart, *pEnde) );
 			if( 1 == aRet.subRegExpressions )
 			{
 				nRet = 1;
@@ -276,10 +286,9 @@
 	int nRet = 0;
 	try
 	{
-		if( xTextSearch.is() )
-		{
-			SearchResult aRet( xTextSearch->searchBackward(
-													rStr, *pStart, *pEnde ));
+        if ( sxTextSearch.is() )
+        {
+            SearchResult aRet( sxTextSearch->searchBackward(rStr, *pStart, *pEnde) );
 			if( aRet.subRegExpressions )
 			{
 				nRet = 1;
