Index: vcl/source/app/svmain.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/app/svmain.cxx,v
retrieving revision 1.57
diff -u -r1.57 svmain.cxx
--- vcl/source/app/svmain.cxx	13 Oct 2005 09:36:37 -0000	1.57
+++ vcl/source/app/svmain.cxx	31 Oct 2005 14:50:18 -0000
@@ -317,6 +317,11 @@
     vos::OStartupInfo   aStartInfo;
     rtl::OUString       aExeFileName;
 
+	// Initialize application instance (should be done after initialization of VCL SAL part)
+    if( pSVData->mpApp )
+        // call init to initialize application class
+        // soffice/sfx implementation creates the global service manager
+        pSVData->mpApp->Init();
 
     // Sal initialisieren
     RTL_LOGFILE_CONTEXT_TRACE( aLog, "{ ::CreateSalInstance" );
@@ -325,12 +330,6 @@
         return FALSE;
     RTL_LOGFILE_CONTEXT_TRACE( aLog, "} ::CreateSalInstance" );
 
-	// Initialize application instance (should be done after initialization of VCL SAL part)
-    if( pSVData->mpApp )
-        // call init to initialize application class
-        // soffice/sfx implementation creates the global service manager
-        pSVData->mpApp->Init();
-
     // Den AppFileName gleich holen und absolut machen, bevor das
     // WorkingDirectory sich aendert...
     aStartInfo.getExecutableFile( aExeFileName );
