Index: desktop/source/app/officeipcthread.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/officeipcthread.cxx,v
retrieving revision 1.50
diff -u -p -u -r1.50 officeipcthread.cxx
--- desktop/source/app/officeipcthread.cxx	19 Oct 2005 12:20:26 -0000	1.50
+++ desktop/source/app/officeipcthread.cxx	3 Jan 2006 17:44:28 -0000
@@ -54,9 +54,12 @@
 #ifndef _UTL_CONFIGMGR_HXX_
 #include <unotools/configmgr.hxx>
 #endif
-#ifndef _THREAD_HXX_
+#ifndef _OSL_THREAD_HXX_
 #include <osl/thread.hxx>
 #endif
+#ifndef _OSL_FILE_HXX_
+#include <osl/file.hxx>
+#endif
 #ifndef _RTL_DIGEST_H_
 #include <rtl/digest.h>
 #endif
@@ -361,7 +364,23 @@ OfficeIPCThread::Status OfficeIPCThread:
 		aUserInstallPathHashCode = OUString( szBuffer, strlen(szBuffer), osl_getThreadTextEncoding() );
     }
 	else
+	{
+#ifdef UNX			
+		// Since we mangle the UID in anyway, the home path adds nothing
+		// but the need to parse bootstraprc => use sys install path instead.
+		rtl::OUString aPath;
+		aInfo.getExecutableFile( aPath );
+		lastIndex = aPath.lastIndexOf('/');
+		if ( lastIndex > 0 )
+			aPath = aPath.copy( 0, lastIndex+1 );
+
+		osl::File::getSystemPathFromFileURL( aPath, aUserInstallPathHashCode);
+		aUserInstallPathHashCode = aUserInstallPathHashCode.replace( '/', '_' );
+#else
 		aUserInstallPathHashCode = CreateMD5FromString( aDummy );
+#endif
+	}
 	
 
 	// Check result to create a hash code from the user install path

