Index: sw/source/ui/dialog/SwSpellDialogChildWindow.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/dialog/SwSpellDialogChildWindow.cxx,v
retrieving revision 1.11.130.1
diff -u -p -r1.11.130.1 SwSpellDialogChildWindow.cxx
--- sw/source/ui/dialog/SwSpellDialogChildWindow.cxx	17 Jan 2008 16:08:49 -0000	1.11.130.1
+++ sw/source/ui/dialog/SwSpellDialogChildWindow.cxx	10 Apr 2008 08:17:11 -0000
@@ -162,6 +162,7 @@ struct SpellState
     SpellState() :
         m_bInitialCall(true),
         m_bLockFocus(false),
+        m_bLostFocus(false),
         m_SpellStartPosition(SPELL_START_BODY),
         m_bBodySpelled(false),
         m_bOtherSpelled(false),
