--- filter/source/config/fragments/fcfg_calc.mk.old	2007-07-17 21:12:07.000000000 +0200
+++ filter/source/config/fragments/fcfg_calc.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,27 +1,5 @@
 all_fragments+=calc
 
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_CALC_BINFILTER = \
-	calc_StarCalc_10 \
-	calc_StarCalc_30 \
-	calc_StarCalc_30_VorlageTemplate \
-	calc_StarCalc_40 \
-	calc_StarCalc_40_VorlageTemplate \
-	calc_StarCalc_50 \
-	calc_StarCalc_50_VorlageTemplate
-F4_CALC_BINFILTER = \
-	StarCalc_1_0 \
-	StarCalc_3_0 \
-	StarCalc_3_0_Vorlage_Template \
-	StarCalc_4_0 \
-	StarCalc_4_0_Vorlage_Template \
-	StarCalc_5_0 \
-	StarCalc_5_0_Vorlage_Template
-.ELSE
-T4_CALC_BINFILTER =
-F4_CALC_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
 # count = 27
 T4_CALC = \
@@ -39,7 +17,6 @@ T4_CALC = \
 	calc_MS_Excel_97_VorlageTemplate \
 	writer_Rich_Text_Format \
 	calc_SYLK \
-	$(T4_CALC_BINFILTER) \
 	calc_StarOffice_XML_Calc \
 	calc_Text_txt_csv_StarCalc \
 	calc_StarOffice_XML_Calc_Template \
@@ -68,7 +45,6 @@ F4_CALC = \
 	MS_Excel_97_Vorlage_Template \
 	Rich_Text_Format__StarCalc_ \
 	SYLK \
-	$(F4_CALC_BINFILTER) \
 	StarOffice_XML__Calc_ \
 	Text___txt___csv__StarCalc_ \
 	calc_HTML_WebQuery \
@@ -82,16 +58,13 @@ F4_CALC = \
 	calc_MS_Excel_2007_XML_Template
 
 # -----------------------------------------------
-# count = 14
+# count = 12
 F4_UI_CALC = \
 	HTML__StarCalc__ui \
 	MS_Excel_4_0_Vorlage_Template_ui \
 	MS_Excel_5_0_95_Vorlage_Template_ui \
 	MS_Excel_95_Vorlage_Template_ui \
 	MS_Excel_97_Vorlage_Template_ui \
-	StarCalc_3_0_Vorlage_Template_ui \
-	StarCalc_4_0_Vorlage_Template_ui \
-	StarCalc_5_0_Vorlage_Template_ui \
 	StarOffice_XML__Calc__ui \
 	Text___txt___csv__StarCalc__ui \
 	calc_HTML_WebQuery_ui \
--- filter/source/config/fragments/fcfg_calc_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_calc_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,59 @@
+all_fragments+=calc_bf
+
+# -----------------------------------------------
+# count = 7
+T4_CALC_BF = \
+    calc_StarCalc_10 \
+    calc_StarCalc_30 \
+    calc_StarCalc_30_VorlageTemplate \
+    calc_StarCalc_40 \
+    calc_StarCalc_40_VorlageTemplate \
+    calc_StarCalc_50 \
+    calc_StarCalc_50_VorlageTemplate
+
+# -----------------------------------------------
+# count = 7
+F4_CALC_BF = \
+	StarCalc_1_0 \
+	StarCalc_3_0 \
+	StarCalc_3_0_Vorlage_Template \
+    StarCalc_4_0 \
+    StarCalc_4_0_Vorlage_Template \
+	StarCalc_5_0 \
+	StarCalc_5_0_Vorlage_Template
+
+# -----------------------------------------------
+# count = 3
+F4_UI_CALC_BF = \
+	StarCalc_3_0_Vorlage_Template_ui \
+    StarCalc_4_0_Vorlage_Template_ui \
+	StarCalc_5_0_Vorlage_Template_ui
+    
+# -----------------------------------------------
+# count = 0
+L4_CALC_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_CALC_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_calc_bf           = $(foreach,i,$(T4_CALC_BF)    types$/$i.xcu                     )
+FILTERS_4fcfg_calc_bf         = $(foreach,i,$(F4_CALC_BF)    filters$/$i.xcu                   )
+UI_FILTERS_4fcfg_calc_bf      = $(foreach,i,$(F4_UI_CALC_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu   )
+FRAMELOADERS_4fcfg_calc_bf    = $(foreach,i,$(L4_CALC_BF)    frameloaders$/$i.xcu              )
+CONTENTHANDLERS_4fcfg_calc_bf = $(foreach,i,$(C4_CALC_BF)    contenthandlers$/$i.xcu           )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_calc_bf = \
+    $(TYPES_4fcfg_calc) \
+    $(FILTERS_4fcfg_calc) \
+    $(UI_FILTERS_4fcfg_calc) \
+    $(FRAMELOADERS_4fcfg_calc) \
+    $(CONTENTHANDLERS_4fcfg_calc)
+    
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_calc_bf)
+
+ALL_PACKAGES+=calc_bf
+
--- filter/source/config/fragments/fcfg_chart.mk.old	2007-05-22 17:23:08.000000000 +0200
+++ filter/source/config/fragments/fcfg_chart.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,30 +1,14 @@
 all_fragments+=chart
 
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_CHART_BINFILTER = \
-	chart_StarChart_30 \
-	chart_StarChart_40 \
-	chart_StarChart_50
-F4_CHART_BINFILTER = \
-    StarChart_3_0 \
-    StarChart_4_0 \
-    StarChart_5_0
-.ELSE
-T4_CHART_BINFILTER =
-F4_CHART_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
-# count = 5
+# count = 3
 T4_CHART = \
-	$(T4_CHART_BINFILTER) \
 	chart_StarOffice_XML_Chart \
 	chart8
 
 # -----------------------------------------------
-# count = 5
+# count = 2
 F4_CHART = \
-    $(F4_CHART_BINFILTER) \
     StarOffice_XML__Chart_\
     chart8
     
--- filter/source/config/fragments/fcfg_chart_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_chart_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,47 @@
+all_fragments+=chart_bf
+
+# -----------------------------------------------
+# count = 3
+T4_CHART_BF = \
+	chart_StarChart_30 \
+	chart_StarChart_40 \
+	chart_StarChart_50
+
+# -----------------------------------------------
+# count = 3
+F4_CHART_BF = \
+    StarChart_3_0 \
+    StarChart_4_0 \
+    StarChart_5_0
+
+# -----------------------------------------------
+# count = 0
+F4_UI_CHART_BF =
+
+# -----------------------------------------------
+# count = 0
+L4_CHART_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_CHART_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_chart_bf           = $(foreach,i,$(T4_CHART_BF)    types$/$i.xcu                   )
+FILTERS_4fcfg_chart_bf         = $(foreach,i,$(F4_CHART_BF)    filters$/$i.xcu                 )
+UI_FILTERS_4fcfg_chart_bf      = $(foreach,i,$(F4_UI_CHART_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu )
+FRAMELOADERS_4fcfg_chart_bf    = $(foreach,i,$(L4_CHART_BF)    frameloaders$/$i.xcu            )
+CONTENTHANDLERS_4fcfg_chart_bf = $(foreach,i,$(C4_CHART_BF)    contenthandlers$/$i.xcu         )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_chart_bf = \
+    $(TYPES_4fcfg_chart_bf) \
+    $(FILTERS_4fcfg_chart_bf) \
+    $(UI_FILTERS_4fcfg_chart_bf) \
+    $(FRAMELOADERS_4fcfg_chart_bf) \
+    $(CONTENTHANDLERS_4fcfg_chart_bf)
+    
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_chart_bf)
+
+ALL_PACKAGES+=chart_bf
--- filter/source/config/fragments/fcfg_draw.mk.old	2007-07-17 21:11:59.000000000 +0200
+++ filter/source/config/fragments/fcfg_draw.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,25 +1,8 @@
 all_fragments+=draw
 
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_DRAW_BINFILTER = \
-	draw_StarDraw_30 \
-	draw_StarDraw_30_Vorlage \
-	draw_StarDraw_50 \
-	draw_StarDraw_50_Vorlage
-F4_DRAW_BINFILTER = \
-    StarDraw_3_0 \
-    StarDraw_3_0_Vorlage \
-    StarDraw_5_0 \
-    StarDraw_5_0_Vorlage
-.ELSE
-T4_DRAW_BINFILTER =
-F4_DRAW_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
 # count = 9
 T4_DRAW = \
-    $(T4_DRAW_BINFILTER) \
     draw_StarOffice_XML_Draw \
     draw_StarOffice_XML_Draw_Template \
     pdf_Portable_Document_Format\
@@ -30,7 +13,6 @@ T4_DRAW = \
 # -----------------------------------------------
 # count = 9
 F4_DRAW = \
-    $(F4_DRAW_BINFILTER) \
     StarOffice_XML__Draw_ \
     draw_StarOffice_XML_Draw_Template \
     draw_pdf_Export \
@@ -40,10 +22,8 @@ F4_DRAW = \
 
 
 # -----------------------------------------------
-# count = 6
+# count = 4
 F4_UI_DRAW = \
-    StarDraw_3_0_Vorlage_ui \
-    StarDraw_5_0_Vorlage_ui \
     StarOffice_XML__Draw__ui \
     draw_StarOffice_XML_Draw_Template_ui \
     draw8_ui \
--- filter/source/config/fragments/fcfg_draw_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_draw_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,51 @@
+all_fragments+=draw_bf
+
+# -----------------------------------------------
+# count = 4
+T4_DRAW_BF = \
+    draw_StarDraw_30 \
+    draw_StarDraw_30_Vorlage \
+    draw_StarDraw_50 \
+    draw_StarDraw_50_Vorlage
+
+# -----------------------------------------------
+# count = 4
+F4_DRAW_BF = \
+    StarDraw_3_0 \
+    StarDraw_3_0_Vorlage \
+    StarDraw_5_0 \
+    StarDraw_5_0_Vorlage
+
+# -----------------------------------------------
+# count = 2
+F4_UI_DRAW_BF = \
+    StarDraw_3_0_Vorlage_ui \
+    StarDraw_5_0_Vorlage_ui
+    
+# -----------------------------------------------
+# count = 0
+L4_DRAW_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_DRAW_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_draw_bf           = $(foreach,i,$(T4_DRAW_BF)    types$/$i.xcu                    )
+FILTERS_4fcfg_draw_bf         = $(foreach,i,$(F4_DRAW_BF)    filters$/$i.xcu                  )
+UI_FILTERS_4fcfg_draw_bf      = $(foreach,i,$(F4_UI_DRAW_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu  )
+FRAMELOADERS_4fcfg_draw_bf    = $(foreach,i,$(L4_DRAW_BF)    frameloaders$/$i.xcu             )
+CONTENTHANDLERS_4fcfg_draw_bf = $(foreach,i,$(C4_DRAW_BF)    contenthandlers$/$i.xcu          )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_draw_bf = \
+    $(TYPES_4fcfg_draw_bf) \
+    $(FILTERS_4fcfg_draw_bf) \
+    $(UI_FILTERS_4fcfg_draw_bf) \
+    $(FRAMELOADERS_4fcfg_draw_bf) \
+    $(CONTENTHANDLERS_4fcfg_draw_bf)
+    
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_draw_bf)
+
+ALL_PACKAGES+=draw_bf
--- filter/source/config/fragments/fcfg_global.mk.old	2006-03-29 10:01:24.000000000 +0200
+++ filter/source/config/fragments/fcfg_global.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,28 +1,8 @@
 all_fragments+=global
 
-
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_GLOBAL_BINFILTER = \
-    writer_StarWriter_30 \
-    writer_StarWriter_40 \
-    writer_globaldocument_StarWriter_40GlobalDocument \
-    writer_StarWriter_50 \
-    writer_globaldocument_StarWriter_50GlobalDocument
-F4_GLOBAL_BINFILTER = \
-	StarWriter_3_0__StarWriter_GlobalDocument_ \
-	StarWriter_4_0__StarWriter_GlobalDocument_ \
-	StarWriter_4_0_GlobalDocument \
-	StarWriter_5_0__StarWriter_GlobalDocument_ \
-	StarWriter_5_0_GlobalDocument
-.ELSE
-T4_GLOBAL_BINFILTER =
-F4_GLOBAL_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
 # count = 12
 T4_GLOBAL = \
-    $(T4_GLOBAL_BINFILTER) \
     writer_Text \
     writer_StarOffice_XML_Writer \
     writer_globaldocument_StarOffice_XML_Writer_GlobalDocument \
@@ -32,7 +12,6 @@ T4_GLOBAL = \
 # -----------------------------------------------
 # count = 12
 F4_GLOBAL = \
-    $(F4_GLOBAL_BINFILTER) \
 	Text__encoded___StarWriter_GlobalDocument_ \
 	writer_globaldocument_StarOffice_XML_Writer \
 	writer_globaldocument_StarOffice_XML_Writer_GlobalDocument \
@@ -44,8 +23,6 @@ F4_GLOBAL = \
 # -----------------------------------------------
 # count = 12
 F4_UI_GLOBAL = \
-	StarWriter_4_0_GlobalDocument_ui \
-	StarWriter_5_0_GlobalDocument_ui \
 	Text__encoded___StarWriter_GlobalDocument__ui \
 	writer_globaldocument_StarOffice_XML_Writer_ui \
 	writer_globaldocument_StarOffice_XML_Writer_GlobalDocument_ui \
--- filter/source/config/fragments/fcfg_global_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_global_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,55 @@
+all_fragments+=global_bf
+
+
+# -----------------------------------------------
+# count = 5
+T4_GLOBAL_BF = \
+    writer_StarWriter_30 \
+    writer_StarWriter_40 \
+    writer_globaldocument_StarWriter_40GlobalDocument \
+    writer_StarWriter_50 \
+    writer_globaldocument_StarWriter_50GlobalDocument
+
+# -----------------------------------------------
+# count = 5
+F4_GLOBAL_BF = \
+	StarWriter_3_0__StarWriter_GlobalDocument_ \
+	StarWriter_4_0__StarWriter_GlobalDocument_ \
+	StarWriter_4_0_GlobalDocument \
+	StarWriter_5_0__StarWriter_GlobalDocument_ \
+	StarWriter_5_0_GlobalDocument
+
+# -----------------------------------------------
+# count = 2
+F4_UI_GLOBAL_BF = \
+	StarWriter_4_0_GlobalDocument_ui \
+	StarWriter_5_0_GlobalDocument_ui
+
+# -----------------------------------------------
+# count = 0
+L4_GLOBAL_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_GLOBAL_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_global_bf           = $(foreach,i,$(T4_GLOBAL_BF)    types$/$i.xcu                     )
+FILTERS_4fcfg_global_bf         = $(foreach,i,$(F4_GLOBAL_BF)    filters$/$i.xcu                   )
+UI_FILTERS_4fcfg_global_bf      = $(foreach,i,$(F4_UI_GLOBAL_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu   )
+FRAMELOADERS_4fcfg_global_bf    = $(foreach,i,$(L4_GLOBAL_BF)    frameloaders$/$i.xcu              )
+CONTENTHANDLERS_4fcfg_global_bf = $(foreach,i,$(C4_GLOBAL_BF)    contenthandlers$/$i.xcu           )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_global_bf = \
+    $(TYPES_4fcfg_global_bf) \
+    $(FILTERS_4fcfg_global_bf) \
+    $(UI_FILTERS_4fcfg_global_bf) \
+    $(FRAMELOADERS_4fcfg_global_bf) \
+    $(CONTENTHANDLERS_4fcfg_global_bf)
+    
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_global_bf)
+    
+ALL_PACKAGES+=global_bf
+
--- filter/source/config/fragments/fcfg_impress.mk.old	2007-07-17 21:12:07.000000000 +0200
+++ filter/source/config/fragments/fcfg_impress.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,39 +1,11 @@
 all_fragments+=impress
 
-
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_IMPRESS_BINFILTER = \
-    draw_StarDraw_30 \
-    draw_StarDraw_30_Vorlage \
-    draw_StarDraw_50 \
-    draw_StarDraw_50_Vorlage \
-    impress_StarImpress_40 \
-    impress_StarImpress_40_Vorlage \
-    impress_StarImpress_50 \
-    impress_StarImpress_50_Vorlage \
-    impress_StarImpress_50_packed
-F4_IMPRESS_BINFILTER = \
-    StarDraw_3_0_Vorlage__StarImpress_ \
-    StarDraw_3_0__StarImpress_ \
-    StarDraw_5_0_Vorlage__StarImpress_ \
-    StarDraw_5_0__StarImpress_ \
-    StarImpress_4_0 \
-    StarImpress_4_0_Vorlage \
-    StarImpress_5_0 \
-    StarImpress_5_0_Vorlage \
-    StarImpress_5_0__packed_
-.ELSE
-T4_IMPRESS_BINFILTER =
-F4_IMPRESS_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
 # count = 19
 T4_IMPRESS = \
     draw_StarOffice_XML_Draw \
     impress_MS_PowerPoint_97 \
     impress_MS_PowerPoint_97_Vorlage \
-    $(T4_IMPRESS_BINFILTER) \
     impress_StarOffice_XML_Impress \
     impress_StarOffice_XML_Impress_Template \
     pdf_Portable_Document_Format \
@@ -49,7 +21,6 @@ T4_IMPRESS = \
 F4_IMPRESS = \
     MS_PowerPoint_97 \
     MS_PowerPoint_97_Vorlage \
-    $(F4_IMPRESS_BINFILTER) \
     impress_StarOffice_XML_Draw \
     StarOffice_XML__Impress_ \
     impress_StarOffice_XML_Impress_Template \
@@ -65,12 +36,7 @@ F4_IMPRESS = \
 # count = 12
 F4_UI_IMPRESS = \
     MS_PowerPoint_97_Vorlage_ui \
-    StarDraw_3_0_Vorlage__StarImpress__ui \
-    StarDraw_5_0_Vorlage__StarImpress__ui \
     impress_StarOffice_XML_Draw_ui \
-    StarImpress_4_0_Vorlage_ui \
-    StarImpress_5_0_Vorlage_ui \
-    StarImpress_5_0__packed__ui \
     StarOffice_XML__Impress__ui \
     impress_StarOffice_XML_Impress_Template_ui \
     impress8_ui \
--- filter/source/config/fragments/fcfg_impress_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_impress_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,66 @@
+all_fragments+=impress_bf
+
+
+# -----------------------------------------------
+# count = 9
+T4_IMPRESS_BF = \
+    draw_StarDraw_30 \
+    draw_StarDraw_30_Vorlage \
+    draw_StarDraw_50 \
+    draw_StarDraw_50_Vorlage \
+    impress_StarImpress_40 \
+    impress_StarImpress_40_Vorlage \
+    impress_StarImpress_50 \
+    impress_StarImpress_50_Vorlage \
+    impress_StarImpress_50_packed
+
+# -----------------------------------------------
+# count = 9
+F4_IMPRESS_BF = \
+    StarDraw_3_0_Vorlage__StarImpress_ \
+    StarDraw_3_0__StarImpress_ \
+    StarDraw_5_0_Vorlage__StarImpress_ \
+    StarDraw_5_0__StarImpress_ \
+    StarImpress_4_0 \
+    StarImpress_4_0_Vorlage \
+    StarImpress_5_0 \
+    StarImpress_5_0_Vorlage \
+    StarImpress_5_0__packed_
+
+# -----------------------------------------------
+# count = 5
+F4_UI_IMPRESS_BF = \
+    StarDraw_3_0_Vorlage__StarImpress__ui \
+    StarDraw_5_0_Vorlage__StarImpress__ui \
+    StarImpress_4_0_Vorlage_ui \
+    StarImpress_5_0_Vorlage_ui \
+    StarImpress_5_0__packed__ui \
+
+# -----------------------------------------------
+# count = 0
+L4_IMPRESS_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_IMPRESS_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_impress_bf           = $(foreach,i,$(T4_IMPRESS_BF)    types$/$i.xcu                   )
+FILTERS_4fcfg_impress_bf         = $(foreach,i,$(F4_IMPRESS_BF)    filters$/$i.xcu                 )
+UI_FILTERS_4fcfg_impress_bf      = $(foreach,i,$(F4_UI_IMPRESS_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu )
+FRAMELOADERS_4fcfg_impress_bf    = $(foreach,i,$(L4_IMPRESS_BF)    frameloaders$/$i.xcu            )
+CONTENTHANDLERS_4fcfg_impress_bf = $(foreach,i,$(C4_IMPRESS_BF)    contenthandlers$/$i.xcu         )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_impress_bf = \
+    $(TYPES_4fcfg_impress_bf) \
+    $(FILTERS_4fcfg_impress_bf) \
+    $(UI_FILTERS_4fcfg_impress_bf) \
+    $(FRAMELOADERS_4fcfg_impress_bf) \
+    $(CONTENTHANDLERS_4fcfg_impress_bf)
+
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_impress_bf)
+
+ALL_PACKAGES+=impress_bf
+
--- filter/source/config/fragments/fcfg_math.mk.old	2005-07-14 17:03:43.000000000 +0200
+++ filter/source/config/fragments/fcfg_math.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,38 +1,20 @@
 all_fragments+=math
 
 
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_MATH_BINFILTER = \
-	math_StarMath_20 \
-	math_StarMath_30 \
-	math_StarMath_40 \
-	math_StarMath_50
-F4_MATH_BINFILTER = \
-    StarMath_2_0 \
-    StarMath_3_0 \
-    StarMath_4_0 \
-    StarMath_5_0
-.ELSE
-T4_MATH_BINFILTER =
-F4_MATH_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
-# count = 9
+# count = 5
 T4_MATH = \
 	math_MathML_XML_Math \
 	math_MathType_3x \
-	$(T4_MATH_BINFILTER) \
 	math_StarOffice_XML_Math \
 	pdf_Portable_Document_Format \
 	math8
 
 # -----------------------------------------------
-# count = 9
+# count = 5
 F4_MATH = \
     MathML_XML__Math_ \
     MathType_3_x \
-    $(F4_MATH_BINFILTER) \
     StarOffice_XML__Math_ \
     math_pdf_Export \
 	math8
--- filter/source/config/fragments/fcfg_math_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_math_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,50 @@
+all_fragments+=math_bf
+
+# -----------------------------------------------
+# count = 4
+T4_MATH_BF = \
+	math_StarMath_20 \
+	math_StarMath_30 \
+	math_StarMath_40 \
+	math_StarMath_50
+
+# -----------------------------------------------
+# count = 4
+F4_MATH_BF = \
+    StarMath_2_0 \
+    StarMath_3_0 \
+    StarMath_4_0 \
+    StarMath_5_0
+
+# -----------------------------------------------
+# count = 0
+F4_UI_MATH_BF =
+    
+# -----------------------------------------------
+# count = 0
+L4_MATH_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_MATH_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_math_bf           = $(foreach,i,$(T4_MATH_BF)    types$/$i.xcu                      )
+FILTERS_4fcfg_math_bf         = $(foreach,i,$(F4_MATH_BF)    filters$/$i.xcu                    )
+UI_FILTERS_4fcfg_math_bf      = $(foreach,i,$(F4_UI_MATH_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu    )
+FRAMELOADERS_4fcfg_math_bf    = $(foreach,i,$(L4_MATH_BF)    frameloaders$/$i.xcu               )
+CONTENTHANDLERS_4fcfg_math_bf = $(foreach,i,$(C4_MATH_BF)    contenthandlers$/$i.xcu            )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_math_bf = \
+    $(TYPES_4fcfg_math_bf) \
+    $(FILTERS_4fcfg_math_bf) \
+    $(UI_FILTERS_4fcfg_math_bf) \
+    $(FRAMELOADERS_4fcfg_math_bf) \
+    $(CONTENTHANDLERS_4fcfg_math_bf)
+
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_math_bf)
+    
+ALL_PACKAGES+=math_bf
+    
--- filter/source/config/fragments/fcfg_web.mk.old	2005-07-14 17:03:56.000000000 +0200
+++ filter/source/config/fragments/fcfg_web.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,28 +1,9 @@
 all_fragments+=web
 
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_WEB_BINFILTER = \
-	writer_StarWriter_30 \
-	writer_StarWriter_40 \
-	writer_StarWriter_50 \
-	writer_web_StarWriterWeb_40_VorlageTemplate \
-	writer_web_StarWriterWeb_50_VorlageTemplate
-F4_WEB_BINFILTER = \
-    StarWriter_3_0__StarWriter_Web_ \
-    StarWriter_4_0__StarWriter_Web_ \
-    StarWriter_5_0__StarWriter_Web_ \
-    StarWriter_Web_4_0_Vorlage_Template \
-    StarWriter_Web_5_0_Vorlage_Template
-.ELSE
-T4_WEB_BINFILTER =
-F4_WEB_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
 # count = 12
 T4_WEB = \
 	writer_web_HTML \
-	$(T4_WEB_BINFILTER) \
 	writer_Text \
 	writer_web_HTML_help \
 	writer_StarOffice_XML_Writer \
@@ -34,7 +15,6 @@ T4_WEB = \
 # count = 14
 F4_WEB = \
     HTML \
-    $(F4_WEB_BINFILTER) \
     Text__StarWriter_Web_ \
     Text__encoded___StarWriter_Web_ \
     writer_web_HTML_help \
@@ -48,8 +28,6 @@ F4_WEB = \
 # count = 9
 F4_UI_WEB = \
     HTML_ui \
-    StarWriter_Web_4_0_Vorlage_Template_ui \
-    StarWriter_Web_5_0_Vorlage_Template_ui \
     Text__StarWriter_Web__ui \
     Text__encoded___StarWriter_Web__ui \
     writer_web_StarOffice_XML_Writer_ui \
--- filter/source/config/fragments/fcfg_web_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_web_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,54 @@
+all_fragments+=web_bf
+
+# -----------------------------------------------
+# count = 5
+T4_WEB_BF = \
+	writer_StarWriter_30 \
+	writer_StarWriter_40 \
+	writer_StarWriter_50 \
+	writer_web_StarWriterWeb_40_VorlageTemplate \
+	writer_web_StarWriterWeb_50_VorlageTemplate
+
+# -----------------------------------------------
+# count = 5
+F4_WEB_BF = \
+    StarWriter_3_0__StarWriter_Web_ \
+    StarWriter_4_0__StarWriter_Web_ \
+    StarWriter_5_0__StarWriter_Web_ \
+    StarWriter_Web_4_0_Vorlage_Template \
+    StarWriter_Web_5_0_Vorlage_Template
+
+# -----------------------------------------------
+# count = 2
+F4_UI_WEB_BF = \
+    StarWriter_Web_4_0_Vorlage_Template_ui \
+    StarWriter_Web_5_0_Vorlage_Template_ui
+    
+# -----------------------------------------------
+# count = 0
+L4_WEB_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_WEB_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_web_bf           = $(foreach,i,$(T4_WEB_BF)    types$/$i.xcu                   )
+FILTERS_4fcfg_web_bf         = $(foreach,i,$(F4_WEB_BF)    filters$/$i.xcu                 )
+UI_FILTERS_4fcfg_web_bf      = $(foreach,i,$(F4_UI_WEB_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu )
+FRAMELOADERS_4fcfg_web_bf    = $(foreach,i,$(L4_WEB_BF)    frameloaders$/$i.xcu            )
+CONTENTHANDLERS_4fcfg_web_bf = $(foreach,i,$(C4_WEB_BF)    contenthandlers$/$i.xcu         )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_web_bf = \
+    $(TYPES_4fcfg_web_bf) \
+    $(FILTERS_4fcfg_web_bf) \
+    $(UI_FILTERS_4fcfg_web_bf) \
+    $(FRAMELOADERS_4fcfg_web_bf) \
+    $(CONTENTHANDLERS_4fcfg_web_bf)
+
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_web_bf)
+    
+ALL_PACKAGES+=web_bf
+    
--- filter/source/config/fragments/fcfg_writer.mk.old	2007-07-17 21:12:07.000000000 +0200
+++ filter/source/config/fragments/fcfg_writer.mk	2007-07-19 12:09:53.000000000 +0200
@@ -1,32 +1,5 @@
 all_fragments+=writer
 
-
-.IF "$(WITH_BINFILTER)" != "NO"
-T4_WRITER_BINFILTER = \
-    writer_StarWriter_10 \
-    writer_StarWriter_20 \
-    writer_StarWriter_30 \
-    writer_StarWriter_30_VorlageTemplate \
-    writer_StarWriter_40 \
-    writer_StarWriter_40_VorlageTemplate \
-    writer_StarWriter_50 \
-    writer_StarWriter_50_VorlageTemplate \
-    writer_StarWriter_DOS
-F4_WRITER_BINFILTER = \
-	StarWriter_1_0 \
-	StarWriter_2_0 \
-	StarWriter_3_0 \
-	StarWriter_3_0_Vorlage_Template \
-	StarWriter_4_0 \
-	StarWriter_4_0_Vorlage_Template \
-	StarWriter_5_0 \
-	StarWriter_5_0_Vorlage_Template \
-	StarWriter_DOS
-.ELSE
-T4_WRITER_BINFILTER =
-F4_WRITER_BINFILTER =
-.ENDIF
-
 # -----------------------------------------------
 # count = 39
 T4_WRITER = \
@@ -44,7 +17,6 @@ T4_WRITER = \
     writer_MS_Word_97_Vorlage \
     writer_Rich_Text_Format \
     writer_StarOffice_XML_Writer \
-    $(T4_WRITER_BINFILTER) \
     writer_WordPerfect_Document \
     writer_MS_Works_Document \
     writer_T602_Document \
@@ -81,7 +53,6 @@ F4_WRITER = \
 	MS_Word_97_Vorlage \
 	Rich_Text_Format \
 	StarOffice_XML__Writer_ \
-	$(F4_WRITER_BINFILTER) \
 	WordPerfect \
 	MS_Works \
 	T602Document \
@@ -108,9 +79,6 @@ F4_UI_WRITER = \
 	MS_Word_95_Vorlage_ui \
 	MS_Word_97_Vorlage_ui \
 	StarOffice_XML__Writer__ui \
-	StarWriter_3_0_Vorlage_Template_ui \
-	StarWriter_4_0_Vorlage_Template_ui \
-	StarWriter_5_0_Vorlage_Template_ui \
 	Text_ui \
 	Text__encoded__ui \
 	writer_JustSystem_Ichitaro_10_template_ui \
--- filter/source/config/fragments/fcfg_writer_bf.mk.old	2007-07-19 12:09:53.000000000 +0200
+++ filter/source/config/fragments/fcfg_writer_bf.mk	2007-07-19 12:09:53.000000000 +0200
@@ -0,0 +1,63 @@
+all_fragments+=writer_bf
+
+# -----------------------------------------------
+# count = 9
+T4_WRITER_BF = \
+    writer_StarWriter_10 \
+    writer_StarWriter_20 \
+    writer_StarWriter_30 \
+    writer_StarWriter_30_VorlageTemplate \
+    writer_StarWriter_40 \
+    writer_StarWriter_40_VorlageTemplate \
+    writer_StarWriter_50 \
+    writer_StarWriter_50_VorlageTemplate \
+    writer_StarWriter_DOS
+
+# -----------------------------------------------
+# count = 9
+F4_WRITER_BF = \
+	StarWriter_1_0 \
+	StarWriter_2_0 \
+	StarWriter_3_0 \
+	StarWriter_3_0_Vorlage_Template \
+	StarWriter_4_0 \
+	StarWriter_4_0_Vorlage_Template \
+	StarWriter_5_0 \
+	StarWriter_5_0_Vorlage_Template \
+	StarWriter_DOS
+
+# -----------------------------------------------
+# count = 3
+F4_UI_WRITER_BF = \
+	StarWriter_3_0_Vorlage_Template_ui \
+	StarWriter_4_0_Vorlage_Template_ui \
+	StarWriter_5_0_Vorlage_Template_ui \
+
+# -----------------------------------------------
+# count = 0
+L4_WRITER_BF =
+
+# -----------------------------------------------
+# count = 0
+C4_WRITER_BF =
+
+# -----------------------------------------------
+TYPES_4fcfg_writer_bf           = $(foreach,i,$(T4_WRITER_BF)    types$/$i.xcu                     )
+FILTERS_4fcfg_writer_bf         = $(foreach,i,$(F4_WRITER_BF)    filters$/$i.xcu                   )
+UI_FILTERS_4fcfg_writer_bf      = $(foreach,i,$(F4_UI_WRITER_BF) $(DIR_LOCFRAG)$/filters$/$i.xcu   )
+FRAMELOADERS_4fcfg_writer_bf    = $(foreach,i,$(L4_WRITER_BF)    frameloaders$/$i.xcu              )
+CONTENTHANDLERS_4fcfg_writer_bf = $(foreach,i,$(C4_WRITER_BF)    contenthandlers$/$i.xcu           )
+
+# -----------------------------------------------
+# needed to get dependencies inside global makefile work!
+ALL_4fcfg_writer_bf = \
+    $(TYPES_4fcfg_writer_bf) \
+    $(FILTERS_4fcfg_writer_bf) \
+    $(UI_FILTERS_4fcfg_writer_bf) \
+    $(FRAMELOADERS_4fcfg_writer_bf) \
+    $(CONTENTHANDLERS_4fcfg_writer_bf)
+
+ALL_UI_FILTERS+=$(UI_FILTERS_4fcfg_writer_bf)
+
+ALL_PACKAGES+=writer_bf
+
--- filter/source/config/fragments/makefile.mk.old	2007-05-10 15:13:05.000000000 +0200
+++ filter/source/config/fragments/makefile.mk	2007-07-19 12:09:53.000000000 +0200
@@ -72,6 +72,18 @@ REALFILTERPACKAGES_TYPES_FLAG = \
     $(DIR_FILTERCFGOUT)$/fcfg_drawgraphics.types_flag       \
     $(DIR_FILTERCFGOUT)$/fcfg_impressgraphics.types_flag
 
+.IF "$(WITH_BINFILTER)" != "NO"
+REALFILTERPACKAGES_TYPES_FLAG += \
+    $(DIR_FILTERCFGOUT)$/fcfg_writer_bf.types_flag          \
+    $(DIR_FILTERCFGOUT)$/fcfg_web_bf.types_flag             \
+    $(DIR_FILTERCFGOUT)$/fcfg_global_bf.types_flag          \
+    $(DIR_FILTERCFGOUT)$/fcfg_calc_bf.types_flag            \
+    $(DIR_FILTERCFGOUT)$/fcfg_draw_bf.types_flag            \
+    $(DIR_FILTERCFGOUT)$/fcfg_impress_bf.types_flag         \
+    $(DIR_FILTERCFGOUT)$/fcfg_chart_bf.types_flag           \
+    $(DIR_FILTERCFGOUT)$/fcfg_math_bf.types_flag
+.ENDIF
+
 REALFILTERPACKAGES_FILTERS_FLAG = \
     $(DIR_FILTERCFGOUT)$/fcfg_base.filters_flag               \
     $(DIR_FILTERCFGOUT)$/fcfg_writer.filters_flag             \
@@ -91,6 +103,18 @@ REALFILTERPACKAGES_FILTERS_FLAG = \
     $(DIR_FILTERCFGOUT)$/fcfg_drawgraphics.filters_flag       \
     $(DIR_FILTERCFGOUT)$/fcfg_impressgraphics.filters_flag
 
+.IF "$(WITH_BINFILTER)" != "NO"
+REALFILTERPACKAGES_FILTERS_FLAG += \
+    $(DIR_FILTERCFGOUT)$/fcfg_writer_bf.filters_flag          \
+    $(DIR_FILTERCFGOUT)$/fcfg_web_bf.filters_flag             \
+    $(DIR_FILTERCFGOUT)$/fcfg_global_bf.filters_flag          \
+    $(DIR_FILTERCFGOUT)$/fcfg_calc_bf.filters_flag            \
+    $(DIR_FILTERCFGOUT)$/fcfg_draw_bf.filters_flag            \
+    $(DIR_FILTERCFGOUT)$/fcfg_impress_bf.filters_flag         \
+    $(DIR_FILTERCFGOUT)$/fcfg_chart_bf.filters_flag           \
+    $(DIR_FILTERCFGOUT)$/fcfg_math_bf.filters_flag
+.ENDIF
+
 REALFILTERPACKAGES_FILTERS_UI_MERGE = \
     $(DIR_LOCMERGE)$/filter_ui.xcu
 
@@ -119,6 +143,18 @@ REALFILTERPACKAGES_OTHERS_FLAG = \
     $(DIR_FILTERCFGOUT)$/fcfg_drawgraphics.others_flag       \
     $(DIR_FILTERCFGOUT)$/fcfg_impressgraphics.others_flag
 
+.IF "$(WITH_BINFILTER)" != "NO"
+REALFILTERPACKAGES_OTHERS_FLAG += \
+    $(DIR_FILTERCFGOUT)$/fcfg_writer_bf.others_flag          \
+    $(DIR_FILTERCFGOUT)$/fcfg_web_bf.others_flag             \
+    $(DIR_FILTERCFGOUT)$/fcfg_global_bf.others_flag          \
+    $(DIR_FILTERCFGOUT)$/fcfg_calc_bf.others_flag            \
+    $(DIR_FILTERCFGOUT)$/fcfg_draw_bf.others_flag            \
+    $(DIR_FILTERCFGOUT)$/fcfg_impress_bf.others_flag         \
+    $(DIR_FILTERCFGOUT)$/fcfg_chart_bf.others_flag           \
+    $(DIR_FILTERCFGOUT)$/fcfg_math_bf.others_flag
+.ENDIF
+
 INTERNALFILTERPACKAGES_TYPES_FLAG = \
     $(DIR_FILTERCFGOUT)$/fcfg_internalgraphics.types_flag
 
--- filter/source/config/fragments/packagedef.mk.old	2006-11-08 13:02:01.000000000 +0100
+++ filter/source/config/fragments/packagedef.mk	2007-07-19 12:09:53.000000000 +0200
@@ -64,4 +64,16 @@ INCLUDE_FRAGMENTS = \
 	 $(PRJ)$/source$/config$/fragments$/fcfg_impressgraphics.mk \
 	 $(PRJ)$/source$/config$/fragments$/fcfg_internalgraphics.mk
 
+.IF "$(WITH_BINFILTER)" != "NO"
+INCLUDE_FRAGMENTS += \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_writer_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_web_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_global_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_calc_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_draw_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_impress_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_chart_bf.mk \
+	 $(PRJ)$/source$/config$/fragments$/fcfg_math_bf.mk
+.ENDIF
+
 .INCLUDE: $(INCLUDE_FRAGMENTS)
