Index: lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk,v
retrieving revision 1.8
diff -u -u -r1.8 makefile.mk
--- lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	17 Jun 2005 09:21:29 -0000	1.8
+++ lingucomponent/source/hyphenator/altlinuxhyph/hyphen/makefile.mk	8 Aug 2005 20:42:23 -0000
@@ -68,7 +68,11 @@
 USE_DEFFILE=TRUE
 
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_HYPH)" == "YES"
+HNJLIB=-lhnj_pic
+.ELSE
 HNJLIB=-lhnj
+.ENDIF
 .ELSE
 HNJLIB=libhnj.lib
 .ENDIF
Index: lingucomponent/source/thesaurus/libnth/makefile.mk
===================================================================
RCS file: /cvs/whiteboard/lingucomponent/source/thesaurus/libnth/makefile.mk,v
retrieving revision 1.4
diff -u -u -r1.4 makefile.mk
--- lingucomponent/source/thesaurus/libnth/makefile.mk	17 Jun 2005 09:22:43 -0000	1.4
+++ lingucomponent/source/thesaurus/libnth/makefile.mk	8 Aug 2005 20:42:23 -0000
@@ -70,7 +70,11 @@
 
 .IF "$(MYTHESLIB)"==""
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_MYTHES)" == "YES"
+MYTHESLIB=-lmythes_pic
+.ELSE
 MYTHESLIB=-lmythes
+.ENDIF
 .ENDIF # unx
 .IF "$(GUI)"=="WNT"
 MYTHESLIB=libmythes.lib
--- solenv/inc/libs.mk.old 2005-12-09 08:25:13.000000000 +0100
+++ solenv/inc/libs.mk     2005-12-09 08:25:28.000000000 +0100
@@ -115,7 +115,7 @@
 CANVASTOOLSLIB=-lcanvastools$(OFFICEUPD)$(DLLPOSTFIX)
 CPPCANVASLIB=-lcppcanvas$(OFFICEUPD)$(DLLPOSTFIX)
 .IF "$(SYSTEM_AGG)" == "YES"
-AGGLIB=-lagg
+AGGLIB=-lagg_pic
 .ELSE
 AGGLIB=-lagg$(OFFICEUPD)$(DLLPOSTFIX)
 .ENDIF

--- lingucomponent/source/spellcheck/spell/makefile.mk-old	2006-03-16 19:03:49.946824328 +0100
+++ lingucomponent/source/spellcheck/spell/makefile.mk	2006-03-16 19:04:44.188578328 +0100
@@ -47,7 +47,11 @@
 
 .IF "$(HUNSPELLLIB)"==""
 .IF "$(GUI)"=="UNX"
+.IF "$(SYSTEM_HUNSPELL)" == "YES"
+HUNSPELLLIB=-lhunspell_pic
+.ELSE
 HUNSPELLLIB=-lhunspell
+.ENDIF
 .ENDIF # unx
 .IF "$(GUI)"=="WNT"
 HUNSPELLLIB=hunspell.lib
