diff -pur basic/inc/sbxobj.hxx basic/inc/sbxobj.hxx
--- basic/inc/sbxobj.hxx	2006-12-22 16:13:20.000000000 +0100
+++ basic/inc/sbxobj.hxx	2006-12-21 12:13:34.000000000 +0100
@@ -37,7 +37,6 @@
 
 #include <svtools/lstner.hxx>
 #include "sbxvar.hxx"
-#include <basic/sb/sbdllapi.h>
 
 ///////////////////////////////////////////////////////////////////////////
 
diff -pur canvas/inc/canvas/base/sprite.hxx canvas/inc/canvas/base/sprite.hxx
--- canvas/inc/canvas/base/sprite.hxx	2006-12-22 16:13:20.000000000 +0100
+++ canvas/inc/canvas/base/sprite.hxx	2006-12-21 11:51:59.000000000 +0100
@@ -38,7 +38,7 @@
 #include <rtl/ref.hxx>
 #include <com/sun/star/lang/XComponent.hpp>
 #include <com/sun/star/rendering/XCanvas.hpp>
-#include <basegfx/vector/b2dpoint.hxx>
+#include <basegfx/point/b2dpoint.hxx>
 #include <basegfx/vector/b2dsize.hxx>
 
 namespace basegfx
diff -pur connectivity/source/drivers/adabas/BGroups.cxx connectivity/source/drivers/adabas/BGroups.cxx
--- connectivity/source/drivers/adabas/BGroups.cxx	2006-12-22 16:13:21.000000000 +0100
+++ connectivity/source/drivers/adabas/BGroups.cxx	2006-12-21 11:57:43.000000000 +0100
@@ -36,13 +36,11 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_connectivity.hxx"
 #include "adabas/BGroups.hxx"
-#ifndef _CONNECTIVITY_ADABAS_GROUP_HXX_
 #include "adabas/BGroup.hxx"
-#endif
 #include "adabas/BTable.hxx"
 #include <com/sun/star/sdbc/XRow.hpp>
 #include <com/sun/star/sdbc/XResultSet.hpp>
-#include "sdbcx/IRefreshable.hxx"
+#include "connectivity/sdbcx/IRefreshable.hxx"
 #include <comphelper/types.hxx>
 
 using namespace ::comphelper;
diff -pur connectivity/source/drivers/dbase/DIndexes.cxx connectivity/source/drivers/dbase/DIndexes.cxx
--- connectivity/source/drivers/dbase/DIndexes.cxx	2006-12-22 16:13:21.000000000 +0100
+++ connectivity/source/drivers/dbase/DIndexes.cxx	2006-12-21 12:00:36.000000000 +0100
@@ -38,11 +38,9 @@
 #include "dbase/DIndexes.hxx"
 #include "dbase/DIndex.hxx"
 #include <connectivity/dbexception.hxx>
-#ifndef _UNOTOOLS_UCBHELPER_HXX
 #include <unotools/ucbhelper.hxx>
-#endif
 #include <comphelper/types.hxx>
-#include "FTable.hxx"
+#include "file/FTable.hxx"
 
 using namespace ::comphelper;
 
diff -pur connectivity/source/drivers/file/FNoException.cxx connectivity/source/drivers/file/FNoException.cxx
--- connectivity/source/drivers/file/FNoException.cxx	2006-12-22 16:13:21.000000000 +0100
+++ connectivity/source/drivers/file/FNoException.cxx	2006-12-21 11:55:11.000000000 +0100
@@ -40,7 +40,7 @@
 #include "file/fanalyzer.hxx"
 #include "file/FResultSet.hxx"
 #include "file/FPreparedStatement.hxx"
-#include "FValue.hxx"
+#include "connectivity/FValue.hxx"
 #include <tools/debug.hxx>
 #include "TKeyValue.hxx"
 
diff -pur connectivity/source/drivers/kab/kfields.hxx connectivity/source/drivers/kab/kfields.hxx
--- connectivity/source/drivers/kab/kfields.hxx	2006-12-22 16:13:21.000000000 +0100
+++ connectivity/source/drivers/kab/kfields.hxx	2006-12-21 12:00:54.000000000 +0100
@@ -35,7 +35,7 @@
 
 #include <vcl/kde_headers.h>
 #include <connectivity/dbexception.hxx>
-#include "ustring.hxx"
+#include <rtl/ustring.hxx>
 
 #define KAB_FIELD_REVISION	0
 #define KAB_DATA_FIELDS		1
diff -pur framework/inc/jobs/job.hxx framework/inc/jobs/job.hxx
--- framework/inc/jobs/job.hxx	2006-12-22 16:13:23.000000000 +0100
+++ framework/inc/jobs/job.hxx	2006-12-21 12:02:55.000000000 +0100
@@ -54,7 +54,7 @@
 #include <com/sun/star/frame/XFrame.hpp>
 #include <com/sun/star/frame/XDesktop.hpp>
 #include <com/sun/star/frame/XDispatchResultListener.hpp>
-#include <com/sun/star/frame/XtERMINATEListener.hpp>
+#include <com/sun/star/frame/XTerminateListener.hpp>
 #include <com/sun/star/task/XJobListener.hpp>
 #include <com/sun/star/util/XCloseListener.hpp>
 #include <com/sun/star/frame/DispatchResultEvent.hpp>
diff -pur sd/source/filter/pptin.hxx sd/source/filter/pptin.hxx
--- sd/source/filter/pptin.hxx	2006-12-22 16:13:30.000000000 +0100
+++ sd/source/filter/pptin.hxx	2006-12-21 15:35:27.000000000 +0100
@@ -41,7 +41,6 @@
 #include <svx/svdtypes.hxx>
 #include <svx/msfiltertracer.hxx>
 #include <com/sun/star/uno/Any.h>
-#include <ppt/pptanimations.hxx>
 #ifndef BOOST_SHARED_PTR_HPP_INCLUDED
 #include <boost/shared_ptr.hpp>
 #endif
diff -pur sfx2/source/appl/shutdownicon.hxx sfx2/source/appl/shutdownicon.hxx
--- sfx2/source/appl/shutdownicon.hxx	2006-12-22 16:13:31.000000000 +0100
+++ sfx2/source/appl/shutdownicon.hxx	2006-12-21 12:15:18.000000000 +0100
@@ -9,17 +10,13 @@
 #include <com/sun/star/lang/XEventListener.hpp>
 #include <com/sun/star/lang/XServiceInfo.hpp>
 #include <com/sun/star/lang/XInitialization.hpp>
-#ifndef _RTL_STRING_HXX
 #include <rtl/string.hxx>
-#endif
-#ifndef _RTL_USTRING_HXX
 #include <rtl/ustring.hxx>
-#endif
 #include <osl/mutex.hxx>
 #include <osl/module.hxx>
 #include <sfxuno.hxx>
 #include <cppuhelper/compbase3.hxx>
-#include "dllapi.h"
+#include "sfx2/dllapi.h"
 
 class ResMgr;
 
diff -pur svx/source/engine3d/obj3d.cxx svx/source/engine3d/obj3d.cxx
--- svx/source/engine3d/obj3d.cxx	2006-12-22 16:13:33.000000000 +0100
+++ svx/source/engine3d/obj3d.cxx	2006-12-21 14:03:07.000000000 +0100
@@ -75,7 +75,7 @@
 #include "xlnwtit.hxx"
 #include <vcl/virdev.hxx>
 #include <tools/poly.hxx>
-#include "b3dtrans.hxx"
+#include <goodies/b3dtrans.hxx>
 
 #ifndef _SVX_SVXIDS_HRC
 #include "svxids.hrc"
diff -pur svx/source/form/datanavi.cxx svx/source/form/datanavi.cxx
--- svx/source/form/datanavi.cxx	2006-12-22 16:13:33.000000000 +0100
+++ svx/source/form/datanavi.cxx	2006-12-21 14:19:06.000000000 +0100
@@ -73,8 +73,8 @@
 #include <sfx2/dispatch.hxx>
 #endif
 #include <com/sun/star/beans/PropertyAttribute.hpp>
-#include <com/sun/star/container/XENumeration.hpp>
-#include <com/sun/star/container/XENumerationAccess.hpp>
+#include <com/sun/star/container/XEnumeration.hpp>
+#include <com/sun/star/container/XEnumerationAccess.hpp>
 #include <com/sun/star/container/XSet.hpp>
 #include <com/sun/star/datatransfer/XTransferable.hpp>
 #include <com/sun/star/frame/XController.hpp>
@@ -84,7 +84,7 @@
 #include <com/sun/star/xforms/XFormsSupplier.hpp>
 #include <com/sun/star/xml/dom/XDocument.hpp>
 #include <com/sun/star/xml/dom/DOMException.hpp>
-#include <com/sun/star/xml/dom/DOMExceptiontype.hpp>
+#include <com/sun/star/xml/dom/DOMExceptionType.hpp>
 #include <com/sun/star/form/binding/XValueBinding.hpp>
 #ifndef _UNOTOOLS_PROCESSFACTORY_HXX
 #include <comphelper/processfactory.hxx>
diff -pur svx/source/tbxctrls/colorwindow.hxx svx/source/tbxctrls/colorwindow.hxx
--- svx/source/tbxctrls/colorwindow.hxx	2006-12-22 16:13:37.000000000 +0100
+++ svx/source/tbxctrls/colorwindow.hxx	2006-12-21 14:48:45.000000000 +0100
@@ -4,7 +4,7 @@
 #define __SVX_COLORWINDOW_HXX_
 
 #include <sfx2/tbxctrl.hxx>
-#include <vcl/valueset.hxx>
+#include <svtools/valueset.hxx>
 #include <svtools/lstner.hxx>
 #include <rtl/ustring.hxx>
 #include <com/sun/star/frame/XFrame.hpp>
diff -pur toolkit/inc/toolkit/controls/eventcontainer.hxx toolkit/inc/toolkit/controls/eventcontainer.hxx
--- toolkit/inc/toolkit/controls/eventcontainer.hxx	2006-12-22 16:13:41.000000000 +0100
+++ toolkit/inc/toolkit/controls/eventcontainer.hxx	2006-12-21 11:39:41.000000000 +0100
@@ -33,14 +33,12 @@
  *
  ************************************************************************/
 
+#ifndef _TOOLKIT_CONTROLS_EVENTCONTAINER_HXX_
+#define _TOOLKIT_CONTROLS_EVENTCONTAINER_HXX_
 
 #include <osl/diagnose.h>
-#ifndef _COM_SUN_STAR_XNAMECONTAINER_HPP_
 #include <com/sun/star/container/XNameContainer.hpp>
-#endif
-#ifndef _COM_SUN_STAR_XCONTAINER_HPP_
 #include <com/sun/star/container/XContainer.hpp>
-#endif
 
 #include <toolkit/helper/listenermultiplexer.hxx>
 
@@ -146,4 +146,4 @@ public:
 
 
 }	// namespace toolkit_namecontainer
-
+#endif
diff -pur toolkit/inc/toolkit/controls/stdtabcontroller.hxx toolkit/inc/toolkit/controls/stdtabcontroller.hxx
--- toolkit/inc/toolkit/controls/stdtabcontroller.hxx	2006-12-22 16:13:41.000000000 +0100
+++ toolkit/inc/toolkit/controls/stdtabcontroller.hxx	2006-12-21 11:37:19.000000000 +0100
@@ -41,7 +41,7 @@
 #include <com/sun/star/awt/XTabController.hpp>
 #include <com/sun/star/awt/XControl.hpp>
 #include <com/sun/star/awt/XControlContainer.hpp>
-#include <com/sun/star/awt/XWINDOW.hpp>
+#include <com/sun/star/awt/XWindow.hpp>
 #include <com/sun/star/lang/XTypeProvider.hpp>
 #include <cppuhelper/weakagg.hxx>
 #include <osl/mutex.hxx>
diff -pur tools/inc/simplerm.hxx tools/inc/simplerm.hxx
--- tools/inc/simplerm.hxx	2006-12-22 16:13:41.000000000 +0100
+++ tools/inc/simplerm.hxx	2006-12-21 11:24:44.000000000 +0100
@@ -43,7 +43,7 @@
 #define _TOOLS_SIMPLERESMGR_HXX_
 
 #include <vos/mutex.hxx>
-#include <tools/resid.hxx>
+#include <resid.hxx>
 #include <i18npool/lang.h>
 #include <string.hxx>
 #include <com/sun/star/lang/Locale.hpp>
diff -pur ucb/source/ucp/file/filrow.hxx ucb/source/ucp/file/filrow.hxx
--- ucb/source/ucp/file/filrow.hxx	2006-12-22 16:13:42.000000000 +0100
+++ ucb/source/ucp/file/filrow.hxx	2006-12-21 11:35:06.000000000 +0100
@@ -34,7 +34,6 @@
  ************************************************************************/
 #ifndef _FILROW_HXX_
 #define _FILROW_HXX_
-#endif
 
 #include <ucbhelper/macros.hxx>
 #include <vos/mutex.hxx>
@@ -207,3 +207,4 @@ namespace fileaccess {
 	};
 
 } // end namespace fileaccess
+#endif
diff -pur xmlhelp/inc/qe/QueryProcessor.hxx xmlhelp/inc/qe/QueryProcessor.hxx
--- xmlhelp/inc/qe/QueryProcessor.hxx	2006-12-22 16:13:42.000000000 +0100
+++ xmlhelp/inc/qe/QueryProcessor.hxx	2006-12-21 11:46:35.000000000 +0100
@@ -35,15 +35,12 @@
 #ifndef _XMLSEARCH_QE_QUERYPROCESSOR_HXX_
 #define _XMLSEARCH_QE_QUERYPROCESSOR_HXX_
 
-#ifndef INCLUDED_STL_VECTOR
 #include <vector>
-#define INCLUDED_STL_VECTOR
-#endif
-#include  <rtl/ustring.hxx>
+#include <rtl/ustring.hxx>
 #include <excep/XmlSearchExceptions.hxx>
 #include <qe/Query.hxx>
 #include <qe/Search.hxx>
-#include <xmlsearch/qe/XmlIndex.hxx>
+#include <qe/XmlIndex.hxx>
 
 
 namespace xmlsearch {
diff -pur xmloff/inc/DomBuilderContext.hxx xmloff/inc/DomBuilderContext.hxx
--- xmloff/inc/DomBuilderContext.hxx	2006-12-22 16:13:42.000000000 +0100
+++ xmloff/inc/DomBuilderContext.hxx	2006-12-21 11:48:16.000000000 +0100
@@ -41,7 +41,7 @@
 //
 
 #include "xmlictxt.hxx"
-#include "Reference.hxx"
+#include "com/sun/star/uno/Reference.hxx"
 
 
 //
diff -pur sw/source/filter/xml/xmlbrshi.hxx sw/source/filter/xml/xmlbrshi.hxx
--- sw/source/filter/xml/xmlbrshi.hxx	2007-01-02 15:30:34.000000000 +0100
+++ sw/source/filter/xml/xmlbrshi.hxx	2007-01-02 15:30:40.000000000 +0100
@@ -36,7 +36,7 @@
 #ifndef _XMLBRSHI_HXX
 #define _XMLBRSHI_HXX
 
-#include "xmlictxt.hxx"
+#include <xmloff/xmlictxt.hxx>
 
 class SvXMLImport;
 class SvXMLUnitConverter;
