--- svx/inc/fntctrl.hxx	2006-11-24 15:41:52.000000000 +0800
+++ svx/inc/fntctrl.hxx	2006-11-24 15:44:25.000000000 +0800
@@ -61,6 +61,8 @@ private:
 	FontPrevWin_Impl*   pImpl;
 
 	SVX_DLLPRIVATE void				InitSettings( BOOL bForeground, BOOL bBackground );
+    sal_Bool bIsCJKUI;
+    sal_Bool bIsCTLUI;
 
 public:
 						SvxFontPrevWindow( Window* pParent, const ResId& rId );
--- svx/source/dialog/fntctrl.cxx	2006-11-24 17:03:03.000000000 +0800
+++ svx/source/dialog/fntctrl.cxx	2006-11-24 17:03:10.000000000 +0800
@@ -486,6 +486,27 @@ SvxFontPrevWindow::SvxFontPrevWindow( Wi
 	initFont(pImpl->aCTLFont);
 	InitSettings( TRUE, TRUE );
     SetBorderStyle( WINDOW_BORDER_MONO );
+
+    LanguageType eLanguage = Application::GetSettings().GetUILanguage();
+    switch( eLanguage )
+    {
+        case LANGUAGE_CHINESE:
+        case LANGUAGE_JAPANESE:
+        case LANGUAGE_KOREAN:
+        case LANGUAGE_KOREAN_JOHAB:
+        case LANGUAGE_CHINESE_SIMPLIFIED:
+        case LANGUAGE_CHINESE_HONGKONG:
+        case LANGUAGE_CHINESE_SINGAPORE:
+        case LANGUAGE_CHINESE_MACAU:
+        case LANGUAGE_CHINESE_TRADITIONAL:
+            bIsCJKUI = sal_True;
+            break;
+        //TODO CTL Locale
+        // bIsCTLUI = sal_True
+        default:
+            bIsCJKUI = bIsCTLUI = sal_False;
+            break;
+    }
 }
 
 // -----------------------------------------------------------------------
@@ -613,6 +634,8 @@ void SvxFontPrevWindow::Paint( const Rec
 {
 	Printer* pPrinter = pImpl->pPrinter;
 	SvxFont& rFont = pImpl->aFont;
+	SvxFont& rCJKFont = pImpl->aCJKFont;
+	SvxFont& rCTLFont = pImpl->aCTLFont;
 
     if ( pImpl->bUseResText )
 		pImpl->aText = GetText();
@@ -629,7 +652,12 @@ void SvxFontPrevWindow::Paint( const Rec
 		}
 
 		if ( !pImpl->bSelection )
+        {
 			pImpl->aText = rFont.GetName();
+            if( bIsCJKUI )
+                pImpl->aText += rCJKFont.GetName();
+            //TODO bIsCTLUI
+        }
 
 		if ( !pImpl->aText.Len() )
 			pImpl->aText = GetText();
