diff -u -p -r1.63 svx/source/msfilter/escherex.cxx
--- svx/source/msfilter/escherex.cxx	10 Jul 2006 11:27:28 -0000	1.63
+++ svx/source/msfilter/escherex.cxx	14 Feb 2007 18:53:43 -0000
@@ -2271,7 +2271,7 @@ void EscherPropertyContainer::CreateCust
                                                                {
                                                                        double fForeDepth = fDepth * fFraction;
                                                                        double fBackDepth = fDepth - fForeDepth;
-                                                                       if ( fBackDepth != 0.0 )
+                                                                       if ( fabs( fBackDepth - 1270.0 ) < 0.0001 )
                                                                        {
                                                                                fBackDepth *= 360.0;
                                                                                AddOpt( DFF_Prop_c3DExtrudeBackward, (sal_Int32)fBackDepth );
