Index: svx/source/msfilter/svxmsbas.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/svxmsbas.cxx,v
retrieving revision 1.15
diff -u -p -r1.15 svxmsbas.cxx
--- svx/source/msfilter/svxmsbas.cxx    3 Feb 2006 18:30:38 -0000       1.15
+++ svx/source/msfilter/svxmsbas.cxx    29 Mar 2006 14:34:21 -0000
@@ -375,10 +379,16 @@ BOOL SvxImportMSVBasic::ImportCode_Impl(
                         const char cLineEnd = bMac ? '\x0D' : '\x0A';
                         const String sAttribute(String::CreateFromAscii(
                             bAsComment ? "Rem Attribute" : "Attribute"));
-						while (STRING_NOTFOUND != (nBegin =	pStr->Search(sAttribute)))
+						nBegin = 0;
+						while (STRING_NOTFOUND != (nBegin =	pStr->Search(sAttribute, nBegin)))
 						{
                             if ((nBegin) && pStr->GetChar(nBegin-1) != cLineEnd)
+							{
+								// Need to skip instances of Attribute
+								// that are NOT Attribute statements
+								nBegin += sAttribute.Len();
                                 continue;
+							}
 							xub_StrLen nEnd = pStr->Search(cLineEnd ,nBegin);
                             // DR #i26521# catch STRING_NOTFOUND, will loop endless otherwise
                             if( nEnd == STRING_NOTFOUND )
