Index: svx/source/editeng/forbiddencharacterstable.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/editeng/forbiddencharacterstable.cxx,v
retrieving revision 1.1
diff -u -p -r1.1 forbiddencharacterstable.cxx
--- svx/source/editeng/forbiddencharacterstable.cxx	9 Mar 2001 16:48:03 -0000	1.1
+++ svx/source/editeng/forbiddencharacterstable.cxx	21 May 2005 16:16:03 -0000
@@ -88,8 +88,12 @@ const com::sun::star::i18n::ForbiddenCha
 	ForbiddenCharactersInfo* pInf = Get( nLanguage );
 	if ( !pInf && bGetDefault && mxMSF.is() )
 	{
+		const SvxForbiddenCharactersTableImpl *pConstImpl = 
+			dynamic_cast<const SvxForbiddenCharactersTableImpl*>(this);
+		SvxForbiddenCharactersTableImpl* pImpl = 
+			const_cast<SvxForbiddenCharactersTableImpl*>(pConstImpl);
 		pInf = new ForbiddenCharactersInfo;
-		((SvxForbiddenCharactersTableImpl*)this)->Insert( nLanguage, pInf );
+		pImpl->Insert( nLanguage, pInf );
 		pInf->bTemporary = TRUE;
 		LocaleDataWrapper aWrapper( mxMSF, SvxCreateLocale( nLanguage ) );
 		pInf->aForbiddenChars = aWrapper.getForbiddenCharacters();
Index: svx/source/unoedit/unotext2.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/unoedit/unotext2.cxx,v
retrieving revision 1.18
diff -u -p -r1.18 unotext2.cxx
--- svx/source/unoedit/unotext2.cxx	23 Mar 2005 13:22:22 -0000	1.18
+++ svx/source/unoedit/unotext2.cxx	21 May 2005 16:16:24 -0000
@@ -95,7 +95,7 @@ using namespace ::com::sun::star;
 SvxUnoTextContentEnumeration::SvxUnoTextContentEnumeration( const SvxUnoTextBase& _rText ) throw()
 : rText( _rText )
 {
-	xParentText =  (text::XText*)&_rText;
+	xParentText = const_cast<SvxUnoTextBase*>(&rText);
 	if( rText.GetEditSource() )
 		pEditSource = rText.GetEditSource()->Clone();
 	else
@@ -173,7 +173,7 @@ SvxUnoTextContent::SvxUnoTextContent( co
 	aDisposeListeners(aDisposeContainerMutex),
 	bDisposing( sal_False )
 {
-	xParentText =  (text::XText*)&rText;
+	xParentText = const_cast<SvxUnoTextBase*>(&rText);
 	SetSelection( ESelection( nParagraph,0, nParagraph, GetEditSource()->GetTextForwarder()->GetTextLen( nParagraph ) ) );
 }
 
@@ -439,7 +439,7 @@ uno::Sequence< OUString > SAL_CALL SvxUn
 // ====================================================================
 
 SvxUnoTextRangeEnumeration::SvxUnoTextRangeEnumeration( const SvxUnoTextBase& rText, sal_uInt16 nPara ) throw()
-:	xParentText(  (text::XText*)&rText ),
+:	xParentText(  const_cast<SvxUnoTextBase*>(&rText) ),
 	rParentText( rText ),
 	nParagraph( nPara ),
 	nNextPortion( 0 )
@@ -530,7 +530,7 @@ uno::Reference< uno::XInterface > SvxUno
 
 SvxUnoTextCursor::SvxUnoTextCursor( const SvxUnoTextBase& rText ) throw()
 :	SvxUnoTextRangeBase(rText),
-	xParentText( (text::XText*)&rText)
+	xParentText(  const_cast<SvxUnoTextBase*>(&rText))
 {
 }
 
