Index: svx/source/msfilter/msdffimp.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/msfilter/msdffimp.cxx,v
retrieving revision 1.135
diff -u -p -r1.135 msdffimp.cxx
--- svx/source/msfilter/msdffimp.cxx	19 Jun 2006 16:19:47 -0000	1.135
+++ svx/source/msfilter/msdffimp.cxx	2 Jan 2007 16:05:09 -0000
@@ -2222,8 +2222,8 @@ void DffPropertyReader::ApplyCustomShape
 		if ( IsProperty( DFF_Prop_c3DExtrudeBackward ) || IsProperty( DFF_Prop_c3DExtrudeForward ) )
 		{
 			const rtl::OUString	sDepth( RTL_CONSTASCII_USTRINGPARAM ( "Depth" ) );
-			double fBackDepth = (double)((sal_Int32)GetPropertyValue( DFF_Prop_c3DExtrudeBackward, 1270 * 360 )) / 360.0;
-			double fForeDepth = (double)((sal_Int32)GetPropertyValue( DFF_Prop_c3DExtrudeForward ), 0 ) / 360.0;
+			double fBackDepth = (double)((sal_Int32)(0x7fffffff & GetPropertyValue( DFF_Prop_c3DExtrudeBackward, 1270 * 360 ))) / 360.0;
+			double fForeDepth = (double)((sal_Int32)(0x7fffffff & GetPropertyValue( DFF_Prop_c3DExtrudeForward )), 0 ) / 360.0;
 			double fDepth = fBackDepth + fForeDepth;
                        double fFraction = fDepth != 0.0 ? fForeDepth / fDepth : 0;
 			EnhancedCustomShapeParameterPair aDepthParaPair;
