diff --git a/svx/source/form/fmshimp.cxx b/svx/source/form/fmshimp.cxx
index 9040ddc..20168a9 100644
--- svx/source/form/fmshimp.cxx
+++ svx/source/form/fmshimp.cxx
@@ -2752,7 +2752,7 @@ void FmXFormShell::SetDesignMode(sal_Bool bDesign)
 	{
         Reference< XIndexAccess > xForms;
 		if ( bDesign )
-			xForms = xForms.query( pPage->GetForms( true ) );
+			xForms = xForms.query( pPage->GetForms( false ) );
 
         ResetForms( xForms, sal_False );
 	}
diff --git a/svx/source/form/fmview.cxx b/svx/source/form/fmview.cxx
index 84f6ae4..9d881db 100644
--- svx/source/form/fmview.cxx
+++ svx/source/form/fmview.cxx
@@ -547,6 +547,10 @@ void FmFormView::DeactivateControls(SdrPageView* pPageView)
 //------------------------------------------------------------------------
 void FmFormView::ObjectCreated(FmFormObj* pObj)
 {
+	// it is valid that m_xForms is not initialized yet - initialize it now
+	if ( IsDesignMode() && pFormShell && !pFormShell->GetImpl()->m_xForms.is() )
+		pFormShell->GetImpl()->SetDesignMode( sal_True );
+
 	Reference< XPropertySet > xSet( pObj->GetUnoControlModel(), UNO_QUERY );
 	if ( !xSet.is() )
 		return;
