--- sw/source/core/doc/dbgoutsw.cxx	2008-02-01 11:57:43.000000000 +0100
+++ sw/source/core/doc/dbgoutsw.cxx	2008-02-01 13:44:53.000000000 +0100
@@ -295,7 +295,7 @@
         if (!bFirst)
             aStr += String(", ", RTL_TEXTENCODING_ASCII_US);
 
-        if ((sal_uInt32)pItem != 0xffffffff)
+        if ((sal_IntPtr)pItem != (sal_IntPtr)-1)
             aStr += lcl_dbg_out(*pItem);
         else
             aStr += String("invalid", RTL_TEXTENCODING_ASCII_US);
--- sw/source/core/text/frmform.cxx	2008-02-01 11:57:43.000000000 +0100
+++ sw/source/core/text/frmform.cxx	2008-02-01 13:57:30.000000000 +0100
@@ -2208,7 +2208,7 @@
     {
         //DBG_LOOP; shadows declaration above.
 		//resolved into:
-#if OSL_DEBUG_LEVEL > 1
+#ifndef PRODUCT
 		DbgLoop aDbgLoop2( (const void*) this );
 #endif
 		nStart = aLine.FormatLine( nStart );
--- sw/source/core/text/frmpaint.cxx	2008-02-01 11:57:43.000000000 +0100
+++ sw/source/core/text/frmpaint.cxx	2008-02-01 14:03:30.000000000 +0100
@@ -789,7 +789,7 @@
 			{
 				//DBG_LOOP; shadows declaration above.
 				//resolved into:
-#if  OSL_DEBUG_LEVEL > 1
+#ifndef PRODUCT
 				DbgLoop aDbgLoop2( (const void*) this );
 #endif
 				aLine.DrawTextLine( rRect, aClip, IsUndersized() );
