Index: sw/source/core/graphic/ndgrf.cxx
===================================================================
RCS file: /cvs/sw/sw/source/core/graphic/ndgrf.cxx,v
retrieving revision 1.33
diff -u -r1.33 ndgrf.cxx
--- sw/source/core/graphic/ndgrf.cxx
+++ sw/source/core/graphic/ndgrf.cxx
@@ -1138,6 +1149,10 @@
 	if( !aUserData.Len() )
         return;
 
+	if (aNewStrmName.Len()>0) {
+	    aUserData=aNewStrmName;
+	}
+
 	String aProt( RTL_CONSTASCII_STRINGPARAM( "vnd.sun.star.Package:" ) );
 	if( 0 == aUserData.CompareTo( aProt, aProt.Len() ) )
 	{
