fcvs diff: Diffing .
Index: sw/source/filter/html/css1atr.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/html/css1atr.cxx,v
retrieving revision 1.30
diff -u -p -r1.30 css1atr.cxx
--- sw/source/filter/html/css1atr.cxx	9 Sep 2005 05:40:17 -0000	1.30
+++ sw/source/filter/html/css1atr.cxx	11 Mar 2006 01:09:24 -0000
@@ -367,7 +367,7 @@ void SwHTMLWriter::OutCSS1_Property( con
 	{
 		bFirstCSS1Rule = FALSE;
 		OutNewLine();
-		HTMLOutFuncs::Out_AsciiTag( Strm(), sHTML_style );
+		Strm() << '<' << sHTML_style << ' ' << sHTML_O_type << "=\"text/css\">";
 		OutNewLine();
 		Strm() << '<' << sHTML_comment;
 
