--- sw_save/source/filter/ww8/ww8par.cxx	2007-11-26 18:30:58.000000000 +0100
+++ sw/source/filter/ww8/ww8par.cxx	2008-02-26 03:55:14.000000000 +0100
@@ -2768,8 +2728,10 @@
             else if (!nInTable)
             {
                 // Always insert a txtnode for a column break, e.g. ##
-                AppendTxtNode(*pPaM->GetPoint());
-                rDoc.Insert(*pPaM, SvxFmtBreakItem(SVX_BREAK_COLUMN_BEFORE, RES_BREAK), 0);
+		SwCntntNode *pCntNd=pPaM->GetCntntNode();
+		if (pCntNd!=NULL && pCntNd->Len()>0) // if par is empty not break is needed                
+		    AppendTxtNode(*pPaM->GetPoint());
+		rDoc.Insert(*pPaM, SvxFmtBreakItem(SVX_BREAK_COLUMN_BEFORE, RES_BREAK), 0);
             }
             break;
         case 0x7:
