Index: source/filter/ww8/ww8par4.cxx
===================================================================
RCS file: /cvs/sw/sw/source/filter/ww8/ww8par4.cxx,v
retrieving revision 1.53
diff -u -r1.53 ww8par4.cxx
--- source/filter/ww8/ww8par4.cxx	19 Apr 2006 13:42:38 -0000	1.53
+++ source/filter/ww8/ww8par4.cxx	14 Jul 2006 14:55:18 -0000
@@ -590,6 +590,7 @@
         {
             pPlcxMan->HasCharSprm(69, aResult);
             pSprmCIbstRMark = aResult.empty() ? 0 : aResult.back();
+            aResult.clear();
             pPlcxMan->HasCharSprm(70, aResult);
             pSprmCDttmRMark = aResult.empty() ? 0 : aResult.back();
         }
@@ -597,6 +598,7 @@
         {
             pPlcxMan->HasCharSprm( bIns ? 0x4804 : 0x4863, aResult);
             pSprmCIbstRMark = aResult.empty() ? 0 : aResult.back();
+            aResult.clear();
             pPlcxMan->HasCharSprm( bIns ? 0x6805 : 0x6864, aResult);
             pSprmCDttmRMark = aResult.empty() ? 0 : aResult.back();
         }
