--- config_office/configure.in.orig	2006-04-26 22:44:35.000000000 +0200
+++ config_office/configure.in	2006-04-26 23:04:34.000000000 +0200
@@ -3723,7 +3723,8 @@
 	test "$with_system_neon" != "no"; then
         AC_MSG_RESULT([external])
         PKG_CHECK_MODULES(NEON, neon >= 0.24.0)
-        PKG_CHECK_MODULES(NEON, neon < 0.26.0, , AC_MSG_ERROR([you need neon 0.24.x or 0.25.x for system-neon]))
+        PKG_CHECK_MODULES(NEON, neon < 0.27.0, , AC_MSG_ERROR([you need neon 0.24.x to 0.26.x for system-neon]))
+        NEON_VERSION="`$PKG_CONFIG --modversion neon | $SED 's/\.//g'`"
         NEON_CFLAGS="$NEON_CFLAGS -DSYSTEM_NEON -DUSE_DAV_LOCKS=1"
         SYSTEM_NEON=YES
 else
@@ -3731,9 +3732,11 @@
     SYSTEM_NEON=NO
     NEON_LIBS=-lneon
     NEON_CFLAGS=
+    NEON_VERSION=0247
     BUILD_TYPE="$BUILD_TYPE NEON"
 fi
 AC_SUBST(SYSTEM_NEON)
+AC_SUBST(NEON_VERSION)
 AC_SUBST(NEON_LIBS)
 AC_SUBST(NEON_CFLAGS)
 fi
--- config_office/set_soenv.in.orig	2006-04-26 22:52:38.000000000 +0200
+++ config_office/set_soenv.in	2006-04-26 22:53:15.000000000 +0200
@@ -1804,6 +1804,7 @@
 ToFile( "SYSTEM_SNDFILE",    "@SYSTEM_SNDFILE@",   "e" );
 ToFile( "SYSTEM_PORTAUDIO",  "@SYSTEM_PORTAUDIO@", "e" );
 ToFile( "SYSTEM_NEON",       "@SYSTEM_NEON@",      "e" );
+ToFile( "NEON_VERSION",      "@NEON_VERSION@",     "e" );
 ToFile( "NEON_LIBS",         "@NEON_LIBS@",        "e" );
 ToFile( "NEON_CFLAGS",       "@NEON_CFLAGS@",      "e" );
 ToFile( "DISABLE_NEON",      "@DISABLE_NEON@",     "e" );
diff -Nur ucb.orig/source/ucp/webdav/LinkSequence.cxx ucb/source/ucp/webdav/LinkSequence.cxx
--- ucb.orig/source/ucp/webdav/LinkSequence.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/LinkSequence.cxx	2006-04-26 23:00:53.000000000 +0200
@@ -177,7 +177,7 @@
                       rInData.getStr() + nStart,
                       nEnd - nStart + TOKEN_LENGTH );
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 	success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
diff -Nur ucb.orig/source/ucp/webdav/LockEntrySequence.cxx ucb/source/ucp/webdav/LockEntrySequence.cxx
--- ucb.orig/source/ucp/webdav/LockEntrySequence.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/LockEntrySequence.cxx	2006-04-26 23:00:53.000000000 +0200
@@ -202,7 +202,7 @@
                        rInData.getStr() + nStart,
                        nEnd - nStart + TOKEN_LENGTH );
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
diff -Nur ucb.orig/source/ucp/webdav/LockSequence.cxx ucb/source/ucp/webdav/LockSequence.cxx
--- ucb.orig/source/ucp/webdav/LockSequence.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/LockSequence.cxx	2006-04-26 23:00:53.000000000 +0200
@@ -335,7 +335,7 @@
                       rInData.getStr() + nStart,
                       nEnd - nStart + TOKEN_LENGTH );
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
diff -Nur ucb.orig/source/ucp/webdav/makefile.mk ucb/source/ucp/webdav/makefile.mk
--- ucb.orig/source/ucp/webdav/makefile.mk	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/makefile.mk	2006-04-26 22:56:47.000000000 +0200
@@ -61,12 +61,9 @@
 CFLAGS+= $(LIBXML_CFLAGS)
 .ENDIF
 
+CFLAGS+= -DNEON_VERSION=$(NEON_VERSION)
 .IF "$(SYSTEM_NEON)" == "YES"
 CFLAGS+= $(NEON_CFLAGS)
-TWOFIVE:=$(shell pkg-config --atleast-version=0.25.0 neon && echo YES || echo NO)
-.IF "$(TWOFIVE)" == "YES"
-CFLAGS+=-DNEONTWOFIVE
-.ENDIF
 .ELSE
 NEONINCDIR=external$/neon
 CFLAGS+= -I$(SOLARINCDIR)$/$(NEONINCDIR)
diff -Nur ucb.orig/source/ucp/webdav/NeonHeadRequest.cxx ucb/source/ucp/webdav/NeonHeadRequest.cxx
--- ucb.orig/source/ucp/webdav/NeonHeadRequest.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/NeonHeadRequest.cxx	2006-04-26 23:00:53.000000000 +0200
@@ -51,7 +51,7 @@
 using namespace webdav_ucp;
 using namespace com::sun::star;
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 static void process_headers(ne_request *req,
                             DAVResource &rResource,
                             const std::vector< ::rtl::OUString > &rHeaderNames)
@@ -190,14 +190,14 @@
                                             inPath,
                                             RTL_TEXTENCODING_UTF8 ) );
 
-#ifndef NEONTWOFIVE
+#if NEON_VERSION < 0250
     NeonHeadRequestContext aCtx( &ioResource, &inHeaderNames );
     ne_add_response_header_catcher( req, NHR_ResponseHeaderCatcher, &aCtx );
 #endif
 
     nError = ne_request_dispatch( req );
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
     process_headers(req, ioResource, inHeaderNames);
 #endif
 
diff -Nur ucb.orig/source/ucp/webdav/NeonPropFindRequest.cxx ucb/source/ucp/webdav/NeonPropFindRequest.cxx
--- ucb.orig/source/ucp/webdav/NeonPropFindRequest.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/NeonPropFindRequest.cxx	2006-04-26 23:08:37.000000000 +0200
@@ -175,11 +175,19 @@
 
 // -------------------------------------------------------------------
 extern "C" void NPFR_propfind_results( void* userdata,
+#if NEON_VERSION >= 0260
+                                       const ne_uri* href_uri,
+#else
                                        const char* href,
+#endif
                                        const NeonPropFindResultSet* set )
 {
 	// @@@ href is not the uri! DAVResource ctor wants uri!
 
+#if NEON_VERSION >= 0260
+	// href should be free'd? says header ...
+	char* href = ne_uri_unparse(href_uri);
+#endif
 	DAVResource theResource(
 						OStringToOUString( href, RTL_TEXTENCODING_UTF8 ) );
 
@@ -209,11 +217,19 @@
 
 // -------------------------------------------------------------------
 extern "C" void NPFR_propnames_results( void* userdata,
+#if NEON_VERSION >= 0260
+                                        const ne_uri* href_uri,
+#else
                                         const char* href,
+#endif
                                         const NeonPropFindResultSet* results )
 {
 	// @@@ href is not the uri! DAVResourceInfo ctor wants uri!
 
+#if NEON_VERSION >= 0260
+	// href should be free'd? says header ...
+	char* href = ne_uri_unparse(href_uri);
+#endif
 	// Create entry for the resource.
 	DAVResourceInfo theResource(
 						OStringToOUString( href, RTL_TEXTENCODING_UTF8 ) );
diff -Nur ucb.orig/source/ucp/webdav/NeonSession.cxx ucb/source/ucp/webdav/NeonSession.cxx
--- ucb.orig/source/ucp/webdav/NeonSession.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/NeonSession.cxx	2006-04-26 23:00:53.000000000 +0200
@@ -221,7 +221,7 @@
 // ResponseBlockReader
 // A simple Neon response_block_reader for use with an XInputStream
 // -------------------------------------------------------------------
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 extern "C" int NeonSession_ResponseBlockReader
 #else
 extern "C" void NeonSession_ResponseBlockReader
@@ -244,7 +244,7 @@
                 xInputStream->AddToStream( inBuf, inLen );
         }
 	}
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 	return 0;
 #endif
 }
@@ -253,7 +253,7 @@
 // ResponseBlockWriter
 // A simple Neon response_block_reader for use with an XOutputStream
 // -------------------------------------------------------------------
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 extern "C" int NeonSession_ResponseBlockWriter
 #else
 extern "C" void NeonSession_ResponseBlockWriter
@@ -273,7 +273,7 @@
             xOutputStream->writeBytes( aSeq );
         }
 	}
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 	return 0;
 #endif
 }
@@ -1538,7 +1538,7 @@
 }
 */
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 static void run_header_handler(ne_request *req, ne_header_handler handler, void *userdata)
 {
     void *cursor = NULL;
@@ -1566,13 +1566,13 @@
     //struct get_context ctx;
     ne_request * req = ne_request_create( sess, "GET", uri );
     int ret;
-#ifndef NEONTWOFIVE
+#if NEON_VERSION < 0250
     ne_add_response_header_catcher( req, handler, userdata );
 #endif
     ne_add_response_body_reader( req, ne_accept_2xx, reader, userdata );
 
     ret = ne_request_dispatch( req );
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
     run_header_handler(req, handler, userdata);
 #endif
 
diff -Nur ucb.orig/source/ucp/webdav/NeonSession.hxx ucb/source/ucp/webdav/NeonSession.hxx
--- ucb.orig/source/ucp/webdav/NeonSession.hxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/NeonSession.hxx	2006-04-26 22:59:20.000000000 +0200
@@ -49,7 +49,7 @@
 #include "NeonTypes.hxx"
 #endif
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
 typedef void (*ne_header_handler)(void *userdata, const char *value);
 #endif
 
diff -Nur ucb.orig/source/ucp/webdav/NeonUri.cxx ucb/source/ucp/webdav/NeonUri.cxx
--- ucb.orig/source/ucp/webdav/NeonUri.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/NeonUri.cxx	2006-04-26 23:08:37.000000000 +0200
@@ -54,9 +54,15 @@
 // the string fields of ne_uri are char*, not const char*
 # pragma disable_warn
 # endif
+#if NEON_VERSION >= 0260
+ne_uri NeonUri::sUriDefaultsHTTP  = { "http",  NULL, NULL, DEFAULT_HTTP_PORT,  NULL };
+ne_uri NeonUri::sUriDefaultsHTTPS = { "https", NULL, NULL, DEFAULT_HTTPS_PORT, NULL };
+ne_uri NeonUri::sUriDefaultsFTP   = { "ftp",   NULL, NULL, DEFAULT_FTP_PORT,   NULL };
+#else
 ne_uri NeonUri::sUriDefaultsHTTP  = { "http",  NULL, DEFAULT_HTTP_PORT,  NULL, NULL };
 ne_uri NeonUri::sUriDefaultsHTTPS = { "https", NULL, DEFAULT_HTTPS_PORT, NULL, NULL };
 ne_uri NeonUri::sUriDefaultsFTP   = { "ftp",   NULL, DEFAULT_FTP_PORT,   NULL, NULL };
+#endif
 # if defined __SUNPRO_CC
 # pragma enable_warn
 #endif
@@ -132,7 +138,11 @@
                     pUri->scheme ? pUri->scheme : pUriDefs->scheme,
                     RTL_TEXTENCODING_UTF8 );
     mUserInfo = rtl::OStringToOUString(
+#if NEON_VERSION >= 0260
+                    pUri->userinfo ? pUri->userinfo : pUriDefs->userinfo,
+#else
                     pUri->authinfo ? pUri->authinfo : pUriDefs->authinfo,
+#endif
                     RTL_TEXTENCODING_UTF8 );
     mHostName = rtl::OStringToOUString(
                     pUri->host ? pUri->host : pUriDefs->host,
diff -Nur ucb.orig/source/ucp/webdav/UCBDeadPropertyValue.cxx ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx
--- ucb.orig/source/ucp/webdav/UCBDeadPropertyValue.cxx	2006-04-26 22:46:11.000000000 +0200
+++ ucb/source/ucp/webdav/UCBDeadPropertyValue.cxx	2006-04-26 23:00:53.000000000 +0200
@@ -390,7 +390,7 @@
 
         ne_xml_parse( parser, rInData.getStr(), rInData.getLength() );
 
-#ifdef NEONTWOFIVE
+#if NEON_VERSION >= 0250
         success = !ne_xml_failed( parser );
 #else
         success = !!ne_xml_valid( parser );
