Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.65
diff -u -r1.65 file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	9 Nov 2004 16:51:31 -0000	1.65
+++ scp2/source/ooo/file_library_ooo.scp	3 Dec 2004 09:32:21 -0000
@@ -687,36 +703,6 @@
 
 STD_UNO_LIB_FILE( gid_File_Lib_Fwl , fwl)
 
-// Temporary solution/hack: at the moment libstdc++.so and libgcc_s.so are
-// needed for unxlngi4 environment (setup is linked against it).
-
-#if ! (defined (FREEBSD) || defined (NETBSD) || defined(MACOSX) )
-#if defined( _gcc3 )
-
-File gid_File_Lib_Gcc
-    TXT_FILE_BODY;
-    Name = STRING(libgcc_s.so.1);
-    Dir = gid_Dir_Program;
-    Styles = (PACKED, SETUPZIP);
-End
-
-#endif
-
-#endif
-
-#if ! (defined (FREEBSD) || defined (NETBSD) || defined(MACOSX) )
-#if defined( _gcc3 )
-
-File gid_File_Lib_Stdc
-    TXT_FILE_BODY;
-    Name = STRING(CONCAT2(libstdc++.so.,SHORTSTDCPP3));
-    Dir = gid_Dir_Program;
-    Styles = (PACKED, SETUPZIP);
-End
-
-#endif
-
-#endif
 
 #if defined(LINUX) || defined(FREEBSD) || defined(NETBSD) || defined(IRIX) || defined(MACOSX)
 
Index: scp2/source/ooo/shortcut_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/shortcut_ooo.scp,v
retrieving revision 1.12
diff -u -r1.12 shortcut_ooo.scp
--- scp2/source/ooo/shortcut_ooo.scp	9 Nov 2004 12:09:23 -0000	1.12
+++ scp2/source/ooo/shortcut_ooo.scp	3 Dec 2004 09:48:56 -0000
@@ -364,19 +364,6 @@
 
 #endif
 
-#if ! (defined (FREEBSD) || defined (NETBSD) || defined(MACOSX) )
-#if ( defined( _gcc3 ) && ( defined( C300 ) || defined( C322 ) ))
-
-Shortcut gid_Shortcut_Lib_Stdc
-    FileID = gid_File_Lib_Stdc;
-    Dir = gid_Dir_Program;
-    Name = STRING(libstdc++.so);
-    Styles = (NETWORK,RELATIVE);
-End
-
-#endif
-#endif
-
 #ifdef UNX
 
 Shortcut gid_Shortcut_Lib_Store_Linker
