Index: source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.51
diff -u -p -r1.51 file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	29 Sep 2004 16:10:25 -0000	1.51
+++ scp2/source/ooo/file_library_ooo.scp	5 Oct 2004 13:29:46 -0000
@@ -1073,7 +1073,8 @@ File gid_File_Lib_Mcnttype
   #endif
 End
 
-#if !defined(MACOSX) && !defined(WITHOUT_MOZILLA)
+#if !defined(MACOSX)
+#ifndef SYSTEM_MOZILLA
 
 File gid_File_Lib_Mozab_2
     TXT_FILE_BODY;
@@ -1087,10 +1088,6 @@ File gid_File_Lib_Mozab_2
   #endif
 End
 
-#endif
-
-#if !defined(MACOSX) && !defined(WITHOUT_MOZILLA)
-
 File gid_File_Lib_Mozabdrv
     TXT_FILE_BODY;
   #ifdef UNX
@@ -1103,6 +1100,7 @@ File gid_File_Lib_Mozabdrv
 End
 
 #endif
+#endif
 
 #ifdef WNT
 

Index: source/ooo/file_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_ooo.scp,v
retrieving revision 1.36
diff -u -p -r1.36 file_ooo.scp
--- scp2/source/ooo/file_ooo.scp	14 Sep 2004 11:32:25 -0000	1.36
+++ scp2/source/ooo/file_ooo.scp	5 Oct 2004 13:29:48 -0000
@@ -264,14 +264,12 @@ End
 
 #ifdef UNX
 
-#ifndef WITHOUT_MOZILLA
 File gid_File_Bin_Pluginapp
     BIN_FILE_BODY;
     Dir = gid_Dir_Program;
     Name = "pluginapp.bin";
     Styles = (PACKED);
 End
-#endif
 
 #endif
 
@@ -900,7 +898,7 @@ File gid_File_Mod_Text
     Name = "text.mod";
 End
 
-#ifndef WITHOUT_MOZILLA
+#ifndef SYSTEM_MOZILLA
 
 #ifndef MACOSX
 

Index: source/ooo/makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.8
diff -u -p -r1.8 makefile.mk
--- scp2/source/ooo/makefile.mk	9 Sep 2004 08:10:36 -0000	1.8
+++ scp2/source/ooo/makefile.mk	5 Oct 2004 13:29:49 -0000
@@ -106,14 +106,14 @@ SCPDEFS+=-DSYSTEM_DB3
 SCPDEFS+=-DUSE_SYSTEM_STL
 .ENDIF
 
-.IF "$(WITH_MOZILLA)" == "NO"
-SCPDEFS+=-DWITHOUT_MOZILA
-.ENDIF
-
 .IF "$(SYSTEM_CURL)" == "YES"
 SCPDEFS+=-DSYSTEM_CURL
 .ENDIF
 
+.IF "$(SYSTEM_MOZILLA)" == "YES"
+SCPDEFS+=-DSYSTEM_MOZILLA
+.ENDIF
+
 SCP_PRODUCT_TYPE=osl
 
 PARFILES=                          \

Index: source/ooo/shortcut_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/shortcut_ooo.scp,v
retrieving revision 1.9
diff -u -p -r1.9 shortcut_ooo.scp
--- scp2/source/ooo/shortcut_ooo.scp	8 Sep 2004 16:47:40 -0000	1.9
+++ scp2/source/ooo/shortcut_ooo.scp	5 Oct 2004 13:29:50 -0000
@@ -62,6 +62,7 @@
 #include "macros.inc"
 
 #ifdef UNX
+#ifndef SYSTEM_LIBXML
 
 Shortcut gid_Shortcut_Lib_Xml2
     FileID = gid_File_Lib_Xml2;
@@ -70,10 +71,6 @@ Shortcut gid_Shortcut_Lib_Xml2
     Styles = (NETWORK,RELATIVE);
 End
 
-#endif
-
-#ifdef UNX
-
 Shortcut gid_Shortcut_Lib_Xml2_2
     ShortcutID = gid_Shortcut_Lib_Xml2;
     Dir = gid_Dir_Program;
@@ -82,6 +79,7 @@ Shortcut gid_Shortcut_Lib_Xml2_2
 End
 
 #endif
+#endif
 
 #ifdef UNX
 
