--- canvas/source/tools/page.cxx.orig	2006-12-27 11:59:14.000000000 +0100
+++ canvas/source/tools/page.cxx	2006-12-27 11:59:03.000000000 +0100
@@ -36,8 +36,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_canvas.hxx"
 
-#include <boost/bind.hpp>
 #include "page.hxx"
+#include <boost/bind.hpp>
 
 namespace canvas
 {
--- canvas/source/tools/pagemanager.cxx.orig	2006-12-27 12:01:26.000000000 +0100
+++ canvas/source/tools/pagemanager.cxx	2006-12-27 12:01:36.000000000 +0100
@@ -36,8 +36,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_canvas.hxx"
 
-#include <boost/bind.hpp>
 #include "pagemanager.hxx"
+#include <boost/bind.hpp>
 
 namespace canvas
 {
--- canvas/source/tools/surfaceproxy.cxx.orig	2006-12-27 12:02:58.000000000 +0100
+++ canvas/source/tools/surfaceproxy.cxx	2006-12-27 12:03:13.000000000 +0100
@@ -36,11 +36,11 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_canvas.hxx"
 
-#include <boost/bind.hpp>
 #include <basegfx/polygon/b2dpolygoncutandtouch.hxx>
 #include <basegfx/polygon/b2dpolygontriangulator.hxx>
 #include <basegfx/polygon/b2dpolypolygontools.hxx>
 #include "surfaceproxy.hxx"
+#include <boost/bind.hpp>
 
 namespace canvas
 {
--- sc/inc/postit.hxx.orig	2006-12-28 01:31:34.000000000 +0100
+++ sc/inc/postit.hxx	2006-12-28 01:31:44.000000000 +0100
@@ -38,8 +38,6 @@
 #ifndef SC_POSTIT_HXX
 #define SC_POSTIT_HXX
 
-#include <boost/shared_ptr.hpp>
-
 #include <svtools/pathoptions.hxx>
 #include <svtools/useroptions.hxx>
 
@@ -71,6 +69,8 @@
 #include <com/sun/star/drawing/XShape.hpp>
 #endif
 
+#include <boost/shared_ptr.hpp>
+
 //	indicative text length for a note object
 #define SC_NOTE_SMALLTEXT	100
 
--- sc/source/filter/html/htmlpars.cxx.orig	2006-12-28 01:49:56.000000000 +0100
+++ sc/source/filter/html/htmlpars.cxx	2006-12-28 01:50:04.000000000 +0100
@@ -36,8 +36,6 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_sc.hxx"
 
-#include <boost/shared_ptr.hpp>
-
 #define SC_HTMLPARS_CXX
 #include "scitems.hxx"
 #include <svx/eeitem.hxx>
@@ -85,6 +83,8 @@
 #include "document.hxx"
 #include "rangelst.hxx"
 
+#include <boost/shared_ptr.hpp>
+
 
 SV_IMPL_VARARR_SORT( ScHTMLColOffset, ULONG );
 
--- sd/source/ui/slidesorter/inc/controller/SlideSorterController.hxx.orig	2006-12-28 12:50:04.000000000 +0100
+++ sd/source/ui/slidesorter/inc/controller/SlideSorterController.hxx	2006-12-28 13:03:30.000000000 +0100
@@ -36,6 +36,7 @@
 #ifndef SD_SLIDESORTER_SLIDE_SORTER_CONTROLLER_HXX
 #define SD_SLIDESORTER_SLIDE_SORTER_CONTROLLER_HXX
 
+#include <memory>
 #include "model/SlsSharedPageDescriptor.hxx"
 #include "ViewShell.hxx"
 
@@ -51,7 +52,6 @@
 #ifndef _SV_GEN_HXX
 #include <tools/gen.hxx>
 #endif
-#include <memory>
 #include <comphelper/implementationreference.hxx>
 
 class TabBar;
--- sd/source/ui/slidesorter/inc/controller/SlsFocusManager.hxx.orig	2006-12-28 13:07:29.000000000 +0100
+++ sd/source/ui/slidesorter/inc/controller/SlsFocusManager.hxx	2006-12-28 13:07:38.000000000 +0100
@@ -35,13 +35,13 @@
 #ifndef SD_SLIDESORTER_FOCUS_MANAGER_HXX
 #define SD_SLIDESORTER_FOCUS_MANAGER_HXX
 
+#include <vector>
 #include <model/SlsSharedPageDescriptor.hxx>
 
 #include <sal/types.h>
 #ifndef _LINK_HXX
 #include <tools/link.hxx>
 #endif
-#include <vector>
 
 namespace sd { namespace slidesorter { namespace controller {
 
--- sd/source/ui/slidesorter/controller/SlsPageObjectFactory.cxx.orig	2006-12-28 13:11:40.000000000 +0100
+++ sd/source/ui/slidesorter/controller/SlsPageObjectFactory.cxx	2006-12-28 13:11:46.000000000 +0100
@@ -36,12 +36,12 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_sd.hxx"
 
-#include "controller/SlsPageObjectFactory.hxx"
-
 #include "view/SlsPageObject.hxx"
 #include "view/SlsPageObjectViewContact.hxx"
 #include "view/SlsPageObjectViewObjectContact.hxx"
 
+#include "controller/SlsPageObjectFactory.hxx"
+
 #include "sdpage.hxx"
 
 
--- sd/source/ui/slidesorter/inc/controller/SlsPageSelector.hxx.orig	2006-12-28 13:14:01.000000000 +0100
+++ sd/source/ui/slidesorter/inc/controller/SlsPageSelector.hxx	2006-12-28 13:14:08.000000000 +0100
@@ -35,11 +35,11 @@
 #ifndef SD_SLIDESORTER_PAGE_SELECTOR_HXX
 #define SD_SLIDESORTER_PAGE_SELECTOR_HXX
 
-#include "model/SlsSharedPageDescriptor.hxx"
-
 #include <set>
 #include <memory>
 
+#include "model/SlsSharedPageDescriptor.hxx"
+
 class SdPage;
 
 namespace sd { namespace slidesorter { namespace model {
--- sd/source/ui/slidesorter/inc/view/SlsViewOverlay.hxx.orig	2006-12-28 13:21:53.000000000 +0100
+++ sd/source/ui/slidesorter/inc/view/SlsViewOverlay.hxx	2006-12-28 13:22:02.000000000 +0100
@@ -36,11 +36,11 @@
 #ifndef SD_SLIDESORTER_VIEW_OVERLAY_HXX
 #define SD_SLIDESORTER_VIEW_OVERLAY_HXX
 
+#include <vector>
 #include "model/SlsSharedPageDescriptor.hxx"
 
 #include <osl/mutex.hxx>
 #include <tools/gen.hxx>
-#include <vector>
 #include <boost/weak_ptr.hpp>
 
 class OutputDevice;
