--- config_office/configure.in.orig	2006-05-30 22:27:54.000000000 +0200
+++ config_office/configure.in	2006-05-30 22:28:40.000000000 +0200
@@ -3749,6 +3749,8 @@
 	   AC_MSG_RESULT(external)
 	   AC_CHECK_HEADER(portaudio.h, [],
 		[AC_MSG_ERROR(portaudio header not found. install portaudio)], [])
+	   # portaudio needs -lpthread
+	   LIBS="$LIBS -lpthread"
 	   AC_CHECK_LIB(portaudio, Pa_Initialize, [],
 		[AC_MSG_ERROR(portaudio library not found or functional)], [])
 	   AC_MSG_CHECKING(portaudio version)
