--- vcl/source/app/settings.cxx.old	2007-08-31 18:54:34.000000000 +0200
+++ vcl/source/app/settings.cxx	2007-09-03 10:29:36.000000000 +0200
@@ -767,7 +767,15 @@ void StyleSettings::SetPreferredSymbolsS
 
 		for( sal_uInt32 n = 0; n <= STYLE_SYMBOLS_THEMES_MAX; n++ )
 			if ( rNameLowCase.indexOf( ImplSymbolsStyleToName( n ) ) != -1 )
-				SetPreferredSymbolsStyle( n );
+			{
+				if (n == STYLE_SYMBOLS_INDUSTRIAL)
+					// The Industrial OOo icon theme is too old.
+					// The OOo Tango icon theme is closer to the current GNOME Industrial theme rules.
+					SetPreferredSymbolsStyle( STYLE_SYMBOLS_TANGO );
+				else
+					SetPreferredSymbolsStyle( n );
+				return;
+			}
 	}
 }
 
