--- officecfg/registry/schema/org/openoffice/Office/Writer.xcs.orig	2006-07-25 19:42:53.000000000 +0800
+++ officecfg/registry/schema/org/openoffice/Office/Writer.xcs	2007-04-20 11:20:34.000000000 +0800
@@ -1589,6 +1589,13 @@
 					</info>
 					<value>1250</value>
 				</prop>
+				<prop oor:name="IsSquaredPageMode" oor:type="xs:boolean">
+					<info>
+						<desc>Specified which kind of paper mode is used in the whole document </desc>
+						<label>Use squared paper mode</label>
+					</info>
+					<value>true</value>
+				</prop>	
 			</group>
 			<group oor:name="Zoom">
 				<info>
