--- /dev/null	2004-07-26 12:51:14.000000000 +0100
+++ solenv/bin/ooidlbuild	2005-05-19 11:49:40.759785792 +0100
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+if test "z$SRC_ROOT" = "z"; then
+	echo "You need to source the environment first"
+	exit 1;
+fi
+
+echo "re-build offapi IDL to types.rdb"
+cd $SRC_ROOT/offapi
+# types.db often breaks things
+rm -f $INPATH/ucr/types.db
+build.pl debug=true && deliver.pl || exit 1;
+
+echo "re-build C++ headers from types.rdb"
+cd $SRC_ROOT/offuh
+build.pl && deliver.pl || exit 1;
+
+echo "done."
\ No newline at end of file
