diff -rup ../src680-m237.orig/vcl/inc/vcl/salinst.hxx vcl/inc/vcl/salinst.hxx
--- ../src680-m237.orig/vcl/inc/vcl/salinst.hxx	2007-12-05 15:02:42.000000000 +0100
+++ vcl/inc/vcl/salinst.hxx	2007-12-10 11:26:59.000000000 +0100
@@ -108,7 +108,7 @@ public:
     virtual void				DestroyFrame( SalFrame* pFrame ) = 0;
 
     // Object (System Child Window)
-    virtual SalObject*			CreateObject( SalFrame* pParent, SystemWindowData* pWindowData ) = 0;
+    virtual SalObject*			CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow = TRUE ) = 0;
     virtual void				DestroyObject( SalObject* pObject ) = 0;
 
     // VirtualDevice
diff -rup ../src680-m237.orig/vcl/inc/vcl/syschild.hxx vcl/inc/vcl/syschild.hxx
--- ../src680-m237.orig/vcl/inc/vcl/syschild.hxx	2007-12-05 15:02:58.000000000 +0100
+++ vcl/inc/vcl/syschild.hxx	2007-12-10 11:39:25.000000000 +0100
@@ -58,7 +58,7 @@ class VCL_DLLPUBLIC SystemChildWindow : 
 {
 private:
     using Window::ImplInit;
-	SAL_DLLPRIVATE void     ImplInitSysChild( Window* pParent, WinBits nStyle, SystemWindowData *pData );
+	SAL_DLLPRIVATE void     ImplInitSysChild( Window* pParent, WinBits nStyle, SystemWindowData *pData, BOOL bShow = TRUE );
 
     // Copy assignment is forbidden and not implemented.
 	SAL_DLLPRIVATE          SystemChildWindow (const SystemChildWindow &);
@@ -67,7 +67,7 @@ private:
 public:
 							SystemChildWindow( Window* pParent, WinBits nStyle = 0 );
                             // create a SystemChildWindow using the given SystemWindowData
-							SystemChildWindow( Window* pParent, WinBits nStyle, SystemWindowData *pData );
+							SystemChildWindow( Window* pParent, WinBits nStyle, SystemWindowData *pData, BOOL bShow = TRUE );
 							SystemChildWindow( Window* pParent, const ResId& rResId );
 							~SystemChildWindow();
 
diff -rup ../src680-m237.orig/vcl/source/window/syschild.cxx vcl/source/window/syschild.cxx
--- ../src680-m237.orig/vcl/source/window/syschild.cxx	2007-12-05 15:02:58.000000000 +0100
+++ vcl/source/window/syschild.cxx	2007-12-10 11:39:52.000000000 +0100
@@ -128,9 +128,9 @@ long ImplSysChildProc( void* pInst, SalO
 
 // =======================================================================
 
-void SystemChildWindow::ImplInitSysChild( Window* pParent, WinBits nStyle, SystemWindowData *pData )
+void SystemChildWindow::ImplInitSysChild( Window* pParent, WinBits nStyle, SystemWindowData *pData, BOOL bShow )
 {
-	mpWindowImpl->mpSysObj = ImplGetSVData()->mpDefInst->CreateObject( pParent->ImplGetFrame(), pData );
+	mpWindowImpl->mpSysObj = ImplGetSVData()->mpDefInst->CreateObject( pParent->ImplGetFrame(), pData, bShow );
 
 	Window::ImplInit( pParent, nStyle, NULL );
 
@@ -153,10 +153,10 @@ SystemChildWindow::SystemChildWindow( Wi
 
 // -----------------------------------------------------------------------
 
-SystemChildWindow::SystemChildWindow( Window* pParent, WinBits nStyle, SystemWindowData *pData ) :
+SystemChildWindow::SystemChildWindow( Window* pParent, WinBits nStyle, SystemWindowData *pData, BOOL bShow ) :
 	Window( WINDOW_SYSTEMCHILDWINDOW )
 {
-	ImplInitSysChild( pParent, nStyle, pData );
+	ImplInitSysChild( pParent, nStyle, pData, bShow );
 }
 
 // -----------------------------------------------------------------------
diff -rup ../src680-m237.orig/vcl/unx/gtk/app/gtkinst.cxx vcl/unx/gtk/app/gtkinst.cxx
--- ../src680-m237.orig/vcl/unx/gtk/app/gtkinst.cxx	2007-08-30 15:55:37.000000000 +0200
+++ vcl/unx/gtk/app/gtkinst.cxx	2007-12-10 11:31:35.000000000 +0100
@@ -216,14 +216,14 @@ SalFrame* GtkInstance::CreateChildFrame(
 	return pFrame;
 }
 
-SalObject* GtkInstance::CreateObject( SalFrame* pParent, SystemWindowData* pWindowData )
+SalObject* GtkInstance::CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow )
 {
     // there is no method to set a visual for a GtkWidget
     // so we need the X11SalObject in that case
     if( pWindowData )
-        return X11SalObject::CreateObject( pParent, pWindowData );
+        return X11SalObject::CreateObject( pParent, pWindowData, bShow );
     
-    return new GtkSalObject( static_cast<GtkSalFrame*>(pParent) );
+    return new GtkSalObject( static_cast<GtkSalFrame*>(pParent), bShow );
 }
 
 GtkYieldMutex::GtkYieldMutex()
diff -rup ../src680-m237.orig/vcl/unx/gtk/window/gtkobject.cxx vcl/unx/gtk/window/gtkobject.cxx
--- ../src680-m237.orig/vcl/unx/gtk/window/gtkobject.cxx	2007-12-05 15:02:58.000000000 +0100
+++ vcl/unx/gtk/window/gtkobject.cxx	2007-12-10 11:56:58.000000000 +0100
@@ -41,7 +41,7 @@
 #include <plugins/gtk/gtkdata.hxx>
 #include <plugins/gtk/gtkinst.hxx>
 
-GtkSalObject::GtkSalObject( GtkSalFrame* pParent ) 
+GtkSalObject::GtkSalObject( GtkSalFrame* pParent, BOOL bShow ) 
         : m_pSocket( NULL ),
           m_pRegion( NULL )
 {
@@ -49,6 +49,7 @@ GtkSalObject::GtkSalObject( GtkSalFrame*
     {
         // our plug window
         m_pSocket = gtk_drawing_area_new();
+        Show( bShow );
         // insert into container
         gtk_fixed_put( pParent->getFixedContainer(),
                        m_pSocket,
diff -rup ../src680-m237.orig/vcl/unx/headless/svpinst.cxx vcl/unx/headless/svpinst.cxx
--- ../src680-m237.orig/vcl/unx/headless/svpinst.cxx	2007-12-05 15:02:42.000000000 +0100
+++ vcl/unx/headless/svpinst.cxx	2007-12-10 11:28:28.000000000 +0100
@@ -245,7 +245,7 @@ void SvpSalInstance::DestroyFrame( SalFr
     delete pFrame;
 }
 
-SalObject* SvpSalInstance::CreateObject( SalFrame*, SystemWindowData* )
+SalObject* SvpSalInstance::CreateObject( SalFrame*, SystemWindowData*, BOOL )
 {
     return new SvpSalObject();
 }
diff -rup ../src680-m237.orig/vcl/unx/headless/svpinst.hxx vcl/unx/headless/svpinst.hxx
--- ../src680-m237.orig/vcl/unx/headless/svpinst.hxx	2007-12-05 15:02:42.000000000 +0100
+++ vcl/unx/headless/svpinst.hxx	2007-12-10 11:28:06.000000000 +0100
@@ -139,7 +139,7 @@ public:
     virtual void			DestroyFrame( SalFrame* pFrame );
 
     // Object (System Child Window)
-    virtual SalObject*		CreateObject( SalFrame* pParent, SystemWindowData* pWindowData );
+    virtual SalObject*		CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow = TRUE );
     virtual void			DestroyObject( SalObject* pObject );
 
     // VirtualDevice
diff -rup ../src680-m237.orig/vcl/unx/inc/plugins/gtk/gtkinst.hxx vcl/unx/inc/plugins/gtk/gtkinst.hxx
--- ../src680-m237.orig/vcl/unx/inc/plugins/gtk/gtkinst.hxx	2006-08-11 19:47:36.000000000 +0200
+++ vcl/unx/inc/plugins/gtk/gtkinst.hxx	2007-12-10 11:29:14.000000000 +0100
@@ -94,7 +94,7 @@ public:
 
     virtual SalFrame*			CreateFrame( SalFrame* pParent, ULONG nStyle );
     virtual SalFrame*           CreateChildFrame( SystemParentData* pParent, ULONG nStyle );
-    virtual SalObject*			CreateObject( SalFrame* pParent, SystemWindowData* pWindowData );
+    virtual SalObject*			CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow = TRUE );
     virtual SalSystem*			CreateSalSystem();
 };
 
diff -rup ../src680-m237.orig/vcl/unx/inc/plugins/gtk/gtkobject.hxx vcl/unx/inc/plugins/gtk/gtkobject.hxx
--- ../src680-m237.orig/vcl/unx/inc/plugins/gtk/gtkobject.hxx	2007-12-05 15:02:58.000000000 +0100
+++ vcl/unx/inc/plugins/gtk/gtkobject.hxx	2007-12-10 11:53:48.000000000 +0100
@@ -60,7 +60,7 @@ class GtkSalObject : public SalObject
     static gboolean		signalFocus( GtkWidget*, GdkEventFocus*, gpointer );
     static void			signalDestroy( GtkObject*, gpointer );
 public:
-    GtkSalObject( GtkSalFrame* pParent );
+    GtkSalObject( GtkSalFrame* pParent, BOOL bShow = TRUE );
     virtual ~GtkSalObject();
 
     // overload all pure virtual methods
diff -rup ../src680-m237.orig/vcl/unx/inc/salinst.h vcl/unx/inc/salinst.h
--- ../src680-m237.orig/vcl/unx/inc/salinst.h	2007-12-05 15:02:42.000000000 +0100
+++ vcl/unx/inc/salinst.h	2007-12-10 11:59:06.000000000 +0100
@@ -89,7 +89,7 @@ public:
     virtual SalFrame*      	CreateFrame( SalFrame* pParent, ULONG nStyle );
     virtual void				DestroyFrame( SalFrame* pFrame );
 
-    virtual SalObject*			CreateObject( SalFrame* pParent, SystemWindowData* pWindowData );
+    virtual SalObject*			CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow = TRUE );
     virtual void				DestroyObject( SalObject* pObject );
 
     virtual SalVirtualDevice*	CreateVirtualDevice( SalGraphics* pGraphics,
diff -rup ../src680-m237.orig/vcl/unx/inc/salobj.h vcl/unx/inc/salobj.h
--- ../src680-m237.orig/vcl/unx/inc/salobj.h	2007-11-09 11:20:33.000000000 +0100
+++ vcl/unx/inc/salobj.h	2007-12-10 11:35:47.000000000 +0100
@@ -93,7 +93,7 @@ public:
 	BOOL			mbVisible;
 
 	static VCL_DLLPUBLIC long Dispatch( XEvent* pEvent );
-    static VCL_DLLPUBLIC X11SalObject* CreateObject( SalFrame* pParent, SystemWindowData* pWindowData );
+    static VCL_DLLPUBLIC X11SalObject* CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow = TRUE );
 
     X11SalObject();
     virtual ~X11SalObject();
diff -rup ../src680-m237.orig/vcl/unx/source/window/salobj.cxx vcl/unx/source/window/salobj.cxx
--- ../src680-m237.orig/vcl/unx/source/window/salobj.cxx	2007-11-09 11:20:46.000000000 +0100
+++ vcl/unx/source/window/salobj.cxx	2007-12-10 11:35:00.000000000 +0100
@@ -74,12 +74,12 @@
 // =======================================================================
 // SalInstance member to create and destroy a SalObject
 
-SalObject* X11SalInstance::CreateObject( SalFrame* pParent, SystemWindowData* pWindowData )
+SalObject* X11SalInstance::CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow )
 {
-    return X11SalObject::CreateObject( pParent, pWindowData );
+    return X11SalObject::CreateObject( pParent, pWindowData, bShow );
 }
 
-X11SalObject* X11SalObject::CreateObject( SalFrame* pParent, SystemWindowData* pWindowData )
+X11SalObject* X11SalObject::CreateObject( SalFrame* pParent, SystemWindowData* pWindowData, BOOL bShow )
 {
 	int error_base, event_base;
 	X11SalObject*		pObject  = new X11SalObject();
@@ -183,8 +185,10 @@ X11SalObject* X11SalObject::CreateObject
     }
 
     pSalDisp->GetXLib()->PushXErrorLevel( true );
-    XMapWindow( pDisp, pObject->maPrimary );
-	XMapWindow( pDisp, pObject->maSecondary );
+    if( bShow ) {
+        XMapWindow( pDisp, pObject->maSecondary );
+        XMapWindow( pDisp, pObject->maPrimary );
+    }
 
 	pObjData->pDisplay		= pDisp;
 	pObjData->aWindow		= pObject->maSecondary;
@@ -421,13 +425,17 @@ X11SalObject::Show( BOOL bVisible )
 	if	( ! maSystemChildData.aWindow )
 		return;
 
-	if ( bVisible ) 
+	if ( bVisible ) {
+		XMapWindow( (Display*)maSystemChildData.pDisplay,
+                    maSecondary );
 		XMapWindow( (Display*)maSystemChildData.pDisplay,
                     maPrimary );
-	else
+	} else {
 		XUnmapWindow( (Display*)maSystemChildData.pDisplay,
-                    maPrimary );
-
+                      maPrimary );
+		XUnmapWindow( (Display*)maSystemChildData.pDisplay,
+                      maSecondary );
+    }
 	mbVisible = bVisible;
 }
 
