diff -rup ../src680-m237.orig/officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu
--- ../src680-m237.orig/officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu	2007-12-05 15:02:57.000000000 +0100
+++ officecfg/registry/data/org/openoffice/Office/UI/Effects.xcu	2007-12-05 16:29:43.000000000 +0100
@@ -1937,6 +1937,31 @@
           <value xml:lang="en-US">Inside turning cube</value>
         </prop>
       </node>
+      <node oor:name="fall" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Fall</value>
+        </prop>
+      </node>
+      <node oor:name="turn-around" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Turn around</value>
+        </prop>
+      </node>
+      <node oor:name="iris" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Iris</value>
+        </prop>
+      </node>
+      <node oor:name="turn-down" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Turn down</value>
+        </prop>
+      </node>
+      <node oor:name="rochade" oor:op="replace">
+        <prop oor:name="Label" oor:type="xs:string">
+          <value xml:lang="en-US">Rochade</value>
+        </prop>
+      </node>
 		</node>
 	</node>	
 	<node oor:name="Presets">
