diff -rup ../src680-m237.orig/slideshow/source/engine/slideshowimpl.cxx slideshow/source/engine/slideshowimpl.cxx
--- ../src680-m237.orig/slideshow/source/engine/slideshowimpl.cxx	2007-11-09 11:16:38.000000000 +0100
+++ slideshow/source/engine/slideshowimpl.cxx	2007-12-10 16:41:26.000000000 +0100
@@ -1448,10 +1448,12 @@ sal_Bool SlideShowImpl::update( double &
         // hold timer, while processing the queues (ensures
         // same time for all activities and events)
         {
-            comphelper::ScopeGuard const scopeGuard(
-                boost::bind( &canvas::tools::ElapsedTime::releaseTimer,
-                             boost::cref(mpPresTimer) ) );
-            mpPresTimer->holdTimer();
+            if( maActivitiesQueue.size() > 1 ) {
+                comphelper::ScopeGuard const scopeGuard(
+                    boost::bind( &canvas::tools::ElapsedTime::releaseTimer,
+                                 boost::cref(mpPresTimer) ) );
+                mpPresTimer->holdTimer();
+            }
                     
             // process queues
             maEventQueue.process();
diff -rup ../src680-m237.orig/slideshow/source/inc/activitiesqueue.hxx slideshow/source/inc/activitiesqueue.hxx
--- ../src680-m237.orig/slideshow/source/inc/activitiesqueue.hxx	2007-07-17 17:01:06.000000000 +0200
+++ slideshow/source/inc/activitiesqueue.hxx	2007-12-10 16:36:33.000000000 +0100
@@ -101,6 +101,10 @@ namespace slideshow
              */
             ::boost::shared_ptr< ::canvas::tools::ElapsedTime > const &
             getTimer() const { return mpTimer; }
+
+            /** returns number of all activities, waiting, reinserted and dequeued
+             */
+            int size() { return maCurrentActivitiesWaiting.size() + maCurrentActivitiesReinsert.size() + maDequeuedActivities.size(); }
             
         private:
             ::boost::shared_ptr< ::canvas::tools::ElapsedTime > mpTimer;
