--- extras/source/wordbook/makefile.mk.old	2005-09-08 23:20:19.000000000 +0200
+++ extras/source/wordbook/makefile.mk	2006-08-28 23:03:19.850718000 +0200
@@ -47,7 +47,7 @@
 
 ZIP1TARGET      = $(WORDBOOK_ALL_TARGET)
 
-ZIP1LIST        = sun.dic
+ZIP1LIST        = Ubuntu.dic
 
 
 ZIP2TARGET      = $(WORDBOOK_USER_TARGET)
--- officecfg/registry/schema/org/openoffice/Office/Linguistic.xcs.old	2005-09-08 17:15:44.000000000 +0200
+++ officecfg/registry/schema/org/openoffice/Office/Linguistic.xcs	2006-08-28 23:03:51.250718000 +0200
@@ -118,7 +118,7 @@
 						<desc>Specifies the custom dictionaries to be used. The Ignore list (IgnoreAllList (All)) is administered in the RAM only for the current spellcheck.</desc>
 						<label>Custom dictionaries</label>
 					</info>
-					<value oor:separator=";">IgnoreAllList;soffice.dic;standard.dic;sun.dic</value>
+					<value oor:separator=";">IgnoreAllList;soffice.dic;standard.dic;Ubuntu.dic</value>
 				</prop>
 				<prop oor:name="IsUseDictionaryList" oor:type="xs:boolean">
 					<!-- OldLocation: NEW -->
