--- ./svx/prj/d.lst.old	2006-02-15 02:21:52.019949000 +0100
+++ ./svx/prj/d.lst	2006-02-15 02:47:48.049949000 +0100
@@ -43,7 +43,7 @@
 ..\res\cjkintro.bmp %COMMON_DEST%\pck%_EXT%\starsuite_bitmap\intro.bmp
 ..\res\ofaabout.bmp %COMMON_DEST%\pck%_EXT%\staroffice_bitmap\about.bmp
 ..\res\soffice.bmp %COMMON_DEST%\pck%_EXT%\staroffice_bitmap\intro.bmp
-..\res\ooonologointro.bmp %COMMON_DEST%\pck%_EXT%\openoffice\intro.bmp
+..\res\openintro_ubuntu.bmp %COMMON_DEST%\pck%_EXT%\openoffice\intro.bmp
 ..\res\ooonologointro.bmp %COMMON_DEST%\pck%_EXT%\openoffice\nologointro.bmp
 
 mkdir: %_DEST%\inc%_EXT%\svx
--- ./svx/source/intro/ooo.src.old	2006-02-10 16:08:03.000000000 +0100
+++ ./svx/source/intro/ooo.src	2006-02-15 02:49:16.979949000 +0100
@@ -50,7 +50,7 @@
 
 Bitmap RID_DEFAULT_ABOUT_BMP_LOGO
 {
-	File = "openabout.bmp" ;
+	File = "openabout_ubuntu.bmp" ;
 };
 
 String RID_APPTITLE
