--- sfx2/source/appl/sfxhelp.cxx~	2006-02-15 02:53:15.839949000 +0100
+++ sfx2/source/appl/sfxhelp.cxx	2006-02-15 02:55:01.389949000 +0100
@@ -151,7 +151,7 @@
 
 NoHelpErrorBox::NoHelpErrorBox( Window* _pParent ) :
 
-    ErrorBox( _pParent, WB_OK, String( SfxResId( RID_STR_HLPFILENOTEXIST ) ) )
+    ErrorBox( _pParent, WB_OK, String( SfxResId( RID_STR_HLPFILENOTEXIST ) ).Append(String(DEFINE_CONST_UNICODE(".\nPlease install the openoffice.org-help-en-us package or the locale specific help package openoffice.org-help-<language-code>."))) )
 {
     // Error message: "No help available"
 }
