--- sfx2/source/appl/appserv.cxx.old	2006-12-20 16:01:19.846229000 +0100
+++ sfx2/source/appl/appserv.cxx	2006-12-20 16:02:27.166229000 +0100
@@ -119,6 +119,12 @@
 #ifndef _SV_MSGBOX_HXX
 #include <vcl/msgbox.hxx>
 #endif
+#ifndef _COM_SUN_STAR_SYSTEM_XSYSTEMSHELLEXECUTE_HPP_
+#include <com/sun/star/system/XSystemShellExecute.hpp>
+#endif
+#ifndef _COM_SUN_STAR_SYSTEM_SYSTEMSHELLEXECUTEFLAGS_HPP_
+#include <com/sun/star/system/SystemShellExecuteFlags.hpp>
+#endif
 #ifndef _SFXCANCEL_HXX //autogen
 #include <svtools/cancel.hxx>
 #endif
@@ -423,9 +429,39 @@
 			break;
 		}
 
+		case SID_HELP_SUPPORTPAGE:
+		{
+			try {
+				Reference< com::sun::star::system::XSystemShellExecute > xSystemShell
+					( ::comphelper::getProcessServiceFactory()->createInstance( DEFINE_CONST_UNICODE("com.sun.star.system.SystemShellExecute" ) ), UNO_QUERY );
+				if ( xSystemShell.is()) {
+					rtl::OUString aURI( DEFINE_CONST_UNICODE( "https://launchpad.net/distros/ubuntu/feisty/+sources/openoffice.org/+gethelp" ) );
+					xSystemShell->execute( aURI, ::rtl::OUString(), com::sun::star::system::SystemShellExecuteFlags::DEFAULTS );
+				}
+
+				break;
+			} catch (const ::com::sun::star::uno::Exception &) {
+ 			}
+			/* drop through on exception */
+		}
+
+		case SID_HELP_TRANSLATEPAGE:
+		{
+			try {
+				Reference< com::sun::star::system::XSystemShellExecute > xSystemShell
+					( ::comphelper::getProcessServiceFactory()->createInstance( DEFINE_CONST_UNICODE("com.sun.star.system.SystemShellExecute" ) ), UNO_QUERY );
+				if ( xSystemShell.is()) {
+					rtl::OUString aURI( DEFINE_CONST_UNICODE( "https://launchpad.net/distros/ubuntu/feisty/+sources/openoffice.org/+translate" ) );
+					xSystemShell->execute( aURI, ::rtl::OUString(), com::sun::star::system::SystemShellExecuteFlags::DEFAULTS );
+				}
+
+				break;
+			} catch (const ::com::sun::star::uno::Exception &) {
+ 			}
+			/* drop through on exception */
+		}
         // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
         case SID_HELPINDEX:
-        case SID_HELP_SUPPORTPAGE:
         {
             Help* pHelp = Application::GetHelp();
             if ( pHelp )
