--- extras/source/palettes/standard.soc.old1	2001-03-16 14:44:46.000000000 +0100
+++ extras/source/palettes/standard.soc	2006-01-30 15:14:54.000000000 +0100
@@ -88,8 +88,7 @@
  <draw:color draw:name="Dark violet" draw:color="#660066"/>
  <draw:color draw:name="Salmon" draw:color="#ff8080"/>
  <draw:color draw:name="Sea blue" draw:color="#0066cc"/>
- <draw:color draw:name="Sun 1" draw:color="#333366"/>
- <draw:color draw:name="Sun 2" draw:color="#666699"/>
- <draw:color draw:name="Sun 3" draw:color="#9999cc"/>
- <draw:color draw:name="Sun 4" draw:color="#ccccff"/>
+ <draw:color draw:name="Ubuntu orange" draw:color="#ff6309"/>
+ <draw:color draw:name="Ubuntu yellow" draw:color="#ffb515"/>
+ <draw:color draw:name="Ubuntu red" draw:color="#c90016"/>
 </office:color-table>
\ No newline at end of file
