--- extras/source/wordbook/makefile.mk~	2006-03-13 13:44:07.259949000 +0100
+++ extras/source/wordbook/makefile.mk	2006-03-14 00:07:22.129949000 +0100
@@ -47,7 +47,7 @@
 
 ZIP1TARGET      = $(WORDBOOK_ALL_TARGET)
 
-ZIP1LIST        = sun.dic Novell.dic
+ZIP1LIST        = Ubuntu.dic
 
 
 ZIP2TARGET      = $(WORDBOOK_USER_TARGET)
--- officecfg/registry/schema/org/openoffice/Office/Linguistic.xcs~	2006-03-13 13:44:07.259949000 +0100
+++ officecfg/registry/schema/org/openoffice/Office/Linguistic.xcs	2006-03-14 00:08:10.669949000 +0100
@@ -118,7 +118,7 @@
 						<desc>Specifies the custom dictionaries to be used. The Ignore list (IgnoreAllList (All)) is administered in the RAM only for the current spellcheck.</desc>
 						<label>Custom dictionaries</label>
 					</info>
-					<value oor:separator=";">IgnoreAllList;soffice.dic;standard.dic;Novell.dic;sun.dic;corporate.dic</value>
+					<value oor:separator=";">IgnoreAllList;soffice.dic;standard.dic;Ubuntu.dic;corporate.dic</value>
 				</prop>
 				<prop oor:name="IsUseDictionaryList" oor:type="xs:boolean">
 					<!-- OldLocation: NEW -->
