--- basctl/source/inc/basidesh.hxx	2007-01-24 18:24:39.000000000 +0100
+++ basctl/source/inc/basidesh.hxx	2007-02-19 18:12:19.000000000 +0100
@@ -200,7 +200,7 @@ public:
 	void				ExecuteBasic( SfxRequest& rReq );
 	void				ExecuteDialog( SfxRequest& rReq );
 
-	virtual	BOOL		HasUIFeature( ULONG nFeature );
+	virtual	sal_Bool	HasUIFeature( sal_uInt32 nFeature );
 
 	long				CallBasicErrorHdl( StarBASIC* pBasic );
 	long				CallBasicBreakHdl( StarBASIC* pBasic );
