--- svx/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ svx/prj/tests.lst	2007-09-07 18:27:08.000000000 +0200
@@ -0,0 +1,5 @@
+#does not build: sx   svx\qa\unoapi                                      nmake - all sx_qa_unoapi NULL
+
+# older tests:
+#does not build: sx   svx\workben                                        nmake - all sx_workben sx_util NULL
+#does not build: sx   svx\workben\msview                                 nmake - all sx_workben_msview sx_util NULL
