--- desktop/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ desktop/prj/tests.lst	2007-09-07 08:12:53.000000000 +0200
@@ -0,0 +1,2 @@
+dt   desktop\test\deployment\locationtest               nmake - all dt_test_deployment_locationtest NULL
+dt   desktop\test\deployment\options\handler\com\sun\star\comp\extensionoptions nmake - all dt_test_deployment_options_handler_com_sun_star_comp_extensionoptions NULL
