--- sw/prj/tests.lst.qa	1970-01-01 01:00:00.000000000 +0100
+++ sw/prj/tests.lst.qa	2007-09-07 14:32:15.000000000 +0200
@@ -0,0 +1,6 @@
+sw   sw\qa\complex\accessibility                        nmake - all sw_qa_complex_accessibility NULL
+sw   sw\qa\complex\checkColor                           nmake - all sw_qa_complex_checkColor NULL
+sw   sw\qa\complex\indeterminateState                   nmake - all sw_qa_complex_indeterminateState NULL
+sw   sw\qa\complex\writer                               nmake - all sw_qa_complex_writer NULL
+sw   sw\qa\core                                         nmake - all sw_qa_core NULL
+sw   sw\qa\unoapi                                       nmake - all sw_qa_unoapi NULL
