--- svtools/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ svtools/prj/tests.lst	2007-09-07 12:40:56.000000000 +0200
@@ -0,0 +1,4 @@
+#does not link: st   svtools\source\config\test                         nmake - all st_source_config_test st_util NULL
+st   svtools\workben\cui                                nmake - all st_workben_cui st_util NULL
+st   svtools\workben                                    nmake - all st_workben st_util NULL
+st   svtools\workben\treecontrol                        nmake - all st_workben_treecontrol st_util NULL
--- svtools/source/config/test/test.cxx	2007-06-27 23:18:32.000000000 +0200
+++ svtools/source/config/test/test.cxx	2007-09-07 12:33:00.000000000 +0200
@@ -98,7 +98,7 @@
 #endif
 
 #ifndef _RTL_USTRING_
-#include <rtl/ustring>
+#include <rtl/ustring.hxx>
 #endif
 
 #ifndef _RTL_USTRBUF_HXX_
