--- xmlscript/prj/build.lst	2006-10-30 10:09:11.000000000 +0100
+++ xmlscript/prj/build.lst	2007-09-07 12:07:52.000000000 +0200
@@ -1,4 +1,4 @@
-xt	xmlscript :	comphelper tools offapi NULL
+xt	xmlscript :	comphelper tools offapi vcl NULL
 xt	xmlscript							usr1	-	all	xt_mkout NULL
 xt	xmlscript\inc						nmake	-	all	xt_inc NULL
 xt	xmlscript\source\xml_helper			nmake	-	all	xt_xml_helper xt_inc NULL
--- xmlscript/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ xmlscript/prj/tests.lst	2007-09-07 12:07:05.000000000 +0200
@@ -0,0 +1 @@
+xt   xmlscript\test                                     nmake - all xt_test xt_util NULL
--- xmlscript/test/makefile.mk	2007-02-06 15:16:47.000000000 +0100
+++ xmlscript/test/makefile.mk	2007-09-07 12:08:25.000000000 +0200
@@ -58,7 +58,6 @@ APP1OBJS=$(OBJFILES)
 APP1STDLIBS= \
 		$(TOOLSLIB)			\
 		$(SOTLIB)			\
-		$(SVTOOLLIB)			\
 		$(COMPHELPERLIB)		\
 		$(VCLLIB)			\
 		$(CPPULIB)			\
