--- toolkit/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ toolkit/prj/tests.lst	2007-09-07 12:24:34.000000000 +0200
@@ -0,0 +1,4 @@
+ti   toolkit\test\accessibility\ov                      nmake - all ti_test_accessibility_ov ti_util NULL
+ti   toolkit\test\accessibility                         nmake - all ti_test_accessibility ti_util NULL
+ti   toolkit\test\accessibility\tools                   nmake - all ti_test_accessibility_tools ti_util NULL
+#depends on svtools: ti   toolkit\workben                                    nmake - all ti_workben ti_util NULL
--- toolkit/test/accessibility/makefile.mk	2007-01-25 14:17:59.000000000 +0100
+++ toolkit/test/accessibility/makefile.mk	2007-09-07 12:21:44.000000000 +0200
@@ -125,3 +125,4 @@ run: all
 runjar: all dist
 	$(JAVA) -classpath $(CLASSPATH) -jar AccessibilityWorkBench.jar -p $(PORT_NUMBER)
 
+check: all
--- toolkit/test/accessibility/ov/makefile.mk	2007-01-25 14:18:24.000000000 +0100
+++ toolkit/test/accessibility/ov/makefile.mk	2007-09-07 12:21:55.000000000 +0200
@@ -49,3 +49,5 @@ JFLAGS = -deprecation -classpath $(CLASS
 	$(JAVAC) $(JFLAGS) $<
 
 ov : $(JAVA_FILES:b:+".class")
+
+check: all
--- toolkit/test/accessibility/tools/makefile.mk	2007-01-25 14:18:37.000000000 +0100
+++ toolkit/test/accessibility/tools/makefile.mk	2007-09-07 12:22:01.000000000 +0200
@@ -40,3 +40,5 @@ JFLAGS = -deprecation -classpath $(CLASS
 	$(JAVAC) $(JFLAGS) $<
 
 tools : $(JAVA_FILES:b:+".class")
+
+check: all
