--- testshl2/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ testshl2/prj/tests.lst	2007-09-07 09:08:29.000000000 +0200
@@ -0,0 +1,8 @@
+t2   testshl2\workben\codegen                           nmake - all t2_workben_codegen NULL
+t2   testshl2\workben\codegen\w32                       nmake - all t2_workben_codegen_w32 NULL
+t2   testshl2\workben                                   nmake - all t2_workben NULL
+t2   testshl2\workben\dlltest                           nmake - all t2_workben_dlltest NULL
+#does not build t2   testshl2\workben\examples                          nmake - all t2_workben_examples NULL
+t2   testshl2\workben\getopt                            nmake - all t2_workben_getopt t2_tstu NULL
+t2   testshl2\workben\skeleton                          nmake - all t2_workben_skeleton t2_tstu NULL
+#depends on offapi t2   testshl2\workben\uno                               nmake - all t2_workben_uno NULL
--- testshl2/workben/examples/testshl_test.cxx	2006-09-17 05:49:44.000000000 +0200
+++ testshl2/workben/examples/testshl_test.cxx	2007-09-07 09:05:08.000000000 +0200
@@ -146,10 +146,10 @@ void usage()
 #include <cppunit/TestFixture.h>
 #include <cppunit/TestCaller.h>
 #include <cppunit/TestSuite.h>
-#include <cppunit/TestResult.h>
-#include <cppunit/TestResultCollector.h> 
 #include <cppunit/TestAssert.h>
-#include <cppunit/TextTestResult.h>
+#include <cppunit/result/TestResult.h>
+#include <cppunit/result/TestResultCollector.h> 
+#include <cppunit/result/TextTestResult.h>
 
 namespace CppunitTest
 {
