--- vcl/prj/build.lst	2007-07-24 12:40:56.000000000 +0200
+++ vcl/prj/build.lst	2007-09-07 12:01:35.000000000 +0200
@@ -32,4 +32,3 @@ vc	vcl\mac\source\window					nmake	-	m	v
 vc	vcl\mac\source\src						nmake	-	m	vc__srcm vc_inc NULL
 vc	vcl\util								nmake	-	all	vc_util vc__plug.u vc__appa.u vc__appm.m vc__appu.u vc__appw.w vc__gdia.u vc__gdim.m vc__gdiu.u vc__gdiw.w vc__srcm.m vc__srcw.w vc__wina.u vc__winm.m vc__winu.u vc__winw.w vc__gtka.u vc__gtky.u vc__gtkw.u vc__gtkg.u vc__kde.u vc__hl.u vc_app vc_ctrl vc_gdi vc_hlp vc_src vc_win vc_glyphs vc_components NULL
 vc  vcl\util\linksvp                        nmake   -   u   vc_lsvp vc_util NULL
-vc	vcl\workben								nmake	-	all	vc_wrkb vc_util NULL
--- vcl/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ vcl/prj/tests.lst	2007-09-07 12:01:32.000000000 +0200
@@ -0,0 +1,2 @@
+vc   vcl\test                                           nmake - all vc_test vc_util NULL
+vc   vcl\workben                                        nmake - all vc_workben vc_util NULL
