--- tools/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ tools/prj/tests.lst	2007-09-07 14:38:11.000000000 +0200
@@ -0,0 +1,5 @@
+tl   tools\qa\urlobj                                    nmake - all tl_qa_urlobj NULL
+
+# older tests:
+tl   tools\test                                         nmake - all tl_test tl_utl NULL
+tl   tools\workben                                      nmake - all tl_workben tl_utl NULL
--- tools/qa/urlobj/tools_urlobj_test.cxx	2007-09-06 23:05:09.000000000 +0200
+++ tools/qa/urlobj/tools_urlobj_test.cxx	2007-09-07 16:21:20.000000000 +0200
@@ -35,7 +35,7 @@
  
  #include <cppunit/simpleheader.hxx>
  
- #include <urlobj.hxx>
+ #include <tools/urlobj.hxx>
  
  namespace tools_urlobj
  {
