--- salhelper/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ salhelper/prj/tests.lst	2007-09-07 09:00:15.000000000 +0200
@@ -0,0 +1,3 @@
+shp  salhelper\test\Symbols                             nmake - all shp_test_Symbols NULL
+shp  salhelper\test\dynamicloader                       nmake - all shp_test_dynamicloader NULL
+#depends on cppu: shp  salhelper\test\rtti                                nmake - all shp_test_rtti NULL
--- salhelper/test/Symbols/loader.cxx	2001-04-19 17:12:00.000000000 +0200
+++ salhelper/test/Symbols/loader.cxx	2007-09-07 08:55:26.000000000 +0200
@@ -1,6 +1,6 @@
 
-#include  <salhelper/dynload.hxx>
-#include <rtl/ustring>
+#include <salhelper/dynload.hxx>
+#include <rtl/ustring.hxx>
 #include <stdio.h>
 #include "samplelib.hxx"
 
