--- framework/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ framework/prj/tests.lst	2007-09-07 17:49:58.000000000 +0200
@@ -0,0 +1,24 @@
+#does not build: fr   framework\qa\complex\ModuleManager                 nmake - all fr_qa_complex_ModuleManager fr_util NULL
+fr   framework\qa\complex\XUserInputInterception        nmake - all fr_qa_complex_XUserInputInterception fr_util NULL
+fr   framework\qa\complex\api_internal                  nmake - all fr_qa_complex_api_internal fr_util NULL
+fr   framework\qa\complex\broken_document               nmake - all fr_qa_complex_broken_document fr_util NULL
+fr   framework\qa\complex\contextMenuInterceptor        nmake - all fr_qa_complex_contextMenuInterceptor fr_util NULL
+fr   framework\qa\complex\desktop                       nmake - all fr_qa_complex_desktop fr_util NULL
+fr   framework\qa\complex\dispatches\helper             nmake - all fr_qa_complex_dispatches_helper fr_util NULL
+fr   framework\qa\complex\dispatches                    nmake - all fr_qa_complex_dispatches fr_qa_complex_dispatches_helper fr_util NULL
+fr   framework\qa\complex\disposing                     nmake - all fr_qa_complex_disposing fr_util NULL
+fr   framework\qa\complex\framework\autosave            nmake - all fr_qa_complex_framework_autosave fr_util NULL
+fr   framework\qa\complex\framework\recovery            nmake - all fr_qa_complex_framework_recovery fr_util NULL
+fr   framework\qa\complex\imageManager\interfaces       nmake - all fr_qa_complex_imageManager_interfaces fr_util NULL
+fr   framework\qa\complex\imageManager                  nmake - all fr_qa_complex_imageManager fr_qa_complex_imageManager_interfaces fr_util NULL
+fr   framework\qa\complex\loadAllDocuments\helper       nmake - all fr_qa_complex_loadAllDocuments_helper fr_util NULL
+fr   framework\qa\complex\loadAllDocuments              nmake - all fr_qa_complex_loadAllDocuments fr_qa_complex_loadAllDocuments_helper fr_util NULL
+fr   framework\qa\complex\path_settings                 nmake - all fr_qa_complex_path_settings fr_util NULL
+fr   framework\qa\complex\path_substitution             nmake - all fr_qa_complex_path_substitution fr_util NULL
+#does not build: fr   framework\qa\complex\sequence                      nmake - all fr_qa_complex_sequence fr_util NULL
+#does not build: fr   framework\qa\unoapi                                nmake - all fr_qa_unoapi fr_util NULL
+
+# older tests:
+#does not build: fr   framework\test\threadtest                          nmake - all fr_test_threadtest fr_util NULL
+#does not build: fr   framework\test                                     nmake - all fr_test fr_util NULL
+#does not build: fr   framework\test\typecfg                             nmake - all fr_test_typecfg fr_util NULL
