--- writerfilter/prj/tests.lst.qa	1970-01-01 01:00:00.000000000 +0100
+++ writerfilter/prj/tests.lst.qa	2007-09-07 14:32:16.000000000 +0200
@@ -0,0 +1,5 @@
+wf   writerfilter\qa\cppunittests\doctok                nmake - all wf_qa_cppunittests_doctok NULL
+wf   writerfilter\qa\cppunittests\odiapi                nmake - all wf_qa_cppunittests_odiapi NULL
+wf   writerfilter\qa\cppunittests\qname                 nmake - all wf_qa_cppunittests_qname NULL
+wf   writerfilter\qa\cppunittests\sl                    nmake - all wf_qa_cppunittests_sl NULL
+wf   writerfilter\qa\cppunittests\xxml                  nmake - all wf_qa_cppunittests_xxml NULL
