Index: profileitem_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/profileitem_ooo.scp,v
retrieving revision 1.20
diff -u -u -r1.20 profileitem_ooo.scp
--- scp2/source/ooo/profileitem_ooo.scp	13 May 2005 08:25:16 -0000	1.20
+++ scp2/source/ooo/profileitem_ooo.scp	25 May 2005 18:46:45 -0000
@@ -531,7 +531,7 @@
     Section = "Bootstrap";
     Order = 2;
     Key = "UNO_SHARED_PACKAGES_CACHE";
-    Value = "$UNO_SHARED_PACKAGES/cache";
+    Value = "file:/var/spool/openoffice2/uno_packages/cache";
 End
 
 ProfileItem gid_Profileitem_Uno_Uno_User_Packages
