Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.128
retrieving revision 1.128.4.4
diff -u -u -r1.128 -r1.128.4.4
--- config_office/configure.in	14 Jul 2005 13:45:35 -0000	1.128
+++ config_office/configure.in	26 Jul 2005 18:01:24 -0000	1.128.4.4
@@ -105,6 +105,9 @@
                           assertions, object counting, etc. Larger build.
                           Independent from --enable-debug
 ],,)
+AC_ARG_ENABLE(xsltproc,
+[  --enable-xsltproc:      Use xsltproc at buildtime, even when java is available
+],,)
 AC_ARG_ENABLE(crashdump,
 [[  --enable-crashdump[=yes|TRUE|STATIC|no]
                           Enable (disable) the crashdump feature code. [default=STATIC]
@@ -1985,7 +1988,7 @@
 fi
 AC_SUBST(JAVACACHE)
 
-if test "$NEEDXSLTPROC" = "no"; then
+if test "$NEEDXSLTPROC" = "no" && test -z "$enable_xsltproc"; then
    XSLTPROC=NO_XSLTPROC
 else
    dnl Check for xsltproc
Index: solenv/inc/tg_config.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/tg_config.mk,v
retrieving revision 1.2
retrieving revision 1.2.154.1
diff -u -u -r1.2 -r1.2.154.1
--- solenv/inc/tg_config.mk	15 Mar 2005 13:00:23 -0000	1.2
+++ solenv/inc/tg_config.mk	26 Jul 2005 15:47:16 -0000	1.2.154.1
@@ -84,7 +84,7 @@
 $(PROCESSOUT)$/registry$/schema$/$(PACKAGEDIR)$/%.xcs : %.xcs
     @+echo -------------+ validating and stripping schema files
 	-$(MKDIRHIER) $(@:d)
-.IF "$(SOLAR_JAVA)"!=""
+.IF "$(XSLTPROC)"=="NO_XSLTPROC"
 .IF "$(NO_INSPECTION)"==""
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(PROCESSORDIR)$/cfgimport.jar -Djavax.xml.parsers.SAXParserFactory=com.sun.xml.parser.SAXParserFactoryImpl org.openoffice.configuration.Inspector $<
 .ENDIF			# "$(NO_INSPECTION)"==""
@@ -124,7 +124,7 @@
 $(PROCESSOUT)$/registry$/res$/{$(alllangiso)}$/$(PACKAGEDIR)$/%.properties :| $(PROCESSOUT)$/merge$/$(PACKAGEDIR)$/%.xcs
     @+echo -------------+ creating locale dependent resource bundles
 	-$(MKDIRHIER) $(@:d)
-.IF "$(SOLAR_JAVA)"!=""
+.IF "$(XSLTPROC)"=="NO_XSLTPROC"
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $< $(XSLDIR)$/resource.xsl $@ locale={$(subst,$/$(PACKAGEDIR)$/$(@:f), $(subst,$(PROCESSOUT)$/registry$/res$/, $@))}
 .ELSE
 	$(XSLTPROC) -o $@ \
@@ -142,7 +142,7 @@
 $(PROCESSOUT)$/registry$/data$/$(PACKAGEDIR)$/%.xcu : %.xcu
     @+echo -------------+ validating and creating a locale independent file
 	-$(MKDIRHIER) $(@:d) 
-.IF "$(SOLAR_JAVA)"!=""
+.IF "$(XSLTPROC)"=="NO_XSLTPROC"
 .IF "$(NO_INSPECTION)"==""
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(PROCESSORDIR)$/cfgimport.jar -Djavax.xml.parsers.SAXParserFactory=com.sun.xml.parser.SAXParserFactoryImpl org.openoffice.configuration.Inspector $<
 .ENDIF			# "$(NO_INSPECTION)"==""
@@ -184,7 +184,7 @@
 $(PROCESSOUT)$/registry$/res$/{$(alllangiso)}$/$(PACKAGEDIR)$/%.xcu :| $(PROCESSOUT)$/merge$/$(PACKAGEDIR)$/%.xcu
     @+echo -------------+ creating locale dependent entries
 	-$(MKDIRHIER) $(@:d)
-.IF "$(SOLAR_JAVA)"!=""
+.IF "$(XSLTPROC)"=="NO_XSLTPROC"
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(PROCESSORDIR)$/cfgimport.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $< $(XSLDIR)$/alllang.xsl $(@:d)$*.tmp xcs=$(XCSROOT)$/registry$/schema$/$(PACKAGEDIR)$/$*.xcs schemaRoot=$(XCSROOT)$/registry$/schema locale={$(subst,$/$(PACKAGEDIR)$/$(@:f), $(subst,$(PROCESSOUT)$/registry$/res$/, $@))}	
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(PROCESSORDIR)$/schema.jar -Djavax.xml.parsers.SAXParserFactory=com.sun.xml.parser.SAXParserFactoryImpl org.openoffice.helper.PrettyPrinter $(@:d)$*.tmp $@
 .ELSE
@@ -220,7 +220,7 @@
 $(PROCESSOUT)$/registry$/spool$/$(PACKAGEDIR)$/%.xcu :| $$(@:b:s/-/./:b).xcu
     @+echo -------------+ creating a module file
 	-$(MKDIRHIER) $(@:d) 
-.IF "$(SOLAR_JAVA)"!=""
+.IF "$(XSLTPROC)"=="NO_XSLTPROC"
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/xt.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(PROCESSORDIR)$/cfgimport.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $< $(XSLDIR)$/alllang.xsl $(@:d)$(@:f:s/.xcu/.tmp/) xcs=$(XCSROOT)$/registry$/schema$/$(PACKAGEDIR)$/$(<:b).xcs schemaRoot=$(XCSROOT)$/registry$/schema module={$(subst,$(<:b)-, $(*))}
 	$(JAVAI) $(JAVACPS) $(SOLARBINDIR)$/jaxp.jar$(PATH_SEPERATOR)$(SOLARBINDIR)$/parser.jar$(PATH_SEPERATOR)$(PROCESSORDIR)$/schema.jar -Djavax.xml.parsers.SAXParserFactory=com.sun.xml.parser.SAXParserFactoryImpl org.openoffice.helper.PrettyPrinter $(@:d)$(@:f:s/.xcu/.tmp/) $@
 .ELSE
