--- filter/source/xsltfilter/makefile.mk.orig	2005-12-28 21:28:18.000000000 +0100
+++ filter/source/xsltfilter/makefile.mk	2005-12-28 22:13:41.000000000 +0100
@@ -73,7 +73,7 @@
 JARFILES 		= ridl.jar unoil.jar jurt.jar juh.jar
 
 .IF "$(SYSTEM_XALAN)" == "YES"
-XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(XALAN_JAR)
+XCLASSPATH!:=$(XCLASSPATH)$(PATH_SEPERATOR)$(XALAN_JAR)$(PATH_SEPERATOR)/usr/share/java/serializer.jar
 .ELSE
 JARFILES += xalan.jar
 .ENDIF
