--- hsqldb/makefile.mk.old	2005-07-12 00:34:50.433505048 +0200
+++ hsqldb/makefile.mk	2005-07-12 00:35:30.248452256 +0200
@@ -68,59 +68,5 @@
 .INCLUDE :	settings.mk
 .INCLUDE :  version.mk
 
-.IF "$(SOLAR_JAVA)" != ""
-# --- Files --------------------------------------------------------
-
-TARFILE_NAME=hsqldb_$(HSQLDB_VERSION)
-      
-TARFILE_ROOTDIR=hsqldb
-
-PATCH_FILE_NAME=hsqldb_1_8_0
-
-# ADDITIONAL_FILES=   src$/org$/hsqldb$/Collation.java \
-#                     src$/org$/hsqldb$/TxManager.java \
-#                     src$/org$/hsqldb$/lib$/LongKeyIntValueHashMap.java \
-#                     src$/org$/hsqldb$/persist$/ScaledRAFileInJar.java \
-#                     src$/org$/hsqldb$/test$/TestCollation.java
-
-.IF "$(ANT_HOME)" == ""
-ANT_HOME*:=$(COMMON_BUILD_TOOLS)$/apache-ant-1.6.1
-.EXPORT : ANT_HOME
-.ENDIF
-ANT_LIB*:=$(ANT_HOME)$/lib
-
-ANT_CLASSPATH:=$(ANT_LIB)$/xercesImpl.jar$(PATH_SEPERATOR)$(ANT_LIB)$/xml-apis.jar$(PATH_SEPERATOR)$(ANT_LIB)$/ant.jar
-PATH!:=$(ANT_HOME)$/bin$(PATH_SEPERATOR)$(PATH)
-
-.IF "$(ANT)" == ""
-ANT*:=$(ANT_HOME)$/bin$/ant
-.ENDIF
-
-ANT_BUILDFILE=build$/build.xml
-
-# Debian: local hack to only use libant1.6-java. see /usr/share/doc/libant1.6-java/README.Debian
-# # FIXME: conditionalize this and don't hardcode it. configure? (configure needs to be fixed anyhow
-# # because we are not allowed to use $ANT_HOME/bin/ant in any way)
-ANT_HOME:=/usr/share/ant1.6
-ANT_LIB:=$(ANT_HOME)$/lib
-ANT:=$(JAVAI) $(JAVACPS) $(CLASSPATH)$(PATH_SEPERATOR)$(ANT_LIB)/ant-launcher.jar$(PATH_SEPERATOR)$(ANT_LIB)/ant.jar -Dant.home=$(ANT_HOME) org.apache.tools.ant.Main
-
-.IF "$(JDK)"=="gcj"
-JAVA_HOME=
-.EXPORT : JAVA_HOME
-BUILD_ACTION=$(ANT) -Dbuild.compiler=gcj -f $(ANT_BUILDFILE) jar
-.ELSE
-BUILD_ACTION=$(ANT) -f $(ANT_BUILDFILE) jar				
-.ENDIF
-
-.ENDIF # $(SOLAR_JAVA)!= ""
-
-# --- Targets ------------------------------------------------------
-
-.INCLUDE : set_ext.mk
-.INCLUDE : target.mk
-
-.IF "$(SOLAR_JAVA)" != ""
-.INCLUDE : tg_ext.mk
-.ENDIF
-
+@all:
+	@echo "disabled"
Index: prj/d.lst
===================================================================
RCS file: /cvs/external/hsqldb/prj/d.lst,v
retrieving revision 1.1
diff -u -u -r1.1 d.lst
--- hsqldb/prj/d.lst	9 Nov 2004 13:59:27 -0000	1.1
+++ hsqldb/prj/d.lst	11 Jul 2005 22:43:22 -0000
@@ -1 +1 @@
-..\%__SRC%\misc\build\hsqldb\lib\hsqldb.jar %_DEST%\bin%_EXT%\hsqldb.jar
+/usr/share/java/hsqldb-1.8.0.0.jar %_DEST%\bin%_EXT%\hsqldb.jar
