--- vcl/source/gdi/virdev.cxx	25 Oct 2007 16:04:52 -0000	1.31
+++ vcl/source/gdi/virdev.cxx	31 Mar 2008 09:54:45 -0000
@@ -191,7 +191,8 @@ VirtualDevice::VirtualDevice( const Outp
 // -----------------------------------------------------------------------
 
 VirtualDevice::VirtualDevice( const OutputDevice& rCompDev, USHORT nBitCount, USHORT nAlphaBitCount )
-	: mpVirDev( NULL )
+	: mpVirDev( NULL ),
+    meRefDevMode( REFDEV_NONE )
 {
 	DBG_TRACE1( "VirtualDevice::VirtualDevice( %hu )", nBitCount );
 
