Index: vcl/source/window/window.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/window.cxx,v
retrieving revision 1.199
diff -u -r1.199 window.cxx
--- vcl/source/window/window.cxx	9 Sep 2004 16:23:08 -0000	1.199
+++ vcl/source/window/window.cxx	15 Oct 2004 15:55:32 -0000
@@ -8769,8 +8769,13 @@
     Color aSelectionBorderCol( GetSettings().GetStyleSettings().GetHighlightColor() );
     Color aSelectionFillCol( aSelectionBorderCol );
 
+#ifdef HORRIBLE_WITH_INDUSTRIAL_THEME
 	BOOL bDark = GetSettings().GetStyleSettings().GetFaceColor().IsDark();
 	BOOL bBright = GetSettings().GetStyleSettings().GetFaceColor().IsBright();
+#else
+	BOOL bDark = FALSE;
+	BOOL bBright = FALSE;
+#endif
 
     int c1 = aSelectionBorderCol.GetLuminance();
     int c2 = GetDisplayBackground().GetColor().GetLuminance();
