--- vcl/source/window/window.cxx	2005-12-14 17:17:52.000000000 +0000
+++ vcl/source/window/window.cxx	2005-12-14 17:20:53.000000000 +0000
@@ -518,6 +518,16 @@ void Window::ImplUpdateGlobalSettings( A
             }
         }
     }
+    else
+    {
+        if( !rSettings.GetStyleSettings().GetFaceColor().IsDark()
+            && !rSettings.GetStyleSettings().GetWindowColor().IsDark() )
+        {
+            StyleSettings aStyleSettings = rSettings.GetStyleSettings();
+            aStyleSettings.SetHighContrastMode( FALSE );
+            rSettings.SetStyleSettings( aStyleSettings );
+        }
+    }
 
 	// Detect if images in menus are allowed or not
     {
