--- vcl/unx/source/app/keysymnames.cxx.orig	2007-06-12 16:19:10.000000000 +0400
+++ vcl/unx/source/app/keysymnames.cxx	2007-06-12 16:33:28.000000000 +0400
@@ -103,6 +103,26 @@
 		{ XK_space, "Space" } 
 	};
 
+	static const struct KeysymNameReplacement aImplReplacements_Turkish[] =
+	{
+		{ XK_Control_L, "Ctrl" },
+		{ XK_Control_R, "Ctrl" },
+		{ XK_Right, "Sağ" },
+		{ XK_Left, "Sol" },
+		{ XK_Up, "Yukarı" },
+		{ XK_Down, "Aşağı" },
+		{ XK_space, "Boşluk" } 
+	};
+
+	static const struct KeysymNameReplacement aImplReplacements_Russian[] =
+	{
+		{ XK_Right, "Вправо" },
+		{ XK_Left, "Влево" },
+		{ XK_Up, "Вверх" },
+		{ XK_Down, "Вниз" },
+		{ XK_space, "Пробел" } 
+	};
+
 	static const struct KeysymNameReplacement aImplReplacements_German[] =
 	{
 		{ XK_Control_L, "Strg" },
@@ -383,6 +403,10 @@
 		{ "Swed", aImplReplacements_Swedish, sizeof(aImplReplacements_Swedish)/sizeof(aImplReplacements_Swedish[0]) },
 		{ "Netherland", aImplReplacements_Dutch, sizeof(aImplReplacements_Dutch)/sizeof(aImplReplacements_Dutch[0]) },
 		{ "Dutch", aImplReplacements_Dutch, sizeof(aImplReplacements_Dutch)/sizeof(aImplReplacements_Dutch[0]) },
+		// Turkish, Turkey
+		{ "Turk", aImplReplacements_Turkish, sizeof(aImplReplacements_Turkish)/sizeof(aImplReplacements_Turkish[0]) },
+		// Russian, Russia
+		{ "Russia", aImplReplacements_Russian, sizeof(aImplReplacements_Russian)/sizeof(aImplReplacements_Russian[0]) },
 		{ "English", aImplReplacements_English, sizeof(aImplReplacements_English)/sizeof(aImplReplacements_English[0]) }
 	};
 
