--- vcl/source/window/window.cxx	2005-12-15 08:51:07.000000000 +0000
+++ vcl/source/window/window.cxx	2005-12-15 09:20:51.000000000 +0000
@@ -491,13 +491,12 @@ void Window::ImplUpdateGlobalSettings( A
     aFont.SetHeight( defFontheight );
     aStyleSettings.SetGroupFont( aFont );
 
-    rSettings.SetStyleSettings( aStyleSettings );
-
-
     // #104427# auto detect HC mode ?
-    if( !rSettings.GetStyleSettings().GetHighContrastMode() )
+    sal_Bool bHighContrast = sal_False;
+    if( rSettings.GetStyleSettings().GetFaceColor().IsDark()
+        || rSettings.GetStyleSettings().GetWindowColor().IsDark() )
     {
-        sal_Bool bTmp, bAutoHCMode = sal_True;
+        sal_Bool bTmp = sal_True;
         utl::OConfigurationNode aNode = utl::OConfigurationTreeRoot::tryCreateWithServiceFactory(
             vcl::unohelper::GetMultiServiceFactory(),
             OUString::createFromAscii( "org.openoffice.Office.Common/Accessibility" ) );    // note: case sensisitive !
@@ -505,19 +504,12 @@ void Window::ImplUpdateGlobalSettings( A
         {
             ::com::sun::star::uno::Any aValue = aNode.getNodeValue( OUString::createFromAscii( "AutoDetectSystemHC" ) );
             if( aValue >>= bTmp )
-                bAutoHCMode = bTmp;
-        }
-        if( bAutoHCMode )
-        {
-            if( rSettings.GetStyleSettings().GetFaceColor().IsDark()
-             || rSettings.GetStyleSettings().GetWindowColor().IsDark() )
-            {
-                StyleSettings aStyleSettings = rSettings.GetStyleSettings();
-                aStyleSettings.SetHighContrastMode( TRUE );
-                rSettings.SetStyleSettings( aStyleSettings );
-            }
+                bHighContrast = bTmp;
         }
     }
+    aStyleSettings.SetHighContrastMode( bHighContrast );
+
+    rSettings.SetStyleSettings( aStyleSettings );
 
 	// Detect if images in menus are allowed or not
     {
