Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.42
diff -u -p -u -r1.42 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	25 Jan 2006 11:40:22 -0000	1.42
+++ vcl/unx/gtk/window/gtkframe.cxx	10 Jul 2006 12:42:59 -0000
@@ -2113,7 +2349,11 @@ gboolean GtkSalFrame::signalKey( GtkWidg
 
     if( pThis->m_pIMHandler )
         if( pThis->m_pIMHandler->handleKeyEvent( pEvent ) )
+		{
+			pThis->m_bSendModChangeOnRelease = false;
+			pThis->m_bSingleAltPress = false;
             return TRUE;
+		}
     GTK_YIELD_GRAB();
 
 	// handle modifiers
@@ -2225,6 +2465,9 @@ gboolean GtkSalFrame::signalKey( GtkWidg
     }
     else
     {
+		pThis->m_bSendModChangeOnRelease = false;
+		pThis->m_bSingleAltPress = false;
+
         pThis->doKeyCallback( pEvent->state,
                               pEvent->keyval,
                               pEvent->hardware_keycode,
@@ -2233,11 +2476,6 @@ gboolean GtkSalFrame::signalKey( GtkWidg
                               sal_Unicode(gdk_keyval_to_unicode( pEvent->keyval )),
                               (pEvent->type == GDK_KEY_PRESS),
                               false );
-		if( ! aDel.isDeleted() )
-		{
-			pThis->m_bSendModChangeOnRelease = false;
-			pThis->m_bSingleAltPress = false;
-		}
     }
 
 	if( !aDel.isDeleted() && pThis->m_pIMHandler )
