Index: vcl/unx/gtk/window/gtkframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/gtk/window/gtkframe.cxx,v
retrieving revision 1.19
diff -u -p -u -r1.19 gtkframe.cxx
--- vcl/unx/gtk/window/gtkframe.cxx	21 Jan 2005 13:36:54 -0000	1.19
+++ vcl/unx/gtk/window/gtkframe.cxx	4 Feb 2005 16:55:13 -0000
@@ -127,7 +130,9 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_Left:			nCode = KEY_LEFT;		break;
             case GDK_KP_Right:
             case GDK_Right:			nCode =	KEY_RIGHT;		break;
+            case GDK_KP_Begin:
             case GDK_KP_Home:
+            case GDK_Begin:
             case GDK_Home:			nCode = KEY_HOME;		break;
             case GDK_KP_End:
             case GDK_End:			nCode = KEY_END;		break;
@@ -144,7 +149,9 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_KP_Add:		nCode = KEY_ADD;		break;
             case GDK_minus:
             case GDK_KP_Subtract:	nCode = KEY_SUBTRACT;	break;
+			case GDK_asterisk:
             case GDK_KP_Multiply:	nCode = KEY_MULTIPLY;	break;
+			case GDK_slash:
             case GDK_KP_Divide:		nCode = KEY_DIVIDE;		break;
             case GDK_period:
             case GDK_decimalpoint:	nCode = KEY_POINT;		break;
@@ -145,12 +180,21 @@ static USHORT GetKeyCode( guint keyval )
             case GDK_comma:			nCode = KEY_COMMA;		break;
             case GDK_less:			nCode = KEY_LESS;		break;
             case GDK_greater:		nCode = KEY_GREATER;	break;
+            case GDK_KP_Equal:
             case GDK_equal:			nCode = KEY_EQUAL;		break;
             case GDK_Find:			nCode = KEY_FIND;		break;
             case GDK_Menu:			nCode = KEY_MENU;		break;
             case GDK_Help:			nCode = KEY_HELP;		break;
             case GDK_Undo:			nCode = KEY_UNDO;		break;
+			case GDK_Redo:			nCode = KEY_REPEAT;		break;
+			case GDK_KP_Decimal:
             case GDK_KP_Separator:	nCode = KEY_DECIMAL;	break;
+// ignored Sun X-Server bits (wrt. numlock etc.) cf saldisp.cxx
+// ignored XK_ISO_Left_Tab, XK_Hangul_Hanja
+// ignored Apollo apXK section
+// ignored DEC section
+// ignored HP & IBM/OSF sections
+// ignored SCO,SGI,SNI,SUN section
         }
     }
     return nCode;


