--- vcl/unx/kde/salnativewidgets-kde.cxx	2005-03-09 13:50:54.551493024 +0100
+++ vcl/unx/kde/salnativewidgets-kde.cxx	2005-03-09 20:38:59.420653245 +0100
@@ -129,6 +129,8 @@
 #endif
 #include <iostream>
 
+#include <pspgraphics.h>
+
 using namespace ::rtl;
 
 /** Cached native widgets.
@@ -1720,49 +1722,95 @@ static Color readColor( KConfig *pConfig
 }
 
 /** Helper function to add information to Font from QFont.
+
+    Mostly grabbed from the Gtk+ vclplug (salnativewidgets-gtk.cxx).
 */
-static void modifyFont( Font &rFont, const QFont &rQFont )
+static Font toFont( const QFont &rQFont, const ::com::sun::star::lang::Locale& rLocale )
 {
+    psp::FastPrintFontInfo aInfo;
     QFontInfo qFontInfo( rQFont );
     
-    // Prepend the KDE font, do not override
-    OUString aQFontName = String( rQFont.family().utf8(), RTL_TEXTENCODING_UTF8 );
-    OUString aFontName = rFont.GetName();
-
-    if ( aQFontName.getLength() > 0 &&
-         aFontName.compareTo( aQFontName, aQFontName.getLength() ) != 0 )
-    {
-        OUStringBuffer aBuffer( 1024 );
-        aBuffer.append( aQFontName );
-        aBuffer.appendAscii( ";", 1 );
-        aBuffer.append( aFontName );
-        
-        rFont.SetName( aBuffer.makeStringAndClear() );
-    }
-    
-    // QFontInfo should give the right point size, but sometimes it does not,
-    // it seems.
-    int nPointSize = qFontInfo.pointSize();
-    if ( nPointSize <= 0 )
-        nPointSize = rQFont.pointSize();
-    if ( nPointSize > 0 )
-        rFont.SetHeight( nPointSize );
-    
-    rFont.SetItalic( qFontInfo.italic()? ITALIC_NORMAL: ITALIC_NONE );
+    // set family name
+    aInfo.m_aFamilyName = String( rQFont.family().utf8(), RTL_TEXTENCODING_UTF8 );
+
+    // set italic
+    aInfo.m_eItalic = ( qFontInfo.italic()? psp::italic::Italic: psp::italic::Upright );
     
-    FontWeight eWeight = WEIGHT_DONTKNOW;
+    // set weight
     int nWeight = qFontInfo.weight();
     if ( nWeight <= QFont::Light )
-        eWeight = WEIGHT_LIGHT;
+        aInfo.m_eWeight = psp::weight::Light;
     else if ( nWeight <= QFont::Normal )
-        eWeight = WEIGHT_NORMAL;
+        aInfo.m_eWeight = psp::weight::Normal;
     else if ( nWeight <= QFont::DemiBold )
-        eWeight = WEIGHT_SEMIBOLD;
+        aInfo.m_eWeight = psp::weight::SemiBold;
     else if ( nWeight <= QFont::Bold )
-        eWeight = WEIGHT_BOLD;
+        aInfo.m_eWeight = psp::weight::Bold;
+    else
+        aInfo.m_eWeight = psp::weight::UltraBold;
+    
+    // set width
+    int nStretch = rQFont.stretch();
+    if ( nStretch <= QFont::UltraCondensed )
+        aInfo.m_eWidth = psp::width::UltraCondensed;
+    else if ( nStretch <= QFont::ExtraCondensed )
+        aInfo.m_eWidth = psp::width::ExtraCondensed;
+    else if ( nStretch <= QFont::Condensed )
+        aInfo.m_eWidth = psp::width::Condensed;
+    else if ( nStretch <= QFont::SemiCondensed )
+        aInfo.m_eWidth = psp::width::SemiCondensed;
+    else if ( nStretch <= QFont::Unstretched )
+        aInfo.m_eWidth = psp::width::Normal;
+    else if ( nStretch <= QFont::SemiExpanded )
+        aInfo.m_eWidth = psp::width::SemiExpanded;
+    else if ( nStretch <= QFont::Expanded )
+        aInfo.m_eWidth = psp::width::Expanded;
+    else if ( nStretch <= QFont::ExtraExpanded )
+        aInfo.m_eWidth = psp::width::ExtraExpanded;
     else
-        eWeight = WEIGHT_BLACK;
-    rFont.SetWeight( eWeight );
+        aInfo.m_eWidth = psp::width::UltraExpanded;
+    
+#if OSL_DEBUG_LEVEL > 1
+    fprintf( stderr, "font name BEFORE system match: \"%s\"\n", OUStringToOString( aInfo.m_aFamilyName, RTL_TEXTENCODING_ISO_8859_1 ).getStr() );
+#endif
+
+    // match font to e.g. resolve "Sans"
+    psp::PrintFontManager::get().matchFont( aInfo, rLocale );
+
+#if OSL_DEBUG_LEVEL > 1
+    fprintf( stderr, "font match %s, name AFTER: \"%s\"\n",
+             aInfo.m_nID != 0 ? "succeeded" : "failed",
+             OUStringToOString( aInfo.m_aFamilyName, RTL_TEXTENCODING_ISO_8859_1 ).getStr() );
+#endif
+
+    // font height
+    int nPointHeight = qFontInfo.pointSize();
+    if ( nPointHeight <= 0 )
+        nPointHeight = rQFont.pointSize();
+    
+    sal_Int32 nDPIX, nDPIY;
+    sal_Int32 nDispDPIY = GetSalData()->GetDisplay()->GetResolution().B();
+    GetSalData()->GetDisplay()->GetScreenFontResolution( nDPIX, nDPIY );
+
+    int nHeight = nPointHeight * nDispDPIY / nDPIY;
+    // allow for rounding in back conversion (at SetFont)
+    while( (nHeight * nDPIY / nDispDPIY) > nPointHeight )
+        nHeight--;
+    while( (nHeight * nDPIY / nDispDPIY) < nPointHeight )
+        nHeight++;
+    
+    // Create the font
+    Font aFont( aInfo.m_aFamilyName, Size( 0, nHeight ) );
+    if( aInfo.m_eWeight != psp::weight::Unknown )
+        aFont.SetWeight( PspGraphics::ToFontWeight( aInfo.m_eWeight ) );
+    if( aInfo.m_eWidth != psp::width::Unknown )
+        aFont.SetWidthType( PspGraphics::ToFontWidth( aInfo.m_eWidth ) );
+    if( aInfo.m_eItalic != psp::italic::Unknown )
+        aFont.SetItalic( PspGraphics::ToFontItalic( aInfo.m_eItalic ) );
+    if( aInfo.m_ePitch != psp::pitch::Unknown )
+        aFont.SetPitch( PspGraphics::ToFontPitch( aInfo.m_ePitch ) );
+
+    return aFont;
 }
 
 /** Implementation of KDE integration's main method.
@@ -1806,8 +1854,7 @@ void KDESalFrame::UpdateSettings( AllSet
         pKey = "titleFont";
         if ( pConfig->hasKey( pKey ) )
         {
-            Font aFont= aStyleSettings.GetTitleFont();
-            modifyFont( aFont, pConfig->readFontEntry( pKey ) );
+            Font aFont = toFont( pConfig->readFontEntry( pKey ), rSettings.GetUILocale() );
             aStyleSettings.SetTitleFont( aFont );
 			bSetTitleFont = true;
         }
@@ -1856,8 +1903,7 @@ void KDESalFrame::UpdateSettings( AllSet
     aStyleSettings.SetHighlightTextColor( toColor( qColorGroup.highlightedText() ) );
 
     // Font
-    Font aFont= aStyleSettings.GetAppFont();
-    modifyFont( aFont, kapp->font() );
+    Font aFont = toFont( kapp->font(), rSettings.GetUILocale() );
 
     aStyleSettings.SetAppFont( aFont );
     aStyleSettings.SetHelpFont( aFont );
@@ -1874,8 +1920,11 @@ void KDESalFrame::UpdateSettings( AllSet
     aStyleSettings.SetIconFont( aFont );
     aStyleSettings.SetGroupFont( aFont );
 
+    KMainWindow qMainWindow;
+    qMainWindow.createGUI( "/dev/null" ); // hack
+
     // Menu
-    QMenuBar *pMenuBar = pWidgetPainter->menuBar( Region() );
+    KMenuBar *pMenuBar = qMainWindow.menuBar();
     if ( pMenuBar )
     {
         // Color
@@ -1887,17 +1936,15 @@ void KDESalFrame::UpdateSettings( AllSet
         aStyleSettings.SetMenuHighlightTextColor( toColor ( qMenuCG.highlightedText() ) );
 
         // Font
-        Font aFont= aStyleSettings.GetMenuFont();
-        modifyFont( aFont, pMenuBar->font() );
+        Font aFont = toFont( pMenuBar->font(), rSettings.GetUILocale() );
         aStyleSettings.SetMenuFont( aFont );
     }
 
     // Tool bar
-    QToolBar *pToolBar = pWidgetPainter->toolBar( Region(), true );
+    KToolBar *pToolBar = qMainWindow.toolBar();
     if ( pToolBar )
     {
-        Font aFont= aStyleSettings.GetToolFont();
-        modifyFont( aFont, pToolBar->font() );
+        Font aFont = toFont( pToolBar->font(), rSettings.GetUILocale() );
         aStyleSettings.SetToolFont( aFont );
     }
 
