--- vcl/unx/kde/salnativewidgets-kde.cxx	2005-03-15 12:51:47.033308709 +0100
+++ vcl/unx/kde/salnativewidgets-kde.cxx	2005-03-15 13:31:44.924184772 +0100
@@ -761,16 +761,20 @@ BOOL WidgetPainter::drawStyledWidget( QW
             QRect qThumbRect = region2QRect( pValue->maGripRect );
             if ( bIsHorizontal )
             {
-                qThumbRect.addCoords( 0, 1, 0, -2 );    // make the thumb a bit nicer
+                qThumbRect.addCoords( 0, 2, 0, -3 );    // make the thumb a bit nicer
                 qRect.setWidth( qThumbRect.width() );   // draw just the thumb part
             }
             else
             {
-                qThumbRect.addCoords( 1, 0, -2, 0 );    // make the thumb a bit nicer
+                qThumbRect.addCoords( 2, 0, -3, 0 );    // make the thumb a bit nicer
                 qRect.setHeight( qThumbRect.height() ); // draw just the thumb part
             }
 
-            if ( kapp->style().inherits( "KStyle" ) )
+            if ( kapp->style().inherits( "HighColorStyle" ) ||
+                 kapp->style().inherits( "HighContrastStyle" ) ||
+                 kapp->style().inherits( "KeramikStyle" ) ||
+                 kapp->style().inherits( "KThemeStyle" ) ||
+                 kapp->style().inherits( "ThinKeramikStyle" ) )
             {
                 // Workaround for the workaround in KStyle::drawPrimitive()
                 KStyle *pStyle = static_cast< KStyle * >( &kapp->style() );
