Index: makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/makefile.mk,v
retrieving revision 1.19
diff -u -u -r1.19 makefile.mk
--- vcl/unx/source/app/makefile.mk	13 Jan 2005 18:14:25 -0000	1.19
+++ vcl/unx/source/app/makefile.mk	16 Mar 2005 02:47:14 -0000
@@ -111,6 +111,12 @@
 .IF "$(ENABLE_PASF)" != ""
 CFLAGS+=-DUSE_PASF
 SLOFILES+=$(SLO)$/pasfsound.obj
+.IF "$(SYSTEM_SNDFILE)" == "YES"
+CFLAGS+=-DSYSTEM_SNDFILE
+.ENDIF
+.IF "$(SYSTEM_PORTAUDIO)" == "YES"
+CFLAGS+=-DSYSTEM_PORTAUDIO
+.ENDIF
 .ENDIF
 
 .IF "$(WITH_LIBSN)"=="YES"
Index: source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.89
diff -u -u -r1.89 file_library_ooo.scp
--- scp2/source/ooo/file_library_ooo.scp	10 Mar 2005 18:23:28 -0000	1.89
+++ scp2/source/ooo/file_library_ooo.scp	16 Mar 2005 03:37:09 -0000
@@ -1925,6 +1925,8 @@
 
 #ifdef USE_PASF
 
+#ifndef SYSTEM_PORTAUDIO
+
 File gid_File_Lib_Portaudio
     TXT_FILE_BODY;
     Name    = STRING(CONCAT3(libportaudio,UNXSUFFIX,.0.0.18));
@@ -1946,6 +1948,10 @@
     Styles      = (NETWORK, RELATIVE);
 End
 
+#endif
+
+#ifndef SYSTEM_SNDFILE
+
 File gid_File_Lib_Sndfile
     TXT_FILE_BODY;
     Name    = STRING(CONCAT3(libsndfile,UNXSUFFIX,.1.0.9));
@@ -1966,6 +1972,8 @@
     Name        = STRING(CONCAT2(libsndfile,UNXSUFFIX));
     Styles      = (NETWORK, RELATIVE);
 End
+
+#endif
 
 #endif
 
Index: makefile.mk
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/makefile.mk,v
retrieving revision 1.25
diff -u -u -r1.25 makefile.mk
--- scp2/source/ooo/makefile.mk	1 Mar 2005 16:17:03 -0000	1.25
+++ scp2/source/ooo/makefile.mk	16 Mar 2005 03:39:32 -0000
@@ -140,6 +140,14 @@
 SCPDEFS+=-DSYSTEM_STDLIBS
 .ENDIF
 
+.IF "$(SYSTEM_SNDFILE)" == "YES"
+SCPDEFS+=-DSYSTEM_SNDFILE
+.ENDIF
+
+.IF "$(SYSTEM_PORTAUDIO)" == "YES"
+SCPDEFS+=-DSYSTEM_PORTAUDIO
+.ENDIF
+
 .IF "$(JDK)" == "gcj"
 SCPDEFS+=-DGCJ
 .ENDIF
