--- instsetoo_native/util/openoffice.lst.old	2005-01-31 11:38:28.000000000 +0100
+++ instsetoo_native/util/openoffice.lst	2005-03-04 15:44:45.000000000 +0100
@@ -5,7 +5,7 @@
 		variables
 		{
 			PRODUCTNAME OpenOffice.org
-			PRODUCTVERSION 1.9.{milestone}
+			PRODUCTVERSION 1.9.129
 			PRODUCTEXTENSION
 			LONG_PRODUCTEXTENSION
 			REGISTRATIONURL http://www.openoffice.org/welcome/registration20.html
@@ -13,7 +13,7 @@
 			FILEFORMATNAME OpenOffice.org
 			FILEFORMATVERSION 1.0
 			WRITERCOMPATIBILITYVERSIONOOO11 OpenOffice.org 1.1
-			PACKAGEVERSION 1.9.{milestone}
+			PACKAGEVERSION 1.9.129
 			PACKAGEREVISION 1
 			LICENSENAME LGPL, SISSL
 	    }
