--- scp2/source/ooo/installation_ooo.scp.bak	2004-11-16 00:20:06.358511584 +0530
+++ scp2/source/ooo/installation_ooo.scp	2004-11-16 00:21:14.575141080 +0530
@@ -64,7 +64,7 @@
 Installation gid_Installation
     ProductName = "${PRODUCTNAME}";
     ProductVersion = "${PRODUCTVERSION}";
-    ProductExtension = "${PRODUCTEXTENSION}";
+    ProductExtension = PRODUCTEXTENSION;
     VendorName = "OpenOffice.org";
     Bitmap = "oo_product.bmp";
   #ifdef WNT
--- scp2/source/ooo/makefile.mk.bak	2004-11-16 00:20:23.230946584 +0530
+++ scp2/source/ooo/makefile.mk	2004-11-16 00:23:35.808670320 +0530
@@ -78,6 +78,13 @@
 SCPDEFS+=-DENABLE_CRASHDUMP
 .ENDIF
 
+.IF "$(OOO_BUILD_EXTENSION)"!=""
+SCPDEFS+=-DPRODUCTEXTENSION=\""$(OOO_BUILD_EXTENSION)"\"
+.ELSE
+SCPDEFS+=-DPRODUCTEXTENSION=\"""\"
+.ENDIF
+
+
 .IF "$(JAVANUMVER)" >= "000100040000"
 SCPDEFS+=-DINCLUDE_JAVA_ACCESSBRIDGE
 .ENDIF
