Index: desktop/source/app/appinit.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/appinit.cxx,v
retrieving revision 1.21
diff -u -p -u -r1.21 appinit.cxx
--- desktop/source/app/appinit.cxx	10 Dec 2004 18:39:15 -0000	1.21
+++ desktop/source/app/appinit.cxx	24 Feb 2005 15:29:10 -0000
@@ -228,7 +228,8 @@ static bool configureUcb(bool bServer, r
                 );
                 rtl::OUString aDesktopEnvironment;
                 if ((aValue >>= aDesktopEnvironment)
-                    && aDesktopEnvironment.equalsAscii("GNOME"))
+                    && (aDesktopEnvironment.equalsAscii("GNOME") ||
+						aDesktopEnvironment.equalsAscii("KDE")))
                 {
                     Reference<XContentProviderManager> xCPM =
                         cb->getContentProviderManagerInterface();
