diff --git a/automation/inc/automation/communi.hxx b/automation/inc/automation/communi.hxx
index 42ed266..1ff4a31 100644
--- automation/inc/automation/communi.hxx
+++ automation/inc/automation/communi.hxx
@@ -98,7 +98,7 @@ public:
 class CommunicationLinkViaSocket : public SimpleCommunicationLinkViaSocket, public osl::Thread
 {
 public:
-	CommunicationLinkViaSocket( CommunicationManager *pMan, NAMESPACE_VOS(OStreamSocket) *pSocket );
+	CommunicationLinkViaSocket( CommunicationManager *pMan, vos::OStreamSocket *pSocket );
 	virtual ~CommunicationLinkViaSocket();
 
 	virtual BOOL IsCommunicationError();
@@ -164,7 +164,7 @@ protected:
 
 private:
 	CommunicationManagerServerViaSocket* pMyServer;
-	NAMESPACE_VOS(OAcceptorSocket) *pAcceptorSocket;
+	vos::OAcceptorSocket *pAcceptorSocket;
 	ULONG nPortToListen;
 	USHORT nMaxConnections;
 	ULONG nAddConnectionEventId;
@@ -193,7 +193,7 @@ private:
 	ByteString aHostToTalk;
 	ULONG nPortToTalk;
 protected:
-	virtual CommunicationLink *CreateCommunicationLink( CommunicationManager *pCM, NAMESPACE_VOS(OConnectorSocket) *pCS ){ return new CommunicationLinkViaSocket( pCM, pCS ); }
+	virtual CommunicationLink *CreateCommunicationLink( CommunicationManager *pCM, vos::OConnectorSocket *pCS ){ return new CommunicationLinkViaSocket( pCM, pCS ); }
 };
 
 #endif
diff --git a/automation/inc/automation/simplecm.hxx b/automation/inc/automation/simplecm.hxx
index 80eb849..c8486f1 100644
--- automation/inc/automation/simplecm.hxx
+++ automation/inc/automation/simplecm.hxx
@@ -357,14 +357,14 @@ private:
 	ByteString aMyName;
 
 	TCPIO* pTCPIO;
-	NAMESPACE_VOS(OStreamSocket) *pStreamSocket;
+	vos::OStreamSocket *pStreamSocket;
 
 protected:
-	SimpleCommunicationLinkViaSocket( CommunicationManager *pMan, NAMESPACE_VOS(OStreamSocket) *pSocket );
+	SimpleCommunicationLinkViaSocket( CommunicationManager *pMan, vos::OStreamSocket *pSocket );
 	virtual ~SimpleCommunicationLinkViaSocket();
 
-    NAMESPACE_VOS(OStreamSocket)* GetStreamSocket() { return pStreamSocket; }
-    void SetStreamSocket( NAMESPACE_VOS(OStreamSocket)* pSocket );
+    vos::OStreamSocket* GetStreamSocket() { return pStreamSocket; }
+    void SetStreamSocket( vos::OStreamSocket* pSocket );
 
 	SvStream *pReceiveStream;
 	BOOL DoReceiveDataStream();				/// Recieve DataPacket from Socket
@@ -378,7 +378,7 @@ protected:
 class SimpleCommunicationLinkViaSocketWithReceiveCallbacks : public SimpleCommunicationLinkViaSocket
 {
 public:
-	SimpleCommunicationLinkViaSocketWithReceiveCallbacks( CommunicationManager *pMan, NAMESPACE_VOS(OStreamSocket) *pSocket );
+	SimpleCommunicationLinkViaSocketWithReceiveCallbacks( CommunicationManager *pMan, vos::OStreamSocket *pSocket );
 	~SimpleCommunicationLinkViaSocketWithReceiveCallbacks();
 	virtual BOOL ReceiveDataStream();
 protected:
@@ -391,7 +391,7 @@ class CommonSocketFunctions
 public:
 	BOOL DoStartCommunication( CommunicationManager *pCM, ICommunicationManagerClient *pCMC, ByteString aHost, ULONG nPort );
 protected:
-	virtual CommunicationLink *CreateCommunicationLink( CommunicationManager *pCM, NAMESPACE_VOS(OConnectorSocket) *pCS )=0;
+	virtual CommunicationLink *CreateCommunicationLink( CommunicationManager *pCM, vos::OConnectorSocket *pCS )=0;
 };
 
 class SingleCommunicationManagerClientViaSocket : public SingleCommunicationManager, public ICommunicationManagerClient, CommonSocketFunctions
@@ -407,7 +407,7 @@ private:
 	ByteString aHostToTalk;
 	ULONG nPortToTalk;
 protected:
-	virtual CommunicationLink *CreateCommunicationLink( CommunicationManager *pCM, NAMESPACE_VOS(OConnectorSocket) *pCS ){ return new SimpleCommunicationLinkViaSocketWithReceiveCallbacks( pCM, pCS ); }
+	virtual CommunicationLink *CreateCommunicationLink( CommunicationManager *pCM, vos::OConnectorSocket *pCS ){ return new SimpleCommunicationLinkViaSocketWithReceiveCallbacks( pCM, pCS ); }
 };
 
 #endif
diff --git a/automation/source/communi/communi.cxx b/automation/source/communi/communi.cxx
index 292b15a..63f1ebb 100644
--- automation/source/communi/communi.cxx
+++ automation/source/communi/communi.cxx
@@ -89,7 +89,7 @@ SV_IMPL_PTRARR_SORT( CommunicationLinkList, CommunicationLink* );
 
 osl::Mutex *pMPostUserEvent=NULL;		// Notwendig, da nicht threadfest
 
-CommunicationLinkViaSocket::CommunicationLinkViaSocket( CommunicationManager *pMan, NAMESPACE_VOS(OStreamSocket) *pSocket )
+CommunicationLinkViaSocket::CommunicationLinkViaSocket( CommunicationManager *pMan, vos::OStreamSocket *pSocket )
 : SimpleCommunicationLinkViaSocket( pMan, pSocket )
 , nConnectionClosedEventId( 0 )
 , nDataReceivedEventId( 0 )
@@ -153,7 +153,7 @@ BOOL CommunicationLinkViaSocket::ShutdownCommunication()
 
 		join();
 
-        NAMESPACE_VOS(OStreamSocket) *pTempSocket = GetStreamSocket();
+        vos::OStreamSocket *pTempSocket = GetStreamSocket();
         SetStreamSocket( NULL );
         delete pTempSocket;
 
@@ -521,8 +521,8 @@ void CommunicationManagerServerAcceptThread::run()
 	if ( !nPortToListen )
 		return;
 
-	pAcceptorSocket = new NAMESPACE_VOS(OAcceptorSocket)();
-	NAMESPACE_VOS(OInetSocketAddr) Addr;
+	pAcceptorSocket = new vos::OAcceptorSocket();
+	vos::OInetSocketAddr Addr;
 	Addr.setPort( nPortToListen );
 	pAcceptorSocket->setReuseAddr( 1 );
 	if ( !pAcceptorSocket->bind( Addr ) )
@@ -535,14 +535,14 @@ void CommunicationManagerServerAcceptThread::run()
 	}
 
 
-	NAMESPACE_VOS(OStreamSocket) *pStreamSocket = NULL;
+	vos::OStreamSocket *pStreamSocket = NULL;
 
 	while ( schedule() )
 	{
-		pStreamSocket = new NAMESPACE_VOS(OStreamSocket);
+		pStreamSocket = new vos::OStreamSocket;
 		switch ( pAcceptorSocket->acceptConnection( *pStreamSocket ) )
 		{
-		case NAMESPACE_VOS(ISocketTypes::TResult_Ok):
+		case vos::ISocketTypes::TResult_Ok:
 			{
 				pStreamSocket->setTcpNoDelay( 1 );
 
@@ -558,17 +558,17 @@ void CommunicationManagerServerAcceptThread::run()
 				}
 			}
 			break;
-		case NAMESPACE_VOS(ISocketTypes::TResult_TimedOut):
+		case vos::ISocketTypes::TResult_TimedOut:
 			delete pStreamSocket;
 			pStreamSocket = NULL;
 			break;
-		case NAMESPACE_VOS(ISocketTypes::TResult_Error):
+		case vos::ISocketTypes::TResult_Error:
 			delete pStreamSocket;
 			pStreamSocket = NULL;
 			break;
 
-		case NAMESPACE_VOS(ISocketTypes::TResult_Interrupted):
-		case NAMESPACE_VOS(ISocketTypes::TResult_InProgress):
+		case vos::ISocketTypes::TResult_Interrupted:
+		case vos::ISocketTypes::TResult_InProgress:
 			break;  // -Wall not handled...
 		}
 	}
diff --git a/automation/source/simplecm/simplecm.cxx b/automation/source/simplecm/simplecm.cxx
index 440179f..4aa883f 100644
--- automation/source/simplecm/simplecm.cxx
+++ automation/source/simplecm/simplecm.cxx
@@ -147,7 +147,7 @@ void CommunicationLink::SetApplication( const ByteString& aApp )
 }
 
 
-SimpleCommunicationLinkViaSocket::SimpleCommunicationLinkViaSocket( CommunicationManager *pMan, NAMESPACE_VOS(OStreamSocket) *pSocket )
+SimpleCommunicationLinkViaSocket::SimpleCommunicationLinkViaSocket( CommunicationManager *pMan, vos::OStreamSocket *pSocket )
 : CommunicationLink( pMan )
 , aCommunicationPartner()
 , aMyName()
@@ -169,7 +169,7 @@ SimpleCommunicationLinkViaSocket::~SimpleCommunicationLinkViaSocket()
     pStreamSocket = NULL;
 }
 
-void SimpleCommunicationLinkViaSocket::SetStreamSocket( NAMESPACE_VOS(OStreamSocket)* pSocket )
+void SimpleCommunicationLinkViaSocket::SetStreamSocket( vos::OStreamSocket* pSocket )
 {
     if ( pTCPIO )
         pTCPIO->SetStreamSocket( pSocket );
@@ -213,9 +213,9 @@ ByteString SimpleCommunicationLinkViaSocket::GetCommunicationPartner( CM_NameTyp
 			case CM_DOTTED:
 				{
 					rtl::OUString aDotted;
-					NAMESPACE_VOS(OSocketAddr) *pPeerAdr = new NAMESPACE_VOS(OSocketAddr);
+					vos::OSocketAddr *pPeerAdr = new vos::OSocketAddr;
 					pStreamSocket->getPeerAddr( *pPeerAdr );
-					((NAMESPACE_VOS(OInetSocketAddr*))pPeerAdr)->getDottedAddr( aDotted );
+					((vos::OInetSocketAddr*)pPeerAdr)->getDottedAddr( aDotted );
 					delete pPeerAdr;
 					return ByteString( UniString(aDotted), RTL_TEXTENCODING_UTF8 );
 				}
@@ -245,9 +245,9 @@ ByteString SimpleCommunicationLinkViaSocket::GetMyName( CM_NameType eType )
 			case CM_DOTTED:
 				{
 					rtl::OUString aDotted;
-					NAMESPACE_VOS(OSocketAddr) *pPeerAdr = new NAMESPACE_VOS(OSocketAddr);
+					vos::OSocketAddr *pPeerAdr = new vos::OSocketAddr;
 					pStreamSocket->getLocalAddr( *pPeerAdr );
-					((NAMESPACE_VOS(OInetSocketAddr*))pPeerAdr)->getDottedAddr( aDotted );
+					((vos::OInetSocketAddr*)pPeerAdr)->getDottedAddr( aDotted );
 					delete pPeerAdr;
 					return ByteString( UniString(aDotted), RTL_TEXTENCODING_UTF8 );
 				}
@@ -363,7 +363,7 @@ BOOL SimpleCommunicationLinkViaSocket::SendHandshake( HandshakeType aHandshakeTy
 	return !bWasError;
 }
 
-SimpleCommunicationLinkViaSocketWithReceiveCallbacks::SimpleCommunicationLinkViaSocketWithReceiveCallbacks( CommunicationManager *pMan, NAMESPACE_VOS(OStreamSocket) *pSocket )
+SimpleCommunicationLinkViaSocketWithReceiveCallbacks::SimpleCommunicationLinkViaSocketWithReceiveCallbacks( CommunicationManager *pMan, vos::OStreamSocket *pSocket )
 : SimpleCommunicationLinkViaSocket( pMan, pSocket )
 {
 }
@@ -407,7 +407,7 @@ BOOL SimpleCommunicationLinkViaSocketWithReceiveCallbacks::ShutdownCommunication
 	if ( GetStreamSocket() )
 		GetStreamSocket()->close();
 
-    NAMESPACE_VOS(OStreamSocket) *pTempSocket = GetStreamSocket();
+    vos::OStreamSocket *pTempSocket = GetStreamSocket();
     SetStreamSocket( NULL );
     delete pTempSocket;
 
@@ -448,7 +448,7 @@ BOOL CommunicationManager::StartCommunication( ByteString aHost, ULONG nPort )
 ByteString CommunicationManager::GetMyName( CM_NameType )
 {
 	rtl::OUString aHostname;
-	NAMESPACE_VOS(OSocketAddr)::getLocalHostname( aHostname );
+	vos::OSocketAddr::getLocalHostname( aHostname );
 	return ByteString( UniString(aHostname), RTL_TEXTENCODING_UTF8 );
 }
 
@@ -683,8 +683,8 @@ SingleCommunicationManagerClientViaSocket::SingleCommunicationManagerClientViaSo
 
 BOOL CommonSocketFunctions::DoStartCommunication( CommunicationManager *pCM, ICommunicationManagerClient *pCMC, ByteString aHost, ULONG nPort )
 {
-	NAMESPACE_VOS(OInetSocketAddr) Addr;
-	NAMESPACE_VOS(OConnectorSocket) *pConnSocket;
+	vos::OInetSocketAddr Addr;
+	vos::OConnectorSocket *pConnSocket;
 
 	Addr.setAddr( rtl::OUString( UniString( aHost, RTL_TEXTENCODING_UTF8 ) ) );
 	Addr.setPort( nPort );
@@ -694,9 +694,9 @@ BOOL CommonSocketFunctions::DoStartCommunication( CommunicationManager *pCM, ICo
 	aTV.Nanosec = 0;
 	do
 	{
-		pConnSocket = new NAMESPACE_VOS(OConnectorSocket)();
+		pConnSocket = new vos::OConnectorSocket();
 		pConnSocket->setTcpNoDelay( 1 );
-		if ( pConnSocket->connect( Addr, &aTV ) == NAMESPACE_VOS(ISocketTypes::TResult_Ok) )
+		if ( pConnSocket->connect( Addr, &aTV ) == vos::ISocketTypes::TResult_Ok )
 		{
 			pConnSocket->setTcpNoDelay( 1 );
 
diff --git a/automation/source/simplecm/tcpio.cxx b/automation/source/simplecm/tcpio.cxx
index d546690..ca640c7 100644
--- automation/source/simplecm/tcpio.cxx
+++ automation/source/simplecm/tcpio.cxx
@@ -71,7 +71,7 @@ comm_USHORT TCPIO::ReceiveBytes( void* pBuffer, comm_UINT32 nLen )
 
 
 // helper
-void TCPIO::SetStreamSocket( NAMESPACE_VOS(OStreamSocket) *pSocket )
+void TCPIO::SetStreamSocket( vos::OStreamSocket *pSocket )
 {
     osl::MutexGuard aRGuard( aMSocketReadAccess );
     osl::MutexGuard aWGuard( aMSocketWriteAccess );
diff --git a/automation/source/simplecm/tcpio.hxx b/automation/source/simplecm/tcpio.hxx
index bb41764..e078875 100644
--- automation/source/simplecm/tcpio.hxx
+++ automation/source/simplecm/tcpio.hxx
@@ -49,14 +49,14 @@
 class TCPIO : public ITransmiter, public IReceiver
 {
 private:
-	NAMESPACE_VOS(OStreamSocket) *pStreamSocket;
+	vos::OStreamSocket *pStreamSocket;
     osl::Mutex aMSocketReadAccess;
     osl::Mutex aMSocketWriteAccess;
 
 public:
 
 	///
-	TCPIO( NAMESPACE_VOS(OStreamSocket) *pSocket ):pStreamSocket( pSocket ){}
+	TCPIO( vos::OStreamSocket *pSocket ):pStreamSocket( pSocket ){}
 	virtual ~TCPIO(){}
 
 	
@@ -67,7 +67,7 @@ public:
 	virtual comm_USHORT ReceiveBytes( void* pBuffer, comm_UINT32 nLen );
 
     // helper
-	void SetStreamSocket( NAMESPACE_VOS(OStreamSocket) *pSocket );
+	void SetStreamSocket( vos::OStreamSocket *pSocket );
 
 };
 
diff --git a/automation/source/testtool/httprequest.cxx b/automation/source/testtool/httprequest.cxx
index 2090466..778be47 100644
--- automation/source/testtool/httprequest.cxx
+++ automation/source/testtool/httprequest.cxx
@@ -91,7 +91,7 @@ BOOL HttpRequest::Execute()
     Init();
 
     // Open channel to standard redir host
-	NAMESPACE_VOS(OInetSocketAddr) aConnectAddr;
+	vos::OInetSocketAddr aConnectAddr;
 
     if ( aProxyHost.Len() )
     {
@@ -108,8 +108,8 @@ BOOL HttpRequest::Execute()
 	aTV.Seconds = 10;		// Warte 10 Sekunden
 	aTV.Nanosec = 0;
 
-    pOutSocket = new NAMESPACE_VOS(OConnectorSocket)();
-	if ( pOutSocket->connect( aConnectAddr, &aTV ) == NAMESPACE_VOS(ISocketTypes::TResult_Ok) )
+    pOutSocket = new vos::OConnectorSocket();
+	if ( pOutSocket->connect( aConnectAddr, &aTV ) == vos::ISocketTypes::TResult_Ok )
 	{
 //		pOutSocket->setTcpNoDelay( 1 );
 	}
diff --git a/basic/inc/pch/precompiled_basic.hxx b/basic/inc/pch/precompiled_basic.hxx
index 1808ad3..41e4383 100644
--- basic/inc/pch/precompiled_basic.hxx
+++ basic/inc/pch/precompiled_basic.hxx
@@ -284,7 +284,7 @@
 #include "vcl/wrkwin.hxx"
 
 #include "osl/diagnose.h"
-#include "vos/macros.hxx"
+#include "sal/macros.h"
 #include "osl/mutex.hxx"
 
 #include "xmlscript/xmldlg_imexp.hxx"
diff --git a/basic/source/classes/eventatt.cxx b/basic/source/classes/eventatt.cxx
index 048a012..fb6a4a0 100644
--- basic/source/classes/eventatt.cxx
+++ basic/source/classes/eventatt.cxx
@@ -335,7 +335,7 @@ Any BasicScriptListener_Impl::approveFiring( const ScriptEvent& aScriptEvent )
 void BasicScriptListener_Impl::disposing(const EventObject& ) throw ( RuntimeException )
 {
 	// TODO: ???
-	//NAMESPACE_VOS(OGuard) guard( Application::GetSolarMutex() );
+	//vos::OGuard guard( Application::GetSolarMutex() );
 	//xSbxObj.Clear();
 }
 
@@ -577,7 +577,7 @@ Any DialogAllListener_Impl::approveFiring( const AllEventObject& Event )
 void DialogAllListener_Impl::disposing(const EventObject& ) throw ( RuntimeException )
 {
 	// TODO: ???
-	//NAMESPACE_VOS(OGuard) guard( Application::GetSolarMutex() );
+	//vos::OGuard guard( Application::GetSolarMutex() );
 	//xSbxObj.Clear();
 }
 
diff --git a/basic/source/classes/sbxmod.cxx b/basic/source/classes/sbxmod.cxx
index ee6f507..a3de51a 100644
--- basic/source/classes/sbxmod.cxx
+++ basic/source/classes/sbxmod.cxx
@@ -38,7 +38,7 @@
 
 #include <list>
 
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #include <vcl/svapp.hxx>
 
 #ifndef _STREAM_HXX //autogen
diff --git a/binfilter/bf_basic/source/classes/sbxmod.cxx b/binfilter/bf_basic/source/classes/sbxmod.cxx
index 9a6b628..373e2f9 100644
--- binfilter/bf_basic/source/classes/sbxmod.cxx
+++ binfilter/bf_basic/source/classes/sbxmod.cxx
@@ -35,7 +35,7 @@
 
 #include <list>
 
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #include <vcl/svapp.hxx>
 
 #ifndef _STREAM_HXX //autogen
diff --git a/comphelper/source/eventattachermgr/eventattachermgr.cxx b/comphelper/source/eventattachermgr/eventattachermgr.cxx
index 1d5b579..1016399 100644
--- comphelper/source/eventattachermgr/eventattachermgr.cxx
+++ comphelper/source/eventattachermgr/eventattachermgr.cxx
@@ -48,8 +48,8 @@
 #ifndef _OSL_DIAGNOSE_H_
 #include <osl/diagnose.h>
 #endif
-#ifndef _VOS_MACROS_HXX_
-#include <vos/macros.hxx>
+#ifndef _SAL_MACROS_H_
+#include <sal/macros.h>
 #endif
 
 #ifndef _COMPHELPER_EVENTATTACHERMGR_HXX_
diff --git a/connectivity/source/drivers/file/FResultSet.cxx b/connectivity/source/drivers/file/FResultSet.cxx
index 6cd4fee..54f777b 100644
--- connectivity/source/drivers/file/FResultSet.cxx
+++ connectivity/source/drivers/file/FResultSet.cxx
@@ -1007,7 +1007,7 @@ BOOL OResultSet::Move(IResultSetHelper::Movement eCursorPosition, INT32 nOffset,
 //
 	INT32 nTempPos = m_nRowPos;
 	// exclusiver zugriff auf die Tabelle
-	//	NAMESPACE_VOS(OGuard)* pGuard = m_pTable->Lock();
+	//	vos::OGuard* pGuard = m_pTable->Lock();
 
 	if (m_aSQLIterator.getStatementType() == SQL_STATEMENT_SELECT &&
 		!isCount())
@@ -1252,7 +1252,7 @@ BOOL OResultSet::OpenImpl()
 	m_nRowCountResult = -1;
 
 	// exclusiver zugriff auf die Tabelle
-	//	NAMESPACE_VOS(OGuard)* pGuard = pTable->Lock();
+	//	vos::OGuard* pGuard = pTable->Lock();
 	m_nLastVisitedPos = m_pTable->getCurrentLastPos();
 
 	switch(m_aSQLIterator.getStatementType())
diff --git a/extensions/inc/pch/precompiled_extensions.hxx b/extensions/inc/pch/precompiled_extensions.hxx
index 54fdd33..c531290 100644
--- extensions/inc/pch/precompiled_extensions.hxx
+++ extensions/inc/pch/precompiled_extensions.hxx
@@ -545,7 +545,7 @@
 
 #include "osl/conditn.hxx"
 #include "osl/diagnose.h"
-#include "vos/macros.hxx"
+#include "sal/macros.h"
 #include "vos/module.hxx"
 #include "osl/mutex.hxx"
 #include "vos/pipe.hxx"
diff --git a/extensions/source/scanner/scanwin.cxx b/extensions/source/scanner/scanwin.cxx
index 1a1014a..cfa1f64 100644
--- extensions/source/scanner/scanwin.cxx
+++ extensions/source/scanner/scanwin.cxx
@@ -143,7 +143,7 @@ class ImpTwain : public ::cppu::WeakImplHelper1< util::XCloseListener >
 	TW_IDENTITY									aSrcIdent;
 	Link										aNotifyLink;
 	DSMENTRYPROC								pDSM;
-	NAMESPACE_VOS( OModule )*					pMod;
+	vos::OModule*					pMod;
 	ULONG										nCurState;
 	HWND						                hTwainWnd;
 	HHOOK						                hTwainHook;
diff --git a/extensions/source/scanner/twain.cxx b/extensions/source/scanner/twain.cxx
index dd9fcad..27988f9 100644
--- extensions/source/scanner/twain.cxx
+++ extensions/source/scanner/twain.cxx
@@ -250,7 +250,7 @@ void ImpTwain::ImplOpenSourceManager()
 {
 	if( 1 == nCurState )
 	{
-		pMod = new NAMESPACE_VOS( OModule )();
+		pMod = new vos::OModule();
 
 		if( pMod->load( TWAIN_LIBNAME ) )
 		{
diff --git a/extensions/source/scanner/twain.hxx b/extensions/source/scanner/twain.hxx
index 5d43b5c..7b8dbdd 100644
--- extensions/source/scanner/twain.hxx
+++ extensions/source/scanner/twain.hxx
@@ -65,7 +65,7 @@ class ImpTwain
 	Link						aNotifyLink;
 	Bitmap						aBitmap;
 	DSMENTRYPROC				pDSM;
-	NAMESPACE_VOS( OModule )*	pMod;
+	vos::OModule*	pMod;
 	ULONG						nCurState;
 
 	void						ImplCreate();
diff --git a/extensions/workben/testpgp.cxx b/extensions/workben/testpgp.cxx
index a81405c..b83aba4 100644
--- extensions/workben/testpgp.cxx
+++ extensions/workben/testpgp.cxx
@@ -47,8 +47,8 @@
 #include <rtl/wstring>
 #endif
 
-#ifndef _VOS_MACROS_HXX_
-#include <vos/macros.hxx>
+#ifndef _SAL_MACROS_H_
+#include <sal/macros.h>
 #endif
 
 #ifndef _USR_SMARTSERVICES_HXX_
@@ -334,7 +334,7 @@ sal_Int32 SAL_CALL DataSource_Impl::readBytes (
 		throw IOException();
 
 	sal_Int32 k = m_buffer.getLength() - m_position;
-	k = VOS_BOUND(k, 0, nBytesToRead);
+	k = SAL_BOUND(k, 0, nBytesToRead);
 	if (k > 0)
 	{
 		rData.realloc(k);
@@ -606,7 +606,7 @@ BOOL install (
 	String aModule ("module://");
 	char   pBuffer[1024];
 
-	NAMESPACE_VOS(ORealDynamicLoader)::computeModuleName (
+	vos::ORealDynamicLoader::computeModuleName (
 		prefix, pBuffer, sizeof(pBuffer));
 	aModule += pBuffer;
 
@@ -624,7 +624,7 @@ BOOL uninstall (
 	String aModule ("module://");
 	char   pBuffer[1024];
 
-	NAMESPACE_VOS(ORealDynamicLoader)::computeModuleName (
+	vos::ORealDynamicLoader::computeModuleName (
 		prefix, pBuffer, sizeof(pBuffer));
 	aModule += pBuffer;
 
diff --git a/goodies/inc/pch/precompiled_goodies.hxx b/goodies/inc/pch/precompiled_goodies.hxx
index 3adc329..92e927b 100644
--- goodies/inc/pch/precompiled_goodies.hxx
+++ goodies/inc/pch/precompiled_goodies.hxx
@@ -174,7 +174,7 @@
 #include "vcl/virdev.hxx"
 #include "vcl/window.hxx"
 
-#include "vos/macros.hxx"
+#include "sal/macros.h"
 #include "osl/mutex.hxx"
 #include "vos/timer.hxx"
 //---MARKER---
diff --git a/goodies/source/graphic/grfmgr2.cxx b/goodies/source/graphic/grfmgr2.cxx
index e547eab..97c8e30 100644
--- goodies/source/graphic/grfmgr2.cxx
+++ goodies/source/graphic/grfmgr2.cxx
@@ -36,7 +36,7 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_goodies.hxx"
 
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #include <vcl/bmpacc.hxx>
 #include <tools/poly.hxx>
 #include <vcl/outdev.hxx>
diff --git a/sal/inc/sal/macros.h b/sal/inc/sal/macros.h
index 18aa63b..e679905 100644
--- sal/inc/sal/macros.h
+++ sal/inc/sal/macros.h
@@ -42,6 +42,9 @@
 #ifndef SAL_MIN
 #	define SAL_MIN(a,b)            (((a) < (b)) ? (a) : (b))
 #endif
+#ifndef SAL_ABS
+#	define SAL_ABS(a)              (((a) < 0) ? (-(a)) : (a))
+#endif
 
 #ifndef SAL_FIELDOFFSET
 #	define SAL_FIELDOFFSET(type, field) ((sal_Int32)(&((type *)16)->field) - 16)
diff --git a/sc/inc/pch/precompiled_sc.hxx b/sc/inc/pch/precompiled_sc.hxx
index 7566c40..bbef9da 100644
--- sc/inc/pch/precompiled_sc.hxx
+++ sc/inc/pch/precompiled_sc.hxx
@@ -251,7 +251,7 @@
 #include <vcl/vclevent.hxx>
 #include <vcl/window.hxx>
 #include <vcl/wintypes.hxx>
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #include <vos/object.hxx>
 #include <vos/types.hxx>
 #include <wchar.h>
diff --git a/sc/source/filter/inc/filt_pch.hxx b/sc/source/filter/inc/filt_pch.hxx
index 7f5d399..d31ae05 100644
--- sc/source/filter/inc/filt_pch.hxx
+++ sc/source/filter/inc/filt_pch.hxx
@@ -173,7 +173,7 @@
 #include <svx/fhgtitem.hxx>
 #include <vos/types.hxx>
 #include <vos/object.hxx>
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #include <tools/unqid.hxx>
 #include <com/sun/star/uno/Sequence.h>
 #include <uno/sequence2.h>
diff --git a/sfx2/source/dialog/filedlghelper.cxx b/sfx2/source/dialog/filedlghelper.cxx
index e2a1312..5b4f7cf 100644
--- sfx2/source/dialog/filedlghelper.cxx
+++ sfx2/source/dialog/filedlghelper.cxx
@@ -2568,7 +2568,7 @@ void FileDialogHelper::SetDisplayDirectory( const String& _rPath )
         if ( sFolder.getLength() == 0 )
         {
             // _rPath is not a valid path -> fallback to home directory
-            NAMESPACE_VOS( OSecurity ) aSecurity;
+            vos::OSecurity aSecurity;
             aSecurity.getHomeDir( sFolder );
         }
         mpImp->displayFolder( sFolder );
diff --git a/so3/source/persist/transprt.cxx b/so3/source/persist/transprt.cxx
index 017b4ff..aad5e17 100644
--- so3/source/persist/transprt.cxx
+++ so3/source/persist/transprt.cxx
@@ -124,8 +124,8 @@
 #include <cppuhelper/weak.hxx>
 #endif
 
-#ifndef _VOS_MACROS_HXX_
-#include <vos/macros.hxx>
+#ifndef _SAL_MACROS_H_
+#include <sal/macros.h>
 #endif
 #ifndef _OSL_MUTEX_HXX_
 #include <osl/mutex.hxx>
@@ -715,7 +715,7 @@ ErrCode UcbTransportLockBytes::ReadAt (
 	Sequence<sal_Int8> aData;
 	sal_Int32          nSize;
 
-	nCount = VOS_MIN(nCount, 0x7FFFFFFF);
+	nCount = SAL_MIN(nCount, 0x7FFFFFFF);
 	try
 	{
 		while (!m_bTerminated)
@@ -747,7 +747,7 @@ ErrCode UcbTransportLockBytes::ReadAt (
 	if (pRead)
 		*pRead = ULONG(nSize);
 
-	sal_uInt32 nRead = VOS_MAX(m_nRead, nPos + nSize);
+	sal_uInt32 nRead = SAL_MAX(m_nRead, nPos + nSize);
 	SAL_CONST_CAST(UcbTransportLockBytes*, this)->m_nRead = nRead;
 
 	return ERRCODE_NONE;
diff --git a/sot/source/sdstor/stgcache.cxx b/sot/source/sdstor/stgcache.cxx
index 1891f9a..054d463 100644
--- sot/source/sdstor/stgcache.cxx
+++ sot/source/sdstor/stgcache.cxx
@@ -43,7 +43,7 @@
 #if defined(_MSC_VER) && (_MSC_VER<1200)
 #include <tools/postsys.h>
 #endif
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 
 #include <string.h>
 
@@ -68,7 +68,7 @@ typedef std::hash_map
 	INT32,
 	StgPage *,
 	std::hash< INT32 >,
-	NAMESPACE_STD(equal_to)< INT32 >
+	std::equal_to< INT32 >
 > UsrStgPagePtr_Impl;
 #ifdef WNT
 #pragma warning( disable: 4786 )
diff --git a/starmath/source/math_pch.cxx b/starmath/source/math_pch.cxx
index 598d33a..0d93acd 100644
--- starmath/source/math_pch.cxx
+++ starmath/source/math_pch.cxx
@@ -131,7 +131,7 @@
 #include <osl/types.h>
 #include <osl/interlck.h>
 #include <vos/object.hxx>
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #include <smdll.hxx>
 #include <sfx2/sfxsids.hrc>
 #include <chaos/cntids.hrc>
diff --git a/svtools/source/dialogs/filedlg2.cxx b/svtools/source/dialogs/filedlg2.cxx
index 1c47520..d748d7b 100644
--- svtools/source/dialogs/filedlg2.cxx
+++ svtools/source/dialogs/filedlg2.cxx
@@ -373,7 +373,7 @@ IMPL_LINK( ImpPathDialog, ClickHdl, Button*, pBtn )
 	if ( pBtn == pHomeBtn )
 	{
 		::rtl::OUString aHomeDir;
-		NAMESPACE_VOS( OSecurity ) aSecurity;
+		vos::OSecurity aSecurity;
 		if ( aSecurity.getHomeDir( aHomeDir ) )
 		{
 			DirEntry aFile ( aHomeDir );
diff --git a/tools/inc/pch/precompiled_tools.hxx b/tools/inc/pch/precompiled_tools.hxx
index b9a1b08..95aa23b 100644
--- tools/inc/pch/precompiled_tools.hxx
+++ tools/inc/pch/precompiled_tools.hxx
@@ -91,7 +91,7 @@
 #include "sys/stat.h"
 #include "sys/types.h"
 
-#include "vos/macros.hxx"
+#include "sal/macros.h"
 #include "osl/mutex.hxx"
 #include "vos/signal.hxx"
 #include "vos/timer.hxx"
diff --git a/tools/source/generic/color.cxx b/tools/source/generic/color.cxx
index 97dcb27..3d1f4dd 100644
--- tools/source/generic/color.cxx
+++ tools/source/generic/color.cxx
@@ -38,8 +38,8 @@
 
 #include <stdlib.h>
 
-#ifndef _VOS_MACROS_HXX_
-#include <vos/macros.hxx>
+#ifndef _SAL_MACROS_H_
+#include <sal/macros.h>
 #endif
 
 #ifndef _TOOLS_COLOR_HXX
@@ -114,18 +114,18 @@ UINT8 Color::GetColorError( const Color& rCompareColor ) const
 
 void Color::IncreaseLuminance( UINT8 cLumInc )
 {
-	SetRed( (UINT8) VOS_BOUND( (long) COLORDATA_RED( mnColor ) + cLumInc, 0L, 255L ) );
-	SetGreen( (UINT8) VOS_BOUND( (long) COLORDATA_GREEN( mnColor ) + cLumInc, 0L, 255L ) );
-	SetBlue( (UINT8) VOS_BOUND( (long) COLORDATA_BLUE( mnColor ) + cLumInc, 0L, 255L ) );
+	SetRed( (UINT8) SAL_BOUND( (long) COLORDATA_RED( mnColor ) + cLumInc, 0L, 255L ) );
+	SetGreen( (UINT8) SAL_BOUND( (long) COLORDATA_GREEN( mnColor ) + cLumInc, 0L, 255L ) );
+	SetBlue( (UINT8) SAL_BOUND( (long) COLORDATA_BLUE( mnColor ) + cLumInc, 0L, 255L ) );
 }
 
 // -----------------------------------------------------------------------
 
 void Color::DecreaseLuminance( UINT8 cLumDec )
 {
-	SetRed( (UINT8) VOS_BOUND( (long) COLORDATA_RED( mnColor ) - cLumDec, 0L, 255L ) );
-	SetGreen( (UINT8) VOS_BOUND( (long) COLORDATA_GREEN( mnColor ) - cLumDec, 0L, 255L ) );
-	SetBlue( (UINT8) VOS_BOUND( (long) COLORDATA_BLUE( mnColor ) - cLumDec, 0L, 255L ) );
+	SetRed( (UINT8) SAL_BOUND( (long) COLORDATA_RED( mnColor ) - cLumDec, 0L, 255L ) );
+	SetGreen( (UINT8) SAL_BOUND( (long) COLORDATA_GREEN( mnColor ) - cLumDec, 0L, 255L ) );
+	SetBlue( (UINT8) SAL_BOUND( (long) COLORDATA_BLUE( mnColor ) - cLumDec, 0L, 255L ) );
 }
 
 // -----------------------------------------------------------------------
@@ -137,9 +137,9 @@ void Color::IncreaseContrast( UINT8 cContInc )
 		const double fM = 128.0 / ( 128.0 - 0.4985 * cContInc );
 		const double fOff = 128.0 - fM * 128.0;
 
-		SetRed( (UINT8) VOS_BOUND( _FRound( COLORDATA_RED( mnColor ) * fM + fOff ), 0L, 255L ) );
-		SetGreen( (UINT8) VOS_BOUND( _FRound( COLORDATA_GREEN( mnColor ) * fM + fOff ), 0L, 255L ) );
-		SetBlue( (UINT8) VOS_BOUND( _FRound( COLORDATA_BLUE( mnColor ) * fM + fOff ), 0L, 255L ) );
+		SetRed( (UINT8) SAL_BOUND( _FRound( COLORDATA_RED( mnColor ) * fM + fOff ), 0L, 255L ) );
+		SetGreen( (UINT8) SAL_BOUND( _FRound( COLORDATA_GREEN( mnColor ) * fM + fOff ), 0L, 255L ) );
+		SetBlue( (UINT8) SAL_BOUND( _FRound( COLORDATA_BLUE( mnColor ) * fM + fOff ), 0L, 255L ) );
 	}
 }
 
@@ -152,9 +152,9 @@ void Color::DecreaseContrast( UINT8 cContDec )
 		const double fM = ( 128.0 - 0.4985 * cContDec ) / 128.0;
 		const double fOff = 128.0 - fM * 128.0;
 
-		SetRed( (UINT8) VOS_BOUND( _FRound( COLORDATA_RED( mnColor ) * fM + fOff ), 0L, 255L ) );
-		SetGreen( (UINT8) VOS_BOUND( _FRound( COLORDATA_GREEN( mnColor ) * fM + fOff ), 0L, 255L ) );
-		SetBlue( (UINT8) VOS_BOUND( _FRound( COLORDATA_BLUE( mnColor ) * fM + fOff ), 0L, 255L ) );
+		SetRed( (UINT8) SAL_BOUND( _FRound( COLORDATA_RED( mnColor ) * fM + fOff ), 0L, 255L ) );
+		SetGreen( (UINT8) SAL_BOUND( _FRound( COLORDATA_GREEN( mnColor ) * fM + fOff ), 0L, 255L ) );
+		SetBlue( (UINT8) SAL_BOUND( _FRound( COLORDATA_BLUE( mnColor ) * fM + fOff ), 0L, 255L ) );
 	}
 }
 
diff --git a/tools/source/rc/resmgr.cxx b/tools/source/rc/resmgr.cxx
index de4655d..e5dcdcf 100644
--- tools/source/rc/resmgr.cxx
+++ tools/source/rc/resmgr.cxx
@@ -882,18 +882,18 @@ void ResMgr::RscError_Impl( const sal_Char* pMessage, ResMgr* pResMgr,
 
 static void RscException_Impl()
 {
-    switch ( NAMESPACE_VOS(OSignalHandler)::raise( OSL_SIGNAL_USER_RESOURCEFAILURE, (void*)"" ) )
+    switch ( vos::OSignalHandler::raise( OSL_SIGNAL_USER_RESOURCEFAILURE, (void*)"" ) )
     {
-        case NAMESPACE_VOS(OSignalHandler)::TAction_CallNextHandler:
+        case vos::OSignalHandler::TAction_CallNextHandler:
             abort();
 
-        case NAMESPACE_VOS(OSignalHandler)::TAction_Ignore:
+        case vos::OSignalHandler::TAction_Ignore:
             return;
 
-        case NAMESPACE_VOS(OSignalHandler)::TAction_AbortApplication:
+        case vos::OSignalHandler::TAction_AbortApplication:
             abort();
 
-        case NAMESPACE_VOS(OSignalHandler)::TAction_KillApplication:
+        case vos::OSignalHandler::TAction_KillApplication:
             exit(-1);
     }
 }
diff --git a/transex3/source/wrdtrans.cxx b/transex3/source/wrdtrans.cxx
index 28ac308..3e26948 100644
--- transex3/source/wrdtrans.cxx
+++ transex3/source/wrdtrans.cxx
@@ -42,7 +42,7 @@
 
 // NOT FULLY DECLARED SERVICES
 #include <vector>
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 
 
 #include <tools/stream.hxx>
@@ -52,7 +52,7 @@
 */
 //**************   Declaration WordTrans_ErrorList   ******************//
 /*
-typedef NAMESPACE_STD(vector)<ByteString> Stl_ByteStringList;
+typedef std::vector<ByteString> Stl_ByteStringList;
 
 class WordTrans_ErrorList
 {
diff --git a/unotools/source/ucbhelper/ucblockbytes.cxx b/unotools/source/ucbhelper/ucblockbytes.cxx
index 07d0541..d3c8412 100644
--- unotools/source/ucbhelper/ucblockbytes.cxx
+++ unotools/source/ucbhelper/ucblockbytes.cxx
@@ -39,6 +39,8 @@
 #include <unotools/ucblockbytes.hxx>
 #include <comphelper/processfactory.hxx>
 
+#include <sal/macros.h>
+
 #ifndef _SALHELPER_CONDITION_HXX_
 #include <salhelper/condition.hxx>
 #endif
@@ -1533,7 +1535,7 @@ ErrCode UcbLockBytes::ReadAt ( ULONG nPos, void *pBuffer, ULONG nCount, ULONG *p
 	Sequence<sal_Int8> aData;
 	sal_Int32          nSize;
 
-	nCount = VOS_MIN(nCount, 0x7FFFFFFF);
+	nCount = SAL_MIN(nCount, 0x7FFFFFFF);
 	try
 	{
 		if ( !m_bTerminated && !IsSynchronMode() )
diff --git a/vcl/source/gdi/bitmap4.cxx b/vcl/source/gdi/bitmap4.cxx
index 88a574e..4e85b73 100644
--- vcl/source/gdi/bitmap4.cxx
+++ vcl/source/gdi/bitmap4.cxx
@@ -37,7 +37,7 @@
 #include "precompiled_vcl.hxx"
 
 #include <stdlib.h>
-#include <vos/macros.hxx>
+#include <sal/macros.h>
 #ifndef _SV_BMPACC_HXX
 #include <vcl/bmpacc.hxx>
 #endif
@@ -486,7 +486,7 @@ BOOL Bitmap::ImplSobelGrey( const BmpFilterParam* /*pFilterParam*/, const Link*
 						nSum2 += nMask332 * nGrey33;
 
 						nSum1 = (long) sqrt( (double)( nSum1 * nSum1 + nSum2 * nSum2 ) );
-						aGrey.SetIndex( ~(BYTE) VOS_BOUND( nSum1, 0, 255 ) );
+						aGrey.SetIndex( ~(BYTE) SAL_BOUND( nSum1, 0, 255 ) );
 						pWriteAcc->SetPixel( nY, nX, aGrey );
 
 						if( nX < ( nWidth - 1 ) )
@@ -561,7 +561,7 @@ BOOL Bitmap::ImplEmbossGrey( const BmpFilterParam* pFilterParam, const Link* /*p
 				const long	nLz = FRound( sin( fElev ) * 255.0 );
 				const long	nZ2 = ( ( 6 * 255 ) / 4 ) * ( ( 6 * 255 ) / 4 );
 				const long	nNzLz = ( ( 6 * 255 ) / 4 ) * nLz;
-				const BYTE	cLz = (BYTE) VOS_BOUND( nLz, 0, 255 );
+				const BYTE	cLz = (BYTE) SAL_BOUND( nLz, 0, 255 );
 
 				// fill mapping tables
 				pHMap[ 0 ] = 0;
@@ -598,7 +598,7 @@ BOOL Bitmap::ImplEmbossGrey( const BmpFilterParam* pFilterParam, const Link* /*p
 						else
 						{
 							const double fGrey = nDotL / sqrt( (double)(nNx * nNx + nNy * nNy + nZ2) );
-							aGrey.SetIndex( (BYTE) VOS_BOUND( fGrey, 0, 255 ) );
+							aGrey.SetIndex( (BYTE) SAL_BOUND( fGrey, 0, 255 ) );
 						}
 						
 						pWriteAcc->SetPixel( nY, nX, aGrey );
@@ -699,7 +699,7 @@ BOOL Bitmap::ImplSepia( const BmpFilterParam* pFilterParam, const Link* /*pProgr
 	{
 		long			nSepiaPercent = ( pFilterParam && pFilterParam->meFilter == BMP_FILTER_SEPIA ) ? 
 									    pFilterParam->mcSolarGreyThreshold : 10;
-		const long		nSepia = 10000 - 100 * VOS_BOUND( nSepiaPercent, 0, 100 );
+		const long		nSepia = 10000 - 100 * SAL_BOUND( nSepiaPercent, 0, 100 );
 		BitmapPalette	aSepiaPal( 256 );
 
 		DBG_ASSERT( nSepiaPercent <= 100, "Bitmap::ImplSepia(): sepia value out of range; defaulting to 100%" );
diff --git a/vcl/source/gdi/gdimtf.cxx b/vcl/source/gdi/gdimtf.cxx
index 7d30239..0d97616 100644
--- vcl/source/gdi/gdimtf.cxx
+++ vcl/source/gdi/gdimtf.cxx
@@ -36,8 +36,8 @@
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_vcl.hxx"
 
-#ifndef _VOS_MACROS_HXX_
-#include <vos/macros.hxx>
+#ifndef _SAL_MACROS_H_
+#include <sal/macros.h>
 #endif
 #ifndef _RTL_CRC_H_
 #include <rtl/crc.h>
@@ -808,12 +808,12 @@ BOOL GDIMetaFile::Mirror( ULONG nMirrorFlags )
     BOOL        bRet;
 
 	if( nMirrorFlags & MTF_MIRROR_HORZ )
-		nMoveX = VOS_ABS( aOldPrefSize.Width() ) - 1, fScaleX = -1.0;
+		nMoveX = SAL_ABS( aOldPrefSize.Width() ) - 1, fScaleX = -1.0;
 	else
 		nMoveX = 0, fScaleX = 1.0;
 
 	if( nMirrorFlags & MTF_MIRROR_VERT )
-		nMoveY = VOS_ABS( aOldPrefSize.Height() ) - 1, fScaleY = -1.0;
+		nMoveY = SAL_ABS( aOldPrefSize.Height() ) - 1, fScaleY = -1.0;
 	else
 		nMoveY = 0, fScaleY = 1.0;
 
diff --git a/vcl/unx/source/app/salimpsound.hxx b/vcl/unx/source/app/salimpsound.hxx
index fcb24e5..b13b65b 100644
--- vcl/unx/source/app/salimpsound.hxx
+++ vcl/unx/source/app/salimpsound.hxx
@@ -235,7 +235,7 @@ class RPTPSound : public VSound
         DECLARE_LIST( RPTPSoundList, RPTPSound* )
 
 	static BOOL								s_bConnected;
-	static NAMESPACE_VOS(OConnectorSocket)		s_aConnector;
+	static vos::OConnectorSocket		s_aConnector;
 	static RPTPSoundList						s_aSounds;
 	static BOOL								s_bFailedOnce;
 
diff --git a/vos/inc/vos/execabl.hxx b/vos/inc/vos/execabl.hxx
index 5e932a1..f229f09 100644
--- vos/inc/vos/execabl.hxx
+++ vos/inc/vos/execabl.hxx
@@ -62,7 +62,7 @@ namespace vos
 	@version 0.1
 */
 
-class IExecutable : public NAMESPACE_VOS(IReference) 
+class IExecutable : public vos::IReference 
 {
 public:
 
@@ -94,8 +94,8 @@ public:
 /** OExecutable
 	added default impl. of IReferenceCounter 
 */
-class OExecutable : public NAMESPACE_VOS(IExecutable),
-					public NAMESPACE_VOS(OReference)
+class OExecutable : public vos::IExecutable,
+					public vos::OReference
 					
 {
 public:
diff --git a/vos/inc/vos/pipe.hxx b/vos/inc/vos/pipe.hxx
index a8336ac..7a1f8ac 100644
--- vos/inc/vos/pipe.hxx
+++ vos/inc/vos/pipe.hxx
@@ -70,10 +70,10 @@ class OStreamPipe;
 
 /** Represents a pipe.
 */
-class OPipe : public NAMESPACE_VOS(OReference),
-              public NAMESPACE_VOS(OObject)
+class OPipe : public vos::OReference,
+              public vos::OObject
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OPipe));
+    VOS_DECLARE_CLASSINFO(vos::OPipe);
 
 public:
     /*
@@ -123,7 +123,7 @@ public:
         @param Security
     */
     OPipe(const ::rtl::OUString& strName, TPipeOption Options,
-          const NAMESPACE_VOS(OSecurity)& rSecurity);
+          const vos::OSecurity& rSecurity);
 
     /** Copy constructor.
     */
@@ -154,7 +154,7 @@ public:
         @return True if socket was successfully created.
     */
     sal_Bool SAL_CALL create(const ::rtl::OUString& strName, TPipeOption Options,
-                   const NAMESPACE_VOS(OSecurity)& rSecurity);
+                   const vos::OSecurity& rSecurity);
 
     /** Assignment operator. If pipe was already created, the old one will
         be discarded.
@@ -213,10 +213,10 @@ public:
 
 /** A pipe to send or receive a stream of data.
 */
-class OStreamPipe : public NAMESPACE_VOS(OPipe),
-                    public NAMESPACE_VOS(IStream)
+class OStreamPipe : public vos::OPipe,
+                    public vos::IStream
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OStreamPipe));
+    VOS_DECLARE_CLASSINFO(vos::OStreamPipe);
 public:
 
     /** Creates an unattached pipe. You must attach the pipe to an oslPipe
diff --git a/vos/inc/vos/process.hxx b/vos/inc/vos/process.hxx
index bb77f96..d8db5b6 100644
--- vos/inc/vos/process.hxx
+++ vos/inc/vos/process.hxx
@@ -301,7 +301,7 @@ class OExtCommandLineImpl;
 class OExtCommandLine : public OObject
 {
     VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OExtCommandLine, vos));
-    static NAMESPACE_VOS(OExtCommandLineImpl)* pExtImpl;
+    static vos::OExtCommandLineImpl* pExtImpl;
 
 public:
 
diff --git a/vos/inc/vos/refernce.hxx b/vos/inc/vos/refernce.hxx
index 84a154e..6256f6e 100644
--- vos/inc/vos/refernce.hxx
+++ vos/inc/vos/refernce.hxx
@@ -96,7 +96,7 @@ private:
 	ORefCount& SAL_CALL operator= (const ORefCount&);
 };
 
-class OReference : public NAMESPACE_VOS(IReference)
+class OReference : public vos::IReference
 {
 public:
 	OReference();
diff --git a/vos/inc/vos/signal.hxx b/vos/inc/vos/signal.hxx
index be36f5e..dd10c07 100644
--- vos/inc/vos/signal.hxx
+++ vos/inc/vos/signal.hxx
@@ -60,7 +60,7 @@ SignalHandlerFunction_impl signalHandlerFunction_impl;
 	@version 1.0
 */
 
-class OSignalHandler : public NAMESPACE_VOS(OObject)
+class OSignalHandler : public vos::OObject
 {
 	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OSignalHandler, vos));
 
diff --git a/vos/inc/vos/socket.hxx b/vos/inc/vos/socket.hxx
index 8db0783..e84c8df 100644
--- vos/inc/vos/socket.hxx
+++ vos/inc/vos/socket.hxx
@@ -211,7 +211,7 @@ public:
 
 /** Base class for socket addresses.
 */
-class ISocketAddr : public NAMESPACE_VOS(ISocketTypes)
+class ISocketAddr : public vos::ISocketTypes
 {
 public:
     virtual ~ISocketAddr() { }
@@ -224,11 +224,11 @@ public:
     virtual sal_Bool SAL_CALL operator== (oslSocketAddr Addr)= 0;
 };
 
-class OSocketAddr : public NAMESPACE_VOS(ISocketAddr),
-            public NAMESPACE_VOS(OObject)
+class OSocketAddr : public vos::ISocketAddr,
+            public vos::OObject
 
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OSocketAddr));
+    VOS_DECLARE_CLASSINFO(vos::OSocketAddr);
 public:
 
     /** Creates socket address of unknown type.
@@ -296,9 +296,9 @@ protected:
 
 /** Represents an internet-address.
 */
-class OInetSocketAddr : public NAMESPACE_VOS(OSocketAddr)
+class OInetSocketAddr : public vos::OSocketAddr
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OInetSocketAddr));
+    VOS_DECLARE_CLASSINFO(vos::OInetSocketAddr);
 public:
 
     /** Creates an empty internet-address (INADDR_ANY).
@@ -381,9 +381,9 @@ public:
 
 /** Represents an IPX/SPX address.
 */
-class OIpxSocketAddr :  public NAMESPACE_VOS(OSocketAddr)
+class OIpxSocketAddr :  public vos::OSocketAddr
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OIpxSocketAddr));
+    VOS_DECLARE_CLASSINFO(vos::OIpxSocketAddr);
 public:
 
     typedef oslSocketIpxNetNumber  TIpxNetNumber;
@@ -452,11 +452,11 @@ public:
 
 /** Represents a socket.
 */
-class OSocket : public NAMESPACE_VOS(ISocketTypes),
-                public NAMESPACE_VOS(OReference),
-                public NAMESPACE_VOS(OObject)
+class OSocket : public vos::ISocketTypes,
+                public vos::OReference,
+                public vos::OObject
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OSocket));
+    VOS_DECLARE_CLASSINFO(vos::OSocket);
 
 protected:
     typedef ORefObj<oslSocket> SockRef;
@@ -831,10 +831,10 @@ public:
 
 /** A socket to send or receive a stream of data.
 */
-class OStreamSocket : public NAMESPACE_VOS(OSocket),
-                      public NAMESPACE_VOS(IStream)
+class OStreamSocket : public vos::OSocket,
+                      public vos::IStream
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OStreamSocket));
+    VOS_DECLARE_CLASSINFO(vos::OStreamSocket);
 public:
 
     /** Creates an unattached socket. You must attach the socket to an oslSocket
@@ -974,9 +974,9 @@ protected:
 
 /** A socket to accept incoming connections.
 */
-class OAcceptorSocket : public NAMESPACE_VOS(OSocket)
+class OAcceptorSocket : public vos::OSocket
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OAcceptorSocket));
+    VOS_DECLARE_CLASSINFO(vos::OAcceptorSocket);
 public:
 
     /** Creates a socket that can accept connections.
@@ -1036,9 +1036,9 @@ public:
 
 /** A socket to initiate a conenction.
 */
-class OConnectorSocket : public NAMESPACE_VOS(OStreamSocket)
+class OConnectorSocket : public vos::OStreamSocket
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(OConnectorSocket));
+    VOS_DECLARE_CLASSINFO(vos::OConnectorSocket);
 public:
 
     /** Creates a socket that can accept connections.
@@ -1072,9 +1072,9 @@ public:
 
 /** A connectionless socket to send and receive datagrams.
 */
-class ODatagramSocket : public NAMESPACE_VOS(OSocket)
+class ODatagramSocket : public vos::OSocket
 {
-    VOS_DECLARE_CLASSINFO(NAMESPACE_VOS(ODatagramSocket));
+    VOS_DECLARE_CLASSINFO(vos::ODatagramSocket);
 public:
 
     /** Creates a datagram socket.
diff --git a/vos/inc/vos/stream.hxx b/vos/inc/vos/stream.hxx
index 8f4e0bc..096f075 100644
--- vos/inc/vos/stream.hxx
+++ vos/inc/vos/stream.hxx
@@ -51,7 +51,7 @@ namespace vos
 
 /** Adds seeking capabilities to IStream
 */
-class IPositionableStream : public NAMESPACE_VOS(IStream)
+class IPositionableStream : public vos::IStream
 {
 
 public:
@@ -89,8 +89,8 @@ protected:
 
 /** Implements IPositionableStream
 */
-class OStream : public NAMESPACE_VOS(OObject), 
-				public NAMESPACE_VOS(IPositionableStream)
+class OStream : public vos::OObject, 
+				public vos::IPositionableStream
 {
 	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OStream, vos));
 
diff --git a/vos/inc/vos/thread.hxx b/vos/inc/vos/thread.hxx
index 60833ba..57fccb9 100644
--- vos/inc/vos/thread.hxx
+++ vos/inc/vos/thread.hxx
@@ -73,8 +73,8 @@ ThreadWorkerFunction_impl threadWorkerFunction_impl;
 	@version 1.0
 */
 
-class OThread : public NAMESPACE_VOS(IRunnable),
-				public NAMESPACE_VOS(OObject)
+class OThread : public vos::IRunnable,
+				public vos::OObject
 {
 
 	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OThread, vos));
@@ -223,7 +223,7 @@ protected:
 	friend void threadWorkerFunction_impl(void *);
 };
 
-class OThreadData : public NAMESPACE_VOS(OObject)
+class OThreadData : public vos::OObject
 {
 	VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OThreadData, vos));
 
diff --git a/vos/source/pipe.cxx b/vos/source/pipe.cxx
index 0db2016..d280fae 100644
--- vos/source/pipe.cxx
+++ vos/source/pipe.cxx
@@ -146,7 +146,7 @@ sal_Bool OPipe::create( const rtl::OUString& strName, TPipeOption Options )
 /*****************************************************************************/
 sal_Bool OPipe::create( const rtl::OUString& strName,
 						TPipeOption Options,
-						const NAMESPACE_VOS(OSecurity)& rSecurity ) 
+						const vos::OSecurity& rSecurity ) 
 {
 	// if this was a valid pipe, decrease reference
 	if ((m_pPipeRef) && (m_pPipeRef->release() == 0))
diff --git a/vos/source/signal.cxx b/vos/source/signal.cxx
index a0f8f57..3a576b1 100644
--- vos/source/signal.cxx
+++ vos/source/signal.cxx
@@ -43,7 +43,7 @@ using namespace vos;
 oslSignalAction vos::signalHandlerFunction_impl(
     void * pthis, oslSignalInfo * pInfo)
 {
-	NAMESPACE_VOS(OSignalHandler)* pThis= (NAMESPACE_VOS(OSignalHandler)*)pthis;
+	vos::OSignalHandler* pThis= (vos::OSignalHandler*)pthis;
 
 	return ((oslSignalAction)pThis->signal(pInfo));
 }
diff --git a/vos/source/thread.cxx b/vos/source/thread.cxx
index a70ef36..324da86 100644
--- vos/source/thread.cxx
+++ vos/source/thread.cxx
@@ -44,7 +44,7 @@ using namespace vos;
 
 void vos::threadWorkerFunction_impl(void * pthis)
 {
-	NAMESPACE_VOS(OThread)* pThis= (NAMESPACE_VOS(OThread)*)pthis;
+	vos::OThread* pThis= (vos::OThread*)pthis;
 
 	// call Handler-Function of OThread-derived class
 	pThis->run();
diff --git a/vos/source/timer.cxx b/vos/source/timer.cxx
index 4750ac6..00a3089 100644
--- vos/source/timer.cxx
+++ vos/source/timer.cxx
@@ -51,7 +51,7 @@
 
 class OTimerManagerCleanup;
 
-class NAMESPACE_VOS(OTimerManager) : public NAMESPACE_VOS(OThread)
+class vos::OTimerManager : public vos::OThread
 {
     
 public:
@@ -63,13 +63,13 @@ public:
   	~OTimerManager();
     
   	/// register timer
-    sal_Bool SAL_CALL registerTimer(NAMESPACE_VOS(OTimer)* pTimer);
+    sal_Bool SAL_CALL registerTimer(vos::OTimer* pTimer);
 
   	/// unregister timer
-    sal_Bool SAL_CALL unregisterTimer(NAMESPACE_VOS(OTimer)* pTimer);
+    sal_Bool SAL_CALL unregisterTimer(vos::OTimer* pTimer);
 
   	/// lookup timer
-    sal_Bool SAL_CALL lookupTimer(const NAMESPACE_VOS(OTimer)* pTimer);
+    sal_Bool SAL_CALL lookupTimer(const vos::OTimer* pTimer);
 
 	/// retrieves the "Singleton" TimerManager Instance
 	static OTimerManager* SAL_CALL getTimerManager();
@@ -87,7 +87,7 @@ protected:
     virtual void SAL_CALL onTerminated();
     
   	// sorted-queue data
-  	NAMESPACE_VOS(OTimer)*		m_pHead;
+  	vos::OTimer*		m_pHead;
     // List Protection
     osl::Mutex					m_Lock;
     // Signal the insertion of a timer
@@ -97,7 +97,7 @@ protected:
 	static osl::Mutex 			m_Access;
 
     // "Singleton Pattern"
-    static NAMESPACE_VOS(OTimerManager)* m_pManager;
+    static vos::OTimerManager* m_pManager;
 
     friend class OTimerManagerCleanup;
     
@@ -277,8 +277,8 @@ TTimeValue OTimer::getRemainingTime() const
 // Timer manager
 //
 
-osl::Mutex NAMESPACE_VOS(OTimerManager)::m_Access;
-OTimerManager* NAMESPACE_VOS(OTimerManager)::m_pManager=0;
+osl::Mutex vos::OTimerManager::m_Access;
+OTimerManager* vos::OTimerManager::m_pManager=0;
 
 OTimerManager::OTimerManager()
 {
