diff --git a/binfilter/bf_sfx2/source/appl/sfx2_appinit.cxx b/binfilter/bf_sfx2/source/appl/sfx2_appinit.cxx
index 6b50346..2a352b6 100644
--- binfilter/bf_sfx2/source/appl/sfx2_appinit.cxx
+++ binfilter/bf_sfx2/source/appl/sfx2_appinit.cxx
@@ -82,8 +82,8 @@
 #ifndef _RTL_USTRBUF_HXX_
 #include <rtl/ustrbuf.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 #ifndef _UCBHELPER_CONFIGURATIONKEYS_HXX_
 #include <ucbhelper/configurationkeys.hxx>
diff --git a/desktop/source/app/app.cxx b/desktop/source/app/app.cxx
index c78329f..b08b8de 100644
--- desktop/source/app/app.cxx
+++ desktop/source/app/app.cxx
@@ -199,8 +199,8 @@
 #ifndef _OSL_MUTEX_HXX_
 #include <osl/mutex.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 #ifndef _RTL_REF_HXX_
 #include <rtl/ref.hxx>
diff --git a/desktop/source/app/appinit.cxx b/desktop/source/app/appinit.cxx
index a9094f4..bef1689 100644
--- desktop/source/app/appinit.cxx
+++ desktop/source/app/appinit.cxx
@@ -136,7 +136,7 @@
 #include <svtools/internaloptions.hxx>
 #endif
 
-#include <vos/security.hxx>
+#include <osl/security.hxx>
 
 
 #define	DEFINE_CONST_OUSTRING(CONSTASCII)		OUString(RTL_CONSTASCII_USTRINGPARAM(CONSTASCII))
@@ -144,7 +144,6 @@
 #define DESKTOP_TEMPDIRNAME						"soffice.tmp"
 
 using namespace rtl;
-using namespace vos;
 using namespace desktop;
 using namespace ::com::sun::star::uno;
 using namespace ::com::sun::star::lang;
@@ -169,7 +168,7 @@ static bool configureUcb(bool bServer, rtl::OUString const & rPortalConnect)
 	}
 
 	rtl::OUString aPipe;
-	vos::OSecurity().getUserIdent(aPipe);
+	osl::Security().getUserIdent(aPipe);
 
 	rtl::OUStringBuffer aPortal;
 	if (rPortalConnect.getLength() != 0)
diff --git a/desktop/source/app/officeipcthread.cxx b/desktop/source/app/officeipcthread.cxx
index 95efc2c..e621e73 100644
--- desktop/source/app/officeipcthread.cxx
+++ desktop/source/app/officeipcthread.cxx
@@ -111,7 +111,7 @@ namespace desktop
 String GetURL_Impl( const String& rName );
 
 OfficeIPCThread*	OfficeIPCThread::pGlobalOfficeIPCThread = 0;
-namespace { struct Security : public rtl::Static<OSecurity, Security> {}; }
+namespace { struct Security : public rtl::Static<osl::Security, Security> {}; }
 ::osl::Mutex*		OfficeIPCThread::pOfficeIPCThreadMutex = 0;
 
 
@@ -375,7 +375,7 @@ OfficeIPCThread::Status OfficeIPCThread::EnableOfficeIPCThread()
 	PipeMode nPipeMode = PIPEMODE_DONTKNOW;
 	do
 	{
-		OSecurity &rSecurity = Security::get();
+        osl::Security &rSecurity = Security::get();
 		// Try to create pipe
 		if ( pThread->maPipe.create( pThread->maPipeIdent.getStr(), OPipe::TOption_Create, rSecurity ))
 		{
diff --git a/desktop/source/app/officeipcthread.hxx b/desktop/source/app/officeipcthread.hxx
index ec6ed08..65cc8b5 100644
--- desktop/source/app/officeipcthread.hxx
+++ desktop/source/app/officeipcthread.hxx
@@ -46,8 +46,8 @@
 #ifndef _VOS_PIPE_HXX_
 #include <vos/pipe.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 #ifndef _THREAD_HXX_
 #include <osl/thread.hxx>
diff --git a/desktop/source/app/userinstall.cxx b/desktop/source/app/userinstall.cxx
index fb952b1..54578f3 100644
--- desktop/source/app/userinstall.cxx
+++ desktop/source/app/userinstall.cxx
@@ -62,8 +62,8 @@
 #include <osl/diagnose.h>
 #endif
 
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 #ifndef _RTL_REF_HXX_
 #include <rtl/ref.hxx>
diff --git a/sal/typesconfig/core b/sal/typesconfig/core
index 97fe99e..e8e58ab 100644
Binary files a/sal/typesconfig/core and b/sal/typesconfig/core differ
diff --git a/sfx2/inc/pch/precompiled_sfx2.hxx b/sfx2/inc/pch/precompiled_sfx2.hxx
index 94de398..ed8e57f 100644
--- sfx2/inc/pch/precompiled_sfx2.hxx
+++ sfx2/inc/pch/precompiled_sfx2.hxx
@@ -673,7 +673,7 @@
 #include "osl/diagnose.h"
 #include "osl/module.hxx"
 #include "osl/mutex.hxx"
-#include "vos/security.hxx"
+#include "osl/security.hxx"
 #include "vos/socket.hxx"
 #include "osl/thread.hxx"
 //---MARKER---
diff --git a/sfx2/source/appl/appinit.cxx b/sfx2/source/appl/appinit.cxx
index 5f4455f..57231b3 100644
--- sfx2/source/appl/appinit.cxx
+++ sfx2/source/appl/appinit.cxx
@@ -90,8 +90,8 @@
 #ifndef _RTL_USTRBUF_HXX_
 #include <rtl/ustrbuf.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 #ifndef _UCBHELPER_CONFIGURATIONKEYS_HXX_
 #include <ucbhelper/configurationkeys.hxx>
diff --git a/sfx2/source/dialog/filedlghelper.cxx b/sfx2/source/dialog/filedlghelper.cxx
index 5b4f7cf..6974d12 100644
--- sfx2/source/dialog/filedlghelper.cxx
+++ sfx2/source/dialog/filedlghelper.cxx
@@ -144,8 +144,8 @@
 #ifndef _OSL_MUTEX_HXX_
 #include <osl/mutex.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 
 #ifndef _SV_CVTGRF_HXX
@@ -2568,7 +2568,7 @@ void FileDialogHelper::SetDisplayDirectory( const String& _rPath )
         if ( sFolder.getLength() == 0 )
         {
             // _rPath is not a valid path -> fallback to home directory
-            vos::OSecurity aSecurity;
+            osl::Security aSecurity;
             aSecurity.getHomeDir( sFolder );
         }
         mpImp->displayFolder( sFolder );
diff --git a/svtools/inc/pch/precompiled_svtools.hxx b/svtools/inc/pch/precompiled_svtools.hxx
index 8324445..f8e7c08 100644
--- svtools/inc/pch/precompiled_svtools.hxx
+++ svtools/inc/pch/precompiled_svtools.hxx
@@ -433,7 +433,7 @@
 #include <osl/mutex.hxx>
 #include <rtl/ref.hxx>
 #include <salhelper/simplereferenceobject.hxx>
-#include <vos/security.hxx>
+#include <osl/security.hxx>
 #include <osl/thread.hxx>
 #include <vos/timer.hxx>
 
diff --git a/svtools/source/dialogs/filedlg2.cxx b/svtools/source/dialogs/filedlg2.cxx
index d748d7b..002c782 100644
--- svtools/source/dialogs/filedlg2.cxx
+++ svtools/source/dialogs/filedlg2.cxx
@@ -61,8 +61,8 @@
 #include <vcl/msgbox.hxx>
 #endif
 
-#ifndef _VOS_SECURITY_HXX_
-#include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#include <osl/security.hxx>
 #endif
 
 #ifndef _COM_SUN_STAR_I18N_XCOLLATOR_HPP_
@@ -373,7 +373,7 @@ IMPL_LINK( ImpPathDialog, ClickHdl, Button*, pBtn )
 	if ( pBtn == pHomeBtn )
 	{
 		::rtl::OUString aHomeDir;
-		vos::OSecurity aSecurity;
+		osl::Security aSecurity;
 		if ( aSecurity.getHomeDir( aHomeDir ) )
 		{
 			DirEntry aFile ( aHomeDir );
diff --git a/vos/inc/vos/pipe.hxx b/vos/inc/vos/pipe.hxx
index 7a1f8ac..3067a42 100644
--- vos/inc/vos/pipe.hxx
+++ vos/inc/vos/pipe.hxx
@@ -55,8 +55,8 @@
 #ifndef _VOS_REFOBJ_HXX_
 #   include <vos/refobj.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#   include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#   include <osl/security.hxx>
 #endif
 #ifndef _RTL_USTRING_HXX_
 #   include <rtl/ustring.hxx>
@@ -123,7 +123,7 @@ public:
         @param Security
     */
     OPipe(const ::rtl::OUString& strName, TPipeOption Options,
-          const vos::OSecurity& rSecurity);
+          const osl::Security& rSecurity);
 
     /** Copy constructor.
     */
@@ -154,7 +154,7 @@ public:
         @return True if socket was successfully created.
     */
     sal_Bool SAL_CALL create(const ::rtl::OUString& strName, TPipeOption Options,
-                   const vos::OSecurity& rSecurity);
+                   const osl::Security& rSecurity);
 
     /** Assignment operator. If pipe was already created, the old one will
         be discarded.
diff --git a/vos/inc/vos/process.hxx b/vos/inc/vos/process.hxx
index d8db5b6..2440c5a 100644
--- vos/inc/vos/process.hxx
+++ vos/inc/vos/process.hxx
@@ -39,8 +39,8 @@
 #ifndef _RTL_USTRING_HXX_
 #   include <rtl/ustring.hxx>
 #endif
-#ifndef _VOS_SECURITY_HXX_
-#   include <vos/security.hxx>
+#ifndef _OSL_SECURITY_HXX_
+#   include <osl/security.hxx>
 #endif
 #ifndef _VOS_OBJECT_HXX_
 #   include <vos/object.hxx>
@@ -211,7 +211,7 @@ public:
         @see OStartupInfo::acceptIOResource
     */
     TProcessError SAL_CALL execute(TProcessOption Options,
-                          const OSecurity &Security,
+                          const osl::Security &Security,
                           const OArgumentList& aArgumentList = OArgumentList(),
                           const OEnvironment&  aEnvironment  = OEnvironment()
                          );
diff --git a/vos/inc/vos/security.hxx b/vos/inc/vos/security.hxx
deleted file mode 100644
index 8f41fd9..0000000
--- vos/inc/vos/security.hxx
+++ /dev/null
@@ -1,135 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: security.hxx,v $
- *
- *  $Revision: 1.6 $
- *
- *  last change: $Author: vg $ $Date: 2006/06/02 12:39:33 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-#ifndef _VOS_SECURITY_HXX_
-#define _VOS_SECURITY_HXX_
-
-#ifndef _VOS_OBJECT_HXX_
-#   include <vos/object.hxx>
-#endif
-#ifndef _RTL_USTRING_HXX_
-#   include <rtl/ustring.hxx>
-#endif
-#ifndef _OSL_SECURITY_H_
-#   include <osl/security.h>
-#endif
-
-namespace vos
-{
-
-/** capsulate security informations for one user.
-    A object of this class is used to execute a process with the rights an
-    security options of a scecified user.
-    @see OProcess::executeProcess
-    @author  Bernd Hofner
-    @version 1.0
-
-*/
-class OSecurity : public OObject
-{
-    VOS_DECLARE_CLASSINFO(VOS_NAMESPACE(OSecurity, vos));
-
-public:
-    /// constructor
-    OSecurity();
-    /// destructor
-    virtual ~OSecurity();
-    /** get the security information for one user.
-        The underlying operating system is asked for this information.
-        @param strName [in] denotes the name of the user
-        @param strPasswd [in] denotes the password of this user
-        @return True, if the specified user is known by the underlying operating system,
-        otherwise False
-    */
-    sal_Bool SAL_CALL logonUser(const ::rtl::OUString& strName,
-                                const ::rtl::OUString& strPasswd);
-    /** get the security information for one user.
-        This method will try to login the user at the denoted file server.
-        If a network resource named \\server\username exists and this resource
-        could be connected by this user, the methos will return true and getHomeDir
-        will return \\server\username.
-        @param strName [in] denotes the name of the user
-        @param strPasswd [in] denotes the password of this user
-        @return True, if the specified user is known by file server and the
-        could be connected, otherwise False
-    */
-    sal_Bool SAL_CALL logonUser(const ::rtl::OUString& strName,
-                                const ::rtl::OUString& strPasswd,
-                                const ::rtl::OUString& strFileServer);
-
-    /** get the ident of the logged in user.
-        @param strName [out] is the buffer which returns the name
-        @param max [in] is the size of this buffer
-        @return True, if any user is successfuly logged in, otherwise False
-    */
-    sal_Bool SAL_CALL getUserIdent(::rtl::OUString& strIdent) const;
-
-    /** get the name of the logged in user.
-        @param strName [out] is the buffer which returns the name
-        @param max [in] is the size of this buffer
-        @return True, if any user is successfuly logged in, otherwise False
-    */
-    sal_Bool SAL_CALL getUserName(::rtl::OUString& strName) const;
-
-    /** get the home directory of the logged in user.
-        @param strDirectory [out] is the buffer which returns the directory name
-        @param max [in] is the size of this buffer
-        @return True, if any user is successfuly logged in, otherwise False
-    */
-    sal_Bool SAL_CALL getHomeDir(::rtl::OUString& strDirectory) const;
-
-    /** get the directory for configuration data of the logged in user.
-        @param strDirectory [out] is the buffer which returns the directory name
-        @param max [in] is the size of this buffer
-        @return True, if any user is successfuly logged in, otherwise False
-    */
-    sal_Bool SAL_CALL getConfigDir(::rtl::OUString& strDirectory) const;
-
-    /** Query if the user who is logged inhas administrator rigths.
-        @return True, if the user has administrator rights, otherwise false.
-    */
-    sal_Bool SAL_CALL isAdministrator() const;
-
-    virtual SAL_CALL operator oslSecurity() const;
-
-protected:
-
-    oslSecurity m_oslSecurity;
-};
-
-}
-
-#endif  // _VOS_SECURITY_HXX_
-
diff --git a/vos/source/makefile.mk b/vos/source/makefile.mk
index 61584e7..0d2ae51 100644
--- vos/source/makefile.mk
+++ vos/source/makefile.mk
@@ -53,7 +53,6 @@ SLOFILES= \
 				$(SLO)$/stream.obj      \
 				$(SLO)$/timer.obj       \
 				$(SLO)$/process.obj     \
-				$(SLO)$/security.obj    \
 				$(SLO)$/pipe.obj
 
 
@@ -66,7 +65,6 @@ OBJFILES= \
 				$(OBJ)$/stream.obj      \
 				$(OBJ)$/timer.obj       \
 				$(OBJ)$/process.obj     \
-				$(OBJ)$/security.obj    \
 				$(OBJ)$/pipe.obj
 
 .ENDIF
diff --git a/vos/source/pipe.cxx b/vos/source/pipe.cxx
index d280fae..1669b80 100644
--- vos/source/pipe.cxx
+++ vos/source/pipe.cxx
@@ -76,12 +76,12 @@ OPipe::OPipe( const rtl::OUString& strName, TPipeOption Options)
 
 OPipe::OPipe( const rtl::OUString& strName,
 			  TPipeOption Options,
-			  const OSecurity& rSecurity)
+			  const osl::Security& rSecurity)
 {
 	m_pPipeRef= 
 		new PipeRef(osl_createPipe(strName.pData, 
 								   (oslPipeOptions)Options, 
-								   (oslSecurity)rSecurity));
+								   rSecurity.getHandle()));
 	
 	OSL_POSTCOND(m_pPipeRef != 0, "OPipe(): new failed.\n");
 	OSL_POSTCOND((*m_pPipeRef)(), "OPipe(): creation of pipe failed!\n");
@@ -146,7 +146,7 @@ sal_Bool OPipe::create( const rtl::OUString& strName, TPipeOption Options )
 /*****************************************************************************/
 sal_Bool OPipe::create( const rtl::OUString& strName,
 						TPipeOption Options,
-						const vos::OSecurity& rSecurity ) 
+						const osl::Security& rSecurity ) 
 {
 	// if this was a valid pipe, decrease reference
 	if ((m_pPipeRef) && (m_pPipeRef->release() == 0))
@@ -159,7 +159,7 @@ sal_Bool OPipe::create( const rtl::OUString& strName,
 	m_pPipeRef= 
 		new PipeRef(osl_createPipe(strName.pData, 
 								   (oslPipeOptions)Options, 
-								   (oslSecurity)rSecurity));
+								   rSecurity.getHandle()));
 
 	OSL_POSTCOND(m_pPipeRef != 0, "OPipe(): new failed.\n");
 
diff --git a/vos/source/process.cxx b/vos/source/process.cxx
index ac6a509..3872bbc 100644
--- vos/source/process.cxx
+++ vos/source/process.cxx
@@ -296,7 +296,7 @@ OProcess::TProcessError OProcess::execute(TProcessOption Options,
 
 
 OProcess::TProcessError OProcess::execute( TProcessOption Options,
-										   const OSecurity &Security,
+										   const osl::Security &Security,
 										   const OArgumentList& aArgumentList,
 										   const OEnvironment&  aEnvironment )
 {
@@ -304,7 +304,7 @@ OProcess::TProcessError OProcess::execute( TProcessOption Options,
 								              aArgumentList.m_aVec,
 											  aArgumentList.n_Args,
 											  Options, 
-								              Security,
+								              Security.getHandle(),
 								              m_strDirectory.pData,
 								              aEnvironment.m_aVec,
 											  aEnvironment.n_Vars,
diff --git a/vos/source/security.cxx b/vos/source/security.cxx
deleted file mode 100644
index cb33330..0000000
--- vos/source/security.cxx
+++ /dev/null
@@ -1,124 +0,0 @@
-/*************************************************************************
- *
- *  OpenOffice.org - a multi-platform office productivity suite
- *
- *  $RCSfile: security.cxx,v $
- *
- *  $Revision: 1.3 $
- *
- *  last change: $Author: vg $ $Date: 2006/06/02 12:43:47 $
- *
- *  The Contents of this file are made available subject to
- *  the terms of GNU Lesser General Public License Version 2.1.
- *
- *
- *    GNU Lesser General Public License Version 2.1
- *    =============================================
- *    Copyright 2005 by Sun Microsystems, Inc.
- *    901 San Antonio Road, Palo Alto, CA 94303, USA
- *
- *    This library is free software; you can redistribute it and/or
- *    modify it under the terms of the GNU Lesser General Public
- *    License version 2.1, as published by the Free Software Foundation.
- *
- *    This library is distributed in the hope that it will be useful,
- *    but WITHOUT ANY WARRANTY; without even the implied warranty of
- *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *    Lesser General Public License for more details.
- *
- *    You should have received a copy of the GNU Lesser General Public
- *    License along with this library; if not, write to the Free Software
- *    Foundation, Inc., 59 Temple Place, Suite 330, Boston,
- *    MA  02111-1307  USA
- *
- ************************************************************************/
-
-
-#include <vos/security.hxx>
-#include <osl/diagnose.h>
-
-using namespace vos;
-
-/////////////////////////////////////////////////////////////////////////////
-// Object super class
-
-VOS_IMPLEMENT_CLASSINFO(VOS_CLASSNAME(OSecurity, vos), VOS_NAMESPACE(OSecurity, vos), VOS_NAMESPACE(OObject, vos), 0);
-
-OSecurity::OSecurity()
-{
-	m_oslSecurity = osl_getCurrentSecurity();
-}
-
-OSecurity::~OSecurity()
-{
-    osl_freeSecurityHandle(m_oslSecurity);
-}
-
-sal_Bool OSecurity::logonUser(const rtl::OUString& strName,
-							  const rtl::OUString& strPasswd)
-{
-    osl_freeSecurityHandle(m_oslSecurity);
-	
-    m_oslSecurity = NULL;
-    
-	return (osl_loginUser( strName.pData, strPasswd.pData, &m_oslSecurity) 
-	        == osl_Security_E_None);
-}
-
-
-sal_Bool OSecurity::logonUser( const rtl::OUString& strName,
-							   const rtl::OUString& strPasswd,
-							   const rtl::OUString& strFileServer )
-{
-    osl_freeSecurityHandle(m_oslSecurity);
-	
-    m_oslSecurity = NULL;
-	
-    return (osl_loginUserOnFileServer(strName.pData, strPasswd.pData, strFileServer.pData, &m_oslSecurity) 
-	        == osl_Security_E_None);
-}
-
-
-sal_Bool OSecurity::getUserIdent( rtl::OUString& strIdent) const
-{
-	OSL_ASSERT(m_oslSecurity);
-	
-    return osl_getUserIdent( m_oslSecurity, &strIdent.pData );
-}
-
-
-sal_Bool OSecurity::getUserName( rtl::OUString& strName ) const
-{
-	OSL_ASSERT(m_oslSecurity);
-
-    return osl_getUserName( m_oslSecurity, &strName.pData );
-}
-
-
-sal_Bool OSecurity::getHomeDir( rtl::OUString& strDirectory) const
-{
-	OSL_ASSERT(m_oslSecurity);
-	
-    return osl_getHomeDir(m_oslSecurity, &strDirectory.pData );
-}
-
-
-sal_Bool OSecurity::getConfigDir( rtl::OUString& strDirectory ) const
-{
-	OSL_ASSERT(m_oslSecurity);
-	
-    return osl_getConfigDir( m_oslSecurity, &strDirectory.pData );
-}
-
-sal_Bool OSecurity::isAdministrator() const
-{
-	OSL_ASSERT(m_oslSecurity);
-
-    return osl_isAdministrator(m_oslSecurity);
-}
-
-OSecurity::operator oslSecurity() const
-{
-    return m_oslSecurity;
-}
-
