--- extensions/source/propctrlr/browserlistbox.cxx.old	2006-09-16 15:13:11.000000000 +0200
+++ extensions/source/propctrlr/browserlistbox.cxx	2007-01-22 13:27:18.000000000 +0100
@@ -790,7 +790,7 @@
 		sal_Int32 nDelta = _nNewThumbPos - m_aVScroll.GetThumbPos();
 		// adjust the scrollbar
 		m_aVScroll.SetThumbPos(_nNewThumbPos);
-		sal_Int32 nThumbPos = nThumbPos = _nNewThumbPos;
+		sal_Int32 nThumbPos = _nNewThumbPos;
 
 		m_nYOffset = -m_aVScroll.GetThumbPos() * m_nRowHeight;
 
--- extensions/source/scanner/sane.cxx.old	2007-01-22 13:28:10.000000000 +0100
+++ extensions/source/scanner/sane.cxx	2007-01-22 13:30:02.000000000 +0100
@@ -489,7 +489,7 @@
 	{
 		SANE_Word nSetTo =
 			mppOptions[n]->type == SANE_TYPE_INT ?
-			nSetTo = (SANE_Word)fSet : nSetTo = SANE_FIX( fSet );
+			(SANE_Word)fSet : SANE_FIX( fSet );
 
 		nStatus = ControlOption( n, SANE_ACTION_SET_VALUE, &nSetTo );
 		if( nStatus == SANE_STATUS_GOOD )
