--- extensions/source/plugin/unx/npwrap.cxx.old	2005-03-29 15:06:21.000000000 +0200
+++ extensions/source/plugin/unx/npwrap.cxx	2005-04-21 16:03:40.000000000 +0200
@@ -148,7 +148,7 @@
   		shell,
         XtNwidth, 200,
         XtNheight, 200,
-  		NULL );
+  		(char *)NULL );
     XtRealizeWidget( shell );
 
     medDebug( 1, "Reparenting new widget %x to %x\n", XtWindow( newWidget ), aParentWindow );
@@ -176,7 +176,7 @@
                             XtNwidth, 200,
                             XtNheight, 200,
                             XtNoverrideRedirect, True,
-							NULL );
+							(char *)NULL );
 	*pShellReturn = newShell;
 
 	char pText[1024];
@@ -319,7 +319,7 @@
         NULL,               /* for missing app-defaults file */
         topLevelShellWidgetClass,
         XtNoverrideRedirect, True,
-        NULL);              /* terminate varargs list */
+        (char *)NULL);      /* terminate varargs list */
 	pAppDisplay = XtDisplay( topLevel );
 
     XtAppAddInput( app_context,
