--- binfilter/bf_svx/source/svdraw/svx_svdobj.cxx.old	2007-07-17 13:44:37.000000000 +0200
+++ binfilter/bf_svx/source/svdraw/svx_svdobj.cxx	2007-07-26 11:45:23.000000000 +0200
@@ -2889,7 +2889,9 @@ class ImpSkeleton;
 /*N*/ 
 /*N*/ //			if(pInterface)
 				SfxObjectFactory& rFac = pObjectShell->GetFactory();
-				if ( rFac.GetShortName() == "sdraw" || rFac.GetShortName() == "simpress" )
+				if ( rFac.GetShortName() &&
+					( strcmp(rFac.GetShortName(), "sdraw") == 0 ||
+					  strcmp(rFac.GetShortName(), "simpress") == 0 ) )
 /*N*/ 			{
 /*N*/ 				//sal_uInt16 nInterfaceID = pInterface->GetInterfaceId();
 /*N*/ 
