--- binfilter/bf_sc/source/core/data/sc_document.cxx.old	2005-12-07 20:24:38.000000000 +0100
+++ binfilter/bf_sc/source/core/data/sc_document.cxx	2005-12-09 16:24:44.000000000 +0100
@@ -1740,7 +1740,9 @@
 
 /*N*/ void ScDocument::GetClipArea(USHORT& nClipX, USHORT& nClipY, BOOL bIncludeFiltered)
 /*N*/ {
-DBG_BF_ASSERT(0, "STRIP"); //STRIP001 //STRIP001 	if (bIsClip)
+DBG_BF_ASSERT(0, "STRIP"); //STRIP001 
+nClipX = 0; nClipY = 0; // hack to avoid warning: ‘’ is used uninitialized; added to fix consequences of //STRIP001
+//STRIP001 	if (bIsClip)
 //STRIP001 	{
 //STRIP001 		nClipX = aClipRange.aEnd.Col() - aClipRange.aStart.Col();
 //STRIP001 
--- binfilter/bf_sc/source/ui/unoobj/sc_viewuno.cxx.old	2005-12-09 12:52:08.000000000 +0100
+++ binfilter/bf_sc/source/ui/unoobj/sc_viewuno.cxx	2005-12-09 13:03:13.000000000 +0100
@@ -1144,7 +1144,7 @@
         ScDBFunc* pView = pViewSh->GetViewData()->GetView();
         if (pView)
         {
-            SvxZoomType eZoomType;
+            SvxZoomType eZoomType = SVX_ZOOM_PERCENT;
             switch (aZoomType)
             {
             case view::DocumentZoomType::BY_VALUE:
--- binfilter/bf_sw/source/core/undo/sw_undobj.cxx.old	2005-09-09 04:43:57.000000000 +0200
+++ binfilter/bf_sw/source/core/undo/sw_undobj.cxx	2005-12-08 15:24:52.000000000 +0100
@@ -755,7 +755,7 @@
 /*N*/ 
 /*N*/ 	// Positionen als SwIndex merken, damit im DTOR dieser Bereich
 /*N*/ 	// entfernt werden kann !!
-/*N*/ 	ULONG nEnd;
+/*N*/ 	ULONG nEnd = 0;
 /*N*/ 	pMvStt = new SwNodeIndex( rRange.aStart );
 /*N*/ 	MoveToUndoNds( aPam, pMvStt, 0, &nEnd, 0 );
 /*N*/ 	nMvLen = nEnd - pMvStt->GetIndex() + 1;
