--- sd/source/ui/animations/CustomAnimationDialog.cxx.old	2007-01-15 18:18:07.000000000 +0100
+++ sd/source/ui/animations/CustomAnimationDialog.cxx	2007-01-17 21:24:17.000000000 +0100
@@ -2488,7 +2488,7 @@
 
 		if( nPos > 1 )
 		{
-			double fTextGroupingAuto = maCBXGroupAuto.IsChecked() ? fTextGroupingAuto = maMFGroupAuto.GetValue() / 10.0 : -1.0;
+			double fTextGroupingAuto = maCBXGroupAuto.IsChecked() ? maMFGroupAuto.GetValue() / 10.0 : -1.0;
 			double fOldTextGroupingAuto = -2.0;
 
 			if(mpSet->getPropertyState( nHandleTextGroupingAuto ) != STLPropertyState_AMBIGUOUS)
--- sd/source/ui/func/fupage.cxx.old	2006-09-16 20:53:19.000000000 +0200
+++ sd/source/ui/func/fupage.cxx	2007-01-22 13:34:56.000000000 +0100
@@ -470,7 +470,7 @@
 		pDoc->SetPageNumType(((const SvxPageItem*) pPoolItem)->GetNumType());
 
 		eOrientation = (((const SvxPageItem*) pPoolItem)->IsLandscape() == ORIENTATION_LANDSCAPE) ? 
-			ORIENTATION_LANDSCAPE : eOrientation = ORIENTATION_PORTRAIT;
+			ORIENTATION_LANDSCAPE : ORIENTATION_PORTRAIT;
 
 		if( mpPage->GetOrientation() != eOrientation )
 			bSetPageSizeAndBorder = TRUE;
