--- filter/source/config/cache/filtercache.cxx.old	2005-09-08 23:28:41.000000000 +0200
+++ filter/source/config/cache/filtercache.cxx	2005-11-28 19:17:53.000000000 +0100
@@ -849,10 +849,13 @@
         case E_CONTENTHANDLER : return m_lContentHandlers; break;
         case E_DETECTSERVICE  : return m_lDetectServices ; break;
 
-        default : throw css::uno::Exception(::rtl::OUString::createFromAscii("unknown sub container requested."),
-                                            css::uno::Reference< css::uno::XInterface >()                      );
+        default : break;
     }
 
+    // throw has been moved from the default section of the switch to
+    // avoid warning: control reaches end of non-void function
+    throw css::uno::Exception(::rtl::OUString::createFromAscii("unknown sub container requested."),
+                                            css::uno::Reference< css::uno::XInterface >()                      );
     // <- SAFE ----------------------------------
 }
 
