--- connectivity/source/drivers/kab/KResultSet.cxx.old	2005-09-22 14:15:29.000000000 +0200
+++ connectivity/source/drivers/kab/KResultSet.cxx	2005-09-22 14:22:01.000000000 +0200
@@ -185,6 +185,8 @@
 	::dbtools::throwGenericSQLException(
 		::rtl::OUString::createFromAscii("Invalid column name: ") + columnName,
 		NULL);
+	// just to avoid warning: control reaches end of non-void function
+	return 1;
 }
 // -------------------------------------------------------------------------
 Reference< XInputStream > SAL_CALL KabResultSet::getBinaryStream(sal_Int32 columnIndex) throw(SQLException, RuntimeException)
--- connectivity/source/drivers/kab/KResultSetMetaData.cxx.old	2005-09-22 14:11:52.000000000 +0200
+++ connectivity/source/drivers/kab/KResultSetMetaData.cxx	2005-09-22 14:21:55.000000000 +0200
@@ -103,6 +103,8 @@
 	::dbtools::throwGenericSQLException(
 		::rtl::OUString::createFromAscii("Invalid column name: ") + columnName,
 		NULL);
+	// just to avoid warning: control reaches end of non-void function
+	return 0;
 }
 // -------------------------------------------------------------------------
 void KabResultSetMetaData::setKabFields(const ::vos::ORef<connectivity::OSQLColumns> &xColumns) throw(SQLException)
--- connectivity/source/drivers/kab/KStatement.cxx.old	2005-09-20 18:10:50.000000000 +0200
+++ connectivity/source/drivers/kab/KStatement.cxx	2005-09-22 14:20:35.000000000 +0200
@@ -237,6 +237,8 @@
 	::dbtools::throwGenericSQLException(
 		::rtl::OUString::createFromAscii("Syntax error or keyword not recognized."),
 		NULL);
+	// just to avoid warning: control reaches end of non-void function
+	return NULL;
 }
 //------------------------------------------------------------------------------
 sal_Bool KabCommonStatement::isTableKnown(KabResultSet *pResult) const
