Index: tools/prj/build.lst
===================================================================
RCS file: /cvs/util/tools/prj/build.lst,v
retrieving revision 1.13
diff -u -p -u -r1.13 build.lst
--- tools/prj/build.lst	24 Apr 2003 13:27:35 -0000	1.13
+++ tools/prj/build.lst	6 Apr 2004 17:44:36 -0000
@@ -11,7 +11,8 @@ tl	tools\prj							usr7	-	all	tl_deliver
 tl	tools\prj							usr7	-	all	tl_deliver NULL
 tl	tools\prj							usr42	-	all	tl_zn NULL
 tl	tools\source\testtoolloader			nmake	-	all	tl_ttloader NULL
-tl	tools\source\generic				nmake	-	all	tl_gen tl_fsys NULL
+tl	tools\source\generic\libart_lgpl			nmake	-	all	tl_libart NULL
+tl	tools\source\generic				nmake	-	all	tl_gen tl_libart tl_fsys tl_sol.u NULL
 tl	tools\source\memtools				nmake	-	all	tl_mem tl_fsys NULL
 tl	tools\source\memmgr					nmake	-	all	tl_mgr tl_fsys NULL
 tl	tools\source\debug					nmake	-	all	tl_deb tl_fsys NULL


Index: tools/prj/d.lst
===================================================================
RCS file: /cvs/util/tools/prj/d.lst,v
retrieving revision 1.24.56.1
diff -u -p -u -r1.24.56.1 d.lst
--- tools/prj/d.lst	28 Jan 2004 10:54:29 -0000	1.24.56.1
+++ tools/prj/d.lst	6 Apr 2004 17:44:36 -0000
@@ -29,6 +29,8 @@ mkdir: %_DEST%\inc%_EXT%\testshl
 ..\%__SRC%\slb\btstrpsh.lib %_DEST%\lib%_EXT%\btstrpsh.lib
 ..\%__SRC%\lib\tstutil.lib %_DEST%\lib%_EXT%\tstutil.lib
 ..\%__SRC%\bin\testshl.exe %_DEST%\bin%_EXT%\testshl.exe
+..\%__SRC%\bin\libart.dll %_DEST%\bin%_EXT%\libart.dll
+..\%__SRC%\lib\libart.lib %_DEST%\lib%_EXT%\libart.lib
 
 
 hedabu: ..\%__SRC%\inc\svconf.h %_DEST%\inc%_EXT%\tools\svconf.h
Index: tools/source/generic/makefile.mk
===================================================================
RCS file: /cvs/util/tools/source/generic/makefile.mk,v
retrieving revision 1.7
diff -u -p -u -r1.7 makefile.mk
--- tools/source/generic/makefile.mk	16 Jul 2003 17:15:14 -0000	1.7
+++ tools/source/generic/makefile.mk	6 Apr 2004 17:44:36 -0000
@@ -71,13 +71,8 @@ TARGET=gen
 .INCLUDE : settings.mk
 .INCLUDE : sv.mk
 
-.IF "$(WITH_GPC)"!="NO"
-CDEFS+=-DHAVE_GPC_H
-.ENDIF
-.IF "$(WITH_LIBART)"!="NO"
 CDEFS+=-DHAVE_LIBART_H
-CFLAGS+=$(LIBART_CFLAGS)
-.ENDIF
+INCPRE+=libart_lgpl
 
 # --- Files --------------------------------------------------------
 

Index: config_office/configure.in
===================================================================
RCS file: /cvs/tools/config_office/configure.in,v
retrieving revision 1.55.6.7
diff -u -p -u -r1.55.6.7 configure.in
--- config_office/configure.in	19 Feb 2004 10:59:25 -0000	1.55.6.7
+++ config_office/configure.in	6 Apr 2004 17:47:13 -0000
@@ -2457,7 +2457,6 @@ WITH_GPC=NO
 WITH_LIBART=NO
 
 if test -n "$enable_libart"; then
-  PKG_CHECK_MODULES( LIBART, libart-2.0 >= 2.3.13 )
   WITH_LIBART=YES
 	
 elif test "$with_gpc" != "no" ; then


Index: tools/util/makefile.mk
===================================================================
RCS file: /cvs/util/tools/util/makefile.mk,v
retrieving revision 1.11.12.1
diff -u -p -u -r1.11.12.1 makefile.mk
--- tools/util/makefile.mk	15 Aug 2003 11:30:50 -0000	1.11.12.1
+++ tools/util/makefile.mk	6 Apr 2004 19:57:47 -0000
@@ -142,7 +142,8 @@ LIB7FILES=  $(LB)$/gen.lib          \
             $(LB)$/ref.lib          \
             $(LB)$/rc.lib           \
             $(LB)$/inet.lib         \
-            $(LB)$/debug.lib
+            $(LB)$/debug.lib	    \
+	    $(LB)$/libart.lib
 
 
 LIB7FILES+= $(LB)$/dll.lib
@@ -199,12 +200,7 @@ SHL1LIBS=       $(LIB1TARGET)
 SHL1DEF=        $(MISC)$/$(SHL1TARGET).def
 SHL1IMPLIB=     itools
 SHL1STDLIBS+=   $(SALLIB) $(VOSLIB) $(BASEGFXLIB)
-
-.IF "$(WITH_LIBART)"=="YES"
-SHL1STDLIBS+=   $(LIBART_LIBS)
-.ELSE
-SHL1STDLIBS+=   $(GPC3RDLIB)
-.ENDIF
+SHL1STDLIBS+=   $(SLB)$/libart.lib
 
 
 .IF "$(GUI)"=="WNT"
