--- shell/source/win32/simplemail/senddoc.cxx
+++ shell/source/win32/simplemail/senddoc.cxx
@@ -99,6 +99,7 @@
     StringList_t gCc;
     StringList_t gBcc;
     StringList_t gAttachments;
+	StringList_t gAttachFilenames;
     int gMapiFlags = 0;
 }
 
@@ -172,7 +173,12 @@
     {
         MapiFileDesc mfd;
         ZeroMemory(&mfd, sizeof(mfd));
+
         mfd.lpszPathName = const_cast<char*>(iter->c_str());    
+		rtl::OString aFullPath( mfd.lpszPathName );
+		sal_Int32 nLastSlash = aFullPath.lastIndexOf('\\');
+		gAttachFilenames.push_back( (const sal_Char *) aFullPath );
+		mfd.lpszFileName = const_cast<char*>(gAttachFilenames.back().c_str());    
         mfd.nPosition = -1;
         pMapiAttachmentList->push_back(mfd);        
     }            
