--- solenv/inc/tg_ext.mk	2005-01-25 11:30:22.781250000 +0530
+++ solenv/inc/tg_ext.mk	2005-01-25 11:30:27.750000000 +0530
@@ -180,6 +180,9 @@
 	@+-$(MKDIR) $(PACKAGE_DIR)
 	+cd $(PACKAGE_DIR) && ( $(shell +$(TYPE) $(PRJ)$/$(ROUT)$/misc$/$(TARFILE_NAME).unpack)) && $(TOUCH) $(UNTAR_FLAG_FILE)
 	@+echo make writeable...
+.IF "$(TARFILE_NAME)"=="curl-7.9.8"
+	@+/usr/bin/dos2unix.exe $(PACKAGE_DIR)/curl-7.9.8/lib/Makefile.vc6
+.ENDIF
 .IF "$(GUI)"=="UNX" || "$(USE_SHELL)"!="4nt"
 	@+cd $(PACKAGE_DIR) && chmod -R +rw $(TARFILE_ROOTDIR) && $(TOUCH) $(UNTAR_FLAG_FILE)
 	@+cd $(PACKAGE_DIR) && find $(TARFILE_ROOTDIR) -type d -exec chmod a+x {{}} \;
--- tools/util/makefile.mk	2005-02-09 09:24:25.546875000 +0530
+++ tools/util/makefile.mk	2005-02-09 11:46:11.296875000 +0530
@@ -202,7 +202,7 @@
 
 .IF "$(WITH_LIBART)"=="YES"
 SHL1STDLIBS+=   $(LIBART_LIBS)
-.ELSE
+.ELIF "$WITH_GPC)"=="YES"
 SHL1STDLIBS+=   $(GPC3RDLIB)
 .ENDIF
 
--- svx/source/intro/makefile.mk	2005-09-28 18:44:58.000000000 +0300
+++ svx/source/intro/makefile.mk.new	2005-11-14 19:37:36.046875000 +0200
@@ -48,7 +48,7 @@
 # setting the compiled by $user$ using $license$ string in the about box
 
 .IF "$(BUILD_SPECIAL)"==""
-RSCDEFS+=-DOOO_VENDOR="$(USER)$(USERNAME)"  -DOOO_LICENSE="LGPL"
+RSCDEFS+=-DOOO_VENDOR="$(USERNAME)"  -DOOO_LICENSE="LGPL"
 .ELSE
 RSCDEFS+=-DBUILD_SPECIAL=True -DOOO_LICENSE="LGPL"
 .ENDIF
