--- solenv/inc/tg_ext.mk	2005-01-25 11:30:22.781250000 +0530
+++ solenv/inc/tg_ext.mk	2005-01-25 11:30:27.750000000 +0530
@@ -180,6 +180,9 @@
 	@+-$(MKDIR) $(PACKAGE_DIR)
 	+cd $(PACKAGE_DIR) && ( $(shell +$(TYPE) $(PRJ)$/$(ROUT)$/misc$/$(TARFILE_NAME).unpack)) && $(TOUCH) $(UNTAR_FLAG_FILE)
 	@+echo make writeable...
+.IF "$(TARFILE_NAME)"=="curl-7.9.8"
+	@+/usr/bin/dos2unix.exe $(PACKAGE_DIR)/curl-7.9.8/lib/Makefile.vc6
+.ENDIF
 .IF "$(GUI)"=="UNX" || "$(USE_SHELL)"!="4nt"
 	@+cd $(PACKAGE_DIR) && chmod -R +rw $(TARFILE_ROOTDIR) && $(TOUCH) $(UNTAR_FLAG_FILE)
 	@+cd $(PACKAGE_DIR) && find $(TARFILE_ROOTDIR) -type d -exec chmod a+x {{}} \;
--- tools/util/makefile.mk	2005-02-09 09:24:25.546875000 +0530
+++ tools/util/makefile.mk	2005-02-09 11:46:11.296875000 +0530
@@ -202,7 +202,7 @@
 
 .IF "$(WITH_LIBART)"=="YES"
 SHL1STDLIBS+=   $(LIBART_LIBS)
-.ELSE
+.ELIF "$WITH_GPC)"=="YES"
 SHL1STDLIBS+=   $(GPC3RDLIB)
 .ENDIF
 
--- setup_native/source/win32/msi-encodinglist.txt	Mon Jan 16 17:53:13 2006
+++ setup_native/source/win32/msi-encodinglist.txt	Mon Jan 16 17:53:18 2006
@@ -94,3 +94,6 @@
 ve       0  1075   # Venda
 ga       0  2108   # Irish
 mk    1251  1071   # Macedonian
+gl    1252  1110   # Galician
+kn       0  1099   # Kannada
+ms       0  1086   # Malay (Malaysian)
--- solenv/inc/postset.mk	Mon Jan 16 18:08:43 2006
+++ solenv/inc/postset.mk	Mon Jan 16 18:09:25 2006
@@ -39,6 +39,6 @@
 defaultlangiso=en-US
 
 # Complete list of all supported ISO codes
-completelangiso=af ar be-BY bg br bn bn-BD bn-IN bs ca cs cy da de el en-GB en-US en-ZA eo es et eu fa fi fr ga gl gu-IN he hi-IN hr hu it ja km kn-IN ko lo lt lv mk ms ne nb nl nn nr ns pa-IN pl pt pt-BR ru rw sk sl sh-YU sr-CS ss st sv sw sw-TZ sx ta-IN th tn tr ts ve vi xh zh-CN zh-TW zu
+completelangiso=af ar be-BY bg br bn bn-BD bn-IN bs ca cs cy da de el en-GB en-US en-ZA es et eu fa fi fr ga gl gu-IN he hi-IN hr hu it ja km ko lo lt lv mk ms ne nb nl nn nr ns pa-IN pl pt pt-BR ru rw sk sl sh-YU sr-CS ss st sv sw sw-TZ ta-IN th tn tr ts ve vi xh zh-CN zh-TW zu
 
 alllangiso=$(strip $(defaultlangiso) $(subst,$(defaultlangiso), $(uniq $(subst,ALL,$(completelangiso) $(WITH_LANG)))))
