--- instsetoo_native/util/openoffice.lst.orig	2007-08-10 19:12:13.000000000 +0200
+++ instsetoo_native/util/openoffice.lst	2007-11-07 20:04:59.955928500 +0100
@@ -47,6 +47,7 @@
 			DOWNLOADBANNER	ooobanner.bmp
 			DOWNLOADBITMAP	ooobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
+			RELATIVE_PATHES_IN_DDF 1
 		}
 		active			1
 		compression		5
@@ -98,6 +99,7 @@
 			DOWNLOADBANNER	ooobanner.bmp
 			DOWNLOADBITMAP	ooobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
+			RELATIVE_PATHES_IN_DDF 1
 		}
 		active			1
 		compression		5
@@ -149,6 +151,7 @@
 			DOWNLOADBITMAP	ooobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			LOCALUSERDIR $ORIGIN/..
+			RELATIVE_PATHES_IN_DDF 1
 		}
 		active			1
 		compression		5
@@ -178,6 +181,7 @@
             DOWNLOADBITMAP urebitmap.bmp
             DOWNLOADSETUPICO ooosetup.ico
             DONTUSESTARTMENUFOLDER 1
+            RELATIVE_PATHES_IN_DDF 1
         }
         active 1
         compression 5
@@ -257,6 +261,7 @@
 			DOWNLOADBANNER	brobanner.bmp
 			DOWNLOADBITMAP	brobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
+			RELATIVE_PATHES_IN_DDF 1
 		}
 		active			1
 		compression		5
@@ -308,6 +313,7 @@
 			DOWNLOADBANNER	brobanner.bmp
 			DOWNLOADBITMAP	brobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
+			RELATIVE_PATHES_IN_DDF 1
 		}
 		active			1
 		compression		5
@@ -358,6 +364,7 @@
 			DOWNLOADBITMAP	brobitmap.bmp
 			DOWNLOADSETUPICO	ooosetup.ico
 			LOCALUSERDIR $ORIGIN/..
+			RELATIVE_PATHES_IN_DDF 1			
 		}
 		active			1
 		compression		5
