--- config_office/configure.in	2005-01-20 14:50:01.328125000 +0530
+++ config_office/configure.in	2005-01-20 14:55:24.046875000 +0530
@@ -929,6 +929,16 @@ dnl  Check which Microsoft C/C++ or MinG
 dnl ===================================================================
 if test "$_os" = "WINNT"; then
 	if test "$WITH_MINGWIN" != "yes"; then
+		AC_MSG_CHECKING([for friendly registry keys])
+		# VS.Net 2003, VS.Net 2002, VC6
+		if test -z "$with_cl_home"; then
+			vctest=`./oowintool --msvc-productdir`;
+			if test -x "$vctest/bin/cl.exe"; then
+				with_cl_home=$vctest;
+			fi
+		fi
+		AC_MSG_RESULT([done])
+
 		AC_MSG_CHECKING([the Microsoft C/C++ Compiler])
 		dnl ===========================================================
 		dnl  Check for mspdb??.dll
@@ -1008,7 +1018,7 @@ dnl      The following find microsoft, m
 							}
 							}"`
 			AC_MSG_RESULT([found Compiler version $CCNUMVER.])
-			if test "$CCNUMVER" -ge "001300100000"; then
+			if test "$CCNUMVER" -ge "001300102240"; then
 				COMEX=10
 			else
 				if test "$CCNUMVER" -ge "001300000000"; then
@@ -1270,6 +1286,11 @@ if test "$COMEX" = "8" -o "$COMEX" = "10
 	if test -x "$with_csc_path/csc.exe"; then
 		CSC_PATH="$with_csc_path"
 	fi
+	csctest=`./oowintool --csc-compilerdir`;
+	if test -x "$csctest/csc.exe"; then
+		CSC_PATH="$csctest"
+	fi
+
 	if test ! -x "$CSC_PATH/csc.exe"; then
 		AC_MSG_ERROR([csc.exe not found. Make sure it's in the path or use --with-csc-path])
 	fi
@@ -1803,6 +1824,12 @@ if test "$_os" = "Linux" && echo $build_
 fi
 AC_SUBST(LANG)
 
+if test -z "$with_jdk_home"; then
+    _jdk_home=`./oowintool --jdk-home`
2
+    if test -d "$_jdk_home"; then
+        with_jdk_home="$_jdk_home"
+    fi
+fi
 
 dnl ===================================================================
 dnl Checks for java
@@ -2206,7 +2216,7 @@ if test \( "$_os" = "WINNT" \) ; then
 	AC_MSG_CHECKING([for PSDK files])
   	if test -z "$with_psdk_home"; then
 		# This first line will detect a February 2003 Microsoft Platform SDK
-		PSDK_HOME=`cat "/proc/registry/HKEY_LOCAL_MACHINE/SOFTWARE/Microsoft/MicrosoftSDK/Directories/Install Dir" 2> /dev/null`
+		PSDK_HOME=`./oowintool --psdk-home`
         # But there might be also an April 2005 PSDK, unfortunately MS changed
         # the registry entry. (we prefer the old version!?)
         if test -z "$PSDK_HOME"; then
@@ -2454,6 +2464,13 @@ may have to search Microsoft's website.)
 fi
 
 dnl ===================================================================
+dnl Windows builds - attempt to auto-copy required dlls into external/
+dnl ===================================================================
+if test "$_os" = "WINNT"; then
+       ./oowintool --msvc-copy-dlls ../external/msvcp
+fi
+
+dnl ===================================================================
 dnl Windows builds with .NET need msvcp70.dll/msvcr70.dll in external/msvcp70/
 dnl ===================================================================
 if test "$_os" = "WINNT" -a "$COMEX" = "8"; then
