--- config_office/configure.in
+++ config_office/configure.in
@@ -122,6 +122,12 @@
 AC_ARG_ENABLE(rpath,
 [  --disable-rpath:        Disable the use of relative paths in shared libraries
 ],,)
+AC_ARG_WITH(myspell-dicts,
+[  --without-myspell-dicts   Removes myspell dictionaries from openoffice.org
+                         installation set, for people building for specific
+                         distributions where the myspell dictionaries are
+                         installed from other sources
+],,)
 AC_ARG_WITH(system-libs,
 [  --with-system-libs      Use libs already on system -- enables all
 			   --with-system-* flags except mozilla, python and
@@ -660,6 +666,18 @@
 fi
 AC_SUBST(ENABLE_RPATH)
 
+dnl Test whether to include MySpell dictionaries
+dnl ===================================================================
+AC_MSG_CHECKING([whether to include MySpell dictionaries])
+if test -z "$with_myspell_dicts"; then
+  AC_MSG_RESULT([yes])
+  WITH_MYSPELL_DICTS=YES
+else
+  AC_MSG_RESULT([no])
+  WITH_MYSPELL_DICTS=NO
+fi
+AC_SUBST(WITH_MYSPELL_DICTS)
+
 dnl ===================================================================
 dnl WINNT uses either 4nt or tcsh, all other O/S use tcsh.
 dnl The following sets the with_use_shell variable.


--- config_office/set_soenv.in
+++ config_office/set_soenv.in
@@ -1707,6 +1707,7 @@
 ToFile( "WITH_GPC",          "@WITH_GPC@",         "e" );
 ToFile( "WITH_LIBART",       "@WITH_LIBART@",      "e" );
 ToFile( "LIBART_CFLAGS",     "@LIBART_CFLAGS@",    "e" );
+ToFile( "WITH_MYSPELL_DICTS","@WITH_MYSPELL_DICTS@","e");
 ToFile( "LIBART_LIBS",       "@LIBART_LIBS@",      "e" );
 ToFile( "WITH_LIBSN",        "@WITH_LIBSN@",       "e" );
 ToFile( "LIBSN_CFLAGS",      "@LIBSN_CFLAGS@",     "e" );
