--- solenv/bin/modules/installer/scriptitems.pm.ark	2005-03-08 04:17:01.000000000 +0100
+++ solenv/bin/modules/installer/scriptitems.pm	2005-03-08 04:17:25.000000000 +0100
@@ -946,7 +946,8 @@
 	if ( $error_occured )
 	{
 		for ( my $i = 0; $i <= $#missingfiles; $i++ ) { print "$missingfiles[$i]"; }
-		installer::exiter::exit_program("ERROR: Missing files", "remove_Files_Without_Sourcedirectory");
+		# _EVIL_ workaround for missing dictionaries
+		# installer::exiter::exit_program("ERROR: Missing files", "remove_Files_Without_Sourcedirectory");
 	}
 
 	return \@newfilesarray;
