--- officecfg/registry/data/org/openoffice/Office/Writer.xcu~	2007-03-28 11:21:09.211200714 +0200
+++ officecfg/registry/data/org/openoffice/Office/Writer.xcu	2007-03-28 11:27:23.606848387 +0200
@@ -742,21 +742,21 @@
     </prop>
   </node>
  </node>
- <node oor:name="DefaultFont" install:module="defaultfontarial">
+ <node oor:name="DefaultFont">
      <prop oor:name="Standard">
-       <value >Arial</value>
+       <value>Times</value>
      </prop>
      <prop oor:name="Heading">
-       <value >Arial</value>
+       <value>Helvetica</value>
      </prop>
      <prop oor:name="List">
-       <value >Arial</value>
+       <value>Times</value>
      </prop>
      <prop oor:name="Caption">
-       <value >Arial</value>
+       <value>Times</value>
      </prop>
      <prop oor:name="Index">
-       <value >Arial</value>
+       <value>Times</value>
      </prop>
  </node>
 </oor:component-data>
