--- config_office/configure.in.old	2006-10-09 20:39:18.000000000 +0200
+++ config_office/configure.in	2006-10-10 18:14:31.000000000 +0200
@@ -3317,10 +3317,25 @@
           AC_CHECK_FILE($XALAN_JAR, [],
                [AC_MSG_ERROR(xalan.jar replacement not found.)], [])
         fi
-        if test -n "$SERIALIZER_JAR"; then
+        if test -z "$SERIALIZER_JAR"; then
+          # FIXME: add check for xalan version to know when serializer.jar
+          #        is required 
+          AC_CHECK_FILE(/usr/share/java/xalan-j2-serializer.jar, 
+            [ SERIALIZER_JAR=/usr/share/java/xalan-j2-serializer.jar ],
+            [
+              AC_CHECK_FILE(/usr/share/java/xalan2-serializer.jar, 
+                [ SERIALIZER_JAR=/usr/share/java/xalan2-serializer.jar ],
+                [ AC_CHECK_FILE(/usr/share/java/serializer.jar,
+                    [ SERIALIZER_JAR=/usr/share/java/serializer.jar ],
+                    [], []
+                  )
+                ]
+              )
+            ]
+          )
+	else
           AC_CHECK_FILE($SERIALIZER_JAR, [],
                [AC_MSG_ERROR(serializer.jar not found.)], [])
-          AC_SUBST(SERIALIZER_JAR)
         fi
 else
     AC_MSG_RESULT([internal])
@@ -3329,6 +3344,7 @@
 fi
 AC_SUBST(SYSTEM_XALAN)
 AC_SUBST(XALAN_JAR)
+AC_SUBST(SERIALIZER_JAR)
 
 if test -n "$NEED_XALAN"; then
    BUILD_TYPE="$BUILD_TYPE XALAN"
