--- config_office/configure.in.orig	2006-05-19 12:11:27.000000000 +0200
+++ config_office/configure.in	2006-05-19 13:27:44.000000000 +0200
@@ -3603,10 +3603,10 @@
    AC_CHECK_HEADERS(X11/Xaw/Label.h,[],[AC_MSG_ERROR([Xaw include headers not found])])
 
    if test "x$x_includes" = "x"; then
-     x_includes="/usr/include"
+     x_includes="default_x_includes"
    fi
    if test "x$x_libraries" = "x"; then
-     x_libraries="/usr/lib"
+     x_libraries="default_x_libraries"
    fi
    dnl The variables $x_libraries and $x_includes are set.
    if test -z "$x_libraries"; then
