diff -ur /home/naren/work/builds/ooo-build-1.9.125.1.pristine/build/src680-m125/xmlsecurity/source/xmlsec/nss/nsspki1t.h /home/naren/work/builds/ooo-build-1.9.125.1/build/src680-m125/xmlsecurity/source/xmlsec/nss/nsspki1t.h
--- xmlsecurity/source/xmlsec/nss/nsspki1t.h	2005-09-27 14:32:03.000000000 -0700
+++ xmlsecurity/source/xmlsec/nss/nsspki1t.h	2005-10-11 09:43:34.740457440 -0700
@@ -57,7 +57,29 @@
  * This is the basic OID that crops up everywhere.
  */
 
-struct NSSOIDStr;
+// This was moved from pki1t.h to here to satisfy GCC 4.0 arrary checking rules
+// Naren Devaiah - 11Oct2005
+/*
+ * NSSOID
+ *
+ * This structure is used to hold our internal table of built-in OID
+ * data.  The fields are as follows:
+ *
+ *  NSSItem     data -- this is the actual DER-encoded multinumber oid
+ *  const char *expl -- this explains the derivation, and is checked
+ *                      in a unit test.  While the field always exists,
+ *                      it is only populated or used in debug builds.
+ *
+ */
+
+struct NSSOIDStr {
+#ifdef DEBUG
+  const NSSUTF8 *tag;
+  const NSSUTF8 *expl;
+#endif /* DEBUG */
+  NSSItem data;
+};
+
 typedef struct NSSOIDStr NSSOID;
 
 /*
diff -ur /home/naren/work/builds/ooo-build-1.9.125.1.pristine/build/src680-m125/xmlsecurity/source/xmlsec/nss/pki1t.h /home/naren/work/builds/ooo-build-1.9.125.1/build/src680-m125/xmlsecurity/source/xmlsec/nss/pki1t.h
--- xmlsecurity/source/xmlsec/nss/pki1t.h	2005-09-27 14:32:03.000000000 -0700
+++ xmlsecurity/source/xmlsec/nss/pki1t.h	2005-10-11 09:43:37.180086560 -0700
@@ -56,27 +56,6 @@
 PR_BEGIN_EXTERN_C
 
 /*
- * NSSOID
- *
- * This structure is used to hold our internal table of built-in OID
- * data.  The fields are as follows:
- *
- *  NSSItem     data -- this is the actual DER-encoded multinumber oid
- *  const char *expl -- this explains the derivation, and is checked
- *                      in a unit test.  While the field always exists,
- *                      it is only populated or used in debug builds.
- *
- */
-
-struct NSSOIDStr {
-#ifdef DEBUG
-  const NSSUTF8 *tag;
-  const NSSUTF8 *expl;
-#endif /* DEBUG */
-  NSSItem data;
-};
-
-/*
  * nssAttributeTypeAliasTable
  *
  * Attribute types are passed around as oids (at least in the X.500
