Index: xmlhelp/source/com/sun/star/help/makefile.mk
===================================================================
--- xmlhelp/source/com/sun/star/help/makefile.mk.orig	2006-01-11 13:16:18.000000000 +0100
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2006-01-16 13:41:11.000000000 +0100
@@ -80,18 +77,25 @@ JAVACLASSFILES += \
 
 .IF "$(JDK)"=="gcj"
 
-.IF "$(SYSTEM_DB)"=="YES"
-$(BIN)$/HelpLinker: $(LB)$/libdb.jar.a
-.ELSE
 $(BIN)$/HelpLinker:
-.ENDIF
 	$(JAVACOMPILER) --main=com.sun.star.help.HelpLinker -o $(BIN)$/HelpLinker \
 		-O2 -Wl,-Bsymbolic -findirect-dispatch -fjni -fno-assert -Wno-deprecated \
 		$(JAVACPS) $(CLASSPATH) $(JAVAFILES) $(SOLARLIB) -L$(LB) \
-		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar
+		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar -lxt-xmlsearch.jar
 
+.IF "$(SYSTEM_DB)"=="YES"
+$(BIN)$/HelpLinker: $(LB)$/libdb.jar.a
+
 $(LB)$/libdb.jar.a:
 	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(DB_JAR) -o $(LB)$/libdb.jar.a
+.ENDIF
+
+.IF "$(SYSTEM_XT)"=="YES"
+$(BIN)$/HelpLinker: $(LB)$/libxt.jar.a
+
+$(LB)$/libxt.jar.a:
+	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(XT_JAR) -o $(LB)$/libxt.jar.a
+.ENDIF
 
 .ELSE
 JARCLASSDIRS	= com
