--- xmlhelp/source/com/sun/star/help/makefile.mk.ark	2006-02-26 23:12:20.000000000 +0100
+++ xmlhelp/source/com/sun/star/help/makefile.mk	2006-02-26 23:14:01.000000000 +0100
@@ -102,7 +102,7 @@
 		-Djavax.xml.validation.SchemaFactory=org.apache.xerces.jaxp.validation.XMLSchemaFactory \
 		-Dorg.w3c.dom.DOMImplementationSourceList=org.apache.xerces.dom.DOMXSImplementationSourceImpl \
 		-Dorg.xml.sax.driver=org.apache.xerces.parsers.SAXParser \
-		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar -lxercesImpl.jar
+		-ldb.jar -lxmlsearch.jar -lxt.jar -lunoil.jar -ljurt.jar -lridl.jar -lxercesImpl.jar -lxt-xmlsearch.jar
 
 .IF "$(SYSTEM_DB)"=="YES"
 $(BIN)$/HelpLinker: $(LB)$/libdb.jar.a $(LB)$/libxercesImpl.jar.a
@@ -116,6 +116,13 @@
 $(LB)$/libxercesImpl.jar.a:
 	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(XERCES_JAR) -o $(LB)$/libxercesImpl.jar.a
 
+.IF "$(SYSTEM_XT)"=="YES"
+$(BIN)$/HelpLinker: $(LB)$/libxt.jar.a
+
+$(LB)$/libxt.jar.a:
+	$(JAVACOMPILER) -c -O2 -findirect-dispatch -fjni $(XT_JAR) -o $(LB)$/libxt.jar.a
+.ENDIF
+
 .ELSE
 JARCLASSDIRS	= com
 JARTARGET		= $(PRJNAME).jar
