Index: svx/inc/galctrl.hxx
===================================================================
RCS file: /cvs/graphics/svx/inc/galctrl.hxx,v
retrieving revision 1.6
retrieving revision 1.6.432.1
diff -u -p -u -p -r1.6 -r1.6.432.1
--- svx/inc/galctrl.hxx	12 Oct 2006 11:43:24 -0000	1.6
+++ svx/inc/galctrl.hxx	10 Aug 2007 12:01:32 -0000	1.6.432.1
@@ -74,7 +74,6 @@ private:
 	
     GraphicObject		aGraphicObj;
 	Rectangle			aPreviewRect;
-	Sound				aSound;
     GalleryTheme*       mpTheme;
 						
 	SVX_DLLPRIVATE BOOL				ImplGetGraphicCenterRect( const Graphic& rGraphic, Rectangle& rResultRect ) const;
@@ -96,7 +95,6 @@ private:
 	SVX_DLLPRIVATE virtual void		StartDrag( sal_Int8 nAction, const Point& rPosPixel );
 
 						DECL_LINK( MenuSelectHdl, Menu* );
-						DECL_LINK( SoundEndHdl, Sound* );
 						
 public:					
 						
Index: svx/source/gallery2/galctrl.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/gallery2/galctrl.cxx,v
retrieving revision 1.25
retrieving revision 1.25.86.1
diff -u -p -u -p -r1.25 -r1.25.86.1
--- svx/source/gallery2/galctrl.cxx	27 Jun 2007 18:19:59 -0000	1.25
+++ svx/source/gallery2/galctrl.cxx	10 Aug 2007 12:01:32 -0000	1.25.86.1
@@ -67,7 +67,6 @@ GalleryPreview::GalleryPreview( GalleryB
 {
 	SetHelpId( HID_GALLERY_WINDOW );
     InitSettings();
-	aSound.SetNotifyHdl( LINK( this, GalleryPreview, SoundEndHdl ) );
 }
 
 // ------------------------------------------------------------------------
@@ -80,7 +79,6 @@ GalleryPreview::GalleryPreview( Window* 
 {
 	SetHelpId( HID_GALLERY_PREVIEW );
     InitSettings();
-    aSound.SetNotifyHdl( LINK( this, GalleryPreview, SoundEndHdl ) );
 }
 
 // ------------------------------------------------------------------------
@@ -281,12 +279,6 @@ void GalleryPreview::PreviewMedia( const
 
 // ------------------------------------------------------------------------
 
-IMPL_LINK( GalleryPreview, SoundEndHdl, Sound*, EMPTYARG )
-{
-	aSound.SetSoundName( String() );
-	return 0L;
-}
-
 // -------------------
 // - GalleryIconView -
 // -------------------
