Index: sc/source/ui/view/tabvwshg.cxx
===================================================================
RCS file: /cvs/sc/sc/source/ui/view/tabvwshg.cxx,v
retrieving revision 1.11
retrieving revision 1.11.236.1
diff -u -p -u -p -r1.11 -r1.11.236.1
--- sc/source/ui/view/tabvwshg.cxx	14 Nov 2006 16:00:31 -0000	1.11
+++ sc/source/ui/view/tabvwshg.cxx	30 Aug 2007 09:22:04 -0000	1.11.236.1
@@ -59,9 +59,7 @@ using namespace com::sun::star;
 #include "document.hxx"
 #include "drawview.hxx"
 #include "globstr.hrc"
-#ifndef _SV_SOUND_HXX
-#include <vcl/sound.hxx>
-#endif
+#include <avmedia/mediawindow.hxx>
 
 //------------------------------------------------------------------------
 
@@ -116,7 +114,7 @@ void ScTabViewShell::InsertURLButton( co
 	aAny <<= eButtonType;
 	xPropSet->setPropertyValue( rtl::OUString::createFromAscii( "ButtonType" ), aAny );
 
-	if ( Sound::IsSoundFile( rURL ) )
+        if ( ::avmedia::MediaWindow::isMediaURL( rURL ) )
 	{
 		// #105638# OJ
 		aAny <<= sal_True;
