Index: scp2/source/ooo/file_library_ooo.scp
===================================================================
RCS file: /cvs/installation/scp2/source/ooo/file_library_ooo.scp,v
retrieving revision 1.262
retrieving revision 1.255.18.3
diff -u -p -u -p -r1.262 -r1.255.18.3
--- scp2/source/ooo/file_library_ooo.scp	25 Oct 2007 16:05:14 -0000	1.262
+++ scp2/source/ooo/file_library_ooo.scp	30 Oct 2007 15:15:01 -0000	1.255.18.3
@@ -2496,7 +2496,7 @@ End
 #endif
 
 // AVMedia
-STD_LIB_FILE_PATCH( gid_File_Lib_AVMedia, avmedia )
+STD_UNO_LIB_FILE_PATCH( gid_File_Lib_AVMedia, avmedia )
 
 #if defined UNX
 #ifdef SOLAR_JAVA
