--- scp2/source/ooo/file_library_ooo.scp	27 Sep 2007 13:10:19 -0000	1.260
+++ scp2/source/ooo/file_library_ooo.scp	5 Oct 2007 05:54:00 -0000	1.222.58.10
@@ -1467,6 +1467,8 @@ End
 
 STD_UNO_LIB_FILE_PATCH(gid_File_Lib_Wpft,wpft)
 
+STD_UNO_LIB_FILE_PATCH(gid_File_Lib_Writerfilter,writerfilter)
+
 #if !defined(MACOSX) && ( !defined(SYSTEM_MOZILLA) && !defined(WITHOUT_MOZILLA) )
 
 File gid_File_Lib_Mozabdrv
@@ -1839,7 +1841,7 @@ File gid_File_Lib_Salhelper
   #endif
 End
 
-File gid_File_Lib_Sax
+File gid_File_Lib_Sax_Uno
     TXT_FILE_BODY;
     Styles = (PACKED,UNO_COMPONENT,DONT_OVERWRITE,PATCH);
     RegistryID = gid_Starregistry_Services_Rdb;
@@ -1851,6 +1853,22 @@ File gid_File_Lib_Sax
   #endif
 End
 
+File gid_File_Lib_Fastsax_Uno
+    TXT_FILE_BODY;
+    Styles = (PACKED,UNO_COMPONENT,DONT_OVERWRITE,PATCH);
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Dir = gid_Dir_Program;
+  #ifdef UNX
+    Name = STRING(CONCAT2(fastsax.uno,UNXSUFFIX));
+  #else
+    Name = "fastsax.uno.dll";
+  #endif
+End
+
+STD_LIB_FILE_PATCH( gid_File_Lib_Sax , sax)
+
+STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Oox , oox)
+
 STD_UNO_LIB_FILE_PATCH( gid_File_Lib_Sb , sb)
 
 //
--- scp2/source/ooo/module_hidden_ooo.scp	27 Sep 2007 13:10:32 -0000	1.74
+++ scp2/source/ooo/module_hidden_ooo.scp	5 Oct 2007 05:55:09 -0000	1.53.20.8
@@ -224,6 +224,7 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Lng,
 	gid_File_Lib_Lnth,
 	gid_File_Lib_Offacc,
+	gid_File_Lib_Oox,
 	gid_File_Lib_Bindet,
 	gid_File_Lib_Pcr,
 	gid_File_Lib_Pdffilter,
@@ -233,6 +234,7 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_Sf_Prothdlr,
 	gid_File_Lib_Psp,
 	gid_File_Lib_Res,
+	gid_File_Lib_Sax,
 	gid_File_Lib_Sb,
 	gid_File_Lib_Scn,
 	gid_File_Lib_Sd,
@@ -389,7 +391,8 @@ Module gid_Module_Root_Files_5
 	gid_File_Lib_Sal,
 	gid_File_Lib_Salhelper,
 	gid_File_Lib_Proxyfac,
-	gid_File_Lib_Sax,
+	gid_File_Lib_Sax_Uno,
+	gid_File_Lib_Fastsax_Uno,
 	gid_File_Lib_Slideshow,
 	gid_File_Lib_Stm,
 	gid_File_Lib_Recentfile,
