--- scp2/source/ooo/profileitem_ooo.scp.orig	2005-01-31 12:01:28.000000000 +0100
+++ scp2/source/ooo/profileitem_ooo.scp	2005-03-04 15:47:33.000000000 +0100
@@ -163,7 +163,7 @@
   #ifdef WNT
     Value = "$SYSUSERCONFIG/%ONEWORDPRODUCTNAME%PRODUCTVERSION";
   #else
-    Value = "$SYSUSERCONFIG/.%LCONEWORDPRODUCTNAME%USERDIRPRODUCTVERSION";
+    Value = "$SYSUSERCONFIG/.ooo-2.0-pre";
   #endif
     StandaloneValue = "$ORIGIN/..";
 End
