--- ./testtools/prj/build.lst.testtools~	2008-06-25 13:49:38.000000000 +0200
+++ ./testtools/prj/build.lst	2008-07-07 22:12:50.947846872 +0200
@@ -1,4 +1,4 @@
-tt testtools : cpputools io remotebridges stoc javaunohelper pyuno cli_ure offapi NULL
+tt testtools : cpputools io remotebridges stoc javaunohelper pyuno cli_ure offapi QADEVOOO:qadevOOo NULL
 tt testtools\inc nmake - all tt_inc NULL
 tt testtools\source\bridgetest nmake - all tt_bridgetest tt_bridgetest_idl tt_javaTestObjs tt_inc NULL
 tt testtools\source\bridgetest\cli nmake - w,vc7 tt_cli tt_bridgetest tt_bridgetest_idl tt_inc NULL
--- ./testtools/prj/tests.lst.testtools~	2008-07-07 22:11:28.989095131 +0200
+++ ./testtools/prj/tests.lst	2008-07-07 22:11:28.989095131 +0200
@@ -0,0 +1 @@
+tt   testtools\qa\cli                                   nmake - all tt_qa_cli NULL
