--- testtools/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ testtools/prj/tests.lst	2007-09-07 14:32:15.000000000 +0200
@@ -0,0 +1 @@
+tt   testtools\qa\cli                                   nmake - all tt_qa_cli NULL
--- testtools/prj/build.lst	2007-09-07 21:19:43.000000000 +0200
+++ testtools/prj/build.lst	2007-09-07 22:01:23.000000000 +0200
@@ -1,4 +1,4 @@
-tt testtools : cpputools io remotebridges stoc javaunohelper pyuno cli_ure NULL
+tt testtools : cpputools io remotebridges stoc javaunohelper pyuno cli_ure QADEVOOO:qadevOOo NULL
 tt testtools\inc nmake - all tt_inc NULL
 tt testtools\source\bridgetest nmake - all tt_bridgetest tt_bridgetest_idl tt_javaTestObjs tt_inc NULL
 tt testtools\source\bridgetest\cli nmake - w,vc7 tt_cli tt_bridgetest tt_bridgetest_idl tt_inc NULL
