--- xmlhelp/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ xmlhelp/prj/tests.lst	2007-09-07 12:52:56.000000000 +0200
@@ -0,0 +1 @@
+xh   xmlhelp\source\cxxhelp\test                        nmake - all xh_source_cxxhelp_test xh_util NULL
--- xmlhelp/source/cxxhelp/test/searchdemo.cxx	2006-09-17 03:20:27.000000000 +0200
+++ xmlhelp/source/cxxhelp/test/searchdemo.cxx	2007-09-07 12:51:05.000000000 +0200
@@ -35,7 +35,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_xmlhelp.hxx"
-#include <rtl/string.hxx>
+#include <rtl/ustring.hxx>
 #include <osl/file.hxx>
 #include <db/Block.hxx>
 #include <db/Schema.hxx>
