--- bridges/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ bridges/prj/tests.lst	2007-09-07 10:34:48.000000000 +0200
@@ -0,0 +1,9 @@
+#does not build: br   bridges\test\com\sun\star\lib\uno\bridges\java_remote nmake - all br_test_com_sun_star_lib_uno_bridges_java_remote NULL
+#does not build: br   bridges\test\com\sun\star\lib                      nmake - all br_test_com_sun_star_lib NULL
+br   bridges\test                                       nmake - all br_test NULL
+br   bridges\test\inter_libs_exc                        nmake - all br_test_inter_libs_exc NULL
+#does not build: br   bridges\test\java_uno\acquire                      nmake - all br_test_java_uno_acquire NULL
+#does not build: br   bridges\test\java_uno\any                          nmake - all br_test_java_uno_any NULL
+#does not build: br   bridges\test\java_uno\equals                       nmake - all br_test_java_uno_equals NULL
+#depends on javaunohelper: br   bridges\test\java_uno\nativethreadpool             nmake - all br_test_java_uno_nativethreadpool NULL
+#does not build: br   bridges\test\performance                           nmake - all br_test_performance NULL
--- bridges/test/java_uno/nativethreadpool/Relay.java	2005-09-08 01:00:52.000000000 +0200
+++ bridges/test/java_uno/nativethreadpool/Relay.java	2007-09-07 10:32:08.000000000 +0200
@@ -106,7 +106,7 @@ public final class Relay implements XRel
         }
     }
 
-    public int get() {
+    public long get() {
         return source.get();
     }
 
--- bridges/test/java_uno/nativethreadpool/testnativethreadpoolclient.cxx	2006-09-16 18:05:35.000000000 +0200
+++ bridges/test/java_uno/nativethreadpool/testnativethreadpoolclient.cxx	2007-09-07 10:28:34.000000000 +0200
@@ -82,7 +82,7 @@ private:
     virtual sal_Int32 SAL_CALL run(css::uno::Sequence< rtl::OUString > const &)
         throw (css::uno::RuntimeException);
 
-    virtual sal_Int32 SAL_CALL get() throw (css::uno::RuntimeException);
+    virtual sal_Int64 SAL_CALL get() throw (css::uno::RuntimeException);
 
     css::uno::Reference< css::uno::XComponentContext > context;
     osl::ThreadData data;
@@ -150,8 +150,8 @@ sal_Int32 Client::run(css::uno::Sequence
     return success ? 0 : 1;
 }
 
-sal_Int32 Client::get() throw (css::uno::RuntimeException) {
-    return reinterpret_cast< sal_Int32 >(data.getData());
+sal_Int64 Client::get() throw (css::uno::RuntimeException) {
+    return static_cast< sal_Int64 >(reinterpret_cast< sal_IntPtr >(data.getData()));
 }
 
 css::uno::Reference< css::uno::XInterface > SAL_CALL create(
--- bridges/test/java_uno/nativethreadpool/testnativethreadpoolserver.cxx	2006-09-16 18:05:51.000000000 +0200
+++ bridges/test/java_uno/nativethreadpool/testnativethreadpoolserver.cxx	2007-09-07 10:30:20.000000000 +0200
@@ -74,12 +74,12 @@ public:
 private:
     virtual ~Server() {}
 
-    virtual sal_Int32 SAL_CALL get() throw (css::uno::RuntimeException);
+    virtual sal_Int64 SAL_CALL get() throw (css::uno::RuntimeException);
 
     css::uno::Reference< css::uno::XComponentContext > context;
 };
 
-sal_Int32 Server::get() throw (css::uno::RuntimeException) {
+sal_Int64 Server::get() throw (css::uno::RuntimeException) {
     css::uno::Reference< css::lang::XMultiComponentFactory > factory(
         context->getServiceManager());
     if (!factory.is()) {
--- bridges/test/java_uno/nativethreadpool/types.idl	2005-09-08 01:02:09.000000000 +0200
+++ bridges/test/java_uno/nativethreadpool/types.idl	2007-09-07 10:27:57.000000000 +0200
@@ -38,7 +38,7 @@
 module test { module javauno { module nativethreadpool {
 
 interface XSource: com::sun::star::uno::XInterface {
-    long get();
+    hyper get();
 };
 
 interface XRelay: com::sun::star::uno::XInterface {
