--- extensions/prj/tests.lst	1970-01-01 01:00:00.000000000 +0100
+++ extensions/prj/tests.lst	2007-09-07 18:46:39.000000000 +0200
@@ -0,0 +1,16 @@
+ex   extensions\qa\complex\extensions                   nmake - all ex_qa_complex_extensions NULL
+ex   extensions\qa\integration\extensions               nmake - all ex_qa_integration_extensions NULL
+#does not build: ex   extensions\qa\unoapi                               nmake - all ex_qa_unoapi NULL
+
+# older tests:
+ex   extensions\source\update\feed\test                 nmake - all ex_source_update_feed_test NULL
+ex   extensions\test\ole\OleClient                      nmake - w   ex_test_ole_OleClient NULL
+ex   extensions\test\ole\OleConverterVar1               nmake - w   ex_test_ole_OleConverterVar1 NULL
+ex   extensions\test\ole\cpnt                           nmake - w   ex_test_ole_cpnt NULL
+ex   extensions\test\ole\cppToUno                       nmake - w   ex_test_ole_cppToUno NULL
+ex   extensions\test\ole\cpptest                        nmake - w   ex_test_ole_cpptest NULL
+ex   extensions\test\ole\unloading                      nmake - w   ex_test_ole_unloading NULL
+#does not build: ex   extensions\test\pgp                                nmake - all ex_test_pgp NULL
+#does not build: ex   extensions\test\sax                                nmake - all ex_test_sax NULL
+#does not build: ex   extensions\test\stm                                nmake - all ex_test_stm NULL
+#does not build: ex   extensions\workben                                 nmake - all ex_workben NULL
--- extensions/source/update/feed/test/updatefeedtest.cxx	2006-12-13 16:19:19.000000000 +0100
+++ extensions/source/update/feed/test/updatefeedtest.cxx	2007-09-07 13:34:06.000000000 +0200
@@ -65,8 +65,8 @@ namespace xml = ::com::sun::star::xml;
 
 SAL_IMPLEMENT_MAIN()
 {
-    (void) argv;
-	(void) argc;
+    //(void) argv;
+	//(void) argc;
 
     if( osl_getCommandArgCount() != 1 )
     {
