--- xmlhelp/prj/tests.lst.old	1970-01-01 00:00:00.000000000 +0000
+++ xmlhelp/prj/tests.lst	2009-04-06 16:42:15.000000000 +0000
@@ -0,0 +1 @@
+xh   xmlhelp\source\cxxhelp\test                        nmake - all xh_source_cxxhelp_test xh_util NULL
--- xmlhelp/source/cxxhelp/test/searchdemo.cxx.old	2009-04-02 11:01:40.000000000 +0000
+++ xmlhelp/source/cxxhelp/test/searchdemo.cxx	2009-04-06 16:42:15.000000000 +0000
@@ -30,7 +30,7 @@
 
 // MARKER(update_precomp.py): autogen include statement, do not remove
 #include "precompiled_xmlhelp.hxx"
-#include <rtl/string.hxx>
+#include <rtl/ustring.hxx>
 #include <osl/file.hxx>
 #include <db/Block.hxx>
 #include <db/BtreeDictParameters.hxx>
