Index: boost/prj/d.lst
===================================================================
RCS file: /cvs/external/boost/prj/d.lst,v
retrieving revision 1.9
diff -u -p -r1.9 d.lst
--- boost/prj/d.lst	2 Nov 2005 13:37:38 -0000	1.9
+++ boost/prj/d.lst	9 Jan 2006 14:53:14 -0000
@@ -244,3 +244,6 @@ mkdir: %_DEST%\inc%_EXT%\boost\tuple
 ..\%__SRC%\misc\build\boost_1_34_1\boost\spirit\utility\* %_DEST%\inc%_EXT%\boost\spirit\utility
 ..\%__SRC%\misc\build\boost_1_34_1\boost\spirit\utility\impl\* %_DEST%\inc%_EXT%\boost\spirit\utility\impl
 ..\%__SRC%\misc\build\boost_1_34_1\boost\spirit\utility\impl\chset\* %_DEST%\inc%_EXT%\boost\spirit\utility\impl\chset
+
+# extra bits that scsolver wants:
+..\%__SRC%\misc\build\boost-1.30.2\boost\shared_ptr.hpp %_DEST%\inc%_EXT%\boost\shared_ptr.hpp
