Index: sc/prj/build.lst
===================================================================
RCS file: /cvs/sc/sc/prj/build.lst,v
retrieving revision 1.10
diff -u -p -u -r1.10 build.lst
--- sc/prj/build.lst	30 Aug 2005 09:35:13 -0000	1.10
+++ sc/prj/build.lst	9 Jan 2006 15:23:50 -0000
@@ -1,4 +1,4 @@
-sc	sc	:	svx stoc uui BOOST:boost sch NULL
+sc	sc	:	svx scsolver stoc uui BOOST:boost sch NULL
 sc	sc										usr1	-	all	sc_mkout NULL
 sc	sc\inc									get		-	all	sc_inc NULL
 sc	sc\prj									get		-	all	sc_prj NULL
