--- /dev/null	
+++ scsolver/README	
@@ -0,0 +1,60 @@
+== HOW TO COMPILE SCSOLVER UNO PACKAGE ==
+
+=== Instructions ===
+
+To compile the separate UNO package named scsolver.uno.zip that can be 
+installed on a version of OO.o that doesn't have the Solver pre-installed, 
+follow these steps.  
+
+1. First, you need to have OO.o itself installed somewhere on your system, and
+   the version of OO.o SDK compatible with your OO.o installation.
+
+2. Download, unpack and build the lp_solve library in a separate directory.  
+   You can download the lp_solve 5.5.0.10 package from our mirror.  To do it, 
+   run these commands in this order:
+
+   wget http://download.go-oo.org/SRC680/lp_solve_5.5.0.10_source.tar.gz
+   tar xvf lp_solve_5.5.0.10_source.tar.gz
+   cd lp_solve_5.5/lpsolve55
+   sh ccc
+
+3. Now, change directory into workben/addon_pkg, and create and run 
+   autogen.sh.
+
+   cd workben/addon_pkg
+   ./autogen.sh \
+       --with-ooo=/path/to/ooo/install \
+       --with-ooo-sdk=/path/to/ooo/sdk \
+       --with-lpsolve=/path/to/lpsolve
+
+   NOTE 1) Use absolute path to specify the location of the OO.o installation,
+   OO.o SDK, and the base lpsolve path.
+
+4. If all goes well, simply run make to build scsolver.uno.zip.
+
+   make
+
+5. At this point, you can launch OO.o Calc and manually install this package
+   from GUI, or you can do this from the command line:
+
+   make install
+
+   this will execute the 'unopkg add' command to register scsolver.uno.zip.
+
+=== FAQ ===
+
+Q. I'm getting the following error:
+
+   ./autogen.sh: line 4: aclocal: command not found
+
+   What am I missing?
+
+A. You need to install 'autoconf' (and 'automake' too?) in order to successfully
+   run the autogen.sh script.
+
+Q. I'm getting boost/shared_ptr.hpp: no such file or directory error.  What 
+   should I do?
+
+A. You need to install boost-devel package to get the boost header files 
+   installed, because scsolver makes use of boost shared_ptr and ublas matrix 
+   libraries.
\ No newline at end of file
