diff --git lpsolve/lp_solve_5.5.patch lpsolve/lp_solve_5.5.patch
index fc446a4..42190cd 100644
--- lpsolve/lp_solve_5.5.patch
+++ lpsolve/lp_solve_5.5.patch
@@ -7,6 +7,15 @@
  
  def=
  so=
+@@ -12,7 +12,7 @@ fi
+ 
+ opts='-O3'
+ 
+-$c -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
++$c -fpic -s -c -I.. -I../shared -I../bfp -I../bfp/bfp_LUSOL -I../bfp/bfp_LUSOL/LUSOL -I../colamd $opts $def -DYY_NEVER_INTERACTIVE -DPARSER_LP -DINVERSE_ACTIVE=INVERSE_LUSOL -DRoleIsExternalInvEngine $src
+ ar rv liblpsolve55.a `echo $src|sed s/[.]c/.o/g|sed 's/[^ ]*\///g'`
+ ranlib liblpsolve55.a
+ 
 --- misc/lp_solve_5.5/lpsolve55/ccc.os2	2008-11-26 12:49:36.000000000 +0100
 +++ misc/build/lp_solve_5.5/lpsolve55/ccc.os2	2008-09-25 17:27:48.000000000 +0200
 @@ -1 +1,13 @@
diff --git lpsolve/makefile.mk lpsolve/makefile.mk
index b36abbe..594e5a5 100644
--- lpsolve/makefile.mk
+++ lpsolve/makefile.mk
@@ -68,14 +68,18 @@ OUT2LIB=$(BUILD_DIR)$/liblpsolve55.lib
 .ELSE
 .IF "$(OS)"=="MACOSX"
 BUILD_ACTION=sh ccc.osx
-OUT2LIB=$(BUILD_DIR)$/liblpsolve55.dylib
+OUT2LIB= \
+        $(BUILD_DIR)$/liblpsolve55.dylib \
+        $(BUILD_DIR)$/liblpsolve55.a
 .ELSE
 .IF "$(COMNAME)"=="sunpro5"
 BUILD_ACTION=sh ccc.solaris
 .ELSE
 BUILD_ACTION=sh ccc
 .ENDIF
-OUT2LIB=$(BUILD_DIR)$/liblpsolve55.so
+OUT2LIB= \
+       $(BUILD_DIR)$/liblpsolve55.so \
+       $(BUILD_DIR)$/liblpsolve55.a
 .ENDIF
 .ENDIF
 
diff --git lpsolve/prj/d.lst lpsolve/prj/d.lst
index 64b8c0c..bd15d44 100644
--- lpsolve/prj/d.lst
+++ lpsolve/prj/d.lst
@@ -2,6 +2,7 @@ mkdir: %_DEST%\inc%_EXT%\lpsolve
 
 ..\%__SRC%\inc\lp_*.h %_DEST%\inc%_EXT%\lpsolve\*
 ..\%__SRC%\lib\liblpsolve55.so %_DEST%\lib%_EXT%\liblpsolve55.so
+..\%__SRC%\lib\liblpsolve55.a %_DEST%\lib%_EXT%\liblpsolve55.a
 ..\%__SRC%\lib\liblpsolve55.dylib %_DEST%\lib%_EXT%\liblpsolve55.dylib
 ..\%__SRC%\lib\liblpsolve55.lib %_DEST%\lib%_EXT%\liblpsolve55.lib
 ..\%__SRC%\lib\lpsolve55.lib %_DEST%\lib%_EXT%\lpsolve55.lib
