--- scp2/source/ooo/file_library_ooo.scp	2004-11-15 11:46:49.141039128 +0100
+++ scp2/source/ooo/file_library_ooo.scp	2004-11-15 11:49:31.741320128 +0100
@@ -625,7 +625,6 @@ End
 
 #endif
 
-#if 0
 #ifdef UNX
 #ifdef ENABLE_GTK
 #ifndef MACOSX
@@ -638,6 +637,23 @@ File gid_File_Lib_Fps_Gnome
 End
 #endif
 #endif
+
+#ifdef ENABLE_KDE
+#ifndef MACOSX
+File gid_File_Lib_Fps_Kde
+    TXT_FILE_BODY;
+    Styles = (PACKED,UNO_COMPONENT);
+    RegistryID = gid_Starregistry_Services_Rdb;
+    Dir = gid_Dir_Program;
+    Name = STRING(CONCAT2(libfps_kde,UNXSUFFIX));
+End
+File gid_File_Bin_KdeFilePicker
+        BIN_FILE_BODY;
+        Styles = (PACKED);
+        Dir = gid_Dir_Program;
+        Name = "kdefilepicker";
+End
+#endif
 #endif
 #endif
 
