--- svx/source/tbxctrls/tbcontrl.cxx	2004-11-16 11:54:00.752627704 +0100
+++ svx/source/tbxctrls/tbcontrl.cxx	2004-11-16 11:53:57.161173688 +0100
@@ -1998,14 +1998,14 @@ void SvxTbxButtonColorUpdater_Impl::Upda
 	    			if( maBmpSize.Width() <= 16 )
 		    			maUpdRect = Rectangle( Point( 0,12 ), Size( maBmpSize.Width(), 4 ) );
 			    	else
-				    	maUpdRect = Rectangle( Point( 1,19 ), Size( 24,6 ) );
+				    	maUpdRect = Rectangle( Point( 1, maBmpSize.Height() - 7 ), Size( maBmpSize.Width() - 2 ,6 ) );
                 }
                 else
                 {
                     if( maBmpSize.Width() <= 16 )
                         maUpdRect = Rectangle( Point( 7, 7 ), Size( 8, 8 ) );
                     else
-                        maUpdRect = Rectangle( Point( 14, 14 ), Size( 11, 11 ) );
+                        maUpdRect = Rectangle( Point( maBmpSize.Width() - 12, maBmpSize.Height() - 12 ), Size( 11, 11 ) );
                 }
 
 				pBmpAcc->DrawRect( maUpdRect );
