--- helpcontent2/util/sdatabase/makefile.mk	2005-02-04 13:45:01.621269920 +0100
+++ helpcontent2/util/sdatabase/makefile.mk	2005-02-04 13:45:18.349379434 +0100
@@ -26,6 +26,7 @@ TARGET  = util
 .INCLUDE : settings.mk
 .INCLUDE : $(PRJ)$/settings.pmk
 
+.IF "$(SOLAR_JAVA)"!=""
 common_build_zip:=
 zip1generatedlangs=TRUE
 zip1langdirs=$(aux_alllangiso)
@@ -33,6 +34,7 @@ ZIP1TARGET=xhp_sdatabase
 ZIP1FLAGS= -u -r
 ZIP1DIR=$(MISC)$/$(LANGDIR)
 ZIP1LIST=$(LANGDIR)$/text$/shared$/explorer$/database$/main.xhp
+.ENDIF
 
 LINKNAME=sdatabase
 LINKADDEDFILES= \
